--- core/xmake.sh
+++ core/xmake.sh
@@ -14,7 +14,7 @@
 
 # disable some compiler errors
 if is_plat "macosx"; then
-    add_cxflags "-Wno-error=deprecated-declarations" "-fno-strict-aliasing" "-Wno-error=nullability-completeness" "-Wno-error=parentheses-equality"
+    add_cxflags "-Wno-error=deprecated-declarations" "-fno-strict-aliasing"
 fi
 
 # add build modes

--- core/xmake.lua
+++ core/xmake.lua
@@ -23,7 +23,7 @@
 end
 
 -- disable some compiler errors
-add_cxflags("-Wno-error=deprecated-declarations", "-fno-strict-aliasing", "-Wno-error=nullability-completeness", "-Wno-error=parentheses-equality")
+add_cxflags("-Wno-error=deprecated-declarations", "-fno-strict-aliasing")
 
 -- add definitions
 add_defines("_GNU_SOURCE=1", "_FILE_OFFSET_BITS=64", "_LARGEFILE_SOURCE")

--- core/src/sv/sv/xmake.lua
+++ core/src/sv/sv/xmake.lua
@@ -3,7 +3,7 @@
 
 set_warnings("all", "error")
 
-if is_plat("macosx") or is_plat("iphoneos") then
+if is_plat("iphoneos") then
     add_cflags("-Wno-nullability-completeness")
     add_cxxflags("-Wno-nullability-completeness")
 end
