--- infix.asd.orig
+++ infix.asd
@@ -16,4 +16,4 @@
 	       (:static-file "COPYING")))
 
 (defmethod source-file-type ((f cl-source-file) (s (eql (find-system 'infix))))
-  "cl")
+  "lisp")
--- infix.system.orig
+++ infix.system
@@ -2,7 +2,7 @@
 
 (mk:defsystem infix
   :source-pathname "cl-library:infix;"
-  :source-extension "cl"
+  :source-extension "lisp"
   :binary-pathname nil
   :binary-extension nil
   :components ((:file "infix")))
