--- Makefile.orig	2012-11-13 23:34:32.000000000 -0600
+++ Makefile	2012-11-13 23:47:27.000000000 -0600
@@ -535,8 +535,8 @@
 all: mplayer$(EXESUF) locales
 
 %.1: %.rst
-	@which rst2man >/dev/null 2>&1 || (printf "\n\trst2man not found. You need the Docutils system to generate the manpages (preferably version >= 0.9; 0.8 and older are known to corrupt the arguments shown in some option descriptions). Alternatively you can use 'install-no-man' rule.\n\n" && exit 1)
-	rst2man $< $@
+	@which rst2man-2.7.py >/dev/null 2>&1 || (printf "\n\trst2man-2.7.py not found. You need the Docutils system to generate the manpages (preferably version >= 0.9; 0.8 and older are known to corrupt the arguments shown in some option descriptions). Alternatively you can use 'install-no-man' rule.\n\n" && exit 1)
+	rst2man-2.7.py $< $@
 
 %.o: %.S
 	$(CC) $(DEPFLAGS) $(CFLAGS) -c -o $@ $<
