--- setup.py.orig	2005-06-22 03:31:11.000000000 +1000
+++ setup.py	2008-03-23 22:06:59.000000000 +1100
@@ -4,8 +4,8 @@
 plist = dict(
     CFBundleName = "ZopeEditManager",
     CFBundleIdentifier = "com.urbanape.zopeeditmanager",
-    CFBundleShortVersionString = "Version 0.9.6",
-    CFBundleGetInfoString = ("ZopeEditManager version 0.9.6, "
+    CFBundleShortVersionString = "Version @VERSION@",
+    CFBundleGetInfoString = ("ZopeEditManager version @VERSION@, "
                              "Copyright 2003-2005 Zope Corporation"),
     NSHumanReadableCopyright = "Copyright 2003-2005 Zope Corporation",
     CFBundleIconFile = "ZEM.icns",
@@ -16,6 +16,8 @@
     NSMainNibFile = "MainMenu",
     NSPrincipalClass = "NSApplication",
     CFBundleDevelopmentRegion = "English",
+    CFBundleHelpBookFolder = "ZopeEditManagerHelp",
+    CFBundleHelpBookName = "ZopeEditManager Help",
     CFBundleDocumentTypes = [
         dict(CFBundleTypeExtensions = ["zem"],
              CFBundleTypeOSTypes = ["ZEMD"],
