Index: src/H5Zbzip2.c
===================================================================
--- src/H5Zbzip2.c (revision 2381)
+++ src/H5Zbzip2.c (revision 4143)
@@ -23,5 +23,11 @@
   char *libver, *versionp, *datep, *sep;
 
-#if H5_VERS_MAJOR == 1 && H5_VERS_MINOR < 7
+/* The conditional below is somewhat messy, but it is necessary because
+  the THG team has decided to fix an API inconsistency in the definition
+  of the H5Z_class_t structure in version 1.8.3 */
+#if (H5_VERS_MAJOR == 1 && H5_VERS_MINOR < 7) || \
+    (H5_USE_16_API && (H5_VERS_MAJOR > 1 || \
+      (H5_VERS_MAJOR == 1 && (H5_VERS_MINOR > 8 || \
+        (H5_VERS_MINOR == 8 && H5_VERS_RELEASE >= 3)))))
    /* 1.6.x */
   H5Z_class_t filter_class = {
@@ -33,5 +39,5 @@
   };
 #else
-   /* 1.7.x */
+   /* 1.8.x where x < 3 */
   H5Z_class_t filter_class = {
     H5Z_CLASS_T_VERS,             /* H5Z_class_t version */
@@ -43,5 +49,5 @@
     (H5Z_func_t)(bzip2_deflate)   /* filter_func */
 };
-#endif /* if H5_VERSION < "1.7" */
+#endif
 
   /* Register the filter class for the bzip2 compressor. */
Index: src/H5Zlzo.c
===================================================================
--- src/H5Zlzo.c (revision 3695)
+++ src/H5Zlzo.c (revision 4143)
@@ -43,5 +43,11 @@
 #ifdef HAVE_LZO_LIB
 
-#if H5_VERS_MAJOR == 1 && H5_VERS_MINOR < 7
+/* The conditional below is somewhat messy, but it is necessary because
+  the THG team has decided to fix an API inconsistency in the definition
+  of the H5Z_class_t structure in version 1.8.3 */
+#if (H5_VERS_MAJOR == 1 && H5_VERS_MINOR < 7) || \
+    (H5_USE_16_API && (H5_VERS_MAJOR > 1 || \
+      (H5_VERS_MAJOR == 1 && (H5_VERS_MINOR > 8 || \
+        (H5_VERS_MINOR == 8 && H5_VERS_RELEASE >= 3)))))
    /* 1.6.x */
   H5Z_class_t filter_class = {
@@ -53,5 +59,5 @@
   };
 #else
-   /* 1.7.x */
+   /* 1.8.x where x < 3 */
   H5Z_class_t filter_class = {
     H5Z_CLASS_T_VERS,             /* H5Z_class_t version */
@@ -63,5 +69,5 @@
     (H5Z_func_t)(lzo_deflate)     /* filter_func */
   };
-#endif /* if H5_VERSION < "1.7" */
+#endif
 
   /* Init the LZO library */
