--- configure.ac~	2026-04-13 17:22:31
+++ configure.ac	2026-04-13 17:55:20
@@ -19,14 +19,15 @@
 AC_PROG_CC
 AC_PROG_CC_C99
 AC_PROG_CXX
+AX_CXX_COMPILE_STDCXX([17], [ext])
 AC_PROG_RANLIB
 
 dnl Find C++ compiler
-AC_CHECK_PROG(HAVE_CXX, $CXX, yes, no)
-if test "$HAVE_CXX" = "no"
-then
-  AC_MSG_ERROR([Could not find a c++ compiler]);
-fi
+dnl AC_CHECK_PROG(HAVE_CXX, $CXX, yes, no)
+dnl if test "$HAVE_CXX" = "no"
+dnl then
+dnl   AC_MSG_ERROR([Could not find a c++ compiler]);
+dnl fi
 
 # Checks for libraries.
 PKG_CHECK_MODULES([GLIB], [glib-2.0])
