--- src/plugins/cppeditor/cppquickfixes.cpp.orig	2024-11-15 19:37:42
+++ src/plugins/cppeditor/cppquickfixes.cpp	2024-11-15 22:04:19
@@ -3716,6 +3716,7 @@
         GenerateProperty = 1 << 5,
         GenerateConstantProperty = 1 << 6,
         HaveExistingQProperty = 1 << 7,
+        Invalid = -1,
     };
 
     GenerateGetterSetterOp(const CppQuickFixInterface &interface,
@@ -4398,7 +4399,7 @@
     };
     using Flag = GenerateGetterSetterOp::GenerateFlag;
     constexpr static Flag ColumnFlag[] = {
-        static_cast<Flag>(-1),
+        Flag::Invalid,
         Flag::GenerateGetter,
         Flag::GenerateSetter,
         Flag::GenerateSignal,
