--- src/http/ngx_http_upstream.c.orig	2017-03-10 18:45:20.000000000 -0600
+++ src/http/ngx_http_upstream.c	2017-03-10 18:47:08.000000000 -0600
@@ -164,8 +164,6 @@
     ngx_http_variable_value_t *v, uintptr_t data);
 static ngx_int_t ngx_http_upstream_response_length_variable(
     ngx_http_request_t *r, ngx_http_variable_value_t *v, uintptr_t data);
-static ngx_int_t ngx_http_upstream_header_variable(ngx_http_request_t *r,
-    ngx_http_variable_value_t *v, uintptr_t data);
 static ngx_int_t ngx_http_upstream_cookie_variable(ngx_http_request_t *r,
     ngx_http_variable_value_t *v, uintptr_t data);
 
@@ -5503,7 +5501,7 @@
 }
 
 
-static ngx_int_t
+ngx_int_t
 ngx_http_upstream_header_variable(ngx_http_request_t *r,
     ngx_http_variable_value_t *v, uintptr_t data)
 {
