--- auto/install	2006-08-30 10:39:17.000000000 +0000
+++ auto/install.new	2007-12-15 02:29:00.000000000 +0000
@@ -7,7 +7,7 @@
     cat << END                                                >> $NGX_MAKEFILE
 
 install_perl_modules:
-	cd $NGX_OBJS/src/http/modules/perl && make install
+	cd $NGX_OBJS/src/http/modules/perl && make pure_install
 END
 
     NGX_INSTALL_PERL_MODULES=install_perl_modules
@@ -19,42 +19,44 @@
 
 install:	$NGX_OBJS${ngx_dirsep}nginx${ngx_binext}	\
 		$NGX_INSTALL_PERL_MODULES
-	test -d '$NGX_PREFIX' || mkdir -p '$NGX_PREFIX'
+	test -d '__DESTROOT__$NGX_PREFIX' || mkdir -p '__DESTROOT__$NGX_PREFIX'
 
-	test -d '`dirname "$NGX_SBIN_PATH"`' \
-		|| mkdir -p '`dirname "$NGX_SBIN_PATH"`'
-	test ! -f '$NGX_SBIN_PATH' || mv '$NGX_SBIN_PATH' '$NGX_SBIN_PATH.old'
-	cp $NGX_OBJS/nginx '$NGX_SBIN_PATH'
+	test -d '`dirname "__DESTROOT__$NGX_SBIN_PATH"`' \
+		|| mkdir -p '`dirname "__DESTROOT__$NGX_SBIN_PATH"`'
+	test ! -f '__DESTROOT__$NGX_SBIN_PATH' || mv '__DESTROOT__$NGX_SBIN_PATH' '__DESTROOT__$NGX_SBIN_PATH.old'
+	cp $NGX_OBJS/nginx '__DESTROOT__$NGX_SBIN_PATH'
 
-	test -d '`dirname "$NGX_CONF_PATH"`' \
-		|| mkdir -p '`dirname "$NGX_CONF_PATH"`'
+	test -d '`dirname "__DESTROOT__$NGX_CONF_PATH"`' \
+		|| mkdir -p '`dirname "__DESTROOT__$NGX_CONF_PATH"`'
 
-	cp conf/koi-win '`dirname "$NGX_CONF_PATH"`'
-	cp conf/koi-utf '`dirname "$NGX_CONF_PATH"`'
-	cp conf/win-utf '`dirname "$NGX_CONF_PATH"`'
+	cp conf/koi-win '`dirname "__DESTROOT__$NGX_CONF_PATH"`'
+	cp conf/koi-utf '`dirname "__DESTROOT__$NGX_CONF_PATH"`'
+	cp conf/win-utf '`dirname "__DESTROOT__$NGX_CONF_PATH"`'
 
-	test -f '`dirname "$NGX_CONF_PATH"`/mime.types' || \
-		cp conf/mime.types '`dirname "$NGX_CONF_PATH"`/mime.types'
-	cp conf/mime.types '`dirname "$NGX_CONF_PATH"`/mime.types.default'
+	test -f '`dirname "__DESTROOT__$NGX_CONF_PATH"`/mime.types' || \
+		cp conf/mime.types '`dirname "__DESTROOT__$NGX_CONF_PATH"`/mime.types'
+	cp conf/mime.types '`dirname "__DESTROOT__$NGX_CONF_PATH"`/mime.types.default'
 
-	test -f '$NGX_CONF_PATH' || cp conf/nginx.conf '$NGX_CONF_PATH'
-	cp conf/nginx.conf '`dirname "$NGX_CONF_PATH"`/nginx.conf.default'
+	test -f '__DESTROOT__$NGX_CONF_PATH' || cp conf/nginx.conf '__DESTROOT__$NGX_CONF_PATH'
+	cp conf/nginx.conf '`dirname "__DESTROOT__$NGX_CONF_PATH"`/nginx.conf.default'
 
-	test -d '`dirname "$NGX_PID_PATH"`' \
-		|| mkdir -p '`dirname "$NGX_PID_PATH"`'
+	test -d '`dirname "__DESTROOT__$NGX_PID_PATH"`' \
+		|| mkdir -p '`dirname "__DESTROOT__$NGX_PID_PATH"`'
 
-	test -d '`dirname "$NGX_HTTP_LOG_PATH"`' || \
-		mkdir -p '`dirname "$NGX_HTTP_LOG_PATH"`'
+	test -d '`dirname "__DESTROOT__$NGX_HTTP_LOG_PATH"`' || \
+		mkdir -p '`dirname "__DESTROOT__$NGX_HTTP_LOG_PATH"`'
 
-	test -d '$NGX_PREFIX/html' || cp -r html '$NGX_PREFIX'
+	test -d '__DESTROOT__$NGX_PREFIX/share/nginx/html' || \
+		mkdir -p '__DESTROOT__$NGX_PREFIX/share/nginx/html' && \
+		cp -r html '__DESTROOT__$NGX_PREFIX/share/nginx'
 END
 
 
-if test -n "$NGX_ERROR_LOG_PATH"; then
+if test -n "__DESTROOT__$NGX_ERROR_LOG_PATH"; then
     cat << END                                                >> $NGX_MAKEFILE
 
-	test -d '`dirname "$NGX_ERROR_LOG_PATH"`' || \
-		mkdir -p '`dirname "$NGX_ERROR_LOG_PATH"`'
+	test -d '`dirname "__DESTROOT__$NGX_ERROR_LOG_PATH"`' || \
+		mkdir -p '`dirname "__DESTROOT__$NGX_ERROR_LOG_PATH"`'
 END
 
 fi
