--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ applnk/miscellaneous.desktop	20 Jan 2003 02:50:02 -0000	1.5
@@ -0,0 +1,35 @@
+[Desktop Entry]
+Icon=edu_miscellaneous
+Name=Miscellaneous
+Name[af]=Allerande
+Name[bg]=Разни
+Name[bs]=Razno
+Name[da]=Diverse
+Name[de]=Verschiedenes
+Name[el]=Διάφορα
+Name[es]=Varios
+Name[et]=Mitmesugust
+Name[fo]=Ymiskt
+Name[fr]=Divers
+Name[he]=שונות
+Name[hu]=Egyéb
+Name[it]=Miscellanea
+Name[ja]=その他
+Name[lt]=Įvairūs dalykai
+Name[lv]=Dažādi
+Name[nb]=Diverse
+Name[nl]=Diversen
+Name[nn]=Ymse
+Name[nso]=Hlakahlakanego
+Name[pt]=Vários
+Name[pt_BR]=Outros
+Name[ru]=Разное
+Name[sk]=Rôzne
+Name[sl]=Razno
+Name[sv]=Diverse
+Name[tr]=Çeşitli
+Name[ven]=Zwo tangananaho
+Name[xh]=Umxhube wezinto
+Name[zh_CN]=杂项
+Name[zh_TW]=雜項
+Name[zu]=Ingxubevange
