--- flashkard/flashkard/main.cpp	13 Mar 2003 19:52:03 -0000	1.1.1.3
+++ flashkard/flashkard/main.cpp	30 Oct 2002 00:52:35 -0000	1.3
@@ -36,7 +36,7 @@
   
   KAboutData aboutData( "flashkard", I18N_NOOP("FlashKard"),
 			"1.0", description, KAboutData::License_GPL,
-			"(c) 2002, Scott Wheeler", 0, "http://edu.kde.org/flashkard/");
+			"(c) 2002, Scott Wheeler", 0, "http://edu.kde.org/flashkard/", "submit@bugs.kde.org");
 
   aboutData.addAuthor("Scott Wheeler", 0, "wheeler@kde.org");
   KCmdLineArgs::init(argc, argv, &aboutData);
