--- kstars/kstars/data/Makefile.am	5 Nov 2002 20:01:24 -0000	1.1.1.5
+++ kstars/kstars/data/Makefile.am	31 Oct 2002 19:19:39 -0000	1.2
@@ -627,9 +627,9 @@
 	-rm -f $(kde_datadir)/kstars/cnames.dat
 	-rm -f $(kde_datadir)/kstars/ngcic.dat
 	-rm -f $(kde_datadir)/kstars/sao.dat
-	-rm -f $(kde_icondir)/hicolor/16x16/apps/kstars.png
-	-rm -f $(kde_icondir)/hicolor/32x32/apps/kstars.png  
-	-rm -f $(kde_icondir)/hicolor/48x48/apps/kstars.png
+	-rm -f $(kde_icondir)/crystalsvg/16x16/apps/kstars.png
+	-rm -f $(kde_icondir)/crystalsvg/32x32/apps/kstars.png  
+	-rm -f $(kde_icondir)/crystalsvg/48x48/apps/kstars.png
 	-rm -f $(kde_icondir)/locolor/16x16/apps/kstars.png 
 	-rm -f $(kde_icondir)/locolor/32x32/apps/kstars.png 
 	-rm -f $(kde_icondir)/locolor/48x48/apps/kstars.png 
