--- kvoctrain/kvoctrain/Makefile.am	27 Nov 2002 00:24:55 -0000	1.1.1.6
+++ kvoctrain/kvoctrain/Makefile.am	31 Oct 2002 19:19:39 -0000	1.2
@@ -49,13 +49,14 @@
 	$(mkinstalldirs) $(DESTDIR)$(kde_appsdir)/Edutainment/Languages/
 	$(INSTALL_DATA) $(srcdir)/kvoctrain.desktop $(DESTDIR)$(kde_appsdir)/Edutainment/Languages/kvoctrain.desktop
 	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/
-	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/48x48/apps/
-	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/32x32/apps/
-	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/hicolor/22x22/apps/
+	$(INSTALL_DATA) $(srcdir)/pics/kvoctrain.xpm $(DESTDIR)$(kde_icondir)/kvoctrain.xpm
+	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/crystalsvg/48x48/apps/
+	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/crystalsvg/32x32/apps/
+	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/crystalsvg/22x22/apps/
 	$(mkinstalldirs) $(DESTDIR)$(kde_icondir)/locolor/16x16/apps/
-	$(INSTALL_DATA) $(srcdir)/pics/48x48-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/hicolor/48x48/apps/kvoctrain.xpm
-	$(INSTALL_DATA) $(srcdir)/pics/32x32-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/hicolor/32x32/apps/kvoctrain.xpm
-	$(INSTALL_DATA) $(srcdir)/pics/22x22-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/hicolor/22x22/apps/kvoctrain.xpm
+	$(INSTALL_DATA) $(srcdir)/pics/48x48-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/crystalsvg/48x48/apps/kvoctrain.xpm
+	$(INSTALL_DATA) $(srcdir)/pics/32x32-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/crystalsvg/32x32/apps/kvoctrain.xpm
+	$(INSTALL_DATA) $(srcdir)/pics/22x22-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/crystalsvg/22x22/apps/kvoctrain.xpm
 	$(INSTALL_DATA) $(srcdir)/pics/16x16-kvoctrain.xpm $(DESTDIR)$(kde_icondir)/locolor/16x16/apps/kvoctrain.xpm
 	$(mkinstalldirs) $(kde_datadir)/kvoctrain
 	for pic in $(APPS_FILES); do \
