--- tests/kakasi-1.orig	2006-09-21 16:32:59.000000000 +0900
+++ tests/kakasi-1	2011-02-28 20:49:46.000000000 +0900
@@ -25,9 +25,9 @@
 		expect=`eval "echo $\`echo $j\`"`
 		for x in $input
 		do
-			/bin/echo -n -e "\\$x"
+			gecho -n -e "\\$x"
 		done | ../src/kakasi -i$i -o$j > tmp
-		ret=`cat tmp|od -t oC -w80|head -1|sed -e 's/^0* //'`
+		ret=`cat tmp|god -t oC -w80|head -1|sed -e 's/^0* //'`
 		if [ ! "x$ret" = "x$expect" ]
 		then
 			echo "unexpected result." >> $LOG
