--- configure.in.orig	2020-12-16 16:43:25.000000000 +0900
+++ configure.in	2020-12-16 19:20:54.000000000 +0900
@@ -84,7 +84,8 @@
     AM_ICONV
     LIBS="$LIBICONV $LIBS"
     AC_DEFINE(KAKASI_SUPPORT_UTF8, 1, [KAKASI_SUPPORT_UTF8])
-    AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>],
+    AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>
+#include <stdlib.h>],
 		    [if (iconv_open("EUC-JP", "UTF-8") == -1) exit(1);])],
 	[],
 	[AC_MSG_ERROR([can not use EUC-JP or UTF-8 encoding on iconv])])
