--- rpm-5.4.9/syck/lib/Makefile.am.orig	2009-04-01 22:43:58.000000000 +0200
+++ rpm-5.4.9/syck/lib/Makefile.am	2013-03-17 13:57:08.000000000 +0100
@@ -4,7 +4,7 @@
 
 LEXLIB = @LEXLIB@
 
-YACC = bison
+YACC = @YACC@ 
 AM_YFLAGS = -d -t -v -p syck
 
 #lib_LTLIBRARIES = libsyck.la
--- rpm-5.4.9/syck/lib/Makefile.in.orig	2012-05-08 05:23:47.000000000 +0200
+++ rpm-5.4.9/syck/lib/Makefile.in	2013-03-17 13:57:21.000000000 +0100
@@ -164,7 +164,7 @@
 SHELL = @SHELL@
 STRIP = @STRIP@
 VERSION = @VERSION@
-YACC = bison
+YACC = @YACC@ 
 YFLAGS = @YFLAGS@
 abs_builddir = @abs_builddir@
 abs_srcdir = @abs_srcdir@
