--- Makefile.orig	2019-08-07 00:39:32.000000000 -0500
+++ Makefile	2019-11-11 23:02:11.000000000 -0600
@@ -1,5 +1,5 @@
 all:
-	cc ota.c 02_decompress.c -o ota.macOS -I./liblzma-master/src/liblzma/api/ -llzma -g2
+	$(CC) ota.c 02_decompress.c -o ota -I. $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -llzma -g2
 linux:
-	cc -DLINUX ota.c 02_decompress.c -o ota.ELF64 -I./liblzma-master/src/liblzma/api/ -llzma -g2
+	$(CC) -DLINUX ota.c 02_decompress.c -o ota -I. $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -llzma -g2
 
