--- AuthorizedTasks.m.orig	2006-07-06 17:18:37.000000000 -0500
+++ AuthorizedTasks.m	2011-05-07 23:11:49.000000000 -0500
@@ -35,7 +35,7 @@
     AuthorizationRights *authorizedRights;
     AuthorizationFlags flags;
     AuthorizationItem items[2];
-    char *crontabWriterPath = "/usr/local/bin/psyncwriter";
+    char *crontabWriterPath = "@PREFIX@/bin/psyncwriter";
     char *rmPath = "/bin/rm";	
     OSStatus err = 0;
     BOOL authorized = NO;
@@ -126,7 +126,7 @@
     AuthorizationRights *authorizedRights;
     AuthorizationFlags flags;
     AuthorizationItem items[2];
-    char *crontabWriterPath = "/usr/local/bin/psyncwriter";
+    char *crontabWriterPath = "@PREFIX@/bin/psyncwriter";
     char *rmPath = "/bin/rm";
     OSStatus err = 0;
     BOOL authorized = NO;
@@ -192,11 +192,11 @@
     //a NULL terminated array of C string pointers.
     args[0] = NULL;
     
-    err = AuthorizationExecuteWithPrivileges(authorizationRef,"/usr/local/bin/psyncwriter", 0, args, NULL);
+    err = AuthorizationExecuteWithPrivileges(authorizationRef,"@PREFIX@/bin/psyncwriter", 0, args, NULL);
     if(err!=0)
     {
         NSBeep();
-        NSLog(@"Couldn't find the psyncwriter helper tool in /usr/local/bin/",err);
+        NSLog(@"Couldn't find the psyncwriter helper tool in @PREFIX@/bin/",err);
 		return FALSE;
     }
 	return TRUE;
