--- runpsync.orig	2006-07-17 00:42:35.000000000 -0500
+++ runpsync	2011-05-07 23:24:24.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!@PERL@
 
 ### Copyright 2003 AcornSoft ###
 
@@ -43,11 +43,11 @@
 
 if(-d $mainHD && -d $secondHD) {
 
-print `touch /var/log/psync.log`;
-open(STDOUT, "| tee /var/log/psync.log");
+print `touch @PREFIX@/var/log/psync.log`;
+open(STDOUT, "| tee @PREFIX@/var/log/psync.log");
 $now=localtime();
 print "Backup begun $now\n";
-open(INPUT, "/usr/local/bin/psync $d $f $r '$mainHD' '$secondHD'|");
+open(INPUT, "@PREFIX@/bin/psync $d $f $r '$mainHD' '$secondHD'|");
 
 while(<INPUT>) {
 print $_;
@@ -58,7 +58,7 @@
 close STDOUT;
 } # end if
 else {
-open(FILE, ">/var/log/psync.log");
+open(FILE, ">@PREFIX@/var/log/psync.log");
 $now=localtime;
 print FILE "$now\n";
 print "$now\n";
