Index: kmrml/kmrml/Makefile.am
===================================================================
RCS file: /home/kde/kdegraphics/kmrml/kmrml/Makefile.am,v
retrieving revision 1.6
diff -u -b -u -r1.6 Makefile.am
--- kmrml/kmrml/Makefile.am	14 Jun 2003 12:53:42 -0000	1.6
+++ kmrml/kmrml/Makefile.am	19 Jun 2003 01:52:29 -0000
@@ -28,7 +28,7 @@
 servicemenudir = $(kde_datadir)/konqueror/servicemenus
 
 #############################################
-CLEANFILES = dummy.cpp
+CLEANFILES = mrmlsearch_main.cpp
 bin_PROGRAMS = mrmlsearch
 lib_LTLIBRARIES = mrmlsearch.la
 
@@ -36,9 +36,12 @@
 mrmlsearch_la_LDFLAGS = $(all_libraries) -module -avoid-version
 mrmlsearch_la_SOURCES = mrmlsearch.cpp
 
-mrmlsearch_SOURCES = dummy.cpp
-mrmlsearch_LDADD = mrmlsearch.la $(LIB_KDECORE)
+mrmlsearch_SOURCES = mrmlsearch_main.cpp
+mrmlsearch_LDADD   = $(LIB_KMRMLSTUFF) $(LIB_KDECORE)
 mrmlsearch_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+
+mrmlsearch_main.cpp: mrmlsearch.cpp
+	cat mrmlsearch.cpp > mrmlsearch_main.cpp
 
 messages:
 	$(EXTRACTRC) */*.ui > rc.cpp
