Index: kghostview/Makefile.am
===================================================================
RCS file: /home/kde/kdegraphics/kghostview/Makefile.am,v
retrieving revision 1.72
diff -u -b -u -r1.72 Makefile.am
--- kghostview/Makefile.am	5 May 2003 17:28:02 -0000	1.72
+++ kghostview/Makefile.am	19 Jun 2003 01:52:04 -0000
@@ -7,21 +7,29 @@
 bin_PROGRAMS = 	kghostview
 kde_module_LTLIBRARIES = libkghostviewpart.la
 noinst_LTLIBRARIES = libdscparse.la
+lib_LTLIBRARIES = libkghostview_shared.la
 
-libkghostviewpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-libkghostviewpart_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) -lkdeprint \
+libkghostview_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkghostview_shared_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) -lkdeprint \
 	libdscparse.la
 
 # Check "make final" after making changes to the following line!!
-libkghostviewpart_la_SOURCES = kgvshell.cpp kgvdocument.cpp kgv_miniwidget.cpp \
+libkghostview_shared_la_SOURCES = kgvshell.cpp kgvdocument.cpp kgv_miniwidget.cpp \
 	marklist.cpp logwindow.cpp infodialog.cpp gotodialog.cpp \
 	kgvpageview.cpp ps.c kgv_view.cpp scrollbox.cpp  kgvpagedecorator.cpp \
 	kgvconfigdialog.cpp kgvmainwidget.cpp qttableview.cpp \
 	kdscerrordialog.cpp displayoptions.cpp kpswidget.cpp
 
+libkghostviewpart_la_SOURCES = libkghostviewpart_la_dummy.cpp
+libkghostviewpart_la_LIBADD  = libkghostview_shared.la
+libkghostviewpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+
+libkghostviewpart_la_dummy.cpp:
+	echo > libkghostviewpart_la_dummy.cpp
+
 kghostview_SOURCES = main.cpp
 kghostview_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kghostview_LDADD = libkghostviewpart.la
+kghostview_LDADD = libkghostview_shared.la
 
 libdscparse_la_LDFLAGS = $(all_libraries) -no-undefined
 libdscparse_la_LIBADD = $(LIB_QT)
