Index: kpovmodeler/configure.in.in
===================================================================
RCS file: /home/kde/kdegraphics/kpovmodeler/configure.in.in,v
retrieving revision 1.7
diff -u -b -u -r1.7 configure.in.in
--- kpovmodeler/configure.in.in	25 Apr 2003 21:13:30 -0000	1.7
+++ kpovmodeler/configure.in.in	19 Jun 2003 01:53:19 -0000
@@ -29,7 +29,7 @@
 fi
 
 
-KDE_FIND_PATH(freetype-config, FREETYPE_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin])
+KDE_FIND_PATH(freetype-config, FREETYPE_CONFIG, [/usr/X11R6/bin ${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin])
 
 if test -n "$FREETYPE_CONFIG"; then
   vers=`$FREETYPE_CONFIG --version 2>/dev/null | sed -e 's/libfreetype //' | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2) * 1000 + $3;}'`
