Index: kpovmodeler/Makefile.am
===================================================================
RCS file: /home/kde/kdegraphics/kpovmodeler/Makefile.am,v
retrieving revision 1.55
diff -u -b -u -r1.55 Makefile.am
--- kpovmodeler/Makefile.am	25 May 2003 18:52:05 -0000	1.55
+++ kpovmodeler/Makefile.am	19 Jun 2003 04:05:02 -0000
@@ -1,12 +1,14 @@
+KDE_OPTIONS = nofinal
 
 INCLUDES = $(LIBFREETYPE_CFLAGS) $(all_includes)
 
 SUBDIRS = pics examples
 
 kde_module_LTLIBRARIES = libkpovmodelerpart.la
+lib_LTLIBRARIES = libkpovmodeler_shared.la
 bin_PROGRAMS = kpovmodeler
 
-libkpovmodelerpart_la_SOURCES = pmpart.cpp pmfactory.cpp \
+libkpovmodeler_shared_la_SOURCES = pmpart.cpp pmfactory.cpp \
 	pmview.cpp pmshell.cpp pmobjectdrag.cpp \
 	pmtreeview.cpp pmtreeviewitem.cpp pmmessage.cpp pmerrordialog.cpp \
 	pminsertpopup.cpp pminserterrordialog.cpp \
@@ -180,13 +182,20 @@
 	\
 	version.h pmdocumentformat.h pmdebug.h
 
-libkpovmodelerpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(KDE_PLUGIN) $(LIBFREETYPE_RPATH) 
-libkpovmodelerpart_la_LIBADD = $(LIBFREETYPE_LIBS) $(LIB_KPARTS) $(GLLIB) -lXmu -lXi
-libkpovmodelerpart_la_METASOURCES = AUTO
+libkpovmodeler_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) $(LIBFREETYPE_RPATH) -no-undefined
+libkpovmodeler_shared_la_LIBADD = $(LIBFREETYPE_LIBS) $(LIB_KPARTS) $(GLLIB) -lXmu -lXi
+libkpovmodeler_shared_la_METASOURCES = AUTO
+
+libkpovmodelerpart_la_SOURCES = libkpovmodelerpart_la_dummy.cpp
+libkpovmodelerpart_la_LDFLAGS = $(all_libraries) -avoid-version -module
+libkpovmodelerpart_la_LIBADD  = libkpovmodeler_shared.la
+
+libkpovmodelerpart_la_dummy.cpp:
+	echo > libkpovmodelerpart_la_dummy.cpp
 
 kpovmodeler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kpovmodeler_SOURCES = main.cpp
-kpovmodeler_LDADD = libkpovmodelerpart.la $(LIB_KFILE)
+kpovmodeler_LDADD = libkpovmodeler_shared.la $(LIB_KFILE)
 
 #kpovmodeler_SOURCES = $(pmallsources) main.cpp
 #kpovmodeler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
