From ea003311a6137abe7bd5737900b9d9babfd0a299 Mon Sep 17 00:00:00 2001
From: sp1rit <sp1rit@disroot.org>
Date: Mon, 30 Jan 2023 19:52:53 +0100
Subject: [PATCH] tweaks to llvm clang build

Signed-off-by: sp1rit <sp1rit@disroot.org>
---
 main.cpp    | 4 ++--
 meson.build | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/main.cpp b/main.cpp
index 98b67a7..43904bc 100644
--- main.cpp
+++ main.cpp
@@ -23,8 +23,8 @@ int main (int argc, char *argv[])
 	Glib::RefPtr<Gtk::Application> app = Gtk::Application::create(argc, argv, "com.github.blackhole89.notekit");
 #endif
 	
-	app->signal_activate().connect( [app,&mainwindow]() {
-			mainwindow=new CMainWindow(app);
+	app->signal_activate().connect( [app]() {
+			auto mainwindow=new CMainWindow(app);
 			app->add_window(*mainwindow);
 		} );
 	
diff --git a/meson.build b/meson.build
index 6720b25..c43cee1 100644
--- meson.build
+++ meson.build
@@ -1,6 +1,6 @@
 project('notekit', 'cpp',
 	version : '0.3.0',
-	default_options : ['warning_level=3', 'cpp_std=c++14']
+	default_options : ['warning_level=3', 'cpp_std=c++17']
 )
 
 has_devenv = meson.version().version_compare('>=0.58.0')
