--- src/stock_crossover.f90	2023-02-22 00:15:58.000000000 +0800
+++ src/stock_crossover.f90	2023-10-18 11:56:36.000000000 +0800
@@ -18,7 +18,7 @@
 
     print *, 'Processing moving average crossover for ' // symbols(n)
 
-    call read_stock('data/' // trim(symbols(n)) //  '.csv', time,&
+    call read_stock('@DATADIR@/' // trim(symbols(n)) //  '.csv', time,&
       open, high, low, close, adjclose, volume)
 
     time = time(size(time):1:-1)

--- src/stock_gain.f90	2023-02-22 00:15:58.000000000 +0800
+++ src/stock_gain.f90	2023-10-18 11:56:49.000000000 +0800
@@ -16,7 +16,7 @@
 
   do n = 1, size(symbols)
 
-    call read_stock('data/' // trim(symbols(n)) //  '.csv', time,&
+    call read_stock('@DATADIR@/' // trim(symbols(n)) //  '.csv', time,&
       open, high, low, close, adjclose, volume)
 
     adjclose = reverse(adjclose)

--- src/stock_volatility.f90	2023-02-22 00:15:58.000000000 +0800
+++ src/stock_volatility.f90	2023-10-18 11:57:03.000000000 +0800
@@ -15,7 +15,7 @@
 
   do n = 1, size(symbols)
 
-    call read_stock('data/' // trim(symbols(n)) //  '.csv', time,&
+    call read_stock('@DATADIR@/' // trim(symbols(n)) //  '.csv', time,&
       open, high, low, close, adjclose, volume)
 
     im = size(time)
