--- Scripts/lego_bluetooth_config.orig	2010-07-27 08:07:46.000000000 -0500
+++ Scripts/lego_bluetooth_config	2011-06-01 22:38:01.000000000 -0500
@@ -54,10 +54,10 @@
 	    
 	    echo "" > $tempfile
 	    echo "device {" >> $tempfile
-	    echo -e "\tbdaddr $addr;" >> $tempfile
-	    echo -e "\tname \"$name\";" >> $tempfile
-	    echo -e "\tkey nokey;" >> $tempfile
-	    echo -e "\tpin \"$pin\";" >> $tempfile
+	    echo "\tbdaddr $addr;" >> $tempfile
+	    echo "\tname \"$name\";" >> $tempfile
+	    echo "\tkey nokey;" >> $tempfile
+	    echo "\tpin \"$pin\";" >> $tempfile
 	    echo "}" >> $tempfile
 	    
 	    hcsecd_conf='/etc/bluetooth/hcsecd.conf'
@@ -72,7 +72,7 @@
 	    bt_hosts='/etc/bluetooth/hosts'
 	    if [ "0`grep $addr $bt_hosts`" = "0" ]; then
 		echo "Adding $addr to $bt_hosts..."
-		echo -e "$addr\t$name" >> $bt_hosts
+		echo "$addr\t$name" >> $bt_hosts
 	    else
 		echo "$addr already in $bt_hosts"
 	    fi
