--- gdk/gdk-sharp.dll.config.in.orig	Sun Feb  6 08:32:24 2005
+++ gdk/gdk-sharp.dll.config.in	Sun Feb  6 08:37:29 2005
@@ -1,6 +1,7 @@
 <configuration>
-  <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
-  <dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libglib-2.0-0.dll" target="@GLIB_DIR@libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgobject-2.0-0.dll" target="@GLIB_DIR@libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgdk-win32-2.0-0.dll" target="@GNOME_DIR@libgdk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="libgdk_pixbuf-2.0-0.dll" target="@GNOME_DIR@libgdk_pixbuf-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="gdksharpglue" target="@GNOME_DIR@libgdksharpglue@LIB_SUFFIX@"/>
 </configuration>
