--- gdk/gdk-sharp.dll.config.in.orig	2008-10-31 17:10:14.000000000 +0100
+++ gdk/gdk-sharp.dll.config.in	2008-10-31 17:14:34.000000000 +0100
@@ -3,4 +3,5 @@
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="gdksharpglue" target="libgdksharpglue.so"/>
 </configuration>
