--- glib/glib-sharp.dll.config.in.orig	2008-10-31 17:10:21.000000000 +0100
+++ glib/glib-sharp.dll.config.in	2008-10-31 17:14:21.000000000 +0100
@@ -2,4 +2,5 @@
   <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="glibsharpglue" target="libglibsharpglue.so"/>
 </configuration>
