--- glib/glue/list.c.orig	2004-06-25 13:42:19.000000000 -0500
+++ glib/glue/list.c	2012-04-30 14:05:08.000000000 -0500
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
--- glib/glue/slist.c.orig	2004-06-25 13:42:19.000000000 -0500
+++ glib/glue/slist.c	2012-04-30 14:05:08.000000000 -0500
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
