--- Makefile.in.orig	2007-03-03 10:43:53.000000000 +1100
+++ Makefile.in	2012-07-13 08:44:17.000000000 +1000
@@ -306,7 +306,7 @@
 #========================================================================
 
 pkgIndex.tcl-hand:
-	@(echo 'package ifneeded $(PACKAGE_NAME) $(PACKAGE_VERSION)\
+	@(echo 'package ifneeded $(PACKAGE_NAME) $(PACKAGE_VERSION) \
 	"load [list [file join $$dir $(PKG_LIB_FILE)]];\
          source [list [file join $$dir tdom.tcl]]"'\
 	) > pkgIndex.tcl
