--- CMakeLists.txt.orig	2013-06-02 06:39:23.000000000 +0900
+++ CMakeLists.txt	2014-06-14 13:44:38.000000000 +0900
@@ -15,8 +15,8 @@
 macro_optional_find_package(QScintilla)
 macro_log_feature(QSCINTILLA_FOUND "QScintilla2" "QScintilla2 libraries" "http://www.riverbankcomputing.co.uk/software/qscintilla/intro" FALSE "" "Needed to compile QScintilla2 bindings")
 
-macro_optional_find_package(Qwt5)
-macro_log_feature(Qwt5_Qt4_FOUND "Qwt5 for Qt4" "Qwt5 libraries for Qt4" "http://qwt.sourceforge.net" FALSE "" "Needed to compile Qwt5 bindings")
+#macro_optional_find_package(Qwt5)
+#macro_log_feature(Qwt5_Qt4_FOUND "Qwt5 for Qt4" "Qwt5 libraries for Qt4" "http://qwt.sourceforge.net" FALSE "" "Needed to compile Qwt5 bindings")
 
 macro_optional_find_package(Phonon)
 macro_log_feature(PHONON_FOUND "Phonon" "Phonon multimedia framework" "http://www.kde.org" FALSE "" "Needed to compile Phonon bindings")
