--- AppController.m	2008-10-04 20:14:37.000000000 +0200
+++ AppController copie.m	2014-07-07 10:23:25.000000000 +0200
@@ -17,14 +17,6 @@
 
 #define _DISPLENGTH 40
 
-enum {
-    NSWindowCollectionBehaviorDefault = 0,
-    NSWindowCollectionBehaviorCanJoinAllSpaces = 1 << 0,
-    NSWindowCollectionBehaviorMoveToActiveSpace = 1 << 1
-};
-
-typedef unsigned NSWindowCollectionBehavior;
-
 @interface NSWindow (NSWindowCollectionBehavior)
 - (void)setCollectionBehavior:(NSWindowCollectionBehavior)behavior;
 @end
