diff -urNp stgit-0.17.1.orig/stgit/templates.py stgit-0.17.1/stgit/templates.py
--- stgit-0.17.1.orig/stgit/templates.py	2006-07-11 12:21:38.000000000 +0300
+++ stgit-0.17.1/stgit/templates.py	2016-11-01 19:01:47.000000000 +0300
@@ -31,7 +31,9 @@ def get_template(tfile):
                   os.path.join(os.path.expanduser('~'), '.stgit', 'templates',
                                tfile),
                   os.path.join(sys.prefix, 'share', 'stgit', 'templates',
-                               tfile) ]
+                               tfile),
+                  os.path.join("@@PREFIX@@", 'share', 'stgit', 'templates',
+                               tfile)]
 
     tmpl = None
     for t in tmpl_list:
