--- doc/pinfo.1.in.orig	2006-03-09 23:54:43.000000000 +0300
+++ doc/pinfo.1.in	2012-08-06 14:44:33.000000000 +0400
@@ -27,7 +27,7 @@
 .IR "filehandling_functions.c".
 .P
 When the search for info pages fails, man is called with the infopage
-argument, and it's output is parsed by pinfo. This means that when you don't
+argument, and its output is parsed by pinfo. This means that when you don't
 have the appropriate info page, but have a man page instead; the man page
 will be viewed.
 .P
@@ -53,10 +53,10 @@
 .P
 You can also call the man function of pinfo in another way. When pinfo is
 called with an argv[0] (the program file name), which contains the word 'man'
-in it's name, the man functions are enabled automatically.
+in its name, the man functions are enabled automatically.
 .P
 Previously there was a symlink to pinfo, called pman, but I had to remove it
-from the distribution, since it's name was in conflict with some other
+from the distribution, since its name was in conflict with some other
 utility. Anyway, you can feel free to create such a link if you wish.
 .P
 .BR "-r",
@@ -188,7 +188,7 @@
 KEY_BACK_1=KEY_LEFT
 KEY_BACK_2='l'
 KEY_FOLLOWLINK_1=KEY_RIGHT
-KEY_FOLLOWLINK_2='\n'
+KEY_FOLLOWLINK_2='\\n'
 # 12 is a code for ctrl+l
 KEY_REFRESH_1=12
 KEY_REFRESH_2='~'
@@ -237,10 +237,10 @@
 blinking attribute, or highlighted background in some other configurations.
 .P
 Now let's move to the key definitions. Here we first put a key name (again
-all keys are present in the example); then we enter it's value -- either
+all keys are present in the example); then we enter its value -- either
 surrounded
 by apostrophes, or a keycode number (like in KEY_REFRESH_1), or its mnemonic
-code name if it's a special key (like i.e. in KEY_FOLLOWLINK_1).
+code name if its a special key (like i.e. in KEY_FOLLOWLINK_1).
 .P
 If you wish to specify key by code value, use the supplied program 'testkey'
 to obtain the needed value. It mainly is a feature, when you want to add
@@ -274,10 +274,10 @@
 Alternate key for repeating the last search.
 .TP
 .B KEY_GOTO_1
-Key for explicitly going to a node (by specifying it's name).
+Key for explicitly going to a node (by specifying its name).
 .TP 
 .B KEY_GOTO_2
-Alternate key for explicitly going to a node (by specifying it's name).
+Alternate key for explicitly going to a node (by specifying its name).
 .TP 
 .B KEY_PREVNODE_1
 Key for going to a node marked as 'Prev' in the header. In man page viewer
@@ -441,7 +441,7 @@
 C/C++ ;).
 .TP
 .B [number]
-\- you can also specify key as it's code number. It is useful e.g. when
+\- you can also specify key as its code number. It is useful e.g. when
 specifying control keys, and some nonstandard keys.
 A numerical value of zero turns given key binding off.
 .sp
