--- nextstep/compile.orig	2007-05-21 07:12:33.000000000 -0400
+++ nextstep/compile	2007-05-21 07:14:25.000000000 -0400
@@ -34,8 +34,10 @@
     export MACOSX_DEPLOYMENT_TARGET=10.3
     export MACOSX_DEPLOYMENT_TARGET_ppc=10.3
     export MACOSX_DEPLOYMENT_TARGET_i386=10.4
+	export SDKROOT_ppc=/Developer/SDKs/MacOSX10.3.9.sdk
+	export SDKROOT_i386=/Developer/SDKs/MacOSX10.4u.sdk
 #    export CFLAGS="$OPTFLAGS -universal -sdk /Developer/SDKs/MacOSX10.4u.sdk"
-    export CFLAGS="$OPTFLAGS -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk"
+    export CFLAGS="$OPTFLAGS -arch ppc
 else
     source /etc/GNUstep/GNUstep.conf
     export GNUSTEP_SYSTEM_ROOT
