diff --git python/lib/moresetuptools.py python/lib/moresetuptools.py
index f639ae88..274b11b7 100644
--- python/lib/moresetuptools.py
+++ python/lib/moresetuptools.py
@@ -111,7 +111,6 @@ def add_external_deps(ext, usedouble = False):
     # add accelerate on darwin
     if sys.platform.startswith('darwin'):
         ext.extra_link_args += ['-framework', 'Accelerate']
-        ext.define_macros += [('HAVE_ACCELERATE', 1)]
         ext.define_macros += [('HAVE_SOURCE_APPLE_AUDIO', 1)]
         ext.define_macros += [('HAVE_SINK_APPLE_AUDIO', 1)]
 
diff --git setup.py setup.py
index 88f8f070..1eb82da7 100755
--- setup.py
+++ setup.py
@@ -17,7 +17,7 @@ __aubio_version__ = get_aubio_version()
 
 include_dirs = []
 library_dirs = []
-define_macros = [('AUBIO_VERSION', '%s' % __aubio_version__)]
+define_macros = [('HAVE_FFTW3F', '1'), ('AUBIO_VERSION', '%s' % __aubio_version__)]
 extra_link_args = []
 
 include_dirs += ['python/ext']
