--- ffmpegaudio.cc.orig	2013-06-06 06:39:06.000000000 -0700
+++ ffmpegaudio.cc	2016-04-21 12:06:50.000000000 -0700
@@ -326,7 +326,11 @@
 
 bool DecoderContext::play( QString & errorString )
 {
+#if LIBAVCODEC_VERSION_MAJOR < 55 || ( LIBAVCODEC_VERSION_MAJOR == 55 && LIBAVCODEC_VERSION_MINOR < 28 )
   AVFrame * frame = avcodec_alloc_frame();
+#else
+  AVFrame * frame = av_frame_alloc();
+#endif
   if ( !frame )
   {
     errorString = QObject::tr( "avcodec_alloc_frame() failed." );
@@ -348,7 +352,11 @@
       }
     }
     // av_free_packet() must be called after each call to av_read_frame()
+#if LIBAVCODEC_VERSION_MAJOR < 57 || ( LIBAVCODEC_VERSION_MAJOR == 57 && LIBAVCODEC_VERSION_MINOR < 7 )
     av_free_packet( &packet );
+#else
+    av_packet_unref( &packet );
+#endif
   }
 
   if ( !isCancelled_ && codecContext_->codec->capabilities & CODEC_CAP_DELAY )
@@ -365,8 +373,10 @@
 
 #if LIBAVCODEC_VERSION_MAJOR < 54
   av_free( frame );
-#else
+#elif LIBAVCODEC_VERSION_MAJOR < 55 || ( LIBAVCODEC_VERSION_MAJOR == 55 && LIBAVCODEC_VERSION_MINOR < 28 )
   avcodec_free_frame( &frame );
+#else
+  av_frame_free( &frame );
 #endif
 
   return true;
