Index: setup.py
===================================================================
--- setup.py	(revision 2980)
+++ setup.py	(working copy)
@@ -105,7 +105,7 @@
 
         self.set_perms()
         self.set_modules_path()
-        self.fix_paths()
+        #self.fix_paths()
         self.create_uninstaller()
         self.finish_banner()
 
@@ -270,4 +270,4 @@
       packages = ['', 'umitCore', 'umitGUI', 'higwidgets'],
       data_files = data_files,
       cmdclass = {"install":umit_install,
-                  "sdist":umit_sdist})
\ No newline at end of file
+                  "sdist":umit_sdist})
