--- bindings/perl/src/CMakeLists.txt.orig	2013-11-02 11:56:51.000000000 -0700
+++ bindings/perl/src/CMakeLists.txt	2013-11-02 11:59:08.000000000 -0700
@@ -14,6 +14,7 @@
 
 target_link_libraries(PLlibproxy ${PERL_LIBRARY} libproxy)
 set_target_properties(PLlibproxy PROPERTIES OUTPUT_NAME "Libproxy")
+set_target_properties(PLlibproxy PROPERTIES INSTALL_NAME_DIR ${PX_PERL_ARCH}/auto/Net/Libproxy )
 set_target_properties(PLlibproxy PROPERTIES PREFIX "")
 
 install( TARGETS PLlibproxy DESTINATION ${PX_PERL_ARCH}/auto/Net/Libproxy )
