--- bindings/perl/src/CMakeLists.txt.orig	2016-04-28 10:04:41.000000000 -0700
+++ bindings/perl/src/CMakeLists.txt	2016-04-30 21:08:29.000000000 -0700
@@ -19,6 +19,7 @@
 
 target_link_libraries(PLlibproxy ${PLlibproxy_LIB_DEPENDENCIES})
 set_target_properties(PLlibproxy PROPERTIES OUTPUT_NAME "Libproxy")
+set_target_properties(PLlibproxy PROPERTIES INSTALL_NAME_DIR ${PX_PERL_ARCH}/auto/Net/Libproxy )
 set_target_properties(PLlibproxy PROPERTIES PREFIX "")
 
 install( TARGETS PLlibproxy DESTINATION ${PX_PERL_ARCH}/auto/Net/Libproxy )
