Don't overwrite MacPorts CFLAGS. Do also use MacPorts LDFLAGS.
--- Makefile.orig	2008-09-07 06:18:50.000000000 -0500
+++ Makefile	2021-11-21 23:57:27.000000000 -0600
@@ -1,6 +1,7 @@
  
 PROGS = codegroup
-CFLAGS = -O -Wall
+CFLAGS ?= -O
+CFLAGS += -Wall
 PAGER = more
 
 all:	$(PROGS)
@@ -9,7 +10,7 @@
 	rm -f $(PROGS) *.o *.bak *.zip core code.* *.out
 
 codegroup: codegroup.o
-	$(CC) codegroup.o -o codegroup $(CFLAGS)
+	$(CC) codegroup.o -o codegroup $(CFLAGS) $(LDFLAGS)
 
 #   Create zipped archive
 
