--- Makefile.in.old	2015-07-21 11:12:46.000000000 +0200
+++ Makefile.in	2015-07-21 11:13:20.000000000 +0200
@@ -6388,7 +6388,7 @@
 	@echo "== done =="
 
 install-data-hook:
-@CMULOCAL_TRUE@	$(INSTALL) -m 644 $(top_srcdir)/depot/depot.conf $(DESTDIR)/
+#@CMULOCAL_TRUE@	$(INSTALL) -m 644 $(top_srcdir)/depot/depot.conf $(DESTDIR)/
 
 install-exec-hook:
 @PERL_TRUE@	for s in installsieve sieveshell; \
