--- src/pffft.c.orig	2020-11-26 15:41:52.000000000 -0800
+++ src/pffft.c	2020-11-26 15:41:57.000000000 -0800
@@ -157,7 +157,7 @@
 /*
   ARM NEON support macros
 */
-#elif !defined(PFFFT_SIMD_DISABLE) && defined(__arm__)
+#elif !defined(PFFFT_SIMD_DISABLE) && ( defined(__arm__) || defined(__arm64__) )
 #  include <arm_neon.h>
 typedef float32x4_t v4sf;
 #  define SIMD_SZ 4
@@ -1732,7 +1732,7 @@
   const v4sf * RESTRICT vb = (const v4sf*)b;
   v4sf * RESTRICT vab = (v4sf*)ab;
 
-#ifdef __arm__
+#if defined(__arm__) || defined(__arm64__)
   __builtin_prefetch(va);
   __builtin_prefetch(vb);
   __builtin_prefetch(vab);
