--- src/workspace.m.orig	2023-02-01 00:45:48.000000000 +0100
+++ src/workspace.m	2023-02-01 00:46:37.000000000 +0100
@@ -103,14 +103,6 @@
 {
     if (!CGDisplayIsBuiltin(did)) return 0;
 
-    if (__builtin_available(macos 12.0, *)) {
-        for (NSScreen *screen in [NSScreen screens]) {
-            if ([[[screen deviceDescription] objectForKey:@"NSScreenNumber"] unsignedIntValue] == did) {
-                return screen.safeAreaInsets.top;
-            }
-        }
-    }
-
     return 0;
 }
 
