--- Makefile.old	2019-01-19 10:17:28.000000000 -0600
+++ Makefile	2022-07-21 13:01:19.000000000 -0500
@@ -1,11 +1,11 @@
-CC=		gcc
-LDFLAGS=	-L/usr/lib	-L/usr/local/lib  	-L/usr/lib/ncurses	-L/usr/local/lib/ncurses
-CPPFLAGS=	-I/usr/include	-I/usr/local/include	-I/usr/include/ncurses	-I/usr/local/include/ncurses
-CFLAGS=		-O3 -Wall #-std=c99 
+#CC=		gcc
+#LDFLAGS=	-L/usr/lib	-L/usr/local/lib  	-L/usr/lib/ncurses	-L/usr/local/lib/ncurses
+#CPPFLAGS=	-I/usr/include	-I/usr/local/include	-I/usr/include/ncurses	-I/usr/local/include/ncurses
+#CFLAGS=		-O3 -Wall #-std=c99 
 #CFLAGS+= -ffunction-sections -fdata-sections
 #LDFLAGS+= --gc-sections 
 LIBS=		-lncurses
-DESTDIR=	/usr/local/
+#DESTDIR=	/usr/local/
 
 OFILES=buffers.o \
 	configfile.o \
@@ -29,10 +29,10 @@
 install:all
 	strip dhex
 	cp dhex $(DESTDIR)/bin
-	cp dhex.1 $(DESTDIR)/man/man1
-	cp dhexrc.5 $(DESTDIR)/man/man5
-	cp dhex_markers.5 $(DESTDIR)/man/man5
-	cp dhex_searchlog.5 $(DESTDIR)/man/man5
+	cp dhex.1 $(DESTDIR)/share/man/man1
+	cp dhexrc.5 $(DESTDIR)/share/man/man5
+	cp dhex_markers.5 $(DESTDIR)/share/man/man5
+	cp dhex_searchlog.5 $(DESTDIR)/share/man/man5
 
 	
 
