--- configure.orig	2021-04-22 08:06:56
+++ configure	2023-07-23 15:24:52
@@ -12601,6 +12601,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_z_compress2" >&5
 $as_echo "$ac_cv_lib_z_compress2" >&6; }
 if test "x$ac_cv_lib_z_compress2" = xyes; then :
+                            HAVE_LIBZ="yes"
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBZ 1
 _ACEOF
@@ -19869,6 +19870,7 @@
 /* end confdefs.h.  */
 
                 #include <stddef.h>
+                #include <stdlib.h>
                 #include <szlib.h>
 
                 int main(void)
@@ -19908,6 +19910,7 @@
 /* end confdefs.h.  */
 
                 #include <stddef.h>
+                #include <stdlib.h>
                 #include <szlib.h>
 
                 int main(void)
@@ -20821,6 +20824,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+            #include <stdlib.h>
             #include <H5Ipublic.h>
             #include <H5Zpublic.h>
 
@@ -20891,6 +20895,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+            #include <stdlib.h>
             #include <H5Ipublic.h>
             #include <H5Zpublic.h>
 
