--- plugins/Makefile.am.orig	2017-01-23 06:31:07.000000000 -0500
+++ plugins/Makefile.am	2017-01-23 10:11:16.000000000 -0500
@@ -41,12 +41,10 @@
 srm7500libusb_la_LDFLAGS    = $(AM_LDFLAGS) @usb_libs@
 srm7500libusb_la_CFLAGS     = $(AM_CFLAGS) $(LIBUSB_CFLAGS)
 
-if LINUX_KERNEL
 plugin_LTLIBRARIES          += commandir.la
 commandir_la_SOURCES        = commandir.c
 commandir_la_LDFLAGS        = $(AM_LDFLAGS) @usb_libs@
 commandir_la_CFLAGS         = $(AM_CFLAGS) $(LIBUSB_CFLAGS)
-endif
 
 endif
