--- test/Makefile.am.orig	2023-04-29 03:53:52.000000000 +0200
+++ test/Makefile.am	2023-07-31 17:14:20.000000000 +0200
@@ -32,37 +32,32 @@
 	# EOL

 check_PROGRAMS = \
-	overlay \
-	bzero \
 	closefrom \
 	endian \
 	explicit_bzero \
 	humanize \
-	fgetln \
-	fparseln \
+	overlay \
 	proctitle-init \
-	pwcache \
-	setmode \
-	strnstr \
-	strtonum \
-	vis \
-	vis-openbsd \
 	# EOL

-if NEED_PROGNAME
-check_PROGRAMS += progname
+if NEED_ARC4RANDOM
+if HAVE_LIBTESTU01
+arc4random_LDADD = $(LDADD) $(TESTU01_LIBS)
+
+check_PROGRAMS += arc4random
+endif
 endif

-if NEED_NLIST
-check_PROGRAMS += nlist
+if NEED_BZERO
+check_PROGRAMS += bzero
 endif

-if NEED_STRL
-check_PROGRAMS += strl
+if NEED_FGETLN
+check_PROGRAMS += fgetln
 endif

-if NEED_STRMODE
-check_PROGRAMS += strmode
+if NEED_FPARSELN
+check_PROGRAMS += fparseln
 endif

 if NEED_FPURGE
@@ -73,12 +68,50 @@
 check_PROGRAMS += funopen
 endif

-if NEED_ARC4RANDOM
-if HAVE_LIBTESTU01
-arc4random_LDADD = $(LDADD) $(TESTU01_LIBS)
+if NEED_ID_FROM_NAME
+check_PROGRAMS += pwcache
+endif

-check_PROGRAMS += arc4random
+if NEED_MD5
+check_PROGRAMS += md5
+
+if NEED_TRANSPARENT_LIBMD
+# On the installed system this is handled via the ld script.
+md5_LDADD = $(LDADD) $(MD5_LIBS)
+endif
+endif
+
+if NEED_NLIST
+check_PROGRAMS += nlist
+endif
+
+if NEED_PROGNAME
+check_PROGRAMS += progname
+endif
+
+if NEED_SETMODE
+check_PROGRAMS += setmode
+endif
+
+if NEED_STRL
+check_PROGRAMS += strl
+endif
+
+if NEED_STRMODE
+check_PROGRAMS += strmode
+endif
+
+if NEED_STRNSTR
+check_PROGRAMS += strnstr
+endif
+
+if NEED_STRTONUM
+check_PROGRAMS += strtonum
 endif
+
+if NEED_VIS
+check_PROGRAMS += vis \
+                  vis-openbsd
 endif

 if BUILD_LIBBSD_CTOR
@@ -91,15 +124,6 @@
 check_PROGRAMS += proctitle
 endif

-if NEED_MD5
-check_PROGRAMS += md5
-
-if NEED_TRANSPARENT_LIBMD
-# On the installed system this is handled via the ld script.
-md5_LDADD = $(LDADD) $(MD5_LIBS)
-endif
-endif
-
 fgetln_SOURCES = test-stream.c test-stream.h fgetln.c
 fgetln_CFLAGS = -Wno-deprecated-declarations
 fparseln_SOURCES = test-stream.c test-stream.h fparseln.c
