--- src/Makefile.am.orig	2023-04-29 03:53:52.000000000 +0200
+++ src/Makefile.am	2023-07-31 21:42:07.000000000 +0200
@@ -74,52 +74,26 @@
 	libbsd.sym \
 	# EOL
 endif
+
 libbsd_la_SOURCES = \
 	closefrom.c \
 	dehumanize_number.c \
-	err.c \
 	expand_number.c \
 	explicit_bzero.c \
-	fgetln.c \
-	fgetwln.c \
 	flopen.c \
-	fmtcheck.c \
-	fparseln.c \
 	freezero.c \
-	getbsize.c \
-	getpeereid.c \
-	heapsort.c \
 	humanize_number.c \
-	inet_net_pton.c \
 	local-elf.h \
 	local-link.h \
-	merge.c \
 	pidfile.c \
-	progname.c \
-	pwcache.c \
-	radixsort.c \
-	readpassphrase.c \
 	reallocarray.c \
-	reallocf.c \
 	recallocarray.c \
-	setmode.c \
 	setproctitle.c \
-	stringlist.c \
-	strnstr.c \
 	strtoi.c \
-	strtonum.c \
 	strtou.c \
 	timeconv.c \
-	unvis.c \
-	vis.c \
 	# EOL
 
-if NEED_BSD_GETOPT
-libbsd_la_SOURCES += \
-	bsd_getopt.c \
-	# EOL
-endif
-
 if NEED_ARC4RANDOM
 if !HAVE_GETENTROPY
 libbsd_la_SOURCES += \
@@ -137,6 +111,80 @@
 	# EOL
 endif
 
+if NEED_BSD_GETOPT
+libbsd_la_SOURCES += \
+	bsd_getopt.c \
+	# EOL
+endif
+
+if NEED_ERR
+libbsd_la_SOURCES += \
+	err.c \
+	# EOL
+endif
+
+if NEED_FGETLN
+libbsd_la_SOURCES += \
+	fgetln.c \
+	fgetwln.c \
+	# EOL
+endif
+
+if NEED_FMTCHECK
+libbsd_la_SOURCES += \
+	fmtcheck.c \
+	# EOL
+endif
+
+if NEED_FPARSELN
+libbsd_la_SOURCES += \
+	fparseln.c \
+	# EOL
+endif
+
+if NEED_FPURGE
+libbsd_la_SOURCES += \
+	fpurge.c \
+	# EOL
+endif
+
+if NEED_FUNOPEN
+libbsd_la_SOURCES += \
+	funopen.c \
+	# EOL
+endif
+
+if NEED_GETBSIZE
+libbsd_la_SOURCES += \
+	getbsize.c \
+	# EOL
+endif
+
+if NEED_GETPEEREID
+libbsd_la_SOURCES += \
+	getpeereid.c \
+	# EOL
+endif
+
+if NEED_HEAPSORT
+libbsd_la_SOURCES += \
+	heapsort.c \
+	merge.c \
+	# EOL
+endif
+
+if NEED_ID_FROM_NAME
+libbsd_la_SOURCES += \
+	pwcache.c \
+	# EOL
+endif
+
+if NEED_INET_NET_PTON
+libbsd_la_SOURCES += \
+	inet_net_pton.c \
+	# EOL
+endif
+
 if NEED_MD5
 libbsd_la_SOURCES += \
 	md5.c \
@@ -149,6 +197,42 @@
 	# EOL
 endif
 
+if NEED_PROGNAME
+libbsd_la_SOURCES += \
+	progname.c \
+	# EOL
+endif
+
+if NEED_RADIXSORT
+libbsd_la_SOURCES += \
+	radixsort.c \
+	# EOL
+endif
+
+if NEED_READPASSPHRASE
+libbsd_la_SOURCES += \
+	readpassphrase.c \
+	# EOL
+endif
+
+if NEED_REALLOCF
+libbsd_la_SOURCES += \
+	reallocf.c \
+	# EOL
+endif
+
+if NEED_SETMODE
+libbsd_la_SOURCES += \
+	setmode.c \
+	# EOL
+endif
+
+if NEED_STRINGLIST
+libbsd_la_SOURCES += \
+	stringlist.c \
+	# EOL
+endif
+
 if NEED_STRL
 libbsd_la_SOURCES += \
 	strlcat.c \
@@ -156,28 +240,35 @@
 	# EOL
 endif
 
-if NEED_WCSL
+if NEED_STRMODE
 libbsd_la_SOURCES += \
-	wcslcat.c \
-	wcslcpy.c \
+	strmode.c \
 	# EOL
 endif
 
-if NEED_STRMODE
+if NEED_STRNSTR
 libbsd_la_SOURCES += \
-	strmode.c \
+	strnstr.c \
 	# EOL
 endif
 
-if NEED_FPURGE
+if NEED_STRTONUM
 libbsd_la_SOURCES += \
-	fpurge.c \
+	strtonum.c \
 	# EOL
 endif
 
-if NEED_FUNOPEN
+if NEED_VIS
 libbsd_la_SOURCES += \
-	funopen.c \
+	unvis.c \
+	vis.c \
+	# EOL
+endif
+
+if NEED_WCSL
+libbsd_la_SOURCES += \
+	wcslcat.c \
+	wcslcpy.c \
 	# EOL
 endif
 
