--- src/libjasper/include/jasper/jas_math.h.orig	2021-04-19 15:06:02.000000000 +1000
+++ src/libjasper/include/jasper/jas_math.h	2021-07-24 17:36:10.000000000 +1000
@@ -209,6 +209,10 @@ inline static int_fast32_t jas_fast32_as
 #pragma GCC diagnostic pop
 #endif
 
+#ifndef __has_builtin
+#  define __has_builtin(x) 0
+#endif
+
 /******************************************************************************\
 * Safe integer arithmetic (i.e., with overflow checking).
 \******************************************************************************/
@@ -216,7 +220,7 @@ inline static int_fast32_t jas_fast32_as
 /* Compute the product of two size_t integers with overflow checking. */
 inline static bool jas_safe_size_mul(size_t x, size_t y, size_t *result)
 {
-#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ > 5)
+#if __has_builtin(__builtin_mul_overflow) || (defined(__GNUC__) && __GNUC__ > 5)
 	size_t result_buffer;
 	if (!result)
 		result = &result_buffer;
@@ -252,7 +256,7 @@ inline static bool jas_safe_size_mul3(si
 /* Compute the sum of two size_t integers with overflow checking. */
 inline static bool jas_safe_size_add(size_t x, size_t y, size_t *result)
 {
-#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ > 5)
+#if __has_builtin(__builtin_add_overflow) || (defined(__GNUC__) && __GNUC__ > 5)
 	size_t result_buffer;
 	if (!result)
 		result = &result_buffer;
@@ -271,7 +275,7 @@ inline static bool jas_safe_size_add(siz
 /* Compute the difference of two size_t integers with overflow checking. */
 inline static bool jas_safe_size_sub(size_t x, size_t y, size_t *result)
 {
-#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ > 5)
+#if __has_builtin(__builtin_sub_overflow) || (defined(__GNUC__) && __GNUC__ > 5)
 	size_t result_buffer;
 	if (!result)
 		result = &result_buffer;
@@ -291,7 +295,7 @@ inline static bool jas_safe_size_sub(siz
 inline static bool jas_safe_intfast32_mul(int_fast32_t x, int_fast32_t y,
   int_fast32_t *result)
 {
-#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ > 5)
+#if __has_builtin(__builtin_mul_overflow) || (defined(__GNUC__) && __GNUC__ > 5)
 	int_fast32_t result_buffer;
 	if (!result)
 		result = &result_buffer;
@@ -350,7 +354,7 @@ inline static bool jas_safe_intfast32_mu
 inline static bool jas_safe_intfast32_add(int_fast32_t x, int_fast32_t y,
   int_fast32_t *result)
 {
-#if defined(__clang__) || (defined(__GNUC__) && __GNUC__ > 5)
+#if __has_builtin(__builtin_add_overflow) || (defined(__GNUC__) && __GNUC__ > 5)
 	int_fast32_t result_buffer;
 	if (!result)
 		result = &result_buffer;
