--- src/app/CMakeLists.txt.orig	2025-03-27 04:54:43
+++ src/app/CMakeLists.txt	2025-03-29 09:09:40
@@ -38,8 +38,8 @@
 if(JAS_HAVE_OPENGL)
 	add_executable(jiv jiv.c)
 	target_link_libraries(jiv libjasper
-	  ${JAS_LIBJPEG_TARGET} ${HEIF_LIBRARIES} ${JAS_OPENGL_TARGETS}
-	  ${JAS_OPENGL_TARGETS} ${MATH_LIBRARY} ${THREAD_LIBRARY})
+	  ${JAS_OPENGL_TARGETS} ${JAS_LIBJPEG_TARGET} ${HEIF_LIBRARIES}
+	  ${MATH_LIBRARY} ${THREAD_LIBRARY})
 	set(public_programs "${public_programs}" jiv)
 	set(man_pages "${man_pages}" jiv.1)
 else()
