--- Modules/Core/GPUCommon/src/CMakeLists.txt.orig	2022-11-24 14:15:41.000000000 +0100
+++ Modules/Core/GPUCommon/src/CMakeLists.txt	2023-06-09 12:30:50.000000000 +0200
@@ -14,5 +14,6 @@
   write_gpu_kernels("${ITKGPUCommon_Kernels}" ITKGPUCommon_SRCS)
 
   itk_module_add_library(ITKGPUCommon ${ITKGPUCommon_SRCS})
-  target_link_libraries(ITKGPUCommon LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUCommon LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
+
--- Modules/Core/GPUCommon/CMakeLists.txt.orig	2022-11-24 14:15:41.000000000 +0100
+++ Modules/Core/GPUCommon/CMakeLists.txt	2023-06-09 12:30:45.000000000 +0200
@@ -25,9 +25,7 @@
 
 if(ITK_USE_GPU)
   set(ITKGPUCommon_LIBRARIES ITKGPUCommon ${OpenCL_LIBRARIES})
-  if(APPLE)
-    list(APPEND ITKGPUCommon_LIBRARIES "-framework OpenCL")
-  elseif(WINDOWS)
+  if(WINDOWS)
     list(APPEND ITKGPUCommon_LIBRARIES cfgmgr32 OneCoreUAP)
   endif()
 
