diff -ur ../marble-4.12.0-orig/src/plugins/designer/latloneditwidget/LatLonEditPlugin.cpp ./src/plugins/designer/latloneditwidget/LatLonEditPlugin.cpp
--- ../marble-4.12.0-orig/src/plugins/designer/latloneditwidget/LatLonEditPlugin.cpp	2013-12-12 04:55:59.000000000 +0900
+++ ./src/plugins/designer/latloneditwidget/LatLonEditPlugin.cpp	2013-12-24 21:45:40.000000000 +0900
@@ -17,7 +17,7 @@
 #include <QtPlugin>
 #include <QWidget>
 #include <QIcon>
-#include <QDesignerCustomWidgetInterface>
+#include <QtDesigner/QDesignerCustomWidgetInterface>
 
 
 LatLonEditPlugin::LatLonEditPlugin(QObject *parent) : QObject(parent)
diff -ur ../marble-4.12.0-orig/src/plugins/designer/latloneditwidget/LatLonEditPlugin.h ./src/plugins/designer/latloneditwidget/LatLonEditPlugin.h
--- ../marble-4.12.0-orig/src/plugins/designer/latloneditwidget/LatLonEditPlugin.h	2013-12-12 04:55:59.000000000 +0900
+++ ./src/plugins/designer/latloneditwidget/LatLonEditPlugin.h	2013-12-24 21:38:20.000000000 +0900
@@ -10,7 +10,7 @@
 #ifndef LATLONEDITPLUGIN_H
 #define LATLONEDITPLUGIN_H
 
-#include <QDesignerCustomWidgetInterface>
+#include <QtDesigner/QDesignerCustomWidgetInterface>
 #include <marble_export.h>
 
 class QObject;
diff -ur ../marble-4.12.0-orig/src/plugins/designer/marblewidget/MarbleWidgetPlugin.h ./src/plugins/designer/marblewidget/MarbleWidgetPlugin.h
--- ../marble-4.12.0-orig/src/plugins/designer/marblewidget/MarbleWidgetPlugin.h	2013-12-12 04:55:59.000000000 +0900
+++ ./src/plugins/designer/marblewidget/MarbleWidgetPlugin.h	2013-12-24 21:42:20.000000000 +0900
@@ -20,7 +20,7 @@
 
 // Workaround: moc on osx is unable to find this file, when prefix with QtDesigner/
 // moc also doesn't respect Q_OS_* macros, otherwise I could ifdef this.
-#include <QDesignerCustomWidgetInterface>
+#include <QtDesigner/QDesignerCustomWidgetInterface>
 
 class MarbleWidgetPlugin : public QObject, public QDesignerCustomWidgetInterface
 {
diff -ur ../marble-4.12.0-orig/src/plugins/designer/navigator/MarbleNavigatorPlugin.h ./src/plugins/designer/navigator/MarbleNavigatorPlugin.h
--- ../marble-4.12.0-orig/src/plugins/designer/navigator/MarbleNavigatorPlugin.h	2013-12-12 04:55:59.000000000 +0900
+++ ./src/plugins/designer/navigator/MarbleNavigatorPlugin.h	2013-12-24 21:42:05.000000000 +0900
@@ -20,7 +20,7 @@
 
 // Workaround: moc on osx is unable to find this file, when prefix with QtDesigner/
 // moc also doesn't respect Q_OS_* macros, otherwise I could ifdef this.
-#include <QDesignerCustomWidgetInterface>
+#include <QtDesigner/QDesignerCustomWidgetInterface>
 
 class MarbleNavigatorPlugin : public QObject, public QDesignerCustomWidgetInterface
 {
