Support DESTDIR.
https://github.com/usnistgov/SCTK/pull/30
--- src/acomp/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/acomp/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -19,7 +19,7 @@
 	@echo "   No tests defined for acomp.pl"
 
 install: all
-	cp acomp.pl $(bindir)
+	cp acomp.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/align2html/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/align2html/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -12,8 +12,8 @@
 	@echo "No Tests for align2html.pl"
 
 install: all
-	perl buildInstallVersion.pl > $(bindir)/align2html.pl
-	chmod +x $(bindir)/align2html.pl
+	perl buildInstallVersion.pl > $(DESTDIR)$(bindir)/align2html.pl
+	chmod +x $(DESTDIR)$(bindir)/align2html.pl
 
 buildPackFile:
 	buildPack.pl > packImageTarFile
--- src/asclite/core/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/asclite/core/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -75,7 +75,7 @@
 
 install: all
 	for p in $(PROGRAMS); do \
-		$(INSTALL) $$p $(bindir)/$$p; \
+		$(INSTALL) $$p $(DESTDIR)$(bindir)/$$p; \
 	done
 
 check:
--- src/chfilt/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/chfilt/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -32,7 +32,7 @@
 	@echo "   chfilt.pl tests successfully completed"
 
 install: all
-	cp chfilt.pl $(bindir)
+	cp chfilt.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/csrfilt/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/csrfilt/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -20,7 +20,7 @@
 	@echo "   csrfilt.sh successfully completed"
 
 install: all
-	cp csrfilt.sh $(bindir)
+	cp csrfilt.sh $(DESTDIR)$(bindir)
 
 clean:
 
--- src/ctmValidator/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/ctmValidator/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -11,7 +11,7 @@
 	(cd test_suite; $(SHELL) RunTest.sh ../ctmValidator.pl)
 
 install: all
-	cp ctmValidator.pl $(bindir)
+	cp ctmValidator.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/def_art/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/def_art/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -19,7 +19,7 @@
 	@echo "   def_art.pl passed without tests"
 
 install: all
-	cp def_art.pl $(bindir)
+	cp def_art.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/hamzaNorm/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/hamzaNorm/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -25,7 +25,7 @@
 	@echo "   hamzaNorm.pl tests successfully completed"
 
 install: all
-	cp hamzaNorm.pl $(bindir)
+	cp hamzaNorm.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/hubscr/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/hubscr/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -21,7 +21,7 @@
 	@echo "   hubscr.pl tests successfully completed"
 
 install: all
-	cp hubscr.pl $(bindir)
+	cp hubscr.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/md-eval/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/md-eval/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for md-eval.pl"
 
 install: all
-	cp md-eval.pl $(bindir)
+	cp md-eval.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/mergectm2rttm/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/mergectm2rttm/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for mergectm2rttm.pl"
 
 install: all
-	cp mergectm2rttm.pl $(bindir)
+	cp mergectm2rttm.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/rfilter1/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/rfilter1/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -31,7 +31,7 @@
 	@echo "   rfilter1 tests successfully completed"
 
 install: all
-	install rfilter1 $(bindir)
+	install rfilter1 $(DESTDIR)$(bindir)
 
 clean:
 	$(RM) rfilter1
--- src/rttm2ctm/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/rttm2ctm/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for rttm2ctm.pl"
 
 install: all
-	cp rttm2ctm.pl $(bindir)
+	cp rttm2ctm.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/rttmSmooth/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/rttmSmooth/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for rttmSmooth.pl"
 
 install: all
-	cp rttmSmooth.pl $(bindir)
+	cp rttmSmooth.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/rttmSort/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/rttmSort/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for rttmSort.pl"
 
 install: all
-	cp rttmSort.pl $(bindir)
+	cp rttmSort.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/rttmValidator/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/rttmValidator/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -11,7 +11,7 @@
 	(cd test_suite; $(SHELL) RunTest.sh ../rttmValidator.pl)
 
 install: all
-	cp rttmValidator.pl $(bindir)
+	cp rttmValidator.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/sclite/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/sclite/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -158,17 +158,17 @@
 
 install: all testinstalldirs
 	set -e; for p in $(PROGRAMS); do \
-	  $(INSTALL) $$p $(bindir)/$$p; \
+	  $(INSTALL) $$p $(DESTDIR)$(bindir)/$$p; \
 	done
 #REenable this later
 #	set -e; for p in `(cd ../doc ; ls *.1)`; do \
-#	  $(INSTALL) ../doc/$$p $(man1dir)/$$p; \
+#	  $(INSTALL) ../doc/$$p $(DESTDIR)$(man1dir)/$$p; \
 #	done
 #	@echo "!!!!  If 'man sclite' does not work, run 'catman -M $(mandir)'"
 #	@echo "!!!!  to make manual pages accessible."
 
 testinstalldirs:
-	@-set -e; for p in $(bindir) $(man1dir) $(man5dir); do \
+	@-set -e; for p in $(DESTDIR)$(bindir) $(DESTDIR)$(man1dir) $(DESTDIR)$(man5dir); do \
 	   if test ! -d $$p ; then \
 		echo "Error: directory $$p doesn't exist"; \
 	   fi ; \
--- src/slatreport/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/slatreport/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -24,7 +24,7 @@
 	@echo "   slatreport.pl tests successfully completed"
 
 install: all
-	cp slatreport.pl $(bindir)
+	cp slatreport.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/spkr2sad/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/spkr2sad/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for spkr2sad.pl"
 
 install: all
-	cp spkr2sad.pl $(bindir)
+	cp spkr2sad.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/stm2rttm/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/stm2rttm/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -15,7 +15,7 @@
 	@echo "No Tests for stm2rttm.pl"
 
 install: all
-	cp stm2rttm.pl $(bindir)
+	cp stm2rttm.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/stmValidator/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/stmValidator/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -13,8 +13,8 @@
 	(cd test_suite; $(SHELL) RunTest.sh ../stmValidator.pl)
 
 install: all
-	perl buildInstallVersion.pl > $(bindir)/stmValidator.pl
-	chmod +x $(bindir)/stmValidator.pl
+	perl buildInstallVersion.pl > $(DESTDIR)$(bindir)/stmValidator.pl
+	chmod +x $(DESTDIR)$(bindir)/stmValidator.pl
 
 STMList.pl:
 
--- src/tanweenFilt/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/tanweenFilt/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -31,7 +31,7 @@
 	@echo "   tanweenFilt.pl tests successfully completed"
 
 install: all
-	cp tanweenFilt.pl $(bindir)
+	cp tanweenFilt.pl $(DESTDIR)$(bindir)
 
 clean:
 
--- src/utf_filt/makefile.in.orig	2021-10-28 07:48:48.000000000 -0500
+++ src/utf_filt/makefile.in	2021-11-07 05:47:31.000000000 -0600
@@ -26,8 +26,8 @@
 	fi	
 
 install: all
-	@sed 's:"nsgmls":"'$(NSGMLS)'":' < utf_filt.pl > $(bindir)/utf_filt.pl
-	@chmod +x $(bindir)/utf_filt.pl
+	@sed 's:"nsgmls":"'$(NSGMLS)'":' < utf_filt.pl > $(DESTDIR)$(bindir)/utf_filt.pl
+	@chmod +x $(DESTDIR)$(bindir)/utf_filt.pl
 
 config.sh: config.in
 	autoconf config.in | sed 's/Makefile/makefile/g' > config.sh
