--- interfaces/cc/aspell.h.orig	2011-07-02 14:53:27.000000000 -0700
+++ interfaces/cc/aspell.h	2014-01-10 23:28:54.000000000 -0800
@@ -237,6 +237,7 @@ void delete_aspell_can_have_error(struct
 /******************************** errors ********************************/
 
 
+#ifndef __cplusplus
 extern const struct AspellErrorInfo * const aerror_other;
 extern const struct AspellErrorInfo * const aerror_operation_not_supported;
 extern const struct AspellErrorInfo * const   aerror_cant_copy;
@@ -322,6 +323,7 @@ extern const struct AspellErrorInfo * co
 extern const struct AspellErrorInfo * const   aerror_bad_magic;
 extern const struct AspellErrorInfo * const aerror_expression;
 extern const struct AspellErrorInfo * const   aerror_invalid_expression;
+#endif
 
 
 /******************************* speller *******************************/
