--- DESCRIPTION	2024-03-22 02:30:02.000000000 +0800
+++ DESCRIPTION	2024-03-23 02:31:43.000000000 +0800
@@ -21,7 +21,7 @@
         purrr (>= 1.0.0), recipes (>= 1.0.10), rlang (>= 1.1.3),
         rsample (>= 1.2.0), stats, tibble (>= 2.1.3), tidyr, tune (>=
         1.2.0), vctrs (>= 0.6.1), workflows (>= 1.1.4)
-Suggests: covr, h2o, kernlab, kknn, knitr, mockr, modeldata, nnet,
+Suggests: covr, kernlab, kknn, knitr, mockr, modeldata, nnet,
         ranger, rmarkdown, testthat (>= 3.0.0), workflowsets (>=
         0.1.0), yardstick (>= 1.1.0)
 VignetteBuilder: knitr
