--- ../kipi-plugins-1.8.0/CMakeLists.txt.orig	2010-11-23 14:08:39.000000000 +0100
+++ ../kipi-plugins-1.8.0/CMakeLists.txt	2010-11-23 14:09:41.000000000 +0100
@@ -128,10 +128,10 @@
 MACRO_OPTIONAL_FIND_PACKAGE(LibXml2)    # For Htmlexport.
 MACRO_OPTIONAL_FIND_PACKAGE(LibXslt)    # For Htmlexport.
 MACRO_OPTIONAL_FIND_PACKAGE(OpenGL)     # For AdvancedSlideshow and ImageViewer.
-MACRO_OPTIONAL_FIND_PACKAGE(Gpod)       # For ipodexport.
-MACRO_OPTIONAL_FIND_PACKAGE(Gdk)        # For ipodexport.
-MACRO_OPTIONAL_FIND_PACKAGE(GLIB2)      # For ipodexport.
-MACRO_OPTIONAL_FIND_PACKAGE(GObject)    # For ipodexport.
+#MACRO_OPTIONAL_FIND_PACKAGE(Gpod)       # For ipodexport.
+#MACRO_OPTIONAL_FIND_PACKAGE(Gdk)        # For ipodexport.
+#MACRO_OPTIONAL_FIND_PACKAGE(GLIB2)      # For ipodexport.
+#MACRO_OPTIONAL_FIND_PACKAGE(GObject)    # For ipodexport.
 MACRO_OPTIONAL_FIND_PACKAGE(KdepimLibs) # For Calendar (libkcal).
 MACRO_OPTIONAL_FIND_PACKAGE(QCA2)       # For Shwup.
 MACRO_OPTIONAL_FIND_PACKAGE(KSane)      # For AcquireImages.
