--- resources/bitmaps_png/CMakeLists.txt.orig	2026-02-16 07:43:57
+++ resources/bitmaps_png/CMakeLists.txt	2026-02-16 07:44:22
@@ -914,9 +914,6 @@
 endif()
 
 set( TARFLAGS cfz )
-if( DEFINED ENV{SOURCE_DATE_EPOCH} )
-    set( TARFLAGS --sort=name --mtime=@$ENV{SOURCE_DATE_EPOCH} --clamp-mtime --owner=0 --group=0 --numeric-owner --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime -czf )
-endif()
 
 # This is a dummy target to bind the custom command to something we can reuse in other calls
 # i.e. we want to dynamically make bitmap_archive depend on generate_pngs target after wards

