--- CMakeLists.txt
+++ CMakeLists.txt
@@ -5,7 +5,7 @@
 #================================================
 # Locations for install targets.
 #================================================
-if( APPLE )
+if( FALSE )
     # Like all variables, CMAKE_INSTALL_PREFIX can be over-ridden on the command line.
     set( CMAKE_INSTALL_PREFIX "/Library/Application Support/kicad/" CACHE PATH "" )
     # Everything without leading / is relative to CMAKE_INSTALL_PREFIX.
