diff --git common/gestfich.cpp common/gestfich.cpp
index cd2d727474..e114b44042 100644
--- common/gestfich.cpp
+++ common/gestfich.cpp
@@ -91,6 +91,7 @@ wxString FindKicadFile( const wxString& shortname )
         wxT( "/usr/bin/" ),
         wxT( "/usr/local/bin/" ),
         wxT( "/usr/local/kicad/bin/" ),
+        wxT( "@PREFIX_BIN@/" ),
 #endif
     };
 
diff --git eeschema/sim/ngspice.cpp eeschema/sim/ngspice.cpp
index 0a3a07ed4b..fe566c9e81 100644
--- eeschema/sim/ngspice.cpp
+++ eeschema/sim/ngspice.cpp
@@ -456,7 +456,7 @@
                 "/../../../../../Contents/PlugIns/sim"
     };
 #else   // Unix systems
-    const std::vector<std::string> dllPaths = { "/usr/local/lib" };
+    const std::vector<std::string> dllPaths = { "@PREFIX_DIR@/lib" };
 #endif

 #if defined(__WINDOWS__) || (__WXMAC__)
