diff --git mesonbuild/compilers/mixins/clang.py.orig mesonbuild/compilers/mixins/clang.py
index acdb352..0a3e879 100644
--- mesonbuild/compilers/mixins/clang.py.orig
+++ mesonbuild/compilers/mixins/clang.py
@@ -81,7 +81,7 @@ class ClangCompiler(GnuLikeCompiler):
         myargs = []  # type: T.List[str]
         if mode is CompileCheckMode.COMPILE:
             myargs.extend(['-Werror=unknown-warning-option', '-Werror=unused-command-line-argument'])
-            if mesonlib.version_compare(self.version, '>=3.6.0'):
+            if mesonlib.version_compare(self.version, '>=9.6.0'):
                 myargs.append('-Werror=ignored-optimization-argument')
         return super().get_compiler_check_args(mode) + myargs
 
