--- mesonbuild/compilers/mixins/clike.py.orig	2024-09-20 13:16:26.000000000 -0500
+++ mesonbuild/compilers/mixins/clike.py	2024-09-28 13:52:12.000000000 -0500
@@ -1185,9 +1185,6 @@
         unless you select a particular macOS SDK with the -isysroot flag.
         You can also add to this by setting -F in CFLAGS.
         '''
-        # TODO: this really needs to be *AppleClang*, not just any clang.
-        if self.id != 'clang':
-            raise mesonlib.MesonException('Cannot find framework path with non-clang compiler')
         # Construct the compiler command-line
         commands = self.get_exelist(ccache=False) + ['-v', '-E', '-']
         commands += self.get_always_args()
