--- mesonbuild/modules/gnome.py.orig	2026-04-13 10:04:57
+++ mesonbuild/modules/gnome.py	2026-04-15 10:50:47
@@ -995,8 +995,9 @@
         # and Windows rules to get the first argument, which breaks if there are
         # backslashes, so best avoid them entirely.
         cc_exelist = [pathlib.Path(exe).as_posix() for exe in cc_exelist]
-        run_env.set('CC', [quote_arg(x) for x in cc_exelist], ' ')
-        run_env.set('CFLAGS', [quote_arg(x) for x in env_flags], ' ')
+        # MACPORTS MOD -- do not set CC and CFLAGS here. MacPorts sets it directly, with modifications
+        #run_env.set('CC', [quote_arg(x) for x in cc_exelist], ' ')
+        #run_env.set('CFLAGS', [quote_arg(x) for x in env_flags], ' ')
         run_env.merge(kwargs['env'])
 
         # response file supported?
