--- common/umutex.h.orig	2013-01-18 01:34:14.000000000 -0800
+++ common/umutex.h	2013-01-18 01:34:38.000000000 -0800
@@ -27,16 +27,8 @@
 #endif
 
 #if U_PLATFORM_IS_DARWIN_BASED
-#if defined(__STRICT_ANSI__)
-#define UPRV_REMAP_INLINE
-#define inline
-#endif
 #include <libkern/OSAtomic.h>
 #define USE_MAC_OS_ATOMIC_INCREMENT 1
-#if defined(UPRV_REMAP_INLINE)
-#undef inline
-#undef UPRV_REMAP_INLINE
-#endif
 #endif
 
 /*
