https://github.com/unicode-org/icu/pull/2239

--- tools/toolutil/writesrc.cpp.orig	2022-10-19 11:53:21.000000000 +1100
+++ tools/toolutil/writesrc.cpp	2022-11-06 22:32:29.000000000 +1100
@@ -19,7 +19,11 @@
 */
 
 #include <stdio.h>
-#include <inttypes.h>
+// cinttypes includes inttypes.h internally for the format macros
+// (PRId64 et al.) On some systems, that follows C99 which says those
+// macros should be hidden to C++ without this define.
+#define __STDC_FORMAT_MACROS 1
+#include <cinttypes>
 #include <time.h>
 #include "unicode/utypes.h"
 #include "unicode/putil.h"
