--- config/mh-darwin.orig	2011-10-14 04:38:41.000000000 -0700
+++ config/mh-darwin	2011-10-14 05:32:55.000000000 -0700
@@ -28,7 +28,7 @@
 ifeq ($(ENABLE_RPATH),YES)
 LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name $(libdir)/$(notdir $(MIDDLE_SO_TARGET))
 else
-LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name $(notdir $(MIDDLE_SO_TARGET))
+LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name __PREFIX__/lib/$(notdir $(MIDDLE_SO_TARGET))
 endif
 
 ## Compiler switch to embed a runtime search path
