--- Makefile.in.orig	2015-03-22 17:32:54.000000000 -0500
+++ Makefile.in	2015-03-22 17:33:14.000000000 -0500
@@ -335,7 +335,6 @@
 	LC_ALL=C sed -f $(top_srcdir)/config/make2sh.sed < $(top_builddir)/config/Makefile.inc | grep -v '#M#' | uniq >> $@
 	LC_ALL=C sed -f $(top_srcdir)/config/make2sh.sed < @platform_make_fragment@ | grep -v '#M#' | uniq >> $@
 	cat $(top_srcdir)/config/icu-config-bottom >> $@
-	echo "# Rebuilt on "`date` >> $@
 	chmod u-w $@
 
 config.status: $(srcdir)/configure $(srcdir)/common/unicode/uvernum.h
