--- makefile.orig	2017-01-18 11:47:58.000000000 +0000
+++ makefile	2017-05-17 07:21:45.000000000 +0000
@@ -144,9 +144,9 @@
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so
+	@rm -f libunrar.dylib
 	@rm -f libunrar.a
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -dynamiclib -o libunrar.dylib -install_name $(DESTDIR)/lib/libunrar.dylib $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
