--- makefile.orig	2024-08-19 11:26:32.000000000 +0300
+++ makefile	2024-11-12 07:15:32.300385000 +0300
@@ -22,7 +22,7 @@
 WHAT=UNRAR
 
 UNRAR_OBJ=filestr.o recvol.o rs.o scantree.o qopen.o
-LIB_OBJ=filestr.o scantree.o dll.o qopen.o
+LIB_OBJ=filestr.o recvol.o rs.o scantree.o dll.o qopen.o
 
 OBJECTS=rar.o strlist.o strfn.o pathfn.o smallfn.o global.o file.o filefn.o filcreat.o \
 	archive.o arcread.o unicode.o system.o crypt.o crc.o rawread.o encname.o \
