--- makefile.orig	2020-08-23 23:52:29.000000000 -0400
+++ makefile	2020-08-23 23:55:50.000000000 -0400
@@ -157,7 +157,7 @@
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
 	@rm -f libunrar.*
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -dynamiclib -o libunrar.dylib -install_name $(DESTDIR)/lib/libunrar.dylib $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
@@ -167,8 +167,8 @@
 			rm -f $(DESTDIR)/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.dylib $(DESTDIR)/lib
 		install libunrar.a $(DESTDIR)/lib
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/lib/libunrar.dylib
