--- consio.cpp.orig
+++ consio.cpp
@@ -1,5 +1,6 @@
 #include "rar.hpp"
 #include "log.cpp"
+#include "consio.hpp"
 
 static MESSAGE_TYPE MsgStream=MSG_STDOUT;
 static RAR_CHARSET RedirectCharset=RCH_DEFAULT;
--- list.cpp.orig
+++ list.cpp
@@ -1,4 +1,5 @@
 #include "rar.hpp"
+#include "consio.hpp"
 
 static void ListFileHeader(Archive &Arc,FileHeader &hd,bool &TitleShown,bool Verbose,bool Technical,bool Bare);
 static void ListSymLink(Archive &Arc);
--- volume.cpp.orig
+++ volume.cpp
@@ -1,4 +1,5 @@
 #include "rar.hpp"
+#include "consio.hpp"
 
 #ifdef RARDLL
 static bool DllVolChange(RAROptions *Cmd,wchar *NextName,size_t NameSize);
--- cmddata.cpp.orig
+++ cmddata.cpp
@@ -1,4 +1,5 @@
 #include "rar.hpp"
+#include "consio.hpp"
 
 CommandData::CommandData()
 {
--- recvol3.cpp.orig
+++ recvol3.cpp
@@ -1,4 +1,5 @@
 #include "rar.hpp"
+#include "consio.hpp"
 
 // Buffer size for all volumes involved.
 static const size_t TotalBufferSize=0x4000000;
--- recvol5.cpp.orig
+++ recvol5.cpp
@@ -1,4 +1,5 @@
 #include "rar.hpp"
+#include "consio.hpp"
 
 static const uint MaxVolumes=65535;
 
