--- configure.orig	2023-03-08 07:16:14.000000000 +0800
+++ configure	2023-04-02 08:58:48.000000000 +0800
@@ -47587,8 +47587,8 @@
 done
 
 if test "x${have_libcurl}" = "xyes"; then
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if libcurl is version 7 and >= 7.28.0" >&5
-printf %s "checking if libcurl is version 7 and >= 7.28.0... " >&6; }
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if libcurl is >= 7.28.0" >&5
+printf %s "checking if libcurl is >= 7.28.0... " >&6; }
 if test ${r_cv_have_curl728+y}
 then :
   printf %s "(cached) " >&6
@@ -47606,7 +47606,7 @@
 {
 #ifdef LIBCURL_VERSION_MAJOR
 #if LIBCURL_VERSION_MAJOR > 7
-  exit(1);
+  exit(0);
 #elif LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR >= 28
   exit(0);
 #else
