--- configure.orig	2016-10-24 06:34:26.000000000 -0500
+++ configure	2017-01-12 02:28:45.000000000 -0600
@@ -35508,6 +35508,7 @@
 #include <zlib.h>
 int main() {
 #ifdef ZLIB_VERSION
+  if(strlen(ZLIB_VERSION) == 6) exit(strncmp(ZLIB_VERSION, "1.2.05", 6) < 0);
 /* Work around Debian bug: it uses 1.2.3.4 even though there was no such
    version on the master site zlib.net */
   exit(strncmp(ZLIB_VERSION, "1.2.5", 5) < 0);
