diff --git Makefile Makefile
index 217b4c2..ae6dff0 100644
--- Makefile
+++ Makefile
@@ -45,12 +45,12 @@ mpicxx=mpicc
 
 # for 32 bit machines
 
-# BITS=
-# MPLRSOBJ2=
+BITS=
+MPLRSOBJ2=
 
 # for 64 bit machines
-BITS=-DB128
-MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
+# BITS=-DB128
+# MPLRSOBJ2=lrslib2-mplrs.o lrslong2-mplrs.o
 
 
 LRSOBJ=lrs.o lrslong1.o lrslong2.o lrslib1.o lrslib2.o lrslibgmp.o lrsgmp.o lrsdriver.o
@@ -216,11 +216,11 @@ SOMINOR ?=.0.0
 SHLIB ?=$(SONAME)$(SOMINOR)
 SHLINK ?=liblrs.so
 
-SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
+# SHLIBOBJ2=lrslib2-shr.o lrslong2-shr.o
 
 # for 32 bit machines
 
-# SHLIBOBJ2=
+SHLIBOBJ2=
 
 SHLIBOBJ=lrslong1-shr.o lrslib1-shr.o  \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o \
