--- tests_and_examples/install.tst/inst_tst.sh.orig	Sat Sep 16 22:32:18 2006
+++ tests_and_examples/install.tst/inst_tst.sh	Sun Sep 17 00:25:34 2006
@@ -1,5 +1,4 @@
 #!/bin/sh 
-# install test procedures
 
 CVER="../../bin/cver -q " 
 
@@ -7,90 +6,162 @@
 then
   echo "ERROR - there is no cver in ../../bin/"
   echo "Must make in src/ directory" 
-  exit;
+  exit 1
 fi
 
 $CVER instid.v >/dev/null
 ./rmlic.pl
-diff verilog.log instid.plg
+cmp -s verilog.log instid.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER instpnd3.v >/dev/null
 ./rmlic.pl
-diff verilog.log instpnd3.plg
+cmp -s verilog.log instpnd3.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER aspike1.v >/dev/null
 ./rmlic.pl
-diff verilog.log aspike1.plg
+cmp -s verilog.log aspike1.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER +warn_canceled_e aspike1.v >/dev/null
 ./rmlic.pl
-diff verilog.log aspike1a.plg
+cmp -s verilog.log aspike1a.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER +show_canceled_e aspike1.v >/dev/null
 ./rmlic.pl
-diff verilog.log aspike1b.plg
+cmp -s verilog.log aspike1b.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER +show_canceled_e +pulse_e_style_ondetect aspike1.v >/dev/null
 ./rmlic.pl
-diff verilog.log aspike1c.plg
+cmp -s verilog.log aspike1c.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER aspike1.v -et >/dev/null
 ./rmlic.pl
-diff verilog.log aspike1d.plg
+cmp -s verilog.log aspike1d.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER +warn_canceled_e -v JK_Q.v -v JK_QBAR.v udpjkff.v >/dev/null
 ./rmlic.pl
-diff verilog.log udpjkff.plg
+cmp -s verilog.log udpjkff.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER xx2bdel.v >/dev/null
 ./rmlic.pl
-diff verilog.log xx2bdel.plg
+cmp -s verilog.log xx2bdel.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER xx2bpth.v >/dev/null
 ./rmlic.pl
-diff verilog.log xx2bpth.plg
+cmp -s verilog.log xx2bpth.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 $CVER xx2bpth2.v >/dev/null
 ./rmlic.pl
-diff verilog.log xx2bpth2.plg
+cmp -s verilog.log xx2bpth2.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER c880.v >/dev/null
 ./rmlic.pl
-diff verilog.log c880.plg
+cmp -s verilog.log c880.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER -i force01.inp force01.v >/dev/null
 ./rmlic.pl
-diff verilog.log force01.plg
+cmp -s verilog.log force01.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER smrd04.v -f smrd04.vc >/dev/null
 ./rmlic.pl
-diff verilog.log smrd04.plg
+cmp -s verilog.log smrd04.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER gatenots.v >/dev/null
 ./rmlic.pl
-diff verilog.log gatenots.plg
+cmp -s verilog.log gatenots.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER arms_sim.v armscnt.v >/dev/null
 ./rmlic.pl
-diff verilog.log arms.plg
+cmp -s verilog.log arms.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER -f dfpsetd.vc >/dev/null
 ./rmlic.pl
-diff verilog.log dfpsetd.plg
+cmp -s verilog.log dfpsetd.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER -f mipdnot1.vc >/dev/null
 ./rmlic.pl
-diff verilog.log mipdnot1.plg
+cmp -s verilog.log mipdnot1.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER -f sdfia04.vc >/dev/null
 ./rmlic.pl
-diff verilog.log sdfia04.plg
+cmp -s verilog.log sdfia04.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER dffn.v >/dev/null
 ./rmlic.pl
-diff verilog.log dffn.plg
+cmp -s verilog.log dffn.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER xplipnd.v >/dev/null
 ./rmlic.pl
-diff verilog.log xplipnd.plg
+cmp -s verilog.log xplipnd.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
 
 $CVER defsplt1.v >/dev/null
 ./rmlic.pl
-diff verilog.log defsplt1.plg
+cmp -s verilog.log defsplt1.plg
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi    
+
 
-echo ">>>> Install test completed (this should be only message printed)."
-echo " "
+exit 0
