--- gtk2/test/run-test.rb.orig	2010-09-25 17:26:22.000000000 +0900
+++ gtk2/test/run-test.rb	2010-09-28 00:03:50.000000000 +0900
@@ -14,9 +14,9 @@
  [pango_base, "pangp"],
  [gdk_pixbuf_base, "gdk_pixbuf2"],
  [gtk_base, "gtk2"]].each do |target, module_name|
-  if system("which make > /dev/null")
-    `make -C #{target.dump} > /dev/null` or exit(false)
-  end
+#  if system("which make > /dev/null")
+#    `make -C #{target.dump} > /dev/null` or exit(false)
+#  end
   $LOAD_PATH.unshift(File.join(target, "ext", module_name))
   $LOAD_PATH.unshift(File.join(target, "lib"))
 end
