--- Makefile.orig	2010-12-24 05:32:56.000000000 +0100
+++ Makefile	2010-12-24 05:33:39.000000000 +0100
@@ -76,8 +76,10 @@
 	@rm -f $(PROGNAME) $(OBJ)
 
 install:
-	install -Dp -m0755 $(PROGNAME) $(DESTDIR)$(BINDIR)/$(PROGNAME)
-	install -Dp -m0644 $(PROGNAME).1 $(DESTDIR)$(MANDIR)/man1/$(PROGNAME).1
+	install -m0755 $(PROGNAME) $(DESTDIR)$(BINDIR)/$(PROGNAME)
+	install -m0644 $(PROGNAME).1 $(DESTDIR)$(MANDIR)/man1/$(PROGNAME).1
 
 .c.o:
 	$(CC) $(CFLAGS) $(OPTFLAGS) -c -o $@ $<
+
+.PHONY: clean install
