--- source/creator/CMakeLists.txt.orig	2021-02-13 12:39:38.000000000 -0500
+++ source/creator/CMakeLists.txt	2021-02-13 13:02:10.000000000 -0500
@@ -924,6 +924,21 @@
   if(NOT WITH_PYTHON_MODULE)
     # Uppercase name for app bundle
     set_target_properties(blender PROPERTIES OUTPUT_NAME Blender)
+
+    if(WITH_DOC_MANPAGE)
+      add_custom_target(
+        blender_man_page ALL
+        COMMAND ${CMAKE_SOURCE_DIR}/doc/manpage/blender.1.py
+                ${EXECUTABLE_OUTPUT_PATH}/Blender.app/Contents/MacOS/Blender
+                ${CMAKE_CURRENT_BINARY_DIR}/blender.1)
+      add_dependencies(blender_man_page blender)
+
+      # manpage only with 'blender' binary
+      install(
+        FILES ${CMAKE_CURRENT_BINARY_DIR}/blender.1
+        DESTINATION @@PREFIX@@/share/man/man1
+      )
+    endif()
   endif()
 
   # handy install macro to exclude files, we use \$ escape for the "to"
