--- source/creator/CMakeLists.txt.orig	2020-09-03 10:32:45.000000000 -0400
+++ source/creator/CMakeLists.txt	2020-09-03 10:33:07.000000000 -0400
@@ -76,9 +76,28 @@
 
 if(WITH_TBB)
   blender_include_dirs(${TBB_INCLUDE_DIRS})
-  link_directories(${LIBDIR}/tbb/lib)
+  link_directories(${LIBDIR}/lib)
 endif()
 
+if(WITH_LZO)
+  if(WITH_SYSTEM_LZO)
+    list(APPEND INC_SYS
+      ${LZO_INCLUDE_DIR}
+    )
+    list(APPEND LIB
+      ${LZO_LIBRARIES}
+    )
+    add_definitions(-DWITH_SYSTEM_LZO)
+  else()
+    list(APPEND INC_SYS
+      ../../extern/lzo/minilzo
+    )
+    list(APPEND LIB
+      extern_minilzo
+    )
+  endif()
+  add_definitions(-DWITH_LZO)
+endif()
 
 if(WITH_PYTHON)
   blender_include_dirs(../blender/python)
@@ -949,14 +968,14 @@
 
   if(WITH_OPENMP AND OPENMP_CUSTOM)
     install(
-      FILES ${LIBDIR}/openmp/lib/libomp.dylib
+      FILES ${LIBDIR}/lib/libomp/libomp.dylib
       DESTINATION Blender.app/Contents/Resources/lib
     )
   endif()
 
   if(WITH_LLVM AND NOT LLVM_STATIC)
     install(
-      FILES ${LIBDIR}/llvm/lib/libLLVM-3.4.dylib
+      FILES ${LIBDIR}/libexec/llvm-3.7/lib/libLLVM-3.7.dylib
       DESTINATION Blender.app/Contents/MacOS
     )
   endif()
