--- src/__init__.py.orig	2017-11-27 07:22:09.000000000 -0600
+++ src/__init__.py	2017-11-27 07:22:28.000000000 -0600
@@ -1,6 +1,6 @@
 __version__ = "1.3"
 __author__ = "phpsh@googlegroups.com"
-__date__ = "Nov 20, 2008"
+__date__ = "May 14, 2014"
 
 from subprocess import Popen, PIPE, call
 from threading import Thread
@@ -201,7 +201,7 @@
         self.partial = ""
 
     def check_syntax(self, line):
-        p = Popen(["php", "-r", "return;" + line], stdout=PIPE, stderr=PIPE)
+        p = Popen(["@@PHP_EXECUTABLE@@", "-r", "return;" + line], stdout=PIPE, stderr=PIPE)
         p.wait()
         # "php -r" lint errors seem to only use stdout, but it might (idk)
         # depend on configuration or change later, so just grab everything.
@@ -398,7 +398,7 @@
                 else:
                     self.xdebug_path = xdebug
 
-        self.comm_base = ["php"]
+        self.comm_base = ["@@PHP_EXECUTABLE@@"]
 
         if self.with_xdebug:
             xdebug_comm_base = self.comm_base[:]
