--- core/utilities/cameragui/devices/gpcamera.cpp.orig	2012-08-26 19:24:51.000000000 +0200
+++ core/utilities/cameragui/devices/gpcamera.cpp	2012-08-27 00:06:15.000000000 +0200
@@ -1331,6 +1331,8 @@
 
     errorCode = gp_camera_folder_put_file(d->camera,
                                           QFile::encodeName(folder).constData(),
+                                          QFile::encodeName(itemName).constData(),
+                                          GP_FILE_TYPE_NORMAL,
                                           cfile,
                                           d->status->context);
 
@@ -1651,7 +1653,9 @@
         for (int i = 0 ; i < numPorts ; ++i)
         {
             gp_port_info_list_get_info(list, i, &info);
-            plist.append(info.path);
+            char *path;
+            gp_port_info_get_path(info, &path);
+            plist.append(path);
         }
     }
 
