https://bugs.kde.org/show_bug.cgi?id=289330
--- core/digikam/album/albumwatch.cpp.orig	2012-03-23 20:43:15.000000000 +0100
+++ core/digikam/album/albumwatch.cpp	2012-03-23 20:43:22.000000000 +0100
@@ -215,7 +215,7 @@
         if (!d->dirWatch->contains(dir))
         {
             d->dirWatchAddedDirs << dir;
-            d->dirWatch->addDir(dir, KDirWatch::WatchFiles | KDirWatch::WatchDirOnly);
+            d->dirWatch->addDir(dir, KDirWatch::WatchDirOnly);
         }
     }
 }
