--- core/libs/dimg/filters/lens/lensfunfilter.cpp.orig	2015-03-06 12:52:39.000000000 -0800
+++ core/libs/dimg/filters/lens/lensfunfilter.cpp	2015-03-06 13:19:33.000000000 -0800
@@ -123,11 +123,6 @@
         modifyFlags |= LF_MODIFY_VIGNETTING;
     }
 
-    if (d->iface->settings().filterCCI)
-    {
-        modifyFlags |= LF_MODIFY_CCI;
-    }
-
     // Init lensfun lib, we are working on the full image.
 
     lfPixelFormat colorDepth = m_orgImage.bytesDepth() == 4 ? LF_PF_U8 : LF_PF_U16;
