--- a/configure.ac	2019-03-04 03:58:11.000000000 +0800
+++ b/configure.ac	2023-08-15 04:33:00.000000000 +0800
@@ -125,9 +125,15 @@
 			TIGER_LAUNCHD=yes
 		;;
 	esac
+	case $host in
+		powerpc*-*-darwin*)
+		;;
+		*)
         AC_CHECK_FUNC(dispatch_async,
                               AC_DEFINE([HAVE_LIBDISPATCH], 1, [Define to 1 if you have the libdispatch (GCD) available]),
                               [])
+        ;;
+    esac
 fi
 
 AC_DEFINE_UNQUOTED(BUNDLE_ID_PREFIX, "$bundleidprefix", [Prefix to use for launchd identifiers])
