--- a/configure.ac.orig	2010-08-24 11:34:21.000000000 -0500
+++ b/configure.ac	2013-01-06 22:57:08.000000000 -0600
@@ -5,7 +5,7 @@
 
 AC_PREREQ([2.54])
 AC_INIT([mutt.h])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 
 mutt_cv_version=`cat $srcdir/VERSION`
 AM_INIT_AUTOMAKE(mutt, $mutt_cv_version)
@@ -27,7 +27,6 @@
 
 AC_PROG_CC
 AC_ISC_POSIX
-AM_C_PROTOTYPES
 if test "x$U" != "x"; then
   AC_MSG_ERROR(Compiler not ANSI compliant)
 fi
