--- a/crypt.c	2009-01-05 11:20:53.000000000 -0800
+++ b/crypt.c	2009-06-29 16:55:12.000000000 -0700
@@ -393,7 +393,14 @@
        */
       if (!ascii_strcasecmp (m->description, "S/MIME Encrypted Message"))
 	return SMIMEENCRYPT;
-      complain = 1;
+
+      /* Thunderbird 2.0.0.19 doesn't seem to be adding a Content-Description
+       * If we make it all the way here, we know it's not signed as Thunderbird
+       * sets the signed-data correctly, so we're going to assuming it's
+       * encrypted
+       */
+
+      return SMIMEENCRYPT;
     }
     else if (ascii_strcasecmp (m->subtype, "octet-stream"))
       return 0;
