--- config/SundialsLapack.cmake.orig	2017-11-07 16:47:45.000000000 -0700
+++ config/SundialsLapack.cmake	2017-11-24 08:06:15.000000000 -0700
@@ -19,15 +19,15 @@
 # a working Fortran compiler) or look for them in the most
 # obvious place...
 if(NOT LAPACK_LIBRARIES)
-  if(F77_FOUND)
+#  if(F77_FOUND)
     include(FindLAPACK)
-  else(F77_FOUND)
-    find_library(LAPACK_LIBRARIES
-      NAMES lapack
-      PATHS /usr/lib /usr/local/lib
-      "$ENV{ProgramFiles}/LAPACK/Lib"
-      )
-  endif(F77_FOUND)
+#  else(F77_FOUND)
+#    find_library(LAPACK_LIBRARIES
+#      NAMES lapack
+#      PATHS /usr/lib /usr/local/lib
+#      "$ENV{ProgramFiles}/LAPACK/Lib"
+#      )
+#  endif(F77_FOUND)
 
   # If the xSDK flag is used, set it to what was found
   if(LAPACK_LIBRARIES AND TPL_ENABLE_LAPACK)
