--- configure.orig	2007-02-19 22:01:46.000000000 -0500
+++ configure	2007-02-19 22:10:26.000000000 -0500
@@ -10769,7 +10769,7 @@
     if test "$?" -gt "0"; then
         with_python="no"
     else
-        checkPYTHON_LIBS=`python -c "import distutils.sysconfig ; print '%s/%s %s' % (distutils.sysconfig.get_config_var('LIBPL'),distutils.sysconfig.get_config_var('LDLIBRARY'),distutils.sysconfig.get_config_var('LIBS'))" 2>/dev/null`
+        checkPYTHON_LIBS="-lpython"
         if test "$?" -gt "0"; then
             with_python="no"
         else
