--- client/Makefile.in.orig	2011-03-08 02:31:49.000000000 -0600
+++ client/Makefile.in	2011-09-15 07:28:03.000000000 -0500
@@ -243,7 +243,7 @@
 display.c main.c win32.ico
 
 lbreakout2_DEPENDENCIES = @win32_deps@
-lbreakout2_LDADD = @win32_deps@ ../common/libcommon.a ../gui/libGui.a ../game/libgame.a @mixer_flag@ @sdlnet_lib_flag@
+lbreakout2_LDADD = @win32_deps@ ../common/libcommon.a ../gui/libGui.a ../game/libgame.a @mixer_flag@ @sdlnet_lib_flag@ -lintl
 EXTRA_DIST = credit.h shine.h extras.h balls.h shrapnells.h  \
 shots.h event.h paddle.h frame.h misc.h bricks.h player.h \
 file.h config.h lbreakout.h setselect.h \
