--- bootstrap.orig	2013-01-15 08:43:05.000000000 -0600
+++ bootstrap	2013-01-15 08:42:39.000000000 -0600
@@ -84,9 +84,9 @@
 }
 
 LIBTOOLIZE=`find_program glibtoolize libtoolize`
-ACLOCAL=`find_program aclocal-1.11 aclocal-1.10 aclocal-1.9 aclocal`
+ACLOCAL=`find_program aclocal`
 AUTOHEADER=`find_program autoheader`
-AUTOMAKE=`find_program automake-1.11 automake-1.10 automake-1.9 automake`
+AUTOMAKE=`find_program automake`
 AUTOCONF=`find_program autoconf`
 
 get_aclocal_dir_list () {
