--- Lib/GSL/SF/ellint/gsl_sf_ellint.pd.orig	2015-11-13 00:10:37.000000000 -0800
+++ Lib/GSL/SF/ellint/gsl_sf_ellint.pd	2015-11-13 00:13:11.000000000 -0800
@@ -84,15 +84,15 @@
 
 pp_def('gsl_sf_ellint_D',
        GenericTypes => [D],
-       Pars=>'double phi(); double k(); double n();
+       Pars=>'double phi(); double k();
               double [o]y(); double [o]e()',
        Code =>'
 gsl_sf_result r;
-GSLERR(gsl_sf_ellint_D_e,($phi(),$k(),$n(),GSL_PREC_DOUBLE,&r))
+GSLERR(gsl_sf_ellint_D_e,($phi(),$k(),GSL_PREC_DOUBLE,&r))
 $y() = r.val;
 $e() = r.err; 
 ',
-       Doc =>'Legendre form of incomplete elliptic integrals D(phi,k,n)'
+       Doc =>'Legendre form of incomplete elliptic integrals D(phi,k)'
       );
 
 pp_def('gsl_sf_ellint_RC',
