--- src/root/port.c.orig	2012-07-30 05:15:04.000000000 +0900
+++ src/root/port.c	2012-08-28 16:36:46.000000000 +0900
@@ -377,7 +377,7 @@
 int Port::isNan(double r)
 {
 #if __APPLE__
-    return __inline_isnan(r);
+    return isnan(r);
 #elif __OpenBSD__
     return isnan(r);
 #else
@@ -389,7 +389,7 @@
 int Port::isNan(longdouble r)
 {
 #if __APPLE__
-    return __inline_isnan(r);
+    return isnan(r);
 #elif __OpenBSD__
     return isnan(r);
 #else
