--- mkpasswd.c.orig	2013-04-10 10:25:05.000000000 -0500
+++ mkpasswd.c	2013-04-18 17:06:25.000000000 -0500
@@ -433,7 +433,7 @@
 "If no SALT is specified, a random one is generated.\n"
 "If TYPE is 'help', available methods are printed.\n"
 "\n"
-"Report bugs to %s.\n"), "<md+whois@linux.it>");
+"Report bugs to %s.\n"), "<https://trac.macports.org/newticket>");
     exit(error);
 }
 
--- whois.c.orig	2013-04-10 10:26:27.000000000 -0500
+++ whois.c	2013-04-18 17:21:01.000000000 -0500
@@ -146,7 +146,7 @@
 	    break;
 	case 1:
 	    fprintf(stdout, _("Version %s.\n\nReport bugs to %s.\n"),
-		    VERSION, "<md+whois@linux.it>");
+		    VERSION, "<https://trac.macports.org/newticket>");
 	    exit(EXIT_SUCCESS);
 	default:
 	    usage(EXIT_FAILURE);
