--- Makefile	2008-05-06 04:41:16.000000000 -0500
+++ Makefile	2008-11-12 00:53:51.000000000 -0600
@@ -31,7 +31,7 @@
 mkpasswd_LDADD += -lcrypt
 endif
 
-all: Makefile.depend whois mkpasswd #pos
+all: Makefile.depend whois mkpasswd pos
 
 whois_OBJECTS := whois.o utils.o
 mkpasswd_OBJECTS := mkpasswd.o utils.o
--- po/Makefile	2008-11-12 00:57:45.000000000 -0600
+++ po/Makefile	2008-11-12 00:58:46.000000000 -0600
@@ -40,7 +40,7 @@
 distclean: clean
 	rm -f whois.pot
 
-install: $(CATALOGS)
+install-pos: $(CATALOGS)
 	for n in $(CATALOGS); do \
 	    l=`basename $$n .mo`; \
 	    $(INSTALL) -m 755 -d $(INSTALLNLSDIR)/$$l; \
