--- src/plugins/position/corelocation/qgeopositioninfosource_cl.mm.orig	2023-03-12 03:16:54.000000000 +0100
+++ src/plugins/position/corelocation/qgeopositioninfosource_cl.mm	2023-12-30 18:29:00.000000000 +0200
@@ -52,12 +52,14 @@
 #ifndef Q_OS_TVOS
     if (newLocation.course >= 0) {
         location.setAttribute(QGeoPositionInfo::Direction, newLocation.course);
+#if !defined(Q_OS_MACOS) || MAC_OS_X_VERSION_MAX_ALLOWED >= 101500
         if (__builtin_available(iOS 13.4, watchOS 6.2, macOS 10.15.4, *)) {
             if (newLocation.courseAccuracy >= 0) {
                 location.setAttribute(QGeoPositionInfo::DirectionAccuracy,
                                       newLocation.courseAccuracy);
             }
         }
+#endif
     }
     if (newLocation.speed >= 0)
         location.setAttribute(QGeoPositionInfo::GroundSpeed, newLocation.speed);
