--- archive.c.orig	2012-08-03 00:00:00.000000000 +0900
+++ archive.c	2012-08-03 10:59:02.000000000 +0900
@@ -213,12 +213,14 @@
 	{"*.tar.Z",	"zcat %C|tar tvf -",	PM_TAR, 0},
 	{"*.tar.gz",	"gzip -cd %C|tar tvf -",	PM_TAR, 0},
 	{"*.tar.bz2",	"bzip2 -cd %C|tar tvf -",	PM_TAR, 0},
+	{"*.tar.lzma",	"lzcat %C|tar tvf -",	PM_TAR, 0},
 	{"*.tar.xz",	"xz -cd %C|tar tvf -",	PM_TAR, 0},
 # ifndef	CYGWIN
 	{"*.taZ",	"zcat %C|tar tvf -",	PM_TAR, 0},
 # endif
 	{"*.taz",	"gzip -cd %C|tar tvf -",	PM_TAR, 0},
 	{"*.tgz",	"gzip -cd %C|tar tvf -",	PM_TAR, 0},
+	{"*.tlz",	"lzcat %C|tar tvf -",	PM_TAR, 0},
 	{"*.txz",	"xz -cd %C|tar tvf -",	PM_TAR, 0},
 #endif	/* !MSDOS */
 	{NULL,		NULL,			PM_NULL, 0}
@@ -255,6 +257,8 @@
 					"gzip -cd %C|tar xf - %TA", 0},
 	{"*.tar.bz2",	"tar cf - %T|bzip2 -c > %C",
 					"bzip2 -cd %C|tar xf - %TA", 0},
+	{"*.tar.lzma",	"tar cf - %T|lzma -c > %C",
+					"lzcat %C|tar xf - %TA", 0},
 	{"*.tar.xz",	"tar cf - %T|xz -c > %C",
 					"xz -cd %C|tar xf - %TA", 0},
 # ifndef	CYGWIN
@@ -265,6 +269,8 @@
 					"gzip -cd %C|tar xf - %TA", 0},
 	{"*.tgz",	"tar cf - %T|gzip -c > %C",
 					"gzip -cd %C|tar xf - %TA", 0},
+	{"*.tlz",	"tar cf - %T|lzma -c > %C",
+					"lzcat %C|tar xf - %TA", 0},
 	{"*.txz",	"tar cf - %T|xz -c > %C",
 					"xz -cd %C|tar xf - %TA", 0},
 #endif	/* !MSDOS */
