--- Makefile.in.orig	2012-07-27 00:00:00.000000000 +0900
+++ Makefile.in	2012-07-27 09:11:13.000000000 +0900
@@ -13,7 +13,7 @@
 BINDIR	= $(BUILD)$(BINTOP)
 DATATOP	= $(PREFIX)/share
 DATADIR	= $(BUILD)$(DATATOP)/fd
-MANTOP	= $(PREFIX)/man
+MANTOP	= $(DATATOP)/man
 MANDIR	= $(BUILD)$(MANTOP)/man$(MANSEC)
 CATDIR	= $(BUILD)$(MANTOP)/cat$(MANSEC)
 JMANDIR	= $(BUILD)$(MANTOP)__LANGDIR__/man$(MANSEC)
@@ -228,7 +228,7 @@
 
 $(PROGRAM)$(EXE): $(OBJ1) $(OBJ2) $(OBJ3) $(OBJ4) $(OBJ5) $(OBJ6) $(ARGS)
 __SLEEP__	@sleep 1
-	$(CC) __MEM__ __LNK__ __OBJLIST__ $(FLDFLAGS)
+	$(CC) $(CFLAGS) __MEM__ __LNK__ __OBJLIST__ $(FLDFLAGS)
 __DJGPP1__	@ren $@ $(PROGRAM)
 __DJGPP1__	@aout2exe $(PROGRAM)
 __DJGPP1__	@del $(PROGRAM)
@@ -466,7 +466,7 @@
 ############################################################
 # to install all
 
-install: $(PROGRAM)$(EXE) mkdir_p$(EXE) install-table install-man install-jman
+install: $(PROGRAM)$(EXE) mkdir_p$(EXE) install-table install-man
 	[ -d $(BINDIR) ] || ./mkdir_p$(EXE) $(BINDIR)
 	$(INSTALL) __INSTSTRIP__ $(PROGRAM)$(EXE) $(BINDIR)
 	$(CHMOD) a+rx $(BINDIR)/$(PROGRAM)$(EXE)
