--- fd.h.orig	2010-09-25 00:00:00.000000000 +0900
+++ fd.h	2012-03-09 18:33:23.000000000 +0900
@@ -106,15 +106,15 @@
  *	variables nor run_com file nor command line option	*
  ****************************************************************/
 #define	BASICCUSTOM		0
-#define	SORTTYPE		0
-#define	DISPLAYMODE		0
-#define	SORTTREE		0
+#define	SORTTYPE		1
+#define	DISPLAYMODE		3
+#define	SORTTREE		1
 #define	WRITEFS			0
 #define	IGNORECASE		0
 #define	INHERITCOPY		0
 #define	PROGRESSBAR		0
 #define	PRECOPYMENU		0
-#define	ADJTTY			0
+#define	ADJTTY			1
 #define	USEGETCURSOR		0
 #define	DEFCOLUMNS		2
 #define	MINFILENAME		12
@@ -150,7 +150,7 @@
 #else
 #define	TMPDIR			"/tmp"
 #endif
-#define	TMPUMASK		022
+#define	TMPUMASK		077
 #define	RRPATH			""
 #define	PRECEDEPATH		""
 #if	FD >= 2
@@ -179,7 +179,7 @@
 #define	HTTPPROXY		""
 #define	HTTPLOGFILE		""
 #define	HTMLLOGFILE		""
-#define	UNICODEBUFFER		0
+#define	UNICODEBUFFER		1
 #define	SJISPATH		""
 #define	EUCPATH			""
 #define	JISPATH			""
