--- Makefile.in.orig	2010-02-27 00:00:00.000000000 +0900
+++ Makefile.in	2010-02-28 02:46:18.000000000 +0900
@@ -13,7 +13,7 @@
 BINDIR	= $(BUILD)$(BINTOP)
 DATATOP	= $(PREFIX)/share
 DATADIR	= $(BUILD)$(DATATOP)/fd
-MANTOP	= $(PREFIX)/man
+MANTOP	= $(DATATOP)/man
 MANDIR	= $(BUILD)$(MANTOP)/man$(MANSEC)
 CATDIR	= $(BUILD)$(MANTOP)/cat$(MANSEC)
 JMANDIR	= $(BUILD)$(MANTOP)__LANGDIR__/man$(MANSEC)
@@ -209,7 +209,7 @@
 
 $(PROGRAM)$(EXE): $(OBJ1) $(OBJ2) $(OBJ3) $(OBJ4) $(OBJ5) $(OBJ6) $(ARGS)
 __SLEEP__	@sleep 1
-	$(CC) __MEM__ __LNK__ __OBJLIST__ $(LDFLAGS)
+	$(CC) $(CFLAGS) __MEM__ __LNK__ __OBJLIST__ $(LDFLAGS)
 __DJGPP1__	@ren $@ $(PROGRAM)
 __DJGPP1__	@aout2exe $(PROGRAM)
 __DJGPP1__	@del $(PROGRAM)
@@ -447,12 +447,6 @@
 		$(INSTALL) $(DICTTBL) __DATADIR__; \
 		$(CHMOD) a+r-x __DATADIR__/$(DICTTBL); \
 	fi
-	[ -d $(JMANDIR) ] || ./mkdir_p$(EXE) $(JMANDIR)
-	$(INSTALL) $(PROGRAM).$(MANSEC) $(JMANDIR)/$(PROGRAM).$(MANSEC)
-	$(CHMOD) a+r-x $(JMANDIR)/$(PROGRAM).$(MANSEC)
-	-$(RM) $(JMANDIR)/$(ALIAS).$(MANSEC)
-	$(LN) $(JMANDIR)/$(PROGRAM).$(MANSEC) $(JMANDIR)/$(ALIAS).$(MANSEC)
-	$(CHMOD) a+r-x $(JMANDIR)/$(ALIAS).$(MANSEC)
 	[ -d $(MANDIR) ] || ./mkdir_p$(EXE) $(MANDIR)
 	$(INSTALL) $(MANSRC) $(MANDIR)/$(PROGRAM).$(MANSEC)
 	$(CHMOD) a+r-x $(MANDIR)/$(PROGRAM).$(MANSEC)
