--- custom.c.orig	2010-09-25 00:00:00.000000000 +0900
+++ custom.c	2012-03-09 19:41:46.000000000 +0900
@@ -509,20 +509,20 @@
 		DEFVAL(UNICODEBUFFER), UNBF_E, T_BOOL},
 #endif
 #if	defined (DEP_KCONV) || (!defined (_NOENGMES) && !defined (_NOJPNMES))
-	{"FD_LANGUAGE", &outputkcode, DEFVAL(NOCNV), LANG_E, _B_(T_KOUT)},
+	{"FD_LANGUAGE", &outputkcode, DEFVAL("utf8"), LANG_E, _B_(T_KOUT)},
 #endif
 #ifdef	DEP_FILECONV
-	{"FD_DEFKCODE", &defaultkcode, DEFVAL(NOCNV), DFKC_E, T_KNAM},
+	{"FD_DEFKCODE", &defaultkcode, DEFVAL("utf8"), DFKC_E, T_KNAM},
 #endif
 #ifdef	DEP_KCONV
-	{"FD_INPUTKCODE", &inputkcode, DEFVAL(NOCNV), IPKC_E, _B_(T_KIN)},
+	{"FD_INPUTKCODE", &inputkcode, DEFVAL("utf8"), IPKC_E, _B_(T_KIN)},
 #endif
 #if	defined (DEP_KCONV) && defined (DEP_PTY)
-	{"FD_PTYINKCODE", &ptyinkcode, DEFVAL(NOCNV), PIKC_E, T_KTERM},
-	{"FD_PTYOUTKCODE", &ptyoutkcode, DEFVAL(NOCNV), POKC_E, T_KTERM},
+	{"FD_PTYINKCODE", &ptyinkcode, DEFVAL("utf8"), PIKC_E, T_KTERM},
+	{"FD_PTYOUTKCODE", &ptyoutkcode, DEFVAL("utf8"), POKC_E, T_KTERM},
 #endif
 #ifdef	DEP_FILECONV
-	{"FD_FNAMEKCODE", &fnamekcode, DEFVAL(NOCNV), FNKC_E, T_KNAM},
+	{"FD_FNAMEKCODE", &fnamekcode, DEFVAL("utf8-mac"), FNKC_E, T_KNAM},
 #endif
 #if	defined (DEP_FILECONV) && defined (DEP_URLPATH)
 	{"FD_URLKCODE", &urlkcode, DEFVAL(NOCNV), URLKC_E, T_KNAM},
@@ -822,6 +822,7 @@
 		case T_KOUT:
 		case T_KNAM:
 		case T_KTERM:
+			if (!cp) cp = def_str(no);
 			n = (1 << (env_type(no) - T_KIN));
 			*((int *)(envlist[no].var)) = getlang(cp, n);
 			break;
