--- termio.c.orig	2008-03-15 00:00:00.000000000 +0900
+++ termio.c	2008-04-14 19:27:13.000000000 +0900
@@ -396,8 +396,10 @@
 	if (*fdp >= 0) fd = *fdp;
 	else if ((fd = newdup(open(_PATH_TTY, flags, 0666))) < 0) return(-1);
 	if (*fpp) fp = *fpp;
-	else if (!(fp = fdopen(fd, "w+b"))
-	&& !(fp = fopen(_PATH_TTY, "w+b"))) {
+#ifndef	SELECTRWONLY
+	else if ((fp = fdopen(fd, "w+b"))) /*EMPTY*/;
+#endif
+	else if (!(fp = fopen(_PATH_TTY, "w+b"))) {
 		close(fd);
 		return(-1);
 	}
