--- DESCRIPTION	2023-09-16 12:50:02.000000000 +0700
+++ DESCRIPTION	2023-09-16 19:41:17.000000000 +0700
@@ -37,7 +37,7 @@
 RoxygenNote: 7.2.3
 Suggests: magrittr, covr, testthat, knitr, rmarkdown, jqr, DT, tibble,
         purrr, lubridate, httr, xml2, jsonlite, repurrrsive,
-        nycflights13, spelling, jsonld
+        nycflights13, spelling
 VignetteBuilder: knitr
 Language: en-US
 NeedsCompilation: no
