--- src/decomposition.c	1995-06-20 11:46:59
+++ src/decomposition.c	2026-05-01 02:53:57
@@ -41,7 +41,7 @@
   static long ltan2_8125 ; /* tan 2.8125 degrees */
   static long ltan1_4063 ; /* tan 1.4063 degrees */
 
-  static reset_fudge = -1;
+  static int reset_fudge = -1;
   int temp = 0;
   int i;
 
--- src/fractals.c	1995-06-20 11:47:02
+++ src/fractals.c	2026-05-01 02:54:32
@@ -1733,7 +1733,7 @@
       register int col;
       for (col = 0; col <= ixstop; col++)  /* look at each point on screen */
         {
-        register color;
+        register int color;
         init.y = dy0[row]+dy1[col];
         init.x = dx0[col]+dx1[row];
         if(check_key() != 0)
--- src/jiim.c	1995-06-20 11:47:04
+++ src/jiim.c	2026-05-01 02:53:45
@@ -48,8 +48,8 @@
 static float luckyy = 0;
 static int cursx;
 static int cursy;
-static oldrootx = 0;
-static oldrooty = 0;
+static int oldrootx = 0;
+static int oldrooty = 0;
 static int oldmyx = 0;
 static int oldmyy = 0;
 static int warned = 0;
