--- etc/Makefile.in.orig	Wed May 25 22:55:29 2005
+++ etc/Makefile.in	Wed May 25 22:55:59 2005
@@ -28,7 +28,7 @@
 VPATH = @srcdir@
 
 prefix = @prefix@
-libdir = $(prefix)/xmfract
+libdir = $(DESTDIR)$(prefix)/share/xmfract/data
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 RM = rm

--- help/Makefile.in.orig	Wed May 25 22:58:25 2005
+++ help/Makefile.in	Wed May 25 22:59:00 2005
@@ -28,7 +28,7 @@
 VPATH = @srcdir@
 
 prefix = @prefix@
-libdir = $(prefix)/xmfract
+libdir = $(DESTDIR)$(prefix)/share/xmfract
 helpdir = $(libdir)/help
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@

--- src/Makefile.in.orig	Wed May 25 22:59:50 2005
+++ src/Makefile.in	Wed May 25 23:01:09 2005
@@ -40,8 +40,8 @@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
-bindir = $(exec_prefix)/bin
-libdir = $(prefix)/xmfract
+bindir = $(DESTDIR)$(exec_prefix)/bin
+libdir = $(DESTDIR)$(prefix)/share/xmfract
 
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
