--- filter/tex/filtersettings.php.orig	2013-07-06 08:49:18.000000000 -0700
+++ filter/tex/filtersettings.php	2014-07-21 13:12:37.000000000 -0700
@@ -43,9 +43,9 @@
 
     } else if (PHP_OS=='Darwin') {
         // most likely needs a fink install (fink.sf.net)
-        $default_filter_tex_pathlatex   = "/sw/bin/latex";
-        $default_filter_tex_pathdvips   = "/sw/bin/dvips";
-        $default_filter_tex_pathconvert = "/sw/bin/convert";
+        $default_filter_tex_pathlatex   = "@PREFIX@/bin/latex";
+        $default_filter_tex_pathdvips   = "@PREFIX@/bin/dvips";
+        $default_filter_tex_pathconvert = "@PREFIX@/bin/convert";
 
     } else if (PHP_OS=='WINNT' or PHP_OS=='WIN32' or PHP_OS=='Windows') {
         // note: you need Ghostscript installed (standard), miktex (standard)
@@ -74,4 +74,4 @@
         $item->set_updatedcallback('filter_tex_updatedcallback');
         $settings->add($item);
     }
-}
\ No newline at end of file
+}
