--- compiler/main/DynFlags.hs.sav	2010-03-09 16:45:42.000000000 -0500
+++ compiler/main/DynFlags.hs	2010-03-09 16:47:38.000000000 -0500
@@ -577,9 +577,7 @@
         hpcDir                  = ".hpc",
 
         opt_L                   = [],
-        opt_P                   = (if opt_PIC
-                                   then ["-D__PIC__"]
-                                   else []),
+        opt_P                   = [],
         opt_F                   = [],
         opt_c                   = [],
         opt_a                   = [],
@@ -2172,7 +2170,7 @@
       --     in dynamic libraries.
 
     | opt_PIC
-        = ["-fno-common", "-D__PIC__"]
+        = ["-fno-common"]
     | otherwise
         = ["-mdynamic-no-pic"]
 #elif mingw32_TARGET_OS
