--- configure.ac.sav	2009-08-29 16:28:58.000000000 -0400
+++ configure.ac	2009-08-29 16:30:11.000000000 -0400
@@ -795,7 +795,7 @@
      AC_MSG_ERROR([GHC is required unless bootstrapping from .hc files.])
   fi
   FP_COMPARE_VERSIONS([$GhcVersion],[-lt],[6.4],
-    [AC_MSG_ERROR([GHC version 6.4 or later is required to compile GHC.])])dnl
+    [AC_MSG_ERROR([GHC version 6.4 or later is required to compile GHC.])])
 fi;
 
 # This uses GHC, so put it after the "GHC is required" check above:
@@ -987,15 +987,6 @@
 dnl ** check for ghc-pkg command
 FP_PROG_GHC_PKG
 
-AC_ARG_WITH(greencard,
-[AC_HELP_STRING([--with-greencard=ARG],
-        [Use ARG as the path to greencard  [default=autodetct]])],
-[
-GreenCardCmd=$withval;
-FPTOOLS_GREENCARD(3.00)
-]
-)
-
 dnl ** check for installed happy binary + version
 dnl    (don't do it if we're booting from .hc files though.)
 if test "$BootingFromHc" = "NO"; then
@@ -1138,19 +1129,19 @@
 
 dnl ** (Mac OS X only: check for GMP.framework)
 HaveFrameworkGMP=NO
-case $HostPlatform in
-*-apple-darwin)
-    AC_MSG_CHECKING([for GMP.framework])
-    save_libs="$LIBS"
-    LIBS="-framework GMP"
-    AC_TRY_LINK_FUNC(__gmpz_fdiv_qr, HaveFrameworkGMP=YES,)
-    if test $HaveFrameworkGMP = YES; then
-      AC_DEFINE([HAVE_FRAMEWORK_GMP], [1], [Define to 1 if GMP.framework is installed (Mac OS X only).])
-    fi;
-    LIBS="$save_libs"
-    AC_MSG_RESULT([$HaveFrameworkGMP])
-    ;;
-esac
+dnl case $HostPlatform in
+dnl *-apple-darwin)
+dnl    AC_MSG_CHECKING([for GMP.framework])
+dnl    save_libs="$LIBS"
+dnl    LIBS="-framework GMP"
+dnl    AC_TRY_LINK_FUNC(__gmpz_fdiv_qr, HaveFrameworkGMP=YES,)
+dnl    if test $HaveFrameworkGMP = YES; then
+dnl      AC_DEFINE([HAVE_FRAMEWORK_GMP], [1], [Define to 1 if GMP.framework is installed (Mac OS X only).])
+dnl    fi;
+dnl    LIBS="$save_libs"
+dnl    AC_MSG_RESULT([$HaveFrameworkGMP])
+dnl    ;;
+dnl esac
 AC_SUBST(HaveFrameworkGMP)
 
 dnl ** check for mingwex library
