--- compiler/Makefile.sav	2009-08-27 08:09:21.000000000 -0400
+++ compiler/Makefile	2009-08-27 08:17:36.000000000 -0400
@@ -190,6 +190,10 @@
 	                   $(CONFIGURE_FLAGS_STAGE$*) \
 	                   $(INSTALL_DIRS_CONFIGURE_FLAGS) \
 	                   $(COMMON_CONFIGURE_FLAGS) \
+			   --ghc-option='-pgmP "XXPATH_TO_GCCXX -E -undef -traditional"' \
+			   --ghc-option='-pgmc XXPATH_TO_GCCXX' \
+			   --ghc-option='-pgml XXPATH_TO_GCCXX' \
+			   --ghc-option='-pgma XXPATH_TO_GCCXX' \
 	                   --ghc-option=-DSTAGE=$*
 	$(RM) -f Makefile-stage$*
 	$(CABAL) makefile  --distpref dist-stage$* -f Makefile-stage$*
