--- ghc/Makefile.sav	2009-08-27 08:20:43.000000000 -0400
+++ ghc/Makefile	2009-08-27 08:21:06.000000000 -0400
@@ -108,10 +108,14 @@
 dist-stage1 dist-stage2 dist-stage3: dist-stage%: boot.stage.%
 
 boot.stage.%:
-	$(CABAL) configure --distpref dist-stage$*         \
+	$(CABAL) configure --verbose=3 --distpref dist-stage$*         \
 	                   $(INSTALL_DIRS_CONFIGURE_FLAGS) \
 	                   $(CONFIGURE_FLAGS_STAGE$*)      \
 	                   $(COMMON_CONFIGURE_FLAGS)       \
+			   --ghc-option='-pgmP XXPATH_TO_GCCXX -E -undef -traditional' \
+			   --ghc-option='-pgmc XXPATH_TO_GCCXX' \
+			   --ghc-option='-pgml XXPATH_TO_GCCXX' \
+			   --ghc-option='-pgma XXPATH_TO_GCCXX' \
 	                   --libsubdir=.                   \
 	                   --datadir='$$libdir'            \
 	                   --datasubdir=.
