--- compiler/cmm/CLabel.hs.sav	2010-03-09 16:45:23.000000000 -0500
+++ compiler/cmm/CLabel.hs	2010-03-09 16:53:31.000000000 -0500
@@ -918,6 +918,10 @@
 pprDynamicLinkerAsmLabel :: DynamicLinkerLabelInfo -> CLabel -> SDoc
 
 #if x86_64_TARGET_ARCH && darwin_TARGET_OS
+pprDynamicLinkerAsmLabel CodeStub lbl
+  = char 'L' <> pprCLabel lbl <> text "$stub"
+pprDynamicLinkerAsmLabel SymbolPtr lbl
+  = char 'L' <> pprCLabel lbl <> text "$non_lazy_ptr"
 pprDynamicLinkerAsmLabel GotSymbolPtr lbl
   = pprCLabel lbl <> text "@GOTPCREL"
 pprDynamicLinkerAsmLabel GotSymbolOffset lbl
