--- compiler/Makefile.local.sav	2009-08-25 10:58:22.000000000 -0400
+++ compiler/Makefile.local	2009-08-25 11:42:30.000000000 -0400
@@ -19,6 +19,10 @@
 GHC_CC_OPTS += $(addprefix -optc, $(MACOSX_DEPLOYMENT_CC_OPTS))
 GHC_OPTS    += $(SRC_HC_OPTS)
 GHC_OPTS    += $(GhcHcOpts)
+GHC_CC_OPTS += -pgmP "XXPATH_TO_GCCXX -E -undef -traditional"
+GHC_CC_OPTS += -pgma XXPATH_TO_GCCXX
+GHC_CC_OPTS += -pgmc XXPATH_TO_GCCXX
+GHC_CC_OPTS += -pgml XXPATH_TO_GCCXX
 GHC_OPTS    += $(GhcStage$(stage)HcOpts)
 GHC_OPTS    += $(addprefix -optc, $(MACOSX_DEPLOYMENT_CC_OPTS))
 LIB_LD_OPTS += $(addprefix -optl, $(MACOSX_DEPLOYMENT_LD_OPTS))
