--- Makefile.orig	2023-11-12 11:56:29
+++ Makefile	2023-11-12 11:56:33
@@ -40,7 +40,7 @@
 
 PACKAGE := github.com/lima-vm/lima
 
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+VERSION=@@VERSION@@
 VERSION_TRIMMED := $(VERSION:v%=%)
 
 GO_BUILD := $(GO) build -ldflags="-s -w -X $(PACKAGE)/pkg/version.Version=$(VERSION)" -tags "$(GO_BUILDTAGS)"
@@ -209,7 +209,7 @@
 install: uninstall
 	mkdir -p "$(DEST)"
 	# Use tar rather than cp, for better symlink handling
-	( cd _output && tar c * | tar Cxv "$(DEST)" )
+	( cd _output && $(TAR) c * | $(TAR) -xv --no-same-owner -C "$(DEST)" )
 	if [ "$(shell uname -s )" != "Linux" -a ! -e "$(DEST)/bin/nerdctl" ]; then ln -sf nerdctl.lima "$(DEST)/bin/nerdctl"; fi
 	if [ "$(shell uname -s )" != "Linux" -a ! -e "$(DEST)/bin/apptainer" ]; then ln -sf apptainer.lima "$(DEST)/bin/apptainer"; fi
 
