--- lib/MIME/Parser.pm.orig	2008-03-12 14:00:20.000000000 -0700
+++ lib/MIME/Parser.pm	2008-09-16 18:27:03.000000000 -0700
@@ -906,6 +906,12 @@
     }
     $good or do { $self->debug("no one made the cut"); return 0 };
 
+	## 20080916 EH:
+	## ignore binhex types, the binhex decoder doesn't (always) work right
+	if ($how_encoded =~ /binhex/i) {
+		$self->debug("Not decoding binhex attachment due to bugs in the binhex decoder");
+		return 0;
+	}
     # If a decoder doesn't exist for this type, forget it!
     my $decoder = MIME::Decoder->new(($how_encoded eq 'uu')?'x-uuencode'
 						     :'binhex');
