--- enthought/kiva/agg/setup.py.orig	2010-11-10 15:04:57.000000000 +0100
+++ enthought/kiva/agg/setup.py	2010-11-10 15:02:29.000000000 +0100
@@ -79,7 +79,8 @@
     if sys.platform=='win32':
         plat = 'win32'
     elif sys.platform == 'darwin':
-        plat = 'gl'
+        #plat = 'gl'
+        plat = 'x11'
     else:
         #plat = 'gtk1'  # use with gtk1, it's fast
         plat = 'x11'  # use with gtk2, it's slow but reliable
