--- compiler/Makefile.unix.orig	2018-01-30 04:50:24.000000000 -0600
+++ compiler/Makefile.unix	2018-01-30 11:13:03.000000000 -0600
@@ -166,8 +166,8 @@
 endif
 
 ifeq ($(system), Darwin)
-    CXXFLAGS += -I /opt/local/include
-    CXXFLAGS += -stdlib=libc++
+    CXXFLAGS += -I$(prefix)/include
+    CXXFLAGS += @STDLIB_FLAG@
 else
     ifneq ($(findstring MINGW32, $(system)),)
         CXXFLAGS += -O1 -I/usr/include
