--- wfview.pro.orig	2021-07-26 11:31:27.000000000 +0200
+++ wfview.pro	2021-08-02 16:50:15.000000000 +0200
@@ -50,8 +50,7 @@
 #linux:DEFINES += __LINUX_PULSE__
 macx:DEFINES += __MACOSX_CORE__
 
-macx:INCLUDEPATH += /usr/local/include /opt/local/include
-macx:LIBS += -L/usr/local/lib -L/opt/local/lib
+macx:LIBS += -lqcustomplot
 
 macx:ICON = ../wfview/resources/wfview.icns
 win32:RC_ICONS = ../wfview/resources/wfview.ico
@@ -102,9 +101,9 @@
 
 #win32:SOURCES += rtaudio/RTAudio.cpp
 #win32:HEADERS += rtaudio/RTAUdio.h
-!linux:SOURCES += ../qcustomplot/qcustomplot.cpp 
-!linux:HEADERS += ../qcustomplot/qcustomplot.h
-!linux:INCLUDEPATH += ../qcustomplot
+#!linux:SOURCES += ../qcustomplot/qcustomplot.cpp
+#!linux:HEADERS += ../qcustomplot/qcustomplot.h
+#!linux:INCLUDEPATH += ../qcustomplot
 
 INCLUDEPATH += resampler
 !linux:INCLUDEPATH += rtaudio
