diff --git wfview.pro wfview.pro
index 275f3c4..c4cb143 100644
--- wfview.pro
+++ wfview.pro
@@ -34,7 +34,7 @@ DEFINES += OUTSIDE_SPEEX
 DEFINES += RANDOM_PREFIX=wf
 
 macx:INCLUDEPATH += /usr/local/include /opt/local/include
-macx:LIBS += -L/usr/local/lib -L/opt/local/lib
+macx:LIBS += -L/usr/local/lib -L/opt/local/lib  -lqcustomplot
 
 macx:ICON = ../wfview/resources/wfview.icns
 win32:RC_ICONS = ../wfview/resources/wfview.ico
@@ -81,10 +81,6 @@ CONFIG(debug, release|debug) {
 
 linux:LIBS += -L./ -l$$QCPLIB
 
-!linux:SOURCES += ../qcustomplot/qcustomplot.cpp
-!linux:HEADERS += ../qcustomplot/qcustomplot.h
-!linux:INCLUDEPATH += ../qcustomplot
-
 SOURCES += main.cpp\
         wfmain.cpp \
     commhandler.cpp \
