--- wfview.pro.orig	2023-07-16 15:05:53.000000000 +0200
+++ wfview.pro	2023-09-19 13:19:41.000000000 +0200
@@ -47,14 +47,11 @@
 #linux:DEFINES += __LINUX_OSS__
 linux:DEFINES += __LINUX_PULSE__
 macx:DEFINES += __MACOSX_CORE__
-!linux:SOURCES += ../rtaudio/RTAudio.cpp
-!linux:HEADERS += ../rtaudio/RTAUdio.h
-!linux:INCLUDEPATH += ../rtaudio
 
 linux:LIBS += -lpulse -lpulse-simple -lrtaudio -lpthread -ludev
 
 win32:INCLUDEPATH += ../portaudio/include
-!win32:LIBS += -lportaudio
+!win32:LIBS += -lportaudio -lrtaudio
 
 # The following define makes your compiler emit warnings if you use
 # any feature of Qt which as been marked as deprecated (the exact warnings
@@ -86,8 +83,7 @@
 
 DEFINES += PREFIX=\\\"$$PREFIX\\\"
 
-macx:INCLUDEPATH += /usr/local/include /opt/local/include 
-macx:LIBS += -L/usr/local/lib -L/opt/local/lib
+macx:LIBS += -lqcustomplot
 
 macx:ICON = ../wfview/resources/wfview.icns
 win32:RC_ICONS = ../wfview/resources/wfview.ico
@@ -214,7 +210,6 @@
 #macx:HEADERS += ../qcustomplot/qcustomplot.h
 
 win32:INCLUDEPATH += ../qcustomplot
-!linux:INCLUDEPATH += ../opus/include
 
 !linux:INCLUDEPATH += ../eigen
 !linux:INCLUDEPATH += ../r8brain-free-src
