--- configure.ac.orig	2014-01-25 17:16:34.000000000 +0100
+++ configure.ac	2014-01-25 17:17:57.000000000 +0100
@@ -228,7 +228,7 @@
 AC_MSG_RESULT([$enable_vala])
 if test "x$enable_vala" != "xno"; then
 	valapkg=""
-	m4_foreach([VERSION], [[0.12], [0.14], [0.16], [0.18], [0.20]],
+	m4_foreach([VERSION], [[0.12], [0.14], [0.16], [0.18], [0.20], [0.22]],
 		   [PKG_CHECK_EXISTS([ libvala-VERSION ],
                            [ valapkg="libvala-VERSION" ])
 		   ])
