--- configure.ac.orig2	2014-01-26 10:13:49.000000000 +0100
+++ configure.ac	2014-01-26 10:19:54.000000000 +0100
@@ -94,17 +94,6 @@
    exec_prefix="${prefix}"
 fi
 
-if test "${libdir}" = '${exec_prefix}/lib'; then
-   case `uname -m` in
-        x86_64|ppc64|powerpc64)
-                libdir='${exec_prefix}/lib64'
-        ;;
-        *)
-                libdir='${exec_prefix}/lib'
-        ;;
-esac
-fi
-
 # Plugins directory
 
 GRL_PLUGINS_DIR=`eval echo ${libdir}/${GRL_NAME}`
