--- configure.ac.orig	2014-02-10 13:36:13.000000000 -0800
+++ configure.ac	2014-02-13 14:19:59.000000000 -0800
@@ -102,17 +102,6 @@
    exec_prefix="${prefix}"
 fi
 
-if test "${libdir}" = '${exec_prefix}/lib'; then
-   case `uname -m` in
-        x86_64|ppc64|powerpc64)
-                libdir='${exec_prefix}/lib64'
-        ;;
-        *)
-                libdir='${exec_prefix}/lib'
-        ;;
-esac
-fi
-
 # Plugins directory
 
 GRL_PLUGINS_DIR=`eval echo ${libdir}/${GRL_NAME}`
