--- configure.ac.orig	2011-04-21 12:35:37.000000000 -0700
+++ configure.ac	2013-05-09 15:04:51.000000000 -0700
@@ -20,7 +20,7 @@
 dnl Autoconf / Automake Initialization
 AC_PREREQ(2.58)
 AC_INIT([evolution-data-server],[eds_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=Evolution-Data-Server])
-AM_INIT_AUTOMAKE([gnu 1.10 tar-ustar -Wall -Werror foreign])
+AM_INIT_AUTOMAKE([gnu 1.10 tar-ustar -Wall  foreign])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR(README)
 AC_CONFIG_HEADERS(config.h)
@@ -164,8 +164,7 @@
 dnl Compiler Warning Flags
 dnl ******************************
 AS_COMPILER_FLAGS(WARNING_FLAGS,
-	"-DG_DISABLE_DEPRECATED
-	-DPANGO_DISABLE_DEPRECATED
+	"-DPANGO_DISABLE_DEPRECATED
 	-DGDK_PIXBUF_DISABLE_DEPRECATED
 	-DGTK_DISABLE_DEPRECATED
 	-DG_DISABLE_SINGLE_INCLUDES
@@ -199,7 +198,7 @@
 AM_CPPFLAGS="$WARNING_FLAGS -fno-strict-aliasing"
 AC_SUBST(AM_CPPFLAGS)
 
-LDFLAGS="$LDFLAGS -Wl,--no-undefined"
+dnl LDFLAGS="$LDFLAGS -Wl,--no-undefined"
 
 dnl ******************************
 dnl Initialize maintainer mode
@@ -325,11 +324,13 @@
 	libsoup-2.4 >= libsoup_minimum_version
 	libgdata >= libgdata_minimum_version])
 
-dnl libgdata 0.7 will break API.  We're not going to deal with
-dnl that on a stable branch, so cap the supported version.
+ac_cv_have_gdata_07=no
+dnl libgdata 0.7 breaks API, thus deal with it.
 if $PKG_CONFIG --atleast-version=0.7 libgdata; then
-	AC_MSG_ERROR([libgdata 0.7 is not supported, use 0.6 instead])
+        ac_cv_have_gdata_07=yes
+        AC_DEFINE(HAVE_GDATA_07, 1, [libgdata is 0.7 or higher])
 fi
+AC_MSG_RESULT([$ac_cv_have_gdata_07])
 
 LIBICAL_REQUIRED=libical_minimum_version
 AC_SUBST(LIBICAL_REQUIRED)
@@ -337,8 +338,8 @@
 dnl ******************************
 dnl regex check
 dnl ******************************
-AC_CHECK_FUNCS(regexec,,[AC_CHECK_LIB(regex,regexec,
-			[REGEX_LIBS=-lregex
+AC_CHECK_FUNCS(regexec,,[AC_CHECK_LIB(gnuregex,regexec,
+			[REGEX_LIBS=-lgnuregex
 			AC_DEFINE(HAVE_REGEXEC,1,[Define to 1 if you have the regexec function.])],
 			[AC_MSG_ERROR([No regex library found])])])
 AC_SUBST(REGEX_LIBS)
@@ -351,8 +352,8 @@
 	[Prefix where libdb is installed]),
 	[libdb_prefix="$withval"], [libdb_prefix='${prefix}'])
 
-DB_CFLAGS="-I$libdb_prefix/include"
-DB_LIBS="-L$libdb_prefix/lib -ldb"
+DB_CFLAGS="-I$prefix/include/db46"
+DB_LIBS="-L$prefix/lib/db46  -ldb"
 
 AC_MSG_CHECKING([Berkeley DB])
 save_cflags=$CFLAGS; CFLAGS=$DB_CFLAGS
