--- m4/evo_krb5_support.m4.orig	2010-08-26 10:41:38.000000000 -0700
+++ m4/evo_krb5_support.m4	2010-08-26 14:02:02.000000000 -0700
@@ -86,12 +86,6 @@
 
 	AM_CONDITIONAL(ENABLE_KRB5, [test "x$with_krb5" != "xno"])
 
-	AC_CHECK_HEADER([et/com_err.h],
-		[AC_DEFINE([HAVE_ET_COM_ERR_H], 1, [Have <et/com_err.h>])],,
-		[[	#if HAVE_ET_COM_ERR_H
-			#include <com_err.h>
-			#endif
-		]])
 	AC_CHECK_HEADER([com_err.h],
 		[AC_DEFINE([HAVE_COM_ERR_H], 1, [Have <com_err.h>])],,
 		[[	#if HAVE_COM_ERR_H
