--- CMakeLists.txt.orig	2018-03-27 14:36:17.000000000 -0700
+++ CMakeLists.txt	2018-03-27 14:37:25.000000000 -0700
@@ -292,12 +292,12 @@
 	message(FATAL_ERROR "You need gperf to build ${PROJECT_NAME}")
 endif(NOT GPERF)
 
-find_program(PERL perl)
+find_program(PERL perl5.26 perl)
 if(NOT PERL)
 	message(FATAL_ERROR "You need perl to build ${PROJECT_NAME}")
 endif(NOT PERL)
 
-find_program(PYTHON python3 python2 python)
+find_program(PYTHON python3.6 python3 python2 python)
 if(NOT PYTHON)
 	message(FATAL_ERROR "You need python to build ${PROJECT_NAME}")
 endif(NOT PYTHON)
@@ -306,7 +306,7 @@
 # db_load checking, it's optional
 # ******************************
 
-find_program(DB_LOAD NAMES db_load db4_load db_load-4 db5_load db_load-5)
+find_program(DB_LOAD NAMES db53_load db_load db4_load db_load-4 db5_load db_load-5)
 if(DB_LOAD)
 	set(HAVE_DB_LOAD ON)
 else(DB_LOAD)
