--- calendar/backends/caldav/e-cal-backend-caldav.c.orig	2011-04-21 12:35:36.000000000 -0700
+++ calendar/backends/caldav/e-cal-backend-caldav.c	2013-06-02 23:29:41.000000000 -0700
@@ -1072,8 +1072,8 @@
 	soup_message_set_request (message,
 				  "application/xml",
 				  SOUP_MEMORY_COPY,
-				  (gchar *) buf->buffer->content,
-				  buf->buffer->use);
+				  (gchar *) xmlBufContent(buf->buffer),
+                                  xmlBufUse(buf->buffer));
 
 	/* Send the request now */
 	send_and_handle_redirection (priv->session, message, NULL);
@@ -1214,8 +1214,8 @@
 	soup_message_set_request (message,
 				  "application/xml",
 				  SOUP_MEMORY_COPY,
-				  (gchar *) buf->buffer->content,
-				  buf->buffer->use);
+				  (gchar *) xmlBufContent(buf->buffer),
+                                  xmlBufUse(buf->buffer)); 
 
 	/* Send the request now */
 	send_and_handle_redirection (priv->session, message, NULL);
@@ -1530,8 +1530,8 @@
 	soup_message_set_request (message,
 				  "application/xml",
 				  SOUP_MEMORY_COPY,
-				  (gchar *) buf->buffer->content,
-				  buf->buffer->use);
+				  (gchar *) xmlBufContent(buf->buffer),
+                                  xmlBufUse(buf->buffer));
 
 	/* Send the request now */
 	send_and_handle_redirection (priv->session, message, NULL);
@@ -1580,8 +1580,8 @@
 		soup_message_set_request (message,
 				  "application/xml",
 				  SOUP_MEMORY_COPY,
-				  (gchar *) buf->buffer->content,
-				  buf->buffer->use);
+				  (gchar *) xmlBufContent(buf->buffer),
+                                  xmlBufUse(buf->buffer));
 
 		/* Send the request now */
 		send_and_handle_redirection (priv->session, message, NULL);
