--- ./calendar/libical/src/libical/icaltz-util.orig	2008-04-02 18:33:48.000000000 -0600
+++ ./calendar/libical/src/libical/icaltz-util.c	2008-04-02 18:34:10.000000000 -0600
@@ -30,8 +30,8 @@
 #define BYTE_ORDER LITTLE_ENDIAN
 #define bswap_32(val) ((((val) & 0x000000ffU) << 24) | (((val) & 0x0000ff00U) <<  8) | (((val) & 0x00ff0000U) >>  8) | (((val) & 0xff000000U) >> 24))
 #else
-#include <byteswap.h>
-#include <endian.h>
+#define bswap_32(val) ((((val) & 0x000000ffU) << 24) | (((val) & 0x0000ff00U) <<  8) | (((val) & 0x00ff0000U) >>  8) | (((val) & 0xff000000U) >> 24))
+#include <machine/endian.h>
 #endif
 #endif
 #include <limits.h>
