--- CMakeLists.txt.orig	2025-12-19 23:27:43
+++ CMakeLists.txt	2025-12-19 23:32:59
@@ -337,12 +337,13 @@
     pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
 endif()

-find_package(tomlplusplus 3.2.0)
-# fallback to pkgconfig, important especially as many distros package toml++ built with meson
-if(NOT tomlplusplus_FOUND)
-    find_package(PkgConfig REQUIRED)
-    pkg_check_modules(tomlplusplus REQUIRED IMPORTED_TARGET tomlplusplus>=3.2.0)
-endif()
+include(FetchContent)
+FetchContent_Declare(
+    tomlplusplus
+    GIT_REPOSITORY https://github.com/marzer/tomlplusplus.git
+    GIT_TAG        v3.2.0
+)
+FetchContent_MakeAvailable(tomlplusplus)

 find_package(ZLIB REQUIRED)
