--- src/Makefile.orig	2022-01-26 09:52:25.000000000 -0500
+++ src/Makefile	2022-01-26 09:53:37.000000000 -0500
@@ -20,13 +20,10 @@
 CPPFLAGS := $(CPPFLAGS) -DNDEBUG -pthread
 
 CXX := g++
-ifeq ($(OS),Darwin)
-	CXX := g++-11
-endif
 
-CXXFLAGS := $(CXXFLAGS) -O3 -fopenmp -ffast-math
+CXXFLAGS := $(CXXFLAGS) -fopenmp -ffast-math
 
-LDFLAGS := $(LDFLAGS) -O3 -fopenmp -pthread -lpthread
+LDFLAGS := $(LDFLAGS) -fopenmp -pthread -lpthread
 ifeq ($(OS),Linux)
     LDFLAGS += -static
 endif
@@ -37,15 +34,12 @@
 
 .PHONY: clean
 
-$(OS)/muscle : gitver.txt $(OS)/ $(OBJS)
+$(OS)/muscle : $(OS)/ $(OBJS)
 	$(CXX) $(LDFLAGS) $(OBJS) -o $@
 
 	# Warning: do not add -d option to strip, this is not portable
 	strip $(OS)/muscle
 
-gitver.txt : $(SRCS)
-	bash ./gitver.bash
-
 $(OS)/ :
 	mkdir -p $(OS)/
 
@@ -53,4 +47,4 @@
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o $@ $<
 
 clean:
-	rm -rf gitver.txt $(OS)/
+	rm -rf $(OS)/
