--- pync/TerminalNotifier.py	2013-07-16 11:43:21.000000000 +0200
+++ pync/TerminalNotifier.py.new	2014-01-09 14:13:40.000000000 +0100
@@ -17,7 +17,7 @@
         Raises if not supported on the current platform or if terminal-notifier.app does not find.
         """
         self.app_path = os.path.join(
             os.path.dirname(__file__),
-            "vendor/terminal-notifier-%s/terminal-notifier.app" % self.TERMINAL_NOTIFIER_VERSION
+            "__PREFIX__/libexec/terminal-notifier.app"
         )
         self.bin_path = os.path.join(self.app_path, "Contents/MacOS/terminal-notifier")
 
