--- pync/TerminalNotifier.py.orig	2017-07-13 13:24:34.000000000 -0700
+++ pync/TerminalNotifier.py	2017-07-13 13:45:09.000000000 -0700
@@ -26,7 +26,7 @@
         else:
             self.app_path = os.path.join(
                 os.path.dirname(__file__),
-                "vendor/terminal-notifier-%s/terminal-notifier.app" % self.TERMINAL_NOTIFIER_VERSION
+                "__PREFIX__/libexec/terminal-notifier.app"
             )
             self.bin_path = os.path.join(self.app_path, "Contents/MacOS/terminal-notifier")
 
