--- setup.py	2013-07-16 11:43:21.000000000 +0200
+++ setup.py.new	2014-01-09 14:11:50.000000000 +0100
@@ -11,10 +11,6 @@
 long_description = '\n\n'.join(file_contents)
 
 terminal_notifier_files = []
-for root, dirs, files in os.walk('pync/vendor/'):
-    root = '/'.join(root.split('/')[1:])
-    for f in files:
-        terminal_notifier_files.append(os.path.join(root, f))
 
 package_data = {
     '': ['LICENSE', 'README.md'] + terminal_notifier_files,
