diff -uNr kdeartwork-3.5.0/kscreensaver/kdesavers/rotation.h kdeartwork-3.5.0-new/kscreensaver/kdesavers/rotation.h
--- kdeartwork-3.5.0/kscreensaver/kdesavers/rotation.h	Mon Oct 10 10:53:54 2005
+++ kdeartwork-3.5.0-new/kscreensaver/kdesavers/rotation.h	Sat Jan 14 13:08:14 2006
@@ -41,10 +41,10 @@
     * @param t Time in seconds, integration variable
     * @param dt Initial time increment in seconds for integration, auto adjusted
     * later to guarantee precision
-    * @param _A Moment of inertia along 1. figure axis
-    * @param _B Moment of inertia along 2. figure axis
-    * @param _C Moment of inertia along 3. figure axis
-    * @param _y Vector of 12 elements containing the initial rotation vector
+    * @param fooA Moment of inertia along 1. figure axis
+    * @param fooB Moment of inertia along 2. figure axis
+    * @param fooC Moment of inertia along 3. figure axis
+    * @param fooy Vector of 12 elements containing the initial rotation vector
     * omega (elements 0 to 2), and the initial rotating systems coordinate
     * vectors e1, e2, e3 (elements 3 to 5, 6 to 8, and 9 to 11).
     * @param eps Relative precision per integration step, see
@@ -52,10 +52,10 @@
    EulerOdeSolver(
       const double &t,
       const double &dt,
-      const double &_A,
-      const double &_B,
-      const double &_C,
-      std::valarray<double> &_y,
+      const double &fooA,
+      const double &fooB,
+      const double &fooC,
+      std::valarray<double> &fooy,
       const double &eps);
 
   protected:
