--- src/cmd/go/build.go.orig	2012-11-17 01:37:21.000000000 +0100
+++ src/cmd/go/build.go	2012-11-17 01:37:33.000000000 +0100
@@ -1387,7 +1387,7 @@
 	// strings returned are "gcc", "-I", objdir (and cuts them off).
 
 	// TODO: HOST_CC?
-	a := []string{"gcc", "-I", objdir, "-g", "-O2"}
+	a := []string{"@CC@", "-I", objdir, "-g", "-O2"}
 
 	// Definitely want -fPIC but on Windows gcc complains
 	// "-fPIC ignored for target (all code is position independent)"
