Index: kicker/applets/launcher/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kicker/applets/launcher/Makefile.am,v
retrieving revision 1.12
diff -u -u -r1.12 Makefile.am
--- kicker/applets/launcher/Makefile.am	29 Dec 2002 17:01:48 -0000	1.12
+++ kicker/applets/launcher/Makefile.am	27 Jun 2003 22:05:58 -0000
@@ -13,8 +13,10 @@
 
 EXTRA_DIST = $(lnk_DATA)
 
-launcher_panelapplet_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-launcher_panelapplet_la_LIBADD = ../../kicker.la $(LIB_KSYCOCA) $(LIB_KDEUI)
+launcher_panelapplet_la_LDFLAGS = -module $(KDE_RPATH) $(all_libraries)
+launcher_panelapplet_la_LIBADD = ../../core/libkicker_core.la  ../../buttons/libkicker_buttons.la \
+				../../ui/libkicker_ui.la ../../share/libkickermain.la $(LIB_KIO) \
+				$(XTESTLIB) $(LIB_KSYCOCA) $(LIB_KDEUI)
 
 messages:
 	$(XGETTEXT) *.cpp *.h -o $(podir)/quicklauncher.pot
