Index: libkonq/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/libkonq/Makefile.am,v
retrieving revision 1.110
diff -u -b -u -r1.110 Makefile.am
--- libkonq/Makefile.am	17 Mar 2003 14:07:25 -0000	1.110
+++ libkonq/Makefile.am	27 Jun 2003 20:16:44 -0000
@@ -21,7 +21,7 @@
 
 lib_LTLIBRARIES = libkonq.la
 libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined
-libkonq_la_LIBADD = $(LIB_KPARTS)
+libkonq_la_LIBADD = $(LIB_KPARTS) $(LIBZ)
 
 libkonq_la_SOURCES = konq_popupmenu.cc knewmenu.cc \
    konq_xmlguiclient.cc\
