Index: kcontrol/kcontrol/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kcontrol/kcontrol/Makefile.am,v
retrieving revision 1.66
diff -u -b -u -r1.66 Makefile.am
--- kcontrol/kcontrol/Makefile.am	9 Apr 2003 12:48:26 -0000	1.66
+++ kcontrol/kcontrol/Makefile.am	17 Jun 2003 17:22:42 -0000
@@ -25,38 +25,68 @@
 CLEANFILES = dummy.cpp
 
 bin_PROGRAMS = kcontrol kcmshell kcminit
-lib_LTLIBRARIES = kcontrol.la kcmshell.la kcminit.la
+lib_LTLIBRARIES = libkcontrol_main.la kcontrol.la libkcmshell_main.la kcmshell.la libkcminit_main.la kcminit.la
 
-kcontrol_la_SOURCES = main.cpp toplevel.cpp indexwidget.cpp searchwidget.cpp \
+libkcontrol_main_la_SOURCES = main.cpp toplevel.cpp indexwidget.cpp searchwidget.cpp \
 	helpwidget.cpp modules.cpp proxywidget.cpp \
 	dockcontainer.cpp aboutwidget.cpp global.cpp \
 	quickhelp.cpp moduletreeview.cpp moduleiconview.cpp \
 	kcrootonly.cpp moduleIface.cpp moduleIface.skel
 
-kcontrol_la_LIBADD  = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils 
+libkcontrol_main_la_LIBADD  = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils 
+libkcontrol_main_la_LDFLAGS = $(all_libraries)  -avoid-version
+
+kcontrol_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcontrol_la_main.cpp
+
+kcontrol_la_SOURCES = kcontrol_la_main.cpp
+kcontrol_la_LIBADD  = libkcontrol_main.la
 kcontrol_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kcmshell_la_SOURCES = kcmshell.cpp proxywidget.cpp \
+
+libkcmshell_main_la_SOURCES = kcmshell.cpp proxywidget.cpp \
 	                  kcdialog.cpp global.cpp kcdialog.skel
-kcmshell_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils 
+libkcmshell_main_la_LIBADD  = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils 
+libkcmshell_main_la_LDFLAGS = $(all_libraries) -avoid-version
+
+kcmshell_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcmshell_la_main.cpp
+
+kcmshell_la_SOURCES = kcmshell_la_main.cpp
+kcmshell_la_LIBADD  = libkcmshell_main.la
 kcmshell_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
+libkcminit_main_la_SOURCES = kcminit.cpp global.cpp
+libkcminit_main_la_LIBADD   = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils
+libkcminit_main_la_LDFLAGS = $(all_libraries) -avoid-version
 
-kcminit_la_SOURCES = kcminit.cpp global.cpp
-kcminit_la_LIBADD   = $(LIB_KDEUI) $(LIB_KSYCOCA) -lkutils
+kcminit_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcminit_la_main.cpp
+
+kcminit_la_SOURCES = kcminit_la_main.cpp
+kcminit_la_LIBADD  = libkcminit_main.la
 kcminit_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
-kcontrol_SOURCES = dummy.cpp
-kcontrol_LDADD = kcontrol.la
+kcontrol_SOURCES = kcontrol_main.cpp
+kcontrol_LDADD   = libkcontrol_main.la
 kcontrol_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic
 
-kcmshell_SOURCES = dummy.cpp
-kcmshell_LDADD = kcmshell.la
+kcontrol_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcontrol_main.cpp
+
+kcmshell_SOURCES = kcmshell_main.cpp
+kcmshell_LDADD   = libkcmshell_main.la
 kcmshell_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic
 
-kcminit_SOURCES = dummy.cpp
-kcminit_LDADD = kcminit.la
+kcmshell_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcmshell_main.cpp
+
+kcminit_SOURCES = kcminit_main.cpp
+kcminit_LDADD   = libkcminit_main.la
 kcminit_LDFLAGS = $(all_libraries) $(KDE_RPATH) -export-dynamic
 
+kcminit_main.cpp: stub_main.cpp
+	cat stub_main.cpp > kcminit_main.cpp
+
 rccontroldir = $(kde_datadir)/kcontrol
 rccontrol_DATA = kcontrolui.rc
 rcinfodir = $(kde_datadir)/kinfocenter
@@ -79,6 +109,3 @@
 
 messages: rc.cpp
 	$(XGETTEXT) *.cpp -o $(podir)/kcontrol.pot
-
-dummy.cpp:
-	echo > dummy.cpp
