--- kdm/backend/client.c	6 May 2003 17:50:39 -0000	1.1.1.5
+++ kdm/backend/client.c	6 May 2003 18:23:16 -0000	1.4
@@ -38,6 +38,7 @@
  * user verification and session initiation.
  */
 
+#include <config.h>
 #include "dm.h"
 #include "dm_auth.h"
 #include "dm_error.h"
@@ -59,7 +60,11 @@
 # endif
 #endif
 #ifdef USE_PAM
-# include <security/pam_appl.h>
+#ifdef HAVE_PAM_PAM_APPL_H
+#include <pam/pam_appl.h>
+#else
+#include <security/pam_appl.h>
+#endif
 #elif defined(AIXV3) /* USE_PAM */
 # include <login.h>
 # include <usersec.h>
