Index: konqueror/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/konqueror/Makefile.am,v
retrieving revision 1.217
diff -u -b -u -r1.217 Makefile.am
--- konqueror/Makefile.am	9 Apr 2003 12:48:28 -0000	1.217
+++ konqueror/Makefile.am	17 Jun 2003 18:21:39 -0000
@@ -5,7 +5,7 @@
 #SUBDIRS = . kfmexec client iconview dirtree listview keditbookmarks shellcmdplugin about pics sidebar
 SUBDIRS = . kfmexec client iconview listview keditbookmarks shellcmdplugin about pics sidebar preloader
 
-lib_LTLIBRARIES = konqueror.la
+lib_LTLIBRARIES = konqueror.la libkonqueror_main.la
 
 konqdatadir = $(kde_datadir)/konqueror
 konqdata_DATA = konqueror.rc
@@ -22,8 +22,7 @@
 
 bin_PROGRAMS = konqueror
 
-konqueror_la_SOURCES = konq_main.cc \
-	    KonquerorIface.cc KonquerorIface.skel \
+libkonqueror_main_la_SOURCES = KonquerorIface.cc KonquerorIface.skel \
 	    KonqMainWindowIface.cc KonqMainWindowIface.skel \
 	    KonqViewIface.cc KonqViewIface.skel \
 	    konq_guiclients.cc \
@@ -37,7 +36,8 @@
 	    konq_combo.cc	\
 	    konq_browseriface.cc \
 	    delayedinitializer.cc \
-	    konq_mainwindow.cc 
+	    konq_mainwindow.cc \
+	    konq_main.cc
 
 noinst_HEADERS = KonqMainWindowIface.h KonqViewIface.h delayedinitializer.h \
 	konq_actions.h konq_browseriface.h konq_combo.h konq_factory.h \
@@ -45,11 +45,15 @@
 	konq_misc.h konq_openurlrequest.h konq_profiledlg.h konq_run.h \
 	konq_view.h konq_viewmgr.h version.h
 
-konqueror_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-konqueror_la_LIBADD = ../libkonq/libkonq.la $(LIBMALLOC)
+libkonqueror_main_la_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+libkonqueror_main_la_LIBADD = ../libkonq/libkonq.la $(LIBMALLOC)
 
-konqueror_SOURCES = dummy.cc
-konqueror_LDADD = konqueror.la
+konqueror_la_SOURCES = konqueror_la_main.cc
+konqueror_la_LIBADD  = libkonqueror_main.la
+konqueror_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module
+
+konqueror_SOURCES = konqueror_main.cc
+konqueror_LDADD   = libkonqueror_main.la
 konqueror_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 # Hmm, this experiment of a static konq failed, don't trust it...
@@ -64,8 +68,8 @@
 #konqueror_static_LDADD = konqueror.la *view/*.la dirtree/*.la $(libdir)/libkhtml.la
 #konqueror_static_LDFLAGS = $(all_libraries) -static
 
-dummy.cc: $(srcdir)/Makefile.am
-	echo "extern \"C\" int kdemain( int argc, char **argv ); int main( int argc, char **argv ) { return kdemain( argc, argv ); }" > dummy.cc
+konqueror_la_main.cc konqueror_main.cc: $(srcdir)/Makefile.am
+	echo "extern \"C\" int kdemain( int argc, char **argv ); int main( int argc, char **argv ) { return kdemain( argc, argv ); }" > $@
 
 messages: rc.cpp
 	$(EXTRACTRC) *.rc */*.rc >> rc.cpp
