Index: startkde
===================================================================
RCS file: /home/kde/kdebase/startkde,v
retrieving revision 1.116
diff -u -u -r1.116 startkde
--- startkde	29 Jun 2003 17:11:28 -0000	1.116
+++ startkde	4 Jul 2003 02:11:19 -0000
@@ -9,6 +9,7 @@
 
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
 unset DYLD_FORCE_FLAT_NAMESPACE
+test -d "/opt/local" && export PATH="/opt/local/bin:${PATH}"
 
 # Boot sequence:
 #
@@ -116,6 +117,11 @@
     export GS_LIB
 fi
 
+# if we're in Apple's X11 and the native window manager is available, use that
+if test -z "$KDEWM" && test -x /usr/X11R6/bin/quartz-wm; then
+    KDEWM="/usr/X11R6/bin/quartz-wm"
+fi
+
 # Link "tmp" resource to directory in /tmp
 # Creates a directory /tmp/kde-$USER and links $KDEHOME/tmp-$HOSTNAME to it.
 lnusertemp tmp >/dev/null
@@ -138,7 +144,11 @@
     # start only dcopserver, don't start whole kdeinit (takes too long)
     echo 'startkde: Running kpersonalizer...'  1>&2
     dcopserver
-    kwin &
+    if test -n "$KDEWM"; then
+        $KDEWM &
+    else
+        kwin &
+    fi
     kpersonalizer --before-session
     # handle kpersonalizer restarts (language change)
     while test $? -eq 1; do
@@ -149,7 +159,6 @@
     sleep 1
 fi
 
-# the splashscreen and progress indicator
 ksplash
 
 # We set LD_BIND_NOW to increase the efficiency of kdeinit.
@@ -160,6 +169,10 @@
   echo 'startkde: Could not start kdeinit. Check your installation.'  1>&2
   xmessage -geometry 500x100 "Could not start kdeinit. Check your installation."
 fi
+
+# fix up KDE themes
+echo 'startkde: updating theme cache'
+kwrapper kinstalltheme
 
 # finally, give the session control to the session manager
 # if the KDEWM environment variable has been set, then it will be used as KDE's
Index: admin/acinclude.m4.in
===================================================================
RCS file: /home/kde/kde-common/admin/acinclude.m4.in,v
retrieving revision 2.372
diff -u -u -r2.372 acinclude.m4.in
--- admin/acinclude.m4.in	30 Jun 2003 11:18:09 -0000	2.372
+++ admin/acinclude.m4.in	4 Jul 2003 02:11:22 -0000
@@ -3611,18 +3611,12 @@
       AC_CACHE_VAL(ac_cv_path_pam,
         [ use_pam=no
           AC_CHECK_LIB(pam, pam_start,
-            [ AC_CHECK_HEADER(security/pam_appl.h,
-                [ use_pam=yes
-		  pam_header=security/pam_appl.h
-                  pam_service=kde ],
-		[ AC_CHECK_HEADER(pam/pam_appl.h,
-			[ use_pam=yes
-			  pam_header=pam/pam_appl.h
-			  pam_service=kde
-    			  AC_DEFINE(HAVE_PAM_PAM_APPL_H, 1, [Define if your pam headers are in pam/ instead of security/])
-			])
-		]) 
-            ], , $LIBDL)
+            [ AC_CHECK_HEADERS(security/pam_appl.h pam/pam_appl.h)
+              if test "$ac_cv_header_security_pam_appl_h" = "yes" || test "$ac_cv_header_pam_pam_appl_h" = "yes"; then
+                use_pam=yes
+                pam_service=kde
+              fi
+          ], , $LIBDL)
           ac_cv_path_pam="use_pam=$use_pam pam_service=$pam_service"
         ])
     ])
@@ -3646,6 +3640,12 @@
                         AC_DEFINE(PAM_MESSAGE_NONCONST, 1, [Define if your PAM support takes non-const arguments (Solaris)])]
                         )],
       [AC_MSG_RESULT([not found - assume const, Linux-type PAM])])
+
+    AC_CHECK_HEADERS(pam/pam_appl.h)
+    if test "$ac_cv_header_pam_pam_appl_h" = "yes"; then
+      AC_DEFINE(HAVE_PAM_PAM_APPL_H, 1, [Define if your pam headers are in pam/ instead of security/])
+    fi
+
   fi
 
   AC_SUBST(PAMLIBS)
Index: kioslave/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kioslave/Makefile.am,v
retrieving revision 1.63
diff -u -u -r1.63 Makefile.am
--- kioslave/Makefile.am	23 Jun 2003 16:41:52 -0000	1.63
+++ kioslave/Makefile.am	4 Jul 2003 02:11:22 -0000
@@ -15,5 +15,5 @@
 endif
 
 
-SUBDIRS = about cgi floppy filter fish info mac man nfs nntp pop3 imap4 smtp \
+SUBDIRS = about cgi floppy filter fish info mac man nntp pop3 imap4 smtp \
   sftp tar finger thumbnail $(LDAP_SUBDIR) $(SMB_SUBDIR) $(DEVICES_SUBDIR)
Index: kioslave/smtp/smtp.cc
===================================================================
RCS file: /home/kde/kdebase/kioslave/smtp/smtp.cc,v
retrieving revision 1.137
diff -u -u -r1.137 smtp.cc
--- kioslave/smtp/smtp.cc	10 May 2003 01:56:52 -0000	1.137
+++ kioslave/smtp/smtp.cc	4 Jul 2003 02:11:22 -0000
@@ -109,7 +109,7 @@
 
 unsigned int SMTPProtocol::sendBufferSize() const {
   // ### how much is eaten by SSL/TLS overhead?
-  const int fd = ::fileno( fp );
+  const int fd = fileno( fp );
   int value = -1;
   ksize_t len = sizeof(value);
   if ( fd < 0 || ::getsockopt( fd, SOL_SOCKET, SO_SNDBUF, (char*)&value, &len ) )
