--- kicker/proxy/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ kicker/proxy/Makefile.am	1 Jun 2003 14:31:55 -0000	1.4
@@ -3,32 +3,48 @@
 CLEANFILES = dummy.cpp
 
 bin_PROGRAMS = appletproxy extensionproxy
-lib_LTLIBRARIES = appletproxy.la extensionproxy.la
+lib_LTLIBRARIES = libappletproxy_main.la appletproxy.la libextensionproxy_main.la extensionproxy.la
 
-appletproxy_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
-appletproxy_la_SOURCES = appletproxy.cpp
-appletproxy_la_METASOURCES = appletproxy.moc
-appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+libappletproxy_main_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
+libappletproxy_main_la_SOURCES = appletproxy.cpp
+libappletproxy_main_la_METASOURCES = appletproxy.moc
+libappletproxy_main_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+
+appletproxy_la_LIBADD  = libappletproxy_main.la
+appletproxy_la_SOURCES = appletproxy_la_main.cpp
+appletproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -module
 
-appletproxy_LDADD = appletproxy.la
-appletproxy_SOURCES = dummy.cpp
+appletproxy_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > appletproxy_la_main.cpp
+
+appletproxy_LDADD = libappletproxy_main.la
+appletproxy_SOURCES = appletproxy_main.cpp
 appletproxy_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
-extensionproxy_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
-extensionproxy_la_SOURCES = extensionproxy.cpp
-extensionproxy_la_METASOURCES = extensionproxy.moc
-extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+appletproxy_main.cpp: stub_main.cpp
+	cat stub_main.cpp > appletproxy_main.cpp
+
+libextensionproxy_main_la_LIBADD = $(LIB_KDEUI) ../share/libkickermain.la
+libextensionproxy_main_la_SOURCES = extensionproxy.cpp
+libextensionproxy_main_la_METASOURCES = extensionproxy.moc
+libextensionproxy_main_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+
+extensionproxy_la_LIBADD = libextensionproxy_main.la
+extensionproxy_la_SOURCES = extensionproxy_la_main.cpp
+extensionproxy_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -module
+
+extensionproxy_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > extensionproxy_la_main.cpp
 
-extensionproxy_LDADD = extensionproxy.la
-extensionproxy_SOURCES = dummy.cpp
+extensionproxy_LDADD = libextensionproxy_main.la
+extensionproxy_SOURCES = extensionproxy_main.cpp
 extensionproxy_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+extensionproxy_main.cpp: stub_main.cpp
+	cat stub_main.cpp > extensionproxy_main.cpp
+
 noinst_HEADERS = appletproxy.h extensionproxy.h
 
 messages:
 	$(XGETTEXT) appletproxy.cpp -o $(podir)/appletproxy.pot
 	$(XGETTEXT) extensionproxy.cpp -o $(podir)/extensionproxy.pot
-
-dummy.cpp:
-	echo > dummy.cpp
-
