Index: kmenuedit/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kmenuedit/Makefile.am,v
retrieving revision 1.33
diff -u -u -r1.33 Makefile.am
--- kmenuedit/Makefile.am	12 Jul 2002 17:52:04 -0000	1.33
+++ kmenuedit/Makefile.am	27 Jun 2003 21:10:37 -0000
@@ -1,7 +1,6 @@
 INCLUDES = $(all_includes)
 
-bin_PROGRAMS = kmenuedit
-lib_LTLIBRARIES = kmenuedit.la
+kdeinit_LTLIBRARIES = kmenuedit.la
 
 CLEANFILES = dummy.cpp
 
@@ -9,10 +8,6 @@
 
 kmenuedit_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO)
 kmenuedit_la_LDFLAGS = $(all_libraries) -module -avoid-version
-
-kmenuedit_SOURCES = dummy.cpp
-kmenuedit_LDADD   = kmenuedit.la
-kmenuedit_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 noinst_HEADERS = kmenuedit.h menueditview.h treeview.h desktopfileeditor.h basictab.h khotkeys.h
 
