--- konqueror/sidebar/trees/history_module/Makefile.am	1 Oct 2002 02:36:50 -0000	1.1.1.3
+++ konqueror/sidebar/trees/history_module/Makefile.am	26 Nov 2002 21:00:53 -0000	1.7
@@ -7,7 +7,7 @@
 konq_sidebartree_history_la_SOURCES = history_module.cpp history_item.cpp \
 	history_settings.cpp history_settings.skel
 konq_sidebartree_history_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-konq_sidebartree_history_la_LIBADD =  ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_history_la_LIBADD =  $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/libkonq_sidebar_tree.la
 
 kcm_history_la_SOURCES = kcmhistory.cpp history_dlg.ui history_settings.cpp history_settings.skel
 kcm_history_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
