Index: kdm/kfrontend/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kdm/kfrontend/Makefile.am,v
retrieving revision 1.32
diff -u -b -u -r1.32 Makefile.am
--- kdm/kfrontend/Makefile.am	18 May 2003 20:39:42 -0000	1.32
+++ kdm/kfrontend/Makefile.am	17 Jun 2003 03:27:16 -0000
@@ -6,7 +6,7 @@
 
 INCLUDES = -I$(srcdir)/../backend -I$(top_srcdir)/kcontrol/background $(all_includes)
 
-bin_PROGRAMS = kdm_config kdm_greet krootimage
+bin_PROGRAMS = kdm_config kdm_greet krootimage genkdmconf
 
 kdm_config_SOURCES = kdm_config.c
 kdm_config_LDADD = $(LIBRESOLV) $(LIBSOCKET)
@@ -37,10 +37,6 @@
 genkdmconf_SOURCES = genkdmconf.c
 genkdmconf_LDFLAGS = $(X_LDFLAGS) $(X_RPATH)
 genkdmconf_LDADD = $(LIB_X11)
-
-install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
