--- kwin/clients/keramik/Makefile.am	1 Oct 2002 02:37:32 -0000	1.1.1.2
+++ kwin/clients/keramik/Makefile.am	26 Nov 2002 21:00:54 -0000	1.2
@@ -13,7 +13,7 @@
 kde_module_LTLIBRARIES    = kwin_keramik.la
 
 kwin_keramik_la_SOURCES   = keramik.cpp
-kwin_keramik_la_LIBADD    = ../../kwin.la
+kwin_keramik_la_LIBADD    = ../../libkwin_main.la
 kwin_keramik_la_COMPILE_FIRST = tiles.h
 kwin_keramik_la_LDFLAGS   = $(all_libraries) $(KDE_PLUGIN) -module
 
