Index: kicker/extensions/childpanel/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kicker/extensions/childpanel/Makefile.am,v
retrieving revision 1.9
diff -u -u -r1.9 Makefile.am
--- kicker/extensions/childpanel/Makefile.am	29 Dec 2002 17:10:53 -0000	1.9
+++ kicker/extensions/childpanel/Makefile.am	27 Jun 2003 22:06:17 -0000
@@ -4,8 +4,10 @@
 
 childpanel_panelextension_la_SOURCES = childpanelextension.cpp childpanelextension.skel
 childpanel_panelextension_la_METASOURCES = AUTO
-childpanel_panelextension_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-childpanel_panelextension_la_LIBADD = $(LIB_KDEUI) $(LIB_KSYCOCA) ../../kicker.la
+childpanel_panelextension_la_LDFLAGS = -module $(KDE_RPATH) $(all_libraries)
+childpanel_panelextension_la_LIBADD = ../../core/libkicker_core.la  ../../buttons/libkicker_buttons.la \
+				../../ui/libkicker_ui.la ../../share/libkickermain.la $(LIB_KIO) \
+				$(XTESTLIB) $(LIB_KSYCOCA) $(LIB_KDEUI)
 
 noinst_HEADERS = childpanelextension.h
 
