--- khotkeys/khotkeys/Makefile.am	25 Jun 2002 02:03:52 -0000	1.1.1.1
+++ khotkeys/khotkeys/Makefile.am	1 Jun 2003 14:31:54 -0000	1.6
@@ -1,15 +1,25 @@
 # $Id: Makefile.am,v 1.12 2002/02/12 14:32:08 ellis Exp $
 
 bin_PROGRAMS = khotkeys
-lib_LTLIBRARIES = khotkeys.la
+lib_LTLIBRARIES = khotkeys.la libkhotkeys_main.la
 
-khotkeys_la_SOURCES = khotkeys.cpp main.cpp khotkeys.skel
-khotkeys_la_LIBADD   = ../shared/libshared.la $(LIB_KSYCOCA)
-khotkeys_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkhotkeys_main_la_SOURCES = khotkeys.cpp main.cpp khotkeys.skel
+libkhotkeys_main_la_LIBADD  = ../shared/libshared.la $(LIB_KSYCOCA) $(LIB_QT) -lkdecore -lDCOP
+libkhotkeys_main_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
 
-khotkeys_SOURCES = dummy.cpp
-khotkeys_LDADD   = khotkeys.la
+khotkeys_la_SOURCES = khotkeys_la_main.cpp
+khotkeys_la_LIBADD  = libkhotkeys_main.la
+khotkeys_la_LDFLAGS = $(all_libraries) -avoid-version -module
+
+khotkeys_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > khotkeys_la_main.cpp
+
+khotkeys_SOURCES = khotkeys_main.cpp
+khotkeys_LDADD = libkhotkeys_main.la
 khotkeys_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+
+khotkeys_main.cpp: stub_main.cpp
+	cat stub_main.cpp > khotkeys_main.cpp
 
 autostart_DATA = khotkeys.desktop
 autostartdir = $(prefix)/share/autostart
