Index: kioslave/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kioslave/Makefile.am,v
retrieving revision 1.62
diff -u -b -u -r1.62 Makefile.am
--- kioslave/Makefile.am	11 Apr 2003 08:46:48 -0000	1.62
+++ kioslave/Makefile.am	17 Jun 2003 03:33:21 -0000
@@ -1,5 +1,9 @@
 ## $Id: Makefile.am,v 1.62 2003/04/11 08:46:48 coolo Exp $
 
+if include_kioslave_nfs
+NFS_SUBDIR=nfs
+endif
+
 if include_kioslave_ldap
 LDAP_SUBDIR=ldap
 endif
@@ -15,5 +19,5 @@
 endif
 
 
-SUBDIRS = about cgi floppy filter fish info mac man nfs nntp pop3 imap4 smtp \
+SUBDIRS = about cgi floppy filter fish info mac man $(NFS_SUBDIR) nntp pop3 imap4 smtp \
   sftp tar zip finger thumbnail $(LDAP_SUBDIR) $(SMB_SUBDIR) $(DEVICES_SUBDIR)
