--- kdeprint/kprinter/Makefile.am	10 Jul 2002 03:20:23 -0000	1.1.1.2
+++ kdeprint/kprinter/Makefile.am	1 Jun 2003 14:31:54 -0000	1.7
@@ -1,21 +1,28 @@
 INCLUDES = $(all_includes)
 
 bin_PROGRAMS = kprinter
-lib_LTLIBRARIES = kprinter.la
+lib_LTLIBRARIES = kprinter.la libkprinter_main.la
 
-kprinter_SOURCES = dummy.cpp
-kprinter_LDADD = kprinter.la
+libkprinter_main_la_SOURCES = printwrapper.cpp
+libkprinter_main_la_LIBADD  = -lkdeprint $(LIB_QT) -lkdecore -lDCOP -lkdeui -lkio
+libkprinter_main_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH) -no-undefined
+libkprinter_main_la_METASOURCES = AUTO
+
+kprinter_SOURCES = main.cpp
+kprinter_LDADD   = libkprinter_main.la
 kprinter_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
-kprinter_la_SOURCES = main.cpp printwrapper.cpp
-kprinter_la_LIBADD = -lkdeprint
-kprinter_la_LDFLAGS = -module -avoid-version $(all_libraries) $(KDE_RPATH)
-kprinter_METASOURCES = AUTO
+kprinter_la_SOURCES = main_la.cpp
+kprinter_la_LIBADD  = libkprinter_main.la
+kprinter_la_LDFLAGS = -avoid-version $(all_libraries) $(KDE_RPATH) -module
+
+main_la.cpp: main.cpp
+	cat main.cpp > main_la.cpp
 
 noinst_HEADERS = printwrapper.h
 
 messages:
 	$(XGETTEXT) *.cpp -o $(podir)/kprinter.pot
 
-dummy.cpp:
-	echo > dummy.cpp
+dummy.cpp: main.cpp
+	cat main.cpp > dummy.cpp
