Index: kwin/Makefile.am
===================================================================
RCS file: /home/kde/kdebase/kwin/Makefile.am,v
retrieving revision 1.52
diff -u -b -u -r1.52 Makefile.am
--- kwin/Makefile.am	9 Apr 2003 12:48:29 -0000	1.52
+++ kwin/Makefile.am	17 Jun 2003 03:50:47 -0000
@@ -3,14 +3,22 @@
 SUBDIRS = . kcmkwin pics clients
 
 bin_PROGRAMS = kwin
-lib_LTLIBRARIES = kwin.la
+lib_LTLIBRARIES = libkwin_main.la
+kde_module_LTLIBRARIES = kwin.la
 
 # workspace.cpp has to be first in order not to break --enable-final
-kwin_la_SOURCES = workspace.cpp client.cpp placement.cpp atoms.cpp  main.cpp \
+libkwin_main_la_SOURCES = workspace.cpp client.cpp placement.cpp atoms.cpp  main.cpp \
 	popupinfo.cpp tabbox.cpp options.cpp plugins.cpp events.cpp KWinInterface.skel \
 	killwindow.cpp kwinbutton.cpp geometrytip.cpp
-kwin_la_LIBADD = $(LIB_KDEUI)
-kwin_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkwin_main_la_LIBADD = $(LIB_KDEUI)
+libkwin_main_la_LDFLAGS = $(all_libraries) -avoid-version
+
+kwin_la_SOURCES = kwin_la_main.cpp
+kwin_la_LIBADD  = libkwin_main.la
+kwin_la_LDFLAGS = $(all_libraries) -avoid-version -module
+
+kwin_la_main.cpp: kwin_main.cpp
+	cat kwin_main.cpp > kwin_la_main.cpp
 
 include_HEADERS = KWinInterface.h
 
@@ -18,7 +26,7 @@
 kwininclude_HEADERS = options.h client.h workspace.h kwinbutton.h
 
 kwin_SOURCES = kwin_main.cpp
-kwin_LDADD = kwin.la
+kwin_LDADD   = libkwin_main.la
 kwin_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 KDE_ICON = kwin
