--- configure.ac.orig	2016-09-21 03:07:06.000000000 -0700
+++ configure.ac	2016-09-21 08:24:32.000000000 -0700
@@ -30,7 +30,7 @@
 AC_SUBST([install_sizes], ["8x8 16x16 22x22 24x24 32x32 48x48 64x64 96x96 256x256 512x512"])
 
 AC_PROG_LN_S
-AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache)
+AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache-3.0)
 
 # need git, icontool, and inkscape for rendering
 allow_rendering=yes
