diff -urN a/make.default b/make.default
--- a/make.default	2017-07-24 19:28:20.000000000 +0100
+++ b/make.default	2023-10-15 16:46:26.000000000 +0100
@@ -19,7 +19,6 @@
 #
 
 # for System V, using new_curse with terminfo
-DEFINES =	-DSYS5 -DNCURSE 
 
 # for BSD, using new_curse with termcap
 #DEFINES = 	-DCAP -DNCURSE 
@@ -28,7 +27,7 @@
 #DEFINES =      -DCAP -DNCURSE -DBSD_SELECT 
 
 # flags for compilation
-CFLAGS = -s -DNO_CATGETS
+CFLAGS = -DHAS_NCURSES -DHAS_STDLIB -DHAS_SYS_WAIT -DHAS_UNISTD
 
 # For Sun systems, remove the '#' from the front of the next two lines:
 #DEFINES = -DSYS5 -DNCURSE 
@@ -37,7 +36,7 @@
 all :	ee
 
 curses :	ee.c
-	cc ee.c -o ee $(CFLAGS) -lcurses 
+	$(CC) ee.c -o ee $(CFLAGS) -lncurses 
 
 ee :	ee.o new_curse.o
 	cc -o ee ee.o new_curse.o $(CFLAGS) 
