--- src/parser.y.orig	2006-07-24 04:03:43.000000000 +1000
+++ src/parser.y	2012-06-11 02:55:49.000000000 +1000
@@ -137,6 +137,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include "shape.h"
 #include "boxes.h"
 #include "tools.h"
--- src/parser.c.orig	2006-07-24 04:03:43.000000000 +1000
+++ src/parser.c	2012-06-11 02:56:30.000000000 +1000
@@ -240,6 +240,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include "shape.h"
 #include "boxes.h"
 #include "tools.h"
