--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -184,8 +184,10 @@

         target_compile_definitions(glfw INTERFACE GLFW_DLL)
     elseif (APPLE)
-        set_target_properties(glfw PROPERTIES
-                              INSTALL_NAME_DIR "${CMAKE_INSTALL_LIBDIR}")
+        if (NOT CMAKE_INSTALL_NAME_DIR)
+            set_target_properties(glfw PROPERTIES
+                INSTALL_NAME_DIR "${CMAKE_INSTALL_LIBDIR}")
+        endif()
     endif()

     if (MINGW)
