--- fpcsrc/packages/fpmake_proc.inc	2020-09-09 23:22:17.000000000 +0800
+++ fpcsrc/packages/fpmake_proc.inc	2023-08-06 06:45:20.000000000 +0800
@@ -779,12 +779,6 @@
 
 {$include odata/fpmake.pp}
 
-procedure add_pastojs(const ADirectory: string);
-begin
-  with Installer do
-{$include pastojs/fpmake.pp}
-end;
-
 procedure add_libgc(const ADirectory: string);
 begin
   with Installer do

--- fpcsrc/packages/fpmake_add.inc	2020-09-09 23:22:17.000000000 +0800
+++ fpcsrc/packages/fpmake_add.inc	2023-08-06 07:00:00.000000000 +0800
@@ -136,7 +136,6 @@
   add_Google(ADirectory+IncludeTrailingPathDelimiter('googleapi'));
   add_fcl_pdf(ADirectory+IncludeTrailingPathDelimiter('fcl-pdf'));
   add_odata(ADirectory+IncludeTrailingPathDelimiter('odata'));
-  add_pastojs(ADirectory+IncludeTrailingPathDelimiter('pastojs'));
   add_libgc(ADirectory+IncludeTrailingPathDelimiter('libgc'));
   add_libfontconfig(ADirectory+IncludeTrailingPathDelimiter('libfontconfig'));
   add_fcl_report(ADirectory+IncludeTrailingPathDelimiter('fcl-report'));

--- fpcsrc/utils/fpmake_proc.inc	2018-08-10 15:12:49.000000000 +0800
+++ fpcsrc/utils/fpmake_proc.inc	2023-08-06 07:22:01.000000000 +0800
@@ -35,6 +35,3 @@
 {$include tply/fpmake.pp}
 
 {$include unicode/fpmake.pp}
-
-{$include pas2js/fpmake.pp}
-

--- fpcsrc/utils/fpmake_add.inc	2018-08-10 15:12:49.000000000 +0800
+++ fpcsrc/utils/fpmake_add.inc	2023-08-06 07:21:47.000000000 +0800
@@ -17,4 +17,3 @@
   add_rmwait(ADirectory+IncludeTrailingPathDelimiter('rmwait'));
   add_tply(ADirectory+IncludeTrailingPathDelimiter('tply'));
   add_unicode(ADirectory+IncludeTrailingPathDelimiter('unicode'));
-  add_pas2js(ADirectory+IncludeTrailingPathDelimiter('pas2js'));
