--- config.sample.orig	2008-01-08 05:56:45.000000000 -0700
+++ config.sample	2008-03-10 00:43:10.000000000 -0600
@@ -2,7 +2,7 @@
 
 # You should not need to use a configuration file; all configuration
 # variables have reasonable defaults.  If you want to use one, you
-# can copy this to /etc/polipo/config or to ~/.polipo and modify.
+# can copy this to @@PREFIX@@/etc/polipo/config or to ~/.polipo and modify.
 
 # This file only contains some of the configuration variables; see the
 # list given by ``polipo -v'' and the manual for more.
