--- forbidden.c.orig	2008-01-08 05:56:45.000000000 -0700
+++ forbidden.c	2008-03-10 00:44:17.000000000 -0600
@@ -304,8 +304,8 @@
     }
 
     if(forbiddenFile == NULL) {
-        if(access("/etc/polipo/forbidden", F_OK) >= 0)
-            forbiddenFile = internAtom("/etc/polipo/forbidden");
+        if(access("@@PREFIX@@/etc/polipo/forbidden", F_OK) >= 0)
+            forbiddenFile = internAtom("@@PREFIX@@/etc/polipo/forbidden");
     }
 
     parseDomainFile(forbiddenFile, &forbiddenDomains, &forbiddenRegex);
@@ -325,8 +325,8 @@
     }
 
     if(uncachableFile == NULL) {
-        if(access("/etc/polipo/uncachable", F_OK) >= 0)
-            uncachableFile = internAtom("/etc/polipo/uncachable");
+        if(access("@@PREFIX@@/etc/polipo/uncachable", F_OK) >= 0)
+            uncachableFile = internAtom("@@PREFIX@@/etc/polipo/uncachable");
     }
 
     parseDomainFile(uncachableFile, &uncachableDomains, &uncachableRegex);
