--- configure	2024-09-05 13:04:35.000000000 +0800
+++ configure	2024-09-06 14:17:31.000000000 +0800
@@ -4344,7 +4344,7 @@
   if test ${TBB_EXISTS} = yes ; then
     TBBFLAGS="-DTBB $(pkg-config --cflags tbb@BIT@)"
 
-    TBBLIBS="$(pkg-config --libs tbb@BIT@)"
+    TBBLIBS="$(pkg-config --libs tbb@BIT@) -latomic"
 
 
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking tbb available for compiling and linking:" >&5
