--- inkscapeExtensions/jessyInk_uninstall.py.orig	2009-04-09 16:31:07.000000000 -0700
+++ inkscapeExtensions/jessyInk_uninstall.py	2009-04-09 16:37:34.000000000 -0700
@@ -15,13 +15,13 @@
 
 # These two lines are only needed if you don't put the script directly into
 # the installation directory
-import sys
+#import sys
 # Unix
-sys.path.append('/usr/share/inkscape/extensions')
+#sys.path.append('/usr/share/inkscape/extensions')
 # OS X
-sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
+#sys.path.append('/Applications/Inkscape.app/Contents/Resources/extensions')
 # Windows
-sys.path.append('C:\Program Files\Inkscape\share\extensions')
+#sys.path.append('C:\Program Files\Inkscape\share\extensions')
 
 # We will use the inkex module with the predefined Effect base class.
 import inkex
