--- src/LoLi2/include/loli_obj.h.orig	2016-07-01 10:02:12.000000000 +0800
+++ src/LoLi2/include/loli_obj.h	2023-01-26 22:33:21.000000000 +0800
@@ -24,6 +24,8 @@
 #include <typeinfo>
 #include <iostream>
 
+typedef unsigned long ulong;
+
 //Macros
 #define lnum(x) 	((loliNum*)x)
 #define lint(x) 	((loliInt*)x)

--- src/LoLi2/Makefile.orig	2016-07-01 10:02:12.000000000 +0800
+++ src/LoLi2/Makefile	2023-01-26 22:31:56.000000000 +0800
@@ -1,7 +1,6 @@
-CC = g++
-CLANG = clang++
-CFLAGSG = -std=c++11 -stdlib=libstdc++ -Wall -Og -Wextra -g
-CFLAGSC = -std=c++11 -stdlib=libstdc++ -Wall -O0 -Wextra -g
+CXX = @CXX@
+CFLAGSG = @ARCHFLAGS@ -std=gnu++11 -Wall -Og -Wextra -g
+CFLAGSC = @ARCHFLAGS@ -std=c++11 -Wall -O0 -Wextra -g
 
 HEADERS = \
 	  include/loli_obj.h\
@@ -45,13 +44,13 @@
        loli2test.cpp
 
 CURRENTG: 
-	$(CC) $(CFLAGSG) -o currentg $(MAIN)
+	$(CXX) $(CFLAGSG) -o currentg $(MAIN) -lstdc++
 
 CURRENTC:
-	$(CLANG) $(CFLAGSC) -o currentc $(MAIN)
+	$(CXX) $(CFLAGSC) -o currentc $(MAIN) -lc++
 
 TEST:	$(HEADERS)
-	$(CLANG) $(CFLAGSC) $(GCINCLUDE) $(GC) -o test $(TEST)
+	$(CXX) $(CFLAGSC) $(GCINCLUDE) $(GC) -o test $(TEST)
 
 .PHONY: clean
 clean:
