--- bushnell.cc.orig	2017-01-01 22:05:19.000000000 -0700
+++ bushnell.cc	2017-10-01 04:42:33.000000000 -0700
@@ -136,7 +136,7 @@
     name = "Waypoint";
   }
 
-  for (t = bushnell_icons; t->icon > 0; t++) {
+  for (t = bushnell_icons; t->icon > (void*)0; t++) {
     if (0 == name.compare(t->icon, Qt::CaseInsensitive)) {
       return t->symbol;
     }
@@ -148,7 +148,7 @@
 bushnell_get_name_from_symbol(signed int s)
 {
   icon_mapping_t* t;
-  for (t = bushnell_icons; t->icon > 0; t++) {
+  for (t = bushnell_icons; t->icon > (void*)0; t++) {
     if (s == t->symbol) {
       return t->icon;
     }
