--- deprecated/delbin.cc.orig
+++ deprecated/delbin.cc
@@ -2614,7 +2614,7 @@ thread_func(void* run_loop_source)
 }
 
 static void
-interrupt_report_cb(void* target, IOReturn result, void* refcon, void* sender, UInt32 bufferSize)
+interrupt_report_cb(void* target, IOReturn result, void* refcon, void* sender, uint32_t bufferSize)
 {
   memcpy(packet_array[packet_array_head], report_buf, delbin_os_packet_size);
   pthread_mutex_lock(&mutex);
