--- src/gtk-about_dialog.ads.orig	2009-12-08 17:25:54.000000000 +0100
+++ src/gtk-about_dialog.ads	2009-12-08 17:21:45.000000000 +0100
@@ -228,11 +228,11 @@
    procedure Set_Name
      (About : access Gtk_About_Dialog_Record; Name  : String)
      renames Set_Program_Name;
-   pragma Obsolescent; --  Set_Name
+   --  pragma Obsolescent; --  Set_Name
    function Get_Name
      (About : access Gtk_About_Dialog_Record) return String
      renames Get_Program_Name;
-   pragma Obsolescent; --  Get_Name
+   --  pragma Obsolescent; --  Get_Name
    --  Returns the program name displayed in the about dialog.
 
    ----------------
