--- src/lib/corelib/tools/tools.pri.orig	2018-03-27 04:40:50.000000000 -0700
+++ src/lib/corelib/tools/tools.pri	2018-04-15 11:17:08.000000000 -0700
@@ -124,6 +124,7 @@
         $$PWD/generateoptions.h \
         $$PWD/generatorpluginmanager.h \
         $$PWD/installoptions.h \
+	$$PWD/projectgeneratormanager.h \
         $$PWD/qbspluginmanager.h \
         $$PWD/setupprojectparameters.h \
         $$PWD/toolchains.h \
