https://github.com/yaml/pyyaml/pull/823

--- pyproject.toml.orig	2024-08-07 06:16:54
+++ pyproject.toml	2025-04-13 12:50:16
@@ -1,7 +1,6 @@
 [build-system]
 requires = [
     "setuptools",  # FIXME: declare min/max setuptools versions?
-    "wheel",
     "Cython; python_version < '3.13'",
     "Cython>=3.0; python_version >= '3.13'"
 ]
--- setup.py.orig	2024-08-07 06:16:54
+++ setup.py	2025-04-13 13:29:21
@@ -93,12 +93,7 @@
     if with_cython:
         raise
 
-try:
-    from wheel.bdist_wheel import bdist_wheel
-except ImportError:
-    bdist_wheel = None
 
-
 try:
     from _pyyaml_pep517 import ActiveConfigSettings
 except ImportError:
@@ -324,8 +319,6 @@
     'build_ext': build_ext,
     'test': test,
 }
-if bdist_wheel:
-    cmdclass['bdist_wheel'] = bdist_wheel
 
 
 if __name__ == '__main__':
