--- sources/iTermApplicationDelegate.m.orig
+++ sources/iTermApplicationDelegate.m
@@ -198,7 +198,6 @@
     IBOutlet NSMenuItem *showFullScreenTabs;
     IBOutlet NSMenuItem *useTransparency;
     IBOutlet NSMenuItem *maximizePane;
-    IBOutlet SUUpdater * suUpdater;
     IBOutlet NSMenuItem *_showTipOfTheDay;  // Here because we must remove it for older OS versions.
     BOOL quittingBecauseLastWindowClosed_;
 
@@ -1547,10 +1546,6 @@
 
     [[iTermBuriedSessions sharedInstance] setMenus:[NSArray arrayWithObjects:_buriedSessions, _statusIconBuriedSessions, nil]];
 
-    item = [[[NSMenuItem alloc] initWithTitle:@"Check For Updates"
-                                       action:@selector(checkForUpdatesFromMenu:)
-                                keyEquivalent:@""] autorelease];
-    [menu addItem:item];
 
     NSMenuItem *mainMenuItem = [[[NSMenuItem alloc] initWithTitle:@"Main Menu" action:nil keyEquivalent:@""] autorelease];
     mainMenuItem.submenu = [[NSApp mainMenu] it_deepCopy];
--- sources/iTermPreferences.m.orig
+++ sources/iTermPreferences.m
@@ -473,7 +473,7 @@
                   kPreferenceKeySavePasteAndCommandHistory: @NO,
                   kPreferenceKeyAddBonjourHostsToProfiles: @NO,
                   kPreferenceKeyNotifyOnlyForCriticalShellIntegrationUpdates: @YES,
-                  kPreferenceKeyCheckForUpdatesAutomatically: @YES,
+                  kPreferenceKeyCheckForUpdatesAutomatically: @NO,
                   kPreferenceKeyCheckForTestReleases: @NO,
                   kPreferenceKeyLoadPrefsFromCustomFolder: @NO,
                   kPreferenceKeyUseCustomScriptsFolder: @NO,
