From a793c88dcfdcd17c42909d8f3ed461bc44aaa53e Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sat, 7 Nov 2020 01:32:23 -0500
Subject: [PATCH] Guard against NS_WARN_UNUSED_RESULT not defined

---
 sources/iTerm2.pch | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git sources/iTerm2.pch sources/iTerm2.pch
index 3108a3a09..028db831c 100644
--- sources/iTerm2.pch
+++ sources/iTerm2.pch
@@ -5,4 +5,6 @@ _Pragma("clang diagnostic ignored \"-Wpartial-availability\"")
 #define ITERM_IGNORE_PARTIAL_END \
 _Pragma("clang diagnostic pop")
 
-
+#ifndef NS_WARN_UNUSED_RESULT
+#define NS_WARN_UNUSED_RESULT
+#endif
-- 
2.29.2

