--- src/llvm-project/llvm/include/llvm/ADT/Triple.h.orig	2021-11-16 09:47:26.000000000 -0700
+++ src/llvm-project/llvm/include/llvm/ADT/Triple.h	2022-03-07 19:38:27.000000000 -0700
@@ -919,7 +919,7 @@
 
   /// Tests whether the target uses emulated TLS as default.
   bool hasDefaultEmulatedTLS() const {
-    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment();
+    return true;
   }
 
   /// Tests whether the target uses -data-sections as default.
--- compiler/rustc_target/src/spec/apple_base.rs.orig	2022-02-22 21:07:21.000000000 -0700
+++ compiler/rustc_target/src/spec/apple_base.rs	2022-04-08 22:29:00.000000000 -0700
@@ -122,7 +122,7 @@
     // TLS is flagged as enabled if it looks to be supported. The architecture
     // only matters for default deployment target which is 11.0 for ARM64 and
     // 10.7 for everything else.
-    let has_thread_local = os == "macos" && macos_deployment_target(Arch::X86_64) >= (10, 7);
+    let has_thread_local = false;
 
     let abi = arch.target_abi();
 
