--- src/llvm-project/llvm/include/llvm/ADT/Triple.h.orig	2021-11-16 09:47:26.000000000 -0700
+++ src/llvm-project/llvm/include/llvm/ADT/Triple.h	2022-03-07 19:38:27.000000000 -0700
@@ -854,7 +854,7 @@
 
   /// Tests whether the target uses emulated TLS as default.
   bool hasDefaultEmulatedTLS() const {
-    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment();
+    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment() || isMacOSXVersionLT(10, 7);
   }
 
   /// Tests whether the target uses -data-sections as default.
