--- src/bootstrap/src/core/build_steps/llvm.rs.orig	2023-12-21 09:55:28
+++ src/bootstrap/src/core/build_steps/llvm.rs	2024-01-01 09:17:49
@@ -575,6 +575,8 @@
     mut ldflags: LdFlags,
     extra_compiler_flags: &[&str],
 ) {
+    cfg.define("CMAKE_LIBTOOL", "__MACPORTS_PREFIX__/bin/libtool");
+
     // Do not print installation messages for up-to-date files.
     // LLVM and LLD builds can produce a lot of those and hit CI limits on log size.
     cfg.define("CMAKE_INSTALL_MESSAGE", "LAZY");
