See https://github.com/llvm/llvm-project/commit/a5e10e248efc72e7909e4067060e89c35a456a18
--- src/llvm-project/llvm/lib/TargetParser/Host.cpp.orig	2023-11-12 21:11:27.000000000 -0700
+++ src/llvm-project/llvm/lib/TargetParser/Host.cpp	2024-01-06 08:02:51.000000000 -0700
@@ -1912,6 +1912,9 @@
 #elif defined(__x86_64__)
   T.setArch(Triple::x86_64);
   T.setArchName("x86_64");
+#elif defined(__i386__)
+  T.setArch(Triple::x86);
+  T.setArchName("i386");
 #elif defined(__powerpc__)
   T.setArch(Triple::ppc);
   T.setArchName("powerpc");
