--- yt_dlp/update.py.orig	2021-09-24 19:59:48.000000000 -0500
+++ yt_dlp/update.py	2021-10-05 19:07:48.000000000 -0500
@@ -104,7 +104,7 @@
         'zip': None,
         'dir': 'Auto-update is not supported for unpackaged windows executable. Re-download the latest release',
         'source': 'You cannot update when running from source code',
-        'unknown': 'It looks like you installed yt-dlp with a package manager, pip, setup.py or a tarball. Use that to update',
+        'unknown': 'It looks like you installed yt-dlp with MacPorts. Please use "sudo port selfupdate && sudo port upgrade yt-dlp" to update.',
     }
     err = ERRORS.get(detect_variant(), ERRORS['unknown'])
     if err:
