--- yt_dlp/update.py.orig	2021-09-01 19:02:31.000000000 -0500
+++ yt_dlp/update.py	2021-09-23 05:22:14.000000000 -0500
@@ -93,7 +93,7 @@
     elif hasattr(sys, 'frozen'):
         pass
     else:
-        err = 'It looks like you installed yt-dlp with a package manager, pip, setup.py or a tarball. Please use that to update'
+        err = 'It looks like you installed yt-dlp with MacPorts. Please use "sudo port selfupdate && sudo port upgrade yt-dlp" to update.'
     if err:
         return report_error(err, expected=True)
 
