--- include/plugins.php.org	2008-02-25 09:47:35.000000000 -0800
+++ include/plugins.php	2008-02-25 09:48:12.000000000 -0800
@@ -32,13 +32,7 @@
     }
 }
 
-if (isset($_SERVER['DOCUMENT_ROOT']) && isset($_SERVER['REMOTE_ADDR'])) {
-	$config['url_path'] = substr(__FILE__, strlen($_SERVER['DOCUMENT_ROOT']), strlen(__FILE__) - strlen($_SERVER['DOCUMENT_ROOT']) - strlen('include/plugins.php'));
-	db_execute("REPLACE INTO settings (name, value) VALUES ('url_path', '" . $config['url_path'] . "')");
-} else {
-	$config['url_path'] = db_fetch_cell("SELECT value FROM settings WHERE name = 'url_path'");
-}
-
+$config['url_path'] = '/cacti/';
 define('URL_PATH', $config['url_path']);
 
 
