--- ref_gl/r_main.c	(revision 1999)
+++ ref_gl/r_main.c	(working copy)
@@ -1185,7 +1185,7 @@
 	gl_flashblend = Cvar_Get ("gl_flashblend", "0", CVAR_ARCHIVE);
 	gl_playermip = Cvar_Get ("gl_playermip", "0", 0);
 #ifdef __unix__
-	gl_driver = Cvar_Get( "gl_driver", "libGL.so.1", CVAR_ARCHIVE );
+	gl_driver = Cvar_Get( "gl_driver", "@PREFIX@/lib/libGL.dylib", CVAR_ARCHIVE );
 #else
 	gl_driver = Cvar_Get( "gl_driver", "opengl32", CVAR_ARCHIVE );
 #endif
