--- unix/q_shunix.c	(revision 1999)
+++ unix/q_shunix.c	(working copy)
@@ -35,7 +35,15 @@
 #ifndef __linux__
 /* For round_page() macro. */
 #include <machine/param.h>
+#ifndef round_page
+size_t round_page (size_t size)
+{
+    int pagesize = sysconf(_SC_PAGESIZE);
+    // This is wrong (too large) when size == 0, oh well
+    return (1 + (size - 1) / pagesize) * pagesize;
+}
 #endif
+#endif
 
 //===============================================================================
 
