--- lib/wx/configure.in.orig
+++ lib/wx/configure.in
@@ -545,10 +545,10 @@ AC_CHECK_TYPES([GLintptr, GLintptrARB, GLchar,
 	       [#ifdef WIN32
 		# include <windows.h>
 	        # include <gl/gl.h>
-	        #elif defined(HAVE_GL_GL_H)
-	        # include <GL/gl.h>
                 #elif defined(HAVE_OPENGL_GL_H)
 		# include <OpenGL/gl.h>
+	        #elif defined(HAVE_GL_GL_H)
+	        # include <GL/gl.h>
 		#endif 
 		])
  
@@ -563,10 +563,10 @@ AC_TRY_COMPILE([
                 #ifdef WIN32
 		# include <windows.h>
 	        # include <gl/glu.h>
-	        #elif defined(HAVE_GL_GL_H)
-	        # include <GL/glu.h>
                 #elif defined(HAVE_OPENGL_GL_H)
 		# include <OpenGL/glu.h>
+	        #elif defined(HAVE_GL_GL_H)
+	        # include <GL/glu.h>
 		#endif 
 		#ifndef CALLBACK
 		# define CALLBACK
@@ -613,10 +613,10 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
 		#ifdef WIN32
 		# include <windows.h>
 	        # include <gl/gl.h>
-	        #elif defined(HAVE_GL_GL_H)
-	        # include <GL/gl.h>
                 #elif defined(HAVE_OPENGL_GL_H)
 		# include <OpenGL/gl.h>
+	        #elif defined(HAVE_GL_GL_H)
+	        # include <GL/gl.h>
 		#endif 
                 #include "wx/wx.h"
 		#include "wx/stc/stc.h"
