--- erts/emulator/hipe/hipe_ppc_bifs.m4.orig	2011-12-14 04:22:11.000000000 -0600
+++ erts/emulator/hipe/hipe_ppc_bifs.m4	2011-12-25 13:18:53.000000000 -0600
@@ -32,12 +32,12 @@
 	.text
 	.p2align 2
 
-define(TEST_GOT_MBUF,`LOAD r4, P_MBUF(P)	# `TEST_GOT_MBUF'
+define(TEST_GOT_MBUF,`LOAD r4, P_MBUF(P)	; `TEST_GOT_MBUF'
 	CMPI r4, 0
 	bne- 3f
 2:')
 define(HANDLE_GOT_MBUF,`
-3:	bl CSYM(nbif_$1_gc_after_bif)	# `HANDLE_GOT_MBUF'
+3:	bl CSYM(nbif_$1_gc_after_bif)	; `HANDLE_GOT_MBUF'
 	b 2b')
 
 
@@ -62,7 +62,7 @@
 
 	/* Save caller-save registers and call the C function. */
 	SAVE_CONTEXT_BIF
-	STORE	r4, P_ARG0(r3)		# Store BIF__ARGS in def_arg_reg[]
+	STORE	r4, P_ARG0(r3)		; Store BIF__ARGS in def_arg_reg[]
 	addi	r4, r3, P_ARG0
 	CALL_BIF($2)
 	TEST_GOT_MBUF
@@ -92,7 +92,7 @@
 
 	/* Save caller-save registers and call the C function. */
 	SAVE_CONTEXT_BIF
-	STORE	r4, P_ARG0(r3)		# Store BIF__ARGS in def_arg_reg[]
+	STORE	r4, P_ARG0(r3)		; Store BIF__ARGS in def_arg_reg[]
 	STORE	r5, P_ARG1(r3)
 	addi	r4, r3, P_ARG0
 	CALL_BIF($2)
@@ -124,7 +124,7 @@
 
 	/* Save caller-save registers and call the C function. */
 	SAVE_CONTEXT_BIF
-	STORE	r4, P_ARG0(r3)		# Store BIF__ARGS in def_arg_reg[]
+	STORE	r4, P_ARG0(r3)		; Store BIF__ARGS in def_arg_reg[]
 	STORE	r5, P_ARG1(r3)
 	STORE	r6, P_ARG2(r3)
 	addi	r4, r3, P_ARG0
@@ -214,7 +214,7 @@
 
 	/* Save caller-save registers and call the C function. */
 	SAVE_CONTEXT_GC
-	STORE	r4, P_ARG0(r3)		# Store BIF__ARGS in def_arg_reg[]
+	STORE	r4, P_ARG0(r3)		; Store BIF__ARGS in def_arg_reg[]
 	addi	r4, r3, P_ARG0
 	CALL_BIF($2)
 	TEST_GOT_MBUF
@@ -244,7 +244,7 @@
 
 	/* Save caller-save registers and call the C function. */
 	SAVE_CONTEXT_GC
-	STORE	r4, P_ARG0(r3)		# Store BIF__ARGS in def_arg_reg[]
+	STORE	r4, P_ARG0(r3)		; Store BIF__ARGS in def_arg_reg[]
 	STORE	r5, P_ARG1(r3)
 	addi	r4, r3, P_ARG0
 	CALL_BIF($2)
