--- lib/erl_interface/src/decode/decode_big.c.orig	2008-11-06 06:57:02.000000000 +0100
+++ lib/erl_interface/src/decode/decode_big.c	2008-11-06 06:58:39.000000000 +0100
@@ -51,7 +51,9 @@
       u = (unsigned char *) s;
       for (i = 0; i < b->arity; ++i) {
 	  dt[i] = u[i*2];
-	  dt[i] |= ((unsigned short) u[(i*2)+1]) << 8;
+        if ((i*2 + 1) < digit_bytes) {
+          dt[i] |= ((unsigned short) u[(i*2)+1]) << 8;
+        }
       }
   } else {
       s++; /* skip sign byte */
