--- setup.py.orig	2026-02-18 08:50:20
+++ setup.py	2026-03-17 16:45:29
@@ -264,13 +264,11 @@
             is_apple_clang = False
 
         if is_apple_clang:
-            ompcompileflags = ['-Xclang', '-fopenmp']
-            omplinkflags = ['-Xclang', '-fopenmp', '-liomp5']
+            ompcompileflags = ['-I__OPENMP_INC__', '-Xclang', '-fopenmp']
+            omplinkflags = ['-L__OPENMP_LIB__', '-Xclang', '-fopenmp', '-liomp5']
         else:
             ompcompileflags = ['-fopenmp']
             omplinkflags = ['-fopenmp=libiomp5']
-        omppath = ['lib', 'clang', '*', 'include', 'omp.h']
-        have_openmp = check_file_at_path(omppath)
     else:
         cpp11flags = ['-std=c++11']
         ompcompileflags = ['-fopenmp']
