diff --git libcxx-5.0.1.src/include/math.h libcxx-5.0.1.src/include/math.h
index 8c30ba8..023943c 100644
--- libcxx-5.0.1.src/include/math.h
+++ libcxx-5.0.1.src/include/math.h
@@ -300,6 +300,32 @@ long double    truncl(long double x);
 
 #include_next <math.h>
 
+#ifdef __APPLE__
+# if __has_include(<Availability.h>)
+#  include <Availability.h>
+#  if __MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+#    define __APPLE_BAD_MATH_H 1
+#  else
+#    define __APPLE_BAD_MATH_H 0
+#  endif
+# else
+#  define __APPLE_BAD_MATH_H 1
+# endif
+
+# if __APPLE_BAD_MATH_H
+/* These prototypes are incorrectly omitted from <math.h> on Snow Leopard despite being available */
+extern "C" {
+    extern long long int llrintl(long double);
+    extern long long int llrint(double);
+    extern long long int llrintf(float);
+
+    extern long long int llroundl(long double);
+    extern long long int llround(double);
+    extern long long int llroundf(float);
+}
+# endif
+#endif // __APPLE__
+
 #ifdef __cplusplus
 
 // We support including .h headers inside 'extern "C"' contexts, so switch
 