diff --git libcxxabi-5.0.1.src/lib/buildit.orig libcxxabi-5.0.1.src/lib/buildit
index 3690482..a1c3f4e 100755
--- libcxxabi-5.0.1.src/lib/buildit.orig
+++ libcxxabi-5.0.1.src/lib/buildit
@@ -85,6 +85,9 @@ set -x
 for FILE in ../src/*.cpp; do
 	$CXX -c -g -O3 $RC_CFLAGS $EXTRA_FLAGS -I../include $OPTIONS $FILE
 done
+for FILE in ../src/*.c; do
+	$CC -c -g -Os $RC_CFLAGS -I../include $FILE
+done
 case $TRIPLE in
   *-*-mingw*)
   for FILE in ../src/support/win32/*.cpp; do
