diff --git libcxxabi-5.0.1.src/include/cxxabi.h libcxxabi-5.0.1.src/include/cxxabi.h
index 2596560..8092e73 100644
--- libcxxabi-5.0.1.src/include/cxxabi.h
+++ libcxxabi-5.0.1.src/include/cxxabi.h
@@ -160,12 +160,12 @@ __cxa_decrement_exception_refcount(void *primary_exception) throw();
 extern _LIBCXXABI_FUNC_VIS bool __cxa_uncaught_exception() throw();
 extern _LIBCXXABI_FUNC_VIS unsigned int __cxa_uncaught_exceptions() throw();
 
-#ifdef __linux__
+//#ifdef __linux__
 // Linux TLS support. Not yet an official part of the Itanium ABI.
 // https://sourceware.org/glibc/wiki/Destructor%20support%20for%20thread_local%20variables
 extern _LIBCXXABI_FUNC_VIS int __cxa_thread_atexit(void (*)(void *), void *,
                                                    void *) throw();
-#endif
+//#endif
 
 } // extern "C"
 } // namespace __cxxabiv1
