--- libcxx-5.0.1.src/lib/buildit	2018-12-23 22:13:07.000000000 -0500
+++ /opt/local/var/macports/sources/rsync.macports.org/release/tarballs/ports/lang/libcxx/work/libcxx-5.0.1.src/lib/buildit	2018-12-23 22:08:51.000000000 -0500
@@ -55,38 +55,24 @@
       RC_CFLAGS="-arch i386 -arch x86_64"
     fi
     SOEXT=dylib
-    if [ "$MACOSX_DEPLOYMENT_TARGET" = "10.6" ]
+
+    if [ -z "${LIBCXXABI_DYLIB_PATH}" ]
+    then
+        LIBCXXABI_DYLIB_PATH=/usr/lib/libc++abi.dylib
+    fi
+
+    if [ "$MACOSX_DEPLOYMENT_TARGET" = "10.5" ] || [ "$MACOSX_DEPLOYMENT_TARGET" = "10.6" ]
     then
-        EXTRA_FLAGS="-nostdinc++ -std=c++11 -U__STRICT_ANSI__"
+        EXTRA_FLAGS="${EXTRA_FLAGS} -U__STRICT_ANSI__"
         LDSHARED_FLAGS="-o libc++.1.dylib \
-            -dynamiclib -nodefaultlibs -current_version 1 \
+            -dynamiclib -nodefaultlibs \
+            -current_version ${RC_ProjectSourceVersion} \
             -compatibility_version 1 \
             -install_name /usr/lib/libc++.1.dylib \
-            -Wl,-reexport_library,/usr/lib/libc++abi.dylib \
+            -Wl,-reexport_library,${LIBCXXABI_DYLIB_PATH} \
             -Wl,-unexported_symbols_list,libc++unexp.exp  \
-            /usr/lib/libSystem.B.dylib"
+            /usr/lib/libSystem.B.dylib /usr/lib/libgcc_s.1.dylib"
     else
-        if [ -n "$SDKROOT" ]
-        then
-            EXTRA_FLAGS+="-isysroot ${SDKROOT} "
-            if echo "${RC_ARCHS}" | grep -q "armv7"  
-            then
-                RE_EXPORT_LINE="${SDKROOT}/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++sjlj-abi.exp"
-            else
-                RE_EXPORT_LINE="-Wl,-reexport_library,${SDKROOT}/usr/lib/libc++abi.dylib"
-            fi
-            CXX=`xcrun -sdk "${SDKROOT}"  -find clang++`
-            CC=`xcrun -sdk "${SDKROOT}"  -find clang`
-        else
-            # Check if we have _LIBCPPABI_VERSION, to determine the reexport list to use.
-            if (echo "#include <cxxabi.h>" | $CXX -E -dM -x c++ - | \
-                grep _LIBCPPABI_VERSION > /dev/null)
-            then
-                RE_EXPORT_LINE="/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++abi2.exp"
-            else
-                RE_EXPORT_LINE="/usr/lib/libc++abi.dylib -Wl,-reexported_symbols_list,libc++abi.exp"
-            fi
-        fi
         LDSHARED_FLAGS="-o libc++.1.dylib \
             -dynamiclib -nodefaultlibs  \
             -current_version ${RC_ProjectSourceVersion} \
@@ -94,7 +80,8 @@
             -install_name /usr/lib/libc++.1.dylib \
             -lSystem  \
             -Wl,-unexported_symbols_list,libc++unexp.exp  \
-            ${RE_EXPORT_LINE}  \
+            -Wl,-reexport_library,${LIBCXXABI_DYLIB_PATH} \
+            -Wl,-reexported_symbols_list,libc++abi2.exp \
             -Wl,-force_symbols_not_weak_list,notweak.exp \
             -Wl,-force_symbols_weak_list,weak.exp"
     fi
