diff --git libcxxabi-5.0.1.src/lib/buildit libcxxabi-5.0.1.src/lib/buildit
index 5a4a710..e912845 100755
--- libcxxabi-5.0.1.src/lib/buildit
+++ libcxxabi-5.0.1.src/lib/buildit
@@ -27,12 +27,12 @@ then
   RC_ProjectSourceVersion=1
 fi
 
-EXTRA_FLAGS="-std=c++11 -stdlib=libc++ -fstrict-aliasing -Wstrict-aliasing=2 \
+EXTRA_FLAGS="-nostdinc++ -std=c++11 -stdlib=libc++ -fstrict-aliasing -Wstrict-aliasing=2 \
              -Wsign-conversion -Wshadow -Wconversion -Wunused-variable \
              -Wmissing-field-initializers -Wchar-subscripts -Wmismatched-tags \
              -Wmissing-braces -Wshorten-64-to-32 -Wsign-compare \
              -Wstrict-aliasing=2 -Wstrict-overflow=4 -Wunused-parameter \
-             -Wnewline-eof"
+             -Wnewline-eof -D_LIBCPP_BUILDING_LIBRARY"
 
 case $TRIPLE in
   *-apple-*)
@@ -52,6 +52,7 @@ case $TRIPLE in
         -current_version ${RC_ProjectSourceVersion} \
         -compatibility_version 1 \
         -install_name /usr/lib/libc++abi.dylib \
+        -Wl,-undefined,dynamic_lookup \
         -lSystem"
 	if [ -f "${SDKROOT}/usr/local/lib/libCrashReporterClient.a" ]
 	then
@@ -91,7 +91,7 @@ case $TRIPLE in
   done
   ;;
 esac
-$CC *.o $RC_CFLAGS $LDSHARED_FLAGS $EXTRA_FLAGS
+$CC *.o $RC_CFLAGS $EXTRA_LDFLAGS $LDSHARED_FLAGS $EXTRA_FLAGS
 
 if [ -z $RC_XBS ]
 then
