--- sccs/sccs/cmd/src/sccs.c.orig	2019-01-04 15:21:20.000000000 +0400
+++ sccs/sccs/cmd/src/sccs.c	2020-09-02 17:28:28.000000000 +0400
@@ -449,7 +449,7 @@
 #ifdef XPG4
 /*static char path[] = NOGETTEXT("PATH=/usr/xpg4/bin:/usr/ccs/bin:/usr/bin");*/
 #ifdef	PROTOTYPES
-static char path[] = NOGETTEXT("PATH=" INS_BASE "/xpg4/bin:" INS_BASE "/ccs/bin:/usr/bin");
+static char path[] = NOGETTEXT("PATH=" INS_BASE "/xpg4/bin:" INS_BASE "/" SCCS_BIN_PRE "libexec/sccs:" INS_BASE "/" SCCS_BIN_PRE "bin:/usr/bin");
 #else
 /*
  * XXX With a K&R compiler, you need to edit the following string in case
