--- sccs/sccs/cmd/src/help.c.orig	2018-04-29 16:36:57.000000000 +0400
+++ sccs/sccs/cmd/src/help.c	2020-09-02 14:39:53.000000000 +0400
@@ -169,16 +169,6 @@
 	strlcat(help_dir, helpdir, sizeof (help_dir));
 	strlcat(help_dir, locale, sizeof (help_dir));
 
-	/*
-	 * The text of the printf statement below should not be wrapped
-	 * with gettext().  Since we don't know what the locale is, we
-	 * don't know how to get the proper translation text.
-	 */
-	if (stat(help_dir, &Statbuf) != 0) { /* Does help directory exist? */
-		printf(
-		    NOGETTEXT("Unrecognized locale... setting to English\n"));
-	}
-
 	if (argc == 1) {
 		char	*he = ask();
 		if (*he == '\0') {
