--- lib/expat_external.h.orig	2026-10-06 01:54:00
+++ lib/expat_external.h	2026-10-11 08:31:54
@@ -128,9 +128,12 @@
 
 /* Marks a function that Expat still provides but that callers should move off
    of. */
-#  if defined(__clang__) || defined(__GNUC__)
+#  if defined(__clang__) || (defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)))
 #    define XML_ATTR_DEPRECATED(message)                                       \
       __attribute__((__deprecated__(message)))
+#  elif defined(__GNUC__)
+#    define XML_ATTR_DEPRECATED(message)                                       \
+      __attribute__((__deprecated__))
 #  elif defined(_MSC_VER)
 #    define XML_ATTR_DEPRECATED(message) __declspec(deprecated(message))
 #  else
