--- cmake/OpenCVFindOpenEXR.cmake.orig	2020-04-02 14:59:58.000000000 -0500
+++ cmake/OpenCVFindOpenEXR.cmake	2020-04-25 13:31:42.000000000 -0500
@@ -21,15 +21,11 @@
         SET(OPENEXR_LIBSEARCH_SUFFIXES Win32/Release Win32 Win32/Debug)
     endif()
 elseif(UNIX)
-    SET(OPENEXR_LIBSEARCH_SUFFIXES ${CMAKE_LIBRARY_ARCHITECTURE})
+    set(OPENEXR_ROOT "@PREFIX@")
 endif()
 
 SET(SEARCH_PATHS
     "${OPENEXR_ROOT}"
-    /usr
-    /usr/local
-    /sw
-    /opt
     "${ProgramFiles_ENV_PATH}/OpenEXR")
 
 MACRO(FIND_OPENEXR_LIBRARY LIBRARY_NAME LIBRARY_SUFFIX)
