--- config.m4.orig	2009-12-24 21:59:32.000000000 +0100
+++ config.m4	2009-12-24 21:59:42.000000000 +0100
@@ -13,18 +13,5 @@
   dnl default to little-endian
 
   CFLAGS="$CFLAGS -DWall -g"
-
-  case $build_os in
-  darwin10.*.*)
-    AC_MSG_CHECKING([whether to compile for recent osx architectures])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64"
-    AC_MSG_RESULT([yes])
-    ;;
-  darwin*)
-    AC_MSG_CHECKING([whether to compile for every osx architecture ever])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64 -arch ppc -arch ppc64"
-    AC_MSG_RESULT([yes])
-    ;;
-  esac
 fi
 
