--- config.m4.orig	2011-02-07 15:12:47.000000000 -0600
+++ config.m4	2011-02-08 21:44:41.000000000 -0600
@@ -18,17 +18,5 @@
   fi
   dnl default to little-endian
 
-  case $build_os in
-  darwin10.*.*)
-    AC_MSG_CHECKING([whether to compile for recent osx architectures])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64 -mmacosx-version-min=10.5"
-    AC_MSG_RESULT([yes])
-    ;;
-  darwin*)
-    AC_MSG_CHECKING([whether to compile for every osx architecture ever])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64 -arch ppc -arch ppc64"
-    AC_MSG_RESULT([yes])
-    ;;
-  esac
 fi
 
