--- config.m4.orig	1970-01-01 03:13:08.000000000 -0600
+++ config.m4	2009-12-11 01:47:21.000000000 -0600
@@ -15,18 +15,5 @@
   dnl extensions prerequisites
   PHP_ADD_EXTENSION_DEP(mongo, date, false)
   PHP_ADD_EXTENSION_DEP(mongo, spl, false)
-
-  case $build_os in
-  darwin10.*.*)
-    AC_MSG_CHECKING([whether to compile for recent osx architectures])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64"
-    AC_MSG_RESULT([yes])
-    ;;
-  darwin*)
-    AC_MSG_CHECKING([whether to compile for every osx architecture ever])
-    CFLAGS="$CFLAGS -arch i386 -arch x86_64 -arch ppc -arch ppc64"
-    AC_MSG_RESULT([yes])
-    ;;
-  esac
 fi
 
