--- src/Makefile.orig	2005-01-27 12:27:28.000000000 +0100
+++ src/Makefile	2009-03-25 15:09:45.000000000 +0100
@@ -84,7 +84,7 @@
           $(if $(OCAMLOPT),*.cmxa sdl*.$(A) *.cmx) \
           $(if $(OCAMLMKLIB),dll*.so)
 ifeq ($(PLATFORM),Apple)
-	$(RANLIB) $$($(OCAMLFIND) printconf destdir)/*.$(A)
+	$(RANLIB) $$($(OCAMLFIND) query sdl)/*.$(A)
 endif
 else
 install-findlib : ; $(error "findlib is not installed")
