--- amavisd.orig	2015-12-12 08:47:21.000000000 -0800
+++ amavisd	2015-12-12 08:59:37.000000000 -0800
@@ -1091,7 +1091,7 @@
   $file = 'file';  # path to the file(1) utility for classifying contents
   $altermime = 'altermime';  # path to the altermime utility (optional)
   @altermime_args_defang     = qw(--verbose --removeall);
-  @altermime_args_disclaimer = qw(--disclaimer=/etc/altermime-disclaimer.txt);
+  @altermime_args_disclaimer = qw(--disclaimer=@@PREFIX@@/etc/amavisd/altermime-disclaimer.txt);
   # @altermime_args_disclaimer =
   #  qw(--disclaimer=/etc/_OPTION_.txt --disclaimer-html=/etc/_OPTION_.html);
   # @disclaimer_options_bysender_maps = ( 'altermime-disclaimer' );
@@ -18579,7 +18579,7 @@
 
 # default location of the config file if none specified
 if (!@config_files) {
-  @config_files = ( '/etc/amavisd.conf' );
+  @config_files = ( '@@PREFIX@@/etc/amavisd/amavisd.conf' );
 # # Debian/Ubuntu specific:
 # @config_files = Amavis::Util::find_config_files('/usr/share/amavis/conf.d',
 #                                                 '/etc/amavis/conf.d');
