--- cmake/Modules/FindGLFW3.cmake.orig
+++ cmake/Modules/FindGLFW3.cmake
@@ -1,18 +1,16 @@
 if(NOT GLFW3_FOUND)
   INCLUDE(FindPkgConfig)
-  pkg_check_modules (GLFW3_PKG glfw3)
-  find_path(GLFW3_INCLUDE_DIRS NAMES GLFW/glfw3.h
+  pkg_check_modules(GLFW3_PKG glfw3)
+  find_path(GLFW3_INCLUDE_DIRS
+    NAMES GLFW/glfw3.h
     HINTS ${GLFW3_PKG_INCLUDE_DIRS}
-    PATHS
-    /usr/include
-    /usr/local/include
+    NO_DEFAULT_PATH
   )
 
-  find_library(GLFW3_LIBRARIES NAMES glfw3 glfw
+  find_library(GLFW3_LIBRARIES
+    NAMES ${GLFW3_PKG_LIBRARIES} glfw3 glfw
     HINTS ${GLFW3_PKG_LIBRARY_DIRS}
-    PATHS
-    /usr/lib
-    /usr/local/lib
+    NO_DEFAULT_PATH
   )
 
   if(GLFW3_INCLUDE_DIRS AND GLFW3_LIBRARIES)
