--- certbot_nginx/_internal/constants.py.orig	2019-10-01 15:04:08.000000000 -0500
+++ certbot_nginx/_internal/constants.py	2019-10-22 15:26:28.000000000 -0500
@@ -1,7 +1,7 @@
 """nginx plugin constants."""
 import platform
 
-FREEBSD_DARWIN_SERVER_ROOT = "/usr/local/etc/nginx"
+FREEBSD_DARWIN_SERVER_ROOT = "%PREFIX%/etc/nginx"
 LINUX_SERVER_ROOT = "/etc/nginx"
 
 if platform.system() in ('FreeBSD', 'Darwin'):
