--- certbot/docs/conf.py.orig	2022-07-05 13:16:39.000000000 -0500
+++ certbot/docs/conf.py	2022-08-13 10:40:34.000000000 -0500
@@ -84,7 +84,7 @@
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = 'en'
 
 # There are two options for replacing |today|: either, you set today to some
 # non-false value, then it is used:
