--- certbot/src/certbot/_internal/plugins/nginx/constants.py.orig	2026-04-07 11:38:17
+++ certbot/src/certbot/_internal/plugins/nginx/constants.py	2026-04-08 07:22:36
@@ -2,7 +2,7 @@
 import platform
 from typing import Any
 
-FREEBSD_DARWIN_SERVER_ROOT = "/usr/local/etc/nginx"
+FREEBSD_DARWIN_SERVER_ROOT = "%PREFIX%/etc/nginx"
 LINUX_SERVER_ROOT = "/etc/nginx"
 PKGSRC_SERVER_ROOT = "/usr/pkg/etc/nginx"
 
