--- CMakeLists.txt	2024-09-01 11:23:28.000000000 +0800
+++ CMakeLists.txt	2024-09-29 18:03:13.000000000 +0800
@@ -100,10 +100,6 @@
 # Platform specifics
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   add_compile_definitions(_XOPEN_SOURCE_EXTENDED)
-  list(APPEND CMAKE_PREFIX_PATH /usr/local/opt/ncurses)
-  list(APPEND CMAKE_PREFIX_PATH /opt/homebrew/opt/ncurses)
-  list(APPEND OPENSSL_ROOT_DIR /usr/local/opt/openssl)
-  list(APPEND OPENSSL_ROOT_DIR /opt/homebrew/opt/openssl)
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Android")
   add_compile_definitions(_XOPEN_SOURCE_EXTENDED)
 endif()
