--- configure.old	2017-10-11 18:38:18.000000000 -0700
+++ configure	2017-10-11 18:45:07.000000000 -0700
@@ -14159,14 +14159,14 @@
 fi
 
 else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for /usr/local/opt/readline/lib" >&5
-$as_echo_n "checking for /usr/local/opt/readline/lib... " >&6; }
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for @@MACPORTS_PREFIX@@/lib" >&5
+$as_echo_n "checking for @@MACPORTS_PREFIX@@/lib... " >&6; }
 if ${ac_cv_file__usr_local_opt_readline_lib+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   test "$cross_compiling" = yes &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "/usr/local/opt/readline/lib"; then
+if test -r "@@MACPORTS_PREFIX@@/lib"; then
   ac_cv_file__usr_local_opt_readline_lib=yes
 else
   ac_cv_file__usr_local_opt_readline_lib=no
@@ -14176,8 +14176,8 @@
 $as_echo "$ac_cv_file__usr_local_opt_readline_lib" >&6; }
 if test "x$ac_cv_file__usr_local_opt_readline_lib" = xyes; then :
   LIBS="-lreadline $LIBS"
-            AM_CPPFLAGS="-I/usr/local/opt/readline/include $AM_CPPFLAGS"
-            AM_LDFLAGS="-L/usr/local/opt/readline/lib $AM_LDFLAGS"
+            AM_CPPFLAGS="-I@@MACPORTS_PREFIX@@/include $AM_CPPFLAGS"
+            AM_LDFLAGS="-L@@MACPORTS_PREFIX@@/lib $AM_LDFLAGS"
 
 else
   as_fn_error $? "libreadline is required for profanity" "$LINENO" 5
