--- src/cmdline.cpp.orig	2019-06-27 09:26:20.000000000 -0700
+++ src/cmdline.cpp	2019-06-27 09:31:37.000000000 -0700
@@ -202,8 +202,7 @@
      "Copies the input streams/files into temporary files to perform diffing. "
      "This is useful if you want to diff FIFOs."
    },
-// Hack for os x - programs are run with an argument line -psn_0_36306945
-   { "prompt-for-files", 'p', true, 'p',
+   { "prompt-for-files", 0, false, 'p',
      "If no files are specified on the command line, show a file dialog so that "
      "the user can select them. This option is ignored if any files are specified."
    },
