--- Makefile.orig	2009-09-16 12:21:18.000000000 +0200
+++ Makefile	2009-09-16 12:21:59.000000000 +0200
@@ -15,8 +15,8 @@
 	MACOS_DEPLOYMENT_TARGET="10.3"; export MACOSX_DEPLOYMENT_TARGET; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS)
 
 install:
-	mkdir -p $(LUA_LIBDIR)
-	cp src/lfs.so $(LUA_LIBDIR)
+	mkdir -p $(DESTDIR)$(LUA_LIBDIR)
+	cp src/lfs.so $(DESTDIR)$(LUA_LIBDIR)
 
 clean:
 	rm -f src/lfs.so $(OBJS)
