--- Makefile.am.orig	2022-02-15 01:47:04.000000000 -0800
+++ Makefile.am	2022-02-15 01:44:58.000000000 -0800
@@ -7,7 +7,7 @@
 	src/drcs.h src/convtable.h			\
 	src/decoder_macro.h
 libaribb24_la_LIBADD = $(PNG_LIBS)
-libaribb24_la_CFLAGS = -Wall -fvisibility=hidden
+libaribb24_la_CFLAGS = -Wall -fvisibility=hidden $(PNG_CFLAGS)
 
 pkginclude_HEADERS = src/aribb24/decoder.h src/aribb24/parser.h	\
 	src/aribb24/bits.h src/aribb24/aribb24.h
