Use MacPorts' include files

--- tiny_gltf.h.orig
+++ tiny_gltf.h
@@ -1703,7 +1703,7 @@
 
 #ifndef TINYGLTF_NO_INCLUDE_JSON
 #ifndef TINYGLTF_USE_RAPIDJSON
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 #else
 #ifndef TINYGLTF_NO_INCLUDE_RAPIDJSON
 #include "document.h"
@@ -1722,13 +1722,13 @@
 
 #ifndef TINYGLTF_NO_STB_IMAGE
 #ifndef TINYGLTF_NO_INCLUDE_STB_IMAGE
-#include "stb_image.h"
+#include <stb/stb_image.h>
 #endif
 #endif
 
 #ifndef TINYGLTF_NO_STB_IMAGE_WRITE
 #ifndef TINYGLTF_NO_INCLUDE_STB_IMAGE_WRITE
-#include "stb_image_write.h"
+#include <stb/stb_image_write.h>
 #endif
 #endif
 


--- tests/tester.cc.orig
+++ tests/tester.cc
@@ -4,7 +4,7 @@
 #include "tiny_gltf.h"
 
 // Nlohmann json(include ../json.hpp)
-#include "json.hpp"
+#include <nlohmann/json.hpp>
 
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
 #include "catch.hpp"

