--- DESCRIPTION	2024-05-01 10:52:27.000000000 +0800
+++ DESCRIPTION	2024-05-10 03:54:28.000000000 +0800
@@ -23,7 +23,7 @@
 Imports: utils, stats, Biobase, S4Vectors, GenomicRanges, BiocIO,
         rtracklayer
 Suggests: GenomicFeatures, Rsamtools, testthat, knitr, rmarkdown,
-        BiocStyle, devtools, BSgenome.Celegans.UCSC.ce2
+        BiocStyle, devtools
 VignetteBuilder: knitr
 git_url: https://git.bioconductor.org/packages/BSgenomeForge
 git_branch: RELEASE_3_19
