--- Makefile.in.orig	2011-02-24 11:43:16.000000000 +0100
+++ Makefile.in	2011-02-24 11:52:54.000000000 +0100
@@ -346,12 +346,12 @@
 install-findlib: META
 ifdef OCAMLFIND
 ifeq (@LABLGNOMECANVAS@,yes)
-	$(OCAMLFIND) install ocamlgraph META \
+	$(OCAMLFIND) install $(OCAMLFIND_INSTFLAGS) ocamlgraph META \
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
 		graph$(LIBEXT) graph.cmx graph.cmo graph.cmi $(CMA) $(CMXA) \
 		$(VIEWER_CMXLIB) $(VIEWER_CMOLIB) $(DGRAPH_CMX) $(DGRAPH_CMO)
 else
-	$(OCAMLFIND) install ocamlgraph META \
+	$(OCAMLFIND) install $(OCAMLFIND_INSTFLAGS) ocamlgraph META \
 		$(SRCDIR)/*.mli $(VIEWER_DIR)/*.mli $(DGRAPH_DIR)/*.mli \
 		graph$(LIBEXT) graph.cmx graph.cmo graph.cmi $(CMA) $(CMXA)
 endif
