--- xattr.c	2005-04-13 07:32:41.000000000 +0200
+++ xattr.c.new	2006-04-19 17:32:57.000000000 +0200
@@ -19,8 +19,10 @@
 	fprintf(stdout, "                   delete a key/value attribute pair\n\n");
 	fprintf(stdout, "           --list: [-l]  list all key/value attribute pairs\n\n");
 	fprintf(stdout, " --list-parseable: [-lp] list all key/value attribute pairs in a parseable format\n\n");
-	fprintf(stdout, "            --get: [-g]  print key/value pair\n\n");
-	fprintf(stdout, "  --get-parseable: [-gp] print key/value pair in a parseable format\n\n");
+	fprintf(stdout, "            --get: [-g]  key path\n");
+	fprintf(stdout, "                   print key/value pair\n\n");
+	fprintf(stdout, "  --get-parseable: [-gp] key path\n");
+	fprintf(stdout, "                   print key/value pair in a parseable format\n\n");
 }
 
 int main(int argc, char **argv)
