--- runfiles/texmf-dist/scripts/epspdf/epspdf.rb.orig	2011-08-12 11:59:51.000000000 -0400
+++ runfiles/texmf-dist/scripts/epspdf/epspdf.rb	2011-08-12 12:00:25.000000000 -0400
@@ -33,14 +33,7 @@
 # WARNING
 # readlink apparently only works right from the directory of the symlink
 
-$SCRIPTDIR = File.dirname( File.expand_path( __FILE__ ))
-if RUBY_PLATFORM !~ /win32|mingw/ and File.symlink?( __FILE__ )
-  savedir = Dir.pwd
-  Dir.chdir( $SCRIPTDIR )
-  # puts File.readlink( __FILE__ )
-  $SCRIPTDIR = File.dirname( File.expand_path( File.readlink( __FILE__ )))
-  Dir.chdir( savedir )
-end
+$SCRIPTDIR = "@@TEXMFDIST@@/scripts/epspdf"
 $:.unshift( $SCRIPTDIR )
 # puts $:
 
