--- Makefile.in.orig	2012-11-06 14:54:10.000000000 -0600
+++ Makefile.in	2013-02-18 19:11:49.000000000 -0600
@@ -31,6 +31,8 @@
 
 RECURSIVE_TARGETS = all allbin alldoc tags clean distclean realclean install
 
+.PHONY: Makefiles all allbin alldoc check clean distclean distclean-local distdir install realclean tags tar.bz2 tar.gz
+
 #-------------------------------------------------------------------------
 
 $(RECURSIVE_TARGETS):
