--- libev/ev.c
+++ libev/ev.c
@@ -4094,6 +4094,7 @@ ev_run (EV_P_ int flags)
           {
             waittime = EV_TS_CONST (MAX_BLOCKTIME);
 
+#if EV_USE_MONOTONIC
             if (ecb_expect_true (have_monotonic))
               {
 #if EV_USE_TIMERFD
@@ -4107,6 +4108,7 @@ ev_run (EV_P_ int flags)
                 waittime = EV_TS_CONST (MAX_BLOCKTIME2);
 #endif
               }
+#endif
 
             if (timercnt)
               {
