--- pyproject.toml.orig	1970-01-01 03:00:00
+++ pyproject.toml	2024-11-14 02:55:09.000000000 +0800
@@ -2,13 +2,13 @@
 # Minimum requirements for the build system to execute.
 # See https://github.com/scipy/scipy/pull/12940 for the AIX issue.
 requires = [
-    "meson-python==0.13.1",
-    "meson==1.2.1",
+    "meson-python",
+    "meson",
     "wheel",
-    "Cython~=3.0.5",  # Note: sync with setup.py, environment.yml and asv.conf.json
+    "Cython",  # Note: sync with setup.py, environment.yml and asv.conf.json
     # Force numpy higher than 2.0, so that built wheels are compatible
     # with both numpy 1 and 2
-    "numpy>=2.0",
+    "numpy",
     "versioneer[toml]"
 ]
 
