--- a/data/enchant.ordering.orig	2016-01-24 01:16:28.000000000 +0100
+++ b/data/enchant.ordering	2016-01-24 01:25:21.000000000 +0100
@@ -1,8 +1,8 @@
-*:myspell,aspell,ispell
-fi:voikko,ispell,myspell,aspell
-fi_FI:voikko,ispell,myspell,aspell
-he:hspell,myspell
-he_IL:hspell,myspell
+*:myspell,aspell,ispell,AppleSpell
+fi:voikko,ispell,myspell,aspell,AppleSpell
+fi_FI:voikko,ispell,myspell,aspell,AppleSpell
+he:hspell,myspell,AppleSpell
+he_IL:hspell,myspell,AppleSpell
 yi:uspell
-tr:zemberek
-tr_TR:zemberek
+tr:zemberek,AppleSpell
+tr_TR:zemberek,AppleSpell
