--- makefile.unix.orig	2009-12-28 08:18:59.000000000 +0100
+++ makefile.unix	2009-12-28 08:28:44.000000000 +0100
@@ -100,7 +100,7 @@
 WHAT=UNRAR
 
 UNRAR_OBJ=filestr.o recvol.o rs.o scantree.o
-LIB_OBJ=filestr.o scantree.o dll.o
+LIB_OBJ=filestr.o recvol.o rs.o scantree.o dll.o
 
 OBJECTS=rar.o strlist.o strfn.o pathfn.o savepos.o smallfn.o global.o file.o filefn.o filcreat.o \
 	archive.o arcread.o unicode.o system.o isnt.o crypt.o crc.o rawread.o encname.o \
@@ -132,8 +132,8 @@
 
 lib:	WHAT=RARDLL
 lib:	$(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	@rm -f libunrar.dylib
+	$(LINK) -shared -o libunrar.dylib $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
 			install unrar $(DESTDIR)/bin
@@ -142,7 +142,7 @@
 			rm -f $(DESTDIR)/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.dylib $(DESTDIR)/lib
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/lib/libunrar.dylib
