--- lib/update.js.orig	2021-02-17 17:02:01.000000000 -0800
+++ lib/update.js	2021-02-20 12:50:37.000000000 -0800
@@ -22,6 +22,20 @@
     ? global
     : npm.prefix
 
+  // Prevent npm package in global from updating. Prevent 'npm update -g'
+  // since it'll implicity upgrade npm and also prevent 'npm update -g npm'.
+  if (npm.flatOptions.global) {
+    if (!args.length || (args.length && args.find(e => e === "npm"))) {
+      log.error('update', 'Ignoring attempt to update npm in ' + global);
+      log.error('update', 'which is part of the MacPorts npm7 base');
+      log.error('update', 'installation. To update npm please run:');
+      log.error('update', 'sudo port selfupdate && sudo port upgrade npm7');
+      throw new Error('Ignoring attempt to update npm in ' + global +
+                      ' which is part of the MacPorts npm7 base' +
+                      ' installation.');
+    }
+  }
+
   if (npm.flatOptions.depth) {
     log.warn('update', 'The --depth option no longer has any effect. See RFC0019.\n' +
       'https://github.com/npm/rfcs/blob/latest/implemented/0019-remove-update-depth-option.md')
