--- libvips/Makefile.in.orig	2015-05-05 08:36:38.000000000 -0400
+++ libvips/Makefile.in	2015-05-18 19:07:54.000000000 -0400
@@ -1106,7 +1106,7 @@
 -include $(INTROSPECTION_MAKEFILE)
 
 # we make the vips8 API
-@HAVE_INTROSPECTION_TRUE@Vips-8.0.gir: libvips.la
+@HAVE_INTROSPECTION_TRUE@Vips-8.0.gir: introspect
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
