diff -urN alpine-2.00-patched/alpine/adrbkcmd.c alpine-2.00/alpine/adrbkcmd.c
--- alpine-2.00-patched/alpine/adrbkcmd.c	2014-09-30 23:25:06.000000000 -0700
+++ alpine-2.00/alpine/adrbkcmd.c	2014-09-21 02:41:20.000000000 -0700
@@ -527,7 +527,7 @@
 	break;
 
       default:
-	panic("Unexpected command in process_abook_view_cmd");
+	uw_panic("Unexpected command in process_abook_view_cmd");
 	break;
     }
 
@@ -2634,7 +2634,7 @@
     }
 
     if(!ps->prc)
-      panic("NULL prc in convert_to_remote_config");
+      uw_panic("NULL prc in convert_to_remote_config");
 
     dprint((2, "convert_to_remote_config\n"));
     
@@ -5011,7 +5011,7 @@
 	    break;
 	  
 	  default:
-	    panic("can't happen in write_single_vcard_entry");
+	    uw_panic("can't happen in write_single_vcard_entry");
 	}
 
 	for(; ll && *ll; ll++){
@@ -5144,7 +5144,7 @@
 	    break;
 
 	  default:
-	    panic("can't happen in write_single_tab_entry");
+	    uw_panic("can't happen in write_single_tab_entry");
 	}
 
 	if(i)
@@ -7366,7 +7366,7 @@
 	break;
 
       default:
-	panic("Unexpected command in process_ldap_cmd");
+	uw_panic("Unexpected command in process_ldap_cmd");
 	break;
     }
 
diff -urN alpine-2.00-patched/alpine/alpine.c alpine-2.00/alpine/alpine.c
--- alpine-2.00-patched/alpine/alpine.c	2014-09-30 23:25:06.000000000 -0700
+++ alpine-2.00/alpine/alpine.c	2014-09-21 02:41:20.000000000 -0700
@@ -2321,7 +2321,7 @@
 	break;
 
       default:
-	panic("Unexpected command in choose_setup_cmd");
+	uw_panic("Unexpected command in choose_setup_cmd");
 	break;
     }
 
@@ -3317,7 +3317,7 @@
   This is also called from imap routines and fs_get and fs_resize.
   ----*/
 void
-panic(char *message)
+uw_panic(char *message)
 {
     char buf[256];
 
diff -urN alpine-2.00-patched/alpine/confscroll.c alpine-2.00/alpine/confscroll.c
--- alpine-2.00-patched/alpine/confscroll.c	2014-09-30 23:25:06.000000000 -0700
+++ alpine-2.00/alpine/confscroll.c	2014-09-21 02:41:20.000000000 -0700
@@ -5236,7 +5236,7 @@
 	}
 	else{
 	    if(reset_character_set_stuff(&err) == -1)
-	      panic(err ? err : "trouble with Character-Set");
+	      uw_panic(err ? err : "trouble with Character-Set");
 	    else if(err){
 		q_status_message(SM_ORDER | SM_DING, 3, 5, err);
 		fs_give((void **) &err);
@@ -5252,7 +5252,7 @@
 	}
 	else{
 	    if(reset_character_set_stuff(&err) == -1)
-	      panic(err ? err : "trouble with Character-Set");
+	      uw_panic(err ? err : "trouble with Character-Set");
 	    else if(err){
 		q_status_message(SM_ORDER | SM_DING, 3, 5, err);
 		fs_give((void **) &err);
diff -urN alpine-2.00-patched/alpine/folder.c alpine-2.00/alpine/folder.c
--- alpine-2.00-patched/alpine/folder.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/folder.c	2014-09-21 02:41:20.000000000 -0700
@@ -1014,7 +1014,7 @@
 		  strncat(servpart, "#news.", sizeof(servpart)-1-strlen(servpart));
 	    }
 	    else
-	      panic("Unexpected invalid server");
+	      uw_panic("Unexpected invalid server");
 	}
 	else
 	  servpart[0] = '\0';
diff -urN alpine-2.00-patched/alpine/help.c alpine-2.00/alpine/help.c
--- alpine-2.00-patched/alpine/help.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/help.c	2014-09-21 02:41:21.000000000 -0700
@@ -412,7 +412,7 @@
 	break;
 
       default :
-	panic("Unhandled case");
+	uw_panic("Unhandled case");
     }
 
     return(rv);
@@ -878,7 +878,7 @@
         break;
 
       default:
-        panic("Unexpected command in journal_processor");
+        uw_panic("Unexpected command in journal_processor");
 	break;
     }
     
diff -urN alpine-2.00-patched/alpine/mailcmd.c alpine-2.00/alpine/mailcmd.c
--- alpine-2.00-patched/alpine/mailcmd.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/mailcmd.c	2014-09-21 02:41:22.000000000 -0700
@@ -1464,7 +1464,7 @@
 
           /*--------- Default, unknown command ----------*/
       default:
-	panic("Unexpected command case");
+	uw_panic("Unexpected command case");
 	break;
     }
 
@@ -2657,7 +2657,7 @@
     ESCKEY_S	      ekey[10];
 
     if(!cntxt)
-      panic("no context ptr in save_prompt");
+      uw_panic("no context ptr in save_prompt");
 
     init_hist(&history, HISTSIZE);
 
@@ -3040,7 +3040,7 @@
 	    break;
 
 	  default :
-	    panic("Unhandled case");
+	    uw_panic("Unhandled case");
 	    break;
 	}
 
@@ -5665,7 +5665,7 @@
 	    break;
 
 	  default :
-	    panic("Unhandled case");
+	    uw_panic("Unhandled case");
 	    break;
 	}
 
diff -urN alpine-2.00-patched/alpine/mailview.c alpine-2.00/alpine/mailview.c
--- alpine-2.00-patched/alpine/mailview.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/mailview.c	2014-09-21 02:41:21.000000000 -0700
@@ -982,7 +982,7 @@
 
 
       default :
-	panic("Unexpected HANDLE type");
+	uw_panic("Unexpected HANDLE type");
     }
 
     return(0);
@@ -1085,7 +1085,7 @@
 	}
 
 	if(l == start_line)
-	  panic("Internal Error: no handle found");
+	  uw_panic("Internal Error: no handle found");
 	else
 	  start_line = l;
     }
diff -urN alpine-2.00-patched/alpine/radio.c alpine-2.00/alpine/radio.c
--- alpine-2.00-patched/alpine/radio.c	2014-09-30 23:25:06.000000000 -0700
+++ alpine-2.00/alpine/radio.c	2014-09-21 02:41:20.000000000 -0700
@@ -279,7 +279,7 @@
 
 	if(flags & RB_RET_HELP){
 	    if(help_text != NO_HELP)
-	      panic("RET_HELP and help in radio_buttons!");
+	      uw_panic("RET_HELP and help in radio_buttons!");
 
 	    button_list[b].ch = '?';
 	    button_list[b].rval = 3;
@@ -359,7 +359,7 @@
     clrbitmap(bitmap);
     memset(fkey_table, NO_OP_COMMAND, 12 * sizeof(int));
     if(flags & RB_RET_HELP && help_text != NO_HELP)
-      panic("RET_HELP and help in radio_buttons!");
+      uw_panic("RET_HELP and help in radio_buttons!");
 
     /* if shown, always at position 0 */
     if(help_text != NO_HELP || flags & RB_RET_HELP){
@@ -397,7 +397,7 @@
 	       && esc_list[i-start].label[0] != '\0'){  /* visible */
 		if(i == 12){  /* special case where we put it in help slot */
 		    if(help_text != NO_HELP)
-		  panic("Programming botch in radio_buttons(): too many keys");
+		  uw_panic("Programming botch in radio_buttons(): too many keys");
 
 		    if(esc_list[i-start].ch != -2)
 		      setbitn(0, bitmap); /* the help slot */
@@ -419,7 +419,7 @@
 		      rb_keymenu.keys[0].label = esc_list[i-start].label;
 		}
 		else
-		  panic("Botch in radio_buttons(): too many keys");
+		  uw_panic("Botch in radio_buttons(): too many keys");
 	    }
 	}
 	else{
@@ -822,16 +822,16 @@
      */
     for(i=0; list1 && list1[i].ch != -1; i++){
 	if(list1[i].rval == list[j].rval)
-	  panic("1bad rval in d_r");
+	  uw_panic("1bad rval in d_r");
 	if(F_OFF(F_USE_FK,ps_global) && list1[i].ch == list[j].ch)
-	  panic("1bad ch in ccl");
+	  uw_panic("1bad ch in ccl");
     }
 
     for(i=0; list2 && list2[i].ch != -1; i++){
 	if(list2[i].rval == list[j].rval)
-	  panic("2bad rval in d_r");
+	  uw_panic("2bad rval in d_r");
 	if(F_OFF(F_USE_FK,ps_global) && list2[i].ch == list[j].ch)
-	  panic("2bad ch in ccl");
+	  uw_panic("2bad ch in ccl");
     }
 
     j++;
@@ -839,7 +839,7 @@
     /* the visible set */
     for(i=0; list1 && list1[i].ch != -1; i++){
 	if(i >= KEYS_PER_LIST && list1[i].label[0] != '\0')
-	  panic("too many visible keys in ccl");
+	  uw_panic("too many visible keys in ccl");
 
 	list[j++] = list1[i];
     }
diff -urN alpine-2.00-patched/alpine/roleconf.c alpine-2.00/alpine/roleconf.c
--- alpine-2.00-patched/alpine/roleconf.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/roleconf.c	2014-09-21 02:41:22.000000000 -0700
@@ -5929,7 +5929,7 @@
     char ***alval;
 
     if(!(v && name))
-      panic("setup_dummy_pattern_var");
+      uw_panic("setup_dummy_pattern_var");
 
     v->name = (char *) fs_get(strlen(name)+NOTLEN+1);
     snprintf(v->name, strlen(name)+NOTLEN+1, "%s%s", (defpat && defpat->not) ? NOT : "", name);
diff -urN alpine-2.00-patched/alpine/signal.c alpine-2.00/alpine/signal.c
--- alpine-2.00-patched/alpine/signal.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/alpine/signal.c	2014-09-21 02:41:22.000000000 -0700
@@ -174,7 +174,7 @@
 /*----------------------------------------------------------------------
      Handle signals caused by aborts -- SIGSEGV, SIGILL, etc
 
-Call panic which cleans up tty modes and then core dumps
+Call uw_panic which cleans up tty modes and then core dumps
   ----------------------------------------------------------------------*/
 static RETSIGTYPE
 auger_in_signal(int sig)
@@ -187,9 +187,9 @@
     snprintf(buf, sizeof(buf), "Received abort signal(sig=%d)", sig);
     buf[sizeof(buf)-1] = '\0';
 
-    panic(buf);				/* clean up and get out */
+    uw_panic(buf);				/* clean up and get out */
 
-    exit(-1);				/* in case panic doesn't kill us */
+    exit(-1);				/* in case uw_panic doesn't kill us */
 }
 
 
diff -urN alpine-2.00-patched/pith/adrbklib.c alpine-2.00/pith/adrbklib.c
--- alpine-2.00-patched/pith/adrbklib.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/adrbklib.c	2014-09-21 02:41:32.000000000 -0700
@@ -4326,7 +4326,7 @@
 
     nn = (adrbk_cntr_t)n;
     if(!exp_head)
-      panic("exp_head not set in exp_set_expanded");
+      uw_panic("exp_head not set in exp_set_expanded");
 
     for(e = exp_head; e->next; e = e->next)
       if(e->next->ent >= nn)
@@ -4358,7 +4358,7 @@
 
     nn = (adrbk_cntr_t)n;
     if(!exp_head)
-      panic("exp_head not set in exp_unset_expanded");
+      uw_panic("exp_head not set in exp_unset_expanded");
 
     for(e = exp_head; e->next; e = e->next)
       if(e->next->ent >= nn)
@@ -4390,7 +4390,7 @@
 
     nn = (adrbk_cntr_t)n;
     if(!exp_head)
-      panic("exp_head not set in exp_del_nth");
+      uw_panic("exp_head not set in exp_del_nth");
 
     e = exp_head->next;
     while(e && e->ent < nn)
@@ -4430,7 +4430,7 @@
 
     nn = (adrbk_cntr_t)n;
     if(!exp_head)
-      panic("exp_head not set in exp_add_nth");
+      uw_panic("exp_head not set in exp_add_nth");
 
     e = exp_head->next;
     while(e && e->ent < nn)
diff -urN alpine-2.00-patched/pith/charset.c alpine-2.00/pith/charset.c
--- alpine-2.00-patched/pith/charset.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/charset.c	2014-09-21 02:41:32.000000000 -0700
@@ -428,7 +428,7 @@
 	    if(utf8_charset(cset)){
 		if(!utf8_text(&src, cset, &xsrc, 0L)){
 		    /* should not happen */
-		    panic("c-client failed to transliterate recognized characterset");
+		    uw_panic("c-client failed to transliterate recognized characterset");
 		}
 	    }
 	    else{
@@ -455,7 +455,7 @@
 	    if(!(cs->type == CT_ASCII || cs->type == CT_UTF8)){
 		if(!utf8_text_cs(&src, cs, &xsrc, 0L, 0L)){
 		    /* should not happen */
-		    panic("c-client failed to transliterate recognized characterset");
+		    uw_panic("c-client failed to transliterate recognized characterset");
 		}
 	    }
 	}
@@ -465,7 +465,7 @@
 		&& utf8_charset(cset)){
 		if(!utf8_text(&src, cset, &xsrc, 0L)){
 		    /* should not happen */
-		    panic("c-client failed to transliterate recognized character set");
+		    uw_panic("c-client failed to transliterate recognized character set");
 		}
 	}
 	else{
diff -urN alpine-2.00-patched/pith/conf.c alpine-2.00/pith/conf.c
--- alpine-2.00-patched/pith/conf.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/conf.c	2014-09-21 02:41:32.000000000 -0700
@@ -1798,7 +1798,7 @@
      */
 
     if(reset_character_set_stuff(&err) == -1)
-      panic(err ? err : "trouble with character set setup");
+      uw_panic(err ? err : "trouble with character set setup");
     else if(err){
 	init_error(ps, SM_ORDER | SM_DING, 3, 5, err);
 	fs_give((void **) &err);
@@ -2583,7 +2583,7 @@
 	      case 4: valptr = v->global_val.l; break;
 	      case 5: valptr = v->fixed_val.l; break;
 	      case 6: valptr = v->cmdline_val.l; break;
-	      default: panic("bad case in convert_configvar");
+	      default: uw_panic("bad case in convert_configvar");
 	    }
 
 	    if(valptr){
@@ -2609,7 +2609,7 @@
 	      case 4: valptr = &v->global_val.p; break;
 	      case 5: valptr = &v->fixed_val.p; break;
 	      case 6: valptr = &v->cmdline_val.p; break;
-	      default: panic("bad case in convert_configvar");
+	      default: uw_panic("bad case in convert_configvar");
 	    }
 
 	    if(valptr && *valptr && (*valptr)[0]){
@@ -7435,7 +7435,7 @@
 
     snprintf(buf1, sizeof(buf1), "%.*s", MAX(sizeof(buf1) - 1 - strlen(message), 0), arg);
     snprintf(buf2, sizeof(buf2), message, buf1);
-    panic(buf2);
+    uw_panic(buf2);
 }
 
 
diff -urN alpine-2.00-patched/pith/filter.c alpine-2.00/pith/filter.c
--- alpine-2.00-patched/pith/filter.c	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/filter.c	2014-09-21 02:41:31.000000000 -0700
@@ -363,10 +363,10 @@
 	    fs_give((void **) &sp);
 	}
 	else
-	  panic("Programmer botch: Can't unstack store readc");
+	  uw_panic("Programmer botch: Can't unstack store readc");
     }
     else
-      panic("Programmer botch: NULL store clearing store readc");
+      uw_panic("Programmer botch: NULL store clearing store readc");
 }
 
 
@@ -397,10 +397,10 @@
 	    fs_give((void **) &sp);
 	}
 	else
-	  panic("Programmer botch: Can't unstack store writec");
+	  uw_panic("Programmer botch: Can't unstack store writec");
     }
     else
-      panic("Programmer botch: NULL store clearing store writec");
+      uw_panic("Programmer botch: NULL store clearing store writec");
 }
 
 
diff -urN alpine-2.00-patched/pith/folder.c alpine-2.00/pith/folder.c
--- alpine-2.00-patched/pith/folder.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/folder.c	2014-09-21 02:41:31.000000000 -0700
@@ -586,7 +586,7 @@
      * collection???
      */
     if(!prime)
-      panic(_("No folder collections defined"));
+      uw_panic(_("No folder collections defined"));
 
     /*
      * At this point, insert the INBOX mapping as the leading
diff -urN alpine-2.00-patched/pith/imap.c alpine-2.00/pith/imap.c
--- alpine-2.00-patched/pith/imap.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/imap.c	2014-09-21 02:41:32.000000000 -0700
@@ -642,7 +642,7 @@
 void
 mm_fatal(char *message)
 {
-    panic(message);
+    uw_panic(message);
 }
 
 
diff -urN alpine-2.00-patched/pith/ldap.c alpine-2.00/pith/ldap.c
--- alpine-2.00-patched/pith/ldap.c	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/ldap.c	2014-09-21 02:41:30.000000000 -0700
@@ -1700,7 +1700,7 @@
 	break;
     
       default:
-	panic("LDAP function not implemented");
+	uw_panic("LDAP function not implemented");
     }
 #endif
 
diff -urN alpine-2.00-patched/pith/mailcap.c alpine-2.00/pith/mailcap.c
--- alpine-2.00-patched/pith/mailcap.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/mailcap.c	2014-09-21 02:41:32.000000000 -0700
@@ -247,7 +247,7 @@
 	return;
 
       default:
-	panic("Programmer botch in mc_process_file");
+	uw_panic("Programmer botch in mc_process_file");
 	/*NOTREACHED*/
     }
 
diff -urN alpine-2.00-patched/pith/mailcmd.c alpine-2.00/pith/mailcmd.c
--- alpine-2.00-patched/pith/mailcmd.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/mailcmd.c	2014-09-21 02:41:32.000000000 -0700
@@ -1228,7 +1228,7 @@
 		break;
 
 	      default:
-		panic("Unexpected incoming startup case");
+		uw_panic("Unexpected incoming startup case");
 		break;
 
 	    }
diff -urN alpine-2.00-patched/pith/mailindx.c alpine-2.00/pith/mailindx.c
--- alpine-2.00-patched/pith/mailindx.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/mailindx.c	2014-09-21 02:41:32.000000000 -0700
@@ -1179,7 +1179,7 @@
 		    break;
 		  
 		  default:
-		    panic("Unhandled fixed case in setup_index_header");
+		    uw_panic("Unhandled fixed case in setup_index_header");
 		    break;
 		}
 	    }
@@ -3279,7 +3279,7 @@
     IELEM_S  *ielem;
 
     if(!buf)
-      panic("NULL buf in simple_index_line()");
+      uw_panic("NULL buf in simple_index_line()");
 
     if(buflen > 0)
       buf[0] = '\0';
@@ -5052,7 +5052,7 @@
 		break;
 	  
 	      default:
-		panic("Unhandled case in prio_str");
+		uw_panic("Unhandled case in prio_str");
 		break;
 	    }
 
diff -urN alpine-2.00-patched/pith/mimetype.c alpine-2.00/pith/mimetype.c
--- alpine-2.00-patched/pith/mimetype.c	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/mimetype.c	2014-09-21 02:41:30.000000000 -0700
@@ -175,7 +175,7 @@
 	    }
 	}
 	else
-	  panic("Unhandled mime type search");
+	  uw_panic("Unhandled mime type search");
     }
 
 
diff -urN alpine-2.00-patched/pith/pattern.c alpine-2.00/pith/pattern.c
--- alpine-2.00-patched/pith/pattern.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/pattern.c	2014-09-21 02:41:32.000000000 -0700
@@ -1294,9 +1294,9 @@
     char  *p;
 
     if(!patgrp)
-      panic("NULL patgrp to parse_patgrp_slash");
+      uw_panic("NULL patgrp to parse_patgrp_slash");
     else if(!(str && *str)){
-	panic("NULL or empty string to parse_patgrp_slash");
+	uw_panic("NULL or empty string to parse_patgrp_slash");
 	patgrp->bogus = 1;
     }
     else if(!strncmp(str, "/NICK=", 6))
@@ -1515,9 +1515,9 @@
     NAMEVAL_S *v;
 
     if(!action)
-      panic("NULL action to parse_action_slash");
+      uw_panic("NULL action to parse_action_slash");
     else if(!(str && *str))
-      panic("NULL or empty string to parse_action_slash");
+      uw_panic("NULL or empty string to parse_action_slash");
     else if(!strncmp(str, "/ROLE=1", 7))
       action->is_a_role = 1;
     else if(!strncmp(str, "/OTHER=1", 8))
diff -urN alpine-2.00-patched/pith/save.c alpine-2.00/pith/save.c
--- alpine-2.00-patched/pith/save.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/save.c	2014-09-21 02:41:31.000000000 -0700
@@ -239,7 +239,7 @@
 	break;
       
       default:
-	panic(botch);
+	uw_panic(botch);
 	break;
     }
 
diff -urN alpine-2.00-patched/pith/send.c alpine-2.00/pith/send.c
--- alpine-2.00-patched/pith/send.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/send.c	2014-09-21 02:41:32.000000000 -0700
@@ -428,7 +428,7 @@
 	fields[++i]   = "X-Our-ReplyTo";	/* ReplyTo is real */
 	fields[++i]   = "Lcc";		/* Lcc: too... */
 	if(++i != FIELD_COUNT)
-	  panic("Fix FIELD_COUNT");
+	  uw_panic("Fix FIELD_COUNT");
 
 	for(pf = *custom; pf && pf->name; pf = pf->next)
 	  if(!pf->standard)
@@ -3087,7 +3087,7 @@
       return((char *)s);
     
     if(dlen < SIZEOF_20KBUF)
-      panic("bad call to encode_header_value");
+      uw_panic("bad call to encode_header_value");
 
     if(!encode_all){
 	/*
diff -urN alpine-2.00-patched/pith/sort.c alpine-2.00/pith/sort.c
--- alpine-2.00-patched/pith/sort.c	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/sort.c	2014-09-21 02:41:31.000000000 -0700
@@ -487,7 +487,7 @@
     dprint((2, "sort_sort_callback\n"));
 
     if(mn_get_total(g_sort.msgmap) < nmsgs)
-      panic("Message count shrank after sort!");
+      uw_panic("Message count shrank after sort!");
 
     /* copy ulongs to array of longs */
     for(i = nmsgs; i > 0; i--)
diff -urN alpine-2.00-patched/pith/stream.c alpine-2.00/pith/stream.c
--- alpine-2.00-patched/pith/stream.c	2014-09-30 23:25:10.000000000 -0700
+++ alpine-2.00/pith/stream.c	2014-09-21 02:41:32.000000000 -0700
@@ -1728,7 +1728,7 @@
 		 * partial text.
 		 */
 		if(!str_to_free)
-		  panic("Programmer botch: partial fetch attempt w/o string pointer");
+		  uw_panic("Programmer botch: partial fetch attempt w/o string pointer");
 		else
 		  *str_to_free = (char *) new_text.data;
 	    }
diff -urN alpine-2.00-patched/pith/string.c alpine-2.00/pith/string.c
--- alpine-2.00-patched/pith/string.c	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/string.c	2014-09-21 02:41:30.000000000 -0700
@@ -2644,7 +2644,7 @@
     char *ans = NULL;
 
     if(!quote_these_chars)
-      panic("bad arg to add_escapes");
+      uw_panic("bad arg to add_escapes");
 
     if(src){
 	char *q, *p, *qchar;
diff -urN alpine-2.00-patched/pith/util.h alpine-2.00/pith/util.h
--- alpine-2.00-patched/pith/util.h	2014-09-30 23:25:09.000000000 -0700
+++ alpine-2.00/pith/util.h	2014-09-21 02:41:31.000000000 -0700
@@ -56,7 +56,7 @@
 /* currently mandatory to implement stubs */
 
 /* called when we detect a serious program error */
-void	  panic(char *);
+void	  uw_panic(char *);
 
 /* called when testing to see if panic state is in effect */
 int	 panicking(void);
diff -urN alpine-2.00-patched/web/src/alpined.d/signal.c alpine-2.00/web/src/alpined.d/signal.c
--- alpine-2.00-patched/web/src/alpined.d/signal.c	2014-09-30 23:25:07.000000000 -0700
+++ alpine-2.00/web/src/alpined.d/signal.c	2014-09-21 02:41:24.000000000 -0700
@@ -117,7 +117,7 @@
 /*----------------------------------------------------------------------
      Handle signals caused by aborts -- SIGSEGV, SIGILL, etc
 
-Call panic which cleans up tty modes and then core dumps
+Call uw_panic which cleans up tty modes and then core dumps
   ----------------------------------------------------------------------*/
 static RETSIGTYPE
 auger_in_signal(int sig)
@@ -128,8 +128,8 @@
       (void) unlink(peSocketName);
 
     snprintf(tmp_20k_buf, SIZEOF_20KBUF, "Abort: signal %d", sig);
-    panic(tmp_20k_buf);		/* clean up and get out */
-    exit(-1);			/* in case panic doesn't kill us */
+    uw_panic(tmp_20k_buf);		/* clean up and get out */
+    exit(-1);			/* in case uw_panic doesn't kill us */
 }
 
 
