--- alpine-2.00/imap/Makefile.orig	2008-06-05 04:43:35.000000000 +1000
+++ alpine-2.00/imap/Makefile	2012-06-18 05:05:25.000000000 +1000
@@ -419,12 +419,12 @@
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
 	PASSWDTYPE=pam \
 	EXTRACFLAGS="$(EXTRACFLAGS) -DMAC_OSX_KLUDGE=1" \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib PAMDLFLAGS=-lpam"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib PAMDLFLAGS=-lpam"
 
 osx:	osxok an
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=$@ IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib"
 
 osxok:
 	@$(SH) -c '(test ! -f /usr/include/pam/pam_appl.h ) || make osxwarn'
