--- configure.ac.orig	2011-05-20 10:19:27.000000000 -0700
+++ configure.ac	2011-05-20 10:32:49.000000000 -0700
@@ -61,7 +61,7 @@
   AC_MSG_ERROR([apu-1-config wasn't found please install http://apr.apache.org ])
 else
   APU_INCLUDES=`$APUCONFIG --includes`
-  APU_LIBS=`$APUCONFIG --link-ld --libs`
+  APU_LIBS=`$APUCONFIG --link-ld --libs --ldflags`
 fi
 AC_SUBST(APU_INCLUDES)
 AC_SUBST(APU_LIBS)
