--- src/xar_internal.h.orig	2022-02-09 16:56:42.000000000 +1100
+++ src/xar_internal.h	2022-03-18 05:43:14.000000000 +1100
@@ -8,15 +8,13 @@
 #ifndef _XAR_INTERNAL_H_
 #define _XAR_INTERNAL_H_
 
-#ifdef XARSIG_BUILDING_WITH_XAR
 #include "xar.h"
-#else
-#include <xar/xar.h>
-#endif // XARSIG_BUILDING_WITH_XAR
 
 // Undeprecate these for internal usage
-xar_t xar_open(const char *file, int32_t flags) API_AVAILABLE(macos(10.4));
-xar_t xar_open_digest_verify(const char *file, int32_t flags, void *expected_toc_digest, size_t expected_toc_digest_len) API_AVAILABLE(macos(10.14.4));
-char *xar_get_path(xar_file_t f) API_AVAILABLE(macos(10.4));
+xar_t xar_open(const char *file, int32_t flags);
+xar_t xar_open_digest_verify(const char *file, int32_t flags, void *expected_toc_digest, size_t expected_toc_digest_len);
+char *xar_get_path(xar_file_t f);
+
+char *xar_get_safe_path(xar_file_t f);
 
 #endif /* _XAR_INTERNAL_H_ */
