--- lib/filetree.c.orig	2022-05-03 23:39:20.000000000 -0500
+++ lib/filetree.c	2022-05-08 06:01:32.000000000 -0500
@@ -740,7 +740,8 @@
 	size_t fspath1_size = 0, fspath2_size = 0;
 	size_t ns1_size = 0, ns2_size = 0;
 	const struct __xar_file_t * child1 = NULL, * child2 = NULL;
-	const uint keys_to_ignore_count = 1;
+	// const uint keys_to_ignore_count = 1;
+	#define keys_to_ignore_count 1
 	char * keys_to_ignore[keys_to_ignore_count] = { "id" }; // ID is allowed ot mismatch
 	
 	// If the two pointers match, call it the same.
