--- include/xar.h.in.orig	2022-03-15 06:02:49.000000000 -0500
+++ include/xar.h.in	2022-03-15 06:07:58.000000000 -0500
@@ -49,6 +49,9 @@
 #include <sys/stat.h>
 
 #ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
 #import <os/availability.h>
 #else
 #define API_DEPRECATED(...)
@@ -264,7 +269,7 @@
 
 void xar_serialize(xar_t x, const char *file);
 char *xar_get_path(xar_file_t f) API_DEPRECATED("Use xar_get_safe_path instead", macos(10.4,12.0));
-char *xar_get_safe_path(xar_file_t f) API_AVAILABLE(macos(12.0));
+char *xar_get_safe_path(xar_file_t f);
 off_t	xar_get_heap_offset(xar_t x);
 uint64_t xar_ntoh64(uint64_t num);
