--- lib/archive.h.orig	2018-03-20 21:09:55.000000000 -0500
+++ lib/archive.h	2020-12-20 20:34:04.000000000 -0600
@@ -40,12 +40,12 @@
 #define _XAR_ARCHIVE_H_
 #include <zlib.h>
 #include <libxml/hash.h>
-#ifdef __APPLE__
+/* #ifdef __APPLE__
 #include <CommonCrypto/CommonDigest.h>
 #include <CommonCrypto/CommonDigestSPI.h>
-#else
+#else */
 #include <openssl/evp.h>
-#endif
+/* #endif */
 #include <sys/types.h>
 #include <sys/stat.h>
 #include "xar.h"
