diff -rupN a/configure.ac b/configure.ac
--- configure.ac	2020-01-25 04:24:35.603846773 +0300
+++ configure.ac	2020-01-25 04:41:05.743109638 +0300
@@ -320,6 +320,16 @@ if test "x${have_libxml2}" = "x0" ; then
 fi
 
 dnl 
+dnl Configure libcrypto (part of OpenSSL).
+dnl 
+have_libcrypto="1"
+AC_CHECK_HEADERS([openssl/evp.h], , [have_libcrypto="0"])
+AC_CHECK_LIB([crypto], [OPENSSL_init_crypto], , [have_libcrypto="0"])
+if test "x${have_libcrypto}" = "x0" ; then
+  AC_MSG_ERROR([Cannot build without libcrypto (OpenSSL)])
+fi
+
+dnl 
 dnl Configure libz.
 dnl 
 have_libz="1"
