--- config.orig	Tue Nov 16 15:07:59 2010 -0800
+++ config	Sat Nov 20 22:36:50 2010 -0600
@@ -1,10 +1,12 @@
 # Configure where libHSPlasma is installed
-PLASMA_PATH=<<EDIT ME>>
+PLASMA_PATH=@PREFIX@
 PLAT_BITS=
 
 # QScintilla Defaults
 win32 {
     DEFINES += PLAT_FONT="\"\\\"Courier New\\\"\""
+} else:macx {
+    DEFINES += PLAT_FONT="\"\\\"@FONT@\\\"\""
 } else {
     DEFINES += PLAT_FONT="\"\\\"Droid Sans Mono\\\"\""
 }
@@ -21,3 +21,7 @@
     INCLUDEPATH *= $${PLASMA_PATH}/include/HSPlasmaNet
     LIBS        *= -L$${PLASMA_PATH}/lib -lHSPlasmaNet
 }
+
+QMAKE_CFLAGS += @CFLAGS@
+QMAKE_CXXFLAGS += @CXXFLAGS@
+QMAKE_LFLAGS += @LDFLAGS@
