--- src/App/FreeCADInit.py.orig	2014-07-13 10:33:02.000000000 -0500
+++ src/App/FreeCADInit.py	2014-07-17 20:19:20.000000000 -0500
@@ -102,6 +102,7 @@
 					Log('Init:      Initializing ' + Dir + '... done\n')
 			else:
 				Log('Init:      Initializing ' + Dir + '(Init.py not found)... ignore\n')
+        sys.path.insert(0,BinDir)
 	sys.path.insert(0,LibDir)
 	sys.path.insert(0,ModDir)
 	Log("Using "+ModDir+" as module path!\n")
