--- dot2tex/dotparsing.py.orig	2009-10-05 12:57:02.000000000 +0100
+++ dot2tex/dotparsing.py	2014-04-28 18:01:42.000000000 +0100
@@ -26,7 +26,7 @@
 from pyparsing import  (Literal, CaselessLiteral, Word, Upcase, OneOrMore, ZeroOrMore,
     Forward, NotAny, delimitedList, oneOf, Group, Optional, Combine, alphas, nums,
     restOfLine, cStyleComment, nums, alphanums, printables, empty, quotedString,
-    ParseException, ParseResults, CharsNotIn, _noncomma, dblQuotedString, QuotedString, ParserElement,
+    ParseException, ParseResults, CharsNotIn, dblQuotedString, QuotedString, ParserElement,
     Suppress,Regex,removeQuotes)
 
 dot_keywords = ['graph', 'subgraph', 'digraph', 'node', 'edge', 'strict']
