--- spyderlib/widgets/sourcecode/codeeditor.py.orig	2011-09-05 18:29:00.000000000 -0500
+++ spyderlib/widgets/sourcecode/codeeditor.py	2011-09-08 08:51:36.000000000 -0500
@@ -2038,6 +2038,18 @@
                     self.emit(SIGNAL('trigger_code_completion(bool)'), True)
         elif key == Qt.Key_Home and not ctrl:
             self.stdkey_home(shift)
+        elif key == Qt.Key_Home:
+            cursor = self.textCursor()
+            cursor.movePosition(QTextCursor.Start)
+            self.setTextCursor(cursor)
+        elif key == Qt.Key_End and not ctrl:
+            cursor = self.textCursor()
+            cursor.movePosition(QTextCursor.EndOfLine)
+            self.setTextCursor(cursor)
+        elif key == Qt.Key_End:
+            cursor = self.textCursor()
+            cursor.movePosition(QTextCursor.End)
+            self.setTextCursor(cursor)
         elif text == '(' and not self.has_selected_text():
             self.hide_completion_widget()
             position = self.get_position('cursor')
