--- src/CMakeLists.txt.orig	2017-03-27 04:04:12.000000000 -0500
+++ src/CMakeLists.txt	2017-06-16 18:11:28.000000000 -0500
@@ -297,7 +297,7 @@
 ENDIF()
 
 IF( EXIV2_ENABLE_NLS )
-    TARGET_LINK_LIBRARIES( exiv2lib ${PRIVATE_VAR}  ${LIBINTL_LIBRARIES} )
+    TARGET_LINK_LIBRARIES( exiv2lib ${PRIVATE_VAR}  ${Intl_LIBRARIES} )
 ENDIF( EXIV2_ENABLE_NLS )
 
 IF( ICONV_FOUND )
@@ -316,7 +316,7 @@
 # ******************************************************************************
 # exiv2 application
 ADD_EXECUTABLE( exiv2 ${EXIV2_SRC}  ${EXIV2_HDR} )
-TARGET_LINK_LIBRARIES( exiv2 exiv2lib )
+TARGET_LINK_LIBRARIES( exiv2 exiv2lib  ${Intl_LIBRARIES} )
 INSTALL( TARGETS exiv2 ${INSTALL_TARGET_STANDARD_ARGS} )
 
 # ******************************************************************************
