--- DESCRIPTION.orig	2023-07-18 15:30:02.000000000 +0800
+++ DESCRIPTION	2023-07-19 01:51:30.000000000 +0800
@@ -17,7 +17,7 @@
 RoxygenNote: 7.2.3
 URL: https://github.com/SeewooLi/IRTest
 BugReports: https://github.com/SeewooLi/IRTest/issues
-Suggests: knitr, rmarkdown, testthat (>= 3.0.0), V8
+Suggests: knitr, rmarkdown, testthat (>= 3.0.0)
 VignetteBuilder: knitr
 Imports: betafunctions, dcurver, ggplot2
 Depends: R (>= 2.10)
