--- DESCRIPTION.orig	2023-09-20 04:20:02.000000000 +0700
+++ DESCRIPTION	2023-09-20 16:12:46.000000000 +0700
@@ -17,7 +17,7 @@
 RoxygenNote: 7.2.3
 URL: https://github.com/SeewooLi/IRTest
 BugReports: https://github.com/SeewooLi/IRTest/issues
-Suggests: knitr, rmarkdown, testthat (>= 3.0.0), V8, gridExtra
+Suggests: knitr, rmarkdown, testthat (>= 3.0.0), gridExtra
 VignetteBuilder: knitr
 Imports: betafunctions, dcurver, ggplot2
 Depends: R (>= 2.10)
