From 12a3eac312d06257ba1dca771ee710eacda6ce1e Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Thu, 14 Jun 2012 19:29:01 +0000
Subject: Fix compilation with clang/clang++ - configure: we can't use CFLAGS for compiling C++ code, especially as CFLAGS would include -std=c99 which makes no sense for C++ - util-osx-cocoa.mm was compiled as a C++ file, not an objective-C++ file, gcc was forgiving about the error, clang++ isn't

---
diff --git mythtv/libs/libmythui/libmythui.pro mythtv/libs/libmythui/libmythui.pro
index 72d5482..4560bb2 100644
--- mythtv/libs/libmythui/libmythui.pro
+++ mythtv/libs/libmythui/libmythui.pro
@@ -95,16 +95,12 @@ using_x11 {
 }
 
 macx {
-    QMAKE_EXT_CPP += .mm
-    QMAKE_OBJECTIVE_CFLAGS += $$QMAKE_CFLAGS
-    HEADERS += screensaver-osx.h   DisplayResOSX.h   util-osx.h   util-osx-cocoa.h
-    SOURCES += screensaver-osx.cpp DisplayResOSX.cpp util-osx.cpp util-osx-cocoa.mm
-
-    QMAKE_CXXFLAGS += -F/System/Library/Frameworks/CoreServices.framework/Frameworks
-    QMAKE_CXXFLAGS += -F/System/Library/Frameworks/ApplicationServices.framework/Frameworks
-    QMAKE_CXXFLAGS += -F/System/Library/Frameworks/Carbon.framework/Frameworks
-    QMAKE_CXXFLAGS += -F/System/Library/Frameworks/Cocoa.framework/Frameworks
-    LIBS           += -framework Carbon -framework Cocoa -framework IOKit
+    HEADERS += screensaver-osx.h   DisplayResOSX.h   util-osx.h
+    SOURCES += screensaver-osx.cpp DisplayResOSX.cpp util-osx.cpp
+    QMAKE_OBJECTIVE_CXXFLAGS += $$QMAKE_CXXFLAGS
+    OBJECTIVE_HEADERS += util-osx-cocoa.h
+    OBJECTIVE_SOURCES += util-osx-cocoa.mm
+    LIBS              += -framework Carbon -framework Cocoa -framework IOKit
 
     using_appleremote {
         HEADERS += AppleRemote.h   AppleRemoteListener.h
