--- mythtv/libs/libmythtv/avformatwriter.cpp	2012-09-18 17:58:13.000000000 -0700
+++ mythtv/libs/libmythtv/avformatwriter.cpp	2012-10-01 19:33:15.000000000 -0700
@@ -24,10 +24,16 @@
 #include "NuppelVideoRecorder.h"
 #include "avformatwriter.h"
 
-#if HAVE_BIGENDIAN
-extern "C" {
-#include "byteswap.h"
-}
+#if HAVE_BYTESWAP_H
+#include <byteswap.h>
+#elif HAVE_SYS_ENDIAN_H
+#include <sys/endian.h>
+#elif CONFIG_DARWIN
+#include <libkern/OSByteOrder.h>
+#define bswap_16 OSSwapInt16
+#define bswap_32 OSSwapInt32
+#elif HAVE_BIGENDIAN
+#error No bswap functions defined
 #endif
 
 #define LOC QString("AVFW(%1): ").arg(m_filename)
