diff --git CMakeLists.txt.orig CMakeLists.txt
index 5c5ca8a..13c29b8 100644
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -210,7 +210,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wno-unreachable-code")
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Wno-documentation")

-  wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=deprecated")
+ # wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=deprecated")
   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=non-pod-varargs")

   wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Qunused-arguments")