--- Makefile	2008-03-30 04:10:14.000000000 -0500
+++ Makefile	2008-04-20 01:40:17.000000000 -0500
@@ -263,6 +263,6 @@
 
 ca-bundle: lib/mk-ca-bundle.pl
 	@echo "generate a fresh ca-bundle.crt"
-	@perl $< -b -l -u lib/ca-bundle.crt
+	@perl $< -n lib/ca-bundle.crt
 
 
--- lib/mk-ca-bundle.pl	2008-02-15 03:28:40.000000000 -0600
+++ lib/mk-ca-bundle.pl	2008-04-20 01:40:17.000000000 -0500
@@ -33,7 +33,6 @@
 #
 use Getopt::Std;
 use MIME::Base64;
-use LWP::UserAgent;
 use strict;
 use vars qw($opt_b $opt_h $opt_i $opt_l $opt_n $opt_q $opt_t $opt_u $opt_v);  
 
