--- config.in.orig	2011-01-27 00:09:43.000000000 -0600
+++ config.in	2011-01-27 00:21:39.000000000 -0600
@@ -9,7 +9,7 @@
 #    Change the path below to point to the directory where db.h is
 #    installed on your system.
 
-INCLUDE	= /usr/local/BerkeleyDB/include
+INCLUDE	= @PREFIX@/include/db48
 #INCLUDE	= /usr/local/include
 #INCLUDE	= /usr/include
 
@@ -18,7 +18,7 @@
 #    Change the path below to point to the directory where libdb is
 #    installed on your system.
 
-LIB	= /usr/local/BerkeleyDB/lib
+LIB	= @PREFIX@/lib/db48
 #LIB	= /usr/local/lib
 #LIB	= /usr/lib
 
