--- CMakeLists.txt.orig	2017-01-24 02:29:17.000000000 -0500
+++ CMakeLists.txt	2017-02-08 09:28:55.000000000 -0500
@@ -25,7 +25,7 @@
 set(PHONON_LIB_MINOR_VERSION "9") # Only change on binary compatible changes with new interfaces
 set(PHONON_LIB_PATCH_VERSION "1") # Bump whenever you feel like it :P
 set(PHONON_LIB_VERSION "${PHONON_LIB_MAJOR_VERSION}.${PHONON_LIB_MINOR_VERSION}.${PHONON_LIB_PATCH_VERSION}")
-set(PHONON_LIB_SOVERSION ${PHONON_LIB_MAJOR_VERSION})
+set(PHONON_LIB_SOVERSION "${PHONON_LIB_MAJOR_VERSION}.4.0")
 set(PHONON_LIB_SONAME phonon)
 set(PHONON_LIB_SONAME_CAMEL Phonon)
 
@@ -230,7 +230,7 @@
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PHONON_LIB_SONAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 endif(NOT WIN32)
 
-set(CMAKECONFIG_INSTALL_DIR ${LIB_INSTALL_DIR}/cmake/${PHONON_LIB_SONAME} )
+set(CMAKECONFIG_INSTALL_DIR @MACPORTS_CMAKE_DIR@/${PHONON_LIB_SONAME} )
 
 # figure out the relative path from the installed Config.cmake file to the install prefix (which may be at
 # runtime different from the chosen CMAKE_INSTALL_PREFIX if under Windows the package was installed anywhere)
