--- macosx/tkMacOSXDraw.c.orig	2024-09-05 01:27:19
+++ macosx/tkMacOSXDraw.c	2024-12-04 03:27:24
@@ -1137,7 +1137,8 @@ TkScrollWindow(
     TKContentView *view = (TKContentView *)TkMacOSXGetNSViewForDrawable(macDraw);
     HIShapeRef srcRgn, dstRgn;
     HIMutableShapeRef dmgRgn = HIShapeCreateMutable();
-    NSRect bounds, viewSrcRect, srcRect, dstRect;
+    NSRect bounds, viewSrcRect;
+    CGRect srcRect, dstRect;
     int result = 0;
 
     if (view) {
--- macosx/tkMacOSXNotify.c.orig	2024-09-05 01:27:19
+++ macosx/tkMacOSXNotify.c	2024-12-04 14:44:55
@@ -378,7 +378,7 @@ TkMacOSXDrawAllViews(
 		if (dirtyCount) {
 		   continue;
 		}
-		[[view layer] setNeedsDisplayInRect:[view tkDirtyRect]];
+		[[view layer] setNeedsDisplayInRect:NSRectToCGRect([view tkDirtyRect])];
 		[view setNeedsDisplay:YES];
 	    }
 	} else {
--- macosx/tkMacOSXRegion.c.orig	2024-09-05 01:27:19
+++ macosx/tkMacOSXRegion.c	2024-12-04 15:00:05
@@ -518,7 +518,7 @@ rectPrinter(
     void *ref)
 {
     if (rect) {
-	fprintf(stderr, "    %s\n", NSStringFromRect(*rect).UTF8String);
+	fprintf(stderr, "    %s\n", NSStringFromRect(NSRectFromCGRect(*rect)).UTF8String);
     }
     return noErr;
 }
