--- macosx/tkMacOSXPrivate.h
+++ macosx/tkMacOSXPrivate.h
@@ -488,6 +488,24 @@ VISIBILITY_HIDDEN
 - (NSMenuItem *)itemInSupermenu;
 @end
 
+/* Forward declaration is required to build using the macOS SDK up to
+ * version 10.9. Moreover, these SDKs lack the 'effectiveAppearance' property.
+ */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1090
+@class NSAppearance;
+
+@interface NSView (AppearanceLegacy)
+@property (readonly) NSAppearance *effectiveAppearance;
+@end
+
+@implementation NSView (AppearanceLegacy)
+- (NSAppearance *)effectiveAppearance
+{
+    return nil;
+}
+@end
+#endif
+
 // Need undocumented appearance: argument
 @interface NSMenu(TKMenu)
 - (BOOL)popUpMenuPositioningItem:(NSMenuItem *)item
