--- Makefile	2018-11-30 22:25:04.000000000 +0800
+++ Makefile	2024-07-17 22:18:35.000000000 +0800
@@ -74,7 +74,7 @@
 #code can be reviewed manually
 #MW=-Wtraditional -Wcast-qual -Wcast-align -Wconversion -pedantic -Wlong-long -Wimplicit -Wuninitialized -W -Wshadow -Wsign-compare -Wstrict-prototypes -Wmissing-declarations
 
-CFLAGS1 = -Wall -Wstrict-prototypes -s -O2 -fomit-frame-pointer -Wno-unused-result $(LARGEFILEFLAGS) $(MW)
+CFLAGS1 = -Wall -Wstrict-prototypes -s -O2 -fomit-frame-pointer $(LARGEFILEFLAGS) $(MW)
 
 CC=gcc
 
