--- osc/conf.py	2017-09-22 09:06:18.000000000 +0200
+++ osc/conf.py	2017-09-28 17:49:56.000000000 +0200
@@ -180,18 +180,12 @@
             'maintained_update_project_attribute': 'OBS:UpdateProject',
             'show_download_progress': '0',
             # path to the vc script
-            'vc-cmd': '/usr/lib/build/vc',
+            'vc-cmd': '@@PREFIX@@/libexec/obs-build/vc',
 
             # heuristic to speedup Package.status
             'status_mtime_heuristic': '0'
 }
 
-# some distros like Debian rename and move build to obs-build
-if not os.path.isfile('/usr/bin/build') and os.path.isfile('/usr/bin/obs-build'):
-    DEFAULTS['build-cmd'] = '/usr/bin/obs-build'
-if not os.path.isfile('/usr/lib/build/vc') and os.path.isfile('/usr/lib/obs-build/vc'):
-    DEFAULTS['vc-cmd'] = '/usr/lib/obs-build/vc'
-
 # being global to this module, this dict can be accessed from outside
 # it will hold the parsed configuration
 config = DEFAULTS.copy()
@@ -538,7 +532,7 @@
         cafile = options.get('cafile', None)
         capath = options.get('capath', None)
         if not cafile and not capath:
-            for i in ['/etc/pki/tls/cert.pem', '/etc/ssl/certs']:
+            for i in ['@@CAFILEPATH@@', '/etc/ssl/certs']:
                 if os.path.isfile(i):
                     cafile = i
                     break
