--- Makefile	2015-03-09 01:31:30.000000000 -0500
+++ Makefile	2015-03-09 01:31:43.000000000 -0500
@@ -31,8 +31,8 @@
 		cp -f bin/$(COMMAND) $(DESTDIR)$(BINPREFIX); \
 	)
 	cp -f man/git-*.1 $(DESTDIR)$(MANPREFIX)
-	@mkdir -p $(DESTDIR)/etc/bash_completion.d
-	cp -f etc/bash_completion.sh $(DESTDIR)/etc/bash_completion.d/git-extras
+	@mkdir -p $(DESTDIR)$(PREFIX)/etc/bash_completion.d
+	cp -f etc/bash_completion.sh $(DESTDIR)$(PREFIX)/etc/bash_completion.d/git-extras
 
 man/%.html: man/%.md
 	ronn \
@@ -56,7 +56,7 @@
 		echo "... uninstalling $(DESTDIR)$(MANPREFIX)/$(notdir $(MAN))"; \
 		rm -f $(DESTDIR)$(MANPREFIX)/$(notdir $(MAN)); \
 	)
-	rm -f $(DESTDIR)/etc/bash_completion.d/git-extras
+	rm -f $(DESTDIR)$(PREFIX)/etc/bash_completion.d/git-extras
 
 clean: docclean
 
