--- Makefile.in.orig	2009-03-09 16:17:54.000000000 -0400
+++ Makefile.in	2009-03-09 16:18:05.000000000 -0400
@@ -827,7 +827,7 @@
 	fi
 	rm -f "$(DESTDIR)$(bindir)/$(bin_SCRIPTS)" || true;
 	chmod +x "$(DESTDIR)$(pkgpythondir)/$(python_debugger_script)"
-	$(LN_S) "$(DESTDIR)$(pkgpythondir)/$(python_debugger_script)" "$(DESTDIR)$(bindir)/$(bin_SCRIPTS)"
+	$(LN_S) "$(pkgpythondir)/$(python_debugger_script)" "$(DESTDIR)$(bindir)/$(bin_SCRIPTS)"
 
 test: check
 @MAINTAINER_MODE_TRUE@ChangeLog:
