--- configure.orig	2018-01-02 21:21:01.000000000 +0400
+++ configure	2018-01-02 21:21:28.000000000 +0400
@@ -310,22 +310,6 @@
   exit 1
 fi
 
-if [ -d '.git' ] && command -v git >/dev/null 2>&1 ; then
-  git submodule init && git submodule update || exit 1
-else
-  if [ ! -f 'nanopass/nanopass.ss' ] ; then
-    rmdir nanopass && (curl  -L -o v1.9.tar.gz https://github.com/nanopass/nanopass-framework-scheme/archive/v1.9.tar.gz && tar -zxf v1.9.tar.gz && mv nanopass-framework-scheme-1.9 nanopass && rm v1.9.tar.gz) || exit 1
-  fi
-
-  if [ ! -f 'zlib/configure' ] ; then
-    rmdir zlib && (curl -L -o v1.2.11.tar.gz https://github.com/madler/zlib/archive/v1.2.11.tar.gz && tar -xzf v1.2.11.tar.gz && mv zlib-1.2.11 zlib && rm v1.2.11.tar.gz) || exit 1
-  fi
-
-  if [ ! -f 'stex/Mf-stex' ] ; then
-    rmdir stex && (curl -L -o v1.2.1.tar.gz https://github.com/dybvig/stex/archive/v1.2.1.tar.gz && tar -zxf v1.2.1.tar.gz && mv stex-1.2.1 stex && rm v1.2.1.tar.gz) || exit 1
-  fi
-fi
-
 ./workarea $m $w
 
 sed -e 's/$(m)/'$m'/g'\
