--- c/Mf-ta6osx.orig	2016-05-13 20:59:20.000000000 +0400
+++ c/Mf-ta6osx	2017-04-26 23:05:58.000000000 +0400
@@ -17,7 +17,7 @@
 Cpu = X86_64
 
 mdclib = -liconv -lm -lncurses
-C = gcc ${CPPFLAGS} -m64 -Wpointer-arith -Wall -Wextra -Werror -O2 -I/opt/X11/include/ ${CFLAGS}
+C = ${CC} ${CPPFLAGS} -m64 -Wpointer-arith -Wall -Wextra -Werror -O2 ${CFLAGS}
 o = o
 mdsrc = i3le.c
 mdobj = i3le.o
@@ -26,16 +26,12 @@
 .SUFFIXES: .c .o
 
 .c.o:
-	$C -c -D${Cpu} -I${Include} -I${ZlibInclude} $*.c
+	$C -c -D${Cpu} -I${Include} $*.c
 
 include Mf-base
 
-${Kernel}: ${kernelobj} ../zlib/libz.a
-	ld -r -o ${Kernel} ${kernelobj} ../zlib/libz.a
+${Kernel}: ${kernelobj}
+	$C -static -nostdlib -o ${Kernel} ${kernelobj} ${LDFLAGS} -lz -Wl,-r
 
 ${Scheme}: ${Kernel} ${Main}
 	$C -o ${Scheme} ${Kernel} ${Main} ${mdclib} ${LDFLAGS}
-
-${ZlibInclude}/zlib.h ${ZlibInclude}/zconf.h ../zlib/libz.a:
-	(cd ../zlib; CFLAGS=-m64 ./configure --64)
-	(cd ../zlib; make)
