Prevent configure script from finding X11 OpenGL (e.g. mesa & freeglut).
--- configure.orig	2022-05-20 11:11:32.000000000 -0500
+++ configure	2022-06-02 15:55:01.000000000 -0500
@@ -20420,7 +20420,7 @@
 CPPFLAGS="${GL_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lopengl32 -lGL"
+ax_check_libs="-lopengl32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
@@ -20635,7 +20635,7 @@
   ax_cv_check_glu_libglu=yes
 else
   LIBS=""
-ax_check_libs="-lglu32 -lGLU"
+ax_check_libs="-lglu32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
@@ -20790,7 +20790,7 @@
 CPPFLAGS="${GLUT_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lglut32 -lglut"
+ax_check_libs="-lglut32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
