--- libtiff/tif_jpeg.c.orig	2012-09-23 00:48:09.000000000 +1000
+++ libtiff/tif_jpeg.c	2013-02-09 12:25:15.000000000 +1100
@@ -81,8 +81,13 @@
 # ifndef __RPCNDR_H__            /* don't conflict if rpcndr.h already read */
    typedef unsigned char boolean;
 # endif
-# define HAVE_BOOLEAN            /* prevent jmorecfg.h from redefining it */
+#else
+/* tiffiop.h defines TRUE and FALSE, which would cause jmorecfg.h's attempted:
+ * typedef enum { FALSE = 0, TRUE = 1 } boolean;
+ * to fail. */
+typedef int boolean;
 #endif
+# define HAVE_BOOLEAN            /* prevent jmorecfg.h from redefining it */
 
 #include "jpeglib.h"
 #include "jerror.h"
