--- libtiff/tif_config.h.in	2006-01-23 12:22:32.000000000 -0600
+++ libtiff/tif_config.h.in	2008-06-05 23:57:37.000000000 -0500
@@ -138,10 +138,10 @@
 
 /* Native cpu byte order: 1 if big-endian (Motorola) or 0 if little-endian
    (Intel) */
-#undef HOST_BIGENDIAN
+@HOST_BIGENDIAN@
 
 /* Set the native cpu bit order (FILLORDER_LSB2MSB or FILLORDER_MSB2LSB) */
-#undef HOST_FILLORDER
+@HOST_FILLORDER@
 
 /* Support JPEG compression (requires IJG JPEG library) */
 #undef JPEG_SUPPORT
@@ -230,7 +230,7 @@
 
 /* Define to 1 if your processor stores words with the most significant byte
    first (like Motorola and SPARC, unlike Intel and VAX). */
-#undef WORDS_BIGENDIAN
+@WORDS_BIGENDIAN@
 
 /* Define to 1 if the X Window System is missing or not being used. */
 #undef X_DISPLAY_MISSING
diff -ru tiff-3.8.2-orig/libtiff/tiffconf.h.in tiff-3.8.2/libtiff/tiffconf.h.in
--- libtiff/tiffconf.h.in	2005-12-27 05:28:23.000000000 -0600
+++ libtiff/tiffconf.h.in	2008-06-05 23:57:47.000000000 -0500
@@ -29,11 +29,11 @@
 #undef HAVE_IEEEFP
 
 /* Set the native cpu bit order (FILLORDER_LSB2MSB or FILLORDER_MSB2LSB) */
-#undef HOST_FILLORDER
+@HOST_FILLORDER@
 
 /* Native cpu byte order: 1 if big-endian (Motorola) or 0 if little-endian
    (Intel) */
-#undef HOST_BIGENDIAN
+@HOST_BIGENDIAN@
 
 /* Support CCITT Group 3 & 4 algorithms */
 #undef CCITT_SUPPORT
