--- libtiff/tif_getimage.c.orig	Wed Jan  5 20:09:34 2005
+++ libtiff/tif_getimage.c	Wed Jan  5 20:10:38 2005
@@ -247,7 +247,7 @@
     TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLESPERPIXEL, &img->samplesperpixel);
     TIFFGetFieldDefaulted(tif, TIFFTAG_EXTRASAMPLES,
 	&extrasamples, &sampleinfo);
-    if (extrasamples > 1)
+    if (extrasamples >= 1)
     {
 	switch (sampleinfo[0]) {
 	case EXTRASAMPLE_UNSPECIFIED:	/* Workaround for some images without */
