--- Makefile.am.orig	2014-03-28 11:08:27.000000000 -0700
+++ Makefile.am	2014-03-28 11:11:08.000000000 -0700
@@ -25,34 +25,6 @@
 lib_LTLIBRARIES = \
 	libepc/libepc-1.0.la \
 	libepc-ui/libepc-ui-1.0.la
-noinst_LTLIBRARIES = \
-	tests/libepc-tests.la
-noinst_PROGRAMS = \
-	examples/list-resources \
-	examples/lookup-resource \
-	examples/service-browser \
-	examples/simple-publisher \
-	examples/consumer-ui \
-	examples/publisher-ui \
-	examples/server-credentials \
-	$(TESTS)
-TESTS = \
-	tests/test-consumer-by-info \
-	tests/test-consumer-by-name \
-	tests/test-dispatcher-local-collision \
-	tests/test-dispatcher-multiple-services \
-	tests/test-dispatcher-rename \
-	tests/test-dispatcher-reset \
-	tests/test-dispatcher-simple-service \
-	tests/test-dispatcher-subtypes \
-	tests/test-dispatcher-unique \
-	tests/test-expand-name \
-	tests/test-progress-hooks \
-	tests/test-publisher-bookmarks \
-	tests/test-publisher-change-name \
-	tests/test-publisher-libsoup-494128 \
-	tests/test-publisher-unique \
-	tests/test-service-type
 
 # ================
 # -- FILE LISTS --
@@ -134,11 +106,6 @@
 	-DLOCALEDIR=\"$(localedir)\" \
 	-DGETTEXT_PACKAGE=\"@GETTEXT_PACKAGE@\"
 
-example_epc_cflags = $(AM_CFLAGS) $(LIBEPC_CFLAGS)
-example_epc_libs = libepc/libepc-1.0.la $(LIBEPC_LIBS)
-example_epc_ui_cflags = $(example_epc_cflags) $(LIBEPC_UI_CFLAGS)
-example_epc_ui_libs = $(example_epc_libs) libepc-ui/libepc-ui-1.0.la $(LIBEPC_UI_LIBS)
-
 libepc_libepc_1_0_la_CFLAGS			= $(AM_CFLAGS) $(LIBEPC_CFLAGS) -DG_LOG_DOMAIN=\"libepc\"
 libepc_libepc_1_0_la_LDFLAGS			= $(LIBEPC_LIBS) -version-info $(LT_VERSION_INFO)
 
@@ -146,61 +113,6 @@
 libepc_ui_libepc_ui_1_0_la_LDFLAGS		= $(LIBEPC_UI_LIBS) -version-info $(LT_VERSION_INFO)
 libepc_ui_libepc_ui_1_0_la_LIBADD		= libepc/libepc-1.0.la
 
-examples_list_resources_CFLAGS			= $(example_epc_cflags)
-examples_list_resources_LDADD			= $(example_epc_libs)
-examples_lookup_resource_CFLAGS			= $(example_epc_cflags)
-examples_lookup_resource_LDADD			= $(example_epc_libs)
-examples_service_browser_CFLAGS			= $(example_epc_cflags)
-examples_service_browser_LDADD			= $(example_epc_libs)
-examples_simple_publisher_CFLAGS		= $(example_epc_cflags)
-examples_simple_publisher_LDADD			= $(example_epc_libs)
-
-examples_consumer_ui_CFLAGS			= $(example_epc_ui_cflags) $(AVAHI_UI_CFLAGS)
-examples_consumer_ui_LDADD			= $(example_epc_ui_libs) $(AVAHI_UI_LIBS)
-examples_publisher_ui_CFLAGS			= $(example_epc_ui_cflags) $(AVAHI_UI_CFLAGS) -rdynamic
-examples_publisher_ui_LDADD			= $(example_epc_ui_libs) $(AVAHI_UI_LIBS)
-examples_server_credentials_CFLAGS		= $(example_epc_ui_cflags)
-examples_server_credentials_LDADD		= $(example_epc_ui_libs)
-
-
-test_epc_libs = tests/libepc-tests.la $(example_epc_libs)
-test_epc_ui_libs = tests/libepc-tests.la $(example_epc_ui_libs)
-
-tests_libepc_tests_la_CFLAGS			= $(example_epc_cflags)
-
-tests_test_consumer_by_info_CFLAGS		= $(example_epc_cflags)
-tests_test_consumer_by_info_LDADD		= $(test_epc_libs)
-tests_test_consumer_by_name_CFLAGS		= $(example_epc_cflags)
-tests_test_consumer_by_name_LDADD		= $(test_epc_libs)
-tests_test_dispatcher_local_collision_CFLAGS	= $(example_epc_cflags)
-tests_test_dispatcher_local_collision_LDADD	= $(test_epc_libs)
-tests_test_dispatcher_multiple_services_CFLAGS	= $(example_epc_cflags)
-tests_test_dispatcher_multiple_services_LDADD	= $(test_epc_libs)
-tests_test_dispatcher_rename_CFLAGS		= $(example_epc_cflags)
-tests_test_dispatcher_rename_LDADD		= $(test_epc_libs)
-tests_test_dispatcher_reset_CFLAGS		= $(example_epc_cflags)
-tests_test_dispatcher_reset_LDADD		= $(test_epc_libs)
-tests_test_dispatcher_simple_service_CFLAGS	= $(example_epc_cflags)
-tests_test_dispatcher_simple_service_LDADD	= $(test_epc_libs)
-tests_test_dispatcher_subtypes_CFLAGS		= $(example_epc_cflags)
-tests_test_dispatcher_subtypes_LDADD		= $(test_epc_libs)
-tests_test_dispatcher_unique_CFLAGS		= $(example_epc_cflags)
-tests_test_dispatcher_unique_LDADD		= $(test_epc_libs)
-tests_test_expand_name_CFLAGS			= $(example_epc_cflags)
-tests_test_expand_name_LDADD			= $(test_epc_libs)
-tests_test_progress_hooks_CFLAGS		= $(example_epc_ui_cflags)
-tests_test_progress_hooks_LDADD			= $(test_epc_ui_libs)
-tests_test_publisher_bookmarks_CFLAGS		= $(example_epc_cflags)
-tests_test_publisher_bookmarks_LDADD		= $(test_epc_libs)
-tests_test_publisher_change_name_CFLAGS		= $(example_epc_cflags)
-tests_test_publisher_change_name_LDADD		= $(test_epc_libs)
-tests_test_publisher_libsoup_494128_CFLAGS	= $(example_epc_cflags)
-tests_test_publisher_libsoup_494128_LDADD	= $(test_epc_libs)
-tests_test_publisher_unique_CFLAGS		= $(example_epc_cflags)
-tests_test_publisher_unique_LDADD		= $(test_epc_libs)
-tests_test_service_type_CFLAGS			= $(example_epc_cflags)
-tests_test_service_type_LDADD			= $(test_epc_libs)
-
 # ==================
 # -- CUSTOM RULES --
 # ==================
