--- configure.orig	2010-11-24 00:25:38.000000000 -0800
+++ configure	2010-11-24 00:27:37.000000000 -0800
@@ -14980,7 +14980,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               \$LIBSOUP
                               \$GIO\"") >&5
   ($PKG_CONFIG --exists --print-errors "avahi-client >= 0.6
@@ -14990,7 +14990,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO") 2>&5
   ac_status=$?
@@ -15003,7 +15003,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO" 2>/dev/null`
 else
@@ -15023,7 +15023,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               \$LIBSOUP
                               \$GIO\"") >&5
   ($PKG_CONFIG --exists --print-errors "avahi-client >= 0.6
@@ -15033,7 +15033,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO") 2>&5
   ac_status=$?
@@ -15046,7 +15046,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO" 2>/dev/null`
 else
@@ -15073,7 +15073,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO" 2>&1`
         else
@@ -15084,7 +15084,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO" 2>&1`
         fi
@@ -15098,7 +15098,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO) were not met:
 
@@ -15118,7 +15118,7 @@
                               gmodule-2.0  >= 2.0
                               gobject-2.0  >= 2.4
                               gnutls       >= 1.4
-                              uuid         >= 1.36
+                              uuid         >= 1.6.2
                               $LIBSOUP
                               $GIO) were not met:
 
