--- srclib/stdint.in.h.orig 2007-11-11 10:00:02.000000000 -0600
+++ srclib/stdint.in.h 2009-01-16 01:49:06.000000000 -0600
@@ -410,10 +410,13 @@
 /* ptrdiff_t limits */
 #undef PTRDIFF_MIN
 #undef PTRDIFF_MAX
-#define PTRDIFF_MIN  \
-   _STDINT_MIN (1, @BITSIZEOF_PTRDIFF_T@, 0@PTRDIFF_T_SUFFIX@)
-#define PTRDIFF_MAX  \
-   _STDINT_MAX (1, @BITSIZEOF_PTRDIFF_T@, 0@PTRDIFF_T_SUFFIX@)
+#ifdef __LP64__
+#define PTRDIFF_MIN  _STDINT_MIN (1, 64, 0l)
+#define PTRDIFF_MAX  _STDINT_MAX (1, 64, 0l)
+#else
+#define PTRDIFF_MIN  _STDINT_MIN (1, 32, 0)
+#define PTRDIFF_MAX  _STDINT_MAX (1, 32, 0)
+#endif
 
 /* sig_atomic_t limits */
 #undef SIG_ATOMIC_MIN
@@ -428,7 +431,11 @@
 
 /* size_t limit */
 #undef SIZE_MAX
-#define SIZE_MAX  _STDINT_MAX (0, @BITSIZEOF_SIZE_T@, 0@SIZE_T_SUFFIX@)
+#ifdef __LP64__
+#define SIZE_MAX  _STDINT_MIN (0, 64, 0@SIZE_T_SUFFIX@)
+#else
+#define SIZE_MAX  _STDINT_MIN (0, 32, 0@SIZE_T_SUFFIX@)
+#endif
 
 /* wchar_t limits */
 #undef WCHAR_MIN
