--- src/geoip2.c.orig	2025-03-31 20:10:17
+++ src/geoip2.c	2025-04-01 07:38:41
@@ -47,6 +47,12 @@
 #include "util.h"
 #include "xmalloc.h"
 
+#if __APPLE__
+  #ifndef HOST_NAME_MAX
+    #define HOST_NAME_MAX 255
+  #endif
+#endif
+
 /* should be reused across lookups */
 static int geoip_asn_type = 0;
 static int geoip_city_type = 0;
--- src/wsauth.c.orig	2025-03-31 20:10:17
+++ src/wsauth.c	2025-04-01 07:41:37
@@ -50,6 +50,12 @@
 #include "pdjson.h"
 #include "settings.h"
 #include "xmalloc.h"
+
+#if __APPLE__
+  #ifndef HOST_NAME_MAX
+    #define HOST_NAME_MAX 255
+  #endif
+#endif
 
 char *
 read_secret_from_file (const char *path) {
