--- tests/Makefile.in.orig	2011-07-21 12:34:27.000000000 -0400
+++ tests/Makefile.in	2011-07-21 12:34:27.000000000 -0400
@@ -89,7 +89,6 @@
 @HAVE_CXX_TRUE@am_autotestkeywords_OBJECTS =  \
 @HAVE_CXX_TRUE@	autotestkeywords-autotestkeywords.$(OBJEXT)
 autotestkeywords_OBJECTS = $(am_autotestkeywords_OBJECTS)
-autotestkeywords_LDADD = $(LDADD)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
 am__v_lt_0 = --silent
@@ -685,6 +684,7 @@
 testtooltips_DEPENDENCIES = $(TEST_DEPS)
 testvolumebutton_DEPENDENCIES = $(TEST_DEPS)
 testwindows_DEPENDENCIES = $(TEST_DEPS)
+autotestkeywords_LDADD = $(LDADDS)
 flicker_LDADD = $(LDADDS)
 simple_LDADD = $(LDADDS)
 print_editor_LDADD = $(LDADDS)
