--- gdk/makegdkalias.pl.orig	2011-06-28 01:13:10.000000000 -0700
+++ gdk/makegdkalias.pl	2011-06-28 01:13:49.000000000 -0700
@@ -14,7 +14,7 @@ print <<EOF;
 
 #include <glib.h>
 
-#ifdef G_HAVE_GNUC_VISIBILITY
+#if defined(G_HAVE_GNUC_VISIBILITY) && !defined(__APPLE__)
 
 EOF
 
--- gtk/makegtkalias.pl.orig	2011-06-28 09:18:59.000000000 -0700
+++ gtk/makegtkalias.pl	2011-06-28 09:19:08.000000000 -0700
@@ -14,7 +14,7 @@ print <<EOF;
 
 #include <glib.h>
 
-#ifdef G_HAVE_GNUC_VISIBILITY
+#if defined(G_HAVE_GNUC_VISIBILITY) && !defined(__APPLE__)
 
 EOF
 
