OS X 10.8 support
http://www.fltk.org/str.php?L2864
--- common/fltk/src/filename_list.cxx.orig	2011-01-05 10:50:10.000000000 -0600
+++ common/fltk/src/filename_list.cxx	2012-08-14 04:23:22.000000000 -0500
@@ -31,7 +22,9 @@
 #include <FL/fl_utf8.h>
 #include "flstring.h"
 #include <stdlib.h>
-
+#ifdef __APPLE__
+#include <FL/x.H>
+#endif
 
 extern "C" {
 #ifndef HAVE_SCANDIR
@@ -115,6 +108,12 @@
   int n = scandir(dirloc, list, 0, (int(*)(void*, void*))sort);
 #elif defined(__sgi)
   int n = scandir(dirloc, list, 0, sort);
+#elif defined(__APPLE__)
+# if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_8
+  int n = scandir(dirloc, list, 0, (int(*)(const struct dirent**,const struct dirent**))sort);
+# else
+  int n = scandir(dirloc, list, 0, (int(*)(const void*,const void*))sort);
+# endif
 #else
   // The vast majority of UNIX systems want the sort function to have this
   // prototype, most likely so that it can be passed to qsort without any
