--- release/makemacapp.in.orig	2023-02-03 02:17:15.000000000 -0600
+++ release/makemacapp.in	2023-02-20 19:18:39.000000000 -0600
@@ -36,8 +36,7 @@
 fi
 
 umask 022
-TMPDIR=`mktemp -d /tmp/$PACKAGE_NAME-build.XXXXXX`
-APPROOT="$TMPDIR/dmg/TigerVNC Viewer $VERSION.app"
+APPROOT="./TigerVNC Viewer.app"
 mkdir -p "$APPROOT/Contents/MacOS"
 mkdir -p "$APPROOT/Contents/Resources"
 
@@ -51,6 +50,8 @@
 		"$APPROOT/Contents/Resources/locale/$lang/LC_MESSAGES/tigervnc.mo"
 done
 
+exit
+
 install -m 644 $SRCDIR/LICENCE.TXT $TMPDIR/dmg/
 install -m 644 $SRCDIR/README.rst $TMPDIR/dmg/
 
