--- vncviewer/CMakeLists.txt.orig	2024-07-23 03:50:25.000000000 -0500
+++ vncviewer/CMakeLists.txt	2024-08-15 20:47:27.000000000 -0500
@@ -80,6 +80,8 @@
     list(APPEND po_FILES "${CMAKE_SOURCE_DIR}/po/${lang}.po")
   endforeach()
 
+endif()
+if(FALSE)
   configure_file(vncviewer.desktop.in.in vncviewer.desktop.in)
   find_program(INTLTOOL_MERGE_EXECUTABLE intltool-merge)
   if("${GETTEXT_VERSION_STRING}" VERSION_GREATER 0.18.99)
