--- unix/tixUnixXpm.c.orig      2006-11-15 14:18:36.000000000 -0800
+++ unix/tixUnixXpm.c   2018-11-07 05:35:43.000000000 -0800
@@ -17,7 +17,11 @@
 #include <tixPort.h>
 #include <tixUnixInt.h>
 #include <tixImgXpm.h>
+#ifndef MAC_OSX_TK
 #include <X11/Xutil.h>
+#else
+#include <X11_tk/Xutil.h>
+#endif
 
 #if !defined(WIN32) && !defined(MAC_OSX_TK)
 #define TkPutImage(colors, ncolors, display, pixels, gc, image, destx, desty, srcx, srcy, width, height) \
