--- externals/bundles/openssl/1.0.1e/CMakeLists.txt.orig	2019-02-12 11:03:39.000000000 -0500
+++ externals/bundles/openssl/1.0.1e/CMakeLists.txt	2019-02-12 11:07:37.000000000 -0500
@@ -20,6 +20,10 @@
 
 if (WITH_OPENSSL)
 
+	find_multiple( "ssl;crypto" OPENSSL_FOUND )
+
+	if ( NOT OPENSSL_FOUND AND NOT PROPER )
+
 	if ( WINDOWS )
   
 		condor_pre_external( OPENSSL openssl-1.0.1e "lib;include" "include/openssl/ssl.h")
@@ -105,6 +109,7 @@
 		endif()
 
 	endif( )
+	endif( )
 
 	if (OPENSSL_FOUND)
 		message (STATUS "external configured (OPENSSL_FOUND=${OPENSSL_FOUND})")
