--- src/python-bindings/CMakeLists.txt
+++ src/python-bindings/CMakeLists.txt
@@ -101,9 +101,9 @@ else()
     set ( CMAKE_LIBRARY_PATH CMAKE_LIBRARY_PATH_ORIG)
 
     if (PROPER AND "${PYTHON_VERSION_MAJOR}" MATCHES "3")
-      set ( PYTHON_BOOST_LIB boost_python3 )
+      set ( PYTHON_BOOST_LIB boost_python3-mt )
     else()
-      set ( PYTHON_BOOST_LIB boost_python )
+      set ( PYTHON_BOOST_LIB boost_python-mt )
     endif()
 
     include_directories(${PYTHON_INCLUDE_DIRS} ${BOOST_INCLUDE})
