--- doc/Makefile.orig	2015-12-15 10:20:16.000000000 -0500
+++ doc/Makefile	2016-01-07 02:54:38.000000000 -0500
@@ -86,7 +86,7 @@
 	latex $(LATEXARGS)  ref.tex
 	makeindex ref
 	latex $(LATEXARGS)  ref.tex
-	latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons ref.tex
+	@prefix@/bin/latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons ref.tex
 	cd ref && perl -pi.bak -e 's,\<HEAD\>,<HEAD>\n<link rel="canonical" href="http://research.cs.wisc.edu/htcondor/manual/current/$$ARGV">\n,' condor_*.html
 	rm -f ref.dvi
 	mv ref condor-$(SUBVER)-Manual
@@ -120,7 +120,7 @@
 	latex $(LATEXARGS)  ref.tex
 	makeindex ref
 	latex $(LATEXARGS)  ref.tex
-	latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons ref.tex
+	@prefix@/bin/latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons ref.tex
 	rm -rf $(LOCALDIR)
 	mv    ref $(LOCALDIR)
 	touch local.html
@@ -170,7 +170,7 @@
 	ln -s symbol-html.tex symbol.tex
 	ln -s figuresizing-html.tex figuresizing.tex
 	cd makeman && make clean makeman
-	latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons just-man-pages.tex
+	@prefix@/bin/latex2html -split 4 -link 2 -show_section_numbers -antialias -tmp /tmp -long_titles 3 -toc_depth 2 -local_icons just-man-pages.tex
 	for html in $(MANPAGES); do \
 		thisfile=just-man-pages/$$html; \
 		makeman/makeman -v -i $$thisfile -s 1; \
