--- ./Makefile	2022-09-18 08:53:58.000000000 -0400
+++ ./Makefile	2022-09-18 08:59:27.000000000 -0400
@@ -80,7 +80,7 @@
 # If a suitable language-c version is not readily available, you can run this
 # to install the included language-c 0.5.0 locally using cabal.
 cabal-install:
-	cd language-c && cabal v1-update && cabal v1-install
+	cd language-c && @CABAL_UPDATE_CMD@ && @CABAL_BUILD_CMD@
 
 # These options work around quirks with various ghc versions.
 GHCXFLAGS += -XTypeSynonymInstances -XFlexibleInstances
@@ -107,7 +107,7 @@
 
 maintainer-build:
 # Install locally into the build directory.
-	cd language-c && HOME=$(shell pwd)/cabal cabal v1-update && HOME=$(shell pwd)/cabal cabal v1-install
+	cd language-c && HOME=$(shell pwd)/cabal @CABAL_UPDATE_CMD@ && HOME=$(shell pwd)/cabal @CABAL_INSTALL_CMD@
 # Make sure to do a static build here, since the cabal-installed language-c
 # will be gone once we're finished.
 	$(MAKE) all GHCFLAGS=-static HOME=$(shell pwd)/cabal
