--- src/Makefile.in.orig	2017-02-12 04:20:01.000000000 -0600
+++ src/Makefile.in	2017-02-13 10:35:44.000000000 -0600
@@ -258,7 +258,7 @@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = $(LIBGCRYPT_CFLAGS)
 LDADD = $(LIBGCRYPT_LIBS)
-AM_CFLAGS = -I.. -Wreturn-type -Wmaybe-uninitialized -Wunused-variable
+AM_CFLAGS = -I.. -Wreturn-type -Wunused-variable
 ffe_SOURCES = ffe.c xmalloc.c parserc.c execute.c endian.c level.c anonymize.c
 noinst_HEADERS = ffe.h
 all: all-am
