--- DESCRIPTION	2023-04-29 00:10:02.000000000 +0800
+++ DESCRIPTION	2023-11-03 19:20:45.000000000 +0800
@@ -18,11 +18,10 @@
 License: GPL-2
 Imports: Rcpp (>= 0.12.12), foreach, iterators
 LinkingTo: Rcpp, BH, RcppArmadillo, testthat
-Suggests: testthat, Matrix, RcppArmadillo, dplyr, RWeka, entropy,
-        FSelector, randomForest, doParallel, rpart, MASS, covr,
-        parallel, htmltools, magrittr, knitr, RTCGA.rnaseq, ggplot2,
-        microbenchmark, pbapply, tibble, rmarkdown, lintr, pkgdown,
-        withr
+Suggests: testthat, Matrix, RcppArmadillo, dplyr, entropy,
+        randomForest, doParallel, rpart, MASS, covr, pkgdown,
+        parallel, htmltools, magrittr, knitr, ggplot2, withr,
+        microbenchmark, pbapply, tibble, rmarkdown, lintr
 RoxygenNote: 7.2.3
 Encoding: UTF-8
 BugReports: https://github.com/mi2-warsaw/FSelectorRcpp/issues
