diff --git a/tests/run.sh b/tests/run.sh
index 683c545..1294733 100755
--- tests/run.sh
+++ tests/run.sh
@@ -10,7 +10,7 @@ mkdir -p tests/tmp
 export PYTHONSUFFIX=$($PYTHON -V 2>&1 |awk '{print $2}' |awk -F. '{print $1 "." $2}')
 export PYTHONPATH=$(ls -d build/lib.*$PYTHONSUFFIX):$PYTHONPATH
 
-extra_attrs=
+extra_attrs="$extra_attrs"
 if test "$CI" = true; then
   if test -n "$USECURL" && echo "$USECURL" |grep -q gssapi; then
     :
