--- setup.py	Sat Aug 16 19:35:17 2003
+++ setup.py.new	Thu Oct 23 23:56:34 2003
@@ -16,7 +16,7 @@
 from distutils.util import split_quoted
 from distutils.version import LooseVersion
 
-include_dirs = []
+include_dirs = ["/opt/local/include"]
 define_macros = []
 library_dirs = []
 libraries = []
@@ -138,7 +138,7 @@
     if sys.platform == "win32":
         datadir = os.path.join("doc", PACKAGE)
     else:
-        datadir = os.path.join("share", "doc", PACKAGE)
+        datadir = os.path.join("share", "doc", "py-curl")
     #
     files = ["ChangeLog", "COPYING", "INSTALL", "README", "TODO",]
     if files:
