--- GNUmakefile.in.orig	2015-10-05 12:19:44.000000000 -0700
+++ GNUmakefile.in	2017-11-12 23:10:07.000000000 -0700
@@ -17,7 +17,6 @@
 	$(MAKE) -C doc all
 
 world:
-	$(MAKE) -C doc all
 	$(MAKE) -C src all
 	$(MAKE) -C config all
 	$(MAKE) -C contrib all
@@ -35,7 +34,6 @@
 	$(MAKE) -C doc install
 
 install-world:
-	$(MAKE) -C doc install
 	$(MAKE) -C src install
 	$(MAKE) -C config install
 	$(MAKE) -C contrib install
