--- stk500v2.c.orig	2021-11-11 14:59:25.000000000 -0600
+++ stk500v2.c	2022-01-09 00:39:31.000000000 -0600
@@ -741,7 +741,8 @@
   DEBUG("STK500V2: stk500v2_getsync()\n");
 
   if (PDATA(pgm)->pgmtype == PGMTYPE_JTAGICE_MKII ||
-      PDATA(pgm)->pgmtype == PGMTYPE_JTAGICE3)
+      PDATA(pgm)->pgmtype == PGMTYPE_JTAGICE3 ||
+      PDATA(pgm)->pgmtype == PGMTYPE_AVRISP_MKII)
     return 0;
 
 retry:
