an upstream patch for clang compile error
--- src/sdk/wxscintilla/src/PlatWX.cpp.orig
+++ src/sdk/wxscintilla/src/PlatWX.cpp
@@ -299,12 +299,14 @@ void SurfaceImpl::Polygon(Point *pts, int npts, ColourDesired fore, ColourDesire
     PenColour(fore);
     BrushColour(back);
 /* C::B begin */
-    // Since Point now has float as x and y, it seems no longer to be interchangeable with wxPoint,
-    // and the polygon is not drawn, so we convert it explicitely
-    wxPoint points[npts];
-    for (int i = 0; i < npts; ++i)
-        points[i] = wxPoint(pts[i].x, pts[i].y);
-    hDC->DrawPolygon(npts, points);
+    wxPoint *p = new wxPoint[npts];
+
+    for (int i=0; i<npts; i++) {
+        p[i].x = pts[i].x;
+        p[i].y = pts[i].y;
+    }
+    hDC->DrawPolygon(npts, p);
+    delete [] p;
 /* C::B end */
 }
 
