Upstream patches (fixed in r9375 & r8639)
--- src/sdk/configmanager.cpp.orig
+++ src/sdk/configmanager.cpp
@@ -36,7 +36,11 @@
 #endif
 
 #ifdef __WXMAC__
+#if wxCHECK_VERSION(2,9,0)
+#include "wx/osx/core/cfstring.h"
+#else
 #include "wx/mac/corefoundation/cfstring.h"
+#endif
 #include "wx/intl.h"
 
 #include <CoreFoundation/CFBundle.h>
@@ -121,7 +125,11 @@
             CFRelease(resourcesURL);
             CFStringRef cfStrPath = CFURLCopyFileSystemPath(absoluteURL,kCFURLPOSIXPathStyle);
             CFRelease(absoluteURL);
-            wxString str = wxMacCFStringHolder(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+            #if wxCHECK_VERSION(2,9,0)
+              wxString str = wxCFStringRef(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+            #else
+              wxString str = wxMacCFStringHolder(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+            #endif
             if (!str.Contains(wxString(_T("/Resources"))))
                return ::DetermineExecutablePath() + _T("/.."); // not a bundle, use relative path
             return str;
--- src/sdk/xtra_res.cpp.orig
+++ src/sdk/xtra_res.cpp
@@ -15,6 +15,7 @@
     #include <wx/wx.h>
 #endif
 
+#include <wx/xml/xml.h>
 
 /////////////////////////////////////////////////////////////////////////////
 // Name:        xh_toolb.cpp
--- src/src/app.cpp.orig
+++ src/src/app.cpp
@@ -271,7 +271,11 @@
 END_EVENT_TABLE()
 
 #ifdef __WXMAC__
+#if wxCHECK_VERSION(2,9,0)
+#include "wx/osx/core/cfstring.h"
+#else
 #include "wx/mac/corefoundation/cfstring.h"
+#endif
 #include "wx/intl.h"
 
 #include <CoreFoundation/CFBundle.h>
@@ -286,7 +290,11 @@
     CFRelease(resourcesURL);
     CFStringRef cfStrPath = CFURLCopyFileSystemPath(absoluteURL,kCFURLPOSIXPathStyle);
     CFRelease(absoluteURL);
-    return wxMacCFStringHolder(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+    #if wxCHECK_VERSION(2,9,0)
+      return wxCFStringRef(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+    #else
+      return wxMacCFStringHolder(cfStrPath).AsString(wxLocale::GetSystemEncoding());
+    #endif
 }
 #endif
 
