--- cmd/gvedit/Makefile.in.orig	2012-03-20 23:46:40.000000000 -0500
+++ cmd/gvedit/Makefile.in	2012-03-21 04:45:06.000000000 -0500
@@ -66,8 +66,7 @@
 	gvedit-mdichild.$(OBJEXT) $(am__objects_1)
 gvedit_OBJECTS = $(am_gvedit_OBJECTS)
 am__DEPENDENCIES_1 =
-gvedit_DEPENDENCIES = $(top_builddir)/lib/gvc/libgvc.la \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
+gvedit_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
@@ -534,7 +533,7 @@
 	-DGVEDIT_DATADIR=\""$(pkgdatadir)/gvedit"\"
 
 gvedit_LDADD = \
-	$(top_builddir)/lib/gvc/libgvc.la \
+	-lgvc \
 	$(QTGUI_LIBS) \
 	$(QTCORE_LIBS) \
 	$(EXPAT_LIBS) $(Z_LIBS) 
