From d94892a0322fcb8f1a2a8b15acb701be9678be39 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 3 Apr 2024 08:31:44 +0800
Subject: [PATCH] Minor fix to CLIENT_DATA_DIR for macOS

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index f3eea016..ae08c726 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -194,7 +194,7 @@ if (NOT SHARE_DIR)
 endif (NOT SHARE_DIR)
 
 if (APPLE)
-    set (CLIENT_DATA_DIR "${PROJECT_NAME_GLOBAL}.app/Contents/Resources/")
+    set (CLIENT_DATA_DIR "${PROJECT_NAME_GLOBAL}.app/Contents/Resources")
     set (CLIENT_DOCS_DIR "${CLIENT_DATA_DIR}/docs")
 elseif (WIN32)
     set (CLIENT_DATA_DIR "${SHARE_DIR}")
