--- cpp/CMakeLists.txt.orig	2010-01-08 12:51:36.000000000 -0600
+++ cpp/CMakeLists.txt	2010-01-08 12:55:27.000000000 -0600
@@ -6,6 +6,7 @@
 SET(SOURCES flann.cpp util/Random.cpp nn/Testing.cpp algorithms/NNIndex.cpp algorithms/dist.cpp util/Logger.cpp)
 
 ADD_LIBRARY(flann SHARED ${SOURCES})
+SET_TARGET_PROPERTIES(flann PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
 ADD_LIBRARY(flann_s STATIC ${SOURCES})
 
 IF(WIN32)
@@ -17,11 +18,6 @@
     TARGETS flann
     RUNTIME DESTINATION python/pyflann/bindings
 )
-ELSE(WIN32)
-INSTALL (
-    TARGETS flann
-    LIBRARY DESTINATION python/pyflann/bindings
-)
 ENDIF(WIN32)
 
 INSTALL (
