From e6b1d05d3c8791e83831f8587b4c54bf97a43f08 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Wed, 26 Jul 2023 19:01:04 +0800
Subject: [PATCH] Altivec code is broken for Darwin, so disable it there

---
 OOps/pffft.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git OOps/pffft.c OOps/pffft.c
index e9696a77c..87862b531 100644
--- OOps/pffft.c
+++ OOps/pffft.c
@@ -100,9 +100,9 @@
 //#define PFFFT_SIMD_DISABLE
 
 /*
-   Altivec support macros
+   Altivec support macros. (The code below is broken for Darwin, so disable it explicitly.)
 */
-#if !defined(PFFFT_SIMD_DISABLE) && (defined(__ppc__) || defined(__ppc64__))
+#if !defined(PFFFT_SIMD_DISABLE) && (defined(__ppc__) || defined(__ppc64__)) && !defined(__APPLE__)
 typedef vector float v4sf;
 #  define SIMD_SZ 4
 #  define VZERO() ((vector float) vec_splat_u8(0))
