--- pluginSDK/SConstruct.orig	2012-02-13 18:17:02.000000000 +1100
+++ pluginSDK/SConstruct	2012-03-08 15:57:17.000000000 +1100
@@ -87,6 +87,7 @@
 opts.Add('customLIBS')
 opts.Add('customLIBPATH')
 opts.Add('customSHLINKFLAGS')
+opts.Add('customLINKFLAGS')
 opts.Add('customSWIGFLAGS')
 opts.Add('useDouble',
     'Set to 1 to use double-precision floating point for audio samples.',
@@ -135,6 +136,8 @@
 commonEnvironment.Prepend(LIBPATH = customLIBPATH)
 customSHLINKFLAGS = commonEnvironment['customSHLINKFLAGS']
 commonEnvironment.Prepend(SHLINKFLAGS = customSHLINKFLAGS)
+customLINKFLAGS = commonEnvironment['customLINKFLAGS']
+commonEnvironment.Prepend(LINKFLAGS = customLINKFLAGS)
 customSWIGFLAGS = commonEnvironment['customSWIGFLAGS']
 commonEnvironment.Prepend(SWIGFLAGS = customSWIGFLAGS)
 
@@ -192,7 +195,7 @@
     commonEnvironment.Prepend(LIBPATH = ['.', '#.', '/usr/local/lib64'])
     commonEnvironment.Append(CCFLAGS = ['-fPIC'])
 else:
-    commonEnvironment.Prepend(LIBPATH = ['.', '#.', '/usr/local/lib'])
+    commonEnvironment.Prepend(LIBPATH = ['.', '#.'])
 
 if commonEnvironment['useDouble'] == '0':
     print 'CONFIGURATION DECISION: Using single-precision floating point for audio samples.'
@@ -214,7 +217,6 @@
     commonEnvironment.Append(LINKFLAGS = ['-Wl,-Bdynamic'])
 elif getPlatform() == 'darwin':
     commonEnvironment.Append(CCFLAGS = "-DMACOSX")
-    commonEnvironment.Append(CPPPATH = '/usr/local/include')
     commonEnvironment.Append(CCFLAGS = "-DPIPES")
     if commonEnvironment['useAltivec'] == '1':
         print 'CONFIGURATION DECISION using Altivec optmisation'
@@ -278,9 +280,8 @@
 #pluginEnvironment.Append(LIBS = ['sndfile'])
 
 if getPlatform() == 'darwin':
-    pluginEnvironment.Append(LINKFLAGS = Split('''
-        -framework CoreMidi -framework CoreFoundation -framework CoreAudio
-    '''))
+    for f in ['CoreMIDI', 'CoreFoundation', 'CoreAudio']:
+        pluginEnvironment.Append(FRAMEWORKS = [f])
     # pluginEnvironment.Append(LINKFLAGS = ['-dynamiclib'])
     pluginEnvironment['SHLIBSUFFIX'] = '.dylib'
     pluginEnvironment.Prepend(CXXFLAGS = "-fno-rtti")
