--- libopenjpeg/Makefile.in.orig	2011-04-18 12:41:35.000000000 -0700
+++ libopenjpeg/Makefile.in	2011-04-18 12:42:08.000000000 -0700
@@ -616,11 +616,11 @@
 
 all-local:
 	@$(INSTALL) -d ../bin
-	@cp -P .libs/libopenjpeg.* ../bin/
+	@cp -PR .libs/libopenjpeg.* ../bin/
 	@rm -f ../bin/libopenjpeg.la*
 	@rm -rf .report_lib
 	@$(INSTALL) -d .report_lib
-	@cp -P .libs/libopenjpeg.* .report_lib
+	@cp -PR .libs/libopenjpeg.* .report_lib
 	@rm -f .report_lib/libopenjpeg.lai
 	@echo "" > .report.txt
 	@echo "Installing: ${prefix}/include/${headerdir}/" >> .report.txt
