--- configure.ac.orig	2024-03-17 15:29:06.000000000 -0500
+++ configure.ac	2024-08-05 21:47:28.000000000 -0500
@@ -492,7 +492,7 @@
 	_gdk_tgt=`$PKG_CONFIG --variable=target gdk-2.0`
 fi
 if test "x$_gdk_tgt" = "xquartz"; then
-	BF_gtkmacintegration="gtk-mac-integration"
+	BF_gtkmacintegration="gtk-mac-integration-gtk3"
 	AC_MSG_RESULT([gdk target is $_gdk_tgt])
 	PKG_CHECK_MODULES(
 		[BLUEFISH_GTKMAC],
