Fix version
https://github.com/cdcseacave/TinyEXIF/pull/18
--- CMakeLists.txt.orig	2019-01-16 07:26:30.000000000 -0600
+++ CMakeLists.txt	2024-12-02 05:50:45.000000000 -0600
@@ -10,7 +10,7 @@
 ################################
 # set lib version here
 
-set(GENERIC_LIB_VERSION "1.0.1")
+set(GENERIC_LIB_VERSION "@MAJOR@.@MINOR@.@PATCH@")
 set(GENERIC_LIB_SOVERSION "1")
 
 ################################
--- TinyEXIF.h.orig	2019-01-16 07:26:30.000000000 -0600
+++ TinyEXIF.h	2024-12-02 05:51:01.000000000 -0600
@@ -38,9 +38,9 @@
 #include <string>
 #include <vector>
 
-#define TINYEXIF_MAJOR_VERSION 1
-#define TINYEXIF_MINOR_VERSION 0
-#define TINYEXIF_PATCH_VERSION 1
+#define TINYEXIF_MAJOR_VERSION @MAJOR@
+#define TINYEXIF_MINOR_VERSION @MINOR@
+#define TINYEXIF_PATCH_VERSION @PATCH@
 
 #ifdef _MSC_VER
 #   ifdef TINYEXIF_EXPORT
