--- Makefile.PL.orig	2013-07-09 20:36:31.000000000 -0700
+++ Makefile.PL	2013-07-09 20:35:35.000000000 -0700
@@ -18,13 +18,7 @@
 
 
 my ($enc2xs, $encode_h) = ();
+$enc2xs = "@@ENC2XS@@";
-PATHLOOP:
-for my $d (split /:/, $ENV{PATH}){
-    for my $f (qw/enc2xs enc2xs5.7.3/){
-        my $path = File::Spec->catfile($d, $f);
-        -x $path and $enc2xs = $path and last PATHLOOP;
-    }
-}
 $enc2xs or die "enc2xs not found!";
 print "enc2xs is $enc2xs\n";
 my %encode_h = ();
