diff --git a/host/lib/usrp/gpsd_iface.cpp b/host/lib/usrp/gpsd_iface.cpp
index b34132f..b439a80 100644
--- a/host/lib/usrp/gpsd_iface.cpp
+++ b/host/lib/usrp/gpsd_iface.cpp
@@ -119,7 +119,7 @@ private: // member functions
                 _timeout_cnt = 0;
                 _detected = true;
 
-                if (gps_read(&_gps_data) < 0)
+                if (gps_read(&_gps_data, NULL, 0) < 0)
                     throw std::runtime_error("error while reading");
             }
         }
