--- host/lib/deps/rpclib/include/rpc/detail/async_writer.h.orig
+++ host/lib/deps/rpclib/include/rpc/detail/async_writer.h
@@ -69,7 +69,7 @@ public:
 
 protected:
     boost::asio::ip::tcp::socket socket_;
-    boost::asio::strand write_strand_;
+    boost::asio::io_service::strand write_strand_;
     std::atomic_bool exit_{false};
     bool exited_ = false;
     std::mutex m_exit_;
--- host/lib/deps/rpclib/include/rpc/detail/server_session.h.orig
+++ host/lib/deps/rpclib/include/rpc/detail/server_session.h
@@ -35,7 +35,7 @@ private:
 private:
     server* parent_;
     boost::asio::io_service *io_;
-    boost::asio::strand read_strand_;
+    boost::asio::io_service::strand read_strand_;
     std::shared_ptr<dispatcher> disp_;
     RPCLIB_MSGPACK::unpacker pac_;
     RPCLIB_MSGPACK::sbuffer output_buf_;
--- host/lib/deps/rpclib/lib/rpc/client.cc.orig
+++ host/lib/deps/rpclib/lib/rpc/client.cc
@@ -133,7 +133,7 @@ struct client::impl {
 
     client *parent_;
     boost::asio::io_service io_;
-    boost::asio::strand strand_;
+    boost::asio::io_service::strand strand_;
     std::atomic<int> call_idx_; /// The index of the last call made
     std::unordered_map<uint32_t, call_t> ongoing_calls_;
     std::string addr_;
