diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 78fff72..9f6e028 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -284,27 +284,60 @@ endif(WIN32)
 # Choose a version of Python
 ########################################################################
 option(ENABLE_PYTHON3 "Enable Python 3. Default is Python 2" OFF)
-set(BOOST_PYTHON_COMPONENT python)
-set(BOOST_PYTHON_VERSION 2.7)
-if(${ENABLE_PYTHON3})
-    set(BOOST_PYTHON_COMPONENT python3)
-    set(BOOST_PYTHON_VERSION 3)
+
+########################################################################
+# Find Boost Version
+########################################################################
+message(STATUS "")
+message(STATUS "Determining Boost Version...")
+unset(UHD_BOOST_OPTIONAL_COMPONENTS)
+unset(UHD_BOOST_REQUIRED_COMPONENTS)
+set(Boost_FIND_QUIET TRUE)
+include(UHDBoost)
+if(Boost_FOUND)
+    message(STATUS "Determining Boost Version... found ${Boost_VERSION}")
+else()
+    message(STATUS "Determining Boost Version... Boost not found!")
+endif()
+
+########################################################################
+# Setup Boost : Optional Component(s)
+########################################################################
+message(STATUS "")
+message(STATUS "Looking for Boost::Python ...")
+if(${Boost_VERSION} VERSION_LESS_EQUAL "1.66.0")
+    if(${ENABLE_PYTHON3})
+        set(BOOST_PYTHON_COMPONENT python3)
+        set(BOOST_PYTHON_VERSION 3)
+    else()
+        set(BOOST_PYTHON_COMPONENT python)
+        set(BOOST_PYTHON_VERSION 2.7)
+    endif()
+else()
+    if(${ENABLE_PYTHON3})
+        set(BOOST_PYTHON_COMPONENT python@PythonVersionNoDot@)
+        set(BOOST_PYTHON_VERSION @PythonVersionWithDot@)
+    else()
+        set(BOOST_PYTHON_COMPONENT python27)
+        set(BOOST_PYTHON_VERSION 2.7)
+    endif()
 endif()
 set(UHD_BOOST_OPTIONAL_COMPONENTS ${BOOST_PYTHON_COMPONENT})
 unset(UHD_BOOST_REQUIRED_COMPONENTS)
 include(UHDBoost)
 
 #Check to see if the Python version we're looking for exists
-set(BOOST_PYTHON_FOUND ${Boost_PYTHON_FOUND})
-set(BOOST_PYTHON_LIBRARY ${Boost_PYTHON_LIBRARY})
-if(${ENABLE_PYTHON3})
-    set(BOOST_PYTHON_FOUND ${Boost_PYTHON3_FOUND})
-    set(BOOST_PYTHON_LIBRARY ${Boost_PYTHON3_LIBRARY})
+string(TOUPPER ${BOOST_PYTHON_COMPONENT} BOOST_PYTHON_COMPONENT_VAR)
+set(BOOST_PYTHON_COMPONENT_FOUND "Boost_${BOOST_PYTHON_COMPONENT_VAR}_FOUND")
+set(BOOST_PYTHON_FOUND ${${BOOST_PYTHON_COMPONENT_FOUND}})
+if(BOOST_PYTHON_FOUND)
+    set(BOOST_PYTHON_COMPONENT_LIBRARY "Boost_${BOOST_PYTHON_COMPONENT_VAR}_LIBRARY")
+    set(BOOST_PYTHON_LIBRARY ${${BOOST_PYTHON_COMPONENT_LIBRARY}})
 endif()
 
-message(STATUS "Boost Python Found: ${BOOST_PYTHON_FOUND}")
+message(STATUS "Boost::Python Found: ${BOOST_PYTHON_FOUND}")
 if(BOOST_PYTHON_FOUND)
-    message(STATUS "Boost Python Libraries: ${BOOST_PYTHON_LIBRARY}")
+    message(STATUS "Boost::Python library: ${BOOST_PYTHON_LIBRARY}")
 endif()
 
 ########################################################################
