diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index ab7f41f..06cd1af 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -342,8 +342,8 @@
 MESSAGE(STATUS "Looking for required Boost components...")
 FIND_PACKAGE(Boost ${BOOST_MIN_VERSION} COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
 
-INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
-LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
+# INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
+# LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
 
 #Check to see if the Python version we're looking for exists
 SET(BOOST_PYTHON_FOUND ${Boost_PYTHON_FOUND})
