diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 91679a3..4ea597f 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -47,7 +47,7 @@ IF(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
         MESSAGE(WARNING "\nCannot determine the version of the compiler selected to build UHD (GCC : ${CMAKE_CXX_COMPILER}). This build may or not work. We highly recommend using GCC version ${GCC_MIN_VERSION} or more recent.")
     ENDIF()
     SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
-ELSEIF(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+ELSEIF(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     EXECUTE_PROCESS(COMMAND
         ${CMAKE_CXX_COMPILER} -v
         RESULT_VARIABLE res ERROR_VARIABLE err
@@ -83,7 +83,7 @@ ENDIF()
 IF(CMAKE_VERSION VERSION_LESS "3.1")
     IF(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
         SET(CMAKE_CXX_FLAGS "--std=gnu++11 ${CMAKE_CXX_FLAGS}")
-    ELSEIF(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+    ELSEIF(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
         IF("${IS_APPLE}" STREQUAL "")
             SET(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
         ELSE()
@@ -231,7 +231,7 @@ SET(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}" CACHE STRING "")
 SET(UHD_RELEASE_MODE "${UHD_RELEASE_MODE}" CACHE STRING "UHD Release Mode")
 
 IF(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" OR
-    ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+    ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     IF(STRIP_BINARIES)
         IF(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
             SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s")
