diff --git a/host/cmake/Modules/UHDVersion.cmake b/host/cmake/Modules/UHDVersion.cmake
index 35185d9..780fb68 100644
--- a/host/cmake/Modules/UHDVersion.cmake
+++ b/host/cmake/Modules/UHDVersion.cmake
@@ -17,11 +17,11 @@ find_package(Git QUIET)
 #  - Increment patch for bugfixes and docs
 #  - set UHD_VERSION_DEVEL to true for master and development branches
 ########################################################################
-set(UHD_VERSION_MAJOR 3)
-set(UHD_VERSION_API   15)
-set(UHD_VERSION_ABI   0)
-set(UHD_VERSION_PATCH git)
-set(UHD_VERSION_DEVEL TRUE)
+set(UHD_VERSION_MAJOR   3)
+set(UHD_VERSION_API    14)
+set(UHD_VERSION_ABI     0)
+set(UHD_VERSION_PATCH   0)
+set(UHD_VERSION_DEVEL FALSE)
 
 ########################################################################
 # If we're on a development branch, we skip the patch version
