diff --git a/host/CMakeLists.txt b/host/CMakeLists.txt
index 37e9eda..1f579f9 100644
--- a/host/CMakeLists.txt
+++ b/host/CMakeLists.txt
@@ -334,8 +334,8 @@
 message(STATUS "Looking for required Boost components...")
 find_package(Boost ${BOOST_MIN_VERSION} COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
 
-include_directories(${Boost_INCLUDE_DIRS})
-link_directories(${Boost_LIBRARY_DIRS})
+# include_directories(${Boost_INCLUDE_DIRS})
+# link_directories(${Boost_LIBRARY_DIRS})
 
 message(STATUS "Boost include directories: ${Boost_INCLUDE_DIRS}")
 message(STATUS "Boost library directories: ${Boost_LIBRARY_DIRS}")
