--- host/utils/uhd_images_downloader.py.in.orig	2012-11-06 21:42:35.000000000 -0500
+++ host/utils/uhd_images_downloader.py.in	2012-11-06 21:58:25.000000000 -0500
@@ -28,18 +28,31 @@
 
     #Command line options
     parser = OptionParser()
-    parser.add_option("--download-location", type="string", default="@CMAKE_INSTALL_PREFIX@/share/uhd/images", help="Set custom download location for images, [default=%default]")
+    parser.add_option("--download-location", type="string", default="@CMAKE_INSTALL_PREFIX@/share/uhd", help="Set custom download location for images, [default=%default]")
+    parser.add_option("--install-location", type="string", default="@CMAKE_INSTALL_PREFIX@/share/uhd/images", help="Set custom install location for images, [default=%default]")
     parser.add_option("--buffer-size", type="int", default=8192, help="Set download buffer size, [default=%default]",)
     (options, args) = parser.parse_args()
     
     #Configuring image download info
     images_src = "@UHD_IMAGES_DOWNLOAD_SRC@"
     filename = images_src.split("/")[-1]
+
+    #Configuring download destination
+    if options.download_location != "":
+        download_dir = options.download_location
+    else:
+        download_dir = "@CMAKE_INSTALL_PREFIX@/share/uhd/tmp"
     
+    #create download_dir if it does not exist
+    if not os.path.exists(download_dir):
+        os.mkdir (download_dir)
+
+    #make sure we download into the correct directory
+    os.chdir (download_dir)
+
     #Configuring image destination
-    cmake_install_prefix = "@CMAKE_INSTALL_PREFIX@"
-    if options.download_location != "":
-        images_dir = options.download_location
+    if options.install_location != "":
+        images_dir = options.install_location
     else:
         images_dir = "@CMAKE_INSTALL_PREFIX@/share/uhd/images"
     
