--- host/lib/usrp/b200/b200_impl.cpp.orig
+++ host/lib/usrp/b200/b200_impl.cpp
@@ -259,6 +259,7 @@
         return device::sptr(new b200_impl(device_addr, handle));
     }
     catch (const uhd::usb_error &e) {
+        UHD_MSG(status) << "Detected bad USB state; resetting." << std::endl;
         libusb::device_handle::sptr dev_handle(libusb::device_handle::get_cached_handle(
             boost::static_pointer_cast<libusb::special_handle>(handle)->get_device()
         ));
