--- setup.py.orig	2013-03-31 01:50:05.000000000 +0400
+++ setup.py	2013-03-31 01:52:15.000000000 +0400
@@ -77,16 +77,7 @@
     )
 
 def cmdline_entrypoints(versioninfo, platform, basename):
-    target = 'pytest:main'
-    if platform.startswith('java'):
-        points = {'py.test-jython': target}
-    else:
-        if basename.startswith("pypy"):
-            points = {'py.test-%s' % basename: target}
-        else: # cpython
-            points = {'py.test-%s.%s' % versioninfo[:2] : target,}
-        points['py.test'] = target
-    return points
+    return {'py.test': 'pytest:main'}
 
 def make_entry_points():
     basename = os.path.basename(sys.executable)
