--- setup.py.orig	2014-07-20 13:06:16.000000000 +0400
+++ setup.py	2014-07-23 23:08:01.000000000 +0400
@@ -45,16 +45,7 @@
 
 
 def cmdline_entrypoints(versioninfo, platform, basename):
-    target = 'pytest:main'
-    if platform.startswith('java'):
-        points = {'py.test-jython': target}
-    else:
-        if basename.startswith('pypy'):
-            points = {'py.test-%s' % basename: target}
-        else: # cpython
-            points = {'py.test-%s.%s' % versioninfo[:2] : target}
-        points['py.test'] = target
-    return points
+    return {'py.test': 'pytest:main'}
 
 
 def make_entry_points():
