--- setup.py.orig	2017-12-05 23:41:58.000000000 +0300
+++ setup.py	2017-12-24 16:31:49.000000000 +0300
@@ -77,7 +77,7 @@
             'Holger Krekel, Bruno Oliveira, Ronny Pfannschmidt, '
             'Floris Bruynooghe, Brianna Laugher, Florian Bruhin and others'),
         entry_points={'console_scripts': [
-            'pytest=pytest:main', 'py.test=pytest:main']},
+            'py.test=pytest:main']},
         classifiers=classifiers,
         keywords="test unittest",
         cmdclass={'test': PyTest},
