--- lib/lib.pro.orig	2010-11-11 12:13:41.000000000 -0500
+++ lib/lib.pro	2010-11-11 12:14:27.000000000 -0500
@@ -1,24 +1,19 @@
 TEMPLATE        = lib
 QT += network
 TARGET                = QtAssistantClient
-isEmpty(QT_MAJOR_VERSION) {
-   VERSION=4.3.0
-} else {
-   VERSION=$${QT_MAJOR_VERSION}.$${QT_MINOR_VERSION}.$${QT_PATCH_VERSION}
-}
+VERSION=4.6.3
 
 CONFIG                += qt warn_on
-mac|win32:CONFIG                += debug_and_release
+CONFIG                += @ARCHES@ @BUILD_TYPE@ create_prl
 mac:unix:CONFIG       += explicitlib
 CONFIG                -= dll
 
+INCLUDEPATH += ./include
+
 HEADERS         = qassistantclient.h \
                   qassistantclient_global.h
 SOURCES         = qassistantclient.cpp
 
-DESTDIR                = ../../../../lib
-DLLDESTDIR             = ../../../../bin
-
 unix {
         QMAKE_CFLAGS += $$QMAKE_CFLAGS_SHLIB
         QMAKE_CXXFLAGS += $$QMAKE_CXXFLAGS_SHLIB
@@ -31,7 +26,7 @@
 
 #load up the headers info
 CONFIG += qt_install_headers
-HEADERS_PRI = $$QT_BUILD_TREE/include/QtAssistant/headers.pri
+HEADERS_PRI = ./include/QtAssistant/headers.pri
 include($$HEADERS_PRI, "", true)|clear(HEADERS_PRI)
 
 #mac frameworks
@@ -76,3 +71,7 @@
    QMAKE_PKGCONFIG_REQUIRES += QtNetwork
 }
 
+# install feature file
+features.files = ../compat/features/assistant.prf
+features.path = $$[QT_INSTALL_DATA]/mkspecs/features
+INSTALLS += features
