--- aes.c.orig	2003-06-03 02:12:09.000000000 +1000
+++ aes.c	2014-08-30 13:33:09.000000000 +1000
@@ -164,7 +164,7 @@ int main(int argc, char **argv)
 				}
 				file_input=1;
 				strncpy(infilename,args,255);
-				infilename[255] = '\0';
+				infilename[254] = '\0';
 				break;
 			case 8:
 				if (sscanf(args,"%d",&bytecount) < 1) {
@@ -176,7 +176,7 @@ int main(int argc, char **argv)
 			case 9:
 				file_output=1;
 				strncpy(outfilename,args,255);
-				outfilename[255] = '\0';
+				outfilename[254] = '\0';
 				break;
 			case 10:
 				fprintf(stderr,"aes %s\n",AES_VERSION);
