--- src/GList.hh.orig	2010-12-10 07:25:09.000000000 +1100
+++ src/GList.hh	2012-09-12 14:04:37.000000000 +1000
@@ -504,7 +504,7 @@ template <class OBJ> int GArray<OBJ>::Ad
    if (Found(*item, result))
       if (fUnique) return -1; //cannot add a duplicate!
    //Found sets result to the position where the item should be!
-   idxInsert(result, *item);
+   this->idxInsert(result, *item);
    }
   else {
    if (fUnique && Found(*item,result)) return -1; //set behaviour
@@ -534,7 +534,7 @@ template <class OBJ> void GArray<OBJ>::A
     for (int i=0;i<list.fCount;i++) Add(&list[i]);
     }
   else { //simply copy
-    setCapacity(this->fCapacity+list.fCount);
+    this->setCapacity(this->fCapacity+list.fCount);
     int s=this->fCount;
     for (int i=0;i<list.fCount;i++)
            this->fArray[s+i]=list.fArray[i];
@@ -769,7 +769,7 @@ template <class OBJ> void GList<OBJ>::Ad
     for (int i=0;i<list.Count();i++) Add(list[i]);
     }
   else { //simply copy
-    setCapacity(this->fCapacity+list.fCount);
+    this->setCapacity(this->fCapacity+list.fCount);
     memcpy( & (this->fList[this->fCount]), list.fList, list.fCount*sizeof(OBJ*));
     this->fCount+=list.fCount;
     }
