https://github.com/bengardner/uncrustify/commit/d519bfa85552cebc5388b979668ea7c5d1c53853
--- src/align.cpp.orig	2011-09-03 12:06:05.000000000 -0500
+++ src/align.cpp	2012-07-11 23:12:46.000000000 -0500
@@ -247,11 +247,6 @@
       align_oc_msg_colons();
    }
 
-   if (cpd.settings[UO_align_oc_decl_colon].b)
-   {
-      align_oc_decl_colon();
-   }
-
    /* Align variable definitions */
    if ((cpd.settings[UO_align_var_def_span].n > 0) ||
        (cpd.settings[UO_align_var_struct_span].n > 0))
@@ -283,6 +278,12 @@
       align_oc_msg_spec(cpd.settings[UO_align_oc_msg_spec_span].n);
    }
 
+   /* Align OC colons */
+   if (cpd.settings[UO_align_oc_decl_colon].b)
+   {
+      align_oc_decl_colon();
+   }
+
    /* Align variable defs in function prototypes */
    if (cpd.settings[UO_align_func_params].b)
    {
@@ -1939,7 +1940,8 @@
                 &&
                 ((tmp->type == CT_WORD) ||
                  (tmp->type == CT_TYPE) ||
-                 (tmp->type == CT_OC_MSG_DECL))
+                 (tmp->type == CT_OC_MSG_DECL) ||
+                 (tmp->type == CT_OC_MSG_SPEC))
                 &&
                 ((tmp2->type == CT_WORD) ||
                  (tmp2->type == CT_TYPE) ||
