--- CMakeLists.txt.orig	2019-10-20 05:39:07.000000000 -0700
+++ CMakeLists.txt	2019-11-28 10:41:01.000000000 -0700
@@ -55,6 +55,20 @@
     endif()
 
     find_package(OpenMP 4.0 REQUIRED)
+
+    if(OpenMP_C_FLAGS)
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+    endif()
+
+    if(OpenMP_CXX_FLAGS)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    endif()
+
+    list(APPEND LIBS ${OpenMP_C_LIBRARIES} ${OpenMP_CXX_LIBRARIES})
+
+    if(WIN32)
+      list(APPEND LIBS gomp)
+    endif(WIN32)
 endif()
 
 include(ConfigureChecks.cmake)
--- src/CMakeLists.txt.orig	2019-10-20 05:39:07.000000000 -0700
+++ src/CMakeLists.txt	2019-11-28 10:41:23.000000000 -0700
@@ -226,22 +226,6 @@
 # Need to explicitly link against math library.
 list(APPEND LIBS "-lm")
 
-if(USE_OPENMP)
-  if(OpenMP_C_FLAGS)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-  endif()
-
-  if(OpenMP_CXX_FLAGS)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-  endif()
-
-  list(APPEND LIBS ${OpenMP_C_LIBRARIES} ${OpenMP_CXX_LIBRARIES})
-
-  if(WIN32)
-    list(APPEND LIBS gomp)
-  endif(WIN32)
-endif(USE_OPENMP)
-
 if(USE_DARKTABLE_PROFILING)
   add_definitions(-DUSE_DARKTABLE_PROFILING)
   set(SOURCES ${SOURCES} "common/profiling.c")
