--- src/common/exif.cc.orig	2018-06-05 14:31:36.000000000 -0700
+++ src/common/exif.cc	2018-12-22 13:20:37.000000000 -0800
@@ -2644,7 +2644,7 @@
     if(Exiv2::XmpParser::encode(xmpPacket, xmpData,
       Exiv2::XmpParser::useCompactFormat | Exiv2::XmpParser::omitPacketWrapper) != 0)
     {
-      throw Exiv2::Error(1, "[xmp_write] failed to serialize xmp data");
+      throw Exiv2::Error(Exiv2::kerErrorMessage, "Failed to serialize XMP data");
     }
     return g_strdup(xmpPacket.c_str());
   }
@@ -2769,7 +2769,7 @@
     if(Exiv2::XmpParser::encode(xmpPacket, xmpData,
        Exiv2::XmpParser::useCompactFormat | Exiv2::XmpParser::omitPacketWrapper) != 0)
     {
-      throw Exiv2::Error(1, "[xmp_write] failed to serialize xmp data");
+      throw Exiv2::Error(Exiv2::kerErrorMessage, "Failed to serialize XMP data");
     }
 
     // hash the new data and compare it to the old hash (if applicable)
