From ab6c1aef13f204af8ecb973f892a9d5ffd0a930f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 24 Mar 2023 03:57:47 +0700
Subject: [PATCH] liblzf: fix build on macOS

---
 src/lzf/lzf/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lzf/lzf/Makefile b/src/lzf/lzf/Makefile
index a49638f..192718e 100644
--- src/lzf/lzf/Makefile
+++ src/lzf/lzf/Makefile
@@ -6,6 +6,7 @@ all: liblzf.a
 liblzf.a: lzf_c.o lzf_d.o
 	rm -f liblzf.a
 	$(AR) cq liblzf.a lzf_c.o lzf_d.o
+	$(RANLIB) liblzf.a
 
 lzf_c.o: lzf_c.c
 	$(CC) $(FLAGS) -c lzf_c.c
-- 
2.39.1

