--- configure.orig	2008-04-10 15:21:51.000000000 +0200
+++ configure	2009-03-18 21:28:50.000000000 +0100
@@ -6005,7 +6005,7 @@
 paths="$QTDIR/include /usr/local/qt/include /usr/include/qt /usr/include/qt3 \
   /usr/include /usr/X11R6/include/X11/qt /usr/X11R6/include/X11/qt \
   /usr/X11R6/include/qt /usr/X11R6/include /sw/include/qt \
-  /usr/X11R6/include/qt2"
+  /usr/X11R6/include/qt2 /opt/local/include/qt3"
 for path in $paths; do
   if test -f "$path/qapplication.h"; then
     QT_INCLUDES=$path
