--- qucs/qucs_actions.cpp.orig	2013-06-23 00:59:40.000000000 -0500
+++ qucs/qucs_actions.cpp	2013-10-01 01:57:16.000000000 -0500
@@ -622,7 +622,7 @@
 void QucsApp::slotCallFilter()
 {
   Q3Process *QucsFilter =
-    new Q3Process(QString(QucsSettings.BinDir + "qucsfilter"));
+    new Q3Process(QString(QucsSettings.AppDir + "qucsfilter.app/Contents/MacOS/qucsfilter"));
   if(!QucsFilter->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start filter synthesis program!"));
@@ -639,7 +639,7 @@
 void QucsApp::slotCallLine()
 {
   Q3Process *QucsLine =
-    new Q3Process(QString(QucsSettings.BinDir + "qucstrans"));
+    new Q3Process(QString(QucsSettings.AppDir + "qucstrans.app/Contents/MacOS/qucstrans"));
   if(!QucsLine->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start line calculation program!"));
@@ -656,7 +656,7 @@
 void QucsApp::slotCallLibrary()
 {
   Q3Process *QucsLibrary =
-    new Q3Process(QString(QucsSettings.BinDir + "qucslib"));
+    new Q3Process(QString(QucsSettings.AppDir + "qucslib.app/Contents/MacOS/qucslib"));
   if(!QucsLibrary->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start library program!"));
@@ -681,7 +681,7 @@
 void QucsApp::slotCallAtt()
 {
   Q3Process *QucsAtt =
-    new Q3Process(QString(QucsSettings.BinDir + "qucsattenuator"));
+    new Q3Process(QString(QucsSettings.AppDir + "qucsattenuator.app/Contents/MacOS/qucsattenuator"));
   if(!QucsAtt->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start attenuator calculation program!"));
@@ -709,7 +709,7 @@
 void QucsApp::showHTML(const QString& Page)
 {
   QStringList com;
-  com << QucsSettings.BinDir + "qucshelp" << Page;
+  com << QucsSettings.AppDir + "qucshelp.app/Contents/MacOS/qucshelp" << Page;
   Q3Process *QucsHelp = new Q3Process(com);
   QucsHelp->setCommunication(0);
   if(!QucsHelp->start()) {
