--- configure.orig	2013-10-13 14:36:47.000000000 +0200
+++ configure	2013-10-13 14:37:10.000000000 +0200
@@ -791,7 +791,6 @@
 CXXCPP'
 ac_subdirs_all='qucs-core
 qucs-doc
-asco
 examples'
 
 # Initialize some variables set by options.
@@ -4699,8 +4698,6 @@
 
 # AC_CONFIG_SUBDIRS(freehdl)
 # AC_CONFIG_SUBDIRS(verilog)
-subdirs="$subdirs asco"
-
 subdirs="$subdirs examples"
 
 
@@ -4708,7 +4705,6 @@
 RELEASEDIRS="$RELEASEDIRS qucs-doc"
 # RELEASEDIRS="$RELEASEDIRS freehdl"
 # RELEASEDIRS="$RELEASEDIRS verilog"
-RELEASEDIRS="$RELEASEDIRS asco"
 RELEASEDIRS="$RELEASEDIRS examples"
 
 
