--- configure.orig	2014-09-27 21:03:27.000000000 +0200
+++ configure	2014-09-27 21:06:51.000000000 +0200
@@ -6136,7 +6136,7 @@
                         with_qt=no
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Qt OS X framework" >&5
 $as_echo_n "checking for Qt OS X framework... " >&6; }
-    paths="$QTDIR/lib /Library/Frameworks /opt/local/lib /usr/local/lib"
+    paths="$QTDIR/lib ${prefix}/Library/Frameworks /opt/local/lib /usr/local/lib"
     for path in $paths; do
       if test -f "$path/QtGui.framework/Headers/QApplication"; then
         with_qt=yes
@@ -6157,13 +6157,14 @@
       CPPFLAGS="$CPPFLAGS -I$path/QtSvg.framework/Headers"
       CPPFLAGS="$CPPFLAGS -I$path/QtScript.framework/Headers"
       CPPFLAGS="$CPPFLAGS -I$path/Qt3Support.framework/Headers"
+      QT_LDFLAGS="$QT_LDFLAGS -F$path"
     else
       as_fn_error $? "not found" "$LINENO" 5
     fi
 
         QT_DEF="-DQT_SHARED -DQT3_SUPPORT"
     CPPFLAGS="$CPPFLAGS $QT_DEF"
-    QT_LDFLAGS="-headerpad_max_install_names"
+    QT_LDFLAGS="$QT_LDFLAGS -headerpad_max_install_names"
 
     if test $USE_MAINTAINER_MODE = yes; then
       CPPFLAGS="$CPPFLAGS -W -Wall"
@@ -6314,7 +6315,7 @@
     DEVELOPER_DIR="${MACOSX_SDK_PATH%/SDKs*}"
     DEVELOPER_DIR="${DEVELOPER_DIR%/Platforms*}"
     export DEVELOPER_DIR
-    FRAMEWORKSHOME="$MACOSX_SDK_PATH/System/Library/Frameworks"
+    FRAMEWORKSHOME="${prefix}/Library/Frameworks"
     MACOSX_DEPLOYMENT_TARGET="$with_macosx_version_min_required"
 
     case "$with_macosx_version_min_required" in
