--- qucs/qucs_actions.cpp.orig	2009-03-24 23:02:53.000000000 +0100
+++ qucs/qucs_actions.cpp	2009-03-25 00:09:46.000000000 +0100
@@ -609,7 +609,7 @@
 void QucsApp::slotCallFilter()
 {
   QProcess *QucsFilter =
-    new QProcess(QString(QucsSettings.BinDir + "qucsfilter"));
+    new QProcess(QString(QucsSettings.AppDir + "qucsfilter.app/Contents/MacOS/qucsfilter"));
   if(!QucsFilter->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start filter synthesis program!"));
@@ -626,7 +626,7 @@
 void QucsApp::slotCallLine()
 {
   QProcess *QucsLine =
-    new QProcess(QString(QucsSettings.BinDir + "qucstrans"));
+    new QProcess(QString(QucsSettings.AppDir + "qucstrans.app/Contents/MacOS/qucstrans"));
   if(!QucsLine->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start line calculation program!"));
@@ -643,7 +643,7 @@
 void QucsApp::slotCallLibrary()
 {
   QProcess *QucsLibrary =
-    new QProcess(QString(QucsSettings.BinDir + "qucslib"));
+    new QProcess(QString(QucsSettings.AppDir + "qucslib.app/Contents/MacOS/qucslib"));
   if(!QucsLibrary->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start library program!"));
@@ -668,7 +668,7 @@
 void QucsApp::slotCallAtt()
 {
   QProcess *QucsAtt =
-    new QProcess(QString(QucsSettings.BinDir + "qucsattenuator"));
+    new QProcess(QString(QucsSettings.AppDir + "qucsattenuator.app/Contents/MacOS/qucsattenuator"));
   if(!QucsAtt->start()) {
     QMessageBox::critical(this, tr("Error"),
                           tr("Cannot start attenuator calculation program!"));
@@ -696,7 +696,7 @@
 void QucsApp::showHTML(const QString& Page)
 {
   QStringList com;
-  com << QucsSettings.BinDir + "qucshelp" << Page;
+  com << QucsSettings.AppDir + "qucshelp.app/Contents/MacOS/qucshelp" << Page;
   QProcess *QucsHelp = new QProcess(com);
   QucsHelp->setCommunication(0);
   if(!QucsHelp->start()) {
