--- builds/unix/detect.mk.orig	2011-11-30 09:30:49.000000000 -0600
+++ builds/unix/detect.mk	2012-03-09 02:27:36.000000000 -0600
@@ -80,9 +80,9 @@
   ifdef must_configure
     ifneq ($(have_Makefile),)
       # we are building FT2 not in the src tree
-	      $(TOP_DIR)/builds/unix/configure $(value CFG)
+	      $(TOP_DIR)/builds/unix/configure $(CFG)
     else
-	      cd builds/unix; ./configure $(value CFG)
+	      cd builds/unix; ./configure $(CFG)
     endif
   endif
 
