--- ./tests/test_source.py.orig	2022-09-27 17:48:13.000000000 +0200
+++ ./tests/test_source.py	2022-09-27 17:48:39.000000000 +0200
@@ -246,7 +246,7 @@
 
 def black_version():
     try:
-        out = subprocess.check_output(("black", "--version"))
+        out = subprocess.check_output(("black-3.10", "--version"))
     except subprocess.CalledProcessError as e:
         out = e.output
 
@@ -257,10 +257,10 @@
     ]
 
 
-@skip_unless_tool_is_at_least("black", black_version, "22.1.0")
+@skip_unless_tool_is_at_least("black-3.10", black_version, "22.1.0")
 def test_code_is_black_clean():
     output = subprocess.check_output(
-        ("black", "--diff", "."), stderr=subprocess.PIPE
+        ("black-3.10", "--diff", "."), stderr=subprocess.PIPE
     ).decode("utf-8")
 
     # Display diff in "captured stdout call"
