Use MacPorts pv and gnutar.
--- noahstrap.orig	2017-01-16 00:27:30.000000000 -0600
+++ noahstrap	2020-05-01 00:02:49.000000000 -0500
@@ -62,7 +62,7 @@
 
 my ($fh, $archive) = tempfile();
 system "curl -o $archive $suites{$suite}" and die "could not fetch the suite";
-system "pv $archive | gtar xzf - -C $target --strip-components=1" and die "could not extract fs tree from suite";
+system "@PREFIX@/bin/pv $archive | @PREFIX@/bin/gnutar xzf - -C $target --strip-components=1" and die "could not extract fs tree from suite";
 
 print "$suite is successfully installed into $target\n";
 
