--- src/gui/osutils/macutils/AppKitImpl.mm
+++ src/gui/osutils/macutils/AppKitImpl.mm
@@ -139,6 +139,7 @@ - (bool) isDarkMode
 //
 - (bool) isStatusBarDark
 {
+#if __clang_major__ >= 9
     if (@available(macOS 10.17, *)) {
         // This is an ugly hack, but I couldn't find a way to access QTrayIcon's NSStatusItem.
         NSStatusItem* dummy = [[NSStatusBar systemStatusBar] statusItemWithLength:0];
@@ -146,6 +147,7 @@ - (bool) isStatusBarDark
         [[NSStatusBar systemStatusBar] removeStatusItem:dummy];
         return [appearance containsString:@"dark"];
     }
+#endif
 
     return [self isDarkMode];
 }
@@ -176,6 +178,7 @@ - (bool) enableAccessibility
 //
 - (bool) enableScreenRecording
 {
+#if __clang_major__ >= 9
     if (@available(macOS 10.15, *)) {
         // Request screen recording permission on macOS 10.15+
         // This is necessary to get the current window title
@@ -193,6 +196,7 @@ - (bool) enableScreenRecording
             return NO;
         }
     }
+#endif
     return YES;
 }
 
--- src/gui/styles/base/BaseStyle.cpp
+++ src/gui/styles/base/BaseStyle.cpp
@@ -53,8 +53,10 @@
 #include <qdrawutil.h>
 
 #ifdef Q_OS_MACOS
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
 #include <QOperatingSystemVersion>
 #endif
+#endif
 
 #include <cmath>
 
@@ -279,16 +281,22 @@ namespace Phantom
 #ifdef Q_OS_MACOS
             QColor tabBarBase(const QPalette& pal)
             {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 10) && QT_VERSION < QT_VERSION_CHECK(5, 13, 0)                               \
+    || QT_VERSION >= QT_VERSION_CHECK(5, 15, 1)
                 if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                     return hack_isLightPalette(pal) ? QRgb(0xD4D4D4) : QRgb(0x2A2A2A);
                 }
+#endif
                 return hack_isLightPalette(pal) ? QRgb(0xDD1D1D1) : QRgb(0x252525);
             }
             QColor tabBarBaseInactive(const QPalette& pal)
             {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 10) && QT_VERSION < QT_VERSION_CHECK(5, 13, 0)                               \
+    || QT_VERSION >= QT_VERSION_CHECK(5, 15, 1)
                 if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                     return hack_isLightPalette(pal) ? QRgb(0xF5F5F5) : QRgb(0x2D2D2D);
                 }
+#endif
                 return hack_isLightPalette(pal) ? QRgb(0xF4F4F4) : QRgb(0x282828);
             }
 #endif
