From 6cfadfaabafed8d0db213044a0bd22537609381c Mon Sep 17 00:00:00 2001
From: tenzap <fabstz-it@yahoo.fr>
Date: Sat, 13 Nov 2021 15:11:35 +0100
Subject: [PATCH] add support for QuaZip >= 1.0

---
 src/keeshare/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/keeshare/CMakeLists.txt b/src/keeshare/CMakeLists.txt
index af4bc61a8d..8749c2e4ec 100644
--- src/keeshare/CMakeLists.txt
+++ src/keeshare/CMakeLists.txt
@@ -23,7 +23,14 @@ if(WITH_XC_KEESHARE)
 
     # Try to find libquazip5, if found, enable secure sharing
     find_package(QuaZip)
-    if(QUAZIP_FOUND)
+    find_package(QuaZip-Qt5)
+    if(QuaZip-Qt5_FOUND)
+        message(STATUS "Using QuaZip-Qt5 (quazip ${QuaZip-Qt5_VERSION})")
+        set(WITH_XC_KEESHARE_SECURE ON PARENT_SCOPE)
+        target_include_directories(keeshare SYSTEM PRIVATE QuaZip::QuaZip)
+        target_link_libraries(keeshare PRIVATE QuaZip::QuaZip)
+    elseif(NOT QuaZip-Qt5_FOUND AND QUAZIP_FOUND)
+        message(STATUS "Using QuaZip (quazip <1.0)")
         set(WITH_XC_KEESHARE_SECURE ON PARENT_SCOPE)
         target_include_directories(keeshare SYSTEM PRIVATE ${QUAZIP_INCLUDE_DIR})
         target_link_libraries(keeshare PRIVATE ${QUAZIP_LIBRARIES})
Subject: [PATCH] Find quazip.h as installed by version 0.9

quazip 0.9 has quazip.h in /opt/local/include/quazip5/quazip.h
quazip 1.1 has quazip.h in /opt/local/include/QuaZip-Qt5-1.1/quazip/quazip.h

--- cmake/FindQuaZip.cmake
+++ cmake/FindQuaZip.cmake
@@ -7,7 +7,7 @@
     find_path(QUAZIP_INCLUDE_DIRS quazip.h PATH_SUFFIXES quazip5)
 elseif(APPLE)
     find_library(QUAZIP_LIBRARIES quazip1-qt5)
-    find_path(QUAZIP_INCLUDE_DIRS quazip.h PATH_SUFFIXES quazip)
+    find_path(QUAZIP_INCLUDE_DIRS quazip.h PATH_SUFFIXES quazip quazip5)
 else()
     # Try pkgconfig first
     find_package(PkgConfig QUIET)
