--- src/gui/osutils/macutils/AppKitImpl.mm	2020-04-09 18:24:20.000000000 +0200
+++ src/gui/osutils/macutils/AppKitImpl.mm.new	2020-04-23 13:24:04.000000000 +0200
@@ -134,6 +134,7 @@
 //
 - (bool) enableScreenRecording
 {
+#ifdef MAC_OS_X_VERSION_10_15
     if (@available(macOS 10.15, *)) {
         // Request screen recording permission on macOS 10.15+
         // This is necessary to get the current window title
@@ -151,6 +152,7 @@
             return NO;
         }
     }
+#endif
     return YES;
 }
 
