--- src/gui/osutils/macutils/AppKitImpl.mm	2020-04-09 18:24:20.000000000 +0200
+++ src/gui/osutils/macutils/AppKitImpl.mm.new	2020-04-23 13:24:04.000000000 +0200
@@ -134,6 +134,7 @@
 //
 - (bool) enableScreenRecording
 {
+#ifdef MAC_OS_X_VERSION_10_15
     if (@available(macOS 10.15, *)) {
         // Request screen recording permission on macOS 10.15+
         // This is necessary to get the current window title
@@ -151,6 +152,7 @@
             return NO;
         }
     }
+#endif
     return YES;
 }
 
--- src/gui/styles/base/BaseStyle.orig.cpp	2021-11-10 11:10:45.000000000 +0100
+++ src/gui/styles/base/BaseStyle.cpp	2021-11-10 11:07:30.000000000 +0100
@@ -292,16 +292,20 @@
 #ifdef Q_OS_MACOS
             QColor tabBarBase(const QPalette& pal)
             {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
                 if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                     return hack_isLightPalette(pal) ? QRgb(0xD4D4D4) : QRgb(0x2A2A2A);
                 }
+#endif
                 return hack_isLightPalette(pal) ? QRgb(0xDD1D1D1) : QRgb(0x252525);
             }
             QColor tabBarBaseInactive(const QPalette& pal)
             {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
                 if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::MacOSBigSur) {
                     return hack_isLightPalette(pal) ? QRgb(0xF5F5F5) : QRgb(0x2D2D2D);
                 }
+#endif
                 return hack_isLightPalette(pal) ? QRgb(0xF4F4F4) : QRgb(0x282828);
             }
 #endif
--- src/gui/osutils/macutils/AppKitImpl.orig.mm	2021-11-10 11:29:13.000000000 +0100
+++ src/gui/osutils/macutils/AppKitImpl.mm	2021-11-10 11:28:28.000000000 +0100
@@ -139,6 +139,7 @@
 //
 - (bool) isStatusBarDark
 {
+#if __clang_major__ >= 9
     if (@available(macOS 10.17, *)) {
         // This is an ugly hack, but I couldn't find a way to access QTrayIcon's NSStatusItem.
         NSStatusItem* dummy = [[NSStatusBar systemStatusBar] statusItemWithLength:0];
@@ -146,7 +147,7 @@
         [[NSStatusBar systemStatusBar] removeStatusItem:dummy];
         return [appearance containsString:@"dark"];
     }
-
+#endif
     return [self isDarkMode];
 }
 
