--- CMakeLists.txt
+++ CMakeLists.txt
@@ -423,12 +423,6 @@
 
 if(APPLE)
     set(CMAKE_MACOSX_RPATH TRUE)
-    find_program(MACDEPLOYQT_EXE macdeployqt HINTS ${Qt5_PREFIX}/bin ENV PATH)
-    if(NOT MACDEPLOYQT_EXE)
-        message(FATAL_ERROR "macdeployqt is required to build on macOS")
-    else()
-        message(STATUS "Using macdeployqt: ${MACDEPLOYQT_EXE}")
-    endif()
 elseif(MINGW)
     find_program(WINDEPLOYQT_EXE windeployqt HINTS ${Qt5_PREFIX}/bin ENV PATH)
     if(NOT WINDEPLOYQT_EXE)
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -408,11 +408,6 @@
     set(CPACK_PACKAGE_FILE_NAME "${PROGNAME}-${KEEPASSXC_VERSION}")
     include(CPack)
 
-    add_custom_command(TARGET ${PROGNAME}
-            POST_BUILD
-            COMMAND ${MACDEPLOYQT_EXE} ${PROGNAME}.app
-            WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src
-            COMMENT "Deploying app bundle")
 endif()
 
 install(TARGETS ${PROGNAME}
--- src/autotype/mac/CMakeLists.txt
+++ src/autotype/mac/CMakeLists.txt
@@ -7,8 +7,8 @@
 if(WITH_APP_BUNDLE)
     add_custom_command(TARGET keepassx-autotype-cocoa
             POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E make_directory ${PLUGIN_INSTALL_DIR}
             COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/libkeepassx-autotype-cocoa.so ${PLUGIN_INSTALL_DIR}
-            COMMAND ${MACDEPLOYQT_EXE} ${PROGNAME}.app -executable=${PLUGIN_INSTALL_DIR}/libkeepassx-autotype-cocoa.so -no-plugins
             WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/src
             COMMENT "Deploying autotype plugin")
 else()
