--- src/pty.cc.orig	2015-10-29 12:42:57.000000000 -0700
+++ src/pty.cc	2015-10-29 12:43:47.000000000 -0700
@@ -748,7 +748,7 @@
 	/* Set it to blocking. */
         /* FIXMEchpe: why?? vte_terminal_set_pty does the inverse... */
         flags = rv & ~(O_NONBLOCK);
-        rv = fcntl(fd, F_SETFL, flags);
+//        rv = fcntl(fd, F_SETFL, flags);
         if (rv < 0) {
                 int errsv = errno;
                 g_set_error(error, VTE_PTY_ERROR,
