https://github.com/TileDB-Inc/TileDB/pull/4690

diff --git tiledb/common/thread_pool/test/unit_thread_pool.cc tiledb/common/thread_pool/test/unit_thread_pool.cc
index 99131dd90..18dfd31da 100644
--- tiledb/common/thread_pool/test/unit_thread_pool.cc
+++ tiledb/common/thread_pool/test/unit_thread_pool.cc
@@ -38,6 +38,7 @@
 #include <cstdio>
 #include <iostream>
 #include <vector>
+#include <random>
 
 #include "tiledb/common/thread_pool.h"
 #include "tiledb/sm/misc/cancelable_tasks.h"

diff --git test/src/unit-tile-metadata-generator.cc test/src/unit-tile-metadata-generator.cc
index cd3a1b07c..b754f7fda 100644
--- test/src/unit-tile-metadata-generator.cc
+++ test/src/unit-tile-metadata-generator.cc
@@ -30,6 +30,8 @@
  * Tests the TileMetadataGenerator class.
  */
 
+#include <random>
+
 #include <test/support/tdb_catch.h>
 #include "test/support/src/helpers.h"
 #include "tiledb/common/common.h"

diff --git test/src/unit-tile-metadata.cc test/src/unit-tile-metadata.cc
index 89281332f..f478b2d30 100644
--- test/src/unit-tile-metadata.cc
+++ test/src/unit-tile-metadata.cc
@@ -31,6 +31,8 @@
  * load_tile_*_values and get_tile_* apis of fragment metadata.
  */
 
+#include <random>
+
 #include <test/support/tdb_catch.h>
 #include "test/support/src/helpers.h"
 #include "tiledb/sm/c_api/tiledb_struct_def.h"

diff --git tiledb/sm/query/readers/aggregators/test/bench_aggregators.cc tiledb/sm/query/readers/aggregators/test/bench_aggregators.cc
index d88e135f4..f714249d8 100644
--- tiledb/sm/query/readers/aggregators/test/bench_aggregators.cc
+++ tiledb/sm/query/readers/aggregators/test/bench_aggregators.cc
@@ -30,6 +30,8 @@
  * Benchmarks the `AggregateWithCount` class.
  */
 
+#include <random>
+
 #include "tiledb/common/common.h"
 #include "tiledb/sm/query/readers/aggregators/aggregate_buffer.h"
 #include "tiledb/sm/query/readers/aggregators/aggregate_with_count.h"
