--- cmake/Options/BuildOptions.cmake	2024-05-03 00:58:42.000000000 +0800
+++ cmake/Options/BuildOptions.cmake	2024-05-10 03:13:43.000000000 +0800
@@ -8,7 +8,7 @@
 include(CMakeDependentOption)
 
 option(TILEDB_SUPERBUILD "If true, perform a superbuild (builds all missing dependencies)." ON)
-option(TILEDB_VCPKG "If true, use vcpkg to download and build dependencies." ON)
+option(TILEDB_VCPKG "If true, use vcpkg to download and build dependencies." OFF)
 cmake_dependent_option(TILEDB_FORCE_ALL_DEPS "If true, force superbuild to download and build all dependencies, even those installed on the system." OFF "NOT TILEDB_VCPKG" OFF)
 option(TILEDB_SANITIZER "Sets the sanitizers to use. Only address is currently supported." "")
 option(TILEDB_VCPKG_BASE_TRIPLET "Sets the base vcpkg triplet when building with sanitizers." "")
@@ -48,10 +48,6 @@
   endif()
 endif()
 
-if (NOT TILEDB_VCPKG)
-  message(FATAL_ERROR "Disabling TILEDB_VCPKG is not supported.")
-endif()
-
 # enable assertions by default for debug builds
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
   set(TILEDB_ASSERTIONS TRUE)
