--- testgui/Makefile.am.orig	2016-12-19 10:55:01.000000000 +0100
+++ testgui/Makefile.am	2016-12-19 10:57:40.000000000 +0100
@@ -20,12 +20,6 @@
 
 if OS_DARWIN
 hidapi_testgui_SOURCES = test.cpp mac_support_cocoa.m mac_support.h
-# Rules for copying the binary and its dependencies into the app bundle.
-TestGUI.app/Contents/MacOS/hidapi-testgui$(EXEEXT): hidapi-testgui$(EXEEXT)
-	$(srcdir)/copy_to_bundle.sh
-
-all: all-am TestGUI.app/Contents/MacOS/hidapi-testgui$(EXEEXT)
-
 endif
 
 EXTRA_DIST = \
