--- source/edu/stanford/ejalbert/testing/BrowserLauncherTestApp.java.orig	2011-12-06 20:42:24.000000000 -0500
+++ source/edu/stanford/ejalbert/testing/BrowserLauncherTestApp.java	2011-12-06 20:42:50.000000000 -0500
@@ -316,10 +316,6 @@
         if (logger.isInfoEnabled()) {
             logger.info("kill browser button clicked");
         }
-        BrowserEvent[] browserEvents = browserCloser.getBrowserEvents();
-        if(browserEvents.length > 0) {
-            browserCloser.closeBrowser(browserEvents[0]);
-        }
     }
 
     private void copyButton_actionPerformed(ActionEvent e) {
