commit 0a721c293a59842368b41e16469e9999859f0329
Author: Adrian Thurston <thurston@complang.org>
Date:   Sun Aug 11 10:41:16 2013 -0400

    added missing std:: using
    
    Patch from Hesham Wahba

diff --git a/ragel/javacodegen.cpp b/ragel/javacodegen.cpp
index adff67e..ff2193c 100644
--- a/ragel/javacodegen.cpp
+++ b/ragel/javacodegen.cpp
@@ -54,6 +54,7 @@ using std::cin;
 using std::cout;
 using std::cerr;
 using std::endl;
+using std::setiosflags;
 
 void javaLineDirective( ostream &out, const char *fileName, int line )
 {
