--- src/pty.c.orig	2016-01-04 04:47:21.000000000 -0800
+++ src/pty.c	2016-01-04 04:47:55.000000000 -0800
@@ -843,7 +843,7 @@
 	/* Set it to blocking. */
         /* FIXMEchpe: why?? vte_terminal_set_pty does the inverse... */
         flags = rv & ~(O_NONBLOCK);
-        rv = fcntl(fd, F_SETFL, flags);
+//        rv = fcntl(fd, F_SETFL, flags);
         if (rv < 0) {
                 int errsv = errno;
                 g_set_error(error, VTE_PTY_ERROR,
