--- Makefile.PL.orig	2017-06-21 08:41:31.000000000 -0700
+++ Makefile.PL	2017-06-21 08:42:43.000000000 -0700
@@ -4,7 +4,7 @@
 use strict;
 use warnings;
 use ExtUtils::MakeMaker;
-require 'common.pl';
+require './common.pl';
 
 my $version = ag_getversion();
 
--- lib/Alien/Gimp.pm.PL	2017-06-21 20:36:20.000000000 -0700
+++ lib/Alien/Gimp.pm.PL.orig	2017-06-21 20:35:58.000000000 -0700
@@ -2,7 +2,7 @@
 
 use strict;
 use warnings;
-require 'common.pl';
+require './common.pl';
 
 my $version = ag_getversion();
 my $config_hash = ag_getconfig();
--- lib/Alien/Gimp/Install/Files.pm.PL.orig	2017-06-21 20:37:48.000000000 -0700
+++ lib/Alien/Gimp/Install/Files.pm.PL	2017-06-21 20:38:01.000000000 -0700
@@ -3,7 +3,7 @@
 use strict;
 use warnings;
 
-require 'common.pl';
+require './common.pl';
 my $build_hash = ag_getbuild();
 
 my $pkg = ExtUtils::Depends->new('Alien::Gimp');
