--- common.pl.orig	2016-05-01 20:41:48.000000000 -0700
+++ common.pl	2018-12-30 18:52:34.000000000 -0800
@@ -10,8 +10,9 @@
 my ($plugindir, $pluginlibs) = split /\n/, `$gimptool --gimpplugindir --libs`;
 
 my $gimpbinname = ExtUtils::PkgConfig->modversion("gimp-2.0");
-$gimpbinname =~ s/^(\d\.\d).*/$1/; # strip off minor versions
-die "Need GIMP version at least 2.8.0\n" unless $gimpbinname >= 2.8;
+$gimpbinname =~ s/^(\d\.\d+).*/$1/; # strip off minor versions
+my ($major, $minor) = split /\./,$gimpbinname;
+die "Need GIMP version at least 2.8.0\n" unless $major == 2 && $minor >= 8;
 
 sub ag_getconfig {
   my %cfg = (
