--- chicken.h.orig	2013-01-30 00:22:08.000000000 -0800
+++ chicken.h	2013-01-30 00:22:10.000000000 -0800
@@ -226,11 +226,7 @@ void *alloca ();
 # ifndef __cplusplus
 #  define C_cblock                ({
 #  define C_cblockend             })
-#  ifdef __clang__
-#   define C_noret
-#  else
-#   define C_noret                __attribute__ ((noreturn))
-#  endif
+#  define C_noret                __attribute__ ((noreturn))
 #  define C_noret_decl(name)
 #  define C_aligned               __attribute__ ((aligned))
 # endif
