--- Makefile.macosx.orig	2023-08-20 15:07:14.000000000 +1000
+++ Makefile.macosx	2023-08-20 15:07:18.000000000 +1000
@@ -28,14 +28,12 @@
 
 # platform configuration
 
-XCODE_DEVELOPER ?= /Applications/Xcode.app/Contents/Developer
-XCODE_TOOL_PATH ?= $(XCODE_DEVELOPER)/Toolchains/XcodeDefault.xctoolchain/usr/bin
-C_COMPILER ?= $(XCODE_DEVELOPER)/usr/bin/gcc
+C_COMPILER ?= /usr/bin/gcc
 ARCH ?= $(shell sh $(SRCDIR)/config-arch.sh)
 
 # commands
 
-POSTINSTALL_PROGRAM = $(XCODE_TOOL_PATH)/install_name_tool
+POSTINSTALL_PROGRAM = /usr/bin/install_name_tool
 
 # options
 
@@ -49,7 +47,7 @@
 C_COMPILER_OPTIMIZATION_OPTIONS ?= -Os -fomit-frame-pointer
 endif
 endif
-LIBRARIAN ?= $(XCODE_TOOL_PATH)/ar
+LIBRARIAN ?= /usr/bin/ar
 LINKER_LINK_SHARED_LIBRARY_OPTIONS = -dynamiclib -compatibility_version 1 -current_version 1.0 -install_name $@
 POSTINSTALL_PROGRAM_FLAGS = -change lib$(PROGRAM_PREFIX)chicken$(PROGRAM_SUFFIX)$(SO) $(LIBDIR)/lib$(PROGRAM_PREFIX)chicken$(PROGRAM_SUFFIX)$(SO)
 LIBRARIAN_OPTIONS = scru
@@ -66,18 +64,12 @@
 
 # architectures
 
+C_COMPILER_OPTIONS += $(CFLAGS)
+LINKER_OPTIONS += $(LDFLAGS)
+
 ifeq ($(ARCH),x86-64)
-C_COMPILER_OPTIONS += -m64
-LINKER_OPTIONS += -m64
 # Avoid bus error in install_name_tool
 LINKER_LINK_SHARED_DLOADABLE_OPTIONS += -Wl,-headerpad -Wl,128
-else
-
-ifeq ($(ARCH),universal)
-C_COMPILER_OPTIONS += -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk
-LINKER_OPTIONS += -arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk
-endif
-
 endif
 
 # select default and internal settings
