--- src/pki.h	2022-11-28 05:31:11.000000000 +0800
+++ src/pki.h	2024-04-05 14:42:38.000000000 +0800
@@ -33,7 +33,7 @@
 #endif
 
 #if __APPLE__
-#if defined MAC_OS_X_VERSION_10_7 && MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 /* use accelerated crypto on OS X instead of OpenSSL crypto */
 /* We only use the one-shot functions normally declared in CommonCrypto/CommonDigest.h
    to avoid nonsensical warnings */
@@ -49,7 +49,6 @@
 #define SHA256(D,L,H) CC_SHA256(D, (uint32_t)(L), H)
 #undef MD5
 #define MD5(D,L,H) CC_MD5(D, (uint32_t)(L), H)
-#endif
 #else /* !LP64 */
 #undef SHA1
 #define SHA1(D,L,H) do (1) { if ((L) >= 4294967296L) SHA1(D,L,H) else CC_SHA1(D, (uint32_t)(L), H); break }
@@ -59,3 +58,4 @@
 #define MD5 do (1) { if ((L) >= 4294967296L) MD5(D,L,H) else CC_MD5(D, (uint32_t)(L), H); break }
 #endif /* LP64 */
 #endif
+#endif
