--- libs/CMakeLists.txt.orig	2014-04-28 06:38:04.000000000 +0200
+++ libs/CMakeLists.txt	2014-06-21 12:10:54.000000000 +0200
@@ -7,7 +7,9 @@
    add_subdirectory(kdm)
    add_subdirectory(plasmagenericshell)
    if(NOT WIN32)
+    if (NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
         add_subdirectory(taskmanager)
+    endif (NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
    endif(NOT WIN32)
 endif(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
 
