--- kstyles/orig.CMakeLists.txt	2014-10-03 18:30:13.000000000 +0200
+++ kstyles/CMakeLists.txt	2014-10-08 11:57:14.000000000 +0200
@@ -1,2 +1,2 @@
-add_subdirectory( oxygen )
+# add_subdirectory( oxygen )
 add_subdirectory( themes )
--- libs/oxy.CMakeLists.txt	2014-12-30 13:32:17.000000000 +0100
+++ libs/CMakeLists.txt	2014-12-30 14:25:25.000000000 +0100
@@ -1,6 +1,6 @@
 add_subdirectory(kephal)
 add_subdirectory(kworkspace)
-add_subdirectory(oxygen)
+#add_subdirectory(oxygen)
 add_subdirectory(plasmaclock)
 
 if(${KDE_PLATFORM_PROFILE} STREQUAL "Desktop")
