diff --git systemsettings/app/SettingsBase.cpp systemsettings/app/SettingsBase.cpp
index 74ccd4f..c08fce7 100644
--- systemsettings/app/SettingsBase.cpp
+++ systemsettings/app/SettingsBase.cpp
@@ -159,7 +159,9 @@ void SettingsBase::initHelpMenu()
     // Add the custom actions
     aboutModuleAction = actionCollection()->addAction( KStandardAction::AboutApp, "help_about_module", this, SLOT(about()) );
     changeAboutMenu( 0, aboutModuleAction, i18n("About Active Module") );
+    aboutModuleAction->setMenuRole(QAction::NoRole);
     aboutViewAction = actionCollection()->addAction( KStandardAction::AboutApp, "help_about_view", this, SLOT(about()) );
+    aboutViewAction->setMenuRole(QAction::NoRole);
 }
 
 void SettingsBase::initConfig()
