--- setup.py.orig	2025-12-08 07:43:58
+++ setup.py	2025-12-08 12:21:23
@@ -186,9 +186,6 @@
         "console_scripts": [
             # Install a script as "coverage".
             "coverage = coverage.cmdline:main",
-            # And as "coverage3", and as "coverage-3.7" (or whatever), but deprecated.
-            "coverage%d = coverage.cmdline:main_deprecated" % sys.version_info[:1],
-            "coverage-%d.%d = coverage.cmdline:main_deprecated" % sys.version_info[:2],
         ],
     },
     extras_require={
