--- setup.py.orig	2021-10-31 12:29:26.000000000 -0400
+++ setup.py	2021-10-31 12:29:43.000000000 -0400
@@ -100,8 +100,6 @@
         # "coverage-3.7" (or whatever).
         'console_scripts': [
             'coverage = coverage.cmdline:main',
-            'coverage%d = coverage.cmdline:main' % sys.version_info[:1],
-            'coverage-%d.%d = coverage.cmdline:main' % sys.version_info[:2],
         ],
     },
 
