--- setup.orig.py
+++ setup.py
@@ -71,13 +71,9 @@
     devstat = "5 - Production/Stable"
 classifier_list.append("Development Status :: " + devstat)
 
-# Install a script as "coverage", and as "coverage[23]", and as
-# "coverage-2.7" (or whatever).
-scripts = [
-    'coverage = coverage:main',
-    'coverage%d = coverage:main' % sys.version_info[:1],
-    'coverage-%d.%d = coverage:main' % sys.version_info[:2],
-    ]
+# 2014-03-28, petr -- changed
+# Install a script as "coverage" only, versions will be handeled by MacPorts
+scripts = ['coverage = coverage:main']
 
 # Create the keyword arguments for setup()
 
