--- src/SDL.xs.orig	2009-11-07 14:47:35.000000000 -0600
+++ src/SDL.xs	2014-12-13 19:54:40.000000000 -0600
@@ -119,6 +119,29 @@
 
 #endif
 
+#ifdef MACOSX
+#include <CoreServices/CoreServices.h>
+
+void CPSEnableForegroundOperation(ProcessSerialNumber* psn);
+void NSApplicationLoad();
+void SDL_macosx_init(void) {
+    Boolean sameProc;
+    ProcessSerialNumber myProc, frProc;
+    if (GetFrontProcess(&frProc) == noErr)
+        if (GetCurrentProcess(&myProc) == noErr)
+            if (SameProcess(&frProc, &myProc, &sameProc) == noErr && sameProc == 0) {
+                /*
+                NSLog(@"creating bad autorelease pool");
+                [[NSAutoreleasePool alloc] init];
+                */
+                NSApplicationLoad();
+                CPSEnableForegroundOperation(&myProc);
+            }
+}
+void SDL_macosx_quit(void) {
+}
+#endif // MACOSX
+
 Uint32 
 sdl_perl_timer_callback ( Uint32 interval, void* param )
 {
