--- DESCRIPTION	2024-01-09 19:20:02.000000000 +0800
+++ DESCRIPTION	2024-01-10 09:52:11.000000000 +0800
@@ -12,7 +12,7 @@
 URL: https://andrewzm.github.io/FRK/, https://github.com/andrewzm/FRK/
 BugReports: https://github.com/andrewzm/FRK/issues/
 Depends: R (>= 3.5.0)
-Suggests: covr, dggrids, gstat, INLA, knitr, lme4, mapproj, parallel,
+Suggests: covr, dggrids, gstat, knitr, lme4, mapproj, parallel,
         sf, spdep, splancs, testthat, verification
 Imports: digest, dplyr, ggplot2, grDevices, Hmisc (>= 4.1), Matrix,
         methods, plyr, Rcpp (>= 0.12.12), sp, spacetime, sparseinv,
