--- src/gui/qgsmapcanvas.h.orig	2013-09-05 10:51:53.000000000 +0200
+++ src/gui/qgsmapcanvas.h	2013-09-05 10:53:04.000000000 +0200
@@ -408,6 +408,12 @@
     /// Handle pattern for implementation object
     std::auto_ptr<CanvasProperties> mCanvasProperties;
 
+    /**debugging member
+       invoked when a connect() is made to this object
+    */
+    void connectNotify( const char * signal );
+
+
   private slots:
     void crsTransformEnabled( bool );
 
@@ -455,11 +461,6 @@
        the last entry in case a lot of resize events arrive in short time*/
     QList< QPair<int, int> > mResizeQueue;
 
-    /**debugging member
-       invoked when a connect() is made to this object
-    */
-    void connectNotify( const char * signal );
-
     //! current layer in legend
     QgsMapLayer* mCurrentLayer;
 
