--- cmake/Python.cmake.orig	2010-05-13 21:47:51.000000000 +0200
+++ cmake/Python.cmake	2010-05-13 21:48:13.000000000 +0200
@@ -148,7 +148,7 @@
     IF (HAVE_SIP_MODULE AND SIP_BINARY_PATH AND SIP_INCLUDE_DIR)
       # check for SIP version
       # minimal version is 4.7 (to support universal builds)
-      SET (SIP_MIN_VERSION 040700)
+      SET (SIP_MIN_VERSION 040A00)
       TRY_RUN_PYTHON (RES "import sip\nprint '%x' % sip.SIP_VERSION" SIP_VERSION)
       IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
         SET (SIP_IS_GOOD TRUE)
