--- sipcorepart3.cpp.orig	2012-03-30 18:56:28.000000000 +0200
+++ sipcorepart3.cpp	2012-03-30 18:56:42.000000000 +0200
@@ -2901,7 +2901,7 @@
     {
         QDomNode * a0;
         QDomDocument * a1;
-        const QgsMapLayer *sipCpp;
+        QgsMapLayer *sipCpp;
 
         if (sipParseArgs(&sipParseErr, sipArgs, "BJ9J9", &sipSelf, sipType_QgsMapLayer, &sipCpp, sipType_QDomNode, &a0, sipType_QDomDocument, &a1))
         {
