--- rak/allocators.h.orig	2011-04-05 20:25:36.000000000 +1000
+++ rak/allocators.h	2011-10-12 11:24:21.000000000 +1100
@@ -77,8 +77,11 @@ public:
 
   static pointer alloc_size(size_type size) {
     pointer ptr = NULL;
+#ifdef HAVE_POSIX_MEMALIGN
     int __UNUSED result = posix_memalign((void**)&ptr, LT_SMP_CACHE_BYTES, size);
-
+#else
+    ptr = (pointer)valloc(size);
+#endif
     return ptr;
   }
 
