--- t-4.1.1.gemspec
+++ t-4.1.1.gemspec
@@ -24,11 +24,11 @@
 
   s.specification_version = 4
 
-  s.add_runtime_dependency(%q<geokit>.freeze, ["~> 1.14".freeze])
-  s.add_runtime_dependency(%q<htmlentities>.freeze, ["~> 4.3".freeze])
-  s.add_runtime_dependency(%q<launchy>.freeze, ["~> 3.0".freeze])
-  s.add_runtime_dependency(%q<oauth>.freeze, ["~> 1.1".freeze])
-  s.add_runtime_dependency(%q<retryable>.freeze, ["~> 3.0".freeze])
-  s.add_runtime_dependency(%q<thor>.freeze, ["~> 1.3".freeze])
-  s.add_runtime_dependency(%q<twitter>.freeze, ["~> 8.1".freeze])
+  s.add_runtime_dependency(%q<geokit>.freeze, [">= 1.14".freeze])
+  s.add_runtime_dependency(%q<htmlentities>.freeze, [">= 4.3".freeze])
+  s.add_runtime_dependency(%q<launchy>.freeze, [">= 3.0".freeze])
+  s.add_runtime_dependency(%q<oauth>.freeze, [">= 1.1".freeze])
+  s.add_runtime_dependency(%q<retryable>.freeze, [">= 3.0".freeze])
+  s.add_runtime_dependency(%q<thor>.freeze, [">= 1.3".freeze])
+  s.add_runtime_dependency(%q<twitter>.freeze, [">= 8.1".freeze])
 end
