diff -ur ../sowing-1.1.16c-orig/Makefile.in ./Makefile.in
--- ../sowing-1.1.16c-orig/Makefile.in	2011-10-27 14:54:31.000000000 -0500
+++ ./Makefile.in	2011-12-15 21:11:45.000000000 -0600
@@ -1,6 +1,6 @@
 ALL: all-redirect
 SUBDIRS = src
-prefix      = @prefix@
+prefix      = $(DESTDIR)@prefix@
 exec_prefix = @exec_prefix@
 bindir      = @bindir@
 datarootdir = @datarootdir@
diff -ur ../sowing-1.1.16c-orig/Makerules.in ./Makerules.in
--- ../sowing-1.1.16c-orig/Makerules.in	2011-10-27 14:54:31.000000000 -0500
+++ ./Makerules.in	2011-12-15 21:12:17.000000000 -0600
@@ -1,6 +1,6 @@
 # Basic makefile rules that all Makes share
 
-prefix=@prefix@
+prefix=$(DESTDIR)@prefix@
 exec_prefix=@exec_prefix@
 datarootdir=@datarootdir@
 libdir=@libdir@
Only in .: config.log
Only in .: config.system
diff -ur ../sowing-1.1.16c-orig/docs/Makefile.in ./docs/Makefile.in
--- ../sowing-1.1.16c-orig/docs/Makefile.in	2011-10-27 14:54:31.000000000 -0500
+++ ./docs/Makefile.in	2011-12-15 21:13:00.000000000 -0600
@@ -1,7 +1,7 @@
 ALL: install.ps tohtml.ps bfort.ps doctext.ps install.pdf tohtml.pdf doctext.pdf
 SHELL = @SHELL@
 TOHTML = ../src/tohtml/tohtml
-prefix = @prefix@
+prefix = $(DESTDIR)@prefix@
 wwwdir = @wwwdir@
 
 doc_POSTSCRIPT = install.ps tohtml.ps bfort.ps doctext.ps
diff -ur ../sowing-1.1.16c-orig/docs/doctext/Makefile.in ./docs/doctext/Makefile.in
--- ../sowing-1.1.16c-orig/docs/doctext/Makefile.in	2011-10-27 14:54:31.000000000 -0500
+++ ./docs/doctext/Makefile.in	2011-12-15 21:13:18.000000000 -0600
@@ -8,7 +8,7 @@
 #TEXPATH=.:/home/gropp/texhome/share/texmf/tex/latex209
 #TOHTML = tohtml
 TOHTML = ../../src/tohtml/tohtml
-prefix = @prefix@
+prefix = $(DESTDIR)@prefix@
 wwwdir = @wwwdir@
 
 # Need to add installation steps.
