Fix installation directories to conform to porthier(7).
--- installer/install.cfg.orig	2018-03-31 01:05:50.000000000 -0500
+++ installer/install.cfg	2021-05-09 10:27:22.000000000 -0500
@@ -30,13 +30,13 @@
 TWPOLICY="${sysconfdir}"
 
 # Tripwire manual pages are stored in TWMAN.
-TWMAN="${prefix}/man"
+TWMAN="${prefix}/share/man"
 
 # Tripwire database files are stored in TWDB.
-TWDB="${prefix}/lib/tripwire"
+TWDB="${prefix}/var/db/tripwire"
 
 # Tripwire documents directory
-TWDOCS="${prefix}/doc/tripwire"
+TWDOCS="${prefix}/share/doc/tripwire"
 
 # The Tripwire site key files are stored in TWSITEKEYDIR.
 TWSITEKEYDIR="${TWPOLICY}"
