--- configure.ac.orig	2011-11-21 10:06:56.000000000 -0600
+++ configure.ac	2013-09-02 18:43:41.000000000 -0500
@@ -6,7 +6,7 @@
 AC_CONFIG_SRCDIR([src/tw/tw.cpp])
 AC_CANONICAL_TARGET([])
 AM_INIT_AUTOMAKE(tripwire, 2.4.2.2)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS([config.h])
 
 dnl #################################
 dnl Cleanup Cruft Leftover From Patch
@@ -38,6 +38,7 @@
 dnl ###################
 dnl Checks for programs
 dnl ###################
+AC_PROG_MKDIR_P
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_RANLIB
@@ -406,6 +407,7 @@
 dnl Export the variables we use
 dnl #####################################
 AC_SUBST(CORE_CRYPT_O)
+AC_SUBST([am__isrc], [' -I$(srcdir)'])
 
 AC_CONFIG_FILES([
 Makefile
