From 981ceacf878ecf0c20f29f7270b150e489235377 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 1 Dec 2024 09:18:34 +0800
Subject: [PATCH] mbtiles.cpp: isinf to std::isinf

---
 mbtiles.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git mbtiles.cpp mbtiles.cpp
index a489e7d..c7a615b 100644
--- mbtiles.cpp
+++ mbtiles.cpp
@@ -700,7 +700,7 @@ metadata make_metadata(const char *fname, int minzoom, int maxzoom, double minla
 
 	m.strategies_json = stringify_strategies(strategies);
 
-	if (isinf(droprate)) {
+	if (std::isinf(droprate)) {
 		droprate = LLONG_MAX;
 	}
 	if (basezoom != maxzoom || droprate != 2.5 || retain_points_multiplier != 1) {
