--- Makefile.in.orig	2008-06-02 12:14:00.000000000 +0200
+++ Makefile.in	2008-06-02 12:14:18.000000000 +0200
@@ -268,7 +268,7 @@
 # Install.
 
 install:
-	ocamlfind install bitstring META *.mli *.cmx *.cma *.cmxa *.a *.so \
+	ocamlfind install -ldconf ignore -destdir $(DESTDIR)@prefix@@libdir@/ocaml/site-lib bitstring META *.mli *.cmx *.cma *.cmxa *.a *.so \
 		bitstring.cmi \
 		bitstring_persistent.cmi \
 		pa_bitstring.cmo
