--- include/mach/i386/_structs.h.orig	2020-02-09 10:59:11.000000000 -0800
+++ include/mach/i386/_structs.h	2020-02-09 10:59:34.000000000 -0800
@@ -627,7 +627,6 @@
 	_STRUCT_XMM_REG		__fpu_ymmh15;		/* YMMH 15  */
 };
 
-#endif /* not __OPEN_SOURCE__ */
 
 #else /* !__DARWIN_UNIX03 */
 #define	_STRUCT_X86_FLOAT_STATE64	struct x86_float_state64
@@ -800,3 +799,4 @@
 };
 #endif /* !__DARWIN_UNIX03 */
 
+#endif /* _MACH_I386__STRUCTS_H_ */
