diff --git libstuff/Makefile libstuff/Makefile
index 975a5f6..da5a411 100644
--- libstuff/Makefile
+++ libstuff/Makefile
@@ -39,7 +39,7 @@ ifeq "$(EFITOOLS)" "efitools"
 else
 	COFF_BYTESEX =
 endif
-CFILES = allocate.c errors.c arch.c execute.c bytesex.c rnd.c hppa.c \
+CFILES = args.c allocate.c errors.c arch.c execute.c bytesex.c rnd.c hppa.c \
 	  swap_headers.c reloc.c version_number.c get_toc_byte_sex.c \
 	  best_arch.c fatals.c arch_usage.c ofile.c ofile_error.c print.c \
 	  set_arch_flag_name.c seg_addr_table.c dylib_table.c write64.c \
