--- mythtv/configure.orig	2013-03-08 09:53:10.000000000 -0500
+++ mythtv/configure	2013-03-08 09:53:42.000000000 -0500
@@ -2554,7 +2554,7 @@
 ranlib="${cross_prefix}${ranlib}"
 strip_default="${cross_prefix}${strip_default}"
 
-sysinclude_default="${sysroot}/usr/include"
+#sysinclude_default="${sysroot}/usr/include"
 
 set_default cc cxx nm pkg_config strip sysinclude qmake python yasmexe
 enabled cross_compile || host_cc_default=$cc
@@ -3609,7 +3609,7 @@
 
 set_default $PATHS_LIST
 
-append CONFIG_INCLUDEPATH $sysinclude "${sysroot}${prefix}"
+#append CONFIG_INCLUDEPATH $sysinclude "${sysroot}${prefix}"
 
 append CCONFIG "$compile_type"
 if test x$compile_type = x"profile" ; then
