From 8929314e77dfc5fe9bcc58f4ace4e7ba6fea9bef Mon Sep 17 00:00:00 2001
From: George Nassas <george@nassas.com>
Date: Tue, 29 May 2012 10:11:38 -0400
Subject: [PATCH] make backend's bonjour service name dns-compliant

---
 mythtv/programs/mythbackend/mediaserver.cpp |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/mythtv/programs/mythbackend/mediaserver.cpp b/mythtv/programs/mythbackend/mediaserver.cpp
index 043140d..19d0eb9 100644
--- mythtv/programs/mythbackend/mediaserver.cpp
+++ mythtv/programs/mythbackend/mediaserver.cpp
@@ -256,13 +256,10 @@ void MediaServer::Init(bool bIsMaster, bool bDisableUPnp /* = false */)
         m_bonjour = new BonjourRegister();
         if (m_bonjour)
         {
-            QByteArray dummy;
             QByteArray name("Mythbackend on ");
             name.append(gCoreContext->GetHostName());
-            m_bonjour->Register(nPort,
-                                bIsMaster ? "_mythbackend-master._tcp" :
-                                            "_mythbackend-slave._tcp",
-                                name, dummy);
+            QByteArray txt(bIsMaster ? "\x0Clevel=master" : "\x0Blevel=slave");
+            m_bonjour->Register(nPort, "_mythbackend._tcp", name, txt);
         }
 #endif
     }
-- 
1.7.2.5

