--- numpy/f2py/setup.py	2011-02-27 23:40:29.000000000 -0600
+++ numpy/f2py/setup.py	2011-05-15 09:18:40.000000000 -0500
@@ -51,7 +51,7 @@
             log.info('Creating %s', target)
             f = open(target,'w')
             f.write('''\
-#!/usr/bin/env %s
+#!@@MPORTS_PYTHON@@
 # See http://cens.ioc.ee/projects/f2py2e/
 import os, sys
 for mode in ["g3-numpy", "2e-numeric", "2e-numarray", "2e-numpy"]:
@@ -75,7 +75,7 @@
     sys.stderr.write("Unknown mode: " + repr(mode) + "\\n")
     sys.exit(1)
 main()
-'''%(os.path.basename(sys.executable)))
+''')
             f.close()
         return target
 
