--- libbitdht.pro.orig	2012-03-02 20:28:30.000000000 -0500
+++ libbitdht.pro	2012-03-02 20:28:48.000000000 -0500
@@ -1,5 +1,5 @@
 TEMPLATE = lib
-CONFIG += staticlib release
+CONFIG += release
 CONFIG -= qt
 TARGET = bitdht
 QMAKE_CXXFLAGS *= -Wall -DBE_DEBUG
@@ -63,7 +63,7 @@
 ################################# MacOSX ##########################################
 
 mac {
-		QMAKE_CC = g++
+		QMAKE_CC = @@CC@@
 		OBJECTS_DIR = temp/obj
 		MOC_DIR = temp/moc
 		DESTDIR = lib
