--- configure.ac.orig	2012-03-23 13:08:56.000000000 -0700
+++ configure.ac	2012-03-23 13:09:45.000000000 -0700
@@ -28,6 +28,8 @@
 		-Wmissing-prototypes -Wnested-externs -Wpointer-arith"
 fi
 
+PYTHON_PREFIX=`$PYTHON -c 'import sys; print sys.prefix;'`
+PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print sys.exec_prefix;'`
 
 AM_PATH_PYTHON()
 AM_CHECK_PYTHON_HEADERS(,AC_MSG_ERROR([install python-devel]))
