--- configure.ac.orig	2015-04-29 00:09:24.000000000 -0700
+++ configure.ac	2015-04-29 00:09:59.000000000 -0700
@@ -438,11 +438,11 @@
 		win32/installer/Makefile
 		win32/installer/pixmaps/Makefile
 		win32/installer/translations/Makefile])
-AM_COND_IF([VIKING_DOC],
-	[AC_CONFIG_FILES([
-		doc/Makefile
-		doc/reference/Makefile
-		doc/examples/Makefile])])
+#AM_COND_IF([VIKING_DOC],
+#	[AC_CONFIG_FILES([
+#		doc/Makefile
+#		doc/reference/Makefile
+#		doc/examples/Makefile])])
 
 AC_OUTPUT([
 	   viking.spec
