--- ./Makefile	2013-10-17 04:52:00.000000000 -0400
+++ ./Makefile	2023-06-24 15:03:49.000000000 -0400
@@ -9,7 +9,7 @@
 #								opposite of a disclaimer.
 ALTERMIME_OPTIONS=-DALTERMIME_PRETEXT
 #ALTERMIME_OPTIONS=
-CFLAGS=-Wall -Werror -g -I. -O2 $(ALTERMIME_OPTIONS)
+CFLAGS+=-Wall -g -I. $(ALTERMIME_OPTIONS)
 OBJS= strstack.o mime_alter.o ffget.o pldstr.o filename-filters.o logger.o MIME_headers.o libmime-decoders.o boundary-stack.o qpe.o
 
 
