--- CMakeLists.txt.orig	2023-02-14 00:10:27.000000000 +0800
+++ CMakeLists.txt	2023-04-20 11:48:16.000000000 +0800
@@ -44,6 +44,3 @@
   target_link_libraries(test_${execid} functional testing)
   add_test(test_${execid} bin/test_${execid})
 endforeach()
-
-install(DIRECTORY "${CMAKE_Fortran_MODULE_DIRECTORY}" DESTINATION "${CMAKE_INSTALL_PREFIX}")
-install(DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}" DESTINATION "${CMAKE_INSTALL_PREFIX}")
