--- waftools/podselect.py.orig	2023-05-12 12:19:00
+++ waftools/podselect.py	2024-05-17 16:57:06
@@ -31,7 +31,10 @@
 		self.bld.install_files(os.path.join(base_path, os.path.dirname(path)), outnode)
 
 class perldoc(Task.Task):
-	run_str='podselect ${SRC} > ${TGT}'
+	run_str='${PODSELECT} ${SRC} > ${TGT}'
 	color='BLUE'
 	ext_in=['.xs']
 	ext_out=['.pod']
+
+def configure(ctx):
+    ctx.find_program('podselect', var='PODSELECT')
