diff -rwub apt-inst/contrib/arfile.cc apt-inst/contrib/arfile.cc
--- apt-inst/contrib/arfile.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/contrib/arfile.cc	Sun Apr 18 03:37:50 2004
@@ -14,7 +14,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/arfile.h"
 #endif
 #include <apt-pkg/arfile.h>
diff -rwub apt-inst/contrib/arfile.h apt-inst/contrib/arfile.h
--- apt-inst/contrib/arfile.h	Tue Feb 20 08:03:16 2001
+++ apt-inst/contrib/arfile.h	Sun Apr 18 03:37:50 2004
@@ -15,7 +15,7 @@
 #ifndef PKGLIB_ARFILE_H
 #define PKGLIB_ARFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/arfile.h"
 #endif
 
diff -rwub apt-inst/contrib/extracttar.cc apt-inst/contrib/extracttar.cc
--- apt-inst/contrib/extracttar.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/contrib/extracttar.cc	Sun Apr 18 03:37:50 2004
@@ -16,7 +16,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/extracttar.h"
 #endif
 #include <apt-pkg/extracttar.h>
diff -rwub apt-inst/contrib/extracttar.h apt-inst/contrib/extracttar.h
--- apt-inst/contrib/extracttar.h	Tue Feb 20 08:03:17 2001
+++ apt-inst/contrib/extracttar.h	Sun Apr 18 03:37:50 2004
@@ -14,7 +14,7 @@
 #ifndef PKGLIB_EXTRACTTAR_H
 #define PKGLIB_EXTRACTTAR_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/extracttar.h"
 #endif
 
diff -rwub apt-inst/database.cc apt-inst/database.cc
--- apt-inst/database.cc	Tue Feb 20 08:03:16 2001
+++ apt-inst/database.cc	Sun Apr 18 03:37:50 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/database.h"
 #endif
 
diff -rwub apt-inst/database.h apt-inst/database.h
--- apt-inst/database.h	Tue Feb 20 08:03:16 2001
+++ apt-inst/database.h	Sun Apr 18 03:37:50 2004
@@ -21,7 +21,7 @@
 #ifndef PKGLIB_DATABASE_H
 #define PKGLIB_DATABASE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/database.h"
 #endif
 
diff -rwub apt-inst/deb/debfile.cc apt-inst/deb/debfile.cc
--- apt-inst/deb/debfile.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/deb/debfile.cc	Sun Apr 18 03:37:50 2004
@@ -16,7 +16,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debfile.h"
 #endif
 
diff -rwub apt-inst/deb/debfile.h apt-inst/deb/debfile.h
--- apt-inst/deb/debfile.h	Tue Feb 20 08:03:17 2001
+++ apt-inst/deb/debfile.h	Sun Apr 18 03:37:50 2004
@@ -23,7 +23,7 @@
 #ifndef PKGLIB_DEBFILE_H
 #define PKGLIB_DEBFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debfile.h"
 #endif
 
diff -rwub apt-inst/deb/dpkgdb.cc apt-inst/deb/dpkgdb.cc
--- apt-inst/deb/dpkgdb.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/deb/dpkgdb.cc	Sun Apr 18 03:37:51 2004
@@ -13,7 +13,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/dpkgdb.h"
 #endif
 
diff -rwub apt-inst/deb/dpkgdb.h apt-inst/deb/dpkgdb.h
--- apt-inst/deb/dpkgdb.h	Tue Feb 20 08:03:17 2001
+++ apt-inst/deb/dpkgdb.h	Sun Apr 18 03:37:51 2004
@@ -19,7 +19,7 @@
 #ifndef PKGLIB_DPKGDB_H
 #define PKGLIB_DPKGDB_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/dpkgdb.h"
 #endif
 
diff -rwub apt-inst/dirstream.cc apt-inst/dirstream.cc
--- apt-inst/dirstream.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/dirstream.cc	Sun Apr 18 03:37:51 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/dirstream.h"
 #endif
 
diff -rwub apt-inst/dirstream.h apt-inst/dirstream.h
--- apt-inst/dirstream.h	Tue Feb 20 08:03:16 2001
+++ apt-inst/dirstream.h	Sun Apr 18 03:37:51 2004
@@ -25,7 +25,7 @@
 #ifndef PKGLIB_DIRSTREAM_H
 #define PKGLIB_DIRSTREAM_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/dirstream.h"
 #endif
 
diff -rwub apt-inst/extract.cc apt-inst/extract.cc
--- apt-inst/extract.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/extract.cc	Sun Apr 18 03:37:51 2004
@@ -44,7 +44,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/extract.h"
 #endif
 #include <apt-pkg/extract.h>
diff -rwub apt-inst/extract.h apt-inst/extract.h
--- apt-inst/extract.h	Tue Feb 20 08:03:16 2001
+++ apt-inst/extract.h	Sun Apr 18 03:37:51 2004
@@ -17,7 +17,7 @@
 #ifndef PKGLIB_EXTRACT_H
 #define PKGLIB_EXTRACT_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/extract.h"
 #endif
 
diff -rwub apt-inst/filelist.cc apt-inst/filelist.cc
--- apt-inst/filelist.cc	Wed Jan  7 21:39:37 2004
+++ apt-inst/filelist.cc	Sun Apr 18 03:37:51 2004
@@ -32,7 +32,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/filelist.h"
 #endif
 
diff -rwub apt-inst/filelist.h apt-inst/filelist.h
--- apt-inst/filelist.h	Tue Feb 20 08:03:16 2001
+++ apt-inst/filelist.h	Sun Apr 18 03:37:51 2004
@@ -28,7 +28,7 @@
 #ifndef PKGLIB_FILELIST_H
 #define PKGLIB_FILELIST_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/filelist.h"
 #endif
 
diff -rwub apt-pkg/acquire-item.cc apt-pkg/acquire-item.cc
--- apt-pkg/acquire-item.cc	Sun Apr 20 18:48:55 2003
+++ apt-pkg/acquire-item.cc	Sun Apr 18 03:37:51 2004
@@ -13,7 +13,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/acquire-item.h"
 #endif
 #include <apt-pkg/acquire-item.h>
diff -rwub apt-pkg/acquire-item.h apt-pkg/acquire-item.h
--- apt-pkg/acquire-item.h	Sun Apr 20 18:48:55 2003
+++ apt-pkg/acquire-item.h	Sun Apr 18 03:37:51 2004
@@ -24,7 +24,7 @@
 #include <apt-pkg/indexfile.h>
 #include <apt-pkg/pkgrecords.h>
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/acquire-item.h"
 #endif 
 
diff -rwub apt-pkg/acquire-method.cc apt-pkg/acquire-method.cc
--- apt-pkg/acquire-method.cc	Tue May 22 06:27:11 2001
+++ apt-pkg/acquire-method.cc	Sun Apr 18 03:37:51 2004
@@ -15,7 +15,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/acquire-method.h"
 #endif
 #include <apt-pkg/acquire-method.h>
diff -rwub apt-pkg/acquire-method.h apt-pkg/acquire-method.h
--- apt-pkg/acquire-method.h	Tue Mar 13 07:51:46 2001
+++ apt-pkg/acquire-method.h	Sun Apr 18 03:37:51 2004
@@ -16,7 +16,7 @@
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/strutl.h>
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/acquire-method.h"
 #endif 
 
diff -rwub apt-pkg/acquire-worker.cc apt-pkg/acquire-worker.cc
--- apt-pkg/acquire-worker.cc	Tue May 22 06:42:54 2001
+++ apt-pkg/acquire-worker.cc	Sun Apr 18 03:37:51 2004
@@ -12,7 +12,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/acquire-worker.h"
 #endif
 #include <apt-pkg/acquire-worker.h>
diff -rwub apt-pkg/acquire-worker.h apt-pkg/acquire-worker.h
--- apt-pkg/acquire-worker.h	Tue Feb 20 08:03:17 2001
+++ apt-pkg/acquire-worker.h	Sun Apr 18 03:37:51 2004
@@ -14,7 +14,7 @@
 
 #include <apt-pkg/acquire.h>
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/acquire-worker.h"
 #endif 
 
diff -rwub apt-pkg/acquire.cc apt-pkg/acquire.cc
--- apt-pkg/acquire.cc	Wed Mar 17 06:17:11 2004
+++ apt-pkg/acquire.cc	Sun Apr 18 03:37:51 2004
@@ -13,7 +13,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/acquire.h"
 #endif       
 #include <apt-pkg/acquire.h>
diff -rwub apt-pkg/acquire.h apt-pkg/acquire.h
--- apt-pkg/acquire.h	Tue May 22 06:17:18 2001
+++ apt-pkg/acquire.h	Sun Apr 18 03:37:51 2004
@@ -38,7 +38,7 @@
 using std::vector;
 using std::string;
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/acquire.h"
 #endif 
 
diff -rwub apt-pkg/algorithms.cc apt-pkg/algorithms.cc
--- apt-pkg/algorithms.cc	Sat Aug  2 21:41:14 2003
+++ apt-pkg/algorithms.cc	Sun Apr 18 03:37:51 2004
@@ -14,7 +14,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/algorithms.h"
 #endif 
 #include <apt-pkg/algorithms.h>
diff -rwub apt-pkg/algorithms.h apt-pkg/algorithms.h
--- apt-pkg/algorithms.h	Sat Aug  2 21:41:14 2003
+++ apt-pkg/algorithms.h	Sun Apr 18 03:37:51 2004
@@ -30,7 +30,7 @@
 #ifndef PKGLIB_ALGORITHMS_H
 #define PKGLIB_ALGORITHMS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/algorithms.h"
 #endif 
 
diff -rwub apt-pkg/cachefile.cc apt-pkg/cachefile.cc
--- apt-pkg/cachefile.cc	Sat May  4 18:09:20 2002
+++ apt-pkg/cachefile.cc	Sun Apr 18 03:37:51 2004
@@ -12,7 +12,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/cachefile.h"
 #endif
 
diff -rwub apt-pkg/cachefile.h apt-pkg/cachefile.h
--- apt-pkg/cachefile.h	Sat May  4 18:09:20 2002
+++ apt-pkg/cachefile.h	Sun Apr 18 03:37:51 2004
@@ -17,7 +17,7 @@
 #ifndef PKGLIB_CACHEFILE_H
 #define PKGLIB_CACHEFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/cachefile.h"
 #endif 
 
diff -rwub apt-pkg/cacheiterators.h apt-pkg/cacheiterators.h
--- apt-pkg/cacheiterators.h	Sun Dec 21 00:49:02 2003
+++ apt-pkg/cacheiterators.h	Sun Apr 18 03:37:51 2004
@@ -31,7 +31,7 @@
 #ifndef PKGLIB_CACHEITERATORS_H
 #define PKGLIB_CACHEITERATORS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/cacheiterators.h"
 #endif 
 
diff -rwub apt-pkg/clean.cc apt-pkg/clean.cc
--- apt-pkg/clean.cc	Tue Feb 20 08:03:17 2001
+++ apt-pkg/clean.cc	Sun Apr 18 03:37:51 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Includes								/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/clean.h"
 #endif
 
diff -rwub apt-pkg/clean.h apt-pkg/clean.h
--- apt-pkg/clean.h	Tue Jul 20 07:53:33 1999
+++ apt-pkg/clean.h	Sun Apr 18 03:37:51 2004
@@ -10,7 +10,7 @@
 #ifndef APTPKG_CLEAN_H
 #define APTPKG_CLEAN_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/clean.h"
 #endif 
 
diff -rwub apt-pkg/contrib/cdromutl.cc apt-pkg/contrib/cdromutl.cc
--- apt-pkg/contrib/cdromutl.cc	Tue Feb 20 08:03:17 2001
+++ apt-pkg/contrib/cdromutl.cc	Sun Apr 18 03:37:51 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/cdromutl.h"
 #endif
 #include <apt-pkg/cdromutl.h>
diff -rwub apt-pkg/contrib/cdromutl.h apt-pkg/contrib/cdromutl.h
--- apt-pkg/contrib/cdromutl.h	Mon May  7 07:06:52 2001
+++ apt-pkg/contrib/cdromutl.h	Sun Apr 18 03:37:51 2004
@@ -14,7 +14,7 @@
 
 using std::string;
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/cdromutl.h"
 #endif 
 
diff -rwub apt-pkg/contrib/cmndline.cc apt-pkg/contrib/cmndline.cc
--- apt-pkg/contrib/cmndline.cc	Sun Apr 20 18:48:55 2003
+++ apt-pkg/contrib/cmndline.cc	Sun Apr 18 03:37:51 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/cmndline.h"
 #endif
 #include <apt-pkg/cmndline.h>
diff -rwub apt-pkg/contrib/cmndline.h apt-pkg/contrib/cmndline.h
--- apt-pkg/contrib/cmndline.h	Sun Oct 31 07:32:28 1999
+++ apt-pkg/contrib/cmndline.h	Sun Apr 18 03:37:51 2004
@@ -44,7 +44,7 @@
 #ifndef PKGLIB_CMNDLINE_H
 #define PKGLIB_CMNDLINE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/cmndline.h"
 #endif 
 
diff -rwub apt-pkg/contrib/configuration.cc apt-pkg/contrib/configuration.cc
--- apt-pkg/contrib/configuration.cc	Sat Jul 26 02:27:36 2003
+++ apt-pkg/contrib/configuration.cc	Sun Apr 18 03:37:52 2004
@@ -15,7 +15,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/configuration.h"
 #endif
 #include <apt-pkg/configuration.h>
diff -rwub apt-pkg/contrib/configuration.h apt-pkg/contrib/configuration.h
--- apt-pkg/contrib/configuration.h	Sun Nov 24 03:22:54 2002
+++ apt-pkg/contrib/configuration.h	Sun Apr 18 03:37:52 2004
@@ -28,7 +28,7 @@
 #ifndef PKGLIB_CONFIGURATION_H
 #define PKGLIB_CONFIGURATION_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/configuration.h"
 #endif 
 
diff -rwub apt-pkg/contrib/crc-16.cc apt-pkg/contrib/crc-16.cc
--- apt-pkg/contrib/crc-16.cc	Mon May 24 00:55:54 1999
+++ apt-pkg/contrib/crc-16.cc	Sun Apr 18 03:37:52 2004
@@ -15,7 +15,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/crc-16.h"
 #endif
 
diff -rwub apt-pkg/contrib/crc-16.h apt-pkg/contrib/crc-16.h
--- apt-pkg/contrib/crc-16.h	Mon May 24 00:55:54 1999
+++ apt-pkg/contrib/crc-16.h	Sun Apr 18 03:37:52 2004
@@ -10,7 +10,7 @@
 #ifndef APTPKG_CRC16_H
 #define APTPKG_CRC16_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/crc-16.h"
 #endif 
 
diff -rwub apt-pkg/contrib/error.cc apt-pkg/contrib/error.cc
--- apt-pkg/contrib/error.cc	Sat May  4 18:09:20 2002
+++ apt-pkg/contrib/error.cc	Sun Apr 18 03:37:52 2004
@@ -14,7 +14,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/error.h"
 #endif 
 
diff -rwub apt-pkg/contrib/error.h apt-pkg/contrib/error.h
--- apt-pkg/contrib/error.h	Mon May  7 07:06:52 2001
+++ apt-pkg/contrib/error.h	Sun Apr 18 03:37:52 2004
@@ -40,11 +40,11 @@
 #ifndef PKGLIB_ERROR_H
 #define PKGLIB_ERROR_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/error.h"
 #endif 
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 // Methods have a hidden this parameter that is visible to this attribute
 #define APT_MFORMAT1 __attribute__ ((format (printf, 2, 3)))
 #define APT_MFORMAT2 __attribute__ ((format (printf, 3, 4)))
diff -rwub apt-pkg/contrib/fileutl.cc apt-pkg/contrib/fileutl.cc
--- apt-pkg/contrib/fileutl.cc	Sun Nov 24 03:22:54 2002
+++ apt-pkg/contrib/fileutl.cc	Sun Apr 18 03:37:52 2004
@@ -14,7 +14,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/fileutl.h"
 #endif 
 #include <apt-pkg/fileutl.h>
diff -rwub apt-pkg/contrib/fileutl.h apt-pkg/contrib/fileutl.h
--- apt-pkg/contrib/fileutl.h	Mon May  7 07:06:52 2001
+++ apt-pkg/contrib/fileutl.h	Sun Apr 18 03:37:52 2004
@@ -21,7 +21,7 @@
 #ifndef PKGLIB_FILEUTL_H
 #define PKGLIB_FILEUTL_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/fileutl.h"
 #endif 
 
diff -rwub apt-pkg/contrib/hashes.cc apt-pkg/contrib/hashes.cc
--- apt-pkg/contrib/hashes.cc	Tue Mar  6 08:15:29 2001
+++ apt-pkg/contrib/hashes.cc	Sun Apr 18 03:37:52 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/hashes.h"
 #endif
 
diff -rwub apt-pkg/contrib/hashes.h apt-pkg/contrib/hashes.h
--- apt-pkg/contrib/hashes.h	Sun Mar 11 06:30:20 2001
+++ apt-pkg/contrib/hashes.h	Sun Apr 18 03:37:52 2004
@@ -13,7 +13,7 @@
 #ifndef APTPKG_HASHES_H
 #define APTPKG_HASHES_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/hashes.h"
 #endif 
 
diff -rwub apt-pkg/contrib/md5.cc apt-pkg/contrib/md5.cc
--- apt-pkg/contrib/md5.cc	Sun May 13 07:15:03 2001
+++ apt-pkg/contrib/md5.cc	Sun Apr 18 03:37:52 2004
@@ -35,7 +35,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/md5.h"
 #endif
 
diff -rwub apt-pkg/contrib/md5.h apt-pkg/contrib/md5.h
--- apt-pkg/contrib/md5.h	Mon May  7 07:06:52 2001
+++ apt-pkg/contrib/md5.h	Sun Apr 18 03:37:52 2004
@@ -23,7 +23,7 @@
 #ifndef APTPKG_MD5_H
 #define APTPKG_MD5_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/md5.h"
 #endif 
 
diff -rwub apt-pkg/contrib/mmap.cc apt-pkg/contrib/mmap.cc
--- apt-pkg/contrib/mmap.cc	Sun May 27 07:19:30 2001
+++ apt-pkg/contrib/mmap.cc	Sun Apr 18 03:37:52 2004
@@ -21,7 +21,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/mmap.h"
 #endif 
 
diff -rwub apt-pkg/contrib/mmap.h apt-pkg/contrib/mmap.h
--- apt-pkg/contrib/mmap.h	Mon May 14 07:16:43 2001
+++ apt-pkg/contrib/mmap.h	Sun Apr 18 03:37:52 2004
@@ -25,7 +25,7 @@
 #ifndef PKGLIB_MMAP_H
 #define PKGLIB_MMAP_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/mmap.h"
 #endif
 
diff -rwub apt-pkg/contrib/progress.cc apt-pkg/contrib/progress.cc
--- apt-pkg/contrib/progress.cc	Sun Jan 12 03:09:29 2003
+++ apt-pkg/contrib/progress.cc	Sun Apr 18 03:37:52 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/progress.h"
 #endif 
 #include <apt-pkg/progress.h>
diff -rwub apt-pkg/contrib/progress.h apt-pkg/contrib/progress.h
--- apt-pkg/contrib/progress.h	Mon May  7 07:06:52 2001
+++ apt-pkg/contrib/progress.h	Sun Apr 18 03:37:52 2004
@@ -21,7 +21,7 @@
 #ifndef PKGLIB_PROGRESS_H
 #define PKGLIB_PROGRESS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/progress.h"
 #endif 
 
diff -rwub apt-pkg/contrib/sha1.cc apt-pkg/contrib/sha1.cc
--- apt-pkg/contrib/sha1.cc	Sun May 13 07:15:03 2001
+++ apt-pkg/contrib/sha1.cc	Sun Apr 18 03:37:52 2004
@@ -29,7 +29,7 @@
  */
 									/*}}} */
 // Include Files                                                        /*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/sha1.h"
 #endif
 
diff -rwub apt-pkg/contrib/sha1.h apt-pkg/contrib/sha1.h
--- apt-pkg/contrib/sha1.h	Mon May  7 07:05:47 2001
+++ apt-pkg/contrib/sha1.h	Sun Apr 18 03:37:52 2004
@@ -14,7 +14,7 @@
 #ifndef APTPKG_SHA1_H
 #define APTPKG_SHA1_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/sha1.h"
 #endif 
 
diff -rwub apt-pkg/contrib/strutl.cc apt-pkg/contrib/strutl.cc
--- apt-pkg/contrib/strutl.cc	Fri Jul 18 16:15:11 2003
+++ apt-pkg/contrib/strutl.cc	Sun Apr 18 03:37:52 2004
@@ -15,7 +15,7 @@
    ##################################################################### */
 									/*}}}*/
 // Includes								/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/strutl.h"
 #endif
 
diff -rwub apt-pkg/contrib/strutl.h apt-pkg/contrib/strutl.h
--- apt-pkg/contrib/strutl.h	Sun Apr 20 18:48:55 2003
+++ apt-pkg/contrib/strutl.h	Sun Apr 18 03:37:52 2004
@@ -16,7 +16,7 @@
 #ifndef STRUTL_H
 #define STRUTL_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/strutl.h"
 #endif 
 
@@ -30,7 +30,7 @@
 using std::vector;
 using std::ostream;
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 // Methods have a hidden this parameter that is visible to this attribute
 #define APT_FORMAT2 __attribute__ ((format (printf, 2, 3)))
 #define APT_FORMAT3 __attribute__ ((format (printf, 3, 4)))
diff -rwub apt-pkg/deb/debindexfile.cc apt-pkg/deb/debindexfile.cc
--- apt-pkg/deb/debindexfile.cc	Sun Jan  4 08:41:30 2004
+++ apt-pkg/deb/debindexfile.cc	Sun Apr 18 03:37:52 2004
@@ -9,7 +9,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debindexfile.h"
 #endif
 
diff -rwub apt-pkg/deb/debindexfile.h apt-pkg/deb/debindexfile.h
--- apt-pkg/deb/debindexfile.h	Sun Apr 29 07:13:51 2001
+++ apt-pkg/deb/debindexfile.h	Sun Apr 18 03:37:52 2004
@@ -16,7 +16,7 @@
 #ifndef PKGLIB_DEBINDEXFILE_H
 #define PKGLIB_DEBINDEXFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debindexfile.h"
 #endif
 
diff -rwub apt-pkg/deb/debrecords.cc apt-pkg/deb/debrecords.cc
--- apt-pkg/deb/debrecords.cc	Tue Mar 13 07:51:46 2001
+++ apt-pkg/deb/debrecords.cc	Sun Apr 18 03:37:52 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debrecords.h"
 #endif
 #include <apt-pkg/debrecords.h>
diff -rwub apt-pkg/deb/debrecords.h apt-pkg/deb/debrecords.h
--- apt-pkg/deb/debrecords.h	Tue Mar 13 07:51:46 2001
+++ apt-pkg/deb/debrecords.h	Sun Apr 18 03:37:52 2004
@@ -14,7 +14,7 @@
 #ifndef PKGLIB_DEBRECORDS_H
 #define PKGLIB_DEBRECORDS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debrecords.h"
 #endif 
 
diff -rwub apt-pkg/deb/debsrcrecords.cc apt-pkg/deb/debsrcrecords.cc
--- apt-pkg/deb/debsrcrecords.cc	Wed Mar 17 06:58:54 2004
+++ apt-pkg/deb/debsrcrecords.cc	Sun Apr 18 03:37:52 2004
@@ -9,7 +9,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debsrcrecords.h"
 #endif 
 
diff -rwub apt-pkg/deb/debsrcrecords.h apt-pkg/deb/debsrcrecords.h
--- apt-pkg/deb/debsrcrecords.h	Wed Mar 17 06:58:54 2004
+++ apt-pkg/deb/debsrcrecords.h	Sun Apr 18 03:37:52 2004
@@ -11,7 +11,7 @@
 #ifndef PKGLIB_DEBSRCRECORDS_H
 #define PKGLIB_DEBSRCRECORDS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debsrcrecords.h"
 #endif 
 
diff -rwub apt-pkg/deb/debsystem.cc apt-pkg/deb/debsystem.cc
--- apt-pkg/deb/debsystem.cc	Mon Jan 26 18:01:53 2004
+++ apt-pkg/deb/debsystem.cc	Sun Apr 18 03:37:52 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debsystem.h"
 #endif
 
diff -rwub apt-pkg/deb/debsystem.h apt-pkg/deb/debsystem.h
--- apt-pkg/deb/debsystem.h	Sun Jan 12 03:09:29 2003
+++ apt-pkg/deb/debsystem.h	Sun Apr 18 03:37:52 2004
@@ -10,7 +10,7 @@
 #ifndef PKGLIB_DEBSYSTEM_H
 #define PKGLIB_DEBSYSTEM_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debsystem.h"
 #endif
 
diff -rwub apt-pkg/deb/debversion.cc apt-pkg/deb/debversion.cc
--- apt-pkg/deb/debversion.cc	Thu Sep 11 01:39:49 2003
+++ apt-pkg/deb/debversion.cc	Sun Apr 18 03:37:53 2004
@@ -11,7 +11,7 @@
 									/*}}}*/
 // Include Files							/*{{{*/
 #define APT_COMPATIBILITY 986
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/debversion.h"
 #endif
 
diff -rwub apt-pkg/deb/debversion.h apt-pkg/deb/debversion.h
--- apt-pkg/deb/debversion.h	Thu May  3 07:25:04 2001
+++ apt-pkg/deb/debversion.h	Sun Apr 18 03:37:53 2004
@@ -12,7 +12,7 @@
 #ifndef PKGLIB_DEBVERSION_H
 #define PKGLIB_DEBVERSION_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/debversion.h"
 #endif 
 
diff -rwub apt-pkg/deb/dpkgpm.cc apt-pkg/deb/dpkgpm.cc
--- apt-pkg/deb/dpkgpm.cc	Tue Jan 27 03:25:01 2004
+++ apt-pkg/deb/dpkgpm.cc	Sun Apr 18 03:37:53 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Includes								/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/dpkgpm.h"
 #endif
 #include <apt-pkg/dpkgpm.h>
diff -rwub apt-pkg/deb/dpkgpm.h apt-pkg/deb/dpkgpm.h
--- apt-pkg/deb/dpkgpm.h	Mon May  7 07:05:13 2001
+++ apt-pkg/deb/dpkgpm.h	Sun Apr 18 03:37:53 2004
@@ -10,7 +10,7 @@
 #ifndef PKGLIB_DPKGPM_H
 #define PKGLIB_DPKGPM_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/dpkgpm.h"
 #endif
 
diff -rwub apt-pkg/depcache.cc apt-pkg/depcache.cc
--- apt-pkg/depcache.cc	Sun May 27 07:36:04 2001
+++ apt-pkg/depcache.cc	Sun Apr 18 03:37:53 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/depcache.h"
 #endif
 #include <apt-pkg/depcache.h>
diff -rwub apt-pkg/depcache.h apt-pkg/depcache.h
--- apt-pkg/depcache.h	Tue Feb 20 08:03:17 2001
+++ apt-pkg/depcache.h	Sun Apr 18 03:37:53 2004
@@ -38,7 +38,7 @@
 #ifndef PKGLIB_DEPCACHE_H
 #define PKGLIB_DEPCACHE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/depcache.h"
 #endif
 
diff -rwub apt-pkg/indexfile.cc apt-pkg/indexfile.cc
--- apt-pkg/indexfile.cc	Tue Feb 20 08:03:17 2001
+++ apt-pkg/indexfile.cc	Sun Apr 18 03:37:53 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/indexfile.h"
 #endif
 
diff -rwub apt-pkg/indexfile.h apt-pkg/indexfile.h
--- apt-pkg/indexfile.h	Sun Aug 11 22:48:56 2002
+++ apt-pkg/indexfile.h	Sun Apr 18 03:37:53 2004
@@ -21,7 +21,7 @@
 #ifndef PKGLIB_INDEXFILE_H
 #define PKGLIB_INDEXFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/indexfile.h"
 #endif
 
diff -rwub apt-pkg/orderlist.cc apt-pkg/orderlist.cc
--- apt-pkg/orderlist.cc	Mon May  7 07:49:43 2001
+++ apt-pkg/orderlist.cc	Sun Apr 18 03:37:53 2004
@@ -63,7 +63,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/orderlist.h"
 #endif 
 #include <apt-pkg/orderlist.h>
diff -rwub apt-pkg/orderlist.h apt-pkg/orderlist.h
--- apt-pkg/orderlist.h	Tue Feb 20 08:03:17 2001
+++ apt-pkg/orderlist.h	Sun Apr 18 03:37:53 2004
@@ -16,7 +16,7 @@
 #ifndef PKGLIB_ORDERLIST_H
 #define PKGLIB_ORDERLIST_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/orderlist.h"
 #endif 
 
diff -rwub apt-pkg/packagemanager.cc apt-pkg/packagemanager.cc
--- apt-pkg/packagemanager.cc	Sat May 17 21:29:09 2003
+++ apt-pkg/packagemanager.cc	Sun Apr 18 03:37:53 2004
@@ -13,7 +13,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/packagemanager.h"
 #endif
 
diff -rwub apt-pkg/packagemanager.h apt-pkg/packagemanager.h
--- apt-pkg/packagemanager.h	Mon May  7 06:24:08 2001
+++ apt-pkg/packagemanager.h	Sun Apr 18 03:37:53 2004
@@ -23,7 +23,7 @@
 #ifndef PKGLIB_PACKAGEMANAGER_H
 #define PKGLIB_PACKAGEMANAGER_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/packagemanager.h"
 #endif
 
diff -rwub apt-pkg/pkgcache.cc apt-pkg/pkgcache.cc
--- apt-pkg/pkgcache.cc	Sun Apr 20 18:48:55 2003
+++ apt-pkg/pkgcache.cc	Sun Apr 18 03:37:53 2004
@@ -20,7 +20,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/pkgcache.h"
 #pragma implementation "apt-pkg/cacheiterators.h"
 #endif 
diff -rwub apt-pkg/pkgcache.h apt-pkg/pkgcache.h
--- apt-pkg/pkgcache.h	Mon Jul  2 00:28:24 2001
+++ apt-pkg/pkgcache.h	Sun Apr 18 03:37:53 2004
@@ -19,7 +19,7 @@
 #ifndef PKGLIB_PKGCACHE_H
 #define PKGLIB_PKGCACHE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/pkgcache.h"
 #endif 
 
diff -rwub apt-pkg/pkgcachegen.cc apt-pkg/pkgcachegen.cc
--- apt-pkg/pkgcachegen.cc	Sun Apr 20 18:48:55 2003
+++ apt-pkg/pkgcachegen.cc	Sun Apr 18 03:37:53 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/pkgcachegen.h"
 #endif
 
diff -rwub apt-pkg/pkgcachegen.h apt-pkg/pkgcachegen.h
--- apt-pkg/pkgcachegen.h	Sun Aug 11 22:48:57 2002
+++ apt-pkg/pkgcachegen.h	Sun Apr 18 03:37:53 2004
@@ -19,7 +19,7 @@
 #ifndef PKGLIB_PKGCACHEGEN_H
 #define PKGLIB_PKGCACHEGEN_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/pkgcachegen.h"
 #endif 
 
diff -rwub apt-pkg/pkgrecords.cc apt-pkg/pkgrecords.cc
--- apt-pkg/pkgrecords.cc	Tue Sep  2 06:52:16 2003
+++ apt-pkg/pkgrecords.cc	Sun Apr 18 03:37:53 2004
@@ -9,7 +9,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/pkgrecords.h"
 #endif
 #include <apt-pkg/pkgrecords.h>
diff -rwub apt-pkg/pkgrecords.h apt-pkg/pkgrecords.h
--- apt-pkg/pkgrecords.h	Tue Sep  2 06:51:56 2003
+++ apt-pkg/pkgrecords.h	Sun Apr 18 03:37:53 2004
@@ -17,7 +17,7 @@
 #ifndef PKGLIB_PKGRECORDS_H
 #define PKGLIB_PKGRECORDS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/pkgrecords.h"
 #endif 
 
diff -rwub apt-pkg/pkgsystem.cc apt-pkg/pkgsystem.cc
--- apt-pkg/pkgsystem.cc	Fri Feb 27 01:43:16 2004
+++ apt-pkg/pkgsystem.cc	Sun Apr 18 03:37:53 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/pkgsystem.h"
 #endif
 
diff -rwub apt-pkg/pkgsystem.h apt-pkg/pkgsystem.h
--- apt-pkg/pkgsystem.h	Sun Nov 24 03:22:54 2002
+++ apt-pkg/pkgsystem.h	Sun Apr 18 03:37:53 2004
@@ -37,7 +37,7 @@
 #ifndef PKGLIB_PKGSYSTEM_H
 #define PKGLIB_PKGSYSTEM_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/pkgsystem.h"
 #endif
 
diff -rwub apt-pkg/policy.cc apt-pkg/policy.cc
--- apt-pkg/policy.cc	Tue Aug 12 02:17:37 2003
+++ apt-pkg/policy.cc	Sun Apr 18 03:37:53 2004
@@ -23,7 +23,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/policy.h"
 #endif
 #include <apt-pkg/policy.h>
diff -rwub apt-pkg/policy.h apt-pkg/policy.h
--- apt-pkg/policy.h	Mon May  7 06:24:08 2001
+++ apt-pkg/policy.h	Sun Apr 18 03:37:53 2004
@@ -33,7 +33,7 @@
 #ifndef PKGLIB_POLICY_H
 #define PKGLIB_POLICY_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/policy.h"
 #endif
 
diff -rwub apt-pkg/sourcelist.cc apt-pkg/sourcelist.cc
--- apt-pkg/sourcelist.cc	Sun Aug 11 22:48:57 2002
+++ apt-pkg/sourcelist.cc	Sun Apr 18 03:37:53 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/sourcelist.h"
 #endif
 
diff -rwub apt-pkg/sourcelist.h apt-pkg/sourcelist.h
--- apt-pkg/sourcelist.h	Sun Aug 11 22:48:57 2002
+++ apt-pkg/sourcelist.h	Sun Apr 18 03:37:53 2004
@@ -35,7 +35,7 @@
 using std::string;
 using std::vector;
     
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/sourcelist.h"
 #endif
 
diff -rwub apt-pkg/srcrecords.cc apt-pkg/srcrecords.cc
--- apt-pkg/srcrecords.cc	Sun Nov 24 03:22:54 2002
+++ apt-pkg/srcrecords.cc	Sun Apr 18 03:37:53 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/srcrecords.h"
 #endif 
 
diff -rwub apt-pkg/srcrecords.h apt-pkg/srcrecords.h
--- apt-pkg/srcrecords.h	Sun Nov  4 18:09:18 2001
+++ apt-pkg/srcrecords.h	Sun Apr 18 03:37:53 2004
@@ -13,7 +13,7 @@
 #ifndef PKGLIB_SRCRECORDS_H
 #define PKGLIB_SRCRECORDS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/srcrecords.h"
 #endif 
 
diff -rwub apt-pkg/tagfile.cc apt-pkg/tagfile.cc
--- apt-pkg/tagfile.cc	Mon Jul 21 07:52:45 2003
+++ apt-pkg/tagfile.cc	Sun Apr 18 03:37:53 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/tagfile.h"
 #endif
 
diff -rwub apt-pkg/tagfile.h apt-pkg/tagfile.h
--- apt-pkg/tagfile.h	Sat Jun 14 19:07:11 2003
+++ apt-pkg/tagfile.h	Sun Apr 18 03:37:53 2004
@@ -20,7 +20,7 @@
 #ifndef PKGLIB_TAGFILE_H
 #define PKGLIB_TAGFILE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/tagfile.h"
 #endif 
 
diff -rwub apt-pkg/version.cc apt-pkg/version.cc
--- apt-pkg/version.cc	Tue Feb 20 08:03:17 2001
+++ apt-pkg/version.cc	Sun Apr 18 03:37:53 2004
@@ -8,7 +8,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/version.h"
 #endif 
 
diff -rwub apt-pkg/version.h apt-pkg/version.h
--- apt-pkg/version.h	Sun May 27 07:55:27 2001
+++ apt-pkg/version.h	Sun Apr 18 03:37:54 2004
@@ -20,7 +20,7 @@
 #ifndef PKGLIB_VERSION_H
 #define PKGLIB_VERSION_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/version.h"
 #endif 
 
diff -rwub apt-pkg/versionmatch.cc apt-pkg/versionmatch.cc
--- apt-pkg/versionmatch.cc	Sat Jun 14 19:07:12 2003
+++ apt-pkg/versionmatch.cc	Sun Apr 18 03:37:54 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-pkg/versionmatch.h"
 #endif
 #include <apt-pkg/versionmatch.h>
diff -rwub apt-pkg/versionmatch.h apt-pkg/versionmatch.h
--- apt-pkg/versionmatch.h	Tue May 29 05:07:12 2001
+++ apt-pkg/versionmatch.h	Sun Apr 18 03:37:54 2004
@@ -31,7 +31,7 @@
 #ifndef PKGLIB_VERSIONMATCH_H
 #define PKGLIB_VERSIONMATCH_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-pkg/versionmatch.h"
 #endif
 
diff -rwub ftparchive/apt-ftparchive.cc ftparchive/apt-ftparchive.cc
--- ftparchive/apt-ftparchive.cc	Fri Jan  2 22:48:13 2004
+++ ftparchive/apt-ftparchive.cc	Sun Apr 18 03:37:54 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "apt-ftparchive.h"
 #endif
 
diff -rwub ftparchive/apt-ftparchive.h ftparchive/apt-ftparchive.h
--- ftparchive/apt-ftparchive.h	Tue May 29 05:51:37 2001
+++ ftparchive/apt-ftparchive.h	Sun Apr 18 03:37:54 2004
@@ -13,7 +13,7 @@
 #ifndef APT_FTPARCHIVE_H
 #define APT_FTPARCHIVE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "apt-ftparchive.h"
 #endif
 
diff -rwub ftparchive/cachedb.cc ftparchive/cachedb.cc
--- ftparchive/cachedb.cc	Sun Apr 20 18:48:56 2003
+++ ftparchive/cachedb.cc	Sun Apr 18 03:37:54 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "cachedb.h"
 #endif
 
diff -rwub ftparchive/cachedb.h ftparchive/cachedb.h
--- ftparchive/cachedb.h	Sun Nov 24 03:22:55 2002
+++ ftparchive/cachedb.h	Sun Apr 18 03:37:54 2004
@@ -12,7 +12,7 @@
 #ifndef CACHEDB_H
 #define CACHEDB_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "cachedb.h"
 #endif 
 
diff -rwub ftparchive/multicompress.cc ftparchive/multicompress.cc
--- ftparchive/multicompress.cc	Sun Apr 20 18:48:56 2003
+++ ftparchive/multicompress.cc	Sun Apr 18 03:37:54 2004
@@ -14,7 +14,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "multicompress.h"
 #endif
 
diff -rwub ftparchive/multicompress.h ftparchive/multicompress.h
--- ftparchive/multicompress.h	Tue Feb 20 08:03:18 2001
+++ ftparchive/multicompress.h	Sun Apr 18 03:37:54 2004
@@ -16,7 +16,7 @@
 #ifndef MULTICOMPRESS_H
 #define MULTICOMPRESS_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "multicompress.h"
 #endif 
 
diff -rwub ftparchive/override.cc ftparchive/override.cc
--- ftparchive/override.cc	Sun Apr 20 18:48:56 2003
+++ ftparchive/override.cc	Sun Apr 18 03:37:54 2004
@@ -10,7 +10,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "override.h"
 #endif
 
diff -rwub ftparchive/override.h ftparchive/override.h
--- ftparchive/override.h	Sun Jan  4 20:06:26 2004
+++ ftparchive/override.h	Sun Apr 18 03:37:54 2004
@@ -12,7 +12,7 @@
 #ifndef OVERRIDE_H
 #define OVERRIDE_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "override.h"
 #endif
 
diff -rwub ftparchive/writer.cc ftparchive/writer.cc
--- ftparchive/writer.cc	Sun Jan  4 20:06:26 2004
+++ ftparchive/writer.cc	Sun Apr 18 03:37:54 2004
@@ -11,7 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma implementation "writer.h"
 #endif
 
diff -rwub ftparchive/writer.h ftparchive/writer.h
--- ftparchive/writer.h	Sun Jan  4 01:19:38 2004
+++ ftparchive/writer.h	Sun Apr 18 03:37:54 2004
@@ -13,7 +13,7 @@
 #ifndef WRITER_H
 #define WRITER_H
 
-#ifdef __GNUG__
+#if defined(__GNUG__) && !defined(__APPLE_CC__)
 #pragma interface "writer.h"
 #endif 
 
