--- tclUnixSock.c.orig	2017-08-13 07:57:43.000000000 -0700
+++ tclUnixSock.c	2017-08-13 07:58:30.000000000 -0700
@@ -700,6 +700,10 @@
  */
 
 #ifndef NEED_FAKE_RFC2553
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+
 static inline int
 IPv6AddressNeedsNumericRendering(
     struct in6_addr addr)
@@ -713,16 +717,14 @@
      * at least some versions of OSX.
      */
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wstrict-aliasing"
     if (!IN6_IS_ADDR_V4MAPPED(&addr)) {
-#pragma GCC diagnostic pop
         return 0;
     }
 
     return (addr.s6_addr[12] == 0 && addr.s6_addr[13] == 0
             && addr.s6_addr[14] == 0 && addr.s6_addr[15] == 0);
 }
+#pragma GCC diagnostic pop
 #endif /* NEED_FAKE_RFC2553 */
 
 static void
