Index: sqliteman/litemanwindow.cpp
===================================================================
--- sqliteman/litemanwindow.cpp	(revision 490)
+++ sqliteman/litemanwindow.cpp	(working copy)
@@ -129,12 +129,9 @@
 		i.next();
 		QSqlDatabase::database(i.value()).rollback();
 		QSqlDatabase::database(i.value()).close();
+		QSqlDatabase::removeDatabase(i.value());
 	}
 
-	// It has to go after writeSettings()!
-	foreach (QWidget *widget, QApplication::topLevelWidgets())
-		widget->close();
-
 	e->accept();
 }
 
