--- sample-config/template-object/minimal.cfg.in.org	2006-03-21 09:04:58.000000000 -0800
+++ sample-config/template-object/minimal.cfg.in	2006-03-31 22:41:34.000000000 -0800
@@ -53,71 +53,78 @@
 ###############################################################################
 ###############################################################################
 
+#
+# The COMMANDS section has been remarked out during the Nagios DarwinPorts
+# port install.  These commands are redundant with the ones in the
+# misccommands.cfg file and creates startup errors.
+#
+
+
 # This is a sample service notification command that can be used to send email 
 # notifications (about service alerts) to contacts.
 
-define command{
-	command_name	notify-by-email
-	command_line	/usr/bin/printf "%b" "***** Nagios @VERSION@ *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$" | @MAIL_PROG@ -s "** $NOTIFICATIONTYPE$ alert - $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
-	}
+#define command{
+#	command_name	notify-by-email
+#	command_line	/usr/bin/printf "%b" "***** Nagios @VERSION@ *****\n\nNotification Type: $NOTIFICATIONTYPE$\n\nService: $SERVICEDESC$\nHost: $HOSTALIAS$\nAddress: $HOSTADDRESS$\nState: $SERVICESTATE$\n\nDate/Time: $LONGDATETIME$\n\nAdditional Info:\n\n$SERVICEOUTPUT$" | @MAIL_PROG@ -s "** $NOTIFICATIONTYPE$ alert - $HOSTALIAS$/$SERVICEDESC$ is $SERVICESTATE$ **" $CONTACTEMAIL$
+#	}
 
 
 # This is a sample host notification command that can be used to send email 
 # notifications (about host alerts) to contacts.
 
-define command{
-	command_name	host-notify-by-email
-	command_line	/usr/bin/printf "%b" "***** Nagios @VERSION@ *****\n\nNotification Type: $NOTIFICATIONTYPE$\nHost: $HOSTNAME$\nState: $HOSTSTATE$\nAddress: $HOSTADDRESS$\nInfo: $HOSTOUTPUT$\n\nDate/Time: $LONGDATETIME$\n" | @MAIL_PROG@ -s "Host $HOSTSTATE$ alert for $HOSTNAME$!" $CONTACTEMAIL$
-	}
+#define command{
+#	command_name	host-notify-by-email
+#	command_line	/usr/bin/printf "%b" "***** Nagios @VERSION@ *****\n\nNotification Type: $NOTIFICATIONTYPE$\nHost: $HOSTNAME$\nState: $HOSTSTATE$\nAddress: $HOSTADDRESS$\nInfo: $HOSTOUTPUT$\n\nDate/Time: $LONGDATETIME$\n" | @MAIL_PROG@ -s "Host $HOSTSTATE$ alert for $HOSTNAME$!" $CONTACTEMAIL$
+#	}
 
 
 # Command to check to see if a host is "alive" (up) by pinging it
 
-define command{
-        command_name    check-host-alive
-        command_line    $USER1$/check_ping -H $HOSTADDRESS$ -w 99,99% -c 100,100% -p 1 
-        }
+#define command{
+#        command_name    check-host-alive
+#        command_line    $USER1$/check_ping -H $HOSTADDRESS$ -w 99,99% -c 100,100% -p 1 
+#        }
 
 
 # Generic command to check a device by pinging it
 
-define command{
-	command_name	check_ping
-	command_line	$USER1$/check_ping -H $HOSTADDRESS$ -w $ARG1$ -c $ARG2$ -p 5
-	}
+#define command{
+#	command_name	check_ping
+#	command_line	$USER1$/check_ping -H $HOSTADDRESS$ -w $ARG1$ -c $ARG2$ -p 5
+#	}
 
 
 # Command used to check disk space usage on local partitions
 
-define command{
-	command_name	check_local_disk
-	command_line	$USER1$/check_disk -w $ARG1$ -c $ARG2$ -p $ARG3$
-	}
+#define command{
+#	command_name	check_local_disk
+#	command_line	$USER1$/check_disk -w $ARG1$ -c $ARG2$ -p $ARG3$
+#	}
 
 
 # Command used to check the number of currently logged in users on the
 # local machine
 
-define command{
-	command_name	check_local_users
-	command_line	$USER1$/check_users -w $ARG1$ -c $ARG2$
-	}
+#define command{
+#	command_name	check_local_users
+#	command_line	$USER1$/check_users -w $ARG1$ -c $ARG2$
+#	}
 
 
 # Command to check the number of running processing on the local machine
 
-define command{
-	command_name	check_local_procs
-	command_line	$USER1$/check_procs -w $ARG1$ -c $ARG2$
-	}
+#define command{
+#	command_name	check_local_procs
+#	command_line	$USER1$/check_procs -w $ARG1$ -c $ARG2$
+#	}
 
 
 # Command to check the load on the local machine
 
-define command{
-	command_name	check_local_load
-	command_line	$USER1$/check_load -w $ARG1$ -c $ARG2$
-	}
+#define command{
+#	command_name	check_local_load
+#	command_line	$USER1$/check_load -w $ARG1$ -c $ARG2$
+#	}
 
 
 
