--- setup.orig.py
+++ setup.py
@@ -455,7 +455,11 @@
     libname = _get_lib_name("tau", add_extension_suffix=False)
     files = glob.glob(os.path.join(path, "*.f"))
     # compiler specific options
-    kwargs = {'libraries': []}
+    # 2014-02-10, petr -- modification needed below
+    kwargs = {}
+    kwargs.update({'libraries': []})
+    kwargs.update({'extra_link_args': []})
+
     # XXX: The build subdirectory is difficult to determine if installed
     # via pypi or other means. I could not find a reliable way of doing it.
     new_interface_path = os.path.join("build", libname + os.extsep + "pyf")
@@ -473,6 +477,20 @@
     # gfortran the option -lgcov is required
     if os.environ.get('OBSPY_C_COVERAGE', ""):
         kwargs['libraries'].append('gcov')
+
+    # 2014-02-10, petr -- workaround for unexpected numpy.distutils behaviour
+    #   This hot fix is needed for Macports. Macports sets `LDFLAGS` via
+    #   `Python Portgroup` to control build architecture setings.
+    #   For Fortran extentions this causes the default link arguments to be
+    #   canceled, which does not correspond to C bevavior. The fix below adds
+    #   these arguments manually.
+    #
+    #   Should probably be adressed in numpy.distutils.
+    #
+    if platform.system() == 'Darwin':
+        kwargs['extra_link_args'].append('-bundle')
+        kwargs['extra_link_args'].append('-undefined dynamic_lookup')
+
     config.add_extension(libname, files, **kwargs)
 
     add_data_files(config)
