--- src/ui/FFMainController.m.orig	2007-08-26 14:40:42.000000000 -0500
+++ src/ui/FFMainController.m	2015-05-06 23:05:10.000000000 -0500
@@ -364,7 +364,7 @@
 }
 
 - (BOOL)validateMenuItem:(id<NSMenuItem>)mitem {
-	switch ([mitem tag]) {
+	switch ((int)[mitem tag]) {
 		// Only if images 
 		case MENU_FILE_CLEAR_IMGLIST :
 		case MENU_FILE_REMOVE :
@@ -813,7 +813,7 @@
 			[encItem setTarget:self];
 			[encItem setTag:*encIDs];
 			
-			if (*encIDs < NSProprietaryStringEncoding)
+			if (*encIDs < 65536 /*NSProprietaryStringEncoding*/)
 				[commonItems addObject:encItem];
 			else
 				[propItems addObject:encItem];	
