--- Configure.orig	2006-01-15 19:21:24.000000000 +0100
+++ Configure	2006-01-15 20:57:02.000000000 +0100
@@ -853,10 +853,10 @@
 case "$fastread$alldone" in
 yescont|yesexit) ;;
 *)
-	if test ! -t 0; then
-		echo "Say 'sh Configure', not 'sh <Configure'"
-		exit 1
-	fi
+#	if test ! -t 0; then
+#		echo "Say 'sh Configure', not 'sh <Configure'"
+#		exit 1
+#	fi
 	;;
 esac
 
@@ -7279,14 +7279,14 @@
 to configure things, do it as a shell escape now (e.g. !vi config.sh).
 
 EOM
-	rp="Press return or use a shell escape to edit config.sh:"
-	. UU/myread
-	nostick=''
-	case "$ans" in
-	'') ;;
-	*) : in case they cannot read
-		sh 1>&4 -c "$ans";;
-	esac
+#	rp="Press return or use a shell escape to edit config.sh:"
+#	. UU/myread
+#	nostick=''
+#	case "$ans" in
+#	'') ;;
+#	*) : in case they cannot read
+#		sh 1>&4 -c "$ans";;
+#	esac
 	;;
 esac
 
