--- GNUmakefile.orig	2009-07-12 21:16:26.000000000 -0700
+++ GNUmakefile	2009-07-12 21:20:08.000000000 -0700
@@ -6,9 +6,9 @@
 
 #FDEBUG= 1
 
-CC?= gcc
-CFLAGS+= -DBUILD="\"$(VERSION)\""
-LDFLAGS+= -L/usr/local/lib
+CC?= __CC__
+CFLAGS+= __CFLAGS__ -DBUILD="\"$(VERSION)\""
+LDFLAGS+= __LDFLAGS__
 LIBS+= -lncurses
 
 # This sort of sucks but gets rid of the stupid warning and should work on
@@ -16,21 +16,21 @@
 ifeq ($(shell (LC_ALL=C $(CC) -v 2>&1|awk '/gcc version 4/') || true), )
 CPPFLAGS:= -I. -I- $(CPPFLAGS)
 else
-CPPFLAGS:= -iquote. $(CPPFLAGS)
+CPPFLAGS:= -I__PREFIX__/include -I__PREFIX__/include/ncurses -iquote. $(CPPFLAGS)
 endif
 
 ifdef FDEBUG
 CFLAGS+= -g -ggdb -DDEBUG
+endif
 CFLAGS+= -Wno-long-long -Wall -W -Wnested-externs -Wformat=2
 CFLAGS+= -Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations
 CFLAGS+= -Wwrite-strings -Wshadow -Wpointer-arith -Wsign-compare
 CFLAGS+= -Wundef -Wbad-function-cast -Winline -Wcast-align
-endif
 
-PREFIX?= /usr/local
+PREFIX?= __PREFIX__
 INSTALLDIR= install -d
-INSTALLBIN= install -g bin -o root -m 555
-INSTALLMAN= install -g bin -o root -m 444
+INSTALLBIN= install -m 555
+INSTALLMAN= install -m 444
 
 SRCS= $(shell echo *.c|sed 's|osdep-[a-z0-9]*.c||g')
 include config.mk
@@ -53,5 +53,5 @@
 install:	all
 		$(INSTALLDIR) $(DESTDIR)$(PREFIX)/bin
 		$(INSTALLBIN) tmux $(DESTDIR)$(PREFIX)/bin/tmux
-		$(INSTALLDIR) $(DESTDIR)$(PREFIX)/man/man1
-		$(INSTALLMAN) tmux.1 $(DESTDIR)$(PREFIX)/man/man1/tmux.1
+		$(INSTALLDIR) $(DESTDIR)$(PREFIX)/share/man/man1
+		$(INSTALLMAN) tmux.1 $(DESTDIR)$(PREFIX)/share/man/man1/tmux.1
