diff -ur -x .DS_Store -x '*.orig' hdhomerun_config_gui/Makefile.in hdhomerun_config_gui/Makefile.in
--- hdhomerun_config_gui/Makefile.in	2013-03-23 13:22:13.000000000 -0400
+++ hdhomerun_config_gui/Makefile.in	2013-12-01 20:38:28.000000000 -0500
@@ -182,7 +182,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = ../libhdhomerun src po
+SUBDIRS = src po
 EXTRA_DIST = \
 	autogen.sh \
 	hdhomerun_config.glade \
diff -ur -x .DS_Store -x '*.orig' hdhomerun_config_gui/src/Makefile.in hdhomerun_config_gui/src/Makefile.in
--- hdhomerun_config_gui/src/Makefile.in	2013-03-23 13:22:13.000000000 -0400
+++ hdhomerun_config_gui/src/Makefile.in	2013-12-01 20:37:55.000000000 -0500
@@ -181,7 +181,6 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 INCLUDES = \
-	-I../../libhdhomerun \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
 	@PACKAGE_CFLAGS@
@@ -197,7 +196,7 @@
 	HDHRConfig.cpp HDHRConfig.h
 
 hdhomerun_config_gui_LDADD = \
-	-L../../libhdhomerun -lhdhomerun \
+	-lhdhomerun \
 	@PACKAGE_LIBS@ $(INTLLIBS)
 
 all: all-am
@@ -421,7 +420,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-binPROGRAMS install-exec-local
+install-exec-am: install-binPROGRAMS
 
 install-html: install-html-am
 
@@ -469,7 +468,7 @@
 	distclean-generic distclean-tags distdir dvi dvi-am html \
 	html-am info info-am install install-am install-binPROGRAMS \
 	install-data install-data-am install-dvi install-dvi-am \
-	install-exec install-exec-am install-exec-local install-html \
+	install-exec install-exec-am install-html \
 	install-html-am install-info install-info-am install-man \
 	install-pdf install-pdf-am install-ps install-ps-am \
 	install-strip installcheck installcheck-am installdirs \
@@ -479,14 +478,6 @@
 	uninstall-local
 
 
-install-exec-local:
-	test -z "$(libdir)" || $(MKDIR_P) "$(DESTDIR)$(libdir)"
-	$(INSTALL) ../../libhdhomerun/libhdhomerun.so $(DESTDIR)$(libdir)
-	$(INSTALL) ../../libhdhomerun/hdhomerun_config $(DESTDIR)$(bindir)
-
-uninstall-local:
-	-rm -f "$(DESTDIR)$(libdir)/libhdhomerun.so"
-	-rm -f "$(DESTDIR)$(bindir)/hdhomerun_config" 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
