--- source/dialogs/filebrowser.c.orig   2022-07-27 14:12:35.000000000 +0800
+++ source/dialogs/filebrowser.c        2022-07-27 14:12:53.000000000 +0800
@@ -50,6 +50,12 @@

 #define FILEBROWSER_CACHE_FILE "rofi3.filebrowsercache"

+#if defined(__APPLE__) || defined(__NetBSD__)
+#define st_atim st_atimespec
+#define st_ctim st_ctimespec
+#define st_mtim st_mtimespec
+#endif
+
 /**
  * The internal data structure holding the private data of the TEST Mode.
  */
