From 977449987412a3d5c59a036dbab8b6d67764bb3e Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Mon, 29 Sep 2025 07:49:28 +0200
Subject: [PATCH] Silent the "not representable as an int" warning

---
 tests/amqpconnection_validation.phpt | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/tests/amqpconnection_validation.phpt b/tests/amqpconnection_validation.phpt
index 9f67cf67..bb1790c0 100644
--- a/tests/amqpconnection_validation.phpt
+++ b/tests/amqpconnection_validation.phpt
@@ -26,7 +26,12 @@ foreach ($parameters as $args) {
     list($prop, $setter, $getter, $values) = $args;
     foreach ($values as $value) {
         try {
-            $con1 = new AMQPConnection([$prop => $value]);
+            if (in_array($prop, ['frame_max', 'heartbeat'])) {
+                // Silent the "not representable as an int" warning
+                $con1 = @new AMQPConnection([$prop => $value]);
+            } else {
+                $con1 = new AMQPConnection([$prop => $value]);
+            }
             echo $getter . " after constructor: ";
             echo $con1->{$getter}();
             echo PHP_EOL;
@@ -109,4 +114,4 @@ AMQPConnectionException: Parameter 'heartbeat' is out of range.
 AMQPConnectionException: Parameter 'heartbeat' is out of range.
 getHeartbeatInterval after constructor: 250
 getHeartbeatInterval after constructor: 0
-==DONE==
\ No newline at end of file
+==DONE==
