--- ./shared/qcommon/q_platform.h.orig	2017-12-16 10:27:37.000000000 -0800
+++ ./shared/qcommon/q_platform.h	2017-12-16 10:30:26.000000000 -0800
@@ -212,7 +212,7 @@
 
 	// gcc versions < 4.9 did not add max_align_t to the std:: namespace, but instead
 	// put it in the global namespace. Need this to provide uniform access to max_align_t
-	#if defined(__GNUC__) && ((__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ < 9))
+	#if 0
 		typedef max_align_t qmax_align_t;
 	#else
 		typedef std::max_align_t qmax_align_t;
