--- ext/openssl/extconf.rb.orig	2021-11-24 20:12:15.000000000 +0900
+++ ext/openssl/extconf.rb	2021-11-25 22:46:16.000000000 +0900
@@ -90,7 +90,9 @@
 end
 
 Logging::message "=== Checking for required stuff... ===\n"
-pkg_config_found = pkg_config("openssl") && have_header("openssl/ssl.h")
+# mkmf#pkg_config() looks to ignore macports' build.env PKG_CONFIG_PATH
+# pkg_config_found = pkg_config("openssl") && have_header("openssl/ssl.h")
+pkg_config_found  = false
 
 if !pkg_config_found && !find_openssl_library
   Logging::message "=== Checking for required stuff failed. ===\n"
