diff --git Plugins/SLACTools/pqSLACDataLoadManager.cxx Plugins/SLACTools/pqSLACDataLoadManager.cxx
index 3ba019c26b..ce38396f68 100644
--- Plugins/SLACTools/pqSLACDataLoadManager.cxx
+++ Plugins/SLACTools/pqSLACDataLoadManager.cxx
@@ -34,6 +34,7 @@
 #include "vtkSMProperty.h"
 #include "vtkSMSourceProxy.h"
 
+#include <QAction>
 #include <QPushButton>
 #include <QtDebug>
 
diff --git Plugins/StreamLinesRepresentation/CMakeLists.txt Plugins/StreamLinesRepresentation/CMakeLists.txt
index 5d656a54b8..86c24964fe 100644
--- Plugins/StreamLinesRepresentation/CMakeLists.txt
+++ Plugins/StreamLinesRepresentation/CMakeLists.txt
@@ -41,7 +41,6 @@ encode_files_as_strings(ENCODED_STRING_FILES
 
 add_paraview_plugin(
   StreamLinesRepresentation "0.1"
-  DOCUMENTATION_DIR "${CMAKE_CURRENT_SOURCE_DIR}/doc"
   SERVER_MANAGER_XML StreamLinesRepresentation.xml
   SERVER_MANAGER_SOURCES
     vtkStreamLinesRepresentation.cxx
diff --git Qt/ApplicationComponents/pqColorMapEditor.cxx Qt/ApplicationComponents/pqColorMapEditor.cxx
index 0395185e0a..88b56745eb 100644
--- Qt/ApplicationComponents/pqColorMapEditor.cxx
+++ Qt/ApplicationComponents/pqColorMapEditor.cxx
@@ -59,6 +59,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QDebug>
 #include <QKeyEvent>
 #include <QPointer>
+#include <QStyle>
 #include <QVBoxLayout>
 
 class pqColorMapEditor::pqInternals
diff --git Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
index 908e4598e0..820361a2b3 100644
--- Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
+++ Qt/ApplicationComponents/pqDoubleRangeSliderPropertyWidget.cxx
@@ -43,6 +43,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "vtkSMUncheckedPropertyHelper.h"
 
 #include <QGridLayout>
+#include <QStyle>
 
 class pqDoubleRangeSliderPropertyWidget::pqInternals
 {
diff --git Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
index c297dc3d9e..501633f63a 100644
--- Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
+++ Qt/ApplicationComponents/pqStandardViewFrameActionsImplementation.cxx
@@ -69,6 +69,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QPushButton>
 #include <QSet>
 #include <QShortcut>
+#include <QStyle>
 
 //-----------------------------------------------------------------------------
 pqStandardViewFrameActionsImplementation::pqStandardViewFrameActionsImplementation(
diff --git Qt/ApplicationComponents/pqTimeInspectorWidget.cxx Qt/ApplicationComponents/pqTimeInspectorWidget.cxx
index 6774c2e48e..3430f70376 100644
--- Qt/ApplicationComponents/pqTimeInspectorWidget.cxx
+++ Qt/ApplicationComponents/pqTimeInspectorWidget.cxx
@@ -51,6 +51,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <QLineF>
 #include <QPainter>
+#include <QStyle>
 #include <QVariant>
 #include <QtDebug>
 
diff --git Qt/ApplicationComponents/pqTransferFunctionWidgetPropertyDialog.cxx Qt/ApplicationComponents/pqTransferFunctionWidgetPropertyDialog.cxx
index 55d3146447..225879cd4c 100644
--- Qt/ApplicationComponents/pqTransferFunctionWidgetPropertyDialog.cxx
+++ Qt/ApplicationComponents/pqTransferFunctionWidgetPropertyDialog.cxx
@@ -34,6 +34,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "pqTransferFunctionWidgetPropertyWidget.h"
 #include "vtkPiecewiseFunction.h"
+#include <qvalidator>
 #include <QString>
 
 class pqTransferFunctionWidgetPropertyDialog::pqInternals
diff --git Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
index 6d2865431e..8d2c4b61cd 100644
--- Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
+++ Qt/ApplicationComponents/pqViewResolutionPropertyWidget.cxx
@@ -39,6 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "vtkSMProxy.h"
 
 #include <QIntValidator>
+#include <QStyle>
 
 class pqViewResolutionPropertyWidget::pqInternals
 {
diff --git Qt/Widgets/pqAnimationWidget.h Qt/Widgets/pqAnimationWidget.h
index 283493a276..1c9a678cfe 100644
--- Qt/Widgets/pqAnimationWidget.h
+++ Qt/Widgets/pqAnimationWidget.h
@@ -36,10 +36,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "pqWidgetsModule.h"
 
 #include <QAbstractScrollArea>
+#include <QHeaderView>
 #include <QStandardItemModel>
 
 class QGraphicsView;
-class QHeaderView;
 class pqAnimationModel;
 class pqAnimationTrack;
 
diff --git Qt/Widgets/pqFlatTreeView.h Qt/Widgets/pqFlatTreeView.h
index a0ba740dbc..e44e0d43a5 100644
--- Qt/Widgets/pqFlatTreeView.h
+++ Qt/Widgets/pqFlatTreeView.h
@@ -39,6 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define _pqFlatTreeView_h
 
 #include "pqWidgetsModule.h"
+#include <QHeaderView>
 #include <QAbstractScrollArea>
 #include <QModelIndex>          // Needed for return type
 #include <QStyleOptionViewItem> // Needed for return type
@@ -50,7 +51,6 @@ class pqFlatTreeViewInternal;
 class QAbstractItemModel;
 class QColor;
 class QFontMetrics;
-class QHeaderView;
 class QItemSelection;
 class QItemSelectionModel;
 class QPoint;
diff --git a/Qt/Python/pqPythonShell.cxx b/Qt/Python/pqPythonShell.cxx
index 3d01fd0a96..7cb356fbc4 100644
--- Qt/Python/pqPythonShell.cxx
+++ Qt/Python/pqPythonShell.cxx
@@ -61,6 +61,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QTextCharFormat>
 #include <QVBoxLayout>
 #include <QtDebug>
+#include <QtWidgets>
 
 QStringList pqPythonShell::Preamble;
 
