--- ThirdParty/IceT/vtkicet/CMakeLists.txt.orig	2016-06-01 20:06:22.000000000 -0400
+++ ThirdParty/IceT/vtkicet/CMakeLists.txt	2016-06-01 20:06:30.000000000 -0400
@@ -26,7 +26,7 @@
 # build and install targets to point to dependent libraries that are not
 # dependent on DYLD_LIBRARY_PATH. This variable turns that behavior on by
 # default and also suppresses CMake policy warning 0042.
-SET(CMAKE_MACOSX_RPATH ON)
+#SET(CMAKE_MACOSX_RPATH ON)
 
 # Set output paths.
 SET(LIBRARY_OUTPUT_PATH ${ICET_BINARY_DIR}/lib CACHE PATH
