diff --git a/kde-modules/KDEInstallDirs.cmake b/kde-modules/KDEInstallDirs.cmake
index d739cf1..cafb7ad 100644
--- a/kde-modules/KDEInstallDirs.cmake
+++ b/kde-modules/KDEInstallDirs.cmake
@@ -18,7 +18,7 @@
 # deprecated variable name in square brackets):
 #
 # ``BUNDLEDIR``
-#     application bundles (``/Applications/KDE``) [``BUNDLE_INSTALL_DIR``]
+#     application bundles (``/Applications/MacPorts/KF5``) [``BUNDLE_INSTALL_DIR``]
 # ``EXECROOTDIR``
 #     executables and libraries (``<empty>``) [``EXEC_INSTALL_PREFIX``]
 # ``BINDIR``
@@ -409,7 +409,7 @@ macro(_define_non_cache varname value)
 endmacro()
 
 if(APPLE)
-    _define_absolute(BUNDLEDIR "/Applications/KDE"
+    _define_absolute(BUNDLEDIR "/Applications/MacPorts/KF5"
         "application bundles"
         BUNDLE_INSTALL_DIR)
 endif()
