diff --git cmake/FindSphinx.cmake cmake/FindSphinx.cmake
index 7d489b6..bd8d2d1 100644
--- cmake/FindSphinx.cmake
+++ cmake/FindSphinx.cmake
@@ -56,6 +56,9 @@ find_program(Sphinx_BUILD_EXECUTABLE
         sphinx-build
         sphinx-build2
         sphinx-build3
+        sphinx-build-2.7
+        sphinx-build-3.4
+        sphinx-build-3.5
     DOC "Sphinx Documentation Builder (https://www.sphinx-doc.org/)"
 )
 
