Index: kde-modules/KDECompilerSettings.cmake
===================================================================
--- kde-modules/KDECompilerSettings.cmake
+++ kde-modules/KDECompilerSettings.cmake
@@ -304,9 +304,9 @@
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions -DQT_NO_EXCEPTIONS")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" AND NOT WIN32)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions -DQT_NO_EXCEPTIONS")
 #elseif (MSVC OR (WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "Intel"))
     # Exceptions appear to be disabled by default for MSVC
     # http://msdn.microsoft.com/en-us/library/1deeycx5.aspx
@@ -324,9 +324,13 @@
             set(${VAR} "${${VAR}} -fexceptions")
         endif()
     elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-        set(${VAR} "${${VAR}} -fexceptions")
+        string(REPLACE " -DQT_NO_EXCEPTIONS " " " ${VAR} " ${${VAR}} ")
+        # it's likely that -DQT_NO_EXCEPTIONS from CMAKE_CXX_FLAGS never made it to the COMPILE_OPTIONS
+        # so we override/undo that global setting for this file
+        set(${VAR} "${${VAR}} -fexceptions -UQT_NO_EXCEPTIONS")
     elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-        set(${VAR} "${${VAR}} -fexceptions")
+        string(REPLACE " -DQT_NO_EXCEPTIONS " " " ${VAR} " ${${VAR}} ")
+        set(${VAR} "${${VAR}} -fexceptions -UQT_NO_EXCEPTIONS")
     endif()
     string(STRIP "${${VAR}}" ${VAR})
 endmacro()
@@ -351,10 +355,10 @@
     if (WIN32)
         target_compile_options(${target} ${mode} "$<$<CXX_COMPILER_ID:Intel>:-EHsc>")
     else()
-        target_compile_options(${target} ${mode} "$<$<CXX_COMPILER_ID:Intel>:-fexceptions>")
+        target_compile_options(${target} ${mode} "$<$<CXX_COMPILER_ID:Intel>:-fexceptions;-UQT_NO_EXCEPTIONS>")
     endif()
     target_compile_options(${target} ${mode}
-        "$<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:-fexceptions>")
+        "$<$<OR:$<CXX_COMPILER_ID:GNU>,$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:-fexceptions;-UQT_NO_EXCEPTIONS>")
 endfunction()
 
 function(KDE_ENABLE_EXCEPTIONS)
