--- a/kde-modules/KDEInstallDirs.cmake.orig	2022-01-27 06:16:22.000000000 -0600
+++ b/kde-modules/KDEInstallDirs.cmake	2022-01-27 06:19:09.000000000 -0600
@@ -18,7 +18,7 @@
 # deprecated variable name in square brackets):
 #
 # ``BUNDLEDIR``
-#     application bundles (``/Applications/KDE``) [``BUNDLE_INSTALL_DIR``]
+#     application bundles (``/@APPLICATIONSDIR@/KDE``) [``BUNDLE_INSTALL_DIR``]
 # ``EXECROOTDIR``
 #     executables and libraries (``<empty>``) [``EXEC_INSTALL_PREFIX``]
 # ``BINDIR``
@@ -409,7 +409,7 @@
 endmacro()
 
 if(APPLE)
-    _define_absolute(BUNDLEDIR "/Applications/KDE"
+    _define_absolute(BUNDLEDIR "/@APPLICATIONSDIR@/KF5"
         "application bundles"
         BUNDLE_INSTALL_DIR)
 endif()
