--- CMakeLists.txt.old	2025-01-07 14:49:30
+++ CMakeLists.txt	2025-01-07 14:49:39
@@ -54,7 +54,7 @@
 # compiler options:
 
 if(APPLE)
-    target_compile_options( astyle PRIVATE -W -Wall -fno-rtti -fno-exceptions -stdlib=libc++)
+    target_compile_options( astyle PRIVATE -W -Wall -fno-rtti -fno-exceptions)
 elseif(NOT WIN32)   # Linux
     target_compile_options(astyle PRIVATE -Wall -Wextra -fno-rtti -fno-exceptions)
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "gnu")
