--- CMakeLists.txt.old	2024-11-02 13:23:29
+++ CMakeLists.txt	2024-11-02 13:23:39
@@ -52,7 +52,7 @@
 # compiler options:
 
 if(APPLE)
-    target_compile_options( astyle PRIVATE -W -Wall -fno-rtti -fno-exceptions -std=c++17 -stdlib=libc++)
+    target_compile_options( astyle PRIVATE -W -Wall -fno-rtti -fno-exceptions -std=c++17)
 elseif(NOT WIN32)   # Linux
     target_compile_options(astyle PRIVATE -Wall -Wextra -fno-rtti -fno-exceptions -std=c++17)
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "gnu")
