--- DESCRIPTION	2024-05-06 05:20:15.000000000 +0800
+++ DESCRIPTION	2024-05-06 07:59:59.000000000 +0800
@@ -63,7 +63,7 @@
 BugReports: https://github.com/m-Py/anticlust/issues
 Depends: R (>= 3.6.0)
 Imports: Matrix, RANN (>= 2.6.0)
-Suggests: knitr, palmerpenguins, Rglpk, rmarkdown, Rsymphony, tinytest
+Suggests: knitr, palmerpenguins, Rglpk, rmarkdown, tinytest
 VignetteBuilder: knitr, rmarkdown
 Encoding: UTF-8
 LazyData: true
