--- attica/kdeplugin/kdeplatformdependent.h.orig	2010-06-07 09:09:48.000000000 -0400
+++ attica/kdeplugin/kdeplatformdependent.h	2010-06-07 09:14:24.000000000 -0400
@@ -47,6 +47,10 @@
     KdePlatformDependent();
     virtual ~KdePlatformDependent();
     virtual QList<QUrl> getDefaultProviderFiles() const;
+    virtual void addDefaultProviderFile(const QUrl& url){} 
+    virtual void removeDefaultProviderFile(const QUrl& url){} 
+    virtual void enableProvider(const QUrl& baseUrl, bool enabled) const{} 
+    virtual bool isEnabled(const QUrl& baseUrl) const{return true;} 
     virtual QNetworkReply* post(const QNetworkRequest& request, const QByteArray& data);
     virtual QNetworkReply* post(const QNetworkRequest& request, QIODevice* data);
     virtual QNetworkReply* get(const QNetworkRequest& request);
