--- frmts/netcdf/netcdfdataset.cpp.orig	2017-08-01 17:04:53.000000000 +0200
+++ frmts/netcdf/netcdfdataset.cpp	2017-08-01 17:06:03.000000000 +0200
@@ -4582,7 +4582,7 @@
 /*                                rint()                                */
 /************************************************************************/
 
-#ifdef HAVE_CXX11
+#ifdef NOT_HAVE_CXX11
 double netCDFDataset::rint( double dfX )
 {
     return std::round(dfX);
