--- patch-swig_python_GNUmakefile.orig	2009-10-08 13:45:07.000000000 +0200
+++ patch-swig_python_GNUmakefile	2009-10-08 11:59:40.000000000 +0200
@@ -10,7 +10,7 @@
 -else
 -	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
 -endif
-+	$(PYTHON) setup.py install --root=$(DESTDIR) --prefix=$(prefix)
++	$(PYTHON) setup.py install --root=$(DESTDIR) --prefix=XXX
  
  	for f in $(SCRIPTS) ; do $(INSTALL) ./scripts/$$f $(DESTDIR)$(INST_BIN) ; done
  	
