--- ogr/ogr_p.h.old	2015-07-19 12:42:55.000000000 +0200
+++ ogr/ogr_p.h	2015-07-19 12:26:52.000000000 +0200
@@ -31,6 +31,8 @@
 #ifndef OGR_P_H_INCLUDED
 #define OGR_P_H_INCLUDED
 
+#include "gdal.h"
+
 /* -------------------------------------------------------------------- */
 /*      Include the common portability library ... lets us do lots      */
 /*      of stuff easily.                                                */
