--- gramps/gen/utils/resourcepath.py.orig	2014-06-15 04:32:07.000000000 -0700
+++ gramps/gen/utils/resourcepath.py	2014-07-09 14:17:21.000000000 -0700
@@ -63,7 +63,7 @@
             try:
                 with io.open(resource_file, encoding='utf-8',
                                 errors='strict') as fp:
-                    resource_path = conv_to_unicode(fp.readline())
+                    resource_path = conv_to_unicode(fp.readline().rstrip('\n'))
             except UnicodeError as err:
                 LOG.exception("Encoding error while parsing resource path", err)
                 sys.exit(1)
