--- ./spyderlib/plugins/externalconsole.py.orig	2012-07-17 15:26:16.000000000 -0500
+++ ./spyderlib/plugins/externalconsole.py	2012-07-17 15:27:08.000000000 -0500
@@ -339,9 +339,9 @@
         ipython_group = QGroupBox(
                             _("IPython interpreter command line options"))
         ipython_layout = QVBoxLayout()
-        ipython_is_installed = programs.is_module_installed('IPython', '0.1')
+        ipython_is_installed = programs.is_module_installed('IPython', '>=0.1')
         if ipython_is_installed:
-            if programs.is_module_installed('IPython', '0.12'):
+            if programs.is_module_installed('IPython', '>=0.12'):
                 ipython_edit_012 = self.create_lineedit("IPython >=v0.12",
                              'ipython_kernel_options', alignment=Qt.Horizontal)
                 ipython_layout.addWidget(ipython_edit_012)
--- ./spyderlib/spyder.py.orig	2012-07-17 15:28:33.000000000 -0500
+++ ./spyderlib/spyder.py	2012-07-17 15:29:35.000000000 -0500
@@ -27,7 +27,7 @@
 try:
     # Test if IPython v0.11+ is installed
     import IPython
-    if IPython.__version__.startswith(('0.11', '0.12'))\
+    if IPython.__version__.startswith(('0.11', '0.12', '0.13'))\
        and os.environ.get('QT_API', 'pyqt') == 'pyqt':
         # If PyQt is the selected GUI toolkit (at this stage, only the
         # bootstrap script has eventually set this option),
@@ -837,7 +837,7 @@
             self.help_menu_actions.append(web_resources)
             
             # IPython frontend action
-            if is_module_installed('IPython', '0.12'):
+            if is_module_installed('IPython', '>=0.12'):
                 ipf_action = create_action(self, _("New IPython frontend..."),
                                            icon="ipython.png",
                                            triggered=self.new_ipython_frontend)
--- ./spyderlib/widgets/explorer.py.orig	2012-07-17 15:31:38.000000000 -0500
+++ ./spyderlib/widgets/explorer.py	2012-07-17 15:30:25.000000000 -0500
@@ -286,7 +286,7 @@
                                triggered=lambda fnames=fnames:
                                self.open_interpreter(fnames))
         actions.append(action)
-        if programs.is_module_installed('IPython', '0.10'):
+        if programs.is_module_installed('IPython', '>=0.10'):
             _title = _("Open IPython here")
             action = create_action(self, _title, icon="ipython.png",
                                    triggered=lambda fnames=fnames:
