--- spyderlib/widgets/sourcecode/syntaxhighlighters.py.orig	2014-05-01 12:42:23.000000000 -0500
+++ spyderlib/widgets/sourcecode/syntaxhighlighters.py	2014-05-01 12:43:40.000000000 -0500
@@ -265,6 +265,10 @@
             if bold:
                 format.setFontWeight(QFont.Bold)
             format.setFontItalic(italic)
+            if not italic and not bold:
+                font = format.font()
+                font.setStyleStrategy(QFont.NoAntialias)
+                format.setFont(font)
             self.formats[name] = format
 
     def _check_color_scheme(self, color_scheme):
