--- setup.py.orig	2024-03-20 21:23:04
+++ setup.py	2024-03-20 21:23:24
@@ -226,7 +226,7 @@
     'pickleshare>=0.4',
     'psutil>=5.3',
     'pygments>=2.0',
-    'pylint>=2.5.0,<3.1',
+    'pylint>=2.5.0',
     'pylint-venv>=3.0.2',
     'python-lsp-black>=2.0.0,<3.0.0',
     'pyls-spyder>=0.4.0',
@@ -237,7 +237,7 @@
     'pyzmq>=24.0.0',
     'qdarkstyle>=3.2.0,<3.3.0',
     'qstylizer>=0.2.2',
-    'qtawesome>=1.2.1,<1.3.0',
+    'qtawesome>=1.2.1',
     'qtconsole>=5.5.1,<5.6.0',
     'qtpy>=2.1.0',
     'rtree>=0.9.7',
--- spyder/dependencies.py.orig	2024-03-20 21:23:38
+++ spyder/dependencies.py	2024-03-20 21:23:51
@@ -56,7 +56,7 @@
 PICKLESHARE_REQVER = '>=0.4'
 PSUTIL_REQVER = '>=5.3'
 PYGMENTS_REQVER = '>=2.0'
-PYLINT_REQVER = '>=2.5.0,<3.1'
+PYLINT_REQVER = '>=2.5.0'
 PYLINT_VENV_REQVER = '>=3.0.2'
 PYLSP_REQVER = '>=1.10.0,<1.11.0'
 PYLSP_BLACK_REQVER = '>=2.0.0,<3.0.0'
@@ -65,7 +65,7 @@
 PYZMQ_REQVER = '>=24.0.0'
 QDARKSTYLE_REQVER = '>=3.2.0,<3.3.0'
 QSTYLIZER_REQVER = '>=0.2.2'
-QTAWESOME_REQVER = '>=1.2.1,<1.3.0'
+QTAWESOME_REQVER = '>=1.2.1'
 QTCONSOLE_REQVER = '>=5.5.1,<5.6.0'
 QTPY_REQVER = '>=2.1.0'
 RTREE_REQVER = '>=0.9.7'
