--- setup.py.orig	2022-11-26 22:10:07
+++ setup.py	2022-11-26 22:10:37
@@ -209,7 +209,7 @@
     'cookiecutter>=1.6.0',
     'diff-match-patch>=20181111',
     'intervaltree>=3.0.2',
-    'ipython>=7.31.1,<8.0.0',
+    'ipython>=7.31.1',
     'jedi>=0.17.2,<0.19.0',
     'jellyfish>=0.7',
     'jsonschema>=3.2.0',
--- spyder/dependencies.py.orig	2022-11-26 22:11:12
+++ spyder/dependencies.py	2022-11-26 22:11:20
@@ -38,7 +38,7 @@
 # None for pynsist install for now
 # (check way to add dist.info/egg.info from packages without wheels available)
 INTERVALTREE_REQVER = None if is_pynsist() else '>=3.0.2'
-IPYTHON_REQVER = ">=7.31.1;<8.0.0"
+IPYTHON_REQVER = ">=7.31.1"
 JEDI_REQVER = '>=0.17.2;<0.19.0'
 JELLYFISH_REQVER = '>=0.7'
 JSONSCHEMA_REQVER = '>=3.2.0'
