--- spyderlib/utils/programs.py.orig	2011-08-11 14:41:11.000000000 -0500
+++ spyderlib/utils/programs.py	2011-08-11 14:43:54.000000000 -0500
@@ -29,6 +29,12 @@
             else:
                 return True
     else:
+        abspath = "/Applications/MacPorts/Qt4/" + basename + ".app/Contents/MacOS/" + basename
+        if osp.isfile(abspath):
+            if get_path:
+                return abspath
+            else:
+                return True
         return False
     
 def run_program(name, args=[]):
