--- spyderlib/utils/bsdsocket.py.orig	2012-06-18 09:50:37.000000000 -0500
+++ spyderlib/utils/bsdsocket.py	2012-06-18 09:50:57.000000000 -0500
@@ -33,6 +33,9 @@
     dlen, data = None, None
     try:
         datalen = sock.recv(SZ)
+        if len(datalen) < SZ:  # Unable to read required size from socket
+          sock.settimeout(None)
+          return
         dlen, = struct.unpack("l", datalen)
         data = ''
         while len(data) < dlen:
