--- CMakeLists.txt.orig	2026-09-18 20:47:46
+++ CMakeLists.txt	2026-10-09 13:10:05
@@ -26,7 +26,7 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.23)
 
 if(POLICY CMP0091)
   cmake_policy(SET CMP0091 NEW)
@@ -163,12 +163,6 @@
   endif(HAVE_CLANG_MAVX)
 endif(SNAPPY_REQUIRE_AVX2)
 
-# Used by googletest.
-check_cxx_compiler_flag(-Wno-missing-field-initializers
-                        SNAPPY_HAVE_NO_MISSING_FIELD_INITIALIZERS)
-check_cxx_compiler_flag(-Wno-implicit-int-float-conversion
-                        SNAPPY_HAVE_NO_IMPLICIT_INT_FLOAT_CONVERSION)
-
 include(CheckCXXSourceCompiles)
 check_cxx_source_compiles("
 int main() {
@@ -381,34 +375,15 @@
 if(SNAPPY_BUILD_TESTS)
   enable_testing()
 
-  # Prevent overriding the parent project's compiler/linker settings on Windows.
-  set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-  set(install_gtest OFF)
-  set(install_gmock OFF)
-  set(build_gmock ON)
-
   # This project is tested using GoogleTest.
-  add_subdirectory("third_party/googletest")
-
-  # GoogleTest triggers a missing field initializers warning.
-  if(SNAPPY_HAVE_NO_MISSING_FIELD_INITIALIZERS)
-    set_property(TARGET gtest
-        APPEND PROPERTY COMPILE_OPTIONS -Wno-missing-field-initializers)
-    set_property(TARGET gmock
-        APPEND PROPERTY COMPILE_OPTIONS -Wno-missing-field-initializers)
-  endif(SNAPPY_HAVE_NO_MISSING_FIELD_INITIALIZERS)
+  find_package(GTest REQUIRED)
 
-  if(SNAPPY_HAVE_NO_IMPLICIT_INT_FLOAT_CONVERSION)
-    set_property(TARGET gtest
-        APPEND PROPERTY COMPILE_OPTIONS -Wno-implicit-int-float-conversion)
-  endif(SNAPPY_HAVE_NO_IMPLICIT_INT_FLOAT_CONVERSION)
-
   add_executable(snappy_unittest "")
   target_sources(snappy_unittest
     PRIVATE
       "snappy_unittest.cc"
   )
-  target_link_libraries(snappy_unittest snappy_test_support gmock_main gtest)
+  target_link_libraries(snappy_unittest snappy_test_support GTest::gmock_main GTest::gtest)
 
   add_test(
     NAME snappy_unittest
