--- a/Makefile	2012-10-27 12:59:04.000000000 -0700
+++ b/Makefile	2012-10-27 13:00:48.000000000 -0700
@@ -69,9 +69,8 @@ endif
 ifeq ($(MAKECMDGOALS),install-clang)
   DIRS := tools/clang/tools/driver tools/clang/lib/Headers \
           tools/clang/tools/libclang \
-          tools/clang/include/clang-c \
-          tools/clang/runtime tools/clang/docs \
-          tools/lto runtime
+          tools/clang/include tools/clang/lib \
+          tools/clang/runtime tools/clang/docs
   ifneq ($(BUILD_CLANG_ONLY),YES)
     DIRS += tools/clang/tools/c-index-test
   endif
