--- build.sh 2007-12-11 08:26:12.000000000 -0600
+++ build.sh 2008-07-25 18:35:33.000000000 -0500
@@ -293,7 +293,6 @@
   ;;
  build*)    echo ;;    #default - nothing to do here, this is the default.
  install)   
-            [ -z "$CYGWIN" ] && [ "`whoami`" != "root" ] && echo "Need to be root to install. try sudo ./build.sh $@" && exit 1
             INSTALL=1;
             ;;
 
@@ -849,15 +848,6 @@
 cd ../bin
 echo -n " "
 
-# Report size and hashes ####
-
-if [ -z "$DARWIN" ]
-then
-    SIZE="`du -sh lisaem 2>/dev/null`"
-else
-    SIZE="`du -sh LisaEm.app 2>/dev/null`"
-fi
-
 
 if [ -n "$DARWIN" ]
 then
@@ -895,15 +885,25 @@
     then
       cd ../bin/
       echo Installing LisaEm.app
-      tar cf - ./LisaEm.app | (cd $PREFIX; tar xf -)
-      mkdir -pm755 /usr/local/bin
-      echo Installing lisafsh-tool and lisadiskinfo to /usr/local/bin
+      tar cf - ./LisaEm.app | (cd $PREFIXAPP; tar xf -)
+      mkdir -pm755 $PREFIXBIN
+      echo Installing lisafsh-tool and lisadiskinfo to $PREFIXBIN
       chmod 755 lisafsh-tool lisadiskinfo
-      cp lisafsh-tool lisadiskinfo /usr/local/bin
+      cp lisafsh-tool lisadiskinfo $PREFIXBIN
       echo Done Installing.
     fi
 fi
 
+
+# Report size and hashes ####
+
+if [ -z "$DARWIN" ]
+then
+    SIZE="`du -sh lisaem 2>/dev/null`"
+else
+    SIZE="`du -sh LisaEm.app 2>/dev/null`"
+fi
+
 # some older OS's don't support du -sh, so fall back to du -sk and convert to MB's
 if [ -z "$SIZE" ]
 then
