--- CMakeLists.txt.orig	2019-09-17 09:54:17.000000000 +0800
+++ CMakeLists.txt	2019-09-17 11:50:53.000000000 +0800
@@ -292,7 +292,7 @@
         ${OPENSSL_ROOT}/lib/libssl.lib
         ${OPENSSL_ROOT}/lib/libcrypto.lib
     )
-elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+elseif (FALSE)
     #Try use 1.1 for the latest features. otherwise use the default
     IF(EXISTS /usr/local/opt/openssl@1.1)
         set (OPENSSL_ROOT /usr/local/opt/openssl@1.1)
@@ -304,7 +304,7 @@
         ${OPENSSL_ROOT}/lib/libssl.a
         ${OPENSSL_ROOT}/lib/libcrypto.a
     )
-elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     set (OPENSSL_LIBS ssl crypto)
 else()
     message (FATAL_ERROR "Couldn't find OpenSSL")
