--- bkb.orig	2006-12-29 14:50:11.000000000 -0800
+++ bkb	2006-12-29 14:50:22.000000000 -0800
@@ -31,7 +31,7 @@ bkb <command> [args] or -h ;# with comma
 	tsec yes		;# wait fractional seconds, then abort
 	unasn foo.der		;# translate asn1/ber-der into a script
 	nint 10			;# write a binary integer
-	no < `randombin`	;# NUL->0xFF, or something else
+	no < \`randombin\`	;# NUL->0xFF, or something else
 	crnl < text		;# ensure CRNL on all lines
 	echo "a%20b" | dehexify ;# HTTP-style dehexify
 	echo "a b" | httphexify ;# HTTP-style hex encoding
