--- maturin/__init__.py	2023-04-06 02:38:20
+++ maturin/__init__.py	2023-05-03 00:03:28
@@ -52,7 +52,7 @@
     # PEP 517 specifies that only `sys.executable` points to the correct
     # python interpreter
     base_command = [
-        "maturin",
+        f"maturin-{sys.version_info.major}.{sys.version_info.minor}",
         "pep517",
         "build-wheel",
         "-i",
@@ -92,7 +92,7 @@
 
 # noinspection PyUnusedLocal
 def build_sdist(sdist_directory: str, config_settings: Optional[Mapping[str, Any]] = None) -> str:
-    command = ["maturin", "pep517", "write-sdist", "--sdist-directory", sdist_directory]
+    command = [f"maturin-{sys.version_info.major}.{sys.version_info.minor}", "pep517", "write-sdist", "--sdist-directory", sdist_directory]
 
     print("Running `{}`".format(" ".join(command)))
     sys.stdout.flush()
@@ -154,7 +154,7 @@
         sys.exit(1)
 
     command = [
-        "maturin",
+        f"maturin-{sys.version_info.major}.{sys.version_info.minor}",
         "pep517",
         "write-dist-info",
         "--metadata-directory",
