--- src/bindings/python/CMakeLists.txt.orig	2020-03-03 03:00:06.000000000 -0500
+++ src/bindings/python/CMakeLists.txt	2021-01-17 11:31:26.000000000 -0500
@@ -77,23 +77,7 @@
 	target_link_libraries(${SWIG_MODULE_pyopenshot_REAL_NAME}
                         PUBLIC ${PYTHON_LIBRARIES} openshot)
 
-	### Check if the following Debian-friendly python module path exists
-	SET(PYTHON_MODULE_PATH "${CMAKE_INSTALL_PREFIX}/lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/dist-packages")
-	if (NOT EXISTS ${PYTHON_MODULE_PATH})
-
-		### Calculate the python module path (using distutils)
-		execute_process ( COMMAND ${PYTHON_EXECUTABLE} -c "\
-from distutils.sysconfig import get_python_lib; \
-print( get_python_lib( plat_specific=True, prefix='${CMAKE_INSTALL_PREFIX}' ) )"
-			OUTPUT_VARIABLE _ABS_PYTHON_MODULE_PATH
-			OUTPUT_STRIP_TRAILING_WHITESPACE )
-
-		GET_FILENAME_COMPONENT(_ABS_PYTHON_MODULE_PATH
-				"${_ABS_PYTHON_MODULE_PATH}" ABSOLUTE)
-		FILE(RELATIVE_PATH _REL_PYTHON_MODULE_PATH
-				${CMAKE_INSTALL_PREFIX} ${_ABS_PYTHON_MODULE_PATH})
-		SET(PYTHON_MODULE_PATH ${_ABS_PYTHON_MODULE_PATH})
-	endif()
+	SET(PYTHON_MODULE_PATH "@@PyModPath@@")
 	message("PYTHON_MODULE_PATH: ${PYTHON_MODULE_PATH}")
 
 	############### INSTALL HEADERS & LIBRARY ################
