--- lib/libpe/Makefile.orig	2024-02-01 07:39:11
+++ lib/libpe/Makefile	2024-05-17 10:24:36
@@ -106,7 +106,7 @@
 	$(LINK) -shared -Wl,-soname,$(LIBNAME).so.1 $(LDFLAGS) -o $(LIBNAME).so $^ $(LIBS)
 else ifeq ($(PLATFORM_OS), Darwin)
 	$(LINK) -headerpad_max_install_names -dynamiclib \
-		-flat_namespace -install_name $(LIBNAME).$(VERSION).dylib \
+		-flat_namespace -install_name $(libdir)/$(LIBNAME).$(VERSION).dylib \
 		-current_version $(VERSION) -compatibility_version $(VERSION) \
 		$(LDFLAGS) -o $(LIBNAME).dylib $^ $(LIBS)
 else ifeq ($(PLATFORM_OS), CYGWIN)
