--- Makefile.in.orig	2017-09-19 12:39:34.000000000 -0400
+++ Makefile.in	2017-09-19 12:40:51.000000000 -0400
@@ -113,8 +113,8 @@
 		@for lib in lib${PACKAGE}.a lib${PACKAGE}${SHLIB_SUFFIX} \
 				${CFITSIO_SHLIB} ${CFITSIO_SHLIB_SONAME}; do \
 		    if [ -f $$lib ]; then \
-			echo "/bin/cp -a $$lib ${CFITSIO_LIB}"; \
-			/bin/cp -a $$lib ${CFITSIO_LIB}; \
+			echo "/bin/cp -pPR $$lib ${CFITSIO_LIB}"; \
+			/bin/cp -pPR $$lib ${CFITSIO_LIB}; \
 		    fi; \
 		done
 		/bin/cp fitsio.h fitsio2.h longnam.h drvrsmem.h ${CFITSIO_INCLUDE}
