--- configure	2010-01-08 12:03:29.000000000 -0600
+++ configure	2010-01-09 07:36:07.000000000 -0600
@@ -3998,7 +3998,7 @@
 
 case $GCC in
 	yes)
-		GCCVERSION="`gcc -dumpversion 2>&1`"
+		GCCVERSION="`$CC -dumpversion 2>&1`"
 		echo "cfitsio: == Using gcc version $GCCVERSION"
 
 
@@ -4049,7 +4049,7 @@
     SHLIB_SUFFIX=".dll"
     ;;
   darwin)
-    SHLIB_LD="$CC -dynamiclib"
+    SHLIB_LD="$CC -dynamiclib -install_name @@MPORTS_PREFIX@@/lib/libcfitsio.dylib"
     SHLIB_SUFFIX=".dylib"
     lhea_shlib_cflags="-fPIC -fno-common"
     ;;
