--- configure.orig	2016-04-29 08:11:26.000000000 -0500
+++ configure	2016-07-11 03:45:18.000000000 -0700
@@ -4486,10 +4486,10 @@
         # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
 
         case $host in
-            *darwin[56789]*)
-                ;;
             *)
-
+            ;;
+            idontthinkso)
+            
                 # Test to see whether the C compiler accepts the "-arch"
                 # flags for building "universal" binaries (Apple XCode only):
                 SAVE_CFLAGS="$CFLAGS"
@@ -4723,10 +4723,10 @@
     CFITSIO_SHLIB="lib\${PACKAGE}.\${CFITSIO_SONAME}.\${CFITSIO_MAJOR}.\${CFITSIO_MINOR}\${SHLIB_SUFFIX}"
     CFITSIO_SHLIB_SONAME="lib\${PACKAGE}.\${CFITSIO_SONAME}\${SHLIB_SUFFIX}"
     case $host in
-        *darwin[56789]*)
-            SHLIB_LD="$CC -dynamiclib -install_name lib\${PACKAGE}.\${CFITSIO_SONAME}\${SHLIB_SUFFIX} -compatibility_version \${CFITSIO_SONAME} -current_version \${CFITSIO_SONAME}.\${CFITSIO_MAJOR}.\${CFITSIO_MINOR}"
-            ;;
-        *)
+         *)
+             SHLIB_LD="$CC -dynamiclib -install_name @@MPORTS_PREFIX@@/lib/libcfitsio.dylib -compatibility_version \${CFITSIO_SONAME} -current_version \${CFITSIO_SONAME}.\${CFITSIO_MAJOR}.\${CFITSIO_MINOR}"
+             ;;
+         idontthinkso)
             # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
             SHLIB_LD="$CC -dynamiclib $C_UNIV_SWITCH -headerpad_max_install_names -install_name @rpath/lib\${PACKAGE}.\${CFITSIO_SONAME}\${SHLIB_SUFFIX} -compatibility_version \${CFITSIO_SONAME} -current_version \${CFITSIO_SONAME}.\${CFITSIO_MAJOR}.\${CFITSIO_MINOR}"
             ;;
