--- configure.orig	2011-05-13 12:58:22.000000000 -0500
+++ configure	2011-06-18 04:10:13.000000000 -0500
@@ -3600,9 +3600,9 @@
         # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
 
         case $system in
-            Darwin-[56789]*)
-                ;;
             *)
+                ;;
+            idontthinkso)
 
                 # Test to see whether the C compiler accepts the "-arch"
                 # flags for building "universal" binaries (Apple XCode only):
@@ -3823,10 +3823,10 @@
   darwin)
 
     case $system in
-        Darwin-[56789]*)
-            SHLIB_LD="$CC -dynamiclib"
-            ;;
         *)
+            SHLIB_LD="$CC -dynamiclib -install_name @@MPORTS_PREFIX@@/lib/libcfitsio.dylib"
+            ;;
+        idontthinkso)
             # Build for i386 & x86_64 architectures on Darwin 10.x or newer:
             SHLIB_LD="$CC -dynamiclib $C_UNIV_SWITCH"
             ;;
