--- src/Makefile.orig	2015-10-30 02:48:27.000000000 -0500
+++ src/Makefile	2016-10-10 11:39:53.000000000 -0500
@@ -25,6 +25,8 @@
 # the developer makefile makefile_control.m
 slicotlibrary.a: slicot.tar.gz
 	tar -xzf slicot.tar.gz
+	sed -i '.orig' 's/DGEGS/DGGES/g' slicot/src/SG03AD.f slicot/src/SG03BD.f
+	sed -i '.orig' 's/DLATZM/DORMRZ/g' slicot/src/AB08NX.f slicot/src/AG08BY.f slicot/src/SB01BY.f slicot/src/SB01FY.f
 	mkdir sltmp
 	mv slicot/src/*.f ./sltmp
 	mv slicot/src_aux/*.f ./sltmp
