From 42c1e6b4305f8bcce4242466acf5d33f07a14734 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 28 Oct 2022 21:50:12 +0800
Subject: [PATCH 2/3] alloca.c: add ifndef

---
 src/runtime/Integer/alloca.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/runtime/Integer/alloca.c b/src/runtime/Integer/alloca.c
index c4dd58e..0705f54 100644
--- a/src/runtime/Integer/alloca.c
+++ b/src/runtime/Integer/alloca.c
@@ -1,9 +1,10 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-
+#ifndef alloca
 void *alloca (unsigned size)
 {
   fprintf(stderr,"Failure alloca %d called:-(\n",size);
   abort();
 }
+#endif
-- 
2.36.1

