# From: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=53964

--- Makefile.inc.orig	2010-07-09 20:40:18.000000000 +0800
+++ Makefile.inc	2022-04-13 07:12:27.000000000 +0800
@@ -1,6 +1,9 @@
 ### Configurable variables:
+# We want C89 semantics plus C++-style comments and things like the
+# 'setjmp_buf' and 'u_short' types.  More importantly, build with '-O1'
+# only to avoid modern optimizations that break the code.
 
-OPT = -O3
+OPT = -O1 -std=gnu89
 #ARCH = -m32
 ARCH = 
 

--- script/nhc98heap.c.orig	2007-03-13 06:03:52.000000000 +0800
+++ script/nhc98heap.c	2022-04-13 07:09:55.000000000 +0800
@@ -1,5 +1,6 @@
 #include <stdlib.h>
 #include <stdio.h>
+#include <ctype.h>
 main(int argc, char **argv)
 {
   int prefix = 1;


--- src/runtime/Kernel/collector.c.orig	2007-03-13 06:03:52.000000000 +0800
+++ src/runtime/Kernel/collector.c	2022-04-13 07:11:12.000000000 +0800
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 #include "node.h"
 /*#include "newmacros.h"  -- already included in node.h */
 /*#include "runtime.h"    -- already included in node.h */


--- script/echo.c.orig	2008-09-04 21:01:02.000000000 +0800
+++ script/echo.c	2022-04-13 09:40:40.000000000 +0800
@@ -1,5 +1,6 @@
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 int main (int argc, char** argv) {
   int i=0;



# From: https://github.com/NetBSD/pkgsrc/blob/trunk/lang/nhc98/patches/patch-ab

--- src/prelude/BinArray/cLowUnboxedArray.c.orig	2007-03-13 06:03:49.000000000 +0800
+++ src/prelude/BinArray/cLowUnboxedArray.c	2022-04-16 17:06:33.000000000 +0800
@@ -1,5 +1,5 @@
-#include "cLowUnboxedArray.h"
 #include <stdlib.h>
+#include "cLowUnboxedArray.h"
 
 void finaliseUBA (UBA uba) {
   free(uba->block);
