--- src/plmap.c.orig	2019-06-01 16:24:33.000000000 -0700
+++ src/plmap.c	2019-08-18 08:42:10.000000000 -0700
@@ -76,6 +76,10 @@
     PLINT i;
     PLFLT *renderX;
     PLFLT *renderY;
+#ifndef PL_DOUBLE
+    PLFLT *fltX;
+    PLFLT *fltY;
+#endif
 
     //we need to do something a bit different with filled polygons. The issue is the poles
     //on lat/lon plots. If we draw Antarctica then we expect it to be filled, but this
@@ -135,7 +139,7 @@
             renderY = newY;
 #else
             fltX = malloc( ( n + nExtraPoints ) * sizeof ( PLFLT ) );
-            fltX = malloc( ( n + nExtraPoints ) * sizeof ( PLFLT ) );
+            fltY = malloc( ( n + nExtraPoints ) * sizeof ( PLFLT ) );
             if ( !fltX || !fltY )
             {
                 free( fltX );
@@ -187,7 +191,7 @@
     renderY = y;
 #else
     fltX = malloc( n * sizeof ( PLFLT ) );
-    fltX = malloc( n * sizeof ( PLFLT ) );
+    fltY = malloc( n * sizeof ( PLFLT ) );
     if ( !fltX || !fltY )
     {
         free( fltX );
