--- examples/ocaml/Makefile.examples.in.orig	2019-06-02 08:24:33.000000000 +0900
+++ examples/ocaml/Makefile.examples.in	2019-08-12 16:02:27.000000000 +0900
@@ -26,7 +26,7 @@
 EXEEXT = @EXEEXT@
 
 PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-install_tree_ocaml_RPATHCMD = @install_tree_ocaml_RPATHCMD@
+install_tree_ocaml_RPATHCMD =
 
 OCAML_CFLAGS = -g -I "@OCAML_INSTALL_DIR@/plplot"
 OCAML_LIBS = $(install_tree_ocaml_RPATHCMD) -ccopt "-L@CMAKE_INSTALL_LIBDIR_ESCAPED@" plplot.@OCAML_EXAMPLE_LIBEXT@ unix.@OCAML_EXAMPLE_LIBEXT@
@@ -52,7 +52,6 @@
 		 x17ocaml$(EXEEXT) \
 		 x18ocaml$(EXEEXT) \
 		 x19ocaml$(EXEEXT) \
-		 x20ocaml$(EXEEXT) \
 		 x21ocaml$(EXEEXT) \
 		 x22ocaml$(EXEEXT) \
 		 x23ocaml$(EXEEXT) \
