From: Stephane Glondu <glondu@debian.org>
Date: Mon, 2 Mar 2026 09:42:46 +0100
Subject: Add 5.4.1 to known OCaml versions

---
 tools/compiler_version.ml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/compiler_version.ml b/tools/compiler_version.ml
index 856f9d1..630af62 100644
--- tools/compiler_version.ml
+++ tools/compiler_version.ml
@@ -88,6 +88,7 @@ let v5_2_0 = mk 5 2 0
 let v5_2_1 = mk 5 2 1
 let v5_3_0 = mk 5 3 0
 let v5_4_0 = mk 5 4 0
+let v5_4_1 = mk 5 4 1
 
 let known_versions =
 [
@@ -135,6 +136,7 @@ let known_versions =
   v5_2_1;
   v5_3_0;
   v5_4_0;
+  v5_4_1;
 ]
 
 let is_known v = List.mem v known_versions
