--- src/cum_median.cpp.orig	2022-08-06 00:36:16.000000000 +0800
+++ src/cum_median.cpp	2023-04-20 02:02:30.000000000 +0800
@@ -7,7 +7,7 @@
   PiecewiseFunction function;
   for(int data_i=0; data_i < n_data; data_i++){
     double data_value = data_vec[data_i];
-    if(!isfinite(data_value)){
+    if(!std::isfinite(data_value)){
       return ERROR_CUM_MEDIAN_DATA_NOT_FINITE;
     }
     double weight_value = weight_vec[data_i];
