--- makefile.orig	2007-09-06 13:10:24.000000000 +0200
+++ makefile	2007-09-06 13:11:55.000000000 +0200
@@ -89,13 +89,13 @@
 endif
 
 .cxx.o:
-	$(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $< -o $@
+	$(CC) $(CXXFLAGS) $(CONFIGFLAGS) -c $< -o $@
 
 clean:
 	rm -f *.o $(PROG)
 
 deps:
-	$(CC) -MM $(CONFIGFLAGS) $(CXXFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
+	$(CC) -MM $(CXXFLAGS) $(CONFIGFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak 
 
 # make should be run in ../../scintilla/gtk to compile all the lexers.
 COMPLIB=../../scintilla/bin/scintilla.a
@@ -103,7 +103,7 @@
 $(PROG): SciTEGTK.o FilePath.o SciTEBase.o SciTEBuffers.o SciTEIO.o StringList.o Exporters.o \
 PropSetFile.o MultiplexExtension.o DirectorExtension.o SciTEProps.o Utf8_16.o \
 	JobQueue.o GTKMutex.o $(COMPLIB) $(LUA_OBJS)
-	$(CC) `$(CONFIGTHREADS)` -rdynamic -Wl,--version-script lua.vers -DGTK $^ -o $@ $(CONFIGLIB)
+	$(CC) `$(CONFIGTHREADS)` $(UNIVERSAL_LDFLAGS) -DGTK $^ -o $@ $(CONFIGLIB)
 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
