--- Makefile.in.orig	2007-11-13 01:43:12.000000000 -0600
+++ Makefile.in	2007-11-13 01:45:35.000000000 -0600
@@ -20,10 +20,11 @@
 
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
+DESTDIR		=
 CFITSIO_PREFIX	= $(prefix)
-CFITSIO_LIB	= $(exec_prefix)/lib
-CFITSIO_INCLUDE	= $(prefix)/include
-INSTALL_DIRS	= @INSTALL_ROOT@ ${CFITSIO_LIB} ${CFITSIO_LIB}/pkgconfig ${CFITSIO_INCLUDE}
+CFITSIO_LIB	= $(DESTDIR)$(exec_prefix)/lib
+CFITSIO_INCLUDE	= $(DESTDIR)$(prefix)/include
+INSTALL_DIRS	= $(DESTDIR)@INSTALL_ROOT@ ${CFITSIO_LIB} ${CFITSIO_LIB}/pkgconfig ${CFITSIO_INCLUDE}
 
 
 SHELL =		/bin/sh
