--- Makefile.orig	2007-08-30 16:01:17.000000000 +0200
+++ Makefile	2007-08-30 16:03:05.000000000 +0200
@@ -17,14 +17,14 @@
 	for d in $(SUBDIRS); do make -C $$d; [ $$? = 0 ] || exit 1 ; done
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/yum-arch
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/yum-arch
 	for p in $(PYFILES) ; do \
-		install -m 644 $$p $(DESTDIR)/usr/share/yum-arch/$$p; \
+		install -m 644 $$p $(DESTDIR)@@PREFIX@@/share/yum-arch/$$p; \
 	done
-	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/usr/share/yum-arch', 1, '$(PYDIR)', 1)"
+	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)@@PREFIX@@/share/yum-arch', 1, '$(PYDIR)', 1)"
 
-	mkdir -p $(DESTDIR)/usr/bin $(DESTDIR)/usr/bin
-	install -m 755 bin/yum-arch $(DESTDIR)/usr/bin/yum-arch
+	mkdir -p $(DESTDIR)@@PREFIX@@/bin $(DESTDIR)@@PREFIX@@/bin
+	install -m 755 bin/yum-arch $(DESTDIR)@@PREFIX@@/bin/yum-arch
 
 	mkdir -p $(DESTDIR)/var/cache/yum
 
--- ./docs/Makefile.orig	2007-08-30 16:05:21.000000000 +0200
+++ ./docs/Makefile	2007-08-30 16:07:46.000000000 +0200
@@ -5,5 +5,5 @@ clean:
 	rm -f *.pyc *.pyo *~
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/man/man8
-	install -m 644 yum-arch.8 $(DESTDIR)/usr/share/man/man8/yum-arch.8
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man8
+	install -m 644 yum-arch.8 $(DESTDIR)@@PREFIX@@/share/man/man8/yum-arch.8
--- ./i18n.py.orig	2007-08-30 16:33:16.000000000 +0200
+++ ./i18n.py	2007-08-30 16:34:44.000000000 +0200
@@ -17,7 +17,7 @@
         t = gettext.translation('yum')
         _ = t.gettext
     else:
-        gettext.bindtextdomain('yum', '/usr/share/locale')
+        gettext.bindtextdomain('yum', '@@PREFIX@@/share/locale')
         gettext.textdomain('yum')
         _ = gettext.gettext
 
--- ./bin/yum-arch.orig	2007-08-30 16:09:50.000000000 +0200
+++ ./bin/yum-arch	2007-08-30 16:15:04.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PYTHON@@
 # yum-arch -- Creates YUM distribution databases
 #
 # This program is free software; you can redistribute it and/or modify
@@ -18,6 +18,6 @@
 
 import sys
 
-sys.path.insert(0,'/usr/share/yum-arch')
+sys.path.insert(0,'@@PREFIX@@/share/yum-arch')
 import pullheaders
 pullheaders.main(sys.argv[1:])
--- ./repomd/Makefile.orig	2007-08-30 16:09:50.000000000 +0200
+++ ./repomd/Makefile	2007-08-30 16:15:04.000000000 +0200
@@ -3,7 +3,7 @@ PYFILES = $(wildcard *.py)
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = /usr/share/yum-arch/$(PACKAGE)
+PKGDIR = @@PREFIX@@/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./rpmUtils/Makefile.orig	2007-08-30 16:09:50.000000000 +0200
+++ ./rpmUtils/Makefile	2007-08-30 16:15:04.000000000 +0200
@@ -3,7 +3,7 @@ PYFILES = $(wildcard *.py)
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = /usr/share/yum-arch/$(PACKAGE)
+PKGDIR = @@PREFIX@@/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./urlgrabber/Makefile.orig	2007-08-30 16:09:50.000000000 +0200
+++ ./urlgrabber/Makefile	2007-08-30 16:15:04.000000000 +0200
@@ -3,7 +3,7 @@ PYFILES = $(wildcard *.py)
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = /usr/share/yum-arch/$(PACKAGE)
+PKGDIR = @@PREFIX@@/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./yum/Makefile.orig	2007-08-30 16:09:50.000000000 +0200
+++ ./yum/Makefile	2007-08-30 16:15:04.000000000 +0200
@@ -3,7 +3,7 @@ PYFILES = $(wildcard *.py)
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = /usr/share/yum-arch/$(PACKAGE)
+PKGDIR = @@PREFIX@@/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
