--- ./docs/Makefile.remi	2004-10-13 07:36:58.000000000 +0200
+++ ./docs/Makefile	2007-02-16 17:11:37.000000000 +0100
@@ -5,7 +5,5 @@
 	rm -f *.pyc *.pyo *~
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/man/man{5,8}
-	install -m 644 yum.8 $(DESTDIR)/usr/share/man/man8/yum.8
+	mkdir -p $(DESTDIR)/usr/share/man/man8
 	install -m 644 yum-arch.8 $(DESTDIR)/usr/share/man/man8/yum-arch.8
-	install -m 644 yum.conf.5 $(DESTDIR)/usr/share/man/man5/yum.conf.5
--- ./yum/Makefile.remi	2004-08-31 06:20:04.000000000 +0200
+++ ./yum/Makefile	2007-02-16 17:11:37.000000000 +0100
@@ -3,7 +3,7 @@
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
+PKGDIR = /usr/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./repomd/Makefile.remi	2004-08-31 06:20:03.000000000 +0200
+++ ./repomd/Makefile	2007-02-16 17:11:37.000000000 +0100
@@ -3,7 +3,7 @@
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
+PKGDIR = /usr/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./bin/yum-arch.remi	2004-10-13 07:57:34.000000000 +0200
+++ ./bin/yum-arch	2007-02-16 17:11:37.000000000 +0100
@@ -18,6 +18,6 @@
 
 import sys
 
-sys.path.insert(1,'/usr/share/yum-cli')
+sys.path.insert(0,'/usr/share/yum-arch')
 import pullheaders
 pullheaders.main(sys.argv[1:])
--- ./Makefile.remi	2005-01-25 06:56:43.000000000 +0100
+++ ./Makefile	2007-02-16 17:13:59.000000000 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = repomd rpmUtils urlgrabber yum etc docs
+SUBDIRS = repomd rpmUtils urlgrabber yum docs
 PYFILES = $(wildcard *.py)
 
 PKGNAME = yum
@@ -17,14 +17,13 @@
 	for d in $(SUBDIRS); do make -C $$d; [ $$? = 0 ] || exit 1 ; done
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/yum-cli
+	mkdir -p $(DESTDIR)/usr/share/yum-arch
 	for p in $(PYFILES) ; do \
-		install -m 644 $$p $(DESTDIR)/usr/share/yum-cli/$$p; \
+		install -m 644 $$p $(DESTDIR)/usr/share/yum-arch/$$p; \
 	done
-	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/usr/share/yum-cli', 1, '$(PYDIR)', 1)"
+	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/usr/share/yum-arch', 1, '$(PYDIR)', 1)"
 
 	mkdir -p $(DESTDIR)/usr/bin $(DESTDIR)/usr/bin
-	install -m 755 bin/yum.py $(DESTDIR)/usr/bin/yum
 	install -m 755 bin/yum-arch $(DESTDIR)/usr/bin/yum-arch
 
 	mkdir -p $(DESTDIR)/var/cache/yum
--- ./urlgrabber/Makefile.remi	2004-08-31 06:20:04.000000000 +0200
+++ ./urlgrabber/Makefile	2007-02-16 17:11:37.000000000 +0100
@@ -3,7 +3,7 @@
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
+PKGDIR = /usr/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- ./rpmUtils/Makefile.remi	2004-08-31 06:20:03.000000000 +0200
+++ ./rpmUtils/Makefile	2007-02-16 17:11:37.000000000 +0100
@@ -3,7 +3,7 @@
 PYVER := $(shell python -c 'import sys; print "%.3s" %(sys.version)')
 PYSYSDIR := $(shell python -c 'import sys; print sys.prefix')
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
-PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
+PKGDIR = /usr/share/yum-arch/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
