--- src/paths.c	2009-03-12 23:23:52.000000000 +0100
+++ src/paths.c	2009-03-12 23:24:09.000000000 +0100
@@ -140,7 +140,7 @@
        seem to be found of */
     default3 = br_build_path(datadir, "dink");
     default4 = "/usr/local/share/games/dink";
-    default5 = "/usr/local/share/dink";
+    default5 = "@PREFIX@/share/dink";
     default6 = "/usr/share/games/dink";
     default7 = "/usr/share/dink";
     lookup[3] = default3;
