--- configure.ac.orig	2016-10-03 11:29:26.000000000 -0700
+++ configure.ac	2016-10-03 11:39:20.000000000 -0700
@@ -71,7 +71,7 @@
 GLIB_COMPILE_RESOURCES=`$PKG_CONFIG --variable glib_compile_resources gio-2.0`
 AC_SUBST(GLIB_COMPILE_RESOURCES)
 
-AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache)
+AC_PATH_PROG(GTK_UPDATE_ICON_CACHE, gtk-update-icon-cache-3.0)
 if test -z "$GTK_UPDATE_ICON_CACHE"; then
     AC_MSG_ERROR([Could not find gtk-update-icon-cache])
 fi
