From 0137b9bd320783264d865a397392b0ee14fd69b3 Mon Sep 17 00:00:00 2001
From: Karl Tomlinson<karlt+@karlt.net>
Date: Wed, 17  Dec  2008  21:40:12  +0000
Subject: [ft] Don't call FT_Done_Face() on faces we did not create

---
diff --git a/src/cairo-ft-font.c b/src/cairo-ft-font.c
index cd11253..3675db6 100644
--- src/cairo-ft-font.c
+++ src/cairo-ft-font.c
@@ -528,7 +528,7 @@ _has_unlocked_face (void *entry)
 {
     cairo_ft_unscaled_font_t *unscaled = entry;
 
-    return (unscaled->lock_count == 0 && unscaled->face);
+    return (!unscaled->from_face && unscaled->lock_count == 0 && unscaled->face);
 }
 
 /* Ensures that an unscaled font has a face object. If we exceed
--
cgit v0.8.1-24-ge5fb
