--- configure.orig	2012-11-04 08:23:41.000000000 -0600
+++ configure	2012-11-06 13:41:24.000000000 -0600
@@ -30512,7 +30512,7 @@
 
 
 
-SPECTRE_VERSION_REQUIRED=0.2.0
+SPECTRE_VERSION_REQUIRED=99999
 test_ps=no
 any2ppm_ps=no
 if test "x$use_ps" = "xyes"; then
@@ -30909,7 +30909,7 @@
 
 
 # poppler-0.17.4 fixes text-pattern and text-transform
-POPPLER_VERSION_REQUIRED=0.17.4
+POPPLER_VERSION_REQUIRED=99999
 test_pdf=no
 any2ppm_pdf=no
 if test "x$use_pdf" = "xyes"; then
@@ -31279,7 +31279,7 @@
 
 
 
-LIBRSVG_VERSION_REQUIRED=2.15.0
+LIBRSVG_VERSION_REQUIRED=99999
 test_svg=no
 any2ppm_svg=no
 if test "x$use_svg" = "xyes"; then
