--- Makefile.orig	2009-02-05 10:00:20.000000000 -0600
+++ Makefile	2009-02-08 18:16:32.000000000 -0600
@@ -33,7 +33,7 @@
 # OSX (untested)
 DLL = .dylib
 shared = -dynamiclib
-LinkGL = -framework OpenGL
+LinkGL = -framework GLUT -framework OpenGL
 endif
 ifneq "$(findstring x86_64-,$(host))" ""
 # 64 bit, needs -fPIC flag
@@ -64,9 +64,9 @@
 	rm -f *$(DLL) *~ *.a *.o
 
 install:
-	test -d "$(DESTDIR)$(libdir)/pure" || mkdir -p "$(DESTDIR)$(libdir)/pure"
+	test -d "$(DESTDIR)`readlink $(libdir)/pure`" || mkdir -p "$(DESTDIR)`readlink $(libdir)/pure`"
 	cp GL.pure GLU.pure GLUT.pure GL_ARB.pure GL_EXT.pure GL_NV.pure GL_ATI.pure \
-	   pure-gl$(DLL) "$(DESTDIR)$(libdir)/pure"
+	   pure-gl$(DLL) "$(DESTDIR)`readlink $(libdir)/pure`"
 
 uninstall:
 	rm -f "$(DESTDIR)$(libdir)/pure/"GL*.pure "$(DESTDIR)$(libdir)/pure/pure-gl$(DLL)"
