From aff729128386a61ac89cf545ddf1b96df0282787 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Tue, 6 Jun 2023 10:19:10 +0200
Subject: [PATCH] add MacPorts' XDG_DATA_DIRS

---
 contrib/asdf/uiop.lisp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git contrib/asdf/uiop.lisp contrib/asdf/uiop.lisp
index 5e52e7745..8a597ec8f 100644
--- contrib/asdf/uiop.lisp
+++ contrib/asdf/uiop.lisp
@@ -7188,7 +7188,7 @@ also \"Configuration DSL\"\) in the ASDF manual."
             (or (remove nil (getenv-absolute-directories "XDG_DATA_DIRS"))
                 (os-cond
                  ((os-windows-p) (mapcar 'get-folder-path '(:appdata :common-appdata)))
-                 (t (mapcar 'parse-unix-namestring '("/usr/local/share/" "/usr/share/")))))))
+                 (t (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/" "/usr/local/share/" "/usr/share/")))))))
 
   (defun xdg-config-dirs (&rest more)
     "The preference-ordered set of additional base paths to search for configuration files.
-- 
2.41.0

