From d773dffdf6f03d4168ae8febbd8468ca8bdda8fa Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Tue, 6 Jun 2023 12:08:08 +0200
Subject: [PATCH] Use CLOCK_MONOTONIC when CLOCK_PROCESS_CPUTIME_ID unavailable

---
 src/runtime/runtime.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/runtime/runtime.h src/runtime/runtime.h
index d18d68e4c..34707a04a 100644
--- src/runtime/runtime.h
+++ src/runtime/runtime.h
@@ -416,4 +416,8 @@ extern struct lisp_startup_options lisp_startup_options;
  */
 #define TINY_BOXED_NWORDS(obj) ((HeaderValue(obj) & 0xFF) | 1)
 
+#ifndef CLOCK_PROCESS_CPUTIME_ID
+#define CLOCK_PROCESS_CPUTIME_ID CLOCK_MONOTONIC
+#endif
+
 #endif /* _SBCL_RUNTIME_H_ */
-- 
2.41.0

