From 5abffa167689977e1236e7489b780194cd67cb21 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Tue, 6 Jun 2023 12:09:09 +0200
Subject: [PATCH] Disable pdf manual

---
 doc/make-doc.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git doc/make-doc.sh doc/make-doc.sh
index fe8f261cd..e592ac342 100644
--- doc/make-doc.sh
+++ doc/make-doc.sh
@@ -3,4 +3,4 @@
 . ../find-gnumake.sh
 find_gnumake
 
-(cd ./manual ; $GNUMAKE html pdf info)
+(cd ./manual ; $GNUMAKE html info)
-- 
2.41.0

