From a6dccb448eabaf91eb2ecbde72e198f4fd7c6e0c Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Tue, 6 Jun 2023 10:31:20 +0200
Subject: [PATCH] x86/Darwin: disable failing stat-mode tests

---
 contrib/sb-posix/posix-tests.lisp | 2 ++
 1 file changed, 2 insertions(+)

diff --git contrib/sb-posix/posix-tests.lisp contrib/sb-posix/posix-tests.lisp
index fc00f6aba..936f6c366 100644
--- contrib/sb-posix/posix-tests.lisp
+++ contrib/sb-posix/posix-tests.lisp
@@ -328,6 +328,7 @@
             (,mode (sb-posix::stat-mode ,stat)))
        ,@body)))
 
+#-(and darwin x86)
 (deftest stat-mode.1
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isreg mode))
@@ -344,6 +345,7 @@
     (sb-posix:s-ischr mode))
   nil)
 
+#-(and darwin x86)
 (deftest stat-mode.4
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isblk mode))
-- 
2.41.0

