From 5bd3a25311bf971106c8a02c95ddf9e147950349 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Tue, 6 Jun 2023 10:32:35 +0200
Subject: [PATCH] x86/Darwin: fix syntax typos

---
 src/runtime/x86-darwin-os.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git src/runtime/x86-darwin-os.c src/runtime/x86-darwin-os.c
index bba95e318..5ac13f970 100644
--- src/runtime/x86-darwin-os.c
+++ src/runtime/x86-darwin-os.c
@@ -91,7 +91,7 @@ int arch_os_thread_init(struct thread *thread) {
      * we've exhausted it */
     sigstack.ss_sp    = calc_altstack_base(thread);
     sigstack.ss_flags = 0;
-    sigstack.ss_size  = calc_altstack_size(thread;
+    sigstack.ss_size  = calc_altstack_size(thread);
     sigaltstack(&sigstack,0);
 #endif
     return 1;                  /* success */
@@ -514,6 +514,8 @@ catch_exception_raise(mach_port_t exception_port,
     return ret;
 }
 
+#endif
+
 void
 os_restore_fp_control(os_context_t *context)
 {
@@ -525,5 +527,3 @@ os_restore_fp_control(os_context_t *context)
     /* reset exception flags and restore control flags on x87 FPU */
     asm ("fldcw %0" : : "m" (fpu_control_word));
 }
-
-#endif
-- 
2.41.0

