--- contrib/sb-posix/posix-tests.lisp.orig	2009-10-25 09:24:59.000000000 +0100
+++ contrib/sb-posix/posix-tests.lisp	2010-01-06 13:08:14.000000000 +0100
@@ -103,6 +103,7 @@
       (sb-posix:syscall-errno c)))
   #.sb-posix::eexist)
 
+#-darwin
 (deftest mkdir.error.2
   (handler-case
       (sb-posix:mkdir #-win32 "/" #+win32 "C:/" 0)
@@ -162,6 +163,7 @@
   #+win32
   #.sb-posix::einval)
 
+#-darwin
 (deftest rmdir.error.3
   (handler-case
       (sb-posix:rmdir #-win32 "/" #+win32 "C:/")
@@ -326,6 +328,7 @@
             (,mode (sb-posix::stat-mode ,stat)))
        ,@body)))
 
+#-darwin
 (deftest stat-mode.1
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isreg mode))
@@ -342,11 +345,13 @@
     (sb-posix:s-ischr mode))
   nil)
 
+#-darwin
 (deftest stat-mode.4
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isblk mode))
   nil)
 
+#-darwin
 (deftest stat-mode.5
   (with-stat-mode (mode *test-directory*)
     (sb-posix:s-isfifo mode))
@@ -506,6 +511,7 @@
         (sb-posix:closedir dir))))
   nil)
 
+#-darwin
 (deftest readdir.1
   (let ((dir (sb-posix:opendir "/")))
     (unwind-protect
@@ -662,6 +668,7 @@
     #.(concatenate 'string "/" (make-string 255 :initial-element #\a)))
 
   ;; The error tests are in the order of exposition from SUSv3.
+  #-darwin
   (deftest readlink.error.1
       (let* ((subdir-pathname (merge-pathnames
                                (make-pathname
