--- src/runtime/Config.arm64-darwin.orig  2022-06-30 00:27:00.000000000 -0700
+++ src/runtime/Config.arm64-darwin       2022-07-12 21:45:30.000000000 -0700
@@ -29,7 +29,9 @@
   OS_LIBS += -lpthread
 endif
 ifdef LISP_FEATURE_SB_CORE_COMPRESSION
-  OS_LIBS += -lzstd
+  OS_LIBS += -L@@PREFIX@@/lib -lzstd
+  DEPEND_FLAGS += -I@@PREFIX@@/include
+  CFLAGS += -I@@PREFIX@@/include
 endif
 ifdef LISP_FEATURE_SB_LINKABLE_RUNTIME
   LIBSBCL = libsbcl.a
