--- base-target-features.lisp-expr.orig	2007-10-16 16:34:50.000000000 -0600
+++ base-target-features.lisp-expr	2007-10-16 16:35:26.000000000 -0600
@@ -157,7 +157,7 @@
  ;; and the %fs segment register.  Note that no consistent effort to
  ;; audit the SBCL library code for thread safety has been performed,
  ;; so caveat executor.
- ; :sb-thread
+:sb-thread
 
  ;; lutex support
  ;;
@@ -167,7 +167,7 @@
  ;; allows for sb-thread support on x86 OS X, Solaris and
  ;; FreeBSD.
  ;;
- ; :sb-lutex
+:sb-lutex
 
  ;; On some operating systems the FS segment register (used for SBCL's
  ;; thread local storage) is not reliably preserved in signal
@@ -337,4 +337,5 @@
  ;; (No others are supported by SBCL as of 1.0.8, but :hpux or :irix
  ;; support could be ported from CMU CL if anyone is sufficiently
  ;; motivated to do so.)
+:darwin
  )
