--- Makefile.in.orig	2013-01-27 10:25:55.000000000 -0800
+++ Makefile.in	2013-01-27 10:28:00.000000000 -0800
@@ -73,7 +73,8 @@ XLIBS=@LIBS@
 
 LDSHARED=@LDSHARED@
 LDFLAGS=@LDFLAGS@
-CFLAGS=$(COMMON_CFLAGS) $(XCFLAGS) @XCFLAGS@
+CFLAGS=@CFLAGS@ @XCFLAGS@ $(COMMON_CFLAGS)
+CPPFLAGS=@CPPFLAGS@
 
 all: mozplugger.so mozplugger-helper mozplugger-controller mozplugger-linker
 
@@ -81,37 +82,37 @@ mozplugger-helper: mozplugger-helper.o m
 	$(CC) $(LDFLAGS) -o mozplugger-helper mozplugger-helper.o mozplugger-common.o child.o debug.o $(XLIBS)
 
 mozplugger-helper.o: mozplugger-helper.c mozplugger.h child.h debug.h config.h
-	$(CC) -c $(CFLAGS) -o mozplugger-helper.o '$(srcdir)/mozplugger-helper.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o mozplugger-helper.o '$(srcdir)/mozplugger-helper.c'
 
 mozplugger.so: mozplugger.o npunix.o mozplugger-common.o debug.o
 	$(LDSHARED) $(LDFLAGS) -o mozplugger.so mozplugger.o mozplugger-common.o npunix.o debug.o $(XLIBS)
 
 mozplugger-common.o: mozplugger-common.c mozplugger.h config.h
-	$(CC) -c $(CFLAGS) -o mozplugger-common.o '$(srcdir)/mozplugger-common.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o mozplugger-common.o '$(srcdir)/mozplugger-common.c'
 
 mozplugger.o: mozplugger.c mozplugger.h config.h
-	$(CC) -c $(CFLAGS) -o mozplugger.o '$(srcdir)/mozplugger.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o mozplugger.o '$(srcdir)/mozplugger.c'
 
 npunix.o: $(srcdir)/npapi/common/npunix.c config.h
-	$(CC) -c $(CFLAGS) -o npunix.o '$(srcdir)/npapi/common/npunix.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o npunix.o '$(srcdir)/npapi/common/npunix.c'
 
 child.o: child.c child.h debug.h mozplugger.h config.h
-	$(CC) -c $(CFLAGS) -o child.o '$(srcdir)/child.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o child.o '$(srcdir)/child.c'
 
 debug.o: debug.c debug.h config.h
-	$(CC) -c $(CFLAGS) -o debug.o '$(srcdir)/debug.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o debug.o '$(srcdir)/debug.c'
 
 mozplugger-controller: mozplugger-controller.o mozplugger-common.o child.o debug.o
 	$(CC) -o mozplugger-controller mozplugger-controller.o mozplugger-common.o child.o debug.o $(LDFLAGS) $(XLIBS) 
 
 mozplugger-controller.o: mozplugger-controller.c mozplugger.h child.h debug.h config.h
-	$(CC) -c $(CFLAGS) -o mozplugger-controller.o '$(srcdir)/mozplugger-controller.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o mozplugger-controller.o '$(srcdir)/mozplugger-controller.c'
 
 mozplugger-linker: mozplugger-linker.o mozplugger-common.o child.o debug.o
 	$(CC) -o mozplugger-linker mozplugger-linker.o mozplugger-common.o child.o debug.o $(LDFLAGS) $(XLIBS)
 
 mozplugger-linker.o: mozplugger-linker.c mozplugger.h child.h debug.h config.h
-	$(CC) -c $(CFLAGS) -o mozplugger-linker.o '$(srcdir)/mozplugger-linker.c'
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) -o mozplugger-linker.o '$(srcdir)/mozplugger-linker.c'
 
 $(srcdir)/configure: configure.ac
 	cd '$(srcdir)' && autoconf
