From 5d32457c53bb0ae8abdafeceb1b622a7b7934de2 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Mon, 15 Apr 2013 22:38:18 -0700
Subject: [PATCH] Comment out SL cctools workaround

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 make/platform/clang_darwin.mk | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/make/platform/clang_darwin.mk b/make/platform/clang_darwin.mk
index a5e22e6..6a4e956 100644
--- a/projects/compiler-rt/make/platform/clang_darwin.mk
+++ a/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -83,12 +83,12 @@ UniversalArchs.ubsan_osx := $(call Check
 # object files. If we are on that platform, strip out all ARM archs. We still
 # build the libraries themselves so that Clang can find them where it expects
 # them, even though they might not have an expected slice.
-ifneq ($(shell test -x /usr/bin/sw_vers && sw_vers -productVersion | grep 10.6),)
-UniversalArchs.ios := $(filter-out armv7, $(UniversalArchs.ios))
-UniversalArchs.cc_kext := $(filter-out armv7, $(UniversalArchs.cc_kext))
-UniversalArchs.cc_kext_ios5 := $(filter-out armv7, $(UniversalArchs.cc_kext_ios5))
-UniversalArchs.profile_ios := $(filter-out armv7, $(UniversalArchs.profile_ios))
-endif
+#ifneq ($(shell test -x /usr/bin/sw_vers && sw_vers -productVersion | grep 10.6),)
+#UniversalArchs.ios := $(filter-out armv7, $(UniversalArchs.ios))
+#UniversalArchs.cc_kext := $(filter-out armv7, $(UniversalArchs.cc_kext))
+#UniversalArchs.cc_kext_ios5 := $(filter-out armv7, $(UniversalArchs.cc_kext_ios5))
+#UniversalArchs.profile_ios := $(filter-out armv7, $(UniversalArchs.profile_ios))
+#endif
 
 # If RC_SUPPORTED_ARCHS is defined, treat it as a list of the architectures we
 # are intended to support and limit what we try to build to that.
-- 
1.8.2.1

