--- release_32/Makefile.orig	2012-11-17 14:30:41.000000000 -0500
+++ release_32/Makefile	2012-11-17 16:17:04.000000000 -0500
@@ -70,9 +70,8 @@ ifeq ($(MAKECMDGOALS),install-clang)
   DIRS := tools/clang/tools/driver tools/clang/lib/Headers \
           tools/clang/tools/libclang \
           tools/clang/tools/c-index-test \
-          tools/clang/include/clang-c \
-          tools/clang/runtime tools/clang/docs \
-          tools/lto runtime
+          tools/clang/include tools/clang/lib \
+          tools/clang/runtime tools/clang/docs
   OPTIONAL_DIRS :=
   NO_INSTALL = 1
 endif
