--- c-blosc/hdf5/blosc_filter.c.orig	2015-08-13 11:58:26.000000000 +0300
+++ c-blosc/hdf5/blosc_filter.c	2015-08-13 11:58:37.000000000 +0300
@@ -175,7 +175,7 @@
     int doshuffle = 1;             /* Shuffle default */
     int compcode;                  /* Blosc compressor */
     int code;
-    char *compname = NULL;
+    char *compname = "blosclz";
     char *complist;
     char errmsg[256];
 
