--- setup.py.orig	2014-03-25 13:49:03.000000000 -0700
+++ setup.py	2014-09-11 17:07:03.000000000 -0700
@@ -123,20 +123,20 @@
 # Check if Cython is installed or not (requisite)
 try:
     from Cython.Distutils import build_ext
-    from Cython.Compiler.Main import Version
+    from Cython.Compiler.Main import version as cython_version
     cmdclass['build_ext'] = build_ext
 except ImportError:
     exit_with_error(
         "You need %(pkgname)s %(pkgver)s or greater to compile PyTables!"
         % {'pkgname': 'Cython', 'pkgver': min_cython_version})
 
-if Version.version < min_cython_version:
+if cython_version < min_cython_version:
     exit_with_error(
         "At least Cython %s is needed so as to generate extensions!"
         % (min_cython_version))
 else:
     print("* Found %(pkgname)s %(pkgver)s package installed."
-          % {'pkgname': 'Cython', 'pkgver': Version.version})
+          % {'pkgname': 'Cython', 'pkgver': cython_version})
 
 VERSION = open('VERSION').read().strip()
 
