--- kfind/Makefile.am	10 Jul 2002 03:21:13 -0000	1.1.1.2
+++ kfind/Makefile.am	9 May 2003 00:40:05 -0000	1.2
@@ -10,7 +10,7 @@
 libkfind_common_la_SOURCES = kftabdlg.cpp kfind.cpp kquery.cpp kdatecombo.cpp
 
 #This section is kfindpart
-lib_LTLIBRARIES = libkfindpart.la
+kde_module_LTLIBRARIES = libkfindpart.la
 
 libkfindpart_la_SOURCES = kfindpart.cpp
 libkfindpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
