--- kioslave/devices/kdedmodule/Makefile.am	29 Oct 2002 03:05:23 -0000	1.1.1.2
+++ kioslave/devices/kdedmodule/Makefile.am	25 Mar 2003 18:16:37 -0000	1.6
@@ -3,7 +3,7 @@
 INCLUDES = $(all_includes)
 kded_mountwatcher_la_SOURCES = mountwatcher.cpp mountwatcher.skel disklist.cpp disks.cpp
 kded_mountwatcher_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_mountwatcher_la_LIBADD = $(LIB_KSYCOCA)
+kded_mountwatcher_la_LIBADD = $(LIB_KSYCOCA) $(LIB_QT) -lDCOP -lkdecore
 
 METASOURCES = AUTO
 
