--- kcontrol/kcontrol/kcmshell.cpp	14 Aug 2002 14:29:19 -0000	1.1.1.2
+++ kcontrol/kcontrol/kcmshell.cpp	26 Nov 2002 21:00:51 -0000	1.2
@@ -129,8 +129,9 @@
         quit();
 }
 
+extern "C" int kdemain(int, char**);
 
-int main(int _argc, char *_argv[])
+int kdemain(int _argc, char *_argv[])
 {
     KAboutData aboutData( "kcmshell", I18N_NOOP("KDE Control Module"),
                           KCONTROL_VERSION,
