--- kicker/menuext/prefmenu/Makefile.am	10 Jul 2002 03:22:41 -0000	1.1.1.1
+++ kicker/menuext/prefmenu/Makefile.am	24 Nov 2002 07:11:31 -0000	1.4
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_prefmenu.la
 
 kickermenu_prefmenu_la_SOURCES = prefmenu.cpp
-kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_prefmenu_la_LDFLAGS = $(all_libraries) -module -avoid-version -undefined suppress -flat_namespace
 kickermenu_prefmenu_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/ui/libkicker_ui.la
 
 kickermenu_prefmenu_la_METASOURCES = AUTO
