--- kicker/menuext/recentdocs/Makefile.am	10 Jul 2002 03:22:42 -0000	1.1.1.1
+++ kicker/menuext/recentdocs/Makefile.am	24 Nov 2002 07:11:31 -0000	1.4
@@ -3,7 +3,7 @@
 kde_module_LTLIBRARIES = kickermenu_recentdocs.la
 
 kickermenu_recentdocs_la_SOURCES = recentdocsmenu.cpp
-kickermenu_recentdocs_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kickermenu_recentdocs_la_LDFLAGS = $(all_libraries) -module -avoid-version -undefined suppress -flat_namespace
 kickermenu_recentdocs_la_LIBADD = $(LIB_KDEUI) $(top_builddir)/kicker/ui/libkicker_ui.la
 
 kickermenu_recentdocs_la_METASOURCES = AUTO
