--- konqueror/sidebar/trees/dirtree_module/Makefile.am	25 Jun 2002 02:03:53 -0000	1.1.1.1
+++ konqueror/sidebar/trees/dirtree_module/Makefile.am	26 Nov 2002 21:00:53 -0000	1.3
@@ -5,6 +5,6 @@
 METASOURCES = AUTO
 konq_sidebartree_dirtree_la_SOURCES = dirtree_module.cpp dirtree_item.cpp
 konq_sidebartree_dirtree_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-konq_sidebartree_dirtree_la_LIBADD = ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_dirtree_la_LIBADD = $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/libkonq_sidebar_tree.la
 
 
