diff -uNr kdm/kfrontend/Makefile.am kdm/kfrontend/Makefile.am
--- kdm/kfrontend/Makefile.am	Wed May  7 04:44:08 2003
+++ kdm/kfrontend/Makefile.am	Tue May 27 10:04:35 2003
@@ -2,7 +2,7 @@
 
 INCLUDES = -I$(srcdir)/../backend -I$(top_srcdir)/kcontrol/background $(all_includes)
 
-bin_PROGRAMS = kdm_config kdm_greet krootimage
+bin_PROGRAMS = kdm_config kdm_greet krootimage genkdmconf
 
 kdm_config_SOURCES = kdm_config.c
 kdm_config_LDADD = $(LIBRESOLV) $(LIBSOCKET)
@@ -24,14 +24,9 @@
 
 METASOURCES = AUTO
 
-noinst_PROGRAMS = genkdmconf
 genkdmconf_SOURCES = genkdmconf.c
 genkdmconf_LDFLAGS = -L$(x_libraries)
 genkdmconf_LDADD = $(LIB_X11)
-
-install-data-local: genkdmconf
-	@$(mkinstalldirs) $(DESTDIR)$(kde_confdir)/kdm
-	@./genkdmconf --in $(DESTDIR)$(kde_confdir)/kdm
 
 messages:
 	$(XGETTEXT) $(kdm_greet_SOURCES) -o $(podir)/kdmgreet.pot
