--- kate/app/Makefile.am	12 Nov 2002 04:16:41 -0000	1.1.1.4
+++ kate/app/Makefile.am	1 Jun 2003 14:31:53 -0000	1.11
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = libkateinterfaces.la kate.la kwrite.la
+lib_LTLIBRARIES = libkateinterfaces.la kate.la kwrite.la libkwrite_main.la
 bin_PROGRAMS = kate kwrite
 
 libkateinterfaces_la_SOURCES = kateapp.cpp kateconfigdialog.cpp kateconfigplugindialogpage.cpp \
@@ -7,34 +7,35 @@
 	kateviewmanager.cpp kateviewspace.cpp \
 	kateIface.skel kategrepdialog.cpp katefiledialog.cpp katemailfilesdialog.cpp \
         kbookmarkhandler.cpp kactionselector.cpp katedockcontainer.cpp
+libkateinterfaces_la_LIBADD   = $(top_builddir)/kate/interfaces/libkateinterfacesprivate.la -lkscript $(top_builddir)/kate/lib/libkmultitabbar.la
+libkateinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
 
-libkateinterfaces_la_LIBADD   = ../interfaces/libkateinterfacesprivate.la -lkscript ../lib/libkmultitabbar.la
-libkateinterfaces_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
-kate_la_SOURCES = katemain.cpp
-kate_la_LIBADD = libkateinterfaces.la
-kate_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version 
-
-kwrite_la_SOURCES = kwritemain.cpp katefiledialog.cpp kwritedialogs.cpp
-kwrite_la_LIBADD   = -lktexteditor
-kwrite_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kate_la_SOURCES = kate_la_main.cpp
+kate_la_LIBADD  = libkateinterfaces.la
+kate_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -module
 
-kate_SOURCES = dummy_kate.cpp
-kate_LDADD = kate.la
-kate_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+kate_la_main.cpp: katemain.cpp
+	cat katemain.cpp > kate_la_main.cpp
 
-kwrite_SOURCES = dummy_kwrite.cpp
-kwrite_LDADD = kwrite.la
-kwrite_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+libkwrite_main_la_SOURCES = kwritemain.cpp katefiledialog.cpp kwritedialogs.cpp
+libkwrite_main_la_LIBADD  = -lktexteditor $(LIB_QT) -lkdecore -lDCOP -lkdeui -lkio -lkparts
+libkwrite_main_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+
+kwrite_la_SOURCES = kwrite_la_main.cpp
+kwrite_la_LIBADD  = libkwrite_main.la
+kwrite_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -avoid-version -module
 
-dummy_kate_la.cpp:
-	echo > dummy_kate_la.cpp
+kwrite_la_main.cpp: kwrite_main.cpp
+	cat kwrite_main.cpp > kwrite_la_main.cpp
 
-dummy_kate.cpp:
-	echo > dummy_kate.cpp
+kate_SOURCES = katemain.cpp
+kate_LDADD = libkateinterfaces.la
+kate_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+
+kwrite_SOURCES = kwrite_main.cpp
+kwrite_LDADD = libkwrite_main.la
+kwrite_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
-dummy_kwrite.cpp:
-	echo > dummy_kwrite.cpp
-                         
 METASOURCES = AUTO  
   
 INCLUDES= -I../interfaces -I$(srcdir)/../lib $(all_includes)
