--- konqueror/sidebar/trees/bookmark_module/Makefile.am	25 Jun 2002 02:03:53 -0000	1.1.1.1
+++ konqueror/sidebar/trees/bookmark_module/Makefile.am	26 Nov 2002 21:00:53 -0000	1.3
@@ -6,5 +6,5 @@
 
 konq_sidebartree_bookmarks_la_SOURCES = bookmark_module.cpp bookmark_item.cpp
 konq_sidebartree_bookmarks_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-konq_sidebartree_bookmarks_la_LIBADD = ../../libkonqsidebarplugin.la ../libkonq_sidebar_tree.la
+konq_sidebartree_bookmarks_la_LIBADD = $(top_builddir)/konqueror/sidebar/libkonqsidebarplugin.la $(top_builddir)/konqueror/sidebar/trees/libkonq_sidebar_tree.la
 
