--- kcheckpass/checkpass_pam.c	25 Jun 2002 02:03:44 -0000	1.1.1.1
+++ kcheckpass/checkpass_pam.c	21 Nov 2002 21:28:36 -0000	1.2
@@ -26,7 +26,12 @@
  *****************************************************************/
 #include <stdlib.h>
 #include <string.h>
+
+#ifdef HAVE_PAM_PAM_APPL_H
+#include <pam/pam_appl.h>
+#else
 #include <security/pam_appl.h>
+#endif
 
 static const char *PAM_username;
 static const char *PAM_password;
