--- kwin/clients/icewm/Makefile.am	25 Jun 2002 02:03:54 -0000	1.1.1.1
+++ kwin/clients/icewm/Makefile.am	26 Nov 2002 21:00:54 -0000	1.2
@@ -5,7 +5,7 @@
 kde_module_LTLIBRARIES = kwin_icewm.la
 
 kwin_icewm_la_SOURCES = icewm.cpp 
-kwin_icewm_la_LIBADD = ../../kwin.la
+kwin_icewm_la_LIBADD = ../../libkwin_main.la
 kwin_icewm_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 METASOURCES = AUTO
