--- framework/sources/HFByteArrayEditScript.m.orig	2012-02-05 23:57:47.000000000 +0100
+++ framework/sources/HFByteArrayEditScript.m	2012-02-05 23:58:02.000000000 +0100
@@ -19,7 +19,7 @@
 #define MAX_RECURSION_DEPTH 64
 
 #if NDEBUG
-#define BYTEARRAY_RELEASE_INLINE __attribute__((always_inline)) static
+#define BYTEARRAY_RELEASE_INLINE inline static
 #else
 #define BYTEARRAY_RELEASE_INLINE __attribute__((noinline)) static
 #endif
