diff --git libarchive/test/test_entry.c libarchive/test/test_entry.c
index f205764..33cefe4 100644
--- libarchive/test/test_entry.c
+++ libarchive/test/test_entry.c
@@ -427,6 +427,9 @@ DEFINE_TEST(test_entry)
 #endif
 
 #if defined(__FreeBSD__) || defined(__APPLE__)
+#ifndef UF_HIDDEN
+#define UF_HIDDEN   0x00008000
+#endif
 	/* Test archive_entry_copy_fflags_text_w() */
 	archive_entry_copy_fflags_text_w(e, L" ,nouappnd, nouchg, dump,hidden");
 	archive_entry_fflags(e, &set, &clear);
diff --git tar/test/test_option_acls.c tar/test/test_option_acls.c
index f7451c8..de9b9b0 100644
--- tar/test/test_option_acls.c
+++ tar/test/test_option_acls.c
@@ -44,8 +44,12 @@ static const acl_perm_t acl_perms[] = {
     ACL_WRITE_EXTATTRIBUTES,
     ACL_READ_SECURITY,
     ACL_WRITE_SECURITY,
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
     ACL_CHANGE_OWNER,
     ACL_SYNCHRONIZE
+#else
+    ACL_CHANGE_OWNER
+#endif
 #else /* !ARCHIVE_ACL_DARWIN */
     ACL_EXECUTE,
     ACL_WRITE,
