--- libarchive/test/test_entry.c	2025-11-17 14:04:25
+++ libarchive/test/test_entry.c	2026-02-27 12:01:42
@@ -426,6 +426,9 @@
 #endif
 
 #if defined(__FreeBSD__) || defined(__APPLE__)
+#ifndef UF_HIDDEN
+#define UF_HIDDEN   0x00008000
+#endif
 	/* Test archive_entry_copy_fflags_text_w() */
 	archive_entry_copy_fflags_text_w(e, L" ,nouappnd, nouchg, dump,hidden");
 	archive_entry_fflags(e, &set, &clear);
--- tar/test/test_option_acls.c	2025-11-17 14:04:25
+++ tar/test/test_option_acls.c	2026-02-27 12:00:23
@@ -25,8 +25,12 @@
     ACL_WRITE_EXTATTRIBUTES,
     ACL_READ_SECURITY,
     ACL_WRITE_SECURITY,
+#if HAVE_DECL_ACL_SYNCHRONIZE
     ACL_CHANGE_OWNER,
     ACL_SYNCHRONIZE
+#else
+    ACL_CHANGE_OWNER
+#endif
 #else /* !ARCHIVE_ACL_DARWIN */
     ACL_EXECUTE,
     ACL_WRITE,
@@ -400,6 +404,12 @@
 		ret = 0;
 		goto exit_free;
 	}
+#if ARCHIVE_ACL_DARWIN
+	if (acl_b == NULL) {
+		ret = 0;
+		goto exit_free;
+	}
+#endif
 	b = acl_get_entry(acl_b, ACL_FIRST_ENTRY, &aclent_b);
 	if (b == -1) {
 		ret = 0;
