--- src/color.c.orig
+++ src/color.c
@@ -517,7 +517,7 @@ cbtick_callback(
 	    if (axis_array[axis].manual_justify)
 		just = axis_array[axis].label.pos;
 	    write_multiline(x2+offsetx, y3+offsety, text,
-			    just, CENTRE, hrotate,
+			    just, JUST_CENTRE, hrotate,
 			    axis_array[axis].ticdef.font);
 	    if (hrotate)
 		(*term->text_angle)(0);
@@ -528,7 +528,7 @@ cbtick_callback(
 	    if (axis_array[axis].manual_justify)
 		just = axis_array[axis].label.pos;	    
 	    write_multiline(x3+offsetx, y2+offsety, text,
-			    just, CENTRE, 0.0,
+			    just, JUST_CENTRE, 0.0,
 			    axis_array[axis].ticdef.font);
 	}
 	term_apply_lp_properties(&border_lp);	/* border linetype */
--- src/graphics.c.orig
+++ src/graphics.c
@@ -1720,7 +1720,7 @@ do_plot(struct curve_points *plots, int pcount)
 	ignore_enhanced(axis_array[FIRST_X_AXIS].label.noenhanced);
 	apply_pm3dcolor(&(axis_array[FIRST_X_AXIS].label.textcolor), t);
 	write_multiline(x, y, axis_array[FIRST_X_AXIS].label.text,
-			JUST_CENTRE, JUST_TOP, 0,
+			CENTRE, JUST_TOP, 0,
 			axis_array[FIRST_X_AXIS].label.font);
 	reset_textcolor(&(axis_array[FIRST_X_AXIS].label.textcolor), t);
 	ignore_enhanced(FALSE);
--- src/plot2d.c.orig
+++ src/plot2d.c
@@ -2127,7 +2127,7 @@ eval_plots()
 
 		    if (this_plot->labels == NULL) {
 			this_plot->labels = new_text_label(-1);
-			this_plot->labels->pos = JUST_CENTRE;
+			this_plot->labels->pos = CENTRE;
 			this_plot->labels->layer = LAYER_PLOTLABELS;
 		    }
 		    parse_label_options(this_plot->labels);
@@ -2316,7 +2316,7 @@ eval_plots()
 	    if (this_plot->plot_style == LABELPOINTS) {
 		if (this_plot->labels == NULL) {
 		    this_plot->labels = new_text_label(-1);
-		    this_plot->labels->pos = JUST_CENTRE;
+		    this_plot->labels->pos = CENTRE;
 		    this_plot->labels->layer = LAYER_PLOTLABELS;
 		}
 		this_plot->labels->place.scalex =
--- src/plot3d.c.orig
+++ src/plot3d.c
@@ -1559,7 +1559,7 @@ eval_3dplots()
 		    int stored_token = c_token;
 		    if (this_plot->labels == NULL) {
 			this_plot->labels = new_text_label(-1);
-			this_plot->labels->pos = JUST_CENTRE;
+			this_plot->labels->pos = CENTRE;
 			this_plot->labels->layer = LAYER_PLOTLABELS;
 		    }
 		    parse_label_options(this_plot->labels);
--- src/wxterminal/wxt_gui.cpp.orig
+++ src/wxterminal/wxt_gui.cpp
@@ -2692,6 +2692,7 @@ void wxtPanel::wxt_cairo_exec_command(gp_command command)
 		text_justification_mode = command.mode;
 		return;
 	case command_put_text :
+	case command_enhanced_put_text :
 		if (wxt_in_key_sample) {
 			int slen = strlen(command.string) * term->h_char * 0.75;
 			if (text_justification_mode == RIGHT) slen = -slen;
--- term/lua.trm.orig
+++ term/lua.trm
@@ -872,7 +872,7 @@ LUA_init()
 {
   fseek(gpoutfile, 0, SEEK_SET);
   /* ignore compiler warnings here, because `gpoutfile' is already open */
-  if (ftruncate(fileno(gpoutfile), 0));
+  (void)ftruncate(fileno(gpoutfile), 0);
 
   /* reset image counter */
   image_cnt = 0;
--- term/x11.trm.orig
+++ term/x11.trm
@@ -617,7 +617,11 @@ X11_options()
     /* Leave the current window unchanged when closing an old window */
     if (X11_ipc) {
 	if (set_close) {
-	    fprintf(X11_ipc, (set_number ? "C%d\n" : "C\n"), new_term_number);
+	    if (set_number) {
+		fprintf(X11_ipc, "C%d\n", new_term_number);
+	    } else {
+		fputs("C\n", X11_ipc);
+	    }
 	    fflush(X11_ipc);
 	} else if (set_number) {
 #ifdef EXTERNAL_X11_WINDOW
