--- archs/Darwin/variables.GNU.orig	2001-11-08 16:53:03.000000000 -0600
+++ archs/Darwin/variables.GNU	2011-09-19 05:50:39.000000000 -0500
@@ -16,8 +16,8 @@
 	$(BUILD)/Darwin-main.o \
 	$(BUILD)/Darwin-main-help.o
 
-CPPFLAGS += $(shell sdl-config --cflags) -I/sw/include
-LDFLAGS  += -L/usr/local/lib -lSDL_image $(shell sdl-config --libs) -lpng
+CPPFLAGS += $(shell sdl-config --cflags)
+LDFLAGS  += -lSDL_image $(shell sdl-config --libs) -lpng
 
 $(CODE)/%.m:	$(NWS)
 	-@$(MKDIR_CMD)
