--- Demo/CMakeLists.txt.orig	2009-01-06 19:44:06.000000000 +0100
+++ Demo/CMakeLists.txt	2009-01-06 19:46:13.000000000 +0100
@@ -34,7 +34,8 @@
 #examples: ITKCommon, VTKRendering, etc
 SET(Libraries
 #list libraries here
-GL glut
+${OPENGL_LIBRARIES}
+${GLUT_LIBRARIES}
 )
 
 #LINK_DIRECTORIES(${CHIPMUNK_SOURCE_DIR}/build/source)
@@ -49,6 +50,8 @@
 
 TARGET_LINK_LIBRARIES(${CurrentExe} ${Libraries} chipmunk_static)
 
+install(TARGETS ${CurrentExe} RUNTIME DESTINATION bin)
+
 #the following line is an example of how to add a test to your project.
 #Testname is the title for this particular test.  ExecutableToRun is the
 #program which will be running this test.  It can either be a part of this
