--- configure.orig	2008-05-07 01:59:49.000000000 +1000
+++ configure	2011-06-06 13:54:59.000000000 +1000
@@ -3101,7 +3101,7 @@
 fi
 #PKG_CHECK_MODULES(SOUNDTOUCH,libSoundTouch,,[AC_MSG_ERROR(libsoundtouch not found.)])
 # PKG_CHECK_MODULES loses when you need --libs-only-[lL]
-SOUNDTOUCH_PKG="soundtouch-1.0"
+SOUNDTOUCH_PKG="soundtouch-1.4"
 if ! $PKG_CONFIG --exists $SOUNDTOUCH_PKG; then
   if ! $PKG_CONFIG --exists libSoundTouch; then
         { { echo "$as_me:$LINENO: error: soundtouch not found" >&5
