From 5cb2d1e544ee22a29cfbc2c9729e2d043eb916b5 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@freedesktop.org>
Date: Tue, 22 Dec 2009 10:58:07 -0800
Subject: [PATCH] darwin: Don't use poll()

Signed-off-by: Jeremy Huddleston <jeremyhu@freedesktop.org>
---
 configure.ac |   13 +++++++++++--
 1 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index e27f8fd..54f1f3f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -39,8 +39,8 @@ AC_PROG_INSTALL
 AC_CANONICAL_HOST
 
 
-AC_CHECK_HEADERS([poll.h pty.h stropts.h sys/param.h sys/select.h])
-AC_CHECK_FUNCS([poll select grantpt])
+AC_CHECK_HEADERS([pty.h stropts.h sys/param.h sys/select.h])
+AC_CHECK_FUNCS([select grantpt])
 
 AC_CHECK_LIB(z, gzopen,, AC_MSG_ERROR([zlib not found]))
 
@@ -55,10 +55,18 @@ AC_SUBST([LOCALEALIASFILE])
 PKG_CHECK_MODULES(LUIT, x11 fontenc)
 
 case $host_os in
+    # darwin has poll() but can't be used to poll character devices (atleast through SnowLeopard)
+    darwin*)
+        OS_CFLAGS=
+        ;;
     linux*)
+        AC_CHECK_HEADERS([poll.h])
+        AC_CHECK_FUNCS([poll])
         OS_CFLAGS="-D_XOPEN_SOURCE=500"
         ;;
     *)
+        AC_CHECK_HEADERS([poll.h])
+        AC_CHECK_FUNCS([poll])
         OS_CFLAGS=
         ;;
 esac
-- 
1.6.2

