--- urlview-0.9.orig/urlview.c.orig	2021-02-20 01:44:03.000000000 +1100
+++ urlview-0.9.orig/urlview.c	2021-02-20 01:50:12.000000000 +1100
@@ -46,6 +46,8 @@
 #include <rx/rxposix.h>
 #endif
 
+#include "quote.h"
+
 #define DEFAULT_REGEXP "(((http|https|ftp|gopher)|mailto):(//)?[^ <>\"\t]*|(www|ftp)[0-9]?\.[-a-z0-9.]+)[^ .,;\t\n\r<\">\):]?[^, <>\"\t]*[^ .,;\t\n\r<\">\):]"
 #define DEFAULT_COMMAND "/etc/urlview/url_handler.sh %s"
 #define SYSTEM_INITFILE "/etc/urlview/system.urlview"
