--- src/onmainwindow.cpp.old	2017-02-22 05:11:35.000000000 +0100
+++ src/onmainwindow.cpp	2017-02-23 09:38:07.000000000 +0100
@@ -5708,14 +5708,14 @@ void ONMainWindow::slotSetModMap()
         /* Let's set a reasonable default value if none is provided. */
         if (path_val.isEmpty ()) {
             /* Prefer the default MacPorts prefix. */
-            path_val = "/opt/local/bin:/opt/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/sbin:/usr/X11R6/bin:/opt/X11/bin";
+            path_val = "@@PREFIX@@/bin:@@PREFIX@@/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/sbin:/usr/X11R6/bin:/opt/X11/bin";
             tmp_env.insert ("PATH", path_val);
         }
         else {
             /* Search for and add specific directories to the PATH value, if necessary. */
             QStringList to_back, to_front;
             to_back << "/opt/X11/bin";
-            to_front << "/opt/local/bin" << "/usr/local/bin";
+            to_front << "@@PREFIX@@/bin" << "/usr/local/bin";
 
             path_val = add_to_path (path_val, to_back);
             path_val = add_to_path (path_val, to_front, false);
--- src/pulsemanager.cpp.old	2017-02-22 05:11:35.000000000 +0100
+++ src/pulsemanager.cpp	2017-02-23 09:38:32.000000000 +0100
@@ -70,7 +70,7 @@ PulseManager::PulseManager () : app_dir_
   QString path_val = tmp_env.value ("PATH");
 
   QStringList to_front, to_back;
-  to_front << "/opt/local/bin"; /* MacPorts default prefix, FIXME: might need to make that configurable. */
+  to_front << "@@PREFIX@@/bin"; /* MacPorts default prefix, FIXME: might need to make that configurable. */
   to_back << "/usr/local/bin"; /* Homebrew or random stuff. Probably even both intermingled... */
 
   path_val = add_to_path (path_val, to_back);
@@ -93,7 +93,7 @@ PulseManager::PulseManager () : app_dir_
 
   if (server_binary_.isEmpty ()) {
     search_paths = QStringList ();
-    search_paths << "/opt/local/bin"; /* MacPorts default prefix, FIXME: might need to make that configurable. */
+    search_paths << "@@PREFIX@@/bin"; /* MacPorts default prefix, FIXME: might need to make that configurable. */
 
     server_binary_ = QStandardPaths::findExecutable ("pulseaudio", search_paths);
 
