--- src/x2goutils.cpp.old	2017-02-22 05:11:35.000000000 +0100
+++ src/x2goutils.cpp	2017-02-23 09:32:43.000000000 +0100
@@ -299,7 +299,7 @@ QString add_to_path (const QString &orig
     }
   }
   else {
-    for (int i = (clean_add.size () - 1); i > 0; --i) {
+    for (int i = (clean_add.size () - 1); i >= 0; --i) {
       if (!found[i]) {
         ret.prepend (QString (clean_add[i] + ":"));
       }
