--- a/Makefile	2013-08-17 08:56:24.000000000 -0700
+++ b/Makefile	2013-08-17 08:56:12.000000000 -0700
@@ -82,14 +82,30 @@
 endif
 
 ifneq ($(POPT_DIR),)
-    CFLAGS += -I$(POPT_DIR)
-    LOADLIBES += -L$(POPT_DIR)
+    CFLAGS += -I$(POPT_DIR)/include
+    LOADLIBES += -L$(POPT_DIR)/lib
 endif
 
 ifneq ($(STATEFILE),)
     CFLAGS += -DSTATEFILE=\"$(STATEFILE)\"
 endif
 
+ifneq ($(COMPRESS_COMMAND),)
+    CFLAGS += -DCOMPRESS_COMMAND=\"$(COMPRESS_COMMAND)\"
+endif
+
+ifneq ($(COMPRESS_EXT),)
+    CFLAGS += -DCOMPRESS_EXT=\"$(COMPRESS_EXT)\"
+endif
+
+ifneq ($(UNCOMPRESS_COMMAND),)
+    CFLAGS += -DUNCOMPRESS_COMMAND=\"$(UNCOMPRESS_COMMAND)\"
+endif
+
+ifneq ($(DEFAULT_MAIL_COMMAND),)
+    CFLAGS += -DDEFAULT_MAIL_COMMAND=\"$(DEFAULT_MAIL_COMMAND)\"
+endif
+
 BINDIR = $(BASEDIR)/sbin
 MANDIR ?= $(BASEDIR)/man
 
