--- config.h.in.orig	2017-05-02 06:16:29.000000000 -0500
+++ config.h.in	2017-05-31 14:36:35.000000000 -0500
@@ -1,13 +1,13 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
 /* default compress command */
-#undef COMPRESS_COMMAND
+#define COMPRESS_COMMAND "@COMPRESS@"
 
 /* default compress extension */
-#undef COMPRESS_EXT
+#define COMPRESS_EXT "@COMPRESS_EXT@"
 
 /* default mail command */
-#undef DEFAULT_MAIL_COMMAND
+#define DEFAULT_MAIL_COMMAND "/usr/bin/mail"
 
 /* Define to 1 if you have the `asprintf' function. */
 #undef HAVE_ASPRINTF
@@ -117,7 +117,7 @@
 #undef STDC_HEADERS
 
 /* default uncompress command */
-#undef UNCOMPRESS_COMMAND
+#define UNCOMPRESS_COMMAND "@UNCOMPRESS@"
 
 /* Enable extensions on AIX 3, Interix.  */
 #ifndef _ALL_SOURCE
