--- configure.org	2010-04-16 00:57:24.000000000 +0200
+++ configure	2010-04-16 00:57:41.000000000 +0200
@@ -7546,7 +7546,7 @@
 	# there are some -arch flags.  Note that *ppc* also matches
 	# ppc64.  This check is also rather less than ideal.
 	case "${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS}" in  #(
-	  *-arch*ppc*|*-arch*i386*|*-arch*x86_64*) ac_cv_c_bigendian=universal;;
+	  *-arch*ppc*) ac_cv_c_bigendian=universal;;
 	esac
 else
   $as_echo "$as_me: failed program was:" >&5
