--- autogen.sh.orig	2015-11-30 12:01:21.000000000 -0800
+++ autogen.sh	2015-11-30 12:01:49.000000000 -0800
@@ -13,7 +13,7 @@
 FILE=src/render.c
 
 AUTOCONF_REQUIRED_VERSION=2.54
-AUTOMAKE_REQUIRED_VERSION=1.6
+AUTOMAKE_REQUIRED_VERSION=1.15
 GLIB_REQUIRED_VERSION=2.0.0
 INTLTOOL_REQUIRED_VERSION=0.17
 
@@ -53,7 +53,10 @@
 fi
 
 echo -n "checking for automake >= $AUTOMAKE_REQUIRED_VERSION ... "
-if (automake-1.7 --version) < /dev/null > /dev/null 2>&1; then
+if (automake-1.15 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.15
+   ACLOCAL=aclocal-1.15
+elif (automake-1.7 --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=automake-1.7
    ACLOCAL=aclocal-1.7
 elif (automake-1.8 --version) < /dev/null > /dev/null 2>&1; then
@@ -130,7 +133,7 @@
 echo
 echo "I am going to run ./configure with the following arguments:"
 echo
-echo "  --enable-maintainer-mode $AUTOGEN_CONFIGURE_ARGS $@"
+echo "  $AUTOGEN_CONFIGURE_ARGS $@"
 echo
 
 if test -z "$*"; then
