--- libexec/test-mecab-ipadic-neologd.sh.orig	2015-03-17 08:48:03.000000000 +0900
+++ libexec/test-mecab-ipadic-neologd.sh	2015-03-19 01:17:14.000000000 +0900
@@ -21,8 +21,8 @@
 
 echo "$ECHO_PREFIX Start.."
 
-echo "$ECHO_PREFIX Replace timestamp from 'git clone' date to 'git commit' date"
-${BASEDIR}/../misc/git-set-file-times
+#echo "$ECHO_PREFIX Replace timestamp from 'git clone' date to 'git commit' date"
+#${BASEDIR}/../misc/git-set-file-times
 
 YMD=`ls -c \`find ${BASEDIR}/../seed/mecab-user-dict-seed.*.csv.xz\` | egrep -o '[0-9]{8}' | tail -1`
 if [ ! -e ${BASEDIR}/../build/mecab-ipadic-2.7.0-20070801-neologd-${YMD} ]; then
@@ -48,14 +48,14 @@
 
 echo "$ECHO_PREFIX Get difference between default system dictionary and mecab-ipadic-neologd"
 
-cat /tmp/buzz_phrase| mecab -Owakati > /tmp/buzz_phrase_tokenized_using_defdic
+cat /tmp/buzz_phrase| mecab -Owakati -d @PREFIX@/lib/mecab/dic/ipadic-utf8 > /tmp/buzz_phrase_tokenized_using_defdic
 cat /tmp/buzz_phrase| mecab -Owakati -d ${MECAB_DIC_DIR} > /tmp/buzz_phrase_tokenized_using_neologismdic
 /usr/bin/diff -y -W60 --side-by-side --suppress-common-lines /tmp/buzz_phrase_tokenized_using_defdic /tmp/buzz_phrase_tokenized_using_neologismdic > /tmp/buzz_phrase_tokenized_diff
 
 if [ -s /tmp/buzz_phrase_tokenized_diff ]; then
     echo "$ECHO_PREFIX Tokenize phrase using default system dictionary"
     echo "default system dictonary" > /tmp/buzz_phrase_tokenized_using_defdic
-    cat /tmp/buzz_phrase| mecab -Owakati >> /tmp/buzz_phrase_tokenized_using_defdic
+    cat /tmp/buzz_phrase| mecab -Owakati -d @PREFIX@/lib/mecab/dic/ipadic-utf8 >> /tmp/buzz_phrase_tokenized_using_defdic
 
     echo "$ECHO_PREFIX Tokenize phrase using mecab-ipadic-neologd"
     echo "mecab-ipadic-neologd" > /tmp/buzz_phrase_tokenized_using_neologismdic
