--- ./10-master.conf	2021-12-23 07:27:14.000000000 -0500
+++ ./10-master.conf	2021-12-23 07:29:11.000000000 -0500
@@ -23,6 +23,10 @@
 # everything.
 #default_vsz_limit = 256M
 
+# Workaround for issue with dovecot version 2.3.17 on macOS 12
+# https://www.mail-archive.com/dovecot@dovecot.org/msg84784.html
+default_vsz_limit = 0
+
 # Login user is internally used by login processes. This is the most untrusted
 # user in Dovecot system. It shouldn't have access to anything at all.
 default_login_user = _dovenull
@@ -54,7 +58,11 @@
   process_min_avail = 6
   # If you set service_count=0, you probably need to grow this.
   #vsz_limit = $default_vsz_limit
-  vsz_limit = 2G
+  # vsz_limit = 2G
+
+  # Workaround for issue with dovecot version 2.3.17 on macOS 12
+  # https://www.mail-archive.com/dovecot@dovecot.org/msg84784.html
+  vsz_limit = 0
 
 }
 
