diff -ur src/2geom.orig/toys/intersect-data.cpp src/2geom/toys/intersect-data.cpp
--- src/2geom.orig/toys/intersect-data.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/intersect-data.cpp	2014-05-14 11:20:07.000000000 -0700
@@ -20,7 +20,7 @@
 #include "topology.cpp"
 
 
-static double exp_rescale(double x){ return pow(10, x);}
+static double exp_rescale(double x){ return pow(10.0, x);}
 std::string exp_formatter(double x){ return default_formatter(exp_rescale(x));}
 
 
diff -ur src/2geom.orig/toys/levelsets-test.cpp src/2geom/toys/levelsets-test.cpp
--- src/2geom.orig/toys/levelsets-test.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/levelsets-test.cpp	2014-05-14 11:21:03.000000000 -0700
@@ -9,7 +9,7 @@
 using namespace std;
 using namespace Geom;
 
-static double exp_rescale(double x){ return std::pow(10, x);}
+static double exp_rescale(double x){ return std::pow(10.0, x);}
 std::string exp_formatter(double x){ return default_formatter(exp_rescale(x));}
 
 
diff -ur src/2geom.orig/toys/pencil-2.cpp src/2geom/toys/pencil-2.cpp
--- src/2geom.orig/toys/pencil-2.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/pencil-2.cpp	2014-05-14 11:21:44.000000000 -0700
@@ -982,7 +982,7 @@
         cairo_stroke(cr);
         ostringstream val_s;
         double scale0=(450-adjuster2.pos[1])/300;
-        double curve_precision = pow(10, scale0*5-2);
+        double curve_precision = pow(10.0, scale0*5-2);
         val_s << curve_precision;
         draw_text(cr, adjuster2.pos, val_s.str().c_str());
         cairo_restore(cr);
diff -ur src/2geom.orig/toys/pencil.cpp src/2geom/toys/pencil.cpp
--- src/2geom.orig/toys/pencil.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/pencil.cpp	2014-05-14 11:22:10.000000000 -0700
@@ -275,7 +275,7 @@
       cairo_stroke(cr);
       ostringstream val_s;
       double scale0=(450-adjuster2.pos[1])/300;
-      double curve_precision = pow(10, scale0*5-2);
+      double curve_precision = pow(10.0, scale0*5-2);
       val_s << curve_precision;
       draw_text(cr, adjuster2.pos, val_s.str().c_str());
       
diff -ur src/2geom.orig/toys/sb-to-bez.cpp src/2geom/toys/sb-to-bez.cpp
--- src/2geom.orig/toys/sb-to-bez.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/sb-to-bez.cpp	2014-05-14 11:22:43.000000000 -0700
@@ -330,7 +330,7 @@
       cairo_stroke(cr);
       ostringstream val_s;
       double scale0=(450-adjuster2.pos[1])/300;
-      double curve_precision = pow(10, scale0*5-2);
+      double curve_precision = pow(10.0, scale0*5-2);
       val_s << curve_precision;
       draw_text(cr, adjuster2.pos, val_s.str().c_str());
       
diff -ur src/2geom.orig/toys/sketch-fitter.cpp src/2geom/toys/sketch-fitter.cpp
--- src/2geom.orig/toys/sketch-fitter.cpp	2014-05-14 11:24:53.000000000 -0700
+++ src/2geom/toys/sketch-fitter.cpp	2014-05-14 11:25:25.000000000 -0700
@@ -285,7 +285,7 @@
 
 double exp_rescale(double x)
 {
-    return pow(10, x*5-2);
+    return pow(10.0, x*5-2);
 }
 std::string exp_formatter(double x)
 {
diff -ur src/2geom.orig/toys/smash-intersector.cpp src/2geom/toys/smash-intersector.cpp
--- src/2geom.orig/toys/smash-intersector.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/smash-intersector.cpp	2014-05-14 11:25:51.000000000 -0700
@@ -59,7 +59,7 @@
 
 #define VERBOSE 0
 
-static double exp_rescale(double x){ return ::pow(10, x);}
+static double exp_rescale(double x){ return ::pow(10.0, x);}
 std::string exp_formatter(double x){ return default_formatter(exp_rescale(x));}
 
 
diff -ur src/2geom.orig/toys/sweeper-toy.cpp src/2geom/toys/sweeper-toy.cpp
--- src/2geom.orig/toys/sweeper-toy.cpp	2014-05-14 11:12:00.000000000 -0700
+++ src/2geom/toys/sweeper-toy.cpp	2014-05-14 11:26:34.000000000 -0700
@@ -22,7 +22,7 @@
 
 #include "sweeper.cpp"
 
-double exp_rescale(double x){ return pow(10, x);}
+double exp_rescale(double x){ return pow(10.0, x);}
 std::string exp_formatter(double x){ return default_formatter(exp_rescale(x));}
 
 
