--- CMakeLists.txt.orig	2013-11-25 11:43:20.000000000 -0800
+++ CMakeLists.txt	2013-11-25 11:48:07.000000000 -0800
@@ -30,7 +30,7 @@
 ENDIF(pycairo_FOUND) 
 
 # Set Compiler Flags
-SET(CXX_WARNINGS_FLAGS "-Wall -Wformat -Wformat-security -W -Wpointer-arith -Wcast-align -Wsign-compare -Woverloaded-virtual -Wswitch -Werror=return-type")
+SET(CXX_WARNINGS_FLAGS "-Wall -Wformat -Wformat-security -W -Wpointer-arith -Wcast-align -Wsign-compare -Woverloaded-virtual -Wswitch -Wreturn-type")
 # set the general CXX flags, note that subproject folders may add to this list for themselves
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GTK2_CFLAGS} ${cairo_CFLAGS} ${gsl_CFLAGS} ${pycairo_CFLAGS} ${CXX_WARNINGS_FLAGS} -O2")
 
@@ -106,7 +106,7 @@
 ADD_CUSTOM_TARGET(uninstall
   "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
-ADD_SUBDIRECTORY (src/gtest)
+#ADD_SUBDIRECTORY (src/gtest)
 ADD_SUBDIRECTORY (src/2geom)
 
 # install config.h
