diff --git a/setup.py b/setup.py
index e35127a..da288fd 100644
--- setup.py
+++ setup.py
@@ -47,7 +47,7 @@ setup(
         'Programming Language :: Python :: Implementation :: PyPy',
     ],
     install_requires=[
-        'h11 ~= 0.7.0',  # means: 0.7.x where x >= 0
+        'h11 ~= 0.8.1',  # means: 0.8.x where x >= 1
     ],
     extras_require={
         ':python_version == "2.7" or python_version == "3.3"':
diff --git a/wsproto/connection.py b/wsproto/connection.py
index 44949cb..777f9c8 100644
--- wsproto/connection.py
+++ wsproto/connection.py
@@ -417,8 +417,12 @@ class WSConnection(object):
                 else:
                     # py34 annoyance: doesn't support bytestring formatting
                     params = params.decode("ascii")
-                    extensions.append(('%s; %s' % (name, params))
-                                      .encode("ascii"))
+                    if params == '':
+                        extensions.append(('%s' % (name))
+                                        .encode("ascii"))
+                    else:
+                        extensions.append(('%s; %s' % (name, params))
+                                        .encode("ascii"))
             return b', '.join(extensions)
 
         return None
