--- setup.py
+++ setup.py
@@ -137,10 +137,10 @@ class build_ext(build_ext_base):
         build_ext_base.initialize_options(self)
         self.poppler_version = None
 
-        self.qmake_bin = 'qmake'
+        self.qmake_bin = '%%QMAKE%%'
 
         self.qt_include_dir = None
-        self.pyqt_sip_dir = None
+        self.pyqt_sip_dir = '%%PYQTSIPDIR%%'
         self.pyqt_sip_flags = None
 
     def finalize_options (self):
