--- configure.orig	2022-03-13 04:55:26.000000000 +0800
+++ configure	2022-05-21 21:25:34.000000000 +0800
@@ -804,7 +804,7 @@
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
 runstatedir='${localstatedir}/run'
-includedir='${prefix}/include'
+includedir='${exec_prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
 infodir='${datarootdir}/info'


--- Makefile.am.orig	2022-03-13 04:55:25.000000000 +0800
+++ Makefile.am	2022-05-21 21:55:19.000000000 +0800
@@ -105,7 +105,7 @@
 ./src/include/a68g-stddef.h \
 ./src/include/a68g-transput.h \
 ./src/include/a68g-types.h
-includedir = /usr/local/include/algol68g
+includedir = /@@PREFIX@@/include/algol68g
 include_HEADERS = \
 ./a68g-config.h \
 ./src/include/a68g-common.h \
