--- configure.ac.orig	2022-03-13 04:55:25.000000000 +0800
+++ configure.ac	2022-05-21 18:47:57.000000000 +0800
@@ -216,7 +216,7 @@
 #
 # Mac OS X.
 #
-*86-*-*darwin* | *86_64-*-*darwin*) 
+*86-*-*darwin* | *86_64-*-*darwin* | powerpc*-*-*darwin*) 
   AC_DEFINE(BUILD_BSD, 1, [Define this if DARWIN was detected]) 
   AC_DEFINE(HAVE_IEEE_754, 1, [Define this if IEEE_754 compliant]) 
   AC_MSG_WARN([configuring interpreter-only])

