--- grafana.ini	2023-07-26 01:55:59
+++ grafana.ini	2023-08-04 07:06:50
@@ -15,19 +15,19 @@
 #################################### Paths ###############################
 [paths]
 # Path to where grafana can store temp files, sessions, and the sqlite3 db (if that is used)
-data = data
+data = @DATAPATH@

 # Temporary files in `data` directory older than given duration will be removed
 temp_data_lifetime = 24h

 # Directory where grafana can store logs
-logs = data/log
+logs = @LOGPATH@

 # Directory where grafana will automatically scan and look for plugins
-plugins = data/plugins
+plugins = @DATAPATH@/plugin

 # folder that contains provisioning config files that grafana will apply on startup and while running.
-provisioning = conf/provisioning
+provisioning = @CONFPATH@provisioning

 #################################### Server ##############################
 [server]
@@ -76,7 +76,7 @@
 socket_mode = 0660

 # Unix socket path
-socket = /tmp/grafana.sock
+socket = @SOCKPATH@/grafana.sock

 # CDN Url
 cdn_url =
