--- lib/Makefile.in.orig	2009-03-09 07:46:18.000000000 +1100
+++ lib/Makefile.in	2012-06-13 23:25:16.000000000 +1000
@@ -352,7 +352,7 @@
 AM_CFLAGS = $(GENERAL_CFLAGS)
 libIL_la_CPPFLAGS = -I $(il_include) -I $(all_include)
 libIL_la_CFLAGS = $(AM_CFLAGS) @IL_CFLAGS@
-libIL_la_CXXFLAGS = $(AM_CFLAGS) @IL_CFLAGS@
+libIL_la_CXXFLAGS = `echo $(AM_CFLAGS) | sed 's|-std=gnu99||'` @IL_CFLAGS@
 libIL_la_SOURCES = $(il_src)/il_alloc.c $(il_src)/il_doom.c \
 	$(il_src)/il_stack.c $(il_src)/il_pcx.c $(il_src)/il_rle.c \
 	$(il_src)/il_bits.c $(il_src)/il_endian.c $(il_src)/il_exr.cpp \
