https://github.com/DentonW/DevIL/issues/39
--- data/ILUT.pc.in.orig	2009-03-08 01:10:07.000000000 -0600
+++ data/ILUT.pc.in	2018-01-26 06:25:13.000000000 -0600
@@ -5,6 +5,6 @@
 Description: Developer's Image Library - ILUT part (connection to higher level libraries)
 Version: @VERSION@
 URL: http://openil.sourceforge.net
-Libs: -L@libdir@ -lILUT -ILU -lIL
+Libs: -L@libdir@ -lILUT -lILU -lIL
 Libs.private: @ILUT_LIBS@
 Cflags: -I@includedir@ @IL_CFLAGS@
