--- src-IL/src/il_jp2.c.orig	2009-03-08 00:10:09.000000000 -0700
+++ src-IL/src/il_jp2.c	2017-09-16 05:34:53.000000000 -0700
@@ -459,7 +459,7 @@
 			/* The buffer must be large enough to accommodate maximum
 			  putback. */
 			assert(bufsize > JAS_STREAM_MAXPUTBACK);
-			stream->bufbase_ = JAS_CAST(uchar *, buf);
+			stream->bufbase_ = JAS_CAST(jas_uchar *, buf);
 			stream->bufsize_ = bufsize - JAS_STREAM_MAXPUTBACK;
 		}
 	} else {
