Install manpage.
https://github.com/P-H-C/phc-winner-argon2/pull/273
--- Makefile.orig
+++ Makefile
@@ -19,6 +19,7 @@ RUN = argon2
 BENCH = bench
 GENKAT = genkat
 ARGON2_VERSION ?= ZERO
+MAN = man/argon2.1
 
 # installation parameters for staging area and final installation path
 # Note; if Linux and not Debian/Ubuntu version also add lib override to make command-line
@@ -136,6 +137,7 @@ ifeq ($(KERNEL_NAME), $(filter $(KERNEL_NAME),DragonFly FreeBSD))
 BINARY_REL ?= bin
 INCLUDE_REL ?= include
 LIBRARY_REL ?= lib
+MAN_REL ?= share/man/man1
 PKGCONFIG_REL ?= libdata
 
 else ifeq ($(KERNEL_NAME)-$(MACHINE_NAME), Linux-x86_64)
@@ -144,6 +146,7 @@ else ifeq ($(KERNEL_NAME)-$(MACHINE_NAME), Linux-x86_64)
 BINARY_REL ?= bin
 INCLUDE_REL ?= include
 LIBRARY_REL ?= lib/x86_64-linux-gnu
+MAN_REL ?= share/man/man1
 PKGCONFIG_REL ?= $(LIBRARY_REL)
 
 else
@@ -152,6 +155,7 @@ else
 BINARY_REL ?= bin
 INCLUDE_REL ?= include
 LIBRARY_REL ?= lib
+MAN_REL ?= share/man/man1
 PKGCONFIG_REL ?= $(LIBRARY_REL)
 
 endif
@@ -160,6 +164,7 @@ endif
 INST_INCLUDE = $(DESTDIR)$(PREFIX)/$(INCLUDE_REL)
 INST_LIBRARY = $(DESTDIR)$(PREFIX)/$(LIBRARY_REL)
 INST_BINARY = $(DESTDIR)$(PREFIX)/$(BINARY_REL)
+INST_MAN = $(DESTDIR)$(PREFIX)/$(MAN_REL)
 INST_PKGCONFIG = $(DESTDIR)$(PREFIX)/$(PKGCONFIG_REL)/pkgconfig
 
 # main target
@@ -242,6 +247,8 @@ ifdef LINKED_LIB_SH
 endif
 	$(INSTALL) -d $(INST_BINARY)
 	$(INSTALL) $(RUN) $(INST_BINARY)
+	$(INSTALL) -d $(INST_MAN)
+	$(INSTALL) -m 0644 $(MAN) $(INST_MAN)
 	$(INSTALL) -d $(INST_PKGCONFIG)
 	$(INSTALL) -m 0644 $(PC_NAME) $(INST_PKGCONFIG)
 
@@ -250,4 +257,5 @@ uninstall:
 	cd $(INST_INCLUDE) && rm -f $(notdir $(HEADERS))
 	cd $(INST_LIBRARY) && rm -f $(notdir $(LIBRARIES) $(LINKED_LIB_SH))
 	cd $(INST_BINARY) && rm -f $(notdir $(RUN))
+	cd $(INST_MAN) && rm -f $(notdir $(MAN))
 	cd $(INST_PKG_CONFIG) && rm -f $(notdir $(PC_NAME))
