From f8e94f86cc8ca0ff24f5d7051a4541df44fe0f30 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 8 Feb 2023 14:59:11 +0700
Subject: [PATCH 3/3] Do not build apps, they depend on chem

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a39b07429..99c208960 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,6 @@ include(CMakeDependentOption)
 include(AddMADLibrary)
 include(AddMADExecutable)
 include(AddUnittests)
-include(AddScriptedTests)
 include(CMakePackageConfigHelpers)
 include(CopyTargetProperties)
 include(FeatureSummary)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 01d359790..b4cc82453 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,10 +1,7 @@
 # src
 
 add_subdirectory(madness)
-if(NOT MADNESS_BUILD_MADWORLD_ONLY AND LAPACK_FOUND)
-	add_subdirectory(apps)
-	add_subdirectory(examples)
-endif()
+
 install(FILES madness.h
     	DESTINATION "${MADNESS_INSTALL_INCLUDEDIR}"
     	COMPONENT madness)
