--- makefile.orig	2012-10-01 00:08:42.000000000 -0400
+++ makefile	2012-10-01 00:20:06.000000000 -0400
@@ -6,7 +6,8 @@
 # Also, before installing, you should change the string "JGRAPH_DIR" in 
 # jgraph.1 to be the directory containing the example jgraphs.
 
-CC = gcc -LCC
+CC := @@CC@@
+CFLAGS = -DLCC -O3 -Wall
 
 OBJS = draw.o \
 		edit.o \
@@ -49,23 +50,22 @@ examples: $(EXAMPLES)
 
 .SUFFIXES: .c .o .h .jgr .jps .eps .tex .dvi
 
-.jgr.jps:
-	jgraph -P < $*.jgr > $*.jps
+%.jps: %.jgr
+	jgraph -P <$< >$@
 
-.jgr.eps:
-	jgraph < $*.jgr > $*.eps
-
-.tex.dvi:
-	latex $*.tex
-.c.o: 
-	$(CC)  -c -g $*.c
+%.eps: %.jgr
+	jgraph <$< >$@
 
+%.dvi: %.tex
+	latex $*
+%.o: %.c 
+	$(CC) -c -o $@ $(CFLAGS) $<
 
 jgraph: $(OBJS)
-	$(CC) -g $(OBJS) -lm -o jgraph
+	$(CC) $(OBJS) -o $@
 
 sin:	sin.o
-	cc -o sin sin.o -lm
+	$(CC) $^ -o $@
 
 ad.jps: cube.eps disk.eps
 bailey.jps: bailey.pts
@@ -93,4 +93,4 @@ process.o: process.c jgraph.h list.h pri
 show.o: show.c jgraph.h list.h prio_list.h 
 token.o: token.c list.h
 clean: 
-	rm -f *.o a.out *.jps
+	$(RM) *.o a.out *.jps jgraph
