--- include/osg/Math.orig	2013-05-04 18:33:49.000000000 -0700
+++ include/osg/Math	2013-05-04 18:34:56.000000000 -0700
@@ -38,6 +38,9 @@
         #define APPLE_PRE_10_2
     #endif
 #endif
+#if !defined(MAC_OS_X_VERSION_10_7) || (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_7) 
+    #define APPLE_PRE_10_7 
+#endif 
 #endif
 
 #if defined(_MSC_VER)
@@ -45,7 +48,7 @@
 #endif
 
 #if defined (sun) || \
-    defined (__APPLE__) || \
+    defined (APPLE_PRE_10_7) || \
     (defined (_AIX) && defined (__GNUC__))
 
     #include <float.h>
