diff -ru bonobo.orig/bonobo.defs bonobo/bonobo.defs
--- bonobo.orig/bonobo.defs	2009-08-08 20:51:13.000000000 +0200
+++ bonobo/bonobo.defs	2009-08-08 20:52:00.000000000 +0200
@@ -777,11 +777,6 @@
   (return-type "gboolean")
 )
 
-(define-function setup_x_error_handler
-  (c-name "bonobo_setup_x_error_handler")
-  (return-type "none")
-)
-
 (define-function orb
   (c-name "bonobo_orb")
   (return-type "CORBA_ORB")
diff -ru bonobo.orig/bonoboui.defs bonobo/bonoboui.defs
--- bonobo.orig/bonoboui.defs	2009-08-08 20:51:13.000000000 +0200
+++ bonobo/bonoboui.defs	2009-08-08 20:52:37.000000000 +0200
@@ -1502,11 +1502,6 @@
   )
 )
 
-(define-function setup_x_error_handler
-  (c-name "bonobo_setup_x_error_handler")
-  (return-type "none")
-)
-
 (define-function debug_shutdown
   (c-name "bonobo_ui_debug_shutdown")
   (return-type "int")
diff -ru bonobo.orig/bonobouimodule.c bonobo/bonobouimodule.c
--- bonobo.orig/bonobouimodule.c	2009-08-08 20:51:13.000000000 +0200
+++ bonobo/bonobouimodule.c	2009-08-08 20:54:21.000000000 +0200
@@ -25,15 +25,6 @@
 
     init_pygtk();
 
-    /* we don't call bonobo_ui_init() here, as all it does is call
-     * bonobo_init() (done by the bonobo module we import),
-     * gnome_program_init() (meant to be done by the user), and
-     * gtk_init() (done by the gtk module we import), and call
-     * bonobo_setup_x_error_handler().  This last call is all that is
-     * left. */
-    bonobo_setup_x_error_handler();
-
-
     pybonoboui_register_classes(d);
 
     /* add anything else to the module dictionary (such as constants) */
