--- pgadmin/pgAdmin3.cpp.orig
+++ pgadmin/pgAdmin3.cpp
@@ -1304,7 +1304,7 @@ wxString pgAdmin3::LocatePath(const wxString &pathToFind, const bool isFile)
 	// When using wxStandardPaths on OSX, wx default to the unix,
 	// not to the mac variants. Therefore, we request wxStandardPathsCF
 	// directly.
-	wxStandardPathsCF stdPaths ;
+	wxStandardPaths stdPaths = wxStandardPaths::Get();
 	dataDir = stdPaths.GetDataDir() ;
 
 #else // other *ixes
--- pgadmin/utils/sysSettings.cpp.orig
+++ pgadmin/utils/sysSettings.cpp
@@ -740,7 +740,7 @@ wxString sysSettings::GetConfigFile(configFileName cfgname)
 {
 	if (cfgname == PGPASS)
 	{
-		wxStandardPaths stdp;
+		wxStandardPaths stdp = wxStandardPaths::Get();
 		wxString fname = stdp.GetUserConfigDir();
 #ifdef WIN32
 		fname += wxT("\\postgresql");
@@ -770,7 +770,7 @@ wxString sysSettings::GetFavouritesFile()
 {
 	wxString s, tmp;
 
-	wxStandardPaths stdp;
+	wxStandardPaths stdp = wxStandardPaths::Get();
 	tmp = stdp.GetUserConfigDir();
 #ifdef WIN32
 	tmp += wxT("\\postgresql");
@@ -791,7 +791,7 @@ wxString sysSettings::GetMacrosFile()
 {
 	wxString s, tmp;
 
-	wxStandardPaths stdp;
+	wxStandardPaths stdp = wxStandardPaths::Get();
 	tmp = stdp.GetUserConfigDir();
 #ifdef WIN32
 	tmp += wxT("\\postgresql");
@@ -812,7 +812,7 @@ wxString sysSettings::GetHistoryFile()
 {
 	wxString s, tmp;
 
-	wxStandardPaths stdp;
+	wxStandardPaths stdp = wxStandardPaths::Get();
 	tmp = stdp.GetUserConfigDir();
 #ifdef WIN32
 	tmp += wxT("\\postgresql");
