--- qmake/Makefile.unix.orig	2008-01-16 06:08:58.000000000 +1100
+++ qmake/Makefile.unix	2010-10-31 06:53:45.000000000 +1100
@@ -49,7 +49,7 @@
 	-cp -f @BUILD_PATH@/bin/qmake @QT_INSTALL_BINS@
 	[ -d @QT_INSTALL_DATA@ ] || mkdir -p @QT_INSTALL_DATA@
 	-cp -R -f @SOURCE_PATH@/mkspecs @QT_INSTALL_DATA@
-	-ln -s -f @QMAKESPEC@ @QT_INSTALL_DATA@/mkspecs/default
+#	-ln -s -f @QMAKESPEC@ @QT_INSTALL_DATA@/mkspecs/default
 
 clean::
 	rm -f $(OBJS) $(QOBJS)
