--- mkspecs/macx-g++/qmake.conf.orig	2010-10-31 04:42:49.000000000 +1100
+++ mkspecs/macx-g++/qmake.conf	2010-10-31 04:46:16.000000000 +1100
@@ -8,7 +8,7 @@
 CONFIG			+= qt warn_on release resource_fork incremental global_init_link_order lib_version_first plugin_no_soname link_prl
 QMAKE_INCREMENTAL_STYLE = sublibs
 
-QMAKE_CC		= cc
+QMAKE_CC		= __CC__
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -17,7 +17,7 @@
 QMAKE_YACC_HEADER       = $base.tab.h
 QMAKE_YACC_SOURCE       = $base.tab.c
 QMAKE_RESOURCE		= /Developer/Tools/Rez
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -pipe __ARCHFLAGS__
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
@@ -27,7 +27,7 @@
 QMAKE_EXTENSION_SHLIB	= dylib
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
-QMAKE_CXX		= c++
+QMAKE_CXX		= __CXX__
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
@@ -45,7 +45,7 @@
 
 QMAKE_LINK		= $$QMAKE_CXX
 QMAKE_LINK_SHLIB	= $$QMAKE_CXX
-QMAKE_LFLAGS		= -headerpad_max_install_names
+QMAKE_LFLAGS		= -headerpad_max_install_names __ARCHFLAGS__
 #QMAKE_LFLAGS_PREBIND    = -seg1addr 0x90000000
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
--- config.tests/mac/mac_version.test.orig	2010-10-31 04:42:49.000000000 +1100
+++ config.tests/mac/mac_version.test	2010-10-31 04:54:48.000000000 +1100
@@ -27,9 +27,9 @@
 
 COMPILE_ERROR=yes
 if [ "$VERBOSE" = "yes" ]; then
-    c++ $TSTFILE -o tst_mac_version -framework Carbon >&2 && COMPILE_ERROR=no
+    __CXX__ $TSTFILE -o tst_mac_version __ARCHFLAGS__ -framework Carbon >&2 && COMPILE_ERROR=no
 else
-    c++ $TSTFILE -o tst_mac_version -framework Carbon >/dev/null 2>&1 && COMPILE_ERROR=no
+    __CXX__ $TSTFILE -o tst_mac_version __ARCHFLAGS__ -framework Carbon >/dev/null 2>&1 && COMPILE_ERROR=no
 fi
 
 # done
