--- common/video/iso-mpeg4/src/sys_tps_enhcbuf.cpp.orig	2005-05-04 13:56:00.000000000 -0600
+++ common/video/iso-mpeg4/src/sys_tps_enhcbuf.cpp	2009-11-18 20:22:28.000000000 -0700
@@ -46,6 +46,21 @@
 #include "vopses.hpp"
 #include "tps_enhcbuf.hpp"
 
+
+Void set_modes(VOLMode*, VOPMode*);
+Void write420_sep(Int, char*, PixelC*, PixelC*, PixelC*, Int, Int);
+Void write420_jnt(FILE*, PixelC*, PixelC*, PixelC*, Int, Int);
+Void write420_jnt_withMask(FILE*, PixelC*, PixelC*, PixelC*, PixelC*, PixelC*, Int, Int);
+Void write_seg_test(Int, char*, PixelC*, Int, Int);
+Void bg_comp_each(PixelC*, PixelC*, PixelC*, PixelC*, PixelC*, PixelC*, Int, Int, Int, Int, Int, Int);
+Void bg_comp_each(PixelC*, PixelC*, PixelC*, PixelC*, Int, Int, Int, CRct);
+Void bg_comp_each_mode3(PixelC*, PixelC*, PixelC*, PixelC*, Int, Int, Int, CRct);
+Void inv_convertYuv(const CVOPU8YUVBA*, PixelC*, PixelC*, PixelC*, Int, Int);
+Void convertYuv(const CVOPU8YUVBA*, PixelC*, PixelC*, PixelC*, Int, Int);
+Void convertSeg(const CVOPU8YUVBA*, PixelC*, PixelC*, Int, Int, Int, Int, Int, Int);
+Void pre_pad(PixelC*, PixelC*, int, int);
+
+
 CEnhcBuffer::CEnhcBuffer(Int iSessionWidth, Int iSessionHeight)
 {
 	Int iNumMBX = iSessionWidth / MB_SIZE;
