--- common/video/iso-mpeg4/src/vtc_ztq_decQM.cpp.orig	2005-05-04 13:56:01.000000000 -0600
+++ common/video/iso-mpeg4/src/vtc_ztq_decQM.cpp	2009-11-19 04:23:19.000000000 -0700
@@ -193,7 +193,7 @@
   err=0;
 
   /* loop through DC */
-  noteDetail("Inverse Quantizing DC band....");
+  noteDetail((char*)"Inverse Quantizing DC band....");
   for (x = 0; x < mzte_codec.m_iDCWidth; ++x)
     for (y = 0; y < mzte_codec.m_iDCHeight; ++y)
     {
@@ -202,7 +202,7 @@
       COEFF_RECVAL(x,y,c) = COEFF_VAL(x,y,c) * mzte_codec.m_iQDC[c];
     }
 
-  noteDetail("Completed inverse Quantizing DC bands.");
+  noteDetail((char*)"Completed inverse Quantizing DC bands.");
 
   return err;
 }
@@ -243,13 +243,13 @@
   err=0;
 
   /* loop through DC */
-  noteDetail("Inverse quantizing AC bands....");
+  noteDetail((char*)"Inverse quantizing AC bands....");
   for (x = 0; x < mzte_codec.m_iDCWidth; ++x)
     for (y = 0; y < mzte_codec.m_iDCHeight; ++y)
     {
       if ((nc = findChild(x, y, xc, yc,c)) != 3)
       {
-	noteError("DC band coefficient has %d children instead of 3.", nc);
+	noteError((char*)"DC band coefficient has %d children instead of 3.", nc);
 	exit(-1);
       }
 	
@@ -258,7 +258,7 @@
       iQuantizeCoeffs(xc[2], yc[2],c);
     }
 
-  noteDetail("Completed inverse quantizing of AC bands.");
+  noteDetail((char*)"Completed inverse quantizing of AC bands.");
 
   return err;
 }
@@ -276,7 +276,7 @@
   err=0;
 
   /* loop through DC */
-  noteDetail("Inverse quantizing AC bands (difference)....");
+  noteDetail((char*)"Inverse quantizing AC bands (difference)....");
 
   hend=mzte_codec.m_SPlayer[c].height;
   wend=mzte_codec.m_SPlayer[c].width;
@@ -301,7 +301,7 @@
     for(w=0;w<wend;w++)
       iQuantizeCoeff(w,h,c);
 
-  noteDetail("Completed inverse quantizing of AC bands.");
+  noteDetail((char*)"Completed inverse quantizing of AC bands.");
 
   return(err);
 
@@ -344,13 +344,13 @@
   err=0;
 
   /* loop through DC */
-  noteDetail("Updating state of AC bands....");
+  noteDetail((char*)"Updating state of AC bands....");
   for (x = 0; x < mzte_codec.m_iDCWidth; ++x)
     for (y = 0; y < mzte_codec.m_iDCHeight; ++y)
     {
       if ((nc = findChild(x, y, xc, yc,c)) != 3)
       {
-	noteError("DC band coefficient has %d children instead of 3.", nc);
+	noteError((char*)"DC band coefficient has %d children instead of 3.", nc);
 	exit(-1);
       }
       
@@ -359,7 +359,7 @@
       updateCoeffAndDescState(xc[2], yc[2], c);
     }
   
-  noteDetail("Completed updating state of AC bands.");
+  noteDetail((char*)"Completed updating state of AC bands.");
 
   return err;
 }
@@ -399,7 +399,7 @@
   err=0;
 
   /* loop through DC */
-  noteDetail("Updating state of AC bands (difference)....");
+  noteDetail((char*)"Updating state of AC bands (difference)....");
 
   hstart=mzte_codec.m_SPlayer[c].height/2;
   wstart=mzte_codec.m_SPlayer[c].width/2;
@@ -414,7 +414,7 @@
     for(w=0;w<wend;w++)
       updateCoeffAndDescState(w, h, c);
    
-  noteDetail("Completed updating state of AC bands.");
+  noteDetail((char*)"Completed updating state of AC bands.");
 
   return err;
 }
