--- common/video/iso-mpeg4/src/tools_entropy_bitstrm.cpp.orig	2005-09-08 16:02:55.000000000 -0600
+++ common/video/iso-mpeg4/src/tools_entropy_bitstrm.cpp	2009-11-18 23:28:07.000000000 -0700
@@ -60,6 +60,9 @@
 #define new DEBUG_NEW				   
 #endif // __MFC_
 
+UInt getbit(UInt, UInt, UInt);
+Void print_bit(UInt, UInt, UInt);
+
 UInt getbit (UInt data, UInt position, UInt num) // get the num-bit field of x starting from position p
 {
 	return ((data >> (position + 1 - num)) & ~(~0 << num));
