--- common/video/iso-mpeg4/src/type_basic.cpp.orig	2005-05-04 13:56:00.000000000 -0600
+++ common/video/iso-mpeg4/src/type_basic.cpp	2009-11-19 00:56:11.000000000 -0700
@@ -317,13 +317,13 @@
 	iHalfY = m_vctTrueHalfPel.y - iMVY * 2;
 }
 
-Void CMotionVector::setToZero (Void)
+Void CMotionVector::setToZero ()
 {
 	memset (this, 0, sizeof (*this));
 }
 
 // RRV insertion
-Void CMotionVector::scaleup (Void)
+Void CMotionVector::scaleup ()
 {
 	if(m_vctTrueHalfPel.x == 0){
 		m_vctTrueHalfPel_x2.x = 0;
