--- common/video/iso-mpeg4/src/sys_decoder_vopsedec.cpp.orig	2005-06-17 14:07:56.000000000 -0600
+++ common/video/iso-mpeg4/src/sys_decoder_vopsedec.cpp	2009-11-17 18:50:55.000000000 -0700
@@ -374,7 +374,7 @@
 #endif
 	  m_pistrm = open(pchStrFile, O_RDONLY);
 		if(m_pistrm < 0)
-			fatal_error("Can't open bitstream file");
+			fatal_error((char *)"Can't open bitstream file");
 #if 0
 	}
 	else {
@@ -602,7 +602,7 @@
 			m_iNumBitsTimeIncr,
 			iDisplayWidth,
 			iDisplayHeight,
-			"",
+			(char *)"",
 			m_volmd.bNewpredSegmentType,
 			m_volmd.fAUsage,
 			m_volmd.bShapeOnly,
@@ -1496,7 +1496,7 @@
 				  m_volmd.bInterCAE ||
 				  m_volmd.bNoUpdate ||
 				  m_volmd.bUpsampling)) {
-				fatal_error("Shape complexity estimation is enabled,\nbut no corresponding flag is enabled.");
+				fatal_error((char *)"Shape complexity estimation is enabled,\nbut no corresponding flag is enabled.");
 			}
 		}
 		else
@@ -1520,7 +1520,7 @@
 				  m_volmd.bInterBlocks ||
 				  m_volmd.bInter4vBlocks ||
 				  m_volmd.bNotCodedBlocks)) {
-				fatal_error("Texture complexity estimation set 1 is enabled,\nbut no corresponding flag is enabled.");
+				fatal_error((char *)"Texture complexity estimation set 1 is enabled,\nbut no corresponding flag is enabled.");
 			}
 		}
 		else
@@ -1542,7 +1542,7 @@
 				  m_volmd.bDCTLines ||
 				  m_volmd.bVLCSymbols ||
 				  m_volmd.bVLCBits)) {
-				fatal_error("Texture complexity estimation set 2 is enabled,\nbut no corresponding flag is enabled.");
+				fatal_error((char *)"Texture complexity estimation set 2 is enabled,\nbut no corresponding flag is enabled.");
 			}
 		}
 		else
@@ -1565,7 +1565,7 @@
 				  m_volmd.bForwBackMCQ ||
 				  m_volmd.bHalfpel2 ||
 				  m_volmd.bHalfpel4)) {
-				fatal_error("Motion complexity estimation is enabled,\nbut no corresponding flag is enabled.");
+				fatal_error((char *)"Motion complexity estimation is enabled,\nbut no corresponding flag is enabled.");
 			}
 		}
 		else
@@ -1587,7 +1587,7 @@
 				m_volmd.bQuarterpel = m_pbitstrmIn -> getBits (1);
 				if (!(m_volmd.bSadct ||
 					  m_volmd.bQuarterpel)) {
-					fatal_error("Version 2 complexity estimation is enabled,\nbut no corresponding flag is enabled.");
+					fatal_error((char *)"Version 2 complexity estimation is enabled,\nbut no corresponding flag is enabled.");
 				}
 			}
 			else
@@ -1608,7 +1608,7 @@
 		    m_volmd.bTextureComplexityEstimationSet2Disable &&
 		    m_volmd.bMotionCompensationComplexityDisable &&
 			m_volmd.bVersion2ComplexityEstimationDisable) {
-			fatal_error("Complexity estimation is enabled,\nbut no correponding flag is enabled.");
+			fatal_error((char *)"Complexity estimation is enabled,\nbut no correponding flag is enabled.");
 		}
 		// END: Complexity Estimation syntax support - Update version 2
 	}
