--- configure.orig	2010-09-14 16:40:50.000000000 -0700
+++ configure	2010-09-14 16:41:40.000000000 -0700
@@ -18805,7 +18805,7 @@
 # the --export-dynamic flag. pkg-config does not provide it, and libtool
 # knows what to do with -export-dynamic.
 # (We need this e.g., for <object class="DasherEditor">)
-	GTK2_LIBS="$GTK2_LIBS -export-dynamic"
+#	GTK2_LIBS="$GTK2_LIBS -export-dynamic"
 
 
 pkg_failed=no
