--- src/PythonTypes.cpp.org	2014-05-05 18:52:04.000000000 +0200
+++ src/PythonTypes.cpp	2018-10-23 14:04:48.000000000 +0200
@@ -49,7 +49,7 @@
 #endif
 #define DONT_HAVE_SYS_STAT_H
 #undef HAVE_STAT_H
-#if defined(__APPLE__) && defined(__MACH__)
+#if defined(__APPLE__) && defined(__MACH__) && defined(HAVE_PYTHON_OSX_FRAMEWORK)
 #include <Python/Python.h>
 #include <Python/pythonrun.h>
 #include <Python/ceval.h>
