--- include/H3D/H3DApi.cmake.org	2014-05-13 15:10:41.000000000 +0200
+++ include/H3D/H3DApi.cmake	2018-10-23 13:38:00.000000000 +0200
@@ -117,6 +117,7 @@
 /// Undef if you do not have Python(www.python.org) installed.
 /// PythonScript node will then not be supported.
 #cmakedefine HAVE_PYTHON
+#cmakedefine HAVE_PYTHON_OSX_FRAMEWORK
 
 /// Undef if you do not have libcurl(http://sourceforge.net/projects/curl/)
 /// installed. URLs using protocols like http and ftp will then not be 
