--- build/CMakeLists.txt.org	2014-04-16 14:14:26.000000000 +0200
+++ build/CMakeLists.txt	2018-10-23 13:33:45.000000000 +0200
@@ -265,37 +265,37 @@
 ENDIF(ZLIB_FOUND)
 
 # Optional extra library. Add python bindings to H3D API, allow for extended prototyping without using C++.
-FIND_PACKAGE(PythonLibs)
-IF( PYTHON_INCLUDE_PATH AND PYTHON_LIBRARIES)
-  SET(PYTHON_FOUND 1)
-ELSE(PYTHON_INCLUDE_PATH AND PYTHON_LIBRARIES)
-  SET(PYTHON_FOUND 0)
-ENDIF(PYTHON_INCLUDE_PATH AND PYTHON_LIBRARIES)
-
-# Report the results.
-IF(NOT PYTHON_FOUND)
-  SET(PYTHON_DIR_MESSAGE
-    "PYTHON was not found. Make sure PYTHON_LIBRARY and PYTHON_INCLUDE_PATH are set.")
-  IF(NOT PYTHON_FIND_QUIETLY)
-    MESSAGE(STATUS "${PYTHON_DIR_MESSAGE}")
-  ELSE(NOT PYTHON_FIND_QUIETLY)
-    IF(PYTHON_FIND_REQUIRED) 
-      MESSAGE(FATAL_ERROR "${PYTHON_DIR_MESSAGE}")
-    ENDIF(PYTHON_FIND_REQUIRED)
-  ENDIF(NOT PYTHON_FIND_QUIETLY)
-ENDIF(NOT PYTHON_FOUND)
-
-IF(PYTHON_FOUND)
-  SET(HAVE_PYTHON 1)
-  INCLUDE_DIRECTORIES( ${PYTHON_INCLUDE_PATH} ) 
-IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  # pyconfig.h is put in different directory from Ubuntu 13.04 (raring) 
-  # and CMake FindPythonLibs module is not updated for this yet. 
-  # Adding it explicitly here in the mean time.
-  INCLUDE_DIRECTORIES( /usr/include/${CMAKE_LIBRARY_ARCHITECTURE}/python2.7 ) 
-ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+IF ( APPLE )
+  SET( CMAKE_FIND_FRAMEWORK_FIRST )
+ENDIF()
+
+FIND_PACKAGE( PythonLibs 2 )
+
+IF( PYTHONLIBS_FOUND )
+  SET( HAVE_PYTHON 1)
+  SET( HAVE_PYTHON_OSX_FRAMEWORK )
+  include_directories( ${PYTHON_INCLUDE_DIRS})
+  IF(APPLE)
+    FOREACH( dir ${PYTHON_INCLUDE_DIRS})
+      IF( dir MATCHES "/System/Library/Frameworks/Python[.]framework")
+        SET( HAVE_PYTHON_OSX_FRAMEWORK 1)
+        BREAK()
+      ENDIF()
+    ENDFOREACH()
+  ENDIF()
+
+  IF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    # pyconfig.h is put in different directory from Ubuntu 13.04 (raring) 
+    # and CMake FindPythonLibs module is not updated for this yet. 
+    # Adding it explicitly here in the mean time.
+    INCLUDE_DIRECTORIES( /usr/include/${CMAKE_LIBRARY_ARCHITECTURE}/python2.7 ) 
+  ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
   SET(optionalLibs ${optionalLibs} ${PYTHON_LIBRARIES} )
-ENDIF(PYTHON_FOUND)
+
+  IF( PYTHONLIBS_VERSION_STRING VERSION_LESS "3.0.0" AND CMAKE_COMPILER_IS_GNUCXX)
+    add_definitions(-fno-strict-aliasing)
+  ENDIF()
+ENDIF()
 
 # Optional extra library. Used for sound in H3D API.
 FIND_PACKAGE(H3DOpenAL)
