--- configure.ac
+++ configure.ac
@@ -490,7 +490,7 @@ dnl ######################
 dnl checking for iniparser
 dnl ######################
 
-  PKG_CHECK_MODULES(INIPARSER, libiniparser, have_iniparser_pkg=yes, have_iniparser_pkg=no)
+  PKG_CHECK_MODULES(INIPARSER, iniparser, have_iniparser_pkg=yes, have_iniparser_pkg=no)
   if [[ $have_iniparser_pkg = "yes" ]] ; then
     LIBS="$LIBS $INIPARSER_LIBS"
     CPPFLAGS="$CPPFLAGS $INIPARSER_CFLAGS"
@@ -500,12 +500,6 @@ dnl ######################
     AC_CHECK_LIB(iniparser,iniparser_load, have_iniparser=yes, have_iniparser=no)
     if [[ $have_iniparser = "yes" ]] ; then
       LIBS="$LIBS -liniparser"
-      if [[ $build_mac = "yes" ]] ; then
-        CPPFLAGS="$CPPFLAGS -I/usr/local/include/iniparser/"
-        CPPFLAGS="$CPPFLAGS -I/opt/homebrew/include/iniparser/"
-      else
-        CPPFLAGS="$CPPFLAGS -I/usr/include/iniparser"
-      fi
       AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <iniparser.h>]],
         [[dictionary* ini;
         const char *keys[3];
