--- configure.orig	2009-09-02 07:55:42.000000000 +1000
+++ configure	2009-09-02 08:02:15.000000000 +1000
@@ -26722,7 +26722,7 @@
 { $as_echo "$as_me:$LINENO: checking for darwin" >&5
 $as_echo_n "checking for darwin... " >&6; }
 case $host_os in
-	rhapsody* | darwin1*)
+	rhapsody* | darwin1.*)
 		{ $as_echo "$as_me:$LINENO: result: yes, but too old" >&5
 $as_echo "yes, but too old" >&6; }
 		{ { $as_echo "$as_me:$LINENO: error: This platform is not supported, please
