--- configure.ac	2014-12-15 17:56:52.000000000 -0500
+++ configure.ac	2015-03-01 17:30:54.000000000 -0500
@@ -467,21 +467,11 @@
                                        [ GUILE1_PROGS ],
                                        [ GUILE_PROGS ])])
                   ], [
-  PKG_CHECK_MODULES(GUILE,
-                  [guile-2.0 >= 2.0.0],
-                  [m4_ifdef([GUILE_PROGS],
-                            [ GUILE_PROGS
-                              gnc_have_guile_2=yes ],
-                            [ AC_MSG_ERROR([ guile 2 is found on your system, but appears
-    to export different autoconf macros than what we expected. Please report this as a bug
-    in GnuCash, so we can fix this for your platform.]) ])
-                  ],
-                  [AC_MSG_ERROR([
+  AC_MSG_ERROR([
     guile does not appear to be installed correctly, or is not in the
     correct version range.  Perhaps you have not installed the guile
     development packages?  Gnucash requires at least version 1.8.5 to build.
   ])])
-])
 
 AM_CONDITIONAL(GNC_HAVE_GUILE_2, test "${gnc_have_guile_2}" = yes)
 AC_SUBST(GUILE_EFFECTIVE_VERSION)
