--- gnucash/environment.in.orig	2020-09-29 18:44:53.000000000 -0700
+++ gnucash/environment.in	2020-09-29 18:59:04.000000000 -0700
@@ -67,7 +67,7 @@
 GNC_LIBRARY_PATH={SYS_LIB};{GNC_LIB}
 LD_LIBRARY_PATH={GNC_LIBRARY_PATH};{LD_LIBRARY_PATH}
 # The same, but for GnuCash on OS X
-DYLD_LIBRARY_PATH={GNC_LIBRARY_PATH};{DYLD_LIBRARY_PATH}
+DYLD_LIBRARY_PATH={GNC_LIB};{DYLD_LIBRARY_PATH}
 
 # For changing the language GnuCash uses, first refer to
 # to https://wiki.gnucash.org/wiki/Locale_Settings.
