--- ./bindings/python/tests/CMakeLists.txt.orig	2020-06-28 10:25:30.000000000 -0700
+++ ./bindings/python/tests/CMakeLists.txt	2020-06-28 10:34:52.000000000 -0700
@@ -5,7 +5,7 @@
     # Because it hasn't been built yet
     set(test_core_dir ${CMAKE_BINARY_DIR}/common/test-core)
   endif()
-  add_custom_target(test-python-bindings ALL DEPENDS unittest_support gnucash-core-c-build gnucash-core-c-py swig-app-utils-python)
+  add_custom_target(test-python-bindings ALL DEPENDS unittest_support gnucash-core-c-build gnucash-core-c-py)
   add_dependencies(check test-python-bindings)
   add_test(python-bindings ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/runTests.py.in)
   set_property(TEST python-bindings PROPERTY ENVIRONMENT
