diff --git bindings/python/CMakeLists.txt bindings/python/CMakeLists.txt
index 7bf9164..3d8a033 100644
--- bindings/python/CMakeLists.txt
+++ bindings/python/CMakeLists.txt
@@ -57,7 +57,7 @@ if(WITH_PYTHON)
     ${CMAKE_SOURCE_DIR}/libgnucash/core-utils
     ${CMAKE_SOURCE_DIR}/libgnucash/gnc-module
     ${GLIB_INCLUDE_DIRS}
-    ${PYTHON_INCLUDE_DIRS}
+    ${PYTHON_INCLUDE_DIR}
   )
 
   add_library(gnucash_core_c MODULE ${SWIG_GNUCASH_CORE_C})
diff --git common/test-core/CMakeLists.txt common/test-core/CMakeLists.txt
index a2fe5b1..9cd9163 100644
--- common/test-core/CMakeLists.txt
+++ common/test-core/CMakeLists.txt
@@ -46,7 +46,7 @@ target_link_libraries(test-core-guile test-core ${GUILE_LDFLAGS} ${GLIB2_LDFLAGS
 if (WITH_PYTHON)
   add_library(unittest_support MODULE ${SWIG_UNITTEST_SUPPORT_PYTHON_C})
   target_link_libraries(unittest_support test-core ${PYTHON_LIBRARIES})
-  target_include_directories(unittest_support PRIVATE ${PYTHON_INCLUDE_DIRS})
+  target_include_directories(unittest_support PRIVATE ${PYTHON_INCLUDE_DIR})
   set_target_properties(unittest_support PROPERTIES PREFIX "_")
   if (HAVE_STRINGOP_TRUNCATION)
     target_compile_options(unittest_support PRIVATE -Wno-error=stringop-truncation)
diff --git libgnucash/app-utils/CMakeLists.txt libgnucash/app-utils/CMakeLists.txt
index 50eedc5..f52bb1e 100644
--- libgnucash/app-utils/CMakeLists.txt
+++ libgnucash/app-utils/CMakeLists.txt
@@ -120,7 +120,7 @@ if (WITH_PYTHON)
 
   target_include_directories (sw_app_utils
     PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
-    PRIVATE ${app_utils_ALL_INCLUDES} ${PYTHON_INCLUDE_DIRS}
+    PRIVATE ${app_utils_ALL_INCLUDES} ${PYTHON_INCLUDE_DIR}
   )
   target_compile_definitions (sw_app_utils PRIVATE -DG_LOG_DOMAIN=\"gnc.app-utils\")
 
