--- gdft.c.orig
+++ gdft.c
@@ -1402,7 +1402,7 @@ fprintf(stderr,"dpi=%d,%d metric_res=%d ptsize=%g\n",hdpi,vdpi,METRIC_RES,ptsize
 
 #include <fontconfig/fontconfig.h>
 
-/* #define NO_POSTSCRIPT_ALIAS 1 */
+#define NO_POSTSCRIPT_ALIAS 1
 #ifndef NO_POSTSCRIPT_ALIAS
 typedef struct _PostscriptAlias {
   char* name;
@@ -1659,7 +1659,7 @@ static char * font_path(char **fontpath, char *name_list)
 BGD_DECLARE(int) gdFTUseFontConfig(int flag)
 {
 #ifdef HAVE_LIBFONTCONFIG
-	fontConfigFlag = 1;
+	fontConfigFlag = flag;
 	return 1;
 #else
 	return 0;
