--- config/gdlib-config.in.orig	2006-10-11 02:46:22.000000000 -0700
+++ config/gdlib-config.in	2008-02-06 21:35:40.000000000 -0800
@@ -71,7 +71,7 @@
 	echo @LDFLAGS@
 	;;
     --libs)
-	echo @LIBS@ @LIBICONV@
+	echo @LIBS@ @LTLIBICONV@
 	;;
     --cflags|--includes)
 	echo -I@includedir@
@@ -84,7 +84,7 @@
 	echo "includedir: $includedir"
 	echo "cflags:     -I@includedir@"
 	echo "ldflags:    @LDFLAGS@"
-	echo "libs:       @LIBS@ @LIBICONV@"
+	echo "libs:       @LIBS@ @LTLIBICONV@"
 	echo "libdir:     $libdir"
 	echo "features:   @FEATURES@"
 	;;
