--- bones-3.9.0.gemspec
+++ bones-3.9.0.gemspec
@@ -25,9 +25,9 @@
 
   s.specification_version = 4
 
-  s.add_runtime_dependency(%q<rake>.freeze, ["~> 13.0".freeze])
-  s.add_runtime_dependency(%q<rdoc>.freeze, ["~> 6.0".freeze])
-  s.add_runtime_dependency(%q<little-plugger>.freeze, ["~> 1.1".freeze])
-  s.add_runtime_dependency(%q<loquacious>.freeze, ["~> 1.9".freeze])
-  s.add_development_dependency(%q<rspec>.freeze, ["~> 3.5".freeze])
+  s.add_runtime_dependency(%q<rake>.freeze, [">= 13.0".freeze])
+  s.add_runtime_dependency(%q<rdoc>.freeze, [">= 6.0".freeze])
+  s.add_runtime_dependency(%q<little-plugger>.freeze, [">= 1.1".freeze])
+  s.add_runtime_dependency(%q<loquacious>.freeze, [">= 1.9".freeze])
+  s.add_development_dependency(%q<rspec>.freeze, [">= 3.5".freeze])
 end
