--- bones-3.9.0.gemspec
+++ bones-3.9.0.gemspec
@@ -28,16 +28,16 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<rake>.freeze, ["~> 13.0"])
-    s.add_runtime_dependency(%q<rdoc>.freeze, ["~> 6.0"])
-    s.add_runtime_dependency(%q<little-plugger>.freeze, ["~> 1.1"])
-    s.add_runtime_dependency(%q<loquacious>.freeze, ["~> 1.9"])
-    s.add_development_dependency(%q<rspec>.freeze, ["~> 3.5"])
+    s.add_runtime_dependency(%q<rake>.freeze, [">= 13.0"])
+    s.add_runtime_dependency(%q<rdoc>.freeze, [">= 6.0"])
+    s.add_runtime_dependency(%q<little-plugger>.freeze, [">= 1.1"])
+    s.add_runtime_dependency(%q<loquacious>.freeze, [">= 1.9"])
+    s.add_development_dependency(%q<rspec>.freeze, [">= 3.5"])
   else
-    s.add_dependency(%q<rake>.freeze, ["~> 13.0"])
-    s.add_dependency(%q<rdoc>.freeze, ["~> 6.0"])
-    s.add_dependency(%q<little-plugger>.freeze, ["~> 1.1"])
-    s.add_dependency(%q<loquacious>.freeze, ["~> 1.9"])
-    s.add_dependency(%q<rspec>.freeze, ["~> 3.5"])
+    s.add_dependency(%q<rake>.freeze, [">= 13.0"])
+    s.add_dependency(%q<rdoc>.freeze, [">= 6.0"])
+    s.add_dependency(%q<little-plugger>.freeze, [">= 1.1"])
+    s.add_dependency(%q<loquacious>.freeze, [">= 1.9"])
+    s.add_dependency(%q<rspec>.freeze, [">= 3.5"])
   end
 end
