--- a/sql/CMakeLists.txt        2019-10-24 09:46:10.000000000 -0500
+++ b/sql/CMakeLists.txt	2019-10-24 09:46:42.000000000 -0500
@@ -20,7 +20,14 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA
 
-INCLUDE_DIRECTORIES(SYSTEM ${BOOST_PATCHES_DIR} ${BOOST_INCLUDE_DIR})
+# Prevent Boost from including external precompiled Boost libraries
+IF(USING_LOCAL_BOOST)
+  ADD_DEFINITIONS(
+    -DBOOST_ALL_NO_LIB
+)
+ENDIF()
+
+INCLUDE_DIRECTORIES(${BOOST_PATCHES_DIR} ${BOOST_INCLUDE_DIR} SYSTEM)
 
 MY_INCLUDE_SYSTEM_DIRECTORIES(ICU)
 
