--- meson.build.orig	2018-03-06 13:46:38.000000000 +0100
+++ meson.build	2018-03-06 14:01:44.000000000 +0100
@@ -12,16 +12,7 @@
 conf.set_quoted('PACKAGE', meson.project_name())
 conf.set_quoted('VERSION', meson.project_version())
 
-iconv = get_option('iconv')
-if iconv == 'false'
-  iconv = false
-elif cc.has_function('iconv')
-  iconv = true
-elif iconv == 'auto'
-  iconv = false
-else
-  error('iconv() not available')
-endif
+iconv = true
 conf.set('HAVE_ICONV', iconv)
 
 configure_file(output: 'config.h', configuration: conf)
@@ -109,6 +100,7 @@
   include_directories: inc,
   dependencies: [
     libmpdclient_dep,
+    cc.find_library('iconv'),
   ],
   install: true
 )
