# Adapted from fix <https://github.com/bitcoin/bitcoin/issues/6056>
diff -r 23cbba64aefc src/rpcrawtransaction.cpp
--- src/rpcrawtransaction.cpp	Sat Apr 25 10:04:50 2015 +0200
+++ src/rpcrawtransaction.cpp	Sat May 09 15:40:41 2015 +0200
@@ -287,7 +287,7 @@
         if (pk.IsPayToScriptHash()) {
             CTxDestination address;
             if (ExtractDestination(pk, address)) {
-                const CScriptID& hash = boost::get<const CScriptID&>(address);
+                const CScriptID& hash = boost::get<CScriptID>(address);
                 CScript redeemScript;
                 if (pwalletMain->GetCScript(hash, redeemScript))
                     entry.push_back(Pair("redeemScript", HexStr(redeemScript.begin(), redeemScript.end())));
