--- ext/vpx/gstvpxdec.c.orig	2019-02-10 12:51:04.000000000 -0800
+++ ext/vpx/gstvpxdec.c	2019-02-10 16:16:21.000000000 -0800
@@ -62,18 +62,6 @@
     {C_FLAGS (VP8_DEBLOCK), "Deblock", "deblock"},
     {C_FLAGS (VP8_DEMACROBLOCK), "Demacroblock", "demacroblock"},
     {C_FLAGS (VP8_ADDNOISE), "Add noise", "addnoise"},
-    {C_FLAGS (VP8_DEBUG_TXT_FRAME_INFO),
-          "Print frame information",
-        "visualize-frame-info"},
-    {C_FLAGS (VP8_DEBUG_TXT_MBLK_MODES),
-          "Show macroblock mode selection overlaid on image",
-        "visualize-macroblock-modes"},
-    {C_FLAGS (VP8_DEBUG_TXT_DC_DIFF),
-          "Show dc diff for each macro block overlaid on image",
-        "visualize-dc-diff"},
-    {C_FLAGS (VP8_DEBUG_TXT_RATE_INFO),
-          "Print video rate info",
-        "visualize-rate-info"},
     {C_FLAGS (VP8_MFQE), "Multi-frame quality enhancement", "mfqe"},
     {0, NULL, NULL}
   };
