--- configure.orig	2010-12-23 10:39:03.000000000 -0600
+++ configure	2010-12-23 10:51:02.000000000 -0600
@@ -106,6 +106,8 @@
 all_platforms="${all_platforms} x86-darwin8-icc"
 all_platforms="${all_platforms} x86-darwin9-gcc"
 all_platforms="${all_platforms} x86-darwin9-icc"
+all_platforms="${all_platforms} x86-darwin10-gcc"
+all_platforms="${all_platforms} x86-darwin10-icc"
 all_platforms="${all_platforms} x86-linux-gcc"
 all_platforms="${all_platforms} x86-linux-icc"
 all_platforms="${all_platforms} x86-solaris-gcc"
@@ -114,6 +116,7 @@
 all_platforms="${all_platforms} x86-win32-vs8"
 all_platforms="${all_platforms} x86-win32-vs9"
 all_platforms="${all_platforms} x86_64-darwin9-gcc"
+all_platforms="${all_platforms} x86_64-darwin10-gcc"
 all_platforms="${all_platforms} x86_64-linux-gcc"
 all_platforms="${all_platforms} x86_64-linux-icc"
 all_platforms="${all_platforms} x86_64-solaris-gcc"
@@ -121,6 +124,7 @@
 all_platforms="${all_platforms} x86_64-win64-vs9"
 all_platforms="${all_platforms} universal-darwin8-gcc"
 all_platforms="${all_platforms} universal-darwin9-gcc"
+all_platforms="${all_platforms} universal-darwin10-gcc"
 all_platforms="${all_platforms} generic-gnu"
 
 # all_targets is a list of all targets that can be configured
@@ -465,12 +469,20 @@
     case $toolchain in
         universal-darwin*)
             local darwin_ver=${tgt_os##darwin}
-            fat_bin_archs="$fat_bin_archs ppc32-${tgt_os}-gcc"
-
-            # Intel
-            fat_bin_archs="$fat_bin_archs x86-${tgt_os}-${tgt_cc}"
-            if [ $darwin_ver -gt 8 ]; then
-                fat_bin_archs="$fat_bin_archs x86_64-${tgt_os}-${tgt_cc}"
+            if [ $darwin_ver -eq 8 ]; then
+               # ppc32 and x86 are defaults for universal on 10.4
+               fat_bin_archs="$fat_bin_archs ppc32-${tgt_os}-${tgt_cc}"
+               fat_bin_archs="$fat_bin_archs x86-${tgt_os}-${tgt_cc}"
+            fi
+            if [ $darwin_ver -eq 9 ]; then
+               # ppc32 and x86 are defaults for universal on 10.5
+               fat_bin_archs="$fat_bin_archs ppc32-${tgt_os}-${tgt_cc}"
+               fat_bin_archs="$fat_bin_archs x86-${tgt_os}-${tgt_cc}"
+            fi
+            if [ $darwin_ver -eq 10 ]; then
+               # x86 and x86_64 are defaults for universal on 10.6
+               fat_bin_archs="$fat_bin_archs x86-${tgt_os}-${tgt_cc}"
+               fat_bin_archs="$fat_bin_archs x86_64-${tgt_os}-${tgt_cc}"
             fi
             ;;
     esac
