--- ../apache-couchdb-1.3.0.orig/etc/launchd/org.apache.couchdb.plist.tpl.in	2013-03-30 09:46:38.000000000 -0500
+++ ./etc/launchd/org.apache.couchdb.plist.tpl.in	2013-04-05 10:12:43.000000000 -0500
@@ -9,8 +9,6 @@
     <dict>
       <key>HOME</key>
       <string>~</string>
-      <key>DYLD_LIBRARY_PATH</key>
-      <string>/opt/local/lib:$DYLD_LIBRARY_PATH</string>
     </dict>
     <key>ProgramArguments</key>
     <array>
