--- ../apache-couchdb-1.3.0.orig/src/couchdb/priv/Makefile.in	2013-03-30 09:46:48.000000000 -0500
+++ ./src/couchdb/priv/Makefile.in	2013-04-05 10:12:43.000000000 -0500
@@ -382,7 +382,7 @@
 	couch_js/util.c
 
 couchjs_SOURCES = $(COUCHJS_SRCS)
-couchjs_CFLAGS = -g -Wall -Werror -D_BSD_SOURCE $(CURL_CFLAGS) $(JS_CFLAGS)
+couchjs_CFLAGS = -g -Wall -D_BSD_SOURCE $(CURL_CFLAGS) $(JS_CFLAGS)
 couchjs_LDADD = $(CURL_LIBS) $(JS_LIBS)
 couchpriv_DATA = stat_descriptions.cfg
 @WINDOWS_TRUE@couchspawnkillable_SOURCES = spawnkillable/couchspawnkillable_win.c
