--- configure.orig	2011-10-24 04:38:37.000000000 -0500
+++ configure	2011-11-14 18:07:08.000000000 -0600
@@ -12557,7 +12557,7 @@
     fi
 fi
 
-has_crypto=`${ERL} -eval "case application:load(crypto) of ok -> ok; _ -> exit(no_crypto) end." -noshell -s init stop`
+has_crypto=""
 
 if test -n "$has_crypto"; then
     { { echo "$as_me:$LINENO: error: Could not find the Erlang crypto library. Has Erlang been compiled with OpenSSL support?" >&5
