--- src/CMakeLists.txt.orig	2018-03-25 07:41:28.000000000 +0800
+++ src/CMakeLists.txt	2023-05-03 20:04:06.000000000 +0800
@@ -3,7 +3,7 @@
 
 set(LIB_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/mod_files/)
 
-add_library(yajl_fort ${SRC})
+add_library(yajl_fort SHARED ${SRC})
 
 set_target_properties(yajl_fort PROPERTIES
     Fortran_MODULE_DIRECTORY ${LIB_MOD_DIR})
@@ -20,6 +20,5 @@
 install(TARGETS yajl_fort
         EXPORT  yajl_fort
         LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib
 )
-install(DIRECTORY ${LIB_MOD_DIR} DESTINATION lib)
+install(DIRECTORY ${LIB_MOD_DIR} DESTINATION include/yajl_fort)
