--- configure.ac.orig	2008-03-02 07:48:54.000000000 +0900
+++ configure.ac	2009-03-17 10:56:51.000000000 +0900
@@ -946,8 +946,7 @@
         AC_MSG_NOTICE([You are running OSX, assuming threadsafe gethostbyname version])
 elif test $CYGWIN_OS -eq 1; then
         AC_MSG_NOTICE([Your are building under Cygwin, assuming threadsafe gethostbyname implementation])
-fi
-
+else
 AC_CHECK_FUNCS([gethostbyname_r],[],
         [
             AC_CHECK_HEADERS([lwres/netdb.h],
@@ -960,6 +959,7 @@
                     fi
                 ])
         ])
+fi
 
 AC_CHECK_FUNCS([if_nameindex],[],
         [
@@ -2327,7 +2327,7 @@
 if test "x$FFMPEG_OK" = xyes; then
     if test -n "$FFMPEG_SEARCH_HEADERS" ; then
         CPPFLAGS="-I$FFMPEG_SEARCH_HEADERS"
-        AC_CHECK_HEADER($FFMPEG_SEARCH_HEADERS/ffmpeg/avformat.h,
+        AC_CHECK_HEADER($FFMPEG_SEARCH_HEADERS/libavformat/avformat.h,
                 [
                     FFMPEG_CXXFLAGS="-I$FFMPEG_SEARCH_HEADERS"
                 ],
@@ -2340,12 +2340,12 @@
                 ]
         )
     else 
-        AC_CHECK_HEADER(ffmpeg/avformat.h,
+        AC_CHECK_HEADER(libavformat/avformat.h,
             [],
             [
                 CPPFLAGS="-I$SEARCH_DIR_HEADERS"
                 unset ac_cv_header_ffmpeg_avformat_h
-                AC_CHECK_HEADER($SEARCH_DIR_HEADERS/ffmpeg/avformat.h,
+                AC_CHECK_HEADER($SEARCH_DIR_HEADERS/libavformat/avformat.h,
                     [
                         FFMPEG_CXXFLAGS="-I$SEARCH_DIR_HEADERS"
                     ],
@@ -2447,7 +2447,7 @@
             [
                 #define __STDC_CONSTANT_MACROS
                 #include <stdint.h>
-                #include <ffmpeg/avformat.h>
+                #include <libavformat/avformat.h>
             ],
             [
                 av_register_all();
@@ -2462,7 +2462,7 @@
                     #define __STDC_CONSTANT_MACROS
                     #include <stdint.h>
                     extern "C" {
-                        #include <ffmpeg/avformat.h>
+                        #include <libavformat/avformat.h>
                     }
                 ],
                 [
