--- src/url.cc	2008/03/01 17:33:21	1714
+++ src/url.cc	2008/04/09 10:32:57	1775
@@ -75,7 +75,7 @@
 
     if (only_header)
     {
-        curl_easy_setopt(curl_handle, CURLOPT_NOBODY);
+        curl_easy_setopt(curl_handle, CURLOPT_NOBODY, 1);
         curl_easy_setopt(curl_handle, CURLOPT_HEADERFUNCTION, URL::dl);
         curl_easy_setopt(curl_handle, CURLOPT_HEADERDATA, 
                          (void *)buffer.getPtr());
