# --disable-warn-error does not work as intended

--- Makefile.orig	2022-03-28 20:27:36.000000000 +0800
+++ Makefile	2022-04-30 02:42:39.000000000 +0800
@@ -48,7 +48,6 @@
 
 COMPFLAGS=-strict-sequence -principal -absname \
           -w +a-4-9-40-41-42-44-45-48-66-70 \
-          -warn-error +a \
           -bin-annot -safe-string -strict-formats $(INCLUDES)
 LINKFLAGS=
 

--- otherlibs/dynlink/Makefile.orig	2022-03-28 20:27:36.000000000 +0800
+++ otherlibs/dynlink/Makefile	2022-04-30 02:41:49.000000000 +0800
@@ -30,10 +30,9 @@
 # COMPFLAGS should be in sync with the toplevel Makefile's COMPFLAGS.
 COMPFLAGS=-strict-sequence -principal -absname \
           -w +a-4-9-40-41-42-44-45-48-66-70 \
-          -warn-error +A \
           -bin-annot -safe-string -strict-formats
 ifeq "$(FLAMBDA)" "true"
-OPTCOMPFLAGS += -O3
+OPTCOMPFLAGS += -O2
 endif
 
 COMPFLAGS += -I byte


--- stdlib/Makefile.orig	2022-03-28 20:27:36.000000000 +0800
+++ stdlib/Makefile	2022-05-04 00:43:33.000000000 +0800
@@ -22,10 +22,10 @@
 COMPILER=$(ROOTDIR)/ocamlc$(EXE)
 CAMLC=$(OCAMLRUN) $(COMPILER)
 COMPFLAGS=-strict-sequence -absname -w +a-4-9-41-42-44-45-48-70 \
-          -g -warn-error +A -bin-annot -nostdlib -principal \
+          -g -bin-annot -nostdlib -principal \
           -safe-string -strict-formats
 ifeq "$(FLAMBDA)" "true"
-OPTCOMPFLAGS += -O3
+OPTCOMPFLAGS += -O2
 endif
 OPTCOMPILER=$(ROOTDIR)/ocamlopt
 CAMLOPT=$(OCAMLRUN) $(OPTCOMPILER)
