--- configure.orig	2022-03-28 21:51:09.000000000 +0800
+++ configure	2022-11-18 15:56:19.000000000 +0800
@@ -14087,7 +14087,7 @@
   case $host in #(
   *-apple-darwin*) :
     mksharedlib="$CC -shared \
-                   -flat_namespace -undefined suppress -Wl,-no_compact_unwind \
+                   -flat_namespace -undefined suppress -Wl,-no_compact_unwind -read_only_relocs suppress \
                    \$(LDFLAGS)"
       supports_shared_libraries=true ;; #(
   *-*-mingw32) :
