--- configure.orig	2009-05-21 00:33:09.000000000 +0900
+++ configure	2009-10-01 19:35:54.000000000 +0900
@@ -373,7 +373,7 @@
 echo "#define SIZEOF_PTR $3" >> m.h
 echo "#define SIZEOF_SHORT $4" >> m.h
 
-if test $2 = 8; then
+if test $2 = 8 && $system != macosx; then
      echo "#define ARCH_INT64_TYPE long" >> m.h
      echo "#define ARCH_UINT64_TYPE unsigned long" >> m.h
      echo '#define ARCH_INT64_PRINTF_FORMAT "l"' >> m.h
