--- CMakeLists.txt.orig	2016-03-15 17:19:38.000000000 +0300
+++ CMakeLists.txt	2016-03-24 15:02:37.000000000 +0300
@@ -32,7 +32,6 @@
 # For the Git SHA1/version code
 list(APPEND CMAKE_MODULE_PATH
     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
-include(GetGitRevisionDescriptionDyND)
 
 find_package(PythonInterp REQUIRED)
 find_package(PythonLibsNew REQUIRED)
@@ -78,8 +77,8 @@
 endif()
 
 # Get the git revision
-get_git_head_revision("${CMAKE_CURRENT_SOURCE_DIR}" GIT_REFSPEC DYND_PYTHON_GIT_SHA1)
-git_describe("${CMAKE_CURRENT_SOURCE_DIR}" DYND_PYTHON_VERSION_STRING --dirty --always --match "v*")
+SET(DYND_PYTHON_GIT_SHA1 "@@DYND_PYTHON_GIT_SHA1@@")
+SET(DYND_PYTHON_VERSION_STRING "@@DYND_PYTHON_VERSION_STRING@@")
 message(STATUS "DyND-Python version: ${DYND_PYTHON_VERSION_STRING}")
 configure_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/dynd/src/git_version.cpp.in"
