--- setup.py.orig	2016-03-15 17:19:38.000000000 +0300
+++ setup.py	2016-03-24 15:14:58.000000000 +0300
@@ -40,7 +40,7 @@
 
   def initialize_options(self):
     build_ext.initialize_options(self)
-    self.extra_cmake_args = ''
+    self.extra_cmake_args = '@@EXTRA_CMAKE_ARGS@@'
 
   def run(self):
     # We don't call the origin build_ext, instead ignore that
@@ -151,8 +151,7 @@
         return subprocess.Popen(args, stdout = subprocess.PIPE).communicate()[0]
 
 # Get the version number to use from git
-ver = check_output(['git', 'describe', '--dirty',
-                               '--always', '--match', 'v*']).decode('ascii').strip('\n')
+ver = "@@DYND_PYTHON_VERSION_STRING@@"
 
 # Same processing as in __init__.py
 if '.' in ver:
