--- name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/preferences/PreferencesInitializer.java.orig	2021-01-02 11:36:29.000000000 +0100
+++ name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/preferences/PreferencesInitializer.java	2021-01-24 14:13:40.000000000 +0100
@@ -13,7 +13,7 @@
     public void initializeDefaultPreferences()
     {
         IPreferenceStore store = PortfolioPlugin.getDefault().getPreferenceStore();
-        store.setDefault(UIConstants.Preferences.AUTO_UPDATE, true);
+        store.setDefault(UIConstants.Preferences.AUTO_UPDATE, false);
         store.setDefault(UIConstants.Preferences.UPDATE_SITE,
                         Platform.ARCH_X86.equals(Platform.getOSArch())
                                         ? "https://updates.portfolio-performance.info/portfolio-x86" //$NON-NLS-1$
