--- Source/WTF/WTF.pri.orig	2014-09-11 03:48:11.000000000 -0700
+++ Source/WTF/WTF.pri	2019-04-19 10:34:17.000000000 -0700
@@ -9,10 +9,8 @@
 INCLUDEPATH += $$PWD
 
 mac {
-    # Mac OS does ship libicu but not the associated header files.
-    # Therefore WebKit provides adequate header files.
-    INCLUDEPATH = $${ROOT_WEBKIT_DIR}/Source/WTF/icu $$INCLUDEPATH
-    LIBS += -licucore
+     CONFIG += link_pkgconfig
+     PKGCONFIG *= icu-i18n
 } else {
     contains(QT_CONFIG,icu) {
         win32: LIBS += -licuin -licuuc -licudt
--- Source/WebCore/Target.pri.orig	2015-10-12 21:37:22.000000000 -0700
+++ Source/WebCore/Target.pri	2016-05-04 09:47:10.000000000 -0700
@@ -2986,9 +2986,7 @@
 }
 
 mac {
-    # For Mac we use the same SmartReplace implementation as the Apple port.
-    SOURCES += editing/SmartReplaceCF.cpp
-    INCLUDEPATH += $$PWD/icu
+    SOURCES += editing/SmartReplaceICU.cpp
 } else {
     SOURCES += editing/SmartReplaceICU.cpp
 }
