diff --git configure configure
index 1bcb362f..68c96f70 100755
--- configure
+++ configure
@@ -2411,7 +2411,9 @@ Additional options:
     -l <string> ........ Add an explicit library.
 
     -no-rpath .......... Do not use the library install path as a runtime
-                         library path.
+                         library path. On Apple platforms, this implies using
+                         absolute install names (based in -libdir) for dynamic
+                         libraries and frameworks.
  +  -rpath ............. Link Qt libraries and executables using the library
                          install path as a runtime library path. Equivalent
                          to -R install_libpath
diff --git mkspecs-save/features/qt_build_config.prf mkspecs-save/features/qt_build_config.prf
index 5fece28c..89d2555e 100644
--- mkspecs-save/features/qt_build_config.prf
+++ mkspecs-save/features/qt_build_config.prf
@@ -47,9 +47,7 @@ QMAKE_DIR_REPLACE_SANE = PRECOMPILED_DIR OBJECTS_DIR MOC_DIR RCC_DIR UI_DIR
 }
 
 mac {
-   !isEmpty(QMAKE_RPATHDIR){
        CONFIG += absolute_library_soname
-   }
 }
 
 cross_compile: \
diff --git mkspecs/features/qt_build_config.prf mkspecs/features/qt_build_config.prf
index 5fece28c..89d2555e 100644
--- mkspecs/features/qt_build_config.prf
+++ mkspecs/features/qt_build_config.prf
@@ -47,9 +47,7 @@ QMAKE_DIR_REPLACE_SANE = PRECOMPILED_DIR OBJECTS_DIR MOC_DIR RCC_DIR UI_DIR
 }
 
 mac {
-   !isEmpty(QMAKE_RPATHDIR){
        CONFIG += absolute_library_soname
-   }
 }
 
 cross_compile: \
