--- src/assistant/assistant/assistant.pro.orig	2019-04-20 07:21:04.000000000 -0700
+++ src/assistant/assistant/assistant.pro	2019-04-20 07:21:55.000000000 -0700
@@ -1,8 +1,5 @@
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
-    QT += webkitwidgets
-} else {
     DEFINES += QT_NO_WEBKIT
-}
+
 QT += widgets network help sql help
 qtHaveModule(printsupport): QT += printsupport
 PROJECTNAME = Assistant
@@ -72,11 +69,9 @@
     openpageswidget.cpp \
     openpagesmanager.cpp \
     openpagesswitcher.cpp
-qtHaveModule(webkitwidgets):!contains(QT_CONFIG, static) {
-    SOURCES += helpviewer_qwv.cpp
-} else {
+
     SOURCES += helpviewer_qtb.cpp
-}
+
 
 FORMS += bookmarkdialog.ui \
     bookmarkmanagerwidget.ui \
--- src/designer/src/plugins/plugins.pro.orig	2016-02-26 03:07:22.000000000 -0700
+++ src/designer/src/plugins/plugins.pro	2016-05-04 16:32:21.000000000 -0700
@@ -1,6 +1,5 @@
 TEMPLATE = subdirs
 
 # qtHaveModule(opengl): SUBDIRS += tools/view3d
-qtHaveModule(webkitwidgets): SUBDIRS += qwebview
 win32: qtHaveModule(axcontainer): SUBDIRS += activeqt
 qtHaveModule(quickwidgets): SUBDIRS += qquickwidget
