--- tools/winebuild/utils.c.orig	2013-02-01 02:10:43.000000000 -0600
+++ tools/winebuild/utils.c	2013-02-04 19:17:05.000000000 -0600
@@ -354,17 +354,11 @@
 
     if (!as_command)
     {
-        as_command = find_tool( "clang", NULL );
+        as_command = "@CLANG@";
         if (as_command) as_is_clang = 1;
     }
 
     if (!as_command)
-    {
-        static const char * const commands[] = { "gas", "as", NULL };
-        as_command = find_tool( "as", commands );
-    }
-
-    if (!as_command)
         fatal_error( "cannot find suitable assembler\n" );
 
     strarray_add_one( args, as_command );
