--- tools/winebuild/utils.c.orig	2013-02-01 02:10:43.000000000 -0600
+++ tools/winebuild/utils.c	2013-02-04 19:16:28.000000000 -0600
@@ -354,14 +354,7 @@
 
     if (!as_command)
     {
-        as_command = find_tool( "clang", NULL );
-        if (as_command) as_is_clang = 1;
-    }
-
-    if (!as_command)
-    {
-        static const char * const commands[] = { "gas", "as", NULL };
-        as_command = find_tool( "as", commands );
+        as_command = find_tool( "as", NULL );
     }
 
     if (!as_command)
