--- dlls/crtdll/crtdll_main.c	2008-07-11 10:55:55.000000000 -0500
+++ dlls/crtdll/crtdll_main.c	2008-07-22 01:39:56.000000000 -0500
@@ -37,14 +37,14 @@
                                  int expand_wildcards, int *new_mode );
 
 /* The following data items are not exported from msvcrt */
-unsigned int CRTDLL__basemajor_dll;
-unsigned int CRTDLL__baseminor_dll;
-unsigned int CRTDLL__baseversion_dll;
-unsigned int CRTDLL__cpumode_dll;
-unsigned int CRTDLL__osmajor_dll;
-unsigned int CRTDLL__osminor_dll;
-unsigned int CRTDLL__osmode_dll;
-unsigned int CRTDLL__osversion_dll;
+unsigned int CRTDLL__basemajor_dll=0;
+unsigned int CRTDLL__baseminor_dll=0;
+unsigned int CRTDLL__baseversion_dll=0;
+unsigned int CRTDLL__cpumode_dll=0;
+unsigned int CRTDLL__osmajor_dll=0;
+unsigned int CRTDLL__osminor_dll=0;
+unsigned int CRTDLL__osmode_dll=0;
+unsigned int CRTDLL__osversion_dll=0;
 
 /* dev_t is a short in crtdll but an unsigned int in msvcrt */
 typedef short crtdll_dev_t;
--- dlls/msvcrt/data.c	2008-07-11 10:55:55.000000000 -0500
+++ dlls/msvcrt/data.c	2008-07-22 01:39:56.000000000 -0500
@@ -29,23 +29,23 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(msvcrt);
 
-int MSVCRT___argc;
-unsigned int MSVCRT_basemajor;/* FIXME: */
-unsigned int MSVCRT_baseminor;/* FIXME: */
-unsigned int MSVCRT_baseversion; /* FIXME: */
-unsigned int MSVCRT__commode;
-unsigned int MSVCRT__fmode;
-unsigned int MSVCRT_osmajor;/* FIXME: */
-unsigned int MSVCRT_osminor;/* FIXME: */
-unsigned int MSVCRT_osmode;/* FIXME: */
-unsigned int MSVCRT__osver;
-unsigned int MSVCRT_osversion; /* FIXME: */
-unsigned int MSVCRT__winmajor;
-unsigned int MSVCRT__winminor;
-unsigned int MSVCRT__winver;
-unsigned int MSVCRT___setlc_active;
-unsigned int MSVCRT___unguarded_readlc_active;
-double MSVCRT__HUGE;
+int MSVCRT___argc=0;
+unsigned int MSVCRT_basemajor=0;/* FIXME: */
+unsigned int MSVCRT_baseminor=0;/* FIXME: */
+unsigned int MSVCRT_baseversion=0; /* FIXME: */
+unsigned int MSVCRT__commode=0;
+unsigned int MSVCRT__fmode=0;
+unsigned int MSVCRT_osmajor=0;/* FIXME: */
+unsigned int MSVCRT_osminor=0;/* FIXME: */
+unsigned int MSVCRT_osmode=0;/* FIXME: */
+unsigned int MSVCRT__osver=0;
+unsigned int MSVCRT_osversion=0; /* FIXME: */
+unsigned int MSVCRT__winmajor=0;
+unsigned int MSVCRT__winminor=0;
+unsigned int MSVCRT__winver=0;
+unsigned int MSVCRT___setlc_active=0;
+unsigned int MSVCRT___unguarded_readlc_active=0;
+double MSVCRT__HUGE=0;
 char **MSVCRT___argv;
 MSVCRT_wchar_t **MSVCRT___wargv;
 char *MSVCRT__acmdln;
--- dlls/msvcrt/locale.c	2008-07-11 10:55:55.000000000 -0500
+++ dlls/msvcrt/locale.c	2008-07-22 01:39:56.000000000 -0500
@@ -45,8 +45,8 @@
 #define MAX_LOCALE_LENGTH 256
 char MSVCRT_current_lc_all[MAX_LOCALE_LENGTH];
 LCID MSVCRT_current_lc_all_lcid;
-int MSVCRT___lc_codepage;
-int MSVCRT___lc_collate_cp;
+int MSVCRT___lc_codepage=0;
+int MSVCRT___lc_collate_cp=0;
 HANDLE MSVCRT___lc_handle[MSVCRT_LC_MAX - MSVCRT_LC_MIN + 1];
 
 /* MT */
