--- src/nco/mpncra.c.orig	2007-12-19 10:11:55.000000000 +0900
+++ src/nco/mpncra.c	2007-12-19 10:12:16.000000000 +0900
@@ -1204,7 +1204,7 @@
 
     /* Add time variable to output file
        NB: nco_cnv_arm_time_install() contains OpenMP critical region */
-    if(CNV_ARM && prg == ncrcat) (void)nco_cnv_arm_time_install(out_id,base_time_srt);
+    if(CNV_ARM && prg == ncrcat) (void)nco_cnv_arm_time_install(out_id,base_time_srt,dfl_lvl);
 #ifdef ENABLE_MPI
     nco_close(out_id); 
     printf("DEBUG: Mgr prc_rnk %d closed out file %d after fixing date, time \n", prc_rnk, out_id);
