--- pylsp/plugins/flake8_lint.py.orig	2021-11-22 12:30:56.000000000 -0500
+++ pylsp/plugins/flake8_lint.py	2021-11-23 12:56:57.000000000 -0500
@@ -55,7 +55,7 @@
         log.debug("using flake8 with config: %s", opts['config'])
 
     # Call the flake8 utility then parse diagnostics from stdout
-    flake8_executable = settings.get('executable', 'flake8')
+    flake8_executable = settings.get('executable', '@@FLAKE8_BIN@@')
 
     args = build_args(opts)
     output = run_flake8(flake8_executable, args, document)
