--- pyproject.toml.orig	2022-12-26 03:45:21
+++ pyproject.toml	2022-12-26 03:46:01
@@ -27,23 +27,23 @@
 
 [project.optional-dependencies]
 all = [
-    "autopep8>=1.6.0,<1.7.0",
-    "flake8>=5.0.0,<5.1.0",
+    "autopep8>=1.6.0",
+    "flake8>=5.0.0",
     "mccabe>=0.7.0,<0.8.0",
-    "pycodestyle>=2.9.0,<2.10.0",
+    "pycodestyle>=2.9.0",
     "pydocstyle>=2.0.0",
-    "pyflakes>=2.5.0,<2.6.0",
+    "pyflakes>=2.5.0",
     "pylint>=2.5.0",
     "rope>=0.10.5",
     "yapf",
     "whatthepatch"
 ]
-autopep8 = ["autopep8>=1.6.0,<1.7.0"]
-flake8 = ["flake8>=5.0.0,<5.1.0"]
+autopep8 = ["autopep8>=1.6.0"]
+flake8 = ["flake8>=5.0.0"]
 mccabe = ["mccabe>=0.7.0,<0.8.0"]
-pycodestyle = ["pycodestyle>=2.9.0,<2.10.0"]
+pycodestyle = ["pycodestyle>=2.9.0"]
 pydocstyle = ["pydocstyle>=2.0.0"]
-pyflakes = ["pyflakes>=2.5.0,<2.6.0"]
+pyflakes = ["pyflakes>=2.5.0"]
 pylint = ["pylint>=2.5.0"]
 rope = ["rope>0.10.5"]
 yapf = ["yapf", "whatthepatch>=1.0.2,<2.0.0"]
