--- setup.cfg.orig	2022-03-18 12:49:09.000000000 -0400
+++ setup.cfg	2022-03-18 12:49:52.000000000 -0400
@@ -11,7 +11,7 @@
 [options]
 packages = find:
 python_requires = >=3.7
-install_requires = 
+install_requires =
 	jedi>=0.17.2,<0.19.0
 	python-lsp-jsonrpc>=1.0.0
 	pluggy>=1.0.0
@@ -23,10 +23,10 @@
 exclude = contrib; docs; test; test.*; test.plugins; test.plugins.*
 
 [options.extras_require]
-all = 
+all =
 	autopep8>=1.6.0,<1.7.0
 	flake8>=4.0.0,<4.1.0
-	mccabe>=0.6.0,<0.7.0
+	mccabe>=0.6.0,<0.8.0
 	pycodestyle>=2.8.0,<2.9.0
 	pydocstyle>=2.0.0
 	pyflakes>=2.4.0,<2.5.0
@@ -35,14 +35,14 @@
 	yapf
 autopep8 = autopep8>=1.6.0,<1.7.0
 flake8 = flake8>=4.0.0,<4.1.0
-mccabe = mccabe>=0.6.0,<0.7.0
+mccabe = mccabe>=0.6.0,<0.8.0
 pycodestyle = pycodestyle>=2.8.0,<2.9.0
 pydocstyle = pydocstyle>=2.0.0
 pyflakes = pyflakes>=2.4.0,<2.5.0
 pylint = pylint>=2.5.0
 rope = rope>0.10.5
 yapf = yapf
-test = 
+test =
 	pylint>=2.5.0
 	pytest
 	pytest-cov
@@ -55,7 +55,7 @@
 
 [options.entry_points]
 console_scripts = pylsp = pylsp.__main__:main
-pylsp = 
+pylsp =
 	autopep8 = pylsp.plugins.autopep8_format
 	folding = pylsp.plugins.folding
 	flake8 = pylsp.plugins.flake8_lint
@@ -84,11 +84,11 @@
 
 [tool:pytest]
 testpaths = test
-addopts = 
+addopts =
 	--cov-report html --cov-report term --junitxml=pytest.xml
 	--cov pylsp --cov test
 
 [egg_info]
-tag_build = 
+tag_build =
 tag_date = 0
 
