--- Makefile.in.orig	2019-11-25 01:08:36
+++ Makefile.in	2026-05-19 00:25:55
@@ -845,48 +845,33 @@
 
 apps: MP4Joiner.app MP4Splitter.app
 
-MP4Joiner.app: Info.plist src/mp4joiner
+MP4Joiner.app: Info.plist all
 	-mkdir MP4Joiner.app
 	-mkdir MP4Joiner.app/Contents
 	-mkdir MP4Joiner.app/Contents/MacOS
 	-mkdir MP4Joiner.app/Contents/Resources
 	-mkdir MP4Joiner.app/Contents/Resources/English.lproj
 	cp Info.plist MP4Joiner.app/Contents/
-	echo -n "APPL????" > MP4Joiner.app/Contents/PkgInfo
+	printf "APPL????" > MP4Joiner.app/Contents/PkgInfo
 	cp src/mp4joiner MP4Joiner.app/Contents/MacOS/MP4Joiner
 	cp resources/mp4joiner.icns MP4Joiner.app/Contents/Resources/MP4Joiner.icns
 	-mkdir MP4Joiner.app/Contents/locale
 	(cd locale && cp -r de es fr hu it pl pt pt_BR ru tr zh_CN zh_TW ../MP4Joiner.app/Contents/locale/)
-	cp /opt/local/bin/ffmpeg MP4Joiner.app/Contents/MacOS/
 	strip MP4Joiner.app/Contents/MacOS/MP4Joiner
-	strip MP4Joiner.app/Contents/MacOS/ffmpeg
-	cp /Applications/Osmo4.app/Contents/MacOS/MP4Box MP4Joiner.app/Contents/MacOS/
-	-mkdir MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libcrypto.1.0.0.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libgpac.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libjpeg.9.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libmozjs185.1.0.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libnspr4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libplc4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libplds4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libpng16.16.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libssl.1.0.0.dylib MP4Joiner.app/Contents/MacOS/lib/
 
-MP4Splitter.app: Info.plist src/mp4splitter
+MP4Splitter.app: Info.plist all
 	-mkdir MP4Splitter.app
 	-mkdir MP4Splitter.app/Contents
 	-mkdir MP4Splitter.app/Contents/MacOS
 	-mkdir MP4Splitter.app/Contents/Resources
 	-mkdir MP4Splitter.app/Contents/Resources/English.lproj
 	cp InfoSplitter.plist MP4Splitter.app/Contents/Info.plist
-	echo -n "APPL????" > MP4Splitter.app/Contents/PkgInfo
+	printf "APPL????" > MP4Splitter.app/Contents/PkgInfo
 	cp src/mp4splitter MP4Splitter.app/Contents/MacOS/MP4Splitter
 	cp resources/mp4splitter.icns MP4Splitter.app/Contents/Resources/MP4Splitter.icns
 	-mkdir MP4Splitter.app/Contents/locale
 	(cd locale && cp -r de es fr hu it pl pt pt_BR ru tr zh_CN zh_TW ../MP4Splitter.app/Contents/locale/)
-	cp /opt/local/bin/ffmpeg MP4Splitter.app/Contents/MacOS/
 	strip MP4Splitter.app/Contents/MacOS/MP4Splitter
-	strip MP4Splitter.app/Contents/MacOS/ffmpeg
 
 libs: MP4Joiner.libs MP4Splitter.libs
 
