diff --git configure.ac configure.ac
index 687bbe91..2aba2da7 100644
--- configure.ac
+++ configure.ac
@@ -669,19 +669,6 @@ fi
 # 
 LIBS=" -lXm -lXt -lX11 $LIBS" 
   
-# 
-# Remove duplicate entries.  Thanks to Paul Lutt, ke7xt, for this! 
-#
-# Don't get rid of dupes on the LIBS line.  We may need the same
-# "-Llibdir" called out several times, immediately prior to each
-# "-llibrary" that needs it.
-# 
-changequote(,) 
-CFLAGS=`echo  "$CFLAGS" | awk '{for(i=1;i<=NF;++i) {if (arg[$i]++ == 0) s = s " " $i} print s}'`
-CPPFLAGS=`echo  "$CPPFLAGS" | awk '{for(i=1;i<=NF;++i) {if (arg[$i]++ == 0) s = s " " $i} print s}'`
-LDFLAGS=`echo "$LDFLAGS" | awk '{for(i=1;i<=NF;++i) {if (arg[$i]++ == 0) s = s " " $i} print s}'`
-#LIBS=`echo "$LIBS" | awk '{for(i=1;i<=NF;++i) {if (arg[$i]++ == 0) s = s " " $i} print s}'`
-changequote([,]) 
 
 
 # 
@@ -900,7 +887,7 @@ echo
 echo RECOMMENDED OPTIONS:
 
 echo "  Xpm / Snapshots ........................... : $use_xpm"
-echo -n "  GraphicsMagick/ImageMagick (Raster maps) .. : "
+printf "  GraphicsMagick/ImageMagick (Raster maps) .. : "
 if test "${use_graphicsmagick}" = "yes"; then 
   echo "yes (GraphicsMagick)"
 else
