--- Makefile.am.orig	2010-09-08 13:13:26.000000000 -0500
+++ Makefile.am	2011-09-29 01:40:00.000000000 -0500
@@ -72,10 +72,10 @@
 # documentation is FHS compliant.
 
 install-exec-hook:
-	-rm -rf ${pkgdatadir}/doc
+	-rm -rf $(DESTDIR)${pkgdatadir}/doc
 
 install-data-hook:
-	cd $(mapdir) && \
+	cd $(DESTDIR)$(mapdir) && \
 	rm -f CC_OpenStreetMap.png && \
 	$(LN_S) CC_OpenStreetMap_logo.png CC_OpenStreetMap.png
  
