--- acinclude.m4.orig
+++ acinclude.m4
@@ -36,14 +36,6 @@
 [
 AC_MSG_CHECKING([for search paths])
 
-test -d /usr/local/include && CPPFLAGS="-I/usr/local/include $CPPFLAGS"
-test -d /usr/local/lib && LDFLAGS="-L/usr/local/lib $LDFLAGS"
-
-for d in /sw /opt /opt/local /usr/dt/share /usr/sfw /opt/sfw; do
-test -d $d/include && CPPFLAGS="$CPPFLAGS -I$d/include"
-test -d $d/lib && LDFLAGS="$LDFLAGS -L$d/lib"
-done
-
 AC_MSG_RESULT([done])
 ])
 
@@ -98,7 +90,7 @@
 [
 BINPATH=$PATH
 
-for d in / /usr /usr/local /usr/X11 /usr/X11R6 /usr/sfw /opt/sfw /opt/local /sw; do
+for d in / /usr; do
 test -d $d/bin && echo $BINPATH | grep -- $d/bin > /dev/null || BINPATH="$BINPATH:$d/bin"
 done
 
