--- configure.in.orig	2012-09-17 12:14:53.000000000 -0500
+++ configure.in	2012-09-17 12:16:00.000000000 -0500
@@ -651,7 +651,13 @@
 fi
 if test "x$pythonprog" != "x"; then
    if test "x$pythonusepthreads" = "x"; then
-      pythonusepthreads=`echo -e "try:\n  import thread\n  print 'yes'\nexcept:\n  print 'no'\n" | python`
+      pythonusepthreads=`echo "
+try:
+  import thread
+  print 'yes'
+except:
+  print 'no'
+" | python`
    fi
    echo "checking for python threads... $pythonusepthreads"
 fi
@@ -675,7 +681,13 @@
 # Check for Tkinter/Tix
 if test "x$tkinter" = "x"; then
    if test "x$pythonprog" != "x"; then
-      tkinter=`echo -e "try:\n  import Tix\n  print 'yes'\nexcept:\n  print 'no'\n" | python`
+      tkinter=`echo "
+try:
+  import Tix
+  print 'yes'
+except:
+  print 'no'
+" | python`
    fi
 fi
 echo "checking for tkinter... $tkinter"
