--- configure.orig	2012-09-13 09:54:24.000000000 -0500
+++ configure	2012-09-13 09:55:31.000000000 -0500
@@ -21762,7 +21762,13 @@
 fi
 if test "x$pythonprog" != "x"; then
    if test "x$pythonusepthreads" = "x"; then
-      pythonusepthreads=`echo -e "try:\n  import thread\n  print 'yes'\nexcept:\n  print 'no'\n" | python`
+      pythonusepthreads=`echo "
+try:
+  import thread
+  print 'yes'
+except:
+  print 'no'
+" | python`
    fi
    echo "checking for python threads... $pythonusepthreads"
 fi
@@ -21786,7 +21792,13 @@
 # Check for Tkinter/Tix
 if test "x$tkinter" = "x"; then
    if test "x$pythonprog" != "x"; then
-      tkinter=`echo -e "try:\n  import Tix\n  print 'yes'\nexcept:\n  print 'no'\n" | python`
+      tkinter=`echo "
+try:
+  import Tix
+  print 'yes'
+except:
+  print 'no'
+" | python`
    fi
 fi
 echo "checking for tkinter... $tkinter"
