--- Makefile.am.orig	2017-12-20 12:11:54.000000000 -0600
+++ Makefile.am	2017-12-20 12:12:01.000000000 -0600
@@ -2,10 +2,10 @@
 RPMFLAGS	= -ta
 
 SUBDIRS    = include utils lib unix $(GLIB_DIR) $(TCL_DIR) libedit cmdlang \
-	     ui lanserv sample doc man $(SWIG_DIR)
+	     ui sample doc man $(SWIG_DIR)
 
 DIST_SUBDIRS = include utils lib unix glib tcl libedit cmdlang \
-	     ui lanserv sample doc man swig
+	     ui sample doc man swig
 
 EXTRA_DIST = FAQ TODO README.Force README.MotorolaMXP OpenIPMI.spec.in \
 	     OpenIPMI.spec ipmi.init ipmi.sysconf COPYING.BSD \
