--- unix/posix_os_hnd.c.orig	2012-09-13 12:00:06.000000000 -0500
+++ unix/posix_os_hnd.c	2012-09-13 12:00:19.000000000 -0500
@@ -51,9 +51,9 @@
 
 /* CHEAP HACK - we don't want the user to have to provide this any
    more. */
-extern void posix_vlog(char                 *format,
+void posix_vlog(char                 *format,
 		       enum ipmi_log_type_e log_type,
-		       va_list              ap);
+		       va_list              ap) {};
 #pragma weak posix_vlog
 
 typedef struct iposix_info_s
--- unix/posix_thread_os_hnd.c.orig	2012-09-13 12:15:55.000000000 -0500
+++ unix/posix_thread_os_hnd.c	2012-09-13 12:16:10.000000000 -0500
@@ -59,9 +59,9 @@
 
 /* CHEAP HACK - we don't want the user to have to provide this any
    more. */
-extern void posix_vlog(char                 *format,
+void posix_vlog(char                 *format,
 		       enum ipmi_log_type_e log_type,
-		       va_list              ap);
+		       va_list              ap) {};
 #pragma weak posix_vlog
 
 typedef struct pt_os_hnd_data_s
