--- setup.py.orig	2020-07-15 23:17:08.000000000 +0300
+++ setup.py	2021-05-04 16:33:14.000000000 +0300
@@ -16,7 +16,7 @@
     name="curtsies",
     version=version(),
     description="Curses-like terminal wrapper, with colored strings!",
-    long_description=open("readme.md", encoding="utf-8").read(),
+    long_description=io.open("readme.md", encoding="utf-8").read(),
     long_description_content_type="text/markdown",
     url="https://github.com/bpython/curtsies",
     author="Thomas Ballinger",
