--- app/gimp-update.c.orig	2023-05-06 15:29:29.000000000 -0400
+++ app/gimp-update.c	2023-05-06 15:30:24.000000000 -0400
@@ -519,7 +519,7 @@
 void
 gimp_update_check (GimpCoreConfig *config)
 {
-#ifdef PLATFORM_OSX
+#if defined(PLATFORM_OSX) && defined(MAC_OS_X_VERSION_10_9)
   const gchar *gimp_versions;
 
   gimp_versions = gimp_get_version_url ();
@@ -569,13 +569,6 @@
 
     g_idle_add ((GSourceFunc) gimp_check_updates_process_idle, (gpointer) update_results);
   }] resume];
-#else
-  GFile *gimp_versions;
-
-  gimp_versions = g_file_new_for_uri (gimp_get_version_url ());
-
-  g_file_load_contents_async (gimp_versions, NULL, gimp_check_updates_callback, config);
-  g_object_unref (gimp_versions);
 #endif /* PLATFORM_OSX */
 }
 
