--- plug-ins/common/web-browser.c.orig	2015-08-15 00:52:41.000000000 -0700
+++ plug-ins/common/web-browser.c	2016-10-09 03:07:01.000000000 -0700
@@ -181,11 +181,25 @@
 
   return TRUE;
 #else
+  gboolean rstatus;
+  gchar *command_line;
+
   gimp_ui_init (PLUG_IN_BINARY, FALSE);
 
-  return gtk_show_uri (gdk_screen_get_default (),
+  rstatus = gtk_show_uri (gdk_screen_get_default (),
                        url,
                        gtk_get_current_event_time(),
                        error);
+
+  if (!rstatus) {
+     command_line = g_strdup_printf ("open %s", url);
+
+     rstatus = g_spawn_command_line_async (command_line,
+                       error);
+
+     g_free (command_line);
+  }
+  
+  return rstatus;
 #endif
 }
