From 7e19906827d301eb70275dba089849a632a0eabe Mon Sep 17 00:00:00 2001
From: Jehan <jehan@girinstud.io>
Date: Sat, 28 Apr 2018 00:27:31 +0200
Subject: gimp.pc: Fix reference to gegl-0.3.

---
 gimp.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gimp.pc.in b/gimp.pc.in
index 9cf1da9..9acf6ca 100644
--- gimp.pc.in
+++ gimp.pc.in
@@ -12,6 +12,6 @@ gimplocaledir=@gimplocaledir@
 Name: GIMP
 Description: GIMP Library
 Version: @GIMP_REAL_VERSION@
-Requires: gdk-pixbuf-2.0 >= @GDK_PIXBUF_REQUIRED_VERSION@ cairo >= @CAIRO_REQUIRED_VERSION@ gegl-0.3 >= @GEGL_REQUIRED_VERSION@
+Requires: gdk-pixbuf-2.0 >= @GDK_PIXBUF_REQUIRED_VERSION@ cairo >= @CAIRO_REQUIRED_VERSION@ gegl-0.4 >= @GEGL_REQUIRED_VERSION@
 Libs: -L${libdir} -lgimp-@GIMP_API_VERSION@ -lgimpmath-@GIMP_API_VERSION@  -lgimpconfig-@GIMP_API_VERSION@ -lgimpcolor-@GIMP_API_VERSION@ -lgimpbase-@GIMP_API_VERSION@ @RT_LIBS@
 Cflags: -I${includedir}/gimp-@GIMP_API_VERSION@
-- 
cgit v0.12
