--- plug-ins/common/web-browser.c.orig	2017-01-28 10:58:03.000000000 -0800
+++ plug-ins/common/web-browser.c	2017-02-07 12:04:30.000000000 -0800
@@ -201,13 +201,25 @@
   return retval;
 
 #else
+  gboolean rstatus;
+  gchar *command_line;
 
   gimp_ui_init (PLUG_IN_BINARY, FALSE);
 
-  return gtk_show_uri (gdk_screen_get_default (),
+  rstatus = gtk_show_uri (gdk_screen_get_default (),
                        url,
                        gtk_get_current_event_time(),
                        error);
 
+  if (!rstatus) {
+     command_line = g_strdup_printf ("open %s", url);
+
+     rstatus = g_spawn_command_line_async (command_line,
+                       error);
+
+     g_free (command_line);
+  }
+  
+  return rstatus;
 #endif
 }
