--- etc/gimprc.orig	2011-12-12 14:10:42.000000000 -0800
+++ etc/gimprc	2011-12-12 14:13:46.000000000 -0800
@@ -619,6 +619,8 @@
 # 
 # (help-browser gimp)
 
+(help-browser web-browser)
+
 # Sets the external web browser to be used.  This can be an absolute path or
 # the name of an executable to search for in the user's PATH. If the command
 # contains '%s' it will be replaced with the URL, else the URL will be
@@ -627,12 +629,16 @@
 # 
 # (web-browser "firefox %s")
 
+(web-browser "open %s")
+
 # When enabled, the online user manual will be used by the help system.
 # Otherwise the locally installed copy is used.  Possible values are yes and
 # no.
 # 
 # (user-manual-online no)
 
+(user-manual-online no)
+
 # The location of the online user manual. This is used if
 # 'user-manual-online' is enabled.  This is a string value.
 # 
@@ -644,12 +650,16 @@
 # 
 # (toolbox-window-hint utility)
 
+(toolbox-window-hint normal)
+
 # The window type hint that is set on dock windows. This may affect the way
 # your window manager decorates and handles dock windows.  Possible values
 # are normal, utility and keep-above.
 # 
 # (dock-window-hint utility)
 
+(dock-window-hint normal)
+
 # When enabled, dock windows (the toolbox and palettes) are set to be
 # transient to the active image window. Most window managers will keep the
 # dock windows above the image window then, but it may also have other
