--- app/composite/gimp-composite-altivec.h.orig	2018-04-20 10:01:59.000000000 -0700
+++ app/composite/gimp-composite-altivec.h	2018-04-20 10:02:50.000000000 -0700
@@ -12,7 +12,7 @@
 #if defined(USE_ALTIVEC)
 #if defined(ARCH_PPC)
 #if __GNUC__ >= 3
-#define COMPILE_ALTIVEC_IS_OKAY (1)
+#undef COMPILE_ALTIVEC_IS_OKAY
 #endif /* __GNUC__ >= 3 */
 #endif /* defined(ARCH_PPC) */
 #endif /* defined(USE_ALTIVEC) */
