From 72942f4e6b4aa25698b6432f7a20494f61ec3fca Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Tue, 04 Feb 2014 22:28:01 +0000
Subject: app: halt the transform tool when switching layer|selection|path transform

(cherry picked from commit 1335bc42c89c7b739bba54400b6adcee3abd7423)
---
diff --git a/app/tools/gimptransformtool.c b/app/tools/gimptransformtool.c
index 009659c..8c31222 100644
--- app/tools/gimptransformtool.c
+++ app/tools/gimptransformtool.c
@@ -667,12 +667,17 @@ gimp_transform_tool_options_notify (GimpTool         *tool,
 
   GIMP_TOOL_CLASS (parent_class)->options_notify (tool, options, pspec);
 
+  if (! strcmp (pspec->name, "type"))
+    {
+      gimp_tool_control (tool, GIMP_TOOL_ACTION_HALT, tool->display);
+      return;
+    }
+
   if (tr_tool->use_grid)
     {
       gimp_draw_tool_pause (GIMP_DRAW_TOOL (tr_tool));
 
-      if (! strcmp (pspec->name, "type") ||
-          ! strcmp (pspec->name, "direction"))
+      if (! strcmp (pspec->name, "direction"))
         {
           if (tr_tool->function != TRANSFORM_CREATING)
             {
--
cgit v0.9.2
