--- app/widgets/gimpdashboard.c.orig	2020-08-29 17:02:04.000000000 +0200
+++ app/widgets/gimpdashboard.c	2020-08-29 17:55:26.000000000 +0200
@@ -2398,6 +2398,13 @@

 #ifdef HAVE_MEMORY_GROUP
 #ifdef PLATFORM_OSX
+  #if MAC_OS_X_VERSION_MAX_ALLOWED < 1080
+    #define MACH_TASK_BASIC_INFO_COUNT TASK_BASIC_INFO_COUNT
+    #define mach_task_basic_info_data_t task_basic_info_data_t
+
+    #define MACH_TASK_BASIC_INFO TASK_BASIC_INFO
+    #define mach_task_basic_info task_basic_info
+  #endif
 static void
 gimp_dashboard_sample_memory_used (GimpDashboard *dashboard,
                                    Variable       variable)
