From 6c2cd97b868e7742f9a3ade611713e46a1759d1c Mon Sep 17 00:00:00 2001
From: Kevin Cozens <kevin@ve3syb.ca>
Date: Mon, 03 Feb 2014 22:30:04 +0000
Subject: Removed extraneous 'x' when writing character constants. Fixes bug #721775.

---
diff --git a/plug-ins/script-fu/tinyscheme/scheme.c b/plug-ins/script-fu/tinyscheme/scheme.c
index 8abddcd..1d288f5 100644
--- plug-ins/script-fu/tinyscheme/scheme.c
+++ plug-ins/script-fu/tinyscheme/scheme.c
@@ -2177,7 +2177,7 @@ static void atom2str(scheme *sc, pointer l, int f, char **pp, int *plen) {
                       snprintf(p,STRBUFFSIZE,"#\\x%x",c); break;
                     }
 #endif
-                    snprintf(p,STRBUFFSIZE,"#\\x%c",c); break;
+                    snprintf(p,STRBUFFSIZE,"#\\%c",c); break;
                }
           }
      } else if (is_symbol(l)) {
--
cgit v0.9.2
