--- etc/gimprc.in.orig	2020-02-16 07:04:57.000000000 -0800
+++ etc/gimprc.in	2020-02-22 17:04:36.000000000 -0800
@@ -831,11 +831,13 @@
 # string value.
 # 
 # (help-locales "")
+(help-locales "en")
 
 # Sets the browser used by the help system.  Possible values are gimp and
 # web-browser.
 # 
 # (help-browser web-browser)
+(help-browser web-browser)
 
 # When enabled, a search of actions will also return inactive actions. 
 # Possible values are yes and no.
@@ -862,6 +864,7 @@
 # windows.  Possible values are normal, utility and keep-above.
 # 
 # (dock-window-hint utility)
+(dock-window-hint normal)
 
 # Sets the handedness for cursor positioning.  Possible values are left and
 # right.
