--- Makefile.in.orig	2009-03-15 00:46:46.000000000 -0500
+++ Makefile.in	2009-03-15 00:48:41.000000000 -0500
@@ -251,7 +251,7 @@
 	$(INSTALL) -m 644 pure.html purelib.html $(DESTDIR)$(libdir)/pure-$(version)
 	ln -sf $(libdir)/pure-$(version) $(DESTDIR)$(libdir)/pure
 	$(INSTALL) -m 644 $(srcdir)/pure.1 $(DESTDIR)$(man1dir)/pure-$(version).1
-	ln -sf $(man1dir)/pure-$(version).1 $(DESTDIR)$(man1dir)/pure.1
+	ln -sf pure-$(version).1 $(DESTDIR)$(man1dir)/pure.1
 else
 	$(INSTALL) $(srcdir)/runtime.h $(DESTDIR)$(includedir)/pure
 	for x in $(srcdir)/lib/*.pure; do $(INSTALL) -m 644 $$x $(DESTDIR)$(libdir)/pure; done
