--- Makefile.in.orig	2010-05-18 01:25:05.000000000 -0500
+++ Makefile.in	2010-05-24 21:41:10.000000000 -0500
@@ -332,7 +332,7 @@
 	ln -sf $(bindir)/$(pure_exe) $(DESTDIR)$(bindir)/pure$(EXE)
 	ln -sf $(pure_includedir) $(DESTDIR)$(includedir)/pure
 	ln -sf $(pure_libdir) $(DESTDIR)$(libdir)/pure
-	ln -sf $(man1dir)/$(pure_1) $(DESTDIR)$(man1dir)/pure.1
+	ln -sf $(pure_1) $(DESTDIR)$(man1dir)/pure.1
 endif
 
 uninstall:
