diff -ur ../kcalc-4.10.0-orig/CMakeLists.txt ./CMakeLists.txt
--- ../kcalc-4.10.0-orig/CMakeLists.txt	2013-01-24 07:02:46.000000000 +0900
+++ ./CMakeLists.txt	2013-02-23 18:56:37.000000000 +0900
@@ -36,6 +36,13 @@
     endif()
 endif(NOT GMP_FOUND)
 
+find_package(MPFR)
+macro_log_feature( MPFR_FOUND "MPFR" "The GNU Multiple Precision Floating-Pointer Library" "http://mpfr.org/" TRUE "" "Required for building KCalc.")
+
+if(MPFR_FOUND)
+ add_definitions (-DKNUMBER_USE_MPFR)
+endif(MPFR_FOUND)
+
 include(CheckTypeSize)
 include(CheckFunctionExists)
 include(CheckIncludeFiles)
diff -ur ../kcalc-4.10.0-orig/knumber/tests/CMakeLists.txt ./knumber/tests/CMakeLists.txt
--- ../kcalc-4.10.0-orig/knumber/tests/CMakeLists.txt	2013-01-24 07:02:47.000000000 +0900
+++ ./knumber/tests/CMakeLists.txt	2013-02-23 18:41:36.000000000 +0900
@@ -5,4 +5,4 @@
 
 kde4_add_unit_test(knumbertest TESTNAME KNumber ${knumbertest_SRCS})
 
-target_link_libraries(knumbertest ${KDE4_KDECORE_LIBS} ${GMP_LIBRARIES})
+target_link_libraries(knumbertest ${KDE4_KDECORE_LIBS} ${GMP_LIBRARIES} ${MPFR_LIBRARIES})
