--- libgps.pc.in.orig	2011-11-22 02:49:33.000000000 -0600
+++ libgps.pc.in	2012-05-02 16:59:47.000000000 -0500
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}@libdir@
+libdir=${exec_prefix}/@libdir@
 includedir=${prefix}/include
 
 Name: GPSD
--- libgpsd.pc.in.orig	2011-11-22 02:49:33.000000000 -0600
+++ libgpsd.pc.in	2012-05-02 16:59:37.000000000 -0500
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}@libdir@
+libdir=${exec_prefix}/@libdir@
 includedir=${prefix}/include
 
 Name: libgpsd
