--- SConstruct.orig	2018-10-02 17:05:28.000000000 -0700
+++ SConstruct	2018-11-17 18:45:31.000000000 -0800
@@ -383,7 +383,8 @@ for key, value in os.environ.items():
 
 # Placeholder so we can kluge together something like VPATH builds.
 # $SRCDIR replaces occurrences for $(srcdir) in the autotools build.
-env['SRCDIR'] = '.'
+# scons can get confused if this is not a full path
+env['SRCDIR'] = os.getcwd()
 
 # We may need to force slow regression tests to get around race
 # conditions in the pty layer, especially on a loaded machine.
@@ -1366,6 +1367,7 @@ if not env['python']:
     python_progs = []
 else:
     python_progs = ["gegps", "gpscat", "gpsfake", "gpsprof", "ubxtool", "zerk"]
+    python_deps = {'gpscat': 'packet'}
 
     if env['xgps']:
         # check for pycairo
@@ -1388,6 +1390,8 @@ else:
 
     if env['xgps']:
         python_progs.extend(["xgps", "xgpsspeed"])
+        python_deps['xgps'] = 'clienthelpers'
+
     python_modules = Glob('gps/*.py')
 
     # Build Python binding
@@ -1471,6 +1475,11 @@ else:
             )
         python_compiled_libs[ext] = python_env.SharedLibrary(
             ext, python_objects[ext])
+
+    # Make sure we know about compiled dependencies
+    for prog, dep in python_deps.items():
+        env.Depends(prog, python_compiled_libs['gps' + os.sep + dep])
+
     # Make PEP 241 Metadata 1.0.
     # Why not PEP 314 (V1.1) or PEP 345 (V1.2)?
     # V1.2 and V1.2 require a Download-URL to an installable binary
@@ -1906,7 +1915,7 @@ if python_progs:
     verenv = env['ENV'].copy()
     verenv['DISPLAY'] = ''  # Avoid launching X11 in X11 progs
     for prog in python_progs:
-        vchk += '$SRCDIR/%s -V\n' % prog
+        vchk += '$PYTHON $SRCDIR/%s -V\n' % prog
     python_versions = Utility('python-versions', [python_progs], vchk,
                               ENV=verenv)
 
