--- SConstruct.orig	2012-01-13 03:41:41.000000000 +1100
+++ SConstruct	2012-01-25 18:50:00.000000000 +1100
@@ -398,7 +399,7 @@
     # use chrpath to edit this out of RPATH.
     if env["shared"]:
         env.Prepend(RPATH=[os.path.realpath(os.curdir)])
-else:
+elif env.subst('$PLATFORM') != 'darwin':
     print "chrpath is not available, forcing static linking."
     env["shared"] = False
 
@@ -1140,7 +1141,7 @@
     binaryinstall.append(LibraryInstall(qt_env, installdir('libdir'), compiled_qgpsmmlib))
 
 # We don't use installdir here in order to avoid having DESTDIR affect the rpath
-if env["shared"]:
+if env["shared"] and env.subst('$PLATFORM') != 'darwin':
     env.AddPostAction(binaryinstall, '$CHRPATH -r "%s" "$TARGET"' \
                       % (installdir('libdir', False), ))
 
@@ -1154,7 +1155,7 @@
     python_module_dir = python_lib_dir + os.sep + 'gps'
     python_extensions_install = python_env.Install( DESTDIR + python_module_dir,
                                                     python_built_extensions)
-    if not env['debug'] or env['profiling']:
+    if not env['debug'] and not env['profiling'] and env['strip']:
         python_env.AddPostAction(python_extensions_install, '$STRIP $TARGET')
 
     python_modules_install = python_env.Install( DESTDIR + python_module_dir,
