diff --git glib/tests/atomic.c.orig glib/tests/atomic.c
index 8bc100f..fb8cdcd 100644
--- glib/tests/atomic.c.orig
+++ glib/tests/atomic.c
@@ -97,12 +97,9 @@ test_types (void)
   /* Note that atomic variables should almost certainly not be marked as
    * `volatile` — see http://isvolatileusefulwiththreads.in/c/. This test exists
    * to make sure that we don’t warn when built against older third party code. */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
   g_atomic_pointer_set (&vp_str_vol, NULL);
   res = g_atomic_pointer_compare_and_exchange (&vp_str_vol, NULL, str);
   g_assert_true (res);
-#pragma GCC diagnostic pop
 
   g_atomic_pointer_set (&ip, 0);
   ip2 = g_atomic_pointer_get (&ip);
