diff --git glib/gi18n-lib.h glib/gi18n-lib.h
index 74e50a3..f6e57c3 100644
--- glib/gi18n-lib.h
+++ glib/gi18n-lib.h
@@ -20,7 +20,7 @@
 
 #include <glib.h>
 
-#include <libintl.h>
+#include <@PREFIX@/include/libintl.h>
 #include <string.h>
 
 #ifndef GETTEXT_PACKAGE
diff --git glib/gi18n.h glib/gi18n.h
index 5f996d2..d198cbd 100644
--- glib/gi18n.h
+++ glib/gi18n.h
@@ -20,7 +20,7 @@
 
 #include <glib.h>
 
-#include <libintl.h>
+#include <@PREFIX@/include/libintl.h>
 #include <string.h>
 
 #define  _(String) gettext (String)
