--- src/convert.c	2024-08-20 15:21:53.000000000 +0800
+++ src/convert.c	2024-12-05 08:28:01.000000000 +0800
@@ -68,7 +68,7 @@
 }
 static R_INLINE void R_mpfr_GET_EXP(mpfr_t r, unsigned int *ex,
                                     unsigned int ex1) {
-    r->_mpfr_exp = (mpft_exp_t) CAST_SIGNED(ex[0], unsigned int, int);
+    r->_mpfr_exp = (mpfr_exp_t) CAST_SIGNED(ex[0], unsigned int, int);
 }
 
 
