--- components/pango_image/src/image_io_exr.cpp.orig	2026-04-11 03:27:37
+++ components/pango_image/src/image_io_exr.cpp	2026-05-12 22:30:06
@@ -56,12 +56,12 @@
         return true;
     }
 
-    virtual uint64_t tellg ()
+    virtual Imf::Int64 tellg ()
     {
         return std::streamoff (_is->tellg());
     }
 
-    virtual void seekg (uint64_t pos)
+    virtual void seekg (Imf::Int64 pos)
     {
         _is->seekg (pos);
     }
