diff --git a/src/video/cocoa/SDL_cocoaopengl.m b/src/video/cocoa/SDL_cocoaopengl.m
index 2cdbc17e9..ff7a9c00b 100644
--- src/video/cocoa/SDL_cocoaopengl.m
+++ src/video/cocoa/SDL_cocoaopengl.m
@@ -44,6 +44,13 @@
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #endif
 
+/* _Nullable is available starting Xcode 7 */
+#ifdef __has_feature
+#if !__has_feature(nullability)
+#define _Nullable
+#endif
+#endif
+
 static SDL_bool SDL_opengl_async_dispatch = SDL_FALSE;
 
 static void SDLCALL
