diff --git a/src/video/cocoa/SDL_cocoaopengl.h b/src/video/cocoa/SDL_cocoaopengl.h
index fe8a8e51c134..9330b5744936 100644
--- src/video/cocoa/SDL_cocoaopengl.h
+++ src/video/cocoa/SDL_cocoaopengl.h
@@ -59,6 +59,9 @@ struct SDL_GLDriverData
 - (SDL_Window*)window;
 - (void)explicitUpdate;
 - (void)dealloc;
+
+@property (retain, nonatomic) NSOpenGLPixelFormat* openglPixelFormat;  // macOS 10.10 has -[NSOpenGLContext pixelFormat] but this handles older OS releases.
+
 @end
 
 /* OpenGL functions */
diff --git a/src/video/cocoa/SDL_cocoaopengl.m b/src/video/cocoa/SDL_cocoaopengl.m
index 607f30716164..2cdbc17e9f0c 100644
--- src/video/cocoa/SDL_cocoaopengl.m
+++ src/video/cocoa/SDL_cocoaopengl.m
@@ -76,6 +76,7 @@ - (id)initWithFormat:(NSOpenGLPixelFormat *)format
 {
     self = [super initWithFormat:format shareContext:share];
     if (self) {
+        self.openglPixelFormat = format;
         SDL_AtomicSet(&self->dirty, 0);
         self->window = NULL;
         SDL_AtomicSet(&self->swapIntervalSetting, 0);
@@ -100,7 +101,7 @@ - (id)initWithFormat:(NSOpenGLPixelFormat *)format
 - (void)movedToNewScreen
 {
     if (self->displayLink) {
-        CVDisplayLinkSetCurrentCGDisplayFromOpenGLContext(self->displayLink, [self CGLContextObj], [[self pixelFormat] CGLPixelFormatObj]);
+        CVDisplayLinkSetCurrentCGDisplayFromOpenGLContext(self->displayLink, [self CGLContextObj], [[self openglPixelFormat] CGLPixelFormatObj]);
     }
 }
 
