--- src/video/SDL_egl.c.orig	2025-01-01 19:47:53
+++ src/video/SDL_egl.c	2025-10-24 21:12:41
@@ -82,10 +82,10 @@
 
 #elif defined(SDL_VIDEO_DRIVER_COCOA)
 /* EGL AND OpenGL ES support via ANGLE */
-#define DEFAULT_EGL        "libEGL.dylib"
-#define DEFAULT_OGL_ES2    "libGLESv2.dylib"
-#define DEFAULT_OGL_ES_PVR "libGLES_CM.dylib"   //???
-#define DEFAULT_OGL_ES     "libGLESv1_CM.dylib" //???
+#define DEFAULT_EGL        "@PREFIX@/lib/libEGL.dylib"
+#define DEFAULT_OGL_ES2    "@PREFIX@/lib/libGLESv2.dylib"
+#define DEFAULT_OGL_ES_PVR "@PREFIX@/lib/libGLES_CM.dylib"   //???
+#define DEFAULT_OGL_ES     "@PREFIX@/lib/libGLESv1_CM.dylib" //???
 
 #elif defined(__OpenBSD__)
 /* OpenBSD */
