--- src/bspeval.cc.orig	2017-04-02 03:37:31.000000000 -0500
+++ src/bspeval.cc	2020-12-19 15:41:44.000000000 -0600
@@ -113,7 +113,7 @@
       error("bspeval: the knot vector should be a real vector."); 
       return true; 
     } 
-  if (!args(3).is_real_type()) 
+  if (!args(3).isreal()) 
     { 
       error("bspeval: the set of parametric points should be an array of doubles."); 
       return true; 
