diff -Naur a/gdk/quartz/gdkquartz.h b/gdk/quartz/gdkquartz.h
--- gdk/quartz/gdkquartz.h      2021-01-11 20:51:03.000000000 +0000
+++ gdk/quartz/gdkquartz.h      2021-01-11 20:42:40.000000000 +0000
@@ -54,7 +54,9 @@
   GDK_OSX_SIERRA = 12,
   GDK_OSX_HIGH_SIERRA = 13,
   GDK_OSX_MOJAVE = 14,
-  GDK_OSX_CURRENT = 14,
+  GDK_OSX_CATALINA = 15,
+  GDK_OSX_BIG_SUR = 16,
+  GDK_OSX_CURRENT = 16,
   GDK_OSX_NEW = 99
 } GdkOSXVersion;
 

