--- configure.ac~	2026-07-12 13:20:29
+++ configure.ac	2026-07-12 13:19:30
@@ -79,7 +79,7 @@
 AM_CONDITIONAL([USE_FTXUI], [test x$with_ftxui == xyes])
 AS_IF([test "x${with_ftxui}" == xyes], [
    AC_DEFINE([HAVE_FTXUI], [1], [Build to use FINAL CUT FTXUI library])
-   PKG_CHECK_MODULES([FTXUI], [ftxui >= 5.0.0 ftxui < 7.0.0])
+   PKG_CHECK_MODULES([FTXUI], [ftxui >= 5.0.0 ftxui < 8.0.0])
 ])
 
 dnl https://wiki.wxwidgets.org/Autoconf
@@ -105,7 +105,7 @@
 LIBS="$LIBS $LIBPUGIXML_LIBS $WX_LIBS $FINALCUT_LIBS $FTXUI_LIBS"
 
 dnl Ensures the selected compiler is clang when compiling wxWidgets on an Apple platform
-FDSD_WX_CHECK_COMPILER
+dnl FDSD_WX_CHECK_COMPILER
 
 # Checks for header files.
 AC_CHECK_HEADERS([getopt.h])
