--- src/main.cpp~	2026-07-10 07:51:40
+++ src/main.cpp	2026-07-10 07:51:55
@@ -91,13 +91,19 @@
   auto usage = [=] {
     simple_usage();
 #ifdef HAVE_FINALCUT
+// --finalcut parameter only needed if there are multiple front-ends
+#ifdef HAVE_FTXUI
     std::cout << "  -t, --finalcut\t\t";
     std::cout << "run with FINALCUT user interface\n";
 #endif
+#endif
 #ifdef HAVE_FTXUI
+// --ftxui parameter only needed if there are multiple front-ends
+#ifdef HAVE_FINALCUT
     std::cout << "  -x, --ftxui\t\t\t";
     std::cout << "run with FTXUI user interface\n";
 #endif
+#endif
   };
 
   std::string filename;
@@ -215,7 +221,7 @@
         usage();
         exit(0xFF);
       }
-      KitListFtxui ftxui_app;
+      fdsd::ftxui::KitListFtxui ftxui_app;
       ftxui_app.run(filename);
     }
 #endif // HAVE_FTXUI
