--- configure.in.orig	2010-06-21 06:51:38.000000000 -0700
+++ configure.in	2010-10-06 16:15:01.000000000 -0700
@@ -235,7 +235,10 @@
 
 pam_status="no"
 if test "$enable_pam" != "no"; then
-	AC_CHECK_HEADER(security/pam_modules.h, have_pam=yes, have_pam=no)
+        AC_CHECK_HEADERS(security/pam_modules.h, have_pam=yes, have_pam=no)
+        if test "$have_pam" = "no"; then
+                AC_CHECK_HEADERS(pam/pam_modules.h, have_pam=yes, have_pam=no)
+        fi
 	if test "$enable_pam" = "yes" -a "$have_pam" = "no"; then
 		AC_MSG_ERROR(The PAM headers are missing)
 	elif test "$have_pam" = "yes"; then
