From c3c3018155f35af5585f79c8c178f4d88165a22d Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Wed, 26 Jul 2023 14:05:08 +0800
Subject: [PATCH] Add some missing headers


diff --git src/dynlibs/mgd/mgd.c src/dynlibs/mgd/mgd.c
index 859bf3d..0e9f4a8 100644
--- src/dynlibs/mgd/mgd.c
+++ src/dynlibs/mgd/mgd.c
@@ -22,7 +22,9 @@
 /* Moscow ML specific includes: */
 
 #include <alloc.h>		/* For alloc_tuple, ...      */
-#include <mlvalues.h>		/* For Val_unit, Long_val, String_val, ... */
+#include <mlvalues.h>	/* For Val_unit, Long_val, String_val, ... */
+#include <fail.h>		/* for failwith */
+#include <memory.h>		/* for modify */
 
 /* Representation of images.  
 
diff --git src/dynlibs/mmysql/mmysql.c src/dynlibs/mmysql/mmysql.c
index e8c079a..083771c 100644
--- src/dynlibs/mmysql/mmysql.c
+++ src/dynlibs/mmysql/mmysql.c
@@ -3,11 +3,11 @@
    sestoft@dina.kvl.dk 1999-08-07, 2000-05-30, 2002-07-25 */
 
 #include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
 
 #ifdef WIN32
 #include <windows.h>
-#include <stdio.h>
-#include <string.h>
 #endif
 
