From 6a910c2115c9a9722a64baee9f2f427d804685c9 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Wed, 26 Jul 2023 14:39:16 +0800
Subject: [PATCH] Address lvalue error


diff --git src/dynlibs/mmysql/mmysql.c src/dynlibs/mmysql/mmysql.c
index 083771c..621c6ab 100644
--- src/dynlibs/mmysql/mmysql.c
+++ src/dynlibs/mmysql/mmysql.c
@@ -76,9 +76,9 @@ void dbresult_finalize(value dbresval)
   MYSQL_ROW_OFFSET* index = DBresultindex_val(dbresval);
   if (dbres != NULL) {
     mysql_free_result(dbres);
-    DBresult_val(dbresval) = NULL;
+    DBresult_val(dbresval) == NULL;
     stat_free((char*)index);
-    DBresultindex_val(dbresval) = NULL;
+    DBresultindex_val(dbresval) == NULL;
   }
 }
 
