From f5d859928130323a016696a8c2ffcd0ffb16b181 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Wed, 26 Jul 2023 16:19:22 +0800
Subject: [PATCH] Placeholders for disabling werrors


diff --git src/dynlibs/mgdbm/Makefile src/dynlibs/mgdbm/Makefile
index 7218150..3fb006e 100644
--- src/dynlibs/mgdbm/Makefile
+++ src/dynlibs/mgdbm/Makefile
@@ -12,7 +12,7 @@ GDBMINCDIR=${HOME}/c/gdbm-1.7.3
 
 include ../../Makefile.inc
 
-OPTS=-fno-defer-pop
+OPTS=-fno-defer-pop @WERROR@
 CFLAGS=-Dunix -O2 $(OPTS) $(ADDDYNLIBCFLAGS) -I$(INCDIR) -I ${GDBMINCDIR}
 BASELIBS=-lm -ldl
 
diff --git src/dynlibs/mpq/Makefile src/dynlibs/mpq/Makefile
index 8d8cec4..c5b4474 100644
--- src/dynlibs/mpq/Makefile
+++ src/dynlibs/mpq/Makefile
@@ -20,7 +20,7 @@ PGSQLINCDIR=/usr/include/postgresql
 
 include ../../Makefile.inc
 
-OPTS=-fno-defer-pop
+OPTS=-fno-defer-pop @WERROR@
 CFLAGS=-Dunix -O2 $(OPTS) $(ADDDYNLIBCFLAGS) -I$(INCDIR) -I ${PGSQLINCDIR}
 
 MOSMLTOOLS=camlrunm $(MOSMLHOME)/tools
diff --git src/dynlibs/mregex/Makefile src/dynlibs/mregex/Makefile
index 2fd659b..1f0675c 100644
--- src/dynlibs/mregex/Makefile
+++ src/dynlibs/mregex/Makefile
@@ -5,7 +5,7 @@
 
 include ../../Makefile.inc
 
-OPTS=-fno-defer-pop
+OPTS=-fno-defer-pop @WERROR@
 CFLAGS=-Dunix -O2 $(OPTS) $(ADDDYNLIBCFLAGS) -I$(INCDIR)
 
 MOSMLTOOLS=camlrunm $(MOSMLHOME)/tools
