--- src/Makefile.inc	2006-08-27 16:20:05.000000000 +0200
+++ src/Makefile.inc	2006-08-27 16:30:56.000000000 +0200
@@ -2,7 +2,7 @@
 
 # Where to install stuff
 
-MOSMLHOME=${HOME}/mosml
+MOSMLHOME=${DESTROOT}
 
 # Various utility programs
 INSTALL_PROGRAM=cp
@@ -22,14 +22,14 @@
 # To compile the runtime system camlrunm under Cygwin with support for
 # dynamically loadable libraries (DDLs), uncomment these:
 
-ADDPRIMS=dynlib.c
-ADDOBJS=dynlib.o
-ADDRUNCFLAGS=
-ADDDYNLIBCFLAGS=
+#ADDPRIMS=dynlib.c
+#ADDOBJS=dynlib.o
+#ADDRUNCFLAGS=
+#ADDDYNLIBCFLAGS=
 
 # For Linux (x86 or Alpha), use:
 # ------------------------------
-CPP=/lib/cpp -P -traditional -Dunix -Umsdos
+CPP=/usr/bin/cpp -P -traditional -Dunix -Umsdos
 STRIP=strip
 LD=gcc -rdynamic
 DYNLD=ld -shared
@@ -114,10 +114,10 @@
 # DOCDIR contains documentation
 
 BINDIR=${MOSMLHOME}/bin
-LIBDIR=${MOSMLHOME}/lib
-INCDIR=${MOSMLHOME}/include
-DOCDIR=${MOSMLHOME}/doc
-TOOLDIR=${MOSMLHOME}/tools
+LIBDIR=${MOSMLHOME}/lib/mosml
+INCDIR=${MOSMLHOME}/include/mosml
+DOCDIR=${MOSMLHOME}/share/doc/mosml
+TOOLDIR=${MOSMLHOME}/bin
 # LIBDIR=${MOSMLHOME}/lib/moscow_ml
 # TOOLDIR=${MOSMLHOME}/libexec/moscow_ml
 
