--- /opt/local/var/macports/build/_opt_local_var_macports_sources_github.com_macports_macports-ports_perl_p5-tcl-ptk/p5.26-tcl-ptk/work/Tcl-pTk-0.92/lib/Tcl/pTk/Widget.pm	2016-08-21 07:34:35.000000000 -0500
+++ lib/Tcl/pTk/Widget.pm	2018-06-14 19:10:26.000000000 -0500
@@ -2417,9 +2417,11 @@
  # events on other platforms.
 
  $mw->bind($class, '<MouseWheel>',
-	       [ sub { $_[0]->yview('scroll',-int(($_[1]/120)),'units') }, Tcl::pTk::Ev("D")]);
+    $mw->windowingsystem eq 'aqua'
+	    ?  [ sub { $_[0]->yview('scroll',-($_[1]),'units') }, Tcl::pTk::Ev("D")]
+	    :  [ sub { $_[0]->yview('scroll',-int(($_[1]/120)),'units') }, Tcl::pTk::Ev("D")]);
 
- if ($Tcl::pTk::platform eq 'unix')
+ if ($mw->windowingsystem eq 'x11')
   {
    # Support for mousewheels on Linux/Unix commonly comes through mapping
    # the wheel to the extended buttons.  If you have a mousewheel, find
