From 46ad2cca58df1115ef75ad8d8e4ef64075d9e271 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 23 Feb 2026 03:17:05 +0800
Subject: [PATCH 3/5] libvpx: fix target on darwin

---
 contrib/libvpx/module.defs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/libvpx/module.defs b/contrib/libvpx/module.defs
index a2c78931a..d3472e733 100644
--- a/contrib/libvpx/module.defs
+++ b/contrib/libvpx/module.defs
@@ -47,5 +47,7 @@ ifeq (darwin,$(HOST.system))
         LIBVPX.CONFIGURE.extra += --target=arm64-darwin20-gcc
     else ifeq (x86_64,$(HOST.machine))
         LIBVPX.CONFIGURE.extra += --target=x86_64-darwin14-gcc
+    else
+        LIBVPX.CONFIGURE.extra += --force-target=generic-darwin10-gcc
     endif
 endif
-- 
2.52.0

