--- configure.ac.orig	2018-06-30 23:20:53.000000000 +1000
+++ configure.ac	2020-01-26 01:19:18.000000000 +1100
@@ -296,7 +296,7 @@
         if test "x$enable_pam" != "xno"; then
             AC_CHECK_HEADERS([security/pam_appl.h pam/pam_appl.h pam_appl.h],[break])
             if test "x$ac_cv_header_security_pam_appl_h" = "xyes" ; then
-                EPAM_CFLAGS="-I/usr/include/security"
+                EPAM_CFLAGS="-I__SDKROOT/usr/include/security"
             elif test "x$ac_cv_header_pam_pam_appl_h" = "xyes" ; then
                 EPAM_CFLAGS="-I/usr/include/pam"
             elif test "x$ac_cv_header_pam_appl_h" = "xno" ; then
