--- src/runtime/CMakeLists.txt	2024-01-04 03:42:26.000000000 +0800
+++ src/runtime/CMakeLists.txt	2024-01-10 14:14:26.000000000 +0800
@@ -94,6 +94,6 @@
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_sin.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_string.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_bit.mod
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/include/lfortran/finclude
   )
 
