--- src/runtime/CMakeLists.txt	2024-10-14 23:38:20.000000000 +0800
+++ src/runtime/CMakeLists.txt	2024-10-29 08:27:41.000000000 +0800
@@ -54,6 +54,6 @@
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_ieee_arithmetic.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/lfortran_intrinsic_iso_c_binding.mod
   ${CMAKE_Fortran_MODULE_DIRECTORY}/omp_lib.mod
-  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/include/lfortran/finclude
   )
 
