--- a/scripts/mysql_install_db.pl.in	2014-05-12 23:43:21.000000000 -0430
+++ b/scripts/mysql_install_db.pl.in	2014-05-12 23:49:54.000000000 -0430
@@ -460,10 +460,10 @@
 
 my $cnfext = ( $^O =~ m/^(MSWin32|cygwin)$/ ) ? "ini" : "cnf";
 
-$config_file= "$basedir/my.$cnfext";
+$config_file= "$basedir@SYSCONFDIR@/my.$cnfext";
 
 my $cfg_template= find_in_basedir($opt,"file","my-default.$cnfext",
-				  ".", "share","share/mysql","support-files");
+                  "@SUPPORTFILESDIR@",".", "share","share/mysql","support-files");
 -e $cfg_template or cannot_find_file("my-default.$cnfext");
 
 $copy_cfg_file= $config_file;
