--- configure.orig	2023-10-05 04:58:30
+++ configure	2023-10-10 07:30:32
@@ -143,10 +143,10 @@
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
 	mkdir -p $(DESTDIR)$(PREFIX)/share/zsh/site-functions
-	mkdir -p $(DESTDIR)bash_complete_dir
+	mkdir -p $(DESTDIR)$(PREFIX)/bash_complete_dir
 	cp ./code2color ./sxw2txt ./archive_color ./lesspipe.sh ./vimcolor ./lesscomplete $(DESTDIR)$(PREFIX)/bin
 	cp ./lesspipe.1 $(DESTDIR)$(PREFIX)/share/man/man1
-	cp ./less_completion $(DESTDIR)bash_complete_dir
+	cp ./less_completion $(DESTDIR)$(PREFIX)/bash_complete_dir
 	cp ./_less $(DESTDIR)$(PREFIX)/share/zsh/site-functions
 	chmod 0755 $(DESTDIR)$(PREFIX)/bin/lesspipe.sh
 	chmod 0755 $(DESTDIR)$(PREFIX)/bin/sxw2txt
