--- doc/libzeitgeist/Makefile.am.orig	2020-03-14 20:22:29.000000000 -0700
+++ doc/libzeitgeist/Makefile.am	2020-03-14 20:24:27.000000000 -0700
@@ -48,7 +48,8 @@
 # with the "--doclet" option (it only works if the directory doesn't
 # exist prior to invocation).
 docs_c: $(FILES) $(srcdir)/zeitgeist-gtkdoc-index.sgml
-	install -D $(srcdir)/zeitgeist-gtkdoc-index.sgml $@/zeitgeist-2.0-docs.xml
+	install -d $@
+	install $(srcdir)/zeitgeist-gtkdoc-index.sgml $@/zeitgeist-2.0-docs.xml
 	$(VALADOC) \
 		-X $(top_builddir)/libzeitgeist/zeitgeist-datamodel.h \
 		-X $(top_builddir)/libzeitgeist/zeitgeist.h \
