diff --git bundler/bundler.py bundler/bundler.py
index 65b8cc0..3ca931f 100644
--- bundler/bundler.py
+++ bundler/bundler.py
@@ -618,7 +618,7 @@ class Bundler:
         self.copy_plist()
 
         # Note: could move this to xml file...
-        self.copy_path(Path("${prefix}/lib/charset.alias"))
+        self.copy_path(Path("/usr/lib/charset.alias", dest="${bundle}/Contents/Resources"))
 
         # Main binary
         path = self.project.get_main_binary()
