--- src/Makefile.orig	2022-11-07 07:13:40.000000000 -0600
+++ src/Makefile	2022-11-08 08:36:11.000000000 -0600
@@ -11,9 +11,10 @@
 #Flags for C++
 CXXFLAGS+=$(FLAGSCOMW) $(FLAGSCOMO)
 #Flags for the linker
-LDFLAGS+=-lproj
+LDFLAGS+=-L%PREFIX%/lib/proj8/lib -lproj
+CPPFLAGS+=-I%PREFIX%/lib/proj8/include
 #Export flags for compilers and linker
-export CFLAGS CXXFLAGS
+#export CFLAGS CXXFLAGS
 
 .PHONY: all
 all: compile
