--- src/Makefile.orig	2016-02-26 09:09:14.000000000 -0600
+++ src/Makefile	2016-06-04 18:21:45.000000000 -0500
@@ -34,7 +34,7 @@
 CC      ?= gcc
 CXX     ?= g++
 CFLAGS += -fopenmp -Wall -Wextra -Wconversion -O2
-OCTMEX	= $(MKOCTFILE) --mex -lgomp
+OCTMEX	= $(MKOCTFILE) --mex
 RM      = rm
 
 # Support Debian Hardening flags
@@ -79,7 +79,7 @@
 
 mexw32 win32:  $(patsubst %.mex, %.mexw32, $(PROGS))
 mexw64 win64:  $(patsubst %.mex, %.mexw64, $(PROGS))
-all:    octave win32 win64 mex4m
+all:    mex4o
 
 clean:
 	-$(RM) *.o *.obj *.o64 core octave-core *.oct *~ *.mex* 
