--- src/Makefile.orig	2014-01-12 13:51:33.000000000 -0600
+++ src/Makefile	2014-11-27 18:44:06.000000000 -0600
@@ -31,15 +31,21 @@
 # Instead of building "mex shortpath.c" and "mex uigetpath.c", I used empty m-functions within argout=argin; 
 ####################################################
 
-CC  	= gcc
-CXX 	= g++
+#CC  	= gcc
+#CXX 	= g++
 CFLAGS 	= -fopenmp -Wall -Wextra -Wconversion -O2 -fPIC 
-OCTMEX	= mkoctfile$(OCTAVE_VERSION) --mex
+#OCTMEX	= mkoctfile$(OCTAVE_VERSION) --mex
+MKOCTFILE = mkoctfile$(OCTAVE_VERSION)
+OCTMEX	= $(MKOCTFILE) --mex
 RM      = rm
 
 MEX_OPTION  += -lgomp CC\#$(CXX) CXX\#$(CXX) CFLAGS\#"$(CFLAGS) " CXXFLAGS\#"$(CFLAGS) "
 MATMEX      = $(MATLABDIR)/bin/mex $(MEX_OPTION)
 
+FLIBS = ${shell $(MKOCTFILE) -p FLIBS}
+INCFLAGS = ${shell $(MKOCTFILE) -p INCFLAGS}
+INC_CSTDLIB = -include cstdlib
+
 PROGS = histo_mex.mex covm_mex.mex kth_element.mex sumskipnan_mex.mex str2array.mex train.mex svmtrain_mex.mex svmpredict_mex.mex xptopen.mex 
 
 ### per default only the mex-files for octave are built
@@ -78,18 +84,16 @@
 #	Octave, MATLAB on Linux 
 #########################################################
 %.oct: %.cc
-	mkoctfile$(OCTAVE_VERSION) $<
+	INCFLAGS="$(INCFLAGS) $(INC_CSTDLIB)" $(MKOCTFILE) $<
 
 %.mex: %.cpp
-	$(OCTMEX) $<  -lgomp         ## Octave
-%.$(MEX_EXT): %.cpp
-	$(MATMEX) $<  -lgomp      ## Matlab
+	INCFLAGS="$(INCFLAGS) $(INC_CSTDLIB)" $(OCTMEX) $< $(FLIBS) -lgomp
 
 svmtrain_mex.mex svmpredict_mex.mex:     svmtrain_mex.cpp svm.h svm.cpp svm_model_matlab.c svmpredict_mex.cpp
 	$(CXX) $(CFLAGS) -c svm.cpp
-	#$(CC)  $(CFLAGS) $(INCOCTAVE) -c svm_model_matlab.c
-	env CC=$(CXX) $(OCTMEX) svmtrain_mex.cpp svm.o svm_model_matlab.c
-	env CC=$(CXX) $(OCTMEX) svmpredict_mex.cpp svm.o svm_model_matlab.c
+	$(CC)  $(CFLAGS) $(INCFLAGS) -c svm_model_matlab.c
+	INCFLAGS="$(INCFLAGS) $(INC_CSTDLIB)" $(OCTMEX) svmtrain_mex.cpp svm.o svm_model_matlab.o
+	INCFLAGS="$(INCFLAGS) $(INC_CSTDLIB)" $(OCTMEX) svmpredict_mex.cpp svm.o svm_model_matlab.o
 
 svmtrain_mex.$(MEX_EXT) svmpredict_mex.$(MEX_EXT):     svmtrain_mex.cpp svm.h svm.cpp svm_model_matlab.c svmpredict_mex.cpp
 	$(CXX) $(CFLAGS) -I $(MATLABDIR)/extern/include -c svm.cpp
@@ -98,12 +102,12 @@
 	$(MATMEX) svmpredict_mex.cpp svm.o svm_model_matlab.o
 
 train.$(MEX_EXT) predict.$(MEX_EXT): train.c linear.h tron.o linear.o linear_model_matlab.c
-	$(CC) $(CFLAGS) -I $(MATLABDIR)/extern/include -c linear_model_matlab.c
+	# $(CC) $(CFLAGS) -I $(MATLABDIR)/extern/include -c linear_model_matlab.c
 	$(MATMEX)  -lblas train.c tron.o linear.o linear_model_matlab.o 
 	#$(MATMEX) -lblas predict.c tron.o linear.o linear_model_matlab.o
 
 train.mex predict.mex: train.c linear.h tron.o linear.o linear_model_matlab.c
-	env CC=$(CXX) $(OCTMEX) -lblas train.c tron.o linear.o linear_model_matlab.c
+	$(OCTMEX) -lblas train.c tron.o linear.o linear_model_matlab.c
 	#env CC=$(CXX) $(OCTMEX) -lblas predict.c tron.o linear.o linear_model_matlab.o
 
 linear.o: linear.cpp linear.h
