--- common/network/TcpSocket.cxx.orig	2008-10-16 10:16:20.000000000 -0500
+++ common/network/TcpSocket.cxx	2013-12-18 17:50:54.000000000 -0600
@@ -24,6 +24,7 @@
 #else
 #define errorNumber errno
 #define closesocket close
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
--- unix/tx/TXImage.cxx.orig	2008-10-16 10:16:21.000000000 -0500
+++ unix/tx/TXImage.cxx	2013-12-18 17:52:06.000000000 -0600
@@ -20,6 +20,7 @@
 //
 
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <strings.h>
 #include <sys/types.h>
--- unix/x0vncserver/Image.cxx.orig	2008-10-16 10:16:21.000000000 -0500
+++ unix/x0vncserver/Image.cxx	2013-12-18 17:53:56.000000000 -0600
@@ -20,6 +20,7 @@
 //
 
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
