--- configure.old	2008-10-07 16:06:40.000000000 +0200
+++ configure	2008-10-07 16:07:08.000000000 +0200
@@ -6220,7 +6220,7 @@
 
 
 
-ac_config_files="$ac_config_files Makefile ext/protoFILE/Makefile ext/protoREGEX/Makefile contrib/inwt/Makefile contrib/inwt/inwt contrib/NativeCalls/Makefile contrib/NewtObjC/Makefile"
+ac_config_files="$ac_config_files Makefile ext/protoFILE/Makefile ext/protoREGEX/Makefile contrib/inwt/Makefile contrib/NativeCalls/Makefile contrib/NewtObjC/Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -6767,7 +6767,6 @@
     "ext/protoFILE/Makefile") CONFIG_FILES="$CONFIG_FILES ext/protoFILE/Makefile" ;;
     "ext/protoREGEX/Makefile") CONFIG_FILES="$CONFIG_FILES ext/protoREGEX/Makefile" ;;
     "contrib/inwt/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/inwt/Makefile" ;;
-    "contrib/inwt/inwt") CONFIG_FILES="$CONFIG_FILES contrib/inwt/inwt" ;;
     "contrib/NativeCalls/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/NativeCalls/Makefile" ;;
     "contrib/NewtObjC/Makefile") CONFIG_FILES="$CONFIG_FILES contrib/NewtObjC/Makefile" ;;
 
