--- SConstruct.orig	2013-06-02 05:41:11.000000000 -0700
+++ SConstruct	2014-06-15 06:55:46.000000000 -0700
@@ -170,11 +170,11 @@
 def CheckLuaLib(env, conf):
   if not 'USE_WIN32' in env['CPPDEFINES']:
      if env.WhereIs('pkg-config'):
-        for packagename in ['lua5.1', 'lua51', 'lua']:
+        for packagename in ['lua-5.1', 'lua51', 'lua']:
            exitcode,_ = ParseConfig(env, 'pkg-config --cflags --libs ' + packagename)
            if exitcode == 0:
               break
-  if conf.CheckLibWithHeader('lua51', 'lua.h', 'c'):
+  if conf.CheckLibWithHeader('lua-5.1', 'lua.h', 'c'):
     return 1
   if conf.CheckLibWithHeader('lua5.1', 'lua.h', 'c'):
     return 1
@@ -285,7 +285,7 @@
 
 debug = env.Clone()
 debug.Append(CPPDEFINES = 'DEBUG')
-debug.Append(CCFLAGS = Split('-g -Wsign-compare -Wall -Werror'))
+debug.Append(CCFLAGS = Split('-g -Wsign-compare -Wall'))
 
 profile = debug.Clone()
 profile.Append(CCFLAGS = Split('-pg'))
