--- src/formats.c.orig	2012-03-24 06:06:33.000000000 +0100
+++ src/formats.c	2012-03-24 06:48:56.000000000 +0100
@@ -348,7 +348,7 @@ static int sox_checkformat(sox_format_t 
   return SOX_SUCCESS;
 }
 
-static sox_bool is_url(char const * text) /* detects only wget-supported URLs */
+static sox_bool is_url(char const * text)
 {
   return !(
       strncasecmp(text, "http:" , (size_t)5) &&
@@ -385,7 +385,7 @@ static FILE * xfopen(char const * identi
   else if (is_url(identifier)) {
     FILE * f = NULL;
 #ifdef HAVE_POPEN
-    char const * const command_format = "wget --no-check-certificate -q -O- \"%s\"";
+    char const * const command_format = "ftp -a -V -o - \"%s\"";
     char * command = lsx_malloc(strlen(command_format) + strlen(identifier));
     sprintf(command, command_format, identifier);
     f = popen(command, POPEN_MODE);
