*** configure.orig	Mon Jun  4 23:15:35 2012
--- configure	Fri Jun  8 22:53:32 2012
***************
*** 17902,17908 ****
  if ac_fn_c_try_link "$LINENO"; then :
    ac_cv_lib_iconv_iconv_open=yes
  else
!   ac_cv_lib_iconv_iconv_open=no
  fi
  rm -f core conftest.err conftest.$ac_objext \
      conftest$ac_exeext conftest.$ac_ext
--- 17902,17908 ----
  if ac_fn_c_try_link "$LINENO"; then :
    ac_cv_lib_iconv_iconv_open=yes
  else
!   ac_cv_lib_iconv_iconv_open=yes
  fi
  rm -f core conftest.err conftest.$ac_objext \
      conftest$ac_exeext conftest.$ac_ext
***************
*** 18036,18043 ****
  fi
  
  if test -n "$GCC"; then
!    CFLAGS="-O3 -Wall "
!    CXXFLAGS="-O3 -Wall "
  fi
  
  
--- 18036,18043 ----
  fi
  
  if test -n "$GCC"; then
!    CFLAGS="$CFLAGS -Wall "
!    CXXFLAGS="$CXXFLAGS -Wall "
  fi
  
  
