--- Cakefile.orig	2011-08-04 22:17:23.000000000 -0500
+++ Cakefile	2011-08-05 06:29:40.000000000 -0500
@@ -45,17 +45,13 @@
   base = options.prefix or '/usr/local'
   lib  = "#{base}/lib/coffee-script"
   bin  = "#{base}/bin"
-  node = "~/.node_libraries/coffee-script"
   console.log   "Installing CoffeeScript to #{lib}"
-  console.log   "Linking to #{node}"
   console.log   "Linking 'coffee' to #{bin}/coffee"
   exec([
     "mkdir -p #{lib} #{bin}"
-    "cp -rf bin lib LICENSE README package.json src #{lib}"
-    "ln -sfn #{lib}/bin/coffee #{bin}/coffee"
-    "ln -sfn #{lib}/bin/cake #{bin}/cake"
-    "mkdir -p ~/.node_libraries"
-    "ln -sfn #{lib}/lib #{node}"
+    "cp -rf bin lib package.json src #{lib}"
+    "ln -sfn ../lib/coffee-script/bin/coffee #{bin}/coffee"
+    "ln -sfn ../lib/coffee-script/bin/cake #{bin}/cake"
   ].join(' && '), (err, stdout, stderr) ->
     if err then console.log stderr.trim() else log 'done', green
   )
