--- bin/cake.orig	2011-05-10 09:04:25.000000000 -0500
+++ bin/cake	2011-05-16 21:29:29.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!@PREFIX@/bin/node
 
 var path = require('path');
 var fs   = require('fs');
--- bin/coffee.orig	2011-05-10 09:04:25.000000000 -0500
+++ bin/coffee	2011-05-16 21:29:25.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!@PREFIX@/bin/node
 
 var path = require('path');
 var fs   = require('fs');
