diff -uNr ../openssh-4.5p1.orig/readconf.c ./readconf.c
--- ../openssh-4.5p1.orig/readconf.c	2006-12-05 21:05:28.000000000 -0800
+++ ./readconf.c	2006-12-05 21:10:59.000000000 -0800
@@ -1113,10 +1113,17 @@
 		options->pubkey_authentication = 1;
 	if (options->challenge_response_authentication == -1)
 		options->challenge_response_authentication = 1;
+#ifdef __APPLE_GSSAPI_ENABLE__
+	if (options->gss_authentication == -1)
+		options->gss_authentication = 1;
+	if (options->gss_keyex == -1)
+		options->gss_keyex = 1;
+#else
 	if (options->gss_authentication == -1)
 		options->gss_authentication = 0;
 	if (options->gss_keyex == -1)
 		options->gss_keyex = 0;
+#endif
 	if (options->gss_deleg_creds == -1)
 		options->gss_deleg_creds = 0;
 	if (options->gss_trust_dns == -1)
diff -uNr ../openssh-4.5p1.orig/servconf.c ./servconf.c
--- ../openssh-4.5p1.orig/servconf.c	2006-12-05 21:05:28.000000000 -0800
+++ ./servconf.c	2006-12-05 21:08:44.000000000 -0800
@@ -204,10 +204,17 @@
 		options->kerberos_ticket_cleanup = 1;
 	if (options->kerberos_get_afs_token == -1)
 		options->kerberos_get_afs_token = 0;
+#ifdef __APPLE_GSSAPI_ENABLE__
+	if (options->gss_authentication == -1)
+		options->gss_authentication = 1;
+	if (options->gss_keyex == -1)
+		options->gss_keyex = 1;
+#else
 	if (options->gss_authentication == -1)
 		options->gss_authentication = 0;
 	if (options->gss_keyex == -1)
 		options->gss_keyex = 0;
+#endif
 	if (options->gss_cleanup_creds == -1)
 		options->gss_cleanup_creds = 1;
 	if (options->gss_strict_acceptor == -1)
