--- a/sandbox-darwin.c.orig	2014-02-12 01:31:10.000000000 +0100
+++ b/sandbox-darwin.c	2014-02-12 01:31:54.000000000 +0100
@@ -62,8 +62,16 @@
 	struct rlimit rl_zero;
 
 	debug3("%s: starting Darwin sandbox", __func__);
+#ifdef __APPLE_SANDBOX_NAMED_EXTERNAL__
+#ifndef SANDBOX_NAMED_EXTERNAL
+#define SANDBOX_NAMED_EXTERNAL (0x3)
+#endif
+	if (sandbox_init("@PREFIX@/share/openssh/org.openssh.sshd.sb",
+		SANDBOX_NAMED_EXTERNAL, &errmsg) == -1)
+#else
 	if (sandbox_init(kSBXProfilePureComputation, SANDBOX_NAMED,
 	    &errmsg) == -1)
+#endif
 		fatal("%s: sandbox_init: %s", __func__, errmsg);
 
 	/*
