--- yenc.c	Sat Feb 22 10:32:58 2003
+++ yenc.c.new	Sun May  9 13:46:00 2004
@@ -1,5 +1,8 @@
 #include "yenc.h"
 
+char err_msg[256];
+int my_err_code;
+
 static int crc_tab[256] = {
 0x00000000, 0x77073096, 0xee0e612c, 0x990951ba, 0x076dc419, 0x706af48f, 0xe963a535, 0x9e6495a3,
 0x0edb8832, 0x79dcb8a4, 0xe0d5e91e, 0x97d2d988, 0x09b64c2b, 0x7eb17cbd, 0xe7b82d07, 0x90bf1d91,
@@ -40,18 +43,18 @@
 
 void throw_exception(int code, char *msg){
 	strncpy(err_msg,msg,256);
-	err_code = code;
+	my_err_code = code;
 }
 
 char *check_exception(){
-	if(err_code)
+	if(my_err_code)
 		return err_msg;
 	else
 		return NULL;
 }
 
 void clear_exception(){
-	err_code = 0;
+	my_err_code = 0;
 }
 
 void crc_init(){
