--- a/libr/Makefile	2011-12-16 06:44:05.000000000 -0800
+++ b/libr/Makefile	2011-12-16 12:44:12.000000000 -0800
@@ -54,7 +54,7 @@
 	done
 	for a in ${LIBLIST} ; do \
 		ln -fs ${PWD}/$${a}/libr_$${a}.${EXT_SO} ${LFX}/libr_$${a}.${EXT_SO} ; \
-		ln -fs ${PWD}/$${a}/libr_$${a}.${EXT_SO} ${LFX}/libr_$${a}.${EXT_SO}.${LIBVERSION} ; \
+		ln -fs ${PWD}/$${a}/libr_$${a}.${EXT_SO} ${LFX}/libr_$${a}.${LIBVERSION}.${EXT_SO} ; \
 		ln -fs ${PWD}/$${a}/libr_$${a}.${EXT_AR} ${LFX}/libr_$${a}.${EXT_AR} ; \
 		for b in $${a}/p/*.${EXT_SO} ; do \
 			if [ -e ${PWD}/$${b} ] ; then \
@@ -68,12 +68,12 @@
 	# libraries
 	@${INSTALL_DIR} ${LFX}
 	@for a in `find * -type f | grep -e '\.${EXT_SO}$$' | grep -v lib/t | grep lib | grep -v /p/` ; do \
-	  b=`echo $$a|cut -d / -f 2`; \
-	  echo " ${LFX}/$$b.${LIBVERSION}"; \
-	  ${INSTALL_LIB} $$a ${LFX}/$$b.${LIBVERSION} ; \
-	  ( cd ${LFX} ; ln -fs $$b.${LIBVERSION} $$b ) ; \
+	  b=`echo $$a|cut -d / -f 2|sed -e 's,\.${EXT_SO},,g'`; \
+	  echo " ${LFX}/$$b.${LIBVERSION}.${EXT_SO}"; \
+	  ${INSTALL_LIB} $$a ${LFX}/$$b.${LIBVERSION}.${EXT_SO} ; \
+	  ( cd ${LFX} ; ln -fs $$b.${LIBVERSION}.${EXT_SO} $$b.${EXT_SO} ) ; \
 	done
-	#Linux only? ( cd ${LFX} ; ln -fs $$b.${LIBVERSION} $$b.0 ; ln -fs $$b.0 $$b )
+	#Linux only? ( cd ${LFX} ; ln -fs $$b.${LIBVERSION}.${EXT_SO} $$b.0.${EXT_SO} ; ln -fs $$b.0.${EXT_SO} $$b.${EXT_SO} )
 	# object archives
 	@for a in `find * -type f | grep -e '\.a$$' | grep -v fs/p` ; do \
 	  echo " ${LFX}/$$a"; ${INSTALL_DATA} $$a ${LFX} ; done
