--- Makefile.orig	2010-05-29 22:23:53.000000000 +0200
+++ Makefile	2010-05-29 22:26:16.000000000 +0200
@@ -21,13 +21,14 @@
 
 # directory for read-only game data (like graphics, sounds, levels)
 # default is '.' to be able to run program without installation
-# RO_GAME_DIR = /usr/games
+RO_GAME_DIR = _PREFIX_/share/games/rocksndiamonds
 # use the following setting for Debian / Ubuntu installations:
 # RO_GAME_DIR = /usr/share/games/rocksndiamonds
 
 # directory for writable game data (like highscore files)
 # default is '.' to be able to run program without installation
 # RW_GAME_DIR = /var/games
+RW_GAME_DIR = _PREFIX_/var/games
 # use the following setting for Debian / Ubuntu installations:
 # RW_GAME_DIR = /var/games/rocksndiamonds
 
@@ -37,7 +38,7 @@
 # choose if more than one global score file entry for one player is allowed
 # (default: MANY_PER_NAME)
 # uncomment to install game in multi-user environment
-# SCORE_ENTRIES = ONE_PER_NAME
+SCORE_ENTRIES = ONE_PER_NAME
 # uncomment to install game in single-user environment (default)
 # SCORE_ENTRIES = MANY_PER_NAME
 
