--- setup.py.orig	2018-03-29 22:57:55.000000000 +1100
+++ setup.py	2018-04-27 05:55:22.000000000 +1000
@@@ -24,7 +24,7 @@ from distutils.spawn import find_executable
 TEST_EXTENSIONS = True

 # This global variable is used to hold the list of modules to be disabled.
-DISABLED_MODULE_LIST = []
+DISABLED_MODULE_LIST = ["_tkinter", "_gdbm"]


 def get_platform():
@@ -1840,8 +1840,8 @@ class PyBuildExt(build_ext):
         # Rather than complicate the code below, detecting and building
         # AquaTk is a separate method. Only one Tkinter will be built on
         # Darwin - either AquaTk, if it is found, or X11 based Tk.
-        if (MACOS and self.detect_tkinter_darwin()):
-            return True
+        self.announce("INFO: Not using Tcl/Tk", 2)
+        return

         # Assume we haven't found any of the libraries or include files
         # The versions with dots are used on Unix, and the versions without
