--- Mac/Tools/pythonw.c	2020-07-13 05:11:53.000000000 -0700
+++ Mac/Tools/pythonw.c	2020-07-18 14:22:39.000000000 -0700
@@ -121,6 +121,8 @@
     cpu_types[0] = CPU_TYPE_POWERPC;
 #elif defined(__i386__)
     cpu_types[0] = CPU_TYPE_X86;
+#elif defined(__arm64__)
+    cpu_types[0] = CPU_TYPE_ARM64;
 #else
 #       error "Unknown CPU"
 #endif
--- Modules/_decimal/libmpdec/mpdecimal.h	2020-07-13 05:11:53.000000000 -0700
+++ Modules/_decimal/libmpdec/mpdecimal.h	2020-07-18 14:24:04.000000000 -0700
@@ -135,6 +135,9 @@
   #elif defined(__x86_64__)
     #define CONFIG_64
     #define ASM
+  #elif defined(__arm64__)
+    #define CONFIG_64
+    #define ANSI
   #else
     #error "unknown architecture for universal build."
   #endif
--- configure	2020-07-13 05:11:53.000000000 -0700
+++ configure	2020-07-18 14:21:25.000000000 -0700
@@ -9260,6 +9260,9 @@
     	esac
     else
     	case `/usr/bin/arch` in
+    	arm64)
+    		MACOSX_DEFAULT_ARCH="arm64"
+    		;;
     	i386)
     		MACOSX_DEFAULT_ARCH="x86_64"
     		;;
