https://bugs.php.net/61375
--- php_stomp.c.orig	2010-10-12 22:28:38.000000000 -0500
+++ php_stomp.c	2012-05-22 19:48:14.000000000 -0500
@@ -327,7 +327,7 @@
 }
 
 	 
-#if (PHP_MAJOR_VERSION == 5 && PHP_MINOR_VERSION == 3 && PHP_RELEASE_VERSION >= 99) || (PHP_MAJOR_VERSION > 5)
+#if (PHP_MAJOR_VERSION == 5 && (PHP_MINOR_VERSION == 3 && PHP_RELEASE_VERSION >= 99) || PHP_MINOR_VERSION >= 4) || (PHP_MAJOR_VERSION > 5)
 #define PHP_STOMP_RUNTIME_CACHE 
 #endif 
 static zend_object_value php_stomp_new(zend_class_entry *ce TSRMLS_DC)
