--- avahi-common/Makefile.in	2007-12-17 09:37:42.000000000 -0600
+++ avahi-common/Makefile.in	2008-11-10 03:45:49.000000000 -0600
@@ -555,7 +555,7 @@
 	i18n.c i18n.h
 
 libavahi_common_la_CFLAGS = $(AM_CFLAGS) $(PTHREAD_CFLAGS) -DAVAHI_LOCALEDIR=\"$(avahilocaledir)\"
-libavahi_common_la_LIBADD = $(AM_LDADD) $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
+libavahi_common_la_LIBADD = $(AM_LDADD) $(PTHREAD_CFLAGS) $(PTHREAD_LIBS) $(LIBINTL)
 libavahi_common_la_LDFLAGS = $(AM_LDFLAGS) -export-dynamic -version-info $(LIBAVAHI_COMMON_VERSION_INFO)
 strlst_test_SOURCES = \
 	strlst.c strlst.h \
