--- a/initscript/darwin/org.freedesktop.avahi-daemon.plist.in	2012-02-15 10:30:25.000000000 -0800
+++ b/initscript/darwin/org.freedesktop.avahi-daemon.plist.in	2012-02-15 10:33:40.000000000 -0800
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>Disabled</key>
+	<true/>
 	<key>Label</key>
 	<string>org.freedesktop.avahi-daemon</string>
 	<key>OnDemand</key>
