--- initscript/darwin/org.freedesktop.avahi-daemon.plist.in.orig	2015-04-01 00:58:00.000000000 -0400
+++ initscript/darwin/org.freedesktop.avahi-daemon.plist.in	2023-02-13 13:09:30.000000000 -0500
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>Disabled</key>
+	<true/>
 	<key>Label</key>
 	<string>org.freedesktop.avahi-daemon</string>
 	<key>OnDemand</key>
@@ -9,6 +11,7 @@
 	<key>ProgramArguments</key>
 	<array>
 			<string>@sbindir@/avahi-daemon</string>
+			<string>--no-drop-root</string>
 	</array>
 	<key>ServiceIPC</key>
 	<false/>
