--- a/avahi-utils/Makefile.in	2012-02-15 10:30:25.000000000 -0800
+++ b/avahi-utils/Makefile.in	2012-02-15 10:33:39.000000000 -0800
@@ -429,16 +429,16 @@
 @HAVE_DBUS_TRUE@avahi_browse_LDADD = $(AM_LDADD) \
 @HAVE_DBUS_TRUE@	../avahi-client/libavahi-client.la \
 @HAVE_DBUS_TRUE@	../avahi-common/libavahi-common.la \
-@HAVE_DBUS_TRUE@	$(am__append_3)
+@HAVE_DBUS_TRUE@	$(am__append_3) $(INTLLIBS)
 @HAVE_DBUS_TRUE@avahi_resolve_SOURCES = avahi-resolve.c sigint.c sigint.h
 @HAVE_DBUS_TRUE@avahi_resolve_CFLAGS = $(AM_CFLAGS)
-@HAVE_DBUS_TRUE@avahi_resolve_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la
+@HAVE_DBUS_TRUE@avahi_resolve_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la $(INTLLIBS)
 @HAVE_DBUS_TRUE@avahi_publish_SOURCES = avahi-publish.c sigint.c sigint.h
 @HAVE_DBUS_TRUE@avahi_publish_CFLAGS = $(AM_CFLAGS)
-@HAVE_DBUS_TRUE@avahi_publish_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la
+@HAVE_DBUS_TRUE@avahi_publish_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la $(INTLLIBS)
 @HAVE_DBUS_TRUE@avahi_set_host_name_SOURCES = avahi-set-host-name.c sigint.c sigint.h
 @HAVE_DBUS_TRUE@avahi_set_host_name_CFLAGS = $(AM_CFLAGS)
-@HAVE_DBUS_TRUE@avahi_set_host_name_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la
+@HAVE_DBUS_TRUE@avahi_set_host_name_LDADD = $(AM_LDADD) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la $(INTLLIBS)
 all: all-am
 
 .SUFFIXES:
