--- avahi-dnsconfd/Makefile.in.orig	2010-04-11 17:00:59.000000000 +0200
+++ avahi-dnsconfd/Makefile.in	2010-04-11 17:01:15.000000000 +0200
@@ -344,7 +344,7 @@
 @HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@	$$3")'
 @HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@avahi_dnsconfd_SOURCES = main.c 
 @HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@avahi_dnsconfd_CFLAGS = $(AM_CFLAGS) $(LIBDAEMON_CFLAGS)
-@HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@avahi_dnsconfd_LDADD = $(AM_LDADD) ../avahi-common/libavahi-common.la $(LIBDAEMON_LIBS)
+@HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@avahi_dnsconfd_LDADD = $(AM_LDADD) ../avahi-common/libavahi-common.la $(LIBDAEMON_LIBS) $(INTLLIBS)
 @HAVE_LIBDAEMON_TRUE@@HAVE_XML_TRUE@pkgsysconf_SCRIPTS = avahi-dnsconfd.action
 EXTRA_DIST = avahi-dnsconfd.action
 all: all-am
