--- src/odbc_wrap.cpp
+++ src/odbc_wrap.cpp
@@ -6461,7 +6461,7 @@
   if (!SWIG_IsOK(res7)) {
     SWIG_exception_fail(SWIG_ArgError(res7), "in method '" "SQLColAttribute" "', argument " "7"" of type '" "SQLPOINTER""'"); 
   }
-  result = (SQLRETURN)SQLColAttribute(arg1,arg2,arg3,arg4,arg5,arg6,arg7);
+  result = (SQLRETURN)SQLColAttribute(arg1,arg2,arg3,arg4,arg5,arg6,(SQLINTEGER*)arg7);
   _outv = SWIG_From_short((short)(result));
   if (_outv.is_defined()) _outp = SWIG_Octave_AppendOutput(_outp, _outv);
 fail:
