diff -ur build.orig/rules/compile build/rules/compile
--- build.orig/rules/compile	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/compile	2008-05-15 12:58:19.000000000 +0200
@@ -32,49 +32,34 @@
 #
 # Rules for compiling
 #	
-$(OBJPATH_DEBUG_STATIC)/%.o: $(SRCDIR)/%.cpp $(DEPPATH)/%.d
+$(OBJPATH_DEBUG_STATIC)/%.o: $(SRCDIR)/%.cpp
 	@echo "** Compiling" $< "(debug)"
 	$(CXX) $(INCLUDE) $(CXXFLAGS) $(DEBUGOPT_CXX) $(STATICOPT_CXX) -c $< -o $@
 
-$(OBJPATH_RELEASE_STATIC)/%.o: $(SRCDIR)/%.cpp $(DEPPATH)/%.d
+$(OBJPATH_RELEASE_STATIC)/%.o: $(SRCDIR)/%.cpp
 	@echo "** Compiling" $< "(release)"
 	$(CXX) $(INCLUDE) $(CXXFLAGS) $(RELEASEOPT_CXX) $(STATICOPT_CXX) -c $< -o $@
 
-$(OBJPATH_DEBUG_STATIC)/%.o: $(SRCDIR)/%.c $(DEPPATH)/%.d
+$(OBJPATH_DEBUG_STATIC)/%.o: $(SRCDIR)/%.c
 	@echo "** Compiling" $< "(debug)"
 	$(CC) $(INCLUDE) $(CFLAGS) $(DEBUGOPT_CC) $(STATICOPT_CC) -c $< -o $@
 
-$(OBJPATH_RELEASE_STATIC)/%.o: $(SRCDIR)/%.c $(DEPPATH)/%.d
+$(OBJPATH_RELEASE_STATIC)/%.o: $(SRCDIR)/%.c
 	@echo "** Compiling" $< "(release)"
 	$(CC) $(INCLUDE) $(CFLAGS) $(RELEASEOPT_CC) $(STATICOPT_CC) -c $< -o $@
 
-$(OBJPATH_DEBUG_SHARED)/%.o: $(SRCDIR)/%.cpp $(DEPPATH)/%.d
+$(OBJPATH_DEBUG_SHARED)/%.o: $(SRCDIR)/%.cpp
 	@echo "** Compiling" $< "(debug, shared)"
 	$(CXX) $(INCLUDE) $(CXXFLAGS) $(DEBUGOPT_CXX) $(SHAREDOPT_CXX) -c $< -o $@
 
-$(OBJPATH_RELEASE_SHARED)/%.o: $(SRCDIR)/%.cpp $(DEPPATH)/%.d
+$(OBJPATH_RELEASE_SHARED)/%.o: $(SRCDIR)/%.cpp
 	@echo "** Compiling" $< "(release, shared)"
 	$(CXX) $(INCLUDE) $(CXXFLAGS) $(RELEASEOPT_CXX) $(SHAREDOPT_CXX) -c $< -o $@
 
-$(OBJPATH_DEBUG_SHARED)/%.o: $(SRCDIR)/%.c $(DEPPATH)/%.d
+$(OBJPATH_DEBUG_SHARED)/%.o: $(SRCDIR)/%.c
 	@echo "** Compiling" $< "(debug, shared)"
 	$(CC) $(INCLUDE) $(CFLAGS) $(DEBUGOPT_CC) $(SHAREDOPT_CC) -c $< -o $@
 
-$(OBJPATH_RELEASE_SHARED)/%.o: $(SRCDIR)/%.c $(DEPPATH)/%.d
+$(OBJPATH_RELEASE_SHARED)/%.o: $(SRCDIR)/%.c
 	@echo "** Compiling" $< "(release, shared)"
 	$(CC) $(INCLUDE) $(CFLAGS) $(RELEASEOPT_CC) $(SHAREDOPT_CC) -c $< -o $@
-
-#
-# Rules for creating dependency information
-#
-$(DEPPATH)/%.d: $(SRCDIR)/%.cpp
-	@echo "** Creating dependency info for" $^
-	$(MKDIR) $(DEPPATH)
-	$(DEP) $(SRCDIR)/$(patsubst %.d,%.cpp,$(notdir $@)) $@ $(OBJPATH_DEBUG_STATIC) $(OBJPATH_RELEASE_STATIC) $(OBJPATH_DEBUG_SHARED) $(OBJPATH_RELEASE_SHARED) $(INCLUDE) $(CXXFLAGS) 
-
-$(DEPPATH)/%.d: $(SRCDIR)/%.c
-	@echo "** Creating dependency info for" $^
-	$(MKDIR) $(DEPPATH)
-	$(DEP) $(SRCDIR)/$(patsubst %.d,%.c,$(notdir $@)) $@ $(OBJPATH_DEBUG_STATIC) $(OBJPATH_RELEASE_STATIC) $(OBJPATH_DEBUG_SHARED) $(OBJPATH_RELEASE_SHARED) $(INCLUDE) $(CFLAGS)
-
-depend: $(addprefix $(DEPPATH)/,$(addsuffix .d,$(objects)))
diff -ur build.orig/rules/dylib build/rules/dylib
--- build.orig/rules/dylib	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/dylib	2008-05-15 12:58:40.000000000 +0200
@@ -70,8 +70,3 @@
 	@echo "** Building dynamic library (release)" $@
 	$(DYLIB) $^ $(LIBRARY) $(TARGET_LIBS_RELEASE) $(SYSLIBS)
 	$(postbuild)
-
-#
-# Include the automatically generated dependency files
-#
-sinclude $(addprefix $(DEPPATH)/,$(addsuffix .d,$(objects)))
diff -ur build.orig/rules/exec build/rules/exec
--- build.orig/rules/exec	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/exec	2008-05-15 12:58:52.000000000 +0200
@@ -57,8 +57,3 @@
 	$(LINK) $(LINKFLAGS) $(RELEASEOPT_LINK) $(SHAREDOPT_LINK) -o $@ $^ $(LIBRARY) $(TARGET_LIBS_RELEASE) $(SYSLIBS)
 	$(STRIPCMD)
 	$(postbuild)
-
-#
-# Include the automatically generated dependency files
-#
-sinclude $(addprefix $(DEPPATH)/,$(addsuffix .d,$(objects)))
diff -ur build.orig/rules/global build/rules/global
--- build.orig/rules/global	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/global	2008-05-15 12:59:10.000000000 +0200
@@ -104,11 +104,9 @@
 LIBDIR   = lib/$(OSNAME)/$(OSARCH)
 BINDIR   = bin/$(OSNAME)/$(OSARCH)
 OBJDIR   = obj/$(OSNAME)/$(OSARCH)
-DEPDIR   = .dep/$(OSNAME)/$(OSARCH)
 LIBPATH  = $(POCO_BUILD)/$(LIBDIR)
 BINPATH  = $(POCO_BUILD)/$(COMPONENT)/$(BINDIR)
 OBJPATH  = $(POCO_BUILD)/$(COMPONENT)/$(OBJDIR)
-DEPPATH  = $(POCO_BUILD)/$(COMPONENT)/$(DEPDIR)
 
 ifeq ($(POCO_BASE),$(PROJECT_BASE))
 POCO_LIBRARY =
diff -ur build.orig/rules/lib build/rules/lib
--- build.orig/rules/lib	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/lib	2008-05-15 12:59:18.000000000 +0200
@@ -70,8 +70,3 @@
 	$(SHLIBLN) $(LIB_RELEASE_SHARED) $(LIB_RELEASE_SHARED_LINK)
 	$(STRIPCMD)
 	$(postbuild)
-
-#
-# Include the automatically generated dependency files
-#
-sinclude $(addprefix $(DEPPATH)/,$(addsuffix .d,$(objects)))
diff -ur build.orig/rules/sample build/rules/sample
--- build.orig/rules/sample	2008-05-15 12:56:50.000000000 +0200
+++ build/rules/sample	2008-05-15 12:59:34.000000000 +0200
@@ -43,12 +43,10 @@
 LIBDIR   = lib
 BINDIR   = bin
 OBJDIR   = obj
-DEPDIR   = .dep
 INCPATH  = $(POCO_BASE)/$(INCDIR)
 LIBPATH  = $(POCO_BASE)/$(LIBDIR)
 BINPATH  = $(BINDIR)
 OBJPATH  = $(OBJDIR)
-DEPPATH  = $(DEPDIR)
 
 #
 # Determine link mode
