--- project.clj.orig   2023-04-16 23:50:00.000000000 -8000
+++ project.clj        2023-04-16 23:50:00.000000000 -8000
@@ -12,6 +12,7 @@
                  "babashka.nrepl/src" "depstar/src" "process/src"
                  "deps.clj/src" "deps.clj/resources"
                  "impl-java/src"]
+  :local-repo "local-m2"
   ;; for debugging Reflector.java code:
   ;; :java-source-paths ["sci/reflector/src-java"]
   :java-source-paths ["src-java"]
