diff -Naur macosx.orig/da.lproj/MainMenu.xib macosx/da.lproj/MainMenu.xib
--- macosx.orig/da.lproj/MainMenu.xib	2014-07-01 13:09:56.000000000 -0400
+++ macosx/da.lproj/MainMenu.xib	2014-07-02 09:53:04.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Søg efter opdateringer..." tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Donér..." tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/de.lproj/MainMenu.xib macosx/de.lproj/MainMenu.xib
--- macosx.orig/de.lproj/MainMenu.xib	2014-07-01 13:10:01.000000000 -0400
+++ macosx/de.lproj/MainMenu.xib	2014-07-02 09:53:13.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Nach Aktualisierung suchen …" tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Spenden …" tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/en.lproj/MainMenu.xib macosx/en.lproj/MainMenu.xib
--- macosx.orig/en.lproj/MainMenu.xib	2014-07-01 13:09:36.000000000 -0400
+++ macosx/en.lproj/MainMenu.xib	2014-07-02 09:51:05.000000000 -0400
@@ -3629,14 +3629,6 @@
 					<string key="id">3075</string>
 				</object>
 				<object class="IBConnectionRecord">
-					<object class="IBActionConnection" key="connection">
-						<string key="label">checkForUpdates:</string>
-						<reference key="source" ref="363870251"/>
-						<reference key="destination" ref="925446525"/>
-					</object>
-					<string key="id">1814</string>
-				</object>
-				<object class="IBConnectionRecord">
 					<object class="IBOutletConnection" key="connection">
 						<string key="label">delegate</string>
 						<reference key="source" ref="823598060"/>
diff -Naur macosx.orig/es.lproj/MainMenu.xib macosx/es.lproj/MainMenu.xib
--- macosx.orig/es.lproj/MainMenu.xib	2014-07-01 13:10:34.000000000 -0400
+++ macosx/es.lproj/MainMenu.xib	2014-07-02 09:51:19.000000000 -0400
@@ -3625,14 +3625,6 @@
 					<string key="id">3075</string>
 				</object>
 				<object class="IBConnectionRecord">
-					<object class="IBActionConnection" key="connection">
-						<string key="label">checkForUpdates:</string>
-						<reference key="source" ref="363870251"/>
-						<reference key="destination" ref="925446525"/>
-					</object>
-					<string key="id">1814</string>
-				</object>
-				<object class="IBConnectionRecord">
 					<object class="IBOutletConnection" key="connection">
 						<string key="label">delegate</string>
 						<reference key="source" ref="823598060"/>
diff -Naur macosx.orig/fr.lproj/MainMenu.xib macosx/fr.lproj/MainMenu.xib
--- macosx.orig/fr.lproj/MainMenu.xib	2014-07-01 13:09:30.000000000 -0400
+++ macosx/fr.lproj/MainMenu.xib	2014-07-02 09:51:40.000000000 -0400
@@ -3637,14 +3637,6 @@
 					<string key="id">3075</string>
 				</object>
 				<object class="IBConnectionRecord">
-					<object class="IBActionConnection" key="connection">
-						<string key="label">checkForUpdates:</string>
-						<reference key="source" ref="363870251"/>
-						<reference key="destination" ref="925446525"/>
-					</object>
-					<string key="id">1814</string>
-				</object>
-				<object class="IBConnectionRecord">
 					<object class="IBOutletConnection" key="connection">
 						<string key="label">delegate</string>
 						<reference key="source" ref="823598060"/>
diff -Naur macosx.orig/it.lproj/MainMenu.xib macosx/it.lproj/MainMenu.xib
--- macosx.orig/it.lproj/MainMenu.xib	2014-07-01 13:09:39.000000000 -0400
+++ macosx/it.lproj/MainMenu.xib	2014-07-02 09:52:00.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Ricerca aggiornamenti..." tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Dona..." tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/nl.lproj/MainMenu.xib macosx/nl.lproj/MainMenu.xib
--- macosx.orig/nl.lproj/MainMenu.xib	2014-07-01 13:10:16.000000000 -0400
+++ macosx/nl.lproj/MainMenu.xib	2014-07-02 09:52:13.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Zoek naar nieuwe versie..." tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Doneer..." tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/pt_PT.lproj/MainMenu.xib macosx/pt_PT.lproj/MainMenu.xib
--- macosx.orig/pt_PT.lproj/MainMenu.xib	2014-07-01 13:10:23.000000000 -0400
+++ macosx/pt_PT.lproj/MainMenu.xib	2014-07-02 09:52:27.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Procurar actualizações…" tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Doar…" tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/ru.lproj/MainMenu.xib macosx/ru.lproj/MainMenu.xib
--- macosx.orig/ru.lproj/MainMenu.xib	2014-07-01 13:10:28.000000000 -0400
+++ macosx/ru.lproj/MainMenu.xib	2014-07-02 09:52:38.000000000 -0400
@@ -189,12 +189,6 @@
                                     <action selector="showAboutWindow:" target="206" id="1927"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Обновления..." tag="1234" id="975">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="checkForUpdates:" target="1812" id="1814"/>
-                                </connections>
-                            </menuItem>
                             <menuItem title="Сделать пожертвование…" tag="5678" id="3212">
                                 <connections>
                                     <action selector="linkDonate:" target="206" id="3213"/>
diff -Naur macosx.orig/tr.lproj/MainMenu.xib macosx/tr.lproj/MainMenu.xib
--- macosx.orig/tr.lproj/MainMenu.xib	2014-07-01 13:09:51.000000000 -0400
+++ macosx/tr.lproj/MainMenu.xib	2014-07-02 09:52:53.000000000 -0400
@@ -3694,14 +3694,6 @@
 					<string key="id">3075</string>
 				</object>
 				<object class="IBConnectionRecord">
-					<object class="IBActionConnection" key="connection">
-						<string key="label">checkForUpdates:</string>
-						<reference key="source" ref="363870251"/>
-						<reference key="destination" ref="925446525"/>
-					</object>
-					<string key="id">1814</string>
-				</object>
-				<object class="IBConnectionRecord">
 					<object class="IBOutletConnection" key="connection">
 						<string key="label">delegate</string>
 						<reference key="source" ref="823598060"/>
