diff --git CMakeLists.txt CMakeLists.txt
index 6a26f19..9011052 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -40,7 +40,6 @@ set(CMAKE_AUTOUIC ON)
 find_package(lxqt-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt5 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets DBus LinguistTools)
 find_package(Qt5Xdg ${QTXDG_MINIMUM_VERSION} REQUIRED)
-find_package(KF5WindowSystem ${KF5_MINIMUM_VERSION} REQUIRED)
 if (BUILD_BACKLIGHT_LINUX_BACKEND)
     find_package(PolkitQt5-1 REQUIRED)
 endif (BUILD_BACKLIGHT_LINUX_BACKEND)
@@ -305,7 +304,6 @@ add_library(${LXQT_LIBRARY_NAME}
 
 target_link_libraries(${LXQT_LIBRARY_NAME}
     PUBLIC
-        KF5::WindowSystem
         Qt5::Widgets
         Qt5::DBus
         Qt5Xdg
diff --git lxqtsingleapplication.cpp lxqtsingleapplication.cpp
index 508ec3d..25bbf8b 100644
--- lxqtsingleapplication.cpp
+++ lxqtsingleapplication.cpp
@@ -27,7 +27,6 @@
 
 #include "lxqtsingleapplication.h"
 #include "singleapplicationadaptor.h"
-#include <KWindowSystem/KWindowSystem>
 #include <QDBusMessage>
 #include <QWidget>
 #include <QDebug>
@@ -96,14 +95,6 @@ void SingleApplication::activateWindow()
 {
     if (mActivationWindow) {
         mActivationWindow->show();
-        WId window = mActivationWindow->effectiveWinId();
-
-        KWindowInfo info(window, KWindowSystem::WMDesktop);
-        int windowDesktop = info.desktop();
-
-        if (windowDesktop != KWindowSystem::currentDesktop())
-            KWindowSystem::setCurrentDesktop(windowDesktop);
-        KWindowSystem::activateWindow(window);
     } else {
         qDebug() << Q_FUNC_INFO << "activationWindow not set or null";
     }
diff --git cmake/lxqt-config.cmake.in cmake/lxqt-config.cmake.in
index 25eaf61..1ff35d4 100644
--- cmake/lxqt-config.cmake.in
+++ cmake/lxqt-config.cmake.in
@@ -12,7 +12,6 @@ find_dependency(Qt5Widgets @QT_MINIMUM_VERSION@)
 find_dependency(Qt5DBus @QT_MINIMUM_VERSION@)
 find_dependency(Qt5LinguistTools @QT_MINIMUM_VERSION@)
 find_dependency(Qt5Xdg @QTXDG_MINIMUM_VERSION@)
-find_dependency(KF5WindowSystem)
 find_dependency(lxqt-build-tools @LXQTBT_MINIMUM_VERSION@)
 if (NOT APPLE)
     find_dependency(Qt5X11Extras @QT_MINIMUM_VERSION@)
