--- package/bin/osmosis.orig	2021-08-27 15:57:04.000000000 +0100
+++ package/bin/osmosis	2021-08-27 15:58:59.000000000 +0100
@@ -80,10 +80,7 @@
 fi
 
 # make it fully qualified
-saveddir=`pwd`
-MYAPP_HOME=`dirname "$PRG"`/..
-MYAPP_HOME=`cd "$MYAPP_HOME" && pwd`
-cd "$saveddir"
+MYAPP_HOME="@PREFIX@/share/osmosis"
 
 # Build up the classpath of required jar files via classworlds launcher.
 MYAPP_CLASSPATH=$MYAPP_HOME/lib/default/plexus-classworlds-*.jar
