diff -pur mkvtoolnix-6.3.0.orig/src/common/command_line.cpp mkvtoolnix-6.3.0/src/common/command_line.cpp
--- mkvtoolnix-6.3.0.orig/src/common/command_line.cpp	2013-08-24 18:35:22.000000000 -0400
+++ mkvtoolnix-6.3.0/src/common/command_line.cpp	2013-08-24 18:36:30.000000000 -0400
@@ -285,6 +285,7 @@ handle_common_cli_args(std::vector<std::
     } else if ((args[i] == "-h") || (args[i] == "-?") || (args[i] == "--help"))
       usage();
 
+#if 0
 #if defined(HAVE_CURL_EASY_H)
     else if (args[i] == "--check-for-updates") {
       mtx_release_version_t rel = get_latest_release_version();
@@ -304,6 +305,7 @@ handle_common_cli_args(std::vector<std::
       mxexit(rel.current_version < rel.latest_source ? 1 : 0);
     }
 #endif  // defined(HAVE_CURL_EASY_H)
+#endif
 
     else
       ++i;
diff -pur mkvtoolnix-6.3.0.orig/src/common/version.cpp mkvtoolnix-6.3.0/src/common/version.cpp
--- mkvtoolnix-6.3.0.orig/src/common/version.cpp	2013-08-24 18:35:22.000000000 -0400
+++ mkvtoolnix-6.3.0/src/common/version.cpp	2013-08-24 18:35:43.000000000 -0400
@@ -220,7 +220,7 @@ get_latest_release_version() {
     std::stringstream urls;
     brng::for_each(release.urls, [&](std::pair<std::string, std::string> const &kv) { urls << " " << kv.first << ":" << kv.second; });
     mxdebug(boost::format("update check: current %1% latest source %2% latest winpre %3% URLs%4%\n")
-            % release.current_version.to_string() % release.latest_source.to_string() % release.latest_windows_build.to_string() % urls);
+            % release.current_version.to_string() % release.latest_source.to_string() % release.latest_windows_build.to_string() % urls.str());
   }
 
   return release;
