--- Rakefile.orig
+++ Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 version = RUBY_VERSION.gsub(/[^0-9\.]+/, "").split(/\./).collect(&:to_i)
 version << 0 while version.size < 3
--- doc/Rakefile.orig
+++ doc/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- doc/guide/Rakefile.orig
+++ doc/guide/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- doc/man/Rakefile.orig
+++ doc/man/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- lib/Rakefile.orig
+++ lib/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- lib/avilib-0.6.10/Rakefile.orig
+++ lib/avilib-0.6.10/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- lib/librmff/Rakefile.orig
+++ lib/librmff/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- rake.d/format_string_verifier.rb.orig
+++ rake.d/format_string_verifier.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 require 'digest/sha1'
 
--- rake.d/gtest.rb.orig
+++ rake.d/gtest.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 $gtest_apps     = %w{common propedit}
 $gtest_internal = c(:GTEST_TYPE) == "internal"
--- rake.d/vendor/drake-0.9.2.0.3.1/bin/drake.orig
+++ rake.d/vendor/drake-0.9.2.0.3.1/bin/drake
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 #--
 # Copyright (c) 2003, 2004, 2005, 2006, 2007  Jim Weirich
--- share/ruby/mkvtoolnix/common.rb.orig
+++ share/ruby/mkvtoolnix/common.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 module MKVToolNix
   module Common
--- share/ruby/mkvtoolnix/merge.rb.orig
+++ share/ruby/mkvtoolnix/merge.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 require "pp"
 
--- share/ruby/mkvtoolnix.rb.orig
+++ share/ruby/mkvtoolnix.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 require File.absolute_path(File.dirname(__FILE__)) + "/mkvtoolnix/common.rb"
 require File.absolute_path(File.dirname(__FILE__)) + "/mkvtoolnix/merge.rb"
--- src/Rakefile.orig
+++ src/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/common/Rakefile.orig
+++ src/common/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/common/chapters/Rakefile.orig
+++ src/common/chapters/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/common/strings/Rakefile.orig
+++ src/common/strings/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/common/tags/Rakefile.orig
+++ src/common/tags/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/common/xml/Rakefile.orig
+++ src/common/xml/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/extract/Rakefile.orig
+++ src/extract/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/info/Rakefile.orig
+++ src/info/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/input/Rakefile.orig
+++ src/input/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/merge/Rakefile.orig
+++ src/merge/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/merge/private/Rakefile.orig
+++ src/merge/private/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/Rakefile.orig
+++ src/mkvtoolnix-gui/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/forms/Rakefile.orig
+++ src/mkvtoolnix-gui/forms/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/job_widget/Rakefile.orig
+++ src/mkvtoolnix-gui/job_widget/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/main_window/Rakefile.orig
+++ src/mkvtoolnix-gui/main_window/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/merge_widget/Rakefile.orig
+++ src/mkvtoolnix-gui/merge_widget/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mkvtoolnix-gui/util/Rakefile.orig
+++ src/mkvtoolnix-gui/util/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mmg/Rakefile.orig
+++ src/mmg/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mmg/header_editor/Rakefile.orig
+++ src/mmg/header_editor/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mmg/options/Rakefile.orig
+++ src/mmg/options/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mmg/tabs/Rakefile.orig
+++ src/mmg/tabs/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/mpegparser/Rakefile.orig
+++ src/mpegparser/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/output/Rakefile.orig
+++ src/output/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/propedit/Rakefile.orig
+++ src/propedit/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- src/tools/Rakefile.orig
+++ src/tools/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 $build_tools = true
 
--- src/tools/changelog_to_xml.rb.orig
+++ src/tools/changelog_to_xml.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 require 'rubygems'
 require 'pp'
--- tests/Rakefile.orig
+++ tests/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 import ['..', '../..', '../../..'].collect { |subdir| FileList[File.dirname(__FILE__) + "/#{subdir}/build-config.in"].to_a }.flatten.compact.first.gsub(/build-config.in/, 'Rakefile')
 
--- tests/run.rb.orig
+++ tests/run.rb
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 # Ruby 1.9.x introduce "require_relative" for local requires. 1.9.2
 # removes "." from $: and forces us to use "require_relative". 1.8.x
--- tests/unit/Rakefile.orig
+++ tests/unit/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 $run_unit_tests = true
 
--- tests/unit/common/Rakefile.orig
+++ tests/unit/common/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 $run_unit_tests = true
 
--- tests/unit/propedit/Rakefile.orig
+++ tests/unit/propedit/Rakefile
@@ -1,4 +1,4 @@
-#!/usr/bin/env ruby
+#!/usr/bin/env ruby2.1
 
 $run_unit_tests = true
 
