--- setup.py.orig	2012-02-29 10:59:12.000000000 -0500
+++ setup.py	2012-02-29 11:00:23.000000000 -0500
@@ -30,21 +30,6 @@
     'Topic :: Software Development :: Libraries :: Python Modules',
 ]
 
-# If we are on Mac OS 10.5 or later, attempt a universal binary, which is the way
-# the original system version of readline.so was compiled. Set up flags here.
-UNIVERSAL = ''
-platform = distutils.util.get_platform()
-if platform.startswith('macosx'):
-    osx_version = platform.split('-')[1]
-    if osx_version == '10.5':
-        UNIVERSAL = '-isysroot /Developer/SDKs/MacOSX10.5.sdk -arch i386 -arch ppc -arch x86_64 -arch ppc64'
-    elif osx_version == '10.6':
-        # Starting with 10.6 (Snow Leopard), only Intel architecture is supported
-        UNIVERSAL = '-isysroot /Developer/SDKs/MacOSX10.6.sdk -arch i386 -arch x86_64'
-    elif osx_version > '10.6':
-        # Starting with 10.7 (Lion) and Xcode 4.3, the developer sysroot is inside the Xcode.app - ignore it
-        UNIVERSAL = '-arch i386 -arch x86_64'
-
 # Since we have the latest readline (post 4.2), enable all readline functionality
 # These macros can be found in pyconfig.h.in in the main directory of the Python tarball
 DEFINE_MACROS = [
@@ -91,6 +76,7 @@
     maintainer_email="ludwig.schwardt@gmail.com; github@srid.name",
     url="http://github.com/ludwigschwardt/python-readline",
     license="GNU GPL",
+    extra_path = 'readline',
     platforms=['MacOS X', 'Posix'],
     include_package_data=True,
     ext_modules=[
@@ -98,8 +84,7 @@
                   sources=["Modules/%s.x/readline.c" % (sys.version_info[0],)],
                   include_dirs=['.'],
                   define_macros=DEFINE_MACROS,
-                  extra_compile_args=['-Wno-strict-prototypes'] + UNIVERSAL.split(),
-                  extra_link_args=UNIVERSAL.split(),
+                  extra_compile_args=['-Wno-strict-prototypes'],
                   extra_objects=['readline/libreadline.a', 'readline/libhistory.a'], 
                   libraries=['ncurses']
         ),
