--- system/configfile.c.orig	2011-02-10 16:49:41.000000000 +0900
+++ system/configfile.c	2011-02-10 16:47:42.000000000 +0900
@@ -22,6 +22,10 @@
 int knp_dict_file_already_defined = 0;
 THESAURUS_FILE THESAURUS[THESAURUS_MAX];
 
+float	SVM_FREQ_SD;
+float	SVM_FREQ_SD_NO;
+float	SVM_R_NUM_S_SD;
+float	SVM_R_NUM_E_SD;
 
 /*==================================================================*/
 	    void check_duplicated(int value, char *string)
--- system/extern.h.orig	2011-02-10 16:50:18.000000000 +0900
+++ system/extern.h	2011-02-10 16:45:47.000000000 +0900
@@ -145,10 +145,10 @@
 extern float	AntecedentDecideThresholdForNi;
 extern float	AntecedentDecideThresholdForNoun;
 extern float	CFSimThreshold;
-extern float	SVM_FREQ_SD;
-extern float	SVM_FREQ_SD_NO;
-extern float	SVM_R_NUM_S_SD;
-extern float	SVM_R_NUM_E_SD;
+//extern float	SVM_FREQ_SD;
+//extern float	SVM_FREQ_SD_NO;
+//extern float	SVM_R_NUM_S_SD;
+//extern float	SVM_R_NUM_E_SD;
 
 extern void *matched_ptr;
 
