--- Makefile.orig	2016-06-28 17:22:59.000000000 -0400
+++ Makefile	2023-05-15 15:53:20.000000000 -0400
@@ -5,9 +5,9 @@
 # as possible. Smake first looks for 'SMakefile' and thus the 
 # command 'psmake/smake $@' will use 'SMakefile' to read rules.
 #
-.PHONY: all clean clobber distclean install ibins depend rmdep config TAGS tags tests rmtarget relink
+.PHONY: all smake
+
+all smake:
+	@echo "NOTICE: Using bootstrap 'Makefile' to run 'Gmake.smake' script"
+	sh ./Gmake.smake
 
-all man lint clean clobber distclean install installman ibins depend rmdep config TAGS tags tests rmtarget relink:
-	@echo "NOTICE: Using bootstrap 'Makefile' to make '$@'"
-	cd psmake && sh ./MAKE-all
-	./psmake/smake -r $@
