--- src/osm_elements/osm2pgrouting.cpp.orig	2021-02-01 18:37:16.000000000 +0100
+++ src/osm_elements/osm2pgrouting.cpp	2021-03-23 19:09:31.000000000 +0100
@@ -140,7 +140,7 @@
                 cout << "Can't open database" << endl;
                 return 1;
             }
-            C.disconnect ();
+            C.close ();
         }catch (const std::exception &e){
             cerr << e.what() << std::endl;
             return 1;
