--- Makefile.orig	2005-08-07 17:18:39.000000000 -0700
+++ Makefile	2014-05-18 15:26:36.000000000 -0700
@@ -263,11 +263,15 @@
 
 install-core: core
 	$(call COPY,$(CORE_LIB_FILE_RELEASE),$(LIB_INSTALL_PATH))
+	$(call COPY,$(CORE_LIB_FILE_RELEASE:.a=.dylib),$(LIB_INSTALL_PATH))
 	$(call COPY,$(CORE_LIB_FILE_DEBUG),$(LIB_INSTALL_PATH))
+	$(call COPY,$(CORE_LIB_FILE_DEBUG:.a=.dylib),$(LIB_INSTALL_PATH))
 
 install-allegro: allegro
 	$(call COPY,$(ALLEGRO_LIB_FILE_RELEASE),$(LIB_INSTALL_PATH))
+	$(call COPY,$(ALLEGRO_LIB_FILE_RELEASE:.a=.dylib),$(LIB_INSTALL_PATH))
 	$(call COPY,$(ALLEGRO_LIB_FILE_DEBUG),$(LIB_INSTALL_PATH))
+	$(call COPY,$(ALLEGRO_LIB_FILE_DEBUG:.a=.dylib),$(LIB_INSTALL_PATH))
 
 ifeq "$(COMSPEC)" ""
 install-core-examples: core-examples
@@ -328,10 +332,10 @@
 include make/Makefile.inc
 
 
-$(CORE_EXAMPLES_EXE): examples/%$(EXE_SUFFIX): examples/%.o $(CORE_LIB_FILE_RELEASE)
+$(CORE_EXAMPLES_EXE): examples/%$(EXE_SUFFIX): examples/%.o $(CORE_LIB_FILE_RELEASE:.a=.dylib)
 	$(CC) $^ -o $@ $(LDFLAGS) $(LINK_MATH)
 
-$(ALLEGRO_EXAMPLES_EXE): examples/%$(EXE_SUFFIX): examples/%.o $(ALLEGRO_LIB_FILE_RELEASE) $(CORE_LIB_FILE_RELEASE)
+$(ALLEGRO_EXAMPLES_EXE): examples/%$(EXE_SUFFIX): examples/%.o $(ALLEGRO_LIB_FILE_RELEASE) $(CORE_LIB_FILE_RELEASE:.a=.dylib)
 	$(CC) $^ -o $@ $(LDFLAGS) $(LINK_ALLEGRO)
 
 $(CORE_EXAMPLES_OBJ): examples/%.o: examples/%.c include/dumb.h
