--- pyls/plugins/flake8_lint.py.orig	2019-11-18 09:36:03.000000000 -0500
+++ pyls/plugins/flake8_lint.py	2019-11-18 09:38:01.000000000 -0500
@@ -40,12 +40,12 @@
     """
     log.debug("Calling flake8 with args: '%s'", args)
     try:
-        cmd = ['flake8']
+        cmd = ['@@FLAKE8_BIN@@']
         cmd.extend(args)
         p = Popen(cmd, stdout=PIPE, stderr=PIPE)
     except IOError:
         log.debug("Can't execute flake8. Trying with 'python -m flake8'")
-        cmd = ['python', '-m', 'flake8']
+        cmd = ['@@PYTHON_BIN@@', '-m', 'flake8']
         cmd.extend(args)
         p = Popen(cmd, stdout=PIPE, stderr=PIPE)
     stderr = p.stderr.read().decode()
