--- runfiles/texmf/dvipdfmx/dvipdfmx.cfg.orig	2011-07-14 17:59:09.000000000 -0400
+++ runfiles/texmf/dvipdfmx/dvipdfmx.cfg	2011-07-14 18:01:52.000000000 -0400
@@ -149,7 +149,7 @@
 %% extant.  You can find the one that is active by running:
 %% kpsewhich -progname=dvipdfmx -format='other text files' dvipdfmx.cfg
 %%
-D  "rungs -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=%v -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -sOutputFile='%o' '%i' -c quit"
+D  "gs -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=%v -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -sOutputFile='%o' '%i' -c quit"
 
 %% Frank Siegert's PStill:
 %D  "/usr/local/bin/pstill -c -o '%o' '%i'"
