From 15c942c8d5432b6804e5b6410dae2cfb483d84cc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 23 Feb 2024 09:09:58 +0800
Subject: [PATCH] Fix install paths

---
 features/communi_installs.prf | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git features/communi_installs.prf features/communi_installs.prf
index a976ba3e..36fecb6a 100644
--- features/communi_installs.prf
+++ features/communi_installs.prf
@@ -21,8 +21,8 @@ isEmpty(COMMUNI_INSTALL_THEMES): \
 isEmpty(COMMUNI_INSTALL_BINS) {
     !isEmpty(COMMUNI_INSTALL_PREFIX):COMMUNI_INSTALL_BINS = $$COMMUNI_INSTALL_PREFIX/bin
     else:win32:COMMUNI_INSTALL_BINS = $$[QT_INSTALL_BINS]
-    else:mac:COMMUNI_INSTALL_BINS = /Applications
-    else:COMMUNI_INSTALL_BINS = /usr/bin
+    else:mac:COMMUNI_INSTALL_BINS = @APPDIR@
+    else:COMMUNI_INSTALL_BINS = @PREFIX@/bin
 }
 
 isEmpty(COMMUNI_INSTALL_LIBS) {
@@ -39,16 +39,16 @@ isEmpty(COMMUNI_INSTALL_PLUGINS) {
 
 isEmpty(COMMUNI_INSTALL_ICONS) {
     !isEmpty(COMMUNI_INSTALL_PREFIX):COMMUNI_INSTALL_ICONS = $$COMMUNI_INSTALL_PREFIX/icons
-    else:COMMUNI_INSTALL_ICONS = /usr/share/icons/hicolor
+    else:COMMUNI_INSTALL_ICONS = @PREFIX@/share/icons/hicolor
 }
 
 isEmpty(COMMUNI_INSTALL_DESKTOP) {
     !isEmpty(COMMUNI_INSTALL_PREFIX):COMMUNI_INSTALL_DESKTOP = $$COMMUNI_INSTALL_PREFIX
-    else:COMMUNI_INSTALL_DESKTOP = /usr/share/applications
+    else:COMMUNI_INSTALL_DESKTOP = @PREFIX@/share/applications
 }
 
 isEmpty(COMMUNI_INSTALL_THEMES) {
     !isEmpty(COMMUNI_INSTALL_PREFIX):COMMUNI_INSTALL_THEMES = $$COMMUNI_INSTALL_PREFIX/themes
     else:win32:COMMUNI_INSTALL_THEMES = $$[QT_INSTALL_BINS]/themes
-    else:COMMUNI_INSTALL_THEMES = /usr/share/themes/communi
+    else:COMMUNI_INSTALL_THEMES = @PREFIX@/share/themes/communi
 }
