--- compiler/parser/enrobage.cpp.orig	2011-07-21 11:02:24.000000000 -0500
+++ compiler/parser/enrobage.cpp	2011-07-25 18:13:35.000000000 -0500
@@ -198,12 +198,6 @@
 		f->open(filename); 
 		if (f->good()) return f; else delete f;
 	}
-	err = chdir(old);
-	if (chdir("/usr/lib/faust")==0) {
-        ifstream* f = new ifstream();
-		f->open(filename); 
-		if (f->good()) return f; else delete f;
-	}
 	
 	return 0;
 }
@@ -387,9 +381,6 @@
     if ((f = fopenat(fullpath, "/usr/local/lib/faust", filename))) { 
     	return f;
     }
-    if ((f = fopenat(fullpath, "/usr/lib/faust", filename))) { 
-    	return f;
-    }
     return 0;
 }
 #endif
