--- compiler/parser/enrobage.cpp.orig	2010-01-23 01:01:19.000000000 -0600
+++ compiler/parser/enrobage.cpp	2010-02-05 19:01:09.000000000 -0600
@@ -102,12 +102,6 @@
 		f->open(filename); 
 		if (f->good()) return f; else delete f;
 	}
-	err = chdir(old);
-	if (chdir("/usr/lib/faust")==0) {
-        ifstream* f = new ifstream();
-		f->open(filename); 
-		if (f->good()) return f; else delete f;
-	}
 	
 	return 0;
 }
@@ -279,9 +273,6 @@
     if ((f = fopenat(fullpath, "/usr/local/lib/faust", filename))) { 
     	return f;
     }
-    if ((f = fopenat(fullpath, "/usr/lib/faust", filename))) { 
-    	return f;
-    }
     return 0;
 }
 #endif
