--- g2clib-1.2.1-patches/enc_png.c.orig	2011-02-10 07:53:25.000000000 +0900
+++ g2clib-1.2.1-patches/enc_png.c	2012-09-06 09:31:00.000000000 +0900
@@ -88,7 +88,7 @@
 
 /*    Set new custom write functions    */
 
-    png_set_write_fn(png_ptr,(voidp)&write_io_ptr,(png_rw_ptr)user_write_data,
+    png_set_write_fn(png_ptr,(png_voidp)&write_io_ptr,(png_rw_ptr)user_write_data,
                     (png_flush_ptr)user_flush_data);
 /*    png_init_io(png_ptr, fptr);   */
 /*    png_set_compression_level(png_ptr, Z_BEST_COMPRESSION);  */
