--- ncarg2d/src/bin/ezmapdemo/yMakefile.orig	2017-02-24 04:11:59.000000000 +0900
+++ ncarg2d/src/bin/ezmapdemo/yMakefile	2017-05-02 16:36:01.000000000 +0900
@@ -7,7 +7,7 @@
 SRC    = EzmapDemo.f
 OBJ    = EzmapDemo.o
 
-PROG_LIBS =	$(NCARGLIBS) $(CAIROLIB) $(PNGLIB)
+PROG_LIBS =	$(NCARGLIBS) $(CAIROLIB) $(PNGLIB) ../../libncarg_gks/awi/ggkwdr_stub.o
 DEP_LIBS =	$(DEPNCARGLIBS)
 SYS_LIBS	= $(XLIB)
 
