--- g2clib-1.2.1-patches/dec_png.c.orig	2011-02-10 07:53:25.000000000 +0900
+++ g2clib-1.2.1-patches/dec_png.c	2012-09-06 09:30:41.000000000 +0900
@@ -88,7 +88,7 @@
 
 /*    Set new custom read function    */
 
-    png_set_read_fn(png_ptr,(voidp)&read_io_ptr,(png_rw_ptr)user_read_data);
+    png_set_read_fn(png_ptr,(png_voidp)&read_io_ptr,(png_rw_ptr)user_read_data);
 /*     png_init_io(png_ptr, fptr);   */
 
 /*     Read and decode PNG stream   */
