--- config/ymake.orig	2009-11-05 14:51:37.000000000 +0900
+++ config/ymake	2009-11-05 14:58:25.000000000 +0900
@@ -415,8 +415,8 @@
             set vendor  = Apple
             breaksw
         case    i386:
-            set model   = $mach
-            set arch    = $mach
+            set model   = x86_64
+            set arch    = x86_64
             set sysincs = Darwin_Intel
             set vendor  = Apple
             breaksw
