--- a/projects/compiler-rt/make/platform/clang_darwin.mk.orig	2015-01-10 03:41:48.000000000 -0800
+++ b/projects/compiler-rt/make/platform/clang_darwin.mk	2015-01-10 04:13:39.000000000 -0800
@@ -115,13 +115,13 @@ UniversalArchs.profile_ios := $(call Che
 UniversalArchs.profile_ios += $(call CheckArches,armv7,profile_ios,$(IOS_SDK))
 
 # Configurations which define the ASAN support functions.
-Configs += asan_osx_dynamic
+#Configs += asan_osx_dynamic
 UniversalArchs.asan_osx_dynamic := $(call CheckArches,i386 x86_64,asan_osx_dynamic,$(OSX_SDK))
 
-Configs += asan_iossim_dynamic
+#Configs += asan_iossim_dynamic
 UniversalArchs.asan_iossim_dynamic := $(call CheckArches,i386 x86_64,asan_iossim_dynamic,$(IOSSIM_SDK))
 
-Configs += ubsan_osx
+#Configs += ubsan_osx
 UniversalArchs.ubsan_osx := $(call CheckArches,i386 x86_64,ubsan_osx,$(OSX_SDK))
 
 # Darwin 10.6 has a bug in cctools that makes it unable to use ranlib on our ARM
--- a/tools/clang/runtime/compiler-rt/Makefile.orig	2015-01-08 23:42:54.000000000 -0800
+++ b/tools/clang/runtime/compiler-rt/Makefile	2015-01-10 04:13:59.000000000 -0800
@@ -78,9 +78,7 @@ ifeq ($(OS),Darwin)
 RuntimeDirs += darwin macho_embedded
 RuntimeLibrary.darwin.Configs := \
 	eprintf.a 10.4.a osx.a cc_kext.a \
-	asan_osx_dynamic.dylib \
-	profile_osx.a \
-	ubsan_osx.a
+	profile_osx.a
 
 XCRunSdkPath = \
   $(shell \
