--- configure.ac.orig	2015-05-17 16:32:52.000000000 -0700
+++ configure.ac	2015-07-30 17:02:42.000000000 -0700
@@ -295,14 +295,14 @@
 if test "x$platform_osx" = xyes; then
   AC_DEFINE([OS_OSX],[1],[Defined if system is Mac OS X])
   dnl check for native osx
-  gdk_windowing=`$PKG_CONFIG --variable=target gdk-2.0`
+  gdk_windowing=`$PKG_CONFIG --variable=targets gdk-3.0`
   AC_MSG_CHECKING([for GDK-Quartz])
 
   if test "x$gdk_windowing" = xquartz; then
         platform=darwin/quartz
         AC_MSG_RESULT(yes)
         AC_DEFINE(GDK_QUARTZ,[1],[Using GDK Quartz (not X11)])
-        PKG_CHECK_MODULES(GTK_MAC, gtk-mac-integration, [os_osx=yes], [os_osx=no])
+        PKG_CHECK_MODULES(GTK_MAC, gtk-mac-integration-gtk3, [os_osx=yes], [os_osx=no])
         dnl AC_MSG_RESULT(${os_osx})
         if test "x$os_osx" = xyes; then
 	      AC_DEFINE([GTKOSXAPPLICATION],[1],[Defined if app is GtkOSXApplication])
