--- goffice/math/go-quad.h.orig	2014-09-21 15:03:01.000000000 -0700
+++ goffice/math/go-quad.h	2014-10-11 07:50:04.000000000 -0700
@@ -46,12 +46,12 @@
 
 GO_VAR_DECL const GOQuad go_quad_zero;
 GO_VAR_DECL const GOQuad go_quad_one;
-GO_VAR_DECL const GOQuad go_quad_pi;
-GO_VAR_DECL const GOQuad go_quad_2pi;
-GO_VAR_DECL const GOQuad go_quad_e;
-GO_VAR_DECL const GOQuad go_quad_ln2;
-GO_VAR_DECL const GOQuad go_quad_sqrt2;
-GO_VAR_DECL const GOQuad go_quad_euler;
+GO_VAR_DECL GOQuad go_quad_pi;
+GO_VAR_DECL GOQuad go_quad_2pi;
+GO_VAR_DECL GOQuad go_quad_e;
+GO_VAR_DECL GOQuad go_quad_ln2;
+GO_VAR_DECL GOQuad go_quad_sqrt2;
+GO_VAR_DECL GOQuad go_quad_euler;
 
 #ifdef GOFFICE_WITH_LONG_DOUBLE
 struct GOQuadl_ {
--- goffice/math/go-quad.c.orig	2014-10-11 07:57:10.000000000 -0700
+++ goffice/math/go-quad.c	2014-10-11 07:57:32.000000000 -0700
@@ -273,12 +273,12 @@
 
 const QUAD SUFFIX(go_quad_zero) = { 0, 0 };
 const QUAD SUFFIX(go_quad_one) = { 1, 0 };
-const QUAD SUFFIX(go_quad_pi);
-const QUAD SUFFIX(go_quad_2pi);
-const QUAD SUFFIX(go_quad_e);
-const QUAD SUFFIX(go_quad_ln2);
-const QUAD SUFFIX(go_quad_sqrt2);
-const QUAD SUFFIX(go_quad_euler);
+QUAD SUFFIX(go_quad_pi);
+QUAD SUFFIX(go_quad_2pi);
+QUAD SUFFIX(go_quad_e);
+QUAD SUFFIX(go_quad_ln2);
+QUAD SUFFIX(go_quad_sqrt2);
+QUAD SUFFIX(go_quad_euler);
 
 /**
  * go_quad_init: (skip)
