--- DESCRIPTION	2023-11-08 08:00:13.000000000 +0800
+++ DESCRIPTION	2023-11-08 12:15:33.000000000 +0800
@@ -54,7 +54,7 @@
         parallel, lifecycle
 Depends: R (>= 3.5.0)
 Suggests: mvtnorm, numDeriv, roxygen2 (>= 6.1), rpf (>= 0.45),
-        snowfall, lme4, covr, testthat, umx, ifaTools, knitr, markdown,
+        snowfall, lme4, covr, testthat, ifaTools, knitr, markdown,
         rmarkdown, reshape2, ggplot2
 VignetteBuilder: knitr
 LazyLoad: yes
