From 45adfb7b1e322eaa6174e88f7d5e27ef20e0596e Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Thu, 7 Aug 2025 21:27:31 +0200
Subject: [PATCH] Use Zend/zend_smart_string.h (#741)

---
 imagick.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/imagick.c b/imagick.c
index c5f2c40b..68e63e13 100644
--- a/imagick.c
+++ b/imagick.c
@@ -24,7 +24,9 @@
 #include "php_imagick_helpers.h"
 #include "php_imagick_shared.h"
 
-#if PHP_VERSION_ID >= 70000
+#if PHP_VERSION_ID >= 70200
+#include "Zend/zend_smart_string.h"
+#elif PHP_VERSION_ID >= 70000
 #include "ext/standard/php_smart_string.h"
 #else
 #include "ext/standard/php_smart_str.h"
