diff -ur kmymoney-4.8.4-orig/kmymoney/dialogs/kexportdlg.cpp kmymoney-4.8.4/kmymoney/dialogs/kexportdlg.cpp
--- kmymoney-4.8.4-orig/kmymoney/dialogs/kexportdlg.cpp	2019-08-13 04:54:24.000000000 +0900
+++ kmymoney-4.8.4/kmymoney/dialogs/kexportdlg.cpp	2020-02-11 13:50:28.000000000 +0900
@@ -145,7 +145,7 @@
   }
 
   m_profileComboBox->setCurrentItem(0);
-  if (list.contains(current) > 0)
+  if (list.contains(current) != NULL)
     m_profileComboBox->setCurrentIndex(m_profileComboBox->findText(current, Qt::MatchExactly));
 }
 
diff -ur kmymoney-4.8.4-orig/kmymoney/mymoney/mymoneyreport.cpp kmymoney-4.8.4/kmymoney/mymoney/mymoneyreport.cpp
--- kmymoney-4.8.4-orig/kmymoney/mymoney/mymoneyreport.cpp	2019-08-13 04:54:24.000000000 +0900
+++ kmymoney-4.8.4/kmymoney/mymoney/mymoneyreport.cpp	2020-02-11 13:46:44.000000000 +0900
@@ -888,7 +888,7 @@
   payees(list);
   tags(list);
 
-  return (list.contains(id) > 0);
+  return (list.contains(id) != NULL);
 }
 
 int MyMoneyReport::m_lineWidth = 2;
