Index: kmymoney/plugins/kbanking/mymoneybanking.cpp
===================================================================
--- kmymoney/kmymoney/plugins/kbanking/mymoneybanking.cpp	(revision 1199177)
+++ kmymoney/kmymoney/plugins/kbanking/mymoneybanking.cpp	(working copy)
@@ -902,7 +902,7 @@
       assert(p);
       s += QString::fromUtf8(p);
       se = GWEN_StringListEntry_Next(se);
-      se = GWEN_StringListEntry_Next(se);
+//      se = GWEN_StringListEntry_Next(se);
     } // while
   }
   kt.m_strPayee = s;
