--- kmymoney/CMakeLists.txt	2015-04-23 23:31:22.000000000 +0200
+++ kmymoney/CMakeLists.txt.new	2015-04-23 23:40:20.000000000 +0200
@@ -91,18 +91,6 @@
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}" )
 
-# check for Doxygen
-find_package(Doxygen)
-if(DOXYGEN_FOUND)
-  set(APIDOC_DIR ${CMAKE_CURRENT_BINARY_DIR}/apidocs)
-
-  make_directory(${APIDOC_DIR})
-
-  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/kmymoney.doxygen.in ${CMAKE_CURRENT_BINARY_DIR}/kmymoney.doxygen IMMEDIATE)
-
-  add_custom_target(apidoc ${DOXYGEN} ${CMAKE_CURRENT_BINARY_DIR}/kmymoney.doxygen)
-endif(DOXYGEN_FOUND)
-
 find_package(LibAlkimia 4.3.2 REQUIRED)
 #TODO: remove when LibAlkimia minimum version will be raised
 if (NOT LIBALKIMIA_LIBRARIES AND LIBALKIMIA_LIBRARY)
@@ -295,7 +283,7 @@
 add_subdirectory( libkdchart )
 add_subdirectory( tools )
 add_subdirectory( kmymoney )
-add_subdirectory( doc )
+#add_subdirectory( doc )
 if (USE_DEVELOPER_DOC)
   add_subdirectory( developer-doc )
 endif (USE_DEVELOPER_DOC)
