diff --git a/kmymoney/dialogs/settings/ksettingshomedecl.ui b/kmymoney/dialogs/settings/ksettingshomedecl.ui
index df9b95c..1193561 100644
--- a/kmymoney/dialogs/settings/ksettingshomedecl.ui
+++ b/kmymoney/dialogs/settings/ksettingshomedecl.ui
@@ -183,12 +183,19 @@ Use the buttons and checkboxes to customize the layout of the home page.</string
             <string>Information display</string>
           </property>
           <layout class="QGridLayout">
-            <item row="0" column="0" rowspan="1" colspan="3">
+            <item row="0" column="0" colspan="3">
               <widget class="QCheckBox" name="kcfg_ShowLimitInfo">
                 <property name="text">
                   <string>Show Account Limit Information</string>
                 </property>
               </widget>
+			</item>
+			<item row="1" column="0" colspan="3">
+              <widget class="QCheckBox" name="kcfg_ShowBalanceStatusOfOnlineAccounts">
+                <property name="text">
+                  <string>Show balance-status of mapped online accounts</string>
+                </property>
+              </widget>
             </item>
           </layout>
         </widget>
diff --git a/kmymoney/kmymoney.kcfg b/kmymoney/kmymoney.kcfg
index 5b66586..a60ccad 100644
--- a/kmymoney/kmymoney.kcfg
+++ b/kmymoney/kmymoney.kcfg
@@ -175,6 +175,10 @@
    <label>Show account limit information on the Homepage</label>
    <default>true</default>
   </entry>
+  <entry name="ShowBalanceStatusOfOnlineAccounts" type="Bool">
+   <label>Show balance-status of mapped online accounts on the Homepage</label>
+   <default>false</default>
+  </entry>
  </group>
  <group name="List Options">
   <entry name="HideUnusedCategory" type="Bool">
diff --git a/kmymoney/views/khomeview.cpp b/kmymoney/views/khomeview.cpp
index 498e4a5..403f0ed 100644
--- a/kmymoney/views/khomeview.cpp
+++ b/kmymoney/views/khomeview.cpp
@@ -841,11 +841,31 @@ void KHomeView::showAccountEntry(const MyMoneyAccount& acc, const MyMoneyMoney&
   tmp = QString("<td>") +
         link(VIEW_LEDGER, QString("?id=%1").arg(acc.id())) + acc.name() + linkend() + "</td>";
 
+
+  QString tmp_os, pathOK, pathNotOK;
+  if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts())
+  {
+    //show account's online-status
+    KIconLoader::global()->loadIcon(QString("dialog-ok-apply"), KIconLoader::Small, KIconLoader::SizeSmall, KIconLoader::DefaultState, QStringList(), &pathOK);
+    KIconLoader::global()->loadIcon(QString("dialog-cancel"), KIconLoader::Small, KIconLoader::SizeSmall, KIconLoader::DefaultState, QStringList(), &pathNotOK);
+
+    if(acc.value("lastImportedTransactionDate").isEmpty()	|| acc.value("lastStatementBalance").isEmpty())
+      tmp_os="-";
+    else if(file->hasMatchingOnlineBalance(acc))
+      tmp_os=QString("<img src=\"%1\" border=\"0\"></a>").arg(pathOK);
+    else
+      tmp_os=QString("<img src=\"%1\" border=\"0\"></a>").arg(pathNotOK);
+
+    tmp += QString("<td class=\"center\">%1</td>").arg(tmp_os);
+  }
+
+
   //show account balance
   tmp += QString("<td class=\"right\">%1</td>").arg(showColoredAmount(amount, value.isNegative()));
 
   //show minimum balance column if requested
   if (showMinBal) {
+	if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts()) tmp +=QString("<td></td>");
     //if it is an investment, show minimum balance empty
     if (acc.accountType() == MyMoneyAccount::Investment) {
       tmp += QString("<td class=\"right\">&nbsp;</td>");
@@ -1218,6 +1238,14 @@ void KHomeView::showAssetsLiabilities(void)
 
   //only do it if we have assets or liabilities account
   if (nameAssetsIdx.count() > 0 || nameLiabilitiesIdx.count() > 0) {
+    QString statusHeader;
+    if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts())
+    {
+      QString pathStatusHeader;
+      KIconLoader::global()->loadIcon(QString("online-banking"), KIconLoader::Small, KIconLoader::SizeSmall, KIconLoader::DefaultState, QStringList(), &pathStatusHeader);
+      statusHeader=QString("<img src=\"%1\" border=\"0\"></a>").arg(pathStatusHeader);
+    }
+
     //print header
     d->m_html += "<div class=\"shadow\"><div class=\"displayblock\"><div class=\"summaryheader\">" + i18n("Assets and Liabilities Summary") + "</div>\n<div class=\"gap\">&nbsp;</div>\n";
     d->m_html += "<table width=\"100%\" cellspacing=\"0\" cellpadding=\"2\" class=\"summarytable\" >";
@@ -1225,6 +1253,11 @@ void KHomeView::showAssetsLiabilities(void)
     d->m_html += "<tr class=\"item\"><td class=\"left\" width=\"30%\">";
     d->m_html += i18n("Asset Accounts");
     d->m_html += "</td>";
+    if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts()) {
+      d->m_html += "<td class=\"setcolor\">";
+      d->m_html += statusHeader;
+      d->m_html += "</td>";
+    }
     d->m_html += "<td width=\"15%\" class=\"right\">";
     d->m_html += i18n("Current Balance");
     d->m_html += "</td>";
@@ -1233,10 +1266,18 @@ void KHomeView::showAssetsLiabilities(void)
     d->m_html += "<td class=\"left\" width=\"30%\">";
     d->m_html += i18n("Liability Accounts");
     d->m_html += "</td>";
+    if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts()) {
+      d->m_html += "<td class=\"setcolor\">";
+      d->m_html += statusHeader;
+      d->m_html += "</td>";
+    }
     d->m_html += "<td width=\"15%\" class=\"right\">";
     d->m_html += i18n("Current Balance");
     d->m_html += "</td></tr>";
 
+    QString tmp_os;
+    if(KMyMoneyGlobalSettings::showBalanceStatusOfOnlineAccounts()) tmp_os="<td></td>";
+
     //get asset and liability accounts
     QMap<QString, MyMoneyAccount>::const_iterator asset_it = nameAssetsIdx.constBegin();
     QMap<QString, MyMoneyAccount>::const_iterator liabilities_it = nameLiabilitiesIdx.constBegin();
@@ -1267,7 +1308,7 @@ void KHomeView::showAssetsLiabilities(void)
         ++asset_it;
       } else {
         //write a white space if we don't
-        d->m_html += "<td></td><td></td>";
+        d->m_html += QString("<td></td>%1<td></td>").arg(tmp_os);
       }
 
       //leave the intermediate column empty
@@ -1292,7 +1333,7 @@ void KHomeView::showAssetsLiabilities(void)
         ++liabilities_it;
       } else {
         //leave the space empty if we run out of liabilities
-        d->m_html += "<td></td><td></td>";
+        d->m_html += QString("<td></td>%1<td></td>").arg(tmp_os);
       }
       d->m_html += "</tr>";
     }
@@ -1311,20 +1352,20 @@ void KHomeView::showAssetsLiabilities(void)
     d->m_html += QString("<tr class=\"row-%1\" style=\"font-weight:bold;\">").arg(i++ & 0x01 ? "even" : "odd");
 
     //print total for assets
-    d->m_html += QString("<td class=\"left\">%1</td><td align=\"right\">%2</td>").arg(i18n("Total Assets")).arg(showColoredAmount(amountAssets, netAssets.isNegative()));
+    d->m_html += QString("<td class=\"left\">%1</td>%2<td align=\"right\">%3</td>").arg(i18n("Total Assets")).arg(tmp_os).arg(showColoredAmount(amountAssets, netAssets.isNegative()));
 
     //leave the intermediate column empty
     d->m_html += "<td class=\"setcolor\"></td>";
 
     //print total liabilities
-    d->m_html += QString("<td class=\"left\">%1</td><td align=\"right\">%2</td>").arg(i18n("Total Liabilities")).arg(showColoredAmount(amountLiabilities, netLiabilities.isNegative()));
+    d->m_html += QString("<td class=\"left\">%1</td>%2<td align=\"right\">%3</td>").arg(i18n("Total Liabilities")).arg(tmp_os).arg(showColoredAmount(amountLiabilities, netLiabilities.isNegative()));
     d->m_html += "</tr>";
 
     //print net worth
     d->m_html += QString("<tr class=\"row-%1\" style=\"font-weight:bold;\">").arg(i++ & 0x01 ? "even" : "odd");
 
-    d->m_html += "<td></td><td></td><td class=\"setcolor\"></td>";
-    d->m_html += QString("<td class=\"left\">%1</td><td align=\"right\">%2</td>").arg(i18n("Net Worth")).arg(showColoredAmount(amountNetWorth, netWorth.isNegative()));
+    d->m_html += QString("<td></td><td></td>%1<td class=\"setcolor\"></td>").arg(tmp_os);
+    d->m_html += QString("<td class=\"left\">%1</td>%2<td align=\"right\">%3</td>").arg(i18n("Net Worth")).arg(tmp_os).arg(showColoredAmount(amountNetWorth, netWorth.isNegative()));
 
     d->m_html += "</tr>";
     d->m_html += "</table>";
