--- src/vmime/encoding.cpp.orig	2024-12-07 13:05:38
+++ src/vmime/encoding.cpp	2024-12-07 13:08:19
@@ -136,8 +136,8 @@
 	(const string::const_iterator begin, const string::const_iterator end)
 {
 	const string::difference_type length = end - begin;
-	const string::difference_type count = std::count_if
-		(begin, end, std::bind2nd(std::less<unsigned char>(), 127));
+	const string::difference_type count = std::count_if(
+		begin, end, [](unsigned char x) { return x < 127; });
 
 	// All is in 7-bit US-ASCII --> 7-bit (or Quoted-Printable...)
 	if (length == count)
--- src/vmime/propertySet.hpp.orig	2017-01-04 00:00:08
+++ src/vmime/propertySet.hpp	2024-12-07 13:10:37
@@ -335,8 +335,7 @@
 	void parse(const string& props);
 
 
-	class propFinder : public std::unary_function <shared_ptr <property>, bool>
-	{
+	class propFinder {
 	public:
 
 		propFinder(const string& name) : m_name(utility::stringUtils::toLower(name)) { }
