--- bee-0.12.2.gemspec
+++ bee-0.12.2.gemspec
@@ -28,10 +28,10 @@
   if s.respond_to? :add_runtime_dependency then
     s.add_runtime_dependency(%q<archive-tar-minitar>.freeze, [">= 0.5.1"])
     s.add_runtime_dependency(%q<highline>.freeze, [">= 1.5.2"])
-    s.add_runtime_dependency(%q<rubyzip>.freeze, [">= 0.9.1", "< 1.0.0"])
+    s.add_runtime_dependency(%q<rubyzip>.freeze, [">= 0.9.1"])
   else
     s.add_dependency(%q<archive-tar-minitar>.freeze, [">= 0.5.1"])
     s.add_dependency(%q<highline>.freeze, [">= 1.5.2"])
-    s.add_dependency(%q<rubyzip>.freeze, [">= 0.9.1", "< 1.0.0"])
+    s.add_dependency(%q<rubyzip>.freeze, [">= 0.9.1"])
   end
 end
