--- Makefile	2012-06-19 22:56:08.000000000 +0200
+++ Makefile	2012-07-31 11:07:15.000000000 +0200
@@ -159,7 +159,7 @@
 		$(PLUGINS_BYTE) $(PLUGINS_OPT) $(PLUGINS_OPT:.cmxs=.cmx) $(PLUGINS_OPT:.cmxs=.o) \
 		$(LIB_CMIFILES) $(LIB_CMXFILES) $(LIB_CMXFILES:.cmx=.o) \
 		$(LIB_BYTE) $(LIB) $(LIB:.cmxa=.a)
-	$(CP) $(MAIN) $(MAIN_BYTE)  `dirname \`which $(OCAMLC)\``/
+	$(CP) $(MAIN) $(MAIN_BYTE)  $(DESTDIR)$(BINDIR)
 
 uninstall:
 	@$(OCAMLFIND) remove stog
