--- DESCRIPTION	2017-06-28 13:43:10.000000000 +0700
+++ DESCRIPTION	2023-09-28 02:54:58.000000000 +0700
@@ -15,7 +15,7 @@
 	'RSQLite', 'RH2', 'RMySQL' and 'RPostgreSQL' backends are supported.
 ByteCompile: true
 Depends: R (>= 3.1.0), gsubfn (>= 0.6), proto, RSQLite
-Suggests: RH2, RMySQL, RPostgreSQL, svUnit, tcltk, MASS
+Suggests: RMySQL, RPostgreSQL, svUnit, tcltk, MASS
 Imports: DBI, chron
 License: GPL-2
 BugReports: https://github.com/ggrothendieck/sqldf/issues
