--- data/scripts/create_apple_bundle.sh.in.orig	2015-03-07 20:47:14.000000000 -0600
+++ data/scripts/create_apple_bundle.sh.in	2015-05-08 04:49:50.000000000 -0500
@@ -11,7 +11,7 @@
 #=========================================================================================
  
 # MacPorts Location
-MACPORTS=/opt/local
+MACPORTS=@PREFIX@
  
  # LMMS project root directory (extracted source)
 CMAKE_SRC=@CMAKE_SOURCE_DIR@
@@ -23,10 +23,10 @@
 CMAKE_BUILD=$(cd @CMAKE_CURRENT_BINARY_DIR@/../..; pwd)
 
 # STK rawwaves directory
-STK_RAWWAVE=${HOME}/stk-*/rawwaves
+STK_RAWWAVE=@PREFIX@/share/stk/rawwaves
  
 # Place to create ".app" bundle
-APP=${HOME}/Desktop/LMMS.app
+APP=@DESTROOT@@APPLICATIONS_DIR@/LMMS.app
  
 # MacPorts installs libreadline with wrong permissions
 LIBREADLINE=${MACPORTS}/lib/libreadline.6.2.dylib
@@ -63,9 +63,9 @@
 mkdir ${APP}/Contents
 cp -R * ${APP}/Contents
  
-# Manually copy STK rawwaves
-mkdir -p ${APP}/Contents/share/stk/rawwaves
-cp ${STK_RAWWAVE}/*.raw ${APP}/Contents/share/stk/rawwaves
+# Symlink STK rawwaves
+mkdir -p ${APP}/Contents/share/stk
+ln -s ${STK_RAWWAVE} ${APP}/Contents/share/stk/rawwaves
  
 # Make all libraries writable for macdeployqt
 cd ${APP}
@@ -107,11 +107,10 @@
 _executables="${_executables} -executable=${APP}/Contents/MacOS/RemoteZynAddSubFx"
 _executables="${_executables} -executable=${APP}/Contents/Frameworks/libZynAddSubFxCore.dylib"
  
-# Build our App Package using "macdeployqt"
-macdeployqt ${APP} $_executables
- 
 # OS X Specific Artwork
+mkdir -p ${APP}/Contents/Resources
 cp ${CMAKE_SRC}/data/*.icns ${APP}/Contents/Resources/
+exit
  
 # Done.  Ready to build DMG
 echo -e "\nFinished.\n\nYou may run LMMS from the Desktop."
