--- cmake/modules/DetectMachine.cmake.orig	2015-03-07 20:47:14.000000000 -0600
+++ cmake/modules/DetectMachine.cmake	2015-05-11 10:32:47.000000000 -0500
@@ -20,7 +20,7 @@
 		SET(IS_X86 TRUE)
 	ENDIF(WIN64)
 ELSE(WIN32)
-	EXEC_PROGRAM( ${CMAKE_C_COMPILER} ARGS "-dumpmachine" OUTPUT_VARIABLE Machine )
+	EXEC_PROGRAM( ${CMAKE_C_COMPILER} ARGS "-dumpmachine ${CMAKE_C_FLAGS}" OUTPUT_VARIABLE Machine )
 	MESSAGE("Machine: ${Machine}")
 	STRING(REGEX MATCH "i.86" IS_X86 "${Machine}")
 	STRING(REGEX MATCH "86_64" IS_X86_64 "${Machine}")
