--- CUnit/Sources/Makefile.am.orig	2014-04-24 10:54:45.000000000 +1000
+++ CUnit/Sources/Makefile.am	2014-12-08 06:19:32.000000000 +1100
@@ -67,6 +67,9 @@
 lib_LTLIBRARIES = libcunit.la
 libcunit_la_SOURCES = 
 libcunit_la_LIBADD = $(OBJECT_FILES_SHARED)
+if ENABLE_CURSES
+libcunit_la_LIBADD += -l$(CURSES_LIB)
+endif
 libcunit_la_LDFLAGS = -version-info @LIBTOOL_CURRENT@:@LIBTOOL_REVISION@:@LIBTOOL_AGE@
 
 
