From cfd1b748e47cf66b7d7921fe9eeae743b9ab8f84 Mon Sep 17 00:00:00 2001
From: Philip Withnall <philip.withnall@collabora.co.uk>
Date: Thu, 30 Oct 2014 17:54:46 +0000
Subject: build: Port from appdata-tools to appstream-glib

https://bugzilla.gnome.org/show_bug.cgi?id=739406

diff --git a/Makefile.am b/Makefile.am
index 4cad8f6..5cc5e53 100644
--- Makefile.am
+++ Makefile.am
@@ -35,7 +35,7 @@ DISTCHECK_CONFIGURE_FLAGS = \
 	--enable-gtk-doc \
 	--with-nautilusdir='$${libdir}/nautilus/extensions-2.0-distcheck' \
 	--enable-introspection \
-	--enable-appdata-validate
+	--enable-appstream-util
 
 # Build ChangeLog from GIT history
 ChangeLog:
diff --git a/configure.ac b/configure.ac
index 0889741..50a3bd7 100644
--- configure.ac
+++ configure.ac
@@ -495,7 +495,7 @@ GNOME_CXX_WARNINGS
 GNOME_MAINTAINER_MODE_DEFINES
 GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
 
-APPDATA_XML
+APPSTREAM_XML
 
 # This macro expands DIR and assigns it to RET.
 # If DIR is NONE, then it's replaced by DEFAULT.
diff --git a/data/appdata/Makefile.am b/data/appdata/Makefile.am
index 50d848f..89ade54 100644
--- data/appdata/Makefile.am
+++ data/appdata/Makefile.am
@@ -1,9 +1,9 @@
 # AppData
-@APPDATA_XML_RULES@
+@APPSTREAM_XML_RULES@
 @INTLTOOL_XML_RULE@
 
-appdata_in_files = org.gnome.Totem.appdata.xml.in
-appdata_XML = $(appdata_in_files:.xml.in=.xml)
+appstream_in_files = org.gnome.Totem.appdata.xml.in
+appstream_XML = $(appstream_in_files:.xml.in=.xml)
 
-EXTRA_DIST = $(appdata_in_files)
-CLEANFILES = $(appdata_XML)
+EXTRA_DIST = $(appstream_in_files)
+CLEANFILES = $(appstream_XML)
-- 
cgit v0.10.1

