--- libmodplug.pc.in.orig	2013-08-22 16:55:58.000000000 -0700
+++ libmodplug.pc.in	2014-04-26 09:40:24.000000000 -0700
@@ -8,5 +8,4 @@
 Version: @VERSION@
 Requires: 
 Libs: -L${libdir} -lmodplug 
-Libs.private: -lstdc++ -lm
 Cflags: -I${includedir}
--- configure.ac.orig	2017-04-24 12:59:31.000000000 +0000
+++ configure.ac	2017-07-21 16:27:50.000000000 +0000
@@ -44,13 +44,6 @@
 esac
 AC_SUBST(LT_LDFLAGS)
 
-# require 10.5+ for osx/x86_64 builds
-case "$host" in
-	x86_64-*-darwin*)
-		CXXFLAGS="$CXXFLAGS -mmacosx-version-min=10.5"
-		LDFLAGS="$LDFLAGS -mmacosx-version-min=10.5" ;;
-esac
-
 # symbol visibility
 ac_save_CXXFLAGS="$CXXFLAGS"
 CXXFLAGS="$CXXFLAGS -fvisibility=hidden -Werror"
