--- build.sh.orig	2014-06-09 15:41:49.000000000 -0500
+++ build.sh	2014-10-16 21:26:27.000000000 -0500
@@ -12,7 +12,7 @@
 # Beware: GNU libtool cannot handle directory names containing whitespace.
 #         Therefore, do not set M_CONF_TMPDIR to such a directory.
 #
-readonly M_CONF_TMPDIR=/tmp
+readonly M_CONF_TMPDIR=@@TMP@@
 readonly M_PLISTSIGNER_TEST_KEY="`dirname $0`/prefpane/autoinstaller/TestKeys/private_key.der"
 
 # Other constants
@@ -104,37 +104,37 @@
 declare M_XCODE_VERSION_REQUIRED=""
 
 # SDK 10.5
-readonly M_SDK_105_ARCHS="ppc ppc64 i386 x86_64"
+readonly M_SDK_105_ARCHS="@@ARCHS@@"
 declare M_SDK_105=""
 declare M_SDK_105_XCODE=""
 declare M_SDK_105_COMPILER=""
 
 # SDK 10.6
-readonly M_SDK_106_ARCHS="i386 x86_64"
+readonly M_SDK_106_ARCHS="@@ARCHS@@"
 declare M_SDK_106=""
 declare M_SDK_106_XCODE=""
 declare M_SDK_106_COMPILER=""
 
 # SDK 10.7
-readonly M_SDK_107_ARCHS="i386 x86_64"
+readonly M_SDK_107_ARCHS="@@ARCHS@@"
 declare M_SDK_107=""
 declare M_SDK_107_XCODE=""
 declare M_SDK_107_COMPILER=""
 
 # SDK 10.8
-readonly M_SDK_108_ARCHS="i386 x86_64"
+readonly M_SDK_108_ARCHS="@@ARCHS@@"
 declare M_SDK_108=""
 declare M_SDK_108_XCODE=""
 declare M_SDK_108_COMPILER=""
 
 # SDK 10.9
-readonly M_SDK_109_ARCHS="i386 x86_64"
+readonly M_SDK_109_ARCHS="@@ARCHS@@"
 declare M_SDK_109=""
 declare M_SDK_109_XCODE=""
 declare M_SDK_109_COMPILER=""
 
 # SDK 10.10
-readonly M_SDK_1010_ARCHS="i386 x86_64"
+readonly M_SDK_1010_ARCHS="@@ARCHS@@"
 declare M_SDK_1010=""
 declare M_SDK_1010_XCODE=""
 declare M_SDK_1010_COMPILER=""
@@ -1880,7 +1880,7 @@
 
     local ms_osxfuse_out="$M_CONF_TMPDIR/osxfuse-homebrew-$ms_os_version-$ms_osxfuse_version"
     local ms_osxfuse_build="$ms_osxfuse_out/build/"
-    local ms_osxfuse_root="$m_prefix"
+    local ms_osxfuse_root="@@DESTROOT@@$m_prefix"
 
     if [ -e "$ms_osxfuse_out" ]
     then
@@ -1984,7 +1984,7 @@
     xcrun make -j4 >$m_stdout 2>$m_stderr
     m_exit_on_error "make failed while compiling the OSXFUSE library."
 
-    xcrun make install >$m_stdout 2>$m_stderr
+    xcrun make install DESTDIR="@@DESTROOT@@" >$m_stdout 2>$m_stderr
     m_exit_on_error "cannot prepare library build for installation."
 
     for f in "$ms_osxfuse_root"/lib/libosxfuse_i64*.dylib; do
