--- kext/common/fuse_param.h.orig	2013-12-22 18:50:41.000000000 +0100
+++ kext/common/fuse_param.h	2013-12-22 18:50:44.000000000 +0100
@@ -59,7 +59,7 @@
 
 /* Paths */
 
-#define OSXFUSE_BUNDLE_PATH "/Library/Filesystems/osxfusefs.fs"
+#define OSXFUSE_BUNDLE_PATH "@@PREFIX@@/Library/Filesystems/osxfusefs.fs"
 #define OSXFUSE_KEXT        OSXFUSE_BUNDLE_PATH "/Support/osxfusefs.kext"
 #define OSXFUSE_LOAD_PROG   OSXFUSE_BUNDLE_PATH "/Support/load_osxfusefs"
 #define OSXFUSE_MOUNT_PROG  OSXFUSE_BUNDLE_PATH "/Support/mount_osxfusefs"
