--- build.sh.orig	2018-09-10 15:28:54.000000000 +0800
+++ build.sh	2018-09-10 15:30:22.000000000 +0800
@@ -353,6 +353,7 @@
 
     local -a command=(/usr/bin/xcodebuild
                       -configuration "${BUILD_TARGET_OPTION_BUILD_CONFIGURATION}"
+                      -IDECustomDerivedDataLocation="@@WORKSRCPATH@@/IDECustomDerivedData"
                       CONFIGURATION_BUILD_DIR="`build_target_get_build_directory "${BUILD_TARGET_NAME}"`"
                       SDKROOT="macosx${BUILD_TARGET_OPTION_SDK}"
                       ARCHS="`array_join BUILD_TARGET_OPTION_ARCHITECTURES " "`"
