--- build.d/targets/packagemanager.sh.old	2017-09-22 19:32:07.000000000 -0700
+++ build.d/targets/packagemanager.sh	2017-09-22 19:32:53.000000000 -0700
@@ -120,11 +120,11 @@
 
     # Build file system bundle
 
-    build_target_invoke fsbundle build "${default_build_options[@]}"
-    common_die_on_error "Failed to build file system bundle"
+    #build_target_invoke fsbundle build "${default_build_options[@]}"
+    #common_die_on_error "Failed to build file system bundle"
 
-    build_target_invoke fsbundle install --debug="${debug_directory}" -- "${stage_directory}/Library/Filesystems"
-    common_die_on_error "Failed to install file system bundle"
+    #build_target_invoke fsbundle install --debug="${debug_directory}" -- "${stage_directory}/Library/Filesystems"
+    #common_die_on_error "Failed to install file system bundle"
 
     # Build library
 
@@ -152,14 +152,14 @@
 
     # Locate file system bundle
 
-    local fsbundle_path=""
-    fsbundle_path="`osxfuse_find "${stage_directory}/Library/Filesystems"/*.fs`"
-    common_die_on_error "Failed to locate file system bundle"
+    #local fsbundle_path=""
+    #fsbundle_path="`osxfuse_find "${stage_directory}/Library/Filesystems"/*.fs`"
+    #common_die_on_error "Failed to locate file system bundle"
 
     # Move debug files into file system bundle
 
-    /bin/mv "${debug_directory}" "${fsbundle_path}/Contents/"
-    common_die_on_error "Failed to move debug files into file system bundle"
+    #/bin/mv "${debug_directory}" "${fsbundle_path}/Contents/"
+    #common_die_on_error "Failed to move debug files into file system bundle"
 }
 
 function packagemanager_install
