Upstream-Status: Inappropriate (Upstream has refactored their build system and this patch would no longer apply)
--- build.sh.orig	2015-10-25 13:04:47.000000000 -0500
+++ build.sh	2016-09-18 22:33:41.000000000 -0500
@@ -117,6 +117,15 @@
 declare M_XCODE71=""
 declare M_XCODE71_VERSION=7.1
 readonly M_XCODE71_COMPILER="com.apple.compilers.llvm.clang.1_0"
+declare M_XCODE72=""
+declare M_XCODE72_VERSION=7.2
+readonly M_XCODE72_COMPILER="com.apple.compilers.llvm.clang.1_0"
+declare M_XCODE73=""
+declare M_XCODE73_VERSION=7.3
+readonly M_XCODE73_COMPILER="com.apple.compilers.llvm.clang.1_0"
+declare M_XCODE80=""
+declare M_XCODE80_VERSION=8.0
+readonly M_XCODE80_COMPILER="com.apple.compilers.llvm.clang.1_0"
 
 declare M_ACTUAL_PLATFORM=""
 declare M_PLATFORMS=""
@@ -2652,6 +2661,30 @@
                     M_XCODE71_VERSION=$m_xcode_version
                 fi
                 ;;
+            7.2*)
+                m_version_compare $M_XCODE72_VERSION $m_xcode_version
+                if [[ $? != 2 ]]
+                then
+                    M_XCODE72="$m_xcode_root"
+                    M_XCODE72_VERSION=$m_xcode_version
+                fi
+                ;;
+            7.3*)
+                m_version_compare $M_XCODE73_VERSION $m_xcode_version
+                if [[ $? != 2 ]]
+                then
+                    M_XCODE73="$m_xcode_root"
+                    M_XCODE73_VERSION=$m_xcode_version
+                fi
+                ;;
+            8.0*)
+                m_version_compare $M_XCODE80_VERSION $m_xcode_version
+                if [[ $? != 2 ]]
+                then
+                    M_XCODE73="$m_xcode_root"
+                    M_XCODE73_VERSION=$m_xcode_version
+                fi
+                ;;
             *)
                 m_log "skip unsupported Xcode version in '$m_xcode_root'."
                 ;;
@@ -2929,6 +2962,33 @@
         M_SDK_1011_COMPILER="$M_XCODE71_COMPILER"
         m_platform_realistic_add "10.11"
     fi
+    if [[ -n "$M_XCODE72" ]]
+    then
+        m_xcode_latest="$M_XCODE72"
+
+        M_SDK_1011="$M_XCODE72/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk"
+        M_SDK_1011_XCODE="$M_XCODE72"
+        M_SDK_1011_COMPILER="$M_XCODE72_COMPILER"
+        m_platform_realistic_add "10.11"
+    fi
+    if [[ -n "$M_XCODE73" ]]
+    then
+        m_xcode_latest="$M_XCODE73"
+
+        M_SDK_1011="$M_XCODE73/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk"
+        M_SDK_1011_XCODE="$M_XCODE73"
+        M_SDK_1011_COMPILER="$M_XCODE73_COMPILER"
+        m_platform_realistic_add "10.11"
+    fi
+    if [[ -n "$M_XCODE80" ]]
+    then
+        m_xcode_latest="$M_XCODE80"
+
+        M_SDK_1012="$M_XCODE80/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk"
+        M_SDK_1012_XCODE="$M_XCODE80"
+        M_SDK_1012_COMPILER="$M_XCODE80_COMPILER"
+        m_platform_realistic_add "10.12"
+    fi
 
     m_read_input "$@"
 
