--- src/google/protobuf/generated_message_tctable_decl.h	2025-03-26 23:20:57.000000000 +0800
+++ src/google/protobuf/generated_message_tctable_decl.h	2025-05-18 08:12:04.000000000 +0800
@@ -154,7 +154,7 @@
  public:
   MapTypeCard() = default;
   constexpr MapTypeCard(int number, WireFormatLite::WireType wiretype,
-                        bool is_signed, bool is_zigzag, bool is_utf8)
+                        uint8_t is_signed, uint8_t is_zigzag, uint8_t is_utf8)
       : tag_(static_cast<uint8_t>(WireFormatLite::MakeTag(number, wiretype))),
         is_signed_(is_signed),
         is_zigzag_(is_zigzag),
@@ -166,22 +166,22 @@
     return static_cast<WireFormatLite::WireType>(tag_ & 7);
   }
 
-  bool is_signed() const { return is_signed_; }
+  uint8_t is_signed() const { return is_signed_; }
 
-  bool is_zigzag() const {
+  uint8_t is_zigzag() const {
     ABSL_DCHECK(wiretype() == WireFormatLite::WIRETYPE_VARINT);
     return is_zigzag_;
   }
-  bool is_utf8() const {
+  uint8_t is_utf8() const {
     ABSL_DCHECK(wiretype() == WireFormatLite::WIRETYPE_LENGTH_DELIMITED);
     return is_utf8_;
   }
 
  private:
   uint8_t tag_;
-  bool is_signed_ : 1;
-  bool is_zigzag_ : 1;
-  bool is_utf8_ : 1;
+  uint8_t is_signed_ : 1;
+  uint8_t is_zigzag_ : 1;
+  uint8_t is_utf8_ : 1;
 };
 
 // Make the map entry type card for a specified field type.
