--- kmymoney-4.6.3/kmymoney/mymoney/mymoneymoney.cpp.old	2013-09-12 20:59:18.000000000 +0200
+++ kmymoney-4.6.3/kmymoney/mymoney/mymoneymoney.cpp	2013-09-12 21:13:43.000000000 +0200
@@ -164,7 +164,7 @@
   // be much better than using KGlobal::locale()->formatMoney.
   bool bNegative = false;
   mpz_class left = value / static_cast<MyMoneyMoney>(convertDenominator(d)).valueRef().get_den();
-  mpz_class right = (valueRef() - mpq_class(left)) * denom;
+  mpz_class right = mpz_class((valueRef() - mpq_class(left)) * denom);
 
   if (right < 0) {
     right = -right;
