diff -ruEN orig/Makefile src/Makefile
--- orig/Makefile	2020-06-25 21:54:53.000000000 +0700
+++ src/Makefile	2020-06-29 11:28:35.000000000 +0700
@@ -10,17 +10,16 @@
 
 OBJS=$(patsubst %.cpp,%.o,$(SRCS))
 
-WX_LIBS = -pthread   `wx-config --libs`
+WX_LIBS=`wx-config --libs`
 
-LIBS=-lstdc++ -lportaudio
-#LIBS=-lstdc++ /usr/lib/x86_64-linux-gnu/libportaudio.so.2
+WX_CFLAGS=`wx-config --cflags`
 
-CPPFLAGS =  -Wall -g -fexceptions `wx-config --cflags`
-	-I/usr/include/wx-2.8 \
-	-DGTK_NO_CHECK_CASTS -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES
-	-D_LARGEFILE_SOURCE=1 -DNO_GCC_PRAGMA -D_ESPEAKEDIT
+LIBS=-lportaudio
 
-CXXFLAGS = -O2  -Wall -fexceptions `wx-config --cflags` \
+CPPFLAGS= -Wall -g -fexceptions $(WX_CFLAGS) \
+   -DGTK_NO_CHECK_CASTS -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES -D_LARGEFILE_SOURCE=1 -DNO_GCC_PRAGMA -D_ESPEAKEDIT
+
+CXXFLAGS= -O2 -Wall -fexceptions $(WX_CFLAGS) \
    -DGTK_NO_CHECK_CASTS -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES -D_LARGEFILE_SOURCE=1 -DNO_GCC_PRAGMA -D_ESPEAKEDIT
 
 all: espeakedit
@@ -29,7 +28,7 @@
 	$(CXX) $(CXXFLAGS) -I. -c -fno-exceptions $<
 
 espeakedit: $(OBJS)
-	$(CXX) -o $@ $(OBJS) $(LIBS) $(WX_LIBS)
+	$(CXX) -o $@ $(OBJS) $(LIBS) -pthread $(WX_LIBS)
 
 clean:
 	rm -f *.o *~
diff -ruEN orig/espeakedit_test src/espeakedit_test
--- orig/espeakedit_test	1970-01-01 07:00:00.000000000 +0700
+++ src/espeakedit_test	2020-06-29 12:00:31.000000000 +0700
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+DISPLAY=:55
+xvfb :55 -screen 0 1024x768x24 &
+XVFBPID=$!
+./espeakedit --help;
+RESULT=$?;
+if [ $RESULT -eq 0 ]; then
+      kill $XVFBPID;
+      exit 0;
+else
+      kill $XVFBPID;
+      exit 1;
+fi
diff -ruEN orig/spectdisplay.cpp src/spectdisplay.cpp
--- orig/spectdisplay.cpp	2020-06-25 21:54:53.000000000 +0700
+++ src/spectdisplay.cpp	2020-06-25 19:43:38.000000000 +0700
@@ -38,6 +38,7 @@
 #include "translate.h"
 #include "spect.h"
 #include "options.h"
+#include "locale.h" // for LC_LOCALE definition
 
 extern MyFrame *frame ;
 #define SCROLLUNITS  20
diff -ruEN orig/speech.h src/speech.h
--- orig/speech.h	2020-06-25 21:54:53.000000000 +0700
+++ src/speech.h	2020-06-27 22:54:34.000000000 +0700
@@ -47,12 +47,16 @@
 //#define ESPEAK_API  extern "C"
 
 #ifdef LIBRARY
+#ifndef __APPLE__
 #define USE_ASYNC
 #endif
+#endif
 
 #ifdef _ESPEAKEDIT
 #define USE_PORTAUDIO
+#ifndef __APPLE__
 #define USE_ASYNC
+#endif
 #define LOG_FRAMES      // write keyframe info to log-espeakedit
 #endif
 
diff -ruEN orig/transldlg.cpp src/transldlg.cpp
--- orig/transldlg.cpp	2020-06-25 21:54:53.000000000 +0700
+++ src/transldlg.cpp	2020-06-25 21:26:22.000000000 +0700
@@ -131,7 +131,7 @@
 	wxSound(wxString(fname,wxConvLocal)).Play(wxSOUND_SYNC);
 #else
 //	wxSound(wxString(fname,wxConvLocal)).Play(wxSOUND_SYNC);  // This gives a click at the start of each play
-	sprintf(command,"aplay %s",fname);
+	sprintf(command,"afplay %s",fname);
 	if(system(command) == -1)
 	{
 		sprintf(command,"play %s",fname);
