--- src/Makefile	Mon Feb 16 23:50:04 2009 +0100
+++ src/Makefile	Tue Feb 17 00:33:14 2009 +0100
@@ -85,8 +85,8 @@
 	@echo " --- Installing global data --- "
 	@mkdir -p ${DESTDIR}${prefix}/share/pal;
 	install -m 0644 -o root ../share/*.pal ${DESTDIR}${prefix}/share/pal
-	@mkdir -p ${DESTDIR}/etc;
-	install -m 0644 -o root ../pal.conf ${DESTDIR}/etc/
+	@mkdir -p ${DESTDIR}${prefix}/etc;
+	install -m 0644 -o root ../pal.conf ${DESTDIR}${prefix}/etc/
 	@echo
 
 # "install" will install the files needed by pal and also check to see
--- src/input.c	2017-04-05 22:00:27.000000000 +0200
+++ src/input.c	2017-04-05 22:01:44.000000000 +0200
@@ -568,7 +568,7 @@
 	    }
 
 	    /* attempt to copy /etc/pal.conf to ~/.pal/pal.conf */
-	    file = fopen("/etc/pal.conf", "r");
+	    file = fopen(PREFIX "/etc/pal.conf", "r");
 
 	    /* if not found, try PREFIX/share/pal/pal.conf instead */
 	    /* NOTE: This is will be removed in the future */
