Don't look in /usr/include/ImageMagick.
--- PerlMagick/Makefile.PL.in.orig	2023-10-09 01:37:25
+++ PerlMagick/Makefile.PL.in	2023-10-11 13:42:44
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../ -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../ -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ @MATH_LIBS@ -L' . $Config{'archlib'} . '/CORE';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
--- PerlMagick/default/Makefile.PL.in.orig	2023-10-09 01:37:25
+++ PerlMagick/default/Makefile.PL.in	2023-10-11 13:43:20
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../.. -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../.. -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ @MATH_LIBS@ -L' . $Config{'archlib'} . '/CORE';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
--- PerlMagick/quantum/Makefile.PL.in.orig	2023-10-09 01:37:25
+++ PerlMagick/quantum/Makefile.PL.in	2023-10-11 13:46:41
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../../ -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../../ -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ @MATH_LIBS@ -L' . $Config{'archlib'} . '/CORE';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
