Don't look in /usr/include/ImageMagick.
--- PerlMagick/Makefile.PL.in.orig	2019-11-29 12:39:22.000000000 -0600
+++ PerlMagick/Makefile.PL.in	2019-12-01 05:51:53.000000000 -0600
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../ -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../ -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ -lperl @MATH_LIBS@';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
--- PerlMagick/default/Makefile.PL.in.orig	2019-11-29 12:39:22.000000000 -0600
+++ PerlMagick/default/Makefile.PL.in	2019-12-01 05:52:00.000000000 -0600
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../.. -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../.. -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ -lperl @MATH_LIBS@';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
--- PerlMagick/quantum/Makefile.PL.in.orig	2019-11-29 12:39:22.000000000 -0600
+++ PerlMagick/quantum/Makefile.PL.in	2019-12-01 05:51:42.000000000 -0600
@@ -161,7 +161,7 @@
 }
 
 # defaults for LIBS & INC & CCFLAGS params that we later pass to Writemakefile
-my $INC_magick = '-I../../ -I@top_srcdir@ @CPPFLAGS@ -I"' . $Config{'usrinc'} . '/ImageMagick"';
+my $INC_magick = '-I../../ -I@top_srcdir@ @CPPFLAGS@';
 my $LIBS_magick = '-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ -lperl @MATH_LIBS@';
 my $CCFLAGS_magick = "$Config{'ccflags'} @CFLAGS@";
 my $LDFLAGS_magick   = "-L../../magick/.libs -lMagickCore-@MAGICK_MAJOR_VERSION@.@MAGICK_ABI_SUFFIX@ $Config{'ldflags'} @LDFLAGS@";
