--- netaddr/tests/eui/test_ieee_parsers.py.orig	2018-07-24 09:41:13.000000000 -0400
+++ netaddr/tests/eui/test_ieee_parsers.py	2018-07-24 09:42:08.000000000 -0400
@@ -12,7 +12,7 @@
 def test_oui_parser_py2():
     from cStringIO import StringIO
     outfile = StringIO()
-    with open(os.path.join(SAMPLE_DIR, 'sample_oui.txt')) as infile:
+    with open(os.path.join(SAMPLE_DIR, 'sample_oui.txt'), 'rb') as infile:
         iab_parser = OUIIndexParser(infile)
         iab_parser.attach(FileIndexer(outfile))
         iab_parser.parse()
@@ -23,7 +23,7 @@
 def test_iab_parser_py2():
     from cStringIO import StringIO
     outfile = StringIO()
-    with open(os.path.join(SAMPLE_DIR, 'sample_iab.txt')) as infile:
+    with open(os.path.join(SAMPLE_DIR, 'sample_iab.txt'), 'rb') as infile:
         iab_parser = IABIndexParser(infile)
         iab_parser.attach(FileIndexer(outfile))
         iab_parser.parse()
@@ -34,7 +34,7 @@
 def test_oui_parser_py3():
     from io import StringIO
     outfile = StringIO()
-    with open(os.path.join(SAMPLE_DIR, 'sample_oui.txt')) as infile:
+    with open(os.path.join(SAMPLE_DIR, 'sample_oui.txt'), 'rb') as infile:
         iab_parser = OUIIndexParser(infile)
         iab_parser.attach(FileIndexer(outfile))
         iab_parser.parse()
@@ -45,7 +45,7 @@
 def test_iab_parser_py3():
     from io import StringIO
     outfile = StringIO()
-    with open(os.path.join(SAMPLE_DIR, 'sample_iab.txt')) as infile:
+    with open(os.path.join(SAMPLE_DIR, 'sample_iab.txt'), 'rb') as infile:
         iab_parser = IABIndexParser(infile)
         iab_parser.attach(FileIndexer(outfile))
         iab_parser.parse()
