--- src/cmd/ksh93/sh/jobs.orig	2010-04-08 20:57:10.000000000 +0700
+++ src/cmd/ksh93/sh/jobs.c	2010-04-08 20:59:19.000000000 +0700
@@ -322,7 +322,9 @@
 #ifdef SIGTSTP
 		else
 			px=job_byjid(pw->p_job);
-		if(WIFSTOPPED(wstat))
+		if(WIFCONTINUED(wstat) && wcontinued)
+			pw->p_flag &= ~(P_NOTIFY|P_SIGNALLED|P_STOPPED);
+                else if(WIFSTOPPED(wstat))
 		{
 			if(px)
 			{
@@ -337,8 +339,6 @@
 				sh_fault(pw->p_exit); 
 			continue;
 		}
-		else if (WIFCONTINUED(wstat) && wcontinued)
-			pw->p_flag &= ~(P_NOTIFY|P_SIGNALLED|P_STOPPED);
 		else
 #endif /* SIGTSTP */
 		{
