--- Makefile.sdl.orig	2000-04-07 02:04:52.000000000 -0500
+++ Makefile.sdl	2011-08-18 16:58:06.000000000 -0500
@@ -4,7 +4,8 @@
 			newsnd.o ini.o input.o monster.o bags.o alpha.o vgagrafx.o \
 			title_gz.o icon.o sdl_kbd.o sdl_vid.o sdl_timer.o sdl_snd.o
 
-ARCH	= "LINUX"
+ARCH	= "DARWIN"
+#ARCH	= "LINUX"
 #ARCH	= "MINGW"
 #ARCH	= "FREEBSD"
 #ARCH	= "FooOS"
@@ -15,6 +16,12 @@
 ESUFFIX	=  .exe
 endif
 
+ifeq ($(ARCH),"DARWIN")
+OBJS   += fbsd_sup.o
+CFLAGS += -DDARWIN `sdl-config --cflags`
+LIBS   += `sdl-config --libs` -lz -lobjc
+endif
+
 ifeq ($(ARCH),"FREEBSD")
 OBJS	+= fbsd_sup.o	# strup()
 CFLAGS	+= -DFREEBSD `sdl-config --cflags`
