--- aux/broker/src/detail/sqlite_backend.cc.orig	2020-02-25 14:27:20.000000000 -0600
+++ aux/broker/src/detail/sqlite_backend.cc	2020-02-28 14:27:45.000000000 -0600
@@ -66,9 +66,9 @@
 
     // Initialize database.  SQLite has a bit of custom memory management
     // that seems to cause some LSANs to lose track and report a leak.
-    BROKER_LSAN_DISABLE();
+    //BROKER_LSAN_DISABLE();
     auto result = sqlite3_open(path.c_str(), &db);
-    BROKER_LSAN_ENABLE();
+    //BROKER_LSAN_ENABLE();
     if (result != SQLITE_OK) {
       sqlite3_close(db);
       BROKER_ERROR("failed to open database:" << path);
