--- configure.orig	1999-01-10 06:25:37.000000000 +1100
+++ configure	2011-04-21 18:35:22.000000000 +1000
@@ -1804,9 +1804,9 @@
 
 if test -n "$GXX"; then
     # GNU C compiler
-    CFLAGS="-Wall -O2"
+    CFLAGS+=" -Wall"
     CFLAGS_D="-Wall -g -DDEBUG"
-    CXXFLAGS="-Wall -O2"
+    CXXFLAGS+=" -Wall"
     CXXFLAGS_D="-Wall -g -DDEBUG"
 else
 if test "`(uname) 2>/dev/null`" = IRIX; then
@@ -1830,9 +1830,7 @@
   fi
 else
     # generic C compiler:
-    CFLAGS="-O"
     CFLAGS_D="-g -DDEBUG"
-    CXXFLAGS="-O"
     CXXFLAGS_D="-g -DDEBUG"
 fi
 fi
