--- configure.in.orig	2000-08-18 03:05:45.000000000 -0500
+++ configure.in	2011-04-21 18:35:22.000000000 +1000
@@ -16,9 +16,9 @@
 dnl# add warnings and optimization to compiler switches:
 if test -n "$GXX"; then
     # GNU C compiler
-    CFLAGS="-Wall -O2"
+    CFLAGS+=" -Wall"
     CFLAGS_D="-Wall -g -DDEBUG"
-    CXXFLAGS="-Wall -O2"
+    CXXFLAGS+=" -Wall"
     CXXFLAGS_D="-Wall -g -DDEBUG"
 else
 if test "`(uname) 2>/dev/null`" = IRIX; then
@@ -42,9 +42,7 @@
   fi
 else
     # generic C compiler:
-    CFLAGS="-O"
     CFLAGS_D="-g -DDEBUG"
-    CXXFLAGS="-O"
     CXXFLAGS_D="-g -DDEBUG"
 fi
 fi
