From 8db894690988f2bac9c0acb96fd39811d2b42f47 Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Date: Fri, 19 Feb 2010 00:05:28 -0600
Subject: [PATCH 07/37] Cygwin/X: Fix make dist after 11252ed82e1f361b99e86521ac9314f868bd1a3a

Signed-off-by: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Acked-by: Adam Jackson <ajax@redhat.com>
Reviewed-by: Keith Packard <keithp@keithp.com>
---
 hw/xwin/Makefile.am |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git xorg-server-1.7.99.902/hw/xwin/Makefile.am xorg-server-1.7.99.902/hw/xwin/Makefile.am
index fbaf092..daba679 100644
--- xorg-server-1.7.99.902/hw/xwin/Makefile.am
+++ xorg-server-1.7.99.902/hw/xwin/Makefile.am
@@ -197,6 +197,7 @@ include $(top_srcdir)/cpprules.in
 EXTRA_DIST = \
 	$(GLX_EXTRAS) \
 	$(MAN_SRCS) \
+	$(xwinconfig_DATA) \
 	X.ico \
 	XWin.rc \
 	xlaunch/config.cc \
-- 
1.7.0.2

