From 0ec78a0cda64453bba2d69abc0c779803a62ba50 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Fri, 3 Jun 2011 02:50:40 -0400
Subject: [PATCH 13/16] XQuartz: AIGLX: Remove unnecessary includes in
 indirect.c

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 hw/xquartz/GL/indirect.c |   32 +++++++++-----------------------
 1 files changed, 9 insertions(+), 23 deletions(-)

diff --git xorg-server-1.10.99.901/hw/xquartz/GL/indirect.c xorg-server-1.10.99.901/hw/xquartz/GL/indirect.c
index 40f1e93..ca25e73 100644
--- xorg-server-1.10.99.901/hw/xquartz/GL/indirect.c
+++ xorg-server-1.10.99.901/hw/xquartz/GL/indirect.c
@@ -35,48 +35,34 @@
 #include <dix-config.h>
 #endif
 
-#include "dri.h"
-
 #include <AvailabilityMacros.h>
 
 #include <dlfcn.h>
 
-#define GL_GLEXT_WUNDEF_SUPPORT
-
 #include <OpenGL/OpenGL.h>
-#include <OpenGL/CGLContext.h>
+#include <OpenGL/gl.h>     /* Just to prevent glxserver.h from loading mesa's and colliding with OpenGL.h */
 
+#include <X11/Xproto.h>
 #include <GL/glxproto.h>
-#include <windowstr.h>
-#include <resource.h>
-#include <GL/glxint.h>
-#include <GL/glxtokens.h>
-#include <scrnintstr.h>
+
 #include <glxserver.h>
-#include <glxscreens.h>
-#include <glxdrawable.h>
-#include <glxcontext.h>
-#include <glxext.h>
 #include <glxutil.h>
-#include <GL/internal/glcore.h>
-#include "x-hash.h"
-#include "x-list.h"
-
-//#include "capabilities.h"
-#include "visualConfigs.h"
 
 typedef unsigned long long GLuint64EXT;
 typedef long long GLint64EXT;
 #include <dispatch.h>
-#include <Xplugin.h>
 #include <glapi.h>
-#include <glapitable.h>
 
-__GLXprovider * GlxGetDRISWrastProvider (void);
+#include "x-hash.h"
+
+#include "visualConfigs.h"
+#include "dri.h"
 
 #include "darwin.h"
 #define GLAQUA_DEBUG_MSG(msg, args...) ASL_LOG(ASL_LEVEL_DEBUG, "GLXAqua", msg, ##args)
 
+__GLXprovider * GlxGetDRISWrastProvider (void);
+
 static void setup_dispatch_table(void);
 GLuint __glFloorLog2(GLuint val);
 void warn_func(void * p1, char *format, ...);
-- 
1.7.5.2

