From d00c3298a698ddddc8967215cef7dd941fa15804 Mon Sep 17 00:00:00 2001
From: Fernando Carrijo <fcarrijo@yahoo.com.br>
Date: Wed, 24 Mar 2010 20:11:34 -0300
Subject: [PATCH 30/37] Cleanup some comments in SpriteRec

Signed-off-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 include/inputstr.h |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

diff --git xorg-server-1.7.99.902/include/inputstr.h xorg-server-1.7.99.902/include/inputstr.h
index 15184d0..de4026c 100644
--- xorg-server-1.7.99.902/include/inputstr.h
+++ xorg-server-1.7.99.902/include/inputstr.h
@@ -386,8 +386,16 @@ typedef struct {
     int spriteTraceSize;
     int spriteTraceGood;
 
-    ScreenPtr pEnqueueScreen; /* screen events are being delivered to */
-    ScreenPtr pDequeueScreen; /* screen events are being dispatched to */
+    /* Due to delays between event generation and event processing, it is
+     * possible that the pointer has crossed screen boundaries between the
+     * time in which it begins generating events and the time when
+     * those events are processed.
+     *
+     * pEnqueueScreen: screen the pointer was on when the event was generated
+     * pDequeueScreen: screen the pointer was on when the event is processed
+     */
+    ScreenPtr pEnqueueScreen;
+    ScreenPtr pDequeueScreen;
 
 } SpriteRec, *SpritePtr;
 
-- 
1.7.0.2

