--- modules/pmg_tk/startup/apbs_tools.py.orig
+++ modules/pmg_tk/startup/apbs_tools.py
@@ -888,7 +888,7 @@
             pass
 
         if not psize_location:
-            psize_location = get_default_location('psize.py')
+            psize_location = get_default_location('apbs-psize.py')
         if not apbs_location:
             apbs_location = get_default_location('apbs.exe')
         if not apbs_location:
