--- layer1/Wizard.cpp.orig	2021-10-15 21:12:37.000000000 -0400
+++ layer1/Wizard.cpp	2021-10-15 21:14:04.000000000 -0400
@@ -153,9 +153,9 @@
 
 template<typename Func, typename... FuncArgs>
 auto WizardCallPython(PyMOLGlobals* G, PyObject* wiz, const char* funcName, Func&& func, FuncArgs&&... fargs)
-  -> pymol::result_of_t<Func(PyObject*, const char*, FuncArgs...)>
+  -> decltype(func(std::declval<PyObject*>(), std::declval<const char*>(), fargs...))
 {
-  using result_t = pymol::result_of_t<Func(PyObject*, const char*, FuncArgs...)>;
+  using result_t = decltype(func(std::declval<PyObject*>(), std::declval<const char*>(), fargs...));
   result_t result{};
   assert(wiz != nullptr);
   if (PyObject_HasAttrString(wiz, funcName)) {
