--- setup.py.orig	2021-10-07 16:20:45.000000000 -0400
+++ setup.py	2021-10-07 16:43:37.000000000 -0400
@@ -480,6 +480,10 @@
     if sys.platform == 'darwin':
         ext_comp_args += ["-fno-strict-aliasing"]

+# MacPorts; need to include the BOOST include directory since msgpack-cpp v4.x
+# needs those and we install them in a non-default location
+inc_dirs += ['@@BOOST_INC_DIR@@',]
+
 def get_pymol_version():
     return re.findall(r'_PyMOL_VERSION "(.*)"', open('layer0/Version.h').read())[0]

