Index: modules/pmg_tk/startup/apbs_tools.py
===================================================================
--- modules/pmg_tk/startup/apbs_tools.py	(revision 4187)
+++ modules/pmg_tk/startup/apbs_tools.py	(working copy)
@@ -169,7 +169,7 @@
 APBS_BINARY_LOCATION = None # corresponding environment variable: APBS_BINARY_DIR
 APBS_WEB_LOCATION = None # corresponding environment variable: APBS_WEB_DIR
 APBS_PSIZE_LOCATION = None # corresponding environment variable: APBS_PSIZE_DIR
-APBS_PDB2PQR_LOCATION = None # corresponding environment variable: APBS_PDB2PQR_DIR
+APBS_PDB2PQR_LOCATION = '@PREFIX@/share/pdb2pqr/' # corresponding environment variable: APBS_PDB2PQR_DIR
 TEMPORARY_FILE_DIR = tempfile.gettempdir() # corresponding environment variable: TEMP
 
 apbs_plea = ("IMPORTANT REQUEST: If you have not already done so, please register\n"
