Index: modules/pmg_tk/PMGApp.py
===================================================================
--- modules/pmg_tk/PMGApp.py	(revision 4106)
+++ modules/pmg_tk/PMGApp.py	(working copy)
@@ -210,7 +210,7 @@ class PMGApp(Pmw.MegaWidget):
         self.fifo.put(cmmd)
 
     def my_show(self,win,center=1):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.show()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -228,7 +228,7 @@ class PMGApp(Pmw.MegaWidget):
             win.deiconify()
 
     def my_withdraw(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.withdraw()
         else: 
             win.destroy()
Index: modules/pmg_tk/skins/normal/__init__.py
===================================================================
--- modules/pmg_tk/skins/normal/__init__.py	(revision 4106)
+++ modules/pmg_tk/skins/normal/__init__.py	(working copy)
@@ -271,7 +271,7 @@ class Normal(PMGSkin):
         self.buttonArea.destroy()
 
     def my_show(self,win,center=1):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.show()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -290,13 +290,13 @@ class Normal(PMGSkin):
 #         win.show()
             
     def my_withdraw(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.withdraw()
         else: 
             win.destroy()
 
     def my_activate(self,win,center=1,focus=None):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.activate()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -317,7 +317,7 @@ class Normal(PMGSkin):
             win.mainloop()
             
     def my_deactivate(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.deactivate()
         else: # autocenter, deiconify, and run mainloop
             win.destroy()
