--- /dev/null	2011-04-22 23:16:48.000000000 -0400
+++ setup/pymol_macports	2011-04-22 23:19:48.000000000 -0400
@@ -0,0 +1,16 @@
+#!/bin/sh
+#
+# PyMOL startup script
+#
+# ==============================================================
+# Set PYMOL_PATH to point at the correct location on your system
+#
+PYMOL_PATH=@@PYTHON_PKGDIR@@/pymol
+export PYMOL_PATH
+#
+# ==============================================================
+# (no changes usually required below here)
+#
+# python modules
+#
+exec @@PYTHON_BINARY@@ @@PYTHON_PKGDIR@@/pymol/__init__.py -d "_ set stereo_double_pump_mono,quiet=1" "$@"
