--- setup.py.orig	2011-04-22 22:47:12.000000000 -0400
+++ setup.py	2011-04-22 22:48:24.000000000 -0400
@@ -101,11 +101,10 @@
         #
         # REMEMEBER to use "./ext/bin/python ..."
         #
-        EXT = os.getcwd()+"/ext"
+        EXT = "@PREFIX@"
         inc_dirs=["ov/src",
                   "layer0","layer1","layer2",
                   "layer3","layer4","layer5", 
-                  "/usr/X11R6/include",
                   EXT+"/include",
                   EXT+"/include/GL",
                   EXT+"/include/freetype2",
@@ -116,13 +115,16 @@
         pyogl_libs = []
         lib_dirs=[]
         def_macros=[("_PYMOL_MODULE",None),
+                    ("_PYMOL_INLINE",None),
                     ("_PYMOL_LIBPNG",None),
                     ("_PYMOL_FREETYPE",None),
+                    ("_PYMOL_NUMPY",None),
+                    ("_PYMOL_OPENGL_SHADERS",None),
+                    ("NO_MMLIBS",None),
                     ]
-        ext_comp_args=[]
+        ext_comp_args=["-ffast-math","-funroll-loops","-O3"]
         ext_link_args=[
-	               "-L/usr/X11R6/lib", "-lGL", "-lXxf86vm",
-                   "-L"+EXT+"/lib", "-lpng", "-lglut", "-lfreetype"
+                   "-L"+EXT+"/lib", "-lpng", "-lGL", "-lglut", "-lGLEW", "-lfreetype"
                    ]
 #============================================================================
 else: # linux or other unix
@@ -181,6 +183,7 @@
                 'pymol/opengl/glu',
                 'pymol/opengl/glut',
                 'pymol/wizard',
+		'pymol2',
 		'web',
 		'web/examples',
 		'web/javascript', ],
