--- setup.py.orig	2018-07-26 09:40:50.000000000 -0400
+++ setup.py	2018-07-26 09:46:35.000000000 -0400
@@ -87,7 +87,7 @@
     X11 = ['/usr/X11'] * (not options.osx_frameworks)
 
     if sys.platform == 'darwin':
-        for prefix in ['/sw', '/opt/local', '/usr/local']:
+        for prefix in ['@PREFIX@', '/usr/local']:
             if sys.executable.startswith(prefix):
                 return [prefix] + X11
 
