Index: setup.py
===================================================================
--- setup.py	(revision 4106)
+++ setup.py	(working copy)
@@ -264,7 +264,7 @@ else: # unix style (linux, mac, ...)
     try:
         prefix_path = os.environ['PREFIX_PATH'].split(os.pathsep)
     except KeyError:
-        prefix_path = ["/usr", "/usr/X11", "/opt/local", "/sw"]
+        prefix_path = ["@PREFIX@", "/usr", "/usr/X11"]
 
     for prefix in prefix_path:
         inc_dirs += filter(os.path.isdir, [prefix + s for s in ["/include", "/include/freetype2", "/include/libxml2"]])
