--- pyproject.toml.orig	2026-07-22 23:05:03
+++ pyproject.toml	2026-07-23 06:33:42
@@ -43,7 +43,8 @@
 dash = ["dash", "typing-extensions"]
 streamlit = ["streamlit"]
 marimo = ["marimo"]
-other_dataframes = ["narwhals", "pyarrow", "modin[dask]"]
+# other_dataframes = ["narwhals", "pyarrow", "modin[dask]"]
+other_dataframes = ["narwhals", "pyarrow"]
 
 config = ["tomli;python_version<\"3.11\"", "platformdirs"]
 typing = ["typing_extensions;python_version<\"3.11\""]
@@ -51,7 +52,8 @@
 
 samples = ["pytz", "world_bank_data"]
 
-all = ["itables[config,pandas,polars,style,narwhals,other_dataframes,samples,notebook,widget,dash,shiny,streamlit,marimo,typing,check_type]"]
+# all = ["itables[config,pandas,polars,style,narwhals,other_dataframes,samples,notebook,widget,dash,shiny,streamlit,marimo,typing,check_type]"]
+all = ["itables[config,pandas,style,narwhals,other_dataframes,notebook,widget,marimo,typing,check_type]"]
 test-base = [
   "itables",
   # Pytest
@@ -97,7 +99,7 @@
 
 [tool.hatch.build.hooks.jupyter-builder]
 enable-by-default = true
-dependencies = ["hatch-jupyter-builder>=0.5.0", "dash==4.4.0", "pyyaml"]
+dependencies = ["hatch-jupyter-builder>=0.5.0", "dash>=4.4.0", "pyyaml"]
 build-function = "hatch_jupyter_builder.npm_builder"
 ensured-targets = [
     "src/itables/html/dt_bundle.js",
@@ -196,7 +198,7 @@
 
 
 [tool.pixi.feature.build.pypi-dependencies]
-dash = "==4.4.0"
+dash = ">=4.4.0"
 hatch = "<1.16"
 virtualenv = "<21"
 
