--- pyproject.toml.orig	2020-02-01 18:00:00
+++ pyproject.toml	2026-06-30 13:32:38
@@ -43,7 +43,8 @@
 dash = ["dash", "typing-extensions"]
 streamlit = ["streamlit"]
 marimo = ["marimo"]
-other_dataframes = ["narwhals", "pyarrow", "modin[dask]"]
+# other_dataframes = ["narwhals", "pyarrow", "modin[dask]"]
+other_dataframes = ["narwhals", "pyarrow"]
 
 config = ["tomli;python_version<\"3.11\"", "platformdirs"]
 typing = ["typing_extensions;python_version<\"3.11\""]
@@ -51,7 +52,8 @@
 
 samples = ["pytz", "world_bank_data"]
 
-all = ["itables[config,pandas,polars,style,narwhals,other_dataframes,samples,notebook,widget,dash,shiny,streamlit,marimo,typing,check_type]"]
+# all = ["itables[config,pandas,polars,style,narwhals,other_dataframes,samples,notebook,widget,dash,shiny,streamlit,marimo,typing,check_type]"]
+all = ["itables[config,pandas,style,narwhals,other_dataframes,notebook,widget,marimo,typing,check_type]"]
 test-base = [
   "itables",
   # Pytest
