diff --git CMakeLists.txt CMakeLists.txt
index e45dbf7c..59ac8d1d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -167,7 +167,7 @@ if(Launcher_LAYOUT_REAL STREQUAL "mac-bundle")
     set(BUNDLE_DEST_DIR ".")
 
     # Apps to bundle
-    set(APPS "\${CMAKE_INSTALL_PREFIX}/${Launcher_Name}.app")
+    set(APPS "\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${Launcher_Name}.app")
 
     # Mac bundle settings
     set(MACOSX_BUNDLE_BUNDLE_NAME "${Launcher_Name}")
@@ -182,6 +182,8 @@ if(Launcher_LAYOUT_REAL STREQUAL "mac-bundle")
     # directories to look for dependencies
     set(DIRS ${QT_LIBS_DIR} ${QT_LIBEXECS_DIR} ${CMAKE_LIBRARY_OUTPUT_DIRECTORY} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
 
+    SET(MultiMC_APP_BINARY_DEFS "-DMULTIMC_LINUX_DATADIR")
+
     # install as bundle
     set(INSTALL_BUNDLE "full")
 
diff --git launcher/CMakeLists.txt launcher/CMakeLists.txt
index 2dfc78b5..2a1c9f01 100644
--- launcher/CMakeLists.txt
+++ launcher/CMakeLists.txt
@@ -964,7 +964,7 @@ install(TARGETS ${Launcher_Name}
 if(INSTALL_BUNDLE STREQUAL "full")
     # Add qt.conf - this makes Qt stop looking for things outside the bundle
     install(
-        CODE "file(WRITE \"\${CMAKE_INSTALL_PREFIX}/${RESOURCES_DEST_DIR}/qt.conf\" \" \")"
+        CODE "file(WRITE \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${RESOURCES_DEST_DIR}/qt.conf\" \" \")"
         COMPONENT Runtime
     )
     # Bundle plugins
@@ -1041,10 +1041,4 @@ if(INSTALL_BUNDLE STREQUAL "full")
             )
         endif()
     endif()
-    configure_file(
-        "${CMAKE_CURRENT_SOURCE_DIR}/install_prereqs.cmake.in"
-        "${CMAKE_CURRENT_BINARY_DIR}/install_prereqs.cmake"
-        @ONLY
-    )
-    install(SCRIPT "${CMAKE_CURRENT_BINARY_DIR}/install_prereqs.cmake" COMPONENT Runtime)
 endif()
