--- pyproj/datadir.py.orig	2019-03-20 12:42:52.000000000 +0300
+++ pyproj/datadir.py	2019-03-20 12:46:37.000000000 +0300
@@ -35,9 +35,7 @@
 
     """
     global _USER_PROJ_DATA
-    internal_datadir = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)), "proj_dir", "share", "proj"
-    )
+    internal_datadir = "%%INTERNAL_DATADIR%%"
 
     def valid_data_dir(potential_data_dir):
         if potential_data_dir is not None and os.path.exists(
