--- configure.orig	2023-08-10 16:57:56.000000000 -0600
+++ configure	2023-09-24 08:05:13.000000000 -0600
@@ -11228,14 +11228,11 @@
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if shared Fortran libraries are supported" >&5
 printf %s "checking if shared Fortran libraries are supported... " >&6; }
   H5_FORTRAN_SHARED="yes"
-  ## tell libtool to do the right thing with COMMON symbols, this fixes
-  ## corrupt values with COMMON and EQUIVALENCE when building shared
-  ## Fortran libraries on OSX with gnu and Intel compilers (HDFFV-2772).
-  case "`uname`" in
-    Darwin*)
-    H5_LDFLAGS="$H5_LDFLAGS -Wl,-commons,use_dylibs"
-    ;;
-  esac
+
+## REMOVED HERE FOR MACPORTS, 2023 SEPT 24:
+## case Darwin ... H5_LDFLAGS="$H5_LDFLAGS -Wl,-commons,use_dylibs"
+## Xcode/CLT 15.0 does not accept "-commons".
+## See upstream PR:  https://github.com/HDFGroup/hdf5/pull/3581
 
   ## Report results of check(s)
 
