--- src/H5Fsuper.c.orig	2019-12-19 10:38:30.000000000 -0600
+++ src/H5Fsuper.c	2020-06-26 10:36:42.000000000 -0500
@@ -53,6 +53,7 @@
 /* Local Prototypes */
 /********************/
 static herr_t H5F__super_ext_create(H5F_t *f, H5O_loc_t *ext_ptr);
+static herr_t H5O__fsinfo_set_version(H5F_t *f, H5O_fsinfo_t *fsinfo);
 static herr_t H5F__update_super_ext_driver_msg(H5F_t *f);
 

--- src/H5Odeprec.c.orig	2019-12-19 10:38:30.000000000 -0600
+++ src/H5Odeprec.c	2020-06-26 10:45:12.000000000 -0500
@@ -54,7 +54,16 @@
 /********************/
 /* Local Prototypes */
 /********************/
-
+static herr_t H5CX_set_apl(hid_t *acspl_id, const H5P_libclass_t *libclass,
+    hid_t
+#ifndef H5_HAVE_PARALLEL
+   H5_ATTR_UNUSED
+#endif /* H5_HAVE_PARALLEL */
+   loc_id, hbool_t
+#ifndef H5_HAVE_PARALLEL
+   H5_ATTR_UNUSED
+#endif /* H5_HAVE_PARALLEL */
+   is_collective);
 
 /*********************/
 /* Package Variables */
--- src/H5Oint.c.orig	2019-12-19 10:38:30.000000000 -0600
+++ src/H5Oint.c	2020-06-26 11:04:56.000000000 -0500
@@ -74,6 +74,7 @@
 /* Local Prototypes */
 /********************/
 
+static herr_t H5CX_get_ohdr_flags(uint8_t *ohdr_flags);
 static herr_t H5O__delete_oh(H5F_t *f, H5O_t *oh);
 static herr_t H5O__obj_type_real(const H5O_t *oh, H5O_type_t *obj_type);
 static herr_t H5O__get_hdr_info_real(const H5O_t *oh, H5O_hdr_info_t *hdr);
--- src/H5Rint.c.orig	2019-12-19 10:38:30.000000000 -0600
+++ src/H5Rint.c	2020-06-26 11:07:46.000000000 -0500
@@ -46,6 +46,8 @@
 /********************/
 /* Local Prototypes */
 /********************/
+static herr_t H5CX_set_libver_bounds(H5F_t *f);
+
 
 /*********************/
 /* Package Variables */
