--- tools/src/misc/h5cc.in.orig	2017-04-26 06:45:02.000000000 +0900
+++ tools/src/misc/h5cc.in	2017-07-16 09:57:18.000000000 +0900
@@ -96,9 +96,6 @@
 
 CC="${HDF5_CC:-$CCBASE}"
 CLINKER="${HDF5_CLINKER:-$CLINKERBASE}"
-CFLAGS="${HDF5_CFLAGS:-$CFLAGSBASE}"
-CPPFLAGS="${HDF5_CPPFLAGS:-$CPPFLAGSBASE}"
-LDFLAGS="${HDF5_LDFLAGS:-$LDFLAGSBASE}"
 LIBS="${HDF5_LIBS:-$LIBSBASE}"
 
 # If a static library is available, the default will be to use it.  If the only
@@ -266,10 +263,12 @@
       ;;
     *\"*)
       qarg="'"$arg"'"
+      compile_args="$compile_args $qarg"
       allargs="$allargs $qarg"
       ;;
     *\'*)
       qarg='\"'"$arg"'\"'
+      compile_args="$compile_args $qarg"
       allargs="$allargs $qarg"
       ;;
     *)
