--- Makefile
+++ Makefile
@@ -1,6 +1,7 @@
 PREFIX=/usr/local
 PREFIX_MAN=$(PREFIX)/man
-CFLAGS?=-g -Wall
+DESTDIR                ?=
+CFLAGS?=-g -Wall -D st_mtim=st_mtimespec
 
 all: snac
 
@@ -18,14 +19,14 @@
 	$(CC) -I/usr/local/include -MM *.c > makefile.depend
 
 install:
-	mkdir -p -m 755 $(PREFIX)/bin
-	install -m 755 snac $(PREFIX)/bin/snac
-	mkdir -p -m 755 $(PREFIX_MAN)/man1
-	install -m 644 doc/snac.1 $(PREFIX_MAN)/man1/snac.1
-	mkdir -p -m 755 $(PREFIX_MAN)/man5
-	install -m 644 doc/snac.5 $(PREFIX_MAN)/man5/snac.5
-	mkdir -p -m 755 $(PREFIX_MAN)/man8
-	install -m 644 doc/snac.8 $(PREFIX_MAN)/man8/snac.8
+	mkdir -p -m 755 $(DESTDIR)$(PREFIX_MAN)/share/man1
+	install -m 755 snac $(DESTDIR)$(PREFIX)/bin/snac
+	mkdir -p -m 755 $(DESTDIR)$(PREFIX_MAN)/share/man1
+	install -m 644 doc/snac.1 $(DESTDIR)$(PREFIX_MAN)/share/man1/snac.1
+	mkdir -p -m 755 $(DESTDIR)$(PREFIX_MAN)/share/man5
+	install -m 644 doc/snac.5 $(DESTDIR)$(PREFIX_MAN)/share/man5/snac.5
+	mkdir -p -m 755 $(DESTDIR)$(PREFIX_MAN)/share/man8
+	install -m 644 doc/snac.8 $(DESTDIR)$(PREFIX_MAN)/share/man8/snac.8
 
 activitypub.o: activitypub.c xs.h xs_encdec.h xs_json.h xs_curl.h \
  xs_mime.h xs_openssl.h xs_regex.h xs_time.h snac.h
