--- Makefile.orig	2007-06-28 00:44:41.000000000 +0200
+++ Makefile	2007-06-28 19:51:54.000000000 +0200
@@ -1,14 +1,15 @@
 MAIN             = Main
-EXEC             = omega.exe
+EXEC             = omega
 GHC              = ghc
 GHCI             = ghci
-GHC_FLAGS        = -auto-all  $(GHC_FLAGS_COMMON)
+GHC_FLAGS        = -O -auto-all  $(GHC_FLAGS_COMMON)
 GHCI_FLAGS       = $(GHC_FLAGS_COMMON)
 GHC_FLAGS_COMMON = -fglasgow-exts -fallow-undecidable-instances
 HUGS             = hugs
 HUGS_FLAGS       = -98 -P:../Lib:../Lib/Parser
+DESTDIR          = 
 
-.PHONY: all clean
+.PHONY: all strip clean install
 
 all: *.hs
 	$(GHC) $(GHC_FLAGS) -o $(EXEC) --make $(MAIN)
@@ -22,8 +23,12 @@
 hugs-%: %.hs
 	$(HUGS) $(HUGS_FLAGS) $*
 
-clean :
+clean:
 	rm -f *.hi *.o *.prof
 
+strip: all
+	strip $(EXEC)
 
+install:
+	install $(EXEC) LangPrelude.prg $(DESTDIR)/bin
 
