--- fig2dev/dev/readpng.c.orig	2013-04-10 15:00:00.000000000 -0400
+++ fig2dev/dev/readpng.c	2013-04-10 15:01:25.000000000 -0400
@@ -62,7 +62,7 @@
     }
 
     /* set long jump here */
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
 	/* if we get here there was a problem reading the file */
 	png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
 	return 0;
@@ -78,15 +78,19 @@
     png_get_IHDR(png_ptr, info_ptr, &w, &h, &bit_depth, &color_type,
 	&interlace_type, &compression_type, &filter_type);
 
-    if (info_ptr->valid & PNG_INFO_gAMA)
-	png_set_gamma(png_ptr, 2.2, info_ptr->gamma);
+    double gamma = 0.45;
+    if (png_get_gAMA(png_ptr,info_ptr,&gamma))
+      png_set_gamma(png_ptr, 2.2, gamma);
     else
-	png_set_gamma(png_ptr, 2.2, 0.45);
+      png_set_gamma(png_ptr, 2.2, 0.45);
 
-    if (info_ptr->valid & PNG_INFO_bKGD)
+    if (png_get_valid(png_ptr,info_ptr,PNG_INFO_bKGD)) {
 	/* set the background to the one supplied */
-	png_set_background(png_ptr, &info_ptr->background,
+    	png_color_16p background;
+	png_get_bKGD(png_ptr,info_ptr,&background);
+	png_set_background(png_ptr, background,
 		PNG_BACKGROUND_GAMMA_FILE, 1, 1.0);
+    }
     else {
 	/* blend the canvas background using the alpha channel */
 	if (bgspec) {
