Index: kwin-styles/cde/Makefile.am
===================================================================
RCS file: /home/kde/kdeartwork/kwin-styles/cde/Makefile.am,v
retrieving revision 1.2
diff -u -b -u -r1.2 Makefile.am
--- kwin-styles/cde/Makefile.am	21 Feb 2002 10:57:12 -0000	1.2
+++ kwin-styles/cde/Makefile.am	17 Jun 2003 22:03:43 -0000
@@ -6,7 +6,7 @@
 kde_module_LTLIBRARIES = kwin_cde.la
 
 kwin_cde_la_SOURCES = cdeclient.cpp
-kwin_cde_la_LIBADD = $(kde_libraries)/kwin.la
+kwin_cde_la_LIBADD = $(kde_libraries)/libkwin_main.la
 kwin_cde_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 METASOURCES = AUTO
