Index: kwin-styles/kde1/Makefile.am
===================================================================
RCS file: /home/kde/kdeartwork/kwin-styles/kde1/Makefile.am,v
retrieving revision 1.4
diff -u -b -u -r1.4 Makefile.am
--- kwin-styles/kde1/Makefile.am	22 Jan 2003 18:55:52 -0000	1.4
+++ kwin-styles/kde1/Makefile.am	17 Jun 2003 22:04:17 -0000
@@ -4,7 +4,7 @@
 kde_module_LTLIBRARIES = kwin_kde1.la
 
 kwin_kde1_la_SOURCES = kde1client.cpp
-kwin_kde1_la_LIBADD = $(kde_libraries)/kwin.la
+kwin_kde1_la_LIBADD = $(kde_libraries)/libkwin_main.la
 kwin_kde1_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 METASOURCES = AUTO
