--- src/mainimpl.cpp.orig	2017-02-26 08:57:00.000000000 +0800
+++ src/mainimpl.cpp	2023-08-19 11:33:29.000000000 +0800
@@ -1734,7 +1734,7 @@
 	const QStringList &local_branches = revision_variables.value(REV_LOCAL_BRANCHES).toStringList();
 
 	if (!selected_name.isEmpty() &&
-	    local_branches.contains(selected_name) > 0 &&
+	    local_branches.contains(selected_name) &&
 	    selected_name != current_branch) {
 		// standard branch switching: directly checkout selected branch
 		rev = selected_name;
