--- meson.build.orig	2023-04-18 05:04:39.000000000 +0800
+++ meson.build	2023-04-18 05:05:00.000000000 +0800
@@ -524,6 +524,7 @@
     # Due to pervasive use of things like GPOINTER_TO_UINT(), we do not support
     # building with -Wbad-function-cast.
     '-Wno-bad-function-cast',
+    '-Wno-declaration-after-statement',
     '-Werror=implicit-function-declaration',
     '-Werror=missing-prototypes',
     '-Werror=pointer-sign',
