--- gio/gdbusprivate.c.orig	2010-10-20 16:08:38.000000000 -0700
+++ gio/gdbusprivate.c	2010-10-20 16:12:16.000000000 -0700
@@ -1750,12 +1750,12 @@
   gchar *ret;
   /* TODO: use PACKAGE_LOCALSTATEDIR ? */
   ret = NULL;
-  if (!g_file_get_contents ("/var/lib/dbus/machine-id",
+  if (!g_file_get_contents ("@@PREFIX@@/var/lib/dbus/machine-id",
                             &ret,
                             NULL,
                             error))
     {
-      g_prefix_error (error, _("Unable to load /var/lib/dbus/machine-id: "));
+      g_prefix_error (error, _("Unable to load @@PREFIX@@/var/lib/dbus/machine-id: "));
     }
   else
     {
