--- glib/gatomic.c	2007-10-16 00:44:49.000000000 -0500
+++ glib/gatomic.c	2007-10-31 01:30:21.000000000 -0500
@@ -281,7 +281,8 @@
 			       gint           val)
 {
   gint result, temp;
-  __asm__ __volatile__ (".Lieaa%=:       lwarx   %0,0,%3\n"
+  __asm__ __volatile__ ("nop\n"
+			".Lieaa%=:       lwarx   %0,0,%3\n"
 			"         add     %1,%0,%4\n"
 			"         stwcx.  %1,0,%3\n"
 			"         bne-    .Lieaa%="
@@ -297,7 +298,8 @@
 		  gint           val)
 {
   gint result, temp;  
-  __asm__ __volatile__ (".Lia%=:       lwarx   %0,0,%3\n"
+  __asm__ __volatile__ ("nop\n"
+			".Lia%=:       lwarx   %0,0,%3\n"
 			"         add     %1,%0,%4\n"
 			"         stwcx.  %1,0,%3\n"
 			"         bne-    .Lia%="
