--- glib-2.0.pc.in.orig 2008-07-01 17:30:18.000000000 -0500
+++ glib-2.0.pc.in 2008-07-02 14:53:01.000000000 -0500
@@ -11,4 +11,4 @@
 Description: C Utility Library
 Version: @VERSION@
 Libs: -L${libdir} -lglib-2.0 @INTLLIBS@ @ICONV_LIBS@
-Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include @GLIB_EXTRA_CFLAGS@
+Cflags: -I${includedir}/glib-2.0 -I${libdir}/glib-2.0/include -I${includedir} @GLIB_EXTRA_CFLAGS@
