--- gstreamer/devicemanager.cpp.orig
+++ gstreamer/devicemanager.cpp
@@ -355,7 +355,7 @@ AbstractRenderer *DeviceManager::createVideoRenderer(VideoWidget *parent)
     if (m_videoSinkWidget == "software") {
         return new WidgetRenderer(parent);
     }
-#ifndef Q_WS_QWS
+#if !defined(Q_WS_QWS) && !defined(Q_OS_MAC)
     else if (m_videoSinkWidget == "xwindow") {
         return new X11Renderer(parent);
     } else {
