--- modules/radldap/radldap.c	2022-01-15 09:09:00.000000000 -0500
+++ modules/radldap/radldap.c	2022-01-15 09:09:55.000000000 -0500
@@ -69,6 +69,7 @@
 #include <sys/time.h>	/* For struct timeval on some systems */
 #include <time.h>	/* For struct timeval on some systems */
 
+#include <limits.h>
 #include <stdio.h>	/* For fprintf() */
 #include <signal.h>	/* For signal() to ignore SIGPIPE during LDAP unbind */
 #include <unistd.h>	/* For getopt() on some systems */
