--- lib/CL/devices/CMakeLists.txt.orig
+++ lib/CL/devices/CMakeLists.txt
@@ -27,6 +27,9 @@
   function(add_pocl_device_library name)
     add_library(${name} SHARED ${ARGN})
     harden("${name}")
+    if(APPLE)
+      set_target_properties(${name} PROPERTIES INSTALL_NAME_DIR "@rpath")
+    endif()
     if(OCL_ICD_INCLUDE_DIRS)
       target_include_directories(${name} SYSTEM PUBLIC ${OCL_ICD_INCLUDE_DIRS})
     endif()
