--- Makefile.orig	2015-10-03 19:37:52.000000000 +0200
+++ Makefile	2015-10-03 19:39:01.000000000 +0200
@@ -19,25 +19,14 @@
   LDFLAGS+=-pg
 endif
 
-ifeq ($(shell pkg-config --atleast-version=0.1.0 unibilium && echo 1),1)
-  CFLAGS +=$(shell pkg-config --cflags unibilium) -DHAVE_UNIBILIUM
-  LDFLAGS+=$(shell pkg-config --libs   unibilium)
-else ifeq ($(shell pkg-config ncursesw && echo 1),1)
-  CFLAGS +=$(shell pkg-config --cflags ncursesw)
-  LDFLAGS+=$(shell pkg-config --libs   ncursesw)
-else
-  LDFLAGS+=-lncurses
-endif
+CFLAGS +=$(shell pkg-config --cflags ncursesw)
+LDFLAGS+=$(shell pkg-config --libs   ncursesw)
 
 OBJECTS=termkey.lo driver-csi.lo driver-ti.lo
 LIBRARY=libtermkey.la
 
 DEMOS=demo demo-async
 
-ifeq ($(shell pkg-config glib-2.0 && echo 1),1)
-  DEMOS+=demo-glib
-endif
-
 DEMO_OBJECTS=$(DEMOS:=.lo)
 
 TESTSOURCES=$(wildcard t/[0-9]*.c)
