--- build_tools/build_detect_platform.orig	2020-10-17 18:33:19.000000000 +0300
+++ build_tools/build_detect_platform	2020-10-17 18:34:58.000000000 +0300
@@ -652,9 +652,9 @@
 
   if [[ "${PLATFORM}" == "OS_MACOSX" ]]; then
     # For portability compile for macOS 10.12 (2016) or newer
-    COMMON_FLAGS="$COMMON_FLAGS -mmacosx-version-min=10.12"
-    PLATFORM_LDFLAGS="$PLATFORM_LDFLAGS -mmacosx-version-min=10.12"
-    PLATFORM_SHARED_LDFLAGS="$PLATFORM_SHARED_LDFLAGS -mmacosx-version-min=10.12"
+    COMMON_FLAGS="-mmacosx-version-min=10.12 $COMMON_FLAGS"
+    PLATFORM_LDFLAGS="-mmacosx-version-min=10.12 $PLATFORM_LDFLAGS"
+    PLATFORM_SHARED_LDFLAGS="-mmacosx-version-min=10.12 $PLATFORM_SHARED_LDFLAGS"
   fi
 fi
 
