--- Makefile.in.orig	2008-08-09 12:51:09.000000000 +0200
+++ Makefile.in	2008-08-09 12:51:42.000000000 +0200
@@ -3,9 +3,9 @@
 
 prefix		=	@prefix@
 exec_prefix	=	@exec_prefix@
-INST_PREFIX	=	@exec_prefix@
-INST_LIB	=	@libdir@
-INST_BIN	=	@bindir@
+INST_PREFIX	=	${DESTDIR}@exec_prefix@
+INST_LIB	=	${DESTDIR}@libdir@
+INST_BIN	=	${DESTDIR}@bindir@
 
 #
 # MS_VERSION - Set by configure based in mapserver.h
