--- renderers/agg/include/agg_renderer_outline_aa.h.orig	2012-04-13 13:16:34.000000000 -0600
+++ renderers/agg/include/agg_renderer_outline_aa.h	2012-04-13 13:17:21.000000000 -0600
@@ -1365,7 +1365,6 @@
         //---------------------------------------------------------------------
         void profile(const line_profile_aa& prof) { m_profile = &prof; }
         const line_profile_aa& profile() const { return *m_profile; }
-        line_profile_aa& profile() { return *m_profile; }
 
         //---------------------------------------------------------------------
         int subpixel_width() const { return m_profile->subpixel_width(); }
