--- src/Appender.cpp.orig	2017-04-18 00:50:54.000000000 -0500
+++ src/Appender.cpp	2017-04-27 21:56:23.000000000 -0500
@@ -18,11 +18,11 @@
 
 	Appender::AppenderMapStorage::AppenderMapStorage()  { 
 		_allAppenders = new AppenderMap(); 
-	};
+	}
 	Appender::AppenderMapStorage::~AppenderMapStorage() { 
 		_deleteAllAppenders(); 
 		delete _allAppenders; 
-	};
+	}
 	
 	Appender::AppenderMapStorageInitializer::AppenderMapStorageInitializer() {
 		 if (appenders_nifty_counter++ == 0) {
