--- tools/meson.build.orig	2022-02-05 15:05:51.000000000 -0500
+++ tools/meson.build	2022-02-05 15:07:11.000000000 -0500
@@ -21,7 +21,7 @@
   tools_conf.set('libdir', libdir_abs)
   tools_conf.set('datarootdir', datadir_abs)
   tools_conf.set('gir_dir_prefix', gir_dir_prefix)
-  tools_conf.set('PYTHON_CMD', python_cmd)
+  tools_conf.set('PYTHON_CMD', '@MP_PYTHON_CMD@')
 
   tools_conf.set('TOOL_MODULE', tool[1])
   tools_conf.set('TOOL_FUNCTION', tool[2])
