--- ./src/gui_main.c.orig 2008-05-23 15:54:28.000000000 +0100
+++ ./src/gui_main.c 2012-10-09 15:29:22.195801538 +0100
@@ -456,7 +456,12 @@

     if (res)
     {
-        gui_tree_load_and_display(res);
+        char *res_realpath = realpath(res, NULL);
+        if (res_realpath)
+        {
+            gui_tree_load_and_display(res_realpath);
+            free(res_realpath);
+        }
         g_free(res);
     }
 }
