--- CMakeLists.txt.orig	2012-10-16 06:45:36.000000000 +0900
+++ CMakeLists.txt	2012-10-20 20:11:32.000000000 +0900
@@ -47,7 +47,6 @@
 
     if(HAVE_UNORDERED_MAP)
         message(STATUS "Enabling c++0x support for unordered map")
-        add_definitions( -std=c++0x ) # For unordered_map
     else(HAVE_UNORDERED_MAP)
         check_include_file_cxx(ext/hash_map HAVE_EXT_HASH_MAP)
     endif(HAVE_UNORDERED_MAP)
