--- DESCRIPTION	2024-04-20 10:20:03.000000000 +0800
+++ DESCRIPTION	2024-04-21 19:11:30.000000000 +0800
@@ -14,7 +14,7 @@
 Depends: R (>= 3.6)
 Imports: MASS, tibble, dplyr, stats, utils, grDevices
 Suggests: aplpack, ash, energy, testthat, colorspace, ggplot2, mgcv,
-        gifski, geozoo, knitr, rmarkdown, tidyr, covr, plotly
+        geozoo, knitr, rmarkdown, tidyr, covr, plotly
 License: MIT + file LICENSE
 LazyData: true
 URL: https://github.com/ggobi/tourr
