--- /Users/takanori/Sources/ptetex3-20080616.orig/md5sum/texsrc.international	2008-06-16 15:02:43.000000000 +0900
+++ ./md5sum/texsrc.international	2009-03-23 00:25:30.000000000 +0900
@@ -72,8 +72,8 @@
 aa8bcb252fdfab1e6164e338b6a791ce  ./texk/dvipdfm/ebb.c
 c98e0303e74020945a281afd04b09182  ./texk/dvipdfm/pdfdoc.c
 31b2da812b271b30405acafb8af18e6b  ./texk/dvipdfm/psimage.c
-bba5e84b40c0d8f1b194dcdc95acb301  ./texk/dvipdfmx/data/Makefile.in
-c5eb635010addfee69655b5bd5a089e7  ./texk/dvipdfmx/data/config/dvipdfmx.cfg
+3abe611cc8205c4ba0b555c8ef04a7fb  ./texk/dvipdfmx/data/Makefile.in
+fb752e6f178ca874204114d7ecf57a20  ./texk/dvipdfmx/data/dvipdfmx.cfg
 383ef540698559ab73f11dd317f111af  ./texk/kpathsea/BUGS
 2e4745385a79e7e3222b1a2d13692d6d  ./texk/kpathsea/ChangeLog
 30a590b862bf12903eb308597cc8bf2d  ./texk/kpathsea/INSTALL
@@ -241,7 +241,7 @@
 df9d869cd778ef524a6d110d3f1185ec  ./texk/pdvipsk/ChangeLog
 49b3e7400dad7585a1f739a16d10528f  ./texk/pdvipsk/Makefile.in
 4a650475d99b54df8aa76781aaa002d2  ./texk/pdvipsk/Makefile.in.jp
-61ebec113e202414044a7e84a192d853  ./texk/pdvipsk/afm2tfm.c
+d4a5ee5e44a3fe87edf713fa860cca77  ./texk/pdvipsk/afm2tfm.c
 0f0f1568ca6cef6e9193c9b42000747a  ./texk/pdvipsk/bbox.c
 5318a19670eba1017248daa85628977b  ./texk/pdvipsk/color.c
 c2a46ffac1534d7767f869c375bb9045  ./texk/pdvipsk/color.lpro
@@ -250,8 +250,8 @@
 28ea2b6d495b6fc506a8739a6a9b4671  ./texk/pdvipsk/dospecial.c
 76cce1c3de62281469971a3e0304329b  ./texk/pdvipsk/download.c
 727a5cb6d89cc7f3fb06e4b6377c011b  ./texk/pdvipsk/drawPS.c
-ede32de2d780924635576382fafb4d63  ./texk/pdvipsk/dvips.c
-6e011a7853ad76f76e8b34f562b5b370  ./texk/pdvipsk/dvips.h
+b4e9c675bd1b2da75cd5afdfb421e793  ./texk/pdvipsk/dvips.c
+6a9394f88aa88c83891e7e0c049acd83  ./texk/pdvipsk/dvips.h
 5228ab9569471530c6e29a916f41d41e  ./texk/pdvipsk/dvips.help
 75285d25caca5d58a1071e86984120e1  ./texk/pdvipsk/dvips.texi
 c4dfb1a03f687dec114b9340274e0195  ./texk/pdvipsk/emspecial.c
@@ -434,7 +434,7 @@
 b7d49ea4978fd1d335a1cec3a0215f9d  ./texk/pxdvik/zeit.c
 57d7ad06f38d6e563770af5405c0abde  ./texk/pxdvik/zeit.h
 10f1f18d90efc226d0db46dc60132752  ./texk/tetex/allcm
-bccacaaf20b5c7d761700101edc3a2de  ./texk/tetex/epstopdf
+f00759fa81cfb55dafed739b0c6ea685  ./texk/tetex/epstopdf
 ebc6b3bdfb294b144444a826cd044442  ./texk/tetex/fmtutil
 5641fcc891193155d3c256de05207f73  ./texk/tetex/texconfig
 c87a7653e7ff3d98faa26d5f6995c6f0  ./texk/tetex/texlinks
