--- ./8test.sh.orig	2008-04-05 17:47:44.000000000 +0900
+++ ./8test.sh	2008-04-13 06:53:32.000000000 +0900
@@ -191,58 +191,6 @@
 ex $PXDVI   babeltestJIS.dvi
 fi
 
-# -dNOKANJI check for dvipdfmx
-echo "-----------------------------"
-echo "[-dNOKANJI] test"
-cat <<EOF > dNOKANJI.tex
-\documentclass[43pt]{jsarticle}
-\setlength{\headsep}{-3em}
-\usepackage{graphicx}
-\begin{document}
-\noindent {\tiny If char and EPS looks similar, OK.}
-
-\noindent
-Char: {\Huge ¤Ñ¤Ô¤×} \newline
-\raisebox{3em}{EPS:~}
-\includegraphics[clip,width=0.7\linewidth]{`pwd`/dNOKANJI.eps}
-\end{document}
-EOF
-
-cat <<EOF > dNOKANJI.eps
-%!PS-Adobe-2.0
-%%BoundingBox: 20 100 330 308
-%%BeginProlog
-/OurCharSet findfont 12 scalefont setfont
-(a) stringwidth pop /mp_a_x exch def
-%%EndProlog
-save
-gsave
-560 11 mp_a_x mul div 306 18 div scale
-4 7 translate 552 560 div 16 36 div scale
-/Ryumin-Light-H findfont 14 scalefont setfont
-0 13 moveto <2451> show
-2 mp_a_x mul 13 moveto <2454> show
-4 mp_a_x mul 13 moveto <2457> show
-0 0 moveto <245a> show
-2 mp_a_x mul 0 moveto <245d> show
-4 mp_a_x mul 0 moveto <396c> show
-grestore
-restore
-showpage
-EOF
-
-nkfjis dNOKANJI.tex > /dev/null
-$BIN/platex dNOKANJI.tex < /dev/null > /dev/null || catexit dNOKANJI.log
-ex $PDVIPS  dNOKANJI.dvi
-ex $PDVIPDF dNOKANJI.dvi
-if test `wc -c < dNOKANJI.pdf` -gt 10000; then
-    cat <<EOF
-Warning: 'gs' seem to embed Japanese fonts in spite of '-dNOKANJI' option.
-Check gs options in $TEXMF/dvipdfm/config/dvipdfmx.cfg.
-We may need some additional patches for ghostscript.
-EOF
-fi
-
 # CVE-2007-0104 check for pdflatex
 echo "-----------------------------"
 echo "[CVE-2007-0104] test"
