--- ./3extract-texmf.sh.orig	2008-06-16 14:43:35.000000000 +0900
+++ ./3extract-texmf.sh	2009-06-11 03:57:51.000000000 +0900
@@ -18,14 +18,16 @@
     chmod -x $TEXMF-dist/scripts/*/*.pl || exit
 fi
 
+cpatch archive/latex-timebomb.diff 0 $TEXMF-dist
+
 # update config.ps with TeX Live 2007 svn (10/25)
 $MKDIR $TEXMF/dvips/config
 $CP archive/config.ps $TEXMF/dvips/config/config.ps
 
 ## for dvips (pstricks)
-tarx archive/dvipsk-5.96-p1.7b.tar.gz $PTETEX_TMP
+tarx archive/dvipsk-5.97-p1.7b-20090509.tar.gz $PTETEX_TMP
 distcp $TEXMF dvips/pstricks pst-text.pro
-cpatch $PTETEX_TMP/PSTricks.patch 0 $TEXMF/dvips/pstricks
+cpatch archive/PSTricks.patch 0 $TEXMF/dvips/pstricks
 
 
 if test "$JAPANESE" != "without"; then
@@ -157,6 +159,56 @@
 distcp $TEXMF tex/latex/marvosym marvosym.sty
 cpatch archive/tetex-3.0-marvosym-rightarrow.patch 2 $TEXMF
 
+# update txfonts from v3.2 to v3.2.1
+$RM -r $TEXMF-dist/doc/fonts/txfonts \
+       $TEXMF-dist/fonts/afm/public/txfonts \
+       $TEXMF-dist/fonts/map/dvips/tetex/txfonts.map \
+       $TEXMF-dist/fonts/tfm/public/txfonts \
+       $TEXMF-dist/fonts/type1/public/txfonts \
+       $TEXMF-dist/fonts/vf/public/txfonts \
+       $TEXMF-dist/tex/latex/txfonts || exit
+$MKDIR $TEXMF-dist/doc/fonts/txfonts \
+       $TEXMF-dist/fonts/afm/public/txfonts \
+       $TEXMF-dist/fonts/enc/dvips/txfonts \
+       $TEXMF-dist/fonts/map/dvips/txfonts \
+       $TEXMF-dist/fonts/tfm/public/txfonts \
+       $TEXMF-dist/fonts/type1/public/txfonts \
+       $TEXMF-dist/fonts/vf/public/txfonts \
+       $TEXMF-dist/tex/latex/txfonts || exit
+tarx archive/txfonts-3.2.1.zip $PTETEX_TMP
+$CP $PTETEX_TMP/txfonts/doc/* $PTETEX_TMP/txfonts/00bug_fix.txt $PTETEX_TMP/txfonts/COPYRIGHT $PTETEX_TMP/txfonts/README $TEXMF-dist/doc/fonts/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/afm/* $TEXMF-dist/fonts/afm/public/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/dvips/*.enc $TEXMF-dist/fonts/enc/dvips/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/dvipdfm/*.map $PTETEX_TMP/txfonts/dvips/*.map $TEXMF-dist/fonts/map/dvips/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/tfm/* $TEXMF-dist/fonts/tfm/public/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/pfb/* $TEXMF-dist/fonts/type1/public/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/vf/* $TEXMF-dist/fonts/vf/public/txfonts/ || exit
+$CP $PTETEX_TMP/txfonts/input/* $TEXMF-dist/tex/latex/txfonts/ || exit
+
+# update pxfonts from v1.1 to v1.1.1
+$RM -r $TEXMF-dist/doc/fonts/pxfonts \
+       $TEXMF-dist/fonts/afm/public/pxfonts \
+       $TEXMF-dist/fonts/map/dvips/tetex/pxfonts.map \
+       $TEXMF-dist/fonts/tfm/public/pxfonts \
+       $TEXMF-dist/fonts/type1/public/pxfonts \
+       $TEXMF-dist/fonts/vf/public/pxfonts \
+       $TEXMF-dist/tex/latex/pxfonts || exit
+$MKDIR $TEXMF-dist/doc/fonts/pxfonts \
+       $TEXMF-dist/fonts/afm/public/pxfonts \
+       $TEXMF-dist/fonts/map/dvips/pxfonts \
+       $TEXMF-dist/fonts/tfm/public/pxfonts \
+       $TEXMF-dist/fonts/type1/public/pxfonts \
+       $TEXMF-dist/fonts/vf/public/pxfonts \
+       $TEXMF-dist/tex/latex/pxfonts || exit
+tarx archive/pxfonts-1.1.1.zip $PTETEX_TMP
+$CP $PTETEX_TMP/pxfonts/doc/* $PTETEX_TMP/pxfonts/00bug_fix.txt $PTETEX_TMP/pxfonts/COPYRIGHT $TEXMF-dist/doc/fonts/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/afm/* $TEXMF-dist/fonts/afm/public/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/dvipdfm/*.map $PTETEX_TMP/pxfonts/dvips/*.map $TEXMF-dist/fonts/map/dvips/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/tfm/* $TEXMF-dist/fonts/tfm/public/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/pfb/* $TEXMF-dist/fonts/type1/public/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/vf/* $TEXMF-dist/fonts/vf/public/pxfonts/ || exit
+$CP $PTETEX_TMP/pxfonts/input/* $TEXMF-dist/tex/latex/pxfonts/ || exit
+
 ## post
 $RM -r $PTETEX_TMP || exit
 (cd $TEXMF || exit
