--- ./1check-archive.sh.orig	2009-05-10 21:14:23.000000000 +0900
+++ ./1check-archive.sh	2009-05-10 21:13:08.000000000 +0900
@@ -103,9 +103,9 @@
 fi
 md5check <<EOF || exit
 92579eca220d2279787ddb98b5b1e439  archive/config.ps
-3429510a849f0736ffa78ce6a5b5a3a7  archive/dvipdfmx-20070409.tar.gz
+c727370d57633eb58618022516d17b07  archive/dvipdfmx-20090708.tar.gz
 92f5fcdf523381c6e30288d0e2c487ab  archive/dvipdfmx-contrib-latex.tar.gz
-02e0b121e2c391efffb46d6485d7899f  archive/dvipsk-5.96-p1.7b.tar.gz
+0c7e46e0562ee2debcf93e313b70e7ac  archive/dvipsk-5.97-p1.7b-20090509.tar.gz
 a7aefa88799df081ca981ce6aa32ec90  archive/freetype-2.3.5.tar.gz
 7d688306677b99ec34cf68ef93562381  archive/jcode.pl
 12fc14c4ed0530eab018c867b383eedb  archive/jis.tar.gz
@@ -147,7 +147,7 @@
 55daff53c4e52be78025dcd0785bcfc3  macro/burasage.lzh
 40f757c7c16bce50915893170ed9b916  macro/eclepsf.sty
 601835d441df357d28f93c4d4534dbe0  macro/epsbox.sty
-bac445d00f782ed55efdfb0380dbb303  macro/jsclasses-071024.zip
+eab726afd7249145578df56daa3d805f  macro/jsclasses-090222.zip
 4c6224bbdc50345999714c32958eb186  macro/otfstable-speedup.patch
 0fd496d0e9b365457e161d18da9dfffb  macro/otfstable.zip
 50382fd31d95532cbc3d6803bc882578  macro/powerdot.tar.gz
