--- ./2extract-src.sh.orig	2008-06-16 13:50:47.000000000 +0900
+++ ./2extract-src.sh	2009-02-25 03:39:56.000000000 +0900
@@ -263,7 +263,11 @@
 
 
 ## check
-cat md5sum/texsrc.$JAPANESE | (cd $SRC; md5check) && exit
+if cat md5sum/texsrc.$JAPANESE | (cd $SRC; md5check); then
+    cpatch archive/dvipng-tiger-#18254.diff        1 $SRC
+    cpatch archive/xdvik-tiger-#18254.diff         1 $SRC
+    exit
+fi
 cat <<EOF
 
 $0: Some processes were performed improperly.
