--- lib/libc-config.h.orig	2020-01-08 20:16:36.000000000 -0800
+++ lib/libc-config.h	2020-01-08 20:17:29.000000000 -0800
@@ -75,7 +75,11 @@
    use __WORDSIZE now, and if Gnulib uses it later the syntax error
    will let us know that __WORDSIZE needs configuring.  */
 #ifndef __WORDSIZE
-# define __WORDSIZE %%%
+#if __LP64__
+#define __WORDSIZE 64
+#else
+#define __WORDSIZE 32
+#endif
 #endif
 /* Undef the macros unconditionally defined by our copy of glibc
    <sys/cdefs.h>, so that they do not clash with any system-defined
