--- setup.py.orig	2014-11-07 01:26:17.000000000 +0300
+++ setup.py	2014-11-07 01:49:53.000000000 +0300
@@ -88,7 +88,9 @@
 # Generate code for build
 #===------------------------------------------------------------------===
 
-if build.have_clang():
+config = build.mkconfig(build.default_config)
+
+if build.have_clang(config.clang):
     # Build llvm asm
     targets = [build.build_llvm]
 else:
@@ -97,7 +99,7 @@
                  "default C compiler")
     targets = []
 
-config = build.mkconfig(build.default_config, targets=targets)
+config = build.mkconfig(config, targets=targets)
 llvmmath.build.build_source(config)
 try:
     llvmmath.build.build_targets(config)
