--- src/libaccountsservice/Makefile.am.orig	2014-03-06 10:51:34.000000000 -0800
+++ src/libaccountsservice/Makefile.am	2014-03-06 10:52:09.000000000 -0800
@@ -82,7 +82,6 @@
 libaccountsservice_la_LIBADD =                                                \
 	../libaccounts-generated.la						\
         $(LIBACCOUNTSSERVICE_LIBS)                                            \
-        -lcrypt                                                               \
         $(END_OF_LIST)
 
 libaccountsservice_la_sources =                                               \
