--- lib/update.js.orig	2015-09-11 02:47:10.000000000 +0200
+++ lib/update.js	2015-10-31 00:33:55.000000000 +0100
@@ -46,6 +46,14 @@
       // use the initial installation method (repo, tar, git) for updating
       if (url.parse(ww.req).protocol) ww.what = ww.req
 
+      if (what.match(/^npm@/) && where.match(/@@NPM_PATH_JSREGEX@@/)) {
+        log.error("Trying to update '" + what + "' in '" + where + "'")
+        log.error("which is part of the MacPorts npm base installation.")
+        log.error("To update npm please run:")
+        log.error("sudo port selfupdate && sudo port upgrade npm\n")
+        return cb()
+      }
+
       var where = ww.dep.parent && ww.dep.parent.path || ww.dep.path
       if (toInstall[where]) {
         toInstall[where].push(ww.what)
