--- lib/Xm/DataF.c.orig	Tue Nov 14 17:30:37 2006
+++ lib/Xm/DataF.c	Tue Nov 14 17:31:40 2006
@@ -58,6 +58,8 @@
 #include "ImageCachI.h"
 
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #include "XmRenderTI.h"
 #endif
--- lib/Xm/TabBox.c.orig	Tue Nov 14 17:31:55 2006
+++ lib/Xm/TabBox.c	Tue Nov 14 17:32:25 2006
@@ -26,6 +26,8 @@
 #include <Xm/DrawUtils.h>
 
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #endif
 
--- lib/Xm/TextF.c.orig	Tue Nov 14 17:32:38 2006
+++ lib/Xm/TextF.c	Tue Nov 14 17:32:59 2006
@@ -89,6 +89,8 @@
 #include "XmStringI.h"
 #include <Xm/PrintSP.h>         /* for XmIsPrintShell */
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #include "XmRenderTI.h"
 #endif
--- lib/Xm/TextFP.h.orig	Tue Nov 14 17:33:34 2006
+++ lib/Xm/TextFP.h	Tue Nov 14 17:33:52 2006
@@ -44,6 +44,8 @@
 #include <Xm/PrimitiveP.h>
 #include <Xm/TextF.h>
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #endif
 
--- lib/Xm/TextIn.c.orig	Tue Nov 14 17:13:39 2006
+++ lib/Xm/TextIn.c	Tue Nov 14 17:22:32 2006
@@ -88,6 +88,8 @@
 #include "TravActI.h"
 #include "TraversalI.h"
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #endif
 
--- lib/Xm/XmRenderT.c.orig	Tue Nov 14 17:34:21 2006
+++ lib/Xm/XmRenderT.c	Tue Nov 14 17:43:35 2006
@@ -75,6 +75,8 @@
 #include "XmStringI.h"
 #include "XmTabListI.h"
 #ifdef USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #endif
 
--- lib/Xm/XmRenderTI.h.orig	Tue Nov 14 17:29:53 2006
+++ lib/Xm/XmRenderTI.h	Tue Nov 14 16:49:05 2006
@@ -43,6 +43,8 @@
 
 #include <Xm/XmP.h>
 #ifdef	USE_XFT
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <X11/Xft/Xft.h>
 #endif
 
