--- autogen.sh.orig	2012-10-23 01:50:34.000000000 +1100
+++ autogen.sh	2012-11-03 18:11:51.000000000 +1100
@@ -7,18 +7,10 @@
 cd "$srcdir"
 DIE=0
 
-libtoolize --force --automake
+glibtoolize --force --automake
 aclocal -I .
 autoconf
 autoheader
 automake --foreign  --include-deps --add-missing
 
-if test -z "$*"; then
-        echo "I am going to run ./configure with no arguments - if you wish "
-        echo "to pass any to it, please specify them on the $0 command line."
-fi
-
-cd "$THEDIR"
-
-$srcdir/configure "$@"
 
