--- ./gnome-session/main.c.orig	2012-01-17 19:30:28.000000000 -0500
+++ ./gnome-session/main.c	2012-01-17 19:31:34.000000000 -0500
@@ -442,11 +442,6 @@
                 { NULL, 0, 0, 0, NULL, NULL, NULL }
         };
 
-        /* Make sure that we have a session bus */
-        if (!require_dbus_session (argc, argv, &error)) {
-                gsm_util_init_error (TRUE, "%s", error->message);
-        }
-
         bindtextdomain (GETTEXT_PACKAGE, LOCALE_DIR);
         bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
         textdomain (GETTEXT_PACKAGE);
