From c5173d46a16220a743593e28359770c54adcc65b Mon Sep 17 00:00:00 2001
From: Nikias Bassen
Date: Sat, 11 Feb 2012 22:51:31 +0000
Subject: bplist: fix invalid memory access in copy_plist_data

---
diff --git a/src/bplist.c b/src/bplist.c
index 43354be..eff44fc 100644
--- a/src/bplist.c
+++ b/src/bplist.c
@@ -521,12 +521,12 @@ static void* copy_plist_data(const void* src)
         break;
     case PLIST_DATA:
     case PLIST_ARRAY:
-        dstdata->buff = (uint8_t *) malloc(sizeof(uint8_t *) * srcdata->length);
-        memcpy(dstdata->buff, srcdata->buff, sizeof(uint8_t *) * srcdata->length);
+        dstdata->buff = (uint8_t*) malloc(sizeof(uint8_t) * srcdata->length);
+        memcpy(dstdata->buff, srcdata->buff, sizeof(uint8_t) * srcdata->length);
         break;
     case PLIST_DICT:
-        dstdata->buff = (uint8_t *) malloc(sizeof(uint8_t *) * srcdata->length * 2);
-        memcpy(dstdata->buff, srcdata->buff, sizeof(uint8_t *) * srcdata->length * 2);
+        dstdata->buff = (uint8_t*) malloc(sizeof(uint8_t) * srcdata->length * 2);
+        memcpy(dstdata->buff, srcdata->buff, sizeof(uint8_t) * srcdata->length * 2);
         break;
     default:
         break;
--
cgit v0.8.3.1-34-gbf3d
