--- ./bfd/coffswap.h.orig	Wed Aug 28 12:38:44 2002
+++ ./bfd/coffswap.h	Thu Jun 30 22:41:15 2005
@@ -454,9 +454,13 @@
 #else
 	  if (numaux > 1)
 	    {
+#if defined(AVR)
+	      memcpy (in->x_file.x_fname, ext->x_file.x_fname, sizeof (AUXENT));
+#else
 	      if (indx == 0)
 		memcpy (in->x_file.x_fname, ext->x_file.x_fname,
 			numaux * sizeof (AUXENT));
+#endif
 	    }
 	  else
 	    memcpy (in->x_file.x_fname, ext->x_file.x_fname, FILNMLEN);

