--- ./binutils/Makefile.in.orig	Tue Feb 22 13:57:27 2005
+++ ./binutils/Makefile.in	Thu Jun 30 22:41:15 2005
@@ -218,7 +218,7 @@
 	rdcoff.c rddbg.c readelf.c rename.c \
 	resbin.c rescoff.c resrc.c resres.c \
 	size.c srconv.c stabs.c strings.c sysdump.c version.c \
-	windres.c winduni.c wrstabs.c
+	windres.c winduni.c wrcoff.c wrstabs.c
 
 
 GENERATED_CFILES = \
@@ -227,7 +227,7 @@
 
 
 DEBUG_SRCS = rddbg.c debug.c stabs.c ieee.c rdcoff.c
-WRITE_DEBUG_SRCS = $(DEBUG_SRCS) wrstabs.c
+WRITE_DEBUG_SRCS = $(DEBUG_SRCS) wrstabs.c wrcoff.c
 
 # Code shared by all the binutils.
 BULIBS = bucomm.c version.c filemode.c
@@ -407,7 +407,7 @@
 ranlib_LDFLAGS = 
 objcopy_OBJECTS =  objcopy.$(OBJEXT) not-strip.$(OBJEXT) \
 rename.$(OBJEXT) rddbg.$(OBJEXT) debug.$(OBJEXT) stabs.$(OBJEXT) \
-ieee.$(OBJEXT) rdcoff.$(OBJEXT) wrstabs.$(OBJEXT) bucomm.$(OBJEXT) \
+ieee.$(OBJEXT) rdcoff.$(OBJEXT) wrcoff.$(OBJEXT) wrstabs.$(OBJEXT) bucomm.$(OBJEXT) \
 version.$(OBJEXT) filemode.$(OBJEXT)
 objcopy_LDADD = $(LDADD)
 objcopy_DEPENDENCIES =  ../bfd/libbfd.la ../libiberty/libiberty.a
@@ -428,7 +428,7 @@
 nm_new_LDFLAGS = 
 strip_new_OBJECTS =  objcopy.$(OBJEXT) is-strip.$(OBJEXT) \
 rename.$(OBJEXT) rddbg.$(OBJEXT) debug.$(OBJEXT) stabs.$(OBJEXT) \
-ieee.$(OBJEXT) rdcoff.$(OBJEXT) wrstabs.$(OBJEXT) bucomm.$(OBJEXT) \
+ieee.$(OBJEXT) rdcoff.$(OBJEXT) wrcoff.$(OBJEXT) wrstabs.$(OBJEXT) bucomm.$(OBJEXT) \
 version.$(OBJEXT) filemode.$(OBJEXT)
 strip_new_LDADD = $(LDADD)
 strip_new_DEPENDENCIES =  ../bfd/libbfd.la ../libiberty/libiberty.a

