--- ./bfd/Makefile.am.orig	Mon May  2 21:45:42 2005
+++ ./bfd/Makefile.am	Thu Jun 30 23:23:09 2005
@@ -188,6 +188,8 @@
 	coff-apollo.lo \
 	coff-arm.lo \
 	coff-aux.lo \
+	coff-avr.lo \
+	coff-ext-avr.lo \
 	coff-h8300.lo \
 	coff-h8500.lo \
 	coff-i386.lo \
@@ -357,6 +359,8 @@
 	coff-apollo.c \
 	coff-arm.c \
 	coff-aux.c \
+	coff-avr.c \
+	coff-ext-avr.c \
 	coff-h8300.c \
 	coff-h8500.c \
 	coff-i386.c \
@@ -882,10 +886,10 @@
 bfdver.h: $(srcdir)/version.h $(srcdir)/Makefile.in
 	@echo "creating $@"
 	@bfd_version=`echo "$(VERSION)" | sed -e 's/\([^\.]*\)\.*\([^\.]*\)\.*\([^\.]*\)\.*\([^\.]*\)\.*\([^\.]*\).*/\1.00\2.00\3.00\4.00\5/' -e 's/\([^\.]*\)\..*\(..\)\..*\(..\)\..*\(..\)\..*\(..\)$$/\1\2\3\4\5/'` ;\
-	bfd_version_string="\"$(VERSION)\"" ;\
+	bfd_version_string="\"$(VERSION) + coff-avr-patch (20050630)\"" ;\
 	if test "x$(RELEASE)" = x ; then \
 	  bfd_version_date=`sed -n -e 's/.*DATE //p' < $(srcdir)/version.h` ;\
-	  bfd_version_string="\"$(VERSION) $${bfd_version_date}\"" ;\
+	  bfd_version_string="\"$(VERSION) $${bfd_version_date} + coff-avr-patch (20050630)\"" ;\
 	fi ;\
 	sed -e "s/@bfd_version@/$$bfd_version/" -e "s/@bfd_version_string@/$$bfd_version_string/" < $(srcdir)/version.h > $@
 
@@ -1055,6 +1059,12 @@
   $(INCDIR)/coff/internal.h $(INCDIR)/coff/m68k.h $(INCDIR)/coff/external.h \
   coff-m68k.c libcoff.h $(INCDIR)/bfdlink.h coffcode.h \
   coffswap.h
+coff-avr.lo: coff-avr.c $(INCDIR)/filenames.h $(INCDIR)/coff/avr.h \
+  $(INCDIR)/coff/external.h $(INCDIR)/coff/internal.h \
+  libcoff.h $(INCDIR)/bfdlink.h coffcode.h coffswap.h
+coff-ext-avr.lo: coff-ext-avr.c $(INCDIR)/filenames.h $(INCDIR)/coff/avr.h \
+  $(INCDIR)/coff/external.h $(INCDIR)/coff/internal.h \
+  libcoff.h $(INCDIR)/bfdlink.h coffcode.h coffswap.h
 coff-h8300.lo: coff-h8300.c $(INCDIR)/filenames.h $(INCDIR)/bfdlink.h \
   genlink.h $(INCDIR)/coff/h8300.h $(INCDIR)/coff/external.h \
   $(INCDIR)/coff/internal.h libcoff.h $(INCDIR)/libiberty.h \

