--- ./binutils/debug.h.orig	Mon Oct 27 14:20:32 2003
+++ ./binutils/debug.h	Thu Jun 30 22:41:15 2005
@@ -440,6 +440,12 @@
 
 extern bfd_boolean debug_start_source (void *, const char *);
 
+/* Register a pathname translation for source (and include) filenames.
+   This is used by the --change-pathname option of objcopy. */
+
+extern void debug_register_pathname_xlat
+  PARAMS ((const char *, const char *));
+
 /* Record a function definition.  This implicitly starts a function
    block.  The debug_type argument is the type of the return value.
    The bfd_boolean indicates whether the function is globally visible.

