--- gcc/graphite-clast-to-gimple.c	(revision 186481)
+++ gcc/graphite-clast-to-gimple.c	(working copy)
@@ -40,6 +40,10 @@
 #include "graphite-dependences.h"
 #include "graphite-cloog-compat.h"
 
+#ifndef CLOOG_LANGUAGE_C
+#define CLOOG_LANGUAGE_C LANGUAGE_C
+#endif
+
 /* This flag is set when an error occurred during the translation of
    CLAST to Gimple.  */
 static bool gloog_error;
@@ -1367,7 +1371,7 @@
   /* Change cloog output language to C.  If we do use FORTRAN instead, cloog
      will stop e.g. with "ERROR: unbounded loops not allowed in FORTRAN.", if
      we pass an incomplete program to cloog.  */
-  options->language = LANGUAGE_C;
+  options->language = CLOOG_LANGUAGE_C;
 
   /* Enable complex equality spreading: removes dummy statements
      (assignments) in the generated code which repeats the
