--- libuuid/meson.build.orig	2026-04-28 19:11:39
+++ libuuid/meson.build	2026-08-22 18:57:47
@@ -53,9 +53,9 @@
 if build_libuuid
   pkgconfig.generate(lib_uuid,
                      description : 'Universally unique id library',
-                     subdirs : 'uuid',
+                     subdirs : 'libuuid/include/uuid',
                      version : pc_version)
-  install_headers('src/uuid.h', subdir : 'uuid')
+  install_headers('src/uuid.h', subdir : 'libuuid/include/uuid')
   if meson.version().version_compare('>=0.54.0')
     meson.override_dependency('uuid', uuid_dep)
   endif
