From 8dac9813b33ce2f4717b0a9df42b306d00987e27 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 2 Feb 2024 13:41:22 +0800
Subject: [PATCH] sys.cpp: add missing AvailabilityMacros.h

---
 src/occa/internal/utils/sys.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/occa/internal/utils/sys.cpp src/occa/internal/utils/sys.cpp
index 01d6b0ed..e1e3722b 100644
--- src/occa/internal/utils/sys.cpp
+++ src/occa/internal/utils/sys.cpp
@@ -19,6 +19,7 @@
 #    include <errno.h>
 #    include <sys/sysinfo.h>
 #  else // OCCA_MACOS_OS
+#    include <AvailabilityMacros.h>
 #    include <mach/mach_host.h>
 #    ifdef __clang__
 #      include <CoreServices/CoreServices.h>
