https://bugs.kde.org/show_bug.cgi?id=295427
--- coregrind/link_tool_exe_darwin.in.orig	2012-08-19 16:30:14.000000000 +0200
+++ coregrind/link_tool_exe_darwin.in	2012-08-19 16:29:50.000000000 +0200
@@ -101,6 +101,11 @@
 my $ala = $ARGV[0];  # the load address to use
 my $cc  = $ARGV[1];  # the C compiler in use
 
+# if ccache is used, the actual compiler is the next argument
+if ("$cc" =~ /ccache$/) {
+    $cc = $ARGV[2];
+}
+
 # check for plausible-ish alt load address
 die "Bogus alt-load address (1)"
     if (length($ala) < 3 || index($ala, "0x") != 0);
