https://bugs.kde.org/show_bug.cgi?id=366138

--- configure.ac.orig	2016-07-26 09:30:11.000000000 -0700
+++ configure.ac	2016-07-26 09:30:27.000000000 -0700
@@ -154,7 +154,7 @@ AM_CONDITIONAL(COMPILER_IS_ICC, test $is
 # Note: m4 arguments are quoted with [ and ] so square brackets in shell
 # statements have to be quoted.
 case "${is_clang}-${gcc_version}" in
-     applellvm-5.1|applellvm-6.*|applellvm-7.*)
+     applellvm-5.1|applellvm-[[6-9]].*|applellvm-[[1-9]][[0-9]].*)
 	AC_MSG_RESULT([ok (Apple LLVM version ${gcc_version})])
 	;;
      icc-1[[3-9]].*)
@@ -167,7 +167,7 @@ case "${is_clang}-${gcc_version}" in
 	AC_MSG_RESULT([ok (clang-${gcc_version})])
 	;;
      *)
-	AC_MSG_RESULT([no (${gcc_version})])
+	AC_MSG_RESULT([no (${is_clang}-${gcc_version})])
 	AC_MSG_ERROR([please use gcc >= 3.0 or clang >= 2.9 or icc >= 13.0])
 	;;
 esac
