https://bugs.kde.org/show_bug.cgi?id=295428
Index: coregrind/m_main.c
===================================================================
--- coregrind/m_main.c	(revision 12418)
+++ coregrind/m_main.c	(working copy)
@@ -2842,6 +2842,7 @@ asm("\n"
     "\tandl  $~15, %eax\n"
     /* install it, and collect the original one */
     "\txchgl %eax, %esp\n"
+    "\tsubl  $12, %esp\n"
     /* call _start_in_C_darwin, passing it the startup %esp */
     "\tpushl %eax\n"
     "\tcall  __start_in_C_darwin\n"
