--- /Users/rhwood/Developer/MacPorts/Repository/dports/devel/liboil/Makefile.in.unpatched	2007-02-07 06:46:38.000000000 -0500
+++ liboil/powerpc/Makefile.in	2007-02-07 06:47:46.000000000 -0500
@@ -56,7 +56,7 @@
 libpowerpc_la_LIBADD =
 am__objects_1 = libpowerpc_la-abs.lo libpowerpc_la-clip.lo \
 	libpowerpc_la-conv.lo libpowerpc_la-copy.lo \
-	libpowerpc_la-md5.lo libpowerpc_la-mix.lo \
+	libpowerpc_la-mix.lo \
 	libpowerpc_la-multsum.lo libpowerpc_la-resample.lo \
 	libpowerpc_la-rgb2bgr.lo libpowerpc_la-rgb2rgba.lo \
 	libpowerpc_la-sad8x8.lo libpowerpc_la-splat.lo \
@@ -251,7 +251,6 @@
 	clip.c \
 	conv.c \
 	copy.c \
-	md5.c \
 	mix.c \
 	multsum.c \
 	resample.c \
@@ -320,7 +319,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-clip.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-conv.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-copy.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-md5.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-mix.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-multsum.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-resample.Plo@am__quote@
@@ -379,13 +377,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -c -o libpowerpc_la-copy.lo `test -f 'copy.c' || echo '$(srcdir)/'`copy.c
 
-libpowerpc_la-md5.lo: md5.c
-@am__fastdepCC_TRUE@	if $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -MT libpowerpc_la-md5.lo -MD -MP -MF "$(DEPDIR)/libpowerpc_la-md5.Tpo" -c -o libpowerpc_la-md5.lo `test -f 'md5.c' || echo '$(srcdir)/'`md5.c; \
-@am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/libpowerpc_la-md5.Tpo" "$(DEPDIR)/libpowerpc_la-md5.Plo"; else rm -f "$(DEPDIR)/libpowerpc_la-md5.Tpo"; exit 1; fi
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='md5.c' object='libpowerpc_la-md5.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -c -o libpowerpc_la-md5.lo `test -f 'md5.c' || echo '$(srcdir)/'`md5.c
-
 libpowerpc_la-mix.lo: mix.c
 @am__fastdepCC_TRUE@	if $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -MT libpowerpc_la-mix.lo -MD -MP -MF "$(DEPDIR)/libpowerpc_la-mix.Tpo" -c -o libpowerpc_la-mix.lo `test -f 'mix.c' || echo '$(srcdir)/'`mix.c; \
 @am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/libpowerpc_la-mix.Tpo" "$(DEPDIR)/libpowerpc_la-mix.Plo"; else rm -f "$(DEPDIR)/libpowerpc_la-mix.Tpo"; exit 1; fi
