Do not look for things in Homebrew paths.
--- setup.py.orig	2022-03-29 14:59:50.000000000 -0500
+++ setup.py	2022-03-29 23:52:53.000000000 -0500
@@ -25,9 +25,6 @@
 
     if 'bsd' in sys.platform:
         extra_includes.append('/usr/local/include')
-    elif 'darwin' in sys.platform and machine() == 'arm64':
-        extra_includes.append('/opt/homebrew/include')
-        extra_library_dirs.append('/opt/homebrew/lib')
 
 try:
     from setuptools_scm import get_version
