--- setup.py.orig	2022-03-21 20:35:06.000000000 -0500
+++ setup.py	2022-03-21 20:35:36.000000000 -0500
@@ -19,8 +19,8 @@
 if 'bsd' in sys.platform:
     extra_includes.append('/usr/local/include')
 elif 'darwin' in sys.platform and machine() == 'arm64':
-    extra_includes.append('/opt/homebrew/include')
-    extra_library_dirs.append('/opt/homebrew/lib')
+    extra_includes.append('%PREFIX%/include')
+    extra_library_dirs.append('%PREFIX%/lib')
 
 try:
     from setuptools_scm import get_version
