--- a/xpdf/SecurityHandler.cc.orig	2014-05-28 13:50:50.000000000 -0500
+++ b/xpdf/SecurityHandler.cc	2015-01-10 11:36:21.000000000 -0600
@@ -421,9 +421,6 @@
     return gFalse;
   }
   permFlags = (*xsh->authorize)(xsh->handlerData, docData, authData);
-  if (!(permFlags & xpdfPermissionOpen)) {
-    return gFalse;
-  }
   if (!(*xsh->getKey)(xsh->handlerData, docData, &key, &length, &encVersion)) {
     return gFalse;
   }
--- a/xpdf/XRef.cc.orig	2014-05-28 13:50:50.000000000 -0500
+++ b/xpdf/XRef.cc	2015-01-10 11:36:21.000000000 -0600
@@ -958,19 +958,19 @@
 }
 
 GBool XRef::okToPrint(GBool ignoreOwnerPW) {
-  return (!ignoreOwnerPW && ownerPasswordOk) || (permFlags & permPrint);
+  return gTrue;
 }
 
 GBool XRef::okToChange(GBool ignoreOwnerPW) {
-  return (!ignoreOwnerPW && ownerPasswordOk) || (permFlags & permChange);
+  return gTrue;
 }
 
 GBool XRef::okToCopy(GBool ignoreOwnerPW) {
-  return (!ignoreOwnerPW && ownerPasswordOk) || (permFlags & permCopy);
+  return gTrue;
 }
 
 GBool XRef::okToAddNotes(GBool ignoreOwnerPW) {
-  return (!ignoreOwnerPW && ownerPasswordOk) || (permFlags & permNotes);
+  return gTrue;
 }
 
 Object *XRef::fetch(int num, int gen, Object *obj, int recursion) {
