--- config.mk.orig	2023-10-23 10:47:34
+++ config.mk	2023-10-23 10:48:26
@@ -2,7 +2,6 @@
 
 # paths (unset $PCPREFIX to not install a pkg-config-file)
 DESTDIR   =
-PREFIX    = /usr/local
 INCPREFIX = $(PREFIX)/include
 LIBPREFIX = $(PREFIX)/lib
 MANPREFIX = $(PREFIX)/share/man
@@ -19,14 +18,14 @@
 
 SHFLAGS   = -fPIC -ffreestanding
 
-SOFLAGS   = -shared -nostdlib -Wl,--soname=libgrapheme.so.$(VERSION_MAJOR).$(VERSION_MINOR)
-SONAME    = libgrapheme.so.$(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH)
-SOSYMLINK = true
+SOFLAGS   = -dynamiclib -install_name libgrapheme.$(VERSION_MAJOR).dylib -current_version $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_PATCH) -compatibility_version $(VERSION_MAJOR).$(VERSION_MINOR).0
+SONAME    = libgrapheme.$(VERSION_MAJOR).dylib
+SOSYMLINK = false
 
 # tools (unset $LDCONFIG to not call ldconfig(1) after install/uninstall)
 CC       = cc
 BUILD_CC = $(CC)
 AR       = ar
 RANLIB   = ranlib
-LDCONFIG = ldconfig
+LDCONFIG = 
 SH       = sh
