--- configure.ac.orig	2015-09-20 00:40:59.000000000 -0700
+++ configure.ac	2016-03-24 09:43:26.000000000 -0700
@@ -26,7 +26,7 @@
 
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR([config/m4])
-AM_CONFIG_HEADER([config.h include/soundtouch_config.h])
+AC_CONFIG_HEADERS([config.h include/soundtouch_config.h])
 AM_INIT_AUTOMAKE
 AM_SILENT_RULES([yes])
 #AC_DISABLE_SHARED	dnl This makes libtool only build static libs 
