--- configure.ac.orig	2013-01-06 02:17:52.000000000 -0800
+++ configure.ac	2013-01-06 02:18:03.000000000 -0800
@@ -21,7 +21,7 @@ dnl Place - Suite 330, Boston, MA  02111
 
 AC_INIT(SoundTouch, 1.6.0, [http://www.surina.net/soundtouch])
 AC_CONFIG_AUX_DIR(config)
-AM_CONFIG_HEADER([include/soundtouch_config.h])
+AC_CONFIG_HEADERS([include/soundtouch_config.h])
 AM_INIT_AUTOMAKE
 AC_DISABLE_SHARED	dnl This makes libtool only build static libs 
 #AC_GNU_SOURCE 		dnl enable posix extensions in glibc
