--- source/SoundTouch/Makefile.am.orig	2015-09-20 00:40:59.000000000 -0700
+++ source/SoundTouch/Makefile.am	2016-03-24 09:45:44.000000000 -0700
@@ -72,4 +72,4 @@
 # noinst_LTLIBRARIES = libSoundTouchOpt.la
 # libSoundTouch_la_LIBADD = libSoundTouchOpt.la
 # libSoundTouchOpt_la_SOURCES = mmx_optimized.cpp sse_optimized.cpp 
-# libSoundTouchOpt_la_CXXFLAGS = -O3 -msse -fcheck-new -I../../include
+# libSoundTouchOpt_la_CXXFLAGS = -O3 -msse -I../../include
