--- src/logging.c.orig	2019-01-26 11:28:32.000000000 -0600
+++ src/logging.c	2019-01-26 11:42:55.000000000 -0600
@@ -18,7 +18,10 @@
     return 0;
 }
 
+#define UNUSED(x) (void)(x)
+
 int ugly_log(int level, const char *tag, const char *format, ...) {
+    UNUSED(tag);
     if (level > max_level) {
         return 0;
     }
@@ -30,19 +33,19 @@
     fprintf(stderr, "%d-%02d-%02dT%02d:%02d:%02d ", tt->tm_year + 1900, tt->tm_mon + 1, tt->tm_mday, tt->tm_hour, tt->tm_min, tt->tm_sec);
     switch (level) {
     case UDEBUG:
-        fprintf(stderr, "DEBUG %s: ", tag);
+        fprintf(stderr, "DEBUG: ");
         break;
     case UINFO:
-        fprintf(stderr, "INFO %s: ", tag);
+        fprintf(stderr, "INFO: ");
         break;
     case UWARN:
-        fprintf(stderr, "WARN %s: ", tag);
+        fprintf(stderr, "WARN: ");
         break;
     case UERROR:
-        fprintf(stderr, "ERROR %s: ", tag);
+        fprintf(stderr, "ERROR: ");
         break;
     default:
-        fprintf(stderr, "%d %s: ", level, tag);
+        fprintf(stderr, "%d: ", level);
         break;
     }
     vfprintf(stderr, format, args);
