--- src/logging.c.orig	2017-01-28 11:55:48.000000000 +0100
+++ src/logging.c	2017-02-03 00:12:16.000000000 +0100
@@ -30,25 +30,25 @@
     fprintf(stderr, "%d-%02d-%02dT%02d:%02d:%02d ", tt->tm_year + 1900, tt->tm_mon + 1, tt->tm_mday, tt->tm_hour, tt->tm_min, tt->tm_sec);
     switch (level) {
     case UDEBUG:
-        fprintf(stderr, "DEBUG %s: ", tag);
+        fprintf(stderr, "DEBUG: ");
         break;
     case UINFO:
-        fprintf(stderr, "INFO %s: ", tag);
+        fprintf(stderr, "INFO: ");
         break;
     case UWARN:
-        fprintf(stderr, "WARN %s: ", tag);
+        fprintf(stderr, "WARN: ");
         break;
     case UERROR:
-        fprintf(stderr, "ERROR %s: ", tag);
+        fprintf(stderr, "ERROR: ");
         break;
     case UFATAL:
-        fprintf(stderr, "FATAL %s: ", tag);
+        fprintf(stderr, "FATAL: ");
         vfprintf(stderr, format, args);
         exit(EXIT_FAILURE);
         // NEVER GETS HERE!!!
         break;
     default:
-        fprintf(stderr, "%d %s: ", level, tag);
+        fprintf(stderr, "%d: ", level);
         break;
     }
     vfprintf(stderr, format, args);
