--- DESCRIPTION.orig	2023-05-18 18:00:09.000000000 +0800
+++ DESCRIPTION	2023-05-18 18:28:47.000000000 +0800
@@ -20,7 +20,7 @@
 Depends: R (>= 2.10)
 Imports: data.table, grDevices, methods, onion, R6, Rcpp (>= 1.0.9),
         rgl, tools, utils
-Suggests: randomcoloR, rmarchingcubes, spray, viridisLite
+Suggests: rmarchingcubes, spray, viridisLite
 LinkingTo: BH, Rcpp, RcppCGAL, RcppColors, RcppEigen
 Encoding: UTF-8
 LazyData: TRUE
