--- src/str_util.c.orig	2010-09-04 05:43:02.000000000 +0200
+++ src/str_util.c	2010-09-04 05:47:53.000000000 +0200
@@ -204,13 +204,13 @@
 
 char *str_strnchr(const char *s, char c, int n)
 {
-	int p = (int)s;
+	const char *p = s;
 	int count = 0;
 
-	while (*(char *)p) {
-		if (*(char *)p == c)
+	while (*p) {
+		if (*p == c)
 			if (++count == n)
-				return (char *)p;
+				return p;
 		p++;
 	}
 	return NULL;
@@ -218,13 +218,13 @@
 
 char *str_strnrchr(const char *s, char c, int n)
 {
-	int p = (int)s + strlen(s);
+	const char *p = s + strlen(s);
 	int count = 0;
 
-	while (p >= (int)s) {
-		if (*(char *)p == c)
+	while (p >= s) {
+		if (*p == c)
 			if (++count == n)
-				return (char *)p;
+				return p;
 		p--;
 	}
 	return NULL;
