--- data/tests/Makefile.am.orig	2015-03-12 12:47:15.000000000 -0700
+++ data/tests/Makefile.am	2015-03-12 12:51:38.000000000 -0700
@@ -53,18 +53,18 @@
 	validate.xml.gz
 
 colorhug-als-2.0.0.cab: $(srcdir)/firmware/2_0_0/firmware.*
-	$(AM_V_GEN) touch -c -m -d"2000-01-01T00:00:00" $?;	\
+	$(AM_V_GEN) touch -c -m -t "200001010000.00" $?;	\
 	$(GCAB) --create --nopath $@			\
 		$(srcdir)/firmware/2_0_0/firmware.bin	\
 		$(srcdir)/firmware/2_0_0/firmware.inf
 colorhug-als-2.0.1.cab: $(srcdir)/firmware/2_0_1/firmware.*
-	$(AM_V_GEN) touch -c -m -d"2000-01-01T00:00:00" $?;	\
+	$(AM_V_GEN) touch -c -m -t "200001010000.00" $?;	\
 	$(GCAB) --create --nopath $@			\
 		$(srcdir)/firmware/2_0_1/firmware.bin	\
 		$(srcdir)/firmware/2_0_1/firmware.inf	\
 		$(srcdir)/firmware/2_0_1/firmware.metainfo.xml
 colorhug-als-2.0.2.cab: $(srcdir)/firmware/2_0_2/firmware.*
-	$(AM_V_GEN) touch -c -m -d"2000-01-01T00:00:00" $?;	\
+	$(AM_V_GEN) touch -c -m -t "200001010000.00" $?;	\
 	$(GCAB) --create --nopath $@			\
 		$(srcdir)/firmware/2_0_2/firmware.bin	\
 		$(srcdir)/firmware/2_0_2/firmware.inf	\
