--- configure.in.orig	2013-07-25 15:10:07.000000000 +0200
+++ configure.in	2013-07-25 15:12:01.000000000 +0200
@@ -6,8 +6,7 @@
 AC_INIT(saga, 2.1.0, BUG-REPORT-ADDRESS)
 AC_CONFIG_SRCDIR([src/saga_core/saga_gui/wksp_data_menu_files.cpp])
 AC_CONFIG_HEADER([config.h])
-AM_INIT_AUTOMAKE(saga, 2.1.0beta)
-AM_INIT_AUTOMAKE(tar-ustar)
+AM_INIT_AUTOMAKE([tar-ustar subdir-objects])
 
 # Checks for programs.
 AC_PROG_CXX
