--- configure.orig	2016-05-12 08:39:14.000000000 +0200
+++ configure	2016-05-12 08:40:48.000000000 +0200
@@ -17863,16 +17863,7 @@
 	#
 	# linking flags needed when embedding
 	#
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking python extra linking flags" >&5
-$as_echo_n "checking python extra linking flags... " >&6; }
-	if test -z "$PYTHON_EXTRA_LDFLAGS"; then
-		PYTHON_EXTRA_LDFLAGS=`$PYTHON -c "import distutils.sysconfig; \
-			conf = distutils.sysconfig.get_config_var; \
-			print (conf('LINKFORSHARED'))"`
-	fi
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_EXTRA_LDFLAGS" >&5
-$as_echo "$PYTHON_EXTRA_LDFLAGS" >&6; }
-
+    PYTHON_EXTRA_LDFLAGS=
 
 	#
 	# final check to see if everything compiles alright
