--- saga-gis/cmake/modules/FindPostgres.cmake.orig	2025-05-12 14:20:06
+++ saga-gis/cmake/modules/FindPostgres.cmake	2025-07-08 19:10:32
@@ -61,13 +61,7 @@
   IF(UNIX) 
 
     SET(POSTGRES_CONFIG_PREFER_PATH "$ENV{POSTGRES_HOME}/bin" CACHE STRING "preferred path to PG (pg_config)")
-    FIND_PROGRAM(POSTGRES_CONFIG pg_config
-      ${POSTGRES_CONFIG_PREFER_PATH}
-      /usr/local/pgsql/bin/
-      /usr/local/bin/
-      /usr/bin/
-      )
-    # MESSAGE("DBG POSTGRES_CONFIG ${POSTGRES_CONFIG}")
+    set(POSTGRES_CONFIG "${POSTGRES_CONFIG_PREFER_PATH}/pg_config")
     
     IF (POSTGRES_CONFIG) 
       # set INCLUDE_DIR
