--- saga-gis/src/saga_core/saga_api/saga_api_python/CMakeLists.txt.orig	2023-05-25 18:05:22.000000000 +0200
+++ saga-gis/src/saga_core/saga_api/saga_api_python/CMakeLists.txt	2023-06-05 14:04:17.000000000 +0200
@@ -25,8 +25,8 @@
 find_package(SWIG QUIET)
 if(SWIG_FOUND)
 	message(STATUS "SWIG for Python found")
-    find_package(Python COMPONENTS Interpreter Development)
-	if(Python_FOUND)
+    find_package(Python3 @PYVER@ EXACT COMPONENTS Interpreter Development)
+	if(Python3_FOUND)
 #		project(saga_api_python)
 
 		message(STATUS "project: ${PROJECT_NAME}")
@@ -43,8 +43,8 @@
 
 		include(UseSWIG)
 
-        include_directories("${Python_INCLUDE_DIRS}")
-        include_directories("${Python_INCLUDE_DIRS}/../PC")	# msvc builds
+        include_directories("${Python3_INCLUDE_DIRS}")
+        include_directories("${Python3_INCLUDE_DIRS}/../PC")	# msvc builds
 		include_directories("..")
 
 		set(CMAKE_SWIG_FLAGS "-D_SAGA_PYTHON -D_SWIG_WCHAR")
@@ -57,14 +57,14 @@
 		set_property(TARGET saga_api_python PROPERTY OUTPUT_NAME saga_api)
 
 		swig_link_libraries(saga_api_python saga_api)
-		swig_link_libraries(saga_api_python ${Python_LIBRARIES})
+		swig_link_libraries(saga_api_python ${Python3_LIBRARIES})
 
 
-        if(EXISTS ${Python_SITELIB})
-            message(STATUS "Python site-packages: ${PYTHON_SITELIB}")
-            install(TARGETS saga_api_python DESTINATION ${Python_SITELIB})
-            install(FILES ${CMAKE_BINARY_DIR}/src/saga_core/saga_api/saga_api_python/saga_api.py DESTINATION ${Python_SITELIB})
-            install(FILES ${CMAKE_SOURCE_DIR}/src/accessories/python/saga.py DESTINATION ${Python_SITELIB})
+        if(EXISTS ${Python3_SITELIB})
+            message(STATUS "Python site-packages: ${Python3_SITELIB}")
+            install(TARGETS saga_api_python DESTINATION ${Python3_SITELIB})
+            install(FILES ${CMAKE_BINARY_DIR}/src/saga_core/saga_api/saga_api_python/saga_api.py DESTINATION ${Python3_SITELIB})
+            install(FILES ${CMAKE_SOURCE_DIR}/src/accessories/python/saga.py DESTINATION ${Python3_SITELIB})
 		endif()
 	endif()
 endif()
