Upstream-Status: Backport


diff --git a/pdf2djvu.cc b/pdf2djvu.cc
index 2b42d16..bb5fd57 100644
--- pdf2djvu.cc.orig
+++ pdf2djvu.cc
@@ -89,7 +89,7 @@ static int get_page_for_goto_link(pdf::link::GoTo *goto_link, pdf::Catalog *cata
 #endif
   }
   else
-    dest.reset(orig_dest->copy());
+    dest.reset(new pdf::link::Destination(*orig_dest));
   if (dest.get() != nullptr)
   {
     int page;
diff --git a/pdf-backend.cc b/pdf-backend.cc
index 3c871b1..8fab764 100644
--- pdf-backend.cc.orig
+++ pdf-backend.cc
@@ -130,7 +130,11 @@ void pdf::Environment::set_antialias(bool value)
  */
 
 pdf::Document::Document(const std::string &file_name)
-: ::PDFDoc(new pdf::String(file_name.c_str()), nullptr, nullptr)
+#if POPPLER_VERSION >= 220300
+: ::PDFDoc(std::make_unique<pdf::String>(file_name.c_str()))
+#else
+: ::PDFDoc(new pdf::String(file_name.c_str()))
+#endif
 {
   if (!this->isOk())
     throw LoadError();
