--- configure.orig	2010-06-12 15:20:48.000000000 -0500
+++ configure	2010-07-23 05:21:04.000000000 -0500
@@ -3017,7 +3017,7 @@
 # Prior to pdf2djvu 0.5.0, pdf2djvu simply dissalowed subsample ratio 12 by
 # itself.
 echo 'P1 12 12' > conftest.458211.pbm
-yes 0 | head -n 144 >> conftest.458211.pbm
+for i in {1..144}; do echo 0; done >> conftest.458211.pbm
 "$djvulibre_bin_path/cjb2" conftest.458211.pbm conftest.458211.djvu
 "$djvulibre_bin_path/ddjvu" -format=rle conftest.458211.djvu conftest.458211.sep
 printf 'P6 1 1 255 xxx' >> conftest.458211.sep
