--- configure.ac.orig	2020-12-09 14:36:36.000000000 -0800
+++ configure.ac	2020-12-09 16:25:08.000000000 -0800
@@ -73,8 +73,8 @@
 done
 AC_MSG_RESULT([ok])
 
-PKG_CHECK_MODULES([POPPLER], [poppler-splash >= 0.35.0])
-poppler_version=$($PKG_CONFIG --modversion poppler-splash)
+PKG_CHECK_MODULES([POPPLER], [poppler >= 0.35.0])
+poppler_version=$($PKG_CONFIG --modversion poppler)
 AC_DEFINE_UNQUOTED([POPPLER_VERSION_STRING], ["$poppler_version"], [Define to the version of Poppler])
 parse_poppler_version()
 {
