--- pdf2djvu.cc.orig	2009-07-13 10:31:59.000000000 -0700
+++ pdf2djvu.cc	2009-07-13 10:41:53.000000000 -0700
@@ -285,28 +285,34 @@
   void drawImage(pdf::gfx::State *state, pdf::Object *object, pdf::Stream *stream, int width, int height,
     pdf::gfx::ImageColorMap *color_map, int *mask_colors, pdf::Bool inline_image)
   {
+    pdf::Bool interpolate = 1;
+
     if (is_foreground_color_map(color_map) || config.no_render)
       return;
-    Renderer::drawImage(state, object, stream, width, height, color_map, mask_colors, inline_image);
+    Renderer::drawImage(state, object, stream, width, height, color_map, interpolate, mask_colors, inline_image);
   }
 
   void drawMaskedImage(pdf::gfx::State *state, pdf::Object *object, pdf::Stream *stream, int width, int height,
     pdf::gfx::ImageColorMap *color_map, pdf::Stream *mask_stream, int mask_width, int mask_height, pdf::Bool mask_invert)
   {
+    pdf::Bool interpolate, maskInterpolate = 1;
+    
     if (is_foreground_color_map(color_map) || config.no_render)
       return;
     Renderer::drawMaskedImage(state, object, stream, width, height,
-      color_map, mask_stream, mask_width, mask_height, mask_invert);
+      color_map, interpolate, mask_stream, mask_width, mask_height, mask_invert, maskInterpolate);
   }
 
   void drawSoftMaskedImage(pdf::gfx::State *state, pdf::Object *object, pdf::Stream *stream,
     int width, int height, pdf::gfx::ImageColorMap *color_map, pdf::Stream *mask_stream,
     int mask_width, int mask_height,	pdf::gfx::ImageColorMap *mask_color_map)
   {
+    pdf::Bool interpolate, maskInterpolate = 1;
+
     if (is_foreground_color_map(color_map) || config.no_render)
       return;
     Renderer::drawSoftMaskedImage(state, object, stream, width, height,
-      color_map, mask_stream, mask_width, mask_height, mask_color_map);
+      color_map, interpolate, mask_stream, mask_width, mask_height, mask_color_map, maskInterpolate);
   }
 
   pdf::Bool interpretType3Chars() { return gFalse; }
