From baacea206bc8f7bae1da2430cfc4eed3796f50be Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 9 Oct 2024 18:58:17 +0700
Subject: [PATCH 06/10] libqcdm: fix build when non-POSIX constants are
 undefined

---
 libqcdm/src/com.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libqcdm/src/com.c b/libqcdm/src/com.c
index d6283e06..11179a38 100644
--- a/libqcdm/src/com.c
+++ b/libqcdm/src/com.c
@@ -23,6 +23,18 @@
 #include "com.h"
 #include "errors.h"
 
+#ifndef CBAUD
+#define CBAUD 0
+#endif
+
+#ifndef IUCLC
+#define IUCLC 0
+#endif
+
+#ifndef OLCUC
+#define OLCUC 0
+#endif
+
 int
 qcdm_port_setup (int fd)
 {
-- 
2.46.1

