--- test/test-stress.c.orig	2025-01-05 00:39:55
+++ test/test-stress.c	2025-01-05 00:43:02
@@ -54,8 +54,8 @@
 static int ITER    = 50;
 #define ALLOW_LARGE true
 #else
-static int THREADS = 32;      // more repeatable if THREADS <= #processors
-static int SCALE   = 50;      // scaling factor
+static int THREADS = 8;       // more repeatable if THREADS <= #processors
+static int SCALE   = 100;     // scaling factor
 static int ITER    = 50;      // N full iterations destructing and re-creating all threads
 #endif
 
