--- ./src/java/org/apache/bcel/verifier/statics/Pass2Verifier.java	2003-04-25 10:06:16.000000000 -0700
+++ ./src-fix/java/org/apache/bcel/verifier/statics/Pass2Verifier.java	2005-11-17 23:58:06.000000000 -0800
@@ -345,7 +345,7 @@
 
 			for (int i=0; i<atts.length; i++){
 				if ((! (atts[i] instanceof SourceFile)) &&
-				    (! (atts[i] instanceof Deprecated))     &&
+				    (! (atts[i] instanceof org.apache.bcel.classfile.Deprecated))     &&
 				    (! (atts[i] instanceof InnerClasses)) &&
 				    (! (atts[i] instanceof Synthetic))){
 					addMessage("Attribute '"+tostring(atts[i])+"' as an attribute of the ClassFile structure '"+tostring(obj)+"' is unknown and will therefore be ignored.");
@@ -519,7 +519,7 @@
 			for (int i=0; i<atts.length; i++){
 				if ((! (atts[i] instanceof ConstantValue)) &&
 				    (! (atts[i] instanceof Synthetic))     &&
-				    (! (atts[i] instanceof Deprecated))){
+				    (! (atts[i] instanceof org.apache.bcel.classfile.Deprecated))){
 					addMessage("Attribute '"+tostring(atts[i])+"' as an attribute of Field '"+tostring(obj)+"' is unknown and will therefore be ignored.");
 				}
 				if  (! (atts[i] instanceof ConstantValue)){
@@ -660,7 +660,7 @@
 				if ((! (atts[i] instanceof Code)) &&
 				    (! (atts[i] instanceof ExceptionTable))     &&
 				    (! (atts[i] instanceof Synthetic)) &&
-				    (! (atts[i] instanceof Deprecated))){
+				    (! (atts[i] instanceof org.apache.bcel.classfile.Deprecated))){
 					addMessage("Attribute '"+tostring(atts[i])+"' as an attribute of Method '"+tostring(obj)+"' is unknown and will therefore be ignored.");
 				}
 				if ((! (atts[i] instanceof Code)) &&
@@ -702,7 +702,7 @@
 				addMessage("SourceFile attribute '"+tostring(obj)+"' has a funny name: remember not to confuse certain parsers working on javap's output. Also, this name ('"+sourcefilename+"') is considered an unqualified (simple) file name only.");
 			}
 		}
-		public void visitDeprecated(Deprecated obj){//vmspec2 4.7.10
+		public void visitDeprecated(org.apache.bcel.classfile.Deprecated obj){//vmspec2 4.7.10
 			checkIndex(obj, obj.getNameIndex(), CONST_Utf8);
 
 			String name = ((ConstantUtf8) cp.getConstant(obj.getNameIndex())).getBytes();
