--- CMakeLists.txt.orig	2014-11-05 07:41:52.000000000 -0800
+++ CMakeLists.txt	2019-06-28 19:00:43.000000000 -0700
@@ -260,7 +260,7 @@
       )
     IF (NOT ICET_INSTALL_NO_DEVELOPMENT)
       INSTALL(EXPORT ${ICET_INSTALL_EXPORT_NAME}
-        DESTINATION ${ICET_INSTALL_LIB_DIR}
+        DESTINATION ${ICET_INSTALL_LIB_DIR}/cmake/IceT
         )
     ENDIF (NOT ICET_INSTALL_NO_DEVELOPMENT)
   ENDIF(NOT ICET_INSTALL_NO_LIBRARIES)
@@ -315,7 +315,7 @@
     ${ICET_LIBRARY_DIR}/IceTConfig.cmake.install
     @ONLY IMMEDIATE)
   INSTALL(FILES ${ICET_LIBRARY_DIR}/IceTConfig.cmake.install
-    DESTINATION ${ICET_INSTALL_LIB_DIR}
+    DESTINATION ${ICET_INSTALL_LIB_DIR}/cmake/IceT
     RENAME IceTConfig.cmake
     )
 ENDIF (NOT ICET_INSTALL_NO_DEVELOPMENT)
@@ -328,7 +328,7 @@
   )
 IF (NOT ICET_INSTALL_NO_DEVELOPMENT)
   INSTALL(FILES ${ICET_LIBRARY_DIR}/IceTConfigVersion.cmake
-    DESTINATION ${ICET_INSTALL_LIB_DIR}
+    DESTINATION ${ICET_INSTALL_LIB_DIR}/cmake/IceT
     )
 ENDIF (NOT ICET_INSTALL_NO_DEVELOPMENT)

--- cmake/IceTConfig.cmake.in.orig	2014-11-05 07:41:52.000000000 -0800
+++ cmake/IceTConfig.cmake.in	2019-06-28 19:01:33.000000000 -0700
@@ -14,7 +14,7 @@

 # This file should be installed in the lib directory.  Find the root directory.
 GET_FILENAME_COMPONENT(_dir "${CMAKE_CURRENT_LIST_FILE}" PATH)
-GET_FILENAME_COMPONENT(_install_dir "${_dir}/.." ABSOLUTE)
+GET_FILENAME_COMPONENT(_install_dir "${_dir}/../../.." ABSOLUTE)

 # Load the targets include (next to this one).
 INCLUDE("${_dir}/IceTTargets.cmake")
