--- kdiff3.pro.orig	2012-03-23 14:31:14.000000000 +0000
+++ kdiff3.pro	2012-03-23 14:24:05.000000000 +0000
@@ -61,11 +61,13 @@
 }
 
 unix {
-  documentation.path = /usr/local/share/doc/kdiff3
+  documentation.path = $$DESTDIR/../share/doc/kdiff3
   documentation.files = ../doc/*
 
   INSTALLS += documentation
 
-  target.path = /usr/local/bin
-  INSTALLS += target
 }
+
+mac {
+  CONFIG -= app_bundle
+}

