--- kdiff3.pro.orig	2012-07-14 05:27:14.000000000 -0500
+++ kdiff3.pro	2012-09-27 11:43:59.000000000 -0500
@@ -64,11 +64,13 @@
 }
 
 unix {
-  documentation.path = /usr/local/share/doc/kdiff3
+  documentation.path = $$DESTDIR/../share/doc/kdiff3
   documentation.files = ../doc/*
 
   INSTALLS += documentation
 
-  target.path = /usr/local/bin
-  INSTALLS += target
+}
+
+mac {
+  CONFIG -= app_bundle
 }
