--- lib/Mac/PropertyList/ReadBinary.pm.orig	2026-09-06 18:18:12
+++ lib/Mac/PropertyList/ReadBinary.pm	2026-09-06 18:42:25
@@ -215,7 +215,7 @@
 				$b -= Math::BigInt->new(2)->bpow(64);
 				}
 
-			@values = ( $b );
+			@values = ( $b->bstr() );
 			}
 		elsif( $power_of_2 == 4 ) { # 128 bits
 		    # 128 bits aren't part of the public API, but apparently
@@ -231,7 +231,7 @@
 				$b -= Math::BigInt->new(2)->bpow(128);
 				}
 
-			@values = ( $b );
+			@values = ( $b->bstr() );
 			}
 
 		return Mac::PropertyList::integer->new( $values[0] );
