--- configure.ac.orig	2015-03-06 05:15:35.000000000 -0800
+++ configure.ac	2015-03-06 05:16:19.000000000 -0800
@@ -184,7 +184,7 @@
 ifdef([GNOME_COMPILE_WARNINGS],[GNOME_COMPILE_WARNINGS],)
 CFLAGS="$CFLAGS $WARN_CFLAGS -DG_DISABLE_DEPRECATED"
 
-set_more_warnings=yes
+set_more_warnings=no
 if test "$GCC" = yes -a "x$set_more_warnings" != xno; then
 	dnl Clang needs to option, or else it will appear to support any
 	dnl warning option
