diff -ru ../fwknop-2.0.4.orig/lib/md5.c ./lib/md5.c
--- ../fwknop-2.0.4.orig/lib/md5.c	2012-12-09 12:55:59.000000000 -0800
+++ ./lib/md5.c	2013-06-09 15:08:13.000000000 -0700
@@ -29,6 +29,12 @@
 */
 #include "md5.h"
 
+#ifndef BYTEORDER
+#ifdef __ppc__
+#define BYTEORDER 4321
+#endif
+#endif
+
 #if BYTEORDER == 1234
   #define byteReverse(buf, len)    /* Nothing */
 #elif BYTEORDER == 4321
diff -ru ../fwknop-2.0.4.orig/lib/sha1.c ./lib/sha1.c
--- ../fwknop-2.0.4.orig/lib/sha1.c	2012-12-09 12:55:59.000000000 -0800
+++ ./lib/sha1.c	2013-06-09 15:08:59.000000000 -0700
@@ -72,6 +72,12 @@
 
 #undef SWAP_DONE
 
+#ifndef BYTEORDER
+#ifdef __ppc__
+#define BYTEORDER 4321
+#endif
+#endif
+
 #if BYTEORDER == 1234
 #define SWAP_DONE
     for (i = 0; i < 16; ++i) {
