--- src/quickjs_backend.cpp
+++ src/quickjs_backend.cpp	2024-08-01 10:35:08
@@ -2149,7 +2149,7 @@
 		{
 			size_t idx WZ_DECL_UNUSED = 0; // unused when Args... is empty
 			quickjs_execution_context execution_context(context);
-			return box(apply(f, UnboxTuple<Args...>(execution_context, idx, context, argc, argv, wrappedFunctionName)()), context);
+			return box(std::apply(f, UnboxTuple<Args...>(execution_context, idx, context, argc, argv, wrappedFunctionName)()), context);
 		}
 
 		template<typename R, typename...Args>
