--- usrp/host/lib/legacy/Makefile.in_old	2009-10-26 16:39:12.000000000 -0400
+++ usrp/host/lib/legacy/Makefile.in	2009-10-26 16:40:01.000000000 -0400
@@ -794,7 +794,7 @@
 STAMPS = $(DEPDIR)/stamp-sources-generate*
 MOSTLYCLEANFILES = $(BUILT_SOURCES) $(STAMPS) *.pyc *.pyo *~ *.tmp \
 	*.loT stamp-sources-generate
-common_INCLUDES = $(USRP_INCLUDES)
+common_INCLUDES = $(USRP_INCLUDES) -I$(USB_INCLUDEDIR)
 lib_LTLIBRARIES = libusrp.la
 libusrp_la_common_LDFLAGS = $(NO_UNDEFINED) -version-info 0:0:0 $(BOOST_LDFLAGS)
 libusrp_la_common_LIBADD = \
