--- usrp/host/apps/Makefile.in_old	2009-10-26 18:49:48.000000000 -0400
+++ usrp/host/apps/Makefile.in	2009-10-26 18:50:36.000000000 -0400
@@ -745,7 +745,7 @@
 STAMPS = 
 MOSTLYCLEANFILES = $(BUILT_SOURCES) $(STAMPS) *.pyc *.pyo *~ *.tmp *.loT
 AM_CPPFLAGS = $(USRP_INCLUDES) $(BOOST_CPPFLAGS) $(CPPUNIT_INCLUDES) \
-	   $(WITH_INCLUDES)
+	   $(WITH_INCLUDES) -I$(USB_INCLUDEDIR)
 
 noinst_HEADERS = \
 	time_stuff.h			
