--- configure.orig	2017-06-13 16:58:00.000000000 -0500
+++ configure	2017-06-24 01:54:11.000000000 -0500
@@ -31260,7 +31260,7 @@
 
 
 
-SPECTRE_VERSION_REQUIRED=0.2.0
+SPECTRE_VERSION_REQUIRED=99999
 test_ps=no
 any2ppm_ps=no
 if test "x$use_ps" = "xyes"; then
@@ -31657,7 +31657,7 @@
 
 
 # poppler-0.17.4 fixes text-pattern and text-transform
-POPPLER_VERSION_REQUIRED=0.17.4
+POPPLER_VERSION_REQUIRED=99999
 test_pdf=no
 any2ppm_pdf=no
 if test "x$use_pdf" = "xyes"; then
@@ -32027,7 +32027,7 @@
 
 
 
-LIBRSVG_VERSION_REQUIRED=2.35.0
+LIBRSVG_VERSION_REQUIRED=99999
 test_svg=no
 any2ppm_svg=no
 if test "x$use_svg" = "xyes"; then
