--- configure.orig	2015-10-28 22:16:21.000000000 -0500
+++ configure	2015-10-29 17:23:41.000000000 -0500
@@ -30906,7 +30906,7 @@
 
 
 
-SPECTRE_VERSION_REQUIRED=0.2.0
+SPECTRE_VERSION_REQUIRED=99999
 test_ps=no
 any2ppm_ps=no
 if test "x$use_ps" = "xyes"; then
@@ -31303,7 +31303,7 @@
 
 
 # poppler-0.17.4 fixes text-pattern and text-transform
-POPPLER_VERSION_REQUIRED=0.17.4
+POPPLER_VERSION_REQUIRED=99999
 test_pdf=no
 any2ppm_pdf=no
 if test "x$use_pdf" = "xyes"; then
@@ -31673,7 +31673,7 @@
 
 
 
-LIBRSVG_VERSION_REQUIRED=2.35.0
+LIBRSVG_VERSION_REQUIRED=99999
 test_svg=no
 any2ppm_svg=no
 if test "x$use_svg" = "xyes"; then
