--- src/openvpn/ssl_openssl.c.orig	2020-04-16 06:26:45.000000000 -0700
+++ src/openvpn/ssl_openssl.c	2020-09-05 19:24:25.000000000 -0700
@@ -1904,7 +1904,7 @@
         crypto_msg(M_FATAL, "Cannot create SSL object");
     }
 
-#if (OPENSSL_VERSION_NUMBER < 0x1010000fL)
+#if (OPENSSL_VERSION_NUMBER < 0x1010000fL || LIBRESSL_VERSION_NUMBER)
     STACK_OF(SSL_CIPHER) *sk = SSL_get_ciphers(ssl);
 #else
     STACK_OF(SSL_CIPHER) *sk = SSL_get1_supported_ciphers(ssl);
