--- gdata/gdata.symbols.orig	2014-09-17 16:22:31.000000000 -0700
+++ gdata/gdata.symbols	2014-09-19 00:22:07.000000000 -0700
@@ -1089,8 +1089,6 @@
 gdata_freebase_topic_result_get_type
 gdata_freebase_topic_result_new
 gdata_freebase_topic_result_dup_object
-gdata_freebase_result_error_get_type
-gdata_freebase_result_error_quark
 gdata_freebase_result_get_type
 gdata_freebase_result_new
 gdata_freebase_result_dup_variant
