--- wscript.orig	2012-09-19 14:17:43.000000000 -0700
+++ wscript	2012-10-28 13:27:44.000000000 -0700
@@ -104,6 +104,7 @@ def configure (conf):
                int(given_major) == major and int(given_minor) == minor and int(given_micro) >= micro
 
     conf.check_tool ('compiler_cc')
+    conf.check_tool ('osx')
     conf.check_tool ('vala')
     conf.check_tool ('glib2')
     if not check_version (conf.env['VALAC_VERSION'], 0, 14, 0):
@@ -398,6 +399,7 @@ def configure (conf):
         Utils.pprint ('RED', 'This combination breaks the download GUI.')
         Utils.pprint ('RED', 'See https://bugs.launchpad.net/midori/+bug/780133/comments/14')
         sys.exit (1)
+    conf.env['macbundle_PATTERN'] = '%s.so'
 
 def set_options (opt):
     def is_maemo (): return os.path.exists ('/etc/osso-af-init/')
@@ -413,6 +415,7 @@ def set_options (opt):
             default=disable, help='Disable ' + desc, dest='disable_' + option_)
 
     opt.tool_options ('compiler_cc')
+    opt.tool_options ('osx')
     opt.get_option_group ('--check-c-compiler').add_option('-d', '--debug-level',
         action = 'store', default = '',
         help = 'Specify the debugging level. [\'none\', \'debug\', \'full\']',
