--- /Users/pierre/Desktop/midori-0.4.2/wscript - copie	2011-11-11 00:21:06.000000000 +0100
+++ wscript	2011-11-22 17:52:37.000000000 +0100
@@ -90,6 +90,7 @@
         return dirvalue
 
     conf.check_tool ('compiler_cc')
+    conf.check_tool ('osx')
     conf.check_tool ('vala')
     conf.check_tool ('glib2')
 
@@ -353,6 +354,7 @@
     if unique == 'yes' and conf.check_cfg (modversion='unique-1.0') == '1.0.4':
         Utils.pprint ('RED', 'unique 1.0.4 found, this version is erroneous.')
         Utils.pprint ('RED', 'Please use an older or newer version.')
+    conf.env['macbundle_PATTERN'] = '%s.so'
 
 def set_options (opt):
     def is_maemo (): return os.path.exists ('/etc/osso-af-init/')
@@ -368,6 +370,7 @@
             default=disable, help='Disable ' + desc, dest='disable_' + option_)
 
     opt.tool_options ('compiler_cc')
+    opt.tool_options ('osx')
     opt.get_option_group ('--check-c-compiler').add_option('-d', '--debug-level',
         action = 'store', default = '',
         help = 'Specify the debugging level. [\'none\', \'debug\', \'full\']',
