--- pecco_conf.h.orig	2011-12-21 14:50:08.000000000 +0900
+++ pecco_conf.h	2011-12-29 15:59:42.000000000 +0900
@@ -26,8 +26,8 @@
 //   - CEDAR (USE_CEDAR)
 //   darts-clone requires 1/2 to 1/3 memory compared to others (recommended)
 // #define USE_DARTS
-// #define USE_DARTS_CLONE
-#define USE_CEDAR
+#define USE_DARTS_CLONE
+// #define USE_CEDAR
 
 // use TR1 unordered_map / unordered_set for map / set type
 //   set to slightly speeding up the initialization
