--- pecco_conf.h.orig	2011-09-03 23:35:03.000000000 +0900
+++ pecco_conf.h	2011-09-03 23:37:46.000000000 +0900
@@ -28,8 +28,8 @@
 //   - CEDAR (USE_CEDAR)
 // EFFECT: Darts-Clone requires 1/2 to 1/3 memory compared to others and is 
 //         highly recommended; either should be set
-// #define USE_DARTS
-#define USE_DARTS_CLONE
+#define USE_DARTS
+// #define USE_DARTS_CLONE
 // #define USE_CEDAR
 
 // use TR1 unordered_map / unordered_set for map / set type
