--- mindy/etc/Makefile.in.orig	Sat Oct  4 13:21:00 2003
+++ mindy/etc/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -50,7 +50,8 @@
 
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
+LDFLAGS	=		@LDFLAGS@
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
@@ -74,7 +75,7 @@
 all:	mindyexec$(exe_ext)
 
 mindyexec$(exe_ext): mindyexec.o
-	${CC} mindyexec.o ${LIBS} -o ,$@
+	${CC} mindyexec.o ${LDFLAGS} ${LIBS} -o ,$@
 	mv ,$@ $@
 
 mindyexec.o: ${SRCDIR}/mindyexec.c
