--- Makefile.in	2014-04-26 12:58:35.000000000 +0200
+++ Makefile.in	2014-05-01 21:36:54.000000000 +0200
@@ -134,7 +134,7 @@
 	sh $(srcdir)/tty.sh tty.c
 
 comm.h: comm.c comm.sh config.h
-	AWK=$(AWK) CC="$(CC) $(CFLAGS)" srcdir=${srcdir} sh $(srcdir)/comm.sh
+	AWK=$(AWK) CC="$(CC)" srcdir=${srcdir} sh $(srcdir)/comm.sh
 
 osdef.h: osdef.sh config.h osdef.h.in
 	CPP="$(CPP) $(CPPFLAGS)" srcdir=${srcdir} sh $(srcdir)/osdef.sh
