Reported upstream: https://savannah.gnu.org/bugs/?42257
--- process.c.orig	2014-05-01 22:01:18.000000000 +0200
+++ process.c	2014-05-01 22:01:50.000000000 +0200
@@ -6282,7 +6282,7 @@
   else
     { strcpy(p = i->pw2, "\377"); l = sizeof(i->pw2) - 1; }
   if (buf && len)
-    strncpy(p, buf, 1 + (l < len) ? l : len);
+    strncpy(p, buf, 1 + ((l < len) ? l : len));
   if (!*i->name)
     Input("Screen User: ", sizeof(i->name) - 1, INP_COOKED, su_fin, (char *)i, 0);
   else if (!*i->pw1)
