--- libgda/sqlite/virtual/gda-vprovider-data-model.c.orig	2016-10-06 14:14:18.000000000 -0700
+++ libgda/sqlite/virtual/gda-vprovider-data-model.c	2016-10-06 14:28:07.000000000 -0700
@@ -24,7 +24,7 @@
 #include "gda-vprovider-data-model.h"
 #include "gda-vconnection-data-model.h"
 #include "gda-vconnection-data-model-private.h"
-#include <sqlite3.h>
+#include "../sqlite-src/sqlite3.h"
 #include <libgda/gda-connection-private.h>
 #include <libgda/gda-connection-internal.h>
 #include <libgda/gda-data-model-iter.h>
