--- configure.ac.orig	2024-11-18 22:11:34
+++ configure.ac	2024-11-18 22:14:32
@@ -381,7 +381,7 @@
 	AC_DEFINE(ENABLE_BINRELOC)
 	br_cv_binreloc=yes
     fi
-    PKG_CHECK_MODULES(MAC_INTEGRATION, ige-mac-integration, have_ige=yes, have_ige=no)
+    PKG_CHECK_MODULES(MAC_INTEGRATION, ige-mac-integration, have_ige=no, have_ige=no)
     if test x"$have_ige" = "xyes"
     then
 	AC_DEFINE(HAVE_MAC_INTEGRATION)
@@ -391,6 +391,7 @@
     LIBTOOL_EXPORT_OPTIONS=$EXPORT_SYM_REGEX
     LIBTOOL_PROV_EXPORT_OPTIONS=$EXPORT_PROV_SYM_REGEX
     LIBTOOL_UI_EXPORT_OPTIONS=$EXPORT_UI_SYM_REGEX
+    linklibext=.dylib
     AC_DEFINE(HAVE_CARBON)
     COREDEPS_LIBS="$COREDEPS_LIBS -framework Carbon"
   ;;
@@ -904,7 +905,7 @@
 #  the deprecated G_STATIC_MUTEX_INIT macro.
 DK_ARG_ENABLE_WARNINGS([COREDEPS_WFLAGS],
 	[-Wall],
-	[-Wall -Wextra -Wformat-security -Wno-address -Wno-unused-variable -Wno-unused-parameter -Wno-pointer-sign -Wno-missing-field-initializers -DGSEAL_ENABLE],
+	[-Wall -Wextra -Wformat-security -Wno-address -Wno-unused-variable -Wno-unused-parameter -Wno-pointer-sign -Wno-missing-field-initializers -Wno-incompatible-function-pointer-types -DGSEAL_ENABLE],
 	[PANGO G GDK GDK_PIXBUF GTK])
 
 AC_SUBST(COREDEPS_CFLAGS)
