diff -ur qalculate-gtk-0.9.6/configure.in qalculate-gtk-0.9.6_patched/configure.in
--- qalculate-gtk-0.9.6/configure.in	2007-05-18 02:36:10.000000000 -0600
+++ qalculate-gtk-0.9.6_patched/configure.in	2009-01-09 13:48:51.000000000 -0700
@@ -63,12 +63,6 @@
 
 AC_LANG([C++])
 
-AC_PATH_CLN(1.1.0, [
-     	LIBS="$LIBS $CLN_LIBS"
-     	CPPFLAGS="$CPPFLAGS $CLN_CPPFLAGS"
-  	AC_DEFINE(HAVE_LIBCLN,1,[If CLN can be used])
-], AC_MSG_ERROR([No suitable installed version of CLN could be found.]))
-
 PKG_CHECK_MODULES(GLIB, [
 	glib-2.0 >= 2.4
 	libxml-2.0
@@ -95,6 +89,17 @@
 AC_SUBST(QALCULATE_CFLAGS)
 AC_SUBST(QALCULATE_LIBS)
 
+PKG_CHECK_MODULES(CLN, [ cln >= 1.2 ],
+                  [have_recent_cln=yes], [have_recent_cln=no])
+AC_SUBST(CLN_CPPFLAGS)
+AC_SUBST(CLN_LIBS)
+
+if test "$have_recent_cln" = "yes" ; then
+   AC_DEFINE(HAVE_LIBCLN12, 1, [Define if CLN can be used])
+   LIBS="$LIBS $CLN_LIBS"
+   CPPFLAGS="$CPPFLAGS $CLN_CPPFLAGS"
+fi
+
 LIBGNOME_REQUIRED=2.0.0
 
 USE_LIBGNOME="auto"
diff -ur qalculate-gtk-0.9.6/src/main.cc qalculate-gtk-0.9.6_patched/src/main.cc
--- qalculate-gtk-0.9.6/src/main.cc	2007-05-18 02:33:00.000000000 -0600
+++ qalculate-gtk-0.9.6_patched/src/main.cc	2009-01-09 13:49:21.000000000 -0700
@@ -92,16 +92,18 @@
 		g_free(icon);
 	}
 	
+	gtk_init(&argc, &argv);
+
 	g_object_get(G_OBJECT(program), GNOME_PARAM_POPT_CONTEXT, &pctx, NULL);
 
-	gtk_init(&argc, &argv);
 
 #else
 
-	gtk_window_set_default_icon_from_file(PACKAGE_DATA_DIR "/pixmaps/qalculate.png", NULL);
 
 	gtk_init(&argc, &argv);
 
+	gtk_window_set_default_icon_from_file(PACKAGE_DATA_DIR "/pixmaps/qalculate.png", NULL);
+
 #endif
 
 	glade_init();
