diff --git protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp
index 2f95eca..3fe279a 100644
--- protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp
+++ protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp
@@ -430,7 +430,7 @@ FileTransfer *FileTransferManager::takeIncoming()
 
 bool FileTransferManager::isActive(const FileTransfer *ft) const
 {
-	return d->list.contains(const_cast<FileTransfer*>(ft)) > 0;
+	return d->list.contains(const_cast<FileTransfer*>(ft)) != NULL;
 }
 
 void FileTransferManager::setDisabled(const QString &ns, bool state)
