diff -Naur ../flickcurl-1.26_orig/src/activity.c src/activity.c
--- ../flickcurl-1.26_orig/src/activity.c	2014-06-01 01:16:49.000000000 +0200
+++ src/activity.c	2024-12-10 00:42:41.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/args.c src/args.c
--- ../flickcurl-1.26_orig/src/args.c	2012-08-29 05:42:02.000000000 +0200
+++ src/args.c	2024-12-10 00:43:02.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/blog.c src/blog.c
--- ../flickcurl-1.26_orig/src/blog.c	2012-08-29 04:58:29.000000000 +0200
+++ src/blog.c	2024-12-10 00:43:16.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/category.c src/category.c
--- ../flickcurl-1.26_orig/src/category.c	2014-05-31 21:54:54.000000000 +0200
+++ src/category.c	2024-12-10 00:43:32.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/collection.c src/collection.c
--- ../flickcurl-1.26_orig/src/collection.c	2014-06-01 01:16:49.000000000 +0200
+++ src/collection.c	2024-12-10 00:43:45.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/comments.c src/comments.c
--- ../flickcurl-1.26_orig/src/comments.c	2012-08-29 04:58:29.000000000 +0200
+++ src/comments.c	2024-12-10 00:44:00.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/contacts.c src/contacts.c
--- ../flickcurl-1.26_orig/src/contacts.c	2014-05-31 21:54:54.000000000 +0200
+++ src/contacts.c	2024-12-10 00:44:17.000000000 +0100
@@ -22,6 +22,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/context.c src/context.c
--- ../flickcurl-1.26_orig/src/context.c	2012-08-29 04:58:29.000000000 +0200
+++ src/context.c	2024-12-10 00:44:30.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/exif.c src/exif.c
--- ../flickcurl-1.26_orig/src/exif.c	2012-08-29 04:58:29.000000000 +0200
+++ src/exif.c	2024-12-10 00:44:43.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/gallery.c src/gallery.c
--- ../flickcurl-1.26_orig/src/gallery.c	2012-08-29 07:15:21.000000000 +0200
+++ src/gallery.c	2024-12-10 00:44:57.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/group.c src/group.c
--- ../flickcurl-1.26_orig/src/group.c	2014-05-31 21:54:54.000000000 +0200
+++ src/group.c	2024-12-10 00:45:10.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/legacy-auth.c src/legacy-auth.c
--- ../flickcurl-1.26_orig/src/legacy-auth.c	2013-06-17 19:42:59.000000000 +0200
+++ src/legacy-auth.c	2024-12-10 00:48:16.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/members.c src/members.c
--- ../flickcurl-1.26_orig/src/members.c	2012-08-29 04:58:29.000000000 +0200
+++ src/members.c	2024-12-10 00:45:35.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/method.c src/method.c
--- ../flickcurl-1.26_orig/src/method.c	2014-06-01 20:02:58.000000000 +0200
+++ src/method.c	2024-12-10 00:45:51.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/note.c src/note.c
--- ../flickcurl-1.26_orig/src/note.c	2012-08-29 04:58:29.000000000 +0200
+++ src/note.c	2024-12-10 00:46:00.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/oauth.c src/oauth.c
--- ../flickcurl-1.26_orig/src/oauth.c	2014-06-01 20:02:58.000000000 +0200
+++ src/oauth.c	2024-12-10 00:48:33.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/perms.c src/perms.c
--- ../flickcurl-1.26_orig/src/perms.c	2014-06-01 20:02:58.000000000 +0200
+++ src/perms.c	2024-12-10 00:47:39.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/photo.c src/photo.c
--- ../flickcurl-1.26_orig/src/photo.c	2014-06-01 20:02:58.000000000 +0200
+++ src/photo.c	2024-12-10 00:46:10.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 #include <ctype.h>
diff -Naur ../flickcurl-1.26_orig/src/photoset.c src/photoset.c
--- ../flickcurl-1.26_orig/src/photoset.c	2012-08-29 04:58:29.000000000 +0200
+++ src/photoset.c	2024-12-10 00:46:23.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/sha1.c src/sha1.c
--- ../flickcurl-1.26_orig/src/sha1.c	2013-04-11 06:52:16.000000000 +0200
+++ src/sha1.c	2024-12-10 00:48:03.000000000 +0100
@@ -89,6 +89,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 #include <stdint.h>
diff -Naur ../flickcurl-1.26_orig/src/size.c src/size.c
--- ../flickcurl-1.26_orig/src/size.c	2012-08-29 04:58:29.000000000 +0200
+++ src/size.c	2024-12-10 00:46:35.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/stamp-h1 src/stamp-h1
--- ../flickcurl-1.26_orig/src/stamp-h1	1970-01-01 01:00:00.000000000 +0100
+++ src/stamp-h1	2024-12-10 00:40:44.000000000 +0100
@@ -0,0 +1 @@
+timestamp for src/config.h
diff -Naur ../flickcurl-1.26_orig/src/stat.c src/stat.c
--- ../flickcurl-1.26_orig/src/stat.c	2012-08-29 04:58:29.000000000 +0200
+++ src/stat.c	2024-12-10 00:46:45.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/tags.c src/tags.c
--- ../flickcurl-1.26_orig/src/tags.c	2013-01-17 01:14:48.000000000 +0100
+++ src/tags.c	2024-12-10 00:47:20.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/ticket.c src/ticket.c
--- ../flickcurl-1.26_orig/src/ticket.c	2012-08-29 04:58:29.000000000 +0200
+++ src/ticket.c	2024-12-10 00:46:57.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
diff -Naur ../flickcurl-1.26_orig/src/user_upload_status.c src/user_upload_status.c
--- ../flickcurl-1.26_orig/src/user_upload_status.c	2014-05-31 21:54:54.000000000 +0200
+++ src/user_upload_status.c	2024-12-10 00:47:10.000000000 +0100
@@ -19,6 +19,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
 
