--- gdk/quartz/gdkdrawable-quartz.c.orig	2007-06-20 13:17:03.000000000 -0400
+++ gdk/quartz/gdkdrawable-quartz.c	2007-06-20 12:17:36.000000000 -0400
@@ -61,7 +61,14 @@
 
   gdk_drawable_get_size (drawable, &width, &height);
 
+/*
+ * Might be able to take this back as far as 1.4.2.
+ */
+#if (CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1,4,8))
+  surface = cairo_quartz_surface_create_for_cg_context (context, width, height);
+#else
   surface = cairo_quartz_surface_create (context, width, height, TRUE);
+#endif
 
   info = g_new (SurfaceInfo, 1);
   info->drawable = drawable;
