--- modules/printbackends/cups/Makefile.am.orig	2008-09-19 06:44:19.000000000 +0200
+++ modules/printbackends/cups/Makefile.am	2008-09-29 15:14:22.000000000 +0200
@@ -16,6 +16,7 @@ INCLUDES = \
 	$(GTK_DEBUG_FLAGS)
 
 LDADDS = \
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
 	$(top_builddir)/gtk/$(gtktargetlib)		\
 	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(GTK_DEP_LIBS)
--- modules/printbackends/cups/Makefile.in.orig	2008-09-24 05:11:42.000000000 +0200
+++ modules/printbackends/cups/Makefile.in	2008-09-29 15:14:34.000000000 +0200
@@ -422,6 +422,7 @@ INCLUDES = \
 
 
 LDADDS = \
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
 	$(top_builddir)/gtk/$(gtktargetlib)		\
 	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(GTK_DEP_LIBS)
--- modules/printbackends/file/Makefile.am.orig	2008-09-19 06:44:19.000000000 +0200
+++ modules/printbackends/file/Makefile.am	2008-09-29 15:09:09.000000000 +0200
@@ -28,6 +28,7 @@ libprintbackend_file_la_LDFLAGS =  \
 	-avoid-version -module $(no_undefined)
 
 libprintbackend_file_la_LIBADD = \
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
 	$(top_builddir)/gtk/$(gtktargetlib)		\
 	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(GTK_DEP_LIBS)
--- modules/printbackends/file/Makefile.in.orig	2008-09-24 05:11:42.000000000 +0200
+++ modules/printbackends/file/Makefile.in	2008-09-29 15:10:38.000000000 +0200
@@ -437,6 +437,7 @@ libprintbackend_file_la_LDFLAGS = \
 
 
 libprintbackend_file_la_LIBADD = \
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
 	$(top_builddir)/gtk/$(gtktargetlib)		\
 	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(GTK_DEP_LIBS)
--- modules/printbackends/lpr/Makefile.am.orig	2008-09-19 06:44:19.000000000 +0200
+++ modules/printbackends/lpr/Makefile.am	2008-09-29 15:13:10.000000000 +0200
@@ -16,6 +16,8 @@ INCLUDES = \
 
 LDADDS = \
 	$(GTK_DEP_LIBS)					\
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
+	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(top_builddir)/gtk/$(gtktargetlib)		
 
 backenddir = $(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/printbackends
--- modules/printbackends/lpr/Makefile.in.orig	2008-09-24 05:11:42.000000000 +0200
+++ modules/printbackends/lpr/Makefile.in	2008-09-29 15:13:20.000000000 +0200
@@ -422,6 +422,8 @@ INCLUDES = \
 
 LDADDS = \
 	$(GTK_DEP_LIBS)					\
+	$(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la  \
+	$(top_builddir)/gdk/$(gdktargetlib)		\
 	$(top_builddir)/gtk/$(gtktargetlib)		
 
 
