--- configure.orig	2008-12-01 08:54:33.000000000 -0800
+++ configure	2008-12-10 14:04:09.000000000 -0800
@@ -5172,11 +5172,6 @@
 if test "$LEX" = :; then
   LEX=${am_missing_run}flex
 fi
-if [ "$LEX" != "flex" ]; then
-	{ { $as_echo "$as_me:$LINENO: error: flex is required to create the libIDL scanner" >&5
-$as_echo "$as_me: error: flex is required to create the libIDL scanner" >&2;}
-   { (exit 1); exit 1; }; }
-fi
 LFLAGS="-8"
 
 
