--- configure.orig	2008-02-06 10:33:41.000000000 -0500
+++ configure	2008-02-06 10:33:50.000000000 -0500
@@ -5034,11 +5034,6 @@
 if test "$LEX" = :; then
   LEX=${am_missing_run}flex
 fi
-if [ "$LEX" != "flex" ]; then
-	{ { echo "$as_me:$LINENO: error: flex is required to create the libIDL scanner" >&5
-echo "$as_me: error: flex is required to create the libIDL scanner" >&2;}
-   { (exit 1); exit 1; }; }
-fi
 LFLAGS="-8"

