--- config_darwin.py.orig	2013-05-20 01:59:25.000000000 +1000
+++ config_darwin.py	2013-05-20 01:59:25.000000000 +1000
@@ -97,7 +97,7 @@
     Dependency('JPEG', 'jpeglib.h', 'libjpeg', ['jpeg']),
     Dependency('SCRAP', '','',[]),
     Dependency('PORTMIDI', 'portmidi.h', 'libportmidi', ['portmidi']),
-    FrameworkDependency('PORTTIME', 'CoreMidi.h', 'CoreMidi', 'CoreMidi'),
+    FrameworkDependency('PORTTIME', 'CoreMIDI.h', 'CoreMIDI', 'CoreMIDI'),
 ]
 
 
