--- src/dune_engine/install.ml.orig	2021-03-08 10:10:16.000000000 -0700
+++ src/dune_engine/install.ml	2021-03-14 07:58:44.000000000 -0700
@@ -133,7 +133,7 @@
       let libexec_root = libdir in
       let share_root = Path.relative destdir "share" in
       let etc_root = Path.relative destdir "etc" in
-      let doc_root = Path.relative destdir "doc" in
+      let doc_root = Path.relative share_root "doc" in
       { lib_root
       ; libexec_root
       ; share_root
