# https://trac.macports.org/ticket/65784

--- scintilla/gtk/Makefile.in.orig	2005-10-15 05:41:50.000000000 +0800
+++ scintilla/gtk/Makefile.in	2022-10-24 18:44:59.000000000 +0800
@@ -1680,7 +1680,7 @@
 
 
 scintilla-marshal.o: scintilla-marshal.h
-	gcc @PACKAGE_CFLAGS@ -c scintilla-marshal.c
+	gcc @PACKAGE_CFLAGS@ $(CFLAGS) -c scintilla-marshal.c
 
 marshallers: scintilla-marshal.list
 	glib-genmarshal --prefix scintilla_marshal $(srcdir)/scintilla-marshal.list --header > scintilla-marshal.h
