Index: amarok-2.2.0/src/app_mac.cpp
===================================================================
--- amarok-2.2.0.orig/src/app_mac.cpp	2009-09-28 19:34:03.000000000 +0200
+++ amarok-2.2.0/src/app_mac.cpp	2009-10-13 14:14:49.000000000 +0200
@@ -91,8 +91,8 @@
 setupEventHandler_mac(long handlerRef)
 {
     appleEventProcessorUPP = AEEventHandlerUPP(appleEventProcessor);
-    AEInstallEventHandler(kCoreEventClass, kAEReopenApplication, appleEventProcessorUPP, handlerRef, true);
+    AEInstallEventHandler(kCoreEventClass, kAEReopenApplication, appleEventProcessorUPP, (void*) handlerRef, true);
     macCallbackUrlHandlerUPP = AEEventHandlerUPP(macCallbackUrlHandler);
-    AEInstallEventHandler(kInternetEventClass, kAEGetURL, macCallbackUrlHandlerUPP, handlerRef, false);
+    AEInstallEventHandler(kInternetEventClass, kAEGetURL, macCallbackUrlHandlerUPP, (void*) handlerRef, false);
 }
 
