--- pcaputil.c.orig	2001-03-15 02:33:04.000000000 -0600
+++ pcaputil.c	2021-07-29 11:08:16.000000000 -0500
@@ -17,6 +17,7 @@
 #include <string.h>
 #include <err.h>
 #include <pcap.h>
+#undef BSD
 #ifdef BSD
 #include <pcap-int.h>
 #endif
@@ -60,7 +61,7 @@
 }
 
 pcap_t *
-pcap_init(char *intf, char *filter, int snaplen)
+pcap_myinit(char *intf, char *filter, int snaplen)
 {
 	pcap_t *pd;
 	u_int net, mask;
