diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index d529a1f5..8e230478 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -9,8 +9,9 @@ find_program(PG_REGRESS pg_regress
   "${PG_PKGLIBDIR}/pgxs/src/test/regress/")
 
 find_program(PG_ISOLATION_REGRESS
-  NAMES pg_isolation_regress
+  NAMES pg_isolation_regress.sh
   HINTS
+  ${PRIMARY_TEST_DIR}
   ${BINDIR}
   ${PG_PKGLIBDIR}/pgxs/src/test/isolation
   ${PG_SOURCE_DIR}/src/test/isolation)
