--- DESCRIPTION	2024-05-01 10:47:25.000000000 +0800
+++ DESCRIPTION	2024-05-04 14:20:45.000000000 +0800
@@ -37,8 +37,7 @@
 Depends: methods, R (>= 3.5.0)
 Imports: stats, utils, futile.logger, parallel, snow, codetools
 Suggests: BiocGenerics, tools, foreach, BBmisc, doParallel,
-        GenomicRanges, RNAseqData.HNRNPC.bam.chr14,
-        TxDb.Hsapiens.UCSC.hg19.knownGene, VariantAnnotation,
+        GenomicRanges, VariantAnnotation,
         Rsamtools, GenomicAlignments, ShortRead, RUnit, BiocStyle,
         knitr, batchtools, data.table
 Enhances: Rmpi
