--- DESCRIPTION	2023-10-24 21:51:58.000000000 +0800
+++ DESCRIPTION	2023-10-26 19:42:16.000000000 +0800
@@ -37,10 +37,8 @@
 Depends: methods, R (>= 3.5.0)
 Imports: stats, utils, futile.logger, parallel, snow, codetools
 Suggests: BiocGenerics, tools, foreach, BBmisc, doParallel,
-    GenomicRanges, RNAseqData.HNRNPC.bam.chr14,
-    TxDb.Hsapiens.UCSC.hg19.knownGene, VariantAnnotation, Rsamtools,
-    GenomicAlignments, ShortRead, RUnit, BiocStyle, knitr, batchtools,
-    data.table
+    GenomicRanges, VariantAnnotation, Rsamtools, data.table,
+    GenomicAlignments, ShortRead, RUnit, BiocStyle, knitr, batchtools
 Enhances: Rmpi
 Collate: AllGenerics.R DeveloperInterface.R prototype.R
     bploop.R ErrorHandling.R log.R
