--- salasaga_screencapture/Makefile.orig	2008-03-26 10:06:43.000000000 +0100
+++ salasaga_screencapture/Makefile	2008-03-26 10:07:15.000000000 +0100
@@ -20,8 +20,8 @@
 	$(CC) $(CFLAGS) $(OFILES) $(LDFLAGS) $(LIBS) -o salasaga_screencapture
 
 install:
-	$(INSTALL) -d $(bindir)
-	$(INSTALL_PROGRAM) -m 0755 salasaga_screencapture $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) -m 0755 salasaga_screencapture $(DESTDIR)$(bindir)
 
 uninstall:
 	if [ -f "$(bindir)/salasaga_screencapture" ]; then \
