--- salasaga_ide/Makefile.orig	2008-03-26 10:20:11.000000000 +0100
+++ salasaga_ide/Makefile	2008-03-26 10:21:22.000000000 +0100
@@ -590,32 +590,32 @@
 	$(CC) $(CFLAGS) $(OFILES) $(LDFLAGS) $(LIBS) -o salasaga
 
 install:
-	$(INSTALL) -d $(bindir)
-	$(INSTALL_PROGRAM) -m 0755 salasaga $(bindir)
-	$(INSTALL) -d $(datadir)/applications/
-	$(INSTALL_DATA) $(DESKTOP_FILE) $(datadir)/applications/
-	$(INSTALL) -d $(datadir)/salasaga/icons/
-	$(INSTALL) -d $(datadir)/salasaga/fonts/
-	$(INSTALL) -d $(datadir)/salasaga/fonts/BitstreamVera
-	$(INSTALL) -d $(datadir)/salasaga/fonts/BitstreamVera/fdb
-	$(INSTALL) -d $(datadir)/pixmaps/
-	cd images; $(INSTALL_DATA) $(ICON_MAIN) $(datadir)/salasaga/icons/
-	cd images; $(INSTALL_DATA) $(ICON_PNG) $(datadir)/pixmaps/
-	$(INSTALL) -d $(datadir)/salasaga/icons/scalable/
-	cd images; $(INSTALL_DATA) $(ICONS_SCALABLE) $(datadir)/salasaga/icons/scalable/
-	$(INSTALL) -d $(datadir)/salasaga/icons/scalable/control_bar/
-	cd images; cd control_bar; $(INSTALL_DATA) $(ICONS_CONTROL_BAR_SCALABLE) $(datadir)/salasaga/icons/scalable/control_bar
-	$(INSTALL) -d $(datadir)/salasaga/icons/scalable/pointers/
-	cd images; cd pointers; $(INSTALL_DATA) $(MOUSE_SCALABLE) $(datadir)/salasaga/icons/scalable/pointers/
-	$(INSTALL) -d $(datadir)/salasaga/icons/72x72/
-	cd images; $(INSTALL_DATA) $(ICONS_PNG) $(datadir)/salasaga/icons/72x72/
-	$(INSTALL) -d $(datadir)/salasaga/icons/72x72/control_bar/
-	cd images; cd control_bar; $(INSTALL_DATA) $(ICONS_CONTROL_BAR_PNG) $(datadir)/salasaga/icons/72x72/control_bar
-	$(INSTALL) -d $(datadir)/salasaga/icons/72x72/pointers/
-	cd images; cd pointers; $(INSTALL_DATA) $(MOUSE_PNG) $(datadir)/salasaga/icons/72x72/pointers/
-	cd fonts; cd BitstreamVera; cd fdb; $(INSTALL_DATA) $(FONT_BITSTREAM_VERA_FDB) $(datadir)/salasaga/fonts/BitstreamVera/fdb
-	$(INSTALL) -d $(datadir)/salasaga/sounds/
-	cd sounds; $(INSTALL_DATA) $(SOUNDS) $(datadir)/salasaga/sounds/
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) -m 0755 salasaga $(DESTDIR)$(bindir)
+	$(INSTALL) -d $(DESTDIR)$(datadir)/applications/
+	$(INSTALL_DATA) $(DESKTOP_FILE) $(DESTDIR)$(datadir)/applications/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/fonts/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/fonts/BitstreamVera
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/fonts/BitstreamVera/fdb
+	$(INSTALL) -d $(DESTDIR)$(datadir)/pixmaps/
+	cd images; $(INSTALL_DATA) $(ICON_MAIN) $(DESTDIR)$(datadir)/salasaga/icons/
+	cd images; $(INSTALL_DATA) $(ICON_PNG) $(DESTDIR)$(datadir)/pixmaps/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/scalable/
+	cd images; $(INSTALL_DATA) $(ICONS_SCALABLE) $(DESTDIR)$(datadir)/salasaga/icons/scalable/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/scalable/control_bar/
+	cd images; cd control_bar; $(INSTALL_DATA) $(ICONS_CONTROL_BAR_SCALABLE) $(DESTDIR)$(datadir)/salasaga/icons/scalable/control_bar
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/scalable/pointers/
+	cd images; cd pointers; $(INSTALL_DATA) $(MOUSE_SCALABLE) $(DESTDIR)$(datadir)/salasaga/icons/scalable/pointers/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/72x72/
+	cd images; $(INSTALL_DATA) $(ICONS_PNG) $(DESTDIR)$(datadir)/salasaga/icons/72x72/
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/72x72/control_bar/
+	cd images; cd control_bar; $(INSTALL_DATA) $(ICONS_CONTROL_BAR_PNG) $(DESTDIR)$(datadir)/salasaga/icons/72x72/control_bar
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/icons/72x72/pointers/
+	cd images; cd pointers; $(INSTALL_DATA) $(MOUSE_PNG) $(DESTDIR)$(datadir)/salasaga/icons/72x72/pointers/
+	cd fonts; cd BitstreamVera; cd fdb; $(INSTALL_DATA) $(FONT_BITSTREAM_VERA_FDB) $(DESTDIR)$(datadir)/salasaga/fonts/BitstreamVera/fdb
+	$(INSTALL) -d $(DESTDIR)$(datadir)/salasaga/sounds/
+	cd sounds; $(INSTALL_DATA) $(SOUNDS) $(DESTDIR)$(datadir)/salasaga/sounds/
 
 uninstall:
 	if [ -f "$(bindir)/salasaga" ]; then \
