--- salasaga_ide/salasaga.c.orig	2008-03-26 09:37:11.000000000 +0100
+++ salasaga_ide/salasaga.c	2008-03-26 09:37:49.000000000 +0100
@@ -287,8 +287,8 @@
 	// Check if the above directory exists
 	if (TRUE != g_file_test(icon_path->str, G_FILE_TEST_IS_DIR))
 	{
-		// First guess of icon directory didn't work, lets try /usr/share/salasaga/icons next
-		g_string_printf(icon_path, g_build_path(G_DIR_SEPARATOR_S, "/", "usr", "share", "salasaga", "icons", "72x72", NULL));
+		// First guess of icon directory didn't work, lets try @@PREFIX@@/share/salasaga/icons next
+		g_string_printf(icon_path, g_build_path(G_DIR_SEPARATOR_S, "@@PREFIX@@", "share", "salasaga", "icons", "72x72", NULL));
 
 		// Display debugging info if requested
 		if (debug_level)
