--- CMakeLists.txt.orig	2014-03-16 17:35:18.000000000 +0900
+++ CMakeLists.txt	2014-06-14 00:18:54.000000000 +0900
@@ -65,6 +65,7 @@
                                "http://www.opengl.org/"
                                FALSE "" "OpenGL rendering is generally faster on hardware with graphics acceleration.")
 add_definitions(-DQT_USE_FAST_CONCATENATION -DQT_USE_FAST_OPERATOR_PLUS)
+include_directories(${OPENGL_INCLUDE_DIR}/Versions/A/Headers)
 
 enable_testing(true)
 
