--- kstars/skyobjects/starobject.cpp.orig	2013-08-29 02:03:16.000000000 +0900
+++ kstars/skyobjects/starobject.cpp	2013-09-14 00:50:58.000000000 +0900
@@ -282,7 +282,7 @@
 
     pmms = pmMagnitudeSquared();
 
-    if( isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
+    if( std::isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
         // Ignore corrections
         *ra = ra0().Degrees();
         *dec = dec0().Degrees();
