--- inews/inews.c.orig	2004-01-29 13:14:19.000000000 +1100
+++ inews/inews.c	2012-08-03 03:29:23.000000000 +1000
@@ -47,6 +47,9 @@ extern	FILE	*ser_wr_fp;
 
 char	host_name[256];
 
+void gen_frompath();
+void append_signature();
+
 main(argc, argv)
 int	argc;
 char	*argv[];
@@ -181,6 +184,7 @@ char	*argv[];
  * The rn-style DOTDIR environmental variable is used if present.
  */
 
+void
 append_signature()
 {
 	char	line[256], sigfile[256];
@@ -254,6 +258,7 @@ append_signature()
  * a From: line in it.
  */
 
+void
 gen_frompath()
 {
 	char	*full_name;
