--- changeidvocab.c.orig	2003-07-03 08:43:20.000000000 +0900
+++ changeidvocab.c	2011-04-23 09:34:17.000000000 +0900
@@ -3,7 +3,7 @@
 #include <signal.h>
 
 int hash_size = 200000;
-char *tmpdir = "/usr/tmp";
+char *tmpdir = "/var/tmp";
 
 SLMIDNgramBuffer *NGBuf;
 int ascii_input,ascii_output;
@@ -18,7 +18,7 @@
     printf("                    [ -hash 200000 ]\n");
     printf("                    [ -buffer 100 ]\n");
     printf("                    [ -verbosity 2 ]\n");
-    printf("                    [ -temp /usr/tmp ]\n");
+    printf("                    [ -temp /var/tmp ]\n");
     printf("                    [ .idngram] [.idngram]\n");
     exit(1);
 }
--- ctext2idngram.c.orig	2009-11-13 17:49:44.000000000 +0900
+++ ctext2idngram.c	2011-04-23 09:35:57.000000000 +0900
@@ -4,7 +4,7 @@
 #include <string.h>
 
 int hash_size = 200000;
-char *tmpdir = "/usr/tmp";
+char *tmpdir = "/var/tmp";
 char delimiter = '+';
 char **vocab_tbl;
 SLMWordID *word_class_id;
--- reverseidngram.c.orig	2002-08-02 09:06:54.000000000 +0900
+++ reverseidngram.c	2011-04-23 09:36:31.000000000 +0900
@@ -12,7 +12,7 @@
 
 SLMIDNgramBuffer *NGBuf;
 int ascii_input, ascii_output;
-char *tmpdir = "/usr/tmp";
+char *tmpdir = "/var/tmp";
 
 void usage()
 {
--- text2idngram.c.orig	2002-11-11 13:38:28.000000000 +0900
+++ text2idngram.c	2011-04-23 09:37:03.000000000 +0900
@@ -3,7 +3,7 @@
 #include <signal.h>
 
 int hash_size = 200000;
-char *tmpdir = "/usr/tmp";
+char *tmpdir = "/var/tmp";
 
 SLMIDNgramBuffer *NGBuf;
 int ascii_output;
@@ -18,7 +18,7 @@
     printf("                    [ -hash 200000 ]\n");
     printf("                    [ -buffer 100 ]\n");
     printf("                    [ -verbosity 2 ]\n");
-    printf("                    [ -temp /usr/tmp ]\n");
+    printf("                    [ -temp /var/tmp ]\n");
     printf("                    [ .text] [.idngram]\n");
     exit(1);
 }
