--- demo-analogy.sh.orig	2014-02-22 20:36:04.000000000 +0900
+++ demo-analogy.sh	2014-02-22 20:27:27.000000000 +0900
@@ -1,11 +1,13 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
 echo -----------------------------------------------------------------------------------------------------
 echo Note that for the word analogy to perform well, the models should be trained on much larger data sets
 echo Example input: paris france berlin
 echo -----------------------------------------------------------------------------------------------------
-time ./word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
-./word-analogy vectors.bin
+if [ ! -e vectors.bin ]; then
+  time @EXECDIR@/word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
+fi
+@EXECDIR@/word-analogy vectors.bin
--- demo-classes.sh.orig	2014-02-22 20:36:09.000000000 +0900
+++ demo-classes.sh	2014-02-22 20:22:53.000000000 +0900
@@ -1,8 +1,8 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
-time ./word2vec -train text8 -output classes.txt -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -classes 500
+time @EXECDIR@/word2vec -train text8 -output classes.txt -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -classes 500
 sort classes.txt -k 2 -n > classes.sorted.txt
 echo The word classes were saved to file classes.sorted.txt
--- demo-phrase-accuracy.sh.orig	2014-02-22 20:36:25.000000000 +0900
+++ demo-phrase-accuracy.sh	2014-02-22 20:29:40.000000000 +0900
@@ -1,12 +1,14 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
 echo ----------------------------------------------------------------------------------------------------------------
 echo Note that the accuracy and coverage of the test set questions is going to be low with this small training corpus
 echo To achieve better accuracy, larger training set is needed
 echo ----------------------------------------------------------------------------------------------------------------
-time ./word2phrase -train text8 -output text8-phrase -threshold 500 -debug 2 -min-count 3
-time ./word2vec -train text8-phrase -output vectors-phrase.bin -cbow 0 -size 300 -window 10 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1 -min-count 3
-./compute-accuracy vectors-phrase.bin <questions-phrases.txt
+if [ ! -e vectors-phrase.bin ]; then
+  time @EXECDIR@/word2phrase -train text8 -output text8-phrase -threshold 500 -debug 2 -min-count 3
+  time @EXECDIR@/word2vec -train text8-phrase -output vectors-phrase.bin -cbow 0 -size 300 -window 10 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1 -min-count 3
+fi
+@EXECDIR@/compute-accuracy vectors-phrase.bin < @EXDIR@/questions-phrases.txt
--- demo-phrases.sh.orig	2014-02-22 20:36:17.000000000 +0900
+++ demo-phrases.sh	2014-02-22 20:30:19.000000000 +0900
@@ -1,8 +1,10 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
-time ./word2phrase -train text8 -output text8-phrase -threshold 500 -debug 2
-time ./word2vec -train text8-phrase -output vectors-phrase.bin -cbow 0 -size 300 -window 10 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
-./distance vectors-phrase.bin
\ No newline at end of file
+if [ ! -e vectors-phrase.bin ]; then
+  time @EXECDIR@/word2phrase -train text8 -output text8-phrase -threshold 500 -debug 2
+  time @EXECDIR@/word2vec -train text8-phrase -output vectors-phrase.bin -cbow 0 -size 300 -window 10 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
+fi
+@EXECDIR@/distance vectors-phrase.bin
--- demo-word-accuracy.sh.orig	2014-02-22 20:36:32.000000000 +0900
+++ demo-word-accuracy.sh	2014-02-22 20:31:16.000000000 +0900
@@ -1,8 +1,10 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
-time ./word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
-./compute-accuracy vectors.bin 30000 < questions-words.txt
-# to compute accuracy with the full vocabulary, use: ./compute-accuracy vectors.bin < questions-words.txt
+if [ ! -e vectors.bin ]; then
+  time @EXECDIR@/word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
+fi
+@EXECDIR@/compute-accuracy vectors.bin 30000 < @EXDIR@/questions-words.txt
+# to compute accuracy with the full vocabulary, use: @EXECDIR@/compute-accuracy vectors.bin < @EXDIR@/questions-words.txt
--- demo-word.sh.orig	2014-02-22 20:36:47.000000000 +0900
+++ demo-word.sh	2014-02-22 20:31:57.000000000 +0900
@@ -1,7 +1,9 @@
-make
+#!/bin/sh
 if [ ! -e text8 ]; then
-  wget http://mattmahoney.net/dc/text8.zip -O text8.gz
-  gzip -d text8.gz -f
+  curl -O http://mattmahoney.net/dc/text8.zip
+  unzip text8.zip
 fi
-time ./word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
-./distance vectors.bin
\ No newline at end of file
+if [ ! -e vectors.bin ]; then
+  time @EXECDIR@/word2vec -train text8 -output vectors.bin -cbow 0 -size 200 -window 5 -negative 0 -hs 1 -sample 1e-3 -threads 12 -binary 1
+fi
+@EXECDIR@/distance vectors.bin
