--- DESCRIPTION	2024-10-05 04:54:43.000000000 +0800
+++ DESCRIPTION	2024-11-25 01:53:01.000000000 +0800
@@ -42,7 +42,6 @@
     glue (>= 1.8.0),
     htmltools (>= 0.5.8.1),
     htmlwidgets (>= 1.6.4),
-    juicyjuice (>= 0.1.0),
     magrittr (>= 2.0.3),
     markdown (>= 1.13),
     reactable (>= 0.4.4),
@@ -58,7 +57,6 @@
     ggplot2,
     grid,
     gtable,
-    katex (>= 1.4.1),
     knitr,
     lubridate,
     magick,
@@ -70,7 +68,6 @@
     shiny (>= 1.9.1),
     testthat (>= 3.1.9),
     tidyr,
-    webshot2 (>= 0.1.0),
     withr
 Config/Needs/coverage: officer
 Config/Needs/website: quarto
