--- DESCRIPTION.orig	2023-04-01 05:59:41.000000000 +0800
+++ DESCRIPTION	2023-04-07 13:09:49.000000000 +0800
@@ -43,7 +43,6 @@
     glue (>= 1.6.2),
     htmltools (>= 0.5.4),
     htmlwidgets (>= 1.6.1),
-    juicyjuice (>= 0.1.0),
     magrittr (>= 2.0.2),
     markdown (>= 1.5),
     reactable (>= 0.4.3),
@@ -57,6 +56,7 @@
     covr,
     digest (>= 0.6.31),
     ggplot2,
+    juicyjuice (>= 0.1.0),
     knitr,
     lubridate,
     paletteer,
