--- DESCRIPTION	2024-01-18 12:28:04.000000000 +0800
+++ DESCRIPTION	2024-01-18 18:20:16.000000000 +0800
@@ -43,7 +43,6 @@
     glue (>= 1.6.2),
     htmltools (>= 0.5.4),
     htmlwidgets (>= 1.6.1),
-    juicyjuice (>= 0.1.0),
     magrittr (>= 2.0.2),
     markdown (>= 1.5),
     reactable (>= 0.4.3),
@@ -67,8 +66,7 @@
     rvest,
     shiny (>= 1.7.4),
     testthat (>= 3.1.9),
-    tidyr,
-    webshot2 (>= 0.1.0)
+    tidyr
 Roxygen: list(markdown = TRUE)
 Collate: 
     'as_data_frame.R'
