--- doc/tinyproxy.conf	Mon Jun 23 23:14:32 2003
+++ doc/tinyproxy.conf.new	Sun Dec 14 22:58:04 2003
@@ -51,20 +51,20 @@
 # HTML file defined with an ErrorFile keyword for the HTTP error
 # that has occured.
 #
-DefaultErrorFile "/usr/share/tinyproxy/default.html"
+DefaultErrorFile "PREFIX/share/tinyproxy/default.html"
 
 #
 # StatFile: The HTML file that gets sent when a request is made
 # for the stathost.  If this file doesn't exist a basic page is
 # hardcoded in tinyproxy.
 #
-StatFile "/usr/share/tinyproxy/stats.html"
+StatFile "PREFIX/share/tinyproxy/stats.html"
 
 #
 # Where to log the information. Either LogFile or Syslog should be set,
 # but not both.
 #
-Logfile "/var/log/tinyproxy.log"
+Logfile "PREFIX/var/log/tinyproxy.log"
 # Syslog On
 
 #
@@ -85,7 +85,7 @@
 # PidFile: Write the PID of the main tinyproxy thread to this file so it
 # can be used for signalling purposes.
 #
-PidFile "/var/run/tinyproxy.pid"
+PidFile "PREFIX/var/run/tinyproxy.pid"
 
 #
 # Include the X-Tinyproxy header, which has the client's IP address when
@@ -167,7 +167,7 @@
 # are tested against the controls based on order.
 #
 Allow 127.0.0.1
-Allow 192.168.1.0/25
+#Allow 192.168.1.0/25
 
 #
 # The "Via" header is required by the HTTP RFC, but using the real host name
