--- macosx/Info.plist.in.orig	2019-02-26 18:37:48.000000000 +0000
+++ macosx/Info.plist.in	2019-11-04 11:15:03.000000000 +0000
@@ -33,6 +33,8 @@
     <true/>
     <key>NSHumanReadableCopyright</key>
     <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+    <key>NSPrincipalClass</key>
+    <string>NSApplication</string>
     <key>NOTE</key>
     <string>Icon is disabled from dock. Only systray is used.</string>
     <key>LSUIElement</key>
