--- ./scripts/build/combine_libs
+++ ./scripts/build/combine_libs
@@ -22,6 +22,15 @@
     mkdir "$dir"
     cd "$dir" >/dev/null
     "${AR:-ar}" x "$abs"
+
+    # Delete the "__.SYMDEF*" file if present, it will be generated
+    # by ranlib at the end on the combined library
+    if [ -f "__.SYMDEF SORTED" ]; then
+        rm -f "__.SYMDEF SORTED"
+    fi
+        if [ -f "__.SYMDEF" ]; then
+        rm -f "__.SYMDEF"
+    fi
 done
 
 cd "$TMPDIR" >/dev/null
