--- src/engine/CMakeLists.txt.orig	2012-08-13 18:10:41.000000000 +0900
+++ src/engine/CMakeLists.txt	2012-11-12 02:04:41.000000000 +0900
@@ -84,20 +84,20 @@
 
 set_target_properties(kanagramengine PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} DEFINE_SYMBOL MAKE_KANAGRAM_ENGINE_LIB)
 
-if(APPLE)
-    #hack for being able to set headers as public in a osx framework
-    list(APPEND KanagramEngine_HEADERS
-        ${KanagramEngine_HEADERS}
-    )
-
-    set_target_properties(KanagramEngine PROPERTIES FRAMEWORK TRUE)
-    set_target_properties(KanagramEngine PROPERTIES BUILD_WITH_INSTALL_RPATH 1 INSTALL_NAME_DIR "@executable_path/../Frameworks")
-    set_target_properties(KanagramEngine PROPERTIES PUBLIC_HEADER "${KanagramEngine_HEADERS}")
-
-    set(MACOSX_FRAMEWORK_IDENTIFIER engine)
-    set(MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${GENERIC_LIB_SOVERSION})
-    set(MACOSX_FRAMEWORK_BUNDLE_VERSION ${GENERIC_LIB_VERSION})
-endif()
+#if(APPLE)
+#    #hack for being able to set headers as public in a osx framework
+#    list(APPEND KanagramEngine_HEADERS
+#        ${KanagramEngine_HEADERS}
+#    )
+#
+#    set_target_properties(KanagramEngine PROPERTIES FRAMEWORK TRUE)
+#    set_target_properties(KanagramEngine PROPERTIES BUILD_WITH_INSTALL_RPATH 1 INSTALL_NAME_DIR "@executable_path/../Frameworks")
+#    set_target_properties(KanagramEngine PROPERTIES PUBLIC_HEADER "${KanagramEngine_HEADERS}")
+
+#    set(MACOSX_FRAMEWORK_IDENTIFIER engine)
+#    set(MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${GENERIC_LIB_SOVERSION})
+#    set(MACOSX_FRAMEWORK_BUNDLE_VERSION ${GENERIC_LIB_VERSION})
+#endif()
 
 install(
     TARGETS     kanagramengine
