*** koffice-1.5.1/karbon/Makefile.am~	Thu May 11 03:35:25 2006
--- koffice-1.5.1/karbon/Makefile.am	Mon May 22 20:38:02 2006
***************
*** 30,36 ****
  	$(LIBART_LIBS) \
  	$(LIBFONTCONFIG_LIBS) \
  	$(LIBMAGICK_LIBS) \
! 	$(LIBFREETYPE_LIBS)
  
  libkarbonpart_la_SOURCES = karbon_factory_init.cc
  libkarbonpart_la_LDFLAGS = $(KDE_PLUGIN) $(LIBFREETYPE_RPATH)
--- 30,37 ----
  	$(LIBART_LIBS) \
  	$(LIBFONTCONFIG_LIBS) \
  	$(LIBMAGICK_LIBS) \
! 	$(LIBFREETYPE_LIBS) \
! 	$(LIB_X11)
  
  libkarbonpart_la_SOURCES = karbon_factory_init.cc
  libkarbonpart_la_LDFLAGS = $(KDE_PLUGIN) $(LIBFREETYPE_RPATH)
