--- CPV/src/Makefile.orig	2015-12-16 14:19:27.000000000 -0500
+++ CPV/src/Makefile	2015-12-16 14:20:02.000000000 -0500
@@ -115,7 +115,7 @@
 manycp.x : manycp.o libcp.a $(QEMODS) $(LIBOBJS)
 	$(LD) $(LDFLAGS) -o manycp.x manycp.o \
 		libcp.a $(QEMODS) $(LIBOBJS) $(LIBS)
-	- ( cd ../../bin ; ln -fs ../CPV/src/cp.x . )
+	- ( cd ../../bin ; ln -fs ../CPV/src/manycp.x . )
 
 cp.x : cprstart.o libcp.a $(QEMODS) $(LIBOBJS)
 	$(LD) $(LDFLAGS) -o cp.x cprstart.o \
