--- install/m4/x_ac_qe_mpif90.m4.orig	2019-10-24 07:06:22.000000000 -0700
+++ install/m4/x_ac_qe_mpif90.m4	2019-10-24 07:06:38.000000000 -0700
@@ -87,7 +87,7 @@
 # check Fortran 90 compiler
 
 # clear cached values
-unset FC ac_cv_prog_ac_ct_FC ac_cv_fc_compiler_gnu ac_cv_prog_fc_g
+unset ac_cv_prog_ac_ct_FC ac_cv_fc_compiler_gnu ac_cv_prog_fc_g
 
 if test "$use_parallel" -eq 0 ; then
 # serial case - use F90 if set
@@ -161,7 +161,6 @@
               AC_MSG_WARN([parallel and serial compiler are the same])
 	   fi
 	fi
-	f90=$f90_in_mpif90
         ;;
 esac
 AC_FC_SRCEXT(f90)
