--- Makefile.orig	2016-10-18 14:56:06.000000000 -0700
+++ Makefile	2016-10-18 14:59:34.000000000 -0700
@@ -64,6 +64,9 @@
 # If "|| exit 1" is not present, the error code from make in subdirectories
 # is not returned and make goes on even if compilation has failed
 
+all-j :
+	$(MAKE) pw && $(MAKE) pp && $(MAKE) ph && $(MAKE) ld1 && $(MAKE) upf && $(MAKE) cp
+
 pw : bindir libfft libla mods liblapack libs libiotk 
 	if test -d PW ; then \
 	( cd PW ; $(MAKE) TLDEPS= all || exit 1) ; fi
@@ -77,8 +80,9 @@
 	( cd CPV ; $(MAKE) TLDEPS= all || exit 1) ; fi
 
 ph : bindir libfft libla mods libs pw lrmods
-	if test -d PHonon; then \
-	(cd PHonon; $(MAKE) all || exit 1) ; fi
+	( cd PHonon/PH && $(MAKE) all ; cd .. || exit 1 )
+	( cd PHonon/Gamma && $(MAKE) all ; cd .. || exit 1 )
+	( cd PHonon/FD && $(MAKE) all ; cd .. || exit 1 )
 
 neb : bindir libfft libla mods libs pw
 	if test -d NEB; then \
