--- config/rules.mk.orig	2006-12-22 14:50:41.000000000 +0200
+++ config/rules.mk	2007-09-22 22:31:47.000000000 +0200
@@ -424,7 +424,7 @@
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS	+= -bundle
 else
-EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @executable_path/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
+EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name $(libdir)/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
 endif
 endif
 endif
--- config/config.mk.orig	2007-02-16 03:19:06.000000000 +0200
+++ config/config.mk	2007-09-22 23:44:44.000000000 +0200
@@ -745,7 +745,7 @@
 # Tell the linker where NSS is, if we're building crypto
 ifeq ($(OS_ARCH),Darwin)
 ifeq (,$(findstring crypto,$(MOZ_META_COMPONENTS)))
-MOZ_COMPONENTLIB_EXTRA_LIBS = $(foreach library, $(patsubst -l%, $(LIB_PREFIX)%$(DLL_SUFFIX), $(filter -l%, $(NSS_LIBS))), -dylib_file @executable_path/$(library):$(DIST)/bin/$(library))
+MOZ_COMPONENTLIB_EXTRA_LIBS = $(foreach library, $(patsubst -l%, $(LIB_PREFIX)%$(DLL_SUFFIX), $(filter -l%, $(NSS_LIBS))), -dylib_file $(libdir)/$(library):$(DIST)/bin/$(library))
 endif
 endif
 endif
--- directory/c-sdk/configure.in.orig	2007-07-15 16:41:07.000000000 +0200
+++ directory/c-sdk/configure.in	2007-09-23 10:38:22.000000000 +0200
@@ -921,7 +921,7 @@ case "$target" in
             CPU_ARCH=ppc
             ;;
     esac
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name \$(libdir)/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
--- directory/c-sdk/configure.orig	2006-02-23 01:58:13.000000000 +0100
+++ directory/c-sdk/configure	2007-09-23 10:38:35.000000000 +0200
@@ -3330,7 +3330,7 @@ EOF
             CPU_ARCH=ppc
             ;;
     esac
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name \$(libdir)/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
--- nsprpub/configure.in.orig	2007-05-03 03:30:08.000000000 +0200
+++ nsprpub/configure.in	2007-09-23 10:39:41.000000000 +0200
@@ -979,7 +979,7 @@ case "$target" in
             ;;
     esac
     DSO_CFLAGS=-fPIC
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@ -headerpad_max_install_names'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name \$(libdir)/$@ -headerpad_max_install_names'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
--- nsprpub/configure.orig	2007-05-03 03:30:08.000000000 +0200
+++ nsprpub/configure	2007-09-23 10:39:24.000000000 +0200
@@ -3326,7 +3326,7 @@ EOF
             ;;
     esac
     DSO_CFLAGS=-fPIC
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@ -headerpad_max_install_names'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name \$(libdir)/$@ -headerpad_max_install_names'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
--- security/coreconf/Darwin.mk.orig	2005-01-20 09:12:48.000000000 +0100
+++ security/coreconf/Darwin.mk	2007-09-23 12:49:02.000000000 +0200
@@ -111,7 +111,7 @@ endif
 
 DSO_CFLAGS	= -fPIC
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@) -headerpad_max_install_names
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name \$(libdir)/$(notdir $@) -headerpad_max_install_names
 
 MKSHLIB		= $(CC) $(DSO_LDOPTS) $(DARWIN_SDK_SHLIBFLAGS)
 DLL_SUFFIX	= dylib
--- security/nss/cmd/platlibs.mk.orig	2004-10-16 01:52:29.000000000 +0200
+++ security/nss/cmd/platlibs.mk	2007-09-23 12:48:37.000000000 +0200
@@ -226,7 +226,7 @@ endif
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file $(libdir)/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 
--- security/nss/lib/smime/config.mk.orig	2004-10-16 01:52:34.000000000 +0200
+++ security/nss/lib/smime/config.mk	2007-09-23 12:48:24.000000000 +0200
@@ -76,7 +76,7 @@ EXTRA_SHARED_LIBS += \
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file $(libdir)/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
--- security/nss/lib/ssl/config.mk.orig	2004-10-16 01:52:35.000000000 +0200
+++ security/nss/lib/ssl/config.mk	2007-09-23 12:48:15.000000000 +0200
@@ -109,7 +109,7 @@ EXTRA_SHARED_LIBS += -lbe
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file $(libdir)/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 ifeq ($(OS_TARGET),SunOS)
