https://github.com/OpenLightingProject/ola/commit/f907b4c992e85bfb30559b5855c41194d2c52050
--- common/base/Init.cpp.orig
+++ common/base/Init.cpp
@@ -397,7 +397,7 @@ void ClockInit() {
   TimeStamp now_monotonic;
   TimeStamp now_realtime;
 #ifndef CLOCK_MONOTONIC
-  OLA_DEBUG << "Monotonic clock unavailable. Falling back to real time clock."
+  OLA_DEBUG << "Monotonic clock unavailable. Falling back to real time clock.";
 #endif
   clock.CurrentMonotonicTime(&now_monotonic);
   clock.CurrentRealTime(&now_realtime);
