--- setup.py.orig	2014-01-15 01:42:58.000000000 +0400
+++ setup.py	2014-01-15 01:43:56.000000000 +0400
@@ -32,8 +32,8 @@
 extens = [Extension('backports/lzma/_lzma',
                     ['backports/lzma/_lzmamodule.c'],
                     libraries = ['lzma'],
-                    include_dirs = [os.path.join(home, 'include'), '/opt/local/include', '/usr/local/include'],
-                    library_dirs = [os.path.join(home, 'lib'), '/opt/local/lib', '/usr/local/lib']
+                    include_dirs = ['@@PREFIX@@/include'],
+                    library_dirs = ['@@PREFIX@@/lib']
                     )]
 
 descr = "Backport of Python 3.3's 'lzma' module for XZ/LZMA compressed files."
