--- Modules/readline.c.orig	2023-09-06 07:57:19
+++ Modules/readline.c	2023-09-10 01:43:53
@@ -442,12 +442,8 @@ readline_set_completion_display_matches_hook_impl(PyOb
        default completion display. */
     rl_completion_display_matches_hook =
         readlinestate_global->completion_display_matches_hook ?
-#if defined(_RL_FUNCTION_TYPEDEF)
         (rl_compdisp_func_t *)on_completion_display_matches_hook : 0;
-#else
-        (VFunction *)on_completion_display_matches_hook : 0;
 #endif
-#endif
     return result;
 
 }
