--- setup.py.orig	2017-08-09 17:08:33.000000000 +1000
+++ setup.py	2017-09-12 21:20:26.000000000 +1000
@@ -39,7 +39,7 @@
 COMPILED_WITH_PYDEBUG = ('--with-pydebug' in sysconfig.get_config_var("CONFIG_ARGS"))
 
 # This global variable is used to hold the list of modules to be disabled.
-disabled_module_list = []
+disabled_module_list = ["_tkinter", "_gdbm"]
 
 def add_dir_to_list(dirlist, dir):
     """Add the directory 'dir' to the list 'dirlist' (after any relative
@@ -1688,7 +1688,8 @@
 
     def detect_tkinter(self, inc_dirs, lib_dirs):
         # The _tkinter module.
-
+        # Skip building; provided by a separate port in MacPorts.
+        return
         # Check whether --with-tcltk-includes and --with-tcltk-libs were
         # configured or passed into the make target.  If so, use these values
         # to build tkinter and bypass the searches for Tcl and TK in standard
