--- etc/hub.bash_completion.sh.orig	2013-10-11 17:03:50.000000000 +0200
+++ etc/hub.bash_completion.sh	2013-10-12 14:24:47.000000000 +0200
@@ -1,15 +1,16 @@
 # hub tab-completion script for bash.
 # This script complements the completion script that ships with git.
 
-# Check that git tab completion is available
-if declare -F _git > /dev/null; then
-  # Duplicate and rename the 'list_all_commands' function
-  eval "$(declare -f __git_list_all_commands | \
-        sed 's/__git_list_all_commands/__git_list_all_commands_without_hub/')"
+# Ensure that git tab completion is available
+_xfunc git __git_list_all_commands
 
-  # Wrap the 'list_all_commands' function with extra hub commands
-  __git_list_all_commands() {
-    cat <<-EOF
+# Duplicate and rename the 'list_all_commands' function
+eval "$(declare -f __git_list_all_commands | \
+    sed 's/__git_list_all_commands/__git_list_all_commands_without_hub/')"
+
+# Wrap the 'list_all_commands' function with extra hub commands
+__git_list_all_commands() {
+cat <<-EOF
 alias
 pull-request
 fork
@@ -17,9 +18,11 @@
 browse
 compare
 EOF
-    __git_list_all_commands_without_hub
-  }
+__git_list_all_commands_without_hub
+}
 
-  # Ensure cached commands are cleared
-  __git_all_commands=""
-fi
+# Ensure cached commands are cleared
+__git_all_commands=""
+
+# Register completion for the hub command
+__git_complete hub __git_main
