--- setup.py.orig	2012-07-18 23:25:03.000000000 +0200
+++ setup.py	2012-07-18 23:25:47.000000000 +0200
@@ -23,7 +23,6 @@
 from distutils import log
 from subprocess import Popen, PIPE
 sys.path.append("lib")
-from contrib.debian.installer import DebianInstaller
 
 #
 # Build the command line script
@@ -365,12 +364,7 @@
             raise OSError("not found: %s" % ", ".join(mis_stys))
 
     def run(self):
-        if self.install_layout == "deb":
-            db = DebianInstaller(self)
-        else:
-            db = None
-
-        if not(db) and not(self.nodeps):
+        if not(self.nodeps):
             try:
                 self.check_xslt_dependencies()
                 self.check_util_dependencies()
@@ -379,16 +373,12 @@
                 print >>sys.stderr, "Error: %s" % e
                 sys.exit(1)
 
-        if db: db.adapt_paths()
-
         # If no build is required, at least build the script
         if self.skip_build:
             self.run_command('build_scripts')
 
         install.run(self)
 
-        if db: db.finalize()
-
 
 class InstallData(install_data):
 
