--- bigfloat/test/test_mpfr.py.orig	2019-10-17 20:39:10.000000000 -0400
+++ bigfloat/test/test_mpfr.py	2019-10-17 20:42:53.000000000 -0400
@@ -26,6 +26,7 @@
     _LONG_MIN, _LONG_MAX,
 
     MPFR_RNDN, MPFR_RNDZ, MPFR_RNDU, MPFR_RNDD, MPFR_RNDA,
+    MPFR_PREC_MIN, MPFR_PREC_MAX,
 
     # Base extension type
     Mpfr_t,
@@ -375,7 +376,10 @@
             Mpfr(10, 11)
 
         with self.assertRaises(ValueError):
-            Mpfr(1)
+            Mpfr(MPFR_PREC_MIN - 1)
+
+        with self.assertRaises(ValueError):
+            Mpfr(MPFR_PREC_MAX + 1)
 
     def test_constructor(self):
         x = Mpfr(10)
