--- cmake/admFFmpegUtil.cmake.orig	2010-09-08 07:09:59.000000000 +0200
+++ cmake/admFFmpegUtil.cmake	2010-09-08 07:10:38.000000000 +0200
@@ -11,9 +11,9 @@
 	getFfmpegVersionFromHeader("libpostproc" "${sourceDir}/libpostproc/postprocess.h" LIBPOSTPROC_VERSION_MAJOR LIBPOSTPROC_VERSION)
 	getFfmpegVersionFromHeader("libswscale" "${sourceDir}/libswscale/swscale.h" LIBSWSCALE_VERSION_MAJOR LIBSWSCALE_VERSION)
 
-	if (UNIX AND NOT APPLE)
+	if (UNIX)
 		set(LIBAVCODEC_ADM ADM5)
-	endif (UNIX AND NOT APPLE)
+	endif (UNIX)
 
 	if (APPLE)
 		set(LIBAVCODEC_LIB lib${LIBAVCODEC_ADM}avcodec.${LIBAVCODEC_VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX})
--- cmake/admFFmpegBuild.cmake.orig	2010-09-08 07:08:21.000000000 +0200
+++ cmake/admFFmpegBuild.cmake	2010-09-08 07:09:09.000000000 +0200
@@ -38,9 +38,9 @@
 		patch_file("${FFMPEG_SOURCE_DIR}" "${patchFile}")
 	endforeach(patchFile)
 
-	if (UNIX AND NOT APPLE)
+	if (UNIX)
 		patch_file("${FFMPEG_SOURCE_DIR}" "${CMAKE_SOURCE_DIR}/cmake/patches/common.mak.diff")
-	endif (UNIX AND NOT APPLE)
+	endif (UNIX)
 
 	message("")
 endif (FFMPEG_PERFORM_PATCH)
@@ -145,10 +145,10 @@
 	execute_process(COMMAND ${CMAKE_COMMAND} -E copy "./libavutil/avconfig.h" "${CMAKE_BINARY_DIR}/config/libavutil"
 					WORKING_DIRECTORY "${FFMPEG_BINARY_DIR}")
 
-	if (UNIX AND NOT APPLE)
+	if (UNIX)
 		find_patch()
 		patch_file("${FFMPEG_BINARY_DIR}" "${CMAKE_SOURCE_DIR}/cmake/patches/config.mak.diff")
-	endif (UNIX AND NOT APPLE)
+	endif (UNIX)
 
 	message("")
 endif (FFMPEG_PERFORM_BUILD)
--- avidemux/CMakeLists.txt.orig	2010-09-08 08:16:50.000000000 +0200
+++ avidemux/CMakeLists.txt	2010-09-08 08:19:55.000000000 +0200
@@ -483,12 +483,12 @@
              ADD_CUSTOM_COMMAND(TARGET ${t}
                       POST_BUILD
                       COMMAND mkdir -p ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
-                      COMMAND find ${CMAKE_BINARY_DIR}/avidemux -name '*.so*' | xargs ln -sft ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
+                      COMMAND find ${CMAKE_BINARY_DIR}/avidemux -name '*.so*' -o -name '*.dylib' | xargs -I{} ln -sf {} ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
                       WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
                       COMMENT "Creating temporary lib directory for proper plugins linking")
        ENDFOREACH (t )
 ENDMACRO (ADM_SYMLINK_LIBS)
-IF (UNIX AND NOT APPLE)
+IF (UNIX)
    # Some plugins link to ADM_coreImage (e.g., ADM_vf_addborders) and thus need the symlink in ${CMAKE_BINARY_DIR}/lib${LIB_SUFFIX}
    ADM_SYMLINK_LIBS(ffmpeg ADM_coreImage)
-ENDIF(UNIX AND NOT APPLE)
+ENDIF(UNIX)
--- cmake/patches/config.mak.diff.orig	2010-09-08 10:07:32.000000000 +0200
+++ cmake/patches/config.mak.diff	2010-09-08 10:06:57.000000000 +0200
@@ -1,11 +1,15 @@
---- config.mak.old	2009-05-01 19:26:13.808782802 +0200
-+++ config.mak	2009-05-01 18:10:25.000085291 +0200
-@@ -60,7 +60,7 @@
- BUILD_SHARED=yes
- PIC=-fPIC -DPIC
+--- config.mak.orig	2010-09-08 09:50:37.000000000 +0200
++++ config.mak	2010-09-08 10:05:45.000000000 +0200
+@@ -59,9 +59,9 @@
+ EXTRALIBS=  -lz -lbz2 -lm
+ INSTALL=install
  LIBTARGET=
 -SLIBNAME=$(SLIBPREF)$(FULLNAME)$(SLIBSUF)
+-SLIBNAME_WITH_VERSION=$(SLIBPREF)$(FULLNAME).$(LIBVERSION)$(SLIBSUF)
+-SLIBNAME_WITH_MAJOR=$(SLIBPREF)$(FULLNAME).$(LIBMAJOR)$(SLIBSUF)
 +SLIBNAME=$(SLIBPREF)ADM5$(FULLNAME)$(SLIBSUF)
- SLIBNAME_WITH_VERSION=$(SLIBNAME).$(LIBVERSION)
- SLIBNAME_WITH_MAJOR=$(SLIBNAME).$(LIBMAJOR)
++SLIBNAME_WITH_VERSION=$(SLIBPREF)ADM5$(FULLNAME).$(LIBVERSION)$(SLIBSUF)
++SLIBNAME_WITH_MAJOR=$(SLIBPREF)ADM5$(FULLNAME).$(LIBMAJOR)$(SLIBSUF)
  SLIB_CREATE_DEF_CMD=
+ SLIB_EXTRA_CMD=
+ SLIB_INSTALL_EXTRA_CMD=
