--- avidemux_plugins/CMakeLists.txt.orig	2021-01-31 09:02:10.000000000 -0500
+++ avidemux_plugins/CMakeLists.txt	2021-01-31 09:03:26.000000000 -0500
@@ -265,7 +265,7 @@
 ADD_SUBDIRECTORY(ADM_videoEncoder)
 ADD_SUBDIRECTORY(ADM_videoFilters6)
 IF(DO_QT4)
-        add_subdirectory(ADM_demuxers/VapourSynth)
+    #    add_subdirectory(ADM_demuxers/VapourSynth)
     #    add_subdirectory(ADM_scriptEngines)
         #IF(USE_OPENGL AND NOT WIN32)
         IF(USE_OPENGL )
