diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0905b5dafd..5d929b3679 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 
 if(APPLE)
     SET(CMAKE_MACOSX_RPATH TRUE)
-    SET(CMAKE_INSTALL_RPATH "@loader_path/../${CMAKE_INSTALL_LIBDIR}/inkscape")
+    SET(CMAKE_INSTALL_RPATH "@loader_path/../${CMAKE_INSTALL_LIBDIR}")
 else()
     SET(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}/inkscape")
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d317cbeb14..6fbf75b1f4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -384,5 +384,5 @@ if(WIN32)
     install(TARGETS inkscape_com inkview_com)
 endif()
 if(BUILD_SHARED_LIBS)
-    install(TARGETS inkscape_base LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/inkscape)
+    install(TARGETS inkscape_base LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
