--- src/zabbix_server/poller/checks_snmp.c.orig	2010-08-17 17:51:56.000000000 +0200
+++ src/zabbix_server/poller/checks_snmp.c	2010-08-17 17:52:51.000000000 +0200
@@ -595,9 +595,9 @@
 				SET_UI64_RESULT(value, (((zbx_uint64_t)vars->val.counter64->high) << 32) +
 						(zbx_uint64_t)vars->val.counter64->low);
 			}
-			else if (vars->type == ASN_INTEGER ||
+			else if (vars->type == ASN_INTEGER
 #ifdef OPAQUE_SPECIAL_TYPES
-					vars->type == ASN_INTEGER64
+					|| vars->type == ASN_INTEGER64
 #endif
 					)
 			{
