diff --git a/src/AboutDialog.cpp b/src/AboutDialog.cpp
index 798e9ee8a..3d32e3787 100644
--- a/src/AboutDialog.cpp
+++ b/src/AboutDialog.cpp
@@ -720,7 +720,7 @@ void AboutDialog::PopulateInformationPage( ShuttleGui & S )
    #ifdef USE_GSTREAMER
    AddBuildinfoRow(&informationStr, wxT("gstreamer"), XO("Import via GStreamer"), enabled);
    #else
-   AddBuildinfoRow(&informationStr, wxT("gstreamer"), XO("Import via GStreamer"), disabled);
+   AddBuildinfoRow(&informationStr, wxT("gstreamer"), XO("Import via GStreamer - removed in version 2.0.6"), disabled);
    #endif
 
    informationStr << wxT("</table>\n");  //end table of file formats supported
