--- mpi.cfg.orig	2022-11-02 05:30:07.000000000 -0500
+++ mpi.cfg	2023-03-08 19:10:08.000000000 -0600
@@ -41,9 +41,9 @@
 # MPICH example
 # -------------
 [mpich]
-mpi_dir              = /home/devel/mpi/mpich/4.0.0
-mpicc                = %(mpi_dir)s/bin/mpicc
-mpicxx               = %(mpi_dir)s/bin/mpicxx
+mpi_dir              = %PREFIX%
+mpicc                = %(mpi_dir)/bin/mpicc-mpich-mp
+mpicxx               = %(mpi_dir)/bin/mpicxx-mpich-mp
 #include_dirs         = %(mpi_dir)s/include
 #libraries            = mpi
 #library_dirs         = %(mpi_dir)s/lib
