--- config/install.sh.orig	2017-06-07 16:06:34.000000000 -0700
+++ config/install.sh	2018-06-25 21:07:28.000000000 -0700
@@ -353,7 +353,18 @@
     vsay $this: Run-time system already exists.
 else
     "$CONFIGDIR"/unpack "$ROOT" runtime
+    echo $this: Adjusting the build scripts to MacPorts requirements
+    cd "$BASEDIR"/runtime/config
+    mv gen-posix-names.sh gen-posix-names.sh.orig
+    awk '/SDK=none/ {print "\t18.*) SDK=MacOSX10.14.sdk ;;\n\t19.*) SDK=MacOSX10.15.sdk ;;"} {print}' gen-posix-names.sh.orig > gen-posix-names.sh
+    chmod +x gen-posix-names.sh
     cd "$BASEDIR"/runtime/objs
+    /usr/bin/sed -E -i '.orig' \
+        -e 's|^CC[ 	]*=[ 	]*cc|CC = @CC@|g' \
+        -e 's|^CPP[ 	]*=[ 	]*cc|CPP = @CPP@|g' \
+        -e 's|^CFLAGS[ 	]*=(.*)$|CFLAGS = \1 @CFLAGS@|g' \
+        -e 's|^LD_LIBS[ 	]*=(.*)$|LD_LIBS = \1 @LDFLAGS@|g' \
+        mk.$ARCH-$OPSYS
     echo $this: Compiling the run-time system.
     $MAKE -f mk.$ARCH-$OPSYS $EXTRA_DEFS
     if [ -x run.$ARCH-$OPSYS ]; then
