--- doc/man/Makefile.am.orig	2017-03-14 01:37:15.000000000 -0700
+++ doc/man/Makefile.am	2017-03-14 01:37:43.000000000 -0700
@@ -7,16 +7,16 @@
 EXTRA_DIST=pam_pwquality.8.pod pwmake.1.pod pwscore.1.pod pwquality.conf.5.pod pwquality.3.pod
 
 %.8: %.8.pod
-	bash -c 'declare -u ucname=$*; pod2man --utf8 --name="$$ucname" --section=8 --center="Linux-PAM Manual" --release="Red Hat, Inc." $< $@'
+	bash -c 'declare ucname=$*; pod2man --utf8 --name="$$ucname" --section=8 --center="Linux-PAM Manual" --release="Red Hat, Inc." $< $@'
 
 %.1: %.1.pod
-	bash -c 'declare -u ucname=$*; pod2man --utf8 --name="$$ucname" --section=1 --center="General Commands Manual" --release="Red Hat, Inc." $< $@'
+	bash -c 'declare ucname=$*; pod2man --utf8 --name="$$ucname" --section=1 --center="General Commands Manual" --release="Red Hat, Inc." $< $@'
 
 %.5: %.5.pod
-	bash -c 'declare -u ucname=$*; pod2man --utf8 --name="$$ucname" --section=5 --center="File Formats Manual" --release="Red Hat, Inc." $< $@'
+	bash -c 'declare ucname=$*; pod2man --utf8 --name="$$ucname" --section=5 --center="File Formats Manual" --release="Red Hat, Inc." $< $@'
 
 %.3: %.3.pod
-	bash -c 'declare -u ucname=$*; pod2man --utf8 --name="$$ucname" --section=5 --center="Libpwquality API Manual" --release="Red Hat, Inc." $< $@'
+	bash -c 'declare ucname=$*; pod2man --utf8 --name="$$ucname" --section=5 --center="Libpwquality API Manual" --release="Red Hat, Inc." $< $@'
 
 manclean:
 	rm *.8 *.1 *.5 *.3
