--- waftools/checks/custom.py.old
+++ waftools/checks/custom.py
@@ -103,7 +103,6 @@
         fragment         = load_fragment('cocoa.m'),
         compile_filename = 'test.m',
         framework_name   = ['Cocoa', 'IOKit', 'OpenGL'],
-        includes         = ctx.srcnode.abspath(),
-        linkflags        = '-fobjc-arc')
+        includes         = ctx.srcnode.abspath())
 
     return fn(ctx, dependency_identifier)
