--- video/out/gl_cocoa.c.old
+++ video/out/gl_cocoa.c
@@ -49,6 +49,8 @@
     struct cgl_context *p = ctx->priv;
     CGLError err;
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+
     CGLOpenGLProfile gl_vers_map[] = {
         [2] = kCGLOGLPVersion_Legacy,
         #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9
@@ -59,15 +61,24 @@
         #endif
     };
 
+#endif
+
     int gl_major = MPGL_VER_GET_MAJOR(ctx->requested_gl_version);
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
     if (gl_major < 2 || gl_major >= MP_ARRAY_SIZE(gl_vers_map)) {
+#else
+    if (gl_major != 2) {
+#endif
         MP_FATAL(ctx->vo, "OpenGL major version %d not supported", gl_major);
         return false;
     }
 
     CGLPixelFormatAttribute attrs[] = {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
         kCGLPFAOpenGLProfile,
         (CGLPixelFormatAttribute) gl_vers_map[gl_major],
+#endif
         kCGLPFADoubleBuffer,
         kCGLPFAAccelerated,
         #if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_8
