--- video/out/cocoa_common.m.old
+++ video/out/cocoa_common.m
@@ -48,6 +48,14 @@
 
 #define CF_RELEASE(a) if ((a) != NULL) CFRelease(a)
 
+/* not used on OS X < 10.8 */
+#if !defined(MAC_OS_X_VERSION_10_8) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_8
+#define kIOPMAssertionTypePreventUserIdleDisplaySleep CFSTR("PreventUserIdleDisplaySleep")
+enum {
+  NSWindowCollectionBehaviorFullScreenPrimary = 0,
+};
+#endif
+
 static void vo_cocoa_fullscreen(struct vo *vo);
 static void vo_cocoa_ontop(struct vo *vo);
 static void cocoa_change_profile(struct vo *vo, char **store, NSScreen *screen);
