--- video/out/cocoa/events_view.h.old	2015-03-04 04:43:55.000000000 +0100
+++ video/out/cocoa/events_view.h	2015-03-04 04:45:38.000000000 +0100
@@ -18,7 +18,7 @@
 #import <Cocoa/Cocoa.h>
 #import "video/out/cocoa/mpvadapter.h"
 
-@interface MpvEventsView : NSView <NSDraggingDestination>
+@interface MpvEventsView : NSView
 @property(nonatomic, retain) MpvCocoaAdapter *adapter;
 - (void)setFullScreen:(BOOL)willBeFullscreen;
 - (void)clear;
