--- video/out/opengl/utils.c.orig
+++ video/out/opengl/utils.c
@@ -1168,6 +1168,7 @@ void gl_pbo_upload_tex(struct gl_pbo_upl
     pbo->index = (pbo->index + 1) % 2;
 
     gl->BindBuffer(GL_PIXEL_UNPACK_BUFFER, pbo->buffers[pbo->index]);
+#if !defined(__APPLE__) || MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
     void *data = gl->MapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, needed_size,
                                     GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT);
     if (!data)
@@ -1185,6 +1186,7 @@ void gl_pbo_upload_tex(struct gl_pbo_upl
     gl->BindBuffer(GL_PIXEL_UNPACK_BUFFER, 0);
 
     return;
+#endif
 
 no_pbo:
     gl_upload_tex(gl, target, format, type, dataptr, stride, x, y, w, h);
