--- video/out/cocoa_common.m.orig
+++ video/out/cocoa_common.m
@@ -135,9 +135,11 @@ static void queue_new_video_size(struct 
     if ([s->window conformsToProtocol: @protocol(MpvSizing)]) {
         id<MpvSizing> win = (id<MpvSizing>) s->window;
         NSRect r = NSMakeRect(0, 0, w, h);
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         if(!opts->hidpi_window_scale) {
             r = [s->current_screen convertRectFromBacking:r];
         }
+#endif
         [win queueNewVideoSize:NSMakeSize(r.size.width, r.size.height)];
     }
 }
@@ -522,8 +524,10 @@ static void create_ui(struct vo *vo, str
     } else {
         NSRect wr =
             NSMakeRect(win->x0, win->y0, win->x1 - win->x0, win->y1 - win->y0);
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         if(!opts->hidpi_window_scale)
             wr = [s->current_screen convertRectFromBacking:wr];
+#endif
         s->window = create_window(wr, s->current_screen, opts->border, adapter);
         parent = [s->window contentView];
     }
