--- video/out/cocoa/events_view.m.old	2015-03-04 04:50:51.000000000 +0100
+++ video/out/cocoa/events_view.m	2015-03-04 04:52:11.000000000 +0100
@@ -202,7 +202,9 @@
 - (NSPoint)convertPointToPixels:(NSPoint)point
 {
     point = [self convertPoint:point fromView:nil];
-    point = [self convertPointToBacking:point];
+    CGFloat scale_factor = [self.window userSpaceScaleFactor];
+    //point = [self convertPointToBacking:point];
+    point = NSMakePoint (point.x * scale_factor, point.y * scale_factor);
     // flip y since isFlipped returning YES doesn't affect the backing
     // coordinate system
     point.y = -point.y;
