--- video/out/opengl/common.c.orig
+++ video/out/opengl/common.c
@@ -227,6 +227,7 @@ static const struct gl_functions gl_func
         .extension = "GL_ARB_texture_rg",
         .provides = MPGL_CAP_TEX_RG,
     },
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
     {
         .ver_core = 320,
         .extension = "GL_ARB_sync",
@@ -237,6 +238,7 @@ static const struct gl_functions gl_func
             {0}
         },
     },
+#endif
     // Swap control, always an OS specific extension
     // The OSX code loads this manually.
     {
