diff --git a/src/AboutDialog.cpp b/src/AboutDialog.cpp
index fe789d6..21d32ad 100644
--- a/src/AboutDialog.cpp
+++ b/src/AboutDialog.cpp
@@ -529,7 +529,7 @@ void AboutDialog::PopulateInformationPage( ShuttleGui & S )
    #ifdef USE_GSTREAMER
    AddBuildinfoRow(&informationStr, wxT("gstreamer"), _("Import via GStreamer"), enabled);
    #else
-   AddBuildinfoRow(&informationStr, wxT("gstreamer"), _("Import via GStreamer"), disabled);
+   AddBuildinfoRow(&informationStr, wxT("gstreamer"), _("Import via GStreamer - removed in version 2.0.6"), disabled);
    #endif
 
    informationStr += wxT("</table>\n");  //end table of file format libraries
