diff --git a/lib-src/libvamp/Makefile.in b/lib-src/libvamp/Makefile.in
index 4fc9378..5e6116d 100644
--- a/lib-src/libvamp/Makefile.in
+++ b/lib-src/libvamp/Makefile.in
@@ -100,21 +100,21 @@ INSTALL_PKGCONFIG	  = $(INSTALL_PREFIX)/lib/pkgconfig
 
 # Flags required to tell the compiler to create a dynamically loadable object
 #
-DYNAMIC_LDFLAGS		= -static-libgcc -shared -Wl,-Bsymbolic
-SDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_SDK_LINK_ABI)
-HOSTSDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_HOSTSDK_LINK_ABI)
+#DYNAMIC_LDFLAGS		= -static-libgcc -shared -Wl,-Bsymbolic
+#SDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_SDK_LINK_ABI)
+#HOSTSDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS) -Wl,-soname=$(INSTALL_HOSTSDK_LINK_ABI)
 
 # Additional flags for making a plugin.  This version script tells the
 # GNU linker to make all symbols in the library hidden except for the
 # public entry point.  It's not essential, but makes a tidier library.
-PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS) -Wl,--version-script=${srcdir}/build/vamp-plugin.map
+#PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS) -Wl,--version-script=${srcdir}/build/vamp-plugin.map
 
 
 ## For OS/X with g++:
-#DYNAMIC_LDFLAGS		= -dynamiclib
-#PLUGIN_LDFLAGS			= $(DYNAMIC_LDFLAGS)
-#SDK_DYNAMIC_LDFLAGS		= $(DYNAMIC_LDFLAGS)
-#HOSTSDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS)
+DYNAMIC_LDFLAGS		= -dynamiclib
+PLUGIN_LDFLAGS			= $(DYNAMIC_LDFLAGS)
+SDK_DYNAMIC_LDFLAGS		= $(DYNAMIC_LDFLAGS)
+HOSTSDK_DYNAMIC_LDFLAGS	= $(DYNAMIC_LDFLAGS)
 
 
 ### End of user-serviceable parts
