--- src/app/CMakeLists.txt.orig	2026-09-03 12:35:12
+++ src/app/CMakeLists.txt	2026-09-21 14:34:07
@@ -401,21 +401,6 @@
         )
     endif()
 
-    qt_generate_deploy_qml_app_script(
-        TARGET audacity
-        OUTPUT_SCRIPT deploy_script
-        NO_TRANSLATIONS
-        EXCLUDE_PLUGIN_TYPES qmltooling sqldrivers styles
-        DEPLOY_TOOL_OPTIONS
-            -qmldir=${PROJECT_SOURCE_DIR}/src
-            -qmldir=${PROJECT_SOURCE_DIR}/muse/framework
-    )
-    install(SCRIPT ${deploy_script})
-
-    install(SCRIPT
-        "${PROJECT_SOURCE_DIR}/buildscripts/packaging/prune_qt_runtime.cmake"
-    )
-
 ###########################################
 # Wasm
 ###########################################
