diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ecc0a47..318195dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,13 +109,13 @@ if( APPLE )
 
    # Define the OSX compatibility parameters
    set( CMAKE_OSX_DEPLOYMENT_TARGET ${MIN_MACOS_VERSION} CACHE INTERNAL "" )
-   set( CMAKE_OSX_SYSROOT macosx CACHE INTERNAL "" )
+#    set( CMAKE_OSX_SYSROOT macosx CACHE INTERNAL "" )
 
    # This prevents a link error when building with the 10.9 or older SDKs
    set( CMAKE_XCODE_ATTRIBUTE_CLANG_LINK_OBJC_RUNTIME OFF )
 
    # Shouldn't cmake do this???
-   string( APPEND CMAKE_CXX_FLAGS " -stdlib=libc++" )
+#    string( APPEND CMAKE_CXX_FLAGS " -stdlib=libc++" )
 endif()
 
 # Add our module path
@@ -140,12 +140,14 @@ set( CMAKE_PREFIX_PATH
 set( CMAKE_CXX_STANDARD 17 )
 set( CMAKE_CXX_STANDARD_REQUIRED ON )
 
-# Use ccache if available
-find_program( CCACHE_PROGRAM ccache )
-mark_as_advanced( FORCE CCACHE_PROGRAM )
+# Use ccache if requested
+if( ${CMAKE_CXX_COMPILER_LAUNCHER} MATCHES "ccache" )
+    set( CCACHE_PROGRAM "${CMAKE_CXX_COMPILER_LAUNCHER}" )
+    mark_as_advanced( FORCE CCACHE_PROGRAM )
 
-if( CCACHE_PROGRAM )
-   set_property( GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}" )
+#     if( CCACHE_PROGRAM )
+#        set_property( GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}" )
+#     endif()
 endif()
 
 # Define option() prefix
@@ -406,6 +408,7 @@ if( CMAKE_CONFIGURATION_TYPES )
 else()
    set( _DESTDIR "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${CMAKE_BUILD_TYPE}" )
 endif()
+set( _DESTDIR "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}" )
 
 set( _DEST "${_DESTDIR}" )
 set( INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" )
@@ -419,9 +422,10 @@ set( _EXEDIR "${CMAKE_INSTALL_BINDIR}" )
 
 # Setup RPATH handling
 set( CMAKE_BUILD_RPATH "$ORIGIN/../${_PKGLIB}" )
-set( CMAKE_BUILD_WITH_INSTALL_RPATH FALSE )
-set( CMAKE_INSTALL_RPATH "$ORIGIN/../${_PKGLIB}" )
-set( CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE )
+set( CMAKE_BUILD_WITH_INSTALL_RPATH TRUE )
+set( CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH};$ORIGIN/../${_PKGLIB}" )
+message(STATUS "CMAKE_INSTALL_RPATH=${CMAKE_INSTALL_RPATH}")
+# set( CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE )
 
 # Adjust them for the Mac
 if( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
@@ -455,9 +459,11 @@ if( HAVE_LIBM )
    list( APPEND CMAKE_REQUIRED_LIBRARIES -lm )
 endif()
 
-check_library_exists( atomic __atomic_fetch_add_4 "" HAVE_LIBATOMIC )
-if( HAVE_LIBATOMIC )
-  list( APPEND CMAKE_REQUIRED_LIBRARIES -latomic )
+if( NOT APPLE)
+   check_library_exists( atomic __atomic_fetch_add_4 "" HAVE_LIBATOMIC )
+   if( HAVE_LIBATOMIC )
+     list( APPEND CMAKE_REQUIRED_LIBRARIES -latomic )
+   endif()
 endif()
 
 # Add the dynamic linker library (if needed) to the list of required libraries
diff --git a/cmake-proxies/lv2/CMakeLists.txt b/cmake-proxies/lv2/CMakeLists.txt
index d4a17715..2e5a837c 100644
--- a/cmake-proxies/lv2/CMakeLists.txt
+++ b/cmake-proxies/lv2/CMakeLists.txt
@@ -192,6 +192,42 @@ macro( bld name packages define sources )
    endif()
 endmacro()
 
+macro( bld2 name packages define sources )
+   set( libs )
+   set( missing )
+   set( lib "${SUIL_MODULE_PREFIX}${name}" )
+   foreach( pkg ${packages} )
+      list( APPEND libs
+         PRIVATE
+            "${pkg}"
+      )
+   endforeach()
+   if( ${name} STREQUAL "suil_qt5_in_cocoa")
+       list( APPEND libs PRIVATE "-framework Cocoa" )
+   endif()
+
+   if( NOT missing )
+      list( APPEND DEFINES
+         PRIVATE
+            ${define}
+      )
+
+      add_library( ${lib} MODULE "${sources}" )
+      add_dependencies( ${TARGET} ${lib} )
+
+      set_target_properties( ${lib}
+         PROPERTIES
+            LIBRARY_OUTPUT_DIRECTORY "${_DEST}/${_PKGLIB}"
+            PREFIX ""
+      )
+
+      organize_source( "${TARGET_ROOT}" "" "${sources}" )
+      target_compile_definitions( ${lib} PRIVATE SUIL_SHARED ${DEFINES} )
+      target_include_directories( ${lib} PRIVATE ${INCLUDES} )
+      target_link_libraries( ${lib} PRIVATE ${libs} )
+   endif()
+endmacro()
+
 if( WIN32 )
    set( LILV_PATH_SEP ";" )
    set( LILV_DIR_SEP "\\\\" )
@@ -206,7 +242,7 @@ if( WIN32 )
 elseif( APPLE )
    set( LILV_PATH_SEP ":" )
    set( LILV_DIR_SEP "/" )
-   set( LILV_DEFAULT_LV2_PATH "~/Library/Audio/Plug-Ins/LV2:~/.lv2:/usr/local/lib/lv2:/usr/lib/lv2:/Library/Audio/Plug-Ins/LV2" )
+   set( LILV_DEFAULT_LV2_PATH "~/Library/Audio/Plug-Ins/LV2:~/.lv2:@PREFIX@/lib/lv2:/Library/Audio/Plug-Ins/LV2" )
 
    set( SUIL_MODULE_DIR "" )
    set( SUIL_DIR_SEP "" )
@@ -214,10 +250,18 @@ elseif( APPLE )
    set( SUIL_GTK3_LIB_NAME "libgtk-x11-3.0.so.0" )
    set( SUIL_MODULE_PREFIX "lib" )
    set( SUIL_MODULE_EXT ".dylib" )
+   option(audacity_use_suil_qt5 "Whether to build the suil_qt5_in_cocoa GUI library for LV2 plugins" OFF)
+   if( audacity_use_suil_qt5 )
+       find_package(Qt5 5.1.0 REQUIRED COMPONENTS Core Widgets Gui)
+       bld2( "suil_qt5_in_cocoa"
+            "Qt5::Core;Qt5::Widgets;Qt5::Gui"
+            "SUIL_WITH_COCOA_IN_QT5"
+            "${TARGET_ROOT}/suil/src/cocoa_in_qt5.mm" )
+   endif()
 elseif( UNIX )
    set( LILV_PATH_SEP ":" )
    set( LILV_DIR_SEP "/" )
-   set( LILV_DEFAULT_LV2_PATH "~/.lv2:/usr/lib/lv2:/usr/local/lib/lv2" )
+   set( LILV_DEFAULT_LV2_PATH "~/.lv2:@PREFIX@/lib/lv2:/usr/lib/lv2:/usr/local/lib/lv2" )
 
    set( SUIL_MODULE_DIR "" )
    set( SUIL_DIR_SEP "" )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0f1a9627..cdbf6bc4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1083,7 +1083,7 @@ list( APPEND DEFINES
 # If we have cmake 3.16 or higher, we can use precompiled headers, but
 # only use them if ccache is not available and the user hasn't disabled
 # it.
-if( CMAKE_VERSION VERSION_GREATER_EQUAL "3.16" AND NOT CCACHE_PROGRAM )
+if( CMAKE_VERSION VERSION_GREATER_EQUAL "3.16"  )
    cmd_option(
       ${_OPT}use_pch
       "Use precompiled headers [yes, no]"
@@ -1269,6 +1269,9 @@ elseif( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
       PRIVATE
          "-framework CoreAudio"
          "-framework CoreAudioKit"
+         "-framework AudioToolbox"
+         "-framework Cocoa"
+         "-framework Carbon"
    )
 
    # Define the Wrapper target
@@ -1497,13 +1500,20 @@ propagate_interesting_dependencies( ${TARGET} "${AUDACITY_LIBRARIES}" )
 
 target_sources( ${TARGET} PRIVATE ${HEADERS} ${SOURCES} ${RESOURCES} ${MAC_RESOURCES} ${WIN_RESOURCES} )
 target_compile_definitions( ${TARGET} PRIVATE ${DEFINES} )
+
+set_property(SOURCE effects/VST/VSTEffect.cpp APPEND PROPERTY COMPILE_DEFINITIONS LIBDIR="${CMAKE_INSTALL_FULL_LIBDIR}")
+set_property(SOURCE effects/ladspa/LadspaEffect.cpp APPEND PROPERTY COMPILE_DEFINITIONS LIBDIR="${CMAKE_INSTALL_FULL_LIBDIR}")
+set_property(SOURCE effects/lv2/LoadLV2.cpp APPEND PROPERTY COMPILE_DEFINITIONS LIBDIR="${CMAKE_INSTALL_FULL_LIBDIR}")
+set_property(SOURCE effects/lv2/LoadLV2.cpp APPEND PROPERTY COMPILE_DEFINITIONS PKGLIBDIR="${PKGLIBDIR}")
+set_property(SOURCE export/ExportMP3.cpp APPEND PROPERTY COMPILE_DEFINITIONS LIBDIR="${CMAKE_INSTALL_FULL_LIBDIR}")
+
 target_compile_options( ${TARGET} PRIVATE ${OPTIONS} )
 target_include_directories( ${TARGET} PRIVATE ${INCLUDES} )
 target_link_options( ${TARGET} PRIVATE ${LDFLAGS} )
 target_link_libraries( ${TARGET} PUBLIC ${AUDACITY_LIBRARIES} )
 target_link_libraries( ${TARGET} ${LIBRARIES} )
 
-if( CMAKE_VERSION VERSION_GREATER_EQUAL "3.16" AND NOT CCACHE_PROGRAM )
+if( CMAKE_VERSION VERSION_GREATER_EQUAL "3.16" )
    if( ${_OPT}use_pch )
       message( STATUS "Using precompiled headers" )
       target_precompile_headers( ${TARGET} PRIVATE
