diff --git a/src/toolbars/ToolManager.cpp b/src/toolbars/ToolManager.cpp
index 66ed25b..11a9e24 100644
--- a/src/toolbars/ToolManager.cpp
+++ b/src/toolbars/ToolManager.cpp
@@ -63,6 +63,8 @@
 #include "../AllThemeResources.h"
 #include "../AudioIO.h"
 #include "../ImageManipulation.h"
+// hack to deactivate a bit in wx/confbase.h on which clang chokes:
+#define __VMS
 #include "../Prefs.h"
 #include "../Project.h"
 #include "../Theme.h"
