diff --git a/CMakeLists.txt b/CMakeLists.txt
index 318195dd..5de81959 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -666,16 +666,16 @@ else()
 endif()
 
 # Generate a picture of module dependencies
-string( JOIN "\n" GRAPH_EDGES ${GRAPH_EDGES} )
-# Choose edge attributes making it easy to hover at either end of edge
-# and see a tooltip describing the edge, in svg image
-file( WRITE "${CMAKE_CURRENT_BINARY_DIR}/modules.dot" "digraph {
-   graph [rankdir=LR] edge [dir=both,arrowtail=inv] \n"
-   "${GRAPH_EDGES}"
-   "\n}\n"
-)
-execute_process( COMMAND
-   dot -O -Tsvg "${CMAKE_CURRENT_BINARY_DIR}/modules.dot" )
+# string( JOIN "\n" GRAPH_EDGES ${GRAPH_EDGES} )
+# # Choose edge attributes making it easy to hover at either end of edge
+# # and see a tooltip describing the edge, in svg image
+# file( WRITE "${CMAKE_CURRENT_BINARY_DIR}/modules.dot" "digraph {
+#    graph [rankdir=LR] edge [dir=both,arrowtail=inv] \n"
+#    "${GRAPH_EDGES}"
+#    "\n}\n"
+# )
+# execute_process( COMMAND
+#    dot -O -Tsvg "${CMAKE_CURRENT_BINARY_DIR}/modules.dot" )
 
 #
 # Code signing
diff --git a/cmake-proxies/cmake-modules/AudacityDependencies.cmake b/cmake-proxies/cmake-modules/AudacityDependencies.cmake
index 1feedb08..196a6249 100644
--- a/cmake-proxies/cmake-modules/AudacityDependencies.cmake
+++ b/cmake-proxies/cmake-modules/AudacityDependencies.cmake
@@ -139,6 +139,9 @@ include (DependenciesList)
 
 # If conan is enabled, run conan_runner.py
 if( ${_OPT}conan_enabled )
+   ## RJVB: cache the MacPorts build type because conan no likey
+   set(CBT ${CMAKE_BUILD_TYPE})
+   set(CMAKE_BUILD_TYPE Release)
    # Deduce the build type
    get_property(is_multi_config GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
    if( is_multi_config )
@@ -200,9 +203,13 @@ if( ${_OPT}conan_enabled )
 
    if( conan_result )
       message( FATAL_ERROR "Conan failed to install dependencies (${conan_result}) ${PYTHON}" )
+   else()
+      message( STATUS "conan_runner.py completed successfully")
    endif()
 
    set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
+   ## RJVB: restore the MacPorts build type
+   set(CMAKE_BUILD_TYPE ${CBT})
 endif()
 
 if( ${_OPT}conan_enabled )
