diff --git a/scripts/build/macOS/fix_bundle.py b/scripts/build/macOS/fix_bundle.py
index f15fc582deca5d937d7922074599bbdba2e4c988..0fbf289d93f709fa09d68faf55e24268879f8413 100644
--- a/scripts/build/macOS/fix_bundle.py
+++ b/scripts/build/macOS/fix_bundle.py
@@ -54,6 +55,12 @@ def macos_collect_dependencies(file):
 
 
 def run_install_name_tool(file, dep_path_prefix):
+    ## RJVB/MacPorts hacked version where we don't embed wxWidgets libraries
+    wxmatch = re.match(r'.*libwx_.*\.dylib', file)
+    if wxmatch:
+        print(f'Skipping wxWidgets library {file}')
+        return
+
     dependencies = macos_collect_dependencies(file)
     rpaths = macos_collect_rpaths(file)
 
@@ -61,15 +68,27 @@ def run_install_name_tool(file, dep_path_prefix):
         return
 
     install_name_tool = ['install_name_tool']
+    int_args_needed = False
 
     for name, lib_line in dependencies:
-        install_name_tool.append('-change')
-        install_name_tool.append(lib_line)
-        install_name_tool.append(dep_path_prefix + '/' + name)
+        wxmatch = re.match(r'.*libwx_.*\.dylib', lib_line)
+        if wxmatch:
+            print(f'Not changing loadpath for wxWidgets library {lib_line}')
+        else:
+            install_name_tool.append('-change')
+            install_name_tool.append(lib_line)
+            install_name_tool.append(dep_path_prefix + '/' + name)
+            int_args_needed = True
+
 
     for rpath in rpaths:
         install_name_tool.append('-delete_rpath')
         install_name_tool.append(rpath)
+        int_args_needed = True
+
+    if not int_args_needed:
+        print(f'No need to invoke install_name_tool on {file}')
+        return
 
     install_name_tool.append(file)
 
