commit ae431bc5bd7c7967e72735779ad2796d43f54ad5
Author: James Crook <james.k.crook@gmail.com>
Date:   Fri Apr 5 17:04:54 2019 +0100

    Build LAME on macOS

--- lib-src/mpglib/config.h
+++ lib-src/mpglib/config.h
@@ -1,8 +1,6 @@
 #ifndef CONFIGMS_H_INCLUDED
 #define CONFIGMS_H_INCLUDED
 
-/*#if defined(_MSC_VER)*/
-#if 0
 /* The number of bytes in a double.  */
 #define SIZEOF_DOUBLE 8
 
@@ -29,7 +27,6 @@
 
 /* The number of bytes in a unsigned short.  */
 #define SIZEOF_UNSIGNED_SHORT 2
-#endif
 
 /* Define if you have the ANSI C header files.  */
 #define STDC_HEADERS
@@ -69,13 +66,6 @@
 
 	typedef float  float32_t;
 	typedef double float64_t;
-#elif defined(STDC_HEADERS)
-
-#include <stdlib.h>
-typedef u_int8_t uint8_t;
-typedef u_int16_t uint16_t;
-typedef u_int32_t uint32_t;
-
 #elif defined (__GNUC__)
 #define __int8_t_defined
 #define uint8_t unsigned char
--- src/export/ExportMP3.h
+++ src/export/ExportMP3.h
@@ -20,7 +20,7 @@
 #define MODE_ABR           2
 #define MODE_CBR           3
 
-#if defined(__WXMSW__) || defined(__WXMAC__)
+#if defined(__WXMSW__)
 #define MP3_EXPORT_BUILT_IN 1
 #endif
 
