--- src/appshell/internal/appshellconfiguration.cpp.orig
+++ src/appshell/internal/appshellconfiguration.cpp
@@ -46,7 +46,7 @@
 static const Settings::Key STARTUP_SCORE_PATH(module_name, "application/startup/startScore");
 
 static const std::string AUDACITY_URL("https://www.audacityteam.org/");
-static const std::string AUDACITY_MANUAL_URL("https://manual.audacityteam.org/");
+static const std::string AUDACITY_MANUAL_URL("https://www.audacityteam.org/manual/getting-started/");
 static const std::string AUDACITY_SUPPORT_URL("https://support.audacityteam.org/");
 static const std::string AUDACITY_FORUM_URL("https://forum.audacityteam.org/");
 static const std::string AUDACITY_CONTRIBUTE_URL("https://support.audacityteam.org/community/contributing");
