diff --git a/src/widgets/NumericTextCtrl.cpp b/src/widgets/NumericTextCtrl.cpp
index bef3431d4..8c26df763 100644
--- a/src/widgets/NumericTextCtrl.cpp
+++ b/src/widgets/NumericTextCtrl.cpp
@@ -1369,8 +1369,8 @@ NumericTextCtrl::NumericTextCtrl(wxWindow *parent, wxWindowID id,
 {
    mAllowInvalidValue = false;
 
-   mDigitBoxW = 11;
-   mDigitBoxH = 19;
+   mDigitBoxW = 8;
+   mDigitBoxH = 14;
 
    mBorderLeft = 1;
    mBorderTop = 1;
diff --git a/src/widgets/Ruler.cpp b/src/widgets/Ruler.cpp
index 93c7cc880..2ffefa48d 100644
--- a/src/widgets/Ruler.cpp
+++ b/src/widgets/Ruler.cpp
@@ -1003,14 +1003,14 @@ static constexpr int MinPixelHeight =
 #ifdef __WXMSW__
    12;
 #else
-   10;
+   7;
 #endif
 
 static constexpr int MaxPixelHeight =
 #ifdef __WXMSW__
    14;
 #elif __WXMAC__
-   10;
+   9;
 #else
    12;
 #endif
