diff --git a/src/Makefile.am b/src/Makefile.am
index 2dea345..eb77b29 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -78,8 +78,9 @@ audacity_CPPFLAGS = \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = \
diff --git a/src/Makefile.in b/src/Makefile.in
index 55cf7ec..d6526bb 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -1161,8 +1161,9 @@ audacity_CPPFLAGS = -std=c++11 -Wno-deprecated-declarations \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = $(EXPAT_LIBS) $(FILEDIALOG_LIBS) $(PORTAUDIO_LIBS) \
