diff --git a/src/Makefile.am b/src/Makefile.am
index e61be0e4e..c45cb3080 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -79,8 +79,9 @@ audacity_CPPFLAGS = \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = \
diff --git a/src/Makefile.in b/src/Makefile.in
index d28e7abc1..63b795932 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -1287,8 +1287,9 @@ audacity_CPPFLAGS = -std=c++11 -Wno-deprecated-declarations \
 #  /usr/local/include/wx-2.8/wx/buffer.h:289:9: note: in expansion of macro 'wxCHECK_RET'
 #           wxCHECK_RET( m_bufdata->m_data, wxT("invalid wxMemoryBuffer") );
 #           ^
-audacity_CXXFLAGS = \
-   -Wno-unused-local-typedefs
+#audacity_CXXFLAGS = \
+#   -Wno-unused-local-typedefs
+audacity_CXXFLAGS =
 
 audacity_LDFLAGS = -rdynamic
 audacity_LDADD = $(EXPAT_LIBS) $(FILEDIALOG_LIBS) $(PORTAUDIO_LIBS) \
