--- src/gxhpng.c.orig	2008-03-13 23:57:19.000000000 +0900
+++ src/gxhpng.c	2008-03-13 23:58:39.000000000 +0900
@@ -377,7 +377,10 @@
   return (retcod);
 }
 	
-int gdCompareInt(const void *a, const void *b);
+int gdCompareInt(const void *a, const void *b)
+{
+  return (*(const int *)a) - (*(const int *)b);
+}
 
 /* Version of gdImageFilledPolygon to invoke my local 
    version of gdImageLne.  Nothing else changed... B.Doty 5/31/01 */
