--- configure.orig	2012-04-04 11:18:20.000000000 +0900
+++ configure	2012-04-04 11:20:45.000000000 +0900
@@ -3232,7 +3232,7 @@
 
   *darwin*)
     is_darwin="yes"
-    LDFLAGS="$LDFLAGS -lSystemStubs"
+    LDFLAGS="$LDFLAGS"
   ;;
 
   cray-*-*)
@@ -8821,18 +8821,8 @@
 if test $ac_cv_lib_readline_tgetent = yes; then
    use_readline=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in "" ; do
-      readline_inc="$readline_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in readline ; do
-      readline_libs="$readline_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  readline_inc=""
+  readline_libs=-lreadline
 
 
 else
@@ -8903,18 +8893,8 @@
 if test $ac_cv_lib_ncurses_tgetent = yes; then
    use_readline=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in "" ; do
-      readline_inc="$readline_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in readline ncurses ; do
-      readline_libs="$readline_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  readline_inc=""
+  readline_libs=-lreadline
 
 
 else
@@ -8985,18 +8965,8 @@
 if test $ac_cv_lib_termcap_tgetent = yes; then
    use_readline=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in "" ; do
-      readline_inc="$readline_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in readline ; do
-      readline_libs="$readline_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  readline_inc=""
+  readline_libs=-lreadline
 
               readline_libadd=" -ltermcap"
 
@@ -9978,18 +9948,8 @@
 if test $ac_cv_lib_geotiff_main = yes; then
    use_geotiff=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in tiff geotiff ; do
-      geotiff_inc="$geotiff_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in tiff geotiff ; do
-      geotiff_libs="$geotiff_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  geotiff_inc=""
+  geotiff_libs="-lgeotiff -ltiff"
 
 
 fi
@@ -10911,18 +10871,8 @@
 if test $ac_cv_lib_shp_main = yes; then
    use_shp=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in shapelib shp ; do
-      shp_inc="$shp_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in shp ; do
-      shp_libs="$shp_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  shp_inc=""
+  shp_libs=-lshp
 
 
 fi
@@ -11217,7 +11167,7 @@
   ga_saved_libs=$LIBS
   CPPFLAGS=""
 
-    for ga_inc_name in libpng12 ; do
+    for ga_inc_name in libpng ; do
       CPPFLAGS="$CPPFLAGS -I${ga_supplib_dir}/include/${ga_inc_name}"
     done
 
@@ -11225,13 +11175,13 @@
   LDFLAGS="-L${ga_supplib_dir}/lib "
   LIBS="$LIBS "
 
-    echo "$as_me:$LINENO: checking for main in -lpng12" >&5
-echo $ECHO_N "checking for main in -lpng12... $ECHO_C" >&6
-if test "${ac_cv_lib_png12_main+set}" = set; then
+    echo "$as_me:$LINENO: checking for main in -lpng" >&5
+echo $ECHO_N "checking for main in -lpng... $ECHO_C" >&6
+if test "${ac_cv_lib_png_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng12  $LIBS"
+LIBS="-lpng  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -11270,20 +11220,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_png12_main=yes
+  ac_cv_lib_png_main=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_png12_main=no
+ac_cv_lib_png_main=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_png12_main" >&5
-echo "${ECHO_T}$ac_cv_lib_png12_main" >&6
-if test $ac_cv_lib_png12_main = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_png_main" >&5
+echo "${ECHO_T}$ac_cv_lib_png_main" >&6
+if test $ac_cv_lib_png_main = yes; then
    have_libpng=yes
 
 fi
@@ -11599,29 +11549,9 @@
 if test $ac_cv_lib_gd_gdImageCreate = yes; then
    use_printim=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in gd ; do
-      printim_inc="$printim_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-	  if test "$have_jpeg" = "yes" ; then
+  printim_inc=""
+  printim_libs=-lgd
 
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in gd png12 z jpeg ; do
-      printim_libs="$printim_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
-
-	  else
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in gd png12 z ; do
-      printim_libs="$printim_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
-
-          fi
 
 fi
 
@@ -12848,7 +12778,7 @@
   ga_saved_libs=$LIBS
   CPPFLAGS=""
 
-    for ga_inc_name in libpng12 ; do
+    for ga_inc_name in libpng ; do
       CPPFLAGS="$CPPFLAGS -I${ga_supplib_dir}/include/${ga_inc_name}"
     done
 
@@ -12856,13 +12786,13 @@
   LDFLAGS="-L${ga_supplib_dir}/lib "
   LIBS="$LIBS "
 
-      echo "$as_me:$LINENO: checking for main in -lpng12" >&5
-echo $ECHO_N "checking for main in -lpng12... $ECHO_C" >&6
-if test "${ac_cv_lib_png12_main+set}" = set; then
+      echo "$as_me:$LINENO: checking for main in -lpng" >&5
+echo $ECHO_N "checking for main in -lpng... $ECHO_C" >&6
+if test "${ac_cv_lib_png_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng12  $LIBS"
+LIBS="-lpng  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -12901,20 +12831,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_png12_main=yes
+  ac_cv_lib_png_main=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_png12_main=no
+ac_cv_lib_png_main=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_png12_main" >&5
-echo "${ECHO_T}$ac_cv_lib_png12_main" >&6
-if test $ac_cv_lib_png12_main = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_png_main" >&5
+echo "${ECHO_T}$ac_cv_lib_png_main" >&6
+if test $ac_cv_lib_png_main = yes; then
    have_libpng=yes
 
 fi
@@ -13227,18 +13157,8 @@
 if test $ac_cv_lib_grib2c_main = yes; then
    use_grib2=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in grib2c ; do
-      grib2_inc="$grib2_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in grib2c jasper png12 z ; do
-      grib2_libs="$grib2_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  grib2_inc=""
+  grib2_libs="-lgrib2c -ljasper -lpng -lz"
 
 
 fi
@@ -13459,13 +13379,13 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_grib2c_main" >&5
 echo "${ECHO_T}$ac_cv_lib_grib2c_main" >&6
 if test $ac_cv_lib_grib2c_main = yes; then
-   echo "$as_me:$LINENO: checking for main in -lpng12" >&5
-echo $ECHO_N "checking for main in -lpng12... $ECHO_C" >&6
-if test "${ac_cv_lib_png12_main+set}" = set; then
+   echo "$as_me:$LINENO: checking for main in -lpng" >&5
+echo $ECHO_N "checking for main in -lpng... $ECHO_C" >&6
+if test "${ac_cv_lib_png_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng12  $LIBS"
+LIBS="-lpng  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -13504,20 +13424,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_png12_main=yes
+  ac_cv_lib_png_main=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_png12_main=no
+ac_cv_lib_png_main=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_png12_main" >&5
-echo "${ECHO_T}$ac_cv_lib_png12_main" >&6
-if test $ac_cv_lib_png12_main = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_png_main" >&5
+echo "${ECHO_T}$ac_cv_lib_png_main" >&6
+if test $ac_cv_lib_png_main = yes; then
    echo "$as_me:$LINENO: checking for main in -lz" >&5
 echo $ECHO_N "checking for main in -lz... $ECHO_C" >&6
 if test "${ac_cv_lib_z_main+set}" = set; then
@@ -14617,18 +14537,8 @@
 if test $ac_cv_lib_mfhdf_main = yes; then
    use_hdf=yes
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in hdf udunits ; do
-      hdf_inc="$hdf_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in mfhdf df udunits sz jpeg z ; do
-      hdf_libs="$hdf_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  hdf_inc=""
+  hdf_libs="-lmfhdf -ldf -ludunits -lsz -ljpeg -lz"
 
 
 fi
@@ -17168,16 +17078,8 @@
 
   ga_include_prefix='-I$(supp_include_dir)'
 
-  for ga_include_name in hdf5 ; do
-      hdf5_inc="$hdf5_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in hdf5 sz z ; do
-      hdf5_libs="$hdf5_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  hdf5_inc=""
+  hdf5_libs=-lhdf5
 
 
 fi
@@ -18653,18 +18555,8 @@
 if test $ac_cv_lib_curl_main = yes; then
    use_nc4="yes"
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in udunits netcdf ; do
-      nc_inc="$nc_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in udunits netcdf hdf5_hl hdf5 z sz curl ; do
-      nc_libs="$nc_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  nc_inc=""
+  nc_libs=-lnetcdf
 
 
 fi
@@ -21604,18 +21496,8 @@
 if test $ac_cv_lib_curl_main = yes; then
    use_gadap="yes"
 
-  ga_include_prefix='-I$(supp_include_dir)'
-
-  for ga_include_name in gadap ; do
-      gadap_inc="$gadap_inc ${ga_include_prefix}/${ga_include_name}"
-  done
-
-
-  ga_lib_prefix='$(supp_lib_dir)/lib'
-  ga_lib_suffix='.a'
-  for ga_lib_name in gadap dapclient dap curl xml2 z ; do
-      gadap_libs="$gadap_libs ${ga_lib_prefix}${ga_lib_name}${ga_lib_suffix}"
-  done
+  gadap_inc=""
+  gadap_libs="-lgadap -ldap -ldapserver -ldapclient -lcurl -lidn -lssl -lcrypto -lz -lxml2 -lpthread -lz -liconv -lm"
 
 
 fi
