--- soundserver/Makefile.am	14 Aug 2002 13:21:26 -0000	1.1.1.4
+++ soundserver/Makefile.am	25 Mar 2003 18:16:37 -0000	1.12
@@ -38,13 +38,13 @@
 libartswavplayobject_la_SOURCES = wavplayobject_impl.cc
 libartswavplayobject_la_LIBADD  = $(top_builddir)/mcop/libmcop.la \
                                   libsoundserver_idl.la $(FLOWLIBS)
-libartswavplayobject_la_LDFLAGS = -no-undefined -module
+libartswavplayobject_la_LDFLAGS = -no-undefined -module -avoid-version
 libartswavplayobject_la_COMPILE_FIRST = soundserver.h ../flow/artsflow.h
 
 libartsgslplayobject_la_SOURCES = gslplayobject_impl.cc
 libartsgslplayobject_la_LIBADD  = $(top_builddir)/mcop/libmcop.la \
                                   libsoundserver_idl.la $(FLOWLIBS)
-libartsgslplayobject_la_LDFLAGS = -no-undefined -module
+libartsgslplayobject_la_LDFLAGS = -no-undefined -module -avoid-version
 libartsgslplayobject_la_COMPILE_FIRST = soundserver.h ../flow/artsflow.h
 
 ###### "real" programs
