--- src/rigcontrol/FreqControl.cxx
+++ src/rigcontrol/FreqControl.cxx
@@ -131,7 +131,7 @@ cFreqControl::cFreqControl(int x, int y, int w, int h, const char *lbl):
 	color(OFFCOLOR);
 
 	minVal = 0;
-	double fmaxval = (pow(10, nD) - 1) * precision;
+	double fmaxval = (pow(10.0, nD) - 1) * precision;
 	long int UMAX = maximum();
 	if (fmaxval > UMAX) fmaxval = UMAX;
 	maxVal = fmaxval;
@@ -341,7 +341,7 @@ void cFreqControl::value(long lv)
 
 long int cFreqControl::maximum(void)
 {
-	return (long int)(pow(2, 31) - 1);
+	return (long int)(pow(2.0, 31) - 1);
 }
 
 
@@ -618,8 +618,8 @@ void cFreqControl::set_ndigits(int nbr)
 	int xpos;
 
 	minVal = 0;
-	double fmaxval = (pow(10, nD) - 1) * precision;
-	long int UMAX = (long int)(pow(2, 31) - 1);
+	double fmaxval = (pow(10.0, nD) - 1) * precision;
+	long int UMAX = (long int)(pow(2.0, 31) - 1);
 	if (fmaxval > UMAX) fmaxval = UMAX;
 	maxVal = fmaxval;
 	fmaxval /= 1000.0;