--- scripts/mkappbundle.sh.old	2019-04-13 12:59:43.000000000 +0200
+++ scripts/mkappbundle.sh	2019-04-13 13:04:38.000000000 +0200
@@ -59,7 +59,6 @@
 
 	$mkinstalldirs "$appname/Contents/Frameworks"
 	cd "$appname/Contents"
-	copy_libs "MacOS/$binary"
 }
 
 if [ $# -ne 2 ]; then
@@ -125,5 +124,3 @@
 
 cd "$build"
 
-echo "creating disk image"
-hdiutil create -ov -srcfolder "$bundle_dir" -format UDZO -tgtimagekey zlib-level=9 "${APPBUNDLE}.dmg"
