--- ./AppController.m.old	2007-11-07 05:15:32.000000000 +0900
+++ ./AppController.m	2007-11-07 08:24:25.000000000 +0900
@@ -76,11 +76,7 @@
   // setup for task
   NSString *askPassPath = [[NSBundle mainBundle] pathForResource:@"askpass"
                                                           ofType:@""];
-  NSString *sshfsPath = [[NSBundle mainBundle] pathForResource:@"sshfs-static"
-                                                        ofType:@""];
-  if ([GoogleShared_SystemVersion isLeopard]) {
-      sshfsPath = [sshfsPath stringByAppendingString:@"-10.5"];
-  }
+  NSString *sshfsPath = @"##PREFIX##/bin/sshfs";
   NSDictionary *sshfsEnv = [NSDictionary dictionaryWithObjectsAndKeys:
     @"NONE", @"DISPLAY", // I know "NONE" is bogus; I just need something
     askPassPath, @"SSH_ASKPASS",
