--- doc/Makefile.in.orig	2014-01-08 08:15:57.000000000 -0600
+++ doc/Makefile.in	2014-06-24 17:37:45.000000000 -0500
@@ -513,7 +513,7 @@
 	  fi; \
 	else \
 	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
-	  $(INSTALL_DATA) $(srcdir)/siegerc $(DESTDIR)$(SIEGERC); \
+	  echo ""; \
 	fi 
 	@if test -f $(DESTDIR)$(URLSTXT); then \
 	  if cmp -s $(srcdir)/siegerc $(DESTDIR)$(URLSTXT); then echo ""; \
