diff -ur libgupnp-dlna/metadata-backends.orig/gstreamer/gstreamer.am libgupnp-dlna/metadata-backends/gstreamer/gstreamer.am
--- libgupnp-dlna/metadata-backends.orig/gstreamer/gstreamer.am	2015-07-03 08:53:52.000000000 -0700
+++ libgupnp-dlna/metadata-backends/gstreamer/gstreamer.am	2015-07-03 08:55:35.000000000 -0700
@@ -21,7 +21,7 @@
 
 metadata_backends_gstreamer_libgstreamer_la_LDFLAGS = \
 	-no-undefined \
-	-shared -fPIC -module -avoid-version -Wl,-z,defs
+	-shared -fPIC -module -avoid-version
 metadata_backends_gstreamer_libgstreamer_la_SOURCES = \
 	metadata-backends/gstreamer/gupnp-dlna-gst-metadata-backend.c \
 	metadata-backends/gstreamer/gupnp-dlna-gst-metadata-extractor.c
diff -ur libgupnp-dlna/metadata-backends.orig/gstreamer-legacy/gstreamer.am libgupnp-dlna/metadata-backends/gstreamer-legacy/gstreamer.am
--- libgupnp-dlna/metadata-backends.orig/gstreamer-legacy/gstreamer.am	2015-07-03 08:53:52.000000000 -0700
+++ libgupnp-dlna/metadata-backends/gstreamer-legacy/gstreamer.am	2015-07-03 08:56:01.000000000 -0700
@@ -21,7 +21,7 @@
 
 metadata_backends_gstreamer_legacy_libgstreamer_legacy_la_LDFLAGS = \
 	-no-undefined \
-	-shared -fPIC -module -avoid-version -Wl,-z,defs
+	-shared -fPIC -module -avoid-version
 metadata_backends_gstreamer_legacy_libgstreamer_legacy_la_SOURCES = \
 	metadata-backends/gstreamer-legacy/gupnp-dlna-gst-metadata-backend.c \
 	metadata-backends/gstreamer-legacy/gupnp-dlna-gst-metadata-extractor.c
