--- sdl2/dll.py.orig	2022-01-12 07:42:03.000000000 +1100
+++ sdl2/dll.py	2022-01-15 23:33:50.000000000 +1100
@@ -141,12 +141,6 @@
     else:
         patterns = ["lib{0}.so"]
 
-    # On Apple Silicon Macs, search the non-standard Homebrew library path if no other
-    # path explicitly set
-    arm_brewpath = "/opt/Homebrew/lib"
-    if not path and platform == "darwin" and os.path.exists(arm_brewpath):
-        path = arm_brewpath
-
     # Adding the potential 'd' suffix that is present on the library
     # when built in debug configuration
     searchfor = libnames + [libname + 'd' for libname in libnames]
@@ -156,6 +150,10 @@
     if path and path.lower() != "system":
         results = _finds_libs_at_path(libnames, path, patterns)
 
+    # MacPorts addition: search in our prefix first (after any explicit path)
+    mp_path = os.path.join("__PREFIX__", "lib")
+    results += _finds_libs_at_path(libnames, mp_path, patterns)
+
     # Next, search for library in system library search paths
     for libname in searchfor:
         dllfile = find_library(libname)
@@ -165,12 +163,6 @@
                 dllfile = "./" + dllfile
             results.append(dllfile)
 
-    # On ARM64 Macs, search the non-standard brew library path as a fallback
-    arm_brewpath = "/opt/Homebrew/lib"
-    is_apple_silicon = platform == "darwin" and cpu_arch() == "arm64"
-    if is_apple_silicon and os.path.exists(arm_brewpath):
-        results += _finds_libs_at_path(libnames, arm_brewpath, patterns)
-
     return results
 
 
