--- sdl2/dll.py.orig	2014-01-25 04:14:32.000000000 +1100
+++ sdl2/dll.py	2014-04-08 12:37:01.000000000 +1000
@@ -32,6 +32,11 @@
             dllfile = os.path.join(path, pattern % libname)
             if os.path.exists(dllfile):
                 results.append(dllfile)
+    # MacPorts addition: search in our prefix first
+    for libname in searchfor:
+        dllfile = os.path.join('__PREFIX__', 'lib', pattern % libname)
+        if os.path.exists(dllfile):
+            results.append(dllfile)
     for libname in searchfor:
         dllfile = find_library(libname)
         if dllfile:
