--- src/mkoctfile.in.cc.orig	2016-11-13 08:16:10.000000000 -0700
+++ src/mkoctfile.in.cc	2016-11-26 14:03:33.000000000 -0700
@@ -586,6 +586,19 @@
         {
           pass_on_options += (" " + arg);
         }
+#if defined __APPLE__ && defined __MACH__
+      else if (arg == "-arch")
+        {
+          if (i < argc-1)
+            {
+              arg = argv[++i];
+              pass_on_options += (" -arch " + arg);
+            }
+          else
+            std::cerr << "mkoctfile: arch type missing"
+                      << std::endl;
+        }
+#endif
       else if (starts_with (arg, "-"))
         {
           // Pass through any unrecognized options
