--- configure.orig	2016-11-13 08:22:50.000000000 -0700
+++ configure	2017-02-26 08:38:24.000000000 -0700
@@ -34697,11 +34697,11 @@
   case "$qt_version" in
     4)
       QT_MODULES="QtCore QtGui QtNetwork QtOpenGL"
-      octave_qscintilla_libnames=qscintilla2
+      octave_qscintilla_libnames="qscintilla2 qscintilla2_qt4"
     ;;
     5)
       QT_MODULES="Qt5Core Qt5Gui Qt5Network Qt5OpenGL Qt5PrintSupport"
-      octave_qscintilla_libnames="qscintilla2-qt5 qt5scintilla2"
+      octave_qscintilla_libnames="qscintilla2-qt5 qt5scintilla2 qscintilla2_qt5"
     ;;
     *)
       as_fn_error $? "Unrecognized Qt version $qt_version" "$LINENO" 5
