--- Makefile.in.orig	2016-11-13 08:23:00.000000000 -0700
+++ Makefile.in	2016-12-10 06:52:04.000000000 -0700
@@ -16690,7 +16690,7 @@
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) libinterp/corefcn/$(DEPDIR)/libcorefcn_la-cdisplay.Tpo libinterp/corefcn/$(DEPDIR)/libcorefcn_la-cdisplay.Plo
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='libinterp/corefcn/cdisplay.c' object='libinterp/corefcn/libcorefcn_la-cdisplay.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libinterp_corefcn_libcorefcn_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libinterp/corefcn/libcorefcn_la-cdisplay.lo `test -f 'libinterp/corefcn/cdisplay.c' || echo '$(srcdir)/'`libinterp/corefcn/cdisplay.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) -x c++ $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libinterp_corefcn_libcorefcn_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libinterp/corefcn/libcorefcn_la-cdisplay.lo `test -f 'libinterp/corefcn/cdisplay.c' || echo '$(srcdir)/'`libinterp/corefcn/cdisplay.c
 
 libinterp/corefcn/libcorefcn_la-coct-hdf5-types.lo: libinterp/corefcn/coct-hdf5-types.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libinterp_corefcn_libcorefcn_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT libinterp/corefcn/libcorefcn_la-coct-hdf5-types.lo -MD -MP -MF libinterp/corefcn/$(DEPDIR)/libcorefcn_la-coct-hdf5-types.Tpo -c -o libinterp/corefcn/libcorefcn_la-coct-hdf5-types.lo `test -f 'libinterp/corefcn/coct-hdf5-types.c' || echo '$(srcdir)/'`libinterp/corefcn/coct-hdf5-types.c
@@ -17047,7 +17047,7 @@
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) src/$(DEPDIR)/octave-display-available.Tpo src/$(DEPDIR)/octave-display-available.Po
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='src/display-available.c' object='src/octave-display-available.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(src_octave_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o src/octave-display-available.o `test -f 'src/display-available.c' || echo '$(srcdir)/'`src/display-available.c
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) -x c++ $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(src_octave_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o src/octave-display-available.o `test -f 'src/display-available.c' || echo '$(srcdir)/'`src/display-available.c
 
 src/octave-display-available.obj: src/display-available.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(src_octave_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT src/octave-display-available.obj -MD -MP -MF src/$(DEPDIR)/octave-display-available.Tpo -c -o src/octave-display-available.obj `if test -f 'src/display-available.c'; then $(CYGPATH_W) 'src/display-available.c'; else $(CYGPATH_W) '$(srcdir)/src/display-available.c'; fi`
