--- libgui/graphics/GLCanvas.cc~	2018-04-30 13:03:56.000000000 -0400
+++ libgui/graphics/GLCanvas.cc	2018-05-15 02:58:43.000000000 -0400
@@ -32,6 +32,9 @@
 #include "GLCanvas.h"
 #include "gl-select.h"
 
+#include <qapplication.h>
+#include <QScreen>
+
 namespace QtHandles
 {
 #if defined (HAVE_QOPENGLWIDGET)
@@ -53,6 +56,9 @@
   {
     setFocusPolicy (Qt::ClickFocus);
     setFocus ();
+
+    QScreen *screen = QApplication::screens().at(0);
+    gh_manager::screen_scale = screen->devicePixelRatio();
   }
 
   GLCanvas::~GLCanvas (void)
--- libinterp/corefcn/gl-render.cc~	2018-04-30 13:03:56.000000000 -0400
+++ libinterp/corefcn/gl-render.cc	2018-05-15 03:05:40.000000000 -0400
@@ -1123,7 +1123,7 @@
     glMultMatrixd (x_mat1.data ());
     glMatrixMode (GL_PROJECTION);
     glLoadIdentity ();
-    glOrtho (0, vw[2], vw[3], 0, xZ1, xZ2);
+    glOrtho (0, roundf((1./gh_manager::screen_scale)*(float)vw[2]), roundf((1./gh_manager::screen_scale)*(float)vw[3]), 0, xZ1, xZ2);
     glMultMatrixd (x_mat2.data ());
     glMatrixMode (GL_MODELVIEW);
 
@@ -3810,7 +3810,7 @@
   {
 #if defined (HAVE_OPENGL)
 
-    glViewport (0, 0, w, h);
+    glViewport (0.0, 0.0, gh_manager::screen_scale*w, gh_manager::screen_scale*h);
 
 #else
 
@@ -3915,7 +3915,7 @@
     txt_renderer.set_font (props.get ("fontname").string_value (),
                            props.get ("fontweight").string_value (),
                            props.get ("fontangle").string_value (),
-                           props.get ("__fontsize_points__").double_value ());
+                           gh_manager::screen_scale*props.get ("__fontsize_points__").double_value ());
   }
 
   void
@@ -4103,7 +4103,7 @@
     glMatrixMode (GL_PROJECTION);
     glPushMatrix ();
     glLoadIdentity ();
-    glOrtho (0, vw[2], vw[3], 0, xZ1, xZ2);
+    glOrtho (0, roundf((1./gh_manager::screen_scale)*(float)vw[2]), roundf((1./gh_manager::screen_scale)*(float)vw[3]), 0, xZ1, xZ2);
     glMatrixMode (GL_MODELVIEW);
     glPushMatrix ();
 
--- libinterp/corefcn/graphics.cc~	2018-04-30 13:03:56.000000000 -0400
+++ libinterp/corefcn/graphics.cc	2018-05-15 03:11:03.000000000 -0400
@@ -6585,7 +6585,7 @@
   txt_renderer.set_font (get ("fontname").string_value (),
                          get ("fontweight").string_value (),
                          get ("fontangle").string_value (),
-                         get ("__fontsize_points__").double_value ());
+                         gh_manager::screen_scale*get ("__fontsize_points__").double_value ());
 }
 
 // The INTERNAL flag defines whether position or outerposition is used.
@@ -8741,7 +8741,7 @@
   txt_renderer.set_font (get ("fontname").string_value (),
                          get ("fontweight").string_value (),
                          get ("fontangle").string_value (),
-                         get ("__fontsize_points__").double_value ());
+                         gh_manager::screen_scale*get ("__fontsize_points__").double_value ());
 
   txt_renderer.set_color (get_color_rgb ());
 }
@@ -9936,6 +9936,8 @@
   return parent_go.get_factory_default (type () + name);
 }
 
+float gh_manager::screen_scale = 1.0;
+
 // We use a random value for the handle to avoid issues with plots and
 // scalar values for the first argument.
 gh_manager::gh_manager (void)
--- libinterp/corefcn/graphics.in.h~	2018-04-30 13:03:56.000000000 -0400
+++ libinterp/corefcn/graphics.in.h	2018-05-15 03:13:32.000000000 -0400
@@ -6047,6 +6047,8 @@
 
 public:
 
+  static float screen_scale;
+
   static void create_instance (void);
 
   static bool instance_ok (void)
