--- libinterp/corefcn/help.h.orig	2023-03-06 23:34:32.000000000 -0600
+++ libinterp/corefcn/help.h	2023-03-08 05:47:48.000000000 -0600
@@ -50,7 +50,7 @@
       m_doc_cache_file (init_doc_cache_file ()),
       m_info_file (init_info_file ()),
       m_info_program (init_info_program ()),
-      m_makeinfo_program ("makeinfo"),
+      m_makeinfo_program ("__MACPORTS_prefix__/bin/makeinfo"),
       m_suppress_verbose_help_message (false),
       m_texi_macros_file (init_texi_macros_file ())
   { }
