--- build-aux/subst-config-vals.in.sh.orig	2018-05-18 22:11:31.000000000 -0700
+++ build-aux/subst-config-vals.in.sh	2018-05-19 05:21:58.000000000 -0700
@@ -20,7 +20,7 @@
 
 : ${SED=@SED@}
 
-canonical_host_type="@canonical_host_type@"
+canonical_host_type="__MACPORTS_canonical_host_type__"
 DEFAULT_PAGER="@DEFAULT_PAGER@"
 EXEEXT="@EXEEXT@"
 man1ext="@man1ext@"
--- configure.ac.orig	2018-05-20 10:22:18.000000000 -0700
+++ configure.ac	2018-05-20 10:24:18.000000000 -0700
@@ -101,37 +101,37 @@
 ## This path usually includes the Octave version and configuration name, so
 ## that configurations for multiple versions of Octave may be installed at once.
 OCTAVE_SET_DEFAULT([archlibdir],
-  '${libexecdir}/octave/${version}/exec/${canonical_host_type}')
+  '${libexecdir}/octave/${version}/exec/__MACPORTS_canonical_host_type__')
 
 ## Where to put executables to be run by Octave, rather than the user, that
 ## are specific to this site.
 OCTAVE_SET_DEFAULT([localarchlibdir],
-  '${libexecdir}/octave/site/exec/${canonical_host_type}')
+  '${libexecdir}/octave/site/exec/__MACPORTS_canonical_host_type__')
 
 OCTAVE_SET_DEFAULT([localapiarchlibdir],
-  '${libexecdir}/octave/${api_version}/site/exec/${canonical_host_type}')
+  '${libexecdir}/octave/${api_version}/site/exec/__MACPORTS_canonical_host_type__')
 
 OCTAVE_SET_DEFAULT([localverarchlibdir],
-  '${libexecdir}/octave/${version}/site/exec/${canonical_host_type}')
+  '${libexecdir}/octave/${version}/site/exec/__MACPORTS_canonical_host_type__')
 
 ## Where to put object files that will by dynamically loaded.
 ## This path includes the Octave version and configuration name, so that
 ## configurations for multiple versions of Octave may be installed at once.
 OCTAVE_SET_DEFAULT([octfiledir],
-  '${libdir}/octave/${version}/oct/${canonical_host_type}')
+  '${libdir}/octave/${version}/oct/__MACPORTS_canonical_host_type__')
 
 ## Directories that Octave should search for object files that will be
 ## dynamically loaded and that are specific to this site (i.e. customizations),
 ## before consulting ${octfiledir}.  This should be a colon-separated list of
 ## directories.
 OCTAVE_SET_DEFAULT([localoctfiledir],
-  '${libdir}/octave/site/oct/${canonical_host_type}')
+  '${libdir}/octave/site/oct/__MACPORTS_canonical_host_type__')
 
 OCTAVE_SET_DEFAULT([localapioctfiledir],
-  '${libdir}/octave/site/oct/${api_version}/${canonical_host_type}')
+  '${libdir}/octave/site/oct/${api_version}/__MACPORTS_canonical_host_type__')
 
 OCTAVE_SET_DEFAULT([localveroctfiledir],
-  '${libdir}/octave/${version}/site/oct/${canonical_host_type}')
+  '${libdir}/octave/${version}/site/oct/__MACPORTS_canonical_host_type__')
 
 ## Where to install Octave's include files.
 OCTAVE_SET_DEFAULT([octincludedir], '${includedir}/octave-${version}/octave')
