--- build-aux/subst-default-vals.in.sh.orig	2016-11-13 08:20:59.000000000 -0700
+++ build-aux/subst-default-vals.in.sh	2016-11-24 08:20:55.000000000 -0700
@@ -22,7 +22,7 @@
 
 archlibdir="@archlibdir@"
 bindir="@bindir@"
-canonical_host_type="@canonical_host_type@"
+canonical_host_type="__MACPORTS_canonical_host_type__"
 datadir="@datadir@"
 datarootdir="@datarootdir@"
 DEFAULT_PAGER="@DEFAULT_PAGER@"
