--- libinterp/corefcn/gl2ps-renderer.cc.orig
+++ libinterp/corefcn/gl2ps-renderer.cc
@@ -34,6 +34,9 @@ along with Octave; see the file COPYING.  If not, see
 #include "gl2ps-renderer.h"
 #include "gl2ps.h"
 
+glps_renderer::glps_renderer (FILE *_fp, const std::string& _term)
+  : opengl_renderer () , fp (_fp), term (_term), fontsize (), fontname () { }
+
 void
 glps_renderer::draw (const graphics_object& go, const std::string print_cmd)
 {
@@ -253,4 +256,31 @@ glps_renderer::draw_text (const text::properties& props)
                 alignment_to_mode (halign, valign), props.get_rotation ());
 }
 
+void
+glps_renderer::set_linestyle (const std::string& s, bool use_stipple)
+{
+  opengl_renderer::set_linestyle (s, use_stipple);
+
+  if (s == "-" && ! use_stipple)
+    gl2psDisable (GL2PS_LINE_STIPPLE);
+  else
+    gl2psEnable (GL2PS_LINE_STIPPLE);
+}
+
+void
+glps_renderer::set_polygon_offset (bool on, double offset)
+{
+  opengl_renderer::set_polygon_offset (on, offset);
+  if (on)
+    gl2psEnable (GL2PS_POLYGON_OFFSET_FILL);
+  else
+    gl2psDisable (GL2PS_POLYGON_OFFSET_FILL);
+}
+
+void
+glps_renderer::set_linewidth (float w)
+{
+  gl2psLineWidth (w);
+}
+
 #endif
--- libinterp/corefcn/gl2ps-renderer.h.orig
+++ libinterp/corefcn/gl2ps-renderer.h
@@ -33,8 +33,7 @@ OCTINTERP_API
 glps_renderer : public opengl_renderer
 {
 public:
-  glps_renderer (FILE *_fp, const std::string& _term)
-    : opengl_renderer () , fp (_fp), term (_term), fontsize (), fontname () { }
+  glps_renderer (FILE *_fp, const std::string& _term);
 
   ~glps_renderer (void) { }
 
@@ -52,29 +51,9 @@ protected:
   void draw_pixels (GLsizei w, GLsizei h, GLenum format,
                     GLenum type, const GLvoid *data);
 
-  void set_linestyle (const std::string& s, bool use_stipple = false)
-  {
-    opengl_renderer::set_linestyle (s, use_stipple);
-
-    if (s == "-" && ! use_stipple)
-      gl2psDisable (GL2PS_LINE_STIPPLE);
-    else
-      gl2psEnable (GL2PS_LINE_STIPPLE);
-  }
-
-  void set_polygon_offset (bool on, double offset = 0.0)
-  {
-    opengl_renderer::set_polygon_offset (on, offset);
-    if (on)
-      gl2psEnable (GL2PS_POLYGON_OFFSET_FILL);
-    else
-      gl2psDisable (GL2PS_POLYGON_OFFSET_FILL);
-  }
-
-  void set_linewidth (float w)
-  {
-    gl2psLineWidth (w);
-  }
+  void set_linestyle (const std::string& s, bool use_stipple = false);
+  void set_polygon_offset (bool on, double offset = 0.0);
+  void set_linewidth (float w);
 
 private:
   int alignment_to_mode (int ha, int va) const;
