diff --git libgui/src/octave-qobject.cc libgui/src/octave-qobject.cc
index 1320283..42cfaca 100644
--- libgui/src/octave-qobject.cc
+++ libgui/src/octave-qobject.cc
@@ -73,7 +73,7 @@ static void disable_app_nap (void)
   unsigned long long NSActivityLatencyCritical = 0xFF00000000ULL;
 
   // Avoid errors on older versions of OS X
-  process_info_class = static_cast<Class> (objc_getClass ("NSProcessInfo"));
+  process_info_class = reinterpret_cast<Class> (objc_getClass ("NSProcessInfo"));
   if (process_info_class == nil)
     return;
 
