--- run-octave.in	2024-03-13 02:00:23.000000000 +0800
+++ run-octave.in	2024-03-23 11:58:26.000000000 +0800
@@ -134,6 +134,13 @@
   export ASAN_OPTIONS
 fi
 
+if [ -n "$DYLD_LIBRARY_PATH" ]; then
+   DYLD_LIBRARY_PATH=%PREFIX%/lib/libgcc:${DYLD_LIBRARY_PATH}
+else
+   DYLD_LIBRARY_PATH=%PREFIX%/lib/libgcc
+fi
+export DYLD_LIBRARY_PATH
+
 exec $builddir/libtool --mode=execute $driver \
   "$octave_executable" --no-init-path --path="$LOADPATH" \
   --image-path="$IMAGEPATH" --doc-cache-file="$DOCFILE" \
