--- Makefile.in.orig	2019-08-28 15:33:26.000000000 -0700
+++ Makefile.in	2019-08-28 15:47:15.000000000 -0700
@@ -27127,7 +27127,7 @@
 $(scripts_java_JAVA_CLASSES) : %.class : %.java | scripts/java/$(octave_dirstamp)
 	$(OCT_V_JAVAC)$(MKDIR_P) scripts/java/$(org_octave_dir) && \
 	( cd $(srcdir)/scripts/java; \
-	  "$(JAVAC)" -source 8 -target 8 -Xlint:-options \
+	  "$(JAVAC)" -source __MACPORTS_JAVA_VERSION__ -target __MACPORTS_JAVA_VERSION__ -Xlint:-options \
 	             -d $(abs_top_builddir)/scripts/java \
 	             $(org_octave_dir)/$(<F) )
 
