--- src/mkoctfile.in.cc.orig	2017-11-08 15:10:37.000000000 -0700
+++ src/mkoctfile.in.cc	2017-11-09 06:09:16.000000000 -0700
@@ -630,6 +630,19 @@
         {
           pass_on_options += (' ' + arg);
         }
+#if defined __APPLE__ && defined __MACH__
+      else if (arg == "-arch")
+        {
+          if (i < argc-1)
+            {
+              arg = argv[++i];
+              pass_on_options += (" -arch " + arg);
+            }
+          else
+            std::cerr << "mkoctfile: arch type missing"
+                      << std::endl;
+        }
+#endif
       else if (starts_with (arg, "-"))
         {
           // Pass through any unrecognized options
