--- src/display.cc.orig	2012-09-21 11:29:13.000000000 -0400
+++ src/display.cc	2012-09-21 11:29:27.000000000 -0400
@@ -41,6 +41,29 @@
 
 display_info *display_info::instance = 0;
 
+#if defined (HAVE_FRAMEWORK_CARBON)
+// for 10.7 and newer only; would work with 10.6, but the function
+// "CGDisplayBitsPerPixel" is still provided by the Carbon framework
+// with a warning that it will be deprecated (but, it works).
+size_t
+CGDisplayBitsPerPixel (CGDirectDisplayID display)
+{
+  CGDisplayModeRef originalMode = CGDisplayCopyDisplayMode (display);
+  CFStringRef pixelEncoding = CGDisplayModeCopyPixelEncoding (originalMode);
+  size_t rv = 8;
+
+  if (CFStringCompare (pixelEncoding, CFSTR (IO32BitDirectPixels), 0) == 0) {
+    rv = 32;
+  } else {
+    if (CFStringCompare (pixelEncoding, CFSTR (IO16BitDirectPixels), 0) == 0) {
+      rv = 16;
+    }
+    // else rv <= 8, by default.
+  }
+  return  (rv);
+}
+#endif
+
 void
 display_info::init (bool query)
 {
