--- liboctave/oct-inttypes.h.orig	2013-04-23 09:15:24.000000000 -0400
+++ liboctave/oct-inttypes.h	2013-04-23 09:15:55.000000000 -0400
@@ -416,7 +416,7 @@
     {
       // Promotion type for multiplication (if exists).
       typedef typename query_integer_type<2*sizeof (T), false>::type mptype;
-      return truncate_int (static_cast<mptype> (x) 
+      return octave_int_base<T>::truncate_int (static_cast<mptype> (x) 
                            * static_cast<mptype> (y));
     }
 
@@ -669,7 +669,7 @@
     {
       // Promotion type for multiplication (if exists).
       typedef typename query_integer_type<2*sizeof (T), true>::type mptype;
-      return truncate_int (static_cast<mptype> (x) 
+      return octave_int_base<T>::truncate_int (static_cast<mptype> (x) 
                            * static_cast<mptype> (y));
     }
 
