--- src/mkoctfile.in.cc.orig	2017-11-08 15:10:37.000000000 -0700
+++ src/mkoctfile.in.cc	2017-11-08 16:16:44.000000000 -0700
@@ -238,16 +238,16 @@
   vars["FLIBS"] = get_variable ("FLIBS", %OCTAVE_CONF_FLIBS%);
 
   vars["OCTAVE_LINK_DEPS"] = get_variable ("OCTAVE_LINK_DEPS",
-                                           %OCTAVE_CONF_MKOCTFILE_OCTAVE_LINK_DEPS%);
+                                           "");
 
   vars["OCTAVE_LINK_OPTS"] = get_variable ("OCTAVE_LINK_OPTS",
-                                           %OCTAVE_CONF_OCTAVE_LINK_OPTS%);
+                                           "");
 
   vars["OCT_LINK_DEPS"] = get_variable ("OCT_LINK_DEPS",
-                                        %OCTAVE_CONF_MKOCTFILE_OCT_LINK_DEPS%);
+                                        "");
 
   vars["OCT_LINK_OPTS"] = get_variable ("OCT_LINK_OPTS",
-                                        %OCTAVE_CONF_OCT_LINK_OPTS%);
+                                        "");
 
   vars["LD_CXX"] = get_variable ("LD_CXX", %OCTAVE_CONF_MKOCTFILE_LD_CXX%);
 
