--- Makefile.orig	2016-06-03 20:25:08.000000000 -0700
+++ Makefile	2016-12-24 21:48:23.000000000 -0800
@@ -63,28 +63,28 @@ strip: $(BINNAME)
 	strip $(BINNAME)
 
 install: $(BINNAME) lib
-	install -d -m 755 $(BINDIR)
-	install -c -m 755 $(BINNAME) $(BINDIR)
-	install -d -m 755 $(LIBDIR)
-	install -c -m 644 $(wildcard lib/*) $(LIBDIR)
-	install -d -m 755 $(INCLUDEDIR)
-	install -c -m 644 $(wildcard include/*) $(INCLUDEDIR)
-	install -d -m 755 $(MANDIR)
-	install -c -m 644 $(MANPAGE) $(MANDIR)
-	install -d -m 755 $(DEMODIR)
-	install -c -m 644 $(wildcard demos/*) $(DEMODIR)
-	install -d -m 755 $(TUTORDIR)
-	install -c -m 644 $(wildcard tutor/*) $(TUTORDIR)
+	install -d -m 755 $(DESTDIR)$(BINDIR)
+	install -c -m 755 $(BINNAME) $(DESTDIR)$(BINDIR)
+	install -d -m 755 $(DESTDIR)$(LIBDIR)
+	install -c -m 644 $(wildcard lib/*) $(DESTDIR)$(LIBDIR)
+	install -d -m 755 $(DESTDIR)$(INCLUDEDIR)
+	install -c -m 644 $(wildcard include/*) $(DESTDIR)$(INCLUDEDIR)
+	install -d -m 755 $(DESTDIR)$(MANDIR)
+	install -c -m 644 $(MANPAGE) $(DESTDIR)$(MANDIR)
+	install -d -m 755 $(DESTDIR)$(DEMODIR)
+	install -c -m 644 $(wildcard demos/*) $(DESTDIR)$(DEMODIR)
+	install -d -m 755 $(DESTDIR)$(TUTORDIR)
+	install -c -m 644 $(wildcard tutor/*) $(DESTDIR)$(TUTORDIR)
 
 install-strip: strip install
 
 
 uninstall:
-	rm -f $(PREFIX)/bin/$(BINNAME)
-	rm -rf $(LIBDIR)
-	rm -rf $(INCLUDEDIR)
-	rm -rf $(DEMODIR)
-	rm -rf $(TUTORDIR)
+	rm -f $(DESTDIR)$(PREFIX)/bin/$(BINNAME)
+	rm -rf $(DESTDIR)$(LIBDIR)
+	rm -rf $(DESTDIR)$(INCLUDEDIR)
+	rm -rf $(DESTDIR)$(DEMODIR)
+	rm -rf $(DESTDIR)$(TUTORDIR)
 
 dist: distclean
 	mkdir $(distdir)
