From 2172257dffa703fd8f567bf6b1268012a4eb7ad7 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 3 Nov 2023 13:21:31 +0800
Subject: [PATCH 1/5] gen.py: do not force random macOS version

---
 build/gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git build/gen.py build/gen.py
index adb622a9..6420b4a1 100755
--- build/gen.py
+++ build/gen.py
@@ -495,7 +495,7 @@ def WriteGNNinja(path, platform, host, options, args_list):
         # This is needed by libc++.
         libs.append('-ldl')
     elif platform.is_darwin():
-      min_mac_version_flag = '-mmacosx-version-min=10.9'
+      min_mac_version_flag = '-mmacosx-version-min=@DEPLOYMENT_TARGET@'
       cflags.append(min_mac_version_flag)
       ldflags.append(min_mac_version_flag)
     elif platform.is_aix():
