diff --git a/flang/include/flang/Support/Fortran-features.h b/flang/include/flang/Support/Fortran-features.h
index 743abf606ab5..db91a3caa74e 100644
--- a/flang/include/flang/Support/Fortran-features.h
+++ b/flang/include/flang/Support/Fortran-features.h
@@ -13,6 +13,11 @@
 #include "flang/Common/enum-set.h"
 #include <string_view>
 #include <vector>
+#include <unordered_map>
+#include <array>
+#include <variant>
+#include <optional>
+#include <string>
 
 namespace Fortran::common {
 
