From 2bf8502dda4a3ec979b15644513b31c0547959b7 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@codesourcery.com>
Date: Sun, 27 Jul 2014 20:35:16 +0100
Subject: [PATCH 1003/1005] Default to ppc7400 for OSX 10.5+

---
 lib/Driver/Tools.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git llvm_master/tools/clang/lib/Driver/Tools.cpp macports_master/tools/clang/lib/Driver/Tools.cpp
index 2819ab3..e7d5a0c 100644
--- llvm_master/tools/clang/lib/Driver/Tools.cpp
+++ macports_master/tools/clang/lib/Driver/Tools.cpp
@@ -1751,6 +1751,9 @@ static std::string getCPUName(const ArgList &Args, const llvm::Triple &T,
         TargetCPUName = "ppc64";
       else if (T.getArch() == llvm::Triple::ppc64le)
         TargetCPUName = "ppc64le";
+      else if (T.isOSDarwin() && !T.isMacOSXVersionLT(10,5))
+        // For 10.5+ Altivec is assumed available.
+        TargetCPUName = "7400";
       else
         TargetCPUName = "ppc";
     }
-- 
2.7.0

