--- llvm-9.0.0.src/tools/clang/lib/DirectoryWatcher/CMakeLists.txt.orig	2019-09-23 17:31:12.000000000 -0700
+++ llvm-9.0.0.src/tools/clang/lib/DirectoryWatcher/CMakeLists.txt	2019-09-23 17:31:35.000000000 -0700
@@ -5,7 +5,7 @@
 set(DIRECTORY_WATCHER_SOURCES DirectoryScanner.cpp)
 set(DIRECTORY_WATCHER_LINK_LIBS "")
 
-if(APPLE)
+if(APPLE AND (CMAKE_OSX_DEPLOYMENT_TARGET GREATER_EQUAL 10.7))
   check_include_files("CoreServices/CoreServices.h" HAVE_CORESERVICES)
   if(HAVE_CORESERVICES)
     list(APPEND DIRECTORY_WATCHER_SOURCES mac/DirectoryWatcher-mac.cpp)
