Index: configure
===================================================================
--- configure	(Revision 24883)
+++ configure	(Arbeitskopie)
@@ -15578,10 +15578,6 @@
 # We enable ffcall by default now.
 #--------------------------------------------------------------------
 do_enable_libffi=no
-# Enable libffi by default on the following machines
-case "$target_os" in
-    darwin*)	do_enable_libffi=yes;;
-esac
 
 # Check whether --enable-libffi was given.
 if test "${enable_libffi+set}" = set; then
Index: configure.ac
===================================================================
--- configure.ac	(Revision 24883)
+++ configure.ac	(Arbeitskopie)
@@ -1804,10 +1804,6 @@
 # We enable ffcall by default now.
 #--------------------------------------------------------------------
 do_enable_libffi=no
-# Enable libffi by default on the following machines
-case "$target_os" in
-    darwin*)	do_enable_libffi=yes;;
-esac
 
 AC_ARG_ENABLE(libffi,
   [  --enable-libffi  		Enable use of libffi library],,
Index: config/config.trampoline.c
===================================================================
--- config/config.trampoline.c	(Revision 24883)
+++ config/config.trampoline.c	(Arbeitskopie)
@@ -60,7 +60,7 @@
 #ifdef __arm__
 register void* env __asm__("r12");
 #endif
-#ifdef __powerpc__
+#if defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__)
 #ifdef __NetBSD__
 register void* env __asm__("r13");
 #else
