Description: Carp 1.25 now has one dot more.
Origin: vendor
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=75526
Bug-Debian: http://bugs.debian.org/711625
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=75526
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-08

--- t/chk_batch.t
+++ t/chk_batch.t
@@ -212,7 +212,7 @@
 sub chop_exception
 {
     my ($ex) = @_;
-    $ex =~ s/ at \S+ line \d+//;
+    $ex =~ s/ at \S+ line \d+\.?//;
     $ex;
 }
