--- doc/Makefile.am.orig	2010-06-13 13:09:32.000000000 -0500
+++ doc/Makefile.am	2013-01-28 00:28:46.000000000 -0600
@@ -24,7 +24,7 @@ CLEANFILES =
 
 if USE_LYNX
 README: README.html
-	$(AM_V_GEN)lynx --dump $^ | sed 's,file://localhost/.*/doc/README.html,README,' > $@
+	$(AM_V_GEN)lynx --dump $^ | LC_CTYPE=C sed 's,file://localhost/.*/doc/README.html,README,' > $@
 
 CLEANFILES += README
 endif
--- doc/Makefile.in.orig	2012-05-15 05:38:54.000000000 -0500
+++ doc/Makefile.in	2013-01-28 00:28:46.000000000 -0600
@@ -455,7 +455,7 @@ uninstall-am:
 
 
 @USE_LYNX_TRUE@README: README.html
-@USE_LYNX_TRUE@	$(AM_V_GEN)lynx --dump $^ | sed 's,file://localhost/.*/doc/README.html,README,' > $@
+@USE_LYNX_TRUE@	$(AM_V_GEN)lynx --dump $^ | LC_CTYPE=C sed 's,file://localhost/.*/doc/README.html,README,' > $@
 
 tidy: README.html
 	tidy -qe < README.html ; true
