From 07370e2c987f02853e367614de6beec285f814f2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 4 Apr 2024 03:31:56 +0800
Subject: [PATCH] Patch gitversion.pri

---
 gitversion.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git gitversion.pri gitversion.pri
index 7202f57..33f7c3b 100644
--- gitversion.pri
+++ gitversion.pri
@@ -53,5 +53,5 @@ DEFINES += GIT_COMMIT_COUNT=\\\"$$GIT_COMMIT_COUNT\\\"
 # This will rewrite Info.plist with full version
 macx {
     INFO_PLIST_PATH = $$shell_quote($${OUT_PWD}/$${TARGET}.app/Contents/Info.plist)
-    QMAKE_POST_LINK += /usr/libexec/PlistBuddy -c \"Set :CFBundleShortVersionString $${VERSION}\" $${INFO_PLIST_PATH}
+    # QMAKE_POST_LINK += /usr/libexec/PlistBuddy -c \"Set :CFBundleShortVersionString $${VERSION}\" $${INFO_PLIST_PATH}
 }
