--- bindings/ocaml/Makefile.ocaml.ocaml-symlinks	2008-03-10 12:15:32.000000000 -0400
+++ bindings/ocaml/Makefile.ocaml	2008-12-02 16:29:01.000000000 -0500
@@ -160,7 +160,7 @@
 
 install-deplibs:
 	$(Verb) for i in $(DestLibs:$(PROJ_libocamldir)/%=%); do \
-	  ln -sf "$(PROJ_libdir)/$$i" "$(PROJ_libocamldir)/$$i"; \
+         ln -sf "${OVERRIDE_libdir}/$$i" "$(PROJ_libocamldir)/$$i"; \
 	done
 
 uninstall-deplibs:
