--- build/src-copyright-check.sh.orig	2015-06-24 16:52:06.000000000 -0700
+++ build/src-copyright-check.sh	2015-06-24 16:53:13.000000000 -0700
@@ -23,7 +23,7 @@
 
 if [ ! -e $copyright_time_file ]
 then
-	touch --date 1980-01-01 $copyright_time_file
+	touch -t 198001010000 $copyright_time_file
 fi
 
 find build/ doc/ src/ test/ Makefile -type f -newer $copyright_time_file | grep -E '(\.(css|hpp|cpp|html|java|pl|js|sh)|Makefile)$' |  xargs build/src-copyright-fix.sh
--- build/src_neaten.sh.orig	2015-06-24 16:57:41.000000000 -0700
+++ build/src_neaten.sh	2015-06-24 16:59:05.000000000 -0700
@@ -23,7 +23,7 @@
 
 if [ ! -e $neaten_time_file ]
 then
-	touch --date 1980-01-01 $neaten_time_file
+	touch -t 198001010000 $neaten_time_file
 fi
 
 find build/ doc/ src/ test/ Makefile -type f -newer $neaten_time_file | grep -vEi 'changelog|copyright|COPYING|README|\.gif|\.png|\.jpg$' | xargs ./build/src_neaten.pl
