--- Makefile.in.orig	2009-07-14 07:07:08.000000000 -0500
+++ Makefile.in	2009-07-14 07:07:29.000000000 -0500
@@ -148,7 +148,7 @@
 #          Creates these directories if they don't exist.
 #          Prefix those paths with ${DESTDIR} (rarely used, usually null;
 #          may be set on a make command line when building contrib RPMs).
-install: 
+install: all
 	cd ${ESLDIR};   ${MAKE} easel_lib
 	cd ${ESLDIR};   ${MAKE} easel_progs
 	cd ${IMPLDIR};  ${MAKE} impl_lib
