diff --git a/cmake/ThirdPartyZlib.cmake b/cmake/ThirdPartyZlib.cmake
index dd8c4811b..c5cf57441 100644
--- cmake/ThirdPartyZlib.cmake.orig
+++ cmake/ThirdPartyZlib.cmake
@@ -16,7 +16,7 @@ IF(WIN32)
     SET(BUILD_ZLIB ON)
 ELSE()
     FIND_PACKAGE(ZLIB QUIET)
-    IF (ZLIB_FOUND AND NOT ZLIB_VERSION_PATCH LESS 7)
+    IF (ZLIB_FOUND AND ZLIB_VERSION_STRING VERSION_GREATER 1.2.8)
         SET(BUILD_ZLIB OFF)
     ELSE ()
         SET(BUILD_ZLIB ON)
