--- Makefile.PL.orig	2016-04-11 13:27:34.000000000 +1000
+++ Makefile.PL	2022-05-29 13:22:09.000000000 +1000
@@ -35,7 +35,7 @@
 } else {
     local $ENV{TMPDIR} = $ENV{TMPDIR};
     my $path_perl = ($^X =~ m!^/!o) ? $^X : `which $^X`;
-    my $cmd = "CC='$Config{cc}' CFLAGS='$Config{ccflags}'";
+    my $cmd = "CC='$ENV{CC}' CFLAGS='$Config{ccflags}'";
     if ($^O eq 'android') {
         # Work around a bug in Android's sh:
         # http://stackoverflow.com/questions/15283220/android-shell-eof
