--- meson.build	2022-02-14 03:47:52.000000000 +0800
+++ meson.build	2023-10-14 02:31:42.000000000 +0800
@@ -44,7 +44,7 @@
 
 i18n = import('i18n')
 gnome= import('gnome')
-python = import('python').find_installation('python3')
+python = import('python').find_installation('@PYTHON_BIN@')
 
 reproducible = get_option('reproducible') ? '-DREPRODUCIBLE' : '-DNOT_REPRODUCIBLE'
 use_deprecated_gtk_api = get_option('webkit') or get_option('thunar') ? '-DGTK_ENABLE_DEPRECATED' : '-DGTK_DISABLE_DEPRECATED'
