--- meson.build	2024-07-02 08:04:06.000000000 +0800
+++ meson.build	2024-07-03 06:12:38.000000000 +0800
@@ -28,7 +28,7 @@
 harfbuzz = dependency('harfbuzz', version: '>= 2.5')
 json = dependency('json-glib-1.0', version: '>= 1.5')
 pangoft2 = dependency('pangoft2', version: '>= 1.45')
-python = import('python').find_installation('python3')
+python = import('python').find_installation('@PYTHON_BIN@')
 sqlite = dependency('sqlite3', version: '>= 3.35')
 xml = dependency('libxml-2.0', version: '>= 2.9.10')
 
