--- configure.orig	2011-09-05 13:36:16.000000000 -0500
+++ configure	2012-06-19 21:57:10.000000000 -0500
@@ -10,7 +10,7 @@
 
 note()
 {
-    echo -n "Looking for $@...  "
+    printf "Looking for $@...  "
     echo    "Looking for $@...  " >&3
 }
 
@@ -150,7 +150,7 @@
 
 # With -O2 or higher you get a warning for strict-aliasing on
 # some versions of gcc.
-echo -n 'Testing for strict aliasing...  '
+printf 'Testing for strict aliasing...  '
 echo 'Testing for strict aliasing...  ' >&3
 cat << EOF > testfile.c
 int main(void)
