--- ./buffer.c.orig	2008-12-11 01:32:56.000000000 -0800
+++ ./buffer.c	2009-01-01 19:58:26.000000000 -0800
@@ -653,7 +653,7 @@
 	int	 count;
 	size_t	 remain, len;
 
-	len = strlcpy(bn, basename(fn), bs);
+	len = strlcpy(bn, basename((char *)fn), bs);
 	if (len >= bs)
 		return (FALSE);
 
--- ./file.c.orig	2008-12-11 01:32:56.000000000 -0800
+++ ./file.c	2009-01-01 19:58:58.000000000 -0800
@@ -676,7 +676,7 @@
 {
 	char *dp;
 	
-	dp = dirname(path);
+	dp = dirname((char *)path);
 	if (*dp && dp[0] == '/' && dp[1] == '\0')
 		return (strdup(""));
 		
