--- src/plugins/debugger/debuggerplugin.cpp.orig	2019-05-31 21:06:24.000000000 +0800
+++ src/plugins/debugger/debuggerplugin.cpp	2019-05-31 21:06:03.000000000 +0800
@@ -1091,7 +1091,7 @@
         const QAction *act = qobject_cast<QAction *>(sender());
         QTC_ASSERT(act, return);
         const BreakpointModelId id = act->data().value<BreakpointModelId>();
-        QTC_ASSERT(id > 0, return);
+        QTC_ASSERT(id, return);
         BreakTreeView::editBreakpoint(id, mainWindow());
     }
 
