--- Makefile.in.orig	2005-01-12 08:12:21.000000000 +1100
+++ Makefile.in	2013-09-24 18:52:28.000000000 +1000
@@ -55,12 +55,12 @@
 
 all:	libquantum.la
 
-libquantum.la: complex.lo measure.lo matrix.lo gates.lo qft.lo classic.lo\
-	qureg.lo decoherence.lo oaddn.lo omuln.lo expn.lo qec.lo version.lo\
+libquantum.la: complex.lo measure.lo matrix.lo gates.lo qft.lo classic.lo \
+	qureg.lo decoherence.lo oaddn.lo omuln.lo expn.lo qec.lo version.lo \
 	objcode.lo density.lo Makefile
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o libquantum.la complex.lo\
-	measure.lo matrix.lo gates.lo oaddn.lo omuln.lo expn.lo qft.lo\
-	classic.lo qureg.lo decoherence.lo qec.lo version.lo objcode.lo\
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o libquantum.la complex.lo \
+	measure.lo matrix.lo gates.lo oaddn.lo omuln.lo expn.lo qft.lo \
+	classic.lo qureg.lo decoherence.lo qec.lo version.lo objcode.lo \
 	density.lo -lm
 
 complex.lo: complex.c complex.h config.h Makefile
@@ -93,7 +93,7 @@
 qureg.lo: qureg.c qureg.h matrix.h config.h complex.h Makefile
 	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c qureg.c
 
-decoherence.lo: decoherence.c decoherence.h measure.h gates.h qureg.h\
+decoherence.lo: decoherence.c decoherence.h measure.h gates.h qureg.h \
 	complex.h config.h Makefile
 	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c decoherence.c
 
@@ -103,7 +103,7 @@
 version.lo: version.c version.h config.h Makefile
 	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c version.c
 
-objcode.lo: objcode.c objcode.h matrix.h gates.h qureg.h measure.h config.h\
+objcode.lo: objcode.c objcode.h matrix.h gates.h qureg.h measure.h config.h \
 	Makefile
 	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c objcode.c
 
@@ -126,11 +126,11 @@
 demos: shor grover
 
 shor: libquantum.la shor.c Makefile
-	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o shor shor.c -I./ -lquantum\
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o shor shor.c -I./ -lquantum \
 	-static	-lm
 
 grover: libquantum.la grover.c Makefile
-	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o grover grover.c -I./\
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o grover grover.c -I./ \
 	-lquantum -static -lm
 
 # Quantum object code tools
@@ -138,7 +138,7 @@
 quobtools: quobprint quobdump
 
 quobprint: libquantum.la quobprint.c objcode.h Makefile
-	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o quobprint quobprint.c\
+	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -o quobprint quobprint.c \
 	-lquantum
 
 quobdump: libquantum.la quobdump.c objcode.h Makefile
@@ -163,7 +163,7 @@
 
 distclean:
 	-rm -rf .libs
-	-rm shor *.lo *.o config.h quantum.h config.status config.log\
+	-rm shor *.lo *.o config.h quantum.h config.status config.log \
 	libquantum.la
 
 # Build the source tarball
@@ -171,7 +171,7 @@
 dist:
 	-rm quantum.h config.h types.h
 	mkdir libquantum-$(VERSION)
-	cp *.c *.h *.in configure COPYING install-sh ltmain.sh config.sub\
+	cp *.c *.h *.in configure COPYING install-sh ltmain.sh config.sub \
 	config.guess aclocal.m4 README INSTALL CHANGES libquantum-$(VERSION)
 	tar czf libquantum-$(VERSION).tar.gz libquantum-$(VERSION)/
 	rm -rf libquantum-$(VERSION)
