--- Source/Core/VideoBackends/Vulkan/CMakeLists.txt.orig
+++ Source/Core/VideoBackends/Vulkan/CMakeLists.txt
@@ -56,8 +56,6 @@
 # Only include the Vulkan headers when building the Vulkan backend
 target_include_directories(videovulkan
 PRIVATE
-  ${CMAKE_SOURCE_DIR}/Externals/Vulkan-Headers/include
-  ${CMAKE_SOURCE_DIR}/Externals/VulkanMemoryAllocator/include
   ${CMAKE_SOURCE_DIR}/Externals/libadrenotools/include
 )

