--- ext/standard/config.m4.orig	2007-07-11 06:56:03.000000000 -0500
+++ ext/standard/config.m4	2008-11-13 02:19:46.000000000 -0600
@@ -311,9 +311,9 @@
 dnl Detect library functions needed by php dns_xxx functions
 dnl ext/standard/dns.h will collect these in a single define: HAVE_DNS_FUNCS
 dnl
-PHP_CHECK_FUNC(res_nmkquery, resolv, bind, socket)
-PHP_CHECK_FUNC(res_nsend, resolv, bind, socket)
-PHP_CHECK_FUNC(dn_expand, resolv, bind, socket)
+PHP_CHECK_FUNC(res_nmkquery, resolv, socket)
+PHP_CHECK_FUNC(res_nsend, resolv, socket)
+PHP_CHECK_FUNC(dn_expand, resolv, socket)
 
 dnl
 dnl Check if atof() accepts NAN
