--- ext/standard/config.m4.orig	2009-01-31 18:05:14.000000000 +0200
+++ ext/standard/config.m4	2009-01-31 18:06:03.000000000 +0200
@@ -249,10 +249,10 @@
 dnl Detect library functions needed by php dns_xxx functions
 dnl ext/standard/dns.h will collect these in a single define: HAVE_DNS_FUNCS
 dnl
-PHP_CHECK_FUNC(res_nmkquery, resolv, bind, socket)
-PHP_CHECK_FUNC(res_nsend, resolv, bind, socket)
+PHP_CHECK_FUNC(res_nmkquery, resolv, socket)
+PHP_CHECK_FUNC(res_nsend, resolv, socket)
 PHP_CHECK_FUNC(res_search, resolv, bind, socket)
-PHP_CHECK_FUNC(dn_expand, resolv, bind, socket)
+PHP_CHECK_FUNC(dn_expand, resolv, socket)
 PHP_CHECK_FUNC(dn_skipname, resolv, bind, socket)
 
 dnl
