--- setup.py.orig	2009-08-05 05:21:24.000000000 -0500
+++ setup.py	2012-01-02 20:10:18.000000000 -0600
@@ -43,8 +43,6 @@
 else:
     update_dirs(cc.library_dirs, [os.path.join(sys.prefix, 'lib'), 
                                   os.path.join(sys.prefix, 'local/lib')])
-    if sys.platform == "darwin":
-        update_list(cc.library_dirs, ["/sw/lib", "/opt/local/lib"])
 
 if 'include_dirs' in os.environ:
     update_dirs(cc.include_dirs, [path.strip() for path in os.environ.get('include_dirs').strip().split(":")])
@@ -52,8 +50,6 @@
     update_dirs(cc.include_dirs, [os.path.join(sys.prefix, 'include'), 
                                   os.path.join(sys.prefix, 'include/freetype2'),
                                   os.path.join(sys.prefix, 'include/fontconfig')])
-    if sys.platform == "darwin":
-        update_dirs(cc.include_dirs, ["/sw/include/freetype2", "/sw/lib/freetype2/include", "/sw/include", "/opt/local/include"])
 
 
 class ConfigScript:
