--- configure.orig	2012-01-02 20:19:33.000000000 -0600
+++ configure	2012-01-02 20:20:12.000000000 -0600
@@ -2075,15 +2075,6 @@
 	CXXFLAGS="$CXXFLAGS -fno-rtti"
 	SHARED="-bundle"
 	SLEXT="bundle"
-	if test -d /opt/local/include && test -d /opt/local/lib; then
-	    CPPFLAGS="${CPPFLAGS} -I/opt/local/include"
-	    LDFLAGS="${LDFLAGS} -L/opt/local/lib"
-	fi
-	# Use fink packages if available.
-	#if test -d /sw/include && test -d /sw/lib; then
-	#    CPPFLAGS="${CPPFLAGS} -I/sw/include"
-	#    LDFLAGS="${LDFLAGS} -L/sw/lib"
-	#fi
 	;;
 esac
 
@@ -8230,7 +8221,7 @@
   ;;
   *)
   as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-as_dummy=""$PATH:/usr/local/bin:/sw/bin:/opt/local/bin""
+as_dummy=""$PATH""
 for as_dir in $as_dummy
 do
   IFS=$as_save_IFS
