--- src/idle-parser.c.orig	2011-05-10 10:08:19.000000000 -0700
+++ src/idle-parser.c	2011-09-13 11:04:07.000000000 -0700
@@ -226,14 +226,22 @@
 	IdleParserPrivate *priv = IDLE_PARSER_GET_PRIVATE(parser);
 	guint i;
 	guint lasti = 0;
-	gchar *tmp;
+	gchar *tmp, *s;
 	gboolean line_ends = FALSE;
 	guint len;
 	gchar concat_buf[2 * (IRC_MSG_MAXLEN + 3)] = {'\0'};
 
 	g_assert(msg != NULL);
 
-	len = strnlen(msg, IRC_MSG_MAXLEN + 3);
+//	len = strnlen(msg, IRC_MSG_MAXLEN + 3);
+
+        s = msg;
+        len = 0;
+
+        while (*s != '\0' && len < IRC_MSG_MAXLEN + 3) {
+                s++;
+                len++;
+        } 
 
 	for (i = 0; i < len; i++) {
 		if ((msg[i] == '\n' || msg[i] == '\r')) {
