--- Makefile.in.org	Fri Nov 19 14:11:38 2004
+++ Makefile.in	Fri Nov 19 14:12:33 2004
@@ -14,7 +14,8 @@
 #uncomment the following if you need libpcap based build under linux
 #(not raccomanded)
 COMPILE_TIME=
-INSTALL_MANPATH=@MANPATH@
+INSTALL_PATH=
+INSTALL_MANPATH=
 @PCAP@
 
 ARSOBJ = ars.o apd.o split.o rapd.o
@@ -72,10 +73,10 @@
 	rm -rf hping3 *.o byteorder byteorder.h systype.h Makefile libars.a .depend
 
 install: hping3
-	cp -f hping3 /usr/sbin/
-	chmod 755 /usr/sbin/hping3
-	ln -s /usr/sbin/hping3 /usr/sbin/hping
-	ln -s /usr/sbin/hping3 /usr/sbin/hping2
+	cp -f hping3 ${INSTALL_PATH}/sbin/
+	chmod 755 ${INSTALL_PATH}/sbin/hping3
+	ln -s ${INSTALL_PATH}/sbin/hping3 ${INSTALL_PATH}/sbin/hping
+	ln -s ${INSTALL_PATH}/sbin/hping3 ${INSTALL_PATH}/sbin/hping2
 	@if [ -d ${INSTALL_MANPATH}/man8 ]; then \
 		cp ./docs/hping3.8 ${INSTALL_MANPATH}/man8; \
 		chmod 644 ${INSTALL_MANPATH}/man8/hping3.8; \
