--- glc_lib/GLC_lib.pro.orig	2010-04-08 11:00:33.000000000 -0400
+++ glc_lib/GLC_lib.pro	2010-04-08 11:01:13.000000000 -0400
@@ -3,7 +3,6 @@
 QT += opengl \
     core
 
-CONFIG+= x86
 CONFIG += exceptions \
     release \
     warn_on
@@ -16,7 +15,7 @@
 
 DEPENDPATH += .
 INCLUDEPATH += .
-INCLUDEPATH += ./zlib
+INCLUDEPATH += ./shading
 
 # Input					
 HEADERS_QUAZIP +=	quazip/crypt.h \
@@ -310,9 +309,10 @@
     			   
 # Linux install configuration
 unix { 
+    LIBS += -lz
     # Location of HEADERS and library
-    LIB_DIR = /usr/lib
-    INCLUDE_DIR = /usr/include
+    LIB_DIR = @@PREFIX@@/lib
+    INCLUDE_DIR = @@PREFIX@@/include
     # Adds a -P to preserve link
 	QMAKE_COPY_FILE = $${QMAKE_COPY_FILE} -P
 	include.path = $${INCLUDE_DIR}/GLC_lib
