diff --git sceneGraph/glc_3dviewcollection.cpp sceneGraph/glc_3dviewcollection.cpp
index 975d49f..b863e6a 100644
--- sceneGraph/glc_3dviewcollection.cpp
+++ sceneGraph/glc_3dviewcollection.cpp
@@ -33,6 +33,7 @@
 #include "glc_spacepartitioning.h"
 
 #include <QtDebug>
+#include <OpenGL/glu.h>
 
 //////////////////////////////////////////////////////////////////////
 // Constructor/Destructor
diff --git shading/glc_material.cpp shading/glc_material.cpp
index 0d02580..42e06b6 100644
--- shading/glc_material.cpp
+++ shading/glc_material.cpp
@@ -27,6 +27,7 @@
 #include "../glc_factory.h"
 
 #include <QtDebug>
+#include <OpenGL/glu.h>
 
 // Class chunk id
 quint32 GLC_Material::m_ChunkId= 0xA703;
diff --git viewport/glc_camera.cpp viewport/glc_camera.cpp
index 05eba54..85b0008 100644
--- viewport/glc_camera.cpp
+++ viewport/glc_camera.cpp
@@ -26,6 +26,7 @@
 #include "glc_camera.h"
 
 #include <QtDebug>
+#include <OpenGL/glu.h>
 
 using namespace glc;
 //////////////////////////////////////////////////////////////////////
diff --git viewport/glc_settargetmover.cpp viewport/glc_settargetmover.cpp
index fce5881..7cb2550 100644
--- viewport/glc_settargetmover.cpp
+++ viewport/glc_settargetmover.cpp
@@ -24,6 +24,8 @@
 #include "glc_viewport.h"
 #include "glc_openglexception.h"
 
+#include <OpenGL/glu.h>
+
 // Default constructor
 GLC_SetTargetMover::GLC_SetTargetMover(GLC_Viewport* pViewport, const QList<GLC_RepMover*>& repsList)
 : GLC_Mover(pViewport, repsList)
diff --git viewport/glc_viewport.cpp viewport/glc_viewport.cpp
index 3fc66e8..5b47c4c 100644
--- viewport/glc_viewport.cpp
+++ viewport/glc_viewport.cpp
@@ -31,6 +31,7 @@
 #include "../sceneGraph/glc_3dviewinstance.h"
 
 #include <QtDebug>
+#include <OpenGL/glu.h>
 
 using namespace glc;
 //////////////////////////////////////////////////////////////////////
