--- Makefile.in.orig	2015-03-29 21:55:55.000000000 -0500
+++ Makefile.in	2015-03-29 22:04:44.000000000 -0500
@@ -884,7 +884,7 @@
 
 # Do not rely on configure.ac AC_CONFIG_FILES for install paths.  The
 # paths should be expanded at make time rather than configure.
-edit_cmd = sed \
+edit_cmd = $(regex_cmd) \
 	 -e 's|@default_style[@]|$(DEFAULT_STYLE)|g' \
 	 -e 's|@default_menu[@]|$(DEFAULT_MENU)|g' \
 	 -e 's|@default_keysfile[@]|$(DEFAULT_KEYSFILE)|g' \
