--- makefile	Mon Aug 18 14:48:23 2003
+++ makefile.new	Wed Feb 11 00:24:18 2004
@@ -1,4 +1,4 @@
-CFLAGS = `sdl-config --cflags` -Wall -DLINUX
+CFLAGS = `sdl-config --cflags` -I${PREFIX}/include -Wall -DLINUX
 LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image
 OBJS = ai.o aliens.o audio.o bullets.o cargo.o collectable.o comms.o debris.o events.o explosions.o game.o globals.o graphics.o init.o intermission.o loadSave.o messages.o misc.o missions.o player.o resources.o script.o shop.o Starfighter.o title.o unpack.o weapons.o
 
@@ -7,9 +7,9 @@
 PACK = starfighter.pak
 DOCS = docs/*
 
-BINDIR = /usr/games/
-DATADIR = /usr/share/games/parallelrealities/
-DOCDIR = /usr/share/doc/starfighter/
+BINDIR = $(PREFIX)/bin/
+DATADIR = $(PREFIX)/share/games/starfighter/
+DOCDIR = $(PREFIX)/share/doc/starfighter/
 # top-level rule to create the program.
 all: $(PROG)
 
@@ -30,8 +30,8 @@
 
 # install
 install:
-	mkdir -p $(DATADIR)
+	install -d $(DATADIR) $(DOCDIR) $(BINDIR)
 	strip $(PROG)
-	install -o root -g games -m 755 $(PROG) $(BINDIR)$(PROG)
-	install -o root -g games -m 644 $(PACK) $(DATADIR)$(PACK)
+	install -m 755 $(PROG) $(BINDIR)$(PROG)
+	install -m 644 $(PACK) $(DATADIR)$(PACK)
 	cp $(DOCS) $(DOCDIR)
