--- CMakeLists.txt.orig	2019-10-07 00:13:09.000000000 -0700
+++ CMakeLists.txt	2019-10-20 16:14:42.000000000 -0700
@@ -312,7 +312,7 @@
 # db_load checking, it's optional
 # ******************************
 
-find_program(DB_LOAD NAMES db_load db4_load db_load-4 db5_load db_load-5)
+find_program(DB_LOAD NAMES db53_load db_load db4_load db_load-4 db5_load db_load-5)
 if(DB_LOAD)
 	set(HAVE_DB_LOAD ON)
 else(DB_LOAD)
