--- config/configure	2024-08-20 11:14:48.000000000 +0800
+++ config/configure	2024-09-29 22:49:13.000000000 +0800
@@ -11440,9 +11440,9 @@
 else
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for native code $LABLGTK_NAME" >&5
 printf %s "checking for native code $LABLGTK_NAME... " >&6; }
-  if test -f $OCAMLLIB/$LABLGTK_NAME/lablgtk.$OCAMLLIB_EXT; then
+  if test -f $OCAMLLIB/site-lib/$LABLGTK_NAME/lablgtk.$OCAMLLIB_EXT; then
      if test "$GUI" = "newgui2"; then
-       if test -f $OCAMLLIB/$LABLGTK_NAME/lablrsvg.$OCAMLLIB_EXT; then
+       if test -f $OCAMLLIB/site-lib/$LABLGTK_NAME/lablrsvg.$OCAMLLIB_EXT; then
          { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
          LABLGTK_CONFIG=yes
@@ -11450,7 +11450,7 @@
          { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
 printf "%s\n" "no" >&6; }
          LABLGTK_CONFIG=no
-         if test -f $OCAMLLIB/$LABLGTK_NAME/lablrsvg.cma; then
+         if test -f $OCAMLLIB/site-lib/$LABLGTK_NAME/lablrsvg.cma; then
            echo "---------------------------------------------------------"
            echo "lablrsvg is not installed properly. See the INSTALL.txt"
            echo "file of mldonkey to see how to compile $LABLGTK_NAME. You"
@@ -11473,7 +11473,7 @@
      { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
 printf "%s\n" "no" >&6; }
      LABLGTK_CONFIG=no
-     if test -f $OCAMLLIB/$LABLGTK_NAME/lablgtk.cma; then
+     if test -f $OCAMLLIB/site-lib/$LABLGTK_NAME/lablgtk.cma; then
        echo "-------------------------------------------------------------"
        echo "$LABLGTK_NAME is not installed properly. See the INSTALL.txt"
        echo "file of mldonkey to see how to compile $LABLGTK_NAME. You"
