--- Makefile.orig	2011-08-08 15:11:57.000000000 +1000
+++ Makefile	2012-07-05 01:32:18.000000000 +1000
@@ -5033,7 +5033,7 @@
 	$(OCAMLDOC) -sort -dump ocamldoc/ocamldocdump -html -d ocamldoc $(INCLUDES) $(libclient_DOC) $(CORE_DOC)
 
 $(LIB)/md4_cc.o: $(LIB)/md4.c
-	$(OCAMLC) -ccopt "$(CFLAGS) -O6 -o $(LIB)/md4_cc.o" -ccopt "" -c $(LIB)/md4.c
+	$(OCAMLC) -ccopt "$(CFLAGS) -O4 -o $(LIB)/md4_cc.o" -ccopt "" -c $(LIB)/md4.c
 
 $(CDK)/heap_c.o: $(CDK)/heap_c.c
 	$(OCAMLC) -ccopt "$(CFLAGS) $(MORECFLAGS) -o $(CDK)/heap_c.o" -ccopt "" -c $(CDK)/heap_c.c
@@ -5447,7 +5447,7 @@
 	$(OCAMLC) $(DEVFLAGS) $(INCLUDES) -c $<
 
 .mlcpp.ml:
-	@cpp -P $< $(FIX_BROKEN_CPP) > $@
+	@$(CPP) -x c -P $< $(FIX_BROKEN_CPP) > $@
 
 %.ml: %.mlp $(BITSTRING)/pa_bitstring.cmo
 	$(CAMLP4OF) build/bitstring.cma $(BITSTRING)/bitstring_persistent.cmo $(BITSTRING)/pa_bitstring.cmo -impl $< -o $@
--- config/Makefile.in.orig	2011-01-24 02:20:25.000000000 +1100
+++ config/Makefile.in	2012-07-05 01:30:23.000000000 +1000
@@ -1434,7 +1434,7 @@
 	$(OCAMLDOC) -sort -dump ocamldoc/ocamldocdump -html -d ocamldoc $(INCLUDES) $(libclient_DOC) $(CORE_DOC)
 
 $(LIB)/md4_cc.o: $(LIB)/md4.c
-	$(OCAMLC) -ccopt "$(CFLAGS) -O6 -o $(LIB)/md4_cc.o" -ccopt "" -c $(LIB)/md4.c
+	$(OCAMLC) -ccopt "$(CFLAGS) -O4 -o $(LIB)/md4_cc.o" -ccopt "" -c $(LIB)/md4.c
 
 $(CDK)/heap_c.o: $(CDK)/heap_c.c
 	$(OCAMLC) -ccopt "$(CFLAGS) $(MORECFLAGS) -o $(CDK)/heap_c.o" -ccopt "" -c $(CDK)/heap_c.c
