From bf4777e5dc00e030ff2015e4d5419079dd79b7d4 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Wed, 31 May 2023 16:15:41 +0200
Subject: [PATCH] fix tests on SBCL-Darwin-arm64

---
 libffi/libffi-types.lisp | 1 +
 1 file changed, 1 insertion(+)

diff --git libffi/libffi-types.lisp libffi/libffi-types.lisp
index b2ec1dd..b8374fe 100644
--- libffi/libffi-types.lisp
+++ libffi/libffi-types.lisp
@@ -73,6 +73,7 @@
  ((:default-abi "FFI_DEFAULT_ABI"))
  #-x86-64
  ((:sysv "FFI_SYSV"))
+ #-arm64
  ((:unix64 "FFI_UNIX64")))
 
 (ctype ffi-abi "ffi_abi")
-- 
2.40.1

