--- src/CMakeLists.txt.orig	2014-05-05 11:17:17.000000000 +0900
+++ src/CMakeLists.txt	2014-05-05 11:17:58.000000000 +0900
@@ -38,7 +38,7 @@
 
 target_link_libraries( lskat kdegames ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTSVG_LIBRARY} )
 
-install(TARGETS lskat  DESTINATION  ${BIN_INSTALL_DIR} )
+install(TARGETS lskat  ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 ########### install files ###############
 
