--- /Users/frank/tmp/carto-1.2.0/lib/carto/tree/rule.js	2019-01-28 18:19:04
+++ lib/carto/tree/rule.js	2025-05-12 11:06:20
@@ -37,7 +37,7 @@
 };
 
 tree.Rule.prototype.toString = function() {
-    return '[' + tree.Zoom.toString(this.zoom) + '] ' + this.name + ': ' + this.value;
+    return '[' + (this.zoom ? this.zoom.toString() : (new tree.Zoom()).toString()) + '] ' + this.name + ': ' + this.value;
 };
 
 tree.Rule.prototype.validate = function (env) {
