--- conf/configure.in.Tcl.orig	2005-07-25 13:26:24.000000000 -0700
+++ conf/configure.in.Tcl	2005-07-25 13:27:11.000000000 -0700
@@ -31,7 +31,7 @@
                 $d/lib"  
 	V_LIB_BTCL=FAIL
 	for dir in $places; do
-		if test -r $dir/libTcl.so -o -r $dir/libTcl.a ; then
+		if test -r $dir/libTcl.so -o -r $dir/libTcl.a -o -r $dir/libtcl.dylib; then
 			V_LIB_BTCL="-L$dir -lTcl"
 			break
 		fi
@@ -41,20 +41,20 @@
 		AC_MSG_ERROR(cannot find libTcl)
 	fi
 
-	AC_MSG_CHECKING(for tcl2c++)
-	places="$d \
-		"
-	V_TCL2CPP=FAIL
-	for dir in $places; do
-		if test -r $dir/tcl2c++ ; then
-			V_TCL2CPP=$dir/tcl2c++
-			break
-		fi
-	done
-	AC_MSG_RESULT($V_TCL2CPP)
-	if test "$V_TCL2CPP" = FAIL ; then
-		AC_MSG_ERROR(cannot find tcl2c++)
-	fi
+#	AC_MSG_CHECKING(for tcl2c++)
+#	places="$d \
+#		"
+#	V_TCL2CPP=FAIL
+#	for dir in $places; do
+#		if test -r $dir/tcl2c++ ; then
+#			V_TCL2CPP=$dir/tcl2c++
+#			break
+#		fi
+#	done
+#	AC_MSG_RESULT($V_TCL2CPP)
+#	if test "$V_TCL2CPP" = FAIL ; then
+#		AC_MSG_ERROR(cannot find tcl2c++)
+#	fi
 else
 	V_INCLUDE_BTCL=FAIL
 	AC_MSG_CHECKING(for Tcl.h)
