--- a/storage/maria/CMakeLists.txt	2012-06-09 10:56:52.000000000 -0700
+++ b/storage/maria/CMakeLists.txt	2012-06-10 14:30:33.000000000 -0700
@@ -92,4 +92,7 @@
 CMAKE_DEPENDENT_OPTION(USE_ARIA_FOR_TMP_TABLES "Use Aria for temporary tables" ON
                        "WITH_ARIA_STORAGE_ENGINE" OFF)
 
-ADD_SUBDIRECTORY(unittest)
+IF(WITH_MARIA_STORAGE_ENGINE)
+  ENABLE_TESTING()
+  ADD_SUBDIRECTORY(unittest)
+ENDIF()
