--- Makefile.orig	2015-10-06 23:52:28.000000000 +0200
+++ Makefile	2015-12-03 12:57:37.000000000 +0100
@@ -21,28 +21,14 @@
   override LDFLAGS+=-pg
 endif
 
-ifeq ($(call pkgconfig, --atleast-version=0.1.0 unibilium && echo 1),1)
-  override CFLAGS +=$(call pkgconfig, --cflags unibilium) -DHAVE_UNIBILIUM
-  override LDFLAGS+=$(call pkgconfig, --libs   unibilium)
-else ifeq ($(call pkgconfig, tinfo && echo 1),1)
-  override CFLAGS +=$(call pkgconfig, --cflags tinfo)
-  override LDFLAGS+=$(call pkgconfig, --libs   tinfo)
-else ifeq ($(call pkgconfig, ncursesw && echo 1),1)
-  override CFLAGS +=$(call pkgconfig, --cflags ncursesw)
-  override LDFLAGS+=$(call pkgconfig, --libs   ncursesw)
-else
-  override LDFLAGS+=-lncurses
-endif
+override CFLAGS +=$(call pkgconfig, --cflags unibilium) -DHAVE_UNIBILIUM
+override LDFLAGS+=$(call pkgconfig, --libs   unibilium)
 
 OBJECTS=termkey.lo driver-csi.lo driver-ti.lo
 LIBRARY=libtermkey.la
 
 DEMOS=demo demo-async
 
-ifeq ($(call pkgconfig, glib-2.0 && echo 1),1)
-  DEMOS+=demo-glib
-endif
-
 DEMO_OBJECTS=$(DEMOS:=.lo)
 
 TESTSOURCES=$(wildcard t/[0-9]*.c)
