Fix parallel build.
https://github.com/egallesio/STklos/issues/493
--- utils/tmpcomp.orig	2023-03-10 11:37:34.000000000 -0600
+++ utils/tmpcomp	2023-03-16 15:07:59.000000000 -0500
@@ -82,7 +82,16 @@
 
 ${prefix}/src/stklos -c -q -b ${prefix}/src/boot.img \
      -f ${prefix}/utils/stklos-compile.stk -- \
-     --no-time $Copt --output=$out $in
+     --no-time $Copt --output="$out.$$" "$in"
+
+status=$?
+if test $status -gt 0
+then
+    rm -f "$out.$$"
+    exit $status
+fi
+
+test -e "$out.$$" && mv "$out.$$" "$out"
 
 if test X$Copt = X -a -e $out
 then
