--- meson.build	2024-05-04 04:12:25
+++ meson.build	2024-11-27 06:07:18
@@ -28,7 +28,7 @@
 
 message('Looking for dependencies')
 # We need pykeepass>=4.0.7.post1 and pyotp>=2.4.0.
-python_bin = python.find_installation('python3', modules:[
+python_bin = import('python').find_installation('@PYTHON@', modules:[
   'pykeepass',
   'pyotp',
   'validators',
@@ -37,9 +37,9 @@
   'yubico',
 ])
 if not python_bin.found()
-  error('No valid python3 binary found')
+  error('No valid python binary found')
 else
-  message('Found python3 binary')
+  message('Found python binary')
 endif
 
 if not python_bin.language_version().version_compare('>= 3.8')
@@ -53,7 +53,7 @@
 dependency('gtk4', version: '>=4.9')
 dependency('libadwaita-1', version: '>=1.5.alpha')
 
-python_dir = python_bin.get_install_dir()
+python_dir = '@PY_PACKAGES@'
 DATA_DIR = join_paths(get_option('prefix'), get_option('datadir'), meson.project_name())
 bindir = join_paths(get_option('prefix'), get_option('bindir'))
 
