diff --git a/libmariadb/libmariadb/CMakeLists.txt b/libmariadb/libmariadb/CMakeLists.txt
index e4217439..6fa6b468 100644
--- a/libmariadb/libmariadb/CMakeLists.txt
+++ b/libmariadb/libmariadb/CMakeLists.txt
@@ -313,9 +313,6 @@
   IF(ICONV_INCLUDE_DIR)
      INCLUDE_DIRECTORIES(BEFORE ${ICONV_INCLUDE_DIR}) 
   ENDIF()
-  IF(NOT CMAKE_SYSTEM_NAME MATCHES AIX)
-    ADD_DEFINITIONS(-DLIBICONV_PLUG)
-  ENDIF()
   IF(APPLE)
     ADD_DEFINITIONS(-Wno-deprecated-declarations)
   ENDIF()
