--- a/storage/columnstore/columnstore/cmake/columnstore_version.cmake.orig	2020-08-06 08:47:37.000000000 -0700
+++ b/storage/columnstore/columnstore/cmake/columnstore_version.cmake	2020-08-24 22:19:07.000000000 -0700
@@ -2,7 +2,7 @@
 
 # Generate "something" to trigger cmake rerun when VERSION changes
 CONFIGURE_FILE(
-    ${ENGINE_SRC_DIR}/VERSION
+    ${ENGINE_SRC_DIR}/VERSION.txt
   ${CMAKE_BINARY_DIR}/VERSION.dep
 )
 
@@ -10,7 +10,7 @@
 
 MACRO(COLUMNSTORE_GET_CONFIG_VALUE keyword var)
  IF(NOT ${var})
-     FILE (STRINGS ${ENGINE_SRC_DIR}/VERSION str REGEX "^[ ]*${keyword}=")
+     FILE (STRINGS ${ENGINE_SRC_DIR}/VERSION.txt str REGEX "^[ ]*${keyword}=")
    IF(str)
      STRING(REPLACE "${keyword}=" "" str ${str})
      STRING(REGEX REPLACE  "[ ].*" ""  str "${str}")
