--- Makefile.orig
+++ Makefile
@@ -1135,7 +1135,7 @@ dist/test/c/Hacl_Test_ECDSA.c: KRML_EXTRA=-drop Lib.IntTypes.Intrinsics -add-inc
 
 copy-kremlib:
 	mkdir -p dist/kremlin
-	(cd $(KREMLIN_HOME) && tar cvf - kremlib/dist/minimal include) | (cd dist/kremlin && tar xf -)
+	(cd $(KREMLIN_HOME) && tar Lcvf - kremlib/dist/minimal include) | (cd dist/kremlin && tar xf -)
 
 compile-%: dist/Makefile.tmpl dist/configure dist/%/Makefile.basic | copy-kremlib
 	cp $< dist/$*/Makefile
