--- mfhdf/examples/Makefile.in.orig	2014-02-10 11:31:26.000000000 +0900
+++ mfhdf/examples/Makefile.in	2014-06-14 17:49:17.000000000 +0900
@@ -266,7 +266,7 @@
 	SD_get_info.c SD_find_sds_by_name.c SD_set_get_dim_info.c        \
 	SD_dimscale_vs_sds.c SD_set_attr.c SD_get_attr.c SD_chunking_example.c
 
-EXAMPLEDIR = $(prefix)/examples/c
+EXAMPLEDIR = $(DESTDIR)$(prefix)/share/hdf4/examples/c
 DISTCLEANFILES = *.chklog *.chkexe .deps
 @BUILD_SHARED_SZIP_CONDITIONAL_TRUE@LD_LIBRARY_PATH = $(LL_PATH)
 
