--- hdf/examples/Makefile.in.orig	2016-06-29 22:23:21.000000000 +0900
+++ hdf/examples/Makefile.in	2016-10-08 12:41:37.000000000 +0900
@@ -561,8 +561,8 @@
 INSTALL_TOP_FILES = README
 
 # Where to install example files
-EXAMPLEDIR = ${DESTDIR}$(prefix)/examples/c
-EXAMPLETOPDIR = ${DESTDIR}$(prefix)/examples
+EXAMPLEDIR = ${DESTDIR}$(prefix)/share/hdf4/examples/c
+EXAMPLETOPDIR = ${DESTDIR}$(prefix)/share/hdf4/examples
 DISTCLEANFILES = *.chkexe *.chklog
 @BUILD_SHARED_SZIP_CONDITIONAL_TRUE@LD_LIBRARY_PATH = $(LL_PATH)
 
