--- src/option.c.orig	2015-10-17 12:05:14.000000000 -0700
+++ src/option.c	2015-10-17 12:06:02.000000000 -0700
@@ -2074,9 +2074,11 @@
     if (zenity_text_font) 
       zenity_option_error (zenity_option_get_name (text_options, &zenity_text_font),
                            ERROR_SUPPORT);
+#ifdef HAVE_WEBKITGTK
     if (zenity_text_enable_html)
       zenity_option_error (zenity_option_get_name (text_options, &zenity_text_enable_html),
                            ERROR_SUPPORT);
+#endif
   }
   return TRUE;
 }
