--- Makefile.am.orig	2019-04-22 21:58:15.000000000 +0300
+++ Makefile.am	2019-04-22 21:58:30.000000000 +0300
@@ -3,10 +3,10 @@
 sbin_PROGRAMS = bwping bwping6
 
 bwping_SOURCES = include/cygwin.h include/features.h src/bwping.c
-bwping_CFLAGS  = -DBUILD_BWPING -Wall -Wextra -Wpedantic
+bwping_CFLAGS  = -DBUILD_BWPING -Wall -Wextra -pedantic
 
 bwping6_SOURCES = include/cygwin.h include/features.h src/bwping.c
-bwping6_CFLAGS  = -DBUILD_BWPING6 -Wall -Wextra -Wpedantic
+bwping6_CFLAGS  = -DBUILD_BWPING6 -Wall -Wextra -pedantic
 
 man_MANS   = man/bwping.8 man/bwping6.8
 EXTRA_DIST = man/bwping.8 man/bwping6.8
--- Makefile.in.orig	2019-04-22 21:58:35.000000000 +0300
+++ Makefile.in	2019-04-22 21:58:51.000000000 +0300
@@ -488,9 +488,9 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = subdir-objects
 bwping_SOURCES = include/cygwin.h include/features.h src/bwping.c
-bwping_CFLAGS = -DBUILD_BWPING -Wall -Wextra -Wpedantic
+bwping_CFLAGS = -DBUILD_BWPING -Wall -Wextra -pedantic
 bwping6_SOURCES = include/cygwin.h include/features.h src/bwping.c
-bwping6_CFLAGS = -DBUILD_BWPING6 -Wall -Wextra -Wpedantic
+bwping6_CFLAGS = -DBUILD_BWPING6 -Wall -Wextra -pedantic
 man_MANS = man/bwping.8 man/bwping6.8
 EXTRA_DIST = man/bwping.8 man/bwping6.8
 dist_check_SCRIPTS = tests/bwping tests/bwping6
