--- gnuradio-core/src/lib/io/ppio_ppdev.h.orig	2014-04-25 12:10:07.000000000 -0400
+++ gnuradio-core/src/lib/io/ppio_ppdev.h	2014-04-25 12:10:28.000000000 -0400
@@ -35,7 +35,7 @@
  */
 
 class GR_CORE_API ppio_ppdev : public ppio {
-  friend GR_CORE_API ppio_ppdev_sptr make_ppio_ppdev (int which = 0);
+  friend GR_CORE_API ppio_ppdev_sptr make_ppio_ppdev (int which);
   ppio_ppdev (int which = 0);
 
  public:
@@ -54,8 +54,8 @@
   int	d_fd;
 };
 
-ppio_ppdev_sptr
-make_ppio_ppdev (int which);
+GR_CORE_API ppio_ppdev_sptr
+make_ppio_ppdev (int which = 0);
 
 
 #endif /* INCLUDED_PPIO_PPDEV_H */
