diff --git a/cmake/Modules/GrSwig.cmake b/cmake/Modules/GrSwig.cmake
index 3c814c5..8cf1265 100644
--- a/cmake/Modules/GrSwig.cmake
+++ b/cmake/Modules/GrSwig.cmake
@@ -130,8 +130,10 @@ macro(GR_SWIG_MAKE name)
     find_package(PythonLibs)
     list(APPEND GR_SWIG_INCLUDE_DIRS ${PYTHON_INCLUDE_PATH}) #deprecated name (now dirs)
     list(APPEND GR_SWIG_INCLUDE_DIRS ${PYTHON_INCLUDE_DIRS})
-    list(APPEND GR_SWIG_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR})
-    list(APPEND GR_SWIG_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR})
+
+    #prepend local swig directories
+    list(INSERT GR_SWIG_INCLUDE_DIRS 0 ${CMAKE_CURRENT_SOURCE_DIR})
+    list(INSERT GR_SWIG_INCLUDE_DIRS 0 ${CMAKE_CURRENT_BINARY_DIR})
 
     #determine include dependencies for swig file
     execute_process(
diff --git a/gnuradio-runtime/lib/CMakeLists.txt b/gnuradio-runtime/lib/CMakeLists.txt
index a87141c..a23202b 100644
--- a/gnuradio-runtime/lib/CMakeLists.txt
+++ b/gnuradio-runtime/lib/CMakeLists.txt
@@ -219,7 +219,8 @@ list(APPEND test_gnuradio_runtime_sources
   qa_runtime.cc
 )
 
-include_directories(${CPPUNIT_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR}/math)
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/math)
+include_directories(${CPPUNIT_INCLUDE_DIRS})
 link_directories(${CPPUNIT_LIBRARY_DIRS})
 
 add_library(test-gnuradio-runtime SHARED ${test_gnuradio_runtime_sources})
diff --git a/gnuradio-runtime/lib/pmt/CMakeLists.txt b/gnuradio-runtime/lib/pmt/CMakeLists.txt
index 1a8cc57..0cc46e4 100644
--- a/gnuradio-runtime/lib/pmt/CMakeLists.txt
+++ b/gnuradio-runtime/lib/pmt/CMakeLists.txt
@@ -20,8 +20,8 @@
 ########################################################################
 # This file included, use CMake directory variables
 ########################################################################
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR})
 
 ########################################################################
 # Generate serial tags header file
@@ -144,4 +144,4 @@ add_executable(gr_pmt_test test_pmt.cc)
 target_link_libraries(gr_pmt_test test-gnuradio-pmt)
 GR_ADD_TEST(gr-pmt-test gr_pmt_test)
 
-endif(ENABLE_TESTING)
\ No newline at end of file
+endif(ENABLE_TESTING)
diff --git a/gr-audio/lib/CMakeLists.txt b/gr-audio/lib/CMakeLists.txt
index 3e29319..4eb6f8e 100644
--- a/gr-audio/lib/CMakeLists.txt
+++ b/gr-audio/lib/CMakeLists.txt
@@ -47,7 +47,8 @@ find_package(ALSA)
 
 if(ALSA_FOUND)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/alsa ${ALSA_INCLUDE_DIRS})
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/alsa)
+    include_directories(${ALSA_INCLUDE_DIRS})
     list(APPEND gr_audio_libs ${ALSA_LIBRARIES})
     list(APPEND gr_audio_sources
         ${CMAKE_CURRENT_SOURCE_DIR}/alsa/alsa_impl.cc
@@ -65,7 +66,8 @@ find_package(OSS)
 
 if(OSS_FOUND)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/oss ${OSS_INCLUDE_DIRS})
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/oss)
+    include_directories(${OSS_INCLUDE_DIRS})
     list(APPEND gr_audio_sources
         ${CMAKE_CURRENT_SOURCE_DIR}/oss/oss_source.cc
         ${CMAKE_CURRENT_SOURCE_DIR}/oss/oss_sink.cc
@@ -82,7 +84,8 @@ find_package(Jack)
 
 if(JACK_FOUND)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/jack ${JACK_INCLUDE_DIRS})
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/jack)
+    include_directories(${JACK_INCLUDE_DIRS})
     list(APPEND gr_audio_libs ${JACK_LIBRARIES})
     add_definitions(${JACK_DEFINITIONS})
     list(APPEND gr_audio_sources
@@ -103,7 +106,7 @@ CHECK_INCLUDE_FILE_CXX(AudioToolbox/AudioToolbox.h AUDIO_TOOLBOX_H)
 
 if(AUDIO_UNIT_H AND AUDIO_TOOLBOX_H)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/osx)
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/osx)
     list(APPEND gr_audio_libs
         "-framework AudioUnit"
         "-framework CoreAudio"
@@ -124,7 +127,8 @@ find_package(Portaudio)
 
 if(PORTAUDIO_FOUND)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/portaudio ${PORTAUDIO_INCLUDE_DIRS})
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/portaudio)
+    include_directories(${PORTAUDIO_INCLUDE_DIRS})
     list(APPEND gr_audio_libs ${PORTAUDIO_LIBRARIES})
     add_definitions(${PORTAUDIO_DEFINITIONS})
     list(APPEND gr_audio_sources
@@ -141,7 +145,7 @@ endif(PORTAUDIO_FOUND)
 ########################################################################
 if(WIN32)
 
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/windows)
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/windows)
     list(APPEND gr_audio_libs winmm.lib)
     list(APPEND gr_audio_sources
         ${CMAKE_CURRENT_SOURCE_DIR}/windows/windows_source.cc
diff --git a/gr-fec/CMakeLists.txt b/gr-fec/CMakeLists.txt
index 05e63cf..9a3e227 100644
--- a/gr-fec/CMakeLists.txt
+++ b/gr-fec/CMakeLists.txt
@@ -40,11 +40,6 @@ GR_SET_GLOBAL(GR_FEC_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/include
 )
 
-if(ENABLE_GR_CTRLPORT)
-  ADD_DEFINITIONS(-DGR_CTRLPORT)
-  include_directories(${ICE_INCLUDE_DIR})
-endif(ENABLE_GR_CTRLPORT)
-
 ########################################################################
 # Begin conditional configuration
 ########################################################################
diff --git a/gr-fec/lib/CMakeLists.txt b/gr-fec/lib/CMakeLists.txt
index 29eb36b..34c0746 100644
--- a/gr-fec/lib/CMakeLists.txt
+++ b/gr-fec/lib/CMakeLists.txt
@@ -33,6 +33,11 @@ include_directories(
     ${Boost_INCLUDE_DIRS}
 )
 
+if(ENABLE_GR_CTRLPORT)
+  ADD_DEFINITIONS(-DGR_CTRLPORT)
+  include_directories(${ICE_INCLUDE_DIR})
+endif(ENABLE_GR_CTRLPORT)
+
 link_directories(${Boost_LIBRARY_DIRS})
 
 ########################################################################
diff --git a/gr-filter/CMakeLists.txt b/gr-filter/CMakeLists.txt
index f9f1f39..9958751 100644
--- a/gr-filter/CMakeLists.txt
+++ b/gr-filter/CMakeLists.txt
@@ -44,11 +44,6 @@ GR_SET_GLOBAL(GR_FILTER_INCLUDE_DIRS
 
 SET(GR_PKG_FILTER_EXAMPLES_DIR ${GR_PKG_DATA_DIR}/examples/filter)
 
-if(ENABLE_GR_CTRLPORT)
-  ADD_DEFINITIONS(-DGR_CTRLPORT)
-  include_directories(${ICE_INCLUDE_DIR})
-endif(ENABLE_GR_CTRLPORT)
-
 ########################################################################
 # Begin conditional configuration
 ########################################################################
diff --git a/gr-pager/CMakeLists.txt b/gr-pager/CMakeLists.txt
index 04c7e2c..a776bbb 100644
--- a/gr-pager/CMakeLists.txt
+++ b/gr-pager/CMakeLists.txt
@@ -44,11 +44,6 @@ GR_SET_GLOBAL(GR_PAGER_SWIG_INCLUDE_DIRS
 
 SET(GR_PKG_PAGER_EXAMPLES_DIR ${GR_PKG_DATA_DIR}/examples/pager)
 
-if(ENABLE_GR_CTRLPORT)
-  ADD_DEFINITIONS(-DGR_CTRLPORT)
-  include_directories(${ICE_INCLUDE_DIR})
-endif(ENABLE_GR_CTRLPORT)
-
 ########################################################################
 # Begin conditional configuration
 ########################################################################
diff --git a/gr-pager/lib/CMakeLists.txt b/gr-pager/lib/CMakeLists.txt
index fa87068..1fcdfe3 100644
--- a/gr-pager/lib/CMakeLists.txt
+++ b/gr-pager/lib/CMakeLists.txt
@@ -27,6 +27,11 @@ include_directories(
     ${Boost_INCLUDE_DIRS}
 )
 
+if(ENABLE_GR_CTRLPORT)
+  ADD_DEFINITIONS(-DGR_CTRLPORT)
+  include_directories(${ICE_INCLUDE_DIR})
+endif(ENABLE_GR_CTRLPORT)
+
 link_directories(${Boost_LIBRARY_DIRS})
 
 include_directories(${LOG4CPP_INCLUDE_DIRS})
diff --git a/gr-vocoder/lib/codec2/CMakeLists.txt b/gr-vocoder/lib/codec2/CMakeLists.txt
index 95e9d25..f4ccd22 100644
--- a/gr-vocoder/lib/codec2/CMakeLists.txt
+++ b/gr-vocoder/lib/codec2/CMakeLists.txt
@@ -22,7 +22,7 @@
 # http://www.vtk.org/Wiki/CMake_Cross_Compiling#Using_executables_in_the_build_created_during_the_build
 ########################################################################
 if(NOT CMAKE_CROSSCOMPILING)
-    include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+    include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR})
     add_executable(generate_codebook ${CMAKE_CURRENT_SOURCE_DIR}/generate_codebook.c)
     target_link_libraries(generate_codebook -lm)
     export(TARGETS generate_codebook APPEND FILE ${EXPORT_FILE})
diff --git a/gr-vocoder/lib/g7xx/CMakeLists.txt b/gr-vocoder/lib/g7xx/CMakeLists.txt
index bde7538..246af0e 100644
--- a/gr-vocoder/lib/g7xx/CMakeLists.txt
+++ b/gr-vocoder/lib/g7xx/CMakeLists.txt
@@ -20,7 +20,7 @@
 ########################################################################
 # Append all sources in this dir
 ########################################################################
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR})
 list(APPEND gr_vocoder_sources
     ${CMAKE_CURRENT_SOURCE_DIR}/g711.c
     ${CMAKE_CURRENT_SOURCE_DIR}/g72x.c
diff --git a/gr-vocoder/lib/gsm/CMakeLists.txt b/gr-vocoder/lib/gsm/CMakeLists.txt
index 8a4813a..dfbade8 100644
--- a/gr-vocoder/lib/gsm/CMakeLists.txt
+++ b/gr-vocoder/lib/gsm/CMakeLists.txt
@@ -20,7 +20,7 @@
 ########################################################################
 # Append all sources in this dir
 ########################################################################
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR})
 
 set(gsm_sources
     ${CMAKE_CURRENT_SOURCE_DIR}/add.c
