--- gnuradio-runtime/CMakeLists.txt.orig
+++ gnuradio-runtime/CMakeLists.txt
@@ -44,7 +44,6 @@
     ${CMAKE_BINARY_DIR}
     ${CMAKE_SOURCE_DIR}/gnuradio-runtime/include
     ${CMAKE_BINARY_DIR}/gnuradio-runtime/include
-    ${LOG4CPP_INCLUDE_DIRS}
 )
 
 GR_SET_GLOBAL(GNURADIO_RUNTIME_SWIG_INCLUDE_DIRS
--- gnuradio-runtime/apps/CMakeLists.txt.orig
+++ gnuradio-runtime/apps/CMakeLists.txt
@@ -23,6 +23,7 @@
 include_directories(
   ${GNURADIO_RUNTIME_INCLUDE_DIRS}
   ${Boost_INCLUDE_DIRS}
+  ${LOG4CPP_INCLUDE_DIRS}
 )
 
 link_directories(
--- gnuradio-runtime/lib/CMakeLists.txt.orig
+++ gnuradio-runtime/lib/CMakeLists.txt
@@ -51,6 +51,7 @@
 		    ${CMAKE_CURRENT_BINARY_DIR}/../include/
 		    ${VOLK_INCLUDE_DIRS}
                     ${Boost_INCLUDE_DIRS}
+                    ${LOG4CPP_INCLUDE_DIRS}
 )
 
 if(ENABLE_CTRLPORT_THRIFT)
--- gnuradio-runtime/swig/CMakeLists.txt.orig
+++ gnuradio-runtime/swig/CMakeLists.txt
@@ -28,6 +28,7 @@
     ${GNURADIO_RUNTIME_SWIG_INCLUDE_DIRS}
     ${CMAKE_CURRENT_BINARY_DIR}
     ${Boost_INCLUDE_DIRS}
+    ${LOG4CPP_INCLUDE_DIRS}
 )
 
 link_directories(${Boost_LIBRARY_DIRS})
--- gr-blocks/CMakeLists.txt.orig
+++ gr-blocks/CMakeLists.txt
@@ -37,7 +37,6 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/include
     ${CMAKE_CURRENT_BINARY_DIR}/lib
     ${CMAKE_CURRENT_BINARY_DIR}/include
-    ${VOLK_INCLUDE_DIRS}
 )
 
 ########################################################################
--- gr-blocks/lib/CMakeLists.txt.orig
+++ gr-blocks/lib/CMakeLists.txt
@@ -65,6 +65,7 @@
     ${CMAKE_CURRENT_BINARY_DIR}/../include
     ${GR_BLOCKS_INCLUDE_DIRS}
     ${GNURADIO_RUNTIME_INCLUDE_DIRS}
+    ${VOLK_INCLUDE_DIRS}
     ${Boost_INCLUDE_DIRS}
 )
 
--- gr-channels/lib/CMakeLists.txt.orig
+++ gr-channels/lib/CMakeLists.txt
@@ -28,6 +28,7 @@
     ${GR_FILTER_INCLUDE_DIRS}
     ${GR_ANALOG_INCLUDE_DIRS}
     ${GNURADIO_RUNTIME_INCLUDE_DIRS}
+    ${VOLK_INCLUDE_DIRS}
     ${Boost_INCLUDE_DIRS}
 )
 
