--- cmake/Modules/GrMiscUtils.cmake.orig
+++ cmake/Modules/GrMiscUtils.cmake
@@ -377,6 +377,13 @@ if __name__ == '__main__':
   endforeach(sig)
   unset(name)
 
+  #try to force generate the headers
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${PYTHON_DASH_B}
+    ${CMAKE_CURRENT_BINARY_DIR}/generate_helper.py
+    ${root} ${root}.h.t ${ARGN}
+  )
+
   #create a command to generate the headers
   add_custom_command(
     OUTPUT ${expanded_files_h}
@@ -422,6 +429,13 @@ if __name__ == '__main__':
   endforeach(sig)
   unset(name)
 
+  #try to force generate the source files
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${PYTHON_DASH_B}
+    ${CMAKE_CURRENT_BINARY_DIR}/generate_helper.py
+    ${root} ${root}.cc.t ${ARGN}
+  )
+
   #create a command to generate the source files
   add_custom_command(
     OUTPUT ${expanded_files_cc}
@@ -431,6 +445,13 @@ if __name__ == '__main__':
     ${root} ${root}.cc.t ${ARGN}
   )
 
+  #try to force generate the header files
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${PYTHON_DASH_B}
+    ${CMAKE_CURRENT_BINARY_DIR}/generate_helper.py
+    ${root} ${root}.h.t ${ARGN}
+  )
+
   #create a command to generate the header files
   add_custom_command(
     OUTPUT ${expanded_files_h}
@@ -484,6 +505,13 @@ if __name__ == '__main__':
   endforeach(sig)
   unset(name)
 
+  #try to force generate the _impl.cc files
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${PYTHON_DASH_B}
+    ${CMAKE_CURRENT_BINARY_DIR}/generate_helper.py
+    ${root} ${root}_impl.cc.t ${ARGN}
+  )
+
   #create a command to generate the _impl.cc files
   add_custom_command(
     OUTPUT ${expanded_files_cc_impl}
@@ -493,6 +521,13 @@ if __name__ == '__main__':
     ${root} ${root}_impl.cc.t ${ARGN}
   )
 
+  #try to force generate the _impl.h files
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} ${PYTHON_DASH_B}
+    ${CMAKE_CURRENT_BINARY_DIR}/generate_helper.py
+    ${root} ${root}_impl.h.t ${ARGN}
+  )
+
   #create a command to generate the _impl.h files
   add_custom_command(
     OUTPUT ${expanded_files_h_impl}
