--- Makefile.in	Fri Mar  5 15:30:23 2004
+++ Makefile.in.new	Thu May 27 17:39:20 2004
@@ -52,11 +52,11 @@
 	  (cd $$dir && $(MAKE) install) \
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
 	done && test -z "$$fail"
-	$(top_srcdir)/mkinstalldirs $(incldir)/libssh
-	$(INSTALL) include/libssh/libssh.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/config.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/sftp.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/crypto.h $(incldir)/libssh/
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(incldir)/libssh
+	$(INSTALL) include/libssh/libssh.h $(DESTDIR)$(incldir)/libssh/
+	$(INSTALL) include/libssh/config.h $(DESTDIR)$(incldir)/libssh/
+	$(INSTALL) include/libssh/sftp.h $(DESTDIR)$(incldir)/libssh/
+	$(INSTALL) include/libssh/crypto.h $(DESTDIR)$(incldir)/libssh/
 clean:
 	/bin/rm -f *~ *.o ssh sftp
 	@for dir in ${subdirs}; do \
