--- Makefile.PL.orig	2015-01-16 09:20:30.000000000 -0800
+++ Makefile.PL	2015-01-16 09:21:11.000000000 -0800
@@ -4,11 +4,11 @@
 use ExtUtils::MakeMaker;
 use Config;
 
-unless ($Config{useperlio} eq 'define' and $Config{usesfio} eq 'false') {
-  die <<BARF;
-You need perl 5.8.0 or later, configured to use perlio (and not to use sfio)
-BARF
-}
+#unless ($Config{useperlio} eq 'define' and $Config{usesfio} eq 'false') {
+#  die <<BARF;
+#You need perl 5.8.0 or later, configured to use perlio (and not to use sfio)
+#BARF
+#}
 
 # OK. Is this worth it? For starters, it's somewhat potty as the "OS types" in
 # RFC 1952 are actaully semi-muddled with FS types.
