--- DESCRIPTION	2024-02-03 07:30:03.000000000 +0800
+++ DESCRIPTION	2024-02-04 10:54:17.000000000 +0800
@@ -25,8 +25,7 @@
         rlang (>= 1.1.0), cli, rstudioapi (>= 0.7), scales (>= 1.1),
         stringr (>= 1.4), tibble (>= 3.1), tidyr (>= 1.1), viridisLite
         (>= 0.4.2), visNetwork (>= 2.1.0)
-Suggests: curl, DiagrammeRsvg, knitr, rmarkdown, rsvg, testthat (>=
-        3.1.0), withr
+Suggests: curl, knitr, rmarkdown, rsvg, testthat (>= 3.1.0), withr
 Config/testthat/edition: 3
 Encoding: UTF-8
 LazyData: true
