--- a/cmake/install_macros.cmake	2019-12-10 15:50:53.000000000 -0500
+++ b/cmake/install_macros.cmake	2019-12-10 16:43:28.000000000 -0500
@@ -324,11 +324,11 @@
     ADD_CUSTOM_COMMAND(TARGET ${TARGET} POST_BUILD
       COMMAND install_name_tool -change
       "@rpath/$<TARGET_FILE_NAME:libprotobuf-lite>"
-      "@loader_path/../lib/$<TARGET_FILE_NAME:libprotobuf-lite>"
+      "@loader_path/../mysql/$<TARGET_FILE_NAME:libprotobuf-lite>"
       "$<TARGET_FILE:${TARGET}>"
       COMMAND install_name_tool -change
       "@rpath/$<TARGET_FILE_NAME:libprotobuf>"
-      "@loader_path/../lib/$<TARGET_FILE_NAME:libprotobuf>"
+      "@loader_path/../mysql/$<TARGET_FILE_NAME:libprotobuf>"
       "$<TARGET_FILE:${TARGET}>"
       )
   ELSEIF(UNIX)
