--- setup.py.orig	2021-02-10 08:43:57.000000000 -0600
+++ setup.py	2021-02-10 08:46:06.000000000 -0600
@@ -62,9 +62,9 @@
     extra_link_args += ["-lfftw3_threads"]
 
 # Only compile with OpenCV if exists (nice for the 'camera' G'MIC command :-D )
-if pkgconfig.exists("opencv"):
+if pkgconfig.exists("opencv4"):
     define_macros += [("cimg_use_opencv", None)]
-    pkgconfig_list += ["opencv"]
+    pkgconfig_list += ["opencv4"]
 
 packages = pkgconfig.parse(" ".join(pkgconfig_list))
 libraries = packages["libraries"] + [
@@ -73,10 +73,10 @@
 
 library_dirs = packages["library_dirs"] + [here, gmic_src_path]
 if sys.platform == "darwin":
-    library_dirs += ["/usr/local/opt/llvm@6/lib"]
+    library_dirs += ["%PREFIX%/libexec/llvm-9.0/lib"]
 include_dirs = packages["include_dirs"] + [here, gmic_src_path]
 if sys.platform == "darwin":
-    include_dirs += ["/usr/local/opt/llvm@6/include"]
+    include_dirs += ["%PREFIX%/libexec/llvm-9.0/include"]
 # Debugging is now set through --global-option --debug and more.
 # debugging_args = [
 #     "-O0",
@@ -94,12 +94,14 @@
     extra_link_args += ["-flto"]
 
 if sys.platform == "darwin":
-    extra_compile_args += ["-fopenmp", "-stdlib=libc++"]
+    extra_compile_args += ["-Xpreprocessor", "-fopenmp", "-stdlib=libc++"]
     extra_link_args += [
         "-lomp",
         "-nodefaultlibs",
         "-lc++",
     ]  # options inspired by https://github.com/explosion/spaCy/blob/master/setup.py
+    library_dirs += ["%PREFIX%/lib/libomp"]
+    include_dirs += ["%PREFIX%/include/libomp"]
 elif sys.platform == "linux":  # Enable openmp for 32bit & 64bit linuxes
     extra_compile_args += ["-fopenmp"]
     extra_link_args += ["-lgomp"]
--- setup.py.orig	2022-08-23 11:18:06.529888929 -0400
+++ setup.py	2022-08-23 11:18:31.909018857 -0400
@@ -9,8 +9,6 @@
 import pkgconfig
 
 here = path.abspath(path.dirname(__file__))
-gmic_src_path = path.abspath("src/gmic/src")
-
 
 # List of non-standard '-l*' compiler parameters
 extra_link_args = []
@@ -71,10 +69,10 @@
     "pthread"
 ]  # removed core-dumping 'gomp' temporarily (for manylinux builds)
 
-library_dirs = packages["library_dirs"] + [here, gmic_src_path]
+library_dirs = packages["library_dirs"] + [here]
 if sys.platform == "darwin":
     library_dirs += ["%PREFIX%/libexec/llvm-9.0/lib"]
-include_dirs = packages["include_dirs"] + [here, gmic_src_path]
+include_dirs = packages["include_dirs"] + [here]
 if sys.platform == "darwin":
     include_dirs += ["%PREFIX%/libexec/llvm-9.0/include"]
 # Debugging is now set through --global-option --debug and more.
@@ -118,7 +116,7 @@
     include_dirs=include_dirs,
     libraries=libraries,
     library_dirs=library_dirs,
-    sources=["gmicpy.cpp", path.join(gmic_src_path, "gmic.cpp")],
+    sources=["gmicpy.cpp"],
     define_macros=define_macros,
     extra_compile_args=extra_compile_args,
     extra_link_args=extra_link_args,
--- setup.py.orig	2022-10-01 11:22:30.000000000 -0400
+++ setup.py	2022-10-01 11:23:51.000000000 -0400
@@ -20,6 +20,7 @@
 # cimg_date and cimg_date are true variables, the value of which is checked in the C/C++ source
 define_macros = [
     ("gmic_build", None),
+    ("cimg_OS", '1'),
     ("cimg_date", '""'),
     ("cimg_time", '""'),
     ("gmic_is_parallel", None),
--- setup.py.orig	2022-10-01 14:25:41.000000000 -0400
+++ setup.py	2022-10-01 14:26:34.000000000 -0400
@@ -71,11 +71,11 @@
 ]  # removed core-dumping 'gomp' temporarily (for manylinux builds)
 
 library_dirs = packages["library_dirs"] + [here]
-if sys.platform == "darwin":
-    library_dirs += ["%PREFIX%/libexec/llvm-9.0/lib"]
+# if sys.platform == "darwin":
+#     library_dirs += ["%PREFIX%/libexec/llvm-9.0/lib"]
 include_dirs = packages["include_dirs"] + [here]
-if sys.platform == "darwin":
-    include_dirs += ["%PREFIX%/libexec/llvm-9.0/include"]
+# if sys.platform == "darwin":
+#     include_dirs += ["%PREFIX%/libexec/llvm-9.0/include"]
 # Debugging is now set through --global-option --debug and more.
 # debugging_args = [
 #     "-O0",
--- setup.py.orig	2022-10-01 14:53:30.000000000 -0400
+++ setup.py	2022-10-01 14:53:58.000000000 -0400
@@ -130,7 +130,7 @@
 
 # Grab the version from the VERSION file, which is also include in MANIFEST.in
 # Technique highlighted here: https://packaging.python.org/guides/single-sourcing-package-version/
-with open(path.join(".", "VERSION")) as version_file:
+with open(path.join(".", "VERSION.ver")) as version_file:
     version = version_file.read().strip()
 
 setup(
