--- build/darwin-config.py.orig	2011-05-31 06:58:29.000000000 -0500
+++ build/darwin-config.py	2012-12-10 05:59:21.000000000 -0600
@@ -27,17 +27,15 @@
 def initEnvironment(env):
 
 		
-	include_dirs = ['/opt/local/include',
-		        '/usr/local/include',
-			'/usr/X11/include',
-			'/opt/local/include/SDL',
-			'/opt/local/include/guichan',
-			'/opt/local/include/boost',
+	include_dirs = ['@PREFIX@/include',
+			'@PREFIX@/include/SDL',
+			'@PREFIX@/include/guichan',
+			'@PREFIX@/include/boost',
 			'/System/Library/Frameworks/OpenAL.framework/Headers',
-			'/opt/local/include/vorbis/',
+			'@PREFIX@/include/vorbis',
 			get_python_inc()]
 
-	lib_dirs = ['/System/Library', '/opt/local/lib', '/usr/X11/lib']
+	lib_dirs = ['/System/Library', '@PREFIX@/lib']
 
 	env.Append(CPPPATH = include_dirs)
 	env.Append(LIBPATH = lib_dirs)
@@ -45,7 +43,7 @@
 	return env
 	
 def addExtras(env, opengl):
-	env.Append(SHLINKFLAGS='-F/opt/local/Library/Frameworks')
+	env.Append(SHLINKFLAGS='-F@FRAMEWORKS_DIR@')
 	env.Append(SHLINKFLAGS='-framework OpenAL')
 	env.Append(SHLINKFLAGS='-framework Cocoa')
 	env.Append(SHLINKFLAGS='-framework CoreFoundation')
