--- libairspy/CMakeLists.txt.orig	2015-02-09 09:40:53.000000000 -0500
+++ libairspy/CMakeLists.txt	2015-02-09 09:40:57.000000000 -0500
@@ -67,7 +67,7 @@
 	set(THREADS_USE_PTHREADS_WIN32 true)
 else()
 	add_definitions(-Wall)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
+#	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
 
 	INCLUDE(TestBigEndian)
 	TEST_BIG_ENDIAN(BIGENDIAN)
--- airspy-tools/CMakeLists.txt.orig	2015-02-09 09:44:03.000000000 -0500
+++ airspy-tools/CMakeLists.txt	2015-02-09 09:44:23.000000000 -0500
@@ -38,7 +38,7 @@
 add_definitions(/D _CRT_SECURE_NO_WARNINGS)
 else()
 add_definitions(-Wall)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
+#set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
 endif()
 
 if(NOT libairspy_SOURCE_DIR)
