--- libcilkrts/runtime/sysdep-unix.c.orig	2018-04-15 15:19:32.000000000 -0700
+++ libcilkrts/runtime/sysdep-unix.c	2018-04-15 15:58:16.000000000 -0700
@@ -102,6 +102,46 @@
 #   include <vxCpuLib.h>  
 #endif
 
+#ifdef __APPLE__
+#ifndef __MAC_OS_X_VERSION_MIN_REQUIRED
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050
+#include <Availability.h>
+#else
+#include <AvailabilityMacros.h>
+#endif
+#endif //#ifndef __MAC_OS_X_VERSION_MIN_REQUIRED
+
+
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#define _SC_NPROCESSORS_ONLN 58
+
+long tigersysconf(int name){
+
+    if (name == _SC_NPROCESSORS_ONLN) {
+		int nm[2];
+		size_t len = 4;
+		uint32_t count;
+		
+		nm[0] = CTL_HW; nm[1] = HW_AVAILCPU;
+		sysctl(nm, 2, &count, &len, NULL, 0);
+		
+		if (count < 1) {
+			nm[1] = HW_NCPU;
+			sysctl(nm, 2, &count, &len, NULL, 0);
+			if (count < 1) { count = 1; }
+			}
+			
+		return (long)count;
+    }
+    return -1;
+}
+#endif //#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#endif //#ifdef __APPLE__
+
+
+
 struct global_sysdep_state
 {
     pthread_t *threads;    ///< Array of pthreads for system workers
@@ -695,7 +735,13 @@
 #ifdef __VXWORKS__      
     fprintf(fp, "System cores: %d\n", (int)__builtin_popcount(vxCpuEnabledGet()));
 #else    
+
+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+    fprintf(fp, "System cores: %d\n", (int)tigersysconf(_SC_NPROCESSORS_ONLN));
+#else
     fprintf(fp, "System cores: %d\n", (int)sysconf(_SC_NPROCESSORS_ONLN));
+#endif //__MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+
 #endif    
     fprintf(fp, "Cilk workers requested: %d\n", n);
 
