--- CMakeLists.txt.orig	2020-05-29 03:52:15.000000000 -0500
+++ CMakeLists.txt	2020-06-30 20:09:34.000000000 -0500
@@ -41,7 +41,6 @@
 find_package(lxqt-build-tools ${LXQTBT_MINIMUM_VERSION} REQUIRED)
 find_package(Qt5 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets DBus LinguistTools)
 find_package(Qt5Xdg ${QTXDG_MINIMUM_VERSION} REQUIRED)
-find_package(KF5WindowSystem ${KF5_MINIMUM_VERSION} REQUIRED)
 if (BUILD_BACKLIGHT_LINUX_BACKEND)
     find_package(PolkitQt5-1 REQUIRED)
 endif (BUILD_BACKLIGHT_LINUX_BACKEND)
@@ -321,7 +320,6 @@
 
 target_link_libraries(${LXQT_LIBRARY_NAME}
     PUBLIC
-        KF5::WindowSystem
         Qt5::Widgets
         Qt5::DBus
         Qt5Xdg
--- lxqtsingleapplication.cpp.orig
+++ lxqtsingleapplication.cpp
@@ -27,7 +27,6 @@
 
 #include "lxqtsingleapplication.h"
 #include "singleapplicationadaptor.h"
-#include <KWindowSystem/KWindowSystem>
 #include <QDBusMessage>
 #include <QWidget>
 #include <QDebug>
@@ -96,14 +95,6 @@ void SingleApplication::activateWindow()
 {
     if (mActivationWindow) {
         mActivationWindow->show();
-        WId window = mActivationWindow->effectiveWinId();
-
-        KWindowInfo info(window, KWindowSystem::WMDesktop);
-        int windowDesktop = info.desktop();
-
-        if (windowDesktop != KWindowSystem::currentDesktop())
-            KWindowSystem::setCurrentDesktop(windowDesktop);
-        KWindowSystem::activateWindow(window);
     } else {
         qDebug() << Q_FUNC_INFO << "activationWindow not set or null";
     }
--- cmake/lxqt-config.cmake.in.orig
+++ cmake/lxqt-config.cmake.in
@@ -12,7 +12,6 @@ find_dependency(Qt5Widgets @QT_MINIMUM_VERSION@)
 find_dependency(Qt5DBus @QT_MINIMUM_VERSION@)
 find_dependency(Qt5LinguistTools @QT_MINIMUM_VERSION@)
 find_dependency(Qt5Xdg @QTXDG_MINIMUM_VERSION@)
-find_dependency(KF5WindowSystem)
 find_dependency(lxqt-build-tools @LXQTBT_MINIMUM_VERSION@)
 if (NOT APPLE)
     find_dependency(Qt5X11Extras @QT_MINIMUM_VERSION@)
