--- bin/dumprecs.cpp.orig	2012-07-27 12:55:00.000000000 +0200
+++ bin/dumprecs.cpp	2012-07-27 13:32:01.000000000 +0200
@@ -84,9 +84,9 @@
     }
 
     std::cout << "Processing file sequentially from beginning..." << std::endl;
-    int i = 0;
-    while(i<MyFile.NoOfRecords()){
-      rc = MyFile.DumpRecord(++i);
+    int j = 0;
+    while(j<MyFile.NoOfRecords()){
+      rc = MyFile.DumpRecord(++j);
       if( rc != XB_NO_ERROR )
         x.DisplayError( rc );
     }
