--- src/ccd2iso.c.orig	2006-12-16 02:58:32
+++ src/ccd2iso.c	2025-09-22 18:07:30
@@ -27,6 +27,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include "clonecd.h"
 
--- configure.orig	2006-12-16 02:51:32
+++ configure	2025-09-22 18:11:53
@@ -3233,14 +3233,14 @@
 
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
 int
-main ()
+main (void)
 {
   int i;
   for (i = 0; i < 256; i++)
     if (XOR (islower (i), ISLOWER (i))
 	|| toupper (i) != TOUPPER (i))
-      exit(2);
-  exit (0);
+      return (2);
+  return (0);
 }
 _ACEOF
 rm -f conftest$ac_exeext
