--- cc65/make/gcc.mak.orig	2009-06-22 18:01:48.000000000 +1000
+++ cc65/make/gcc.mak	2009-06-22 18:02:39.000000000 +1000
@@ -14,7 +14,7 @@
 
 # The compiler include search path. Default is "/usr/lib/cc65/include/" if
 # nothing is defined. You may use CC65_INC=foo on the command line to override it.
-CC65_INC = \"/usr/lib/cc65/include/\"
+CC65_INC = \"${PREFIX}/lib/cc65/include/\"
 
 #
 CFLAGS = -O2 -g -Wall -W -std=c89 -I$(COMMON) -DCC65_INC=$(CC65_INC)
--- ld65/make/gcc.mak.orig	2005-12-11 22:40:51.000000000 +1000
+++ ld65/make/gcc.mak	2009-06-22 18:04:30.000000000 +1000
@@ -7,7 +7,7 @@
 
 # The linker library search path. Default is "/usr/lib/cc65/lib/" if nothing
 # is defined. You may use CC65_LIB=foo on the command line to override it.
-CC65_LIB = \"/usr/lib/cc65/lib/\"
+CC65_LIB = \"${PREFIX}/lib/cc65/lib/\"
 
 #
 CFLAGS = -g -O2 -Wall -W -std=c89 -I$(COMMON) -DCC65_LIB=$(CC65_LIB)
