--- Makefile.orig	2026-02-19 15:50:21
+++ Makefile	2026-02-19 16:32:11
@@ -8,12 +8,12 @@
 # SPDX-FileCopyrightText: Copyright (C) Eric S. Raymond <esr@thyrsus.com>
 # SPDX-License-Identifier: MIT
 
-PREFIX ?= /usr/local
-BINDIR = $(PREFIX)/bin
-INCDIR = $(PREFIX)/include
-LIBDIR = $(PREFIX)/lib
-MANDIR = $(PREFIX)/share/man
-DOCDIR = $(PREFIX)/share/doc/giflib
+PREFIX ?= /opt/local
+BINDIR = $(PREFIX)/bin/giflib6/bin
+INCDIR = $(PREFIX)/include/include/giflib6
+LIBDIR = $(PREFIX)/lib/giflib6/lib
+MANDIR = $(PREFIX)/share/man/giflib6/man
+DOCDIR = $(PREFIX)/share/doc/giflib6/doc
 
 CC ?= gcc
 OFLAGS = -g -fno-inline #-fsanitize=address
@@ -38,7 +38,7 @@
 HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
 OBJECTS = $(SOURCES:.c=.o)
 
-USOURCES = qprintf.c getarg.c
+USOURCES = qprintf.c getarg.c gif_err.c
 UHEADERS = getarg.h
 UOBJECTS = $(USOURCES:.c=.o)
 
@@ -53,6 +53,8 @@
 
 # Build
 
+UNAME:=$(shell uname)
+
 # Some utilities are installed
 INSTALLABLE = \
 	gifbuild \
