--- kdebase-3.5.10/kcontrol/info/info_osx.cpp~	2005-10-11 00:03:43.000000000 +0900
+++ kdebase-3.5.10/kcontrol/info/info_osx.cpp	2009-08-22 19:01:45.000000000 +0900
@@ -117,7 +117,7 @@
 #define kMaxStringSize 1024
 	OSStatus status;
 	AudioDeviceID gOutputDeviceID;
-	unsigned long propertySize;
+	UInt32 propertySize;
 	char deviceName[kMaxStringSize];
 	char manufacturer[kMaxStringSize];
 	propertySize = sizeof(gOutputDeviceID);
