--- Include/pyport.h.old	2012-04-10 01:25:33.000000000 +0200
+++ Include/pyport.h	2016-08-01 05:50:34.000000000 +0200
@@ -619,6 +619,12 @@
 #endif
 
 #ifdef _PY_PORT_CTYPE_UTF8_ISSUE
+#ifndef __cplusplus
+   /* The workaround below is unsafe in C++ because
+    * the <locale> defines these symbols as real functions,
+    * with a slightly different signature.
+    * See issue #10910
+    */
 #include <ctype.h>
 #include <wctype.h>
 #undef isalnum
@@ -636,6 +642,7 @@
 #undef toupper
 #define toupper(c) towupper(btowc(c))
 #endif
+#endif
 
 
 /* Declarations for symbol visibility.
