--- Makefile.orig	2010-04-15 19:04:26.000000000 -0500
+++ Makefile	2010-04-16 01:59:09.000000000 -0500
@@ -2,7 +2,7 @@
 PREFIX?=/usr/local
 INSTALLBIN?=$(PREFIX)/bin
 INSTALLLIB?=$(PREFIX)/lib
-INSTALLMAN?=$(PREFIX)/man
+INSTALLMAN?=$(PREFIX)/share/man
 INSTALLINCLUDE?=$(PREFIX)/include
 
 DPREFIX=$(DESTDIR)$(PREFIX)
@@ -26,8 +26,8 @@
  
 CFLAGS?=-pipe -O2 $(WARNFLAGS)
 
-DEFAULT_LIBS=-L/usr/X11R6/lib -L/usr/local/lib -lX11 -lXtst
-DEFAULT_INC=-I/usr/X11R6/include -I/usr/local/include
+DEFAULT_LIBS=-L$(PREFIX)/lib -lX11 -lXtst
+DEFAULT_INC=-I$(PREFIX)/include
 
 LIBS=$(shell pkg-config --libs x11 xtst 2> /dev/null || echo "$(DEFAULT_LIBS)")
 INC=$(shell pkg-config --cflags x11 xtst 2> /dev/null || echo "$(DEFAULT_INC)")
