http://code.google.com/p/semicomplete/issues/detail?id=28
--- Makefile.orig	2010-04-15 19:04:26.000000000 -0500
+++ Makefile	2010-04-16 02:25:22.000000000 -0500
@@ -22,7 +22,7 @@
 LIBSUFFIX=$(shell sh platform.sh libsuffix)
 VERLIBSUFFIX=$(shell sh platform.sh libsuffix $(MAJOR))
 DYNLIBFLAG=$(shell sh platform.sh dynlibflag)
-LIBNAMEFLAG=$(shell sh platform.sh libnameflag $(MAJOR))
+LIBNAMEFLAG=$(shell sh platform.sh libnameflag $(MAJOR) $(INSTALLLIB))
  
 CFLAGS?=-pipe -O2 $(WARNFLAGS)
 
