--- Makefile.orig	2011-05-30 02:29:58.000000000 -0500
+++ Makefile	2011-05-30 06:58:46.000000000 -0500
@@ -28,8 +28,8 @@
 CFLAGS?=-pipe -O2 $(WARNFLAGS)
 CFLAGS+=-g # TODO(sissel): Comment before release
 
-DEFAULT_LIBS=-L/usr/X11R6/lib -L/usr/local/lib -lX11 -lXtst -lXinerama
-DEFAULT_INC=-I/usr/X11R6/include -I/usr/local/include
+DEFAULT_LIBS=-L$(PREFIX)/lib -lX11 -lXtst -lXinerama
+DEFAULT_INC=-I$(PREFIX)/include
 
 LIBS+=$(shell pkg-config --libs x11 xtst xinerama 2> /dev/null || echo "$(DEFAULT_LIBS)")
 INC=$(shell pkg-config --cflags x11 xtst xinerama 2> /dev/null || echo "$(DEFAULT_INC)")
