--- src/corelib/global/qcompilerdetection.h.orig        2023-02-14 09:26:04.000000000 +0100
+++ src/corelib/global/qcompilerdetection.h     2023-06-01 17:09:55.000000000 +0200
@@ -125,7 +125,7 @@
 #      elif __apple_build_version__ >= 11000033 // Xcode 11.0
 #        define Q_CC_CLANG 800
 #      else
-#        error "Unsupported Apple Clang version"
+         static_assert(false, "Unsupported Apple Clang version" #__apple_build_version__);
 #      endif
 #    else
        // Non-Apple Clang, so we trust the versions reported
