--- designer-Qt4Qt5/designer.pro.orig
+++ designer-Qt4Qt5/designer.pro
@@ -4,7 +4,7 @@
 TEMPLATE = lib
 TARGET = qscintillaplugin
 
-CONFIG += release plugin qscintilla2
+CONFIG += plugin qscintilla2 @ARCHES@ @BUILD_TYPE@
 
 greaterThan(QT_MAJOR_VERSION, 4) {
     QT += designer
@@ -24,3 +24,15 @@
 
 target.path = $$[QT_INSTALL_PLUGINS]/designer
 INSTALLS += target
+
+QSCI_LIB = qscintilla2
+
+!debug_and_release|build_pass {
+	CONFIG(debug, debug|release):macx {
+		TARGET = $$member(TARGET, 0)_debug
+		QSCI_LIB = $$member(QSCI_LIB, 0)_debug
+	}
+}
+
+INCLUDEPATH += ../Qt4Qt5
+LIBS += -L../Qt4Qt5 -l$$QSCI_LIB
--- Qt4Qt5/qscintilla.pro.orig
+++ Qt4Qt5/qscintilla.pro
@@ -24,8 +24,14 @@
 
 TEMPLATE = lib
 TARGET = qscintilla2
-CONFIG += qt warn_off release thread exceptions
-INCLUDEPATH += . ../include ../lexlib ../src
+CONFIG += qt warn_off thread exceptions @ARCHES@ @BUILD_TYPE@
+INCLUDEPATH += . .. ../include ../lexlib ../src
+
+!debug_and_release|build_pass {
+    CONFIG(debug, debug|release) {
+        macx:TARGET = $$member(TARGET, 0)_debug
+    }
+}
 
 DEFINES += QSCINTILLA_MAKE_DLL SCINTILLA_QT SCI_LEXER
 greaterThan(QT_MAJOR_VERSION, 3) {
