--- configure.py.orig
+++ configure.py
@@ -963,24 +963,24 @@
                 self.qmake_spec = 'win32-msvc.net'
             else:
                 self.qmake_spec = 'win32-msvc'
-        else:
-            # Otherwise use the default.
-            self.qmake_spec = ''
+        #else:
+        #    # Otherwise use the default.
+        #    self.qmake_spec = ''
 
         # The binary MacOS/X Qt installer used to default to XCode.  If so then
         # use macx-clang (Qt v5) or macx-g++ (Qt v4).
-        if sys.platform == 'darwin':
-            try:
-                # Qt v5.
-                if qt_config.QMAKE_SPEC == 'macx-xcode':
-                    # This will exist (and we can't check anyway).
-                    self.qmake_spec = 'macx-clang'
-                else:
-                    # No need to explicitly name the default.
-                    self.qmake_spec = ''
-            except AttributeError:
-                # Qt v4.
-                self.qmake_spec = 'macx-g++'
+        #if sys.platform == 'darwin':
+        #    try:
+        #        # Qt v5.
+        #        if qt_config.QMAKE_SPEC == 'macx-xcode':
+        #            # This will exist (and we can't check anyway).
+        #            self.qmake_spec = 'macx-clang'
+        #        else:
+        #            # No need to explicitly name the default.
+        #            self.qmake_spec = ''
+        #    except AttributeError:
+        #        # Qt v4.
+        #        self.qmake_spec = 'macx-g++'
 
         self.api_dir = os.path.join(qt_config.QT_INSTALL_DATA, 'qsci')
         self.qt_inc_dir = qt_config.QT_INSTALL_HEADERS
