--- configure.ac.orig	2022-11-11 02:31:27.000000000 +0100
+++ configure.ac	2023-08-10 13:45:44.000000000 +0200
@@ -12,9 +12,9 @@
 AC_PREREQ([2.67])
 
 dnl Ensures that ./configure will be automatically re-run when VERSION changes.
-m4_include([VERSION])
+m4_include([VERSION.txt])
 
-AC_INIT([LibFixPOSIX], m4_esyscmd_s([cat VERSION]),
+AC_INIT([LibFixPOSIX], m4_esyscmd_s([cat VERSION.txt]),
         [iolib-devel@common-lisp.net],
         [libfixposix], [https://github.com/sionescu/libfixposix])
 AC_COPYRIGHT([Copyright © 2010-2020 Stelian Ionescu <sionescu@cddr.org>])
