--- setup.cfg.orig	2012-10-22 23:11:09.000000000 +1100
+++ setup.cfg	2012-10-23 22:02:00.000000000 +1100
@@ -5,7 +5,7 @@
 name = altgraph
 version = 0.10
 description = Python graph (network) package
-long_description_file = ReadMe.txt doc/changelog.rst
+long_description_file = README.txt doc/changelog.rst
 maintainer = Ronald Oussoren
 maintainer_email = ronaldoussoren@mac.com
 url = http://packages.python.org/altgraph
