--- gettext-tools/configure	2007-04-07 02:57:25.000000000 -0500
+++ gettext-tools/configure	2007-04-07 02:57:51.000000000 -0500
@@ -23076,6 +23076,19 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
+      gt_cv_int_divbyzero_sigfpe=
+      case "$host_os" in
+        macos* | darwin[6-9]* | darwin[1-9][0-9]*)
+          # On MacOS X 10.2 or newer, just assume the same as when cross-
+          # compiling. If we were to perform the real test, 1 Crash Report
+          # dialog window would pop up.
+          case "$host_cpu" in
+            i[34567]86 | x86_64)
+              gt_cv_int_divbyzero_sigfpe="guessing yes" ;;
+          esac
+          ;;
+      esac
+      if test -z "$gt_cv_int_divbyzero_sigfpe"; then
       if test "$cross_compiling" = yes; then
 
           # Guess based on the CPU.
@@ -23164,6 +23177,7 @@
 fi
 
 
+      fi
 
 fi
 { echo "$as_me:$LINENO: result: $gt_cv_int_divbyzero_sigfpe" >&5
