--- gettext-tools/src/write-java.c.orig	2016-03-20 18:37:53.000000000 +1100
+++ gettext-tools/src/write-java.c	2019-04-10 22:00:47.000000000 +1000
@@ -150,7 +150,7 @@ msgid_hashcode (const char *msgctxt, con
       char *combined;
       unsigned int result;
 
-      combined = (char *) xmalloca (combined_len);
+      combined = (char *) xmalloca (combined_len+1);
       memcpy (combined, msgctxt, msgctxt_len);
       combined[msgctxt_len] = MSGCTXT_SEPARATOR;
       memcpy (combined + msgctxt_len + 1, msgid, msgid_len + 1);
