--- configure.ac.orig	2012-04-27 08:34:22.000000000 -0700
+++ configure.ac	2013-10-02 09:16:37.000000000 -0700
@@ -352,7 +352,7 @@
 	AC_DEFINE(ENABLE_BINRELOC)
 	br_cv_binreloc=yes
     fi
-    PKG_CHECK_MODULES(MAC_INTEGRATION, ige-mac-integration, have_ige=yes, have_ige=no)
+    PKG_CHECK_MODULES(MAC_INTEGRATION, ige-mac-integration, have_ige=no, have_ige=no)
     if test x"$have_ige" = "xyes"
     then
 	AC_DEFINE(HAVE_MAC_INTEGRATION)
@@ -362,6 +362,7 @@
     LIBTOOL_EXPORT_OPTIONS=$EXPORT_SYM_REGEX
     LIBTOOL_PROV_EXPORT_OPTIONS=$EXPORT_PROV_SYM_REGEX
     LIBTOOL_UI_EXPORT_OPTIONS=$EXPORT_UI_SYM_REGEX
+    linklibext=.dylib
     AC_DEFINE(HAVE_CARBON)
     LIBGDA_LIBS="$LIBGDA_LIBS -framework Carbon"
   ;;
--- m4/bdb.m4.orig	2013-10-02 09:05:59.000000000 -0700
+++ m4/bdb.m4	2013-10-02 09:07:29.000000000 -0700
@@ -119,6 +119,9 @@
 	        if test $platform_win32 = yes
 		then
 		    sversion=`echo $version | sed -e 's,\.,,g'`
+		elif test $platform_carbon = yes
+		then
+		    sversion=`echo $version | sed -e 's,\.,,g'`
 		else
 		    sversion=`echo $version | sed -e 's,\..*,,g'`
 		fi
@@ -135,6 +138,12 @@
         	        db_lib="-ldb$sversion"
 		        db_libfile="$d/$bdb_loclibdir/libdb$sversion$bdb_libext"
         	        try_headers="db.h db$version/db.h db$sversion/db.h"
+		    elif test $platform_carbon = yes
+		    then
+		        db_libfilename="libdb-$version$bdb_libext"
+        	        db_lib="-ldb-$version"
+		        db_libfile="$d/$bdb_loclibdir/libdb-$version$bdb_libext"
+        	        try_headers="db.h db$version/db.h db$sversion/db.h"
 		    else
 		        db_libfilename="libdb-$version$bdb_libext"
         	        db_lib="-ldb-$version"
@@ -145,6 +154,7 @@
 
 		for db_hdr in $try_headers
 		do
+		    echo "Checking for files $d/include/$db_hdr and $db_libfile"
 		    if test -f $d/include/$db_hdr -a -f $db_libfile
 		    then
   	                save_CFLAGS="$CFLAGS"
@@ -221,6 +231,9 @@
         if test "x$platform_win32" = xyes
 	then
 	    bdbsql_loclibdir=bin
+        elif test "x$platform_carbon" = xyes
+	then
+	    bdbsql_loclibdir=$bdb_loclibdir
 	else
 	    bdbsql_loclibdir=lib
 	fi
@@ -231,14 +244,22 @@
     if test "x$BDB_DIR" != x
     then
         AC_MSG_CHECKING([for Berkeley DB SQL files along with found BDB installation])
-	if test -f $BDB_DIR/include/dbsql.h
+	echo "Checking $BDB_DIR/include/db$sversion/dbsql.h, $BDB_DIR/include/dbsql.h and $BDB_DIR/include/libdb/dbsql.h"
+	if test -f $BDB_DIR/include/db$sversion/dbsql.h -o $BDB_DIR/include/dbsql.h -o -f $BDB_DIR/include/libdb/dbsql.h
 	then
-	    BDBSQL_CFLAGS="$BDB_CFLAGS"
+	    if test -f $BDB_DIR/include/db$sversion/dbsql.h
+	    then
+	        BDBSQL_CFLAGS="$BDB_CFLAGS/db$sversion"
+	    elif test -f $BDB_DIR/include/libdb/dbsql.h
+	    then
+	        BDBSQL_CFLAGS="$BDB_CFLAGS/libdb"
+	    else
+	        BDBSQL_CFLAGS="$BDB_CFLAGS"
+	    fi
 	    BDBSQL_LIBS="-L$BDB_DIR/$bdb_loclibdir -ldb_sql"
 	    BDBSQL_PATH="$BDB_DIR/$bdb_loclibdir"
 	    AC_MSG_RESULT([found])
 	    AC_CHECK_LIB(db_sql, sqlite3_table_column_metadata,[bdbsql_api=1], [bdbsql_api=0], $BDBSQL_CFLAGS $BDBSQL_LIBS -pthread -ldl)
-
 	    if test $bdbsql_api = 0
 	    then
 		BDBSQL_CFLAGS=""
