--- configure.in.orig	2009-04-18 05:03:50.000000000 -0700
+++ configure.in	2009-06-13 15:19:24.000000000 -0700
@@ -176,6 +176,7 @@
     AC_MSG_RESULT([Mac OS X - carbon])
     platform_carbon=yes
     SOPREFIX='lib'
+    linklibext=".dylib"
     AM_BINRELOC
     LIBTOOL_EXPORT_OPTIONS=$EXPORT_SYM_REGEX
     LIBTOOL_PROV_EXPORT_OPTIONS=$EXPORT_PROV_SYM_REGEX
@@ -355,6 +356,25 @@
 				BDB_LIBS="-L${bdbdir}/lib -ldb"
 				break
 			fi
+                        if test $platform_carbon = yes  -a -f $d/include/db47/db.h -a -f $d/lib/db47/libdb$linklibext
+                        then    
+                                AC_MSG_RESULT(found Berkeley DB 4.7 in $d)
+                                AC_DEFINE(HAVE_BDB, 1, [Have Berkeley DB])
+                                bdbdir=$d
+                                BDB_CFLAGS="-I${bdbdir}/include/db47"
+                                BDB_LIB="$d/lib/db47/libdb$linklibext"
+                                BDB_LIBS="-L${bdbdir}/lib/db47 -ldb"
+                                break
+			elif test $platform_carbon = yes  -a -f $d/include/db46/db.h -a -f $d/lib/db46/libdb$linklibext
+			then
+				AC_MSG_RESULT(found Berkeley DB 4.6 in $d)
+				AC_DEFINE(HAVE_BDB, 1, [Have Berkeley DB])
+				bdbdir=$d
+				BDB_CFLAGS="-I${bdbdir}/include/db46"
+				BDB_LIB="$d/lib/db46/libdb$linklibext"
+				BDB_LIBS="-L${bdbdir}/lib/db46 -ldb"
+				break
+			fi
 			dnl FIXME: check for db1 (non-sleepycat implementation)
 		done
 		if test x$bdbdir = x
