--- pdftoipe-20110916-src/xmloutputdev.cpp.orig	2012-07-28 14:38:53.000000000 +0200
+++ pdftoipe-20110916-src/xmloutputdev.cpp	2012-07-28 14:39:31.000000000 +0200
@@ -30,7 +30,7 @@
   FILE *f;
 
   if (!(f = fopen(fileName, "wb"))) {
-    error(-1, "Couldn't open output file '%s'", fileName);
+    fprintf(stderr, "Couldn't open output file '%s'\n", fileName);
     ok = gFalse;
     return;
   }
@@ -336,7 +336,7 @@
     // dump JPEG stream
     std::vector<char> buffer;
     // initialize stream
-    str = ((DCTStream *)str)->getRawStream();
+    str = str->getNextStream();
     str->reset();
     // copy the stream
     while ((c = str->getChar()) != EOF)
