error: #pragma GCC diagnostic not allowed inside functions
--- src/funcs.c.orig	2017-10-06 09:09:54.000000000 -0500
+++ src/funcs.c	2017-11-20 20:33:56.000000000 -0600
@@ -1118,14 +1118,11 @@
 /* The `start' and `end' arguments are fake, hence their string type,
    so they can be ignored, hence the pragma (FIXME: work out how to
    apply just to `start' and `end'). */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 DEFUN_ARGS ("shell-command-on-region", shell_command_on_region,
             STR_ARG (start)
             STR_ARG (end)
             STR_ARG (cmd)
             BOOL_ARG (insert))
-#pragma GCC diagnostic pop
 /*+
 Execute string command in inferior shell with region as input.
 Normally display output (if any) in temp buffer `*Shell Command Output*';
--- src/minibuf.c.orig	2017-10-06 09:09:54.000000000 -0500
+++ src/minibuf.c	2017-11-20 20:36:56.000000000 -0600
@@ -287,10 +287,7 @@
             }
           else
             {
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
               minibuf_error (invalid_err, astr_cstr (ms));
-#pragma GCC diagnostic pop
               waitkey ();
             }
         }
