--- Makefile.dynamic.in.orig	2008-11-10 13:43:37.000000000 +0100
+++ Makefile.dynamic.in	2008-11-10 14:06:41.000000000 +0100
@@ -127,8 +127,7 @@
 # You may need to be root to do the install
 #explicit use of sh to make sure we don't invoke csh with different "if" syntax
 	sh -c "if test -d $(srcdir)/htdocs >/dev/null ; then $(MAKE) install_docs ; fi"
-	sh -c "if test -d fontforge/FontForge.app >/dev/null ; then rm -rf /Applications/FontForge.app ; mv fontforge/FontForge.app /Applications ; ln -s /usr/local/bin/fontforge /Applications/FontForge.app/Contents/MacOS/FontForge ; fi"
-	sh < systemspecific
+	sh -c "if test -d fontforge/FontForge.app >/dev/null ; then mkdir -p $(DESTDIR)/Applications/MacPorts ; mv fontforge/FontForge.app $(DESTDIR)/Applications/MacPorts ; ln -sf $(bindir)/fontforge $(DESTDIR)/Applications/MacPorts/FontForge.app/Contents/MacOS/FontForge ; fi"
 #we don't always create a setup.py
 	-(cd pyhook ; if test -r setup.py ; then python setup.py install --prefix=$(prefix) --root=$(DESTDIR); for dir in * ; do if test -d $$dir ; then  rm -rf $$dir ; fi; done ; fi )
 
