--- configure.orig	2008-11-10 09:16:23.000000000 +0100
+++ configure	2008-11-10 09:21:43.000000000 +0100
@@ -2684,8 +2684,6 @@
 
 fi;
 
-LDFLAGS="$LDFLAGS -L/usr/local/lib"
-
 case "$host" in
 *-apple-darwin*)
  gww_ismac="yes"
@@ -2705,13 +2703,9 @@
  gww_define_caps_for_alt="1"
 
 
- if test -d /sw/include >/dev/null ; then
-  CPPFLAGS="$CPPFLAGS -I/sw/include"
- fi
- if test -d /sw/lib >/dev/null ; then
-  LDFLAGS="$LDFLAGS -L/sw/lib"
-  gww_rpath="-rpath /sw/lib"
- fi
+ CPPFLAGS="$CPPFLAGS -I@PREFIX@/include"
+ LDFLAGS="$LDFLAGS -L@PREFIX@/lib"
+ gww_rpath="-rpath @PREFIX@/lib"
  LIBS="-Wl,/System/Library/Frameworks/Carbon.framework/Carbon $LIBS"
  LIBS="-Wl,/System/Library/Frameworks/CoreServices.framework/CoreServices $LIBS"
 
@@ -19738,55 +19732,8 @@
 
 fi
 
-echo "$as_me:$LINENO: checking for /usr/include/libxml2/libxml" >&5
-echo $ECHO_N "checking for /usr/include/libxml2/libxml... $ECHO_C" >&6
-if test "${ac_cv_file__usr_include_libxml2_libxml+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r "/usr/include/libxml2/libxml"; then
-  ac_cv_file__usr_include_libxml2_libxml=yes
-else
-  ac_cv_file__usr_include_libxml2_libxml=no
-fi
-fi
-echo "$as_me:$LINENO: result: $ac_cv_file__usr_include_libxml2_libxml" >&5
-echo "${ECHO_T}$ac_cv_file__usr_include_libxml2_libxml" >&6
-if test $ac_cv_file__usr_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/include/libxml2/"
-else
-
- echo "$as_me:$LINENO: checking for /usr/local/include/libxml2/libxml" >&5
-echo $ECHO_N "checking for /usr/local/include/libxml2/libxml... $ECHO_C" >&6
-if test "${ac_cv_file__usr_local_include_libxml2_libxml+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r "/usr/local/include/libxml2/libxml"; then
-  ac_cv_file__usr_local_include_libxml2_libxml=yes
-else
-  ac_cv_file__usr_local_include_libxml2_libxml=no
-fi
-fi
-echo "$as_me:$LINENO: result: $ac_cv_file__usr_local_include_libxml2_libxml" >&5
-echo "${ECHO_T}$ac_cv_file__usr_local_include_libxml2_libxml" >&6
-if test $ac_cv_file__usr_local_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"
-else
-  cat >>confdefs.h <<\_ACEOF
-#define _NO_LIBXML 1
-_ACEOF
-
-fi
-
-fi
 
+CFLAGS="$CFLAGS -I@PREFIX@/include -I@PREFIX@/include/libxml2"
 
 
 echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
