--- fontforge/macbinary.c.orig	2012-03-14 15:43:22.000000000 -0400
+++ fontforge/macbinary.c	2012-03-14 15:51:15.000000000 -0400
@@ -37,7 +37,7 @@
 #include "psfont.h"
 #if __Mac
 # include <ctype.h>
-# include </Developer/Headers/FlatCarbon/Files.h>
+# include <@@DEVDIR@@/Headers/FlatCarbon/Files.h>
 #else
 # include <utype.h>
 #undef __Mac
--- fontforge/startui.c.orig	2012-03-14 15:49:32.000000000 -0400
+++ fontforge/startui.c	2012-03-14 15:51:51.000000000 -0400
@@ -47,10 +47,10 @@
 #  endif
 #endif
 #ifdef __Mac
-# include </Developer/Headers/FlatCarbon/Files.h>
+# include <@@DEVDIR@@/Headers/FlatCarbon/Files.h>
 # define FontInfo	MacFontInfo
 # define KernPair	MacKernPair
-# include </Developer/Headers/FlatCarbon/CarbonEvents.h>
+# include <@@DEVDIR@@/Headers/FlatCarbon/CarbonEvents.h>
 /* For reasons obscure to me RunApplicationEventLoop is not defined in */
 /*  the mac header files if we are in 64 bit mode. Strangely it seems to */
 /*  be in the libraries and functional */
--- gutils/giomime.c.orig	2012-03-14 15:49:53.000000000 -0400
+++ gutils/giomime.c	2012-03-14 15:52:03.000000000 -0400
@@ -65,7 +65,7 @@
 unichar_t fontsnf[] = { 'a','p','p','l','i','c','a','t','i','o','n','/','x','-','f','o','n','t','-','s','n','f', '\0' };
 
 #ifdef __Mac
-#include </Developer/Headers/FlatCarbon/Files.h>
+#include <@@DEVDIR@@/Headers/FlatCarbon/Files.h>
 #define CHR(ch1,ch2,ch3,ch4) (((ch1)<<24)|((ch2)<<16)|((ch3)<<8)|(ch4))
 
 unichar_t *_GioMacMime(const char *path) {
