--- mkfls/unix/configure.orig	2014-08-22 11:09:23.000000000 -0400
+++ mkfls/unix/configure	2026-07-14 14:26:09.000000000 -0400
@@ -4594,6 +4594,8 @@
 
 #include <sys/types.h>
 #include <signal.h>
+#include <unistd.h>
+#include <stdlib.h>
 
 #ifndef SIGCHLD
 # define SIGCHLD SIGCLD
@@ -4604,15 +4606,12 @@
 
 int got;
 
-#ifdef SIGVOID
-void
-#endif
-hand()
+void hand()
 {
   got++;
 }
 
-main()
+int main()
 {
   (void)signal(SIGCHLD, hand);
   kill(getpid(), SIGCHLD);
