diff -rubw examples/gldemo.cpp examples/gldemo.cpp
--- examples/gldemo.cpp	2015-01-21 13:58:00.000000000 -0600
+++ examples/gldemo.cpp	2006-10-03 10:55:34.000000000 -0500
@@ -21,7 +21,7 @@
  * use of this library.
  */
 
-#include <GL/glut.h>
+#include <GLUT/glut.h>
 #include <SOLID.h>
 
 #define DISPLAY           // Interactively draw new placements 
Only in examples/: gldemo.o
Only in examples/: mnm
diff -rubw examples/mnm.cpp examples/mnm.cpp
--- examples/mnm.cpp	2015-01-21 13:58:00.000000000 -0600
+++ examples/mnm.cpp	2006-10-03 10:55:34.000000000 -0500
@@ -29,7 +29,7 @@
 //#define DRAW_COORD
 
 #include <new>
-#include <GL/glut.h>
+#include <GLUT/glut.h>
 
 #include "GEN_MinMax.h"
 #include "MT_Point3.h"
Only in examples/: mnm.o
Only in examples/: physics
diff -rubw examples/physics.cpp examples/physics.cpp
--- examples/physics.cpp	2015-01-21 13:58:00.000000000 -0600
+++ examples/physics.cpp	2006-10-03 10:55:34.000000000 -0500
@@ -28,7 +28,7 @@
 //#define USE_QUADS           // Idem, but now with quads  
 
 #include <new>
-#include <GL/glut.h>
+#include <GLUT/glut.h>
 
 #include "GEN_MinMax.h"
 #include "MT_Point3.h"
