--- examples/Makefile.in	2006-10-03 11:12:39.000000000 +0200
+++ examples/Makefile.in	2012-07-22 22:43:55.000000000 +0200
@@ -192,7 +192,7 @@
 gldemo_SOURCES = gldemo.cpp
 physics_SOURCES = physics.cpp
 mnm_SOURCES = mnm.cpp
-GLLIBS = -lglut -lGLU -lGL -L/usr/X11R6/lib -lXmu -lXi -lX11
+GLLIBS = -framework OpenGL -framework GLUT
 sample_LDADD = ../src/libsolid.la  
 gldemo_LDADD = ../src/libsolid.la $(GLLIBS)
 physics_LDADD = dynamics/libdynamics.la ../src/libsolid.la $(GLLIBS)
