--- DESCRIPTION	2024-05-01 12:19:26.000000000 +0800
+++ DESCRIPTION	2024-05-04 14:13:59.000000000 +0800
@@ -23,8 +23,7 @@
 Imports: stats4
 LinkingTo: S4Vectors
 Suggests: XVector, GenomicRanges, Rsamtools, GenomicAlignments,
-        GenomicFeatures, BSgenome.Celegans.UCSC.ce2, pasillaBamSubset,
-        RUnit, BiocStyle
+        GenomicFeatures, RUnit, BiocStyle
 Collate: range-squeezers.R Vector-class-leftovers.R
         DataFrameList-class.R DataFrameList-utils.R AtomicList-class.R
         AtomicList-utils.R Ranges-and-RangesList-classes.R
