--- DESCRIPTION	2023-10-24 21:32:51.000000000 +0800
+++ DESCRIPTION	2023-10-26 18:40:26.000000000 +0800
@@ -23,7 +23,7 @@
 Imports: stats4
 LinkingTo: S4Vectors
 Suggests: XVector, GenomicRanges, Rsamtools, GenomicAlignments, GenomicFeatures,
-	BSgenome.Celegans.UCSC.ce2, pasillaBamSubset, RUnit, BiocStyle
+	RUnit, BiocStyle
 Collate: range-squeezers.R
 	Vector-class-leftovers.R
 	DataFrameList-class.R
