--- utils/pypi/setup-wheel.py.in.orig	2022-01-04 08:01:40.000000000 -0600
+++ utils/pypi/setup-wheel.py.in	2022-01-22 13:25:42.000000000 -0600
@@ -7,8 +7,8 @@
 import os
 import setuptools
 
-setupdir = os.environ["GMSH_SETUP_DIR"]
-sdkdir = os.environ["GMSH_SDK_DIR"]
+setupdir = "WORKSRCPATH/utils/pypi/gmsh/"
+sdkdir = "DESTROOT/opt/local/"
 
 version = '${GMSH_MAJOR_VERSION}.${GMSH_MINOR_VERSION}.${GMSH_PATCH_VERSION}'
 
