see https://gitlab.onelab.info/gmsh/gmsh/issues/716 
--- Mesh/gmshCrossFields.cpp.orig	2019-12-20 19:59:26.000000000 -0700
+++ Mesh/gmshCrossFields.cpp	2020-01-04 07:08:02.000000000 -0700
@@ -20,10 +20,6 @@
 #include "PView.h"
 #include "PViewDataGModel.h"
 
-#ifdef HAVE_FLTK
-#include "FlGui.h"
-#endif
-
 #if defined(_OPENMP)
 #include <omp.h>
 #endif
@@ -37,6 +33,10 @@
 #include "linearSystemFull.h"
 #include "linearSystemPETSc.h"
 
+#ifdef HAVE_FLTK
+#include "FlGui.h"
+#endif
+
 static inline double lifting(double a, double _a)
 {
   double D = M_PI * .5;
