diff --git lib_pypy/_tkinter/tklib_build.py lib_pypy/_tkinter/tklib_build.py
--- lib_pypy/_tkinter/tklib_build.py
+++ lib_pypy/_tkinter/tklib_build.py
@@ -18,7 +18,7 @@ elif sys.platform == 'win32':
     linklibs = ['tcl85', 'tk85']
     libdirs = []
 elif sys.platform == 'darwin':
-    incdirs = ['/System/Library/Frameworks/Tk.framework/Versions/Current/Headers/']
+    incdirs = ['__PREFIX__/include']
     linklibs = ['tcl', 'tk']
     libdirs = []
 else:
diff -ur ./extra_tests/ctypes_tests/conftest.py ../pypy2.7-v7.0.0-src.new/extra_tests/ctypes_tests/conftest.py
--- ./extra_tests/ctypes_tests/conftest.py	2019-02-05 14:42:41.000000000 +0100
+++ ../pypy2.7-v7.0.0-src.new/extra_tests/ctypes_tests/conftest.py	2019-02-12 14:57:18.000000000 +0100
@@ -58,7 +58,7 @@
         link_extra = link_extra + ['/DEBUG']  # generate .pdb file
     if sys.platform == 'darwin':
         # support Fink & Darwinports
-        for s in ('/sw/', '/opt/local/'):
+        for s in ('__PREFIX__',):
             if (s + 'include' not in include_dirs
                     and os.path.exists(s + 'include')):
                 include_dirs.append(s + 'include')
diff -ur ./pypy/tool/cpyext/extbuild.py ../pypy2.7-v7.0.0-src.new/pypy/tool/cpyext/extbuild.py
--- ./pypy/tool/cpyext/extbuild.py	2019-02-05 14:42:41.000000000 +0100
+++ ../pypy2.7-v7.0.0-src.new/pypy/tool/cpyext/extbuild.py	2019-02-12 14:57:18.000000000 +0100
@@ -175,7 +175,7 @@
         link_extra = link_extra + ['/DEBUG']  # generate .pdb file
     if sys.platform == 'darwin':
         # support Fink & Darwinports
-        for s in ('/sw/', '/opt/local/'):
+        for s in ('__PREFIX__',):
             if (s + 'include' not in include_dirs
                     and os.path.exists(s + 'include')):
                 include_dirs.append(s + 'include')
