--- utils/Math/PariBuild.pm.orig	2010-03-06 17:26:46.000000000 -0600
+++ utils/Math/PariBuild.pm	2012-04-21 12:13:31.000000000 -0500
@@ -486,6 +486,7 @@
 			      'patches/diff_2.1.7_-O',
 			      'patches/diff_2.1.7_div',
 			      'patches/diff_2.1.6_align_power_of_2',
+			      'patches/patch-pari-2-1-7_src_kernel_ix86_level0.h.diff',
 			      'patches/diff_2.1.7_restart'],
 		 '2.3.5' =>  [
 			($^O eq 'MSWin32' ? 'patches/diff_2.3.5_mingw-w64' : ())], 
@@ -1319,7 +1320,7 @@
 sub extra_includes {
   my $pari_dir = shift;
   # Some #include directives assume us inside $pari_dir/OARCH; replace by src
-  return join ' -I ', '', grep -d, "$pari_dir/src/systems/$^O", "$pari_dir/src";
+  return join ' -I', '', grep -d, "$pari_dir/src/systems/$^O", "$pari_dir/src";
 }
 
 sub build_funclists_ourselves ($) {
