--- alex.cabal.sav	2008-11-21 14:05:49.000000000 -0500
+++ alex.cabal	2008-11-21 14:06:15.000000000 -0500
@@ -59,9 +59,9 @@
   main-is: Main.hs
 
   if flag(small_base)
-    build-depends: base >= 2.1, array, containers, directory
+    build-depends: base >= 2.1, base < 4, array, containers, directory
   else
-    build-depends: base >= 1.0
+    build-depends: base >= 1.0, base < 4
 
   build-depends: haskell98 >= 1.0
 
