--- configure.orig	2011-06-23 09:13:45.000000000 +0200
+++ configure	2011-08-02 12:18:03.000000000 +0200
@@ -17231,7 +17231,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_date_time" = xyes; then
-        boost_cv_lib_date_time_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_date_time_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_date_time_LDPATH="$boost_ldpath"
         break 6
       else
@@ -17449,7 +17449,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_program_options" = xyes; then
-        boost_cv_lib_program_options_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_program_options_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_program_options_LDPATH="$boost_ldpath"
         break 6
       else
@@ -17667,7 +17667,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_iostreams" = xyes; then
-        boost_cv_lib_iostreams_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_iostreams_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_iostreams_LDPATH="$boost_ldpath"
         break 6
       else
@@ -17889,7 +17889,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_system" = xyes; then
-        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_system_LDPATH="$boost_ldpath"
         break 6
       else
@@ -18112,7 +18112,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_filesystem" = xyes; then
-        boost_cv_lib_filesystem_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_filesystem_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_filesystem_LDPATH="$boost_ldpath"
         break 6
       else
@@ -18336,7 +18336,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_system" = xyes; then
-        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -Wl,-rpath,$boost_ldpath"
         boost_cv_lib_system_LDPATH="$boost_ldpath"
         break 6
       else
