--- Makefile.orig
+++ Makefile
@@ -636,7 +636,7 @@
 ifeq ($(target),Darwin)
 
 # default BINDIR for macOS
-BINDIR= /usr/local/bin
+BINDIR= ___PREFIX___/bin
 
 else
 #endif	/* end of skip for non-Gnu makefiles */
@@ -652,7 +652,7 @@
 ifeq ($(target),Darwin)
 
 # default LIBDIR for macOS
-LIBDIR= /usr/local/lib
+LIBDIR= ___PREFIX___/lib
 
 else
 #endif	/* end of skip for non-Gnu makefiles */
@@ -668,7 +668,7 @@
 ifeq ($(target),Darwin)
 
 # default CALC_SHAREDIR for macOS
-CALC_SHAREDIR= /usr/local/share/calc
+CALC_SHAREDIR= ___PREFIX___/share/calc
 
 else
 #endif	/* end of skip for non-Gnu makefiles */
@@ -688,7 +688,7 @@
 #
 #CALC_INCDIR= /usr/local/include/calc
 #CALC_INCDIR= /dev/env/DJDIR/include/calc
-CALC_INCDIR= ${INCDIR}/calc
+CALC_INCDIR= ___PREFIX___/include/calc
 
 # By default, these values are based CALC_SHAREDIR, INCDIR, BINDIR
 # ---------------------------------------------------------------
@@ -758,7 +758,7 @@
 #MANDIR=
 #MANDIR= /usr/local/man/man1
 #MANDIR= /usr/man/man1
-MANDIR= /usr/share/man/man1
+MANDIR= ___PREFIX___/share/man/man1
 #MANDIR= /dev/env/DJDIR/man/man1
 #MANDIR= /usr/man/u_man/man1
 #MANDIR= /usr/contrib/man/man1
@@ -893,11 +893,11 @@
 #
 # If in doubt, set USE_READLINE, READLINE_LIB and READLINE_INCLUDE to nothing.
 #
-USE_READLINE=
-#USE_READLINE= -DUSE_READLINE
+#USE_READLINE=
+USE_READLINE= -DUSE_READLINE
 #
-READLINE_LIB=
-READLINE_EXTRAS=
+#READLINE_LIB=
+#READLINE_EXTRAS=
 #
 #READLINE_LIB= -lreadline
 #READLINE_EXTRAS= -lhistory -lncurses
@@ -922,6 +922,8 @@
 #
 #READLINE_LIB= -L/usr/local/opt/readline/lib -lreadline
 #READLINE_EXTRAS= -lhistory -lncurses
+READLINE_LIB= -L___PREFIX___/lib -lreadline
+READLINE_EXTRAS= -lhistory -lncurses
 #
 READLINE_INCLUDE=
 #READLINE_INCLUDE= -I/usr/gnu/include
@@ -1274,7 +1274,7 @@
 BLD_TYPE= calc-dynamic-only
 #
 CC_SHARE= -fPIC
-DEFAULT_LIB_INSTALL_PATH= ${PWD}:${LIBDIR}:/usr/local/lib
+DEFAULT_LIB_INSTALL_PATH= ${PWD}:${LIBDIR}:___PREFIX___/lib
 LD_SHARE= ${DARWIN_ARCH}
 #SET_INSTALL_NAME= no
 SET_INSTALL_NAME= yes
