--- configure~	2006-08-05 07:49:16.000000000 -0400
+++ configure	2006-08-05 07:50:34.000000000 -0400
@@ -38134,7 +38134,7 @@
 
 
 
-if test $have_posix_acl_support = yes; then
+if test x"$have_posix_acl_support" = x"yes"; then
   HAVE_POSIX_ACL_TRUE=
   HAVE_POSIX_ACL_FALSE='#'
 else
@@ -38144,7 +38144,7 @@
 
 
 
-if test $have_solaris_acl_support = yes; then
+if test x"$have_solaris_acl_support" = x"yes"; then
   HAVE_SOLARIS_ACL_TRUE=
   HAVE_SOLARIS_ACL_FALSE='#'
 else
