--- configure.orig	2013-07-27 16:01:41.000000000 +0900
+++ configure	2013-07-27 16:03:56.000000000 +0900
@@ -5097,7 +5097,7 @@
     am__obj=sub/conftest.${OBJEXT-o}
     am__minus_obj="-o $am__obj"
     case $depmode in
-    gcc)
+    *gcc* | *clang*)
       # This depmode causes a compiler race in universal mode.
       test "$am__universal" = false || continue
       ;;
@@ -9925,7 +9925,7 @@
     # check size of mp_limb_t
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking size of GNU-MP mp_limb_t" >&5
 $as_echo_n "checking size of GNU-MP mp_limb_t... " >&6; }
-    lSz=`qcCC utils/csMPlimb.c ./csMPlimb`
+    lSz=`qcCC buildutils/csMPlimb.c ./csMPlimb`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: ${lSz}" >&5
 $as_echo "${lSz}" >&6; }
     omniLimbT=""
@@ -9950,7 +9950,7 @@
     # check how many mp_limb_t the GNU-MP needed for N bit precision.
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking how many mp_limb_t needed for $omniQPrec bit precision" >&5
 $as_echo_n "checking how many mp_limb_t needed for $omniQPrec bit precision... " >&6; }
-    omniQRealPrecInLimb=`qcCC utils/cMPlb.c ./cMPlb '' $omniQPrec`
+    omniQRealPrecInLimb=`qcCC buildutils/cMPlb.c ./cMPlb '' $omniQPrec`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $omniQRealPrecInLimb" >&5
 $as_echo "$omniQRealPrecInLimb" >&6; }
     cat >>confdefs.h <<_ACEOF
