--- configure.orig	2012-04-17 22:22:37.000000000 +0900
+++ configure	2012-04-17 22:24:04.000000000 +0900
@@ -5051,7 +5051,7 @@
 
 chk=`$BECC --version | head -n1`
 case $chk in
-gcc*)
+*gcc* | *clang*)
     OMC_CONF_LX2X_OPT=""
     OMC_CONF_NTV_OPT="-std=gnu99 -Wno-implicit"
     OMC_CONF_LNK_OPT=""
@@ -9229,7 +9229,7 @@
     # check size of mp_limb_t
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking size of GNU-MP mp_limb_t" >&5
 $as_echo_n "checking size of GNU-MP mp_limb_t... " >&6; }
-    lSz=`qcCC utils/csMPlimb.c ./csMPlimb`
+    lSz=`qcCC buildutils/csMPlimb.c ./csMPlimb`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: ${lSz}" >&5
 $as_echo "${lSz}" >&6; }
     omniLimbT=""
@@ -9254,7 +9254,7 @@
     # check how many mp_limb_t the GNU-MP needed for N bit precision.
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking how many mp_limb_t needed for $omniQPrec bit precision" >&5
 $as_echo_n "checking how many mp_limb_t needed for $omniQPrec bit precision... " >&6; }
-    omniQRealPrecInLimb=`qcCC utils/cMPlb.c ./cMPlb '' $omniQPrec`
+    omniQRealPrecInLimb=`qcCC buildutils/cMPlb.c ./cMPlb '' $omniQPrec`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $omniQRealPrecInLimb" >&5
 $as_echo "$omniQRealPrecInLimb" >&6; }
     cat >>confdefs.h <<_ACEOF
