--- mkspecs/common/clang.conf.orig	2016-12-01 01:17:04.000000000 -0700
+++ mkspecs/common/clang.conf	2016-12-26 05:05:56.000000000 -0700
@@ -4,8 +4,8 @@
 
 QMAKE_COMPILER          = gcc clang llvm   # clang pretends to be gcc
 
-QMAKE_CC                = clang
-QMAKE_CXX               = clang++
+QMAKE_CC                = __MACPORTS_CC__
+QMAKE_CXX               = __MACPORTS_CXX__
 
 QMAKE_LINK_C            = $$QMAKE_CC
 QMAKE_LINK_C_SHLIB      = $$QMAKE_CC
--- mkspecs/common/macx.conf.orig	2016-12-01 01:17:04.000000000 -0700
+++ mkspecs/common/macx.conf	2017-02-05 05:10:55.000000000 -0700
@@ -3,6 +3,6 @@
 #
 
 QMAKE_PLATFORM         += osx macx
-QMAKE_MAC_SDK           = macosx
+QMAKE_MAC_SDK           = __MACPORTS_MAC_SDK__
 
 include(mac.conf)
--- mkspecs/macx-clang/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-clang/qmake.conf	2016-11-12 07:33:45.000000000 -0700
@@ -11,6 +11,10 @@
 include(../common/clang.conf)
 include(../common/clang-mac.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
+
+QMAKE_CFLAGS           += -arch x86_64
+QMAKE_CXXFLAGS         += -arch x86_64
+QMAKE_LFLAGS           += -arch x86_64
 
 load(qt_config)
--- mkspecs/macx-clang-32/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-clang-32/qmake.conf	2016-11-12 07:33:29.000000000 -0700
@@ -11,7 +11,7 @@
 include(../common/clang.conf)
 include(../common/clang-mac.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 QMAKE_CFLAGS           += -arch i386
 QMAKE_OBJECTIVE_CFLAGS += -arch i386
--- mkspecs/macx-g++/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-g++/qmake.conf	2016-11-12 07:33:12.000000000 -0700
@@ -14,6 +14,10 @@
 include(../common/gcc-base-mac.conf)
 include(../common/g++-macx.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
+
+QMAKE_CFLAGS           += -arch x86_64
+QMAKE_CXXFLAGS         += -arch x86_64
+QMAKE_LFLAGS           += -arch x86_64
 
 load(qt_config)
--- mkspecs/macx-g++-32/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-g++-32/qmake.conf	2016-11-12 07:32:47.000000000 -0700
@@ -14,7 +14,7 @@
 include(../common/gcc-base-mac.conf)
 include(../common/g++-macx.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 QMAKE_CFLAGS           += -arch i386
 QMAKE_OBJECTIVE_CFLAGS += -arch i386
--- mkspecs/macx-g++40/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-g++40/qmake.conf	2016-11-12 07:32:23.000000000 -0700
@@ -14,7 +14,7 @@
 include(../common/gcc-base-mac.conf)
 include(../common/g++-macx.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 QMAKE_CC                = gcc-4.0
 QMAKE_CXX               = g++-4.0
--- mkspecs/macx-g++42/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-g++42/qmake.conf	2016-11-12 07:31:58.000000000 -0700
@@ -14,7 +14,7 @@
 include(../common/gcc-base-mac.conf)
 include(../common/g++-macx.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 QMAKE_CC                = gcc-4.2
 QMAKE_CXX               = g++-4.2
--- mkspecs/macx-icc/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-icc/qmake.conf	2016-11-12 07:31:31.000000000 -0700
@@ -84,7 +84,7 @@
 QMAKE_CFLAGS_HIDESYMS   += -fvisibility=hidden
 QMAKE_CXXFLAGS_HIDESYMS += $$QMAKE_CFLAGS_HIDESYMS -fvisibility-inlines-hidden
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.9
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 include(../common/macx.conf)
 
--- mkspecs/macx-llvm/qmake.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/macx-llvm/qmake.conf	2016-11-12 07:30:53.000000000 -0700
@@ -14,7 +14,7 @@
 include(../common/gcc-base-mac.conf)
 include(../common/llvm.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+QMAKE_MACOSX_DEPLOYMENT_TARGET = __MACPORTS_DEPLOYMENT_TARGET__
 
 QMAKE_XCODE_GCC_VERSION = com.apple.compilers.llvmgcc42
 
--- mkspecs/common/clang-mac.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/common/clang-mac.conf	2016-11-12 07:30:23.000000000 -0700
@@ -6,5 +6,7 @@
 
 QMAKE_XCODE_GCC_VERSION = com.apple.compilers.llvm.clang.1_0
 
-QMAKE_CXXFLAGS_CXX11 += -stdlib=libc++
-QMAKE_LFLAGS_CXX11   += -stdlib=libc++
+QMAKE_CXXFLAGS_CXX11 += -stdlib=__MACPORTS_CXX_STDLIB__
+QMAKE_LFLAGS_CXX11   += -stdlib=__MACPORTS_CXX_STDLIB__
+QMAKE_CXXFLAGS += -stdlib=__MACPORTS_CXX_STDLIB__
+QMAKE_LFLAGS   += -stdlib=__MACPORTS_CXX_STDLIB__
--- mkspecs/common/gcc-base.conf.orig	2015-10-12 21:35:32.000000000 -0700
+++ mkspecs/common/gcc-base.conf	2017-10-12 10:22:53.000000000 -0700
@@ -31,7 +31,7 @@
 # you can use the manual test in tests/manual/mkspecs.
 #
 
-QMAKE_CFLAGS_OPTIMIZE      = -O2
+QMAKE_CFLAGS_OPTIMIZE      = __MACPORTS_OPTFLAGS__
 QMAKE_CFLAGS_OPTIMIZE_FULL = -O3
 
 QMAKE_CFLAGS               += -pipe
