--- tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp.orig	2014-06-28 00:25:11.000000000 -0700
+++ tests/auto/testlib/selftests/benchlibcallgrind/tst_benchlibcallgrind.cpp	2014-06-19 03:08:01.000000000 -0700
@@ -73,9 +73,9 @@
     QBENCHMARK {
         __asm__ __volatile__(
             "mov $100000000,%%eax   \n"
-            "LOOPTOP:               \n"
+            "1:                     \n"
             "dec %%eax              \n"
-            "jnz LOOPTOP            \n"
+            "jnz 1b                 \n"
             : /* no output */
             : /* no input */
             : /* clobber */ "eax"
