--- work/boolector-1.4.1-376e6b0-110304/configure	2010-05-14 15:52:49.000000000 +0200
+++ configure	2012-09-08 16:44:05.000000000 +0200
@@ -23,11 +23,6 @@
  [ -f $1 ] || die "can not find '$1'" 
 }
 
-check ../picosat/picosat.h
-check ../picosat/version.o
-check ../picosat/picosat.o
-echo "picosat backend"
-
 if [ $precosat = yes ]
 then
   check ../precosat/precosat.hh
@@ -58,8 +53,8 @@
 rm -f makefile
 sed \
 -e "s,@CC@,$CC," \
--e "s,@CFLAGS@,$CFLAGS," \
--e "s,@LIBS@,$LIBS," \
+-e "s,@CFLAGS@,$CFLAGS $CPPFLAGS," \
+-e "s,@LIBS@,$LDFLAGS -lpicosat $LIBS," \
 -e "s,@PRECOSAT@,$PRECOSAT," \
 makefile.in > makefile
 
