--- mycli/main.py
+++ mycli/main.py
@@ -387,7 +387,7 @@ class MyCli(object):

         database = database or cnf['database']
         # Socket interface not supported for SSH connections
-        if port or host or ssh_host or ssh_port:
+        if (port and host) or (ssh_host and ssh_port):
             socket = ''
         else:
             socket = socket or cnf['socket'] or guess_socket_location()
