--- src/element.c.orig	2012-03-15 02:40:30.000000000 +1100
+++ src/element.c	2013-09-18 15:12:04.000000000 +1000
@@ -306,7 +306,7 @@ clear_element_ptlist(chart_ptlist *ptlst
   *ptlst = NULL;
 }
 
-
+void
 clear_element_lnlist(chart_lnlist *lnlst)
 {
   GList *l;
--- src/remote.c.orig	2007-11-12 14:09:03.000000000 +1100
+++ src/remote.c	2013-09-18 15:18:33.000000000 +1000
@@ -36,7 +36,7 @@ int
 remove_fifos(gpointer data)
 {
   // if not even enabled manually, just go away
-  if (pref.rem_mode != REMOTE_SOCK) return;
+  if (pref.rem_mode != REMOTE_SOCK) return 0;
   
 printf("Removing the remote pipes\n");
   close(pref.inpipe);
