--- lib/QoreObject.cpp.orig	2014-05-17 15:37:03.000000000 +0200
+++ lib/QoreObject.cpp	2014-05-17 15:37:13.000000000 +0200
@@ -1388,7 +1388,7 @@
 	    printd(QRO_LVL, "ObjectRSet::canDelete() this: %p cannot delete graph obj %p '%s' rcount: %d refs: %d\n", this, *i, (*i)->getClassName(), (*i)->priv->rcount, (*i)->references);
 	    return 0;
 	 }
-	 if (!(*i)->priv->status == OS_OK || (*i)->priv->in_destructor) {
+	 if ((*i)->priv->status != OS_OK || (*i)->priv->in_destructor) {
 	    printd(QRO_LVL, "ObjectRSet::canDelete() this: %p cannot delete graph obj %p '%s' status: %d in_destructor: %d\n", this, *i, (*i)->getClassName(), (*i)->priv->status, (*i)->priv->in_destructor);
 	    return 0;
 	 }
