--- configure.orig	2013-07-31 13:35:18.000000000 +0200
+++ configure	2013-07-31 13:36:33.000000000 +0200
@@ -5191,7 +5191,7 @@
 $as_echo_n "checking for flex 2.5.31 or greater... " >&6; }
 if test -z "`echo $LEX | grep flex`"; then
         as_fn_error $? "no flex version found" "$LINENO" 5
-elif test -z "`$LEX --version | grep -e ' 2\.5\.3[1-9]' -e ' 2\.5\.[4-9][0-9]' -e ' 2\.\[6-9]' -e ' [3-9]'`" ; then
+elif test -z "`$LEX --version | grep -e ' 2\.5\.3[1-9]' -e ' 2\.5\.[4-9][0-9]' -e ' 2\.[6-9]' -e ' [3-9]'`" ; then
         flexver=`$LEX --version`
 	as_fn_error $? "incompatible $flexver found - you can download flex 2.5.35 at http://sourceforge.net/projects/flex" "$LINENO" 5
 else
@@ -5249,7 +5249,7 @@
 $as_echo_n "checking for bison 1.85 or greater... " >&6; }
 if test -z "`echo $YACC | grep bison`"; then
 	as_fn_error $? "bison not found" "$LINENO" 5
-elif test -z "`$YACC --version | head -n 1 | grep -e ' 2\.' -e '1\.8[5-9]*' -e '1\.9'`"; then
+elif test -z "`$YACC --version | head -n 1 | grep -e ' [2-9]\.' -e ' 1\.8[5-9]*' -e ' 1\.9'`"; then
         bisonver=`$YACC --version | head -n 1`
 	as_fn_error $? "incompatible $bisonver found" "$LINENO" 5
 else
