From bd63c0f54468c01ac3c8909e3f14aef31da0ed1f Mon Sep 17 00:00:00 2001
From: Ronald Wampler <rdwampler@gmail.com>
Date: Sat, 19 Sep 2015 10:25:00 -0400
Subject: [PATCH] Adopt WebFrameLoadDelegate when building with 10.11 SDK
Upstream-Status: Backport

---
 Commands/tooltip/TMDHTMLTips.mm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Commands/tooltip/TMDHTMLTips.mm b/Commands/tooltip/TMDHTMLTips.mm
index 662ab2f..ff07bc5 100644
--- PlugIns/dialog/Commands/tooltip/TMDHTMLTips.mm
+++ PlugIns/dialog/Commands/tooltip/TMDHTMLTips.mm
@@ -13,7 +13,11 @@
 
 NSString* const TMDTooltipPreferencesIdentifier = @"TM Tooltip";
 
+#if !defined(MAC_OS_X_VERSION_10_11) || (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_11)
 @interface TMDHTMLTip ()
+#else
+@interface TMDHTMLTip () <WebFrameLoadDelegate>
+#endif
 {
 	WebView*	webView;
 	WebPreferences* webPreferences;
