--- bin/create_default_bundles_tbz.orig	2012-11-29 14:31:53.000000000 +1100
+++ bin/create_default_bundles_tbz	2012-11-29 14:44:48.000000000 +1100
@@ -11,7 +11,7 @@
 	BZIP2_FLAG="--use-compress-prog=pbzip2"
 fi
 
-ninja "$bl" && \
+ninja -v "$bl" && \
 	mkdir -p "$BUNDLES_DST/Managed" && \
 	"$bl" -C "$BUNDLES_DST/Managed" install Apache Bundle\ Development Bundle\ Support C CSS Diff Git HTML Hyperlink\ Helper JavaScript JSON Mail Make Markdown Math Mercurial Objective-C PHP Property\ List Python Ruby SCM Shell\ Script Source SQL Subversion Text TextMate Themes TODO XML && \
 	gnutar -cf "$TBZ_DST" "$BZIP2_FLAG" -C "$BUNDLES_DST" "Managed"
