diff --git a/Makefile.am b/Makefile.am
index 6cdbece..265ccfc 100644
--- Makefile.am
+++ Makefile.am
@@ -6,14 +6,22 @@ pkgconfig_DATA =
 if BUILD_XAW6
 
 pkgconfig_DATA += xaw6.pc
-DEFAULT_LIB = libXaw6.so
+if PLATFORM_DARWIN
+DEFAULT_LIB = libXaw.6.@LIBEXT@
+else
+DEFAULT_LIB = libXaw6.@LIBEXT@
+endif
 
 endif
 
 if BUILD_XAW7
 
 pkgconfig_DATA += xaw7.pc
-DEFAULT_LIB = libXaw7.so
+if PLATFORM_DARWIN
+DEFAULT_LIB = libXaw.7.@LIBEXT@
+else
+DEFAULT_LIB = libXaw7.@LIBEXT@
+endif
 
 endif
 
@@ -26,7 +34,7 @@ EXTRA_DIST = xaw6.pc.in \
 
 if !PLATFORM_WIN32
 install-exec-hook:
-	cd $(DESTDIR)$(libdir) && rm -f libXaw.so && ln -s $(DEFAULT_LIB) libXaw.so
+	cd $(DESTDIR)$(libdir) && rm -f libXaw.@LIBEXT@ && ln -s $(DEFAULT_LIB) libXaw.@LIBEXT@
 endif
 
 MAINTAINERCLEANFILES=ChangeLog
diff --git a/configure.ac b/configure.ac
index 5633a46..9ba2414 100644
--- configure.ac
+++ configure.ac
@@ -35,16 +35,22 @@ xorglibxawname="libXaw"
 	rm -f libtool_
 ])
 
-# Win32 DLL rules are different.
+# OSX/Win32 rules are different.
+platform_win32=no
+platform_darwin=no
+LIBEXT=so
 case $host_os in
     cygwin*|mingw*)
 	platform_win32=yes
 	;;
-    *)
-	platform_win32=no
+    darwin*)
+	LIBEXT=dylib
+	platform_darwin=yes
 	;;
 esac
+AC_SUBST(LIBEXT)
 AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AM_CONDITIONAL(PLATFORM_DARWIN, test "x$platform_darwin" = "xyes")
 
 # Whether to build Xaw6
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 85573ce..af4be50 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -77,8 +77,13 @@ libXaw6_la_LIBADD = $(XAW6_LIBS)
 
 if !PLATFORM_WIN32
 install-exec-hook::
-	-rm -f $(DESTDIR)$(libdir)/libXaw.so.6
-	(cd $(DESTDIR)$(libdir) && ln -s libXaw6.so.6 libXaw.so.6)
+if PLATFORM_DARWIN
+	-rm -f $(DESTDIR)$(libdir)/libXaw.6.@LIBEXT@
+	(cd $(DESTDIR)$(libdir) && ln -s libXaw6.6.@LIBEXT@ libXaw.6.@LIBEXT@)
+else
+	-rm -f $(DESTDIR)$(libdir)/libXaw.@LIBEXT@.6
+	(cd $(DESTDIR)$(libdir) && ln -s libXaw6.@LIBEXT@.6 libXaw.@LIBEXT@.6)
+endif
 endif
 
 endif
@@ -101,8 +106,13 @@ libXaw7_la_LIBADD = $(XAW7_LIBS)
 
 if !PLATFORM_WIN32
 install-exec-hook::
-	-rm -f $(DESTDIR)$(libdir)/libXaw.so.7
-	(cd $(DESTDIR)$(libdir) && ln -s libXaw7.so.7 libXaw.so.7)
+if PLATFORM_DARWIN
+	-rm -f $(DESTDIR)$(libdir)/libXaw.7.@LIBEXT@
+	(cd $(DESTDIR)$(libdir) && ln -s libXaw7.7.@LIBEXT@ libXaw.7.@LIBEXT@)
+else
+	-rm -f $(DESTDIR)$(libdir)/libXaw.@LIBEXT@.7
+	(cd $(DESTDIR)$(libdir) && ln -s libXaw7.@LIBEXT@.7 libXaw.@LIBEXT@.7)
+endif
 endif
 
 endif
diff --git a/xaw6.pc.in b/xaw6.pc.in
index ba5a90a..537df32 100644
--- xaw6.pc.in
+++ xaw6.pc.in
@@ -7,6 +7,6 @@ Name: Xaw
 Description: X Athena Widgets Library, version 6
 Version: @PACKAGE_VERSION@
 Requires: xproto xt xmu
-Requires.private: x11 xext xt xmu xpm
+Requires.private: x11 xext xt xmu
 Cflags: -I${includedir}
 Libs: -L${libdir} -lXaw6
