Install headers into subdirectory.
https://savannah.nongnu.org/bugs/?65950
--- libsim/CMakeLists.txt.orig	2019-12-20 07:13:39.000000000 -0600
+++ libsim/CMakeLists.txt	2024-07-03 18:05:54.000000000 -0500
@@ -42,12 +42,12 @@
 file(GLOB files_elfio "${CMAKE_SOURCE_DIR}/include/elfio/elfio/*.hpp")
 
 install(TARGETS sim DESTINATION "lib")
-install(FILES ${files_common} DESTINATION "include")
-install(FILES ${files_hwtimer} DESTINATION "include/hwtimer")
-install(FILES ${files_gdb} DESTINATION "include/gdb")
-install(FILES ${files_ui} DESTINATION "include/ui")
-install(FILES ${files_cmd} DESTINATION "include/cmd")
-install(FILES ${files_elfio} DESTINATION "include/elfio/elfio")
+install(FILES ${files_common} DESTINATION "include/simulavr")
+install(FILES ${files_hwtimer} DESTINATION "include/simulavr/hwtimer")
+install(FILES ${files_gdb} DESTINATION "include/simulavr/gdb")
+install(FILES ${files_ui} DESTINATION "include/simulavr/ui")
+install(FILES ${files_cmd} DESTINATION "include/simulavr/cmd")
+install(FILES ${files_elfio} DESTINATION "include/simulavr/elfio/elfio")
 
 if(BUILD_VERILOG)
 
