Index: kdelibs-4.3.1/kdeui/util/kkeyserver_mac.cpp
===================================================================
--- kdelibs-4.3.1.orig/kdeui/util/kkeyserver_mac.cpp	2008-05-21 13:08:51.000000000 +0200
+++ kdelibs-4.3.1/kdeui/util/kkeyserver_mac.cpp	2009-10-06 07:44:24.000000000 +0200
@@ -68,6 +68,7 @@
     static KeyboardLayoutRef lastLayout = NULL;
     
     void updateScancodes() {
+#ifdef Q_OS_MAC32
         KeyboardLayoutRef layout;
         if (KLGetCurrentKeyboardLayout(&layout) != noErr) {
             kWarning() << "Error retrieving current layout";
@@ -93,6 +94,9 @@
                 }
             }
         }
+#else
+#warning port me to Mac64
+#endif
     }
     
 #define SCANCODE(name, value) { Qt::Key_ ## name, value }    
