Index: kdelibs-4.3.2/cmake/modules/KDE4Macros.cmake
===================================================================
--- kdelibs-4.3.2.orig/cmake/modules/KDE4Macros.cmake	2009-07-21 17:16:04.000000000 +0200
+++ kdelibs-4.3.2/cmake/modules/KDE4Macros.cmake	2009-10-14 10:38:21.000000000 +0200
@@ -1155,11 +1155,7 @@
                 # main target
                 list(APPEND ${appsources} ${_outfilename}.icns)
 
-                #            this doesn't seem to work for me - Use manual "install" instead
-                # TODO: test again with cmake 2.6 ?
-                #           SET_SOURCE_FILES_PROPERTIES(${CMAKE_CURRENT_BINARY_DIR}/${target}.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-
-                install(FILES ${_outfilename}.icns DESTINATION ${BIN_INSTALL_DIR}/${target}.app/Contents/Resources/)
+                set_source_files_properties(${_outfilename}.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
 
             else(_icon)
                 # TODO - try to scale a non-128 icon...? Try to convert an SVG on the fly?
