Index: kdelibs-4.3.1/kio/misc/kpac/discovery.cpp
===================================================================
--- kdelibs-4.3.1.orig/kio/misc/kpac/discovery.cpp	2009-10-06 08:41:03.000000000 +0200
+++ kdelibs-4.3.1/kio/misc/kpac/discovery.cpp	2009-10-06 08:41:19.000000000 +0200
@@ -30,6 +30,9 @@
 #include <netinet/in.h>
 #endif
 #include <arpa/nameser.h>
+#ifdef HAVE_ARPA_NAMESER_COMPAT_H
+#include <arpa/nameser_compat.h>
+#endif
 #ifdef HAVE_ARPA_NAMESER8_COMPAT_H
 #include <arpa/nameser8_compat.h>
 #endif
Index: kdelibs-4.3.1/ConfigureChecks.cmake
===================================================================
--- kdelibs-4.3.1.orig/ConfigureChecks.cmake	2009-10-06 08:41:17.000000000 +0200
+++ kdelibs-4.3.1/ConfigureChecks.cmake	2009-10-06 08:41:19.000000000 +0200
@@ -80,6 +80,7 @@
 check_include_files(crtdbg.h      HAVE_CRTDBG_H)                       # kjs
 check_include_files(langinfo.h    HAVE_LANGINFO_H)                     # kdecore
 
+check_include_files(arpa/nameser_compat.h  HAVE_ARPA_NAMESER_COMPAT_H)  # kio
 check_include_files(arpa/nameser8_compat.h HAVE_ARPA_NAMESER8_COMPAT_H) # kio
 
 macro_bool_to_01(X11_XTest_FOUND HAVE_XTEST)                                                   # kdecore
Index: kdelibs-4.3.1/config.h.cmake
===================================================================
--- kdelibs-4.3.1.orig/config.h.cmake	2009-04-15 12:26:51.000000000 +0200
+++ kdelibs-4.3.1/config.h.cmake	2009-10-06 08:59:22.000000000 +0200
@@ -72,6 +72,7 @@
 #cmakedefine HAVE_SYS_MMAN_H 1
 #cmakedefine HAVE_SYS_UCRED_H 1
 #cmakedefine HAVE_UNISTD_H 1
+#cmakedefine HAVE_ARPA_NAMESER_COMPAT_H
 #cmakedefine HAVE_ARPA_NAMESER8_COMPAT_H
 #cmakedefine HAVE_LANGINFO_H 1
 
