--- kdelibs-4.12.5-orig/cmake/modules/CMakeLists.txt	2014-04-28 13:37:51.000000000 +0900
+++ kdelibs-4.12.5/cmake/modules/CMakeLists.txt	2014-06-24 21:32:09.000000000 +0900
@@ -41,7 +41,6 @@
                FindFAM.cmake
                FindFFmpeg.cmake
                FindFlac.cmake
-               FindFlex.cmake
                FindFontconfig.cmake
                FindGLIB2.cmake
                FindGMP.cmake
