diff -ur kdelibs-4.14.3-orig/kdecore/localization/klocale_kde.cpp kdelibs-4.14.3/kdecore/localization/klocale_kde.cpp
--- kdelibs-4.14.3-orig/kdecore/localization/klocale_kde.cpp	2014-11-07 07:33:50.000000000 +0900
+++ kdelibs-4.14.3/kdecore/localization/klocale_kde.cpp	2019-04-21 11:21:16.000000000 +0900
@@ -2438,8 +2438,8 @@
 
 bool KLocalePrivate::use12Clock() const
 {
-    if ((timeFormat().contains(QString::fromLatin1("%I")) > 0) ||
-        (timeFormat().contains(QString::fromLatin1("%l")) > 0)) {
+    if ((timeFormat().contains(QString::fromLatin1("%I")) != NULL) ||
+        (timeFormat().contains(QString::fromLatin1("%l")) != NULL)) {
         return true;
     } else {
         return false;
diff -ur kdelibs-4.14.3-orig/kio/kfile/kopenwithdialog.cpp kdelibs-4.14.3/kio/kfile/kopenwithdialog.cpp
--- kdelibs-4.14.3-orig/kio/kfile/kopenwithdialog.cpp	2014-11-07 07:33:50.000000000 +0900
+++ kdelibs-4.14.3/kio/kfile/kopenwithdialog.cpp	2019-04-21 11:45:29.000000000 +0900
@@ -713,7 +713,7 @@
         // ### indicate that default value was restored
         d->terminal->setChecked(d->curService->terminal());
         QString terminalOptions = d->curService->terminalOptions();
-        d->nocloseonexit->setChecked((terminalOptions.contains(QLatin1String("--noclose")) > 0));
+        d->nocloseonexit->setChecked((terminalOptions.contains(QLatin1String("--noclose")) != NULL));
         d->m_terminaldirty = false; // slotTerminalToggled changed it
     }
 }
diff -ur kdelibs-4.14.3-orig/kio/kfile/kpropertiesdialog.cpp kdelibs-4.14.3/kio/kfile/kpropertiesdialog.cpp
--- kdelibs-4.14.3-orig/kio/kfile/kpropertiesdialog.cpp	2014-11-07 07:33:50.000000000 +0900
+++ kdelibs-4.14.3/kio/kfile/kpropertiesdialog.cpp	2019-04-21 13:04:23.000000000 +0900
@@ -3306,7 +3306,7 @@
 
     if (preferredTerminal == "konsole")
     {
-        terminalCloseBool = (d->m_terminalOptionStr.contains( "--noclose" ) > 0);
+        terminalCloseBool = (d->m_terminalOptionStr.contains( "--noclose" ) != NULL);
         w.terminalCloseCheck->setChecked(terminalCloseBool);
         d->m_terminalOptionStr.remove( "--noclose");
     }
