diff -ur kdelibs-4.14.3-orig/kdeui/itemviews/kselectionproxymodel.cpp kdelibs-4.14.3/kdeui/itemviews/kselectionproxymodel.cpp
--- kdelibs-4.14.3-orig/kdeui/itemviews/kselectionproxymodel.cpp	2014-11-07 07:33:50.000000000 +0900
+++ kdelibs-4.14.3/kdeui/itemviews/kselectionproxymodel.cpp	2021-03-27 16:59:42.000000000 +0900
@@ -2144,7 +2144,7 @@
     if (!proxyIndex.isValid() || !sourceModel() || d->m_rootIndexList.isEmpty())
         return QModelIndex();
 
-    Q_ASSERT(proxyIndex.internalPointer() >= 0);
+    Q_ASSERT(proxyIndex.internalPointer() != NULL);
     Q_ASSERT(proxyIndex.model() == this);
 
     if (proxyIndex.internalPointer() == 0)
