--- kdelibs-4.8.3.orig/kjs/wtf/HashSet.h		2012-09-04 10:11:08.000000000 +0100
+++ kdelibs-4.8.3/kjs/wtf/HashSet.h				2012-09-06 15:41:13.000000000 +0100
@@ -181,7 +181,7 @@
     inline HashSet<Value, HashFunctions, Traits>::find(const T& value)
     {
         typedef HashSetTranslatorAdapter<ValueType, ValueTraits, T, Translator> Adapter;
-        return m_impl.find<T, Adapter>(value);
+        return m_impl.template find<T, Adapter>(value);
     }
 
     template<typename Value, typename HashFunctions, typename Traits>
@@ -190,7 +190,7 @@
     inline HashSet<Value, HashFunctions, Traits>::find(const T& value) const
     {
         typedef HashSetTranslatorAdapter<ValueType, ValueTraits, T, Translator> Adapter;
-        return m_impl.find<T, Adapter>(value);
+        return m_impl.template find<T, Adapter>(value);
     }
 
     template<typename Value, typename HashFunctions, typename Traits>
@@ -198,7 +198,7 @@
     inline bool HashSet<Value, HashFunctions, Traits>::contains(const T& value) const
     {
         typedef HashSetTranslatorAdapter<ValueType, ValueTraits, T, Translator> Adapter;
-        return m_impl.contains<T, Adapter>(value);
+        return m_impl.template contains<T, Adapter>(value);
     }
 
     template<typename T, typename U, typename V>
