diff -ur kdelibs-4.14.3-orig/kimgio/jp2.cpp kdelibs-4.14.3/kimgio/jp2.cpp
--- kdelibs-4.14.3-orig/kimgio/jp2.cpp	2014-11-07 07:33:50.000000000 +0900
+++ kdelibs-4.14.3/kimgio/jp2.cpp	2021-08-27 11:42:24.000000000 +0900
@@ -103,13 +103,13 @@
         stream->bufmode_ |= bufmode & JAS_STREAM_BUFMODEMASK;
 }
 
-static int qiodevice_read(jas_stream_obj_t *obj, char *buf, int cnt)
+static int qiodevice_read(jas_stream_obj_t *obj, char *buf, unsigned int cnt)
 {
         QIODevice *io = (QIODevice*) obj;
         return io->read(buf, cnt);
 }
 
-static int qiodevice_write(jas_stream_obj_t *obj, char *buf, int cnt)
+static int qiodevice_write(jas_stream_obj_t *obj, const char *buf, unsigned int cnt)
 {
         QIODevice *io = (QIODevice*) obj;
         return io->write(buf, cnt);
