--- kdelibs-4.4.2/CMakeLists.txt.orig	2010-04-30 11:58:10.000000000 -0400
+++ kdelibs-4.4.2/CMakeLists.txt	2010-04-30 11:58:20.000000000 -0400
@@ -81,7 +81,7 @@
 macro_optional_find_package(BZip2)
 macro_log_feature(BZIP2_FOUND "BZip2" "A high-quality data compressor" "http://www.bzip.org" FALSE "" "STRONGLY RECOMMENDED: Provides the ability to read and write bzip2 compressed data files.")
 
-macro_optional_find_package(LibLZMA)
+#macro_optional_find_package(LibLZMA)
 macro_log_feature(LIBLZMA_FOUND "LZMA/XZ" "A very high compression ratio data compressor" "http://tukaani.org/xz/" FALSE "" "Provides the ability to read and write xz compressed data files.")
 
 macro_optional_find_package(OpenSSL)
