--- stats-bsd.c.orig	2003-05-27 17:03:43.000000000 +1000
+++ stats-bsd.c	2010-09-28 23:24:05.000000000 +1000
@@ -48,7 +48,6 @@
 #include <err.h>        /* FreeBSD */
 #include <errno.h>
 #include <fcntl.h>
-#include <kvm.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
--- Makefile.orig	2003-05-27 17:13:02.000000000 +1000
+++ Makefile	2010-09-28 23:32:33.000000000 +1000
@@ -207,7 +207,7 @@
 
 darwin:
 	$(MAKE) $(MAKEFLAGS) CC=cc			\
-		LDFLAGS='$(LDFLAGS) -lkvm -prebind'	\
+		LDFLAGS='$(LDFLAGS) -prebind'	\
 		CFLAGS='$(CFLAGS) -prebind'		\
 		PLATFORMFLAG='-DPLATFORM_BSD'		\
 		all
