--- CMakeLists.txt.orig	2016-11-22 03:27:08.000000000 +0900
+++ CMakeLists.txt	2017-07-10 18:44:16.000000000 +0900
@@ -82,8 +82,7 @@
 ENDIF()
 
 #Add custom CMake Module
-SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/"
-  CACHE INTERNAL "Location of our custom CMake modules.")
+SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
 # auto-configure style checks, other CMake modules.
 INCLUDE(${CMAKE_ROOT}/Modules/CheckLibraryExists.cmake)
