--- libdap2/Makefile.in.orig	2014-04-24 05:44:02.000000000 +0900
+++ libdap2/Makefile.in	2014-10-05 14:21:49.000000000 +0900
@@ -130,7 +130,7 @@
 	libdap2_la-dapdebug.lo libdap2_la-dapattr.lo \
 	libdap2_la-ncd2dispatch.lo libdap2_la-getvara.lo \
 	libdap2_la-dceconstraints.lo libdap2_la-dcetab.lo \
-	libdap2_la-dceparse.lo libdap2_la-dcelex.lo
+	libdap2_la-dceparse.lo libdap2_la-dcelex.lo libdap2_la-ncdap.lo
 am__objects_2 =
 @BUILD_DAP_TRUE@am_libdap2_la_OBJECTS = $(am__objects_1) \
 @BUILD_DAP_TRUE@	$(am__objects_2)
@@ -593,6 +593,13 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libdap2_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libdap2_la-dcelex.lo `test -f 'dcelex.c' || echo '$(srcdir)/'`dcelex.c
 
+libdap2_la-ncdap.lo: ncdap.c
+@am__fastdepCC_TRUE@	$(AM_V_CC)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libdap2_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT libdap2_la-ncdap.lo -MD -MP -MF $(DEPDIR)/libdap2_la-ncdap.Tpo -c -o libdap2_la-ncdap.lo `test -f 'ncdap.c' || echo '$(srcdir)/'`ncdap.c
+@am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) $(DEPDIR)/libdap2_la-ncdap.Tpo $(DEPDIR)/libdap2_la-ncdap.Plo
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	$(AM_V_CC)source='ncdap.c' object='libdap2_la-ncdap.lo' libtool=yes @AMDEPBACKSLASH@
+@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
+@am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libdap2_la_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o libdap2_la-ncdap.lo `test -f 'ncdap.c' || echo '$(srcdir)/'`ncdap.c
+
 mostlyclean-libtool:
 	-rm -f *.lo
 
