--- gcc/config/darwin-driver.c.orig	2020-08-01 15:12:48.000000000 +0200
+++ gcc/config/darwin-driver.c	2020-08-01 15:12:48.000000000 +0200
@@ -65,7 +65,7 @@
   major = strtoul (version_str, &end, 10);
   version_str = end + ((*end == '.') ? 1 : 0);
 
-  if (major != 10) /* So far .. all MacOS 10 ... */
+  if ((major < 10) || (major > 11)) /* So far .. all MacOS 10 ... */
     return NULL;
 
   /* Version string components must be present and numeric.  */
@@ -104,7 +104,7 @@
   if (need_rewrite)
     {
       char *new_version;
-      asprintf (&new_version, "10.%lu.%lu", minor, tiny);
+      asprintf (&new_version, "%lu.%lu.%lu", major, minor, tiny);
       return new_version;
     }
 
