--- gcc/config/darwin-c.c.save	2020-07-23 08:35:17.000000000 +0200
+++ gcc/config/darwin-c.c	2020-07-27 17:58:34.000000000 +0200
@@ -692,13 +692,18 @@
   if (!version_array)
     goto fail;
 
-  if (version_array[MAJOR] != 10)
-    goto fail;
-
-  if (version_array[MINOR] < 10)
-    version_macro = version_as_legacy_macro (version_array);
+  if (version_array[MAJOR] == 11)
+     version_macro = version_as_modern_macro (version_array);
+  
   else
-    version_macro = version_as_modern_macro (version_array);
+
+    if (version_array[MAJOR] != 10)
+      goto fail;
+
+    if (version_array[MINOR] < 10)
+      version_macro = version_as_legacy_macro (version_array);
+    else
+      version_macro = version_as_modern_macro (version_array);
 
   if (!version_macro)
     goto fail;
