--- CMakeLists.txt.orig	2019-03-21 19:51:46.000000000 -0700
+++ CMakeLists.txt	2019-08-10 15:00:38.000000000 -0700
@@ -682,3 +682,6 @@
 
 install (EXPORT glog-targets NAMESPACE glog:: DESTINATION
   ${_glog_CMake_INSTALLDIR})
+
+configure_file ("libglog.pc.in" "${PROJECT_BINARY_DIR}/lib${CPACK_PACKAGE_NAME}.pc" @ONLY)
+install (FILES "${PROJECT_BINARY_DIR}/lib${CPACK_PACKAGE_NAME}.pc" DESTINATION "${_glog_CMake_LIBDIR}/pkgconfig")
--- libglog.pc.in.orig	2019-03-21 19:51:46.000000000 -0700
+++ libglog.pc.in	2019-08-10 15:04:27.000000000 -0700
@@ -1,10 +1,10 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: libglog
 Description: Google Log (glog) C++ logging framework
-Version: @VERSION@
+Version: @PROJECT_VERSION@
 Libs: -L${libdir} -lglog
 Cflags: -I${includedir}
