--- src/glog/stl_logging.h.in.orig	2013-05-04 15:12:09.000000000 +0000
+++ src/glog/stl_logging.h.in
@@ -53,7 +53,9 @@
 #ifdef __GNUC__
 # include <ext/hash_set>
 # include <ext/hash_map>
-# include <ext/slist>
+# ifndef _LIBCPP_VERSION
+#  include <ext/slist>
+# endif
 #endif
 
 // Forward declare these two, and define them after all the container streams
@@ -81,7 +83,9 @@ OUTPUT_TWO_ARG_CONTAINER(std::vector)
 OUTPUT_TWO_ARG_CONTAINER(std::deque)
 OUTPUT_TWO_ARG_CONTAINER(std::list)
 #ifdef __GNUC__
+# ifndef _LIBCPP_VERSION
 OUTPUT_TWO_ARG_CONTAINER(__gnu_cxx::slist)
+# endif
 #endif
 
 #undef OUTPUT_TWO_ARG_CONTAINER
