From e872961958aed867a3fed3e6bccc6e0969552f69 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 11 Jun 2024 03:21:02 +0800
Subject: [PATCH] cpu_features_macros.h: fix PowerPC macros

---
 include/cpu_features_macros.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git include/cpu_features_macros.h include/cpu_features_macros.h
index 4fe7c3d..59897a4 100644
--- cpu_features/include/cpu_features_macros.h
+++ cpu_features/include/cpu_features_macros.h
@@ -59,7 +59,7 @@
 #define CPU_FEATURES_ARCH_MIPS
 #endif
 
-#if defined(__powerpc__)
+#if defined(__powerpc__) || defined(__POWERPC__)
 #define CPU_FEATURES_ARCH_PPC
 #endif
 
