--- usr/bin/makepasswd.orig
+++ usr/bin/makepasswd
@@ -30,7 +30,7 @@ $Chars = "";
 $Clear = "";
 $Count = "";
 $Crypt = 0;
-$CryptMd5 = 0;
+$CryptMd5 = 1;
 $CryptSalt = "";
 $MaxChars = "";
 $MinChars = "";
@@ -470,7 +470,7 @@ Options are:
 --count=N        Produce a total of N passwords (the default is one).
 --crypt          Produce encrypted passwords.
 --crypt-md5      Produce encrypted passwords using the MD5 digest (hash)
-                       algorithm.
+                       algorithm (the default).
 --cryptsalt=N    Use crypt() salt N, a positive number <= 4096.  If random
                        seeds are desired, specify a zero value (the default).
 --help           Ignore other operands and produce only this help display.
