--- setup.py	2020-07-07 11:58:27.000000000 -0700
+++ setup.py	2020-10-09 20:33:36.000000000 -0700
@@ -332,25 +332,16 @@
         main_file('oids'),
         main_file('cython_converters'),
         main_file('chan_bindings'),
-        extension_file('s4u', 'gss_acquire_cred_impersonate_name'),
-        extension_file('cred_store', 'gss_store_cred_into'),
-        extension_file('rfc4178', 'gss_set_neg_mechs'),
         extension_file('rfc5587', 'gss_indicate_mechs_by_attrs'),
-        extension_file('rfc5588', 'gss_store_cred'),
         extension_file('rfc5801', 'gss_inquire_saslname_for_mech'),
         extension_file('cred_imp_exp', 'gss_import_cred'),
-        extension_file('dce', 'gss_wrap_iov'),
-        extension_file('iov_mic', 'gss_get_mic_iov'),
         extension_file('ggf', 'gss_inquire_sec_context_by_oid'),
         extension_file('set_cred_opt', 'gss_set_cred_option'),
 
         # see ext_rfc6680_comp_oid for more information on this split
-        extension_file('rfc6680', 'gss_display_name_ext'),
-        extension_file('rfc6680_comp_oid', 'GSS_C_NT_COMPOSITE_EXPORT'),
 
         # see ext_password{,_add}.pyx for more information on this split
         extension_file('password', 'gss_acquire_cred_with_password'),
-        extension_file('password_add', 'gss_add_cred_with_password'),
     ]),
     keywords=['gssapi', 'security'],
     install_requires=install_requires
