--- Makefile	2024-09-27 19:06:27
+++ Makefile.new	2024-12-27 01:31:54
@@ -71,7 +71,7 @@
 LIBDIR_Darwin = lib
 
 ifeq ($(PREFIX),)
-    PREFIX := /usr/local
+    PREFIX := @@DEFAULT_PREFIX@@
 endif
 
 .PHONY: install clean
@@ -96,11 +96,11 @@
 else
 ifeq ($(EFI),1)
 ifeq ($(OS),Darwin)
-	install_name_tool -id libkrun-efi.dylib target/release/libkrun.dylib
+	install_name_tool -id $(PREFIX)/lib/libkrun-efi.dylib target/$(CARGO_BUILD_TARGET)/release/libkrun.dylib
 endif
-	mv target/release/libkrun.dylib target/release/$(KRUN_BASE_$(OS))
+	mv target/$(CARGO_BUILD_TARGET)/release/libkrun.dylib target/$(CARGO_BUILD_TARGET)/release/$(KRUN_BASE_$(OS))
 endif
-	cp target/release/$(KRUN_BASE_$(OS)) $(LIBRARY_RELEASE_$(OS))
+	cp target/$(CARGO_BUILD_TARGET)/release/$(KRUN_BASE_$(OS)) $(LIBRARY_RELEASE_$(OS))
 endif
 
 $(LIBRARY_DEBUG_$(OS)): $(INIT_BINARY)
