--- configure.in.orig	2011-08-20 20:56:49.000000000 -0700
+++ configure.in	2011-08-20 20:57:24.000000000 -0700
@@ -9263,19 +9263,7 @@ xpcom/xpcom-config.h
 xpcom/xpcom-private.h
 )
 
-# Hack around an Apple bug that effects the egrep that comes with OS X 10.7.
-# "arch -arch i386 egrep" always uses the 32-bit Intel part of the egrep fat
-# binary, even on 64-bit systems.  It should work on OS X 10.4.5 and up.  We
-# (apparently) only need this hack when egrep's "pattern" is particularly
-# long (as in the following code).  See bug 655339.
-case "$host" in
-*-apple-darwin*)
-    FIXED_EGREP="arch -arch i386 egrep"
-    ;;
-*)
     FIXED_EGREP="egrep"
-    ;;
-esac
 
 # Save the defines header file before autoconf removes it.
 # (Do not add AC_DEFINE calls after this line.)
