diff -Nur wml-2.0.11.orig/Makefile.in wml-2.0.11/Makefile.in
--- wml-2.0.11.orig/wml_common/Makefile.in	2011-07-27 17:30:03.000000000 +0200
+++ wml-2.0.11/wml_common/Makefile.in		2011-07-27 17:31:34.000000000 +0200
@@ -45,14 +45,9 @@
      have_gd=0
 
 SUBDIRS = \
-     bitvector \
      filepathconvert \
      gd \
-     getoptlong \
-     htmlclean \
-     imagesize \
-     io \
-     termreadkey
+     htmlclean
 
 # ------------------------------------------------
 #   TARGETS
