--- qwtconfig.pri.orig	2010-06-28 10:39:44.000000000 -0600
+++ qwtconfig.pri	2010-06-28 10:40:28.000000000 -0600
@@ -7,17 +7,13 @@
 VER_PAT      = 1
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
-unix {
-    INSTALLBASE    = /usr/local/qwt-5.2.1
-}
-
 win32 {
     INSTALLBASE    = C:/Qwt-5.2.1
 }
 
 target.path    = $$INSTALLBASE/lib
-headers.path   = $$INSTALLBASE/include
-doc.path       = $$INSTALLBASE/doc
+headers.path   = $$INSTALLBASE/include/qwt
+doc.path       = $$INSTALLBASE/share/qwt/doc
 
 ######################################################################
 # qmake internal options
