--- d2dbs/cmdline_parse.c	Wed Dec 26 02:10:05 2001
+++ d2dbs/cmdline_parse.c.new	Wed May 12 15:24:07 2004
@@ -64,18 +64,18 @@
 static t_param cmdline_param;
 
 static char help_message[]="\
-Usage: d2cs [<options>]
-	-m <FILE>:		set memory debug logging file to FILE
-	-c <FILE>:		set configuration file to FILE
-	-l <FILE>:		set log to FILE
-	-h, --help:		show this help message and exit
-	-v, --version:		show version information and exit
-	-f, --foreground:	start in foreground mode (don`t daemonize)
-	-s, --stderr:		log to stderr instead of logging to file
-
-Notes:
-	1.You should always use absolute path here for all FILE names
-	2.-m option only works when compiled with USE_CHECK_ALLOC defined
+Usage: d2cs [<options>] \
+	-m <FILE>:		set memory debug logging file to FILE \
+	-c <FILE>:		set configuration file to FILE \
+	-l <FILE>:		set log to FILE \
+	-h, --help:		show this help message and exit \
+	-v, --version:		show version information and exit \
+	-f, --foreground:	start in foreground mode (don`t daemonize) \
+	-s, --stderr:		log to stderr instead of logging to file \
+\
+Notes: \
+	1.You should always use absolute path here for all FILE names \
+	2.-m option only works when compiled with USE_CHECK_ALLOC defined \
 ";
 
 extern void cmdline_show_help(void)
