--- setupext.py.orig	2019-02-27 23:27:59.000000000 -0500
+++ setupext.py	2019-02-27 23:28:57.000000000 -0500
@@ -1541,10 +1541,10 @@
 
 class BackendTkAgg(OptionalBackendPackage):
     name = "tkagg"
-    force = True
-
-    def check(self):
-        return "installing; run-time loading from Python Tcl / Tk"
+##    force = True
+##
+##    def check(self):
+##        return "installing; run-time loading from Python Tcl / Tk"
 
     def runtime_check(self):
         """ Checks whether TkAgg runtime dependencies are met
