--- setup.py.orig	2019-02-25 19:18:32.000000000 -0500
+++ setup.py	2019-05-21 10:36:03.000000000 -0400
@@ -188,9 +188,9 @@
         _download_jquery_to("lib/matplotlib/backends/web_backend/")
 
 
-cmdclass['sdist'] = sdist_with_jquery
-cmdclass['install_lib'] = install_lib_with_jquery
-cmdclass['develop'] = develop_with_jquery
+##cmdclass['sdist'] = sdist_with_jquery
+##cmdclass['install_lib'] = install_lib_with_jquery
+##cmdclass['develop'] = develop_with_jquery
 
 
 # One doesn't normally see `if __name__ == '__main__'` blocks in a setup.py,
