--- kabc/plugins/ldap/configure.in.in	14 Jan 2003 19:45:52 -0000	1.1.1.3
+++ kabc/plugins/ldap/configure.in.in	21 Feb 2003 21:42:06 -0000	1.9
@@ -19,7 +19,7 @@
   fi
   if test $with_ldap = FOUND ; then
     with_ldap=NOTFOUND
-    for ext in la so sl a ; do
+    for ext in la so sl a dylib; do
       AC_FIND_FILE(libldap.$ext, $kde_libraries /usr/lib /usr/local/lib,
         ldap_libdir)
       if test -r $ldap_libdir/libldap.$ext ; then
@@ -38,6 +38,12 @@
 case "$with_ldap" in
 no) AC_MSG_RESULT(no) ;;
 NOTFOUND) AC_MSG_RESULT(searched but not found) ;;
+framework)
+  LDAP_LIBS="-Xlinker -framework -Xlinker LDAP"
+  AC_DEFINE_UNQUOTED(HAVE_LIBLDAP, 1, [Define if you have LDAP libraries])
+  LDAP_SUBDIR="ldap"
+  AC_MSG_RESULT(Apple framework)
+  ;;
 *)
   if test "x$with_ldap" = "xFOUND" ; then
     msg="incs=$ldap_incdir libs=$ldap_libdir"
