--- kinit/kinit.cpp	13 Mar 2003 02:18:42 -0000	1.1.1.6
+++ kinit/kinit.cpp	13 Mar 2003 02:52:40 -0000	1.8
@@ -81,11 +81,6 @@
 # endif
 #endif
 
-#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)
-# include <crt_externs.h>
-# define environ (*_NSGetEnviron())
-#endif
-
 extern char **environ;
 
 extern int lt_dlopen_flag;
@@ -1437,6 +1432,7 @@
    kdeinit_setproctitle("Starting up...");
    kdeinit_library_path();
    unsetenv("LD_BIND_NOW");
+   unsetenv("DYLD_BIND_AT_LAUNCH");
    KApplication::loadedByKdeinit = true;
 
    d.maxname = strlen(argv[0]);
