--- src/gui/tabledialog.cpp.orig	2009-07-29 00:50:26.000000000 -0500
+++ src/gui/tabledialog.cpp	2014-06-08 15:33:41.000000000 -0500
@@ -186,7 +186,7 @@
 void TableDialog::chooseDir(void) {
   QString path = QFileDialog::getExistingDirectory(this, 
 						   tr("Select the directory which contains the tables."), 
-						   QString(), 
+						   QString("@TABLES_DIR@"), 
 						   QFileDialog::ShowDirsOnly);
   
   if (!path.isNull()) emit install(path, true, true);
