--- makefile.orig	2019-02-08 02:53:47.000000000 +0900
+++ makefile	2019-11-29 15:00:31.000000000 +0900
@@ -115,8 +115,8 @@
 
 
 # Warning do not set both USE_NETCDF3 and USE_NETCDF4 to one
-USE_NETCDF3=1
-USE_NETCDF4=0
+USE_NETCDF3=0
+USE_NETCDF4=1
 USE_REGEX=1
 USE_TIGGE=1
 USE_MYSQL=0
@@ -124,13 +124,13 @@
 USE_SPECTRAL=0
 USE_UDF=0
 USE_OPENMP=1
-USE_PROJ4=0
+USE_PROJ4=1
 USE_WMO_VALIDATION=0
 DISABLE_TIMEZONE=0
 MAKE_FTN_API=1
 DISABLE_ALARM=0
 
-USE_G2CLIB=0
+USE_G2CLIB=1
 USE_PNG=1
 USE_JASPER=1
 USE_AEC=1
@@ -446,8 +446,8 @@
 # gctpc library
 gctpcdir:=${cwd}/gctpc
 gctpcsrc:=gctpc20a.tgz
-gctpclib:=${lib}/libgeo.a
-wLDFLAGS+=-lgeo
+gctpclib:=${lib}/libgctpc.a
+wLDFLAGS+=-lgctpc
 # wCPPFLAGS+=-I${gctpc}/source
 
 # proj4 library
@@ -613,11 +613,11 @@
 w=wgrib2
 prog=$w/wgrib2
 
-all:	${netcdf4src} ${hdf5src} ${prog} aux_progs/gmerge aux_progs/smallest_grib2 aux_progs/smallest_4
+all:	${prog} aux_progs/gmerge aux_progs/smallest_grib2 aux_progs/smallest_4
 
 
-${prog}:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
-	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE}
+${prog}:        $w/*.c $w/*.h ${g2clib} ${iplib}
+	cd "$w" && ${MAKE} LDFLAGS="${LDFLAGS} ${wLDFLAGS}" CPPFLAGS="-I${g2cdir} ${CPPFLAGS}"
 
 fast:        $w/*.c $w/*.h ${jlib} ${aeclib} ${netcdf3lib} ${pnglib} ${hdf5lib} ${g2clib} ${netcdf4lib} ${iplib} ${spectrallib} ${gctpclib} ${proj4lib}
 	cd "$w" && export LDFLAGS="${wLDFLAGS}" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE} fast
@@ -677,8 +677,11 @@
 ${zlib}:	${zdir}
 	cd ${zdir} && export CFLAGS="${wCPPFLAGS}" && ./configure --prefix=${cwd} --static && ${MAKE} install
 
-${g2clib}:	${jlib} ${pnglib} ${zlib}
-	cd "$g2cdir" && export CPPFLAGS="${wCPPFLAGS}" && ${MAKE} && cp libgrib2c.a ${lib}
+${g2clib}:         
+	touch ${g2clib}    
+	rm ${g2clib}    
+	cd "${g2cdir}" && ${MAKE} && cp libgrib2c.a ${lib}    
+	ranlib ${g2clib} 
 
 ${gctpcdir}/source/makefile.gctpc:
 	cp ${gctpcsrc} tmpgctpc.tar.gz
@@ -733,7 +736,7 @@
 	cd "${hdf5dir}" && export CFLAGS="${hdf5CFLAGS}" && export LDFLAGS="${LDFLAGS}" && ./configure --disable-shared --with-zlib=$z --prefix=${cwd} && ${MAKE} all check install
 
 ${iplib}:
-	cd "${ipdir}" && export FFLAGS="${wFFLAGS}" && export FTN_REAL8=${FTN_REAL8}  && ${MAKE} && cp $(notdir ${iplib}) ${iplib}
+	cd "${ipdir}" && ${MAKE} && cp $(notdir ${iplib}) ${iplib}
 
 ${spectrallib}:
 	cd "${spectraldir}" && export FFLAGS="${wFFLAGS}" && export FTN_REAL8=${FTN_REAL8}  && ${MAKE} && cp $(notdir ${spectrallib}) ${spectrallib}
