--- wgrib2/unpk.c.orig	2021-05-04 03:12:43.000000000 +0900
+++ wgrib2/unpk.c	2022-09-11 11:48:25.000000000 +0900
@@ -202,7 +202,8 @@
         image = NULL;
 	opts = NULL;
         jpcstream=jas_stream_memopen((char *) sec[7]+5, (int) GB2_Sec7_size(sec)-5);
-	image = jpc_decode(jpcstream, opts);
+        int fmt = jas_image_strtofmt((char*)"jpc");
+	image = jas_image_decode(jpcstream, fmt, opts);
 	if (image == NULL) fatal_error("jpeg2000 decoding", "");
 	pcmpt = image->cmpts_[0];
         if (image->numcmpts_ != 1 ) 
