--- sp_v2.0.2_d/Makefile.orig	2020-09-24 01:11:32.000000000 +0900
+++ sp_v2.0.2_d/Makefile	2021-07-24 18:12:13.000000000 +0900
@@ -37,11 +37,11 @@
 	-rm -f *.o *.mod *.a
 
 fftpack.o:	fftpack.f
-	$(FC) $(FFLAGS) ${FTN_REAL8} ${FTN_LEGACY} $<
+	$(FC) $(FFLAGS) ${FTN_REAL8} ${FTN_LEGACY} $< -c
 
 .F.o:
-	$(FC) $(FFLAGS) ${FTN_REAL8} $(FPPFLAGS) $<
+	$(FC) $(FFLAGS) ${FTN_REAL8} $(FPPFLAGS) $< -c
 
 .f.o:
-	$(FC) $(FFLAGS) ${FTN_REAL8} $<
+	$(FC) $(FFLAGS) ${FTN_REAL8} $< -c
 
