--- g2clib-1.4.0/dec_jpeg2000.c.orig	2012-02-15 22:54:35.000000000 +0900
+++ g2clib-1.4.0/dec_jpeg2000.c	2022-09-11 13:20:35.000000000 +0900
@@ -70,9 +70,10 @@
 //   
 //     Decode JPEG200 codestream into jas_image_t structure.
 //       
-    image=jpc_decode(jpcstream,opts);
+    int fmt = jas_image_strtofmt((char*)"jpc");
+    image=jas_image_decode(jpcstream, fmt, opts);
     if ( image == 0 ) {
-       printf(" jpc_decode return\n");
+       printf(" jas_image_decode return\n");
        return -3;
     }
     
