diff -Naurp grib2.orig/g2clib-1.4.0/makefile grib2/g2clib-1.4.0/makefile
--- g2clib-1.4.0/makefile	2012-09-04 05:27:30.000000000 -0700
+++ g2clib-1.4.0/makefile	2013-11-17 22:57:54.000000000 -0800
@@ -86,6 +86,7 @@ $(LIB):	$(LIB)(gridtemplates.o) \
 #	$(LIB)(getgb2.o) \
 #	$(LIB)(putgb2.o) \
 #	$(LIB)(params.o)
+	ranlib ${LIB}
 
 .c.a:
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) ${DEFS} $<
diff -Naurp grib2.orig/iplib.2012/makefile grib2/iplib.2012/makefile
--- iplib.2012/makefile	2012-02-24 11:57:49.000000000 -0800
+++ iplib.2012/makefile	2013-11-17 22:58:32.000000000 -0800
@@ -32,6 +32,7 @@ gdswizcd.o  gdswzdcd.o
 
 $(LIB):	$(objs)
 	ar $(ARFLAGS) ${LIB} $(objs)
+	ranlib ${LIB}
 
 .f.o:
 	$(FC) ${FFLAGS} -c $<
diff -Naurp grib2.orig/makefile.gctpc grib2/makefile.gctpc
--- makefile.gctpc	2012-04-20 10:52:55.000000000 -0700
+++ makefile.gctpc	2013-11-17 22:57:28.000000000 -0800
@@ -23,6 +23,7 @@ objs=gctp.o alberfor.o alberinv.o alconf
 
 $(LIB): $(objs)
 	ar $(ARFLAGS) ${LIB} $(objs)
+	ranlib ${LIB}
 
 clean:
 	touch junk.o
