--- libexec/bats-core/bats
+++ libexec/bats-core/bats
@@ -133,9 +133,11 @@
 if [[ ! -d "${BATS_TMPDIR}" ]]; then
   printf "Error: BATS_TMPDIR (%s) does not exist or is not a directory" "${BATS_TMPDIR}" >&2
   exit 1
-elif [[ ! -w "${BATS_TMPDIR}" ]]; then
+elif ! touch "${BATS_TMPDIR}/.bats_tmp_test_$$" 2>/dev/null; then
   printf "Error: BATS_TMPDIR (%s) is not writable" "${BATS_TMPDIR}" >&2
   exit 1
+else
+  rm -f "${BATS_TMPDIR}/.bats_tmp_test_$$"
 fi
 
 arguments=()
