--- netcdf-fortran.pc.in.orig	2012-05-04 06:28:17.000000000 +0900
+++ netcdf-fortran.pc.in	2012-05-04 06:31:10.000000000 +0900
@@ -1,10 +1,10 @@
-prefix=/usr/local
+prefix=@prefix@
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
-ccompiler=gcc
+ccompiler=@CC@
 cppcompiler=@CXX@
-fcompiler=gfortran
+fcompiler=@FC@
 fflags=-g -O2 -I${includedir}
 flibs=-lnetcdff -lnetcdf
 
