--- docs/Makefile.in.orig	2023-05-20 05:19:16
+++ docs/Makefile.in	2024-07-27 17:54:38
@@ -653,11 +653,11 @@
 
 # Create the F77 API man page from the m4 source.
 netcdf_f77.3: netcdf.m4
-	m4 $(M4FLAGS) $(ARGS_MANPAGE) $? >$@ || rm $@
+	gm4 $(M4FLAGS) $(ARGS_MANPAGE) $? >$@ || rm $@
 
 # Create the F90 API man page from the m4 source.
 netcdf_fortran.3: netcdf.m4
-	m4 $(M4FLAGS) $(ARGS_MANPAGE) $? >$@ || rm $@
+	gm4 $(M4FLAGS) $(ARGS_MANPAGE) $? >$@ || rm $@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
