--- tests/Makefile.in	2008-07-31 12:24:20.000000000 +0000
+++ tests/Makefile.in.new	2008-09-29 12:19:00.000000000 +0000
@@ -218,7 +218,8 @@
 	      $(LIBGCRYPT_CFLAGS) $(KSBA_CFLAGS)
 
 AM_LDFLAGS = $(LDFLAGS)
-LDADD = ../jnlib/libjnlib.a $(LIBASSUAN_LIBS) $(LIBGCRYPT_LIBS) $(KSBA_LIBS)
+LDADD = ../jnlib/libjnlib.a $(LIBASSUAN_LIBS) $(LIBGCRYPT_LIBS) $(KSBA_LIBS) \
+	$(LIBICONV) $(LIBINTL)
 asschk_SOURCES = asschk.c
 show_multi_SOURCERS = show-multi.c
 show_multi_LDADD = 
