--- unix/CMakeLists.txt	2022-04-02 21:10:02
+++ unix/CMakeLists.txt	2023-01-20 20:37:11
@@ -15,15 +15,11 @@
   # a local better alternative
   ../utils/ltime.c)
 
-set(CURSES_NEED_WIDE TRUE)
-find_package(Curses)
-if(CURSES_FOUND)
-  set(CURSES_WIDE TRUE)
-else()
+# Previous version disabled ncursesw - for now do the same here
   set(CURSES_NEED_WIDE FALSE)
   find_package(Curses)
   set(CURSES_WIDE FALSE)
-endif()
+
 if(NOT CURSES_FOUND)
   message(FATAL_ERROR "Cannot build IPBT without a curses library")
 endif()
