--- picocom.c.orig	2010-10-27 15:40:52.000000000 +0200
+++ picocom.c	2010-10-27 15:39:02.000000000 +0200
@@ -455,8 +455,8 @@
 	if ( baud > 115200 )
 		baud = 115200;
 #else
-	if ( baud > 921600 )
-		baud = 921600;
+	if ( baud > 230400 )
+		baud = 230400;
 #endif
 
 	return baud;
@@ -469,8 +469,8 @@
 	if ( baud > 115200 )
 		baud = 115200;
 #else
-	if ( baud > 921600 )
-		baud = 921600;
+	if ( baud > 230400 )
+		baud = 230400;
 #endif
 	else if ( baud == 57600 )
 		baud = 38400;
