Index: INSTALL.im
===================================================================
--- INSTALL.im.orig
+++ INSTALL.im
@@ -76,8 +76,8 @@ void setLocations()
     MAN             = BASE + "/share/man";    
     STD_INCLUDE     = BASE + "/share/yodl";
 
-    COMPILER = "gcc";
-    CXX = "g++";
+    COMPILER = "__MP_CC__";
+    CXX = "__MP_CXX__";
 }
 
 
Index: build
===================================================================
--- build.orig
+++ build
@@ -13,6 +13,7 @@
 string  CLASSES;
 string  g_lopt;
 string  g_copt;
+string  g_cxxopt;
 string  g_cwd;
 string  g_wip =     "tmp/wip";          // no slash here
 string  g_install = "tmp/install";      // no slash here
@@ -61,6 +62,8 @@ void main(int argc, list argv)
 
     g_copt = setOpt(COPT + " -g", "CFLAGS");
 
+    g_cxxopt = setOpt(g_cxxopt, "CXXFLAGS");
+
 #ifdef PROFILING
     g_copt = COPT + " -pg";
     g_lopt = "-pg";
Index: icmake/program
===================================================================
--- icmake/program.orig
+++ icmake/program
@@ -27,7 +27,7 @@ void _programYodlverbinsert()
     if ("src/verbinsert/verbinsert.cc" younger 
             g_install + BIN + "/yodlverbinsert")
         run(CXX + " -o " + g_install + BIN + "/yodlverbinsert " + 
-            g_copt + " src/verbinsert/verbinsert.cc " + g_lopt);
+            g_cxxopt + " src/verbinsert/verbinsert.cc " + g_lopt);
 }
 
 void program(string target)
