Index: INSTALL.im
===================================================================
--- INSTALL.im.orig
+++ INSTALL.im
@@ -65,14 +65,14 @@ string CXX;
 
 void setLocations()
 {
-    BASE    = "/usr";
+    BASE    = "__MP_PREFIX__";
 
     // make sure that BIN, STD_INCLUDE, MAN, DOC and DOCDOC all are
     // absolute paths
 
     BIN             = BASE + "/bin";
     DOC             = BASE + "/share/doc/yodl";
-    DOCDOC          = BASE + "/share/doc/yodl-doc";
+    DOCDOC          = DOC;
     MAN             = BASE + "/share/man";    
     STD_INCLUDE     = BASE + "/share/yodl";
 
Index: build
===================================================================
--- build.orig
+++ build
@@ -1,4 +1,4 @@
-#!/usr/bin/icmake -qt/tmp/yodl
+#!__MP_PREFIX__/bin/icmake -qt/tmp/yodl
 
 #include "VERSION" 
 #include "INSTALL.im"
