diff -u src/dvd_input.c.orig src/dvd_input.c
--- src/dvd_input.c.orig	2020-03-05 14:18:33.000000000 -0500
+++ src/dvd_input.c	2021-01-26 17:06:33.000000000 -0500
@@ -318,7 +318,7 @@
   /* dlopening libdvdcss */
 
 #ifdef __APPLE__
-  #define CSS_LIB "libdvdcss.2.dylib"
+  #define CSS_LIB "@@PREFIX@@/lib/libdvdcss.2.dylib"
 #elif defined(_WIN32)
   #define CSS_LIB "libdvdcss-2.dll"
 #elif defined(__OS2__)
