--- dvdread/dvd_reader.h.orig	2006-10-06 02:12:31.000000000 -0600
+++ dvdread/dvd_reader.h	2008-05-28 02:21:03.000000000 -0600
@@ -240,6 +240,8 @@
  */
 int DVDFileSeek( dvd_file_t *, int );
 
+int DVDFileSeekForce( dvd_file_t *, int, int );
+
 /**
  * Reads the given number of bytes from the file.  This call can only be used
  * on the information files, and may not be used for reading from a VOB.  This
