--- http-replicator.conf	Sat May  1 19:16:06 2004
+++ http-replicator.conf.new	Sun May  2 00:10:19 2004
@@ -39,17 +39,17 @@
 #  Where in this directory the files are actually put depends on if the server is in flat mode.
 #  By default the entire directory structure is copied.
 
-DIR = '/home/cache'
+DIR = '__PREFIX__/var/http-replicator/'
 
 #  The process id of the running process is saved in [PID].
 #  As this is done before changing into [USER], write permission for [USER] for this file is not needed.
 
-PID = '/var/run/http-replicator.pid'
+PID = '__PREFIX__/var/run/http-replicator.pid'
 
 #  All messages on stdout and stderr are in daemon mode written to the [LOG].
 #  Just as for [PID], write permission for [LOG] is not necessary.
 
-LOG = '/var/log/http-replicator'
+LOG = '__PREFIX__/var/log/http-replicator'
 
 #  When the proxy server is used to maintain a package cache a cron script can delete the oldest packages.
 #  The value of [KEEP] sets the maximum number of versions of each package to be kept.
