diff -ur remind-03.01.09.orig/configure remind-03.01.09/configure
--- configure	2010-06-02 15:45:50.000000000 -0400
+++ configure	2010-10-12 14:55:48.000000000 -0400
@@ -1691,21 +1691,6 @@
 ac_config_headers="$ac_config_headers src/config.h"
 
 
-if test "`uname -s`" = "Darwin" ; then
-   trap 'echo Be patient...' INT TERM
-   cat <<'EOF'
-
-Please don't use Apple products.  This script will continue in 30 seconds
-if you insist on compiling Remind on Mac OS X.
-
-EOF
-    for i in 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 ; do
-        sleep 1
-    done
-    trap - INT
-    trap - TERM
-fi
-
 if uname -s | grep -i -q 'cygwin' ; then
    trap 'echo Be patient...' INT TERM
    cat <<'EOF'
diff -ur remind-03.01.09.orig/configure.in remind-03.01.09/configure.in
--- configure.in	2010-06-02 15:45:50.000000000 -0400
+++ configure.in	2010-10-12 14:56:06.000000000 -0400
@@ -14,21 +14,6 @@
 
 AC_CONFIG_HEADER(src/config.h)
 
-if test "`uname -s`" = "Darwin" ; then
-   trap 'echo Be patient...' INT TERM
-   cat <<'EOF'
-
-Please don't use Apple products.  This script will continue in 30 seconds
-if you insist on compiling Remind on Mac OS X.
-
-EOF
-    for i in 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 ; do
-        sleep 1
-    done
-    trap - INT
-    trap - TERM
-fi
-
 if uname -s | grep -i -q 'cygwin' ; then
    trap 'echo Be patient...' INT TERM
    cat <<'EOF'
diff -ur remind-03.01.09.orig/src/init.c remind-03.01.09/src/init.c
--- src/init.c	2010-05-18 20:45:23.000000000 -0400
+++ src/init.c	2010-10-12 14:44:12.000000000 -0400
@@ -139,12 +139,6 @@
 
     int jul, tim;
 
-#if defined(__APPLE__)
-    rkrphgvba(0);
-#elif defined(__CYGWIN__)
-    rkrphgvba(1);
-#endif
-
     jul = NO_DATE;
     tim = NO_TIME;
 
