From 892683b4173a4ad4ecd463f548db1e2a79b42b9e Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 26 Jan 2024 21:34:27 +0800
Subject: [PATCH] AppleSMC.h: fix includes

---
 addons/AppleSensors/source/AppleSMC.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git addons/AppleSensors/source/AppleSMC.h addons/AppleSensors/source/AppleSMC.h
index 565d3731..1acccc73 100644
--- addons/AppleSensors/source/AppleSMC.h
+++ addons/AppleSensors/source/AppleSMC.h
@@ -1,5 +1,8 @@
+#include <AvailabilityMacros.h>
 #include <mach/mach.h>
-//#include <Carbon/Carbon.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1090
+#include <Carbon/Carbon.h>
+#endif
 #include <IOKit/IOKitLib.h>
 #include <CoreFoundation/CoreFoundation.h>
 
