From 787c8b090934f7c9d362b70fcb50b065f35be7c0 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 26 Jan 2024 22:09:47 +0800
Subject: [PATCH] CMakeLists: drop breaking simd flag

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index ce1ca176..6cac2b27 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -39,7 +39,7 @@ IF(CMAKE_COMPILER_IS_GNUCC)
 	SET(CMAKE_BUILD_TYPE_DebugFast)
 	SET(CMAKE_CXX_FLAGS_DEBUGFAST "-g -O0")
 	SET(CMAKE_C_FLAGS_DEBUGFAST "-g -O0")
-	SET(CMAKE_C_FLAGS "-msse2 -fobjc-exceptions")
+	SET(CMAKE_C_FLAGS "-D__SIMD_NO_SIMD__ -fobjc-exceptions")
 	if(NOT CMAKE_BUILD_TYPE)
 	        SET(CMAKE_BUILD_TYPE "DebugFast")
 	endif(NOT CMAKE_BUILD_TYPE)
