--- Makefile.orig	2014-12-06 01:12:55.000000000 +0900
+++ Makefile	2014-12-21 18:19:42.000000000 +0900
@@ -167,10 +167,10 @@
 endif
 LIBRARIES += glog gflags protobuf leveldb snappy \
 	lmdb \
-	boost_system \
+	boost_system-mt \
 	hdf5_hl hdf5 \
 	opencv_core opencv_highgui opencv_imgproc pthread
-PYTHON_LIBRARIES := boost_python python2.7
+PYTHON_LIBRARIES := boost_python-mt python2.7
 WARNINGS := -Wall -Wno-sign-compare
 
 ##############################
@@ -239,13 +239,13 @@
 # clang++ instead of g++
 # libstdc++ instead of libc++ for CUDA compatibility on 10.9
 ifeq ($(OSX), 1)
-	CXX := /usr/bin/clang++
+	CXX ?= /usr/bin/clang++
 	# clang throws this warning for cuda headers
 	WARNINGS += -Wno-unneeded-internal-declaration
-	ifneq ($(findstring 10.9, $(shell sw_vers -productVersion)),)
-		CXXFLAGS += -stdlib=libstdc++
-		LINKFLAGS += -stdlib=libstdc++
-	endif
+#	ifneq ($(findstring 10.9, $(shell sw_vers -productVersion)),)
+#		CXXFLAGS += -stdlib=libstdc++
+#		LINKFLAGS += -stdlib=libstdc++
+#	endif
 	# boost::thread is called boost_thread-mt to mark multithreading on OS X
 	LIBRARIES += boost_thread-mt
 endif
@@ -303,6 +303,14 @@
 		BLAS_INCLUDE ?= /System/Library/Frameworks/vecLib.framework/Versions/Current/Headers/
 		LIBRARIES += cblas
 		LDFLAGS += -framework vecLib
+		# fix for vecLib path on Xcode 6
+		XCODEVERSION := $(shell xcodebuild -version | head -1 | cut -f2 -d\ | cut -f1 -d.)
+		ifeq ($(shell echo $(XCODEVERSION) \>= 6  | bc), 1)
+			FRAMEWORKS_DIR = /System/Library/Frameworks/Accelerate.framework/Versions/Current/Frameworks
+			BLAS_INCLUDE = $(FRAMEWORKS_DIR)/vecLib.framework/Headers
+			LDFLAGS += -F$(FRAMEWORKS_DIR)
+		endif
+
 	endif
 endif
 INCLUDE_DIRS += $(BLAS_INCLUDE)
@@ -310,7 +318,7 @@
 
 # Complete build flags.
 COMMON_FLAGS += $(foreach includedir,$(INCLUDE_DIRS),-I$(includedir))
-CXXFLAGS += -pthread -fPIC $(COMMON_FLAGS) $(WARNINGS)
+CXXFLAGS += $(_CXXFLAGS) $(COMMON_FLAGS) $(WARNINGS)
 NVCCFLAGS += -ccbin=$(CXX) -Xcompiler -fPIC $(COMMON_FLAGS)
 # mex may invoke an older gcc that is too liberal with -Wuninitalized
 MATLAB_CXXFLAGS := $(CXXFLAGS) -Wno-uninitialized
@@ -485,7 +493,7 @@
 # Target for extension-less symlinks to tool binaries with extension '*.bin'.
 $(TOOL_BUILD_DIR)/%: $(TOOL_BUILD_DIR)/%.bin | $(TOOL_BUILD_DIR)
 	@ $(RM) $@
-	@ ln -s $(abspath $<) $@
+	@ cd $(TOOL_BUILD_DIR); ln -s $(notdir $<) $(notdir $@)
 
 $(TOOL_BINS): %.bin : %.o $(STATIC_NAME)
 	$(CXX) $< $(STATIC_NAME) -o $@ $(LINKFLAGS) $(LDFLAGS)
