--- Makefile.orig	2021-01-28 12:52:07.000000000 +0800
+++ Makefile	2022-11-30 21:32:17.000000000 +0800
@@ -35,6 +35,7 @@
 .PHONY: byte
 byte:
 	$(OCB) src/abella.byte
+	cp -a _build/src/abella.byte abella
 
 .PHONY: gitclean
 gitclean:
@@ -49,3 +50,8 @@
 test: all
 	$(OCB) -no-links test/test.native
 	_build/test/test.native
+
+.PHONY: test_byte
+test_byte: byte
+	$(OCB) test/test.byte
+	_build/test/test.byte
