--- Makefile.in.orig	2011-12-15 18:30:18.000000000 -0600
+++ Makefile.in	2011-12-15 18:36:53.000000000 -0600
@@ -50,13 +50,13 @@
 INSTALL         = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA    = @INSTALL_DATA@
-prefix          = @prefix@
+prefix          = $(DESTDIR)@prefix@
 exec_prefix     = ${prefix}
 bindir          = @bindir@
 mandir          = @mandir@
 man1dir         = @mandir@/man1
 srcdir          = @srcdir@
-docdir          = @DOCDIR@
+docdir          = @datadir@/doc
 CPPFLAGS        = @CPPFLAGS@
 
 CFLAGS += -O2 -Wall
