diff --git src/xxdiff.pro src/xxdiff.pro
index 289adb6..bf0358d 100644
--- src/xxdiff.pro
+++ src/xxdiff.pro
@@ -117,6 +117,10 @@ irix-n32:QMAKE_CFLAGS_RELEASE += -OPT:Olimit=4000
 macx {
    QMAKE_CFLAGS -= -O2
    QMAKE_CXXFLAGS -= -O2
+   QMAKE_CFLAGS += @@CFLAGS@@
+   QMAKE_CXXFLAGS += @@CXXFLAGS@@
+   QMAKE_LFLAGS += @@LFLAGS@@
+   INCLUDEPATH += -I@@QT_INCLUDES_DIR@@
 
    # Icon used for the application bundle
    ICON = xxdiff.icns
