--- test/run_test.sh.orig	2010-02-22 05:27:16.000000000 +1100
+++ test/run_test.sh	2010-07-08 06:09:54.000000000 +1000
@@ -108,7 +108,7 @@
 [ -d "$diffs_dir" ] || mkdir "$diffs_dir"
 [ -d "$out_dir" ] || mkdir "$out_dir"
 
-if tmp_dir=`mktemp -p /tmp -d l2h_t2h_XXXXXXXX`; then
+if tmp_dir=`mktemp -d l2h_t2h_XXXXXXXX`; then
   echo "\$L2H_TMP = '$tmp_dir';" > l2h_tmp_dir.init
   echo "1;" >> l2h_tmp_dir.init
 else
@@ -182,21 +182,21 @@
       $out_dir/$dir/*_tex4ht_tex.html
   fi
   if [ $ret = 0 ]; then
-    sed -i -e 's/^texexpand.*/texexpand /' "$out_dir/$dir/$basename.2"
-    sed -i '/is no longer supported at.*line/d' "$out_dir/$dir/$basename.2"
+    sed -i '' -e 's/^texexpand.*/texexpand /' "$out_dir/$dir/$basename.2"
+    sed -i '' '/is no longer supported at.*line/d' "$out_dir/$dir/$basename.2"
     if [ "$use_latex2html" = 'yes' ]; then
-      sed -i -e 's/CONTENT="LaTeX2HTML.*/CONTENT="LaTeX2HTML">/' -e \
+      sed -i '' -e 's/CONTENT="LaTeX2HTML.*/CONTENT="LaTeX2HTML">/' -e \
        's/with LaTeX2HTML.*/with LaTeX2HTML/' "$out_dir/$dir/"*"_l2h.html"
       # "*"_images.pl" files are not guaranteed to be present
       for file in "${outdir}$dir/"*"_images.pl" "${outdir}$dir/"*"_labels.pl"; do
        if [ -f "$file" ]; then
-        sed -i -e 's/^# LaTeX2HTML.*/# LaTeX2HTML/' "$file"
+        sed -i '' -e 's/^# LaTeX2HTML.*/# LaTeX2HTML/' "$file"
        fi
       done
       for file in "${outdir}$dir/"*.htm* "${outdir}$dir/"*-l2h_cache.pm "${outdir}$dir/"*_l2h_images.pl; do
        if [ -f "$file" ]; then
        # different rounding on different computers !
-        sed -i -e 's/WIDTH="\([0-9]*\)\([0-9]\)"/WIDTH="100"/' -e 's/HEIGHT="\([0-9]*\)\([0-9]\)"/HEIGHT="\10"/' "$file"
+        sed -i '' -e 's/WIDTH="\([0-9]*\)\([0-9]\)"/WIDTH="100"/' -e 's/HEIGHT="\([0-9]*\)\([0-9]\)"/HEIGHT="\10"/' "$file"
        fi
       done
       rm -f "$out_dir/$dir/"*".aux"  "$out_dir/$dir/"*"_images.out"
--- test/run_test_all.sh.orig	2010-02-22 05:27:07.000000000 +1100
+++ test/run_test_all.sh	2010-07-08 06:16:53.000000000 +1000
@@ -134,7 +134,7 @@
   [ -d "${outdir}" ] || mkdir "${outdir}"
 done
 
-if tmp_dir=`mktemp -p /tmp -d l2h_t2h_XXXXXXXX`; then
+if tmp_dir=`mktemp -d l2h_t2h_XXXXXXXX`; then
   echo "\$L2H_TMP = '$tmp_dir';" > l2h_tmp_dir.init
   echo "1;" >> l2h_tmp_dir.init
 else
--- test/many_input_files/tex_l2h.sh.orig	2010-02-21 09:26:08.000000000 +1100
+++ test/many_input_files/tex_l2h.sh	2010-07-08 06:16:16.000000000 +1000
@@ -18,7 +18,7 @@
 echo "$basename" > $logfile
 : > $stdout_file
 
-if tmp_dir=`mktemp -p /tmp -d l2h_t2h_XXXXXXXX`; then
+if tmp_dir=`mktemp -d l2h_t2h_XXXXXXXX`; then
   echo "\$L2H_TMP = '$tmp_dir';" > l2h_tmp_dir.init
   echo "1;" >> l2h_tmp_dir.init
 else
--- test/many_input_files/tex_t4ht.sh.orig	2009-09-16 04:22:02.000000000 +1000
+++ test/many_input_files/tex_t4ht.sh	2010-07-08 06:15:28.000000000 +1000
@@ -18,7 +18,7 @@
 echo "$basename" > $logfile
 : > $stdout_file
 
-if tmp_dir=`mktemp -p /tmp -d l2h_t2h_XXXXXXXX`; then
+if tmp_dir=`mktemp -d l2h_t2h_XXXXXXXX`; then
   echo "\$L2H_TMP = '$tmp_dir';" > l2h_tmp_dir.init
   echo "1;" >> l2h_tmp_dir.init
 else
