--- configure.in.orig	2012-03-30 00:16:23.000000000 +0300
+++ configure.in	2012-07-09 02:15:10.000000000 +0300
@@ -199,23 +199,9 @@
 *darwin*)
     AC_DEFINE(HAVE_SOCKADDR_SA_LEN, 1,
         [Define if the sockaddr structure includes a sa_len member.])
-    LIBNET_CONFIG_DEFINES="-DHAVE_SOCKADDR_SA_LEN"
-
-dnl
-dnl Check to see if x86
-dnl
-
-    case "$target" in
-    i?86-*-*darwin*)
-        AC_DEFINE(LIBNET_BSDISH_OS)
-        AC_DEFINE(LIBNET_BSD_BYTE_SWAP)
-        LIBNET_CONFIG_DEFINES="$LIBNET_CONFIG_DEFINES -DLIBNET_BSDISH_OS -DLIBNET_BSD_BYTE_SWAP"
-        ;;
-
-    *)
-        ;;
-
-    esac
+    AC_DEFINE(LIBNET_BSDISH_OS)
+    AC_DEFINE(LIBNET_BSD_BYTE_SWAP)
+    LIBNET_CONFIG_DEFINES="-DHAVE_SOCKADDR_SA_LEN -DLIBNET_BSDISH_OS -DLIBNET_BSD_BYTE_SWAP"
 
     ;;
 
