--- ./setup.py	2020-07-05 09:50:12.000000000 -0400
+++ ./setup.py	2020-07-07 06:55:37.000000000 -0400
@@ -90,11 +90,11 @@
       os.makedirs(self.build_temp)
 
     bazel_argv = [
-        'bazel',
+        @BAZEL_CMD@
         'build',
+        @BAZEL_BUILD_OPTS@
         ext.bazel_target,
         '--symlink_prefix=' + os.path.join(self.build_temp, 'bazel-'),
-        '--compilation_mode=' + ('dbg' if self.debug else 'opt'),
     ]
 
     if IS_WINDOWS:
