Patch to make libfilezilla compile with MacPorts' gcc7.

--- ./lib/invoker.cpp.orig	2020-10-13 14:16:48.000000000 +0200
+++ ./lib/invoker.cpp	2023-03-25 18:38:23.000000000 +0100
@@ -25,7 +25,7 @@
 
 invoker_factory get_invoker_factory(event_loop& loop)
 {
-	return [handler = std::optional<thread_invoker>(), &loop](std::function<void()> const& cb) mutable {
+	return [handler = std::optional<thread_invoker>(), &loop=loop](std::function<void()> const& cb) mutable {
 		if (!handler) {
 			handler.emplace(loop);
 		}
--- ./lib/libfilezilla/uri.hpp.orig	2022-04-01 14:35:07.000000000 +0200
+++ ./lib/libfilezilla/uri.hpp	2023-03-25 19:00:19.000000000 +0100
@@ -21,7 +21,7 @@
 class FZ_PUBLIC_SYMBOL uri final
 {
 public:
-	uri() noexcept = default;
+	uri() = default;
 	explicit uri(std::string_view const& in);
 
 	void clear();
