--- cpp/wxPerl.osx/Makefile.orig
+++ cpp/wxPerl.osx/Makefile
@@ -30,7 +30,7 @@ $(BUILD)/wxPerl.app/Contents/Info.plist: $(BUILD)/wxPerl.app/Contents
 	cp Info.plist $@
 
 $(BUILD)/wxPerl.app/Contents/MacOS/wxPerl: $(BUILD)/wxPerl.app/Contents/MacOS
-	gcc -o $@ $(ARCH_FLAGS) main.c
+	@@CC@@ -o $@ $(ARCH_FLAGS) main.c
 
 $(BUILD)/wxPerl.app/Contents/Resources/wxPerl.rsrc: $(BUILD)/wxPerl.app/Contents/Resources
 	$(RESTOOLDIR)/Rez -d __DARWIN__ -useDF -o $(BUILD)/wxPerl.rsrc $(ARCH_FLAGS) wxPerl.r
