--- client/sguil.tk.orig	2018-04-01 07:41:21.000000000 +1000
+++ client/sguil.tk	2022-09-28 14:32:40.000000000 +1000
@@ -2099,11 +2099,11 @@
     set CONF_FILE $env(HOME)/sguil.conf
   } elseif { [file exists ./sguil.conf] } {
     set CONF_FILE ./sguil.conf
-  } elseif { [file exists /etc/sguil] &&\
-             [file isdirectory /etc/sguil] &&\
-             [file exists /etc/sguil/sguil.conf] &&\
-             [file readable /etc/sguil/sguil.conf] } {
-    set CONF_FILE /etc/sguil/sguil.conf
+  } elseif { [file exists "@PREFIX@/etc/sguil"] &&\
+             [file isdirectory "@PREFIX@/etc/sguil"] &&\
+             [file exists "@PREFIX@/etc/sguil/sguil.conf"] &&\
+             [file readable "@PREFIX@/etc/sguil/sguil.conf"] } {
+    set CONF_FILE "@PREFIX@/etc/sguil/sguil.conf"
   } else {
     puts "Couldn't determine where the sguil config file is"
     puts "Looked for $env(HOME)/sguil.conf and ./sguil.conf."
