--- Makefile.in.org	2007-12-28 17:53:12.000000000 +0100
+++ Makefile.in	2007-12-29 16:53:00.000000000 +0100
@@ -675,8 +675,11 @@
 	@echo "------------------------------------------------------------"
 	@echo "Installing include files to $(includedir)"
 	@echo "------------------------------------------------------------"
+	for dir in `cd $(srcdir)/include && find . -type d -print | grep -v \.svn`; \
+	  do mkdir -p "$(DESTDIR)$(includedir)/$$dir"; \
+	done
 	for inc in `cd $(srcdir)/include && find . -type f -print | grep -v \.svn`; \
-	  do $(INSTALL_HEADER) -D "$(srcdir)/include/$$inc" "$(includedir)/$$inc"; \
+	  do $(INSTALL_HEADER) "$(srcdir)/include/$$inc" "$(DESTDIR)$(includedir)/$$inc"; \
 	done 
 
 uninstall-local:
