From af3d036dbc3c295a05789267db2002dafc170ad3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 28 Mar 2023 00:46:39 +0700
Subject: [PATCH 1/5] Encoding.hh: define __STDC_FORMAT_MACROS

---
 src/Encoding.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Encoding.hh b/src/Encoding.hh
index 41343e9..2c1beee 100644
--- src/Encoding.hh
+++ src/Encoding.hh
@@ -1,5 +1,6 @@
 #pragma once
 
+#define __STDC_FORMAT_MACROS
 #include <stdint.h>
 #include <inttypes.h>
 
