From b3fe3a1675100660da408f76f22bddfb1af4d9d8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 28 Mar 2023 01:04:07 +0700
Subject: [PATCH 2/5] Image.hh: define __darwin_ssize_t

---
 src/Image.hh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Image.hh b/src/Image.hh
index 76448cd..8aa9417 100644
--- src/Image.hh
+++ src/Image.hh
@@ -9,6 +9,9 @@
 
 #include "Platform.hh"
 
+#ifdef __APPLE__
+typedef __darwin_ssize_t ssize_t;
+#endif
 
 // an Image represents a drawing canvas. this class is fairly simple; it
 // supports reading/writing individual pixels, drawing lines, and saving the
