https://sourceforge.net/p/dar/patches/34/
--- configure.orig	2013-08-25 17:05:49.000000000 -0500
+++ configure	2013-12-25 02:29:48.000000000 -0600
@@ -24605,84 +24605,84 @@
 echo ""
 echo "  LIBDAR parameters:"
 
-echo -n "   Zlib compression (gzip)    : "
+printf "   Zlib compression (gzip)    : "
 if [ "$local_libz" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Libbz2 compression (bzip2) : "
+printf "   Libbz2 compression (bzip2) : "
 if [ "$local_libbz2" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Liblzo2 compression (lzo)  : "
+printf "   Liblzo2 compression (lzo)  : "
 if [ "$local_liblzo2" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Strong encryption support  : "
+printf "   Strong encryption support  : "
 if [ "$local_crypto" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Extended Attributes support: "
+printf "   Extended Attributes support: "
 if [ "$local_ea_support" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Large files support (> 2GB): "
+printf "   Large files support (> 2GB): "
 if [ ! -z "$ac_cv_sys_file_offset_bits" -o ! -z "$ac_cv_sys_large_files" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   ext2fs NODUMP flag support : "
+printf "   ext2fs NODUMP flag support : "
 if [ "$local_nodump_feature" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Special allocation scheme  : "
+printf "   Special allocation scheme  : "
 if [ "$local_special_alloc" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Integer size used          : "
+printf "   Integer size used          : "
 if [ -z "$build_mode" -o \( ! "$build_mode" = "32" -a ! "$build_mode" = "64" \) ] ; then
   echo "infinint"
 else
   echo "$build_mode"
 fi
 
-echo -n "   Thread safe support        : "
+printf "   Thread safe support        : "
 if [ \( "$local_mutex_works" = "yes" -o ! "$local_special_alloc" = "yes" \) -a -z "$local_test_memory" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Furtive read mode          : "
+printf "   Furtive read mode          : "
 if [ "$local_furtive_read_mode" = "yes" ]; then
    echo "YES"
 else
    echo "NO"
 fi
 
-echo -n "   Large directory optim.     : "
+printf "   Large directory optim.     : "
 if [ "$local_fast_dir" = "yes" ] ; then
    echo "YES"
 else
@@ -24691,21 +24691,21 @@
 
 echo ""
 echo "  DAR SUITE command line programs:"
-echo -n "   Long options available : "
+printf "   Long options available : "
 if [ "$local_have_getopt_long" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Building examples      : "
+printf "   Building examples      : "
 if [ "$examples" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   Building dar_static    : "
+printf "   Building dar_static    : "
 if  [ "$build_static" = "yes" ]; then
   if [ "$static_pb" = "yes" ]; then
      echo "NO  (system does not support static linking, see note below)"
@@ -24716,14 +24716,14 @@
   echo "NO"
 fi
 
-echo -n "   using upx at install   : "
+printf "   using upx at install   : "
 if  [ "$upx" = "yes" ] ; then
   echo "YES"
 else
   echo "NO"
 fi
 
-echo -n "   building documentation : "
+printf "   building documentation : "
 if  [ "$doxygen" = "yes" ] ; then
   echo "YES"
 else
