--- cpp/src/arrow/util/io_util.cc.orig	2023-01-18 16:38:47.000000000 +0800
+++ cpp/src/arrow/util/io_util.cc	2023-04-11 07:44:03.000000000 +0800
@@ -2138,6 +2138,7 @@
 
 #elif defined(__APPLE__)
   // OSX ------------------------------------------------------
+  #ifdef MACH_TASK_BASIC_INFO
   struct mach_task_basic_info info;
   mach_msg_type_number_t infoCount = MACH_TASK_BASIC_INFO_COUNT;
   if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO, (task_info_t)&info, &infoCount) !=
@@ -2145,6 +2146,14 @@
     ARROW_LOG(WARNING) << "Can't resolve RSS value";
     return 0;
   }
+  #else
+  struct task_basic_info info;
+  mach_msg_type_number_t infoCount = TASK_BASIC_INFO_COUNT;
+  if (task_info(mach_task_self(), TASK_BASIC_INFO, (task_info_t)&info, &infoCount) != KERN_SUCCESS) {
+    ARROW_LOG(WARNING) << "Can't resolve RSS value";
+    return 0;
+  }
+  #endif
   return static_cast<int64_t>(info.resident_size);
 
 #elif defined(__linux__)
