http://trac.macports.org/ticket/44125

--- Modules/Platform/Darwin.cmake.orig	2014-06-09 08:46:52.000000000 -0700
+++ Modules/Platform/Darwin.cmake	2014-07-19 20:43:20.000000000 -0700
@@ -207,6 +207,8 @@ if(CMAKE_OSX_DEPLOYMENT_TARGET)
     set(_sdk_ver "${CMAKE_MATCH_1}")
   elseif("${_CMAKE_OSX_SYSROOT_ORIG}" MATCHES "^macosx([0-9]+\\.[0-9]+)$")
     set(_sdk_ver "${CMAKE_MATCH_1}")
+  elseif("${_CMAKE_OSX_SYSROOT_ORIG}" STREQUAL "/")
+    set(_sdk_ver "${_CURRENT_OSX_VERSION}")
   else()
     message(FATAL_ERROR
       "CMAKE_OSX_DEPLOYMENT_TARGET is '${CMAKE_OSX_DEPLOYMENT_TARGET}' "
