diff --git Modules/CMakeFindFrameworks.cmake Modules/CMakeFindFrameworks.cmake
index 8b9d803b..1da2fb7e 100644
--- Modules/CMakeFindFrameworks.cmake
+++ Modules/CMakeFindFrameworks.cmake
@@ -17,12 +17,6 @@ if(NOT CMAKE_FIND_FRAMEWORKS_INCLUDED)
   macro(CMAKE_FIND_FRAMEWORKS fwk)
     set(${fwk}_FRAMEWORKS)
     if(APPLE)
-      # 'Frameworks' directory from Brew (Apple Silicon and Intel)
-      if(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
-        set(_brew_framework_path /opt/homebrew/Frameworks)
-      else()
-        set(_brew_framework_path /usr/local/Frameworks)
-      endif()
 
       file(TO_CMAKE_PATH "$ENV{CMAKE_FRAMEWORK_PATH}" _cmff_CMAKE_FRAMEWORK_PATH)
       set(_cmff_search_paths
diff --git Modules/FindGTK2.cmake Modules/FindGTK2.cmake
index 46348768..00bfc29c 100644
--- Modules/FindGTK2.cmake
+++ Modules/FindGTK2.cmake
@@ -313,7 +313,6 @@ function(_GTK2_FIND_INCLUDE_DIR _var _hdr)
             /usr/openwin/lib
             /sw/lib
             /opt/local/lib
-            /opt/homebrew/lib
             /usr/pkg/lib
             /usr/pkg/include/glib
             $ENV{GTKMM_BASEPATH}/include
diff --git Modules/FindPython/Support.cmake Modules/FindPython/Support.cmake
index d483d7f8..7a015bc3 100644
--- Modules/FindPython/Support.cmake
+++ Modules/FindPython/Support.cmake
@@ -116,7 +116,6 @@ macro (_PYTHON_FIND_FRAMEWORKS)
                          ${_pff_CMAKE_FRAMEWORK_PATH}
                          ~/Library/Frameworks
                          /usr/local/Frameworks
-                         /opt/homebrew/Frameworks
                          ${CMAKE_SYSTEM_FRAMEWORK_PATH})
     list (REMOVE_DUPLICATES _pff_frameworks)
     foreach (_pff_implementation IN LISTS _${_PYTHON_PREFIX}_FIND_IMPLEMENTATIONS)
diff --git Modules/Platform/Darwin.cmake Modules/Platform/Darwin.cmake
index 7955c3ba..c9d71283 100644
--- Modules/Platform/Darwin.cmake
+++ Modules/Platform/Darwin.cmake
@@ -245,12 +245,6 @@ unset(_apps_paths)
 
 include(Platform/UnixPaths)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
-  list(PREPEND CMAKE_SYSTEM_PREFIX_PATH
-    /opt/homebrew # Brew on Apple Silicon
-    )
-endif()
-
 if(_CMAKE_OSX_SYSROOT_PATH)
   if(EXISTS ${_CMAKE_OSX_SYSROOT_PATH}/usr/include)
     list(INSERT CMAKE_SYSTEM_PREFIX_PATH 0 ${_CMAKE_OSX_SYSROOT_PATH}/usr)
