--- CMakeLists.txt	2012-05-11 11:39:53.000000000 +0200
+++ CMakeLists.txt	2012-05-11 11:48:15.000000000 +0200
@@ -542,7 +542,7 @@
 ENDIF(BUILD_CursesDialog)
 
 IF(BUILD_QtDialog)
-  IF(APPLE)
+  IF(FALSE)
     SET(CMAKE_BUNDLE_NAME
       "CMake ${CMake_VERSION_MAJOR}.${CMake_VERSION_MINOR}-${CMake_VERSION_PATCH}")
     SET(CMAKE_BUNDLE_LOCATION "${CMAKE_INSTALL_PREFIX}")
@@ -555,7 +555,7 @@
     ENDIF(NOT "${ENDCH}" STREQUAL "/")
     SET(CMAKE_INSTALL_PREFIX 
       "${CMAKE_INSTALL_PREFIX}${CMAKE_BUNDLE_NAME}.app/Contents")
-  ENDIF(APPLE)
+  ENDIF(FALSE)
   
   SET(QT_NEED_RPATH FALSE)
   IF(NOT "${QT_LIBRARY_DIR}" STREQUAL "/lib" AND NOT "${QT_LIBRARY_DIR}" STREQUAL "/usr/lib" AND NOT "${QT_LIBRARY_DIR}" STREQUAL "/lib64" AND NOT "${QT_LIBRARY_DIR}" STREQUAL "/usr/lib64")
--- Source/QtDialog/CMakeLists.txt	2012-05-11 11:39:53.000000000 +0200
+++ Source/QtDialog/CMakeLists.txt	2012-05-11 11:48:15.000000000 +0200
@@ -59,23 +59,23 @@
   IF(Q_WS_WIN)
     SET(SRCS ${SRCS} CMakeSetup.rc)
   ENDIF(Q_WS_WIN)
-  IF(Q_WS_MAC)
+  IF(FALSE)
     SET(SRCS ${SRCS} CMakeSetup.icns)
     SET(MACOSX_BUNDLE_ICON_FILE CMakeSetup.icns)
     SET_SOURCE_FILES_PROPERTIES(CMakeSetup.icns PROPERTIES
       MACOSX_PACKAGE_LOCATION Resources)
-  ENDIF(Q_WS_MAC)
+  ENDIF(FALSE)
 
   INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
   INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
-  ADD_EXECUTABLE(cmake-gui WIN32 MACOSX_BUNDLE ${SRCS})
+  ADD_EXECUTABLE(cmake-gui WIN32 ${SRCS})
   TARGET_LINK_LIBRARIES(cmake-gui CMakeLib ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES})
   IF(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 2.4)
-    IF(APPLE)
+    IF(FALSE)
       SET_TARGET_PROPERTIES(cmake-gui PROPERTIES
        OUTPUT_NAME ${CMAKE_BUNDLE_NAME})
-    ENDIF(APPLE)
+    ENDIF(FALSE)
     SET(CMAKE_INSTALL_DESTINATION_ARGS 
       BUNDLE DESTINATION "${CMAKE_BUNDLE_LOCATION}")
   ENDIF(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 2.4)
@@ -90,7 +90,7 @@
     INSTALL(FILES cmakecache.xml DESTINATION share/mime/packages )
   ENDIF(UNIX)
 
-  IF(APPLE)
+  IF(FALSE)
     SET(CMAKE_POSTFLIGHT_SCRIPT
       "${CMake_BINARY_DIR}/Source/QtDialog/postflight.sh")
     SET(CMAKE_POSTUPGRADE_SCRIPT
@@ -101,9 +101,9 @@
       "${CMake_BINARY_DIR}/Source/QtDialog/postupgrade.sh")
     INSTALL(CODE "execute_process(COMMAND ln -s \"../MacOS/${CMAKE_BUNDLE_NAME}\" cmake-gui
                   WORKING_DIRECTORY \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin)")
-  ENDIF(APPLE)
+  ENDIF(FALSE)
 
-  if(APPLE OR WIN32)
+  if(WIN32)
     # install rules for including 3rd party libs such as Qt
     # if a system Qt is used (e.g. installed in /usr/lib/), it will not be included in the installation
     set(fixup_exe "\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/cmake-gui${CMAKE_EXECUTABLE_SUFFIX}")
@@ -114,7 +114,7 @@
       include(\"${CMake_SOURCE_DIR}/Modules/BundleUtilities.cmake\")
       fixup_bundle(\"${fixup_exe}\" \"\" \"${QT_LIBRARY_DIR};${QT_BINARY_DIR}\")
     ")
-  endif(APPLE OR WIN32)
+  endif(WIN32)
 
   CONFIGURE_FILE("${QtDialog_SOURCE_DIR}/QtDialogCPack.cmake.in"
     "${QtDialog_BINARY_DIR}/QtDialogCPack.cmake" @ONLY)    
