--- Modules/FindQt4.cmake.orig	2013-01-03 21:15:16.000000000 -0500
+++ Modules/FindQt4.cmake	2013-01-03 21:24:55.000000000 -0500
@@ -550,7 +550,8 @@
     set(QT_MKSPECS_DIR NOTFOUND)
     find_path(QT_MKSPECS_DIR NAMES qconfig.pri
       HINTS ${qt_cross_paths} ${qt_mkspecs_dirs}
-      DOC "The location of the Qt mkspecs containing qconfig.pri")
+      DOC "The location of the Qt mkspecs containing qconfig.pri"
+      NO_DEFAULT_PATH)
   endif()
 
   if(EXISTS "${QT_MKSPECS_DIR}/qconfig.pri")
@@ -644,6 +645,7 @@
       find_path(QT_QTCORE_INCLUDE_DIR QtCore
                 HINTS ${qt_headers} ${QT_LIBRARY_DIR}
                 PATH_SUFFIXES QtCore qt4/QtCore
+                NO_DEFAULT_PATH
         )
 
       # Set QT_HEADERS_DIR based on finding QtCore header
@@ -696,7 +698,8 @@
     endforeach()
     find_path(QT_PLUGINS_DIR NAMES accessible imageformats sqldrivers codecs designer
       HINTS ${qt_cross_paths} ${qt_plugins_dir}
-      DOC "The location of the Qt plugins")
+      DOC "The location of the Qt plugins"
+      NO_DEFAULT_PATH)
   endif ()
 
   # ask qmake for the translations directory
@@ -716,8 +719,7 @@
       find_path(QT_IMPORTS_DIR NAMES Qt
         HINTS ${qt_cross_paths} ${qt_imports_dir}
         DOC "The location of the Qt imports"
-        NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH
-        NO_CMAKE_SYSTEM_PATH)
+        NO_DEFAULT_PATH)
       mark_as_advanced(QT_IMPORTS_DIR)
     endif()
   endif ()
