--- Modules/FindQt4.cmake.orig	2010-11-18 09:49:17.000000000 -0500
+++ Modules/FindQt4.cmake	2010-11-18 09:53:21.000000000 -0500
@@ -580,7 +580,8 @@
     SET(QT_MKSPECS_DIR NOTFOUND)
     FIND_PATH(QT_MKSPECS_DIR NAMES qconfig.pri
       HINTS ${qt_cross_paths} ${qt_mkspecs_dirs}
-      DOC "The location of the Qt mkspecs containing qconfig.pri")
+      DOC "The location of the Qt mkspecs containing qconfig.pri"
+      NO_DEFAULT_PATH)
   ENDIF()
 
   IF(EXISTS "${QT_MKSPECS_DIR}/qconfig.pri")
@@ -604,12 +605,12 @@
     FIND_LIBRARY(QT_QTCORE_LIBRARY_RELEASE
                  NAMES QtCore${QT_LIBINFIX} QtCore${QT_LIBINFIX}4
                  HINTS ${QT_LIBRARY_DIR_TMP}
-                 NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                 NO_DEFAULT_PATH
         )
     FIND_LIBRARY(QT_QTCORE_LIBRARY_DEBUG
                  NAMES QtCore${QT_LIBINFIX}_debug QtCore${QT_LIBINFIX}d QtCore${QT_LIBINFIX}d4
                  HINTS ${QT_LIBRARY_DIR_TMP}
-                 NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                 NO_DEFAULT_PATH
         )
 
     # try dropping a hint if trying to use Visual Studio with Qt built by mingw
@@ -663,6 +664,7 @@
                 HINTS ${qt_headers}
                 ${QT_LIBRARY_DIR}/QtCore.framework/Headers
                 PATH_SUFFIXES QtCore
+		NO_DEFAULT_PATH
         )
 
       # Set QT_HEADERS_DIR based on finding QtCore header
@@ -711,7 +713,8 @@
     endforeach(qt_cross_path)
     FIND_PATH(QT_PLUGINS_DIR NAMES accessible imageformats sqldrivers codecs designer
       HINTS ${qt_cross_paths} ${qt_plugins_dir}
-      DOC "The location of the Qt plugins")
+      DOC "The location of the Qt plugins"
+      NO_DEFAULT_PATH)
   ENDIF (QT_LIBRARY_DIR AND NOT QT_PLUGINS_DIR  OR  QT_QMAKE_CHANGED)
 
   # ask qmake for the translations directory
@@ -731,8 +734,7 @@
       FIND_PATH(QT_IMPORTS_DIR NAMES Qt
         HINTS ${qt_cross_paths} ${qt_imports_dir}
         DOC "The location of the Qt imports"
-        NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH
-        NO_CMAKE_SYSTEM_PATH)
+        NO_DEFAULT_PATH)
       mark_as_advanced(QT_IMPORTS_DIR)
     endif(qt_imports_dir)
   ENDIF (QT_LIBRARY_DIR AND NOT QT_IMPORTS_DIR  OR  QT_QMAKE_CHANGED)
