--- CMakeLists.txt	2013-10-11 18:37:11.000000000 +0200
+++ CMakeLists.txt	2013-10-11 18:39:22.000000000 +0200
@@ -528,7 +528,7 @@ if(BUILD_CursesDialog)
 endif()
 
 if(BUILD_QtDialog)
-  if(APPLE)
+  if(FALSE)
     set(CMAKE_BUNDLE_NAME
       "CMake ${CMake_VERSION_MAJOR}.${CMake_VERSION_MINOR}-${CMake_VERSION_PATCH}")
     set(CMAKE_BUNDLE_LOCATION "${CMAKE_INSTALL_PREFIX}")
--- Source/QtDialog/CMakeLists.txt	2013-10-11 18:37:12.000000000 +0200
+++ Source/QtDialog/CMakeLists.txt	2013-10-11 18:43:24.000000000 +0200
@@ -95,7 +95,7 @@ set(SRCS ${SRCS} ${UI_SRCS} ${MOC_SRCS} 
 if(WIN32)
   set(SRCS ${SRCS} CMakeSetup.rc)
 endif()
-if(APPLE)
+if(FALSE)
   set(SRCS ${SRCS} CMakeSetup.icns)
   set(MACOSX_BUNDLE_ICON_FILE CMakeSetup.icns)
   set_source_files_properties(CMakeSetup.icns PROPERTIES
@@ -104,13 +104,13 @@ endif()
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-add_executable(cmake-gui WIN32 MACOSX_BUNDLE ${SRCS})
+add_executable(cmake-gui WIN32 ${SRCS})
 target_link_libraries(cmake-gui CMakeLib ${QT_QTMAIN_LIBRARY} ${QT_LIBRARIES})
 if(Qt_BIN_DIR)
   set_property(TARGET cmake-gui PROPERTY Qt_BIN_DIR ${Qt_BIN_DIR})
 endif()
 
-if(APPLE)
+if(FALSE)
   set_target_properties(cmake-gui PROPERTIES
     OUTPUT_NAME ${CMAKE_BUNDLE_NAME})
 endif()
@@ -127,7 +127,7 @@ if(UNIX)
   install(FILES cmakecache.xml DESTINATION share/mime/packages )
 endif()
 
-if(APPLE)
+if(FALSE)
   set(CMAKE_POSTFLIGHT_SCRIPT
     "${CMake_BINARY_DIR}/Source/QtDialog/postflight.sh")
   set(CMAKE_POSTUPGRADE_SCRIPT
@@ -140,7 +140,7 @@ if(APPLE)
                 WORKING_DIRECTORY \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin)")
 endif()
 
-if(APPLE OR WIN32)
+if(WIN32)
   # install rules for including 3rd party libs such as Qt
   # if a system Qt is used (e.g. installed in /usr/lib/), it will not be included in the installation
   set(fixup_exe "\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin/cmake-gui${CMAKE_EXECUTABLE_SUFFIX}")
