--- src/plugins/script/script-repo.h.orig	2012-12-16 00:15:16.000000000 +0100
+++ src/plugins/script/script-repo.h	2012-12-16 00:15:36.000000000 +0100
@@ -57,7 +57,7 @@
 extern struct t_script_repo *scripts_repo;
 extern struct t_script_repo *last_script_repo;
 extern int script_repo_count, script_repo_count_displayed;
-struct t_hashtable *script_repo_max_length_field;
+extern struct t_hashtable *script_repo_max_length_field;
 extern char *script_repo_filter;
 
 extern int script_repo_script_valid (struct t_script_repo *script);
