--- oldMakefile	2004-08-18 02:18:45.000000000 +0200
+++ Makefile	2008-08-16 17:55:20.000000000 +0200
@@ -50,7 +50,7 @@
 # These four are typical settings for optimized code with gcc.  The
 # last two also work with icc/ecc.
 #ICFLAGS	= -g -O6 -Wall
-ICFLAGS	= -g -O6
+ICFLAGS	= -g -O6 -dynamiclib
 #ICFLAGS	= -g -O3 -Wall
 #ICFLAGS	= -g -O3
 
@@ -62,7 +62,7 @@
 #  Linux
 #
 # Gcc 2.8.1 or higher on i686.
-XCFLAGS	= -mcpu=pentiumpro -malign-double -DHAVE_IEEE_754 -DBSD
+XCFLAGS	= -DHAVE_IEEE_754
 # Gcc 3.2.2 or higher on i686.
 #XCFLAGS	= -mcpu=pentium4 -malign-double -DHAVE_IEEE_754 -DBSD
 # Icc on i686.
