--- test/unit/test_style.py
+++ test/unit/test_style.py
@@ -184,7 +184,7 @@ def test_create_child_rule_list(mocker, style_class, css):
     )
     mocked_set_child_rule = mocker.patch.object(style_class, "set_child_rule")
     assert css.create_child_rule_list(name) == style_list
-    mocked_set_child_rule.called_once_with(name, style_list)
+    mocked_set_child_rule.assert_called_once_with(name, style_list)
 
 
 @pytest.mark.parametrize(
