--- Source/JavaScriptCore/API/JSBase.h.orig	2015-10-12 21:37:10.000000000 -0700
+++ Source/JavaScriptCore/API/JSBase.h	2016-11-13 06:22:44.000000000 -0700
@@ -141,7 +141,7 @@
 
 /* Enable the Objective-C API for platforms with a modern runtime. */
 #if !defined(JSC_OBJC_API_ENABLED)
-#define JSC_OBJC_API_ENABLED (defined(__clang__) && defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1090 && !defined(__i386__))
+#define JSC_OBJC_API_ENABLED 0
 #endif
 
 #endif /* JSBase_h */
--- Source/WebKit2/Platform/CoreIPC/mac/ConnectionMac.cpp.orig	2015-10-12 21:37:14.000000000 -0700
+++ Source/WebKit2/Platform/CoreIPC/mac/ConnectionMac.cpp	2016-11-13 08:00:01.000000000 -0700
@@ -423,10 +419,6 @@
     OwnPtr<MessageDecoder> decoder = createMessageDecoder(header);
     ASSERT(decoder);
 
-#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 1090
-    decoder->setImportanceAssertion(ImportanceAssertion::create(header));
-#endif
-
     if (decoder->messageReceiverName() == "IPC" && decoder->messageName() == "InitializeConnection") {
         ASSERT(m_isServer);
         ASSERT(!m_isConnected);
