--- ddi/compddi.orig	2018-08-07 00:23:20.000000000 -0700
+++ ddi/compddi	2018-11-03 10:13:34.000000000 -0700
@@ -832,8 +832,8 @@
 
      set FORTRAN=$GMS_FORTRAN       # will be g77 or gfortran
 
-     set CC = 'gcc'
-                           set CFLAGS = "-DMACOSX"
+     set CC = '__MACPORTS_CC__'
+                           set CFLAGS = "-DMACOSX -m32"
      if ($TARGET == mac64) set CFLAGS = "-DLINUX -m64"
      set CFLAGS = "$CFLAGS -O3 -fstrict-aliasing -I./"
      set CLIBS  = "-lpthread"
@@ -847,12 +847,12 @@
            set FORT_FLAGS = '-O2'
            set FORT_LIBS  = ''
            breaksw
-        case gfortran:
+        case *gfortran*:
            set CFLAGS = "$CFLAGS -Dgetarg_=_gfortran_getarg_i4"
            set CFLAGS = "$CFLAGS -Diargc_=_gfortran_iargc"
            set NumUS=1
            set FORT_FLAGS = '-O2 -m64 -fdefault-integer-8 -std=legacy'
-           if ($TARGET == mac32) set FORT_FLAGS = '-O2 -std=legacy'
+           if ($TARGET == mac32) set FORT_FLAGS = '-O2 -m32 -std=legacy'
            set FORT_LIBS  = ''
            breaksw
         default:
