https://trac.macports.org/ticket/43480
--- Configure.orig
+++ Configure
@@ -4360,6 +4360,8 @@ dos|vms)
 *)
 	version=`echo $revision $patchlevel $subversion | \
 		 $awk '{ printf "%d.%d.%d", $1, $2, $3 }'`
+	version_short=`echo $revision $patchlevel | \
+		 $awk '{ printf "%d.%d\n", $1, $2 }'`
 	api_versionstring=`echo $api_revision $api_version $api_subversion | \
 		 $awk '{ printf "%d.%d.%d", $1, $2, $3 }'`
 	;;
@@ -7310,7 +7312,7 @@ esac
 : /opt/perl/lib/perl5... would be redundant.
 : The default "style" setting is made in installstyle.U
 case "$installstyle" in
-*lib/perl5*) set dflt privlib lib/$package/$version ;;
+*lib/perl5*) set dflt privlib lib/$package/$version_short ;;
 *)	 set dflt privlib lib/$version ;;
 esac
 eval $prefixit
@@ -7558,7 +7560,7 @@ siteprefixexp="$ansexp"
 prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 case "$sitelib" in
 '') case "$installstyle" in
-	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version ;;
+	*lib/perl5*) dflt=$siteprefix/lib/$package/site_$prog/$version_short ;;
 	*)	 dflt=$siteprefix/lib/site_$prog/$version ;;
 	esac
 	;;
@@ -7976,7 +7978,7 @@ case "$vendorprefix" in
 	'')
 		prog=`echo $package | $sed 's/-*[0-9.]*$//'`
 		case "$installstyle" in
-		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version ;;
+		*lib/perl5*) dflt=$vendorprefix/lib/$package/vendor_$prog/$version_short ;;
 		*)	     dflt=$vendorprefix/lib/vendor_$prog/$version ;;
 		esac
 		;;
