--- src/constants.h	Thu Sep  4 17:54:02 2003
+++ src/constants.h.new	Wed Dec 17 22:52:39 2003
@@ -114,18 +114,6 @@
 #define	KEY_BACKSPACE	0402
 #endif
 
-#if !sun
-#define cbreak()	crmode()			/* name change */
-#endif
-
-#if UNIXPC
-#define	erasechar()	(_tty.c_cc[VERASE])		/* equivalent */
-#define	killchar()	(_tty.c_cc[VKILL])		/* equivalent */
-#else
-#define	erasechar()	(_tty.sg_erase)			/* equivalent */
-#define	killchar()	(_tty.sg_kill)			/* equivalent */
-#endif	/* if UNIXPC */
-
 #endif	/* if !TERMINFO */
 #endif	/* ifndef __FreeBSD__ */
 
