--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -497,7 +497,7 @@
 ########### Set some global variables ###########
 IF(UNIX AND NOT WIN32)
      IF(APPLE)
-          SET(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/Stellarium.app/Contents")
+          SET(CMAKE_INSTALL_PREFIX "${MP_APPLICATIONS_DIR}/Stellarium.app/Contents")
      ELSE()
           ADD_DEFINITIONS(-DINSTALL_DATADIR="${CMAKE_INSTALL_PREFIX}/share/stellarium")
           ADD_DEFINITIONS(-DINSTALL_LOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale")
