--- src/core/StelLocationMgr.cpp.orig
+++ src/core/StelLocationMgr.cpp
@@ -158,12 +158,20 @@
 				//qDebug() << "newdata->online=" << newdata->online;
 				qDebug() << "Solution from " << newdata->satellites_used << "out of " << newdata->satellites_visible << " visible Satellites.";
 				dop_t dop=newdata->dop;
+#if GPSD_API_MAJOR_VERSION < 9
 				qDebug() << "GPSD data: Long" << newdata->fix.longitude << "Lat" << newdata->fix.latitude << "Alt" << newdata->fix.altitude;
+#else
+				qDebug() << "GPSD data: Long" << newdata->fix.longitude << "Lat" << newdata->fix.latitude << "Alt" << newdata->fix.altHAE;
+#endif
 				qDebug() << "Dilution of Precision:";
 				qDebug() << " - xdop:" << dop.xdop << "ydop:" << dop.ydop;
 				qDebug() << " - pdop:" << dop.pdop << "hdop:" << dop.hdop;
 				qDebug() << " - vdop:" << dop.vdop << "tdop:" << dop.tdop << "gdop:" << dop.gdop;
+#if GPSD_API_MAJOR_VERSION < 8
 				qDebug() << "Spherical Position Error (epe):" << newdata->epe;
+#else
+				qDebug() << "Spherical Position Error (epe):" << newdata->fix.sep;
+#endif
 			}
 			loc.longitude=newdata->fix.longitude;
 			loc.latitude=newdata->fix.latitude;
@@ -175,7 +183,11 @@
 			}
 			else
 			{
+#if GPSD_API_MAJOR_VERSION < 9
 				loc.altitude=newdata->fix.altitude;
+#else
+				loc.altitude=newdata->fix.altHAE;
+#endif
 				if (verbose)
 				{
 					qDebug() << "GPSDfix " << fixmode << ": Location" << QString("lat %1, long %2, alt %3").arg(loc.latitude).arg(loc.longitude).arg(loc.altitude);
