--- a/tests/CMakeLists.txt.orig	2021-01-17 20:01:58.000000000 +0800
+++ b/tests/CMakeLists.txt	2023-02-01 06:21:48.000000000 +0800
@@ -248,7 +248,7 @@
     endif()
   else()
     # per-test directories not generated on OS X / Darwin
-    if(NOT ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang.*")
+    if(NOT APPLE)
       link_directories(${test} PRIVATE "${ZeroMQ_SOURCE_DIR}/../lib")
     endif()
   endif()
