--- CMakeLists.txt.orig	2018-11-08 16:10:15.000000000 -0800
+++ CMakeLists.txt	2018-11-08 16:10:20.000000000 -0800
@@ -1223,7 +1223,7 @@
 	set(HAVE_ZLIB 1)
 	# Always include the "true" zlib includes first. This works around a
 	# bug in the Windows setup of GTK[23] which has a faulty zconf.h.
-	include_directories(BEFORE ${ZLIB_INCLUDE_DIRS})
+	# include_directories(BEFORE ${ZLIB_INCLUDE_DIRS})
 endif()
 if(HAVE_LIBLZ4)
 	set(HAVE_LZ4 1)
