diff --git a/epan/dissectors/CMakeLists.txt b/epan/dissectors/CMakeLists.txt
index ff8b3c6365..5a4fa25fce 100644
--- epan/dissectors/CMakeLists.txt
+++ epan/dissectors/CMakeLists.txt
@@ -1888,8 +1888,7 @@ set(ALL_DISSECTOR_SRC
 )
 
 add_executable(make-dissectors make-dissectors.c)
-# wsutil is only required for glib-compat.c
-target_link_libraries(make-dissectors ${GLIB2_LIBRARIES} wsutil)
+target_link_libraries(make-dissectors ${GLIB2_LIBRARIES})
 
 if(WIN32)
        add_dependencies(make-dissectors copy_cli_dlls)
