--- ggml/src/ggml-metal/ggml-metal-device.m.orig	2025-10-01 12:51:12
+++ ggml/src/ggml-metal/ggml-metal-device.m	2025-10-01 12:52:18
@@ -229,7 +229,10 @@
             if (path_resource) {
                 path_source = [path_resource stringByAppendingPathComponent:@"ggml-metal.metal"];
             } else {
-                path_source = [bundle pathForResource:@"ggml-metal" ofType:@"metal"];
+		path_resource = @"__PREFIX__/share/whisper/ggml";
+		GGML_METAL_LOG_INFO("%s: GGML_METAL_PATH_RESOURCES = %s\n", __func__, path_resource ? [path_resource UTF8String] : "nil");
+		path_source = [path_resource stringByAppendingPathComponent:@"ggml-metal.metal"];
+		/* path_source = [bundle pathForResource:@"ggml-metal" ofType:@"metal"];*/
             }
 
             if (path_source == nil) {
