--- src/whisper.cpp.orig	2025-04-13 16:33:17
+++ src/whisper.cpp	2025-04-13 16:53:25
@@ -22,30 +22,32 @@
 #include <climits>
 #include <codecvt>
 #include <cstdarg>
 #include <cstdio>
 #include <cstring>
 #include <fstream>
 #include <functional>
 #include <map>
 #include <mutex>
 #include <random>
 #include <regex>
 #include <set>
 #include <string>
 #include <thread>
 #include <vector>
+#include <dirent.h>
+#include <sys/types.h>
 
 // dummy
 
 #if defined(_MSC_VER)
 #pragma warning(disable: 4244 4267) // possible loss of data
 #endif
 
 #if defined(WHISPER_BIG_ENDIAN)
 template<typename T>
 static T byteswap(T value) {
     T value_swapped;
     char * source = reinterpret_cast<char *>(&value);
     char * target = reinterpret_cast<char *>(&value_swapped);
     int size = sizeof(T);
     for (int i = 0; i < size; i++) {
