--- configure.ac.orig	2012-06-15 14:52:45.000000000 +0200
+++ configure.ac	2012-07-18 22:57:38.000000000 +0200
@@ -29,7 +29,7 @@
 AC_INIT([pam_yubico], [2.12], [yubico-devel@googlegroups.com])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([1.10 foreign -Wall -Werror])
+AM_INIT_AUTOMAKE([1.10 foreign -Wall -Werror -Wno-extra-portability])
 AM_SILENT_RULES([yes])
 AM_PROG_CC_C_O
 AC_LIBTOOL_WIN32_DLL
