Fix implicit declaration of functions.
--- wvSummary.c.orig	2005-07-31 14:06:17.000000000 -0500
+++ wvSummary.c	2021-03-23 07:02:13.000000000 -0500
@@ -25,6 +25,7 @@
 #include <stdio.h>
 #include "wv.h"
 
+#include <gsf/gsf-doc-meta-data.h>
 #include <gsf/gsf-input-stdio.h>
 #include <gsf/gsf-input-memory.h>
 #include <gsf/gsf-utils.h>
--- wvparse.c.orig	2006-03-11 17:01:59.000000000 -0600
+++ wvparse.c	2021-03-23 06:56:51.000000000 -0500
@@ -101,6 +101,8 @@
 }
 
 static void tokenTreeInit (void);
+int wvOLEDecode_gsf (wvParseStruct * ps, GsfInput *path, wvStream ** mainfd, wvStream ** tablefd0,
+		     wvStream ** tablefd1, wvStream ** data, wvStream ** summary);
 
 int wvInitParser_gsf (wvParseStruct * ps, GsfInput *path)
 {
