--- ./cabal.project	2022-09-03 20:04:07.000000000 -0400
+++ ./cabal.project	2022-09-03 20:06:31.000000000 -0400
@@ -1,9 +1,18 @@
 packages: ./
 
+allow-newer: base, template-haskell, ghc-bignum
+
+constraints:
+  primitive == 0.7.4.0,
+  hashable == 1.4.0.2
+
 -- This essentially freezes the build plan for hadrian
-index-state: 2022-03-26T18:46:55Z
+index-state: 2022-08-09T12:46:55Z
 
 -- N.B. Compile with -O0 since this is not a performance-critical executable
 -- and the Cabal takes nearly twice as long to build with -O1. See #16817.
 package Cabal
   optimization: False
+
+package hadrian
+  flags: -selftest
