--- libraries/Makefile.sav	2008-11-11 13:35:39.000000000 -0500
+++ libraries/Makefile	2008-11-11 13:39:03.000000000 -0500
@@ -145,7 +145,7 @@
 	cd Cabal    && $(CABAL) build     --distpref=dist-bootstrapping
 	cd Cabal    && $(CABAL) install   --distpref=dist-bootstrapping --inplace
 	-cd hpc    && $(CABAL) clean     --distpref=dist-bootstrapping
-	cd hpc    && $(CABAL) configure --distpref=dist-bootstrapping --with-compiler=$(GHC) --with-hc-pkg=$(GHC_PKG) --package-db=$(HERE_ABS)/$@.tmp
+	cd hpc    && $(CABAL) configure --distpref=dist-bootstrapping --with-compiler=$(GHC) --with-hc-pkg=$(GHC_PKG) --with-hsc2hs=XXPATH_TO_HSC2HSXX --package-db=$(HERE_ABS)/$@.tmp
 	cd hpc    && $(CABAL) build     --distpref=dist-bootstrapping
 	cd hpc    && $(CABAL) install   --distpref=dist-bootstrapping --inplace
 	mv $@.tmp $@
