--- libraries/cabal-bin.hs.sav	2009-08-26 15:29:20.000000000 -0400
+++ libraries/cabal-bin.hs	2009-08-26 15:30:51.000000000 -0400
@@ -53,6 +53,10 @@
     unless cleaning $
         rawSystemExit verbosity ghc ["-package-conf", packageConf,
                                      "--make", "Setup",
+				     "-pgmP", "XXPATH_TO_GCCXX -E -undef -traditional",
+				     "-pgmc", "XXPATH_TO_GCCXX",
+				     "-pgml", "XXPATH_TO_GCCXX",
+				     "-pgma", "XXPATH_TO_GCCXX",
                                      "-package", "Cabal-" ++ useCabalVersion,
                                      "-o", "Setup"]
     rawSystemExit verbosity "./Setup" args
