diff -Naur ./libraries/integer-gmp/gmp/ghc.mk ./libraries/integer-gmp/gmp/ghc.mk
--- ./libraries/integer-gmp/gmp/ghc.mk	2020-09-22 14:03:12.000000000 -0400
+++ ./libraries/integer-gmp/gmp/ghc.mk	2020-09-22 14:04:35.000000000 -0400
@@ -130,7 +130,7 @@
 	#       `./configure`, not `HOSTPLATFORM`: the 'host' on which GMP will
 	#       run is the 'target' platform of the compiler we're building.
 	cd libraries/integer-gmp/gmp/gmpbuild; \
-	    CC=$(CCX) CXX=$(CCX) NM=$(NM) AR=$(AR_STAGE1) ./configure \
+	    NM=$(NM) AR=$(AR_STAGE1) ./configure \
 	          --enable-shared=no --with-pic=yes \
 	          --host=$(TARGETPLATFORM) --build=$(BUILDPLATFORM)
 	$(MAKE) -C libraries/integer-gmp/gmp/gmpbuild MAKEFLAGS=
