diff -Naur ./libraries/integer-gmp/gmp/ghc.mk ./libraries/integer-gmp/gmp/ghc.mk
--- ./libraries/integer-gmp/gmp/ghc.mk	2021-06-07 13:51:36.000000000 -0400
+++ ./libraries/integer-gmp/gmp/ghc.mk	2021-06-07 13:52:05.000000000 -0400
@@ -133,7 +133,7 @@
 	#        dependency on ncurses through gmp.  readline is only relevant
 	#        for gmp test programs. (See gmp's configure)
 	cd libraries/integer-gmp/gmp/gmpbuild; \
-	    CC=$(CCX) CXX=$(CCX) NM=$(NM) AR=$(AR_STAGE1) ./configure \
+	    NM=$(NM) AR=$(AR_STAGE1) ./configure \
 	          --enable-shared=no --with-pic=yes --with-readline=no \
 	          --host=$(TARGETPLATFORM) --build=$(BUILDPLATFORM)
 	$(MAKE) -C libraries/integer-gmp/gmp/gmpbuild MAKEFLAGS=
