--- libraries/base/configure.ac.sav	2007-12-17 16:35:12.000000000 -0500
+++ libraries/base/configure.ac	2007-12-17 16:36:07.000000000 -0500
@@ -5,6 +5,27 @@
 
 AC_CONFIG_HEADERS([include/HsBaseConfig.h])
 
+dnl--------------------------------------------------------------------
+dnl * Deal with arguments telling us gmp is somewhere odd
+dnl--------------------------------------------------------------------
+
+FP_ARG_GMP
+
+GMP_INCLUDE_DIRS=
+GMP_LIB_DIRS=
+if test "x$gmp_libraries" != "xNONE"; then
+  LDFLAGS="-L$gmp_libraries $LDFLAGS"
+  GMP_LIB_DIRS=$gmp_libraries
+fi
+if test "x$gmp_includes" != "xNONE"; then
+  CPPFLAGS="-I$gmp_includes $CPPFLAGS"
+  GMP_INCLUDE_DIRS=$gmp_includes
+fi
+AC_SUBST(GMP_INCLUDE_DIRS)
+AC_SUBST(GMP_LIB_DIRS)
+AC_CONFIG_FILES(base.buildinfo)
+dnl--------------------------------------------------------------------
+
 AC_ARG_WITH([cc],
             [C compiler],
             [CC=$withval])
