Index: alps/applications/dmft/qmc/CMakeLists.txt
===================================================================
--- alps/applications/dmft/qmc/CMakeLists.txt	(revision 7830)
+++ alps/applications/dmft/qmc/CMakeLists.txt	(working copy)
@@ -80,7 +80,7 @@
   install(TARGETS hirschfye  RUNTIME DESTINATION bin COMPONENT applications)
   install(TARGETS hybridization RUNTIME DESTINATION bin COMPONENT applications)
   install(TARGETS interaction RUNTIME DESTINATION bin COMPONENT applications)
-  install(FILES hybridization/Documentation/hybdoc.pdf DESTINATION doc)
+  install(FILES hybridization/Documentation/hybdoc.pdf DESTINATION share/alps)
 
 else(LAPACK_FOUND)
   message(STATUS "dmft will not be built since the lapack library has not been found")
