--- src/fcatomic.h.orig
+++ src/fcatomic.h
@@ -69,13 +69,21 @@ typedef LONG fc_atomic_int_t;
 
 #elif !defined(FC_NO_MT) && defined(__APPLE__)
 
+#include <AvailabilityMacros.h>
 #include <libkern/OSAtomic.h>
 
 typedef int fc_atomic_int_t;
 #define fc_atomic_int_add(AI, V)	(OSAtomicAdd32Barrier ((V), &(AI)) - (V))
 
 #define fc_atomic_ptr_get(P)		(OSMemoryBarrier (), (void *) *(P))
+
+#ifdef MAC_OS_X_VERSION_10_5
 #define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwapPtrBarrier ((void *) (O), (void *) (N), (void **) (P))
+#elif defined(__LP64__)
+#define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwap64Barrier ((void *) (O), (void *) (N), (void **) (P))
+#else
+#define fc_atomic_ptr_cmpexch(P,O,N)	OSAtomicCompareAndSwap32Barrier ((void *) (O), (void *) (N), (void **) (P))
+#endif
 
 
 #elif !defined(FC_NO_MT) && defined(HAVE_INTEL_ATOMIC_PRIMITIVES)
