Index: configure
===================================================================
--- configure	(revision 12596)
+++ configure	(working copy)
@@ -1556,7 +1556,7 @@
 
 asm_cfi_supported=false
 
-export aspp
+export as aspp
 
 if sh ./tryassemble cfi.S; then
   echo "#define ASM_CFI_SUPPORTED" >> m.h
Index: config/auto-aux/tryassemble
===================================================================
--- config/auto-aux/tryassemble	(revision 12596)
+++ config/auto-aux/tryassemble	(working copy)
@@ -5,3 +5,13 @@
 else
 $aspp -o tst $* 2> /dev/null || exit 100
 fi
+
+# test as also (if differs)
+if test "$aspp" != "$as"; then
+if test "$verbose" = yes; then
+echo "tryassemble: $as -o tst $*" >&2
+$as -o tst $* || exit 100
+else
+$as -o tst $* 2> /dev/null || exit 100
+fi
+fi
