--- DESCRIPTION	2024-03-23 03:30:06.000000000 +0800
+++ DESCRIPTION	2024-03-24 04:42:11.000000000 +0800
@@ -47,7 +47,7 @@
 Imports: doParallel, foreach, graph, lme4, mclogit, methods, nnet,
         Rcpp, Rgraphviz, rjags, stringi
 Suggests: bookdown, boot, brglm, devtools (>= 2.4.5), entropy, ggplot2,
-        gridExtra, INLA, knitr, Matrix (>= 1.6.3), MatrixModels (>=
+        gridExtra, knitr, Matrix (>= 1.6.3), MatrixModels (>=
         0.5.3), microbenchmark, moments, R.rsp, RhpcBLASctl, rmarkdown,
         testthat (>= 3.0.0)
 LinkingTo: Rcpp, RcppArmadillo
