diff --git a/src/graph_tool/draw/gtk_draw.py b/src/graph_tool/draw/gtk_draw.py
index 9f60075d..cd3edf97 100644
--- src/graph_tool/draw/gtk_draw.py
+++ src/graph_tool/draw/gtk_draw.py
@@ -1182,7 +1182,7 @@ _window_list = []
 
 def interactive_window(g, pos=None, vprops=None, eprops=None, vorder=None,
                        eorder=None, nodesfirst=False, geometry=(500, 400),
-                       update_layout=True, async=False, no_main=False, **kwargs):
+                       update_layout=True, sync=True, no_main=False, **kwargs):
     r"""
     Display an interactive GTK+ window containing the given graph.
 
@@ -1211,8 +1211,8 @@ def interactive_window(g, pos=None, vprops=None, eprops=None, vorder=None,
         Window geometry.
     update_layout : bool (optional, default: ``True``)
         If ``True``, the layout will be updated dynamically.
-    async : bool (optional, default: ``False``)
-        If ``True``, run asynchronously. (Requires :mod:`IPython`)
+    sync : bool (optional, default: ``True``)
+        If ``False``, run asynchronously. (Requires :mod:`IPython`)
     no_main : bool (optional, default: ``False``)
         If ``True``, the GTK+ main loop will not be called.
     **kwargs
@@ -1244,7 +1244,7 @@ def interactive_window(g, pos=None, vprops=None, eprops=None, vorder=None,
     win.show_all()
     _window_list.append(win)
     if not no_main:
-        if async:
+        if not sync:
             # just a placeholder for a proper main loop integration with gtk3 when
             # ipython implements it
             import IPython.lib.inputhook
