Index: kinit/CMakeLists.txt
===================================================================
--- kinit/CMakeLists.txt	(revision 900481)
+++ kinit/CMakeLists.txt	(working copy)
@@ -34,7 +34,7 @@
   set(kdeinit_SRCS kinit_win.cpp  )
   set(kdeinit_LIBS kparts psapi)
 else (WIN32)
-  set(kdeinit_SRCS kinit.cpp proctitle.cpp )
+  set(kdeinit_SRCS kinit.cpp proctitle.cpp ../kdecore/kernel/kkernel_mac.cpp)
   set(kdeinit_LIBS kparts)
 endif (WIN32)
 
Index: kdeui/CMakeLists.txt
===================================================================
--- kdeui/CMakeLists.txt	(revision 900481)
+++ kdeui/CMakeLists.txt	(working copy)
@@ -239,7 +239,7 @@
                                          dialogs/kcupsoptionspageswidget_p.cpp )
 endif (Q_WS_X11)
 if (Q_WS_MAC)
-    set(kdeui_LIB_SRCS ${kdeui_LIB_SRCS} util/kkeyserver_mac.cpp )
+    set(kdeui_LIB_SRCS ${kdeui_LIB_SRCS} util/kkeyserver_mac.cpp ../kdecore/kernel/kkernel_mac.cpp)
 endif (Q_WS_MAC)
 if (Q_WS_WIN)
     set(kdeui_LIB_SRCS ${kdeui_LIB_SRCS} kernel/kapplication_win.cpp )
