diff --git a/Makefile b/Makefile
index 866264595..e51c9d59b 100644
--- Makefile
+++ Makefile
@@ -18,9 +18,6 @@ include scripts/lib.mk
 
 CFLAGS += -D_FILE_OFFSET_BITS=64
 
-FFMPEG_CFLAGS += $(shell pkg-config --cflags libswresample)
-FFMPEG_LIBS += $(shell pkg-config --libs libswresample)
-
 CMUS_LIBS = $(PTHREAD_LIBS) $(NCURSES_LIBS) $(ICONV_LIBS) $(DL_LIBS) $(DISCID_LIBS) \
 			-lm $(COMPAT_LIBS) $(LIBSYSTEMD_LIBS)
 
diff --git a/configure b/configure
index 49de7b31f..cd663d61d 100755
--- configure
+++ configure
@@ -471,21 +471,14 @@ check_aac()
 
 check_ffmpeg()
 {
-	HAVE_FFMPEG_AVCODEC_H=y
-	pkg_config FFMPEG "libavformat libavcodec" || return $?
-	if check_header "libavcodec/avcodec.h" $FFMPEG_CFLAGS
-	then
-		HAVE_FFMPEG_AVCODEC_H=n
-	else
-		check_header "ffmpeg/avcodec.h" $FFMPEG_CFLAGS || return $?
-	fi
+	pkg_config FFMPEG "libavformat libavcodec libswresample" || return $?
+	# check the existence of specific headers since they've been renamed before
+	check_header "libavformat/avformat.h" $FFMPEG_CFLAGS || return $?
+	check_header "libavcodec/avcodec.h" $FFMPEG_CFLAGS || return $?
+	check_header "libswresample/swresample.h" $FFMPEG_CFLAGS || return $?
 	# ffmpeg api changes so frequently that it is best to compile the module
 	libs="$LDDLFLAGS $FFMPEG_LIBS"
 	cflags="$SOFLAGS $FFMPEG_CFLAGS"
-	if test "$HAVE_FFMPEG_AVCODEC_H" = y
-	then
-		cflags="$cflags -DHAVE_FFMPEG_AVCODEC_H"
-	fi
 	topdir=`dirname "$0"`
 	ffmpeg_code=`cat "$topdir"/ip/ffmpeg.c | sed 's/\\\n//g'`
 	msg_checking "for successful build of ffmpeg.c"
