--- configure.orig	2017-01-20 03:13:17.000000000 -0700
+++ configure	2018-08-25 07:59:16.000000000 -0700
@@ -28373,7 +28373,7 @@
   P4EST_SC_MK_USE=yes
   P4EST_SC_MK_INCLUDE="include $P4EST_SC_ETC/Makefile.sc.mk"
   P4EST_SC_CPPFLAGS="\$(SC_CPPFLAGS)"
-  P4EST_SC_LDADD="$P4EST_SC_DIR/lib/libsc.la"
+  P4EST_SC_LDADD="$P4EST_SC_DIR/lib/libsc.dylib"
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: Building with source of package sc" >&5
 $as_echo "$as_me: Building with source of package sc" >&6;}
@@ -28395,7 +28395,7 @@
   P4EST_SC_AMFLAGS="-I \$(top_srcdir)/$P4EST_SC_SOURCE/config"
   P4EST_SC_MK_INCLUDE="include \${p4est_sysconfdir}/Makefile.sc.mk"
   P4EST_SC_CPPFLAGS="-I$P4EST_SC_SOURCE/src -I\$(top_srcdir)/$P4EST_SC_SOURCE/src"
-  P4EST_SC_LDADD="$P4EST_SC_SOURCE/src/libsc.la"
+  P4EST_SC_LDADD="$P4EST_SC_SOURCE/src/libsc.dylib"
 fi
 
 
