--- plugins/gpgme-vala/vapi/gpg-error.vapi	2024-07-01 05:01:20
+++ plugins/gpgme-vala/vapi/gpg-error.vapi	2024-07-01 16:22:11
@@ -442,7 +442,7 @@
 		public ErrorCode code { [CCode (cname = "gpg_err_code")] get; }
 		public ErrorSource source { [CCode (cname = "gpg_err_source")] get; }
 
-		[CCode (cname = "gpg_strerror")]
+		[CCode (cname = "gpgme_strerror")]
 		public unowned string to_string ();
 
 		[CCode (cname = "gpg_strsource")]
\ No newline at end of file
