--- src/gnc-module/gnc-module.c	2011-10-11 11:47:40.706200237 -0400
+++ src/gnc-module/gnc-module.c	2011-10-11 11:48:21.254201370 -0400
@@ -323,7 +323,7 @@ gnc_module_get_info(const char * fullpat
     info->module_age         = *(int *)age;
     info->module_revision    = *(int *)revision;
 
-
+    g_module_make_resident(gmodule);
 get_info_close:
     /*   g_debug("(init) closing '%s'\n", fullpath); */
     g_module_close(gmodule);
