diff -ur src/osgAnimation.i.orig src/osgAnimation.i
--- src/osgAnimation.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgAnimation.i	2009-12-27 16:46:38.000000000 +0100
@@ -18,6 +18,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 %{
 #include <osgAnimation/Animation>
diff -ur src/osgDB.i.orig src/osgDB.i
--- src/osgDB.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgDB.i	2009-12-27 16:47:14.000000000 +0100
@@ -20,6 +20,7 @@
 
 %include osg_header.i
 %import osg.i
+%pythoncode {import osg}
 
 /* instantiate the required template specializations */
 %template() std::vector<std::string>;    
diff -ur src/osgFX.i.orig src/osgFX.i
--- src/osgFX.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgFX.i	2009-12-27 16:47:40.000000000 +0100
@@ -18,6 +18,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 
 %{
diff -ur src/osgGA.i.orig src/osgGA.i
--- src/osgGA.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgGA.i	2009-12-27 16:48:41.000000000 +0100
@@ -22,6 +22,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 %include osgGA_header.i
 
diff -ur src/osgManipulator.i.orig src/osgManipulator.i
--- src/osgManipulator.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgManipulator.i	2009-12-27 16:49:08.000000000 +0100
@@ -17,6 +17,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 
 %{
diff -ur src/osgSim.i.orig src/osgSim.i
--- src/osgSim.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgSim.i	2009-12-27 16:49:20.000000000 +0100
@@ -17,6 +17,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 
 
diff -ur src/osgText.i.orig src/osgText.i
--- src/osgText.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgText.i	2009-12-27 16:50:32.000000000 +0100
@@ -20,6 +20,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 
 %{
diff -ur src/osgUtil.i.orig src/osgUtil.i
--- src/osgUtil.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgUtil.i	2009-12-27 16:50:46.000000000 +0100
@@ -28,6 +28,7 @@
 
 /* import stuff from OpenSceneGraph */
 %import osg.i
+%pythoncode {import osg}
 
 
 %{
diff -ur src/osgViewer.i.orig src/osgViewer.i
--- src/osgViewer.i.orig	2009-12-26 02:04:30.000000000 +0100
+++ src/osgViewer.i	2009-12-27 17:05:09.000000000 +0100
@@ -19,6 +19,10 @@
 /* import stuff from OpenSceneGraph */
 %import osg.i
 %import osgGA.i
+%pythoncode {
+    import osg
+    import osgGA
+}
 
 
 %include osgViewer_header.i
