--- src/osg.i.orig	2009-12-21 18:16:02.000000000 +0100
+++ src/osg.i	2009-12-21 22:01:15.000000000 +0100
@@ -462,8 +463,12 @@
 %ignore osg::GraphicsContext::getCurrentOperation;
 
 
+%ignore osg::Texture::getTextureObjectManager;
 %ignore osg::Texture::getExtensions;
 %ignore osg::Texture::setExtensions;
+%ignore osg::Texture::s_numberTextureReusedLastInLastFrame;
+%ignore osg::Texture::s_numberNewTextureInLastFrame;
+%ignore osg::Texture::s_numberDeletedTextureInLastFrame;
 
 %ignore osg::Texture1D::setSubloadCallback;
 %ignore osg::Texture1D::getSubloadCallback;
@@ -500,8 +505,11 @@
 %ignore osg::BlendEquation::setExtensions;
 %ignore osg::BlendColor::getExtensions;
 %ignore osg::BlendColor::setExtensions;
-%ignore osg::BufferObject::getExtensions;
-%ignore osg::BufferObject::setExtensions;
+
+%ignore osg::GLBufferObject::_extensions;
+%ignore osg::GLBufferObject::Extensions;
+%ignore osg::GLBufferObject::getExtensions;
+%ignore osg::GLBufferObject::setExtensions;
 
 %ignore osg::Point::Extensions;
 %ignore osg::Point::getExtensions;
@@ -679,7 +687,7 @@
     if (PyString_Check($input)) {
         int len;
         char *buf;
-        PyString_AsStringAndSize($input, &buf, &len);
+        PyString_AsStringAndSize($input, &buf, (Py_ssize_t *)&len);
         $1 = (unsigned char *)malloc(len);
         memcpy($1, buf, len);
     } else {
@@ -733,7 +742,14 @@
 %extend osg::Viewport {	value_type getHeight() {return $self->height();}};
 %include osg/Viewport
 
+%ignore osg::Shader::Extensions;
+%ignore osg::Shader::getExtensions;
+%ignore osg::Shader::setExtensions;
 %include osg/Shader
+
+%ignore osg::Program::Extensions;
+%ignore osg::Program::getExtensions;
+%ignore osg::Program::setExtensions;
 %include osg/Program
 
 %extend osg::Program { 
@@ -940,6 +956,9 @@
 %include osg/ArgumentParser
 %include osg/Array
 
+%ignore osg::PrimitiveSet::Extensions;
+%ignore osg::PrimitiveSet::getExtensions;
+%ignore osg::PrimitiveSet::setExtensions;
 %include osg/PrimitiveSet
 
 
@@ -1020,6 +1038,9 @@
 %ignore osg::Uniform::getUpdateCallback;
 %ignore osg::Uniform::setEventCallback;
 %ignore osg::Uniform::getEventCallback;
+%ignore osg::Uniform::Extensions;
+%ignore osg::Uniform::getExtensions;
+%ignore osg::Uniform::setExtensions;
 
 
 %include osg/Uniform
