--- CMakeLists.txt.orig	2009-06-29 17:26:22.000000000 +0200
+++ CMakeLists.txt	2010-01-29 22:48:58.000000000 +0100
@@ -28,13 +28,13 @@
 
 include(UseQt4)
 
-if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 5   )
+if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} GREATER 4   )
 	MESSAGE (FATAL_ERROR 
-			"Qt version 4.5 required.\n"
+			"Qt version >= 4.5 required.\n"
 			"Qt version installed is: ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}\n"
 			"Please install required version.\n"
 		)
-endif( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 5   )
+endif( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} GREATER 4   )
 
 
 
