--- CMakeLists.txt.orig	2010-10-23 08:58:23.000000000 +0200
+++ CMakeLists.txt	2010-10-23 11:39:03.000000000 +0200
@@ -28,9 +28,9 @@
 
 include(UseQt4)
 
-if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 6   )
+if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} GREATER 5   )
 	MESSAGE (FATAL_ERROR 
-			"Qt version 4.6 required.\n"
+			"Qt version greater than or equal to 4.6 required.\n"
 			"Qt version installed is: ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}\n"
 			"Please install required version.\n"
 		)
