--- CMakeLists.txt.orig	2010-11-12 08:21:53.000000000 +0100
+++ CMakeLists.txt	2010-11-12 08:24:17.000000000 +0100
@@ -28,13 +28,13 @@
 
 include(UseQt4)
 
-if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 6   )
+if( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} GREATER 5   )
 	MESSAGE (FATAL_ERROR 
-			"Qt version 4.6 required.\n"
+			"Qt version greater than or equal to 4.6 required.\n"
 			"Qt version installed is: ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}\n"
 			"Please install required version.\n"
 		)
-endif( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} EQUAL 6   )
+endif( NOT ${QT_VERSION_MAJOR} EQUAL 4 OR NOT ${QT_VERSION_MINOR} GREATER 5   )
 
 add_subdirectory (qtoctave_pkg/src)
 add_subdirectory (xmlwidget/qt4)
