--- qtoctave/src/table.cpp.orig	2012-05-18 14:50:15.000000000 +0200
+++ qtoctave/src/table.cpp	2012-05-18 14:51:14.000000000 +0200
@@ -258,7 +258,7 @@
 		QStringList numbers=list.at(3).trimmed().split(" ", QString::SkipEmptyParts);
 		
 		long thisValue=numbers.at(0).toLong();
-		if( ((long)this)!=thisValue )
+		if( (((long)this) & 0xFFFFFFFF)!=thisValue)
                 {
 		   qDebug("PROBLEM: value mismatch; %ld != %ld\n", (long)this, thisValue);
 		   return;
