--- lib/libutil/asprintf.c.orig	2015-12-30 15:22:14
+++ lib/libutil/asprintf.c	2025-08-18 06:49:53
@@ -1,4 +1,3 @@
-#define _XOPEN_SOURCE 600  /* Make sure strdup() is in <string.h> */
 #define _GNU_SOURCE  /* But only when HAVE_ASPRINTF */
 #include <stdarg.h>
 #include <stdio.h>
--- lib/abyss/src/server.c.orig	2023-12-27 06:30:28
+++ lib/abyss/src/server.c	2025-08-18 06:56:32
@@ -1,7 +1,6 @@
 /* Copyright information is at end of file */
 
 #define _DEFAULT_SOURCE /* New name for SVID & BSD source defines */
-#define _XOPEN_SOURCE 600  /* Make sure strdup() is in <string.h> */
 #define _BSD_SOURCE  /* Make sure setgroups()is in <grp.h> */
 
 #include "xmlrpc_config.h"
