--- test/cpp/server_abyss.cpp.orig	2015-12-30 15:22:14
+++ test/cpp/server_abyss.cpp	2025-08-18 07:21:02
@@ -85,7 +85,7 @@ class boundSocket { (public)
         sockAddr.sin_port   = htons(portNumber);
         sockAddr.sin_addr.s_addr = 0;
 
-        rc = bind(this->fd, (struct sockaddr *)&sockAddr, sizeof(sockAddr));
+        rc = ::bind(this->fd, (struct sockaddr *)&sockAddr, sizeof(sockAddr));
         
         if (rc != 0) {
             closesock(this->fd);
