--- config.mk.in.orig	2011-03-26 03:58:35.000000000 +1100
+++ config.mk.in	2011-05-06 10:58:03.000000000 +1000
@@ -276,7 +276,7 @@
   shlibfn = $(1:%=%.$(MAJ).$(MIN).$(SHLIB_SUFFIX))
   shliblefn = $(1:%=%.$(SHLIB_SUFFIX))
   LDFLAGS_SHLIB = -dynamiclib -undefined suppress -single_module \
-	-flat_namespace $(SHLIB_CLIB)
+	-flat_namespace -install_name $(LIBINST_DIR)/$(SONAME) $(SHLIB_CLIB)
 endif
 
 ifeq ($(patsubst irix%,irix,$(HOST_OS)),irix)
