diff -ur xmlrpc-c-1.51.07/lib/abyss/src/conf.c xmlrpc-c-1.51.07_ed/lib/abyss/src/conf.c
--- xmlrpc-c-1.51.07/lib/abyss/src/conf.c	2018-06-26 04:47:58.000000000 +0300
+++ xmlrpc-c-1.51.07_ed/lib/abyss/src/conf.c	2021-04-28 01:02:04.000000000 +0300
@@ -32,7 +32,6 @@
 **
 ******************************************************************************/
 
-#define _XOPEN_SOURCE 600   /* For strdup() */
 #define _BSD_SOURCE  /* For xmlrpc_strcaseeq */
 
 #include "xmlrpc_config.h"
diff -ur xmlrpc-c-1.51.07/lib/abyss/src/http.c xmlrpc-c-1.51.07_ed/lib/abyss/src/http.c
--- xmlrpc-c-1.51.07/lib/abyss/src/http.c	2019-03-29 06:33:33.000000000 +0300
+++ xmlrpc-c-1.51.07_ed/lib/abyss/src/http.c	2021-04-28 01:03:37.000000000 +0300
@@ -1,6 +1,5 @@
 /* Copyright information is at the end of the file */
 
-#define _XOPEN_SOURCE 600   /* For strdup() */
 #define _BSD_SOURCE   /* For xmlrpc_strcaseeq() */
 
 #include <ctype.h>
diff -ur xmlrpc-c-1.51.07/lib/abyss/src/server.c xmlrpc-c-1.51.07_ed/lib/abyss/src/server.c
--- xmlrpc-c-1.51.07/lib/abyss/src/server.c	2019-03-29 06:33:33.000000000 +0300
+++ xmlrpc-c-1.51.07_ed/lib/abyss/src/server.c	2021-04-28 01:03:58.000000000 +0300
@@ -1,6 +1,5 @@
 /* Copyright information is at end of file */
 
-#define _XOPEN_SOURCE 600  /* Make sure strdup() is in <string.h> */
 #define _BSD_SOURCE  /* Make sure setgroups()is in <grp.h> */
 
 #include "xmlrpc_config.h"
diff -ur xmlrpc-c-1.51.07/lib/abyss/src/sessionReadRequest.c xmlrpc-c-1.51.07_ed/lib/abyss/src/sessionReadRequest.c
--- xmlrpc-c-1.51.07/lib/abyss/src/sessionReadRequest.c	2021-02-21 01:18:09.000000000 +0300
+++ xmlrpc-c-1.51.07_ed/lib/abyss/src/sessionReadRequest.c	2021-04-28 01:04:19.000000000 +0300
@@ -1,4 +1,3 @@
-#define _XOPEN_SOURCE 600   /* For strdup() */
 #define _BSD_SOURCE   /* For xmlrpc_strcaseeq() */
 
 #include <ctype.h>
diff -ur xmlrpc-c-1.51.07/lib/libutil/asprintf.c xmlrpc-c-1.51.07_ed/lib/libutil/asprintf.c
--- xmlrpc-c-1.51.07/lib/libutil/asprintf.c	2015-12-30 07:22:14.000000000 +0300
+++ xmlrpc-c-1.51.07_ed/lib/libutil/asprintf.c	2021-04-28 01:07:52.000000000 +0300
@@ -1,4 +1,3 @@
-#define _XOPEN_SOURCE 600  /* Make sure strdup() is in <string.h> */
 #define _GNU_SOURCE  /* But only when HAVE_ASPRINTF */
 #include <stdarg.h>
 #include <stdio.h>
