--- a/src/managesieve-login/managesieve-proxy.c	2012-02-12 16:48:38.000000000 -0800
+++ b/src/managesieve-login/managesieve-proxy.c	2012-02-12 16:48:45.000000000 -0800
@@ -325,13 +325,13 @@
 
 			(void)client_skip_line(msieve_client);
 			client_proxy_finish_destroy_client(client);
-			
+
 			return 1;
-		} 
-		
+		}
+
 		/* Login failed */
 
-		if ( client->set->verbose_auth ) {
+		if ( client->set->auth_verbose ) {
 			const char *log_line = line;
 
 			if (strncasecmp(log_line, "NO ", 3) == 0)
