--- enthought/traits/ui/wx/color_trait.py.orig	2006-08-17 20:25:30.000000000 -0700
+++ enthought/traits/ui/wx/color_trait.py	2007-04-25 17:24:12.000000000 -0700
@@ -34,6 +34,8 @@
     if type( value ) is int:
         num = int( value )
         return wx.Colour( num / 0x10000, (num / 0x100) & 0xFF, num & 0xFF )
+    if isinstance( value, str ) and value in standard_colors:
+        return wx.NamedColour(value)
     raise TraitError
 
 convert_to_color.info = ('a wx.Colour instance, an integer which in hex is of '
