diff -Naur configure configure
--- configure	2008-11-25 20:28:41.000000000 -0800
+++ configure	2009-01-12 07:53:52.000000000 -0800
@@ -5,7 +5,7 @@
 [ -n "$VERSION" ] && echo "Running configure for SHOGUN version $VERSION"
 
 # some hopefully reasonable defaults
-MAKEDEPEND="gcc -MM"
+MAKEDEPEND="/opt/local/bin/gcc-mp-4.3 -MM"
 MAKEDEPENDOPTS=""
 SRCDIR="`dirname $0`"
 EXT_SRC_HEADER=h
@@ -63,7 +63,7 @@
 _hmmdebug=no
 _pathdebug=no
 _profile=no
-_prefix="/usr/local"
+_prefix="/opt/local"
 _destdir="/"
 _bindir=""
 _datadir=""
@@ -385,8 +385,8 @@
   if test -f "$TMPO"
   then
 	  echo >> "$TMPLOG"
-	  echo "ldd $TMPO" >> "$TMPLOG"
-	  ( ldd "$TMPO" ) >> "$TMPLOG" 2>&1
+	  echo "otool -L $TMPO" >> "$TMPLOG"
+	  ( otool -L "$TMPO" ) >> "$TMPLOG" 2>&1
   else
 	  echo "no object file created" >> "$TMPLOG"
   fi
@@ -405,8 +405,8 @@
   if test -f "$TMPO"
   then
 	  echo >> "$TMPLOG"
-	  echo "ldd $TMPO" >> "$TMPLOG"
-	  ( ldd "$TMPO" ) >> "$TMPLOG" 2>&1
+	  echo "otool -L $TMPO" >> "$TMPLOG"
+	  ( otool -L "$TMPO" ) >> "$TMPLOG" 2>&1
   else
 	  echo "no object file created" >> "$TMPLOG"
   fi
@@ -2391,7 +2391,7 @@
 			LINKFLAGS="-bundle -Wl,-flat_namespace -L`octave-config -p OCTLIBDIR` -loctinterp -loctave -lcruft $LINKFLAGS"
 		else
 			OCTLIBDIR=`octave-config -p OCTLIBDIR`
-			LINKFLAGS="-shared -L $OCTLIBDIR -loctinterp -loctave -lcruft $LINKFLAGS"
+			LINKFLAGS="-shared -L$OCTLIBDIR -loctinterp -loctave -lcruft $LINKFLAGS"
 		fi
 		TARGET="sg.oct"
 		INCLUDES="$INCLUDES -I`octave-config -p OCTINCLUDEDIR` -I`octave-config -p INCLUDEDIR`"
@@ -2667,7 +2667,7 @@
 }
 EOF
 		echocheck "lpsolve Developer Files"
-		LPSOLVELINKFLAGS="-L /usr/lib/lp_solve -llpsolve55"
+		LPSOLVELINKFLAGS="-L/opt/local/lib/ -llpsolve55"
 		if cc_check $LPSOLVELINKFLAGS
 		then 
 			echores "yes"
