From 5140e4f2b9f85292954ada4662ac351935a179a4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 29 Oct 2024 11:12:26 +0800
Subject: [PATCH] socket.c: define AI_NUMERICSERV if undefined

---
 src/socket.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/socket.c src/socket.c
index 100562c..1518954 100644
--- src/socket.c
+++ src/socket.c
@@ -83,6 +83,10 @@ static int wsa_init = 0;
 #define ETIMEDOUT 138
 #endif
 
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
 static int verbose = 0;
 
 void socket_set_verbose(int level)
