--- setup.py.orig	2020-05-30 18:50:56.000000000 -0500
+++ setup.py	2020-08-02 10:19:30.000000000 -0500
@@ -82,7 +82,10 @@
 EXTENSIONS = {
     'shared': dict(),
     'imcd': dict(sources=['imagecodecs/imcd.c']),
-    'aec': dict(libraries=['aec']),
+    'aec': dict(libraries=['aec'], 
+        include_dirs=['%PREFIX%/lib/libaec/include'],
+        library_dirs=['%PREFIX%/lib/libaec/lib']
+        ),
     'bitshuffle': dict(
         sources=[
             'bitshuffle-0.3.5/bitshuffle_core.c',
@@ -93,15 +96,18 @@
     'blosc': dict(libraries=['blosc']),
     'brotli': dict(libraries=['brotlienc', 'brotlidec', 'brotlicommon']),
     'bz2': dict(libraries=['bz2']),
-    'gif': dict(libraries=['gif']),
+    'gif': dict(libraries=['gif'], 
+    	include_dirs=['%PREFIX%/include/giflib5']
+    	),
     'jpeg2k': dict(
         sources=['imagecodecs/opj_color.c'],
+        include_dirs=['%PREFIX%/include/openjpeg-2.3'], 
         libraries=['openjp2', 'lcms2'],
         define_macros=[('OPJ_HAVE_LIBLCMS2', 1)]
         ),
     'jpeg8': dict(
         libraries=['jpeg'],
-        cython_compile_env={'HAVE_LIBJPEG_TURBO': True}
+        cython_compile_env={'HAVE_LIBJPEG_TURBO': False}
         ),
     'jpeg12': dict(
         libraries=['jpeg12'],
@@ -111,10 +117,10 @@
     'jpegls': dict(libraries=['charls']),
     'jpegsof3': dict(sources=['imagecodecs/jpegsof3.cpp']),
     'jpegxl': dict(libraries=['brunslidec-c', 'brunslienc-c']),
-    'jpegxr': dict(
-        libraries=['jpegxr', 'jxrglue'],
-        define_macros=[('__ANSI__', 1)] if sys.platform != 'win32' else []
-        ),
+    #'jpegxr': dict(
+    #    libraries=['jpegxr', 'jxrglue'],
+    #    define_macros=[('__ANSI__', 1)] if sys.platform != 'win32' else []
+    #    ),
     'lerc': dict(libraries=['lerc']),
     'lz4': dict(libraries=['lz4']),
     'lzf': dict(
