--- setup.py.orig	2021-01-11 18:31:44.000000000 -0600
+++ setup.py	2021-01-11 18:33:58.000000000 -0600
@@ -381,7 +381,7 @@
     del EXTENSIONS['lerc']
     del EXTENSIONS['lz4f']
     del EXTENSIONS['zfp']
-    del EXTENSIONS['zopfli']  # zopfli/zopfli.h does not exist
+#    del EXTENSIONS['zopfli']  # zopfli/zopfli.h does not exist
 
     EXTENSIONS['aec']['library_dirs'] = ['/opt/local/lib/libaec/lib']
     EXTENSIONS['aec']['include_dirs'] = ['/opt/local/lib/libaec/include']
