--- Makefile	Sun Nov 17 10:52:53 2002
+++ Makefile	Tue Jul  1 23:22:29 2003
@@ -15,14 +15,13 @@
 endif
 
 TREETOP = .
-PILOT_LINK_DIR = $(TREETOP)/pilot-link-0.11.7
 
 CFLAGS = -Wall -g -DMALSYNC -I$(TREETOP)/mal/common \
-  -I$(TREETOP)/mal/client/common -I$(PILOT_LINK_DIR)/include \
+  -I$(TREETOP)/mal/client/common -I$(PREFIX)/include \
   -I$(TREETOP)/mal/client/unix
 
 ifeq ($(PLAT),Darwin)
-CFLAGS += --traditional-cpp -DDARWIN -D__unix__ -D__FreeBSD__ -D_HPUX_SOURCE
+CFLAGS += -no-cpp-precomp -DDARWIN -D__unix__ -D__FreeBSD__ -D_HPUX_SOURCE
 endif
 
 SRC = malsync.c  \
@@ -73,7 +72,7 @@
 
 $(TARGET): $(OBJS)
 	$(CC) $(DYNLINK) $(CFLAGS) -o $@ $(OBJS) \
-          ${PILOT_LINK_DIR}/libpisock/.libs/libpisock.a $(LINKLIBS)
+          -L$(PREFIX)/lib -lpisock $(LINKLIBS)
 clean:
 	rm -f $(OBJS) $(TARGET)
 
--- mal/client/common/AGCommandProcessor.c	Sun May 14 14:17:35 2000
+++ mal/client/common/AGCommandProcessor.c	Tue Jul  1 23:10:28 2003
@@ -907,9 +907,9 @@
     case AG_DEVICEINFO_CMD:      
     case AG_NEWIDS_CMD:      
     default:
-        #ifdef DEBUG_COMMANDS
+#ifdef DEBUG_COMMANDS
         printf("\tThe server should have NEVER sent a %s command.\n", name);
-        #endif
+#endif
         rc = AGCLIENT_ERR; 
         break;
     }
--- mal/common/AGBufferedNet.c	Sun May 14 14:17:36 2000
+++ mal/common/AGBufferedNet.c	Tue Jul  1 23:09:47 2003
@@ -26,19 +26,19 @@
 
 #ifdef __unix__
 
-    #include <errno.h>
+#include <errno.h>
     
 #endif // __unix__
 
 /*************************************************************************************/
 #ifdef __palmos__
 
-    #define BUFSIZE 512
-    #define ALMOST_INFINITE_TIMEOUT (10 * sysTicksPerSecond)
+#define BUFSIZE 512
+#define ALMOST_INFINITE_TIMEOUT (10 * sysTicksPerSecond)
 
 #else
 
-    #define BUFSIZE 4096
+#define BUFSIZE 4096
     
 #endif
 
--- mal/common/AGNet.c	Sat Feb  9 11:14:20 2002
+++ mal/common/AGNet.c	Tue Jul  1 23:24:11 2003
@@ -26,7 +26,8 @@
 //#include <stdio.h>	(adam) temporary hack - comment out so it will build on ce
 
 #ifndef _WIN32
-#include <ctype.h> //PENDING(klobad) not needed on palm or win remove??
+#include <ctype.h>
+//PENDING(klobad) not needed on palm or win remove??
 #include <errno.h>
 #endif
 
@@ -455,9 +456,9 @@
     case EINPROGRESS:
       return AG_NET_WOULDBLOCK;
 #ifdef _HPUX_SOURCE
-   #ifndef DARWIN
+#ifndef DARWIN
       case EWOULDBLOCK:
-   #endif
+#endif
 #endif
     case EAGAIN:
       return AG_NET_WOULDBLOCK;
