--- Makefile.orig	2013-12-30 18:07:04.000000000 -0600
+++ Makefile	2014-01-05 02:41:31.000000000 -0600
@@ -32,7 +32,7 @@
 
 RELEASE=r110
 DESTDIR=
-PREFIX=${DESTDIR}/usr
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man/man1
 DISTRIBNAME=lz4-$(RELEASE).tar.gz
@@ -90,13 +90,11 @@
 	@echo Cleaning completed
 
 
-ifeq ($(shell uname),Linux)
-
 install: lz4 lz4c
-	@install -d -m 755 $(BINDIR)/ $(MANDIR)/
-	@install -m 755 lz4 $(BINDIR)/lz4
-	@install -m 755 lz4c $(BINDIR)/lz4c
-	@install -m 644 lz4.1 $(MANDIR)/lz4.1
+	@install -d -m 755 $(DESTDIR)$(BINDIR)/ $(DESTDIR)$(MANDIR)/
+	@install -m 755 lz4 $(DESTDIR)$(BINDIR)/lz4
+	@install -m 755 lz4c $(DESTDIR)$(BINDIR)/lz4c
+	@install -m 644 lz4.1 $(DESTDIR)$(MANDIR)/lz4.1
 
 uninstall:
 	[ -x $(BINDIR)/lz4 ] && rm -f $(BINDIR)/lz4
@@ -116,5 +114,3 @@
 	@tar -czf $(DISTRIBNAME) lz4-$(RELEASE)/
 	@rm -rf lz4-$(RELEASE)
 	@echo Distribution $(DISTRIBNAME) built
-
-endif
