--- configure.orig	2022-10-21 18:04:07.000000000 -0500
+++ configure	2022-10-23 19:22:10.000000000 -0500
@@ -20123,6 +20123,7 @@
 /* end confdefs.h.  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <pwd.h>
 #include <limits.h>
@@ -22512,6 +22513,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <fnmatch.h>
+#include <stdlib.h>
 int main() { return(fnmatch("/*/bin/echo *", "/usr/bin/echo just a test", FNM_CASEFOLD)); }
 _ACEOF
 if ac_fn_c_try_run "$LINENO"
@@ -23442,6 +23444,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 int main(void)
@@ -23485,6 +23488,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
 
@@ -24363,6 +24367,7 @@
 #error "OpenSSL too old"
 #endif
 
+#include <stdio.h>
 int
 main (void)
 {
@@ -28482,7 +28487,7 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-
+#include <stdio.h>
 int
 main (void)
 {
