--- extensions/sse-fixups.c.orig	2011-06-28 20:01:39.000000000 -0700
+++ extensions/sse-fixups.c	2011-06-28 20:01:29.000000000 -0700
@@ -21,7 +21,7 @@
 
 #include "config.h"
 
-#if defined(__GNUC__) && (__GNUC__ >= 4) && defined(USE_SSE) && defined(USE_MMX)
+#if !defined(__clang__) && defined(__GNUC__) && (__GNUC__ >= 4) && defined(USE_SSE) && defined(USE_MMX)
 
 #include <stdint.h>
 #include <stdlib.h>
@@ -173,7 +173,7 @@ int init (void);
 int
 init (void)
 {
-#if defined(__GNUC__) && (__GNUC__ >= 4) && defined(USE_SSE) && defined(USE_MMX)
+#if !defined(__clang__) && defined(__GNUC__) && (__GNUC__ >= 4) && defined(USE_SSE) && defined(USE_MMX)
 
   Babl *rgbaF_linear = babl_format_new (
     babl_model ("RGBA"),
