From 8cefd02a39bfc0ca87b7d90187bf2068e28ad083 Mon Sep 17 00:00:00 2001
From: Alex Samorukov <samm@os2.kiev.ua>
Date: Thu, 31 Oct 2019 08:13:07 +0100
Subject: [PATCH] Add dylib extension for macOS build

---
 meson.build | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 0befaefc6..8c7d5222f 100644
--- meson.build
+++ meson.build
@@ -124,7 +124,13 @@ platform_android = host_os.contains('android')
 
 path_sep = ( platform_win32 ? ';'     : ':' )
 dirs_sep = ( platform_win32 ? '\\\\'  : '/' )
-lib_ext  = ( platform_win32 ? '.dll'  : '.so' )
+if platform_win32
+  lib_ext = '.dll'
+elif platform_osx
+  lib_ext = '.dylib'
+else
+  lib_ext = '.so'
+endif
 
 conf.set('BABL_PATH_SEPARATOR', '\'' + path_sep + '\'', description:
   'separator between paths in BABL_PATH')
-- 
2.22.0

