diff --git a/meson.build b/meson.build
index 55054f441a0378845e0d3655b7eb18f6d140e3e1..8c93be9d1b22a7b105a51d4eb9f4db7f75702a24 100644
--- meson.build.orig
+++ meson.build
@@ -55,7 +55,7 @@ lib_name    = meson.project_name() + '-' + api_version
 stability_version_number = (major_version != 0 ? minor_version : micro_version)
 stable = (stability_version_number % 2 == 0)
 
-conf.set10('BABL_UNSTABLE', not stable, Description:
+conf.set10('BABL_UNSTABLE', not stable, description:
   'Define to 1 if this is an unstable version of BABL.')
 
 conf.set       ('BABL_MAJOR_VERSION',    '@0@'.format(major_version))
