--- configure.orig	2021-08-19 00:40:10.000000000 +0800
+++ configure	2022-12-06 05:19:05.000000000 +0800
@@ -27,6 +27,9 @@
     ;;
 esac
 
+PKG_CPPFLAGS="$PKG_CPPFLAGS -I@PREFIX@/include/LegacySupport"
+EXTRA_PKG_LIBS="$EXTRA_PKG_LIBS -lMacportsLegacySupport"
+
 # Write to Makevars
 sed -e "s|@extra_pkg_libs@|$EXTRA_PKG_LIBS|" -e "s|@pkg_cppflags@|$PKG_CPPFLAGS|" \
     src/Makevars.in > src/Makevars
