# https://github.com/libhangul/libhangul/issues/63

--- data/keyboards/CMakeLists.txt.orig	2021-12-31 19:15:38.000000000 +0700
+++ data/keyboards/CMakeLists.txt	2022-08-18 03:58:42.000000000 +0700
@@ -72,7 +72,7 @@
         OUTPUT
             "${keyboard_name}.xml"
         COMMAND
-            sed -i -e '1 D'
+            sed -i '' -e '1 D'
                     -e '/^<hangul-keyboard/D'
                     -e '/^<\\/hangul-keyboard/D'
                     -e 's/^ *<name/    <name/' "${keyboard_name}.name.xml"


--- data/keyboards/Makefile.am.orig	2021-12-31 19:15:38.000000000 +0700
+++ data/keyboards/Makefile.am	2022-08-18 03:59:05.000000000 +0700
@@ -38,7 +38,7 @@
 # name.xml로 만들어 (msgfmt 입출력은 valid XML만 가능하므로 root node도 포함.)
 # 번역한후 template과 name.xml을 병합하여 키보드 xml 파일을 생성한다.
 hangul-keyboard-%.xml: hangul-keyboard-%.name.xml hangul-keyboard-%.xml.template
-	sed -i -e '1 D' \
+	sed -i '' -e '1 D' \
 		-e '/^<hangul-keyboard/D' \
 		-e '/^<\/hangul-keyboard/D' \
 		-e 's/^ *<name/    <name/'  $<

