diff --git modules/codec/x264.c modules/codec/x264.c
index 872269a..3d220c5 100644
--- modules/codec/x264.c
+++ modules/codec/x264.c
@@ -847,7 +847,7 @@ static int  Open ( vlc_object_t *p_this )
     char *psz_profile = var_GetString( p_enc, SOUT_CFG_PREFIX "profile" );
     if( psz_profile )
     {
-        const int mask = x264_bit_depth > 8 ? X264_CSP_HIGH_DEPTH : 0;
+        const int mask = X264_BIT_DEPTH > 8 ? X264_CSP_HIGH_DEPTH : 0;
 
 
 # ifdef MODULE_NAME_IS_x26410b
