diff --git modules/codec/fluidsynth.c modules/codec/fluidsynth.c
index bdc8d24..9859aa1 100644
--- modules/codec/fluidsynth.c
+++ modules/codec/fluidsynth.c
@@ -125,7 +125,7 @@ static int Open (vlc_object_t *p_this)
     {
         glob_t gl;
 
-        glob ("/usr/share/sounds/sf2/*.sf2", GLOB_NOESCAPE, NULL, &gl);
+        glob ("@PREFIX@/share/sounds/sf2/*.sf2", GLOB_NOESCAPE, NULL, &gl);
         for (size_t i = 0; i < gl.gl_pathc; i++)
         {
             const char *path = gl.gl_pathv[i];
