commit 042b1f0783d447d966c4cd314f7920ede7ef0610
Author: Jon Beniston <jon@beniston.com>
Date:   Sun Sep 18 16:53:35 2022 +0100

    Remote Control: Fix Mac build

diff --git a/plugins/feature/remotecontrol/remotecontrolgui.cpp b/plugins/feature/remotecontrol/remotecontrolgui.cpp
index 2ee4b0198..2b41ace75 100644
--- plugins/feature/remotecontrol/remotecontrolgui.cpp.orig
+++ plugins/feature/remotecontrol/remotecontrolgui.cpp
@@ -1005,7 +1005,7 @@
     QString format = sensor->m_format.trimmed();
     if (format.contains("%s"))
     {
-        formattedValue = QString::asprintf(format.toUtf8(), value.toString().toUtf8());
+        formattedValue = QString::asprintf(format.toUtf8(), value.toString().toUtf8().data());
     }
     else if (format.contains("%d") || format.contains("%u") || format.contains("%x") || format.contains("%X"))
     {
diff --git a/plugins/feature/remotecontrol/remotecontrolworker.h b/plugins/feature/remotecontrol/remotecontrolworker.h
index 34fabc98b..b3eb3112c 100644
--- plugins/feature/remotecontrol/remotecontrolworker.h.orig
+++ plugins/feature/remotecontrol/remotecontrolworker.h
@@ -16,7 +16,7 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.          //
 ///////////////////////////////////////////////////////////////////////////////////
 
-#ifndef INCLUDE_FEATURE_PERTESTERWORKER_H_
+#ifndef INCLUDE_FEATURE_REMOTECONTROLWORKER_H_
 #define INCLUDE_FEATURE_REMOTECONTROLWORKER_H_
 
 #include <QObject>
