--- build/moz.configure/toolchain.configure.orig	2021-09-20 18:53:22.000000000 -0400
+++ build/moz.configure/toolchain.configure	2021-09-20 18:53:40.000000000 -0400
@@ -145,7 +145,7 @@
         sdk_max_version = Version('10.15.4')
 
         if sdk:
-            sdk = sdk[0]
+            return sdk[0]
         elif host.os == 'OSX':
             sdk = check_cmd_output('xcrun', '--show-sdk-path', onerror=lambda: '').rstrip()
             if not sdk:
