--- src/WLTerminalView.m.orig	2019-07-01 07:00:29.646912382 +0800
+++ src/WLTerminalView.m	2019-07-01 07:00:38.729230261 +0800
@@ -741,7 +741,7 @@
             [NSBundle loadNibNamed:kLongScreenshotPanelNibFilename owner:self];
         }
         [NSApp beginSheet:_longScreenshotWindow
-           modalForWindow:_window
+           modalForWindow:self.window
             modalDelegate:nil
            didEndSelector:NULL
               contextInfo:nil];
