--- track.cc	2013-03-24 18:51:08.000000000 +0100
+++ track.cc	2013-12-14 12:15:57.000000000 +0100
@@ -19,6 +19,7 @@
 #include <algorithm>
 #include <cstdio>
 #include <vector>
+#include <cstdlib>
 
 #include "common.h"
 #include "rectangle.h"
@@ -43,12 +44,12 @@
       {
       if( 9 * r1.height() <= 10 * mean_height &&
           9 * r2.height() <= 10 * mean_height &&
-          10 * std::abs( r1.bottom() - r2.bottom() ) <= mean_height )
+          10 * abs( r1.bottom() - r2.bottom() ) <= mean_height )
         { val = 0; return ( r1.height() <= r2.height() ) ? 0 : 1; }
-      if( val > 1 && 10 * std::abs( r1.vcenter() - r2.vcenter() ) <= mean_height )
+      if( val > 1 && 10 * abs( r1.vcenter() - r2.vcenter() ) <= mean_height )
         { val = 1; return ( r1.bottom() <= r2.bottom() ) ? 0 : 1; }
       }
-    if( val > 2 && 10 * std::abs( r1.vcenter() - r2.vcenter() ) <= mean_height )
+    if( val > 2 && 10 * abs( r1.vcenter() - r2.vcenter() ) <= mean_height )
       { val = 2; return ( r1.bottom() <= r2.bottom() ) ? 0 : 1; }
     }
   return -1;
