--- libguile/posix.c.orig	2023-01-23 18:47:35.000000000 +0800
+++ libguile/posix.c	2023-04-17 01:33:47.000000000 +0800
@@ -108,6 +108,24 @@
 verify (WEXITSTATUS (W_EXITCODE (127, 0)) == 127);
 #endif
 
+/* Configure misdetects it as being available even when it is not. */
+#ifndef dprintf
+#include <stdarg.h>
+#include <sys/errno.h>
+int dprintf(int fildes, const char * __restrict format, ...) {
+  FILE *stream = fdopen(fildes, "w");
+  if (stream == NULL) {
+    errno = EBADF;
+    return -1;
+  }
+  va_list ap;
+  va_start(ap, format);
+  int result = vfprintf(stream, format, ap);
+  va_end(ap);
+  fclose(stream);
+  return result;
+}
+#endif
 
 #include <signal.h>
 
