--- src/aircrack-ng.c.orig	2015-04-08 23:47:28.000000000 -0500
+++ src/aircrack-ng.c	2015-05-09 02:03:04.000000000 -0500
@@ -3929,7 +3929,7 @@
 	int i, j, len, slen;
 	int nparallel = 1;
 
-#if defined(__i386__) || defined(__x86_64__)
+#if 0
 	// Check for SSE2, with SSE2 the algorithm works with 4 keys
 	if (shasse2_cpuid()>=2)
 		nparallel = 4;
@@ -5042,7 +5042,7 @@
 
 			case 'u' :
 				printf("Nb CPU detected: %d ", cpu_count);
-#if defined(__i386__) || defined(__x86_64__)
+#if 0
 				unused = shasse2_cpuid();
 
 				if (unused == 1) {
