--- common.mak.orig	2015-12-31 13:42:10.000000000 -0600
+++ common.mak	2017-01-18 02:49:47.000000000 -0600
@@ -46,8 +46,8 @@
 endif
 
 ifeq ($(subst TRUE,true,$(filter TRUE true,$(macport) $(MACPORT))),true)
-	COMMON_CFLAGS	+= -I/opt/local/include -I../..
-	LDFLAGS		+= -L/opt/local/lib
+	COMMON_CFLAGS	+= -isystem$(prefix)/include
+	LDFLAGS		+= -L$(prefix)/lib
 endif
 
 ifeq ($(subst TRUE,true,$(filter TRUE true,$(sqlite) $(SQLITE))),true)
@@ -59,7 +59,7 @@
 endif
 
 ifeq ($(PCRE), true)
-COMMON_CFLAGS += $(shell $(PKG_CONFIG) --cflags libpcre) -DHAVE_PCRE
+COMMON_CFLAGS += $(shell $(PKG_CONFIG) --cflags libpcre | sed -e 's%-I$(prefix)%-isystem$(prefix)%g') -DHAVE_PCRE
 endif
 
 ifeq ($(OSNAME), cygwin)
