--- src/aircrack-ng.c.orig	2010-04-09 09:50:14.000000000 -0500
+++ src/aircrack-ng.c	2014-04-26 17:21:12.000000000 -0500
@@ -3852,7 +3852,7 @@
 	int i, j, len, slen;
 	int nparallel = 1;
 
-#if defined(__i386__) || defined(__x86_64__)
+#if 0
 	// Check for SSE2, with SSE2 the algorithm works with 4 keys
 	if (shasse2_cpuid()>=2)
 		nparallel = 4;
@@ -4626,7 +4626,7 @@
 
 			case 'u' :
 				printf("Nb CPU detected: %d ", cpu_count);
-#if defined(__i386__) || defined(__x86_64__)
+#if 0
 				unused = shasse2_cpuid();
 
 				if (unused == 1) {
