--- pyproject.toml.orig	2023-01-08 08:13:55.000000000 -0600
+++ pyproject.toml	2023-01-29 10:15:28.000000000 -0600
@@ -2,8 +2,8 @@
 requires = ["setuptools",
             "setuptools_scm>=6.2",
             "wheel",
-            "cython==0.29.30",
-            "oldest-supported-numpy",
+            "cython",
+            "numpy",
             "extension-helpers"]
 build-backend = 'setuptools.build_meta'
 
