--- ext/gmp/gmp.c.orig	2008-12-31 05:15:49.000000000 -0600
+++ ext/gmp/gmp.c	2010-02-19 02:31:47.000000000 -0600
@@ -1359,7 +1359,7 @@
 
 		GMPG(rand_initialized) = 1;
 	}
-	mpz_urandomb(*gmpnum_result, GMPG(rand_state), GMP_ABS (limiter) * __GMP_BITS_PER_MP_LIMB);
+	mpz_urandomb(*gmpnum_result, GMPG(rand_state), GMP_ABS (limiter) * GMP_LIMB_BITS);
 
 	ZEND_REGISTER_RESOURCE(return_value, gmpnum_result, le_gmp);
 }
