--- posix.mak.orig	2015-10-04 11:19:24.000000000 -0700
+++ posix.mak	2016-04-30 07:44:08.000000000 -0700
@@ -1,6 +1,6 @@
 DMD = ../dmd/src/dmd
 CC = gcc
-INSTALL_DIR = ../install
+INSTALL_DIR = $(DESTDIR)$(PREFIX)
 DRUNTIME_PATH = ../druntime
 PHOBOS_PATH = ../phobos

@@ -67,7 +67,7 @@
 # default include/link paths, override by setting DMD (e.g. make -f posix.mak DMD=dmd)
 DMD += -I$(DRUNTIME_PATH)/import -I$(PHOBOS_PATH) -L-L$(PHOBOS_PATH)/generated/$(OS)/release/$(MODEL)

-DFLAGS = -w
+DFLAGS = -w -O -release

 TOOLS = \
     $(ROOT)/rdmd \
