--- configure.ac.orig	2015-07-20 08:37:46.000000000 -0700
+++ configure.ac	2015-07-20 21:40:49.000000000 -0700
@@ -177,8 +177,9 @@
 
 IT_PROG_INTLTOOL([0.50.1])
 
-AM_GNU_GETTEXT_VERSION([0.18])
-AM_GNU_GETTEXT([external])
+dnl AM_GNU_GETTEXT macros cannot be used with IT_PROG_INTLTOOL
+dnl AM_GNU_GETTEXT_VERSION([0.18])
+dnl AM_GNU_GETTEXT([external])
 
 AC_SUBST([GETTEXT_PACKAGE], [gedit])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE], "$GETTEXT_PACKAGE", [Define to the gettext package name.])
