--- configure.ac.orig	2009-04-13 08:46:20.000000000 -0700
+++ configure.ac	2009-05-10 17:41:50.000000000 -0700
@@ -93,6 +93,25 @@
 	AC_DEFINE([PLATFORM_OSX],[1],[Defined if platform is Mac OSX])
 fi
 
+dnl check for gdk quartz support
+AC_MSG_CHECKING([if GDK supports quartz])
+gdk_has_quartz=no
+if $PKG_CONFIG --exists gdk-quartz-2.0; then
+       gdk_has_quartz=yes
+fi
+AC_MSG_RESULT([$gdk_has_quartz])
+
+dnl check if osx quartz platform
+AC_MSG_CHECKING([if platform is osx quartz])
+platform_osx_quartz=no
+if test "$platform_osx" = "yes" &&
+   test "$gdk_has_quartz" = "yes" ; then
+        platform_osx_quartz=yes
+fi
+AC_MSG_RESULT([$platform_osx_quartz])
+
+AM_CONDITIONAL(PLATFORM_OSX_QUARTZ, test "$platform_osx_quartz" = "yes")
+
 dnl ===============================================================
 dnl Expanded dirs
 dnl ===============================================================
@@ -233,8 +252,8 @@
 
 AM_CONDITIONAL(BUILD_SEXY, test "$gtkatleast" = "no")
 
-if test "$platform_osx" = "no" &&
-   test "$os_win32" = no; then
+if test "$platform_osx_quartz" = "no" &&
+   test "$os_win32" = "no"; then
 	PKG_CHECK_MODULES(EGG_SMCLIENT, [
 		sm >= 1.0.0
 	])
