--- gedit/meson.build.orig	2020-06-10 12:49:37.000000000 -0700
+++ gedit/meson.build	2020-06-10 12:49:49.000000000 -0700
@@ -122,7 +122,7 @@
   libgd_dep,
 ]
 
-if host_machine.system() == 'darwin'
+if host_machine.system() == 'mydarwin'
   libgedit_sources += files(
     'gedit-app-osx.m',
     'gedit-file-chooser-dialog-osx.m',
@@ -291,7 +291,7 @@
 
 gedit_c_args = []
 
-if host_machine.system() == 'darwin'
+if host_machine.system() == 'mydarwin'
   gedit_c_args += [
     '-DOS_OSX=1',
   ]
