--- tetex-src-3.0/texk/dvipsk/dvips.c.orig	2009-03-25 14:45:01.000000000 +0900
+++ tetex-src-3.0/texk/dvipsk/dvips.c	2009-03-25 15:01:19.000000000 +0900
@@ -1217,6 +1217,7 @@
       oname = nextstring ;
 #ifndef VMCMS  /* get stuff before LAST "." */
       lastext = strlen(iname) - 1 ;
+      if (lastext >= 0) { // lastext should be a positive number
       while (iname[lastext] != '.' && lastext > 0)
          lastext-- ;
       if (iname[lastext] != '.')
@@ -1253,6 +1254,7 @@
 #ifndef VMCMS
       *nextstring++ = 'p' ;
       *nextstring++ = 's' ;
+      } // lastext should be a positive number
 #else  /* might as well keep things uppercase */
       *nextstring++ = 'P' ;
       *nextstring++ = 'S' ;
