--- tetex-src-3.0/texk/dvipng/configure.orig	2009-02-22 20:49:42.000000000 -0500
+++ tetex-src-3.0/texk/dvipng/configure	2009-02-22 20:50:08.000000000 -0500
@@ -2488,7 +2488,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lpng -lz -lm $LIBS"
+LIBS="-lpng -lz -lc -lm $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 2494 "configure"
 #include "confdefs.h"
@@ -2601,7 +2601,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lt1 -lz -lm $LIBS"
+LIBS="-lt1 -lz -lc -lm $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 2607 "configure"
 #include "confdefs.h"
@@ -2638,7 +2638,7 @@
 
   fi
 
-  LIBS="-lt1 -lm $LIBS"
+  LIBS="-lt1 -lc -lm $LIBS"
   if test "$with_system_t1lib" = yes; then
       echo $ac_n "checking whether installed T1 lib is >= v 5.0.0 ""... $ac_c" 1>&6
 echo "configure:2645: checking whether installed T1 lib is >= v 5.0.0 " >&5
@@ -3122,7 +3122,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lm  $LIBS"
+LIBS="-lc -lm  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 3128 "configure"
 #include "confdefs.h"
@@ -3152,7 +3152,7 @@
 #define $ac_tr_lib 1
 EOF
 
-  LIBS="-lm $LIBS"
+  LIBS="-lc -lm $LIBS"
 
 else
   echo "$ac_t""no" 1>&6
