--- tetex-src-3.0/texk/web2c/cpascal.h.orig	2004-08-29 05:49:59.000000000 +0900
+++ tetex-src-3.0/texk/web2c/cpascal.h	2011-03-01 07:00:18.000000000 +0900
@@ -272,4 +272,12 @@
 /* version.c */
 extern string versionstring;
 
+/* Apparently POSIX 2008 has getline and glibc 2.9.90 exports it.
+   tangle, weave, et al. use that symbol; try to define it away so
+   something that a standard won't usurp.  */
+#ifdef getline
+#undef getline
+#endif
+#define getline web2c_getline
+
 #endif /* not CPASCAL_H */
--- tetex-src-3.0/texk/dvipsk/afm2tfm.c.orig	2011-03-01 07:02:54.000000000 +0900
+++ tetex-src-3.0/texk/dvipsk/afm2tfm.c	2011-03-01 07:14:42.000000000 +0900
@@ -42,6 +42,14 @@
 #endif
 #endif /* KPATHSEA */
 
+/* Apparently POSIX 2008 has getline and glibc 2.9.90 exports it.
+   tangle, weave, et al. use that symbol; try to define it away so
+   something that a standard won't usurp.  */
+#ifdef getline
+#undef getline
+#endif
+#define getline web2c_getline
+
 /* JLH: added these to make the code easier to read and remove some
    ascii<->ebcdic dependencies */
 #define ASCII_A 65
