*** ptex-base.ch.orig	2010-05-22 14:56:08.000000000 +0900
--- ptex-base.ch	2010-05-22 14:56:41.000000000 +0900
***************
*** 3524,3529 ****
--- 3524,3537 ----
  new_noad:=p;
  @z
  
+ @x [34.687] accent with japanese char
+ @d accent_noad_size=5 {number of |mem| words in an accent noad}
+ @d accent_chr(#)==#+4 {the |accent_chr| field of an accent noad}
+ @y
+ @d accent_noad_size=6 {number of |mem| words in an accent noad}
+ @d accent_chr(#)==#+5 {the |accent_chr| field of an accent noad}
+ @z
+ 
  @x [34.691] l.14236 - pTeX: print_fam_and_char
  procedure print_fam_and_char(@!p:pointer); {prints family and character}
  begin print_esc("fam"); print_int(fam(p)); print_char(" ");
***************
*** 5623,5629 ****
      p:=nucleus(info(p)); q:=kcode_noad_nucleus(p);
      end;
    math_type(p):=math_jchar; fam(p):=cur_jfam; character(p):=qi(0);
!   info(q):=KANJI(cx);
    if font_dir[fam_fnt(fam(p)+cur_size)]=dir_default then
      begin print_err("Not two-byte family");
      help1("IGNORE.");@/
--- 5631,5637 ----
      p:=nucleus(info(p)); q:=kcode_noad_nucleus(p);
      end;
    math_type(p):=math_jchar; fam(p):=cur_jfam; character(p):=qi(0);
!   math_kcode(p-1):=KANJI(cx);
    if font_dir[fam_fnt(fam(p)+cur_size)]=dir_default then
      begin print_err("Not two-byte family");
      help1("IGNORE.");@/
