*** ptex-base.ch.orig	2010-05-22 14:53:20.000000000 +0900
--- ptex-base.ch	2010-05-22 14:54:53.000000000 +0900
***************
*** 4487,4494 ****
        ("\box register that now have a different direction.")@/
        ("Proceed, and I'll discard its present contents.");
      box_error(n)
!     end;
!   height(r):=height(box(n))+depth(box(n));
    end;
  @z
  
--- 4487,4495 ----
        ("\box register that now have a different direction.")@/
        ("Proceed, and I'll discard its present contents.");
      box_error(n)
!     end
!   else
!     height(r):=height(box(n))+depth(box(n));
    end;
  @z
  
***************
*** 5397,5404 ****
    begin KANJI(cx):=cur_val;
    if direction=dir_tate then f:=cur_tfont else f:=cur_jfont;
    p:=new_character(f,get_jfm_pos(KANJI(cx),f));
!   {DEBUG if p<>null then ...}
!   link(p):=get_avail; info(link(p)):=KANJI(cx);
    end
  else begin f:=cur_font; p:=new_character(f,cur_val);
    end;
--- 5398,5407 ----
    begin KANJI(cx):=cur_val;
    if direction=dir_tate then f:=cur_tfont else f:=cur_jfont;
    p:=new_character(f,get_jfm_pos(KANJI(cx),f));
!   if p<>null then
!     begin
!       link(p):=get_avail; info(link(p)):=KANJI(cx);
!     end;
    end
  else begin f:=cur_font; p:=new_character(f,cur_val);
    end;
***************
*** 7545,7551 ****
      {@<Append |disp_node| at begin of displace area@>;}
      ins_kp:=false; goto main_loop;
    end else begin { Kanji -> cs }
!     {@<Append |disp_node| at begin of displace area@>;}
      goto reswitch;
    end;
  
--- 7548,7554 ----
      {@<Append |disp_node| at begin of displace area@>;}
      ins_kp:=false; goto main_loop;
    end else begin { Kanji -> cs }
!     @<Append |disp_node| at begin of displace area@>;
      goto reswitch;
    end;
  
