--- tetex-src-3.0/texk/xdvik/configure.orig	2009-01-30 02:01:00.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/configure	2009-01-30 02:05:56.000000000 +0900
@@ -4289,7 +4289,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lt1 -lz -lm $LIBS"
+LIBS="-lt1 -lz -lc -lm $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 4295 "configure"
 #include "confdefs.h"
@@ -4326,7 +4326,7 @@
 
   fi
 
-  LIBS="-lt1 -lm $LIBS"
+  LIBS="-lt1 -lc -lm $LIBS"
   if test "$with_system_t1lib" = yes; then
       echo $ac_n "checking whether installed T1 lib is >= v 5.0.0 ""... $ac_c" 1>&6
 echo "configure:4333: checking whether installed T1 lib is >= v 5.0.0 " >&5
@@ -6313,7 +6313,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lm  $LIBS"
+LIBS="-lc -lm  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 6319 "configure"
 #include "confdefs.h"
@@ -6343,7 +6343,7 @@
 #define $ac_tr_lib 1
 EOF
 
-  LIBS="-lm $LIBS"
+  LIBS="-lc -lm $LIBS"
 
 else
   echo "$ac_t""no" 1>&6
