--- tetex-src-3.0/texk/xdvik/xdvi.c.orig	2010-03-31 09:13:16.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/xdvi.c	2010-03-31 09:41:39.000000000 +0900
@@ -1002,6 +1002,34 @@
     "b10r","44x31mm",
     "", "0",
 
+    /* JIS `B' formats, Portrait */
+    "jisb0", "1030x1456mm",
+    "jisb1", "728x1030mm",
+    "jisb2", "515x728mm",
+    "jisb3", "364x515mm",
+    "jisb4", "257x364mm",
+    "jisb5", "182x257mm",
+    "jisb6", "128x182mm",
+    "jisb7", "91x128mm",
+    "jisb8", "64x91mm",
+    "jisb9", "45x64mm",
+    "jisb10","32x45mm",
+    "", "0",
+
+    /* JIS `B' formats, Landscape */
+    "jisb0r", "1456x1030mm",
+    "jisb1r", "1030x728mm",
+    "jisb2r", "728x515mm",
+    "jisb3r", "515x364mm",
+    "jisb4r", "364x257mm",
+    "jisb5r", "257x182mm",
+    "jisb6r", "182x128mm",
+    "jisb7r", "128x91mm",
+    "jisb8r", "91x64mm",
+    "jisb9r", "64x45mm",
+    "jisb10r","45x32mm",
+    "", "0",
+
     /* ISO `C' formats, Portrait */
     "c0", "917x1297mm",
     "c1", "648x917mm",
--- tetex-src-3.0/texk/xdvik/gui/xm_prefs_page.c.orig	2010-03-31 09:13:16.000000000 +0900
+++ tetex-src-3.0/texk/xdvik/gui/xm_prefs_page.c	2010-03-31 09:48:28.000000000 +0900
@@ -95,9 +95,14 @@
     { "A3",		"a3",		"297x420mm",	"420x297mm"	},
     { "A4",		"a4",		"210x297mm",	"297x210mm"	},
     { "A5",		"a5",		"148x210mm",	"210x148mm"	},
+    { "B4",		"b4",		"250x353mm",	"353x250mm"	},
     { "B5",		"b5",		"176x250mm",	"250x176mm"	},
     { "B6",		"b6",		"125x176mm",	"176x125mm"	},
     { "B7",		"b7",		"88x125mm",	"125x88mm"	},
+    { "JIS B4",		"jisb4",	"257x364mm",	"364x257mm"	},
+    { "JIS B5",		"jisb5",	"182x257mm",	"257x182mm"	},
+    { "JIS B6",		"jisb6",	"128x182mm",	"182x128mm"	},
+    { "JIS B7",		"jisb7",	"91x128mm",	"128x91mm"	},
     { NULL,		NULL,		NULL,		NULL		} /* terminate */
 };
 
