--- work/ocaml-buddy-0.5/myocamlbuild.ml	2011-10-04 20:44:28.000000000 +0200
+++ myocamlbuild.ml	2012-07-26 22:34:34.000000000 +0200
@@ -44,7 +44,7 @@
 
        (* When one link an OCaml library/binary/package, one should use -linkpkg *)
        flag ["ocaml"; "link"] & A"-linkpkg";
-       flag ["c"; "compile"; "gcc"] (S[A"-cc"; A"gcc"; A"-ccopt"; A"-fPIC"]);
+       flag ["c"; "compile"; "gcc"] (S[A"-cc"; A"gcc"; A"-ccopt"; A"-fPIC"; A"-ccopt"; A"-I@PREFIX@/include"; A"-ccopt"; A"-L@PREFIX@/lib"; A"-ccopt"; A"-lbuddy"; A"-ccopt"; A"-fnested-functions"]);
        flag ["cc"; "compile"; "g++"] (S[A"-cc"; A"g++"; A"-ccopt"; A"-fPIC"]);
        (* flag ["ocamlmklib"] (S[A"-lminisat"; A"-oc"; A"minisat_stubs" ]); *)
 
@@ -67,10 +67,10 @@
        end (find_syntaxes ());
 
        flag ["ocaml"; "link"; "c_use_bdd"; "byte"]
-       (S[A"-custom"; A"-ccopt"; A("-Lbdd"); A"-cclib"; A("-lbdd")]);
+       (S[A"-custom"; A"-ccopt"; A("-L@PREFIX@/lib"); A"-cclib"; A("-lbdd")]);
 
        flag ["ocaml"; "link"; "c_use_bdd"; "native"]
-       (S[A"-ccopt"; A("-Lbdd"); A"-cclib"; A("-lbdd")]);
+       (S[A"-ccopt"; A("-L@PREFIX@/lib"); A"-cclib"; A("-lbdd")]);
 
        flag [ "byte"; "library"; "link" ]
        (S[A"-dllib"; A("-lbdd"); A"-cclib"; A("-lbuddystubs")]);
