--- Makefile.orig	2014-08-15 01:40:05.000000000 -0700
+++ Makefile	2019-01-26 08:50:20.000000000 -0700
@@ -4,8 +4,10 @@
 #   make git-stamp
 VERSION=
 
-CC=			gcc
-CFLAGS=		-g -Wall $(VERSION) -O2
+include Makefile.macports.inc
+
+CC ?=			gcc
+CFLAGS ?=		-g -Wall $(VERSION) -O2
 #LDFLAGS=		-Wl,-rpath,\$$ORIGIN/../lib
 DFLAGS=		-D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_USE_KNETFILE -D_CURSES_LIB=1
 KNETFILE_O=	knetfile.o
@@ -91,7 +93,7 @@
 
 
 libbam.1.dylib-local:$(LOBJS)
-		libtool -dynamic $(LOBJS) -o libbam.1.dylib -lc -lz
+		libtool -dynamic -install_name $(PREFIX)/lib/libbam.1.dylib -current_version 0.1.20 -compatibility_version 0.1  $(LOBJS) -o libbam.1.dylib -lc -lz
 
 libbam.so.1-local:$(LOBJS)
 		$(CC) -shared -Wl,-soname,libbam.so -o libbam.so.1 $(LOBJS) -lc -lz
