--- setup.py.orig	2022-03-11 13:42:28.000000000 -0500
+++ setup.py	2022-03-11 13:42:53.000000000 -0500
@@ -75,8 +75,6 @@
         'Topic :: Software Development :: Libraries :: Python Modules',
       ],
       keywords = ['python', 'sparql', 'rdf', 'rdflib'],
-      use_2to3 = True,
-      use_2to3_fixers = ['custom_fixers'],
       project_urls={
         'Home': 'https://rdflib.github.io/sparqlwrapper/',
         'Documentation': 'https://rdflib.github.io/sparqlwrapper/doc/',
