--- configure.ac.orig	2004-09-04 09:42:10.000000000 -0500
+++ configure.ac	2013-02-16 04:31:46.000000000 -0600
@@ -2,12 +2,11 @@
 
 AC_INIT
 AC_CONFIG_SRCDIR([giblib/gib_list.c])
-AM_CONFIG_HEADER(giblib/giblib_config.h)
+AC_CONFIG_HEADERS([giblib/giblib_config.h])
 AM_INIT_AUTOMAKE(giblib, 1.2.4, no-define)
 
 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AM_PROG_LIBTOOL
 AC_PROG_INSTALL
