--- configure.in.orig	2011-03-05 21:46:37.000000000 +1100
+++ configure.in	2012-04-25 06:30:19.000000000 +1000
@@ -436,6 +436,7 @@
 #--------------------------------------------------------------------
 
 AC_MSG_CHECKING(whether ... arguments behave correctly)
+SAVE_CXXFLAGS="$CXXFLAGS"
 if test -z "$GUILE_CFLAGS"; then
   CXXFLAGS="`guile-config compile`"
 else
@@ -454,7 +455,6 @@
 ],[
   AC_MSG_RESULT(no)
 ])
-CXXFLAGS=""
 
 AC_MSG_CHECKING(the size_t of guile strings)
 if test -z "$GUILE_CFLAGS"; then
@@ -476,7 +476,7 @@
   AC_DEFINE(guile_str_size_t, size_t, [Guile string size type])
   AC_MSG_RESULT(size_t)
 ])
-CXXFLAGS=""
+CXXFLAGS="$SAVE_CXXFLAGS"
 
 AC_MSG_CHECKING([if statically linking with guile works])
 SAVE_CPPFLAGS="$CPPFLAGS"
@@ -932,6 +932,7 @@
     STD_DEBUG_FLAGS="-fno-exceptions"
 fi
 
+SAVE_CXXFLAGS="$CXXFLAGS"
 AC_MSG_CHECKING(if compiler supports $STD_DEBUG_FLAGS)
 CXXFLAGS="$STD_DEBUG_FLAGS"
 AC_TRY_COMPILE([
@@ -944,7 +945,7 @@
     CONFIG_CXXDIALECT=""
     AC_MSG_RESULT(no)
 ])
-CXXFLAGS=""
+CXXFLAGS="$SAVE_CXXFLAGS"
 AC_SUBST(CONFIG_CXXDIALECT)
 
 AC_ARG_ENABLE(debug,
