--- aclocal.m4.orig	2014-03-06 10:43:02.000000000 -0600
+++ aclocal.m4	2015-03-29 20:18:58.000000000 -0500
@@ -8,15 +8,15 @@
 ## The GUILE_FLAGS macro.
   ## First, let's just see if we can find Guile at all.
   AC_MSG_CHECKING(for Guile)
-  guile-config link > /dev/null || {
-    echo "configure: cannot find guile-config; is Guile installed?" 1>&2
+  guile18-config link > /dev/null || {
+    echo "configure: cannot find guile18-config; is Guile installed?" 1>&2
     exit 1
   }
-  GUILE_ORIGINAL_CFLAGS="`guile-config compile`"
+  GUILE_ORIGINAL_CFLAGS="`guile18-config compile`"
   GUILE_CFLAGS="$GUILE_ORIGINAL_CFLAGS"
   GUILE_VARIANT_CFLAGS="$GUILE_ORIGINAL_CFLAGS $GUILE_ORIGINAL_CFLAGS/guile $GUILE_ORIGINAL_CFLAGS/libguile"
-  GUILE_LDFLAGS="`guile-config link`"
-  GUILE_VARIANT_LDFLAGS="-L`guile-config info libdir` -lguile -lreadline -ltermcap"
+  GUILE_LDFLAGS="`guile18-config link`"
+  GUILE_VARIANT_LDFLAGS="-L`guile18-config info libdir` -lguile -lreadline -ltermcap"
   AC_SUBST(GUILE_CFLAGS)
   AC_SUBST(GUILE_LDFLAGS)
   AC_MSG_RESULT(yes)
@@ -28,7 +28,7 @@
 
 AC_DEFUN(TEXMACS_LINK_GUILE,
 [AC_TRY_LINK([
-#include <guile/gh.h>
+#include <guile18/gh.h>
 $CONFIG_DOTS
 ],[
   struct dummy {
