--- configure.sav	2006-11-26 18:47:29.000000000 -0500
+++ configure	2006-11-26 18:48:01.000000000 -0500
@@ -6150,8 +6150,8 @@
 #    CXX="export CYGWIN=check_case:strict; $CXX"
 #    AC_SUBST(CXX)
   ;;
-  powerpc-*-darwin*)
-    echo "$ac_t""final adjustments for a PowerPC/Darwin host" 1>&6
+  *-*-darwin*)
+    echo "$ac_t""final adjustments for a Darwin host" 1>&6
     CONFIG_OS="DARWIN"
     CONFIG_CXXFLAGS="-I${prefix}/include"
     CONFIG_BSHARED=""
