--- CMakeLists.txt.orig	2013-09-14 15:10:53.000000000 +0900
+++ CMakeLists.txt	2013-09-14 15:10:57.000000000 +0900
@@ -22,7 +22,7 @@
 add_subdirectory(pics)
 add_subdirectory(toolbar)
 add_subdirectory(icon)
-add_subdirectory(doc)
+#add_subdirectory(doc)
 
 # use the project-specific _SOURCE_DIR variable (set by the project() call above), Alex
 include_directories(${kuser_SOURCE_DIR} ${KDEPIMLIBS_INCLUDE_DIR})
