diff -NaurdwB ./bin/_build.sh ./bin/_build.sh
--- ./bin/_build.sh	2024-12-28 13:40:15
+++ ./bin/_build.sh	2024-12-28 13:40:10
@@ -788,7 +788,7 @@
       virtualenv==15.0.2  \
   ; do
       ruler "Installing ${pkg}";
-      "${bootstrap_python}" -m pip install -I ${NESTED} "${pkg}";
+      "${bootstrap_python}" -m pip install -I ${NESTED} --cert @PREFIX@/share/curl/curl-ca-bundle.crt "${pkg}";
   done;
 }
 
@@ -802,6 +802,7 @@
     --pre                                  \
     --no-cache-dir                         \
     --log-file="${dev_home}/pip.log"       \
+    --cert @PREFIX@/share/curl/curl-ca-bundle.crt \
     "$@";
 }
 
@@ -815,6 +816,7 @@
     --find-links="${dev_patches}"            \
     --find-links="${dev_home}/pip_downloads" \
     --log-file="${dev_home}/pip.log"         \
+    --cert @PREFIX@/share/curl/curl-ca-bundle.crt \
     "$@";
 }
 
@@ -826,6 +828,7 @@
     --no-cache-dir                           \
     --find-links="${dev_patches}"            \
     --log-file="${dev_home}/pip.log"         \
+    --cert @PREFIX@/share/curl/curl-ca-bundle.crt \
     "$@";
 }
 
diff -NaurdwB ./requirements-cs.txt ./requirements-cs.txt
--- ./requirements-cs.txt	2024-12-28 13:40:15
+++ ./requirements-cs.txt	2024-12-28 13:40:10
@@ -37,5 +37,5 @@
 
     psutil==5.0.0
     setproctitle==1.1.10
-    # xattr==0.7.5  # Only needed for upgrades from ancient versions.  Added in _cache_deps.
+    xattr==0.7.5  # Only needed for upgrades from ancient versions.  Added in _cache_deps.
         #cffi
diff -NaurdwB ./requirements-twisted-default.txt ./requirements-twisted-default.txt
--- ./requirements-twisted-default.txt	2024-12-28 13:40:15
+++ ./requirements-twisted-default.txt	2024-12-28 20:47:02
@@ -10,7 +10,7 @@
     #   but doesn't specify them as dependencies, so that are explicitly
     #   added to calendarserver.
     #pyOpenSSL
-    service_identity==16.0.0
+    service_identity
         characteristic==14.3.0
         pyasn1==0.1.9
         pyasn1-modules==0.0.8
@@ -18,7 +18,7 @@
     pycrypto==2.6.1
 
     pyOpenSSL==16.2.0
-        cryptography==1.6
+        cryptography
         	idna
             #pyasn1
             #cffi
