--- c++std.pri.orig	2023-06-28 11:53:41
+++ c++std.pri	2023-06-28 11:54:24
@@ -1,22 +1,6 @@
 macx {
-  # Mac needs special care to link against the correct C++ library
-  # We attempt to auto-detect it by inspecting Boost
-  dirs = $${BOOSTDIR} $${QMAKE_LIBDIR}
-  for(dir, dirs) {
-    system(otool -L $${dir}/libboost_thread*  | grep libc++ >& /dev/null ) {
-      message("Using libc++")
-      CONFIG += libc++
-    }
-    else {
-      message("Using libstdc++")
-      CONFIG += libstdc++
-      c++std {
-        # libc++ is a requirement for using C++14 
-        warning("Disabling C++14 since libstdc++ dependencies were found")
-        CONFIG -= c++std
-      }
-    }
-  }
+  message("Using libc++")
+  CONFIG += libc++
 
   libc++ {
     QMAKE_CXXFLAGS += -stdlib=libc++
