--- rtree/core.py.orig	2014-11-04 14:14:05.000000000 +0300
+++ rtree/core.py	2014-11-04 14:20:22.000000000 +0300
@@ -103,6 +103,8 @@
 elif os.name == 'posix':
     platform = os.uname()[0]
     lib_name = find_library('spatialindex_c')
+    if not lib_name and platform == 'Darwin':
+        lib_name = '@PREFIX@/lib/libspatialindex_c.dylib'
     rt = ctypes.CDLL(lib_name)
 else:
     raise RTreeError('Unsupported OS "%s"' % os.name)
