diff -ruw fetchmail-6.2.5/config.h.in fetchmail-6.2.5-macosx/config.h.in
--- fetchmail-6.2.5/config.h.in	Wed Jul 16 21:03:21 2003
+++ fetchmail-6.2.5-macosx/config.h.in	Fri Dec 19 17:14:31 2003
@@ -22,6 +22,9 @@
 /* Define as 1 if you have gettext and don't want to use GNU gettext.  */
 #undef HAVE_GETTEXT
 
+/* Define if you have krb5_init_etc */
+#undef HAVE_KRB5_INIT_ETS
+
 /* Define if you have MIT kerberos */
 #undef HAVE_GSS_C_NT_HOSTBASED_SERVICE
 
diff -ruw fetchmail-6.2.5/configure.in fetchmail-6.2.5-macosx/configure.in
--- fetchmail-6.2.5/configure.in	Fri Oct 10 04:36:57 2003
+++ fetchmail-6.2.5-macosx/configure.in	Fri Dec 19 23:11:07 2003
@@ -471,13 +471,20 @@
                 AC_CHECK_LIB(k5crypto,
                    krb5_des_string_to_key,
                    libk5crypto=-lk5crypto,
+                      AC_CHECK_LIB(k5crypto,
+                         krb5_c_string_to_key,
+                         libk5crypto=-lk5crypto,
                       AC_ERROR([Kerberos 5 DES libraries not found]),
                    ${LDEFLAGS} ${ac_krblibs}),
+                   ${LDEFLAGS} ${ac_krblibs}),
                  ${LDEFLAGS} ${ac_krblibs})
         fi
         AC_CHECK_LIB(krb5, krb5_init_context,[],
            continue,
            ${LDEFLAGS} ${ac_krblibs})
+        AC_CHECK_LIB(krb5, krb5_init_ets,
+           AC_DEFINE(HAVE_KRB5_INIT_ETS),
+           AC_MSG_RESULT([Kerberos V library loads its own error tables]))
         AC_DEFINE(KERBEROS_V5)
         test "$dir" != "/usr" && CEFLAGS="$CEFLAGS -I$dir/include"
         LDEFLAGS="$LDEFLAGS -L$dir/lib"
Only in fetchmail-6.2.5-macosx: configure.in~
diff -ruw fetchmail-6.2.5/driver.c fetchmail-6.2.5-macosx/driver.c
--- fetchmail-6.2.5/driver.c	Wed Oct 15 15:22:31 2003
+++ fetchmail-6.2.5-macosx/driver.c	Fri Dec 19 17:05:47 2003
@@ -210,7 +210,9 @@
     krb5_auth_context auth_context = NULL;
 
     krb5_init_context(&context);
+#ifdef HAVE_KRB5_INIT_ETS
     krb5_init_ets(context);
+#endif
     krb5_auth_con_init(context, &auth_context);
 
     if (retval = krb5_cc_default(context, &ccdef)) {
diff -ruw fetchmail-6.2.5/fetchmail.c fetchmail-6.2.5-macosx/fetchmail.c
--- fetchmail-6.2.5/fetchmail.c	Wed Oct 15 15:22:31 2003
+++ fetchmail-6.2.5-macosx/fetchmail.c	Fri Dec 19 17:02:37 2003
@@ -387,7 +387,7 @@
 	{
 	    fprintf(stderr,GT_("fetchmail: %s fetchmail at %d killed.\n"),
 		    bkgd ? GT_("background") : GT_("foreground"), pid);
-	    lock_release();
+	    lock_do_release();
 	    if (argc == 2)
 		exit(0);
 	    else
@@ -1300,7 +1300,7 @@
 	  memset(ctl->password, '\0', strlen(ctl->password));
 
 #if !defined(HAVE_ATEXIT) && !defined(HAVE_ON_EXIT)
-    lock_release();
+    lock_do_release();
 #endif
 
     if (activecount == 0)
diff -ruw fetchmail-6.2.5/fetchmail.h fetchmail-6.2.5-macosx/fetchmail.h
--- fetchmail-6.2.5/fetchmail.h	Wed Oct 15 15:22:31 2003
+++ fetchmail-6.2.5-macosx/fetchmail.h	Fri Dec 19 17:01:54 2003
@@ -462,7 +462,7 @@
 
 /* lock.c: concurrency locking */
 void lock_setup(void), lock_assert(void);
-void lock_or_die(void), lock_release(void);
+void lock_or_die(void), lock_do_release(void);
 int lock_state(void);
 void lock_dispose(void);
 
diff -ruw fetchmail-6.2.5/lock.c fetchmail-6.2.5-macosx/lock.c
--- fetchmail-6.2.5/lock.c	Wed Aug 13 04:08:17 2003
+++ fetchmail-6.2.5-macosx/lock.c	Fri Dec 19 17:01:01 2003
@@ -125,7 +125,7 @@
     }
 }
 
-void lock_release(void)
+void lock_do_release(void)
 /* release a lock on a given host */
 {
     unlink(lockfile);
