--- socket.c.orig	2013-04-24 04:00:45.000000000 +0800
+++ socket.c	2019-09-09 09:11:25.000000000 +0800
@@ -914,7 +914,12 @@
 			return -1;
 #endif
 		} else if(!strcasecmp("ssl3",myproto)) {
+#ifndef OPENSSL_NO_SSL3 
 			_ctx[sock] = SSL_CTX_new(SSLv3_client_method());
+#else
+			report(stderr, GT_("Your operating system does not support SSLv3.\n"));
+			return -1;
+#endif
 		} else if(!strcasecmp("tls1",myproto)) {
 			_ctx[sock] = SSL_CTX_new(TLSv1_client_method());
 		} else if (!strcasecmp("ssl23",myproto)) {
