--- a/plugins/perl/generate_header.py	2018-03-23 01:22:55.000000000 +0100
+++ b/plugins/perl/generate_header.py	2018-03-23 01:23:05.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!@PYTHON3@
 
 import sys
 from os.path import basename
--- a/src/common/make-te.py	2018-03-23 01:23:48.000000000 +0100
+++ b/src/common/make-te.py	2018-03-23 01:23:55.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!@PYTHON3@
 
 import sys
 
--- a/plugins/perl/generate_header	2018-03-23 01:27:25.000000000 +0100
+++ b/plugins/perl/generate_header	2018-03-23 01:27:58.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!@PERL@

 use strict;
 use warnings;
--- a/plugins/python/generate_plugin.py
--- b/plugins/python/generate_plugin.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!@PYTHON3@
 
 import sys
 import cffi
