--- hexchat-2.9.5/configure.ac
+++ hexchat-2.9.5/configure.ac
@@ -968,8 +968,6 @@
 plugins/doat/Makefile
 plugins/fishlim/Makefile
 plugins/sysinfo/Makefile
-intl/Makefile
-po/Makefile.in
 man/Makefile
 ])
 dnl plugins/tcl/Makefile
--- hexchat-2.9.5/Makefile.am
+++ hexchat-2.9.5/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = po intl src plugins man share
+SUBDIRS = src plugins man share
 
 EXTRA_DIST = autogen.sh
 
