--- jmorecfg.h.orig	2013-02-08 22:31:48.000000000 -0800
+++ jmorecfg.h	2013-02-08 22:33:47.000000000 -0800
@@ -252,17 +252,16 @@ typedef void noreturn_t;
  * Defining HAVE_BOOLEAN before including jpeglib.h should make it work.
  */
 
-#ifdef HAVE_BOOLEAN
+#ifndef HAVE_BOOLEAN
+typedef int boolean;
+#endif
+
 #ifndef FALSE			/* in case these macros already exist */
 #define FALSE	0		/* values of boolean */
 #endif
 #ifndef TRUE
 #define TRUE	1
 #endif
-#else
-typedef enum { FALSE = 0, TRUE = 1 } boolean;
-#endif
-
 
 /*
  * The remaining options affect code selection within the JPEG library,
