From 403ae6f35431d66686196d2e2b34bd280f459dab Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 8 Jun 2024 15:21:00 +0800
Subject: [PATCH] CMakeLists: add ppc case

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git CMakeLists.txt CMakeLists.txt
index 5881ea3f..1e3a8aa6 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -80,6 +80,13 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(ppc64|powerpc64)")
     if(${OQS_DIST_BUILD})
         set(OQS_DIST_PPC64_BUILD ON)
     endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(ppc|powerpc)")
+    message(WARNING "There is currently no CI for: " ${CMAKE_SYSTEM_PROCESSOR})
+    set(ARCH "ppc")
+    set(ARCH_PPC ON)
+    if(${OQS_DIST_BUILD})
+        set(OQS_DIST_PPC_BUILD ON)
+    endif()
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
     set(ARCH "s390x")
     set(ARCH_S390X ON)
