--- config.h.in.orig	2013-07-23 01:10:46.000000000 -0400
+++ config.h.in	2013-07-23 09:20:05.000000000 -0400
@@ -705,7 +705,7 @@
 /* Define to 1 if the compiler supports one of the keywords 'inline',
    '__inline__', '__inline' and effectively inlines functions marked as such.
    */
-#undef HAVE_INLINE
+/* #undef HAVE_INLINE */
 
 /* Define if you have the 'intmax_t' type in <stdint.h> or <inttypes.h>. */
 #undef HAVE_INTMAX_T
