--- gildas-src-may15a/admin/define-system.sh.orig	2014-12-09 11:31:56.000000000 +0100
+++ gildas-src-may15a/admin/define-system.sh	2015-05-26 10:30:40.000000000 +0200
@@ -213,8 +213,8 @@
                 fi
         	if which ifort > /dev/null 2>&1; then
         	    DEFAULT_FCOMPILER=ifort
-        	elif which gfortran > /dev/null 2>&1; then
-        	    DEFAULT_FCOMPILER=gfortran
+        	elif which @FC@ > /dev/null 2>&1; then
+        	    DEFAULT_FCOMPILER=@FC@
         	fi
             fi ;;
         CYGWIN*)
@@ -375,10 +375,12 @@
             fi
         fi
         #
-        CPP_VERSION=`cpp --version | head -1 | sed "s%.*[^0-9]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*%\1%"`
+        CPP_VERSION=`@CPP@ --version | head -1 | sed "s%.*[^0-9]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*%\1%"`
         if [ "$CPP_VERSION" ">" "4.7.9" ]; then
             # See http://gcc.gnu.org/gcc-4.8/porting_to.html
             GAG_COMPILER_CPPFLAGS="$GAG_COMPILER_CPPFLAGS -ffreestanding"
+	elif [ "$CPP_VERSION" != "${CPP_VERSION/clang/}" ]; then
+            GAG_COMPILER_CPPFLAGS="$GAG_COMPILER_CPPFLAGS -ffreestanding"
         fi
         #
         export GAG_COMPILER_FFLAGS GAG_COMPILER_FLIBS GAG_COMPILER_CPPFLAGS
--- gildas-src-may15a/admin/define-system.sh.orig	2015-05-27 10:53:14.000000000 +0200
+++ gildas-src-may15a/admin/define-system.sh	2015-05-27 10:53:29.000000000 +0200
@@ -403,7 +403,7 @@
     #  source admin/gildas-env.sh -c ifort  (use default ifort e.g. ifort 11) and
     #  source admin/gildas-env.sh -c ifort14  (i.e. use namely non default ifort 14)
     # does not build their binaries in the same integration branch.
-    GAG_EXEC_SYSTEM=$GAG_MACHINE-$GAG_TARGET_VERS-$GAG_COMPILER_FEXE$GAG_CONFIG
+    GAG_EXEC_SYSTEM=$GAG_MACHINE-$GAG_TARGET_VERS-$GAG_COMPILER_FKIND$GAG_CONFIG
     export GAG_COMP_SYSTEM GAG_EXEC_SYSTEM
     export GAG_MACHINE GAG_OS_KIND GAG_OS_VERS GAG_CONFIG
     export GAG_COMPILER_CKIND GAG_COMPILER_CEXE
