--- gildas-src-may15a/admin/Makefile.def.orig	2015-03-17 10:55:49.000000000 +0100
+++ gildas-src-may15a/admin/Makefile.def	2015-05-21 23:36:25.000000000 +0200
@@ -72,12 +72,12 @@
 # The following variables are provided to the user so that he can override
 # the preprocessing, compilation and linking options defined by default
 # later in this file. User can modify them here.
-CPPFLAGS = $(GAG_CPPFLAGS)
-SLDFLAGS = $(GAG_SLDFLAGS)
-FLDFLAGS = $(GAG_FLDFLAGS)
-CLDFLAGS = $(GAG_CLDFLAGS)
-FFLAGS = $(GAG_FFLAGS)
-CFLAGS = $(GAG_CFLAGS)
+CPPFLAGS = @CPPFLAGS@ $(GAG_CPPFLAGS)
+SLDFLAGS = @LDFLAGS@ $(GAG_SLDFLAGS)
+FLDFLAGS = @LDFLAGS@ $(GAG_FLDFLAGS)
+CLDFLAGS = @LDFLAGS@ $(GAG_CLDFLAGS)
+FFLAGS = @FCFLAGS@ $(GAG_FFLAGS)
+CFLAGS = @CCFLAGS@ $(GAG_CFLAGS)
 
 # The following variables are for mandatory GLOBAL definitions.
 # They should be modified only in this file.
@@ -219,7 +219,7 @@
 # Fortran preprocessing
 
 # Preprocessor command
-CPP = cpp
+CPP = @CPP@
 
 # Preprocessor generic flags
 GLOBAL_CPPFLAGS += -P -traditional -C 
@@ -335,7 +335,7 @@
 # C++ compilation
 
 # C++ compiler name
-CXX = g++
+CXX = @CXX@
 ifdef GAG_COMPILER_CXXEXE
   CXX = $(GAG_COMPILER_CXXEXE)
 endif
@@ -538,7 +538,7 @@
 SIC_SYS_LIBS =
 SIC_LIB_DEPENDS += $(_GAG_LIBS) $(LEGACY_LIBS)
 ifeq ($(PYTHON_PRESENT),yes)
-  SIC_SYS_LIBS += $(shell python $(gagadmdir)/python-config-ldflags.py)
+  SIC_SYS_LIBS += $(shell @PYTHON@ $(gagadmdir)/python-config-ldflags.py)
 endif
 ifeq ($(CFITSIO_PRESENT),yes)
   SIC_LIB_DEPENDS += -lcfitsio