--- gildas-src-jan20a/admin/Makefile.build.orig	2020-01-08 09:56:44.000000000 +0100
+++ gildas-src-jan20a/admin/Makefile.build	2020-01-08 10:02:52.000000000 +0100
@@ -296,10 +296,17 @@
 # Copy into integration branch
 ifdef GAG_USE_PYTHON
 
+# Pre-touch'ing the output file below is needed when building under
+# MacPorts to create the file with proper permissions (umask). Then
+# patch will keep those permissions. Without this pre-touch, patch
+# does not create a file with proper permissions (readable for root
+# only, i.e. files are not accessible to users, even if umask is
+# 0022).
 PYRECIPE = \
 	if [ "$(PYTHON3_PRESENT)" = "yes" ]; then \
 		2to3 $< 2>/dev/null > cl; \
 		if test -s cl; then \
+			touch $@; \
 			patch -i cl -o $@; $(RM) cl; \
 		else \
 			$(CP) $< $@; \
