--- gildas-src-jul18a/admin/gildas-env.sh.orig	2018-06-20 12:05:33.000000000 +0200
+++ gildas-src-jul18a/admin/gildas-env.sh	2018-07-02 17:34:15.000000000 +0200
@@ -252,11 +252,11 @@
 	OLDIFS=$IFS
 	IFS=:
 	for DIR in $GAGENV_SEARCH_PATH; do
-	    if file_present "liblapack." "${DIR}"; then
+	    if file_present "libLAPACK." "${DIR}"; then
 		LAPACK_PRESENT=yes
 		LAPACK_LIB_DIR=$DIR
 	    fi
-	    if file_present "libblas." "${DIR}"; then
+	    if file_present "libBLAS." "${DIR}"; then
 		BLAS_PRESENT=yes
 		BLAS_LIB_DIR=$DIR
 	    fi
@@ -265,13 +265,13 @@
 		SLATEC_LIB_DIR=$DIR
 	    fi
 	    # Note: libfftw3 and libfftw3f must be in the same directory
-	    if file_present "libfftw3." "${DIR}"; then
-		FFTW3_PRESENT=yes
-		FFTW3_LIB_DIR=$DIR
-		if file_present "libfftw3f." "${DIR}"; then
-		    FFTW3F_PRESENT=yes
-		fi
-	    fi
+# 	    if file_present "libfftw3." "${DIR}"; then
+# 		FFTW3_PRESENT=yes
+# 		FFTW3_LIB_DIR=$DIR
+# 		if file_present "libfftw3f." "${DIR}"; then
+# 		    FFTW3F_PRESENT=yes
+# 		fi
+# 	    fi
 	done
 	IFS=$OLDIFS
 	#
@@ -432,24 +432,24 @@
 	    PYTHON_LIB=no
 	    NUMPY_PRESENT=no
 	    SQLITE3_PRESENT=no
-	    if \which python > /dev/null 2>&1; then
+	    if \which @PYTHON@ > /dev/null 2>&1; then
 		# Check the version of python the user is currently using
-		if python -V 2>&1 > /dev/null 2>&1; then
-		    PYTHON_VERS_FULL=`python -V 2>&1 | cut -d' ' -f2`
+		if @PYTHON@ -V 2>&1 > /dev/null 2>&1; then
+		    PYTHON_VERS_FULL=`@PYTHON@ -V 2>&1 | cut -d' ' -f2`
 		    PYTHON_VERS=`echo $PYTHON_VERS_FULL | \sed "s%\([0-9]\.[0-9][0-9]*\).*%\1%"`
 		    # Search for Python header and library. kernel/python/binding/setup.py
 		    # does exactly the same, so this should be consistent
-		    PYTHON_INC_DIR=`python -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'))"`
+		    PYTHON_INC_DIR=`@PYTHON@ -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'))"`
 		    if [ -e "$PYTHON_INC_DIR/Python.h" ]; then PYTHON_INC=yes; else PYTHON_INC=no; fi
-		    PYTHON_LIB_DIR=`python -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBDIR'))"`
+		    PYTHON_LIB_DIR=`@PYTHON@ -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBDIR'))"`
 		    PYTHON_LIB=yes  # libpython name on the various supported systems is just
 				    # unpredictible. Can not check in advance if the linker will
 				    # find it or not. See also our script python-config-ldflags.py
 		    #
 		    # Now search for Numpy
-		    if python -c "import numpy" > /dev/null 2>&1; then
+		    if @PYTHON@ -c "import numpy" > /dev/null 2>&1; then
 			NUMPY_PRESENT=yes
-			NUMPY_INC_DIR=`python -c "import numpy; print(numpy.__path__[0] + '/core/include')"`
+			NUMPY_INC_DIR=`@PYTHON@ -c "import numpy; print(numpy.__path__[0] + '/core/include')"`
 			if [ -e "$NUMPY_INC_DIR/numpy/arrayobject.h" ]; then
 			    NUMPY_INC_PRESENT=yes
 			else
@@ -457,7 +457,7 @@
 			fi
 		    fi
 		    # Search for sqlite3
-		    if python -c "import sqlite3" > /dev/null 2>&1; then
+		    if @PYTHON@ -c "import sqlite3" > /dev/null 2>&1; then
 			SQLITE3_PRESENT=yes
 		    fi
 		fi
@@ -760,6 +760,8 @@
     # environment variables:
     #
     unset GAG_LIB_DEP_PATH GAG_INC_DEP_PATH  # Incremented by detection functions
+    GAG_INC_DEP_PATH=@PREFIX@/include
+    GAG_LIB_DEP_PATH=@PREFIX@/lib
     #
     gagenv_detect_atm
     gagenv_message ""
