From 158b00c3fe7fafe1bcf21471fbe53bc36ea322c5 Mon Sep 17 00:00:00 2001
From: David Martinez <kendersec@gmail.com>
Date: Tue, 19 Jul 2011 19:28:43 +0200
Subject: [PATCH 09/23] Fix for fullscreen on Mac issue931

---
 src/browsermainwindow.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git src/browsermainwindow.cpp src/browsermainwindow.cpp
index ffcb8b8..05f23b1 100644
--- src/browsermainwindow.cpp
+++ src/browsermainwindow.cpp
@@ -1379,6 +1379,7 @@ void BrowserMainWindow::zoomOut()
 void BrowserMainWindow::viewFullScreen(bool makeFullScreen)
 {
     if (makeFullScreen) {
+        setUnifiedTitleAndToolBarOnMac(false);
         setWindowState(windowState() | Qt::WindowFullScreen);
 
         menuBar()->hide();
@@ -1386,6 +1387,7 @@ void BrowserMainWindow::viewFullScreen(bool makeFullScreen)
     } else {
         setWindowState(windowState() & ~Qt::WindowFullScreen);
 
+        setUnifiedTitleAndToolBarOnMac(true);
         menuBar()->setVisible(m_menuBarVisible);
         statusBar()->setVisible(m_statusBarVisible);
     }
-- 
2.44.0

