--- qtbindings/qtbindingsbase.pri.orig	2010-11-03 14:23:27.000000000 -0400
+++ qtbindings/qtbindingsbase.pri	2010-11-03 14:27:39.000000000 -0400
@@ -3,6 +3,8 @@
 INCLUDEPATH += .
 DESTDIR = $$PWD/../plugins/script
 QT += script
-CONFIG += debug_and_release
+CONFIG += @ARCHES@ @BUILD_TYPE@
 GENERATEDCPP = $$PWD/../generated_cpp
 TARGET=$$qtLibraryTarget($$TARGET)
+target.path = $$[QT_INSTALL_PLUGINS]/script
+INSTALLS += target
