--- update-smart-drivedb.in.orig	2010-03-05 06:56:41.000000000 +0900
+++ update-smart-drivedb.in	2010-11-22 05:18:16.000000000 +0900
@@ -40,7 +40,8 @@
 SRCEXPR='http://smartmontools.svn.sourceforge.net/viewvc/smartmontools/$location/smartmontools/drivedb.h?revision=HEAD'
 
 # Convert version into branch name: 5.41[.X] -> "RELEASE_5_41_DRIVEDB"
-BRANCH="`echo $VERSION | sed -n 's|^\([0-9][0-9]*\)\.\([0-9][0-9]*\)\([^0-9].*\)\?$|RELEASE_\1_\2_DRIVEDB|p'`"
+BRANCH="`echo $VERSION | sed 's|^\([0-9]*\.[0-9]*\)\..*$|\1|' \
+         | sed -n 's|^\([0-9][0-9]*\)\.\([0-9][0-9]*\)$|RELEASE_\1_\2_DRIVEDB|p'`"
 
 if [ -z "$BRANCH" ]; then
   echo "$0: syntax error in version number: $VERSION" >&2; exit 1
