--- install.py.old	2014-07-10 20:19:31.000000000 +0200
+++ install.py	2014-07-10 20:20:08.000000000 +0200
@@ -240,7 +240,7 @@
         print '    !!! PyGTK .................... Not found'
         required_found = False
     try:
-        import Image
+        from PIL import Image
         assert Image.VERSION >= '1.1.5'
         print '    Python Imaging Library ....... OK'
     except ImportError:
--- mime/comicthumb.old	2014-07-10 20:19:32.000000000 +0200
+++ mime/comicthumb	2014-07-10 20:20:08.000000000 +0200
@@ -22,7 +22,7 @@
 import subprocess
 
 try:
-    import Image
+    from PIL import Image
 except ImportError:
     print '! Could not import the Image module (PIL).'
     print __doc__
--- src/comix.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/comix.py	2014-07-10 20:20:08.000000000 +0200
@@ -49,7 +49,7 @@
     sys.exit(1)
 
 try:
-    import Image
+    from PIL import Image
     assert Image.VERSION >= '1.1.5'
 except AssertionError:
     print "You don't have the required version of the Python Imaging",
--- src/histogram.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/histogram.py	2014-07-10 20:20:08.000000000 +0200
@@ -1,9 +1,9 @@
 """histogram.py - Draw histograms (RGB) from pixbufs."""
 
 import gtk
-import Image
-import ImageDraw
-import ImageOps
+from PIL import Image
+from PIL import ImageDraw
+from PIL import ImageOps
 
 import image
 
--- src/image.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/image.py	2014-07-10 20:20:08.000000000 +0200
@@ -1,10 +1,10 @@
 """image.py - Various image manipulations."""
 
 import gtk
-import Image
-import ImageEnhance
-import ImageOps
-import ImageStat
+from PIL import Image
+from PIL import ImageEnhance
+from PIL import ImageOps
+from PIL import ImageStat
 
 from preferences import prefs
 
--- src/library.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/library.py	2014-07-10 20:20:08.000000000 +0200
@@ -8,8 +8,8 @@
 import gtk
 import gobject
 import pango
-import Image
-import ImageDraw
+from PIL import Image
+from PIL import ImageDraw
 
 import archive
 import encoding
--- src/thumbbar.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/thumbbar.py	2014-07-10 20:20:08.000000000 +0200
@@ -4,8 +4,8 @@
 
 import gtk
 import gobject
-import Image
-import ImageDraw
+from PIL import Image
+from PIL import ImageDraw
 
 import image
 from preferences import prefs
--- src/thumbnail.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/thumbnail.py	2014-07-10 20:20:08.000000000 +0200
@@ -15,7 +15,7 @@
 import tempfile
 
 import gtk
-import Image
+from PIL import Image
 
 import archive
 import constants
--- src/thumbremover.py.old	2014-07-10 20:19:32.000000000 +0200
+++ src/thumbremover.py	2014-07-10 20:20:08.000000000 +0200
@@ -7,7 +7,7 @@
 
 import gtk
 import pango
-import Image
+from PIL import Image
 
 import encoding
 import labels
