--- configure.orig	2011-07-26 08:42:18.000000000 +1000
+++ configure	2012-07-08 01:55:16.000000000 +1000
@@ -4417,7 +4417,7 @@
 
 
 
-    if test "$GCC" = "yes"; then
+    if false; then
 
     r="`eval echo '$'"CC"`"
     { $as_echo "$as_me:$LINENO: checking $r version" >&5
@@ -4497,7 +4497,6 @@
     else
 	warn="$CC (Please install *GNU* cc)"
 
-    eval "REQUIRED"=\"`eval echo \"'$'REQUIRED\" \"$warn\"`\"
 
     fi
 
@@ -4910,7 +4909,7 @@
 
 
 
-    if test "$GXX" = "yes"; then
+    if false; then
 
     r="`eval echo '$'"CXX"`"
     { $as_echo "$as_me:$LINENO: checking $r version" >&5
@@ -4990,7 +4989,6 @@
     else
 	warn="$CXX (Please install *GNU* c++)"
 
-    eval "REQUIRED"=\"`eval echo \"'$'REQUIRED\" \"$warn\"`\"
 
     fi
 
