--- flex/FlexLexer.h.orig	2021-05-30 16:27:52.000000000 +0200
+++ flex/FlexLexer.h	2021-05-30 16:29:01.000000000 +0200
@@ -129,7 +129,7 @@
 
   virtual ~yyFlexLexer();
 
-  void yy_switch_to_buffer( yy_buffer_state* new_buffer );
+  void yy_switch_to_buffer( yy_buffer_state* new_buffer ) override;
   yy_buffer_state* yy_create_buffer( std::istream* s, int size );
   yy_buffer_state* yy_create_buffer( std::istream& s, int size );
   void yy_delete_buffer( yy_buffer_state* b );
