--- configure.ac.orig	2010-08-06 06:45:34.000000000 -0500
+++ configure.ac	2010-11-14 19:11:45.000000000 -0600
@@ -13,8 +13,24 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_LN_S
+AC_PROG_GREP
+
+# Are we using GNU ld?
+AC_MSG_CHECKING([for GNU ld])
+LD=`$CC -print-prog-name=ld 2>&5`
+
+if test `$LD -v 2>&1 | $ac_cv_path_GREP -c "GNU ld"` = 0; then
+	# Not
+	GNU_LD=""
+	AC_MSG_RESULT([no])
+else
+	GNU_LD="$LD"
+	AC_MSG_RESULT([yes])
+	AC_SUBST([GNU_LD])
+fi
 
 AC_CHECK_HEADERS([sys/select.h alloca.h])
+AC_CHECK_HEADERS([popt.h libintl.h])
 
 AC_MSG_CHECKING([for python versions])
 PYTHONVERS=$(echo /usr/include/python*/Python.h | sed "s|/usr/include/||g"| sed "s|/Python.h||g" 2> /dev/null)
