--- src/dbarts/Makefile.orig	2023-01-21 01:54:34.000000000 +0800
+++ src/dbarts/Makefile	2023-04-13 05:47:53.000000000 +0800
@@ -13,6 +13,7 @@
 
 library : $(LOCAL_OBJECTS)
 	$(AR) $(ARFLAGS) ../dbarts.a $(LOCAL_OBJECTS)
+	$(RANLIB) ../dbarts.a
 
 clean :
 	@-rm -f config.hpp

--- src/external/Makefile.orig	2023-01-21 01:54:34.000000000 +0800
+++ src/external/Makefile	2023-04-13 05:48:13.000000000 +0800
@@ -9,6 +9,7 @@
 
 library : $(LOCAL_OBJECTS)
 	$(AR) $(ARFLAGS) ../external.a $(LOCAL_OBJECTS)
+	$(RANLIB) ../external.a
 
 clean :
 	@-rm -f config.h

--- src/misc/Makefile.orig	2023-01-21 01:54:34.000000000 +0800
+++ src/misc/Makefile	2023-04-13 05:48:41.000000000 +0800
@@ -57,6 +57,7 @@
 
 library : $(LOCAL_OBJECTS)
 	$(AR) $(ARFLAGS) ../misc.a $(LOCAL_OBJECTS)
+	$(RANLIB) ../misc.a
 
 clean :
 	@-rm -f config.h

--- src/rc/Makefile.orig	2023-01-21 01:54:34.000000000 +0800
+++ src/rc/Makefile	2023-04-13 05:48:57.000000000 +0800
@@ -9,6 +9,7 @@
 
 library : $(LOCAL_OBJECTS)
 	$(AR) $(ARFLAGS) ../rc.a $(LOCAL_OBJECTS)
+	$(RANLIB) ../rc.a
 
 clean :
 	@-rm -f config.h
