--- detect_cpu_asm.s.orig	2006-09-13 15:51:20.000000000 -0600
+++ detect_cpu_asm.s	2008-10-21 23:25:59.000000000 -0600
@@ -52,9 +52,9 @@
     check if cpuid is available
 */
 
-    .globl  x86_check_cpuid_flag
+    .globl  _x86_check_cpuid_flag
 
-x86_check_cpuid_flag:
+_x86_check_cpuid_flag:
 	/* save ecx register in the stack */
 	pushl	%ecx
 
@@ -109,9 +109,9 @@
     returns: pointer to static buffer
 */
 
-    .globl x86_exec_cpuid
+    .globl _x86_exec_cpuid
 
-x86_exec_cpuid:
+_x86_exec_cpuid:
     /* get function number */
     movl    4(%esp),%eax
 
@@ -122,10 +122,10 @@
     cpuid
 
     /* copy register values */
-    movl    %eax,x86_cpu_reg_eax
-    movl    %ebx,x86_cpu_reg_ebx
-    movl    %ecx,x86_cpu_reg_ecx
-    movl    %edx,x86_cpu_reg_edx
+    movl    %eax,_x86_cpu_reg_eax
+    movl    %ebx,_x86_cpu_reg_ebx
+    movl    %ecx,_x86_cpu_reg_ecx
+    movl    %edx,_x86_cpu_reg_edx
 
     popl    %edx
     popl    %ecx
@@ -152,9 +152,9 @@
     check if cpuid is available
 */
 
-    .globl  x86_check_cpuid_flag
+    .globl  _x86_check_cpuid_flag
 
-x86_check_cpuid_flag:
+_x86_check_cpuid_flag:
 	pushq	%rcx        /* save rcx register */
 
 	pushfq
@@ -186,9 +186,9 @@
     returns: pointer to static buffer
 */
 
-    .globl x86_exec_cpuid
+    .globl _x86_exec_cpuid
 
-x86_exec_cpuid:
+_x86_exec_cpuid:
     /* get function number (arg0 in rdi register) */
     movq    %rdi,%rax
 
@@ -199,10 +199,10 @@
     cpuid
 
     /* copy register values */
-    movl    %eax,x86_cpu_reg_eax
-    movl    %ebx,x86_cpu_reg_ebx
-    movl    %ecx,x86_cpu_reg_ecx
-    movl    %edx,x86_cpu_reg_edx
+    movl    %eax,_x86_cpu_reg_eax
+    movl    %ebx,_x86_cpu_reg_ebx
+    movl    %ecx,_x86_cpu_reg_ecx
+    movl    %edx,_x86_cpu_reg_edx
 
     popq    %rdx
     popq    %rcx
@@ -218,20 +218,20 @@
 
 .data
 
-    .globl x86_cpu_reg_eax
-x86_cpu_reg_eax:
+    .globl _x86_cpu_reg_eax
+_x86_cpu_reg_eax:
     .long 0
 
-    .globl x86_cpu_reg_ebx
-x86_cpu_reg_ebx:
+    .globl _x86_cpu_reg_ebx
+_x86_cpu_reg_ebx:
     .long 0
 
-    .globl x86_cpu_reg_ecx
-x86_cpu_reg_ecx:
+    .globl _x86_cpu_reg_ecx
+_x86_cpu_reg_ecx:
     .long 0
 
-    .globl x86_cpu_reg_edx
-x86_cpu_reg_edx:
+    .globl _x86_cpu_reg_edx
+_x86_cpu_reg_edx:
     .long 0
 
 #endif /* ARCH_X86_32 || ARCH_X86_64 */
