--- pyproject.toml.orig	2026-04-07 23:30:48
+++ pyproject.toml	2026-04-08 07:55:09
@@ -1,6 +1,6 @@
 [build-system]
 build-backend = "setuptools.build_meta"
-requires = ["setuptools>=82.0.1"]
+requires = ["setuptools"]
 
 [project]
 name = "types-PyYAML"
@@ -49,5 +49,5 @@
 packages = ['yaml-stubs']
 include-package-data = false
 
-[tool.setuptools.package-data]
-'yaml-stubs' = ['__init__.pyi', '_yaml.pyi', 'composer.pyi', 'constructor.pyi', 'cyaml.pyi', 'dumper.pyi', 'emitter.pyi', 'error.pyi', 'events.pyi', 'loader.pyi', 'nodes.pyi', 'parser.pyi', 'reader.pyi', 'representer.pyi', 'resolver.pyi', 'scanner.pyi', 'serializer.pyi', 'tokens.pyi', 'METADATA.toml', 'py.typed']
+# [tool.setuptools.package-data]
+# 'yaml-stubs' = ['__init__.pyi', '_yaml.pyi', 'composer.pyi', 'constructor.pyi', 'cyaml.pyi', 'dumper.pyi', 'emitter.pyi', 'error.pyi', 'events.pyi', 'loader.pyi', 'nodes.pyi', 'parser.pyi', 'reader.pyi', 'representer.pyi', 'resolver.pyi', 'scanner.pyi', 'serializer.pyi', 'tokens.pyi', 'METADATA.toml', 'py.typed']
