--- ace/CDR_Base.inl.orig	2011-06-03 21:31:31.000000000 -0700
+++ ace/CDR_Base.inl	2011-06-03 21:30:57.000000000 -0700
@@ -174,10 +174,10 @@
   // 64 bit architecture.
   register unsigned long x =
     * reinterpret_cast<const unsigned long*> (orig);
-  register unsigned long x84 = (x & 0x000000ff000000ffUL) << 24;
-  register unsigned long x73 = (x & 0x0000ff000000ff00UL) << 8;
-  register unsigned long x62 = (x & 0x00ff000000ff0000UL) >> 8;
-  register unsigned long x51 = (x & 0xff000000ff000000UL) >> 24;
+  register unsigned long x84 = (x & 0x000000ff000000ffLLU) << 24;
+  register unsigned long x73 = (x & 0x0000ff000000ff00LLU) << 8;
+  register unsigned long x62 = (x & 0x00ff000000ff0000LLU) >> 8;
+  register unsigned long x51 = (x & 0xff000000ff000000LLU) >> 24;
   x = (x84 | x73 | x62 | x51);
   x = (x << 32) | (x >> 32);
   *reinterpret_cast<unsigned long*> (target) = x;
