--- src/CMakeLists.txt.orig	2021-12-07 21:15:06.000000000 -0800
+++ src/CMakeLists.txt	2021-12-07 21:32:38.000000000 -0800
@@ -443,7 +443,7 @@
       set(LLDB_LIBRARY_INSTALL_NAME "${LLDB_LIBRARY}/Versions/Current/LLDB")
 	  add_custom_command(TARGET ${KCOV}
         POST_BUILD COMMAND
-        "${CMAKE_INSTALL_NAME_TOOL}" -change @rpath/LLDB.framework/LLDB "${LLDB_LIBRARY_INSTALL_NAME}" $<TARGET_FILE:${KCOV}>
+        "${CMAKE_INSTALL_NAME_TOOL}" -change @rpath/LLDB.framework/LLDB "${LLDB_LIBRARY_INSTALL_NAME}" -change @rpath/LLDB.framework/Versions/A/LLDB "${LLDB_LIBRARY_INSTALL_NAME}" $<TARGET_FILE:${KCOV}>
         COMMENT "Changing LLDB install name from @rpath/LLDB.framework/LLDB to ${LLDB_LIBRARY_INSTALL_NAME}")
 	endif()
 endif()
