--- utils/rwphase.hpp.orig	2012-05-29 19:07:52.000000000 -0700
+++ utils/rwphase.hpp	2019-09-09 20:51:12.000000000 -0700
@@ -17,6 +17,8 @@
 
 #include <boost/thread.hpp>
 
+#include <boost/core/noncopyable.hpp>
+
 #include <utils/atomicop.hpp>
 
 namespace utils
--- utils/rwspinlock.hpp.orig	2012-05-27 01:08:52.000000000 -0700
+++ utils/rwspinlock.hpp	2019-09-09 20:53:38.000000000 -0700
@@ -22,6 +22,8 @@
 
 #include <boost/thread.hpp>
 
+#include <boost/core/noncopyable.hpp>
+
 #include <utils/atomicop.hpp>
 
 namespace utils
--- utils/rwticket.hpp.orig	2012-05-29 17:29:22.000000000 -0700
+++ utils/rwticket.hpp	2019-09-09 20:54:27.000000000 -0700
@@ -12,6 +12,8 @@
 
 #include <boost/thread.hpp>
 
+#include <boost/core/noncopyable.hpp>
+
 #include <utils/atomicop.hpp>
 
 namespace utils
--- utils/json_string_main.cpp.orig	2012-04-08 23:58:37.000000000 -0700
+++ utils/json_string_main.cpp	2019-09-09 20:58:21.000000000 -0700
@@ -2,6 +2,8 @@
 #include "utils/json_string_parser.hpp"
 #include "utils/json_string_generator.hpp"
 
+#include <boost/numeric/conversion/bounds.hpp>
+
 #include <iostream>
 #include <iterator>
 
