--- libntp/work_fork.c.orig	2018-08-29 17:27:26.000000000 -0400
+++ libntp/work_fork.c	2018-08-29 17:29:00.000000000 -0400
@@ -88,8 +88,9 @@
 	return (size_t)(b - (const char *)vb);
 }
 
-
+#if defined(HAVE_DROPROOT)
 int set_user_group_ids(void);
+#endif
 
 /* === functions === */
 /*
@@ -594,7 +595,10 @@
 	init_logging("ntp_intres", 0, FALSE);
 	setup_logfile(NULL);
 
+#if defined(HAVE_DROPROOT)
 	(void) set_user_group_ids();
+#endif
+
 
 	/*
 	 * And now back to the portable code
