From eb0fcfd1534b8bd92f8008ae8c9beb5deb5ad353 Mon Sep 17 00:00:00 2001
From: Juan Lang <juan.lang@gmail.com>
Date: Fri, 4 Dec 2009 16:41:53 -0800
Subject: [PATCH] Use sk_ functions rather than accessing an OpenSSL stack type directly

---
 dlls/winhttp/net.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/dlls/winhttp/net.c b/dlls/winhttp/net.c
index 3ed3b1b..4a3cc90 100644
--- dlls/winhttp/net.c
+++ dlls/winhttp/net.c
@@ -124,6 +124,8 @@ MAKE_FUNCPTR( CRYPTO_set_locking_callback );
 MAKE_FUNCPTR( ERR_get_error );
 MAKE_FUNCPTR( ERR_error_string );
 MAKE_FUNCPTR( i2d_X509 );
+MAKE_FUNCPTR( sk_value );
+MAKE_FUNCPTR( sk_num );
 #undef MAKE_FUNCPTR
 
 static CRITICAL_SECTION *ssl_locks;
@@ -339,11 +341,11 @@ static int netconn_secure_verify( int preverify_ok, X509_STORE_CTX *ctx )
             PCCERT_CONTEXT endCert = NULL;
 
             ret = TRUE;
-            for (i = 0; ret && i < ctx->chain->num; i++)
+            for (i = 0; ret && i < psk_num((struct stack_st *)ctx->chain); i++)
             {
                 PCCERT_CONTEXT context;
 
-                cert = (X509 *)ctx->chain->data[i];
+                cert = (X509 *)psk_value((struct stack_st *)ctx->chain, i);
                 if ((context = X509_to_cert_context( cert )))
                 {
                     if (i == 0)
@@ -441,6 +443,8 @@ BOOL netconn_init( netconn_t *conn, BOOL secure )
     LOAD_FUNCPTR( ERR_get_error );
     LOAD_FUNCPTR( ERR_error_string );
     LOAD_FUNCPTR( i2d_X509 );
+    LOAD_FUNCPTR( sk_value );
+    LOAD_FUNCPTR( sk_num );
 #undef LOAD_FUNCPTR
 
     pSSL_library_init();
-- 
1.5.2.2
