--- sq/Makefile.orig	2012-02-11 11:55:55.000000000 -0500
+++ sq/Makefile	2013-02-26 00:49:04.000000000 -0500
@@ -11,8 +11,8 @@
 SRCS= sq.c
 	
 	
-sq32:
-	g++ -O2 -fno-exceptions -fno-rtti -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
+all:
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -fno-exceptions -fno-rtti -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
 
 sqprof:
 	g++ -O2 -pg -fno-exceptions -fno-rtti -pie -gstabs -g3 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
