--- setup.py	2008-02-14 21:48:02.000000000 +0100
+++ setup.py	2008-02-23 14:56:34.000000000 +0100
@@ -28,4 +28,4 @@
     long_description = "\n".join(pydot.__doc__.split('\n')),
     py_modules = ['pydot', 'dot_parser'],
     install_requires = ['pyparsing', 'setuptools'],
-    data_files = [('.', ['ChangeLog', 'LICENSE', 'README'])] )
+    data_files = [('share/doc/py-dot', ['ChangeLog', 'LICENSE', 'README'])] )
