--- flute/main.c.orig	Sun May  2 23:22:01 2004
+++ flute/main.c	Wed May 19 15:34:20 2004
@@ -834,7 +834,7 @@
                                 exit(-1);
                         }
 
-                        if(errno == ERANGE && a->toi == 0xFFFFFFFFFFFFFFFF) {
+                        if(errno == ERANGE && a->toi == 0xFFFFFFFFFFFFFFFFULL) {
                                 printf("TOI too big for unsigned long long (64 bits)\n");
                                 fflush(stdout);
                                 exit(-1);
@@ -851,8 +851,8 @@
 #ifdef WIN32			  
 						a->duration = _atoi64(&argv[i][3]);
 
-						if(a->duration > 0xFFFFFFFFFFFFFFFF) {
-							printf("Duration: %I64u too big (max=%I64u)\n", a->duration,  0xFFFFFFFFFFFFFFFF);
+						if(a->duration > 0xFFFFFFFFFFFFFFFFULL) {
+							printf("Duration: %I64u too big (max=%I64u)\n", a->duration,  0xFFFFFFFFFFFFFFFFULL);
                             fflush(stdout);
                             exit(-1);
 						}
@@ -867,7 +867,7 @@
                             exit(-1);
                         }
 
-                        if(errno == ERANGE && a->duration == 0xFFFFFFFFFFFFFFFF) {
+                        if(errno == ERANGE && a->duration == 0xFFFFFFFFFFFFFFFFULL) {
                             printf("Duration too big for unsigned long long (64 bits)\n");
                             fflush(stdout);
                             exit(-1);
@@ -917,17 +917,17 @@
                             exit(-1);
 						}
 
-           				if(errno == ERANGE && tmp_tsi == 0xFFFFFFFFFFFFFFFF) {   
+           				if(errno == ERANGE && tmp_tsi == 0xFFFFFFFFFFFFFFFFULL) {   
 							printf("TSI too big for unsigned long long (64 bits)\n");
                             fflush(stdout);
                             exit(-1);
 						}
 #endif 
-						if(tmp_tsi > 0xFFFFFFFFFFFF) {
+						if(tmp_tsi > 0xFFFFFFFFFFFFULL) {
 #ifdef WIN32
 							printf("TSI: %I64u too big (max=%I64u)\n", tmp_tsi, 0xFFFFFFFFFFFF);
 #else
-							printf("TSI: %llu too big (max=%llu)\n", tmp_tsi, 0xFFFFFFFFFFFF);
+							printf("TSI: %llu too big (max=%llu)\n", tmp_tsi, 0xFFFFFFFFFFFFULL);
 #endif
                             fflush(stdout);
                             exit(-1);
