diff -r -U2 -bwB evince-3.15.90_org/backend/dvi/mdvi-lib/fontmap.c evince-3.15.90/backend/dvi/mdvi-lib/fontmap.c
--- backend/dvi/mdvi-lib/fontmap.c	2014-04-14 18:19:45.000000000 +0200
+++ backend/dvi/mdvi-lib/fontmap.c	2015-03-23 06:59:24.319595671 +0100
@@ -513,5 +513,4 @@
 		ps_name     = NULL;
 		vec_name    = NULL;
-		is_encoding = 0;
 
 		if(ent == NULL) {
@@ -522,4 +521,5 @@
 		}
 		while(*ptr) {
+		        is_encoding = 0;
 			char	*hdr_name = NULL;
 			
