--- configure.ac	2008/03/10 20:06:22	2964
+++ configure.ac	2008/03/14 11:13:01	2965
@@ -218,6 +218,15 @@
 	    AC_CHECK_FUNCS(poppler_page_get_image)
 	    LIBS=$evince_save_LIBS
 
+	    POPPLER_MAJOR_VERSION=`$PKG_CONFIG --modversion poppler-glib | cut -d '.' -f 2` 
+	    POPPLER_MINOR_VERSION=`$PKG_CONFIG --modversion poppler-glib | cut -d '.' -f 3`
+	    if test -z "$POPPLER_MINOR_VERSION"; then
+	      POPPLER_MINOR_VERSION=0
+	    fi
+
+	    AC_DEFINE_UNQUOTED([POPPLER_MAJOR_VERSION], $POPPLER_MAJOR_VERSION, [Poppler major version number])
+	    AC_DEFINE_UNQUOTED([POPPLER_MINOR_VERSION], $POPPLER_MINOR_VERSION, [Poppler minor version number])
+
 	    PKG_CHECK_MODULES(CAIRO_PDF, cairo-pdf, enable_cairo_pdf=yes, enable_cairo_pdf=no)
 	    if test x$enable_cairo_pdf == xyes; then
 	            AC_DEFINE([HAVE_CAIRO_PDF], [1], [defined if cairo-pdf is available])
