diff --git a/libtool-kill-dependency_libs.sh b/libtool-kill-dependency_libs.sh
index 73489df33..f55fa0cdb 100755
--- libtool-kill-dependency_libs.sh
+++ libtool-kill-dependency_libs.sh
@@ -50,10 +50,10 @@ done
 #echo "${args[@]}"
 "${args[@]}"
 
-if [ -n "$output" ]; then
+case $output in *.la)
     cp -p "$output" "$output.tmp"
 
     # Remove dependency_libs from output.
     sed "s/^dependency_libs=.*/dependency_libs=''/" < "$output.tmp" > "$output"
     rm "$output.tmp"
-fi
+esac
