--- convmv.orig	2008-12-11 18:38:34.000000000 -0600
+++ convmv	2011-05-24 11:53:29.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!/usr/bin/env perl
 # convmv 1.14 - converts filenames from one encoding to another
 # Copyright © 2003-2008 Bjoern JACKE <bjoern@j3e.de>
 #
@@ -577,10 +577,8 @@
 	if (-e $newname and !$opt_exec) {
 		if ($opt_replace and !&compare($oldfile,$newname)) {
 			if ($opt_notest) {
-				unlink $newname or print $outerr "Error: $!\n";
 				rename ($oldfile, $newname) or print $outerr "Error: $!\n";
 			} elsif ($opt_parsable) {
-				print "unlink".$del.$dir."/".$oldfile.$fin_del;
 				print "rename".$del.$dir."/".$oldfile.$del.$dir."/".$newname.$fin_del;
 			}
 		} else {
