--- config.mk	2013-12-28 04:43:43.000000000 -0800
+++ config.mk	2014-02-19 09:51:56.000000000 -0800
@@ -6,7 +6,7 @@
 PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/share/man
 
-INCS = -I. -I/usr/include -I/usr/local/include
+INCS = -I. -I${PREFIX}/include
 LIBS = -lc -lutil -lncursesw
 # NetBSD
 #LIBS = -lc -lutil -lcurses
@@ -16,7 +16,7 @@
 #INCS += -I/usr/include/ncurses
 
 CFLAGS += -std=c99 -Os ${INCS} -DVERSION=\"${VERSION}\" -DNDEBUG
-LDFLAGS += -L/usr/lib -L/usr/local/lib ${LIBS}
+LDFLAGS += -L${PREFIX}/lib ${LIBS}
 
 DEBUG_CFLAGS = ${CFLAGS} -UNDEBUG -O0 -g -ggdb -Wall -Wextra -Wno-missing-field-initializers -Wno-unused-parameter
 
