--- src/initiator/libkmod.c	2011-08-26 06:51:09.000000000 -0700
+++ src/initiator/libkmod.c	2011-08-26 07:52:37.000000000 -0700
@@ -26,6 +26,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#if defined(__NetBSD__) && defined(USE_LIBKMOD)
 #include <sys/cdefs.h>
 #ifndef lint
 __RCSID("$NetBSD: libkmod.c,v 1.2 2009/06/30 02:44:52 agc Exp $");
@@ -45,8 +46,6 @@
 
 #include "libkmod.h"
 
-#ifdef USE_LIBKMOD
-
 static const char *classes[] = {
 	"any",
 	"misc",
