--- Modules/Filtering/LabelMap/wrapping/itkShapeLabelMapFilter.wrap.orig	2020-07-24 03:47:25
+++ Modules/Filtering/LabelMap/wrapping/itkShapeLabelMapFilter.wrap	2025-08-22 11:29:49
@@ -1,6 +1,6 @@
 itk_wrap_class("itk::ShapeLabelMapFilter" POINTER)
   itk_wrap_filter_dims(restricted_dims 2+)
   foreach(d ${restricted_dims})
-    itk_wrap_template("${ITKM_LM${d}}" "${ITKT_LM${d}}")
+    itk_wrap_template("${ITKM_LM${d}}${ITKM_IUL${d}}" "${ITKT_LM${d}}, ${ITKT_IUL${d}}")
   endforeach()
 itk_end_wrap_class()
--- Modules/Segmentation/LevelSets/wrapping/itkNarrowBandLevelSetImageFilter.wrap.orig	2020-07-24 03:47:25
+++ Modules/Segmentation/LevelSets/wrapping/itkNarrowBandLevelSetImageFilter.wrap	2025-08-22 11:33:40
@@ -2,7 +2,8 @@
 
   foreach(d ${ITK_WRAP_IMAGE_DIMS})
     foreach(t ${WRAP_ITK_REAL})
-      itk_wrap_template("${ITKM_I${t}${d}}${ITKM_I${t}${d}}${ITKM_${t}}" "${ITKT_I${t}${d}},${ITKT_I${t}${d}},${ITKT_${t}}")
+      itk_wrap_template("${ITKM_I${t}${d}}${ITKM_I${t}${d}}${ITKM_${t}}${ITKM_I${t}${d}}"
+                        "${ITKT_I${t}${d}},${ITKT_I${t}${d}},${ITKT_${t}},${ITKT_I${t}${d}}")
     endforeach()
   endforeach()
 
