From 2dc5256e43045a97589a0cfb3cc872df0e2c5fa2 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Thu, 18 Jan 2024 03:28:42 +0800
Subject: [PATCH] test_tcp_port: ensure no type error on macOS

---
 r/src/test_tcp_port.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git r/src/test_tcp_port.c r/src/test_tcp_port.c
index 991a6d0..288ed8a 100644
--- src/test_tcp_port.c
+++ src/test_tcp_port.c
@@ -8,6 +8,10 @@
 #include <netinet/in.h>
 #endif
 
+#ifdef __APPLE__
+  typedef uint32_t u_int32_t;
+#endif
+
 // Adopted from https://github.com/ropensci/ssh/blob/master/src/tunnel.c
 // which is released under the MIT license
 static int test_tcp_port(int port) {
