From 2fed131a3e4a52b9cd9ef07d400ee16915d23d2a Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Thu, 17 Jul 2025 11:25:16 +0200
Subject: [PATCH] use zend_ce_exception instead of zend_exception_get_default()
 for 8.5

---
 gmagick.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gmagick.c b/gmagick.c
index 5102a42..72571c0 100644
--- a/gmagick.c
+++ b/gmagick.c
@@ -272,11 +272,11 @@ PHP_MINIT_FUNCTION(gmagick)
 
 	/* Exception */
 	INIT_CLASS_ENTRY(ce, "GmagickException", NULL);
-	php_gmagick_exception_class_entry = zend_register_internal_class_ex(&ce, zend_exception_get_default());
+	php_gmagick_exception_class_entry = zend_register_internal_class_ex(&ce, zend_ce_exception);
 	php_gmagick_exception_class_entry->ce_flags |= ZEND_ACC_FINAL;
 
 	INIT_CLASS_ENTRY(ce, "GmagickPixelException", NULL);
-	php_gmagickpixel_exception_class_entry = zend_register_internal_class_ex(&ce, zend_exception_get_default());
+	php_gmagickpixel_exception_class_entry = zend_register_internal_class_ex(&ce, zend_ce_exception);
 	php_gmagickpixel_exception_class_entry->ce_flags |= ZEND_ACC_FINAL;
 
 	/* Class entry */
