From 135e00737095b01b0becfb8ef7410654ef1937fe Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Tue, 30 Jan 2024 10:30:19 +0100
Subject: [PATCH] Fix incompatible pointer types

---
 gmagick_methods.c     | 2 +-
 gmagickdraw_methods.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gmagick_methods.c b/gmagick_methods.c
index 2f60add..270d5fa 100644
--- a/gmagick_methods.c
+++ b/gmagick_methods.c
@@ -4975,7 +4975,7 @@ PHP_METHOD(Gmagick, getimagechannelextrema)
 {
 	php_gmagick_object *intern;
 	long channel_type;
-	size_t minima, maxima;
+	unsigned long minima, maxima;
 	unsigned int status;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &channel_type) == FAILURE) {
diff --git a/gmagickdraw_methods.c b/gmagickdraw_methods.c
index b6f4f4f..eb18913 100644
--- a/gmagickdraw_methods.c
+++ b/gmagickdraw_methods.c
@@ -1409,7 +1409,7 @@ static zend_bool php_gmagick_check_font(char *font, int font_len TSRMLS_DC)
 	zend_bool retval = 0;
 	char **fonts;
 	unsigned long i = 0;
-	size_t num_fonts = 0;
+	unsigned long num_fonts = 0;
 
 	/* Check that user is only able to set a proper font */
 	fonts = MagickQueryFonts("*", &num_fonts);
