--- widget/src/gtk2/nsFilePicker.cpp.bak	2009-04-03 18:35:03.000000000 -0700
+++ widget/src/gtk2/nsFilePicker.cpp	2009-04-03 18:38:58.000000000 -0700
@@ -203,7 +203,11 @@ nsFilePicker::LoadSymbolsGTK24()
     _gtk_file_chooser_get_filename = (_gtk_file_chooser_get_filename_fn)func;
   } else {
     // XXX hmm, this seems to fail when gtk 2.4 is already loaded...
+#ifdef __APPLE__
+    mGTK24 = PR_LoadLibrary("libgtk-x11-2.0.dylib");
+#else
     mGTK24 = LoadVersionedLibrary("gtk-2", ".4");
+#endif
     if (!mGTK24) {
       return NS_ERROR_NOT_AVAILABLE;
     }
