--- config/rules.mk.orig	2006-11-28 18:03:37.000000000 +0100
+++ config/rules.mk	2007-11-05 10:38:28.000000000 +0100
@@ -1018,9 +1018,10 @@
 ifndef NO_COMPONENT_LINK_MAP
 ifndef MOZ_COMPONENTS_VERSION_SCRIPT_LDFLAGS
 ifndef MOZ_DEBUG
-ifeq ($(OS_ARCH)_$(IS_COMPONENT),Darwin_1)
-	nmedit -s $(BUILD_TOOLS)/gnu-ld-scripts/components-export-list $@
-endif
+# Export list looks empty
+#ifeq ($(OS_ARCH)_$(IS_COMPONENT),Darwin_1)
+#	nmedit -s $(BUILD_TOOLS)/gnu-ld-scripts/components-export-list $@
+#endif
 endif
 endif
 endif
