--- CMakeLists.txt.orig	2022-12-14 13:06:39.000000000 -0600
+++ CMakeLists.txt	2022-12-15 07:24:53.000000000 -0600
@@ -88,18 +88,18 @@
                          "${WARNING_BORDER}")
 
 # Python 3 is only needed here to check for configuration warnings.
-if(NOT CMAKE_VERSION VERSION_LESS 3.15.0)
-    set(Python3_FIND_STRATEGY LOCATION)
-    find_package(Python3 COMPONENTS Interpreter)
-    if(Python3_Interpreter_FOUND)
-        set(MBEDTLS_PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
-    endif()
-else()
-    find_package(PythonInterp 3)
-    if(PYTHONINTERP_FOUND)
-        set(MBEDTLS_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE})
-    endif()
-endif()
+# if(NOT CMAKE_VERSION VERSION_LESS 3.15.0)
+#     set(Python3_FIND_STRATEGY LOCATION)
+#     find_package(Python3 COMPONENTS Interpreter)
+#     if(Python3_Interpreter_FOUND)
+#         set(MBEDTLS_PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
+#     endif()
+# else()
+#     find_package(PythonInterp 3)
+#     if(PYTHONINTERP_FOUND)
+#         set(MBEDTLS_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE})
+#     endif()
+# endif()
 if(MBEDTLS_PYTHON_EXECUTABLE)
 
     # If 128-bit keys are configured for CTR_DRBG, display an appropriate warning
