diff --git gegl/meson.build gegl/meson.build
index 12c753c..95ed1b6 100644
--- gegl/meson.build
+++ gegl/meson.build
@@ -80,7 +80,7 @@ install_headers(gegl_headers,
   subdir: api_name
 )
 
-gegl_ldflags = os_osx ? ['-framework', 'OpenCL'] : []
+gegl_ldflags = []
 
 gegl_lib = library(api_name,
   gegl_sources,
diff --git gegl/opencl/gegl-cl-init.c gegl/opencl/gegl-cl-init.c
index 22bdbb8..07d8ec8 100644
--- gegl/opencl/gegl-cl-init.c
+++ gegl/opencl/gegl-cl-init.c
@@ -689,7 +689,7 @@ gegl_cl_init_common (cl_device_type          requested_device_type,
           if (!gegl_cl_init_get_gl_sharing_props (gl_contex_props, error))
             return FALSE;
 
-#ifdef __APPLE__
+#if __APPLE__ && MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__POWERPC__)
           /* Create context */
           ctx = gegl_clCreateContext (gl_contex_props, 0, 0, NULL, 0, &err);
 
