--- operations/external/ff-load.c	2012-04-01 13:17:57.000000000 +0200
+++ operations/external/ff-load.c	2012-07-11 12:42:05.174756560 +0200
@@ -271,7 +271,7 @@
       gint err;
 
       ff_cleanup (o);
-      err = av_open_input_file (&p->ic, o->path, NULL, 0, NULL);
+      err = avformat_open_input (&p->ic, o->path, NULL, NULL);
       if (err < 0)
         {
           print_error (o->path, err);
