From b26ef9a733938ed4693d7b3158f13e46c2c9173e Mon Sep 17 00:00:00 2001
From: Stefan Westerfeld <stefan@space.twc.de>
Date: Sun, 25 Feb 2024 11:07:38 +0100
Subject: [PATCH] Add missing include for macOS High Sierra build.

See #50.

Signed-off-by: Stefan Westerfeld <stefan@space.twc.de>
--- src/sfinputstream.cc
+++ src/sfinputstream.cc
@@ -20,6 +20,7 @@
 #include <assert.h>
 #include <string.h>
 #include <unistd.h>
+#include <errno.h>
 
 using std::string;
 using std::vector;
