--- gpgwrap/src/Makefile.orig	2011-12-10 01:16:21.000000000 +0100
+++ gpgwrap/src/Makefile	2011-12-10 01:19:41.000000000 +0100
@@ -1,4 +1,6 @@
-CC=${DIET} gcc -s -Wall -O2 -fomit-frame-pointer
+CC ?= gcc
+CFLAGS ?= -Wall -O2
+CFLAGS += -s -fomit-frame-pointer
 STRIP=strip -R .note -R .comment
 RM=rm -f
 TARGET=../bin/gpgwrap
@@ -8,7 +10,7 @@
 all: ${TARGET}
 
 ${TARGET}: gpgwrap.c
-	${CC} -o ${TARGET} gpgwrap.c
+	${CC} ${CFLAGS} -o ${TARGET} gpgwrap.c
 	${STRIP} ${TARGET} 2>/dev/null || true
 
 clean:
