--- keyanalyze/Makefile.orig	2014-06-21 17:33:36.000000000 +0200
+++ keyanalyze/Makefile	2014-07-13 13:16:02.000000000 +0200
@@ -7,7 +7,7 @@
 process_keys: process_keys.o
 
 pgpring/pgpring:
-	cd pgpring && touch AUTHORS ChangeLog INSTALL NEWS README && automake-1.11 --add-missing && automake-1.11
+	cd pgpring && touch AUTHORS ChangeLog INSTALL NEWS README && aclocal-17 && automake-17 --add-missing && automake-17
 	cd pgpring && autoconf && CFLAGS="${CFLAGS}" ./configure
 	make -C pgpring pgpring
 
