--- myocamlbuild.ml.orig	2009-09-28 15:27:50.000000000 +0100
+++ myocamlbuild.ml	2009-09-28 15:28:06.000000000 +0100
@@ -305,7 +305,7 @@
              Echo([substitute [("@VERSION@", get_version ())] (read_file "META.in")], "META"));
 
         (* Generation of the lwt.odocl file *)
-        let deps = ["src/lwt_extra.mllib"; "src/lwt.mllib"; "src/lwt_preemptive.mllib"; "src/lwt_ssl.mllib"]
+        let deps = ["src/lwt_extra.mllib"; "src/lwt.mllib"; "src/lwt_preemptive.mllib"; "src/lwt_ssl.mllib"; "src/lwt_unix.mllib"]
         and prod = "lwt.odocl" in
         rule "lwt_doc" ~prod ~deps
           (fun _ _ -> Echo(List.map (sprintf "src/%s\n")
