--- CMakeLists.txt.orig	2021-12-09 03:25:45.000000000 -0600
+++ CMakeLists.txt	2021-12-09 05:58:41.000000000 -0600
@@ -98,7 +98,7 @@
 
 list(APPEND GMIC_CXX_COMPILE_FLAGS -Dgmic_build -Dcimg_use_vt100 -Dgmic_is_parallel -Dcimg_use_abort)
 if(APPLE)
-    list(APPEND GMIC_CXX_COMPILE_FLAGS -mmacosx-version-min=10.8 -stdlib=libc++ -Wno-error=c++11-narrowing -Wc++11-extensions -fpermissive)
+    list(APPEND GMIC_CXX_COMPILE_FLAGS -Wno-error=c++11-narrowing -Wc++11-extensions -fpermissive)
 else()
     list(APPEND GMIC_CXX_COMPILE_FLAGS -Wno-error=narrowing -fno-ipa-sra -fpermissive)
 endif()
