--- gmic-qt/CMakeLists.txt.orig	2022-06-16 10:26:07.000000000 -0500
+++ gmic-qt/CMakeLists.txt	2022-06-17 05:45:25.000000000 -0500
@@ -545,16 +545,16 @@
     )
 endif(ENABLE_DYNAMIC_LINKING)
 
-message("Producing translation .qm files")
-execute_process(COMMAND make WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/translations OUTPUT_QUIET)
-message("Producing filter translation .qm files")
-execute_process(COMMAND make WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/translations/filters OUTPUT_QUIET)
-
-set(gmic_qt_QRC
-    gmic_qt.qrc
-    translations.qrc
-    wip_translations.qrc
-)
+# message("Producing translation .qm files")
+# execute_process(COMMAND make WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/translations OUTPUT_QUIET)
+# message("Producing filter translation .qm files")
+# execute_process(COMMAND make WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/translations/filters OUTPUT_QUIET)
+# 
+# set(gmic_qt_QRC
+#     gmic_qt.qrc
+#     translations.qrc
+#     wip_translations.qrc
+# )
 
 if (${GMIC_QT_HOST} STREQUAL "gimp" OR ${GMIC_QT_HOST} STREQUAL "gimp3")
     if (${GMIC_QT_HOST} STREQUAL "gimp3")
