--- Makefile	2016-03-25 05:58:15.000000000 -0400
+++ Makefile	2016-04-13 12:26:19.000000000 -0400
@@ -88,12 +88,7 @@
 endif
 
 ifeq ($(OS),Darwin)
-PLUGINDIR = ~/Library/Application\ Support/Gimp/plug-ins
-ifeq (,$(wildcard /opt/local))
-USR = /usr/local
-else
-USR = /opt/local
-endif
+USR = @PREFIX@
 WGET = curl -k -L --silent -o
 endif
 
@@ -176,8 +171,8 @@
 # (keep /usr/ dirname here since X11 is located in /usr/ on Mac too).
 # This requires the presence of the X11 include and library files.
 ifneq ($(OS),Darwin)
-X11_CFLAGS = -Dcimg_display=1 -Dcimg_appname=\\\"gmic\\\" `pkg-config --cflags x11 || echo -I/usr/X11R6/include` #-Dcimg_use_xrandr
-X11_LIBS = `pkg-config --libs x11 || echo -L/usr/X11R6/lib -lX11` -lpthread #`pkg-config --libs xrandr || echo -lXrandr`
+X11_CFLAGS = -Dcimg_display=1 -Dcimg_appname=\\\"gmic\\\" `pkg-config --cflags x11 || echo -I$(USR)/include` #-Dcimg_use_xrandr
+X11_LIBS = `pkg-config --libs x11 || echo -L$(USR)/lib -lX11` -lpthread #`pkg-config --libs xrandr || echo -lXrandr`
 else
 ifeq (,$(wildcard /usr/X11R6))
 X11_CFLAGS = -Dcimg_display=0 -Dcimg_appname=\\\"gmic\\\" #-Dcimg_use_xrandr
@@ -190,7 +185,7 @@
 # Flags to enable fast display, using XShm.
 # This requires the presence of the X11 extension include and library files.
 XSHM_CFLAGS = -Dcimg_use_xshm `pkg-config --cflags xcb-shm`
-XSHM_LIBS = `pkg-config --libs xcb-shm || echo -L$(USR)/X11R6/lib -lXext`
+XSHM_LIBS = `pkg-config --libs xcb-shm || echo -L$(USR)/lib -lXext`
 
 # Flags to enable image display, using GDI32.
 # This requires the presence of the GDI32 include and library files (Windows only).
@@ -386,7 +381,7 @@
 	@echo "**"
 	@echo "** Start building G'MIC with default MacOSX configuration."
 	@echo "**"
-	$(MAKE) cli gimp lib libc zart
+	$(MAKE) cli gimp lib libc
 else
 	@echo "**"
 	@echo "** Start building G'MIC with default Windows configuration."
@@ -605,8 +600,7 @@
 	mkdir -p $(DESTDIR)$(USR)/share/man/fr/man1/
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/man1/gmic.1.gz
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/fr/man1/gmic.1.gz
-	if [ -d /etc/bash_completion.d/ ]; then mkdir -p $(DESTDIR)/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/etc/bash_completion.d/gmic; fi
-	if [ -d /opt/local/etc/bash_completion.d/ ]; then mkdir -p $(DESTDIR)/opt/local/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/opt/local/etc/bash_completion.d/gmic; fi
+	mkdir -p $(DESTDIR)$(USR)/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)$(USR)/etc/bash_completion.d/gmic
 
 uninstall:
 	rm -f $(DESTDIR)$(USR)/$(BIN)/gmic
