--- SDL_ttf.c.orig	2007-04-08 14:37:18.000000000 -0700
+++ SDL_ttf.c	2007-04-08 14:40:43.000000000 -0700
@@ -48,11 +48,6 @@
 #include <freetype/ftoutln.h>
 #include <freetype/ttnameid.h>
 */
-#include <freetype/internal/ftobjs.h>
-
-#ifndef FT_OPEN_STREAM
-#define FT_OPEN_STREAM ft_open_stream
-#endif
 
 #include "SDL.h"
 #include "SDL_endian.h"
@@ -278,7 +273,7 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = library->memory;
+	stream->memory = NULL;  /* set by FT_Open_Face */
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
