--- CMakeLists.txt.orig	2022-03-31 19:39:11.000000000 +0700
+++ CMakeLists.txt	2022-09-09 00:55:21.000000000 +0700
@@ -306,11 +306,11 @@
                             PROPERTIES GENERATED 1)
 
 install(FILES ${PROJECT_BINARY_DIR}/doc/edge.8.gz
-        DESTINATION /usr/share/man/man8)
+        DESTINATION @PREFIX@/share/man/man8)
 install(FILES ${PROJECT_BINARY_DIR}/doc/supernode.1.gz
-        DESTINATION /usr/share/man/man1)
+        DESTINATION @PREFIX@/share/man/man1)
 install(FILES ${PROJECT_BINARY_DIR}/doc/n2n.7.gz
-        DESTINATION /usr/share/man/man7)
+        DESTINATION @PREFIX@/share/man/man7)
 endif(DEFINED UNIX)
 
 if (CMAKE_SYSTEM_NAME STREQUAL Linux)
