--- src/SConscript.client.orig	2014-08-14 14:16:38.000000000 -0500
+++ src/SConscript.client	2014-08-22 22:29:42.000000000 -0500
@@ -328,8 +328,8 @@
         if darwin:
             sharedLibEnv.AddPostAction(
                 mongoClientSharedLibPrefixInstall,
-                "install_name_tool -id %s %s" % (
-                    mongoClientSharedLibPrefixInstall[0],
+                "install_name_tool -id @PREFIX@/lib/%s %s" % (
+                    mongoClientSharedLibPrefixInstall[0].name,
                     mongoClientSharedLibPrefixInstall[0]
                 ))
         mongoClientPrefixInstalls.append(mongoClientSharedLibPrefixInstall)
