--- src/SConscript.client.orig	2015-01-28 15:59:55.000000000 -0600
+++ src/SConscript.client	2015-02-16 04:49:17.000000000 -0600
@@ -405,8 +405,8 @@
     if darwin:
         sharedLibEnv.AddPostAction(
             mongoClientSharedLibPrefixInstall,
-            "install_name_tool -id %s %s" % (
-                mongoClientSharedLibPrefixInstall[0],
+            "install_name_tool -id @PREFIX@/lib/%s %s" % (
+                mongoClientSharedLibPrefixInstall[0].name,
                 mongoClientSharedLibPrefixInstall[0]
             ))
     mongoClientPrefixInstalls.append(mongoClientSharedLibPrefixInstall)
