--- src/setenv.c.orig	2008-03-09 14:25:10.000000000 +0900
+++ src/setenv.c	2008-03-09 14:28:22.000000000 +0900
@@ -39,7 +39,11 @@
  */
 int
 setenv(name, value, rewrite)
+#ifdef __APPLE__
+	const char *name, *value;
+#else
 	register char *name, *value;
+#endif
 	int rewrite;
 {
 	extern char **environ;
@@ -97,7 +101,11 @@
  */
 void
 unsetenv(name)
+#ifdef __APPLE__
+	const char	*name;
+#else
 	char	*name;
+#endif
 {
 	extern char **environ;
 	register char **P;
