--- setup.py	2009-05-16 17:56:46.000000000 +0300
+++ setup.py	2009-05-16 18:01:54.000000000 +0300
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+#ident "@(#) $Id: setup.py,v 1.28 2006/06/07 17:21:28 jwa Exp $"
+# vi:set sw=4 ts=8 showmode ai:
+#-----------------------------------------------------------------------+
+# Name:		setup.py						|
+#									|
+# Synopsis:	python setup.py build	# Build the module.		|
+#		python setup.py install	# Install the module.		|
+#									|
+#		See http://www.python.org/sigs/distutils-sig/doc/ for	|
+#		more information on using distutils to install Python	|
+#		programs and modules.					|
+#									|
+# Description:	Setup script (using the distutils framework) for	|
+#		Midgard2.						|
+#=======================================================================|
+# Copyright 2009 by Jyrki Wahlstedt.					|
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#=======================================================================|
+# People who have worked on this code.					|
+#									|
+# Ini Name								|
+# --- ----------------------------------------------------------------- |
+# jwa Jyrki Wahlstedt <jwa@wahlstedt.fi>				|
+#=======================================================================|
+# Revision History:							|
+#									|
+# Date       Ini Description						|
+# ---------  --- ------------------------------------------------------ |
+# 2009-05-15 jwa Initial version created by Jyrki Wahlstedt.		|
+#-----------------------------------------------------------------------+
+import os, os.path, sys
+
+from distutils.core import setup
+from distutils.extension import Extension
+
+__version__ = "9.03.0"
+
+# Define the runtime library path for this module.  It starts out as None.
+
+def main():
+    # Default settings, may be overriden for specific platforms
+    pymidgard_rt_dirs = None
+    optional_libs = []
+    modname = "python-midgard2"
+
+    sources = ["midgardmodule.c",	"py_midgard_object_attachment.c",
+               "py_midgard_blob.c",	"py_midgard_object_class.c",
+               "py_midgard_collector.c",	"py_midgard_object_parameter.c",
+               "py_midgard_config.c",		"py_midgard_query_builder.c",
+               "py_midgard_connection.c",	"py_midgard_reflection_property.c",
+               "py_midgard_dbobject.c",		"py_midgard_replicator.c",
+               "py_midgard_dbus.c",		"py_midgard_user.c",
+               "py_midgard_object.c"]
+
+    define_macros = [('MGD_HAVE_DBUS', None)]
+    include_dirs = os.popen("pkg-config --cflags-only-I pygobject-2.0 midgard-2.0 | sed 's;-I;;g'").read().strip().split()
+    library_dirs = [os.popen("pkg-config --libs midgard-2.0").read().strip()]
+
+    # patch distutils if it can't cope with the "classifiers" keyword
+    if sys.version < '2.2.3':
+	from distutils.dist import DistributionMetadata
+	DistributionMetadata.classifiers = None
+	DistributionMetadata.download_url = None
+
+    classifiers = [
+	"Development Status :: 5 - Production/Stable",
+	"Environment :: Other Environment",
+	"Intended Audience :: Developers",
+	"License :: OSI Approved :: LGPL",
+	"Natural Language :: English",
+	"Operating System :: POSIX",
+	"Programming Language :: C",
+	"Programming Language :: Python",
+	"Topic :: Content :: Front-Ends"]
+
+    setup (
+	name = "python-midgard2",
+	version = __version__,
+	description = \
+	    "python-midgard2 - Python bindings to Midgard2.",
+	maintainer = "Midgard developers",
+	maintainer_email = "jwa@macports.org",
+	url = "http://www.midgard-project.org/",
+	license = "Python",
+#	packages = ["python-midgard2"],
+	ext_modules = [Extension(
+	    name = modname,
+	    sources = sources,
+	    define_macros = define_macros,
+	    include_dirs = include_dirs,
+	    library_dirs = library_dirs,
+	    runtime_library_dirs = pymidgard_rt_dirs,
+	    )],
+	classifiers = classifiers
+    )
+
+if __name__ == "__main__":
+    main()
