--- mozilla/security/nss/lib/ckfw/builtins/config.mk.orig	2008-01-05 21:38:36.000000000 +0100
+++ mozilla/security/nss/lib/ckfw/builtins/config.mk	2008-01-05 21:38:55.000000000 +0100
@@ -60,7 +60,7 @@
 # To create a loadable module on Darwin, we must use -bundle.
 #
 ifeq ($(OS_TARGET),Darwin)
-DSO_LDOPTS = -bundle
+DSO_LDOPTS = -arch i386 -arch ppc -bundle -L@@PREFIX@@/lib
 endif
 
 ifeq ($(OS_TARGET),SunOS)
