# See: https://bugzilla.mozilla.org/show_bug.cgi?id=1769063
--- nss/lib/freebl/mpi/mpcpucache.c.orig	2022-05-26 17:54:33.000000000 +0800
+++ nss/lib/freebl/mpi/mpcpucache.c	2022-06-04 07:52:38.000000000 +0800
@@ -727,10 +727,7 @@
 static inline void
 dcbzl(char *array)
 {
-    register char *a asm("r2") = array;
-    __asm__ __volatile__("dcbzl %0,0"
-                         : "=r"(a)
-                         : "0"(a));
+  __asm__ ("dcbzl %0, %1" : /*no result*/ : "b%" (array), "r" (0) : "memory");
 }
 
 #define PPC_DO_ALIGN(x, y) ((char *)((((long long)(x)) + ((y)-1)) & ~((y)-1)))
