--- mozilla/security/coreconf/UNIX.mk.orig	2007-11-14 19:41:08.000000000 +0100
+++ mozilla/security/coreconf/UNIX.mk	2007-11-14 19:41:24.000000000 +0100
@@ -46,7 +46,7 @@
 	DEFINES    += -UDEBUG -DNDEBUG
 else
 	OPTIMIZER  += -g
-	DEFINES    += -DDEBUG -UNDEBUG -DDEBUG_$(shell whoami)
+	DEFINES    += -DDEBUG -UNDEBUG -DDEBUG_$(shell whoami) -I@@PREFIX@@/include/nspr -L@@PREFIX@@/lib/nspr
 endif
 
 ifdef BUILD_TREE
