--- src/GLSDL.cpp	2006-11-08 14:13:50.000000000 +0100
+++ src/GLSDL.cpp	2008-02-05 00:10:22.000000000 +0100
@@ -1741,7 +1741,7 @@
 
 static int InitTexture(SDL_Surface *datasurf, glSDL_TexInfo *txi, int tex)
 {
-	glGenTextures(1, (unsigned int *)&txi->texture[tex]);
+	glGenTextures(1, (GLuint *)&txi->texture[tex]);
 	glBindTexture(GL_TEXTURE_2D, txi->texture[tex]);
 	glPixelStorei(GL_UNPACK_ROW_LENGTH, datasurf->pitch /
 			datasurf->format->BytesPerPixel);
@@ -1970,7 +1970,7 @@
 {
 	int i;
 	for(i = 0; i < txi->textures; ++i)
-		glDeleteTextures(1, (unsigned int *)&txi->texture[i]);
+		glDeleteTextures(1, (GLuint *)&txi->texture[i]);
 	memset(&txi->invalid_area, 0, sizeof(txi->invalid_area));
 }
 
