diff -ru Ice-3.4.2.orig/cpp/src/Ice/DynamicLibrary.cpp Ice-3.4.2/cpp/src/Ice/DynamicLibrary.cpp
--- Ice-3.4.2.orig/cpp/src/Ice/DynamicLibrary.cpp	2011-06-15 12:43:58.000000000 -0700
+++ Ice-3.4.2/cpp/src/Ice/DynamicLibrary.cpp	2013-08-30 22:10:32.000000000 -0700
@@ -112,12 +112,18 @@
 
     lib += ".dll";
 #elif defined(__APPLE__)
+    // Look for libZeroc... to handle the MacPorts
+    // s/libIce/libZerocIce/ rename.
+    string libZeroc = lib;
     lib = "lib" + libName;
+    libZeroc += "libZeroc" + libName;
     if(!version.empty()) 
     {
         lib += "." + version;
+        libZeroc += "." + version;
     }
     lib += ".dylib";
+    libZeroc += ".dylib";
 #elif defined(__hpux)
     lib = "lib" + libName;
     if(!version.empty())
@@ -145,7 +151,10 @@
 
     if(!load(lib))
     {
-        return 0;
+        if (!load(libZeroc))
+        {
+            return 0;
+        }
     }
 
     return getSymbol(funcName);
