--- DESCRIPTION.orig	2022-10-29 15:52:42.000000000 +0700
+++ DESCRIPTION	2023-03-25 05:12:57.000000000 +0700
@@ -15,7 +15,7 @@
 BugReports: https://github.com/stla/MeshesTools/issues
 Depends: R (>= 2.10)
 Imports: data.table, gmp, PolygonSoup, Rcpp (>= 1.0.9), rgl, Rvcg
-Suggests: randomcoloR, rmarchingcubes
+Suggests: rmarchingcubes
 LinkingTo: BH, Rcpp, RcppCGAL, RcppEigen
 Encoding: UTF-8
 LazyData: true
