--- configure.orig	2008-03-04 04:41:58.000000000 +1100
+++ configure	2012-08-15 09:46:18.000000000 +1000
@@ -20194,7 +20194,7 @@
                 PROFILE="-v -xpg -g -xO4 -fast -native -fsimple -xcg92 -Dsuncc" ;;
         *)
                 DEBUG="-g"
-                CFLAGS="-O"
+                CFLAGS+=" -O"
                 PROFILE="-g -p" ;;
         esac
 else
@@ -20202,16 +20202,16 @@
         case $host in
         *-*-linux*)
                 DEBUG="-g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
-                CFLAGS="-O20 -ffast-math -D_REENTRANT -fsigned-char"
-                PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char";;
+                CFLAGS+=" -O4 -ffast-math -D_REENTRANT -fsigned-char"
+                PROFILE="-pg -g -O4 -ffast-math -D_REENTRANT -fsigned-char";;
         sparc-sun-*)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8"
-                CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8"
-                PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
+                CFLAGS+=" -O4 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8"
+                PROFILE="-pg -g -O4 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
         *)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
-                CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char"
-                PROFILE="-O20 -g -pg -D__NO_MATH_INLINES -fsigned-char" ;;
+                CFLAGS+=" -O4 -D__NO_MATH_INLINES -fsigned-char"
+                PROFILE="-O4 -g -pg -D__NO_MATH_INLINES -fsigned-char" ;;
         esac
 fi
 
