--- Makefile.in.orig	2005-03-06 06:28:31
+++ Makefile.in	2023-02-08 23:48:28
@@ -79,7 +79,7 @@
 scanssh_SOURCES = scanssh.c atomicio.c exclude.c connecter.c xmalloc.c 	interface.c socks.c http.c telnet.c exclude.h interface.h 	scanssh.h socks.h xmalloc.h
 
 scanssh_LDADD = @LIBOBJS@ @PCAPLIB@ @EVENTLIB@ @DNETLIB@
-CFLAGS = -O2 -Wall -g
+CFLAGS = @CFLAGS@ -Wall
 
 INCLUDES = -I$(top_srcdir)/@DNETCOMPAT@ -I$(top_srcdir)/compat 	@EVENTINC@ @PCAPINC@ @DNETINC@
 
--- configure.orig	2005-03-06 06:21:31
+++ configure	2023-02-09 00:07:24
@@ -2874,7 +2874,7 @@
 fi
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 
-CFLAGS=""
+CFLAGS=$ac_save_CFLAGS
 LIBS=""
 
 ac_ext=c
