--- biabam	Fri Jun 11 06:49:05 2004
+++ biabam.new	Fri Jun 11 12:14:38 2004
@@ -27,6 +27,19 @@
 # no user serviceable parts below this point
 VERSION=0.9.6
 
+#missing seq() function
+seq()
+{
+	local I=$1;
+	while [ $2 != $I ]; do
+	{
+		echo -n "$I ";
+		I=$(( $I + 1 ))
+	};
+	done;
+	echo $2
+}
+
 if  [ $# -lt 2 ]; then
   echo "Usage:"
   echo "$0 filename1[${SP}filename2${SP}filenameN] [-s subject] recipient1[${SP}recipient2${SP}recipientN]"
