--- src/gen-posix-lock-obj.c.orig	2016-11-16 13:22:03.000000000 +0100
+++ src/gen-posix-lock-obj.c	2017-09-04 15:03:39.000000000 +0200
@@ -137,8 +137,10 @@ main (void)
   p = (unsigned char *)&mtx;
   for (i=0; i < sizeof mtx; i++)
     {
+#ifndef NO_LINE_CONT
       if (i && !(i % 8))
         printf (" \\\n%*s", 36, "");
+#endif
       printf ("%u", p[i]);
       if (i < sizeof mtx - 1)
         putchar (',');
