--- Configure.orig	2023-11-28 15:57:27
+++ Configure	2024-05-19 17:27:30
@@ -1511,7 +1511,7 @@
 usereentrant='undef'
 : List of libraries we want.
 : If anyone needs extra -lxxx, put those in a hint file.
-libswanted="cl pthread socket bind inet ndbm gdbm dbm db malloc dl ld"
+libswanted="pthread socket inet ndbm gdbm dbm db malloc dl ld"
 libswanted="$libswanted sun m crypt sec util c cposix posix ucb bsd BSD"
 : We probably want to search /usr/shlib before most other libraries.
 : This is only used by the lib/ExtUtils/MakeMaker.pm routine extliblist.
