--- core/vole_config.h.orig	2015-03-19 23:58:24.000000000 +0100
+++ core/vole_config.h	2015-03-19 23:59:03.000000000 +0100
@@ -12,7 +12,9 @@
 
 #ifdef WITH_BOOST_PROGRAM_OPTIONS
 #include <boost/version.hpp>
+#ifndef Q_MOC_RUN /* https://bugreports.qt.io/browse/QTBUG-22829 */
 #include <boost/program_options.hpp>
+#endif
 #endif // WITH_BOOST_PROGRAM_OPTIONS
 
 #include "hashes.h"
--- ./shell/command.h.orig	2015-03-19 23:59:33.000000000 +0100
+++ ./shell/command.h	2015-03-19 23:59:57.000000000 +0100
@@ -3,7 +3,9 @@
 
 #include <vector>
 #include <iostream>
+#ifndef Q_MOC_RUN /* https://bugreports.qt.io/browse/QTBUG-22829 */
 #include <boost/program_options.hpp>
+#endif
 #include "vole_config.h"
 #include "progress_observer.h"
 
--- gui/controller/controller.h.orig	2015-03-20 00:11:28.000000000 +0100
+++ gui/controller/controller.h	2015-03-20 00:12:07.000000000 +0100
@@ -6,7 +6,9 @@
 
 #include <QObject>
 #include <QMap>
+#ifndef Q_MOC_RUN /* https://bugreports.qt.io/browse/QTBUG-22829 */
 #include <boost/thread.hpp>
+#endif
 
 class DockController;
 class DistViewController;
