--- work/mathomatic-15.0.3/makefile	2010-01-08 16:17:44.000000000 +0100
+++ makefile	2010-01-12 13:57:45.000000000 +0100
@@ -2,7 +2,7 @@
 # Remove the -DUNIX define in CFLAGS when not using Mathomatic for desktop UNIX/Linux.
 # This makefile does not compile and install the Prime Number Tools in the "primes" directory.
 # This makefile does create and install all documentation.
-
+include flags
 VERSION		= `cat VERSION`
 
 CC		= gcc
@@ -10,7 +10,7 @@
 
 CFLAGS		+= -Wall -Wshadow -Wno-char-subscripts # gcc specific flags; can be removed
 #CFLAGS		+= -fomit-frame-pointer # Uncomment this line to optimize further.
-CFLAGS		+= -O -DUNIX -DVERSION=\"$(VERSION)\"
+CFLAGS		+= -DUNIX -DVERSION=\"$(VERSION)\"
 CFLAGS          += -D__C99FEATURES__ # for a successful compile under Solaris
 LDLIBS		+= -lm # libraries to link
 
