--- makefile	2009-12-15 13:41:22.000000000 +0100
+++ makefile	2009-12-30 12:34:16.000000000 +0100
@@ -4,11 +4,12 @@
 # Remove the -DUNIX define in CFLAGS when not using Mathomatic for desktop UNIX/Linux.
 # This makefile does not compile and install the Prime Number Tools in the "primes" directory.
 # This makefile does create and install all documentation.
+include flags
 
 VERSION		= `cat VERSION`
 CFLAGS		+= -Wall -Wshadow -Wno-char-subscripts # gcc specific flags
 #CFLAGS		+= -fomit-frame-pointer # Uncomment this line to optimize further.
-CFLAGS		+= -O -DUNIX -DVERSION=\"$(VERSION)\"
+CFLAGS		+= -DUNIX -DVERSION=\"$(VERSION)\"
 LDLIBS		+= -lm # libraries to link
 
 # "make READLINE=1" to include readline support:
