--- makefile	2009-05-31 20:01:26.000000000 +0200
+++ makefile	2009-06-09 09:22:05.000000000 +0200
@@ -3,10 +3,11 @@
 # Remove the -DUNIX define in CFLAGS when not using Mathomatic for desktop UNIX/Linux.
 # This makefile does not compile and install the Prime Number Tools in the "primes" directory.
 # This makefile does create and install all documentation.
+include flags
 
 VERSION		= `cat VERSION`
 CFLAGS		+= -Wall -Wshadow -Wno-char-subscripts -fomit-frame-pointer # gcc specific flags
-CFLAGS		+= -O -DUNIX -DVERSION=\"$(VERSION)\"
+CFLAGS		+= -DUNIX -DVERSION=\"$(VERSION)\"
 LDLIBS		+= -lm # libraries to link
 
 # "make READLINE=1" to include readline support:
