--- makefile	2009-04-17 12:25:14.000000000 +0200
+++ makefile	2009-04-18 11:20:32.000000000 +0200
@@ -1,10 +1,11 @@
 # Makefile for compiling, testing, and installing Mathomatic under any UNIX-like (POSIX) OS.
 # Currently uses gcc only options in CFLAGS, just remove them for other C compilers.
+include flags
 
 VERSION		= `cat VERSION`
 #CFLAGS		+= -Wall # Copy cproto.h to proto.h to compile with -Wall with no complaints.
 CFLAGS		+= -Wunused -Wshadow -Wformat -Wparentheses -Wno-char-subscripts # gcc specific flags
-CFLAGS		+= -O -DUNIX -DVERSION=\"$(VERSION)\"
+CFLAGS		+= -DUNIX -DVERSION=\"$(VERSION)\"
 LDLIBS		+= -lm # libraries to link
 
 # "make READLINE=1" to include readline support:
