http://hg.research.att.com/graphviz/rev/a7620a110c4e
--- cmd/gvedit/csettings.cpp.orig	Wed Aug 03 16:07:10 2011 -0400
+++ cmd/gvedit/csettings.cpp	Wed Aug 03 16:19:05 2011 -0400
@@ -373,11 +373,11 @@
     if (!graph)
 	return false;
     QString sfx = WIDGET(QComboBox, cbExtension)->currentText();
+    QString fileName(WIDGET(QLineEdit, leOutput)->text());
 
-    if (sfx == QString("NONE"))
+    if ((fileName == QString("")) || (sfx == QString("NONE")))
 	doPreview(QString());
     else {
-	QString fileName(WIDGET(QLineEdit, leOutput)->text());
 	fileName = stripFileExtension(fileName);
 	fileName = fileName + "." + sfx;
 	if (fileName != activeWindow->outputFile)
