Index: src/do/make_checkpoint_pareto.h
===================================================================
RCS file: /cvsroot/eodev/eo/src/do/make_checkpoint_pareto.h,v
retrieving revision 1.10
diff -u -r1.10 make_checkpoint_pareto.h
--- src/do/make_checkpoint_pareto.h	10 Oct 2005 13:34:21 -0000	1.10
+++ src/do/make_checkpoint_pareto.h	31 Mar 2008 13:22:07 -0000
@@ -27,7 +27,8 @@
 #ifndef _make_checkpoint_pareto_h
 #define _make_checkpoint_pareto_h

-#include <stdlib.h>
+#include <climits>
+#include <cstdlib>
 #include <sstream>

 #include "EO.h"
