--- src/utils/eoGnuplot1DSnapshot.h.orig	2016-01-26 12:13:11.000000000 +0100
+++ src/utils/eoGnuplot1DSnapshot.h	2016-01-26 12:13:53.000000000 +0100
@@ -56,7 +56,7 @@
                       std::string _filename = "gen", std::string _delim = " ",
                       unsigned _counter = 0, bool _rmFiles = true) :
       eoFileSnapshot(_dirname, _frequency, _filename, _delim, _counter, _rmFiles),
-      eoGnuplot(_filename,"set data style points"),
+      eoGnuplot(_filename,"set style data points"),
       pointSize(5)
   {}
 
@@ -65,7 +65,7 @@
 		       unsigned _frequency = 1, std::string _filename = "gen",
 		       std::string _delim = " ", unsigned _counter = 0, bool _rmFiles = true ) :
       eoFileSnapshot(_dirname, _frequency, _filename, _delim, _counter, _rmFiles),
-      eoGnuplot(_filename,"set data style points"),
+      eoGnuplot(_filename,"set style data points"),
       pointSize(5)
   {
     handleBounds(_bounds);
@@ -73,14 +73,14 @@
     // Ctor
   eoGnuplot1DSnapshot(eoFileSnapshot & _fSnapshot) :
       eoFileSnapshot(_fSnapshot),
-      eoGnuplot(_fSnapshot.baseFileName(),"set data style points"),
+      eoGnuplot(_fSnapshot.baseFileName(),"set style data points"),
       pointSize(5)
   {}
 
     // Ctor with range
   eoGnuplot1DSnapshot(eoFileSnapshot & _fSnapshot, eoRealVectorBounds & _bounds) :
       eoFileSnapshot(_fSnapshot),
-      eoGnuplot(_fSnapshot.baseFileName(),"set data style points"),
+      eoGnuplot(_fSnapshot.baseFileName(),"set style data points"),
       pointSize(5)
   {
     handleBounds(_bounds);
