--- src/utils/make_help.cpp	2008-03-31 21:26:37.000000000 +0200
+++ src/utils/make_help.cpp	2008-03-31 21:08:17.000000000 +0200
@@ -28,10 +28,12 @@
 #pragma warning(disable:4786)
 #endif 
 
+#include <cstdlib>
-#include <utils/eoParser.h>
 #include <fstream>
 #include <stdexcept>
 
+#include <utils/eoParser.h>
+
 using namespace std;
 
 /** Generation of the status file, and output of the help message if needed
