--- ./configure	2024-03-15 16:41:55
+++ ./configure	2024-03-15 16:42:09
@@ -21173,51 +21173,9 @@
 else
   HAVE_SGML_TRUE='#'
   HAVE_SGML_FALSE=
-fi
-
-
-save_LDFLAGS="$LDFLAGS"
-LDFLAGS="-Wl,--enable-new-dtags $LDFLAGS"
-
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking --enable-new-dtags option" >&5
-printf %s "checking --enable-new-dtags option... " >&6; }
-if test ${authlib_cv_enable_new_dtags+y}
-then :
-  printf %s "(cached) " >&6
-else $as_nop
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-int
-main (void)
-{
-return 0;
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"
-then :
-  authlib_cv_enable_new_dtags=yes
-else $as_nop
-  authlib_cv_enable_new_dtags=no
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.beam \
-    conftest$ac_exeext conftest.$ac_ext
-
-
 fi
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $authlib_cv_enable_new_dtags" >&5
-printf "%s\n" "$authlib_cv_enable_new_dtags" >&6; }
 
-if test "$authlib_cv_enable_new_dtags" = "yes"
-then
-   courierauth_ldflags="-Wl,--enable-new-dtags $courierauth_ldflags"
-fi
-
 
-
 LIBVERSION_CURRENT=0
 LIBVERSION_RELEASE=0
 LIBVERSION_AGE=0
