--- tools/pngio.c.orig	2011-09-28 03:15:53.000000000 +1000
+++ tools/pngio.c	2014-07-09 06:31:55.000000000 +1000
@@ -204,7 +204,7 @@ void SavePNG(Image *pi, char *name)
 
    /* supply image info to png library */
 
-   png_set_compression_level(pi->png_write, Z_BEST_COMPRESSION);
+   png_set_compression_level(pi->png_write, 9);
 
    png_set_IHDR(pi->png_write, pi->png_info,
 		pi->width, pi->height, 8,
