--- ./configure.in.orig	2008-10-26 23:46:06.000000000 +0100
+++ ./configure.in	2008-10-31 12:14:07.000000000 +0100
@@ -38,6 +38,10 @@ AM_CONDITIONAL([HAVE_OS_CYGWIN], [test x
 if test x"`uname | grep \"CYGWIN\"`" != x""; then
 	AC_DEFINE([HAVE_OS_CYGWIN], [1], [Compiling under cygwin])
 fi
+AM_CONDITIONAL([HAVE_OS_DARWIN], [test x"`uname`" = x"Darwin"])
+if test x"`uname`" = x"Darwin"; then
+	AC_DEFINE([HAVE_OS_DARWIN], [1], [Compiling under darwin])
+fi
 
 dnl check for basic programs
 AC_PROG_CC()
--- ./configure.orig	2008-10-26 23:46:17.000000000 +0100
+++ ./configure	2008-10-31 12:14:43.000000000 +0100
@@ -910,6 +910,8 @@ INTLTOOL_EXTRACT
 INTLTOOL_MERGE
 INTLTOOL_UPDATE
 USE_NLS
+HAVE_OS_DARWIN_FALSE
+HAVE_OS_DARWIN_TRUE
 HAVE_OS_CYGWIN_FALSE
 HAVE_OS_CYGWIN_TRUE
 EGREP
@@ -4996,6 +4998,21 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
 fi
+ if test x"`uname`" = x"Darwin"; then
+  HAVE_OS_DARWIN_TRUE=
+  HAVE_OS_DARWIN_FALSE='#'
+else
+  HAVE_OS_DARWIN_TRUE='#'
+  HAVE_OS_DARWIN_FALSE=
+fi
+
+if test x"`uname`" = x"Darwin"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_OS_DARWIN 1
+_ACEOF
+
+fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
@@ -7130,7 +7147,7 @@ ia64-*-hpux*)
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 7133 "configure"' > conftest.$ac_ext
+  echo '#line 7150 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -9728,11 +9745,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9731: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9748: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9735: \$? = $ac_status" >&5
+   echo "$as_me:9752: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -10018,11 +10035,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:10021: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:10038: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:10025: \$? = $ac_status" >&5
+   echo "$as_me:10042: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -10122,11 +10139,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:10125: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:10142: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:10129: \$? = $ac_status" >&5
+   echo "$as_me:10146: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -12524,7 +12541,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12527 "configure"
+#line 12544 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12624,7 +12641,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12627 "configure"
+#line 12644 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -15033,11 +15050,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:15036: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:15053: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:15040: \$? = $ac_status" >&5
+   echo "$as_me:15057: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -15137,11 +15154,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:15140: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:15157: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:15144: \$? = $ac_status" >&5
+   echo "$as_me:15161: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -16722,11 +16739,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16725: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16742: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16729: \$? = $ac_status" >&5
+   echo "$as_me:16746: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16826,11 +16843,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16829: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16846: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:16833: \$? = $ac_status" >&5
+   echo "$as_me:16850: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -19043,11 +19060,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:19046: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:19063: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:19050: \$? = $ac_status" >&5
+   echo "$as_me:19067: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -19333,11 +19350,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:19336: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:19353: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:19340: \$? = $ac_status" >&5
+   echo "$as_me:19357: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -19437,11 +19454,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:19440: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:19457: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:19444: \$? = $ac_status" >&5
+   echo "$as_me:19461: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -27325,6 +27342,13 @@ $as_echo "$as_me: error: conditional \"H
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${HAVE_OS_DARWIN_TRUE}" && test -z "${HAVE_OS_DARWIN_FALSE}"; then
+  { { $as_echo "$as_me:$LINENO: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+$as_echo "$as_me: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${am__fastdepCC_TRUE}" && test -z "${am__fastdepCC_FALSE}"; then
   { { $as_echo "$as_me:$LINENO: error: conditional \"am__fastdepCC\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
--- ./libxfsm/Makefile.am.orig	2008-10-26 23:08:04.000000000 +0100
+++ ./libxfsm/Makefile.am	2008-10-31 12:16:25.000000000 +0100
@@ -28,6 +28,10 @@ libxfsm_4_2_la_LIBADD =							\
 	@LIBXFCEGUI4_LIBS@
 endif
 
+if HAVE_OS_DARWIN
+libxfsm_4_2_la_LIBADD =							\
+	@LIBXFCEGUI4_LIBS@
+endif
 
 libxfsmincludedir =							\
 	$(includedir)/xfce4/xfce4-session-4.2/libxfsm
--- ./libxfsm/Makefile.in.orig	2008-10-26 23:46:15.000000000 +0100
+++ ./libxfsm/Makefile.in	2008-10-31 12:17:20.000000000 +0100
@@ -300,6 +300,9 @@ libxfsm_4_2_la_LDFLAGS = -export-dynamic
 @HAVE_OS_CYGWIN_TRUE@	@LIBX11_LIBS@							\
 @HAVE_OS_CYGWIN_TRUE@	@LIBXFCEGUI4_LIBS@
 
+@HAVE_OS_DARWIN_TRUE@libxfsm_4_2_la_LIBADD = \
+@HAVE_OS_DARWIN_TRUE@	@LIBXFCEGUI4_LIBS@
+
 libxfsmincludedir = \
 	$(includedir)/xfce4/xfce4-session-4.2/libxfsm
 
