--- xfce4-session/main.c.orig	2016-01-02 15:02:42.000000000 +0100
+++ xfce4-session/main.c	2016-01-02 15:04:23.000000000 +0100
@@ -243,8 +243,12 @@ main (int argc, char **argv)
   XfsmShutdown     *shutdown_helper;
   gboolean          succeed = TRUE;
 
+#if defined(__APPLE__) && defined(__MACH__)
+  /* dbus is being started by launchd */
+#else
   if (!xfsm_dbus_require_session (argc, argv))
     return EXIT_SUCCESS;
+#endif
 
   xfce_textdomain (GETTEXT_PACKAGE, PACKAGE_LOCALE_DIR, "UTF-8");
 
