From 2e2b1f4241a92d11e99b39b4c514173df3b2dcd4 Mon Sep 17 00:00:00 2001
From: Aaron Madlon-Kay <aaron@wovn.io>
Date: Fri, 18 Sep 2020 09:16:48 +0900
Subject: [PATCH] Fix shell invocations for MacPorts

diff --git a/libviews/callgraphview.cpp b/libviews/callgraphview.cpp
index 16996d9..180a76f 100644
--- a/libviews/callgraphview.cpp
+++ b/libviews/callgraphview.cpp
@@ -938,7 +938,7 @@ bool GraphExporter::savePrompt(QWidget *parent, TraceData *data,
         if (wrote && mime != "text/vnd.graphviz") {
             QProcess proc;
             proc.setStandardOutputFile(intendedName, QFile::Truncate);
-            proc.start("dot", { dotRenderType, dotName }, QProcess::ReadWrite);
+            proc.start("@@PREFIX@@/bin/dot", { dotRenderType, dotName }, QProcess::ReadWrite);
             proc.waitForFinished();
             wrote = proc.exitStatus() == QProcess::NormalExit;
 
@@ -2118,9 +2118,9 @@ void CallGraphView::refresh()
     QString renderProgram;
     QStringList renderArgs;
     if (_layout == GraphOptions::Circular)
-        renderProgram = QStringLiteral("twopi");
+        renderProgram = QStringLiteral("@@PREFIX@@/bin/twopi");
     else
-        renderProgram = QStringLiteral("dot");
+        renderProgram = QStringLiteral("@@PREFIX@@/bin/dot");
     renderArgs << QStringLiteral("-Tplain");
 
     _unparsedOutput = QString();
-- 
2.28.0

