--- ./Makefile.orig	2007-07-24 11:57:31.000000000 +0200
+++ ./Makefile	2007-08-28 15:15:43.000000000 +0200
@@ -12,10 +12,10 @@ clean:
 	rm -f *.tar.gz
 
 install:
-	mkdir -p $(DESTDIR)/usr/bin/
-	mkdir -p $(DESTDIR)/usr/share/man/man1
+	mkdir -p $(DESTDIR)@@PREFIX@@/bin/
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man1
 	for util in $(UTILS); do \
-		install -m 755 $$util.py $(DESTDIR)/usr/bin/$$util; \
+		install -m 755 $$util.py $(DESTDIR)@@PREFIX@@/bin/$$util; \
 	done
 
 	for d in $(SUBDIRS); do make DESTDIR=`cd $(DESTDIR); pwd` -C $$d install; [ $$? = 0 ] || exit 1; done
--- docs/Makefile.orig	2007-07-24 09:55:07.000000000 +0200
+++ docs/Makefile	2007-08-28 15:19:11.000000000 +0200
@@ -9,16 +9,16 @@
 	rm -f *.pyc *.pyo *~
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/man/man1
-	mkdir -p $(DESTDIR)/usr/share/man/man5
-	mkdir -p $(DESTDIR)/usr/share/man/man8
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man1
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man5
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man8
 	for doc in $(DOCS); do \
-		install -m 644 $$doc.1 $(DESTDIR)/usr/share/man/man1/; \
+		install -m 644 $$doc.1 $(DESTDIR)@@PREFIX@@/share/man/man1/; \
 	done
 	for doc in $(DOCS5); do \
-		install -m 644 $$doc.5 $(DESTDIR)/usr/share/man/man5/; \
+		install -m 644 $$doc.5 $(DESTDIR)@@PREFIX@@/share/man/man5/; \
 	done
 	for doc in $(DOCS8); do \
-		install -m 644 $$doc.8 $(DESTDIR)/usr/share/man/man8/; \
+		install -m 644 $$doc.8 $(DESTDIR)@@PREFIX@@/share/man/man8/; \
 	done
 	
\ No newline at end of file
--- ./debuginfo-install.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./debuginfo-install.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python 
+#!@@PYTHON@@ 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -15,7 +15,7 @@
 # Copyright 2007 Seth Vidal
 
 import sys
-sys.path.insert(0,'/usr/share/yum-cli/')
+sys.path.insert(0,'@@PREFIX@@/share/yum-cli/')
 
 import yum
 import yum.Errors
--- ./package-cleanup.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./package-cleanup.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PYTHON@@
 #
 # (C) 2005 Gijs Hollestelle, released under the GPL
 #
--- ./repo-graph.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repo-graph.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PYTHON@@
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repo-rss.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repo-rss.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PYTHON@@
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./repoclosure.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repoclosure.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PYTHON@@ -t
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./repomanage.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repomanage.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PYTHON@@
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./repoquery.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repoquery.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PYTHON@@ -tt
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -17,7 +17,7 @@
 
 
 import sys
-sys.path.insert(0, '/usr/share/yum-cli')
+sys.path.insert(0, '@@PREFIX@@/share/yum-cli')
 
 import signal
 import re
--- ./reposync.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./reposync.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PYTHON@@ -tt
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./repotrack.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./repotrack.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PYTHON@@ -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yum-builddep.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./yum-builddep.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PYTHON@@ -tt
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -15,7 +15,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
 import sys
-sys.path.insert(0,'/usr/share/yum-cli')
+sys.path.insert(0,'@@PREFIX@@/share/yum-cli')
 
 import yum
 from yum.misc import getCacheDir
--- ./yumdownloader.py.orig	2007-07-24 09:55:07.000000000 +0200
+++ ./yumdownloader.py	2007-08-28 15:15:13.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python 
+#!@@PYTHON@@ 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -14,7 +14,7 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
 import sys
-sys.path.insert(0,'/usr/share/yum-cli')
+sys.path.insert(0,'@@PREFIX@@/share/yum-cli')
 
 import yum
 from yum.misc import getCacheDir
