--- CMakeLists.txt.orig	2020-12-02 02:58:35.000000000 -0600
+++ CMakeLists.txt	2020-12-02 07:49:25.000000000 -0600
@@ -148,9 +148,9 @@
 if(BUILD_CLIENT)
     OPTION(GUI_TESTING "Build with gui introspection features of socket api" OFF)
 
-    if(APPLE)
-        find_package(Sparkle REQUIRED)
-    endif(APPLE)
+#    if(APPLE)
+#        find_package(Sparkle REQUIRED)
+#    endif(APPLE)
 
     if(UNIX AND NOT APPLE)
         find_package(Inotify REQUIRED)
