--- setup.py.orig	2019-03-01 11:39:55.000000000 +0100
+++ setup.py	2019-03-01 11:54:55.000000000 +0100
@@ -10,6 +10,7 @@
 
 iswindows = os.name == 'nt'
 WERROR = "/WX" if iswindows else "-Werror"
+WNOREGISTER = "" if iswindows else "-Wno-error=deprecated-register"
 WSTRICTPROTOTYPES = None if iswindows else "-Werror=strict-prototypes"
 WALL = "/Wall" if iswindows else "-Wall"
 STDC99 = "" if iswindows else "-std=c99"
@@ -86,6 +87,7 @@
     cflags.extend([NOOPTIMIZATION, PRODUCEDEBUGSYMBOLS])
 else:
     cflags.append(WERROR)
+    cflags.append(WNOREGISTER)
 
 def get_env_path_list(var_name, default=None):
     '''Get a path list from an environment variable.  The variable is parsed as
