--- src/lib/VSDMetaData.cpp.orig	2015-08-05 19:02:21.000000000 -0700
+++ src/lib/VSDMetaData.cpp	2015-08-05 19:02:49.000000000 -0700
@@ -312,7 +312,7 @@
   uint32_t firstDirSectorLocation = readU32(input);
 
   // Seek to the Root Directory Entry
-  size_t sectorSize = std::pow(2, sectorShift);
+  size_t sectorSize = std::pow(2.0, sectorShift);
   input->seek((firstDirSectorLocation + 1) * sectorSize, librevenge::RVNG_SEEK_SET);
   // DirectoryEntryName: 64 bytes
   // DirectoryEntryNameLength: 2 bytes
