--- src/lib/VSDMetaData.cpp.orig	2015-02-21 09:56:49.000000000 -0800
+++ src/lib/VSDMetaData.cpp	2015-02-21 09:57:33.000000000 -0800
@@ -226,7 +226,7 @@
   uint64_t modifiedTime = readU64(input);
 
   // modifiedTime is number of 100ns since Jan 1 1601
-  static const uint64_t epoch = 11644473600;
+  static const uint64_t epoch = 11644473600ULL;
   time_t sec = (modifiedTime / 10000000) - epoch;
   const struct tm *time = localtime(&sec);
   if (time)
