--- OSXvnc-server/rfb.h.orig	2009-09-10 10:13:09.000000000 -0500
+++ OSXvnc-server/rfb.h	2011-05-17 07:34:28.000000000 -0500
@@ -30,6 +30,8 @@
 #include "Xmd.h"
 #include "regionstr.h"
 
+#include <pthread.h>
+#include <machine/types.h>
 #include <rfbproto.h>
 #include <vncauth.h>
 #include <zlib.h>
@@ -41,7 +43,7 @@
 //#import <Carbon/Carbon.h>
 //#include <ApplicationServices/ApplicationServices.h>
 //#include "CoreGraphics/CGGeometry.h"
-#ifndef __COREGRAPHICS__
+#ifndef COREGRAPHICS_H_
 struct CGPoint {
     float x;
     float y;
