--- src/polkitbackend/polkitbackendinteractiveauthority.c.orig	2013-04-04 11:16:50.000000000 -0700
+++ src/polkitbackend/polkitbackendinteractiveauthority.c	2014-10-17 10:38:48.000000000 -0700
@@ -2113,11 +2113,7 @@
   ret = NULL;
   name = polkit_unix_netgroup_get_name (POLKIT_UNIX_NETGROUP (group));
 
-  if (setnetgrent (name) == 0)
-    {
-      g_warning ("Error looking up net group with name %s: %s", name, g_strerror (errno));
-      goto out;
-    }
+  setnetgrent (name);
 
   for (;;)
     {
