--- configure.ac.orig	2019-04-02 04:20:42.000000000 -0700
+++ configure.ac	2019-04-28 18:44:12.000000000 -0700
@@ -101,10 +101,6 @@
 
 AC_CHECK_FUNCS(clearenv fdatasync)
 
-if test "x$GCC" = "xyes"; then
-  LDFLAGS="-Wl,--as-needed $LDFLAGS"
-fi
-
 dnl ---------------------------------------------------------------------------
 dnl - Check whether setnetgrent has a return value
 dnl ---------------------------------------------------------------------------
@@ -433,7 +429,7 @@
 		;;
 	  *)
 		with_os_type=unknown
-		::
+		;;
 	esac
     fi
 fi
