--- makefile.orig	2008-02-23 15:40:42.000000000 +0100
+++ makefile	2008-04-25 22:26:21.000000000 +0200
@@ -12,7 +12,7 @@
 #
 #	Disregard compiler warnings on rdmolf.f
 #
-CC = cc
+CC = gcc-mp-4.3
 LIBS =  -lX11 -lm
 LIBSG = -L/usr/X11R6/lib -lGLU -lGL -lXmu -lX11 -lm
 
@@ -52,11 +52,12 @@
 # with GCC4.0 or higher replace the line with:
 # 'external gfortran_iargc'
 #
-CFLAGS = -c -I/usr/X11R6/include -DDOBACK -DHASTIMER -DCOLOR_OFFSET=0.0
-FC = g77
-FFLAGS =
-LIBS =  -L/usr/X11R6/lib -lX11 -lm
-LDR = g77   
+FC = gfortran-mp-4.3
+FFLAGS += -fomit-frame-pointer -ftree-vectorize -funroll-loops -O3
+CFLAGS += -fomit-frame-pointer -ftree-vectorize -funroll-loops -O3 -c -I/usr/X11R6/include -I/usr/include/malloc -DDOBACK -DHASTIMER -DCOLOR_OFFSET=0.0
+LIBS = -L/usr/X11R6/lib -lX11 -lm
+LDR = gfortran-mp-4.3
+
 # 64 bit machines:
 #LIBS =  -L/usr/X11R6/lib64 -lX11 -lm
 #
@@ -191,7 +192,7 @@
 # and the full opengl version of molden 'gmolden', (does not need glut)
 # (make gmolden)
 #
-LIBSOGL = -lglut -lGLU -lGL -lXmu -lX11 -lm
+LIBSOGL = -L/opt/local/lib -L/usr/X11R6/lib -lglut -lGLU -lGL -lXmu -lX11 -lm -Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
 #
 # on linux :
 #
