--- molden.f.orig	2007-06-26 22:10:20.000000000 +0200
+++ molden.f	2007-06-26 22:10:28.000000000 +0200
@@ -267,7 +267,7 @@
       common /vrmlhl/ spcdfil
       common /pmflvl/ ipmfm,ipmfh,pmflev(mxpmfl),levcol(mxpmfl)
       common /comsrf/ vo(3), rt(3),v1t(3),v2t(3),v3t(3),wo(3)
-      external iargc
+c      external iargc
 c GCC 4.0 or higher: 
 c      external gfortran_iargc
       dimension fcnt(mxcntr)
