Index: libtiff/configure.ac
===================================================================
--- libtiff/configure.ac	(revision 666)
+++ libtiff/configure.ac	(revision 667)
@@ -67,7 +67,7 @@
     TIFFHOST="$host"
 fi
 ${srcdir}/../compat/libtiff/configure --disable-zlib --disable-zstd --disable-lzma \
-	--disable-pixarlog --disable-jpeg --disable-jbig --disable-webp \
+	--disable-pixarlog --disable-jpeg --disable-jbig --disable-webp --disable-lerc \
 	"--with-CC=$TIFFCC" $TIFFHOST
 mv -f tifftcl_confdefs.h confdefs.h
 
Index: libtiff/configure
===================================================================
--- libtiff/configure	(revision 666)
+++ libtiff/configure	(revision 667)
@@ -5747,7 +5747,7 @@
     TIFFHOST="$host"
 fi
 ${srcdir}/../compat/libtiff/configure --disable-zlib --disable-zstd --disable-lzma \
-	--disable-pixarlog --disable-jpeg --disable-jbig --disable-webp \
+	--disable-pixarlog --disable-jpeg --disable-jbig --disable-webp --disable-lerc \
 	"--with-CC=$TIFFCC" $TIFFHOST
 mv -f tifftcl_confdefs.h confdefs.h
 
