--- pixmap/pixmapUnix.c.orig	2017-12-30 09:09:00.000000000 -0700
+++ pixmap/pixmapUnix.c	2018-11-11 21:31:49.000000000 -0700
@@ -13,11 +13,15 @@
 
 #include "pixmapInt.h"
 #include <stdlib.h>
-#include "X11/Xlib.h"
-#include "X11/Xutil.h"
+
 
 #ifdef MAC_OSX_TK
+#include <X11_tk/Xlib.h>
+#include <X11_tk/Xutil.h>
 #define XPutImage
+#else
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
 #endif
 
 typedef struct PixmapData {
--- window/window.c.orig	2017-12-30 09:09:03.000000000 -0700
+++ window/window.c	2018-11-11 21:40:25.000000000 -0700
@@ -13,13 +13,14 @@
 
 #include "init.c"
 
-#include "X11/Xutil.h"
 #ifndef __WIN32__
 #  ifndef MAC_OSX_TK
-#   include "X11/Xproto.h"
+#   include <X11/Xutil.h>
+#   include <X11/Xproto.h>
 #  else
-#   include "X11/Xlib.h"
-#   include "X11/Xfuncproto.h"
+#   include <X11_tk/Xutil.h>
+#   include <X11_tk/Xlib.h>
+#   include <X11_tk/Xfuncproto.h>
 #  endif
 #else
 #   include "X11/Xlib.h"
