Index: pixmap/pixmapUnix.c
==================================================================
--- pixmap/pixmapUnix.c (revision 716)
+++ pixmap/pixmapUnix.c (working copy)
@@ -13,8 +13,13 @@
 
 #include "pixmapInt.h"
 #include <stdlib.h>
-#include "X11/Xlib.h"
-#include "X11/Xutil.h"
+#ifdef MAC_OSX_TK
+#include <X11_tk/Xlib.h>
+#include <X11_tk/Xutil.h>
+#else
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+#endif
 
 typedef struct PixmapData {
     Pixmap mask;		/* Mask: only display pixmap pixels where
--- window/window.c.orig	2023-04-03 08:24:00.000000000 +0000
+++ window/window.c	2023-08-21 12:42:04.000000000 -0500
@@ -13,13 +13,14 @@
 
 #include "init.c"
 
-#include "X11/Xutil.h"
 #if !defined(_WIN32)
 #  if !defined(MAC_OSX_TK)
-#   include "X11/Xproto.h"
+#   include <X11/Xutil.h>
+#   include <X11/Xproto.h>
 #  else
-#   include "X11/Xlib.h"
-#   include "X11/Xfuncproto.h"
+#   include <X11_tk/Xutil.h>
+#   include <X11_tk/Xlib.h>
+#   include <X11_tk/Xfuncproto.h>
 #   undef X_GetImage
 #  endif
 #else
