From 83ad41e58895019d5a02dee080317eb3a6ed8e1c Mon Sep 17 00:00:00 2001
From: Eric S. Raymond <esr@thyrsus.com>
Date: Tue, 11 Apr 2017 00:58:56 -0400
Subject: [PATCH] Attempted fix for Mac OS X Sierra port bug.

---
 cvs.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cvs.h b/cvs.h
index 5722b0a..618ab2d 100644
--- cvs.h
+++ cvs.h
@@ -432,7 +432,8 @@ typedef struct _cvs_author {
 #define _alignof(T)  sizeof(long double)
 #endif
 
-#if defined(__APPLE__)
+/* For versions of Mac OS X before Sierra */
+#if defined(__APPLE__) && !defined(CLOCK_REALTIME)
 /* we mock this in utils.c; the CLOCK_REALTIME value is not used */
 typedef int clockid_t;
 int clock_gettime(clockid_t clock_id, struct timespec *tp);
--
libgit2 0.24.0

