--- Makefile.orig
+++ Makefile
@@ -4,9 +4,11 @@
 # For blob compression you will also need zlib.
 
 INSTALL = install
-prefix?=/usr/local
+prefix?=@@PREFIX@@
 target=$(DESTDIR)$(prefix)
-LEX=/usr/bin/flex
+LEX=@@FLEX@@
+CC=@@CC@@
+A2X=@@A2X@@
 
 VERSION=1.8
 
@@ -14,7 +16,7 @@ GCC_WARNINGS1=-Wall -Wpointer-arith -Wstrict-prototypes
 GCC_WARNINGS2=-Wmissing-prototypes -Wmissing-declarations
 GCC_WARNINGS3=-Wno-unused-function -Wno-unused-label -Wno-format-zero-length
 GCC_WARNINGS=$(GCC_WARNINGS1) $(GCC_WARNINGS2) $(GCC_WARNINGS3)
-CFLAGS=$(GCC_WARNINGS) -DVERSION=\"$(VERSION)\"
+CFLAGS+=$(GCC_WARNINGS) -DVERSION=\"$(VERSION)\"
 
 # To enable debugging of the Yacc grammar, uncomment the following line
 #CFLAGS += -DYYDEBUG=1
@@ -36,7 +38,7 @@ OBJS=gram.o lex.o rbtree.o main.o cvsutil.o revdir.o \
 	nodehash.o tags.o authormap.o graph.o utils.o
 
 cvs-fast-export: $(OBJS)
-	cc $(CFLAGS) -o $@ $(OBJS) $(LDFLAGS)
+	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LDFLAGS)
 
 $(OBJS): cvs.h
 
@@ -55,9 +57,9 @@ y.tab.h: gram.c
 
 # Requires asciidoc
 .asc.1:
-	a2x --doctype manpage --format manpage $*.asc
+	$(A2X) --doctype manpage --format manpage $*.asc
 .asc.html:
-	a2x --doctype manpage --format xhtml $*.asc
+	$(A2X) --doctype manpage --format xhtml $*.asc
 
 man: cvssync.1 cvs-fast-export.1
 
@@ -90,7 +92,7 @@ SUPPRESSIONS = -U__UNUSED__ -UYYPARSE_PARAM -UYYTYPE_INT16 -UYYTYPE_INT8 \
 	-UYYTYPE_UINT16 -UYYTYPE_UINT8 -UYY_USER_INIT \
 	-Ushort -Usize_t -Uyytext_ptr -Uyyoverflow
 cppcheck:
-	cppcheck -I. --template gcc --enable=all $(SUPPRESSIONS) --suppress=unusedStructMember --suppress=unusedFunction --suppress=unreadVariable --suppress=uselessAssignmentPtrArg --suppress=missingIncludeSystem *.[ch]
+	cppcheck -I. --template $(CC) --enable=all $(SUPPRESSIONS) --suppress=unusedStructMember --suppress=unusedFunction --suppress=unreadVariable --suppress=uselessAssignmentPtrArg --suppress=missingIncludeSystem *.[ch]
 
 SOURCES = Makefile *.[ch] *.[yl] cvssync
 DOCS = README COPYING NEWS AUTHORS TODO control cvs-fast-export.asc cvssync.asc
