--- spyderplugins/widgets/pylintgui.py.orig	2011-06-10 17:01:04.000000000 -0500
+++ spyderplugins/widgets/pylintgui.py	2011-08-10 11:27:23.000000000 -0500
@@ -42,7 +42,7 @@
 from spyderlib.utils.translations import get_translation
 _ = get_translation("p_pylint", dirname="spyderplugins")
 
-PYLINT_PATH = programs.get_nt_program_name('pylint')
+PYLINT_PATH = programs.get_nt_program_name('pylint-2.7')
 
 def is_pylint_installed():
     return programs.is_program_installed(PYLINT_PATH)
@@ -353,7 +353,10 @@
             i2 = line.find(':', i1+1)
             if i2 == -1:
                 continue
-            line_nb = line[i1+1:i2].strip()
+            i3 = line.find(',', i1+1)
+            if i3 == -1:
+                continue
+            line_nb = line[i1+1:i3].strip()
             if not line_nb:
                 continue
             line_nb = int(line_nb)
