--- meson.build.orig	2026-05-05 12:50:39
+++ meson.build	2026-05-05 12:52:22
@@ -343,24 +343,7 @@
 endif
 
 has_core_text = false
-if host_system == 'darwin' and subsystem == 'macos'
-  # for viewer-cocoa.m
-  add_languages('objc', native : false)
 
-  has_core_text = cc.links('''#include <CoreText/CoreText.h>
-                              int main (void) {
-                                CTGetCoreTextVersion ();
-                                return 0;
-                              }''',
-                           name: 'CoreText availability',
-                           dependencies: dependency('appleframeworks', modules: 'ApplicationServices'))
-  if has_core_text
-    pango_conf.set('HAVE_CORE_TEXT', 1)
-  endif
-
-  pango_deps += dependency('appleframeworks', modules: [ 'CoreFoundation', 'ApplicationServices' ])
-endif
-
 # Check for dwrite_3.h (from more recent Windows SDK or mingw-w64)
 if host_system == 'windows' and not cpp.has_header('dwrite_3.h')
   error('Windows backend enabled but dwrite_3.h not found.')
