--- bin/smokeping_cgi.orig	2021-08-11 10:07:32.000000000 -0500
+++ bin/smokeping_cgi	2021-08-12 13:20:57.000000000 -0500
@@ -6,7 +6,7 @@
 
 use FindBin;
 use lib (split /:/, q{}); # PERL5LIB
-use lib "$FindBin::RealBin/../lib";use lib "$FindBin::RealBin/../thirdparty/lib/perl5"; # LIBDIR
+use lib "__PREFIX__/lib/smokeping/lib";use lib "__PERL5__ARCHLIB__/auto/RRDs"; # LIBDIR
 
 # don't bother with zombies
 $SIG{CHLD} = 'IGNORE';
@@ -17,7 +17,7 @@
 
 use CGI::Fast;
 
-my $cfg = (shift @ARGV) || "$FindBin::Bin/../etc/config";
+my $cfgfile = "__PREFIX__/etc/smokeping/config";
 
 
 while (my $q = new CGI::Fast) {
