--- bin/smokeping_cgi.orig	2018-04-13 07:13:05.000000000 -0500
+++ bin/smokeping_cgi	2019-05-04 09:11:25.000000000 -0500
@@ -6,7 +6,7 @@
 
 use lib (split /:/, q{}); # PERL5LIB
 use FindBin;
-use lib "$FindBin::RealBin/../lib";use lib "$FindBin::RealBin/../thirdparty/lib/perl5"; # LIBDIR
+use lib "__PREFIX__/lib/smokeping/lib";use lib "__PERL5__ARCHLIB__/auto/RRDs"; # LIBDIR
 
 # don't bother with zombies
 $SIG{CHLD} = 'IGNORE';
