--- bin/smokeping.orig	2021-08-11 10:07:59.000000000 -0500
+++ bin/smokeping	2021-08-12 13:14:27.000000000 -0500
@@ -6,11 +6,11 @@
 
 use FindBin;
 use lib (split /:/, q{}); # PERL5LIB
-use lib "$FindBin::RealBin/../lib";use lib "$FindBin::RealBin/../thirdparty/lib/perl5"; # LIBDIR
+use lib "__PREFIX__/lib/smokeping";use lib "__PERL5__ARCHLIB__/auto/RRDs"; # LIBDIR
 
 use Smokeping;
  
-Smokeping::main("$FindBin::RealBin/../etc/config");
+Smokeping::main("__PREFIX__/etc/smokeping/config");
 
 =head1 NAME
 
@@ -130,22 +130,7 @@
 
 =head1 SETUP
 
-When installing SmokePing, this file has to be adjusted to fit your
-local system. Three paths have to be entered.
 
-One pointing to your B<rrdtool> installation
-
- use lib qw(/usr/pack/rrdtool-1.0.33-to/lib/perl);
-
-One pointing to the place where you have installed the SmokePing libraries
-
- use lib qw(/home/oetiker/public_html/smokeping/lib);
-
-The third path is the argument to the Smokeping::main command. It points to
-the SmokePing configuration file.
-
- use Smokeping;
- Smokeping::main("/home/oetiker/.smokeping/config");
 
 =head1 SEE ALSO
 
