--- Makefile.in.orig	2018-12-20 04:22:43.000000000 -0600
+++ Makefile.in	2019-05-04 13:34:54.000000000 -0500
@@ -313,7 +313,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = thirdparty doc etc htdocs lib bin
+SUBDIRS = doc etc htdocs lib bin
 EXTRA_DIST = COPYRIGHT CHANGES CONTRIBUTORS LICENSE PERL_MODULES
 THIRDPARTY_DIR := $(shell pwd)/thirdparty
 all: all-recursive
