--- playxm/xmload.c	2024-04-05 03:54:03.000000000 +0800
+++ playxm/xmload.c	2024-12-03 10:27:37.000000000 +0800
@@ -453,8 +453,8 @@
 			cpifaceSession->cpiDebug (cpifaceSession, "[XM/XM] warning, Panning loop end point (%d) >= Number of panning points (%d), truncating\n", ins2.ploope, ins2.pnum);
 			ins2.ploope=ins2.pnum-1;
 		}
-		for (k=0;k<12;k++)
-			for (j=0;j<2;j++)
+		for (k=0;k<2;k++)
+			for (j=0;j<12;j++)
 			{
 				ins2.venv[j][k] = uint16_little (ins2.venv[j][k]);
 				ins2.penv[j][k] = uint16_little (ins2.penv[j][k]);
