--- imageindex	2007-04-04 15:55:51.000000000 -0400
+++ imageindex	2016-04-01 22:10:01.000000000 -0400
@@ -325,7 +325,7 @@
 use File::Copy;
 use English;
 use Carp;
-require 'flush.pl';
+require IO::Handle;
 
 # to shut up -w
 use vars qw($opt_recurse);
@@ -523,7 +523,7 @@
 
 # Extract info
 print "Extracting image info";
-flush (STDOUT);
+STDOUT->flush();
 
 foreach my $file (@files) {
 
@@ -1708,7 +1708,7 @@
 	# at the user so that the video regexp might be adjusted
 	if ($retval == -1) {
 	    print "\nwarning: $pathname identified by extension as video file but mplayer doesn't recognize it\n";
-	    flush (STDOUT);
+	    STDOUT->flush();
 	}
     } else  {
 	&extract_image_info ($filename);
@@ -1731,15 +1731,15 @@
     my ($arate, $anch, $length, $is_video);
 
     print ".";
-    flush (STDOUT);
+    STDOUT->flush();
 
     if ($mplayer_prog eq '' || ($do_video_files == 0)) {
 	if (($do_video_files != 0) && $mplayer_prog eq '') {
 	    print "\nwarning: Trying to process video files but cannot find mplayer in \$path!\n";
-	    flush (STDOUT);
+	    STDOUT->flush();
 	}
 	print "\nSkipping $pathname";
-	flush (STDOUT);
+	STDOUT->flush();
 	return 0;
     } else {
 	$object_counter++;
@@ -1893,14 +1893,14 @@
     my $i;
 
     print ".";
-    flush (STDOUT);
+    STDOUT->flush();
 
     $retval = $image->Read($pathname);
 
 
     if ($retval ne "") {
 	print "\nSkipping $pathname";
-	flush (STDOUT);
+	STDOUT->flush();
 	return;
     } else {
 	$object_counter++;
@@ -2298,7 +2298,7 @@
     if (defined($lastdate)) {
 	print "<META NAME=\"$enddatemetatag\" CONTENT=\"$lastdate\">\n";
     }
-    if (!defined ($opt_includeall) && defined (@opt_exclude) && scalar (@opt_exclude)) {
+    if (!defined ($opt_includeall) && (@opt_exclude) && scalar (@opt_exclude)) {
 	my $tmp = join (',', @opt_exclude);
 	my $etmp;
 
@@ -2311,7 +2311,7 @@
     }
     printf ("<META NAME=\"$numimagesmetatag\" CONTENT=\"%d\">\n", $image_counter);
 
-    if (defined (@opt_skipmont) && scalar (@opt_skipmont)) {
+    if ((@opt_skipmont) && scalar (@opt_skipmont)) {
 	my $tmp = join (',', @opt_skipmont);
 	printf ("<META NAME=\"$skipmetatag\" CONTENT=\"%s\">\n", $tmp);
     }
@@ -2475,7 +2475,7 @@
     # then process. Check to see if any of the -skipmont options were given as
     # strings of filenames concatenated with ',' characters. If so, support it.
     #
-    if (defined (@opt_skipmont)) {
+    if ((@opt_skipmont)) {
 	foreach (@opt_skipmont) {
 	    (@tokens) = split (/,/, $_);
 	    foreach $token (@tokens) {
@@ -2499,7 +2499,7 @@
     # then process. Check to see if any of the -exclude options were given as
     # strings of filenames concatenated with ',' characters. If so, support it.
     #
-    if (defined (@opt_exclude)) {
+    if ((@opt_exclude)) {
 	# -includeall takes priority over -exclude on the commandline if they are
 	# used together (wierd, but ...)
 	#
