https://github.com/NGSolve/netgen/pull/172

--- ng/encoding.hpp.orig	2026-04-24 17:35:17
+++ ng/encoding.hpp	2026-05-10 13:47:12
@@ -3,6 +3,11 @@
 
 #ifdef FFMPEG
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
+
 extern "C" {
 #include <libavutil/avassert.h>
 #include <libavcodec/avcodec.h>
@@ -16,6 +21,10 @@
 #include <libswresample/swresample.h>
 }
 
+using std::cerr;
+using std::endl;
+using std::string;
+
 constexpr int BITRATE = 50000000;
 
 class Mpeg {
