diff --git CMakeLists.txt CMakeLists.txt
index 0a05ebb1..6726480c 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -280,7 +280,7 @@ if (USE_GUI)
        target_include_directories(netgen_gui INTERFACE ${TK_INCLUDE_PATH}/../xlib)
     endif()
 
-    target_link_libraries(nggui PUBLIC nglib togl PRIVATE "$<BUILD_INTERFACE:netgen_python>" )
+    target_link_libraries(nggui PUBLIC nglib Togl PRIVATE "$<BUILD_INTERFACE:netgen_python>" )
 
     if(WIN32)
       target_compile_definitions(netgen_gui INTERFACE -DTOGL_WGL)
diff --git ng/CMakeLists.txt ng/CMakeLists.txt
index 3ad3d8be..b14a3d1d 100644
--- ng/CMakeLists.txt
+++ ng/CMakeLists.txt
@@ -70,6 +70,6 @@ if(USE_GUI)
         DESTINATION ${NG_INSTALL_DIR_BIN} COMPONENT netgen)
   endif()
 
-  add_subdirectory(Togl2.1)
+  # add_subdirectory(Togl2.1)
 endif(USE_GUI)
 
