do not install another version of pybind11

diff --git cmake/SuperBuild.cmake cmake/SuperBuild.cmake
index 13d6915a..5abdd844 100644
--- cmake/SuperBuild.cmake
+++ cmake/SuperBuild.cmake
@@ -154,7 +154,7 @@ endif()
 #######################################################################
 if (USE_PYTHON)
   find_path(PYBIND_INCLUDE_DIR pybind11/pybind11.h PATHS ${CMAKE_CURRENT_SOURCE_DIR}/external_dependencies/pybind11/include NO_DEFAULT_PATH)
-    set(NG_INSTALL_PYBIND ON)
+    set(NG_INSTALL_PYBIND OFF)
     if( NOT PYBIND_INCLUDE_DIR )
       # if the pybind submodule is missing, try to initialize and update all submodules
       execute_process(COMMAND git submodule update --init --recursive WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
