--- ng/CMakeLists.txt.orig	2018-11-23 02:52:27.000000000 -0700
+++ ng/CMakeLists.txt	2018-12-10 05:04:29.000000000 -0700
@@ -26,7 +26,7 @@
 
     target_link_libraries( gui PUBLIC nglib )
     target_link_libraries( gui PRIVATE ${LIBTOGL} ${ZLIB_LIBRARIES} ${JPEG_LIBRARIES} ${FFMPEG_LIBRARIES} ${X11_Xmu_LIB} ${X11_X11_LIB} ${OCC_LIBRARIES} )
-    if(NOT APPLE)
+    if(TRUE OR NOT APPLE)
       target_link_libraries( gui PRIVATE ${TCL_LIBRARY} ${TK_LIBRARY})
     endif(NOT APPLE)
 
