See https://github.com/NGSolve/netgen/pull/143

diff --git CMakeLists.txt CMakeLists.txt
index 0a05ebb1..6726480c 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -280,7 +280,7 @@ if (USE_GUI)
        target_include_directories(netgen_gui INTERFACE ${TK_INCLUDE_PATH}/../xlib)
     endif()
 
-    target_link_libraries(nggui PUBLIC nglib Togl PRIVATE "$<BUILD_INTERFACE:netgen_python>" )
+    target_link_libraries(nggui PUBLIC nglib Togl PRIVATE "$<BUILD_INTERFACE:netgen_python>" "$<BUILD_INTERFACE:netgen_gui>" )
 
     if(WIN32)
       target_compile_definitions(netgen_gui INTERFACE -DTOGL_WGL)
