diff --git libsrc/core/CMakeLists.txt libsrc/core/CMakeLists.txt
index ee89b4f6..37a1535c 100644
--- libsrc/core/CMakeLists.txt
+++ libsrc/core/CMakeLists.txt
@@ -18,7 +18,7 @@ target_compile_options(ngcore PUBLIC "${NG_COMPILE_FLAGS}")
 
 # Pybind11 2.3 Issue https://github.com/pybind/pybind11/issues/1604
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-  target_compile_options(ngcore PUBLIC -fsized-deallocation -faligned-allocation)
+  target_compile_options(ngcore PUBLIC -faligned-allocation)
 endif()
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
     target_link_libraries(ngcore PUBLIC stdc++fs)
