--- setup.py.orig	2019-12-18 11:48:21.000000000 -0500
+++ setup.py	2020-02-04 22:29:28.000000000 -0500
@@ -33,10 +33,6 @@
 
 lib_dirs = []
 inc_dirs = ["src"]
-# some OS, such as FreeBSD, uses /usr/local folders
-if path.exists("/usr/local"):
-    lib_dirs.append("/usr/local/lib")
-    inc_dirs.append("/usr/local/include")
 source_files = [
     "src/ck_attribute_smart.cpp",
     "src/pkcs11lib.cpp",
