--- meson.build.orig	2016-12-21 00:51:59.000000000 -0800
+++ meson.build	2017-03-16 15:42:05.000000000 -0700
@@ -64,7 +64,7 @@
 
 httpseverywhere_test = executable('httpseverywhere_test', httpseverywhere_test_source,
                                   dependencies: [glib, gobject, gio, gee, libxml],
-                                  include_directories: include_directories('@0@/httpseverywhere-@1@'.format(meson.current_build_dir(), api)),
+                                  include_directories: include_directories('build/httpseverywhere-'+api),
                                   link_with: httpseverywhere_lib)
 
 g_ir_compiler = find_program('g-ir-compiler')
