--- Makefile.in.org	2009-11-09 07:42:09.000000000 -0600
+++ Makefile.in	2010-01-26 11:34:20.000000000 -0600
@@ -3,7 +3,8 @@
 CFLAGS += -I. -fvisibility=hidden -DIGUANAIR_EXPORTS
 LINKARGS = -Wl,-soname=$(LIBNAME),--version-script=$(srcdir)/ig.ver
 ifeq ($(shell uname -s),Darwin)
-  LINKARGS = -Wl,-dylib_current_version,1.0.0,-dylib_compatibility_version,1.0.0
+  LINKARGS = -Wl,-dylib_current_version,1.0.0,-dylib_compatibility_version,1.0.0 \
+		-install_name $(prefix)/$(LIBDIR)/$(LIBNAME)
 endif
 
 .PHONY: all drivers
@@ -19,7 +20,7 @@
 igdaemon: daemon.o client-interface.o device-interface.o driver.o \
           list.o protocol-versions.o pipes.o dataPackets.o compat.o \
           support.c $(OSSUPPORT)
-	$(CC) -Wl,-export-dynamic $(CPPFLAGS) $(CFLAGS) -DSUPPORT_EXPORTS \
+	$(CC)  $(CPPFLAGS) $(CFLAGS) -DSUPPORT_EXPORTS \
               $(LDFLAGS) $^ -lpopt -lusb -lpthread -ldl $(LIBNAME) -o $@
 
 igclient: client.o list.o support.o compat.o
