--- drivers/Makefile.in.org	2009-11-04 20:53:12.000000000 -0600
+++ drivers/Makefile.in	2010-02-15 10:13:32.000000000 -0600
@@ -6,11 +6,10 @@
 all: $(DRIVERS)
 
 libusb.so: libusb.c $(LISTS) Makefile
-#	$(CC) $(CPPFLAGS) $(CFLAGS) $(SHAREARG) $(LDFLAGS) -lusb-1.0 $< $(LISTS) -o $@
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(SHAREARG) -lusb-1.0 $< $(LISTS) -o $@
+	$(CC) -undefined dynamic_lookup $(CPPFLAGS) $(CFLAGS) $(SHAREARG) $(LDFLAGS) -lusb-1.0 $< $(LISTS) -o $@
 
 libusbpre1.so: libusbpre1.c $(LISTS) Makefile
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(SHAREARG) -lusb $< $(LISTS) -o $@
+	$(CC) -undefined dynamic_lookup $(CPPFLAGS) $(CFLAGS) $(SHAREARG) $(LDFLAGS) -lusb $< $(LISTS) -o $@
 
 .PHONY: clean
 clean:
