--- src/Makefile.am.orig	2015-12-16 16:00:15.000000000 -0800
+++ src/Makefile.am	2015-12-16 16:02:57.000000000 -0800
@@ -62,9 +62,6 @@
     gcal-utils.c                                          \
     gcal-utils.h
 
-gnome_calendar_LDFLAGS =                                  \
-  -Wl,--export-dynamic
-
 gnome_calendar_LDADD =                                    \
   -lm                                                     \
   $(CALENDAR_LIBS)
