--- common/sysutils.c.old	2017-08-28 12:22:54.000000000 +0200
+++ common/sysutils.c	2017-11-05 11:45:20.000000000 +0100
@@ -36,6 +36,11 @@
 # undef USE_NPTH
 #endif
 
+/* Needed for the int-returning version of unsetenv(). */
+#if defined(__APPLE__) && ((!defined(_POSIX_C_SOURCE)) || (_POSIX_C_SOURCE < 200112L))
+#define _POSIX_C_SOURCE 200112L
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
