--- agent/gpg-agent.c.old	2016-03-31 12:46:07.000000000 +0200
+++ agent/gpg-agent.c	2016-06-02 07:26:26.000000000 +0200
@@ -2505,6 +2505,7 @@
         }
 
       infostr = xstrdup (infostr);
+#ifndef __APPLE__
       if ( !(p = strchr (infostr, PATHSEP_C)) || p == infostr)
         {
           xfree (infostr);
@@ -2530,6 +2531,11 @@
             return 0; /* Okay, its running on the standard socket. */
           return -1;
         }
+#else
+      // Currently unused by assuan.
+      pid = (pid_t) (-1);
+      prot = 1;
+#endif
     }
   else /* MODE != 0 */
     {
--- common/asshelp.c.old	2016-03-24 17:32:51.000000000 +0100
+++ common/asshelp.c	2016-06-02 07:26:26.000000000 +0200
@@ -398,6 +398,7 @@
       int pid;
 
       infostr = xstrdup (infostr);
+#ifndef __APPLE__
       if ( !(p = strchr (infostr, PATHSEP_C)) || p == infostr)
         {
           log_error (_("malformed GPG_AGENT_INFO environment variable\n"));
@@ -418,6 +419,11 @@
           force_pipe_server = 1;
           goto restart;
         }
+#else
+      // Currently unused by assuan.
+      pid = (pid_t) (-1);
+      prot = 1;
+#endif
 
       err = assuan_socket_connect (ctx, infostr, pid, 0);
       xfree (infostr);
--- common/simple-pwquery.c.old	2016-03-24 17:32:51.000000000 +0100
+++ common/simple-pwquery.c	2016-06-02 07:26:26.000000000 +0200
@@ -345,19 +345,27 @@
   strcpy (p, infostr);
   infostr = p;
 
+#ifndef __APPLE__
   if ( !(p = strchr ( infostr, PATHSEP_C)) || p == infostr
        || (p-infostr)+1 >= sizeof client_addr.sun_path ) 
+#else
+  if (((p-infostr) + 1) >= sizeof (client_addr.sun_path))
+#endif
     {
 #ifdef SPWQ_USE_LOGGING
       log_error ( _("malformed GPG_AGENT_INFO environment variable\n"));
 #endif
       return SPWQ_NO_AGENT;
     }
+#ifndef __APPLE__
   *p++ = 0;
 
   while (*p && *p != PATHSEP_C)
     p++;
   prot = *p? atoi (p+1) : 0;
+#else
+  prot = 1;
+#endif
   if ( prot != 1)
     {
 #ifdef SPWQ_USE_LOGGING
--- tools/gpg-connect-agent.c.old	2016-03-24 17:32:51.000000000 +0100
+++ tools/gpg-connect-agent.c	2016-06-02 07:26:26.000000000 +0200
@@ -2163,6 +2163,7 @@
       int pid;
 
       infostr = xstrdup (infostr);
+#ifndef __APPLE__
       if ( !(p = strchr (infostr, PATHSEP_C)) || p == infostr)
         {
           log_error (_("malformed GPG_AGENT_INFO environment variable\n"));
@@ -2174,6 +2175,11 @@
       while (*p && *p != PATHSEP_C)
         p++;
       prot = *p? atoi (p+1) : 0;
+#else
+      // Currently unused by assuan.
+      pid = (pid_t) (-1);
+      prot = 1;
+#endif
       if (prot != 1)
         {
           log_error (_("gpg-agent protocol version %d is not supported\n"),
--- tools/gpgconf.c.old	2016-03-24 17:32:51.000000000 +0100
+++ tools/gpgconf.c	2016-06-02 07:26:26.000000000 +0200
@@ -317,6 +317,7 @@
             char *tmp;
 
             infostr = xstrdup (infostr);
+#ifndef __APPLE__
             tmp = strchr (infostr, PATHSEP_C);
             if (!tmp || tmp == infostr)
               {
@@ -325,6 +326,7 @@
               }
             else
               *tmp = 0;
+#endif
           }
         fprintf (outfp, "agent-socket:%s\n",
                  infostr? gc_percent_escape (infostr) : "");
