--- configure.ac.orig	2015-06-09 14:59:40.000000000 -0700
+++ configure.ac	2015-07-01 12:08:41.000000000 -0700
@@ -1126,7 +1126,7 @@
 GTK_CHECK_BACKEND([x11], [gtk_required_version],
                   [have_gtk_x11=yes], [have_gtk_x11=no])
 
-have_libxmu="no (building for `pkg-config --variable=targets gtk-3.0`)"
+have_libxmu="no (building for `pkg-config --variable=targets gtk+-3.0`)"
 
 if test "x$have_gtk_x11" = "xyes"; then
 
