--- app/widgets/gimpcriticaldialog.c.orig	2018-07-25 08:01:38.000000000 -0700
+++ app/widgets/gimpcriticaldialog.c	2018-07-25 14:39:33.000000000 -0700
@@ -276,26 +276,25 @@
 
   return TRUE;
 
-#elif defined(PLATFORM_OSX)
-
-  NSURL    *ns_url;
-  gboolean  retval;
-
-  @autoreleasepool
-    {
-      ns_url = [NSURL URLWithString: [NSString stringWithUTF8String: url]];
-      retval = [[NSWorkspace sharedWorkspace] openURL: ns_url];
-    }
-
-  return retval;
-
 #else
+  gboolean rstatus;
+  gchar *command_line;
 
-  return gtk_show_uri_on_window (window,
+  rstatus = gtk_show_uri_on_window (window,
                                  url,
                                  GDK_CURRENT_TIME,
                                  error);
 
+  if (!rstatus) {
+     command_line = g_strdup_printf ("open %s", url);
+
+     rstatus = g_spawn_command_line_async (command_line,
+                       error);
+
+     g_free (command_line);
+  }
+  
+  return rstatus;
 #endif
 }
 
