--- libgimpwidgets/gimppickbutton-quartz.c.orig	2018-10-25 12:24:03.000000000 -0700
+++ libgimpwidgets/gimppickbutton-quartz.c	2018-10-25 12:58:21.000000000 -0700
@@ -53,6 +53,10 @@
 {
   GimpPickButton *button;
   NSMutableArray *windows;
+#ifndef __LP64__
+  BOOL firstBecameKey;
+  NSCursor *cursor;
+#endif
 }
 
 @property (nonatomic, assign) BOOL firstBecameKey;
@@ -67,6 +71,9 @@
 {
   GimpPickButton *button;
   GimpPickWindowController *controller;
+#ifndef __LP64__
+  NSTrackingArea *area;
+#endif
 }
 
 @property (readonly,assign) NSTrackingArea *area;
@@ -191,7 +198,7 @@
   rect = [self.window convertRectToScreen:rect];
   rect.origin.y = [[[NSScreen screens] objectAtIndex:0] frame].size.height - rect.origin.y;
 
-  root_image_ref = CGWindowListCreateImage (rect,
+  root_image_ref = CGWindowListCreateImage (NSRectToCGRect(rect),
                                             kCGWindowListOptionOnScreenOnly,
                                             kCGNullWindowID,
                                             kCGWindowImageDefault);
