--- setup.cfg.orig	2013-06-23 06:59:44.000000000 -0500
+++ setup.cfg	2013-11-25 20:34:23.000000000 -0600
@@ -11,7 +11,7 @@
 # build psycopg2. If pg_config is not in the path or is installed under a
 # different name uncomment the following option and set it to the pg_config
 # full path.
-#pg_config=
+pg_config=@PG_CONFIG@
 
 # Set to 1 to use Python datetime objects for default date/time representation.
 use_pydatetime=1
