--- CMakeLists.txt.orig	2012-01-30 11:14:36.000000000 +0900
+++ CMakeLists.txt	2012-01-30 11:14:40.000000000 +0900
@@ -18,7 +18,7 @@
 find_package( KdepimLibs REQUIRED ) 
 
 add_subdirectory( icons )
-add_subdirectory( doc )
+#add_subdirectory( doc )
 include_directories( ${KDE4_INCLUDES} ${KDEPIMLIBS_INCLUDE_DIR} core editor model transactions )
 
 option(KGPG_DEBUG_TRANSACTIONS "show commands and results of gpg calls in debug log" Off)
