--- CMakeLists.txt.orig	2012-08-13 18:35:52.000000000 +0900
+++ CMakeLists.txt	2012-11-12 01:51:48.000000000 +0900
@@ -18,7 +18,7 @@
 find_package( KdepimLibs REQUIRED ) 
 
 add_subdirectory( icons )
-add_subdirectory( doc )
+#add_subdirectory( doc )
 include_directories( ${KDE4_INCLUDES} ${KDEPIMLIBS_INCLUDE_DIR} )
 
 option(KGPG_DEBUG_TRANSACTIONS "show commands and results of gpg calls in debug log" Off)
