--- src/bindings/python/CMakeLists.txt.dist	2017-12-01 20:10:51.000000000 +0900
+++ src/bindings/python/CMakeLists.txt	2017-12-08 01:03:41.000000000 +0900
@@ -333,6 +333,11 @@
 else()
     set(PYTHON_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/python)
 endif()
+if (APPLE)
+    execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; from distutils.sysconfig import get_python_lib; sys.stdout.write(get_python_lib())"
+        OUTPUT_VARIABLE PYTHON_PACKAGE_INSTALL_DIR)
+endif(APPLE)
+MESSAGE( STATUS "PYTHON_PACKAGE_INSTALL_DIR: " ${PYTHON_PACKAGE_INSTALL_DIR} )
 
 ####################################################################
 #
