--- src/bindings/python/CMakeLists.txt.dist	2018-05-10 20:29:51.000000000 +0900
+++ src/bindings/python/CMakeLists.txt	2018-05-29 22:08:05.000000000 +0900
@@ -330,6 +330,11 @@
 else()
     set(PYTHON_PACKAGE_INSTALL_DIR ${MISC_PREFIX}bindings/python)
 endif()
+if (APPLE)
+    execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; from distutils.sysconfig import get_python_lib; sys.stdout.write(get_python_lib())"
+    OUTPUT_VARIABLE PYTHON_PACKAGE_INSTALL_DIR)
+endif(APPLE)
+MESSAGE( STATUS "PYTHON_PACKAGE_INSTALL_DIR: " ${PYTHON_PACKAGE_INSTALL_DIR} )
 
 ####################################################################
 #
