--- Makefile.in.orig	2009-05-05 01:22:33.000000000 -0700
+++ Makefile.in	2009-05-05 01:41:43.000000000 -0700
@@ -37,9 +37,10 @@
 
 .PHONY: install
 install: all
-	@cp -f $(BINARIES) $(BIN_DIR)
-	@cp -f $(LIBRARIES) $(LIB_DIR)
-	@cp -f $(HEADERS) $(INCLUDE_DIR)
+	@cp -f $(BINARIES) $(DESTDIR)$(BIN_DIR)
+	@cp -f $(LIBRARIES) $(DESTDIR)$(LIB_DIR)
+	@mkdir -m 755 -p $(DESTDIR)$(INCLUDE_DIR)
+	@cp -f $(HEADERS) $(DESTDIR)$(INCLUDE_DIR)
 	@echo "STP installed successfully."
 
 .PHONY: clean
