--- configure.ac.orig	2015-06-29 19:28:12.000000000 -0700
+++ configure.ac	2015-07-01 11:04:28.000000000 -0700
@@ -646,6 +646,17 @@
 AM_PATH_PYTHON([2.5.0],, [:])
 AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
 
+dnl Override standard python locations with MacPorts version specific ones
+
+AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print(sys.prefix);'`])
+AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print(sys.exec_prefix)'`])
+
+am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+am_cv_python_pyexecdir=$PYTHON_EXEC_PREFIX/lib/python$PYTHON_VERSION/site-packages
+
+AC_SUBST([pythondir], [$am_cv_python_pythondir])
+AC_SUBST([pyexecdir], [$am_cv_python_pyexecdir])
+
 #################
 # Check for Cairo
 #################
