--- configure.ac.orig	2015-12-01 08:33:44.000000000 -0800
+++ configure.ac	2015-12-01 10:04:11.000000000 -0800
@@ -646,6 +646,17 @@
 fi
 AM_CONDITIONAL(HAVE_DOT, test "x$have_dot" = "xyes")
 
+dnl Override standard python locations with MacPorts version specific ones
+
+AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print(sys.prefix);'`])
+AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print(sys.exec_prefix)'`])
+
+am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+am_cv_python_pyexecdir=$PYTHON_EXEC_PREFIX/lib/python$PYTHON_VERSION/site-packages
+
+AC_SUBST([pythondir], [$am_cv_python_pythondir])
+AC_SUBST([pyexecdir], [$am_cv_python_pyexecdir])
+
 #################
 # Check for Python (optional, used for introspection unittests)
 #################
