--- a/gst/goom/Makefile.am	2011-06-12 15:35:19.000000000 -0700
+++ b/gst/goom/Makefile.am	2011-06-12 15:37:03.000000000 -0700
@@ -4,8 +4,8 @@ plugin_LTLIBRARIES = libgstgoom.la
 PPC_FILES=ppc_zoom_ultimate.s ppc_drawings.s ppc_drawings.h ppc_zoom_ultimate.h
 MMX_FILES=mmx.c xmmx.c mmx.h xmmx.h
 
-# disable until someone figures out how to build these correctly on ppc
-ARCH_FILES = $(MMX_FILES)
+# disable until someone figures out how to build these correctly on ppc and x86_64
+#ARCH_FILES = $(MMX_FILES)
 #ARCH_FILES += $(PPC_FILES)
 				
 libgstgoom_la_SOURCES =						\
@@ -31,6 +31,9 @@ libgstgoom_la_LIBADD = $(GST_BASE_LIBS) 
 libgstgoom_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstgoom_la_LIBTOOLFLAGS = --tag=disable-static
 
+# mmx code doesn't build on x86_64, so disable it
+libgstgoom_la_CFLAGS += -mno-mmx
+
 EXTRA_DIST = $(PPC_FILES) $(MMX_FILES)
 
 .NOTPARALLEL:
