--- tests/meson.build
+++ tests/meson.build
@@ -71,7 +71,7 @@ installed_tests = {
   'pixbuf-construction': { 'suites': ['conform'], },
   'animation': {
     'suites': ['format'],
-    'skip': not (enabled_loaders.contains('ani') and enabled_loaders_contains('gif')),
+    'skip': not (enabled_loaders.contains('ani') and enabled_loaders.contains('gif')),
   },
   'cve-2015-4491': {
     'suites': ['security'],
