--- ngx_lua_distname.tmp/src/ngx_http_lua_headers_in.c.orig	2022-03-02 00:54:22.000000000 -0600
+++ ngx_lua_distname.tmp/src/ngx_http_lua_headers_in.c	2022-06-22 15:17:11.000000000 -0500
@@ -152,9 +152,15 @@
                  ngx_http_set_builtin_header },
 #endif
 
+#if defined(nginx_version) && nginx_version >= 1023000
+     { ngx_string("Cookie"),
+                  offsetof(ngx_http_headers_in_t, cookie),
+                  ngx_http_set_builtin_multi_header },
+#else
     { ngx_string("Cookie"),
                  offsetof(ngx_http_headers_in_t, cookies),
                  ngx_http_set_builtin_multi_header },
+#endif
 
     { ngx_null_string, 0, ngx_http_set_header }
 };
