--- DESCRIPTION.orig	2023-07-03 15:10:06.000000000 +0800
+++ DESCRIPTION	2023-07-04 09:43:04.000000000 +0800
@@ -15,7 +15,7 @@
 License: GPL-2
 Depends: R (>= 2.15.1), coda (>= 0.13), parallel, Matrix
 Imports: methods, stats, rjags (>= 4-4), rstan, R2OpenBUGS
-Suggests: MASS, lattice, R2WinBUGS, BRugs, rlecuyer
+Suggests: MASS, lattice, R2WinBUGS, rlecuyer
 SystemRequirements: none, one or more of JAGS (>= 3.0.0), WinBUGS (>=
         1.4), OpenBUGS (>= 3.2.2), Stan
 URL: https://groups.google.com/forum/#!forum/dclone-users,
