--- exec-wrapper.in.orig	2010-02-09 19:47:32.000000000 -0600
+++ exec-wrapper.in	2010-02-09 19:49:25.000000000 -0600
@@ -18,6 +18,8 @@
 #
 ################################################################################
 
+CC=${CC-cc}
+
 OPT_DEST="DEFAULT_DEST"
 OPT_MODE=4750 # -rwsr-x---
 OPT_MODE_A=4755 # -rwsr-xr-x
@@ -165,7 +167,7 @@
 checkUtil chmod
 checkUtil install
 checkUtil id
-checkUtil cc
+checkUtil ${CC}
 
 # Default owner and group.
 OPT_OWNER=$( id -un )
@@ -248,7 +250,7 @@
 WRAPPER_SRC="${tmp}/wrapper.c"
 WRAPPER_BIN="${tmp}/wrapper"
 genSource "${OPT_REAL}" >"${WRAPPER_SRC}"
-cc -o "${WRAPPER_BIN}" "${WRAPPER_SRC}" || die "Can't compile wrapper"
+${CC} -o "${WRAPPER_BIN}" "${WRAPPER_SRC}" || die "Can't compile wrapper"
 
 # Install wrapper.
 install -o "${OPT_OWNER}" -g "${OPT_GROUP}" -m "${OPT_MODE}" -s \
