--- bindings/lua/Makefile.in.orig	2014-09-29 15:46:44.000000000 -0700
+++ bindings/lua/Makefile.in	2014-12-03 05:41:40.000000000 -0800
@@ -416,7 +416,7 @@
 @LUA_NEED_OUR_COMPAT51_TRUE@compatdir = $(LUA_INSTALL_LMOD)
 @LUA_NEED_OUR_COMPAT51_TRUE@compat_DATA = compat-5.1r5/compat-5.1.lua
 rrd_la_LIBADD = -lrrd -lm
-rrd_la_LDFLAGS = -module -version-info $(LIB_VERSION_INFO) $(LUA_LFLAGS) -L$(top_builddir)/src/.libs
+rrd_la_LDFLAGS = -module -version-info $(LIB_VERSION_INFO) -L$(top_builddir)/src/.libs $(LUA_LFLAGS) 
 rrd_la_CFLAGS = -I$(top_srcdir)/src -I$(top_srcdir) $(LUA_CFLAGS) $(LUA_DEFINES) -DLIB_VERSION=\"$(LIB_VERSION)\"
 all: all-am
 
