From 7311dc1ca5eccf8ccfedd1d06c825b105f057e65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Adam=20Engstr=C3=B6m?= <adaengst@gmail.com>
Date: Sat, 20 Dec 2014 17:59:11 +0100
Subject: [PATCH 3/4] elflink: Possible use of unintialized variable

---
 bfd/elflink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bfd/elflink.c b/bfd/elflink.c
index f445912..6f79173 100644
--- a/bfd/elflink.c
+++ b/bfd/elflink.c
@@ -12413,7 +12413,7 @@ _bfd_elf_section_already_linked (bfd *abfd, asection *sec,
 		   abfd, sec);
 	      else if (sec->size != 0)
 		{
-		  bfd_byte *sec_contents, *l_sec_contents;
+		  bfd_byte *sec_contents, *l_sec_contents = NULL;
 
 		  if (!bfd_malloc_and_get_section (abfd, sec, &sec_contents))
 		    (*_bfd_error_handler)
-- 
2.1.2

