--- tnm/library/init.tcl.org	2006-05-06 12:29:49.000000000 -0700
+++ tnm/library/init.tcl	2006-05-06 12:30:53.000000000 -0700
@@ -166,6 +166,9 @@
 lappend tnm(mibs) TUBS-IBR-NFS-MIB TUBS-IBR-PROC-MIB
 lappend tnm(mibs) TUBS-IBR-LINUX-MIB
 
+# Add the MIB for NET-SNMP
+lappend tnm(mibs) /usr/share/snmp/mibs/NET-SNMP-MIB.txt
+
 # Define a proc to handle background errors if none exists. First try
 # to auto_load a definition so that we don't overwrite a definition
 # provided by another package.
