--- libmetrics/darwin/metrics.c.orig	2010-03-16 17:10:25.000000000 +0100
+++ libmetrics/darwin/metrics.c	2010-03-16 20:09:54.000000000 +0100
@@ -11,7 +11,12 @@
 
 #include <stdlib.h>
 #include "interface.h"
+/* kvm.h is no longer in /usr/include for Leopard */
+#include <AvailabilityMacros.h>
+#ifndef MAC_OS_X_VERSION_10_5
 #include <kvm.h>
+#endif
+
 #include <sys/sysctl.h>
 
 #include <mach/mach_init.h>
