Index: src/cpu/callback.cpp
===================================================================
--- src/cpu/callback.cpp	(Revision 3693)
+++ src/cpu/callback.cpp	(Revision 3694)
@@ -65,7 +65,7 @@
 	Bit16u oldcs=SegValue(cs);
 	Bit32u oldeip=reg_eip;
 	SegSet16(cs,CB_SEG);
-	reg_eip=call_idle*CB_SIZE;
+	reg_eip=CB_SOFFSET+call_idle*CB_SIZE;
 	DOSBOX_RunMachine();
 	reg_eip=oldeip;
 	SegSet16(cs,oldcs);
