--- mesonbuild/modules/gnome.py.orig
+++ mesonbuild/modules/gnome.py
@@ -816,6 +816,13 @@
         for incdir in typelib_includes:
             typelib_cmd += ["--includedir=" + incdir]
 
+        for target in girtargets:
+            if isinstance(target, build.SharedLibrary):
+                typelib_cmd += ["--shared-library=" +
+                        os.path.join(state.environment.get_prefix(),
+                                     state.environment.get_shared_lib_dir(),
+                                     target.filename)]
+
         typelib_target = self._make_typelib_target(state, typelib_output, typelib_cmd, kwargs)
 
         rv = [scan_target, typelib_target]
