From 8577d3ed558de0aa88afe7d047ee6b1132b049bb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 23 Feb 2024 07:00:55 +0800
Subject: [PATCH] fpm.toml: fix parsing

---
 fpm.toml | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git fpm.toml fpm.toml
index b2807d2..66eeb58 100644
--- fpm.toml
+++ fpm.toml
@@ -17,37 +17,37 @@ simulated-annealing = { git="https://github.com/jacobwilliams/simulated-annealin
 [library]
 source-dir="src/"
 
-[[ test ]]
+[[test]]
 name="root-finding-single-test"
 source-dir="test/"
 main="root_finding_single_test.f90"
 
-[[ test ]]
+[[test]]
 name="root-finding-mult-test"
 source-dir="test/"
 main="root_finding_mult_test.f90"
 
-[[ test ]]
+[[test]]
 name="static-opt-test"
 source-dir="test/"
 main="static_opt_test.f90"
 
-[[ test ]]
+[[test]]
 name="tauchen-test"
 source-dir="test/"
 main="tauchen_test.f90"
 
-[[ test ]]
+[[test]]
 name="markov-test"
 source-dir="test/"
 main="markov_test.f90"
 
-[[ test ]]
+[[test]]
 name="io-test"
 source-dir="test/"
 main="io_test.f90"
 
-[[ test ]]
+[[test]]
 name="global-opt-test"
 source-dir="test/"
 main="global_opt_test.f90"
