--- DESCRIPTION	2018-08-14 03:40:03.000000000 +0800
+++ DESCRIPTION	2024-01-10 18:34:27.000000000 +0800
@@ -6,7 +6,7 @@
 Author: Xiaofei Wang, Chandra Erdman, and John W. Emerson
 Maintainer: Xiaofei Wang <xiaofei.wang@yale.edu>
 Depends: graphics, methods, grid
-Suggests: DNAcopy, coda, strucchange, vegan, ggplot2, igraph
+Suggests: coda, strucchange, vegan, ggplot2, igraph
 Description: Provides an implementation of the Barry and Hartigan (1993) product
     partition model for the normal errors change point problem using Markov Chain
     Monte Carlo. It also extends the methodology to regression models on a connected
