--- cmake/CXXSniffer.cmake	2023-12-31 01:15:05.000000000 +0800
+++ cmake/CXXSniffer.cmake	2024-07-07 12:19:06.000000000 +0800
@@ -4,7 +4,6 @@
   set (LINUX true)
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set (DARWIN true)
-  set (CMAKE_CXX_FLAGS "-stdlib=libc++ ${CMAKE_CXX_FLAGS}")
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "kFreeBSD")
   set (KFREEBSD true)
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
