--- openvdb/openvdb/python/CMakeLists.txt.orig	2021-02-05 14:21:38.000000000 -0500
+++ openvdb/openvdb/python/CMakeLists.txt	2021-12-05 13:11:13.000000000 -0500
@@ -90,11 +90,11 @@
     list(APPEND OPENVDB_PYTHON_DEPS Python::NumPy)
   endif()
 else()
-  find_package(Python QUIET COMPONENTS ${OPENVDB_PYTHON_REQUIRED_COMPONENTS})
+  find_package(Python __MACPORTS_PYTHON_VERSION__ EXACT QUIET COMPONENTS ${OPENVDB_PYTHON_REQUIRED_COMPONENTS})
   OPENVDB_CHECK_PYTHON_VERSION(${Python_VERSION} ${Python_INCLUDE_DIRS})
 
   if(USE_NUMPY)
-    find_package(Python QUIET COMPONENTS NumPy)
+    find_package(Python __MACPORTS_PYTHON_VERSION__ EXACT QUIET COMPONENTS NumPy)
     if(NOT TARGET Python::NumPy)
         message(FATAL_ERROR "Could NOT find NumPy (Required is at least version "
           "\"${MINIMUM_NUMPY_VERSION}\")"
