--- openvdb/python/CMakeLists.txt.orig	2019-12-16 22:50:43.000000000 -0700
+++ openvdb/python/CMakeLists.txt	2019-12-16 22:51:32.000000000 -0700
@@ -90,7 +90,7 @@
   # Development and NumPy, specify all components at the same time when using
   # FindPython
   if(USE_NUMPY)
-    find_package(Python ${MINIMUM_PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development NumPy)
+    find_package(Python __MACPORTS_PYTHON_VERSION__ EXACT REQUIRED COMPONENTS Interpreter Development NumPy)
     if(Python_NumPy_VERSION VERSION_LESS MINIMUM_NUMPY_VERSION)
       message(FATAL_ERROR "Could NOT find NumPy (Required is at least version "
         "\"${MINIMUM_NUMPY_VERSION}\")"
@@ -103,7 +103,7 @@
     endif()
     list(APPEND OPENVDB_PYTHON_DEPS Python::NumPy)
   else()
-    find_package(Python ${MINIMUM_PYTHON_VERSION} REQUIRED COMPONENTS Interpreter Development)
+    find_package(Python __MACPORTS_PYTHON_VERSION__ EXACT REQUIRED COMPONENTS Interpreter Development)
   endif()
   list(APPEND OPENVDB_PYTHON_DEPS Python::Python)
 endif()
