--- openvdb/openvdb/CMakeLists.txt
+++ openvdb/openvdb/CMakeLists.txt	2024-04-28 09:18:21.000000000 +0800
@@ -250,6 +250,8 @@
   list(APPEND OPENVDB_CORE_DEPENDENT_LIBS Boost::headers)
 endif()
 
+list(APPEND OPENVDB_CORE_DEPENDENT_LIBS atomic)
+
 if(WIN32)
   # Boost headers contain #pragma commands on Windows which causes Boost
   # libraries to be linked in automatically. Custom boost installations
