--- old/common.h	2022-03-10 23:10:00.000000000 -0500
+++ common.h	2022-03-10 23:05:40.000000000 -0500
@@ -42,6 +42,7 @@
 #define MANUAL_SYNTAX_URL "http://www.lichteblau.com/ldapvi/manual#syntax"
 #define MANUAL_LDIF_URL "http://www.lichteblau.com/ldapvi/manual#syntax-ldif"
 #define RFC_2849_URL "http://www.rfc-editor.org/rfc/rfc2849.txt"
+char ldap_bv2dn_x ();
 
 typedef struct tschema {
 	GHashTable *classes;
@@ -282,6 +283,7 @@
 int adjoin_ptr(GPtrArray *, void *);
 void init_dialog(tdialog *, enum dialog_mode, char *, char *);
 void dialog(char *header, tdialog *, int, int);
+void fdcp(int fdsrc, int fddst);
 
 /*
  * schema.c
@@ -343,6 +345,7 @@
 int g_string_append_ssha(GString *string, char *key);
 int g_string_append_md5(GString *string, char *key);
 int g_string_append_smd5(GString *string, char *key);
+int on_exit(void (*function)(int, void *), void *arg);
 
 /*
  * base64.c
