--- GNUmakefile.in.orig	2007-05-05 13:17:26.000000000 +0300
+++ GNUmakefile.in	2007-10-21 01:42:20.000000000 +0300
@@ -25,10 +25,10 @@
 
 .PHONY: install
 install: ldapvi
-	mkdir -p @bindir@ @mandir@/man1/ @prefix@/share/doc/ldapvi
+	mkdir -p @bindir@ @datarootdir@/man/man1/ @datarootdir@/doc/ldapvi
 	@INSTALL_PROGRAM@ ldapvi @bindir@
-	@INSTALL_PROGRAM@ -m 644 ldapvi.1 @mandir@/man1/
-	@INSTALL_PROGRAM@ -m 644 manual/manual.css manual/manual.xml manual/bg.png manual/html.xsl @prefix@/share/doc/ldapvi
+	@INSTALL_PROGRAM@ -m 644 ldapvi.1 @datarootdir@/man/man1/
+	@INSTALL_PROGRAM@ -m 644 manual/manual.css manual/manual.xml manual/bg.png manual/html.xsl @datarootdir@/doc/ldapvi
 
 distclean:
 	rm -rf configure GNUmakefile config.h config.log autom4te.cache config.status
