--- a/gdb/tui/tui-io.c	2010-05-25 10:48:44.000000000 -0500
+++ b/gdb/tui/tui-io.c	2019-08-23 01:51:22.000000000 -0500
@@ -129,10 +129,10 @@
 struct ui_out *tui_old_uiout;
 
 /* Readline previous hooks.  */
-static Function *tui_old_rl_getc_function;
-static VFunction *tui_old_rl_redisplay_function;
-static VFunction *tui_old_rl_prep_terminal;
-static VFunction *tui_old_rl_deprep_terminal;
+static rl_intfunc_t *tui_old_rl_getc_function;
+static rl_voidfunc_t *tui_old_rl_redisplay_function;
+static rl_voidfunc_t *tui_old_rl_prep_terminal;
+static rl_voidfunc_t *tui_old_rl_deprep_terminal;
 static int tui_old_readline_echoing_p;
 
 /* Readline output stream.
