--- lib/procutils.c	2021-02-10 03:26:03.000000000 -0500
+++ lib/procutils.c	2021-02-21 23:24:02.000000000 -0500
@@ -13,7 +13,9 @@
 #include <string.h>
 #include <errno.h>
 #include <sys/stat.h>
-#include <sys/vfs.h>
+//#include <sys/vfs.h>
+#include <sys/param.h>
+#include <sys/mount.h>
 #include <sys/types.h>
 #include <dirent.h>
 #include <ctype.h>
--- include/statfs_magic.h	2021-02-02 10:41:37.000000000 -0500
+++ include/statfs_magic.h	2021-02-21 23:43:37.000000000 -0500
@@ -1,7 +1,7 @@
 #ifndef UTIL_LINUX_STATFS_MAGIC_H
 #define UTIL_LINUX_STATFS_MAGIC_H
 
-#include <sys/statfs.h>
+//#include <sys/statfs.h>
 
 /*
  * If possible then don't depend on internal libc __SWORD_TYPE type.
