--- gl/m4/gc.m4.orig	2012-05-21 09:07:06.000000000 -0500
+++ gl/m4/gc.m4	2014-07-22 06:35:50.000000000 -0500
@@ -15,7 +15,7 @@
 /* GCRY_MODULE_ID_USER was added in 1.4.4 and gc-libgcrypt.c
    will fail on startup if we don't have 1.4.4 or later, so
    test for it early. */
-#if !defined GCRY_MODULE_ID_USER
+#if !defined GCRY_MODULE_ID_USER && !defined GCRYPT_VERSION_NUMBER
 error too old libgcrypt
 #endif
 ])
