--- Makefile.PL.orig	2014-11-22 07:28:22.000000000 -0800
+++ Makefile.PL	2014-12-04 15:32:06.000000000 -0800
@@ -13,16 +13,16 @@
 
 requires_external_cc();
 
-    inc '-I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include';
+    inc '-I@PREFIX@/include/ssl';
 
-    libs '-L/usr/lib -L/usr/local/lib -L/usr/local/ssl/lib -lcrypto';
+    libs '-L@PREFIX@/lib -lcrypto';
 
     if ($Config::Config{myuname} =~ /darwin/i) {
-      cc_optimize_flags('-O2 -g -Wall -Werror -Wno-deprecated-declarations');
+      cc_optimize_flags('-O2 -g -Wall -Wno-deprecated-declarations');
     } elsif ($Config::Config{myuname} =~ /sunos|solaris/i) {
       # Any SunStudio flags?
     } else {
-      cc_optimize_flags('-O2 -g -Wall -Werror');
+      cc_optimize_flags('-O2 -g -Wall');
     }
 
 auto_install();
