Index: kig/misc/lists.cc
===================================================================
RCS file: /home/kde/kdeedu/kig/misc/lists.cc,v
retrieving revision 1.11
diff -u -u -r1.11 lists.cc
--- kig/misc/lists.cc	4 Jun 2003 12:43:56 -0000	1.11
+++ kig/misc/lists.cc	10 Jul 2003 02:14:47 -0000
@@ -234,7 +234,7 @@
 
   QDomElement docelem = doc.createElement( "KigMacroFile" );
   docelem.setAttribute( "Version", "0.5.1" );
-  docelem.setAttribute( "Number", ms.size() );
+  docelem.setAttribute( "Number", (unsigned int)ms.size() );
 
   for ( uint i = 0; i < ms.size(); ++i )
   {
Index: kig/misc/object_hierarchy.cc
===================================================================
RCS file: /home/kde/kdeedu/kig/misc/object_hierarchy.cc,v
retrieving revision 1.11
diff -u -u -r1.11 object_hierarchy.cc
--- kig/misc/object_hierarchy.cc	3 Jul 2003 11:42:39 -0000	1.11
+++ kig/misc/object_hierarchy.cc	10 Jul 2003 02:14:47 -0000
@@ -464,7 +464,7 @@
       ObjectImp* imp = ObjectImpFactory::instance()->deserialize( typen, e );
       newnode = new PushStackNode( imp );
     };
-    mnodes.resize( kMax( id - mnumberofargs, mnodes.size() ) );
+    mnodes.resize( kMax( id - mnumberofargs, (unsigned int)mnodes.size() ) );
     mnodes[id - mnumberofargs - 1] = newnode;
   };
 }
Index: kstars/kstars/kstarsdata.h
===================================================================
RCS file: /home/kde/kdeedu/kstars/kstars/kstarsdata.h,v
retrieving revision 1.50
diff -u -u -r1.50 kstarsdata.h
--- kstars/kstars/kstarsdata.h	4 Jul 2003 17:53:43 -0000	1.50
+++ kstars/kstars/kstarsdata.h	10 Jul 2003 02:14:47 -0000
@@ -54,6 +54,10 @@
 #include "jupitermoons.h"
 #include "indidriver.h"
 
+#include <iostream>
+using std::cout;
+using std::endl;
+
 #define NSAOFILES 40
 #define NMWFILES  11
 #define NNGCFILES 13
