--- src/Makefile.in.orig	2003-04-04 14:28:32.000000000 -0600
+++ src/Makefile.in	2014-10-31 05:03:10.000000000 -0500
@@ -9,6 +9,7 @@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
 PREFIX = @prefix@
+MANDIR = @mandir@
 
 INSTALL = @INSTALL@
 
@@ -21,9 +22,9 @@
 	$(CC) -c $(INCLUDES) $(CFLAGS) $(LDFLAGS) $< -o $@
 
 real_install:
-	../mkinstalldirs $(DESTDIR)$(PREFIX)/bin/ $(DESTDIR)$(PREFIX)/man/man1
+	../mkinstalldirs $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/share/man/man1
 	$(INSTALL) $(PROG) $(DESTDIR)$(PREFIX)/bin/$(PROG)
-	$(INSTALL) ../doc/*.1 $(DESTDIR)$(PREFIX)/man/man1	
+	$(INSTALL) ../doc/*.1 $(DESTDIR)$(PREFIX)/share/man/man1
 
 install: real_install
 
