--- src/profiles.c.orig	2014-09-17 10:09:56.000000000 -0700
+++ src/profiles.c	2014-09-17 10:11:37.000000000 -0700
@@ -287,7 +287,7 @@
     return NULL;
   }
 
-  if (av_find_stream_info (ctx) < 0)
+  if (avformat_find_stream_info (ctx, NULL) < 0)
   {
     if (dlna->verbosity)
       fprintf (stderr, "can't find stream info\n");
@@ -334,7 +334,7 @@
     p = p->next;
   }
 
-  av_close_input_file (ctx);
+  avformat_close_input (&ctx);
   free (codecs);
   return profile;
 }
