--- src/Makefile	2007-11-26 15:47:43.000000000 -0500
+++ src/Makefile.new	2008-01-12 15:03:09.000000000 -0500
@@ -13,7 +13,7 @@
 endif
 
 LIBNAME = libdlna
-LIBNAME_SHARED = ${LIBNAME}.so
+LIBNAME_SHARED = ${LIBNAME}.dylib
 LIBNAME_STATIC = ${LIBNAME}.a
 LIBNAME_VERSION = $(LIBNAME_SHARED).$(VERSION)
 LIBNAME_MAJOR = $(LIBNAME_SHARED).$(shell echo $(VERSION) | cut -f1 -d.)
@@ -89,13 +89,12 @@
 	@echo "#############################################"
 
 lib_shared: lib_shared_info_pre $(LOBJS) lib_shared_info_post
-	$(CC) -shared -Wl,-soname,$(LIBNAME_MAJOR) $(LOBJS) \
-	  $(LDFLAGS) $(EXTRALIBS) -o $(LIBNAME_VERSION)
+	$(CC) -dynamiclib $(LOBJS) $(LDFLAGS) $(EXTRALIBS) -o $(LIBNAME_VERSION)
 	$(LN) -sf $(LIBNAME_VERSION) $(LIBNAME_MAJOR)
 	$(LN) -sf $(LIBNAME_MAJOR) $(LIBNAME_SHARED)
 
 clean:
-	-$(RM) -f *.o *.lo *.a *.so*
+	-$(RM) -f *.o *.lo *.a *.dylib*
 	-$(RM) -f .depend
 
 install_static: lib_static
