--- gst-libs/gst/gl/gstglfuncs.h.old	2018-03-23 21:44:42.000000000 +0100
+++ gst-libs/gst/gl/gstglfuncs.h	2018-07-04 11:23:52.000000000 +0200
@@ -54,7 +54,7 @@
 
 /* OpenGL for desktop systems */
 #if GST_GL_HAVE_OPENGL
-# ifdef __APPLE__
+# if GST_GL_HAVE_WINDOW_COCOA
 #  include <OpenGL/OpenGL.h>
 #  include <OpenGL/gl.h>
 #  if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
--- m4/gst-gl.m4.orig	2018-07-18 06:32:30.000000000 -0700
+++ m4/gst-gl.m4	2018-08-04 18:21:58.000000000 -0700
@@ -338,12 +338,16 @@
 if test "x$HAVE_GL" = "xyes" -a "x$HAVE_GLES2" = "xyes"; then
   GLES3_H_DEFINE=0
   GLES3EXT3_H_DEFINE=0
+  NEED_COCOA_DEFINE=0
   if test "x$HAVE_GLES3_H" == "xyes"; then
     GLES3_H_DEFINE=1
   fi
   if test "x$HAVE_GLES3EXT3_H" == "xyes"; then
     GLES3EXT3_H_DEFINE=1
   fi
+  if test "x$NEED_COCOA" == "xyes"; then
+    NEED_COCOA_DEFINE=1
+  fi
   GL_INCLUDES="
 #ifdef __GNUC__
 #  pragma GCC diagnostic push
@@ -367,7 +371,7 @@
 #   include <GLES2/gl2ext.h>
 #  endif
 # endif
-# ifdef __APPLE__
+# if $NEED_COCOA_DEFINE
 #  include <OpenGL/OpenGL.h>
 #  include <OpenGL/gl.h>
 #  if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
@@ -877,7 +881,7 @@
 
 if test "x$USE_OPENGL" = "xyes"; then
   GL_INCLUDES="$GL_INCLUDES
-# ifdef __APPLE__
+# if $GST_GL_HAVE_WINDOW_COCOA
 #  include <OpenGL/OpenGL.h>
 #  include <OpenGL/gl.h>
 # else
