--- CMakeLists.txt.orig	2019-05-23 09:57:03.000000000 +0200
+++ CMakeLists.txt	2019-05-23 09:59:46.000000000 +0200
@@ -1271,7 +1271,7 @@
         find_package(Gnuradio)
     endif()
 endif()
-
+set(PC_GNURADIO_RUNTIME_VERSION "3.7.13")
 
 ################################################################################
 # Doxygen - http://www.stack.nl/~dimitri/doxygen/index.html (OPTIONAL, used if found)
