--- protoMakefile.orig	2020-03-14 21:01:24.000000000 +0000
+++ protoMakefile	2021-12-29 20:43:38.000000000 +0000
@@ -47,7 +47,7 @@
 
 CPPFLAGS =
 CFLAGS =
-CC = cc $(CPPFLAGS) $(CFLAGS) -c
+CC = cc
 
 # Define LINK1 and LINK2 so that the command
 #
@@ -61,7 +61,7 @@
 # LINK1 = cc -s
 # LINK2 = -o
 
-LINK1 = cc
+LINK1 = $(CC)
 LINK2 = -o
 
 # Define RM so that the command
@@ -93,10 +93,10 @@
 OBJS = buffer$O charset$O errmsg$O par$O reformat$O
 
 .c$O:
-	$(CC) $<
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) $<
 
 par$E: $(OBJS)
-	$(LINK1) $(OBJS) $(LINK2) par$E
+	$(LINK1) $(OBJS) $(LINK2) $@ $(LDFLAGS)
 
 buffer$O: buffer.c buffer.h errmsg.h
 
