--- Source/WTF/wtf/Assertions.h.orig	2013-01-18 15:03:57.000000000 -0800
+++ Source/WTF/wtf/Assertions.h	2013-01-18 15:04:46.000000000 -0800
@@ -381,7 +381,8 @@ while (0)
 // a function. Hence it uses macro naming convention.
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wmissing-noreturn"
-static inline void UNREACHABLE_FOR_PLATFORM()
+__attribute__ ((__always_inline__))
+static __inline__ void UNREACHABLE_FOR_PLATFORM()
 {
     ASSERT_NOT_REACHED();
 }
