--- configure.ac.orig	2012-12-09 12:01:04.000000000 -0600
+++ configure.ac	2013-01-30 16:08:56.000000000 -0600
@@ -1215,14 +1215,6 @@
    AC_DEFINE([NDEBUG], [1], [Define to disable debugging features])
 fi
 
-# Add the appropriate 'O' level for optimized builds
-if test "$enable_optimizations" = "yes"; then
-   CXXFLAGS="$CXXFLAGS -O2"
-   CFLAGS="$CFLAGS -O2"
-else
-   CXXFLAGS="$CXXFLAGS -O0"
-   CFLAGS="$CFLAGS -O0"
-fi
 
 PKG_CHECK_MODULES([LIBSOUP],
                   [libsoup-2.4 >= $LIBSOUP_REQUIRED_VERSION])
