--- configure.ac.orig	2012-08-06 06:45:10.000000000 -0700
+++ configure.ac	2012-08-17 16:40:51.000000000 -0700
@@ -409,10 +409,6 @@ AC_MSG_RESULT([$with_unicode_backend])
 if test "$with_unicode_backend" = "icu"; then
     # TODO: use pkg-config (after CFLAGS in their .pc files are cleaned up)
     case "$host" in
-        *-*-darwin*)
-            UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
-            UNICODE_LIBS="-licucore"
-            ;;
         *-*-mingw*)
             UNICODE_CFLAGS=""
             UNICODE_LIBS="-licui18n -licuuc"
