--- configure.orig	2012-04-16 15:59:17.000000000 -0700
+++ configure	2012-04-16 15:59:37.000000000 -0700
@@ -19205,10 +19205,6 @@ $as_echo "$with_unicode_backend" >&6; }
 
 if test "$with_unicode_backend" = "icu"; then
         case "$host" in
-            *-*-darwin*)
-		UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
-		UNICODE_LIBS="-licucore"
-                ;;
             *-*-mingw*)
 		UNICODE_CFLAGS=""
 		UNICODE_LIBS="-licuin -licuuc"
