--- Source/autotools/webkit.m4.orig	2012-02-01 12:37:41.000000000 -0800
+++ Source/autotools/webkit.m4	2012-06-03 12:05:52.000000000 -0700
@@ -134,10 +134,6 @@ AC_MSG_RESULT([$with_unicode_backend])
 
 if test "$with_unicode_backend" = "icu"; then
         case "$host" in
-            *-*-darwin*)
-		UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
-		UNICODE_LIBS="-licucore"
-                ;;
             *-*-mingw*)
 		UNICODE_CFLAGS=""
 		UNICODE_LIBS="-licuin -licuuc"
