--- Source/WebCore/GNUmakefile.am	(revision 106782)
+++ Source/WebCore/GNUmakefile.am	(revision 106847)
@@ -836,4 +836,5 @@
 	$(LIBXML_CFLAGS) \
 	$(LIBXSLT_CFLAGS) \
+	$(PANGO_CFLAGS) \
 	$(SQLITE3_CFLAGS) \
 	$(UNICODE_CFLAGS) \
--- Tools/GNUmakefile.am	(revision 106786)
+++ Tools/GNUmakefile.am	(revision 106847)
@@ -66,4 +66,5 @@
 	$(CAIRO_CFLAGS) \
 	$(LIBSOUP_CFLAGS) \
+	$(PANGO_CFLAGS) \
 	-I$(top_builddir)/DerivedSources \
 	-I$(top_builddir)/DerivedSources/WebCore
--- Source/WebCore/GNUmakefile.am.orig	2012-04-19 22:57:39.000000000 -0700
+++ Source/WebCore/GNUmakefile.am	2012-04-19 22:58:17.000000000 -0700
@@ -134,6 +134,7 @@ endif # END USE_FREETYPE
 # ---
 if USE_PANGO
 webcoregtk_cppflags += \
+	$(PANGO_CFLAGS) \
 	-DWTF_USE_PANGO=1 \
 	-I$(srcdir)/Source/WebCore/platform/graphics/pango
 endif # END USE_PANGO
