--- JavaScriptCore/wtf/FastMalloc.cpp.orig	2010-05-27 19:00:10.000000000 +0200
+++ JavaScriptCore/wtf/FastMalloc.cpp	2010-05-27 19:00:40.000000000 +0200
@@ -1381,14 +1381,12 @@
   // Bytes allocated from system
   uint64_t system_bytes_;
 
-#if USE_BACKGROUND_THREAD_TO_SCAVENGE_MEMORY
   // Number of pages kept in free lists that are still committed.
   Length free_committed_pages_;
 
   // Minimum number of free committed pages since last scavenge. (Can be 0 if
   // we've committed new pages since the last scavenge.)
   Length min_free_committed_pages_since_last_scavenge_;
-#endif
 
   bool GrowHeap(Length n);
 
