--- WebKitTools/DumpRenderTree/gtk/TestNetscapePlugin/TestNetscapePlugin.cpp.orig	2009-06-11 17:05:32.000000000 -0700
+++ WebKitTools/DumpRenderTree/unix/TestNetscapePlugin/TestNetscapePlugin.cpp	2009-06-15 13:59:40.000000002 -0700
@@ -26,6 +26,13 @@
  */
 
 #include "config.h"
+
+#define Cursor       QD_Cursor
+#define WindowPtr    QD_WindowPtr
+#define Picture      QD_Picture
+#define BOOL         OSX_BOOL
+#define EventType    HIT_EventType
+
 #include "PluginObject.h"
 
 #include "npapi.h"
@@ -36,6 +43,13 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+
+#undef Cursor
+#undef WindowPtr
+#undef Picture
+#undef BOOL
+#undef EventType
+
 #include <X11/Xlib.h>
 
 static void log(NPP instance, const char* format, ...)
