--- Source/WebCore/plugins/PluginView.cpp.orig	2015-05-20 02:03:24.000000000 -0700
+++ Source/WebCore/plugins/PluginView.cpp	2015-08-08 15:02:30.000000000 -0700
@@ -839,7 +839,7 @@
 #if defined(XP_MACOSX)
     , m_contextRef(0)
 #endif
-#if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
+#if PLATFORM(X11) && ENABLE(NETSCAPE_PLUGIN_API)
     , m_hasPendingGeometryChange(true)
     , m_drawable(0)
     , m_visual(0)
--- Source/WebCore/plugins/PluginView.h.orig	2015-05-20 02:03:24.000000000 -0700
+++ Source/WebCore/plugins/PluginView.h	2015-08-08 15:02:30.000000000 -0700
@@ -378,7 +378,7 @@
         void setNPWindowIfNeeded();
 #endif
 
-#if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
+#if PLATFORM(X11) && ENABLE(NETSCAPE_PLUGIN_API)
         bool m_hasPendingGeometryChange;
         Pixmap m_drawable;
         Visual* m_visual;
--- Source/WebCore/GNUmakefile.list.am.orig	2015-06-16 09:12:37.000000000 +0200
+++ Source/WebCore/GNUmakefile.list.am	2015-06-16 09:13:38.000000000 +0200
@@ -6315,11 +6315,9 @@
 
 if TARGET_QUARTZ
 if !TARGET_X11
-if ENABLE_WEBKIT2
 webcore_sources += \
 	Source/WebCore/plugins/PluginPackageNone.cpp \
 	Source/WebCore/plugins/PluginViewNone.cpp
-endif # END ENABLE_WEBKIT2
 endif # END !TARGET_X11
 platformgtk_sources += \
 	Source/WebCore/platform/cairo/WidgetBackingStoreCairo.h \
