--- configure.orig	2012-04-18 17:55:13.000000000 -0500
+++ configure	2012-04-18 17:56:11.000000000 -0500
@@ -22086,7 +22086,7 @@
 
 # check for XRender under Linux/Unix. Some linkers require explicit
 # linkage (like GNU Gold), so we cannot rely on GTK+ pulling XRender
-if test "$os_win32" = "no"; then
+if test "$with_target" = "x11"; then
 
 pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for XRENDER" >&5
