--- src/Makefile.orig	2014-04-08 13:39:40.000000000 -0500
+++ src/Makefile	2014-06-23 20:35:19.000000000 -0500
@@ -321,8 +321,8 @@
 ifeq (freebsd,$(OS))
 	# Running under FreeBSD
 ifdef ENABLE_NLS
-	CFLAGS_OS     = -I/usr/local/include
-	LDFLAGS_EXTRA = -L/usr/local/lib
+	CFLAGS_OS     = -I$(PREFIX)/include
+	LDFLAGS_EXTRA = -L$(PREFIX)/lib
 	LIBS_EXTRA    = -lintl
 endif
 endif
@@ -330,8 +330,8 @@
 ifeq (Darwin, $(shell uname -s))
 	OS = Darwin
 ifdef ENABLE_NLS
-	CFLAGS_OS     = -I/usr/local/include
-	LDFLAGS_EXTRA = -L/usr/local/lib
+	CFLAGS_OS     = -I$(PREFIX)/include
+	LDFLAGS_EXTRA = -L$(PREFIX)/lib
 	LIBS_EXTRA    = -lintl
 endif
 endif
