--- Makefile.in.orig	2012-01-04 03:39:27.000000000 -0600
+++ Makefile.in	2012-01-05 23:57:47.000000000 -0600
@@ -310,7 +310,7 @@
 	@ln -sf $(PWD)/priv temp_ts/lib/$(APPLICATION)-$(VERSION)/priv
 	@ln -sf $(PWD)/builder.beam temp_ts/lib/$(APPLICATION)-$(VERSION)/
 	@ln -sf $(PWD) temp_ts/lib/$(APPLICATION)-$(VERSION)
-	@echo -n "build main app boot script ... "
+	@printf "build main app boot script ... "
 	@(cd temp_ts/lib/$(APPLICATION)-$(VERSION) \
 	 && echo $(BUILD_OPTIONS) > $(BUILD_OPTIONS_FILE) \
 	 && $(ERL) -noshell -s builder go -s init stop >> $(BUILDER_LOG) 2>&1 \
@@ -328,7 +328,7 @@
 	@ln -sf $(PWD)/include temp_tsc/lib/$(CONTROLLER_APPLICATION)-$(VERSION)/include
 	@ln -sf $(PWD)/priv temp_tsc/lib/$(CONTROLLER_APPLICATION)-$(VERSION)/priv
 	@ln -sf $(PWD)/builder.beam temp_tsc/lib/$(CONTROLLER_APPLICATION)-$(VERSION)/
-	@echo -n "build controller boot script ... "
+	@printf "build controller boot script ... "
 	@(cd temp_tsc/lib/$(CONTROLLER_APPLICATION)-$(VERSION) \
 	 && echo $(BUILD_OPTIONS) > $(BUILD_OPTIONS_FILE) \
 	 && $(ERL) -noshell -s builder go -s init stop >> $(BUILDER_LOG) 2>&1 \
@@ -346,7 +346,7 @@
 	@ln -sf $(PWD)/include temp_tsr/lib/$(RECORDER_APPLICATION)-$(VERSION)/include
 	@ln -sf $(PWD)/priv temp_tsr/lib/$(RECORDER_APPLICATION)-$(VERSION)/priv
 	@ln -sf $(PWD)/builder.beam temp_tsr/lib/$(RECORDER_APPLICATION)-$(VERSION)/
-	@echo -n "build recorder boot script ... "
+	@printf "build recorder boot script ... "
 	@(cd temp_tsr/lib/$(RECORDER_APPLICATION)-$(VERSION) \
 	 && echo $(BUILD_OPTIONS) > $(BUILD_OPTIONS_FILE) \
 	 && $(ERL) -noshell -s builder go -s init stop  >> $(BUILDER_LOG) 2>&1 \
