--- DESCRIPTION	2024-03-12 19:20:03
+++ DESCRIPTION	2024-03-13 06:10:18
@@ -28,10 +28,10 @@
 Imports: fastglm, forecast, glmnet, Matrix, methods, Rcpp (>= 0.11.0),
         stats, tseries, utils
 Suggests: abind, breakfast, changepoint, cpm, CptNonPar, ecp, fpop,
-        gfpop, ggplot2, gridExtra, InspectChangepoint, jointseg, knitr,
+        gfpop, ggplot2, gridExtra, InspectChangepoint, knitr,
         lubridate, matrixStats, mcp, mockthat, mosum, mvtnorm, not,
-        numDeriv, Rbeast, rmarkdown, segmented, stepR, strucchange,
-        testthat (>= 3.0.0), VARDetect, wbs, xml2, zoo
+        numDeriv, rmarkdown, segmented, stepR, strucchange,
+        testthat (>= 3.0.0), wbs, xml2, zoo
 LinkingTo: progress, Rcpp, RcppArmadillo, testthat
 VignetteBuilder: knitr
 Config/testthat/edition: 3
