--- yorick/Makefile.in.orig	2014-10-21 07:15:34.000000000 -0500
+++ yorick/Makefile.in	2014-10-31 05:15:45.000000000 -0500
@@ -9,7 +9,7 @@
 # ----------------------------------------------------- optimization flags
 
 # options for make command line, e.g.-   make COPT=-g TGT=exe
-COPT=$(COPT_DEFAULT) -L../lib -I../include
+COPT=$(COPT_DEFAULT) -L../lib/.libs -I../include
 TGT=$(DEFAULT_TGT)
 
 # ------------------------------------------------ macros for this package
