--- ./build-release.sh	2023-03-22 11:32:06.000000000 -0400
+++ ./build-release.sh	2023-03-22 11:32:33.000000000 -0400
@@ -161,26 +161,4 @@
   echo "REPORTED:    $BUILT_VER"
 fi
 
-# pack
-cp $SRC/LICENSE $DIR
-case $EXT in
-  tar.bz2)
-    tar -C $DIR -cjf $OUT $(basename $BIN) LICENSE
-  ;;
-  zip)
-    zip $OUT -j $BIN LICENSE
-  ;;
-esac
-
-# report
-echo "PACKED:      $OUT ($(du -sh $OUT|awk '{print $1}'))"
-case $EXT in
-  tar.bz2)
-    tar -jvtf $OUT
-  ;;
-  zip)
-    unzip -l $OUT
-  ;;
-esac
-
 popd &> /dev/null
