Index: zzuf/trunk/src/libzzuf/lib-mem.c
===================================================================
--- a/zzuf/trunk/src/libzzuf/lib-mem.c
+++ b/zzuf/trunk/src/libzzuf/lib-mem.c
@@ -132,8 +132,4 @@
 /* setrlimit(RLIMIT_AS) is ignored on OS X, we need to check memory usage
  * from inside the process. Oh, and getrusage() doesn't work either. */
-#if defined HAVE_MACH_TASK_H
-vm_size_t mach_page_size;
-#endif
-
 static int memory_exceeded(void)
 {
@@ -144,6 +140,5 @@
     if (task_info(mach_task_self(), TASK_BASIC_INFO,
                   (task_info_t)&tbi, &mmtn) == KERN_SUCCESS
-         && (int64_t)tbi.resident_size * mach_page_size / 1048576
-                  > (int64_t)_zz_memory)
+         && (int64_t)tbi.resident_size / 1048576 > (int64_t)_zz_memory)
         return 1;
 #endif
@@ -157,8 +152,4 @@
     LOADSYM(malloc);
     LOADSYM(realloc);
-
-#if defined HAVE_MACH_TASK_H
-    host_page_size(mach_host_self(), &mach_page_size);
-#endif
 }
 