--- setup.py	Thu Apr 29 14:56:28 2004
+++ setup.py.new	Thu Apr 29 23:36:55 2004
@@ -21,8 +21,5 @@
                      """,
       version="0.1",
       scripts=["linda_server.py"],
-      data_files=[("/usr/share/linda/examples/", glob.glob("examples/*.py")),
-                  ("/usr/share/linda/doc", glob.glob("doc/html/*")),
-                  ("/usr/share/linda/", ["README"])],
       packages=["linda", "linda.typeconvert"],
       package_dir={"linda": ""})
