--- setup.py	Mon May  3 13:02:51 2004
+++ setup.py.new	Mon May  3 21:02:59 2004
@@ -113,9 +113,6 @@
                      Topic :: System :: Distributed Computing""".split("\n"),
       version="0.2",
       scripts=["linda_server.py"],
-      data_files=[(root_path+"/examples", glob.glob("examples/*.py")),
-                  (root_path+"/doc", glob.glob("doc/html/*")),
-                  (root_path, ["README"])],
       packages=["linda", "linda.typeconvert"],
       package_dir={"linda": ""},
       cmdclass=mycmdclass)
