--- CMakeLists.txt.orig	2021-12-25 09:18:20.000000000 -0500
+++ CMakeLists.txt	2021-12-25 09:40:15.000000000 -0500
@@ -1052,6 +1052,7 @@
 
 ## Install stuff
 
+option(ENABLE_APP_BUNDLING "Build an .app bundle" ON)
 option(DISABLE_CPACK_BUNDLING "Build an .app bundle without CPack" OFF)
 if(WIN32 AND NOT UNIX)
 
@@ -1074,8 +1075,8 @@
     include(InstallRequiredSystemLibraries)
   endif()
 
-else(WIN32 AND NOT UNIX)
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND DISABLE_CPACK_BUNDLING)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  if(ENABLE_APP_BUNDLING AND DISABLE_CPACK_BUNDLING)
 
     set(INFOPLIST_CFBUNDLEEXECUTABLE "supertux2")
 
@@ -1089,7 +1090,8 @@
 
     install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/data/images/engine/icons/supertux.png ${CMAKE_CURRENT_SOURCE_DIR}/data/images/engine/icons/supertux.icns DESTINATION "SuperTux.app/Contents/Resources/")
 
-  else(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND DISABLE_CPACK_BUNDLING)
+  endif(ENABLE_APP_BUNDLING AND DISABLE_CPACK_BUNDLING)
+else(WIN32 AND NOT UNIX)
 
     set(INSTALL_SUBDIR_BIN "games" CACHE STRING "Installation subdir for binaries")
     set(INSTALL_SUBDIR_SHARE "share/games/supertux2" CACHE STRING "Installation subdir for data")
@@ -1113,7 +1115,6 @@
 
     install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/supertux2.svg DESTINATION "share/icons/hicolor/scalable/apps")
 
-  endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND DISABLE_CPACK_BUNDLING)
 endif(WIN32 AND NOT UNIX)
 
 if(UBUNTU_TOUCH)
@@ -1188,7 +1189,7 @@
 include(InstallRequiredSystemLibraries)
 set(DIRS ${Boost_LIBRARY_DIRS} ${CMAKE_CURRENT_BINARY_DIR}/external/squirrel ${CMAKE_CURRENT_BINARY_DIR}/external/tinygettext)
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT DISABLE_CPACK_BUNDLING)
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND ENABLE_APP_BUNDLING AND NOT DISABLE_CPACK_BUNDLING)
   set(INFOPLIST_CFBUNDLEEXECUTABLE "SuperTux")
 
   find_package(png)
