--- configure.orig	2012-04-16 05:14:10.000000000 +0900
+++ configure	2012-04-16 05:15:04.000000000 +0900
@@ -5573,7 +5573,7 @@
      if test -z "$GCCUPC_VERSION" ; then # auto-detect version by running the compiler
        { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gccupc version" >&5
 $as_echo_n "checking for gccupc version... " >&6; }
-       if test -z "`$TRANSLATOR -v 2>&1 < /dev/null | grep 'GCC UPC'`" ; then
+       if test -z "`$TRANSLATOR -v 2>&1 < /dev/null | grep 'GNU UPC'`" ; then
          as_fn_error $? "'--with-gccupc' requires pathname to 'upc' or 'xupc' executable" "$LINENO" 5
        fi
        # Old GCC UPC version string:
@@ -5583,7 +5583,7 @@
        # Above, in a production compiler, xgcc is gcc,
        # and dev (with optional parens) is not present.
        GCCUPC_VERSION=`$TRANSLATOR -v 2>&1 | \
-         $PERL -ne 'if (/upc-(\S+) .GCC UPC/ or /\(GCC UPC (\S+)?\)/) {print $1;}'`
+         $PERL -ne 'if (/upc-(\S+) .GNU UPC/ or /\(GNU UPC (\S+)?\)/) {print $1;}'`
      fi
      GCCUPC_VERSION_MAJOR=`echo "$GCCUPC_VERSION" | \
          $PERL -ne 'if (m/(\d+)\.\d+\.\d+[.-]\d+/) {print $1;}'`
