--- src/procproperties.cpp.orig	2013-04-05 13:20:20.000000000 -0700
+++ src/procproperties.cpp	2013-09-19 14:01:51.000000000 -0700
@@ -29,6 +29,9 @@
 #elif defined (__OpenBSD__)
 #include <sys/param.h>
 #include <sys/sysctl.h>
+#elif defined (__APPLE__)
+#include <sys/param.h>
+#include <sys/sysctl.h>
 #endif
 
 #include "procman-app.h"
@@ -124,6 +127,8 @@
         HZ = 100;
     else
         HZ = (cinf.stathz ? cinf.stathz : cinf.hz);
+#elif defined (__APPLE__)
+    int HZ = 100;
 #endif
     proc_arg proc_props[] = {
         { N_("Process Name"), g_strdup_printf("%s", info->name)},
