--- Makefile.in.orig	2017-10-09 15:11:34.000000000 +0200
+++ Makefile.in	2017-10-09 15:13:50.000000000 +0200
@@ -28,61 +28,12 @@
 	$(CC) -o tinyows $(SRC) $(XMLFLAGS) $(CFLAGS) $(PGFLAGS)  $(FCGIFLAGS) $(GIT_FLAGS) -lfl
 	@rm -rf tinyows.dSYM
 
-flex:
-	lex -i -osrc/mapfile/mapfile.c src/mapfile/mapfile.l
-
-svn-clean: clean doc-clean 
-	@find . -name '*~' -exec rm {} \;	
-	@rm -f configure
-
-clean: 
-	@rm -f tinyows Makefile src/ows_define.h
-	@rm -rf tinyows.dSYM
-	@rm -f demo/tinyows.xml demo/install.sh
-	@rm -f test/tinyows.xml test/install.sh
-
 install:
 	@echo "-----"
 	@echo "TinyOWS Schema install dir in $(PREFIX)/share/tinyows/schema"
-	mkdir -p $(PREFIX)/share/tinyows
-	rm -rf $(PREFIX)/share/tinyows/schema
-	cp -rf schema $(PREFIX)/share/tinyows/
-	@echo "-----"
-	@echo "Now:"
-	@echo " 1) copy 'tinyows' binary to cgi-bin directory"
-	@echo " 2) - put a workable tinyows.xml file in /etc dir"
-	@echo "    - OR launch 'make install-demo' as a superuser"
-
-install-demo:
-	@chmod +x demo/install.sh
-	@demo/install.sh
-	cp -i demo/tinyows.xml /etc/tinyows.xml
-
-install-test100:
-	@chmod +x test/wfs_100/install_wfs_100.sh
-	@test/wfs_100/install_wfs_100.sh
-	cp -i test/wfs_100/config_wfs_100.xml /etc/tinyows.xml
-
-install-test110:
-	@chmod +x test/wfs_110/install_wfs_110.sh
-	@test/wfs_110/install_wfs_110.sh
-	cp -i test/wfs_110/config_wfs_110.xml /etc/tinyows.xml
-
-doc-clean:
-	@rm -rf doc/doxygen
-
-doxygen: doc-clean
-	@(which doxygen 2> /dev/null > /dev/null	\
-	&& mkdir -p doc/doxygen				\
-	&& doxygen doc/Doxyfile				\
-	) || echo "doxygen seems not installed"
-
-test-valgrind110:
-	@test/unit_test test/wfs_110/cite 1
-
-test-valgrind100:
-	@test/unit_test test/wfs_100/cite 1
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	cp tinyows $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/share/tinyows
+	cp -rf schema $(DESTDIR)$(PREFIX)/share/tinyows/
+	cp -i demo/tinyows.xml $(DESTDIR)$(PREFIX)/etc/tinyows.xml
 
-astyle:
-	astyle --style=k/r --indent=spaces=2 -c --lineend=linux -S $(SRC) src/*.h*
-	rm -f src/*.orig src/*/*.orig
