--- Makefile.orig	2023-07-09 10:10:38
+++ Makefile	2023-07-09 10:11:51
@@ -21,9 +21,9 @@
 # liking.  The commented-out example for USRDIR installs files in
 # subdirectories under /usr/local instead.
 #
-USRDIR = usr
-# USRDIR = usr/local
-PREFIX = $(DESTDIR)/$(USRDIR)
+USRDIR ?= /usr
+# USRDIR = /usr/local
+PREFIX = $(DESTDIR)$(USRDIR)
 PKGDEST = $(PREFIX)/share/unifont
 
 HEXFILES = hangul-syllables.hex plane00-nonprinting.hex pua.hex spaces.hex \
