--- configure.orig	2024-10-22 21:05:41.000000000 +0800
+++ configure	2024-11-17 15:35:57.000000000 +0800
@@ -19123,15 +19123,11 @@
 printf "%s\n" "disabled" >&6; }
 else
     CFLAGS="$CFLAGS -O -std=c99 -pedantic -pedantic-errors -W -Wall -Wextra"
-    CFLAGS="$CFLAGS -fno-inline -Wnull-dereference -Wdouble-promotion"
+    CFLAGS="$CFLAGS -fno-inline"
     CFLAGS="$CFLAGS -Wformat=2 -Wformat-security -Wformat-nonliteral"
     CFLAGS="$CFLAGS -Wconversion -Wunused -Wshadow -Wpointer-arith -Wcast-align"
     CFLAGS="$CFLAGS -Wwrite-strings -Wmissing-prototypes"
     CFLAGS="$CFLAGS -Wnested-externs -Wredundant-decls"
-
-    if test "z$build_with_gcc" = "zyes" ; then
-        CFLAGS="$CFLAGS -Wformat-overflow=2 -Wformat-signedness"
-    fi
     { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
 fi
