=== modified file 'src/extension/effect.cpp'
--- src/extension/effect.cpp	2010-09-28 21:11:41 +0000
+++ src/extension/effect.cpp	2013-11-19 23:57:37 +0000
@@ -36,7 +36,7 @@
 Effect::Effect (Inkscape::XML::Node * in_repr, Implementation::Implementation * in_imp)
     : Extension(in_repr, in_imp),
       _id_noprefs(Glib::ustring(get_id()) + ".noprefs"),
-      _name_noprefs(Glib::ustring(get_name()) + _(" (No preferences)")),
+      _name_noprefs(Glib::ustring(_(get_name())) + _(" (No preferences)")),
       _verb(get_id(), get_name(), NULL, NULL, this, true),
       _verb_nopref(_id_noprefs.c_str(), _name_noprefs.c_str(), NULL, NULL, this, false),
       _menu_node(NULL), _workingDialog(true),

=== modified file 'src/extension/system.cpp'
--- src/extension/system.cpp	2013-09-21 22:20:09 +0000
+++ src/extension/system.cpp	2013-11-19 23:57:37 +0000
@@ -20,6 +20,7 @@
 
 #include <interface.h>
 
+#include <unistd.h>
 #include "system.h"
 #include "preferences.h"
 #include "extension.h"

