--- src/helper/png-write.cpp.orig	2012-09-04 16:04:30.000000000 -0400
+++ src/helper/png-write.cpp	2012-09-04 16:04:30.000000000 -0400
@@ -165,7 +165,7 @@
     /* Set error handling.  REQUIRED if you aren't supplying your own
      * error hadnling functions in the png_create_write_struct() call.
      */
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
         /* If we get here, we had a problem reading the file */
         fclose(fp);
         png_destroy_write_struct(&png_ptr, &info_ptr);
--- src/sp-image.cpp.orig	2012-09-04 16:04:30.000000000 -0400
+++ src/sp-image.cpp	2012-09-04 16:04:30.000000000 -0400
@@ -396,7 +396,7 @@
                 {
                     char* name = 0;
                     int compression_type = 0;
-                    char* profile = 0;
+                    png_bytep profile = 0;
                     png_uint_32 proflen = 0;
                     if ( png_get_iCCP(pngPtr, infoPtr, &name, &compression_type, &profile, &proflen) ) {
 //                                         g_message("Found an iCCP chunk named [%s] with %d bytes and comp %d", name, proflen, compression_type);
--- src/extension/internal/pdfinput/svg-builder.cpp.orig	2012-09-04 16:04:30.000000000 -0400
+++ src/extension/internal/pdfinput/svg-builder.cpp	2012-09-04 16:04:30.000000000 -0400
@@ -1443,7 +1443,7 @@
         return NULL;
     }
     // Set error handler
-    if (setjmp(png_ptr->jmpbuf)) {
+    if (setjmp(png_jmpbuf(png_ptr))) {
         png_destroy_write_struct(&png_ptr, &info_ptr);
         return NULL;
     }
