=== modified file 'src/color-profile.cpp'
--- src/color-profile.cpp	2014-03-18 11:45:34 +0000
+++ src/color-profile.cpp	2014-10-30 18:35:32 +0000
@@ -13,6 +13,7 @@
 #include <gtk/gtk.h>
 #endif // DEBUG_LCMS
 
+#include <unistd.h>
 #include <cstring>
 #include <string>
 #include <io/sys.h>

=== modified file 'src/io/inkjar.cpp'
--- src/io/inkjar.cpp	2013-11-12 20:52:11 +0000
+++ src/io/inkjar.cpp	2014-09-12 15:05:40 +0000
@@ -107,7 +107,7 @@
 
 bool JarFile::open()
 {
-    if ((fd = fopen(_filename, O_RDONLY)) < 0) {
+    if (!(fd = fopen(_filename, "r"))) {
 	fprintf(stderr, "open failed.\n");
 	return false;
     }
@@ -119,7 +119,7 @@
 
 bool JarFile::close()
 {
-    if (fd >= 0 && !fclose(fd)) {
+    if (fd && !fclose(fd)) {
 	inflateEnd(&_zs);
 	return true;
     }

=== modified file 'src/object-snapper.cpp'
--- src/object-snapper.cpp	2010-07-19 06:51:04 +0000
+++ src/object-snapper.cpp	2014-09-12 15:05:40 +0000
@@ -561,7 +561,7 @@
                         // When it's within snapping range, then return it
                         // (within snapping range == between p_min_on_cl and p_max_on_cl == 0 < ta < 1)
                         Geom::Coord dist = Geom::L2(_snapmanager->getDesktop()->dt2doc(p_proj_on_cl) - p_inters);
-                        SnappedPoint s(_snapmanager->getDesktop()->doc2dt(p_inters), p.getSourceType(), p.getSourceNum(), k->target_type, dist, getSnapperTolerance(), getSnapperAlwaysSnap(), true, k->target_bbox);
+                        SnappedPoint s(_snapmanager->getDesktop()->doc2dt(p_inters), p.getSourceType(), p.getSourceNum(), k->target_type, dist, getSnapperTolerance(), getSnapperAlwaysSnap(), true, false, k->target_bbox);
                         sc.points.push_back(s);
                     }
                 }

