--- configure.ac.orig	2013-11-13 13:32:49.000000000 -0800
+++ configure.ac	2013-11-13 13:32:50.000000000 -0800
@@ -434,11 +434,11 @@
             [with_python=$withval], [with_python=skipped])
 
 if test "x$with_python" = "xyes"; then
-    checkPYTHON_CFLAGS=`python -c "import distutils.sysconfig ; print '-I%s' % distutils.sysconfig.get_config_var('INCLUDEPY')" 2>/dev/null`
+    checkPYTHON_CFLAGS=`${prefix}/bin/python@@MP_PYTHON_VERSION@@ -c "import distutils.sysconfig ; print '-I%s' % distutils.sysconfig.get_config_var('INCLUDEPY')" 2>/dev/null`
     if test "$?" -gt "0"; then
         with_python="no"
     else
-        checkPYTHON_LIBS=`python -c "import distutils.sysconfig ; print '%s/%s %s' % (distutils.sysconfig.get_config_var('LIBPL'),distutils.sysconfig.get_config_var('LDLIBRARY'),distutils.sysconfig.get_config_var('LIBS'))" 2>/dev/null`
+        checkPYTHON_LIBS=-lpython@@MP_PYTHON_VERSION@@
         if test "$?" -gt "0"; then
             with_python="no"
         else
