--- Defaults.in.orig	Sat Oct  4 13:21:01 2003
+++ Defaults.in	Sat Oct  4 13:23:16 2003
@@ -96,10 +96,10 @@
 };
 
 if ($enable_bootstrap && $bootstrap_compiler eq "d2c") {
-    $CPPFLAGS = '-I@LOCAL_RUNTIME@';
+    $CPPFLAGS = '-I@LOCAL_RUNTIME@ @CPPFLAGS@';
     $d2c_runtime = '-L@LOCAL_RUNTIME@ ';
 } else {
-    $CPPFLAGS = '-I'.$srcroot.'/d2c/runtime';
+    $CPPFLAGS = '-I'.$srcroot.'/d2c/runtime @CPPFLAGS@';
     @runtime_directories = ('c-code', 'dylan', 'melange');
     foreach (@runtime_directories) {
 	$d2c_runtime .= "-L$buildroot/d2c/runtime/$_ ";
