From 9d03794e9e610520d903d50a104bd7168c67a6dd Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 4 Dec 2022 15:27:33 +0800
Subject: [PATCH] api-macos.c: fix missing stdbool

---
 src/api-macos.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/api-macos.c b/src/api-macos.c
index a893ef9..93c7bbb 100644
--- a/src/api-macos.c
+++ b/src/api-macos.c
@@ -21,6 +21,8 @@
 #include "ps-internal.h"
 #include "arch/macos/process_info.h"
 
+#include <stdbool.h>
+
 #define PS__TV2DOUBLE(t) ((t).tv_sec + (t).tv_usec / 1000000.0)
 
 #define PS__CHECK_KINFO(kp, handle)				      \
-- 
2.38.1

