From 7387cb6f3e5f958af99eaf51a14a892a8fb1bf52 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 21 Mar 2023 22:33:49 +0700
Subject: [PATCH] api-macos.c: restore PPC support

---
 src/api-macos.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/api-macos.c b/src/api-macos.c
index 93c7bbb..f862589 100644
--- a/src/api-macos.c
+++ b/src/api-macos.c
@@ -839,6 +839,14 @@ static bool ps_in_shared_region(mach_vm_address_t addr, cpu_type_t type) {
     base = SHARED_REGION_BASE_X86_64;
     size = SHARED_REGION_SIZE_X86_64;
     break;
+  case CPU_TYPE_POWERPC:
+    base = SHARED_REGION_BASE_PPC;
+    size = SHARED_REGION_SIZE_PPC;
+    break;
+  case CPU_TYPE_POWERPC64:
+    base = SHARED_REGION_BASE_PPC64;
+    size = SHARED_REGION_SIZE_PPC64;
+    break;
   default:
     return false;
   }
-- 
2.39.1

