--- contrib/torify.in	2009-09-21 05:33:32.000000000 +0000
+++ contrib/torify.in.new	2009-11-21 12:06:12.000000000 +0000
@@ -50,7 +50,7 @@
 	! [ "$verbose" -ge 1 ] || echo "Using tsocks as socksifier." >&2
 
 	# Define our tsocks config file
-	TSOCKS_CONF_FILE="/etc/tor/tor-tsocks.conf"
+	TSOCKS_CONF_FILE="@CONFDIR@/tor-tsocks.conf"
 	export TSOCKS_CONF_FILE
 
 	# Check that we've got a tsocks config file
