--- rtree/core.py.orig	2009-08-04 21:42:57.000000000 +0400
+++ rtree/core.py	2009-12-26 11:07:14.000000000 +0300
@@ -92,9 +92,15 @@
     lib_name = 'libsidx.so'
     if platform == 'Darwin':
         free = ctypes.CDLL(find_library('libc')).free
+        lib_name_darwin = os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', lib_name))
+        if os.path.exists(lib_name_darwin):
+            try:
+                rt = ctypes.CDLL(lib_name_darwin)
+            except OSError:
+                rt = ctypes.CDLL(lib_name)
     else:        
         free = ctypes.CDLL(find_library('libc.so.6')).free
-    rt = ctypes.CDLL(lib_name)
+        rt = ctypes.CDLL(lib_name)
 else:
     raise RTreeError('Unsupported OS "%s"' % os.name)
 
