Use MacPorts CPPFLAGS and LDFLAGS.
--- Makefile.in.orig	2006-02-13 18:38:23.000000000 -0600
+++ Makefile.in	2024-10-05 02:07:37.000000000 -0500
@@ -9,6 +9,8 @@
 LIBS=@LIBS@
 CC=@CC@
 CFLAGS=@CFLAGS@
+CPPFLAGS=@CPPFLAGS@
+LDFLAGS=@LDFLAGS@
 
 INSTALLCMD=@INSTALL@
 
@@ -24,7 +26,7 @@
 
 # note that the -I. is needed to handle config.h when using VPATH
 .c.o:
-	$(CC) -I. -I$(srcdir) $(CFLAGS) -c $< -o $@
+	$(CC) -I. -I$(srcdir) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
 
 all: rzip
 
@@ -37,7 +39,7 @@
 	${INSTALLCMD} -m 644 $(srcdir)/rzip.1 ${INSTALL_MAN}/man1/
 
 rzip: $(OBJS)
-	$(CC) $(CFLAGS) -o rzip $(OBJS) $(LIBS)
+	$(CC) -o rzip $(OBJS) $(LDFLAGS) $(LIBS)
 
 rzip.1: rzip.yo
 	yodl2man -o rzip.1 rzip.yo
