--- devhelp/meson.build.orig	2020-11-20 05:48:48.000000000 -0800
+++ devhelp/meson.build	2022-07-18 17:06:59.000000000 -0700
@@ -96,7 +96,7 @@
 libdevhelp_shared_lib = shared_library(
   'devhelp-@0@'.format(LIBDEVHELP_API_VERSION),
   link_whole: libdevhelp_static_lib,
-  link_args: '-Wl,--version-script,' + symbol_map,
+  link_args: '-export-symbols ' + symbol_map,
   link_depends: symbol_map,
   version: LIBDEVHELP_LT_VERSION,
   install: true
