--- Makefile.orig	2022-05-27 15:45:34.000000000 -0400
+++ Makefile	2023-04-29 08:03:15.037784865 -0400
@@ -26,7 +26,7 @@
 EXDIR=${prefix}/bin

 # This is where the man page goes.
-MANDIR=${prefix}/man/man1
+MANDIR=${prefix}/share/man/man1
 MANEXT=1
 MANMODE=644

@@ -499,32 +499,32 @@
 	 $(MANDIR)/p$(name).$(MANEXT)

 $(EXDIR)/$(name): $(name)
-	cp $(name) $(EXDIR)
-	strip $(EXDIR)/$(name)
+	cp $(name) $(DESTDIR)$(EXDIR)/
+	strip $(DESTDIR)$(EXDIR)/$(name)

 $(EXDIR)/$(name)qref: $(name)qref
-	cp $(name)qref $(EXDIR)
-	strip $(EXDIR)/$(name)qref
+	cp $(name)qref $(DESTDIR)$(EXDIR)/
+	strip $(DESTDIR)$(EXDIR)/$(name)qref

 $(EXDIR)/p$(name): p$(name)
-	cp p$(name) $(EXDIR)
-	strip $(EXDIR)/p$(name)
+	cp p$(name) $(DESTDIR)$(EXDIR)/
+	strip $(DESTDIR)$(EXDIR)/p$(name)

 $(LIBDIR)/tutorial: tutorial.sc $(LIBDIR)
-	-mkdir -p $(LIBDIR)/plugins
-	cp tutorial.sc $(LIBDIR)/tutorial.$(name)
-	chmod $(MANMODE) $(LIBDIR)/tutorial.$(name)
+	-mkdir -p $(DESTDIR)$(LIBDIR)/plugins
+	cp tutorial.sc $(DESTDIR)$(LIBDIR)/tutorial.$(name)
+	chmod $(MANMODE) $(DESTDIR)$(LIBDIR)/tutorial.$(name)

 $(LIBDIR):
-	mkdir $(LIBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)

 $(MANDIR)/$(name).$(MANEXT): $(name).1
-	cp $(name).1 $(MANDIR)/$(name).$(MANEXT)
-	chmod $(MANMODE) $(MANDIR)/$(name).$(MANEXT)
+	cp $(name).1 $(DESTDIR)$(MANDIR)/$(name).$(MANEXT)
+	chmod $(MANMODE) $(DESTDIR)$(MANDIR)/$(name).$(MANEXT)

 $(MANDIR)/p$(name).$(MANEXT): p$(name).1
-	cp p$(name).1 $(MANDIR)/p$(name).$(MANEXT)
-	chmod $(MANMODE) $(MANDIR)/p$(name).$(MANEXT)
+	cp p$(name).1 $(DESTDIR)$(MANDIR)/p$(name).$(MANEXT)
+	chmod $(MANMODE) $(DESTDIR)$(MANDIR)/p$(name).$(MANEXT)

 uninstall:
 	rm -f $(EXDIR)/$(name)
