--- libkdegames/kggzmod/misc_private.h~	2009-01-21 06:17:49.000000000 -0500
+++ libkdegames/kggzmod/misc_private.h	2009-02-03 18:58:09.000000000 -0500
@@ -36,7 +36,7 @@
 #endif
 
 // work around a bug in the Mac OS X 10.4.0 SDK
-#if defined(__APPLE__)
+/* #if defined(__APPLE__)
 # ifndef __DARWIN_ALIGNBYTES
 #  define __DARWIN_ALIGNBYTES     (sizeof(__darwin_size_t) - 1)
 # endif
@@ -45,7 +45,7 @@
 # endif
 # undef CMSG_SPACE
 # define CMSG_SPACE(l) (ALIGN(sizeof(struct cmsghdr)) + ALIGN(l))
-#endif
+#endif */
 
 namespace KGGZMod
 {
