--- work/lemon-1.39/lemon.c	2006-07-20 08:09:51.000000000 +0200
+++ lemon.c	2008-02-05 09:49:51.000000000 +0100
@@ -2994,7 +2994,7 @@
 PRIVATE FILE *tplt_open(lemp)
 struct lemon *lemp;
 {
-  static char templatename[] = "lempar.c";
+  static char templatename[] = LEMPAR;
   char buf[1000];
   FILE *in;
   char *tpltname;
@@ -3466,7 +3466,7 @@
 
   in = tplt_open(lemp);
   if( in==0 ) return;
-  out = file_open(lemp,".c","wb");
+  out = file_open(lemp,CSUFFIX,"wb");
   if( out==0 ){
     fclose(in);
     return;
@@ -3877,7 +3877,7 @@
 
   if( lemp->tokenprefix ) prefix = lemp->tokenprefix;
   else                    prefix = "";
-  in = file_open(lemp,".h","rb");
+  in = file_open(lemp,HSUFFIX,"rb");
   if( in ){
     for(i=1; i<lemp->nterminal && fgets(line,LINESIZE,in); i++){
       sprintf(pattern,"#define %s%-30s %2d\n",prefix,lemp->symbols[i]->name,i);
@@ -3889,7 +3889,7 @@
       return;
     }
   }
-  out = file_open(lemp,".h","wb");
+  out = file_open(lemp,HSUFFIX,"wb");
   if( out ){
     for(i=1; i<lemp->nterminal; i++){
       fprintf(out,"#define %s%-30s %2d\n",prefix,lemp->symbols[i]->name,i);
