--- lib/error.c.orig	2014-05-02 13:26:30.000000000 -0400
+++ lib/error.c	2014-05-02 13:27:28.000000000 -0400
@@ -377,8 +377,10 @@
   __fxprintf (NULL, file_name != NULL ? "%s:%d: " : " ",
               file_name, line_number);
 #else
-  fprintf (stderr, file_name != NULL ? "%s:%d: " : " ",
-           file_name, line_number);
+  if (file_name != NULL)
+    fprintf (stderr, "%s:%d: ", file_name, line_number);
+  else
+    fprintf (stderr, " ");
 #endif
 
   va_start (args, message);
