--- lib/error.c.orig	2017-03-13 08:34:51.000000000 +0000
+++ lib/error.c	2017-03-13 08:38:18.000000000 +0000
@@ -382,8 +382,10 @@
   __fxprintf (NULL, file_name != NULL ? "%s:%u: " : " ",
               file_name, line_number);
 #else
-  fprintf (stderr, file_name != NULL ? "%s:%u: " : " ",
-           file_name, line_number);
+  if (file_name != NULL)
+    fprintf (stderr, "%s:%u: ", file_name, line_number);
+  else
+    fprintf (stderr, " ");
 #endif
 
   va_start (args, message);
