--- common/llist.c.orig	2012-05-03 19:14:50.000000000 -0400
+++ common/llist.c	2014-05-02 13:15:38.000000000 -0400
@@ -281,7 +281,7 @@
 {
 	LNode *entry;
 
-	if (index < 0 || index >= list->size)
+	if (index >= list->size)
 		return NULL;
 
 	entry = llist_get_entry(list, index);
@@ -294,7 +294,7 @@
 	LNode *entry;
 	void *old_data;
 
-	if (index < 0 || index >= list->size)
+	if (index >= list->size)
 		return NULL;
 
 	entry = llist_get_entry(list, index);
@@ -316,7 +316,7 @@
 {
 	LNode *entry;
 	
-	if (index < 0 || index > list->size)
+	if (index > list->size)
 		return;
 
 	entry = xmalloc(sizeof(LNode));
@@ -343,7 +343,7 @@
 	LNode *entry;
 	void *data;
 
-	if (index < 0 || index >= list->size)
+	if (index >= list->size)
 		return NULL;
 
 	entry = llist_get_entry(list, index);
