--- sip/iqt5qt4/IQtModule.sip.orig
+++ sip/iqt5qt4/IQtModule.sip
@@ -29,7 +29,7 @@
 // PyQwt becomes a free plug-in for a non-free program.
 
 
-%Module PyQt4.Qwt5._iqt 0
+%Module(name=PyQt4.Qwt5._iqt)
 
 %ModuleCode
 #include <qcoreapplication.h>
--- sip/qwt5qt4/QwtModule.sip.orig
+++ sip/qwt5qt4/QwtModule.sip
@@ -29,7 +29,7 @@
 // PyQwt becomes a free plug-in for a non-free program.
 
 
-%Module PyQt4.Qwt5.Qwt 0
+%Module(name=PyQt4.Qwt5.Qwt)
 
 %Timeline {Qwt_5_0_0 Qwt_5_0_1 Qwt_5_0_3 Qwt_5_1_0 Qwt_5_1_1 Qwt_5_2_0}
 
@@ -49,7 +49,7 @@
 %End // HAS_QWT_SVG
 
 
-%Include QwtTypes.sip
+ // %Include QwtTypes.sip
 
 %ModuleHeaderCode
 #if PY_VERSION_HEX < 0x02050000 && !defined(PY_SSIZE_T_MIN)
