From f87a09a7b12c8c6eeaf74df7ecefd9d6c81763cf Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sun, 18 Jan 2015 02:04:36 -0800
Subject: [PATCH 1010/1012] Fix dylib install name when building on Tiger

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 tools/libclang/Makefile | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git llvm_release_34/tools/clang/tools/libclang/Makefile macports_release_34/tools/clang/tools/libclang/Makefile
index 43ecbd1..971cc43 100644
--- llvm_release_34/tools/clang/tools/libclang/Makefile
+++ macports_release_34/tools/clang/tools/libclang/Makefile
@@ -49,12 +49,8 @@ ifeq ($(HOST_OS),Darwin)
     # Extra options to override libtool defaults.
     LLVMLibsOptions += -Wl,-dead_strip
 
-    # Mac OS X 10.4 and earlier tools do not allow a second -install_name on command line
-    DARWIN_VERS := $(shell echo $(TARGET_TRIPLE) | sed 's/.*darwin\([0-9]*\).*/\1/')
-    ifneq ($(DARWIN_VERS),8)
-       LLVMLibsOptions += -Wl,-install_name \
-                          -Wl,"@rpath/lib$(LIBRARYNAME)$(SHLIBEXT)"
-    endif
+    # $(CLANG_LEVEL)/../../Makefile.rules has been patched to set the
+    # install name for all dylibs.
 
     # If we're doing an Apple-style build, add the LTO object path.
     ifeq ($(RC_XBS),YES)
-- 
2.10.1

