--- a/Makefile	2014-05-29 08:40:41.000000000 -0500
+++ b/Makefile	2014-05-29 08:41:52.000000000 -0500
@@ -72,7 +72,9 @@
           tools/clang/tools/c-index-test \
           tools/clang/runtime tools/clang/docs \
           tools/clang/include tools/clang/lib \
-          tools/clang/tools/clang-format
+          tools/clang/tools/clang-format \
+          tools/clang/tools/extra/clang-modernize \
+          tools/clang/tools/extra/clang-apply-replacements
   OPTIONAL_DIRS :=
   NO_INSTALL = 1
 endif
