Index: llvm-3.4.2.src/projects/compiler-rt/make/platform/clang_darwin.mk
===================================================================
--- llvm-3.4.2.src.orig/projects/compiler-rt/make/platform/clang_darwin.mk
+++ llvm-3.4.2.src/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -85,8 +85,8 @@ Configs += profile_ios
 UniversalArchs.profile_ios := $(call CheckArches,i386 x86_64 armv7,profile_ios)
 
 # Configurations which define the ASAN support functions.
-Configs += asan_osx_dynamic
-UniversalArchs.asan_osx_dynamic := $(call CheckArches,i386 x86_64,asan_osx_dynamic)
+#Configs += asan_osx_dynamic
+#UniversalArchs.asan_osx_dynamic := $(call CheckArches,i386 x86_64,asan_osx_dynamic)
 
 IOSSIM_SDK_PATH := $(call XCRunSdkPath,iphonesimulator)
 ifneq ($(IOSSIM_SDK_PATH),)
@@ -94,8 +94,8 @@ Configs += asan_iossim_dynamic
 UniversalArchs.asan_iossim_dynamic := $(call CheckArches,i386 x86_64,asan_iossim_dynamic)
 endif
 
-Configs += ubsan_osx
-UniversalArchs.ubsan_osx := $(call CheckArches,i386 x86_64,ubsan_osx)
+#Configs += ubsan_osx
+#UniversalArchs.ubsan_osx := $(call CheckArches,i386 x86_64,ubsan_osx)
 
 # Darwin 10.6 has a bug in cctools that makes it unable to use ranlib on our ARM
 # object files. If we are on that platform, strip out all ARM archs. We still
Index: llvm-3.4.2.src/tools/clang/runtime/compiler-rt/Makefile
===================================================================
--- llvm-3.4.2.src.orig/tools/clang/runtime/compiler-rt/Makefile
+++ llvm-3.4.2.src/tools/clang/runtime/compiler-rt/Makefile
@@ -82,9 +82,7 @@ ifeq ($(OS),Darwin)
 RuntimeDirs += darwin darwin_embedded
 RuntimeLibrary.darwin.Configs := \
 	eprintf.a 10.4.a osx.a ios.a cc_kext.a cc_kext_ios5.a \
-	asan_osx_dynamic.dylib \
-	profile_osx.a profile_ios.a \
-	ubsan_osx.a
+	profile_osx.a profile_ios.a
 RuntimeLibrary.darwin_embedded.Configs := \
 	soft_static.a hard_static.a soft_pic.a hard_pic.a
 
