Index: llvm-3.3/projects/compiler-rt/make/platform/clang_darwin.mk
===================================================================
--- llvm-3.3/projects/compiler-rt/make/platform/clang_darwin.mk	(revision 185755)
+++ llvm-3.3/projects/compiler-rt/make/platform/clang_darwin.mk	(working copy)
@@ -70,12 +70,12 @@
 UniversalArchs.profile_ios := $(call CheckArches,i386 x86_64 armv7,profile_ios)
 
 # Configurations which define the ASAN support functions.
-Configs += asan_osx
-UniversalArchs.asan_osx := $(call CheckArches,i386 x86_64,asan_osx)
+#Configs += asan_osx
+#UniversalArchs.asan_osx := $(call CheckArches,i386 x86_64,asan_osx)
+#
+#Configs += asan_osx_dynamic
+#UniversalArchs.asan_osx_dynamic := $(call CheckArches,i386 x86_64,asan_osx_dynamic)
 
-Configs += asan_osx_dynamic
-UniversalArchs.asan_osx_dynamic := $(call CheckArches,i386 x86_64,asan_osx_dynamic)
-
 Configs += ubsan_osx
 UniversalArchs.ubsan_osx := $(call CheckArches,i386 x86_64,ubsan_osx)
 
Index: llvm-3.3/tools/clang/runtime/compiler-rt/Makefile
===================================================================
--- llvm-3.3/tools/clang/runtime/compiler-rt/Makefile	(revision 185755)
+++ llvm-3.3/tools/clang/runtime/compiler-rt/Makefile	(working copy)
@@ -75,7 +75,6 @@
 RuntimeDirs += darwin
 RuntimeLibrary.darwin.Configs := \
 	eprintf.a 10.4.a osx.a ios.a cc_kext.a cc_kext_ios5.a \
-	asan_osx.a asan_osx_dynamic.dylib \
 	profile_osx.a profile_ios.a \
 	ubsan_osx.a
 endif
