--- DESCRIPTION	2024-04-18 06:10:02.000000000 +0800
+++ DESCRIPTION	2024-04-19 16:01:25.000000000 +0800
@@ -27,8 +27,8 @@
 Depends: R (>= 2.14.0)
 Imports: ca, cluster, colorspace, foreach, gclus, grDevices, grid,
         MASS, qap, registry, stats, TSP, vegan
-Suggests: dbscan, DendSer, dendextend, doParallel, GA, ggplot2, keras,
-        Rtsne, scales, smacof, tensorflow, testthat, umap
+Suggests: dbscan, DendSer, dendextend, doParallel, GA, ggplot2,
+        Rtsne, scales, smacof, testthat, umap
 Encoding: UTF-8
 RoxygenNote: 7.3.1
 License: GPL-3
