--- src/build.mk.orig	2005-03-27 10:34:32.000000000 +1000
+++ src/build.mk	2012-05-09 15:41:08.000000000 +1000
@@ -11,7 +11,7 @@
 OBJS		= $(UPKG).o $(UPKG)/*.o
 
 HC_OPTS		= -package-name $(PKG)
-HC_OPTS		+= -O -Wall -Werror -fno-warn-missing-signatures $(GHC_EXTRA_OPTS)
+HC_OPTS		+= -O -Wall -fno-warn-missing-signatures $(GHC_EXTRA_OPTS)
 
 CLEANS 		+= $(LIBRARY) $(GHCI_LIBRARY)
 CLEAN_FILES     += *.conf.inplace* *.conf.*.old *.conf.in *.h *.in
