--- XXAutoConnectBag.m.orig	2012-11-15 22:16:19.000000000 +0100
+++ XXAutoConnectBag.m	2012-11-15 22:19:02.000000000 +0100
@@ -82,13 +82,13 @@
         return;
     
     // Move everything up
-    for (i = index; i < 5; i++)
+    for (i = index; i < 4; i++)
     {
         names[i] = names[i+1];
         positions[i] = positions[i+1];
         types[i] = types[i+1];
     }
-    names[5] = nil;
+    names[4] = nil;
     
     size--;
 }
