--- XXView.m.orig	2008-10-11 19:18:55.000000000 +1100
+++ XXView.m	2018-09-01 04:57:06.000000000 +1000
@@ -37,6 +37,10 @@
 #import "XXAbstractRemote.h"
 #import <Carbon/Carbon.h>
 #import "XXController.h"
+#import <AvailabilityMacros.h>
+#if defined(MAC_OS_X_VERSION_MAX_ALLOWED) && defined(MAC_OS_X_VERSION_10_9) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_9
+typedef int32_t CGMouseDelta;
+#endif
 
 @implementation XXView
 
