--- conf/zabbix_agentd.conf.orig	2016-02-16 14:03:55.000000000 -0600
+++ conf/zabbix_agentd.conf	2016-02-16 14:08:16.000000000 -0600
@@ -10,6 +10,10 @@
 # Default:
 # PidFile=/tmp/zabbix_agentd.pid
 
+# MACPORTS CONFIG. PLEASE ONLY EDIT IF YOU KNOW WHAT YOU ARE DOING
+PidFile=%%PREFIX%%/var/run/zabbix/zabbix_agentd.pid
+# END MACPORTS CONFIG
+
 ### Option: LogType
 #	Specifies where log messages are written to:
 #		system  - syslog
@@ -27,7 +31,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agentd.log
+LogFile=%%PREFIX%%/var/log/zabbix/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
--- conf/zabbix_server.conf.orig	2016-02-16 14:04:00.000000000 -0600
+++ conf/zabbix_server.conf	2016-02-16 14:08:34.000000000 -0600
@@ -35,7 +35,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_server.log
+LogFile=%%PREFIX%%/var/log/zabbix/zabbix_server.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -67,6 +67,10 @@
 # Default:
 # PidFile=/tmp/zabbix_server.pid
 
+# MACPORTS CONFIG. PLEASE ONLY EDIT IF YOU KNOW WHAT YOU ARE DOING
+PidFile=%%PREFIX%%/var/run/zabbix/zabbix_server.pid
+# END MACPORTS CONFIG
+ 
 ### Option: SocketDir
 #	IPC socket directory.
 #       Directory to store IPC sockets used by internal Zabbix services.
