--- src/configure.in.old	2008-12-07 02:58:32.000000000 +0100
+++ src/configure.in	2008-12-07 02:58:58.000000000 +0100
@@ -949,7 +949,11 @@
 AC_MSG_RESULT($enable_python3interp)
 if test "$enable_python3interp" = "yes" -o "$enable_python3interp" = "dynamic"; then
   dnl -- find the python3 executable
-  AC_PATH_PROG(vi_cv_path_python3, python3)
+  AC_MSG_CHECKING(--with-python argument)
+  AC_ARG_WITH(python, [  --with-python=PATH       which python to use (default: python)],
+        python_name="$withval"; AC_MSG_RESULT($python_name),
+        python_name="python"; AC_MSG_RESULT(no))
+  AC_SUBST(vi_cv_path_python, $python_name)
   if test "X$vi_cv_path_python3" != "X"; then
 
     dnl -- get its version number
