--- src/lib/CMakeLists.txt.orig	2020-07-17 23:54:44.000000000 -0700
+++ src/lib/CMakeLists.txt	2020-07-17 23:54:58.000000000 -0700
@@ -21,7 +21,7 @@
 endif()
 
 add_library(ktoblzcheck ${MODE} ${ktoblzcheck_LIB_SRCS})
-target_link_libraries(ktoblzcheck stdc++ ${LIBS})
+target_link_libraries(ktoblzcheck ${LIBS})
 if(WIN32)
     set_target_properties(ktoblzcheck PROPERTIES SUFFIX "-${LIBKTOBLZCHECK_EFFECTIVE}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 else()
