--- configure.orig      2013-12-06 08:51:37.000000000 -0700
+++ configure	2018-10-28 11:16:01.000000000 -0700
@@ -20148,7 +20148,7 @@
 
 # Build hypre module if HYPRE is installed
 hypre=no
-if test "x$use_mpicc" = "xyes" ; then # HYPRE needs MPI
+if test "x" = "x" ; then # HYPRE does *not* need MPI
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for HYPRE_IJMatrixCreate in -lHYPRE" >&5
 $as_echo_n "checking for HYPRE_IJMatrixCreate in -lHYPRE... " >&6; }
 if ${ac_cv_lib_HYPRE_HYPRE_IJMatrixCreate+:} false; then :
