--- doc/auto-opts.tpl.orig	2011-03-28 22:46:20.000000000 -0500
+++ doc/auto-opts.tpl	2011-03-28 22:48:40.000000000 -0500
@@ -104,7 +104,7 @@
 @item
 @code{autogen checkopt.def}
 @item
-@code{cc -o check -g $@{cflags@} checkopt.c $@{ldflags@}}
+@code{%%CONFIGURE.CC%% -o check -g $@{cflags@} checkopt.c $@{ldflags@}}
 @item
 @code{./check --help}
 @end enumerate
@@ -188,7 +188,7 @@
 autogen default-test.def
 copts="-DTEST_DEFAULT_TEST_OPTS `autoopts-config cflags`"
 lopts="`autoopts-config ldflags`"
-cc -o default-test $@{copts@} default-test.c $@{lopts@}
+%%CONFIGURE.CC%% -o default-test $@{copts@} default-test.c $@{lopts@}
 @end example
 
 @noindent
@@ -304,7 +304,7 @@
 With that text in a file named ``hello.c'', this short script:
 
 @example
-cc -o hello hello.c `autoopts-config cflags ldflags`
+%%CONFIGURE.CC%% -o hello hello.c `autoopts-config cflags ldflags`
 ./hello
 echo 'greeting Buzz off' > hello.conf
 ./hello
@@ -318,7 +318,7 @@
 @example
 [= (texi-escape-encode (shell "
 cd ${tmp_dir}
-cc -o hello hello.c ${CFLAGS} ${LIBS} ${LDFLAGS} || \
+%%CONFIGURE.CC%% -o hello hello.c ${CFLAGS} ${LIBS} ${LDFLAGS} || \
   die cannot compile hello
 ./hello
 echo 'greeting Buzz off' > hello.conf
