--- a/ShichiZip/QuickActions/Extensions/QuickActionRequestHandlers.swift.old	2026-05-23 18:10:05
+++ a/ShichiZip/QuickActions/Extensions/QuickActionRequestHandlers.swift	2026-05-23 18:05:54
@@ -2,7 +2,7 @@
 import Foundation
 import UniformTypeIdentifiers
 
-class ShichiZipQuickActionRequestHandler: NSObject, @MainActor NSExtensionRequestHandling {
+class ShichiZipQuickActionRequestHandler: NSObject, @MainActor @preconcurrency NSExtensionRequestHandling {
     private enum LoadedFileReference {
         case durable(URL)
         case temporary(URL)
