Fix relative path in install_name.
--- CMakeLists.txt.orig	2022-03-04 01:56:56.000000000 +1100
+++ CMakeLists.txt	2022-08-28 06:15:00.000000000 +1000
@@ -74,7 +74,6 @@
     target_link_libraries(SDL PRIVATE dl)
 endif()
 if(APPLE)
-    set_target_properties(SDL PROPERTIES INSTALL_NAME_DIR "")
     set_target_properties(SDL PROPERTIES COMPILE_DEFINITIONS "_THREAD_SAFE")
     set_target_properties(SDL PROPERTIES LINK_FLAGS
             "-Wl,-compatibility_version,${DYLIB_COMPAT_VERSION} -Wl,-current_version,${DYLIB_CURRENT_VERSION}")
--- CMakeLists.txt.orig	2022-03-04 01:56:56.000000000 +1100
+++ CMakeLists.txt	2022-09-05 12:05:50.000000000 +1000
@@ -21,7 +21,7 @@
   set(OSX_SRCS "src/SDL12_compat_objc.m")
   set_source_files_properties(${OSX_SRCS} PROPERTIES LANGUAGE C)
   # the following matches SDL-1.2 Xcode project file
-  set(DYLIB_COMPAT_VERSION 1.0.0 CACHE STRING "library compatibility version")
+  set(DYLIB_COMPAT_VERSION 12.0.0 CACHE STRING "library compatibility version")
   set(DYLIB_CURRENT_VERSION 12.52.0 CACHE STRING "library current version")
   include_directories("/opt/X11/include")  # hack.
   if(CMAKE_VERSION VERSION_LESS 3.9)
