--- src/SDL12_compat.c.orig	2022-03-04 01:56:56.000000000 +1100
+++ src/SDL12_compat.c	2022-08-28 10:08:52.000000000 +1000
@@ -1043,6 +1043,7 @@ static char loaderror[256];
     static SDL_bool LoadSDL20Library(void) {
         /* I don't know if this is the _right_ order to try, but this seems reasonable */
         static const char * const dylib_locations[] = {
+            "__PREFIX__/lib/" SDL20_LIBNAME, /* MacPorts */
             "@loader_path/" SDL20_LIBNAME, /* MyApp.app/Contents/MacOS/libSDL2-2.0.0.dylib */
             "@loader_path/../Frameworks/" SDL20_FRAMEWORK, /* MyApp.app/Contents/Frameworks/SDL2.framework */
             "@executable_path/" SDL20_LIBNAME, /* MyApp.app/Contents/MacOS/libSDL2-2.0.0.dylib */
