*** ./ARmake.inc	Fri Apr 26 12:12:57 1996
--- ./ARmake.inc	Thu Dec  7 21:01:43 2006
***************
*** 34,39 ****
--- 34,44 ----
  #
  PLAT = SUN4
  #
+ #  The communication library ( need for parallel library only )
+ #  Current availible platforms are: BLACS or MPI
+ #
+ COMMLIB       = MPI
+ #
  #  %------------------------------------------------------%
  #  | The directories to find the various pieces of ARPACK |
  #  %------------------------------------------------------%
***************
*** 42,47 ****
--- 47,54 ----
  LAPACKdir    = $(home)/LAPACK
  UTILdir      = $(home)/UTIL
  SRCdir       = $(home)/SRC
+ PSRCdir      = $(home)/PARPACK/SRC/$(COMMLIB)
+ PUTILdir     = $(home)/PARPACK/UTIL/$(COMMLIB)
  #
  DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
  #
***************
*** 51,67 ****
  # | NOTE: ARPACK assumes the use of LAPACK version 2 codes.           |
  # %-------------------------------------------------------------------%
  #
! #DIRS         = $(UTILdir) $(SRCdir)
  #
  # %---------------------------------------------------%
  # | The name of the libraries to be created/linked to |
  # %---------------------------------------------------%
  #
! ARPACKLIB  = $(home)/libarpack_$(PLAT).a
! LAPACKLIB = 
  BLASLIB = 
  #
  ALIBS =  $(ARPACKLIB) $(LAPACKLIB) $(BLASLIB) 
  #
  # 
  # %---------------------------------------------------------%
--- 58,78 ----
  # | NOTE: ARPACK assumes the use of LAPACK version 2 codes.           |
  # %-------------------------------------------------------------------%
  #
! DIRS         = $(UTILdir) $(SRCdir)
  #
  # %---------------------------------------------------%
  # | The name of the libraries to be created/linked to |
  # %---------------------------------------------------%
  #
! ARPACKLIB  = $(home)/libarpack.a
! PARPACKLIB = $(home)/libparpack.a
! LAPACKLIB = -framework Accelerate
  BLASLIB = 
  #
  ALIBS =  $(ARPACKLIB) $(LAPACKLIB) $(BLASLIB) 
+ MPILIBS = 
+ PLIBS = $(PARPACKLIB) $(ALIBS) $(MPILIBS)
+ 
  #
  # 
  # %---------------------------------------------------------%
***************
*** 102,108 ****
  # %-----------------------------------------%
  #
  FC      = f77
! FFLAGS	= -O -cg89
  
  LDFLAGS = 
  CD      = cd
--- 113,119 ----
  # %-----------------------------------------%
  #
  FC      = f77
! FFLAGS	= -O2 -ff2c -fno-underscoring
  
  LDFLAGS = 
  CD      = cd
***************
*** 112,118 ****
  LN      = ln
  LNFLAGS = -s
  
! MAKE    = /bin/make
  
  RM      = rm
  RMFLAGS = -f
--- 123,129 ----
  LN      = ln
  LNFLAGS = -s
  
! MAKE    = make
  
  RM      = rm
  RMFLAGS = -f
*** ./UTIL/second.f       Mon Oct  2 11:56:44 1995
--- ./UTIL/second.f       Fri Dec  8 10:56:45 2006
***************
*** 20,34 ****
        REAL               TARRAY( 2 )
  *     ..
  *     .. External Functions ..
!       REAL               ETIME
!       EXTERNAL           ETIME
  *     ..
  *     .. Executable Statements ..
  *

!       T1 = ETIME( TARRAY )
!       T  = TARRAY( 1 )
! 
        RETURN
  *
  *     End of SECOND
--- 20,34 ----
        REAL               TARRAY( 2 )
  *     ..
  *     .. External Functions ..
!       REAL               SECOND_
!       EXTERNAL           SECOND_
  *     ..
  *     .. Executable Statements ..
  *

! *     T1 = ETIME( TARRAY )
! *      T  = TARRAY( 1 )
!       T  = SECOND_()
        RETURN
  *
  *     End of SECOND
q