--- ./SRC/cgetv0.f.orig	2012-08-07 11:23:20.000000000 +1000
+++ ./SRC/cgetv0.f	2012-08-07 10:48:58.000000000 +1000
@@ -168,7 +168,7 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   ccopy, cgemv, clarnv, cvout, arscnd
+      external   ccopy, cdotc, cgemv, clarnv, cvout, arscnd
 c
 c     %--------------------%
 c     | External Functions |
@@ -176,9 +176,7 @@
 c
       Real 
      &           scnrm2, slapy2
-      Complex
-     &           cdotc
-      external   cdotc, scnrm2, slapy2
+      external   scnrm2, slapy2
 c
 c     %-----------------%
 c     | Data Statements |
@@ -291,7 +289,7 @@
 c 
       first = .FALSE.
       if (bmat .eq. 'G') then
-          cnorm  = cdotc (n, resid, 1, workd, 1)
+          call cdotc(cnorm, n, resid, 1, workd, 1)
           rnorm0 = sqrt(slapy2(real(cnorm),aimag(cnorm)))
       else if (bmat .eq. 'I') then
            rnorm0 = scnrm2(n, resid, 1)
@@ -348,7 +346,7 @@
       end if
 c 
       if (bmat .eq. 'G') then
-         cnorm = cdotc (n, resid, 1, workd, 1)
+         call cdotc(cnorm, n, resid, 1, workd, 1)
          rnorm = sqrt(slapy2(real(cnorm),aimag(cnorm)))
       else if (bmat .eq. 'I') then
          rnorm = scnrm2(n, resid, 1)
--- ./SRC/cnaitr.f.orig	2012-08-07 11:24:18.000000000 +1000
+++ ./SRC/cnaitr.f	2012-08-07 10:50:20.000000000 +1000
@@ -272,18 +272,16 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   caxpy, ccopy, cscal, csscal, cgemv, cgetv0, 
+      external   caxpy, ccopy, cdotc, cscal, csscal, cgemv, cgetv0, 
      &           slabad, cvout, cmout, ivout, arscnd
 c
 c     %--------------------%
 c     | External Functions |
 c     %--------------------%
 c
-      Complex
-     &           cdotc 
       Real            
      &           slamch,  scnrm2, clanhs, slapy2
-      external   cdotc, scnrm2, clanhs, slamch, slapy2
+      external   scnrm2, clanhs, slamch, slapy2
 c
 c     %---------------------%
 c     | Intrinsic Functions |
@@ -550,7 +548,7 @@
 c        %-------------------------------------%
 c
          if (bmat .eq. 'G') then  
-             cnorm = cdotc (n, resid, 1, workd(ipj), 1)
+             call cdotc(cnorm, n, resid, 1, workd(ipj), 1)
              wnorm = sqrt( slapy2(real(cnorm),aimag(cnorm)) )
          else if (bmat .eq. 'I') then
              wnorm = scnrm2(n, resid, 1)
@@ -622,7 +620,7 @@
 c        %------------------------------%
 c
          if (bmat .eq. 'G') then         
-            cnorm = cdotc (n, resid, 1, workd(ipj), 1)
+            call cdotc(cnorm, n, resid, 1, workd(ipj), 1)
             rnorm = sqrt( slapy2(real(cnorm),aimag(cnorm)) )
          else if (bmat .eq. 'I') then
             rnorm = scnrm2(n, resid, 1)
@@ -722,7 +720,7 @@
 c        %-----------------------------------------------------%
 c 
          if (bmat .eq. 'G') then         
-             cnorm  = cdotc (n, resid, 1, workd(ipj), 1)
+             call cdotc(cnorm, n, resid, 1, workd(ipj), 1)
              rnorm1 = sqrt( slapy2(real(cnorm),aimag(cnorm)) )
          else if (bmat .eq. 'I') then
              rnorm1 = scnrm2(n, resid, 1)
--- ./SRC/cnaup2.f.orig	2012-08-07 11:25:22.000000000 +1000
+++ ./SRC/cnaup2.f	2012-08-07 10:51:23.000000000 +1000
@@ -239,18 +239,16 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   ccopy, cgetv0, cnaitr, cneigh, cngets, cnapps,
+      external   ccopy, cdotc, cgetv0, cnaitr, cneigh, cngets, cnapps,
      &           csortc, cswap, cmout, cvout, ivout, arscnd
 c
 c     %--------------------%
 c     | External functions |
 c     %--------------------%
 c
-      Complex 
-     &           cdotc
       Real   
      &           scnrm2, slamch, slapy2
-      external   cdotc, scnrm2, slamch, slapy2
+      external   scnrm2, slamch, slapy2
 c
 c     %---------------------%
 c     | Intrinsic Functions |
@@ -754,7 +752,7 @@
          end if
 c 
          if (bmat .eq. 'G') then         
-            cmpnorm = cdotc (n, resid, 1, workd, 1)
+            call cdotc(cmpnorm, n, resid, 1, workd, 1)
             rnorm = sqrt(slapy2(real (cmpnorm),aimag(cmpnorm)))
          else if (bmat .eq. 'I') then
             rnorm = scnrm2(n, resid, 1)
--- ./SRC/zgetv0.f.orig	2012-08-07 11:40:04.000000000 +1000
+++ ./SRC/zgetv0.f	2012-08-07 10:10:13.000000000 +1000
@@ -168,7 +168,7 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   zcopy, zgemv, zlarnv, zvout, arscnd
+      external   zcopy, zdotc, zgemv, zlarnv, zvout, arscnd
 c
 c     %--------------------%
 c     | External Functions |
@@ -176,9 +176,7 @@
 c
       Double precision 
      &           dznrm2, dlapy2
-      Complex*16
-     &           zdotc
-      external   zdotc, dznrm2, dlapy2
+      external   dznrm2, dlapy2
 c
 c     %-----------------%
 c     | Data Statements |
@@ -291,7 +289,7 @@
 c 
       first = .FALSE.
       if (bmat .eq. 'G') then
-          cnorm  = zdotc (n, resid, 1, workd, 1)
+          call zdotc(cnorm, n, resid, 1, workd, 1)
           rnorm0 = sqrt(dlapy2(dble(cnorm),dimag(cnorm)))
       else if (bmat .eq. 'I') then
            rnorm0 = dznrm2(n, resid, 1)
@@ -348,7 +346,7 @@
       end if
 c 
       if (bmat .eq. 'G') then
-         cnorm = zdotc (n, resid, 1, workd, 1)
+         call zdotc(cnorm, n, resid, 1, workd, 1)
          rnorm = sqrt(dlapy2(dble(cnorm),dimag(cnorm)))
       else if (bmat .eq. 'I') then
          rnorm = dznrm2(n, resid, 1)
--- ./SRC/znaitr.f.orig	2012-08-07 11:40:35.000000000 +1000
+++ ./SRC/znaitr.f	2012-08-07 10:12:42.000000000 +1000
@@ -272,18 +272,16 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   zaxpy, zcopy, zscal, zdscal, zgemv, zgetv0, 
+      external   zaxpy, zcopy, zdotc, zscal, zdscal, zgemv, zgetv0, 
      &           dlabad, zvout, zmout, ivout, arscnd
 c
 c     %--------------------%
 c     | External Functions |
 c     %--------------------%
 c
-      Complex*16
-     &           zdotc 
       Double precision            
      &           dlamch,  dznrm2, zlanhs, dlapy2
-      external   zdotc, dznrm2, zlanhs, dlamch, dlapy2
+      external   dznrm2, zlanhs, dlamch, dlapy2
 c
 c     %---------------------%
 c     | Intrinsic Functions |
@@ -550,7 +548,7 @@
 c        %-------------------------------------%
 c
          if (bmat .eq. 'G') then  
-             cnorm = zdotc (n, resid, 1, workd(ipj), 1)
+             call zdotc(cnorm,n, resid, 1, workd(ipj), 1)
              wnorm = sqrt( dlapy2(dble(cnorm),dimag(cnorm)) )
          else if (bmat .eq. 'I') then
              wnorm = dznrm2(n, resid, 1)
@@ -622,7 +620,7 @@
 c        %------------------------------%
 c
          if (bmat .eq. 'G') then         
-            cnorm = zdotc (n, resid, 1, workd(ipj), 1)
+            call zdotc(cnorm,n, resid, 1, workd(ipj), 1)
             rnorm = sqrt( dlapy2(dble(cnorm),dimag(cnorm)) )
          else if (bmat .eq. 'I') then
             rnorm = dznrm2(n, resid, 1)
@@ -722,7 +720,7 @@
 c        %-----------------------------------------------------%
 c 
          if (bmat .eq. 'G') then         
-             cnorm  = zdotc (n, resid, 1, workd(ipj), 1)
+             call zdotc (cnorm, n, resid, 1, workd(ipj), 1)
              rnorm1 = sqrt( dlapy2(dble(cnorm),dimag(cnorm)) )
          else if (bmat .eq. 'I') then
              rnorm1 = dznrm2(n, resid, 1)
--- ./SRC/znaup2.f.orig	2012-08-07 11:41:31.000000000 +1000
+++ ./SRC/znaup2.f	2012-08-07 10:21:53.000000000 +1000
@@ -239,18 +239,16 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   zcopy , zgetv0 , znaitr , zneigh , zngets , znapps ,
-     &           zsortc , zswap , zmout , zvout , ivout, arscnd
+      external   zcopy , zdotc , zgetv0 , znaitr , zneigh , zngets ,
+     &           znapps , zsortc , zswap , zmout , zvout , ivout, arscnd
 c
 c     %--------------------%
 c     | External functions |
 c     %--------------------%
 c
-      Complex*16
-     &           zdotc
       Double precision
      &           dznrm2 , dlamch , dlapy2
-      external   zdotc , dznrm2 , dlamch , dlapy2
+      external   dznrm2 , dlamch , dlapy2
 c
 c     %---------------------%
 c     | Intrinsic Functions |
@@ -754,7 +752,7 @@
          end if
 c
          if (bmat .eq. 'G') then
-            cmpnorm = zdotc  (n, resid, 1, workd, 1)
+            call zdotc(cmpnorm, n, resid, 1, workd, 1)
             rnorm = sqrt(dlapy2 (dble (cmpnorm),dimag (cmpnorm)))
          else if (bmat .eq. 'I') then
             rnorm = dznrm2 (n, resid, 1)
--- ./SRC/zneupd.f.orig	2012-08-07 11:42:11.000000000 +1000
+++ ./SRC/zneupd.f	2012-08-07 11:10:29.000000000 +1000
@@ -312,7 +312,7 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   zcopy , zgeru, zgeqr2, zlacpy, zmout,
+      external   zcopy, zdotc, zgeru, zgeqr2, zlacpy, zmout,
      &           zunm2r, ztrmm, zvout, ivout,
      &           zlahqr
 c  
@@ -324,10 +324,6 @@
      &           dznrm2, dlamch, dlapy2
       external   dznrm2, dlamch, dlapy2
 c
-      Complex*16
-     &           zdotc
-      external   zdotc
-c
 c     %-----------------------%
 c     | Executable Statements |
 c     %-----------------------%
@@ -731,7 +727,7 @@
 c                 | inner product can be set to j.           |
 c                 %------------------------------------------%
 c 
-                  workev(j) = zdotc(j, workl(ihbds), 1,
+                  call zdotc(workev(j), j, workl(ihbds), 1,
      &                        workl(invsub+(j-1)*ldq), 1)
  40         continue
 c
--- ./SRC/cneupd.f.orig	2012-08-07 15:21:02.000000000 +1000
+++ ./SRC/cneupd.f	2012-08-07 15:21:17.000000000 +1000
@@ -312,7 +312,7 @@
 c     | External Subroutines |
 c     %----------------------%
 c
-      external   ccopy , cgeru, cgeqr2, clacpy, cmout,
+      external   ccopy , cdotc, cgeru, cgeqr2, clacpy, cmout,
      &           cunm2r, ctrmm, cvout, ivout,
      &           clahqr
 c  
@@ -324,10 +324,6 @@
      &           scnrm2, slamch, slapy2
       external   scnrm2, slamch, slapy2
 c
-      Complex
-     &           cdotc
-      external   cdotc
-c
 c     %-----------------------%
 c     | Executable Statements |
 c     %-----------------------%
@@ -731,7 +727,7 @@
 c                 | inner product can be set to j.           |
 c                 %------------------------------------------%
 c 
-                  workev(j) = cdotc(j, workl(ihbds), 1,
+                  call cdotc( workev(j), j, workl(ihbds), 1,
      &                        workl(invsub+(j-1)*ldq), 1)
  40         continue
 c
