--- configure.orig	2012-06-08 11:18:01.000000000 -0500
+++ configure	2012-06-08 11:18:32.000000000 -0500
@@ -14917,13 +14917,13 @@
 
 # BLAS in ATLAS library? (http://math-atlas.sourceforge.net/)
 if test $ax_blas_ok = no; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ATL_xerbla in -latlas" >&5
-$as_echo_n "checking for ATL_xerbla in -latlas... " >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ATL_xerbla in -ltatlas" >&5
+$as_echo_n "checking for ATL_xerbla in -ltatlas... " >&6; }
 if ${ac_cv_lib_atlas_ATL_xerbla+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-latlas  $LIBS"
+LIBS="-ltatlas  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -14969,7 +14969,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lf77blas -latlas $LIBS"
+LIBS="-lf77blas -ltatlas $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15015,7 +15015,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcblas -lf77blas -latlas $LIBS"
+LIBS="-lcblas -lf77blas -ltatlas $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15055,7 +15055,7 @@
 $as_echo "$ac_cv_lib_cblas_cblas_dgemm" >&6; }
 if test "x$ac_cv_lib_cblas_cblas_dgemm" = xyes; then :
   ax_blas_ok=yes
-			 BLAS_LIBS="-lcblas -lf77blas -latlas"
+			 BLAS_LIBS="-lcblas -lf77blas -ltatlas"
 fi
 
 fi
