--- ./ARmake.inc	2007-03-29 16:08:05.000000000 +0200
+++ ./ARmake.inc	2007-03-29 16:08:15.000000000 +0200
@@ -34,6 +34,11 @@
 #
 PLAT = SUN4
 #
+#  The communication library ( need for parallel library only )
+#  Current availible platforms are: BLACS or MPI
+#
+COMMLIB       = MPI
+#
 #  %------------------------------------------------------%
 #  | The directories to find the various pieces of ARPACK |
 #  %------------------------------------------------------%
@@ -42,6 +47,8 @@
 LAPACKdir    = $(home)/LAPACK
 UTILdir      = $(home)/UTIL
 SRCdir       = $(home)/SRC
+PSRCdir      = $(home)/PARPACK/SRC/$(COMMLIB)
+PUTILdir     = $(home)/PARPACK/UTIL/$(COMMLIB)
 #
 DIRS        = $(BLASdir) $(LAPACKdir) $(UTILdir) $(SRCdir)
 #
@@ -51,17 +58,21 @@
 # | NOTE: ARPACK assumes the use of LAPACK version 2 codes.           |
 # %-------------------------------------------------------------------%
 #
-#DIRS         = $(UTILdir) $(SRCdir)
+DIRS         = $(UTILdir) $(SRCdir)
 #
 # %---------------------------------------------------%
 # | The name of the libraries to be created/linked to |
 # %---------------------------------------------------%
 #
-ARPACKLIB  = $(home)/libarpack_$(PLAT).a
-LAPACKLIB = 
+ARPACKLIB  = $(home)/libarpack.a
+PARPACKLIB = $(home)/libparpack.a
+LAPACKLIB = -framework Accelerate
 BLASLIB = 
 #
 ALIBS =  $(ARPACKLIB) $(LAPACKLIB) $(BLASLIB) 
+MPILIBS = 
+PLIBS = $(PARPACKLIB) $(ALIBS) $(MPILIBS)
+
 #
 # 
 # %---------------------------------------------------------%
@@ -102,7 +113,7 @@
 # %-----------------------------------------%
 #
 FC      = f77
-FFLAGS	= -O -cg89
+FFLAGS	= -O2 -ff2c
 
 LDFLAGS = 
 CD      = cd
@@ -112,7 +123,7 @@
 LN      = ln
 LNFLAGS = -s
 
-MAKE    = /bin/make
+MAKE    = make
 
 RM      = rm
 RMFLAGS = -f
diff -rubw ARPACK.orig/ARPACK/UTIL/second.f ARPACK/UTIL/second.f
--- ./UTIL/second.f	2007-03-29 16:08:05.000000000 +0200
+++ ./UTIL/second.f	2007-03-29 16:08:15.000000000 +0200
@@ -20,15 +20,15 @@
       REAL               TARRAY( 2 )
 *     ..
 *     .. External Functions ..
-      REAL               ETIME
-      EXTERNAL           ETIME
+*      REAL               ETIME
+*      EXTERNAL           ETIME
 *     ..
 *     .. Executable Statements ..
 *
 
-      T1 = ETIME( TARRAY )
-      T  = TARRAY( 1 )
-
+*     T1 = ETIME( TARRAY )
+*      T  = TARRAY( 1 )
+      T  = 1
       RETURN
 *
 *     End of SECOND
