--- makefile	2024-04-01 04:58:47.000000000 +0800
+++ makefile	2024-05-25 09:15:48.000000000 +0800
@@ -2,8 +2,8 @@
 .PHONY: all install install-newsraft install-man install-examples man html clean check cppcheck clang-tidy
 
 CC            = cc
-CFLAGS        = -O3
-LDFLAGS       =
+CFLAGS        = @CFLAGS@
+LDFLAGS       = @LDFLAGS@
 CURL_CFLAGS   = `pkg-config --cflags libcurl  2>/dev/null`
 CURL_LIBS     = `pkg-config --libs   libcurl  2>/dev/null || echo '-lcurl'`
 CURSES_CFLAGS = `pkg-config --cflags ncursesw 2>/dev/null`
