--- lib/encoder_impl.cc.orig	2015-03-20 13:14:26.000000000 -0400
+++ lib/encoder_impl.cc	2015-03-20 13:14:37.000000000 -0400
@@ -392,7 +392,7 @@
 void encoder_impl::prepare_group1a(void) {
 	std::cout << "preparing group 1" << std::endl;
 	//infoword[1] = infoword[1] | (1 << 4); // TMC in 8A
-	infoword[2] = (0b10000000 << 8) | 0xE0;
+	infoword[2] = (0x80 << 8) | 0xE0;
 	infoword[3] = 0; // time
 }
 
