--- Makefile.in.orig	2015-08-31 17:04:27.000000000 -0400
+++ Makefile.in	2015-08-31 17:04:58.000000000 -0400
@@ -162,7 +162,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = -lcrypt @LIBS@
+LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
 MAKE = @MAKE@
 MAKEINFO = @MAKEINFO@
@@ -272,7 +272,7 @@
 stamp-h1: $(srcdir)/config.h.in $(top_builddir)/config.status
 	@rm -f stamp-h1
 	cd $(top_builddir) && $(SHELL) ./config.status config.h
-$(srcdir)/config.h.in:  $(am__configure_deps) 
+$(srcdir)/config.h.in:  $(am__configure_deps)
 	($(am__cd) $(top_srcdir) && $(AUTOHEADER))
 	rm -f stamp-h1
 	touch $@
@@ -319,7 +319,7 @@
 
 clean-binPROGRAMS:
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
-mkpwd$(EXEEXT): $(mkpwd_OBJECTS) $(mkpwd_DEPENDENCIES) $(EXTRA_mkpwd_DEPENDENCIES) 
+mkpwd$(EXEEXT): $(mkpwd_OBJECTS) $(mkpwd_DEPENDENCIES) $(EXTRA_mkpwd_DEPENDENCIES)
 	@rm -f mkpwd$(EXEEXT)
 	$(LINK) $(mkpwd_OBJECTS) $(mkpwd_LDADD) $(LIBS)
 
