--- src/sqliteInt.h	2024-01-31
+++ src/sqliteInt.h	2024-01-31
@@ -886,7 +886,7 @@
 #   define SQLITE_PTRSIZE __SIZEOF_POINTER__
 # elif defined(i386)     || defined(__i386__)   || defined(_M_IX86) ||    \
        defined(_M_ARM)   || defined(__arm__)    || defined(__x86)   ||    \
-      (defined(__APPLE__) && defined(__POWERPC__)) ||                     \
+      (defined(__APPLE__) && defined(__ppc__))  ||                        \
       (defined(__TOS_AIX__) && !defined(__64BIT__))
 #   define SQLITE_PTRSIZE 4
 # else

--- src/tclsqlite.c	2024-01-31
+++ src/tclsqlite.c	2024-01-31
@@ -60,7 +60,7 @@
 #     define SQLITE_PTRSIZE __SIZEOF_POINTER__
 #   elif defined(i386)     || defined(__i386__)   || defined(_M_IX86) ||    \
          defined(_M_ARM)   || defined(__arm__)    || defined(__x86)   ||    \
-        (defined(__APPLE__) && defined(__POWERPC__)) ||                     \
+        (defined(__APPLE__) && defined(__ppc__))  ||                        \
         (defined(__TOS_AIX__) && !defined(__64BIT__))
 #     define SQLITE_PTRSIZE 4
 #   else

--- config.guess	2024-01-31
+++ config.guess	2024-01-31
@@ -1425,7 +1425,7 @@
 		esac
 	    fi
 	    # On 10.4-10.6 one might compile for PowerPC via gcc -arch ppc
-	    if (echo '#ifdef __POWERPC__'; echo IS_PPC; echo '#endif') | \
+	    if (echo '#ifdef __ppc__'; echo IS_PPC; echo '#endif') | \
 		   (CCOPTS="" $CC_FOR_BUILD -E - 2>/dev/null) | \
 		   grep IS_PPC >/dev/null
 	    then
