--- configure.orig	2022-12-28 08:26:45.000000000 -0600
+++ configure	2023-01-18 16:21:25.000000000 -0600
@@ -7898,7 +7898,7 @@
 
 			# JH: static-libgcc is necessary for core Tcl, but may
 			# not be necessary for extensions.
-			SHLIB_LD="$SHLIB_LD -m64 -static-libgcc"
+			# SHLIB_LD="$SHLIB_LD -m64 -static-libgcc"
 
 fi
 fi
@@ -8893,9 +8893,9 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
+	    # if test "$GCC" = "yes"; then
+		# SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
+	    # fi
 	    eval eval "PKG_LIB_FILE8=${PACKAGE_LIB_PREFIX8}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	    eval eval "PKG_LIB_FILE9=${PACKAGE_LIB_PREFIX9}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
