--- configure.orig	2016-08-11 14:17:12.000000000 -0500
+++ configure	2016-08-11 19:08:51.000000000 -0500
@@ -13187,7 +13187,7 @@
   return 0;
 }
 _ACEOF
-for ac_lib in '' termcap curses ncurses ncursesw; do
+for ac_lib in '' termcap ncurses ncursesw curses; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
@@ -13243,7 +13243,7 @@
   return 0;
 }
 _ACEOF
-for ac_lib in '' readline edit; do
+for ac_lib in '' edit readline; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
