--- configure.orig	2012-01-16 08:05:06.000000000 -0600
+++ configure	2012-01-21 02:25:14.000000000 -0600
@@ -19623,7 +19623,7 @@
   return 0;
 }
 _ACEOF
-for ac_lib in '' curses ncurses ncursesw; do
+for ac_lib in '' ncurses ncursesw curses; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
@@ -19679,7 +19679,7 @@
   return 0;
 }
 _ACEOF
-for ac_lib in '' readline; do
+for ac_lib in '' edit readline; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
