From fcf8f7c7aa1b0484486a7838403694df081b8833 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 26 May 2023 19:18:44 +0800
Subject: [PATCH 4/5] Fix macOS linking

---
 src/notify.c   | 3 ---
 src/x11focus.c | 4 ----
 2 files changed, 7 deletions(-)

diff --git src/notify.c src/notify.c
index 8863ed1..0cefb7a 100644
--- src/notify.c
+++ src/notify.c
@@ -39,9 +39,6 @@
 
 #if defined(AUDIO) || defined(SOUND_NOTIFY)
 #ifdef __APPLE__
-#include <OpenAL/al.h>
-#include <OpenAL/alc.h>
-#else
 #include <AL/al.h>
 #include <AL/alc.h>
 /* compatibility with older versions of OpenAL */
diff --git src/x11focus.c src/x11focus.c
index 0c4b1af..6faa0d2 100644
--- src/x11focus.c
+++ src/x11focus.c
@@ -22,8 +22,6 @@
 
 #include "x11focus.h"
 
-#ifndef __APPLE__
-
 #include <X11/Xlib.h>
 
 static struct Focus {
@@ -83,5 +81,3 @@ void terminate_x11focus(void)
     XCloseDisplay(Focus.display);
     XUnlockDisplay(Focus.display);
 }
-
-#endif /* !__APPLE__ */
