--- plugins/language-support-vala/Makefile.am.orig	2014-03-12 22:11:19.000000000 -0700
+++ plugins/language-support-vala/Makefile.am	2014-03-12 22:11:54.000000000 -0700
@@ -40,8 +40,7 @@
 	$(LIBANJUTA_CFLAGS) \
 	$(VALA_CFLAGS) \
 	-DPACKAGE_SRC_DIR=\"$(srcdir)\" \
-	-DG_LOG_DOMAIN=\"language-support-vala\" \
-	--no-warn
+	-DG_LOG_DOMAIN=\"language-support-vala\"
 
 # The plugin
 plugindir = $(anjuta_plugin_dir)
