--- plugins/language-support-vala/Makefile.in.orig	2013-11-14 13:21:14.000000000 -0800
+++ plugins/language-support-vala/Makefile.in	2013-11-25 20:46:49.000000000 -0800
@@ -509,8 +509,7 @@
 @ENABLE_VALA_TRUE@	$(LIBANJUTA_CFLAGS) \
 @ENABLE_VALA_TRUE@	$(VALA_CFLAGS) \
 @ENABLE_VALA_TRUE@	-DPACKAGE_SRC_DIR=\"$(srcdir)\" \
-@ENABLE_VALA_TRUE@	-DG_LOG_DOMAIN=\"language-support-vala\" \
-@ENABLE_VALA_TRUE@	--no-warn
+@ENABLE_VALA_TRUE@	-DG_LOG_DOMAIN=\"language-support-vala\"
 
 
 # The plugin
