--- plugins/language-support-vala/Makefile.in.orig	2013-09-18 17:59:47.000000000 -0700
+++ plugins/language-support-vala/Makefile.in	2013-09-18 18:00:11.000000000 -0700
@@ -453,8 +453,7 @@
 @ENABLE_VALA_TRUE@	$(LIBANJUTA_CFLAGS) \
 @ENABLE_VALA_TRUE@	$(VALA_CFLAGS) \
 @ENABLE_VALA_TRUE@	-DPACKAGE_SRC_DIR=\"$(srcdir)\" \
-@ENABLE_VALA_TRUE@	-DG_LOG_DOMAIN=\"language-support-vala\" \
-@ENABLE_VALA_TRUE@	--no-warn
+@ENABLE_VALA_TRUE@	-DG_LOG_DOMAIN=\"language-support-vala\"
 
 
 # The plugin
