--- configure.orig	2013-11-14 13:21:08.000000000 -0800
+++ configure	2013-11-25 20:46:49.000000000 -0800
@@ -5538,7 +5538,7 @@
 
 
 ANJUTA_LDFLAGS="-no-undefined"
-ANJUTA_PLUGIN_LDFLAGS="-module -avoid-version -no-undefined"
+ANJUTA_PLUGIN_LDFLAGS="-module -avoid-version"
 
 
 
@@ -18120,9 +18120,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
