--- plugins/python-loader/Makefile.am.orig	2014-01-12 04:37:43.000000000 -0500
+++ plugins/python-loader/Makefile.am	2014-11-24 14:48:13.000000000 -0500
@@ -38,12 +38,11 @@ plugin_LTLIBRARIES = libanjuta-python-lo
 # Plugin sources
 libanjuta_python_loader_la_SOURCES = plugin.c plugin.h
 
-libanjuta_python_loader_la_LDFLAGS = $(ANJUTA_PLUGIN_LDFLAGS)
+libanjuta_python_loader_la_LDFLAGS = $(ANJUTA_PLUGIN_LDFLAGS) $(PYTHON_LDFLAGS)
 
 # Plugin dependencies
 libanjuta_python_loader_la_LIBADD = \
 	$(LIBANJUTA_LIBS) \
-	$(PYTHON_LIBS) \
 	$(PYGOBJECT_LIBS)
 
 endif # HAVE_PYTHON
--- configure.ac.orig	2015-11-22 20:14:09.000000000 -0800
+++ configure.ac	2015-11-22 20:19:29.000000000 -0800
@@ -368,14 +368,14 @@
 if test x$have_python = xyes; then
 	if test -x $PYTHON-config; then
 		PYTHON_CFLAGS=`$PYTHON-config --cflags | sed -e "s|-Wno-unused-result||"`
-		PYTHON_LIBS=`$PYTHON-config --libs`
+		PYTHON_LDFLAGS=`$PYTHON-config --ldflags`
 		AC_SUBST(PYTHON_CFLAGS)
-		AC_SUBST(PYTHON_LIBS)
+		AC_SUBST(PYTHON_LDFLAGS)
 	elif test -x $PYTHON$PYTHON_VERSION-config; then
 		PYTHON_CFLAGS=`$PYTHON$PYTHON_VERSION-config --cflags | sed -e "s|-Wno-unused-result||"`
-		PYTHON_LIBS=`$PYTHON$PYTHON_VERSION-config --libs`
+		PYTHON_LDFLAGS=`$PYTHON-config --ldflags`
 		AC_SUBST(PYTHON_CFLAGS)
-		AC_SUBST(PYTHON_LIBS)		
+		AC_SUBST(PYTHON_LDFLAGS)
 	else
 		have_python="no"
 		echo "$PYTHON-config or $PYTHON$PYTHON_VERSION-config not found - disabling python"
