--- cli/build.clj.orig	2022-09-29 18:43:01.000000000 -0700
+++ cli/build.clj	2022-09-29 18:42:48.000000000 -0700
@@ -11,7 +11,7 @@
 (def clojars-lib 'com.github.clojure-lsp/clojure-lsp-standalone)
 (def current-version (string/trim (slurp (io/resource "CLOJURE_LSP_VERSION"))))
 (def class-dir "target/classes")
-(def basis {:project "deps.edn"})
+(def basis {:project "deps.edn" :extra {:mvn/local-repo "%%WORKSRCPATH%%/.m2/repository"}})
 (def uber-file (format "target/%s-standalone.jar" (name lib)))
 
 (defn clean [_]
