--- Makefile.in.orig	2010-10-07 14:39:41.000000000 +0200
+++ Makefile.in	2010-10-07 14:40:09.000000000 +0200
@@ -56,7 +56,7 @@
 install-doc:
 	${INSTALL} -d ${DESTDIR}$(MANDIR)
 	${INSTALL_DATA} $(DOCDIR)/$(manpage).1 ${DESTDIR}$(MANDIR)
-	/usr/bin/gzip -nf -9 $(MANDIR)/$(manpage).1
+	/usr/bin/gzip -nf -9 ${DESTDIR}$(MANDIR)/$(manpage).1
 	for dir in $(DOCDIR)/*/; do \
 	  if [ -f $${dir}$(manpage).1 ]; then \
 	    LOCMANDIR=${DESTDIR}$(prefix)/share/man$${dir#$(DOCDIR)}/man1 &&\
