--- configure.ac.orig	2010-10-08 17:47:04.000000000 +0200
+++ configure.ac	2010-10-08 17:47:24.000000000 +0200
@@ -196,7 +196,7 @@
 AC_TYPE_SIZE_T
 
 CFLAGS="$CFLAGS -Wall -Wshadow"
-CFLAGS="$CFLAGS -pedantic-errors -Wpointer-arith -Waggregate-return"
+CFLAGS="$CFLAGS -Wpointer-arith -Waggregate-return"
 CFLAGS="$CFLAGS -Wlong-long -Winline -Wredundant-decls -Wcast-qual -Wcast-align"
 DEFS="$DEFS -D__STRICT_ANSI__ -DNDEBUG"
 OPTS="-Wmissing-prototypes -Wstrict-prototypes -Wmissing-declarations"
