--- js/src/build/js.pc.in.orig
+++ js/src/build/js.pc.in
@@ -8,4 +8,4 @@
 Version: @MOZILLA_VERSION@
 @PKGCONF_REQUIRES_PRIVATE@
 Libs: -L${libdir} -l@JS_LIBRARY_NAME@ -lmozglue
-Cflags: -include ${includedir}/@JS_LIBRARY_NAME@/js/RequiredDefines.h -I${includedir}/@JS_LIBRARY_NAME@
+Cflags: -mone-byte-bool -include ${includedir}/@JS_LIBRARY_NAME@/js/RequiredDefines.h -I${includedir}/@JS_LIBRARY_NAME@
--- js/src/build/js-config.in.orig  
+++ js/src/build/js-config.in
@@ -108,7 +108,7 @@
 fi

 if test "$echo_cflags" = "yes"; then
-    echo "-std=gnu++11 -include $includedir/$JS_LIBRARY_NAME/js/RequiredDefines.h -I$includedir/$JS_LIBRARY_NAME $NSPR_CFLAGS"
+    echo "-mone-byte-bool -std=gnu++11 -include $includedir/$JS_LIBRARY_NAME/js/RequiredDefines.h -I$includedir/$JS_LIBRARY_NAME $NSPR_CFLAGS"
 fi

 if test "$echo_libs" = "yes"; then
