--- meson.build	2024-08-21 06:43:12.000000000 +0800
+++ meson.build	2024-11-10 14:02:31.000000000 +0800
@@ -88,10 +88,8 @@
 endif
 
 if host_machine.system() == 'darwin'
-  if cc.has_link_argument('-Wl,-ld_classic')
-    # New linker introduced in macOS 14 not working yet, see gh-19357 and gh-19387
-    add_project_link_arguments('-Wl,-ld_classic', language : ['c', 'cpp', 'fortran'])
-  endif
+  add_project_arguments('-D__STDC_NO_THREADS__', language : ['c', 'cpp'])
+  add_project_arguments('-Wno-error=incompatible-pointer-types', language: 'c')
   if cc.has_link_argument('-Wl,-dead_strip')
     # Allow linker to strip unused symbols
     add_project_link_arguments('-Wl,-dead_strip', language : ['c', 'cpp', 'fortran'])
