--- pyproject.toml.orig	2026-07-04 22:34:14
+++ pyproject.toml	2026-07-04 22:34:50
@@ -20,16 +20,16 @@
     # The upper bound on meson-python is pre-emptive only (looser
     # on purpose, since chance of breakage in 0.18/0.19 is low with
     # 0.17.1 working at time of writing)
-    "meson-python>=0.15.0,<0.20.0",
+    "meson-python>=0.15.0",
     # we need at least Cython 3.1.0a1 for free-threaded CPython;
     # for other CPython versions, the regular pre-emptive
     # Cython version bounds policy applies
-    "Cython>=3.0.8,<3.1.0",
+    "Cython>=3.0.8",
     # The upper bound on pybind11 is pre-emptive only
-    "pybind11>=2.13.2,<2.14.0",     # when updating version, also update check in scipy/meson.build
+    "pybind11>=2.13.2",     # when updating version, also update check in scipy/meson.build
     # The upper bound on pythran is pre-emptive only; 0.17.0
     # is released/working at time of writing
-    "pythran>=0.14.0,<0.18.0",
+    "pythran>=0.14.0",
 
     # numpy requirement for wheel builds for distribution on PyPI - building
     # against 2.x yields wheels that are also compatible with numpy 1.x at
@@ -38,7 +38,7 @@
     # redistributors can do this by installing the numpy version they like and
     # disabling build isolation.
     # NOTE: need numpy>=2.1.3 for free-threaded CPython 3.13 support
-    "numpy>=2.0.0,<2.5",
+    "numpy",
 ]
 
 [project]
