--- configure.org	2004-01-12 07:47:18.000000000 -0800
+++ configure	2006-09-21 23:33:37.000000000 -0700
@@ -39,7 +39,7 @@
 }
 EOF
 CXX=unknown;
-for i in "CC" "g++" "cc" "$CC"; do
+for i in "g++" "c++" "$CXX"; do
   if $i -c $tmp_file.C 2>/dev/null; then
     CXX="$i";
     break;
@@ -245,7 +245,7 @@
 EOF
 if eval "$CXX -c $LIBSTDCXX_INCLUDES $EXPLICIT $BOOL_DEFINITION $tmp_file.C" 2>/dev/null; then
   $echo 'defined in <memory>, good';
-else
+
   $echo 'not defined or not working, use "./libstd/include/auto_ptr.h"';
   AUTO_PTR_BROKEN="-DAUTO_PTR_BROKEN";
 fi;
