--- configure.orig	2010-12-29 10:07:16.000000000 -0600
+++ configure	2012-02-20 23:13:41.000000000 -0600
@@ -13713,7 +13713,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking link options" >&5
 $as_echo_n "checking link options... " >&6; }
 ZZIPLIB_LDFLAGS=""
-test ".$can_build_shared" != ".no" && ZZIPLIB_LDFLAGS="--export-dynamic"
+test ".$can_build_shared" != ".no" && ZZIPLIB_LDFLAGS="-export-dynamic"
 RESOLVES=" # "
 case "$host_os" in mingw*)
     ZZIPLIB_LDFLAGS="-no-undefined -export-all-symbols -mconsole"
