--- bins/unzzip.c.orig	2017-04-23 18:46:28.000000000 -0500
+++ bins/unzzip.c	2017-05-27 02:01:15.000000000 -0500
@@ -14,6 +14,23 @@
 extern int unzzip_print(int argc, char** argv);
 extern int unzzip_extract(int argc, char** argv);
 
+size_t strnlen(const char *s, size_t len)
+{
+    size_t i;
+    for(i=0; i<len && *(s+i); i++);
+    return i;
+}
+
+char* strndup (char const *s, size_t n)
+{
+    size_t len = strnlen (s, n);
+    char *new = malloc (len + 1);
+    if (new == NULL)
+    return NULL;
+    new[len] = '\0';
+    return memcpy (new, s, len);
+}
+
 static const char usage[] = 
 {
     "unzzip <dir>.. \n"
