--- modules/regexp/regexp_Regexp.c	2009-07-17 14:59:19.000000000 +0800
+++ modules/regexp/regexp_Regexp.c	2024-08-31 15:10:36.000000000 +0800
@@ -88,7 +88,7 @@
 		}
 		
 		/* get the number of subparts */
-		captured_str_cnt = pcre_info( rgx->compiled_re, NULL, NULL ) + 1;
+		captured_str_cnt = pcre_fullinfo( rgx->compiled_re, NULL, 0, NULL ) + 1;
 		/* create an offset array */
 		size_offsets = (int)(captured_str_cnt * 3);
 		offsets = (int *)fmalloc(size_offsets * sizeof(int));
