--- scripts/build/macosx_makebundle.py.orig	2013-03-25 14:55:00.000000000 +0100
+++ scripts/build/macosx_makebundle.py	2013-03-25 15:01:00.000000000 +0100
@@ -38,7 +38,7 @@
     CONTENTS_DIR = os.path.dirname(EXE_NAME) + "/"
 CONTENTS_DIR += BUNDLE_NAME + ".app/Contents/"
 
-BUNDLE_VERSION = commands.getoutput("git --git-dir="+SOURCE_DIR+"/.git/ describe")
+BUNDLE_VERSION = "@@VERSION@@"
 ICON_FILE = "pics/quassel.icns"
 
 def createBundle():
