--- pyobjc_setup.py.orig	2013-01-10 15:26:32.000000000 -0600
+++ pyobjc_setup.py	2014-08-12 06:20:25.000000000 -0500
@@ -186,7 +186,7 @@
 class pyobjc_install_lib (install_lib.install_lib):
     def get_exclusions(self):
         result = install_lib.install_lib.get_exclusions(self)
-        for fn in install_lib._install_lib.get_outputs(self):
+        for fn in install_lib.orig.install_lib.get_outputs(self):
             if 'PyObjCTest' in fn:
                 result[fn] = 1
 
