diff -ru ../Ice-3.3.0.orig/config/TestUtil.py ./config/TestUtil.py
--- ../Ice-3.3.0.orig/config/TestUtil.py	2008-05-16 10:24:00.000000000 -0700
+++ ./config/TestUtil.py	2008-05-28 14:39:16.000000000 -0700
@@ -838,7 +838,7 @@
             print >>output, "-Djava.net.preferIPv4Stack=true",
         print >>output,  exe,
     elif config.lang == "py":
-        print >>output, "python", exe,
+        print >>output, sys.executable, exe,
     elif config.lang == "php" and config.type == "client":
         print >>output, phpCmd, "-c tmp.ini -f", exe, " -- ",
     else:
@@ -923,13 +923,13 @@
 
             global keepGoing
             if script:
-                print "if ! python %s %s; then" % (os.path.join(dir, "run.py"), args)
+                print "if ! %s %s %s; then" % (sys.executable, os.path.join(dir, "run.py"), args)
                 print "  echo 'test in %s failed'" % os.path.abspath(dir)
                 if not keepGoing:
                     print "  exit 1"
                 print "fi"
             else:
-                status = os.system("python " + os.path.join(dir, "run.py " + args))
+                status = os.system(sys.executable + " " + os.path.join(dir, "run.py " + args))
 
                 if status:
                     if(num > 0):
