From: Sune Vuorela <sune@debian.org>
From: Modestas Vainius <modax@debian.org>
Subject: define Qt/KDE definitions when building standalone
Forwarded: no
Last-Update: 2011-06-30
Origin: vendor

Index: pykde4-4.13.97/CMakeLists.txt
===================================================================
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -33,6 +33,8 @@ set_package_properties(PythonInterp PROP
 
 include_directories(${KDE4_INCLUDES})
 
+add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS} -DQT3_SUPPORT)
+
 find_package(SIP)
 set_package_properties(SIP PROPERTIES DESCRIPTION "The SIP binding generator"
                        URL "http://riverbankcomputing.com" TYPE REQUIRED
