Use unversioned docdir.
--- Makefile.in.orig	2020-06-16 19:00:39.000000000 -0500
+++ Makefile.in	2021-03-08 18:30:47.000000000 -0600
@@ -424,7 +424,7 @@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
-htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
+htmldir = ${datadir}/doc/@PACKAGE@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
--- html/Makefile.in.orig	2020-06-16 19:00:39.000000000 -0500
+++ html/Makefile.in	2021-03-08 18:30:35.000000000 -0600
@@ -339,7 +339,7 @@
 host_cpu = @host_cpu@
 host_os = @host_os@
 host_vendor = @host_vendor@
-htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
+htmldir = ${datadir}/doc/@PACKAGE@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
@@ -362,7 +362,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-htmldeveldir = ${datadir}/doc/@PACKAGE@-@VERSION@/devel
+htmldeveldir = ${datadir}/doc/@PACKAGE@/devel
 html_DATA = $(wildcard *.html)
 htmldevel_DATA = $(wildcard devel/*)
 EXTRA_DIST = $(html_DATA) $(htmldevel_DATA)
