--- Makefile.in.orig	Wed Nov 24 23:12:12 2004
+++ Makefile.in	Wed Nov 24 23:12:22 2004
@@ -61,8 +61,8 @@
 
 prefix		= @prefix@
 bindir		= $(prefix)/bin
-catdir		= $(prefix)/man/cat$(manext)
-mandir		= $(prefix)/man/man$(manext)
+catdir		= $(prefix)/share/man/cat$(manext)
+mandir		= $(prefix)/share/man/man$(manext)
 manext		= 1
 shrlibdir	= $(shrlibtop)/$(PROGNAME)-$(VERSION)
 shrlibtop       = $(prefix)/share/lib/$(PROGNAME)
