diff --git lib/easyjour.rb lib/easyjour.rb
index 21d569b..dc5d996 100644
--- lib/easyjour.rb
+++ lib/easyjour.rb
@@ -28,7 +28,7 @@ module Easyjour
         text_record[key] = value
       end
       
-      @service = DNSSD.register(name, Easyjour.type_from_parts(service, protocol), 'local', port, text_record.encode, DNSSD::Flags::Add) do |reply|
+      @service = DNSSD.register(name, Easyjour.type_from_parts(service, protocol), 'local', port, text_record, DNSSD::Flags::Add) do |reply|
       end
     end
     
