--- llvmCore/lib/CodeGen/MachineSSAUpdater.cpp.orig	2015-10-12 19:33:17.000000000 -0400
+++ llvmCore/lib/CodeGen/MachineSSAUpdater.cpp	2015-10-12 19:34:31.000000000 -0400
@@ -38,7 +38,7 @@
   return *static_cast<IncomingPredInfoTy*>(IPI);
 }
 
-
+#pragma clang optimize off
 MachineSSAUpdater::MachineSSAUpdater(MachineFunction &MF,
                                      SmallVectorImpl<MachineInstr*> *NewPHI)
   : AV(0), IPI(0), InsertedPHIs(NewPHI) {
@@ -50,6 +50,7 @@
   delete &getAvailableVals(AV);
   delete &getIncomingPredInfo(IPI);
 }
+#pragma clang optimize on
 
 /// Initialize - Reset this object to get ready for a new set of SSA
 /// updates.  ProtoValue is the value used to name PHI nodes.
--- llvmCore/lib/Transforms/Utils/SSAUpdater.cpp.orig	2015-10-12 19:34:50.000000000 -0400
+++ llvmCore/lib/Transforms/Utils/SSAUpdater.cpp	2015-10-12 19:35:24.000000000 -0400
@@ -32,7 +32,7 @@
   return *static_cast<IncomingPredInfoTy*>(IPI);
 }
 
-
+#pragma clang optimize off
 SSAUpdater::SSAUpdater(SmallVectorImpl<PHINode*> *NewPHI)
   : AV(0), PrototypeValue(0), IPI(0), InsertedPHIs(NewPHI) {}
 
@@ -40,6 +40,7 @@
   delete &getAvailableVals(AV);
   delete &getIncomingPredInfo(IPI);
 }
+#pragma clang optimize on
 
 /// Initialize - Reset this object to get ready for a new set of SSA
 /// updates.  ProtoValue is the value used to name PHI nodes.
