--- llvmCore/Makefile.rules.orig	2010-03-17 14:29:48.000000000 -0700
+++ llvmCore/Makefile.rules	2014-01-01 17:51:22.000000000 -0800
@@ -555,11 +555,9 @@ endif
 #--------------------------------------------------------------------
 
 ifeq ($(HOST_OS),Darwin)
-  DARWIN_VERSION := `sw_vers -productVersion`
-  # Strip a number like 10.4.7 to 10.4
-  DARWIN_VERSION := $(shell echo $(DARWIN_VERSION)| sed -E 's/(10.[0-9]).*/\1/')
+  DARWIN_VERSION := @@DEPLOYMENT_TARGET@@
   # Get "4" out of 10.4 for later pieces in the makefile.
-  DARWIN_MAJVERS := $(shell echo $(DARWIN_VERSION)| sed -E 's/10.([0-9]).*/\1/')
+  DARWIN_MAJVERS := $(shell echo $(DARWIN_VERSION) | sed -E 's/10\.//')
 
   SharedLinkOptions=-Wl,-flat_namespace -Wl,-undefined -Wl,suppress \
                     -dynamiclib
