--- Makefile.orig	2007-09-03 19:29:51.000000000 -0400
+++ Makefile	2007-09-03 19:31:13.000000000 -0400
@@ -4,7 +4,7 @@
 #######################################################################
 
 # Directory where qiv will be installed under.
-PREFIX = /usr/local
+PREFIX = $(DESTDIR)@PREFIX@
 
 # Font to use for statusbar in fullscreen mode
 STATUSBAR_FONT = "fixed"
@@ -124,9 +124,11 @@
 
 install: $(PROGRAM)
 	@echo "Installing QIV..."
+	mkdir -p $(PREFIX)/bin
 	install -s -m 0755 $(PROGRAM) $(PREFIX)/bin
-	install -m 0644 $(PROGRAM).1 $(PREFIX)/man/man1
-	$(COMPRESS_PROG) $(PREFIX)/man/man1/$(PROGRAM).1
+	mkdir -p $(PREFIX)/share/man/man1
+	install -m 0644 $(PROGRAM).1 $(PREFIX)/share/man/man1
+	$(COMPRESS_PROG) $(PREFIX)/share/man/man1/$(PROGRAM).1
 	@if ./qiv -f ./intro.jpg ; \
 	then echo "-- Test Passed --" ; \
 	else echo "-- Test Failed --" ; \
