--- Makefile.orig	2017-11-03 19:27:27.000000000 +0100
+++ Makefile	2021-01-02 00:16:46.000000000 +0100
@@ -4,7 +4,7 @@
 #######################################################################
 
 # Directory where qiv will be installed under.
-PREFIX = /usr/local
+PREFIX = $(DESTDIR)@PREFIX@
 
 # Fonts to use for statusbar and comments
 STATUSBAR_FONT = "Monospace 9"
@@ -57,8 +57,7 @@
 PKG_CONFIG ?= pkg-config
 #CFLAGS    = -O0 -g -Wall
 CFLAGS    = -O2 -Wall \
-	    -fcaller-saves -ffast-math -fno-strength-reduce \
-	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
+	    -ffast-math -fno-strength-reduce
 #CFLAGS    = -O2 -Wall -fomit-frame-pointer -finline-functions \
 #	    -fcaller-saves -ffast-math -fno-strength-reduce \
 #	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
@@ -149,9 +148,9 @@
 	  echo install -d -m 0755 $(PREFIX)/bin; \
         fi
 	install -s -m 0755 $(PROGRAM) $(PREFIX)/bin
-	@if [ ! -e $(PREFIX)/man/man1 ]; then \
-	  echo install -d -m 0755 $(PREFIX)/man/man1; \
-	  install -d -m 0755 $(PREFIX)/man/man1; \
+	@if [ ! -e $(PREFIX)/share/man/man1 ]; then \
+	  echo install -d -m 0755 $(PREFIX)/share/man/man1; \
+	  install -d -m 0755 $(PREFIX)/share/man/man1; \
 	fi
 	install -m 0644 $(PROGRAM).1 $(PREFIX)/share/man/man1
 	$(COMPRESS_PROG) $(PREFIX)/share/man/man1/$(PROGRAM).1
@@ -165,10 +164,7 @@
 	  install -d -m 0755 $(PREFIX)/share/applications; \
 	fi
 	install -m 0644 qiv.desktop $(PREFIX)/share/applications/qiv.desktop
-	@if ./qiv -f ./intro.jpg ; \
-	then echo "-- Test Passed --" ; \
-	else echo "-- Test Failed --" ; \
-	fi
+	@echo "-- Test Skipped --"
 	@echo "\nDont forget to look into the \"qiv-command\" file and install it!\n-> cp qiv-command.example $(PREFIX)/bin/qiv-command\n\n"
 
 # the end... ;-)
