--- CMakeLists.txt.orig	2018-10-11 00:06:15.000000000 -0700
+++ CMakeLists.txt	2018-11-10 08:51:02.000000000 -0700
@@ -595,14 +595,6 @@
     endif()
   endif()
 
-  if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
-    if(CMAKE_Fortran_COMPILER MATCHES "g77")
-      list(APPEND LAPACK_LIBRARIES -lg2c)
-    elseif(NOT EXTERNAL_LIBRARIES MATCHES "gfortran")
-      list(APPEND LAPACK_LIBRARIES -lgfortran)
-    endif()
-  endif()
-
 else()
   set_config_option(HAVE_NO_FORTRAN "NoFortran")
 endif()
