--- src/netplex/Makefile.pre	2007-11-01 17:12:32.000000000 -0500
+++ src/netplex/Makefile.pre	2009-07-31 11:58:14.000000000 -0500
@@ -4,7 +4,7 @@
 PKGNAME = netplex
 GENERATE = netplex_ctrl_aux.ml netplex_ctrl_clnt.ml netplex_ctrl_srv.ml META
 PACKLIST = netplex-packlist
-RPCGEN = ../rpc-generator/ocamlrpcgen
+RPCGEN = ../rpc-generator/ocamlrpcgen -cpp @CPP@
 
 INSTALL_EXTRA = netplex-packlist netplex_mt.cmo netplex_mt.cmx netplex_mt.o
 INSTOTHER = install-netplex-admin
--- src/rpc/Makefile.pre	2007-11-01 17:12:32.000000000 -0500
+++ src/rpc/Makefile.pre	2009-07-31 11:58:11.000000000 -0500
@@ -5,7 +5,7 @@
 
 GENERATE = rpc_portmapper_aux.ml rpc_portmapper_clnt.ml META
 
-RPCGEN = ../rpc-generator/ocamlrpcgen
+RPCGEN = ../rpc-generator/ocamlrpcgen -cpp @CPP@
 
 include $(TOP_DIR)/Makefile.rules
 
--- src/rpc-auth-dh/Makefile.pre	2007-11-01 17:12:33.000000000 -0500
+++ src/rpc-auth-dh/Makefile.pre	2009-07-31 11:58:08.000000000 -0500
@@ -18,4 +18,4 @@
 	sed -e "s/@CONNECTOR@/$(RPC_AUTH_DH_CONNECTOR)/" \
 		rpc_key_service.mlp >rpc_key_service.ml
 
-RPCGEN = ../rpc-generator/ocamlrpcgen
+RPCGEN = ../rpc-generator/ocamlrpcgen -cpp @CPP@
