--- Makefile.PL.orig	2009-12-01 22:07:46.000000000 +0100
+++ Makefile.PL	2009-12-09 22:16:43.000000000 +0100
@@ -122,10 +122,10 @@
 	
 	# These might fail if a regular user is installing into a home
 	# directory.
-	-install -d $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 wikilist $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto.setup $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto-blog.setup $(DESTDIR)/etc/ikiwiki
+	-install -d $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 wikilist $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto-blog.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
 }
 }
 
