--- Makefile.PL.orig	2008-08-21 23:43:32.000000000 +0200
+++ Makefile.PL	2008-08-21 23:44:03.000000000 +0200
@@ -48,9 +48,9 @@
 	$(MAKE) -C po clean
 
 extra_install:
-	install -d $(DESTDIR)/etc/ikiwiki
-	install -m 0644 wikilist $(DESTDIR)/etc/ikiwiki
-	install -m 0644 auto.setup $(DESTDIR)/etc/ikiwiki
+	install -d $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	install -m 0644 wikilist $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	install -m 0644 auto.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
 
 	install -d $(DESTDIR)$(PREFIX)/share/ikiwiki
 	for dir in `cd underlays && find . -follow -type d ! -regex '.*\.svn.*'`; do \
