--- Makefile.PL.orig	2009-01-28 22:20:14.000000000 +0100
+++ Makefile.PL	2009-02-03 21:42:52.000000000 +0100
@@ -57,7 +57,7 @@
 	for dir in `cd underlays && find . -follow -type d ! -regex '.*\.svn.*'`; do \
 		install -d $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir; \
 		for file in `find underlays/$$dir -follow -maxdepth 1 -type f`; do \
-			cp -aL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir || \
+			cp -aL $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir 2>/dev/null || \
 			install -m 644 $$file $(DESTDIR)$(PREFIX)/share/ikiwiki/$$dir; \
 		done; \
 	done
@@ -71,11 +71,11 @@
 	done
 	
 	# Install example sites.
-	for dir in `cd doc/examples; find -type d ! -regex '.*\.svn.*'`; do \
+	for dir in `cd doc/examples; find . -type d ! -regex '.*\.svn.*'`; do \
 		install -d $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$dir; \
 	done
-	for file in `cd doc/examples; find -type f ! -regex '.*\.svn.*'`; do \
-		cp -aL doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file || \
+	for file in `cd doc/examples; find . -type f ! -regex '.*\.svn.*'`; do \
+		cp -aL doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file 2>/dev/null || \
 		install -m 644 doc/examples/$$file $(DESTDIR)$(PREFIX)/share/ikiwiki/examples/$$file; \
 	done
 
@@ -87,10 +87,10 @@
 	done
 	
 	install -d $(DESTDIR)$(PREFIX)/lib/ikiwiki/plugins
-	for file in `find plugins -maxdepth 1 -type f ! -wholename plugins/.\* ! -name \*demo\* -name \*.py`; do \
+	for file in `find plugins -maxdepth 1 -type f ! -path plugins/.\* ! -name \*demo\* -name \*.py`; do \
 		install -m 644 $$file $(DESTDIR)$(PREFIX)/lib/ikiwiki/plugins; \
 	done
-	for file in `find plugins -maxdepth 1 -type f ! -wholename plugins/.\* ! -name \*demo\* ! -name \*.py ! -name \*.pyc`; do \
+	for file in `find plugins -maxdepth 1 -type f ! -path plugins/.\* ! -name \*demo\* ! -name \*.py ! -name \*.pyc`; do \
 		install -m 755 $$file $(DESTDIR)$(PREFIX)/lib/ikiwiki/plugins; \
 	done
 
@@ -117,10 +117,10 @@
 	
 	# These might fail if a regular user is installing into a home
 	# directory.
-	-install -d $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 wikilist $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto.setup $(DESTDIR)/etc/ikiwiki
-	-install -m 0644 auto-blog.setup $(DESTDIR)/etc/ikiwiki
+	-install -d $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 wikilist $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
+	-install -m 0644 auto-blog.setup $(DESTDIR)$(PREFIX)/etc/ikiwiki
 }
 }
 
