--- Source/autotools/FindDependencies.m4.orig	2013-03-07 19:18:41.000000000 -0800
+++ Source/autotools/FindDependencies.m4	2013-03-07 19:18:48.000000000 -0800
@@ -108,10 +108,6 @@ GLIB_GSETTINGS
 
 # TODO: use pkg-config (after CFLAGS in their .pc files are cleaned up).
 case "$host" in
-    *-*-darwin*)
-        UNICODE_CFLAGS="-I$srcdir/Source/JavaScriptCore/icu -I$srcdir/Source/WebCore/icu"
-        UNICODE_LIBS="-licucore"
-        ;;
     *-*-mingw*)
         UNICODE_CFLAGS=""
         UNICODE_LIBS="-licui18n -licuuc"
