--- configure.orig	2009-09-18 19:51:51.000000000 +0200
+++ configure	2009-09-18 19:55:23.000000000 +0200
@@ -25,7 +25,7 @@
   lua_share=$lua_root/share/lua/5.1
   lua_lib=$lua_root/lib/lua/5.1
   bin_dir=$lua_root/bin
-  echo "LIB_OPTION= -shared" > config
+  echo "LIB_OPTION= -dynamiclib" > config
   echo "LUA_DIR= $lua_share" >> config
   echo "BIN_DIR= $bin_dir" >> config
   echo "LUA_LIBDIR= $lua_lib" >> config
