--- configure.orig	2019-01-02 08:43:31.000000000 -0600
+++ configure	2020-06-25 15:31:24.000000000 -0500
@@ -12902,6 +12902,7 @@
 #endif
 #include <stdio.h>
 #include <ctype.h>
+#include <stdlib.h>
 
 main(c, v)
 int	c;
@@ -12961,6 +12962,7 @@
 
 #include <sys/types.h>
 #include <fcntl.h>
+#include <stdlib.h>
 main()
 {
   int fd1, fd2, fl;
@@ -13012,6 +13014,7 @@
 #ifdef HAVE_UNISTD_H
 #  include <unistd.h>
 #endif
+#include <stdlib.h>
 main()
 {
 # ifdef GETPGRP_VOID
@@ -13225,6 +13228,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#include <stdlib.h>
 #if !HAVE_DECL_SYS_SIGLIST
 extern char *sys_siglist[];
 #endif
@@ -13309,6 +13313,7 @@
 #ifndef UNDER_SYS_SIGLIST_DECLARED
 extern char *_sys_siglist[];
 #endif
+#include <stdlib.h>
 main()
 {
 char *msg = (char *)_sys_siglist[2];
@@ -14626,6 +14631,7 @@
 #  include <ndir.h>
 # endif
 #endif /* HAVE_DIRENT_H */
+#include <stdlib.h>
 main()
 {
 DIR *dir;
@@ -14677,6 +14683,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 main()
 {
 long maxfds = ulimit(4, 0L);
@@ -14754,6 +14761,7 @@
 {
 return "42";
 }
+#include <stdlib.h>
 main()
 {
 char *s;
@@ -14805,7 +14813,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-
+#include <stdlib.h>
 main()
 {
 	char	*xpwd;
@@ -14860,7 +14868,7 @@
 #include <sys/types.h>
 #include <signal.h>
 #include <setjmp.h>
-
+#include <stdlib.h>
 main()
 {
 #if !defined (_POSIX_VERSION) || !defined (HAVE_POSIX_SIGNALS)
@@ -14929,7 +14937,8 @@
 #if defined (HAVE_LOCALE_H)
 #include <locale.h>
 #endif
-
+#include <stdlib.h>
+#include <string.h>
 main(c, v)
 int     c;
 char    *v[];
@@ -14999,7 +15008,7 @@
 /* end confdefs.h.  */
 
 #include <stdio.h>
-
+#include <stdlib.h>
 main()
 {
   int n;
@@ -15237,6 +15246,7 @@
 /* end confdefs.h.  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 int
@@ -15380,6 +15390,7 @@
   nsigint++;
 }
 
+#include <stdlib.h>
 main()
 {
 	nsigint = 0;
@@ -15501,7 +15512,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-
+#include <stdlib.h>
 /* Add more tests in here as appropriate. */
 main()
 {
@@ -15697,7 +15708,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include <errno.h>
-
+#include <stdlib.h>
 #ifndef errno
 extern int errno;
 #endif
@@ -15811,7 +15822,7 @@
 
 #include <sys/types.h>
 #include <signal.h>
-
+#include <stdlib.h>
 #ifndef NSIG
 #  define NSIG 64
 #endif
