--- relax.orig	2007-02-19 00:36:06.000000000 -0500
+++ relax	2008-08-11 02:13:30.000000000 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!@PREFIX@/bin/python2.6
 
 ###############################################################################
 #                                                                             #
--- scons/install.py.orig	2008-08-10 15:31:01.000000000 -0400
+++ scons/install.py	2008-08-10 16:01:22.000000000 -0400
@@ -112,8 +112,8 @@
     ###############
 
     # Run relax to create the *.pyc files.
-    print "\nRunning relax to create the byte-compiled *.pyc files."
-    system(env['SYMLINK'] + " --test")
+    # print "\nRunning relax to create the byte-compiled *.pyc files."
+    # system(env['SYMLINK'] + " --test")
 
     # Final print out.
     print "\n\n\n"
--- sconstruct.orig	2007-02-19 00:36:06.000000000 -0500
+++ sconstruct	2008-08-11 02:02:56.000000000 -0400
@@ -94,7 +94,7 @@
     SYS = SYSTEM
 
     # Mac OS X installation path.
-    INSTALL_PATH = sys.prefix + sep + 'local'
+    INSTALL_PATH = '@DEST_ROOT@@PREFIX@' + sep + 'lib'
 
 
 # All other operating systems.
@@ -111,16 +111,15 @@
 ###############
 
 # Relax installation directory.
-RELAX_PATH = INSTALL_PATH + sep + 'relax'
+RELAX_PATH = INSTALL_PATH + sep + 'relax'
 
 # Installation path for binaries.
-BIN_PATH = INSTALL_PATH + sep + 'bin'
+BIN_PATH = '@DEST_ROOT@@PREFIX@' + sep + 'bin'
 
 # Symbolic link installation path.
 SYMLINK = BIN_PATH + sep + 'relax'
 
 
-
 # The distribution files.
 #########################
 
@@ -486,7 +485,7 @@
 
         # Catch Mac OS X and send the correct command line options to the linker (these may become redundant as SCons improves).
         if env['PLATFORM'] == 'darwin':
-            env.Append(LINKFLAGS = ['-bundle', '-bundle_loader', sys.prefix + path.sep + 'bin' + path.sep + 'python', '-dynamic'])
+            env.Append(LINKFLAGS = ['-bundle', '-bundle_loader', sys.prefix + path.sep + 'bin' + path.sep + 'python2.6', '-dynamic', '-undefined', 'dynamic_lookup'])
             env['SHLINKFLAGS'] = SCons.Util.CLVar('$LINKFLAGS')
             env['SHLIBSUFFIX'] = '.so'
 
