Index: src/lib/platform/COSXScreen.cpp
===================================================================
--- src/lib/platform/COSXScreen.cpp	(revision 909)
+++ src/lib/platform/COSXScreen.cpp	(working copy)
@@ -620,7 +620,7 @@
 		// FIXME -- start watching jump zones
 		
 		// kCGEventTapOptionDefault = 0x00000000 (Missing in 10.4, so specified literally)
-		m_eventTapPort=CGEventTapCreate(kCGHIDEventTap, kCGHIDEventTap, 0, 
+		m_eventTapPort=CGEventTapCreate(kCGHIDEventTap, kCGHeadInsertEventTap, 0, 
 										kCGEventMaskForAllEvents, 
 										handleCGInputEvent, 
 										this);
@@ -640,7 +640,7 @@
                 // there may be a better way to do this, but we register an event handler even if we're
                 // not on the primary display (acting as a client). This way, if a local event comes in
                 // (either keyboard or mouse), we can make sure to show the cursor if we've hidden it. 
-		m_eventTapPort=CGEventTapCreate(kCGHIDEventTap, kCGHIDEventTap, 0, 
+		m_eventTapPort=CGEventTapCreate(kCGHIDEventTap, kCGHeadInsertEventTap, 0, 
 										kCGEventMaskForAllEvents, 
 										handleCGInputEventSecondary, 
 										this);
Index: src/lib/platform/COSXKeyState.cpp
===================================================================
--- src/lib/platform/COSXKeyState.cpp	(revision 909)
+++ src/lib/platform/COSXKeyState.cpp	(working copy)
@@ -284,22 +284,6 @@
 #endif
 
 	if (layoutValid) {
-
-		// choose action
-		UInt16 action;
-		switch (eventKind) {
-		case kEventRawKeyDown:
-			action = kUCKeyActionDown;
-			break;
-
-		case kEventRawKeyRepeat:
-			action = kUCKeyActionAutoKey;
-			break;
-
-		default:
-			return 0;
-		}
-
 		// translate key
 		UniCharCount count;
 		UniChar chars[2];
