--- CMakeLists.txt.orig	2016-11-21 15:45:07.000000000 -0800
+++ CMakeLists.txt	2016-11-21 15:45:27.000000000 -0800
@@ -173,16 +173,6 @@
 			message(FATAL_ERROR "Mac OS X target must be 10.x")
 		endif ()
 
-		if (OSX_TARGET_MINOR LESS 6)
-			# <= 10.5: 32-bit Intel and PowerPC
-			set(CMAKE_OSX_ARCHITECTURES "ppc;i386"
-				CACHE STRING "" FORCE)
-		else()
-			# >= 10.6: Intel only
-			set(CMAKE_OSX_ARCHITECTURES "i386"
-				CACHE STRING "" FORCE)
-		endif()
-		
 		set(CMAKE_CXX_FLAGS "--sysroot ${CMAKE_OSX_SYSROOT} ${CMAKE_CXX_FLAGS} -DGTEST_USE_OWN_TR1_TUPLE=1")
 
 		find_library(lib_ScreenSaver ScreenSaver)
