--- src/CMakeLists.txt.orig	2016-12-06 18:41:58.000000000 -0800
+++ src/CMakeLists.txt	2016-12-06 18:42:28.000000000 -0800
@@ -14,37 +14,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-if (WIN32)
-	if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-		set(OPENSSL_PLAT_DIR openssl-win64)
-	else()
-		set(OPENSSL_PLAT_DIR openssl-win32)
-	endif()
-	set(OPENSSL_INCLUDE ${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/inc32)
-endif()
-
-if (APPLE)
-	set(OPENSSL_PLAT_DIR openssl-osx)
-	set(OPENSSL_INCLUDE ${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/include)
-endif()
-
-if (WIN32)
-	set(OPENSSL_LIBS
-		${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/out32dll/libeay32.lib
-		${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/out32dll/ssleay32.lib
-	)
-endif()
-
-if (UNIX)
-	if (APPLE)
-		set(OPENSSL_LIBS
-		${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/libssl.a
-		${CMAKE_SOURCE_DIR}/ext/${OPENSSL_PLAT_DIR}/libcrypto.a
-		)
-	else()
-		set(OPENSSL_LIBS ssl crypto)
-	endif()
-endif()
+set(OPENSSL_INCLUDE @@PREFIX@@/include)
+set(OPENSSL_LIBS ssl crypto)
 
 add_subdirectory(lib)
 add_subdirectory(cmd)
