--- mddisplay.c.orig	2003-08-05 00:48:13
+++ mddisplay.c	2025-03-15 22:09:04
@@ -44,6 +44,7 @@
 #include <assert.h>
 #include <errno.h>
 #include <time.h>
+#include <unistd.h>
 
 #ifdef __APPLE__ // "Think different" for OS/X :) 
 #include <GLUT/glut.h>
@@ -3598,7 +3599,7 @@ void saveit(void) // so the user can resume later, whe
 
    // Write the movie and view structs, which contain transformation matrices,
    // as well as other info such as scaling
-   printf("   Writing session file %s (%d bytes).\n", sesFileName,sizeof(movie) + sizeof (view));
+   printf("   Writing session file %s (%ld bytes).\n", sesFileName,sizeof(movie) + sizeof (view));
    f = fopen_or_exit_if_error(sesFileName,"wb");
    fwrite_or_exit_if_error(&movie,sizeof (movie),1,f);
    fwrite_or_exit_if_error(&view,sizeof (view),1,f);
@@ -5225,7 +5226,7 @@ void readparms_and_allocate_globals(int recoverFlag)
       sizeof(atomattrib) *
             fread(&attribs[nat+nres+1],sizeof(atomattrib),1,f);
 
-   printf("%d attribute bytes read (%d attribute records).\n",
+   printf("%d attribute bytes read (%ld attribute records).\n",
       attribytesread,attribytesread/sizeof(atomattrib));
 
    /* CHECK FOR BOX COORDS */
