--- displib.c.orig	2003-08-04 07:50:36.000000000 -0700
+++ displib.c	2009-09-30 11:43:57.000000000 -0700
@@ -76,7 +76,7 @@
 
    if ( nread != n )
       {
-      fprintf(stderr,"Unable to fread(%d elements of size %d).  fread() returned %d  Error: %s\n",n,size,nread,strerror(errno));
+      fprintf(stderr,"Unable to fread(%ld elements of size %ld).  fread() returned %ld  Error: %s\n",n,size,nread,strerror(errno));
       exit(EXIT_FAILURE);
       }
 }
@@ -89,7 +89,7 @@
 
    if ( nwritten != n )
       {
-      fprintf(stderr,"Unable to fwrite(%d elements of size %d): %s\n",n,size,strerror(errno));
+      fprintf(stderr,"Unable to fwrite(%ld elements of size %ld): %s\n",n,size,strerror(errno));
       exit(EXIT_FAILURE);
       }
 }
