--- PKG-INFO	2011-03-04 15:34:07.000000000 -0600
+++ PKG-INFO	2012-12-06 14:29:46.000000000 -0600
@@ -1,6 +1,6 @@
 Metadata-Version: 1.0
 Name: VOEventLib
-Version: 0.2
+Version: 0.3
 Summary: A Python library for VOEvent
 Home-page: http://lib.skyalert.org/VOEventLib
 Author: Roy Williams, Dave Kuhlman
--- setup.py	2011-04-24 20:36:18.000000000 -0500
+++ setup.py	2012-12-06 14:29:53.000000000 -0600
@@ -1,7 +1,7 @@
 from distutils.core import setup
 
 setup(name='VOEventLib',
-      version='0.2',
+      version='0.3',
       description='A Python library for VOEvent',
       long_description='A Python library for VOEvent',
       author='Roy Williams, Dave Kuhlman',
