--- GNUmakefile.orig	2016-07-26 15:43:33.000000000 +0200
+++ GNUmakefile	2016-07-26 15:43:38.000000000 +0200
@@ -53,7 +53,7 @@
 
 ac_jpeg_ver = $(shell \
 	$(call ac_init,for libjpeg version);\
-	$(call ac_s_cmd,echo JPEG_LIB_VERSION \
+	$(call ac_s_cmd,printf JPEG_LIB_VERSION \
 		| cpp -include jpeglib.h | tail -n 1);\
 	$(call ac_fini))
 
@@ -61,19 +61,19 @@
 LIB		:= $(LIB)
 RESDIR		:= $(call ac_resdir)
 HAVE_ENDIAN_H	:= $(call ac_header,endian.h)
-HAVE_LINUX_FB_H	:= $(call ac_header,linux/fb.h)
+HAVE_LINUX_FB_H	:= no
 HAVE_NEW_EXIF	:= $(call ac_header,libexif/exif-log.h)
 HAVE_GLIBC	:= $(call ac_func,fopencookie)
 HAVE_STRSIGNAL	:= $(call ac_func,strsignal)
-HAVE_LIBPCD	:= $(call ac_lib,pcd_open,pcd)
-HAVE_LIBGIF	:= $(call ac_lib,DGifOpenFileName,gif)
-HAVE_LIBPNG	:= $(call ac_pkg_config,libpng)
-HAVE_LIBTIFF	:= $(call ac_pkg_config,libtiff-4)
-HAVE_LIBWEBP	:= $(call ac_pkg_config,libwebp)
-HAVE_LIBSANE	:= $(call ac_lib,sane_init,sane)
-HAVE_LIBCURL	:= $(call ac_lib,curl_easy_init,curl)
-HAVE_LIBLIRC	:= $(call ac_lib,lirc_init,lirc_client)
-HAVE_MOTIF	:= $(call ac_lib,XmStringGenerate,Xm,-L/usr/X11R6/$(LIB) -lXpm -lXt -lXext -lX11)
+HAVE_LIBPCD	:= no
+HAVE_LIBGIF	:= no
+HAVE_LIBPNG	:= no
+HAVE_LIBTIFF	:= no
+HAVE_LIBWEBP	:= no
+HAVE_LIBSANE	:= no
+HAVE_LIBCURL	:= no
+HAVE_LIBLIRC	:= no
+HAVE_MOTIF	:= no
 JPEG_VER        := $(call ac_jpeg_ver)
 endef
 
