--- setup.cfg.orig	2023-08-17 11:33:02
+++ setup.cfg	2023-08-17 11:33:33
@@ -34,15 +34,15 @@
     numpy>=1.15.0
     # We also need our Cython packages here to compile against 
     cymem>=2.0.2,<2.1.0
-    preshed>=3.0.2,<3.1.0
+    preshed>=3.0.2
     murmurhash>=1.0.2,<1.1.0
-    blis>=0.7.8,<0.8.0
+    blis>=0.7.8
 install_requires =
     # Explosion-provided dependencies
-    blis>=0.7.8,<0.8.0
+    blis>=0.7.8
     murmurhash>=1.0.2,<1.1.0
     cymem>=2.0.2,<2.1.0
-    preshed>=3.0.2,<3.1.0
+    preshed>=3.0.2
     wasabi>=0.8.1,<1.2.0
     srsly>=2.4.0,<3.0.0
     catalogue>=2.0.4,<2.1.0
