diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index ed0fe3d..be0089e 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -37,7 +37,7 @@ endif(ENABLE_DOXYGEN)
 ########################################################################
 # Convert manual pages from rst to roff and install them
 ########################################################################
-find_program(RST2MAN_EXECUTABLE NAMES rst2man rst2man.py rst2man2 rst2man2.py)
+find_program(RST2MAN_EXECUTABLE NAMES rst2man-2.7.py)
 
 function(BUILD_MAN_PAGE _sources _src _dst)
     add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${_dst}"
