--- cmake/generate_version_txt.cmake.orig	2026-03-26 22:43:00
+++ cmake/generate_version_txt.cmake	2026-03-26 22:43:00
@@ -10,7 +10,7 @@
 
 if(NOT ${armtoolchain_COMMIT} MATCHES "^[a-f0-9]+$")
     execute_process(
-        COMMAND git -C ${ArmToolchainForEmbedded_SOURCE_DIR} rev-parse HEAD
+        COMMAND git -c safe.directory=* -C ${ArmToolchainForEmbedded_SOURCE_DIR} rev-parse HEAD
         OUTPUT_VARIABLE armtoolchain_COMMIT
         OUTPUT_STRIP_TRAILING_WHITESPACE
         COMMAND_ERROR_IS_FATAL ANY
@@ -30,7 +30,7 @@
         endif()
 
         execute_process(
-            COMMAND git -C ${${base_library}_SOURCE_DIR} rev-parse HEAD
+            COMMAND git -c safe.directory=* -C ${${base_library}_SOURCE_DIR} rev-parse HEAD
             OUTPUT_VARIABLE ${base_library}_COMMIT
             OUTPUT_STRIP_TRAILING_WHITESPACE
             COMMAND_ERROR_IS_FATAL ANY
