--- tcp.c.orig	2024-05-08 01:54:44
+++ tcp.c	2024-05-10 16:16:28
@@ -161,14 +161,14 @@
 	/* connect to peer */
 	if (tfo && *tfo)
 	{
-#if defined(__FreeBSD__)
+/* #if defined(__FreeBSD__) */
 		int enable = 1;                                                                                                                                                                                             
 		setsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN, &enable, sizeof(enable));                                                                                                                                         
 
 		rc = sendto(fd, msg, msg_len, 0, ai->ai_addr, ai->ai_addrlen);                                                                                                                                              
-#else
+/* #else
 		rc = sendto(fd, msg, msg_len, MSG_FASTOPEN, ai -> ai_addr, ai -> ai_addrlen);
-#endif
+#endif */
 		
 		if (rc == msg_len)
 			*msg_accepted = 1;
