--- librawstudio/rawstudio.h.orig	2021-06-03 05:43:01.000000000 +0800
+++ librawstudio/rawstudio.h	2023-06-09 06:18:23.000000000 +0800
@@ -23,10 +23,6 @@
 #include <sys/types.h>
 #include "rs-types.h"
 
-#ifdef  __cplusplus
-extern "C" {
-#endif
-
 #include "rs-macros.h"
 
 #include "rs-debug.h"
@@ -78,8 +74,4 @@
 
 #include "x86-cpu.h"
 
-#ifdef  __cplusplus
-} /* extern "c" */
-#endif
-
 #endif /* RAWSTUDIO_H */

--- librawstudio/rs-macros.h.orig	2021-06-03 05:43:01.000000000 +0800
+++ librawstudio/rs-macros.h	2023-06-09 06:26:30.000000000 +0800
@@ -51,12 +51,10 @@
 #if __GNUC__ >= 3
 #define likely(x) __builtin_expect (!!(x), 1)
 #define unlikely(x) __builtin_expect (!!(x), 0)
-#define align(x) __attribute__ ((aligned (x)))
 #define __deprecated __attribute__ ((deprecated))
 #else
 #define likely(x) (x)
 #define unlikely(x) (x)
-#define align(x)
 #define __deprecated
 #endif
 
--- src/rs-tag-gui.c.orig	2023-06-09 07:20:35.000000000 +0800
+++ src/rs-tag-gui.c	2023-06-09 07:20:46.000000000 +0800
@@ -90,11 +90,6 @@
 	return box;
 }
 
-GtkWidget *
-rs_library_tag_entry_new(RSLibrary *library)
-{
-	g_assert(RS_IS_LIBRARY(library));
-
 	gboolean
 	selected(GtkEntryCompletion *completion, GtkTreeModel *model, GtkTreeIter *iter, gpointer user_data)
 	{
@@ -176,6 +171,11 @@
 		return found;
 	}
 
+GtkWidget *
+rs_library_tag_entry_new(RSLibrary *library)
+{
+	g_assert(RS_IS_LIBRARY(library));
+
 	GtkWidget *entry = gtk_entry_new();
 	GtkEntryCompletion *completion = gtk_entry_completion_new();
 	GtkListStore *store = gtk_list_store_new(1, G_TYPE_STRING);

