From ff6f56b68e3be3a3436134b7a0270cb6c92fb7cc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 17 Mar 2024 16:44:27 +0800
Subject: [PATCH] Fix for missing _exposure_ramp

---
 plugins/dcp/dcp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git plugins/dcp/dcp.c plugins/dcp/dcp.c
index 9fabc1e4..fef4b1bf 100644
--- plugins/dcp/dcp.c
+++ plugins/dcp/dcp.c
@@ -693,7 +693,7 @@ neutral_to_xy(RSDcp *dcp, const RS_VECTOR3 *neutral)
 	return last;
 }
 
-inline void
+static void
 RGBtoHSV(gfloat r, gfloat g, gfloat b, gfloat *h, gfloat *s, gfloat *v)
 {
 	*v = MAX(r, MAX (g, b));
@@ -723,7 +723,7 @@ RGBtoHSV(gfloat r, gfloat g, gfloat b, gfloat *h, gfloat *s, gfloat *v)
 	}
 }
 
-inline gfloat
+static gfloat
 exposure_ramp (RSDcp *dcp, gfloat x)
 {
 	if (x <= dcp->exposure_black - dcp->exposure_radius)
@@ -738,7 +738,7 @@ exposure_ramp (RSDcp *dcp, gfloat x)
 }
 
 
-inline void
+static void
 HSVtoRGB(gfloat h, gfloat s, gfloat v, gfloat *r, gfloat *g, gfloat *b)
 {
 	if (s > 0.0f)
