diff -ru ./src/libmdb/backend.c ../mdbtools-0.6pre1/src/libmdb/backend.c
--- ./src/libmdb/backend.c	2004-06-11 15:56:41.000000000 +0200
+++ ../mdbtools-0.6pre1/src/libmdb/backend.c	2009-02-26 21:06:17.000000000 +0100
@@ -28,7 +28,7 @@
 #endif
 
 static int is_init;
-static GHashTable *mdb_backends;
+GHashTable *mdb_backends;
 
    /*    Access data types */
 static MdbBackendType mdb_access_types[] = {
