diff -ur cdparanoia-III-10.2/paranoia/overlap.c cdparanoia-III-10.2/paranoia/overlap.c
--- cdparanoia-III-10.2/paranoia/overlap.c	2008-08-07 02:27:41.000000000 +0800
+++ cdparanoia-III-10.2/paranoia/overlap.c	2021-01-04 20:12:11.000000000 +0800
@@ -107,7 +107,7 @@
        sector, frob it.  We just want a little hysteresis [sp?]*/
     long av=(p->stage2.offpoints?p->stage2.offaccum/p->stage2.offpoints:0);
     
-    if(abs(av)>p->dynoverlap/4){
+    if(labs(av)>p->dynoverlap/4){
       av=(av/MIN_SECTOR_EPSILON)*MIN_SECTOR_EPSILON;
       
       if(callback)(*callback)(ce(p->root.vector),PARANOIA_CB_DRIFT);
@@ -207,7 +207,7 @@
   if(o->offpoints!=-1){
 
     /* Track the average magnitude of jitter (in either direction) */
-    o->offdiff+=abs(value);
+    o->offdiff+=labs(value);
     o->offpoints++;
     o->newpoints++;
 
