--- cdparanoia-III-10.2/paranoia/paranoia.c.orig	2008-09-11 14:09:40.000000000 -0600
+++ cdparanoia-III-10.2/paranoia/paranoia.c	2009-06-04 14:16:21.000000000 -0600
@@ -127,7 +127,7 @@
   FLAGS_EDGE    =0x1, /**< first/last N words of frame */
   FLAGS_UNREAD  =0x2, /**< unread, hence missing and unmatchable */
   FLAGS_VERIFIED=0x4  /**< block read and verified */
-} paranoia_read_flags;
+} paranoia_read_flags=FLAGS_UNREAD;
 
 /**** matching and analysis code *****************************************/
 
