--- CMakeLists.txt.orig	2017-10-12 18:11:01.000000000 -0400
+++ CMakeLists.txt	2023-07-26 20:30:23.000000000 -0400
@@ -72,13 +72,16 @@
 )
 
 add_library(caption ${CAPTION_SOURCES})
+add_library(caption_shared SHARED ${CAPTION_SOURCES})
 
 if(ENABLE_RE2C)
   add_dependencies(caption re2c)
+  add_dependencies(caption_shared re2c)
 endif()
 
 if(CMAKE_VERSION VERSION_EQUAL 2.8.12 OR CMAKE_VERSION VERSION_GREATER 2.8.12)
 target_include_directories(caption PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(caption_shared PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
 option(BUILD_EXAMPLES "Build examples" ON)
@@ -93,7 +96,10 @@
 #add_executable(test_captions unit_tests/test_sei.c )
 #target_link_libraries(test_captions caption)
 
+set_target_properties(caption_shared PROPERTIES OUTPUT_NAME caption)
+
 install (TARGETS caption DESTINATION lib EXPORT caption-targets)
+install (TARGETS caption_shared DESTINATION lib EXPORT caption-targets)
 install (FILES ${CAPTION_HEADERS} DESTINATION include/caption)
 
 find_package(Doxygen)
