--- configure.in.orig	2013-11-05 01:45:25.000000000 +0100
+++ configure.in	2013-11-05 01:46:40.000000000 +0100
@@ -161,25 +161,12 @@
 if test "$CYGWIN32" = yes; then
     LIBS="$LIBS -L../nxcomp -lXcomp -lstdc++ -Wl,-e,_mainCRTStartup -ljpeg -lpng -lz"
 else
-    LIBS="$LIBS -L../nxcomp -lXcomp"
+    LIBS="$LIBS -lXcomp"
 fi
 
 dnl Find makedepend somewhere.
 
 AC_SUBST(MAKEDEPEND)
-
-if test -x "../nx-X11/config/makedepend/makedepend" ; then
-  MAKEDEPEND=../nx-X11/config/makedepend/makedepend
-else
-  if test -x "/usr/X11R6/bin/makedepend" ; then
-    MAKEDEPEND=/usr/X11R6/bin/makedepend
-  else
-    if test -x "/usr/openwin/bin/makedepend" ; then
-      MAKEDEPEND=/usr/openwin/bin/makedepend
-    else
-      MAKEDEPEND=makedepend
-    fi
-  fi
-fi
+MAKEDEPEND=`which makedepend`
 
 AC_OUTPUT(Makefile)
