--- configure.in.orig	2013-11-05 01:12:25.000000000 +0100
+++ configure.in	2013-11-05 01:12:56.000000000 +0100
@@ -187,7 +187,7 @@
 dnl the options -G -h.
 
 if test "$DARWIN" = yes; then
-  LDFLAGS="$LDFLAGS -bundle"
+  LDFLAGS="$LDFLAGS -dynamiclib"
 elif test "$SUN" = yes; then
   LDFLAGS="$LDFLAGS -G -h \$(LIBLOAD)"
 else
@@ -362,20 +362,7 @@
 dnl Find makedepend somewhere.
 
 AC_SUBST(MAKEDEPEND)
-
-if test -x "../nx-X11/config/makedepend/makedepend" ; then
-  MAKEDEPEND=../nx-X11/config/makedepend/makedepend
-else
-  if test -x "/usr/X11R6/bin/makedepend" ; then
-    MAKEDEPEND=/usr/X11R6/bin/makedepend
-  else
-    if test -x "/usr/openwin/bin/makedepend" ; then
-      MAKEDEPEND=/usr/openwin/bin/makedepend
-    else
-      MAKEDEPEND=/usr/bin/makedepend
-    fi
-  fi
-fi
+MAKEDEPEND=`which makedepend`
 
 dnl Determine what to build based on the platform.
 dnl Override the LIBS settings on Cygwin32 so that
