--- Makefile.in.orig	2013-03-05 00:09:10.000000000 +0100
+++ Makefile.in	2013-03-05 00:16:19.000000000 +0100
@@ -91,9 +91,9 @@
 LIBRARY = Xcomp
 
 LIBNAME    = lib$(LIBRARY)
-LIBFULL    = lib$(LIBRARY).so.$(VERSION)
-LIBLOAD    = lib$(LIBRARY).so.$(LIBVERSION)
-LIBSHARED  = lib$(LIBRARY).so
+LIBFULL    = lib$(LIBRARY)$(VERSION).dylib
+LIBLOAD    = lib$(LIBRARY)$(LIBVERSION).dylib
+LIBSHARED  = lib$(LIBRARY).dylib
 LIBARCHIVE = lib$(LIBRARY).a
 
 LIBCYGSHARED  = cyg$(LIBRARY).dll
@@ -231,7 +231,11 @@
 CXXOBJ = $(CXXSRC:.cpp=.o)
 
 $(LIBFULL):	 $(CXXOBJ) $(COBJ)
-		 $(CXX) -o $@ $(LDFLAGS) $(CXXOBJ) $(COBJ) $(LIBS)
+		 $(CXX) -o $@ \
+			 -install_name $(libdir)/nx/$@ \
+			 -compatibility_version $(LIBVERSION) \
+			 -current_version $(LIBVERSION) \
+			 $(LDFLAGS) $(CXXOBJ) $(COBJ) $(LIBS)
 
 $(LIBLOAD):	 $(LIBFULL)
 		 rm -f $(LIBLOAD)
@@ -277,9 +281,9 @@
 	./mkinstalldirs $(DESTDIR)${libdir}/nx
 	./mkinstalldirs $(DESTDIR)${includedir}/nx
 	$(INSTALL_DATA) $(LIBFULL)              $(DESTDIR)${libdir}/nx
-	$(INSTALL_LINK) libXcomp.so.3           $(DESTDIR)${libdir}/nx
-	$(INSTALL_LINK) libXcomp.so             $(DESTDIR)${libdir}/nx
-	$(INSTALL_DATA) libXcomp.a              $(DESTDIR)${libdir}/nx
+	$(INSTALL_LINK) $(LIBLOAD)              $(DESTDIR)${libdir}/nx
+	$(INSTALL_LINK) $(LIBSHARED)            $(DESTDIR)${libdir}/nx
+	$(INSTALL_DATA) $(LIBARCHIVE)           $(DESTDIR)${libdir}/nx
 	$(INSTALL_DATA) NX*.h                   $(DESTDIR)${includedir}/nx
 	$(INSTALL_DATA) MD5.h                   $(DESTDIR)${includedir}/nx
 	echo "Running ldconfig tool, this may take a while..." && ldconfig || true
@@ -292,9 +296,9 @@
 
 uninstall.lib:
 	$(RM_FILE) $(DESTDIR)${libdir}/nx/$(LIBFULL)
-	$(RM_FILE) $(DESTDIR)${libdir}/nx/libXcomp.so.3
-	$(RM_FILE) $(DESTDIR)${libdir}/nx/libXcomp.so
-	$(RM_FILE) $(DESTDIR)${libdir}/nx/libXcomp.a
+	$(RM_FILE) $(DESTDIR)${libdir}/nx/$(LIBLOAD)
+	$(RM_FILE) $(DESTDIR)${libdir}/nx/$(LIBSHARED)
+	$(RM_FILE) $(DESTDIR)${libdir}/nx/$(LIBARCHIVE)
 	$(RM_FILE) $(DESTDIR)${includedir}/nx/NXalert.h
 	$(RM_FILE) $(DESTDIR)${includedir}/nx/NX.h
 	$(RM_FILE) $(DESTDIR)${includedir}/nx/NXmitshm.h
