--- iodev/network/eth_socket.cc.orig	2020-02-16 15:46:49.000000000 -0500
+++ iodev/network/eth_socket.cc	2020-02-16 16:05:10.000000000 -0500
@@ -95,7 +95,6 @@
 #include <netinet/in.h>
 #include <net/ethernet.h>
 #include <net/if.h>
-#include <linux/types.h>
 #include <netdb.h>
 #define closesocket(s) close(s)
 typedef int SOCKET;
@@ -105,6 +104,8 @@
 #endif
 };
 
+#define MSG_NOSIGNAL 0
+
 #ifdef WIN32
 #define MSG_NOSIGNAL 0
 #define MSG_DONTWAIT 0
