--- libs/random/random_device.cpp.orig	2009-06-11 15:27:21.000000000 +0200
+++ libs/random/random_device.cpp	2009-06-11 15:28:01.000000000 +0200
@@ -22,7 +22,7 @@
 #endif
 
 
-#if defined(__linux__) || defined (__FreeBSD__)
+#if defined(__linux__) || defined (__FreeBSD__) || defined(__APPLE__)
 
 // the default is the unlimited capacity device, using some secure hash
 // try "/dev/random" for blocking when the entropy pool has drained
