--- tools/build/v2/engine/build.jam.orig	2011-03-21 22:43:37.000000000 -0500
+++ tools/build/v2/engine/build.jam	2011-03-21 22:45:23.000000000 -0500
@@ -194,7 +194,7 @@
     -I$(--python-include) -I$(--extra-include)
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
 ## MacOSX Darwin, using GCC 2.9.x, 3.x
-toolset darwin cc :  "-o " : -D
+toolset darwin %%CONFIGURE.CC%% :  "-o " : -D
     :
     [ opt --release : -Wl,-x -O3 -finline-functions ]
     [ opt --debug : -g -O0 -fno-inline -pg ]
