diff -ruN libs/context/build/Jamfile.v2 libs/context/build/Jamfile.v2
--- libs/context/build/Jamfile.v2	2013-01-27 04:25:34.000000000 +0400
+++ libs/context/build/Jamfile.v2	2013-03-08 13:32:34.000000000 +0400
@@ -379,6 +379,16 @@
    ;
 
 alias asm_context_sources
+   : asm/make_i386_x86_64_sysv_macho_gas.S
+     asm/jump_i386_x86_64_sysv_macho_gas.S
+   : <address-model>32_64
+     <architecture>x86
+     <binary-format>mach-o
+     <target-os>darwin
+     <toolset>darwin
+   ;
+
+alias asm_context_sources
    : asm/make_x86_64_ms_pe_masm.asm
      asm/jump_x86_64_ms_pe_masm.asm
      dummy.cpp
diff -ruN libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S
--- libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S	1970-01-01 03:00:00.000000000 +0300
+++ libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S	2013-03-08 13:37:13.000000000 +0400
@@ -0,0 +1,17 @@
+/*
+            Copyright Sergue E. Leontiev <lse@cryptopro.ru> 2013 :)
+   Distributed under the Boost Software License, Version 1.0.
+      (See accompanying file LICENSE_1_0.txt or copy at
+          http://www.boost.org/LICENSE_1_0.txt)
+*/
+/*
+ * Stub file for universal binary
+ */
+
+#if defined(__i386__)
+    #include "jump_i386_sysv_macho_gas.S"
+#elif defined(__x86_64__)
+    #include "jump_x86_64_sysv_macho_gas.S"
+#else
+    #error "No arch's"
+#endif
diff -ruN libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S
--- libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S	1970-01-01 03:00:00.000000000 +0300
+++ libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S	2013-03-08 13:37:52.000000000 +0400
@@ -0,0 +1,17 @@
+/*
+            Copyright Sergue E. Leontiev <lse@cryptopro.ru> 2013 :)
+   Distributed under the Boost Software License, Version 1.0.
+      (See accompanying file LICENSE_1_0.txt or copy at
+          http://www.boost.org/LICENSE_1_0.txt)
+*/
+/*
+ * Stub file for universal binary
+ */
+
+#if defined(__i386__)
+    #include "make_i386_sysv_macho_gas.S"
+#elif defined(__x86_64__)
+    #include "make_x86_64_sysv_macho_gas.S"
+#else
+    #error "No arch's"
+#endif
