diff -ru boost/exception/exception.hpp /opt/local/include/boost/exception/exception.hpp
--- boost/exception/exception.hpp	2011-07-10 01:45:57.000000000 -0400
+++ /opt/local/include/boost/exception/exception.hpp	2011-07-15 13:12:46.000000000 -0400
@@ -310,6 +310,11 @@
     namespace
     exception_detail
         {
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility push (default) 
+# endif
+#endif
         template <class T>
         struct
         error_info_injector:
@@ -326,6 +331,11 @@
                 {
                 }
             };
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility pop 
+# endif
+#endif
 
         struct large_size { char c[256]; };
         large_size dispatch_boost_exception( exception const * );
@@ -373,6 +383,11 @@
     namespace
     exception_detail
         {
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility push (default) 
+# endif
+#endif
         class
         clone_base
             {
@@ -439,6 +454,11 @@
                 throw*this;
                 }
             };
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility pop 
+# endif
+#endif
         }
 
     template <class T>
diff -ru boost/thread/exceptions.hpp /opt/local/include/boost/thread/exceptions.hpp
--- boost/thread/exceptions.hpp	2009-10-19 05:18:13.000000000 -0400
+++ /opt/local/include/boost/thread/exceptions.hpp	2011-07-15 12:52:20.000000000 -0400
@@ -21,6 +21,12 @@
 
 #include <boost/config/abi_prefix.hpp>
 
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility push (default) 
+# endif
+#endif
+
 namespace boost
 {
 
@@ -177,6 +183,12 @@
 
 } // namespace boost
 
+#if defined(__GNUC__) 
+# if (__GNUC__ == 4 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 4) 
+#  pragma GCC visibility pop 
+# endif
+#endif
+
 #include <boost/config/abi_suffix.hpp>
 
 #endif
