diff -Naur ./tools/build/src/tools/clang-darwin.jam ./tools/build/src/tools/clang-darwin.jam
--- ./tools/build/src/tools/clang-darwin.jam	2021-06-08 12:28:19.000000000 -0400
+++ ./tools/build/src/tools/clang-darwin.jam	2021-06-08 12:29:08.000000000 -0400
@@ -54,7 +54,7 @@
 #   compile and link options allow you to specify addition command line options for each version
 rule init ( version ? :  command * : options * )
 {
-    command = [ common.get-invocation-command clang-darwin : clang++
+    command = [ common.get-invocation-command clang-darwin : __MACPORTS_CXX__
         : $(command) : /usr/bin /usr/local/bin ] ;
 
     # Determine the version
