--- Source/WebCore/platform/sql/SQLiteDatabase.cpp.orig	2026-05-11 14:00:40
+++ Source/WebCore/platform/sql/SQLiteDatabase.cpp	2026-05-11 14:00:45
@@ -70,23 +70,7 @@
         // aren't confident that it really is, and we still support ancient versions of SQLite. So
         // std::call_once is used to stay on the safe side. See bug #143245.
 
-#if OS(DARWIN)
-        int ret;
-        callOnMainThreadAndWait([&] {
-            // In the Network process, this function can be called on a background thread when
-            // creating WebKit::ResourceLoadStatisticsStore, which then races with
-            // WebKit::NetworkProcess::initializeNetworkProcess(). Since both of those calls query
-            // the Darwin user temp directory via confstr(), this should only be called from the
-            // main thread.
-            ret = sqlite3_initialize();
-        });
-#else
-        // On non-Darwin systems confstr() is MT-safe and it does not try to fiddle with environment
-        // variables, and it is better initialize directly. This is true at least on Linux with the
-        // supported C libraries (glibc, Musl, uClibc), the "big" BSDs (FreeBSD, NetBSD, OpenBSD),
-        // and the Android C library (Bionic) does not even provide confstr().
         int ret = sqlite3_initialize();
-#endif
 
         if (ret != SQLITE_OK) {
 #if SQLITE_VERSION_NUMBER >= 3007015
