--- Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp.orig	2026-04-29 22:00:00
+++ Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp	2026-04-29 22:00:00
@@ -1364,14 +1364,20 @@
         if (gst_structure_get_int(structure, key.utf8(), &value))
             return value;
     } else if constexpr(std::is_same_v<T, int64_t>) {
-        if (gst_structure_get_int64(structure, key.utf8(), &value))
+        gint64 gstValue;
+        if (gst_structure_get_int64(structure, key.utf8(), &gstValue)) {
+            value = gstValue;
             return value;
+        }
     } else if constexpr(std::is_same_v<T, unsigned>) {
         if (gst_structure_get_uint(structure, key.utf8(), &value))
             return value;
     } else if constexpr(std::is_same_v<T, uint64_t>) {
-        if (gst_structure_get_uint64(structure, key.utf8(), &value))
+        guint64 gstValue;
+        if (gst_structure_get_uint64(structure, key.utf8(), &gstValue)) {
+            value = gstValue;
             return value;
+        }
     } else if constexpr(std::is_same_v<T, double>) {
         if (gst_structure_get_double(structure, key.utf8(), &value))
             return value;
--- Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp.orig	2026-04-29 23:00:00
+++ Source/WebCore/platform/graphics/gstreamer/mse/WebKitMediaSourceGStreamer.cpp	2026-04-29 23:00:00
@@ -828,7 +828,7 @@
         double rate;
         GstFormat format;
         GstSeekType startType;
-        int64_t start;
+        gint64 start;
         gst_event_parse_seek(event.get(), &rate, &format, nullptr, &startType, &start, nullptr, nullptr);
         if (format != GST_FORMAT_TIME || startType != GST_SEEK_TYPE_SET) {
             GST_ERROR_OBJECT(element, "Rejecting unsupported seek event: %" GST_PTR_FORMAT, event.get());
--- Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp.orig	2026-04-29 23:00:00
+++ Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp	2026-04-29 23:00:00
@@ -644,7 +644,7 @@
             GstFormat format;
             GstSeekFlags flags;
             GstSeekType startType, stopType;
-            int64_t start, stop;
+            gint64 start, stop;
             gst_event_parse_seek(data->event.get(), &rate, &format, &flags, &startType, &start, &stopType, &stop);
             if (!gst_element_send_event(pipeline, data->event.leakRef()))
                 return;
@@ -1408,7 +1408,7 @@
     if (GST_STATE(m_pipeline.get()) < GST_STATE_PAUSED)
         return MediaTime::invalidTime();
 
-    int64_t duration = 0;
+    gint64 duration = 0;
     if (!gst_element_query_duration(m_pipeline.get(), GST_FORMAT_TIME, &duration) || !GST_CLOCK_TIME_IS_VALID(duration)) {
         GST_DEBUG_OBJECT(pipeline(), "Time duration query failed for %s", m_url.string().utf8().data());
         // https://www.w3.org/TR/2011/WD-html5-20110113/video.html#getting-media-metadata
@@ -4688,7 +4688,7 @@
     if (GST_BUFFER_PTS_IS_VALID(buffer)) {
         auto segment = gst_sample_get_segment(m_sample.get());
         RELEASE_ASSERT(segment);
-        uint64_t streamTime;
+        guint64 streamTime;
         if (int sign = gst_segment_to_stream_time_full(segment, GST_FORMAT_TIME, GST_BUFFER_PTS(buffer), &streamTime))
             metadata.mediaTime = sign * fromGstClockTime(streamTime).toDouble();
     }
--- Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp.orig	2026-04-29 23:00:00
+++ Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp	2026-04-29 23:00:00
@@ -174,7 +174,7 @@
 
 void MediaRecorderPrivateBackend::positionUpdated()
 {
-    int64_t position;
+    gint64 position;
     if (!gst_element_query_position(m_pipeline.get(), GST_FORMAT_TIME, &position)) {
         GST_LOG_OBJECT(m_pipeline.get(), "Could not query position");
         return;
