--- Source/JavaScriptCore/runtime/MachineContext.h.orig
+++ Source/JavaScriptCore/runtime/MachineContext.h
@@ -39,8 +39,10 @@
 
 template<typename T = void*> T stackPointer(const PlatformRegisters&);
 
-#if OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
+#if OS(DAWRIN) || OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
 template<typename T = void*> void setStackPointer(PlatformRegisters&, T);
+#endif // OS(DAWRIN) || OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
+#if OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
 template<typename T = void*> T framePointer(const PlatformRegisters&);
 template<typename T = void*> void setFramePointer(PlatformRegisters&, T);
 inline MacroAssemblerCodePtr<PlatformRegistersLRPtrTag> linkRegister(const PlatformRegisters&);
@@ -79,7 +81,7 @@
 #endif // HAVE(MACHINE_CONTEXT)
 #endif // OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
 
-#if OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
+#if OS(DARWIN) || OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
 
 #if !USE(PLATFORM_REGISTERS_WITH_PROFILE)
 static inline void*& stackPointerImpl(PlatformRegisters& regs)
@@ -156,7 +158,7 @@
 #endif
 }
 
-#else // not OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
+#else // not OS(DARWIN) || OS(WINDOWS) || HAVE(MACHINE_CONTEXT)
 
 template<typename T>
 inline T stackPointer(const PlatformRegisters& regs)
--- Source/JavaScriptCore/tools/SigillCrashAnalyzer.cpp.orig
+++ Source/JavaScriptCore/tools/SigillCrashAnalyzer.cpp
@@ -65,7 +65,7 @@
 #endif
 };
 
-#if OS(DARWIN)
+#if OS(DARWIN) && HAVE(MACHINE_CONTEXT)
 
 #if USE(OS_LOG)
 
