From f0109e5107e58760ec357fd1a9945553b0e2d8a3 Mon Sep 17 00:00:00 2001
From: Stephan Boyer <stepchowfun@users.github.com>
Date: Tue, 12 Mar 2024 20:39:49 -0700
Subject: [PATCH] Use `set -x` for shell scripts

---
 src/toastfile.rs                              | 12 +++----

diff --git a/src/toastfile.rs b/src/toastfile.rs
index 1b149ae1..a2a0fdb0 100644
--- a/src/toastfile.rs
+++ b/src/toastfile.rs
@@ -2009,13 +2009,13 @@ tasks:
             default: None,
             location: UnixPath::new(DEFAULT_LOCATION).to_owned(),
             user: DEFAULT_USER.to_owned(),
-            command_prefix: "set -euo pipefail".to_owned(),
+            command_prefix: "set -euxo pipefail".to_owned(),
             tasks,
         };

         assert_eq!(
             command(&toastfile, &toastfile.tasks["foo"]),
-            "set -euo pipefail".to_owned(),
+            "set -euxo pipefail".to_owned(),
         );
     }

@@ -2079,7 +2079,7 @@ tasks:
                 location: None,
                 user: None,
                 command: String::new(),
-                command_prefix: Some("set -euo pipefail".to_owned()),
+                command_prefix: Some("set -euxo pipefail".to_owned()),
                 extra_docker_arguments: vec![],
             },
         );
@@ -2095,7 +2095,7 @@ tasks:

         assert_eq!(
             command(&toastfile, &toastfile.tasks["foo"]),
-            "set -euo pipefail".to_owned(),
+            "set -euxo pipefail".to_owned(),
         );
     }

@@ -2119,7 +2119,7 @@ tasks:
                 location: None,
                 user: None,
                 command: "echo hello".to_owned(),
-                command_prefix: Some("set -euo pipefail".to_owned()),
+                command_prefix: Some("set -euxo pipefail".to_owned()),
                 extra_docker_arguments: vec![],
             },
         );
@@ -2135,7 +2135,7 @@ tasks:

         assert_eq!(
             command(&toastfile, &toastfile.tasks["foo"]),
-            "set -euo pipefail\necho hello".to_owned(),
+            "set -euxo pipefail\necho hello".to_owned(),
         );
     }
 }
