From a19b52ad583137b1acd6eef116a668512b68976f Mon Sep 17 00:00:00 2001
From: Stephan Boyer <stepchowfun@users.github.com>
Date: Thu, 28 Nov 2024 13:03:40 -0800
Subject: [PATCH] Update Rust to v1.83.0

---
 src/failure.rs           |  2 +-
 src/toastfile.rs         |  2 +-

diff --git a/src/failure.rs b/src/failure.rs
index bb01653..212eb9d 100644
--- a/src/failure.rs
+++ b/src/failure.rs
@@ -24,7 +24,7 @@ impl fmt::Display for Failure {
 }

 impl error::Error for Failure {
-    fn source<'a>(&'a self) -> Option<&(dyn error::Error + 'static)> {
+    fn source<'a>(&'a self) -> Option<&'a (dyn error::Error + 'static)> {
         match self {
             Self::System(_, source) => source.as_ref().map(|e| &**e),
             Self::User(_, source) => source.as_ref().map(|e| &**e),
diff --git a/src/toastfile.rs b/src/toastfile.rs
index a2a0fdb..79bf0d5 100644
--- a/src/toastfile.rs
+++ b/src/toastfile.rs
@@ -79,7 +79,7 @@ impl Display for MappingPath {

 struct MappingPathVisitor;

-impl<'de> serde::de::Visitor<'de> for MappingPathVisitor {
+impl serde::de::Visitor<'_> for MappingPathVisitor {
     type Value = MappingPath;

     fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
