--- lib/isc/include/isc/util.h.orig	2022-05-09 19:32:19.000000000 +1000
+++ lib/isc/include/isc/util.h	2022-05-20 02:36:59.000000000 +1000
@@ -35,6 +35,10 @@
 #define __has_attribute(x) 0
 #endif /* if !defined(__has_attribute) */
 
+#if !defined(__has_c_attribute)
+#define __has_c_attribute(x) 0
+#endif /* if !defined(__has_c_attribute) */
+
 #if !defined(__has_feature)
 #define __has_feature(x) 0
 #endif /* if !defined(__has_feature) */
@@ -61,7 +65,9 @@
 #define ISC_NONSTRING
 #endif /* __GNUC__ */
 
-#if __GNUC__ >= 7 || __has_attribute(fallthrough)
+#if __has_c_attribute(fallthrough)
+#define FALLTHROUGH [[fallthrough]]
+#elif !defined(__clang__) && (__GNUC__ >= 7 || __has_attribute(fallthrough))
 #define FALLTHROUGH __attribute__((fallthrough))
 #else
 /* clang-format off */
