--- configure.in.orig	2013-01-23 16:21:42.000000000 -0800
+++ configure.in	2013-01-23 16:23:14.000000000 -0800
@@ -385,17 +385,15 @@ AC_CHECK_FUNCS(setegid setresgid)
 # is reported to not support "static inline" (RT #1212).
 #
 AC_MSG_CHECKING(for static inline breakage)
-AC_TRY_COMPILE(, [
-		foo1();
-	}
-
+AC_TRY_COMPILE([
 	static inline int foo1() {
 		return 0;
 	}
 
 	static inline int foo2() {
 		return foo1();
-	],
+	}
+	], [foo1();],
 	[AC_MSG_RESULT(no)],
 	[AC_MSG_RESULT(yes)
 	 AC_DEFINE(inline, ,[Define to empty if your compiler does not support "static inline".])])
