--- configure.orig	2014-09-22 22:08:21.000000000 -0400
+++ configure	2014-09-22 22:08:45.000000000 -0400
@@ -14064,6 +14064,7 @@
 		# problems start to show up.
 		saved_libs="$LIBS"
 		for TRY_LIBS in \
+		    "-lgssapi -lkrb5 -lcrypto -lasn1 -lroken -lcom_err" \
 		    "-lgssapi_krb5" \
 		    "-lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err" \
 		    "-lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lresolv" \
