--- src/fsearch_query_parser.c.orig	2023-08-05 16:56:19
+++ src/fsearch_query_parser.c	2024-12-13 16:32:02
@@ -317,7 +317,7 @@
                                   flags,
                                   "date-modified",
                                   fsearch_query_node_new_date_modified,
-                                  fsearch_date_time_parse_interval);
+                                  (FsearchQueryIntegerParserFunc*)fsearch_date_time_parse_interval);
 }
 
 static GList *
