--- src/BulletJoints.cpp.org	2014-04-04 11:37:15.000000000 +0200
+++ src/BulletJoints.cpp	2018-10-24 10:45:49.000000000 +0200
@@ -565,10 +565,20 @@
     doubleHingeJointParameters->haveSuspensionErrorCorrection() )
     updateSuspension ( *doubleHingeJointParameters );
 
+#if BT_BULLET_VERSION < 283
   // stopBounce1 - restitution
   if ( doubleHingeJointParameters->haveStopBounce1() )
     doubleHingeJoint->getRotationalLimitMotor(2)->m_limitSoftness= btScalar ( doubleHingeJointParameters->getStopBounce1() );
-
+#else
+#ifdef _MSC_VER
+#pragma message("Bullet version is equal to or above 2.83. stopBounce1 field of DoubleAxisHingeJoint will not be used.")
+#else
+#ifdef __GNUC__
+#warning Bullet version is equal to or above 2.83. stopBounce1 field of DoubleAxisHingeJoint will not be used.
+#endif
+#endif
+#endif
+  
   // stopConstantForceMix1
   if ( doubleHingeJointParameters->haveStopConstantForceMix1() )
     doubleHingeJoint->getRotationalLimitMotor(2)->m_stopCFM= btScalar ( doubleHingeJointParameters->getStopConstantForceMix1() );
