--- configure.old	2015-03-04 16:56:11.000000000 +1300
+++ configure	2015-03-04 17:08:33.000000000 +1300
@@ -31054,7 +31054,10 @@
 if test "$have_gtk" = "yes" ; then
 	# If we have GTK then add flags for it.
 
-	CPPFLAGS="-DGDK_PIXBUF_DISABLE_DEPRECATED $CPPFLAGS"
+	# 2015-03-04 - disabled to fix SegFault, see
+	# https://trac.macports.org/ticket/46850
+	# https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=195980#c45
+	#CPPFLAGS="-DGDK_PIXBUF_DISABLE_DEPRECATED $CPPFLAGS"
 	CPPFLAGS="-DGDK_DISABLE_DEPRECATED $CPPFLAGS"
 	if test \( $gtk_config_major_version -eq 3 -a $gtk_config_minor_version -ge 10 \) ; then
 		## Allow use of deprecated & disable deprecated warnings if Gtk >= 3.10;
