--- t/perldoc.t.orig	2016-03-30 03:50:11.000000000 -0700
+++ t/perldoc.t	2016-03-30 03:51:06.000000000 -0700
@@ -30,7 +30,7 @@
 
 my $X = $^X =~ m/\s/ ? qq{"$^X"} : $^X;
 my $Mblib = Mblib();
-my $perldoc = File::Spec->catfile($Config{installbin}, 'perldoc');
+my $perldoc = File::Spec->catfile($Config{installbin}, 'perldoc-@@MP_PERL5_MAJOR@@');
 my $perlcc = "$X $Mblib blib/script/perlcc";
 $perlcc .= " -Wb=-fno-fold,-fno-warnings" if $] > 5.013;
 $perlcc .= " -UB -uFile::Spec -uCwd";
