--- dolphin/src/CMakeLists.txt.orig	2014-04-03 16:03:47.000000000 +0900
+++ dolphin/src/CMakeLists.txt	2014-04-24 23:41:31.000000000 +0900
@@ -1,11 +1,11 @@
-macro_optional_find_package(Baloo)
+#macro_optional_find_package(Baloo)
 set_package_properties(Baloo PROPERTIES DESCRIPTION "Baloo Core libraries"
                        URL "http://www.kde.org"
                        TYPE OPTIONAL
                        PURPOSE "For adding desktop-wide search and tagging support to dolphin"
                       )
 
-macro_optional_find_package(BalooWidgets)
+#macro_optional_find_package(BalooWidgets)
 set_package_properties(BalooWidgets PROPERTIES DESCRIPTION "Baloos Widgets"
                        URL "http://www.kde.org"
                        TYPE OPTIONAL
