--- texk/xdvik/configure.orig~	2009-01-26 08:10:11.000000000 -0700
+++ texk/xdvik/configure	2009-01-26 08:11:30.000000000 -0700
@@ -4222,7 +4222,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lt1 -lz -lm $LIBS"
+LIBS="-lt1 -lz -lc -lm $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 4228 "configure"
 #include "confdefs.h"
@@ -4259,7 +4259,7 @@
 
   fi
 
-  LIBS="-lt1 -lm $LIBS"
+  LIBS="-lt1 -lc -lm $LIBS"
   if test "$with_system_t1lib" = yes; then
       echo $ac_n "checking whether installed T1 lib is >= v 5.0.0 ""... $ac_c" 1>&6
 echo "configure:4266: checking whether installed T1 lib is >= v 5.0.0 " >&5
@@ -5978,7 +5978,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lm  $LIBS"
+LIBS="-lc -lm  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 5984 "configure"
 #include "confdefs.h"
@@ -6008,7 +6008,7 @@
 #define $ac_tr_lib 1
 EOF
 
-  LIBS="-lm $LIBS"
+  LIBS="-lc -lm $LIBS"
 
 else
   echo "$ac_t""no" 1>&6
