diff -Naurp krb5-1.18.2.orig/src/aclocal.m4 krb5-1.18.2/src/aclocal.m4
--- aclocal.m4	2020-06-09 22:39:34.000000000 -0700
+++ aclocal.m4	2020-06-10 12:38:04.000000000 -0700
@@ -1599,7 +1599,9 @@ fi
 a=no
 b=no
 # blindly assume we have 'unlink'...
-AC_TRY_RUN([void foo1() __attribute__((constructor));
+AC_TRY_RUN([
+#include <unistd.h>
+void foo1() __attribute__((constructor));
 void foo1() { unlink("conftest.1"); }
 void foo2() __attribute__((destructor));
 void foo2() { unlink("conftest.2"); }
