--- ./display/GPI/Matplotlib_GPI.py	2019-07-30 09:55:58.000000000 -0500
+++ ./display/GPI/Matplotlib_GPI.py	2019-08-09 10:09:15.000000000 -0500
@@ -104,7 +104,7 @@
                 if tooltip_text is not None:
                     a.setToolTip(tooltip_text)

-        self.buttons = {}
+        #self.buttons = {}

         # Add the x,y location widget at the right side of the toolbar
         # The stretch factor is 1 which means any resizing of the toolbar
@@ -120,7 +120,7 @@
             labelAction.setVisible(True)

         # reference holder for subplots_adjust window
-        self.adj_window = None
+        #self.adj_window = None

 ###############################################################################
 # -*- coding: utf-8 -*-
