--- spiral/bnispiralgmn.c.orig	2018-05-25 10:46:18.000000000 -0500
+++ spiral/bnispiralgmn.c	2018-05-25 10:46:42.000000000 -0500
@@ -892,7 +892,7 @@
     gradtrap(gdwell, maxgrad, Am, Ac, N1c, am, n1m, n2m);
     gradtrap(gdwell, maxgrad, Ar, Ac, N1c, ar, n1r, n2r);
 
-    if (am < 0 || ar < 0 || n1m < 0 || n2m < 0 || n1r < 0 || n2r < 0)
+    if (*am < 0 || *ar < 0 || *n1m < 0 || *n2m < 0 || *n1r < 0 || *n2r < 0)
     {
       printf("LINE #%d: Function solution_analytical failed\n", __LINE__);
       return 0;
