Index: fftw-2.1.5/configure.in
===================================================================
--- fftw-2.1.5.orig/configure.in	2003-03-24 08:01:13.000000000 +0100
+++ fftw-2.1.5/configure.in	2011-11-29 02:16:51.531836400 +0100
@@ -27,7 +27,7 @@
 
 AC_ARG_ENABLE(float, [  --enable-float          compile fftw for single precision], enable_float=$enableval, enable_float=no)
 if test "$enable_float" = "yes"; then
-	AC_DEFINE(FFTW_ENABLE_FLOAT)
+	AC_DEFINE([FFTW_ENABLE_FLOAT],[],[desc])
 fi
 
 FFTW_PREFIX=""
@@ -56,27 +56,27 @@
 
 AC_ARG_ENABLE(i386-hacks, [  --enable-i386-hacks     enable gcc/x86 specific performance hacks], ok=$enableval, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(FFTW_ENABLE_I386_HACKS)
+	AC_DEFINE([FFTW_ENABLE_I386_HACKS],[],[desc])
 fi
 
 AC_ARG_ENABLE(pentium-timer, [  --enable-pentium-timer  enable high resolution Pentium timer], ok=$enableval, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(FFTW_ENABLE_PENTIUM_TIMER)
+	AC_DEFINE([FFTW_ENABLE_PENTIUM_TIMER],[],[desc])
 fi
 
 AC_ARG_ENABLE(debug, [  --enable-debug          compile fftw with extra runtime checks for debugging], ok=$enableval, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(FFTW_DEBUG)
+	AC_DEFINE([FFTW_DEBUG],[],[desc])
 fi
 
 AC_ARG_ENABLE(debug-alignment, [  --enable-debug-alignment          enable runtime checks for alignment on x86], ok=$enableval, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(FFTW_DEBUG_ALIGNMENT)
+	AC_DEFINE([FFTW_DEBUG_ALIGNMENT],[],[desc])
 fi
 
 AC_ARG_ENABLE(vec-recurse, [  --enable-vec-recurse    enable experimental performance hack], ok=$enableval, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(FFTW_ENABLE_VECTOR_RECURSE)
+	AC_DEFINE([FFTW_ENABLE_VECTOR_RECURSE],[],[desc])
 fi
 
 dnl -----------------------------------------------------------------------
@@ -115,7 +115,7 @@
 dnl -----------------------------------------------------------------------
 
 AC_SUBST(SHARED_VERSION_INFO)
-AC_DEFINE_UNQUOTED(FFTW_VERSION, "$VERSION")
+AC_DEFINE_UNQUOTED([FFTW_VERSION], "$VERSION",[Version])
 
 # Get the version number that will be appended to shared libraries:
 SHARED_VERSION=`echo $SHARED_VERSION_INFO | awk -F':' '{ print $1 "." $3 "." $2 }'`
@@ -123,7 +123,7 @@
 
 ACX_PROG_CC_MAXOPT
 
-ACX_GCC_ALIGNS_STACK(AC_DEFINE(FFTW_GCC_ALIGNS_STACK), [
+ACX_GCC_ALIGNS_STACK(AC_DEFINE([FFTW_GCC_ALIGNS_STACK],[],[desc]), [
 	if test "$enable_i386_hacks" = yes; then
 	if test "${acx_gcc_stack_align_bug-no}" = yes; then
 		# we are using a gcc with a stack alignment bug, and we should
@@ -183,7 +183,7 @@
 AC_TRY_LINK([#include <math.h>
 ], if (!isnan(3.14159)) isnan(2.7183);, ok=yes, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(HAVE_ISNAN)
+	AC_DEFINE([HAVE_ISNAN],[],[desc])
 fi
 AC_MSG_RESULT(${ok})
 
@@ -196,7 +196,7 @@
 #endif
 ], [hrtime_t foobar;], ok=yes, ok=no)
 if test "$ok" = "yes"; then
-	AC_DEFINE(HAVE_HRTIME_T)
+	AC_DEFINE([HAVE_HRTIME_T],[],[desc])
 fi
 AC_MSG_RESULT(${ok})
 
@@ -205,7 +205,7 @@
 
 AC_ARG_ENABLE(unsafe-mulmod, [  --enable-unsafe-mulmod  risk overflow for large prime sizes], enable_unsafe_mulmod=$enableval, enable_unsafe_mulmod=no)
 if test "$enable_unsafe_mulmod" = "yes"; then
-        AC_DEFINE(FFTW_ENABLE_UNSAFE_MULMOD)
+        AC_DEFINE([FFTW_ENABLE_UNSAFE_MULMOD],[],[desc])
 fi
 
 
@@ -243,7 +243,7 @@
 		if test -z "$THREADLIBS"; then
 			AC_MSG_ERROR([don't know how to enable OpenMP])
 		fi
-		AC_DEFINE(FFTW_USING_OPENMP_THREADS)
+		AC_DEFINE([FFTW_USING_OPENMP_THREADS],[],[desc])
 	fi
 	if test "$with_sgimp"x != nox; then
 		AC_MSG_CHECKING(how to enable SGI MP)
@@ -261,7 +261,7 @@
 		if test -z "$THREADLIBS"; then 
 			AC_MSG_ERROR([don't know how to enable SGI MP])
 		fi
-		AC_DEFINE(FFTW_USING_SGIMP_THREADS)
+		AC_DEFINE([FFTW_USING_SGIMP_THREADS],[],[desc])
 	fi
 
 	# POSIX threads, the default choice:
@@ -270,38 +270,38 @@
 		ACX_PTHREAD([THREADLIBS="$PTHREAD_LIBS "
 	                     CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 	                     CC="$PTHREAD_CC"
-	                     AC_DEFINE(FFTW_USING_POSIX_THREADS)])
+	                     AC_DEFINE([FFTW_USING_POSIX_THREADS],[],[desc])])
 	fi
 	# Solaris threads:
 	if test -z "$THREADLIBS"; then
 		AC_CHECK_LIB(thread, thr_create,
 		             [THREADLIBS="-lthread"
-                	      AC_DEFINE(FFTW_USING_SOLARIS_THREADS)])
+                	      AC_DEFINE([FFTW_USING_SOLARIS_THREADS],[],[desc])])
 	fi
 	# Mach C threads:
 	if test -z "$THREADLIBS"; then
 		AC_CHECK_FUNC(cthread_fork,
 		             [THREADLIBS=" "
-                	      AC_DEFINE(FFTW_USING_MACH_THREADS)])
+                	      AC_DEFINE([FFTW_USING_MACH_THREADS],[],[desc])])
 		AC_CHECK_HEADERS(mach/cthreads.h cthreads.h cthread.h)
 	fi
 	if test -z "$THREADLIBS"; then
 		AC_CHECK_LIB(cthread, cthread_fork,
 		             [THREADLIBS="-lcthread"
-                	      AC_DEFINE(FFTW_USING_MACH_THREADS)])
+                	      AC_DEFINE([FFTW_USING_MACH_THREADS],[],[desc])])
 		AC_CHECK_HEADERS(mach/cthreads.h cthreads.h cthread.h)
 	fi
 	if test -z "$THREADLIBS"; then
 		AC_CHECK_LIB(cthreads, cthread_fork,
 		             [THREADLIBS="-lcthreads"
-                	      AC_DEFINE(FFTW_USING_MACH_THREADS)])
+                	      AC_DEFINE([FFTW_USING_MACH_THREADS],[],[desc])])
 		AC_CHECK_HEADERS(mach/cthreads.h cthreads.h cthread.h)
 	fi
 	# BeOS threads:
 	if test -z "$THREADLIBS"; then
 		AC_CHECK_FUNC(spawn_thread,
 		             [THREADLIBS=" "
-                	      AC_DEFINE(FFTW_USING_BEOS_THREADS)])
+                	      AC_DEFINE([FFTW_USING_BEOS_THREADS],[],[desc])])
 	fi
 	if test -z "$THREADLIBS"; then
 		AC_MSG_ERROR(couldn't find threads library for --enable-threads)
@@ -334,7 +334,7 @@
 	CC="$MPICC"
 	ok=yes
 	AC_TRY_LINK([#include <mpi.h>
-	], [MPI_Comm_f2c(0);], [AC_DEFINE(HAVE_MPI_COMM_F2C)], [ok=no])
+	], [MPI_Comm_f2c(0);], [AC_DEFINE([HAVE_MPI_COMM_F2C],[],[desc])], [ok=no])
 	AC_MSG_RESULT($ok)
 	CC="$save_CC"
 else
