--- configure.in.orig	2003-01-19 11:19:41.000000000 -0600
+++ configure.in	2013-01-15 22:30:09.000000000 -0600
@@ -4,10 +4,10 @@
 dnl     generate the file "configure", which is run during the build
 dnl     to configure the system for the local environment.
 
-AC_INIT
-AM_INIT_AUTOMAKE(fcgi, 2.4.0)
+AC_INIT([fcgi],[2.4.0])
+AM_INIT_AUTOMAKE([-Wall foreign])
 
-AM_CONFIG_HEADER(fcgi_config.h)
+AC_CONFIG_HEADERS([fcgi_config.h])
 
 AC_PROG_CC
 AC_PROG_CPP 
