--- libfcgi/Makefile.am.orig	2001-12-21 20:16:20.000000000 -0700
+++ libfcgi/Makefile.am	2009-01-18 23:57:40.000000000 -0700
@@ -18,10 +18,11 @@
                      os_@SYSTEM@.c
 libfcgi_la_CC      = @PTHREAD_CC@
 libfcgi_la_CFLAGS  = @PTHREAD_CFLAGS@
+libfcgi_la_LDFLAGS = -version-number $(subst .,:,@VERSION@)
 
 libfcgi___la_SOURCES = $(INCLUDE_FILES)       \
                        $(INCLUDEDIR)/fcgio.h  \
                        fcgio.cpp
 libfcgi___la_CFLAGS  = @PTHREAD_CFLAGS@
-libfcgi___la_LDFLAGS = -lfcgi -rpath @libdir@
+libfcgi___la_LDFLAGS = -lfcgi -rpath @libdir@ -version-number $(subst .,:,@VERSION@)
 
