--- Makefile.in.orig	2011-12-10 06:35:15.000000000 -0600
+++ Makefile.in	2012-01-17 17:33:35.000000000 -0600
@@ -835,7 +835,7 @@
 	$(CC) $(COMPAT) $(CFLAGS) $(XFLAGS) -c ../unix/xpmputty.c
 
 version.o: FORCE
-	if test -z "$(VER)" && (cd ..; md5sum -c manifest); then \
+	if test -z "$(VER)"; then \
 		$(CC) $(COMPAT) $(XFLAGS) $(CFLAGS) `cat ../version.def` -c ../version.c; \
 	else \
 		$(CC) $(COMPAT) $(XFLAGS) $(CFLAGS) $(VER) -c ../version.c; \
