diff -r 07ebf629df5f pure-lilv/Makefile
--- Makefile.orig	Fri Sep 26 07:46:15 2014 +0200
+++ Makefile	Sun Sep 28 11:15:14 2014 +0200
@@ -29,7 +29,7 @@
 all: $(objects)
 
 lilv$(DLL): lilv.c lv2_evbuf.c symap.c
-	gcc $(shared) $(dllname) -o $@ $(MOD_CFLAGS) -std=c99 $^ $(MOD_LDFLAGS) $(LIBS)
+	$(CC) $(shared) $(dllname) -o $@ $(MOD_CFLAGS) -std=c99 $^ $(MOD_LDFLAGS) $(LIBS)
 
 clean:
 	rm -rf *$(DLL)*
