diff --git a/configure b/configure
index f282bb34d91..6d90a87985a 100755
--- a/configure
+++ b/configure
@@ -1544,6 +1544,7 @@ enable_winegstreamer
 enable_winehid_sys
 enable_winemac_drv
 enable_winemapi
+enable_winemetal
 enable_wineoss_drv
 enable_wineps_drv
 enable_winepulse_drv
@@ -23213,6 +23214,7 @@ wine_fn_config_makefile dlls/winegstreamer enable_winegstreamer
 wine_fn_config_makefile dlls/winehid.sys enable_winehid_sys
 wine_fn_config_makefile dlls/winemac.drv enable_winemac_drv
 wine_fn_config_makefile dlls/winemapi enable_winemapi
+wine_fn_config_makefile dlls/winemetal enable_winemetal
 wine_fn_config_makefile dlls/wineoss.drv enable_wineoss_drv
 wine_fn_config_makefile dlls/wineps.drv enable_wineps_drv
 wine_fn_config_makefile dlls/wineps16.drv16 enable_win16
diff --git a/configure.ac b/configure.ac
index 624e6ce6b6b..8514c579a5d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3358,6 +3358,7 @@ WINE_CONFIG_MAKEFILE(dlls/winegstreamer)
 WINE_CONFIG_MAKEFILE(dlls/winehid.sys)
 WINE_CONFIG_MAKEFILE(dlls/winemac.drv)
 WINE_CONFIG_MAKEFILE(dlls/winemapi)
+WINE_CONFIG_MAKEFILE(dlls/winemetal)
 WINE_CONFIG_MAKEFILE(dlls/wineoss.drv)
 WINE_CONFIG_MAKEFILE(dlls/wineps.drv)
 WINE_CONFIG_MAKEFILE(dlls/wineps16.drv16)
diff --git a/dlls/winemetal/Makefile.in b/dlls/winemetal/Makefile.in
new file mode 100644
index 00000000000..5f8ab6f673a
--- /dev/null
+++ b/dlls/winemetal/Makefile.in
@@ -0,0 +1 @@
+MODULE = winemetal.dll
diff --git a/dlls/winemetal/winemetal.spec b/dlls/winemetal/winemetal.spec
new file mode 100644
index 00000000000..9ccfb1a000b
--- /dev/null
+++ b/dlls/winemetal/winemetal.spec
@@ -0,0 +1 @@
+@ stub WineMetalEntry
