http://svn.php.net/viewvc?view=revision&revision=297236
--- esmtp.c	2010/03/31 20:39:12	297235
+++ esmtp.c	2010/03/31 20:39:48	297236
@@ -141,7 +141,7 @@
 }
 /* }}} */
 
-function_entry esmtp_functions[] = {
+zend_function_entry esmtp_functions[] = {
     ZEND_FALIAS(esmtp_version,                  smtp_version,             NULL)
     ZEND_FALIAS(esmtp_errno,                    smtp_errno,               NULL)
     ZEND_FALIAS(esmtp_strerror,                 smtp_strerror,            NULL)
