http://svn.php.net/viewvc?view=revision&revision=309844
--- esmtp.c	2011/03/30 10:31:59	309843
+++ esmtp.c	2011/03/30 11:32:09	309844
@@ -35,6 +35,10 @@
 #include <auth-client.h>
 #include <libesmtp.h>
 
+#ifndef Z_ADDREF_P
+# define Z_ADDREF_P ZVAL_ADDREF
+#endif
+
 static const int do_serialize = 1;
 
 static int le_esmtp_session;
@@ -326,10 +330,10 @@
     MAKE_STD_ZVAL(z_data);
     array_init(z_data);
 
-    ZVAL_ADDREF(cb);
+    Z_ADDREF_P(cb);
     add_assoc_zval_ex(z_data, "funcname", sizeof("funcname"), cb);
     if (arg) {
-        ZVAL_ADDREF(arg);
+        Z_ADDREF_P(arg);
         add_assoc_zval_ex(z_data, "userdata", sizeof("userdata"), arg);
     }
 
@@ -758,10 +762,10 @@
     MAKE_STD_ZVAL(z_data);
     array_init(z_data);
 
-    ZVAL_ADDREF(cb);
+    Z_ADDREF_P(cb);
     add_assoc_zval_ex(z_data, "funcname", sizeof("funcname"), cb);
     if (arg) {
-        ZVAL_ADDREF(arg);
+        Z_ADDREF_P(arg);
         add_assoc_zval_ex(z_data, "userdata", sizeof("userdata"), arg);
     }
 
@@ -841,11 +845,11 @@
     MAKE_STD_ZVAL(z_data);
     array_init(z_data);
 
-    ZVAL_ADDREF(cb);
+    Z_ADDREF_P(cb);
     add_assoc_zval_ex(z_data, "funcname", sizeof("funcname"), cb);
     add_property_zval(getThis(), "header_regex_funcname", cb);
     if (arg) {
-        ZVAL_ADDREF(arg);
+        Z_ADDREF_P(arg);
         add_assoc_zval_ex(z_data, "userdata", sizeof("userdata"), arg);
         add_property_zval(getThis(), "header_regex_userdata", arg);
     }
@@ -1189,10 +1193,10 @@
     MAKE_STD_ZVAL(z_data);
     array_init(z_data);
 
-    ZVAL_ADDREF(cb);
+    Z_ADDREF_P(cb);
     add_assoc_zval_ex(z_data, "funcname", sizeof("funcname"), cb);
     if (arg) {
-        ZVAL_ADDREF(arg);
+        Z_ADDREF_P(arg);
         add_assoc_zval_ex(z_data, "userdata", sizeof("userdata"), arg);
     }
 
