--- DESCRIPTION	2023-10-25 07:30:00.000000000 +0800
+++ DESCRIPTION	2023-11-05 23:11:22.000000000 +0800
@@ -13,8 +13,7 @@
 License: GPL-2
 Imports: matrixStats, limma, parallel, shiny, qvalue, grDevices, stats,
         MultiAssayExperiment, graphics
-Suggests: knitr, yaml, testthat (>= 3.0.0), rmarkdown, BiocStyle,
-        clusterProfiler
+Suggests: knitr, yaml, testthat (>= 3.0.0), rmarkdown, BiocStyle
 LinkingTo: Rcpp
 biocViews: Clustering, Annotation, PrincipalComponent,
         DifferentialExpression, Visualization, Proteomics, Metabolomics
