--- Makefile.orig	2003-12-08 20:28:02.000000000 +1100
+++ Makefile	2012-08-29 19:59:24.000000000 +1000
@@ -3,9 +3,9 @@
 #
 
 # use always:
-DEFINES :=  
+#DEFINES :=  
 LIBS    := -lm 
-INCLUDES :=
+#INCLUDES :=
 
 ### enable ppm support ###
 DEFINES  += -D_HAVE_LIB_PPM_
@@ -70,19 +70,19 @@
 # Target
 subtitle2pgm: subtitle2pgm.o spudec.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) $^ -o $@  
+	@$(CC) $(LDFLAGS) $(LIBS) $^ -o $@
 
 subtitle2vobsub: subtitle2vobsub.o vobsub.o 
 	@echo "Linking $@"
-	@$(CC) $(LIBS) $^ -o $@  
+	@$(CC) $(LDFLAGS) $(LIBS) $^ -o $@
 
 srttool: srttool.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) -g $^ -o $@  
+	@$(CC) $(LDFLAGS) $(LIBS) -g $^ -o $@
 
 vobsub2pgm: vobsub2pgm.o vobsub.o spudec.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) -g $^ -o $@  
+	@$(CC) $(LDFLAGS) $(LIBS) -g $^ -o $@
 
 .PHONY: clean dist rpm
 clean:
