--- gegl.pc.in.orig	2023-08-25 13:07:46
+++ gegl.pc.in	2023-08-25 13:08:48
@@ -7,6 +7,6 @@
 Name: GEGL
 Description: Generic Graphics Library 
 Version: @GEGL_REAL_VERSION@
-Requires: @GLIB_PACKAGES@ babl
+Requires: @GLIB_PACKAGES@ babl-0.1
 Libs: -L${libdir} -l@PACKAGE_NAME@-@GEGL_API_VERSION@ -l@PACKAGE_NAME@-npd-@GEGL_API_VERSION@ @MATH_LIB@
 Cflags: -I${includedir}/@PACKAGE_NAME@-@GEGL_API_VERSION@
--- seamless-clone/gegl-sc.pc.in.orig	2023-08-25 13:46:53
+++ seamless-clone/gegl-sc.pc.in	2023-08-25 13:48:21
@@ -6,6 +6,6 @@
 Name: GEGL Seamless Cloning
 Description: Seamless Cloning Library Based on GEGL
 Version: @GEGL_REAL_VERSION@
-Requires: @GLIB_PACKAGES@ babl @P2TC_PACKAGES@ @PACKAGE_NAME@-@GEGL_API_VERSION@
+Requires: @GLIB_PACKAGES@ babl-0.1 @P2TC_PACKAGES@ @PACKAGE_NAME@-@GEGL_API_VERSION@
 Libs: -L${libdir} -l@PACKAGE_NAME@-sc-@GEGL_API_VERSION@
 Cflags: -I${includedir}/@PACKAGE_NAME@-@GEGL_API_VERSION@/sc
