--- contrib/python-bindings/source/CMakeLists.txt.orig	2019-05-26 19:07:17.000000000 -0700
+++ contrib/python-bindings/source/CMakeLists.txt	2019-06-21 08:14:22.000000000 -0700
@@ -35,7 +35,7 @@
 LIST(REMOVE_ITEM CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
 SET(Boost_FOUND)
 # Use the low level _FIND_PACKAGE function instead of our wrapper
-_FIND_PACKAGE(Boost 1.54 COMPONENTS python)
+_FIND_PACKAGE(Boost 1.54 COMPONENTS python3)
 LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules/)
 
 IF(NOT Boost_FOUND)
