--- Makefile.orig	2022-07-30 19:42:53.000000000 +0800
+++ Makefile	2022-12-04 05:34:57.000000000 +0800
@@ -1,6 +1,6 @@
 # Inform the location to install the modules
-LUAPATH  ?= /usr/share/lua/5.1
-LUACPATH ?= /usr/lib/lua/5.1
+# LUAPATH  ?= /usr/share/lua/5.1
+# LUACPATH ?= /usr/lib/lua/5.1
 
 # Compile with build-in LuaSocket's help files.
 # Comment this lines if you will link with non-internal LuaSocket's help files
@@ -10,13 +10,13 @@
 
 # Edit the lines below to inform new path, if necessary.
 # Path below points to internal LuaSocket's help files.
-INC_PATH ?= -I/usr/include
-LIB_PATH ?= -L/usr/lib
+INC_PATH ?= -I@PREFIX@/include
+LIB_PATH ?= -L@PREFIX@/lib
 INCDIR    = -I. $(INC_PATH)
-LIBDIR    = -L./luasocket $(LIB_PATH)
+LIBDIR    = -L./luasocket -L@SSLDIR@ $(LIB_PATH)
 
 # For Mac OS X: set the system version
-MACOSX_VERSION=10.11
+MACOSX_VERSION=@MACOS_VER@
 
 #----------------------
 # Do not edit this part

--- src/Makefile.orig	2022-07-30 19:42:53.000000000 +0800
+++ src/Makefile	2022-11-26 17:27:12.000000000 +0800
@@ -24,8 +24,7 @@
 MAC_LDFLAGS=-bundle -undefined dynamic_lookup $(LIBDIR)
 
 INSTALL  = install
-CC      ?= cc
-CCLD      ?= $(MYENV) $(CC)
+CCLD    ?= $(MYENV) $(CC)
 CFLAGS  += $(MYCFLAGS)
 LDFLAGS += $(MYLDFLAGS)
 
