--- Makefile	Mon Aug 18 07:43:45 2003
+++ Makefile.new	Tue Jan  6 15:16:47 2004
@@ -38,8 +38,8 @@
 	$(CC) $(CFLAGS) $(DEFS) -o bcwipe wipe.c $(BC_OBJ) 
 
 install:
-	$(INSTALL) -m 644 bcwipe.1      $(root)/$(MAN_PATH)/man1/
-	$(INSTALL) -o root -m 755 -s bcwipe   $(root)/usr/bin/
+	$(INSTALL) -m 644 bcwipe.1      $(DESTDIR)/opt/local/share/man/man1/
+	$(INSTALL) -m 755 -s bcwipe   $(DESTDIR)/opt/local/bin/
 
 clean:
 	rm -f *.o *.ver *~ $(TARGETS)
