--- Makefile	2008-07-03 07:07:55.000000000 +0200
+++ Makefile.new	2008-10-25 14:04:33.000000000 +0200
@@ -66,9 +66,8 @@
 	$(CC) $(CFLAGS) $(DEFS) -o bcwipe wipe.c $(BC_OBJ) 
 
 install: bcwipe
-	./install.sh $(root)
-# $(INSTALL) -m 644 bcwipe.1      $(root)/$(MAN_PATH)/man1/
-# $(INSTALL) -o root -m 755 -s bcwipe   $(root)/usr/bin/
+	$(INSTALL) -m 644 bcwipe.1      $(DESTDIR)@@prefix@@/share/man/man1/
+	$(INSTALL) -o root -m 755 -s bcwipe   $(DESTDIR)@@prefix@@/bin/
 
 clean:
 	rm -f *.o *.ver *~ $(TARGETS)
