--- Makefile	2007-06-13 03:23:32.000000000 +0000
+++ Makefile.new	2007-06-13 03:28:05.000000000 +0000
@@ -68,9 +68,8 @@
 	$(CC) $(CFLAGS) $(DEFS) -o bcwipe wipe.c $(BC_OBJ) 
 
 install:
-	./install.sh $(root)
-# $(INSTALL) -m 644 bcwipe.1      $(root)/$(MAN_PATH)/man1/
-# $(INSTALL) -o root -m 755 -s bcwipe   $(root)/usr/bin/
+	$(INSTALL) -m 644 bcwipe.1      $(DESTDIR)@@prefix@@/share/man/man1/
+	$(INSTALL) -o root -m 755 -s bcwipe   $(DESTDIR)@@prefix@@/bin/
 
 clean:
 	rm -f *.o *.ver *~ $(TARGETS)
