--- ./build.sh	2023-05-29 17:29:20.000000000 -0400
+++ ./build.sh	2023-05-29 17:34:27.000000000 -0400
@@ -59,7 +59,7 @@
 	shift
 done
 
-configdir="${destdir}etc/maddy"
+configdir="${destdir}${prefix}/etc/maddy"
 
 if [ "$1" = "" ]; then
 	print_help
@@ -122,7 +122,19 @@
 			-o "${builddir}/maddy" ${GOFLAGS} ./cmd/maddy
 	else
 		echo "-- Building main server executable..." >&2
-		go build -tags "$tags" -trimpath -ldflags="-X \"github.com/foxcpp/maddy.Version=${version}\"" -o "${builddir}/maddy" ${GOFLAGS} ./cmd/maddy
+		go build \
+                    -tags "$tags" \
+                    -trimpath \
+                    -ldflags="
+                        -X \"github.com/foxcpp/maddy.ConfigDirectory=${prefix}/etc/maddy\"
+                        -X \"github.com/foxcpp/maddy.DefaultStateDirectory=${prefix}/var/lib/maddy\"
+                        -X \"github.com/foxcpp/maddy.DefaultRuntimeDirectory=${prefix}/var/run/maddy\"
+                        -X \"github.com/foxcpp/maddy.DefaultLibexecDirectory=${prefix}/libexec/maddy\"
+                        -X \"github.com/foxcpp/maddy.Version=${version}\"
+                    " \
+                    -o "${builddir}/maddy" \
+                    ${GOFLAGS} \
+                    ./cmd/maddy
 	fi
 
 		build_man_pages
