--- SConstruct.orig	2019-08-24 20:16:35.000000000 +0800
+++ SConstruct	2022-06-03 22:38:38.000000000 +0800
@@ -6,12 +6,12 @@
 EnsurePythonVersion(2,7)
 
 # set defaults
-PREFIX='/usr/local'
+PREFIX='@PREFIX@'
 VERSION='2.8.1b29'
 EFENCE=False
 CLIENTS=True
 
-env = Environment(CPPPATH='#/include', LIBPATH=['/usr/lib','/usr/local/lib'])
+env = Environment(CPPPATH='#/include', LIBPATH=['/usr/lib','@PREFIX@/lib'])
 
 # Import GZip builder
 import gzipBuilder
 