--- pyproject.toml.orig	2026-08-18 12:41:08
+++ pyproject.toml	2026-08-18 12:41:32
@@ -34,8 +34,7 @@
 
 [build-system]
 build-backend = "setuptools.build_meta"
-requires = ["setuptools>=62; python_version>='3.12'",
-            "setuptools<60; python_version<'3.12'"]
+requires = ["setuptools"]
 
 [tool.setuptools]
 packages = ['pythran', 'pythran.analyses', 'pythran.transformations',
--- requirements.txt.orig	2026-08-18 12:41:51
+++ requirements.txt	2026-08-18 12:41:59
@@ -1,6 +1,5 @@
 ply>=3.4
-setuptools<60; python_version < '3.12'
-setuptools; python_version >= '3.12'
+setuptools
 gast~=0.7.0
 numpy
 beniget~=0.5.0
