diff --git gtk2/common/config_file.cpp gtk2/common/config_file.cpp
index d54c1b58..85963936 100644
--- gtk2/common/config_file.cpp
+++ gtk2/common/config_file.cpp
@@ -1577,6 +1577,7 @@ static const char * getSystemConfigFile()
            "/etc/qt3/"OLD_CONFIG_FILE,
            "/etc/qt/"OLD_CONFIG_FILE, */
         "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
         nullptr};
 
     for (int i = 0;constFiles[i];i++) {
diff --git qt4/common/config_file.cpp qt4/common/config_file.cpp
index d51ad2f3..8793bdcc 100644
--- qt4/common/config_file.cpp
+++ qt4/common/config_file.cpp
@@ -1427,7 +1427,11 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ /*"/etc/qt4/" OLD_CONFIG_FILE, "/etc/qt3/" OLD_CONFIG_FILE, "/etc/qt/" OLD_CONFIG_FILE,*/ "/etc/" OLD_CONFIG_FILE, nullptr };
+    static const char * constFiles[]={
+        "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
+        nullptr
+    };
 
     int i;
 
@@ -1626,7 +1630,7 @@ void qtcDefaultSettings(Options *opts)
         systemFilename=getSystemConfigFile();
 
     if(systemFilename)
-        qtcReadConfig(systemFilename, opts, opts);
+        qtcReadConfig(QFile::decodeName(systemFilename), opts, opts);
     }
 }
 
diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index e5c6541a..5ec9d08d 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -1437,10 +1437,8 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
 static const char * getSystemConfigFile()
 {
     static const char * constFiles[]={
-        /*"/etc/qt4/" OLD_CONFIG_FILE,
-          "/etc/qt/" OLD_CONFIG_FILE,
-        */
         "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
         nullptr
     };
 
@@ -1641,7 +1639,7 @@ void qtcDefaultSettings(Options *opts)
         systemFilename=getSystemConfigFile();
 
     if(systemFilename)
-        qtcReadConfig(systemFilename, opts, opts);
+        qtcReadConfig(QFile::decodeName(systemFilename), opts, opts);
     }
 }
 
