diff --git qt5/common/config_file.cpp qt5/common/config_file.cpp
index 7630f4e..afbc82b 100644
--- qt5/common/config_file.cpp
+++ qt5/common/config_file.cpp
@@ -1432,10 +1432,8 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
 static const char * getSystemConfigFile()
 {
     static const char * constFiles[]={
-        /*"/etc/qt4/" OLD_CONFIG_FILE,
-          "/etc/qt/" OLD_CONFIG_FILE,
-        */
         "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
         nullptr
     };
 
@@ -1640,7 +1638,7 @@ void qtcDefaultSettings(Options *opts)
         systemFilename=getSystemConfigFile();
 
     if(systemFilename)
-        qtcReadConfig(systemFilename, opts, opts);
+        qtcReadConfig(QFile::decodeName(systemFilename), opts, opts);
     }
 }
 

diff --git qt4/common/config_file.cpp qt4/common/config_file.cpp
index 7a0a6bc..c6992d8 100644
--- qt4/common/config_file.cpp
+++ qt4/common/config_file.cpp
@@ -1436,7 +1436,11 @@ bool qtcReadConfig(const QString &file, Options *opts, Options *defOpts, bool ch
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ /*"/etc/qt4/" OLD_CONFIG_FILE, "/etc/qt3/" OLD_CONFIG_FILE, "/etc/qt/" OLD_CONFIG_FILE,*/ "/etc/" OLD_CONFIG_FILE, nullptr };
+    static const char * constFiles[]={
+        "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
+        nullptr
+    };
 
     int i;
 
@@ -1639,7 +1643,7 @@ void qtcDefaultSettings(Options *opts)
         systemFilename=getSystemConfigFile();
 
     if(systemFilename)
-        qtcReadConfig(systemFilename, opts, opts);
+        qtcReadConfig(QFile::decodeName(systemFilename), opts, opts);
     }
 }
 
diff --git gtk2/common/config_file.cpp gtk2/common/config_file.cpp
index ce83b07..8c453cf 100644
--- gtk2/common/config_file.cpp
+++ gtk2/common/config_file.cpp
@@ -1577,6 +1577,7 @@ static const char * getSystemConfigFile()
            "/etc/qt3/"OLD_CONFIG_FILE,
            "/etc/qt/"OLD_CONFIG_FILE, */
         "/etc/" OLD_CONFIG_FILE,
+        "@@PREFIX@@/etc/xdg/qtcurve/" CONFIG_FILE,
         nullptr};
 
     for (int i = 0;constFiles[i];i++) {
