--- src/CMakeLists.txt.orig	2013-09-15 18:19:18.000000000 +0900
+++ src/CMakeLists.txt	2013-09-15 18:19:28.000000000 +0900
@@ -5,8 +5,8 @@
 find_package(KDeclarative REQUIRED)
 
 # Currently it is called by FindQt4, but call it explicitly
-find_package(X11)
-macro_log_feature(X11_Xkbfile_FOUND "libxkbfile" "X Keyboard Extension" "http://www.x.org/wiki/XKB" FALSE "" "Support for keyboard layout auto-detection in KTouch")
+#find_package(X11)
+#macro_log_feature(X11_Xkbfile_FOUND "libxkbfile" "X Keyboard Extension" "http://www.x.org/wiki/XKB" FALSE "" "Support for keyboard layout auto-detection in KTouch")
 
 add_subdirectory(qml)
 add_subdirectory(schemata)
