diff --git CMakeLists.txt CMakeLists.txt
index 4d1a64c..a40b6cc 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -75,7 +78,8 @@ endif()
 
 find_package(Gnuradio "3.8" REQUIRED audio blocks PATHS ${CMAKE_INSTALL_PREFIX} )
 list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_SOURCE_DIR}/cmake/Modules")
-FIND_LIBRARY(HIDAPI_LIBRARY hidapi-libusb)
+FIND_PATH(HIDAPI_INCLUDES hidapi.h PATH $ENV{CPATH}/hidapi)
+FIND_LIBRARY(HIDAPI_LIBRARY hidapi)
 
 include(GrVersion)
 
