--- configure.orig	2024-10-19 11:41:35
+++ configure	2024-12-02 21:16:20
@@ -3,7 +3,7 @@
 # Use xml2-config or pkg-config to get the include directories for libxml-2.0
 if xml2-config --version >/dev/null 2>&1; then
     libxml2_cflags=$(xml2-config --cflags)
-    libxml2_libs=$(xml2-config --libs)
+    libxml2_libs=$(xml2-config --libs --dynamic)
 elif pkg-config --version >/dev/null 2>&1; then
     libxml2_cflags=$(pkg-config --cflags libxml-2.0)
     libxml2_libs=$(pkg-config --libs --static libxml-2.0)
