--- a/clang/lib/DirectoryWatcher/mac/DirectoryWatcher-mac.cpp
+++ b/clang/lib/DirectoryWatcher/mac/DirectoryWatcher-mac.cpp
@@ -23,7 +23,9 @@
 # define TARGET_OS_OSX 1
 #endif
 
-#if TARGET_OS_OSX
+#if TARGET_OS_OSX &&                                                         \
+    (!defined(MAC_OS_X_VERSION_MAX_ALLOWED) ||                               \
+     MAC_OS_X_VERSION_MAX_ALLOWED >= 1070)
 
 static void stopFSEventStream(FSEventStreamRef);
 
