--- Makefile.in	2006-05-24 08:29:46.000000000 +0200
+++ Makefile.in	2011-02-20 19:38:13.000000000 +0100
@@ -3,7 +3,7 @@
 # Configuration part
 ###################################################################
 
-TARGETDIR = @OCAMLLIB@
+TARGETDIR = $(DESTDIR)@OCAMLLIB@
 
 CAMLC   = @OCAMLC@ -g
 CAMLOPT = @OCAMLOPT@
@@ -35,12 +35,15 @@
 install: install-@OCAMLBEST@
 
 install-common:
+	mkdir -p $(TARGETDIR)
 	cp $(COMMFILES) $(TARGETDIR)
 
 install-byte: install-common
+	mkdir -p $(TARGETDIR)
 	cp $(BYTEFILES) $(TARGETDIR)
 
 install-opt: install-common install-byte
+	mkdir -p $(TARGETDIR)
 	cp $(OPTFILES) $(TARGETDIR)
 
 MAJORVN=0
