--- config/jail.conf.orig	2013-02-20 06:00:19.000000000 +0100
+++ config/jail.conf	2013-02-20 06:07:15.000000000 +0100
@@ -226,6 +226,18 @@
 logpath  = /var/log/auth.log
 ignoreip = 168.192.0.1
 
+# This jail uses pf, the standard firewall of Mac OS X (>=10.7). It uses
+# the table called "bruteforce" in the IceFloor firewall main anchor.
+# To be used with icefloor: http://www.hanynet.com/icefloor/
+
+[ssh-pf]
+
+enabled  = false
+filter   = sshd
+action   = pf-icefloor
+	   sendmail-whois[name="SSH,PF", dest=you@example.com]
+logpath  = /var/log/system.log
+
 # These jails block attacks against named (bind9). By default, logging is off
 # with bind9 installation. You will need something like this:
 #
