--- fail2ban/tests/utils.py.orig	2015-09-28 15:34:25.000000000 +0200
+++ fail2ban/tests/utils.py	2015-09-28 15:36:10.000000000 +0200
@@ -42,7 +42,7 @@
 	if os.path.exists(os.path.join('config','fail2ban.conf')):
 		CONFIG_DIR = 'config'
 	else:
-		CONFIG_DIR = '/etc/fail2ban'
+		CONFIG_DIR = '@@PREFIX@@/etc/fail2ban'
 
 
 def mtimesleep():
@@ -263,4 +263,4 @@
 			kernel32.CloseHandle(process)
 			return True
 		else:
-			return False
\ No newline at end of file
+			return False
