--- server/server.py.orig	2013-06-12 19:21:12.000000000 +0200
+++ server/server.py	2013-08-15 21:54:21.000000000 +0200
@@ -352,7 +352,7 @@
 				# Syslog daemons already add date to the message.
 				formatter = logging.Formatter("%(name)-16s: %(levelname)-6s %(message)s")
 				facility = logging.handlers.SysLogHandler.LOG_DAEMON
-				hdlr = logging.handlers.SysLogHandler("/dev/log", 
+				hdlr = logging.handlers.SysLogHandler("/var/run/syslog", 
 													  facility = facility)
 			elif target == "STDOUT":
 				hdlr = logging.StreamHandler(sys.stdout)
