--- config/fail2ban.conf.orig	2015-09-28 15:42:15.000000000 +0200
+++ config/fail2ban.conf	2015-09-28 15:44:48.000000000 +0200
@@ -46,14 +46,14 @@
 #         communicate with the server afterwards.
 # Values: [ FILE ]  Default: /var/run/fail2ban/fail2ban.sock
 #
-socket = /var/run/fail2ban/fail2ban.sock
+socket = @@PREFIX@@/var/run/fail2ban/fail2ban.sock
 
 # Option: pidfile
 # Notes.: Set the PID file. This is used to store the process ID of the
 #         fail2ban server.
 # Values: [ FILE ]  Default: /var/run/fail2ban/fail2ban.pid
 #
-pidfile = /var/run/fail2ban/fail2ban.pid
+pidfile = @@PREFIX@@/var/run/fail2ban/fail2ban.pid
 
 # Options: dbfile
 # Notes.: Set the file for the fail2ban persistent data to be stored.
@@ -61,7 +61,7 @@
 #         and data is lost when fail2ban is stopped.
 #         A value of "None" disables the database.
 # Values: [ None :memory: FILE ] Default: /var/lib/fail2ban/fail2ban.sqlite3
-dbfile = /var/lib/fail2ban/fail2ban.sqlite3
+dbfile = @@PREFIX@@/var/run/fail2ban/fail2ban.sqlite3
 
 # Options: dbpurgeage
 # Notes.: Sets age at which bans should be purged from the database
