--- fail2ban-client.orig	2012-04-26 10:59:30.000000000 +0200
+++ fail2ban-client	2012-04-26 11:27:03.000000000 +0200
@@ -33,7 +33,7 @@
 
 # Inserts our own modules path first in the list
 # fix for bug #343821
-sys.path.insert(1, "/usr/share/fail2ban")
+sys.path.insert(1, "@@PREFIX@@/share/fail2ban")
 
 # Now we can import our modules
 from common.version import version
@@ -59,7 +59,7 @@
 		self.__stream = None
 		self.__configurator = Configurator()
 		self.__conf = dict()
-		self.__conf["conf"] = "/etc/fail2ban"
+		self.__conf["conf"] = "@@PREFIX@@/etc/fail2ban"
 		self.__conf["dump"] = False
 		self.__conf["force"] = False
 		self.__conf["verbose"] = 1
