--- fail2ban-regex.orig	2013-02-19 16:30:48.000000000 +0100
+++ fail2ban-regex	2013-02-19 19:37:00.000000000 +0100
@@ -25,7 +25,7 @@
 # fix for bug #343821
 if os.path.abspath(__file__).startswith('/usr/'):
 	# makes sense to use system-wide library iff -regex is also under /usr/
-    sys.path.insert(1, "/usr/share/fail2ban")
+    sys.path.insert(1, "@@PREFIX@@/share/fail2ban")
 
 from client.configparserinc import SafeConfigParserWithIncludes
 from ConfigParser import NoOptionError, NoSectionError, MissingSectionHeaderError
@@ -62,7 +62,7 @@
 	
 	test = None
 
-	CONFIG_DEFAULTS = {'configpath' : "/etc/fail2ban/"}
+	CONFIG_DEFAULTS = {'configpath' : "@@PREFIX@@/etc/fail2ban/"}
 
 	def __init__(self):
 		self.__filter = Filter(None)
