--- fail2ban-regex.orig	2013-06-12 19:21:12.000000000 +0200
+++ fail2ban-regex	2013-08-15 23:36:34.000000000 +0200
@@ -29,7 +29,7 @@
 try:
 	from common.version import version
 except ImportError, e:
-	sys.path.insert(1, "/usr/share/fail2ban")
+	sys.path.insert(1, "@@PREFIX@@/share/fail2ban")
 	from common.version import version
 
 from client.configparserinc import SafeConfigParserWithIncludes
@@ -70,7 +70,7 @@
 
 	test = None
 
-	CONFIG_DEFAULTS = {'configpath' : "/etc/fail2ban/"}
+	CONFIG_DEFAULTS = {'configpath' : "@@PREFIX@@/etc/fail2ban/"}
 
 	def __init__(self):
 		self.__filter = Filter(None)
@@ -116,7 +116,7 @@
 		print
 		print "Log:"
 		print "    string                  a string representing a log line"
-		print "    filename                path to a log file (/var/log/auth.log)"
+		print "    filename                path to a log file (/var/log/secure.log)"
 		print
 		print "Regex:"
 		print "    string                  a string representing a 'failregex'"
