--- Makefile.orig	2010-04-24 04:38:35.000000000 +0200
+++ Makefile	2010-04-24 20:00:38.000000000 +0200
@@ -131,7 +131,7 @@
 #
 # Define NO_R_TO_GCC_LINKER if your gcc does not like "-R/path/lib"
 # that tells runtime paths to dynamic libraries;
-# "-Wl,-rpath=/path/lib" is used instead.
+# "-Wl,-rpath,/path/lib" is used instead.
 #
 # Define USE_NSEC below if you want git to care about sub-second file mtimes
 # and ctimes. Note that you need recent glibc (at least 2.2.4) for this, and
@@ -270,7 +270,7 @@
 gitexecdir = libexec/git-core
 sharedir = $(prefix)/share
 template_dir = share/git-core/templates
-htmldir = share/doc/git-doc
+htmldir = share/doc/git-core
 ifeq ($(prefix),/usr)
 sysconfdir = /etc
 ETC_GITCONFIG = $(sysconfdir)/gitconfig
@@ -1098,7 +1098,7 @@
 	ifdef NO_R_TO_GCC_LINKER
 		# Some gcc does not accept and pass -R to the linker to specify
 		# the runtime dynamic library path.
-		CC_LD_DYNPATH = -Wl,-rpath,
+		CC_LD_DYNPATH = -L
 	else
 		CC_LD_DYNPATH = -R
 	endif
@@ -1116,7 +1116,7 @@
 	REMOTE_CURL_NAMES =
 else
 	ifdef CURLDIR
-		# Try "-Wl,-rpath=$(CURLDIR)/$(lib)" in such a case.
+		# Try "-Wl,-rpath,$(CURLDIR)/$(lib)" in such a case.
 		BASIC_CFLAGS += -I$(CURLDIR)/include
 		CURL_LIBCURL = -L$(CURLDIR)/$(lib) $(CC_LD_DYNPATH)$(CURLDIR)/$(lib) -lcurl
 	else
