--- a/contrib/credential/osxkeychain/Makefile.orig	2012-02-22 20:04:18.000000000 -0600
+++ b/contrib/credential/osxkeychain/Makefile	2012-03-06 00:41:11.000000000 -0600
@@ -5,7 +5,7 @@
 CFLAGS = -g -Wall
 
 git-credential-osxkeychain: git-credential-osxkeychain.o
-	$(CC) -o $@ $< -Wl,-framework -Wl,Security
+	$(CC) $(LDFLAGS) -o $@ $< -Wl,-framework -Wl,Security
 
 git-credential-osxkeychain.o: git-credential-osxkeychain.c
 	$(CC) -c $(CFLAGS) $<
