--- Makefile.orig	2007-07-03 02:06:07.000000000 -0700
+++ Makefile	2007-07-03 02:06:32.000000000 -0700
@@ -137,8 +137,6 @@ uname_P := $(shell sh -c 'uname -p 2>/de
 
 # CFLAGS and LDFLAGS are for the users to override from the command line.
 
-CFLAGS = -g -O2 -Wall
-LDFLAGS =
 ALL_CFLAGS = $(CFLAGS)
 ALL_LDFLAGS = $(LDFLAGS)
 STRIP ?= strip
@@ -507,7 +505,7 @@ ifdef NO_R_TO_GCC_LINKER
 	# the runtime dynamic library path.
 	CC_LD_DYNPATH = -Wl,-rpath=
 else
-	CC_LD_DYNPATH = -R
+	CC_LD_DYNPATH = -L
 endif
 
 ifndef NO_CURL
