--- Makefile.orig	2008-10-17 16:10:07.000000000 +0200
+++ Makefile	2008-10-17 16:13:08.000000000 +0200
@@ -164,8 +164,6 @@
 
 # CFLAGS and LDFLAGS are for the users to override from the command line.
 
-CFLAGS = -g -O2 -Wall
-LDFLAGS =
 ALL_CFLAGS = $(CFLAGS)
 ALL_LDFLAGS = $(LDFLAGS)
 STRIP ?= strip
@@ -187,7 +185,7 @@
 gitexecdir = $(prefix)/libexec/git-core
 sharedir = $(prefix)/share
 template_dir = $(sharedir)/git-core/templates
-htmldir=$(sharedir)/doc/git-doc
+htmldir=$(sharedir)/doc/git-core
 ifeq ($(prefix),/usr)
 sysconfdir = /etc
 else
@@ -793,7 +791,7 @@
 	# the runtime dynamic library path.
 	CC_LD_DYNPATH = -Wl,-rpath=
 else
-	CC_LD_DYNPATH = -R
+	CC_LD_DYNPATH = -L
 endif
 
 ifdef NO_CURL
