From 9cd99f330a94e4cd68eeee53525f13d9b7b9648a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 4 Feb 2024 03:17:36 +0800
Subject: [PATCH 2/2] Fix static liblv5

---
 iv/lv5/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git iv/lv5/CMakeLists.txt iv/lv5/CMakeLists.txt
index e4ba404b..3c0b3e4f 100644
--- iv/lv5/CMakeLists.txt
+++ iv/lv5/CMakeLists.txt
@@ -72,10 +72,10 @@ add_library(liblv5 STATIC
     radio/core.cc
     )
 set_target_properties(liblv5 PROPERTIES PREFIX "")
-mz_merge_static_libraries(liblv5 ALL gc-lib v8_dtoa netlib_dtoa)
+# mz_merge_static_libraries(liblv5 ALL gc-lib v8_dtoa netlib_dtoa)
 add_dependencies(liblv5 gc-lib v8_dtoa netlib_dtoa)
 
-set(LV5_LIBRARIES liblv5 ${IV_LIBRARIES})
+set(LV5_LIBRARIES liblv5 gc-lib v8_dtoa netlib_dtoa ${IV_LIBRARIES})
 
 # lv5 shell
 
