From 44474128ab6c711f2f4038835d3396b85adcf773 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 5 Jan 2024 18:44:18 +0800
Subject: [PATCH] ppc.h: use mftb on ppc

---
 include/urcu/arch/ppc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git include/urcu/arch/ppc.h include/urcu/arch/ppc.h
index 80462c6..be9d158 100644
--- include/urcu/arch/ppc.h
+++ include/urcu/arch/ppc.h
@@ -51,7 +51,7 @@ extern "C" {
 	__extension__					\
 	({ 						\
 		unsigned long rval;			\
-		__asm__ __volatile__ ("mftbl %0" : "=r" (rval));	\
+		__asm__ __volatile__ ("mftb %0" : "=r" (rval));	\
 		rval;					\
 	})
 
