--- gen.tab/Makefile.am.orig	2020-03-24 21:49:33.864576589 -0400
+++ gen.tab/Makefile.am	2020-03-24 21:49:51.680898782 -0400
@@ -23,9 +23,7 @@
 gen_brackets_tab_CPPFLAGS = $(AM_CPPFLAGS)
 gen_brackets_type_tab_CPPFLAGS = $(AM_CPPFLAGS)

-CFLAGS_FOR_BUILD += -DHAVE_CONFIG_H -I$(top_builddir) -I$(top_builddir)/lib -I$(top_srcdir)/lib
 CC = $(CC_FOR_BUILD)
-CFLAGS = $(CFLAGS_FOR_BUILD)
 LDFLAGS = $(LDFLAGS_FOR_BUILD)

 CLEANFILES = $(EXTRA_PROGRAMS)
