--- gen.tab/Makefile.am.orig	2018-07-23 07:09:31.000000000 +1000
+++ gen.tab/Makefile.am	2019-02-09 10:54:31.000000000 +1100
@@ -23,9 +23,7 @@
 gen_brackets_tab_CPPFLAGS = $(AM_CPPFLAGS)
 gen_brackets_type_tab_CPPFLAGS = $(AM_CPPFLAGS)
 
-CFLAGS_FOR_BUILD += -DHAVE_CONFIG_H -I$(top_builddir) -I$(top_builddir)/lib -I$(top_srcdir)/lib
 CC = $(CC_FOR_BUILD)
-CFLAGS = $(CFLAGS_FOR_BUILD)
 
 CLEANFILES = $(EXTRA_PROGRAMS)
 DISTCLEANFILES =
