--- gqrx.pro.orig	2016-01-15 15:12:02.000000000 -0500
+++ gqrx.pro	2016-01-15 15:13:05.000000000 -0500
@@ -48,10 +48,12 @@
     PREFIX=/usr/local
 }
 
-target.path  = $$PREFIX/bin
+target.path  = @APPSDIR@
 INSTALLS    += target 
 
-#CONFIG += debug
+CONFIG -= debug
+VER = @VERSION@
+CONFIG += @ARCHES@
 
 # disable debug messages in release
 CONFIG(debug, debug|release) {
@@ -59,13 +61,13 @@
     #QMAKE_CFLAGS_DEBUG += '-g -O0'
 
     # Define version string (see below for releases)
-    VER = $$system(git describe --abbrev=8)
+    # VER = $$system(git describe --abbrev=8)
     ##VER = 2.4
 
 } else {
     DEFINES += QT_NO_DEBUG
     DEFINES += QT_NO_DEBUG_OUTPUT
-    VER = $$system(git describe --abbrev=1)
+    # VER = $$system(git describe --abbrev=1)
     ##VER = 2.4
 
     # Release binaries with gr bundled
