--- gqrx.pro.orig	2014-12-04 12:58:29.000000000 -0500
+++ gqrx.pro	2014-12-04 12:58:33.000000000 -0500
@@ -47,24 +47,20 @@
     PREFIX=/usr/local
 }
 
-target.path  = $$PREFIX/bin
+target.path  = @APPSDIR@
 INSTALLS    += target 
 
-#CONFIG += debug
+CONFIG -= debug
+VER = @VERSION@
+CONFIG += @ARCHES@
 
 # disable debug messages in release
 CONFIG(debug, debug|release) {
     # Use for valgrind
     #QMAKE_CFLAGS_DEBUG += '-g -O0'
-
-    # Define version string (see below for releases)
-    VER = $$system(git describe --abbrev=8)
-
 } else {
     DEFINES += QT_NO_DEBUG
     DEFINES += QT_NO_DEBUG_OUTPUT
-    VER = $$system(git describe --abbrev=1)
-    ## VER = v2.3.1-37-g78ea9469
 
     # Release binaries with gr bundled
     # QMAKE_RPATH & co won't work with origin
