--- src/botantools/botan/botan/secmem.h.orig	2012-09-26 21:16:22.000000000 -0400
+++ src/botantools/botan/botan/secmem.h	2012-09-26 21:16:38.000000000 -0400
@@ -214,7 +214,7 @@
 
       SecureVector(u32bit n = 0) { MemoryRegion<T>::init(true, n); }
       SecureVector(const T in[], u32bit n)
-         { MemoryRegion<T>::init(true); set(in, n); }
+	 { MemoryRegion<T>::init(true); MemoryRegion<T>::set(in, n); }
       SecureVector(const MemoryRegion<T>& in)
          { MemoryRegion<T>::init(true); set(in); }
       SecureVector(const MemoryRegion<T>& in1, const MemoryRegion<T>& in2)
