From cc33f40f29dcd250e2dc3df77412fdec91f2d4eb Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yen@chyen.cc>
Date: Sat, 29 Dec 2018 21:24:46 +0800
Subject: [PATCH] Fix a failing test on macOS Mojave

---
 test/mitmproxy/test_proxy.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git test/mitmproxy/test_proxy.py test/mitmproxy/test_proxy.py
index 00086c4b5..c8cf6c334 100644
--- test/mitmproxy/test_proxy.py
+++ test/mitmproxy/test_proxy.py
@@ -1,4 +1,5 @@
 import argparse
+import platform
 from unittest import mock
 import pytest
 
@@ -52,8 +53,11 @@ def test_certs(self, tdata):
 class TestProxyServer:
 
     @skip_windows
+    @pytest.mark.skipif(platform.mac_ver()[0].split('.')[:2] == ['10', '14'],
+                        reason='Skipping due to macOS Mojave')
     def test_err(self):
-        # binding to 0.0.0.0:1 works without special permissions on Windows
+        # binding to 0.0.0.0:1 works without special permissions on Windows and
+        # macOS Mojave
         conf = ProxyConfig(options.Options(listen_port=1))
         with pytest.raises(Exception, match="Error starting proxy server"):
             ProxyServer(conf)
