diff --git mitmproxy/addons/__init__.py mitmproxy/addons/__init__.py
index 838fba9b..5d21bf68 100644
--- mitmproxy/addons/__init__.py
+++ mitmproxy/addons/__init__.py
@@ -8,7 +8,6 @@ from mitmproxy.addons import core
 from mitmproxy.addons import cut
 from mitmproxy.addons import disable_h2c
 from mitmproxy.addons import export
-from mitmproxy.addons import onboarding
 from mitmproxy.addons import proxyauth
 from mitmproxy.addons import replace
 from mitmproxy.addons import script
@@ -33,7 +32,6 @@ def default_addons():
         cut.Cut(),
         disable_h2c.DisableH2C(),
         export.Export(),
-        onboarding.Onboarding(),
         proxyauth.ProxyAuth(),
         replace.Replace(),
         script.ScriptLoader(),
