--- CMakeLists.txt.orig	2019-06-09 12:18:38.000000000 -0700
+++ CMakeLists.txt	2019-06-15 08:38:16.000000000 -0700
@@ -6,14 +6,14 @@
 PROJECT("Luminance HDR")
 
 string(TOLOWER ${CMAKE_BUILD_TYPE} LOWERCASE_CMAKE_BUILD_TYPE)
-if(LOWERCASE_CMAKE_BUILD_TYPE MATCHES "^(debug|release|relwithdebinfo|minsizerel)$")
+if(LOWERCASE_CMAKE_BUILD_TYPE MATCHES "^(debug|release|relwithdebinfo|minsizerel|macports)$")
     message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 else()
     message(FATAL_ERROR "Invalid value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}\n"
         "CMAKE_BUILD_TYPE should be one of:\nDebug, Release, RelWithDebInfo, MinSizeRel")
 endif()
 
-if(LOWERCASE_CMAKE_BUILD_TYPE MATCHES "^(release|minsizerel)$")
+if(LOWERCASE_CMAKE_BUILD_TYPE MATCHES "^(release|minsizerel|macports)$")
     message(STATUS "Adding definitions: -DQT_NO_DEBUG_OUTPUT")
     add_definitions("-DQT_NO_DEBUG_OUTPUT")
 endif()
