--- build/PackagingApple.cmake.orig	2014-01-27 13:20:46.000000000 -0700
+++ build/PackagingApple.cmake	2016-05-22 04:16:52.000000000 -0700
@@ -13,13 +13,6 @@
 SET(APPS "\${CMAKE_INSTALL_PREFIX}/${LHDR_OSX_EXECUTABLE_NAME}.app")
 
 #--------------------------------------------------------------------------------
-# Install needed Qt plugins by copying directories from the qt installation
-# One can cull what gets copied by using 'REGEX "..." EXCLUDE'
-install(DIRECTORY "${QT_PLUGINS_DIR}/sqldrivers"
-        DESTINATION ${plugin_dest_dir} COMPONENT Runtime)
-install(FILES ${LIBS}
-        DESTINATION ${LHDR_OSX_EXECUTABLE_NAME}.app/Contents/MacOS)
-
 # install align_image_stack
 install(FILES ${CMAKE_SOURCE_DIR}/build/macosx/align_image_stack
         DESTINATION ${LHDR_OSX_EXECUTABLE_NAME}.app/Contents/MacOS)
@@ -30,10 +23,6 @@
 install(DIRECTORY ${CMAKE_SOURCE_DIR}/html
         DESTINATION ${LHDR_OSX_EXECUTABLE_NAME}.app/Contents/Resources)
 
-# install a qt.conf file
-install(FILES ${CMAKE_SOURCE_DIR}/build/macosx/qt.conf
-        DESTINATION ${LHDR_OSX_EXECUTABLE_NAME}.app/Contents/Resources)
-
 # install README and other files
 install(FILES
         ${CMAKE_SOURCE_DIR}/AUTHORS
@@ -48,13 +37,6 @@
 # directories to look for dependencies
 set(DIRS ${QT_LIBRARY_DIRS})
 
-INSTALL(CODE "
-    file(GLOB_RECURSE QTPLUGINS
-      \"\${CMAKE_INSTALL_PREFIX}/${plugin_dest_dir}/*${CMAKE_SHARED_LIBRARY_SUFFIX}\")
-    include(BundleUtilities)
-    fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
-    " COMPONENT Runtime)
-
 # create drag and drop installer
 set(CPACK_BINARY_DRAGNDROP ON)
 include(CPack)
