--- gdk/Makefile.am.orig	2009-08-24 14:16:36.000000000 +0400
+++ gdk/Makefile.am	2009-08-22 05:20:42.000000000 +0400
@@ -2,7 +2,7 @@
 ## Makefile.am for gtkglext/gdk
 
 SUBDIRS = glext $(gdktarget)
-DIST_SUBDIRS = glext x11 win32
+DIST_SUBDIRS = glext x11 win32 quartz
 
 EXTRA_DIST = \
 	gdkglversion.h.in	\
@@ -131,12 +131,17 @@
 
 EXTRA_LTLIBRARIES = \
 	libgdkglext-x11-@API_MJ@.@API_MI@.la	\
+	libgdkglext-quartz-@API_MJ@.@API_MI@.la  \
 	libgdkglext-win32-@API_MJ@.@API_MI@.la
 
 libgdkglext_x11_@API_MJ@_@API_MI@_la_SOURCES = $(gdkglext_sources)
 libgdkglext_x11_@API_MJ@_@API_MI@_la_LDFLAGS = $(common_ldflags)
 libgdkglext_x11_@API_MJ@_@API_MI@_la_LIBADD = x11/libgdkglext-x11.la $(common_libadd)
 
+libgdkglext_quartz_@API_MJ@_@API_MI@_la_SOURCES = $(gdkglext_sources)
+libgdkglext_quartz_@API_MJ@_@API_MI@_la_LDFLAGS = $(common_ldflags)
+libgdkglext_quartz_@API_MJ@_@API_MI@_la_LIBADD = quartz/libgdkglext-quartz.la $(common_libadd)
+
 libgdkglext_win32_@API_MJ@_@API_MI@_la_SOURCES = $(gdkglext_sources)
 libgdkglext_win32_@API_MJ@_@API_MI@_la_LDFLAGS = $(common_ldflags)
 libgdkglext_win32_@API_MJ@_@API_MI@_la_LIBADD = win32/libgdkglext-win32.la $(common_libadd)
@@ -152,7 +157,7 @@
 	$(gdkglext_built_sources)
 
 # Generate built header without using automake BUILT_SOURCES
-$(libgdkglext_x11_@API_MJ@_@API_MI@_la_OBJECTS) $(libgdkglext_win32_@API_MJ@_@API_MI@_la_OBJECTS): $(gdkglext_built_public_h_sources)
+$(libgdkglext_x11_@API_MJ@_@API_MI@_la_OBJECTS) $(libgdkglext_quartz_@API_MJ@_@API_MI@_la_OBJECTS) $(libgdkglext_win32_@API_MJ@_@API_MI@_la_OBJECTS): $(gdkglext_built_public_h_sources)
 
 $(srcdir)/gdkglenumtypes.h: stamp-gdkglenumtypes-h
 	@true
