--- ./gdk/x11/gdkglfont-x11.c.orig	2023-01-24 12:50:29
+++ ./gdk/x11/gdkglfont-x11.c	2023-01-24 12:54:28
@@ -18,8 +18,6 @@
 
 #include <string.h>
 
-#include <pango/pangox.h>
-
 #include "gdkglx.h"
 #include "gdkglprivate-x11.h"
 #include "gdkglfont.h"
@@ -75,55 +73,9 @@
                                    int                         count,
                                    int                         list_base)
 {
-  PangoFont *font = NULL;
-  gchar *charset = NULL;
-  PangoXSubfont subfont_id;
-  gchar *xlfd = NULL;
-  PangoXFontCache *font_cache;
-  XFontStruct *fs;
-
   GDK_GL_NOTE_FUNC_PRIVATE ();
 
-  font = pango_font_map_load_font (font_map, NULL, font_desc);
-  if (font == NULL)
-    {
-      g_warning ("cannot load PangoFont");
-      goto FAIL;
-    }
-
-  charset = gdk_gl_font_charset_for_locale ();
-  if (!pango_x_find_first_subfont (font, &charset, 1, &subfont_id))
-    {
-      g_warning ("cannot find PangoXSubfont");
-      font = NULL;
-      goto FAIL;
-    }
-
-  xlfd = pango_x_font_subfont_xlfd (font, subfont_id);
-  if (xlfd == NULL)
-    {
-      g_warning ("cannot get XLFD");
-      font = NULL;
-      goto FAIL;
-    }
-
-  font_cache = pango_x_font_map_get_font_cache (font_map);
-
-  fs = pango_x_font_cache_load (font_cache, xlfd);
-
-  glXUseXFont (fs->fid, first, count, list_base);
-
-  pango_x_font_cache_unload (font_cache, fs);
-
- FAIL:
-
-  if (charset != NULL)
-    g_free (charset);
-
-  if (xlfd != NULL)
-    g_free (xlfd);
-
-  return font;
+  return  pango_font_map_load_font (font_map, NULL, font_desc);
 }
 
 /**
@@ -149,12 +101,8 @@
 
   GDK_GL_NOTE_FUNC ();
 
-#ifdef GDKGLEXT_MULTIHEAD_SUPPORT
-  font_map = pango_x_font_map_for_display (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()));
-#else  /* GDKGLEXT_MULTIHEAD_SUPPORT */
-  font_map = pango_x_font_map_for_display (gdk_x11_get_default_xdisplay ());
-#endif /* GDKGLEXT_MULTIHEAD_SUPPORT */
 
+  font_map = pango_cairo_font_map_get_default ();
   return gdk_gl_font_use_pango_font_common (font_map, font_desc,
                                             first, count, list_base);
 }
@@ -187,7 +135,7 @@
 
   GDK_GL_NOTE_FUNC ();
 
-  font_map = pango_x_font_map_for_display (GDK_DISPLAY_XDISPLAY (display));
+  font_map = pango_cairo_font_map_get_default ();
 
   return gdk_gl_font_use_pango_font_common (font_map, font_desc,
                                             first, count, list_base);
