--- rpc_clntout.c.orig
+++ rpc_clntout.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_clntout.c 1.11 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_clntout.c, Client-stub outputter for the RPC protocol compiler
--- rpc_cout.c.orig
+++ rpc_cout.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_cout.c 1.13 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_cout.c, XDR routine outputter for the RPC protocol compiler
--- rpc_hout.c.orig
+++ rpc_hout.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_hout.c 1.12 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_hout.c, Header file outputter for the RPC protocol compiler
--- rpc_main.c.orig
+++ rpc_main.c
@@ -36,7 +36,7 @@ static char sccsid[] = "@(#)rpc_main.c 1.30 89/03/30 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_main.c, Top level of the RPC protocol compiler.
--- rpc_parse.c.orig
+++ rpc_parse.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_parse.c 1.8 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_parse.c, Parser for the RPC protocol compiler
--- rpc_sample.c.orig
+++ rpc_sample.c
@@ -30,7 +30,7 @@
 /* #pragma ident	"@(#)rpc_sample.c	1.9	94/04/25 SMI" */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_sample.c, Sample client-server code outputter for the RPC protocol compiler
--- rpc_scan.c.orig
+++ rpc_scan.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_scan.c 1.11 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_scan.c, Scanner for the RPC protocol compiler
--- rpc_svcout.c.orig
+++ rpc_svcout.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_svcout.c 1.29 89/03/30 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_svcout.c, Server-skeleton outputter for the RPC protocol compiler
--- rpc_tblout.c.orig
+++ rpc_tblout.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_tblout.c 1.4 89/02/22 (C) 1988 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_tblout.c, Dispatch table outputter for the RPC protocol compiler
--- rpc_util.c.orig
+++ rpc_util.c
@@ -35,7 +35,7 @@ static char sccsid[] = "@(#)rpc_util.c 1.11 89/02/22 (C) 1987 SMI";
 #endif
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
+/*__FBSDID("$FreeBSD$");*/
 
 /*
  * rpc_util.c, Utility routines for the RPC protocol compiler
