--- setup.py.orig	2009-03-04 03:13:23.000000000 +1100
+++ setup.py	2012-06-25 12:54:38.000000000 +1000
@@ -21,7 +21,7 @@
 # Common configuration #
 ########################
 
-extra_link_args = ['--Wl,--no-undefined']
+extra_link_args = ['-Wl,--no-undefined']
 include_dirs = []
 library_dirs = []
 defines = []
@@ -45,11 +45,8 @@
 else:
     # try to look for nifticlibs in some place
     if not sys.platform.startswith('win'):
-        include_dirs += ['/usr/include/nifti',
-                         '/usr/include/nifticlibs',
-                         '/usr/local/include/nifti',
-                         '/usr/local/include/nifticlibs',
-                         '/usr/local/include']
+        include_dirs += ['@@PREFIX@@/include/nifti']
+        library_dirs.append('@@PREFIX@@/lib')
     else:
         # no clue on windows
         pass
