--- setup.py.old	2005-08-19 00:18:38.000000000 +0900
+++ setup.py	2005-08-19 00:18:58.000000000 +0900
@@ -43,7 +43,7 @@
                     'Topic :: Software Development :: Testing',
                   ]
 
-    install_dir = sysconfig.get_python_lib() + os.sep + 'pychecker'
+    install_dir = 'PYTHONLIB' + os.sep + 'pychecker'
     checker_py = install_dir + os.sep + 'checker.py'
     py_exe = sys.executable
 
@@ -78,7 +78,6 @@
           author_email          = "pychecker@metaslash.com",
           url                   = "http://pychecker.sourceforge.net/",
           packages              = [ 'pychecker' ],
-          data_files            = [ (install_dir, DOC_FILES) ],
           scripts               = [ LOCAL_SCRIPT, ],
           # classifiers was added in Python 2.3, but needed for PyPI
           #classifiers           = CLASSIFIERS,
