--- SConstruct.orig	2010-06-16 08:40:25.000000000 -0700
+++ SConstruct	2010-06-22 15:08:14.000000000 -0700
@@ -530,7 +530,7 @@
 		print ('Ardour cannot be compiled without the curl headers, which do not seem to be installed')
 		sys.exit (1)            
 	else:
-		libraries['curl'].ParseConfig('pkg-config --cflags --libs libcurl')
+		libraries['curl'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs libcurl')
 	conf.Finish()
 else:
 	print 'FREESOUND support is not enabled.  Build with \'scons FREESOUND=1\' to enable.'
@@ -540,7 +540,7 @@
 	
 	if conf.CheckPKGVersion('slv2', '0.6.1'):
 		libraries['slv2'] = LibraryInfo()
-		libraries['slv2'].ParseConfig('pkg-config --cflags --libs slv2')
+		libraries['slv2'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs slv2')
                 env.Append (CCFLAGS="-DHAVE_LV2")
 	else:
 		print 'LV2 support is not enabled (SLV2 not found or older than 0.6.0)'
@@ -879,7 +879,7 @@
 
 conf = Configure (env)
 
-if conf.CheckHeader ('fftw3.h'):
+if conf.CheckHeader ('@MP_PREFIX@/include/fftw3.h'):
     env['RUBBERBAND'] = True
     libraries['rubberband'] = LibraryInfo (LIBS='rubberband',
                                            LIBPATH='#libs/rubberband',
@@ -940,7 +940,7 @@
 
 libraries['boost'] = LibraryInfo ()
 prep_libcheck(env, libraries['boost'])
-libraries['boost'].Append(CPPPATH="/usr/local/include", LIBPATH="/usr/local/lib")
+libraries['boost'].Append(CPPPATH="@MP_PREFIX@/include", LIBPATH="@MP_PREFIX@/lib")
 conf = Configure (libraries['boost'])
 if conf.CheckHeader ('boost/shared_ptr.hpp', language='CXX') == False:
         print "Boost header files do not appear to be installed. You also might be running a buggy version of scons. Try scons 0.97 if you can."
@@ -954,6 +954,7 @@
 if env['LIBLO']:
     libraries['lo'] = LibraryInfo ()
     prep_libcheck(env, libraries['lo'])
+    libraries['lo'].Append(CPPPATH="@MP_PREFIX@/include", LIBPATH="@MP_PREFIX@/lib")
 
     conf = Configure (libraries['lo'])
     if conf.CheckLib ('lo', 'lo_server_new') == False:
@@ -1055,30 +1056,30 @@
     env = conf.Finish()
     
     libraries['sigc2'] = LibraryInfo()
-    libraries['sigc2'].ParseConfig('pkg-config --cflags --libs sigc++-2.0')
+    libraries['sigc2'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs sigc++-2.0')
     libraries['glibmm2'] = LibraryInfo()
-    libraries['glibmm2'].ParseConfig('pkg-config --cflags --libs glibmm-2.4')
+    libraries['glibmm2'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs glibmm-2.4')
     libraries['cairomm'] = LibraryInfo()
-    libraries['cairomm'].ParseConfig('pkg-config --cflags --libs cairomm-1.0')
+    libraries['cairomm'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs cairomm-1.0')
     libraries['gdkmm2'] = LibraryInfo()
-    libraries['gdkmm2'].ParseConfig ('pkg-config --cflags --libs gdkmm-2.4')
+    libraries['gdkmm2'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs gdkmm-2.4')
     libraries['gtkmm2'] = LibraryInfo()
-    libraries['gtkmm2'].ParseConfig ('pkg-config --cflags --libs gtkmm-2.4')
+    libraries['gtkmm2'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs gtkmm-2.4')
     libraries['atkmm'] = LibraryInfo()
-    libraries['atkmm'].ParseConfig ('pkg-config --cflags --libs atkmm-1.6')
+    libraries['atkmm'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs atkmm-1.6')
     libraries['pangomm'] = LibraryInfo()
-    libraries['pangomm'].ParseConfig ('pkg-config --cflags --libs pangomm-1.4')
+    libraries['pangomm'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs pangomm-1.4')
     libraries['libgnomecanvasmm'] = LibraryInfo()
-    libraries['libgnomecanvasmm'].ParseConfig ('pkg-config --cflags --libs libgnomecanvasmm-2.6')
+    libraries['libgnomecanvasmm'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs libgnomecanvasmm-2.6')
 
 #    libraries['libglademm'] = LibraryInfo()
-#    libraries['libglademm'].ParseConfig ('pkg-config --cflags --libs libglademm-2.4')
+#    libraries['libglademm'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs libglademm-2.4')
 
 #    libraries['flowcanvas'] = LibraryInfo(LIBS='flowcanvas', LIBPATH='#/libs/flowcanvas', CPPPATH='#libs/flowcanvas')
     libraries['soundtouch'] = LibraryInfo()
-    libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs soundtouch-1.0')
+    libraries['soundtouch'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs soundtouch-1.0')
     # Comment the previous line and uncomment this for old versions of Debian:
-    #libraries['soundtouch'].ParseConfig ('pkg-config --cflags --libs libSoundTouch')
+    #libraries['soundtouch'].ParseConfig('@MP_PREFIX@/bin/pkg-config --cflags --libs libSoundTouch')
 
     libraries['appleutility'] = LibraryInfo(LIBS='libappleutility',
                                             LIBPATH='#libs/appleutility',
@@ -1276,7 +1277,7 @@
     else:
         print "Found msgmerge"
     
-    if not conf.CheckCHeader('libintl.h'):
+    if not conf.CheckCHeader('@MP_PREFIX@/include/libintl.h'):
         nls_error += ' No libintl.h.'
         env['NLS'] = 0
         
