--- gtk2_ardour/SConscript.orig	2009-06-19 21:48:51.000000000 -0700
+++ gtk2_ardour/SConscript	2009-06-19 21:55:28.000000000 -0700
@@ -405,9 +405,9 @@
 # 
 
 if env['GTKOSX']:
-	ardour_menus = env.Command ('ardour.menus', 'ardour.menus.in', "cpp -E -P -DGTKOSX -DTOP_MENUBAR ardour.menus.in ardour.menus", chdir=1)
+	ardour_menus = env.Command ('ardour.menus', 'ardour.menus.in', "@MP_CPP@ -E -P -DGTKOSX -DTOP_MENUBAR ardour.menus.in ardour.menus", chdir=1)
 else:
-	ardour_menus = env.Command ('ardour.menus', 'ardour.menus.in', "cpp -E -P ardour.menus.in ardour.menus", chdir=1)
+	ardour_menus = env.Command ('ardour.menus', 'ardour.menus.in', "@MP_CPP@ -E -P ardour.menus.in ardour.menus", chdir=1)
 
 ardour_dark_theme = env.SubstInFile ('ardour2_ui_dark.rc', 'ardour2_ui_dark.rc.in', SUBST_DICT = my_font_dict)
 ardour_light_theme = env.SubstInFile ('ardour2_ui_light.rc', 'ardour2_ui_light.rc.in', SUBST_DICT = my_font_dict)
