--- gtk2_ardour/main.cc.orig	2010-07-21 23:36:34.000000000 -0700
+++ gtk2_ardour/main.cc	2010-07-21 23:38:09.000000000 -0700
@@ -62,7 +62,7 @@
 static ARDOUR_UI  *ui = 0;
 static const char* localedir = LOCALEDIR;
 
-#ifdef __APPLE__
+#ifdef GTKOSX
 
 #include <mach-o/dyld.h>
 #include <sys/param.h>
@@ -313,7 +313,7 @@
 {
 	vector<Glib::ustring> null_file_list;
 	
-#ifdef __APPLE__
+#ifdef GTKOSX
 	fixup_bundle_environment ();
 #endif
 
