--- src/config.h	Thu May 13 21:11:37 2004
+++ src/config.h.new	Mon May 24 15:01:52 2004
@@ -28,10 +28,17 @@
 #else
 #  define RESOLVFILE "/etc/resolv.conf"
 #endif
-#define RUNFILE "/var/run/dnsmasq.pid"
+#ifdef __APPLE__
+#  define RUNFILE "__PREFIX__/var/run/dnsmasq.pid"
+#else
+#  define RUNFILE "/var/run/dnsmasq.pid"
+#endif
 #if defined(__FreeBSD__) || defined (__OpenBSD__)
 #   define LEASEFILE "/var/db/dnsmasq.leases"
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
+#elif __APPLE__
+#   define LEASEFILE "__PREFIX__/var/db/dnsmasq.leases"
+#   define CONFFILE "__PREFIX__/etc/dnsmasq.conf"
 #else
 #   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
 #   define CONFFILE "/etc/dnsmasq.conf"
