--- src/config.h.orig	2012-03-04 15:04:22.000000000 -0500
+++ src/config.h	2012-03-06 11:00:17.000000000 -0500
@@ -32,7 +32,7 @@
 #define SMALLDNAME 40 /* most domain names are smaller than this */
 #define HOSTSFILE "/etc/hosts"
 #define ETHERSFILE "/etc/ethers"
-#define RUNFILE "/var/run/dnsmasq.pid"
+#define RUNFILE "@PREFIX@/var/run/dnsmasq.pid"
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
 #define CHGRP "dip"
@@ -137,7 +137,7 @@
 #   elif defined(__ANDROID__)
 #      define LEASEFILE "/data/misc/dhcp/dnsmasq.leases"
 #   else
-#      define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#      define LEASEFILE "@PREFIX@/var/lib/misc/dnsmasq.leases"
 #   endif
 #endif
 
@@ -145,7 +145,7 @@
 #   if defined(__FreeBSD__)
 #      define CONFFILE "/usr/local/etc/dnsmasq.conf"
 #   else
-#      define CONFFILE "/etc/dnsmasq.conf"
+#      define CONFFILE "@PREFIX@/etc/dnsmasq.conf"
 #   endif
 #endif
 
@@ -153,7 +153,7 @@
 #   if defined(__uClinux__)
 #      define RESOLVFILE "/etc/config/resolv.conf"
 #   else
-#      define RESOLVFILE "/etc/resolv.conf"
+#      define RESOLVFILE "@PREFIX@/etc/resolv.conf"
 #   endif
 #endif
 
