--- Makefile.orig	2008-06-08 00:22:47.000000000 -0500
+++ Makefile	2008-06-08 00:24:34.000000000 -0500
@@ -17,6 +17,7 @@
 BINDIR = ${PREFIX}/sbin
 MANDIR = ${PREFIX}/share/man
 LOCALEDIR = ${PREFIX}/share/locale
+CONFDIR = ${PREFIX}/etc
 
 SRC = src
 PO = po
@@ -53,6 +54,7 @@
 	$(INSTALL) -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL) -m 644 $(MAN)/dnsmasq.8 $(DESTDIR)$(MANDIR)/man8 
 	$(INSTALL) -m 755 $(SRC)/dnsmasq $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 644 dnsmasq.conf.example $(DESTDIR)$(CONFDIR)/dnsmasq.conf.sample
 
 all-i18n :
 	cd $(SRC) && $(MAKE) \
