--- Makefile.orig	2008-06-06 07:04:47.000000000 -0400
+++ Makefile	2008-07-16 23:56:35.000000000 -0400
@@ -17,13 +17,14 @@
 BINDIR = ${PREFIX}/sbin
 MANDIR = ${PREFIX}/share/man
 LOCALEDIR = ${PREFIX}/share/locale
+CONFDIR = ${PREFIX}/etc
 
 SRC = src
 PO = po
 MAN = man
 
 PKG_CONFIG = pkg-config
-AWK = nawk
+AWK = awk
 INSTALL = install
 
 DBUS_MINOR=" `echo $(COPTS) | ../bld/pkg-wrapper $(PKG_CONFIG) --modversion dbus-1 | $(AWK) -F . -- '{ if ($$(NF-1)) print \"-DDBUS_MINOR=\"$$(NF-1) }'`" 
@@ -53,6 +54,7 @@
 	$(INSTALL) -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL) -m 644 $(MAN)/dnsmasq.8 $(DESTDIR)$(MANDIR)/man8 
 	$(INSTALL) -m 755 $(SRC)/dnsmasq $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 644 dnsmasq.conf.example $(DESTDIR)$(CONFDIR)/dnsmasq.conf.sample
 
 all-i18n :
 	cd $(SRC) && $(MAKE) \
