--- Makefile	Sun May  9 17:50:11 2004
+++ Makefile.new	Mon May 24 15:09:16 2004
@@ -1,6 +1,7 @@
 PREFIX?=/usr/local
 BINDIR = ${PREFIX}/sbin
-MANDIR = ${PREFIX}/man
+MANDIR = ${PREFIX}/share/man
+CONFDIR = ${PREFIX}/etc
 
 SRC = src
 
@@ -16,7 +17,7 @@
 	install -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
 	install -m 644 dnsmasq.8 $(DESTDIR)$(MANDIR)/man8 
 	install -m 755 $(SRC)/dnsmasq $(DESTDIR)$(BINDIR)
-
+	install -m 644 dnsmasq.conf.example $(DESTDIR)$(CONFDIR)/dnsmasq.conf.sample
 
 
 
