--- ddd/ddd.C	2024/01/12 21:23:32	7409
+++ ddd/ddd.C	2024/01/13 13:23:07	7410
@@ -193,7 +193,6 @@
 #include "graph/GraphEdit.h"
 #include "GDBAgent.h"
 #include "HistoryD.h"
-#include "motif/LabelH.h"
 #include "motif/MakeMenu.h"
 #include "PlotAgent.h"
 #include "SourceView.h"
@@ -299,6 +298,7 @@
 }
 #endif
 
+#include "box/FontTable.h" // for the define USE_XFT_LIB
 
 //-----------------------------------------------------------------------------
 // Forward function decls
@@ -2322,10 +2322,6 @@
     }
 #endif
 
-    // Setup label hack
-    arg = 0;
-    XtCreateWidget("label_hack", xmLabelHackWidgetClass, toplevel, args, arg);
-
     // Set key bindings
     setup_cut_copy_paste_bindings(XtDatabase(XtDisplay(toplevel)));
     setup_select_all_bindings(XtDatabase(XtDisplay(toplevel)));
