--- src/kilestdtools.cpp.orig	2014-06-17 22:11:28.000000000 +0900
+++ src/kilestdtools.cpp	2014-06-17 22:29:16.000000000 +0900
@@ -377,6 +377,8 @@
 		addDict("%dir_target", QString());
 		addDict("%target", relativeTarget);
 		addDict("%absolute_target", absoluteTarget);
+		addDict("%sourceFileName", filepath);
+		addDict("%sourceLine", QString::number(para + 1));
 		KILE_DEBUG() << "==KileTool::ForwardDVI::determineTarget()=============\n";
 		KILE_DEBUG() << "\tusing  (absolute)" << absoluteTarget;
 		KILE_DEBUG() << "\tusing  (relative)" << relativeTarget;
