--- CMakeLists.txt.orig	2013-12-26 13:58:47.000000000 +0400
+++ CMakeLists.txt	2013-12-26 13:59:29.000000000 +0400
@@ -36,7 +36,7 @@
 # For out-of-the-box support on MacOS:
 IF (ENABLE_OPENCL)
   IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    INCLUDE_DIRECTORIES("/opt/local/include")
+    INCLUDE_DIRECTORIES("@@PREFIX@@/include")
     set(CMAKE_EXE_LINKER_FLAGS "-framework OpenCL")
   ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 ENDIF(ENABLE_OPENCL)
