Index: fnc.bld.mk
=======================================================================
hash - d5125d91b651e4115400e52857e80fd35cd8fb25ccf50e3828db32c27f119b63
hash + 0461726b7bef3caa495e6e17283581bc33b1f0d439da29c2ef477439a27d6d45
--- fnc.bld.mk
+++ fnc.bld.mk
@@ -37,13 +37,13 @@ FOSSIL_CFLAGS =	${CFLAGS} -Wall -Werror -Wsign-compare 
 		-DSQLITE_TRUSTED_SCHEMA=0
 
 # FLAGS NEEDED TO BUILD LIBFOSSIL
-FOSSIL_CFLAGS =	${CFLAGS} -Wall -Werror -Wsign-compare -pedantic -std=c99
+FOSSIL_CFLAGS =	${CFLAGS} -Wall -Wsign-compare -pedantic -std=c99
 
 # On SOME Linux (e.g., Ubuntu 18.04.6), we have to include wchar curses from
 # I/.../ncursesw, but linking to -lncursesw (w/ no special -L path) works fine.
 # FLAGS NEEDED TO BUILD FNC
 FNC_CFLAGS =	${CFLAGS} -Wall -Werror -Wsign-compare -pedantic -std=c99 \
-		-I./lib -I./include -I/usr/include/ncursesw \
+		-I./lib -I./include \
 		-D_XOPEN_SOURCE_EXTENDED -DVERSION=${VERSION} -DHASH=${HASH} \
 		-DDATE="${DATE}"
 
