From 8adbdaccf78012598a73030db7b227e5c180425b Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Tue, 29 Dec 2009 02:40:55 +0200
Subject: [PATCH] miscfuncs.c: (show_status): Show date in ISO 8601 format
Forwarded: not-needed

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 miscfuncs.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/miscfuncs.c b/miscfuncs.c
index fad6fa3..1f858b6 100644
--- miscfuncs.c.orig
+++ miscfuncs.c
@@ -217,10 +217,12 @@ void show_status(void)
 {
     int i, n;
     struct tm *tp;
+/*
     static char *monname[] = {
 	"Jan", "Feb", "Mar", "Apr", "May", "Jun",
 	"Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
     };
+*/
     static time_t tval0;
     time_t tval;
 
@@ -234,9 +236,13 @@ void show_status(void)
     for (i = 0, n = 0 ; i < List_file->num_entries ; ++i) {
 	if (List_file->list[i]->fd > 0) {
 	    tp = localtime(&List_file->list[i]->mtime);
-	    (void) printf("%4d  %2d-%3s-%02d %02d:%02d:%02d  %s\n",
+	    (void) printf("%4d  %d-%02d-%02d %02d:%02d:%02d  %s\n",
 		++n,
-		tp->tm_mday, monname[tp->tm_mon], (tp->tm_year % 100),
+		tp->tm_year + 1900,
+		tp->tm_mon + 1,
+                tp->tm_mday,
+/*              monname[tp->tm_mon], */
+/*              (tp->tm_year % 100), */
 		tp->tm_hour, tp->tm_min, tp->tm_sec,
 		List_file->list[i]->name
 	    );
-- 
1.6.5

