--- nextstep/templates/Info.plist.in.orig	2025-08-16 16:16:20
+++ nextstep/templates/Info.plist.in	2025-08-16 16:16:23
@@ -21,6 +21,39 @@
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+  <key>LSEnvironment</key>
+  <dict>
+    <key>PATH</key>
+    <string>@PATH@</string>
+  </dict>
+	<key>NSBluetoothAlwaysUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to use Bluetooth.</string>
+	<key>NSCalendarsFullAccessUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your calendars.</string>
+	<key>NSCalendarsUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your calendars.</string>
+	<key>NSCameraUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to use the camera.</string>
+	<key>NSContactsUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your contacts.</string>
+	<key>NSLocalNetworkUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access devices on your local network.</string>
+	<key>NSLocationUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to use your location information.</string>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to use your location information while active.</string>
+	<key>NSMicrophoneUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to use the microphone.</string>
+	<key>NSPhotoLibraryAddUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to add to your photo library.</string>
+	<key>NSPhotoLibraryUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your photo library.</string>
+	<key>NSRemindersFullAccessUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your reminders.</string>
+	<key>NSRemindersUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires permission to access your reminders.</string>
+	<key>NSSystemAdministrationUsageDescription</key>
+	<string>Emacs, or a program started from within Emacs, requires elevated permission to perform system administration tasks.</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>English</string>
 	<key>CFBundleDocumentTypes</key>
