--- src/dbusbind.c.orig	2013-03-13 16:52:59.000000000 -0700
+++ src/dbusbind.c	2013-03-13 16:53:29.000000000 -0700
@@ -294,8 +294,6 @@
 	if (!(EQ (bus, QCdbus_system_bus) || EQ (bus, QCdbus_session_bus))) \
 	  XD_SIGNAL2 (build_string ("Wrong bus name"), bus);		\
 	/* We do not want to have an autolaunch for the session bus.  */ \
-	if (EQ (bus, QCdbus_session_bus) && session_bus_address == NULL) \
-	  XD_SIGNAL2 (build_string ("No connection to bus"), bus);	\
       }									\
   } while (0)
 
