--- src/dbusbind.c.orig	2016-10-31 01:50:23.000000000 -0700
+++ src/dbusbind.c	2016-10-31 02:00:51.000000000 -0700
@@ -267,8 +267,6 @@
 	if (!(EQ (bus, QCsystem) || EQ (bus, QCsession)))		\
 	  XD_SIGNAL2 (build_string ("Wrong bus name"), bus);		\
 	/* We do not want to have an autolaunch for the session bus.  */ \
-	if (EQ (bus, QCsession) && session_bus_address == NULL)		\
-	  XD_SIGNAL2 (build_string ("No connection to bus"), bus);	\
       }									\
   } while (0)
 
