--- mindy/interp/Makefile.in.orig	Sat Oct  4 13:21:00 2003
+++ mindy/interp/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -52,8 +52,9 @@
 
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} -I. @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DVERSION=\"@version@\" -DTARGET=\"@d2c_target@\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} -I. @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DVERSION=\"@version@\" -DTARGET=\"@d2c_target@\"
 CFLAGS_NO_WERROR =      `echo $(CFLAGS) | sed 's/-Werror//g'`
+LDFLAGS	=		@LDFLAGS@ 
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
@@ -89,7 +90,7 @@
 EXTERN-INCLUDES = ${SRCDIR}/malloc.inc ${TIME}/time.inc
 
 mindy$(exe_ext): ${OBJS} ../compat/libcompat.a
-	${CC} ${OBJS} ${LIBS} -o ,$@
+	${CC} ${OBJS} ${LDFLAGS} ${LIBS} -o ,$@
 	mv -f ,$@ $@
 
 ${SRCDIR}/parser-tab.c: ${SRCDIR}/parser.y
