--- CMakeLists.txt.old	2014-07-18 17:06:55.000000000 +0900
+++ CMakeLists.txt	2018-06-20 01:05:28.000000000 +0900
@@ -7,10 +7,10 @@
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
 
-find_package(PopplerQt4 0.12.1)
-set_package_properties(PopplerQt4 PROPERTIES DESCRIPTION "A PDF rendering library"
-                       URL "http://poppler.freedesktop.org" TYPE OPTIONAL
-                       PURPOSE "Support for PDF files")
+#find_package(PopplerQt4 0.12.1)
+#set_package_properties(PopplerQt4 PROPERTIES DESCRIPTION "A PDF rendering library"
+#                       URL "http://poppler.freedesktop.org" TYPE OPTIONAL
+#                       PURPOSE "Support for PDF files")
 
 find_package(Taglib)
 set_package_properties(Taglib PROPERTIES DESCRIPTION "Id3 tag reader"
