--- Makefile.orig	2010-09-21 18:05:36.000000000 -0500
+++ Makefile	2011-01-05 01:13:29.000000000 -0600
@@ -4,6 +4,8 @@
 #
 # OPENSSL := ../openssl-0.9.8i
 
+PREFIX=/usr
+
 ifdef RPM_OPT_FLAGS
 OPT_FLAGS := $(RPM_OPT_FLAGS)
 else
@@ -17,10 +19,10 @@
 SSL_CFLAGS += -I$(OPENSSL)/include
 SSL_LDFLAGS += $(OPENSSL)/libssl.a $(OPENSSL)/libcrypto.a
 else
-ifeq ($(wildcard /usr/include/openssl),)
-$(error "No OpenSSL in /usr/include/openssl. Cannot continue");
+ifeq ($(wildcard $(PREFIX)/include/openssl),)
+$(error "No OpenSSL in $(PREFIX)/include/openssl. Cannot continue");
 endif
-SSL_CFLAGS += -I/usr/include/openssl
+SSL_CFLAGS += -I$(PREFIX)/include/openssl
 SSL_LDFLAGS += -lcrypto -lssl
 endif
 
@@ -106,10 +106,10 @@
 	rm -f *~
 
 install: all
-	mkdir -p $(DESTDIR)/usr/bin $(DESTDIR)/usr/libexec
-	install -m0755 openconnect $(DESTDIR)/usr/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/libexec
+	install -m0755 openconnect $(DESTDIR)$(PREFIX)/bin
 ifeq ($(MISSINGPKGS),)
-	install -m0755 nm-openconnect-auth-dialog $(DESTDIR)/usr/libexec
+	install -m0755 nm-openconnect-auth-dialog $(DESTDIR)$(PREFIX)/libexec
 endif
 
 include /dev/null $(wildcard .*.o.dep)
