--- Makefile.orig	2022-03-07 14:11:27.000000000 -0600
+++ Makefile	2022-03-07 14:12:16.000000000 -0600
@@ -124,7 +124,7 @@
    SHARED_LIB         := libdeflate.$(SOVERSION).dylib
    SHARED_LIB_SYMLINK := libdeflate.dylib
    SHARED_LIB_CFLAGS  := -fPIC
-   SHARED_LIB_LDFLAGS := -install_name $(SHARED_LIB)
+   SHARED_LIB_LDFLAGS := -install_name $(LIBDIR)/$(SHARED_LIB)
 
 # Compiling for Android?
 else ifneq ($(findstring -android,$(TARGET_MACHINE)),)
