diff -r 52c432bb18b1 src/org/armedbear/lisp/abcl-contrib.lisp
--- src/org/armedbear/lisp/abcl-contrib.lisp	Sun Oct 23 09:57:18 2011 +0000
+++ xsrc/org/armedbear/lisp/abcl-contrib.lisp	Mon Oct 24 09:49:13 2011 +0200
@@ -8,7 +8,11 @@
 (defun find-system-jar () 
   (dolist (loader (java:dump-classpath))
     (let ((abcl-jar
-	   (find-if (lambda (p) (and (equal (pathname-name p) "abcl")
+	   (find-if (lambda (p) (and 
+                                 (or (equal (pathname-name p) "abcl")
+                                     (equal (pathname-name p) 
+                                            (format nil "abcl-~A" 
+                                                    (lisp-implementation-version))))
 				     (equal (pathname-type p) "jar")))
 		    (rest loader))))
       (when abcl-jar
