--- src/org/armedbear/lisp/asdf.lisp
+++ src/org/armedbear/lisp/asdf.lisp
@@ -7591,9 +7591,9 @@
                  ;; So, when running Genera on macOS, only search /usr/local/share.
                  ((os-genera-p)
                   #+Genera (sys:system-case
-                            (darwin-vlm (mapcar 'parse-unix-namestring '("/usr/local/share/")))
-                            (otherwise (mapcar 'parse-unix-namestring '("/usr/local/share/" "/usr/share/")))))
-                 (t (mapcar 'parse-unix-namestring '("/usr/local/share/" "/usr/share/")))))))
+                            (darwin-vlm (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/" "/usr/local/share/")))
+                            (otherwise (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/" "/usr/local/share/" "/usr/share/")))))
+                 (t (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/" "/usr/local/share/" "/usr/share/")))))))
 
   (defun xdg-config-dirs (&rest more)
     "The preference-ordered set of additional base paths to search for configuration files.
