--- a/rapid/plugin/x/CMakeLists.txt.orig	2023-05-18 09:26:35.000000000 +0800
+++ b/rapid/plugin/x/CMakeLists.txt	2023-05-18 12:13:03.000000000 +0800
@@ -135,6 +135,10 @@
     SET(BOOST_LIB_SOURCES
       ${BOOST_SOURCES_DIR}/libs/atomic/src/lockpool.cpp)
   ENDIF()
+  IF(APPLE AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc|ppc64")
+    SET(BOOST_LIB_SOURCES
+      ${BOOST_SOURCES_DIR}/libs/atomic/src/lockpool.cpp)
+  ENDIF()
   IF (CMAKE_CXX_COMPILER_ID STREQUAL "SunPro")
     SET(BOOST_LIB_SOURCES
       ${BOOST_SOURCES_DIR}/libs/atomic/src/lockpool.cpp)
