--- a/rapid/plugin/x/CMakeLists.txt	2016-11-28 06:32:52.000000000 -0700
+++ b/rapid/plugin/x/CMakeLists.txt	2017-01-11 05:04:43.000000000 -0700
@@ -124,6 +124,10 @@
   # Prevent missing symbols scoped_lock() CTOR/DTOR in mysqlx.so
   #   boost::atomics::detail::lockpool::scoped_lock::~scoped_lock()
   UNSET(BOOST_LIB_SOURCES)
+  IF(APPLE AND CMAKE_OSX_ARCHITECTURES AND CMAKE_OSX_ARCHITECTURES MATCHES i386 AND CMAKE_CXX_COMPILER MATCHES clang)
+    SET(BOOST_LIB_SOURCES
+      ${BOOST_SOURCES_DIR}/libs/atomic/src/lockpool.cpp)
+  ENDIF()
   IF (CMAKE_CXX_COMPILER_ID STREQUAL "SunPro")
     SET(BOOST_LIB_SOURCES
       ${BOOST_SOURCES_DIR}/libs/atomic/src/lockpool.cpp)
