Workaround for https://bugs.mysql.com/bug.php?id=100217
--- a/plugin/innodb_memcached/innodb_memcache/CMakeLists.txt
+++ b/plugin/innodb_memcached/innodb_memcache/CMakeLists.txt
@@ -41,7 +41,7 @@
 # Set extra flags for the C compiler
 IF(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
   SET(CMAKE_C_FLAGS
-    "${CMAKE_C_FLAGS} ${CMAKE_SHARED_LIBRARY_C_FLAGS} -std=gnu99")
+    "${CMAKE_C_FLAGS} ${CMAKE_SHARED_LIBRARY_C_FLAGS} -std=gnu99 -fcommon")
 ENDIF()
 
 SET(MCD__UTILITITY_SOURCES
