--- a/cmake/install_layout.cmake.orig	2016-02-25 22:41:33.000000000 -0430
+++ b/cmake/install_layout.cmake	2016-02-25 22:41:54.000000000 -0430
@@ -293,6 +293,9 @@
 #
 # MACPORTS layout
 #
+INCLUDE_DIRECTORIES(@WORKSRCPATH@/extra/yassl/include)
+LINK_DIRECTORIES(@WORKSRCPATH@/extra/yassl/lib)
+#
 SET(INSTALL_BINDIR_MACPORTS                       "lib/@NAME@/bin")
 SET(INSTALL_SBINDIR_MACPORTS                      "lib/@NAME@/bin")
 SET(INSTALL_SCRIPTDIR_MACPORTS                    "lib/@NAME@/bin")
