--- CMakeLists.txt
+++ CMakeLists.txt
@@ -278,7 +278,7 @@ ENDIF(GATE_DOWNLOAD_BENCHMARKS_DATA)
 # such warning related to clhep/g4 system of units. Additionally,
 # we force the c++11 std. This is mandatory for Geant4 >= 10.2.
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-shadow")
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=@@CXX_STANDARD@@")
 #=========================================================
 
 #=========================================================
