--- pidgin/gtkdocklet-gtk.c.orig	2012-04-30 16:27:39.000000000 +0300
+++ pidgin/gtkdocklet-gtk.c	2012-04-30 17:27:58.000000000 +0300
@@ -47,7 +47,6 @@
 static gboolean
 docklet_gtk_embed_timeout_cb(gpointer data)
 {
-#if !GTK_CHECK_VERSION(2,12,0)
 	if (gtk_status_icon_is_embedded(docklet)) {
 		/* Older GTK+ (<2.12) don't implement the embedded signal, but the
 		   information is still accessable through the above function. */
@@ -57,7 +56,6 @@
 		purple_prefs_set_bool(PIDGIN_PREFS_ROOT "/docklet/gtk/embedded", TRUE);
 	}
 	else
-#endif
 	{
 		/* The docklet was not embedded within the timeout.
 		 * Remove it as a visibility manager, but leave the plugin
