--- pidgin/gtkdocklet-x11.c	2008-08-31 06:36:31.000000000 +0200
+++ pidgin/gtkdocklet-x11.c	2008-10-18 22:18:05.000000000 +0200
@@ -30,14 +30,20 @@
 
 #include "gtkdialogs.h"
 
+#ifdef GDK_WINDOWING_X11
 #include "eggtrayicon.h"
+#endif
 #include "gtkdocklet.h"
 
 #define SHORT_EMBED_TIMEOUT 5000
 #define LONG_EMBED_TIMEOUT 15000
 
 /* globals */
+#ifdef GDK_WINDOWING_X11
 static EggTrayIcon *docklet = NULL;
+#else
+static void* docklet = NULL;
+#endif
 static GtkWidget *image = NULL;
 static GtkTooltips *tooltips = NULL;
 static GdkPixbuf *blank_icon = NULL;
@@ -256,7 +262,8 @@
 		docklet_x11_destroy();
 	}
 
-	docklet = egg_tray_icon_new(PIDGIN_NAME);
+	// removed because uses X11 calls
+	docklet = NULL; // egg_tray_icon_new(PIDGIN_NAME);
 	box = gtk_event_box_new();
 	image = gtk_image_new();
 

