--- src/71_bse/haydock.F90.orig	2014-10-27 01:17:45.000000000 +0100
+++ src/71_bse/haydock.F90	2014-10-27 01:14:25.000000000 +0100
@@ -2229,7 +2229,9 @@
 
  c = czero
 
- !$OMP PARALLEL DO DEFAULT(none) PRIVATE(A,b,c,is1,iv1,ic1,ibnd_coarse,it,allindices,is,iv,ic,idense,ineighbour,ind_with_nb) SHARED(BSp,overlaps,phi,nbnd_coarse,ophi,grid)
+!$OMP PARALLEL DO DEFAULT(none) PRIVATE(A,b,c,is1,iv1,ic1,ibnd_coarse,it,allindices, &
+!$OMP is,iv,ic,idense,ineighbour,ind_with_nb), &
+!$OMP SHARED(BSp,overlaps,phi,nbnd_coarse,ophi,grid)
  do ik_dense = 1,grid%nbz_dense
  ! if( ik_dense is not in my set of k-points)
  !   ! continue
@@ -2292,7 +2294,9 @@
  ABI_MALLOC(b,(grid%ndiv))
  ABI_MALLOC(c,(grid%ndiv))
 
- !$OMP PARALLEL DO COLLAPSE(2) DEFAULT(none) PRIVATE(is, it_coarse, ibnd_coarse, ineighbour, b, c, tmp) SHARED(allp,BSp,interp_factors,ophi,indices)
+ !$OMP PARALLEL DO COLLAPSE(2) DEFAULT(none), &
+ !$OMP PRIVATE(is, it_coarse, ibnd_coarse, ineighbour, b, c, tmp), &
+ !$OMP SHARED(allp,BSp,interp_factors,ophi,indices)
  do is = 1, BSp%nsppol
    do ineighbour = 1,8
 
@@ -2348,7 +2352,10 @@
 
  c = czero
 
- !$OMP PARALLEL DO COLLAPSE(2) DEFAULT(none) PRIVATE(ineighbour,ik_dense,is1,iv1,ic1,is,iv,ic,A,b,c,ibnd_coarse,ik_coarse,it,idense) SHARED(test,overlaps,ophi,BSp,grid,corresp,nbnd_coarse)
+ !$OMP PARALLEL DO COLLAPSE(2) DEFAULT(none), &
+ !$OMP PRIVATE(ineighbour,ik_dense,is1,iv1,ic1,is,iv,ic,A,b,c,ibnd_coarse, &
+ !$OMP ik_coarse,it,idense), &
+ !$OMP SHARED(test,overlaps,ophi,BSp,grid,corresp,nbnd_coarse)
  do ineighbour = 1,8
    do ik_dense = 1,grid%nbz_dense
      do is1 = 1, Bsp%nsppol
