--- fallbacks/sources/bigdft-1.7.0.93/src/Makefile.in.orig	2014-09-16 18:10:18.000000000 -0400
+++ fallbacks/sources/bigdft-1.7.0.93/src/Makefile.in	2014-09-16 18:33:20.000000000 -0400
@@ -485,8 +485,6 @@
 	convolutions \
 	xcPSolver \
 	$(sub_dirs_libconfig) \
-	CUDA \
-	OpenCL \
 	bindings \
 	art \
 	$(sub_dirs_bin)
@@ -506,10 +504,8 @@
 # put all subs
 subs_static_library = \
 	modules/libmodules.a \
-	CUDA/libCUDA.a \
 	convolutions/libconvolutions.a \
 	xcPSolver/libxcPSolver.a \
-	OpenCL/libOpenCL.a \
 	bindings/libbindings.a
 
 out_static_library = \
