--- qwtfunctions.pri.orig
+++ qwtfunctions.pri
@@ -14,7 +14,7 @@
    unset(LIBRARY_NAME)
    LIBRARY_NAME = $$1
 
-   mac:!static:contains(QT_CONFIG, qt_framework) {
+   mac:!static:contains(QWT_CONFIG, QwtFramework) {
       QMAKE_FRAMEWORK_BUNDLE_NAME = $$LIBRARY_NAME
       export(QMAKE_FRAMEWORK_BUNDLE_NAME)
    }
@@ -31,25 +31,5 @@
 }
 
 defineTest(qwtAddLibrary) {
-
-    LIB_NAME = $$1
-
-    unset(LINKAGE)
-
-    if(!debug_and_release|build_pass):CONFIG(debug, debug|release) {
-       win32:LINKAGE = -l$${LIB_NAME}$${QT_LIBINFIX}d
-       mac:LINKAGE = -l$${LIB_NAME}$${QT_LIBINFIX}_debug
-    }
-
-    isEmpty(LINKAGE):LINKAGE = -l$${LIB_NAME}$${QT_LIBINFIX}
-
-    !isEmpty(QMAKE_LSB) {
-        QMAKE_LFLAGS *= --lsb-shared-libs=$${LIB_NAME}$${QT_LIBINFIX}
-    }
-
-    LIBS += $$LINKAGE
-    export(LIBS)
-    export(QMAKE_LFLAGS)
-
-    return(true)
+    qtAddLibrary($$1)
 }
