--- qwtconfig.pri.orig	2013-01-25 19:21:02.000000000 -0500
+++ qwtconfig.pri	2013-02-08 13:30:10.000000000 -0500
@@ -16,26 +16,16 @@
 # Install paths
 ######################################################################
 
-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
+QWT_INSTALL_PREFIX = $$INSTALLBASE
 
-unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION-rc3
-}
-
-win32 {
-    QWT_INSTALL_PREFIX    = C:/Qwt-$$QWT_VERSION-rc3
-}
-
-QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
-QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
-QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib
+QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/share/qwt/doc
+QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include/qwt
 
 ######################################################################
 # Designer plugin
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
-# QWT_INSTALL_PLUGINS   = $${QT_INSTALL_PREFIX}/plugins/designer
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 ######################################################################
 # Features
@@ -49,8 +39,7 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $${QT_INSTALL_PREFIX}/features
+QWT_INSTALL_FEATURES  = $$[QT_INSTALL_DATA]/mkspecs/features
 
 ######################################################################
 # Build the static/shared libraries.
@@ -96,7 +85,7 @@
 # to your qmake project file.
 ######################################################################
 
-#QWT_CONFIG     += QwtMathML
+QWT_CONFIG     += QwtMathML
 
 ######################################################################
 # If you want to build the Qwt designer plugin, 
@@ -130,6 +119,8 @@
 ######################################################################
 
 macx:CONFIG(qt_framework, qt_framework|qt_no_framework) {
-
-    QWT_CONFIG += QwtFramework
-}  
+  QWT_CONFIG += QwtFramework
+  QWT_INSTALL_LIBS = $$[QT_INSTALL_FRAMEWORKS]
+} else {
+  QWT_INSTALL_LIBS = $$[QT_INSTALL_LIBS]
+}
