--- qwtconfig.pri.orig
+++ qwtconfig.pri
@@ -16,20 +16,10 @@
 # Install paths
 ######################################################################

-QWT_INSTALL_PREFIX = $$[QT_INSTALL_PREFIX]
+QWT_INSTALL_PREFIX = $$INSTALLBASE

-unix {
-    QWT_INSTALL_PREFIX    = /usr/local/qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = /usr/local/qwt-$$QWT_VERSION-qt-$$QT_VERSION
-}
-
-win32 {
-    QWT_INSTALL_PREFIX    = C:/Qwt-$$QWT_VERSION
-    # QWT_INSTALL_PREFIX = C:/Qwt-$$QWT_VERSION-qt-$$QT_VERSION
-}
-
-QWT_INSTALL_DOCS      = $${QWT_INSTALL_PREFIX}/doc
-QWT_INSTALL_HEADERS   = $${QWT_INSTALL_PREFIX}/include
+QWT_INSTALL_DOCS      = @INSTALL_DOCS@@
+QWT_INSTALL_HEADERS   = @INSTALL_HEADERS@@
 QWT_INSTALL_LIBS      = $${QWT_INSTALL_PREFIX}/lib

 ######################################################################
@@ -42,7 +32,7 @@
 # runtime environment of designer/creator.
 ######################################################################
 
-QWT_INSTALL_PLUGINS   = $${QWT_INSTALL_PREFIX}/plugins/designer
+QWT_INSTALL_PLUGINS   = $$[QT_INSTALL_PLUGINS]/designer
 
 # linux distributors often organize the Qt installation
 # their way and QT_INSTALL_PREFIX doesn't offer a good
@@ -63,8 +53,7 @@
 # with every Qt upgrade. 
 ######################################################################
 
-QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
-# QWT_INSTALL_FEATURES  = $$[QT_INSTALL_PREFIX]/features
+QWT_INSTALL_FEATURES  = $$[QT_INSTALL_DATA]/mkspecs/features
 
 ######################################################################
 # Build the static/shared libraries.
@@ -110,7 +99,7 @@
 # to your qmake project file.
 ######################################################################
 
-#QWT_CONFIG     += QwtMathML
+QWT_CONFIG     += QwtMathML
 
 ######################################################################
 # If you want to build the Qwt designer plugin, 
@@ -141,7 +130,7 @@
 # Otherwise you have to build them from the examples directory.
 ######################################################################
 
-#QWT_CONFIG     += QwtExamples
+QWT_CONFIG     += QwtExamples
 
 ######################################################################
 # The playground is primarily intended for the Qwt development 
@@ -161,8 +150,11 @@
 
 macx:!static:CONFIG(qt_framework, qt_framework|qt_no_framework) {
 
-    QWT_CONFIG += QwtFramework
-}  
+  QWT_CONFIG += QwtFramework
+  QWT_INSTALL_LIBS = $$[QT_INSTALL_FRAMEWORKS]
+} else {
+  QWT_INSTALL_LIBS = $$[QT_INSTALL_LIBS]
+}
 
 ######################################################################
 # Create and install pc files for pkg-config
