--- a/docs/conf/httpd.conf.in.orig	2016-08-16 18:12:07.000000000 -0500
+++ b/docs/conf/httpd.conf.in	2017-11-13 16:42:24.000000000 -0600
@@ -167,12 +167,18 @@
 </IfModule>
 
 #
-# The following lines prevent .htaccess and .htpasswd files from being 
-# viewed by Web clients. 
+# The following lines prevent .htaccess, .htpasswd and .DS_Store files and 
+# Mac resource forks and named forks from being viewed by Web clients.
 #
-<Files ".ht*">
+<Files ~ "^\.([Hh][Tt]|[Dd][Ss]_[Ss])">
     Require all denied
 </Files>
+<Files "rsrc">
+    Require all denied
+</Files>
+<DirectoryMatch ".*\.\.namedfork">
+    Require all denied
+</DirectoryMatch>
 
 #
 # ErrorLog: The location of the error log file.
