--- /opt/local/var/macports/build/_Users_chris_Projects_MacPorts_portfiles_science_root/root/work/root/configure	2011-12-02 11:42:22.000000000 +0000
+++ configure	2011-12-04 13:57:40.000000000 +0000
@@ -433,19 +433,19 @@
             chklib64file=`ar t $chklib64 | awk '{if (NR == 2) print $1}'`
             logmsg " ar x $chklib64 $chklib64file"
             ar x $chklib64 $chklib64file
-            logmsg " file $filearg $chklib64file | grep '64-bit'"
-            if file $filearg $chklib64file | grep '64-bit' > /dev/null 2>& 1 ; then
+            logmsg " /usr/bin/file $filearg $chklib64file | grep '64-bit'"
+            if /usr/bin/file $filearg $chklib64file | grep '64-bit' > /dev/null 2>& 1 ; then
                 ret=1
             fi
             rm -f $chklib64file
         fi
     else
-        if file $filearg $chklib64 | grep 'ASCII' > /dev/null 2>& 1 ; then
+        if /usr/bin/file $filearg $chklib64 | grep 'ASCII' > /dev/null 2>& 1 ; then
             check_link $chklib64
             ret=$link_result
         else
-            logmsg " file $filearg $chklib64 | grep '64-bit'"
-            if file $filearg $chklib64 | grep '64-bit' > /dev/null 2>& 1 ; then
+            logmsg " /usr/bin/file $filearg $chklib64 | grep '64-bit'"
+            if /usr/bin/file $filearg $chklib64 | grep '64-bit' > /dev/null 2>& 1 ; then
                 ret=1
             fi
         fi
