--- texk/kpathsea/configure.orig	2020-09-01 11:22:00.000000000 +0100
+++ texk/kpathsea/configure	2020-09-01 11:23:36.000000000 +0100
@@ -14057,6 +14057,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdarg.h>
+#include <stdio.h>
                                                   char buf[16];
                                                   va_list ap;
 int
@@ -14086,6 +14087,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 char buf[4] = "abc";
 int
 main ()
