--- texk/ps2pkm/type1.c.orig	Mon Jun 22 23:33:58 2009
+++ texk/ps2pkm/type1.c	Sun Jul 17 22:33:35 2011
@@ -127,6 +127,8 @@
 #define Error {errflag = TRUE; return;}
  
 #define Error0(errmsg) { CC; IfTrace0(TRUE, errmsg); Error;}
+
+#define Error01(errmsg) { CC; IfTrace0(TRUE, errmsg); errflag = TRUE; return -1.0;}
  
 #define Error1(errmsg,arg) { CC; IfTrace1(TRUE, errmsg, arg); Error;}
  
@@ -780,7 +782,7 @@
 static DOUBLE PSFakePop ()
 {
   if (PSFakeTop >= 0) return(PSFakeStack[PSFakeTop--]);
-  else Error0("PSFakePop : Stack empty\n");
+  else Error01("PSFakePop : Stack empty\n");
   /*NOTREACHED*/
 }
