Index: texk/dvipdfm-x/epdf.c
===================================================================
--- texk/dvipdfm-x/epdf.c	(revision 74686)
+++ texk/dvipdfm-x/epdf.c	(revision 74687)
@@ -119,7 +119,7 @@
   xform_info info;
   pdf_obj *contents = NULL, *catalog;
   pdf_obj *page = NULL, *resources = NULL, *markinfo = NULL;
-  pdf_obj *group = NULL;
+  pdf_obj *group = NULL, *group_obj = NULL;
 
   pf = pdf_open(ident, image_file);
   if (!pf)
@@ -155,7 +155,9 @@
   /*
    * Handle page's Group
    */
-  group = pdf_deref_obj(pdf_lookup_dict(page, "Group"));
+  group_obj = pdf_lookup_dict(page, "Group");
+  if (group_obj)
+    group = pdf_import_object(group_obj);
   /*
    * Handle page content stream.
    */
