--- texk/texlive/linked_scripts/Makefile.in.orig	2011-06-20 17:00:53.000000000 -0400
+++ texk/texlive/linked_scripts/Makefile.in	2011-06-20 20:31:41.000000000 -0400
@@ -291,6 +291,7 @@
 #
 # We support both multiplatform and non-multiplatform builds.
 #
+# MacPorts: hard-code link destinations instead of using relative ones
 runscript = $(top_srcdir)/w32_wrapper/runscript.exe
 @WIN32_TRUE@WINEXT = $(EXEEXT)
 all: all-am
@@ -619,12 +620,7 @@
 @WIN32_TRUE@	  echo "$(INSTALL_SCRIPT) '$(runscript)' '$(DESTDIR)$(bindir)/$$link.exe'"; \
 @WIN32_TRUE@	  $(INSTALL_SCRIPT) $(runscript) $(DESTDIR)$(bindir)/$$link.exe || exit 1; \
 @WIN32_TRUE@	done
-@WIN32_FALSE@	case "$(bindir)" in \
-@WIN32_FALSE@	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. install-links;; \
-@WIN32_FALSE@	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. install-links;; \
-@WIN32_FALSE@	  *) echo "strange directory '$(bindir)' for linked scripts" >&2; \
-@WIN32_FALSE@	     exit 1;; \
-@WIN32_FALSE@	esac
+@WIN32_FALSE@	$(MAKE) $(AM_MAKEFLAGS) install-links
 
 .PHONY: install-links
 install-links:
@@ -632,16 +628,16 @@
 	  for s in $(nobase_dist_texmf_scripts_SCRIPTS); do \
 	    target=`basename $$s | sed 's,\.[^/]*$$,,' | tr '[A-Z]' '[a-z]'`; \
 	    rm -f $$target; \
-	    echo "creating link '$$target' -> '$(REL)/texmf/scripts/$$s'"; \
-	    $(LN_S) $(REL)/texmf/scripts/$$s $$target || exit 1; \
+	    echo "creating link '$$target' -> '@@TEXMFMAIN@@/scripts/$$s'"; \
+	    $(LN_S) @@TEXMFMAIN@@/scripts/$$s $$target || exit 1; \
 	  done && \
 	  for s in $(nobase_dist_texmf_dist_scripts_SCRIPTS); do \
 	    target=`basename $$s | tr '[A-Z]' '[a-z]'`; \
 	    echo "$$s" | grep listings-ext.sh >/dev/null \
 	    || target=`echo $$target | sed 's,\.[^/]*$$,,'`; \
 	    rm -f $$target; \
-	    echo "creating link '$$target' -> '$(REL)/texmf-dist/scripts/$$s'"; \
-	    $(LN_S) $(REL)/texmf-dist/scripts/$$s $$target || exit 1; \
+	    echo "creating link '$$target' -> '@@TEXMFDIST@@/scripts/$$s'"; \
+	    $(LN_S) @@TEXMFDIST@@/scripts/$$s $$target || exit 1; \
 	  done && \
 	  for s in $(bin_links); do \
 	    link=`echo $$s | sed 's,.*:,,'`; \
@@ -649,12 +645,7 @@
 	    rm -f $$link; \
 	    echo "creating link '$$link' -> '$$file'"; \
 	    $(LN_S) $$file $$link || exit 1; \
-	  done && \
-	  rm -f man && \
-	  if test -d $(REL)/texmf/doc/man; then \
-	    echo "creating link 'man' -> '$(REL)/texmf/doc/man'" && \
-	    (ln -s $(REL)/texmf/doc/man man || :); \
-	  else :; fi
+	  done
 
 uninstall-hook:
 	@for s in $(nobase_dist_texmf_scripts_SCRIPTS) $(nobase_dist_texmf_dist_scripts_SCRIPTS); do \
