Index: source/texk/dvipdf-x/dpxcrypt.c
===================================================================
--- texk/dvipdfmx/src/dpxcrypt.c	(revision 31699)
+++ texk/dvipdfmx/src/dpxcrypt.c	(revision 31700)
@@ -28,12 +28,6 @@
 # include <sys/types.h>
 #endif
 #include <string.h>
-#ifdef HAVE_INTTYPES_H
-# include <inttypes.h>
-#endif
-#ifdef HAVE_STDINT_H
-# include <stdint.h>
-#endif
 
 #include "dpxcrypt.h"
 
@@ -289,7 +283,7 @@
 #define X(a) do { *p++ = hd->a; *p++ = hd->a >> 8; \
 	          *p++ = hd->a >> 16; *p++ = hd->a >> 24; } while (0)
 #else /* little endian */
-#define X(a) do { *(unsigned long *)p = (*hd).a ; p += 4; } while (0)
+#define X(a) do { *(uint32_t *)p = (*hd).a ; p += sizeof(uint32_t); } while (0)
 #endif
   X(A);
   X(B);
Index: source/texk/dvipdf-x/dpxcrypt.h
===================================================================
--- texk/dvipdfmx/src/dpxcrypt.h	(revision 31699)
+++ texk/dvipdfmx/src/dpxcrypt.h	(revision 31700)
@@ -24,10 +24,16 @@
 #define _DPXCRYPT_H_
 
 #include <stdio.h>
+#ifdef HAVE_INTTYPES_H
+# include <inttypes.h>
+#endif
+#ifdef HAVE_STDINT_H
+# include <stdint.h>
+#endif
 
 /* libgcrypt md5 */
 typedef struct {
-  unsigned long A,B,C,D; /* chaining variables */
+  uint32_t A,B,C,D; /* chaining variables */
   unsigned long nblocks;
   unsigned char buf[64];
   int count;
Index: source/texk/dvipdf-x/dpxcrypt.c
===================================================================
--- texk/xdvipdfmx/src/dpxcrypt.c	(revision 31699)
+++ texk/xdvipdfmx/src/dpxcrypt.c	(revision 31700)
@@ -28,12 +28,6 @@
 # include <sys/types.h>
 #endif
 #include <string.h>
-#ifdef HAVE_INTTYPES_H
-# include <inttypes.h>
-#endif
-#ifdef HAVE_STDINT_H
-# include <stdint.h>
-#endif
 
 #include "dpxcrypt.h"
 
@@ -289,7 +283,7 @@
 #define X(a) do { *p++ = hd->a; *p++ = hd->a >> 8; \
 	          *p++ = hd->a >> 16; *p++ = hd->a >> 24; } while (0)
 #else /* little endian */
-#define X(a) do { *(unsigned long *)p = (*hd).a ; p += 4; } while (0)
+#define X(a) do { *(uint32_t *)p = (*hd).a ; p += sizeof(uint32_t); } while (0)
 #endif
   X(A);
   X(B);
Index: source/texk/dvipdf-x/dpxcrypt.h
===================================================================
--- texk/xdvipdfmx/src/dpxcrypt.h	(revision 31699)
+++ texk/xdvipdfmx/src/dpxcrypt.h	(revision 31700)
@@ -24,10 +24,16 @@
 #define _DPXCRYPT_H_
 
 #include <stdio.h>
+#ifdef HAVE_INTTYPES_H
+# include <inttypes.h>
+#endif
+#ifdef HAVE_STDINT_H
+# include <stdint.h>
+#endif
 
 /* libgcrypt md5 */
 typedef struct {
-  unsigned long A,B,C,D; /* chaining variables */
+  uint32_t A,B,C,D; /* chaining variables */
   unsigned long nblocks;
   unsigned char buf[64];
   int count;
