--- texk/ptexenc/Makefile.in.orig	2011-07-11 18:49:12.000000000 -0400
+++ texk/ptexenc/Makefile.in	2011-07-11 18:49:02.000000000 -0400
@@ -253,7 +253,7 @@
 AM_CFLAGS = $(WARNING_CFLAGS)
 lib_LTLIBRARIES = libptexenc.la
 libptexenc_la_CPPFLAGS = -DMAKE_PTENC_DLL
-libptexenc_la_LDFLAGS = -no-undefined -version-info $(PTEXENC_LT_VERSINFO)
+libptexenc_la_LDFLAGS = -version-info $(PTEXENC_LT_VERSINFO)
 
 # Note: jisx0208.h is a copy of ../../libs/gd/gd-2.0.35/jisx0208.h.
 #
