--- texk/tex4htk/Makefile.in.orig	2010-11-01 20:10:42.000000000 -0400
+++ texk/tex4htk/Makefile.in	2010-11-01 22:31:56.000000000 -0400
@@ -274,8 +274,8 @@
 	wcontext wlatex wmex wtex wtexi wxelatex wxetex xhcontext \
 	xhlatex xhmcontext xhmex xhmlatex xhmmex xhmtex xhmtexi \
 	xhmxelatex xhmxetex xhtex xhtexi xhxelatex xhxetex xv4ht.java
-tex4ht_subdir = texmf-dist/scripts/tex4ht
-tex4htdir = ${prefix}/$(tex4ht_subdir)
+tex4ht_subdir = scripts/tex4ht
+tex4htdir = @@TEXMFDIST@@/$(tex4ht_subdir)
 tex4ht_SCRIPTS = 
 bin_SCRIPTS = 
 all: c-auto.h
@@ -837,25 +837,20 @@
 	  echo "$(INSTALL_SCRIPT) $(srcdir)/$$p '$(DESTDIR)$(tex4htdir)/$$p.sh'"; \
 	  $(INSTALL_SCRIPT) $(srcdir)/$$p "$(DESTDIR)$(tex4htdir)/$$p.sh"; \
 	done
-	case "$(bindir)" in \
-	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. install-links;; \
-	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. install-links;; \
-	  *) echo "strange directory '$(bindir)' for linked scripts" >&2; \
-	     exit 1;; \
-	esac
+	$(MAKE) $(AM_MAKEFLAGS) install-links
 
 .PHONY: install-links
 install-links:
 	@cd $(DESTDIR)$(bindir) && \
 	  for p in $(perl_scripts); do \
 	    rm -f $$p; \
-	    echo "creating link '$$p' -> '$(REL)/$(tex4ht_subdir)/$$p.pl'"; \
-	    $(LN_S) $(REL)/$(tex4ht_subdir)/$$p.pl $$p; \
+	    echo "creating link '$$p' -> '$(tex4htdir)/$$p.pl'"; \
+	    $(LN_S) $(tex4htdir)/$$p.pl $$p; \
 	  done && \
 	  for p in $(shell_scripts); do \
 	    rm -f $$p; \
-	    echo "creating link '$$p' -> '$(REL)/$(tex4ht_subdir)/$$p.sh'"; \
-	    $(LN_S) $(REL)/$(tex4ht_subdir)/$$p.sh $$p; \
+	    echo "creating link '$$p' -> '$(tex4htdir)/$$p.sh'"; \
+	    $(LN_S) $(tex4htdir)/$$p.sh $$p; \
 	  done
 
 uninstall-hook:
