reverted:
--- texk/web2c/pdftexdir/pdftosrc-newpoppler.cc.orig	2018-06-16 15:37:11.000000000 -0700
+++ texk/web2c/pdftexdir/pdftosrc-newpoppler.cc	2018-06-16 15:37:18.000000000 -0700
@@ -68,7 +68,7 @@
     Stream *s;
     Object srcStream, srcName, catalogDict;
     FILE *outfile;
+    const char *outname;
-    char *outname;
     int objnum = 0, objgen = 0;
     bool extract_xref_table = false;
     int c;
