reverted:
--- texk/kpathsea/Makefile.in.orig
+++ texk/kpathsea/Makefile.in
@@ -703,7 +703,6 @@
 	$(am__append_1)
 libkpathsea_la_LDFLAGS = -bindir @bindir@ -no-undefined -version-info $(KPSE_LT_VERSINFO)
 libkpathsea_la_LIBADD = $(LTLIBOBJS)
-BUILT_SOURCES = paths.h
 libkpathsea_la_SOURCES = tex-file.c absolute.c atou.c cnf.c concat.c \
 	concat3.c concatn.c db.c debug.c dir.c elt-dirs.c expand.c \
 	extend-fname.c file-p.c find-suffix.c fn.c fontmap.c hash.c \
@@ -788,7 +787,7 @@
 
 pkgconfigdir = ${libdir}/pkgconfig
 pkgconfig_DATA = kpathsea.pc
+DISTCLEANFILES = paths.h kpathsea.h
-DISTCLEANFILES = paths.h stamp-paths kpathsea.h
 kpseaccess_SOURCES = access.c
 kpsereadlink_SOURCES = readlink.c
 kpsewhich_LDADD = libkpathsea.la
@@ -800,7 +799,7 @@
 dist_web2c_SCRIPTS = mktexdir mktexnam mktexupd
 dist_web2c_DATA = mktex.opt mktexdir.opt mktexnam.opt
 dist_noinst_SCRIPTS = mktexlsr mktexmf mktexpk mktextfm
+dist_noinst_DATA =
-dist_noinst_DATA = texmf.cnf
 bin_links = mktexlsr:texhash
 TESTS = tests/cnfnewline.test tests/kpseaccess.test \
 	tests/kpsereadlink.test tests/kpsestat.test \
@@ -2264,26 +2263,6 @@
 
 $(libkpathsea_la_OBJECTS): paths.h kpathsea.h
 
-paths.h: stamp-paths
-stamp-paths: texmf.cnf bsnl.awk cnf-to-paths.awk
-# ensure grep is not completely broken, e.g., from GREP_OPTIONS.
-	echo a | grep -v b >/dev/null || { echo "*** grep broken, goodbye."; exit 1; }
-#
-# generate paths.h without build machine directories.
-	echo "/* paths.h: Generated from texmf.cnf. */" >paths.tmp
-	$(AWK) -f $(srcdir)/bsnl.awk $(srcdir)/texmf.cnf \
-	| sed -e 's/%.*//' -e 's/^[ 	]*//' -e 's/[ 	]*$$//' \
-	| $(AWK) -f $(srcdir)/cnf-to-paths.awk \
-	>>paths.tmp
-	@if cmp -s paths.h paths.tmp 2>/dev/null; then \
-	  echo "paths.h is unchanged"; \
-	else \
-	  echo "cp paths.tmp paths.h"; \
-	  cp paths.tmp paths.h; \
-	fi
-	rm -f paths.tmp
-	date >$@
-
 kpathsea.h: stamp-kpathsea
 stamp-kpathsea: Makefile paths.h
 	$(AM_V_GEN)rm -f $@; \
