--- texk/texlive/context/mtxrun.orig	2012-06-14 18:34:57.000000000 -0400
+++ texk/texlive/context/mtxrun	2012-06-14 18:35:30.000000000 -0400
@@ -12340,7 +12340,7 @@
     resolvers.luacnfspec = environment.default_texmfcnf
 else
  -- resolvers.luacnfspec = "selfautoparent:texmf{-local,-context,}/web2c"
-    resolvers.luacnfspec = "{selfautoloc:,selfautodir:,selfautoparent:}{,/texmf{-local,}/web2c}"
+    resolvers.luacnfspec = '@@TEXMFSYSCONFIG'
 end
 
 resolvers.luacnfspec = 'home:texmf/web2c;' .. resolvers.luacnfspec
