Don't accidentally create flat-namespace dylibs on Yosemite (#44596).
--- dist/configure.orig	2014-01-24 13:09:41.000000000 -0600
+++ dist/configure	2017-02-10 05:03:23.000000000 -0600
@@ -10164,7 +10164,7 @@
       case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
 	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
-	10.[012]*)
+	10.[012][,.]*)
 	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
 	10.*)
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
