diff --git CMake/Findpugixml.cmake CMake/Findpugixml.cmake
new file mode 100644
index 0000000000..b715568503
--- /dev/null
+++ CMake/Findpugixml.cmake
@@ -0,0 +1,13 @@
+find_path(pugixml_INCLUDE_DIRS
+    "pugixml.hpp"
+)
+
+find_library(pugixml_LIBRARIES
+    NAMES pugixml
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(pugixml
+    REQUIRED_VARS pugixml_LIBRARIES pugixml_INCLUDE_DIRS
+)
+
