--- ThirdParty/libxml2/vtklibxml2/dict.c
+++ ThirdParty/libxml2/vtklibxml2/dict.c	2024-11-29 05:44:11.000000000 +0800
@@ -929,12 +929,9 @@
   #include <windows.h>
   #include <bcrypt.h>
 #elif defined(HAVE_GETENTROPY)
-  #ifdef HAVE_UNISTD_H
     #include <unistd.h>
-  #endif
-  #ifdef HAVE_SYS_RANDOM_H
+    #include <sys/types.h>
     #include <sys/random.h>
-  #endif
 #else
   #include <time.h>
 #endif
