--- wand/api.py.orig	2015-11-29 22:02:15.000000000 +0300
+++ wand/api.py	2015-12-24 16:13:28.000000000 +0300
@@ -58,7 +58,7 @@
     versions = '', '-6', '-Q16', '-Q8', '-6.Q16'
     options = '', 'HDRI', 'HDRI-2'
     system = platform.system()
-    magick_home = os.environ.get('MAGICK_HOME')
+    magick_home = os.environ.get('MAGICK_HOME') or '@PREFIX@'
 
     if system == 'Windows':
         # ImageMagick installers normally install coder and filter DLLs in
