--- src/fetch/fetchmanager.h.orig	2012-07-14 21:55:34.000000000 +0300
+++ src/fetch/fetchmanager.h	2012-08-06 18:57:16.000000000 +0300
@@ -65,6 +65,8 @@
   typedef QString (*FETCHER_ICON_FN)(void);
   typedef StringHash (*FETCHER_OPTIONALFIELDS_FN)(void);
   typedef ConfigWidget* (*FETCHER_CONFIGWIDGET_FN)(QWidget*);
+
+public:
   struct FetcherFunction  {
     FETCHER_CREATE_FN create;
     FETCHER_NAME_FN name;
@@ -73,7 +75,6 @@
     FETCHER_CONFIGWIDGET_FN configWidget;
   };
 
-public:
   static Manager* self() { if(!s_self) s_self = new Manager(); return s_self; }
 
   ~Manager();

