--- setup.py.orig	2012-04-17 20:03:37.000000000 -0500
+++ setup.py	2012-04-17 20:04:08.000000000 -0500
@@ -258,18 +258,6 @@
         'shortcuts': ["quodlibet.desktop", "exfalso.desktop"],
         'man_pages': ["man/quodlibet.1", "man/exfalso.1"],
         'gobject_modules': [
-                    GObjectExtension("quodlibet._mmkeys",
-                            "gobject_ext/mmkeys/mmkeys.defs",
-                            "gobject_ext/mmkeys/mmkeys.override",
-                            ["gobject_ext/mmkeys/mmkeys.c",
-                            "gobject_ext/mmkeys/mmkeysmodule.c"],
-                            include_dirs=["gobject_ext/mmkeys"]),
-                    GObjectExtension("quodlibet._trayicon",
-                            "gobject_ext/trayicon/trayicon.defs",
-                            "gobject_ext/trayicon/trayicon.override",
-                            ["gobject_ext/trayicon/eggtrayicon.c",
-                             "gobject_ext/trayicon/trayiconmodule.c"],
-                            include_dirs=["gobject_ext/trayicon"])
                     ],
         }
     if os.name == 'nt':
