--- cpack/macos.cmake.orig
+++ cpack/macos.cmake
@@ -25,9 +25,11 @@ RUNTIME DESTINATION . COMPONENT Runtime
     BUNDLE DESTINATION . COMPONENT Runtime
   )
 
-  install(CODE "
-    include(BundleUtilities)
-    fixup_bundle(\"\${CMAKE_INSTALL_PREFIX}/${CPACK_BUNDLE_NAME}.app\"  \"\"  \"\")
-    execute_process(COMMAND codesign --deep --force --sign - \"\${CMAKE_INSTALL_PREFIX}/${CPACK_BUNDLE_NAME}.app\")
-    " COMPONENT Runtime)
+  if(MACOS_BUNDLE_FIXUP_AND_SIGN)
+    install(CODE "
+      include(BundleUtilities)
+      fixup_bundle(\"\${CMAKE_INSTALL_PREFIX}/${CPACK_BUNDLE_NAME}.app\"  \"\"  \"\")
+      execute_process(COMMAND codesign --deep --force --sign - \"\${CMAKE_INSTALL_PREFIX}/${CPACK_BUNDLE_NAME}.app\")
+      " COMPONENT Runtime)
+  endif()
 endif()
--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -7,6 +7,7 @@ option(PREFER_SYSTEM_Lua "Prefer system Lua" ON)
 option(PREFER_SYSTEM_XDG "Prefer system XDG" ON)
 option(ALLOW_DEV "Enable some development/debug features" OFF)
 option(BUILD_MACOS_BUNDLE "Build xmoto as a macOS Bundle" OFF)
+option(MACOS_BUNDLE_FIXUP_AND_SIGN "Execute fixup_bundle and codesign for a macOS bundle" OFF)
 
 if(CMAKE_GENERATOR MATCHES "Ninja")
   message("-- Using Ninja, try to colorize")
