--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -517,7 +517,6 @@ target_link_libraries(xmoto PUBLIC
   ode
   glad
   ${CMAKE_DL_LIBS}
-  ${OPENGL_LIBRARIES}
   ${PNG_LIBRARIES}
   ${SDL2_LIBRARIES}
   ${SDL2_MIXER_LIBRARY}
@@ -530,6 +530,14 @@ target_link_libraries(xmoto PUBLIC
   $<$<PLATFORM_ID:Darwin>:${COREFOUNDATION_LIBRARY}>
 )
 
+# Fixes an on-boot issue on OS X 10.9, at least.
+if(APPLE)
+  set(OPENGL_LIBRARIES "-framework OpenGL")
+  target_link_libraries(xmoto PUBLIC ${OPENGL_LIBRARIES})
+else()
+  target_link_libraries(xmoto PUBLIC OpenGL::GL)
+endif()
+
 set_property(TARGET xmoto PROPERTY CXX_STANDARD 11)
 # strip release builds
 set_target_properties(xmoto PROPERTIES LINK_FLAGS_RELEASE -s)
