--- src/meson.build.orig	2024-03-24 03:16:15.000000000 +0800
+++ src/meson.build	2024-08-10 09:06:26.000000000 +0800
@@ -12,7 +12,7 @@
 python = import('python')
 
 conf = configuration_data()
-conf.set('PYTHON', python.find_installation('python3').path())
+conf.set('PYTHON', python.find_installation('@PYTHON@').path())
 conf.set('VERSION', meson.project_version())
 conf.set('localedir', join_paths(get_option('prefix'), get_option('localedir')))
 conf.set('pkgdatadir', pkgdatadir)
