--- configure.orig	2005-07-23 19:55:57.000000000 +0900
+++ configure	2005-07-23 19:57:01.000000000 +0900
@@ -1277,11 +1277,11 @@
 
 # Building paths
 pathtmp="$PATH"
-PATH="$HOME/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
+PATH="${exec_prefix}/bin:${exec_prefix}/sbin:/usr/bin:/usr/sbin:/bin:/sbin"
 PATH="$PATH:/usr/ccs/bin:/usr/ucb:$pathtmp"
-LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LIBRARY_PATH"
-LD_LIBRARY_PATH="$HOME/lib:/usr/local/lib:$LD_LIBRARY_PATH"
-CPATH="$HOME/include:/usr/local/include:$CPATH"
+LIBRARY_PATH="${exec_prefix}/lib:$LIBRARY_PATH"
+LD_LIBRARY_PATH="${exec_prefix}/lib:$LD_LIBRARY_PATH"
+CPATH="${exec_prefix}/include:$CPATH"
 export PATH LIBRARY_PATH LD_LIBRARY_PATH CPATH
 
 
