--- Makefile	2021-06-13 12:26:27.000000000 -0400
+++ Makefile.new	2021-06-13 14:03:02.000000000 -0400
@@ -40,20 +40,13 @@
 CFLAGS ?= -O2
 CXXFLAGS ?= -O2
 
-# try to find right names for OpenCV libs 
+BASEDIR=@@MACPORTS_PREFIX@@
 
-ifeq ($(GCC_OPENCV_LIBS),auto)
-ifeq ($(shell echo 'int main(){}' | gcc -xc -lopencv_imgcodecs - && ( rm -f a.out ; echo "OK")),OK)
-  GCC_OPENCV_LIBS:=-lopencv_core -lopencv_imgproc -lopencv_imgcodecs
-else ifeq ($(shell echo 'int main(){}' | gcc -xc -lopencv_core - && ( rm -f a.out ; echo "OK")),OK)
-  GCC_OPENCV_LIBS:=-lopencv_core -lopencv_highgui -lopencv_imgproc
-else
-  GCC_OPENCV_LIBS:=-lcv -lhighgui -lcxcore
-endif
-endif
+OPENCV_LIBDIR=$(BASEDIR)/lib/opencv3
+OPENCV_PKGCONFIG=$(BASEDIR)/lib/opencv3/pkgconfig/opencv.pc
 
-GCC_OPENCV ?= $(shell pkg-config --cflags opencv)
-GCC_OPENCV_LIBS ?= $(shell pkg-config --libs opencv)
+GCC_OPENCV=$(shell pkg-config --cflags $(OPENCV_PKGCONFIG))
+GCC_OPENCV_LIBS=$(shell pkg-config --libs $(OPENCV_PKGCONFIG))
 GCC_PDF ?= $(shell pkg-config --cflags --libs cairo pangocairo poppler-glib)
 GCC_POPPLER ?= $(shell pkg-config --cflags --libs poppler-glib gio-2.0)
 
