--- shapely/geos.py.orig	2014-01-09 16:00:29.000000000 -0500
+++ shapely/geos.py	2014-01-09 16:00:51.000000000 -0500
@@ -66,7 +66,7 @@
             # The Framework build from Kyng Chaos:
             "/Library/Frameworks/GEOS.framework/Versions/Current/GEOS",
             # macports
-            '/opt/local/lib/libgeos_c.dylib',
+            '@PREFIX@/lib/libgeos_c.dylib',
         ]
     _lgeos = load_dll('geos_c', fallbacks=alt_paths)
     free = load_dll('c').free
