--- shapely/geos.py.orig	2015-03-16 02:09:25.000000000 -0400
+++ shapely/geos.py	2015-04-20 09:34:35.000000000 -0400
@@ -76,7 +76,7 @@
                 # The Framework build from Kyng Chaos
                 "/Library/Frameworks/GEOS.framework/Versions/Current/GEOS",
                 # macports
-                '/opt/local/lib/libgeos_c.dylib',
+                '@PREFIX@/lib/libgeos_c.dylib',
             ]
         _lgeos = load_dll('geos_c', fallbacks=alt_paths)
 
