--- shapely/geos.py.orig	2012-06-27 13:42:14.000000000 -0400
+++ shapely/geos.py	2012-07-06 09:38:36.000000000 -0400
@@ -54,7 +54,7 @@
             # The Framework build from Kyng Chaos:
             "/Library/Frameworks/GEOS.framework/Versions/Current/GEOS",
             # macports
-            '/opt/local/lib/libgeos_c.dylib',
+            '@PREFIX@/lib/libgeos_c.dylib',
     ]
     _lgeos = load_dll('geos_c', fallbacks=alt_paths)
     free = load_dll('c').free
