From 13eb912483be1b044439a764c6f744773c6197d1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 27 Jun 2023 11:52:28 +0800
Subject: [PATCH 1/2] Use endian macros to set endianness

---
 src/crc32c/crc32c_config.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/crc32c/crc32c_config.h src/crc32c/crc32c_config.h
index fa8ba50..aa63445 100644
--- src/crc32c/crc32c_config.h
+++ src/crc32c/crc32c_config.h
@@ -6,7 +6,11 @@
 #define CRC32C_CRC32C_CONFIG_H_
 
 // Define to 1 if building for a big-endian platform.
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#define BYTE_ORDER_BIG_ENDIAN 1
+#else
 #define BYTE_ORDER_BIG_ENDIAN 0
+#endif
 
 // Define to 1 if the compiler has the __builtin_prefetch intrinsic.
 #define HAVE_BUILTIN_PREFETCH 1
-- 
2.41.0

