From 3849b1960440e0c3540c537d761a3838b4770513 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 27 Jun 2023 12:00:56 +0800
Subject: [PATCH 2/2] Fix HAVE_MM_PREFETCH

---
 src/crc32c/crc32c_config.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/crc32c/crc32c_config.h src/crc32c/crc32c_config.h
index aa63445..8eb6c87 100644
--- src/crc32c/crc32c_config.h
+++ src/crc32c/crc32c_config.h
@@ -16,7 +16,11 @@
 #define HAVE_BUILTIN_PREFETCH 1
 
 // Define to 1 if targeting X86 and the compiler has the _mm_prefetch intrinsic.
+#if defined(__i386__) || defined(__x86_64__)
 #define HAVE_MM_PREFETCH 1
+#else
+#define HAVE_MM_PREFETCH 0
+#endif
 
 // Define to 1 if targeting X86 and the compiler has the _mm_crc32_u{8,32,64}
 // intrinsics.
-- 
2.41.0

