diff --git CMakeLists.txt CMakeLists.txt
index 63f903c..71c494b 100644
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -730,7 +730,7 @@ feature_option(build_tests "build tests" OFF)
 feature_option(build_examples "build examples" OFF)
 feature_option(build_tools "build tools" OFF)
 feature_option(python-bindings "build python bindings" OFF)
-feature_option(python-egg-info "generate python egg info" OFF)
+feature_option(python-dist-info "generate python dist info" OFF)
 feature_option(python-install-system-dir "Install python bindings to the system installation directory rather than the CMake installation prefix" OFF)
 
 # these options require existing target

diff --git bindings/python/CMakeLists.txt bindings/python/CMakeLists.txt
index 689c274..f9c5906 100644
--- bindings/python/CMakeLists.txt.orig
+++ bindings/python/CMakeLists.txt
@@ -110,7 +110,7 @@ message(STATUS "Python 3 extension suffix: ${Python3_SOABI}")
 
 install(TARGETS python-libtorrent DESTINATION "${_PYTHON3_SITE_ARCH}")
 
-if (python-egg-info)
+if (python-dist-info)
 	set(SETUP_PY_IN "${CMAKE_CURRENT_SOURCE_DIR}/setup.py.cmake.in")
 	set(SETUP_PY    "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
 	set(OUTPUT      "${CMAKE_CURRENT_BINARY_DIR}/timestamp")
@@ -119,12 +119,15 @@ if (python-egg-info)
 	configure_file(${SETUP_PY_IN} ${SETUP_PY} @ONLY)
 
 	add_custom_command(OUTPUT ${OUTPUT}
-		COMMAND ${Python3_EXECUTABLE} ${SETUP_PY} egg_info
+		COMMAND CC=${CMAKE_C_COMPILER} CXX=${CMAKE_CXX_COMPILER} ${Python3_EXECUTABLE} ${SETUP_PY} dist_info
+		COMMAND echo 'libtorrent' > "${CMAKE_CURRENT_BINARY_DIR}/libtorrent-${CMAKE_PROJECT_VERSION}.dist-info/top_level.txt"
 		COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
 		DEPENDS ${DEPS}
 	)
 
-	add_custom_target(python_bindings ALL DEPENDS ${OUTPUT})
+	if (NOT TARGET python_bindings)
+		add_custom_target(python_bindings ALL DEPENDS ${OUTPUT})
+	endif()
 
-	install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/libtorrent.egg-info" DESTINATION "${_PYTHON3_SITE_ARCH}")
+	install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/libtorrent-${CMAKE_PROJECT_VERSION}.dist-info" DESTINATION "${_PYTHON3_SITE_ARCH}")
 endif()
