--- bindings/python/CMakeLists.txt.old	2022-04-17 17:46:50.000000000 -0400
+++ bindings/python/CMakeLists.txt	2022-04-17 17:50:36.000000000 -0400
@@ -116,7 +116,7 @@
 	configure_file(${SETUP_PY_IN} ${SETUP_PY} @ONLY)
 
 	add_custom_command(OUTPUT ${OUTPUT}
-		COMMAND ${Python3_EXECUTABLE} ${SETUP_PY} egg_info
+		COMMAND CC=${CMAKE_C_COMPILER} CXX=${CMAKE_CXX_COMPILER} ${Python3_EXECUTABLE} ${SETUP_PY} egg_info
 		COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
 		DEPENDS ${DEPS}
 	)
