--- bindings/python/src/session.cpp.orig	2016-04-19 12:56:30.000000000 -0700
+++ bindings/python/src/session.cpp	2016-04-19 12:57:29.000000000 -0700
@@ -810,7 +810,7 @@
 #ifndef TORRENT_NO_DEPRECATE
         .def("add_feed", &add_feed)
         .def("status", allow_threads(&lt::session::status))
-        .def("settings", &session_get_settings)
+        .def("settings", &lt::session::settings)
         .def("set_settings", &session_set_settings)
 #endif
         .def("get_settings", &session_get_settings)
