--- configure.ac.orig	2019-04-30 12:55:17.000000000 -0700
+++ configure.ac	2019-08-12 18:34:59.000000000 -0700
@@ -445,6 +445,16 @@
       AC_MSG_RESULT([yes])
 
       AM_PATH_PYTHON([2.4], [], AC_MSG_ERROR([Python interpreter not found.]))
+
+      AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print(sys.prefix);'`])
+      AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print(sys.exec_prefix)'`])
+
+      am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+      am_cv_python_pyexecdir=$PYTHON_EXEC_PREFIX/lib/python$PYTHON_VERSION/site-packages
+
+      AC_SUBST([pythondir], [$am_cv_python_pythondir])
+      AC_SUBST([pyexecdir], [$am_cv_python_pyexecdir])
+
       AX_PYTHON_DEVEL([>= '2.4'])
       AX_BOOST_PYTHON()
 
