--- pyproject.toml.orig	2022-03-23 10:12:03.000000000 +0200
+++ pyproject.toml	2022-03-25 11:27:17.000000000 +0200
@@ -1,8 +1,6 @@
 [build-system]
 requires = [
-  "setuptools >= 42.0.0", # required by pyproject+setuptools_scm integration
-  "setuptools_scm[toml] >= 3.5.0", # required for "no-local-version" scheme
-  "setuptools_scm_git_archive >= 1.0",
+  "setuptools >= 42.0.0",
   "wheel",
 ]
 build-backend = "setuptools.build_meta"

--- setup.cfg.orig	2022-11-21 02:17:27.000000000 +0200
+++ setup.cfg	2022-11-22 21:17:08.322183925 +0200
@@ -1,6 +1,7 @@
 # spell-checker:ignore apidoc programoutput sphinxcontrib
 [metadata]
 name = ansible-lint
+version = @@VERSION@@
 url = https://github.com/ansible/ansible-lint
 project_urls =
   Bug Tracker = https://github.com/ansible/ansible-lint/issues
@@ -55,7 +56,6 @@
   lint
 
 [options]
-use_scm_version = True
 python_requires = >=3.8
 package_dir =
   = src

