--- /dev/null	2010-03-29 00:03:17.000000000 +0400
+++ siteconf.py	2010-03-28 21:17:52.000000000 +0400
@@ -0,0 +1,13 @@
+BOOST_INC_DIR = ['@@PREFIX@@/include']
+BOOST_LIB_DIR = ['@@PREFIX@@/lib']
+BOOST_COMPILER = '@@GCC@@'
+BOOST_PYTHON_LIBNAME = ['boost_python-mt']
+BOOST_THREAD_LIBNAME = ['boost_thread-mt']
+CL_TRACE = False
+CL_ENABLE_GL = True
+SHIPPED_CL_HEADERS = False
+CL_INC_DIR = []
+CL_LIB_DIR = []
+CL_LIBNAME = []
+CXXFLAGS = []
+LDFLAGS = ['-Wl,-framework,OpenCL']
