--- ctypes_configure/cbuild.py.orig	2016-03-10 02:34:50.000000000 +1100
+++ ctypes_configure/cbuild.py	2016-03-14 01:56:29.000000000 +1100
@@ -208,7 +208,7 @@
     library_dirs = list(eci.library_dirs)
     if (sys.platform == 'darwin' or    # support Fink & Darwinports
             sys.platform.startswith('freebsd')):
-        for s in ('/sw/', '/opt/local/', '/usr/local/'):
+        for s in ['__PREFIX__']:
             if s + 'include' not in include_dirs and \
                os.path.exists(s + 'include'):
                 include_dirs.append(s + 'include')
@@ -383,7 +383,7 @@
             self.link_extra += ['/DEBUG'] # generate .pdb file
         if (sys.platform == 'darwin' or    # support Fink & Darwinports
                 sys.platform.startswith('freebsd')):
-            for s in ('/sw/', '/opt/local/', '/usr/local/'):
+            for s in ['__PREFIX__']:
                 if s + 'include' not in self.include_dirs and \
                    os.path.exists(s + 'include'):
                     self.include_dirs.append(s + 'include')
