--- pypy/translator/platform/darwin.py.orig	2011-08-26 11:44:13.000000000 +1000
+++ pypy/translator/platform/darwin.py	2011-08-26 11:44:56.000000000 +1000
@@ -70,10 +70,10 @@
 
 class Darwin_i386(Darwin):
     name = "darwin_i386"
-    link_flags = ('-arch', 'i386')
-    cflags = ('-arch', 'i386', '-O3', '-fomit-frame-pointer')
+    link_flags = ('-m32')
+    cflags = ('-m32', '-O3', '-fomit-frame-pointer')
 
 class Darwin_x86_64(Darwin):
     name = "darwin_x86_64"
-    link_flags = ('-arch', 'x86_64')
-    cflags = ('-arch', 'x86_64', '-O3', '-fomit-frame-pointer')
+    link_flags = ('-m64')
+    cflags = ('-m64', '-O3', '-fomit-frame-pointer')
