--- lib_pypy/cffi/ffiplatform.py.orig	2016-03-19 04:52:25.000000000 +1100
+++ lib_pypy/cffi/ffiplatform.py	2016-03-22 14:49:10.000000000 +1100
@@ -47,6 +47,8 @@
     options['force'] = ('ffiplatform', True)
     options['build_lib'] = ('ffiplatform', tmpdir)
     options['build_temp'] = ('ffiplatform', tmpdir)
+    options['include_dirs'] = ('ffiplatform', '__PREFIX__/include')
+    options['library_dirs'] = ('ffiplatform', '__PREFIX__/lib')
     #
     try:
         old_level = distutils.log.set_threshold(0) or 0
