--- lfd_legacy_encrypt.c	2008-01-07 16:35:33.000000000 -0600
+++ lfd_legacy_encrypt.c	2008-08-31 19:31:25.000000000 -0500
@@ -49,14 +49,9 @@
 
 #ifdef HAVE_SSL
 
-#ifndef __APPLE_CC__
 /* OpenSSL includes */
 #include <openssl/md5.h>
 #include <openssl/blowfish.h>
-#else /* YAY - We're MAC OS */
-#include <sys/md5.h>
-#include <crypto/blowfish.h>
-#endif  /* __APPLE_CC__ */
 
 #define ENC_BUF_SIZE VTUN_FRAME_SIZE + 16 
 #define ENC_KEY_SIZE 16
