--- llvmlite/binding/targets.py.orig	2018-08-13 23:13:18.000000000 +0300
+++ llvmlite/binding/targets.py	2018-08-13 23:14:33.000000000 +0300
@@ -64,7 +64,8 @@
         content = str(out)
         if content:  # protect against empty string
             for feat in content.split(','):
-                outdict[feat[1:]] = flag_map[feat[0]]
+                if feat:  # protect against empty features in list
+                    outdict[feat[1:]] = flag_map[feat[0]]
         return outdict
 
 
