--- ffi/Makefile.osx.orig	2023-04-23 21:20:38.000000000 +0200
+++ ffi/Makefile.osx	2023-04-23 21:21:04.000000000 +0200
@@ -1,9 +1,9 @@
 
-CXX = clang++
+CXX = %%CXX%%
 CXXFLAGS = $(LLVM_CXXFLAGS)
 # Only export the LLVMPY symbols we require and exclude everything else.
 EXPORT = "-Wl,-exported_symbol,_LLVMPY_*"
-LDFLAGS :=  $(LDFLAGS) $(EXPORT) $(LLVM_LDFLAGS)
+LDFLAGS :=  $(LDFLAGS) $(EXPORT) $(LLVM_LDFLAGS) %%MP_EXTRA_LDFLAGS%%
 LIBS = $(LLVM_LIBS)
 INCLUDE = core.h
 SRC = assembly.cpp bitcode.cpp core.cpp initfini.cpp module.cpp value.cpp \
