--- CMakeLists.txt.orig	2010-05-07 15:21:05.000000000 -0400
+++ CMakeLists.txt	2010-05-07 15:22:04.000000000 -0400
@@ -22,7 +22,7 @@
   macro_optional_find_package( QImageBlitz )
   macro_log_feature( QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "kdesupport" FALSE "" "Required for building SuperKaramba.")
 endif(X11_FOUND)
-macro_optional_find_package( GMP )
+#macro_optional_find_package( GMP )
 macro_log_feature( GMP_FOUND "GMP" "The GNU Multiple Precision Arithmetic Library" "http://gmplib.org/" FALSE "" "Required for building KCalc.")
 macro_optional_find_package( QCA2 )
 macro_log_feature( QCA2_FOUND "QCA2" "Qt Cryptographic Architecture" "http://delta.affinix.com/qca" FALSE "2.0.0" "Needed for most of the algorithms of the checksum tool in Okteta." )
