--- CMakeLists.txt.orig	2010-05-07 15:21:05.000000000 -0400
+++ CMakeLists.txt	2010-05-07 15:21:21.000000000 -0400
@@ -19,7 +19,7 @@
 
 # --- check for dependencies ---
 if(X11_FOUND)
-  macro_optional_find_package( QImageBlitz )
+  #macro_optional_find_package( QImageBlitz )
   macro_log_feature( QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "kdesupport" FALSE "" "Required for building SuperKaramba.")
 endif(X11_FOUND)
 macro_optional_find_package( GMP )
