--- src/auto/configure.orig	2008-07-24 06:40:36.000000000 -0600
+++ src/auto/configure	2009-01-18 21:15:03.000000000 -0700
@@ -15458,7 +15458,7 @@
 if test "$enable_multibyte" = "yes"; then
   cflags_save=$CFLAGS
   ldflags_save=$LDFLAGS
-  if test -n "$x_includes" ; then
+  if test -n "$x_includes" && test "$x_includes" != "NONE"; then
     CFLAGS="$CFLAGS -I$x_includes"
     LDFLAGS="$X_LIBS $LDFLAGS -lX11"
     { $as_echo "$as_me:$LINENO: checking whether X_LOCALE needed" >&5
