--- giscanner/shlibs.py.orig	2012-05-19 04:01:42.000000000 +0200
+++ giscanner/shlibs.py	2012-05-19 04:03:23.000000000 +0200
@@ -49,7 +49,7 @@
 # is crazy enough to name a library liblib<foo> when lib<foo> exists.
 #
 def _ldd_library_pattern(library_name):
-    return re.compile("(?<![A-Za-z0-9_-])(lib*%s[^A-Za-z0-9_-][^\s\(\)]*)"
+    return re.compile("([^\s]*lib*%s[^A-Za-z0-9_-][^\s\(\)]*)"
                       % re.escape(library_name))
 
 # This is a what we do for non-la files. We assume that we are on an
