diff --git giscanner/ccompiler.py giscanner/ccompiler.py
index c003828..87f98d2 100644
--- giscanner/ccompiler.py
+++ giscanner/ccompiler.py
@@ -126,7 +126,7 @@ class CCompiler(object):
 
             if not libtool:
                 # https://bugzilla.gnome.org/show_bug.cgi?id=625195
-                args.append('-Wl,-rpath,.')
+#               args.append('-Wl,-rpath,.')
 
                 # Ensure libraries are always linked as we are going to use ldd to work
                 # out their names later
@@ -152,12 +152,12 @@ class CCompiler(object):
                 args.append('-libpath:' + library_path)
             else:
                 args.append('-L' + library_path)
-                if os.path.isabs(library_path):
-                    if libtool:
-                        args.append('-rpath')
-                        args.append(library_path)
-                    else:
-                        args.append('-Wl,-rpath,' + library_path)
+#               if os.path.isabs(library_path):
+#                   if libtool:
+#                       args.append('-rpath')
+#                       args.append(library_path)
+#                   else:
+#                       args.append('-Wl,-rpath,' + library_path)
 
             runtime_paths.append(library_path)
 
