--- QGLViewer/QGLViewer.pro.orig	2010-07-05 11:27:55.000000000 +0200
+++ QGLViewer/QGLViewer.pro	2010-07-05 11:29:09.000000000 +0200
@@ -232,9 +232,6 @@
 #		--  M a c O S X  --
 #       -------------------
 macx|darwin-g++ {
-  # This setting creates a Mac framework. Comment out this line to create a dylib instead.
-  CONFIG *= lib_bundle
-
   include.files *= qglviewer.icns
 
   lib_bundle {
@@ -252,7 +249,7 @@
     # Framework already installed, with includes
     INSTALLS -= include target
   } else {
-    #QMAKE_LFLAGS_SONAME  = -Wl,-install_name,libQGLViewer.dylib
+    QMAKE_LFLAGS_SONAME  = -Wl,-install_name,@@PREFIX@@/lib/
   }
 
   # GLUT for Macintosh architecture
