--- net/tap-bsd.c.orig	2012-09-05 16:03:06.000000000 +0200
+++ net/tap-bsd.c	2012-09-07 17:53:09.000000000 +0200
@@ -43,7 +43,7 @@
     struct stat s;
 #endif
 
-#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__OpenBSD__) || defined(__APPLE__)
     /* if no ifname is given, always start the search from tap0/tun0. */
     int i;
     char dname[100];
