--- configure.orig	2014-04-17 15:44:44.000000000 +0200
+++ configure	2014-04-19 15:04:58.000000000 +0200
@@ -577,10 +577,6 @@
   bsd="yes"
   darwin="yes"
   LDFLAGS_SHARED="-bundle -undefined dynamic_lookup"
-  if [ "$cpu" = "x86_64" ] ; then
-    QEMU_CFLAGS="-arch x86_64 $QEMU_CFLAGS"
-    LDFLAGS="-arch x86_64 $LDFLAGS"
-  fi
   cocoa="yes"
   audio_drv_list="coreaudio"
   audio_possible_drivers="coreaudio sdl fmod"
