--- configure.orig	2009-05-21 06:46:58.000000000 +1000
+++ configure	2009-08-12 12:15:52.000000000 +1000
@@ -244,17 +244,7 @@
 Darwin)
 bsd="yes"
 darwin="yes"
-# on Leopard most of the system is 32-bit, so we have to ask the kernel it if we can run 64-bit userspace code
-if [ "$cpu" = "i386" ] ; then
-    is_x86_64=`sysctl -n hw.optional.x86_64`
-    [ "$is_x86_64" = "1" ] && cpu=x86_64
-fi
-if [ "$cpu" = "x86_64" ] ; then
-    OS_CFLAGS="-arch x86_64"
-    LDFLAGS="-arch x86_64"
-else
-    OS_CFLAGS="-mdynamic-no-pic"
-fi
+OS_CFLAGS="-mdynamic-no-pic"
 darwin_user="yes"
 cocoa="yes"
 audio_drv_list="coreaudio"
