--- Makefile.orig	2014-02-26 16:33:10.000000000 -0600
+++ Makefile	2014-03-07 01:04:08.000000000 -0600
@@ -15,10 +15,10 @@
 
 CFLAGSOPT ?= -DNDEBUG -O3 -ffast-math -funroll-loops -fomit-frame-pointer
 
-CFLAGS ?= -Wall -Wno-unknown-pragmas -I. -I/usr/local/include/ -I/usr/include/ $(CFLAGSOPT)
+CFLAGS ?= -Wall -Wno-unknown-pragmas -I. -I$(PREFIX)/include -I/usr/include $(CFLAGSOPT)
 CFLAGS += -std=c99 $(CFLAGSADD)
 
-LDFLAGS ?= -L/usr/local/lib/ -L/usr/lib/
+LDFLAGS ?= -L$(PREFIX)/lib -L/usr/lib
 
 ifneq "$(wildcard $(CUSTOMZLIB))" ""
 LDFLAGS += -L$(CUSTOMZLIB) -L$(CUSTOMZLIB)/lib
@@ -29,19 +29,6 @@
 LDFLAGS += -L$(CUSTOMLIBPNG) -L$(CUSTOMLIBPNG)/lib
 CFLAGS += -I$(CUSTOMLIBPNG) -I$(CUSTOMLIBPNG)/include
 
-else ifneq "$(wildcard /usr/local/include/png.h)" ""
-
-else ifneq "$(wildcard /opt/local/include/libpng15)" ""
-LDFLAGS += -L/opt/local/lib
-CFLAGS += -I/opt/local/include/libpng15
-
-else ifneq "$(wildcard /opt/X11/include/libpng15/)" ""
-LDFLAGS += -L/opt/X11/lib
-CFLAGS += -I/opt/X11/include/libpng15/
-
-else ifneq "$(wildcard /usr/X11/lib/)" ""
-LDFLAGS += -L/usr/X11/lib/
-CFLAGS += -I/usr/X11/include/
 endif
 
 LDFLAGS += -lpng -lz -lm lib/libimagequant.a -lm $(LDFLAGSADD)
