--- .//qttools/src/macdeployqt/shared/shared.cpp.orig	2014-09-11 03:48:25.000000000 -0700
+++ .//qttools/src/macdeployqt/shared/shared.cpp	2014-10-26 11:14:49.000000000 -0700
@@ -168,7 +168,7 @@
                 state = DylibName;
                 continue;
             } else if (part < parts.count() && parts.at(part).endsWith(".framework")) {
-                info.installName += "/" + (qtPath + "lib/").simplified();
+                info.installName += "/" + (qtPath + "Frameworks/").simplified();
                 info.frameworkDirectory = info.installName;
                 state = FrameworkName;
                 continue;
