--- qtbase/tests/auto/other/qaccessibilitymac/tst_qaccessibilitymac_helpers.mm.orig	2014-06-19 03:08:01.000000000 -0700
+++ qtbase/tests/auto/other/qaccessibilitymac/tst_qaccessibilitymac_helpers.mm	2014-06-28 00:58:02.000000000 -0700
@@ -116,7 +116,7 @@
 
 - (AXUIElementRef) ref { return reference; }
 - (void) print {
-    NSLog(@"Accessible Object role: '%@', description: '%@', value: '%@', rect: '%@'", self.role, self.description, self.value, NSStringFromRect(self.rect));
+    NSLog(@"Accessible Object role: '%@', description: '%@', value: '%@', rect: '%@'", self.role, self.description, self.value, NSStringFromRect(NSRectFromCGRect(self.rect)));
     NSLog(@"    Children: %ld", [[self childList] count]);
 }
 
