--- qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro.orig	2014-06-19 03:08:00.000000000 -0700
+++ qtbase/tests/auto/corelib/plugin/qpluginloader/machtest/machtest.pro	2014-06-27 11:09:55.000000000 -0700
@@ -4,10 +4,10 @@
     generate-bad.pl
 
 i386.target = good.i386.dylib
-i386.commands = $(CXX) $(CXXFLAGS) -shared -arch i386 -o $@ -I$$[QT_INSTALL_HEADERS/get] $<
+i386.commands = $(CXX) $(filter-out -arch i386 x86_64,$(CXXFLAGS)) -shared -arch i386 -o $@ -I$$[QT_INSTALL_HEADERS/get] $<
 i386.depends += $$PWD/../fakeplugin.cpp
 x86_64.target = good.x86_64.dylib
-x86_64.commands = $(CXX) $(CXXFLAGS) -shared -arch x86_64 -o $@ -I$$[QT_INSTALL_HEADERS/get] $<
+x86_64.commands = $(CXX) $(filter-out -arch i386 x86_64,$(CXXFLAGS)) -shared -arch x86_64 -o $@ -I$$[QT_INSTALL_HEADERS/get] $<
 x86_64.depends += $$PWD/../fakeplugin.cpp
 
 # Current Mac OS X toolchains have no compiler for PPC anymore
