https://trac.macports.org/ticket/73979
--- buildbot/status/web/waterfall.py.orig 2016-07-15 21:42:03
+++ buildbot/status/web/waterfall.py 2026-05-08 13:32:30
@@ -613,6 +613,9 @@
             minTime = util.now() - 24 * 60 * 60
         else:
             minTime = 0
+        # cap at one month
+        if maxTime - minTime > (30 * 24 * 60 * 60):
+            minTime = maxTime - (30 * 24 * 60 * 60)
         spanLength = 10  # ten-second chunks
         req_events = int(request.args.get("num_events", [self.num_events])[0])
         if self.num_events_max and req_events > self.num_events_max:
