--- buildbot/status/web/grid.py.orig	2015-04-21 00:53:41.000000000 -0500
+++ buildbot/status/web/grid.py	2016-08-05 15:42:30.000000000 -0500
@@ -13,6 +13,7 @@
 #
 # Copyright Buildbot Team Members
 
+from buildbot import util
 from buildbot.sourcestamp import SourceStamp
 from buildbot.status.web.base import HtmlResource
 from buildbot.status.web.base import build_get_class
@@ -204,7 +205,7 @@
                     'stamps': [map(SourceStamp.asDict, sstamp) for sstamp in stamps],
                     })
 
-        sortedBuilderNames = sorted(status.getBuilderNames())
+        sortedBuilderNames = util.naturalSort(status.getBuilderNames())
 
         cxt['builders'] = []
 
@@ -273,7 +274,7 @@
                     'stamps': [map(SourceStamp.asDict, sstamp) for sstamp in stamps],
                     })
 
-        sortedBuilderNames = sorted(status.getBuilderNames())
+        sortedBuilderNames = util.naturalSort(status.getBuilderNames())
 
         cxt['sorted_builder_names'] = sortedBuilderNames
         cxt['builder_builds'] = builder_builds = []
