*** gdk/gdk-sharp.dll.config.in.orig	2008-01-18 05:33:19.000000000 +0100
--- gdk/gdk-sharp.dll.config.in	2008-01-18 05:34:06.000000000 +0100
***************
*** 3,6 ****
--- 3,7 ----
    <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
    <dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-x11-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
    <dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+   <dllmap dll="gdksharpglue-2" target="libgdksharpglue-2.so"/>
  </configuration>
