--- Makefile.orig	2021-08-21 14:39:52.000000000 +0800
+++ Makefile	2023-07-07 08:04:35.000000000 +0800
@@ -34,7 +34,7 @@
 shared: lib/$(SO_TARGET)
 
 lib/$(SO_TARGET): $(OBJ_FILES)
-	$(CC99) $(CF) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+	@CC@ $(CF) $(SO_OPTS) $^ -o $@ $(LDLIBS)
 	( cd lib; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd lib; ln -sf $(SO_TARGET) $(SO_MAIN) )
 
@@ -43,14 +43,14 @@
 #-------------------------------------------------------------------------------
 
 example/maxvolume: example/maxvolume.c example/mmio.c lib/$(AR_TARGET)
-	$(CC99) $(CF) -I./include -o $@ $^ $(LDLIBS)
+	@CC@ $(CF) -I./include -o $@ $^ $(LDLIBS)
 
 #-------------------------------------------------------------------------------
 # compile each object file from its source file
 #-------------------------------------------------------------------------------
 
 build/%.o: src/%.c $(DEP_FILES)
-	$(CC99) $(CF) -I./include -c $< -o $@
+	@CC@ $(CF) -I./include -c $< -o $@
 
 #-------------------------------------------------------------------------------
 # clean and purge
