--- cmake/Macros.cmake.orig	2016-11-28 01:35:55.000000000 -0800
+++ cmake/Macros.cmake	2016-11-28 01:36:45.000000000 -0800
@@ -103,11 +103,6 @@
                                   MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}
                                   MACOSX_FRAMEWORK_BUNDLE_VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
         endif()
-
-        # adapt install directory to allow distributing dylibs/frameworks in user's frameworks/application bundle
-        set_target_properties(${target} PROPERTIES
-                              BUILD_WITH_INSTALL_RPATH 1
-                              INSTALL_NAME_DIR "@rpath")
     endif()
 
     # enable automatic reference counting on iOS
