--- configure.orig	2009-12-07 23:46:30.000000000 +0100
+++ configure	2009-12-07 23:46:59.000000000 +0100
@@ -1845,7 +1845,7 @@
 fi
 
 for ac_func in mkdir strdup strstr strtol memcpy memset lstat strcasecmp \
-               strcasestr getpwuid getopt snprintf memmove strerror
+               strcasestr getpwuid getopt printf vprintf vfprintf sprintf vsprintf snprintf vsnprintf memmove strerror
 do
 echo $ac_n "checking for $ac_func""... $ac_c" 1>&6
 echo "configure:1852: checking for $ac_func" >&5
