Fix echo -n stupidity.
--- configure.orig	2008-01-08 10:41:30.000000000 -0600
+++ configure	2021-10-27 17:33:19.000000000 -0500
@@ -10885,56 +10885,56 @@
 
 echo
 echo    "Keyring API.................... $with_keyring_api"
-echo -n "LDAP Schema support............ "
+printf "LDAP Schema support............ "
 if test -z "$ac_cv_func_ldap_str2objectclass" -o "x$ac_cv_func_ldap_str2objectclass" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "TLS support.................... "
+printf "TLS support.................... "
 if test -z "$gq_have_tls" -o "x$gq_have_tls" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "Encrypted passwords............ "
+printf "Encrypted passwords............ "
 if test -z "$gq_have_crypt" -o "x$gq_have_crypt" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "SASL binds..................... "
+printf "SASL binds..................... "
 if test -z "$SASL" -o "x$SASL" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "Kerberos binds................. "
+printf "Kerberos binds................. "
 if test -z "$HAVE_KERBEROS" -o "x$HAVE_KERBEROS" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "Browser Drag and drop.......... "
+printf "Browser Drag and drop.......... "
 if test -z "$enable_browser_dnd" -o "x$enable_browser_dnd" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "OpenLDAP client-side caching... "
+printf "OpenLDAP client-side caching... "
 if test -z "$HAVE_OLCACHE" -o "x$HAVE_OLCACHE" = "xno"; then
     echo "no"
 else
     echo "yes"
 fi
 
-echo -n "Debugging support ............. "
+printf "Debugging support ............. "
 if test -z "$DEBUG" -o "x$DEBUG" = "xno"; then
     echo "no"
 else
