--- DESCRIPTION	2023-11-09 15:10:06.000000000 +0800
+++ DESCRIPTION	2023-11-10 11:09:15.000000000 +0800
@@ -22,7 +22,7 @@
         data.table(>= 1.9.6), rsparse (>= 0.3.3.4), stringi (>= 1.1.5),
         mlapi (>= 0.1.0), lgr (>= 0.2), digest (>= 0.6.8)
 LinkingTo: Rcpp, digest (>= 0.6.8)
-Suggests: magrittr, udpipe (>= 0.6), glmnet, testthat, covr, knitr,
+Suggests: magrittr, glmnet, testthat, covr, knitr,
         rmarkdown, proxy
 URL: http://text2vec.org
 BugReports: https://github.com/dselivanov/text2vec/issues
