diff --git a/projects/compiler-rt/make/platform/clang_darwin.mk b/projects/compiler-rt/make/platform/clang_darwin.mk
index 6ed3230..b4917c8 100644
--- a/projects/compiler-rt/make/platform/clang_darwin.mk
+++ b/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -47,6 +47,10 @@ XCRun = \
 XCRunSdkPath = \
   $(shell \
     result=`xcrun --sdk $(1) --show-sdk-path 2> /dev/null`; \
+    if [ "$$?" != "0" ]; then \
+      result=`xcodebuild -sdk $(1) -version 2> /dev/null \
+              | sed -n 's/^Path: \(..*\)$$/\1/p'`; \
+    fi; \
     if [ "$$?" != "0" ]; then result=""; fi; \
     echo $$result)
 ###
diff --git a/tools/clang/runtime/compiler-rt/Makefile b/tools/clang/runtime/compiler-rt/Makefile
index 95c9fb7..50f0a81 100644
--- a/tools/clang/runtime/compiler-rt/Makefile
+++ b/tools/clang/runtime/compiler-rt/Makefile
@@ -82,8 +82,18 @@ RuntimeLibrary.darwin.Configs := \
 	profile_osx.a \
 	ubsan_osx.a
 
-IOS_SDK := $(shell xcrun --show-sdk-path -sdk iphoneos 2> /dev/null)
-IOSSIM_SDK := $(shell xcrun --show-sdk-path -sdk iphonesimulator 2> /dev/null)
+XCRunSdkPath = \
+  $(shell \
+    result=`xcrun --sdk $(1) --show-sdk-path 2> /dev/null`; \
+    if [ "$$?" != "0" ]; then \
+      result=`xcodebuild -sdk $(1) -version 2> /dev/null \
+              | sed -n 's/^Path: \(..*\)$$/\1/p'`; \
+    fi; \
+    if [ "$$?" != "0" ]; then result=""; fi; \
+    echo $$result)
+
+IOS_SDK := $(call XCRunSdkPath,iphoneos)
+IOSSIM_SDK := $(call XCRunSdkPath,iphonesimulator)
 
 ifneq ($(IOS_SDK)$(IOSSIM_SDK),)
 RuntimeLibrary.darwin.Configs += ios.a profile_ios.a
