http://llvm.org/bugs/show_bug.cgi?id=21636

Index: lib/CodeGen/AsmPrinter/AsmPrinter.cpp
===================================================================
--- a/lib/CodeGen/AsmPrinter/AsmPrinter.cpp
+++ b/lib/CodeGen/AsmPrinter/AsmPrinter.cpp
@@ -186,7 +186,7 @@
   // use the directive, where it would need the same conditionalization
   // anyway.
   Triple TT(getTargetTriple());
-  if (TT.isOSDarwin()) {
+  if (TT.isOSDarwin() && MAI->useIntegratedAssembler()) {
     unsigned Major, Minor, Update;
     TT.getOSVersion(Major, Minor, Update);
     // If there is a version specified, Major will be non-zero.
