--- Makefile.orig	2012-01-05 17:31:42.000000000 +0200
+++ Makefile	2013-07-26 19:19:21.000000000 +0300
@@ -13,7 +13,6 @@
 BINARY = postgresql_autodoc
 SOURCE = ${BINARY}.pl
 MANPAGE = ${BINARY}.1
-MANPAGE_SOURCE = ${MANPAGE}.in
 RELEASE_FILES = Makefile ChangeLog ${SOURCE} ${TEMPLATES} ${MANPAGE}
 RELEASE_DIR=postgresql_autodoc
 
@@ -23,11 +22,7 @@
 SED = $$(which sed)
 
 
-all: ${BINARY} ${MANPAGE}
-
-${MANPAGE}: ${MANPAGE_SOURCE}
-	${SED} -e "s,@@TEMPLATE-DIR@@,${DATADIR}," \
-		${MANPAGE_SOURCE} > ${MANPAGE}
+all: ${BINARY}
 
 
 ${BINARY}: ${SOURCE}
