Use the correct MacPorts prefix.
Do not use Homebrew.
--- configure.orig	2018-09-18 22:16:47.000000000 -0500
+++ configure	2018-10-04 13:15:21.000000000 -0500
@@ -19553,11 +20148,11 @@
 
 
        if test x$PORT = xport; then
-                  CPPFLAGS="$CPPFLAGS -isystem /opt/local/include"
-         LIBS="$LIBS -L/opt/local/lib"
-         if test -d /opt/local/include/db48; then
-           CPPFLAGS="$CPPFLAGS -I/opt/local/include/db48"
-           LIBS="$LIBS -L/opt/local/lib/db48"
+                  CPPFLAGS="$CPPFLAGS -isystem @PREFIX@/include"
+         LIBS="$LIBS -L@PREFIX@/lib"
+         if test -d @PREFIX@/include/db48; then
+           CPPFLAGS="$CPPFLAGS -I@PREFIX@/include/db48"
+           LIBS="$LIBS -L@PREFIX@/lib/db48"
          fi
        fi
 
@@ -19607,61 +20202,6 @@
 done
 test -n "$RSVG_CONVERT" || RSVG_CONVERT="rsvg-convert"
 
-       # Extract the first word of "brew", so it can be a program name with args.
-set dummy brew; ac_word=$2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-$as_echo_n "checking for $ac_word... " >&6; }
-if ${ac_cv_prog_BREW+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  if test -n "$BREW"; then
-  ac_cv_prog_BREW="$BREW" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
-    ac_cv_prog_BREW="brew"
-    $as_echo "$as_me:${as_lineno-$LINENO}: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-fi
-fi
-BREW=$ac_cv_prog_BREW
-if test -n "$BREW"; then
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $BREW" >&5
-$as_echo "$BREW" >&6; }
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-
-
-       if test x$BREW = xbrew; then
-
-         openssl_prefix=`$BREW --prefix openssl 2>/dev/null`
-         bdb_prefix=`$BREW --prefix berkeley-db4 2>/dev/null`
-         qt5_prefix=`$BREW --prefix qt5 2>/dev/null`
-         if test x$openssl_prefix != x; then
-           PKG_CONFIG_PATH="$openssl_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
-           export PKG_CONFIG_PATH
-         fi
-         if test x$bdb_prefix != x; then
-           CPPFLAGS="$CPPFLAGS -I$bdb_prefix/include"
-           LIBS="$LIBS -L$bdb_prefix/lib"
-         fi
-         if test x$qt5_prefix != x; then
-           PKG_CONFIG_PATH="$qt5_prefix/lib/pkgconfig:$PKG_CONFIG_PATH"
-           export PKG_CONFIG_PATH
-         fi
-       fi
      else
        case $build_os in
          *darwin*)
