Don't use -Werror.
--- artnet/Makefile.in.orig	2013-02-27 10:59:11.000000000 -0600
+++ artnet/Makefile.in	2019-01-11 23:16:54.000000000 -0600
@@ -284,7 +284,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = -I$(top_builddir) -I$(top_srcdir) -Wall -Werror -Wformat -W
+AM_CFLAGS = -I$(top_builddir) -I$(top_srcdir) -Wall -Wformat -W
 EXTRA_DIST = tod.h misc.h
 pkginclude_HEADERS = artnet.h packets.h common.h
 lib_LTLIBRARIES = libartnet.la
