--- setup.py.orig	2010-10-22 17:23:47.000000000 +0200
+++ setup.py	2010-10-22 17:24:21.000000000 +0200
@@ -114,7 +114,7 @@
     from distutils.ccompiler import new_compiler
 
     compiler = new_compiler()
-    inc_dirs = compiler.include_dirs + ['/usr/include']
+    inc_dirs = compiler.include_dirs + ['@@PREFIX@@/include']
 
     search_for_ssl_incs_in = [
                           '/usr/local/ssl/include',
@@ -130,7 +130,7 @@
             ssl_incs += krb5_h
 
     ssl_libs = find_library_file(compiler, 'ssl',
-                                 ['/usr/lib'],
+                                 ['@@PREFIX@@/lib'],
                                  ['/usr/local/lib',
                                   '/usr/local/ssl/lib',
                                   '/usr/contrib/ssl/lib/'
