--- Makefile	2015-08-01 20:02:46.000000000 -0400
+++ Makefile	2017-01-02 11:55:03.000000000 -0500
@@ -2,8 +2,6 @@
 # Makefile to build brightness
 # originally by Jon Stacey <jon@jonsview.com>
 
-prefix=/usr/local
-
 override CFLAGS += -mmacosx-version-min=10.6
 
 all: build
@@ -23,5 +21,4 @@
 	/bin/rm -f brightness *.o
 
 install:
-	/bin/mkdir -p $(prefix)/bin
-	/usr/bin/install -s -m 0755 brightness $(prefix)/bin
+	/usr/bin/install -s -m 0755 brightness $(PREFIX)/bin
