--- include/rapidjson_include.orig.h	2022-12-22 15:28:15.000000000 +0100
+++ include/rapidjson_include.h	2022-12-22 15:28:47.000000000 +0100
@@ -11,12 +11,12 @@
 #include "Exceptions.h"
 #include "CoolPropTools.h"
 
-#include "externals/rapidjson/include/rapidjson/rapidjson.h"
-#include "externals/rapidjson/include/rapidjson/document.h"
-#include "externals/rapidjson/include/rapidjson/filewritestream.h"  // wrapper of C stream for prettywriter as output
-#include "externals/rapidjson/include/rapidjson/prettywriter.h"     // for stringify JSON
-#include "externals/rapidjson/include/rapidjson/stringbuffer.h"     // for string buffer
-#include "externals/rapidjson/include/rapidjson/schema.h"
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/document.h>
+#include <rapidjson/filewritestream.h>  // wrapper of C stream for prettywriter as output
+#include <rapidjson/prettywriter.h>     // for stringify JSON
+#include <rapidjson/stringbuffer.h>     // for string buffer
+#include <rapidjson/schema.h>
 
 #include <cassert>
 
