--- src/HumidAirProp.orig.cpp	2022-12-22 22:34:28.000000000 +0100
+++ src/HumidAirProp.cpp	2022-12-22 22:35:18.000000000 +0100
@@ -24,7 +24,7 @@
 #include <string.h>
 #include <iostream>
 #include <list>
-#include "externals/IF97/IF97.h"
+#include <IF97.h>
 
 /// This is a stub overload to help with all the strcmp calls below and avoid needing to rewrite all of them
 std::size_t strcmp(const std::string& s, const std::string& e) {
