--- rubberband/RubberBandStretcher.h.orig	2022-10-21 08:51:10.000000000 -0500
+++ rubberband/RubberBandStretcher.h	2022-11-01 18:51:15.000000000 -0500
@@ -24,7 +24,7 @@
 #ifndef RUBBERBAND_STRETCHER_H
 #define RUBBERBAND_STRETCHER_H
     
-#define RUBBERBAND_VERSION "3.1.0"
+#define RUBBERBAND_VERSION "3.1.1"
 #define RUBBERBAND_API_MAJOR_VERSION 2
 #define RUBBERBAND_API_MINOR_VERSION 7
 
--- rubberband/rubberband-c.h.orig	2022-10-21 08:51:10.000000000 -0500
+++ rubberband/rubberband-c.h	2022-11-01 18:51:05.000000000 -0500
@@ -28,7 +28,7 @@
 extern "C" {
 #endif
 
-#define RUBBERBAND_VERSION "3.1.0"
+#define RUBBERBAND_VERSION "3.1.1"
 #define RUBBERBAND_API_MAJOR_VERSION 2
 #define RUBBERBAND_API_MINOR_VERSION 7
 
