--- applets/clock/clock.c.bak	Tue Apr 20 07:47:26 2004
+++ applets/clock/clock.c	Tue Apr 20 07:51:19 2004
@@ -47,7 +47,9 @@
 #include <dirent.h>
 #include <string.h>
 #include <time.h>
+#if !defined(__APPLE__)
 #include <langinfo.h>
+#endif
 
 #include <panel-applet.h>
 #include <panel-applet-gconf.h>
@@ -525,8 +527,12 @@
 {
         const char *am;
 
+#if !defined(__APPLE__)
         am = nl_langinfo (AM_STR);
         return (am[0] == '\0') ? CLOCK_FORMAT_24 : CLOCK_FORMAT_12;
+#else
+        return CLOCK_FORMAT_24;
+#endif
 }
 
 #ifdef HAVE_LIBECAL
