--- applets/clock/clock-location.c.orig	2013-11-16 00:58:04.000000000 -0800
+++ applets/clock/clock-location.c	2013-11-16 01:08:25.000000000 -0800
@@ -542,9 +542,7 @@
 	}
 
 	priv->weather_info =
-		gweather_info_new_for_world (priv->world,
-					     priv->loc,
-					     GWEATHER_FORECAST_LIST);
+		gweather_info_new (priv->loc, GWEATHER_FORECAST_LIST);
 	g_signal_connect (priv->weather_info, "updated",
 			  G_CALLBACK (weather_info_updated), loc);
 
