--- CPU.xs.orig	2013-11-27 14:40:07.000000000 -0800
+++ CPU.xs	2017-02-14 15:30:16.000000000 -0800
@@ -172,6 +172,39 @@
 #ifndef INTEL_6_13
 #define INTEL_6_13 0xaa33392b
 #endif
+
+#ifndef INTEL_PENRYN
+#define INTEL_PENRYN 0x78ea4fbc
+#endif
+
+#ifndef INTEL_NEHALEM
+#define INTEL_NEHALEM 0x6b5a4cd2
+#endif
+
+#ifndef INTEL_WESTMERE
+#define INTEL_WESTMERE 0x573b5eec
+#endif
+
+#ifndef INTEL_SANDYBRIDGE
+#define INTEL_SANDYBRIDGE 0x5490b78c
+#endif
+
+#ifndef INTEL_IVYBRIDGE
+#define INTEL_IVYBRIDGE 0x1f65e835
+#endif
+
+#ifndef INTEL_HASWELL
+#define INTEL_HASWELL 0x10b282dc
+#endif
+
+#ifndef INTEL_BROADWELL
+#define INTEL_BROADWELL 0x582ed09c
+#endif
+
+#ifndef INTEL_SKYLAKE
+#define INTEL_SKYLAKE 0x37fc219f
+#endif
+
 #ifndef ARM_9
 #define ARM_9 0xe73283ae
 #endif
@@ -180,24 +213,44 @@
 #define ARM_11 0x8ff620d8
 #endif
 
-#ifndef INTEL_PENRYN
-#define INTEL_PENRYN 0x78ea4fbc
+#ifndef ARM_XSCALE
+#define ARM_XSCALE 0x53b005f5
 #endif
 
-#ifndef INTEL_NEHALEM
-#define INTEL_NEHALEM 0x6b5a4cd2
+#ifndef ARM_12
+#define ARM_12 0xbd1b0ae9
+#endif
+
+#ifndef ARM_13
+#define ARM_13 0x0cc90e64
+#endif
+
+#ifndef ARM_14
+#define ARM_14 0x96077ef1
+#endif
+
+#ifndef ARM_15
+#define ARM_15 0xa8511bca
+#endif
+
+#ifndef ARM_SWIFT
+#define ARM_SWIFT 0x1e2d6381
 #endif
 
-#ifndef INTEL_CORE
-#define INTEL_CORE 0x73d67300
+#ifndef ARM_CYCLONE
+#define ARM_CYCLONE 0x37a09642
 #endif
 
-#ifndef INTEL_CORE2
-#define INTEL_CORE2 0x426f69ef
+#ifndef ARM_TYPHOON
+#define ARM_TYPHOON 0x2c91a47e
 #endif
 
-#ifndef INTEL_COREI7
-#define INTEL_COREI7 0x5490B78C
+#ifndef ARM_TWISTER
+#define ARM_TWISTER 0x92fb37c8
+#endif
+
+#ifndef ARM_HURRICANE
+#define ARM_HURRICANE 0x67ceee93
 #endif
 
 char *apple_get_type_name() {
@@ -217,20 +270,46 @@
                    return "POWERPC_G5";
                 case INTEL_6_13:
                    return "INTEL_6_13";
-                case ARM_9:
-                   return "ARM_9";
-                case ARM_11:
-                   return "ARM_11";
                 case INTEL_PENRYN:
                    return "INTEL_PENRYN";
                 case INTEL_NEHALEM:
                    return "INTEL_NEHALEM";
-                case INTEL_CORE:
-                   return "INTEL_CORE";
-                case INTEL_CORE2:
-                   return "INTEL_CORE2";
-                case INTEL_COREI7:
-                   return "INTEL_COREI7";
+                case INTEL_WESTMERE:
+                   return "INTEL_WESTMERE";
+                case INTEL_SANDYBRIDGE:
+                   return "INTEL_SANDYBRIDGE";
+                case INTEL_IVYBRIDGE:
+                   return "INTEL_IVYBRIDGE";
+                case INTEL_HASWELL:
+                   return "INTEL_HASWELL";
+                case INTEL_BROADWELL:
+                   return "INTEL_BROADWELL";
+                case INTEL_SKYLAKE:
+                   return "INTEL_SKYLAKE";
+                case ARM_9:
+                   return "ARM_9";
+                case ARM_11:
+                   return "ARM_11";
+                case ARM_XSCALE:
+                   return "ARM_XSCALE";
+                case ARM_12:
+                   return "ARM_12";
+                case ARM_13:
+                   return "ARM_13";
+                case ARM_14:
+                   return "ARM_14";
+                case ARM_15:
+                   return "ARM_15";
+                case ARM_SWIFT:
+                   return "ARM_SWIFT";
+                case ARM_CYCLONE:
+                   return "ARM_CYCLONE";
+                case ARM_TYPHOON:
+                   return "ARM_TYPHOON";
+                case ARM_TWISTER:
+                   return "ARM_TWISTER";
+                case ARM_HURRICANE:
+                   return "ARM_HURRICANE";
     default:
        return "UNKNOWN";
         }
