--- work/camldbm-1.0/Makefile	2011-11-22 16:56:49.000000000 +0100
+++ Makefile	2012-07-26 08:39:14.000000000 +0200
@@ -21,7 +21,6 @@
 O=o
 A=a
 SO=so
-LIBDIR=`ocamlc -where`
 STUBLIBDIR=$(LIBDIR)/stublibs
 
 
@@ -57,13 +56,14 @@
 	$(OCAMLDEP) *.ml *.mli > .depend
 
 install::
-	if test -f dllcamldbm.$(SO); then cp dllcamldbm.$(SO) $(STUBLIBDIR)/; fi 
-	cp libcamldbm.$(A) $(LIBDIR)/
-	cd $(LIBDIR) && ranlib libcamldbm.$(A)
-	cp dbm.cma dbm.cmxa dbm.cmi dbm.mli $(LIBDIR)/
-	cp dbm.$(A) $(LIBDIR)/
-	cd $(LIBDIR) && ranlib dbm.$(A)
-	if test -f dbm.cmxs; then cp dbm.cmxs $(LIBDIR)/; fi
+	install -m 755 -d $(DESTDIR)$(STUBLIBDIR) $(DESTDIR)$(LIBDIR)
+	if test -f dllcamldbm.$(SO); then cp dllcamldbm.$(SO) $(DESTDIR)$(STUBLIBDIR)/; fi 
+	cp libcamldbm.$(A) $(DESTDIR)$(LIBDIR)/
+	cd $(DESTDIR)$(LIBDIR) && ranlib libcamldbm.$(A)
+	cp dbm.cma dbm.cmxa dbm.cmi dbm.mli $(DESTDIR)$(LIBDIR)/
+	cp dbm.$(A) $(DESTDIR)$(LIBDIR)/
+	cd $(DESTDIR)$(LIBDIR) && ranlib dbm.$(A)
+	if test -f dbm.cmxs; then cp dbm.cmxs $(DESTDIR)$(LIBDIR)/; fi
 
 clean::
 	rm -f *.cm* *.$(O) *.$(A) *.$(SO)
