--- test/gvfs-test.orig	2016-02-18 11:43:03.000000000 -0800
+++ test/gvfs-test	2016-02-18 11:44:43.000000000 -0800
@@ -68,7 +68,7 @@
 have_httpd = httpd_cmd is not None
 sshd_path = subprocess.check_output(['which', 'sshd'], universal_newlines=True).strip()
 
-local_ip = subprocess.check_output("ip -4 addr | sed -nr '/127\.0\.0/ n; "
+local_ip = subprocess.check_output("ifconfig -a | gsed -nr '/127\.0\.0/ n; "
                                    "/inet / {  s/^.*inet ([0-9.]+).*$/\\1/; p; q  }'",
                                    shell=True, universal_newlines=True)
 
