diff -ur before/configure lp_solve_5.5/configure
--- configure	2008-07-21 14:10:03.000000000 -0400
+++ configure	2008-11-17 00:01:48.000000000 -0500
@@ -1357,7 +1357,7 @@
   linux)
     CCSHARED="-fPIC";;
   apple-darwin)
-    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME -Wno-long-double
+    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME
     ;;
   OpenUNIX*|UnixWare*)
     DEF='-dy -DNOLONGLONG'
diff -ur before/configure.ac lp_solve_5.5/configure.ac
--- configure.ac	2008-07-21 14:10:17.000000000 -0400
+++ configure.ac	2008-11-17 00:01:55.000000000 -0500
@@ -21,7 +21,7 @@
   linux)
     CCSHARED="-fPIC";;
   apple-darwin)
-    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME -Wno-long-double
+    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME
     ;;
   OpenUNIX*|UnixWare*)
     DEF='-dy -DNOLONGLONG'
diff -ur before/demo/ccc.osx lp_solve_5.5/demo/ccc.osx
--- demo/ccc.osx	2005-06-11 16:27:44.000000000 -0400
+++ demo/ccc.osx	2008-11-17 00:02:06.000000000 -0500
@@ -9,6 +9,6 @@
 else dl=-ldl
 fi
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -I.. -I../bfp -I../bfp/bfp_etaPFI -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP $src -o demo $math $dl
diff -ur before/lp_solve/ccc.osx lp_solve_5.5/lp_solve/ccc.osx
--- lp_solve/ccc.osx	2005-06-11 16:27:38.000000000 -0400
+++ lp_solve/ccc.osx	2008-11-17 00:02:22.000000000 -0500
@@ -9,6 +9,6 @@
 else dl=-ldl
 fi
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src -o lp_solve $math $dl
diff -ur before/lpsolve55/ccc.osx lp_solve_5.5/lpsolve55/ccc.osx
--- lpsolve55/ccc.osx	2005-06-23 17:53:08.000000000 -0400
+++ lpsolve55/ccc.osx	2008-11-17 00:02:38.000000000 -0500
@@ -9,7 +9,7 @@
      so=y
 fi
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 libtool -static -o liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
diff -ur before/lpsolve55/cccLUSOL.osx lp_solve_5.5/lpsolve55/cccLUSOL.osx
--- lpsolve55/cccLUSOL.osx	2005-06-23 17:53:18.000000000 -0400
+++ lpsolve55/cccLUSOL.osx	2008-11-17 00:02:42.000000000 -0500
@@ -9,7 +9,7 @@
      so=y
 fi
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 libtool -static -o liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
