--- configure	2009-08-04 17:22:58.000000000 -0500
+++ configure	2009-12-30 01:29:33.000000000 -0600
@@ -1357,7 +1357,7 @@
   linux)
     CCSHARED="-fPIC";;
   apple-darwin)
-    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME -Wno-long-double
+    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME
     ;;
   OpenUNIX*|UnixWare*)
     DEF='-dy -DNOLONGLONG'
--- configure.ac	2009-01-02 12:15:34.000000000 -0600
+++ configure.ac	2009-12-30 01:29:33.000000000 -0600
@@ -21,7 +21,7 @@
   linux)
     CCSHARED="-fPIC";;
   apple-darwin)
-    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME -Wno-long-double
+    DEF=-ldl -idirafter /usr/include/sys -DINTEGERTIME
     ;;
   OpenUNIX*|UnixWare*)
     DEF='-dy -DNOLONGLONG'
--- demo/ccc.osx	2005-06-11 15:27:44.000000000 -0500
+++ demo/ccc.osx	2009-12-30 01:29:33.000000000 -0600
@@ -9,6 +9,6 @@
 else dl=-ldl
 fi
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -I.. -I../bfp -I../bfp/bfp_etaPFI -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP $src -o demo $math $dl
--- lp_solve/ccc.osx	2009-01-25 12:39:09.000000000 -0600
+++ lp_solve/ccc.osx	2009-12-30 01:29:33.000000000 -0600
@@ -31,6 +31,6 @@
 def=
 dl=-ldl
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src -o bin/$PLATFORM/lp_solve $math $dl
--- lpsolve55/ccc.osx	2009-08-09 10:36:19.000000000 -0500
+++ lpsolve55/ccc.osx	2009-12-30 01:29:33.000000000 -0600
@@ -30,7 +30,7 @@
 dl=-ldl
 so=y
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 libtool -static -o bin/$PLATFORM/liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
--- lpsolve55/cccLUSOL.osx	2009-01-25 12:39:54.000000000 -0600
+++ lpsolve55/cccLUSOL.osx	2009-12-30 01:29:33.000000000 -0600
@@ -30,7 +30,7 @@
 dl=-ldl
 so=y
 
-opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME -Wno-long-double'
+opts='-idirafter /usr/include/sys -O3 -DINTEGERTIME'
 
 $c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 libtool -static -o bin/$PLATFORM/liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
