--- runfiles/texmf-dist/scripts/crossrefware/bibmradd.pl.orig
+++ runfiles/texmf-dist/scripts/crossrefware/bibmradd.pl
@@ -69,10 +69,7 @@
 
 BEGIN {
     # find files relative to our installed location within TeX Live
-    chomp(my $TLMaster = `kpsewhich -var-value=TEXMFROOT`); # TL root
-    if (length($TLMaster)) {
-	unshift @INC, "$TLMaster/texmf-dist/scripts/bibtexperllibs";
-    }
+    unshift @INC, "@@TEXMFDIST@@/scripts/bibtexperllibs";
      # find development bibtexperllibs in sibling checkout to this script,
      # even if $0 is a symlink. Irrelevant when using from an installation.
      my $real0 = Cwd::abs_path($0);
