--- work/smpeg-0.4.4/Makefile.in	2001-06-18 22:59:14.000000000 +0200
+++ Makefile.in	2005-09-30 08:56:31.000000000 +0200
@@ -195,7 +195,7 @@
 SOURCES = $(libsmpeg_la_SOURCES) $(plaympeg_SOURCES) $(gtv_SOURCES) $(glmovie_SOURCES)
 OBJECTS = $(libsmpeg_la_OBJECTS) $(plaympeg_OBJECTS) $(gtv_OBJECTS) $(glmovie_OBJECTS)
 
-all: all-redirect
+all: all-redirect libsmpeg.$(VERSION).dylib
 .SUFFIXES:
 .SUFFIXES: .S .c .cpp .lo .o .s
 $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
@@ -283,6 +283,9 @@
 libsmpeg.la: $(libsmpeg_la_OBJECTS) $(libsmpeg_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libsmpeg_la_LDFLAGS) $(libsmpeg_la_OBJECTS) $(libsmpeg_la_LIBADD) $(LIBS)
 
+libsmpeg.$(VERSION).dylib:
+	$(CXX) -dynamiclib MPEG*.o smpeg.o audio/*.o video/*.o -o $@ -current_version $(VERSION) -compatibility_version $(VERSION) -install_name $(libdir)/$@ $(LIBS)
+
 mostlyclean-binPROGRAMS:
 
 clean-binPROGRAMS:
