--- configure.in.orig	2006-01-16 22:23:49.000000000 -0600
+++ configure.in	2013-01-22 09:25:05.000000000 -0600
@@ -21,7 +21,7 @@
 AC_INIT(configure.in)
 AM_INIT_AUTOMAKE(libast, 0.7)
 AC_DEFINE_UNQUOTED(LIBAST_VERSION, "$VERSION", [Version])
-AM_CONFIG_HEADER(config.h include/libast/sysdefs.h)
+AC_CONFIG_HEADERS([config.h include/libast/sysdefs.h])
 
 LIBTOOLIZE="${SAVE_LIBTOOLIZE:-libtoolize}"
 ACLOCAL="${SAVE_ACLOCAL:-aclocal}"
