--- src/qt/src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro.orig	2013-06-07 07:16:58.000000000 +0200
+++ src/qt/src/3rdparty/webkit/Source/WebKit/qt/QtWebKit.pro	2013-10-30 10:10:17.000000000 +0100
@@ -278,6 +278,8 @@
             LIBS += $$SOURCE_DIR/../WebKitLibraries/libWebKitSystemInterfaceLion.a
         } else: equals(DARWIN_MAJOR_VERSION, "12") | contains(QMAKE_MAC_SDK, "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.8.sdk") {
             LIBS += $$SOURCE_DIR/../WebKitLibraries/libWebKitSystemInterfaceMountainLion.a
+        } else: equals(DARWIN_MAJOR_VERSION, "13") | contains(QMAKE_MAC_SDK, "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk") {
+            LIBS += $$SOURCE_DIR/../WebKitLibraries/libWebKitSystemInterfaceMavericks.a
         }
     }
 }
