--- src/metamail/codes.c.orig	1993-09-20 08:13:22.000000000 -0500
+++ src/metamail/codes.c	2022-01-25 23:28:47.000000000 -0600
@@ -13,6 +13,7 @@
 WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES.
 */
 #include <stdio.h>
+#include <string.h>
 #include <ctype.h>
 #include <config.h>
 
@@ -43,6 +44,8 @@
 }
 */
 
+void output64chunk(int c1, int c2, int c3, int pads, FILE *outfile);
+
 /* the following gets a character, but fakes it properly into two chars if there's a newline character */
 static int InNewline=0;
 
@@ -69,6 +72,7 @@
 #endif
 }
 
+void
 to64(infile, outfile, PortableNewlines) 
 FILE *infile, *outfile;
 int PortableNewlines;
@@ -97,6 +101,7 @@
     fflush(outfile);
 }
 
+void
 output64chunk(c1, c2, c3, pads, outfile)
 FILE *outfile;
 {
@@ -174,6 +179,7 @@
 }
 #endif
 
+void
 from64(infile, outfile, boundaries, boundaryct, PortableNewlines) 
 FILE *infile, *outfile;
 char **boundaries;
@@ -275,6 +281,7 @@
 }
 */
 
+void
 toqp(infile, outfile) 
 FILE *infile, *outfile;
 {
@@ -350,6 +357,7 @@
     }
 }
 
+void
 fromqp(infile, outfile, boundaries, boundaryct) 
 FILE *infile, *outfile;
 char **boundaries;
--- src/metamail/mmencode.c.orig	1994-01-26 12:47:37.000000000 -0600
+++ src/metamail/mmencode.c	2022-01-25 23:28:53.000000000 -0600
@@ -13,6 +13,7 @@
 WITHOUT ANY EXPRESS OR IMPLIED WARRANTIES.
 */
 #include <stdio.h>
+#include <stdlib.h>
 #include <config.h>
 #ifdef MSDOS
 #include <fcntl.h>
@@ -21,6 +22,11 @@
 #define BASE64 1
 #define QP 2 /* quoted-printable */
 
+void to64(FILE *infile, FILE *outfile, int PortableNewlines);
+void from64(FILE *infile, FILE *outfile, char **boundaries, int *boundaryct, int PortableNewlines);
+void toqp(FILE *infile, FILE *outfile);
+void fromqp(FILE *infile, FILE *outfile, char **boundaries, int *boundaryct);
+
 main(argc, argv)
 int argc;
 char **argv;
