--- src/cxx/cxxsupport/arr.h.orig	2010-11-24 04:14:42.000000000 -0800
+++ src/cxx/cxxsupport/arr.h	2011-03-06 00:36:49.000000000 -0800
@@ -54,7 +54,7 @@
       using namespace std;
       if (sz==0) return 0;
       void *res;
-      planck_assert(posix_memalign(&res,align,sz*sizeof(T))==0,
+      planck_assert((res = valloc(sz*sizeof(T))),
         "error in posix_memalign()");
       return static_cast<T *>(res);
       }
