--- src/Makefile.orig	2010-10-02 11:21:01.000000000 -0500
+++ src/Makefile	2012-08-16 20:57:58.000000000 -0500
@@ -68,8 +68,8 @@
 
 GKRELLM_INCLUDES = gkrellm.h gkrellm-public-proto.h $(SHARED_PATH)/log.h
 
-PKG_INCLUDE = `$(PKG_CONFIG) --cflags gtk+-2.0 gthread-2.0`
-PKG_LIB = `$(PKG_CONFIG) --libs gtk+-2.0 gthread-2.0`
+PKG_INCLUDE = `$(PKG_CONFIG) --cflags gtk+-2.0 gthread-2.0 gmodule-2.0`
+PKG_LIB = `$(PKG_CONFIG) --libs gtk+-2.0 gthread-2.0 gmodule-2.0`
 
 FLAGS = -O2 -I.. -I$(SHARED_PATH) $(PKG_INCLUDE) $(GTOP_INCLUDE) $(PTHREAD_INC)\
  ${SSL_INCLUDE} ${NTLM_INCLUDE} -DGKRELLM_CLIENT
