--- lib/in2tsv.pl.orig
+++ lib/in2tsv.pl
@@ -1,4 +1,4 @@
-#!/bin/env perl
+#!@PERL@
 
 #
 # Copyright(C) 2007-2012 National Institute of Information and Communications Technology
--- lib/tsv2out.pl.orig
+++ lib/tsv2out.pl
@@ -1,4 +1,4 @@
-#!/bin/env perl
+#!@PERL@
 
 #
 # Copyright(C) 2007-2012 National Institute of Information and Communications Technology
--- makemodel/csv/csv2tsv.pl.orig
+++ makemodel/csv/csv2tsv.pl
@@ -1,4 +1,4 @@
-#!/bin/env perl
+#!@PERL@
 
 #
 # Copyright(C) 2007-2012 National Institute of Information and Communications Technology
--- makemodel/csv/csv2tsv.sh.orig
+++ makemodel/csv/csv2tsv.sh
@@ -8,7 +8,7 @@
 for f in *.csv
 do
   f=`basename $f .csv`
-  perl -I ../../lib ./csv2tsv.pl $f.csv > ../tsv/$f.tsv
+  @PERL@ -I ../../lib ./csv2tsv.pl $f.csv > ../tsv/$f.tsv
 done
 
 
