--- inc/Devel/CheckLib.pm.orig	2016-12-18 18:49:37.000000000 +1100
+++ inc/Devel/CheckLib.pm	2022-05-29 14:24:29.000000000 +1000
@@ -372,7 +372,7 @@
     my @ccflags = grep { length } quotewords('\s+', 1, $Config_ccflags||'');
     my @ldflags = grep { length } quotewords('\s+', 1, @Config_ldflags);
     my @paths = split(/$Config{path_sep}/, $ENV{PATH});
-    my @cc = split(/\s+/, $Config{cc});
+    my @cc = split(/\s+/, $ENV{CC});
     return ( [ @cc, @ccflags ], \@ldflags ) if -x $cc[0];
     foreach my $path (@paths) {
         my $compiler = File::Spec->catfile($path, $cc[0]) . $Config{_exe};
