--- CMakeLists.txt.orig	2018-09-28 14:20:56.000000000 -0700
+++ CMakeLists.txt	2019-07-19 17:49:59.000000000 -0700
@@ -81,3 +81,23 @@
 
 install                     (DIRECTORY   ${CMAKE_SOURCE_DIR}/include/diy
                              DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+
+add_library(diy INTERFACE)
+target_include_directories(
+    diy
+    INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+)
+
+install(
+    TARGETS diy
+    EXPORT diyExport
+    INCLUDES DESTINATION "${CMAKE_INSTALL_PREFIX}/include/diy"
+)
+
+install(
+    EXPORT diyExport
+    DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/cmake/diy"
+    FILE "diyConfig.cmake"
+)
