--- setup.cfg	Tue Jun 29 12:19:07 2004
+++ setup.cfg.new	Wed Oct 13 09:45:12 2004
@@ -6,8 +6,8 @@
 # Section for compiling the C extension module
 # for wrapping OpenLDAP 2 libs
 
-library_dirs = /usr/local/openldap-REL_ENG_2_1/lib /usr/lib/sasl /usr/local/sasl/lib
-include_dirs = /usr/local/openldap-REL_ENG_2_1/include /usr/include/sasl /usr/local/sasl/include/sasl
+library_dirs = LIBDIR
+include_dirs = INCDIR
 
 extra_compile_args = 
 extra_objects =
@@ -16,7 +16,7 @@
 # Support for StartTLS/LDAPS, SASL bind and reentrant libldap_r.
 # This needs recent OpenLDAP 2.0.26+ or 2.1.3+ built with
 # ./configure --with-cyrus-sasl --with-tls
-libs = ldap_r lber sasl2 ssl crypto
+libs = LIBS
 
 [install]
 # Installation options
