--- setup.cfg	2005-04-29 13:47:32.000000000 +0200
+++ setup.cfg.new	2005-05-23 18:13:15.000000000 +0200
@@ -6,8 +6,8 @@
 # Section for compiling the C extension module
 # for wrapping OpenLDAP 2 libs
 
-library_dirs = /opt/openldap-OPENLDAP_REL_ENG_2_2/lib
-include_dirs = /opt/openldap-OPENLDAP_REL_ENG_2_2/include /usr/include/sasl
+library_dirs = __LIBDIR__
+include_dirs = __INCDIR__
 
 extra_compile_args = 
 extra_objects =
@@ -16,7 +16,7 @@
 # Support for StartTLS/LDAPS, SASL bind and reentrant libldap_r.
 # This needs recent OpenLDAP 2.0.26+ or 2.1.3+ built with
 # ./configure --with-cyrus-sasl --with-tls
-libs = ldap_r lber ssl crypto sasl2
+libs = __LIBS__
 
 [install]
 # Installation options
