--- Makefile.orig	2020-02-13 06:58:50.000000000 +0000
+++ Makefile	2020-02-16 14:30:00.000000000 +0000
@@ -451,8 +451,10 @@
 OPTIONS_CFLAGS += $(foreach opt,$(use_opts),$(if $($(opt)),-D$(opt),))
 
 ifneq ($(USE_LIBCRYPT),)
+ifneq ($(shell uname -s),Darwin)
 OPTIONS_LDFLAGS += -lcrypt
 endif
+endif
 
 ifneq ($(USE_SLZ),)
 # Use SLZ_INC and SLZ_LIB to force path to zlib.h and libz.{a,so} if needed.
