--- ProjectFilesDialog.cpp.orig	2010-10-22 19:17:12.000000000 +0200
+++ ProjectFilesDialog.cpp	2010-10-22 19:17:16.000000000 +0200
@@ -697,9 +697,13 @@
 			bind(&QItemSelectionRange::bottom, _1)
 		)
 	);
+
+	// This hack is required to make it build with boost 1.44.
+	typedef int const Range::* IntMemPtr;
+
 	std::sort(
 		sorted_ranges.begin(), sorted_ranges.end(),
-		bind(&Range::first, _1) < bind(&Range::first, _2)
+		bind((IntMemPtr)&Range::first, _1) < bind((IntMemPtr)&Range::first, _2)
 	);
 	
 	QVectorIterator<Range> it(sorted_ranges);
