--- src/daemon/digestauth.c.orig	2010-12-26 11:21:16.000000000 -0500
+++ src/daemon/digestauth.c	2010-12-26 11:22:28.000000000 -0500
@@ -56,6 +56,24 @@
  */
 #define MAX_AUTH_RESPONSE_LENGTH 128
 
+/* begin test code */ 
+size_t strnlen(const char *s, size_t len) 
+{ 
+    size_t i; 
+    for(i=0; i<len && *(s+i); i++); 
+    return i; 
+} 
+char* strndup (char const *s, size_t n) 
+{ 
+    size_t len = strnlen (s, n); 
+    char *new = malloc (len + 1); 
+    if (new == NULL) 
+    return NULL; 
+    new[len] = '\0'; 
+    return memcpy (new, s, len); 
+} 
+/* end test code */ 
+
 /**
  * convert bin to hex 
  *
