--- scripts/build.py.orig	2026-04-03 05:21:13
+++ scripts/build.py	2026-04-14 14:36:26
@@ -82,15 +82,15 @@
     file = "libcurl-impersonate.tar.gz"
     sysname = "linux-" + arch["libc"] if arch["system"] == "Linux" else arch["sysname"]
 
-    url = (
-        f"https://github.com/lexiforest/curl-impersonate/releases/download/"
-        f"v{__version__}/libcurl-impersonate-v{__version__}"
-        f".{arch['arch']}-{sysname}.tar.gz"
-    )
+#     url = (
+#         f"https://github.com/lexiforest/curl-impersonate/releases/download/"
+#         f"v{__version__}/libcurl-impersonate-v{__version__}"
+#         f".{arch['arch']}-{sysname}.tar.gz"
+#     )
+# 
+#     print(f"Downloading libcurl-impersonate from {url}...")
+#     urlretrieve(url, file)
 
-    print(f"Downloading libcurl-impersonate from {url}...")
-    urlretrieve(url, file)
-
     print("Unpacking downloaded files...")
     os.makedirs(libdir, exist_ok=True)
     shutil.unpack_archive(file, libdir)
