--- gui/cpp/gwengui-cpp-config.cmake.in	2016-01-15 21:09:17.000000000 +0100
+++ gui/cpp/gwengui-cpp-config.cmake.in.new	2016-10-21 02:21:38.000000000 +0200
@@ -25,7 +25,11 @@
 set_and_check(exec_prefix "@exec_prefix@")
 set_and_check(includedir "@includedir@")
 set_and_check(GWENGUI_CPP_INCLUDE_DIRS "@gwenhywfar_headerdir@")
-set_and_check(GWENGUI_CPP_LIBRARIES "@libdir@/libgwengui-cpp.so")
+if(APPLE)
+  set_and_check(GWENGUI_CPP_LIBRARIES "@libdir@/libgwengui-cpp.dylib")
+else()
+  set_and_check(GWENGUI_CPP_LIBRARIES "@libdir@/libgwengui-cpp.so")
+endif(APPLE)
 set(GWENGUI_CPP_FOUND true)
 
 if(GWENGUI_CPP_FOUND AND NOT TARGET gwenhywfar::gui-cpp)
