--- gui/qt4/gwengui-qt4-config.cmake.in	2016-10-20 23:45:28.000000000 +0200
+++ gui/qt4/gwengui-qt4-config.cmake.in.new	2016-10-21 01:31:16.000000000 +0200
@@ -44,7 +44,11 @@
 set_and_check(exec_prefix "@exec_prefix@")
 set_and_check(includedir "@includedir@")
 set_and_check(${CMAKE_FIND_PACKAGE_NAME}_INCLUDE_DIRS "@gwenhywfar_headerdir@")
-set_and_check(${CMAKE_FIND_PACKAGE_NAME}_LIBRARIES "@libdir@/libgwengui-qt4.so")
+if(APPLE)
+  set_and_check(${CMAKE_FIND_PACKAGE_NAME}_LIBRARIES "@libdir@/libgwengui-qt4.dylib")
+else()
+  set_and_check(${CMAKE_FIND_PACKAGE_NAME}_LIBRARIES "@libdir@/libgwengui-qt4.so")
+endif(APPLE)
 set(${CMAKE_FIND_PACKAGE_NAME}_FOUND true)
 
 if(${CMAKE_FIND_PACKAGE_NAME}_FOUND AND NOT TARGET gwenhywfar::gui-qt4)
