--- configure.orig	2014-10-13 10:00:17.000000000 -0500
+++ configure	2020-07-01 18:35:17.000000000 -0500
@@ -3932,6 +3932,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
   main() {
   #ifdef __SUNPRO_C
         exit(0);
@@ -4603,6 +4604,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
     main() {
     #ifdef _WIN32
 	  exit(0);
@@ -4633,6 +4635,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
     main() {
     #ifdef _WIN32
 	  exit(0);
@@ -4685,6 +4688,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <stdio.h>
+#include <stdlib.h>
       main() {
 	  long long ll;
 	  unsigned long long ull;
