--- lib/parser-yang.h.orig	2013-06-04 07:34:00.000000000 -0500
+++ lib/parser-yang.h	2020-07-01 18:59:26.000000000 -0500
@@ -5,6 +5,7 @@
 
 #include <stdio.h>
 
+#include "smi-data.h"
 #include "yang-data.h"
 
 
@@ -15,7 +16,11 @@
 
 extern int yangparse();
 
+extern void smiyyerror(char *msg, Parser *parserPtr);
 
+void yangSemanticAnalysis(_YangNode *module);
+
+void yangValidateInclude(_YangNode *module, _YangNode *extModule);
 
 #endif /* _PARSER_YANG_H */
 
