--- configure.orig	2007-07-19 08:24:45.000000000 +1000
+++ configure	2018-04-21 22:33:59.000000000 +1000
@@ -21171,6 +21171,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
+#include <cstddef>
 int
 main ()
 {
@@ -21878,7 +21879,7 @@
 echo "${ECHO_T}yes" >&6; }
 	hasgnutlsextra=yes
 fi
-if test $hasgnutls = "yes" ; then
+if test $hasgnutlsextra = "yes" ; then
     CPPFLAGS="$CPPFLAGS $GNUTLSEXTRA_CFLAGS"
     LDFLAGS="$LDFLAGS $GNUTLSEXTRA_LIBS"
 
