--- src/IMG_webp.c	2024-01-02 05:35:11.000000000 +0800
+++ src/IMG_webp.c	2024-03-31 17:21:16.000000000 +0800
@@ -71,11 +71,11 @@
     if (lib.FUNC == NULL) { IMG_SetError("Missing webpdemux.framework"); return -1; }
 #endif
 
-int IMG_InitWEBP()
 #ifdef __APPLE__
     /* Need to turn off optimizations so weak framework load check works */
     __attribute__ ((optnone))
 #endif
+int IMG_InitWEBP()
 {
     if (lib.loaded == 0) {
 #if defined(LOAD_WEBP_DYNAMIC) && defined(LOAD_WEBPDEMUX_DYNAMIC)
