From: Santiago Vila <sanvila@debian.org>
Subject: Fixed typo in procmail(1)
Bug-Debian: http://bugs.debian.org/142983
X-Debian-version: 3.22-4

--- src/manconf.c.orig
+++ src/manconf.c
@@ -233,7 +233,7 @@
  \2-@PRESERVOPT@\1and\1.BR \2-@FROMWHOPT@ .\1");
   pc("LMTPOPT",LMTPOPT);
 #else
-  ps("LMTPOPTdesc","");ps("LMTPusage","");
+  ps("LMTPOPTdesc","");ps("LMTPusage","\1");
 #endif
   pname("INIT_UMASK",0);printf("0%lo/g\n",(unsigned long)INIT_UMASK);lines--;
   pn("DEFlinebuf",DEFlinebuf);
