scons: Use MacPorts' Python interpreter

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Lawrence Velázquez <larryv@macports.org>

Index: script/scons
===================================================================
--- script/scons.orig
+++ script/scons
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!__PYTHON__
 #
 # SCons - a Software Constructor
 #
Index: script/scons-configure-cache
===================================================================
--- script/scons-configure-cache.orig
+++ script/scons-configure-cache
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!__PYTHON__
 #
 # SCons - a Software Constructor
 #
Index: script/scons-time
===================================================================
--- script/scons-time.orig
+++ script/scons-time
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!__PYTHON__
 #
 # scons-time - run SCons timings and collect statistics
 #
Index: script/sconsign
===================================================================
--- script/sconsign.orig
+++ script/sconsign
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!__PYTHON__
 #
 # SCons - a Software Constructor
 #
