--- makefile	Sat Sep  4 20:18:44 2004
+++ makefile.new	Sun Sep  5 01:42:05 2004
@@ -8,12 +8,12 @@
 USEPAK = 1
 
 PREFIX=$(DESTDIR)/usr
-BINDIR = $(PREFIX)/games/
+BINDIR = $(PREFIX)/bin/
 DATADIR = $(PREFIX)/share/games/parallelrealities/
-DOCDIR = $(PREFIX)/share/doc/$(PROG)-$(VERSION)-$(RELEASE)/
-ICONDIR = $(PREFIX)/share/icons/
-KDE = $(PREFIX)/share/applnk/Games/Arcade/
-GNOME = $(PREFIX)/share/gnome/apps/Games/
+DOCDIR = $(PREFIX)/share/doc/$(PROG)/
+ICONDIR = $(PREFIX)/crap/
+KDE = $(PREFIX)/crap/
+GNOME = $(PREFIX)/crap/
 
 CFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DTEMPDIR=\"/tmp/\" -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
@@ -63,8 +63,8 @@
 	mkdir -p $(GNOME)
 
 	strip $(PROG)
-	install -o root -g games -m 755 $(PROG) $(BINDIR)$(PROG)
-	install -o root -g games -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
+	install -m 755 $(PROG) $(BINDIR)$(PROG)
+	install -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
 	cp $(DOCS) $(DOCDIR)
 	cp $(ICONS)$(PROG).png $(ICONDIR)
 	cp $(ICONS)$(PROG)-mini.png $(ICONDIR)mini/$(PROG).png
@@ -81,4 +81,4 @@
 	$(RM) $(ICONDIR)large/$(PROG).png
 	$(RM) $(KDE)$(ICONS)$(PROG).desktop
 	$(RM) $(GNOME)$(ICONS)$(PROG).desktop
-	
\ No newline at end of file
+	
