--- makefile.orig	2011-04-17 09:56:56.000000000 -0500
+++ makefile	2014-10-15 13:57:31.000000000 -0500
@@ -9,7 +9,7 @@
 USEPAK ?= 0
 
 PREFIX ?= /usr
-BINDIR ?= $(PREFIX)/games/
+BINDIR ?= $(PREFIX)/libexec/blobwars/
 DATADIR ?= $(PREFIX)/share/games/blobwars/
 DOCDIR ?= $(PREFIX)/share/doc/$(PROG)/
 ICONDIR ?= $(PREFIX)/share/icons/hicolor/
@@ -21,8 +21,8 @@
 CXXFLAGS += `sdl-config --cflags` -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DUSEPAK=$(USEPAK)
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\" -DMEDAL_SERVER_HOST=\"$(MEDAL_SERVER_HOST)\" -DMEDAL_SERVER_PORT=$(MEDAL_SERVER_PORT)
-CXXFLAGS += $(CFLAGS) -Werror
-LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz
+CXXFLAGS += $(CFLAGS)
+LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz -lintl
 
 OBJS += CAudio.o
 OBJS += CBoss.o
@@ -85,7 +85,7 @@
 	$(CXX) $(LIBS) $(GAMEOBJS) -o $(PROG)
 	
 pak: $(PAKOBJS)
-	$(CXX) $(LIBS) $(PAKOBJS) -o pak
+	$(CXX) -lz $(PAKOBJS) -o pak
 
 %.mo: %.po
 	msgfmt -c -o $@ $<
