--- makefile	Wed Oct 27 09:39:20 2004
+++ makefile.new	Wed Oct 27 09:39:49 2004
@@ -8,7 +8,7 @@
 USEPAK = 1
 
 PREFIX=$(DESTDIR)/usr
-BINDIR = $(PREFIX)/games/
+BINDIR = $(PREFIX)/bin/
 DATADIR = $(PREFIX)/share/games/parallelrealities/
 DOCDIR = $(PREFIX)/share/doc/$(PROG)/
 ICONDIR = $(PREFIX)/share/icons/
@@ -63,9 +63,9 @@
 	mkdir -p $(GNOME)
 
 	strip $(PROG)
-	install -o root -g games -m 755 $(PROG) $(BINDIR)$(PROG)
-	install -o root -g games -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
-	install -o root -g games -m 644 $(DOCS) $(DOCDIR)
+	install -m 755 $(PROG) $(BINDIR)$(PROG)
+	install -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
+	install -m 644 $(DOCS) $(DOCDIR)
 	cp $(ICONS)$(PROG).png $(ICONDIR)
 	cp $(ICONS)$(PROG)-mini.png $(ICONDIR)mini/$(PROG).png
 	cp $(ICONS)$(PROG)-large.png $(ICONDIR)large/$(PROG).png
@@ -81,4 +81,4 @@
 	$(RM) $(ICONDIR)large/$(PROG).png
 	$(RM) $(KDE)$(ICONS)$(PROG).desktop
 	$(RM) $(GNOME)$(ICONS)$(PROG).desktop
-	
\ No newline at end of file
+	
