--- makefile.orig	2010-03-07 04:58:04.000000000 -0600
+++ makefile	2010-03-30 04:08:11.000000000 -0500
@@ -8,7 +8,7 @@
 USEPAK = 1
 
 PREFIX=$(DESTDIR)/usr
-BINDIR = $(PREFIX)/games/
+BINDIR = $(PREFIX)/bin/
 DATADIR = $(PREFIX)/share/games/blobwars/
 DOCDIR = $(PREFIX)/share/doc/$(PROG)/
 ICONDIR = $(PREFIX)/share/icons/hicolor/
@@ -22,7 +22,7 @@
 CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEPLAYMANUAL=\"$(DOCDIR)index.html\" -Wall
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\" -DMEDAL_SERVER_HOST=\"$(MEDAL_SERVER_HOST)\" -DMEDAL_SERVER_PORT=$(MEDAL_SERVER_PORT)
 CXXFLAGS += $(CFLAGS) -Werror
-LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz
+LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz -lintl
 
 OBJS += CAudio.o
 OBJS += CBoss.o
@@ -108,9 +108,9 @@
 	mkdir -p $(ICONDIR)64x64/apps
 	mkdir -p $(DESKTOPDIR)
 
-	install -o root -g games -m 755 $(PROG) $(BINDIR)$(PROG)
-	install -o root -g games -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
-	install -o root -g games -m 644 $(DOCS) $(DOCDIR)
+	install -m 755 $(PROG) $(BINDIR)$(PROG)
+	install -m 644 $(PAKNAME) $(DATADIR)$(PAKNAME)
+	install -m 644 $(DOCS) $(DOCDIR)
 	cp $(ICONS)$(PROG).png $(ICONDIR)32x32/apps/
 	cp $(ICONS)$(PROG)-mini.png $(ICONDIR)16x16/apps/$(PROG).png
 	cp $(ICONS)$(PROG)-large.png $(ICONDIR)64x64/apps/$(PROG).png
