--- configure.orig	2002-12-11 20:42:38.000000000 +1100
+++ configure	2011-01-24 14:42:54.000000000 +1100
@@ -10734,24 +10734,6 @@
 fi
 
 
-if test "_$trpath" != _; then
-if test "_$sortpath" != _; then
-if test "_$uniqpath" != _; then
-
-echo $GUI_CFLAGS > cflags
-tr " " "\n" < cflags > cflags2
-sort cflags2 > cflags
-uniq cflags > cflags2
-tr "\n" " " < cflags2 > cflags
-GUI_CFLAGS=`cat cflags`
-rm -f cflags cflags2
-
-fi
-fi
-fi
-
-
-
 
 ac_config_files="$ac_config_files Makefile src/Makefile src/common/Makefile src/fe-text/Makefile src/fe-gtk/Makefile intl/Makefile po/Makefile.in"
 cat >confcache <<\_ACEOF
