--- orig.Makefile.in	2017-03-21 19:40:17.000000000 +0100
+++ Makefile.in	2017-04-30 14:14:43.000000000 +0200
@@ -93,7 +93,7 @@
 LIB_OBJS=effect_util.o util.o effect.o effect_chain.o init.o resource_pool.o fp16.o ycbcr.o $(INPUTS:=.o) $(EFFECTS:=.o)
 
 # Default target:
-all: libmovit.la $(TESTS)
+all: libmovit.la
 
 ifeq ($(with_demo_app),yes)
 all: demo
@@ -112,8 +112,8 @@
 $(TESTS): %: %.o $(TEST_OBJS) libmovit.la
 	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -o $@ $^ $(TEST_LDLIBS)
 
-OWN_OBJS=$(DEMO_OBJS) $(LIB_OBJS) $(OWN_TEST_OBJS) $(TESTS:=.o)
-OBJS=$(DEMO_OBJS) $(LIB_OBJS) $(TEST_OBJS) $(TESTS:=.o)
+OWN_OBJS=$(DEMO_OBJS) $(LIB_OBJS)
+OBJS=$(DEMO_OBJS) $(LIB_OBJS)
 
 # A small demo program.
 demo: libmovit.la $(DEMO_OBJS)
