--- Makefile.in	2023-07-15 23:41:56.000000000 +0800
+++ Makefile.in	2023-07-25 13:39:39.000000000 +0800
@@ -92,7 +92,7 @@
 LIB_OBJS=effect_util.o util.o effect.o effect_chain.o init.o resource_pool.o ycbcr.o bundled_shaders.o $(INPUTS:=.o) $(EFFECTS:=.o)
 
 # Default target:
-all: libmovit.la $(TESTS)
+all: libmovit.la
 
 ifeq ($(with_demo_app),yes)
 all: demo
@@ -111,8 +111,8 @@
 $(TESTS): %: %.o $(TEST_OBJS) libmovit.la
 	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -o $@ $^ $(TEST_LDLIBS)
 
-OWN_OBJS=$(DEMO_OBJS) $(LIB_OBJS) $(OWN_TEST_OBJS) $(TESTS:=.o) make_bundled_shaders.o
-OBJS=$(DEMO_OBJS) $(LIB_OBJS) $(TEST_OBJS) $(TESTS:=.o) make_bundled_shaders.o
+OWN_OBJS=$(DEMO_OBJS) $(LIB_OBJS) make_bundled_shaders.o
+OBJS=$(DEMO_OBJS) $(LIB_OBJS) make_bundled_shaders.o
 
 # A small demo program.
 demo: libmovit.la $(DEMO_OBJS)
