--- orig.Makefile.in	2017-03-21 19:40:17.000000000 +0100
+++ Makefile.in	2017-04-30 14:14:43.000000000 +0200
@@ -29,7 +29,7 @@
 LDFLAGS=@LDFLAGS@
 LDLIBS=@epoxy_LIBS@ @FFTW3_LIBS@ -lpthread
 TEST_LDLIBS=@epoxy_LIBS@ @SDL2_LIBS@ @SDL_LIBS@ -lpthread
-DEMO_LDLIBS=@SDL2_image_LIBS@ @SDL_image_LIBS@ -lrt -lpthread @libpng_LIBS@ @FFTW3_LIBS@
+DEMO_LDLIBS=@SDL2_image_LIBS@ @SDL_image_LIBS@ -lpthread @libpng_LIBS@ @FFTW3_LIBS@
 SHELL=@SHELL@
 LIBTOOL=@LIBTOOL@ --tag=CXX
 RANLIB=ranlib
