--- doc/Makefile.orig	2012-03-15 04:26:00.000000000 +0900
+++ doc/Makefile	2012-08-21 14:06:16.000000000 +0900
@@ -1,6 +1,5 @@
 all:	latex2rtf.info latex2rtf.pdf latex2rtf.txt \
-		latex2pn.txt latex2rt.txt \
-		web/usage.html web/manpage.html web/latex2png.html
+		latex2pn.txt latex2rt.txt latex2rtf.html
 
 latex2rtf.html:	latex2rtf.texi
 	makeinfo --html --no-split --no-headers latex2rtf.texi > latex2rtf.html
