diff --git src/CMakeLists.txt src/CMakeLists.txt
index 6f8168a..5252dbd 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -137,18 +137,3 @@ install(TARGETS freedv
     RUNTIME DESTINATION bin
 )
 endif(APPLE)
-
-# Custom commands to build OSX images.
-if(APPLE)
-    add_custom_command(
-        TARGET FreeDV
-        POST_BUILD
-        COMMAND rm -rf dist_tmp || true
-        COMMAND DYLD_LIBRARY_PATH=${CODEC2_BUILD_DIR}/src:${LPCNET_BUILD_DIR}/src:${DYLD_LIBRARY_PATH} ${CMAKE_SOURCE_DIR}/macdylibbundler/dylibbundler ARGS -od -b -x FreeDV.app/Contents/MacOS/FreeDV -d FreeDV.app/Contents/libs -p @loader_path/../libs/ -i /usr/lib -s ${LPCNET_BUILD_DIR}/src -s ${CODEC2_BUILD_DIR}/src 
-        COMMAND cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/freedv.icns FreeDV.app/Contents/Resources
-        COMMAND mkdir dist_tmp
-        COMMAND cp -r FreeDV.app dist_tmp
-        COMMAND hdiutil create -srcfolder dist_tmp/ -volname FreeDV -format UDZO -fs HFS+ ./FreeDV.dmg
-        COMMAND rm -rf dist_tmp
-    )
-endif(APPLE)
