--- src/CMakeLists.txt.old	2019-04-10 21:29:07.000000000 +0200
+++ src/CMakeLists.txt	2019-04-10 21:29:37.000000000 +0200
@@ -70,10 +70,5 @@
         COMMAND cp ARGS ${CMAKE_CURRENT_SOURCE_DIR}/freedv.icns FreeDV.app/Contents/Resources
         COMMAND echo ARGS -n "APPL????" > FreeDV.app/Contents/PkgInfo
         COMMAND cp ARGS freedv FreeDV.app/Contents/MacOS/FreeDV
-        COMMAND dylibbundler ARGS -od -b -x FreeDV.app/Contents/MacOS/FreeDV -d FreeDV.app/Contents/libs -p @executable_path/../libs/
-        COMMAND mkdir dist_tmp
-        COMMAND cp -r FreeDV.app dist_tmp
-        COMMAND hdiutil create -srcfolder dist_tmp/ -volname FreeDV -format UDZO ./FreeDV.dmg
-        COMMAND rm -rf dist_tmp
     )
 endif(APPLE)
