--- Makefile.in	2007-12-23 16:01:39.000000000 -0600
+++ Makefile.in	2007-12-23 16:02:52.000000000 -0600
@@ -20,6 +20,7 @@
 INSTALL_DATA	= @INSTALL_DATA@
 MKINSTALLDIRS = @top_srcdir@/mkinstalldirs
 
+install_prefix = @install_prefix@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 libdir = $(exec_prefix)/lib
@@ -53,13 +54,13 @@
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I$(srcdir) -I. $(CFLAGS) $(PICFLAGS) $<
 
 install: all
-	$(MKINSTALLDIRS) $(includedir)
-	$(MKINSTALLDIRS) $(libdir)
-	$(MKINSTALLDIRS) $(mandir)
-	$(MKINSTALLDIRS) $(man3dir)
-	$(INSTALL) -m 0555 $(LIB) $(libdir)
-	$(INSTALL) -m 0644 krbafs.h $(includedir)
-	$(INSTALL) -m 0644 $(srcdir)/krbafs.3 $(man3dir)/krbafs.3
+	$(MKINSTALLDIRS) $(install_prefix)$(includedir)
+	$(MKINSTALLDIRS) $(install_prefix)$(libdir)
+	$(MKINSTALLDIRS) $(install_prefix)$(mandir)
+	$(MKINSTALLDIRS) $(install_prefix)$(man3dir)
+	$(INSTALL) -m 0555 $(LIB) $(install_prefix)$(libdir)
+	$(INSTALL) -m 0644 krbafs.h $(install_prefix)$(includedir)
+	$(INSTALL) -m 0644 $(srcdir)/krbafs.3 $(install_prefix)$(man3dir)/krbafs.3
 
 uninstall:
 
