diff --git c/src/FTFont/FTBufferFont.cpp i/src/FTFont/FTBufferFont.cpp
index ce04cf56ae0f..b330a3b119f7 100644
--- c/src/FTFont/FTBufferFont.cpp
+++ i/src/FTFont/FTBufferFont.cpp
@@ -230,11 +230,11 @@ inline FTPoint FTBufferFontImpl::RenderI(const T* string, const int len,
     int width, height, texWidth, texHeight;
     int cacheIndex = -1;
     bool inCache = false;
 
     // Protect blending functions, GL_TEXTURE_2D and optionally GL_BLEND
-    glPushAttrib(GL_COLOR_BUFFER_BIT | GL_ENABLE_BIT | GL_TEXTURE_ENV_MODE);
+    glPushAttrib(GL_COLOR_BUFFER_BIT | GL_ENABLE_BIT | GL_TEXTURE_BIT);
 
     // Protect glPixelStorei() calls
     glPushClientAttrib(GL_CLIENT_PIXEL_STORE_BIT);
 
     if(FTLibrary::Instance().GetLegacyOpenGLStateSet())
diff --git c/src/FTFont/FTTextureFont.cpp i/src/FTFont/FTTextureFont.cpp
index 97e876876e12..a5145cf05f53 100644
--- c/src/FTFont/FTTextureFont.cpp
+++ i/src/FTFont/FTTextureFont.cpp
@@ -239,11 +239,11 @@ template <typename T>
 inline FTPoint FTTextureFontImpl::RenderI(const T* string, const int len,
                                           FTPoint position, FTPoint spacing,
                                           int renderMode)
 {
     // Protect GL_TEXTURE_2D and optionally GL_BLEND
-    glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT | GL_TEXTURE_ENV_MODE);
+    glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT | GL_TEXTURE_BIT);
 
     if(FTLibrary::Instance().GetLegacyOpenGLStateSet())
       {
         glEnable(GL_BLEND);
         /*
