--- configure	Mon Aug 25 06:18:22 2003
+++ configure	Sun Feb  8 18:43:43 2004
@@ -1032,9 +1032,9 @@
   --with-x                use the X Window System
   --with-ft-prefix=PFX      Prefix where FreeType is installed (optional)
   --with-ft-exec-prefix=PFX Exec prefix where FreeType is installed (optional)
-  --with-gl-inc=DIR       Directory where GL/gl.h is installed
+  --with-gl-inc=DIR       Directory where OpenGL/gl.h is installed
   --with-gl-lib=DIR       Directory where OpenGL libraries are installed
-  --with-glut-inc=DIR     Directory where GL/glut.h is installed (optional)
+  --with-glut-inc=DIR     Directory where GLUT/glut.h is installed (optional)
   --with-glut-lib=DIR     Directory where GLUT libraries are installed
                           (optional)
 
@@ -9342,8 +9342,8 @@
 
 CPPFLAGS="$GL_CFLAGS"
 if test "${ac_cv_header_GL_gl_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for GL/gl.h" >&5
-echo $ECHO_N "checking for GL/gl.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for OpenGL/gl.h" >&5
+echo $ECHO_N "checking for OpenGL/gl.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_gl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -9351,8 +9351,8 @@
 echo "${ECHO_T}$ac_cv_header_GL_gl_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking GL/gl.h usability" >&5
-echo $ECHO_N "checking GL/gl.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking OpenGL/gl.h usability" >&5
+echo $ECHO_N "checking OpenGL/gl.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9361,7 +9361,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <GL/gl.h>
+#include <OpenGL/gl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -9387,8 +9387,8 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking GL/gl.h presence" >&5
-echo $ECHO_N "checking GL/gl.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking OpenGL/gl.h presence" >&5
+echo $ECHO_N "checking OpenGL/gl.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9396,7 +9396,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <GL/gl.h>
+#include <OpenGL/gl.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -9429,10 +9429,10 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: GL/gl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: GL/gl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/gl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/gl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/gl.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: OpenGL/gl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/gl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: OpenGL/gl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9443,12 +9443,12 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: GL/gl.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: GL/gl.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/gl.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: GL/gl.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/gl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/gl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/gl.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: OpenGL/gl.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/gl.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: OpenGL/gl.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/gl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: OpenGL/gl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9459,8 +9459,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for GL/gl.h" >&5
-echo $ECHO_N "checking for GL/gl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for OpenGL/gl.h" >&5
+echo $ECHO_N "checking for OpenGL/gl.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_gl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -9473,8 +9473,8 @@
 if test $ac_cv_header_GL_gl_h = yes; then
   :
 else
-  { { echo "$as_me:$LINENO: error: GL/gl.h is needed" >&5
-echo "$as_me: error: GL/gl.h is needed" >&2;}
+  { { echo "$as_me:$LINENO: error: OpenGL/gl.h is needed" >&5
+echo "$as_me: error: OpenGL/gl.h is needed" >&2;}
    { (exit please specify its location with --with-gl-inc.  If this still fails); exit please specify its location with --with-gl-inc.  If this still fails; }; }
 fi
 
@@ -9484,12 +9484,12 @@
 echo $ECHO_N "checking for GL library... $ECHO_C" >&6
 if test "x$with_gl_lib" != "x" ; then
     if test -d "$with_gl_lib" ; then
-        LIBS="-L$with_gl_lib -lGL"
+        LIBS="-L$with_gl_lib -framework OpenGL -lobjc"
     else
         LIBS="$with_gl_lib"
     fi
 else
-    LIBS="-lGL"
+    LIBS="-framework OpenGL -lobjc"
 fi
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
@@ -9536,7 +9536,7 @@
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 if test "x$HAVE_GL" = xno ; then
     if test "x$GL_X_LIBS" != x ; then
-        LIBS="-lGL $GL_X_LIBS"
+        LIBS="-framework OpenGL -lobjc $GL_X_LIBS"
         cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9595,8 +9595,8 @@
 fi
 
 if test "${ac_cv_header_GL_glu_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for GL/glu.h" >&5
-echo $ECHO_N "checking for GL/glu.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for OpenGL/glu.h" >&5
+echo $ECHO_N "checking for OpenGL/glu.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_glu_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -9604,8 +9604,8 @@
 echo "${ECHO_T}$ac_cv_header_GL_glu_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking GL/glu.h usability" >&5
-echo $ECHO_N "checking GL/glu.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking OpenGL/glu.h usability" >&5
+echo $ECHO_N "checking OpenGL/glu.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9614,7 +9614,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <GL/glu.h>
+#include <OpenGL/glu.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -9640,8 +9640,8 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking GL/glu.h presence" >&5
-echo $ECHO_N "checking GL/glu.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking OpenGL/glu.h presence" >&5
+echo $ECHO_N "checking OpenGL/glu.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9649,7 +9649,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <GL/glu.h>
+#include <OpenGL/glu.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -9682,10 +9682,10 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: GL/glu.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: GL/glu.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glu.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/glu.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/glu.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: OpenGL/glu.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/glu.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: OpenGL/glu.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9696,12 +9696,12 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: GL/glu.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: GL/glu.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glu.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: GL/glu.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glu.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/glu.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/glu.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: OpenGL/glu.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/glu.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: OpenGL/glu.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: OpenGL/glu.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: OpenGL/glu.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9712,8 +9712,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for GL/glu.h" >&5
-echo $ECHO_N "checking for GL/glu.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for OpenGL/glu.h" >&5
+echo $ECHO_N "checking for OpenGL/glu.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_glu_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -9734,7 +9734,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <GL/glu.h>
+#include <OpenGL/glu.h>
 int
 main ()
 {
@@ -9776,7 +9776,7 @@
 
 echo "$as_me:$LINENO: checking for GLU library" >&5
 echo $ECHO_N "checking for GLU library... $ECHO_C" >&6
-LIBS="-lGLU $GL_LIBS"
+LIBS="-framework OpenGL -lobjc $GL_LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9822,7 +9822,7 @@
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 if test "x$HAVE_GLU" = xno ; then
     if test "x$GL_X_LIBS" != x ; then
-        LIBS="-lGLU $GL_LIBS $GL_X_LIBS"
+        LIBS="-framework OpenGL -lobjc $GL_LIBS $GL_X_LIBS"
         cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9936,8 +9936,8 @@
 
 CPPFLAGS="$GLUT_CFLAGS"
 if test "${ac_cv_header_GL_glut_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for GL/glut.h" >&5
-echo $ECHO_N "checking for GL/glut.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for GLUT/glut.h" >&5
+echo $ECHO_N "checking for GLUT/glut.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_glut_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -9945,8 +9945,8 @@
 echo "${ECHO_T}$ac_cv_header_GL_glut_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking GL/glut.h usability" >&5
-echo $ECHO_N "checking GL/glut.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking GLUT/glut.h usability" >&5
+echo $ECHO_N "checking GLUT/glut.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9955,7 +9955,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <GL/glut.h>
+#include <GLUT/glut.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -9981,8 +9981,8 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking GL/glut.h presence" >&5
-echo $ECHO_N "checking GL/glut.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking GLUT/glut.h presence" >&5
+echo $ECHO_N "checking GLUT/glut.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9990,7 +9990,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <GL/glut.h>
+#include <GLUT/glut.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -10023,10 +10023,10 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: GL/glut.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: GL/glut.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glut.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/glut.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: GLUT/glut.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: GLUT/glut.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: GLUT/glut.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: GLUT/glut.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -10037,12 +10037,12 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: GL/glut.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: GL/glut.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glut.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: GL/glut.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: GL/glut.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: GL/glut.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: GLUT/glut.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: GLUT/glut.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: GLUT/glut.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: GLUT/glut.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: GLUT/glut.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: GLUT/glut.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -10053,8 +10053,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for GL/glut.h" >&5
-echo $ECHO_N "checking for GL/glut.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for GLUT/glut.h" >&5
+echo $ECHO_N "checking for GLUT/glut.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_glut_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -10083,12 +10083,12 @@
 echo $ECHO_N "checking for GLUT library... $ECHO_C" >&6
     if test "x$with_glut_lib" != "x" ; then
         if test -d "$with_glut_lib" ; then
-            LIBS="-L$with_glut_lib -lglut"
+            LIBS="-L$with_glut_lib -framework GLUT -framework OpenGL -lobjc"
         else
             LIBS="$with_glut_lib"
         fi
     else
-        LIBS="-lglut"
+        LIBS="-framework GLUT -framework OpenGL -lobjc"
     fi
     cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
@@ -10136,7 +10136,7 @@
 
     if test "x$HAVE_GLUT" = xno ; then
         # Try again with the GL libs
-        LIBS="-lglut $GL_LIBS"
+        LIBS="-framework GLUT -framework OpenGL -lobjc $GL_LIBS"
         cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -10184,7 +10184,7 @@
 
     if test "x$HAVE_GLUT" = xno && test "x$GLUT_X_LIBS" != x ; then
         # Try again with the GL and X11 libs
-        LIBS="-lglut $GL_LIBS $GLUT_X_LIBS"
+        LIBS="-framework GLUT -framework OpenGL -lobjc $GL_LIBS $GLUT_X_LIBS"
         cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
