diff --git a/src/applicationinfo.cpp b/src/applicationinfo.cpp
index b3cbc09..39a0d42 100644
--- a/src/applicationinfo.cpp
+++ b/src/applicationinfo.cpp
@@ -26,8 +26,8 @@
 // Should also be overridable through an optional file.
 
 #define PROG_NAME "Psi"
 //#define PROG_VERSION "0.14-dev" " (" __DATE__ ")"; //CVS Builds are dated
-#define PROG_VERSION "0.14";
+#define PROG_VERSION "0.14-patched";
 #define PROG_CAPS_NODE "http://psi-im.org/caps";
 #define PROG_CAPS_VERSION "caps-b75d8d2b25";
 #define PROG_IPC_NAME "org.psi-im.Psi"	// must not contain '\\' character on Windows
