--- src/CocoaUtilities/CocoaTrayClick.cpp.orig	2018-11-01 17:15:39.000000000 -0700
+++ src/CocoaUtilities/CocoaTrayClick.cpp	2018-11-27 19:04:29.000000000 -0700
@@ -46,7 +46,7 @@
 CocoaTrayClick::CocoaTrayClick()
 	: QObject(qApp)
 {
-	Class cls = objc_getClass("NSApplication");
+	id cls = objc_getClass("NSApplication");
 	objc_object *appInst = objc_msgSend((objc_object*)cls, sel_registerName("sharedApplication"));
 
 	if(appInst != NULL) {
