--- configure	2008-12-30 16:39:07.000000000 +0100
+++ configure	2008-12-30 16:49:59.000000000 +0100
@@ -40,6 +40,8 @@
   --with-aspell-inc=[path]     Path to Aspell include files
   --with-aspell-lib=[path]     Path to Aspell library files
   --enable-tests               Enable use of Tests
+  --disable-plugins            Disable use of Plugins
+  --enable-plugins             Enable Psi Plugin support
   --enable-debug               Enable debugging support
 
 EOT
@@ -242,6 +244,16 @@
 			shift
 			;;
 
+        --disable-plugins)
+            QC_DISABLE_plugins="Y"
+            shift
+            ;;
+
+        --enable-plugins)
+            QC_ENABLE_PLUGINS="Y"
+            shift
+            ;;
+
 		--enable-debug)
 			QC_ENABLE_DEBUG="Y"
 			shift
@@ -285,6 +297,8 @@
 echo QC_WITH_ASPELL_INC=$QC_WITH_ASPELL_INC
 echo QC_WITH_ASPELL_LIB=$QC_WITH_ASPELL_LIB
 echo QC_ENABLE_tests=$QC_ENABLE_tests
+echo QC_DISABLE_plugins=$QC_DISABLE_plugins
+echo QC_ENABLE_PLUGINS=$QC_ENABLE_PLUGINS
 echo QC_ENABLE_DEBUG=$QC_ENABLE_DEBUG
 echo
 fi
@@ -1061,6 +1075,48 @@
 		return true;
 	}
 };
+#line 1 "plugins.qcm"
+/*
+-----BEGIN QCMOD-----
+name: Plugins
+arg: enable-plugins,Enable Psi Plugin support
+
+-----END QCMOD-----
+*/
+
+//----------------------------------------------------------------------------
+// qc_plugins
+//----------------------------------------------------------------------------
+class qc_plugins : public ConfObj
+{
+public:
+   qc_plugins(Conf *c) : ConfObj(c) {}
+   QString name() const { return "Plugins"; }
+   QString shortname() const { return "Plugins"; }
+   bool exec()
+   {
+       QString s;
+      
+       // Check if Jingle was enabled explicitly
+       s = conf->getenv("QC_ENABLE_PLUGINS");
+       if(s.isEmpty())
+           return false;
+       
+       conf->addDefine("PSI_PLUGINS");
+       
+       // Finish
+       conf->addExtra("CONFIG += psi_plugins");                              
+
+       qWarning("");
+       qWarning("");
+       qWarning("        !!!!!!!!!!!!!! WARNING !!!!!!!!!!!!!!!");
+       qWarning("        PLUGIN SUPPORT IS STILL UNFINISHED !!!");
+       qWarning("        THE PLUGIN INTERFACE /WILL/ CHANGE !!!");
+       qWarning("        USE AT YOUR OWN RISK !!!");
+			 
+       return true;
+   }
+};
 #line 1 "tests.qcm"
 /*
 -----BEGIN QCMOD-----
@@ -1185,6 +1241,9 @@
     o = new qc_tests(conf);
     o->required = false;
     o->disabled = true;
+    o = new qc_plugins(conf);
+    o->required = false;
+    o->disabled = false;
     o = new qc_debug(conf);
     o->required = true;
     o->disabled = false;
@@ -2176,6 +2235,8 @@
 export QC_WITH_ASPELL_INC
 export QC_WITH_ASPELL_LIB
 export QC_ENABLE_tests
+export QC_DISABLE_plugins
+export QC_ENABLE_PLUGINS
 export QC_ENABLE_DEBUG
 export QC_VERBOSE
 rm -rf .qconftemp
