Index: src/pluginmanager.h
===================================================================
--- src/pluginmanager.h	(Revision 1195)
+++ src/pluginmanager.h	(Arbeitskopie)
@@ -4,9 +4,12 @@
 #include <QtCore>
 #include <QList>
 #include <QMap>
+#include <QToolButton>
+#include <QAction>
 
 #include "userlist.h"
 #include "optionstree.h"
+#include "../iris/include/im.h"
 
 class PsiPlugin;
 
@@ -22,6 +25,8 @@
 
 class QPluginLoader;
 
+using namespace XMPP;
+
 class PluginManager : public QObject
 {
 	Q_OBJECT
@@ -30,9 +35,15 @@
 
 	QStringList availablePlugins();
 
+	QList<QAction*> getChatDlgMenuEntries(QWidget* parent, 
+									      PsiAccount* account, 
+	                                      const XMPP::Jid& otherjid);
+
 	void addAccount( const PsiAccount* account, XMPP::Client* client);
-	void message( PsiAccount* account, const XMPP::Jid& from, 
-	const UserListItem*, const QString& message );
+	QString incomingMessage( PsiAccount* account, const XMPP::Jid& from, QString message );
+	HTMLElement incomingMessage( PsiAccount* account, const XMPP::Jid& from, HTMLElement htmlMessage );
+	QString outgoingMessage( PsiAccount* account, const XMPP::Jid& from, QString message );
+				
 	bool loadPlugin( const QString& file );
 	void loadEnabledPlugins();
 	bool unloadPlugin( const QString& file );
@@ -41,6 +52,13 @@
 	QString shortName( const QString& plugin );
 	QWidget* getOptionsWidget( const QString& plugin );
 	bool processEvent( const PsiAccount* account, QDomElement &event );
+	void statusChanged(QString account, QString jid, QString resource, int prio, QString status, 
+			   QString text);
+	void contactOffline(QString account, QString contact);
+	void contactOnline(QString account, QString contact);
+	void login(QString jid);
+	void logout(QString jid);
+	void quitProgram();
 	
 	const QVariant getGlobalOption(const QString& option);
 	
@@ -80,6 +98,8 @@
 	void optionChanged(const QString& option);
 	void sendStanza( const PsiAccount* account, const QDomElement& stanza);
 	void sendStanza( const PsiAccount* account, const QString& stanza);
+	void sendStanza(const QString& fromJid, const QString& stanza);
+	void getHomeDir(QString&);
 };
 
 #endif
