--- pyproject.toml.orig	2023-03-11 15:00:28.000000000 +0300
+++ pyproject.toml	2023-03-11 15:00:20.000000000 +0300
@@ -1,6 +1,6 @@
 [build-system]
 requires = ["flit_core >=2,<4", "wheel"]
-build-backend = ["flit_core.buildapi", "flit_core.wheel"]
+build-backend = "flit_core.buildapi"
 
 [tool.flit.metadata]
 module = "pyglet"
@@ -17,4 +17,4 @@
 
 [tool.flit.sdist]
 include = ["pyglet"]
-#exclude = ["doc/*.html"]
\ No newline at end of file
+#exclude = ["doc/*.html"]
