--- configure.orig	2017-09-15 23:26:41.000000000 +0900
+++ configure	2017-09-15 23:26:59.000000000 +0900
@@ -10614,6 +10614,7 @@
 ac_res=$ac_cv_search___gmpz_init
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
+  $as_echo "#define HAVE_LIBGMP 1" >>confdefs.h
 
 fi
 
@@ -10683,6 +10684,7 @@
 ac_res=$ac_cv_search_malloc_conf
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
+  $as_echo "#define HAVE_LIBJEMALLOC 1" >>confdefs.h
 
 else
   with_jemalloc=no
