--- src/bin/Makefile.am.orig	2012-07-27 22:49:43.000000000 +0900
+++ src/bin/Makefile.am	2012-09-04 01:37:00.000000000 +0900
@@ -4,13 +4,13 @@
 bin_PROGRAMS = train-lader lader evaluate-lader label-ranking
 
 train_lader_SOURCES = train-lader.cc
-train_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+train_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 
 lader_SOURCES = lader.cc
-lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 
 evaluate_lader_SOURCES = evaluate-lader.cc
-evaluate_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+evaluate_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 
 label_ranking_SOURCES = label-ranking.cc
-label_ranking_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+label_ranking_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
--- src/bin/Makefile.in.orig	2012-09-04 01:39:38.000000000 +0900
+++ src/bin/Makefile.in	2012-09-04 01:52:05.000000000 +0900
@@ -200,13 +200,13 @@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I$(srcdir)/../include
 train_lader_SOURCES = train-lader.cc
-train_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+train_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 lader_SOURCES = lader.cc
-lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 evaluate_lader_SOURCES = evaluate-lader.cc
-evaluate_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+evaluate_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 label_ranking_SOURCES = label-ranking.cc
-label_ranking_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+label_ranking_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 all: all-am
 
 .SUFFIXES:
--- src/test/Makefile.am.orig	2012-07-28 08:11:33.000000000 +0900
+++ src/test/Makefile.am	2012-09-04 02:14:03.000000000 +0900
@@ -5,4 +5,4 @@
 bin_PROGRAMS = test-lader
 
 test_lader_SOURCES = test-lader.cc ${LADERTH}
-test_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+test_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
--- src/test/Makefile.in.orig	2012-07-28 08:13:02.000000000 +0900
+++ src/test/Makefile.in	2012-09-04 02:14:23.000000000 +0900
@@ -199,7 +199,7 @@
 LADERTH = test-alignments-and-ranks.h test-feature-vector.h test-base.h test-hyper-graph.h test-dictionary.h test-loss-bracket.h test-feature-align.h test-loss-chunk.h test-feature-parse.h test-loss-tau.h test-feature-sequence.h test-reorderer-model.h test-feature-set.h
 AM_CPPFLAGS = -I$(srcdir)/../include -DPKGDATADIR='"$(pkgdatadir)"'
 test_lader_SOURCES = test-lader.cc ${LADERTH}
-test_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt
+test_lader_LDADD = ../lib/liblader.la -lboost_regex-mt -lboost_thread-mt -lboost_system-mt
 all: all-am
 
 .SUFFIXES:
