--- build.gradle.orig	2025-06-30 07:16:11
+++ build.gradle	2026-09-13 16:24:54
@@ -36,7 +36,7 @@
 
 dependencies {
     jaxenSupportApi 'jaxen:jaxen:1.1.6'
-    xsdlibSupportApi 'net.java.dev.msv:xsdlib:2022.7'
+    xsdlibSupportApi files('@@PREFIX@@/share/java/msv/xsdlib.jar')
     jaxbSupportApi 'jakarta.xml.bind:jakarta.xml.bind-api:4.0.2'
     pullParserSupportApi 'pull-parser:pull-parser:2.1.10'
     xppSupportApi 'xpp3:xpp3:1.1.4c'
@@ -44,8 +44,8 @@
     testImplementation(
             'org.testng:testng:7.10.2',
 
-            'xerces:xercesImpl:2.12.2',
-            'xalan:xalan:2.7.2',
+            files('@@PREFIX@@/share/java/xercesj/xercesImpl.jar'),
+            files('@@PREFIX@@/share/java/xalanj/xalan.jar'),
     )
 }
 
