https://llvm.org/bugs/show_bug.cgi?id=23238

--- a/src/cxa_handlers.cpp	2014-10-03 13:03:47.000000000 -0700
+++ b/src/cxa_handlers.cpp	2015-04-15 05:26:25.000000000 -0700
@@ -102,7 +102,7 @@ terminate() _NOEXCEPT
     __terminate(get_terminate());
 }
 
-new_handler __cxa_new_handler = 0;
+extern "C" new_handler __cxa_new_handler = 0;
 // In the future these will become:
 // std::atomic<std::new_handler>  __cxa_new_handler(0);
 
