--- Makefile.PL.orig	2019-02-20 23:20:39.000000000 -0500
+++ Makefile.PL	2020-05-20 21:44:45.000000000 -0400
# from
# https://github.com/DragonFlyBSD/DPorts/blob/master/textproc/p5-xmltv/files/patch-Makefile.PL
@@ -715,14 +715,14 @@
         # Guess a default value for {install} based on whether
         # prerequisites were found.
         #
-        $info->{install} = (not $info->{exclude}) && ($opt_yes || not $info->{missing});
+        $info->{install} = 1;(not $info->{exclude}) && ($opt_yes || not $info->{missing});
 
         print STDERR ($s, ' ' x (1 + $width - length $s),
                       $info->{install} ? '[yes]' : '[no]',
                       "\n");
     }
     print STDERR "\n";
-    if (not ask(0, 'Do you want to proceed with this configuration?', 1)) {
+    if (1 or not ask(0, 'Do you want to proceed with this configuration?', 1)) {
         # Need to set {install} for each component by prompting.
         foreach my $info (@opt_components) {
             my $missing = $info->{missing};
@@ -751,8 +751,8 @@
                 die;
             }
 
-            $info->{install} =
-              ask(0, $msg, not $missing);
+            $info->{install} = 1;
+            #  ask(0, $msg, not $missing);
         }
     }
 }
@@ -956,8 +956,8 @@
     }
 
     # Remove existing non-working 'uninstall' target.
-    $inherited =~ s!^uninstall\s:.*$!!m
-      or die "no uninstall target in: $inherited";
+    $inherited =~ s!^uninstall\s::?.*?\n\t+.*$!!m;
+    $inherited =~ s!^uninstall\s:.*$!!m;
 
     # For each *_install create a corresponding _uninstall.
     my $targets = ::targets($inherited);
