--- src/formats.cpp	2015-03-23 04:27:46.000000000 +0800
+++ src/formats.cpp	2024-05-28 16:25:38.000000000 +0800
@@ -36,7 +36,7 @@
 
 void Formats::loadFormats()
 {
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_DARWIN)
     list<CoderInfo> coderList;
     coderInfoList(&coderList,
                   CoderInfo::TrueMatch,
@@ -106,7 +106,6 @@
             s_writableFilters << line;
         } while (!line.isNull());
     }
-    s_writableFilters.removeLast();
 #endif
 
     s_writableFilters.prepend(tr("Don't change the format"));
