Index: jcc2/sources/JArray.cpp
===================================================================
--- jcc2/sources/JArray.cpp	(revision 1812454)
+++ jcc2/sources/JArray.cpp	(revision 1812455)
@@ -312,7 +312,7 @@
     PyObject *list = toSequence<U>(self);
 
     if (list != NULL &&
-        PyList_Type.tp_as_sequence->sq_inplace_concat(list, arg) < 0)
+        PyList_Type.tp_as_sequence->sq_inplace_concat(list, arg) == NULL)
     {
         Py_DECREF(list);
         return NULL;
@@ -327,7 +327,7 @@
     PyObject *list = toSequence<U>(self);
 
     if (list != NULL &&
-        PyList_Type.tp_as_sequence->sq_inplace_repeat(list, n) < 0)
+        PyList_Type.tp_as_sequence->sq_inplace_repeat(list, n) == NULL)
     {
         Py_DECREF(list);
         return NULL;
Index: jcc3/sources/JArray.cpp
===================================================================
--- jcc3/sources/JArray.cpp	(revision 1812454)
+++ jcc3/sources/JArray.cpp	(revision 1812455)
@@ -312,7 +312,7 @@
     PyObject *list = toSequence<U>(self);
 
     if (list != NULL &&
-        PyList_Type.tp_as_sequence->sq_inplace_concat(list, arg) < 0)
+        PyList_Type.tp_as_sequence->sq_inplace_concat(list, arg) == NULL)
     {
         Py_DECREF(list);
         return NULL;
@@ -327,7 +327,7 @@
     PyObject *list = toSequence<U>(self);
 
     if (list != NULL &&
-        PyList_Type.tp_as_sequence->sq_inplace_repeat(list, n) < 0)
+        PyList_Type.tp_as_sequence->sq_inplace_repeat(list, n) == NULL)
     {
         Py_DECREF(list);
         return NULL;
