--- work/highlight-3.9/makefile	2010-07-28 23:16:48.000000000 +0200
+++ makefile	2012-06-09 10:20:54.000000000 +0200
@@ -5,10 +5,10 @@
 # Installation directories:
 
 # Destination directory for installation (intended for packagers)
-DESTDIR =
+DESTDIR = @destroot@
 
 # Root directory for final installation
-PREFIX = /usr
+PREFIX = @prefix@
 
 # Location of the highlight data files:
 data_dir = ${PREFIX}/share/highlight/
@@ -26,7 +26,7 @@
 examples_dir = ${doc_dir}examples/
 
 # Location of the highlight config files:
-conf_dir = /etc/highlight/
+conf_dir = ${PREFIX}/etc/highlight/
 #conf_dir = ${PREFIX}/etc/highlight/
 
 # Location of additional gui files
