Index: ARBDB/arbdb.cxx
===================================================================
--- ARBDB/arbdb.cxx	(revision 11958)
+++ ARBDB/arbdb.cxx	(working copy)
@@ -447,6 +447,7 @@
     return inside_shell;
 }
 
+#if 0
 struct GB_test_shell_closed {
     ~GB_test_shell_closed() {
         if (GB_shell::in_shell()) { // leave that call
@@ -455,6 +456,7 @@
     }
 };
 static GB_test_shell_closed test;
+#endif
 
 #if defined(UNIT_TESTS)
 static bool closed_open_shell_for_unit_tests() {
