--- Makefile.orig	2011-11-10 10:19:41.000000000 -0500
+++ Makefile	2012-03-24 12:16:11.000000000 -0400
@@ -42,10 +42,15 @@
 
 # ---------------------- [unconditionally used options]
 
-GCC:=gcc
-GPP:=g++ 
-CPPreal:=cpp
-
+ifdef DARWIN
+	#GCC:=gcc      <Now set in the Portfile>
+	#GPP:=g++      <Now set in the Portfile>
+	#CPPreal:=cpp  <Now set in the Portfile>
+else
+	GCC:=gcc
+	GPP:=g++ 
+	CPPreal:=cpp
+endif
 
 
 # ---------------------- compiler version detection
@@ -71,10 +76,10 @@
 #---------------------- define special directories for non standard builds
 
 ifdef DARWIN
-OSX_SDK:=/Developer/SDKs/MacOSX10.5.sdk
-OSX_FW:=/System/Library/Frameworks
-OSX_FW_OPENGL:=$(OSX_FW)/OpenGL.framework/Versions/A/Libraries
-OSX_FW_IMAGEIO:=$(OSX_FW)/ApplicationServices.framework/Versions/A/Frameworks/ImageIO.framework/Versions/A/Resources
+	OSX_FW:=/System/Library/Frameworks
+	OSX_FW_OPENGL:=$(OSX_FW)/OpenGL.framework/Versions/A/Libraries
+	OSX_FW_GLUT:=$(OSX_FW)/GLUT.framework/Versions/A/Libraries
+	OSX_FW_IMAGEIO:=$(OSX_FW)/ApplicationServices.framework/Versions/A/Frameworks/ImageIO.framework/Versions/A/Resources
 endif
 
 #----------------------
@@ -186,7 +191,6 @@
 dflags += -D$(MACH) # define machine
 
 ifdef DARWIN
-	cflags += -no-cpp-precomp 
 	shared_cflags += -fno-common
 else
 	dflags +=  $(shell getconf LFS_CFLAGS)
@@ -199,16 +203,23 @@
 
 #---------------------- X11 location
 
-XHOME:=/usr/X11R6
+ifdef DARWIN
+	XHOME:=$(PREFIX)
+else
+	XHOME:=/usr/X11R6
+endif
+
+
 XINCLUDES:=-I$(XHOME)/include
 
 ifdef DARWIN
-	XINCLUDES := -I/sw/include -I$(OSX_SDK)/usr/X11/include -I$(OSX_SDK)/usr/include/krb5 -I/usr/OpenMotif/include #Snow Leopard couldn't find OpenMotif
+	XINCLUDES += -I$(OSX_FW)/GLUT.framework/Headers -I$(OSX_FW)/OpenGL.framework/Headers -I$(OSX_SDK)/usr/include/krb5
 endif
 
 ifdef DARWIN
-	XLIBS := -L/usr/OpenMotif/lib -lXm -L$(XHOME)/lib -lpng -lXt -lX11 -lXext -lc -lXmu -lXi
-	XLIBS += -lGLU -lGL -Wl,-dylib_file,$(OSX_FW_OPENGL)/libGL.dylib:$(OSX_FW_OPENGL)/libGL.dylib
+	XLIBS := -L$(XHOME)/lib -lXm -lpng -lz -lXt -lX11 -lXext -lXp -lXmu -lXi
+	XLIBS += -Wl,-dylib_file,$(OSX_FW_OPENGL)/libGL.dylib:$(OSX_FW_OPENGL)/libGL.dylib
+	XLIBS += -Wl,-dylib_file,$(OSX_FW_OPENGL)/libGLU.dylib:$(OSX_FW_OPENGL)/libGLU.dylib
 else
 	XLIBS:=-L$(XHOME)/$(CROSS_LIB) -lXm -lXpm -lXt -lXext -lX11
 endif
@@ -226,13 +237,8 @@
 
 GL_PNGLIBS := -L$(ARBHOME)/GL/glpng -lglpng_arb -lpng
 
- ifdef DARWIN
-	GLEWLIB := -L/usr/lib -lGLEW -L$(OSX_SDK)/usr/X11/lib -lGLw
-	GLUTLIB := -L$(XHOME)/lib -lglut
- else 
 	GLEWLIB := -lGLEW -lGLw
-	GLUTLIB := -lglut
- endif
+	GLUTLIB := -glut
 
 GL_LIBS := $(GL_LIB) $(GLEWLIB) $(GLUTLIB) $(GL_PNGLIBS)
 
@@ -248,11 +254,7 @@
 
 #---------------------- tiff lib:
 
-ifdef DARWIN
-	TIFFLIBS := -L/usr/local/lib -ltiff -L$(OSX_FW_IMAGEIO) -lTIFF  
-else
-	TIFFLIBS := -ltiff
-endif
+TIFFLIBS := -ltiff
 
 #---------------------- basic libs:
 
@@ -318,11 +320,13 @@
 # other used tools
 
 CTAGS := etags
+
 ifdef DARWIN
-	XMKMF := /usr/X11/bin/xmkmf
+	XMKMF := $(PREFIX)/bin/xmkmf
 else
 	XMKMF := /usr/bin/X11/xmkmf
 endif
+
 MAKEDEPEND_PLAIN = makedepend
 
 MAKEDEPEND = $(FORCEMASK);$(MAKEDEPEND_PLAIN)
@@ -532,7 +536,12 @@
 ARBDBPP_LIB=-lARBDBPP
 
 LIBS = $(ARBDB_LIB) $(SYSLIBS)
-GUI_LIBS = $(LIBS) -lAW -lAWT $(XLIBS)
+
+ifdef DARWIN
+	GUI_LIBS = -framework GLUT -framework OpenGL $(LIBS) -lAW -lAWT $(XLIBS)
+else
+	GUI_LIBS = $(LIBS) -lAW -lAWT $(XLIBS)
+endif
 
 LIBPATH = -L$(ARBHOME)/LIBLINK
 
