--- CMakeLists.txt.orig	2012-06-01 07:52:21.000000000 -0400
+++ CMakeLists.txt	2013-07-27 00:16:47.000000000 -0400
@@ -127,7 +127,7 @@
 
 if( APPLE OR ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   # apple doesn't have 64bit versions of file opening functions, so add them
-  add_definitions("-Dfopen64=fopen -Dfseeko64=fseeko -Dfseek64=fseek -Dftell64=ftell -Dftello64=ftello")
+  add_definitions(-DUSE_FILE32API)
 endif()
 
 set(COLLADA_DOM_INCLUDE_INSTALL_DIR "include/collada-dom${COLLADA_DOM_SOVERSION}")
