--- configure.orig	2008-12-05 07:52:18.000000000 +1100
+++ configure	2009-01-12 23:12:39.000000000 +1100
@@ -23135,16 +23135,16 @@
   dbusint64_constant='(val)'
   dbusuint64_constant='(val)'
   ;;
-$ac_cv_sizeof_long)
-  dbusint64=long
-  dbusint64_constant='(val##L)'
-  dbusuint64_constant='(val##UL)'
-  ;;
 $ac_cv_sizeof_long_long)
   dbusint64='long long'
   dbusint64_constant='(val##LL)'
   dbusuint64_constant='(val##ULL)'
   ;;
+$ac_cv_sizeof_long)
+  dbusint64=long
+  dbusint64_constant='(val##L)'
+  dbusuint64_constant='(val##UL)'
+  ;;
 $ac_cv_sizeof___int64)
   dbusint64=__int64
   dbusint64_constant='(val##i64)'
