--- doc/Makefile.in.orig	2009-10-01 13:04:07.000000000 -0500
+++ doc/Makefile.in	2011-10-21 05:52:45.000000000 -0500
@@ -161,7 +161,7 @@
 
 SUBDIRS = spec
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 static_docs = vp3-format.txt color.html \
 	draft-ietf-avt-rtp-theora-00.xml \
--- doc/spec/Makefile.in.orig	2009-10-01 13:04:07.000000000 -0500
+++ doc/spec/Makefile.in	2011-10-21 05:57:02.000000000 -0500
@@ -162,7 +162,7 @@
 ltdl_LTLIBOBJS = @ltdl_LTLIBOBJS@
 sys_symbol_underscore = @sys_symbol_underscore@
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 built_docs = Theora.pdf
 
