--- Makefile.orig	2011-07-22 19:20:21.000000000 -0500
+++ Makefile	2020-11-21 17:34:35.000000000 -0600
@@ -578,9 +578,6 @@
 	@echo + In order to rectify this problem, you MUST build with:
 	@echo ++ SSLTYPE=$(SSLTYPE).nopwd
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-	@echo
-	@echo Do you want to continue this build anyway?  Type y or n please:
-	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make nounenc;exit 1);; esac'
 
 nounenc:
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
