--- configure.ac.orig	2018-03-24 11:33:13.000000000 -0700
+++ configure.ac	2018-03-24 11:33:55.000000000 -0700
@@ -123,7 +123,7 @@
 fi
 
 dnl get rid of the -export-dynamic stuff from the configure flags ...
-export_dynamic=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
+export_dynamic=`(./glibtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
 
 dnl glib
 AM_PATH_GLIB_2_0(glib_required_version,,[AC_MSG_ERROR(maybe you want the pygobject-2-4 branch?)],gobject $extra_mods)
