--- configure	1998-05-13 11:39:40.000000000 -0400
+++ configure	2026-04-30 16:19:00.000000000 -0400
@@ -769,7 +769,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 771 "configure"
 #include "confdefs.h"
-main(){return(0);}
+int main(){return(0);}
 EOF
 if { (eval echo configure:775: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then
   ac_cv_prog_cc_works=yes
@@ -1797,7 +1797,7 @@
     }
   }
 }
-main() {
+int main() {
   pid_t parent = getpid ();
   pid_t child;
 
