--- cephes/Make.common	1998-05-12 08:17:15.000000000 -0400
+++ cephes/Make.common	2026-05-01 21:01:53.000000000 -0400
@@ -9,7 +9,7 @@
 	jn.c jv.c k0.c k1.c kn.c log2.c mtherr.c ndtri.c pdtr.c polevl.c \
 	polyn.c psi.c revers.c rgamma.c round.c shichi.c sici.c \
 	spence.c stdtr.c struve.c unity.c yn.c zeta.c zetac.c \
-	acosh.c asinh.c atanh.c ndtr.c isfinite.c cbrt.c
+	acosh.c asinh.c atanh.c ndtr.c cbrt.c

 OBJS =	airy$(O) beta$(O) chbevl$(O) chdtr$(O) const$(O) dawsn$(O) ellie$(O) \
         ellik$(O) ellpe$(O) ellpk$(O) expn$(O) fac$(O) fdtr$(O) fresnl$(O) \
@@ -19,5 +19,5 @@
 	polevl$(O) polyn$(O) psi$(O) revers$(O) rgamma$(O) round$(O) \
 	shichi$(O) sici$(O) spence$(O) stdtr$(O) struve$(O) \
 	unity$(O) yn$(O) zeta$(O) zetac$(O) \
-	acosh$(O) asinh$(O) atanh$(O) ndtr$(O) isfinite$(O) cbrt$(O)
+	acosh$(O) asinh$(O) atanh$(O) ndtr$(O) cbrt$(O)

--- cmath.h	1998-05-03 18:20:04.000000000 -0400
+++ cmath.h	2026-05-01 20:55:29.000000000 -0400
@@ -102,10 +102,3 @@
 extern double jn ( int n, double x );
 #endif

-#ifndef HAVE_FINITE
-#  define finite isfinite
-#  ifndef HAVE_ISFINITE
-extern int isfinite ( double x );
-#  endif
-#endif
-
--- src/fit.c	1997-11-14 17:13:35.000000000 -0500
+++ src/fit.c	2026-05-01 21:09:37.000000000 -0400
@@ -117,7 +117,7 @@
 	dofitcurve(n, x, y, ideg, result);
 	for (i = 0; i < n; i++) {
 	    fitted[i] = leasev(result, ideg, x[i]);
-	    if (!finite(fitted[i])) {
+	    if (!isfinite(fitted[i])) {
 	        errmsg("Linear_regression - all values of x or y are the same");
 		ifail = 3;
 		return ifail;
