--- u2f-server/Makefile.in.orig	2016-12-24 09:50:51.000000000 +0000
+++ u2f-server/Makefile.in	2016-12-24 09:51:13.000000000 +0000
@@ -384,7 +384,7 @@
 	internal.h u2f-server.pc.in u2f-server.map global.c version.c \
 	error.c core.c sha256.h sha256.c cencode.c cdecode.c \
 	b64/cencode.h b64/cdecode.h crypto.h openssl.c
-libu2f_server_la_LIBADD = $(HIDAPI_LIBS) $(LIBJSON_LIBS) $(LIBSSL_LIBS)
+libu2f_server_la_LIBADD = $(HIDAPI_LIBS) $(LIBJSON_LIBS) $(LIBSSL_LIBS) $(LIBCRYPTO_LIBS)
 libu2f_server_la_LDFLAGS = -no-undefined -version-info \
 	$(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(am__append_1) \
 	$(am__append_2)
