--- tls/tls_internal.h.orig	2015-02-22 13:39:36.000000000 -0800
+++ tls/tls_internal.h	2015-03-24 04:34:38.000000000 -0700
@@ -23,7 +23,7 @@
 
 #define HTTPS_PORT "443"
 
-#define _PATH_SSL_CA_FILE "/etc/ssl/cert.pem"
+#define _PATH_SSL_CA_FILE OPENSSLDIR"/cert.pem"
 
 #define TLS_CIPHERS_COMPAT	"ALL:!aNULL:!eNULL"
 #define TLS_CIPHERS_DEFAULT	"TLSv1.2+AEAD+ECDHE:TLSv1.2+AEAD+DHE"
--- tls/Makefile.am.orig	2015-08-07 20:41:08.000000000 -0700
+++ tls/Makefile.am	2015-08-07 20:42:32.000000000 -0700
@@ -5,6 +5,14 @@ lib_LTLIBRARIES = libtls.la
 EXTRA_DIST = VERSION
 EXTRA_DIST += CMakeLists.txt
 
+libtls_la_CPPFLAGS = $(AM_CPPFLAGS)
+
+if OPENSSLDIR_DEFINED
+libtls_la_CPPFLAGS += -DOPENSSLDIR=\"@OPENSSLDIR@\"
+else
+libtls_la_CPPFLAGS += -DOPENSSLDIR=\"$(sysconfdir)/ssl\"
+endif
+
 libtls_la_LDFLAGS = -version-info @LIBTLS_VERSION@ -no-undefined
 libtls_la_LIBADD = ../crypto/libcrypto.la ../ssl/libssl.la $(PLATFORM_LDADD)
 
