Index: ./kab/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kab/Makefile.am,v
retrieving revision 1.45
diff -u -b -u -r1.45 Makefile.am
--- ./kab/Makefile.am	29 Dec 2002 13:27:45 -0000	1.45
+++ ./kab/Makefile.am	16 Jun 2003 14:07:29 -0000
@@ -9,7 +9,7 @@
 	addressbook.cc \
 	qconfigDB.cc 
 
-libkab_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined
+libkab_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -no-undefined
 libkab_la_LIBADD = ../kdeui/libkdeui.la
 
 libkab_la_METASOURCES = AUTO
