Index: kio/kio/kprotocolmanager.cpp
===================================================================
RCS file: /home/kde/kdelibs/kio/kio/kprotocolmanager.cpp,v
retrieving revision 1.127
diff -u -b -u -r1.127 kprotocolmanager.cpp
--- kio/kio/kprotocolmanager.cpp	21 Jun 2003 00:24:26 -0000	1.127
+++ kio/kio/kprotocolmanager.cpp	26 Jun 2003 14:54:03 -0000
@@ -428,7 +428,17 @@
     }
     if( modifiers.contains('p') )
     {
-      supp += QString::fromLatin1("; X11");  // TODO: determine this valye instead of hardcoding...
+#ifdef Q_WS_X11
+      supp += QString::fromLatin1("; X11");
+#elif defined(Q_WS_MACX)
+      supp += QString::fromLatin1("; Mac OS X");
+#elif defined(Q_WS_QWS)
+      supp += QString::fromLatin1("; Embedded");
+#elif defined(Q_WS_WIN)
+      supp += QString::fromLatin1("; Windows");
+#else
+      supp += QString::fromLatin1("; Unknown");
+#endif
     }
     if( modifiers.contains('m') )
     {
