Index: kabc/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kabc/Makefile.am,v
retrieving revision 1.89
diff -u -u -r1.89 Makefile.am
--- kabc/Makefile.am	28 Jun 2003 07:31:56 -0000	1.89
+++ kabc/Makefile.am	28 Jun 2003 21:18:29 -0000
@@ -1,5 +1,7 @@
 SUBDIRS = scripts vcard vcardparser . formats plugins
 
+KDE_OPTIONS = nofinal
+
 # Make sure $(all_includes) remains last!
 INCLUDES = -I$(top_builddir)/kabc -I$(srcdir)/vcard/include -I$(srcdir)/vcard/include/generated \
   -I$(top_srcdir)/kabc -I$(top_srcdir)/kab -I$(srcdir)/vcardparser $(all_includes)
@@ -17,7 +19,7 @@
 
 lib_LTLIBRARIES   = libkabc.la
 libkabc_la_COMPILE_FIRST = addressee.h
-libkabc_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 3:0:2 -module
+libkabc_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 3:0:2
 libkabc_la_LIBADD  = vcard/libvcard.la vcardparser/libvcards.la ../kdeui/libkdeui.la ../kio/libkio.la \
   ../kresources/libkresources.la
 
