Index: kio/kio/kurlcompletion.cpp
===================================================================
RCS file: /home/kde/kdelibs/kio/kio/kurlcompletion.cpp,v
retrieving revision 1.51
diff -u -b -u -r1.51 kurlcompletion.cpp
--- kio/kio/kurlcompletion.cpp	2 Jun 2003 13:27:29 -0000	1.51
+++ kio/kio/kurlcompletion.cpp	26 Jun 2003 14:54:05 -0000
@@ -54,11 +54,6 @@
 
 #include "kurlcompletion.h"
 
-#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)
-# include <crt_externs.h>
-# define environ (*_NSGetEnviron())
-#endif
-
 static bool expandTilde(QString &);
 static bool expandEnv(QString &);
 
