Index: kdeprint/cups/cupsdconf2/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf2/Makefile.am,v
retrieving revision 1.9
diff -u -b -u -r1.9 Makefile.am
--- kdeprint/cups/cupsdconf2/Makefile.am	11 Sep 2002 12:24:07 -0000	1.9
+++ kdeprint/cups/cupsdconf2/Makefile.am	27 Jun 2003 13:14:44 -0000
@@ -17,7 +17,7 @@
 			  locationdialog.cpp addressdialog.cpp cups-util.c \
 			  qdirmultilineedit.cpp sizewidget.cpp
 libcupsdconf_la_METASOURCES = AUTO
-libcupsdconf_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
+libcupsdconf_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
 libcupsdconf_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
 
 noinst_HEADERS = cupsdconf.h cupsdpage.h cupsddialog.h cupsdcomment.h cupsdsplash.h \
