Index: kdeui/kedittoolbar.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeui/kedittoolbar.cpp,v
retrieving revision 1.76
diff -u -b -u -r1.76 kedittoolbar.cpp
--- kdeui/kedittoolbar.cpp	11 Jun 2003 17:28:00 -0000	1.76
+++ kdeui/kedittoolbar.cpp	26 Jun 2003 14:53:16 -0000
@@ -151,13 +151,13 @@
    */
   ToolbarList findToolbars(QDomElement elem)
   {
-    static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
-    static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
+    // static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
+    // static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
     ToolbarList list;
 
     for( ; !elem.isNull(); elem = elem.nextSibling().toElement() )
     {
-      if (elem.tagName() == tagToolbar && elem.attribute( attrNoEdit ) != "true" )
+      if (elem.tagName() == KGlobal::staticQString( "ToolBar" ) && elem.attribute( KGlobal::staticQString( "noEdit" ) ) != "true" )
         list.append(elem);
 
       QDomElement child = elem.firstChild().toElement();
