--- kcert/Makefile.am	4 Jan 2003 04:28:18 -0000	1.1.1.4
+++ kcert/Makefile.am	8 Mar 2003 23:48:52 -0000	1.2
@@ -1,4 +1,4 @@
-INCLUDES= $(SSL_INCLUDES) -I$(top_srcdir)/kio/kssl -I$(top_builddir)/kio/kssl -I$(top_srcdir) $(all_includes)
+INCLUDES= -I$(top_srcdir)/kio/kssl -I$(top_builddir)/kio/kssl -I$(top_srcdir) $(all_includes) $(SSL_INCLUDES)
 
 lib_LTLIBRARIES = libkcertpart.la
 KCERTPARTVERSION=	2.2.0
