--- kdeprint/cups/cupsdconf2/Makefile.am	1 Oct 2002 03:05:37 -0000	1.1.1.3
+++ kdeprint/cups/cupsdconf2/Makefile.am	1 Jun 2003 14:59:02 -0000	1.5
@@ -3,22 +3,32 @@
 INCLUDES= -I$(top_srcdir) -I$(top_srcdir)/kio -I$(top_srcdir)/kfile $(all_includes)
 
 bin_PROGRAMS = cupsdconf
-cupsdconf_SOURCES = main.cpp
 
-cupsdconf_LDADD   =  libcupsdconf.la
+cupsdconf_SOURCES = cupsdconf_main.cpp
+cupsdconf_LDADD   = libcupsdconf_main.la
 cupsdconf_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+cupsdconf_main.cpp: stub_main.cpp
+	cat stub_main.cpp > cupsdconf_main.cpp
+
 # library creation
-lib_LTLIBRARIES = libcupsdconf.la
-libcupsdconf_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
+lib_LTLIBRARIES = libcupsdconf_main.la cupsdconf.la
+libcupsdconf_main_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
 			  cupsdserverpage.cpp cupsdlogpage.cpp cupsdjobspage.cpp cupsdfilterpage.cpp \
 			  qdirlineedit.cpp cupsddirpage.cpp portdialog.cpp cupsdnetworkpage.cpp \
 			  editlist.cpp cupsdbrowsingpage.cpp browsedialog.cpp cupsdsecuritypage.cpp \
 			  locationdialog.cpp addressdialog.cpp cups-util.c \
-			  qdirmultilineedit.cpp sizewidget.cpp
-libcupsdconf_la_METASOURCES = AUTO
-libcupsdconf_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-libcupsdconf_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
+			  qdirmultilineedit.cpp sizewidget.cpp main.cpp
+libcupsdconf_main_la_METASOURCES = AUTO
+libcupsdconf_main_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
+libcupsdconf_main_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
+
+cupsdconf_la_SOURCES = cupsdconf_la_main.cpp
+cupsdconf_la_LDFLAGS = $(all_libraries) -avoid-version -module $(KDE_RPATH)
+cupsdconf_la_LIBADD  = libcupsdconf_main.la
+
+cupsdconf_la_main.cpp: stub_main.cpp
+	cat stub_main.cpp > cupsdconf_la_main.cpp
 
 noinst_HEADERS = cupsdconf.h cupsdpage.h cupsddialog.h cupsdcomment.h cupsdsplash.h \
 		 cupsdserverpage.h cupsdlogpage.h cupsdjobspage.h cupsdfilterpage.h \
