Index: khtml/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/khtml/Makefile.am,v
retrieving revision 1.144
diff -u -b -u -r1.144 Makefile.am
--- khtml/Makefile.am	8 Apr 2003 13:29:34 -0000	1.144
+++ khtml/Makefile.am	16 Jun 2003 14:30:31 -0000
@@ -28,7 +28,7 @@
 
 CLEANFILES = dummy.cpp
 
-lib_LTLIBRARIES = libkhtml.la
+lib_LTLIBRARIES = libkhtml.la khtml.la libkhtmlimagepart_main.la
 kde_module_LTLIBRARIES = khtmlimagepart.la
 
 libkhtml_la_SOURCES = khtmlview.cpp khtml_part.cpp khtml_run.cpp khtml_factory.cpp khtml_settings.cc \
@@ -46,7 +46,7 @@
 VSCRIPT = -Wl,--version-script=$(srcdir)/libkhtml.map
 endif
 
-libkhtml_la_LDFLAGS = -version-info 6:0:2 -module -no-undefined $(VSCRIPT) $(all_libraries)
+libkhtml_la_LDFLAGS = -version-info 6:0:2 -no-undefined $(VSCRIPT) $(all_libraries)
 libkhtml_la_LIBADD = ./xml/libkhtmlxml.la ./html/libkhtmlhtml.la \
 	./rendering/libkhtmlrender.la ./css/libkhtmlcss.la \
 	./misc/libkhtmlmisc.la ecma/libkjs_html.la ./dom/libkhtmldom.la $(LIBKJAVA) \
@@ -54,12 +54,23 @@
 	$(top_builddir)/kdeprint/libkdeprint.la  \
 	$(top_builddir)/kutils/libkutils.la
 
+khtml_la_LDFLAGS = -avoid-version -no-undefined $(all_libraries) -module
+khtml_la_LIBADD  = libkhtml.la
+khtml_la_SOURCES = dummy.cpp
+
 dummy.cpp:
 	echo > $@
 
-khtmlimagepart_la_SOURCES = dummy.cpp
-khtmlimagepart_la_LDFLAGS = -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
-khtmlimagepart_la_LIBADD = libkhtml.la 
+libkhtmlimagepart_main_la_SOURCES = dummy.cpp
+libkhtmlimagepart_main_la_LDFLAGS = -avoid-version -no-undefined -module $(KDE_PLUGIN) $(all_libraries)
+libkhtmlimagepart_main_la_LIBADD = libkhtml.la 
+
+khtmlimagepart_la_SOURCES = khtmlimagepart_dummy.cpp
+khtmlimagepart_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
+khtmlimagepart_la_LIBADD  = $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KPARTS) libkhtml.la
+
+khtmlimagepart_dummy.cpp:
+	echo > khtmlimagepart_dummy.cpp
 
 INCLUDES = -I$(top_srcdir)/kjs -I$(top_srcdir)/kimgio \
 	-I$(srcdir)/java -I$(top_srcdir)/dcop -I$(srcdir)/misc \
