--- fop.orig	2013-08-11 23:05:43.000000000 -0700
+++ fop	2013-08-11 23:06:31.000000000 -0700
@@ -80,7 +80,7 @@ darwin=false;
 case "`uname`" in
   CYGWIN*) cygwin=true ;;
   Darwin*) darwin=true
-           if [ -z "$JAVA_HOME" ] ; then
+           if [ -z "$JAVA_HOME" -a -e /System/Library/Frameworks/JavaVM.framework/Home ] ; then
              JAVA_HOME=/System/Library/Frameworks/JavaVM.framework/Home
            fi
            ;;
