diff --git Makefile Makefile
index 67b6d88..94120f6 100644
--- Makefile
+++ Makefile
@@ -23,7 +23,7 @@ PLUM_DATA = bin/rime-install \
 OPENCC_DATA = data/opencc/TSCharacters.ocd \
 	data/opencc/TSPhrases.ocd \
 	data/opencc/t2s.json
-DEPS_CHECK = $(RIME_LIBRARY) $(PLUM_DATA) $(OPENCC_DATA)
+DEPS_CHECK = $(PLUM_DATA)
 
 OPENCC_DATA_OUTPUT = librime/thirdparty/share/opencc/*.*
 PLUM_DATA_OUTPUT = plum/output/*.*
@@ -53,7 +53,7 @@ copy-rime-binaries:
 
 .PHONY: data plum-data opencc-data copy-plum-data copy-opencc-data
 
-data: plum-data opencc-data
+data: plum-data
 
 $(PLUM_DATA):
 	$(MAKE) plum-data
@@ -78,7 +78,7 @@ copy-opencc-data:
 	mkdir -p data/opencc
 	cp $(OPENCC_DATA_OUTPUT) data/opencc/
 
-deps: librime data
+deps: data
 
 release: $(DEPS_CHECK)
 	bash package/add_data_files
