--- ACE_wrappers.orig/ace/config.h	1969-12-31 16:00:00.000000000 -0800
+++ ACE_wrappers/ace/config.h	2012-02-10 20:59:39.000000000 -0800
@@ -0,0 +1,8 @@
+// ACE_LACKS_CLOCKID_T and ACE_LACKS_CLOCK_MONOTONIC were needed sometime before Mountain Lion
+// but are not needed for Sierra. Do not yet have info for releases in between.
+// #define ACE_LACKS_CLOCKID_T
+// #define ACE_LACKS_CLOCK_MONOTONIC
+// ACE_LACKS_CLOCK_REALTIME not necessary for Sierra. Do not yet have info for earlier releases.
+// #define ACE_LACKS_CLOCK_REALTIME
+
+#include "ace/config-macosx-@MACOSX@.h"
