diff -uNr gnokii-0.6.13/configure gnokii-0.6.13-new/configure
--- gnokii-0.6.13/configure	2006-06-18 13:02:08.000000000 -0400
+++ gnokii-0.6.13-new/configure	2006-08-15 11:32:58.000000000 -0400
@@ -27521,7 +27521,7 @@
 _ACEOF
 
 		USE_BLUETOOTH="yes"
-		BLUETOOTH_LIBS="$PTHREAD_LIBS -framework CoreFoundation -framework IOBluetooth -framework Foundation"
+		BLUETOOTH_LIBS="$PTHREAD_LIBS -Wl,-framework,CoreFoundation -Wl,-framework,IOBluetooth -Wl,-framework,Foundation"
 
 	fi
 fi

diff -uNr gnokii-0.6.13/xgnokii/xgnokii_data.c gnokii-0.6.13-new/xgnokii/xgnokii_data.c
--- gnokii-0.6.13/xgnokii/xgnokii_data.c	2006-06-18 13:02:00.000000000 -0400
+++ gnokii-0.6.13-new/xgnokii/xgnokii_data.c	2006-08-15 11:33:18.000000000 -0400
@@ -65,7 +65,7 @@
 }
 
 
-inline void GUI_ShowData(void)
+__attribute__ ((visibility("hidden"))) void GUI_ShowData(void)
 {
 	if (!phoneMonitor.supported & PM_DATA)
 		return;
@@ -81,7 +81,7 @@
 }
 
 
-inline void GUI_HideData(void)
+__attribute__ ((visibility("hidden"))) void GUI_HideData(void)
 {
 	gtk_widget_hide(GUI_DataWindow);
 }
diff -uNr gnokii-0.6.13/xgnokii/xgnokii_data.h gnokii-0.6.13-new/xgnokii/xgnokii_data.h
--- gnokii-0.6.13/xgnokii/xgnokii_data.h	2006-06-18 13:02:00.000000000 -0400
+++ gnokii-0.6.13-new/xgnokii/xgnokii_data.h	2006-08-15 11:33:18.000000000 -0400
@@ -31,7 +31,7 @@
 #define XGNOKII_DATA_H
 
 void GUI_CreateDataWindow(void);
-inline void GUI_ShowData(void);
-inline void GUI_HideData(void);
+__attribute__ ((visibility("hidden"))) void GUI_ShowData(void);
+__attribute__ ((visibility("hidden"))) void GUI_HideData(void);
 
 #endif
diff -uNr gnokii-0.6.13/xgnokii/xgnokii_lowlevel.h gnokii-0.6.13-new/xgnokii/xgnokii_lowlevel.h
--- gnokii-0.6.13/xgnokii/xgnokii_lowlevel.h	2006-06-18 13:02:00.000000000 -0400
+++ gnokii-0.6.13-new/xgnokii/xgnokii_lowlevel.h	2006-08-15 11:32:58.000000000 -0400
@@ -230,6 +230,6 @@
 extern void GUI_InsertEvent(PhoneEvent * event);
 extern gn_error GUI_InitSMSFolders(void);
 extern int isSMSactivated;
-struct gn_statemachine *statemachine;
+extern struct gn_statemachine *statemachine;
 
 #endif
