--- src/astrcplot.c.orig	2014-07-06 16:10:08.000000000 -0700
+++ src/astrcplot.c	2014-07-06 16:10:17.000000000 -0700
@@ -101,7 +101,7 @@
 /* Draw meridians and parallels */
   plfont(2);
   plschr(0.0, 0.3);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   mark[0] = 500;
   space[0] = 500;
@@ -148,7 +148,7 @@
 
 /* Plot fields */
   pllsty(1);
-  plwid(lwid);
+  plwidth(lwid);
   for (g=0; g<ngroup; g++)
     {
     field = fgroups[g]->field;
@@ -464,7 +464,7 @@
 
 /* Draw meridians */
   plschr(0.0, 0.33);
-  plwid(0);
+  plwidth(0);
   pllsty(2);
   xmd = xmu = xdo = -0.5;
   ymd = ymu = ydo = -0.5;
@@ -679,7 +679,7 @@
 
 /* Draw meridians */
   plschr(0.0, 0.33);
-  plwid(0);
+  plwidth(0);
   pllsty(2);
   xmd = xmu = xdo = -0.5;
   ymd = ymu = ydo = -0.5;
@@ -832,7 +832,7 @@
   plcol0(15);
   plschr(0.0, 0.67);
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
   sprintf(str, "#uGroup ##%d: detections", fgroup->no);
   pllab("","", str);
@@ -865,7 +865,7 @@
       }
     }
   plssym(0.0, psize);
-  plwid(lsize);
+  plwidth(lsize);
   plcol0(3);
   plpoin((PLINT)npoint, x,y, 11);
   plcol0(1);
@@ -902,7 +902,7 @@
             *(xt++) = (PLFLT)samp->projpos[lng];
             *(yt++) = (PLFLT)samp->projpos[lat];
             }
-        plwid(lsize);
+        plwidth(lsize);
         plcol0(8);
         if (lsize)
           plpoin((PLINT)(xt-x), x,y, 17);
@@ -913,7 +913,7 @@
         free(x);
         free(y);
         }
-      plwid(2*lwid);
+      plwidth(2*lwid);
       if (field->cplot_colour==15)
         {
         plcol0(15);
@@ -928,7 +928,7 @@
     field = fgroup->field[f];
     if (field->cplot_colour!=15)
       {
-      plwid(3*lwid);
+      plwidth(3*lwid);
       plcol0(field->cplot_colour);
       for (s=0; s<field->nset; s++)
         {
@@ -939,7 +939,7 @@
     }
 
   plcol0(7);
-  plwid(lwid);
+  plwidth(lwid);
   cplot_drawcoordgrid(wcs, xmin, xmax, ymin, ymax);
   plflush();
 
@@ -1041,13 +1041,13 @@
   ymin = 0.5;
   ymax = 100.5;
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plfont(2);
   plcol0(15);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
   sprintf(str, "#uInstrument A%d: distortion map", field->astromlabel+1);
   pllab("","", str);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   cplot_drawloccoordgrid(wcs, xmin, xmax, ymin, ymax);
 
@@ -1128,11 +1128,11 @@
              ystep/2.0+0.5, set->wcs->naxisn[lat]-ystep/2.0+0.5,
 	     clevel, CPLOT_NSHADES, 1, 0, 0, plfill, 0, distort_map, &distort);
 /*
-    plwid(0);
+    plwidth(0);
     cplot_drawfgrids(set->wcs, wcs);
 */
     plcol0(7);
-    plwid(lwid);
+    plwidth(lwid);
     cplot_drawbounds(set->wcs, wcs);
     }
 
@@ -1326,14 +1326,14 @@
   ymin = 0.5;
   ymax = 100.5;
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plfont(2);
   plcol0(15);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
   sprintf(str, "#uInstrument A%d: map of astrometric systematics (internal)", 
 	field0->astromlabel+1);
   pllab("","", str);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   cplot_drawloccoordgrid(wcs0, xmin, xmax, ymin, ymax);
   y[0] = y[1] = 3.0;
@@ -1341,9 +1341,9 @@
   x[1] = x[0] + scalel;
   pllsty(1);
   plcol0(15);
-  plwid(lwid*3);
+  plwidth(lwid*3);
   plline(2,x,y);
-  plwid(lwid);
+  plwidth(lwid);
   plschr(0.0, 0.5);
   plptex((x[0] + x[1]) / 2.0, y[0] + 2.5, 1.0, 0.0, 0.5, dispunit);
   x[0] = x[1] = 7.0;
@@ -1480,12 +1480,12 @@
           x[1] = vecpos2[lng] + minscale*(vecpos2[lng] - vecpos[lng]);
           y[0] = vecpos[lat];
           y[1] = vecpos2[lat] + minscale*(vecpos2[lat] - vecpos[lat]);
-          plwid(lwid*2);
+          plwidth(lwid*2);
           plcol0(3);
           plline(2, x,y);
           plcol0(15);
           plpoin(1,x,y,1);
-          plwid(lwid);
+          plwidth(lwid);
           }
       }
     }
@@ -1647,7 +1647,7 @@
   ymin = 0.5;
   ymax = 100.5;
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plfont(2);
   plcol0(15);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
@@ -1655,7 +1655,7 @@
   sprintf(str, "#uInstrument A%d: map of astrometric systematics (reference)",
 	field0->astromlabel+1);
   pllab("","", str);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   cplot_drawloccoordgrid(wcs0, xmin, xmax, ymin, ymax);
   y[0] = y[1] = 3.0;
@@ -1663,9 +1663,9 @@
   x[1] = x[0] + scalel;
   pllsty(1);
   plcol0(15);
-  plwid(lwid*3);
+  plwidth(lwid*3);
   plline(2,x,y);
-  plwid(lwid);
+  plwidth(lwid);
   plschr(0.0, 0.5);
   plptex((x[0] + x[1]) / 2.0, y[0] + 2.5, 1.0, 0.0, 0.5, dispunit);
   x[0] = x[1] = 7.0;
@@ -1791,12 +1791,12 @@
           x[1] = vecpos2[lng] + minscale*(vecpos2[lng] - vecpos[lng]);
           y[0] = vecpos[lat];
           y[1] = vecpos2[lat] + minscale*(vecpos2[lat] - vecpos[lat]);
-          plwid(lwid*2);
+          plwidth(lwid*2);
           plcol0(1);
           plline(2, x,y);
           plcol0(15);
           plpoin(1,x,y,1);
-          plwid(lwid);
+          plwidth(lwid);
           }
       }
     }
@@ -1869,7 +1869,7 @@
   plcol0(15);
   plschr(0.0,0.5);
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv(0.0, nfield + 1.0, ymin, ymax, 0, 0);
   sprintf(xlabel, "Field ##");
   sprintf(ylabel, "#gx#u2#d / d.o.f.");
@@ -1890,7 +1890,7 @@
   pllsty(1);
 /* Reference chi2/d.o.f. */
   plschr(0.0,0.5);
-  plwid(lwid);
+  plwidth(lwid);
   plenv(0.0, nfield + 1.0, ymin2, ymax2, 0, 0);
   sprintf(xlabel, "Field ##");
   sprintf(ylabel, "#gx#u2#d / d.o.f.");
@@ -2007,7 +2007,7 @@
   plcol0(15);
   plschr(0.0,0.5);
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv(airmassmin, airmassmax, shearmin, shearmax, 0, 0);
   sprintf(xlabel, "Airmass");
   sprintf(ylabel, "Contraction (%%)");
@@ -2245,7 +2245,7 @@
         cuty_hsn[d3][i] = fgroup->projposmin[d2] - margin
 			+ cuty_hsn[d3][i]/cutymax_hsn[d3] * 0.9*margin;
         }
-      plwid(lwid);
+      plwidth(lwid);
       plenv(fgroup->projposmin[d2]-margin, fgroup->projposmax[d2],
 		-maxlim, maxlim, 0, -2);
 /* Use a non-linear shade level distribution */
@@ -2283,13 +2283,13 @@
       plscol0(15, 0,0,0);	/* Force the foreground colour to black */
 /* 1D histograms */
       plcol0(3);
-      plwid(2*lwid);
+      plwidth(2*lwid);
       plline(CPLOT_NADERRHISTBIN, cuty[d3], cutbin);
       plcol0(7);
       plline(CPLOT_NADERRHISTBIN, cuty_hsn[d3], cutbin);
-      plwid(lwid);
+      plwidth(lwid);
       plcol0(15);
-      plwid(lwid);
+      plwidth(lwid);
       xl[0] = fgroup->projposmin[d2] - margin;
       xl[1] = fgroup->projposmax[d2];
       yl[0] = yl[1] = 0.0;
@@ -2494,7 +2494,7 @@
     }
 
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv(-maxlim, maxlim, -maxlim, maxlim, 1, -2);
 
 /* Use a non-linear shade level distribution */
@@ -2540,13 +2540,13 @@
   pllsty(1);
 /* 1D histograms */
   plcol0(3);
-  plwid(2*lwid);
+  plwidth(2*lwid);
   plline(CPLOT_NADERRHISTBIN, cutbin, cutx);
   plline(CPLOT_NADERRHISTBIN, cuty, cutbin);
   plcol0(7);
   plline(CPLOT_NADERRHISTBIN, cutbin, cutx_hsn);
   plline(CPLOT_NADERRHISTBIN, cuty_hsn, cutbin);
-  plwid(lwid);
+  plwidth(lwid);
   plcol0(15);
   plbox("bcnst", 0.0, 0.0, "bcnst", 0.0, 0.0);
   sprintf(str, "Group ##%d: 2D internal astrometric errors", fgroup->no);
@@ -2767,7 +2767,7 @@
         cuty_hsn[d3][i] = fgroup->projposmin[d2] - margin
 			+ cuty_hsn[d3][i]/cutymax_hsn[d3] * 0.9*margin;
         }
-      plwid(lwid);
+      plwidth(lwid);
       plenv(fgroup->projposmin[d2] - margin, fgroup->projposmax[d2],
 		-maxlim, maxlim, 0, -2);
 /*---- Use a non-linear shade level distribution */
@@ -2806,13 +2806,13 @@
       sprintf(ylabel, "#gDAXIS%d [\"]", d+1);
 /*---- 1D histograms */
       plcol0(1);
-      plwid(2*lwid);
+      plwidth(2*lwid);
       plline(CPLOT_NREFERRHISTBIN, cuty[d3], cutbin);
       plcol0(7);
       plline(CPLOT_NREFERRHISTBIN, cuty_hsn[d3], cutbin);
-      plwid(lwid);
+      plwidth(lwid);
       plcol0(15);
-      plwid(lwid);
+      plwidth(lwid);
       xl[0] = fgroup->projposmin[d2] - margin;
       xl[1] = fgroup->projposmax[d2];
       yl[0] = yl[1] = 0.0;
@@ -3014,7 +3014,7 @@
     }
 
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv(-maxlim, maxlim, -maxlim, maxlim, 1, -2);
 
 /* Use a non-linear shade level distribution */
@@ -3058,13 +3058,13 @@
   pllsty(1);
 /* 1D histograms */
   plcol0(1);
-  plwid(2*lwid);
+  plwidth(2*lwid);
   plline(CPLOT_NREFERRHISTBIN, cutbin, cutx);
   plline(CPLOT_NREFERRHISTBIN, cuty, cutbin);
   plcol0(7);
   plline(CPLOT_NREFERRHISTBIN, cutbin, cutx_hsn);
   plline(CPLOT_NREFERRHISTBIN, cuty_hsn, cutbin);
-  plwid(lwid);
+  plwidth(lwid);
   plcol0(15);
   plbox("bcnst", 0.0, 0.0, "bcnst", 0.0, 0.0);
   sprintf(str, "Group ##%d: 2D reference astrometric errors", fgroup->no);
@@ -3367,7 +3367,7 @@
         cuty_hsn[d3][i] = 0.5 - margin
 			+ cuty_hsn[d3][i]/cutymax_hsn[d3] * 0.9*margin;
         }
-      plwid(lwid);
+      plwidth(lwid);
       plenv(-0.5-margin, maxwidth, -maxlim, maxlim, 0, -2);
 /* Use a non-linear shade level distribution */
       if (zmax[d3]>=1.0)
@@ -3406,7 +3406,7 @@
       plscol0(15, 0,0,0);	/* Force the foreground colour to black */
 /* 1D histograms */
       plcol0(3);
-      plwid(2*lwid);
+      plwidth(2*lwid);
       plline(CPLOT_NPIXERRHISTBIN, cuty[d3], cutbin);
       plcol0(7);
       plline(CPLOT_NPIXERRHISTBIN, cuty_hsn[d3], cutbin);
@@ -3416,16 +3416,16 @@
         for (i=0; i<CPLOT_PIXERR1DNX+1; i++)
           if (weight[d3][i]>0.0)
             line[d3][i] /= weight[d3][i];
-        plwid(6*lwid);
+        plwidth(6*lwid);
         plcol0(15);
         plline(CPLOT_PIXERR1DNX+1, cutx, line[d3]);
-        plwid(3*lwid);
+        plwidth(3*lwid);
         plcol0(3);
         plline(CPLOT_PIXERR1DNX+1, cutx, line[d3]);
         }
-      plwid(lwid);
+      plwidth(lwid);
       plcol0(15);
-      plwid(lwid);
+      plwidth(lwid);
       xl[0] = 0.5 - margin;
       xl[1] = maxwidth;
       yl[0] = yl[1] = 0.0;
@@ -3717,7 +3717,7 @@
         cuty_hsn[d3][i] = -0.5 - margin
 			+ cuty_hsn[d3][i]/cutymax_hsn[d3] * 0.9*margin;
         }
-      plwid(lwid);
+      plwidth(lwid);
       plenv(-0.5-margin, 0.5, -maxlim, maxlim, 0, -2);
 /* Use a non-linear shade level distribution */
       if (zmax[d3]>=1.0)
@@ -3757,7 +3757,7 @@
       plscol0(15, 0,0,0);	/* Force the foreground colour to black */
 /* 1D histograms */
       plcol0(3);
-      plwid(2*lwid);
+      plwidth(2*lwid);
       plline(CPLOT_NSUBPIXERRHISTBIN, cuty[d3], cutbin);
       plcol0(7);
       plline(CPLOT_NSUBPIXERRHISTBIN, cuty_hsn[d3], cutbin);
@@ -3768,16 +3768,16 @@
           if (weight[d3][i]>0.0)
             line[d3][i] /= weight[d3][i];
         line[d3][CPLOT_SUBPIXERR1DNX] = line[d3][0];
-        plwid(6*lwid);
+        plwidth(6*lwid);
         plcol0(15);
         plline(CPLOT_SUBPIXERR1DNX+1, cutx, line[d3]);
-        plwid(3*lwid);
+        plwidth(3*lwid);
         plcol0(3);
         plline(CPLOT_SUBPIXERR1DNX+1, cutx, line[d3]);
         }
-      plwid(lwid);
+      plwidth(lwid);
       plcol0(15);
-      plwid(lwid);
+      plwidth(lwid);
       xl[0] = -0.5 - margin;
       xl[1] = 0.5;
       yl[0] = yl[1] = 0.0;
@@ -4060,7 +4060,7 @@
           cuty_hsn[d][i] = dmagmin - margin
 			+ cuty_hsn[d][i]/cutymax_hsn[d] * 0.9*margin;
           }
-        plwid(lwid);
+        plwidth(lwid);
         gra = instru2 + (instru1*naxis + d)*ninstru;
         if (gra)
           pladv(gra);
@@ -4100,13 +4100,13 @@
         plscol0(15, 0,0,0);	/* Force the foreground colour to black */
 /*------ 1D histograms */
         plcol0(13);
-        plwid(2*lwid);
+        plwidth(2*lwid);
         plline(CPLOT_NADERRHISTBIN, cuty[d], cutbin);
         plcol0(7);
         plline(CPLOT_NADERRHISTBIN, cuty_hsn[d], cutbin);
-        plwid(lwid);
+        plwidth(lwid);
         plcol0(15);
-        plwid(lwid);
+        plwidth(lwid);
         xl[0] = dmagmin;
         xl[1] = dmagmax;
         ymin = ymax = 0.0;
@@ -4305,7 +4305,7 @@
   plschr(0.0,0.5);
   for (d=0; d<2; d++)
     {
-    plwid(lwid);
+    plwidth(lwid);
     pladv(1);
     plvpas(0.07+d*0.5,0.48+d*0.5,0.1,0.9,1.0);
     plwind(-maxlim,maxlim,-maxlim,maxlim);
@@ -4520,7 +4520,7 @@
   yl[0] = yl[1] = 0.0;
   plcol0(15);
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   pladv(0);
   plvpor(0.0, 1.0, 0.0, 0.95);
   plwind(-0.75, 0.75, -0.6, 1.1);
@@ -4554,13 +4554,13 @@
     plfill3(5, focplanet->x, focplanet->y, focplanet->z);
     if (focplanet->colour!=15)
       {
-      plwid(2*lwid);
+      plwidth(2*lwid);
       plcol0(focplanet->colour);
       }
     else
       plcol0(15);
     plline3(5, focplanet->x, focplanet->y, focplanet->z); 
-    plwid(lwid);
+    plwidth(lwid);
 /*-- Find the point with highest projpos[lng] and lowest projpos[lat] */
     if (focplanet->str)
       {
@@ -4753,7 +4753,7 @@
     }
 
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plenv(-maxlim, maxlim, -maxlim, maxlim, 1, -2);
 
 /* Use a non-linear shade level distribution */
@@ -4799,13 +4799,13 @@
   pllsty(1);
 /* 1D histograms */
   plcol0(13);
-  plwid(2*lwid);
+  plwidth(2*lwid);
   plline(CPLOT_NADERRHISTBIN, cutbin, cutx);
   plline(CPLOT_NADERRHISTBIN, cuty, cutbin);
   plcol0(7);
   plline(CPLOT_NADERRHISTBIN, cutbin, cutx_hsn);
   plline(CPLOT_NADERRHISTBIN, cuty_hsn, cutbin);
-  plwid(lwid);
+  plwidth(lwid);
   plcol0(15);
   plbox("bcnst", 0.0, 0.0, "bcnst", 0.0, 0.0);
   sprintf(str, "Group ##%d: 2D proper motions", fgroup->no);
@@ -4945,14 +4945,14 @@
   ymin = 0.5;
   ymax = 100.5;
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plfont(2);
   plcol0(15);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
   sprintf(str, "#uInstrument A%d: pixel coordinate residuals (x)",
 	field->astromlabel+1);
   pllab("","", str);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   cplot_drawloccoordgrid(wcs, xmin, xmax, ymin, ymax);
 
@@ -5063,7 +5063,7 @@
              ystep/2.0+0.5, wcs0->naxisn[1]-ystep/2.0+0.5,
 	     clevel, CPLOT_NSHADES, 1, 0, 0, plfill, 0, distort_map, &distort);
     plcol0(7);
-    plwid(lwid);
+    plwidth(lwid);
     cplot_drawbounds(wcs0, wcs);
     }
 
@@ -5210,14 +5210,14 @@
   ymin = 0.5;
   ymax = 100.5;
   lwid = plotaaflag? ((CPLOT_AAFAC+1)/2) : 1;
-  plwid(lwid);
+  plwidth(lwid);
   plfont(2);
   plcol0(15);
   plenv((PLFLT)xmin, (PLFLT)xmax, (PLFLT)ymin, (PLFLT)ymax, 1, -1);
   sprintf(str, "#uInstrument A%d: pixel coordinate residuals (y)",
 	field->astromlabel+1);
   pllab("","", str);
-  plwid(0);
+  plwidth(0);
   plcol0(7);
   cplot_drawloccoordgrid(wcs, xmin, xmax, ymin, ymax);
 
@@ -5328,7 +5328,7 @@
              ystep/2.0+0.5, wcs0->naxisn[1]-ystep/2.0+0.5,
 	     clevel, CPLOT_NSHADES, 1, 0, 0, plfill, 0, distort_map, &distort);
     plcol0(7);
-    plwid(lwid);
+    plwidth(lwid);
     cplot_drawbounds(wcs0, wcs);
     }
 
