--- Makefile.orig	2010-06-14 18:04:03.000000000 +0900
+++ Makefile	2011-06-10 23:22:40.000000000 +0900
@@ -5,7 +5,7 @@
 SHVER = 1
 #LIBS = -lblas
 
-all: train predict
+all: train predict liblinear.a liblinear.dylib
 
 lib: linear.o tron.o blas/blas.a
 	$(CXX) -shared -dynamiclib linear.o tron.o blas/blas.a -o liblinear.so.$(SHVER)
@@ -25,7 +25,16 @@
 blas/blas.a:
 	cd blas; make OPTFLAGS='$(CFLAGS)' CC='$(CC)';
 
+liblinear.a: linear.o tron.o blas/blas.a
+	ar rc liblinear.a linear.o tron.o blas/blas.a
+	ranlib liblinear.a
+liblinear.dylib: linear.o tron.o blas/blas.a
+	$(CXX) $(CFLAGS) -dynamiclib -install_name $(PREFIX)/lib/liblinear.dylib \
+                         -current_version $(VERSION) \
+                         -compatibility_version $(VERSION) \
+                         -o liblinear.$(VERSION).dylib linear.o tron.o blas/blas.a
 clean:
 	cd blas;	make clean
 	cd matlab;	make clean
 	rm -f *~ tron.o linear.o train predict liblinear.so.$(SHVER)
+	rm -f *.a *.dylib
