--- makefile.orig	2017-05-01 12:43:14.000000000 +0000
+++ makefile	2019-07-28 03:00:00.000000000 +0000
@@ -820,10 +820,10 @@
 #
 # The following symbols are used to specify library and header file locations:
 # 
-prefix  = /usr/local
+prefix  = @@prefix@@
 srproot = $(prefix)
 sslroot = $(prefix)
-manroot = $(prefix)
+manroot = $(prefix)/share
 
 K4LIB=-L/usr/kerberos/lib
 K4INC=-I/usr/kerberos/include
@@ -974,7 +974,7 @@
 MANDIR = $(manroot)/man/man1
 MANEXT = 1
 SRCDIR =
-INFODIR =
+INFODIR = $(prefix)/share/doc/kermit
 CERTDIR =
 
 TEXTFILES = COPYING.TXT ckcbwr.txt ckubwr.txt ckuins.txt ckccfg.txt \
@@ -1076,7 +1076,7 @@
 	cp $(BINARY) $(DESTDIR)$(BINDIR)/kermit || exit 1;\
 	chmod 755    $(DESTDIR)$(BINDIR)/kermit || exit 1;\
 	rm -f        $(DESTDIR)$(BINDIR)/kermit-sshsub;\
-	ln -s        $(DESTDIR)$(BINDIR)/kermit\
+	ln -s        kermit\
 		     $(DESTDIR)$(BINDIR)/kermit-sshsub || exit 1;\
 	echo 'set flag=f\nPrC Removing binaries' >&3;\
 	echo "RmF $(DESTDIR)$(BINDIR)/kermit-sshsub" >&3;\
