--- makefile.orig	2017-05-01 12:43:14.000000000 +0000
+++ makefile	2019-07-28 03:00:00.000000000 +0000
@@ -842,7 +842,7 @@
 prefix  = $${PREFIX:-/usr/local}
 srproot = $(prefix)
 sslroot = $(prefix)
-manroot = $(prefix)
+manroot = $(prefix)/share
 
 # The default Kerberos settings seem to be based on AIX, Irix, and Solaris
 # but are not appropriate for (e.g.) Linux.
@@ -996,7 +996,7 @@
 MANDIR = $(manroot)/man/man1
 MANEXT = 1
 SRCDIR =
-INFODIR =
+INFODIR = $(prefix)/share/doc/kermit
 CERTDIR =
 
 TEXTFILES = COPYING.TXT ckcbwr.txt ckubwr.txt ckuins.txt ckccfg.txt \
@@ -1116,7 +1116,7 @@
 	cp $(BINARY) $(DESTDIR)$(BINDIR)/kermit || exit 1;\
 	chmod 755    $(DESTDIR)$(BINDIR)/kermit || exit 1;\
 	rm -f        $(DESTDIR)$(BINDIR)/kermit-sshsub;\
-	ln -s        $(DESTDIR)$(BINDIR)/kermit\
+	ln -s        kermit\
 		     $(DESTDIR)$(BINDIR)/kermit-sshsub || exit 1;\
 	echo 'set flag=f\nPrC Removing binaries' >&3;\
 	echo "RmF $(DESTDIR)$(BINDIR)/kermit-sshsub" >&3;\
