--- src/Csdp/lib/Makefile.orig	2022-12-13 23:17:49.000000000 +0700
+++ src/Csdp/lib/Makefile	2023-03-18 22:18:36.000000000 +0700
@@ -105,6 +105,7 @@
 #
 libsdp.a: readprob.o sdp.o op_o.o psd_feas.o op_a.o op_at.o  Fnorm.o calc_pobj.o calc_dobj.o trace_prod.o zero_mat.o mat_mult.o sym_mat.o copy_mat.o addscaledmat.o  user_exit.o make_i.o allocmat.o initsoln.o initparams.o add_mat.o writesol.o readsol.o easysdp.o writeprob.o solvesys.o makefill.o mat_multsp.o norms.o linesearch.o matvec.o chol.o  qreig.o tweakgap.o freeprob.o packed.o  sortentries.o
 	${AR} cr libsdp.a  readprob.o sdp.o op_o.o psd_feas.o op_a.o op_at.o  Fnorm.o calc_pobj.o calc_dobj.o trace_prod.o zero_mat.o mat_mult.o sym_mat.o copy_mat.o addscaledmat.o  user_exit.o make_i.o allocmat.o initsoln.o initparams.o add_mat.o writesol.o readsol.o easysdp.o writeprob.o solvesys.o makefill.o mat_multsp.o norms.o linesearch.o matvec.o chol.o qreig.o tweakgap.o freeprob.o packed.o sortentries.o
+	${RANLIB} libsdp.a
 #
 # On some systems, you might need to add after "ar cr libsdp.a ..."
 #        ranlib libsdp.a
@@ -115,8 +116,3 @@
 clean:
 	rm -f *.o
 	rm -f libsdp.a
-
-
-
-
-
