--- setup.py.orig	2022-03-12 23:11:26.000000000 -0500
+++ setup.py	2022-03-12 23:12:55.000000000 -0500
@@ -58,10 +58,10 @@
     if not os.path.exists(target_dir):
         os.makedirs(target_dir)
 
-    CC = "gcc"
+    CC = os.environ["CC"]
 
     DEFAULT_CFLAGS = (
-        "-lm -pthread -Ofast -Wall -march=native -funroll-loops -Wno-unused-result"
+        "-lm -pthread -O2 -Wall -funroll-loops -Wno-unused-result"
     )
 
     if sys.platform == "darwin":
