--- ../src/mac/carbon/morefilex/MoreFilesX.cpp	2009-07-25 20:34:52.000000000 +0300
+++ MoreFilesX.cpp	2009-07-25 20:32:02.000000000 +0300
@@ -6,7 +6,7 @@
 
 	Version:	MoreFilesX 1.0.1
 
-	Copyright:	© 1992-2002 by Apple Computer, Inc., all rights reserved.
+	Copyright:	Â© 1992-2002 by Apple Computer, Inc., all rights reserved.
 
 	Disclaimer:	IMPORTANT:  This Apple software is supplied to you by Apple Computer, Inc.
 				("Apple") in consideration of your agreement to the following terms, and your
@@ -15,7 +15,7 @@
 				please do not use, install, modify or redistribute this Apple software.
 
 				In consideration of your agreement to abide by the following terms, and subject
-				to these terms, Apple grants you a personal, non-exclusive license, under AppleÕs
+				to these terms, Apple grants you a personal, non-exclusive license, under AppleÃ•s
 				copyrights in this original Apple software (the "Apple Software"), to use,
 				reproduce, modify and redistribute the Apple Software, with or without
 				modifications, in source and/or binary forms; provided that if you redistribute
@@ -2304,7 +2304,7 @@
 }
 
 /*****************************************************************************/
-
+/*
 OSErr
 FSVolumeMount(
 	const void *volMountInfo,
@@ -2312,23 +2312,23 @@
 {
 	OSErr			result;
 	ParamBlockRec	pb;
-
+*/
 	/* check parameters */
-	require_action(NULL != volRefNum, BadParameter, result = paramErr);
+	/*	require_action(NULL != volRefNum, BadParameter, result = paramErr);
 	
 	pb.ioParam.ioBuffer = (Ptr)volMountInfo;
 	result = PBVolumeMount(&pb);
 	require_noerr(result, PBVolumeMount);
-	
+	*/	
 	/* return the volume reference number */
-	*volRefNum = pb.ioParam.ioVRefNum;
+	/*	*volRefNum = pb.ioParam.ioVRefNum;
 
 PBVolumeMount:
 BadParameter:
 
 	return ( result );
 }
-
+	*/
 /*****************************************************************************/
 
 OSErr
