--- ../src/mac/carbon/morefilex/MoreFilesX.h	2009-07-25 20:34:52.000000000 +0300
+++ MoreFilesX.h	2009-07-25 20:36:09.000000000 +0300
@@ -1479,14 +1479,14 @@
 */
 
 /*****************************************************************************/
-
+/*
 #pragma mark FSVolumeMount
 
 OSErr
 FSVolumeMount(
-	const void *volMountInfo,
+	UInt8 *volMountInfo,
 	FSVolumeRefNum *volRefNum);
-
+*/
 /*
 	The VolumeMount function mounts a volume using a volume mounting
 	information record.
