--- modules/avcodec/encode.c.orig	2016-02-29 16:17:00.000000000 -0800
+++ modules/avcodec/encode.c	2016-02-29 16:17:26.000000000 -0800
@@ -171,7 +171,7 @@
 	st->ctx->width     = size->w;
 	st->ctx->height    = size->h;
 	st->ctx->gop_size  = DEFAULT_GOP_SIZE;
-	st->ctx->pix_fmt   = PIX_FMT_YUV420P;
+	st->ctx->pix_fmt   = AV_PIX_FMT_YUV420P;
 	st->ctx->time_base.num = 1;
 	st->ctx->time_base.den = prm->fps;
 
