From 62b0988520ebc24b101249c593443abe9db6711a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 19 Jun 2024 15:13:22 +0800
Subject: [PATCH] file_utils.cpp: add a missing <cstdint> include

---
 lib/file_utils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git lib/file_utils.cpp lib/file_utils.cpp
index 375e9c7d..75dd3d4f 100644
--- lib/file_utils.cpp
+++ lib/file_utils.cpp
@@ -21,6 +21,7 @@
 #include <cstring>
 #include <sstream>
 #include <string>
+#include <cstdint>
 
 #include <minizinc/_thirdparty/b64/decode.h>
 #include <minizinc/_thirdparty/b64/encode.h>
