--- libwindow-settings/Makefile.am.orig	2012-04-22 02:55:05.000000000 -0700
+++ libwindow-settings/Makefile.am	2012-04-22 02:55:24.000000000 -0700
@@ -17,7 +17,7 @@ libgnome_window_settings_la_LDFLAGS = 	\
 	-no-undefined			\
         -version-info 1:0:0
 
-libgnome_window_settings_la_LIBADD = $(GNOMECC_LIBS)
+libgnome_window_settings_la_LIBADD = $(GNOMECC_LIBS) -lgmodule-2.0
 
 libgnome_window_settings_la_SOURCES = \
 	gnome-window-manager.c	\
