--- configure.ac.orig	2013-09-13 05:09:33.000000000 -0700
+++ configure.ac	2013-09-16 09:28:36.000000000 -0700
@@ -146,10 +146,10 @@
 PKG_CHECK_MODULES(ONLINE_ACCOUNTS_PANEL, $COMMON_MODULES goa-1.0 goa-backend-1.0 >= $GOA_REQUIRED_VERSION)
 PKG_CHECK_MODULES(POWER_PANEL, $COMMON_MODULES upower-glib >= 0.9.1
                   gnome-settings-daemon >= $GSD_REQUIRED_VERSION)
-PKG_CHECK_MODULES(COLOR_PANEL, $COMMON_MODULES
-                  colord >= $COLORD_REQUIRED_VERSION
-                  colord-gtk >= $COLORD_GTK_REQUIRED_VERSION
-                  gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
+dnl PKG_CHECK_MODULES(COLOR_PANEL, $COMMON_MODULES
+dnl                   colord >= $COLORD_REQUIRED_VERSION
+dnl                   colord-gtk >= $COLORD_GTK_REQUIRED_VERSION
+dnl                   gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
 PKG_CHECK_MODULES(PRINTERS_PANEL, $COMMON_MODULES
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION)
 PKG_CHECK_MODULES(PRIVACY_PANEL, $COMMON_MODULES)
--- panels/Makefile.am.orig	2013-01-25 06:42:18.000000000 -0800
+++ panels/Makefile.am		2013-09-16 09:32:32.000000000 -0700
@@ -2,7 +2,6 @@
 	common \
 	background \
 	power \
-	color \
 	display \
 	mouse \
 	notifications \
--- shell/Makefile.am.orig	2013-09-16 17:47:34.000000000 -0700
+++ shell/Makefile.am	2013-09-16 17:48:10.000000000 -0700
@@ -51,7 +51,6 @@
 	$(top_builddir)/libgd/libgd.la					\
 	$(top_builddir)/panels/common/liblanguage.la			\
 	$(top_builddir)/panels/background/libbackground.la		\
-	$(top_builddir)/panels/color/libcolor.la			\
 	$(top_builddir)/panels/datetime/libdate_time.la			\
 	$(top_builddir)/panels/display/libdisplay.la			\
 	$(top_builddir)/panels/info/libinfo.la				\
--- shell/cc-panel-loader.c.orig	2013-09-16 17:57:33.000000000 -0700
+++ shell/cc-panel-loader.c	2013-09-16 17:58:28.000000000 -0700
@@ -77,7 +77,9 @@
 #ifdef BUILD_BLUETOOTH
   PANEL_TYPE("bluetooth",        cc_bluetooth_panel_get_type    ),
 #endif
+#ifdef BUILD_COLOR
   PANEL_TYPE("color",            cc_color_panel_get_type        ),
+#endif
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type    ),
   PANEL_TYPE("display",          cc_display_panel_get_type      ),
   PANEL_TYPE("info",             cc_info_panel_get_type         ),
