https://github.com/assimp/assimp/issues/4334
--- code/Common/ZipArchiveIOSystem.cpp.orig	2022-03-17 05:35:44.000000000 -0500
+++ code/Common/ZipArchiveIOSystem.cpp	2022-04-18 05:10:56.000000000 -0500
@@ -196,7 +196,9 @@
     zlib_filefunc_def mapping;
 
     mapping.zopen_file = (open_file_func)open;
+#ifdef ZOPENDISK64
     mapping.zopendisk_file = (opendisk_file_func)opendisk;
+#endif
     mapping.zread_file = (read_file_func)read;
     mapping.zwrite_file = (write_file_func)write;
     mapping.ztell_file = (tell_file_func)tell;
