--- setup.py.orig	2008-12-25 10:39:57.000000000 -0700
+++ setup.py	2009-02-09 20:51:24.000000000 -0700
@@ -61,7 +61,7 @@
             fn = os.path.join(documentation_path, fn)
             if os.path.isfile(fn):
                 documentation_files.append(fn)
-    data_files.append(('docs', documentation_files))
+    data_files.append(('share/doc/@@NAME@@', documentation_files))
 
 
 def get_terminal_width():
