From 4eb592815d4222c78fab6132ce8f4f9a5e24396b Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 4 Mar 2012 17:38:13 -0800
Subject: [PATCH] Fix potential error during make install

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 lib/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index bd2ccfc..c857434 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -35,7 +35,7 @@ $(LIBRARY_BASENAME) : $(DIST_LIB)
 install-exec-hook : $(LIBRARY_BASENAME)
 	install -d -m 755 $(DESTDIR)$(libdir)
 	install -m 755 $(LIBRARY_BASENAME) $(DESTDIR)$(libdir)
-	rm $(DESTDIR)$(libdir)/$(LIBRARY_SYMLINK)
+	rm -f $(DESTDIR)$(libdir)/$(LIBRARY_SYMLINK)
 	ln -s $(LIBRARY_BASENAME) $(DESTDIR)$(libdir)/$(LIBRARY_SYMLINK)
 
 CLEANFILES = $(LIBRARY_SYMLINK) $(LIBRARY_BASENAME)
-- 
1.7.9.2

