--- /dev/null	2009-10-11 21:18:38.000000000 +0200
+++ configure.ac	2009-10-11 21:08:45.000000000 +0200
@@ -0,0 +1,5 @@
+AC_INIT([poster], [20060221])
+AM_INIT_AUTOMAKE
+AC_PROG_CC
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
--- /dev/null	2009-10-11 21:18:38.000000000 +0200
+++ Makefile.am	2009-10-11 21:17:54.000000000 +0200
@@ -0,0 +1,4 @@
+bin_PROGRAMS = poster
+poster_SOURCES = poster.c
+man_MANS = poster.1
+EXTRA_DIST = $(man_MANS) 
