--- ../exonerate-2.2.0.orig/src/program/Makefile.in	2008-10-17 11:04:45.000000000 +0100
+++ ./src/program/Makefile.in	2012-02-24 11:53:57.000000000 +0000
@@ -361,9 +361,9 @@
 mostlyclean distclean maintainer-clean
 
 c4_model_archive.h: $(BOOTSTRAPPER)
-	$(BOOTSTRAPPER) --compiled no
+	CFLAGS="$(CFLAGS)" $(BOOTSTRAPPER) --compiled no
 c4_model_archive.a: $(BOOTSTRAPPER)
-	$(BOOTSTRAPPER) --compiled no
+	CFLAGS="$(CFLAGS)" $(BOOTSTRAPPER) --compiled no
 
 # Special viterbi.o and scheduler.o for compiled model linking
 viterbi.o: $(top_srcdir)/src/c4/viterbi.c
