--- src/Makefile.in.orig	2011-03-21 18:40:31.000000000 +0000
+++ src/Makefile.in	2011-03-21 18:53:08.000000000 +0000
@@ -200,7 +200,7 @@
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
 
-@IS_WIN32_FALSE@AM_CFLAGS = -export-dynamic -Wall
+@IS_WIN32_FALSE@AM_CFLAGS = -Wall
 @IS_WIN32_TRUE@AM_CFLAGS = -export-all-symbols -Wall
 @IS_WIN32_FALSE@klavaro_LDADD = @PACKAGE_LIBS@
 @IS_WIN32_TRUE@klavaro_LDADD = @PACKAGE_LIBS@ -lcurldll
