--- configure.orig	2012-07-07 08:36:20.000000000 -0700
+++ configure	2013-03-26 14:09:09.000000000 -0700
@@ -18089,8 +18089,8 @@
 if test "x$ac_cv_lib_m_pow" = xyes; then :
   GUTENPRINT_LIBDEPS="${GUTENPRINT_LIBDEPS} -lm"
              gutenprint_libdeps="${gutenprint_libdeps} -lm"
-             GUTENPRINTUI2_LIBDEPS="${GUTENPRINT_LIBDEPS} -lm"
-             gutenprintui2_libdeps="${gutenprint_libdeps} -lm"
+             GUTENPRINTUI2_LIBDEPS="${GUTENPRINTUI2_LIBDEPS} -lm"
+             gutenprintui2_libdeps="${gutenprintui2_libdeps} -lm"
 	     LIBM=-lm
 
 fi
