--- src/cf.data.pre	Tue Feb 10 22:01:21 2004
+++ src/cf.data.pre.new	Thu May 13 18:15:22 2004
@@ -2152,10 +2152,10 @@
 # We strongly recommend to uncomment the following to protect innocent
 # web applications running on the proxy server who think that the only
 # one who can access services on "localhost" is a local user
-#http_access deny to_localhost
+http_access deny to_localhost
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
-
+http_access allow localhost
 # Example rule allowing access from your local networks. Adapt
 # to list your (internal) IP networks from where browsing should
 # be allowed
