--- src/cf.data.pre	Fri Oct  8 19:41:10 2004
+++ src/cf.data.pre.new	Fri Dec  3 13:50:20 2004
@@ -2181,10 +2181,10 @@
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
 # one who can access services on "localhost" is a local user
-#http_access deny to_localhost
+http_access deny to_localhost
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
-
+http_access allow localhost
 # Example rule allowing access from your local networks. Adapt
 # to list your (internal) IP networks from where browsing should
 # be allowed
