--- src/ssl_support.c.orig	2010-03-15 04:20:45.000000000 +1100
+++ src/ssl_support.c	2016-04-07 02:04:53.000000000 +1000
@@ -446,10 +446,6 @@ sslCreateServerContext(const char *certf
     ERR_clear_error();
     debug(83, 1) ("Initialising SSL.\n");
     switch (version) {
-    case 2:
-	debug(83, 5) ("Using SSLv2.\n");
-	method = SSLv2_server_method();
-	break;
     case 3:
 	debug(83, 5) ("Using SSLv3.\n");
 	method = SSLv3_server_method();
@@ -459,6 +455,7 @@ sslCreateServerContext(const char *certf
 	method = TLSv1_server_method();
 	break;
     case 1:
+    case 2:
     default:
 	debug(83, 5) ("Using SSLv2/SSLv3.\n");
 	method = SSLv23_server_method();
@@ -609,10 +606,6 @@ sslCreateClientContext(const char *certf
     ERR_clear_error();
     debug(83, 1) ("Initialising SSL.\n");
     switch (version) {
-    case 2:
-	debug(83, 5) ("Using SSLv2.\n");
-	method = SSLv2_client_method();
-	break;
     case 3:
 	debug(83, 5) ("Using SSLv3.\n");
 	method = SSLv3_client_method();
@@ -622,6 +615,7 @@ sslCreateClientContext(const char *certf
 	method = TLSv1_client_method();
 	break;
     case 1:
+    case 2:
     default:
 	debug(83, 5) ("Using SSLv2/SSLv3.\n");
 	method = SSLv23_client_method();
