--- t/local/http.t.orig	2019-07-26 13:41:12.000000000 -0700
+++ t/local/http.t	2019-08-03 20:51:08.000000000 -0700
@@ -1,6 +1,6 @@
 use Test::Needs 'LWP::UserAgent';
 
-if ($^O eq "MacOS") {
+if ($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
--- t/robot/ua-get.t.orig	2019-07-26 13:41:12.000000000 -0700
+++ t/robot/ua-get.t	2019-08-03 20:55:38.000000000 -0700
@@ -1,6 +1,6 @@
 use Test::Needs 'LWP::RobotUA';
 
-if ($^O eq "MacOS") {
+if ($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
--- t/robot/ua.t.orig	2019-07-26 13:41:12.000000000 -0700
+++ t/robot/ua.t	2019-08-03 21:01:03.000000000 -0700
@@ -1,6 +1,6 @@
 use Test::Needs 'LWP::RobotUA';
 
-if ($^O eq "MacOS") {
+if ($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
