diff -ur t.orig/local/http.t t/local/http.t
--- t.orig/local/http.t	2019-04-01 16:51:16.000000000 -0700
+++ t/local/http.t	2019-04-01 16:56:50.000000000 -0700
@@ -2,7 +2,7 @@
 
 use Test::Needs 'LWP::UserAgent';
 
-if ($^O eq "MacOS") {
+if ($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
diff -ur t.orig/robot/ua-get.t t/robot/ua-get.t
--- t.orig/robot/ua-get.t	2019-04-01 16:51:16.000000000 -0700
+++ t/robot/ua-get.t	2019-04-01 16:55:43.000000000 -0700
@@ -1,4 +1,4 @@
-if($^O eq "MacOS") {
+if($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
diff -ur t.orig/robot/ua.t t/robot/ua.t
--- t.orig/robot/ua.t	2019-04-01 16:51:16.000000000 -0700
+++ t/robot/ua.t	2019-04-01 16:56:23.000000000 -0700
@@ -1,4 +1,4 @@
-if($^O eq "MacOS") {
+if($^O eq "darwin") {
     print "1..0\n";
     exit(0);
 }
