From 3ee1d3ebb81de199fc630a86933ac18c0a869482 Mon Sep 17 00:00:00 2001
From: Patrick Walton <pcwalton@mimiga.net>
Date: Sat, 22 Dec 2012 16:24:19 -0500
Subject: [PATCH] rustllvm: Fix symbol resolution on Mac for rusti. rs=bugfix

---
 src/rustllvm/RustWrapper.cpp |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/rustllvm/RustWrapper.cpp b/src/rustllvm/RustWrapper.cpp
index 86a302e..bb00f04 100644
--- a/src/rustllvm/RustWrapper.cpp
+++ b/src/rustllvm/RustWrapper.cpp
@@ -290,6 +290,13 @@ void *RustMCJITMemoryManager::getPointerToNamedFunction(const std::string &Name,
   void *Ptr = sys::DynamicLibrary::SearchForAddressOfSymbol(NameStr);
   if (Ptr) return Ptr;
 
+  // If it wasn't found and if it starts with an underscore ('_') character,
+  // try again without the underscore.
+  if (NameStr[0] == '_') {
+    Ptr = sys::DynamicLibrary::SearchForAddressOfSymbol(NameStr+1);
+    if (Ptr) return Ptr;
+  }
+
   if (AbortOnFailure)
     report_fatal_error("Program used external function '" + Name +
                       "' which could not be resolved!");
-- 
1.7.10

