--- src/llvm-project/llvm/include/llvm/TargetParser/Triple.h.orig	2023-05-31 12:29:26
+++ src/llvm-project/llvm/include/llvm/TargetParser/Triple.h	2023-06-02 18:55:27
@@ -960,7 +960,7 @@
 
   /// Tests whether the target uses emulated TLS as default.
   bool hasDefaultEmulatedTLS() const {
-    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment();
+    return isAndroid() || isOSOpenBSD() || isWindowsCygwinEnvironment() || isMacOSXVersionLT(10, 7);
   }
 
   /// Tests whether the target uses -data-sections as default.
