--- src/bootstrap/src/lib.rs	2025-09-19 20:08:27
+++ src/bootstrap/src/lib.rs	2025-09-19 20:10:37
@@ -1841,6 +1841,11 @@
                 metadata = t!(fs::metadata(&src), format!("target = {}", src.display()));
             } else {
                 let link = t!(fs::read_link(src));
+                if dst.is_dir() {
+                    t!(fs::remove_dir_all(&dst));
+                } else if dst.exists() {
+                    t!(fs::remove_file(dst));
+                }
                 t!(self.symlink_file(link, dst));
                 return;
             }
