--- src/llvm-project/llvm/include/llvm/TargetParser/Triple.h.orig	2023-12-21 09:56:09
+++ src/llvm-project/llvm/include/llvm/TargetParser/Triple.h	2024-01-04 01:32:33
@@ -1029,8 +1029,8 @@
   ///
   /// Note: Android API level 29 (10) introduced ELF TLS.
   bool hasDefaultEmulatedTLS() const {
-    return (isAndroid() && isAndroidVersionLT(29)) || isOSOpenBSD() ||
-           isWindowsCygwinEnvironment() || isOHOSFamily();
+    return
+           true;
   }
 
   /// True if the target supports both general-dynamic and TLSDESC, and TLSDESC
--- compiler/rustc_target/src/spec/base/apple/mod.rs.orig	2023-12-21 09:55:28
+++ compiler/rustc_target/src/spec/base/apple/mod.rs	2024-01-04 01:39:44
@@ -161,7 +161,7 @@
         archive_format: "darwin".into(),
         // Thread locals became available with iOS 8 and macOS 10.7,
         // and both are far below our minimum.
-        has_thread_local: true,
+        has_thread_local: false,
         abi_return_struct_as_int: true,
         emit_debug_gdb_scripts: false,
         eh_frame_header: false,
