--- src/bootstrap/bootstrap.py	2022-02-24 14:11:08.000000000 -0500
+++ src/bootstrap/bootstrap.py	2022-02-24 14:11:49.000000000 -0500
@@ -774,6 +774,7 @@
             args.append("--locked")
         if self.use_vendored_sources:
             args.append("--frozen")
+        args.append("--offline")
         if self.get_toml("metrics", "build"):
             args.append("--features")
             args.append("build-metrics")
