See https://github.com/llvm/llvm-project/issues/54846

--- src/llvm-project/llvm/tools/llvm-objcopy/MachO/MachOWriter.cpp.orig	2022-03-09 01:50:54.000000000 -0700
+++ src/llvm-project/llvm/tools/llvm-objcopy/MachO/MachOWriter.cpp	2022-04-10 11:32:24.000000000 -0700
@@ -567,7 +567,7 @@
   while (CurrHashReadPosition < HashReadEnd) {
     StringRef Block(reinterpret_cast<char *>(CurrHashReadPosition),
                     std::min(HashReadEnd - CurrHashReadPosition,
-                             static_cast<ssize_t>(CodeSignature.BlockSize)));
+                             static_cast<std::ptrdiff_t>(CodeSignature.BlockSize)));
     SHA256 Hasher;
     Hasher.update(Block);
     StringRef Hash = Hasher.final();
