--- ./Makefile	2013-09-03 04:20:45.000000000 -0500
+++ /Users/m028373/src/macports/trunk/science/hdf5-lz4-plugin/Makefile	2015-01-02 16:15:07.000000000 -0600
@@ -1,7 +1,7 @@
-CFLAGS = -fPIC -g -O3 -std=c99
+CFLAGS += -fPIC -std=c99
-CC = gcc
+CC ?= gcc
 # install directory of HDF5 1.8.11 #
-HDF5_INSTALL = /home/det/hdf5-1.8.11/hdf5/
+HDF5_INSTALL = %%PREFIX%%
 MAJOR = 0
 MINOR = 2
 NAME1 = h5zlz4
@@ -10,7 +10,7 @@
 
 INCLUDES = -I./ -I$(HDF5_INSTALL)/include 
 
-lib: lib$(NAME1).so.$(VERSION)
+lib: lib$(NAME1).dylib.$(VERSION)
 
 $(NAME1).o: $(NAME1).c
 	$(CC) $(CFLAGS) $(INCLUDES) -c $(NAME1).c 
@@ -19,13 +19,15 @@
 	$(CC) $(CFLAGS) $(INCLUDES) -c lz4.c
 
 
-lib$(NAME1).so.$(VERSION): $(NAME1).o lz4.o
-	#$(CC) -shared -Wl,soname,lib$(NAME1).so.$(MAJOR) $^ -o $@ 
-	$(CC) -shared  $^ -o $@ 
-	ln -sf lib$(NAME1).so.$(VERSION) lib$(NAME1).so
-	ln -sf lib$(NAME1).so.$(VERSION) lib$(NAME1).so.$(MAJOR)
+lib$(NAME1).dylib.$(VERSION): $(NAME1).o 
+	$(CC) $^ -o $@ \
+	      -dynamiclib \
+	      -Wl,-install_name,%%PREFIX%%/lib/hdf5/$@ \
+	      -L%%PREFIX%%/lib/ -llz4
+	ln -sf lib$(NAME1).dylib.$(VERSION) lib$(NAME1).dylib
+	ln -sf lib$(NAME1).dylib.$(VERSION) lib$(NAME1).dylib.$(MAJOR)
 clean:
-	$(RM) *.o *.so*
+	$(RM) *.o *.dylib*
 
 distclean:		
-	$(RM) *.o *.so*
\ No newline at end of file
+	$(RM) *.o *.dylib*
