--- DESCRIPTION	2024-05-03 06:49:42.000000000 +0800
+++ DESCRIPTION	2024-05-04 21:03:38.000000000 +0800
@@ -26,8 +26,8 @@
 Imports: Biobase, BiocBaseUtils, BiocGenerics, DelayedArray,
         GenomicRanges (>= 1.25.93), IRanges, methods, S4Vectors (>=
         0.23.19), tidyr, utils
-Suggests: BiocStyle, HDF5Array (>= 1.19.17), knitr, maftools (>=
-        2.7.10), R.rsp, RaggedExperiment, reshape2, rmarkdown,
+Suggests: BiocStyle, HDF5Array (>= 1.19.17), knitr, R.rsp,
+        RaggedExperiment, reshape2, rmarkdown,
         survival, survminer, testthat, UpSetR
 VignetteBuilder: knitr, R.rsp
 biocViews: Infrastructure, DataRepresentation
