From bcdc65f7c2aad91d7838189070323cbcff429496 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Mon, 13 Nov 2023 08:54:49 +0800
Subject: [PATCH 2/2] CMakeLists.txt: add ppc/ppc64 to known archs

---
 src/tapi/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/tapi/CMakeLists.txt src/tapi/CMakeLists.txt
index d08557213..59b740893 100644
--- src/tapi/CMakeLists.txt
+++ src/tapi/CMakeLists.txt
@@ -42,7 +42,7 @@ elseif(NOT DEFINED TAPI_SUPPORTED_ARCHS)
 endif()
 message(STATUS "Supported Architectures: ${TAPI_SUPPORTED_ARCHS}")
 
-set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 arm64e)
+set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 arm64e ppc ppc64)
 
 set (CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/include/tapi/Core/ArchitectureConfig.h)
 file(WRITE ${CONFIG_FILE} "#ifndef TAPI_CORE_ARCHITECTURE_CONFIG_H\n")
