--- src/tovid.orig	2015-06-03 09:33:01
+++ src/tovid	2026-05-20 22:47:48
@@ -68,7 +68,7 @@
 class Frontend:
     def __init__(self, args):
         # Set attributes
-        self.path = os.path.abspath(args.pop(0))
+        self.path = os.path.realpath(args.pop(0))
         self.prefix = self.get_prefix(self.path)
         self.script_dir = os.path.join(self.prefix, 'lib', 'tovid')
         self.version = self.get_version()
