--- docs/scripts/txt2tags.orig	2026-05-20 20:44:06
+++ docs/scripts/txt2tags	2026-05-20 20:46:59
@@ -2047,10 +2047,10 @@
 	bank['linkmark'] = re.compile(
 		r'\[(?P<label>%s|[^]]+) (?P<link>%s|%s|%s)\]'%(
 			patt_img, retxt_url, patt_email, retxt_url_local),
-		re.L+re.I)
+		re.I)
 	
 	# Image
-	bank['img'] = re.compile(patt_img, re.L+re.I)
+	bank['img'] = re.compile(patt_img, re.I)
 	
 	# Special things
 	bank['special'] = re.compile(r'^%!\s*')
