From 7194c874a9674ce9b5d6a715a24ef048195a29f9 Mon Sep 17 00:00:00 2001
From: Adam Mercer <adam.mercer@ligo.org>
Date: Tue, 25 Mar 2014 20:01:11 +0000
Subject: don't use -mfpmath=387 for pylal build

---
diff --git a/pylal/setup.py b/pylal/setup.py
index 3422f9a..8620e70 100644
--- a/pylal/setup.py
+++ b/pylal/setup.py
@@ -457,7 +457,7 @@ setup(
 			library_dirs = gsl_pkg_config.libdirs + lalsimulation_pkg_config.libdirs + lal_pkg_config.libdirs,
 			runtime_library_dirs = gsl_pkg_config.libdirs + lalsimulation_pkg_config.libdirs + lal_pkg_config.libdirs,
 			libraries = gsl_pkg_config.libs + lalsimulation_pkg_config.libs + lal_pkg_config.libs,
-			extra_compile_args = ["-std=c99", "-ffast-math", "-mfpmath=387", "-O3", "-Wno-unknown-pragmas"] + gsl_pkg_config.extra_cflags + lalsimulation_pkg_config.extra_cflags + lal_pkg_config.extra_cflags,  # , "-fopenmp"
+			extra_compile_args = ["-std=c99", "-ffast-math", "-O3", "-Wno-unknown-pragmas"] + gsl_pkg_config.extra_cflags + lalsimulation_pkg_config.extra_cflags + lal_pkg_config.extra_cflags,  # , "-fopenmp"
 			#extra_link_args=['-fopenmp']
 		),
 		Extension(
--
cgit v0.9
