--- src/gpu.cc.orig	2007-12-16 15:05:56.000000000 -0800
+++ src/gpu.cc	2007-12-16 16:23:18.000000000 -0800
@@ -22,12 +22,15 @@
 
 #include "gpu.h"
 
+//OSX Hack
+#include <dlfcn.h>
+
 using std::cerr;
 using std::cout;
 using std::endl;
 
 static GLuint GlutWindowHandle;
-static int MaxTextureSize;
+static GLint MaxTextureSize;
 static GLuint PiTexture;
 static GLuint ETexture;
 static GLuint OutTexture;
@@ -71,8 +74,8 @@
 }
 
 void printInfoLog(GLhandleARB obj) {
-    int infologLength = 0;
-    int charsWritten = 0;
+    GLint infologLength = 0;
+    GLsizei charsWritten = 0;
     char *infoLog;
     glGetObjectParameterivARB(obj, GL_OBJECT_INFO_LOG_LENGTH_ARB, &infologLength);
     if (infologLength > 1) {
@@ -116,6 +119,14 @@
 }
 
 bool initGPU() {
+    // OSX hack
+    // http://www.ogre3d.org/wiki/index.php/MacLibrary
+    void* cocoa_lib; 
+    cocoa_lib = dlopen( "/System/Library/Frameworks/Cocoa.framework/Cocoa", RTLD_LAZY ); 
+    void (*nsappload)(void); 
+    nsappload = (void(*)()) dlsym( cocoa_lib, "NSApplicationLoad"); 
+    nsappload(); 
+
     glutInitDisplayMode(GLUT_RGBA | GLUT_ALPHA);
     GlutWindowHandle = glutCreateWindow("Enblend");
 
