--- mcrypt.c.orig
+++ mcrypt.c
@@ -38,7 +38,7 @@
 #include "php_ini.h"
 #include "php_globals.h"
 #include "ext/standard/info.h"
-#include "ext/standard/php_rand.h"
+#include "ext/random/php_random.h"
 #include "zend_smart_str.h"
 #include "php_mcrypt_filter.h"

@@ -1414,7 +1414,7 @@
	} else {
		n = (int)size;
		while (size) {
-			iv[--size] = (char) (255.0 * php_rand() / RAND_MAX);
+			iv[--size] = (char) (255.0 * php_mt_rand() / RAND_MAX);
		}
	}
	RETVAL_STRINGL(iv, n);
