--- libraries/libldap/tls2.c.orig	2020-04-28 07:05:54.000000000 -0700
+++ libraries/libldap/tls2.c	2020-06-10 13:16:50.000000000 -0700
@@ -74,6 +74,9 @@ static oid_name oids[] = {
 	{ BER_BVNULL, BER_BVNULL }
 };
 
+static int
+ldap_pvt_tls_check_hostname( LDAP *ld, void *s, const char *name_in );
+
 #ifdef HAVE_TLS
 
 void
@@ -503,7 +506,7 @@ ldap_pvt_tls_get_peer_dn( void *s, struc
 	return rc;
 }
 
-int
+static int
 ldap_pvt_tls_check_hostname( LDAP *ld, void *s, const char *name_in )
 {
 	tls_session *session = s;
