diff -Naurp stardict-3.0.4.orig/dict/src/dictmanagedlg.cpp stardict-3.0.4/dict/src/dictmanagedlg.cpp
--- dict/src/dictmanagedlg.cpp	2012-12-24 02:04:24.000000000 -0800
+++ dict/src/dictmanagedlg.cpp	2015-10-12 23:36:06.000000000 -0700
@@ -26,6 +26,8 @@
 #include <glib/gi18n.h>
 #include <algorithm>
 
+#include <stdlib.h>
+
 #ifdef _WIN32
 #  include <gdk/gdkwin32.h>
 #endif
diff -Naurp stardict-3.0.4.orig/dict/src/lib/dict_client.cpp stardict-3.0.4/dict/src/lib/dict_client.cpp
--- dict/src/lib/dict_client.cpp	2012-01-12 22:21:32.000000000 -0800
+++ dict/src/lib/dict_client.cpp	2015-10-12 23:36:10.000000000 -0700
@@ -27,6 +27,7 @@
 #endif
 
 #include <glib.h>
+#include <stdlib.h>
 
 #include "sockets.h"
 
diff -Naurp stardict-3.0.4.orig/dict/src/lib/stddict.cpp stardict-3.0.4/dict/src/lib/stddict.cpp
--- dict/src/lib/stddict.cpp	2012-02-17 00:58:47.000000000 -0800
+++ dict/src/lib/stddict.cpp	2015-10-12 23:36:20.000000000 -0700
@@ -34,6 +34,7 @@
 #include <glib/gstdio.h>
 #include <algorithm>
 #include <memory>
+#include <stdlib.h>
 
 #include "ifo_file.h"
 #include "edit-distance.h"
diff -Naurp stardict-3.0.4.orig/dict/src/mainwin.cpp stardict-3.0.4/dict/src/mainwin.cpp
--- dict/src/mainwin.cpp	2012-12-24 02:50:21.000000000 -0800
+++ dict/src/mainwin.cpp	2015-10-12 23:36:23.000000000 -0700
@@ -23,6 +23,7 @@
 #include <glib/gi18n.h>
 #include <gdk/gdkkeysyms.h>
 #include <algorithm>
+#include <stdlib.h>
 
 #ifdef _WIN32
 #define VERSION "3.0.4"
