--- src/Makevars	2024-05-31 02:47:55.000000000 +0800
+++ src/Makevars	2024-06-01 18:04:16.000000000 +0800
@@ -1,7 +1,7 @@
-PKG_CPPFLAGS = -I"../inst/include/" -I"../inst/include/quickjs"
+PKG_CPPFLAGS = -I"../inst/include/" -I"../inst/include/quickjs" -I@PREFIX@/include/LegacySupport
 PKG_CPPFLAGS += -D_GNU_SOURCE -DCONFIG_BIGNUM
 PKG_CPPFLAGS += -DCONFIG_VERSION=\"$(shell cat quickjs/VERSION)\"
-PKG_LIBS = ../inst/lib/$(R_ARCH)/libquickjs.a
+PKG_LIBS = ../inst/lib/$(R_ARCH)/libquickjs.a -L@PREFIX@/lib -lMacportsLegacySupport
 
 ifeq ($(OS),Windows_NT)
   DLL := .dll
