--- src/Makevars	2024-05-31 02:47:55.000000000 +0800
+++ src/Makevars	2024-06-01 18:08:54.000000000 +0800
@@ -31,9 +31,8 @@
 		PKG_CPPFLAGS += -DDISABLE_ATOMICS
 	else
 # Explicit -latomic is needed for 32-bit builds only on GCC
-		ifeq ($(shell getconf LONG_BIT), 32)
-			PKG_LIBS += -latomic
-		endif
+# However we still need -latomic when building for 32-bit on 64-bit with GCC
+		PKG_LIBS += -latomic
 	endif
 endif
 
