--- src/Makevars	2024-01-22 13:31:51.000000000 +0800
+++ src/Makevars	2024-01-22 13:31:41.000000000 +0800
@@ -1,6 +1,6 @@
-PKG_CPPFLAGS = -I"../inst/include/" -DSTRICT_R_HEADERS -D_GNU_SOURCE -DCONFIG_BIGNUM 
+PKG_CPPFLAGS = -I"../inst/include/" -I@PREFIX@/include/LegacySupport -DSTRICT_R_HEADERS -D_GNU_SOURCE -DCONFIG_BIGNUM 
 PKG_CPPFLAGS += -DCONFIG_VERSION=\"$(shell cat quickjs/VERSION)\"
-PKG_LIBS = ../inst/lib/$(R_ARCH)/libquickjs.a
+PKG_LIBS = ../inst/lib/$(R_ARCH)/libquickjs.a -L@PREFIX@/lib -lMacportsLegacySupport
 
 ifeq ($(shell getconf LONG_BIT), 32)
 	PKG_LIBS += -latomic
