Index: src/main.cpp
===================================================================
--- src/main.cpp	(revision 391)
+++ src/main.cpp	(revision 392)
@@ -71,6 +71,9 @@
 	app.setOrganizationDomain("schmufu.dyndns.org");
 	app.setApplicationName("AB-Transfers");
 
+	//On MacOS the QSharedMemory seems not to work as assumed.
+	//Therefore, as a fast workaround, we do not use the protection on MacOS.
+#if !defined(Q_OS_MAC)
 	//The program should be started only once!
 	//Therefore we create a sharedMemory and check if it already exists.
 	//As key for the sharedMemory the program name and the "Key-ID" from
@@ -122,6 +125,7 @@
 				   << "Error:" << myMem.errorString();
 		}
 	}
+#endif // !defined(Q_OS_MAC)
 
 
 	//creation of global objects
