--- ./pgmodeler.pri	2023-10-17 15:23:08
+++ ./pgmodeler.pri	2023-10-17 15:26:09
@@ -231,10 +231,10 @@
 }
 
 macx {
-  !defined(PGSQL_LIB, var): PGSQL_LIB = /Library/PostgreSQL/14/lib/libpq.dylib
-  !defined(PGSQL_INC, var): PGSQL_INC = /Library/PostgreSQL/14/include
-  !defined(XML_INC, var): XML_INC = /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libxml2
-  !defined(XML_LIB, var): XML_LIB = /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libxml2.tbd
+  !defined(PGSQL_LIB, var): PGSQL_LIB = @@PREFIX@@/lib/postgresql15/libpq.dylib
+  !defined(PGSQL_INC, var): PGSQL_INC = @@PREFIX@@/include/postgresql15
+  !defined(XML_INC, var): XML_INC = @@PREFIX@@/include/libxml2
+  !defined(XML_LIB, var): XML_LIB = @@PREFIX@@/lib/libxml2.dylib
   INCLUDEPATH += "$$PGSQL_INC" "$$XML_INC"
 }
 
