--- cmake/OpenCVFindOpenEXR.cmake.orig	2017-12-16 18:52:33.000000000 -0700
+++ cmake/OpenCVFindOpenEXR.cmake	2017-12-16 18:53:30.000000000 -0700
@@ -21,24 +21,16 @@
         SET(OPENEXR_LIBSEARCH_SUFFIXES Win32/Release Win32 Win32/Debug)
     endif()
 else()
-    set(OPENEXR_ROOT "")
+    set(OPENEXR_ROOT "@PREFIX@")
 endif()
 
 SET(LIBRARY_PATHS
-    /usr/lib
-    /usr/local/lib
-    /sw/lib
-    /opt/local/lib
     "${ProgramFiles_ENV_PATH}/OpenEXR/lib/static"
     "${OPENEXR_ROOT}/lib")
 
 FIND_PATH(OPENEXR_INCLUDE_PATH ImfRgbaFile.h
     PATH_SUFFIXES OpenEXR
     PATHS
-    /usr/include
-    /usr/local/include
-    /sw/include
-    /opt/local/include
     "${ProgramFiles_ENV_PATH}/OpenEXR/include"
     "${OPENEXR_ROOT}/include")
 
