--- CMakeLists.txt.orig	2011-08-17 14:42:55.000000000 -0500
+++ CMakeLists.txt	2011-10-29 01:10:32.000000000 -0500
@@ -664,11 +664,7 @@
                     RESULT_VARIABLE PYTHON_CVPY_PROCESS
                     OUTPUT_VARIABLE PYTHON_STD_PACKAGES_PATH
                     OUTPUT_STRIP_TRAILING_WHITESPACE)
-    if("${PYTHON_STD_PACKAGES_PATH}" MATCHES "site-packages")
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages CACHE PATH "Where to install the python packages.")
-    else() #debian based assumed, install to the dist-packages.
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/dist-packages CACHE PATH "Where to install the python packages.")
-    endif()
+    set(PYTHON_PACKAGES_PATH @@PYTHON_PKGD@@ CACHE PATH "Where to install the python packages.")
 endif()
 if(CMAKE_HOST_WIN32)
     get_filename_component(PYTHON_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\${PYTHON_VERSION_MAJOR_MINOR}\\InstallPath]" ABSOLUTE CACHE)
