--- modules/highgui/src/window_cocoa.mm.orig	2016-12-23 05:54:44.000000000 -0700
+++ modules/highgui/src/window_cocoa.mm	2017-01-20 16:07:47.000000000 -0700
@@ -83,11 +83,26 @@
 static bool wasInitialized = false;
 
 @interface CVView : NSView
+#ifndef __LP64__
+{
+NSImage *image;
+}
+#endif
 @property(strong) NSImage *image;
 - (void)setImageData:(CvArr *)arr;
 @end
 
 @interface CVSlider : NSView
+#ifndef __LP64__
+{
+NSSlider *slider;
+NSTextField *name;
+int *value;
+void *userData;
+CvTrackbarCallback callback;
+CvTrackbarCallback2 callback2;
+}
+#endif
 @property(strong) NSSlider *slider;
 @property(strong) NSTextField *name;
 @property(assign) int *value;
@@ -97,6 +112,16 @@
 @end
 
 @interface CVWindow : NSWindow
+#ifndef __LP64__
+{
+CvMouseCallback mouseCallback;
+void *mouseParam;
+BOOL autosize;
+BOOL firstContent;
+NSMutableDictionary *sliders;
+int status;
+}
+#endif
 @property(assign) CvMouseCallback mouseCallback;
 @property(assign) void *mouseParam;
 @property(assign) BOOL autosize;
