diff --git CMakeLists.txt CMakeLists.txt
index 3ee50ff..357a700 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -162,7 +162,7 @@ endif()
 # Optional 3rd party components
 # ===================================================
 OCV_OPTION(WITH_1394           "Include IEEE1394 support"                    ON   IF (NOT ANDROID AND NOT IOS AND NOT WINRT) )
-OCV_OPTION(WITH_AVFOUNDATION   "Use AVFoundation for Video I/O"              ON   IF IOS)
+OCV_OPTION(WITH_AVFOUNDATION   "Use AVFoundation for Video I/O"              ON   IF (APPLE OR IOS))
 OCV_OPTION(WITH_CARBON         "Use Carbon for UI instead of Cocoa"          OFF  IF APPLE )
 OCV_OPTION(WITH_VTK            "Include VTK library support (and build opencv_viz module eiher)"             ON  IF (NOT ANDROID AND NOT IOS AND NOT WINRT AND NOT CMAKE_CROSSCOMPILING) )
 OCV_OPTION(WITH_CUDA           "Include NVidia Cuda Runtime support"                                         ON  IF (NOT IOS AND NOT WINRT) )
diff --git cmake/OpenCVFindLibsVideo.cmake cmake/OpenCVFindLibsVideo.cmake
index 2e09cf2..978d9de 100644
--- cmake/OpenCVFindLibsVideo.cmake
+++ cmake/OpenCVFindLibsVideo.cmake
@@ -316,7 +316,7 @@ if (NOT IOS)
   if(WITH_QUICKTIME)
     set(HAVE_QUICKTIME YES)
   elseif(APPLE AND CMAKE_COMPILER_IS_CLANGCXX)
-    set(HAVE_QTKIT YES)
+    set(HAVE_QTKIT NO)
   endif()
 endif()
 
diff --git modules/videoio/CMakeLists.txt modules/videoio/CMakeLists.txt
index b5da8d0..e4ca82b 100644
--- modules/videoio/CMakeLists.txt
+++ modules/videoio/CMakeLists.txt
@@ -167,7 +167,7 @@ endif(HAVE_GIGE_API)
 
 if(HAVE_AVFOUNDATION)
   list(APPEND videoio_srcs ${CMAKE_CURRENT_LIST_DIR}/src/cap_avfoundation.mm)
-  list(APPEND VIDEOIO_LIBRARIES "-framework AVFoundation" "-framework QuartzCore")
+  list(APPEND VIDEOIO_LIBRARIES "-framework AVFoundation" "-framework QuartzCore" "-framework CoreMedia" "-framework CoreFoundation" "-framework AppKit")
 endif()
 
 if(HAVE_QUICKTIME)
