--- modules/python/CMakeLists.txt.orig	2015-04-27 19:27:04.000000000 +0300
+++ modules/python/CMakeLists.txt	2015-04-27 19:26:31.000000000 +0300
@@ -2,6 +2,9 @@
 #  CMake file for python support
 # ----------------------------------------------------------------------------
 
+ocv_module_disable(python2)
+ocv_module_disable(python3)
+
 if((WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
     OR BUILD_opencv_world
     )
