--- OpenCVModule.cmake.orig	2011-08-02 07:48:41.000000000 -0500
+++ OpenCVModule.cmake	2011-11-02 02:35:15.000000000 -0500
@@ -83,7 +83,6 @@
         DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
         ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
         RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-        INSTALL_NAME_DIR lib
         )
 
     add_opencv_precompiled_headers(${the_target})
--- modules/gpu/CMakeLists.txt.orig	2011-08-11 16:17:14.000000000 -0500
+++ modules/gpu/CMakeLists.txt	2011-11-02 02:35:42.000000000 -0500
@@ -113,7 +113,6 @@
     DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
     ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR lib
     )
 
 # Add the required libraries for linking:
--- modules/haartraining/CMakeLists.txt.orig	2011-06-14 05:37:20.000000000 -0500
+++ modules/haartraining/CMakeLists.txt	2011-11-02 02:35:38.000000000 -0500
@@ -42,7 +42,6 @@
     DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
     ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR lib
     )
 
 if(NOT ANDROID)
--- modules/highgui/CMakeLists.txt.orig	2011-08-11 16:17:08.000000000 -0500
+++ modules/highgui/CMakeLists.txt	2011-11-02 02:35:34.000000000 -0500
@@ -294,7 +294,6 @@
     DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
     ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR lib
     LINK_INTERFACE_LIBRARIES ""
     )
 
--- modules/stitching/CMakeLists.txt.orig	2011-06-14 05:37:15.000000000 -0500
+++ modules/stitching/CMakeLists.txt	2011-11-02 02:35:25.000000000 -0500
@@ -28,7 +28,6 @@
                       DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
                       ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
                       RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-                      INSTALL_NAME_DIR lib
                       OUTPUT_NAME "opencv_stitching")
 					  
 if(ENABLE_SOLUTION_FOLDERS)
--- modules/traincascade/CMakeLists.txt.orig	2011-08-02 07:48:38.000000000 -0500
+++ modules/traincascade/CMakeLists.txt	2011-11-02 02:35:19.000000000 -0500
@@ -32,7 +32,6 @@
                       DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
                       ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
                       RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-                      INSTALL_NAME_DIR lib
                       OUTPUT_NAME "opencv_traincascade")
 					  
 if(ENABLE_SOLUTION_FOLDERS)
