--- cmake/OpenCVFindLibsPerf.cmake.orig	2013-03-20 12:39:40.000000000 +0400
+++ cmake/OpenCVFindLibsPerf.cmake	2013-03-20 13:08:15.000000000 +0400
@@ -56,8 +56,13 @@
 endif()
 
 # --- GCD ---
-if(APPLE AND NOT HAVE_TBB AND NOT HAVE_CSTRIPES AND NOT HAVE_OPENMP)
-  set(HAVE_GCD 1)
+if(WITH_GCD AND APPLE AND NOT HAVE_TBB AND NOT HAVE_CSTRIPES AND NOT HAVE_OPENMP)
+  find_path(GCD_INCLUDE_PATH dispatch/dispatch.h)
+  if(GCD_INCLUDE_PATH)
+    set(HAVE_GCD 1)
+  else()
+    set(HAVE_GCD 0)
+  endif()
 else()
   set(HAVE_GCD 0)
 endif()
