--- modules/python/test/test.py.orig	2012-05-13 10:33:44.000000000 +0200
+++ modules/python/test/test.py	2012-05-13 10:35:24.000000000 +0200
@@ -1821,7 +1821,7 @@
 
     def local_test_Haar(self):
         import os
-        hcfile = os.environ['OPENCV_ROOT'] + '/share/opencv/haarcascades/haarcascade_frontalface_default.xml'
+        hcfile = os.environ['OPENCV_ROOT'] + '/share/OpenCV/haarcascades/haarcascade_frontalface_default.xml'
         hc = cv.Load(hcfile)
         img = cv.LoadImage('Stu.jpg', 0)
         faces = cv.HaarDetectObjects(img, hc, cv.CreateMemStorage())
