--- macros2/gnome-autogen.sh.orig	2014-03-24 05:16:46.000000000 -0700
+++ macros2/gnome-autogen.sh	2014-03-24 22:43:38.000000000 -0700
@@ -110,7 +110,7 @@
     if [ "$vc_status" != 0 ]; then
 	printerr "***Error***: You must have $vc_package >= $vc_min_version installed"
 	printerr "  to build $PKG_NAME.  Download the appropriate package for"
-	printerr "  from your distribution or get the source tarball at"
+	printerr "  from your distribution.  For example, you can install it with"
         printerr "    $vc_source"
 	printerr
 	exit $vc_status
