--- macros2/gnome-autogen.sh.orig	2009-05-27 14:29:07.000000000 -0700
+++ macros2/gnome-autogen.sh	2010-08-08 14:25:56.000000000 -0700
@@ -112,7 +112,7 @@
     if [ "$vc_status" != 0 ]; then
 	printerr "***Error***: You must have $vc_package $vc_comparator $vc_min_version installed"
 	printerr "  to build $PKG_NAME.  Download the appropriate package for"
-	printerr "  from your distribution or get the source tarball at"
+	printerr "  from your distribution.  For example, you can install it with"
         printerr "    $vc_source"
 	printerr
 	exit $vc_status
@@ -329,44 +329,44 @@
 ACLOCAL=`echo $AUTOMAKE | sed s/automake/aclocal/`
 
 if $want_libtool; then
-    version_check libtool LIBTOOLIZE libtoolize $REQUIRED_LIBTOOL_VERSION \
-        "http://ftp.gnu.org/pub/gnu/libtool/libtool-$REQUIRED_LIBTOOL_VERSION.tar.gz"
+    version_check libtool LIBTOOLIZE glibtoolize $REQUIRED_LIBTOOL_VERSION \
+        "sudo port install libtool"
     require_m4macro libtool.m4
 fi
 
 if $want_gettext; then
     version_check gettext GETTEXTIZE gettextize $REQUIRED_GETTEXT_VERSION \
-        "http://ftp.gnu.org/pub/gnu/gettext/gettext-$REQUIRED_GETTEXT_VERSION.tar.gz"
+        "sudo port install gettext"
     require_m4macro gettext.m4
 fi
 
 if $want_glib_gettext; then
     version_check glib-gettext GLIB_GETTEXTIZE glib-gettextize $REQUIRED_GLIB_GETTEXT_VERSION \
-        "ftp://ftp.gtk.org/pub/gtk/v2.2/glib-$REQUIRED_GLIB_GETTEXT_VERSION.tar.gz"
+        "sudo port install glib2"
     require_m4macro glib-gettext.m4
 fi
 
 if $want_intltool; then
     version_check intltool INTLTOOLIZE intltoolize $REQUIRED_INTLTOOL_VERSION \
-        "http://ftp.gnome.org/pub/GNOME/sources/intltool/"
+        "sudo port install intltool"
     require_m4macro intltool.m4
 fi
 
 if $want_pkg_config; then
     version_check pkg-config PKG_CONFIG pkg-config $REQUIRED_PKG_CONFIG_VERSION \
-        "'http://www.freedesktop.org/software/pkgconfig/releases/pkgconfig-$REQUIRED_PKG_CONFIG_VERSION.tar.gz"
+        "sudo port install pkgconfig"
     require_m4macro pkg.m4
 fi
 
 if $want_gtk_doc; then
     version_check gtk-doc GTKDOCIZE gtkdocize $REQUIRED_GTK_DOC_VERSION \
-        "http://ftp.gnome.org/pub/GNOME/sources/gtk-doc/"
+        "sudo port install gtk-doc"
     require_m4macro gtk-doc.m4
 fi
 
 if $want_gnome_doc_utils; then
     version_check gnome-doc-utils GNOME_DOC_PREPARE gnome-doc-prepare $REQUIRED_GNOME_DOC_UTILS_VERSION \
-        "http://ftp.gnome.org/pub/GNOME/sources/gnome-doc-utils/"
+        "sudo port install gnome-doc-utils"
 fi
 
 if [ "x$USE_COMMON_DOC_BUILD" = "xyes" ]; then
