--- macros2/gnome-autogen.sh.orig	2013-09-03 13:32:52.000000000 -0700
+++ macros2/gnome-autogen.sh	2013-11-23 11:58:37.000000000 -0800
@@ -109,7 +109,7 @@
     if [ "$vc_status" != 0 ]; then
 	printerr "***Error***: You must have $vc_package >= $vc_min_version installed"
 	printerr "  to build $PKG_NAME.  Download the appropriate package for"
-	printerr "  from your distribution or get the source tarball at"
+	printerr "  from your distribution.  For example, you can install it with"
         printerr "    $vc_source"
 	printerr
 	exit $vc_status
