From 63aac6889ba53ee03c4dab2e1e45e68f48f3b00a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 14 Feb 2023 17:46:18 +0800
Subject: [PATCH] archive_read.cpp: unbreak build for Big-endian platforms

Signed-off-by: Sergey Fedorov <vital.had@gmail.com>
---
 src/archive_read.cpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/archive_read.cpp b/src/archive_read.cpp
index b4f03ae..c61ffda 100644
--- src/archive_read.cpp
+++ src/archive_read.cpp
@@ -1,5 +1,26 @@
 #include "r_archive.h"
 
+/* Define BSWAP_32 on Big Endian systems */
+#ifdef WORDS_BIGENDIAN
+#if (defined(__sun) && defined(__SVR4))
+#include <sys/byteorder.h>
+#elif (defined(__APPLE__) && defined(__ppc__) || defined(__ppc64__))
+#include <libkern/OSByteOrder.h>
+#define BSWAP_32 OSSwapInt32
+#elif (defined(__OpenBSD__))
+#define BSWAP_32(x) swap32(x)
+#elif (defined(__NetBSD__))
+#include <sys/types.h>
+#include <machine/bswap.h>
+#define BSWAP_32(x) bswap32(x)
+#elif (defined(__GLIBC__))
+#include <byteswap.h>
+#define BSWAP_32(x) bswap_32(x)
+#elif (defined(_AIX))
+#define BSWAP_32(x) __builtin_bswap32(x)
+#endif
+#endif
+
 /* Read archives
  *
  * The custom R connection code was adapted from curl package by Jeroen Ooms
-- 
2.39.1

