--- DESCRIPTION.orig	2023-08-07 20:00:02.000000000 +0800
+++ DESCRIPTION	2023-08-12 03:03:05.000000000 +0800
@@ -28,7 +28,7 @@
 Imports: methods, Rcpp (>= 1.0.8), nanotime, spdl
 LinkingTo: Rcpp
 Suggests: tinytest, simplermarkdown, curl, bit64, Matrix,
-        palmerpenguins, nycflights13, data.table, tibble, arrow
+        nycflights13, data.table, tibble, arrow
 VignetteBuilder: simplermarkdown
 RoxygenNote: 7.2.3
 Encoding: UTF-8
