--- common/aiccu.h.orig	2009-03-29 18:05:03.000000000 +0200
+++ common/aiccu.h	2009-03-29 18:07:00.000000000 +0200
@@ -75,7 +75,8 @@
 /* GetWindowsDirectory() is used to figure out the directory to store the config */
 #define AICCU_CONFIG	"aiccu.conf"
 #else
-#define AICCU_CONFIG	"/etc/aiccu.conf"
+#define as_string(x)	#x
+#define AICCU_CONFIG	MPPREFIX "/etc/aiccu.conf"
 #endif
 
 /* Inbound listen queue */
--- unix-console/Makefile.orig	2009-03-29 18:28:01.000000000 +0200
+++ unix-console/Makefile	2009-03-29 18:29:03.000000000 +0200
@@ -115,7 +115,7 @@
 
 # Darwin
 ifeq ($(shell uname | grep -c "Darwin"),1)
-CFLAGS	+= -D_DARWIN -D NEED_IFHEAD -D AICCU_TYPE="\"darwin\""
+CFLAGS	+= -D_DARWIN -D NEED_IFHEAD -D AICCU_TYPE="\"darwin\"" -D MPPREFIX="\"${MPPREFIX}\""
 SRCS	+= ../common/aiccu_darwin.c
 OBJS	+= ../common/aiccu_darwin.o
 LDFLAGS	+= -lresolv
