--- Makefile.orig	2009-03-29 16:25:59.000000000 +0200
+++ Makefile	2009-03-29 16:28:07.000000000 +0200
@@ -57,6 +57,7 @@
 export dirdoc
 export RPM_OPT_FLAGS
 export CFLAGS
+export MPPREFIX
 
 ####################
 ## Makefile Targets
--- unix-console/Makefile.orig	2009-03-29 16:31:22.000000000 +0200
+++ unix-console/Makefile	2009-03-29 16:31:45.000000000 +0200
@@ -44,6 +44,12 @@
 CFLAGS	+= -D AICCU_GNUTLS
 LDFLAGS += -lgnutls
 endif
+ifeq ($(shell uname | grep -c "Darwin"),1)
+CFLAGS	+= -D AICCU_GNUTLS
+CFLAGS  += -I${MPPREFIX}/include
+LDFLAGS += -lgnutls
+LDFLAGS += -L${MPPREFIX}/lib
+endif
 
 # Linux
 ifeq ($(shell uname | grep -c "Linux"),1)
