diff -upr .depend .depend
--- .depend	2014-10-21 19:07:15.000000000 -0400
+++ .depend	2014-10-21 19:13:27.000000000 -0400
@@ -326,11 +326,9 @@ ubase/trace.cmx: ubase/util.cmx system.c
     ubase/trace.cmi 
 ubase/uarg.cmo: ubase/util.cmi system.cmi ubase/safelist.cmi ubase/uarg.cmi 
 ubase/uarg.cmx: ubase/util.cmx system.cmx ubase/safelist.cmx ubase/uarg.cmi 
-ubase/uprintf.cmo: ubase/uprintf.cmi 
-ubase/uprintf.cmx: ubase/uprintf.cmi 
-ubase/util.cmo: ubase/uprintf.cmi system.cmi ubase/safelist.cmi \
+ubase/util.cmo: system.cmi ubase/safelist.cmi \
     ubase/util.cmi 
-ubase/util.cmx: ubase/uprintf.cmx system.cmx ubase/safelist.cmx \
+ubase/util.cmx: system.cmx ubase/safelist.cmx \
     ubase/util.cmi 
 lwt/lwt.cmi: 
 lwt/lwt_unix.cmi: lwt/lwt.cmi 
@@ -343,7 +341,6 @@ ubase/rx.cmi: 
 ubase/safelist.cmi: 
 ubase/trace.cmi: ubase/prefs.cmi 
 ubase/uarg.cmi: 
-ubase/uprintf.cmi: 
 ubase/util.cmi: system.cmi 
 lwt/example/editor.cmo: lwt/lwt_unix.cmi 
 lwt/example/editor.cmx: lwt/lwt_unix.cmx 
diff -upr Makefile.OCaml Makefile.OCaml
--- Makefile.OCaml	2014-10-21 19:07:15.000000000 -0400
+++ Makefile.OCaml	2014-10-21 19:13:36.000000000 -0400
@@ -211,7 +211,7 @@ OCAMLOBJS += \
           system.cmo \
           \
           ubase/projectInfo.cmo ubase/myMap.cmo ubase/safelist.cmo \
-          ubase/uprintf.cmo ubase/util.cmo ubase/uarg.cmo \
+          ubase/util.cmo ubase/uarg.cmo \
           ubase/prefs.cmo ubase/trace.cmo ubase/proplist.cmo \
           \
           lwt/pqueue.cmo lwt/lwt.cmo lwt/lwt_util.cmo \
diff -upr ubase/Makefile ubase/Makefile
--- ubase/Makefile	2014-10-21 19:07:15.000000000 -0400
+++ ubase/Makefile	2014-10-21 19:13:52.000000000 -0400
@@ -1,7 +1,7 @@
 NAME = ubase
 
 OBJECTS  = \
-  safelist.cmo uprintf.cmo util.cmo uarg.cmo prefs.cmo trace.cmo rx.cmo \
+  safelist.cmo util.cmo uarg.cmo prefs.cmo trace.cmo rx.cmo \
   myMap.cmo
 
 OCAMLC = ocamlfind ocamlc -g 
@@ -54,4 +54,4 @@ clean::
 
 # Used by BCP to update Harmony's copy of these files from Unison's
 update:
-	cp $(HOME)/current/unison/trunk/src/ubase/{*.ml,*.mli,Makefile} .
\ No newline at end of file
+	cp $(HOME)/current/unison/trunk/src/ubase/{*.ml,*.mli,Makefile} .
diff -upr ubase/depend ubase/depend
--- ubase/depend	2014-10-21 19:07:15.000000000 -0400
+++ ubase/depend	2014-10-21 19:13:47.000000000 -0400
@@ -12,10 +12,8 @@ trace.cmo: util.cmi safelist.cmi prefs.c
 trace.cmx: util.cmx safelist.cmx prefs.cmx trace.cmi 
 uarg.cmo: util.cmi safelist.cmi uarg.cmi 
 uarg.cmx: util.cmx safelist.cmx uarg.cmi 
-uprintf.cmo: uprintf.cmi 
-uprintf.cmx: uprintf.cmi 
-util.cmo: uprintf.cmi safelist.cmi util.cmi 
-util.cmx: uprintf.cmx safelist.cmx util.cmi 
+util.cmo: safelist.cmi util.cmi 
+util.cmx: safelist.cmx util.cmi 
 myMap.cmi: 
 prefs.cmi: util.cmi 
 proplist.cmi: 
@@ -23,5 +21,4 @@ rx.cmi: 
 safelist.cmi: 
 trace.cmi: prefs.cmi 
 uarg.cmi: 
-uprintf.cmi: 
 util.cmi: 
Only in unison-2.40.102.orig/ubase: uprintf.ml
Only in unison-2.40.102.orig/ubase: uprintf.mli
diff -upr ubase/util.ml ubase/util.ml
--- ubase/util.ml	2014-10-21 19:07:15.000000000 -0400
+++ ubase/util.ml	2014-10-21 19:10:16.000000000 -0400
@@ -62,7 +62,7 @@ let set_infos s =
   if s <> !infos then begin clear_infos (); infos := s; show_infos () end
 
 let msg f =
-  clear_infos (); Uprintf.eprintf (fun () -> flush stderr; show_infos ()) f
+  clear_infos (); Printf.kfprintf (fun c -> flush c; show_infos ()) stderr f
 
 let msg : ('a, out_channel, unit) format -> 'a = msg
 
