--- Makefile.in	2013-07-15 17:27:28.000000000 -0400
+++ Makefile.in	2017-04-07 22:40:26.000000000 -0400
@@ -832,8 +832,6 @@
 	tar cBf - doc | ( cd $(distdir); tar xBf - ; find doc -name CVS | xargs rm -rf; )
 	tar cBf - jemboss | ( cd $(distdir); tar xBf - ; find jemboss -name CVS | xargs rm -rf; find jemboss -name Makefile | xargs rm -rf; find jemboss -name .cvsignore | xargs rm -rf )
 
-install-exec-hook:
-	$(bindir)/embossupdate
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
