diff -ur ../openbabel-2.3.1-orig/include/openbabel/obmolecformat.h ./include/openbabel/obmolecformat.h
--- ../openbabel-2.3.1-orig/include/openbabel/obmolecformat.h	2010-10-10 04:05:12.000000000 +0200
+++ ./include/openbabel/obmolecformat.h	2014-05-29 12:05:22.000000000 +0200
@@ -23,7 +23,10 @@
   #include <hash_map>
 #endif
 
-#if __GNUC__ == 4 && __GNUC_MINOR__ >= 1
+#include <ciso646>  // detect std::lib 
+#ifdef _LIBCPP_VERSION
+  #include <unordered_map>
+#elif __GNUC__ == 4 && __GNUC_MINOR__ >= 1
   #include <tr1/unordered_map>
 #elif defined(USE_BOOST)
   #include <boost/tr1/unordered_map.hpp>
diff -ur ../openbabel-2.3.1-orig/include/openbabel/shared_ptr.h ./include/openbabel/shared_ptr.h
--- ../openbabel-2.3.1-orig/include/openbabel/shared_ptr.h	2011-10-12 22:24:02.000000000 +0200
+++ ./include/openbabel/shared_ptr.h	2014-05-29 12:00:36.000000000 +0200
@@ -22,9 +22,13 @@
 #else
   #include <memory>
   #if __GNUC__ == 4  //&& __GNUC_MINOR__ < 3  removed at the suggestion of Konstantin Tokarev
-    #include <tr1/memory>
+    #ifdef _LIBCPP_VERSION
+      using std::shared_ptr;
+    #else
+      #include <tr1/memory>
+      using std::tr1::shared_ptr;
+    #endif
   #endif
-  using std::tr1::shared_ptr;
 #endif
 
 #endif // OB_SHARED_PTR_H
