--- CMakeLists.txt	2022-12-20 22:30:27.000000000 +0800
+++ CMakeLists.txt	2024-08-14 11:45:13.000000000 +0800
@@ -166,10 +166,10 @@
 	list(APPEND STRICT_OPTIONS_CPP "-Wall" "-Wuninitialized")
 	list(APPEND STRICT_OPTIONS_C "-Wdeclaration-after-statement" "-Wstrict-prototypes" "-Wno-error=strict-prototypes")
 	if(CMAKE_C_COMPILER_ID MATCHES "Clang")
-		list(APPEND STRICT_OPTIONS_CPP "-Qunused-arguments" "-Wno-array-bounds")
+		list(APPEND STRICT_OPTIONS_CPP "-Wno-error=unknown-warning-option" "-Qunused-arguments" "-Wno-array-bounds")
 	endif()
 	if(APPLE)
-		list(APPEND STRICT_OPTIONS_CPP "-Wno-error=unknown-warning-option" "-Qunused-arguments" "-Wno-tautological-compare" "-Wno-unused-function" "-Wno-array-bounds")
+		list(APPEND STRICT_OPTIONS_CPP "-Wno-tautological-compare" "-Wno-unused-function" "-Wno-array-bounds")
 	endif()
 	if(ENABLE_STRICT)
 		list(APPEND STRICT_OPTIONS_CPP "-Werror" "-Wextra" "-Wno-unused-parameter" "-Wno-missing-field-initializers" "-fno-strict-aliasing")
