--- AIO.xs.orig	2016-01-29 05:55:48.000000000 -0800
+++ AIO.xs	2016-01-29 06:05:11.000000000 -0800
@@ -21,6 +21,18 @@
 # include <sys/mman.h>
 #endif
 
+#ifndef MAP_GROWSDOWN
+#define MAP_GROWSDOWN 0
+#endif
+
+#ifndef MAP_32BIT
+#define MAP_32BIT 0
+#endif
+
+#ifndef MAP_STACK
+#define MAP_STACK 0
+#endif
+
 /* the incompetent fool that created musl keeps __linux__, refuses
  * to implement any linux standard apis, and also has no way to test
  * for his broken iplementation. on't complain if this fails for you.
