--- AIO.xs.orig	2018-02-19 22:26:58.000000000 -0800
+++ AIO.xs	2018-02-20 22:46:24.000000000 -0800
@@ -24,6 +24,18 @@
 # include <sys/mman.h>
 #endif
 
+#ifndef MAP_GROWSDOWN
+#define MAP_GROWSDOWN 0
+#endif
+
+#ifndef MAP_32BIT
+#define MAP_32BIT 0
+#endif
+
+#ifndef MAP_STACK
+#define MAP_STACK 0
+#endif
+
 /* the incompetent fool that created musl keeps __linux__, refuses
  * to implement any linux standard apis, and also has no way to test
  * for his broken iplementation. don't complain to me if this fails
