--- configure.in.orig	2009-04-05 13:02:43.000000000 -0700
+++ configure.in	2015-02-05 06:34:32.000000000 -0800
@@ -20,7 +20,7 @@
 # Initialize GNOME environment
 GNOME_COMMON_INIT
 GNOME_COMPILE_WARNINGS(error)
-DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
+DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 AC_SUBST(DEPRECATED_FLAGS)
 
 PKG_CHECK_MODULES(DEP,
@@ -39,7 +39,7 @@
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+CFLAGS="${CFLAGS} -Wall -std=c99 -Wmissing-prototypes"
 
 
 ######################################################################
