--- configure.orig	2006-03-03 09:02:03.000000000 -0600
+++ configure	2006-03-03 09:03:44.000000000 -0600
@@ -23128,8 +23128,8 @@
 
 if test x"$with_libsndfile" != xno; then
   if test x"$with_libsndfile" != xyes; then
-    sim_ac_libsndfile_cppflags="-I${with_libsndfile}/src"
-    sim_ac_libsndfile_ldflags="-L${with_libsndfile}"
+    sim_ac_libsndfile_cppflags="-I${with_libsndfile}/include"
+    sim_ac_libsndfile_ldflags="-L${with_libsndfile}/lib"
   fi
 
   sim_ac_save_cppflags=$CPPFLAGS
