--- makefile.in.orig	2014-02-17 08:00:02.000000000 -0500
+++ makefile.in	2014-02-17 08:00:31.000000000 -0500
@@ -1116,7 +1116,7 @@
 #
 # gcc -dynamiclib -install_name libliquid.dylib -o libliquid.dylib libmodem.a libutility.a 
 libliquid.dylib: $(objects)
-	$(CC) -dynamiclib -install_name $@ -o $@ $^ $(LDFLAGS)
+	$(CC) -dynamiclib -install_name $(exec_prefix)/lib/$@ -o $@ $^ $(LDFLAGS)
 
 # linux, et al
 libliquid.so: libliquid.a
@@ -1139,11 +1139,10 @@
 
 install:
 	@echo "installing..."
-	mkdir -p $(exec_prefix)/lib
-	install -m 644 -p $(SHARED_LIB) libliquid.a $(exec_prefix)/lib
-	mkdir -p $(prefix)/include
-	mkdir -p $(prefix)/include/liquid
-	install -m 644 -p $(addprefix include/,$(headers_install)) $(prefix)/include/liquid
+	mkdir -p $(DESTDIR)$(exec_prefix)/lib
+	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)$(exec_prefix)/lib
+	mkdir -p $(DESTDIR)$(prefix)/include/liquid
+	install -m 644 -p $(addprefix include/,$(headers_install)) $(DESTDIR)$(prefix)/include/liquid
 	@echo ""
 	@echo "---------------------------------------------------------"
 	@echo "  liquid-dsp was successfully installed.     "
