--- gl/stdint_.h.orig	2015-02-17 06:00:57.000000000 -0800
+++ gl/stdint_.h	2015-06-15 11:47:56.000000000 -0700
@@ -38,11 +38,7 @@
       diagnostics.  */
 #  define __STDINT_H__
 # endif
-  /* Other systems may have an incomplete or buggy <stdint.h>.
-     Include it before <inttypes.h>, since any "#include <stdint.h>"
-     in <inttypes.h> would reinclude us, skipping our contents because
-     _GL_STDINT_H is defined.  */
-# include @ABSOLUTE_STDINT_H@
+# include_next <stdint.h>
 #endif
 
 /* <sys/types.h> defines some of the stdint.h types as well, on glibc,
