--- thirdparty/CMakeLists.txt.orig	2025-07-21 15:06:49
+++ thirdparty/CMakeLists.txt	2025-07-21 15:09:44
@@ -18,13 +18,7 @@ Don't stop building due to missing submodules (save for snappy that is always required, according to docs)
 
 set (SUBMODULES_MISSING FALSE)
 foreach (path IN ITEMS
-    brotli/LICENSE
-    gtest/LICENSE
-    libbacktrace/LICENSE
-    libpng/LICENSE
     snappy/COPYING
-    zlib/README
-    directxmath/LICENSE
 )
     if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${path}")
         message (SEND_ERROR "error: ${CMAKE_CURRENT_SOURCE_DIR}/${path} does not exist")
