diff -urN a/doc/Makefile.in b/doc/Makefile.in
--- a/doc/Makefile.in	2016-05-17 15:50:52.000000000 -0400
+++ b/doc/Makefile.in	2023-11-22 22:57:43.000000000 -0500
@@ -331,16 +331,16 @@
 @ENABLE_GROP_TRUE@	grop.desktop
 
 @ENABLE_GROP_TRUE@grop_appsdir = \
-@ENABLE_GROP_TRUE@	$(DESTDIR)$(datadir)/applications
+@ENABLE_GROP_TRUE@	$(datadir)/applications
 
 @ENABLE_GROP_TRUE@grop_icons_DATA = \
 @ENABLE_GROP_TRUE@	grop.png
 
 @ENABLE_GROP_TRUE@grop_iconsdir = \
-@ENABLE_GROP_TRUE@	$(DESTDIR)$(datadir)/pixmaps
+@ENABLE_GROP_TRUE@	$(datadir)/pixmaps
 
 @ENABLE_GROP_FALSE@porg_icon_path = \
-@ENABLE_GROP_FALSE@	$(DESTDIR)$(pkgdatadir)/porg.png
+@ENABLE_GROP_FALSE@	$(pkgdatadir)/porg.png
 
 @ENABLE_GROP_TRUE@porg_icon_path = \
 @ENABLE_GROP_TRUE@	$(grop_iconsdir)/grop.png
