--- src/makefile.orig	2012-04-18 17:46:29.000000000 -0500
+++ src/makefile	2012-09-12 14:08:50.000000000 -0500
@@ -88,6 +88,8 @@
 NE_DEBUG=
 NE_TEST=
 
+CC=__CC__
+
 ifeq ($(CC),gcc)
 NO_SIGN=-Wno-pointer-sign
 endif
@@ -97,9 +99,10 @@
 	$(if $(NE_POSIX),   -D_POSIX_C_SOURCE=199506L,) \
 	$(if $(NE_NOWCHAR), -DNOWCHAR,) \
 	$(if $(NE_TEST),    -DNE_TEST -coverage,) \
-	$(if $(NE_DEBUG),   -g,-O3 -DNDEBUG) \
+	__CFLAGS__ \
 	$(if $(NE_TERMCAP), -DTERMCAP,) \
 	$(if $(NE_ANSI),    -DTERMCAP -DANSI,)
+LDFLAGS=__LDFLAGS__
 
 
 LIBS=$(if $(NE_TERMCAP)$(NE_ANSI),,-lcurses)
