Install documentation in unversioned directory.
--- Makefile.am.orig	2015-10-12 16:13:52.000000000 -0500
+++ Makefile.am	2018-02-01 13:14:56.000000000 -0600
@@ -11,7 +11,7 @@
 AM_CXXFLAGS = -Wall -W -Wwrite-strings -Woverloaded-virtual -Wshadow
 endif
 
-docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
+docdir = $(prefix)/share/doc/$(PACKAGE)
 ## This is for HTML and other documentation you want to install.
 ## Add your documentation files (in doc/) in addition to these boilerplate
 ## Also add a TODO file if you have one
