From 7f6351fb06241b96fdb39ae3aff53c2acb1cd7a4 Mon Sep 17 00:00:00 2001
From: Donovan Hide <donovanhide@gmail.com>
Date: Wed, 23 Sep 2015 17:46:42 +0100
Subject: [PATCH] use std::swap in hashtable test

---
 src/hashtable_test.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hashtable_test.cc b/src/hashtable_test.cc
index 21c60a7..ede7c65 100644
--- a/src/hashtable_test.cc
+++ b/src/hashtable_test.cc
@@ -901,5 +901,5 @@ TYPED_TEST(HashtableAllTest, Swap) {
 #ifdef _MSC_VER
   other_ht.swap(this->ht_);
 #else
-  swap(this->ht_, other_ht);
+  std::swap(this->ht_, other_ht);
 #endif

   EXPECT_EQ(this->UniqueKey(1), this->ht_.deleted_key());
