--- CMakeLists.txt.orig	2020-06-14 00:08:29.000000000 -0400
+++ CMakeLists.txt	2020-06-22 01:22:28.000000000 -0400
@@ -24,6 +24,8 @@
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
+include(FeatureSummary)
+
 add_definitions(-std=c++11)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
@@ -72,7 +74,7 @@
 find_package(Qt5 REQUIRED COMPONENTS Concurrent Gui LinguistTools Network PrintSupport Test Widgets Xml)
 
 if(NOT FORCE_INTERNAL_QSCINTILLA)
-    find_package(QScintilla 2.8.10 QUIET)
+    find_package(QScintilla 2.8.10)
 endif()
 if(FORCE_INTERNAL_QCUSTOMPLOT)
     set(QCUSTOMPLOT_FOUND FALSE)
@@ -351,7 +353,7 @@
 
 #enable version check for MacOS
 if(APPLE)
-	add_definitions(-DCHECKNEWVERSION)
+	#add_definitions(-DCHECKNEWVERSION)
 endif(APPLE)
 
 # SQLCipher option
@@ -399,7 +401,7 @@
 if(QSCINTILLA_FOUND)
     include_directories(${QSCINTILLA_INCLUDE_DIR})
 else()
-    include_directories(${QSCINTILLA_DIR})
+    #include_directories(${QSCINTILLA_DIR})
 endif()
 
 add_executable(${PROJECT_NAME}
@@ -566,3 +568,5 @@
 			distri/winlaunch.bat
 		DESTINATION "/")
 endif()
+
+feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
