--- src/Makefile.in	2021-03-27 16:38:19.000000000 +0900
+++ src/Makefile.in	2021-04-02 16:50:00.000000000 +0900
@@ -346,7 +346,7 @@
 
 ## ARM Macs require that all code have a valid signature.  Since pump
 ## invalidates the signature, we must re-sign to fix it.
-DO_CODESIGN=$(patsubst arm-apple-darwin%,yes,@configuration@)
+DO_CODESIGN=$(patsubst aarch64-apple-darwin%,yes,@configuration@)
 
 # 'make' verbosity.
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
