--- gtksourceview/gtksourceviewmm/sourceiter.cc.orig	2007-05-01 13:32:36.000000000 -0500
+++ gtksourceview/gtksourceviewmm/sourceiter.cc	2012-06-27 16:41:02.000000000 -0500
@@ -27,7 +27,7 @@
  */
 
 #include "sourceiter.h"
-#include <glib/gmessages.h>
+#include <glib.h>
 namespace gtksourceview
 {
 
--- gtksourceview/src/sourceiter.ccg.orig	2007-04-23 02:08:40.000000000 -0500
+++ gtksourceview/src/sourceiter.ccg	2012-06-27 16:41:02.000000000 -0500
@@ -20,7 +20,7 @@
  */
 
 #include "sourceiter.h"
-#include <glib/gmessages.h>
+#include <glib.h>
 namespace gtksourceview
 {
 
--- tests/test-search.cpp.orig	2007-04-23 02:08:40.000000000 -0500
+++ tests/test-search.cpp	2012-06-27 16:41:02.000000000 -0500
@@ -5,7 +5,7 @@
 #include <string>
 #include <gtkmm.h>
 #include <gtksourceviewmm.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 
 using namespace std ;
 using namespace Gtk;
