--- configure.orig	2017-09-17 18:58:20.000000000 +0000
+++ configure	2017-09-19 12:34:30.000000000 +0000
@@ -5063,8 +5063,8 @@
         *-*-osf*)	platform=TRU64 ;;
         *-apple-*)	platform=MACOSX
 			# Appease 10.7 Lion (OpenSSL deprecated)
-			CXXFLAGS="-Wno-deprecated-declarations"
-			CFLAGS="-Wno-deprecated-declarations"
+			CXXFLAGS="$CXXFLAGS -Wno-deprecated-declarations"
+			CFLAGS="$CFLAGS -Wno-deprecated-declarations"
 			;;
 	*-nto-qnx)	platform=QNX
 			SAMPLE_EXTRA_LIBS="-lsocket -lm"
