--- work/ospec-0.2.1/Makefile	2012-03-20 14:44:04.000000000 +0100
+++ Makefile	2012-07-06 15:16:36.000000000 +0200
@@ -1,10 +1,11 @@
 install:
-	ocamlfind install ospec src/META _build/src/pa_spec.cmo \
+	ocamlfind install -destdir $(DESTDIR)$(PREFIX)/lib/ocaml/site-lib -ldconf ignore \
+	        ospec src/META _build/src/pa_spec.cmo \
 		_build/src/pa_spec.cmi _build/src/spec.cmo _build/src/spec.cmi \
 		_build/src/helpers.cmo _build/src/helpers.cmi _build/src/report.cmo \
 		_build/src/report.cmi _build/src/gen.cmo _build/src/gen.cmi \
 		_build/src/prop.cmo _build/src/prop.cmi
-	install -m 755 _build/src/ospec $(EXEC_PREFIX)/bin
+	install -m 755 _build/src/ospec $(DESTDIR)$(PREFIX)/bin
 
 uninstall:
 	ocamlfind remove ospec
