--- Makefile.PL.orig	2015-07-11 16:15:07.000000000 -0700
+++ Makefile.PL	2015-07-11 16:17:38.000000000 -0700
@@ -24,7 +24,7 @@
 if($use_64_bit_int == -1) {}
 elsif($use_64_bit_int == 1) {$defines .= " -DMATH_GMPZ_NEED_LONG_LONG_INT"}
 else {
-  unless($Config::Config{ivsize} < 8 || $Config{ivtype} eq 'long') {
+  unless($Config::Config{ivsize} < 8) {
     $defines .= " -DMATH_GMPZ_NEED_LONG_LONG_INT";
   }
 }
