--- configure.orig	2011-11-22 09:17:41.000000000 +1100
+++ configure	2011-11-22 09:28:46.000000000 +1100
@@ -3820,13 +3820,13 @@
 	    if test "$Py_DEBUG" = 'true' ; then
 		# Optimization messes up debuggers, so turn it off for
 		# debug builds.
-		OPT="-g -Wall -Wstrict-prototypes"
+		OPT="-g -fwrapv -Wall -Wstrict-prototypes"
 	    else
-		OPT="-g -O3 -Wall -Wstrict-prototypes"
+		OPT="-g -fwrapv -O3 -Wall -Wstrict-prototypes"
 	    fi
 	    ;;
 	*)
-	    OPT="-O3 -Wall -Wstrict-prototypes"
+	    OPT="-O3 -fwrapv -Wall -Wstrict-prototypes"
 	    ;;
 	esac
 	case $ac_sys_system in
