--- configure.orig	2010-09-28 02:39:37.000000000 -0700
+++ configure	2013-04-03 23:01:24.000000000 -0700
@@ -11034,9 +11034,9 @@
 # Check whether --with-idl-compiler was given.
 if test "${with_idl_compiler+set}" = set; then :
   withval=$with_idl_compiler; if test "x$withval" != x; then
-		if test -f "$withval"; then
+		#if test -f "$withval"; then
 		    IDL_COMPILER=$withval
-		fi
+		#fi
 	fi
 
 fi
@@ -12950,7 +12950,7 @@
 
 if test "$enable_maintainer_mode" = "yes"; then
 	DISABLE_DEPRECATED_CFLAGS=" \
-		-DG_DISABLE_DEPRECATED -DG_DISABLE_SINGLE_INCLUDES"
+		-DG_DISABLE_SINGLE_INCLUDES"
 
 fi
 
