--- Docs/src/bin/halibut/misc.c.orig	2022-06-08 13:40:13.000000000 +0200
+++ Docs/src/bin/halibut/misc.c	2022-06-08 13:43:24.000000000 +0200
@@ -367,7 +367,7 @@ unsigned long getutcunixtime()
   if (0 == clock_gettime(CLOCK_REALTIME, &ts))
     return ts.tv_sec;
 #endif
-#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && defined(TIME_UTC)
   if (timespec_get(&ts, TIME_UTC)) /* implementation defined epoch :( */
     return ts.tv_sec;
 #endif
