--- Makefile.orig	2014-11-30 22:56:18.000000000 -0600
+++ Makefile	2014-12-22 02:00:06.000000000 -0600
@@ -16,10 +16,10 @@
 
 SRCDIR = src
 EXPDIR = exp
-export CUDD_ROOT = extern/cudd-2.5.0
-CUDD_LIB = $(CUDD_ROOT)/cudd/libcudd.a $(CUDD_ROOT)/mtr/libmtr.a $(CUDD_ROOT)/st/libst.a $(CUDD_ROOT)/util/libutil.a $(CUDD_ROOT)/epd/libepd.a
-export CUDD_XCFLAGS = -mtune=native -DHAVE_IEEE_754 -DBSD -DSIZEOF_VOID_P=8 -DSIZEOF_LONG=8
-CUDD_INC = -I$(CUDD_ROOT)/include
+export CUDD_PREFIX = $(prefix)
+CUDD_LIB = $(CUDD_PREFIX)/lib/cudd/libcudd.dylib
+export CUDD_XCFLAGS = -DHAVE_IEEE_754 -DBSD
+CUDD_INC = -I$(CUDD_PREFIX)/include/cudd
 
 LEX = flex
 LFLAGS = 
@@ -30,7 +30,7 @@
 LD = ld -r
 
 CFLAGS = -g -Wall -pedantic -ansi
-ALL_CFLAGS = $(CFLAGS) $(CUDD_XCFLAGS) $(CUDD_INC) -Isrc
+ALL_CFLAGS = $(CFLAGS) $(CUDD_XCFLAGS) -Isrc $(CUDD_INC)
 LDFLAGS = $(CUDD_LIB) -lm $(CUDD_XCFLAGS)
 
 # To use and statically link with GNU Readline
@@ -124,7 +124,7 @@
 	rm -f $(DESTDIR)$(bindir)/gr1c $(DESTDIR)$(bindir)/rg $(DESTDIR)$(bindir)/grpatch
 
 check: $(CORE_PROGRAMS) $(EXP_PROGRAMS)
-	$(MAKE) -C tests CC=$(CC)
+	$(MAKE) -C tests CC="$(CC)"
 
 .PHONY: doc
 doc:
