--- tests/Makefile.orig	2013-12-02 10:21:56.000000000 -0600
+++ tests/Makefile	2014-10-02 00:22:39.000000000 -0500
@@ -3,10 +3,9 @@
 # SCL; 2012, 2013.
 
 
-CUDD_LIB = ../$(CUDD_ROOT)/cudd/libcudd.a ../$(CUDD_ROOT)/mtr/libmtr.a ../$(CUDD_ROOT)/st/libst.a ../$(CUDD_ROOT)/util/libutil.a ../$(CUDD_ROOT)/epd/libepd.a
-
-CFLAGS = $(CFLAGS_BASE) -I../src -I../$(CUDD_ROOT)/include
-LDFLAGS = $(CUDD_LIB) -lm
+CC := gcc
+CFLAGS = $(CFLAGS_BASE) -I../src -I$(CUDD_PREFIX)/include/cudd
+LDFLAGS = -L$(CUDD_PREFIX)/lib/cudd -lcudd -lm
 PROGRAMS = test_logging test_automaton test_automaton_io test_ptree test_ptree_to_BDD test_solve_support test_patching
 
 all: $(PROGRAMS)
@@ -25,25 +24,25 @@
 COMMON_BINS = ../util.o ../automaton.o ../automaton_io.o ../ptree.o ../solve_support.o ../logging.o
 
 test_logging: test_logging.c
-	gcc $(CFLAGS) $^ ../logging.o -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ ../logging.o -o $@ $(LDFLAGS)
 
 test_ptree: test_ptree.c
-	gcc $(CFLAGS) $^ ../ptree.o -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ ../ptree.o -o $@ $(LDFLAGS)
 
 test_ptree_to_BDD: test_ptree_to_BDD.c
-	gcc $(CFLAGS) $^ ../ptree.o -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ ../ptree.o -o $@ $(LDFLAGS)
 
 test_automaton: test_automaton.c
-	gcc $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
 
 test_automaton_io: test_automaton_io.c
-	gcc $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
 
 test_solve_support: test_solve_support.c
-	gcc $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ $(COMMON_BINS) -o $@ $(LDFLAGS)
 
 test_patching: test_patching.c
-	gcc $(CFLAGS) $^ $(COMMON_BINS) ../patching.o ../patching_support.o -o $@ $(LDFLAGS)
+	$(CC) $(CFLAGS) $^ $(COMMON_BINS) ../patching.o ../patching_support.o -o $@ $(LDFLAGS)
 
 clean:
 	rm -fv *~ *.o $(PROGRAMS) temp_*_dump*
