diff --git src/dopt.c src/dopt.c
index d5338169891d7416b3a06b09b77e49c057357b19..7324b66131009f9b21276d4a504654ad10d3e9ff 100644
--- src/dopt.c
+++ src/dopt.c
@@ -95,7 +95,7 @@ int opt_log_stderr = 0;
 int opt_log_level_num = RS_LOG_NOTICE;
 
 /**
- * If true, do not check if a link to distcc exists in /usr/lib/distcc
+ * If true, do not check if a link to distcc exists in @PREFIX@/lib/distcc
  * for every program executed remotely.
  **/
 int opt_make_me_a_botnet = 0;
@@ -135,7 +135,7 @@ static const char *dcc_private_networks[] = {"192.168.0.0/16",
 
 const struct poptOption options[] = {
     { "allow", 'a',      POPT_ARG_STRING, 0, 'a', 0, 0 },
-    { "allow-private", 0,POPT_ARG_STRING, &opt_allow_private, 0, 0, 0 },
+    { "allow-private", 0,POPT_ARG_NONE, &opt_allow_private, 0, 0, 0 },
 #ifdef HAVE_GSSAPI
     { "auth", 0,	 POPT_ARG_NONE, &opt_auth_enabled, 'A', 0, 0 },
     { "blacklist", 0,    POPT_ARG_STRING, &arg_list_file, 'b', 0, 0 },
