--- src/orig.compile.c	2018-03-08 00:01:17.000000000 +0100
+++ src/compile.c	2018-05-17 18:35:33.000000000 +0200
@@ -446,6 +446,7 @@
     return dcc_note_discrepancy(discrepancy_filename);
 }
 
+#ifdef AT_SYMLINK_NOFOLLOW
 /* Re-write "cc" to directly call gcc or clang
  */
 static void dcc_rewrite_generic_compiler(char **argv)
@@ -517,6 +518,7 @@
     } else
         return;
 }
+#endif
 
 
 /* Clang is a native cross-compiler, but needs to be told to what target it is
@@ -682,7 +684,9 @@
     dcc_free_argv(argv);
     argv = new_argv;
     if (!getenv("DISTCC_NO_REWRITE_CROSS")) {
+#ifdef AT_SYMLINK_NOFOLLOW
         dcc_rewrite_generic_compiler(new_argv);
+#endif
         dcc_add_clang_target(new_argv);
         dcc_gcc_rewrite_fqn(new_argv);
     }
