--- src/orig.daemon.c	2018-03-08 00:01:17.000000000 +0100
+++ src/daemon.c	2018-05-17 18:53:57.000000000 +0200
@@ -159,13 +159,13 @@
                        " --make-me-a-botnet. To set up masquerade automatically" \
                        " run update-distcc-symlinks.";
 
-    d = opendir("/usr/lib/distcc");
+    d = opendir("@PREFIX@/lib/distcc");
     if (!d) {
-        rs_log_crit("/usr/lib/distcc not found. %s", warn);
+        rs_log_crit("@PREFIX@/lib/distcc not found. %s", warn);
         dcc_exit(EXIT_COMPILER_MISSING);
     }
     if (!readdir(d)) {
-        rs_log_crit("/usr/lib/distcc empty. %s", warn);
+        rs_log_crit("@PREFIX@/lib/distcc empty. %s", warn);
         dcc_exit(EXIT_COMPILER_MISSING);
     }
 }
