--- src/hosts.c.orig	2009-09-16 20:45:37.000000000 -0400
+++ src/hosts.c	2009-09-16 20:41:32.000000000 -0400
@@ -125,13 +125,14 @@
 int dcc_compare_container(const void *a, const void *b);
 
 
-#ifndef HAVE_STRNDUP
+#if !HAVE_STRNDUP
 /**
  * Copy at most @p size characters from @p src, plus a terminating nul.
  *
  * Really this needs to be in util.c, but it's only used here.
  **/
-static char *strndup(const char *src, size_t size)
+char *strndup(const char *src, size_t size);
+char *strndup(const char *src, size_t size)
 {
     char *dst;
 
