--- Makefile.orig	2013-03-18 17:43:21.000000000 +0400
+++ Makefile	2013-03-18 17:47:25.000000000 +0400
@@ -1,9 +1,6 @@
-CC=cc
-CFLAGS=-O3 -Wall -Wextra
-
 
 pigz: pigz.o yarn.o zopfli/deflate.o zopfli/blocksplitter.o zopfli/tree.o zopfli/lz77.o zopfli/cache.o zopfli/hash.o zopfli/util.o zopfli/squeeze.o zopfli/katajainen.o
-	$(CC) -o pigz $^ -lpthread -lz
+	$(CC) ${CFLAGS} ${LDFLAGS} -o pigz $^ -lpthread -lz
 	ln -f pigz unpigz
 
 pigz.o: pigz.c yarn.h zopfli/deflate.h zopfli/util.h
@@ -31,19 +28,19 @@
 dev: pigz pigzt pigzn
 
 pigzt: pigzt.o yarnt.o
-	$(CC) -o pigzt pigzt.o yarnt.o -lpthread -lz
+	$(CC) ${CFLAGS} ${LDFLAGS} -o pigzt pigzt.o yarnt.o -lpthread -lz
 
 pigzt.o: pigz.c yarn.h
-	$(CC) -Wall -O3 -DDEBUG -g -c -o pigzt.o pigz.c
+	$(CC) ${CFLAGS} ${LDFLAGS} -Wall -O3 -DDEBUG -g -c -o pigzt.o pigz.c
 
 yarnt.o: yarn.c yarn.h
-	$(CC) -Wall -O3 -DDEBUG -g -c -o yarnt.o yarn.c
+	$(CC) ${CFLAGS} ${LDFLAGS} -Wall -O3 -DDEBUG -g -c -o yarnt.o yarn.c
 
 pigzn: pigzn.o
-	$(CC) -o pigzn pigzn.o -lz
+	$(CC) ${CFLAGS} ${LDFLAGS} -o pigzn pigzn.o -lz
 
 pigzn.o: pigz.c
-	$(CC) -Wall -O3 -DDEBUG -DNOTHREAD -g -c -o pigzn.o pigz.c
+	$(CC) ${CFLAGS} ${LDFLAGS} -Wall -O3 -DDEBUG -DNOTHREAD -g -c -o pigzn.o pigz.c
 
 test: pigz
 	./pigz -kf pigz.c ; ./pigz -t pigz.c.gz
