--- pom/pom.h.orig	2000-05-25 16:38:55.000000000 -0500
+++ pom/pom.h	2010-01-07 12:04:24.000000000 -0600
@@ -410,11 +410,11 @@
 };
 ////////////////////////////////////////////////////////////////////
 class Entities {
-  friend IEntity;
-  friend ISubentity;
-  friend IAtom;
-  friend IMonomer;
-  friend IBond;
+  friend class IEntity;
+  friend class ISubentity;
+  friend class IAtom;
+  friend class IMonomer;
+  friend class IBond;
  public:
   static DB *db_default;
   DB *db;
@@ -502,7 +502,7 @@
 };
 ////////////////////////////////////////////////////////////////////
 class ISubentity {
-  friend IMonomer;
+  friend class IMonomer;
   int iE;
   int iS;
   int iM;
@@ -576,9 +576,9 @@
 };
 ////////////////////////////////////////////////////////////////////
 class IAtom {
-  friend ISubentity;
-  friend IMonomer;
-  friend IBond;
+  friend class ISubentity;
+  friend class IMonomer;
+  friend class IBond;
   int iE;
   int iS;
   int iA;
