--- Makefile.orig	2000-05-25 20:26:03.000000000 -0500
+++ Makefile	2010-01-13 08:49:40.000000000 -0600
@@ -1,4 +1,4 @@
-all: make_pom make_ce make_copy
+all: make_pom make_ce
 
 make_ce:
 	$(MAKE) -C ce
@@ -16,3 +16,7 @@
 use_exe_sgi:
 	cp ce/ce_sgi ./CE
 	cp pom/mkDB_sgi pom/mkDB
+
+install:
+	cp ce/ce ${DESTDIR}${PREFIX}/bin/ce
+	cp pom/mkDB ${DESTDIR}${PREFIX}/bin/mkDB
--- ce/Makefile.orig	2014-03-13 11:48:06.000000000 -0500
+++ ce/Makefile	2014-03-13 11:52:06.000000000 -0500
@@ -7,7 +7,7 @@
 LINTFILES=$(SOURCES:.C=.ln)
 INCLUDE_DIR=
 
-CC = g++   ${INCLUDE_DIR} -DREAD_WRITE 
+CC = @CXX@   ${INCLUDE_DIR} -DREAD_WRITE 
 CFLAGS = -DFUNCPROTO  
 
 
--- pom/Makefile.orig	2014-03-13 11:49:41.000000000 -0500
+++ pom/Makefile	2014-03-13 11:52:06.000000000 -0500
@@ -23,7 +23,7 @@
 	  calc.h \
 	  linkedid.h
 
-CC = g++
+CC = @CXX@
 
 MKOBJS	= $(MKSRCS:.C=.o)
 
