diff --git deps/build_deps.sh deps/build_deps.sh
index 62cffaf6c..ff2e5903e 100755
--- deps/build_deps.sh
+++ deps/build_deps.sh
@@ -63,9 +63,9 @@ SDKROOT = $SDKROOT
 MACOSX_DEPLOYMENT_TARGET = $MACOSX_DEPLOYMENT_TARGET
 CONFIGURATION = $CONFIGURATION
 
-sed = $GSED
-yacc = $YACC
-curl = $CURL
+sed = @@PREFIX@@/bin/gsed
+yacc = @@PREFIX@@/bin/yacc
+curl = @@PREFIX@@/bin/curl
 
 top_srcdir = `pwd`
 builddir = $DEPS_BUILD_DIR
@@ -81,7 +81,7 @@ EOT
 trap 'exit 0' INT
 
 for ARCH in $ARCHS; do
-    ninja -f build.$ARCH.ninja $1
+    @@PREFIX@@/bin/ninja -f build.$ARCH.ninja $1
 done
 
 ../macos/merge-archs.sh "$DEPS_BUILD_DIR/$1"
