diff --git a/src/plugins/rssyl/libfeed/parser.c b/src/plugins/rssyl/libfeed/parser.c
index 14f4a36..6439245 100644
--- a/src/plugins/rssyl/libfeed/parser.c
+++ b/src/plugins/rssyl/libfeed/parser.c
@@ -34,6 +34,15 @@
 
 #include "parser.h"
 
+enum {
+	FEED_TYPE_NONE,
+	FEED_TYPE_RDF,
+	FEED_TYPE_RSS_20,
+	FEED_TYPE_ATOM_03,
+	FEED_TYPE_ATOM_10,
+	FEED_TYPE_OPML
+} FeedTypes;
+
 static void _handler_set(XML_Parser parser, guint type)
 {
 	if( parser == NULL )
diff --git a/src/plugins/rssyl/libfeed/parser.h b/src/plugins/rssyl/libfeed/parser.h
index f8cbb9a..02cd0e5 100644
--- a/src/plugins/rssyl/libfeed/parser.h
+++ b/src/plugins/rssyl/libfeed/parser.h
@@ -32,14 +32,4 @@ gchar *feed_parser_get_attribute_value(const gchar **attr, const gchar *name);
 int feed_parser_unknown_encoding_handler(void *encdata, const XML_Char *name,
 		XML_Encoding *info);
 
-
-enum {
-	FEED_TYPE_NONE,
-	FEED_TYPE_RDF,
-	FEED_TYPE_RSS_20,
-	FEED_TYPE_ATOM_03,
-	FEED_TYPE_ATOM_10,
-	FEED_TYPE_OPML
-} FeedTypes;
-
 #endif /* __PARSER_H */
diff --git a/src/plugins/rssyl/libfeed/parser_atom10.c b/src/plugins/rssyl/libfeed/parser_atom10.c
index 9cf4ffa..8f9f4d6 100644
--- a/src/plugins/rssyl/libfeed/parser_atom10.c
+++ b/src/plugins/rssyl/libfeed/parser_atom10.c
@@ -29,6 +29,14 @@
 #include "parser.h"
 #include "parser_atom10.h"
 
+enum {
+	FEED_LOC_ATOM10_NONE,
+	FEED_LOC_ATOM10_ENTRY,
+	FEED_LOC_ATOM10_AUTHOR,
+	FEED_LOC_ATOM10_SOURCE,
+	FEED_LOC_ATOM10_CONTENT
+} FeedAtom10Locations;
+
 void feed_parser_atom10_start(void *data, const gchar *el, const gchar **attr)
 {
 	FeedParserCtx *ctx = (FeedParserCtx *)data;
diff --git a/src/plugins/rssyl/libfeed/parser_atom10.h b/src/plugins/rssyl/libfeed/parser_atom10.h
index 85e95b4..e3c71fb 100644
--- a/src/plugins/rssyl/libfeed/parser_atom10.h
+++ b/src/plugins/rssyl/libfeed/parser_atom10.h
@@ -23,12 +23,4 @@
 void feed_parser_atom10_start(void *data, const char *el, const char **attr);
 void feed_parser_atom10_end(void *data, const char *el);
 
-enum {
-	FEED_LOC_ATOM10_NONE,
-	FEED_LOC_ATOM10_ENTRY,
-	FEED_LOC_ATOM10_AUTHOR,
-	FEED_LOC_ATOM10_SOURCE,
-	FEED_LOC_ATOM10_CONTENT
-} FeedAtom10Locations;
-
 #endif /* __PARSER_ATOM03_H */
diff --git a/src/plugins/rssyl/libfeed/parser_rdf.c b/src/plugins/rssyl/libfeed/parser_rdf.c
index d514bf0..77e1df5 100644
--- a/src/plugins/rssyl/libfeed/parser_rdf.c
+++ b/src/plugins/rssyl/libfeed/parser_rdf.c
@@ -28,6 +28,12 @@
 #include "date.h"
 #include "parser_rdf.h"
 
+enum {
+	FEED_LOC_RDF_NONE,
+	FEED_LOC_RDF_CHANNEL,
+	FEED_LOC_RDF_ITEM
+} FeedRdfLocations;
+
 void feed_parser_rdf_start(void *data, const gchar *el, const gchar **attr)
 {
 	FeedParserCtx *ctx = (FeedParserCtx *)data;
diff --git a/src/plugins/rssyl/libfeed/parser_rdf.h b/src/plugins/rssyl/libfeed/parser_rdf.h
index 1c3809f..66299cb 100644
--- a/src/plugins/rssyl/libfeed/parser_rdf.h
+++ b/src/plugins/rssyl/libfeed/parser_rdf.h
@@ -23,10 +23,4 @@
 void feed_parser_rdf_start(void *data, const char *el, const char **attr);
 void feed_parser_rdf_end(void *data, const char *el);
 
-enum {
-	FEED_LOC_RDF_NONE,
-	FEED_LOC_RDF_CHANNEL,
-	FEED_LOC_RDF_ITEM
-} FeedRdfLocations;
-
 #endif /* __PARSER_RDF_H */
