diff --git a/lib/baz_burster.cc b/lib/baz_burster.cc
index 39d4d44..c106a5c 100644
--- a/lib/baz_burster.cc
+++ b/lib/baz_burster.cc
@@ -222,7 +222,7 @@ fprintf(stderr, "[%s<%li>] updated time\n", name().c_str(), unique_id());
 						if (d_config.sample_interval)
 							limit /= (double)d_config.sample_rate;
 						
-						if (diff >= boost::posix_time::microseconds(limit * 1e6))
+						if (diff >= boost::posix_time::microseconds(long(limit * 1e6)))
 						//if (d >= limit)
 						{
 //fprintf(stderr, "[%s<%i>] host elapsed %f\n", name().c_str(), unique_id(), ((float)diff.ticks() / (float)d_system_time_ticks_per_second));
