diff -uNr ../yum-1.0.1/yummain.py ./yummain.py
--- ../yum-1.0.1/yummain.py	Mon May 19 22:54:59 2003
+++ ./yummain.py	Sat May 31 18:32:23 2003
@@ -35,8 +35,8 @@
 
     # our default config file location
     yumconffile=None
-    if os.access("/etc/yum.conf", os.R_OK):
-        yumconffile="/etc/yum.conf"
+    if os.access("/opt/local/etc/yum.conf", os.R_OK):
+        yumconffile="/opt/local/etc/yum.conf"
         
     try:
         gopts, cmds = getopt.getopt(args, 'Cc:hR:e:d:y', ['help'])
