--- etc/Makefile.in.orig	Sat Jan 24 21:13:29 2004
+++ etc/Makefile.in	Sat Jan 24 20:58:59 2004
@@ -36,14 +36,14 @@
 
 
 install: all installdirs
-	mkdir -p $(DESTDIR)/etc/{cron.daily,init.d,logrotate.d}
-	$(INSTALL_DATA) $(srcdir)/yum.conf $(DESTDIR)/etc/yum.conf
-	$(INSTALL_SCRIPT) $(srcdir)/yum.cron $(DESTDIR)/etc/cron.daily/yum.cron
-	$(INSTALL_SCRIPT) $(srcdir)/yum.init $(DESTDIR)/etc/init.d/yum
-	$(INSTALL_DATA) $(srcdir)/yum.logrotate $(DESTDIR)/etc/logrotate.d/yum
+	mkdir -p $(DESTDIR)@@PREFIX@@/etc/{cron.daily,init.d,logrotate.d}
+	$(INSTALL_DATA) $(srcdir)/yum.conf $(DESTDIR)@@PREFIX@@/etc/yum.conf
+	$(INSTALL_SCRIPT) $(srcdir)/yum.cron $(DESTDIR)@@PREFIX@@/etc/cron.daily/yum.cron
+	$(INSTALL_SCRIPT) $(srcdir)/yum.init $(DESTDIR)@@PREFIX@@/etc/init.d/yum
+	$(INSTALL_DATA) $(srcdir)/yum.logrotate $(DESTDIR)@@PREFIX@@/etc/logrotate.d/yum
 
 uninstall:
-	$(RM) /etc/yum.conf
+	$(RM) @@PREFIX/etc/yum.conf
 
 
 clean:
