--- ./bin/yum-arch.jbj	2005-08-20 16:12:32.000000000 -0400
+++ ./bin/yum-arch	2005-08-20 16:12:51.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 # yum-arch -- Creates YUM distribution databases
 #
 # This program is free software; you can redistribute it and/or modify
@@ -18,6 +18,6 @@
 
 import sys
 
-sys.path.insert(1,'/usr/share/yum-cli')
+sys.path.insert(1,'@@PREFIX@@/share/yum-cli')
 import pullheaders
 pullheaders.main(sys.argv[1:])
--- ./bin/yum.py.jbj	2005-08-20 15:48:36.000000000 -0400
+++ ./bin/yum.py	2005-08-20 15:47:55.000000000 -0400
@@ -1,7 +1,7 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 
 import sys
-sys.path.insert(0, '/usr/share/yum-cli')
+sys.path.insert(0, '@@PREFIX@@/share/yum-cli')
 try:
     import yummain
     yummain.main(sys.argv[1:])
--- ./callback.py.jbj	2005-08-20 16:14:26.000000000 -0400
+++ ./callback.py	2005-08-20 16:14:39.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./cli.py.jbj	2005-08-20 16:14:56.000000000 -0400
+++ ./cli.py	2005-08-20 16:34:06.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -99,8 +99,8 @@
     
         # our default config file location
         yumconffile = None
-        if os.access("/etc/yum.conf", os.R_OK):
-            yumconffile = "/etc/yum.conf"
+        if os.access("@@PREFIX@@/etc/yum.conf", os.R_OK):
+            yumconffile = "@@PREFIX@@/etc/yum.conf"
     
         try:
             gopts, self.cmds = getopt.getopt(args, 'tCc:hR:e:d:y', ['help',
@@ -126,7 +126,7 @@
         sleeptime=0
         root = '/'
         installroot = None
-        conffile = '/etc/yum.conf'
+        conffile = '@@PREFIX@@/etc/yum.conf'
         
         try: 
             for o,a in gopts:
--- ./docs/Makefile.jbj	2005-08-20 16:13:23.000000000 -0400
+++ ./docs/Makefile	2005-08-20 16:13:51.000000000 -0400
@@ -5,7 +5,7 @@
 	rm -f *.pyc *.pyo *~
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/man/man{5,8}
-	install -m 644 yum.8 $(DESTDIR)/usr/share/man/man8/yum.8
-	install -m 644 yum-arch.8 $(DESTDIR)/usr/share/man/man8/yum-arch.8
-	install -m 644 yum.conf.5 $(DESTDIR)/usr/share/man/man5/yum.conf.5
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/man/man{5,8}
+	install -m 644 yum.8 $(DESTDIR)@@PREFIX@@/share/man/man8/yum.8
+	install -m 644 yum-arch.8 $(DESTDIR)@@PREFIX@@/share/man/man8/yum-arch.8
+	install -m 644 yum.conf.5 $(DESTDIR)@@PREFIX@@/share/man/man5/yum.conf.5
--- ./docs/yum.8.jbj	2005-08-20 16:20:35.000000000 -0400
+++ ./docs/yum.8	2005-08-20 16:21:06.000000000 -0400
@@ -245,8 +245,8 @@
 .PP
 .SH "FILES"
 .nf 
-/etc/yum.conf
-/var/cache/yum/
+@@PREFIX@@/etc/yum.conf
+@@PREFIX@@/var/cache/yum/
 .fi 
 
 .PP 
--- ./docs/yum.conf.5.jbj	2005-08-20 16:21:22.000000000 -0400
+++ ./docs/yum.conf.5	2005-08-20 16:22:18.000000000 -0400
@@ -4,10 +4,10 @@
 \fByum.conf\fR \- Configuration file for \fByum(8)\fR.
 .SH "DESCRIPTION"
 .LP 
-Yum uses a configuration file at \fB/etc/yum.conf\fR. 
+Yum uses a configuration file at \fB@@PREFIX@@/etc/yum.conf\fR. 
 .LP
 Additional configuration files are also read from the directory set by the
-\fBreposdir\fR option (default is `/etc/yum.repos.d'). See the \fBreposdir\fR
+\fBreposdir\fR option (default is `@@PREFIX@@/etc/yum.repos.d'). See the \fBreposdir\fR
 option below for further details.
 
 .SH "PARAMETERS"
@@ -24,14 +24,14 @@
 
 .IP \fBcachedir\fR
 Directory where yum should store its cache and db files. The default is
-`/var/cache/yum'.
+`@@PREFIX@@/var/cache/yum'.
 
 .IP \fBreposdir\fR
 Directory where yum should look for .repo files for its configuration
-of repositories. Default is `/etc/yum.repos.d'. Each file in this directory
+of repositories. Default is `@@PREFIX@@/etc/yum.repos.d'. Each file in this directory
 should contain one or more repository sections as documented in the
 repository options below. These will be merged with the repositories defined
-in /etc/yum.conf to form the complete set of repositories that yum will use.
+in @@PREFIX@@/etc/yum.conf to form the complete set of repositories that yum will use.
 
 .IP \fBdebuglevel\fR
 Debug message output level. Practical range is 0\-10. Default is `2'.
@@ -300,8 +300,8 @@
 
 .SH "URL INCLUDE SYNTAX"
 .LP 
-The inclusion of external configuration files is supported for /etc/yum.conf
-and the .repo files in the /etc/yum.repos.d directory. To include a URL, use a
+The inclusion of external configuration files is supported for @@PREFIX@@/etc/yum.conf
+and the .repo files in the @@PREFIX@@/etc/yum.repos.d directory. To include a URL, use a
 line of the following format:
 
 include=url://to/some/location
@@ -338,7 +338,7 @@
 
 .SH "FILES"
 .LP 
-/etc/yum.conf
+@@PREFIX@@/etc/yum.conf
 .SH "SEE ALSO"
 .LP 
 yum(8)
--- ./etc/Makefile.jbj	2005-08-20 16:32:18.000000000 -0400
+++ ./etc/Makefile	2005-08-20 16:33:04.000000000 -0400
@@ -5,21 +5,21 @@
 	rm -f *.pyc *.pyo *~
 
 install:
-	#mkdir -p $(DESTDIR)/etc/security/console.apps 
-	#install -m 644 yum.console $(DESTDIR)/etc/security/console.apps/yum
+	#mkdir -p $(DESTDIR)@@PREFIX@@/etc/security/console.apps 
+	#install -m 644 yum.console $(DESTDIR)@@PREFIX@@/etc/security/console.apps/yum
 
-	#mkdir -p $(DESTDIR)/etc/pam.d
-	#install -m 644 yum.pam $(DESTDIR)/etc/pam.d/yum
+	#mkdir -p $(DESTDIR)@@PREFIX@@/etc/pam.d
+	#install -m 644 yum.pam $(DESTDIR)@@PREFIX@@/etc/pam.d/yum
 
-	mkdir -p $(DESTDIR)/etc/yum.repos.d
-	install -m 644 yum.conf $(DESTDIR)/etc/yum.conf
+	mkdir -p $(DESTDIR)@@PREFIX@@/etc/yum.repos.d
+	install -m 644 yum.conf $(DESTDIR)@@PREFIX@@/etc/yum.conf
 
-	mkdir -p $(DESTDIR)/etc/cron.daily
-	install -m 755 yum.cron $(DESTDIR)/etc/cron.daily/yum.cron
+	mkdir -p $(DESTDIR)@@PREFIX@@/etc/cron.daily
+	install -m 755 yum.cron $(DESTDIR)@@PREFIX@@/etc/cron.daily/yum.cron
 
-	mkdir -p $(DESTDIR)/etc/rc.d/init.d
-	install -m 755 yum.init $(DESTDIR)/etc/rc.d/init.d/yum
+	mkdir -p $(DESTDIR)@@PREFIX@@/etc/rc.d/init.d
+	install -m 755 yum.init $(DESTDIR)@@PREFIX@@/etc/rc.d/init.d/yum
 
-	mkdir -p $(DESTDIR)/etc/logrotate.d
-	install -m 644 yum.logrotate $(DESTDIR)/etc/logrotate.d/yum
+	mkdir -p $(DESTDIR)@@PREFIX@@/etc/logrotate.d
+	install -m 644 yum.logrotate $(DESTDIR)@@PREFIX@@/etc/logrotate.d/yum
 
--- ./etc/yum.conf.jbj	2005-08-20 16:24:52.000000000 -0400
+++ ./etc/yum.conf	2005-08-20 16:26:42.000000000 -0400
@@ -1,23 +1,18 @@
 [main]
-cachedir=/var/cache/yum
+cachedir=@@PREFIX@@/var/cache/yum
 debuglevel=2
-logfile=/var/log/yum.log
+logfile=@@PREFIX@@/var/log/yum.log
 pkgpolicy=newest
-distroverpkg=redhat-release
+distroverpkg=dports
 tolerant=1
 exactarch=1
 obsoletes=1
 
 
 # PUT YOUR REPOS HERE OR IN separate files named file.repo
-# in /etc/yum.repos.d
-#[base]
-#name=Red Hat Linux $releasever - $basearch - Base
-#baseurl=http://mirror.dulug.duke.edu/pub/yum-repository/redhat/$releasever/$basearch/
-
-
-#[updates]
-#name=Red Hat Linux $releasever - Updates
-#baseurl=http://mirror.dulug.duke.edu/pub/yum-repository/redhat/updates/$releasever/
+# in @@PREFIX@@/etc/yum.repos.d
+[base]
+name=Open Darwin dports - $basearch - Base
+baseurl=http://wraptastic.org/dports
 
 
--- ./etc/yum.console.jbj	2005-08-20 16:11:32.000000000 -0400
+++ ./etc/yum.console	2005-08-20 16:12:02.000000000 -0400
@@ -1,4 +1,4 @@
 USER=root
-PROGRAM=/usr/sbin/yum
+PROGRAM=@@PREFIX@@/sbin/yum
 GUI=no
 SESSION=true
--- ./etc/yum.cron.jbj	2005-08-20 16:10:21.000000000 -0400
+++ ./etc/yum.cron	2005-08-20 16:10:56.000000000 -0400
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-if [ -f /var/lock/subsys/yum ]; then
-	/usr/bin/yum -R 10 -e 0 -d 0 -y update yum
-	/usr/bin/yum -R 120 -e 0 -d 0 -y update
+if [ -f @@PREFIX@@/var/lock/subsys/yum ]; then
+	@@PREFIX@@/bin/yum -R 10 -e 0 -d 0 -y update yum
+	@@PREFIX@@/bin/yum -R 120 -e 0 -d 0 -y update
 fi
--- ./etc/yum.init.jbj	2005-08-20 16:27:03.000000000 -0400
+++ ./etc/yum.init	2005-08-20 16:27:21.000000000 -0400
@@ -14,7 +14,7 @@
 # source function library
 . /etc/rc.d/init.d/functions
 
-lockfile=/var/lock/subsys/yum
+lockfile=@@PREFIX@@/var/lock/subsys/yum
 
 RETVAL=0
 
--- ./etc/yum.logrotate.jbj	2005-08-20 16:27:45.000000000 -0400
+++ ./etc/yum.logrotate	2005-08-20 16:27:55.000000000 -0400
@@ -1,4 +1,4 @@
-/var/log/yum.log {
+@@PREFIX@@/var/log/yum.log {
     missingok
     notifempty
     size 30k
--- ./i18n.py.jbj	2005-08-20 16:15:27.000000000 -0400
+++ ./i18n.py	2005-08-20 16:15:56.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@@PREFIX@@/bin/env python
 """i18n abstraction
 
 License: GPL
@@ -17,7 +17,7 @@
         t = gettext.translation('yum')
         _ = t.gettext
     else:
-        gettext.bindtextdomain('yum', '/usr/share/locale')
+        gettext.bindtextdomain('yum', '@@PREFIX@@/share/locale')
         gettext.textdomain('yum')
         _ = gettext.gettext
 
--- ./Makefile.jbj	2005-08-20 16:19:01.000000000 -0400
+++ ./Makefile	2005-08-20 16:28:28.000000000 -0400
@@ -17,17 +17,17 @@
 	for d in $(SUBDIRS); do make -C $$d; [ $$? = 0 ] || exit 1 ; done
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/yum-cli
+	mkdir -p $(DESTDIR)@@PREFIX@@/share/yum-cli
 	for p in $(PYFILES) ; do \
-		install -m 644 $$p $(DESTDIR)/usr/share/yum-cli/$$p; \
+		install -m 644 $$p $(DESTDIR)@@PREFIX@@/share/yum-cli/$$p; \
 	done
-	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/usr/share/yum-cli', 1, '$(PYDIR)', 1)"
+	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)@@PREFIX@@/share/yum-cli', 1, '$(PYDIR)', 1)"
 
-	mkdir -p $(DESTDIR)/usr/bin $(DESTDIR)/usr/bin
-	install -m 755 bin/yum.py $(DESTDIR)/usr/bin/yum
-	install -m 755 bin/yum-arch $(DESTDIR)/usr/bin/yum-arch
+	mkdir -p $(DESTDIR)@@PREFIX@@/bin $(DESTDIR)@@PREFIX@@/bin
+	install -m 755 bin/yum.py $(DESTDIR)@@PREFIX@@/bin/yum
+	install -m 755 bin/yum-arch $(DESTDIR)@@PREFIX@@/bin/yum-arch
 
-	mkdir -p $(DESTDIR)/var/cache/yum
+	mkdir -p $(DESTDIR)@@PREFIX@@/var/cache/yum
 
 	for d in $(SUBDIRS); do make PYTHON=$(PYTHON) DESTDIR=`cd $(DESTDIR); pwd` -C $$d install; [ $$? = 0 ] || exit 1; done
 
--- ./output.py.jbj	2005-08-20 16:16:10.000000000 -0400
+++ ./output.py	2005-08-20 16:16:22.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 
 """This handles actual output from the cli"""
 
--- ./po/pygettext.py.jbj	2005-08-20 15:49:06.000000000 -0400
+++ ./po/pygettext.py	2005-08-20 15:49:25.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 # Originally written by Barry Warsaw <bwarsaw@python.org>
 #
 # minimally patched to make it even more xgettext compatible 
--- ./posts/bootloadercfg.py.jbj	2005-08-20 15:49:52.000000000 -0400
+++ ./posts/bootloadercfg.py	2005-08-20 15:50:10.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 
 # base class for boot loader updating code code for Update Agent
 # Copyright (c) 2001 Red Hat, Inc.  Distributed under GPL.
--- ./posts/checkbootloader.py.jbj	2005-08-20 15:50:21.000000000 -0400
+++ ./posts/checkbootloader.py	2005-08-20 15:51:01.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 #
 # Check to see whether it looks like GRUB or LILO is the boot loader
 # being used on the system.
--- ./posts/grubcfg.py.jbj	2005-08-20 15:51:14.000000000 -0400
+++ ./posts/grubcfg.py	2005-08-20 15:51:25.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 #
 # Module for munging lilo.conf lilo configurations file in the case
 #   of installing new or updated kernel packages
--- ./posts/iutil.py.jbj	2005-08-20 16:36:33.000000000 -0400
+++ ./posts/iutil.py	2005-08-20 16:37:08.000000000 -0400
@@ -16,6 +16,11 @@
         arch[2:4] == "86"):
         arch = "i386"
 
+    if arch == "Power Macintosh":
+        arch = "ppc"
+    if arch == "powerpc":
+        arch = "ppc"
+
     if arch == "sparc64":
         arch = "sparc"
 
--- ./posts/lilo.py.jbj	2005-08-20 15:51:46.000000000 -0400
+++ ./posts/lilo.py	2005-08-20 15:52:01.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 #
 # Module for manipulation of lilo.conf files. Original found
 #    in the ananaconda installer
--- ./posts/lilocfg.py.jbj	2005-08-20 15:52:12.000000000 -0400
+++ ./posts/lilocfg.py	2005-08-20 15:52:31.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 #
 # Module for munging lilo.conf lilo configurations file in the case
 #   of installing new or updated kernel packages
--- ./pullheaders.py.jbj	2005-08-20 16:16:33.000000000 -0400
+++ ./pullheaders.py	2005-08-20 16:16:49.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repomd/mdErrors.py.jbj	2005-08-20 15:53:15.000000000 -0400
+++ ./repomd/mdErrors.py	2005-08-20 15:53:35.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repomd/mdUtils.py.jbj	2005-08-20 15:53:52.000000000 -0400
+++ ./repomd/mdUtils.py	2005-08-20 15:54:25.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./repomd/packageObject.py.jbj	2005-08-20 16:01:02.000000000 -0400
+++ ./repomd/packageObject.py	2005-08-20 16:01:15.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repomd/packageSack.py.jbj	2005-08-20 16:01:34.000000000 -0400
+++ ./repomd/packageSack.py	2005-08-20 16:01:48.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repomd/repoMDObject.py.jbj	2005-08-20 16:02:02.000000000 -0400
+++ ./repomd/repoMDObject.py	2005-08-20 16:02:14.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./repomd/test.py.jbj	2005-08-20 16:02:28.000000000 -0400
+++ ./repomd/test.py	2005-08-20 16:02:40.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./rpmUtils/__init__.py.jbj	2005-08-20 16:03:01.000000000 -0400
+++ ./rpmUtils/__init__.py	2005-08-20 16:03:14.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 
 import rpm
 import miscutils
--- ./rpmUtils/arch.py.jbj	2005-08-21 13:07:17.000000000 -0400
+++ ./rpmUtils/arch.py	2005-08-21 13:08:06.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 #
 
 import os
@@ -186,9 +186,9 @@
     return arch
         
 def getCanonArch(skipRpmPlatform = 0):
-    if not skipRpmPlatform and os.access("/etc/rpm/platform", os.R_OK):
+    if not skipRpmPlatform and os.access("@@PREFIX@@/etc/rpm/platform", os.R_OK):
         try:
-            f = open("/etc/rpm/platform", "r")
+            f = open("@@PREFIX@@/etc/rpm/platform", "r")
             line = f.readline()
             f.close()
             (arch, vendor, opersys) = line.split("-", 2)
@@ -197,6 +197,8 @@
             pass
         
     arch = os.uname()[4]
+    if arch == "Power Macintosh":
+	arch = "ppc"
 
     if (len(arch) == 4 and arch[0] == "i" and arch[2:4] == "86"):
         return getCanonX86Arch(arch)
--- ./rpmUtils/miscutils.py.jbj	2005-08-20 16:04:05.000000000 -0400
+++ ./rpmUtils/miscutils.py	2005-08-20 16:04:20.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- ./rpmUtils/oldUtils.py.jbj	2005-08-20 16:04:30.000000000 -0400
+++ ./rpmUtils/oldUtils.py	2005-08-20 16:04:43.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 
 import rpm
 import types
--- ./rpmUtils/transaction.py.jbj	2005-08-20 16:04:55.000000000 -0400
+++ ./rpmUtils/transaction.py	2005-08-20 16:05:16.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 
 #
 # Client code for Update Agent
--- ./rpmUtils/updates.py.jbj	2005-08-20 16:06:50.000000000 -0400
+++ ./rpmUtils/updates.py	2005-08-20 16:05:59.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./serverStuff.py.jbj	2005-08-20 16:17:02.000000000 -0400
+++ ./serverStuff.py	2005-08-20 16:17:14.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./translate.py.jbj	2005-08-20 16:17:25.000000000 -0400
+++ ./translate.py	2005-08-20 16:37:49.000000000 -0400
@@ -5,7 +5,7 @@
 
 import os, string
 
-prefix = '/usr'
+prefix = '@@PREFIX@@'
 localedir = prefix + '/share/locale'
 _cat = None
 _cats = {}
@@ -246,6 +246,11 @@
     if (len (arch) == 4 and arch[0] == 'i' and arch[2:4] == "86"):
         arch = "i386"
 
+    if arch == "Power Macintosh":
+        arch = "ppc"
+    if arch == "powerpc":
+        arch = "ppc"
+
     if arch == "sparc64":
         arch = "sparc"
 
--- ./yum/__init__.py.jbj	2005-08-20 16:07:22.000000000 -0400
+++ ./yum/__init__.py	2005-08-20 16:34:42.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -60,7 +60,7 @@
     def filelog(self, value, msg):
         print msg
     
-    def doConfigSetup(self, fn='/etc/yum.conf'):
+    def doConfigSetup(self, fn='@@PREFIX@@/etc/yum.conf'):
         """basic stub function for doing configuration setup"""
         
         self.conf = config.yumconf(configfile=fn)
@@ -898,7 +898,7 @@
         # if it even vaguely matches
         for arg in args:
             matched = 0
-            globs = ['.*bin\/.*', '.*\/etc\/.*', '^\/usr\/lib\/sendmail$']
+            globs = ['.*bin\/.*', '.*\/etc\/.*', '^\@@PREFIX@@\/lib\/sendmail$']
             for glob in globs:
                 globc = re.compile(glob)
                 if globc.match(arg):
--- ./yum/archwork.py.jbj	2005-08-20 15:44:37.000000000 -0400
+++ ./yum/archwork.py	2005-08-20 16:09:23.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -32,6 +32,10 @@
         newarch = 'ppc'
     if re.search('x86_64', arch):
         newarch = 'x86_64'
+    if re.search('Power Macintosh', arch):
+        arch = 'ppc'
+    if re.search('powerpc', arch):
+        arch = 'ppc'
     if not newarch:
         newarch = arch
     return newarch
@@ -67,10 +71,10 @@
 
 def compatArchList():
     archdict = {}
-    archdict['i386']=['i386','i486','i586','i686','athlon','noarch']
+    archdict['i386']=['i386','i486','i586','i686','athlon','noarch','fat']
     archdict['alpha']=['alpha','alphaev6','noarch']
     archdict['sparc']=['sparc','sparc64','noarch','sun4c','sun4u','sun4d','sun4m','sparcv9']
-    archdict['ppc']=['ppc','noarch','ppc64','powerpc','powerppc','osfmach3_ppc','ppciseries','ppcpseries','rs6000']
+    archdict['ppc']=['ppc','noarch','ppc64','powerpc','powerppc','osfmach3_ppc','ppciseries','ppcpseries','rs6000','Power Macintosh','fat']
     archdict['ia64']=['ia64','noarch', 'i686']
     archdict['s390']=['noarch','s390']
     archdict['s390x']=['noarch','s390','s390x']
--- ./yum/comps.py.jbj	2005-08-20 15:55:27.000000000 -0400
+++ ./yum/comps.py	2005-08-20 15:55:41.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.2
+#!@@PREFIX@@/bin/python2.2
 # -*- mode: python -*-
 
 
--- ./yum/config.py.jbj	2005-08-20 15:55:55.000000000 -0400
+++ ./yum/config.py	2005-08-20 16:29:26.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -171,7 +171,7 @@
 class yumconf(object):
     """primary config class for yum"""
     
-    def __init__(self, configfile = '/etc/yum.conf', root='/'):
+    def __init__(self, configfile = '@@PREFIX@@/etc/yum.conf', root='/'):
         self.cfg = CFParser()
         configh = confpp(configfile)
         try:
@@ -193,9 +193,9 @@
                       
                       
         #defaults -either get them or set them
-        optionstrings = [('cachedir', '/var/cache/yum'), 
-                         ('logfile', '/var/log/yum.log'), 
-                         ('reposdir', '/etc/yum.repos.d'),
+        optionstrings = [('cachedir', '@@PREFIX@@/var/cache/yum'), 
+                         ('logfile', '@@PREFIX@@/var/log/yum.log'), 
+                         ('reposdir', '@@PREFIX@@/etc/yum.repos.d'),
                          ('rss-filename', 'yum-rss.xml'),
                          ('pkgpolicy', 'newest'),
                          ('syslog_ident', None),
--- ./yum/depsolve.py.jbj	2005-08-20 15:56:34.000000000 -0400
+++ ./yum/depsolve.py	2005-08-20 15:57:58.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -57,11 +57,11 @@
 
         self.log(4, 'Searching pkgSack for dep: %s' % name)
         # we need to check the name - if it doesn't match:
-        # /etc/* bin/* or /usr/lib/sendmail then we should fetch the 
+        # /etc/* bin/* or @@PREFIX@@/lib/sendmail then we should fetch the 
         # filelists.xml for all repos to make the searchProvides more complete.
         if name[0] == '/':
             matched = 0
-            globs = ['.*bin\/.*', '^\/etc\/.*', '^\/usr\/lib\/sendmail$']
+            globs = ['.*bin\/.*', '^\@@PREFIX@@\/etc\/.*', '^\@@PREFIX@@\/lib\/sendmail$']
             for glob in globs:
                 globc = re.compile(glob)
                 if globc.match(name):
--- ./yum/Errors.py.jbj	2005-08-20 16:08:46.000000000 -0400
+++ ./yum/Errors.py	2005-08-20 16:08:59.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yum/failover.py.jbj	2005-08-20 15:58:16.000000000 -0400
+++ ./yum/failover.py	2005-08-20 15:58:30.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yum/groups.py.jbj	2005-08-20 15:58:43.000000000 -0400
+++ ./yum/groups.py	2005-08-20 15:58:57.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@@PREFIX@@/bin/python
 
 import comps
 import sys
--- ./yum/mdcache.py.jbj	2005-08-20 15:59:09.000000000 -0400
+++ ./yum/mdcache.py	2005-08-20 15:59:24.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yum/packages.py.jbj	2005-08-20 15:59:41.000000000 -0400
+++ ./yum/packages.py	2005-08-20 15:59:55.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yum/repos.py.jbj	2005-08-20 16:00:06.000000000 -0400
+++ ./yum/repos.py	2005-08-20 16:00:24.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -tt
+#!@@PREFIX@@/bin/python -tt
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
--- ./yummain.py.jbj	2005-08-20 16:18:10.000000000 -0400
+++ ./yummain.py	2005-08-20 16:20:21.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python -t
+#!@@PREFIX@@/bin/python -t
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
@@ -41,7 +41,7 @@
     def unlock():
         try:
             base.closeRpmDB()
-            base.doUnlock('/var/run/yum.pid')
+            base.doUnlock('@@PREFIX@@/var/run/yum.pid')
         except Errors.LockError, e:
             sys.exit(200)
 
@@ -58,7 +58,7 @@
         sys.exit(1)
         
     try:
-        base.doLock('/var/run/yum.pid')
+        base.doLock('@@PREFIX@@/var/run/yum.pid')
     except Errors.LockError, e:
         base.errorlog(0,'%s' % e.msg)
         sys.exit(200)
