--- bin/yum.orig	Sat Jan 24 21:05:00 2004
+++ bin/yum	Sat Jan 24 20:52:48 2004
@@ -18,12 +18,13 @@
 
 import sys
 import os
-sys.path.insert(1,'/usr/share/yum')
+sys.path.insert(1,'@@PREFIX@@/share/yum')
+sys.path.insert(1,'@@PREFIX@@/lib/python2.2/site-packages')
 import yummain
 import yumlock
 
 uid = os.geteuid()
-lockfile = '/var/run/yum.pid'
+lockfile = '@@PREFIX@@/var/run/yum.pid'
 mypid = str(os.getpid())
 
 if uid == 0:
