diff -uNr ../yum-1.0.1/up2datetheft.py ./up2datetheft.py
--- ../yum-1.0.1/up2datetheft.py	Mon May 19 22:54:59 2003
+++ ./up2datetheft.py	Sat May 31 18:32:23 2003
@@ -21,7 +21,7 @@
 
 
 def openrpmdb(option=0, dbpath=None):
-    dbpath = "/var/lib/rpm/"
+    dbpath = "/opt/local/var/lib/rpm/"
     rpm.addMacro("_dbpath", dbpath)
 
     #log.log_me("Opening rpmdb in %s with option %s" % (dbpath,option))
