diff -uNr ../yum-1.0.1/etc/Makefile.in ./etc/Makefile.in
--- ../yum-1.0.1/etc/Makefile.in	Mon May 19 22:55:00 2003
+++ ./etc/Makefile.in	Sat May 31 18:32:23 2003
@@ -36,14 +36,14 @@
 
 
 install: all installdirs
-	mkdir -p $(DESTDIR)/etc/{cron.daily,init.d,logrotate.d}
-	$(INSTALL_DATA) $(srcdir)/yum.conf $(DESTDIR)/etc/yum.conf
-	$(INSTALL_SCRIPT) $(srcdir)/yum.cron $(DESTDIR)/etc/cron.daily/yum.cron
-	$(INSTALL_SCRIPT) $(srcdir)/yum.init $(DESTDIR)/etc/init.d/yum
-	$(INSTALL_DATA) $(srcdir)/yum.logrotate $(DESTDIR)/etc/logrotate.d/yum
+	mkdir -p $(DESTDIR)/opt/local/etc/{cron.daily,init.d,logrotate.d}
+	$(INSTALL_DATA) $(srcdir)/yum.conf $(DESTDIR)/opt/local/etc/yum.conf
+	$(INSTALL_SCRIPT) $(srcdir)/yum.cron $(DESTDIR)/opt/local/etc/cron.daily/yum.cron
+	$(INSTALL_SCRIPT) $(srcdir)/yum.init $(DESTDIR)/opt/local/etc/init.d/yum
+	$(INSTALL_DATA) $(srcdir)/yum.logrotate $(DESTDIR)/opt/local/etc/logrotate.d/yum
 
 uninstall:
-	$(RM) /etc/yum.conf
+	$(RM) /opt/local/etc/yum.conf
 
 
 clean:
