diff -uNr ../yum-1.0.1/bin/yum ./bin/yum
--- ../yum-1.0.1/bin/yum	Mon May 19 22:55:00 2003
+++ ./bin/yum	Sat May 31 18:32:23 2003
@@ -18,12 +18,13 @@
 
 import sys
 import os
-sys.path.insert(1,'/usr/share/yum')
+sys.path.insert(1,'/opt/local/share/yum')
+sys.path.insert(1,'/opt/local/lib/python2.2/site-packages')
 import yummain
 import yumlock
 
 uid = os.geteuid()
-lockfile = '/var/run/yum.pid'
+lockfile = '/opt/local/var/run/yum.pid'
 mypid = str(os.getpid())
 
 if uid == 0:
