--- ../yum-1.0.1/clientStuff.py	Mon May 19 22:54:59 2003
+++ clientStuff.py	Mon Jun  2 16:17:10 2003
@@ -152,15 +152,17 @@
     return 0
 
 def openrpmdb(option=0, dbpath=None):
-    dbpath = "/var/lib/rpm/"
-    rpm.addMacro("_dbpath", dbpath)
+    for dbpath in ["/opt/local/var/lib/rpm/", "/var/lib/rpm/"]:
+        if os.path.exists(dbpath + "Packages"):
+            rpm.addMacro("_dbpath", dbpath)
 
-    try:
-        db = rpm.opendb(option)
-    except rpm.error, e:
-        raise RpmError(_("Could not open RPM database for reading. Perhaps it is already in use?"))
-    
-    return db
+            try:
+                db = rpm.opendb(option)
+            except rpm.error, e:
+                raise RpmError(_("Could not open RPM database for reading.  Perhaps it is already in use?"))
+            return db
+ 
+    raise RpmError(_("Could not find RPM database!"))
 
 
 def rpmdbNevralLoad(nevral):
