--- ../yum-1.0.1/etc/yum.conf	Mon May 19 22:55:00 2003
+++ etc/yum.conf	Mon Jun  2 18:51:00 2003
@@ -1,16 +1,10 @@
 [main]
-cachedir=/var/cache/yum
+cachedir=/opt/local/var/cache/yum
 debuglevel=2
-logfile=/var/log/yum.log
+logfile=/opt/local/var/log/yum.log
 pkgpolicy=newest
+distroverpkg=xnu
 
-[base]
-name=Red Hat Linux $releasever base
-baseurl=http://mirror.dulug.duke.edu/pub/yum-repository/redhat/$releasever/$basearch/
-
-
-[updates]
-name=Red Hat Linux $releasever updates
-baseurl=http://mirror.dulug.duke.edu/pub/yum-repository/redhat/updates/$releasever/
-
-
+#[example]
+#name=Example Site
+#baseurl=http://www.example.com/$basearch/
