diff -uNr ../yum-1.0.1/config.py ./config.py
--- ../yum-1.0.1/config.py	Mon May 19 22:54:59 2003
+++ ./config.py	Sat May 31 18:32:23 2003
@@ -42,7 +42,7 @@
 
 
 class yumconf:
-    def __init__(self, configfile = '/etc/yum.conf'):
+    def __init__(self, configfile = '/opt/local/etc/yum.conf'):
         self.cfg = yumConfigParser()
         (s,b,p,q,f,o) = urlparse.urlparse(configfile)
         if s in ('http', 'ftp','file'):
@@ -69,9 +69,9 @@
         self.excludes=[]
         
         #defaults
-        self.cachedir = '/var/cache/yum'
+        self.cachedir = '/opt/local/var/cache/yum'
         self.debuglevel = 2
-        self.logfile = '/var/log/yum.log'
+        self.logfile = '/opt/local/var/log/yum.log'
         self.pkgpolicy = 'newest'
         self.gpghome = '/root/.gnupg'
         self.gpgkeyring = None
