diff -uNr ../yum-1.0.1/archwork.py ./archwork.py
--- ../yum-1.0.1/archwork.py	Mon May 19 22:54:59 2003
+++ ./archwork.py	Sat May 31 18:33:49 2003
@@ -35,6 +35,10 @@
         arch = 'alpha'
     if re.search('ppc', arch):
         arch = 'ppc'
+    if re.search('Power Macintosh', arch):
+        arch = 'ppc'
+    if re.search('powerpc', arch):
+        arch = 'ppc'
     return arch
 
 def betterarch(arch1, arch2):
@@ -67,10 +71,10 @@
 
 def compatArchList():
     archdict = {}
-    archdict['i386']=['i386','i486','i586','i686','athlon','noarch']
+    archdict['i386']=['i386','i486','i586','i686','athlon','noarch','fat']
     archdict['alpha']=['alpha','alphaev6','noarch']
     archdict['sparc']=['sparc','sparc64','noarch','sun4c','sun4u','sun4d','sun4m','sparcv9']
-    archdict['ppc']=['ppc','noarch','ppc64','powerpc','powerppc','osfmach3_ppc','ppciseries','ppcpseries','rs6000']
+    archdict['ppc']=['ppc','noarch','ppc64','powerpc','powerppc','osfmach3_ppc','ppciseries','ppcpseries','rs6000','Power Macintosh','fat']
     archdict['ia64']=['ia64','noarch', 'i686']
     archdict['s390']=['noarch','s390']
     archdict['s390x']=['noarch','s390','s390x']
