--- libcryptui/Makefile.am.orig	2013-05-13 17:52:22.000000000 -0700
+++ libcryptui/Makefile.am	2013-05-13 17:52:52.000000000 -0700
@@ -97,6 +97,9 @@ BUILT_SOURCES = \
 BUILT_GIRSOURCES =
 
 if HAVE_INTROSPECTION
+
+_gir_libtool = $(if $(findstring MINGW32,$(shell uname -s)),--libtool="$(top_builddir)/libtool",$(if $(LIBTOOL),--libtool="$(LIBTOOL)"))
+
 CryptUI-@LIBCRYPTUI_MAJOR@.gir: $(INTROSPECTION_SCANNER) libcryptui.la
 	$(QUIET_GEN)$(INTROSPECTION_SCANNER) -v \
 	--namespace CryptUI --nsversion=@LIBCRYPTUI_MAJOR@ \
@@ -104,6 +107,7 @@ CryptUI-@LIBCRYPTUI_MAJOR@.gir: $(INTROS
 	$(INCLUDES) \
 	--include=Gtk-2.0 \
 	--library=libcryptui.la \
+	$(_gir_libtool) \
     --pkg gtk+-2.0 \
     --output $@ \
 	$(inc_HEADERS) \
