--- gkr/seahorse-gkr-backend.c.orig	2013-09-18 10:09:34.000000000 -0700
+++ gkr/seahorse-gkr-backend.c	2013-09-18 10:11:32.000000000 -0700
@@ -162,7 +162,7 @@
 	SeahorseGkrBackend *self = SEAHORSE_GKR_BACKEND (user_data);
 	GError *error = NULL;
 
-	self->service = secret_service_new_finish (result, &error);
+	self->service = secret_service_open_finish (result, &error);
 	if (error == NULL) {
 		g_signal_connect (self->service, "notify::collections", G_CALLBACK (on_notify_collections), self);
 		refresh_collection (self);
@@ -185,7 +185,7 @@
 
 	G_OBJECT_CLASS (seahorse_gkr_backend_parent_class)->constructed (obj);
 
-	secret_service_new (my_secret_service_get_type (), NULL, SECRET_SERVICE_OPEN_SESSION,
+	secret_service_open (my_secret_service_get_type (), NULL, SECRET_SERVICE_OPEN_SESSION,
 	                    NULL, on_service_new, g_object_ref (self));
 }
 
