--- ./CMakeLists.txt.orig	2023-11-11 02:06:57
+++ ./CMakeLists.txt	2023-11-15 13:09:46
@@ -50,11 +50,6 @@
 find_package(folly CONFIG REQUIRED)
 include_directories(${FOLLY_INCLUDE_DIR})
 
-find_package(GMock MODULE REQUIRED)
-include_directories(${GMOCK_INCLUDEDIR} ${LIBGMOCK_INCLUDE_DIR})
-include(GoogleTest)
-enable_testing()
-
 add_subdirectory(eden/common/os)
 add_subdirectory(eden/common/utils)
 
--- eden/common/utils/CMakeLists.txt.orig	2023-11-11 02:06:57
+++ eden/common/utils/CMakeLists.txt	2023-11-15 13:06:08
@@ -36,5 +36,3 @@
   FILES ${utils_headers}
   DESTINATION ${INCLUDE_INSTALL_DIR}/eden/common/utils
 )
-
-add_subdirectory(test)
--- ./eden/common/os/CMakeLists.txt.orig	2023-11-15 13:27:03
+++ ./eden/common/os/CMakeLists.txt	2023-11-15 13:27:12
@@ -41,5 +41,3 @@
   FILES ${os_headers}
   DESTINATION ${INCLUDE_INSTALL_DIR}/eden/common/os
 )
-
-add_subdirectory(test)
