--- src/Makevars.in.orig	2023-02-22 04:39:49.000000000 +0700
+++ src/Makevars.in	2023-03-21 02:49:00.000000000 +0700
@@ -1,5 +1,5 @@
 PKG_CPPFLAGS=-DRCPP_USE_UNWIND_PROTECT -DRCPP_NO_RTTI -DRCPP_NO_SUGAR -I. @ZSTD_INCLUDE_PATH@ @LZ4_INCLUDE_PATH@ @SIMD_FLAG@
-PKG_LIBS=-L. -lpthread @ZSTD_LIBS@ @LZ4_LIBS@
+PKG_LIBS=-L. -lpthread -latomic @ZSTD_LIBS@ @LZ4_LIBS@
 
 # LIBZSTD = ZSTD/common/debug.o ZSTD/common/entropy_common.o ZSTD/common/error_private.o ZSTD/common/fse_decompress.o ZSTD/common/pool.o ZSTD/common/threading.o ZSTD/common/xxhash.o ZSTD/common/zstd_common.o ZSTD/compress/fse_compress.o ZSTD/compress/hist.o ZSTD/compress/huf_compress.o ZSTD/compress/zstd_compress.o ZSTD/compress/zstd_double_fast.o ZSTD/compress/zstd_fast.o ZSTD/compress/zstd_lazy.o ZSTD/compress/zstd_ldm.o ZSTD/compress/zstdmt_compress.o ZSTD/compress/zstd_opt.o ZSTD/decompress/huf_decompress.o ZSTD/decompress/zstd_ddict.o ZSTD/decompress/zstd_decompress_block.o ZSTD/decompress/zstd_decompress.o ZSTD/dictBuilder/cover.o ZSTD/dictBuilder/divsufsort.o ZSTD/dictBuilder/fastcover.o ZSTD/dictBuilder/zdict.o
 LIBZSTD = ZSTD/zstd.o
