--- CMakeLists.txt.orig	2022-10-06 07:27:29.000000000 +0800
+++ CMakeLists.txt	2023-05-24 20:10:23.000000000 +0800
@@ -14,7 +14,7 @@
 cmake_minimum_required(VERSION 3.7)
 
 # Has to be set before `project()`, and ignored on non-macos:
-set(CMAKE_OSX_DEPLOYMENT_TARGET 10.12 CACHE STRING "macOS deployment target (Apple clang only)")
+set(CMAKE_OSX_DEPLOYMENT_TARGET @MACOS@ CACHE STRING "macOS deployment target (Apple clang only)")
 
 project(liboxenmq
     VERSION 1.2.14
