--- project.pro.orig	2009-03-29 08:25:15.000000000 -0500
+++ project.pro	2009-12-31 17:08:44.000000000 -0600
@@ -315,7 +315,11 @@
 	HTMLDIR = $${DOCDIR}
 }
 
-target.path      = $${PREFIX}/bin
+mac {
+	target.path      = $${APPLICATIONS_DIR}
+} else {
+	target.path      = $${PREFIX}/bin
+}
 menu.files       = qtpfsgui.desktop
 menu.path        = $${PREFIX}/share/applications
 icon.files       = images/qtpfsgui.png
