--- Makefile.PL.orig	2017-06-21 06:36:09.000000000 -0700
+++ Makefile.PL	2017-06-21 06:36:39.000000000 -0700
@@ -10,7 +10,7 @@
 
     # returns a reference to anonymous hash which is then interpreted as
     # additional options to the WriteMakeFile
-    $options = require "pam.cfg";
+    $options = require "./pam.cfg";
 
     if ( $Config{'osname'} eq 'solaris' && $Config{'osvers'} eq '2.6') {
       print "Adding a workaround for a bug in the Solaris 2.6 pam library\n";
