--- CMakeLists.txt.orig	2012-02-03 11:03:32.000000000 -0600
+++ CMakeLists.txt	2012-02-03 11:03:47.000000000 -0600
@@ -242,7 +242,6 @@
     qhull_TARGETS 
         ${qhull_CPP} ${qhull_STATIC} ${qhull_STATICP} ${qhull_SHARED} ${qhull_SHAREDP}
         qhull rbox qconvex qdelaunay qvoronoi qhalf
-        user_eg user_eg2 user_eg3 testqset
 )
 
 
@@ -342,33 +341,21 @@
 
 add_executable(qhull ${qhull_SOURCES})
 target_link_libraries(qhull ${qhull_STATIC})
-set_target_properties(qhull PROPERTIES 
-    VERSION ${qhull_VERSION})
 
 add_executable(rbox ${rbox_SOURCES})
 target_link_libraries(rbox ${qhull_STATIC})
-set_target_properties(rbox PROPERTIES
-    VERSION ${qhull_VERSION})
 
 add_executable(qconvex ${qconvex_SOURCES})
 target_link_libraries(qconvex ${qhull_STATIC})
-set_target_properties(qconvex PROPERTIES
-    VERSION ${qhull_VERSION})
 
 add_executable(qdelaunay ${qdelaunay_SOURCES})
 target_link_libraries(qdelaunay ${qhull_STATIC})
-set_target_properties(qdelaunay PROPERTIES
-    VERSION ${qhull_VERSION})
 
 add_executable(qvoronoi ${qvoronoi_SOURCES})
 target_link_libraries(qvoronoi ${qhull_STATIC})
-set_target_properties(qvoronoi PROPERTIES
-    VERSION ${qhull_VERSION})
 
 add_executable(qhalf ${qhalf_SOURCES})
 target_link_libraries(qhalf ${qhull_STATIC})
-set_target_properties(qhalf PROPERTIES
-    VERSION ${qhull_VERSION})
 
 # ---------------------------------------
 # Define testqset linked to qset.o and mem.o
