--- src/CMakeLists.txt.orig	2010-01-19 08:44:09.000000000 -0600
+++ src/CMakeLists.txt	2010-01-19 09:23:29.000000000 -0600
@@ -23,8 +23,21 @@
 
 file(GLOB qhull_hdr *.h)
 
-add_library(qhull ${qhull_src})
+add_library(qhull SHARED ${qhull_src})
 target_link_libraries(qhull m)
+if(UNIX)
+	if(APPLE)
+		set_target_properties(qhull PROPERTIES 
+			INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+	else(APPLE)
+		set_target_properties(qhull PROPERTIES 
+			INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib"
+			INSTALL_RPATH_USE_LINK_PATH TRUE
+			BUILD_WITH_INSTALL_RPATH FALSE)
+	endif(APPLE)
+endif(UNIX)
+add_library(qhullstatic STATIC ${qhull_src})
+set_property(TARGET qhullstatic PROPERTY OUTPUT_NAME "qhull")
 
 set(qhullcmd_SOURCES  unix.c)
 set(rbox_SOURCES  rbox.c)
@@ -36,6 +49,7 @@
 
 add_executable(qhullcmd ${qhullcmd_SOURCES})
 target_link_libraries(qhullcmd qhull)
+set_property(TARGET qhullcmd PROPERTY OUTPUT_NAME "qhull")
 
 add_executable(rbox ${rbox_SOURCES})
 target_link_libraries(rbox qhull)
@@ -52,3 +66,8 @@
 add_executable(qhalf ${qhalf_SOURCES})
 target_link_libraries(qhalf qhull)
 
+install(TARGETS qhull qhullstatic qhullcmd rbox qconvex qdelaunay qvoronoi qhalf
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+	ARCHIVE DESTINATION lib)
+install(FILES libqhull.h DESTINATION include)
