diff -rubw src/libqhullcpp.orig/Coordinates.h src/libqhullcpp/Coordinates.h
--- src/libqhullcpp.orig/Coordinates.h	2012-02-03 11:06:38.000000000 -0600
+++ src/libqhullcpp/Coordinates.h	2012-02-03 11:09:06.000000000 -0600
@@ -148,7 +148,7 @@
 
     private:
         std::vector<coordT>::iterator i;
-        friend class    const_iterator;
+        friend class    Coordinates::const_iterator;
 
     public:
         typedef std::random_access_iterator_tag  iterator_category;
diff -rubw src/libqhullcpp.orig/QhullLinkedList.h src/libqhullcpp/QhullLinkedList.h
--- src/libqhullcpp.orig/QhullLinkedList.h	2012-02-03 11:06:38.000000000 -0600
+++ src/libqhullcpp/QhullLinkedList.h	2012-02-03 11:09:08.000000000 -0600
@@ -112,7 +112,7 @@
 
     private:
         T               i;
-        friend class    const_iterator;
+        friend class    QhullLinkedList::const_iterator;
 
     public:
         typedef std::bidirectional_iterator_tag  iterator_category;
diff -rubw src/libqhullcpp.orig/QhullPointSet.h src/libqhullcpp/QhullPointSet.h
--- src/libqhullcpp.orig/QhullPointSet.h	2012-02-03 11:06:38.000000000 -0600
+++ src/libqhullcpp/QhullPointSet.h	2012-02-03 11:09:10.000000000 -0600
@@ -111,7 +111,7 @@
 
     // before const_iterator for conversion with comparison operators
     class iterator {
-        friend class    const_iterator;
+        friend class    QhullPointSet::const_iterator;
 
     private:
         coordT        **i;
