diff --git FDTD/excitation.cpp FDTD/excitation.cpp
index ba0ac95..970500a 100644
--- FDTD/excitation.cpp
+++ FDTD/excitation.cpp
@@ -19,6 +19,7 @@
 #include "tools/useful.h"
 #include <iostream>
 #include <fstream>
+#include <clocale>
 #include "fparser.hh"
 #include "excitation.h"
 
diff --git FDTD/extensions/operator_ext_upml.cpp FDTD/extensions/operator_ext_upml.cpp
index 6683459..4a5bdc0 100644
--- FDTD/extensions/operator_ext_upml.cpp
+++ FDTD/extensions/operator_ext_upml.cpp
@@ -20,6 +20,7 @@
 #include "engine_ext_upml.h"
 #include "tools/array_ops.h"
 #include "fparser.hh"
+#include <clocale>
 
 using namespace std;
 
diff --git nf2ff/nf2ff.cpp nf2ff/nf2ff.cpp
index 883b622..7f1f30b 100644
--- nf2ff/nf2ff.cpp
+++ nf2ff/nf2ff.cpp
@@ -30,6 +30,7 @@
 #include <complex>
 #include <iostream>
 #include <sstream>
+#include <clocale>
 
 //external libs
 #include "tinyxml.h"
diff --git openems.cpp openems.cpp
index d9de070..3d640de 100644
--- openems.cpp
+++ openems.cpp
@@ -45,6 +45,7 @@
 #include <hdf5.h>            // only for H5get_libversion()
 #include <boost/version.hpp> // only for BOOST_LIB_VERSION
 #include <vtkVersion.h>
+#include <clocale>
 
 //external libs
 #include "tinyxml.h"
