--- libsrc/zlib/gzguts.h	2013-12-06 22:44:34.000000000 +0800
+++ libsrc/zlib/gzguts.h	2023-07-31 15:41:10.000000000 +0800
@@ -19,6 +19,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 #include "zlib.h"
 #ifdef STDC
 #  include <string.h>

--- libsrc/Wi/bif_search_excerpt.c	2013-12-09 20:28:32.000000000 +0800
+++ libsrc/Wi/bif_search_excerpt.c	2023-07-31 15:57:06.000000000 +0800
@@ -24,6 +24,7 @@
  */
 
 #ifndef SELF_INCLUDE
+#include <math.h>
 #include "http.h" /* for dks_esc_write and DKS_ESC_... */
 #include "langfunc.h"
 #include "multibyte.h"

