--- configure.orig	2006-07-16 11:48:28.000000000 +1000
+++ configure	2011-08-20 09:11:58.000000000 +1000
@@ -1664,7 +1664,7 @@
 
 
 # Define the identity of the package.
- PACKAGE="Critical_Mass"
+ PACKAGE="criticalmass"
  VERSION=1.0.2
 
 
@@ -3864,13 +3864,13 @@
     LIBS="-L/usr/local/lib -L/usr/X11R6/lib $LIBS"
     CXXFLAGS="$CXXFLAGS -I/usr/X11R6/include"
 
-echo "$as_me:$LINENO: checking for main in -lpng12" >&5
-echo $ECHO_N "checking for main in -lpng12... $ECHO_C" >&6
-if test "${ac_cv_lib_png12_main+set}" = set; then
+echo "$as_me:$LINENO: checking for main in -lpng" >&5
+echo $ECHO_N "checking for main in -lpng... $ECHO_C" >&6
+if test "${ac_cv_lib_png_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng12  $LIBS"
+LIBS="-lpng  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -3908,25 +3908,25 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_png12_main=yes
+  ac_cv_lib_png_main=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_png12_main=no
+ac_cv_lib_png_main=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_png12_main" >&5
-echo "${ECHO_T}$ac_cv_lib_png12_main" >&6
-if test $ac_cv_lib_png12_main = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_png_main" >&5
+echo "${ECHO_T}$ac_cv_lib_png_main" >&6
+if test $ac_cv_lib_png_main = yes; then
   cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBPNG12 1
+#define HAVE_LIBPNG 1
 _ACEOF
 
-  LIBS="-lpng12 $LIBS"
+  LIBS="-lpng $LIBS"
 
 else
   { { echo "$as_me:$LINENO: error: libpng is needed" >&5
