--- theseus_align.orig	2020-03-03 21:42:45.000000000 -0500
+++ theseus_align	2020-03-03 21:43:11.000000000 -0500
@@ -44,13 +44,13 @@
 CAT="/bin/cat";
 SED="/usr/bin/sed";
 TEE="/usr/bin/tee";
-theseus="/usr/local/bin/theseus"; # where to find the THESEUS binary executable
+theseus="@@PREFIX@@/bin/theseus"; # where to find the THESEUS binary executable
 fastafile="theseus.fasta";
 filemapfile="theseus.filemap";
 alignmentfile="theseus.aln";
 
 # for MUSCLE -- http://www.drive5.com/muscle/
-alignprog="/usr/local/bin/muscle";
+alignprog="@@PREFIX@@/bin/muscle";
 align_cmd="${alignprog} -maxiters 32 -in ${fastafile} -out ${alignmentfile} -clwstrict";
 
 # for PROBCONS -- http://probcons.stanford.edu/
