--- net6emuconf.orig	2002-08-10 04:40:28.000000000 -0500
+++ net6emuconf	2011-11-18 23:32:24.000000000 -0600
@@ -101,14 +101,6 @@
 
 
 
-if [ `uname` = "Darwin" ]; then
-  case `uname -r` in [012345]*)   # up to Mac OS X 10.1.x / Darwin 5.x
-    test_getaddrinfo=0            #  getaddrinfo() is somewhat broken
-    echo "Skipping check for getaddrinfo() because it is known to be buggy on this"
-    echo "system (Mac OS X up to 10.1.x alias Darwin up to 5.x). It will be emulated."
-    ;;
-  esac
-fi
 
 if [ $test_getaddrinfo -eq 1 ]; then
   echo "Checking getaddrinfo()..."
