--- config.mk.orig	2017-08-26 12:36:07.000000000 -0700
+++ config.mk	2017-08-26 12:43:42.000000000 -0700
@@ -2,11 +2,11 @@
 VERSION = 4.7
 
 # paths
-PREFIX = /usr/local
+PREFIX = @@MACPORTS_PREFIX@@
 MANPREFIX = ${PREFIX}/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = @@MACPORTS_PREFIX@@/include/X11
+X11LIB = @@MACPORTS_PREFIX@@/lib
 
 # Xinerama, comment if you don't want it
 XINERAMALIBS  = -lXinerama
@@ -14,7 +14,7 @@
 
 # freetype
 FREETYPELIBS = -lfontconfig -lXft
-FREETYPEINC = /usr/include/freetype2
+FREETYPEINC = @@MACPORTS_PREFIX@@/include/freetype2
 # OpenBSD (uncomment)
 #FREETYPEINC = ${X11INC}/freetype2
 
