--- ScriptExecController.m.orig	2014-01-14 09:42:58.000000000 -0800
+++ ScriptExecController.m	2014-01-14 09:43:56.000000000 -0800
@@ -55,7 +55,7 @@
     NSEnumerator *enumerator = [files objectEnumerator];
     id filename;
 
-    args[i++] = strdup([[[NSBundle mainBundle] pathForResource:@"script"
+    args[i++] = strdup([[[NSBundle mainBundle] pathForResource:@"GIMP"
                                                ofType:nil] UTF8String]);
     while (filename = [enumerator nextObject]) {
         args[i++] = strdup([filename UTF8String]);
