--- kdenetwork-3.5.8/krfb/libvncserver/rfb.h	2005-09-10 17:20:13.000000000 +0900
+++ kdenetwork-3.5.8-new/krfb/libvncserver/rfb.h	2008-05-04 17:59:50.000000000 +0900
@@ -38,9 +38,7 @@
 #include "keysym.h"
 
 /* TODO: this stuff has to go into autoconf */
-typedef unsigned char CARD8;
-typedef unsigned short CARD16;
-typedef unsigned int CARD32;
+#include <X11/Xmd.h>
 typedef CARD32 Pixel;
 /* typedef CARD32 KeySym; */
 #ifndef __osf__
