From 2f38eb376744ba51a4c3ee1235dcce0ee538f4e4 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 1 Aug 2023 21:39:52 +0800
Subject: [PATCH 2/2] Use EV_TRIGGER when NOTE_TRIGGER unavailable

---
 src/poller/iwn_poller.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git src/poller/iwn_poller.c src/poller/iwn_poller.c
index 0532ac2..5cf268c 100644
--- src/poller/iwn_poller.c
+++ src/poller/iwn_poller.c
@@ -680,7 +680,13 @@ void iwn_poller_poke(struct iwn_poller *p) {
   {
     struct kevent ev[] = {
       { p->fd, EVFILT_USER, EV_ADD | EV_ONESHOT },
+#if defined(NOTE_TRIGGER)
       { p->fd, EVFILT_USER, 0, NOTE_TRIGGER     }
+#elif defined(EV_TRIGGER)
+      { p->fd, EVFILT_USER, EV_TRIGGER, 0       }
+#else
+#error "Either NOTE_TRIGGER or EV_TRIGGER is required."
+#endif
     };
     if (kevent(p->fd, ev, sizeof(ev) / sizeof(ev[0]), 0, 0, 0) == -1) {
       iwlog_ecode_error3(iwrc_set_errno(IW_ERROR_ERRNO, errno));
