--- DESCRIPTION	2023-10-24 21:54:02.000000000 +0800
+++ DESCRIPTION	2023-10-26 20:18:39.000000000 +0800
@@ -14,7 +14,7 @@
 	S4Vectors (>= 0.27.12), IRanges (>= 2.23.9)
 Imports: methods, utils, tools, zlibbioc, BiocGenerics, S4Vectors, IRanges
 LinkingTo: S4Vectors, IRanges
-Suggests: Biostrings, drosophila2probe, RUnit
+Suggests: Biostrings, RUnit
 Collate: io-utils.R
 	RDS-random-access.R
 	SharedVector-class.R
