--- vlib/net/openssl/c.v.original	2023-02-07 11:21:08.000000000 +0000
+++ vlib/net/openssl/c.v	2023-02-07 11:25:02.000000000 +0000
@@ -14,20 +14,8 @@
 	#flag -lssl -lcrypto
 	#flag linux -ldl -lpthread
 	// MacPorts
-	#flag darwin -I/opt/local/include
-	#flag darwin -L/opt/local/lib
-	// Brew
-	#flag darwin -I/usr/local/opt/openssl/include
-	#flag darwin -L/usr/local/opt/openssl/lib
-	// brew on macos-12 (ci runner)
-	#flag darwin -I/usr/local/opt/openssl@3/include
-	#flag darwin -L/usr/local/opt/openssl@3/lib
-	// Brew arm64
-	#flag darwin -I /opt/homebrew/opt/openssl/include
-	#flag darwin -L /opt/homebrew/opt/openssl/lib
-	// Procursus
-	#flag darwin -I/opt/procursus/include
-	#flag darwin -L/opt/procursus/lib
+	#flag darwin -I@PREFIX@/include
+	#flag darwin -L@PREFIX@/lib
 }
 
 #include <openssl/rand.h> # Please install OpenSSL development headers
