--- vlib/net/openssl/c.v.original	2022-04-13 10:54:47.000000000 +0100
+++ vlib/net/openssl/c.v	2022-04-13 10:59:22.000000000 +0100
@@ -15,17 +15,8 @@
 #flag -lssl -lcrypto
 #flag linux -ldl -lpthread
 // MacPorts
-#flag darwin -I/opt/local/include
-#flag darwin -L/opt/local/lib
-// Brew
-#flag darwin -I/usr/local/opt/openssl/include
-#flag darwin -L/usr/local/opt/openssl/lib
-// Brew arm64
-#flag darwin -I /opt/homebrew/opt/openssl/include
-#flag darwin -L /opt/homebrew/opt/openssl/lib
-// Procursus
-#flag darwin -I/opt/procursus/include
-#flag darwin -L/opt/procursus/lib
+#flag darwin -I@PREFIX@/include
+#flag darwin -L@PREFIX@/lib
 //
 #include <openssl/rand.h> # Please install OpenSSL development headers
 #include <openssl/ssl.h>
