--- vlib/v/pkgconfig/pkgconfig.v.original	2022-04-23 12:58:33.000000000 +0100
+++ vlib/v/pkgconfig/pkgconfig.v	2022-04-23 13:50:41.000000000 +0100
@@ -5,17 +5,7 @@
 
 const (
 	default_paths = [
-		'/usr/local/lib/x86_64-linux-gnu/pkgconfig',
-		'/usr/local/lib64/pkgconfig',
-		'/usr/local/lib/pkgconfig',
-		'/usr/local/share/pkgconfig',
-		'/usr/lib/x86_64-linux-gnu/pkgconfig',
-		'/usr/lib/aarch64-linux-gnu/pkgconfig',
-		'/usr/lib64/pkgconfig',
-		'/usr/lib/pkgconfig',
-		'/usr/share/pkgconfig',
-		'/opt/homebrew/lib/pkgconfig', // Brew on macOS
-		'/usr/local/libdata/pkgconfig', // FreeBSD
+		'@PREFIX@/lib/pkgconfig'
 	]
 	version       = '0.3.2'
 )
