--- src/input/vcd/libcdio/_cdio_osx.c.orig	2008-04-18 02:48:58.000000000 +1000
+++ src/input/vcd/libcdio/_cdio_osx.c	2011-02-17 19:26:11.000000000 +1100
@@ -53,7 +53,7 @@ static const char _rcsid[] = "$Id: _cdio
 
 #include <mach/mach.h>
 #include <Carbon/Carbon.h>
-#include <IOKit/scsi-commands/SCSITaskLib.h>
+#include <IOKit/scsi/SCSITaskLib.h>
 #include <IOKit/IOCFPlugIn.h>
 #include <mach/mach_error.h>
 
@@ -72,7 +72,7 @@ static const char _rcsid[] = "$Id: _cdio
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
 #include <IOKit/IOBSD.h>
-#include <IOKit/scsi-commands/IOSCSIMultimediaCommandsDevice.h>
+#include <IOKit/scsi/IOSCSIMultimediaCommandsDevice.h>
 #include <IOKit/storage/IOCDTypes.h>
 #include <IOKit/storage/IODVDTypes.h>
 #include <IOKit/storage/IOMedia.h>
