--- src/xine-engine/Makefile.in.orig	2012-06-09 12:44:05.000000000 -0500
+++ src/xine-engine/Makefile.in	2012-07-21 06:16:59.000000000 -0500
@@ -607,7 +607,11 @@
 	  rm -f "$${dir}/so_locations"; \
 	done
 libxine-interface.la: $(libxine_interface_la_OBJECTS) $(libxine_interface_la_DEPENDENCIES) $(EXTRA_libxine_interface_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libxine_interface_la_LINK)  $(libxine_interface_la_OBJECTS) $(libxine_interface_la_LIBADD) $(LIBS)
+	if test -n "$(libxine_interface_la_OBJECTS)" ; then \
+	    $(AM_V_CCLD)$(libxine_interface_la_LINK)  $(libxine_interface_la_OBJECTS) $(libxine_interface_la_LIBADD) $(LIBS); \
+	else \
+	    touch $@; \
+	fi
 libxine.la: $(libxine_la_OBJECTS) $(libxine_la_DEPENDENCIES) $(EXTRA_libxine_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libxine_la_LINK) -rpath $(libdir) $(libxine_la_OBJECTS) $(libxine_la_LIBADD) $(LIBS)
 
