--- DESCRIPTION	2023-10-06 17:50:02.000000000 +0800
+++ DESCRIPTION	2023-11-05 22:06:11.000000000 +0800
@@ -5,7 +5,7 @@
 Authors@R: c(person("Nicolas", "Robette", role = c("aut", "cre"), email = "nicolas.robette@uvsq.fr"))
 Imports: descriptio (>= 1.2), FactoMineR, ggplot2, ggrepel, rlang
 Suggests: R.rsp, rmarkdown, knitr, rmdformats, TraMineR, sf, shiny,
-        miniUI, esquisse, rclipboard, factoextra, ade4
+        miniUI, rclipboard, factoextra, ade4
 Description: Many tools for Geometric Data Analysis (Le Roux & Rouanet (2005) <doi:10.1007/1-4020-2236-0>), such as MCA variants (Specific Multiple Correspondence Analysis, Class Specific Analysis), many graphical and statistical aids to interpretation (structuring factors, concentration ellipses, inductive tests, bootstrap validation, etc.) and multiple-table analysis (Multiple Factor Analysis, between- and inter-class analysis, Principal Component Analysis and Correspondence Analysis with Instrumental Variables, etc.).
 VignetteBuilder: R.rsp
 License: GPL (>= 2)
