--- autosetup/cc.orig	2013-02-16 22:48:58.000000000 +0100
+++ autosetup/cc.tcl	2013-02-16 22:49:52.000000000 +0100
@@ -493,12 +493,6 @@
 	}
 	lappend cmdline {*}$opts(-cflags)
 
-	switch -glob -- [get-define host] {
-		*-*-darwin* {
-			# Don't generate .dSYM directories
-			lappend cmdline -gstabs
-		}
-	}
 	lappend cmdline $src -o $tmp {*}$opts(-libs)
 
 	# At this point we have the complete command line and the
