--- ompl/src/ompl/control/ODESolver.h	2013-07-09 19:54:58.000000000 -0500
+++ ompl/src/ompl/control/ODESolver.h	2014-08-18 09:10:11.000000000 -0500
@@ -318,7 +317,11 @@
             {
                 ODESolver::ODEFunctor odefunc (ode_, control);
 
+#if BOOST_VERSION < 105600
                 odeint::controlled_runge_kutta< Solver > solver (odeint::default_error_checker<double>(maxError_, maxEpsilonError_));
+#else
+                typename boost::numeric::odeint::result_of::make_controlled< Solver >::type solver = make_controlled( 1.0e-6 , 1.0e-6 , Solver() );
+#endif
                 odeint::integrate_adaptive (solver, odefunc, state, 0.0, duration, intStep_);
             }
 
