--- CMakeLists.txt	2015-10-28 20:18:02.000000000 -0500
+++ CMakeLists.txt	2015-11-01 14:42:31.000000000 -0600
@@ -27,6 +27,7 @@
 endif (MSVC OR MSVC90 OR MSVC10)
 
 set(CMAKE_MODULE_PATH
+    "${CMAKE_MODULE_PATH}"
     "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules"
     "${CMAKE_CURRENT_SOURCE_DIR}/ompl/CMakeModules")
 include(GNUInstallDirs)
@@ -177,7 +178,7 @@
 
 if (EIGEN3_FOUND)
     set(OMPL_HAVE_EIGEN3 1)
-    include_directories("${EIGEN3_INCLUDE_DIRS}")
+    include_directories("${EIGEN3_INCLUDE_DIR}")
 endif()
 
 if (FLANN_FOUND)
