diff -r 72ff4ee4fea6 CMakeLists.txt
--- ompl/CMakeLists.txt	Wed Oct 28 20:04:41 2015 -0500
+++ ompl/CMakeLists.txt	Fri Oct 30 15:48:19 2015 -0500
@@ -23,7 +23,8 @@
     set(MSVC ON)
 endif (MSVC OR MSVC90 OR MSVC10)
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
+message(${CMAKE_MODULE_PATH})
 include(GNUInstallDirs)
 include(CompilerSettings)
 include(OMPLVersion)
@@ -112,7 +113,7 @@
 find_package(Eigen3 QUIET)
 if (EIGEN3_FOUND)
     set(OMPL_HAVE_EIGEN3 1)
-    include_directories("${EIGEN3_INCLUDE_DIRS}")
+    include_directories("${EIGEN3_INCLUDE_DIR}")
 endif()
 
 # MORSE is only needed for Modular OpenRobots Simulation Engine bindings
diff -r 72ff4ee4fea6 CMakeModules/FindEigen3.cmake
--- ompl/CMakeModules/FindEigen3.cmake	Wed Oct 28 20:04:41 2015 -0500
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,8 +0,0 @@
-include(FindPackageHandleStandardArgs)
-
-find_package(PkgConfig)
-if(PKG_CONFIG_FOUND)
-    pkg_check_modules(EIGEN3 eigen3)
-endif()
-
-find_package_handle_standard_args(Eigen3 DEFAULT_MSG EIGEN3_INCLUDE_DIRS)
diff -r 72ff4ee4fea6 CMakeModules/PythonBindingsUtils.cmake
--- ompl/CMakeModules/PythonBindingsUtils.cmake	Wed Oct 28 20:04:41 2015 -0500
+++ ompl/CMakeModules/PythonBindingsUtils.cmake	Fri Oct 30 15:48:19 2015 -0500
@@ -80,7 +80,7 @@
         PYTHONPATH="${PROJECT_BINARY_DIR}/pyplusplus/lib/python${PYTHON_VERSION}/site-packages:$ENV{PYTHONPATH}"
         ${PYTHON_EXEC}
         "${CMAKE_CURRENT_SOURCE_DIR}/generate_bindings.py" "${module}"
-        "1>${CMAKE_BINARY_DIR}/pyplusplus_${module}.log" "2>&1"
+        "|tee" "${CMAKE_BINARY_DIR}/pyplusplus_${module}.log" "2>&1"
         COMMAND ${CMAKE_COMMAND} -D "PATH=${dir}/bindings/${module}"
         -P "${OMPL_CMAKE_UTIL_DIR}/workaround_for_gccxml_bug.cmake"
         COMMAND ${CMAKE_COMMAND} ${CMAKE_BINARY_DIR}
diff -r 72ff4ee4fea6 py-bindings/bindings_generator.py.in
--- ompl/py-bindings/bindings_generator.py.in	Wed Oct 28 20:04:41 2015 -0500
+++ ompl/py-bindings/bindings_generator.py.in	Fri Oct 30 15:48:19 2015 -0500
@@ -112,7 +112,7 @@
         @dep name of another module this module depends on"""
         module_builder.set_logger_level( logging.INFO )
         candidate_include_paths = [ "@OMPL_INCLUDE_DIR@", "@OMPLAPP_INCLUDE_DIR@",
-            "@PYTHON_INCLUDE_DIRS@", "@Boost_INCLUDE_DIR@", "@ASSIMP_INCLUDE_DIRS@", "@ODEINT_INCLUDE_DIR@", "@Eigen_INCLUDE_DIRS@"]
+            "@PYTHON_INCLUDE_DIRS@", "@Boost_INCLUDE_DIR@", "@ASSIMP_INCLUDE_DIRS@", "@ODEINT_INCLUDE_DIR@", "@EIGEN3_INCLUDE_DIR@"]
 
         # Adding standard windows headers
         if platform == 'win32':
