--- configure.orig	2009-03-25 14:43:25.000000000 +0100
+++ configure	2009-03-25 15:01:53.000000000 +0100
@@ -6308,6 +6308,13 @@
 echo "$as_me: updating makefile.platform" >&6;}
 sed -i -e "s/^PLATFORM.*/PLATFORM = $PLATFORM/" makefile.platform
 
+##### Macports patches #####
+echo
+echo "Patching makefile.config.gcc for Macports..."
+mv makefile.config.gcc makefile.config.gcc.orig
+/usr/bin/perl -e "while (<>){ s/-Wl,-framework,Cocoa/-framework Cocoa/; print; }" < makefile.config.gcc.orig > makefile.config.gcc
+##### end of patches #####
+
 # Output the configuration summary
 echo
 echo "Configure summary:"
