--- configs/macos_config.janet	2022-06-11 22:57:59.000000000 +1000
+++ configs/macos_config.janet	2022-06-13 21:51:53.000000000 +1000
@@ -1,4 +1,4 @@
-(def prefix "/usr/local")
+(def prefix "@@MACPORTS_PREFIX@@")

 (def config
   {:ar "ar"
@@ -26,7 +26,7 @@
    :local false
    :manpath (string prefix "/share/man/man1")
    :modext ".so"
-   :modpath (string prefix "/lib/janet")
+   :modpath (string prefix "/lib/jpm")
    :nocolor false
    :optimize 2
    :pkglist "https://github.com/janet-lang/pkgs.git"
