--- configs/macos_config.janet.orig	2022-06-16 01:46:34
+++ configs/macos_config.janet	2023-08-08 22:48:50
@@ -1,4 +1,4 @@
-(def prefix "/usr/local")
+(def prefix "@@MACPORTS_PREFIX@@")
 
 (def config 
   {:ar "ar"
@@ -26,7 +26,7 @@
    :local false
    :manpath (string prefix "/share/man/man1")
    :modext ".so"
-   :modpath (string prefix "/lib/janet")
+   :modpath (string prefix "/lib/jpm")
    :nocolor false
    :optimize 2
    :pkglist "https://github.com/janet-lang/pkgs.git"
