--- meson.build.orig	2018-06-13 09:49:45.000000000 -0700
+++ meson.build	2018-06-25 14:49:23.000000000 -0700
@@ -246,7 +246,7 @@
 sse2_cflags = []
 if get_option('sse2')
   sse_prog = '''
-#if defined(__GNUC__) && (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 9))
+#if !defined(__APPLE__) && defined(__GNUC__) && (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 9))
 # if !defined(__amd64__) && !defined(__x86_64__)
 #   error "Need GCC >= 4.9 for SSE2 intrinsics on x86"
 # endif
