--- meson.build.orig	2023-02-07 18:04:51.000000000 -0800
+++ meson.build	2023-02-07 18:04:59.000000000 -0800
@@ -293,11 +293,7 @@
 sse2_cflags = []
 if get_option('sse2')
   sse_prog = '''
-#if defined(__GNUC__)
-# if !defined(__amd64__) && !defined(__x86_64__)
-#   error "SSE2 intrinsics are only available on x86_64"
-# endif
-#elif defined (_MSC_VER) && !defined (_M_X64) && !defined (_M_AMD64)
+#if defined (_MSC_VER) && !defined (_M_X64) && !defined (_M_AMD64)
 # error "SSE2 intrinsics not supported on x86 MSVC builds"
 #endif
 #if defined(__SSE__) || (_M_X64 > 0)
