--- scons/scons-local-3.0.1/SCons/Tool/javac.py.orig	2023-11-15 16:08:31.000000000 +0000
+++ scons/scons-local-3.0.1/SCons/Tool/javac.py	2023-11-15 16:10:14.000000000 +0000
@@ -34,6 +34,7 @@
 
 import os
 import os.path
+from collections import OrderedDict
 
 import SCons.Action
 import SCons.Builder
@@ -70,7 +71,7 @@
         if isinstance(entry, SCons.Node.FS.File):
             slist.append(entry)
         elif isinstance(entry, SCons.Node.FS.Dir):
-            result = SCons.Util.OrderedDict()
+            result = OrderedDict()
             dirnode = entry.rdir()
             def find_java_files(arg, dirpath, filenames):
                 java_files = sorted([n for n in filenames
