--- redis.conf.orig	2020-05-14 09:42:51.094603573 -0400
+++ redis.conf	2020-05-14 09:50:15.671701491 -0400
@@ -224,7 +224,7 @@
 #
 # Creating a pid file is best effort: if Redis is not able to create it
 # nothing bad happens, the server will start and run normally.
-pidfile /var/run/redis_6379.pid
+pidfile @PREFIX@/var/run/redis_6379.pid

 # Specify the server verbosity level.
 # This can be one of:
@@ -237,7 +237,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Redis to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile @PREFIX@/var/log/redis.log

 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
@@ -342,7 +342,7 @@
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir @PREFIX@/var/db/redis

 ################################# REPLICATION #################################

--- src/Makefile.orig	2020-05-14 12:03:25.183608552 -0400
+++ src/Makefile	2020-05-14 12:03:51.752513404 -0400
@@ -103,8 +103,6 @@
 ifeq ($(uname_S),Darwin)
 	# Darwin
 	FINAL_LIBS+= -ldl
-	OPENSSL_CFLAGS=-I/usr/local/opt/openssl/include
-	OPENSSL_LDFLAGS=-L/usr/local/opt/openssl/lib
 else
 ifeq ($(uname_S),AIX)
         # AIX
@@ -182,8 +180,7 @@
 endif

 ifeq ($(BUILD_TLS),yes)
-    FINAL_CFLAGS+=-DUSE_OPENSSL $(OPENSSL_CFLAGS)
-    FINAL_LDFLAGS+=$(OPENSSL_LDFLAGS)
+    FINAL_CFLAGS+=-DUSE_OPENSSL
     FINAL_LIBS += ../deps/hiredis/libhiredis_ssl.a -lssl -lcrypto
 endif
