--- kbackgammon/Makefile.am	10 Dec 2002 20:31:41 -0000	1.1.1.4
+++ kbackgammon/Makefile.am	10 Dec 2002 20:56:04 -0000	1.7
@@ -9,6 +9,9 @@
 			kbgstatus.cpp
 kbackgammon_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
+kbackgammon_main.cpp: main.cpp
+	cat main.cpp > kbackgammon_main.cpp
+
 noinst_HEADERS = version.h kbg.h kbgboard.h kbgtextview.h kbgstatus.h
 
 SUBDIRS = pics sounds icons engines
@@ -22,9 +25,3 @@
 messages: rc.cpp	
 	LIST=`find . -name \*.h -o -name \*.hh -o -name \*.H -o -name \*.hxx -o -name \*.hpp -o -name \*.cpp -o -name \*.cc -o -name \*.cxx -o -name \*.ecpp -o -name \*.C`; \
 	$(XGETTEXT) $$LIST -o $(podir)/kbackgammon.pot
-
-dummy.cpp:
-	echo > dummy.cpp
-
-install-data-local:
-	rm -rf $(DESTDIR)$(kde_appsdir)/Games/kbackgammon.desktop
