--- CMakeLists.txt.orig	2009-12-20 23:45:08.000000000 +0100
+++ CMakeLists.txt	2009-12-20 23:48:36.000000000 +0100
@@ -464,34 +464,10 @@
     ADD_DEFINITIONS(-DGLU_TESS_CALLBACK_TRIPLEDOT)
 ENDIF(OSG_GLU_TESS_CALLBACK_TRIPLEDOT)
 
-# Platform specific:
-# (We can approach this one of two ways. We can try to FIND everything
-# and simply check if we found the packages before actually building
-# or we can hardcode the cases. The advantage of the former is that
-# packages that are installed on platforms that don't require them
-# will still get built (presuming no compatibility issues). But this
-# also means modules that are redundant may get built. For example,
-# OS X doesn't need GIF, JPEG, PNG, TIFF, etc because it uses QuickTime.
-# Also, it will clutter the CMake menu with "NOT_FOUND".
-# The downside to the latter is that it is harder to build those
-# potentially redundant modules.)
-
-# Image readers/writers depend on 3rd party libraries except for OS X which
-# can use Quicktime.
-IF(NOT APPLE)
-    FIND_PACKAGE(GIFLIB)
-    FIND_PACKAGE(JPEG)
-    FIND_PACKAGE(PNG)
-    FIND_PACKAGE(TIFF)
-
-    # QuickTime is required for OS X, but optional for Windows.
-    IF(WIN32)
-        FIND_PACKAGE(QuickTime)
-    ENDIF()
-
-ELSE()
-    FIND_PACKAGE(QuickTime)
-ENDIF()
+FIND_PACKAGE(GIFLIB)
+FIND_PACKAGE(JPEG)
+FIND_PACKAGE(PNG)
+FIND_PACKAGE(TIFF)
 
 ################################################################################
 # Create bin and lib directories if required
