Use MaPports libs.
--- CMakeLists.txt.orig	2026-02-03 22:26:30.000000000 -0600
+++ CMakeLists.txt	2026-04-11 13:25:40.000000000 -0500
@@ -65,11 +65,11 @@
 target_link_libraries(finitediff_finitediff PRIVATE finitediff::warnings)
 
 # Eigen
-include(eigen)
+find_package(Eigen3 3.3 REQUIRED NO_MODULE)
 target_link_libraries(finitediff_finitediff PUBLIC Eigen3::Eigen)
 
 # Logger
-include(spdlog)
+find_package(spdlog REQUIRED)
 target_link_libraries(finitediff_finitediff PUBLIC spdlog::spdlog)
 
 ################################################################################
--- tests/CMakeLists.txt.orig	2026-02-03 22:26:30.000000000 -0600
+++ tests/CMakeLists.txt	2026-04-11 13:25:40.000000000 -0500
@@ -20,7 +20,7 @@
 include(finitediff_warnings)
 target_link_libraries(finitediff_tests PRIVATE finitediff::warnings)
 
-include(catch2)
+find_package(Catch2)
 target_link_libraries(finitediff_tests PUBLIC Catch2::Catch2WithMain)
 
 ################################################################################
@@ -33,8 +33,6 @@
 # Register tests
 ################################################################################
 
-FetchContent_GetProperties(catch2)
-list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/extras)
 include(Catch)
 
 # Register tests
