--- textio/txCommands.c.orig	2015-04-16 11:48:14.000000000 -0500
+++ textio/txCommands.c	2015-09-07 19:42:06.000000000 -0500
@@ -1141,7 +1141,7 @@
     if (argc > TX_MAXARGS)
     {
 	TxError("Error: number of command arguments exceeds %d!\n", TX_MAXARGS);
-	return;
+	return 0;
     }
 
     SigIOReady = FALSE;
@@ -1158,7 +1158,7 @@
 	{
 	    TxError("Error: command length exceeds %d characters!\n", TX_MAX_CMDLEN);
 	    TxFreeCommand(tclcmd);
-	    return;
+	    return 0;
 	}
 	strcpy(&tclcmd->tx_argstring[asize], argv[n]);
 	tclcmd->tx_argv[n] = &tclcmd->tx_argstring[asize];
