--- qt-x11-free-3.3.8/mkspecs/darwin-g++/qmake.conf.orig	2009-05-31 20:31:18.000000000 +1000
+++ qt-x11-free-3.3.8/mkspecs/darwin-g++/qmake.conf	2009-05-31 20:30:34.000000000 +1000
@@ -42,12 +42,12 @@
 
 QMAKE_INCDIR		= @PREFIX@/lib/qt3/include
 QMAKE_LIBDIR		= @PREFIX@/lib
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	= @PREFIX@/include
+QMAKE_LIBDIR_X11	= @PREFIX@/lib
 QMAKE_INCDIR_QT		= $(QTDIR)/include
 QMAKE_LIBDIR_QT		= $(QTDIR)/lib
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	= @PREFIX@/include
+QMAKE_LIBDIR_OPENGL	= @PREFIX@/lib
 
 QMAKE_LINK		= g++-4.0
 QMAKE_LINK_SHLIB	= g++-4.0
