--- src/params.h.orig	2009-09-13 20:13:01.000000000 -0500
+++ src/params.h	2010-04-08 00:42:24.000000000 -0500
@@ -49,15 +49,15 @@
  * notes above.
  */
 #ifndef JOHN_SYSTEMWIDE
-#define JOHN_SYSTEMWIDE			0
+#define JOHN_SYSTEMWIDE			1
 #endif
 
 #if JOHN_SYSTEMWIDE
 #ifndef JOHN_SYSTEMWIDE_EXEC /* please refer to the notes above */
-#define JOHN_SYSTEMWIDE_EXEC		"/usr/libexec/john"
+#define JOHN_SYSTEMWIDE_EXEC		"@PREFIX@/share/john"
 #endif
 #ifndef JOHN_SYSTEMWIDE_HOME
-#define JOHN_SYSTEMWIDE_HOME		"/usr/share/john"
+#define JOHN_SYSTEMWIDE_HOME		"@PREFIX@/share/john"
 #endif
 #define JOHN_PRIVATE_HOME		"~/.john"
 #endif
