Index: Modules/objc/function.m
===================================================================
--- Modules/objc/function.m	(revision 1995)
+++ Modules/objc/function.m	(working copy)
@@ -132,8 +132,8 @@
 			plain_count + byref_in_count, PyTuple_Size(args));
 		return NULL;
 	} else {
-		if (PyErr_WarnEx(PyExc_DeprecationWarning, 
-			"Not all arguments to an Objective-C function are present", 1) < 0) {
+		if (PyErr_Warn(PyExc_DeprecationWarning, 
+			"Not all arguments to an Objective-C function are present") < 0) {
 			return NULL;
 		}
 	}
Index: Modules/objc/module.m
===================================================================
--- Modules/objc/module.m	(revision 1995)
+++ Modules/objc/module.m	(working copy)
@@ -495,8 +495,8 @@
 		return NULL;
 	}
 
-	if (PyErr_WarnEx(PyExc_DeprecationWarning,
-		"Use the new metadata machinery", 1) < 0) {
+	if (PyErr_Warn(PyExc_DeprecationWarning,
+		"Use the new metadata machinery") < 0) {
 
 		return NULL;
 	}
Index: Modules/objc/libffi_support.m
===================================================================
--- Modules/objc/libffi_support.m	(revision 1995)
+++ Modules/objc/libffi_support.m	(working copy)
@@ -1708,7 +1708,7 @@
 				"Not all Objective-C arguments are present in the Python argument-list of %s", PyString_AsString(repr));
 			Py_DECREF(repr);
 
-			if (PyErr_WarnEx(PyExc_DeprecationWarning, buffer, 1) < 0) {
+			if (PyErr_Warn(PyExc_DeprecationWarning, buffer) < 0) {
 				Py_DECREF(methinfo);
 				PyMem_Free(stubUserdata);
 				return NULL;
@@ -1769,8 +1769,8 @@
 			snprintf(buffer, sizeof(buffer),
 				"Not all Objective-C arguments are present in the Python argument-list of %s", PyString_AsString(repr));
 			Py_DECREF(repr);
-			if (PyErr_WarnEx(PyExc_DeprecationWarning,
-				buffer, 1) < 0) {
+			if (PyErr_Warn(PyExc_DeprecationWarning,
+				buffer) < 0) {
 				Py_DECREF(methinfo);
 				PyMem_Free(stubUserdata);
 				return NULL;
@@ -3413,7 +3413,7 @@
 			plain_count + byref_in_count, PyTuple_Size(args));
 		goto error_cleanup;
 	} else {
-		if (PyErr_WarnEx(PyExc_DeprecationWarning, "Not all arguments to an Objective-C method are present", 1) < 0) {
+		if (PyErr_Warn(PyExc_DeprecationWarning, "Not all arguments to an Objective-C method are present") < 0) {
 			goto error_cleanup;
 		}
 	}
