--- core/vil/tests/test_image_resource.cxx.orig	2012-06-14 21:41:57.000000000 +0200
+++ core/vil/tests/test_image_resource.cxx	2012-06-14 21:42:55.000000000 +0200
@@ -4,7 +4,6 @@
 #include <vcl_iostream.h>
 #include <vcl_complex.h>
 #include <vil/vil_image_resource.h>
-#include <vil/vil_math.h>
 #include <vil/vil_new.h>
 #include <vil/vil_crop.h>
 #include <vil/vil_clamp.h>
@@ -21,6 +20,10 @@
 static inline bool operator> (vcl_complex<float> const&, vcl_complex<float> const&) { return false; }
 static inline bool operator> (vcl_complex<double>const&, vcl_complex<double>const&) { return false; }
 
+// vil_math.h needs to be included _after_ declaring the operators, or the
+// operator need to be in namespace that vcl_complex is in
+#include <vil/vil_math.h>
+
 template <class T>
 void test_image_resource(vcl_string type, vil_pixel_format format, T /*dummy*/)
 {
