--- core/vil/CMakeLists.txt.orig	2013-08-11 09:58:38.000000000 -0700
+++ core/vil/CMakeLists.txt	2013-08-11 09:59:49.000000000 -0700
@@ -61,7 +61,7 @@
 ##############################
 INCLUDE( ${VXL_CMAKE_DIR}/FindOpenJPEG2.cmake )
 IF(OPENJPEG2_FOUND)
-  INCLUDE_DIRECTORIES(${OPENJPEG2_INCLUDE_DIR})
+  INCLUDE_DIRECTORIES(BEFORE ${OPENJPEG2_INCLUDE_DIR})
   ADD_DEFINITIONS( ${OPENJPEG2_DEFINITIONS} )
   SET( vil_sources ${vil_sources}
     file_formats/vil_openjpeg.h  file_formats/vil_openjpeg.cxx
