--- core/vil/file_formats/vil_jpeg_compressor.cxx.orig	2010-09-06 22:37:58.000000000 -0500
+++ core/vil/file_formats/vil_jpeg_compressor.cxx	2017-10-16 21:50:15.000000000 -0500
@@ -71,8 +71,7 @@
     jpeg_set_quality(&jobj, quality, TRUE);
 
     // start compression
-    bool write_all_tables = true;
-    jpeg_start_compress (&jobj, write_all_tables);
+    jpeg_start_compress (&jobj, TRUE);
 
     //
     ready = true;
--- core/vil1/file_formats/vil1_jpeg_compressor.cxx.orig	2003-05-21 19:31:50.000000000 -0500
+++ core/vil1/file_formats/vil1_jpeg_compressor.cxx	2017-10-16 21:55:39.000000000 -0500
@@ -54,8 +54,7 @@
     jpeg_set_defaults(&jobj);
 
     // start compression
-    bool write_all_tables = true;
-    jpeg_start_compress (&jobj, write_all_tables);
+    jpeg_start_compress (&jobj, TRUE);
 
     //
     ready = true;
