--- contrib/tbl/vipl/filter/vipl_filter_2d.txx.orig	2012-06-15 19:50:50.000000000 +0200
+++ contrib/tbl/vipl/filter/vipl_filter_2d.txx	2012-06-15 19:51:14.000000000 +0200
@@ -112,7 +112,7 @@
       vipl_section_descriptor<DataOut>& secDesc = *dstitr;
       put_secp( new vipl_section_descriptor<DataOut> (secDesc) );
 #else
-      put_secp( new vipl_section_descriptor<DataOut> (*dstitr));
+      this->put_secp( new vipl_section_descriptor<DataOut> (*dstitr));
 #endif
       this->ref_dst_section()->ref_overlap()[0] = this->image_border_size();
       this->ref_dst_section()->ref_overlap()[1] = this->image_border_size();
@@ -121,7 +121,7 @@
         vipl_section_descriptor<DataIn>& inSecDesc = *it;
         put_insecp( new vipl_section_descriptor<DataIn> (inSecDesc));
 #else
-        put_insecp( new vipl_section_descriptor<DataIn> (*it));
+        this->put_insecp( new vipl_section_descriptor<DataIn> (*it));
 #endif
         this->ref_src_section()->ref_overlap()[0] = this->image_border_size();
         this->ref_src_section()->ref_overlap()[1] = this->image_border_size();
@@ -151,7 +151,7 @@
       vipl_section_descriptor<DataOut>& secDesc2 = *it;
       put_secp( new vipl_section_descriptor<DataOut> (secDesc2));
 #else
-      put_secp( new vipl_section_descriptor<DataOut> (*it));
+      this->put_secp( new vipl_section_descriptor<DataOut> (*it));
 #endif
       this->ref_dst_section()->ref_overlap()[0] = this->image_border_size();
       this->ref_dst_section()->ref_overlap()[1] = this->image_border_size();
@@ -160,7 +160,7 @@
         vipl_section_descriptor<DataIn>& inSecDesc2 = *srcitr;
         put_insecp( new vipl_section_descriptor<DataIn> (inSecDesc2));
 #else
-        put_insecp( new vipl_section_descriptor<DataIn> (*srcitr));
+        this->put_insecp( new vipl_section_descriptor<DataIn> (*srcitr));
 #endif
         this->ref_src_section()->ref_overlap()[0] = this->image_border_size();
         this->ref_src_section()->ref_overlap()[1] = this->image_border_size();
