--- contrib/prip/vmap/vmap_2_map.txx.orig	2012-06-15 00:08:27.000000000 +0200
+++ contrib/prip/vmap/vmap_2_map.txx	2012-06-15 00:13:24.000000000 +0200
@@ -214,7 +214,7 @@
 template <class D>
 void vmap_2_map<D>::contraction(contraction_kernel const& arg_kernel)
 {
-  int ld=get_dart_pointer.size(), id ;
+  int ld=nb_darts(), id ;
 
   typename contraction_kernel::const_iterator itk ;
   for (itk=arg_kernel.begin(); itk!=arg_kernel.end(); ++itk)
@@ -247,7 +247,7 @@
     }
   }
 
-  ld=get_dart_pointer.size()-ld ;
+  ld=nb_darts()-ld ;
   for (id=0; id<ld ; ++id )
   {
 #if 0
@@ -261,7 +261,7 @@
 template <class D>
 void vmap_2_map<D>::removal(removal_kernel const& arg_kernel)
 {
-  int ld=get_dart_pointer.size(), id ;
+  int ld=nb_darts(), id ;
 
   typename contraction_kernel::const_iterator itk ;
   for (itk=arg_kernel.begin(); itk!=arg_kernel.end(); ++itk)
@@ -294,7 +294,7 @@
     }
   }
 
-  ld=get_dart_pointer.size()-ld ;
+  ld=nb_darts()-ld ;
   for (id=0; id<ld ; ++id )
   {
 #if 0
