--- pr/include/md/_pth.h-orig	2007-10-29 05:25:26.000000000 +0900
+++ pr/include/md/_pth.h	2007-10-29 05:26:05.000000000 +0900
@@ -205,10 +205,6 @@
 /*
  * These platforms don't have pthread_kill()
  */
-#if defined(DARWIN)
-#define pthread_kill(thread, sig) ENOSYS
-#endif
-
 #if defined(OSF1) || defined(VMS)
 #define PT_PRIO_MIN            PRI_OTHER_MIN
 #define PT_PRIO_MAX            PRI_OTHER_MAX
