--- Makefile.orig	2004-06-02 11:36:58.000000000 +0200
+++ Makefile	2010-10-03 01:33:55.000000000 +0200
@@ -1,6 +1,6 @@
 # vim: set noexpandtab :
 
-INCLUDES = -I/usr/local/include -I/usr/include/pgsql -I/usr/include/pgsql/internal -I/usr/local/pgsql/include/internal -I/usr/include/xbase -I/usr/local/pgsql/include -I/usr/local/include/xbase
+INCLUDES = -I@@PREFIX@@/include -I@@PREFIX@@/include/postgresql81 -I@@PREFIX@@/include/postgresql81/internal -I@@PREFIX@@/include/xbase64
 OPTIMIZE=-O2
 CFLAGS = -g -Wall $(OPTIMIZE)
 CFLAGS += -Wno-deprecated
@@ -8,7 +8,7 @@
 #CFLAGS = -g -ggdb -Wall
 # for xbase sys/locking problem in 1.8.0a
 CFLAGS += -DHAVE_CONFIG_H
-LIBS = -L/usr/local/pgsql/lib -lpq -lxbase
+LIBS = -L@@PREFIX@@/lib -L@@PREFIX@@/lib/postgresql81 -lpq -lxbase64
 
 ifeq ($(OSTYPE),solaris)
 # uncomment for Solaris
@@ -16,9 +16,9 @@
 prefix=/usr/local
 else
 # -- LINUX DEFS --
-LIBS += -lcrypt
-CFLAGS += -DLINUX
-prefix=/usr
+#LIBS += -lcrypt
+#CFLAGS += -DLINUX
+prefix=@@PREFIX@@
 endif
 
 # CFLAGS += -static
@@ -60,9 +60,9 @@
 	tar cvf - README INSTALL COPYING-2.0 *.h *.hh *.cpp *.1 convenctable Makefile | gzip -c > xbase2pg.tgz
 
 install: xbase2pg pg2xbase xbasenew xbaseconv
-	mkdir -p $(prefix)/bin/ $(prefix)/man/man1/
+	mkdir -p $(prefix)/bin/ $(prefix)/share/man/man1/
 	mkdir -p $(prefix)/share/pg2xbase
 	$(CP) $(PROGS) $(prefix)/bin/
 	$(CP) convenctable $(convtable)
-	$(CP) xbase2pg.1 pg2xbase.1 xbaseconv.1 xbasenew.1 $(prefix)/man/man1/
+	$(CP) xbase2pg.1 pg2xbase.1 xbaseconv.1 xbasenew.1 $(prefix)/share/man/man1/
 	
