--- Makefile.in.orig	2012-12-22 05:03:04.000000000 -0600
+++ Makefile.in	2016-12-05 20:07:16.000000000 -0600
@@ -1233,11 +1233,9 @@
 
 
 install-exec-hook:
-	-chgrp games $(DESTDIR)$(bindir)/toppler
-	-chmod 2755 $(DESTDIR)$(bindir)/toppler
+	-chmod 0755 $(DESTDIR)$(bindir)/toppler
 
 install-data-hook:
-	-chgrp games $(DESTDIR)$(pkglocalstatedir)/$(PACKAGE).hsc
 	-chmod 0664 $(DESTDIR)$(pkglocalstatedir)/$(PACKAGE).hsc
 
 dist-win32: $(PACKAGE).exe
