--- configure.orig	2009-04-12 10:07:30.000000000 -0700
+++ configure	2009-04-22 22:53:09.000000000 -0700
@@ -4196,6 +4196,9 @@ echores "$_x11_headers"
 
 
 echocheck "X11"
+if test "$_x11" = yes ; then
+  libs_mplayer="$libs_mplayer -lXext -lX11"
+fi
 if test "$_x11" = auto && test "$_x11_headers" = yes ; then
   cat > $TMPC <<EOF
 #include <X11/Xlib.h>
@@ -6760,6 +6763,7 @@ fi
 if test "$_qtx" = yes ; then
     def_qtx='#define CONFIG_QTX_CODECS 1'
     win32 && _qtx_codecs_win32=yes && def_qtx_win32='#define CONFIG_QTX_CODECS_WIN32 1'
+    darwin && extra_ldflags="$extra_ldflags -framework Carbon -framework QuickTime" && def_quicktime='#define CONFIG_QUICKTIME 1'
     _codecmodules="qtx $_codecmodules"
     darwin || win32 || _qtx_emulation=yes
 else
