--- configure.orig	2013-04-01 00:55:08.000000000 -0700
+++ configure	2013-04-01 01:08:51.000000000 -0700
@@ -6694,6 +6694,9 @@ if test "$_tremor" = auto; then
   _tremor=no
   statement_check tremor/ivorbiscodec.h 'vorbis_synthesis(0, 0)' -logg -lvorbisidec && _tremor=yes && _libvorbis=no
 fi
+if test "$_libvorbis" = yes; then
+  vorbislibs=$($_pkg_config --libs vorbisenc)
+fi
 if test "$_libvorbis" = auto; then
   _libvorbis=no
   for vorbislibs in '-lvorbisenc -lvorbis -logg' '-lvorbis -logg' ; do
