--- ./configure.in.orig	2007-07-03 00:05:52.000000000 +0200
+++ ./configure.in	2007-08-29 21:19:06.000000000 +0200
@@ -360,11 +360,16 @@ dnl - util.h (NetBSD)
 AC_CHECK_FUNC(openpty, have_openpty=yes, [
   AC_CHECK_LIB(util, openpty, have_openpty=yes need_libutil=yes)])
 
+UTILLIBS=
 if test "${have_openpty}." = "yes."; then
   AC_DEFINE([HAVE_OPENPTY],1,[defined if openpty() is avialable])
   AC_CHECK_HEADERS([pty.h libutil.h util.h], break)
-  test "$need_libutil" = "yes" && LIBS="-lutil $LIBS"
+  if test "$need_libutil" = "yes" ; then
+    UTILLIBS="-lutil"
+    LIBS="$UTILLIBS $LIBS"
+  fi
 fi
+AC_SUBST(UTILLIBS)
 
 AC_CHECK_FUNCS(EVP_DigestInit,,
 	[AC_CHECK_LIB(crypto, EVP_DigestInit,,
--- ./vfile/Makefile.am.orig	2007-07-02 20:30:49.000000000 +0200
+++ ./vfile/Makefile.am	2007-08-29 21:21:48.000000000 +0200
@@ -23,7 +23,7 @@ pkgincludedir = $(includedir)/vfile
 pkginclude_HEADERS = vfile.h p_open.h vopen3.h
 
 noinst_PROGRAMS = vfget test_vopen3 
-TEST_LDADDS    = libvfile.la -lutil
+TEST_LDADDS    = libvfile.la @UTILLIBS@
 
 vfget_SOURCES  = vfget.c 
 vfget_LDADD    = $(TEST_LDADDS)
--- ./vfile/Makefile.in.orig	2007-07-02 20:31:01.000000000 +0200
+++ ./vfile/Makefile.in	2007-08-29 21:19:06.000000000 +0200
@@ -190,6 +190,7 @@ TRURL_DIR = @TRURL_DIR@
 TRURL_INCLUDE = @TRURL_INCLUDE@
 TRURL_LDFLAG = @TRURL_LDFLAG@
 USE_NLS = @USE_NLS@
+UTILLIBS = @UTILLIBS@
 VERSION = @VERSION@
 VERSION_CVSTAG = @VERSION_CVSTAG@
 WITH_METADATA_REPOSITORY_FALSE = @WITH_METADATA_REPOSITORY_FALSE@
@@ -260,7 +261,7 @@ libvfile_la_SOURCES = vfile.c fetch.c vf
 
 libvfile_la_LIBADD = vfff/libvfff.la
 pkginclude_HEADERS = vfile.h p_open.h vopen3.h
-TEST_LDADDS = libvfile.la -lutil
+TEST_LDADDS = libvfile.la @UTILLIBS@
 vfget_SOURCES = vfget.c 
 vfget_LDADD = $(TEST_LDADDS)
 test_vopen3_SOURCES = test_vopen3.c 
