--- autogen.sh.orig	2009-10-08 11:26:39.000000000 -0500
+++ autogen.sh	2012-11-04 05:35:55.000000000 -0600
@@ -18,7 +18,13 @@
 	DIE=1
 }
 
-if automake-1.10 --version < /dev/null > /dev/null 2>&1 ; then
+if automake-1.12 --version < /dev/null > /dev/null 2>&1 ; then
+    AUTOMAKE=automake-1.12
+    ACLOCAL=aclocal-1.12
+elif automake-1.11 --version < /dev/null > /dev/null 2>&1 ; then
+    AUTOMAKE=automake-1.11
+    ACLOCAL=aclocal-1.11
+elif automake-1.10 --version < /dev/null > /dev/null 2>&1 ; then
     AUTOMAKE=automake-1.10
     ACLOCAL=aclocal-1.10
 elif automake-1.9 --version < /dev/null > /dev/null 2>&1 ; then
@@ -38,7 +44,7 @@
         DIE=1
 fi
 
-(libtool --version) < /dev/null > /dev/null 2>&1 || {
+(glibtool --version) < /dev/null > /dev/null 2>&1 || {
 	echo
 	echo "You must have libtool installed to compile $PROJECT."
 	echo "Get http://ftp.gnu.org/gnu/libtool/libtool-1.5.22.tar.gz"
@@ -49,15 +55,11 @@
 	exit 1
 fi
 
-libtoolize --force --copy
+glibtoolize --force --copy --quiet
 
 $ACLOCAL $ACLOCAL_FLAGS
 autoheader
-$AUTOMAKE --add-missing
+$AUTOMAKE --add-missing < /dev/null > /dev/null 2>&1
 autoconf
 cd $THEDIR
 
-$srcdir/configure --enable-maintainer-mode "$@"
-
-echo 
-echo "Now type 'make' to compile $PROJECT"
