--- DESCRIPTION	2023-10-30 18:20:02.000000000 +0800
+++ DESCRIPTION	2023-10-31 08:43:29.000000000 +0800
@@ -44,7 +44,7 @@
 BugReports: https://github.com/duckdb/duckdb-r/issues
 Depends: DBI, R (>= 3.6.0)
 Imports: methods, utils
-Suggests: arrow (>= 13.0.0), bit64, callr, DBItest, dplyr, dbplyr,
+Suggests: arrow (>= 13.0.0), bit64, callr, dplyr, dbplyr,
         rlang, testthat, tibble, vctrs, withr, adbcdrivermanager
 Encoding: UTF-8
 RoxygenNote: 7.2.3
