--- ./programs/Xserver/hw/darwin/darwinKeyboard.c.orig	2005-04-29 13:17:48.000000000 -0700
+++ ./programs/Xserver/hw/darwin/darwinKeyboard.c	2005-04-29 13:17:56.000000000 -0700
@@ -64,7 +64,8 @@
 /* Define this to use Alt for Mode_switch. */
 #define ALT_IS_MODE_SWITCH 1
 
-#include <drivers/event_status_driver.h>
+#include <sys/cdefs.h>
+#include <IOKit/hidsystem/event_status_driver.h>
 #include <IOKit/hidsystem/ev_keymap.h>
 #include <stdio.h>
 #include <stdlib.h>
