--- ./lib/GL/apple/dri_driver.h.orig	2005-04-29 13:18:29.000000000 -0700
+++ ./lib/GL/apple/dri_driver.h	2005-04-29 13:20:03.000000000 -0700
@@ -38,6 +38,12 @@
 
 #include "Xplugin.h"
 #include "Xthreads.h"
+
+#ifndef GL_TYPEDEFS_2_0
+#define GL_TYPEDEFS_2_0
+    typedef char GLchar;
+#endif
+
 #include <CoreGraphics/CoreGraphics.h>
 #include <OpenGL/OpenGL.h>
 #include <OpenGL/CGLContext.h>
--- ./lib/GL/apple/dri_glx.c.orig	2005-04-29 13:20:40.000000000 -0700
+++ ./lib/GL/apple/dri_glx.c	2005-04-29 13:20:56.000000000 -0700
@@ -50,11 +50,11 @@
 
 
 /* Apple OpenGL "driver" information. */
-static const char *__driDriverName = "apple";
-static const char __driConfigOptions[] = "";
-static const int __driDriverMajor = 1;
-static const int __driDriverMinor = 0;
-static const int __driDriverPatch = 0;
+const char *__driDriverName = "apple";
+const char __driConfigOptions[] = "";
+const int __driDriverMajor = 1;
+const int __driDriverMinor = 0;
+const int __driDriverPatch = 0;
 
 
 /*
