diff --git a/fizz/experimental/psp/PSP.cpp b/fizz/experimental/psp/PSP.cpp
index 61ab9f1..c428877 100644
--- a/fizz/experimental/psp/PSP.cpp
+++ b/fizz/experimental/psp/PSP.cpp
@@ -107,7 +107,7 @@ class AsyncPSPUpgradeImpl : public ::fizz::psp::AsyncPSPUpgradeFrame,
     }
   }
   void start(Callback* cb, std::chrono::milliseconds timeout) override {
-    FIZZ_CHECK_EQ(awaiter_, nullptr);
+    FIZZ_CHECK(awaiter_ == nullptr);
     awaiter_ = cb;
     timeout_ = timeout;
 
diff --git a/fizz/extensions/delegatedcred/DelegatedCredentialUtils.cpp b/fizz/extensions/delegatedcred/DelegatedCredentialUtils.cpp
index dcdcc85..414951a 100644
--- a/fizz/extensions/delegatedcred/DelegatedCredentialUtils.cpp
+++ b/fizz/extensions/delegatedcred/DelegatedCredentialUtils.cpp
@@ -61,7 +61,7 @@ Status DelegatedCredentialUtils::hasDelegatedExtension(
   FIZZ_RETURN_ON_ERROR(generateCredentialOid(credentialOid, err));
   // To be valid for a credential, it has to have the delegated credential
   // extension and the digitalSignature KeyUsage.
-  FIZZ_DCHECK_NE(credentialOid, nullptr);
+  FIZZ_DCHECK(credentialOid != nullptr);
   auto credentialIdx = X509_get_ext_by_OBJ(cert.get(), credentialOid, -1);
   ret = (credentialIdx != -1);
   return Status::Success;
diff --git a/fizz/extensions/exportedauth/ExportedAuthenticator.cpp b/fizz/extensions/exportedauth/ExportedAuthenticator.cpp
index 555aa43..55db226 100644
--- a/fizz/extensions/exportedauth/ExportedAuthenticator.cpp
+++ b/fizz/extensions/exportedauth/ExportedAuthenticator.cpp
@@ -50,7 +50,7 @@ Status ExportedAuthenticator::getAuthenticator(
   FIZZ_RETURN_ON_ERROR(
       ::fizz::DefaultFactory().makeHasherFactory(
           makeHasher, err, hashFunction));
-  FIZZ_DCHECK_NE(makeHasher, nullptr);
+  FIZZ_DCHECK(makeHasher != nullptr);
 
   auto supportedSchemes = transport.getSupportedSigSchemes();
   Buf handshakeContext;
diff --git a/fizz/server/AeadCookieCipher.h b/fizz/server/AeadCookieCipher.h
index 1e959f8..4a1de27 100644
--- a/fizz/server/AeadCookieCipher.h
+++ b/fizz/server/AeadCookieCipher.h
@@ -27,7 +27,7 @@ class AeadCookieCipher : public CookieCipher {
  public:
   explicit AeadCookieCipher(std::unique_ptr<TokenCipher> tokenCipher)
       : tokenCipher_{std::move(tokenCipher)} {
-    FIZZ_DCHECK_NE(tokenCipher_.get(), nullptr);
+    FIZZ_DCHECK(tokenCipher_.get() != nullptr);
   }
 
   /**
