--- tools/getlibs.sh.orig	2009-01-07 04:29:54.000000000 +1100
+++ tools/getlibs.sh	2009-09-22 06:22:54.000000000 +1000
@@ -104,6 +104,9 @@
 
 	'CYGWIN_NT-5.0'|'CYGWIN_NT-5.1')
 		echo "" ;;
+	
+	'Darwin')
+		echo "-lresolv" ;;
 
 	'FreeBSD'|'NetBSD')
 		echo "-lpthread" ;;
--- makefile.orig	2009-07-21 01:02:22.000000000 +1000
+++ makefile	2009-09-22 06:38:58.000000000 +1000
@@ -1409,7 +1409,7 @@
 $(DYLIBNAME):	$(OBJS) $(EXTRAOBJS) $(TESTOBJS)
 				@$(LD) -dynamiclib -compatibility_version $(MAJ).$(MIN) \
 					-current_version $(MAJ).$(MIN).$(PLV) \
-					-o $(DYLIBNAME) $(OBJS) $(EXTRAOBJS)
+					-o $(DYLIBNAME) $(OBJS) $(EXTRAOBJS) `./tools/getlibs.sh autodetect`
 
 # If installing cryptlib as a systemwide lib, run ldconfig (which normally
 # reads /etc/ld.so.conf, sets up the appropriate symbolic links in the
