--- makefile.orig	2018-02-08 12:39:38.000000000 -0600
+++ makefile	2018-03-28 08:16:58.000000000 -0500
@@ -93,7 +93,7 @@
 # Further cc flags are gathered dynamically at runtime via the ccopts.sh
 # script.
 
-CFLAGS		= -c -D__UNIX__ -DNDEBUG -I.
+CFLAGS		+= -c -D__UNIX__ -DNDEBUG -I.
 CFLAGS_ANALYSE = -c -D__UNIX__ -I.
 CFLAGS_COVERAGE = -c -D__UNIX__ -I. -ggdb3 -fno-omit-frame-pointer -O1 --coverage -fprofile-arcs -ftest-coverage
 CFLAGS_DEBUG = -c -D__UNIX__ -I. -ggdb3 -fno-omit-frame-pointer -O0
@@ -450,6 +450,7 @@
 	@if [ ! -d "$(DESTDIR)$(PATH_LIB)" ] ; then \
 		mkdir -p "$(DESTDIR)$(PATH_LIB)" ; \
 		chmod 755 "$(DESTDIR)$(PATH_LIB)" ; \
+	fi
 	@if [ ! -d "$(DESTDIR)$(PATH_INCLUDE)" ] ; then \
 		mkdir -p "$(DESTDIR)$(PATH_INCLUDE)" ; \
 		chmod 755 "$(DESTDIR)$(PATH_INCLUDE)" ; \
@@ -470,6 +471,8 @@
 	if [ -f "$(DYLIBNAME)" ] ; then \
 		cp "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)" ; \
 		chmod 755 "$(DESTDIR)$(PATH_LIB)/$(DYLIBNAME)" ; \
+		ln -s "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)/lib$(PROJ).$(MAJ).dylib" ; \
+		ln -s "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)/lib$(PROJ).dylib" ; \
 	fi
 	cp cryptlib.h "$(DESTDIR)$(PATH_INCLUDE)"
 	chmod 644 "$(DESTDIR)$(PATH_INCLUDE)/cryptlib.h"
@@ -1557,8 +1560,9 @@
 					$(STRIP) $(MAJ).$(MIN) $(OBJS) $(EXTRAOBJS)
 
 $(DYLIBNAME):	$(OBJS) $(EXTRAOBJS) $(TESTOBJS)
-				@$(LD) -dynamiclib -compatibility_version $(MAJ).$(MIN) \
+				$(LD) $(LDFLAGS) -dynamiclib -compatibility_version $(MAJ).$(MIN) \
 					-current_version $(MAJ).$(MIN).$(PLV) \
+					-install_name $(PATH_LIB)/$(DYLIBNAME) \
 					`./tools/getlibs.sh autodetect` \
 					-o $(DYLIBNAME) $(OBJS) $(EXTRAOBJS)
 
