--- makefile.orig	2021-09-11 00:27:18.000000000 -0500
+++ makefile	2021-11-25 09:51:35.000000000 -0600
@@ -92,7 +92,7 @@
 #
 # Standard build flags
 
-CFLAGS			= -c -D__UNIX__ -DNDEBUG -I.
+CFLAGS			+= -c -D__UNIX__ -DNDEBUG -I.
 CFLAGS_DEBUG	= -c -D__UNIX__ -I. -g -O1
 CFLAGS_DEBUGGCC	= -c -D__UNIX__ -I. -ggdb3 -fno-omit-frame-pointer -O1
 
@@ -553,6 +553,8 @@
 	if [ -f "$(DYLIBNAME)" ] ; then \
 		cp "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)" ; \
 		chmod 755 "$(DESTDIR)$(PATH_LIB)/$(DYLIBNAME)" ; \
+		ln -s "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)/lib$(PROJ).$(MAJ).dylib" ; \
+		ln -s "$(DYLIBNAME)" "$(DESTDIR)$(PATH_LIB)/lib$(PROJ).dylib" ; \
 	fi
 	cp cryptlib.h "$(DESTDIR)$(PATH_INCLUDE)"
 	chmod 644 "$(DESTDIR)$(PATH_INCLUDE)/cryptlib.h"
@@ -1716,8 +1718,9 @@
 				fi
 
 $(DYLIBNAME):	$(OBJS) $(EXTRAOBJS) $(TESTOBJS)
-				@$(LD) -dynamiclib -compatibility_version $(MAJ).$(MIN) \
+				$(LD) $(LDFLAGS) -dynamiclib -compatibility_version $(MAJ).$(MIN) \
 					-current_version $(MAJ).$(MIN).$(PLV) \
+					-install_name $(PATH_LIB)/$(DYLIBNAME) \
 					`./tools/getlibs.sh $(LD) $(OSNAME)` \
 					-o $(DYLIBNAME) $(OBJS) $(EXTRAOBJS)
 
