--- tools/ccopts.sh.orig	2016-03-04 18:54:34.000000000 -0600
+++ tools/ccopts.sh	2016-10-08 11:16:50.000000000 -0500
@@ -375,7 +375,7 @@
 # worked since at least 2.7.2 although it wasn't actually documented until
 # the first 3.x releases).
 
-GCC_VER=`gcc -dumpversion | tr -d  '.' | cut -c 1-2`
+GCC_VER=`$CC -dumpversion | tr -d  '.' | cut -c 1-2`
 
 # Try and determine the CPU type.  This is made more complex by a pile of
 # *BSE's which, along with antideluvian tools like an as that doesn't
