--- configure.ac.orig	2015-05-30 09:46:38.000000000 -0700
+++ configure.ac	2015-08-26 12:03:38.000000000 -0700
@@ -8,9 +8,9 @@
 dnl Boilerplace and versioning
 dnl ###########################################################################
 
-# if autoconf is used not in a git source tree, use version=2.5.0
+# if autoconf is used not in a git source tree, use version=2.5.2
 AC_INIT([balsa],
-        m4_esyscmd([test -d .git && echo -n `git describe --tags` || echo -n 2.5.1]),
+        [2.5.2],
         [])
 
 AM_INIT_AUTOMAKE
