--- configure.ac.orig	2023-08-25 12:51:20
+++ configure.ac	2023-08-25 12:52:06
@@ -19,6 +19,7 @@
 
 AX_CHECK_ENABLE_DEBUG([yes],,, [$ax_is_release])
 AX_COMPILER_FLAGS([WARN_CFLAGS], [WARN_LDFLAGS], [$ax_is_release])
+WARN_LDFLAGS=""
 
 AS_IF([test "x$ax_is_release" != "xyes"], [PACKAGE_COMMIT_ID=$(git describe)])
 AS_IF([test -n "$PACKAGE_COMMIT_ID"],
@@ -91,7 +92,7 @@
 AC_CHECK_FUNCS([shm_open], [LIBRT=], [AC_CHECK_LIB([rt], [shm_open], [LIBRT=-lrt], [LIBRT=])])
 AC_SUBST(LIBRT)
 
-PKG_CHECK_MODULES(BABL, [babl])
+PKG_CHECK_MODULES(BABL, [babl-0.1])
 PKG_CHECK_MODULES(CAIRO, [cairo >= $CAIRO_MIN_VERSION cairo-gobject])
 PKG_CHECK_MODULES(GFBGRAPH, [libgfbgraph-0.2 >= $GFBGRAPH_MIN_VERSION])
 PKG_CHECK_MODULES(GDATA, [libgdata >= $GDATA_MIN_VERSION])
