--- inc/Devel/CheckLib.pm.orig	2023-12-28 11:47:02
+++ inc/Devel/CheckLib.pm	2023-12-28 18:58:54
@@ -456,7 +456,7 @@
     my @ccflags = grep { length } _parsewords($Config_ccflags||'', $user_ccflags||'');
     my @ldflags = grep { length && $_ !~ m/^-Wl/ } _parsewords(@Config_ldflags, $user_ldflags||'');
     my @paths = split(/$Config{path_sep}/, $ENV{PATH});
-    my @cc = _parsewords($Config{cc});
+    my @cc = _parsewords($ENV{CC});
     if (check_compiler ($cc[0], $debug)) {
 	return ( [ @cc, @ccflags ], \@ldflags );
     }
