--- inc/Devel/CheckLib.pm.orig	2011-05-24 21:39:14.000000000 -0600
+++ inc/Devel/CheckLib.pm	2014-07-21 14:04:44.000000000 -0600
@@ -332,6 +332,8 @@
 }
     
 sub _findcc {
+    my @cc = split(/\s+/, $ENV{CC});
+    return @cc if -x $cc[0];
     # Need to use $keep=1 to work with MSWin32 backslashes and quotes
     my @Config_ccflags_ldflags =  @Config{qw(ccflags ldflags)};  # use copy so ASPerl will compile
     my @flags = grep { length } map { quotewords('\s+', 1, $_ || ()) } @Config_ccflags_ldflags;
