--- makefile.orig	2010-09-12 15:47:10.000000000 +0200
+++ makefile	2011-12-06 01:09:54.000000000 +0100
@@ -20,7 +20,7 @@
 CXXBFLAGS = $(CXXFLAGS) -fno-common  
 
 DELETER = rm -f
-COPIER = cp -a
+COPIER = cp -Rp
 
 SRCNAMES = \
   as_atomic.cpp \
@@ -65,6 +65,8 @@
 
 
 all: $(TARG)
+	touch -m -r makefile $(OBJDIR) $(LIBDIR)
+
 $(LIBDIR)/$(LIB): $(OBJDIR) $(LIBDIR) $(OBJ)
 	rm -f $(LIBDIR)/$(LIB)
 	ar r $(LIBDIR)/$(LIB) $(OBJ)
@@ -73,7 +75,7 @@
 
 $(LIBDIR)/$(DEVLIB): $(OBJDIR) $(LIBDIR) $(OBJ)
 	rm -f $(LIBDIR)/$(DEVLIB)
-	$(CXX) $(AFLAGS) -dynamiclib -o $(DEVLIB) $(OBJ)
+	$(CXX) $(AFLAGS) -dynamiclib -install_name $(LOCAL)/lib/$(DEVLIB) -o $(DEVLIB) $(OBJ)
 	mv $(DEVLIB) $(LIBDIR)/$(DEVLIB)
 	file  $(LIBDIR)/$(DEVLIB) 
 
