--- configure.orig	2011-01-15 22:27:58.000000000 +1100
+++ configure	2011-09-21 18:33:45.000000000 +1000
@@ -15635,37 +15635,13 @@
 
 # Check for the availabiity of libreadline
 
-for ac_header in readline/readline.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "readline/readline.h" "ac_cv_header_readline_readline_h" "$ac_includes_default"
-if test "x$ac_cv_header_readline_readline_h" = x""yes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_READLINE_READLINE_H 1
-_ACEOF
- HAVE_READLINE_H=1
-fi
-
-done
-
-for ac_header in readline/history.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "readline/history.h" "ac_cv_header_readline_history_h" "$ac_includes_default"
-if test "x$ac_cv_header_readline_history_h" = x""yes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_READLINE_HISTORY_H 1
-_ACEOF
- HAVE_HISTORY_H=1
-fi
-
-done
-
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for readline in -lreadline" >&5
-$as_echo_n "checking for readline in -lreadline... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for readline in -ledit" >&5
+$as_echo_n "checking for readline in -ledit... " >&6; }
 if test "${ac_cv_lib_readline_readline+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline  $LIBS"
+LIBS="-ledit  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -15868,15 +15844,7 @@
 # Similarly for --enable-pcretest-readline
 
 if test "$enable_pcretest_libreadline" = "yes"; then
-  if test "$HAVE_READLINE_H" != "1"; then
-    echo "** Cannot --enable-pcretest-readline because readline/readline.h was not found."
-    exit 1
-  fi
-  if test "$HAVE_HISTORY_H" != "1"; then
-    echo "** Cannot --enable-pcretest-readline because readline/history.h was not found."
-    exit 1
-  fi
-  LIBREADLINE="-lreadline"
+  LIBREADLINE="-ledit"
 fi
 
 
@@ -18564,7 +18532,7 @@
     Build static libs ............... : ${enable_static}
     Link pcregrep with libz ......... : ${enable_pcregrep_libz}
     Link pcregrep with libbz2 ....... : ${enable_pcregrep_libbz2}
-    Link pcretest with libreadline .. : ${enable_pcretest_libreadline}
+    Link pcretest with libedit ...... : ${enable_pcretest_libreadline}
 
 EOF
 
