--- tests/TestInt64.cc	2015-09-09 06:42:20.000000000 +0800
+++ tests/TestInt64.cc	2024-07-31 19:48:25.000000000 +0800
@@ -105,7 +105,7 @@
             d_buf = 64;
     }
     else {
-        d_buf = 0x00ffffffffffffff;
+        d_buf = 0x00ffffffffffffffll;
     }
 
     set_read_p(true);

--- tests/TestUInt64.cc	2015-09-09 06:42:20.000000000 +0800
+++ tests/TestUInt64.cc	2024-07-31 19:48:10.000000000 +0800
@@ -99,7 +99,7 @@
         d_buf = 64 * d_buf;
     }
     else {
-        d_buf = 0xffffffffffffffff;		// really big
+        d_buf = 0xffffffffffffffffull;		// really big
     }
 
     set_read_p(true);
