--- meson.build.orig	2021-10-24 12:11:14.000000000 +0100
+++ meson.build	2021-10-24 12:11:45.000000000 +0100
@@ -47,7 +47,7 @@
     libm = cc.find_library('m', required : false)
     libdl = cc.find_library('dl', required : false)
     threads_dep = dependency('threads')
-    lua_dep = dependency('lua5.2', fallback: ['lua', 'lua_dep'],
+    lua_dep = dependency('lua-5.2', fallback: ['lua', 'lua_dep'],
         default_options: ['shared=false', 'use_readline=false', 'app=false']
     )
     pcre2_dep = dependency('libpcre2-8')
