--- meson.build.orig	2021-07-24 17:23:10.000000000 +0100
+++ meson.build	2021-07-24 17:23:08.000000000 +0100
@@ -8,7 +8,7 @@
 libm = cc.find_library('m', required : false)
 libdl = cc.find_library('dl', required : false)
 libx11 = dependency('x11', required : false)
-lua_dep = dependency('lua5.2', required : false)
+lua_dep = dependency('lua-5.2', required : false)
 
 if not lua_dep.found()
     lua_subproject = subproject('lua', default_options: ['shared=false', 'use_readline=false', 'app=false'])
@@ -22,7 +22,7 @@
     lite_docdir = 'doc'
     lite_datadir = 'data'
 else
-    lite_docdir = 'share/doc'
+    lite_docdir = 'share/doc/lite-xl'
     lite_datadir = 'share/lite-xl'
 endif
 
