--- DESCRIPTION.orig	2023-04-01 04:40:02.000000000 +0800
+++ DESCRIPTION	2023-04-28 15:31:00.000000000 +0800
@@ -26,7 +26,7 @@
         (>= 0.1.2), stringr, tibble, tidyr (>= 1.0.0), timetk (>=
         2.4.0), timeDate, TTR, xts, rlang, tidyverse, tidyselect
 Suggests: tibbletime, forcats, broom, knitr, rmarkdown, testthat (>=
-        2.1.0), scales, Rblpapi, covr, janitor
+        2.1.0), scales, covr, janitor
 RoxygenNote: 7.2.3
 VignetteBuilder: knitr
 NeedsCompilation: no
