--- configure.orig	2026-09-17 02:00:11
+++ configure	2026-09-18 23:58:47
@@ -655,7 +655,7 @@
 #line 656 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(void) { return 0; }
 EOF
 if { (eval echo configure:661: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -1059,7 +1059,7 @@
 #line 1060 "configure"
 #include "confdefs.h"
 
-int main() {
+int main(void) {
 ${x_direct_test_function}()
 ; return 0; }
 EOF
@@ -1175,9 +1175,7 @@
 #line 1176 "configure"
 #include "confdefs.h"
 
-int main() {
-
-; return 0; }
+int main(void) { return 0; }
 EOF
 if { (eval echo configure:1183: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1198,9 +1196,7 @@
 #line 1199 "configure"
 #include "confdefs.h"
 
-int main() {
-
-; return 0; }
+int main(void) { return 0; }
 EOF
 if { (eval echo configure:1206: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1249,9 +1245,7 @@
     builtin and then its argument prototype would still apply.  */
 char dnet_ntoa();
 
-int main() {
-dnet_ntoa()
-; return 0; }
+int main(void) { dnet_ntoa(); return 0; }
 EOF
 if { (eval echo configure:1257: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1290,9 +1284,7 @@
     builtin and then its argument prototype would still apply.  */
 char dnet_ntoa();
 
-int main() {
-dnet_ntoa()
-; return 0; }
+int main(void) { dnet_ntoa(); return 0; }
 EOF
 if { (eval echo configure:1298: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1338,7 +1330,7 @@
     builtin and then its argument prototype would still apply.  */
 char gethostbyname();
 
-int main() {
+int main(void) {
 
 /* The GNU C library defines this for functions which it implements
     to always fail with ENOSYS.  Some functions are actually named
@@ -1387,9 +1379,7 @@
     builtin and then its argument prototype would still apply.  */
 char gethostbyname();
 
-int main() {
-gethostbyname()
-; return 0; }
+int main(void) { gethostbyname(); return 0; }
 EOF
 if { (eval echo configure:1395: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1436,7 +1426,7 @@
     builtin and then its argument prototype would still apply.  */
 char connect();
 
-int main() {
+int main(void) {
 
 /* The GNU C library defines this for functions which it implements
     to always fail with ENOSYS.  Some functions are actually named
@@ -1485,9 +1475,7 @@
     builtin and then its argument prototype would still apply.  */
 char connect();
 
-int main() {
-connect()
-; return 0; }
+int main(void) { connect(); return 0; }
 EOF
 if { (eval echo configure:1493: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1528,7 +1516,7 @@
     builtin and then its argument prototype would still apply.  */
 char remove();
 
-int main() {
+int main(void) {
 
 /* The GNU C library defines this for functions which it implements
     to always fail with ENOSYS.  Some functions are actually named
@@ -1577,9 +1565,7 @@
     builtin and then its argument prototype would still apply.  */
 char remove();
 
-int main() {
-remove()
-; return 0; }
+int main(void) { remove(); return 0; }
 EOF
 if { (eval echo configure:1585: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1620,7 +1606,7 @@
     builtin and then its argument prototype would still apply.  */
 char shmat();
 
-int main() {
+int main(void) {
 
 /* The GNU C library defines this for functions which it implements
     to always fail with ENOSYS.  Some functions are actually named
@@ -1669,9 +1655,7 @@
     builtin and then its argument prototype would still apply.  */
 char shmat();
 
-int main() {
-shmat()
-; return 0; }
+int main(void) { shmat(); return 0; }
 EOF
 if { (eval echo configure:1677: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1721,9 +1705,7 @@
     builtin and then its argument prototype would still apply.  */
 char IceConnectionNumber();
 
-int main() {
-IceConnectionNumber()
-; return 0; }
+int main(void) { IceConnectionNumber(); return 0; }
 EOF
 if { (eval echo configure:1729: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1752,9 +1734,9 @@
 LIBS="$LIBS $X_LIBS"
 CFLAGS="$CFLAGS $X_CFLAGS"
 
-echo $ac_n "checking for main in -lX11""... $ac_c" 1>&6
-echo "configure:1757: checking for main in -lX11" >&5
-ac_lib_var=`echo X11'_'main | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for linker success using -lX11""... $ac_c" 1>&6
+echo "configure:1757: checking for linker success using -lX11" >&5
+ac_lib_var=`echo X11'_'link | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
@@ -1764,9 +1746,7 @@
 #line 1765 "configure"
 #include "confdefs.h"
 
-int main() {
-main()
-; return 0; }
+int main(void) { return 0; }
 EOF
 if { (eval echo configure:1772: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1795,9 +1775,9 @@
   echo "$ac_t""no" 1>&6
 fi
 
-echo $ac_n "checking for main in -lm""... $ac_c" 1>&6
-echo "configure:1800: checking for main in -lm" >&5
-ac_lib_var=`echo m'_'main | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for linker success using -lm""... $ac_c" 1>&6
+echo "configure:1800: checking for linker success using -lm" >&5
+ac_lib_var=`echo m'_'libm | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
@@ -1807,9 +1787,7 @@
 #line 1808 "configure"
 #include "confdefs.h"
 
-int main() {
-main()
-; return 0; }
+int main(void) { return 0; }
 EOF
 if { (eval echo configure:1815: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
@@ -1850,9 +1828,8 @@
 #line 1851 "configure"
 #include "confdefs.h"
 
-int main() {
-} $ac_kw foo() {
-; return 0; }
+int main(void) { return 0; }
+$ac_kw int foo(void) { return 0; }
 EOF
 if { (eval echo configure:1858: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
   rm -rf conftest*
@@ -1956,7 +1933,7 @@
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
-int main () { int i; for (i = 0; i < 256; i++)
+int main(void) { int i; for (i = 0; i < 256; i++)
 if (XOR (islower (i), ISLOWER (i)) || toupper (i) != TOUPPER (i)) exit(2);
 exit (0); }
 
@@ -1995,9 +1972,7 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <time.h>
-int main() {
-struct tm *tp;
-; return 0; }
+int main(void) { struct tm *tp; return 0; }
 EOF
 if { (eval echo configure:2002: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
   rm -rf conftest*
@@ -2030,9 +2005,7 @@
 #include "confdefs.h"
 #include <sys/types.h>
 #include <time.h>
-int main() {
-struct tm *tp; tp->tm_sec;
-; return 0; }
+int main(void) { struct tm *tp; tp->tm_sec; return 0; }
 EOF
 if { (eval echo configure:2037: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
   rm -rf conftest*
@@ -2064,9 +2037,7 @@
 #include "confdefs.h"
 #include <sys/types.h>
 #include <time.h>
-int main() {
-struct tm tm; tm.tm_gmtoff;
-; return 0; }
+int main(void) { struct tm tm; tm.tm_gmtoff; return 0; }
 EOF
 if { (eval echo configure:2071: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
   rm -rf conftest*
