--- DESCRIPTION.orig	2023-05-08 17:20:02.000000000 +0800
+++ DESCRIPTION	2023-05-08 21:17:35.000000000 +0800
@@ -11,7 +11,7 @@
   time series classes: 'ts','xts', 'data.frame', 'data.table', 'tibble', 'zoo',
   'timeSeries', 'tsibble', 'tis' or 'irts'. Also converts reliably between these classes.
 Imports: data.table (>= 1.10.0), anytime
-Suggests: testthat, dplyr, tibble, tidyr, forecast, seasonal, dygraphs,
+Suggests: testthat, dplyr, tibble, tidyr, forecast, dygraphs,
         xts, ggplot2, scales, knitr, rmarkdown, tsibble (>= 0.8.2),
         tsibbledata, tibbletime, tseries, units, zoo, tis, timeSeries,
         nycflights13, imputeTS, spelling, covr
