--- meson.build.orig	2019-03-13 16:59:12.000000000 -0700
+++ meson.build	2019-03-13 17:03:58.000000000 -0700
@@ -185,9 +185,10 @@
 
 # Will automatically pick it up from the cross file if defined
 gio_querymodules = find_program('gio-querymodules', required : false)
-if gio_querymodules.found()
-  meson.add_install_script('meson_post_install.py', gio_querymodules.path(), gio_module_dir)
-endif
+# don't run gio-querymodules during destroot -- we'll do it in post-activate
+#if gio_querymodules.found()
+#  meson.add_install_script('meson_post_install.py', gio_querymodules.path(), gio_module_dir)
+#endif
 
 output = '\n\n'
 output += '  gnutls support:      ' + backends.contains('gnutls').to_string() + '\n'
