--- tls/gnutls/gtlsconnection-gnutls.c	2012-06-22 16:10:11.000000000 +0200
+++ tls/gnutls/gtlsconnection-gnutls.c	2011-05-23 21:40:54.000000000 +0200
@@ -974,6 +974,7 @@
       g_simple_async_result_set_op_res_gboolean (simple, TRUE);
       g_simple_async_result_complete_in_idle (simple);
       g_object_unref (simple);
+      return;
     }
   else if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_WOULD_BLOCK))
     {
@@ -981,7 +982,11 @@
       g_error_free (error);
       g_simple_async_result_complete_in_idle (simple);
       g_object_unref (simple);
+      return;
     }
+  else if (error)
+    g_error_free (error);
+    
 
   source = g_tls_connection_gnutls_create_source (gnutls, 0, cancellable);
   g_source_set_callback (source,
