--- t/CairoFt.t.orig	2019-10-07 17:38:01.000000000 -0700
+++ t/CairoFt.t	2019-10-07 17:43:13.000000000 -0700
@@ -19,8 +19,8 @@
 }
 
 my @files_to_try = qw(
-  /usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf
-  /usr/share/fonts/truetype/ttf-dejavu/DejaVuSerif.ttf
+  @@MP_PREFIX@@/share/fonts/Vera.ttf
+  @@MP_PREFIX@@/share/fonts/dejavu-fonts/DejaVuSerif.ttf
 );
 my @files_found = grep { -r $_ } @files_to_try;
 my $file = $files_found[0];
