--- magic/Magdir/mach	2012-11-06 12:37:19.000000000 -0800
+++ magic/Magdir/mach	2009-09-19 09:28:10.000000000 -0700
@@ -1,133 +1,86 @@
 
 #------------------------------------------------------------
-# $File: mach,v 1.12 2012/11/01 04:26:40 christos Exp $
+# $File: mach,v 1.9 2009/09/19 16:28:10 christos Exp $
 # Mach has two magic numbers, 0xcafebabe and 0xfeedface.
 # Unfortunately the first, cafebabe, is shared with
 # Java ByteCode, so they are both handled in the file "cafebabe".
 # The "feedface" ones are handled herein.
 #------------------------------------------------------------
-
-0	name		mach-o-bed
+0	lelong&0xfffffffe	0xfeedface	Mach-O
 >0	byte		0xcf		64-bit
+>12	lelong		1		object
+>12	lelong		2		executable
+>12	lelong		3		fixed virtual memory shared library
+>12	lelong		4		core
+>12	lelong		5		preload executable
+>12	lelong		6		dynamically linked shared library
+>12	lelong		7		dynamic linker
+>12	lelong		8		bundle
+>12	lelong		9		dynamically linked shared library stub
+>12	lelong		>9
+>>12	lelong		x		filetype=%ld
+>4	lelong		<0
+>>4	lelong		x		architecture=%ld
+>4	lelong		1		vax
+>4	lelong		2		romp
+>4	lelong		3		architecture=3
+>4	lelong		4		ns32032
+>4	lelong		5		ns32332
+>4	lelong		6		m68k
+>4	lelong		7		i386
+>4	lelong		8		mips
+>4	lelong		9		ns32532
+>4	lelong		10		architecture=10
+>4	lelong		11		hppa
+>4	lelong		12		acorn
+>4	lelong		13		m88k
+>4	lelong		14		sparc
+>4	lelong		15		i860-big
+>4	lelong		16		i860
+>4	lelong		17		rs6000
+>4	lelong		18		ppc
+>4	lelong		16777234	ppc64
+>4	lelong		>16777234
+>>4	lelong		x		architecture=%ld
+#
+0	belong&0xfffffffe	0xfeedface	Mach-O
+>3	byte		0xcf		64-bit
+>12	belong		1		object
+>12	belong		2		executable
+>12	belong		3		fixed virtual memory shared library
+>12	belong		4		core
+>12	belong		5		preload executable
+>12	belong		6               dynamically linked shared library
+>12	belong		7               dynamic linker
+>12	belong		8		bundle
+>12	belong		9		dynamically linked shared library stub
+>12	belong		>9
+>>12	belong		x		filetype=%ld
 >4	belong		<0
 >>4	belong		x		architecture=%ld
->4	belong		1
->>8	belong		0		vax
->>8	belong		1		vax11/780
->>8	belong		2		vax11/785
->>8	belong		3		vax11/750
->>8	belong		4		vax11/730
->>8	belong		5		uvaxI
->>8	belong		6		uvaxII
->>8	belong		7		vax8200
->>8	belong		8		vax8500
->>8	belong		9		vax8600
->>8	belong		10		vax8650
->>8	belong		11		vax8800
->>8	belong		12		uvaxIII
+>4	belong		1		vax
 >4	belong		2		romp
 >4	belong		3		architecture=3
 >4	belong		4		ns32032
 >4	belong		5		ns32332
->4	belong		6		m68k
->4	belong		7
->>8	belong		3		i386
->>8	belong		4		i486
->>8	belong		132		i486sx
->>8	belong		5		i586
->>8	belong		8		pentium_3
->>8	belong		9		pentiumM
->>8	belong		10		pentiumIV
->>8	belong		11		itanium
->>8	belong		12		xeon
->>8	belong		14		pentiumIV_M
->>8	belong		15		itanium2
->>8	belong		21		pentiumpro
->>8	belong		23		pentiumIII
->>8	belong		24		pentium_3_M
->>8	belong		25		pentiumV
->>8	belong		26		xeon_mp
->>8	belong		40		pentium_3_xeon
->>8	belong		113		celeron
->>8	belong		119		celeron_mobile
+>4	belong		6		for m68k architecture
+# from NeXTstep 3.0 <mach/machine.h>
+# i.e. mc680x0_all, ignore
+# >>8	belong		1		(mc68030)
+>>8	belong		2		(mc68040)
+>>8	belong		3		(mc68030 only)
+>4	belong		7		i386
 >4	belong		8		mips
->>8	belong		1		R2300
->>8	belong		2		R2600
->>8	belong		3		R2800
->>8	belong		4		R2000a
->>8	belong		5		R2000
->>8	belong		6		R3000a
->>8	belong		7		R3000
 >4	belong		9		ns32532
->4	belong		10		mc98000
+>4	belong		10		architecture=10
 >4	belong		11		hppa
->>8	belong		0		7100
->>8	belong		1		7100LC
->4	belong		12
->>8	belong		0		arm
->>8	belong		1		arm_v4t
->>8	belong		2		arm_v6
->>8	belong		3		arm_v5tej
->>8	belong		4		arm_xscale
->>8	belong		5		arm_v7
->>8	belong		6		arm_v7f
->>8	belong		7		arm_v7k
->4	belong		13
->>8	belong		0		mc88000
->>8	belong		1		mc88100
->>8	belong		2		mc88110
+>4	belong		12		acorn
+>4	belong		13		m88k
 >4	belong		14		sparc
->4	belong		15		i860g
->4	belong		16		alpha
+>4	belong		15		i860-big
+>4	belong		16		i860
 >4	belong		17		rs6000
->4	belong		18
->>8	belong		0		ppc
->>8	belong		1		ppc_601
->>8	belong		2		ppc_602
->>8	belong		3		ppc_603
->>8	belong		4		ppc_603e
->>8	belong		5		ppc_603ev
->>8	belong		6		ppc_604
->>8	belong		7		ppc_604e
->>8	belong		8		ppc_620
->>8	belong		9		ppc_650
->>8	belong		10		ppc_7400
->>8	belong		11		ppc_7450
->>8	belong		100		ppc_970
->4	belong		16777223
->>8	belong		3		x86_64
->>8	belong		4		x86_64_arch1
->4	belong		16777234
->>8	belong		0		ppc64
->>8	belong		1		ppc64_601
->>8	belong		2		ppc64_602
->>8	belong		3		ppc64_603
->>8	belong		4		ppc64_603e
->>8	belong		5		ppc64_603ev
->>8	belong		6		ppc64_604
->>8	belong		7		ppc64_604e
->>8	belong		8		ppc64_620
->>8	belong		9		ppc64_650
->>8	belong		10		ppc64_7400
->>8	belong		11		ppc64_7450
->>8	belong		100		ppc64_970
+>4	belong		18		ppc
+>4	belong		16777234	ppc64
 >4	belong		>16777234
 >>4	belong		x		architecture=%ld
->12	belong		1		object
->12	belong		2		executable
->12	belong		3		fixed virtual memory shared library
->12	belong		4		core
->12	belong		5		preload executable
->12	belong		6		dynamically linked shared library
->12	belong		7		dynamic linker
->12	belong		8		bundle
->12	belong		9		dynamically linked shared library stub
->12	belong		>9
->>12	belong		x		filetype=%ld
-
-#
-0	lelong&0xfffffffe	0xfeedface	Mach-O 
->0	use	\^mach-o-be
-
-0	belong&0xfffffffe	0xfeedface	Mach-O 
->0	use	mach-o-be
