--- src/readcdf.c.orig	2012-06-20 15:18:33.000000000 -0700
+++ src/readcdf.c	2013-02-10 17:03:26.000000000 -0800
@@ -136,7 +136,7 @@ cdf_file_property_info(struct magic_set 
                                         char *c, *ec;
                                         cdf_timestamp_to_timespec(&ts, tp);
                                         c = cdf_ctime(&ts.tv_sec, tbuf);
-                                        if ((ec = strchr(c, '\n')) != NULL)
+                                        if (c != NULL && (ec = strchr(c, '\n')) != NULL)
                                                 *ec = '\0';
 
                                         if (NOTMIME(ms) && file_printf(ms,
