--- magic/Magdir/msdos.orig	2008-10-22 10:44:44.000000000 -0700
+++ magic/Magdir/msdos	2008-10-22 10:46:35.000000000 -0700
@@ -671,8 +671,11 @@
 0	string	\224\246\056		Microsoft Word Document
 !:mime	application/msword
 
-512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	Microsoft Word Document
-!:mime	application/msword
+## this is not completely accurate, there are Excel documents that have the same pattern
+## 512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	Microsoft Word Document
+## !:mime	application/msword
+512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	Microsoft Office Document
+!:mime	application/x-msoffice
 
 # From: "Nelson A. de Oliveira" <naoliv@gmail.com>
 # Magic type for Dell's BIOS .hdr files
