--- Makefile.in.orig	2009-06-28 14:59:36.000000000 +0900
+++ Makefile.in	2009-06-28 15:00:53.000000000 +0900
@@ -316,8 +316,8 @@
 # debugging. 
 CXXFLAGS_DEBUG = -g3 -O0 -Wall -fno-defer-pop
 TEST_COV_FLAGS = -ftest-coverage -fprofile-arcs
-SUBDIRS = . ncdump
-DIST_SUBDIRS = nc_test unit-tests ncdump 
+SUBDIRS = .
+DIST_SUBDIRS = nc_test unit-tests
 lib_LTLIBRARIES = libnc-dap.la
 bin_SCRIPTS = ncdap-config ncdap-config-pkgconfig
 LIBSRC = NCArray.cc NCByte.cc NCFloat64.cc NCGrid.cc NCUInt32.cc	 \
