--- src/libical/icaltz-util.c.orig	2009-01-08 08:50:21.000000000 -0800
+++ src/libical/icaltz-util.c	2009-05-30 20:24:36.000000000 -0700
@@ -41,26 +41,26 @@
 # endif
 # ifdef HAVE_ENDIAN_H
 #  include <endian.h>
 # endif 
+# if !defined(HAVE_BYTESWAP_H) && !defined(HAVE_SYS_ENDIAN_H) && !defined(HAVE_ENDIAN_H)
+#  define bswap_16(x) (((x) << 8) & 0xff00) | (((x) >> 8 ) & 0xff)
+#  define bswap_32(x) (((x) << 24) & 0xff000000)  \
+                      | (((x) << 8) & 0xff0000)   \
+                      | (((x) >> 8) & 0xff00)     \
+                      | (((x) >> 24) & 0xff )
+#  define bswap_64(x) ((((x) & 0xff00000000000000ull) >> 56) \
+                      | (((x) & 0x00ff000000000000ull) >> 40) \
+                      | (((x) & 0x0000ff0000000000ull) >> 24) \
+                      | (((x) & 0x000000ff00000000ull) >> 8) \
+                      | (((x) & 0x00000000ff000000ull) << 8) \
+                      | (((x) & 0x0000000000ff0000ull) << 24) \
+                      | (((x) & 0x000000000000ff00ull) << 40) \
+                      | (((x) & 0x00000000000000ffull) << 56))
+# endif
 #endif
 
 #ifdef WIN32
-#if !defined(HAVE_BYTESWAP_H) && !defined(HAVE_SYS_ENDIAN_H) && !defined(HAVE_ENDIAN_H)
-#define bswap_16(x) (((x) << 8) & 0xff00) | (((x) >> 8 ) & 0xff)
-#define bswap_32(x) (((x) << 24) & 0xff000000)  \
-                    | (((x) << 8) & 0xff0000)   \
-                    | (((x) >> 8) & 0xff00)     \
-                    | (((x) >> 24) & 0xff )
-#define bswap_64(x) ((((x) & 0xff00000000000000ull) >> 56) \
-                    | (((x) & 0x00ff000000000000ull) >> 40) \
-                    | (((x) & 0x0000ff0000000000ull) >> 24) \
-                    | (((x) & 0x000000ff00000000ull) >> 8) \
-                    | (((x) & 0x00000000ff000000ull) << 8) \
-                    | (((x) & 0x0000000000ff0000ull) << 24) \
-                    | (((x) & 0x000000000000ff00ull) << 40) \
-                    | (((x) & 0x00000000000000ffull) << 56))
-#endif
 #include <io.h>
 #endif
 
 #ifndef PATH_MAX
