--- cli-plugins/manager/manager_unix.go
+++ cli-plugins/manager/manager_unix.go
@@ -5,5 +5,5 @@
 
 var defaultSystemPluginDirs = []string{
 	"/usr/local/lib/docker/cli-plugins", "/usr/local/libexec/docker/cli-plugins",
-	"/usr/lib/docker/cli-plugins", "/usr/libexec/docker/cli-plugins",
+	"__PREFIX__/lib/docker/cli-plugins", "__PREFIX__/libexec/docker/cli-plugins",
 }

Diff finished.  Fri Aug  4 19:09:34 2023
