--- AMD/Makefile.orig	2011-07-25 11:00:33.000000000 +1000
+++ AMD/Makefile	2011-07-25 11:00:51.000000000 +1000
@@ -6,6 +6,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library fortran cross clean purge distclean doc dist ccode lib mex
+
 # Compile all C code, including the C-callable routines.
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
--- CAMD/Makefile.orig	2007-05-06 00:28:33.000000000 +1000
+++ CAMD/Makefile	2011-07-25 11:03:59.000000000 +1000
@@ -6,6 +6,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library clean purge distclean doc dist ccode lib mex
+
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
--- CCOLAMD/Makefile.orig	2007-05-05 23:57:17.000000000 +1000
+++ CCOLAMD/Makefile	2011-07-25 11:06:00.000000000 +1000
@@ -6,6 +6,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library clean purge distclean dist ccode lib mex
+
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not the MATLAB interface.
 demo:
--- COLAMD/Makefile.orig	2007-05-05 23:40:14.000000000 +1000
+++ COLAMD/Makefile	2011-07-25 11:08:03.000000000 +1000
@@ -6,6 +6,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library clean purge distclean dist ccode lib mex
+
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not the MATLAB interface.
 demo:
--- LDL/Makefile.orig	2007-05-06 03:24:14.000000000 +1000
+++ LDL/Makefile	2011-07-25 11:11:26.000000000 +1000
@@ -6,6 +6,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library clean purge distclean doc dist ccode lib mex
+
 # Compile all C code, including the C-callable routine.
 demo:
 	( cd Lib    ; $(MAKE) )
--- SPQR/Makefile.orig	2008-08-21 05:59:59.000000000 +1000
+++ SPQR/Makefile	2011-07-25 11:13:36.000000000 +1000
@@ -4,6 +4,8 @@
 
 include ../UFconfig/UFconfig.mk
 
+.PHONY : default all demo library cov clean purge distclean ccode mex
+
 default: all
 
 all:
