--- SuiteSparse_config/SuiteSparse_config_Mac.mk.orig	2013-10-24 11:47:48.000000000 -0400
+++ SuiteSparse_config/SuiteSparse_config_Mac.mk	2013-10-24 12:04:00.000000000 -0400
@@ -84,8 +84,8 @@
 # LIB = -lm
 
 # For "make install"
-INSTALL_LIB = /usr/local/lib
-INSTALL_INCLUDE = /usr/local/include
+INSTALL_LIB = @DESTROOT@@PREFIX@/lib
+INSTALL_INCLUDE = @DESTROOT@@PREFIX@/include
 
 # Which version of MAKE you are using (default is "make")
 # MAKE = make
@@ -108,8 +108,8 @@
 # naming the BLAS and LAPACK library (*.a or *.so) files.
 
 # This is probably slow ... it might connect to the Standard Reference BLAS:
-BLAS = -lblas -lgfortran
-LAPACK = -llapack
+# BLAS = -lblas -lgfortran
+# LAPACK = -llapack
 
 # NOTE: this next option for the "Goto BLAS" has nothing to do with a "goto"
 # statement.  Rather, the Goto BLAS is written by Dr. Kazushige Goto.
@@ -164,8 +164,8 @@
 # The path is relative to where it is used, in CHOLMOD/Lib, CHOLMOD/MATLAB, etc.
 # You may wish to use an absolute path.  METIS is optional.  Compile
 # CHOLMOD with -DNPARTITION if you do not wish to use METIS.
-METIS_PATH = ../../metis-4.0
-METIS = ../../metis-4.0/libmetis.a
+METIS_PATH = @METIS_PATH@
+METIS = @METIS@
 
 #------------------------------------------------------------------------------
 # UMFPACK configuration:
@@ -225,7 +225,7 @@
 # -DNSUNPERF	    for Solaris only.  If defined, do not use the Sun
 #			Performance Library
 
-CHOLMOD_CONFIG = $(GPU_CONFIG)
+CHOLMOD_CONFIG = @CHOLMOD_CONFIG@
 
 # uncomment this line to compile CHOLMOD without METIS:
 # CHOLMOD_CONFIG = -DNPARTITION
@@ -313,9 +313,9 @@
 # on my MacBook Air, and they work fine.
 
   F77 = gfortran
-  CF = $(CFLAGS) -O3 -fno-common -fexceptions -DNTIMER
-  BLAS = -framework Accelerate
-  LAPACK = -framework Accelerate
+  CF = @ARCHES@ $(CFLAGS) -O3 -fno-common -fexceptions -DNTIMER
+  BLAS = @BLAS_LAPACK@
+  LAPACK = @BLAS_LAPACK@
   LIB = -lm
 
 #------------------------------------------------------------------------------
