--- base/stdpre.h.orig	2007-09-25 22:31:24.000000000 +0900
+++ base/stdpre.h	2010-03-16 23:06:55.000000000 +0900
@@ -17,9 +17,6 @@
 #ifndef stdpre_INCLUDED
 #  define stdpre_INCLUDED
 
-/* Ghostscript uses transitional LFS functions. */
-#define _LARGEFILE64_SOURCE 1
-
 /*
  * Here we deal with the vagaries of various C compilers.  We assume that:
  *      ANSI-standard Unix compilers define __STDC__.
