--- base/macosx.mak.orig	2008-10-03 04:33:22.000000000 +0900
+++ base/macosx.mak	2009-02-10 08:39:57.000000000 +0900
@@ -54,7 +54,7 @@
 
 FRAMEWORK_NAME=Ghostscript
 FRAMEWORK_EXT=.framework
-prefix = /Library/Frameworks/$(FRAMEWORK_NAME)$(FRAMEWORK_EXT)
+prefix = @@FRAMEWORKS_DIR@@/$(FRAMEWORK_NAME)$(FRAMEWORK_EXT)
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 scriptdir = $(bindir)
@@ -159,7 +159,7 @@
 # what its name is (usually libz, but sometimes libgz).
 # See gs.mak and Make.htm for more information.
 
-SHARE_ZLIB=1
+SHARE_ZLIB=0
 #ZLIB_NAME=gz
 ZLIB_NAME=z
 
