--- Sources/TSDocument.m	2012-01-04 14:20:33.000000000 +0200
+++ Sources/TSDocument.m	2012-01-04 14:30:23.000000000 +0200
@@ -1094,7 +1094,8 @@
 				if ([theChecker setLanguage:spellcheckString]) {
 					spellLanguageChanged = YES;
 					if ([theChecker respondsToSelector:@selector(setAutomaticallyIdentifiesLanguages:)])
-						automaticLanguage = [theChecker setAutomaticallyIdentifiesLanguages:NO];
+						automaticLanguage = NO;
+						[theChecker setAutomaticallyIdentifiesLanguages:NO];
 					if (spellLanguage != nil)
 						[spellLanguage release];
 					spellLanguage = [spellcheckString retain];
@@ -1311,12 +1312,14 @@
 	if (spellLanguage != nil) {
 		[theChecker setLanguage:spellLanguage]; 
 		if ([theChecker respondsToSelector:@selector(setAutomaticallyIdentifiesLanguages:)])
-			automaticLanguage = [theChecker setAutomaticallyIdentifiesLanguages:NO];
+			automaticLanguage = NO;
+			[theChecker setAutomaticallyIdentifiesLanguages:NO];
 	}
 	else {
 		[theChecker setLanguage:defaultLanguage]; 
 		if ([theChecker respondsToSelector:@selector(setAutomaticallyIdentifiesLanguages:)])
-			automaticLanguage = [theChecker setAutomaticallyIdentifiesLanguages:automaticLanguage];
+			automaticLanguage = NO;
+			[theChecker setAutomaticallyIdentifiesLanguages:automaticLanguage];
 	}
 }
 
