--- Sources/TSPreferences.h.orig	2022-07-13 13:22:08
+++ Sources/TSPreferences.h	2023-11-01 19:04:57
@@ -133,8 +133,8 @@
 	IBOutlet NSTextField *_bibTeXengineField; // added by Terada
 //	IBOutlet NSButton *_makeatletterButton; // added by Terada
     
-    IBOutlet NSButton  *_sparkleAutomaticButton;
-    IBOutlet NSMatrix  *_sparkleIntervalMatrix;
+//    IBOutlet NSButton  *_sparkleAutomaticButton;
+//    IBOutlet NSMatrix  *_sparkleIntervalMatrix;
     
     IBOutlet NSButton  *_useNewToolbarButton;
     IBOutlet NSButton  *_useNewToolbarIconsButton;
@@ -164,8 +164,8 @@
 	BOOL			bibDeskCompleteTouched;
 	BOOL			oldAutoComplete;
 	BOOL			oldBibDeskComplete;
-    BOOL            oldSparkleAutomaticUpdate;
-    NSInteger       oldSparkleInterval;
+//     BOOL            oldSparkleAutomaticUpdate;
+//     NSInteger       oldSparkleInterval;
     BOOL            oldNewToolbarIcons;
 	BOOL			magnificationTouched;
 	BOOL			externalEditorTouched;
@@ -176,7 +176,7 @@
 	BOOL            kpsetoolTouched; // added by Terada
 	BOOL            bibTeXengineTouched; // added by Terada
 //	BOOL            makeatletterTouched; // added by Terada
-    BOOL            sparkleTouched;
+//     BOOL            sparkleTouched;
     BOOL            newToolbarIconsTouched;
     BOOL            xmlTagsTouched;
    
@@ -365,8 +365,8 @@
 - (IBAction)bibTeXengineChanged:sender; // added by Terada
 // - (IBAction)makeatletterChanged:sender; // added by Terada
 
-- (IBAction)sparkleAutomaticCheck:sender;
-- (IBAction)sparkleInterval:sender;
+// - (IBAction)sparkleAutomaticCheck:sender;
+// - (IBAction)sparkleInterval:sender;
 
 - (IBAction)NewToolbarIconsCheck:sender;
 
--- Sources/TSPreferences.m.orig	2022-07-13 13:22:08
+++ Sources/TSPreferences.m	2023-11-01 19:04:14
@@ -33,7 +33,7 @@
 #import "TSAppDelegate.h" // mitsu 1.29 (O)
 #import "TSDocument.h"
 #import "TSConsoleWindow.h"
-#import <Sparkle/SUUpdater.h> 
+//#import <Sparkle/SUUpdater.h> 
 
 //#import "MyPDFView.h" // mitsu 1.29 (O)
 
@@ -156,7 +156,7 @@
 	kpsetoolTouched = NO; // added by Terada
 	bibTeXengineTouched = NO; // added by Terada
 //	makeatletterTouched = NO; // added by Terada
-    sparkleTouched = NO;
+//    sparkleTouched = NO;
 	// end addition
 	// prepare undo manager: forget all the old undo information and begin a new group.
 	[_undoManager removeAllActions];
@@ -1656,6 +1656,7 @@
 
 /*" Sparkle Actions 
 "*/
+/* 
 - (IBAction)sparkleAutomaticCheck:sender
 {
     sparkleTouched = YES;
@@ -1697,6 +1698,7 @@
  
     
 }
+ */
 
 
 
@@ -1712,7 +1714,7 @@
     [SUD setBool:theValue forKey:NewToolbarIconsKey];
     
     
-    [[SUUpdater sharedUpdater] setAutomaticallyChecksForUpdates: theValue ];
+//     [[SUUpdater sharedUpdater] setAutomaticallyChecksForUpdates: theValue ];
     
 }
 
@@ -2086,6 +2088,7 @@
 	}
     
   
+/* 
     if (sparkleTouched) {
         [[SUUpdater sharedUpdater] setAutomaticallyChecksForUpdates: oldSparkleAutomaticUpdate];
         
@@ -2101,6 +2104,7 @@
                 break;
         }
     }
+ */
    
     
 	// added by mitsu --(G) TSEncodingSupport
@@ -2265,9 +2269,9 @@
 	[_convertUTFButton setState:[defaults boolForKey:AutomaticUTF8MACtoUTF8ConversionKey]];
     [_openRootFileButton  setState:[defaults boolForKey:AutoOpenRootFileKey]];
     [_miniaturizeRootFileButton setState:[defaults boolForKey:MiniaturizeRootFileKey]];
-    [_sparkleAutomaticButton setState: [defaults boolForKey: SparkleAutomaticUpdateKey]];
-    [_sparkleIntervalMatrix setEnabled: [defaults boolForKey: SparkleAutomaticUpdateKey]];
-    [_sparkleIntervalMatrix selectCellWithTag: [defaults integerForKey: SparkleIntervalKey]];
+//     [_sparkleAutomaticButton setState: [defaults boolForKey: SparkleAutomaticUpdateKey]];
+//     [_sparkleIntervalMatrix setEnabled: [defaults boolForKey: SparkleAutomaticUpdateKey]];
+//     [_sparkleIntervalMatrix selectCellWithTag: [defaults integerForKey: SparkleIntervalKey]];
     [_useNewToolbarIconsButton setState: [defaults boolForKey: NewToolbarIconsKey]];
     [_consoleMatrix selectCellWithTag: [defaults integerForKey: ConsoleBehaviorKey]];
     [_spellCheckCommands setState:[defaults boolForKey:TurnOffCommandSpellCheckKey]];
