--- DESCRIPTION.orig	2023-03-08 20:30:02.000000000 +0800
+++ DESCRIPTION	2023-04-22 19:28:12.000000000 +0800
@@ -22,7 +22,7 @@
 License: GPL (>= 2)
 Imports: methods, Rcpp (>= 0.11.0), RApiSerialize
 LinkingTo: Rcpp, RApiSerialize
-Suggests: RcppMsgPack, rredis, tinytest
+Suggests: RcppMsgPack, tinytest
 Additional_repositories: https://ghrr.github.io/drat
 NeedsCompilation: yes
 Packaged: 2023-03-08 12:06:55 UTC; edd
