--- configure.orig	2023-03-12 04:38:02.000000000 +0800
+++ configure	2024-07-30 13:55:21.000000000 +0800
@@ -6021,9 +6021,9 @@
 
   if test "x$BUNDLE" != x; then
     if test "x$MODULE_XLIBS" = x; then
-      MODULE_XLIBS="$BUNDLE"
+      MODULE_XLIBS=" "
     else
-      MODULE_XLIBS="$MODULE_XLIBS $BUNDLE"
+      MODULE_XLIBS="$MODULE_XLIBS"
     fi
   fi
 
@@ -6241,7 +6241,7 @@
     Darwin)
       # Mac OS X
       SHLIB_CC="$CC -fPIC"
-      SHLIB_LD="ld -bundle -undefined error"
+      SHLIB_LD="ld -shared -undefined error"
 
 $as_echo "#define BIND_8_COMPAT 1" >>confdefs.h
 
