--- Makefile.in.orig    2011-02-10 11:32:02.000000000 -0800
+++ Makefile.in 2011-02-10 11:32:28.000000000 -0800
@@ -70,7 +70,7 @@
 # Extra libraries
 # XLIBS will be linked with everything
 # MODULE_XLIBS will only be linked with the module objects
-XLIBS = @TCL_LIBS@ @LIBS@
+XLIBS = @TCL_LIBS@ @LIBS@ -framework CoreFoundation -lresolv
 MODULE_XLIBS = @MODULE_XLIBS@
 
 # You shouldn't need to edit anything below this line.

