Use LDFLAGS.
--- src/Makefile.in.orig	2018-02-04 10:08:11.000000000 -0600
+++ src/Makefile.in	2018-10-07 19:08:37.000000000 -0500
@@ -16,6 +16,7 @@
 STRIP = @STRIP@
 CFLAGS = @CFLAGS@ -I.. -I$(top_srcdir) @SSL_INCLUDES@ @DEFS@ $(CFLGS)
 CPPFLAGS = @CPPFLAGS@
+LDFLAGS = @LDFLAGS@
 
 eggdrop_objs = bg.o botcmd.o botmsg.o botnet.o chanprog.o cmds.o dcc.o \
 dccutil.o dns.o flags.o language.o match.o main.o mem.o misc.o misc_file.o \
@@ -41,7 +42,7 @@
 	touch mod/mod.xlibs
 
 link:
-	$(LD) -o ../$(EGGEXEC) $(eggdrop_objs) $(MODOBJS) $(XLIBS) md5/md5c.o compat/*.o `cat mod/mod.xlibs`
+	$(LD) $(LDFLAGS) -o ../$(EGGEXEC) $(eggdrop_objs) $(MODOBJS) $(XLIBS) md5/md5c.o compat/*.o `cat mod/mod.xlibs`
 
 linkfinish:
 	@$(STRIP) ../$(EGGEXEC) && \
@@ -63,7 +64,7 @@
 main.o:
 	$(CC) $(CFLAGS) $(CPPFLAGS) \
 	'-DCCFLAGS="$(CC) $(CFLAGS) $(CPPFLAGS)"' \
-	'-DLDFLAGS="$(LD)"' \
+	'-DLDFLAGS="$(LD) $(LDFLAGS)"' \
 	'-DSTRIPFLAGS="$(STRIP)"' -c $(srcdir)/main.c
 
 compatibility:
--- src/mod/Makefile.in.orig	2018-02-04 10:08:11.000000000 -0600
+++ src/mod/Makefile.in	2018-10-09 03:21:12.000000000 -0500
@@ -17,6 +17,7 @@
 STRIP = @STRIP@
 CFLAGS = @CFLAGS@ -I../.. -I$(top_srcdir) @SSL_INCLUDES@ @DEFS@ $(CFLGS)
 CPPFLAGS = @CPPFLAGS@
+LDFLAGS = @LDFLAGS@
 MOD_CFLAGS = @CFLAGS@ -I. -I../../.. -I$(MOD_UPDIR)$(top_srcdir) \
   -I$(MOD_UPDIR)$(top_srcdir)/src/mod @SSL_INCLUDES@ @DEFS@ $(CFLGS)
 MOD_CPPFLAGS = @CPPFLAGS@
@@ -33,7 +34,7 @@
   --top_srcdir=$(top_srcdir)
 egg_ac_parameters = @egg_ac_parameters@
 
-MAKE_MOD = $(MAKE) 'MAKE=$(MAKE)' 'CC=$(CC)' 'LD=$(LD)' \
+MAKE_MOD = $(MAKE) 'MAKE=$(MAKE)' 'CC=$(CC)' 'LD=$(LD) $(LDFLAGS)' \
 'STRIP=$(STRIP)' 'CFLAGS=$(MOD_CFLAGS)' 'CPPFLAGS=$(MOD_CPPFLAGS)' \
 'XLIBS=$(XLIBS)' 'MOD_EXT=$(MOD_EXT)' 'SHELL=$(SHELL)' \
 'MODULE_XLIBS=$(MODULE_XLIBS)'
