diff --git a/configure.ac b/configure.ac
index 051eb139..e2a00ac8 100644
--- configure.ac
+++ configure.ac
@@ -151,8 +151,6 @@ AC_PROG_MKDIR_P
 
 PKG_PROG_PKG_CONFIG([0.20])
 
-AM_PATH_PYTHON([2.7],, [:])
-
 if [test "x$request_lib_only" = "xyes"]; then
   request_app=no
   request_hpack_tools=no
@@ -161,6 +159,7 @@ if [test "x$request_lib_only" = "xyes"]; then
 fi
 
 if [test "x$request_python_bindings" != "xno"]; then
+  AM_PATH_PYTHON([2.7],, [:])
   AX_PYTHON_DEVEL([>= '2.7'])
 fi
