--- python/Makefile.in.orig	2015-07-17 10:28:43.000000000 -0500
+++ python/Makefile.in	2015-07-19 07:37:58.000000000 -0500
@@ -524,8 +524,7 @@
 @ENABLE_PYTHON_BINDINGS_TRUE@	$(PYTHON) setup.py build
 
 @ENABLE_PYTHON_BINDINGS_TRUE@install-exec-local:
-@ENABLE_PYTHON_BINDINGS_TRUE@	test -d $(pyexecdir) || mkdir -p $(pyexecdir)
-@ENABLE_PYTHON_BINDINGS_TRUE@	PYTHONUSERBASE=$(DESTDIR)/$(prefix) $(PYTHON) setup.py install --user
+@ENABLE_PYTHON_BINDINGS_TRUE@	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
 
 @ENABLE_PYTHON_BINDINGS_TRUE@uninstall-local:
 @ENABLE_PYTHON_BINDINGS_TRUE@	rm -f $(DESTDIR)$(libdir)/python*/site-packages/nghttp2.so
