--- src/shrpx_client_handler.cc.orig	2019-03-08 06:24:40.000000000 -0600
+++ src/shrpx_client_handler.cc	2019-03-09 05:59:32.000000000 -0600
@@ -980,7 +980,7 @@
   auto http2session = get_http2_session(group, addr);
   auto dconn = std::make_unique<Http2DownstreamConnection>(http2session);
   dconn->set_client_handler(this);
-  return dconn;
+  return std::move(dconn);
 }
 
 MemchunkPool *ClientHandler::get_mcpool() { return worker_->get_mcpool(); }
