From a42253a08e3686a90e676773819f0dccd2f98fb0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 29 Feb 2024 02:26:33 +0700
Subject: [PATCH 1/3] shrpx_config.cc: if undefined, define AI_NUMERICSERV to 0

---
 src/shrpx_config.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/shrpx_config.cc src/shrpx_config.cc
index 8bc3a280..1e9cd4c5 100644
--- src/shrpx_config.cc
+++ src/shrpx_config.cc
@@ -66,6 +66,10 @@
 #include "ssl_compat.h"
 #include "xsi_strerror.h"
 
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
 namespace shrpx {
 
 namespace {

From 1942eb8ffeee28efb46a7472f002b6e0aeedb8d5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 29 Feb 2024 02:29:36 +0700
Subject: [PATCH 2/3] shrpx_tls_test.cc: use AI_NUMERICSERV when defined

---
 src/shrpx_tls_test.cc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git src/shrpx_tls_test.cc src/shrpx_tls_test.cc
index 7fe50dde..04d16daa 100644
--- src/shrpx_tls_test.cc
+++ src/shrpx_tls_test.cc
@@ -225,7 +225,10 @@ static Address parse_addr(const char *ipaddr) {
   addrinfo hints{};
 
   hints.ai_family = AF_UNSPEC;
-  hints.ai_flags = AI_NUMERICHOST | AI_NUMERICSERV;
+  hints.ai_flags = AI_NUMERICHOST;
+#ifdef AI_NUMERICSERV
+  hints.ai_flags |= AI_NUMERICSERV;
+#endif
 
   addrinfo *res = nullptr;
 
