--- src/util.c.orig	2011-09-25 08:25:23.000000000 -0500
+++ src/util.c	2011-10-01 20:54:36.000000000 -0500
@@ -199,6 +199,12 @@
       exit(1);
     }
 
+#if defined(__APPLE__)
+	if (limit.rlim_max > OPEN_MAX) {
+		limit.rlim_max = OPEN_MAX;
+	}
+#endif
+
     /* set the current to the maximum or specified value */
     if (max_desired_fds) limit.rlim_cur = max_desired_fds;
     else limit.rlim_cur = limit.rlim_max;
