--- configure.ac
+++ configure.ac	2024-12-18 18:09:12.000000000 +0800
@@ -75,17 +75,6 @@
 esac
 
 if test x"${mingw}" == "xno" ; then
-  # Bring additional directories where things might be found into our
-  # search path. I don't know why autoconf doesn't do this by default
-  for spfx in /usr/local /opt/local /sw /usr/local/ssl /usr/boost/include /usr/X11R6 ; do
-    AC_MSG_NOTICE([checking ${spfx}/include])
-    if test -d ${spfx}/include; then
-        CPPFLAGS="-I${spfx}/include $CPPFLAGS"
-        LDFLAGS="-L${spfx}/lib $LDFLAGS"
-        AC_MSG_NOTICE([ *** ADDING ${spfx}/include to CPPFLAGS *** ])
-        AC_MSG_NOTICE([ *** ADDING ${spfx}/lib to LDFLAGS *** ])
-    fi
-  done
   AC_MSG_NOTICE([ CPPFLAGS = ${CPPFLAGS} ])
   AC_MSG_NOTICE([ LDFLAGS = ${LDFLAGS} ])
 fi
