--- configure.orig	2022-12-11 09:21:24.000000000 -0800
+++ configure	2022-12-11 09:24:30.000000000 -0800
@@ -43016,6 +43016,9 @@
 int
 main ()
 {
+/* just skip this */
+exit (0);
+
 #ifndef PATH_MAX
   /* The Hurd doesn't define this, so getcwd can't exhibit the bug --
      at least not on a local file system.  And if we were to start worrying
@@ -43297,7 +43300,7 @@
   /* The bug is triggered when PATH_MAX < getpagesize (), so skip
      this relatively expensive and invasive test if that's not true.  */
 #ifdef PATH_MAX
-  int bug_possible = PATH_MAX < getpagesize ();
+  int bug_possible = 0;
 #else
   int bug_possible = 0;
 #endif
