--- ./golang.mk	2023-06-26 11:21:51.000000000 -0400
+++ ./golang.mk	2023-06-26 11:22:58.000000000 -0400
@@ -4,12 +4,12 @@
 PACKAGE=$(shell GOPATH= go list $(TARGET))
 NAME=$(notdir $(shell echo $(PACKAGE) | sed 's/\/v2//'))
 
-BUILD_VERSION=$(shell git describe --always --dirty --tags | tr '-' '.' )
-BUILD_DATE=$(shell LC_ALL=C date)
-BUILD_HASH=$(shell git rev-parse HEAD)
-BUILD_MACHINE=$(shell uname -n)
-BUILD_USER=$(shell whoami)
-BUILD_ENVIRONMENT=$(BUILD_USER)@$(BUILD_MACHINE)
+BUILD_VERSION := $(shell git describe --always --dirty --tags | tr '-' '.' )
+BUILD_DATE := $(shell LC_ALL=C date)
+BUILD_HASH := $(shell git rev-parse HEAD)
+BUILD_MACHINE := $(shell uname -n)
+BUILD_USER := $(shell whoami)
+BUILD_ENVIRONMENT := $(BUILD_USER)@$(BUILD_MACHINE)
 
 BUILD_XDST=$(PACKAGE)/cmd
 BUILD_FLAGS=-ldflags "\
