--- tools/alter/auxfunc_alter.c.orig	2016-08-13 19:29:16.000000000 +1000
+++ tools/alter/auxfunc_alter.c	2021-10-21 21:01:14.000000000 +1100
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #ifdef __MINGW32__
 #include <winsock2.h>
 #endif
--- tools/postp/auxfunc_updatelis.c.orig	2016-08-13 19:29:16.000000000 +1000
+++ tools/postp/auxfunc_updatelis.c	2021-10-21 21:09:15.000000000 +1100
@@ -78,6 +78,7 @@ void ReadKeyPrintLine(char *Result, char
 
 
 
+int DetectsTransistorColumns(char *lelement, int index[]);
 
 /*
  *
--- auxfunc_alter.c.orig	2016-08-13 19:29:16.000000000 +1000
+++ auxfunc_alter.c	2021-10-21 21:10:48.000000000 +1100
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #ifdef __MINGW32__
 #include <winsock2.h>
 #endif
