diff --git src/scripts/install.py src/scripts/install.py
index 52a1ebdf7..e3df1b20a 100755
--- src/scripts/install.py
+++ src/scripts/install.py
@@ -222,7 +222,7 @@ def main(args):
                                '-change',
                                os.path.join('@executable_path', soname_abi),
                                os.path.join(lib_dir, soname_abi),
-                               os.path.join(bin_dir, cfg['cli_exe_name'])])
+                               prepend_destdir(os.path.join(bin_dir, cfg['cli_exe_name']))])
 
     if 'botan_pkgconfig' in cfg:
         pkgconfig_dir = os.path.join(options.prefix, options.libdir, options.pkgconfigdir)
