--- src/build-data/cc/gcc	2008-12-08 02:22:23.000000000 +0100
+++ gcc	2009-01-13 12:55:00.000000000 +0100
@@ -31,7 +31,7 @@
 # AIX doesn't seem to have soname support (weird...)
 aix     -> "$(CXX) -shared -fPIC"
 
-darwin  -> "$(CXX) -dynamiclib -fPIC -install_name $(SONAME)"
+darwin  -> "$(CXX) -dynamiclib -fPIC -install_name $(LIBDIR)/$(SONAME)"
 hpux    -> "$(CXX) -shared -fPIC -Wl,+h,$(SONAME)"
 solaris -> "$(CXX) -shared -fPIC -Wl,-h,$(SONAME)"
 # Gotta use ld directly on BeOS, their GCC is busted
