diff --git src/lib/utils/os_utils.cpp src/lib/utils/os_utils.cpp
index f30f18ae9..f5a08ca4c 100644
--- src/lib/utils/os_utils.cpp
+++ src/lib/utils/os_utils.cpp
@@ -425,7 +425,7 @@ std::vector<void*> OS::allocate_locked_pages(size_t count)
 
       ptr = ::mmap(nullptr, 2*page_size,
                    PROT_READ | PROT_WRITE,
-                   MAP_ANONYMOUS | MAP_PRIVATE | MAP_NOCORE,
+                   MAP_ANON | MAP_PRIVATE | MAP_NOCORE,
                    /*fd=*/-1, /*offset=*/0);
 
       if(ptr == MAP_FAILED)
