--- src/MYOSGLUE.c.orig	2011-09-18 23:59:33.000000000 -0500
+++ src/MYOSGLUE.c	2013-05-21 01:53:34.000000000 -0500
@@ -3410,55 +3410,19 @@
 #if IncludeSonyNew
 LOCALPROC MakeNewDisk(ui5b L, CFStringRef NewDiskName)
 {
-	NavDialogRef theSaveDialog;
-	NavDialogCreationOptions dialogOptions;
-	NavReplyRecord theReply;
-	NavEventUPP gEventProc = NewNavEventUPP(NavigationEventProc);
-
-	if (noErr == NavGetDefaultDialogCreationOptions(&dialogOptions)) {
-		dialogOptions.modality = kWindowModalityAppModal;
-		dialogOptions.saveFileName = NewDiskName;
-		if (noErr == NavCreatePutFileDialog(&dialogOptions,
-			'TEXT', 'MPS ',
-			gEventProc, NULL,
-			&theSaveDialog))
-		{
-			MyBeginDialog();
-			(void) NavDialogRun(theSaveDialog);
-			MyEndDialog();
-			if (noErr == NavDialogGetReply(theSaveDialog,
-				&theReply))
-			{
-				if (theReply.validRecord) {
-					long itemsInList;
-					AEKeyword keyword;
-					DescType typeCode;
-					FSRef theRef;
-					Size actualSize;
-
-					if (noErr == AECountItems(
-						&theReply.selection, &itemsInList))
-					{
-						if (itemsInList == 1) {
-							if (noErr == AEGetNthPtr(
-								&theReply.selection, 1, typeFSRef,
-								&keyword, &typeCode, (Ptr)&theRef,
-								sizeof(FSRef), &actualSize))
+	ProcessSerialNumber currentProcess = {0, kCurrentProcess};
+	FSRef fsRef;
+	FSRef theRef;
+
+	if (noErr == GetProcessBundleLocation(&currentProcess,
+		&fsRef))
+	if (noErr == FSGetCatalogInfo(&fsRef, kFSCatInfoNone,
+		NULL, NULL, NULL, &theRef))
 							{
 								ReportStandardOpenDiskError(
 									MakeNewDisk0(&theRef,
-										theReply.saveFileName, L));
+										NewDiskName, L));
 							}
-						}
-					}
-				}
-				(void) NavDisposeReply(&theReply);
-			}
-			NavDialogDispose(theSaveDialog);
-		}
-	}
-
-	DisposeNavEventUPP(gEventProc);
 
 	if (NewDiskName != NULL) {
 		CFRelease(NewDiskName);
