--- a/Makefile.orig
+++ b/Makefile
@@ -1065,7 +1065,7 @@
 ifeq ($(COMPUTE_HEADER_DEPENDENCIES),auto)
 dep_check = $(shell $(CC) $(ALL_CFLAGS) \
 	-c -MF /dev/null -MQ /dev/null -MMD -MP \
-	-x c /dev/null -o /dev/null 2>&1; \
+	-x c /dev/null -o dummy 2>&1; \
 	echo $$?)
 ifeq ($(dep_check),0)
 override COMPUTE_HEADER_DEPENDENCIES = yes
@@ -1133,7 +1133,7 @@
 	ifdef NO_R_TO_GCC_LINKER
 		# Some gcc does not accept and pass -R to the linker to specify
 		# the runtime dynamic library path.
-		CC_LD_DYNPATH = -Wl,-rpath,
+		CC_LD_DYNPATH = -L
 	else
 		CC_LD_DYNPATH = -R
 	endif
