--- mono/interpreter/interp.h.orig	2011-11-15 11:22:44.000000000 -0600
+++ mono/interpreter/interp.h	2012-04-30 13:09:17.000000000 -0500
@@ -120,8 +120,8 @@
 void
 mono_interp_transform_init (void);
 
-void inline stackval_from_data (MonoType *type, stackval *result, char *data, gboolean pinvoke);
-void inline stackval_to_data (MonoType *type, stackval *val, char *data, gboolean pinvoke);
+void stackval_from_data (MonoType *type, stackval *result, char *data, gboolean pinvoke);
+void stackval_to_data (MonoType *type, stackval *val, char *data, gboolean pinvoke);
 void ves_exec_method (MonoInvocation *frame);
 
 /*
