--- libgc/darwin_stop_world.c.orig	2009-09-25 07:58:09.000000000 -0700
+++ libgc/darwin_stop_world.c	2010-05-06 15:13:15.000000000 -0700
@@ -2,7 +2,6 @@
 
 # if defined(GC_DARWIN_THREADS)
 
-#include <AvailabilityMacros.h>
 #include "mono/utils/mono-compiler.h"
 
 #ifdef MONO_DEBUGGER_SUPPORTED
@@ -112,7 +111,7 @@ void GC_push_all_stacks() {
 	if(r != KERN_SUCCESS) ABORT("thread_get_state failed");
 	
 #if defined(I386)
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_5
+#if __DARWIN_UNIX03
 
 	lo = state.__esp;
 
@@ -135,7 +134,7 @@ void GC_push_all_stacks() {
 	GC_push_one(state.ebp); 
 #endif
 #elif defined(POWERPC)
-#if defined(_STRUCT_PPC_EXCEPTION_STATE)
+#if __DARWIN_UNIX03
 	lo = (void*)(state.__r1 - PPC_RED_ZONE_SIZE);
         
 	GC_push_one(state.__r0); 
@@ -278,7 +277,7 @@ void GC_push_all_stacks() {
 			     (natural_t *)&info, &outCount);
 	if(r != KERN_SUCCESS) continue;
 
-#if defined(_STRUCT_PPC_EXCEPTION_STATE)
+#if __DARWIN_UNIX03
 	lo = (void*)(info.__r1 - PPC_RED_ZONE_SIZE);
 	hi = (ptr_t)FindTopOfStack(info.__r1);
 
@@ -358,7 +357,7 @@ void GC_push_all_stacks() {
 			     (natural_t *)&info, &outCount);
 	if(r != KERN_SUCCESS) continue;
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_5
+#if __DARWIN_UNIX03
 	lo = (void*)info.__esp;
 	hi = (ptr_t)FindTopOfStack(info.__esp);
 
