--- mono/mini/mini-ppc.h	2008-03-25 00:04:36.000000000 +0100
+++ mono/mini/mini-ppc.h	2008-03-25 00:04:47.000000000 +0100
@@ -178,13 +178,13 @@
 	#define UCONTEXT_REG_FPRn(ctx, n) ((ctx)->uc_mcontext.uc_regs->fpregs.fpregs [(n)])
 	#define UCONTEXT_REG_NIP(ctx)     ((ctx)->uc_mcontext.uc_regs->gregs [PT_NIP])
 	#define UCONTEXT_REG_LNK(ctx)     ((ctx)->uc_mcontext.uc_regs->gregs [PT_LNK])
-#elif defined (__APPLE__) && defined (_STRUCT_MCONTEXT)
+#elif defined (__APPLE__) && __DARWIN_UNIX03
 	typedef struct __darwin_ucontext os_ucontext;
 
 	#define UCONTEXT_REG_Rn(ctx, n)   ((ctx)->uc_mcontext->__ss.__r##n)
 	#define UCONTEXT_REG_FPRn(ctx, n) ((ctx)->uc_mcontext->__fs.__fpregs [(n)])
 	#define UCONTEXT_REG_NIP(ctx)     ((ctx)->uc_mcontext->__ss.__srr0)
-#elif defined (__APPLE__) && !defined (_STRUCT_MCONTEXT)
+#elif defined (__APPLE__) && !__DARWIN_UNIX03
 	typedef struct ucontext os_ucontext;
 
 	#define UCONTEXT_REG_Rn(ctx, n)   ((ctx)->uc_mcontext->ss.r##n)
