--- googlemock/CMakeLists.txt.orig	2023-08-24 02:44:59.000000000 +0300
+++ googlemock/CMakeLists.txt	2023-08-24 02:45:04.306946991 +0300
@@ -37,6 +37,7 @@
 # ${gmock_BINARY_DIR}.
 # Language "C" is required for find_package(Threads).
 cmake_minimum_required(VERSION 3.13)
+set(GOOGLETEST_VERSION @@GOOGLETEST_VERSION@@)
 project(gmock VERSION ${GOOGLETEST_VERSION} LANGUAGES CXX C)
 
 if (COMMAND set_up_hermetic_build)
--- googletest/CMakeLists.txt.orig	2023-08-24 02:44:41.000000000 +0300
+++ googletest/CMakeLists.txt	2023-08-24 02:44:49.524343655 +0300
@@ -47,6 +47,7 @@
 # Project version:
 
 cmake_minimum_required(VERSION 3.13)
+set(GOOGLETEST_VERSION @@GOOGLETEST_VERSION@@)
 project(gtest VERSION ${GOOGLETEST_VERSION} LANGUAGES CXX C)
 
 if (COMMAND set_up_hermetic_build)
