--- filestat.c_orig	2026-04-11 21:14:50
+++ filestat.c	2026-04-11 21:14:17
@@ -44,7 +44,7 @@
   if (file->device != s.st_dev) return 1;
   if (file->mode != s.st_mode) return 1;
 #ifndef NO_MTIME
-  if (file->mtime != s.st_mtim.tv_sec) return 1;
+  if (file->mtime != s.st_mtime) return 1;
 #endif
 #ifndef NO_PERMS
   if (file->uid != s.st_uid) return 1;
@@ -75,10 +75,10 @@
   file->inode = s.st_ino;
   file->device = s.st_dev;
 #ifndef NO_MTIME
-  file->mtime = s.st_mtim.tv_sec;
+  file->mtime = s.st_mtime;
 #endif
 #ifndef NO_ATIME
-  file->atime = s.st_atim.tv_sec;
+  file->atime = s.st_atime;
 #endif
   file->mode = s.st_mode;
 #ifndef NO_HARDLINKS
