--- Makefile.orig	2024-06-09 00:24:35
+++ Makefile	2024-06-09 00:44:20
@@ -90,7 +90,7 @@
 
 INCLUDES = -I. -Isimio -Iformats -Itransport -Idrivers -Iutil -Iui
 GCC_CFLAGS = -O1 -Wall -Wno-char-subscripts -ggdb
-CONFIG_CFLAGS = -DLIB_DIR=\"$(LIBDIR)\"
+CONFIG_CFLAGS = -DLIB_DIR=\"$(LIBDIR)\" -D__APPLE__ -D__MACPORTS__ -DMACPORTS_LIB_DIR=\"$(PREFIX)/lib\"
 
 MSPDEBUG_LDFLAGS = $(LDFLAGS) $(PORTS_LDFLAGS)
 MSPDEBUG_LIBS = -L. -lusb $(READLINE_LIBS) $(OS_LIBS)
