--- config/config.pl.orig	2020-01-01 08:29:53.000000000 -0800
+++ config/config.pl	2020-02-06 17:50:17.000000000 -0800
@@ -1153,7 +1153,7 @@
 
       while (<GS>) {
         chomp;
-        if($flag == 0 && /Ghostscript\s*(?:Version|(?:PRE-|BETA |TESTER )RELEASE|)\s*(\d+[.]?\d*)/i) {
+        if($flag == 0 && /Ghostscript\s*(?:Version|(?:PRE-|BETA |TESTER )RELEASE|RELEASE CANDIDATE|)\s*(\d+[.]?\d*)/i) {
           $gs_version = $1;
           $flag = 1;
         }
