--- data/Makefile.in.orig	2010-11-17 03:15:12.000000000 -0800
+++ data/Makefile.in	2010-11-21 15:32:07.000000000 -0800
@@ -457,7 +457,7 @@
 uninstall-hook: update-icon-cache
 
 install-icons:			
-	for size in 16x16 22x22 24x24 32x32 48x48 scalable; do mkdir --parents $(DESTDIR)/$(datadir)/icons/hicolor/$$size/apps; done
+	for size in 16x16 22x22 24x24 32x32 48x48 scalable; do mkdir -p $(DESTDIR)/$(datadir)/icons/hicolor/$$size/apps; done
 
 	$(INSTALL_DATA) $(srcdir)/pessulus-16.png $(DESTDIR)/$(datadir)/icons/hicolor/16x16/apps/pessulus.png
 	$(INSTALL_DATA) $(srcdir)/pessulus-22.png $(DESTDIR)/$(datadir)/icons/hicolor/22x22/apps/pessulus.png
