--- Makefile.orig	2011-12-07 01:56:23.000000000 +0900
+++ Makefile	2011-12-29 22:34:48.000000000 +0900
@@ -5,7 +5,7 @@
 PREFIX         = $(HOME)/local
 PROGRAM_SUFFIX =
 CC             = g++
-TARGET         = opal rand_shuf opal-prune opal-pmt opal-prune-pmt opalk
+TARGET         = opal rand_shuf opal-prune opal-pmt opal-prune-pmt opalk opal-multiclass
 OBJS           = timer.o
 CFLAGS         = -Wall -O2 -DNDEBUG # -m64 -march=native
 
@@ -21,6 +21,8 @@
 	$(CC) $(CFLAGS) -o $@$(PROGRAM_SUFFIX) $(OBJS) opal.cc
 opalk: $(OBJS) Makefile
 	$(CC) $(CFLAGS) -DUSE_POLYK -o $@$(PROGRAM_SUFFIX) $(OBJS) opal.cc
+opal-multiclass: $(OBJS) Makefile
+	$(CC) $(CFLAGS) -DUSE_MULTICLASS -o $@$(PROGRAM_SUFFIX) $(OBJS) opal.cc
 rand_shuf: $(OBJS) Makefile
 	$(CC) $(CFLAGS) -o $@$(PROGRAM_SUFFIX) rand_shuf.cc
 
@@ -31,6 +33,7 @@
 	install -c -s opal-pmt$(PROGRAM_SUFFIX) $(PREFIX)/bin
 	install -c -s opal-prune-pmt$(PROGRAM_SUFFIX) $(PREFIX)/bin
 	install -c -s opalk$(PROGRAM_SUFFIX) $(PREFIX)/bin
+	install -c -s opal-multiclass$(PROGRAM_SUFFIX) $(PREFIX)/bin
 	install -c -s rand_shuf$(PROGRAM_SUFFIX) $(PREFIX)/bin
 
 .cc.o: Makefile
