diff --git setup.py setup.py
index 4ec944e..a4777b5 100644
--- setup.py
+++ setup.py
@@ -21,13 +21,8 @@ setup(
         'bpylist2',
     ],
     setup_requires=[
-        "pytest-runner==5.3.0",
-        "pytest-pylint==0.18.0",
-        "pytest-pycodestyle==2.2.0",
-        "pytest-mypy==0.8.1",
-        'dataclasses;python_version<"3.7"',
     ],
-    tests_require=["pytest==6.2.3"],
+    tests_require=[],
     install_requires=[
         'dataclasses;python_version<"3.7"',
     ],
