--- Makefile.PL.orig
+++ Makefile.PL
@@ -29,12 +29,7 @@ my @possible_libraries = qw(
 );
 
 my %possible_install_paths = (
-  "../js/src/*"      => "../js/src",
-  "/usr/lib"         => "/usr/include",
-  "/usr/lib64"       => "/usr/include",          # CentOS 5 / 64 bit
-  "/usr/local/lib"   => "/usr/local/include",
-  "/usr/lib/firefox" => "/usr/include/firefox",
-  "/opt/local/lib"   => "/opt/local/include",    # MacPorts
+  "@@PREFIX@@/lib"   => "@@PREFIX@@/include/js"
 );
 
 my ($JS_LIB_DIR, @JS_INCL_DIRS, $JS_LIB_NAME);
