--- Makefile.orig	2013-07-02 17:22:26.000000000 -0400
+++ Makefile	2013-07-03 19:18:27.000000000 -0400
@@ -7,21 +7,21 @@
 TEXTFILES = words.txt rhymes.txt multiple.txt
 DBFILES = words.db rhymes.db multiple.db
 
-RHYMEPATH = /usr/share/rhyme
-BINPATH = /usr/bin
-MANPATH = /usr/share/man/man1
+RHYMEPATH = $(PREFIX)/share/rhyme
+BINPATH   = $(PREFIX)/bin
+MANPATH   = $(PREFIX)/share/man/man1
 
 OBJECTS = rhyme.o setup.o alloc.o getopt1.o getopt.o
 
 all: words.db rhymes.db multiple.db rhyme
 
 install: all
-	install --mode=755 rhyme $(BINPATH)
-	install --mode=755 -d $(RHYMEPATH)
-	install --mode=644 words.db $(RHYMEPATH)
-	install --mode=644 rhymes.db $(RHYMEPATH)
-	install --mode=644 multiple.db $(RHYMEPATH)
-	install --mode=644 rhyme.1 $(MANPATH)
+	install -m 755 rhyme $(DESTDIR)$(BINPATH)
+	install -m 755 -d $(DESTDIR)$(RHYMEPATH)
+	install -m 644 words.db $(DESTDIR)$(RHYMEPATH)
+	install -m 644 rhymes.db $(DESTDIR)$(RHYMEPATH)
+	install -m 644 multiple.db $(DESTDIR)$(RHYMEPATH)
+	install -m 644 rhyme.1 $(DESTDIR)$(MANPATH)
 
 uninstall:
 	rm -f $(BINPATH)/rhyme
