--- configure.orig	2013-09-24 10:08:36.000000000 -0700
+++ configure	2013-11-06 16:10:24.000000000 -0800
@@ -5861,9 +5861,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -5936,9 +5936,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -6011,9 +6011,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -6086,9 +6086,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
