--- configure.orig	2010-09-15 02:55:39.000000000 -0700
+++ configure	2010-09-15 11:43:27.000000000 -0700
@@ -5736,9 +5736,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print sys.prefix;'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print sys.exec_prefix;'`
 
 
 
@@ -5811,9 +5811,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print sys.prefix;'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print sys.exec_prefix;'`
 
 
 
@@ -5886,9 +5886,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print sys.prefix;'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print sys.exec_prefix;'`
 
 
 
