--- configure.ac.orig	2014-03-03 05:01:44.000000000 -0800
+++ configure.ac	2014-03-14 19:20:22.000000000 -0700
@@ -1,7 +1,7 @@
 dnl -*- Mode: autoconf -*-
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.53)
-AC_INIT(alacarte, 3.11.91, http://bugzilla.gnome.org/enter_bug.cgi?product=alacarte)
+AC_INIT(alacarte.py, 3.11.91, http://bugzilla.gnome.org/enter_bug.cgi?product=alacarte)
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
 AC_CONFIG_SRCDIR(alacarte.in)
 AC_CONFIG_HEADERS([config.h])
