--- Makefile.in.orig	2010-03-29 01:30:58.000000000 -0700
+++ Makefile.in	2010-08-25 18:54:12.000000000 -0700
@@ -250,8 +250,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = data po Alacarte
-CLEANFILES = alacarte
-bin_SCRIPTS = alacarte
+CLEANFILES = alacarte.py
+bin_SCRIPTS = alacarte.py
 DISTCLEANFILES = ChangeLog
 EXTRA_DIST = \
 	alacarte.in \
@@ -782,11 +782,12 @@
 	uninstall uninstall-am uninstall-binSCRIPTS
 
 
-alacarte: alacarte.in
+alacarte.py: alacarte.in
 	sed -e s!\@PYTHON\@!@PYTHON@!		\
 	    -e s!\@PYOPTIONS\@!-OOt!		\
+	    -e s!\@PYEXECDIR\@!@pyexecdir@!     \
 	    < $< > $@
-alacarte: Makefile
+alacarte.py: Makefile
 
 ChangeLog:
 	@echo Creating $@
