--- setup.py	2012-07-06 14:03:32.000000000 +0200
+++ setup.py.new	2012-09-03 23:56:13.000000000 +0200
@@ -11,13 +11,12 @@
         'README-fr',
         'COPYING',
         'AUTHORS',
-        'ChangeLog',
         'NEWS',
         'screenshot.png']),
     ('share/doc/glances/doc', glob('doc/*.png')),
-    ('share/glances/html', glob('src/html/*.html')),
-    ('share/glances/html/css', glob('src/css/*.css')),
-    ('share/glances/html/img', glob('src/img/*.png')),
+    ('share/glances/html', glob('glances/html/*.html')),
+    ('share/glances/html/css', glob('glances/css/*.css')),
+    ('share/glances/html/img', glob('glances/img/*.png')),
 ]
 for mo in glob('i18n/*/LC_MESSAGES/*.mo'):
     data_files.append((dirname(mo).replace('i18n/', 'share/locale/'), [mo]))
