--- tests/test_arraydatatype.py.orig	2025-01-20 11:30:48
+++ tests/test_arraydatatype.py	2025-01-20 18:56:12
@@ -73,10 +73,10 @@
             glVertexPointer(4, GL_INT, 0, s)
 
         def test_texture(self):
-            """Test texture (requires OpenGLContext and PIL)"""
+            """Test texture (requires OpenGLContext and PIL or Pillow)"""
             try:
                 from OpenGLContext import texture
-                import Image
+                from PIL import Image
                 from OpenGL.GLUT import glutSolidTeapot
             except ImportError:
                 pass
