--- src/backend/utils/init/postinit.c.orig	2025-05-05 13:30:08.000000000 -0700
+++ src/backend/utils/init/postinit.c	2025-07-06 20:10:05.091360497 -0700
@@ -469,10 +469,16 @@
 
 		actual_versionstr = get_collation_actual_version(dbform->datlocprovider, dbform->datlocprovider == COLLPROVIDER_ICU ? iculocale : collate);
 		if (!actual_versionstr)
+			/*
+			 * Reduce the log level from WARNING to LOG to to silence warnings
+			 * doing physical replication from a Linux machine to a macOS:
+			 *   WARNING: database "bz_trader" with provider "libc" has no
+			 *   actual collation version, but version 2.39 was recorded
+			 */
 			/* should not happen */
-			elog(WARNING,
-				 "database \"%s\" has no actual collation version, but a version was recorded",
-				 name);
+			elog(LOG,
+				 "database \"%s\" with provider \"%s\" has no actual collation version, but version %s was recorded",
+				 name, collprovider_name(dbform->datlocprovider), collversionstr);
 		else if (strcmp(actual_versionstr, collversionstr) != 0)
 			ereport(WARNING,
 					(errmsg("database \"%s\" has a collation version mismatch",
