--- vendor/github.com/containers/common/pkg/config/config_darwin.go.orig	2024-03-19 23:36:25.000000000 +0200
+++ vendor/github.com/containers/common/pkg/config/config_darwin.go	2024-03-19 23:56:09.275397878 +0200
@@ -15,6 +15,9 @@
 var defaultHelperBinariesDir = []string{
 	// Relative to the binary directory
 	"$BINDIR/../libexec/podman",
+	// MacPorts install paths
+	"@@PREFIX@@/libexec",
+	"@@PREFIX@@/bin",
 	// Homebrew install paths
 	"/usr/local/opt/podman/libexec/podman",
 	"/opt/homebrew/opt/podman/libexec/podman",
