--- vendor/github.com/containers/common/pkg/config/config_darwin.go.orig	2023-02-08 21:03:18.000000000 +0200
+++ vendor/github.com/containers/common/pkg/config/config_darwin.go	2023-02-12 20:02:19.697007565 +0200
@@ -29,6 +29,9 @@
 }
 
 var defaultHelperBinariesDir = []string{
+	// MacPorts install paths
+	"@@PREFIX@@/libexec",
+	"@@PREFIX@@/bin",
 	// Homebrew install paths
 	"/usr/local/opt/podman/libexec",
 	"/opt/homebrew/bin",

