--- vendor/go.podman.io/common/pkg/config/config_darwin.go.orig	2025-12-09 15:11:03
+++ vendor/go.podman.io/common/pkg/config/config_darwin.go	2025-12-16 20:34:26
@@ -15,6 +15,9 @@
 var defaultHelperBinariesDir = []string{
 	// Relative to the binary directory
 	"$BINDIR/../libexec/podman",
+	// MacPorts install paths
+	"@@PREFIX@@/libexec",
+	"@@PREFIX@@/bin",
 	// Homebrew install paths
 	"/usr/local/opt/podman/libexec/podman",
 	"/opt/homebrew/opt/podman/libexec/podman",
