--- Makefile.in.orig	2018-04-05 20:23:27.000000000 +0900
+++ Makefile.in	2018-08-19 11:36:33.000000000 +0900
@@ -396,7 +396,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . src inst_headers tests examples perf
+SUBDIRS = . src inst_headers
 DIST_SUBDIRS = src inst_headers tests examples perf .
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = util/sunf95preproc-wrapper \
