--- Makefile.in.orig	2015-10-30 01:43:17.000000000 +0900
+++ Makefile.in	2015-11-22 16:51:55.000000000 +0900
@@ -320,7 +320,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . src inst_headers tests examples perf
+SUBDIRS = . src inst_headers
 DIST_SUBDIRS = src inst_headers tests examples perf .
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST = util/mpi_launch_poe util/sunf95preproc-wrapper \
