--- examples/Makefile.in.orig	2014-03-06 23:32:58.000000000 +0900
+++ examples/Makefile.in	2014-06-08 09:06:01.000000000 +0900
@@ -355,7 +355,7 @@
 row2col_f_LDADD = ../src/libyaxt.la $(MPI_FC_LIB)
 tr_gp2fs_SOURCES = tr_gp2fs.c
 tr_gp2fs_LDADD = $(LDADD) -lm
-AM_CFLAGS = -I../inst_headers $(MPI_C_INCLUDE)
+AM_CPPFLAGS = -I../inst_headers $(MPI_C_INCLUDE)
 AM_FCFLAGS = $(FC_MOD_FLAG)../inst_headers/f90 $(MPI_FC_INCLUDE) $(MPI_FC_MOD)
 LDADD = ../src/libyaxt.la $(MPI_C_LIB)
 all: all-am
