diff -Naurd ./mlx_lm/tool_parsers/qwen3_coder.py ./mlx_lm/tool_parsers/qwen3_coder.py
--- ./mlx_lm/tool_parsers/qwen3_coder.py	2026-09-23 14:57:40
+++ ./mlx_lm/tool_parsers/qwen3_coder.py	2026-09-24 11:13:21
@@ -54,7 +54,15 @@
         or param_type.startswith("short")
         or param_type.startswith("unsigned")
     ):
-        return int(param_value)
+        # Handle mlx_lm.server errors observed on "majentik/Qwen3-Coder-Next-MLX-8bit"
+        # E.g. `WARNING - Failed to parse tool call (ValueError: invalid literal for int() with base 10: '10.0') — tool text was likely truncated mid-generation.`
+        # This issue is observed to cause agent looping and appears to be exacerbated by larger (~120k) token contexts on 128 GB machines
+        # Fixed in https://github.com/ml-explore/mlx-lm/pull/1417
+        float_param_value = float(param_value)
+        int_param_value = int(float_param_value)
+        if float_param_value - int_param_value != 0:
+            raise ValueError(f"Invalid integer literal {param_value!r}")
+        return int_param_value
     elif param_type.startswith("num") or param_type.startswith("float"):
         float_param_value = float(param_value)
         int_param_value = int(float_param_value)
