--- configure.orig	2014-12-08 14:23:37.000000000 -0600
+++ configure	2014-12-19 18:25:57.000000000 -0600
@@ -23928,7 +23928,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Perl cc" >&5
 $as_echo_n "checking for Perl cc... " >&6; }
 
-        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/\.]+).;\s*/$1/);'`
 
         if test "x$PERLCC" != "x" ; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERLCC" >&5
