--- configure.orig	2009-09-23 19:11:54.000000000 -0500
+++ configure	2010-01-20 23:31:57.000000000 -0600
@@ -27279,7 +27279,7 @@
         { $as_echo "$as_me:$LINENO: checking for Perl cc" >&5
 $as_echo_n "checking for Perl cc... " >&6; }
 
-        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/\.]+).;\s*/$1/);'`
 
         if test "x$PERLCC" != "x" ; then
             { $as_echo "$as_me:$LINENO: result: $PERLCC" >&5
