--- configure.orig	2012-10-09 17:35:37.000000000 -0500
+++ configure	2012-10-20 18:10:02.000000000 -0500
@@ -23410,7 +23410,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Perl cc" >&5
 $as_echo_n "checking for Perl cc... " >&6; }
 
-        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/\.]+).;\s*/$1/);'`
 
         if test "x$PERLCC" != "x" ; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERLCC" >&5
