--- configure.orig	2014-12-08 14:23:37.000000000 -0600
+++ configure	2016-01-03 03:47:24.000000000 -0600
@@ -10906,7 +10906,7 @@
       case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
 	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
-	10.[012]*)
+	10.[012][,.]*)
 	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
 	10.*)
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
@@ -15658,19 +15658,10 @@
 	fi
 	;;
 
-    darwin8*|darwin9*|darwin10*|darwin11*|darwin12*|darwin13*)
-        #   Flat Namespace linking                      (Mac OS/X)
+    darwin*)
+        #   Add frameworks                              (OS X)
         #
-        { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether darwin ld might need -flat_namespace" >&5
-$as_echo_n "checking whether darwin ld might need -flat_namespace... " >&6; }
-	if test -z "$MACOSX_DEPLOYMENT_TARGET"; then
-		LDFLAGS="-flat_namespace -framework CoreFoundation -framework CoreServices -framework DiskArbitration -framework IOKit $LDFLAGS"
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-	else
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	fi
+		LDFLAGS="-framework CoreFoundation -framework CoreServices -framework DiskArbitration -framework IOKit $LDFLAGS"
 	;;
 
     *)
@@ -23928,7 +23919,7 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Perl cc" >&5
 $as_echo_n "checking for Perl cc... " >&6; }
 
-        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+        PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/\.]+).;\s*/$1/);'`
 
         if test "x$PERLCC" != "x" ; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PERLCC" >&5
