--- test/filter_designer_hw.c.orig
+++ test/filter_designer_hw.c
@@ -7,11 +7,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#ifdef __APPLE__
-#include <iio/iio.h>
-#else
 #include <iio.h>
-#endif
 
 int main(void)
 {
--- test/auto_rate_test_hw.c.orig
+++ test/auto_rate_test_hw.c
@@ -8,11 +8,7 @@
 #include <string.h>
 #include <errno.h>
 
-#ifdef __APPLE__
-#include <iio/iio.h>
-#else
 #include <iio.h>
-#endif
 
 #define RATE_TOLERANCE_HZ 2
 
