diff --git a/test/auto_rate_test_hw.c b/test/auto_rate_test_hw.c
index 20edf25..b86f0e4 100644
--- a/test/auto_rate_test_hw.c
+++ b/test/auto_rate_test_hw.c
@@ -8,11 +8,7 @@
 #include <string.h>
 #include <errno.h>
 
-#ifdef __APPLE__
-#include <iio/iio.h>
-#else
 #include <iio.h>
-#endif
 
 #define RATE_TOLERANCE_HZ 2
 
diff --git a/test/filter_designer_hw.c b/test/filter_designer_hw.c
index f441ee6..a54cda1 100644
--- a/test/filter_designer_hw.c
+++ b/test/filter_designer_hw.c
@@ -7,11 +7,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#ifdef __APPLE__
-#include <iio/iio.h>
-#else
 #include <iio.h>
-#endif
 
 int main(void)
 {
diff --git a/test/fmcomms5_sync_test.c b/test/fmcomms5_sync_test.c
index 054a71f..b925c88 100644
--- a/test/fmcomms5_sync_test.c
+++ b/test/fmcomms5_sync_test.c
@@ -3,11 +3,7 @@
 #include <math.h>
 #include <stdbool.h>
 #include <stdio.h>
-#ifdef __APPLE__
-#include <iio/iio.h>
-#else
 #include <iio.h>
-#endif
 #include <stdlib.h>
 #include <stdint.h>
 #include <string.h>
