--- src/Makefile_libltfat.orig	2022-05-24 15:55:33
+++ src/Makefile_libltfat	2025-03-29 13:12:17
@@ -39,7 +39,7 @@
 
 # Base CFLAGS
 CFLAGS+=-Wall -Wextra -std=c99
-CXXFLAGS+=-Wall -Wextra -std=c++11 -fno-exceptions -fno-rtti
+CXXFLAGS+=-Wall -Wextra -std=c++17 -fno-exceptions -fno-rtti
 LFLAGS = -Wl,--no-undefined -Lbuild/$(CROSS) $(OPTLPATH) -Wl,-rpath,$$ORIGIN
 
 MATLABROOT ?= /usr/local/MATLAB_R2017a
