--- oct/Makefile_unix.orig	2022-05-24 15:55:33
+++ oct/Makefile_unix	2024-03-19 18:30:51
@@ -36,7 +36,7 @@
 export CXXFLAGS := $(shell $(MKOCTFILE) -p CXXFLAGS) -std=gnu++11 -Wall -DLTFAT_LARGEARRAYS $(OPTCXXFLAGS)
 # export is necessary, otherwise LFLAGS won't have any effect
 # at least on Windows and on Mac
-export LDFLAGS := -L$(shell $(MKOCTFILE) -p LIBDIR) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS) 
+export LDFLAGS := -L$(shell $(MKOCTFILE) -p LIBDIR) $(LDFLAGS) $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FFTW_LIBS) 
 undefine LFLAGS
 unexport LFLAGS
 
