From 89ec99a02fe593384469d91c1b6435f2c86b587e Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Mon, 21 Nov 2022 01:08:19 +0800
Subject: [PATCH 1/2] time_stamp_counter_stubs.c: add legacysupport header for
 clock_gettime

--- a/time_stamp_counter/src/dune	2023-04-25 21:12:26.000000000 +0800
+++ b/time_stamp_counter/src/dune	2024-07-06 22:52:24.000000000 +0800
@@ -1,7 +1,7 @@
 (library (name time_stamp_counter) (public_name core_unix.time_stamp_counter)
  (preprocess (pps ppx_jane ppx_optcomp)) (preprocessor_deps config.h)
- (c_names time_stamp_counter_stubs)
- (libraries core core_unix ocaml_intrinsics))
+ (c_names time_stamp_counter_stubs) (c_flags -I@PREFIX@/include/LegacySupport)
+ (c_library_flags -lMacportsLegacySupport) (libraries core core_unix ocaml_intrinsics))
 
 (rule (targets config.h) (deps)
- (action (bash "cp %{lib:jst-config:config.h} %{targets}")))
\ No newline at end of file
+ (action (bash "cp %{lib:jst-config:config.h} %{targets}")))
