--- sources/Makefile.orig	2018-12-26 03:31:52.000000000 -0600
+++ sources/Makefile	2019-05-31 13:02:55.000000000 -0500
@@ -22,24 +22,13 @@
 MANDIR=/usr/local/man
 
 sleepwatcher: sleepwatcher.c
-	$(CC) $(CFLAGS_I386) -o sleepwatcher.i386 sleepwatcher.c $(LIBS)
-	$(CC) $(CFLAGS_X86_64) -o sleepwatcher.x86_64 sleepwatcher.c $(LIBS)
-	lipo -create sleepwatcher.i386 sleepwatcher.x86_64 -output sleepwatcher
-	rm sleepwatcher.i386 sleepwatcher.x86_64
+	$(CC) $(CFLAGS) -o sleepwatcher sleepwatcher.c $(LIBS)
 
-fat: sleepwatcher sleepwatcher.ppc
-	lipo -create sleepwatcher sleepwatcher.ppc -output sleepwatcher.fat
-	mv sleepwatcher.fat sleepwatcher
-	rm sleepwatcher.ppc
-
-sleepwatcher.ppc: sleepwatcher.c
-	$(CC) $(CFLAGS_PPC) -o sleepwatcher.ppc sleepwatcher.c $(LIBS)
-
-install: sleepwatcher sleepwatcher.8
+install: sleepwatcher ../sleepwatcher.8
 	mkdir -p $(BINDIR)
-	install -o root -g wheel -m 755 sleepwatcher $(BINDIR)
+	install -m 755 sleepwatcher $(BINDIR)
 	mkdir -p $(MANDIR)/man8
-	install -o root -g wheel -m 644 sleepwatcher.8 $(MANDIR)/man8
+	install -m 644 ../sleepwatcher.8 $(MANDIR)/man8
 
 clean:
 	rm -f sleepwatcher
