--- src/discover.ml	2011-07-12 10:21:07.000000000 +0200
+++ src/discover.ml	2011-09-28 11:33:04.000000000 +0200
@@ -35,8 +35,6 @@
 let get_includes = filter_map (function I s -> Some s | _ -> None)
 
 let config_defines = [
-  "_POSIX_C_SOURCE 200112L";
-  "_XOPEN_SOURCE 600";
   "_BSD_SOURCE";
   "_LARGEFILE64_SOURCE";
   "WIN32_LEAN_AND_MEAN";
@@ -243,6 +241,6 @@
     "EXECINFO", L[ I"execinfo.h"; S"backtrace"; S"backtrace_symbols"; ];
     "SETENV", L[ I"stdlib.h"; S"setenv"; S"unsetenv"; ];
     "CLEARENV", L[ I"stdlib.h"; S"clearenv"; ];
-    "MKDTEMP", L[ I"stdlib.h"; S"mkdtemp"; ];
+    "MKDTEMP", L[ I"stdlib.h"; I"unistd.h"; S"mkdtemp"; ];
   ]
 
