--- Makefile.unix-install.orig	2017-01-26 12:42:23.000000000 +0000
+++ Makefile.unix-install	2017-01-26 12:42:23.000000000 +0000
@@ -1,5 +1,5 @@
 DESTDIR		=
-prefix		= /usr/local
+prefix		= __PREFIX__
 exec_prefix	= $(prefix)
 man_prefix	= $(prefix)/share
 
@@ -49,10 +49,8 @@
 	done;
 
 install-man:
-	$(INSTALL_BIN) -d $(MANDIR1)
 	$(INSTALL_BIN) -d $(MANDIR3)
 	$(INSTALL_BIN) -d $(MANDIR8)
-	$(INSTALL_DATA) man/*.1 $(MANDIR1)
 	$(INSTALL_DATA) man/*.3 $(MANDIR3)
 	$(INSTALL_DATA) man/*.8 $(MANDIR8)
 
