--- vvp/Makefile.in.orig	2025-02-19 14:26:54
+++ vvp/Makefile.in	2025-02-19 14:27:14
@@ -159,7 +159,7 @@
 	$(CXX) $(LDFLAGS) -o vvp@EXEEXT@ main.o -L. -lvvp $(LIBS)
 
 libvvp.so: $O
-	$(CXX) -shared $(LDFLAGS) -o libvvp.so $O $(LIBS) $(dllib)
+	$(CXX) -shared $(LDFLAGS) -install_name $(libdir)/libvvp.so -o libvvp.so $O $(LIBS) $(dllib)
 else
 vvp@EXEEXT@: $O main.o
 	$(CXX) $(LDFLAGS) -o vvp@EXEEXT@ main.o $O $(LIBS) $(dllib)
