--- plugins/check_load.c.orig	2017-01-19 17:01:31.000000000 +0100
+++ plugins/check_load.c	2017-11-16 18:10:11.000000000 +0100
@@ -157,7 +157,7 @@
 	    sscanf (input_buffer, "%*[^l]load average: %lf, %lf, %lf", &la1, &la5, &la15);
     }
     else if(strstr(input_buffer, "load averages:")) {
-	    sscanf (input_buffer, "%*[^l]load averages: %lf, %lf, %lf", &la1, &la5, &la15);
+	    sscanf (input_buffer, "%*[^l]load averages: %lf %lf %lf", &la1, &la5, &la15);
     }
     else {
 		printf (_("could not parse load from uptime: %s\n"), result, PATH_TO_UPTIME);
