--- ext/nokogiri/extconf.rb.orig	2019-08-11 19:06:54.000000000 +0000
+++ ext/nokogiri/extconf.rb	2019-10-23 16:00:09.000000000 +0000
@@ -393,7 +393,7 @@
 end
 
 if darwin?
-  ENV['CFLAGS'] = "#{ENV['CFLAGS']} -I /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libxml2"
+  ENV['CFLAGS'] = "#{ENV['CFLAGS']} -I @@PREFIX@@/include/libxml2"
 end
 
 if openbsd? && !using_system_libraries?
