Do not copy libraries into app bundle.
--- scons/bundle.py.orig	2009-06-14 05:11:34.000000000 -0500
+++ scons/bundle.py	2018-10-18 00:26:09.000000000 -0500
@@ -30,8 +30,6 @@
 	# add icon -- TODO generate .icns file from png or svg
 	iconFile = env['BUNDLE_ICON']
 	run('cp %s %s/Contents/Resources' % (iconFile, bundleDir) )
-	# add dependent libraries, fixing all absolute paths
-	addDependentLibsToBundle( bundleDir )
 	
 
 def createBundleMessage(target, source, env) :
