From 9b1e24faf2bf6df2434757102f11b0cc7eb64788 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Wed, 11 Feb 2015 12:32:56 -0800
Subject: [PATCH 3/3] glext.h: Add missing include of stddef.h for ptrdiff_t

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 include/GL/glext.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/GL/glext.h b/include/GL/glext.h
index 907a582..babe4eb 100644
--- a/include/GL/glext.h
+++ b/include/GL/glext.h
@@ -4662,6 +4662,7 @@ GLAPI void APIENTRY glVertexBlendARB (GLint count);
 
 #ifndef GL_ARB_vertex_buffer_object
 #define GL_ARB_vertex_buffer_object 1
+#include <stddef.h>
 typedef ptrdiff_t GLsizeiptrARB;
 typedef ptrdiff_t GLintptrARB;
 #define GL_BUFFER_SIZE_ARB                0x8764
-- 
2.8.2

