--- a/configs/darwin.orig	2012-03-23 01:10:16.000000000 +0800
+++ b/configs/darwin	2014-01-09 20:28:20.000000000 +0800
@@ -13,8 +13,7 @@
 CXX = $(shell xcrun -find c++)
 PIC_FLAGS = -fPIC
 DEFINES =  -D_DARWIN_C_SOURCE -DPTHREADS -D_GNU_SOURCE \
-	   -DGLX_ALIAS_UNSUPPORTED \
-	   -DGLX_DIRECT_RENDERING -DGLX_USE_APPLEGL
+	   -DGLX_ALIAS_UNSUPPORTED -DGLX_INDIRECT_RENDERING
 
 # -DGLX_INDIRECT_RENDERING \
 
@@ -47,12 +46,12 @@
 OSMESA_LIB_GLOB = lib$(OSMESA_LIB).*dylib
 VG_LIB_GLOB = lib$(VG_LIB).*dylib
 
-GL_LIB_DEPS = -L$(INSTALL_DIR)/$(LIB_DIR) -L$(X11_DIR)/$(LIB_DIR) -lX11-xcb -lxcb -lX11 -lXext $(EXTRA_LDFLAGS)
+GL_LIB_DEPS = -L$(INSTALL_DIR)/$(LIB_DIR) -L$(X11_DIR)/$(LIB_DIR) -lX11-xcb -lxcb -lxcb-glx -lX11 -lXext $(EXTRA_LDFLAGS)
 OSMESA_LIB_DEPS = $(EXTRA_LDFLAGS)
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) $(EXTRA_LDFLAGS)
 GLW_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) -L$(INSTALL_DIR)/$(LIB_DIR) -L$(X11_DIR)/$(LIB_DIR) -lX11 -lXt $(EXTRA_LDFLAGS)
 
-SRC_DIRS = glsl mapi/glapi mapi/vgapi glx/apple mesa gallium glu
+SRC_DIRS = glsl mapi/glapi mapi/vgapi glx mesa gallium glu
 GLU_DIRS = sgi
 DRIVER_DIRS = osmesa
 #DRIVER_DIRS = dri
