diff --git a/configs/darwin b/configs/darwin
index 3cf1110..aa5a22c 100644
--- a/configs/darwin
+++ b/configs/darwin
@@ -13,8 +13,7 @@ CC = gcc
 CXX = g++
 PIC_FLAGS = -fPIC
 DEFINES =  -D_DARWIN_C_SOURCE -DPTHREADS -D_GNU_SOURCE \
-	   -DGLX_ALIAS_UNSUPPORTED \
-	   -DGLX_DIRECT_RENDERING -DGLX_USE_APPLEGL
+	   -DGLX_ALIAS_UNSUPPORTED -DGLX_INDIRECT_RENDERING
 
 # -DGLX_INDIRECT_RENDERING \
 
@@ -54,7 +53,7 @@ GLW_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) -L$(INSTALL_DIR)/$(LIB_DIR) -L$(X
 APP_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L$(INSTALL_DIR)/$(LIB_DIR) -L$(X11_DIR)/$(LIB_DIR) -lX11 -lXmu -lXt -lXi -lm
 
 # omit glw lib for now:
-SRC_DIRS = glsl mapi/glapi mapi/vgapi glx/apple mesa gallium glu glut/glx
+SRC_DIRS = glsl mapi/glapi mapi/vgapi glx mesa gallium glu glut/glx
 GLU_DIRS = sgi
 DRIVER_DIRS = osmesa
 #DRIVER_DIRS = dri
