--- Mesa-7.8.2/src/mesa/Makefile.orig	2010-12-01 00:08:30.000000000 +0900
+++ Mesa-7.8.2/src/mesa/Makefile	2010-12-01 00:11:20.000000000 +0900
@@ -20,7 +20,7 @@
 
 # Default: build dependencies, then asm_subdirs, GLSL built-in lib,
 # then convenience libs (.a) and finally the device drivers:
-default: depend asm_subdirs glsl_builtin libmesa.a libmesagallium.a \
+default: depend asm_subdirs glsl_builtin libmesa.a \
 	libglapi.a driver_subdirs
 
 
@@ -42,7 +42,7 @@
 
 ######################################################################
 # Device drivers
-driver_subdirs: libmesa.a libglapi.a libmesagallium.a
+driver_subdirs: libmesa.a libglapi.a
 	@ (cd drivers && $(MAKE))
 
 
