--- meson.build.orig	2022-09-29 15:08:18.000000000 -0400
+++ meson.build	2022-09-29 15:08:57.000000000 -0400
@@ -22,7 +22,7 @@
   'mesa',
   ['c', 'cpp'],
   version : run_command(
-    [find_program('python3', 'python'), 'bin/meson_get_version.py'],
+    [find_program('@@python3@@'), 'bin/meson_get_version.py'],
     check : true
   ).stdout(),
   license : 'MIT',
@@ -1002,7 +1002,7 @@
   ]
 endif
 
-prog_python = import('python').find_installation('python3')
+prog_python = import('python').find_installation('@@python3@@')
 has_mako = run_command(
   prog_python, '-c',
   '''
