--- a/src/egl/drivers/dri2/egl_dri2.c	2019-06-27 03:14:08.000000000 +0700
+++ b/src/egl/drivers/dri2/egl_dri2.c	2022-09-01 20:21:19.000000000 +0700
@@ -3003,6 +3003,7 @@
          return NULL;
       }
 
+   #if !defined(__APPLE__) // no pthread_condattr_setclock on MacOS
       /* change clock attribute to CLOCK_MONOTONIC */
       ret = pthread_condattr_setclock(&attr, CLOCK_MONOTONIC);
 
@@ -3011,6 +3012,7 @@
          free(dri2_sync);
          return NULL;
       }
+   #endif
 
       ret = pthread_cond_init(&dri2_sync->cond, &attr);
 
