--- a/../AppleSGLX-58/gen_gl_h.sh.orig	2009-03-19 16:08:01.000000000 -0500
+++ b/../AppleSGLX-58/gen_gl_h.sh	2009-12-20 22:04:14.000000000 -0600
@@ -13,7 +13,7 @@
       echo "#define GL_GLEXT_LEGACY 1"
       generate_macros
       echo '#include "/System/Library/Frameworks/OpenGL.framework/Headers/gl.h"'
-    } | gcc -E - | grep typedef.*PFN
+    } | ${CC} -E - | grep typedef.*PFN
 }
 
 cat ${INFILE} | while IFS= read LINE ; do
