--- main.m	Wed Jan 28 00:34:22 2004
+++ main.m.new	Wed Jan 28 00:31:47 2004
@@ -22,21 +22,28 @@
  */
 
 #include "Controller.h"
+#include "GNUstep.h"
 
 int main (int argc, char **argv)
 {
    NSAutoreleasePool *pool;
    NSApplication *app;
+#ifdef GNUSTEP
    NSMenu *mainMenu;
    NSMenu *menu;
    NSMenuItem *menuItem;
+#endif
    Controller *controller;
    int cnt;
 
    pool = [NSAutoreleasePool new];
    app = [NSApplication sharedApplication];
    [app setApplicationIconImage: [NSImage imageNamed: @"Gomoku"]];
+#ifndef GNUSTEP
+   NSApplicationMain(argc, argv);
+#endif
 
+#ifdef GNUSTEP
    //
    // Create the Menu 
    //
@@ -110,6 +117,7 @@
    controller = [Controller new: cnt];
    [app setDelegate: controller];
    [app run];
+#endif
    RELEASE (controller);
    RELEASE (pool);
    return 0;
