--- Controller.h	Wed Jan 28 00:33:14 2004
+++ Controller.h.new	Wed Jan 28 00:31:47 2004
@@ -29,6 +29,16 @@
 
 @interface Controller : NSWindow
 {
+  // Outlets
+  IBOutlet NSMenu *menu;
+
+  IBOutlet NSMenuItem *game;
+  IBOutlet NSMenuItem *window;
+  IBOutlet NSMenuItem *help;
+  IBOutlet NSMenuItem *aboutPanelMenuItem;
+  IBOutlet NSMenuItem *preferencesMenuItem;
+  IBOutlet NSMenuItem *newGameMenuItem;
+
   MainBoard *board;
   DifficultyPanel *diffPanel;
 }
@@ -36,8 +46,8 @@
 + new: (int) cnt;
 - init: (int) cnt;
 - (void) applicationDidFinishLaunching: (NSNotification *)aNotification;
-- (void) runDifficultyLevelPanel: (id)sender;
-- (void) newGame: (id)sender;
+- (IBAction) runDifficultyLevelPanel: (id)sender;
+- (IBAction) newGame: (id)sender;
 @end
 
 #endif
