--- setup.py.old	2018-05-24 19:42:16.000000000 -0700
+++ setup.py	2018-05-24 19:43:21.000000000 -0700
@@ -189,6 +189,8 @@
     if sys.platform == 'darwin':
       extra_compile_args.append("-Wno-shorten-64-to-32");
       extra_compile_args.append("-Wno-deprecated-register");
+      extra_compile_args.append('@@MACPORTS_STDLIB@@')
+      extra_compile_args.append('@@MACPORTS_EXTRAARG@@')
 
     # https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes
     # C++ projects must now migrate to libc++ and are recommended to set a
