--- setup.py.orig	2023-05-27 16:30:12.000000000 -0500
+++ setup.py	2023-05-27 16:33:07.000000000 -0500
@@ -264,6 +264,8 @@
     if sys.platform == 'darwin':
       extra_compile_args.append('-Wno-shorten-64-to-32')
       extra_compile_args.append('-Wno-deprecated-register')
+      extra_compile_args.append('@@MACPORTS_STDLIB@@')
+      extra_compile_args.append('@@MACPORTS_EXTRAARG@@')
 
     # https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes
     # C++ projects must now migrate to libc++ and are recommended to set a
