--- m3-sys/m3cc/src/m3makefile	Tue Oct  7 17:19:51 2003
+++ m3-sys/m3cc/src/m3makefile	Tue Oct  7 17:27:11 2003
@@ -52,16 +52,6 @@
   end
 end
 
-if equal(TARGET, "PPC_DARWIN")
-  write(EOL)
-  write("  Please note that you need a bootstrapped gcc 3.2.1 in order to" &
-        EOL)
-  write("  compile this package. All attempts to build cm3cg with Apple's" &
-        EOL)
-  write("  native version of gcc 3.1 have failed so far." & EOL)
-  write(EOL)
-end
-
 readonly proc GNU_platform (x) is
   if Platform_info contains x
     return Platform_info{x}
@@ -113,6 +103,12 @@
 
 % HACK: avoid the use of DW_FORM_strp debugging information for the time being
 M3CC_CFLAGS = M3CC_CFLAGS & " -DAVOID_DW_FORM_STRP"
+
+% HACK: Apple's cpp-precomp in gcc 3.1 does not correctly handle the poisoned token pragma
+if equal(TARGET, "PPC_DARWIN")
+  M3CC_CFLAGS = M3CC_CFLAGS & " -no-cpp-precomp"
+end
+
 
 % figure out where we're going to build the beast
 local build_dir = "."  % let m3build set the build directory
