--- CMakeLists.txt.orig	2015-01-20 12:53:01.000000000 -0500
+++ CMakeLists.txt	2015-01-20 12:53:20.000000000 -0500
@@ -145,6 +145,10 @@
 ########################################################################
 # Install cmake search helper for this library
 ########################################################################
+if(NOT CMAKE_MODULES_DIR)
+    set(CMAKE_MODULES_DIR ${GR_LIBRARY_DIR}/cmake)
+endif(NOT CMAKE_MODULES_DIR)
+
 install(FILES cmake/Modules/dvbs2Config.cmake        
-    DESTINATION lib/cmake/dvbs2
+    DESTINATION ${CMAKE_MODULES_DIR}/dvbs2
 )
