--- sys/unix/Makefile.src.orig	2008-05-12 09:35:18.000000000 +0900
+++ sys/unix/Makefile.src	2008-05-12 09:36:38.000000000 +0900
@@ -36,7 +36,7 @@
 # SHELL=E:/GEMINI2/MUPFEL.TTP
 
 # Normally, the C compiler driver is used for linking:
-LINK=$(CC)
+LINK=$(CC) __CFLAGS__
 
 # Pick the SYSSRC and SYSOBJ lines corresponding to your desired operating
 # system.
@@ -154,8 +154,8 @@
 # flags for debugging:
 # CFLAGS = -g -I../include
 
-CFLAGS = -W -g -O -I../include
-LFLAGS = 
+CFLAGS = __CFLAGS__ -I../include
+LFLAGS = __LDFLAGS__
 
 # The Qt and Be window systems are written in C++, while the rest of
 # NetHack is standard C.  If using Qt, uncomment the LINK line here to get
