--- src/modules/macosx/module-coreaudio-device.c.old
+++ src/modules/macosx/module-coreaudio-device.c
@@ -77,7 +78,7 @@
 
 struct userdata {
     AudioObjectID object_id;
-    AudioDeviceIOProcID proc_id;
+    AudioDeviceIOProc proc_id;
 
     pa_thread_mq thread_mq;
     pa_asyncmsgq *async_msgq;
@@ -885,7 +896,8 @@
     pa_log_debug("%u frames per IOProc\n", (unsigned int) frames);
 
     /* create one ioproc for both directions */
-    err = AudioDeviceCreateIOProcID(u->object_id, io_render_proc, u, &u->proc_id);
+    err = AudioDeviceAddIOProc(u->object_id, io_render_proc, u);
+    u->proc_id = io_render_proc;
     if (err) {
         pa_log("AudioDeviceCreateIOProcID() failed (err = %08x\n).", (int) err);
         goto fail;
@@ -967,7 +979,7 @@
 
     if (u->proc_id) {
         AudioDeviceStop(u->object_id, u->proc_id);
-        AudioDeviceDestroyIOProcID(u->object_id, u->proc_id);
+        AudioDeviceRemoveIOProc(u->object_id, u->proc_id);
     }
 
     property_address.mSelector = kAudioDevicePropertyStreamFormat;
