Upstream-Status: Submitted [https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/365]
--- src/modules/macosx/module-coreaudio-device.c.orig	2020-09-10 19:42:18.000000000 +0200
+++ src/modules/macosx/module-coreaudio-device.c	2020-09-10 19:46:35.000000000 +0200
@@ -27,6 +27,7 @@
 #include <config.h>
 #endif
 
+#include <pulse/util.h>
 #include <pulse/xmalloc.h>
 
 #include <pulsecore/sink.h>
