--- src/Makefile.am.old	2017-01-13 04:02:18.000000000 +0100
+++ src/Makefile.am	2017-02-07 00:50:33.000000000 +0100
@@ -780,19 +780,10 @@ libpulsecommon_@PA_MAJORMINOR@_la_SOURCE
 		pulsecore/thread-win32.c pulsecore/thread.h \
 		pulsecore/semaphore-win32.c pulsecore/semaphore.h
 else !OS_IS_WIN32
-if OS_IS_DARWIN
-libpulsecommon_@PA_MAJORMINOR@_la_SOURCES += \
-		pulsecore/mutex-posix.c pulsecore/mutex.h \
-		pulsecore/thread-posix.c pulsecore/thread.h \
-		pulsecore/semaphore-osx.c pulsecore/semaphore.h
-libpulsecommon_@PA_MAJORMINOR@_la_CFLAGS += "-I/Developer/Headers/FlatCarbon/"
-#libpulsecommon_@PA_MAJORMINOR@_la_LDFLAGS += "-framework CoreServices"
-else !OS_IS_DARWIN
 libpulsecommon_@PA_MAJORMINOR@_la_SOURCES += \
 		pulsecore/mutex-posix.c pulsecore/mutex.h \
 		pulsecore/thread-posix.c pulsecore/thread.h \
 		pulsecore/semaphore-posix.c pulsecore/semaphore.h
-endif !OS_IS_DARWIN
 endif !OS_IS_WIN32
 
 if HAVE_LIBASYNCNS
