--- src/daemon/default.pa.in.old	2016-05-03 08:17:39.000000000 +0200
+++ src/daemon/default.pa.in	2016-09-27 03:23:14.000000000 +0200
@@ -140,12 +140,14 @@ load-module module-suspend-on-idle
 
 ### If autoexit on idle is enabled we want to make sure we only quit
 ### when no local session needs us anymore.
+ifelse(@HAVE_COREAUDIO@, 1, [], [dnl
 .ifexists module-console-kit@PA_SOEXT@
 load-module module-console-kit
 .endif
 .ifexists module-systemd-login@PA_SOEXT@
 load-module module-systemd-login
 .endif
+])dnl
 
 ### Enable positioned event sounds
 load-module module-position-event-sounds
