--- t/00-report-prereqs.t.orig	2018-01-16 19:25:38.000000000 -0800
+++ t/00-report-prereqs.t	2018-01-16 19:25:57.000000000 -0800
@@ -69,7 +69,7 @@
 );
 
 # Add static prereqs to the included modules list
-my $static_prereqs = do 't/00-report-prereqs.dd';
+my $static_prereqs = do './t/00-report-prereqs.dd';
 
 # Merge all prereqs (either with ::Prereqs or a hashref)
 my $full_prereqs = _merge_prereqs(
