--- MANIFEST.orig	2019-02-20 14:53:15.000000000 -0800
+++ MANIFEST	2019-02-21 18:12:59.000000000 -0800
@@ -337,41 +337,6 @@
 PPPort_pm.PL
 PPPort_xs.PL
 soak
-t/call.t
-t/cop.t
-t/exception.t
-t/format.t
-t/grok.t
-t/gv.t
-t/HvNAME.t
-t/limits.t
-t/magic.t
-t/memory.t
-t/mess.t
-t/misc.t
-t/mPUSH.t
-t/MY_CXT.t
-t/newCONSTSUB.t
-t/newRV.t
-t/newSV_type.t
-t/newSVpv.t
-t/podtest.t
-t/ppphtest.t
-t/pv_tools.t
-t/pvs.t
-t/shared_pv.t
-t/snprintf.t
-t/sprintf.t
-t/strlfuncs.t
-t/Sv_set.t
-t/sv_xpvf.t
-t/SvPV.t
-t/SvREFCNT.t
-t/testutil.pl
-t/threads.t
-t/uv.t
-t/variables.t
-t/warn.t
 typemap
 META.yml                                 Module YAML meta-data (added by MakeMaker)
 META.json                                Module JSON meta-data (added by MakeMaker)
