--- fftw.conf.orig	2015-03-04 07:08:12.000000000 -0800
+++ fftw.conf	2015-03-04 07:09:23.000000000 -0800
@@ -6,10 +6,10 @@
 %FFTW_CONFIG = (
 
   # Location to search for the FFTW libs, overrides default if defined
-  FFTW_LIBS => undef,    # [ '/lib','/usr/lib','/usr/local/lib'],
+  FFTW_LIBS => [ '__PREFIX__/lib' ],    # [ '/lib','/usr/lib','/usr/local/lib'],
 
   # Location to find FFTW includes, overrides default if defined
-  FFTW_INC => undef,     # ['/usr/include/','/usr/local/include'],
+  FFTW_INC => [ '__PREFIX__/include' ],  # ['/usr/include/','/usr/local/include'],
 
   # FFTW Numeric Precision Type to link in: (double or single precision)
   FFTW_TYPE => 'double', # 'single'
