From 280ee5c25061803665db57389f94a3262fcbef7d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 10 Sep 2024 01:23:46 +0800
Subject: [PATCH] tcpkali_iface.c: define AI_NUMERICSERV when undefined

---
 src/tcpkali_iface.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/tcpkali_iface.c src/tcpkali_iface.c
index 33d2a49..8e98355 100644
--- src/tcpkali_iface.c
+++ src/tcpkali_iface.c
@@ -46,6 +46,10 @@
 #include "tcpkali_logging.h"
 #include "tcpkali_iface.h"
 
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
 /*
  * Note: struct sockaddr_in6 is larger than struct sockaddr, hence
  * the storage should be bigger. However, we shall not dereference
-- 
2.46.0

