--- source/dub/dub.d.orig	2017-02-14 02:35:14.000000000 -0700
+++ source/dub/dub.d	2017-03-17 22:27:45.000000000 -0700
@@ -166,7 +166,7 @@
 			m_dirs.systemSettings = Path(environment.get("ProgramData")) ~ "dub/";
 			m_dirs.userSettings = Path(environment.get("APPDATA")) ~ "dub/";
 		} else version(Posix){
-			m_dirs.systemSettings = Path("/var/lib/dub/");
+			m_dirs.systemSettings = Path("__MACPORTS_PREFIX__/var/lib/dub/");
 			m_dirs.userSettings = Path(environment.get("HOME")) ~ ".dub/";
 			if (!m_dirs.userSettings.absolute)
 				m_dirs.userSettings = Path(getcwd()) ~ m_dirs.userSettings;
