--- configure	2024-12-04 11:36:10.000000000 -0500
+++ configure.orig	2024-09-27 19:32:37.000000000 -0400
@@ -15923,13 +15923,13 @@
 
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
-	RUST_TARGET_SUBDIR=debug
+	RUST_TARGET_SUBDIR="$CARGO_BUILD_TARGET/debug"
 
 else case e in #(
   e)
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
 printf "%s\n" "no" >&6; }
-	RUST_TARGET_SUBDIR=release
+	RUST_TARGET_SUBDIR="$CARGO_BUILD_TARGET/release"
  ;;
 esac
 fi
@@ -15956,7 +15956,7 @@
 if test "x$cross_compiling" = "xyes"
 then :
 
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
+	RUST_TARGET_SUBDIR="$RUST_TARGET_SUBDIR"
 
 fi
 
