--- configure.orig	2021-10-05 11:54:20.000000000 -0400
+++ configure	2021-10-05 11:54:30.000000000 -0400
@@ -14734,13 +14734,13 @@
 
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	RUST_TARGET_SUBDIR=debug
+	RUST_TARGET_SUBDIR="$CARGO_BUILD_TARGET/debug"
 
 else
 
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
-	RUST_TARGET_SUBDIR=release
+	RUST_TARGET_SUBDIR="$CARGO_BUILD_TARGET/release"
 
 fi
  if test "x$debug_release" = "xyes"; then
@@ -14765,7 +14765,7 @@
 
 if test "x$cross_compiling" = "xyes"; then :
 
-	RUST_TARGET_SUBDIR="$RUST_TARGET/$RUST_TARGET_SUBDIR"
+	RUST_TARGET_SUBDIR="$RUST_TARGET_SUBDIR"
 
 fi
 
