--- src/otrconnection.cpp.orig	2007-11-13 02:21:18.000000000 +0100
+++ src/otrconnection.cpp	2008-02-02 16:11:06.000000000 +0100
@@ -54,6 +54,9 @@
 	ui_ops.gone_insecure = (*OtrConnection::cb_gone_insecure);
 	ui_ops.still_secure = (*OtrConnection::cb_still_secure);
 	ui_ops.log_message = (*OtrConnection::cb_log_message);
+	ui_ops.max_message_size = NULL;
+	ui_ops.account_name = NULL;
+	ui_ops.account_name_free = NULL;
 }
 
 
