--- colors-bright.c.orig	2009-01-01 20:46:45.000000000 -0600
+++ colors-bright.c	2011-06-12 19:33:09.000000000 -0500
@@ -24,7 +24,7 @@
  * activation is done by bit-masking with TM_ACTIVE color. Certain combinations
  * may work, but you have been warned :-)
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 
 #include "colors.h"
 #include "items.h"
--- colors-dark.c.orig	2009-01-01 20:46:45.000000000 -0600
+++ colors-dark.c	2011-06-12 19:33:07.000000000 -0500
@@ -24,7 +24,7 @@
  * activation is done by bit-masking with TM_ACTIVE color. Certain combinations
  * may work, but you have been warned :-)
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 
 #include "colors.h"
 #include "items.h"
--- links.c.orig	2009-01-01 20:46:47.000000000 -0600
+++ links.c	2011-06-12 19:33:04.000000000 -0500
@@ -9,7 +9,7 @@
  * (C) 2002 antrik
  *
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <stdlib.h>
 
 #include "colors.h"
--- main.c.orig	2009-01-01 20:46:48.000000000 -0600
+++ main.c	2011-06-12 19:33:02.000000000 -0500
@@ -13,7 +13,7 @@
  * as argument), and either dump it and quit immediatly (with --dump), or
  * display it interactively and load new files (requested by user) in a loop
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <errno.h>
 #include <locale.h>
 #include <stdio.h>
--- pager.c.orig	2009-01-01 20:46:48.000000000 -0600
+++ pager.c	2011-06-12 19:33:00.000000000 -0500
@@ -11,7 +11,7 @@
  *
  * Displays output page and handles user commands.
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
--- parse-syntax.c.orig	2009-01-01 20:46:49.000000000 -0600
+++ parse-syntax.c	2011-06-12 19:32:58.000000000 -0500
@@ -12,7 +12,7 @@
  * all content. (Text inside the elements.)
  */
 #include <ctype.h>
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- render.c.orig	2009-01-01 20:46:49.000000000 -0600
+++ render.c	2011-06-12 19:32:55.000000000 -0500
@@ -17,9 +17,9 @@
  * like dump().
  */
 
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <stdlib.h>
-#include <ncursesw/term.h>
+#include <term.h>
 #include <string.h>
 
 #include "cfg.h"
--- screen.c.orig	2009-01-01 20:46:49.000000000 -0600
+++ screen.c	2011-06-12 19:32:51.000000000 -0500
@@ -10,11 +10,11 @@
  *
  * Presently the full screen functions aren't used by the main program...
  */
-#include <ncursesw/curses.h>
+#include <curses.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ncursesw/term.h>
+#include <term.h>
 
 #include "cfg.h"
 #include "screen.h"
