https://jira.mongodb.org/browse/SERVER-5575
--- src/SConscript.client.orig	2012-08-28 00:28:11.000000000 -0500
+++ src/SConscript.client	2012-09-24 08:02:47.000000000 -0500
@@ -134,7 +134,7 @@
 # install
 prefix = GetOption("prefix")
 
-env.Install(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
+env.InstallAs(prefix + "/lib", '${LIBPREFIX}mongoclient${LIBSUFFIX}')
 
 for x in clientHeaderDirectories:
     env.Install(prefix + "/include/mongo/" + x,
