--- src/third_party/IntelRDFPMathLib20U1/LIBRARY/float128/dpml_ux_ops.c.orig	2023-06-16 20:55:56.000000000 +0800
+++ src/third_party/IntelRDFPMathLib20U1/LIBRARY/float128/dpml_ux_ops.c	2024-04-10 09:39:36.000000000 +0800
@@ -376,7 +376,7 @@
                 lsd = -lsd;
                 carry = (lsd == 0) ? 0 : -1;
 
-#           if NUM_UX_FRACTION_DIGITS == 4
+#           if NUM_UX_FRACTION_DIGITS == 4  && !defined(__ppc__)
 
                     _F2 = carry - F2;
                     carry = (_F2 == 0) ? carry : -1;
