--- src/third_party/SConscript.orig	2012-08-28 00:28:11.000000000 -0500
+++ src/third_party/SConscript	2012-09-24 08:00:44.000000000 -0500
@@ -33,6 +33,10 @@
     env.StaticLibrary('shim_snappy', ['shim_snappy.cpp'], LIBDEPS=['snappy/snappy'])
 
 if use_system_version_of_library("sm"):
+    if windows:
+        env.Append(CPPDEFINES=['XP_WIN'])
+    else:
+        env.Append(CPPDEFINES=['XP_UNIX'])
     env.StaticLibrary("shim_spidermonkey", ['shim_spidermonkey.cpp'], SYSLIBDEPS=['js'])
 else:
     if windows:
