--- t/Taxonomy.t.orig	2021-12-21 12:52:06.000000000 -0800
+++ t/Taxonomy.t	2021-12-21 13:42:49.000000000 -0800
@@ -166,8 +166,8 @@
     # 'Chlorella vulgaris'   | 3077          | 3077
     # 'Phygadeuon solidus'   | 1763951       | 1763951
     # 'Ovatus'               | 666060        | 666060
-    # 'Phygadeuon ovatus'    | "No hit"      | 666060
-    # 'Trimorus ovatus'      | "No hit"      | 666060
+    # 'Phygadeuon ovatus'    | 2890685       | 2890685
+    # 'Trimorus ovatus'      | "No hit"      | "No hit"
 
     my @ids;
     @ids = $db->get_taxonids('Lissotriton vulgaris');
@@ -179,7 +179,7 @@
     @ids = $db->get_taxonids('Ovatus');
     is $ids[0], 666060, 'Correct: Ovatus';
     @ids = $db->get_taxonids('Phygadeuon ovatus');
-    is $ids[0], 'No hit', 'Correct: No hit';
+    is $ids[0], 2890685, 'Correct: Phygadeuon ovatus';
     @ids = $db->get_taxonids('Trimorus ovatus');
     is $ids[0], 'No hit', 'Correct: No hit';
 }
