--- po/Makefile.in.in.orig	1998-04-26 13:22:40.000000000 +0000
+++ po/Makefile.in.in	2018-08-23 15:55:49.000000000 +0000
@@ -122,7 +122,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r $(MKINSTALLDIRS); then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
