--- CMakeLists.txt.orig	2014-11-06 17:45:47.000000000 +0900
+++ CMakeLists.txt	2015-01-01 20:49:17.000000000 +0900
@@ -250,11 +250,11 @@
   add_subdirectory(messagelist)
   add_subdirectory(templateparser)
 
-  if(QGPGME_FOUND)
-    if(Boost_TOPOLOGICAL_SORT_DIR)
-      macro_optional_add_subdirectory(kleopatra)
-    endif()
-  endif()
+#  if(QGPGME_FOUND)
+#    if(Boost_TOPOLOGICAL_SORT_DIR)
+#      macro_optional_add_subdirectory(kleopatra)
+#    endif()
+#  endif()
 
   # The following components depend on QGpgME.
   add_subdirectory(messageviewer)
