--- blogilo/src/blogger.h.orig	2015-03-13 22:14:25.000000000 +0900
+++ blogilo/src/blogger.h	2015-03-13 22:13:38.000000000 +0900
@@ -42,7 +42,8 @@
 
 namespace KBlog
 {
-
+typedef QMap<QString, QString> AuthData_t;
+    
 class BloggerPrivate;
 
 /**
@@ -93,7 +94,7 @@
                       signal emitted after previous authenticate() call.
       @see authenticated(QMap\<QString,QString\>)
     */
-    void authenticate(const QMap<QString, QString> &authData = QMap<QString, QString>());
+    void authenticate(const AuthData_t &authData = AuthData_t());
 
     /**
       Sets Google OAuth application API key.
