--- kleopatra/uiserver/assuanserverconnection.cpp.orig	2013-03-08 00:43:55.000000000 +0900
+++ kleopatra/uiserver/assuanserverconnection.cpp	2013-03-08 01:31:27.000000000 +0900
@@ -79,7 +79,7 @@
 #include <kleo-assuan.h>
 
 #include <boost/type_traits/remove_pointer.hpp>
-#ifndef _WIN32_WCE
+#if !defined(_WIN32_WCE) && !defined(Q_MOC_RUN)
 #include <boost/lexical_cast.hpp>
 #endif
 #include <boost/bind.hpp>
