--- test.pl.orig	2019-01-24 23:17:53.000000000 -0800
+++ test.pl	2019-01-24 23:18:11.000000000 -0800
@@ -35,7 +35,7 @@
 
    print "============== Running test$jjj.p ==============\n";
    %@ = ();       # Clear error status
-   do "test$jjj.p";
+   do "./test$jjj.p";
    warn $@ if $@; # Report any error detected
    if($dev eq '/PNG' && $^O =~ /mswin32/i) {
      system("ren pgplot.png pgplot_$jjj.png");
