--- tardiff.orig	2005-05-17 07:52:27.000000000 -0500
+++ tardiff	2011-09-09 19:19:16.000000000 -0500
@@ -41,7 +41,7 @@
 			$opt_modified = 1;
 		}elsif(($arg eq "--list") or ($arg eq "-l")){
 			$opt_list = 1;
-		}elsif(($arg eq "--autoskip") or ($arg eq "-s")){
+		}elsif(($arg eq "--autoskip") or ($arg eq "-a")){
 			$opt_autoskip = 1;
 		}elsif(($arg eq "--stats") or ($arg eq "-s")){
 			$opt_stats = 1;
@@ -67,9 +67,9 @@
 	my $tarball = shift(@_);
 
 	my $flag = "";
-	if($tarball =~ /\.gz$/){
+	if($tarball =~ /\.t?gz$/){
 		$flag = "-z";
-	}elsif($tarball =~ /\.bz2$/){
+	}elsif($tarball =~ /\.t?bz2$/){
 		$flag = "-j";
 	}
 
