--- mythtv/configure.orig	2013-06-04 22:25:57.000000000 -0400
+++ mythtv/configure	2013-06-05 20:48:04.000000000 -0400
@@ -3001,7 +3001,7 @@
 ranlib="${cross_prefix}${ranlib}"
 strip_default="${cross_prefix}${strip_default}"
 
-sysinclude_default="${sysroot}/usr/include"
+#sysinclude_default="${sysroot}/usr/include"
 
 case "$toolchain" in
     clang-asan)
@@ -4389,7 +4389,7 @@
 set_default nm
 
 set_default $MYTHTV_PATHS_LIST
-append CONFIG_INCLUDEPATH $sysinclude "${sysroot}${prefix}"
+#append CONFIG_INCLUDEPATH $sysinclude "${sysroot}${prefix}"
 
 append CCONFIG "$compile_type"
 if test x$compile_type = x"profile" ; then
