--- n3-mode.el.orig	2010-10-07 19:24:02.000000000 +0200
+++ n3-mode.el          2014-10-30 15:56:44.000000000 +0100
@@ -42,7 +42,8 @@
 )
 
 ;;(define-generic-mode 'n3-mode
-(define-derived-mode n3-mode fundamental-mode
+;;;###autoload
+(define-derived-mode n3-mode fundamental-mode "n3-mode"
   ;; setup tab key not working :/
   ;;(setq c-basic-offset 4)
 
@@ -57,4 +58,11 @@
 
   ;; description
   "Mode for Notation 3 documents."
 )
+
+;;;###autoload
+(add-to-list 'auto-mode-alist '("\\.n3\\'" . n3-mode))
+;;;###autoload
+(add-to-list 'auto-mode-alist '("\\.owl\\'" . n3-mode))
+
+(provide 'n3-mode)
