--- cmulocal/bsd_sockets.m4~	2025-03-09 12:44:21
+++ cmulocal/bsd_sockets.m4	2025-03-09 12:50:47
@@ -9,19 +9,10 @@
 AC_DEFUN(CMU_SOCKETS, [
 	save_LIBS="$LIBS"
 	LIB_SOCKET=""
-	AC_CHECK_FUNC(connect, :,
-		AC_CHECK_LIB(nsl, gethostbyname,
-			     LIB_SOCKET="-lnsl $LIB_SOCKET")
-		AC_CHECK_LIB(socket, connect,
-			     LIB_SOCKET="-lsocket $LIB_SOCKET")
-	)
-	LIBS="$LIB_SOCKET $save_LIBS"
-	AC_CHECK_FUNC(res_search, :,
-                AC_CHECK_LIB(resolv, res_search,
-                              LIB_SOCKET="-lresolv $LIB_SOCKET") 
-        )
-	LIBS="$LIB_SOCKET $save_LIBS"
+	AC_SEARCH_LIBS([gethostbyname], [nsl])
+	AC_SEARCH_LIBS([connect], [socket])
+	AC_SEARCH_LIBS([res_search], [resolv])
 	AC_CHECK_FUNCS(dn_expand dns_lookup)
 	LIBS="$save_LIBS"
-	AC_SUBST(LIB_SOCKET)
+	AC_SUBST([LIB_SOCKET])
 	])
