--- README.unix.orig	2009-01-03 23:00:57.000000000 +0000
+++ README.unix	2009-01-03 23:03:01.000000000 +0000
@@ -188,7 +188,7 @@
 
      TMPDIR
           Root of directory to store partial messages awaiting 
-          reassembly.  Default is "/usr/tmp".   Partial messages
+          reassembly.  Default is "/tmp".   Partial messages
 	  are stored in subdirectories of $TMPDIR/m-prts-$USER/
 
 
--- mkreadme.pl.orig	2009-01-03 23:01:26.000000000 +0000
+++ mkreadme.pl	2009-01-03 23:03:38.000000000 +0000
@@ -413,7 +413,7 @@
 =unix
      TMPDIR
           Root of directory to store partial messages awaiting 
-          reassembly.  Default is "/usr/tmp".   Partial messages
+          reassembly.  Default is "/tmp".   Partial messages
 	  are stored in subdirectories of $TMPDIR/m-prts-$USER/
 
 =pc os2
--- unixos.c.orig	2009-01-03 23:01:38.000000000 +0000
+++ unixos.c	2009-01-03 23:04:09.000000000 +0000
@@ -90,7 +90,7 @@
 	strcpy(buf, getenv("TMPDIR"));
     }
     else {
-	strcpy(buf, "/usr/tmp");
+	strcpy(buf, "/tmp");
     }
     strcat(buf, "/m-prts-");
     p = getenv("USER");
--- unixpk.c.orig	2009-01-03 23:01:52.000000000 +0000
+++ unixpk.c	2009-01-03 23:04:35.000000000 +0000
@@ -164,7 +164,7 @@
 	    strcpy(fnamebuf, getenv("TMPDIR"));
 	}
 	else {
-	    strcpy(fnamebuf, "/usr/tmp");
+	    strcpy(fnamebuf, "/tmp");
 	}
 	strcat(fnamebuf, "/mpackXXXXXX");
 	mktemp(fnamebuf);
--- unixpk.man.orig	2009-01-03 23:02:10.000000000 +0000
+++ unixpk.man	2009-01-03 23:04:55.000000000 +0000
@@ -136,7 +136,7 @@
 .SH ENVIRONMENT
 .TP
 .B TMPDIR
-Directory to store temporary files.  Default is /usr/tmp.
+Directory to store temporary files.  Default is /tmp.
 .TP
 .B SPLITSIZE
 Default value of the -m switch.
