--- configure.in.orig	2009-03-28 01:39:07.000000000 -0700
+++ configure.in	2009-03-30 16:31:47.000000000 -0700
@@ -1152,7 +1152,9 @@
   sdl_minor_version=1
   sdl_micro_version=6
   save_CPPFLAGS="$CPPFLAGS"
+  save_LIBS="$LIBS"
   CPPFLAGS="$CPPFLAGS $SDL_CFLAGS"
+  LIBS="$LIBS $SDL_LIBS"
   AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdio.h>
 #include <stdlib.h>
@@ -1182,6 +1184,7 @@
   [AC_MSG_RESULT([cross compiling; assumed OK...])
    have_sdl="yes"])
   CPPFLAGS="$save_CPPFLAGS"
+  LIBS="$save_LIBS"
 fi
 TC_PKG_HAVE(sdl, SDL)
 
