--- transcode-1.1.7/configure.in.orig	2011-11-19 20:50:27
+++ transcode-1.1.7/configure.in	2026-09-05 13:48:31
@@ -795,7 +795,7 @@
   lame_version_t lv;
   get_lame_version_numerical(&lv);
   if(lv.alpha || lv.beta) lv.minor--;
-  printf("%d%d\n", lv.major, lv.minor);
+  printf("%d%02d\n", lv.major, lv.minor);
   return 0;
 }
 ]])],
@@ -1075,6 +1075,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "SDL.h"
+#define main main
 int main (int argc, char *argv[])
 {
    if ( (SDL_MAJOR_VERSION > $sdl_major_version) ||
