--- spyder/config/main.py.orig	2019-01-05 10:50:15.000000000 -0500
+++ spyder/config/main.py	2019-01-07 10:04:33.000000000 -0500
@@ -668,7 +668,7 @@
             ('lsp-server', {
                 'python': {
                     'index': 0,
-                    'cmd': 'pyls',
+                    'cmd': '@@PYLS_BIN_NAME@@',
                     'args': '--host {host} --port {port} --tcp',
                     'host': '127.0.0.1',
                     'port': 2087,
