--- spyder/app/mainwindow.py.orig	2018-10-19 15:04:42.000000000 -0400
+++ spyder/app/mainwindow.py	2018-10-20 11:20:38.000000000 -0400
@@ -219,7 +219,7 @@
             return file_uri(osp.join(doc_path, python_chm[0]))
     else:
         vinf = sys.version_info
-        doc_path = '/usr/share/doc/python%d.%d/html' % (vinf[0], vinf[1])
+        doc_path = '@@PREFIX@@/share/doc/python%d%d-doc' % (vinf[0], vinf[1])
     python_doc = osp.join(doc_path, "index.html")
     if osp.isfile(python_doc):
         return file_uri(python_doc)
