--- spyder/plugins/editor/lsp/transport/main.py.orig	2018-10-19 15:04:42.000000000 -0400
+++ spyder/plugins/editor/lsp/transport/main.py	2018-10-20 11:20:38.000000000 -0400
@@ -46,7 +46,7 @@
                     help="Initial current working directory used to "
                          "initialize ls-server")
 parser.add_argument('--server',
-                    default='pyls',
+                    default='@@PYLS_BIN_NAME@@',
                     help='Instruction executed to start the language server')
 parser.add_argument('--external-server',
                     action="store_true",
