--- spyder/app/mainwindow.py.orig	2018-12-15 10:39:50.000000000 -0500
+++ spyder/app/mainwindow.py	2018-12-18 10:17:21.000000000 -0500
@@ -220,7 +220,7 @@
             return file_uri(osp.join(doc_path, python_chm[0]))
     else:
         vinf = sys.version_info
-        doc_path = '/usr/share/doc/python%d.%d/html' % (vinf[0], vinf[1])
+        doc_path = '@@PREFIX@@/share/doc/python%d%d-doc' % (vinf[0], vinf[1])
     python_doc = osp.join(doc_path, "index.html")
     if osp.isfile(python_doc):
         return file_uri(python_doc)
