--- spyplugins/ui/pylint/widgets/pylintgui.py.orig	2016-06-07 13:57:54.000000000 -0500
+++ spyplugins/ui/pylint/widgets/pylintgui.py	2016-06-07 13:59:10.000000000 -0500
@@ -47,16 +47,9 @@
     _ = gettext.gettext
 
 
-PYLINT = 'pylint'
-if PY3:
-    if programs.find_program('pylint3'):
-        PYLINT = 'pylint3'
-    elif programs.find_program('python3-pylint'):
-        PYLINT = 'python3-pylint'
-
-
-locale_codec = QTextCodec.codecForLocale()
-PYLINT_PATH = programs.find_program(PYLINT)
+PYLINT = '@@LINT_BIN_NAME@@'
+locale_codec = QTextCodec.codecForLocale()
+PYLINT_PATH = programs.find_program('@@LINT_BIN_NAME@@')
 
 
 def get_pylint_version():
