--- spyder/plugins/editor/lsp/transport/producer.py.orig	2018-12-15 10:39:50.000000000 -0500
+++ spyder/plugins/editor/lsp/transport/producer.py	2018-12-18 10:10:55.000000000 -0500
@@ -42,7 +42,7 @@
 
     def __init__(self, host='127.0.0.1', port=2087, workspace=getcwd(),
                  use_external_server=False, zmq_in_port=7000,
-                 zmq_out_port=7001, server='pyls', server_args=['--tcp']):
+                 zmq_out_port=7001, server='@@PYLS_BIN_NAME@@', server_args=['--tcp']):
         self.req_status = {}
         self.host = host
         self.port = port
