--- spyderlib/app/spyder.py.orig	2011-09-05 18:29:00.000000000 -0500
+++ spyderlib/app/spyder.py	2011-09-08 08:36:55.000000000 -0500
@@ -142,7 +142,7 @@
             return osp.join(doc_path, python_chm[0])
     else:
         vinf = sys.version_info
-        doc_path = '/usr/share/doc/python%d.%d/html' % (vinf[0], vinf[1])
+        doc_path = '@@PREFIX@@/share/doc/python%d%d-doc' % (vinf[0], vinf[1])
     python_doc = osp.join(doc_path, "index.html")
     if osp.isfile(python_doc):
         return python_doc
