--- spyderlib/widgets/sourcecode/base.py.orig	2013-06-10 14:24:56.000000000 -0500
+++ spyderlib/widgets/sourcecode/base.py	2013-06-10 14:26:04.000000000 -0500
@@ -156,7 +156,8 @@
     
     def focusOutEvent(self, event):
         event.ignore()
-        self.hide()
+        if event.reason() != Qt.ActiveWindowFocusReason:
+            self.hide()
         
     def item_selected(self, item=None):
         if item is None:
