--- DESCRIPTION.orig	2023-02-05 23:25:30.000000000 +0800
+++ DESCRIPTION	2023-04-28 11:39:25.000000000 +0800
@@ -14,8 +14,8 @@
 Imports: Rcpp, parallel, slam (>= 0.1-37), stats, tools, utils,
         graphics, xml2
 LinkingTo: BH, Rcpp
-Suggests: antiword, filehash, methods, pdftools, Rcampdf, Rgraphviz,
-        Rpoppler, SnowballC, testthat, tm.lexicon.GeneralInquirer
+Suggests: antiword, filehash, methods, pdftools, Rgraphviz,
+        Rpoppler, SnowballC, testthat
 Description: A framework for text mining applications within R.
 License: GPL-3
 URL: https://tm.r-forge.r-project.org/
