--- src/include/utils/pg_locale.h.orig	2022-08-09 04:44:29.000000000 +0800
+++ src/include/utils/pg_locale.h	2022-10-25 01:49:09.000000000 +0800
@@ -17,6 +17,9 @@
 #endif
 #ifdef USE_ICU
 #include <unicode/ucol.h>
+#ifdef bool
+#undef bool
+#endif
 #endif
 
 #include "utils/guc.h"

--- src/backend/utils/adt/formatting.c.orig	2022-10-25 02:07:10.000000000 +0800
+++ src/backend/utils/adt/formatting.c	2022-10-25 02:07:21.000000000 +0800
@@ -81,6 +81,9 @@
 
 #ifdef USE_ICU
 #include <unicode/ustring.h>
+#ifdef bool
+#undef bool
+#endif
 #endif
 
 #include "catalog/pg_collation.h"
