diff --git a/sslh-fork.c b/sslh-fork.c
index 74335d7..40a1b5a 100644
--- a/sslh-fork.c
+++ b/sslh-fork.c
@@ -164,6 +164,13 @@ void set_listen_procname(struct listen_endpoint *listen_socket)
 }
 
 
+/* At least MacOS does not know these two options, so define them to something
+ * equivalent for our use case */
+#ifndef ENONET
+#define ENONET EWOULDBLOCK
+#endif
+/* /MacOS kludge */
+
 /* TCP listener: connections, fork a child for each new connection 
  * IN: 
  *      endpoint: array of listening endpoint objects
