From 574d9b9f0c151c7aa0bb06f1d3d46b1d4ee275f6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 21 Jan 2024 05:46:30 +0800
Subject: [PATCH] sqliteInt.h: fix pointer size for ppc64

---
 src/sqliteInt.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/sqliteInt.h src/sqliteInt.h
index 40af5a18..00fb9ad9 100644
--- src/sqliteInt.h
+++ src/sqliteInt.h
@@ -873,7 +873,7 @@ typedef INT16_TYPE LogEst;
 #   define SQLITE_PTRSIZE __SIZEOF_POINTER__
 # elif defined(i386)     || defined(__i386__)   || defined(_M_IX86) ||    \
        defined(_M_ARM)   || defined(__arm__)    || defined(__x86)   ||    \
-      (defined(__APPLE__) && defined(__POWERPC__)) ||                     \
+      (defined(__APPLE__) && defined(__ppc__))  ||                        \
       (defined(__TOS_AIX__) && !defined(__64BIT__))
 #   define SQLITE_PTRSIZE 4
 # else
