--- gnome-dictionary/libgdict/Makefile.am.orig	2011-11-30 16:58:33.000000000 -0800
+++ gnome-dictionary/libgdict/Makefile.am	2011-11-30 16:59:38.000000000 -0800
@@ -12,9 +12,7 @@
 	-DGNOMELOCALEDIR=\""$(gnomeutilslocaledir)"\" 		\
 	-DGDICTSOURCESDIR=\""$(datadir)/gdict-1.0/sources"\" 	\
 	-DGDICT_ENABLE_INTERNALS=1 				\
-	-DG_DISABLE_DEPRECATED					\
 	-DGDK_DISABLE_DEPRECATED				\
-	-DGTK_DISABLE_DEPRECATED				\
 	-DPANGO_DISABLE_DEPRECATED				\
 	-DG_DISABLE_SINGLE_INCLUDES				\
 	-DGTK_DISABLE_SINGLE_INCLUDES				\
