--- make-rules/inputenc/make-inp-rules.pl.orig	2014-05-12 01:05:16.000000000 -0700
+++ make-rules/inputenc/make-inp-rules.pl	2018-01-13 04:58:08.000000000 -0700
@@ -8,7 +8,7 @@
 
 
 while (<STDIN>) {
-  if (/\\indexentry{(.+)--([8-9a-f][0-9a-f])\}\{1\}/) {
+  if (/\\indexentry\{(.+)--([8-9a-f][0-9a-f])\}\{1\}/) {
     if (!($1 =~ /inputenc Error|\@inpenc\@undefined/)) {
       $i = hex($2);
       $macro = $1;
