--- make-rules/inputenc/Makefile.am.orig	2011-09-25 02:53:44.000000000 -0400
+++ make-rules/inputenc/Makefile.am	2011-09-25 02:54:23.000000000 -0400
@@ -43,7 +43,7 @@
 latin.xdy : latin1.xdy latin2.xdy latin3.xdy
 	echo ';; Generated from latin?.xdy' >latin.xdy
 	echo >>latin.xdy
-	sort -u latin?.xdy | grep -v '^;' >>latin.xdy
+	LC_ALL=C sort -u latin?.xdy | grep -v '^;' >>latin.xdy
 
 %.xdy %.idx :
 	$(PERL) $(srcdir)/make-testidx.pl TS1,T2A,T1 $* >$*.tex
