--- libscidavis/src/ColorMapEditor.cpp.orig	2016-06-28 21:54:40.000000000 +0900
+++ libscidavis/src/ColorMapEditor.cpp	2016-06-28 21:55:06.000000000 +0900
@@ -124,7 +124,7 @@
 		
 	QColor c = QColor(map.rgb(QwtDoubleInterval(0, 1), colors[i]));
 	it = new QTableWidgetItem(c.name());
-	it->setFlags(!Qt::ItemIsEditable);
+//	it->setFlags(!Qt::ItemIsEditable);
 	it->setBackground(QBrush(c));
 	it->setForeground(QBrush(c));
     table->setItem(i, 1, it);
@@ -156,7 +156,7 @@
 table->setItem(row, 0, it);
 		
 it = new QTableWidgetItem(c.name());
-it->setFlags(!Qt::ItemIsEditable);
+//it->setFlags(!Qt::ItemIsEditable);
 it->setBackground(QBrush(c));
 it->setForeground(QBrush(c));
 table->setItem(row, 1, it);
