diff -ur ../kate-4.13.3-orig/part/view/kateviewinternal.cpp ./part/view/kateviewinternal.cpp
--- ../kate-4.13.3-orig/part/view/kateviewinternal.cpp	2014-07-10 12:10:57.000000000 +0900
+++ ./part/view/kateviewinternal.cpp	2014-09-23 11:34:13.000000000 +0900
@@ -3275,7 +3275,7 @@
 void KateViewInternal::wheelEvent(QWheelEvent* e)
 {
   // zoom text, if ctrl is pressed
-  if (e->modifiers() == Qt::ControlModifier) {
+  if (e->modifiers() == Qt::ControlModifier && KGlobalSettings::wheelMouseZooms() ) {
     if (e->delta() > 0) {
       slotIncFontSizes();
     } else {
Only in ./part/view: kateviewinternal.cpp.orig
