--- autogen.sh.orig	2017-11-19 19:01:02.000000000 -0800
+++ autogen.sh	2017-11-19 19:01:32.000000000 -0800
@@ -6,12 +6,6 @@
   exit 1
 fi
 
-GIT=$(which git 2>/dev/null)
-if test $? -ne 0; then
-  echo "No 'git' found. You must install the git package."
-  exit 1
-fi
-
 #  create m4 before gtkdocize
 mkdir -p m4 2>/dev/null
 
@@ -27,8 +21,6 @@
   $GTKDOCIZE
 fi
 
-$GIT submodule init
-$GIT submodule update
 $AUTORECONF --install --force --symlink
 
 echo
