--- CMakeLists.txt.orig	2013-07-19 09:32:26.000000000 -0700
+++ CMakeLists.txt	2013-08-25 02:13:57.000000000 -0700
@@ -87,7 +87,7 @@
   if (WIN32)
     set (Boost_USE_STATIC_LIBS ON)
   endif ()
-  find_package (Boost 1.40.0 COMPONENTS date_time thread)
+  find_package (Boost 1.40.0 COMPONENTS date_time thread system)
   if (NOT Boost_FOUND)
     print_error ("Boost Datetime/Thread" "Boost")
   endif ()
@@ -318,7 +318,7 @@
 endif ()
 
 if (NOT WIN32)
-  add_definitions ("-Wall -Werror")
+  add_definitions ("-Wall")
 endif ()
 
 include_directories ("src")
@@ -401,8 +401,8 @@
   list (APPEND GEOCODER_DEPS ${COMMON_DEPS})
   # Note that the subset of base/ on which the geocoder relies is implemented
   # on top of Boost header-only libraries (e.g. scoped_ptr.hpp).
-  target_link_libraries (geocoding ${LIBRARY_DEPS})
-  target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
+  target_link_libraries (geocoding phonenumber ${LIBRARY_DEPS})
+  target_link_libraries (geocoding-shared phonenumber-shared ${LIBRARY_DEPS})
 endif ()
 
 # Build a specific library for testing purposes.
