--- CMakeLists.txt.orig	2014-09-25 05:05:46.000000000 -0700
+++ CMakeLists.txt	2014-10-17 09:59:32.000000000 -0700
@@ -331,7 +331,7 @@
 endif ()
 
 if (NOT WIN32)
-  add_definitions ("-Wall -Werror")
+  add_definitions ("-Wall")
 endif ()
 
 include_directories ("src")
@@ -414,8 +414,8 @@
   list (APPEND GEOCODER_DEPS ${COMMON_DEPS})
   # Note that the subset of base/ on which the geocoder relies is implemented
   # on top of Boost header-only libraries (e.g. scoped_ptr.hpp).
-  target_link_libraries (geocoding ${LIBRARY_DEPS})
-  target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
+  target_link_libraries (geocoding phonenumber ${LIBRARY_DEPS})
+  target_link_libraries (geocoding-shared phonenumber-shared ${LIBRARY_DEPS})
 endif ()
 
 # Build a specific library for testing purposes.
