--- configure.orig	2008-05-21 10:10:32.000000000 -0600
+++ configure	2008-11-06 19:32:24.000000000 -0700
@@ -51,6 +51,16 @@
    exit 1 ;;
 esac
 
+case $MACHINE in
+  *Darwin[56]) CCC=${CC-cc};;
+  *)           CCC=${CC-gcc};;
+esac
+#case $OS in
+# SunOS)  LIBCOMPAT=" -lsocket"  ;;
+# NetBSD) LIBCOMPAT=" -lcompat"  ;;
+# *)      LIBCOMPAT=  ;;
+#esac
+
 # We need a working `echo' command: at least Solaris2.6 may not have it.
 case `echo -n hello | wc -c | ( read n ; echo $n )` in
     5) ;;
@@ -164,16 +174,6 @@
   *) ;;
 esac
 
-case $MACHINE in
-  *Darwin[56]) CCC=${CC-cc};;
-  *)           CCC=${CC-gcc};;
-esac
-#case $OS in
-# SunOS)  LIBCOMPAT=" -lsocket"  ;;
-# NetBSD) LIBCOMPAT=" -lcompat"  ;;
-# *)      LIBCOMPAT=  ;;
-#esac
-
 # Detect Haskell compilers and choose one for building with.
 # (Also generates a little script for configuring hmake later.)
 if [ -d src ]	# Delay this step if it is a binary (src-less) distribution.
