--- Makefile.orig	2008-03-18 02:16:56.000000000 +0100
+++ Makefile	2008-03-18 02:17:41.000000000 +0100
@@ -172,6 +172,7 @@
 	$(MAKE)                    -C libavdevice install-libs
 	$(MAKE-$(CONFIG_PP))       -C libpostproc install-libs
 	$(MAKE-$(CONFIG_SWSCALER)) -C libswscale  install-libs
+	$(MAKE-$(CONFIG_AVFILTER)) -C libavfilter  install-libs
 
 install-headers:
 	$(MAKE)              -C libavutil   install-headers
