--- libavcodec/ppc/check_altivec.c.orig	2008-12-05 20:07:27.000000000 +0900
+++ libavcodec/ppc/check_altivec.c	2008-12-08 05:20:52.000000000 +0900
@@ -23,6 +23,18 @@
  */
 
 #ifdef __APPLE__
+#ifndef u_char
+typedef unsigned char           u_char;
+#endif /* u_char */
+#ifndef u_short
+typedef unsigned short          u_short;
+#endif /* u_short */
+#ifndef u_int
+typedef unsigned int            u_int;
+#endif /* u_int */
+#ifndef u_long
+typedef unsigned long           u_long;
+#endif /* u_long */
 #include <sys/sysctl.h>
 #elif __AMIGAOS4__
 #include <exec/exec.h>
