--- libf95.a-0.93/intrinsics/matmul10.c.orig	2008-06-23 10:44:29.000000000 +0900
+++ libf95.a-0.93/intrinsics/matmul10.c	2014-10-18 11:32:28.000000000 +0900
@@ -899,7 +899,7 @@
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "movsbl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (q), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (q), "m" (temp) : EAX
 
 #include "matmul10.c"
 
@@ -913,7 +913,7 @@
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "movswl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (q), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (q), "m" (temp) : EAX
 
 #include "matmul10.c"
 
@@ -925,7 +925,7 @@
                  "fldt ( %" EAX ")\n" : : "m" (p) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
-                 "fild ( %" EAX ")\n" : : "m" (q) : EAX
+                 "fildl ( %" EAX ")\n" : : "m" (q) : EAX
 
 #include "matmul10.c"
 
@@ -972,7 +972,7 @@
 #define LOAD_A   "mov %0, %" EAX "\n" \
                  "movsbl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (p), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (p), "m" (temp) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" : : "m" (q) : EAX
@@ -987,7 +987,7 @@
 #define LOAD_A   "mov %0, %" EAX "\n" \
                  "movswl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (p), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (p), "m" (temp) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" : : "m" (q) : EAX
@@ -999,7 +999,7 @@
 #define SUBROUTINE22     prefix(matmul22_i4r10)
 
 #define LOAD_A   "mov %0, %" EAX "\n" \
-                 "fild ( %" EAX ")\n" : : "m" (p), "m" (temp) : EAX
+                 "fildl ( %" EAX ")\n" : : "m" (p), "m" (temp) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" : : "m" (q) : EAX
@@ -1113,7 +1113,7 @@
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "movsbl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (q), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (q), "m" (temp) : EAX
 
 #define LOAD_RESULT "mov %0, %" EAX "\n" \
                     "fldt (%" EAX ")\n" \
@@ -1140,7 +1140,7 @@
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "movswl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (q), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (q), "m" (temp) : EAX
 
 #define LOAD_RESULT "mov %0, %" EAX "\n" \
                     "fldt (%" EAX ")\n" \
@@ -1166,7 +1166,7 @@
                   : : "m" (p) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
-                 "fild (%" EAX ")\n" : : "m" (q) : EAX
+                 "fildl (%" EAX ")\n" : : "m" (q) : EAX
 
 #define LOAD_RESULT "mov %0, %" EAX "\n" \
                     "fldt (%" EAX ")\n" \
@@ -1340,7 +1340,7 @@
 #define LOAD_A   "mov %0, %" EAX "\n" \
                  "movsbl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (p), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (p), "m" (temp) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" \
@@ -1368,7 +1368,7 @@
 #define LOAD_A   "mov %0, %" EAX "\n" \
                  "movswl (%" EAX "), %%eax\n" \
                  "mov %%eax, %1\n" \
-                 "fild %1\n" : : "m" (p), "m" (temp) : EAX
+                 "filds %1\n" : : "m" (p), "m" (temp) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" \
@@ -1394,7 +1394,7 @@
 #define SUBROUTINE22    prefix(matmul22_i4z10)
 
 #define LOAD_A   "mov %0, %" EAX "\n" \
-                 "fild (%" EAX ")\n" : : "m" (p) : EAX
+                 "fildl (%" EAX ")\n" : : "m" (p) : EAX
 
 #define LOAD_B   "mov %0, %" EAX "\n" \
                  "fldt ( %" EAX ")\n" \
