--- gcc.c.orig	2009-10-27 22:53:00.000000000 +0900
+++ gcc.c	2009-10-27 22:59:17.000000000 +0900
@@ -107,6 +107,8 @@
 #define TARGET_OBJECT_SUFFIX ".o"
 #endif
 
+int target_flags;
+
 static const char dir_separator_str[] = { DIR_SEPARATOR, 0 };
 
 /* Most every one is fine with LIBRARY_PATH.  For some, it conflicts.  */
@@ -6095,8 +6097,6 @@
 
   expandargv (&argc, &argv);
 
-  prune_options (&argc, &argv);
-
 #ifdef GCC_DRIVER_HOST_INITIALIZATION
   /* Perform host dependent initialization when needed.  */
   GCC_DRIVER_HOST_INITIALIZATION;
