From 344d0baed0f6c7e5833cad57f2019e3975c373f7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 16 Apr 2024 11:30:58 +0800
Subject: [PATCH] BuildJaxws.gmk: mkdir that actually works

---
 jaxws/make/BuildJaxws.gmk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git jaxws/make/BuildJaxws.gmk jaxws/make/BuildJaxws.gmk
index 204dd47fb5..ca0aac119a 100644
--- jaxws/make/BuildJaxws.gmk
+++ jaxws/make/BuildJaxws.gmk
@@ -78,7 +78,7 @@ BUILD_JAXWS += $(JAXWS_OUTPUTDIR)/jaxws_classes/META-INF/services/com.sun.tools.
 # for now, even though it actually breaks properties containing # in the value.
 # Using nawk to avoid solaris sed.
 $(JAXWS_OUTPUTDIR)/jaxws_classes/%.properties: $(JAXWS_TOPDIR)/src/share/jaxws_classes/%.properties
-	$(MKDIR) -p $(@D)
+	mkdir -p $(@D)
 	$(RM) $@ $@.tmp
 	$(CAT) $< | LANG=C $(NAWK) '{ sub(/#.*$$/,"#"); print }' > $@.tmp
 	$(MV) $@.tmp $@
@@ -88,7 +88,7 @@ TARGET_PROP_FILES := $(patsubst $(JAXWS_TOPDIR)/src/share/jaxws_classes/%, \
     $(JAXWS_OUTPUTDIR)/jaxws_classes/%, $(JAXWS_SRC_PROP_FILES))
 
 $(JAXWS_OUTPUTDIR)/jaf_classes/%.properties: $(JAXWS_TOPDIR)/src/share/jaf_classes/%.properties
-	$(MKDIR) -p $(@D)
+	mkdir -p $(@D)
 	$(RM) $@ $@.tmp
 	$(CAT) $< | LANG=C $(NAWK) '{ sub(/#.*$$/,"#"); print }' > $@.tmp
 	$(MV) $@.tmp $@
