--- configure.orig	2011-03-22 23:16:57.000000000 +1100
+++ configure	2012-02-16 14:40:36.000000000 +1100
@@ -33191,7 +33191,7 @@
     elif test "$wxUSE_MSW" = 1; then
         OPENGL_LIBS="-lopengl32 -lglu32"
     elif test "$USE_DARWIN" = 1; then
-        OPENGL_LIBS=""
+        OPENGL_LIBS="-lGL -lGLU"
     else
                         echo "$as_me:$LINENO: checking for OpenGL headers" >&5
 echo $ECHO_N "checking for OpenGL headers... $ECHO_C" >&6
