--- lang/sql/sqlite/src/main.c.orig	2012-02-29 19:48:35.000000000 +0100
+++ lang/sql/sqlite/src/main.c	2012-03-24 13:46:37.000000000 +0100
@@ -2647,6 +2647,7 @@ int sqlite3_test_control(int op, ...){
     }
 #endif 
 
+#ifdef SQLITE_TESTCTRL_PGHDRSZ
     /* sqlite3_test_control(SQLITE_TESTCTRL_PGHDRSZ)
     **
     ** Return the size of a pcache header in bytes.
@@ -2655,6 +2656,7 @@ int sqlite3_test_control(int op, ...){
       rc = sizeof(PgHdr);
       break;
     }
+#endif
 
     /* sqlite3_test_control(SQLITE_TESTCTRL_SCRATCHMALLOC, sz, &pNew, pFree);
     **
--- lang/sql/sqlite/src/shell.c.orig	2012-02-29 19:48:35.000000000 +0100
+++ lang/sql/sqlite/src/shell.c	2012-03-24 13:47:32.000000000 +0100
@@ -2238,7 +2238,9 @@ static int do_meta_command(char *zLine, 
       { "reserve",               SQLITE_TESTCTRL_RESERVE                },
       { "optimizations",         SQLITE_TESTCTRL_OPTIMIZATIONS          },
       { "iskeyword",             SQLITE_TESTCTRL_ISKEYWORD              },
+#ifdef SQLITE_TESTCTRL_PGHDRSZ
       { "pghdrsz",               SQLITE_TESTCTRL_PGHDRSZ                },
+#endif
       { "scratchmalloc",         SQLITE_TESTCTRL_SCRATCHMALLOC          },
     };
     int testctrl = -1;
@@ -2283,7 +2285,9 @@ static int do_meta_command(char *zLine, 
         case SQLITE_TESTCTRL_PRNG_SAVE:           
         case SQLITE_TESTCTRL_PRNG_RESTORE:        
         case SQLITE_TESTCTRL_PRNG_RESET:
+#ifdef SQLITE_TESTCTRL_PGHDRSZ
         case SQLITE_TESTCTRL_PGHDRSZ:             
+#endif
           if( nArg==2 ){
             rc = sqlite3_test_control(testctrl);
             printf("%d (0x%08x)\n", rc, rc);
