--- SConstruct.orig	2021-11-08 11:54:15.000000000 +0100
+++ SConstruct	2021-11-08 11:55:19.000000000 +0100
@@ -376,8 +376,8 @@
     apr_libs = ''
     apu_libs = ''
 
-  env.Append(CPPPATH=['$OPENSSL/include'])
-  env.Append(LIBPATH=['$OPENSSL/lib'])
+  env.Prepend(CPPPATH=['$OPENSSL/include'])
+  env.Prepend(LIBPATH=['$OPENSSL/lib'])
 
 
 # If build with gssapi, get its information and define SERF_HAVE_GSSAPI
