--- kdelibs-3.5.10/khtml/ecma/xmlhttprequest.cpp.orig	2007-10-08 18:52:07.000000000 +0900
+++ kdelibs-3.5.10/khtml/ecma/xmlhttprequest.cpp	2010-04-30 10:16:24.000000000 +0900
@@ -342,16 +342,16 @@
 {
   aborted = false;
 
-  if (method == "post") {
-    QString protocol = url.protocol().lower();
+  const QString protocol = url.protocol().lower();
+  // Abandon the request when the protocol is other than "http",
+  // instead of blindly doing a KIO::get on other protocols like file:/.
+  if (!protocol.startsWith("http") && !protocol.startsWith("webdav"))
+  {
+    abort();
+    return;
+  }
 
-    // Abondon the request when the protocol is other than "http",
-    // instead of blindly changing it to a "get" request.
-    if (!protocol.startsWith("http") && !protocol.startsWith("webdav"))
-    {
-      abort();
-      return;
-    }
+  if (method == "post") {
 
     // FIXME: determine post encoding correctly by looking in headers
     // for charset.
@@ -763,11 +763,11 @@
         if (obj.isValid() && obj.inherits(&DOMDocument::info)) {
           DOM::Node docNode = static_cast<KJS::DOMDocument *>(obj.imp())->toNode();
           DOM::DocumentImpl *doc = static_cast<DOM::DocumentImpl *>(docNode.handle());
-          
+
           try {
             body = doc->toString().string();
             // FIXME: also need to set content type, including encoding!
-  
+
           } catch(DOM::DOMException& e) {
             Object err = Error::create(exec, GeneralError, "Exception serializing document");
             exec->setException(err);
