--- pylib/wscript.orig	2018-03-14 20:28:15.000000000 -0700
+++ pylib/wscript	2018-03-30 16:01:44.000000000 -0700
@@ -36,7 +36,7 @@
     ctx(
         before=['pyc', 'pyo'],
         cwd=srcnode,
-        rule='${SRC} >${TGT}',
+        rule='${PYTHON} ${SRC} >${TGT}',
         source=["../wafhelpers/pythonize-header", "../include/ntp_control.h"],
         target=target1,
     )
@@ -44,7 +44,7 @@
     ctx(
         before=['pyc', 'pyo'],
         cwd=srcnode,
-        rule='${SRC} >${TGT}',
+        rule='${PYTHON} ${SRC} >${TGT}',
         source=["../wafhelpers/pythonize-header", "../include/ntp.h"],
         target=target2,
     )
