diff -Naurp libXt-1.0.5.orig/configure.ac libXt-1.0.5/configure.ac
--- configure.ac	2008-02-12 19:02:35.000000000 -0800
+++ configure.ac	2008-02-12 19:02:41.000000000 -0800
@@ -43,8 +43,6 @@ PKG_CHECK_MODULES(XT, sm x11 xproto kbpr
 AC_CHECK_HEADER([alloca.h], AC_DEFINE(INCLUDE_ALLOCA_H, 1, [Define to 1 if Xalloca.h should include <alloca.h>]))
 
 # Map function checks to old Imake #defines
-AC_CHECK_FUNC(poll, AC_DEFINE(USE_POLL,1,
-        [Define to 1 if you have the `poll' function.]))
 AC_CHECK_FUNC(snprintf, AC_DEFINE(USE_SNPRINTF,1,
         [Define to 1 if you have the `snprintf' function.]))
 
