Index: artsc/Makefile.am
===================================================================
RCS file: /home/kde/arts/artsc/Makefile.am,v
retrieving revision 1.14
diff -u -b -u -r1.14 Makefile.am
--- artsc/Makefile.am	23 Mar 2003 20:32:08 -0000	1.14
+++ artsc/Makefile.am	16 Jun 2003 01:57:01 -0000
@@ -9,11 +9,11 @@
 bin_SCRIPTS = artsc-config artsdsp
 
 libartsdsp_la_SOURCES = artsdsp.c
-libartsdsp_la_LDFLAGS = -no-undefined -module $(all_libraries)
+libartsdsp_la_LDFLAGS = -no-undefined $(all_libraries)
 libartsdsp_la_LIBADD = libartsc.la
 
 libartsdsp_st_la_SOURCES = artsc.c artsdsp.c
-libartsdsp_st_la_LDFLAGS = -no-undefined -module $(all_libraries)
+libartsdsp_st_la_LDFLAGS = -no-undefined $(all_libraries)
 libartsdsp_st_la_LIBADD = $(top_builddir)/libltdl/libltdlc.la
 
 libartsc_la_SOURCES = artsc.c
@@ -21,7 +21,7 @@
 libartsc_la_LIBADD = $(top_builddir)/libltdl/libltdlc.la $(LIBPTHREAD) $(USE_THREADS)
 
 libartscbackend_la_SOURCES = artscbackend.cc
-libartscbackend_la_LDFLAGS = -no-undefined -module $(KDE_RPATH) $(all_libraries)
+libartscbackend_la_LDFLAGS = -no-undefined $(KDE_RPATH) $(all_libraries)
 libartscbackend_la_LIBADD = $(FLOWLIBS) \
     $(top_builddir)/soundserver/libsoundserver_idl.la
 
