Index: artsc/artsc-config.in
===================================================================
RCS file: /home/kde/arts/artsc/artsc-config.in,v
retrieving revision 1.7
diff -u -b -u -r1.7 artsc-config.in
--- artsc/artsc-config.in	18 Mar 2003 12:31:09 -0000	1.7
+++ artsc/artsc-config.in	16 Jun 2003 01:57:23 -0000
@@ -27,7 +27,7 @@
 do
   case $1 in
     --libs)
-	  flags="$flags -L$libdir $libdl -lartsc @USE_THREADS@ @LIBPTHREAD@ @GLIB_LDFLAGS@ @GLIB_LIBADD@"
+	  flags="$flags -L$libdir $libdl -lartsc @USE_THREADS@ @LIBPTHREAD@ @GLIB_LDFLAGS@ @GLIB_LIBADD@ @FRAMEWORK_COREAUDIO@"
 	  ;;
     --cflags)
 	  flags="$flags -I$includedir/artsc @GLIB_CFLAGS@"
