--- configure.orig	2014-06-08 15:27:19.000000000 -0400
+++ configure	2014-06-08 15:27:45.000000000 -0400
@@ -9612,8 +9612,8 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for PHP header files" >&5
 $as_echo_n "checking for PHP header files... " >&6; }
         case $PHP in
-      *5)
-	PHPCONFIG=`echo "$PHP"|sed 's/5$/-config5/'` ;;
+      *5*)
+	PHPCONFIG=`echo "$PHP"|sed 's/\(5.*\)$/-config\1/'` ;;
       *)
 	PHPCONFIG=$PHP-config ;;
     esac
