--- examples/gmsk_radio.grc.orig
+++ examples/gmsk_radio.grc
@@ -1,23 +1,23 @@
-<?xml version='1.0' encoding='ASCII'?>
-<?grc format='1' created='3.7.6'?>
+<?xml version='1.0' encoding='utf-8'?>
+<?grc format='1' created='3.7.11'?>
 <flow_graph>
   <timestamp>Fri Sep 12 22:58:12 2014</timestamp>
   <block>
     <key>options</key>
     <param>
-      <key>id</key>
-      <value>gmsk_radio</value>
+      <key>author</key>
+      <value></value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>window_size</key>
+      <value>1480, 1424</value>
     </param>
     <param>
-      <key>title</key>
-      <value>GMSK Modem</value>
+      <key>category</key>
+      <value>gr-mac</value>
     </param>
     <param>
-      <key>author</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
@@ -25,65 +25,73 @@
       <value></value>
     </param>
     <param>
-      <key>window_size</key>
-      <value>1480, 1424</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
-      <key>generate_options</key>
-      <value>hb</value>
+      <key>_coordinate</key>
+      <value>(10, 10)</value>
     </param>
     <param>
-      <key>category</key>
-      <value>gr-mac</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>run_options</key>
-      <value>prompt</value>
+      <key>generate_options</key>
+      <value>hb</value>
     </param>
     <param>
-      <key>run</key>
-      <value>True</value>
+      <key>hier_block_src_path</key>
+      <value>.:</value>
+    </param>
+    <param>
+      <key>id</key>
+      <value>gmsk_radio</value>
     </param>
     <param>
       <key>max_nouts</key>
       <value>0</value>
     </param>
     <param>
-      <key>realtime_scheduling</key>
+      <key>qt_qss_theme</key>
       <value></value>
     </param>
     <param>
-      <key>alias</key>
+      <key>realtime_scheduling</key>
       <value></value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(10, 10)</value>
+      <key>run_command</key>
+      <value>{python} -u {filename}</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>run_options</key>
+      <value>prompt</value>
     </param>
-  </block>
-  <block>
-    <key>variable</key>
     <param>
-      <key>id</key>
-      <value>samp_rate</value>
+      <key>run</key>
+      <value>True</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>thread_safe_setters</key>
+      <value></value>
     </param>
     <param>
-      <key>value</key>
-      <value>rate</value>
+      <key>title</key>
+      <value>GMSK Modem</value>
     </param>
+  </block>
+  <block>
+    <key>variable</key>
     <param>
-      <key>alias</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
       <key>_coordinate</key>
       <value>(120, 123)</value>
     </param>
@@ -91,632 +99,685 @@
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>parameter</key>
     <param>
       <key>id</key>
-      <value>rate</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
-      <key>label</key>
-      <value>Sample rate</value>
+      <value>samp_rate</value>
     </param>
     <param>
       <key>value</key>
-      <value>1e6</value>
+      <value>rate</value>
     </param>
+  </block>
+  <block>
+    <key>parameter</key>
     <param>
-      <key>type</key>
-      <value>eng_float</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>short_id</key>
-      <value>r</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(8, 123)</value>
+      <value>(8, 387)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>parameter</key>
     <param>
       <key>id</key>
-      <value>samps_per_sym</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <value>access_code_threshold</value>
     </param>
     <param>
       <key>label</key>
-      <value>Samples/symbol</value>
+      <value>Access code threshold</value>
     </param>
     <param>
-      <key>value</key>
-      <value>4</value>
+      <key>short_id</key>
+      <value></value>
     </param>
     <param>
       <key>type</key>
       <value>intx</value>
     </param>
     <param>
-      <key>short_id</key>
-      <value></value>
+      <key>value</key>
+      <value>0</value>
     </param>
+  </block>
+  <block>
+    <key>parameter</key>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
+      <key>comment</key>
+      <value></value>
+    </param>
+    <param>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
       <key>_coordinate</key>
-      <value>(8, 251)</value>
+      <value>(168, 251)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>parameter</key>
     <param>
       <key>id</key>
       <value>ampl</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
       <key>label</key>
       <value>TX BB amp</value>
     </param>
     <param>
-      <key>value</key>
-      <value>0.7</value>
+      <key>short_id</key>
+      <value></value>
     </param>
     <param>
       <key>type</key>
       <value>eng_float</value>
     </param>
     <param>
-      <key>short_id</key>
-      <value></value>
+      <key>value</key>
+      <value>0.7</value>
     </param>
+  </block>
+  <block>
+    <key>blocks_multiply_const_vxx</key>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(168, 251)</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>const</key>
+      <value>ampl</value>
     </param>
-  </block>
-  <block>
-    <key>parameter</key>
     <param>
-      <key>id</key>
-      <value>access_code_threshold</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>label</key>
-      <value>Access code threshold</value>
+      <key>_coordinate</key>
+      <value>(592, 635)</value>
     </param>
     <param>
-      <key>value</key>
+      <key>_rotation</key>
       <value>0</value>
     </param>
     <param>
-      <key>type</key>
-      <value>intx</value>
+      <key>id</key>
+      <value>blocks_multiply_const_vxx_0</value>
     </param>
     <param>
-      <key>short_id</key>
-      <value></value>
+      <key>type</key>
+      <value>complex</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(8, 387)</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
   </block>
   <block>
-    <key>note</key>
+    <key>blocks_pdu_to_tagged_stream</key>
     <param>
-      <key>id</key>
-      <value>note_0</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>note</key>
-      <value>Tradeoff between detecting possibly slightly corrupted header and still decoding packet vs. starting decode and have state machine decoding while another packet starts later (and it is missed)</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(184, 435)</value>
+      <value>(944, 491)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_sink</key>
     <param>
       <key>id</key>
-      <value>pad_sink_0_0</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
-      <key>label</key>
-      <value>msg_out</value>
+      <value>blocks_pdu_to_tagged_stream_0</value>
     </param>
     <param>
       <key>type</key>
-      <value>message</value>
+      <value>byte</value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>tag</key>
+      <value>length</value>
     </param>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>optional</key>
-      <value>False</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
+  </block>
+  <block>
+    <key>digital_gmsk_demod</key>
     <param>
-      <key>_coordinate</key>
-      <value>(1272, 235)</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>comment</key>
+      <value></value>
     </param>
-  </block>
-  <block>
-    <key>mac_packet_to_pdu</key>
     <param>
-      <key>id</key>
-      <value>mac_packet_to_pdu_0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>dewhiten</key>
-      <value>True</value>
+      <key>freq_error</key>
+      <value>0.0</value>
     </param>
     <param>
-      <key>output_invalid</key>
-      <value>False</value>
-    </param>
+      <key>_coordinate</key>
+      <value>(632, 195)</value>
+    </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>gain_mu</key>
+      <value>0.175</value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>id</key>
+      <value>digital_gmsk_demod_0</value>
+    </param>
+    <param>
+      <key>log</key>
+      <value>False</value>
     </param>
     <param>
       <key>maxoutbuf</key>
       <value>0</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(1088, 227)</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>mu</key>
+      <value>0.5</value>
     </param>
-  </block>
-  <block>
-    <key>mac_packet_deframer</key>
     <param>
-      <key>id</key>
-      <value>mac_packet_deframer_0</value>
+      <key>omega_relative_limit</key>
+      <value>0.005</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>samples_per_symbol</key>
+      <value>samps_per_sym</value>
     </param>
     <param>
-      <key>access_code</key>
-      <value></value>
+      <key>verbose</key>
+      <value>True</value>
     </param>
+  </block>
+  <block>
+    <key>digital_gmsk_mod</key>
     <param>
-      <key>threshold</key>
-      <value>access_code_threshold</value>
+      <key>bt</key>
+      <value>0.35</value>
     </param>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(888, 227)</value>
+      <value>(1184, 483)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>blocks_pdu_to_tagged_stream</key>
     <param>
       <key>id</key>
-      <value>blocks_pdu_to_tagged_stream_0</value>
+      <value>digital_gmsk_mod_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>log</key>
+      <value>False</value>
     </param>
     <param>
-      <key>type</key>
-      <value>byte</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>tag</key>
-      <value>length</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
+      <key>samples_per_symbol</key>
+      <value>samps_per_sym</value>
+    </param>
+    <param>
+      <key>verbose</key>
+      <value>False</value>
+    </param>
+  </block>
+  <block>
+    <key>mac_burst_tagger</key>
+    <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(944, 491)</value>
+      <value>(768, 611)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>digital_gmsk_mod</key>
     <param>
       <key>id</key>
-      <value>digital_gmsk_mod_0</value>
+      <value>mac_burst_tagger_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>samples_per_symbol</key>
-      <value>samps_per_sym</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>bt</key>
-      <value>0.35</value>
+      <key>mult</key>
+      <value>samps_per_sym*8</value>
     </param>
     <param>
-      <key>verbose</key>
-      <value>False</value>
+      <key>pad_front</key>
+      <value>32*0+ 0</value>
     </param>
     <param>
-      <key>log</key>
-      <value>False</value>
+      <key>pad_rear</key>
+      <value>16*0 + 16</value>
+    </param>
+    <param>
+      <key>tag_name</key>
+      <value>'length'</value>
+    </param>
+  </block>
+  <block>
+    <key>mac_packet_deframer</key>
+    <param>
+      <key>access_code</key>
+      <value></value>
     </param>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(1184, 483)</value>
+      <value>(888, 227)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>blocks_multiply_const_vxx</key>
     <param>
       <key>id</key>
-      <value>blocks_multiply_const_vxx_0</value>
+      <value>mac_packet_deframer_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>type</key>
-      <value>complex</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>const</key>
-      <value>ampl</value>
+      <key>threshold</key>
+      <value>access_code_threshold</value>
     </param>
+  </block>
+  <block>
+    <key>mac_packet_framer</key>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>access_code</key>
+      <value></value>
     </param>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(592, 635)</value>
+      <value>(680, 451)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_sink</key>
     <param>
       <key>id</key>
-      <value>pad_sink_0</value>
+      <value>mac_packet_framer_0</value>
     </param>
     <param>
-      <key>_enabled</key>
+      <key>maxoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>minoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>postamble</key>
+      <value>''.join(['\x00']*(16/8/samps_per_sym)*0)</value>
+    </param>
+    <param>
+      <key>preamble</key>
+      <value>''.join(['\x55']*((256*1)/8/samps_per_sym))</value>
+    </param>
+    <param>
+      <key>rotate_whitener_offset</key>
+      <value>False</value>
+    </param>
+    <param>
+      <key>whiten</key>
       <value>True</value>
     </param>
     <param>
-      <key>label</key>
-      <value>out</value>
+      <key>whitener_offset</key>
+      <value>0</value>
     </param>
+  </block>
+  <block>
+    <key>mac_packet_to_pdu</key>
     <param>
-      <key>type</key>
-      <value>complex</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>optional</key>
-      <value>False</value>
+      <key>dewhiten</key>
+      <value>True</value>
+    </param>
+    <param>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(968, 635)</value>
+      <value>(1088, 227)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
+    <param>
+      <key>id</key>
+      <value>mac_packet_to_pdu_0</value>
+    </param>
+    <param>
+      <key>maxoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>minoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>output_invalid</key>
+      <value>False</value>
+    </param>
   </block>
   <block>
-    <key>digital_gmsk_demod</key>
+    <key>note</key>
     <param>
-      <key>id</key>
-      <value>digital_gmsk_demod_0</value>
+      <key>alias</key>
+      <value></value>
+    </param>
+    <param>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>samples_per_symbol</key>
-      <value>samps_per_sym</value>
+      <key>_coordinate</key>
+      <value>(184, 435)</value>
     </param>
     <param>
-      <key>gain_mu</key>
-      <value>0.175</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>mu</key>
-      <value>0.5</value>
+      <key>id</key>
+      <value>note_0</value>
     </param>
     <param>
-      <key>omega_relative_limit</key>
-      <value>0.005</value>
+      <key>note</key>
+      <value>Tradeoff between detecting possibly slightly corrupted header and still decoding packet vs. starting decode and have state machine decoding while another packet starts later (and it is missed)</value>
     </param>
+  </block>
+  <block>
+    <key>pad_sink</key>
     <param>
-      <key>freq_error</key>
-      <value>0.0</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>verbose</key>
+      <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>log</key>
-      <value>False</value>
+      <key>_coordinate</key>
+      <value>(968, 635)</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>id</key>
+      <value>pad_sink_0</value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>type</key>
+      <value>complex</value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>label</key>
+      <value>out</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(632, 195)</value>
+      <key>num_streams</key>
+      <value>1</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>optional</key>
+      <value>False</value>
+    </param>
+    <param>
+      <key>vlen</key>
+      <value>1</value>
     </param>
   </block>
   <block>
-    <key>mac_packet_framer</key>
+    <key>pad_sink</key>
     <param>
-      <key>id</key>
-      <value>mac_packet_framer_0</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>access_code</key>
-      <value></value>
+      <key>_coordinate</key>
+      <value>(1272, 235)</value>
     </param>
     <param>
-      <key>whitener_offset</key>
+      <key>_rotation</key>
       <value>0</value>
     </param>
     <param>
-      <key>whiten</key>
-      <value>True</value>
+      <key>id</key>
+      <value>pad_sink_0_0</value>
     </param>
     <param>
-      <key>rotate_whitener_offset</key>
-      <value>False</value>
+      <key>type</key>
+      <value>message</value>
     </param>
     <param>
-      <key>preamble</key>
-      <value>''.join(['\x55']*((256*1)/8/samps_per_sym))</value>
+      <key>label</key>
+      <value>msg_out</value>
     </param>
     <param>
-      <key>postamble</key>
-      <value>''.join(['\x00']*(16/8/samps_per_sym)*0)</value>
+      <key>num_streams</key>
+      <value>1</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>optional</key>
+      <value>False</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
+  </block>
+  <block>
+    <key>pad_source</key>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(680, 451)</value>
+      <value>(456, 235)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_source</key>
     <param>
       <key>id</key>
       <value>pad_source_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
       <key>label</key>
       <value>bb_in</value>
     </param>
     <param>
+      <key>num_streams</key>
+      <value>1</value>
+    </param>
+    <param>
+      <key>optional</key>
+      <value>False</value>
+    </param>
+    <param>
       <key>type</key>
       <value>complex</value>
     </param>
@@ -724,38 +785,42 @@
       <key>vlen</key>
       <value>1</value>
     </param>
+  </block>
+  <block>
+    <key>pad_source</key>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>optional</key>
-      <value>False</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(456, 235)</value>
+      <value>(504, 491)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_source</key>
     <param>
       <key>id</key>
       <value>pad_source_1</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
       <key>label</key>
       <value>msg_in</value>
     </param>
     <param>
+      <key>num_streams</key>
+      <value>1</value>
+    </param>
+    <param>
+      <key>optional</key>
+      <value>False</value>
+    </param>
+    <param>
       <key>type</key>
       <value>message</value>
     </param>
@@ -763,99 +828,106 @@
       <key>vlen</key>
       <value>1</value>
     </param>
+  </block>
+  <block>
+    <key>parameter</key>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>optional</key>
-      <value>False</value>
+      <key>comment</key>
+      <value></value>
+    </param>
+    <param>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(504, 491)</value>
+      <value>(8, 123)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>mac_burst_tagger</key>
     <param>
       <key>id</key>
-      <value>mac_burst_tagger_0</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <value>rate</value>
     </param>
     <param>
-      <key>tag_name</key>
-      <value>'length'</value>
+      <key>label</key>
+      <value>Sample rate</value>
     </param>
     <param>
-      <key>mult</key>
-      <value>samps_per_sym*8</value>
+      <key>short_id</key>
+      <value>r</value>
     </param>
     <param>
-      <key>pad_front</key>
-      <value>32*0+ 0</value>
+      <key>type</key>
+      <value>eng_float</value>
     </param>
     <param>
-      <key>pad_rear</key>
-      <value>16*0 + 16</value>
+      <key>value</key>
+      <value>1e6</value>
     </param>
+  </block>
+  <block>
+    <key>parameter</key>
     <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
-    </param>
-    <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(768, 611)</value>
+      <value>(8, 251)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
+    <param>
+      <key>id</key>
+      <value>samps_per_sym</value>
+    </param>
+    <param>
+      <key>label</key>
+      <value>Samples/symbol</value>
+    </param>
+    <param>
+      <key>short_id</key>
+      <value></value>
+    </param>
+    <param>
+      <key>type</key>
+      <value>intx</value>
+    </param>
+    <param>
+      <key>value</key>
+      <value>4</value>
+    </param>
   </block>
   <connection>
-    <source_block_id>blocks_pdu_to_tagged_stream_0</source_block_id>
-    <sink_block_id>digital_gmsk_mod_0</sink_block_id>
+    <source_block_id>blocks_multiply_const_vxx_0</source_block_id>
+    <sink_block_id>mac_burst_tagger_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>mac_packet_framer_0</source_block_id>
-    <sink_block_id>blocks_pdu_to_tagged_stream_0</sink_block_id>
-    <source_key>out</source_key>
-    <sink_key>pdus</sink_key>
-  </connection>
-  <connection>
-    <source_block_id>mac_packet_to_pdu_0</source_block_id>
-    <sink_block_id>pad_sink_0_0</sink_block_id>
-    <source_key>pdu</source_key>
+    <source_block_id>blocks_pdu_to_tagged_stream_0</source_block_id>
+    <sink_block_id>digital_gmsk_mod_0</sink_block_id>
+    <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>mac_packet_deframer_0</source_block_id>
-    <sink_block_id>mac_packet_to_pdu_0</sink_block_id>
-    <source_key>msg</source_key>
-    <sink_key>msg</sink_key>
-  </connection>
-  <connection>
     <source_block_id>digital_gmsk_demod_0</source_block_id>
     <sink_block_id>mac_packet_deframer_0</sink_block_id>
     <source_key>0</source_key>
@@ -868,18 +940,30 @@
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>blocks_multiply_const_vxx_0</source_block_id>
-    <sink_block_id>mac_burst_tagger_0</sink_block_id>
-    <source_key>0</source_key>
-    <sink_key>0</sink_key>
-  </connection>
-  <connection>
     <source_block_id>mac_burst_tagger_0</source_block_id>
     <sink_block_id>pad_sink_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
   <connection>
+    <source_block_id>mac_packet_deframer_0</source_block_id>
+    <sink_block_id>mac_packet_to_pdu_0</sink_block_id>
+    <source_key>msg</source_key>
+    <sink_key>msg</sink_key>
+  </connection>
+  <connection>
+    <source_block_id>mac_packet_framer_0</source_block_id>
+    <sink_block_id>blocks_pdu_to_tagged_stream_0</sink_block_id>
+    <source_key>out</source_key>
+    <sink_key>pdus</sink_key>
+  </connection>
+  <connection>
+    <source_block_id>mac_packet_to_pdu_0</source_block_id>
+    <sink_block_id>pad_sink_0_0</sink_block_id>
+    <source_key>pdu</source_key>
+    <sink_key>in</sink_key>
+  </connection>
+  <connection>
     <source_block_id>pad_source_0</source_block_id>
     <sink_block_id>digital_gmsk_demod_0</sink_block_id>
     <source_key>0</source_key>
@@ -888,7 +972,7 @@
   <connection>
     <source_block_id>pad_source_1</source_block_id>
     <sink_block_id>mac_packet_framer_0</sink_block_id>
-    <source_key>0</source_key>
+    <source_key>out</source_key>
     <sink_key>in</sink_key>
   </connection>
 </flow_graph>
--- examples/ofdm_radio.grc.orig
+++ examples/ofdm_radio.grc
@@ -1,23 +1,23 @@
-<?xml version='1.0' encoding='ASCII'?>
-<?grc format='1' created='3.7.6'?>
+<?xml version='1.0' encoding='utf-8'?>
+<?grc format='1' created='3.7.11'?>
 <flow_graph>
   <timestamp>Fri Sep 12 16:57:41 2014</timestamp>
   <block>
     <key>options</key>
     <param>
-      <key>id</key>
-      <value>ofdm_radio</value>
+      <key>author</key>
+      <value></value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>window_size</key>
+      <value>1280, 1024</value>
     </param>
     <param>
-      <key>title</key>
-      <value>OFDM Modem</value>
+      <key>category</key>
+      <value>gr-mac</value>
     </param>
     <param>
-      <key>author</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
@@ -25,257 +25,243 @@
       <value></value>
     </param>
     <param>
-      <key>window_size</key>
-      <value>1280, 1024</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
-      <key>generate_options</key>
-      <value>hb</value>
+      <key>_coordinate</key>
+      <value>(10, 10)</value>
     </param>
     <param>
-      <key>category</key>
-      <value>gr-mac</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>run_options</key>
-      <value>prompt</value>
+      <key>generate_options</key>
+      <value>hb</value>
     </param>
     <param>
-      <key>run</key>
-      <value>True</value>
+      <key>hier_block_src_path</key>
+      <value>.:</value>
+    </param>
+    <param>
+      <key>id</key>
+      <value>ofdm_radio</value>
     </param>
     <param>
       <key>max_nouts</key>
       <value>0</value>
     </param>
     <param>
-      <key>realtime_scheduling</key>
+      <key>qt_qss_theme</key>
       <value></value>
     </param>
     <param>
-      <key>alias</key>
+      <key>realtime_scheduling</key>
       <value></value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(10, 10)</value>
+      <key>run_command</key>
+      <value>{python} -u {filename}</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>run_options</key>
+      <value>prompt</value>
     </param>
-  </block>
-  <block>
-    <key>variable</key>
     <param>
-      <key>id</key>
-      <value>samp_rate</value>
+      <key>run</key>
+      <value>True</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>thread_safe_setters</key>
+      <value></value>
     </param>
     <param>
-      <key>value</key>
-      <value>rate</value>
+      <key>title</key>
+      <value>OFDM Modem</value>
     </param>
+  </block>
+  <block>
+    <key>variable</key>
     <param>
-      <key>alias</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
       <key>_coordinate</key>
-      <value>(123, 98)</value>
+      <value>(844, 456)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>variable</key>
     <param>
       <key>id</key>
       <value>backoff_db</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
       <key>value</key>
       <value>-15.0</value>
     </param>
+  </block>
+  <block>
+    <key>variable</key>
     <param>
-      <key>alias</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
       <key>_coordinate</key>
-      <value>(844, 456)</value>
+      <value>(123, 98)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>parameter</key>
     <param>
       <key>id</key>
-      <value>rate</value>
+      <value>samp_rate</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>value</key>
+      <value>rate</value>
     </param>
+  </block>
+  <block>
+    <key>blocks_multiply_const_vxx</key>
     <param>
-      <key>label</key>
-      <value>Sample rate</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>value</key>
-      <value>1e6</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>type</key>
-      <value>eng_float</value>
+      <key>const</key>
+      <value>10.0**(1.0*backoff_db/10.0)</value>
     </param>
     <param>
-      <key>short_id</key>
-      <value>r</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(10, 98)</value>
+      <value>(810, 414)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>blocks_tagged_stream_to_pdu</key>
     <param>
       <key>id</key>
-      <value>blocks_tagged_stream_to_pdu_0</value>
+      <value>blocks_multiply_const_vxx_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>type</key>
+      <value>complex</value>
     </param>
     <param>
-      <key>type</key>
-      <value>byte</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>tag</key>
-      <value>length</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
+      <key>vlen</key>
+      <value>1</value>
+    </param>
+  </block>
+  <block>
+    <key>blocks_pdu_to_tagged_stream</key>
+    <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(826, 44)</value>
+      <value>(457, 237)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_sink</key>
     <param>
       <key>id</key>
-      <value>pad_sink_3</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
-    </param>
-    <param>
-      <key>label</key>
-      <value>msg_out</value>
+      <value>blocks_pdu_to_tagged_stream_0</value>
     </param>
     <param>
       <key>type</key>
-      <value>message</value>
-    </param>
-    <param>
-      <key>vlen</key>
-      <value>1</value>
-    </param>
-    <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <value>byte</value>
     </param>
     <param>
-      <key>optional</key>
-      <value>True</value>
+      <key>tag</key>
+      <value>length</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(1074, 44)</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_rotation</key>
+      <key>minoutbuf</key>
       <value>0</value>
     </param>
   </block>
   <block>
-    <key>note</key>
+    <key>blocks_tagged_stream_to_pdu</key>
     <param>
-      <key>id</key>
-      <value>note_0</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>note</key>
-      <value>Front: 128 is too short, Tail: 8 is adequate</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(1080, 172)</value>
+      <value>(826, 44)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>blocks_pdu_to_tagged_stream</key>
     <param>
       <key>id</key>
-      <value>blocks_pdu_to_tagged_stream_0</value>
-    </param>
-    <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <value>blocks_tagged_stream_to_pdu_0</value>
     </param>
     <param>
       <key>type</key>
@@ -286,267 +272,335 @@
       <value>length</value>
     </param>
     <param>
+      <key>maxoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>minoutbuf</key>
+      <value>0</value>
+    </param>
+  </block>
+  <block>
+    <key>digital_ofdm_rx</key>
+    <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
+      <key>comment</key>
+      <value></value>
+    </param>
+    <param>
       <key>affinity</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>cp_len</key>
+      <value>16</value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
+      <key>fft_len</key>
+      <value>64</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(457, 237)</value>
+      <value>(560, 9)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>pad_source</key>
+    <param>
+      <key>header_mod</key>
+      <value>"BPSK"</value>
+    </param>
     <param>
       <key>id</key>
-      <value>pad_source_0</value>
+      <value>digital_ofdm_rx_1</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>log</key>
+      <value>False</value>
     </param>
     <param>
-      <key>label</key>
-      <value>msg_in</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>type</key>
-      <value>message</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>occupied_carriers</key>
+      <value>()</value>
     </param>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>packet_len_key</key>
+      <value>"length"</value>
     </param>
     <param>
-      <key>optional</key>
-      <value>True</value>
+      <key>payload_mod</key>
+      <value>"BPSK"</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(274, 244)</value>
+      <key>pilot_carriers</key>
+      <value>()</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>pilot_symbols</key>
+      <value>()</value>
+    </param>
+    <param>
+      <key>scramble_bits</key>
+      <value>False</value>
+    </param>
+    <param>
+      <key>sync_word1</key>
+      <value>()</value>
+    </param>
+    <param>
+      <key>sync_word2</key>
+      <value>()</value>
     </param>
   </block>
   <block>
-    <key>foo_burst_tagger</key>
+    <key>digital_ofdm_tx</key>
     <param>
-      <key>id</key>
-      <value>foo_burst_tagger_0</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>False</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>alias</key>
+      <key>affinity</key>
       <value></value>
     </param>
     <param>
+      <key>cp_len</key>
+      <value>16</value>
+    </param>
+    <param>
+      <key>_enabled</key>
+      <value>True</value>
+    </param>
+    <param>
+      <key>fft_len</key>
+      <value>64</value>
+    </param>
+    <param>
       <key>_coordinate</key>
-      <value>(936, 237)</value>
+      <value>(696, 202)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
     <param>
-      <key>tag_name</key>
-      <value>pmt.intern("length")</value>
+      <key>header_mod</key>
+      <value>"BPSK"</value>
     </param>
     <param>
-      <key>mult</key>
-      <value>1</value>
+      <key>id</key>
+      <value>digital_ofdm_tx_0</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>packet_len_key</key>
+      <value>length</value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>log</key>
+      <value>False</value>
     </param>
     <param>
       <key>maxoutbuf</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>foo_packet_pad</key>
     <param>
-      <key>id</key>
-      <value>foo_packet_pad_0</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>False</value>
+      <key>occupied_carriers</key>
+      <value>()</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>payload_mod</key>
+      <value>"BPSK"</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(1141, 216)</value>
+      <key>pilot_carriers</key>
+      <value>()</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>pilot_symbols</key>
+      <value>()</value>
     </param>
     <param>
-      <key>debug</key>
-      <value>False</value>
+      <key>rolloff</key>
+      <value>0 + 4</value>
     </param>
     <param>
-      <key>delay</key>
+      <key>scramble_bits</key>
       <value>False</value>
     </param>
     <param>
-      <key>delay_sec</key>
-      <value>0.001</value>
+      <key>sync_word1</key>
+      <value>()</value>
     </param>
     <param>
-      <key>pad_front</key>
-      <value>16*4 * 4</value>
+      <key>sync_word2</key>
+      <value>()</value>
     </param>
+  </block>
+  <block>
+    <key>foo_burst_tagger</key>
     <param>
-      <key>pad_tail</key>
-      <value>16</value>
+      <key>alias</key>
+      <value></value>
+    </param>
+    <param>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>affinity</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
+      <key>_enabled</key>
+      <value>False</value>
+    </param>
+    <param>
+      <key>_coordinate</key>
+      <value>(936, 237)</value>
+    </param>
+    <param>
+      <key>_rotation</key>
       <value>0</value>
     </param>
     <param>
+      <key>id</key>
+      <value>foo_burst_tagger_0</value>
+    </param>
+    <param>
       <key>maxoutbuf</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>digital_ofdm_tx</key>
     <param>
-      <key>id</key>
-      <value>digital_ofdm_tx_0</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>mult</key>
+      <value>1</value>
     </param>
     <param>
-      <key>fft_len</key>
-      <value>64</value>
+      <key>tag_name</key>
+      <value>pmt.intern("length")</value>
     </param>
+  </block>
+  <block>
+    <key>foo_packet_pad</key>
     <param>
-      <key>cp_len</key>
-      <value>16</value>
+      <key>alias</key>
+      <value></value>
     </param>
     <param>
-      <key>packet_len_key</key>
-      <value>length</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>occupied_carriers</key>
-      <value>()</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>pilot_carriers</key>
-      <value>()</value>
+      <key>debug</key>
+      <value>False</value>
     </param>
     <param>
-      <key>pilot_symbols</key>
-      <value>()</value>
+      <key>delay_sec</key>
+      <value>0.001</value>
     </param>
     <param>
-      <key>sync_word1</key>
-      <value>()</value>
+      <key>delay</key>
+      <value>False</value>
     </param>
     <param>
-      <key>sync_word2</key>
-      <value>()</value>
+      <key>_enabled</key>
+      <value>False</value>
     </param>
     <param>
-      <key>header_mod</key>
-      <value>"BPSK"</value>
+      <key>_coordinate</key>
+      <value>(1141, 216)</value>
     </param>
     <param>
-      <key>payload_mod</key>
-      <value>"BPSK"</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>rolloff</key>
-      <value>0 + 4</value>
+      <key>id</key>
+      <value>foo_packet_pad_0</value>
     </param>
     <param>
-      <key>scramble_bits</key>
-      <value>False</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>log</key>
-      <value>False</value>
+      <key>minoutbuf</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>pad_front</key>
+      <value>16*4 * 4</value>
     </param>
     <param>
+      <key>pad_tail</key>
+      <value>16</value>
+    </param>
+  </block>
+  <block>
+    <key>mac_burst_tagger</key>
+    <param>
       <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>affinity</key>
+      <key>comment</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>affinity</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(696, 202)</value>
+      <value>(588, 393)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
-  </block>
-  <block>
-    <key>mac_burst_tagger</key>
     <param>
       <key>id</key>
       <value>mac_burst_tagger_0</value>
     </param>
     <param>
-      <key>_enabled</key>
-      <value>True</value>
+      <key>maxoutbuf</key>
+      <value>0</value>
     </param>
     <param>
-      <key>tag_name</key>
-      <value>'length'</value>
+      <key>minoutbuf</key>
+      <value>0</value>
     </param>
     <param>
       <key>mult</key>
@@ -561,51 +615,70 @@
       <value>16</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>tag_name</key>
+      <value>'length'</value>
     </param>
+  </block>
+  <block>
+    <key>note</key>
     <param>
-      <key>affinity</key>
+      <key>alias</key>
       <value></value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
       <key>_coordinate</key>
-      <value>(588, 393)</value>
+      <value>(1080, 172)</value>
     </param>
     <param>
       <key>_rotation</key>
       <value>0</value>
     </param>
+    <param>
+      <key>id</key>
+      <value>note_0</value>
+    </param>
+    <param>
+      <key>note</key>
+      <value>Front: 128 is too short, Tail: 8 is adequate</value>
+    </param>
   </block>
   <block>
-    <key>pad_source</key>
+    <key>pad_sink</key>
     <param>
-      <key>id</key>
-      <value>pad_source_1</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>label</key>
-      <value>in</value>
+      <key>_coordinate</key>
+      <value>(1008, 427)</value>
+    </param>
+    <param>
+      <key>_rotation</key>
+      <value>0</value>
+    </param>
+    <param>
+      <key>id</key>
+      <value>pad_sink_0</value>
     </param>
     <param>
       <key>type</key>
       <value>complex</value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>label</key>
+      <value>out</value>
     </param>
     <param>
       <key>num_streams</key>
@@ -616,186 +689,191 @@
       <value>False</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(408, 67)</value>
-    </param>
-    <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
   </block>
   <block>
-    <key>digital_ofdm_rx</key>
+    <key>pad_sink</key>
     <param>
-      <key>id</key>
-      <value>digital_ofdm_rx_1</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>fft_len</key>
-      <value>64</value>
+      <key>_coordinate</key>
+      <value>(1074, 44)</value>
     </param>
     <param>
-      <key>cp_len</key>
-      <value>16</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>packet_len_key</key>
-      <value>"length"</value>
+      <key>id</key>
+      <value>pad_sink_3</value>
     </param>
     <param>
-      <key>occupied_carriers</key>
-      <value>()</value>
+      <key>type</key>
+      <value>message</value>
     </param>
     <param>
-      <key>pilot_carriers</key>
-      <value>()</value>
+      <key>label</key>
+      <value>msg_out</value>
     </param>
     <param>
-      <key>pilot_symbols</key>
-      <value>()</value>
+      <key>num_streams</key>
+      <value>1</value>
     </param>
     <param>
-      <key>sync_word1</key>
-      <value>()</value>
+      <key>optional</key>
+      <value>True</value>
     </param>
     <param>
-      <key>sync_word2</key>
-      <value>()</value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
+  </block>
+  <block>
+    <key>pad_source</key>
     <param>
-      <key>header_mod</key>
-      <value>"BPSK"</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
-      <key>payload_mod</key>
-      <value>"BPSK"</value>
+      <key>_enabled</key>
+      <value>True</value>
     </param>
     <param>
-      <key>scramble_bits</key>
-      <value>False</value>
+      <key>_coordinate</key>
+      <value>(274, 244)</value>
     </param>
     <param>
-      <key>log</key>
-      <value>False</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>id</key>
+      <value>pad_source_0</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>label</key>
+      <value>msg_in</value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>num_streams</key>
+      <value>1</value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>optional</key>
+      <value>True</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(560, 9)</value>
+      <key>type</key>
+      <value>message</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
   </block>
   <block>
-    <key>blocks_multiply_const_vxx</key>
+    <key>pad_source</key>
     <param>
-      <key>id</key>
-      <value>blocks_multiply_const_vxx_0</value>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>type</key>
-      <value>complex</value>
-    </param>
-    <param>
-      <key>const</key>
-      <value>10.0**(1.0*backoff_db/10.0)</value>
+      <key>_coordinate</key>
+      <value>(408, 67)</value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>alias</key>
-      <value></value>
+      <key>id</key>
+      <value>pad_source_1</value>
     </param>
     <param>
-      <key>affinity</key>
-      <value></value>
+      <key>label</key>
+      <value>in</value>
     </param>
     <param>
-      <key>minoutbuf</key>
-      <value>0</value>
+      <key>num_streams</key>
+      <value>1</value>
     </param>
     <param>
-      <key>maxoutbuf</key>
-      <value>0</value>
+      <key>optional</key>
+      <value>False</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(810, 414)</value>
+      <key>type</key>
+      <value>complex</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>vlen</key>
+      <value>1</value>
     </param>
   </block>
   <block>
-    <key>pad_sink</key>
+    <key>parameter</key>
     <param>
-      <key>id</key>
-      <value>pad_sink_0</value>
+      <key>alias</key>
+      <value></value>
+    </param>
+    <param>
+      <key>comment</key>
+      <value></value>
     </param>
     <param>
       <key>_enabled</key>
       <value>True</value>
     </param>
     <param>
-      <key>label</key>
-      <value>out</value>
+      <key>_coordinate</key>
+      <value>(10, 98)</value>
     </param>
     <param>
-      <key>type</key>
-      <value>complex</value>
+      <key>_rotation</key>
+      <value>0</value>
     </param>
     <param>
-      <key>vlen</key>
-      <value>1</value>
+      <key>id</key>
+      <value>rate</value>
     </param>
     <param>
-      <key>num_streams</key>
-      <value>1</value>
+      <key>label</key>
+      <value>Sample rate</value>
     </param>
     <param>
-      <key>optional</key>
-      <value>False</value>
+      <key>short_id</key>
+      <value>r</value>
     </param>
     <param>
-      <key>_coordinate</key>
-      <value>(1008, 427)</value>
+      <key>type</key>
+      <value>eng_float</value>
     </param>
     <param>
-      <key>_rotation</key>
-      <value>0</value>
+      <key>value</key>
+      <value>1e6</value>
     </param>
   </block>
   <connection>
-    <source_block_id>digital_ofdm_rx_1</source_block_id>
-    <sink_block_id>blocks_tagged_stream_to_pdu_0</sink_block_id>
+    <source_block_id>blocks_multiply_const_vxx_0</source_block_id>
+    <sink_block_id>pad_sink_0</sink_block_id>
+    <source_key>0</source_key>
+    <sink_key>0</sink_key>
+  </connection>
+  <connection>
+    <source_block_id>blocks_pdu_to_tagged_stream_0</source_block_id>
+    <sink_block_id>digital_ofdm_tx_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
@@ -803,23 +881,23 @@
     <source_block_id>blocks_tagged_stream_to_pdu_0</source_block_id>
     <sink_block_id>pad_sink_3</sink_block_id>
     <source_key>pdus</source_key>
-    <sink_key>0</sink_key>
+    <sink_key>in</sink_key>
   </connection>
   <connection>
-    <source_block_id>blocks_pdu_to_tagged_stream_0</source_block_id>
-    <sink_block_id>digital_ofdm_tx_0</sink_block_id>
+    <source_block_id>digital_ofdm_rx_1</source_block_id>
+    <sink_block_id>blocks_tagged_stream_to_pdu_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>pad_source_0</source_block_id>
-    <sink_block_id>blocks_pdu_to_tagged_stream_0</sink_block_id>
+    <source_block_id>digital_ofdm_tx_0</source_block_id>
+    <sink_block_id>foo_burst_tagger_0</sink_block_id>
     <source_key>0</source_key>
-    <sink_key>pdus</sink_key>
+    <sink_key>0</sink_key>
   </connection>
   <connection>
     <source_block_id>digital_ofdm_tx_0</source_block_id>
-    <sink_block_id>foo_burst_tagger_0</sink_block_id>
+    <sink_block_id>mac_burst_tagger_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
@@ -836,26 +914,20 @@
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>digital_ofdm_tx_0</source_block_id>
-    <sink_block_id>mac_burst_tagger_0</sink_block_id>
-    <source_key>0</source_key>
-    <sink_key>0</sink_key>
-  </connection>
-  <connection>
     <source_block_id>mac_burst_tagger_0</source_block_id>
     <sink_block_id>blocks_multiply_const_vxx_0</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
   <connection>
-    <source_block_id>pad_source_1</source_block_id>
-    <sink_block_id>digital_ofdm_rx_1</sink_block_id>
-    <source_key>0</source_key>
-    <sink_key>0</sink_key>
+    <source_block_id>pad_source_0</source_block_id>
+    <sink_block_id>blocks_pdu_to_tagged_stream_0</sink_block_id>
+    <source_key>out</source_key>
+    <sink_key>pdus</sink_key>
   </connection>
   <connection>
-    <source_block_id>blocks_multiply_const_vxx_0</source_block_id>
-    <sink_block_id>pad_sink_0</sink_block_id>
+    <source_block_id>pad_source_1</source_block_id>
+    <sink_block_id>digital_ofdm_rx_1</sink_block_id>
     <source_key>0</source_key>
     <sink_key>0</sink_key>
   </connection>
