--- src/SimpleRandomizer.cc.orig	2023-11-15 03:46:08
+++ src/SimpleRandomizer.cc	2023-11-15 15:57:52
@@ -41,9 +41,9 @@
 #include <cstring>
 #include <iostream>
 
-#ifdef __APPLE__
+/*#ifdef __APPLE__
 #  include <Security/SecRandom.h>
-#endif // __APPLE__
+#endif // __APPLE__*/
 
 #ifdef HAVE_LIBGNUTLS
 #  include <gnutls/crypto.h>
@@ -106,9 +106,9 @@
     assert(r);
     abort();
   }
-#elif defined(__APPLE__)
+/*#elif defined(__APPLE__)
   auto rv = SecRandomCopyBytes(kSecRandomDefault, len, buf);
-  assert(errSecSuccess == rv);
+  assert(errSecSuccess == rv); */
 #elif defined(HAVE_LIBGNUTLS)
   auto rv = gnutls_rnd(GNUTLS_RND_RANDOM, buf, len);
   if (rv != 0) {
