--- host/libhackrf/CMakeLists.txt.orig	2013-06-08 14:49:37.000000000 -0400
+++ host/libhackrf/CMakeLists.txt	2013-06-08 14:49:50.000000000 -0400
@@ -34,7 +34,6 @@
 	set(THREADS_USE_PTHREADS_WIN32 true)
 else()
 	add_definitions(-Wall)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
 
 	INCLUDE(TestBigEndian)
 	TEST_BIG_ENDIAN(BIGENDIAN)
--- host/hackrf-tools/CMakeLists.txt.orig	2013-06-08 14:49:42.000000000 -0400
+++ host/hackrf-tools/CMakeLists.txt	2013-06-08 14:49:54.000000000 -0400
@@ -34,7 +34,6 @@
 include_directories(getopt)
 else()
 add_definitions(-Wall)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")
 endif()
 
 if(NOT libhackrf_SOURCE_DIR)
