--- doc/info/build_html.sh.in.orig	2020-01-17 20:50:17.000000000 +0100
+++ doc/info/build_html.sh.in	2020-01-17 20:50:29.000000000 +0100
@@ -34,12 +34,12 @@
 
 for f in @abs_srcdir@/*.texi
 do
-  @SED@ -e 's/^@\(deffn\|defvr\)  *{[^}]*}  *\([^[:blank:]]*\).*/@anchor{Item: \2}\n&/;' -e 's/^@node  *\([^,]*\).*/@anchor{Item: \1}\n&/' $f > $(basename $f)
+  @SED@ -e 's/^@\(deffn\|defvr\)  *{[^}]*}  *\([^[:blank:]]*\).*/@anchor{Item: \2}\'$'\n&/;' -e 's/^@node  *\([^,]*\).*/@anchor{Item: \1}\'$'\n&/' $f > $(basename $f)
 done 
 
 cat *.texi | @AWK@ '!/^@c / && !/^@c$/ && (/^@deffn/ || /^@defvr/ || /^@end deffn/ || /^@end defvr/ || /@category/ || /@node/)' | @SED@ -f @abs_srcdir@/extract_categories1.sed | @AWK@ -f @abs_srcdir@/extract_categories1.awk > make-categories.py 
 python make-categories.py 
-@SED@ -e 's/^@bye/@node Documentation Categories, , Function and Variable Index\n@chapter Documentation Categories/' @abs_srcdir@/maxima.texi > maxima.texi 
+@SED@ -e 's/^@bye/@node Documentation Categories, , Function and Variable Index\'$'\n@chapter Documentation Categories/' @abs_srcdir@/maxima.texi > maxima.texi 
 ( for f in Category-*.texi; do echo '@include' $f; done ; echo @bye ) >> maxima.texi 
 
 if [ "X$DEBUG" = "Xyes" ]; then

