From 2bdaafc2c80fbed7ffef54f1a85f4d5eea5f881a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 4 Jan 2023 07:52:40 +0700
Subject: [PATCH 07/10] src/Makefile: there is no tuning dir

---
 src/Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 3503b91..81cc0cc 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,24 +1,24 @@
 CFLAGS += $(INC) $(DEFINE)
 CPPFLAGS += $(INC) $(DEFINE)
 
-SUBDIRS = core kernel util tuning
+SUBDIRS = core kernel util
 
-ifeq ($(HIP_ON),1)
-SUBDIRS += util_cpp
-SUBDIRS += tuning
-endif
 ifeq ($(PLAIN_ON),1)
 SUBDIRS += plain
 endif
+ifeq ($(HIP_ON),1)
+SUBDIRS += util_cpp
+# SUBDIRS += tuning
+endif
 ifeq ($(ARM_ON),1)
-SUBDIRS += tuning
+# SUBDIRS += tuning
 endif
 
 .PHONY : lib
 
-lib : 
+lib :
 	@for d in $(SUBDIRS) ; \
 	do if test -d $$d; then \
 	  $(MAKE) -C $$d $(@F) || exit 1 ; \
 	fi; \
-	done
\ No newline at end of file
+	done
-- 
2.39.0

