--- pyproject.toml.orig	2026-02-05 11:11:12
+++ pyproject.toml	2026-08-29 12:10:12
@@ -45,13 +45,13 @@
 Changelog = "https://pip.pypa.io/en/stable/news/"
 
 [build-system]
-requires = ["flit-core >=3.11,<4"]
+requires = ["flit-core >=3.11"]
 build-backend = "flit_core.buildapi"
 
 [dependency-groups]
 test = [
     "cryptography",
-    "flit-core >= 3.11, < 4",
+    "flit-core >= 3.11",
     "freezegun",
     "installer",
     # pytest-subket requires 7.0+
@@ -72,7 +72,7 @@
 ]
 
 test-common-wheels = [
-    "flit-core >= 3.11, < 4",
+    "flit-core >= 3.11",
     # We pin setuptools<80 because our test suite currently
     # depends on setup.py develop to generate egg-link files.
     "setuptools >= 70.1.0, <80",
