--- src/geoclue.js.orig1	2014-07-24 20:40:44.000000000 +0200
+++ src/geoclue.js	2014-07-24 20:40:52.000000000 +0200
@@ -79,6 +79,18 @@
 
 const Geoclue = new Lang.Class({
     Name: 'Geoclue',
+    Extends: GObject.Object,
+    Properties: {
+        'connected': GObject.ParamSpec.boolean('connected',
+                                               'Connected',
+                                               'Connected to DBus service',
+                                               GObject.ParamFlags.READABLE,
+                                               false)
+    },
+
+    get connected() {
+        return this._connected;
+    },
 
     overrideLocation: function(location) {
         if (this._clientProxy && this._locationUpdatedId > 0) {
@@ -110,6 +122,9 @@
     },
 
     _init: function() {
+        this.parent();
+        this._connected = false;
+
         let lastLocation = Application.settings.get('last-location');
         if (lastLocation.length >= 3) {
             let [lat, lng, accuracy] = lastLocation;
@@ -149,6 +164,9 @@
 
         if (!this.userSetLocation)
             this.findLocation();
+
+        this._connected = true;
+        this.notify('connected');
     },
 
     _onLocationUpdated: function(proxy, sender, [oldPath, newPath]) {
