--- newmail.c	Thu Oct 14 06:49:42 1999
+++ newmail.c.new	Tue May 18 03:38:08 2004
@@ -148,19 +148,19 @@
 
 static void usage() {
 
-	fprintf(stderr,"usage: newmail [-deirv] directory|file directory|file ...
-
-where:
-	d: Prints messages when you do not have new mail
-	e: Exits with 0 on no new mail, 1 on new mail, > 1 on error
-		(normally exits with a value from sysexits(3)). May not be used
-		with -d or -i.
-	i: Prints a message for each mailbox that has new mail (and those
-		that don't if -d is also used)
-	r: Warns if it finds a non regular file
-	v: Print version number and exit
-	
-directory|file are, respectively, directories to look in for mailboxes
+	fprintf(stderr,"usage: newmail [-deirv] directory|file directory|file ...\
+\
+where:\
+	d: Prints messages when you do not have new mail\
+	e: Exits with 0 on no new mail, 1 on new mail, > 1 on error\
+		(normally exits with a value from sysexits(3)). May not be used\
+		with -d or -i.\
+	i: Prints a message for each mailbox that has new mail (and those\
+		that don't if -d is also used)\
+	r: Warns if it finds a non regular file\
+	v: Print version number and exit\
+	\
+directory|file are, respectively, directories to look in for mailboxes\
 and mailboxes.\n");
 
 	exit(EX_USAGE);
