From 01079f677de1048f9447dd32a73e664a65ce4c51 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 6 Mar 2023 10:49:26 +0800
Subject: [PATCH 3/3] Remove valgrind from tests

---
 test/unittest/CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/test/unittest/CMakeLists.txt b/test/unittest/CMakeLists.txt
index 565ed982..bfd90afa 100644
--- test/unittest/CMakeLists.txt
+++ test/unittest/CMakeLists.txt
@@ -81,11 +81,6 @@ add_test(NAME unittest
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)
 
 if(NOT MSVC)
-    # Not running SIMD.* unit test cases for Valgrind
-    add_test(NAME valgrind_unittest
-        COMMAND valgrind --suppressions=${CMAKE_SOURCE_DIR}/test/valgrind.supp --leak-check=full --error-exitcode=1 ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/unittest --gtest_filter=-SIMD.*
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)
-
     if(CMAKE_BUILD_TYPE STREQUAL "Debug")
         add_test(NAME symbol_check
         COMMAND sh -c "objdump -t -C libnamespacetest.a | grep rapidjson ; test $? -ne 0"
