--- DESCRIPTION	2015-10-24 14:21:16.000000000 +0800
+++ DESCRIPTION	2024-04-19 15:30:19.000000000 +0800
@@ -14,7 +14,7 @@
 Depends: R (>= 2.10)
 Imports: proxy, RJSONIO, parallel
 License: MIT + file LICENSE
-Suggests: mallet, lda, topicmodels, gistr (>= 0.0.8.99), servr, shiny,
+Suggests: lda, topicmodels, gistr (>= 0.0.8.99), servr, shiny,
         knitr, rmarkdown, digest, htmltools
 LazyData: true
 VignetteBuilder: knitr
