--- CMakeLists.txt.orig	2009-06-24 15:30:01.000000000 -0500
+++ CMakeLists.txt	2011-09-14 03:26:48.000000000 -0500
@@ -20,11 +20,6 @@
    # with cmake < 2.6 install(TARGETS ... BUNDLE DESTINATION ... ) doesn't work:
    cmake_minimum_required(VERSION 2.6 FATAL_ERROR )
    set(exeType MACOSX_BUNDLE)
-   set(binInstallDir /Applications )
-   # hint to linker for indirect referenced libs inside qt3support
-   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -dylib_file QtSql.framework/Versions/4/QtSql:/Library/Frameworks/QtSql.framework/Versions/4/QtSql")
-   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -dylib_file QtNetwork.framework/Versions/4/QtNetwork:/Library/Frameworks/QtNetwork.framework/Versions/4/QtNetwork")
-   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -dylib_file QtXml.framework/Versions/4/QtXml:/Library/Frameworks/QtXml.framework/Versions/4/QtXml")
 endif(APPLE)
 
 add_executable(cutecom ${exeType} ${cutecomSrcs} ${uiHeaders})
