--- example/solr/conf/velocity/head.vm.orig	2012-07-17 18:39:39.000000000 +0900
+++ example/solr/conf/velocity/head.vm	2012-08-02 13:50:10.000000000 +0900
@@ -20,7 +20,7 @@
 <title>#param('title')</title>
     <meta http-equiv="content-type" content="text/html; charset=UTF-8"/>
 
-<script type="text/javascript" src="#{url_for_solr}/admin/jquery-1.4.3.min.js"></script>
+<script type="text/javascript" src="#{url_for_solr}/admin/file?file=/velocity/jquery-1.7.2.min.js&contentType=text/javascript"></script>
   <link rel="stylesheet" type="text/css" href="#{url_for_solr}/admin/file?file=/velocity/main.css&contentType=text/css"/>
   <link rel="stylesheet" href="#{url_for_solr}/admin/file?file=/velocity/jquery.autocomplete.css&contentType=text/css" type="text/css" />
   <script type="text/javascript" src="#{url_for_solr}/admin/file?file=/velocity/jquery.autocomplete.js&contentType=text/javascript"></script>
