--- configure.ac.orig	2014-08-29 07:49:29.000000000 -0700
+++ configure.ac	2014-08-29 07:50:15.000000000 -0700
@@ -44,6 +44,15 @@
 AM_PROG_LIBTOOL
 AC_PATH_XTRA
 AM_PATH_PYTHON(2.4)
+
+dnl Override PYTHON_PREFIX, PYTHON_EXEC_PREFIX and pythondir
+
+AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print sys.prefix;'`])
+AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print sys.exec_prefix;'`])
+
+am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+AC_SUBST([pythondir], [$am_cv_python_pythondir])
+
 AM_CHECK_PYTHON_HEADERS(HAVE_PYHDRS="yes", HAVE_PYHDRS="no")
 
 X_LIBS="$X_LIBS $X_PRE_LIBS -lX11 $X_EXTRA_LIBS"
@@ -540,7 +549,8 @@
 dnl *** invest-applet specific checks                                       ***
 dnl ***************************************************************************
 
-AM_CONDITIONAL(BUILD_INVEST_APPLET, test "x$HAVE_PYGOBJECT" = "xyes")
+build_invest_applet=$HAVE_PYGOBJECT
+AM_CONDITIONAL(BUILD_INVEST_APPLET, test "x$build_invest_applet" = "xyes")
 
 dnl ***************************************************************************
 dnl *** Check if IPv6 is available                                          ***
@@ -723,7 +733,7 @@
 	 - drivemount			always
 	 - geyes			always
 	 - gweather			$build_libgweather_applets
-	 - invest-applet		$BUILD_INVEST_APPLET
+	 - invest-applet		$build_invest_applet
 	 - mini-commander		$enable_mini_commander
 	 - modemlights			$BUILD_MODEM_LIGHTS
 	 - mixer			$enable_mixer
