--- configure.ac.orig	2014-10-28 16:37:39.000000000 -0700
+++ configure.ac	2014-10-29 15:52:28.000000000 -0700
@@ -42,6 +42,17 @@
 AC_PATH_XTRA
 AM_PATH_PYTHON(3)
 
+dnl Override standard python locations with MacPorts version specific ones
+
+AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print(sys.prefix);'`])
+AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print(sys.exec_prefix)'`])
+
+am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+am_cv_python_pyexecdir=$PYTHON_EXEC_PREFIX/lib/python$PYTHON_VERSION/site-packages
+
+AC_SUBST([pythondir], [$am_cv_python_pythondir])
+AC_SUBST([pyexecdir], [$am_cv_python_pyexecdir])
+
 X_LIBS="$X_LIBS $X_PRE_LIBS -lX11 $X_EXTRA_LIBS"
 AC_SUBST(X_LIBS)
