--- config/Makefile.darwin.orig	2012-07-18 13:46:32.000000000 +1000
+++ config/Makefile.darwin	2012-07-19 12:09:47.000000000 +1000
@@ -1,10 +1,10 @@
 NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
-CFLAGS.EXTRA = -dynamic -fno-common
+CFLAGS.EXTRA = -dynamic -fno-common $(CFLAGS.ARCH)
 #CFLAGS.EXTRA += -no-cpp-precomp
 PICFLAG = -fPIC
-LDFLAGS.EXTRA =
+LDFLAGS.EXTRA = $(LDFLAGS.ARCH)
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
 LDFLAGS.GL = -L/usr/X11R6/lib -lXmu -lXi -lGL -lXext -lX11
