--- config/Makefile.darwin.orig	2017-07-31 21:25:39.000000000 +1000
+++ config/Makefile.darwin	2017-08-05 23:39:51.000000000 +1000
@@ -1,11 +1,12 @@
-GLEW_DEST = /usr/local
+GLEW_DEST ?= /usr/local
 NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
 CFLAGS.EXTRA += -dynamic -fno-common
 CFLAGS.EXTRA += -pedantic
 CFLAGS.EXTRA += -fPIC
-LDFLAGS.EXTRA =
+CFLAGS.EXTRA += $(CFLAGS.ARCH)
+LDFLAGS.EXTRA = $(LDFLAGS.ARCH)
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
 CFLAGS.EXTRA += -std=c99
 CFLAGS.EXTRA += -I/usr/X11R6/include -D'GLEW_APPLE_GLX'
