--- DESCRIPTION.orig	2023-04-26 07:00:59.000000000 +0800
+++ DESCRIPTION	2023-04-30 10:06:15.000000000 +0800
@@ -7,7 +7,7 @@
 Author: Adi Laurentiu Tarca <atarca@med.wayne.edu>, Purvesh Kathri
         <purvesh@cs.wayne.edu> and Sorin Draghici <sorin@wayne.edu>
 Depends: R (>= 2.14.0), graphics, KEGGgraph
-Suggests: graph, Rgraphviz, hgu133plus2.db
+Suggests: graph, Rgraphviz
 Maintainer: Adi Laurentiu Tarca <atarca@med.wayne.edu>
 Description: This package implements the Signaling Pathway Impact
         Analysis (SPIA) which uses the information form a list of
