Use MacPorts zlib instead of the bundled copy.
--- CMakeLists.txt.orig	2021-11-10 17:05:19.000000000 -0600
+++ CMakeLists.txt	2025-10-13 09:10:59.000000000 -0500
@@ -23,16 +23,7 @@
 	stb/stb_image_write.c
 	)
 
-add_library(zlib STATIC
-	zlib/adler32.c
-	zlib/crc32.c
-	zlib/deflate.c
-	zlib/inffast.c
-	zlib/inflate.c
-	zlib/inftrees.c
-	zlib/trees.c
-	zlib/zutil.c
-	)
+find_package(ZLIB REQUIRED)
 
 add_library(MacRomanConversion STATIC
 	MacRomanConversion/MacRomanConversion.cpp
@@ -138,7 +129,7 @@
 
 target_compile_options(PortabilityLayer PRIVATE -Wno-multichar)
 
-target_link_libraries(PortabilityLayer zlib MacRomanConversion stb)
+target_link_libraries(PortabilityLayer ZLIB::ZLIB MacRomanConversion stb)
 
 
 add_library(GpShell STATIC
