--- C/CONFIGS/macosx-x86-64.orig	2012-01-16 23:14:47.000000000 +0000
+++ C/CONFIGS/macosx-x86-64	2013-03-02 17:48:14.000000000 +0000
@@ -161,18 +161,18 @@
 if [ -n "$CC" ]; then
 	cc=$CC
 else
-	cc='gcc'
+	cc='cc'
 fi
 if [ -n "$CFLAGS" ]; then
 	ccflags=$CFLAGS
 else
-	ccflags='-arch x86_64 -pipe -no-cpp-precomp -fno-common -fPIC'
+	ccflags='-pipe -fno-common -fPIC'
 fi
-ccldflags='-arch x86_64'
+ccldflags=''
 if [ -n "$CPP" ]; then
 	cpp=$CPP
 else
-	cpp='g++'
+	cpp='c++'
 fi
 if [ -n "$CPPFLAGS" ]; then
 	cppflags=$CPPFLAGS
@@ -211,7 +211,7 @@
 else
 	ld='ld'
 fi
-ldflags='-arch x86_64'
+ldflags=''
 ldsharedflags="$ldflags -dynamiclib -flat_namespace -undefined suppress -o"
 libs='-lm'
 ln='/bin/ln'
@@ -235,7 +235,7 @@
 sharedlibs='-lm'
 mtsharedlibs='-lm -lpthread'
 sharedlibversion='.7.1'
-sharedlink='gcc'
+sharedlink=$cc
 suffix='.a'
 wkeiflib='wkbench'
 
