--- doc/Makefile.in.orig	2010-05-02 06:33:06.000000000 -0500
+++ doc/Makefile.in	2010-09-01 02:56:03.000000000 -0500
@@ -493,7 +493,7 @@
 install-data-hook:
 	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)
 	if [ ! -f $(DESTDIR)$(sysconfdir)/ngircd.conf ]; then \
-	  $(INSTALL) -m 600 -c $(srcdir)/sample-ngircd.conf $(DESTDIR)$(sysconfdir)/ngircd.conf; \
+	  $(INSTALL) -m 644 -c $(srcdir)/sample-ngircd.conf $(DESTDIR)$(sysconfdir)/ngircd.conf.sample; \
 	 fi
 	$(mkinstalldirs) $(DESTDIR)$(docdir)
 	for f in $(documents); do \
