--- doc/Makefile.in.orig	2014-03-25 08:48:41.000000000 -0500
+++ doc/Makefile.in	2014-07-18 19:33:07.000000000 -0500
@@ -586,10 +586,6 @@
 all: $(generated_docs)
 
 install-data-hook: $(static_docs) $(toplevel_docs) $(generated_docs)
-	$(MKDIR_P) -m 755 $(DESTDIR)$(sysconfdir)
-	@if [ ! -f $(DESTDIR)$(sysconfdir)/ngircd.conf ]; then \
-	  make install-config; \
-	 fi
 	$(MKDIR_P) -m 755 $(DESTDIR)$(docdir)
 	for f in $(static_docs) $(toplevel_docs); do \
 	  $(INSTALL) -m 644 -c $(srcdir)/$$f $(DESTDIR)$(docdir)/; \
