--- autogen.sh.orig	2015-08-19 14:53:23.000000000 -0700
+++ autogen.sh	2015-08-19 14:53:44.000000000 -0700
@@ -6,7 +6,7 @@
 intltoolize --force --copy
 
 echo "Running libtoolize"
-libtoolize --force --copy
+glibtoolize --force --copy
 
 echo "Removing autom4te.cache"
 rm -fr autom4te.cache
@@ -35,7 +35,7 @@
 ##export ACLOCAL=aclocal-1.4
 ##export AUTOMAKE=automake-1.4
 
-autoreconf -v -f
+autoreconf -v -f -i
 
 echo "======================================"
 echo "Now you are ready to run './configure'"
