--- setup.py.orig	2009-07-11 19:34:42.000000000 +0200
+++ setup.py	2009-07-11 19:34:56.000000000 +0200
@@ -79,8 +79,8 @@
     keywords=['python.templating.engines'],
     packages=['jinja', 'jinja.translators'],
     data_files=[
-        ('docs/html', list(list_files('docs/html'))),
-        ('docs/txt', list(list_files('docs/src')))
+        ('share/doc/@@NAME@@/html', list(list_files('docs/html'))),
+        ('share/doc/@@NAME@@/txt', list(list_files('docs/src')))
     ],
     entry_points='''
     [python.templating.engines]
