--- platforms/darwin/org.apcupsd.apcupsd.plist.in.orig	2022-05-09 11:44:10
+++ platforms/darwin/org.apcupsd.apcupsd.plist.in	2023-08-14 15:51:50
@@ -8,12 +8,13 @@
 	<string>root</string>
 	<key>GroupName</key>
 	<string>wheel</string>
+	<key>StandardErrorPath</key>
+	<string>/var/log/apcupsd.events</string>
 	<key>Program</key>
 	<string>@sbindir@/apcupsd-start</string>
 	<key>ProgramArguments</key>
 	<array>
-		<string>/bin/sh</string>
-		<string>-f/etc/apcupsd/apcupsd.conf</string>
+		<string>-f@sysconfdir@/apcupsd.conf</string>
 		<string>-b</string>
 	</array>
 	<key>KeepAlive</key>
