--- options.mk.orig	2013-12-08 16:40:11.000000000 -0800
+++ options.mk	2013-12-08 16:46:43.000000000 -0800
@@ -13,10 +13,10 @@
 # in $(INSTALL_PREFIX_DIR)/bin, libraries in $(INSTALL_PREFIX_DIR)/lib
 # etc,.
 #
-INSTALL_PREFIX_DIR=/usr/local
+INSTALL_PREFIX_DIR=@@MP_INSTALL_PREFIX@@
 
 # Set RELEASE to 1 to compile with optimizations and without debug info.
-RELEASE=0
+RELEASE=1
 
 # Set DEBUG to 1 if you want diagnostic messages turned on
 DEBUG=0
@@ -25,8 +25,8 @@
 # Specify an alternate C++ and C compiler. The default compiler is 
 # platform dependent
 #
-# CXX = 
-# CC = 
+CXX = @@MP_CXX@@
+CC = @@MP_CC@@
 
 # Set BENCHMARK or PROFILE to 1 for framerate diagnostics
 BENCHMARK=0
@@ -38,14 +38,14 @@
 # linker normally checks. The linker will search the directories
 # in the order specified.
 #
-LIB_SEARCH_DIRS = 
+LIB_SEARCH_DIRS = @@MP_PREFIX@@/lib 
 
 # Set INC_SEARCH_DIRS to a list of directories containing include files
 # not on the default search path for the compiler. Typically 3rd party 
 # dependencies (e.g. netCDF, IDL) are not installed in a location where 
 # compiler normally checks. The compiler will search the directories 
 # in the order specified.  
-INC_SEARCH_DIRS =
+INC_SEARCH_DIRS = @@MP_PREFIX@@/include @@MP_PREFIX@@/include/udunits2 @@MP_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 @@MP_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/numarray/include @@MP_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/core/include
 
 # Set NETCDF_LIBS to the name of your netCDF library and any library
 # dependencies that netcdf has. The default is simply 'netcdf'. Starting
@@ -66,7 +66,7 @@
 # http://www.trolltech.com, and when possible, from the vapor
 # web site: www.vapor.ucar.edu. Qt version 4.6.1 or higher is required.
 #
-QTDIR =
+QTDIR = @@MP_PREFIX@@
 
 # If BUILD_GUI is set to 1 **and** this is a Mac system, set HAVE_QT_FRAMEWORK
 # to 1 if your Qt libraries are built as Mac Frameworks, or to 0 if 
@@ -92,7 +92,7 @@
 # Python directory where the sub directories 'lib', and 'include' 
 # may be found. Python version 2.6.5 or higher is required. 
 #
-PYTHONDIR =
+PYTHONDIR = @@MP_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.7
 
 # If BUILD_PYTHON is set to 1, set PYTHONVERSION to the version 
 # number (both major and minor, e.g. "2.6") of Python.
