--- dispenso/CMakeLists.txt	2023-12-28 08:49:50.000000000 +0800
+++ dispenso/CMakeLists.txt	2024-01-12 15:15:20.000000000 +0800
@@ -22,7 +22,7 @@
 
 target_compile_options(dispenso PRIVATE
   $<$<CXX_COMPILER_ID:MSVC>:/W3 /WX>
-  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -pedantic -Wconversion -Wno-sign-conversion -Werror>
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -pedantic -Wconversion -Wno-sign-conversion>
   )
 
 target_include_directories(dispenso

--- tests/CMakeLists.txt	2023-12-28 08:49:50.000000000 +0800
+++ tests/CMakeLists.txt	2024-01-12 15:15:55.000000000 +0800
@@ -21,7 +21,7 @@
     target_compile_features(${TEST_NAME} PRIVATE cxx_std_14)
     target_compile_options(${TEST_NAME} PRIVATE
       $<$<CXX_COMPILER_ID:MSVC>:/W3 /WX>
-      $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -pedantic -Wconversion -Wno-sign-conversion -Werror>
+      $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall -Wextra -pedantic -Wconversion -Wno-sign-conversion>
       )
     target_link_libraries(${TEST_NAME} gmock_main dispenso)
     gtest_discover_tests(${TEST_NAME}
