--- CMakeLists.txt.orig	2011-05-04 14:40:19.000000000 +0200
+++ CMakeLists.txt	2011-05-04 14:41:33.000000000 +0200
@@ -955,9 +955,9 @@
   # set these so that the files get installed nicely in the MacOSX
   # .app bundle (FIXME: this screws up a standard "make install" on
   # MacOS)
-  set(GMSH_BIN ../MacOS)
-  set(GMSH_DOC ../../..)
-  set(GMSH_MAN ../../..)
+  set(GMSH_BIN bin)
+  set(GMSH_DOC share/doc/gmsh)
+  set(GMSH_MAN share/man/man1)
 else(WIN32)
   set(GMSH_BIN bin)
   set(GMSH_DOC share/doc/gmsh)
@@ -1057,11 +1057,11 @@
   file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/Info.plist "${F1}")
   set(CPACK_BUNDLE_PLIST ${CMAKE_CURRENT_BINARY_DIR}/Info.plist)
   set(CPACK_BUNDLE_ICON ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIcons.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsGeo.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsGeo.icns DESTINATION share/gmsh
           RENAME GmshGeo.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsMsh.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsMsh.icns DESTINATION share/gmsh
           RENAME GmshMsh.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsPos.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsPos.icns DESTINATION share/gmsh
           RENAME GmshPos.icns)
   set(CPACK_PACKAGE_ICON ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIcons.icns)
 elseif(WIN32)
