--- src/log.c.orig	2015-04-11 15:08:14.000000000 -0500
+++ src/log.c	2015-04-16 19:41:58.000000000 -0500
@@ -266,15 +266,7 @@
         enter_critical_section(CRIT_ID);
         my_seq=seq++;
         leave_critical_section(CRIT_ID);
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat"
-#pragma GCC diagnostic ignored "-Wformat-extra-args"
-#endif /* __GNUC__ */
         return str_printf("%llu", my_seq);
-#ifdef __GNUC__
-#pragma GCC diagnostic pop
-#endif /* __GNUC__ */
     case LOG_ID_UNIQUE:
         if(RAND_bytes(rnd, sizeof rnd)<=0) /* log2(62^22)=130.99 */
             return str_dup("error");
@@ -302,10 +294,6 @@
 
 /* critical problem handling */
 /* str.c functions are not safe to use here */
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-result"
-#endif /* __GNUC__ */
 void fatal_debug(char *txt, const char *file, int line) {
     char msg[80];
 #ifdef USE_WIN32
