--- configure.ac.orig	2009-11-23 11:51:18.000000000 -0600
+++ configure.ac	2011-06-29 05:24:40.000000000 -0500
@@ -22,15 +22,6 @@
 
 
 
-# Bring additional directories where things might be found into our
-# search path. I don't know why autoconf doesn't do this by default
-for spfx in /usr/local /opt/local /sw ; do
-    echo checking ${spfx}/include
-    if test -d ${spfx}/include; then
-        CPPFLAGS="-I${spfx}/include $CPPFLAGS"
-        LDFLAGS="-L${spfx}/lib $LDFLAGS"
-    fi
-done
         
 
 # Specific headers that I plan to use
