--- F77.pm.orig	2019-01-04 21:56:24.000000000 -0800
+++ F77.pm	2019-01-24 22:37:46.000000000 -0800
@@ -368,6 +368,8 @@
 ### Darwin (Mac OS X) ###
 
 $F77config{Darwin}{GNU} = $F77config{Generic}{GNU};
+$F77config{Darwin}{GNU}{Link} = '-L@@GCCLIB@@ -lgfortran';
+$F77config{Darwin}{GNU}{Compiler} = '@@F77@@';
 $F77config{Darwin}{DEFAULT}     = 'GNU';
 
 ############ End of database is here ############
