--- request.c.orig	2004-11-28 01:26:46.000000000 +1100
+++ request.c	2013-09-13 10:22:02.000000000 +1000
@@ -209,7 +209,7 @@ char *rfctime(time_t t, char *buf)
 	return buf;
 }
 
-static char *getline(struct pool *p, int fold)
+static char *private_getline(struct pool *p, int fold)
 {
 	char *s, *olds, *sp, *end;
 	int f;
@@ -1090,7 +1090,7 @@ static int process_headers(struct reques
 	unsigned long cl;
 
 	do {
-		l = getline(&r->cn->header_input, 0);
+		l = private_getline(&r->cn->header_input, 0);
 		if (l == 0)
 			return -1;
 	} while (*l == 0);
@@ -1119,7 +1119,7 @@ static int process_headers(struct reques
 	if (r->protocol_major && r->protocol_minor)
 		r->cn->keepalive = 1;
 	n = 0;
-	while ((l = getline(&r->cn->header_input, 1)) != 0) {
+	while ((l = private_getline(&r->cn->header_input, 1)) != 0) {
 		s = strchr(l, ':');
 		if (s == 0)
 			continue;
