--- test_scripts.py.orig
+++ test_scripts.py
@@ -84,7 +84,7 @@
         major = str(version[0])
         minor = str(version[1])
 
-        f2py_cmds = ('f2py', 'f2py' + major, 'f2py' + major + '.' + minor)
+        f2py_cmds = ('f2py', 'f2py' + major, 'f2py' + major + '.' + minor, 'f2py-' + major + '.' + minor)
         success = False
 
         for f2py_cmd in f2py_cmds:
@@ -95,5 +95,5 @@
                 break
             except:
                 pass
-        msg = "Warning: neither %s nor %s nor %s found in path" % f2py_cmds
+        msg = "Warning: none of %s, %s, %s, nor %s found in path" % f2py_cmds
         assert_(success, msg)
