From b9364db1b1be024cc834240b59d9b065a67c66f1 Mon Sep 17 00:00:00 2001
From: Mark Mentovai <mark@mentovai.com>
Date: Fri, 20 Oct 2023 12:05:52 -0400
Subject: [PATCH] Allow the path to cython to be configured in the environment
 as CYTHON

---
 mesonbuild/envconfig.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git vendored-meson/meson/mesonbuild/envconfig.py vendored-meson/meson/mesonbuild/envconfig.py
index 7e0c56703121..3545ccb6fb31 100644
--- vendored-meson/meson/mesonbuild/envconfig.py
+++ vendored-meson/meson/mesonbuild/envconfig.py
@@ -138,6 +138,8 @@ ENV_VAR_TOOL_MAP: T.Mapping[str, str] = {
     'make': 'MAKE',
     'vapigen': 'VAPIGEN',
     'llvm-config': 'LLVM_CONFIG',
+
+    'cython': 'CYTHON',
 }
 
 ENV_VAR_PROG_MAP = {**ENV_VAR_COMPILER_MAP, **ENV_VAR_TOOL_MAP}
-- 
2.42.0

