--- numpy/linalg/setup.py.orig
+++ numpy/linalg/setup.py
@@ -64,6 +64,7 @@
         sources=['lapack_litemodule.c', get_lapack_lite_sources],
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
+        extra_link_args=['-Wl,-undefined,dynamic_lookup', '-bundle']
     )
 
     # umath_linalg module
@@ -73,6 +74,7 @@
         depends=['lapack_lite/f2c.h'],
         extra_info=lapack_info,
         libraries=['npymath'],
+        extra_link_args=['-Wl,-undefined,dynamic_lookup', '-bundle']
     )
     config.add_data_files('*.pyi')
     return config
