--- numpy/distutils/fcompiler/g95.py	2007-11-07 17:05:14.000000000 -0500
+++ numpy/distutils/fcompiler/g95.py	2008-01-30 21:26:47.000000000 -0500
@@ -21,7 +21,7 @@
         'compiler_f77' : ["g95", "-ffixed-form"],
         'compiler_fix' : ["g95", "-ffixed-form"],
         'compiler_f90' : ["g95"],
-        'linker_so'    : ["<F90>","-shared"],
+        'linker_so'    : ["<F90>","-dynamiclib -L@@MPORTS_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.4/lib -lpython2.4"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"]
         }
