--- numpy/core/setup.py.orig
+++ numpy/core/setup.py
@@ -980,7 +980,8 @@
                                     join('blasdot', 'cblas.h'),
                                   ],
                          include_dirs = ['blasdot'],
-                         extra_info = blas_info
+                         extra_info = blas_info,
+                         extra_link_args=['-undefined dynamic_lookup -bundle']
                          )
 
     #######################################################################
