--- numpy/_build_utils/src/apple_sgemv_fix.c.orig
+++ numpy/_build_utils/src/apple_sgemv_fix.c
@@ -120,7 +120,7 @@
      * AVX and the OS version is Mavericks */
     AVX_and_10_9 = AVX && MAVERICKS;
     /* load vecLib */
-    veclib = dlopen(VECLIB_FILE, RTLD_LOCAL | RTLD_FIRST);
+    veclib = dlopen(VECLIB_FILE, 0x4 | 0x100);
     if (!veclib) {
         veclib = NULL;
         snprintf(errormsg, sizeof(errormsg),
