From d2109a772ed650ea76f31bb285952f8efee75ff1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 17 Feb 2025 10:52:21 +0800
Subject: [PATCH] display: fix

---
 src/osx/core/display.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/osx/core/display.cpp b/src/osx/core/display.cpp
index 6b8004da4c..8878ceb51b 100644
--- a/src/osx/core/display.cpp
+++ b/src/osx/core/display.cpp
@@ -31,6 +31,8 @@
 
 #include "wx/osx/private.h"
 
+#ifdef __WXMAC__
+
 // ----------------------------------------------------------------------------
 // common helpers compiled even in wxUSE_DISPLAY==0 case
 // ----------------------------------------------------------------------------
@@ -441,4 +443,13 @@ protected:
     return new wxDisplayFactorySingleMacOSX;
 }
 
+#else
+
+/* static */ wxDisplayFactory *wxDisplay::CreateFactory()
+{
+    return new wxDisplayFactorySingle;
+}
+
+#endif
+
 #endif // wxUSE_DISPLAY
