--- a/src/common/powercmn.cpp	2025-05-26 00:15:56
+++ b/src/common/powercmn.cpp	2025-11-24 21:03:44
@@ -39,7 +39,7 @@
 #endif
 
 // Provide stubs for systems without power resource management functions
-#if !defined(__WINDOWS__) && !defined(__APPLE__)
+#if !defined(__WINDOWS__)
 
 bool
 wxPowerResource::Acquire(wxPowerResourceKind WXUNUSED(kind),
@@ -52,7 +52,7 @@
 {
 }
 
-#endif // !(__WINDOWS__ || __APPLE__)
+#endif // !(__WINDOWS__)
 
 // provide stubs for the systems not implementing these functions
 #if !defined(__WINDOWS__)
