--- src/Makefile.in.sav	2008-09-22 10:47:26.000000000 -0400
+++ src/Makefile.in	2008-09-22 10:48:04.000000000 -0400
@@ -100,17 +100,17 @@
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/lib
 	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/bin
-	@INSTALL@ -m 755 -d ${DEVDESTDIR}@prefix@/include/modglue
+	@INSTALL@ -m 755 -d ${DESTDIR}@prefix@/include/modglue
 	${IPHASE} libmodglue.la ${DESTDIR}@prefix@/lib
 	${IPHASE} libmid.la ${DESTDIR}@prefix@/lib
 	${FPHASE}
 	${IPHASE} ptywrap ${DESTDIR}@prefix@/bin
 	${IPHASE} prompt ${DESTDIR}@prefix@/bin
 	${IPHASE} isatty ${DESTDIR}@prefix@/bin
-	@INSTALL@ -m 0644 ../include/modglue/ext_process.hh ${DEVDESTDIR}@prefix@/include/modglue
-	@INSTALL@ -m 0644 ../include/modglue/process.hh ${DEVDESTDIR}@prefix@/include/modglue
-	@INSTALL@ -m 0644 ../include/modglue/pipe.hh ${DEVDESTDIR}@prefix@/include/modglue
-	@INSTALL@ -m 0644 ../include/modglue/main.hh ${DEVDESTDIR}@prefix@/include/modglue
+	@INSTALL@ -m 0644 ../include/modglue/ext_process.hh ${DESTDIR}@prefix@/include/modglue
+	@INSTALL@ -m 0644 ../include/modglue/process.hh ${DESTDIR}@prefix@/include/modglue
+	@INSTALL@ -m 0644 ../include/modglue/pipe.hh ${DESTDIR}@prefix@/include/modglue
+	@INSTALL@ -m 0644 ../include/modglue/main.hh ${DESTDIR}@prefix@/include/modglue
 
 clean:
 	rm -f *.o *.lo *~ libmodglue* libmid* 
