diff -Naurp xorg-cf-files-1.0.2.orig/Imake.cf xorg-cf-files-1.0.2/Imake.cf
--- xorg-cf-files-1.0.2.orig/Imake.cf	2005-11-07 22:33:24.000000000 -0800
+++ xorg-cf-files-1.0.2/Imake.cf	2009-06-08 17:06:14.000000000 -0700
@@ -33,6 +33,10 @@ XCOMM $XFree86: xc/config/cf/Imake.cf,v 
 #  define i386DarwinArchitecture
 #  undef __i386__
 # endif
+# ifdef __x86_64__
+#  define x86_64DarwinArchitecture
+#  undef __x86_64__
+# endif
 #endif
 
 #if defined(clipper) || defined(__clipper__)
diff -Naurp xorg-cf-files-1.0.2.orig/darwin.cf xorg-cf-files-1.0.2/darwin.cf
--- xorg-cf-files-1.0.2.orig/darwin.cf	2005-03-31 17:22:07.000000000 -0800
+++ xorg-cf-files-1.0.2/darwin.cf	2009-06-08 17:04:09.000000000 -0700
@@ -230,6 +230,17 @@ XCOMM operating system:  OSName (OSMajor
 # define ByteOrder              X_LITTLE_ENDIAN
 #endif /* i386DarwinArchitecture */
 
+
+#ifdef x86_64DarwinArchitecture
+# define DarwinMachineDefines   -D__x86_64__
+# if HasGcc3
+#  define OptimizedCDebugFlags  -Os
+# else
+#  define OptimizedCDebugFlags  -O2 -fno-strength-reduce
+# endif
+# define ByteOrder              X_LITTLE_ENDIAN
+#endif /* x86_64DarwinArchitecture */
+
 #if OSMajorVersion < 7
 # define DarwinLocaleDefines    -DX_LOCALE
 #else
