--- darwin.cf.orig	2010-09-04 09:19:43.000000000 +1000
+++ darwin.cf	2012-06-01 15:29:58.000000000 +1000
@@ -169,23 +169,31 @@
  * The gcc 3.3 cpp that Apple ships inserts spurious #pragmas,
  * so we use the 3.1-based version in this case.
  */
+#ifndef CppCmd
 #if (GccMajorVersion == 3) && (GccMinorVersion == 3)
 # define CppCmd                 /usr/bin/cpp3
 #else
 # define CppCmd                 /usr/bin/cpp
 #endif
+#endif
 #if (GccMajorVersion == 3) && (GccMinorVersion == 3) || (GccMajorVersion > 3)
 # define StandardCppOptions     -traditional -D__GNUC__
 #endif
+#ifndef RawCppCmd
 #if HasGcc3
 # define RawCppCmd              CppCmd -undef
 #else
 # define RawCppCmd              CppCmd
 #endif
+#endif
 
 /* we have a gcc compatible compiler, but its not called gcc */
+#ifndef CcCmd
 #define CcCmd                   /usr/bin/cc
+#endif
+#ifndef CplusplusCmd
 #define CplusplusCmd            /usr/bin/c++
+#endif
 
 /* default flags to pass to cc */
 #ifndef DefaultCCOptions
