--- PKG-INFO.orig	2020-02-01 18:00:00
+++ PKG-INFO	2026-06-20 16:38:33
@@ -26,7 +26,7 @@
 Classifier: Topic :: Internet :: WWW/HTTP
 Requires-Python: >=3.10
 Requires-Dist: anyio
-Requires-Dist: httpcore2==2.6.0
+Requires-Dist: httpcore2
 Requires-Dist: idna>=3.18
 Requires-Dist: truststore>=0.10
 Requires-Dist: typing-extensions>=4.5.0; python_version < '3.13'
--- pyproject.toml.orig	2020-02-01 18:00:00
+++ pyproject.toml	2026-06-20 16:14:09
@@ -45,7 +45,7 @@
 [tool.hatch.metadata.hooks.uv-dynamic-versioning]
 dependencies = [
     "truststore>=0.10",
-    "httpcore2=={{ version }}",
+    "httpcore2",
     "anyio",
     "idna>=3.18",
     "typing_extensions>=4.5.0; python_version < '3.13'",
