--- mgzip.c.orig	2009-02-21 21:04:31.000000000 +0900
+++ mgzip.c	2009-02-21 21:05:04.000000000 +0900
@@ -37,7 +37,7 @@
 /* -------------------------------------------------------- */
 
 /* gzip flag byte needs bit 0x04 set for extra field present */
-char gz_header[] = { 0x1f, 0x8b, 0x08, 0x04,  /* stolen from an existing .gz file */
+char mgzip_gz_header[] = { 0x1f, 0x8b, 0x08, 0x04,  /* stolen from an existing .gz file */
                        0x00, 0x00, 0x00, 0x00,
                        0x00, 0x03, 0x04, 0x00 };  /* modified for a 4 byte extra field */
                       
@@ -219,7 +219,7 @@
 		crc = crc32(crc, gzip_in_buffer, length); 
 
 		/* write length of entire wadge to output queue */
-		outlength = s.total_out + sizeof(gz_header) + sizeof(outlength) + sizeof(crc) + sizeof(length);
+		outlength = s.total_out + sizeof(mgzip_gz_header) + sizeof(outlength) + sizeof(crc) + sizeof(length);
 
 #ifdef DEBUG
 fprintf(stderr, "worker %d: sending on %d bytes.\n", worker_number, outlength); 
@@ -229,7 +229,7 @@
 		/* gzip_writer_thread will be reading it natively. 			  */
 		enqueue(outq, &outlength, sizeof(outlength), Q_ALL_OR_NOTHING | Q_BLOCK); 
 		/* write valid gzip format file to output queue */
-		enqueue(outq, gz_header, sizeof(gz_header), Q_ALL_OR_NOTHING | Q_BLOCK);
+		enqueue(outq, mgzip_gz_header, sizeof(mgzip_gz_header), Q_ALL_OR_NOTHING | Q_BLOCK);
 		outlength |= 0x7d000000;  /* mgzip magic byte */ 
 #ifdef WORDS_BIGENDIAN
 		longswap(&outlength);
