--- configure.orig	2008-02-02 03:38:42.000000000 +1100
+++ configure	2008-12-29 17:07:30.000000000 +1100
@@ -728,7 +728,7 @@
 SHELL=${CONFIG_SHELL-/bin/sh}
 
 # Identity of this package.
-PACKAGE_NAME='libtool'
+PACKAGE_NAME='libtool1'
 PACKAGE_TARNAME='libtool'
 PACKAGE_VERSION='1.5.26'
 PACKAGE_STRING='libtool 1.5.26'
@@ -2322,7 +2322,7 @@
 
 
 # Define the identity of the package.
- PACKAGE=libtool
+ PACKAGE=libtool1
  VERSION=1.5.26
 
 
