--- Makefile.in.orig	2008-02-02 03:38:41.000000000 +1100
+++ Makefile.in	2008-12-29 16:59:33.000000000 +1100
@@ -215,7 +215,7 @@
 DIST_SUBDIRS = $(BUILD_SUBDIRS) $(CONF_SUBDIRS)
 CPPLAGS = @CPPFLAGS@
 FLIBS = @FLIBS@
-aclocal_macros = libtool.m4 ltdl.m4
+aclocal_macros = libtool1.m4 ltdl1.m4
 EXTRA_DIST = $(aclocal_macros) bootstrap libtoolize.in ltmain.in \
 	mkstamp ChangeLog.0 ChangeLog.1
 
@@ -253,7 +253,7 @@
 
 # TSDEPS will be defined to TSDEPS_DIST at `make dist' time
 TSDEPS = 
-TSDEPS_DIST = ChangeLog libtool.m4
+TSDEPS_DIST = ChangeLog libtool1.m4
 CVS = cvs # set it to `:' to avoid CVS operations
 WGET = wget
 WGETSGO = $(WGET) 'http://savannah.gnu.org/cgi-bin/viewcvs/~checkout~'
@@ -805,14 +805,14 @@
 # All our rules should depend on these demo files.
 all-recursive: $(srcdir)/libltdl/acinclude.m4 $(ACINCLUDE_M4_LIST)
 
-@srcdir@/libltdl/acinclude.m4: $(srcdir)/libtool.m4 $(srcdir)/ltdl.m4
+@srcdir@/libltdl/acinclude.m4: $(srcdir)/libtool1.m4 $(srcdir)/ltdl1.m4
 	-rm -f $@ $@T
-	cat $(srcdir)/libtool.m4 $(srcdir)/ltdl.m4 > $@T
+	cat $(srcdir)/libtool1.m4 $(srcdir)/ltdl1.m4 > $@T
 	mv $@T $@
 
-@ACINCLUDE_M4_LIST@: $(srcdir)/libtool.m4
+@ACINCLUDE_M4_LIST@: $(srcdir)/libtool1.m4
 	-rm -f $@ $@T
-	cat $(srcdir)/libtool.m4 > $@T
+	cat $(srcdir)/libtool1.m4 > $@T
 	mv $@T $@
 
 .PHONY: configure-subdirs
