--- configure.ac	2025-06-11 00:04:17.000000000 +0800
+++ configure.ac	2025-10-15 06:40:37.000000000 +0800
@@ -349,7 +349,7 @@
 	AC_SUBST([HARDENING_LDFLAGS])
 fi
 
-AM_CFLAGS="$CFLAGS $COVERAGE_CFLAGS -Wall -Werror -Wshadow -Wreturn-type -Wsign-compare -Wno-self-assign -Wmissing-prototypes"
+AM_CFLAGS="$CFLAGS $COVERAGE_CFLAGS -Wall -Wshadow -Wreturn-type -Wsign-compare -Wmissing-prototypes"
 AM_CFLAGS="$AM_CFLAGS"
 AM_LDFLAGS="$LDFLAGS $COVERAGE_LDFLAGS"
 
