--- configure.orig	2016-06-18 22:35:29.000000000 +1000
+++ configure	2016-06-30 18:41:02.000000000 +1000
@@ -13623,6 +13623,23 @@
 fi
 
 
+# getline
+ac_fn_c_check_func "$LINENO" "getline" "ac_cv_func_getline"
+if test "x$ac_cv_func_getline" = xyes; then :
+  found_getline=yes
+else
+  found_getline=no
+fi
+
+ if test "x$found_getline" = xyes; then
+  HAVE_GETLINE_TRUE=
+  HAVE_GETLINE_FALSE='#'
+else
+  HAVE_GETLINE_TRUE='#'
+  HAVE_GETLINE_FALSE=
+fi
+
+
 # vis
 ac_fn_c_check_func "$LINENO" "vis" "ac_cv_func_vis"
 if test "x$ac_cv_func_vis" = xyes; then :
