--- configure.orig	2016-06-18 22:35:29.000000000 +1000
+++ configure	2016-06-30 21:15:07.000000000 +1000
@@ -640,6 +640,7 @@
 HAVE_STRLCAT_TRUE
 HAVE_STRLCPY_FALSE
 HAVE_STRLCPY_TRUE
+HAVE_GETLINE_FALSE
 LIBOBJS
 ENABLE_EXAMPLES_FALSE
 ENABLE_EXAMPLES_TRUE
@@ -13623,6 +13624,23 @@
 fi
 
 
+# getline
+ac_fn_c_check_func "$LINENO" "getline" "ac_cv_func_getline"
+if test "x$ac_cv_func_getline" = xyes; then :
+  found_getline=yes
+else
+  found_getline=no
+fi
+
+ if test "x$found_getline" = xyes; then
+  HAVE_GETLINE_TRUE=
+  HAVE_GETLINE_FALSE='#'
+else
+  HAVE_GETLINE_TRUE='#'
+  HAVE_GETLINE_FALSE=
+fi
+
+
 # vis
 ac_fn_c_check_func "$LINENO" "vis" "ac_cv_func_vis"
 if test "x$ac_cv_func_vis" = xyes; then :
