--- CMakeLists.txt	2014-10-21 23:47:40.000000000 +0800
+++ CMakeLists.txt	2023-12-18 22:03:24.000000000 +0800
@@ -195,11 +195,11 @@
     if(APPLE)
         # this is a must to load the lib correctly
         set_target_properties(qtermwidget4plugin PROPERTIES
-            INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/qt4/plugins/designer"
+            INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/libexec/qt4/share/plugins/designer"
         )
     endif()
 
-    install(TARGETS qtermwidget4plugin DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/qt4/plugins/designer")
+    install(TARGETS qtermwidget4plugin DESTINATION "${CMAKE_INSTALL_PREFIX}/libexec/qt4/share/plugins/designer")
 
 endif (BUILD_DESIGNER_PLUGIN)
 # end of designer plugin
