--- meson.build.orig	2022-05-10 08:57:22.000000000 -0400
+++ meson.build	2022-05-10 08:57:32.000000000 -0400
@@ -20,15 +20,13 @@
 glib = dependency('glib-2.0')
 cairo = dependency('cairo')
 mupdf = cc.find_library('mupdf')
-mupdfthird = cc.find_library('mupdf-third')
 
 build_dependencies = [
   zathura,
   girara,
   glib,
   cairo,
-  mupdf,
-  mupdfthird
+  mupdf
 ]
 
 if get_option('link-external')
