diff --git a/config.mk b/config.mk
index 6407ae6..404f4f1 100644
--- a/config.mk
+++ b/config.mk
@@ -32,10 +32,16 @@ endif
 OPENSSL_INC ?= $(shell pkg-config --cflags libcrypto)
 OPENSSL_LIB ?= $(shell pkg-config --libs libcrypto)
 
+FREETYPE_INC ?= $(shell pkg-config --cflags freetype2)
+FREETYPE_LIB ?= $(shell pkg-config --libs freetype2)
+
+HARFBUZZ_INC ?= $(shell pkg-config --cflags harfbuzz)
+HARFBUZZ_LIB ?= $(shell pkg-config --libs harfbuzz)
+
 MUPDF_LIB ?= -lmupdf -lmupdfthird
 
-INCS = ${GTK_INC} ${GIRARA_INC} ${OPENSSL_INC} ${ZATHURA_INC}
-LIBS = ${GTK_LIB} ${GIRARA_LIB} ${MUPDF_LIB} ${OPENSSL_LIB} -ljbig2dec -lopenjp2 -ljpeg
+INCS = ${GTK_INC} ${GIRARA_INC} ${OPENSSL_INC} ${ZATHURA_INC} ${FREETYPE_INC} ${HARFBUZZ_INC}
+LIBS = ${GTK_LIB} ${GIRARA_LIB} ${MUPDF_LIB} ${OPENSSL_LIB} ${FREETYPE_LIB} ${HARFBUZZ_LIB} -ljbig2dec -lopenjp2 -ljpeg -lz
 
 # compiler flags
 CFLAGS += -std=c99 -fPIC -pedantic -Wall -Wno-format-zero-length $(INCS)
