--- generators/CMakeLists.txt.old	2018-06-20 01:15:20.000000000 +0900
+++ generators/CMakeLists.txt	2018-06-20 01:14:01.000000000 +0900
@@ -2,7 +2,7 @@
 
 set(LIBSPECTRE_MINIMUM_VERSION "0.2")
 
-macro_optional_find_package(Poppler)
+#macro_optional_find_package(Poppler)
 macro_log_feature(HAVE_POPPLER_0_12_1 "Poppler-Qt4" "A PDF rendering library" "http://poppler.freedesktop.org" FALSE "0.12.1" "Support for PDF files in okular.")
 
 macro_optional_find_package(LibSpectre)
