--- texstudio.pro.orig	2017-08-05 17:42:45.000000000 +0800
+++ texstudio.pro	2024-06-23 17:55:39
@@ -65,6 +65,8 @@
 CONFIG += qt
 CONFIG -= precompile_header
 
+CONFIG -= debug
+
 # allow loading extra config by file for automatic compilations (OBS)
 exists(texstudio.pri):include(texstudio.pri)
 QT += network \
@@ -158,11 +160,10 @@
     config += unix
 
     # #universal tiger
-    CONFIG += link_prl \
-        x86_64
+    CONFIG += link_prl
 
     # QMAKE_MAC_SDK = /Developer/SDKs/MacOSX10.4u.sdk
-    QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.15
+    # QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.15
     target.path = /Applications
     manual.path = Contents/Resources
     manualOSX.path = Contents/Resources
@@ -330,7 +331,7 @@
         utilities/dictionaries/th_de_DE_v2.dat \
         utilities/AUTHORS \
         utilities/COPYING \
-        utilities/manual/CHANGELOG.md
+        utilities/manual/source/CHANGELOG.md
     manual.files = \
         utilities/manual/build/html/*.html
         utilities/manual/build/html/searchindex.js
@@ -435,7 +436,7 @@
 
   QMAKE_CXXFLAGS += -fno-omit-frame-pointer
   win32: QMAKE_CXXFLAGS += -fpermissive
-  !win32:!haiku: QMAKE_LFLAGS += -rdynamic # option not supported by mingw and haiku
+  !macx:!haiku: QMAKE_LFLAGS += -rdynamic # option not supported by macOS and haiku
 } else {
   DEFINES += _CRT_SECURE_NO_WARNINGS
 }
