--- src/libtracker-sparql/Makefile.am.orig	2016-07-16 17:33:00.000000000 -0700
+++ src/libtracker-sparql/Makefile.am	2016-08-25 00:23:19.000000000 -0700
@@ -115,7 +115,7 @@
 
 if HAVE_INTROSPECTION
 Tracker-@TRACKER_API_VERSION@.typelib: Tracker-$(TRACKER_API_VERSION).gir
-	@INTROSPECTION_COMPILER@ --shared-library=libtracker-sparql-@TRACKER_API_VERSION@.so.0 -o $@ $^
+	@INTROSPECTION_COMPILER@ --shared-library=${prefix}/lib/libtracker-sparql-@TRACKER_API_VERSION@.dylib -o $@ $^
 
 girdir = $(datadir)/gir-1.0
 gir_DATA = Tracker-$(TRACKER_API_VERSION).gir
