--- video/out/cocoa_common.m.orig
+++ video/out/cocoa_common.m
@@ -22,6 +22,7 @@
 #import <IOKit/IOKitLib.h>
 #import <AppKit/AppKit.h>
 #include <mach/mach.h>
+#import <CoreVideo/CoreVideo.h>
 
 #import "cocoa_common.h"
 #import "video/out/cocoa/window.h"
--- waftools/fragments/cocoa.m.orig
+++ waftools/fragments/cocoa.m
@@ -1,6 +1,7 @@
 #import <CoreServices/CoreServices.h>
 #import <OpenGL/OpenGL.h>
 #import <Cocoa/Cocoa.h>
+#import <CoreVideo/CoreVideo.h>
 
 #include "osdep/macosx_compat.h"
 
