--- configure.ac.orig	2013-12-31 23:26:06.000000000 -0800
+++ configure.ac	2013-12-31 23:35:58.000000000 -0800
@@ -138,22 +138,11 @@ case "${host_os}" in
     SYS=dragonfly
     ;;
   darwin*)
-
-    dnl Force gcc "-arch" flag
-    ARCH_flag=""
     case "${host}" in
       i?86*)
-        ARCH_flag="-arch i386"
         YASMFLAGS="-f macho32"
       ;;
-      ppc64*)
-        ARCH_flag="-arch ppc64"
-      ;;
-      ppc*)
-        ARCH_flag="-arch ppc"
-      ;;
       x86_64*)
-        ARCH_flag="-arch x86_64"
         YASMFLAGS="-f macho64"
       ;;
       arm*)
@@ -163,11 +152,10 @@ case "${host_os}" in
     esac
 
     SYS=darwin
-    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS ${ARCH_flag}"
-    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS ${ARCH_flag}"
-    CPPFLAGS="${CPPFLAGS} ${ARCH_flag}"
-    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu99 ${ARCH_flag}"
-    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names ${ARCH_flag}"
+    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS"
+    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS"
+    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu99"
+    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names"
     VLC_ADD_LIBS([libvlc vlc],[-Wl,-undefined,dynamic_lookup,-framework,AppKit])
     VLC_ADD_LIBS([avcodec access_avio swscale postproc i420_rgb_mmx x262 x264 x26410b],[-Wl,-read_only_relocs,suppress])
     VLC_ADD_LIBS([libvlccore],[-Wl,-framework,CoreFoundation,-framework,CoreServices])
