diff --git src/Core/Unix/MacOSX/CoreMacOSX.cpp src/Core/Unix/MacOSX/CoreMacOSX.cpp
index 251e4c65..2caa7898 100644
--- src/Core/Unix/MacOSX/CoreMacOSX.cpp
+++ src/Core/Unix/MacOSX/CoreMacOSX.cpp
@@ -129,7 +129,7 @@ namespace VeraCrypt
 
 		// look for OSXFuse dynamic library
 		struct stat sb;
-		if (0 != stat("/usr/local/lib/libosxfuse_i64.2.dylib", &sb))
+		if (0 != stat("@PREFIX@/lib/libosxfuse_i64.2.dylib", &sb))
 		{
 			throw HigherFuseVersionRequired (SRC_POS);
 		}
