diff -r -u src.orig/password-store.sh src/password-store.sh
--- src.orig/password-store.sh	2014-07-06 18:51:05.000000000 +0200
+++ src/password-store.sh	2014-07-06 18:51:45.000000000 +0200
@@ -168,7 +168,7 @@
 	[[ $1 == "nowarn" ]] && warn=0
 	local template="$PROGRAM.XXXXXXXXXXXXX"
 	if [[ -d /dev/shm && -w /dev/shm && -x /dev/shm ]]; then
-		SECURE_TMPDIR="$(mktemp -d "/dev/shm/$template")"
+		SECURE_TMPDIR="$(${MKTEMP} -d "/dev/shm/$template")"
 		remove_tmpfile() {
 			rm -rf "$SECURE_TMPDIR"
 		}
@@ -182,7 +182,7 @@
 		Are you sure you would like to continue?
 		_EOF
 		)"
-		SECURE_TMPDIR="$(mktemp -d "${TMPDIR:-/tmp}/$template")"
+		SECURE_TMPDIR="$(${MKTEMP} -d "${TMPDIR:-/tmp}/$template")"
 		shred_tmpfile() {
 			find "$SECURE_TMPDIR" -type f -exec $SHRED {} +
 			rm -rf "$SECURE_TMPDIR"
@@ -421,7 +421,7 @@
 	local passfile="$PREFIX/$path.gpg"
 
 	tmpdir #Defines $SECURE_TMPDIR
-	local tmp_file="$(mktemp -u "$SECURE_TMPDIR/XXXXX")-${path//\//-}.txt"
+	local tmp_file="$(${MKTEMP} -u "$SECURE_TMPDIR/XXXXX")-${path//\//-}.txt"
 
 
 	local action="Add"
diff -r -u src.orig/platform/darwin.sh src/platform/darwin.sh
--- src.orig/platform/darwin.sh	2014-07-06 18:53:23.000000000 +0200
+++ src/platform/darwin.sh	2014-07-06 18:53:02.000000000 +0200
@@ -24,7 +24,7 @@
 		rm -rf "$SECURE_TMPDIR"
 	}
 	trap unmount_tmpdir INT TERM EXIT
-	SECURE_TMPDIR="$(mktemp -d "${TMPDIR:-/tmp}/$PROGRAM.XXXXXXXXXXXXX")"
+	SECURE_TMPDIR="$(${MKTEMP} -d "${TMPDIR:-/tmp}/$PROGRAM.XXXXXXXXXXXXX")"
 	DARWIN_RAMDISK_DEV="$(hdid -drivekey system-image=yes -nomount 'ram://32768' | cut -d ' ' -f 1)" # 32768 sectors = 16 mb
 	[[ -z $DARWIN_RAMDISK_DEV ]] && die "Error: could not create ramdisk."
 	newfs_hfs -M 700 "$DARWIN_RAMDISK_DEV" &>/dev/null || die "Error: could not create filesystem on ramdisk."
@@ -33,3 +33,4 @@
 
 GETOPT=@@PREFIX@@/bin/getopt
 SHRED="srm -f -z"
+MKTEMP="/usr/bin/mktemp"
