--- a/src/turtle_lexer.l	2011-07-23 17:17:24.000000000 -0700
+++ b/src/turtle_lexer.l	2011-07-23 17:18:14.000000000 -0700
@@ -345,7 +345,7 @@
                   yyterminate();  }
 
 
-{QUOTEDURI}[\ \t\v\n]*(":-")?[\ \t\v\n]*"{"   {
+{QUOTEDURI}[\ \t\v\n]*("=")?[\ \t\v\n]*"{"   {
                   raptor_stringbuffer* sb;
                   unsigned char* uri_string;
 
@@ -377,11 +377,11 @@
                     TURTLE_LEXER_OOM();
                 return GRAPH_NAME_LEFT_CURLY; }
 
-{QNAME}[\ \t\v\n]*(":-")?[\ \t\v\n]*"{"   {
+{QNAME}[\ \t\v\n]*("=")?[\ \t\v\n]*"{"   {
                while(1) {
                   int c = yytext[yyleng - 1];
                   if(c == '{' || c == ' ' || c=='\t' || c == '\v' || c == '\n' ||
-                     c == ':' || c == '-') {
+                     c == '=') {
                     yyleng--;
                   } else
                     break;
