--- posix.mak.orig	2018-04-30 07:38:51.000000000 -0700
+++ posix.mak	2018-05-12 11:48:28.000000000 -0700
@@ -40,10 +40,6 @@
 
 include $(DMD_DIR)/src/osmodel.mak
 
-ifeq (osx,$(OS))
-	export MACOSX_DEPLOYMENT_TARGET=10.7
-endif
-
 # Default to a release build, override with BUILD=debug
 ifeq (,$(BUILD))
 BUILD_WAS_SPECIFIED=0
@@ -112,12 +108,7 @@
 endif
 
 # Set CFLAGS
-CFLAGS=$(MODEL_FLAG) -fPIC -DHAVE_UNISTD_H
-ifeq ($(BUILD),debug)
-	CFLAGS += -g
-else
-	CFLAGS += -O3
-endif
+CFLAGS=$(MACPORTS_CFLAGS) $(MODEL_FLAG) -fPIC -DHAVE_UNISTD_H
 
 # Set DFLAGS
 DFLAGS=-conf= -I$(DRUNTIME_PATH)/import $(DMDEXTRAFLAGS) -w -de -dip25 $(MODEL_FLAG) $(PIC) -transition=complex
