--- share/functions/alias.fish.orig	2012-11-07 14:52:32.000000000 +0100
+++ share/functions/alias.fish	2012-11-07 14:52:44.000000000 +0100
@@ -14,9 +14,8 @@
 	switch (count $argv)
 
 		case 1
-			set -l tmp (echo $argv|sed -e "s/\([^=]\)=/\1\n/")
-			set name $tmp[1]
-			set body $tmp[2]
+			set name (echo $argv| cut -d\= -f 1)
+			set body (echo $argv| cut -d\= -f 2-)
 
 		case 2
 			set name $argv[1]
