--- configure.ac	2008-01-12 19:47:44.000000000 -0600
+++ configure.ac	2008-07-19 00:56:11.000000000 -0500
@@ -16,22 +16,22 @@
 # List of output variables produced by this configure script
 #
 
-AC_SUBST( docdir )
-AC_SUBST( HAVE_GETTEXT )
-AC_SUBST( LDFLAGS_FISH )
-AC_SUBST( LIBS_FISH )
-AC_SUBST( LIBS_FISH_INDENT )
-AC_SUBST( LIBS_FISH_PAGER )
-AC_SUBST( LIBS_FISHD )
-AC_SUBST( LIBS_MIMEDB )
-AC_SUBST( LIBS_SET_COLOR )
-AC_SUBST( localedir )
-AC_SUBST( optbindirs )
-AC_SUBST( prefix )
-AC_SUBST( SEQ_FALLBACK )
-AC_SUBST( XSEL )
-AC_SUBST( XSEL_MAN )
-AC_SUBST( XSEL_MAN_PATH )
+AC_SUBST(docdir)
+AC_SUBST(HAVE_GETTEXT)
+AC_SUBST(LDFLAGS_FISH)
+AC_SUBST(LIBS_FISH)
+AC_SUBST(LIBS_FISH_INDENT)
+AC_SUBST(LIBS_FISH_PAGER)
+AC_SUBST(LIBS_FISHD)
+AC_SUBST(LIBS_MIMEDB)
+AC_SUBST(LIBS_SET_COLOR)
+AC_SUBST(localedir)
+AC_SUBST(optbindirs)
+AC_SUBST(prefix)
+AC_SUBST(SEQ_FALLBACK)
+AC_SUBST(XSEL)
+AC_SUBST(XSEL_MAN)
+AC_SUBST(XSEL_MAN_PATH)
 
 #
 # If needed, run autoconf to regenerate the configure file
@@ -519,7 +519,6 @@
 fi
 
 
-AC_SEARCH_LIBS( iconv_open, iconv, , [AC_MSG_ERROR([Could not find an iconv implementation, needed to build fish])] )
 LIBS_FISH=$LIBS
 LIBS=$LIBS_COMMON
 
