diff -ru fish-1.23.1.orig/configure.ac fish-1.23.1/configure.ac
--- configure.ac-old	2009-03-08 10:46:47.000000000 -0400
+++ configure.ac	2009-03-08 15:40:25.000000000 -0400
@@ -90,43 +90,6 @@
 fi
 
 
-#
-# Detect directories which may contain additional headers, libraries
-# and commands. This needs to be done early - before Autoconf starts
-# to mess with CFLAGS and all the other environemnt variables.
-#
-# This mostly helps OS X users, since fink usually installs out of
-# tree and doesn't update CFLAGS.
-#
-
-for i in /usr/pkg /sw /opt /opt/local; do
-
-	AC_MSG_CHECKING([for $i/include include directory])
-	if test -d $i/include; then
-		AC_MSG_RESULT(yes)
-		CPPFLAGS="$CPPFLAGS -I$i/include/"
-		CFLAGS="$CFLAGS -I$i/include/"
-	else
-	AC_MSG_RESULT(no)
-	fi
-
-	AC_MSG_CHECKING([for $i/lib library directory])
-	if test -d $i/lib; then
-		AC_MSG_RESULT(yes)
-		LDFLAGS="$LDFLAGS -L$i/lib/ -R$i/lib/"
-	else
-		AC_MSG_RESULT(no)
-	fi
-
-	AC_MSG_CHECKING([for $i/bin command directory])
-	if test -d $i/bin; then
-		AC_MSG_RESULT(yes)
-		optbindirs="$optbindirs $i/bin"
-	else
-		AC_MSG_RESULT(no)
-	fi
-
-done
 
 
 #
@@ -521,7 +484,6 @@
 
 # Check for libiconv_open if we can't find iconv_open. Silly OS X does
 # weird macro magic for the sole purpose of amusing me.
-AC_SEARCH_LIBS( iconv_open, iconv, , [AC_SEARCH_LIBS( libiconv_open, iconv, , [AC_MSG_ERROR([Could not find an iconv implementation, needed to build fish])] )] )
 
 LIBS_FISH=$LIBS
 LIBS=$LIBS_COMMON
@@ -547,7 +509,6 @@
 if test x$local_gettext != xno; then 
 	AC_SEARCH_LIBS( gettext, intl,,)
 fi
-AC_SEARCH_LIBS( iconv_open, iconv, , [AC_MSG_ERROR([Could not find an iconv implementation, needed to build fish])] )
 LIBS_FISH_PAGER=$LIBS
 LIBS=$LIBS_COMMON
 
@@ -560,7 +521,6 @@
 if test x$local_gettext != xno; then 
 	AC_SEARCH_LIBS( gettext, intl,,)
 fi
-AC_SEARCH_LIBS( iconv_open, iconv, , [AC_MSG_ERROR([Could not find an iconv implementation, needed to build fish])] )
 LIBS_FISHD=$LIBS
 LIBS=$LIBS_COMMON
 
