diff --git a/tools/clang/lib/Driver/ToolChains/Darwin.cpp b/tools/clang/lib/Driver/ToolChains/Darwin.cpp
index 40944be0..34799210 100644
--- a/tools/clang/lib/Driver/ToolChains/Darwin.cpp
+++ b/tools/clang/lib/Driver/ToolChains/Darwin.cpp
@@ -679,8 +679,8 @@ types::ID MachO::LookupTypeForExtension(StringRef Ext) const {
 bool MachO::HasNativeLLVMSupport() const { return true; }
 
 ToolChain::CXXStdlibType Darwin::GetDefaultCXXStdlibType() const {
-  // Default to use libc++ on OS X 10.9+ and iOS 7+.
-  if ((isTargetMacOS() && !isMacosxVersionLT(10, 9)) ||
+  // Default to use libc++ on OS X 10.4+ and iOS 7+.
+  if ((isTargetMacOS() && !isMacosxVersionLT(10, 4)) ||
        (isTargetIOSBased() && !isIPhoneOSVersionLT(7, 0)) ||
        isTargetWatchOSBased())
     return ToolChain::CST_Libcxx;
