--- Python/CMakeLists.txt.orig	2010-10-05 22:37:28.000000000 -0500
+++ Python/CMakeLists.txt	2010-10-05 23:02:22.000000000 -0500
@@ -451,10 +451,10 @@
     )
 endif()
 
-EXECUTE_PROCESS(COMMAND python -c "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
+EXECUTE_PROCESS(COMMAND @PYTHON@ -c "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
                 OUTPUT_VARIABLE _PYTHON_VERSION
 )
-EXECUTE_PROCESS(COMMAND python -c "import sys; sys.stdout.write(sys.prefix)"
+EXECUTE_PROCESS(COMMAND @PYTHON@ -c "import sys; sys.stdout.write(sys.prefix)"
                 OUTPUT_VARIABLE _PYTHON_PREFIX
 )
 
