--- src/Makefile.orig	2008-01-16 07:05:14.000000000 +1100
+++ src/Makefile	2008-02-29 11:11:54.000000000 +1100
@@ -35,8 +35,8 @@
 
 ############ ...AND... activate the lines below for your platform... (X11 libs & header files)
 #### linux
-XLIBS = -L/usr/X11R6/lib -lX11
-XINCLUDEDIR = -I/usr/X11R6/include
+# XLIBS = -L/usr/X11R6/lib -lX11
+# XINCLUDEDIR = -I/usr/X11R6/include
 #### alt linux
 # XLIBS = -lX11
 # XINCLUDEDIR = -I/usr/include/X11
@@ -47,8 +47,8 @@
 # XINCLUDEDIR = -I/usr/openwin/include
 #### mac osx
 # CC = cc
-# XLIBS = -L/usr/X11R6/lib -lX11
-# XINCLUDEDIR = -I/usr/X11R6/include
+XLIBS = -L/usr/X11R6/lib -lX11
+XINCLUDEDIR = -I/usr/X11R6/include
 #### cygwin 
 # XLIBS = -L/usr/X11R6/lib -lX11
 # XINCLUDEDIR = -I/usr/X11R6/include
@@ -92,10 +92,10 @@
 # exetarget: pl 
 
 ########### Option 2: use bundled GD16 (PNG only).  Requires libpng and zlib.
-exetarget: plpng 
-GD16LIBS = -lpng -lz
-GD16H = 
-ZFLAG = -DWZ
+# exetarget: plpng 
+# GD16LIBS = -lpng -lz
+# GD16H = 
+# ZFLAG = -DWZ
 
 ########### Option 3: use your own GD resource.  Requires GD 1.84+, libpng, zlib, libjpeg
 # exetarget: plgd18 
@@ -106,11 +106,11 @@
 
 ########### Option 4: use your own GD resource with FreeType2 fonts enabled.  
 ###########           Requires GD 1.84+, libpng, zlib, libjpeg and libfreetype
-# exetarget: plgd18 
-# GD18LIBS = -lgd -lpng -lz -ljpeg -lfreetype
-# GD18H = 
-# GDFREETYPE = -DGDFREETYPE
-# ZFLAG = -DWZ
+exetarget: plgd18 
+GD18LIBS = -L__PREFIX__/lib -lgd -lpng -lz -ljpeg -lfreetype
+GD18H = -I__PREFIX__/include
+GDFREETYPE = -DGDFREETYPE
+ZFLAG = -DWZ
 
 ########### Option 5: don't use GD at all.
 # exetarget: plnogd 
@@ -131,12 +131,12 @@
 
 
 #### If you do a "make install", where do you want the executable(s) to be moved to?
-INSTALLBIN = /usr/local/bin
+INSTALLBIN = $(DESTDIR)__PREFIX__/bin
 
 
 #### For LOCALE support (non-roman alphabets & collation), uncomment the following..
-# LOCALEOBJ = localef.o
-# LOCALE_FLAG = -DLOCALE
+LOCALEOBJ = localef.o
+LOCALE_FLAG = -DLOCALE
 
 #### SVG output is always available by default.
 #### If you want compressed SVG (.svgz), and you selected GD13 or no GD above, 
