--- src/Makefile.orig	2010-09-22 22:28:21.000000000 -0500
+++ src/Makefile	2010-09-22 22:33:21.000000000 -0500
@@ -25,13 +25,13 @@
 
 
 ############ Option 1: build the pl executable (unix), no X11 ....
-NOX11 = -DNOX11
+# NOX11 = -DNOX11
 
 
 
 ############ Option 2: build the pl executable (unix), with X11 enabled .... 
-# NOX11 = 
-# XOBJ = x11.o interact.o
+NOX11 = 
+XOBJ = x11.o interact.o
 
 ############ ...AND... activate the lines below for your platform... (X11 libs & header files)
 #### linux
@@ -53,7 +53,8 @@
 # XLIBS = -L/usr/X11R6/lib -lX11
 # XINCLUDEDIR = -I/usr/X11R6/include
 # PLATFORM = WIN32
-
+XLIBS = -L$(PREFIX)/lib -lX11
+XINCLUDEDIR = -I$(PREFIX)/include
 
 
 ############ Option 3: build pl executable (native win32 pl.exe)
@@ -89,10 +90,10 @@
 
 
 ########### Option 1: use bundled GD16 (PNG only).  Requires libpng and zlib.
-exetarget: plpng 
-GD16LIBS = -lpng -lz
-GD16H = 
-ZFLAG = -DWZ
+# exetarget: plpng 
+# GD16LIBS = -lpng -lz
+# GD16H = 
+# ZFLAG = -DWZ
 
 ########### Option 2: use bundled GD13 (pseudoGIF only). No additional lib dependencies.
 # exetarget: pl 
@@ -106,11 +107,11 @@
 
 ########### Option 4: use your own GD resource with FreeType2 fonts enabled.  
 ###########           Requires GD 1.84+, libpng, zlib, libjpeg and libfreetype
-# exetarget: plgd18 
-# GD18LIBS = -lgd -lpng -lz -ljpeg -lfreetype
-# GD18H = 
-# GDFREETYPE = -DGDFREETYPE
-# ZFLAG = -DWZ
+exetarget: plgd18 
+GD18LIBS = -lgd -lpng -lz -ljpeg -lfreetype
+GD18H = 
+GDFREETYPE = -DGDFREETYPE
+ZFLAG = -DWZ
 
 ########### Option 5: don't use GD at all.
 # exetarget: plnogd 
@@ -137,12 +138,12 @@
 
 
 #### If you do a "make install", where do you want the executable(s) to be moved to?
-INSTALLBIN = /usr/local/bin
+INSTALLBIN = $(DESTDIR)$(PREFIX)/bin
 
 
 #### For LOCALE support (non-roman alphabets & collation), uncomment the following..
-# LOCALEOBJ = localef.o
-# LOCALE_FLAG = -DLOCALE
+LOCALEOBJ = localef.o
+LOCALE_FLAG = -DLOCALE
 
 #### SVG output is always available by default.
 #### If you want compressed SVG (.svgz), and you selected GD13 or no GD above, 
