--- configure.ac.orig	2023-07-06 08:57:20.199117372 -0400
+++ configure.ac	2023-07-06 08:59:19.531579439 -0400
@@ -277,8 +277,6 @@
   src/maple2c/gga_vxc/Makefile
   src/maple2c/mgga_exc/Makefile
   src/maple2c/mgga_vxc/Makefile
-  testsuite/Makefile
-  testsuite/pytest/Makefile
   pkgconfig/Makefile
   pkgconfig/libxc.pc
   pkgconfig/libxcf90.pc
--- Makefile.am.orig	2023-07-06 09:08:02.678245720 -0400
+++ Makefile.am	2023-07-06 09:08:22.076046605 -0400
@@ -8,7 +8,7 @@
 ##
 
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = pkgconfig scripts maple src testsuite cmake pylibxc
+SUBDIRS = pkgconfig scripts maple src cmake pylibxc
 
 include_HEADERS = xc_version.h libxc.bib
 EXTRA_DIST = ChangeLog.md PACKAGING setup.py config.h.cmake.in CMakeLists.txt CITATION
