--- bearssl/bearssl.h	2018-08-15 04:41:54.000000000 +0800
+++ bearssl/bearssl.h	2023-08-01 22:57:29.000000000 +0800
@@ -124,18 +124,18 @@
  * BearSSL source archive also comes with sample code.
  */
 
-#include "bearssl_hash.h"
-#include "bearssl_hmac.h"
-#include "bearssl_kdf.h"
-#include "bearssl_rand.h"
-#include "bearssl_prf.h"
-#include "bearssl_block.h"
-#include "bearssl_aead.h"
-#include "bearssl_rsa.h"
-#include "bearssl_ec.h"
-#include "bearssl_ssl.h"
-#include "bearssl_x509.h"
-#include "bearssl_pem.h"
+#include "bearssl/bearssl_hash.h"
+#include "bearssl/bearssl_hmac.h"
+#include "bearssl/bearssl_kdf.h"
+#include "bearssl/bearssl_rand.h"
+#include "bearssl/bearssl_prf.h"
+#include "bearssl/bearssl_block.h"
+#include "bearssl/bearssl_aead.h"
+#include "bearssl/bearssl_rsa.h"
+#include "bearssl/bearssl_ec.h"
+#include "bearssl/bearssl_ssl.h"
+#include "bearssl/bearssl_x509.h"
+#include "bearssl/bearssl_pem.h"
 
 /** \brief Type for a configuration option.
  *
