--- umbrello/uml.cpp.orig	2014-01-03 04:37:28.000000000 +0900
+++ umbrello/uml.cpp	2014-01-24 01:03:47.000000000 +0900
@@ -1018,7 +1018,15 @@
     m_toolsbar->applySettings(m_config->group( "workbar") );
     fileOpenRecent->loadEntries(m_config->group( "Recent Files") );
     setImageMimeType( UmbrelloSettings::imageMimeType() );
-    resize( UmbrelloSettings::geometry());
+    QSize size = UmbrelloSettings::geometry();
+    if (size.isEmpty()) {
+        //Forcing window to fixed size
+        size.setWidth(800);
+        size.setHeight(600);
+        resize(size);
+    }
+    else
+        resize( UmbrelloSettings::geometry());
 }
 
 /**
