diff --git a/h/compprotos.h b/h/compprotos.h
index 1347f02d3..4a6316e84 100644
--- a/h/compprotos.h
+++ b/h/compprotos.h
@@ -85,11 +85,6 @@ char object_to_char(object);
 void not_a_symbol(object);
 object number_expt(object,object);
 object fLrow_major_aref(object,fixnum);
-#if SIZEOF_SIZE_T == SIZEOF_INT
-void *alloca(unsigned);
-#else
-void *alloca(unsigned long);
-#endif
 object cmod(object);
 object ctimes(object,object);
 object cdifference(object,object);
diff --git a/h/cstack.h b/h/cstack.h
index b833a2652..7e1a81cbb 100644
--- a/h/cstack.h
+++ b/h/cstack.h
@@ -30,6 +30,13 @@
 #define SS getpagesize()
 #endif
 
+#ifndef MAP_GROWSDOWN
+#define MAP_GROWSDOWN 0
+#endif
+#ifndef MAP_STACK
+#define MAP_STACK 0
+#endif
+
 #ifdef SET_STACK_POINTER
 {
   void *p,*p1,*b,*s;
diff --git a/o/sfaslmacho.c b/o/sfaslmacho.c
index cf2c64c22..a228c3a19 100644
--- a/o/sfaslmacho.c
+++ b/o/sfaslmacho.c
@@ -66,6 +66,14 @@ store_val(ul *w,ul m,ul v) {
 
 }
 
+static int
+add_val(ul *w,ul m,ul v) {
+
+  return store_val(w,m,v+(*w&m));
+
+}
+
+
 #ifndef _LP64
 /*redirect trampolines gcc-4.0 gives no reloc for stub sections on x86 only*/
 static int
