diff --git a/lsp/gcl_dl.lsp b/lsp/gcl_dl.lsp
index 6f6b35f44..4d8f9962e 100644
--- a/lsp/gcl_dl.lsp
+++ b/lsp/gcl_dl.lsp
@@ -3,9 +3,18 @@
 
 (export '(mdlsym mdl lib-name))
 
+(defvar *dll-extension* #+darwin ".dylib" #+cygwin ".dll" #-(or darwin cygwin) ".so")
+
+;; NOTE: In some versions of macOS (Darwin), GCL loads libSystem.B.dylib, which is in /usr/lib;
+;; And in other versions of macOS, GCL loads libsystem_*.dylib, which is in /usr/lib/system.
 (defun lib-name (p)
-  (if (or (string= p "") (string= p "libc") (string= p "libm")) "" 
-    (string-concatenate #+darwin "/usr/lib/system/" p #+darwin ".dylib" #+cygwin ".dll" #-(or darwin cygwin) ".so")));FIXME
+  (cond ((or (string= p "") (string= p "libc") (string= p "libm"))
+         "")
+        #+darwin ; "/usr/lib/system/" is macOS-specific
+        ((string<= "libsystem" p)
+         (string-concatenate "/usr/lib/system/" p *dll-extension*))
+        (t
+         (string-concatenate #+darwin "/usr/lib/" p *dll-extension*))))
 
 (defun mdl (n p vad)
   (let* ((sym (mdlsym n (lib-name p)))
