--- configure.orig	2013-08-19 13:31:42.000000000 -0700
+++ configure	2013-08-20 06:40:39.000000000 -0700
@@ -13258,7 +13258,7 @@
 	  libgtop_use_machine_h=yes
 	  libgtop_need_server=yes
 	  libgtop_have_sysinfo=yes
-	  libgtop_postinstall='chgrp kmem $(bindir)/libgtop_server2 && chmod g+s $(bindir)/libgtop_server2'
+	  libgtop_postinstall='chgrp kmem $(DESTDIR)$(bindir)/libgtop_server2 && chmod g+s $(DESTDIR)$(bindir)/libgtop_server2'
 	  ;;
 	cygwin*)
 	  libgtop_sysdeps_dir=cygwin
