--- mk/re.mk.orig	2010-11-03 06:34:14.000000000 -0500
+++ mk/re.mk	2011-03-28 01:10:55.000000000 -0500
@@ -271,6 +271,7 @@
 	AR		:= ar
 	AFLAGS		:= cru
 	LIB_SUFFIX	:= .dylib
+	SH_LFLAGS	+= -install_name $(PREFIX)/lib/libre$(LIB_SUFFIX)
 	ARCH            := $(shell echo $(MACHINE) | \
 				sed -e 's/\([^-]*\)-.*/\1/')
 endif
@@ -488,9 +488,6 @@
 #
 # Misc tools section
 #
-CCACHE	:= $(shell [ -e /usr/bin/ccache ] 2>/dev/null \
-	|| [ -e /opt/local/bin/ccache ] \
-	&& echo "ccache")
 CFLAGS  += -DVERSION=\"$(VERSION)\"
 
 # Enable gcov Coverage testing
