--- src/player.h.orig	2018-04-25 14:15:44.000000000 -0700
+++ src/player.h	2018-04-25 14:15:58.000000000 -0700
@@ -34,7 +34,6 @@
 #include <ao/ao.h>
 #include <libavformat/avformat.h>
 #include <libavfilter/avfilter.h>
-#include <libavfilter/avfiltergraph.h>
 #include <piano.h>
 
 #include "settings.h"
--- src/player.c.orig	2018-04-25 14:00:42.000000000 -0700
+++ src/player.c	2018-04-25 14:01:58.000000000 -0700
@@ -36,7 +36,6 @@
 #include <libavcodec/avcodec.h>
 #include <libavutil/avutil.h>
 #include <libavfilter/avfilter.h>
-#include <libavfilter/avfiltergraph.h>
 #include <libavfilter/buffersink.h>
 #include <libavfilter/buffersrc.h>
 #ifdef HAVE_LIBAVFILTER_AVCODEC_H
