--- Makefile.orig	2010-06-18 11:29:59.000000000 -0700
+++ Makefile	2010-06-18 12:15:22.000000000 -0700
@@ -1,6 +1,6 @@
 # makefile of pianobar
 
-PREFIX:=/usr/local
+PREFIX:=@PREFIX@
 BINDIR:=${PREFIX}/bin
 LIBDIR:=${PREFIX}/lib
 MANDIR:=${PREFIX}/share/man
@@ -87,7 +87,7 @@
 pianobar: ${PIANOBAR_OBJ} ${PIANOBAR_HDR} ${LIBPIANO_OBJ} ${LIBWAITRESS_OBJ} \
 		${LIBWAITRESS_HDR} ${LIBEZXML_OBJ} ${LIBEZXML_HDR}
 	${CC} ${CFLAGS} ${PIANOBAR_OBJ} ${LIBPIANO_OBJ} ${LIBWAITRESS_OBJ} \
-			${LIBEZXML_OBJ} ${LIBAO_LIB} ${LIBFAAD_LIB} ${LIBMAD_LIB} \
+			${LIBEZXML_OBJ} -L@PREFIX@/lib ${LIBAO_LIB} ${LIBFAAD_LIB} ${LIBMAD_LIB} \
 			${PTHREAD_LIB} ${LIBM_LIB} -o $@
 
 # build shared libpiano
@@ -98,7 +98,7 @@
 
 %.o: %.c
 	${CC} ${CFLAGS} -I ${LIBPIANO_INCLUDE} -I ${LIBWAITRESS_INCLUDE} \
-			-I ${LIBEZXML_INCLUDE} ${LIBFAAD_SWITCH} ${LIBMAD_SWITCH} -c \
+			-I ${LIBEZXML_INCLUDE} -I @PREFIX@/include ${LIBFAAD_SWITCH} ${LIBMAD_SWITCH} -c \
 			-o $@ $<
 
 # create position independent code (for shared libraries)
@@ -113,7 +113,9 @@
 
 all: pianobar libpiano
 
-install: pianobar
+install: pianobar ${DESTDIR}/${BINDIR}/pianobar
+
+${DESTDIR}/${BINDIR}/pianobar:
 	install -d ${DESTDIR}/${BINDIR}/
 	install -m755 pianobar ${DESTDIR}/${BINDIR}/
 	install -d ${DESTDIR}/${MANDIR}/man1/
