--- Makefile.orig	2012-06-24 09:34:26.000000000 -0500
+++ Makefile	2012-07-11 09:32:45.000000000 -0500
@@ -88,7 +88,7 @@
 
 # build pianobar
 ifeq (${DYNLINK},1)
-pianobar: ${PIANOBAR_OBJ} ${PIANOBAR_HDR} libpiano.so.0
+pianobar: ${PIANOBAR_OBJ} ${PIANOBAR_HDR} libpiano.0.dylib
 	@echo "  LINK  $@"
 	@${CC} -o $@ ${PIANOBAR_OBJ} ${LDFLAGS} -lao -lpthread -lm -L. -lpiano \
 			${LIBFAAD_LDFLAGS} ${LIBMAD_LDFLAGS} ${LIBGNUTLS_LDFLAGS}
@@ -103,15 +103,15 @@
 endif
 
 # build shared and static libpiano
-libpiano.so.0: ${LIBPIANO_RELOBJ} ${LIBPIANO_HDR} ${LIBWAITRESS_RELOBJ} \
+libpiano.0.dylib: ${LIBPIANO_RELOBJ} ${LIBPIANO_HDR} ${LIBWAITRESS_RELOBJ} \
 		${LIBWAITRESS_HDR} ${LIBPIANO_OBJ} ${LIBWAITRESS_OBJ}
 	@echo "  LINK  $@"
-	@${CC} -shared -Wl,-soname,libpiano.so.0 ${CFLAGS} ${LDFLAGS} \
-			-o libpiano.so.0.0.0 ${LIBPIANO_RELOBJ} \
+	@${CC} -dynamiclib -Wl,-install_name,${LIBDIR}/libpiano.0.dylib ${CFLAGS} ${LDFLAGS} \
+			-o libpiano.0.0.0.dylib ${LIBPIANO_RELOBJ} \
 			${LIBWAITRESS_RELOBJ} ${LIBGNUTLS_LDFLAGS} ${LIBGCRYPT_LDFLAGS} \
 			${LIBJSONC_LDFLAGS}
-	@ln -s libpiano.so.0.0.0 libpiano.so.0
-	@ln -s libpiano.so.0 libpiano.so
+	@ln -s libpiano.0.0.0.dylib libpiano.0.dylib
+	@ln -s libpiano.0.dylib libpiano.dylib
 	@echo "    AR  libpiano.a"
 	@${AR} rcs libpiano.a ${LIBPIANO_OBJ} ${LIBWAITRESS_OBJ}
 
@@ -131,7 +131,7 @@
 clean:
 	@echo " CLEAN"
 	@${RM} ${PIANOBAR_OBJ} ${LIBPIANO_OBJ} ${LIBWAITRESS_OBJ} ${LIBWAITRESS_OBJ}/test.o \
-			${LIBPIANO_RELOBJ} ${LIBWAITRESS_RELOBJ} pianobar libpiano.so* \
+			${LIBPIANO_RELOBJ} ${LIBWAITRESS_RELOBJ} pianobar libpiano*.dylib \
 			libpiano.a waitress-test
 
 all: pianobar
@@ -183,9 +183,9 @@
 
 install-libpiano:
 	install -d ${DESTDIR}/${LIBDIR}/
-	install -m644 libpiano.so.0.0.0 ${DESTDIR}/${LIBDIR}/
-	ln -s libpiano.so.0.0.0 ${DESTDIR}/${LIBDIR}/libpiano.so.0
-	ln -s libpiano.so.0 ${DESTDIR}/${LIBDIR}/libpiano.so
+	install -m644 libpiano.0.0.0.dylib ${DESTDIR}/${LIBDIR}/
+	ln -s libpiano.0.0.0.dylib ${DESTDIR}/${LIBDIR}/libpiano.0.dylib
+	ln -s libpiano.0.dylib ${DESTDIR}/${LIBDIR}/libpiano.dylib
 	install -m644 libpiano.a ${DESTDIR}/${LIBDIR}/
 	install -d ${DESTDIR}/${INCDIR}/
 	install -m644 src/libpiano/piano.h ${DESTDIR}/${INCDIR}/
