--- src/libpiano/piano.c.orig	2012-04-27 07:52:09.000000000 -0700
+++ src/libpiano/piano.c	2012-04-27 07:52:24.000000000 -0700
@@ -42,7 +42,7 @@
 #include "crypt.h"
 #include "config.h"
 
-#define PIANO_PROTOCOL_VERSION "33"
+#define PIANO_PROTOCOL_VERSION "34"
 #define PIANO_RPC_HOST "www.pandora.com"
 #define PIANO_RPC_PORT "80"
 #define PIANO_RPC_PATH "/radio/xmlrpc/v" PIANO_PROTOCOL_VERSION "?"
@@ -272,6 +272,7 @@
 							"<?xml version=\"1.0\"?><methodCall>"
 							"<methodName>listener.authenticateListener</methodName>"
 							"<params><param><value><int>%lu</int></value></param>"
+							"<param><value><string></string></value></param>"
 							/* user */
 							"<param><value><string>%s</string></value></param>"
 							/* password */
