--- configure	2024-08-27 13:33:10.000000000 +0800
+++ configure	2024-08-27 15:36:45.000000000 +0800
@@ -109,38 +109,11 @@
 
   if [ "$(uname)" = "Darwin" ]; then
 
-    # https://mac.r-project.org/openmp
-    printf "%s" "* checking if R installation supports OpenMP with \"-Xclang -fopenmp\" ... "
-    if CPPFLAGS="${CPPFLAGS} -Xclang -fopenmp" PKG_LIBS="-lomp" "${R_HOME}/bin/R" CMD SHLIB test-omp.c >> config.log 2>&1; then
-      echo "yes"
-      export PKG_CFLAGS="${PKG_CFLAGS} -Xclang -fopenmp"
-      export PKG_LIBS="${PKG_LIBS} -lomp"
-      export R_OPENMP_ENABLED=1
-      return
-    else
-      echo "no"
-    fi
-
-    if [ "$(uname -m)" = "arm64" ]; then
-      HOMEBREW_PREFIX=/opt/homebrew
-    else
-      HOMEBREW_PREFIX=/usr/local
-    fi
-
-    if [ -e "${HOMEBREW_PREFIX}/opt/libomp" ]; then
-      printf "%s" "* checking if libomp installation at ${HOMEBREW_PREFIX}/opt/libomp can be used... "
-      LIBOMP_INCLUDE="-I${HOMEBREW_PREFIX}/opt/libomp/include -Xclang -fopenmp"
-      LIBOMP_LINK="-L${HOMEBREW_PREFIX}/opt/libomp/lib -lomp"
-      if ${CC} ${CFLAGS} ${LIBOMP_INCLUDE} ${LIBOMP_LINK} test-omp.c >> config.log 2>&1; then
-        echo "yes"
-        export PKG_CFLAGS="${PKG_CFLAGS} ${LIBOMP_INCLUDE}"
-        export PKG_LIBS="${PKG_LIBS} ${LIBOMP_LINK}"
-        export R_OPENMP_ENABLED=1
-        return
-      else
-        echo "no"
-      fi
-    fi
+    # Instead of using a defunct check, just set the right variables for OpenMP to work.
+    export PKG_CFLAGS="${PKG_CFLAGS} @MP_OPENMP_CFLAGS@"
+    export PKG_LIBS="${PKG_LIBS} @MP_OPENMP_LFLAGS@"
+    export R_OPENMP_ENABLED=1
+    return
 
   fi # uname=Darwin
 
@@ -162,7 +135,7 @@
   echo "***"
   sed -e "s|@openmp_cflags@||" src/Makevars.in > src/Makevars
 else
-  sed -e "s|@openmp_cflags@|\$(SHLIB_OPENMP_CFLAGS)|" src/Makevars.in > src/Makevars
+  sed -e "s|@openmp_cflags@|@MP_OPENMP_FLAGS@|" src/Makevars.in > src/Makevars
 fi
 
 # retain user supplied PKG_ env variables, #4664. See comments in Makevars.in too.
