diff -ru Ice-3.5.0.orig/cpp/test/IceGrid/deployer/run.py Ice-3.5.0/cpp/test/IceGrid/deployer/run.py
--- Ice-3.5.0.orig/cpp/test/IceGrid/deployer/run.py	2013-03-11 08:19:47.000000000 -0700
+++ Ice-3.5.0/cpp/test/IceGrid/deployer/run.py	2013-03-16 20:28:41.401227641 -0700
@@ -30,9 +30,9 @@
     iceBox = os.path.join(os.getcwd(), "iceboxwrapper")
     iceBoxWrapper = open(iceBox, "w")
     if TestUtil.x64:
-        iceBoxWrapper.write("#!/bin/sh\narch -x86_64 " + TestUtil.getIceBox() + " \"$@\"\n")
+        iceBoxWrapper.write("#!/bin/sh\narch -x86_64 -ppc64 " + TestUtil.getIceBox() + " \"$@\"\n")
     else:
-        iceBoxWrapper.write("#!/bin/sh\narch -i386 -x86_64 " + TestUtil.getIceBox() + " \"$@\"\n")
+        iceBoxWrapper.write("#!/bin/sh\narch -i386 -ppc -x86_64 -ppc64 " + TestUtil.getIceBox() + " \"$@\"\n")
     iceBoxWrapper.close()
     os.chmod(iceBox, 0o700)
 else:
