--- configure.orig	2011-12-01 00:15:46.000000000 -0800
+++ configure	2011-12-01 00:25:45.000000000 -0800
@@ -13497,9 +13497,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print sys.prefix;'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print sys.exec_prefix;'`
 
 
 
@@ -14870,6 +14870,7 @@
 	HAVE_UUID=yes
 fi
 
+HAVE_UUID=no
 
 if test x"$HAVE_UUID" = xyes; then
 
