--- mkspecs/common/mac.conf.orig	2010-11-02 10:44:30.000000000 -0400
+++ mkspecs/common/mac.conf	2010-11-02 10:48:37.000000000 -0400
@@ -40,5 +40,9 @@
 QMAKE_MKDIR		= mkdir -p
 QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
 
+# allow CC and CXX to use CCACHE;
+# use "make CCACHE=ccache" when building
+QMAKE_CC = $(CCACHE) $$QMAKE_CC
+QMAKE_CXX = $(CCACHE) $$QMAKE_CXX
 
 include(unix.conf)
