--- mkspecs/features/qt_functions.prf.orig	2010-09-10 05:04:49.000000000 -0400
+++ mkspecs/features/qt_functions.prf	2010-11-02 14:23:13.000000000 -0400
@@ -16,8 +16,18 @@
 }
 
 defineTest(qtAddLibrary) {
+#    message('LIBS' before $$1 is '$${LIBS}')
+#    message('INCLUDEPATH' before $$1 is '$${INCLUDEPATH}')
+#    message('QMAKE_LFLAGS' before $$1 is '$${QMAKE_LFLAGS}')
+
+    # reorder the includes path: user, this library, Qt headers
+    INCLUDEPATH -= $$QMAKE_INCDIR_QT
     INCLUDEPATH -= $$QMAKE_INCDIR_QT/$$1
-    INCLUDEPATH = $$QMAKE_INCDIR_QT/$$1 $$INCLUDEPATH
+    INCLUDEPATH += $$QMAKE_INCDIR_QT/$$1 $$QMAKE_INCDIR_QT
+
+    # first time through, append the path for Qt's installed libraries:
+    # -after- all of the other necessary libraries and paths
+    !contains(LIBS,-L$$QMAKE_LIBDIR_QT):LIBS = $$LIBS -L$$QMAKE_LIBDIR_QT
 
     LIB_NAME = $$1
     unset(LINKAGE)
@@ -85,6 +95,9 @@
         QMAKE_LFLAGS *= --lsb-shared-libs=$${LIB_NAME}$${QT_LIBINFIX}
     }
     LIBS += $$LINKAGE
+#    message('LIBS' after $$1 is now '$${LIBS}')
+#    message('INCLUDEPATH' after $$1 is now '$${INCLUDEPATH}')
+#    message('QMAKE_LFLAGS' after $$1 is now '$${QMAKE_LFLAGS}')
     export(LIBS)
     export(INCLUDEPATH)
     export(QMAKE_FRAMEWORKPATH)
