--- a/dialog.c.orig	2012-05-15 18:02:39.000000000 -0500
+++ b/dialog.c	2012-05-15 18:03:30.000000000 -0500
@@ -740,7 +740,7 @@
             line[ML];
     int     y,
             i;
-    int	    getline();
+    int	    xgraph_getline();
     unsigned long wamask;
     XSetWindowAttributes wattr;
     XFontStruct *bigFont = PM_FONT("TitleFont");
@@ -778,7 +778,7 @@
     memset(new_info->lines, 0, sizeof(Window) * E_LINES);
 
     lineptr = text;
-    while (getline(&lineptr, line)) {
+    while (xgraph_getline(&lineptr, line)) {
 	if (new_info->num_lines >= new_info->alloc_lines) {
 	    int old_alloc_lines_size = new_info->alloc_lines * sizeof(Window);
 	    new_info->alloc_lines *= 2;
@@ -889,7 +889,7 @@
 
 
 int 
-getline(tptr, lptr)
+xgraph_getline(tptr, lptr)
 char  **tptr;
 char   *lptr;
 
