--- as/driver.c.orig	2018-09-19 08:01:56.000000000 +1000
+++ as/driver.c	2019-01-25 01:35:21.000000000 +1100
@@ -295,11 +295,18 @@ char **envp)
 	    arch_flag.cputype == CPU_TYPE_ARM64 ||
 	    arch_flag.cputype == CPU_TYPE_ARM64_32 ||
 	    arch_flag.cputype == CPU_TYPE_ARM)){
-	    as = makestr(prefix, CLANG, NULL);
-	    if(access(as, F_OK) != 0){
-		printf("%s: assembler (%s) not installed\n", progname, as);
-		exit(1);
-	    }
+        as = makestr(prefix, "__MP_CLANG_NAME__", NULL);
+        if(access(as, F_OK) != 0){
+#if __TRY_SYSTEM_CLANG__
+            as = "/usr/bin/clang";
+            if(access(as, F_OK) != 0){
+                as = NULL;
+            }
+#else
+            as = NULL;
+#endif
+        }
+        if (as != NULL) {
 	    new_argv = allocate((argc + 8) * sizeof(char *));
 	    new_argv[0] = as;
 	    j = 1;
@@ -360,6 +367,7 @@ char **envp)
 		exit(0);
 	    else
 		exit(1);
+        } /* as != NULL */
 	}
 
 	/*
