--- as/driver.c.orig	2012-10-10 17:17:27.000000000 -0700
+++ as/driver.c	2012-10-10 17:21:11.000000000 -0700
@@ -107,8 +107,15 @@ char **envp)
 		    case 'I':	/* -I directory */
 		    case 'm':	/* -mc68000, -mc68010 and mc68020 */
 		    case 'N':	/* -NEXTSTEP-deployment-target */
+			/* We want to skip the next argv if the value is not contained in this argv
+			 * eg: -I dir
+			 */
 			if(p[1] == '\0')
 			    i++;
+
+			/* And in case the value is contained in this argv (eg: -Idir), skip the rest */
+			while(p[1])
+			    p++;
 			break;
 
 		    case 'a':
