diff -Naurp as/as.h as/as.h
--- as/as.h	2007-06-05 18:00:45.000000000 -0700
+++ as/as.h	2012-04-03 10:27:49.000000000 -0700
@@ -155,7 +155,7 @@ extern cpu_subtype_t archflag_cpusubtype
 extern char *specific_archflag;
 
 /* TRUE if the .subsections_via_symbols directive was seen */
-int subsections_via_symbols;
+extern int subsections_via_symbols;
 
 /* -I path options for .includes */
 struct directory_stack {
diff -Naurp otool/print_objc2_32bit.c otool/print_objc2_32bit.c
--- otool/print_objc2_32bit.c	2008-03-31 17:16:58.000000000 -0700
+++ otool/print_objc2_32bit.c	2012-04-03 10:27:55.000000000 -0700
@@ -229,7 +229,7 @@ struct category_t {
     uint32_t protocols;		/* struct protocol_list_t * (32-bit pointer) */
     uint32_t instanceProperties; /* struct objc_property_list *
 				    (32-bit pointer) */
-} category_t;
+};
 
 static
 void
diff -Naurp otool/print_objc2_64bit.c otool/print_objc2_64bit.c
--- otool/print_objc2_64bit.c	2008-03-31 17:17:01.000000000 -0700
+++ otool/print_objc2_64bit.c	2012-04-03 10:27:55.000000000 -0700
@@ -232,7 +232,7 @@ struct category_t {
     uint64_t protocols;		/* struct protocol_list_t * (64-bit pointer) */
     uint64_t instanceProperties; /* struct objc_property_list *
 				    (64-bit pointer) */
-} category_t;
+};
 
 static
 void
