diff -Naurp misc/libtool.c misc/libtool.c
--- misc/libtool.c	1969-12-31 16:00:00.000000000 -0800
+++ misc/libtool.c	1969-12-31 16:00:00.000000000 -0800
@@ -2326,7 +2326,7 @@ struct ofile *ofile)
 			if(archs[0].members[j].offset == 
 			   archs[0].toc_ranlibs[i].ran_off){
 			    archs[0].toc_ranlibs[i].ran_off = 
-				archs[0].members[i].input_member_offset;
+				archs[0].members[j].input_member_offset;
 			    break;
 			}
 		    }
