--- configure.orig	2010-04-05 20:00:26.000000000 -0600
+++ configure	2010-05-02 17:52:37.000000000 -0600
@@ -14971,9 +14971,9 @@
 
 if test $has_python = true
 then
-    PYTHON_CFLAGS="-I${PYTHON_PREFIX}"
+    PYTHON_CFLAGS="-I`@@PREFIX@@/bin/python2.6 -c 'from distutils import sysconfig; print sysconfig.get_python_inc()'`"
 
-    PYTHON_SITE_PACKAGES=`python -c 'from distutils import sysconfig; print sysconfig.get_python_lib()'`
+    PYTHON_SITE_PACKAGES=`@@PREFIX@@/bin/python2.6 -c 'from distutils import sysconfig; print sysconfig.get_python_lib()'`
 
 
 $as_echo "#define HAVE_PYTHON 1" >>confdefs.h
--- src/python/Makefile.in.orig	2010-04-05 20:00:25.000000000 -0600
+++ src/python/Makefile.in	2010-05-23 15:38:52.000000000 -0600
@@ -594,21 +594,22 @@
 	rm -rf $(top_builddir)/lib/python
 
 install-exec-local:
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/_quickfix.so
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/_quickfix.dylib
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix40.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix41.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix42.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix43.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix44.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix50.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix50sp1.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfix50sp2.py
-	rm -rf $(PYTHON_SITE_PACKAGES)/python/quickfixt11.py
-	ln -sf $(prefix)/lib/python/_quickfix.so $(PYTHON_SITE_PACKAGES)/_quickfix.so
-	ln -sf $(prefix)/lib/python/_quickfix.dylib $(PYTHON_SITE_PACKAGES)/_quickfix.dylib
-	cp $(top_builddir)/lib/python/*.py $(PYTHON_SITE_PACKAGES)
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/_quickfix.so
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/_quickfix.dylib
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix40.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix41.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix42.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix43.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix44.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix50.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix50sp1.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfix50sp2.py
+	rm -rf $(DESTDIR)$(PYTHON_SITE_PACKAGES)/python/quickfixt11.py
+	$(mkinstalldirs) $(DESTDIR)$(PYTHON_SITE_PACKAGES)
+	cp $(top_builddir)/lib/python/*.py $(DESTDIR)$(PYTHON_SITE_PACKAGES)
+	cp $(top_builddir)/lib/python/_quickfix.dylib $(DESTDIR)$(PYTHON_SITE_PACKAGES)
+	ln -sf $(PYTHON_SITE_PACKAGES)/_quickfix.dylib $(DESTDIR)$(PYTHON_SITE_PACKAGES)/_quickfix.so
 
 uninstall-local:
 	rm -rf $(PYTHON_SITE_PACKAGES)/_quickfix.so
