--- UnitTest++/Makefile.orig	2014-06-23 23:49:36.000000000 -0500
+++ UnitTest++/Makefile	2014-09-03 15:24:07.000000000 -0500
@@ -1,6 +1,6 @@
-CXX = g++
-CXXFLAGS ?= -g -Wall -W -Winline -Wno-unused-private-field -Wno-overloaded-virtual -ansi 
-LDFLAGS ?= 
+CXX = @CXX@
+CXXFLAGS ?= -g -Wall -W -Winline -Wno-overloaded-virtual -ansi @ARCHFLAGS@
+LDFLAGS ?= @ARCHFLAGS@
 SED = sed
 MV = mv
 RM = rm
@@ -73,11 +73,11 @@
 
 $(lib): $(objects) 
 	@echo Creating $(lib) library...
-	@ar crs $(lib) $(objects)
+	ar crs $(lib) $(objects)
     
 $(test): $(lib) $(test_objects)
 	@echo Linking $(test)...
-	@$(CXX) $(LDFLAGS) -o $(test) $(test_objects) $(lib)
+	$(CXX) $(LDFLAGS) -o $(test) $(test_objects) $(lib)
 	@echo Running unit tests...
 	@./$(test)
 
@@ -85,9 +85,8 @@
 	-@$(RM) $(objects) $(test_objects) $(dependencies) $(test_dependencies) $(test) $(lib) 2> /dev/null
 
 %.o : %.cpp
-	@echo $<
 	@$(call make-depend,$<,$@,$(subst .o,.d,$@))
-	@$(CXX) $(CXXFLAGS) -c $< -o $(patsubst %.cpp, %.o, $<)
+	$(CXX) $(CXXFLAGS) -c $< -o $(patsubst %.cpp, %.o, $<)
 
 install:
 
