--- configure.orig	2010-04-05 20:00:26.000000000 -0600
+++ configure	2010-05-02 15:37:33.000000000 -0600
@@ -14938,9 +14938,9 @@
 
 if test $has_postgresql = true
 then
-    POSTGRESQL_CFLAGS="-I${POSTGRESQL_PREFIX}/include -I${POSTGRESQL_PREFIX}/include/postgresql"
+    POSTGRESQL_CFLAGS="-I${POSTGRESQL_PREFIX}/include -I${POSTGRESQL_PREFIX}/include/postgresql84"
 
-    POSTGRESQL_LIBS="-L${POSTGRESQL_PREFIX}/lib -lpq"
+    POSTGRESQL_LIBS="-L${POSTGRESQL_PREFIX}/lib/postgresql84 -lpq"
 
 
 $as_echo "#define HAVE_POSTGRESQL 1" >>confdefs.h
