--- a/GNUmakefile
+++ b/GNUmakefile
@@ -41,7 +41,7 @@ ifdef SUMO
 	$(RANLIB) $(LIBDIR)/libnwchem-sumo.a
 endif
 	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL)
+#	$(INSTALL)
 #
 # This to link only and avoid tedious subdir makes
 #
@@ -62,7 +62,7 @@ ifdef SUMO
 	$(RANLIB) $(LIBDIR)/libnwchem-sumo.a
 endif
 #	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL)
+#	$(INSTALL)
 
 ccalink:
 	@echo "NWCHEM_CORE_LIBS = $(LIBS)" > ./cca/makefile.h
@@ -79,7 +79,7 @@ ifeq ($(TARGET),SOLARIS)
 endif
 	$(LINK.f) -p -o nwchem_prof nwchem.o stubs.o -lnwctask $(LIBS)
 	/bin/rm -f nwchem.o stubs.o
-	$(INSTALL)
+#	$(INSTALL)
 #
 # This for linking test version
 #
@@ -88,7 +88,7 @@ ifeq ($(TARGET),SOLARIS)
 	/bin/rm -f nwchem_test
 endif
 	$(LINK.f) -o nwchem_test nwchem.o stubs.o -lnwctask $(LIBS)
-	$(INSTALL)
+#	$(INSTALL)
 
 configure_ga:
 ifdef OLD_GA
@@ -255,7 +255,7 @@
 endif
 
 .PHONY:	install
-install: link
+install: all
 	@test $${INSTALL_PREFIX?Please set the environment variable INSTALL_PREFIX}
 	@echo install_prefix is $(INSTALL_PREFIX)
 	@mkdir -p $(INSTALL_PREFIX)/bin  $(INSTALL_PREFIX)/share
