--- NSAttributedString-Ellipsis.m.orig	2004-09-03 08:24:57.000000000 +1000
+++ NSAttributedString-Ellipsis.m	2014-01-30 01:08:31.000000000 +1100
@@ -11,7 +11,7 @@
 
 - ( NSAttributedString * )ellipsisAbbreviatedStringForWidth: ( double )width
 {
-    NSAttributedString          *attrString = nil;
+    /*NSAttributedString          *attrString = nil;
     NSMutableString             *string = [[[ self string ] mutableCopy ] autorelease ];
     OSStatus                    status;
     double                      paddedWidth = ( width - 24.0 );
@@ -19,13 +19,13 @@
     if (( status = TruncateThemeText(( CFMutableStringRef )string,
                     kThemeViewsFont, kThemeStateActive, paddedWidth, truncMiddle, NULL ))
                     != noErr ) {
-        NSLog( @"TruncateThemeText %@ returned error %d", string, ( int )status );
+        NSLog( @"TruncateThemeText %@ returned error %d", string, ( int )status );*/
         return( self );
-    }
+    /*}
     
     attrString = [[ NSAttributedString alloc ] initWithString: string ];
     
-    return( [ attrString autorelease ] );
+    return( [ attrString autorelease ] );*/
 }
 
 @end
