diff -ur src.orig/mlview-source-view.h src/mlview-source-view.h
--- src.orig/mlview-source-view.h	2013-07-11 13:11:00.000000000 -0700
+++ src/mlview-source-view.h	2013-07-11 13:12:48.000000000 -0700
@@ -53,7 +53,7 @@
 
 	enum MlViewStatus set_default_options () ;
 
-	MlViewXMLDocument* SourceView::get_document () ;
+	MlViewXMLDocument* get_document () ;
 
 	GtkUIManager * get_ui_manager () ;
 
diff -ur src.orig/mlview-tree-view.h src/mlview-tree-view.h
--- src.orig/mlview-tree-view.h	2013-07-11 13:11:00.000000000 -0700
+++ src/mlview-tree-view.h	2013-07-11 13:12:01.000000000 -0700
@@ -178,7 +178,7 @@
 
 	enum MlViewStatus build_app_edit_menu () ;
 
-	GtkDialog * TreeView::get_expand_tree_dialog ()  ;
+	GtkDialog * get_expand_tree_dialog ()  ;
 
 	void clear_completion_popup_submenus () ;
 
--- src/mlview-plugin.h.orig	2013-07-11 13:20:43.000000000 -0700
+++ src/mlview-plugin.h	2013-07-11 13:21:23.000000000 -0700
@@ -46,12 +46,12 @@
 		
 		//forbid assignation/copy
 		Plugin& operator= (Plugin const&) ;
-		Plugin::Plugin (Plugin const&) ;
+		Plugin (Plugin const&) ;
 
 	public:
 		Plugin (const PluginDescriptor &a_descr) ;
 
-		Plugin::Plugin (const UString &a_url) ;
+		Plugin (const UString &a_url) ;
 		
 		virtual ~Plugin () ;
 		
--- src/mlview-editor.cc.orig	2013-07-11 13:28:57.000000000 -0700
+++ src/mlview-editor.cc	2013-07-11 13:29:35.000000000 -0700
@@ -157,7 +157,7 @@
 			contextual_menu (NULL)
 	{}
 
-	static void EditorPriv::schemas_window_destroy_cb
+	static void schemas_window_destroy_cb
 	(GtkWidget *a_widget, struct DocumentWindowData *a_win) ;
 
 	static void validation_window_destroy_cb
