--- kccommon.h.orig	2014-09-15 00:42:51.000000000 +0400
+++ kccommon.h	2014-09-15 01:20:07.000000000 +0400
@@ -82,7 +82,7 @@
 using ::snprintf;
 }
 
-#if __cplusplus > 199711L || defined(__GXX_EXPERIMENTAL_CXX0X__) || defined(_MSC_VER)
+#if __cplusplus > 199711L || defined(__GXX_EXPERIMENTAL_CXX0X__) || defined(_MSC_VER) || (defined(__clang__) && !defined(__GLIBCXX__))
 
 #include <unordered_map>
 #include <unordered_set>
