diff -Naurp guile-1.8.8.orig/libguile/__scm.h guile-1.8.8/libguile/__scm.h
--- libguile/__scm.h	2010-12-13 09:25:01.000000000 -0800
+++ libguile/__scm.h	2012-10-29 13:54:26.000000000 -0700
@@ -141,7 +141,7 @@
 
 /* Random options (not yet supported or in final form). */
 
-#define STACK_CHECKING
+#undef STACK_CHECKING
 #undef NO_CEVAL_STACK_CHECKING
 
 
diff -Naurp guile-1.8.8.orig/libguile/debug.c guile-1.8.8/libguile/debug.c
--- libguile/debug.c	2010-12-13 09:24:39.000000000 -0800
+++ libguile/debug.c	2012-10-29 13:55:44.000000000 -0700
@@ -69,7 +69,9 @@ SCM_DEFINE (scm_debug_options, "debug-op
       SCM_OUT_OF_RANGE (1, setting);
     }
   SCM_RESET_DEBUG_MODE;
+#ifdef STACK_CHECKING
   scm_stack_checking_enabled_p = SCM_STACK_CHECKING_P;
+#endif
   scm_debug_eframe_size = 2 * SCM_N_FRAMES;
 
   scm_dynwind_end ();
