--- Modules/_hacl/Lib_Memzero0.c.orig	2025-05-07 01:33:52
+++ Modules/_hacl/Lib_Memzero0.c	2025-05-09 23:44:44
@@ -11,7 +11,7 @@
 #if defined(__APPLE__) && defined(__MACH__)
 #include <AvailabilityMacros.h>
 // memset_s is available from macOS 10.9, iOS 7, watchOS 2, and on all tvOS and visionOS versions.
-#  if (defined(MAC_OS_X_VERSION_MIN_REQUIRED) && (MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_9))
+#  if (defined(MAC_OS_X_VERSION_MIN_REQUIRED) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1090))
 #    define APPLE_HAS_MEMSET_S 1
 #  elif (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && (__IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_7_0))
 #    define APPLE_HAS_MEMSET_S 1
