--- ./cycler.py.orig	2021-10-25 14:20:29.000000000 -0700
+++ ./cycler.py	2021-11-15 14:14:07.000000000 -0800
@@ -40,6 +40,8 @@ Results in::
     {'color': 'b', 'linestyle': '-.'}
 """
 
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
 
 import copy
 from functools import reduce
@@ -101,7 +103,7 @@ def concat(left, right):
     return reduce(add, (_cycler(k, _l[k] + _r[k]) for k in left.keys))
 
 
-class Cycler:
+class Cycler(object):
     """
     Composable cycles.
 
@@ -225,7 +227,7 @@ class Cycler:
         """
         ret = cls(None)
         ret._left = list({label: v} for v in itr)
-        ret._keys = {label}
+        ret._keys = set([label])
         return ret
 
     def __getitem__(self, key):
@@ -257,7 +259,7 @@ class Cycler:
         """
         if len(self) != len(other):
             raise ValueError("Can only add equal length cycles, "
-                             f"not {len(self)} and {len(other)}")
+                             "not {0} and {1}".format(len(self), len(other)))
         return Cycler(self, other, zip)
 
     def __mul__(self, other):
@@ -341,7 +343,7 @@ class Cycler:
         if self._right is None:
             lab = self.keys.pop()
             itr = list(v[lab] for v in self)
-            return f"cycler({lab!r}, {itr!r})"
+            return "cycler({lab!r}, {itr!r})".format(lab=lab, itr=itr)
         else:
             op = op_map.get(self._op, '?')
             msg = "({left!r} {op} {right!r})"
@@ -352,11 +354,11 @@ class Cycler:
         output = "<table>"
         sorted_keys = sorted(self.keys, key=repr)
         for key in sorted_keys:
-            output += f"<th>{key!r}</th>"
+            output += "<th>{key!r}</th>".format(key=key)
         for d in iter(self):
             output += "<tr>"
             for k in sorted_keys:
-                output += f"<td>{d[k]!r}</td>"
+                output += "<td>{val!r}</td>".format(val=d[k])
             output += "</tr>"
         output += "</table>"
         return output
--- ./doc/source/conf.py.orig	2021-10-28 20:39:12.000000000 -0700
+++ ./doc/source/conf.py	2021-11-15 14:14:07.000000000 -0800
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+# -*- coding: utf-8 -*-
 #
 # cycler documentation build configuration file, created by
 # sphinx-quickstart on Wed Jul  1 13:32:53 2015.
--- ./test_cycler.py.orig	2021-10-25 14:20:29.000000000 -0700
+++ ./test_cycler.py	2021-11-15 14:14:07.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import (absolute_import, division, print_function)
+
 from collections import defaultdict
 from operator import add, iadd, mul, imul
 from itertools import product, cycle, chain
