--- src/emulate.c.orig	2011-11-15 04:03:40.000000000 +0100
+++ src/emulate.c	2015-08-30 15:34:08.000000000 +0200
@@ -695,7 +695,7 @@
   }
 }
 
-inline int
+/*inline*/ int
 #ifdef __FunctionProto__
 decode_8_thru_f(int op1)
 #else
@@ -1965,7 +1965,7 @@
   }
 }
 
-inline int
+/*inline*/ int
 #ifdef __FunctionProto__
 step_instruction(void)
 #else
@@ -2214,7 +2214,7 @@
   return stop;
 }
 
-inline void
+/*inline*/ void
 #ifdef __FunctionProto__
 schedule(void)
 #else
