--- src/posix.h.orig	2014-06-08 07:44:32.000000000 -0700
+++ src/posix.h	2014-06-11 13:33:49.000000000 -0700
@@ -101,6 +101,19 @@
 
 #include "strnlen.h"
 
+#ifdef __APPLE__
+#include <Availability.h>
+#if __MAC_OS_X_VERSION_MIN_REQUIRED <= 1060
+static size_t
+strnlen (const char *string, size_t maxlen)
+{
+    const char *end = memchr (string, '\0', maxlen);
+    return end ? (size_t) (end - string) : maxlen;
+}
+# define p_strnlen strnlen
+#endif
+#endif
+
 #ifdef NO_READDIR_R
 #	include <dirent.h>
 GIT_INLINE(int) p_readdir_r(DIR *dirp, struct dirent *entry, struct dirent **result)
