--- pm_config.in.h.orig	2012-07-25 08:48:16.000000000 -0700
+++ pm_config.in.h	2012-07-25 08:48:44.000000000 -0700
@@ -242,7 +242,7 @@
    stuck with pamflip_sse.c.  So we say the feature exists only on 4.2
    and up.
 */
-#if GCCVERSION >=402 && defined(__SSE__) && defined(__SSE2__)
+#if GCCVERSION >=402 && defined(__SSE__) && defined(__SSE2__) && !defined(__clang__)
   #define HAVE_GCC_SSE2 1
 #else
   #define HAVE_GCC_SSE2 0
