--- buildtools/makeman.orig	2017-08-19 09:26:14.000000000 -0700
+++ buildtools/makeman	2017-08-19 09:31:57.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!__MACPORTS_PYTHON__
 #
 # makeman -- compile netpbm's stereotyped HTML to troff markup
 #
--- buildtools/manpage.mk.orig	2017-08-19 09:56:44.000000000 -0700
+++ buildtools/manpage.mk	2017-08-19 10:00:29.000000000 -0700
@@ -127,11 +127,11 @@
 # to standard error.
 %.1 %.3 %.5: $(USERGUIDE)/%.html
 	@echo Converting $< to $@
-	@python $(MAKEMAN) -d $(USERGUIDE) $(<F) 
+	@__MACPORTS_PYTHON__ $(MAKEMAN) -d $(USERGUIDE) $(<F)
 
 netpbm.1: $(USERGUIDE)/index.html
 	@echo Converting $< to $@
-	@python $(MAKEMAN) -d $(USERGUIDE) index.html
+	@__MACPORTS_PYTHON__ $(MAKEMAN) -d $(USERGUIDE) index.html
 	@mv index.1 netpbm.1
 
 # Generate man pages
