--- GNUmakefile.orig	2015-04-16 22:17:30.000000000 -0500
+++ GNUmakefile	2015-06-22 23:17:31.000000000 -0500
@@ -61,7 +61,7 @@
 include $(BUILDDIR)/config.mk
 
 PROG_SUBDIRS = converter analyzer editor generator other test
-PRODUCT_SUBDIRS = lib $(PROG_SUBDIRS)
+PRODUCT_SUBDIRS = $(PROG_SUBDIRS)
 SUPPORT_SUBDIRS = urt buildtools
 
 SUBDIRS = $(PRODUCT_SUBDIRS) $(SUPPORT_SUBDIRS)
@@ -103,7 +103,7 @@
 # for the aforementioned reason for other invocations.
 
 $(SUBDIRS:%=%/all) lib/util/all: pm_config.h inttypes_netpbm.h version.h
-$(PROG_SUBDIRS:%=%/all): lib/all $(SUPPORT_SUBDIRS:%=%/all)
+$(PROG_SUBDIRS:%=%/all): $(SUPPORT_SUBDIRS:%=%/all)
 lib/all: lib/util/all
 
 .PHONY: lib/util/all
