Fix build failure on 10.6 and earlier
--- lib/libpbmfont2.c.orig	2018-10-02 09:29:55.000000000 -0500
+++ lib/libpbmfont2.c	2018-10-05 18:20:21.000000000 -0500
@@ -749,7 +749,7 @@
         pm_message("CHARSET_ENCODING in BDF font file is too long. "
                    "Truncating");
 
-    *encodingP = strndup(readlineP->arg[1], maxTokenLen);
+    *encodingP = pm_strdup(readlineP->arg[1]);
     *gotEncodingP = true;
 }
 
