--- ufraw_exiv2.cc.orig	2015-06-15 20:58:38.000000000 -0700
+++ ufraw_exiv2.cc	2018-12-22 18:59:36.000000000 -0800
@@ -67,7 +67,7 @@
         if (exifData.empty()) {
             std::string error(uf->filename);
             error += ": No Exif data found in the file";
-            throw Exiv2::Error(1, error);
+            throw Exiv2::Error(Exiv2::kerErrorMessage, error);
         }
 
         /* List of tag names taken from exiv2's printSummary() in actions.cpp */
