--- uf_gtk.cc.orig	2015-06-15 20:58:38.000000000 -0700
+++ uf_gtk.cc	2015-11-05 01:49:48.000000000 -0800
@@ -19,6 +19,7 @@
 #include <Carbon/Carbon.h>
 #include <ApplicationServices/ApplicationServices.h>
 #include <CoreServices/CoreServices.h>
+#include <Availability.h>
 #endif
 
 #ifdef G_OS_WIN32
@@ -163,6 +164,15 @@
         g_free(atom_name);
 
 #elif defined GDK_WINDOWING_QUARTZ
+#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
+//
+// ColorSync Manager API, deprecated in Mac OS X 10.6,
+// was finally removed in Mac OS X 10.11 (El Capitan).
+//
+// If 10.11 or later just return with no profile.
+//
+        return;
+#else
         GdkScreen *screen = gtk_widget_get_screen(widget);
         if (screen == NULL)
             screen = gdk_screen_get_default();
@@ -184,7 +194,7 @@
         *buffer_size = CFDataGetLength(data);
 
         CFRelease(data);
-
+#endif
 #elif defined G_OS_WIN32
         (void)widget;
         HDC hdc = GetDC(NULL);
