--- dcraw.cc.orig	2016-10-26 11:00:19.000000000 -0700
+++ dcraw.cc	2016-10-26 12:20:29.000000000 -0700
@@ -2404,7 +2404,7 @@
 #endif
   cinfo->src->next_input_byte = jpeg_buffer;
   cinfo->src->bytes_in_buffer = nbytes;
-  return TRUE;
+  return (boolean) TRUE;
 }
 
 void CLASS kodak_jpeg_load_raw()
@@ -2420,7 +2420,7 @@
   jpeg_create_decompress (&cinfo);
   jpeg_stdio_src (&cinfo, ifp);
   cinfo.src->fill_input_buffer = fill_input_buffer;
-  jpeg_read_header (&cinfo, TRUE);
+  jpeg_read_header (&cinfo, (boolean) TRUE);
   jpeg_start_decompress (&cinfo);
   if ((cinfo.output_width      != width  ) ||
       (cinfo.output_height*2   != height ) ||
@@ -2493,7 +2493,7 @@
     if (tile_length < INT_MAX)
       fseek (ifp, get4(), SEEK_SET);
     jpeg_stdio_src (&cinfo, ifp);
-    jpeg_read_header (&cinfo, TRUE);
+    jpeg_read_header (&cinfo, (boolean) TRUE);
     jpeg_start_decompress (&cinfo);
     buf = (*cinfo.mem->alloc_sarray)
 	((j_common_ptr) &cinfo, JPOOL_IMAGE, cinfo.output_width*3, 1);
@@ -9256,7 +9256,7 @@ canon_a5:
     if (make[0] == 'O') {
       i = find_green (12, 32, 1188864, 3576832);
       c = find_green (12, 32, 2383920, 2387016);
-      if (abs(i) < abs(c)) {
+      if (i < c) {
 	SWAP(i,c);
 	load_flags = 24;
       }
