--- ufraw-gimp.c.orig	2014-01-16 10:32:12.000000000 -0800
+++ ufraw-gimp.c	2014-01-16 10:38:12.000000000 -0800
@@ -260,7 +260,10 @@
         uf->conf->embeddedImage = TRUE;
     }
 #else
-    if (run_mode == GIMP_RUN_NONINTERACTIVE) uf->conf->shrink = 8;
+/*   this loads images at 1/8 size in non-interactive mode
+ *   which causes problems with gimp-gap's processing of
+ *   videos created from RAW frame sequences
+ *   if (run_mode == GIMP_RUN_NONINTERACTIVE) uf->conf->shrink = 8; */
 #endif
     /* UFRaw already issues warnings.
      * With GIMP_PDB_CANCEL, Gimp won't issue another one. */
