--- dcraw.cc.orig	2015-07-05 19:35:57.000000000 -0700
+++ dcraw.cc	2015-07-05 19:38:28.000000000 -0700
@@ -41,6 +41,20 @@ extern "C" {
 #include <time.h>
 #include <sys/types.h>
 
+extern "C" {
+static size_t
+strnlen(const char *s, size_t maxlen)
+{
+	size_t len;
+
+	for (len = 0; len < maxlen; len++, s++) {
+		if (!*s)
+			break;
+	}
+	return (len);
+}
+}
+
 #ifndef HAVE_CONFIG_H /*fseeko() is handled by the UFRaw config system - NKBJ*/
 #if defined(DJGPP) || defined(__MINGW32__)
 #define fseeko fseek
