http://bugs.mysql.com/bug.php?id=55846
--- include/Makefile.in.orig	2010-08-03 12:28:49.000000000 -0500
+++ include/Makefile.in	2010-09-15 09:19:54.000000000 -0500
@@ -405,7 +405,8 @@
 			decimal.h errmsg.h my_global.h my_net.h \
 			my_getopt.h sslopt-longopts.h my_dir.h \
 			sslopt-vars.h sslopt-case.h sql_common.h keycache.h \
-			m_ctype.h my_attribute.h $(HEADERS_GEN_CONFIGURE) \
+			m_ctype.h my_attribute.h my_compiler.h \
+			$(HEADERS_GEN_CONFIGURE) \
 			$(HEADERS_GEN_MAKE)
 
 noinst_HEADERS = config-win.h config-netware.h my_bit.h \
@@ -416,7 +417,7 @@
 			my_aes.h my_tree.h my_trie.h hash.h thr_alarm.h \
 			thr_lock.h t_ctype.h violite.h my_md5.h base64.h \
 			my_handler.h my_time.h my_vle.h my_user.h \
-			my_libwrap.h my_stacktrace.h my_compiler.h
+			my_libwrap.h my_stacktrace.h
 
 EXTRA_DIST = mysql.h.pp mysql/plugin.h.pp
 
