--- CONFIG/src/atlconf_misc.c.orig	2012-05-18 13:20:52.000000000 +0200
+++ CONFIG/src/atlconf_misc.c	2012-05-18 13:21:43.000000000 +0200
@@ -715,6 +715,7 @@
  *    match with pgcc
  */
       if (strstr(res, "(GCC)") || strstr(res, " GCC ") ||
+		  strstr(res, "gcc-mp") ||
           strstr(res, "GNU Fortran") || strstr(res, "(GCC ") ||
           (res[0] == 'g' && res[1] == 'c' && res[2] == 'c' && res[3] == ' ')
          )
