--- build/tune/blas/gemv/ATL_dmvnk.c.orig	2011-07-24 13:50:01.000000000 +0200
+++ build/tune/blas/gemv/ATL_dmvnk.c	2011-07-24 13:50:29.000000000 +0200
@@ -109,7 +109,7 @@
    movl   $3*8, -56(%rsp)       /* mem = MU */
    fildl  -56(%rsp)                     /* ST = MU */
    movl   %edi, -56(%rsp)               /* mem = M */
-   fidivr -56(%rsp)                     /* ST = M/MU */
+   fidivrl -56(%rsp)                     /* ST = M/MU */
    fisttpl -60(%rsp)                    /* mem = TRUNC(M/MU) */
    movl    -60(%rsp), %ebx              /* rbx = TRUNC(M/MU) */
    imul    $3*8, %ebx, %ebx     /* rbx = MU*TRUNC(M/MU) */
