--- include/atlas_mmparse.h.orig	2011-08-17 22:35:24.000000000 +0200
+++ include/atlas_mmparse.h	2011-08-17 22:35:57.000000000 +0200
@@ -243,17 +243,6 @@
  *  use an old GOODGCC that lacks something like -pg.
  */
 {
-   ATL_mmnode_t *kp;  /* queue to make sub in */
-   char *gcc;
-   gcc = GetGoodGcc();
-   for (kp=bp; kp; kp = kp->next)
-   {
-      if (kp->comp && !strcmp(kp->comp, gcc))
-      {
-         free(kp->comp);
-	 kp->comp = DupString("gcc");
-      }
-   }
 }
 
 /* procedure 11 */
@@ -266,17 +255,6 @@
  * to help portability
  */
 {
-   ATL_mmnode_t *kp;  /* queue to make sub in */
-   char *gcc;
-   gcc = GetGoodGcc();
-   for (kp=bp; kp; kp = kp->next)
-   {
-      if (kp->comp && !strcmp(kp->comp, gcc))
-      {
-         free(kp->comp);
-	 kp->comp = DupString("gcc");
-      }
-   }
 }
 
 /* procedure 12 */
