diff -ur ../kdepimlibs-4.13.2-orig/CMakeLists.txt ./CMakeLists.txt
--- ../kdepimlibs-4.13.2-orig/CMakeLists.txt	2014-06-06 07:12:42.000000000 +0900
+++ ./CMakeLists.txt	2014-06-13 02:39:46.000000000 +0900
@@ -138,47 +138,15 @@
 
 ############### Now, we add the KDEPIMLibs components ###############
 # These targets will always be built
-add_subdirectory(cmake)
-add_subdirectory(gpgme++)
-add_subdirectory(qgpgme)
 
 if (NOT KDEPIM_ONLY_KLEO)
-  add_subdirectory(kabc)
-  add_subdirectory(kalarmcal)
-  if (NOT KDEPIM_NO_KCAL)
-    add_subdirectory(kblog)
-    add_subdirectory(kcal)
-  endif()
-  add_subdirectory(kcalcore)
-  add_subdirectory(kcalutils)
-  add_subdirectory(akonadi) # depends on kabc is build
-  add_subdirectory(kholidays)
   add_subdirectory(kimap)
   add_subdirectory(kioslave)
   add_subdirectory(kldap)
-  add_subdirectory(kmbox)
 endif()
   add_subdirectory(kmime)
-if (NOT KDEPIM_ONLY_KLEO)
-  add_subdirectory(kpimidentities)
-  add_subdirectory(kpimutils)
-  add_subdirectory(kpimtextedit)
-  if (NOT KDEPIM_NO_KRESOURCES)
-    add_subdirectory(kresources)
-  endif()
-  add_subdirectory(ktnef)
-  add_subdirectory(kxmlrpcclient)
-  add_subdirectory(mailtransport)
-  add_subdirectory(microblog)
-  add_subdirectory(syndication)
-  add_subdirectory(kontactinterface)
-  # Build the CamelCase headers
-  add_subdirectory(includes)
-endif()
-
 
 # doc must be a subdir of kdepimlibs
-macro_optional_add_subdirectory(doc)
 
 # All done, let's display what we found...
 feature_summary(WHAT ALL
@@ -259,18 +227,3 @@
    set(_KdepimLibsConfig_INSTALL_DIR ${LIB_INSTALL_DIR}/cmake/KdepimLibs)
 endif()
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/KdepimLibsConfigVersion.cmake
-              ${CMAKE_CURRENT_BINARY_DIR}/KdepimLibsConfig.cmake
-        DESTINATION ${_KdepimLibsConfig_INSTALL_DIR} )
-
-# Install the file with the exported targets, use ${KDEPIMLIBS_TARGET_PREFIX} as prefix for the names of these targets, Alex
-install(EXPORT kdepimlibsLibraryTargets
-        NAMESPACE ${KDEPIMLIBS_TARGET_PREFIX}
-        DESTINATION ${_KdepimLibsConfig_INSTALL_DIR}
-        FILE KDEPimLibsLibraryTargetsWithPrefix.cmake )
-
-# Install a KDEPimLibsDependencies.cmake so people using kdepimlibs 4.2 with kdelibs < 4.2 get a useful error message, Alex
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/KDEPimLibsDependencies.cmake "\n message(FATAL_ERROR \"For using this version of kdepimlibs (${KDEPIMLIBS_VERSION}) you need a newer version of kdelibs, please update.\")\n")
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/KDEPimLibsDependencies.cmake
-        DESTINATION ${DATA_INSTALL_DIR}/cmake/modules)
-
diff -ur ../kdepimlibs-4.13.2-orig/kimap/CMakeLists.txt ./kimap/CMakeLists.txt
--- ../kdepimlibs-4.13.2-orig/kimap/CMakeLists.txt	2014-06-06 07:12:42.000000000 +0900
+++ ./kimap/CMakeLists.txt	2014-06-13 02:39:13.000000000 +0900
@@ -2,7 +2,6 @@
 
 add_definitions( -DKDE_DEFAULT_DEBUG_AREA=5327 )
 
-add_subdirectory( tests )
 add_definitions( -DQT_NO_CAST_FROM_ASCII )
 add_definitions( -DQT_NO_CAST_TO_ASCII )
 
@@ -65,12 +64,6 @@
 )
 
 set_target_properties(kimap PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
-install(TARGETS kimap EXPORT kdepimlibsLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
-install( FILES ${CMAKE_CURRENT_BINARY_DIR}/kimap_export.h acl.h imapset.h job.h appendjob.h capabilitiesjob.h fetchjob.h idlejob.h listjob.h loginjob.h logoutjob.h namespacejob.h rfccodecs.h
- selectjob.h closejob.h expungejob.h deletejob.h createjob.h subscribejob.h unsubscribejob.h renamejob.h
- session.h sessionuiproxy.h storejob.h copyjob.h searchjob.h acljobbase.h setacljob.h getacljob.h deleteacljob.h
- myrightsjob.h listrightsjob.h quotajobbase.h setquotajob.h getquotajob.h getquotarootjob.h metadatajobbase.h setmetadatajob.h getmetadatajob.h
- DESTINATION ${INCLUDE_INSTALL_DIR}/kimap  COMPONENT Devel)
diff -ur ../kdepimlibs-4.13.2-orig/kldap/CMakeLists.txt ./kldap/CMakeLists.txt
--- ../kdepimlibs-4.13.2-orig/kldap/CMakeLists.txt	2014-06-06 07:12:42.000000000 +0900
+++ ./kldap/CMakeLists.txt	2014-06-13 02:37:51.000000000 +0900
@@ -85,26 +85,6 @@
   VERSION ${GENERIC_LIB_VERSION}
   SOVERSION ${GENERIC_LIB_SOVERSION}
 )
-install(TARGETS kldap EXPORT kdepimlibsLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
-install(FILES
-  ber.h
-  ${CMAKE_CURRENT_BINARY_DIR}/kldap_export.h
-  ldapattributeproxymodel.h
-  ldapconfigwidget.h
-  ldapconnection.h
-  ldapcontrol.h
-  ldapdefs.h
-  ldapdn.h
-  ldapmodel.h
-  ldapobject.h
-  ldapoperation.h
-  ldapsearch.h
-  ldapserver.h
-  ldapstructureproxymodel.h
-  ldapurl.h
-  ldif.h
-  DESTINATION ${INCLUDE_INSTALL_DIR}/kldap COMPONENT Devel
-)
diff -ur ../kdepimlibs-4.13.2-orig/kmime/CMakeLists.txt ./kmime/CMakeLists.txt
--- ../kdepimlibs-4.13.2-orig/kmime/CMakeLists.txt	2014-06-06 07:12:42.000000000 +0900
+++ ./kmime/CMakeLists.txt	2014-06-13 02:38:49.000000000 +0900
@@ -38,21 +38,3 @@
 target_link_libraries(kmime  ${KDE4_KDECORE_LIBS} )
 
 set_target_properties(kmime PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-
-install(TARGETS kmime EXPORT kdepimlibsLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
-
-install( FILES
-         boolflags.h
-         ${CMAKE_CURRENT_BINARY_DIR}/kmime_export.h
-         kmime_charfreq.h
-         kmime_codecs.h
-         kmime_content.h
-         kmime_contentindex.h
-         kmime_header_parsing.h
-         kmime_headers.h
-         kmime_message.h
-         kmime_mdn.h
-         kmime_newsarticle.h
-         kmime_dateformatter.h
-         kmime_util.h
-         DESTINATION ${INCLUDE_INSTALL_DIR}/kmime COMPONENT Devel)
