--- src/Makefile.in.orig	2003-09-17 07:55:56.000000000 -0600
+++ src/Makefile.in	2010-05-03 00:46:00.000000000 -0600
@@ -16,9 +16,9 @@
 SHELL = /bin/sh
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 CC = @CC@
-CFLAGS = @CFLAGS@ -Wall
+CFLAGS = @CFLAGS@ -Wall @CPPFLAGS@
 DEFS = @DEFS@ -DLOCALEDIR=\"$(localedir)\"
-LIBS = @LIBS@
+LIBS = @LIBS@ @LDFLAGS@ -lintl
 
 objects = util.o desproxy.o desproxy-inetd.o desproxy-socksserver.o \
           desproxy-dns.o socket2socket.o desproxy-test.o
