--- Makefile.orig	2020-07-28 11:27:54.000000000 -0600
+++ Makefile	2020-07-28 11:29:39.000000000 -0600
@@ -11,14 +11,14 @@
 
 # Linux
 #
-DEFINES	= -DPNG
-INCLUDES= 
-LIB_DIRS= 
-NCH_DEP	= /usr/include/netcdf.h
-LIBS    = -lXm -lXt -lX11 -lnetcdf -lhdf5 -lhdf5_hl -lfl -lgsl -lgslcblas -lpng -lpthread
-BIN	= ${JLOCAL}/bin
-ANONFTP	= /net/ftp/pub/archive/RAF-src/bin.RHEL7
-
+#DEFINES	= -DPNG
+#INCLUDES= 
+#LIB_DIRS= 
+#NCH_DEP	= /usr/include/netcdf.h
+#LIBS    = -lXm -lXt -lX11 -lnetcdf -lhdf5 -lhdf5_hl -lfl -lgsl -lgslcblas -lpng -lpthread
+#BIN	= ${JLOCAL}/bin
+#ANONFTP	= /net/ftp/pub/archive/RAF-src/bin.RHEL7
+#
 # Mac OS X
 #
 # Requires XCode from Apple App Store.  Don't forget to run Xcode after it is
@@ -34,12 +34,12 @@
 #  brew install gmt  ; for geo-politcal maps.
 #  brew install whatever else
 #
-#DEFINES	= -DPNG
-#INCLUDES= -I/opt/X11/include
-#LIB_DIRS= -L/opt/X11/lib
-#NCH_DEP	= /usr/local/include/netcdf.h
-#LIBS    = -lXm -lXt -lX11 -lnetcdf -ll -lgsl -lpng -lpthread
-#BIN	= /usr/local/bin
+DEFINES	= -DPNG
+INCLUDES= 
+LIB_DIRS= 
+NCH_DEP	= @@PREFIX@@/include/netcdf.h
+LIBS    = -lXm -lXt -lX11 -lnetcdf -ll -lgsl -lpng -lpthread
+BIN	= @@PREFIX@@/bin
 
 CFLAGS	= -Wall -g -O2 ${INCLUDES} ${DEFINES} -Wno-write-strings -Wno-overflow
 
