--- Makefile.orig	2022-02-02 12:46:06.000000000 -0700
+++ Makefile	2022-02-02 12:47:09.000000000 -0700
@@ -11,13 +11,13 @@
 # yum install flex-devel netcdf-devel motif-devel gsl-devel \
 #    xorg-x11-fonts-ISO8859-1-75dpi xorg-x11-fonts-ISO8859-1-100dpi
 #
-DEFINES	= -DPNG
-INCLUDES= 
-LIB_DIRS= 
-NCH_DEP	= /usr/include/netcdf.h
-LIBS    = -lXm -lXt -lX11 -lnetcdf -lhdf5 -lhdf5_hl -lfl -lgsl -lgslcblas -lpng -lpthread
-BIN	= ${JLOCAL}/bin
-WWW	= /net/www/docs/raf/Software
+#DEFINES	= -DPNG
+#INCLUDES= 
+#LIB_DIRS= 
+#NCH_DEP	= /usr/include/netcdf.h
+#LIBS    = -lXm -lXt -lX11 -lnetcdf -lhdf5 -lhdf5_hl -lfl -lgsl -lgslcblas -lpng -lpthread
+#BIN	= ${JLOCAL}/bin
+#WWW	= /net/www/docs/raf/Software
 
 
 # Linux Ubuntu
@@ -41,9 +41,9 @@
 #  brew install gmt  ; for geo-politcal maps.
 #  brew install whatever else
 #
-#DEFINES	= -DPNG
-#LIBS    = -lXm -lXt -lX11 -lnetcdf -ll -lgsl -lpng -lpthread
-#BIN	= /usr/local/bin
+DEFINES	= -DPNG
+LIBS    = -lXm -lXt -lX11 -lnetcdf -ll -lgsl -lpng -lpthread
+BIN	= @@PREFIX@@/bin
 
 # Homebrew
 #INCLUDES=
@@ -51,12 +51,12 @@
 #NCH_DEP	= /usr/local/include/netcdf.h
 
 # MacPorts
-#INCLUDES= -I/opt/local/include
-#LIB_DIRS= -L/opt/local/lib
-#NCH_DEP	= /opt/local/include/netcdf.h
+INCLUDES= -I@@PREFIX@@/include
+LIB_DIRS= -L@@PREFIX@@/lib
+NCH_DEP	= @@PREFIX@@/include/netcdf.h
 
 
-CXXFLAGS	= -Wall -g -O2 ${INCLUDES} ${DEFINES} -Wno-write-strings -Wno-overflow
+CXXFLAGS	= -Wall -g -Os ${INCLUDES} ${DEFINES} -Wno-write-strings -Wno-overflow
 
 PROG	= ncplot
 HDRS	= define.h extern.h
