--- CMakeLists.txt.orig	2014-05-13 23:42:49.000000000 +0900
+++ CMakeLists.txt	2014-06-14 00:59:48.000000000 +0900
@@ -8,8 +8,8 @@
 macro_optional_find_package(Analitza)
 macro_log_feature(Analitza_FOUND "libanalitza" "A library provided by KAlgebra" FALSE "" "Backend to use KAlgebra with Cantor.")
 
-macro_optional_find_package(R)
-macro_log_feature(R_FOUND "R" "A free software environment for statistical computing and graphics" "http://www.r-project.org/" FALSE "" "Backend to use R with Cantor.")
+#macro_optional_find_package(R)
+#macro_log_feature(R_FOUND "R" "A free software environment for statistical computing and graphics" "http://www.r-project.org/" FALSE "" "Backend to use R with Cantor.")
 
 if(NOT WIN32)
   macro_optional_find_package(LibSpectre)
