--- wscript.orig	2014-03-12 08:19:28.000000000 -0700
+++ wscript	2014-05-05 01:32:00.000000000 -0700
@@ -103,8 +103,6 @@
         ctx.env['cshlib_PATTERN'] = 'lib%s.dll'
 
     if target_platform == 'darwin':
-        ctx.env.CFLAGS += ['-arch', 'i386', '-arch', 'x86_64']
-        ctx.env.LINKFLAGS += ['-arch', 'i386', '-arch', 'x86_64']
         ctx.env.FRAMEWORK = ['CoreFoundation', 'AudioToolbox', 'Accelerate']
         ctx.define('HAVE_ACCELERATE', 1)
 
