diff -ur python/aubio.orig/bench/onset.py python/aubio/bench/onset.py
--- python/aubio.orig/bench/onset.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/bench/onset.py	2011-10-31 10:54:20.000000000 -0700
@@ -111,7 +111,7 @@
 		for i in self.vlist:
 			gd.append(i['GD']) 
 			fp.append(i['FP']) 
-		d.append(Gnuplot.Data(fp, gd, with='linespoints', 
+		d.append(Gnuplot.Data(fp, gd, _with='linespoints', 
 			title="%s %s" % (plottitle,i['mode']) ))
 
 	def plotplotroc(self,d,outplot=0,extension='ps'):
@@ -147,7 +147,7 @@
 		for i in self.vlist:
 			x.append(i['prec']) 
 			y.append(i['recl']) 
-		d.append(Gnuplot.Data(x, y, with='linespoints', 
+		d.append(Gnuplot.Data(x, y, _with='linespoints', 
 			title="%s %s" % (plottitle,i['mode']) ))
 
 	def plotplotpr(self,d,outplot=0,extension='ps'):
@@ -172,7 +172,7 @@
 		for i in self.vlist:
 			x.append(i['thres']) 
 			y.append(i['dist']) 
-		d.append(Gnuplot.Data(x, y, with='linespoints', 
+		d.append(Gnuplot.Data(x, y, _with='linespoints', 
 			title="%s %s" % (plottitle,i['mode']) ))
 
 	def plotplotfmeas(self,d,outplot="",extension='ps', title="F-measure"):
@@ -205,7 +205,7 @@
 		for i in self.vlist:
 			x.append(i[var]) 
 			y.append(i['dist']) 
-		d.append(Gnuplot.Data(x, y, with='linespoints', 
+		d.append(Gnuplot.Data(x, y, _with='linespoints', 
 			title="%s %s" % (plottitle,i['mode']) ))
 	
 	def plotplotfmeasvar(self,d,var,outplot="",extension='ps', title="F-measure"):
@@ -244,7 +244,7 @@
 		total = v['Torig']
 		for i in range(len(per)): per[i] /= total/100.
 
-		d.append(Gnuplot.Data(val, per, with='fsteps', 
+		d.append(Gnuplot.Data(val, per, _with='fsteps', 
 			title="%s %s" % (plottitle,v['mode']) ))
 		#d.append('mean=%f,sigma=%f,eps(x) title \"\"'% (mean,smean))
 		#d.append('mean=%f,sigma=%f,eps(x) title \"\"'% (amean,samean))
@@ -275,7 +275,7 @@
 		total = v['Torig']
 		for i in range(len(per)): per[i] /= total/100.
 
-		d.append(Gnuplot.Data(val, per, with='fsteps', 
+		d.append(Gnuplot.Data(val, per, _with='fsteps', 
 			title="%s %s" % (plottitle,v['mode']) ))
 		#d.append('mean=%f,sigma=%f,eps(x) title \"\"'% (mean,smean))
 		#d.append('mean=%f,sigma=%f,eps(x) title \"\"'% (amean,samean))
diff -ur python/aubio.orig/gnuplot.py python/aubio/gnuplot.py
--- python/aubio.orig/gnuplot.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/gnuplot.py	2011-10-31 10:55:35.000000000 -0700
@@ -155,7 +155,7 @@
   """ create gnuplot plot from an audio file """
   import Gnuplot, Gnuplot.funcutils
   x,y = downsample_audio(time,data,maxpoints=maxpoints)
-  return Gnuplot.Data(x,y,with='lines')
+  return Gnuplot.Data(x,y,_with='lines')
 
 def make_audio_envelope(time,data,maxpoints=10000):
   """ create gnuplot plot from an audio file """
@@ -165,7 +165,7 @@
   x = [i.mean() for i in numarray.array(time).resize(len(time)/bufsize,bufsize)] 
   y = [i.mean() for i in numarray.array(data).resize(len(time)/bufsize,bufsize)] 
   x,y = downsample_audio(x,y,maxpoints=maxpoints)
-  return Gnuplot.Data(x,y,with='lines')
+  return Gnuplot.Data(x,y,_with='lines')
 
 def gnuplot_addargs(parser):
   """ add common gnuplot argument to OptParser object """
diff -ur python/aubio.orig/task/beat.py python/aubio/task/beat.py
--- python/aubio.orig/task/beat.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/task/beat.py	2011-10-31 10:56:25.000000000 -0700
@@ -247,7 +247,7 @@
 
 	def plot(self,oplots,results):
 		import Gnuplot
-		oplots.append(Gnuplot.Data(results,with='linespoints',title="auto"))
+		oplots.append(Gnuplot.Data(results,_with='linespoints',title="auto"))
 
 	def plotplot(self,wplot,oplots,outplot=None,extension=None,xsize=1.,ysize=1.,spectro=False):
 		import Gnuplot
@@ -258,5 +258,5 @@
 		#f = make_audio_plot(time,data)
 
 		g = gnuplot_create(outplot=outplot, extension=extension)
-		oplots = [Gnuplot.Data(self.gettruth(),with='linespoints',title="orig")] + oplots
+		oplots = [Gnuplot.Data(self.gettruth(),_with='linespoints',title="orig")] + oplots
 		g.plot(*oplots)
diff -ur python/aubio.orig/task/notes.py python/aubio/task/notes.py
--- python/aubio.orig/task/notes.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/task/notes.py	2011-10-31 10:57:15.000000000 -0700
@@ -95,15 +95,15 @@
 		import numarray
 		import Gnuplot
 
-		oplots.append(Gnuplot.Data(now,freq,with='lines',
+		oplots.append(Gnuplot.Data(now,freq,_with='lines',
 			title=self.params.pitchmode))
-		oplots.append(Gnuplot.Data(now,ifreq,with='lines',
+		oplots.append(Gnuplot.Data(now,ifreq,_with='lines',
 			title=self.params.pitchmode))
 
 		temponsets = []
 		for i in onset:
 			temponsets.append(i*1000)
-		oplots.append(Gnuplot.Data(now,temponsets,with='impulses',
+		oplots.append(Gnuplot.Data(now,temponsets,_with='impulses',
 			title=self.params.pitchmode))
 
 	def plotplot(self,wplot,oplots,outplot=None,multiplot = 0):
@@ -117,10 +117,10 @@
 		# check if ground truth exists
 		#timet,pitcht = self.gettruth()
 		#if timet and pitcht:
-		#	oplots = [Gnuplot.Data(timet,pitcht,with='lines',
+		#	oplots = [Gnuplot.Data(timet,pitcht,_with='lines',
 		#		title='ground truth')] + oplots
 
-		t = Gnuplot.Data(0,0,with='impulses') 
+		t = Gnuplot.Data(0,0,_with='impulses') 
 
 		g = gnuplot_init(outplot)
 		g('set title \'%s\'' % (re.sub('.*/','',self.input)))
diff -ur python/aubio.orig/task/onset.py python/aubio/task/onset.py
--- python/aubio.orig/task/onset.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/task/onset.py	2011-10-31 10:57:49.000000000 -0700
@@ -103,7 +103,7 @@
 		self.maxofunc = max(ofunc)
 		# onset detection function 
 		downtime = numarray.arange(len(ofunc))*self.params.step
-		oplot.append(Gnuplot.Data(downtime,ofunc,with='lines',title=self.params.onsetmode))
+		oplot.append(Gnuplot.Data(downtime,ofunc,_with='lines',title=self.params.onsetmode))
 
 		# detected onsets
 		if not nplot:
@@ -114,8 +114,8 @@
 			#x1 = numarray.array(onsets)*self.params.step
 			#y1 = self.maxofunc*numarray.ones(len(onsets))
 			if x1:
-				oplot.append(Gnuplot.Data(x1,y1,with='impulses'))
-				wplot.append(Gnuplot.Data(x1,y1p,with='impulses'))
+				oplot.append(Gnuplot.Data(x1,y1,_with='impulses'))
+				wplot.append(Gnuplot.Data(x1,y1p,_with='impulses'))
 
 		oplots.append((oplot,self.params.onsetmode,self.maxofunc))
 
@@ -128,7 +128,7 @@
 			t_onsets = aubio.txtfile.read_datafile(datafile)
 			x2 = numarray.array(t_onsets).resize(len(t_onsets))
 			y2 = self.maxofunc*numarray.ones(len(t_onsets))
-			wplot.append(Gnuplot.Data(x2,y2,with='impulses'))
+			wplot.append(Gnuplot.Data(x2,y2,_with='impulses'))
 			
 			tol = 0.050 
 
diff -ur python/aubio.orig/task/pitch.py python/aubio/task/pitch.py
--- python/aubio.orig/task/pitch.py	2011-10-31 10:52:53.000000000 -0700
+++ python/aubio/task/pitch.py	2011-10-31 10:59:25.000000000 -0700
@@ -156,7 +156,7 @@
 
 		time = [ (i+self.params.pitchdelay)*self.params.step for i in range(len(pitch)) ]
 		pitch = [aubio_freqtomidi(i) for i in pitch]
-		oplots.append(Gnuplot.Data(time,pitch,with='lines',
+		oplots.append(Gnuplot.Data(time,pitch,_with='lines',
 			title=self.params.pitchmode))
 		titles.append(self.params.pitchmode)
 
@@ -170,7 +170,7 @@
 		if truth:
 			timet,pitcht = self.gettruth()
 			if timet and pitcht:
-				oplots = [Gnuplot.Data(timet,pitcht,with='lines',
+				oplots = [Gnuplot.Data(timet,pitcht,_with='lines',
 					title='ground truth')] + oplots
 
 		g = gnuplot_create(outplot=outplot, extension=extension)
