From 986aa6c080c564464b2d53a228142e8eeaad53bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lawrence=20Vel=C3=A1zquez?= <vq@larryv.me>
Date: Tue, 1 Jun 2021 19:49:05 -0400
Subject: [PATCH 5/6] Build verbosely

---
 icmake/bootstrap/functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/icmake/bootstrap/functions b/icmake/bootstrap/functions
index 3b1e7d7..1de053b 100644
--- a/icmake/bootstrap/functions
+++ b/icmake/bootstrap/functions
@@ -2,7 +2,7 @@
 
 try()
 {
-    # echo $*
+    printf '%s\n' "$*" >&2
     $* || exit 1
 }
 
-- 
2.31.1

