--- ligo/lw/tokenizer.RowBuilder.c.orig
+++ ligo/lw/tokenizer.RowBuilder.c
@@ -228,7 +228,7 @@


 PyTypeObject ligolw_RowBuilder_Type = {
-	PyObject_HEAD_INIT(NULL)
+	PyVarObject_HEAD_INIT(NULL, 0)
 	.tp_basicsize = sizeof(ligolw_RowBuilder),
 	.tp_dealloc = __del__,
 	.tp_doc =
--- ligo/lw/tokenizer.RowDumper.c.orig
+++ ligo/lw/tokenizer.RowDumper.c
@@ -280,7 +280,7 @@


 PyTypeObject ligolw_RowDumper_Type = {
-	PyObject_HEAD_INIT(NULL)
+	PyVarObject_HEAD_INIT(NULL, 0)
 	.tp_basicsize = sizeof(ligolw_RowDumper),
 	.tp_dealloc = __del__,
 	.tp_doc =
--- ligo/lw/tokenizer.Tokenizer.c.orig
+++ ligo/lw/tokenizer.Tokenizer.c
@@ -738,7 +738,7 @@


 PyTypeObject ligolw_Tokenizer_Type = {
-	PyObject_HEAD_INIT(NULL)
+	PyVarObject_HEAD_INIT(NULL, 0)
 	.tp_basicsize = sizeof(ligolw_Tokenizer),
 	.tp_dealloc = __del__,
 	.tp_doc =
