--- src/CMakeLists.txt	2022-10-23 10:20:16.000000000 +0800
+++ src/CMakeLists.txt	2024-01-05 21:22:45.000000000 +0800
@@ -39,7 +39,6 @@
 else(CMAKE_C_COMPILER_ID MATCHES "GNU|Clang")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c89")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wconversion")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wpedantic")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wextra")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 endif()
