--- platform/osx/os_osx.mm.orig	2021-05-24 07:39:13.000000000 -0400
+++ platform/osx/os_osx.mm	2021-11-08 16:09:55.000000000 -0500
@@ -48,9 +48,6 @@
 #include <IOKit/IOKitLib.h>
 #include <IOKit/hid/IOHIDKeys.h>
 #include <IOKit/hid/IOHIDLib.h>
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
-#include <os/log.h>
-#endif
 
 #include <dlfcn.h>
 #include <fcntl.h>
@@ -493,11 +490,7 @@
 	trackingArea = nil;
 	imeInputEventInProgress = false;
 	[self updateTrackingAreas];
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
-	[self registerForDraggedTypes:[NSArray arrayWithObject:NSPasteboardTypeFileURL]];
-#else
 	[self registerForDraggedTypes:[NSArray arrayWithObject:NSFilenamesPboardType]];
-#endif
 	markedText = [[NSMutableAttributedString alloc] init];
 	return self;
 }
@@ -643,18 +636,6 @@
 	Vector<String> files;
 	NSPasteboard *pboard = [sender draggingPasteboard];
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
-	NSArray *items = pboard.pasteboardItems;
-	for (NSPasteboardItem *item in items) {
-		NSString *path = [item stringForType:NSPasteboardTypeFileURL];
-		NSString *ns = [NSURL URLWithString:path].path;
-		char *utfs = strdup([ns UTF8String]);
-		String ret;
-		ret.parse_utf8(utfs);
-		free(utfs);
-		files.push_back(ret);
-	}
-#else
 	NSArray *filenames = [pboard propertyListForType:NSFilenamesPboardType];
 	for (NSString *ns in filenames) {
 		char *utfs = strdup([ns UTF8String]);
@@ -663,7 +644,6 @@
 		free(utfs);
 		files.push_back(ret);
 	}
-#endif
 
 	if (files.size()) {
 		OS_OSX::singleton->main_loop->drop_files(files, 0);
@@ -1837,69 +1817,7 @@
 	return "OSX";
 }
 
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
-class OSXTerminalLogger : public StdLogger {
-public:
-	virtual void log_error(const char *p_function, const char *p_file, int p_line, const char *p_code, const char *p_rationale, ErrorType p_type = ERR_ERROR) {
-		if (!should_log(true)) {
-			return;
-		}
-
-		const char *err_details;
-		if (p_rationale && p_rationale[0])
-			err_details = p_rationale;
-		else
-			err_details = p_code;
-
-		switch (p_type) {
-			case ERR_WARNING:
-				if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_12) {
-					os_log_info(OS_LOG_DEFAULT,
-							"WARNING: %{public}s: %{public}s\nAt: %{public}s:%i.",
-							p_function, err_details, p_file, p_line);
-				}
-				logf_error("\E[1;33mWARNING: %s: \E[0m\E[1m%s\n", p_function,
-						err_details);
-				logf_error("\E[0;33m   At: %s:%i.\E[0m\n", p_file, p_line);
-				break;
-			case ERR_SCRIPT:
-				if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_12) {
-					os_log_error(OS_LOG_DEFAULT,
-							"SCRIPT ERROR: %{public}s: %{public}s\nAt: %{public}s:%i.",
-							p_function, err_details, p_file, p_line);
-				}
-				logf_error("\E[1;35mSCRIPT ERROR: %s: \E[0m\E[1m%s\n", p_function,
-						err_details);
-				logf_error("\E[0;35m   At: %s:%i.\E[0m\n", p_file, p_line);
-				break;
-			case ERR_SHADER:
-				if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_12) {
-					os_log_error(OS_LOG_DEFAULT,
-							"SHADER ERROR: %{public}s: %{public}s\nAt: %{public}s:%i.",
-							p_function, err_details, p_file, p_line);
-				}
-				logf_error("\E[1;36mSHADER ERROR: %s: \E[0m\E[1m%s\n", p_function,
-						err_details);
-				logf_error("\E[0;36m   At: %s:%i.\E[0m\n", p_file, p_line);
-				break;
-			case ERR_ERROR:
-			default:
-				if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_12) {
-					os_log_error(OS_LOG_DEFAULT,
-							"ERROR: %{public}s: %{public}s\nAt: %{public}s:%i.",
-							p_function, err_details, p_file, p_line);
-				}
-				logf_error("\E[1;31mERROR: %s: \E[0m\E[1m%s\n", p_function, err_details);
-				logf_error("\E[0;31m   At: %s:%i.\E[0m\n", p_file, p_line);
-				break;
-		}
-	}
-};
-
-#else
-
 typedef UnixTerminalLogger OSXTerminalLogger;
-#endif
 
 void OS_OSX::alert(const String &p_alert, const String &p_title) {
 	if (is_no_window_mode_enabled()) {
