--- platform/osx/os_osx.mm.orig	2021-12-21 06:57:28.000000000 -0500
+++ platform/osx/os_osx.mm	2022-02-10 20:20:14.000000000 -0500
@@ -2958,7 +2958,7 @@
 }
 
 Error OS_OSX::execute(const String &p_path, const List<String> &p_arguments, bool p_blocking, ProcessID *r_child_id, String *r_pipe, int *r_exitcode, bool read_stderr, Mutex *p_pipe_mutex) {
-	if (@available(macOS 10.15, *)) {
+	#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101500
 		NSString *nsappname = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleName"];
 		// If executable is bundled, always execute editor instances using NSWorkspace to ensure app window is registered and activated correctly.
 		if (p_path == get_executable_path() && nsappname != nil) {
@@ -3006,9 +3006,9 @@
 		} else {
 			return OS_Unix::execute(p_path, p_arguments, p_blocking, r_child_id, r_pipe, r_exitcode, read_stderr, p_pipe_mutex);
 		}
-	} else {
+	#else
 		return OS_Unix::execute(p_path, p_arguments, p_blocking, r_child_id, r_pipe, r_exitcode, read_stderr, p_pipe_mutex);
-	}
+	#endif
 }
 
 // Returns string representation of keys, if they are printable.
