--- modules/camera/camera_osx.mm.orig	2022-08-04 21:40:16.000000000 -0400
+++ modules/camera/camera_osx.mm	2023-01-01 14:22:18.000000000 -0500
@@ -257,7 +257,7 @@
 		// Already recording!
 	} else {
 		// Start camera capture, check permission.
-		if (@available(macOS 10.14, *)) {
+		#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101400
 			AVAuthorizationStatus status = [AVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo];
 			if (status == AVAuthorizationStatusAuthorized) {
 				capture_session = [[MyCaptureSession alloc] initForFeed:this andDevice:device];
@@ -270,9 +270,9 @@
 											 }
 										 }];
 			}
-		} else {
+		#else
 			capture_session = [[MyCaptureSession alloc] initForFeed:this andDevice:device];
-		}
+		#endif
 	};
 
 	return true;
