--- bournal.orig	2011-03-22 21:50:10.000000000 +0100
+++ bournal	2011-03-22 21:54:09.000000000 +0100
@@ -1,6 +1,14 @@
 #!/bin/bash
 VERSION="Bournal 1.5, Codename: Sudo-cide"
 
+unset LANG
+unset LC_COLLATE
+unset LC_CTYPE
+unset LC_MESSAGES
+unset LC_MONETARY
+unset LC_NUMERIC
+unset LC_TIME
+
 # parameter check
 if [ -n "$1" -a "$1" != "-h" -a "$1" != "-i" -a "$1" != "-l" -a "$1" != "-ls" -a "$1" != "-lst" -a "$1" != "-d"  -a "$1" != "-f" -a "$1" != "-fi" -a "$1" != "-r" -a "$1" != "-p" -a "$1" != "--help" -a "$1" != "-v" -a "$1" != "--version" -a "$1" != "--heh" ]; then
    echo 'Invalid parameter. Type bournal -h for help.'
@@ -166,9 +174,9 @@
 
 posty ()
 {
-  EDCHK="$(md5sum "$FILE" | cut -d" " -f1)"
+  EDCHK="$(md5 "$FILE" | cut -d" " -f4)"
   "$EDITOR" "$FILE"
-  EDCHK2="$(md5sum "$FILE" | cut -d" " -f1)"
+  EDCHK2="$(md5 "$FILE" | cut -d" " -f4)"
   gpg --passphrase "$PASS" -c "$FILE" 2>/dev/null
   rm -f "$FILE"
   if [ "$EDCHK" = "$EDCHK2" ]; then
