diff --git include/crypto/rand.h include/crypto/rand.h
index ac41a9f..38c3888 100644
--- include/crypto/rand.h
+++ include/crypto/rand.h
@@ -23,9 +23,8 @@
 # include "crypto/rand_pool.h"
 
 # if defined(__APPLE__) && !defined(OPENSSL_NO_APPLE_CRYPTO_RANDOM)
-#  include <Availability.h>
-#  if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 101000) || \
-     (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 80000)
+#  if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101200 || \
+      __ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED >= 80000
 #   define OPENSSL_APPLE_CRYPTO_RANDOM 1
 #   include <CommonCrypto/CommonCryptoError.h>
 #   include <CommonCrypto/CommonRandom.h>
