--- include/ns.mak.in.orig	2018-10-12 14:28:30.000000000 -0700
+++ include/ns.mak.in	2018-10-12 14:28:43.000000000 -0700
@@ -88,7 +88,7 @@
 CFLAGS_WARNING	= @TCL_CFLAGS_WARNING@
 CFLAGS_INCLUDE	= -I$(INCDIR) @TCL_INCLUDE_SPEC@
 CFLAGS_EXTRA	= @TCL_SHLIB_CFLAGS@ @TCL_EXTRA_CFLAGS@
-DEFS      	= -DNO_CONST @TCL_DEFS@ @DEFS@
+DEFS      	= -DUSE_INTERP_ERRORLINE @TCL_DEFS@ @DEFS@
 CFLAGS		+= @CFLAGS_DEFAULT@ $(CFLAGS_WARNING) $(CFLAGS_EXTRA) $(CFLAGS_INCLUDE) @CPPFLAGS@ $(DEFS)
 ifndef SRCDIR
     UTILS	 = $(AOLSERVER)/bin
--- nsd/pools.c.orig	2018-10-12 14:45:54.000000000 -0700
+++ nsd/pools.c	2018-10-12 14:46:11.000000000 -0700
@@ -250,7 +250,7 @@
  */
 
 int
-NsTclListPoolsObjCmd(ClientData arg, Tcl_Interp *interp, int objc, Tcl_Obj **objv)
+NsTclListPoolsObjCmd(ClientData arg, Tcl_Interp *interp, int objc, Tcl_Obj *const *objv)
 {
     AppendData data;
 
--- nsdb/dbtcl.c.orig	2018-10-12 14:43:37.000000000 -0700
+++ nsdb/dbtcl.c	2018-10-12 14:43:56.000000000 -0700
@@ -157,7 +157,7 @@
  */
 
 static int
-DbObjCmd(ClientData data, Tcl_Interp *interp, int objc, Tcl_Obj **objv)
+DbObjCmd(ClientData data, Tcl_Interp *interp, int objc, Tcl_Obj *const *objv)
 {
 #define MAXHANDLES 4
     InterpData	   *idataPtr = data;
