--- a/libmariadb/libmariadb/CMakeLists.txt	2026-05-31 09:01:44
+++ b/libmariadb/libmariadb/CMakeLists.txt	2026-05-31 09:02:23
@@ -345,9 +345,6 @@
   IF(ICONV_INCLUDE_DIR)
      INCLUDE_DIRECTORIES(BEFORE ${ICONV_INCLUDE_DIR}) 
   ENDIF()
-  IF(NOT CMAKE_SYSTEM_NAME MATCHES AIX)
-    add_compile_definitions(LIBICONV_PLUG)
-  ENDIF()
   IF(APPLE)
     add_compile_options(-Wno-deprecated-declarations)
   ENDIF()
