--- pack/CMakeLists.txt.orig	2011-11-17 16:21:18.000000000 +0100
+++ pack/CMakeLists.txt	2011-11-19 17:52:15.000000000 +0100
@@ -72,7 +72,7 @@
 		install(CODE "
 			file(GLOB_RECURSE QTPLUGINS \"\${CMAKE_INSTALL_PREFIX}/${EXE_DIR}/plugins/*${CMAKE_SHARED_LIBRARY_SUFFIX}\")
 		include(BundleUtilities)
-		fixup_bundle(\"${APPS}\" \"${QTPLUGINS}\" \"${QT_LIBRARY_DIRS}\")" COMPONENT Runtime)
+		 " COMPONENT Runtime)
 	else(APPLE)
 		set(CPACK_PACKAGE_NAME "tagainijisho")	
 		set(ROOT_DIR share/tagainijisho)
