--- CMakeLists.txt.orig	2012-05-10 17:42:15.000000000 -0500
+++ CMakeLists.txt	2012-08-27 20:00:35.000000000 -0500
@@ -30,8 +30,6 @@
 # 64 bits Intel binary with 10.6 compatibility 
 if(APPLE)
 	set(CMAKE_OSX_ARCHITECTURES "${ARCHS_STANDARD_64_BIT}")
-	set(CMAKE_OSX_SYSROOT "/Developer/SDKs/MacOSX10.6.sdk")
-	set(CMAKE_OSX_DEPLOYMENT_TARGET "10.6")
 	set(CMAKE_PREFIX_PATH "${CMAKE_OSX_SYSROOT}/usr")  
 	set(CMAKE_FRAMEWORK_PATH "${CMAKE_OSX_SYSROOT}/Library/Frameworks:${CMAKE_OSX_SYSROOT}/System/")
 	set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -sectcreate __TEXT __info_plist ${CMAKE_BINARY_DIR}/src/gui/Tagaini\\ Jisho.app/Contents/Info.plist")
