--- CMakeLists.txt	2022-11-26 10:23:59
+++ CMakeLists.txt	2024-08-14 15:22:51
@@ -1,8 +1,8 @@
 # 64 bits Intel binary with 10.13 compatibility (these variables should be set before any project command)
-if(APPLE)
-	set(CMAKE_OSX_ARCHITECTURES "${ARCHS_STANDARD_64_BIT}")
-	set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
-endif(APPLE)
+#if(APPLE)
+#	set(CMAKE_OSX_ARCHITECTURES "${ARCHS_STANDARD_64_BIT}")
+#	set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13")
+#endif(APPLE)
 
 # The project name decides the naming pattern of many things - choose it according
 # to the standard of the platform we run on.
