--- pack/CMakeLists.txt	2022-11-26 10:23:59
+++ pack/CMakeLists.txt	2024-08-14 15:21:30
@@ -103,9 +103,9 @@
 		COMPONENT Runtime
 		DESTINATION ".")
 	# Qt config file
-	install(CODE
-		"file(WRITE \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${RESOURCES_DIR}/qt.conf\" \"[Paths]\n\rPlugins = plugins\n\rTranslations = i18n\n\r\")"
-		COMPONENT Runtime)
+#install(CODE "file(WRITE \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${RESOURCES_DIR}/qt.conf\" \"[Paths]\nPlugins = PlugIns\nTranslations = Translations\n\")" COMPONENT Runtime)
+#		"file(WRITE \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${RESOURCES_DIR}/qt.conf\" \"[Paths]\n\rPlugins = plugins\n\rTranslations = i18n\n\r\")"
+#		COMPONENT Runtime)
 endif(UNIX AND APPLE)
 
 # Databases
