--- make.sh.orig	2020-05-04 13:08:18.000000000 -0400
+++ make.sh	2020-05-04 13:08:06.000000000 -0400
@@ -3,43 +3,17 @@
 set -e
 shopt -s extglob
 
-if [[ $# == 0 ]]; then
-    ios=false
-else
-    ios=$1
-    shift
-fi
-
-export DEVELOPER_DIR=/Applications/Xcode-5.1.1.app
-
 os=()
 
-if "${ios}"; then
-
-out=ios
-flags=(cycc -- -miphoneos-version-min=2.0 -arch armv6 -arch arm64)
-
-flags+=(-Xarch_armv6 -Isysroot32/usr/include)
-flags+=(-Xarch_arm64 -Isysroot64/usr/include)
-
-flags+=(-Xarch_armv6 -Lsysroot32/usr/lib)
-flags+=(-Xarch_arm64 -Lsysroot64/usr/lib)
-
-static=false
-flags+=(-framework CoreFoundation)
-
-flags+=(-lplist)
-flags+=(-lcrypto)
-
-else
-
 out=out
 
 if which xcrun &>/dev/null; then
     flags=(xcrun -sdk macosx g++)
-    flags+=(-mmacosx-version-min=10.4)
+    flags+=(-mmacosx-version-min="$os_ver")
+    flags+=(-I"$PREFIX"/include)
+    flags+=(-L"$PREFIX"/lib)
 
-    for arch in i386 x86_64; do
+    for arch in x86_64; do
         flags+=(-arch "${arch}")
     done
 else
@@ -54,8 +28,6 @@
 flags+=(-lcrypto)
 #flags+=(-Wl,/usr/lib/libcrypto.42.dylib)
 
-fi
-
 sdk=$(xcodebuild -sdk iphoneos -version Path)
 
 flags+=(-I.)
@@ -77,14 +49,8 @@
 
 fi
 
-flags+=("$@")
-
 mkdir -p "${out}"
 set -x
 
 "${flags[@]}" -O3 -g0 -c -std=c++11 -o "${out}"/ldid.o ldid.cpp
 "${flags[@]}" -O3 -g0 -o "${out}"/ldid "${out}"/ldid.o "${os[@]}" -x c lookup2.c -lxml2 -framework Security
-
-if ! "${ios}"; then
-    ln -sf out/ldid .
-fi
