--- node.h.orig	2007-10-23 17:09:10.000000000 +0200
+++ node.h	2007-10-23 17:08:19.000000000 +0200
@@ -374,9 +374,17 @@
 void rb_add_event_hook _((rb_event_hook_func_t,rb_event_t));
 int rb_remove_event_hook _((rb_event_hook_func_t));
 
-#if defined(HAVE_GETCONTEXT) && defined(HAVE_SETCONTEXT)
-#include <ucontext.h>
-#define USE_CONTEXT
+#if defined(HAVE_GETCONTEXT) && defined(HAVE_SETCONTEXT) 
+# if defined(__APPLE__)
+#   include <AvailabilityMacros.h>
+#   if MAC_OS_X_VERSION_MAX_ALLOWED <= 1040
+#     include <ucontext.h>
+#     define USE_CONTEXT
+#   endif
+# else
+#   include <ucontext.h>
+#   define USE_CONTEXT
+# endif
 #endif
 #include <setjmp.h>
 #include "st.h"
