diff -ur ../ruby-1.8.7-p160.org/mkconfig.rb ./mkconfig.rb
--- ../ruby-1.8.7-p160.org/mkconfig.rb	2008-06-06 19:39:57.000000000 +0900
+++ ./mkconfig.rb	2009-04-19 01:21:16.000000000 +0900
@@ -147,6 +147,12 @@
   CONFIG["sitearchdir"] = "$(sitelibdir)/$(sitearch)"
   CONFIG["vendorlibdir"] = "$(vendordir)/$(ruby_version)"
   CONFIG["vendorarchdir"] = "$(vendorlibdir)/$(sitearch)"
+  # macports -rvendor-specific support
+  if defined?(VENDOR_SPECIFIC) && VENDOR_SPECIFIC
+    CONFIG["sitedir"] = CONFIG["vendordir"]
+    CONFIG["sitelibdir"] = CONFIG["vendorlibdir"]
+    CONFIG["sitearchdir"] = CONFIG["vendorarchdir"]
+  end
   CONFIG["topdir"] = File.dirname(__FILE__)
   MAKEFILE_CONFIG = {}
   CONFIG.each{|k,v| MAKEFILE_CONFIG[k] = v.dup}
