--- lib/gvc/Makefile.am.orig	2011-06-26 03:07:07.000000000 -0700
+++ lib/gvc/Makefile.am	2011-06-26 03:07:27.000000000 -0700
@@ -55,7 +55,6 @@ libgvc_C_la_DEPENDENCIES +=  $(top_build
 endif
 if ENABLE_LTDL
-libgvc_C_la_LIBADD += $(LIBLTDL) $(LIBLTDL_LDFLAGS)
+libgvc_C_la_LIBADD += @LIBLTDL@ $(LIBLTDL_LDFLAGS)
-libgvc_C_la_DEPENDENCIES += $(LIBLTDL)
 endif
 
 libgvc_la_LDFLAGS = -version-info $(GVC_VERSION) -no-undefined

