--- cmd/gvedit/Makefile.in.orig	2012-08-13 23:46:56.000000000 -0500
+++ cmd/gvedit/Makefile.in	2012-08-14 13:57:37.000000000 -0500
@@ -67,8 +67,7 @@
 	gvedit-mdichild.$(OBJEXT) $(am__objects_1)
 gvedit_OBJECTS = $(am_gvedit_OBJECTS)
 am__DEPENDENCIES_1 =
-gvedit_DEPENDENCIES = $(top_builddir)/lib/gvc/libgvc.la \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
+gvedit_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_$(V))
 am__v_lt_ = $(am__v_lt_$(AM_DEFAULT_VERBOSITY))
@@ -535,7 +534,7 @@
 	-DGVEDIT_DATADIR=\""$(pkgdatadir)/gvedit"\"
 
 gvedit_LDADD = \
-	$(top_builddir)/lib/gvc/libgvc.la \
+	-lgvc \
 	$(QTGUI_LIBS) \
 	$(QTCORE_LIBS) \
 	$(EXPAT_LIBS) $(Z_LIBS) 
