http://www.graphviz.org/mantisbt/view.php?id=2257
--- plugin/ming/Makefile.in.orig	2013-02-10 23:46:19.000000000 -0600
+++ plugin/ming/Makefile.in	2013-02-12 23:44:36.000000000 -0600
@@ -510,7 +510,7 @@
 	-I$(top_srcdir)/lib/common \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/gvc \
-	-I$(top_srcdir)/lib/graph \
+	-I$(top_srcdir)/lib/cgraph \
 	-I$(top_srcdir)/lib/cdt \
 	$(MING_CFLAGS)
 
