--- cmd/gvedit/Makefile.in.orig	2023-07-07 02:42:43.000000000 -0500
+++ cmd/gvedit/Makefile.in	2023-07-10 04:52:31.000000000 -0500
@@ -123,9 +123,7 @@
 	gvedit-mdichild.$(OBJEXT) $(am__objects_1)
 gvedit_OBJECTS = $(am_gvedit_OBJECTS)
 am__DEPENDENCIES_1 =
-gvedit_DEPENDENCIES = $(top_builddir)/lib/gvc/libgvc.la \
-	$(top_builddir)/lib/cgraph/libcgraph.la \
-	$(top_builddir)/lib/cdt/libcdt.la $(am__DEPENDENCIES_1) \
+gvedit_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_@AM_V@)
@@ -620,9 +618,9 @@
 	-DDEMAND_LOADING=1
 
 gvedit_LDADD = \
-	$(top_builddir)/lib/gvc/libgvc.la \
-	$(top_builddir)/lib/cgraph/libcgraph.la \
-	$(top_builddir)/lib/cdt/libcdt.la \
+	-lgvc \
+	-lcgraph \
+	-lcdt \
 	$(QTGUI_LIBS) \
 	$(QTCORE_LIBS) \
 	$(EXPAT_LIBS) $(Z_LIBS)
