--- pyproject.toml.orig	2025-07-02 15:49:28
+++ pyproject.toml	2025-07-15 14:55:02
@@ -9,7 +9,7 @@
     # Used by cffi (which import distutils, and in Python 3.12, distutils has
     # been removed from the stdlib, but installing setuptools puts it back) as
     # well as our build.rs for the rust/cffi bridge.
-    "setuptools!=74.0.0,!=74.1.0,!=74.1.1,!=74.1.2",
+    "setuptools",
 ]
 build-backend = "maturin"
 
