--- src/_cffi_src/build_openssl.py.orig	2018-08-14 12:24:23.000000000 -0500
+++ src/_cffi_src/build_openssl.py	2018-08-19 08:05:13.000000000 -0500
@@ -46,10 +46,7 @@
     When we drop support for CRYPTOGRAPHY_OPENSSL_LESS_THAN_110 we can
     revisit this.
     """
-    if platform not in ["win32", "hp-ux11"]:
-        return ["-Wconversion", "-Wno-error=sign-conversion"]
-    else:
-        return []
+    return []
 
 
 ffi = build_ffi_for_binding(
