--- src/_cffi_src/build_openssl.py.orig	2022-11-10 18:16:14.000000000 +0300
+++ src/_cffi_src/build_openssl.py	2022-11-10 18:18:19.000000000 +0300
@@ -65,13 +65,7 @@
             "gcc" in cmd.compiler.compiler[0]
             or "clang" in cmd.compiler.compiler[0]
         )
-    if is_gcc or not (
-        platform in ["win32", "hp-ux11", "sunos5"]
-        or platform.startswith("aix")
-    ):
-        return ["-Wconversion", "-Wno-error=sign-conversion"]
-    else:
-        return []
+    return []
 
 
 ffi = build_ffi_for_binding(
