--- a2ps-j.orig	2010-06-25 04:54:48.000000000 +0900
+++ a2ps-j	2010-11-15 18:01:02.000000000 +0900
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/env perl
 ;#
 ;# a2ps: ascii to ps
 ;#
@@ -192,7 +192,6 @@
 
 if ($debug == 2) {
     require('dumpvar.pl');
-    local($#) = '%.6g';
     &dumpvar('main',
 	     'width', 'height', 'lmargin', 'smargin', 'font_size',
 	     'sheet_height', 'sheet_width', 'char_width', 'skip_column',
@@ -410,7 +409,6 @@
     unless ($selfconvert) {
 	shift while @_ > 0 && $_[0] eq '';
 	exec('nkf', '-b', '-j', @_);
-	exec('jconv', '-j', @_);
     }
     if (($file = shift) ne '') {
 	open(STDIN, $file) || die "$file: $!\n";
