diff --git cmake/BTApplication.cmake cmake/BTApplication.cmake
index 9c5eaa6..9da73dd 100644
--- cmake/BTApplication.cmake
+++ cmake/BTApplication.cmake
@@ -55,20 +55,6 @@ IF(MSVC)
   SET(CMAKE_CXX_FLAGS_DEBUG   "-DSWUSINGDLL /MDd /Zc:wchar_t- /W1 /D_UNICODE /DUNICODE /Zc:wchar_t /Od")
 ELSE()
   ADD_COMPILE_OPTIONS("-fPIE" "-fexceptions")
-  SET(CMAKE_CXX_FLAGS_RELEASE "")
-  SET(CMAKE_CXX_FLAGS_DEBUG "")
-  IF(APPLE)
-    SET(T "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/")
-    LIST(APPEND BibleTime_CXXFLAGS
-        "-mmacosx-version-min=10.12"
-        "-stdlib=libc++"
-        "-isysroot" "${T}")
-    LIST(APPEND BibleTime_LDFLAGS
-        "-mmacosx-version-min=10.12"
-        "-stdlib=libc++"
-    )
-    UNSET(T)
-  ENDIF()
 ENDIF()
 
 
 
