From d6ed57b7c265a4abc018297e942b653d2a0bf005 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 14 Aug 2024 11:01:54 +0800
Subject: [PATCH 2/3] port.c: define AI_NUMERICSERV if undefined

---
 src/utils/port.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/utils/port.c src/utils/port.c
index 76982ac..823f7f1 100644
--- src/utils/port.c
+++ src/utils/port.c
@@ -62,6 +62,10 @@
 #define MIN(a,b) a<=b ? a : b
 #endif
 
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
 static void *bctbx_libc_malloc(size_t sz){
 	return malloc(sz);
 }
