--- src/CMakeLists.txt	2023-01-16 16:38:20.000000000 +0800
+++ src/CMakeLists.txt	2024-08-14 11:13:46.000000000 +0800
@@ -105,7 +105,7 @@
 	add_library(bctoolbox-static STATIC ${BCTOOLBOX_SOURCE_FILES} ${BCTOOLBOX_HEADER_FILES} ${BCTOOLBOX_PRIVATE_HEADER_FILES})
 	target_link_libraries(bctoolbox-static INTERFACE ${CMAKE_THREAD_LIBS_INIT})
 	if(APPLE)
-		target_link_libraries(bctoolbox-static INTERFACE "-framework Foundation" "-framework AVFoundation")
+		target_link_libraries(bctoolbox-static INTERFACE "-framework Foundation")
 		target_include_directories(bctoolbox-static SYSTEM PRIVATE ${ICONV_INCLUDE_DIRS})
 		target_link_libraries(bctoolbox-static INTERFACE ${ICONV_LIBRARIES})
 	endif()
@@ -149,7 +149,7 @@
 		if (IOS)
 			target_link_libraries(bctoolbox PRIVATE "-framework UIKit" "-framework Foundation" "-framework AVFoundation")
 		else()
-			target_link_libraries(bctoolbox PRIVATE "-framework Foundation" "-framework AVFoundation")
+			target_link_libraries(bctoolbox PRIVATE "-framework Foundation")
 		endif()
 		target_include_directories(bctoolbox SYSTEM PRIVATE ${ICONV_INCLUDE_DIRS})
 		target_link_libraries(bctoolbox PRIVATE ${ICONV_LIBRARIES})
