--- meson.build	2024-07-11 01:03:14.000000000 +0800
+++ meson.build	2024-10-28 22:14:39.000000000 +0800
@@ -248,7 +248,6 @@
   'rint',
   'log2',
   'exp2',
-  'sincos',
   'trunc',
   'localtime_r',
   'fmin',
@@ -260,6 +259,10 @@
   endif
 endforeach
 
+if cc.has_function('sincos', prefix: '#define _GNU_SOURCE\n#include <math.h>', dependencies: libm)
+  cdata.set('HAVE_SINCOS', 1)
+endif
+
 cdata.set('HAVE_DECL_ISINF', cc.has_header_symbol('math.h', 'isinf') ? 1 : false)
 cdata.set('HAVE_DECL_ISNAN', cc.has_header_symbol('math.h', 'isnan') ? 1 : false)
 
