--- gdk/quartz/GdkQuartzView.c.orig	2022-01-01 14:50:31.000000000 -0500
+++ gdk/quartz/GdkQuartzView.c	2022-01-01 14:51:28.000000000 -0500
@@ -296,11 +296,13 @@
    * has the additional cost of an 8-bit (ARGB) to 10-bit conversion.
    * Let's disable this mode to regain the lost performance.
    */
+#if defined(MAC_OS_X_VERSION_10_12) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12
   if(gdk_quartz_osx_version() >= GDK_OSX_BIGSUR)
   {
     CALayer* layer = self.layer;
     layer.contentsFormat = kCAContentsFormatRGBA8Uint;
   }
+#endif
 
   [super viewWillDraw];
 }
