--- gdk/quartz/gdkinternal-quartz.h.orig	2021-04-05 15:46:38.000000000 +0200
+++ gdk/quartz/gdkinternal-quartz.h	2021-04-05 15:46:49.000000000 +0200
@@ -75,7 +75,7 @@
   GDK_QUARTZ_EVENT_SUBTYPE_EVENTLOOP
 } GdkQuartzEventSubType;
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 10130
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101300
 #define GDK_QUARTZ_EVENT_TABLET_PROXIMITY NSEventTypeTabletProximity
 #define GDK_QUARTZ_EVENT_SUBTYPE_TABLET_PROXIMITY NSEventSubtypeTabletProximity
 #define GDK_QUARTZ_EVENT_SUBTYPE_TABLET_POINT NSEventSubtypeTabletPoint
--- ./modules/input/imquartz.c.orig	2021-04-05 16:09:16.000000000 +0200
+++ ./modules/input/imquartz.c	2021-04-05 16:09:25.000000000 +0200
@@ -37,7 +37,7 @@
 #define GTK_IM_CONTEXT_QUARTZ(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), GTK_IM_CONTEXT_TYPE_QUARTZ, GtkIMContextQuartz))
 #define GTK_IM_CONTEXT_QUARTZ_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS((obj), GTK_IM_CONTEXT_TYPE_QUARTZ, GtkIMContextQuartzClass))
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 10120
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 101200
 #define NS_EVENT_KEY_DOWN NSKeyDown
 #else
 #define NS_EVENT_KEY_DOWN NSEventTypeKeyDown
