--- src/configure.orig	2009-06-07 04:26:53.000000000 -0700
+++ src/configure	2018-05-21 05:24:47.000000000 -0700
@@ -2867,7 +2867,7 @@
 #if test -z "$mpath" || test -z "$opath" || test -z "$xpath" || test -z "$altmpath" || test -z "$altopath" || test -z "$ver" ; then
 if test -z "$mpath" || test -z "$opath" || test -z "$xpath" || test -z "$ver" ; then
       cat > conftest.cc <<EOF
-#include <octave/config.h>
+#include <octave/octave-config.h>
 #include <octave/version.h>
 #include <octave/defaults.h>
 
@@ -2956,7 +2956,7 @@
 { $as_echo "$as_me:$LINENO: checking for F77_FUNC" >&5
 $as_echo_n "checking for F77_FUNC... " >&6; }
 cat > conftest.cc << EOF
-#include <octave/config.h>
+#include <octave/octave-config.h>
 int F77_FUNC (hello, HELLO) (const int &n);
 EOF
 ac_try="$MKOCTFILE -c conftest.cc"
