diff --git a/build.zig b/build.zig
--- a/build.zig
+++ b/build.zig
@@ -775,7 +775,7 @@
         // This number should never be raised. If the value is exceeded, then
         // it is considered a bug in the Zig project that building the
         // compiler takes more than 8G of memory.
-        .max_rss = 8_000_000_000,
+        .max_rss = 7_500_000_000,
         .root_module = addCompilerMod(b, options),
     });
     exe.stack_size = stack_size;
