diff -u dist/aclocal/mutex.m4.orig dist/aclocal/mutex.m4
--- dist/aclocal/mutex.m4.orig	2007-05-19 01:08:50.000000000 +1000
+++ dist/aclocal/mutex.m4	2021-02-02 23:11:42.000000000 +1100
@@ -9,7 +9,7 @@
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
 	pthread_mutexattr_t mutexattr;
-	exit (
+	return (
 	pthread_condattr_init(&condattr) ||
 	pthread_condattr_setpshared(&condattr, PTHREAD_PROCESS_SHARED) ||
 	pthread_mutexattr_init(&mutexattr) ||
@@ -29,7 +29,7 @@
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
 	pthread_mutexattr_t mutexattr;
-	exit (
+	return (
 	pthread_condattr_init(&condattr) ||
 	pthread_condattr_setpshared(&condattr, PTHREAD_PROCESS_SHARED) ||
 	pthread_mutexattr_init(&mutexattr) ||
@@ -51,7 +51,7 @@
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
 	pthread_mutexattr_t mutexattr;
-	exit (
+	return (
 	pthread_condattr_init(&condattr) ||
 	pthread_mutexattr_init(&mutexattr) ||
 	pthread_cond_init(&cond, &condattr) ||
@@ -69,7 +69,7 @@
 	pthread_mutex_t mutex;
 	pthread_condattr_t condattr;
 	pthread_mutexattr_t mutexattr;
-	exit (
+	return (
 	pthread_condattr_init(&condattr) ||
 	pthread_mutexattr_init(&mutexattr) ||
 	pthread_cond_init(&cond, &condattr) ||
@@ -145,7 +145,7 @@
 		static lwp_cond_t ci = SHAREDCV;
 		lwp_mutex_t mutex = mi;
 		lwp_cond_t cond = ci;
-		exit (
+		return (
 		_lwp_mutex_lock(&mutex) ||
 		_lwp_mutex_unlock(&mutex));
 	], [db_cv_mutex=Solaris/lwp])
@@ -185,7 +185,7 @@
 		mutex_t mutex;
 		cond_t cond;
 		int type = USYNC_PROCESS;
-		exit (
+		return (
 		mutex_init(&mutex, type, NULL) ||
 		cond_init(&cond, type, NULL) ||
 		mutex_lock(&mutex) ||
@@ -200,7 +200,7 @@
 		mutex_t mutex;
 		cond_t cond;
 		int type = USYNC_PROCESS;
-		exit (
+		return (
 		mutex_init(&mutex, type, NULL) ||
 		cond_init(&cond, type, NULL) ||
 		mutex_lock(&mutex) ||
@@ -220,7 +220,7 @@
 	# x86/gcc: FreeBSD, NetBSD, BSD/OS, Linux
 	AC_TRY_COMPILE(,[
 	#if (defined(i386) || defined(__i386__)) && defined(__GNUC__)
-		exit(0);
+		return (0);
 	#else
 		FAIL TO COMPILE/LINK
 	#endif
@@ -229,7 +229,7 @@
 	# x86_64/gcc: FreeBSD, NetBSD, BSD/OS, Linux
 	AC_TRY_COMPILE(,[
 	#if (defined(x86_64) || defined(__x86_64__)) && defined(__GNUC__)
-		exit(0);
+		return (0);
 	#else
 		FAIL TO COMPILE/LINK
 	#endif
@@ -251,7 +251,7 @@
 	# Sparc/gcc: SunOS, Solaris
 	AC_TRY_COMPILE(,[
 	#if defined(__sparc__) && defined(__GNUC__)
-		exit(0);
+		return (0);
 	#else
 		FAIL TO COMPILE/LINK
 	#endif
@@ -293,7 +293,7 @@
 	msem_init(&x, 0);
 	msem_lock(&x, 0);
 	msem_unlock(&x, 0);
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -310,7 +310,7 @@
 	msem_init(&x, 0);
 	msem_lock(&x, 0);
 	msem_unlock(&x, 0);
-	exit(0);
+	return (0);
 ], [db_cv_mutex=UNIX/msem_init])
 fi
 
@@ -332,7 +332,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__USLC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -389,7 +389,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__alpha) && defined(__DECC)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -400,7 +400,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__alpha) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -411,7 +411,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__arm__) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -422,7 +422,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if (defined(__mips) || defined(__mips__)) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -433,7 +433,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if (defined(__hppa) || defined(__hppa__)) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -444,7 +444,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if (defined(__powerpc__) || defined(__ppc__)) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -455,7 +455,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if (defined(mc68020) || defined(sun3)) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -466,7 +466,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__MVS__) && defined(__IBMC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -477,7 +477,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__s390__) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -488,7 +488,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(__ia64) && defined(__GNUC__)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
@@ -499,7 +499,7 @@
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
 #if defined(_UTS)
-	exit(0);
+	return (0);
 #else
 	FAIL TO COMPILE/LINK
 #endif
diff -u dist/aclocal/sequence.m4.orig dist/aclocal/sequence.m4
--- dist/aclocal/sequence.m4.orig	2005-11-24 11:49:24.000000000 +1100
+++ dist/aclocal/sequence.m4	2021-02-02 23:06:31.000000000 +1100
@@ -43,7 +43,11 @@
 	# test, which won't test for the appropriate printf format strings.
 	if test "$db_cv_build_sequence" = "yes"; then
 		AC_TRY_RUN([
-		main() {
+		#include <stdio.h>
+        #if HAVE_STRING_H
+        #include <string.h>
+        #endif
+		int main() {
 			$db_cv_seq_type l;
 			unsigned $db_cv_seq_type u;
 			char buf@<:@100@:>@;
@@ -59,7 +63,12 @@
 				return (1);
 			return (0);
 		}],, [db_cv_build_sequence="no"],
-		AC_TRY_LINK(,[
+		AC_TRY_LINK([
+		#include <stdio.h>
+        #if HAVE_STRING_H
+        #include <string.h>
+        #endif
+		],[
 			$db_cv_seq_type l;
 			unsigned $db_cv_seq_type u;
 			char buf@<:@100@:>@;
--- db_dump185/db_dump185.c.orig	2007-05-18 01:15:00.000000000 +1000
+++ db_dump185/db_dump185.c	2021-02-02 23:55:31.000000000 +1100
@@ -19,6 +19,7 @@ static const char copyright[] =
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #include <db.h>
 
