--- renderer/CMakeLists.txt.orig	2011-07-29 20:03:33.000000000 -0500
+++ renderer/CMakeLists.txt	2011-07-29 20:07:12.000000000 -0500
@@ -25,5 +25,5 @@
   ${QT_QTDBUS_LIBRARY}
 )
 
-install(TARGETS kdenlive_render DESTINATION ${BIN_INSTALL_DIR})
+install(TARGETS kdenlive_render DESTINATION ${BUNDLE_INSTALL_DIR})
 
--- src/CMakeLists.txt.orig	2011-07-29 20:03:33.000000000 -0500
+++ src/CMakeLists.txt	2011-07-29 20:07:15.000000000 -0500
@@ -363,7 +363,7 @@
   endif(OPENGL_FOUND)
 endif(APPLE)
 
-install(TARGETS kdenlive DESTINATION ${BIN_INSTALL_DIR})
+install(TARGETS kdenlive DESTINATION ${BUNDLE_INSTALL_DIR})
 install(FILES kdenliveui.rc kdenlive.notifyrc DESTINATION ${DATA_INSTALL_DIR}/kdenlive)
 install(FILES kdenlivesettings.kcfg DESTINATION ${KCFG_INSTALL_DIR})
 install(FILES kdenlive.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
