diff --git renderer/CMakeLists.txt renderer/CMakeLists.txt
index 0bd7a74..277c351 100644
--- renderer/CMakeLists.txt
+++ renderer/CMakeLists.txt
@@ -18,4 +18,4 @@ target_link_libraries(kdenlive_render
   ${QT_LIBRARIES}
 )
 
-install(TARGETS kdenlive_render DESTINATION ${BIN_INSTALL_DIR})
+install(TARGETS kdenlive_render DESTINATION ${BUNDLE_INSTALL_DIR})
diff --git src/CMakeLists.txt src/CMakeLists.txt
index c6e2ff3..ccaa01b 100644
--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -306,4 +306,4 @@ if(BUILD_JogShuttle)
   )
 endif(BUILD_JogShuttle)
 
-install(TARGETS kdenlive DESTINATION ${BIN_INSTALL_DIR})
+install(TARGETS kdenlive DESTINATION ${BUNDLE_INSTALL_DIR})
