https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=700958
--- rtf-fix/rtf2rtf.l.orig	2008-05-28 18:16:36.000000000 +0200
+++ rtf-fix/rtf2rtf.l	2013-02-19 11:10:25.367016000 +0100
@@ -72,7 +72,14 @@
 int skipnewline = 0;
 
 extern char *yytext;
+
+#if   YY_FLEX_MAJOR_VERSION > 2 \
+ ||  (YY_FLEX_MAJOR_VERSION == 2 && YY_FLEX_MINOR_VERSION > 5) \
+ ||  (YY_FLEX_MAJOR_VERSION == 2 && YY_FLEX_MINOR_VERSION == 5 && YY_FLEX_SUBMINOR_VERSION > 35)
+extern yy_size_t yyleng;
+#else
 extern int yyleng;
+#endif
 
 #define CHAPTER 1
 #define SECTION 2
