--- gcstar/install.orig	2023-10-14 11:47:28.000000000 -0700
+++ gcstar/install	2023-10-14 11:47:37.000000000 -0700
@@ -188,8 +188,6 @@
     copy 'share/applications/gcstar.xml', $baseDir.'share/mime/packages'
         if (-w $baseDir.'share/mime/packages');
 
-    system 'update-desktop-database '.$baseDir.'share/applications';
-    system 'update-mime-database '.$baseDir.'share/mime';
 }
 
 sub doInstall
