From 82b9ed62d1682d9736a5f2b7ad08e7f6adfaa3d2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 24 Nov 2023 17:17:31 +0800
Subject: [PATCH] CMakeLists: do not force i386

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 213cfaa..b067a5d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -20,7 +20,7 @@ ENDIF(UNIX AND NOT APPLE)
 IF(APPLE)
 # 	SET (CMAKE_INSTALL_PREFIX /Applications CACHE PATH "Installation path")
 # 	SET (FONTMATRIX_BIN bin "")
-	set( CMAKE_OSX_ARCHITECTURES i386 )
+#	set( CMAKE_OSX_ARCHITECTURES i386 )
 ENDIF(APPLE)
 
 IF(WIN32)
