--- gdl/gdl-dock-object.h.orig	2013-10-28 14:12:04.000000000 -0700
+++ gdl/gdl-dock-object.h	2013-10-28 14:12:37.000000000 -0700
@@ -381,7 +381,7 @@
     G_STMT_START {                            \
     g_log (G_LOG_DOMAIN,                      \
 	   G_LOG_LEVEL_DEBUG,                 \
-           "%s:%d (%s) %s [%p %d%s:%d]: "format, \
+           "%s:%d (%s) %s [%p %d%s:%d]: " format, \
 	   __FILE__,                          \
 	   __LINE__,                          \
 	   __PRETTY_FUNCTION__,               \
