--- encfs/encfssh.orig	2013-11-02 23:30:01.000000000 +0100
+++ encfs/encfssh	2013-11-02 23:30:23.000000000 +0100
@@ -63,5 +63,5 @@
 cd $unenc_dir
 
 # Set the shell up
-exec /bin/sh -c "$SHELL ; cd $orig_dir ; fusermount -u $unenc_dir ; if ! $unenc_dir_given; then rmdir $unenc_dir; fi"
+exec /bin/sh -c "$SHELL ; cd $orig_dir ; umount $unenc_dir ; if ! $unenc_dir_given; then rmdir $unenc_dir; fi"
 
