--- etc/freshclam.conf.macports
+++ etc/freshclam.conf.macports	2023-12-09 13:29:07
@@ -5,7 +5,7 @@
 
 
 # Comment or remove the line below.
-Example
+#Example
 
 # Path to the database directory.
 # WARNING: It must match clamd.conf's directive!
@@ -14,7 +14,7 @@
 
 # Path to the log file (make sure it has proper permissions)
 # Default: disabled
-#UpdateLogFile /var/log/freshclam.log
+UpdateLogFile @PREFIX@/var/log/clamav/freshclam.log
 
 # Maximum size of the log file.
 # Value of 0 disables the limit.
@@ -52,7 +52,7 @@
 # It is recommended that the directory where this file is stored is
 # also owned by root to keep other users from tampering with it.
 # Default: disabled
-#PidFile /run/clamav/freshclam.pid
+PidFile @PREFIX@/var/run/clamav/freshclam.pid
 
 # By default when started freshclam drops privileges and switches to the
 # "clamav" user. This directive allows you to change the database owner.
@@ -148,7 +148,7 @@
 
 # Send the RELOAD command to clamd.
 # Default: no
-#NotifyClamd /path/to/clamd.conf
+NotifyClamd @PREFIX@/etc/clamd.conf
 
 # Run command after successful database update.
 # Use EXIT_1 to return 1 after successful database update.
@@ -166,7 +166,7 @@
 
 # Don't fork into background.
 # Default: no
-#Foreground yes
+Foreground yes
 
 # Enable debug messages in libclamav.
 # Default: no
