--- VHFShared/vhfCommonFunctions.m.orig	2015-12-29 05:50:50.000000000 -0600
+++ VHFShared/vhfCommonFunctions.m	2019-02-15 05:44:06.000000000 -0600
@@ -188,7 +188,7 @@
      * We do this to allow systems like Debian to place the library in the system folder
      */
     /* "/Library/Application Support/..." */
-    lPath = vhfPathWithPathComponents(
+    lPath = vhfPathWithPathComponents(@"%PREFIX%", 
             [NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSLocalDomainMask, YES) objectAtIndex:0],
             @"Application Support", append, nil );
     if ( lPath && [fileManager fileExistsAtPath:lPath] )
@@ -215,7 +215,7 @@
     NSString		*lPath;
 
     /* "$HOME/Library/Application Support/..." */
-    lPath = vhfPathWithPathComponents(
+    lPath = vhfPathWithPathComponents(@"%PREFIX%", 
             [NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES) objectAtIndex:0],
             @"Application Support", append, nil );
     if ( lPath && [fileManager fileExistsAtPath:lPath] )
