--- Modules/Wrappers/SWIG/src/otbPythonLogOutput.i.orig
+++ Modules/Wrappers/SWIG/src/otbPythonLogOutput.i
@@ -105,7 +105,7 @@
 /** Create the required objects for logging. Logger.Instance() is reset in 
  * order to replace the itkStdStreamLogOutput by a PythonLogOutput */
 %pythoncode {
-  _libraryLogOutput = PythonLogOutput_New()
+  _libraryLogOutput = PythonLogOutput.New()
   _libraryLogCallback = PythonPrintCallback()
 
   Logger.Instance().ResetOutputs()


--- Modules/Wrappers/SWIG/src/otbApplication.i.orig
+++ Modules/Wrappers/SWIG/src/otbApplication.i
@@ -229,7 +229,7 @@
           logger = self.GetLogger()
           logger.AddLogOutput(_libraryLogOutput.GetPointer())
 
-          self.progressReportManager = ProgressReporterManager_New()
+          self.progressReportManager = ProgressReporterManager.New()
           self.progressReportManager.SetLogOutputCallback(_libraryLogCallback)
           self.AddObserver(AddProcessToWatchEvent(),
                            self.progressReportManager.GetAddProcessCommand()
