--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -121,7 +121,7 @@
     if(WIN32)
       set(Python_ROOT_DIR "C:/tools/Python37-x64")
     endif()
-    find_package(Python REQUIRED COMPONENTS Interpreter Development)
+    find_package(Python @PYVER@ EXACT REQUIRED COMPONENTS Interpreter Development)
     if(Python_FOUND)
       message(STATUS "Python Library directory : ${Python_LIBRARY_DIRS}")
       message(STATUS "Python Headers directory : ${Python_INCLUDE_DIRS}")
