--- Modules/Wrappers/QGIS/src/CMakeLists.txt.orig
+++ Modules/Wrappers/QGIS/src/CMakeLists.txt
@@ -17,7 +17,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
 add_executable(otbQgisDescriptor otbQgisDescriptor.cxx)
+
+if(NOT "${CMAKE_BUILD_RPATH}" STREQUAL "")
+  set_property(TARGET otbQgisDescriptor PROPERTY BUILD_WITH_INSTALL_RPATH ON)
+endif()
+set_property(TARGET otbQgisDescriptor PROPERTY INSTALL_RPATH "${OTB_BINARY_DIR}/lib")
+
 target_link_libraries(otbQgisDescriptor ${OTBQGIS_LIBRARIES})
 otb_module_target(otbQgisDescriptor)
 
