--- Modules/Learning/Applications/app/CMakeLists.txt.orig
+++ Modules/Learning/Applications/app/CMakeLists.txt
@@ -131,4 +131,4 @@
   LINK_LIBRARIES ${${otb-module}_LIBRARIES})
 
 # Mantis-1427 : temporary fix
-add_dependencies(${otb-module}-all otbapp_ImageEnvelope)
\ No newline at end of file
+# add_dependencies(${otb-module}-all otbapp_ImageEnvelope)
\ No newline at end of file


--- CMake/OTBApplicationMacros.cmake.orig
+++ CMake/OTBApplicationMacros.cmake
@@ -42,6 +42,9 @@
    # Remove the usual 'lib' prefix to make it clear it is a plugin
    # and not a shared library to link against
    set_property(TARGET ${APPLICATION_TARGET_NAME} PROPERTY PREFIX "")
+   if (APPLE)
+     set_property(TARGET ${APPLICATION_TARGET_NAME} PROPERTY SUFFIX ".dylib")
+   endif()
 
    # When called from the OTB build system, use OTB_INSTALL_APP_DIR
    if (NOT APPLICATION_INSTALL_PATH AND OTB_INSTALL_APP_DIR)


