diff -u -r DNSUpdate2.8s.orig/DUDynDNSUpdater.m DNSUpdate2.8s/DUDynDNSUpdater.m
--- DNSUpdate2.8s.orig/DUDynDNSUpdater.m	2003-09-24 09:50:44.000000000 +1000
+++ DNSUpdate2.8s/DUDynDNSUpdater.m	2007-11-29 11:08:53.000000000 +1100
@@ -212,7 +212,7 @@
                     [host setStatus:@"No such host"];
                     break;
                 case _good:
-                    NSLog(@"%@: update suceeded",host);
+                    NSLog(@"%@: update succeeded",host);
                     [host updateAddressTo:[[host interface] getAddress]]; // XXX must cache all those addresses
                     [host setStatus:@"Ok"];
                     break;
@@ -283,7 +283,7 @@
                 [theUser setActive:NO];
                 break;
             case _dnserr:
-                NSLog(@"DynDNS.org DNS error. Report the following line to DynDNS.org support departement :\nPacket ID: %@",(argument?argument:@"NoPacketId"));
+                NSLog(@"DynDNS.org DNS error. Report the following line to DynDNS.org support department :\nPacket ID: %@",(argument?argument:@"NoPacketId"));
                 [theUser setUserStatus:@"DNS Error"];
                 [theUser setActive:NO];
                 break;
diff -u -r DNSUpdate2.8s.orig/DUEasyDNSUpdater.m DNSUpdate2.8s/DUEasyDNSUpdater.m
--- DNSUpdate2.8s.orig/DUEasyDNSUpdater.m	2003-09-24 09:50:44.000000000 +1000
+++ DNSUpdate2.8s/DUEasyDNSUpdater.m	2007-11-29 11:08:53.000000000 +1100
@@ -131,7 +131,7 @@
             NSString *returnCode=[words objectAtIndex:0];
 
             if([returnCode isEqualToString:@"NOACCESS"]) {
-                NSLog(@"Access denyied for %@ or host is not yours (Deactivating this user)",theUser);
+                NSLog(@"Access denied for %@ or host is not yours (Deactivating this user)",theUser);
                 [theUser setUserStatus:@"Bad Auth"];
                 [theUser setActive:NO];
                 return;
diff -u -r DNSUpdate2.8s.orig/DUExternalInterface.m DNSUpdate2.8s/DUExternalInterface.m
--- DNSUpdate2.8s.orig/DUExternalInterface.m	2003-09-24 09:50:44.000000000 +1000
+++ DNSUpdate2.8s/DUExternalInterface.m	2007-11-29 11:13:19.000000000 +1100
@@ -43,7 +43,7 @@
 }
 
 - (void)cacheAddress {
-    NSLog(@"Caching external address");
+    //NSLog(@"Caching external address");
     if([lastUpdate timeIntervalSinceNow]<=-EXTIPTEST_INTERVAL*60.0) {
         NSDictionary *serverReturn;
 
diff -u -r DNSUpdate2.8s.orig/DUMachBootstrapServer.m DNSUpdate2.8s/DUMachBootstrapServer.m
--- DNSUpdate2.8s.orig/DUMachBootstrapServer.m	2006-04-22 01:08:54.000000000 +1000
+++ DNSUpdate2.8s/DUMachBootstrapServer.m	2007-11-29 11:08:53.000000000 +1100
@@ -45,9 +45,9 @@
         NSLog(@"Can't get bootstrap port");
     } else {
 #if 0
-        NSLog(@"Getting priviledged bootstrap port");
+        NSLog(@"Getting privileged bootstrap port");
         if(bootstrap_parent(bPort, &_bootstrapPort)) {
-            NSLog(@"Can't get priviledged bootstrap port");
+            NSLog(@"Can't get privileged bootstrap port");
             mach_port_deallocate(mach_task_self(), bPort);
             _bootstrapPort = MACH_PORT_NULL;
         }
