--- HTTPLoader.m.orig	2006-04-16 22:24:32.000000000 +1000
+++ HTTPLoader.m	2009-09-11 19:54:58.000000000 +1000
@@ -137,10 +137,8 @@
 
     if(user&&password) {
         // get URL with authentification
-        char auth[512],theUser[256];
-
-        strcpy(theUser,[[NSString stringWithFormat:@"%@:%@", user, password] lossyCString]);
-        base64Encode(theUser,auth);
+        char auth[512];
+        base64Encode([[NSString stringWithFormat:@"%@:%@", user, password] lossyCString],auth);
 
         CFHTTPMessageSetHeaderFieldValue(requestMessage, CFSTR("Authorization"), (CFStringRef)[NSString stringWithFormat:@"Basic %s", auth]);
     }
