From ffdad4cf451075651a179cb73991e6646c5ec83f Mon Sep 17 00:00:00 2001
From: edzer <edzer.pebesma@uni-muenster.de>
Date: Wed, 26 Jun 2024 15:10:54 +0200
Subject: [PATCH] address #692

---
 R/init.R | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/R/init.R b/R/init.R
index dffab70d..953b4b39 100644
--- R/init.R
+++ R/init.R
@@ -17,4 +17,6 @@ setOldClass("stars_proxy")
 
 .onLoad = function(libname, pkgname) {
 	register_all_s3_methods() # dynamically registers non-imported pkgs (tidyverse) # nocov
+	if (nrow(sf::st_drivers("raster", "netCDF")) != 1)
+		message("Note: sf is linked to a GDAL version without netCDF driver, some tests or examples may fail")
 }
