--- CMakeLists.txt.orig	2012-10-22 22:41:04.000000000 -0500
+++ CMakeLists.txt	2015-08-26 00:00:37.000000000 -0500
@@ -35,12 +35,6 @@
 #INCLUDE_DIRECTORIES(archivers)
 
 IF(MACOSX)
-    # Fallback to older OS X on PowerPC to support wider range of systems...
-    IF(CMAKE_OSX_ARCHITECTURES MATCHES ppc)
-        ADD_DEFINITIONS(-DMAC_OS_X_VERSION_MIN_REQUIRED=1020)
-        SET(OTHER_LDFLAGS ${OTHER_LDFLAGS} " -mmacosx-version-min=10.2")
-    ENDIF(CMAKE_OSX_ARCHITECTURES MATCHES ppc)
-
     # Need these everywhere...
     ADD_DEFINITIONS(-fno-common)
     SET(OTHER_LDFLAGS ${OTHER_LDFLAGS} " -framework Carbon -framework IOKit")
@@ -49,7 +43,7 @@
 # Add some gcc-specific command lines.
 IF(CMAKE_COMPILER_IS_GNUCC)
     # Always build with debug symbols...you can strip it later.
-    ADD_DEFINITIONS(-g -pipe -Werror -fsigned-char)
+    ADD_DEFINITIONS(-g -pipe -fsigned-char)
 
     # Stupid BeOS generates warnings in the system headers.
     IF(NOT BEOS)
