--- CMakeLists.txt.orig	2011-08-05 17:46:19.596723448 -0400
+++ CMakeLists.txt	2011-08-05 17:46:27.744621586 -0400
@@ -44,7 +44,7 @@
 # Add some gcc-specific command lines.
 IF(CMAKE_COMPILER_IS_GNUCC)
     # Always build with debug symbols...you can strip it later.
-    ADD_DEFINITIONS(-g -pipe -Werror -fsigned-char)
+    ADD_DEFINITIONS(-g -pipe -fsigned-char)
 
     # Stupid BeOS generates warnings in the system headers.
     IF(NOT BEOS)
