--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,16 +78,17 @@
 # Enable C++17 support in Clang >= 6.0.0 or AppleClang >= 900
 # Enable C++14 support in 6.0.0 > Clang >= 3.5.0 or 900 > AppleClang >= 600
 # Fallback to C++11 if older version
+
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
         # See https://trac.macports.org/wiki/XcodeVersionInfo for Apple Clang version equivalences
         if(CLANG_VERSION VERSION_LESS "600")
-            set(MY_CXX_FLAGS "${MY_CXX_FLAGS} -std=c++11")
+            set(CMAKE_CXX_STANDARD 11)
         else(CLANG_VERSION VERSION_LESS "600")
             if(CLANG_VERSION VERSION_LESS "900")
-                set(MY_CXX_FLAGS "${MY_CXX_FLAGS} -std=c++14")
+                set(CMAKE_CXX_STANDARD 14)
             else(CLANG_VERSION VERSION_LESS "900")
-                set(MY_CXX_FLAGS "${MY_CXX_FLAGS} -std=c++17")
+                set(CMAKE_CXX_STANDARD 17)
             endif(CLANG_VERSION VERSION_LESS "900")
         endif(CLANG_VERSION VERSION_LESS "600")
     else(CMAKE_SYSTEM_NAME MATCHES "Darwin")
