--- GNUmakefile.orig
+++ GNUmakefile
@@ -3,9 +3,9 @@
 ############################################################################
 
 # Set this to change the target installation path
-DESTDIR ?= /usr/local
-BINDIR   = ${DESTDIR}/bin
-MANDIR   = ${DESTDIR}/share/man
+PREFIX ?= /usr/local
+BINDIR   = ${PREFIX}/bin
+MANDIR   = ${PREFIX}/share/man
 
 # Override the go executable
 GO = go
@@ -150,10 +150,10 @@
 
 .PHONY: install
 install: all
-	install -d $(BINDIR)
-	install direnv $(BINDIR)
-	install -d $(MANDIR)/man1
-	cp -R man/*.1 $(MANDIR)/man1
+	install -d $(DESTDIR)$(BINDIR)
+	install direnv $(DESTDIR)$(BINDIR)
+	install -d $(DESTDIR)$(MANDIR)/man1
+	cp -R man/*.1 $(DESTDIR)$(MANDIR)/man1
 
 .PHONY: dist
 dist: | $(base)
