diff -urbB conf/configure.in.fns conf/configure.in.fns
--- conf/configure.in.fns	Tue Sep 19 18:30:10 2000
+++ conf/configure.in.fns	Mon Jul 26 10:33:00 2004
@@ -117,7 +117,7 @@
 	for dir in $places; do
 		for file in $full_lib_name $simple_lib_name $other_simple_lib_name $simpler_lib_name
 		do
-			if test -r $dir/lib$file.so -o -r $dir/lib$file.a; then
+			if test -r $dir/lib$file.so -o -r $dir/lib$file.a -o -r $dir/lib$file.dylib; then
 				$5="-L$dir -l$file"
 				double_break=true
 				break
