--- lib/database/src/so/engines/SoCalcParse.y.orig	2008-12-07 18:31:30.000000000 -0500
+++ lib/database/src/so/engines/SoCalcParse.y	2008-12-07 18:32:19.000000000 -0500
@@ -119,7 +119,7 @@
 %type <expr> asgn primary_expression postfix_expression
 %type <expr> unary_expression
 %type <expr> multiplicative_expression additive_expression
-%type <expr> additive_expression relational_expression
+%type <expr> relational_expression
 %type <expr> equality_expression logical_AND_expression
 %type <expr> logical_OR_expression conditional_expression
 %type <list> args
