--- lib/database/src/so/nodes/SoText3.c++.orig	2005-08-31 13:30:03.000000000 -0500
+++ lib/database/src/so/nodes/SoText3.c++	2005-08-31 13:30:40.000000000 -0500
@@ -2291,7 +2291,7 @@
 	size_t outbytes = 2*inbytes+2;
 	char* output = (char*)UCSStrings[i];
     
-	if ((iconv(conversionCode, &input, &inbytes, &output, &outbytes) == (size_t)-1)){
+	if ((iconv(conversionCode, const_cast<const char**>(&input), &inbytes, &output, &outbytes) == (size_t)-1)){
 #ifdef DEBUG
 	    SoDebugError::post("SoOutlineFontCache::convertToUCS", 
 		"Error converting text to UCS-2");
