Find the Pashua library where MacPorts installed it.
--- example.sh.orig	2015-02-23 08:01:43.000000000 -0600
+++ example.sh	2020-06-19 17:22:08.000000000 -0500
@@ -3,7 +3,8 @@
 MYDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # Include pashua.sh to be able to use the 2 functions defined in that file
-source "$MYDIR/pashua.sh"
+#source "$MYDIR/pashua.sh"
+source "$(dirname $(dirname $(which port)))/share/Pashua-Binding-Bash/pashua.sh"
 
 # Define what the dialog should be like
 # Take a look at Pashua's Readme file for more info on the syntax
