diff -ur ../ruby-1.9.3-p551.org/Makefile.in ./Makefile.in
--- ../ruby-1.9.3-p551.org/Makefile.in	2014-02-06 14:20:52.000000000 +0900
+++ ./Makefile.in	2021-11-09 22:03:25.000000000 +0900
@@ -73,6 +73,9 @@
 BUILTIN_TRANSOBJS = @BUILTIN_TRANSOBJS@
 POSTLINK = @POSTLINK@
 
+PKG_CONFIG = @PKG_CONFIG@
+PKG_CONFIG_PATH = @PKG_CONFIG_PATH@
+
 RUBY_BASE_NAME=@RUBY_BASE_NAME@
 RUBY_PROGRAM_VERSION=@RUBY_PROGRAM_VERSION@
 RUBY_INSTALL_NAME=@RUBY_INSTALL_NAME@
diff -ur ../ruby-1.9.3-p551.org/common.mk ./common.mk
--- ../ruby-1.9.3-p551.org/common.mk	2014-05-14 17:38:00.000000000 +0900
+++ ./common.mk	2021-11-09 22:04:51.000000000 +0900
@@ -170,7 +170,8 @@
 
 EXTS_MK = exts.mk
 $(EXTS_MK): $(MKFILES) incs $(PREP) $(RBCONFIG) $(LIBRUBY)
-	@$(MINIRUBY) $(srcdir)/ext/extmk.rb --make="$(MAKE)" --command-output=$(EXTS_MK) $(EXTMK_ARGS) configure
+	@ PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) \
+	$(MINIRUBY) $(srcdir)/ext/extmk.rb --make="$(MAKE)" --command-output=$(EXTS_MK) $(EXTMK_ARGS) configure
 
 configure-ext: $(EXTS_MK)
 
diff -ur ../ruby-1.9.3-p551.org/configure ./configure
--- ../ruby-1.9.3-p551.org/configure	2014-11-13 22:42:20.000000000 +0900
+++ ./configure	2021-11-09 22:03:25.000000000 +0900
@@ -869,7 +869,8 @@
 CXX
 CXXFLAGS
 CCC
-CPP'
+CPP
+PKG_CONFIG_PATH'
 
 
 # Initialize some variables set by options.
