--- configure.orig	2014-11-13 22:42:20.000000000 +0900
+++ configure	2020-12-30 17:13:04.000000000 +0900
@@ -17826,11 +17826,11 @@
 _ACEOF
 
     cat >>confdefs.h <<_ACEOF
-#define RUBY_ARCH "universal-"RUBY_PLATFORM_OS
+#define RUBY_ARCH "universal-" RUBY_PLATFORM_OS
 _ACEOF
 
     cat >>confdefs.h <<_ACEOF
-#define RUBY_PLATFORM "universal."RUBY_PLATFORM_CPU"-"RUBY_PLATFORM_OS
+#define RUBY_PLATFORM "universal." RUBY_PLATFORM_CPU "-" RUBY_PLATFORM_OS
 _ACEOF
 
 else
