Index: st.c
===================================================================
--- st.c	(revision 39495)
+++ st.c	(working copy)
@@ -515,7 +515,7 @@
     st_index_t hash_val, bin_pos;
 
     if (table->entries_packed) {
-        int i;
+        st_index_t i;
         if (MORE_PACKABLE_P(table)) {
             i = table->num_entries++;
             table->bins[i*2] = (struct st_table_entry*)key;
Index: thread.c
===================================================================
--- thread.c	(revision 39495)
+++ thread.c	(working copy)
@@ -2161,7 +2161,7 @@
 static int
 vm_living_thread_num(rb_vm_t *vm)
 {
-    return vm->living_threads->num_entries;
+    return (int)vm->living_threads->num_entries;
 }
 
 int
