diff --git a/CMakeLists.txt b/CMakeLists.txt
index 822ea2a2..79eea9ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -287,7 +287,7 @@ macro (configure_files srcDir destDir)
 endmacro (configure_files)
 
 # Make a bundle for mac os
-if (APPLE)
+if (APPLE__NOPE)
     set (CMAKE_INSTALL_RPATH "@loader_path/../Libraries;@loader_path/../Frameworks")
     set(INPUTLEAP_BUNDLE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/dist/macos/bundle)
     set(INPUTLEAP_BUNDLE_DIR ${CMAKE_BINARY_DIR}/bundle)
