--- configure.orig	2015-11-15 08:07:06.000000000 -0600
+++ configure	2015-11-16 10:58:34.000000000 -0600
@@ -4011,7 +4011,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $MKDIR_P" >&5
 $as_echo "$MKDIR_P" >&6; }
 
-for ac_prog in gawk mawk nawk awk
+for ac_prog in gawk mawk awk
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
