--- src/quvi-0.9/quvi/Makefile.in.orig	2013-08-02 17:30:06.000000000 -0400
+++ src/quvi-0.9/quvi/Makefile.in	2013-08-02 17:30:22.000000000 -0400
@@ -562,7 +562,6 @@
 		echo "** Old headers found in $(libquvisubincludedir).  You should" ; \
 		echo "** remove the contents of this directory and type 'make " ; \
 		echo "** install' again." ; \
-		false ; \
 	fi
 
 # vim: set ts=2 sw=2 tw=72 expandtab:
