diff --git a/pydantic/v1/typing.py b/pydantic/v1/typing.py
index a690a053a2..5075f97f23 100644
--- pydantic/v1/typing.py
+++ pydantic/v1/typing.py
@@ -54,17 +54,21 @@
 
 
 if sys.version_info < (3, 9):
-
     def evaluate_forwardref(type_: ForwardRef, globalns: Any, localns: Any) -> Any:
         return type_._evaluate(globalns, localns)
 
-else:
-
+elif sys.version_info < (3, 12, 4):
     def evaluate_forwardref(type_: ForwardRef, globalns: Any, localns: Any) -> Any:
         # Even though it is the right signature for python 3.9, mypy complains with
         # `error: Too many arguments for "_evaluate" of "ForwardRef"` hence the cast...
         return cast(Any, type_)._evaluate(globalns, localns, set())
 
+else:
+    def evaluate_forwardref(type_: ForwardRef, globalns: Any, localns: Any) -> Any:
+        # For 3.12.4+ provide a default `recursive_guard` to resolve:
+        # TypeError: ForwardRef._evaluate() missing 1 required keyword-only argument: 'recursive_guard'
+        return cast(Any, type_)._evaluate(globalns, localns, set(), recursive_guard=set())
+
 
 if sys.version_info < (3, 9):
     # Ensure we always get all the whole `Annotated` hint, not just the annotated type.
