--- CMakeLists.txt.orig	2019-04-20 08:28:43.000000000 +0900
+++ CMakeLists.txt	2019-07-03 17:16:53.000000000 +0900
@@ -35,10 +35,10 @@
 #
 cmake_minimum_required(VERSION 3.8)
 
-if (UNIX)
-    set(CMAKE_C_COMPILER "clang")
-    set(CMAKE_CXX_COMPILER "clang++")
-endif()
+#if (UNIX)
+#    set(CMAKE_C_COMPILER "clang")
+#    set(CMAKE_CXX_COMPILER "clang++")
+#endif()
 
 set(PROJECT_NAME ispc)
 set(ISPC_BUILD TRUE)
