Fix "Unknown architecture i386" on systems running the 32-bit kernel.
--- examples/cmake/AddISPCExample.cmake.orig	2019-04-19 18:28:43.000000000 -0500
+++ examples/cmake/AddISPCExample.cmake	2019-08-23 04:09:46.000000000 -0500
@@ -44,7 +44,7 @@
     set(ISPC_OBJ_NAME "${CMAKE_CURRENT_BINARY_DIR}/${ISPC_SRC_NAME}_ispc${CMAKE_CXX_OUTPUT_EXTENSION}")
     set(ISPC_FLAGS ${example_ISPC_FLAGS})
     if (UNIX)
-        execute_process( COMMAND bash "-c" "uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/arm.*/arm/ -e s/sa110/arm/" OUTPUT_VARIABLE ARCH)
+        execute_process( COMMAND bash "-c" "uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/i386/x86/ -e s/arm.*/arm/ -e s/sa110/arm/" OUTPUT_VARIABLE ARCH)
         string(STRIP ${ARCH} ARCH)
         execute_process( COMMAND getconf LONG_BIT OUTPUT_VARIABLE ARCH_BIT)
         string(STRIP ${ARCH_BIT} ARCH_BIT)
--- examples/common.mk.orig	2019-04-19 18:28:43.000000000 -0500
+++ examples/common.mk	2019-08-23 04:13:03.000000000 -0500
@@ -13,7 +13,7 @@
 ISPC_FLAGS+=-O2
 ISPC_HEADER=objs/$(ISPC_SRC:.ispc=_ispc.h)
 
-ARCH:=$(shell uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/arm.*/arm/ -e s/sa110/arm/)
+ARCH:=$(shell uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/i386/x86/ -e s/arm.*/arm/ -e s/sa110/arm/)
 
 ifeq ($(ARCH),x86)
   ISPC_OBJS=$(addprefix objs/, $(ISPC_SRC:.ispc=)_ispc.o)
--- examples/portable/common_cpu.mk.orig	2019-04-19 18:28:43.000000000 -0500
+++ examples/portable/common_cpu.mk	2019-08-23 04:14:22.000000000 -0500
@@ -17,7 +17,7 @@
 ISPC_FLAGS+=-O2
 ISPC_HEADER=objs/$(ISPC_SRC:.ispc=_ispc.h)
 
-ARCH:=$(shell uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/arm.*/arm/ -e s/sa110/arm/)
+ARCH:=$(shell uname -m | sed -e s/x86_64/x86/ -e s/i686/x86/ -e s/i386/x86/ -e s/arm.*/arm/ -e s/sa110/arm/)
 
 ifeq ($(ARCH),x86)
   ISPC_OBJS=$(addprefix objs/, $(ISPC_SRC:.ispc=)_ispc.o)
