--- configure.ac.orig	2006-08-22 07:56:08.000000000 +1000
+++ configure.ac	2010-07-17 01:45:44.000000000 +1000
@@ -218,6 +218,7 @@
 dnl End of command line options configuration
 
 dnl Start DX configuration
+AC_PROG_CXX
 AC_LANG([C++])
 AC_CHECK_HEADERS(stdlib.h)
 AC_LANG([C])
@@ -276,7 +277,6 @@
 
 dnl Checks for programs.
 dnl AC_PROG_RANLIB
-AC_PROG_CXX
 if test -n "$CXXCPP" ; then
     ac_cv_prog_CXXCPP=$CXXCPP
 fi
