--- src/uipp/base/Application.h.orig	2005-11-10 05:26:29.000000000 +1100
+++ src/uipp/base/Application.h	2012-06-15 02:19:19.000000000 +1000
@@ -49,7 +49,7 @@ class Application : public UIComponent, 
     //
     // The main program needs to access protected member functions.
     //
-    friend int main(unsigned int argc,
+    friend int main(int argc,
 		     char**       argv);
 
 
--- src/uipp/dxui/Main.C.orig	2005-12-08 04:50:53.000000000 +1100
+++ src/uipp/dxui/Main.C	2012-06-15 02:23:01.000000000 +1000
@@ -48,9 +48,10 @@ extern unsigned long _etext;
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
  
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
+    unsigned int argc_unsigned = argc;
 #if defined(HAVE_HCLXMINIT)
 // The following is not needed with the recent Exceed XDK. If you're
 // using an XDK < 11.0 then uncomment the following command.
@@ -104,7 +105,7 @@ int main(unsigned int argc,
 
     // add *tearOffModel:: XmTEAR_OFF_ENABLED/XmTEAR_OFF_DISABLED
     XmRepTypeInstallTearOffModelConverter();
-    if (!theApplication->initialize(&argc, argv))
+    if (!theApplication->initialize(&argc_unsigned, argv))
 	exit(1);
 
     theApplication->handleEvents();
--- src/uipp/mb/Main.C.orig	2000-05-21 03:49:40.000000000 +1000
+++ src/uipp/mb/Main.C	2012-06-15 02:22:27.000000000 +1000
@@ -22,9 +22,10 @@ extern "C" void HCLXmInit();
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
+    unsigned int argc_unsigned = argc;
 #if defined(HAVE_HCLXMINIT)
     HCLXmInit();
 #endif
@@ -39,7 +40,7 @@ int main(unsigned int argc,
 	theApplication = new MBApplication("MB");
     }
 
-    theApplication->initialize(&argc, argv);
+    theApplication->initialize(&argc_unsigned, argv);
     theApplication->handleEvents();
 
     delete theApplication;
--- src/uipp/prompter/Main.C.orig	2000-05-17 04:52:59.000000000 +1000
+++ src/uipp/prompter/Main.C	2012-06-15 02:21:29.000000000 +1000
@@ -17,9 +17,10 @@ extern "C" void HCLXmInit();
 #endif
 
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
+    unsigned int argc_unsigned = argc;
 #if defined(HAVE_HCLXMINIT)
      HCLXmInit();
 #endif
@@ -34,7 +35,7 @@ int main(unsigned int argc,
 	theApplication = new GARApplication("GAR");
     }
 
-    theApplication->initialize(&argc, argv);
+    theApplication->initialize(&argc_unsigned, argv);
     theApplication->handleEvents();
 
     delete theApplication;
--- src/uipp/startup/Main.C.orig	2000-05-17 04:53:07.000000000 +1000
+++ src/uipp/startup/Main.C	2012-06-15 02:21:18.000000000 +1000
@@ -14,9 +14,10 @@
 extern "C" void HCLXmInit();
 #endif
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
+    unsigned int argc_unsigned = argc;
 #if defined(HAVE_HCLXMINIT)
     HCLXmInit();
 #endif
@@ -31,7 +32,7 @@ int main(unsigned int argc,
 	theApplication = new StartupApplication("Startup");
     }
 
-    theApplication->initialize(&argc, argv);
+    theApplication->initialize(&argc_unsigned, argv);
     theApplication->handleEvents();
 
     delete theApplication;
--- src/uipp/tutor/Main.C.orig	2000-05-17 04:53:15.000000000 +1000
+++ src/uipp/tutor/Main.C	2012-06-15 02:20:48.000000000 +1000
@@ -31,9 +31,10 @@ extern "C" void HCLXmInit();
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
  
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
+    unsigned int argc_unsigned = argc;
 #if defined(HAVE_HCLXMINIT)
     HCLXmInit();
 #endif
@@ -58,7 +59,7 @@ int main(unsigned int argc,
 	theApplication = new TutorApplication("DXTutor");
     }
 
-    if (!theApplication->initialize(&argc, argv))
+    if (!theApplication->initialize(&argc_unsigned, argv))
 	exit(1);
 
     theApplication->handleEvents();
