--- src/uipp/base/Application.h.orig	2005-11-10 05:26:29.000000000 +1100
+++ src/uipp/base/Application.h	2012-06-15 01:20:01.000000000 +1000
@@ -49,7 +49,7 @@ class Application : public UIComponent, 
     //
     // The main program needs to access protected member functions.
     //
-    friend int main(unsigned int argc,
+    friend int main(int argc,
 		     char**       argv);
 
 
--- src/uipp/dxui/Main.C.orig	2005-12-08 04:50:53.000000000 +1100
+++ src/uipp/dxui/Main.C	2012-06-15 01:23:08.000000000 +1000
@@ -48,7 +48,7 @@ extern unsigned long _etext;
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
  
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
 #if defined(HAVE_HCLXMINIT)
--- src/uipp/mb/Main.C.orig	2000-05-21 03:49:40.000000000 +1000
+++ src/uipp/mb/Main.C	2012-06-15 01:24:36.000000000 +1000
@@ -22,7 +22,7 @@ extern "C" void HCLXmInit();
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
 #if defined(HAVE_HCLXMINIT)
--- src/uipp/prompter/Main.C.orig	2000-05-17 04:52:59.000000000 +1000
+++ src/uipp/prompter/Main.C	2012-06-15 01:26:01.000000000 +1000
@@ -17,7 +17,7 @@ extern "C" void HCLXmInit();
 #endif
 
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
 #if defined(HAVE_HCLXMINIT)
--- src/uipp/startup/Main.C.orig	2000-05-17 04:53:07.000000000 +1000
+++ src/uipp/startup/Main.C	2012-06-15 01:26:42.000000000 +1000
@@ -14,7 +14,7 @@
 extern "C" void HCLXmInit();
 #endif
 
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
 #if defined(HAVE_HCLXMINIT)
--- src/uipp/tutor/Main.C.orig	2000-05-17 04:53:15.000000000 +1000
+++ src/uipp/tutor/Main.C	2012-06-15 01:27:42.000000000 +1000
@@ -31,7 +31,7 @@ extern "C" void HCLXmInit();
 //
 const char *AssertMsgString = "Internal error detected at \"%s\":%d.\n";
  
-int main(unsigned int argc,
+int main(int argc,
 	  char**       argv)
 {
 #if defined(HAVE_HCLXMINIT)
