http://www.opendx.org/bugs/view.php?id=251
--- src/exec/dpexec/packet.c.orig 2006-04-27 10:34:53.000000000 -0500
+++ src/exec/dpexec/packet.c 2010-03-05 10:48:41.000000000 -0600
@@ -143,6 +143,7 @@
	{
	    if (sts > 0 && (!tmpbufferused || *tmpbufferused == 0))
	    {
+ sts=0; // MHM change in case IOCTL fails
		if(SFILEIoctl(_dxd_exSockFD, FIONBIO, &one) >= 0)
	        sts = writeToSFILE(_dxd_exSockFD, packet, length);
		SFILEIoctl(_dxd_exSockFD, FIONBIO, &zero);
