--- src/uipp/java/Makefile.am~	2018-11-03 21:02:54.000000000 -0700
+++ src/uipp/java/Makefile.am	2018-11-03 21:13:11.000000000 -0700
@@ -165,7 +165,7 @@
 	        -c -Dintelnt $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
 	else \
 		$(CC) $(DXABI) -o $$javaobj/DXLink.o -I$$srcincs -Iserver \
-		-I$(BUILDBASE)/include $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
+		-c -I$(BUILDBASE)/include $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
 	fi ;
 
 ##SAMPLE_OBJS = AS400VSMonth.class AutoInsurance.class CCNet2.class \
--- src/uipp/java/Makefile.in~	2018-11-03 21:16:38.000000000 -0700
+++ src/uipp/java/Makefile.in	2018-11-03 21:17:49.000000000 -0700
@@ -715,7 +715,7 @@
 	        -c -Dintelnt $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
 	else \
 		$(CC) $(DXABI) -o $$javaobj/DXLink.o -I$$srcincs -Iserver \
-		-I$(BUILDBASE)/include -c $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
+		-c -I$(BUILDBASE)/include -c $(JNI_CFLAGS) @JINC@ $(srcdir)/server/DXLink.c ; \
 	fi ;
 
 distclean-local:
