From: Carl Hayter <zengargoyle@gmail.com>
Subject: perldoc: look for .pod in addition to .pm files
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=708173
Origin: vendor, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=708173
Forwarded: no

---
 helpers/perl |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- helpers/perl
+++ helpers/perl
@@ -24,8 +24,8 @@ sub print_modules_real {
     chdir($dir) or return;
 
     # print each file
-    foreach my $file (glob('*.pm')) {
-        $file =~ s/\.pm$//;
+    foreach my $file (glob('*.{pm,pod}')) {
+        $file =~ s/\.(?:pm|pod)$//;
         my $module = $base . $file;
         next if $module !~ /^\Q$word/;
         next if $seen{$module}++;
