From: Ville Skyttä <ville.skytta@iki.fi>
Subject: don't return from sourced script
Origin: upstream, http://anonscm.debian.org/gitweb/?p=bash-completion/bash-completion.git;a=commitdiff;h=867282a
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=741657
Forwarded: not-needed

---
 bash_completion.sh.in |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- bash_completion.sh.in
+++ bash_completion.sh.in
@@ -1,5 +1,5 @@
 # Check for interactive bash and that we haven't already been sourced.
-[ -z "$BASH_VERSION" -o -z "$PS1" -o -n "$BASH_COMPLETION_COMPAT_DIR" ] && return
+if [ -n "$BASH_VERSION" -a -n "$PS1" -a -z "$BASH_COMPLETION_COMPAT_DIR" ]; then
 
 # Check for recent enough version of bash.
 bash=${BASH_VERSION%.*}; bmajor=${bash%.*}; bminor=${bash#*.}
@@ -12,3 +12,5 @@ if [ $bmajor -gt 4 ] || [ $bmajor -eq 4
     fi
 fi
 unset bash bmajor bminor
+
+fi
