--- completions/man.orig	2011-02-10 20:00:21.000000000 +0100
+++ completions/man	2011-02-10 20:00:58.000000000 +0100
@@ -27,7 +27,7 @@
     fi
 
     uname=$( uname -s )
-    if [[ $uname == @(Linux|GNU|GNU/*|FreeBSD|Cygwin|CYGWIN_*) ]]; then
+    if [[ $uname == @(Linux|GNU|GNU/*|FreeBSD|Darwin|Cygwin|CYGWIN_*) ]]; then
         manpath=$( manpath 2>/dev/null || command man --path )
     else
         manpath=$MANPATH
