--- contrib/man.orig	2010-03-04 16:02:52.000000000 +0100
+++ contrib/man	2010-03-04 16:03:38.000000000 +0100
@@ -28,7 +28,7 @@
     fi
 
     uname=$( uname -s )
-    if [[ $uname == @(Linux|GNU|GNU/*|FreeBSD|Cygwin|CYGWIN_*) ]]; then
+    if [[ $uname == @(Linux|GNU|GNU/*|FreeBSD|Darwin|Cygwin|CYGWIN_*) ]]; then
         manpath=$( manpath 2>/dev/null || command man --path )
     else
         manpath=$MANPATH
