--- h5py/api_functions.txt
+++ h5py/api_functions.txt
@@ -535,7 +535,6 @@ hdf5:
 
   MPI herr_t H5Pset_fapl_mpio(hid_t fapl_id, MPI_Comm comm, MPI_Info info)
   MPI herr_t H5Pget_fapl_mpio(hid_t fapl_id, MPI_Comm *comm, MPI_Info *info)
-  MPI herr_t H5Pset_fapl_mpiposix(hid_t fapl_id, MPI_Comm comm, hbool_t hints)
   MPI 1.8.9 herr_t H5Fset_mpi_atomicity(hid_t file_id, hbool_t flag)
   MPI 1.8.9 herr_t H5Fget_mpi_atomicity(hid_t file_id, hbool_t *flag)
 
--- h5py/h5fd.pxd
+++ h5py/h5fd.pxd
@@ -33,7 +33,7 @@ cdef extern from "hdf5.h":
 # hid_t H5FD_GASS  not in 1.8.X
   hid_t H5FD_LOG
   hid_t H5FD_MPIO
-  hid_t H5FD_MPIPOSIX
+#  hid_t H5FD_MPIPOSIX  removed in 1.8.13
   hid_t H5FD_MULTI
   hid_t H5FD_SEC2
   hid_t H5FD_STDIO
--- h5py/h5fd.pyx
+++ h5py/h5fd.pyx
@@ -31,7 +31,7 @@ CORE = H5FD_CORE
 FAMILY = H5FD_FAMILY
 LOG = H5FD_LOG
 MPIO = H5FD_MPIO
-MPIPOSIX = H5FD_MPIPOSIX
+MPIPOSIX = -1
 MULTI = H5FD_MULTI
 SEC2 = H5FD_SEC2
 STDIO = H5FD_STDIO
--- h5py/h5p.pyx
+++ h5py/h5p.pyx
@@ -976,14 +976,9 @@ cdef class PropFAID(PropInstanceID):
             return (pycomm, pyinfo)
 
         def set_fapl_mpiposix(self, Comm comm not None, bint use_gpfs_hints=0):
-            """ (Comm comm, BOOL use_gpfs_hints=0)
-
-            Set MPI-POSIX driver.
-
-            Comm: An mpi4py.MPI.Comm instance
-            use_gpfs_hints: Enable internal hints for GPFS file system
+            """ Obsolete.
             """
-            H5Pset_fapl_mpiposix(self.id, comm.ob_mpi, use_gpfs_hints)
+            raise RuntimeError("MPI-POSIX driver is broken; removed in h5py 2.3.1")
 
     def get_mdc_config(self):
         """() => CacheConfig
