--- pyproject.toml.orig	2024-09-27 12:13:25
+++ pyproject.toml	2024-09-27 12:13:37
@@ -1,7 +1,7 @@
 [build-system]
 requires = [
     "Cython >=0.29.31,<4",
-    "numpy >=2.0.0, <3",
+    "numpy",
     "pkgconfig",
     "setuptools >=61",
 ]
--- setup_build.py.orig	2025-06-06 15:46:36
+++ setup_build.py	2025-06-06 15:47:18
@@ -24,7 +24,7 @@
     return op.abspath(op.join(op.dirname(__file__), *args))
 
 
-MODULES_NUMPY2 = ['_npystrings']
+MODULES_NUMPY2 = [] #'_npystrings']
 MODULES = ['defs', '_errors', '_objects', '_proxy', 'h5fd', 'h5z',
             'h5', 'h5i', 'h5r', 'utils', '_selector',
             '_conv', 'h5t', 'h5s',
@@ -127,8 +127,8 @@
         settings = copy.deepcopy(settings)
         settings['libraries'] += EXTRA_LIBRARIES.get(module, [])
 
-        assert numpy.__version__ >= '2.0'  # See build dependencies in pyproject.toml
         if module in MODULES_NUMPY2:
+            assert numpy.__version__ >= '2.0'  # See build dependencies in pyproject.toml
             # Enable NumPy 2.0 C API for modules that require it.
             # These modules will not be importable when NumPy 1.x is installed.
             settings['define_macros'].append(('NPY_TARGET_VERSION', 0x00000012))
