--- src/engine/Makefile.am.orig	2026-04-10 12:55:55
+++ src/engine/Makefile.am	2026-04-18 16:01:10
@@ -143,15 +143,17 @@
 		storj/storjcontrolsocket.h
 endif
 
-if !HAVE_LIBPUGIXML
-libfzclient_private_la_SOURCES += ../pugixml/pugixml.cpp
-endif
-
 libfzclient_private_la_CXXFLAGS = -fvisibility=hidden
 libfzclient_private_la_LDFLAGS = -no-undefined -release $(ENGINE_VERSION_MAJOR).$(ENGINE_VERSION_MINOR).$(ENGINE_VERSION_MICRO)
 libfzclient_private_la_LDFLAGS += $(LIBFZSSH_LIBS) $(LIBFILEZILLA_LIBS)
 libfzclient_private_la_LDFLAGS += $(IDN_LIB)
 
+if HAVE_LIBPUGIXML
+libfzclient_private_la_LDFLAGS += $(PUGIXML_LIBS)
+else
+libfzclient_private_la_SOURCES += ../pugixml/pugixml.cpp
+endif
+
 if FZ_MAC
 libfzclient_private_la_LDFLAGS += -framework CoreServices
 endif
