--- configure.orig	2008-02-22 15:25:27.000000000 +1100
+++ configure	2012-10-06 03:17:48.000000000 +1000
@@ -1725,7 +1725,7 @@
 echo "$as_me: error: Wyrd requires OCaml version 3.08 or greater." >&2;}
    { (exit 1); exit 1; }; }
 else
-   if test $OCAMLMINORVERSION -lt 8 ; then
+   if test $OCAMLMAJORVERSION -eq 3 -a $OCAMLMINORVERSION -lt 8 ; then
       { { echo "$as_me:$LINENO: error: Wyrd requires OCaml version 3.08 or greater." >&5
 echo "$as_me: error: Wyrd requires OCaml version 3.08 or greater." >&2;}
    { (exit 1); exit 1; }; }
