--- ./baseboards/unix.exp.orig	2017-10-15 20:44:19.000000000 -0700
+++ ./baseboards/unix.exp	2020-11-02 19:16:18.000000000 -0800
@@ -39,3 +39,9 @@ if { [istarget "alpha*-*"] } {
 if { [istarget "sparc64-*-linux-gnu"] } {
     set_board_info gdb,no_hardware_watchpoints 1
 }
+
+# Avoid multitop on Darwin, where it's at best useless, and attempts to
+# determine it may cause failures with the use of the -dumpspecs option.
+if { [istarget "*-*-darwin*"] } {
+    set_board_info multitop ""
+}
