--- cmake/wasi-sdk-sysroot.cmake.orig	2026-05-01 22:25:57
+++ cmake/wasi-sdk-sysroot.cmake	2026-05-01 22:26:02
@@ -354,13 +354,13 @@
     USES_TERMINAL_PATCH ON
     PATCH_COMMAND
       ${CMAKE_COMMAND} -E chdir .. bash -c
-        "git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-168449.patch || git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-168449.patch -R --check"
+        "patch -N -p1 < ${CMAKE_SOURCE_DIR}/src/llvm-pr-168449.patch || true"
     COMMAND
       ${CMAKE_COMMAND} -E chdir .. bash -c
-        "git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-186054.patch || git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-186054.patch -R --check"
+        "patch -N -p1 < ${CMAKE_SOURCE_DIR}/src/llvm-pr-186054.patch || true"
     COMMAND
       ${CMAKE_COMMAND} -E chdir .. bash -c
-        "git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-185770.patch || git apply ${CMAKE_SOURCE_DIR}/src/llvm-pr-185770.patch -R --check"
+        "patch -N -p1 < ${CMAKE_SOURCE_DIR}/src/llvm-pr-185770.patch || true"
   )
   add_dependencies(libcxx-${target} libcxx-${target}${target_suffix}-build)
 endfunction()
