--- ltmain.sh.orig	2016-08-29 05:15:09.000000000 -0700
+++ ltmain.sh	2016-09-01 05:17:06.000000000 -0700
@@ -7070,7 +7070,7 @@
 	continue
 	;;
 
-      -mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
+      -mt|-mthreads|-kthread|-Kthread|-lpthread|-pthreads|--thread-safe \
       |-threads|-fopenmp|-openmp|-mp|-xopenmp|-omp|-qsmp=*)
 	func_append compiler_flags " $arg"
 	func_append compile_command " $arg"
@@ -7604,7 +7604,7 @@
 	lib=
 	found=false
 	case $deplib in
-	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe \
+	-mt|-mthreads|-kthread|-Kthread|-lpthread|-pthreads|--thread-safe \
         |-threads|-fopenmp|-openmp|-mp|-xopenmp|-omp|-qsmp=*)
 	  if test prog,link = "$linkmode,$pass"; then
 	    compile_deplibs="$deplib $compile_deplibs"
