--- pango/pangocoretext-fontmap.c.orig	2016-09-01 18:23:39.000000000 -0700
+++ pango/pangocoretext-fontmap.c	2016-09-01 18:28:09.000000000 -0700
@@ -119,7 +119,7 @@
  */
 
 static const PangoCTWeight ct_weight_map[] = {
-    { ct_weight_min, PANGO_WEIGHT_THIN },
+    { -0.7, PANGO_WEIGHT_THIN },
     { -0.5, PANGO_WEIGHT_ULTRALIGHT },
     { -0.23, PANGO_WEIGHT_LIGHT },
     { -0.115, PANGO_WEIGHT_SEMILIGHT },
@@ -128,7 +128,7 @@
     {  0.3, PANGO_WEIGHT_SEMIBOLD },
     {  0.4, PANGO_WEIGHT_BOLD },
     {  0.6, PANGO_WEIGHT_ULTRABOLD },
-    {  ct_weight_max, PANGO_WEIGHT_HEAVY }
+    {  0.8, PANGO_WEIGHT_HEAVY }
 };
 
 static const char *
