--- configure.orig	2009-07-21 06:59:52.000000000 +1000
+++ configure	2009-10-22 06:14:52.000000000 +1100
@@ -19806,12 +19806,14 @@
 $as_echo "$ac_cv_header_Carbon_Carbon_h" >&6; }
 
 fi
-if test "x$ac_cv_header_Carbon_Carbon_h" = x""yes; then
-  have_atsui=true
-else
-  have_atsui=false
-fi
 
+if test "x$have_atsui" = "x"; then
+  if test "x$ac_cv_header_Carbon_Carbon_h" = x""yes; then
+    have_atsui=true
+  else
+    have_atsui=false
+  fi
+fi
 
 
 #
@@ -23206,7 +23208,8 @@
 if test -n "$CONFIG_FILES"; then
 
 
-ac_cr=''
+ac_cr='
+'
 ac_cs_awk_cr=`$AWK 'BEGIN { print "a\rb" }' </dev/null 2>/dev/null`
 if test "$ac_cs_awk_cr" = "a${ac_cr}b"; then
   ac_cs_awk_cr='\\r'
