--- src/core/misc.cpp.orig	2009-10-12 10:37:32.000000000 +0200
+++ src/core/misc.cpp	2009-10-12 10:51:35.000000000 +0200
@@ -64,22 +64,11 @@
     #define EIT( x ) { OSErr err = x; if (err != noErr) throw 1; }
     try
     {
-        short vRefNum = 0;
-        long dirId;
-        EIT( ::FindFolder( kOnAppropriateDisk, 
+        FSRef fsref;
+        EIT( ::FSFindFolder( kOnAppropriateDisk, 
                            kApplicationSupportFolderType,
                            kDontCreateFolder, 
-                           &vRefNum, 
-                           &dirId ) );
-
-        // Now we have a vRefNum and a dirID - but *not* an Unix-Path as string.
-        // Lets make one based from this:
-        FSSpec fsspec;
-        EIT( ::FSMakeFSSpec( vRefNum, dirId, NULL, &fsspec ) );
-
-        // ...and build an FSRef based on thes FSSpec.
-        FSRef fsref;
-        EIT( ::FSpMakeFSRef( &fsspec, &fsref ) );
+                           &fsref ) );
 
         // ...then extract the Unix Path as a C-String from the FSRef
         unsigned char path[512];
