Index: Makefile.in
===================================================================
--- Makefile.in	(revision 996)
+++ Makefile.in	(working copy)
@@ -58,6 +58,7 @@
 -DHAVE_CUFFT=@CUFFT32_OK@ \
 -Wreturn-type -Wformat @M32@ -Wunused -Wcomment -Wchar-subscripts -Wshadow \
 -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror -ffast-math \
+$(CFLAGS) \
 -Wundef -fomit-frame-pointer -c -o
 
 CFLN	= -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
@@ -67,6 +68,7 @@
 -Wconversion \
 -Wreturn-type -Wformat @M32@ -Wunused -Wcomment -Wchar-subscripts -Wshadow \
 -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror -ffast-math \
+$(CFLAGS) \
 -Wundef -fomit-frame-pointer -c -o
 
 CFLOCL	= -O2 -DOSNUM=1 @X11_32_INC@ -W -Wall -DCPU=@CPU@ @OSSINC32@ \
@@ -76,6 +78,7 @@
 @OCLPROGS32@ -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DHAVE_CUFFT=@CUFFT32_OK@ \
 -Wreturn-type -Wformat @M32@ -Wunused -Wcomment -Wchar-subscripts -Wshadow \
 -Wuninitialized -Wparentheses -Werror -ffast-math \
+$(CFLAGS) \
 -Wundef -fomit-frame-pointer -c -o
 
 CFW	= -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=0 -DHAVE_CUFFT=0 \
@@ -82,6 +85,7 @@
 -DCPU=CPU_INTEL -DSERVER=@MAKE_SERVER@ -DHAVE_OSS=0 -DOPENCL_PRESENT=0 \
 @M32@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
+$(CFLAGS) \
 -ffast-math -Wundef -fomit-frame-pointer -c -o
 
 # -fsyntax-only -pedantic -pedantic-errors -w -Wextra -Wall -Waddress -Waggrega
@@ -91,6 +95,7 @@
 @M32@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wconversion \
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
+$(CFLAGS) \
 -ffast-math -Wundef -fomit-frame-pointer -c -o
 
 CFW64	= -O2 -DOSNUM=2 -W -Wall -Wimplicit -DIA64=1 -DHAVE_CUFFT=0 \
@@ -97,6 +102,7 @@
 -DCPU=CPU_INTEL -DSERVER=@MAKE_SERVER@ -DHAVE_OSS=0 -DOPENCL_PRESENT=0 \
 @M64@ -Wreturn-type -Wformat -Wunused -Wcomment -Wchar-subscripts \
 -Wshadow -Wuninitialized -Wparentheses -Wstrict-prototypes -Werror \
+$(CFLAGS) \
 -ffast-math -Wundef -fomit-frame-pointer -c -o
 
 CFL64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
@@ -106,6 +112,7 @@
 -Wimplicit -Wreturn-type -Wformat @M64@ -Wunused -Wcomment -DOSSD=@OSSHDR64@ \
 -Wchar-subscripts -Wshadow -Wuninitialized -Wparentheses @OSSINC64@ \
 -Wstrict-prototypes -Werror -ffast-math -Wundef -fomit-frame-pointer \
+$(CFLAGS) \
 -c -o
 
 CFLN64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
@@ -115,6 +122,7 @@
 -Wconversion -Wimplicit -Wreturn-type -Wformat @M64@ -Wunused -Wcomment \
 -Wchar-subscripts -Wshadow -Wuninitialized -Wparentheses \
 -Wstrict-prototypes -Werror -ffast-math -Wundef -fomit-frame-pointer \
+$(CFLAGS) \
 -c -o
 
 CFLOCL64	= -O2 -DOSNUM=1 @X11_64_INC@ -W -Wall -DHAVE_OSS=@OSS64_H@ \
@@ -124,6 +132,7 @@
 -Wimplicit -Wreturn-type -Wformat @M64@ -Wunused -Wcomment -DOSSD=@OSSHDR64@ \
 -Wchar-subscripts -Wshadow -Wuninitialized -Wparentheses @OSSINC64@ \
 -Werror -ffast-math -Wundef -fomit-frame-pointer \
+$(CFLAGS) \
 -c -o
 
 # Define dependencies for routines used both in Linux and Windows
@@ -1732,7 +1741,7 @@
 @LIBXEXT32_LD@ \
 @X11_32_LD@ \
 @CUFFTPATH32@ \
--lpthread @LIBRT@ -lm -ldl $(LIBMACOSX)
+-lpthread $(LDFLAGS) @LIBRT@ -lm -ldl $(LIBMACOSX)
 
 	@echo " "
 	@echo "You may want to disable pulseaudio in Ubuntu with this command:"
@@ -2083,7 +2092,7 @@
 @LIBXEXT64_LD@ \
 @X11_64_LD@ \
 @CUFFTPATH64@ \
--lpthread @LIBRT@ -lm -ldl $(LIBMACOSX)
+-lpthread $(LDFLAGS) @LIBRT@ -lm -ldl $(LIBMACOSX)
 
 	@echo " "
 	@echo "You may want to disable pulseaudio in Ubuntu with this command:"
