diff --git configure.in configure.in
index 79547e7..53d0268 100644
--- configure.in
+++ configure.in
@@ -74,19 +74,6 @@ AC_SUBST(S_OS_TYPE)
 if test "${CC}" = ""; then
   CC=gcc
 fi
-AC_PATH_PROG(HAVE_GCC, ${CC} )
-if test "$HAVE_GCC" = ""; then
-  echo -------------------------------------------------------
-  echo Can not find ${CC} , the GNU C compiler.
-  echo Install gcc and development libraries. Then run configure again.
-  echo Debian, Ubuntu: apt-get install gcc
-  echo Old Fedora: yum install gcc
-  echo Fedora 22 and later: dnf install gcc
-  echo Mandriva, Mageia: urpmi gcc
-  echo SuSE: zypper install gcc
-  echo Sabayon: equo install sys-devel/gcc
-  exit
-fi
 ${CC} $DARWIN libfind1.c -c -o libfind1.ol 2> x.x
 chmod 777 x.x
 Y=`cat x.x`
