Use MacPorts noahstrap.
--- bin/noah.in.orig	2018-03-04 22:20:42.000000000 -0600
+++ bin/noah.in	2020-05-01 00:07:50.000000000 -0500
@@ -34,7 +34,7 @@
   $root = "$noahdir/tree";
   if (! -d $root) {
     prompt_yn("Noah is installing the initial filesystem in ~/.noah/tree. Proceed? [Y/n]", "y") or exit 1;
-    system "sudo noahstrap -p ubuntu $root";
+    system "sudo @CMAKE_INSTALL_PREFIX@/bin/noahstrap -p ubuntu $root";
   }
 }
 
