--- plplot.pd.orig	2020-07-16 10:09:48.000000000 -0700
+++ plplot.pd	2020-10-05 18:42:31.000000000 -0700
@@ -2340,6 +2340,7 @@
 
 # Necessary includes for .xs file
 pp_addhdr(<<'EOH');
+#define  PL_DEPRECATED
 #include <plplot.h>
 #include <plplotP.h>
 #include <plevent.h>
--- t/x15c.c.orig	2020-10-05 18:57:56.000000000 -0700
+++ t/x15c.c	2020-10-05 19:00:15.000000000 -0700
@@ -5,6 +5,7 @@
 //      31 Aug 1993
 //
 
+#define PL_DEPRECATED
 #include "plcdemos.h"
 
 #define XPTS    35              // Data points in x
