From e339056c2591476b180d5e7c3f5992a3ce3d5b69 Mon Sep 17 00:00:00 2001
From: Ryan Kavanagh <rak@rak.ac>
Date: Thu, 29 Jul 2021 18:43:55 -0400
Subject: [PATCH] Install sandman

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git Makefile Makefile
index 373e7d5..00e4efb 100644
--- Makefile
+++ Makefile
@@ -27,7 +27,7 @@ OBJS += xdg.o
 
 dev: tags all
 
-all: catgirl
+all: catgirl scripts/sandman
 
 catgirl: ${OBJS}
 	${CC} ${LDFLAGS} ${OBJS} ${LDLIBS} -o $@
@@ -40,7 +40,7 @@ tags: *.[ch]
 clean:
 	rm -f catgirl ${OBJS} tags
 
-install: catgirl catgirl.1
+install: catgirl catgirl.1 install-sandman
 	install -d ${DESTDIR}${BINDIR} ${DESTDIR}${MANDIR}/man1
 	install catgirl ${DESTDIR}${BINDIR}
 	install -m 644 catgirl.1 ${DESTDIR}${MANDIR}/man1
-- 
2.32.0

