--- Makefile.orig	2017-12-12 11:29:54.000000000 -0700
+++ Makefile	2019-01-27 04:52:46.000000000 -0700
@@ -2,16 +2,18 @@
 # Makefile for bowtie, bowtie-build, bowtie-inspect
 #
 
-prefix = /usr/local
+include Makefile.macports.inc
+
+prefix ?= /usr/local
 bindir = $(prefix)/bin
 
 SEQAN_DIR = ./SeqAn-1.1
 # treat SeqAn as a sysdir to suppress warnings
 SEQAN_INC = -isystem $(SEQAN_DIR)
 INC = $(if $(RELEASE_BUILD),-I$(CURDIR)/.include) $(SEQAN_INC) -I third_party
-CPP = g++
-CXX = $(CPP)
-CC = gcc
+CXX ?= g++
+CPP ?= $(CXX)
+CC ?= gcc
 LIBS = $(LDFLAGS) $(if $(RELEASE_BUILD),-L$(CURDIR)/.lib) -lz
 HEADERS = $(wildcard *.h)
 BOWTIE_MM = 1
@@ -19,8 +21,8 @@
 EXTRA_FLAGS =
 EXTRA_CFLAGS =
 EXTRA_CXXFLAGS =
-CFLAGS += $(EXTRA_CFLAGS)
-CXXFLAGS += $(EXTRA_CXXFLAGS)
+CFLAGS += $(EXTRA_CFLAGS) $(CPPFLAGS)
+CXXFLAGS += $(EXTRA_CXXFLAGS) $(CPPFLAGS)
 WARNING_FLAGS = -Wall -Wno-unused-parameter -Wno-reorder \
 				-Wno-unused-local-typedefs
 
@@ -49,14 +51,6 @@
 MACOS =
 ifneq (,$(findstring Darwin,$(shell uname)))
     MACOS = 1
-	ifneq (,$(findstring 13,$(shell uname -r)))
-		CPP = clang++
-		CC = clang
-		override EXTRA_FLAGS += -stdlib=libstdc++
-	endif
-	ifeq (1, $(RELEASE_BUILD))
-		EXTRA_FLAGS += -mmacosx-version-min=10.9
-	endif
 endif
 
 LINUX =
@@ -159,29 +153,8 @@
 SEARCH_FRAGMENTS = $(wildcard search_*_phase*.c)
 VERSION = $(shell cat VERSION)
 
-BITS=32
-ifeq (x86_64,$(shell uname -m))
-	BITS=64
-endif
-# msys will always be 32 bit so look at the cpu arch instead.
-ifneq (,$(findstring AMD64,$(PROCESSOR_ARCHITEW6432)))
-	ifeq (1,$(MINGW))
-		BITS=64
-	endif
-endif
-
-ifeq (1,$(LINUX))
-    ifeq (x86_64, $(shell uname -p))
-        BITS=64
-    endif
-endif
-
-ifeq (32,$(BITS))
-    $(error bowtie2 compilation requires a 64-bit platform )
-endif
-
-DEBUG_FLAGS = -O0 -g3 -m64
-RELEASE_FLAGS = -O3 -m64
+DEBUG_FLAGS =
+RELEASE_FLAGS =
 NOASSERT_FLAGS = -DNDEBUG
 FILE_FLAGS = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 
