Respect MacPorts compiler & compiler flags.
--- Makefile.orig	2012-12-14 20:15:42.000000000 -0700
+++ Makefile	2019-01-19 07:27:34.000000000 -0700
@@ -2,14 +2,16 @@
 # Makefile for bowtie, bowtie-build, bowtie-inspect
 #
 
+include Makefile.macports.inc
+
 SEQAN_DIR = SeqAn-1.1
 SEQAN_INC = -I $(SEQAN_DIR)
 INC = $(SEQAN_INC)
 GCC_PREFIX = $(shell dirname `which gcc`)
 GCC_SUFFIX =
-CC = $(GCC_PREFIX)/gcc$(GCC_SUFFIX)
-CPP = $(GCC_PREFIX)/g++$(GCC_SUFFIX)
-CXX = $(CPP)
+CC ?= $(GCC_PREFIX)/gcc$(GCC_SUFFIX)
+CXX ?= $(GCC_PREFIX)/g++$(GCC_SUFFIX)
+CPP ?= $(CXX)
 HEADERS = $(wildcard *.h)
 BOWTIE_PTHREADS = 1
 BOWTIE_MM = 1
@@ -99,8 +101,8 @@
 BITS_FLAG = -m64
 endif
 
-DEBUG_FLAGS = -O0 -g3 $(BITS_FLAG)
-RELEASE_FLAGS = -O3 $(BITS_FLAG)
+DEBUG_FLAGS = $(BITS_FLAG)
+RELEASE_FLAGS = $(BITS_FLAG)
 NOASSERT_FLAGS = -DNDEBUG
 FILE_FLAGS = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 
