--- src/dune	2023-08-10 15:27:45.000000000 +0800
+++ src/dune	2023-10-27 18:20:47.000000000 +0800
@@ -4,4 +4,4 @@
  (wrapped false)
  (flags
   (:standard -safe-string))
- (libraries react result uchar uutf uucp uuseg))
+ (libraries react result uutf uucp uuseg))

--- dune-project	2023-08-10 15:27:45.000000000 +0800
+++ dune-project	2023-10-27 18:22:16.000000000 +0800
@@ -29,7 +29,6 @@
    (>= 4.02.3))
   react
   result
-  uchar
   uutf
   (uucp
    (>= 2.0.0))

--- zed.opam	2023-10-27 18:21:11.000000000 +0800
+++ zed.opam	2023-10-27 18:21:22.000000000 +0800
@@ -19,7 +19,6 @@
   "ocaml" {>= "4.02.3"}
   "react"
   "result"
-  "uchar"
   "uutf"
   "uucp" {>= "2.0.0"}
   "uuseg"
