diff --git a/wangle/CMakeLists.txt b/wangle/CMakeLists.txt
index 4a9ed54..2efdb36 100644
--- a/wangle/CMakeLists.txt
+++ b/wangle/CMakeLists.txt
@@ -77,18 +77,22 @@ set(WANGLE_SOURCES
   ssl/TLSTicketKeyManager.cpp
 )
 
-add_library(wangle STATIC
+add_library(wangle
   ${WANGLE_HEADERS}
   ${WANGLE_SOURCES}
 )
 
+if (BUILD_SHARED_LIBS)
+  set_target_properties(wangle
+    PROPERTIES VERSION 1.0.0 SOVERSION 1)
+endif()
+
 target_link_libraries(wangle
   ${FOLLY_LIBRARIES}
   ${Boost_LIBRARIES}
   ${OPENSSL_LIBRARIES}
   ${GLOG_LIBRARY_PATH}
-  ${GFLAGS_LIBRARY_PATH}
-  $<$<NOT:$<CXX_COMPILER_ID:Clang>>:-latomic>)
+  ${GFLAGS_LIBRARY_PATH})
 
 install(TARGETS wangle DESTINATION lib)
 foreach(dir ${WANGLE_HEADER_DIRS})
