From 52fc7c650949d409ac1c1cce78e022c0e1860589 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 13 Dec 2022 18:01:32 +0700
Subject: [PATCH 3/3] Link to LegacySupport

---
 src/Makevars | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makevars b/src/Makevars
index 3c95cb1..01e7269 100644
--- a/src/Makevars
+++ b/src/Makevars
@@ -77,7 +77,7 @@
 		sh autogen.sh; \
 	fi; \
 	chmod +x configure; \
-	CC="$(CC)" CFLAGS="$(CFLAGS) $(CPICFLAGS) $(C_VISIBILITY) -DNDEBUG" CPPFLAGS="$(CPPFLAGS)" AR="$(AR)" RANLIB="$(RANLIB)" LDFLAGS="$(LDFLAGS)" ./configure $(CONFIGURE_FLAGS)
+	CC="$(CC)" CFLAGS="$(CFLAGS) -I@PREFIX@/include/LegacySupport $(CPICFLAGS) $(C_VISIBILITY) -DNDEBUG" CPPFLAGS="$(CPPFLAGS)" AR="$(AR)" RANLIB="$(RANLIB)" LDFLAGS="$(LDFLAGS) -lMacportsLegacySupport" ./configure $(CONFIGURE_FLAGS)
 
 libuv/.libs/libuv.a: libuv/Makefile
 	$(MAKE) --directory=libuv \
