--- Makefile.orig	2018-07-12 02:49:49.000000000 -0500
+++ Makefile	2023-10-26 19:47:50.000000000 -0500
@@ -1,3 +1,4 @@
+include Makefile.macports.inc
 
 CURRENT_DATE=$(shell date +"%Y-%m-%d")
 SOURCE_FILES=trash.m HGUtils.m HGCLIUtils.m fileSize.m
@@ -10,7 +11,7 @@
 	@echo
 	@echo ---- Compiling:
 	@echo ======================================
-	$(CC) -O2 -Wall -Wextra -Wpartial-availability -Wno-unguarded-availability -force_cpusubtype_ALL -mmacosx-version-min=10.7 -arch i386 -arch x86_64 -framework AppKit -framework ScriptingBridge -o $@ $(SOURCE_FILES)
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -Wall -Wextra -Wpartial-availability -Wno-unguarded-availability -framework AppKit -framework ScriptingBridge -o $@ $(SOURCE_FILES)
 
 analyze:
 	@echo
