--- Makefile.2LaTeX.orig	Sat Jan 15 16:22:38 2005
+++ Makefile.2LaTeX	Sat Jan 15 16:24:52 2005
@@ -42,10 +42,10 @@
 rtf2LaTeX: $(LATEXOBJS)
 	$(LD) $(LDFLAGS) -o $@ $(LATEXOBJS)
 install: rtf2LaTeX $(DOCFILS) $(SUPPORT)
-	$(CP) rtf2LaTeX $(BINDIR)
-	$(CP) rtf2LaTeX.man $(MANDIR)/rtf2LaTeX.1
-	-mkdir $(RTFDIR)
-	$(CP) ansi.code mac.code german.land english.land $(RTFDIR)
+	$(CP) rtf2LaTeX $(DESTDIR)$(BINDIR)
+	$(CP) rtf2LaTeX.man $(DESTDIR)$(MANDIR)/rtf2LaTeX.1
+	-mkdir $(DESTDIR)$(RTFDIR)
+	$(CP) ansi.code mac.code german.land english.land $(DESTDIR)$(RTFDIR)
 tags:
 	etags -et *.[ch]
 clean::
