XMLScanner::loadGrammar's toCache argument is optional
--- src/types/schema/SchemaValidatorFilter.h.orig	2012-03-20 19:56:21.000000000 -0500
+++ src/types/schema/SchemaValidatorFilter.h	2012-05-27 00:36:41.000000000 -0500
@@ -238,7 +238,7 @@
   virtual XERCES_CPP_NAMESPACE_QUALIFIER Grammar* loadGrammar(
         const XERCES_CPP_NAMESPACE_QUALIFIER InputSource &src,
         const short grammarType,
-        const bool toCache) 
+        const bool toCache = false) 
   {
     return 0;
   }
