diff -u -r xqf-1.0.5/src/Makefile.in xqf-1.0.5-patch/src/Makefile.in
--- xqf-1.0.5/src/Makefile.in	2006-11-04 13:44:12.000000000 +0100
+++ xqf-1.0.5-patch/src/Makefile.in	2007-05-20 22:59:44.000000000 +0200
@@ -313,7 +313,7 @@
 	    $(GDK_PIXBUF_LIBS) \
 	    -lz
 
-xqf_LDFLAGS = -Wl,-rpath,$(pkglibdir) -Wl,--export-dynamic -Wl,--version-script=$(top_srcdir)/src/xqf.map
+xqf_LDFLAGS = 
 
 # we need to specify all deps manually due to dummy libs
 xqf_DEPENDENCIES = $(top_builddir)/src/zip/libunzip.a \
