--- drivers/macosx-ups.c	2024-04-02 08:31:18.000000000 +0800
+++ drivers/macosx-ups.c	2024-12-13 15:56:30.000000000 +0800
@@ -28,6 +28,13 @@
 #include "IOKit/ps/IOPowerSources.h"
 #include "IOKit/ps/IOPSKeys.h"
 
+#ifndef kIOPSTypeKey
+#define kIOPSTypeKey "Type"
+#endif
+#ifndef kIOPSInternalBatteryType
+#define kIOPSInternalBatteryType "InternalBattery"
+#endif
+
 #define DRIVER_NAME	"Mac OS X UPS meta-driver"
 #define DRIVER_VERSION	"1.41"
 
