--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -195,11 +195,13 @@
 # Check for Cairo rendering backend
 macro_optional_find_package(Cairo ${CAIRO_VERSION})
 
-find_package(Boost 1.58.0)
-if(Boost_FOUND)
-  set(USE_BOOST_HEADERS ON)
-elseif(ENABLE_BOOST)
-  message(FATAL_ERROR "-- Boost recommended for Splash. Use ENABLE_BOOST=OFF to skip.")
+if(ENABLE_BOOST)
+  find_package(Boost 1.58.0)
+  if(Boost_FOUND)
+    set(USE_BOOST_HEADERS ON)
+  else()
+    message(FATAL_ERROR "-- Boost enabled for Splash but not found.")
+  endif()
 endif()
 
 if(CAIRO_FOUND)
