--- DESCRIPTION.orig	2023-09-10 11:30:03.000000000 +0700
+++ DESCRIPTION	2023-09-10 13:28:35.000000000 +0700
@@ -25,7 +25,7 @@
 ByteCompile: true
 URL: https://github.com/leonawicz/memery
 BugReports: https://github.com/leonawicz/memery/issues
-Suggests: testthat, knitr, rmarkdown, magick, gifski
+Suggests: testthat, knitr, rmarkdown, magick
 Depends: showtext
 Imports: sysfonts, grid, png, jpeg, Cairo, ggplot2, cowplot, magrittr,
         purrr, shiny, shinycssloaders, shinyBS, colourpicker
