diff --git inc/stdtp.h inc/stdtp.h
index 4789b68..53855ae 100644
--- inc/stdtp.h
+++ inc/stdtp.h
@@ -51,7 +51,7 @@
 
 #if defined(__APPLE__) && !defined(__FreeBSD__)
 // MAC OS X is Free BSD
-#define __FreeBSD__ 5
+//#define __FreeBSD__ 5
 #endif
 
 #ifndef _WINCE
diff --git src/sync.cpp src/sync.cpp
index 4ae2a6a..24f2392 100644
--- src/sync.cpp
+++ src/sync.cpp
@@ -340,7 +340,7 @@ bool dbWatchDog::create(char const* name) {
 
 
 
-#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__FreeBSD__) 
+#if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) || defined(__FreeBSD__)  || defined(__APPLE__)
 /* union semun is defined by including <sys/sem.h> */  
 #else
 union semun {
