https://github.com/google/re2/pull/354

--- re2/re2.h.orig	2021-10-29 01:42:34.000000000 +1100
+++ re2/re2.h	2022-01-28 03:00:30.000000000 +1100
@@ -971,7 +971,7 @@ namespace hooks {
 // As per https://github.com/google/re2/issues/325, thread_local support in
 // MinGW seems to be buggy. (FWIW, Abseil folks also avoid it.)
 #define RE2_HAVE_THREAD_LOCAL
-#if (defined(__APPLE__) && !TARGET_OS_OSX) || defined(__MINGW32__)
+#if (defined(__APPLE__) && (!defined(TARGET_OS_OSX) || !TARGET_OS_OSX)) || defined(__MINGW32__)
 #undef RE2_HAVE_THREAD_LOCAL
 #endif
 
