--- src/Makefile.osx.orig	2020-12-29 13:17:51.000000000 -0600
+++ src/Makefile.osx	2020-12-29 13:30:06.000000000 -0600
@@ -18,7 +18,7 @@
 WARNINGS = -W -Wall -Wno-unused-parameter -Wno-missing-field-initializers \
 	-Wunused-macros
 JUST_C = -std=c99
-OBJ_CFLAGS = -std=c99 -x objective-c -fobjc-arc -mmacosx-version-min=10.9
+OBJ_CFLAGS = -std=c99 -x objective-c -fobjc-arc 
 CFLAGS = -g -I. $(WARNINGS) $(OPT) -DMACH_O_CARBON -DHAVE_MKSTEMP \
 	-fno-stack-protector $(ARCHFLAGS)
 LIBS = -framework Cocoa
@@ -63,7 +63,6 @@
 
 $(EXE).o: $(OBJS)
 	@for A in $(ARCHS); do \
-		printf "%10s %-20s\n" LD $@.$$A; \
 		$(LD) -r -arch $$A -o $@.$$A $(OBJS); \
 		LIPO_ARGS="$$LIPO_ARGS -arch $$A $@.$$A"; \
 	done; \
@@ -72,7 +71,7 @@
 izzywizzy: $(OBJS) izzywizzy.o
 
 $(EXE): $(EXE).o $(OSX_OBJS)
-	$(DEPLOYMENT_TARGET) $(CC) $(CFLAGS) $(LDFLAGS) -o $(EXE) $(EXE).o $(OSX_OBJS) $(LIBS) 
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(EXE) $(EXE).o $(OSX_OBJS) $(LIBS) 
 
 #
 # Clean up old junk
@@ -87,11 +86,9 @@
 #
 
 %.o : %.m
-	@printf "%10s %-20s (objective-c)\n" CC $<
-	@$(CC) $(CFLAGS) $(OBJ_CFLAGS) -c -o $@ $<
+	$(CC) $(CFLAGS) $(OBJ_CFLAGS) -c -o $@ $<
 %.o : %.c
-	@printf "%10s %-20s\n" CC $<
-	@$(CC) $(JUST_C) $(CFLAGS) -c -o $@ $<
+	$(CC) $(JUST_C) $(CFLAGS) -c -o $@ $<
 
 
 #
