diff -Naurp ./src/common/servlist.c ./src/common/servlist.c
--- ./src/common/servlist.c	2010-05-16 00:24:26.000000000 -0700
+++ ./src/common/servlist.c	2012-04-09 14:02:12.000000000 -0700
@@ -24,7 +24,7 @@
 #include <unistd.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 
 #include "cfgfiles.h"
 #include "fe.h"
diff -Naurp ./src/common/text.c ./src/common/text.c
--- ./src/common/text.c	2010-05-29 19:14:41.000000000 -0700
+++ ./src/common/text.c	2012-04-09 14:02:12.000000000 -0700
@@ -28,7 +28,7 @@
 #include <sys/mman.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 #include "cfgfiles.h"
 #include "chanopt.h"
 #include "plugin.h"
diff -Naurp ./src/common/util.c ./src/common/util.c
--- ./src/common/util.c	2009-08-16 02:40:16.000000000 -0700
+++ ./src/common/util.c	2012-04-09 14:02:12.000000000 -0700
@@ -39,7 +39,7 @@
 #include <errno.h>
 #include "xchat.h"
 #include "xchatc.h"
-#include <glib/gmarkup.h>
+#include <glib.h>
 #include <ctype.h>
 #include "util.h"
 #include "../../config.h"
diff -Naurp ./src/common/xchat.h ./src/common/xchat.h
--- ./src/common/xchat.h	2009-08-16 02:40:16.000000000 -0700
+++ ./src/common/xchat.h	2012-04-09 14:01:05.000000000 -0700
@@ -1,10 +1,6 @@
 #include "../../config.h"
 
-#include <glib/gslist.h>
-#include <glib/glist.h>
-#include <glib/gutils.h>
-#include <glib/giochannel.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <time.h>			/* need time_t */
 
 #ifndef XCHAT_H
