#
# Fix for building xchat 2.6.0 with ./configure --enable-textfe
#
--- src/fe-text/fe-text.c	28 Sep 2005 07:49:46 -0000	1.26
+++ src/fe-text/fe-text.c	4 Nov 2005 03:19:46 -0000
@@ -109,12 +109,6 @@
 	fe_print_text (sess, buf);
 
 	fe_print_text (sess, "\n\nCompiled in Features\0032:\017 "
-#ifdef USE_PERL
-	"Perl "
-#endif
-#ifdef USE_PYTHON
-	"Python "
-#endif
 #ifdef USE_PLUGIN
 	"Plugin "
 #endif
@@ -124,9 +118,6 @@
 #ifdef USE_OPENSSL
 	"OpenSSL "
 #endif
-#ifdef SOCKS
-	"Socks5 "
-#endif
 #ifdef USE_IPV6
 	"IPv6"
 #endif
@@ -406,7 +397,7 @@
 			return 0;
 		}
 	}
-	return 1;
+	return -1;
 }
 
 void
@@ -557,7 +548,7 @@
 }
 
 void
-fe_message (char *msg, int wait)
+fe_message (char *msg, int flags)
 {
 	puts (msg);
 }
@@ -565,7 +556,7 @@
 void
 fe_close_window (struct session *sess)
 {
-	kill_session_callback (sess);
+	session_free (sess);
 	done = TRUE;
 }
 
@@ -803,6 +794,11 @@
 {
 	return -1;
 }
+void *
+fe_gui_info_ptr (session *sess, int info_type)
+{
+	return NULL;
+}
 void fe_confirm (const char *message, void (*yesproc)(void *), void (*noproc)(void *), void *ud)
 {
 }
