#
# Fix proxy
#
diff -u -r1.71 server.c
--- src/common/server.c	29 May 2006 06:53:25 -0000	1.71
+++ src/common/server.c	8 Jun 2006 15:22:07 -0000
@@ -1534,7 +1534,10 @@
 	if (!serv->dont_use_proxy && (prefs.proxy_type == 5))
 		error = net_connect (ns_server, serv->proxy_sok4, serv->proxy_sok6, &psok);
 	else
+	{
 		error = net_connect (ns_server, serv->sok4, serv->sok6, &sok);
+		psok = sok;
+	}
 
 	if (error != 0)
 	{
