--- libavcodec/libsvtav1.c.orig	2024-07-29 08:54:52
+++ libavcodec/libsvtav1.c	2025-02-21 22:34:45
@@ -262,7 +262,7 @@
 
     svt_enc->eos_flag = EOS_NOT_REACHED;
 
-    svt_ret = svt_av1_enc_init_handle(&svt_enc->svt_handle, svt_enc, &svt_enc->enc_params);
+    svt_ret = svt_av1_enc_init_handle(&svt_enc->svt_handle, &svt_enc->enc_params);
     if (svt_ret != EB_ErrorNone) {
         return svt_print_error(avctx, svt_ret, "Error initializing encoder handle");
     }
