--- libavcodec/videotoolboxenc.c	2026-03-17 05:13:05
+++ libavcodec/videotoolboxenc.c	2026-04-30 22:38:13
@@ -1648,7 +1648,7 @@
         return AVERROR(EINVAL);
     }
 
-#if defined(MAC_OS_X_VERSION_10_9) && !TARGET_OS_IPHONE && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_9)
+#if defined(MAC_OS_X_VERSION_10_10) && !TARGET_OS_IPHONE && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_10)
     if (avctx->codec_id == AV_CODEC_ID_PRORES) {
         if (__builtin_available(macOS 10.10, *)) {
             VTRegisterProfessionalVideoWorkflowVideoEncoders();
