--- misc.c.orig	2012-07-31 23:44:04.000000000 +0900
+++ misc.c	2012-07-31 23:44:20.000000000 +0900
@@ -299,7 +299,7 @@
   /* Clear the stack up to about limit.  Return arg.  This function is  */
   /* not static because it could also be errorneously defined in .S     */
   /* file, so this error would be caught by the linker.                 */
-  void * GC_clear_stack_inner(void *arg, ptr_t limit)
+  void * GC_clear_stack_inner(void *arg, volatile ptr_t limit)
   {
     word dummy[CLEAR_SIZE];
 
