--- pdel/pd_port.h.orig	2009-05-14 07:36:03.000000000 +1000
+++ pdel/pd_port.h	2018-04-20 21:41:33.000000000 +1000
@@ -47,7 +47,7 @@
 
 /** BSD Common **/
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 #define HAVE_SIN_LEN 1
 #define PDEL_NET_SUPPORT 1
 #define HAVE_FUNOPEN 1
@@ -266,8 +266,10 @@ __THROW __attribute__ ((__format__ (__pr
 #endif
 
 #ifndef __FreeBSD__
+#ifndef __FBSDID
 #define __FBSDID(s)	static const char *pd_fbsdid   __attribute__ ((unused)) = (s);
 #endif
+#endif
 
 /* DLL Export/Import Tricks */
 #if defined(NEED_WINDLL_DECLSPEC) && (defined(PDEL_SHARED) && PDEL_SHARED != 0)
