--- src/meson.build	2024-08-14 21:52:30
+++ src/meson.build	2024-10-11 00:52:39
@@ -54,7 +54,7 @@
 cmatrix_lib = both_libraries('cmatrix',
                              cmatrix_sources,
                              include_directories: [ root_inc, src_inc ],
-                             soversion: '0u',
+                             soversion: '0',
                              dependencies: cmatrix_deps,
                              install: install_lib,
 )
