--- Common/common-rules.mk.orig	2016-09-09 10:35:06.000000000 -0700
+++ Common/common-rules.mk	2016-09-09 10:35:16.000000000 -0700
@@ -333,4 +333,5 @@
 clean-keepmod clean-keepmod-common \
 clean-keepmod-utilities clean-cpp clean-wfnutils \
 qhull clean-qhull cleanall-qhull clean-keepmod-qhull \
-abi2bgw clean-abi2bgw cleanall-abi2bgw clean-keepmod-abi2bgw
+abi2bgw clean-abi2bgw cleanall-abi2bgw clean-keepmod-abi2bgw \
+sapo epm icm utilities
