Index: Common/common-rules.mk
===================================================================
--- Common/common-rules.mk	(revision 5608)
+++ Common/common-rules.mk	(working copy)
@@ -46,7 +46,16 @@
 F90_CMD = $(F90free) $(FTNINC) -c $(FOPTS) $(basename $<).p.f -o $(basename $<).o $(MOD_OPT)$(dir $<)
 F90_CMD_NOOPT = $(F90free) $(FTNINC) -c $(FNOOPTS) $(basename $<).p.f -o $(basename $<).o $(MOD_OPT)$(dir $<)
 # $(MOD_OPT) directs where to put the resulting *.mod file
-f90_CPP = $(FCPP) $(INCLUDE) $(CPPOPT) $< > $(basename $<).p.f
+
+#clang on MacOS treats files incorrectly if they have .F90 extension; but MacOS is case-insensitive
+#anyway, so we can refer to the file either way
+ifeq ($(findstring clang,$(FCPP)),clang)
+  f90_name = $(basename $<).F90
+else
+  f90_name = $<
+endif
+
+f90_CPP = $(FCPP) $(INCLUDE) $(CPPOPT) $(f90_name) > $(basename $<).p.f
 F90_CPP = $(FCPP) -P $(INCLUDE) $(CPPOPT) $< > $(basename $<).p.f
 ifneq (,$(filter $(COMPFLAG),-DOPEN64 -DPATH -DABSOFT -DCRAY))
 # these compilers name all modules uppercase
