diff --git a/src/menus/FileMenus.cpp b/src/menus/FileMenus.cpp
index b36c10fa3..80ec737b4 100644
--- a/src/menus/FileMenus.cpp
+++ b/src/menus/FileMenus.cpp
@@ -200,6 +200,7 @@ void OnClose(const CommandContext &context )
 void OnCompact(const CommandContext &context)
 {
    ProjectFileManager::Get(context.project).Compact();
+   UndoManager::Get(context.project).ClearStates();
 }
 
 void OnSave(const CommandContext &context )
@@ -618,15 +619,15 @@ BaseItemSharedPtr FileMenu()
                AudioIONotBusyFlag() ),
             Command( wxT("SaveCopy"), XXO("&Backup Project..."), FN(OnSaveCopy),
                AudioIONotBusyFlag() )
-         )//,
+         ),
 
          // Bug 2600: Compact has interactions with undo/history that are bound
          // to confuse some users.  We don't see a way to recover useful amounts 
          // of space and not confuse users using undo.
          // As additional space used by aup3 is 50% or so, perfectly valid
          // approach to this P1 bug is to not provide the 'Compact' menu item.
-         //Command( wxT("Compact"), XXO("Co&mpact Project"), FN(OnCompact),
-         //   AudioIONotBusyFlag(), wxT("Shift+A") )
+         Command( wxT("Compact"), XXO("Compact Project..."), FN(OnCompact),
+           AudioIONotBusyFlag() )
       ),
 
       Section( "Import-Export",
