--- config/configure.py.orig	2009-02-10 07:52:42.000000000 -0600
+++ config/configure.py	2009-02-10 07:52:54.000000000 -0600
@@ -161,7 +161,7 @@
   sys.exit('ERROR: SLEPc cannot be configured for non-source installation if PETSc is not configured in the same way.')
 
 # Create architecture directory and configuration file
-archdir = os.sep.join([slepcdir,petscconf.ARCH])
+archdir = slepcdir
 if not os.path.exists(archdir):
   try:
     os.mkdir(archdir)
