--- src/usocket.c.orig	2016-05-30 12:23:04.000000000 +0900
+++ src/usocket.c	2016-05-30 12:38:37.000000000 +0900
@@ -438,7 +438,9 @@
         case EAI_FAMILY: return PIE_FAMILY;
         case EAI_MEMORY: return PIE_MEMORY;
         case EAI_NONAME: return PIE_NONAME;
+#ifdef EAI_OVERFLOW
         case EAI_OVERFLOW: return PIE_OVERFLOW;
+#endif
 #ifdef EAI_PROTOCOL
         case EAI_PROTOCOL: return PIE_PROTOCOL;
 #endif
