--- config.old	2009-09-16 10:42:22.000000000 +0200
+++ config	2009-09-16 10:44:04.000000000 +0200
@@ -15,9 +15,7 @@
 #------
 # Lua includes and libraries
 #
-#LUAINC=-I/usr/local/include/lua50
-#LUAINC=-I/usr/local/include/lua5.1
-#LUAINC=-Ilua-5.1.1/src
+LUAINC=-I@PREFIX@/include
 
 #------
 # Compat-5.1 directory
@@ -28,10 +26,8 @@
 # Top of your Lua installation
 # Relative paths will be inside the src tree
 #
-#INSTALL_TOP_SHARE=/usr/local/share/lua/5.0
-#INSTALL_TOP_LIB=/usr/local/lib/lua/5.0
-INSTALL_TOP_SHARE=/usr/local/share/lua/5.1
-INSTALL_TOP_LIB=/usr/local/lib/lua/5.1
+INSTALL_TOP_SHARE=$(DESTDIR)@PREFIX@/share/lua/5.1
+INSTALL_TOP_LIB=$(DESTDIR)@PREFIX@/lib/lua/5.1
 
 INSTALL_DATA=cp
 INSTALL_EXEC=cp
@@ -40,20 +36,20 @@
 # Compiler and linker settings
 # for Mac OS X
 #
-#CC=gcc
-#DEF= -DLUASOCKET_DEBUG -DUNIX_HAS_SUN_LEN
-#CFLAGS= $(LUAINC) -I$(COMPAT) $(DEF) -pedantic -Wall -O2 -fno-common
-#LDFLAGS=-bundle -undefined dynamic_lookup
-#LD=export MACOSX_DEPLOYMENT_TARGET="10.3"; gcc
+CC=gcc
+DEF= -DLUASOCKET_DEBUG -DUNIX_HAS_SUN_LEN
+CFLAGS= $(LUAINC) $(DEF) -pedantic -Wall -O2 -fno-common
+LDFLAGS=-bundle -undefined dynamic_lookup
+LD=export MACOSX_DEPLOYMENT_TARGET="10.3"; gcc
 
 #------
 # Compiler and linker settings
 # for Linux
-CC=gcc
-DEF=-DLUASOCKET_DEBUG 
-CFLAGS= $(LUAINC) $(DEF) -pedantic -Wall -O2 -fpic
-LDFLAGS=-O -shared -fpic
-LD=gcc 
+#CC=gcc
+#DEF=-DLUASOCKET_DEBUG 
+#CFLAGS= $(LUAINC) $(DEF) -pedantic -Wall -O2 -fpic
+#LDFLAGS=-O -shared -fpic
+#LD=gcc 
 
 #------
 # End of makefile configuration
