--- kghostview/Makefile.am	2 Oct 2002 23:47:21 -0000	1.1.1.2
+++ kghostview/Makefile.am	1 Jun 2003 14:31:58 -0000	1.3
@@ -7,8 +7,9 @@
 
 bin_PROGRAMS = 	kghostview
 kde_module_LTLIBRARIES = libkghostviewpart.la
+lib_LTLIBRARIES = libkghostview_shared.la kghostview.la
 
-libkghostviewpart_la_SOURCES = dscparse.cpp kgv_miniwidget.cpp marklist.cpp \
+libkghostview_shared_la_SOURCES = dscparse.cpp kgv_miniwidget.cpp marklist.cpp \
 	messages.cpp infodialog.cpp gotodialog.cpp kgvshell.cpp kpswidget.cpp \
 	kgvpageview.cpp ps.c kgv_view.cpp scrollbox.cpp  kgvpagedecorator.cpp \
 	kgvconfigdialog.cpp kgvmainwidget.cpp dscparse_adapter.cpp \
@@ -19,12 +20,26 @@
 	kgvpagedecorator.h kgvconfigdialog.h kgvmainwidget.h dscparse.h \
 	dscparse_adapter.h qttableview.h kdscerrordialog.h
 
+kghostview_la_SOURCES = kghostview_la_main.cpp
+kghostview_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+kghostview_la_LIBADD  = libkghostview_shared.la
+
+kghostview_la_main.cpp: main.cpp
+	cat main.cpp > kghostview_la_main.cpp
+
 kghostview_SOURCES = main.cpp
 kghostview_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kghostview_LDADD = libkghostviewpart.la
+kghostview_LDADD = libkghostview_shared.la
+
+libkghostviewpart_la_SOURCES = libkghostviewpart_la_dummy.cpp
+libkghostviewpart_la_LIBADD  = libkghostview_shared.la
+libkghostviewpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -module -avoid-version
+
+libkghostviewpart_la_dummy.cpp:
+	echo > libkghostviewpart_la_dummy.cpp
 
-libkghostviewpart_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-libkghostviewpart_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) -lkdeprint
+libkghostview_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -avoid-version -no-undefined
+libkghostview_shared_la_LIBADD = $(LIB_KFILE) $(LIB_KPARTS) -lkdeprint $(LIB_QT) $(LIB_KDECORE) -lkdeui
 
 METASOURCES = AUTO
 EXTRA_DIST = kghostview.desktop 
