--- kview/Makefile.am	29 Oct 2002 16:46:58 -0000	1.1.1.4
+++ kview/Makefile.am	29 Nov 2002 19:01:21 -0000	1.5
@@ -5,21 +5,28 @@
 METASOURCES = AUTO
 
 bin_PROGRAMS = kview
-lib_LTLIBRARIES = kview.la libkviewsupport.la
+lib_LTLIBRARIES = kview.la libkview_main.la libkviewsupport.la
 
 noinst_HEADERS = kview.h kpreferencesdialog.h kpreferencesmodule.h kpluginselector.h kviewconfmodules.h
 
 libkviewsupport_la_SOURCES = kpreferencesdialog.cpp kpreferencesmodule.cpp kpluginselector.cpp
-libkviewsupport_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 0:0
+libkviewsupport_la_LDFLAGS = $(all_libraries) -no-undefined -avoid-version
 libkviewsupport_la_LIBADD  = $(LIB_KDEUI)
 
-kview_SOURCES = dummy.cpp
+libkview_main_la_SOURCES = kviewconfmodules.cpp kview.cpp
+libkview_main_la_LDFLAGS = $(all_libraries) -no-undefined -avoid-version
+libkview_main_la_LIBADD  = libkviewsupport.la $(LIB_KPARTS) $(LIB_KIO)
+
+kview_SOURCES = kview_main.cpp
 kview_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kview_LDADD = kview.la
+kview_LDADD = libkview_main.la
+
+kview_main.cpp: main.cpp
+	cat main.cpp > kview_main.cpp
 
-kview_la_SOURCES = kviewconfmodules.cpp kview.cpp main.cpp
+kview_la_SOURCES = main.cpp
 kview_la_LDFLAGS = $(KDE_RPATH) $(all_libraries) -module -avoid-version
-kview_la_LIBADD = $(LIB_KPARTS) libkviewsupport.la
+kview_la_LIBADD = libkview_main.la
 
 lnkdir = $(kde_appsdir)/Graphics
 lnk_DATA = kview.desktop
