--- kghostview/kpswidget.cpp	6 May 2003 21:20:48 -0000	1.1.1.3
+++ kghostview/kpswidget.cpp	9 Apr 2003 23:17:42 -0000	1.2
@@ -513,8 +513,7 @@
         *_process <<
         // The following two lines are their to ensure that we are allowed to read _fileName
         "-dDELAYSAFER" << "-sInputFile="+_fileName << "-c" <<
-        "<< /PermitFileReading [ InputFile ] /PermitFileWriting [] /PermitFileControl [] >> setuserparams .locksafe" <<
-        "-";
+        "<< /PermitFileReading [ InputFile ] /PermitFileWriting [] /PermitFileControl [] >> setuserparams .locksafe";
     else
 	*_process << _fileName << "-c" << "quit";
 
@@ -580,8 +579,6 @@
     kdDebug(4500) << "KPSWidget: process exited" << endl;
     
     delete process;
-    if (process == _process)
-        _process = 0;
 }
 
 /**
