--- kviewshell/Makefile.am	29 Oct 2002 16:46:59 -0000	1.1.1.4
+++ kviewshell/Makefile.am	1 Jun 2003 14:31:59 -0000	1.8
@@ -19,8 +19,8 @@
 kviewerpart_la_LIBADD = ./libifaces.la $(LIB_KFILE) -lkparts
 
 libkmultipage_la_SOURCES = kmultipage.cpp centeringScrollview.cpp kmultipageInterface.skel
-libkmultipage_la_LDFLAGS = $(all_libraries)
-libkmultipage_la_LIBADD = $(LIB_KFILE) -lkparts
+libkmultipage_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
+libkmultipage_la_LIBADD = $(LIB_KFILE) -lkparts -lDCOP $(LIB_QT) $(LIB_KDEUI) $(LIB_KDECORE)
 
 libkpagetest_la_SOURCES = kpagetest.cpp 
 libkpagetest_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN)
