--- dist/configure.orig	2020-06-24 14:58:52.000000000 -0700
+++ dist/configure	2020-06-24 15:09:30.000000000 -0700
@@ -18756,6 +18756,7 @@
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 int
 main ()
 {
@@ -18792,6 +18793,7 @@
 /* end confdefs.h.  */
 
 #include <pthread.h>
+#include <stdlib.h>
 main() {
 	pthread_cond_t cond;
 	pthread_mutex_t mutex;
@@ -19159,6 +19161,7 @@
 	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 int
 main ()
 {
@@ -20198,6 +20201,7 @@
 	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 int
 main ()
 {
@@ -21600,6 +21604,8 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdio.h>
+#include <string.h>
 int
 main ()
 {
@@ -21634,6 +21640,8 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdio.h>
+#include <string.h>
 		main() {
 			$db_cv_seq_type l;
 			unsigned $db_cv_seq_type u;
