--- a/Makefile.orig	2012-01-22 17:34:02.000000000 -0800
+++ b/Makefile	2012-01-22 17:34:14.000000000 -0800
@@ -70,8 +70,7 @@ ifeq ($(MAKECMDGOALS),install-clang)
   DIRS := tools/clang/tools/driver tools/clang/lib/Headers \
           tools/clang/tools/libclang tools/clang/tools/c-index-test \
-          tools/clang/include/clang-c \
-          tools/clang/runtime tools/clang/docs \
-          tools/lto runtime
+          tools/clang/include tools/clang/lib \
+          tools/clang/runtime tools/clang/docs
   OPTIONAL_DIRS :=
   NO_INSTALL = 1
 endif
