--- CronJob.m.orig	2006-07-16 15:24:03.000000000 -0500
+++ CronJob.m	2011-05-07 23:14:39.000000000 -0500
@@ -15,7 +15,7 @@
 {
     if(self = [super init]) {
 		[self setEnabledDisabledString:@"#"];
-		[self setCommand:@"/usr/local/bin/runpsync"];
+		[self setCommand:@"@PREFIX@/bin/runpsync"];
 /*		[self setFFlag:@""];
 		[self setDFlag:@""];
 		[self setRFlag:@""];*/
@@ -344,7 +344,7 @@
 		return;
 	}
     NSMutableString *temp2=[NSMutableString stringWithString:cmd];
-    [temp2 replaceOccurrencesOfString:@"/usr/bin/runpsync" withString:@"/usr/local/bin/runpsync" options:NSLiteralSearch range:NSMakeRange(0, [temp2 length])];
+    [temp2 replaceOccurrencesOfString:@"/usr/bin/runpsync" withString:@"@PREFIX@/bin/runpsync" options:NSLiteralSearch range:NSMakeRange(0, [temp2 length])];
     [command release];
     command = [[NSString stringWithString:temp2] retain];
 	[self setNeedToSave:TRUE];
