--- src/tools/hunspell.cxx.orig	2011-04-09 11:32:43.000000000 -0500
+++ src/tools/hunspell.cxx	2011-04-09 11:34:15.000000000 -0500
@@ -60,11 +60,6 @@
 #include "manparser.hxx"
 #include "firstparser.hxx"
 
-#define LIBDIR \
-    "/usr/share/hunspell:" \
-    "/usr/share/myspell:" \
-    "/usr/share/myspell/dicts:" \
-    "/Library/Spelling"
 #define USEROOODIR \
     ".openoffice.org/3/user/wordbook:" \
     ".openoffice.org2/user/wordbook:" \
