--- src/tools/hunspell.cxx.orig	2016-12-22 12:20:34.000000000 +1100
+++ src/tools/hunspell.cxx	2017-01-19 08:23:49.000000000 +1100
@@ -118,11 +118,6 @@
 #include "xmlparser.hxx"
 #include "odfparser.hxx"
 
-#define LIBDIR                \
-  "/usr/share/hunspell:"      \
-  "/usr/share/myspell:"       \
-  "/usr/share/myspell/dicts:" \
-  "/Library/Spelling"
 #define USEROOODIR                    \
   ".openoffice.org/3/user/wordbook:"  \
   ".openoffice.org2/user/wordbook:"   \
