--- DESCRIPTION	2023-09-18 00:10:02.000000000 +0700
+++ DESCRIPTION	2023-09-18 01:07:15.000000000 +0700
@@ -19,8 +19,7 @@
 Imports: dplyr (>= 1.1.0), ggplot2, ggraph, igraph (>= 1.5.0.1),
         network, tidygraph
 Suggests: gganimate, methods, readxl, roxygen2, RSiena, testthat, xml2,
-        patchwork, knitr, rmarkdown, learnr, grDevices, BiocManager,
-        minMSE
+        patchwork, knitr, rmarkdown, grDevices, BiocManager, minMSE
 Enhances: Rgraphviz
 Authors@R: 
     c(person(given = "James",
