$ diff -u configure.ac.orig configure.ac
--- configure.ac.orig	2014-10-15 09:00:52.000000000 -0400
+++ configure.ac	2014-10-15 09:04:11.000000000 -0400
@@ -23,7 +23,7 @@
 
 dnl add features to CFLAGS
 CFLAGS_dist="-Wall -Werror --std=gnu99 -D_GNU_SOURCE"
-CFLAGS_dist="${CFLAGS_dist} -Wpointer-arith -Wcast-align -Wcast-qual -Wstrict-prototypes -Wshadow -Waggregate-return -Wmissing-prototypes -Wnested-externs -Wsign-compare"
+CFLAGS_dist="${CFLAGS_dist} -Wno-tautological-constant-out-of-range-compare -Wpointer-arith -Wcast-align -Wcast-qual -Wstrict-prototypes -Wshadow -Waggregate-return -Wmissing-prototypes -Wnested-externs -Wsign-compare"
 
 dnl check the operating system
 case "${target_os}" in
