$ diff -u Makefile.am.orig Makefile.am
--- Makefile.am.orig	2014-10-14 14:57:02.000000000 -0400
+++ Makefile.am	2014-10-14 14:57:46.000000000 -0400
@@ -2,7 +2,7 @@
 
 ACLOCAL_AMFLAGS=-I m4
 
-SUBDIRS = src examples misc
+SUBDIRS = src misc
 DIST_SUBDIRS = $(SUBDIRS) doc wince
 
 EXTRA_DIST = libdvbpsi.spec libdvbpsi.spec.in libdvbpsi.pc.in bootstrap
$ diff -u configure.ac.orig configure.ac
--- configure.ac.orig	2014-10-14 14:57:26.000000000 -0400
+++ configure.ac	2014-10-14 14:57:59.000000000 -0400
@@ -105,8 +105,6 @@
 dnl
 AC_OUTPUT([Makefile
            src/Makefile
-           examples/Makefile
-           examples/dvbinfo/Makefile
            misc/Makefile
            doc/Makefile
            wince/Makefile
