--- doc/Makefile.in.orig	2010-11-01 15:29:11.000000000 -0500
+++ doc/Makefile.in	2011-10-21 06:00:49.000000000 -0500
@@ -171,7 +171,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
--- doc/libvorbis/Makefile.in.orig	2010-11-01 15:29:11.000000000 -0500
+++ doc/libvorbis/Makefile.in	2011-10-21 06:01:48.000000000 -0500
@@ -158,7 +158,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/libvorbis
+docdir = $(datadir)/doc/$(PACKAGE)/libvorbis
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
--- doc/vorbisenc/Makefile.in.orig	2010-11-01 15:29:11.000000000 -0500
+++ doc/vorbisenc/Makefile.in	2011-10-21 06:02:12.000000000 -0500
@@ -158,7 +158,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisenc
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisenc
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
--- doc/vorbisfile/Makefile.in.orig	2010-11-01 15:29:11.000000000 -0500
+++ doc/vorbisfile/Makefile.in	2011-10-21 06:02:23.000000000 -0500
@@ -158,7 +158,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisfile
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisfile
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
