--- Applications/Cxx/CMakeLists.txt.orig	2019-07-22 07:36:02.000000000 -0700
+++ Applications/Cxx/CMakeLists.txt	2019-12-21 09:13:08.000000000 -0700
@@ -66,7 +66,7 @@
   set(CMAKE_REQUIRED_INCLUDES ${POPPLER_INCLUDE_DIRS})
   set(CMAKE_REQUIRED_LIBRARIES ${POPPLER_LIBRARIES})
   CHECK_CXX_SOURCE_COMPILES(
-    "\#include <poppler/GlobalParams.h>\nint main() { globalParams = new GlobalParams(0); return 0;}"
+    "\#include <poppler/GlobalParams.h>\nint main() { globalParams = std::make_unique<GlobalParams>(nullptr); return 0;}"
     LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM)
   set(libpoppler_flags)
   if(LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM)
--- Applications/Cxx/gdcmpdf.cxx.orig	2021-10-06 14:38:42.000000000 -0500
+++ Applications/Cxx/gdcmpdf.cxx	2022-02-27 06:05:39.000000000 -0600
@@ -345,12 +345,12 @@
   ownerPW = NULL;
   userPW = NULL;
 #ifdef LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM
-  globalParams = new GlobalParams(0);
+  globalParams = std::make_unique<GlobalParams>(nullptr);
 #else
 #ifdef LIBPOPPLER_GLOBALPARAMS_HAS_RESET
   globalParams.reset(new GlobalParams());
 #else
-  globalParams = new GlobalParams();
+  globalParams = std::make_unique<GlobalParams>();
 #endif
 #endif
   uMap = globalParams->getTextEncoding();
 
--- Applications/Cxx/gdcminfo.cxx.orig	2021-10-06 14:38:42.000000000 -0500
+++ Applications/Cxx/gdcminfo.cxx	2022-02-27 06:07:16.000000000 -0600
@@ -519,12 +519,12 @@
     UnicodeMap *uMap;
 #endif
 #ifdef LIBPOPPLER_GLOBALPARAMS_CSTOR_HAS_PARAM
-    globalParams = new GlobalParams(0);
+    globalParams = std::make_unique<GlobalParams>(nullptr);
 #else
 #ifdef LIBPOPPLER_GLOBALPARAMS_HAS_RESET
     globalParams.reset(new GlobalParams());
 #else
-    globalParams = new GlobalParams();
+    globalParams = std::make_unique<GlobalParams>();
 #endif
 #endif
     uMap = globalParams->getTextEncoding();
 
