--- setup.cfg	1969-12-31 18:00:00.000000000 -0600
+++ setup.cfg	2018-02-09 19:17:18.000000000 -0600
@@ -0,0 +1,16 @@
+[global]
+salt-root-dir=@@PREFIX@@
+salt-share-dir=@@PREFIX@@/share
+salt-config-dir=@@PREFIX@@/etc/salt
+salt-cache-dir=@@PREFIX@@/var/cache/salt
+salt-sock-dir=@@PREFIX@@/var/run
+salt-srv-root-dir=@@PREFIX@@/srv
+salt-base-file-roots-dir=@@PREFIX@@/srv/salt
+salt-base-pillar-roots-dir=@@PREFIX@@/srv/pillar
+salt-base-master-roots-dir=@@PREFIX@@/srv/salt-master
+salt-base-thorium-roots-dir=@@PREFIX@@/srv/thorium
+salt-logs-dir=@@PREFIX@@/var/log/salt
+salt-pidfile-dir=@@PREFIX@@/var/run
+salt-spm-formula-dir=@@PREFIX@@/srv/spm/salt
+salt-spm-pillar-dir=@@PREFIX@@/srv/spm/pillar
+salt-spm-reactor-dir=@@PREFIX@@/srv/spm/reactor
