--- configure.in	2014-09-05 11:05:27.000000000 -0400
+++ configure.in	2017-07-13 21:54:03.000000000 -0400
@@ -307,9 +307,6 @@
 dnl idempotent
 AC_SUBST(INTLOBJS)
 
-localedir='${prefix}/${DATADIRNAME}/locale'
-AC_SUBST(localedir)
-
 dnl
 dnl Locate the gnome-xml library
 dnl
@@ -481,7 +478,7 @@
   xsltproc=false
 fi
 AM_CONDITIONAL(HAVE_XSLTPROC, $xsltproc)
-AC_SUBST(DB2MAN, "http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl")
+AC_SUBST(DB2MAN, "file://__PREFIX__/share/xsl/docbook-xsl/manpages/docbook.xsl")
 
 dnl Handling for Python
 AC_ARG_WITH(python,
@@ -540,8 +537,8 @@
   AM_CONDITIONAL(WITH_PSDOC, test "xno" != "xno")
 else
   AM_CONDITIONAL(WITH_HTMLDOC, test "x$xsltproc" != "xno")
-  AM_CONDITIONAL(WITH_PDFDOC, test "x$dblatex" != "xno")
-  AM_CONDITIONAL(WITH_PSDOC, test "x$dblatex" != "xno")
+  AM_CONDITIONAL(WITH_PDFDOC, test "xno" != "xno")
+  AM_CONDITIONAL(WITH_PSDOC, test "xno" != "xno")
   if test "x$dblatex" != "xno" ; then
     hardbook_ok=yes
   else 
