--- src/configure.ac.orig	2016-09-12 16:31:10.000000000 +0200
+++ src/configure.ac	2016-09-12 19:26:47.000000000 +0200
@@ -1337,7 +1337,11 @@
   fi
 
   dnl -- find the python3 executable
-  AC_PATH_PROGS(vi_cv_path_python3, python3 python)
+  AC_MSG_CHECKING(--with-python3 argument)
+  AC_ARG_WITH(python3, [  --with-python3=PATH       which python to use (default: python3)],
+        python_name="$withval"; AC_MSG_RESULT($python_name),
+        python_name="python3"; AC_MSG_RESULT(no))
+  AC_SUBST(vi_cv_path_python3, $python_name)
   if test "X$vi_cv_path_python3" != "X"; then
 
     dnl -- get its version number
@@ -1427,7 +1431,7 @@
 	@echo "python3_LIBS='$(LIBS)'"
 	@echo "python3_SYSLIBS='$(SYSLIBS)'"
 	@echo "python3_DLLLIBRARY='$(DLLLIBRARY)'"
-	@echo "python3_INSTSONAME='$(INSTSONAME)'"
+	@echo "python3_INSTSONAME='$(PYTHONFRAMEWORKPREFIX)/$(INSTSONAME)'"
 eof
 	    dnl -- delete the lines from make about Entering/Leaving directory
 	    eval "`cd ${PYTHON3_CONFDIR} && make -f "${tmp_mkf}" __ | sed '/ directory /d'`"
