--- src/configure.in.old	2009-04-11 00:05:39.000000000 +0200
+++ src/configure.in	2009-04-11 00:06:09.000000000 +0200
@@ -2952,7 +2952,7 @@
 if test "$enable_multibyte" = "yes"; then
   cflags_save=$CFLAGS
   ldflags_save=$LDFLAGS
-  if test -n "$x_includes" ; then
+  if test -n "$x_includes" && test "$x_includes" != "NONE"; then
     CFLAGS="$CFLAGS -I$x_includes"
     LDFLAGS="$X_LIBS $LDFLAGS -lX11"
     AC_MSG_CHECKING(whether X_LOCALE needed)
