--- src/MacVim/Base.lproj/MainMenu.xib	2021-04-05 15:55:00.000000000 +0200
+++ src/MacVim/Base.lproj/MainMenu.xib	2021-04-05 15:55:20.000000000 +0200
@@ -24,11 +24,6 @@
                                     <action selector="orderFrontStandardAboutPanel:" target="-2" id="142"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Check for Updates…" id="255">
-                                <connections>
-                                    <action selector="checkForUpdates:" target="235" id="Vif-VP-y50"/>
-                                </connections>
-                            </menuItem>
                             <menuItem isSeparatorItem="YES" id="196">
                                 <modifierMask key="keyEquivalentModifierMask" command="YES"/>
                             </menuItem>
--- src/MacVim/Base.lproj/Preferences.xib	2021-04-05 16:02:16.000000000 +0200
+++ src/MacVim/Base.lproj/Preferences.xib	2021-04-05 16:04:16.000000000 +0200
@@ -11,13 +11,9 @@
                 <outlet property="advancedPreferences" destination="620" id="632"/>
                 <outlet property="allowForceClickLookUpButton" destination="rlt-zw-mfW" id="xCv-HS-zyJ"/>
                 <outlet property="appearancePreferences" destination="hr4-G4-3ZG" id="G54-DD-ACh"/>
-                <outlet property="autoInstallUpdateButton" destination="UYM-W0-Kgl" id="cX5-tk-9WJ"/>
                 <outlet property="generalPreferences" destination="115" id="143"/>
                 <outlet property="inputPreferences" destination="Bnq-Nx-GJH" id="FES-rQ-Fpa"/>
                 <outlet property="layoutPopUpButton" destination="427" id="596"/>
-                <outlet property="sparklePrereleaseButton" destination="4Y5-mA-blQ" id="baN-Dn-dgO"/>
-                <outlet property="sparklePrereleaseDesc" destination="ere-oJ-WLd" id="fTe-GS-5qR"/>
-                <outlet property="sparkleUpdaterPane" destination="0hT-y8-Hge" id="e0L-sv-OCW"/>
             </connections>
         </customObject>
         <customObject id="-1" userLabel="First Responder" customClass="FirstResponder"/>
@@ -205,49 +201,6 @@
                             <rect key="frame" x="-2" y="0.0" width="187" height="17"/>
                             <autoresizingMask key="autoresizingMask" flexibleMinY="YES"/>
                             <textFieldCell key="cell" sendsActionOnEndEditing="YES" alignment="right" title="Resizing window:" id="ExQ-6y-pyN" userLabel="Window resizing:">
-                                <font key="font" metaFont="system"/>
-                                <color key="textColor" name="controlTextColor" catalog="System" colorSpace="catalog"/>
-                                <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
-                            </textFieldCell>
-                        </textField>
-                    </subviews>
-                </customView>
-                <customView id="0hT-y8-Hge" userLabel="Updater">
-                    <rect key="frame" x="20" y="20" width="444" height="36"/>
-                    <autoresizingMask key="autoresizingMask" flexibleMaxX="YES" flexibleMinY="YES"/>
-                    <subviews>
-                        <button id="122">
-                            <rect key="frame" x="188" y="18" width="258" height="18"/>
-                            <autoresizingMask key="autoresizingMask" flexibleMinY="YES"/>
-                            <string key="toolTip">Checks for updates and presents a dialog box showing the release notes and prompt for whether you want to install the new version.</string>
-                            <buttonCell key="cell" type="check" title="Check for updates" bezelStyle="regularSquare" imagePosition="left" alignment="left" inset="2" id="975">
-                                <behavior key="behavior" changeContents="YES" doesNotDimImage="YES" lightByContents="YES"/>
-                                <font key="font" metaFont="system"/>
-                            </buttonCell>
-                            <connections>
-                                <action selector="checkForUpdatesChanged:" target="-2" id="YjS-ig-M1j"/>
-                                <binding destination="58" name="value" keyPath="values.SUCheckAtStartup" id="169"/>
-                            </connections>
-                        </button>
-                        <button id="UYM-W0-Kgl" userLabel="Auto-install updates">
-                            <rect key="frame" x="202" y="-2" width="244" height="18"/>
-                            <autoresizingMask key="autoresizingMask" flexibleMinY="YES"/>
-                            <string key="toolTip">MacVim will automatically download and install updates without prompting. The updated version will be used the next time MacVim starts.</string>
-                            <buttonCell key="cell" type="check" title="Automatically install updates" bezelStyle="regularSquare" imagePosition="left" alignment="left" inset="2" id="GfP-vg-mec">
-                                <behavior key="behavior" changeContents="YES" doesNotDimImage="YES" lightByContents="YES"/>
-                                <font key="font" metaFont="system"/>
-                                <connections>
-                                    <binding destination="58" name="enabled" keyPath="values.SUCheckAtStartup" id="5oY-Gf-XJN"/>
-                                </connections>
-                            </buttonCell>
-                            <connections>
-                                <binding destination="58" name="value" keyPath="values.SUAutomaticallyUpdate" id="kyZ-ah-zKf"/>
-                            </connections>
-                        </button>
-                        <textField verticalHuggingPriority="750" horizontalCompressionResistancePriority="250" id="124">
-                            <rect key="frame" x="-2" y="19" width="187" height="17"/>
-                            <autoresizingMask key="autoresizingMask" flexibleMinY="YES"/>
-                            <textFieldCell key="cell" sendsActionOnEndEditing="YES" alignment="right" title="When MacVim launches:" id="976">
                                 <font key="font" metaFont="system"/>
                                 <color key="textColor" name="controlTextColor" catalog="System" colorSpace="catalog"/>
                                 <color key="backgroundColor" name="controlColor" catalog="System" colorSpace="catalog"/>
