--- configure.orig	2018-10-21 21:45:50.000000000 +0800
+++ configure	2023-03-08 17:55:47.000000000 +0800
@@ -2437,48 +2437,6 @@
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
 
-
-
-## Default the OpenMP flag to the empty string.
-## If and only if OpenMP is found, expand to $(SHLIB_OPENMP_CXXFLAGS)
-openmp_flag=""
-openmp_cflag=""
-
-## Check for broken systems produced by a corporation based in Cupertino
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for macOS" >&5
-$as_echo_n "checking for macOS... " >&6; }
-RSysinfoName=$("${R_HOME}/bin/Rscript" --vanilla -e 'cat(Sys.info()["sysname"])')
-if test x"${RSysinfoName}" == x"Darwin"; then
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
-$as_echo "found" >&6; }
-   { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: OpenMP unavailable and turned off." >&5
-$as_echo "$as_me: WARNING: OpenMP unavailable and turned off." >&2;}
-   openmp_flag="-DARMA_DONT_USE_OPENMP"
-else
-   { $as_echo "$as_me:${as_lineno-$LINENO}: result: not found as on ${RSysinfoName}" >&5
-$as_echo "not found as on ${RSysinfoName}" >&6; }
-   ## Check for OpenMP
-   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for OpenMP" >&5
-$as_echo_n "checking for OpenMP... " >&6; }
-   ## if R has -fopenmp we should be good
-   allldflags=$(${R_HOME}/bin/R CMD config --ldflags)
-   hasOpenMP=$(echo ${allldflags} | grep -- -fopenmp)
-   if test x"${hasOpenMP}" == x""; then
-	  { $as_echo "$as_me:${as_lineno-$LINENO}: result: missing" >&5
-$as_echo "missing" >&6; }
-	  openmp_flag="-DARMA_DONT_USE_OPENMP"
-   else
-	  { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
-$as_echo "found" >&6; }
-	  openmp_flag='$(SHLIB_OPENMP_CXXFLAGS)'
-	  openmp_cflag='$(SHLIB_OPENMP_CFLAGS)'
-   fi
-fi
-
-OPENMP_CFLAG="${openmp_cflag}"
-
-OPENMP_FLAG="${openmp_flag}"
-
 ac_config_files="$ac_config_files src/Makevars"
 
 cat >confcache <<\_ACEOF
