--- glib/giochannel.ml.orig	2009-12-03 23:31:14.000000000 +0100
+++ glib/giochannel.ml	2009-12-03 23:31:25.000000000 +0100
@@ -24,7 +24,7 @@
 
 external read_chars : t -> ?off:int -> ?len:int -> string -> 
   [ `NORMAL of int | `AGAIN | `EOF]
-  = "ml_g_io_channel_read_chars"
+  = "my_ml_g_io_channel_read_chars"
 external write_chars : t -> ?bytes_written:int ref -> ?off:int -> ?len:int -> string -> 
   [ `NORMAL of int | `AGAIN]
   = "ml_g_io_channel_write_chars"
