--- configure.orig	2012-09-21 11:41:47.000000000 -0400
+++ configure	2012-09-21 11:42:36.000000000 -0400
@@ -32450,8 +32450,8 @@
     LDFLAGS="$LDFLAGS -framework Carbon"
     ac_ext=cpp
 ac_cpp='$CXXCPP $CPPFLAGS'
-ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compile='$APPLE_CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$APPLE_CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -37034,6 +37034,21 @@
 
   ## fltk (www.fltk.org)
 
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for fltk..." >&5
+$as_echo_n "checking for fltk... " >&6; }
+
+# Check whether FLTK_CONFIG is set and equal to "no"
+use_fltk="auto detect"
+if test "${FLTK_CONFIG+set}" = set ; then
+    if test "${FLTK_CONFIG}" = "no" ; then
+	use_fltk="disabled"
+    fi
+fi
+
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ${use_fltk}" >&5
+$as_echo "${use_fltk}" >&6; }
+
+if test "${use_fltk}" != "disabled"; then
 
 # Check whether --with-fltk-prefix was given.
 if test "${with_fltk_prefix+set}" = set; then :
@@ -37110,6 +37125,7 @@
 $as_echo "no" >&6; }
 fi
 
+fi
 
 
   warn_fltk_config=""
