--- m4/acinclude.m4.orig	2016-11-08 13:21:14.000000000 -0700
+++ m4/acinclude.m4	2016-11-11 09:31:48.000000000 -0700
@@ -1351,7 +1351,7 @@
     ;;
     5)
       QT_MODULES="Qt5Core Qt5Gui Qt5Network Qt5OpenGL Qt5PrintSupport"
-      octave_qscintilla_libnames="qscintilla2-qt5 qt5scintilla2"
+      octave_qscintilla_libnames="qscintilla2-qt5 qt5scintilla2 qscintilla2"
     ;;
     *)
       AC_MSG_ERROR([Unrecognized Qt version $qt_version])
@@ -2738,4 +2738,3 @@
 dnl         End of macros written by Octave developers
 dnl ------------------------------------------------------------
 dnl
-
