--- configure.ac.orig	2016-05-08 13:35:28.000000000 -0700
+++ configure.ac	2016-05-10 06:04:35.000000000 -0700
@@ -1266,84 +1266,6 @@
   fi
 fi
 
-## Check for FreeType 2 library
-
-warn_freetype=""
-PKG_CHECK_MODULES([FT2], [freetype2], [
-  min_ft2_version=9.03
-  AC_MSG_CHECKING([for FreeType2 version >= $min_ft2_version])
-  $PKG_CONFIG freetype2 --atleast-version=$min_ft2_version
-  ac_status=$?
-  if test "$ac_status" = 0; then
-    AC_MSG_RESULT(yes)
-    AC_DEFINE(HAVE_FREETYPE, 1, [Define to 1 if you have Freetype library.])
-    save_LIBS="$LIBS"
-    LIBS="$FT2_LIBS $LIBS"
-    AC_CHECK_FUNCS([FT_Reference_Face])
-    LIBS="$save_LIBS"
-  else
-    AC_MSG_RESULT(no)
-    warn_freetype="FreeType library >= 9.03 not found.  OpenGL graphics will not be fully functional."
-  fi])
-
-if test -n "$warn_freetype"; then
-  OCTAVE_CONFIGURE_WARNING([warn_freetype])
-else
-  FT2_CPPFLAGS="$FT2_CFLAGS"
-  ## Alias CPPFLAGS to CFLAGS.  This is closer to the true meaning
-  ## of `pkg-config --cflags` output.
-  FT2_CPPFLAGS="$FT2_CFLAGS"
-  AC_SUBST(FT2_CPPFLAGS)
-fi
-
-## Check for Xft library (when using X11)
-
-warn_xft=""
-if test "$have_x" = yes; then
-  PKG_CHECK_MODULES(XFT, [xft],
-    [AC_DEFINE(HAVE_XFT, 1, [Define to 1 if Xft is present.])],
-    [warn_xft="Xft library not found.  OpenGL graphics will not be fully functional."])
-fi
-
-if test -n "$warn_xft"; then
-  OCTAVE_CONFIGURE_WARNING([warn_xft])
-fi
-
-## Check for fontconfig library
-
-warn_fontconfig=""
-PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
-  [AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present.])],
-  [warn_fontconfig="Fontconfig library not found.  OpenGL graphics will not be fully functional."])
-
-if test -n "$warn_fontconfig"; then
-  OCTAVE_CONFIGURE_WARNING([warn_fontconfig])
-else
-  FONTCONFIG_CPPFLAGS="$FONTCONFIG_CFLAGS"
-  ## Alias CPPFLAGS to CFLAGS.  This is closer to the true meaning
-  ## of `pkg-config --cflags` output.
-  FONTCONFIG_CPPFLAGS="$FONTCONFIG_CFLAGS"
-  AC_SUBST(FONTCONFIG_CPPFLAGS)
-fi
-
-## Check for OSMesa which is needed for offscreen rendering with OpenGL
-## FIXME: Check for version >= 9.0.0
-OCTAVE_CHECK_LIB(OSMesa, OSMesa,
-  [OSMesa library not found.  Offscreen rendering with OpenGL will be disabled.],
-  [osmesa.h GL/osmesa.h], [OSMesaCreateContext])
-
-## Check for gl2ps which is required for printing with OpenGL graphics
-
-AC_CHECK_HEADERS([gl2ps.h],
-  [GL2PS_LIBS="-lgl2ps"],
-  [warn_gl2ps="gl2ps library not found.  Printing of OpenGL graphics will be disabled."])
-
-if test -n "$warn_gl2ps"; then
-  OCTAVE_CONFIGURE_WARNING([warn_gl2ps])
-fi
-
-AC_SUBST(GL2PS_LIBS)
-
 ### GUI/Qt related tests.
 
 QT_CPPFLAGS=
@@ -1652,6 +1574,92 @@
   opengl_graphics=yes
 fi
 
+## Check for FreeType 2 library
+
+warn_freetype=""
+if test "$opengl_graphics" = yes; then
+  PKG_CHECK_MODULES([FT2], [freetype2], [
+    min_ft2_version=9.03
+    AC_MSG_CHECKING([for FreeType2 version >= $min_ft2_version])
+    $PKG_CONFIG freetype2 --atleast-version=$min_ft2_version
+    ac_status=$?
+    if test "$ac_status" = 0; then
+      AC_MSG_RESULT(yes)
+      AC_DEFINE(HAVE_FREETYPE, 1, [Define to 1 if you have Freetype library.])
+      save_LIBS="$LIBS"
+      LIBS="$FT2_LIBS $LIBS"
+      AC_CHECK_FUNCS([FT_Reference_Face])
+      LIBS="$save_LIBS"
+    else
+      AC_MSG_RESULT(no)
+      warn_freetype="FreeType library >= 9.03 not found.  OpenGL graphics will not be fully functional."
+    fi])
+fi
+
+if test -n "$warn_freetype"; then
+  OCTAVE_CONFIGURE_WARNING([warn_freetype])
+else
+  FT2_CPPFLAGS="$FT2_CFLAGS"
+  ## Alias CPPFLAGS to CFLAGS.  This is closer to the true meaning
+  ## of `pkg-config --cflags` output.
+  FT2_CPPFLAGS="$FT2_CFLAGS"
+  AC_SUBST(FT2_CPPFLAGS)
+fi
+
+## Check for Xft library (when using X11)
+
+warn_xft=""
+if test "$opengl_graphics" = yes && "$have_x" = yes; then
+  PKG_CHECK_MODULES(XFT, [xft],
+    [AC_DEFINE(HAVE_XFT, 1, [Define to 1 if Xft is present.])],
+    [warn_xft="Xft library not found.  OpenGL graphics will not be fully functional."])
+fi
+
+if test -n "$warn_xft"; then
+  OCTAVE_CONFIGURE_WARNING([warn_xft])
+fi
+
+## Check for fontconfig library
+
+warn_fontconfig=""
+if test "$opengl_graphics" = yes; then
+  PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
+    [AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present.])],
+    [warn_fontconfig="Fontconfig library not found.  OpenGL graphics will not be fully functional."])
+fi
+
+if test -n "$warn_fontconfig"; then
+  OCTAVE_CONFIGURE_WARNING([warn_fontconfig])
+else
+  FONTCONFIG_CPPFLAGS="$FONTCONFIG_CFLAGS"
+  ## Alias CPPFLAGS to CFLAGS.  This is closer to the true meaning
+  ## of `pkg-config --cflags` output.
+  FONTCONFIG_CPPFLAGS="$FONTCONFIG_CFLAGS"
+  AC_SUBST(FONTCONFIG_CPPFLAGS)
+fi
+
+## Check for OSMesa which is needed for offscreen rendering with OpenGL
+## FIXME: Check for version >= 9.0.0
+if test "$opengl_graphics" = yes; then
+  OCTAVE_CHECK_LIB(OSMesa, OSMesa,
+    [OSMesa library not found.  Offscreen rendering with OpenGL will be disabled.],
+    [osmesa.h GL/osmesa.h], [OSMesaCreateContext])
+fi
+
+## Check for gl2ps which is required for printing with OpenGL graphics
+
+if test "$opengl_graphics" = yes; then
+  AC_CHECK_HEADERS([gl2ps.h],
+    [GL2PS_LIBS="-lgl2ps"],
+    [warn_gl2ps="gl2ps library not found.  Printing of OpenGL graphics will be disabled."])
+fi
+
+if test -n "$warn_gl2ps"; then
+  OCTAVE_CONFIGURE_WARNING([warn_gl2ps])
+fi
+
+AC_SUBST(GL2PS_LIBS)
+
 ### Start determination of shared vs. static libraries
 
 ## Use -static if compiling on Alpha OSF/1 1.3 systems.
