--- src/Makefile.am.orig	2018-03-26 16:27:01.000000000 +0200
+++ src/Makefile.am	2018-03-27 15:20:55.000000000 +0200
@@ -25,7 +25,7 @@
 gnome_internet_radio_locator_CFLAGS  = $(GNOME_INTERNET_RADIO_LOCATOR_CFLAGS) \
 	       -DGNOME_INTERNET_RADIO_LOCATOR_DATADIR=\"$(datadir)/gnome-internet-radio-locator\" \
 	       -DDATADIR=\"$(datadir)\" \
-	       -DGNOME_INTERNET_RADIO_LOCATOR_DEBUG=1 \
+	       -DGNOME_INTERNET_RADIO_LOCATOR_DEBUG=0 \
 	       -DGNOME_INTERNET_RADIO_LOCATOR_CFG \
 	       -DGNOMELOCALEDIR=\"$(datadir)/locale\"
 
