diff --git a/aclocal.m4 b/aclocal.m4
index 8fd4cd1..f6e74d2 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -40,7 +40,7 @@ AC_DEFUN([DUTI_CHECK_SDK],
 	    macosx_arches=""
 	    ;;
 
-	darwin15*|darwin16*|darwin17*)
+	darwin15*|darwin16*|darwin17*|darwin18*|darwin19*|darwin20*)
 	    sdk_path="${sdk_path}/MacOSX.sdk"
 	    macosx_arches=""
 	    ;;
@@ -106,6 +106,18 @@ AC_DEFUN([DUTI_CHECK_DEPLOYMENT_TARGET],
 	darwin17*)
 	    dep_target="10.13"
 	    ;;
+
+	darwin18*)
+		dep_target="10.14"
+		;;
+
+	darwin19*)
+		dep_target="10.15"
+		;;
+
+	darwin20*)
+		dep_target="11.0"
+		;;
     esac
 
     if test -z "$macosx_dep_target"; then
