--- setup.rb.orig       2008-10-22 00:52:26.000000000 +0200
+++ setup.rb    2008-10-22 00:52:30.000000000 +0200
@@ -121,19 +121,8 @@
   lib_dir = Gem::ConfigMap[site_or_vendor]
   bin_dir = Gem::ConfigMap[:bindir]
 else
-  # Apple installed RubyGems into libdir, and RubyGems <= 1.1.0 gets confused
-  # about installation location, so switch back to sitelibdir/vendorlibdir.
-  if defined?(APPLE_GEM_HOME) and
-      # just in case Apple and RubyGems don't get this patched up proper.
-     (prefix == Gem::ConfigMap[:libdir] or
-      # this one is important
-      prefix == File.join(Gem::ConfigMap[:libdir], 'ruby')) then
     lib_dir = Gem::ConfigMap[site_or_vendor]
     bin_dir = Gem::ConfigMap[:bindir]
-  else
-    lib_dir = File.join prefix, 'lib'
-    bin_dir = File.join prefix, 'bin'
-  end
 end
 
 unless install_destdir.empty?
@@ -141,8 +130,13 @@
   lib_dir_p = Pathname.new(lib_dir)
   bin_dir_p = Pathname.new(bin_dir)
 
-  lib_dir = File.join install_destdir, lib_dir_p.relative_path_from(top_dir)
-  bin_dir = File.join install_destdir, bin_dir_p.relative_path_from(top_dir)
+  unless prefix.empty?
+    lib_dir = File.join(install_destdir, prefix, lib_dir_p.relative_path_from(top_dir))
+    bin_dir = File.join(install_destdir, prefix, bin_dir_p.relative_path_from(top_dir))
+  else
+    lib_dir = File.join(install_destdir, lib_dir_p.relative_path_from(top_dir))
+    bin_dir = File.join(install_destdir, bin_dir_p.relative_path_from(top_dir))
+  end
 end
 
 mkdir_p lib_dir
