--- subprojects/keycodemapdb/tools/keymap-gen.orig	2023-01-20 14:40:33
+++ subprojects/keycodemapdb/tools/keymap-gen	2023-07-17 15:04:20
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!@@PYTHON_BIN@@
 # -*- python -*-
 #
 # Keycode Map Generator
--- subprojects/spice-common/meson.build
+++ subprojects/spice-common/meson.build
@@ -127,7 +127,7 @@ endforeach
 # Python
 if spice_common_generate_client_code or spice_common_generate_server_code
   py_module = import('python')
-  python = py_module.find_installation('python3')
+  python = py_module.find_installation('@@PYTHON_BIN@@')
 
   if get_option('python-checks')
     foreach module : ['six', 'pyparsing']
