--- curl-config.in.orig	2012-08-12 00:48:14.000000000 +0000
+++ curl-config.in	2012-08-12 00:49:18.000000000 +0000
@@ -135,9 +135,9 @@
           CPPFLAG_CURL_STATICLIB=""
         fi
        	if test "X@includedir@" = "X/usr/include"; then
-          echo "$(CPPFLAG_CURL_STATICLIB)"
+          echo "${CPPFLAG_CURL_STATICLIB}"
         else
-          echo "$(CPPFLAG_CURL_STATICLIB)-I@includedir@"
+          echo "${CPPFLAG_CURL_STATICLIB}-I@includedir@"
         fi
        	;;
 
