--- lib/vtls/openssl.c.orig	2018-04-21 18:01:03.000000000 +0200
+++ lib/vtls/openssl.c	2018-04-21 18:01:03.000000000 +0200
@@ -117,14 +117,6 @@
 #define X509_get0_notBefore(x) X509_get_notBefore(x)
 #define X509_get0_notAfter(x) X509_get_notAfter(x)
 #define CONST_EXTS /* nope */
-#ifdef LIBRESSL_VERSION_NUMBER
-static unsigned long OpenSSL_version_num(void)
-{
-  return LIBRESSL_VERSION_NUMBER;
-}
-#else
-#define OpenSSL_version_num() SSLeay()
-#endif
 #endif
 
 #if (OPENSSL_VERSION_NUMBER >= 0x1000200fL) && /* 1.0.2 or later */ \
