Disable silent rules.
--- generate/unix/Makefile.rules.orig	2020-02-14 11:33:53.000000000 -0600
+++ generate/unix/Makefile.rules	2020-04-03 00:24:38.000000000 -0500
@@ -11,13 +11,11 @@
 	$(COPYPROG)
 
 $(PROG) : $(INTERMEDIATES) $(MISC) $(OBJECTS)
-	@echo "- Link" $(PROG)
-	@$(LINKPROG)
+	$(LINKPROG)
 	$(RENAMEPROG)
 
 $(OBJDIR)/%.o : %.c $(HEADERS) $(ACPICA_HEADERS)
-	@echo $(PROG)  $<
-	@$(COMPILEOBJ)
+	$(COMPILEOBJ)
 
 clean :
 	@rm -f $(PROG) $(PROG).exe $(OBJECTS) $(OBJDIR)/*.o $(INTERMEDIATES) $(MISC)
