--- compiler/Makefile.orig	2012-01-12 05:23:14.000000000 +1100
+++ compiler/Makefile	2012-07-12 04:57:45.000000000 +1000
@@ -230,7 +230,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
@@ -248,15 +247,11 @@
     -Waggregate-return \
     -Wchar-subscripts \
     -Wempty-body \
-    -Wlogical-op \
     -Wmissing-declarations \
     -Wmissing-field-initializers \
-    -Wmissing-parameter-type \
     -Wnested-externs \
-    -Wold-style-declaration \
     -Wold-style-definition \
-    -Wredundant-decls \
-    -Wtype-limits
+    -Wredundant-decls
 
 #
 # Bison/Flex configuration
@@ -306,16 +301,16 @@
 # by the utilities above and they are not necessarily ANSI C, etc.
 #
 aslcompilerlex.o :   aslcompilerlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -Wstrict-aliasing=0 -o$@ $?
 
 aslcompilerparse.o : aslcompilerparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -Wstrict-aliasing=0 -o$@ $?
 
 dtparserlex.o :      dtparserlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -Wstrict-aliasing=0 -o$@ $?
 
 dtparserparse.o :    dtparserparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -Wstrict-aliasing=0 -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -Wstrict-aliasing=0 -o$@ $?
 
 
 %.o : %.c $(HEADERS) $(ACPICA_HEADERS)
--- generate/unix/iasl/Makefile.orig	2012-01-12 05:23:17.000000000 +1100
+++ generate/unix/iasl/Makefile	2012-07-12 04:34:52.000000000 +1000
@@ -235,16 +235,16 @@
 # by the utilities above and they are not necessarily ANSI C, etc.
 #
 $(OBJDIR)/aslcompilerlex.o :   $(OBJDIR)/aslcompilerlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -o$@ $?
 
 $(OBJDIR)/aslcompilerparse.o : $(OBJDIR)/aslcompilerparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -o$@ $?
 
 $(OBJDIR)/dtparserlex.o :      $(OBJDIR)/dtparserlex.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -o$@ $?
 
 $(OBJDIR)/dtparserparse.o :    $(OBJDIR)/dtparserparse.c
-	$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $?
+	$(CC) -c $(CFLAGS) -Wall -o$@ $?
 
 
 $(OBJDIR)/%.o : %.c $(HEADERS) $(ACPICA_HEADERS)
--- generate/unix/Makefile.config.orig	2012-01-12 05:23:17.000000000 +1100
+++ generate/unix/Makefile.config	2012-07-12 04:35:40.000000000 +1000
@@ -101,7 +101,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
--- tools/acpibin/Makefile.orig	2012-01-12 05:23:21.000000000 +1100
+++ tools/acpibin/Makefile	2012-07-12 04:36:12.000000000 +1000
@@ -77,7 +77,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
--- tools/acpiexec/Makefile.orig	2012-01-12 05:23:22.000000000 +1100
+++ tools/acpiexec/Makefile	2012-07-12 05:01:31.000000000 +1000
@@ -241,7 +241,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
@@ -259,15 +258,11 @@
     -Waggregate-return \
     -Wchar-subscripts \
     -Wempty-body \
-    -Wlogical-op \
     -Wmissing-declarations \
     -Wmissing-field-initializers \
-    -Wmissing-parameter-type \
     -Wnested-externs \
-    -Wold-style-declaration \
     -Wold-style-definition \
-    -Wredundant-decls \
-    -Wtype-limits
+    -Wredundant-decls
 
 #
 # Rules
--- tools/acpihelp/Makefile.orig	2012-01-12 05:23:22.000000000 +1100
+++ tools/acpihelp/Makefile	2012-07-12 04:37:08.000000000 +1000
@@ -68,7 +68,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
--- tools/acpinames/Makefile.orig	2012-01-12 05:23:22.000000000 +1100
+++ tools/acpinames/Makefile	2012-07-12 04:37:37.000000000 +1000
@@ -141,7 +141,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
--- tools/acpisrc/Makefile.orig	2012-01-12 05:23:22.000000000 +1100
+++ tools/acpisrc/Makefile	2012-07-12 04:38:06.000000000 +1000
@@ -70,7 +70,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
--- tools/acpixtract/Makefile.orig	2012-01-12 05:23:22.000000000 +1100
+++ tools/acpixtract/Makefile	2012-07-12 05:05:36.000000000 +1000
@@ -61,7 +61,6 @@
     -Wall \
     -Wbad-function-cast \
     -Wdeclaration-after-statement \
-    -Werror \
     -Wformat=2 \
     -Wmissing-declarations \
     -Wmissing-prototypes \
@@ -79,15 +78,11 @@
     -Waggregate-return \
     -Wchar-subscripts \
     -Wempty-body \
-    -Wlogical-op \
     -Wmissing-declarations \
     -Wmissing-field-initializers \
-    -Wmissing-parameter-type \
     -Wnested-externs \
-    -Wold-style-declaration \
     -Wold-style-definition \
-    -Wredundant-decls \
-    -Wtype-limits
+    -Wredundant-decls
 
 #
 # Rules
