--- src.pro.orig	2014-04-08 13:50:14.000000000 -0500
+++ src.pro	2014-09-29 22:32:16.000000000 -0500
@@ -153,7 +153,7 @@
 
 
 contains(USE_ASPELL,true){
-exists("/usr/include/aspell.h") {
+exists("$$PREFIX/include/aspell.h") {
     message ("ASpell enabled")
     LIBS += -laspell
     DEFINES += ASPELL_ENABLE
@@ -163,7 +163,7 @@
 
 
 contains(USE_HUNSPELL,true){
-exists("/usr/include/hunspell/hunspell.hxx") { 
+exists("$$PREFIX/include/hunspell/hunspell.hxx") { 
     message ("hunspell enabled")
         PKGCONFIG += hunspell
             DEFINES += HUNSPELL_ENABLE
