--- tea-qmake.pro.orig	2021-05-04 12:36:58.000000000 -0500
+++ tea-qmake.pro	2021-05-10 19:16:46.000000000 -0500
@@ -265,7 +265,7 @@
                }
 
        contains (USE_ASPELL,true) {
-                exists ("/usr/include/aspell.h") {
+                exists ("$$PREFIX/include/aspell.h") {
                         message ("ASpell enabled")
                         LIBS += -laspell
                         DEFINES += ASPELL_ENABLE
