--- CMakeLists.txt.orig	2017-10-31 11:28:14.000000000 -0700
+++ CMakeLists.txt	2018-11-02 08:17:24.000000000 -0700
@@ -66,7 +66,7 @@
   set(DEF_INSTALL_CMAKE_DIR CMake)
   set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W0" )
 else()
-  set(DEF_INSTALL_CMAKE_DIR lib/CMake/AcesContainer)
+  set(DEF_INSTALL_CMAKE_DIR lib/cmake/AcesContainer)
 endif()
 set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Install directory for project CMake files" )
 
