diff --git a/old/rule_bison.py b/src/3rdparty/chromium/third_party/blink/renderer/build/scripts/rule_bison.py
index f75e25f..6d96b7e 100755
--- src/3rdparty/chromium/third_party/blink/renderer/build/scripts/rule_bison.py
+++ src/3rdparty/chromium/third_party/blink/renderer/build/scripts/rule_bison.py
@@ -101,7 +101,7 @@ assert outputHTmp != None
 outputHFile = open(outputHTmp)
 outputHContents = outputHFile.read()
 outputHFile.close()
-os.unlink(outputHTmp)
+# not working for bison-3.7: os.unlink(outputHTmp)
 
 # Rewrite the generated header with #include guards.
 outputH = os.path.join(outputDir, inputRoot + '.h')
