--- configure.orig	2012-08-26 19:51:10.000000000 +0200
+++ configure	2012-08-26 19:58:47.000000000 +0200
@@ -1961,13 +1961,20 @@
     11\.*)
       darwin_ver="10.7" # Lion
 
+      CXX_FLAGS="-mmacosx-version-min=10.6 -isysroot $sdk -Wl,-syslibroot,$sdk"
+
       if { test -d "/Developer/SDKs/MacOSX10.6.sdk"; } then
+          # Developer tools are not bundled in Xcode.app
           sdk="/Developer/SDKs/MacOSX10.6.sdk"
-      else
+      elif { test -d "`xcode-select -print-path`/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.6.sdk"; } then
+          # Developer tools are in Xcode.app and there's 10.6 SDK
           sdk="`xcode-select -print-path`/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.6.sdk"
+      else
+          # Developer tools are in Xcode.app and there's no 10.6 SDK, i.e. we're on 10.7 using Xcode >4.4
+          # that only sports 10.7 and 10.8, we need to change CXX_FLAGS too
+          sdk="`xcode-select -print-path`/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.7.sdk"
+          CXX_FLAGS="-mmacosx-version-min=10.7 -isysroot $sdk -Wl,-syslibroot,$sdk"
       fi
-
-      CXX_FLAGS="-mmacosx-version-min=10.6 -isysroot $sdk -Wl,-syslibroot,$sdk"
       ;;
     10\.*)
       darwin_ver="10.6" # Snow Leopard
