--- src/VBox/Runtime/r3/posix/fileaio-posix.cpp.orig	2011-10-27 02:10:41.000000000 -0400
+++ src/VBox/Runtime/r3/posix/fileaio-posix.cpp	2011-10-27 02:10:41.000000000 -0400
@@ -307,7 +307,7 @@
                          &cReqsOutstandingMax,  /* Where to store the old value. */
                          &cbParameter,          /* Size of the memory pointed to. */
                          NULL,                  /* Where the new value is located. */
-                         NULL);                 /* Where the size of the new value is stored. */
+                         0);                    /* Where the size of the new value is stored. */
     if (rcBSD == -1)
         return RTErrConvertFromErrno(errno);
 
@@ -326,7 +326,7 @@
                          &cReqsOutstandingMax,       /* Where to store the old value. */
                          &cbParameter,               /* Size of the memory pointed to. */
                          NULL,                       /* Where the new value is located. */
-                         NULL);                      /* Where the size of the new value is stored. */
+                         0);                         /* Where the size of the new value is stored. */
     if (rcBSD == -1)
     {
         /* ENOENT means the value is unknown thus the module is not loaded. */
