--- skins/PloneWebMail/top_bar.pt.orig	Mon Feb 16 15:59:15 2004
+++ skins/PloneWebMail/top_bar.pt	Mon Feb 16 15:59:34 2004
@@ -29,7 +29,7 @@
   <tr>
     <td width="40">
       <div align="center">
-      <a href="http://www.1000asa.com" target="_blank"><img src="img/PloneWebMail_logo.png" alt="logo PloneWebMail"></a>
+      <a href="http://plonewebmail.1000asa.com/" target="_blank"><img src="img/PloneWebMail_logo.png" alt="logo PloneWebMail"></a>
       </div>
     </td>
     <td >
