--- skins/PloneWebMail/PloneWebMail_addressbook_view.pt.orig	Mon Feb 16 16:00:21 2004
+++ skins/PloneWebMail/PloneWebMail_addressbook_view.pt	Mon Feb 16 16:00:37 2004
@@ -33,7 +33,7 @@
 			<h1 i18n:translate="">Addressbook content:</h1>
 		</td>
 		<td align="right">
-			<h1><a tal:define="url python:'javascript:add_contact()'" tal:attributes="href url"><img src="img/addcontact_de.png" alt="add"></a></h1>
+			<h1><a tal:define="url python:'javascript:add_contact()'" tal:attributes="href url"><img src="img/addcontact.png" alt="add"></a></h1>
 		</td>
 	</tr>
 </table>
