--- enthought/traits/ui/wx/helper.py    (revision 23572)
+++ enthought/traits/ui/wx/helper.py    (working copy)
@@ -413,5 +413,8 @@
             This method overrides the original in the base class so that
             nested subpanels are handled correctly.
         """
+        if child is None:
+            return
+
         sppux, sppuy = self.GetScrollPixelsPerUnit()
         vsx, vsy     = self.GetViewStart()
