diff -NaurdwB ./pyproject.toml ./pyproject.toml
--- ./pyproject.toml	2025-11-11 07:23:47
+++ ./pyproject.toml	2025-11-11 07:24:54
@@ -4,9 +4,7 @@
 requires = [
     # 70.1.0: min version for integrated bdist_wheel command from wheel package
     # 77.0.0: min version for SPDX expression support for project.license
-    "setuptools>=70.1.0,<80.0",
-    "cmake>=3.27",
-    "ninja",
+    "setuptools",
     "numpy",
     "packaging",
     "pyyaml",
