--- autogen.sh.orig	2015-02-11 06:21:07.000000000 -0800
+++ autogen.sh	2015-02-11 06:23:56.000000000 -0800
@@ -88,7 +88,13 @@
 
 
 echo -n "checking for automake >= $AUTOMAKE_REQUIRED_VERSION ... "
-if (automake-1.13 --version) < /dev/null > /dev/null 2>&1; then
+if (automake-1.15 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.15
+   ACLOCAL=aclocal-1.15
+elif (automake-1.14 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.14
+   ACLOCAL=aclocal-1.14
+elif (automake-1.13 --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=automake-1.13
    ACLOCAL=aclocal-1.13
 elif (automake-1.12 --version) < /dev/null > /dev/null 2>&1; then
