--- configure.in.orig	2009-03-11 13:03:02.000000000 -0700
+++ configure.in	2009-03-11 13:09:59.000000000 -0700
@@ -178,7 +178,7 @@
 AC_ARG_ENABLE(ff_libfaad,
               [  --disable-ff-libfaad          configure libavformat without libfaad])
   if test "x$enable_ff_libfaad" != "xno"; then
-    AC_CHECK_LIB(faad, faacDecOpen,
+    AC_CHECK_LIB(faad, NeAACDecOpen,
         [AC_CHECK_HEADER(faad.h,
           FF_LIBFAAD="-lfaad",
           faad_warn="$NEW_LINE ** faad header file (faad.h) not found (not critical)")],
