--- autogen.sh.orig	2013-02-09 22:46:18.000000000 -0800
+++ autogen.sh	2013-02-09 22:50:53.000000000 -0800
@@ -88,7 +88,10 @@
 
 
 echo -n "checking for automake >= $AUTOMAKE_REQUIRED_VERSION ... "
-if (automake-1.12 --version) < /dev/null > /dev/null 2>&1; then
+if (automake-1.13 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.13
+   ACLOCAL=aclocal-1.13
+elif (automake-1.12 --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=automake-1.12
    ACLOCAL=aclocal-1.12
 elif (automake-1.11 --version) < /dev/null > /dev/null 2>&1; then
