--- autogen.sh.orig	2015-08-10 20:16:26.000000000 -0700
+++ autogen.sh	2015-08-10 20:16:52.000000000 -0700
@@ -24,12 +24,12 @@
         exit 1
 fi
 
-git submodule update --init --recursive
+#git submodule update --init --recursive
 
-if [ $? != 0 ]; then
-    echo "*** Failed to download submodules. Maybe you have a bad connection or a submodule was not forked?"
-    exit 1
-fi
+#if [ $? != 0 ]; then
+#    echo "*** Failed to download submodules. Maybe you have a bad connection or a submodule was not forked?"
+#    exit 1
+#fi
 
 autopoint --force
 AUTOPOINT='intltoolize --automake --copy' autoreconf --force --install --verbose
