--- libgitg-ext/meson.build.orig	2020-08-22 14:51:30.000000000 -0700
+++ libgitg-ext/meson.build	2020-08-22 15:31:43.000000000 -0700
@@ -125,10 +125,9 @@
 
 enable_python = get_option('python')
 if enable_python
-  python = import('python3')
-
+  python = import('python').find_installation('@@PYTHON3_BIN@@')
   install_data(
     'GitgExt.py',
-    install_dir: join_paths(python.sysconfig_path('purelib'), 'gi', 'overrides'),
+    install_dir: join_paths(python.get_path('purelib'), 'gi', 'overrides'),
   )
 endif
