--- autogen.sh.orig	2016-08-25 15:20:16.000000000 -0700
+++ autogen.sh	2016-09-02 02:44:20.000000000 -0700
@@ -24,18 +24,18 @@
         exit 1
 fi
 
-LIBTOOL=`which libtoolize`
+LIBTOOL=`which glibtoolize`
 if test -z $LIBTOOL; then
         echo "*** No libtool found, please install it ***"
         exit 1
 fi
 
-git submodule update --init --recursive
+#git submodule update --init --recursive
 
-if [ $? != 0 ]; then
-    echo "*** Failed to download submodules. Maybe you have a bad connection or a submodule was not forked?"
-    exit 1
-fi
+#if [ $? != 0 ]; then
+#    echo "*** Failed to download submodules. Maybe you have a bad connection or a submodule was not forked?"
+#    exit 1
+#fi
 
 autopoint --force
 AUTOPOINT='intltoolize --automake --copy' autoreconf --force --install --verbose
