--- lib/mrwfile.cpp.orig	2023-06-28 08:40:14.000000000 -0400
+++ lib/mrwfile.cpp	2023-06-28 08:54:24.000000000 -0400
@@ -26,6 +26,9 @@
 #include <string>
 #include <memory>
 
+#include <machine/endian.h>
+#define be16toh(x) ntohs(x)
+
 #include <libopenraw/cameraids.h>
 #include <libopenraw/debug.h>
 
--- lib/peffile.cpp.orig	2023-06-28 08:40:22.000000000 -0400
+++ lib/peffile.cpp	2023-06-28 08:55:07.000000000 -0400
@@ -21,6 +21,9 @@
 
 #include <libopenraw/cameraids.h>
 
+#include <machine/endian.h>
+#define be16toh(x) ntohs(x)
+
 #include "rawdata.hpp"
 #include "ifd.hpp"
 #include "ifdfilecontainer.hpp"
