--- lib/jfifcontainer.cpp.orig	2011-11-13 01:00:09.000000000 -0600
+++ lib/jfifcontainer.cpp	2016-03-31 11:09:29.000000000 -0500
@@ -145,7 +145,7 @@
 		{
 			int ret = 0;
 			if (::setjmp(m_jpegjmp) == 0) {
-				ret = JPEG::jpeg_read_header(&m_cinfo, TRUE);
+				ret = JPEG::jpeg_read_header(&m_cinfo, JPEG::TRUE);
 				//Trace(DEBUG1) << "jpeg_read_header " << ret << "\n";
 				
 				JPEG::jpeg_calc_output_dimensions(&m_cinfo);
@@ -181,7 +181,7 @@
 				src->pub.next_input_byte = NULL;
 				src->pub.bytes_in_buffer = 0;
 			}
-			return TRUE;
+			return JPEG::TRUE;
 		}
 
 
