https://bugs.php.net/bug.php?id=62006
--- cairo_wrapper.c.orig	2008-03-28 17:20:13.000000000 -0500
+++ cairo_wrapper.c	2012-05-11 14:01:27.000000000 -0500
@@ -127,7 +127,7 @@
 /* }}} */
 
 /* {{{ cairo_wrapper_functions[] */
-function_entry cairo_wrapper_functions[] = {
+zend_function_entry cairo_wrapper_functions[] = {
 	PHP_FE(cairo_image_surface_create, cairo_image_surface_create_arg_info)
 	PHP_FE(cairo_surface_destroy, cairo_surface_destroy_arg_info)
 	PHP_FE(cairo_create        , cairo_create_arg_info)
