--- Makefile.ref.orig	2006-03-06 03:31:04.000000000 +0000
+++ Makefile.ref	2006-03-06 03:33:03.000000000 +0000
@@ -61,11 +61,11 @@
 
 ifdef JS_THREADSAFE
 DEFINES += -DJS_THREADSAFE
-INCLUDES += -I../../dist/$(OBJDIR)/include
+INCLUDES += -I__PREFIX__/include/nspr
 ifdef USE_MSVC
 OTHER_LIBS += ../../dist/$(OBJDIR)/lib/libnspr${NSPR_LIBSUFFIX}.lib
 else
-OTHER_LIBS += -L../../dist/$(OBJDIR)/lib -lnspr${NSPR_LIBSUFFIX}
+OTHER_LIBS += -L__PREFIX__$(LIBDIR) -lnspr${NSPR_LIBSUFFIX}
 endif
 endif
 
