--- Makefile.ref.orig	2008-03-27 00:51:10.000000000 +0900
+++ Makefile.ref	2008-03-27 00:52:01.000000000 +0900
@@ -63,11 +63,11 @@
 
 ifdef JS_THREADSAFE
 DEFINES += -DJS_THREADSAFE
-INCLUDES += -I$(DIST)/include/nspr
+INCLUDES += -I__PREFIX__/include/nspr
 ifdef USE_MSVC
 OTHER_LIBS += $(DIST)/lib/libnspr$(NSPR_LIBSUFFIX).lib
 else
-OTHER_LIBS += -L$(DIST)/lib -lnspr$(NSPR_LIBSUFFIX)
+OTHER_LIBS += -L__PREFIX__/lib -lnspr$(NSPR_LIBSUFFIX)
 endif
 endif
 
