--- Makefile.orig	2008-05-11 00:03:56.000000000 +0200
+++ Makefile	2008-06-09 13:46:27.000000000 +0200
@@ -1,23 +1,26 @@
-BINS=isutf8 ifdata ifne pee sponge mispipe lckdo
+BINS=isutf8 pee sponge mispipe lckdo
 PERLSCRIPTS=vidir vipe ts combine zrun
-MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 mispipe.1 lckdo.1
+MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 pee.1 zrun.1 mispipe.1 lckdo.1
 CFLAGS=-O2 -g -Wall
 INSTALL_BIN?=install -s
 
 DOCBOOK2XMAN="docbook2x-man"
 
-all: $(BINS) $(MANS)
+all: $(BINS)
+
+doc: $(MANS)
 
 clean:
 	rm -f $(BINS) $(MANS)
 
 install:
-	mkdir -p $(DESTDIR)/usr/bin
-	$(INSTALL_BIN) $(BINS) $(DESTDIR)/usr/bin
-	install $(PERLSCRIPTS) $(DESTDIR)/usr/bin
+	mkdir -p $(DESTDIR)/bin
+	$(INSTALL_BIN) $(BINS) $(DESTDIR)/bin
+	install $(PERLSCRIPTS) $(DESTDIR)/bin
 	
-	mkdir -p $(DESTDIR)/usr/share/man/man1
-	install $(MANS) $(DESTDIR)/usr/share/man/man1
+install-doc:
+	mkdir -p $(DESTDIR)/share/man/man1
+	install $(MANS) $(DESTDIR)/share/man/man1
 
 check: isutf8
 	./check-isutf8
