--- Makefile	2017-01-23 09:50:26.000000000 -0500
+++ Makefile	2017-01-23 09:52:52.000000000 -0500
@@ -1,6 +1,6 @@
-BINS=isutf8 ifdata ifne pee sponge mispipe lckdo errno
+BINS=ifdata ifne pee sponge mispipe lckdo errno
 PERLSCRIPTS=vidir vipe ts combine zrun chronic
-MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 chronic.1 mispipe.1 lckdo.1 errno.1
+MANS=sponge.1 vidir.1 vipe.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.1 chronic.1 mispipe.1 lckdo.1 errno.1
 CFLAGS?=-O2 -g -Wall
 INSTALL_BIN?=install -s
 PREFIX?=/usr
@@ -19,10 +19,6 @@
 clean:
 	rm -f $(BINS) $(MANS) dump.c errnos.h errno.o
 
-isutf8: is_utf8/*.c is_utf8/*.h
-	$(MAKE) -C is_utf8/
-	cp is_utf8/isutf8 .
-
 install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	$(INSTALL_BIN) $(BINS) $(DESTDIR)$(PREFIX)/bin
@@ -32,8 +28,6 @@
 	mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1
 	install $(MANS) $(DESTDIR)$(PREFIX)/share/man/man1
 
-check: isutf8
-	./is_utf8/test.sh
 
 %.1: %.docbook
 	xmllint --noout --valid $<
