--- virtualenvwrapper.sh.orig	2011-07-12 17:47:34.000000000 -0600
+++ virtualenvwrapper.sh	2011-07-12 17:50:50.000000000 -0600
@@ -47,13 +47,13 @@
 # Locate the global Python where virtualenvwrapper is installed.
 if [ "$VIRTUALENVWRAPPER_PYTHON" = "" ]
 then
-    VIRTUALENVWRAPPER_PYTHON="$(\which python)"
+    VIRTUALENVWRAPPER_PYTHON="$(\which __PYTHON_BIN__)"
 fi
 
 # Set the name of the virtualenv app to use.
 if [ "$VIRTUALENVWRAPPER_VIRTUALENV" = "" ]
 then
-    VIRTUALENVWRAPPER_VIRTUALENV="virtualenv"
+    VIRTUALENVWRAPPER_VIRTUALENV="__VIRTUALENV_BIN__"
 fi
 
 virtualenvwrapper_derive_workon_home() {