--- virtualenvwrapper_bashrc.orig	2009-08-13 22:32:24.000000000 +1000
+++ virtualenvwrapper_bashrc	2010-11-02 10:44:22.000000000 +1100
@@ -91,7 +91,7 @@
     eval "envname=\$$#"
     virtualenvwrapper_verify_workon_home || return 1
     (cd "$WORKON_HOME" &&
-        virtualenv "$@" &&
+        __VIRTUALENV_BIN__ "$@" &&
         virtualenvwrapper_run_hook "./premkvirtualenv" "$envname"
         )
     # If they passed a help option or got an error from virtualenv,
