--- Makefile.in.orig	2008-11-13 21:23:33.000000000 -0800
+++ Makefile.in	2008-11-13 21:26:01.000000000 -0800
@@ -6,6 +6,7 @@
 LIBDIR = @libdir@
 EMACSDIR = @emacsdir@
 BUILDDIR = @builddir@
+DESTDIR =
 COMPILE= $(JAVAC) $(JFLAGS) -d src $(TOOLSLIB) $(SRC)
 
 SHELL = /bin/sh
@@ -103,16 +104,16 @@
 	fi
 
 install: $(JARFILE)
-	-mkdir -p $(LIBDIR)
-	-cp -p $(JARFILE) $(LIBDIR);
-	-mkdir -p $(BINDIR)
-	-cp bin/gjdb $(BINDIR)/gjdb
-	-cp bin/meta-gjdb $(BINDIR)/meta-gjdb
-	chmod 755 $(BINDIR)/gjdb $(BINDIR)/meta-gjdb
-	-mkdir -p $(EMACSDIR)
-	-cp $(SRCDIR)/lib/emacs/lisp/gjdb.el $(EMACSDIR)
-	-$(EMACS) -batch -f batch-byte-compile $(EMACSDIR)/gjdb.el
-	-chmod 644 $(EMACSDIR)/gjdb.el $(EMACSDIR)/gjdb.elc
+	-mkdir -p $(DESTDIR)$(LIBDIR)
+	-cp -p $(JARFILE) $(DESTDIR)$(LIBDIR);
+	-mkdir -p $(DESTDIR)$(BINDIR)
+	-cp bin/gjdb $(DESTDIR)$(BINDIR)/gjdb
+	-cp bin/meta-gjdb $(DESTDIR)$(BINDIR)/meta-gjdb
+	chmod 755 $(DESTDIR)$(BINDIR)/gjdb $(DESTDIR)$(BINDIR)/meta-gjdb
+	-mkdir -p $(DESTDIR)$(EMACSDIR)
+	-cp $(SRCDIR)/lib/emacs/lisp/gjdb.el $(DESTDIR)$(EMACSDIR)
+	-$(EMACS) -batch -f batch-byte-compile $(DESTDIR)$(EMACSDIR)/gjdb.el
+	-chmod 644 $(DESTDIR)$(EMACSDIR)/gjdb.el $(DESTDIR)$(EMACSDIR)/gjdb.elc
 
 install-doc:
 	cd doc; $(MAKE) install
