diff --git a/cpp/poppler-global.h b/cpp/poppler-global.h
index 782e07585f1bad69aafe8cc3ee40e7df03ed5ffa..e11849bef78437162ddbf2b3d0dad21328a19f09 100644
--- cpp/poppler-global.h
+++ cpp/poppler-global.h
@@ -98,7 +98,7 @@ typedef unsigned int /* time_t */ time_type;
 #    pragma warning(push)
 #    pragma warning(disable : 4251) /* class 'A' needs to have dll interface for to be used by clients of class 'B'. */
 #endif
-class POPPLER_CPP_EXPORT ustring : public std::basic_string<unsigned short>
+class POPPLER_CPP_EXPORT ustring : public std::basic_string<char16_t>
 {
 public:
     ustring();
