--- glib/CMakeLists.txt.orig
+++ glib/CMakeLists.txt
@@ -126,30 +126,38 @@
 # GObject Introspection
 if (HAVE_INTROSPECTION AND BUILD_SHARED_LIBS)
   include(GObjectIntrospectionMacros)
+
+  # General gir: Reset object-list for introspection & load tool args
   set(INTROSPECTION_GIRS)
-  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR} --warn-all")
-  set(INTROSPECTION_SCANNER_ARGS "--library-path=${CMAKE_CURRENT_BINARY_DIR}")
+  set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR}" "--warn-all")
+  list(APPEND INTROSPECTION_SCANNER_ARGS "--library-path=${CMAKE_CURRENT_BINARY_DIR}")
   set(INTROSPECTION_COMPILER_ARGS "--includedir=${CMAKE_CURRENT_SOURCE_DIR}")
 
-  set(introspection_files ${poppler_glib_SRCS} ${poppler_glib_public_headers})
+  # Poppler: Assign package to gir & export keys
   set(Poppler_0_18_gir "poppler-glib")
-  set(Poppler_0_18_gir_INCLUDES GObject-2.0 Gio-2.0 cairo-1.0)
+  set(Poppler_0_18_gir_EXPORT_PACKAGES "poppler-glib")
+  # Then load library and header lists
+  set(Poppler_0_18_gir_LIBS "poppler-glib" "poppler")
+  set(Poppler_0_18_gir_INCLUDES "GObject-2.0" "Gio-2.0" "cairo-1.0")
+
+  # Format list of include directories as compiler flags
   get_directory_property(_tmp_includes INCLUDE_DIRECTORIES)
   _list_prefix(_includes _tmp_includes "-I")
+  # And set flags for gir compiler and scanner
   set(Poppler_0_18_gir_CFLAGS ${_includes} -L${CMAKE_BINARY_DIR} -L${CMAKE_CURRENT_BINARY_DIR})
-  set(Poppler_0_18_gir_LIBS poppler-glib poppler)
-  _list_prefix(_abs_introspection_files introspection_files "${CMAKE_CURRENT_SOURCE_DIR}/")
-  list(APPEND _abs_introspection_files
-    ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.c
-    ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.h
-    ${CMAKE_CURRENT_BINARY_DIR}/poppler-features.h
-  )
-  set(Poppler_0_18_gir_FILES ${_abs_introspection_files})
-  set(Poppler_0_18_gir_SCANNERFLAGS --c-include poppler.h)
-  set(Poppler_0_18_gir_EXPORT_PACKAGES poppler-glib)
-
+  set(Poppler_0_18_gir_SCANNERFLAGS "--c-include=poppler.h")
+
+  # Load temporary source-file lists, including a few generated at build
+  set(orig_introspect_srcs ${poppler_glib_SRCS} ${poppler_glib_public_headers})
+  set(gen_introspect_srcs "poppler-enums.c" "poppler-enums.h" "poppler-features.h")
+  # Prefix the files with their correct directories for full paths
+  _list_prefix(_orig_introspect_paths orig_introspect_srcs "${CMAKE_CURRENT_SOURCE_DIR}/")
+  _list_prefix(_gen_introspect_paths gen_introspect_srcs "${CMAKE_CURRENT_BINARY_DIR}/")
+  # Now load them to the final file list
+  set(Poppler_0_18_gir_FILES ${_orig_introspect_paths} ${_gen_introspect_paths})
+
+  # Finally, load the list of objects for introspection & invoke macro
   list(APPEND INTROSPECTION_GIRS Poppler-0.18.gir)
-
   gir_add_introspections(INTROSPECTION_GIRS)
 endif ()
 
