--- glib/CMakeLists.txt.orig	2023-11-04 19:56:34
+++ glib/CMakeLists.txt	2023-11-04 19:58:09
@@ -123,6 +123,8 @@
   set(INTROSPECTION_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR}" "--warn-all")
   list(APPEND INTROSPECTION_SCANNER_ARGS "--library-path=${CMAKE_CURRENT_BINARY_DIR}")
   set(INTROSPECTION_COMPILER_ARGS ${INTROSPECTION_COMPILER_ARGS} "--includedir=${CMAKE_CURRENT_SOURCE_DIR}")
+  get_target_property(POPPLER_GLIB_SOVERSION poppler-glib SOVERSION)
+  list(APPEND INTROSPECTION_COMPILER_ARGS "--shared-library=${CMAKE_INSTALL_NAME_DIR}/libpoppler-glib.${POPPLER_GLIB_SOVERSION}" "--shared-library=${CMAKE_INSTALL_NAME_DIR}/libpoppler.${POPPLER_SOVERSION_NUMBER}")
 
   # Poppler: Assign package to gir & export keys
   set(Poppler_0_18_gir "poppler-glib")
