--- trunk/CMakeLists.txt.orig	2015-02-24 20:21:03.000000000 +0700
+++ trunk/CMakeLists.txt	2023-01-05 01:27:14.000000000 +0700
@@ -31,13 +31,16 @@
  OPTION( USE_64_BIT "Set to on if you want to compile Win64" OFF )
 ENDIF(WIN32)
 
-
 # include specific modules
 SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
+OPTION(BUILD_DOCS "Build documentation" OFF)
+
 ADD_SUBDIRECTORY (src)
 ADD_SUBDIRECTORY (scripts)
+IF (BUILD_DOCS)
 ADD_SUBDIRECTORY (doc)
+ENDIF()
 
 IF (WIN32)
 MESSAGE( STATUS "-------------------------------------------------------------------------------" )
@@ -49,4 +52,3 @@
 SET( USE_64_BIT "${USE_64_BIT}" CACHE BOOL
     "Set to ON to build Win64" FORCE )
 ENDIF()
-

--- trunk/src/CMakeLists.txt.orig	2015-01-10 23:36:16.000000000 +0700
+++ trunk/src/CMakeLists.txt	2023-01-05 01:41:40.000000000 +0700
@@ -1,6 +1,6 @@
 # Set output directory
-SET(EXECUTABLE_OUTPUT_PATH ${CMAKE_INSTALL_PREFIX}/bin)
-SET(LIBRARY_OUTPUT_PATH ${CMAKE_INSTALL_PREFIX}/lib)
+SET(EXECUTABLE_OUTPUT_PATH @OUTPUT_PREFIX@/bin)
+SET(LIBRARY_OUTPUT_PATH @OUTPUT_PREFIX@/lib)
 
 ADD_DEFINITIONS("-D_LARGE_FILES")
 ADD_DEFINITIONS("-D_FILE_OFFSET_BITS=64")
