--- DESCRIPTION	2024-02-21 06:50:07.000000000 +0800
+++ DESCRIPTION	2024-03-15 12:34:01.000000000 +0800
@@ -5,7 +5,7 @@
 Date: 2024-02-20
 Depends: Matrix, terra, R (>= 3.5.0)
 Imports: abind, numDeriv, methods, stats
-Suggests: RandomFields, parallel, mapmisc, ellipse, pracma, knitr
+Suggests: RandomFields, parallel, ellipse, pracma, knitr
 Enhances: INLA, diseasemapping, geoR, mvtnorm
 LinkingTo: Matrix (>= 1.6-2)
 Additional_repositories: https://inla.r-inla-download.org/R/testing
