--- install.sh.orig	2023-05-11 23:59:51.000000000 +0800
+++ install.sh	2023-05-14 01:49:41.000000000 +0800
@@ -31,16 +31,7 @@
     fi
 }
 
-# Return value of the latest published release on GitHub, with no heading "v" (e.g., "0.7.0")
-get_latest_release()
-{
-     $2 "https://api.github.com/repos/$1/releases/latest" | # Get latest release from GitHub api
-     grep '"tag_name":'        |                            # Get tag line
-     sed -E 's/.*"([^"]+)".*/\1/' |                         # Pluck JSON value
-     sed -E 's/^v//'                                        # Remove heading "v" if present
-}
-
-PREFIX="$HOME/.local"
+PREFIX="@PREFIX@"
 
 while [ "$1" != "" ]; do
     PARAM=$(echo "$1" | awk -F= '{print $1}')
@@ -71,18 +62,11 @@
   exit 2
 fi
 
-LATEST_RELEASE=$(get_latest_release "fortran-lang/fpm" "$FETCH")
-
-if [ -z "$LATEST_RELEASE" ]; then
-  echo "Could not fetch the latest release from GitHub. Install curl or wget, and ensure network connectivity."
-  exit 3
-fi
-
-SOURCE_URL="https://github.com/fortran-lang/fpm/releases/download/v${LATEST_RELEASE}/fpm-${LATEST_RELEASE}.F90"
+SOURCE_URL="https://github.com/fortran-lang/fpm/releases/download/v@VERSION@/fpm-@VERSION@.F90"
 BOOTSTRAP_DIR="build/bootstrap"
 
 if [ -z ${FC+x} ]; then
-    FC="gfortran"
+    FC=@FC@
 fi
 if [ -z ${FFLAGS+x} ]; then
     FFLAGS="-g -fbacktrace -O3"
