--- install.sh.orig	2023-04-12 21:20:20.000000000 +0800
+++ install.sh	2023-04-15 16:46:33.000000000 +0800
@@ -40,7 +40,7 @@
      sed -E 's/^v//'                                        # Remove heading "v" if present
 }
 
-PREFIX="$HOME/.local"
+PREFIX="@PREFIX@"
 
 while [ "$1" != "" ]; do
     PARAM=$(echo "$1" | awk -F= '{print $1}')
@@ -82,7 +82,7 @@
 BOOTSTRAP_DIR="build/bootstrap"
 
 if [ -z ${FC+x} ]; then
-    FC="gfortran"
+    FC=@FC@
 fi
 if [ -z ${FFLAGS+x} ]; then
     FFLAGS="-g -fbacktrace -O3"
