--- configure.orig	2022-12-19 20:03:06.000000000 +0400
+++ configure	2022-12-29 19:50:49.000000000 +0400
@@ -788,17 +788,8 @@
         else
             LDFLAGS="$LDFLAGS -L$prefix/$libdir"
         fi
-        if test -d /sw/bin ; then
-            alt_macosx_dir="/sw"
-            CFLAGS_DIR="-I/sw/include $CFLAGS_DIR"
-            LDFLAGS="-L/sw/lib $LDFLAGS"
-        elif test -d /opt/local/bin ; then
-            alt_macosx_dir="/opt/local"
-            CFLAGS_DIR="-I/opt/local/include $CFLAGS_DIR"
-            LDFLAGS="-L/opt/local/lib $LDFLAGS"
-        fi
         CFLAGS="$CFLAGS_DIR $CFLAGS"
-        Mac_Applications="/Applications"
+        Mac_Applications="@APPLICATIONS_DIR@"
         SHFLAGS="-dynamiclib"
         DYN_LIB_SUFFIX=".dylib"
         extralibs=""
