--- src/filter_core/filter_props.c.orig
+++ src/filter_core/filter_props.c
@@ -30,6 +30,24 @@
 //for base64 decode
 #include <gpac/base_coding.h>
 
+void *
+memrchr(s, c, n)
+    const void *s;
+    int c;
+    size_t n;
+{
+    const unsigned char *cp;
+
+    if (n != 0) {
+	cp = (unsigned char *)s + n;
+	do {
+	    if (*(--cp) == (unsigned char)c)
+		return((void *)cp);
+	} while (--n != 0);
+    }
+    return((void *)0);
+}
+
 typedef u32(*cst_parse_proto)(const char *val);
 typedef const char *(*cst_name_proto)(u32 val);
 
