--- configure.orig	2016-02-19 07:01:44.000000000 -0600
+++ configure	2016-02-23 10:40:59.000000000 -0600
@@ -546,17 +546,7 @@
         xul_flags=-DXP_MAC
         CFLAGS_DIR="-I$prefix/include"
         LDFLAGS="-L$prefix/lib"
-        if test -d /sw/bin ; then
-            alt_macosx_dir="/sw"
-            CFLAGS_DIR="-I/sw/include $CFLAGS_DIR"
-            LDFLAGS="-L/sw/lib $LDFLAGS"
-        elif test -d /opt/local/bin ; then
-            alt_macosx_dir="/opt/local"
-            CFLAGS_DIR="-I/opt/local/include $CFLAGS_DIR"
-            LDFLAGS="-L/opt/local/lib $LDFLAGS"
-        fi
-        cc="cc"
-        Mac_Applications="/Applications"
+        Mac_Applications="@APPLICATIONS_DIR@"
         SHFLAGS="-dynamiclib"
         DYN_LIB_SUFFIX=".dylib"
         extralibs=""
