--- configure.orig
+++ configure
@@ -5981,7 +5981,6 @@ $as_echo "yes" >&6; }
 			else
 				{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
-				as_fn_error $? "Run 'sudo mkdir -p /usr/local/include/sandbox' and 'sudo touch /usr/local/include/sandbox/private.h' to build CUPS." "$LINENO" 5
 			fi
 		fi
 
@@ -6905,7 +6904,7 @@ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
 	fi
 
 	# Add useful warning options for tracking down problems...
-	WARNING_OPTIONS="-Wall -Wno-format-y2k -Wunused -Wno-unused-result -Wsign-conversion"
+	WARNING_OPTIONS="-Wall -Wno-format-y2k -Wunused"
 
 	# Test GCC version for certain warning flags since -Werror
 	# doesn't trigger...
