Upstream: https://github.com/OpenPrinting/cups/pull/229

--- cups/ppd-util.c.orig
+++ cups/ppd-util.c
@@ -226,16 +226,20 @@ cupsGetPPD3(http_t     *http,		/* I  - HTTP connection or @code CUPS_HTTP_DEFAUL
 	* per-process TMPDIR value.
 	*/
 
+#ifdef _CS_DARWIN_USER_TEMP_DIR
         char		tmppath[1024];	/* Temporary directory */
+#endif /* _CS_DARWIN_USER_TEMP_DIR */
 
 	if ((tmpdir = getenv("TMPDIR")) != NULL && access(tmpdir, W_OK))
 	  tmpdir = NULL;
 
 	if (!tmpdir)
 	{
+#ifdef _CS_DARWIN_USER_TEMP_DIR
 	  if (confstr(_CS_DARWIN_USER_TEMP_DIR, tmppath, sizeof(tmppath)))
 	    tmpdir = tmppath;
 	  else
+#endif /* _CS_DARWIN_USER_TEMP_DIR */
 	    tmpdir = "/private/tmp";		/* This should never happen */
 	}
 #else
--- cups/tempfile.c.orig
+++ cups/tempfile.c
@@ -38,9 +38,9 @@ cupsTempFd(char *filename,		/* I - Pointer to buffer */
   int		fd;			/* File descriptor for temp file */
   int		tries;			/* Number of tries */
   const char	*tmpdir;		/* TMPDIR environment var */
-#if defined(__APPLE__) || defined(_WIN32)
+#if (defined(__APPLE__) && defined(_CS_DARWIN_USER_TEMP_DIR)) || defined(_WIN32)
   char		tmppath[1024];		/* Temporary directory */
-#endif /* __APPLE__ || _WIN32 */
+#endif /* (__APPLE__ && _CS_DARWIN_USER_TEMP_DIR) || _WIN32 */
 #ifdef _WIN32
   DWORD		curtime;		/* Current time */
 #else
@@ -72,9 +72,11 @@ cupsTempFd(char *filename,		/* I - Pointer to buffer */
 
   if (!tmpdir)
   {
+#ifdef _CS_DARWIN_USER_TEMP_DIR
     if (confstr(_CS_DARWIN_USER_TEMP_DIR, tmppath, sizeof(tmppath)))
       tmpdir = tmppath;
     else
+#endif
       tmpdir = "/private/tmp";		/* This should never happen */
   }
 
