--- apps/CMakeLists.txt.orig	2016-12-11 11:56:14.000000000 -0800
+++ apps/CMakeLists.txt	2016-12-11 11:57:49.000000000 -0800
@@ -21,7 +21,7 @@
 
 # Inspired by http://bloerg.net/2012/11/10/cmake-and-distutils.html
 
-FIND_PROGRAM(PYTHON "python3")
+FIND_PROGRAM(PYTHON "python@@PYTHON_VERSION@@")
 IF(PYTHON)
     SET(SETUP_PY_IN "${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in")
     SET(SETUP_PY    "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
