diff -r d5d9177c0045 setup.py
--- a/setup.py	Mon Mar 08 16:55:39 2021 -0500
+++ b/setup.py	Fri Apr 16 18:33:21 2021 +0200
@@ -1411,7 +1411,9 @@
 class RustExtension(Extension):
     """Base classes for concrete Rust Extension classes."""
 
-    rusttargetdir = os.path.join('rust', 'target', 'release')
+    rusttargetdir = os.path.join(
+        'rust', 'target', os.getenv('CARGO_BUILD_TARGET', '.'), 'release'
+    )
 
     def __init__(
         self, mpath, sources, rustlibname, subcrate, py3_features=None, **kw
@@ -1462,7 +1464,7 @@
 
             env['HOME'] = pwd.getpwuid(os.getuid()).pw_dir
 
-        cargocmd = ['cargo', 'rustc', '--release']
+        cargocmd = ['cargo', 'rustc', '--release', '--frozen', '-j', '@BUILD_JOBS@']
 
         feature_flags = []
 
@@ -1764,7 +1766,7 @@
     # XCode 4.0 dropped support for ppc architecture, which is hardcoded in
     # distutils.sysconfig
     if xcode4:
-        os.environ['ARCHFLAGS'] = ''
+        pass
 
 setup(name='mercurial',
       version=setupversion,
