--- setuptools/_distutils/spawn.py.orig	2021-03-19 19:19:54.000000000 +1100
+++ setuptools/_distutils/spawn.py	2021-03-19 19:54:58.000000000 +1100
@@ -64,11 +64,26 @@
             # than that used when the interpreter was built. This ensures
             # extension modules are built with correct compatibility values
             cur_target = os.environ.get('MACOSX_DEPLOYMENT_TARGET', _cfg_target)
-            if _cfg_target_split > [int(x) for x in cur_target.split('.')]:
-                my_msg = ('$MACOSX_DEPLOYMENT_TARGET mismatch: '
-                          'now "%s" but "%s" during configure'
-                                % (cur_target, _cfg_target))
-                raise DistutilsPlatformError(my_msg)
+            cur_target_split = [int(x) for x in cur_target.split('.')]
+            if _cfg_target_split > cur_target_split:
+                # Before macOS 11 Big Sur, major OS version was 10.x
+                # Now it's 11, 12, etc. But Big Sur can sometimes identify
+                # itself as 10.16 for compatibility.
+                target_majors = []
+                for vers in [_cfg_target_split, cur_target_split]:
+                    if vers[0] >= 11:
+                        target_majors.append(vers[0])
+                    elif (vers[0], vers[1]) == (10, 16):
+                        target_majors.append(11)
+                    else:
+                        target_majors.append((vers[0], vers[1]))
+
+                if target_majors[0] != target_majors[1]:
+                    # not just a difference in minor version
+                    my_msg = ('$MACOSX_DEPLOYMENT_TARGET mismatch: '
+                              'now "%s" but "%s" during configure'
+                                    % (cur_target, _cfg_target))
+                    raise DistutilsPlatformError(my_msg)
             env.update(MACOSX_DEPLOYMENT_TARGET=cur_target)
 
     try:
