--- Makefile.in.orig	2009-03-09 02:26:24.000000000 -0500
+++ Makefile.in	2012-09-07 03:11:36.000000000 -0500
@@ -184,7 +184,7 @@
 man_MANS = xteddy.6
 bin_SCRIPTS = xtoys xteddy_test
 AM_CFLAGS = @IMLIB2_CFLAGS@
-AM_LDFLAGS = @IMLIB2_LIBS@ -lX11
+AM_LDFLAGS = @IMLIB2_LIBS@ -lX11 -lXext
 SUBDIRS = html images
 EXTRA_DIST = $(man_MANS)  $(bin_SCRIPTS) \
              README xteddy.README images.credit \
