--- DESCRIPTION	2023-10-27 04:30:02.000000000 +0800
+++ DESCRIPTION	2023-10-27 08:17:51.000000000 +0800
@@ -42,7 +42,7 @@
         methods, progressr, Rcpp (>= 1.0.5), rlang (>= 0.4.7), stats,
         tools, utils, spam, lifecycle, generics
 Suggests: DelayedArray, fs (>= 1.5.2), ggplot2, HDF5Array, rmarkdown,
-        testthat, BPCells, sf (>= 1.0.0)
+        testthat, sf (>= 1.0.0)
 Collate: 'RcppExports.R' 'zzz.R' 'generics.R' 'keymixin.R' 'graph.R'
         'default.R' 'assay.R' 'logmap.R' 'layers.R' 'assay5.R'
         'centroids.R' 'command.R' 'compliance.R' 'data.R' 'jackstraw.R'
