--- configure.orig  2008-06-19 13:44:34.000000000 -0600
+++ configure   2008-09-23 16:28:11.000000000 -0600
@@ -17486,15 +17486,10 @@
                            MYSQL_LIBDIR=/usr/lib
                    fi
                    MYSQL_BINDIR=/usr/bin
-           elif test -f /usr/local/include/mysql/mysql.h; then
-                   MYSQL_INCDIR=/usr/local/include/mysql
-                   if test -f /usr/local/lib64/mysql/libmysqlclient_r.a \
-                        -o -f /usr/local/lib64/mysql/libmysqlclient_r.so; then
-                           MYSQL_LIBDIR=/usr/local/lib64/mysql
-                   else
-                           MYSQL_LIBDIR=/usr/local/lib/mysql
-                   fi
-                   MYSQL_BINDIR=/usr/local/bin
+           elif test -f @@PREFIX@@/include/mysql5/mysql/mysql.h; then
+                  MYSQL_INCDIR=@@PREFIX@@/include/mysql5/mysql
+                  MYSQL_LIBDIR=@@PREFIX@@/lib/mysql5/mysql
+                  MYSQL_BINDIR=@@PREFIX@@/lib/mysql5/bin
            elif test -f /usr/local/include/mysql.h; then
                    MYSQL_INCDIR=/usr/local/include
                    if test -f /usr/local/lib64/libmysqlclient_r.a \
