From f1a1910fe0c0ce45027bc957bba85235baaae611 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 30 Jan 2024 09:36:29 +0800
Subject: [PATCH] libmmg3d_private.h: include inttypes.h for PRId

---
 src/mmg3d/libmmg3d_private.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git src/mmg3d/libmmg3d_private.h src/mmg3d/libmmg3d_private.h
index 4061b48f..0401f01d 100644
--- src/mmg3d/libmmg3d_private.h
+++ src/mmg3d/libmmg3d_private.h
@@ -24,6 +24,12 @@
 #ifndef LIBMMG3D_PRIVATE_H
 #define LIBMMG3D_PRIVATE_H
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
+#include <inttypes.h>
+
 #include "libmmgcommon_private.h"
 #include "PRoctree_3d_private.h"
 
