--- DESCRIPTION	2024-05-06 02:10:02.000000000 +0800
+++ DESCRIPTION	2024-05-06 03:31:28.000000000 +0800
@@ -40,7 +40,7 @@
 BugReports: https://github.com/davidgohel/officer/discussions
 Imports: graphics, grDevices, openssl, R6, ragg, stats, utils, uuid,
         xml2 (>= 1.1.0), zip (>= 2.1.0)
-Suggests: devEMF, doconv (>= 0.3.0), ggplot2, knitr, magick, rmarkdown,
+Suggests: devEMF, ggplot2, knitr, magick, rmarkdown,
         rsvg, testthat
 Encoding: UTF-8
 RoxygenNote: 7.3.1
