--- src/mathfunc.c.orig	2018-12-08 09:58:08.000000000 -0800
+++ src/mathfunc.c	2018-12-08 09:58:43.000000000 -0800
@@ -5033,7 +5033,7 @@
 {
 	gnm_float w;
 	static const gnm_float one_over_e = 1 / M_Egnum;
-	static const gnm_float sqrt_one_over_e = gnm_sqrt (1 / M_Egnum);
+	const gnm_float sqrt_one_over_e = gnm_sqrt (1 / M_Egnum);
 	static const gboolean debug = FALSE;
 	gnm_float wmin, wmax;
 	int i, imax = 20;
