--- configure.ac.orig	2014-12-30 18:35:24.000000000 -0800
+++ configure.ac	2015-01-26 11:16:37.000000000 -0800
@@ -266,11 +266,11 @@
 AC_CHECK_PROG(GDK_PIXBUF_CSOURCE, gdk-pixbuf-csource$ac_exeext, gdk-pixbuf-csource$ac_exeext)
 
 ## this should come after `AC_PROG_CC'
-ifdef([GNOME_COMPILE_WARNINGS],
-	[GNOME_COMPILE_WARNINGS] CFLAGS="$CFLAGS $WARN_CFLAGS",
-	[]
-)
-set_more_warnings=yes
+dnl ifdef([GNOME_COMPILE_WARNINGS],
+dnl	[GNOME_COMPILE_WARNINGS] CFLAGS="$CFLAGS $WARN_CFLAGS",
+dnl	[]
+dnl)
+set_more_warnings=no
 if test "$GCC" = yes -a "x$set_more_warnings" != xno; then
 	dnl Clang needs this option, or else it will appear to support any
 	dnl warning option, only to spew warnings about them later.
