--- src/GNOME_Gnumeric-gtk-orig.xml.in	2013-05-28 16:38:55.000000000 -0700
+++ src/GNOME_Gnumeric-gtk.xml.in	2013-09-08 23:35:27.000000000 -0700
@@ -96,14 +96,11 @@
       </menu>
       <menuitem action="EditRecalc"/>
       <placeholder name="ops"/>
-      <separator/>
       <menuitem action="EditPreferences"/>
     </menu>
     <menu name="View" action="MenuView">
       <menuitem action="ViewNew"/>
       <menuitem action="ViewFreezeThawPanes"/>
-      <menu name="Windows" action="MenuViewWindows">
-      </menu>
       <separator/>
       <menu name="Toolbars" action="MenuViewToolbars">
       </menu>
@@ -326,7 +323,6 @@
       <menuitem action="HelpWeb"/>
       <menuitem action="HelpIRC"/>
       <menuitem action="HelpBug"/>
-      <separator/>
       <menuitem action="HelpAbout"/>
     </menu>
   </menubar>
