--- configure.ac.orig	2018-05-04 08:02:09.000000000 -0700
+++ configure.ac	2018-05-04 08:03:27.000000000 -0700
@@ -870,7 +870,7 @@
     AC_MSG_CHECKING([for python module gi])
     if $PYTHON -c 'import gi' 2>/dev/null; then
         have_python_gi=yes
-	[PYTHON_GIOVERRIDESDIR=`python -c 'import gi; import os.path; print(os.path.dirname(gi.__file__)+"/overrides")'`]
+	[PYTHON_GIOVERRIDESDIR=`$PYTHON -c 'import gi; import os.path; print(os.path.dirname(gi.__file__)+"/overrides")'`]
     fi
     AC_MSG_RESULT($have_python_gi)
 fi
