From 798755af32c42e395d167d9d49d14cbe449d2842 Mon Sep 17 00:00:00 2001
From: Morten Welinder <terra@gnome.org>
Date: Tue, 31 Jan 2017 18:26:55 -0500
Subject: Compilation: destdir fix from Redhat.

---
 doc/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/Makefile.am b/doc/Makefile.am
index 7e3f9a4..52a011d 100644
--- doc/Makefile.am
+++ doc/Makefile.am
@@ -527,7 +527,7 @@ HELP_MEDIA= \
 
 install-exec-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		install -d $$helpdir; \
 		$(RM) $$helpdir/index.docbook; \
 		$(LN_S) gnumeric.xml $$helpdir/index.docbook; \
@@ -535,7 +535,7 @@ install-exec-hook:
 
 uninstall-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		$(RM) $$helpdir/index.docbook; \
 	done
 
-- 
cgit v0.12
