--- man/Makefile.am.orig	2013-04-04 00:10:24.000000000 -0500
+++ man/Makefile.am	2013-11-04 20:22:14.000000000 -0600
@@ -53,7 +53,7 @@
 
 REL: $(top_srcdir)/configure
 	@{ echo '.ds Rv $(PACKAGE_VERSION)' ;		\
-	   date -u +'.ds Dt %Y-%m-%d' -r $< ; } > $@t ;	\
+	   gdate -u +'.ds Dt %Y-%m-%d' -r $< ; } > $@t ;\
 	if [ -f $@ ] && cmp -s $@t $@ ;			\
 	  then rm -f $@t ;				\
 	  else mv $@t $@ ; echo Created $@ ;		\
