https://savannah.gnu.org/bugs/?66220

diff --git a/Makefile.in b/Makefile.in
index 5f5b496..5865660 100644
--- Makefile.in.orig
+++ Makefile.in
@@ -233,3 +233,7 @@ dist : force
 
 force :
 
+# This Makefile contains rules which don't work with parallel make.
+# See <https://savannah.gnu.org/bugs/?66220>.
+# So, turn off parallel execution in this Makefile.
+.NOTPARALLEL:
diff --git a/callback/Makefile.in b/callback/Makefile.in
index d453d6b..64908e8 100644
--- callback/Makefile.in.orig
+++ callback/Makefile.in
@@ -234,3 +234,8 @@ distdir : $(DISTFILES)
 
 
 force :
+
+# This Makefile contains rules which don't work with parallel make.
+# See <https://savannah.gnu.org/bugs/?66220>.
+# So, turn off parallel execution in this Makefile.
+.NOTPARALLEL:
