--- autogen.sh.orig	2012-02-20 04:35:43.000000000 -0600
+++ autogen.sh	2013-02-07 18:25:43.000000000 -0600
@@ -202,29 +202,11 @@
   GNU_OR_DIE=0
 }
 
-WGET=`find_command 'wget*'`
-($WGET --version) < /dev/null > /dev/null 2>&1 ||
-{
-  echo
-  echo "You must have wget installed to compile $progname."
-  echo "Download the appropriate package for your distribution, or get the"
-  echo "source tarball from ftp://ftp.gnu.org/pub/gnu/wget"
-  GNU_OR_DIE=0
-}
 
 if test "$GNU_OR_DIE" -eq 0; then
   exit 1
 fi
 
-SVN=`find_command 'svn'`
-($SVN --version) < /dev/null > /dev/null 2>&1 ||
-{
-  echo
-  echo "You must have svn/subversion installed to compile $progname."
-  echo "Download the appropriate package for your distribution, or get the"
-  echo "source from http://subversion.tigris.org"
-  GNU_OR_DIE=0
-}
 
 if test "$GNU_OR_DIE" -eq 0; then
   exit 1
@@ -325,9 +307,9 @@
 
 if [ ! -f libtool.m4.in ]; then
   echo "    Finding libtool.m4.in"
-  if [ -f /usr/local/share/aclocal/libtool.m4 ]; then
-     echo "        .... found /usr/local/share/aclocal/libtool.m4"
-     cp /usr/local/share/aclocal/libtool.m4 libtool.m4.in
+  if [ -f __PREFIX__/share/aclocal/libtool.m4 ]; then
+     echo "        .... found __PREFIX__/share/aclocal/libtool.m4"
+     cp __PREFIX__/share/aclocal/libtool.m4 libtool.m4.in
   else
      if [ -f /usr/share/aclocal/libtool.m4 ]; then
       echo "        .... found /usr/share/aclocal/libtool.m4"
@@ -444,13 +426,14 @@
 else
     svn co $NDPI_URL
 fi
+echo
 
-dnl> nDPI compilation
+# nDPI compilation
 if test -f NDPI_LIB; then
     echo "nDPI already compiled"
 else
     echo "10. Compiling nDPI..."
-    cd nDPI; ./configure --with-pic; make; cd ..
+    cd nDPI; ./configure --with-pic || exit $?; make || exit $?; cd ..
 fi
 
 
