--- Makefile.am.orig	2011-03-03 07:44:59.000000000 -0600
+++ Makefile.am	2011-11-12 23:33:32.000000000 -0600
@@ -67,9 +67,7 @@
 NTOPDATA = ntop-cert.pem \
            $(ETTER_PASSIVE) \
            oui.txt.gz \
-           specialMAC.txt.gz \
-	   GeoIPASNum.dat \
-	   GeoLiteCity.dat
+           specialMAC.txt.gz
 
 NTOPHTML = html html/*.js html/*.html  html/*.gif html/*.jpg html/*.ico html/*.png \
            html/*.css html/*.dtd \
