--- configure.in.orig	2009-08-05 19:30:31.141160363 +0530
+++ configure.in	2009-08-05 19:34:11.076160092 +0530
@@ -1962,40 +1962,23 @@
     }'`
 fi
 
-dnl> GeoIP (http://www.maxmind.com/)
-if test -f "GeoIP.tar.gz"; then
-   echo "GeoIP already present on this machine"
-else
-   wget http://www.maxmind.com/download/geoip/api/c/GeoIP.tar.gz
-fi
-tar xvfz GeoIP.tar.gz
-GEO_DIR=`find $PWD -type d -name "GeoIP-*"`
-cd $GEO_DIR; ./configure --prefix=${prefix}; make; cd ..
-# OSX Fix
-GEO_DYLIB="$GEO_DIR/libGeoIP/.libs/libGeoIP.dylib"
-if test -f $GEO_DYLIB; then
-   ln -s $GEO_DYLIB .
-fi
-
-if test -f "GeoLiteCity.dat"; then
-   echo "GeoLiteCity.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz
-   gunzip GeoLiteCity.dat.gz
-fi
-
-if test -f "GeoIPASNum.dat"; then
-   echo "GeoIPASNum.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/asnum/GeoIPASNum.dat.gz
-   gunzip GeoIPASNum.dat.gz
+#
+#GeoIP
+#
+AC_CHECK_HEADERS(
+    [GeoIP.h],
+    [geoip_header="yes"],
+    [AC_MSG_FAILURE([GeoIP header file is missing])]
+)
+if test "$geoip_header" = "yes"; then
+    AC_CHECK_LIB(GeoIP, GeoIP_open,
+        [geoip_libs="yes"],
+	[AC_MSG_FAILURE([GeoIP library is missing])]
+    )
+fi
+if test "$geoip_libs" = "yes"; then
+    LDFLAGS="${LDFLAGS} -lGeoIP"
 fi
-
-
-GEO_DIR=`find $PWD -type d -name "GeoIP-*"`
-GEO_IP="$GEO_DIR/libGeoIP/"
-CFLAGS="$CFLAGS -I$GEO_IP"
-LDFLAGS="-L$GEO_IP.libs/ -lGeoIP $LDFLAGS"
 
 dnl> NTOPCONFIGDEBUG_SETTINGS([precet])
 
