diff -Nru ntop-3.3.9.orig/configure.in ntop-3.3.9/configure.in
--- ntop-3.3.9.orig/configure.in	2009-02-18 21:04:10.000000000 +0000
+++ ntop-3.3.9/configure.in	2009-02-18 21:05:58.000000000 +0000
@@ -1883,39 +1883,7 @@
 fi
 
 dnl> GeoIP (http://www.maxmind.com/)
-if test -f "GeoIP.tar.gz"; then
-   echo "GeoIP already present on this machine"
-else
-   wget http://www.maxmind.com/download/geoip/api/c/GeoIP.tar.gz
-   tar xvfz GeoIP.tar.gz
-   GEO_DIR=`find $PWD -type d -name "GeoIP-*"`
-   cd $GEO_DIR; ./configure; make; cd ..
-   # OSX Fix
-   GEO_DYLIB="$GEO_DIR/libGeoIP/.libs/libGeoIP.dylib"
-   if test -f $GEO_DYLIB; then
-      ln -s $GEO_DYLIB .
-   fi
-fi
-
-if test -f "GeoLiteCity.dat"; then
-   echo "GeoLiteCity.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz
-   gunzip GeoLiteCity.dat.gz
-fi
-
-if test -f "GeoIPASNum.dat"; then
-   echo "GeoIPASNum.dat already present"
-else
-   wget http://www.maxmind.com/download/geoip/database/asnum/GeoIPASNum.dat.gz
-   gunzip GeoIPASNum.dat.gz
-fi
-
-
-GEO_DIR=`find $PWD -type d -name "GeoIP-*"`
-GEO_IP="$GEO_DIR/libGeoIP/"
-CFLAGS="$CFLAGS -I$GEO_IP"
-LDFLAGS="$LDFLAGS -L$GEO_IP.libs/ -lGeoIP"
+LDFLAGS="$LDFLAGS -lGeoIP"
 
 dnl> NTOPCONFIGDEBUG_SETTINGS([precet])
 
@@ -2080,7 +2048,6 @@
 AC_SUBST(SO_VERSION_PATCH)
 AC_SUBST(RRD_LIB)
 AC_SUBST(RRD_INC)
-AC_SUBST(GEO_DIR)
 
 AC_CONFIG_FILES([Makefile])
 
diff -Nru ntop-3.3.9.orig/initialize.c ntop-3.3.9/initialize.c
--- ntop-3.3.9.orig/initialize.c	2008-12-03 22:00:22.000000000 +0000
+++ ntop-3.3.9/initialize.c	2009-02-18 21:04:44.000000000 +0000
@@ -467,16 +467,15 @@
 #endif
 
   /* Initialize GeoIP databases */
-  for(i=0; myGlobals.configFileDirs[i] != NULL; i++) {
+  {
     char path[256];
     
     safe_snprintf(__FILE__, __LINE__, path, sizeof(path),
 		  "%s%c%s",
-		  myGlobals.configFileDirs[i], CONST_PATH_SEP, GEO_IP_FILE);
+		  myGlobals.dbPath, CONST_PATH_SEP, GEO_IP_FILE);
     revertSlashIfWIN32(path, 0);
     if((myGlobals.geo_ip_db = GeoIP_open(path, GEOIP_CHECK_CACHE)) != NULL) {
       traceEvent(CONST_TRACE_INFO, "GeoIP: loaded config file %s", path);
-      break;
     }
   }
   
@@ -485,16 +484,15 @@
   
   /* *************************** */
 
-  for(i=0; myGlobals.configFileDirs[i] != NULL; i++) {
+  {
     char path[256];
     
     safe_snprintf(__FILE__, __LINE__, path, sizeof(path),
 		  "%s%c%s",
-		  myGlobals.configFileDirs[i], CONST_PATH_SEP, GEO_IP_ASN_FILE);
+		  myGlobals.dbPath, CONST_PATH_SEP, GEO_IP_ASN_FILE);
     revertSlashIfWIN32(path, 0);
     if((myGlobals.geo_ip_asn_db = GeoIP_open(path, GEOIP_CHECK_CACHE)) != NULL) {
       traceEvent(CONST_TRACE_INFO, "GeoIP: loaded ASN config file %s", path);
-      break;
     }
   }
   
diff -Nru ntop-3.3.9.orig/Makefile.am ntop-3.3.9/Makefile.am
--- ntop-3.3.9.orig/Makefile.am	2008-12-03 22:00:22.000000000 +0000
+++ ntop-3.3.9/Makefile.am	2009-02-18 21:05:33.000000000 +0000
@@ -76,9 +76,7 @@
 NTOPDATA = ntop-cert.pem \
            $(ETTER_PASSIVE) \
            oui.txt.gz \
-           specialMAC.txt.gz \
-	   GeoIPASNum.dat \
-	   GeoLiteCity.dat
+           specialMAC.txt.gz
 
 NTOPHTML = html html/*.js html/*.html  html/*.gif html/*.jpg html/*.ico html/*.png \
            html/*.css html/*.dtd \
@@ -270,8 +268,6 @@
 
 install: install-recursive
 
-	cd @GEO_DIR@; make install
-
 	@mkdir -p $(DESTDIR)/$(CFG_DBFILE_DIR)
 
 	@echo ""
