--- configure.in.orig	2011-08-15 03:34:28.000000000 -0500
+++ configure.in	2011-11-12 23:59:24.000000000 -0600
@@ -469,10 +469,6 @@
     LOCALEDIR="/usr/lib/locale"
 fi
 
-dnl> Add /usr/local/ /opt/local
-CFLAGS="${CFLAGS} -I/usr/local/include -I/opt/local/include"
-CPPFLAGS="${CPPFLAGS} -I/usr/local/include -I/opt/local/include"
-LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/opt/local/lib"
 
 PWD=`pwd`
 
@@ -1450,25 +1446,8 @@
 dnl> PYTHON
 dnl>
   AC_CHECK_TOOL(PYTHON, python-config)
-  PYTHON_CONFIG=""
+  PYTHON_CONFIG="$ac_cv_prog_ac_ct_PYTHON"
 
-  if test "x$ac_cv_prog_ac_ct_PYTHON" != "xpython-config"; then   
-       if test -f "/etc/debian_version"; then
-       	   AC_MSG_RESULT(Please install python-dev and rerun configure)
-	   exit 1
-	else
-	AC_MSG_RESULT(>>>> Unable to locate python-config: using workaround <<<<)
-        dnl> Silly workaround for RedHat-like distro
-        AC_CHECK_TOOL(PYTHON, python)
-        if test "x$ac_cv_prog_ac_ct_PYTHON" = "xpython"; then
-          PYTHON_CONFIG="./configureextra/python-config"
-        else
-          PYTHON_CONFIG="python-config"
-        fi
-     fi
-  else
-     PYTHON_CONFIG="python-config"
-  fi
 
   if test "x$PYTHON_CONFIG" != "x"; then
     PYTHON_LIBS="`$PYTHON_CONFIG --libs`"
@@ -1480,7 +1459,7 @@
 
     dnl remove unecessary path
     dnl line below workaround for OSX 10.6 (Snow Leopard)/10.7 (Lion)
-    PYTHON_INCS=`$PYTHON_CONFIG --cflags | sed -e "s/-arch i386// | sed -e "s/-arch ppc// | sed -e "s/-arch x86_64//"`
+    PYTHON_INCS=`$PYTHON_CONFIG --cflags | sed -e "s/-arch i386//" | sed -e "s/-arch ppc//" | sed -e "s/-arch x86_64//"`
     INCS="${INCS} ${PYTHON_INCS}"
 
 OLD_CFLAGS=$CFLAGS
@@ -1956,16 +1935,8 @@
 echo "#endif /* MEMORY_DEBUG */" >> version.c
 
 case "${host}" in
-    *darwin* )
-        echo
-	echo "Copying patched versions of some configure-related files for Mac OS X"
-        cp packages/MacOSX/lt* .
-	rm -f libtool
-	ln -s  /usr/bin/glibtool libtool
-        echo
-        ;;
 	*)
-	LIBTOOL_PATH="/usr/bin/libtool"
+	LIBTOOL_PATH="${prefix}/bin/glibtool"
 	if test -x ${LIBTOOL_PATH}; then
 	   rm -f libtool
 	   ln -s ${LIBTOOL_PATH} libtool
