--- plugins/Makefile.am.orig	2010-01-01 10:53:43.000000000 -0600
+++ plugins/Makefile.am	2011-11-13 00:47:52.000000000 -0600
@@ -57,24 +57,24 @@
                   libsflowPlugin.la
 
 libicmpPlugin_la_SOURCES = icmpPlugin.c
-libicmpPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libicmpPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libicmpPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 libnetflowPlugin_la_SOURCES = netflowPlugin.c
-libnetflowPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libnetflowPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libnetflowPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 libcpacketPlugin_la_SOURCES = cpacketPlugin.c
-libcpacketPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libcpacketPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libcpacketPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 librrdPlugin_la_SOURCES = rrdPlugin.c
-librrdPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+librrdPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 librrdPlugin_la_LIBADD = ${RRD_LIB}
 librrdPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 libsflowPlugin_la_SOURCES = sflowPlugin.c
-libsflowPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libsflowPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libsflowPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 #
