--- Makefile.am.orig	2009-04-17 17:27:57.000000000 -0300
+++ Makefile.am	2009-09-15 13:07:19.000000000 -0300
@@ -76,9 +76,7 @@
 NTOPDATA = ntop-cert.pem \
            $(ETTER_PASSIVE) \
            oui.txt.gz \
-           specialMAC.txt.gz \
-	   GeoIPASNum.dat \
-	   GeoLiteCity.dat
+           specialMAC.txt.gz
 
 NTOPHTML = html html/*.js html/*.html  html/*.gif html/*.jpg html/*.ico html/*.png \
            html/*.css html/*.dtd \
@@ -270,8 +268,6 @@
 
 install: install-recursive
 
-	cd @GEO_DIR@; make install
-
 	@mkdir -p $(DESTDIR)/$(CFG_DBFILE_DIR)
 
 	@echo ""
@@ -341,7 +337,6 @@
 	chcon -t textrel_shlib_t $(DESTDIR)$(libdir)/*ntop*.so
 
 install-data-local: ntop.txt ntop.html faq.html
-	@$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(libdir)/plugins;
 	@if ! test -d $(DESTDIR)/$(datadir)/ntop; then \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/ntop; \
 	  touch $(DESTDIR)/$(datadir)/ntop/warnuser; \
