--- plugins/Makefile.am.orig	2012-02-02 05:17:12.000000000 -0600
+++ plugins/Makefile.am	2013-02-07 17:55:09.000000000 -0600
@@ -52,16 +52,16 @@
                   libsflowPlugin.la 
 
 libnetflowPlugin_la_SOURCES = netflowPlugin.c
-libnetflowPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libnetflowPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libnetflowPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 librrdPlugin_la_SOURCES = rrdPlugin.c
-librrdPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+librrdPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 librrdPlugin_la_LIBADD = ${RRD_LIB}
 librrdPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 libsflowPlugin_la_SOURCES = sflowPlugin.c
-libsflowPlugin_la_LDFLAGS = -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
+libsflowPlugin_la_LDFLAGS = $(AM_LDFLAGS) -shared -release @PACKAGE_VERSION@ @DYN_FLAGS@
 libsflowPlugin_la_CFLAGS = $(AM_CFLAGS)
 
 #
