Index: XBMC/system/scrapers/video/ofdb.xml
===================================================================
--- system/scrapers/video/ofdb.xml	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ system/scrapers/video/ofdb.xml	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -122,23 +122,23 @@
 	</RegExp>
 
 	<!--URL to IMDB-->
-	<RegExp input="$$1" output="&lt;url function=&quot;IMDB-Details&quot;&gt;http://german.imdb.com/title/tt\1/&lt;/url&gt;" dest="5+">
-		<expression>http://german.imdb.com/Title\?([0-9]+)</expression>
+	<RegExp input="$$1" output="&lt;url function=&quot;IMDB-Details&quot;&gt;http://www.imdb.com/title/tt\2/&lt;/url&gt;" dest="5+">
+		<expression>(imdb.com/)Title\?([0-9]+)</expression>
 	</RegExp>
 
 	<!--URL to IMDBcredits-->
-	<RegExp input="$$1" output="&lt;url function=&quot;GetMovieCast&quot;&gt;http://german.imdb.com/title/tt\1/fullcredits#cast&lt;/url&gt;" dest="5+">
-		<expression>http://german.imdb.com/Title\?([0-9]+)</expression>
+	<RegExp input="$$1" output="&lt;url function=&quot;GetMovieCast&quot;&gt;http://www.imdb.com/title/tt\2/fullcredits#cast&lt;/url&gt;" dest="5+">
+		<expression>(imdb.com/)Title\?([0-9]+)</expression>
 	</RegExp>
 
 	<!--URL to Fanart-->
-	<RegExp conditional="fanart" input="$$1" output="&lt;url function=&quot;GetFanart&quot;&gt;http://api.themoviedb.org/backdrop.php?imdb=tt\1&lt;/url&gt;" dest="5+">
-		<expression>http://german.imdb.com/Title\?([0-9]+)</expression>
+	<RegExp conditional="fanart" input="$$1" output="&lt;url function=&quot;GetFanart&quot;&gt;http://api.themoviedb.org/backdrop.php?imdb=tt\2&lt;/url&gt;" dest="5+">
+		<expression>(imdb.com/)Title\?([0-9]+)</expression>
 	</RegExp>
 
 	<!--URL to MoviePosterDB -->
-	<RegExp conditional="movieposterdb" input="$$1" output="&lt;url function=&quot;GetMoviePosterDBLink&quot;&gt;http://www.movieposterdb.com/browse/search?title=\1&lt;/url&gt;" dest="5+">
-  		<expression>http://german.imdb.com/Title\?([0-9]+)</expression> 
+	<RegExp conditional="movieposterdb" input="$$1" output="&lt;url function=&quot;GetMoviePosterDBLink&quot;&gt;http://www.movieposterdb.com/browse/search?title=\2&lt;/url&gt;" dest="5+">
+  		<expression>(imdb.com/)Title\?([0-9]+)</expression> 
 	</RegExp>
 	
 	<!-- Ofdb Poster -->
Index: XBMC/XBMC.xcodeproj/project.pbxproj
===================================================================
--- XBMC.xcodeproj/project.pbxproj	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ XBMC.xcodeproj/project.pbxproj	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -5838,7 +5838,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "# Update version in Info.plist with SVN revision\nSVN_REVISION=\"8.10-b1\"\nSVNVERSION=\"\"\nBUNDLE_NAME=\"XBMC\"\n\n# only check for macports subversion or collab.net's svn\n# since OS X's bundled subversion client is too old\n# http://www.collab.net/downloads/community/\nif [ -x /opt/local/bin/svnversion ]; then\n\tSVVERSION=\"/opt/local/bin/svnversion\"\nelif [ -x /opt/subversion/bin/svnversion ]; then\n\tSVNVERSION=\"/opt/subversion/bin/svnversion\"\nfi\n\n# we check /usr/local/bin since svn bundled with OS X is too old\nif [ \"x$SVNVERSION\" != \"x\" ]; then\n\tSVN_REVISION=\"SVN-\"$($SVNVERSION \"$TARGET_BUILD_DIR/../../\")\nfi\nperl -p -i -e \"s/r####/$SVN_REVISION/\" \"$TARGET_BUILD_DIR/$BUNDLE_NAME.app/Contents/Info.plist\"\n";
+			shellScript = "# Update version in Info.plist with SVN revision\nSVN_REVISION=\"8.10-bf1\"\nSVNVERSION=\"\"\nBUNDLE_NAME=\"XBMC\"\n\n# only check for macports subversion or collab.net's svn\n# since OS X's bundled subversion client is too old\n# http://www.collab.net/downloads/community/\nif [ -x /opt/local/bin/svnversion ]; then\n\tSVVERSION=\"/opt/local/bin/svnversion\"\nelif [ -x /opt/subversion/bin/svnversion ]; then\n\tSVNVERSION=\"/opt/subversion/bin/svnversion\"\nfi\n\n# we check /usr/local/bin since svn bundled with OS X is too old\nif [ \"x$SVNVERSION\" != \"x\" ]; then\n\tSVN_REVISION=\"SVN-\"$($SVNVERSION \"$TARGET_BUILD_DIR/../../\")\nfi\nperl -p -i -e \"s/r####/$SVN_REVISION/\" \"$TARGET_BUILD_DIR/$BUNDLE_NAME.app/Contents/Info.plist\"\n";
 		};
 		882630930E1B6FAD00DC9524 /* ShellScript */ = {
 			isa = PBXShellScriptBuildPhase;
@@ -6831,7 +6831,9 @@
 					"_FILE_OFFSET_BITS=64",
 					_LARGEFILE64_SOURCE,
 					HAS_SDL_JOYSTICK,
+					"$(GCC_PREPROCESSOR_DEFINITIONS_QUOTED_FOR_TARGET_1)",
 				);
+				GCC_PREPROCESSOR_DEFINITIONS_QUOTED_FOR_TARGET_1 = "VERSION_STRING='\"8.10 bf1:ATLANTIS\"'";
 				GCC_SYMBOLS_PRIVATE_EXTERN = NO;
 				GCC_UNROLL_LOOPS = YES;
 				HEADER_SEARCH_PATHS = (
Index: XBMC/xbmc/Util.cpp
===================================================================
--- xbmc/Util.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/Util.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -837,6 +837,7 @@
   if (strProtocol == "cdda" || strProtocol == "iso9660") return false;
   if (strProtocol == "special") return IsRemote(TranslateSpecialPath(strFile));
   if (strProtocol.Left(3) == "mem") return false;   // memory cards
+  if (strProtocol == "stack") return IsRemote(CStackDirectory::GetFirstStackedFile(strFile));
   if (strProtocol == "virtualpath")
   { // virtual paths need to be checked separately
     CVirtualPathDirectory dir;
Index: XBMC/xbmc/ApplicationMessenger.h
===================================================================
--- xbmc/ApplicationMessenger.h	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/ApplicationMessenger.h	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -126,8 +126,8 @@
   void ProcessMessage(ThreadMessage *pMsg);
 
 
-  std::vector<ThreadMessage*> m_vecMessages;
-  std::vector<ThreadMessage*> m_vecWindowMessages;
+  std::queue<ThreadMessage*> m_vecMessages;
+  std::queue<ThreadMessage*> m_vecWindowMessages;
   CCriticalSection m_critSection;
   CCriticalSection m_critBuffer;
   CStdString bufferResponse;
Index: XBMC/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
===================================================================
--- xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -436,6 +436,11 @@
   {
     if (m_ioContext)
     {
+      if(m_pFormatContext->pb && m_pFormatContext->pb != m_ioContext)
+      {
+        CLog::Log(LOGWARNING, "CDVDDemuxFFmpeg::Dispose - demuxer changed our byte context behind our back, possible memleak");
+        m_ioContext = m_pFormatContext->pb;
+      }
       m_dllAvFormat.av_close_input_stream(m_pFormatContext);
       m_dllAvFormat.url_fclose(m_ioContext);
     }
Index: XBMC/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp
===================================================================
--- xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoPPFFmpeg.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -147,6 +147,7 @@
   m_pTarget->iDisplayHeight = m_pSource->iDisplayHeight;
   m_pTarget->iDisplayWidth = m_pSource->iDisplayWidth;
   m_pTarget->pts = m_pSource->pts;
+  m_pTarget->iGroupId = m_pSource->iGroupId;
 
 
 }
Index: XBMC/xbmc/win32/XBMC_PC.rc
===================================================================
--- xbmc/win32/XBMC_PC.rc	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/win32/XBMC_PC.rc	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -53,8 +53,8 @@
 //
 
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION 2,1,0,0
- PRODUCTVERSION 2,1,0,0
+ FILEVERSION 0,8,1,0
+ PRODUCTVERSION 0,8,1,0
  FILEFLAGSMASK 0x3fL
 #ifdef _DEBUG
  FILEFLAGS 0x1L
@@ -71,12 +71,12 @@
         BEGIN
             VALUE "CompanyName", "Team XBMC"
             VALUE "FileDescription", "XBMC "
-            VALUE "FileVersion", "pre-2.1"
+            VALUE "FileVersion", "8.10.bf1"
             VALUE "InternalName", "XBMC.exe"
             VALUE "LegalCopyright", "CopyRight (c) Team XBMC.  All rights reserved."
             VALUE "OriginalFilename", "XBMC.exe"
             VALUE "ProductName", "XBMC for Windows"
-            VALUE "ProductVersion", "pre-2.1"
+            VALUE "ProductVersion", "8.10.bf1"
         END
     END
     BLOCK "VarFileInfo"
Index: XBMC/xbmc/GUISettings.cpp
===================================================================
--- xbmc/GUISettings.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/GUISettings.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -599,13 +599,7 @@
 
   AddString(3, "videoscreen.guicalibration",214,"", BUTTON_CONTROL_STANDARD);
   AddString(4, "videoscreen.testpattern",226,"", BUTTON_CONTROL_STANDARD);
-  AddInt(6, "videoscreen.vsync", 13105,
-#ifdef _LINUX
-         VSYNC_DISABLED,
-#else /* _LINUX */
-         VSYNC_ALWAYS,
-#endif /* _LINUX */
-         VSYNC_DISABLED, 1, VSYNC_DRIVER, SPIN_CONTROL_TEXT);
+  AddInt(6, "videoscreen.vsync", 13105, VSYNC_ALWAYS, VSYNC_DISABLED, 1, VSYNC_DRIVER, SPIN_CONTROL_TEXT);
 
   AddCategory(7, "filelists", 14018);
   AddBool(1, "filelists.hideparentdiritems", 13306, false);
Index: XBMC/xbmc/ApplicationMessenger.cpp
===================================================================
--- xbmc/ApplicationMessenger.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/ApplicationMessenger.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -53,20 +53,18 @@
 
 void CApplicationMessenger::Cleanup()
 {
-  vector<ThreadMessage*>::iterator it = m_vecMessages.begin();
-  while (it != m_vecMessages.end())
+  while (m_vecMessages.size() > 0)
   {
-    ThreadMessage* pMsg = *it;
+    ThreadMessage* pMsg = m_vecMessages.front();
     delete pMsg;
-    it = m_vecMessages.erase(it);
+    m_vecMessages.pop();
   }
 
-  it = m_vecWindowMessages.begin();
-  while (it != m_vecWindowMessages.end())
+  while (m_vecWindowMessages.size() > 0)
   {
-    ThreadMessage* pMsg = *it;
+    ThreadMessage* pMsg = m_vecWindowMessages.front();
     delete pMsg;
-    it = m_vecWindowMessages.erase(it);
+    m_vecWindowMessages.pop();
   }
 }
 
@@ -98,16 +96,14 @@
   CSingleLock lock (m_critSection);
   if (msg->dwMessage == TMSG_DIALOG_DOMODAL ||
       msg->dwMessage == TMSG_WRITE_SCRIPT_OUTPUT)
-  {
-    m_vecWindowMessages.push_back(msg);
-  }
-  else m_vecMessages.push_back(msg);
+    m_vecWindowMessages.push(msg);
+  else 
+    m_vecMessages.push(msg);
+  lock.Leave();
 
   if (message.hWaitEvent)
   {
-    lock.Leave();
     WaitForSingleObject(message.hWaitEvent, INFINITE);
-    lock.Enter();
     CloseHandle(message.hWaitEvent);
     message.hWaitEvent = NULL;
   }
@@ -119,24 +115,20 @@
   CSingleLock lock (m_critSection);
   while (m_vecMessages.size() > 0)
   {
-    vector<ThreadMessage*>::iterator it = m_vecMessages.begin();
-
-    ThreadMessage* pMsg = *it;
+    ThreadMessage* pMsg = m_vecMessages.front();
     //first remove the message from the queue, else the message could be processed more then once
-    it = m_vecMessages.erase(it);
+    m_vecMessages.pop();
 
     //Leave here as the message might make another
     //thread call processmessages or sendmessage
     lock.Leave();
-    ProcessMessage(pMsg);
 
-    // lock again to make sure nothing happened to pMsg->hWaitEvent
-    lock.Enter();
-
+    ProcessMessage(pMsg);
     if (pMsg->hWaitEvent)
       SetEvent(pMsg->hWaitEvent);
-
     delete pMsg;
+
+    lock.Enter();
   }
 }
 
@@ -540,25 +532,21 @@
 {
   CSingleLock lock (m_critSection);
   //message type is window, process window messages
-  if (m_vecWindowMessages.size() > 0)
+  while (m_vecWindowMessages.size() > 0)
   {
-    vector<ThreadMessage*>::iterator it = m_vecWindowMessages.begin();
-    while (it != m_vecWindowMessages.end())
-    {
-      ThreadMessage* pMsg = *it;
-      //first remove the message from the queue, else the message could be processed more then once
-      it = m_vecWindowMessages.erase(it);
+    ThreadMessage* pMsg = m_vecWindowMessages.front();
+    //first remove the message from the queue, else the message could be processed more then once
+    m_vecWindowMessages.pop();
 
-      // leave here in case we make more thread messages from this one
-      lock.Leave();
-      ProcessMessage(pMsg);
-      if (pMsg->hWaitEvent)
-        SetEvent(pMsg->hWaitEvent);
+    // leave here in case we make more thread messages from this one
+    lock.Leave();
 
-      delete pMsg;
-      // reenter
-      lock.Enter();
-    }
+    ProcessMessage(pMsg);
+    if (pMsg->hWaitEvent)
+      SetEvent(pMsg->hWaitEvent);
+    delete pMsg;
+
+    lock.Enter();
   }
 }
 
Index: XBMC/xbmc/XBVideoConfig.cpp
===================================================================
--- xbmc/XBVideoConfig.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/XBVideoConfig.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -327,7 +327,7 @@
      m_ResInfo[m_iNumResolutions].iSubtitles = (int)(0.9*h);
      snprintf(m_ResInfo[m_iNumResolutions].strMode,
                sizeof(m_ResInfo[m_iNumResolutions].strMode),
-               "%d x %d (Full screen #%d)", w, h, i+1);
+               "%d x %d (Full Screen #%d)", w, h, i+1);
      if ((float)w / (float)h >= 1.59)
        m_ResInfo[m_iNumResolutions].dwFlags = D3DPRESENTFLAG_WIDESCREEN;
      else
Index: XBMC/xbmc/GUIWindowSettingsCategory.cpp
===================================================================
--- xbmc/GUIWindowSettingsCategory.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/GUIWindowSettingsCategory.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -116,6 +116,7 @@
   m_pOriginalButton = NULL;
   m_pOriginalCategoryButton = NULL;
   m_pOriginalImage = NULL;
+  m_pOriginalEdit = NULL;
   // set the correct ID range...
   m_dwIDRange = 8;
   m_iScreen = 0;
@@ -893,7 +894,7 @@
       if (pControl)
       {
         int value = g_guiSettings.GetInt("videoscreen.resolution");
-        if (strstr(g_settings.m_ResInfo[value].strMode, "Full screen") != 0)
+        if (strstr(g_settings.m_ResInfo[value].strMode, "Full Screen") != 0)
           pControl->SetEnabled(true);
         else
           pControl->SetEnabled(false);
Index: XBMC/xbmc/FileSystem/cddb.cpp
===================================================================
--- xbmc/FileSystem/cddb.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/FileSystem/cddb.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -1206,9 +1206,9 @@
 
   //##########################################################
   // Send the Hello message
-  if ( ! Send("cddb hello xbox xbox XboxMediaCenter pre-2.1"))
+  if ( ! Send("cddb hello xbmc xbmc XBMC 8.10.bf1"))
   {
-    CLog::Log(LOGERROR, "Xcddb::queryCDinfo Error sending \"%s\"", "cddb hello xbox xbox XboxMediaCenter pre-2.1");
+    CLog::Log(LOGERROR, "Xcddb::queryCDinfo Error sending \"%s\"", "cddb hello xbmc xbmc XBMC 8.10");
     m_lastError = E_NETWORK_ERROR_SEND;
     return false;
   }
Index: XBMC/xbmc/FileSystem/FileCurl.cpp
===================================================================
--- xbmc/FileSystem/FileCurl.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/FileSystem/FileCurl.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -389,7 +389,7 @@
   if (m_userAgent.length() > 0)
     g_curlInterface.easy_setopt(h, CURLOPT_USERAGENT, m_userAgent.c_str());
   else /* set some default agent as shoutcast doesn't return proper stuff otherwise */
-    g_curlInterface.easy_setopt(h, CURLOPT_USERAGENT, "XBMC/pre-2.1 (compatible; MSIE 6.0; Windows NT 5.1; WinampMPEG/5.09)");
+    g_curlInterface.easy_setopt(h, CURLOPT_USERAGENT, "XBMC/8.10.bf1 (compatible; MSIE 6.0; Windows NT 5.1; WinampMPEG/5.09)");
   
   if (m_useOldHttpVersion)
     g_curlInterface.easy_setopt(h, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_0);
Index: XBMC/xbmc/FileSystem/FileSmb.cpp
===================================================================
--- xbmc/FileSystem/FileSmb.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/FileSystem/FileSmb.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -512,7 +512,6 @@
   struct stat info;
 #endif
 
-  CLog::Log(LOGDEBUG, "%s(%s)", __FUNCTION__, strFileName.c_str());
   CSingleLock lock(smb);
   int iResult = smbc_stat(strFileName, &info);
 
Index: XBMC/xbmc/GUILargeTextureManager.h
===================================================================
--- xbmc/GUILargeTextureManager.h	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/GUILargeTextureManager.h	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -104,7 +104,10 @@
     {
       assert(m_texture == NULL);
 #ifdef HAS_SDL_OPENGL
-      m_texture = new CGLTexture(texture, false, true);
+      if (texture)
+        m_texture = new CGLTexture(texture, false, true);
+      else
+        m_texture = NULL; // unable to load the textures
 #else
       m_texture = texture;
 #endif
Index: XBMC/xbmc/NfoFile.cpp
===================================================================
--- xbmc/NfoFile.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/NfoFile.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -166,7 +166,15 @@
   if (file.Open(strFile, true))
   {
     m_size = (int)file.GetLength();
-    m_doc = new char[m_size+1];
+    try
+    {
+      m_doc = new char[m_size+1];
+    }
+    catch (...)
+    {
+      CLog::Log(LOGERROR, "%s: Exception while creating file buffer",__FUNCTION__);
+      return E_FAIL;
+    }
     if (!m_doc)
     {
       file.Close();
Index: XBMC/xbmc/lib/libRTMP/rtmp.cpp
===================================================================
--- xbmc/lib/libRTMP/rtmp.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/lib/libRTMP/rtmp.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -621,7 +621,8 @@
 
     CLog::Log(LOGDEBUG,"%s, onStatus: %s", __FUNCTION__, code.c_str() );
     if (code == "NetStream.Failed"
-    ||  code == "NetStream.Play.Failed")
+    ||  code == "NetStream.Play.Failed"
+    ||  code == "NetStream.Play.Stop")
       Close();
   }
   else
Index: XBMC/xbmc/osx/CocoaUtils.m
===================================================================
--- xbmc/osx/CocoaUtils.m	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/osx/CocoaUtils.m	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -116,6 +116,45 @@
   CFNumberGetValue(CFDictionaryGetValue(mode, kCGDisplayHeight), kCFNumberSInt32Type, h);
 }
 
+int Cocoa_GetScreenIndex(void)
+{
+  // return one based screen index
+  int screen_index = 0;
+  int numDisplays = [[NSScreen screens] count];
+  NSScreen* current_Screen = nil;
+  
+  if (numDisplays > 1)
+  {
+    NSOpenGLContext* context = (NSOpenGLContext*)Cocoa_GL_GetCurrentContext();
+    if (context)
+    {
+      NSView* view;
+    
+      view = [context view];
+      if (view) {
+        NSWindow* window;
+        window = [view window];
+        if (window)
+        {
+          // Get the screen we are using for display.
+          current_Screen = [window screen];
+        }
+      }
+    }
+    
+    for (screen_index = 0; screen_index < numDisplays; screen_index++)
+    {
+      if (current_Screen == [[NSScreen screens] objectAtIndex:screen_index])
+      {
+        break;
+      }
+    }
+  }
+  screen_index++;
+  
+  return(screen_index);
+}
+
 void Cocoa_GetScreenResolution(int* w, int* h)
 {
   // Figure out the screen size. (default to main screen)
Index: XBMC/xbmc/osx/Info.plist
===================================================================
--- xbmc/osx/Info.plist	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/osx/Info.plist	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -1,28 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-	<key>CFBundleInfoDictionaryVersion</key>
-	<string>6.0</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>English</string>
 	<key>CFBundleExecutable</key>
 	<string>XBMC</string>
 	<key>CFBundleGetInfoString</key>
-	<string>8.10.b1</string>
+	<string>8.10.bf1</string>
 	<key>CFBundleIconFile</key>
 	<string>xbmc.icns</string>
 	<key>CFBundleIdentifier</key>
 	<string>com.teamxbmc.xbmc</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
 	<key>CFBundleName</key>
 	<string>XBMC</string>
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>8.10-b1</string>
+	<string>8.10-bf1</string>
+	<key>CFBundleSignature</key>
+	<string>xbmc</string>
 	<key>CFBundleVersion</key>
 	<string>r####</string>
-	<key>CFBundleSignature</key>
-	<string>xbmc</string>
 </dict>
 </plist>
Index: XBMC/xbmc/osx/CocoaUtils.h
===================================================================
--- xbmc/osx/CocoaUtils.h	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/osx/CocoaUtils.h	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -29,6 +29,7 @@
   //
   // Graphics.
   //
+  int Cocoa_GetScreenIndex(void);
   void Cocoa_GetScreenResolution(int* w, int* h);
   double Cocoa_GetScreenRefreshRate(int screen_id);
   void Cocoa_GetScreenResolutionOfAnotherScreen(int display, int* w, int* h);
Index: XBMC/xbmc/GUIWindowMusicInfo.cpp
===================================================================
--- xbmc/GUIWindowMusicInfo.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/GUIWindowMusicInfo.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -459,7 +459,7 @@
     item->SetIconImage("defaultPicture.png");
     item->GetVideoInfoTag()->m_strPictureURL.m_url.push_back(url.m_url[i]);
     item->SetLabel(g_localizeStrings.Get(415));
-    item->SetProperty("labelonthumbload", g_localizeStrings.Get(20055));
+    item->SetProperty("labelonthumbload", g_localizeStrings.Get(20015));
     // make sure any previously cached thumb is removed
     if (CFile::Exists(item->GetCachedPictureThumb()))
       CFile::Delete(item->GetCachedPictureThumb());
Index: XBMC/xbmc/utils/GUIInfoManager.h
===================================================================
--- xbmc/utils/GUIInfoManager.h	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ xbmc/utils/GUIInfoManager.h	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -393,10 +393,14 @@
 #define BUTTON_SCROLLER_HAS_ICON    30001
 
 #ifdef SVN_REV
-#define VERSION_STRING "pre-8.10 SVN:"SVN_REV
+#ifndef VERSION_STRING
+#define VERSION_STRING "8.10 SVN:"SVN_REV
+#endif
 #else
-#define VERSION_STRING "pre-8.10"
+#ifndef VERSION_STRING
+#define VERSION_STRING "8.10 bf1:ATLANTIS"
 #endif
+#endif
 
 #define LISTITEM_START              35000
 #define LISTITEM_THUMB              (LISTITEM_START)
Index: XBMC/known_issues.txt
===================================================================
--- known_issues.txt	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ known_issues.txt	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -1,30 +1,48 @@
 Known Issues for 8.10 'Atlantis' Release of XBMC for Linux, Mac OS X and Windows
--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
  
-*) There is no offloading of Video Decoding (commonly referred to as Hardware Acceleration) in XBMC at this time. This means that it takes powerful hardware to decode High Definition H.264 videos.
+*) There is no offloading of Video Decoding (commonly referred to as
+ Hardware Acceleration) in XBMC at this time. This means that it takes
+ powerful hardware to decode High Definition H.264 videos.
  
-*) PulseAudio (Ubuntu component) causes various audio related problems for XBMC, so you might want to remove this component to be on the safe side.
+*) PulseAudio (Ubuntu component) causes various audio related problems
+ for XBMC, so you might want to disable this component to be on the
+ safe side.
  
-*) 'Settings/Videos/Player/High Quality Software Upscaling' is not recommended to use with older IGP solutions/slow hardware, safest is to not enable it at all unless you are fully aware of what you're doing.
+*) 'Settings -> Videos -> Player -> High Quality Software Upscaling'
+ is not recommended for use with older IGP solutions/slow hardware.
+ Safest is to not enable it at all unless you are fully aware of what
+ you're doing.
  
-*) Intel GMA X3100 (GM965) and slower/older IGP's barely handle 1080p Display Output under Linux due to (what I pike suspect) poor  drivers.
+*) Intel GMA X3100 (GM965) and older IGPs have issues handling 1080p
+ Display Output under Linux.
  
-*) Some H.264 encoded videos experiences blocking. It's currently believed this is related to the unapproved FFMpeg 'Cabac patch' we use to speedup (multithread) H.264 decoding.
+*) Some H.264 encoded videos exhibit video "blocking" artifacts. It's
+ currently believed that this is related to the unapproved FFMpeg
+ 'Cabac patch' we use to speedup (multithread) H.264 decoding.
  
-*) A running Screensaver behind Fullscreen XBMC on Linux can cause graphical glitches in XBMC.
+*) An active screensaver in Linux, behind XBMC running fullscreen, can
+ cause graphical glitches.
  
-*) Compiz / Desktop Effects and XBMC does not mix well, atleast not without serious tweaking to Compiz.
+*) Compiz / Desktop Effects and XBMC do not mix well, atleast not
+ without serious tweaking to Compiz.
  
-*) Eventclients are not to be considered stable in this release.
+*) Event clients are not to be considered stable in this release.
  
-*) Shoutcast/AAC streaming do not work correctly, it may re-buffer every ~30 secs.
+*) Shoutcast/AAC streaming does not work correctly, it may re-buffer
+ every 30 or so seconds.
  
-*) Not all of XBMC's settings are active or do what they are supposed to.
+*) Not all of XBMC's settings are active or do what they are supposed
+ to.
  
-*) "Rar sets" needs to be completely downloaded before XBMC can stream content from them.
+*) "Rar sets" needs to be completely downloaded before XBMC can stream
+ content from them.
  
-*) Not all the old xbox XBMC scripts work with Linux/OSX/Win ports. Some don't even work on XBOX anymore.
+*) Not all the old XBox XBMC scripts work with Linux/OS X/Win32
+ ports. Some don't even work on XBox anymore.
  
-*) XBMC crashing on Linux will sometimes lock the mousepointer (you need to restart XBMC and quit it again to release it).
+*) XBMC crashing on Linux will sometimes lock the mouse pointer. To fix
+ this, restart XBMC and quit it again to release it.
  
-*) Visualisations are less polished than other components and may cause various glitches, like resizing issues.
+*) Visualisations are less polished than other components and may
+ cause various glitches, like resizing issues.
Index: XBMC/guilib/GraphicContext.cpp
===================================================================
--- guilib/GraphicContext.cpp	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ guilib/GraphicContext.cpp	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -615,6 +615,41 @@
 
   if (res==WINDOW || (m_Resolution != res))
   {
+#if defined(__APPLE__)
+    // In going FullScreen, m_Resolution == DESKTOP but if using multiple displays
+    // the display resolution will be wrong if the windowed display is moved to
+    // a display with a different resolution. So we have to resort to    
+    // Hack, hack, hack. The basic problem is the resolution is not linked to the 
+    // display so we have to find which display we are going fs on, then search
+    // through the m_ResInfo resolutions to find a matching "Full Screen"
+    // descriptor, then use that index to setup m_Resolution as there are multiple
+    // "DESKTOP" with multiple displays. If the strMode descriptor changes, this
+    // will break but the resolution really need to be linked to a display index.
+    if (m_bFullScreenRoot)
+    {
+      // going to fullscreen desktop but which display if multiple displays?
+      // need to find the m_ResInfo index for that display.
+      int screen_index = Cocoa_GetScreenIndex();
+      char        test_string[256];
+
+      if (screen_index == 1)
+      {
+        strcpy(test_string, "(Full Screen)");
+      }
+      else
+      {
+        sprintf(test_string, "(Full Screen #%d)", screen_index);
+      }
+      for (int i = (int)DESKTOP ; i< (CUSTOM+g_videoConfig.GetNumberOfResolutions()) ; i++)
+      {
+        if (strstr(g_settings.m_ResInfo[i].strMode, test_string) != 0)
+        {
+            res = (RESOLUTION)i;
+            break;
+        }
+      }
+    }
+#endif
     Lock();
     m_iScreenWidth  = g_settings.m_ResInfo[res].iWidth;
     m_iScreenHeight = g_settings.m_ResInfo[res].iHeight;
@@ -926,6 +961,7 @@
                                        g_settings.m_ResInfo[res].iHeight,
                                        g_settings.m_ResInfo[res].fRefreshRate);
     g_settings.m_ResInfo[res].iSubtitles = (int)(0.965 * g_settings.m_ResInfo[res].iHeight);
+  
     snprintf(g_settings.m_ResInfo[res].strMode, sizeof(g_settings.m_ResInfo[res].strMode), 
              "%dx%d (Full Screen)", g_settings.m_ResInfo[res].iWidth, g_settings.m_ResInfo[res].iHeight);
     if ((float)g_settings.m_ResInfo[res].iWidth/(float)g_settings.m_ResInfo[res].iHeight >= fOptimalSwitchPoint)
@@ -1449,9 +1485,6 @@
 #endif
     
 #ifdef __APPLE__
-    ResetScreenParameters(m_Resolution);
-    m_iFullScreenWidth = m_iScreenWidth= g_settings.m_ResInfo[m_Resolution].iWidth;
-    m_iFullScreenHeight = m_iScreenHeight = g_settings.m_ResInfo[m_Resolution].iHeight;
     Cocoa_GL_SetFullScreen(m_iFullScreenWidth, m_iFullScreenHeight, true, blankOtherDisplays, g_advancedSettings.m_osx_GLFullScreen);
 #elif defined(_WIN32PC)
     DEVMODE settings;
Index: XBMC/skin/PM3.HD/language/English/strings.xml
===================================================================
--- skin/PM3.HD/language/English/strings.xml	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ skin/PM3.HD/language/English/strings.xml	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -42,7 +42,7 @@
   <string id="31041">Current TV Show[CR]Fanart Image[CR][CR]Click to Change</string>
   <string id="31042">PLAYING</string>
   <string id="31043">PAUSED</string>
-  <string id="31044">FAST FOWARD</string>
+  <string id="31044">FAST FORWARD</string>
   <string id="31045">REWIND</string>
   <string id="31046">Welcome to XBMC - Please Login</string>
   <string id="31047">Last Logged in</string>
Index: XBMC/skin/PM3.HD/language/German/strings.xml
===================================================================
--- skin/PM3.HD/language/German/strings.xml	(.../tags/8.10_Atlantis-linux-osx-win32)	(revision 17004)
+++ skin/PM3.HD/language/German/strings.xml	(.../branches/8.10_Atlantis-linux-osx-win32)	(revision 17004)
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="ISO-8859-1" standalone="yes"?>
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
 <!--Translator: T045T-->
 <!--Email: t045tbr0t@yahoo.de-->
 <!--Date of translation: 10/25/2008-->
