Index: xbmc/XboxMediaCenter.cpp
===================================================================
--- xbmc/XboxMediaCenter.cpp	(revision 16965)
+++ xbmc/XboxMediaCenter.cpp	(working copy)
@@ -85,7 +85,16 @@
   // if we're on a Mac or if XBMC_PLATFORM_MODE is set, enable platform
   // specific directories.
 #ifdef __APPLE__
-  system("XBMC.app/Contents/Resources/XBMC/tools/preflight");
+  if(getenv("XBMC_HOME")) {
+    char *preflight = (char *)malloc((strlen(getenv("XBMC_HOME")) + strlen("/tools/preflight") + 1) * sizeof(char));
+    if(!preflight)
+      _exit(1);
+    strcpy(preflight, getenv("XBMC_HOME"));
+    strcat(preflight, "/tools/preflight");
+    system(preflight);
+  } else {
+    system("XBMC.app/Contents/Resources/XBMC/tools/preflight");
+  }
   if (1)
 #else
   if (getenv("XBMC_PLATFORM_MODE"))
