Index: xbmc/linux/PlatformDefs.h
===================================================================
--- xbmc/linux/PlatformDefs.h	(revision 16965)
+++ xbmc/linux/PlatformDefs.h	(working copy)
@@ -314,7 +314,7 @@
 #define _O_WRONLY O_WRONLY
 #define _off_t off_t
 
-#ifdef __APPLE__
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED < 1050
 #define __stat64 stat
 #define stat64 stat
 #define statfs64 statfs
