diff -Naurp wol-0.7.1.orig/configure.ac wol-0.7.1/configure.ac
--- wol-0.7.1.orig/configure.ac	2011-06-15 18:31:34.000000000 -0700
+++ wol-0.7.1/configure.ac	2011-06-15 18:31:52.000000000 -0700
@@ -63,12 +63,12 @@ AC_CHECK_SIZEOF(unsigned char, 1)
 ether_includes=""
 AC_CHECK_TYPE([struct ether_addr], , , [#include <netinet/ether.h>]) dnl Linux
 if test "$ac_cv_type_struct_ether_addr" = "yes"; then
-	AC_DEFINE_UNQUOTED(HAVE_STRUCT_ETHER_ADDR, 1)
+	AC_DEFINE(HAVE_STRUCT_ETHER_ADDR, 1, [struct ether_addr])
 	ether_includes="#include <netinet/ether.h>"
 else dnl BSD
 	AC_CHECK_TYPE([struct ether_addr], , , [#include <net/ethernet.h>])
 	if test "$ac_cv_type_struct_ether_addr" = "yes"; then
-		AC_DEFINE_UNQUOTED(HAVE_STRUCT_ETHER_ADDR, 1)
+		AC_DEFINE(HAVE_STRUCT_ETHER_ADDR, 1, [struct ether_addr])
 		ether_includes="#include <net/ethernet.h>"
 	else dnl Solaris
 		ac_cv_type_struct_ether_addr="no"
@@ -83,7 +83,7 @@ else dnl BSD
 				[ac_cv_type_struct_ether_addr="yes"], [])
 		AC_MSG_RESULT($ac_cv_type_struct_ether_addr)
 		if test "$ac_cv_type_struct_ether_addr" = "yes"; then
-			AC_DEFINE_UNQUOTED(HAVE_STRUCT_ETHER_ADDR, 1)
+			AC_DEFINE(HAVE_STRUCT_ETHER_ADDR, 1, [struct ether_addr])
 			ether_includes="#include <sys/types.h>
 #include <sys/socket.h>
 #include <net/if.h>
@@ -99,11 +99,11 @@ dnl check struct members
 if test "$ac_cv_type_struct_ether_addr" = "yes"; then
 	AC_CHECK_MEMBER([struct ether_addr.octet], , , [$ether_includes])
 	if test "$ac_cv_member_struct_ether_addr_octet" = "yes"; then
-		AC_DEFINE_UNQUOTED(HAVE_STRUCT_ETHER_ADDR_OCTET, 1)
+		AC_DEFINE(HAVE_STRUCT_ETHER_ADDR_OCTET, 1, [struct ether_addr.octet])
 	else
 		AC_CHECK_MEMBER([struct ether_addr.ether_addr_octet], , , [$ether_includes])
 		if test "$ac_cv_member_struct_ether_addr_ether_addr_octet" = "yes"; then
-			AC_DEFINE_UNQUOTED(HAVE_STRUCT_ETHER_ADDR_ETHER_ADDR_OCTET, 1)
+			AC_DEFINE(HAVE_STRUCT_ETHER_ADDR_ETHER_ADDR_OCTET, 1, [struct ether_addr.ether_addr_octet])
 		else
 			AC_MSG_ERROR([*** Found struct ether_addr, but no usable octet member... ***])
 		fi
@@ -221,7 +221,7 @@ AC_CHECK_FUNC(ether_hostton, [ether_host
 ])
 AC_SUBST(EXTRA_LIBS)
 if test x$ether_hostton = xyes; then
-	AC_DEFINE_UNQUOTED(HAVE_ETHER_HOSTTON, 1)
+	AC_DEFINE_UNQUOTED(HAVE_ETHER_HOSTTON, 1, [ether_hostton])
 fi
 
 
@@ -242,17 +242,6 @@ dnl *** Because of that, you won't be ab
 dnl fi
 
 
-dnl config.h.in defines
-AC_DEFINE([HAVE_STRUCT_ETHER_ADDR], 0, [struct ether_addr])
-AC_DEFINE([HAVE_STRUCT_ETHER_ADDR_ETHER_ADDR_OCTET], 0, [struct ether_addr.ether_addr_octet])
-AC_DEFINE([HAVE_STRUCT_ETHER_ADDR_OCTET], 0, [struct ether_addr.octet])
-AC_DEFINE([HAVE_ETHER_HOSTTON], 0, [ether_hostton])
-
-
-dnl full featured warnings
-CFLAGS="$CFLAGS -W -Wall -Wpointer-arith -Wimplicit-function-declaration"
-
-
 dnl create the Makefiles
 AC_OUTPUT([intl/Makefile po/Makefile.in
 Makefile
