--- setup.cfg.orig	2008-12-12 18:29:21.000000000 -0800
+++ setup.cfg	2008-12-12 18:29:45.000000000 -0800
@@ -1,9 +1,9 @@
 [_ldap]
 extra_objects = 
 extra_compile_args = 
-libs = ldap_r lber sasl2 ssl crypto
-library_dirs = /usr/local/openldap-2.3/lib
-include_dirs = /usr/local/openldap-2.3/include /usr/include/sasl
+libs = __LIBS__ 
+library_dirs = __LIBDIR__ 
+include_dirs = __INCDIR__ 
 
 [egg_info]
 tag_build = 
