Upstream-Status: Backport [https://github.com/libdynd/libdynd/pull/1371]
--- src/dynd/type.cpp
+++ src/dynd/type.cpp
@@ -680,7 +680,7 @@ void dynd::hexadecimal_print_summarized(std::ostream &o, const char *data, intpt
 
 static intptr_t line_count(const std::string &s)
 {
-  return 1 + count_if(s.begin(), s.end(), bind1st(equal_to<char>(), '\n'));
+  return 1 + count_if(s.begin(), s.end(), bind(equal_to<char>(), '\n', placeholders::_1));
 }
 
 static void summarize_stats(const std::string &s, intptr_t &num_rows, intptr_t &max_num_cols)
@@ -695,7 +695,7 @@ void dynd::print_indented(ostream &o, const std::string &indent, const std::stri
   const char *end = s.data() + s.size();
   const char *cur = begin;
   while (cur != end) {
-    const char *next = find_if(cur, end, bind1st(equal_to<char>(), '\n'));
+    const char *next = find_if(cur, end, bind(equal_to<char>(), '\n', placeholders::_1));
     if (*next == '\n')
       ++next;
     if (!skipfirstline || cur != begin)
