--- vendor/raylib/raylib.odin.orig	2024-01-03 22:32:45
+++ vendor/raylib/raylib.odin	2024-01-11 03:26:16
@@ -139,7 +139,7 @@
 	when ODIN_ARCH == .arm64 {
 		when RAYLIB_SHARED {
 			foreign import lib {
-				"macos-arm64/libraylib.500.dylib",
+				"system:raylib",
 				"system:Cocoa.framework",
 				"system:OpenGL.framework",
 				"system:IOKit.framework",
@@ -155,7 +155,7 @@
 	} else {
 		when RAYLIB_SHARED {
 			foreign import lib {
-				"macos/libraylib.500.dylib",
+				"system:raylib",
 				"system:Cocoa.framework",
 				"system:OpenGL.framework",
 				"system:IOKit.framework",
