--- vendor/raylib/raylib.odin.orig	2024-07-01 20:56:07
+++ vendor/raylib/raylib.odin	2024-07-07 20:11:56
@@ -117,9 +117,7 @@
 	}
 } else when ODIN_OS == .Darwin {
 	foreign import lib {
-		"macos" +
-			("-arm64" when ODIN_ARCH == .arm64 else "") +
-			"/libraylib" + (".500.dylib" when RAYLIB_SHARED else ".a"),
+		"system:raylib",
 		"system:Cocoa.framework",
 		"system:OpenGL.framework",
 		"system:IOKit.framework",
