--- lib/IMProto.c.orig	2002-10-03 18:35:28.000000000 +0900
+++ lib/IMProto.c	2009-09-21 03:20:53.000000000 +0900
@@ -281,7 +281,7 @@
 	 * The unix domain socket pathname has the following form:
 	 *   <UNIX_SOCKET_DIR>/<Display Name>-<Language>
 	 */
-	(void)mkdir(UNIX_SOCKET_DIR, 0777);
+	(void)mkdir(UNIX_SOCKET_DIR, 01777);
 #ifdef S_IFLNK
 	{
 	    /*
@@ -292,11 +292,11 @@
 	    struct stat st;
 	    if (lstat(UNIX_SOCKET_DIR, &st) == 0 &&
 		(st.st_mode & S_IFMT) == S_IFDIR) {
-		(void)chmod(UNIX_SOCKET_DIR, 0777);
+		(void)chmod(UNIX_SOCKET_DIR, 01777);
 	    }
 	}
 #else
-	(void)chmod(UNIX_SOCKET_DIR, 0777);
+	(void)chmod(UNIX_SOCKET_DIR, 01777);
 #endif
 	(void)sprintf(path, "%s/%s", UNIX_SOCKET_DIR,
 		      DisplayString(XtDisplay(new)));
