--- cmd/Kinput2.ad.orig	2002-10-03 18:35:26.000000000 +0900
+++ cmd/Kinput2.ad	2007-10-17 08:00:36.000000000 +0900
@@ -96,12 +96,13 @@
 ! XimpProtocol
 *XimpProtocol.ForceDefaultServer: true
 ! X Input Method Protocol
-*IMProtocol.locales: ja_JP.SJIS, ja_JP.EUC, ja_JP, japanese, japan, ja
+*IMProtocol.locales: ja_JP.UTF-8, ja_JP.eucJP, ja_JP.SJIS, ja_JP, japanese, japan, ja
 ! common to XimpProtocol and IMProtocol
 *defaultFontList: -misc-fixed-medium-r-normal--14-*-*-*-c-*
 *ConversionStartKeys: \
 	Shift<Key>space \n\
-	Ctrl<Key>Kanji
+	Ctrl<Key>Kanji \n\
+	Mod2<Key>space
 !!
 !! sending back unused keypress events
 !!
