--- src/3rdparty/webkit/Source/WebCore/WebCore.pro.orig	2012-08-08 16:27:24.000000000 -0400
+++ src/3rdparty/webkit/Source/WebCore/WebCore.pro	2012-08-08 16:30:43.000000000 -0400
@@ -2975,7 +2975,7 @@
             platform/mac/WebWindowAnimation.mm
 
         DEFINES+=NSGEOMETRY_TYPES_SAME_AS_CGGEOMETRY_TYPES
-        contains(CONFIG, "x86") {
+        contains(CONFIG, "x86")|contains(CONFIG, "ppc") {
             DEFINES+=NS_BUILD_32_LIKE_64
         }
 
