--- src/plugins/bearer/corewlan/corewlan.pro.orig	2012-03-14 10:01:12.000000000 -0400
+++ src/plugins/bearer/corewlan/corewlan.pro	2012-04-24 09:23:47.000000000 -0400
@@ -5,7 +5,7 @@
 LIBS += -framework Foundation -framework SystemConfiguration
 
 contains(QT_CONFIG, corewlan) {
-    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, "/Developer/SDKs/MacOSX10\.[67]\.sdk") {
+    isEmpty(QMAKE_MAC_SDK)|contains(QMAKE_MAC_SDK, "@DEVELOPER_SDK@") {
          LIBS += -framework CoreWLAN -framework Security
     }
 }
