--- src/3rdparty/webkit/Source/WebKit/qt/Api/qgraphicswebview.cpp.orig	2013-06-07 07:16:58.000000000 +0200
+++ src/3rdparty/webkit/Source/WebKit/qt/Api/qgraphicswebview.cpp	2013-10-28 18:54:43.000000000 +0100
@@ -471,7 +471,7 @@
         return;
 
     page->d->view.clear();
-    page->d->client = 0;
+    page->d->client = nullptr;
 
     // if the page was created by us, we own it and need to
     // destroy it as well.
