--- src/imports/shaders/shadereffectitem.cpp.orig
+++ src/imports/shaders/shadereffectitem.cpp
@@ -509,7 +509,7 @@ void ShaderEffectItem::updateEffectState(const QMatrix4x4 &matrix)
             m_program.setUniformValue(name.constData(), qvariant_cast<QTransform>(v));
             break;
         case QVariant::Int:
-            m_program.setUniformValue(name.constData(), v.toInt());
+            m_program.setUniformValue(name.constData(), GLint(v.toInt()));
             break;
         case QVariant::Bool:
             m_program.setUniformValue(name.constData(), GLint(v.toBool()));
@@ -852,7 +852,7 @@ void ShaderEffectItem::updateShaderProgram()
     if (m_program.isLinked()) {
         m_program.bind();
         for (int i = 0; i < m_sources.size(); ++i)
-            m_program.setUniformValue(m_sources.at(i).name.constData(), i);
+            m_program.setUniformValue(m_sources.at(i).name.constData(), (GLint) i);
     }
 
     m_program_dirty = false;