--- src/plugins/bearer/corewlan/qcorewlanengine.mm.orig	2011-08-04 16:33:29.000000000 +0200
+++ src/plugins/bearer/corewlan/qcorewlanengine.mm	2011-08-04 16:34:21.000000000 +0200
@@ -52,6 +52,10 @@
 #include <QtCore/qdebug.h>
 
 #include <QDir>
+#ifdef MAC_SDK_10_7
+   #include <CoreWLAN/CoreWLANConstants.h>
+   #include <CoreWLAN/CoreWLANTypes.h>
+#endif 
 #include <CoreWLAN/CoreWLAN.h>
 #include <CoreWLAN/CWInterface.h>
 #include <CoreWLAN/CWNetwork.h>
