--- Src5.1.6/makefile.orig	2007-11-10 19:45:40.000000000 -0500
+++ Src5.1.6/makefile	2008-10-08 12:01:56.000000000 -0400
@@ -12,7 +12,7 @@
 
 CC=cc -DNXT 	# -DNXT enables the X operator in LTL
 # CC=cc -m32 -DNXT 	# for 32bit compilation on a 64bit system
-CFLAGS=-ansi -D_POSIX_SOURCE	# on some systems add: -I/usr/include
+CFLAGS=-ansi -D_POSIX_SOURCE -DMAC-DCPP="\"/usr/bin/cpp -xassembler-with-cpp\""	# on some systems add: -I/usr/include
 
 # for a more picky compilation:
 # CFLAGS=-std=c99 -Wstrict-prototypes -pedantic -fno-strength-reduce -fno-builtin -W -Wshadow -Wpointer-arith -Wcast-qual -Winline -Wall -g
