--- aclocal.m4.orig
+++ aclocal.m4
@@ -506,9 +506,9 @@
         *-apple-darwin*)
             SHLIB_CFLAGS="-fno-common"
             SHLIB_SUFFIX=".dylib"
-            SHLIB_LD="${CC} -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name @rpath/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
+            SHLIB_LD="${CC} -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name \${INST_DIR}/lib/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
             LDFLAGS="-Wl,-rpath,${INSTDIR}/lib,-rpath,\${GISBASE}/lib"
-            LD_LIBRARY_PATH_VAR="LD_RUN_PATH"
+            LD_LIBRARY_PATH_VAR="DYLD_LIBRARY_PATH"
             ;;
 	*-sun-solaris*)
 	    # Note: If _REENTRANT isn't defined, then Solaris


--- configure.orig
+++ configure
@@ -4045,9 +4045,9 @@
         *-apple-darwin*)
             SHLIB_CFLAGS="-fno-common"
             SHLIB_SUFFIX=".dylib"
-            SHLIB_LD="${CC} -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name @rpath/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
+            SHLIB_LD="${CC} -dynamiclib -compatibility_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -current_version \${GRASS_VERSION_MAJOR}.\${GRASS_VERSION_MINOR} -install_name \${INST_DIR}/lib/lib\${LIB_NAME}\${SHLIB_SUFFIX}"
             LDFLAGS="-Wl,-rpath,${INSTDIR}/lib,-rpath,\${GISBASE}/lib"
-            LD_LIBRARY_PATH_VAR="LD_RUN_PATH"
+            LD_LIBRARY_PATH_VAR="DYLD_LIBRARY_PATH"
             ;;
     *-sun-solaris*)
         # Note: If _REENTRANT isn't defined, then Solaris
