--- Makefile.orig
+++ Makefile
@@ -24,7 +24,7 @@ include $(MODULE_TOPDIR)/include/Make/Grass.make
 
 # Install directories
 exec_prefix=            ${prefix}
-BINDIR=			${UNIX_BIN}
+BINDIR=			${DESTDIR}${UNIX_BIN}
 
 # Shell commands
 MAKE_DIR_CMD=		mkdir -p -m 755
@@ -81,6 +81,9 @@ DOXNAME=grass
 include $(MODULE_TOPDIR)/include/Make/Docs.make
 include $(MODULE_TOPDIR)/include/Make/Doxygen.make
 
+FINAL_GISBASE:=${INST_DIR}
+INST_DIR:=${DESTDIR}${INST_DIR}
+
 default: builddemolocation
 	@echo "GRASS GIS compilation log"     > $(ERRORLOG)
 	@echo "-------------------------"    >> $(ERRORLOG)
@@ -272,7 +275,7 @@ real-install: FORCE
 	test -d ${INST_DIR} || ${MAKE_DIR_CMD} ${INST_DIR}
 	@##### test -d ${INST_DIR}/dev || ${MAKE_DIR_CMD} ${INST_DIR}/dev
 	test -d ${BINDIR} || ${MAKE_DIR_CMD} ${BINDIR}
-	-sed -e "s#GISBASE=.*#GISBASE=${INST_DIR}#" ${ARCH_BINDIR}/grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR} > ${BINDIR}/grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}
+	-sed -e "s#GISBASE=.*#GISBASE=${FINAL_GISBASE}#" ${ARCH_BINDIR}/grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR} > ${BINDIR}/grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}
 	-chmod a+x ${BINDIR}/grass${GRASS_VERSION_MAJOR}${GRASS_VERSION_MINOR}
 	-cd ${GISBASE} ; tar cBf - $(FILES) | (cd ${INST_DIR} ; tar xBf - ) 2>/dev/null
 	-cd ${GISBASE} ; tar cBf - bin | (cd ${INST_DIR} ; tar xBf - ) 2>/dev/null
@@ -310,8 +313,6 @@ ifeq ($(strip $(MINGW)),)
 	-tar cBf - gem/skeleton | (cd ${INST_DIR}/etc ; tar xBf - ) 2>/dev/null
 	-${INSTALL} gem/gem$(GRASS_VERSION_MAJOR)$(GRASS_VERSION_MINOR) ${BINDIR} 2>/dev/null
 endif
-	@# enable OSX Help Viewer
-	@if [ "`cat include/Make/Platform.make | grep -i '^ARCH.*darwin'`" ] ; then /bin/ln -sfh "${INST_DIR}/docs/html" /Library/Documentation/Help/GRASS-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR} ; fi
 
 
 install-strip: FORCE
