--- configure.orig
+++ configure
@@ -1424,9 +1424,9 @@ fi
 # Set INST_DIR
 
 if test -n "$MACOSX_APP" ; then
-    INSTDIR='${prefix}'"/GRASS-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.app/Contents/MacOS"
+    INSTDIR='@@APPLICATIONS_DIR@@'"/GRASS-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.app/Contents/MacOS"
 else
-    INSTDIR='${prefix}'"/grass-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.${GRASS_VERSION_RELEASE}"
+    INSTDIR='${prefix}'"/share/grass-${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.${GRASS_VERSION_RELEASE}"
 fi
 
 
