http://trac.osgeo.org/grass/ticket/2170

--- raster/r.terraflow/direction.cc.orig
+++ raster/r.terraflow/direction.cc
@@ -53,11 +53,11 @@ encodeDirectionMFD(const genericWindow<elevation_type>& elevwin,
   
   if(!is_nodata(elevwin.get())) {
     dir = 0;
-    if (elevwin.get(5) < elevwin.get() && !is_void(elevwin.get(5))) dir |= 1;
-    if (elevwin.get(3) < elevwin.get() && !is_void(elevwin.get(3))) dir |= 16;
+    if (elevwin.get(5) < elevwin.get() && !elevation_type_is_void(elevwin.get(5))) dir |= 1;
+    if (elevwin.get(3) < elevwin.get() && !elevation_type_is_void(elevwin.get(3))) dir |= 16;
     for(int i=0; i<3; i++) {
-      if(elevwin.get(i) < elevwin.get() && !is_void(elevwin.get(i))) dir |= 32<<i;
-      if(elevwin.get(i+6) < elevwin.get() && !is_void(elevwin.get(6+i))) dir |= 8>>i;
+      if(elevwin.get(i) < elevwin.get() && !elevation_type_is_void(elevwin.get(i))) dir |= 32<<i;
+      if(elevwin.get(i+6) < elevwin.get() && !elevation_type_is_void(elevwin.get(6+i))) dir |= 8>>i;
     }
   }
   
--- raster/r.terraflow/nodata.cc.orig
+++ raster/r.terraflow/nodata.cc
@@ -73,7 +73,7 @@ is_nodata(float x) {
 
 
 int
-is_void(elevation_type el) {
+elevation_type_is_void(elevation_type el) {
   return (el == nodataType::ELEVATION_NODATA);
 }
 
--- raster/r.terraflow/nodata.h.orig
+++ raster/r.terraflow/nodata.h
@@ -37,7 +37,7 @@
 int is_nodata(elevation_type el);
 int is_nodata(int x);
 int is_nodata(float x);
-int is_void(elevation_type el);
+int elevation_type_is_void(elevation_type el);
 
 
 class nodataType : public ijBaseType {
