--- support/pma.c.orig	2022-08-03 04:54:28.000000000 -0500
+++ support/pma.c	2022-09-04 15:22:29.000000000 -0500
@@ -337,7 +337,7 @@
   p->fprev = p->fnext = NULL;
 }
 
-#define MMAP(N) mmap(NULL, (N), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_NORESERVE, -1, 0)
+#define MMAP(N) mmap(NULL, (N), PROT_NONE, MAP_PRIVATE | MAP_ANON | MAP_NORESERVE, -1, 0)
 #define MUNMAP(A, N) do { if (0 != munmap((A), (N))) { ERR("munmap()" ERN); SERN; } } while (0)
 static void * addrgap(off_t n) {  // find big gap in address space to map n bytes
   void *A, *Amax = NULL;  size_t L = 0, U, Max = 0, N = (size_t)n;  char *r;
