From c406e6637a50b4a03ba36e0d304bca76dad9a8a5 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 3 Nov 2023 18:22:45 +0800
Subject: [PATCH 02/12] inspector: unbreak build with gcc

---
 src/inspector/string-16.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/inspector/string-16.h b/src/inspector/string-16.h
index b4cbd016a6c..45f756e3360 100644
--- a/v8/src/inspector/string-16.h
+++ b/v8/src/inspector/string-16.h
@@ -147,7 +147,7 @@ String16 String16::concat(T... args) {
 
 }  // namespace v8_inspector
 
-#if !defined(__APPLE__) || defined(_LIBCPP_VERSION)
+#if !(defined(__APPLE__) && defined(__clang__)) || defined(_LIBCPP_VERSION)
 
 namespace std {
 template <>
