From 87e9a3995ad3a0cd95716b5489b176e4356fe875 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Thu, 9 Nov 2023 03:04:09 +0800
Subject: [PATCH 05/12] runtime-utils.h: fix static_cast

---
 src/runtime/runtime-utils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/runtime/runtime-utils.h b/src/runtime/runtime-utils.h
index 170c0bcdbcc..a7e02aee1a8 100644
--- a/v8/src/runtime/runtime-utils.h
+++ b/v8/src/runtime/runtime-utils.h
@@ -136,7 +136,7 @@ static inline ObjectPair MakePair(Object x, Object y) {
 #if defined(V8_TARGET_LITTLE_ENDIAN)
   return x.ptr() | (static_cast<ObjectPair>(y.ptr()) << 32);
 #elif defined(V8_TARGET_BIG_ENDIAN)
-  return y->ptr() | (static_cast<ObjectPair>(x->ptr()) << 32);
+  return y.ptr() | (static_cast<ObjectPair>(x.ptr()) << 32);
 #else
 #error Unknown endianness
 #endif
