--- src/Makevars.orig	2022-05-03 22:01:45.000000000 +0800
+++ src/Makevars	2023-03-23 14:49:42.000000000 +0800
@@ -1,6 +1,6 @@
 PKG_CXXFLAGS = -I../inst/include  -DPRIMME_INT_SIZE=0 -DF77UNDERSCORE -DUSE_XHEEV -DUSE_ZGESV -DUSE_XHEGV -DPRIMME_INT_SIZE=0 -DPRIMME_WITHOUT_FLOAT -DPRIMME_BLAS_RCOMPLEX
 # Linker will discard primmeext if R provides a full LAPACK
-PKG_LIBS = -Lprimme -lprimme $(LAPACK_LIBS) -lprimmeext $(BLAS_LIBS) $(FLIBS)
+PKG_LIBS = ../src/primme/libprimme.a $(LAPACK_LIBS) ../src/primme/libprimmeext.a $(BLAS_LIBS) $(FLIBS)
 
 $(SHLIB): primme/libprimme.a primme/libprimmeext.a
 
