--- scripts/configure.in.orig	2012-04-20 09:12:44.000000000 -0500
+++ scripts/configure.in	2012-09-17 20:51:22.000000000 -0500
@@ -23,6 +23,7 @@
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_CPP
+AC_PROG_CXX
 AC_ISC_POSIX
 AC_PROG_INSTALL
 AC_PROG_RANLIB
@@ -39,7 +40,7 @@
 dnl stage will not execute correctly on the ".in" files in the scmos
 dnl directory.
 
-if test "$CPP" = "gcc -E" ; then
+if test "$CPP" = "$CC -E" ; then
    CPP="$CPP -x c"
 fi
 
@@ -251,7 +252,6 @@
 LD_RUN_PATH=""
 WISH_EXE=""
 TCLSH_EXE=""
-CXX=
 
 modules=""
 unused=""
@@ -1290,7 +1290,7 @@
       SHDLIB_EXT=".dylib"
       LDDL_FLAGS="-dynamiclib -flat_namespace -undefined suppress -noprebind"
       LDFLAGS="${LDFLAGS} ${LIB_SPECS}"
-      CFLAGS="${CFLAGS} ${X_CFLAGS} ${INC_SPECS} -I/sw/include -fno-common"
+      CFLAGS="${CFLAGS} ${X_CFLAGS} ${INC_SPECS} -fno-common"
       ;;
 
     *cygwin*)
@@ -1529,9 +1529,6 @@
    DEPEND_FLAG="-MM"
 fi
 
-if test $usingOA ; then
-   AC_PROG_CXX
-fi
 
 dnl Substitute all variables
 
@@ -1583,7 +1580,7 @@
 echo "Configuration Summary (principle requirements):"
 echo
 
-echo -n "X11:          "
+printf "X11:          "
 if test $usingX11 ; then
    echo "yes"
 else
@@ -1597,7 +1594,7 @@
    echo
 fi
 
-echo -n "OpenGL:       "
+printf "OpenGL:       "
 if test $usingOGL ; then
    echo "yes"
 else
@@ -1614,9 +1611,9 @@
 fi
 
 if test $usingTcl ; then
-   echo -n "Tcl/Tk:       "
+   printf "Tcl/Tk:       "
    echo "yes (version ${TCL_VERSION})"
-   echo -n "BLT:          "
+   printf "BLT:          "
    if test "x${TCLSH_EXE}" = "x" ; then
       echo "Cannot find 'tclsh' executable to check existance of the BLT package"
    else
@@ -1639,10 +1636,10 @@
    fi
 else
    if test $usingScheme ; then
-     echo -n "Scheme:       "
+     printf "Scheme:       "
      echo "yes"
    else
-     echo -n "Tcl/Tk:       "
+     printf "Tcl/Tk:       "
      echo "no"
      echo
      echo "  Without Tcl/Tk, you cannot run the enhanced version of magic"
