From 489e9799103dca140558916cb39d64430c3f5566 Mon Sep 17 00:00:00 2001
From: "Mike L." <cl.jeremy@gmail.com>
Date: Tue, 16 Sep 2025 11:17:25 +0200
Subject: [PATCH] s4:kdc/db-glue: Use realloc_p macro

Signed-off-by: Mike L. <cl.jeremy@gmail.com>
---
 source4/kdc/db-glue.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/source4/kdc/db-glue.c b/source4/kdc/db-glue.c
index a776dc00..8201de4f 100644
--- source4/kdc/db-glue.c
+++ source4/kdc/db-glue.c
@@ -601,7 +601,7 @@ static krb5_error_code samba_kdc_merge_keys(struct sdb_keys *keys,
 	num_old_keys = old_keys->len;
 	total_keys = num_keys + num_old_keys;
 
-	skeys = realloc(keys->val, total_keys * sizeof keys->val[0]);
+	skeys = realloc_p(keys->val, struct sdb_key, total_keys);
 	if (skeys == NULL) {
 		return ENOMEM;
 	}
@@ -1396,9 +1396,9 @@ static krb5_error_code get_key_trust_public_keys(TALLOC_CTX *mem_ctx,
 					goto pub_keys_oom;
 				}
 			} else {
-				pub_keys.keys = reallocarray(pub_keys.keys,
-							pub_keys.len + 1,
-							sizeof(pub_key));
+				pub_keys.keys = realloc_p(pub_keys.keys,
+							struct sdb_pub_key,
+							pub_keys.len + 1);
 				if (pub_keys.keys == NULL) {
 					SAFE_FREE(pub_key.exponent.data);
 					SAFE_FREE(pub_key.modulus.data);
