--- third_party/ngtcp2/wscript.orig	2025-08-07 16:57:35.000000000 +0200
+++ third_party/ngtcp2/wscript	2025-09-19 09:46:32.000000000 +0200
@@ -5,11 +5,6 @@
 from samba_utils import save_file
 
 def configure(conf):
-    if not sys.platform.startswith('linux'):
-        Logs.warn('libngtcp2 requires linux')
-        conf.SET_TARGET_TYPE('ngtcp2', 'EMPTY')
-        return
-
     if not conf.CONFIG_GET('HAVE_GNUTLS_CB_TLS_SERVER_END_POINT'):
         Logs.warn('libngtcp2 requires gnutls 3.7.2')
         conf.SET_TARGET_TYPE('ngtcp2', 'EMPTY')
