--- src/svg_image.c.orig	2026-03-21 08:30:18.000000000 -0500
+++ src/svg_image.c	2026-03-21 08:30:18.000000000 -0500
@@ -272,7 +272,7 @@
 
     /* expand gray bit depth if needed */
     if (color_type == PNG_COLOR_TYPE_GRAY && depth < 8)
-	png_set_gray_1_2_4_to_8 (png);
+	png_set_expand_gray_1_2_4_to_8 (png);
 
     /* transform transparency to alpha */
     if (png_get_valid(png, info, PNG_INFO_tRNS))
