--- configure.ac.orig	2012-12-29 13:17:57.000000000 -0800
+++ configure.ac	2012-12-29 13:28:51.000000000 -0800
@@ -473,11 +473,11 @@
             [with_python=$withval], [with_python=skipped])
 
 if test "x$with_python" = "xyes"; then
-    checkPYTHON_CFLAGS=`python -c "import distutils.sysconfig ; print '-I%s' % distutils.sysconfig.get_config_var('INCLUDEPY')" 2>/dev/null`
+    checkPYTHON_CFLAGS=`${prefix}/bin/python2.6 -c "import distutils.sysconfig ; print '-I%s' % distutils.sysconfig.get_config_var('INCLUDEPY')" 2>/dev/null`
     if test "$?" -gt "0"; then
         with_python="no"
     else
-        checkPYTHON_LIBS=`python -c "import distutils.sysconfig ; print '%s/%s %s' % (distutils.sysconfig.get_config_var('LIBPL'),distutils.sysconfig.get_config_var('LDLIBRARY'),distutils.sysconfig.get_config_var('LIBS'))" 2>/dev/null`
+        checkPYTHON_LIBS=-lpython2.6
         if test "$?" -gt "0"; then
             with_python="no"
         else
