--- src/Makefile.am.orig	2013-09-08 23:55:35.000000000 -0700
+++ src/Makefile.am	2013-09-09 00:03:45.000000000 -0700
@@ -221,7 +221,7 @@
 # someone updates the BZR working directory.
 inkscape-version.cpp: $(inkscape_version_deps)
 	VER_PREFIX="$(VERSION)";\
-	VER_BZRREV=" r`bzr revno`"; \
+	VER_BZRREV=" r`bzr revno --tree`"; \
 	if test ! -z "`bzr status -S -V $(srcdir)`"; then \
 	    VER_CUSTOM=" custom"; \
 	fi; \
