--- src/extension/dbus/wrapper/inkscape-dbus-wrapper.c.orig	2013-04-22 12:42:00.000000000 -0700
+++ src/extension/dbus/wrapper/inkscape-dbus-wrapper.c	2013-04-22 12:44:28.000000000 -0700
@@ -95,8 +95,10 @@
     DBusGConnection *connection;
     GError *error;
     DBusGProxy *proxy;
-  
+ 
+#ifndef GLIB_VERSION_2_36 
     g_type_init ();
+#endif
 
     error = NULL;
     connection = dbus_g_bus_get (DBUS_BUS_SESSION,
