--- src/Makefile.am.orig	2012-12-29 13:17:57.000000000 -0800
+++ src/Makefile.am	2012-12-29 13:28:51.000000000 -0800
@@ -223,7 +223,7 @@
 # someone updates the BZR working directory.
 inkscape-version.cpp: $(inkscape_version_deps)
 	VER_PREFIX="$(VERSION)";\
-	VER_BZRREV=" r`bzr revno`"; \
+	VER_BZRREV=" r`bzr revno --tree`"; \
 	if test ! -z "`bzr status -S -V $(srcdir)`"; then \
 	    VER_CUSTOM=" custom"; \
 	fi; \
