--- src/Makefile.am.orig	2013-07-24 07:33:16.000000000 -0700
+++ src/Makefile.am	2013-07-24 07:41:03.000000000 -0700
@@ -217,7 +217,7 @@
 # someone updates the BZR working directory.
 inkscape-version.cpp: $(inkscape_version_deps)
 	VER_PREFIX="$(VERSION)";\
-	VER_BZRREV=" r`bzr revno`"; \
+	VER_BZRREV=" r`bzr revno --tree`"; \
 	if test ! -z "`bzr status -S -V $(srcdir)`"; then \
 	    VER_CUSTOM=" custom"; \
 	fi; \
