--- configure.ac.orig	2018-10-24 11:50:49.000000000 -0700
+++ configure.ac	2018-10-24 18:39:09.000000000 -0700
@@ -798,7 +798,7 @@
 fi
 AC_MSG_RESULT($have_glib_2_43)
 
-AC_PATH_PROG(GEGL, gegl, no)
+AC_PATH_PROG(GEGL, gegl-0.4, no)
 if test "x$GEGL" = xno; then
   add_deps_error([gegl], [Could not find native gegl executable in your PATH.])
 fi
@@ -2238,7 +2238,7 @@
 # Check for perl
 ################
 
-AC_PATH_PROGS(PERL,perl5 perl,perl)
+AC_PATH_PROGS(PERL,perl5.26 perl5 perl,perl)
 AX_PROG_PERL_VERSION([perl_required_version],,
                      [add_deps_error([perl >= perl_required_version])])
 
