--- app/widgets/gimpcriticaldialog.c.orig	2018-07-23 09:25:57.000000000 -0700
+++ app/widgets/gimpcriticaldialog.c	2018-07-23 22:07:29.000000000 -0700
@@ -274,26 +274,25 @@
 
   return TRUE;
 
-#elif defined(PLATFORM_OSX)
-
-  NSURL    *ns_url;
-  gboolean  retval;
-
-  @autoreleasepool
-    {
-      ns_url = [NSURL URLWithString: [NSString stringWithUTF8String: url]];
-      retval = [[NSWorkspace sharedWorkspace] openURL: ns_url];
-    }
-
-  return retval;
-
 #else
+  gboolean rstatus;
+  gchar *command_line;
 
-  return gtk_show_uri (gdk_screen_get_default (),
+  rstatus = gtk_show_uri (gdk_screen_get_default (),
                        url,
                        gtk_get_current_event_time(),
                        error);
 
+  if (!rstatus) {
+     command_line = g_strdup_printf ("open %s", url);
+
+     rstatus = g_spawn_command_line_async (command_line,
+                       error);
+
+     g_free (command_line);
+  }
+  
+  return rstatus;
 #endif
 }
 
