--- Makefile.orig	2010-11-13 20:47:18.000000000 -0500
+++ Makefile	2010-11-13 20:47:45.000000000 -0500
@@ -1,7 +1,7 @@
 # Makefile for jam
 
 CC = cc
-CFLAGS =
+#CFLAGS =
 EXENAME = ./jam0
 TARGET = -o $(EXENAME)
 
@@ -59,7 +59,7 @@
 	rules.c scan.c search.c timestamp.c variable.c
 
 all: $(EXENAME)
-	$(EXENAME)
+	$(EXENAME) -sCCFLAGS="@CFLAGS@ $(CFLAGS)" @EXTRA@
 
 $(EXENAME):
 	$(CC) $(TARGET) $(CFLAGS) $(SOURCES) $(LINKLIBS)
