--- include/CGAL/config.h.orig	2012-03-14 15:41:38.000000000 +0000
+++ include/CGAL/config.h	2012-03-14 15:37:42.000000000 +0000
@@ -26,6 +26,15 @@
 #ifndef CGAL_CONFIG_H
 #define CGAL_CONFIG_H
 
+
+// Workaround to the following bug: 
+//   https://bugreports.qt.nokia.com/browse/QTBUG-22829
+#ifdef Q_MOC_RUN 
+// When Qt moc runs on CGAL files, do not process 
+// <boost/type_traits/has_operator.hpp> 
+#  define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED 
+#endif
+
 #ifdef CGAL_INCLUDE_WINDOWS_DOT_H
 // Mimic users including this file which defines min max macros
 // and other names leading to name clashes
