--- demo/Mesh_3/CMakeLists.txt.orig	2011-09-05 15:47:29.000000000 +0200
+++ demo/Mesh_3/CMakeLists.txt	2011-09-29 16:51:17.000000000 +0200
@@ -202,7 +202,7 @@
     else()
       set(other_sources ${ARGN_TAIL})
     endif()
-    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" ${plugin_implementation_base_name}.moc )
+    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.moc" )
     add_file_dependencies( ${plugin_implementation_base_name}.moc "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" )
 
     add_library(${plugin_name} MODULE ${option} ${plugin_implementation_base_name}.moc ${plugin_implementation_base_name}.cpp ${other_sources})
--- demo/Mesh_3/implicit_functions/CMakeLists.txt.orig	2011-09-05 15:47:29.000000000 +0200
+++ demo/Mesh_3/implicit_functions/CMakeLists.txt	2011-09-29 16:51:17.000000000 +0200
@@ -51,7 +51,7 @@
     else()
       set(other_sources ${ARGN_TAIL})
     endif()
-    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" ${plugin_implementation_base_name}.moc )
+    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.moc" )
     add_file_dependencies( ${plugin_implementation_base_name}.moc "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" )
 
     add_library(${plugin_name} MODULE ${option} ${plugin_implementation_base_name}.moc ${plugin_implementation_base_name}.cpp ${other_sources})
--- demo/Surface_reconstruction_points_3/CMakeLists.txt.orig	2011-09-05 15:47:56.000000000 +0200
+++ demo/Surface_reconstruction_points_3/CMakeLists.txt	2011-09-29 16:51:17.000000000 +0200
@@ -219,7 +219,7 @@
     else()
       set(other_sources ${ARGN_TAIL})
     endif()
-    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" ${plugin_implementation_base_name}.moc )
+    qt4_generate_moc( "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.moc" )
     add_file_dependencies( ${plugin_implementation_base_name}.moc "${CMAKE_CURRENT_SOURCE_DIR}/${plugin_implementation_base_name}.cpp" )
 
     add_library(${plugin_name} MODULE ${option} ${plugin_implementation_base_name}.moc ${plugin_implementation_base_name}.cpp ${other_sources})
