--- Makefile	2011-08-15 16:00:48.000000000 +0200
+++ Makefile	2012-07-04 13:32:08.000000000 +0200
@@ -74,8 +74,7 @@
 
 install: all
 	if [ -x "$(PATH_OCAMLFIND)" ]; then \
-	  $(PATH_OCAMLFIND) query $(PROJECT_NAME) && $(PATH_OCAMLFIND) remove $(PROJECT_NAME) || true; \
-	  $(PATH_OCAMLFIND) install $(PROJECT_NAME) META -optional \
+	  $(PATH_OCAMLFIND) install -destdir $(DESTROOT)$(PATH_OCAML_PREFIX)/lib/ocaml/site-lib $(PROJECT_NAME) META -optional \
 	    $(PATH_BUILD)/src/syntax/$(PROJECT_NAME)_pp.cmo \
 	    $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.cm* \
 	    $(PATH_BUILD)/src/threads/$(PROJECT_NAME)Thread.o \
@@ -86,7 +85,8 @@
 	    $(PATH_BUILD)/$(PROJECT_NAME).cmo \
 	    $(PATH_BUILD)/$(PROJECT_NAME).cmxa \
 	    $(PATH_BUILD)/$(PROJECT_NAME).cmja \
-	    $(PATH_BUILD)/$(PROJECT_NAME).ja; \
+	    $(PATH_BUILD)/$(PROJECT_NAME).ja \
+	    -ldconf ignore; \
 	else \
 	  mkdir -p $(PATH_INSTALL); \
 	  cp $(PATH_BUILD)/src/syntax/$(PROJECT_NAME)_pp.cmo $(PATH_INSTALL); \
