--- src/AtExit.cc	2015-12-17 00:34:48.000000000 +0100
+++ src/AtExit.cc	2017-10-08 10:40:59.000000000 +0200
@@ -194,7 +194,7 @@
 
   private:
     typedef std::map< int, ExitQueueNode, std::less< int >,
-		      malloc_allocator< std::pair< int, ExitQueueNode > > >
+		      malloc_allocator< std::pair< const int, ExitQueueNode > > >
     exit_queue_type;
 
     bool			is_exiting;
--- src/MemChecker.cc	2016-06-17 23:36:24.000000000 +0200
+++ src/MemChecker.cc	2017-10-08 10:40:40.000000000 +0200
@@ -245,7 +245,7 @@
 
   private:
     typedef std::map< int, GCQueueNode, std::less< int >,
-		      malloc_allocator< std::pair< int, GCQueueNode > > >
+		      malloc_allocator< std::pair< const int, GCQueueNode > > >
     gc_queue_type;
 
     bool			is_exiting;
