# https://github.com/AcademySoftwareFoundation/openexr/commit/663708e53d7e8b33668e70117c5c632332e768ce

--- src/lib/OpenEXRCore/internal_xdr.h.orig	2023-03-28 23:25:15.000000000 +0800
+++ src/lib/OpenEXRCore/internal_xdr.h	2023-05-16 20:46:39.000000000 +0800
@@ -57,7 +57,11 @@
 #  define le32toh(x) OSSwapLittleToHostInt32(x)
 #  define htole64(x) OSSwapHostToLittleInt64(x)
 #  define le64toh(x) OSSwapLittleToHostInt64(x)
-#  define EXR_HOST_IS_NOT_LITTLE_ENDIAN (BYTE_ORDER != LITTLE_ENDIAN)
+#  if defined(__m68k__) || defined(__POWERPC__)
+#    define EXR_HOST_IS_NOT_LITTLE_ENDIAN 1
+#  else
+#    define EXR_HOST_IS_NOT_LITTLE_ENDIAN 0
+#  endif
 
 #elif defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
 
