--- Source/kwsys/kwsysPlatformTestsC.c.orig	2020-03-04 06:50:44.000000000 -0700
+++ Source/kwsys/kwsysPlatformTestsC.c	2020-03-17 03:35:40.000000000 -0700
@@ -56,12 +56,6 @@
 #endif
 
 #ifdef TEST_KWSYS_C_HAS_CLOCK_GETTIME_MONOTONIC
-#  if defined(__APPLE__)
-#    include <AvailabilityMacros.h>
-#    if MAC_OS_X_VERSION_MIN_REQUIRED < 101200
-#      error "clock_gettime not available on macOS < 10.12"
-#    endif
-#  endif
 #  include <time.h>
 int KWSYS_PLATFORM_TEST_C_MAIN()
 {
