diff --git packaging/osx/Info.plist.in packaging/osx/Info.plist.in
index 2aa1a238..de48e5e4 100644
--- packaging/osx/Info.plist.in
+++ packaging/osx/Info.plist.in
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+    <key>NSPrincipalClass</key>
+    <string>NSApplication</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>English</string>
 	<key>CFBundleInfoDictionaryVersion</key>
