From a31cbdab3d9994c3371ba4ad5996fe01dc314229 Mon Sep 17 00:00:00 2001
From: Uday Bondhugula <udayreddy@gmail.com>
Date: Tue, 14 Jan 2014 21:41:16 +0530
Subject: [PATCH 07/10] expose clast_pprint_expr

Signed-off-by: Uday Bondhugula <udayreddy@gmail.com>
Signed-off-by: Cedric Bastoul <cedric.bastoul@unistra.fr>
---
 include/cloog/pprint.h | 1 +
 source/pprint.c        | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/include/cloog/pprint.h b/include/cloog/pprint.h
index 55e2b5e..627e929 100644
--- include/cloog/pprint.h
+++ include/cloog/pprint.h
@@ -60,6 +60,7 @@ extern "C"
 void clast_pprint(FILE *foo, struct clast_stmt *root, int indent, 
 	    CloogOptions *options);
 
+void clast_pprint_expr(struct cloogoptions *i, FILE *dst, struct clast_expr *e);
 
 #if defined(__cplusplus)
   }
diff --git a/source/pprint.c b/source/pprint.c
index 83fcc32..ab4b22f 100644
--- source/pprint.c
+++ source/pprint.c
@@ -626,3 +626,9 @@ void clast_pprint(FILE *foo, struct clast_stmt *root,
 {
     pprint_stmt_list(options, foo, indent, root);
 }
+
+
+void clast_pprint_expr(struct cloogoptions *i, FILE *dst, struct clast_expr *e)
+{
+    pprint_expr(i, dst, e);
+}
-- 
1.9.3

