--- m4/ax_create_pkgconfig_info.m4.orig	2012-05-10 13:39:22.000000000 -0700
+++ m4/ax_create_pkgconfig_info.m4	2012-05-10 13:39:36.000000000 -0700
@@ -214,8 +214,8 @@
 Description: @PACKAGE_DESCRIPTION@
 Version: @PACKAGE_VERSION@
 Requires: @PACKAGE_REQUIRES@
-Libs: -L\${libdir} @LDFLAGS@ @LIBS@
-Cflags: -I\${includedir} @CPPFLAGS@
+Libs: -L\${libdir} @LIBS@
+Cflags: -I\${includedir}
 AXEOF
 fi # DONE generate $pkgconfig_generate.in
 AC_MSG_NOTICE(creating $pkgconfig_generate)
