--- F/gengeom.f.orig	2018-10-12 21:10:28.000000000 -0500
+++ F/gengeom.f	2018-10-12 21:21:49.000000000 -0500
@@ -47,8 +47,8 @@
       include 'param.inc'
       CHARACTER*256 MODE1, MODE2, MODE3,
      *     ANX, ANY, ANZ, FILE2, INPUT
-      REAL*8 PC(3,1),AC(3,2),BC(3,2),CC(3,2),FC(3,4),IC(3,2),RC(3,3),
-     *     HC(3,3),CSTMC(3,4),P2A(3,3),P2B(3,3),P2C(3,3),P2F(3,3),
+      REAL*8 PC(3,1),AC(3,4),BC(3,4),CC(3,4),FC(3,4),IC(3,4),RC(3,4),
+     *     HC(3,4),CSTMC(3,4),P2A(3,3),P2B(3,3),P2C(3,3),P2F(3,3),
      *     P2I(3,3),R2H(3,3),P2AI(3,3),P2BI(3,3),P2CI(3,3),P2FI(3,3),
      *     P2II(3,3),R2HI(3,3),P2H(3,3),P2HI(3,3),
      *     DV(3,3),SOP(3,3,48),TRX(48),TRY(48),TRZ(48),
@@ -674,7 +674,7 @@
       include 'param.inc'
       CHARACTER*80 FILE2
       INTEGER NAT(NAC)
-      REAL*8 AC(3,2),BC(3,2),CC(3,2),FC(3,4),IC(3,2),RC(3,3),HC(3,3),
+      REAL*8 AC(3,4),BC(3,4),CC(3,4),FC(3,4),IC(3,4),RC(3,4),HC(3,4),
      $     CSTMC(3,4),
      *     DVC(3,3),SOP(3,3,48),TRX(48),TRY(48),TRZ(48),
      *     XC(NAC,4),YC(NAC,4),ZC(NAC,4),
