--- Makefile.orig	2017-10-19 21:30:17.000000000 -0500
+++ Makefile	2018-10-27 16:15:38.000000000 -0500
@@ -7,10 +7,10 @@
 endif
 
 wcluster: $(files)
-	g++ -Wall -g -std=c++0x -O3 -o wcluster $(files) -lpthread
+	$(CXX) $(CXXFLAGS) -Wall -g -std=c++0x -o wcluster $(files) -lpthread
 
 %.o: %.cc
-	g++ -Wall -g -O3 -std=c++0x -o $@ -c $< 
+	$(CXX) $(CXXFLAGS) -Wall -g -std=c++0x -o $@ -c $< 
 
 clean:
 	rm wcluster basic/*.o *.o
