--- libclamav/Makefile.in.orig	2018-12-04 21:28:59.000000000 -0500
+++ libclamav/Makefile.in	2018-12-04 21:30:40.000000000 -0500
@@ -718,7 +718,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_builddir)/libclammspack
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir) -I$(top_builddir)/libclammspack -I../libclammspack/mspack
 depcomp = $(SHELL) $(top_srcdir)/config/depcomp
 am__depfiles_maybe = depfiles
 am__mv = mv -f
