diff --git configure.ac configure.ac
index 73633d0359..eec327f49f 100644
--- configure.ac
+++ configure.ac
@@ -108,6 +108,7 @@ m4_include([m4/reorganization/libs/bzip.m4])
 m4_include([m4/reorganization/libs/unrar.m4])
 
 m4_include([m4/reorganization/libs/systemd.m4])
+m4_include([m4/reorganization/milter/enable.m4])
 m4_include([m4/reorganization/milter/check.m4])
 m4_include([m4/reorganization/clamdtop.m4])
 
@@ -147,7 +148,6 @@ m4_include([m4/reorganization/code_checks/ipv6.m4])
 m4_include([m4/reorganization/code_checks/dns.m4])
 m4_include([m4/reorganization/code_checks/fanotify.m4])
 m4_include([m4/reorganization/code_checks/setpgrp.m4])
-m4_include([m4/reorganization/milter/enable.m4])
 m4_include([m4/reorganization/utility_checks/id.m4])
 m4_include([m4/reorganization/utility_checks/yp.m4])
 m4_include([m4/reorganization/usergroup.m4])
