=== modified file 'bzrlib/export/__init__.py'
--- bzrlib/export/__init__.py	2006-09-20 23:43:18 +0000
+++ bzrlib/export/__init__.py	2007-10-28 17:30:25 +0000
@@ -92,7 +92,11 @@
 
     if format not in _exporters:
         raise errors.NoSuchExportFormat(format)
-    return _exporters[format](tree, dest, root)
+    tree.lock_read()
+    try:
+        return _exporters[format](tree, dest, root)
+    finally:
+        tree.unlock()
 
 
 def get_root_name(dest):
