diff -uNr kdeedu-3.4.0/kalzium/src/Makefile.am kdeedu-3.4.0-new/kalzium/src/Makefile.am
--- kdeedu-3.4.0/kalzium/src/Makefile.am	2005-01-03 09:52:32.000000000 -0500
+++ kdeedu-3.4.0-new/kalzium/src/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,8 +1,11 @@
 # set the include path for X, qt and KDE
 INCLUDES= -I$(top_srcdir)/libkdeedu/kdeeduplot $(all_includes)
 
-bin_PROGRAMS = kalzium
-kalzium_SOURCES = kalzium.cpp \
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kalzium.la
+
+kalzium_la_SOURCES = kalzium.cpp \
 			main.cpp \
 			pse.cpp \
 			element.cpp \
@@ -25,7 +28,7 @@
 			tempslider.ui \
 			isotopewidget.cpp
 			
-kalzium_LDADD   = $(LIB_KIO) $(LIB_KDEUI)\
+kalzium_la_LIBADD   = $(LIB_KIO) $(LIB_KDEUI)\
 		  $(top_builddir)/libkdeedu/kdeeduplot/libkdeeduplot.la
 		  
 SUBDIRS = icons elementpics data
@@ -71,7 +74,7 @@
 KDE_ICON = AUTO
 
 # the library search path. 
-kalzium_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kalzium_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 elementnames:
 	  rm -f kalzium_i18n.cpp
diff -uNr kdeedu-3.4.0/kalzium/src/main.cpp kdeedu-3.4.0-new/kalzium/src/main.cpp
--- kdeedu-3.4.0/kalzium/src/main.cpp	2005-02-23 04:56:01.000000000 -0500
+++ kdeedu-3.4.0-new/kalzium/src/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -18,7 +18,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
     KAboutData about("kalzium", I18N_NOOP("Kalzium"), version, description,
                      KAboutData::License_GPL, "(C) 2002-2005 Carsten Niehaus", 0, "http://edu.kde.org/kalzium");
diff -uNr kdeedu-3.4.0/kbruch/src/Makefile.am kdeedu-3.4.0-new/kbruch/src/Makefile.am
--- kdeedu-3.4.0/kbruch/src/Makefile.am	2005-01-03 09:52:34.000000000 -0500
+++ kdeedu-3.4.0-new/kbruch/src/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,10 +1,14 @@
-bin_PROGRAMS = kbruch
 
-kbruch_COMPILE_FIRST = version.h
-kbruch_SOURCES = exercisebase.cpp taskvieweroptionsbase.ui task.cpp ratio.cpp taskview.cpp statisticsview.cpp primenumber.cpp kbruch.cpp mainqtwidget.cpp taskwidget.cpp fractionbasewidget.cpp resultwidget.cpp exercisecompare.cpp ratiowidget.cpp rationalwidget.cpp exerciseconvert.cpp exercisefactorize.cpp factorizedwidget.cpp settingsclass.kcfgc
-kbruch_LDADD   =  $(LIB_KDEUI)
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kbruch.la
+
+kbruch.cpp: version.h
+
+kbruch_la_SOURCES = exercisebase.cpp taskvieweroptionsbase.ui task.cpp ratio.cpp taskview.cpp statisticsview.cpp primenumber.cpp kbruch.cpp mainqtwidget.cpp taskwidget.cpp fractionbasewidget.cpp resultwidget.cpp exercisecompare.cpp ratiowidget.cpp rationalwidget.cpp exerciseconvert.cpp exercisefactorize.cpp factorizedwidget.cpp settingsclass.kcfgc
+kbruch_la_LIBADD  =  $(LIB_KDEUI)
 # the library search path.
-kbruch_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kbruch_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 EXTRA_DIST = kbruch.cpp kbruch.h kbruch.desktop hi32-app-kbruch.png hi16-app-kbruch.png lo32-app-kbruch.png lo16-app-kbruch.png mainqtwidget.cpp mainqtwidget.h primenumber.cpp primenumber.h statisticsview.cpp statisticsview.h exercisebase.cpp exercisebase.h taskview.cpp taskview.h ratio.cpp ratio.h task.cpp task.h taskwidget.cpp taskwidget.h fractionbasewidget.cpp fractionbasewidget.h resultwidget.cpp resultwidget.h exercisecompare.cpp exercisecompare.h ratiowidget.cpp ratiowidget.h rationalwidget.cpp rationalwidget.h exerciseconvert.cpp exerciseconvert.h exercisefactorize.cpp exercisefactorize.h factorizedwidget.cpp factorizedwidget.h kbruch.rc taskvieweroptionsbase.ui
 
diff -uNr kdeedu-3.4.0/kbruch/src/kbruch.cpp kdeedu-3.4.0-new/kbruch/src/kbruch.cpp
--- kdeedu-3.4.0/kbruch/src/kbruch.cpp	2004-07-16 12:12:01.000000000 -0400
+++ kdeedu-3.4.0-new/kbruch/src/kbruch.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -28,7 +28,7 @@
 #include <kaboutdata.h>
 
 /* the main program */
-int main(int argc, char * argv[])
+extern "C" int kdemain(int argc, char * argv[])
 {
 	// init random generator
 	srand(time(NULL));
diff -uNr kdeedu-3.4.0/keduca/keduca/Makefile.am kdeedu-3.4.0-new/keduca/keduca/Makefile.am
--- kdeedu-3.4.0/keduca/keduca/Makefile.am	2004-12-02 04:37:34.000000000 -0500
+++ kdeedu-3.4.0-new/keduca/keduca/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,11 +1,13 @@
 # set the include path for X, qt and KDE
-INCLUDES= -I../libkeduca $(all_includes)
+INCLUDES= -I$(srcdir)/../libkeduca $(all_includes)
 
 # The binary program keduca
-bin_PROGRAMS   = keduca
-keduca_LDFLAGS = $(all_libraries)
-keduca_LDADD   = $(LIB_KFILE) $(LIB_KPARTS) ../libkeduca/libkeduca.la
-keduca_SOURCES = main.cpp keduca.cpp 
+bin_PROGRAMS   = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = keduca.la
+keduca_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+keduca_la_LIBADD  = $(LIB_KFILE) $(LIB_KPARTS) ../libkeduca/libkeduca.la
+keduca_la_SOURCES = main.cpp keduca.cpp 
 
 # Convenience library
 #noinst_LTLIBRARIES = libkeduca_common.la
diff -uNr kdeedu-3.4.0/keduca/keduca/main.cpp kdeedu-3.4.0-new/keduca/keduca/main.cpp
--- kdeedu-3.4.0/keduca/keduca/main.cpp	2005-02-23 04:56:02.000000000 -0500
+++ kdeedu-3.4.0-new/keduca/keduca/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -31,7 +31,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
     KAboutData aboutData( "keduca", I18N_NOOP("KEduca"),
                           version, description, KAboutData::License_GPL,
diff -uNr kdeedu-3.4.0/keduca/keducabuilder/Makefile.am kdeedu-3.4.0-new/keduca/keducabuilder/Makefile.am
--- kdeedu-3.4.0/keduca/keducabuilder/Makefile.am	2004-12-02 04:37:35.000000000 -0500
+++ kdeedu-3.4.0-new/keduca/keducabuilder/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,6 +1,10 @@
-bin_PROGRAMS = keducabuilder
-keducabuilder_SOURCES = kcontroladdeditbase.ui keducaeditorstartdialogbase.ui keducaeditorstartdialog.cpp ktagcombobox.cpp klangcombo.cpp kcontrolheader.cpp keducabuilder.cpp kcontroladdedit.cpp  main.cpp
-keducabuilder_LDADD   =  $(LIB_KFILE) ../libkeduca/libkeduca.la 
+
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = keducabuilder.la
+
+keducabuilder_la_SOURCES = kcontroladdeditbase.ui keducaeditorstartdialogbase.ui keducaeditorstartdialog.cpp ktagcombobox.cpp klangcombo.cpp kcontrolheader.cpp keducabuilder.cpp kcontroladdedit.cpp  main.cpp
+keducabuilder_la_LIBADD  =  $(LIB_KFILE) ../libkeduca/libkeduca.la 
 
 EXTRA_DIST = main.cpp kcontroladdedit.cpp kcontroladdedit.h keducabuilder.cpp keducabuilder.h kcontrolheader.cpp kcontrolheader.h keducabuilderui.rc klangcombo.cpp klangcombo.h ktagcombobox.cpp ktagcombobox.h keducaeditorstartdialog.cpp keducaeditorstartdialog.h 
 
@@ -18,12 +22,12 @@
 # kde_wallpaperdir  Where general wallpapers should go to.
 
 # set the include path for X, qt and KDE
-INCLUDES= -I../libkeduca $(all_includes)
+INCLUDES= -I$(srcdir)/../libkeduca $(all_includes)
 
 METASOURCES = AUTO
 
 # the library search path. 
-keducabuilder_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+keducabuilder_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 # Uncomment the following two lines if you add a ui.rc file for your application to make use of
 # KDE´s XML GUI builing
diff -uNr kdeedu-3.4.0/keduca/keducabuilder/main.cpp kdeedu-3.4.0-new/keduca/keducabuilder/main.cpp
--- kdeedu-3.4.0/keduca/keducabuilder/main.cpp	2005-02-23 04:56:02.000000000 -0500
+++ kdeedu-3.4.0-new/keduca/keducabuilder/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -32,7 +32,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
     KAboutData aboutData( "keduca", I18N_NOOP("KEducaBuilder"),
                           version, description, KAboutData::License_GPL,
diff -uNr kdeedu-3.4.0/khangman/khangman/Makefile.am kdeedu-3.4.0-new/khangman/khangman/Makefile.am
--- kdeedu-3.4.0/khangman/khangman/Makefile.am	2004-10-06 22:04:29.000000000 -0400
+++ kdeedu-3.4.0-new/khangman/khangman/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,7 +1,10 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
-bin_PROGRAMS = khangman
-khangman_SOURCES = mainw.ui khangman.cpp main.cpp khangmanview.cpp prefs.kcfgc khnewstuff.cpp normal.ui advanced.ui
-khangman_LDADD   =  $(LIB_KDEUI) $(top_builddir)/libkdeedu/kdeeducore/libkdeeducore.la -lknewstuff
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = khangman.la
+
+khangman_la_SOURCES = mainw.ui khangman.cpp main.cpp khangmanview.cpp prefs.kcfgc khnewstuff.cpp normal.ui advanced.ui
+khangman_la_LIBADD  =  $(LIB_KDEUI) $(top_builddir)/libkdeedu/kdeeducore/libkdeeducore.la -lknewstuff
 
 SUBDIRS = data pics sounds icons
 
@@ -17,7 +20,7 @@
 METASOURCES = AUTO
 
 # the library search path.
-khangman_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+khangman_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 rc_DATA = khangmanui.rc
 rcdir = $(kde_datadir)/khangman
diff -uNr kdeedu-3.4.0/khangman/khangman/main.cpp kdeedu-3.4.0-new/khangman/khangman/main.cpp
--- kdeedu-3.4.0/khangman/khangman/main.cpp	2005-02-23 04:56:02.000000000 -0500
+++ kdeedu-3.4.0-new/khangman/khangman/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -31,7 +31,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
     KAboutData aboutData( "khangman", I18N_NOOP("KHangMan"),
                               KHM_VERSION, description, KAboutData::License_GPL,
diff -uNr kdeedu-3.4.0/kig/kig/Makefile.am kdeedu-3.4.0-new/kig/kig/Makefile.am
--- kdeedu-3.4.0/kig/kig/Makefile.am	2004-06-19 17:26:27.000000000 -0400
+++ kdeedu-3.4.0-new/kig/kig/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -19,12 +19,14 @@
 #########################################################################
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-bin_PROGRAMS = kig
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kig.la
 
 # the application source, library search path, and link libraries
-kig_SOURCES = main.cpp kig.cpp
-kig_LDFLAGS = $(KDE_RPATH) $(all_libraries)
-kig_LDADD   = $(LIB_KPARTS)
+kig_la_SOURCES = main.cpp kig.cpp
+kig_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
+kig_la_LIBADD  = $(LIB_KPARTS)
 
 # this is where the desktop file will go 
 xdg_apps_DATA = kig.desktop
diff -uNr kdeedu-3.4.0/kig/kig/main.cpp kdeedu-3.4.0-new/kig/kig/main.cpp
--- kdeedu-3.4.0/kig/kig/main.cpp	2004-06-26 14:10:31.000000000 -0400
+++ kdeedu-3.4.0-new/kig/kig/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -101,7 +101,7 @@
   return (*converterfunction)( file, outfile );
 }
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
   KAboutData *about = kigAboutData( "kig", I18N_NOOP("Kig") );
 
diff -uNr kdeedu-3.4.0/kiten/Makefile.am kdeedu-3.4.0-new/kiten/Makefile.am
--- kdeedu-3.4.0/kiten/Makefile.am	2005-02-03 17:36:23.000000000 -0500
+++ kdeedu-3.4.0-new/kiten/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -2,15 +2,16 @@
 
 SUBDIRS = pics
 
-kiten_SOURCES = main.cpp kiten.cpp learn.cpp kloader.cpp ksaver.cpp \
+kiten_la_SOURCES = main.cpp kiten.cpp learn.cpp kloader.cpp ksaver.cpp \
 	configsearching.ui configlearn.ui configfont.ui configdictionariesbase.ui \
 	configdictionaries.cpp optiondialog.cpp kitenconfig.kcfgc
 
-kiten_LDFLAGS = $(all_libraries) $(KDE_RPATH) 
-kiten_LDADD = libkiten.la $(LIB_KIO)
+kiten_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kiten_la_LIBADD = libkiten.la $(LIB_KIO)
 
-bin_PROGRAMS = kiten kitengen
+bin_PROGRAMS = kitengen
 lib_LTLIBRARIES = libkiten.la
+kdeinit_LTLIBRARIES = kiten.la
 
 libkiten_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined -version-info 1:0:0
 libkiten_la_LIBADD = $(LIB_KDEUI) $(LIB_KDEPRINT)
diff -uNr kdeedu-3.4.0/kiten/main.cpp kdeedu-3.4.0-new/kiten/main.cpp
--- kdeedu-3.4.0/kiten/main.cpp	2005-02-03 17:36:23.000000000 -0500
+++ kdeedu-3.4.0-new/kiten/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -37,7 +37,7 @@
 	// INSERT YOUR COMMANDLINE OPTIONS HERE
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 	qInstallMsgHandler(noMessageOutput);
 	KAboutData aboutData( "kiten", I18N_NOOP("Kiten"),
diff -uNr kdeedu-3.4.0/klatin/klatin/Makefile.am kdeedu-3.4.0-new/klatin/klatin/Makefile.am
--- kdeedu-3.4.0/klatin/klatin/Makefile.am	2004-10-27 20:11:16.000000000 -0400
+++ kdeedu-3.4.0-new/klatin/klatin/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -6,19 +6,21 @@
 
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-bin_PROGRAMS = klatin
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = klatin.la
 
 # set the include path for X, qt and KDE
 INCLUDES         = $(all_includes)
 
 # the library search path.
-klatin_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+klatin_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
 
 # the libraries to link against.
-klatin_LDADD   = $(LIB_KFILE) $(top_builddir)/libkdeedu/kdeeducore/libkdeeducore.la
+klatin_la_LIBADD  = $(LIB_KFILE) $(top_builddir)/libkdeedu/kdeeducore/libkdeeducore.la
 
 # which sources should be compiled for klatin
-klatin_SOURCES =	main.cpp klatin.cpp klatingrammar.cpp klatinverbs.cpp klatinvocab.cpp \
+klatin_la_SOURCES =	main.cpp klatin.cpp klatingrammar.cpp klatinverbs.cpp klatinvocab.cpp \
 	klatinchoose.ui settings.kcfgc klatingrammarwidget.ui klatinvocabwidget.ui klatinverbswidget.ui \
 	klatinresultsdialog.ui klatinsettings_vocabpage.ui
 
diff -uNr kdeedu-3.4.0/klatin/klatin/main.cpp kdeedu-3.4.0-new/klatin/klatin/main.cpp
--- kdeedu-3.4.0/klatin/klatin/main.cpp	2005-02-23 04:56:02.000000000 -0500
+++ kdeedu-3.4.0-new/klatin/klatin/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -29,7 +29,7 @@
 	KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 	KAboutData aboutData( "klatin", I18N_NOOP("KLatin"),
diff -uNr kdeedu-3.4.0/klettres/klettres/Makefile.am kdeedu-3.4.0-new/klettres/klettres/Makefile.am
--- kdeedu-3.4.0/klettres/klettres/Makefile.am	2005-02-03 17:36:24.000000000 -0500
+++ kdeedu-3.4.0-new/klettres/klettres/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,10 +1,12 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
-bin_PROGRAMS = klettres
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = klettres.la
 
 ## INCLUDES were found outside kdevelop specific part
 
-klettres_SOURCES = fontsdlg.ui klettres.cpp klnewstuff.cpp main.cpp klettresview.cpp soundfactory.cpp prefs.kcfgc timerdlg.ui
-klettres_LDADD   = $(LIB_KDEUI) -lknewstuff
+klettres_la_SOURCES = fontsdlg.ui klettres.cpp klnewstuff.cpp main.cpp klettresview.cpp soundfactory.cpp prefs.kcfgc timerdlg.ui
+klettres_la_LIBADD  = $(LIB_KDEUI) -lknewstuff
 
 SUBDIRS = pics data icons
 
@@ -24,7 +26,7 @@
 KDE_ICON = AUTO
 
 # the library search path.
-klettres_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+klettres_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 messages: rc.cpp
 	$(XGETTEXT) *.cpp -o $(podir)/klettres.pot;
diff -uNr kdeedu-3.4.0/klettres/klettres/main.cpp kdeedu-3.4.0-new/klettres/klettres/main.cpp
--- kdeedu-3.4.0/klettres/klettres/main.cpp	2005-02-23 04:56:03.000000000 -0500
+++ kdeedu-3.4.0-new/klettres/klettres/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -25,7 +25,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
     KAboutData about("klettres", I18N_NOOP("KLettres"), KLETTRES_VERSION, description,
                      KAboutData::License_GPL, "(C) 2001-2003 Anne-Marie Mahfouf",0, "http://edu.kde.org/klettres", "submit@bugs.kde.org");
diff -uNr kdeedu-3.4.0/kmathtool/kmathtool/Makefile.am kdeedu-3.4.0-new/kmathtool/kmathtool/Makefile.am
--- kdeedu-3.4.0/kmathtool/kmathtool/Makefile.am	2003-08-29 15:59:44.000000000 -0400
+++ kdeedu-3.4.0-new/kmathtool/kmathtool/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,7 +1,9 @@
 ## Makefile.am for kmathtool
 
 # This is the program that gets installed.
-bin_PROGRAMS = kmathtool
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kmathtool.la
 
 # Set the include path for X, qt and KDE
 AM_CPPFLAGS = -I$(srcdir)/../libmodule $(all_includes)
@@ -12,13 +14,13 @@
   kmtmoduleiface.h version.h
 
 # Which sources should be compiled for kmathtool
-kmathtool_SOURCES = \
+kmathtool_la_SOURCES = \
   kmathtool.cpp kmathtoolview.cpp kmttreeview.cpp kmtmessagewidget.cpp \
   kmtmoduleiface.cpp main.cpp
 
 # The libraries to link against.
-kmathtool_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kmathtool_LDADD   = ../libmodule/libkmathtoolmodule.la $(LIB_KHTML) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
+kmathtool_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kmathtool_la_LIBADD  = ../libmodule/libkmathtoolmodule.la $(LIB_KHTML) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
 
 # Let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
diff -uNr kdeedu-3.4.0/kmessedwords/kmessedwords/Makefile.am kdeedu-3.4.0-new/kmessedwords/kmessedwords/Makefile.am
--- kdeedu-3.4.0/kmessedwords/kmessedwords/Makefile.am	2005-01-03 09:52:50.000000000 -0500
+++ kdeedu-3.4.0-new/kmessedwords/kmessedwords/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,9 +1,12 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
-bin_PROGRAMS = kmessedwords
-kmessedwords_SOURCES = minparser.cpp answer.cpp mainwindow.cpp frontend.ui \
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kmessedwords.la
+
+kmessedwords_la_SOURCES = minparser.cpp answer.cpp mainwindow.cpp frontend.ui \
 	kmwdict.ui scores.cpp kwritewords.cpp main.cpp kmwview.cpp prefs.kcfgc \
 	interface.ui game.ui
-kmessedwords_LDADD   =  $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+kmessedwords_la_LIBADD  =  $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
 
 SUBDIRS = pics 
 
@@ -24,7 +27,7 @@
 METASOURCES = AUTO
 
 # the library search path. 
-kmessedwords_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kmessedwords_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 stringstocpp:
 	rm -f dictionary_i18n.cpp; \
diff -uNr kdeedu-3.4.0/kmessedwords/kmessedwords/main.cpp kdeedu-3.4.0-new/kmessedwords/kmessedwords/main.cpp
--- kdeedu-3.4.0/kmessedwords/kmessedwords/main.cpp	2005-02-03 17:36:24.000000000 -0500
+++ kdeedu-3.4.0-new/kmessedwords/kmessedwords/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -30,7 +30,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
   KAboutData aboutData( "kmessedwords", I18N_NOOP("KMessedWords"),
diff -uNr kdeedu-3.4.0/kmplot/kmplot/Makefile.am kdeedu-3.4.0-new/kmplot/kmplot/Makefile.am
--- kdeedu-3.4.0/kmplot/kmplot/Makefile.am	2004-10-29 08:29:08.000000000 -0400
+++ kdeedu-3.4.0-new/kmplot/kmplot/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -21,12 +21,14 @@
 #########################################################################
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-bin_PROGRAMS = kmplot
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kmplot.la
 
 # the application source, library search path, and link libraries
-kmplot_SOURCES = main.cpp kmplot.cpp kmplotprogress.cpp kmplotIface.skel
-kmplot_LDFLAGS = $(KDE_RPATH) $(all_libraries)
-kmplot_LDADD   = $(LIB_KPARTS)
+kmplot_la_SOURCES = main.cpp kmplot.cpp kmplotprogress.cpp kmplotIface.skel
+kmplot_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
+kmplot_la_LIBADD  = $(LIB_KPARTS)
 
 # this is where the desktop file will go
 appmimedir = $(kde_mimedir)/application
diff -uNr kdeedu-3.4.0/kmplot/kmplot/main.cpp kdeedu-3.4.0-new/kmplot/kmplot/main.cpp
--- kdeedu-3.4.0/kmplot/kmplot/main.cpp	2005-01-03 09:52:52.000000000 -0500
+++ kdeedu-3.4.0-new/kmplot/kmplot/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -50,7 +50,7 @@
     };
 
 
-int main( int argc, char **argv )
+extern "C" int kdemain( int argc, char **argv )
 {
 	KAboutData aboutData(
 	    "kmplot",
diff -uNr kdeedu-3.4.0/kpercentage/kpercentage/Makefile.am kdeedu-3.4.0-new/kpercentage/kpercentage/Makefile.am
--- kdeedu-3.4.0/kpercentage/kpercentage/Makefile.am	2003-08-29 15:59:44.000000000 -0400
+++ kdeedu-3.4.0-new/kpercentage/kpercentage/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,6 +1,10 @@
-bin_PROGRAMS = kpercentage
-kpercentage_SOURCES =  kanimation.cpp kanswer.cpp kpercentmain.cpp  kpercentage.cpp ksplashscreen.cpp main.cpp
-kpercentage_LDADD   =  $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kpercentage.la
+
+kpercentage_la_SOURCES =  kanimation.cpp kanswer.cpp kpercentmain.cpp  kpercentage.cpp ksplashscreen.cpp main.cpp
+kpercentage_la_LIBADD  =  $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
 
 SUBDIRS = icons pics
 
@@ -30,7 +34,7 @@
 
 
 # the library search path.
-kpercentage_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kpercentage_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 
 stringstocpp:
diff -uNr kdeedu-3.4.0/kpercentage/kpercentage/main.cpp kdeedu-3.4.0-new/kpercentage/kpercentage/main.cpp
--- kdeedu-3.4.0/kpercentage/kpercentage/main.cpp	2005-02-23 04:56:03.000000000 -0500
+++ kdeedu-3.4.0-new/kpercentage/kpercentage/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -34,7 +34,7 @@
         KCmdLineLastOption
     };
 
-int main( int argc, char *argv[] )
+extern "C" int kdemain( int argc, char *argv[] )
 {
     KAboutData aboutData
     (
diff -uNr kdeedu-3.4.0/kstars/kstars/Makefile.am kdeedu-3.4.0-new/kstars/kstars/Makefile.am
--- kdeedu-3.4.0/kstars/kstars/Makefile.am	2005-02-23 04:56:04.000000000 -0500
+++ kdeedu-3.4.0-new/kstars/kstars/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,6 +1,9 @@
-bin_PROGRAMS = kstars
 
-kstars_SOURCES = addcatdialog.cpp addlinkdialog.cpp colorscheme.cpp conbridlg.cpp \
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kstars.la
+
+kstars_la_SOURCES = addcatdialog.cpp addlinkdialog.cpp colorscheme.cpp conbridlg.cpp \
 	csegment.cpp deepskyobject.cpp detaildialog.cpp devicemanager.cpp dms.cpp \
 	dmsbox.cpp filesource.cpp finddialog.cpp fitshistogram.cpp fitsimage.cpp \
 	fitsprocess.cpp fitsviewer.cpp focusdialog.cpp fov.cpp fovdialog.cpp \
@@ -29,11 +32,11 @@
 kde_kcfg_DATA = kstars.kcfg
 kde_conf_DATA = kstarsrc
 
-kstars_LDADD = indi/liblilxml.a indi/libindicom.a tools/libkstarstools.a $(top_builddir)/libkdeedu/kdeeduplot/libkdeeduplot.la $(top_builddir)/libkdeedu/extdate/libextdate.la -lkdeprint -lknewstuff -lm -lz
+kstars_la_LIBADD = indi/liblilxml.a indi/libindicom.a tools/libkstarstools.a $(top_builddir)/libkdeedu/kdeeduplot/libkdeeduplot.la $(top_builddir)/libkdeedu/extdate/libextdate.la -lkdeprint -lknewstuff -lm -lz
 
 SUBDIRS = indi tools . data icons 
 
-kstars_COMPILE_FIRST = Options.kcfgc
+kstars_la_COMPILE_FIRST = Options.kcfgc
 
 EXTRA_DIST = kstars.desktop kstarsui.rc fitsviewer.rc
 
@@ -53,7 +56,7 @@
 METASOURCES = AUTO
 
 # the library search path.
-kstars_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kstars_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 # Uncomment the following two lines if you add a ui.rc file for your application to make use of
 # KDE
diff -uNr kdeedu-3.4.0/kstars/kstars/indi/fli_ccd.c kdeedu-3.4.0-new/kstars/kstars/indi/fli_ccd.c
--- kdeedu-3.4.0/kstars/kstars/indi/fli_ccd.c	2005-02-23 04:56:05.000000000 -0500
+++ kdeedu-3.4.0-new/kstars/kstars/indi/fli_ccd.c	2005-03-15 09:40:35.000000000 -0500
@@ -19,6 +19,7 @@
 
 #endif
 
+#include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -uNr kdeedu-3.4.0/kstars/kstars/indi/lx200generic.cpp kdeedu-3.4.0-new/kstars/kstars/indi/lx200generic.cpp
--- kdeedu-3.4.0/kstars/kstars/indi/lx200generic.cpp	2005-02-23 04:56:05.000000000 -0500
+++ kdeedu-3.4.0-new/kstars/kstars/indi/lx200generic.cpp	2005-03-15 09:41:16.000000000 -0500
@@ -41,7 +41,7 @@
 
 
 #if !(TIMEZONE_IS_INT)
-static int daylight = 0;
+int daylight = 0;
 #endif
 
 static inline double timezoneOffset()
diff -uNr kdeedu-3.4.0/kstars/kstars/main.cpp kdeedu-3.4.0-new/kstars/kstars/main.cpp
--- kdeedu-3.4.0/kstars/kstars/main.cpp	2005-02-23 04:56:05.000000000 -0500
+++ kdeedu-3.4.0-new/kstars/kstars/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -49,7 +49,7 @@
 	KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 	KAboutData aboutData( "kstars", I18N_NOOP("KStars"),
 		KSTARS_VERSION, description, KAboutData::License_GPL,
diff -uNr kdeedu-3.4.0/ktouch/src/Makefile.am kdeedu-3.4.0-new/ktouch/src/Makefile.am
--- kdeedu-3.4.0/ktouch/src/Makefile.am	2004-11-01 22:54:33.000000000 -0500
+++ kdeedu-3.4.0-new/ktouch/src/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -24,7 +24,7 @@
 	ktouchstatus.h ktouchtrainer.h ktouchtrainingsession.h
 
 # let automoc handle all of the meta source files (moc)
-ktouch_METASOURCES = AUTO
+ktouch_la_METASOURCES = AUTO
 
 messages: rc.cpp
 	$(XGETTEXT) *.cpp -o $(podir)/ktouch.pot
@@ -35,10 +35,12 @@
 #########################################################################
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-bin_PROGRAMS = ktouch
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = ktouch.la
 
 # the application source, library search path, and link libraries
-ktouch_SOURCES = ktouch.cpp ktouchcharstats.cpp ktouchchartwidget.cpp \
+ktouch_la_SOURCES = ktouch.cpp ktouchcharstats.cpp ktouchchartwidget.cpp \
 	ktouchkeyboard.cpp ktouchkeys.cpp ktouchlecture.cpp ktouchleveldata.cpp \
 	ktouchslideline.cpp ktouchstatistics.cpp ktouchstatus.cpp ktouchtrainer.cpp \
 	ktouchtrainingsession.cpp main.cpp ktouchprefcolorslayout.ui ktouchprefgenerallayout.ui \
@@ -48,11 +50,11 @@
 	ktouchcolorscheme.cpp ktouchutils.cpp prefs.kcfgc 
 
 # the library search path.
-ktouch_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+ktouch_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
 
 # the libraries to link against. Be aware of the order. First the libraries,
 # that depend on the following ones.
-ktouch_LDADD = $(LIB_KIO)
+ktouch_la_LIBADD = $(LIB_KIO)
 
 rcdir = $(kde_datadir)/ktouch
 rc_DATA = ktouchui.rc
diff -uNr kdeedu-3.4.0/ktouch/src/main.cpp kdeedu-3.4.0-new/ktouch/src/main.cpp
--- kdeedu-3.4.0/ktouch/src/main.cpp	2004-11-08 19:47:29.000000000 -0500
+++ kdeedu-3.4.0-new/ktouch/src/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -26,7 +26,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
     KAboutData about("ktouch",
                      I18N_NOOP("KTouch"),
diff -uNr kdeedu-3.4.0/kturtle/src/Makefile.am kdeedu-3.4.0-new/kturtle/src/Makefile.am
--- kdeedu-3.4.0/kturtle/src/Makefile.am	2004-10-22 10:38:07.000000000 -0400
+++ kdeedu-3.4.0-new/kturtle/src/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -6,20 +6,22 @@
 # added from kjots
 AM_LDFLAGS   = $(all_libraries) $(KDE_RPATH)
 
-kturtle_LDFLAGS = $(KDE_RPATH) $(all_libraries)
-kturtle_LDADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_KDEPRINT) $(LIB_KPARTS) -lkatepartinterfaces
+kturtle_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
+kturtle_la_LIBADD = $(LIB_KDEUI) $(LIB_KIO) $(LIB_KDEPRINT) $(LIB_KPARTS) -lkatepartinterfaces
 
 KDE_ICON = kturtle
 
 # this is the program that gets installed.  it's name is used for all
 # of the other Makefile.am variables
-bin_PROGRAMS = kturtle
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kturtle.la
 
 # the application source, library search path, and link libraries
-kturtle_SOURCES = main.cpp kturtle.cpp canvas.cpp dialogs.cpp value.cpp lexer.cpp parser.cpp executer.cpp treenode.cpp token.cpp translate.cpp settings.kcfgc
+kturtle_la_SOURCES = main.cpp kturtle.cpp canvas.cpp dialogs.cpp value.cpp lexer.cpp parser.cpp executer.cpp treenode.cpp token.cpp translate.cpp settings.kcfgc
 
 # let automoc handle all the meta source files (moc)
-kturtle_METASOURCES = AUTO
+kturtle_la_METASOURCES = AUTO
 
 # these are the headers for your project
 noinst_HEADERS = kturtle.h canvas.h dialogs.cpp value.h lexer.h parser.h executer.h treenode.h token.h translate.h
diff -uNr kdeedu-3.4.0/kturtle/src/main.cpp kdeedu-3.4.0-new/kturtle/src/main.cpp
--- kdeedu-3.4.0/kturtle/src/main.cpp	2005-02-03 17:36:26.000000000 -0500
+++ kdeedu-3.4.0-new/kturtle/src/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -43,7 +43,7 @@
 };
 
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
 	KAboutData about("kturtle", I18N_NOOP("KTurtle"), version, description, KAboutData::License_GPL, copyright, 0, website);
 	about.addAuthor("Cies Breijs",
diff -uNr kdeedu-3.4.0/kverbos/kverbos/Makefile.am kdeedu-3.4.0-new/kverbos/kverbos/Makefile.am
--- kdeedu-3.4.0/kverbos/kverbos/Makefile.am	2004-07-03 23:42:45.000000000 -0400
+++ kdeedu-3.4.0-new/kverbos/kverbos/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,8 +1,11 @@
 SUBDIRS = icons data
 
-bin_PROGRAMS = kverbos
-kverbos_SOURCES = kfeedercontrol.cpp kresult.cpp qresult.ui verbspanish.cpp kverbosuser.cpp kerfassen.cpp qerfassen.ui kverbedit.cpp qverbedit.ui qverbosoptions.ui kverbosoptions.cpp qlernen.ui kstartuplogo.cpp kverbosview.cpp kverbosdoc.cpp kverbos.cpp main.cpp prefs.kcfgc
-kverbos_LDADD   =  $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kverbos.la
+
+kverbos_la_SOURCES = kfeedercontrol.cpp kresult.cpp qresult.ui verbspanish.cpp kverbosuser.cpp kerfassen.cpp qerfassen.ui kverbedit.cpp qverbedit.ui qverbosoptions.ui kverbosoptions.cpp qlernen.ui kstartuplogo.cpp kverbosview.cpp kverbosdoc.cpp kverbos.cpp main.cpp prefs.kcfgc
+kverbos_la_LIBADD  =  $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
 
 
 EXTRA_DIST = kfeedercontrol.cpp kfeedercontrol.h main.cpp kverbos.cpp kverbos.h kverbosdoc.cpp kverbosdoc.h kverbosview.cpp kverbosview.h kverbosui.rc kverbos.desktop kstartuplogo.cpp kstartuplogo.h startuplogo.png qlernen.ui kverbosoptions.cpp kverbosoptions.h qverbosoptions.ui qverbedit.ui kverbedit.cpp kverbedit.h qerfassen.ui kerfassen.cpp kerfassen.h deu16.png span16.png kverbosuser.cpp kverbosuser.h verbspanish.cpp verbspanish.h qresult.ui kverbosuser.png kresult.cpp kresult.h
@@ -19,7 +22,7 @@
 METASOURCES = AUTO
 
 # the library search path. 
-kverbos_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kverbos_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 rcdir = $(kde_datadir)/kverbos
 rc_DATA = kverbosui.rc
diff -uNr kdeedu-3.4.0/kverbos/kverbos/main.cpp kdeedu-3.4.0-new/kverbos/kverbos/main.cpp
--- kdeedu-3.4.0/kverbos/kverbos/main.cpp	2004-06-26 14:10:36.000000000 -0400
+++ kdeedu-3.4.0-new/kverbos/kverbos/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -32,7 +32,7 @@
 
 
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 	KAboutData aboutData( "kverbos", I18N_NOOP("KVerbos"),
diff -uNr kdeedu-3.4.0/kwordquiz/src/Makefile.am kdeedu-3.4.0-new/kwordquiz/src/Makefile.am
--- kdeedu-3.4.0/kwordquiz/src/Makefile.am	2004-11-22 21:45:22.000000000 -0500
+++ kdeedu-3.4.0-new/kwordquiz/src/Makefile.am	2005-03-15 09:40:35.000000000 -0500
@@ -1,9 +1,11 @@
 ####### kdevelop will overwrite this part!!! (begin)##########
-bin_PROGRAMS = kwordquiz
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kwordquiz.la
 
 ## INCLUDES were found outside kdevelop specific part
 
-kwordquiz_SOURCES = kwordquizprefs.cpp kwordquizview.cpp kwordquizdoc.cpp \
+kwordquiz_la_SOURCES = kwordquizprefs.cpp kwordquizview.cpp kwordquizdoc.cpp \
 		kwordquiz.cpp main.cpp dlglanguagebase.ui flashviewbase.ui multipleviewbase.ui \
 		qaviewbase.ui qaview.cpp flashview.cpp multipleview.cpp wqquiz.cpp wqlistitem.cpp \
 		wqscore.cpp keduvocdata.cpp prefeditorbase.ui prefquizbase.ui prefeditor.cpp \
@@ -11,7 +13,7 @@
 		kvtmlwriter.cpp dlglanguage.cpp wqprintdialogpage.cpp prefcharacter.cpp \
 		prefcharacterbase.ui paukerreader.cpp wqlreader.cpp wqlwriter.cpp wqundo.cpp prefs.kcfgc \
 	prefcardappearance.cpp prefcardappearancebase.ui
-kwordquiz_LDADD   =  $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE)  $(LIBSOCKET) $(LIB_KDEPRINT)
+kwordquiz_la_LIBADD  =  $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE)  $(LIBSOCKET) $(LIB_KDEPRINT)
 
 
 EXTRA_DIST = main.cpp kwordquiz.cpp kwordquiz.h kwordquizdoc.cpp kwordquizdoc.h kwordquizview.cpp kwordquizview.h kwordquizui.rc kwordquiz.desktop kwordquizprefs.cpp kwordquizprefs.h
@@ -49,7 +51,7 @@
 METASOURCES = AUTO
 
 # the library search path. 
-kwordquiz_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kwordquiz_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 rcdir = $(kde_datadir)/kwordquiz
 rc_DATA = kwordquizui.rc eventsrc
diff -uNr kdeedu-3.4.0/kwordquiz/src/main.cpp kdeedu-3.4.0-new/kwordquiz/src/main.cpp
--- kdeedu-3.4.0/kwordquiz/src/main.cpp	2005-02-03 17:36:28.000000000 -0500
+++ kdeedu-3.4.0-new/kwordquiz/src/main.cpp	2005-03-15 09:40:35.000000000 -0500
@@ -33,7 +33,7 @@
   { 0, 0, 0 }
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
   KAboutData aboutData( "kwordquiz", I18N_NOOP("KWordQuiz"), KWQ_VERSION, description, KAboutData::License_GPL, "(c) 2003-2005, Peter Hedlund", 0, 0, "peter@peterandlinda.com");
   aboutData.addAuthor("Peter Hedlund", 0, "peter@peterandlinda.com");
