--- ClamAV.pm.orig	2009-04-29 16:02:43.000000000 -0700
+++ ClamAV.pm	2011-11-04 16:49:01.000000000 -0700
@@ -191,7 +191,7 @@
     struct cl_stat st;
     char is_dir;
     char *path;
-    int signatures;
+    unsigned int signatures;
 };
 
 SV *clamav_perl_new(char *class, char *path)
@@ -286,7 +286,7 @@
     reclevel = cl_engine_get_num(c->root, CL_ENGINE_MAX_RECURSION, &err);
     if (err != CL_SUCCESS) {
         error(err);
-        return;
+        return -1;
     }
     return reclevel;
 }
@@ -313,7 +313,7 @@
     maxfiles = cl_engine_get_num(c->root, CL_ENGINE_MAX_FILES, &err);
     if (err != CL_SUCCESS) {
         error(err);
-        return;
+        return -1;
     }
     return maxfiles;
 }
@@ -334,7 +334,7 @@
     maxfsize = cl_engine_get_num(c->root, CL_ENGINE_MAX_FILESIZE, &err);
     if (err != CL_SUCCESS) {
         error(err);
-        return;
+        return -1;
     }
     return maxfsize;
 }
