--- plugins/classbrowser/documentclassesfolder.h.orig	2015-04-29 01:16:28.000000000 +0900
+++ plugins/classbrowser/documentclassesfolder.h	2015-04-29 01:16:54.000000000 +0900
@@ -23,9 +23,11 @@
 #define KDEVPLATFORM_PLUGIN_DOCUMENTCLASSESFOLDER_H
 
 #include "classmodelnode.h"
+#ifndef Q_MOC_RUN
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/member.hpp>
 #include <boost/multi_index/ordered_index.hpp>
+#endif
 
 namespace ClassModelNodes
 {
