diff -ur ../kdevplatform-1.6.0-orig/language/duchain/duchain.cpp ./language/duchain/duchain.cpp
--- ../kdevplatform-1.6.0-orig/language/duchain/duchain.cpp	2013-12-08 03:45:38.000000000 +0900
+++ ./language/duchain/duchain.cpp	2013-12-11 11:54:50.000000000 +0900
@@ -42,7 +42,7 @@
 #include <interfaces/foregroundlock.h>
 #include <interfaces/isession.h>
 
-#include <util/google/dense_hash_map>
+#include <google/dense_hash_map>
 
 #include "../interfaces/ilanguagesupport.h"
 #include "../interfaces/icodehighlighting.h"
diff -ur ../kdevplatform-1.6.0-orig/language/duchain/ducontextdynamicdata.h ./language/duchain/ducontextdynamicdata.h
--- ../kdevplatform-1.6.0-orig/language/duchain/ducontextdynamicdata.h	2013-12-08 03:45:38.000000000 +0900
+++ ./language/duchain/ducontextdynamicdata.h	2013-12-11 11:51:39.000000000 +0900
@@ -23,7 +23,7 @@
 #define DUCONTEXTDYNAMICDATA_H
 
 #include "ducontextdata.h"
-#include <util/google/dense_hash_map>
+#include <google/dense_hash_map>
 
 namespace KDevelop {
 
