--- plugins/CMakeLists.txt.orig	2012-04-14 04:49:31.000000000 +0900
+++ plugins/CMakeLists.txt	2012-07-14 20:52:42.000000000 +0900
@@ -32,8 +32,9 @@
     macro_optional_add_subdirectory(classbrowser)
 endif(Boost_FOUND)
 
-find_package(SubversionLibrary)
-macro_log_feature( SUBVERSION_FOUND "Subversion" "Support for Subversion integration" "http://subversion.tigris.org" FALSE "1.3.0" "The subversion libraries are needed for the Subversion support" )
+#find_package(SubversionLibrary)
+#macro_log_feature( SUBVERSION_FOUND "Subversion" "Support for Subversion integration" "http://subversion.tigris.org" FALSE "1.3.0" "The #subversion libraries are needed for the Subversion support" )
+SET(SUBVERSION_FOUND false)
 
 if(SUBVERSION_FOUND)
     macro_optional_add_subdirectory(subversion)
