--- plugins/standardoutputview/outputwidget.cpp.orig	2014-10-17 22:25:08.000000000 +0900
+++ plugins/standardoutputview/outputwidget.cpp	2014-10-17 22:25:12.000000000 +0900
@@ -422,7 +422,7 @@
         ? iface->previousHighlightIndex( index )
         : iface->nextHighlightIndex( index );
 
-    kDebug() << "selecting item" << select << index << newIndex;
+//    kDebug() << "selecting item" << select << index << newIndex;
     activateIndex(newIndex, view, iface);
 }
 
