--- interpreter.cc.orig	2012-03-17 18:35:43.000000000 -0500
+++ interpreter.cc	2012-03-22 18:48:39.000000000 -0500
@@ -10552,8 +10552,8 @@
     // The -disable-cfi seems to be needed on OSX as of LLVM 3.0.
     custom_opts = "-disable-cfi ";
 #endif
-    string cmd = "opt -f -std-compile-opts "+quote(target)+
-      " | llc "+custom_opts+string(pic?"-relocation-model=pic ":"")+
+    string cmd = "@LLVM_PREFIX@/bin/opt -f -std-compile-opts "+quote(target)+
+      " | @LLVM_PREFIX@/bin/llc "+custom_opts+string(pic?"-relocation-model=pic ":"")+
       "-o "+quote(asmfile);
     if (vflag) std::cerr << cmd << '\n';
     unlink(asmfile.c_str());
