--- interpreter.cc.orig	2012-03-05 20:54:23.000000000 -0600
+++ interpreter.cc	2012-03-09 00:42:45.000000000 -0600
@@ -10539,8 +10539,8 @@
     /* Call llc (and opt) to create a native assembler file which can then be
        passed to gcc to handle assembly and linkage (if requested). */
     string asmfile = (ext==".s")?out:out+".s";
-    string cmd = "opt -f -std-compile-opts "+quote(target)+
-      " | llc "+string(pic?"-relocation-model=pic ":"")+
+    string cmd = "@LLVM_PREFIX@/bin/opt -f -std-compile-opts "+quote(target)+
+      " | @LLVM_PREFIX@/bin/llc -disable-cfi "+string(pic?"-relocation-model=pic ":"")+
       "-o "+quote(asmfile);
     if (vflag) std::cerr << cmd << '\n';
     unlink(asmfile.c_str());
