--- Makefile.in.orig	2009-03-30 05:55:34.000000000 -0500
+++ Makefile.in	2009-04-09 16:56:57.000000000 -0500
@@ -87,7 +87,7 @@
 # the runtime ABI changes, and the $(minor) version number is used to indicate
 # bugfix releases.
 
-major = 0
+major = 1
 minor = 0
 
 libpure_name    = $(name)
@@ -175,7 +175,7 @@
 
 ifeq ($(sharedlib), yes)
 pure$(EXE): pure.o $(libpure)
-	$(CXX) -o $@ $(LDFLAGS) pure.o -L. -l$(libpure_name) $(LIBS)
+	$(CXX) -o $@ -L. $(LDFLAGS) pure.o -l$(libpure_name) $(LIBS)
 else
 pure$(EXE): pure.o $(OBJECT)
 	$(CXX) -o $@ $(LDFLAGS) pure.o $(OBJECT) $(LLVM_LIBS) $(LIBS)
@@ -292,7 +292,7 @@
 	$(INSTALL) -m 644 $(srcdir)/pure.html $(srcdir)/purelib.html $(DESTDIR)$(libdir)/pure-$(version)
 	ln -sf $(libdir)/pure-$(version) $(DESTDIR)$(libdir)/pure
 	$(INSTALL) -m 644 $(srcdir)/pure.1 $(DESTDIR)$(man1dir)/pure-$(version).1
-	ln -sf $(man1dir)/pure-$(version).1 $(DESTDIR)$(man1dir)/pure.1
+	ln -sf pure-$(version).1 $(DESTDIR)$(man1dir)/pure.1
 else
 	$(INSTALL) -m 644 $(srcdir)/runtime.h $(DESTDIR)$(includedir)/pure
 	$(INSTALL) -m 644 $(srcdir)/pure_main.c pure_main.o $(DESTDIR)$(libdir)/pure
