--- README.orig	2013-09-25 05:43:09.000000000 -0500
+++ README	2014-03-08 00:18:42.000000000 -0600
@@ -164,8 +164,9 @@
 addon modules and libraries. You can invoke the manpage with 'man pure' after
 installation, and the online manual by using the 'help' command inside the
 interpreter. The former needs the 'man' program, the latter a html browser,
-w3m by default. See the INSTALL file or the manpage for details. Or just point
-your browser at the following URL to read the manual on the web:
+your OS X web browser by default. See the INSTALL file or the manpage for
+details. Or just point your browser at the following URL to read the manual
+on the web:
 
 http://puredocs.bitbucket.org/
 
--- lexer.cc.orig	2013-09-15 01:13:44.000000000 -0500
+++ lexer.cc	2014-03-08 00:18:42.000000000 -0600
@@ -9363,7 +9363,7 @@
       } else
 	cerr << "help: memory allocation error\n";
     } else {
-      if (!browser) browser = "w3m"; // default
+      if (!browser) browser = "openbrowser"; // default
       string helpcmd = string(browser) + " \"" + docname + "\"";
       if (system(helpcmd.c_str()) == -1) perror("system");
     }
--- pure.1.orig	2013-09-15 00:54:21.000000000 -0500
+++ pure.1	2014-03-08 00:18:42.000000000 -0600
@@ -131,9 +131,9 @@
 .PP
 The full Pure manual can be read inside the interpreter, by typing
 .B help
-on the interpreter's command line. This requires
-.BR w3m (1)
-to work. (You can also set the name of another html browser to use with the
+on the interpreter's command line.
+This opens the documentation into your OS X web browser.
+(You can also set the name of another html browser to use with the
 .B PURE_HELP
 or the
 .B BROWSER
@@ -185,7 +185,7 @@
 .TP
 .B PURE_HELP
 Browser to use to read online documentation in the interpreter. By default,
-.BR w3m (1)
+.BR openbrowser (1)
 is used.
 .TP
 .B PURE_INCLUDE
