--- configure.ac.orig	2010-08-09 15:24:50.000000000 -0500
+++ configure.ac	2012-05-13 21:38:53.000000000 -0500
@@ -539,7 +539,7 @@
 dnl #########################################################################
 
 AC_REPLACE_FUNCS(strdup strcasecmp strncasecmp memrchr)
-AC_CHECK_FUNCS(strtol uname gethostname gettimeofday)
+AC_CHECK_FUNCS(strtol uname gethostname gettimeofday snprintf vsnprintf)
 AC_SEARCH_LIBS(socket, socket)
 
 dnl Conditional check for functions needed in threaded code
