--- src/BulletMultiThreaded/GpuSoftBodySolvers/OpenCL/btSoftBodySolverBuffer_OpenCL.h.orig	2010-09-24 15:17:03.000000000 +0400
+++ src/BulletMultiThreaded/GpuSoftBodySolvers/OpenCL/btSoftBodySolverBuffer_OpenCL.h	2010-09-24 15:17:19.000000000 +0400
@@ -22,7 +22,7 @@
 	#include "MiniCL/cl.h"
 #else //USE_MINICL
 	#ifdef __APPLE__
-		#include <OpenCL/OpenCL.h>
+		#include <OpenCL/opencl.h>
 	#else
 		#include <CL/cl.h>
 	#endif //__APPLE__
@@ -191,4 +191,4 @@
 }; // class btOpenCLBuffer
 
 
-#endif // #ifndef BT_SOFT_BODY_SOLVER_BUFFER_OPENCL_H
\ No newline at end of file
+#endif // #ifndef BT_SOFT_BODY_SOLVER_BUFFER_OPENCL_H
