Index: src/BulletMultiThreaded/GpuSoftBodySolvers/OpenCL/CMakeLists.txt
===================================================================
--- src/BulletMultiThreaded/GpuSoftBodySolvers/OpenCL/CMakeLists.txt	(revision 2278)
+++ src/BulletMultiThreaded/GpuSoftBodySolvers/OpenCL/CMakeLists.txt	(working copy)
@@ -11,6 +11,6 @@
 	SUBDIRS(NVidia)
 ENDIF()
 
-IF(APPLE)
+IF(APPLE AND OPENCL_LIBRARY) 
 	SUBDIRS(Apple)
 ENDIF()
