--- cmake/NNGHelpers.cmake.orig	2021-08-11 08:09:53.000000000 +0800
+++ cmake/NNGHelpers.cmake	2023-03-05 15:21:14.000000000 +0800
@@ -98,7 +98,7 @@
                 ${PROJECT_SOURCE_DIR}/tests
                 ${PROJECT_SOURCE_DIR}/src
                 ${PROJECT_SOURCE_DIR}/include)
-        add_test(NAME ${NNG_TEST_PREFIX}.${NAME} COMMAND ${NAME} -t -v)
+        add_test(NAME ${NNG_TEST_PREFIX}.${NAME} COMMAND ${NAME} -v)
         set_tests_properties(${NNG_TEST_PREFIX}.${NAME} PROPERTIES TIMEOUT 180)
     endif ()
 endfunction()
@@ -111,7 +111,7 @@
                 ${PROJECT_SOURCE_DIR}/tests
                 ${PROJECT_SOURCE_DIR}/src
                 ${PROJECT_SOURCE_DIR}/include)
-        add_test(NAME ${NNG_TEST_PREFIX}.${NAME} COMMAND ${NAME} -t -v)
+        add_test(NAME ${NNG_TEST_PREFIX}.${NAME} COMMAND ${NAME} -v)
         set_tests_properties(${NNG_TEST_PREFIX}.${NAME} PROPERTIES TIMEOUT 180)
     endif ()
 endfunction()
