--- src/ap.cpp.orig	2023-12-28 04:25:40.000000000 +0800
+++ src/ap.cpp	2024-01-29 23:20:48.000000000 +0800
@@ -291,7 +291,9 @@
  * you can remove them, if you want - they are not used anywhere.
  *
  */
+#ifndef __ppc__
 static char     _ae_bool_must_be_8_bits_wide [1-2*((int)(sizeof(ae_bool))-1)*((int)(sizeof(ae_bool))-1)];
+#endif
 static char  _ae_int32_t_must_be_32_bits_wide[1-2*((int)(sizeof(ae_int32_t))-4)*((int)(sizeof(ae_int32_t))-4)];
 static char  _ae_int64_t_must_be_64_bits_wide[1-2*((int)(sizeof(ae_int64_t))-8)*((int)(sizeof(ae_int64_t))-8)];
 static char _ae_uint64_t_must_be_64_bits_wide[1-2*((int)(sizeof(ae_uint64_t))-8)*((int)(sizeof(ae_uint64_t))-8)];
@@ -313,7 +315,9 @@
 *************************************************************************/
 void ae_never_call_it()
 {
+#ifndef __ppc__
     ae_touch_ptr((void*)_ae_bool_must_be_8_bits_wide);
+#endif
     ae_touch_ptr((void*)_ae_int32_t_must_be_32_bits_wide);
     ae_touch_ptr((void*)_ae_int64_t_must_be_64_bits_wide);
     ae_touch_ptr((void*)_ae_uint64_t_must_be_64_bits_wide);
