--- a/lib/Fuzzer/CMakeLists.txt	2017-07-13 17:06:27.000000000 -0700
+++ b/lib/Fuzzer/CMakeLists.txt	2017-10-17 04:18:38.000000000 -0700
@@ -28,7 +28,7 @@
 
 # Compile libFuzzer if the compilation is specifically requested, OR
 # if the platform is known to be working.
-if ( LLVM_USE_SANITIZE_COVERAGE OR CMAKE_SYSTEM_NAME MATCHES "Darwin|Linux" )
+if ( LLVM_USE_SANITIZE_COVERAGE OR CMAKE_SYSTEM_NAME STREQUAL "Linux" OR (CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT ${CMAKE_SYSTEM_VERSION} VERSION_LESS 11) )
   add_library(LLVMFuzzerNoMainObjects OBJECT
       FuzzerCrossOver.cpp
       FuzzerDriver.cpp
