--- Eigen/src/QR/QrInstantiations.cpp.orig	2011-12-06 22:18:54.000000000 +0900
+++ Eigen/src/QR/QrInstantiations.cpp	2014-05-09 23:13:07.000000000 +0900
@@ -33,10 +33,10 @@
 namespace Eigen
 {
 
-template static void ei_tridiagonal_qr_step(float* , float* , int, int, float* , int);
-template static void ei_tridiagonal_qr_step(double* , double* , int, int, double* , int);
-template static void ei_tridiagonal_qr_step(float* , float* , int, int, std::complex<float>* , int);
-template static void ei_tridiagonal_qr_step(double* , double* , int, int, std::complex<double>* , int);
+template void ei_tridiagonal_qr_step(float* , float* , int, int, float* , int);
+template void ei_tridiagonal_qr_step(double* , double* , int, int, double* , int);
+template void ei_tridiagonal_qr_step(float* , float* , int, int, std::complex<float>* , int);
+template void ei_tridiagonal_qr_step(double* , double* , int, int, std::complex<double>* , int);
 
 EIGEN_QR_MODULE_INSTANTIATE();
 
