--- cmd/gvedit/Makefile.in.orig	2022-10-11 13:31:41.000000000 -0500
+++ cmd/gvedit/Makefile.in	2022-10-30 06:35:18.000000000 -0500
@@ -121,9 +121,7 @@
 	gvedit-mdichild.$(OBJEXT) $(am__objects_1)
 gvedit_OBJECTS = $(am_gvedit_OBJECTS)
 am__DEPENDENCIES_1 =
-gvedit_DEPENDENCIES = $(top_builddir)/lib/gvc/libgvc.la \
-	$(top_builddir)/lib/cgraph/libcgraph.la \
-	$(top_builddir)/lib/cdt/libcdt.la $(am__DEPENDENCIES_1) \
+gvedit_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_@AM_V@)
@@ -630,9 +628,9 @@
 	-DGVEDIT_DATADIR='"$(pkgdatadir)/gvedit"'
 
 gvedit_LDADD = \
-	$(top_builddir)/lib/gvc/libgvc.la \
-	$(top_builddir)/lib/cgraph/libcgraph.la \
-	$(top_builddir)/lib/cdt/libcdt.la \
+	-lgvc \
+	-lcgraph \
+	-lcdt \
 	$(QTGUI_LIBS) \
 	$(QTCORE_LIBS) \
 	$(EXPAT_LIBS) $(Z_LIBS)
