diff --git macosx/English.lproj/Attributes.xib macosx/English.lproj/Attributes.xib
index e75ef53..a3127c8 100644
--- macosx/English.lproj/Attributes.xib
+++ macosx/English.lproj/Attributes.xib
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9B18</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949</string>
diff --git macosx/English.lproj/Document.xib macosx/English.lproj/Document.xib
index 117e457..0e3b12c 100644
--- macosx/English.lproj/Document.xib
+++ macosx/English.lproj/Document.xib
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9C7010</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.26</string>
diff --git macosx/English.lproj/Export.xib macosx/English.lproj/Export.xib
index 1ae8aa4..315dd2a 100644
--- macosx/English.lproj/Export.xib
+++ macosx/English.lproj/Export.xib
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9C7010</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.26</string>
diff --git macosx/English.lproj/MainMenu.xib macosx/English.lproj/MainMenu.xib
index dc01a03..4365902 100644
--- macosx/English.lproj/MainMenu.xib
+++ macosx/English.lproj/MainMenu.xib
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <archive type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="7.01">
 	<data>
-		<int key="IBDocument.SystemTarget">1050</int>
+		<int key="IBDocument.SystemTarget">1060</int>
 		<string key="IBDocument.SystemVersion">9E17</string>
 		<string key="IBDocument.InterfaceBuilderVersion">629</string>
 		<string key="IBDocument.AppKitVersion">949.33</string>

