--- cmd/gvedit/Makefile.in.orig	2013-02-14 07:30:24.000000000 -0600
+++ cmd/gvedit/Makefile.in	2013-02-18 19:16:50.000000000 -0600
@@ -84,8 +84,7 @@
 	gvedit-mdichild.$(OBJEXT) $(am__objects_1)
 gvedit_OBJECTS = $(am_gvedit_OBJECTS)
 am__DEPENDENCIES_1 =
-gvedit_DEPENDENCIES = $(top_builddir)/lib/gvc/libgvc.la \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
+gvedit_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
@@ -563,7 +562,7 @@
 	-DGVEDIT_DATADIR=\""$(pkgdatadir)/gvedit"\"
 
 gvedit_LDADD = \
-	$(top_builddir)/lib/gvc/libgvc.la \
+	-lgvc -lcgraph \
 	$(QTGUI_LIBS) \
 	$(QTCORE_LIBS) \
 	$(EXPAT_LIBS) $(Z_LIBS) 
