--- configure.orig	2021-06-23 10:40:34.000000000 -0500
+++ configure	2021-10-27 13:06:59.000000000 -0500
@@ -10312,7 +10312,7 @@
         archive_expsym_cmds='$CC -o $output_objdir/$soname $libobjs $deplibs $wl'$no_entry_flag' $compiler_flags `if test -n "$allow_undefined_flag"; then func_echo_all "$wl$allow_undefined_flag"; else :; fi` $wl'$exp_sym_flag:\$export_symbols' '$shared_flag
       else
 	if test ia64 = "$host_cpu"; then
-	  hardcode_libdir_flag_spec='$wl-R $libdir:/usr/lib:/lib'
+	  hardcode_libdir_flag_spec='${wl}-rpath $libdir:/usr/lib:/lib'
 	  allow_undefined_flag="-z nodefs"
 	  archive_expsym_cmds="\$CC $shared_flag"' -o $output_objdir/$soname $libobjs $deplibs '"\$wl$no_entry_flag"' $compiler_flags $wl$allow_undefined_flag '"\$wl$exp_sym_flag:\$export_symbols"
 	else
@@ -10521,7 +10521,7 @@
     # extra space).
     freebsd2.2*)
       archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags /usr/lib/c++rt0.o'
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10537,7 +10537,7 @@
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
     freebsd* | dragonfly*)
       archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10723,7 +10723,7 @@
       else
 	archive_cmds='$LD -shared -o $lib $libobjs $deplibs $linker_flags'      # ELF
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10840,7 +10840,7 @@
 	  ;;
 	esac
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_shlibpath_var=no
       case $host_os in
       solaris2.[0-5] | solaris2.[0-5].*) ;;
