--- configure.orig	2018-11-29 05:45:16.000000000 -0600
+++ configure	2019-02-15 15:56:08.000000000 -0600
@@ -9928,7 +9928,7 @@
         archive_expsym_cmds='$CC -o $output_objdir/$soname $libobjs $deplibs '"\${wl}$no_entry_flag"' $compiler_flags `if test "x${allow_undefined_flag}" != "x"; then func_echo_all "${wl}${allow_undefined_flag}"; else :; fi` '"\${wl}$exp_sym_flag:\$export_symbols $shared_flag"
       else
 	if test "$host_cpu" = ia64; then
-	  hardcode_libdir_flag_spec='${wl}-R $libdir:/usr/lib:/lib'
+	  hardcode_libdir_flag_spec='${wl}-rpath $libdir:/usr/lib:/lib'
 	  allow_undefined_flag="-z nodefs"
 	  archive_expsym_cmds="\$CC $shared_flag"' -o $output_objdir/$soname $libobjs $deplibs '"\${wl}$no_entry_flag"' $compiler_flags ${wl}${allow_undefined_flag} '"\${wl}$exp_sym_flag:\$export_symbols"
 	else
@@ -10124,7 +10124,7 @@
     # extra space).
     freebsd2.2*)
       archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags /usr/lib/c++rt0.o'
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10140,7 +10140,7 @@
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
     freebsd* | dragonfly*)
       archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10315,7 +10315,7 @@
       else
 	archive_cmds='$LD -shared -o $lib $libobjs $deplibs $linker_flags'      # ELF
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_direct=yes
       hardcode_shlibpath_var=no
       ;;
@@ -10345,7 +10345,7 @@
 	  case $host_os in
 	   openbsd[01].* | openbsd2.[0-7] | openbsd2.[0-7].*)
 	     archive_cmds='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'
-	     hardcode_libdir_flag_spec='-R$libdir'
+	     hardcode_libdir_flag_spec='-rpath $libdir'
 	     ;;
 	   *)
 	     archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
@@ -10420,7 +10420,7 @@
 	  ;;
 	esac
       fi
-      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_libdir_flag_spec='-rpath $libdir'
       hardcode_shlibpath_var=no
       case $host_os in
       solaris2.[0-5] | solaris2.[0-5].*) ;;
@@ -10517,7 +10517,7 @@
       allow_undefined_flag='${wl}-z,nodefs'
       archive_cmds_need_lc=no
       hardcode_shlibpath_var=no
-      hardcode_libdir_flag_spec='${wl}-R,$libdir'
+      hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
       hardcode_libdir_separator=':'
       link_all_deplibs=yes
       export_dynamic_flag_spec='${wl}-Bexport'
