--- src/files_operators.c.orig	2010-06-06 09:31:47.000000000 -0700
+++ src/files_operators.c	2010-06-30 15:01:45.000000000 -0700
@@ -588,7 +588,7 @@
 
 Debug("VerifyFinderType of %s for %s\n", file,a.perms.findertype);
 
-if (strncmp(a.perms.findertype,"*",CF_BUFSIZE) == 0 || strncmp(a.perms.findertype,"",CF_BUFSIZE) == 0)
+if (!a.perms.findertype || strncmp(a.perms.findertype,"*",CF_BUFSIZE) == 0 || strncmp(a.perms.findertype,"",CF_BUFSIZE) == 0)
    {
    return 0;
    }
