--- setuptools/command/build_ext.py.orig	2008-09-24 11:10:28.000000000 -0600
+++ setuptools/command/build_ext.py	2009-10-06 16:03:03.000000000 -0600
@@ -82,6 +82,8 @@
 
     def get_ext_filename(self, fullname):
         filename = _build_ext.get_ext_filename(self,fullname)
+        if fullname not in self.ext_map:
+            return filename
         ext = self.ext_map[fullname]
         if isinstance(ext,Library):
             fn, ext = os.path.splitext(filename)
