--- install.pl.org	2007-11-21 01:12:04.000000000 -0800
+++ install.pl	2008-02-09 20:52:23.000000000 -0800
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!__PREFIX__/bin/perl -I __PREFIX__/libexec/nfsen -I __DESTROOT____PREFIX__/libexec/nfsen
 #
 #
 #  Copyright (c) 2004, SWITCH - Teleinformatikdienste fuer Lehre und Forschung
@@ -283,8 +283,8 @@
 sub PatchAllScripts {
 
 	my %GlobList = ( 
-		"bin/*"		=> 	"$NfConf::BINDIR",
-		"libexec/*"	=>	"$NfConf::LIBEXECDIR"
+		"bin/*"		=> 	"__DESTDIR____PREFIX__/bin",
+		"libexec/*"	=>	"__DESTDIR____PREFIX__/libexec/nfsen"
 	);
 
 	foreach my $glob_list ( keys %GlobList ) {
@@ -334,15 +334,14 @@
 	my ( $nfsen_uid, $www_gid ) = @_;
 
 	print "Setup php and html files.\n";
-	mkdir "$NfConf::HTMLDIR" unless -d "$NfConf::HTMLDIR";
-	die "Could not create HTMl directory '$NfConf::HTMLDIR': $!\n"  unless -d $NfConf::HTMLDIR;
+	mkdir "__DESTDIR____PREFIX__/www/nfsen";
 
 	$CopyRecursive::CopyLink = 1;
 	$CopyRecursive::MODE 	 = 0644;
 	$CopyRecursive::UID 	 = 0;
 	$CopyRecursive::GID 	 = $www_gid;
-	CopyRecursive::dircopy("html", "$NfConf::HTMLDIR");
-	open CONF, ">$NfConf::HTMLDIR/conf.php" || die "Can't open conf.php for writing: $!\n";
+	CopyRecursive::dircopy("html", "__DESTDIR____PREFIX__/www/nfsen");
+	open CONF, ">__DESTDIR____PREFIX__/www/nfsen/conf.php" || die "Can't open conf.php for writing: $!\n";
 	print CONF "<?\n";
 	print CONF "/* This file was automatically created by the NfSen install.pl script */\n\n";
 	print CONF "\$COMMSOCKET = \"$NfConf::COMMSOCKET\";\n";
@@ -362,11 +361,11 @@
 	umask 0002;
 	print "\nSetup diretories:\n";
 	my @dirs = (
-		"$NfConf::VARDIR",
-		"$NfConf::PIDDIR",
-		"$NfConf::VARDIR/tmp",
-		"$NfConf::VARDIR/filters",
-		"$NfConf::VARDIR/fmt",
+		"__DESTDIR____PREFIX__/var/nfsen",
+		"__DESTDIR____PREFIX__/var/nfsen/run",
+		"__DESTDIR____PREFIX__/var/nfsen/tmp",
+		"__DESTDIR____PREFIX__/var/nfsen/filters",
+		"__DESTDIR____PREFIX__/var/nfsen/fmt",
 		"$NfConf::PROFILESTATDIR",
 		"$NfConf::PROFILESTATDIR/live",
 		"$NfConf::PROFILEDATADIR",
@@ -585,21 +584,15 @@
 	$CopyRecursive::UID 	 = 0;
 	$CopyRecursive::GID 	 = $www_gid;
 	$CopyRecursive::MODE 	 = 0755;
-	unlink "$NfConf::BINDIR/nfsen.rc";
-	CopyRecursive::dircopy("bin", "$NfConf::BINDIR");
-	CopyRecursive::dircopy("libexec", "$NfConf::LIBEXECDIR");
+	unlink "__DESTDIR____PREFIX__/bin/nfsen.rc";
+	CopyRecursive::dircopy("bin", "__DESTDIR____PREFIX__/bin");
+	CopyRecursive::dircopy("libexec", "__DESTDIR____PREFIX__/libexec/nfsen");
 	$CopyRecursive::MODE 	 = 0644;
-	CopyRecursive::dircopy("etc", "$NfConf::CONFDIR");
-	CopyRecursive::dircopy("plugins/backend",  "$NfConf::BACKEND_PLUGINDIR");
-	CopyRecursive::dircopy("plugins/frontend", "$NfConf::FRONTEND_PLUGINDIR");
-	CopyRecursive::dircopy("doc", "$NfConf::DOCDIR");
+	CopyRecursive::dircopy("etc", "__DESTDIR____PREFIX__/etc");
+	CopyRecursive::dircopy("plugins/backend",  "__DESTDIR____PREFIX__/libexec/nfsen/plugins");
+	CopyRecursive::dircopy("plugins/frontend", "__DESTDIR____PREFIX__/www/nfsen/plugins");
+	CopyRecursive::dircopy("doc", "__DESTDIR____PREFIX__/www/nfsen/doc");
 
-	if ( $ConfigFile eq "$NfConf::CONFDIR/nfsen.conf" ) {
-		print "Keep config file '$ConfigFile'\n";
-	} else {
-		print "Copy config file '$ConfigFile'\n";
-		CopyRecursive::fcopy("$ConfigFile", "$NfConf::CONFDIR/nfsen.conf");
-	}
 	print "\n";
 
 } # End of CopyAllFiles
@@ -669,12 +662,6 @@
 	exit 1;
 }
 
-# check for extra errornoous nfsen.conf file, which may overwrite existing files
-if ( -f "$NfConf::CONFDIR/nfsen.conf" && -f "etc/nfsen.conf" &&
-	( (stat($ConfigFile))[1] != (stat("etc/nfsen.conf"))[1] )) {
-	die "Extra nfsen.conf file in etc directory found. Remove errornous file first"
-}
-
 my $hints = NfSen::LoadHints();
 if ( $$$hints{'version'} == -1 ) {
 	# initial NfSen install or upgrade from old version without hints
