--- install.pl.orig	2011-12-30 06:23:35.000000000 -0800
+++ install.pl	2014-12-03 07:26:35.000000000 -0800
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!__PREFIX__/bin/perl5.16 -I __PREFIX__/libexec/nfsen -I __DESTDIR____PREFIX__/libexec/nfsen
 #
 #
 #  Copyright (c) 2004, SWITCH - Teleinformatikdienste fuer Lehre und Forschung
@@ -277,8 +277,8 @@
 sub PatchAllScripts {
 
 	my %GlobList = ( 
-		"bin/*"		=> 	"$NfConf::INSTPREFIX$NfConf::BINDIR",
-		"libexec/*"	=>	"$NfConf::INSTPREFIX$NfConf::LIBEXECDIR"
+		"bin/*"		=> 	"__DESTDIR____PREFIX__/bin",
+		"libexec/*"	=>	"__DESTDIR____PREFIX__/libexec/nfsen"
 	);
 
 	foreach my $glob_list ( keys %GlobList ) {
@@ -328,16 +328,15 @@
 	my ( $nfsen_uid, $www_gid ) = @_;
 
 	print "Setup php and html files.\n";
-	mkpath("$NfConf::INSTPREFIX$NfConf::HTMLDIR", 1, 0755) unless -d "$NfConf::INSTPREFIX$NfConf::HTMLDIR";
-	die "Could not create HTMl directory '$NfConf::INSTPREFIX$NfConf::HTMLDIR': $!\n"  unless -d "$NfConf::INSTPREFIX$NfConf::HTMLDIR";
+        mkdir "__DESTDIR____PREFIX__/www/nfsen";
 
 	$CopyRecursive::CopyLink = 1;
 	$CopyRecursive::MODE 	 = 0644;
 	$CopyRecursive::UID 	 = 0;
 	$CopyRecursive::GID 	 = $www_gid;
-	CopyRecursive::dircopy("html", "$NfConf::INSTPREFIX$NfConf::HTMLDIR");
-	open CONF, ">$NfConf::INSTPREFIX$NfConf::HTMLDIR/conf.php" || die "Can't open conf.php for writing: $!\n";
-	print CONF "<?php\n";
+        CopyRecursive::dircopy("html", "__DESTDIR____PREFIX__/www/nfsen");
+        open CONF, ">__DESTDIR____PREFIX__/www/nfsen/conf.php" || die "Can't open conf.php for writing: $!\n";
+        print CONF "<?php\n";
 	print CONF "/* This file was automatically created by the NfSen $nfsen_version install.pl script */\n\n";
 	print CONF "\$COMMSOCKET = \"$NfConf::COMMSOCKET\";\n";
 	print CONF "\n\$DEBUG=0;\n";
@@ -356,11 +355,11 @@
 	umask 0002;
 	print "\nSetup diretories:\n";
 	my @dirs = (
-		"$NfConf::INSTPREFIX$NfConf::VARDIR",
-		"$NfConf::INSTPREFIX$NfConf::VARDIR/tmp",
-		"$NfConf::INSTPREFIX$NfConf::PIDDIR",
-		"$NfConf::INSTPREFIX$NfConf::FILTERDIR",
-		"$NfConf::INSTPREFIX$NfConf::FORMATDIR",
+                "__DESTDIR____PREFIX__/var/nfsen",
+                "__DESTDIR____PREFIX__/var/nfsen/tmp",
+                "__DESTDIR____PREFIX__/var/nfsen/run",
+                "__DESTDIR____PREFIX__/var/nfsen/filters",
+                "__DESTDIR____PREFIX__/var/nfsen/fmt",
 		"$NfConf::INSTPREFIX$NfConf::PROFILESTATDIR",
 		"$NfConf::INSTPREFIX$NfConf::PROFILESTATDIR/live",
 		"$NfConf::INSTPREFIX$NfConf::PROFILEDATADIR",
@@ -579,20 +578,14 @@
 	$CopyRecursive::UID 	 = 0;
 	$CopyRecursive::GID 	 = $www_gid;
 	$CopyRecursive::MODE 	 = 0755;
-	unlink "$NfConf::BINDIR/nfsen.rc";
-	CopyRecursive::dircopy("bin", "$NfConf::INSTPREFIX$NfConf::BINDIR");
-	CopyRecursive::dircopy("libexec", "$NfConf::INSTPREFIX$NfConf::LIBEXECDIR");
-	$CopyRecursive::MODE 	 = 0644;
-	CopyRecursive::dircopy("etc", "$NfConf::INSTPREFIX$NfConf::CONFDIR");
-	CopyRecursive::dircopy("plugins/backend",  "$NfConf::INSTPREFIX$NfConf::BACKEND_PLUGINDIR");
-	CopyRecursive::dircopy("plugins/frontend", "$NfConf::INSTPREFIX$NfConf::FRONTEND_PLUGINDIR");
+        unlink "__DESTDIR____PREFIX__/bin/nfsen.rc";
+        CopyRecursive::dircopy("bin", "__DESTDIR____PREFIX__/bin");
+        CopyRecursive::dircopy("libexec", "__DESTDIR____PREFIX__/libexec/nfsen");
+        $CopyRecursive::MODE 	 = 0644;
+        CopyRecursive::dircopy("etc", "__DESTDIR____PREFIX__/etc");
+        CopyRecursive::dircopy("plugins/backend",  "__DESTDIR____PREFIX__/libexec/nfsen/plugins");
+        CopyRecursive::dircopy("plugins/frontend", "__DESTDIR____PREFIX__/www/nfsen/plugins");
 
-	if ( $ConfigFile eq "$NfConf::CONFDIR/nfsen.conf" ) {
-		print "Keep config file '$ConfigFile'\n";
-	} else {
-		print "Copy config file '$ConfigFile'\n";
-		CopyRecursive::fcopy("$ConfigFile", "$NfConf::CONFDIR/nfsen.conf");
-	}
 	print "\n";
 
 } # End of CopyAllFiles
@@ -663,12 +656,6 @@
 	exit 1;
 }
 
-# check for extra errornoous nfsen.conf file, which may overwrite existing files
-if ( -f "$NfConf::CONFDIR/nfsen.conf" && -f "etc/nfsen.conf" &&
-	( (stat($ConfigFile))[1] != (stat("etc/nfsen.conf"))[1] )) {
-	die "Extra nfsen.conf file in etc directory found. Remove errornous file first"
-}
-
 Log::LogInit();
 
 my $hints = NfSen::LoadHints();
