--- Makefile.orig	2008-01-14 18:18:23.000000000 -0600
+++ Makefile	2008-01-14 18:20:47.000000000 -0600
@@ -13,9 +13,9 @@
 # CPVALINK = Y
 
 # where to install the binary
-BINDIR = /usr/local/bin
+BINDIR = @DESTROOT@@PREFIX@/bin
 # where to install the man page
-MANDIR = /usr/local/share/man/man1
+MANDIR = @DESTROOT@@PREFIX@/share/man/man1
 # extension of the man page
 MANEXT = 1
 
@@ -26,11 +26,11 @@
 # of work. Switching on optimization *does* change the output
 # slightly, probably owing to slight differences in floating point
 # calculations.
-CDEBUGFLAGS=-O2
+CDEBUGFLAGS=@CFLAGS@ @CPPFLAGS@
 # if your ppm libraries are not in a standard place, uncomment
 # and change this. You may also need thing like -R, but if you're
 # on that sort of system you probably know it!
-# LDLIBS = -L/usr/local/lib
+LDLIBS = @LDFLAGS@
 
 # install commands
 BININSTALL = install -m 0555
@@ -38,7 +38,7 @@
 
 OBJS = ppmtomd.o mddata.o photocolcor.o vphotocolcor.o dyesubcolcor.o
 ppmtomd: $(OBJS)
-	$(CC) -o ppmtomd $(OBJS) $(LDLIBS) -lpnm -lppm -lpgm -lpbm -lm
+	$(CC) -o ppmtomd $(OBJS) $(LDLIBS) -lnetpbm -lm
 
 ppmtomd.o: ppmtomd.c mddata.h
 	$(CC) $(CDEBUGFLAGS) $(DEFAULTMODEL) -W -Wall -Wstrict-prototypes -c ppmtomd.c 
