diff --git trunk/lib/Support/Unix/Path.inc.orig trunk/lib/Support/Unix/Path.inc
index c64c0df..aaa2226 100644
--- trunk/lib/Support/Unix/Path.inc.orig
+++ trunk/lib/Support/Unix/Path.inc
@@ -38,6 +38,9 @@
 #include <mach-o/dyld.h>
 #include <sys/attr.h>
 #include <copyfile.h>
+#ifndef COPYFILE_CLONE
+#define COPYFILE_CLONE		(1<<24)
+#endif
 #elif defined(__DragonFly__)
 #include <sys/mount.h>
 #endif
