--- runtime/keiko.iset	2010-12-27 21:55:07.000000000 -0600
+++ runtime/keiko.iset.new	2010-12-27 21:57:48.000000000 -0600
@@ -136,9 +136,11 @@
 # LDIs :: a, b --> mem_s[a+s*b]
 # STIs :: a, b, c --> []; mem_s[b+s*c] := a
 inst LDIW 0		B.* {subs($1.x, $2.i, type($t))}
+inst LDIF 0        B.f {subs($1.x, $2.i, float)}
 inst LDIS 0		B.i {subs($1.x, $2.i, short)}
 inst LDIC 0		B.i {subs($1.x, $2.i, uchar)}
 inst STIW 0		S3  {subs($2.x, $3.i, type($u1)) = $1.*;}
+inst STIF 0        S3  {subs($2.x, $3.i, float) = $1.f;}
 inst STIS 0		S3  {subs($2.x, $3.i, short) = $1.i;}
 inst STIC 0		S3  {subs($2.x, $3.i, uchar) = $1.i;}
 
@@ -754,10 +756,12 @@
 expand LDIC 0		{INDEXC, LOADC}
 expand LDIS 0		{INDEXS, LOADS}
 expand LDIW 0		{INDEXW, LOADW}
+expand LDIF 0      {INDEXW, LOADF}
 expand LDID 0		{INDEXD, LOADD}
 expand STIC 0		{INDEXC, STOREC}
 expand STIS 0		{INDEXS, STORES}
 expand STIW 0		{INDEXW, STOREW}
+expand STIF 0      {INDEXW, STOREF}
 expand STID 0		{INDEXD, STORED}
 
 expand LDEW 1		{LDLW -4, LDNW $a}
