Index: configure.ac
===================================================================
--- configure.ac	(revision 2980)
+++ configure.ac	(revision 2984)
@@ -4,6 +4,7 @@
 AC_PREREQ(2.56)
 AC_INIT(ldns, 1.6.0, libdns@nlnetlabs.nl,libdns)
 AC_CONFIG_SRCDIR([drill.c])
+sinclude(../acx_nlnetlabs.m4)
 
 OURCPPFLAGS=''
 CPPFLAGS=${CPPFLAGS:-${OURCPPFLAGS}}
@@ -194,71 +195,7 @@
 AC_CHECK_LIB(socket, socket)
 AC_CHECK_LIB(nsl, inet_pton)
 
-# Checks for libraries.
-# Check for SSL, original taken from
-# http://www.gnu.org/software/ac-archive/htmldoc/check_ssl.html and
-# modified for NSD and 
-# copied again for use in ldns
-AC_ARG_WITH(ssl, AC_HELP_STRING([--with-ssl=pathname],
-                                    [enable SSL (will check /usr/local/ssl
-                            /usr/lib/ssl /usr/ssl /usr/pkg /usr/local /usr/sfw /usr)]),[
-        ],[
-            withval="yes"
-    if test x_$withval != x_no; then
-        AC_MSG_CHECKING(for SSL)
-        if test x_$withval = x_ -o x_$withval = x_yes; then
-            withval="/usr/local/ssl /usr/lib/ssl /usr/ssl /usr/pkg /usr/local /usr/sfw /usr"
-        fi
-        for dir in $withval; do
-            ssldir="$dir"
-            if test -f "$dir/include/openssl/ssl.h"; then
-                found_ssl="yes";
-                AC_DEFINE_UNQUOTED([HAVE_SSL], [], [Define if you have the SSL libraries installed.])
-                CPPFLAGS="$CPPFLAGS -I$ssldir/include";
-                break;
-            fi
-        done
-        if test x_$found_ssl != x_yes; then
-            AC_MSG_ERROR(Cannot find the SSL libraries in $withval)
-        else
-            AC_MSG_RESULT(found in $ssldir)
-            HAVE_SSL=yes
-            LDFLAGS="$LDFLAGS -L$ssldir/lib";
-            if test "x$enable_rpath" = xyes; then
-	        RUNTIME_PATH="$RUNTIME_PATH -R$ssldir/lib"
-            fi
-            AC_MSG_CHECKING([for HMAC_CTX_init in -lcrypto])
-            LIBS="$LIBS -lcrypto"
-            AC_TRY_LINK(, [
-                int HMAC_CTX_init(void);
-                (void)HMAC_CTX_init();
-              ], [
-                AC_MSG_RESULT(yes)
-                AC_DEFINE([HAVE_HMAC_CTX_INIT], 1,
-                        [If you have HMAC_CTX_init])
-              ], [
-                AC_MSG_RESULT(no)
-                # check if -lwsock32 or -lgdi32 are needed.     
-                LIBS="$LIBS -lgdi32 -lws2_32"
-                AC_MSG_CHECKING([if -lcrypto needs -lgdi32])
-                AC_TRY_LINK([], [
-                    int HMAC_CTX_init(void);
-                    (void)HMAC_CTX_init();
-                  ],[
-                    AC_DEFINE([HAVE_HMAC_CTX_INIT], 1,
-                        [If you have HMAC_CTX_init])
-                    AC_MSG_RESULT(yes)
-                  ],[
-                    AC_MSG_RESULT(no)
-                    AC_MSG_ERROR([OpenSSL found in $ssldir, but version 0.9.7 or
- higher is required])
-                ])
-            ])
-        fi
-        AC_SUBST(HAVE_SSL)
-	AC_SUBST(RUNTIME_PATH)
-    fi
-])
+ACX_WITH_SSL_OPTIONAL
 
 LIBS_STC="$LIBS"
 AC_SUBST(LIBS_STC)
