--- src/plugins/plugin.pri.orig	2013-01-27 16:35:01.000000000 -0600
+++ src/plugins/plugin.pri	2013-03-02 00:56:59.000000000 -0600
@@ -7,8 +7,6 @@
 contains(QT_CONFIG, reduce_exports): CONFIG += hide_symbols
 win32 {
     DESTDIR = $$OUT_PWD/../../../plugins/tiled
-} else:macx {
-    DESTDIR = $$OUT_PWD/../../../bin/Tiled.app/Contents/PlugIns
 } else {
     DESTDIR = $$OUT_PWD/../../../lib/tiled/plugins
 }
@@ -18,9 +16,7 @@
 INSTALLS += target
 
 include(../libtiled/libtiled.pri)
-macx {
-    QMAKE_LIBDIR += $$OUT_PWD/../../../bin/Tiled.app/Contents/Frameworks
-} else:win32 {
+win32 {
     LIBS += -L$$OUT_PWD/../../../lib
 } else {
     QMAKE_LIBDIR += $$OUT_PWD/../../../lib
