diff --git src/gui/qgslibloader.cpp src/gui/qgslibloader.cpp
index 007d049..f9212af 100644
--- src/gui/qgslibloader.cpp
+++ src/gui/qgslibloader.cpp
@@ -259,19 +259,7 @@ int GSLibFunctions::loadLibrary(const QString& location, QString& last_error) {
 	gsapi_revision_t rv;
 	/* Try to load the library */
 	if (location == "") {
-		#ifdef Q_WS_X11
-		#if defined(__x86_64__) || defined(__ppc64__) || defined (__s390x__) || defined (__sparc64__)
-		// try 64 bit libraries on 64 bit system
-      tryLocationLoop("/usr/lib64");
-		tryLocationLoop("/usr/local/lib64");
-		#endif // 64 bit
-      tryLocationLoop("/usr/lib");
-		tryLocationLoop("/usr/local/lib");
-		#endif // Q_WS_X11
-		#ifdef Q_WS_MAC
-		tryLocation("/usr/lib/libgs.dylib");
-		tryLocation("/usr/local/lib/libgs.dylib");
-		#endif
+		tryLocation("@@PREFIX@@/lib/libgs.dylib");
 	} else {
 		m_LibGSLocation = location;
 		QString libloc = location;
