--- src/plugins/platforms/cocoa/qcocoahelpers.h.orig	2017-01-18 07:20:58.000000000 -0700
+++ src/plugins/platforms/cocoa/qcocoahelpers.h	2017-05-29 06:54:43.000000000 -0700
@@ -168,6 +168,14 @@
 @end
 
 @interface QT_MANGLE_NAMESPACE(QNSPanelContentsWrapper) : NSView
+#ifndef __LP64__
+{
+NSButton *_okButton;
+NSButton *_cancelButton;
+NSView *_panelContents;
+NSEdgeInsets _panelContentsMargins;
+}
+#endif
 
 @property (nonatomic, readonly) NSButton *okButton;
 @property (nonatomic, readonly) NSButton *cancelButton;
--- src/plugins/platforms/cocoa/qcocoahelpers.mm.orig	2017-01-18 07:20:58.000000000 -0700
+++ src/plugins/platforms/cocoa/qcocoahelpers.mm	2017-05-14 12:37:30.000000000 -0700
@@ -312,6 +312,13 @@
  */
 @implementation QNSPanelContentsWrapper
 
+#ifndef __LP64__
+@synthesize okButton = _okButton;
+@synthesize cancelButton = _cancelButton;
+@synthesize panelContents = _panelContents;
+@synthesize panelContentsMargins = _panelContentsMargins;
+#endif
+
 - (instancetype)initWithPanelDelegate:(id<QT_MANGLE_NAMESPACE(QNSPanelDelegate)>)panelDelegate
 {
     if ((self = [super initWithFrame:NSZeroRect])) {
