--- src/plugins/sqldrivers/sqlite/sqlite.pro.orig	2017-01-18 07:20:58.000000000 -0700
+++ src/plugins/sqldrivers/sqlite/sqlite.pro	2017-06-22 04:52:40.000000000 -0700
@@ -5,11 +5,11 @@
 HEADERS += $$PWD/qsql_sqlite_p.h
 SOURCES += $$PWD/qsql_sqlite.cpp $$PWD/smain.cpp
 
-qtConfig(system-sqlite) {
+#qtConfig(system-sqlite) {
     QMAKE_USE += sqlite
-} else {
-    include($$PWD/../../../3rdparty/sqlite.pri)
-}
+#} else {
+#    include($$PWD/../../../3rdparty/sqlite.pri)
+#}
 
 OTHER_FILES += sqlite.json
 
