--- src/virtualkeyboard/virtualkeyboard.pro.orig	2016-11-28 02:45:25.000000000 -0700
+++ src/virtualkeyboard/virtualkeyboard.pro	2017-06-20 07:13:44.000000000 -0700
@@ -207,10 +207,10 @@
     SOURCES += desktopinputpanel.cpp inputview.cpp
     HEADERS += desktopinputpanel.h inputview.h
     DEFINES += QT_VIRTUALKEYBOARD_DESKTOP
-    !no-pkg-config:packagesExist(xcb) {
-        PKGCONFIG += xcb xcb-xfixes
-        DEFINES += QT_VIRTUALKEYBOARD_HAVE_XCB
-    }
+    #!no-pkg-config:packagesExist(xcb) {
+    #    PKGCONFIG += xcb xcb-xfixes
+    #    DEFINES += QT_VIRTUALKEYBOARD_HAVE_XCB
+    #}
 }
 SOURCES += appinputpanel.cpp
 HEADERS += appinputpanel.h
