Honor MacPorts CXXFLAGS.
--- setup.py.orig	2020-08-27 09:12:55.000000000 -0500
+++ setup.py	2020-10-05 22:57:09.000000000 -0500
@@ -79,11 +79,15 @@
                 cxx_sources.append(source)
         extra_args = ext.extra_compile_args or []
 
+        cxxflags = os.environ['CXXFLAGS'].split()
+
         objects = []
         for lang, sources in (('c', c_sources), ('c++', cxx_sources)):
             if lang == 'c++':
                 if self.compiler.compiler_type == 'msvc':
                     extra_args.append('/EHsc')
+                else:
+                    extra_args.extend(cxxflags)
 
             macros = ext.define_macros[:]
             if platform.system() == 'Darwin':
@@ -110,6 +114,7 @@
         if ext.extra_objects:
             objects.extend(ext.extra_objects)
         extra_args = ext.extra_link_args or []
+        extra_args.extend(cxxflags)
         # when using GCC on Windows, we statically link libgcc and libstdc++,
         # so that we don't need to package extra DLLs
         if self.compiler.compiler_type == 'mingw32':
