--- Makefile.orig	Fri Feb 28 23:16:55 2003
+++ Makefile	Fri Feb 28 23:18:00 2003
@@ -570,9 +570,9 @@
 	@echo + In order to rectify this problem, you MUST build with:
 	@echo ++ SSLTYPE=$(SSLTYPE).nopwd
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-	@echo
-	@echo Do you want to continue this build anyway?  Type y or n please:
-	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make nounenc;exit 1);; esac'
+#	@echo
+#	@echo Do you want to continue this build anyway?  Type y or n please:
+#	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make nounenc;exit 1);; esac'
 
 nounenc:
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
