diff -u -r DUDynDNSUpdater.m.orig DUDynDNSUpdater.m
--- DUDynDNSUpdater.m.orig	2003-09-24 09:50:44.000000000 +1000
+++ DUDynDNSUpdater.m	2007-11-29 11:08:53.000000000 +1100
@@ -212,7 +212,7 @@
                     [host setStatus:@"No such host"];
                     break;
                 case _good:
-                    NSLog(@"%@: update suceeded",host);
+                    NSLog(@"%@: update succeeded",host);
                     [host updateAddressTo:[[host interface] getAddress]]; // XXX must cache all those addresses
                     [host setStatus:@"Ok"];
                     break;
@@ -283,7 +283,7 @@
                 [theUser setActive:NO];
                 break;
             case _dnserr:
-                NSLog(@"DynDNS.org DNS error. Report the following line to DynDNS.org support departement :\nPacket ID: %@",(argument?argument:@"NoPacketId"));
+                NSLog(@"DynDNS.org DNS error. Report the following line to DynDNS.org support department :\nPacket ID: %@",(argument?argument:@"NoPacketId"));
                 [theUser setUserStatus:@"DNS Error"];
                 [theUser setActive:NO];
                 break;
diff -u -r DUEasyDNSUpdater.m.orig DUEasyDNSUpdater.m
--- DUEasyDNSUpdater.m.orig	2003-09-24 09:50:44.000000000 +1000
+++ DUEasyDNSUpdater.m	2007-11-29 11:08:53.000000000 +1100
@@ -131,7 +131,7 @@
             NSString *returnCode=[words objectAtIndex:0];
 
             if([returnCode isEqualToString:@"NOACCESS"]) {
-                NSLog(@"Access denyied for %@ or host is not yours (Deactivating this user)",theUser);
+                NSLog(@"Access denied for %@ or host is not yours (Deactivating this user)",theUser);
                 [theUser setUserStatus:@"Bad Auth"];
                 [theUser setActive:NO];
                 return;
diff -u -r DUExternalInterface.m.orig DUExternalInterface.m
--- DUExternalInterface.m.orig	2003-09-24 09:50:44.000000000 +1000
+++ DUExternalInterface.m	2007-11-29 11:13:19.000000000 +1100
@@ -43,7 +43,7 @@
 }
 
 - (void)cacheAddress {
-    NSLog(@"Caching external address");
+    //NSLog(@"Caching external address");
     if([lastUpdate timeIntervalSinceNow]<=-EXTIPTEST_INTERVAL*60.0) {
         NSDictionary *serverReturn;
 
