diff --git Makefile.mac Makefile.mac
index e3d09cb..2e6fa87 100644
--- Makefile.mac
+++ Makefile.mac
@@ -32,8 +32,8 @@ MIDI_INCLUDE=MIDI
 # uncomment the line below for various debug facilities
 #DEBUG_OPTION=-D DEBUG
 
-CC=gcc
-LINK=gcc
+CC?=gcc
+LINK?=$(CC)
 
 ifeq ($(CONTROLLER2_V2_INCLUDE),CONTROLLER2_V2)
 CONTROLLER2_OPTIONS=-D CONTROLLER2_V2
@@ -145,7 +145,7 @@ GTKLIBS=`pkg-config --libs gtk+-3.0`
 AUDIO_OPTIONS=-DPORTAUDIO
 AUDIO_LIBS=-lportaudio -O3
 
-CFLAGS=	-g -Wno-deprecated-declarations
+CFLAGS?=-g -Wno-deprecated-declarations
 OPTIONS=$(MIDI_OPTIONS) $(PURESIGNAL_OPTIONS) $(REMOTE_OPTIONS) $(USBOZY_OPTIONS) \
 	$(SOAPYSDR_OPTIONS) $(LOCALCW_OPTIONS) \
 	$(STEMLAB_OPTIONS) \
@@ -408,7 +408,7 @@ clean:
 	-rm -rf $(PROGRAM).app
 
 install: $(PROGRAM)
-	cp $(PROGRAM) /usr/local/bin
+	cp $(PROGRAM) $(DESTDIR)/usr/local/bin
 
 release: $(PROGRAM)
 	cp $(PROGRAM) release/pihpsdr
