--- Makefile.orig	2015-12-23 13:09:27.000000000 -0600
+++ Makefile	2019-03-09 08:32:04.000000000 -0600
@@ -1,4 +1,4 @@
-VERSION=v2.4
+VERSION=2.4
 
 prefix=/usr/local
 
@@ -8,9 +8,9 @@
 SYS=posix
 #SYS=mingw
 
-CRYPTO=OPENSSL
+#CRYPTO=OPENSSL
 #CRYPTO=POLARSSL
-#CRYPTO=GNUTLS
+CRYPTO=GNUTLS
 LIBZ=-lz
 LIB_GNUTLS=-lgnutls -lhogweed -lnettle -lgmp $(LIBZ)
 LIB_OPENSSL=-lssl -lcrypto $(LIBZ)
@@ -26,7 +26,7 @@
 
 bindir=$(prefix)/bin
 sbindir=$(prefix)/sbin
-mandir=$(prefix)/man
+mandir=$(prefix)/share/man
 
 BINDIR=$(DESTDIR)$(bindir)
 SBINDIR=$(DESTDIR)$(sbindir)
