--- bytes.h.orig	2009-04-30 05:52:02.000000000 +0200
+++ bytes.h	2009-04-30 06:00:47.000000000 +0200
@@ -34,6 +34,20 @@
       | (((x) & 0x00000000000000ffull) << 56))
 
 #else
+
+#ifdef __APPLE__
+#include <machine/endian.h>
+#define __BIG_ENDIAN		BIG_ENDIAN
+#define __LITTLE_ENDIAN		LITTLE_ENDIAN
+
+#include <libkern/OSByteOrder.h>
+#define __BYTE_ORDER		BYTE_ORDER
+#define __FLOAT_WORD_ORDER	BYTE_ORDER
+
+#define __bswap_32			OSSwapConstInt32
+#define __bswap_64			OSSwapConstInt64
+
+#else
 #include <endian.h>
 #include <byteswap.h>
 
@@ -41,6 +55,8 @@
 typedef __uint32_t uint32_t;
 #endif
 
+#endif
+
 #if !defined(__BYTE_ORDER) || !defined(__FLOAT_WORD_ORDER)
 #error "Undefined byte and float word order!"
 #endif
