Fix overlinking.
http://bugs.debian.org/745216
--- librtmp/librtmp.pc.in.orig	2021-02-19 11:54:54.000000000 -0600
+++ librtmp/librtmp.pc.in	2022-04-22 21:48:03.000000000 -0500
@@ -6,8 +6,8 @@
 Name: librtmp
 Description: RTMP implementation
 Version: @VERSION@
-Requires: @CRYPTO_REQ@
+Requires.private: @CRYPTO_REQ@
 URL: http://rtmpdump.mplayerhq.hu
-Libs: -L${libdir} -lrtmp -lz @PUBLIC_LIBS@
-Libs.private: @PRIVATE_LIBS@
+Libs: -L${libdir} -lrtmp
+Libs.private: @PUBLIC_LIBS@ @PRIVATE_LIBS@
 Cflags: -I${incdir}
