--- src/main.cpp.orig	2008-07-18 13:37:22.000000000 +0900
+++ src/main.cpp	2008-07-18 13:39:30.000000000 +0900
@@ -41,6 +41,16 @@
 
 bool MyApp::OnInit()
 {
+  {
+    wxString oldpath;
+    wxGetEnv(wxT("PATH"), &oldpath);
+    if (oldpath.Find(wxT("@@PREFIX@@/bin")) == -1)
+    {
+      wxString newpath = oldpath.Prepend(wxT("@@PREFIX@@/bin:"));
+      wxSetEnv(wxT("PATH"), newpath);
+    }
+  }
+
   int lang = wxLANGUAGE_UNKNOWN;
 
   wxConfig *config = new wxConfig(wxT("wxMaxima"));
