--- src/Dirstructure.cpp.orig	2018-06-01 18:24:00.000000000 +0900
+++ src/Dirstructure.cpp	2018-06-03 23:04:26.000000000 +0900
@@ -296,10 +296,8 @@
     command = wxT("/Applications/Maxima.app");
   if (wxFileExists("/Applications/maxima.app"))
     command = wxT("/Applications/maxima.app");
-  else if (wxFileExists("/usr/local/bin/maxima"))
-    command = wxT("/usr/local/bin/maxima");
-  else if (wxFileExists("/usr/bin/maxima"))
-    command = wxT("/usr/bin/maxima");
+  else if (wxFileExists("@@PREFIX@@/bin/maxima"))
+    command = wxT("@@PREFIX@@/bin/maxima");
   else
     command = wxT("maxima");
   return command;
