--- src/Dirstructure.h.old	2015-07-31 02:10:34.000000000 +0200
+++ src/Dirstructure.h	2015-07-31 02:16:47.000000000 +0200
@@ -110,7 +110,7 @@
   
   //! The path maxima is found at by default.
 #if defined __WXMAC__
-  wxString MaximaDefaultName() {return wxT("/Applications/Maxima.app");}
+  wxString MaximaDefaultName() {return wxT("maxima");}
 #elif defined __WXMSW__
   wxString MaximaDefaultName() {return wxStandardPaths::Get().GetExecutablePath()+wxT("/bin/maxima.bat");}
 #else
