--- src/main.cpp.orig	2009-04-20 08:33:02.000000000 +0900
+++ src/main.cpp	2009-04-20 08:36:51.000000000 +0900
@@ -46,6 +46,16 @@
 
 bool MyApp::OnInit()
 {
+  {
+    wxString oldpath;
+    wxGetEnv(wxT("PATH"), &oldpath);
+    if (oldpath.Find(wxT("@@PREFIX@@/bin")) == -1)
+    {
+      wxString newpath = oldpath.Prepend(wxT("@@PREFIX@@/bin:"));
+      wxSetEnv(wxT("PATH"), newpath);
+    }
+  }
+
   int lang = wxLANGUAGE_UNKNOWN;
 
 #if defined __WXMSW__
