--- data/Info.plist.in.orig	2008-04-19 23:29:52.000000000 +0900
+++ data/Info.plist.in	2008-11-21 00:42:54.000000000 +0900
@@ -15,7 +15,7 @@
     <key>CFBundleDevelopmentRegion</key>
     <string>English</string>
     <key>CFBundleExecutable</key>
-    <string>wxMaxima</string>
+    <string>wxmaxima</string>
     <key>CFBundleIconFile</key>
     <string>wxmac.icns</string>
     <key>CFBundlePackageType</key>
