--- pyproject.toml.orig	2025-09-30 15:58:13
+++ pyproject.toml	2025-09-30 15:58:49
@@ -9,21 +9,7 @@
 requires = [
     "meson-python>=0.14.0",
     "Cython>=0.29.35",
-
-    # When numpy 2.0.0rc1 comes out, we should update this to build against 2.0,
-    # and then runtime depend on the range 1.22.X to <2.3. No need to switch to
-    # 1.25.2 in the meantime (1.25.x is the first version which exports older C
-    # API versions by default).
-
-    # default numpy requirements
-    "numpy==1.22.4; python_version<='3.10' and platform_python_implementation != 'PyPy'",
-    "numpy==1.23.3; python_version=='3.11' and platform_python_implementation != 'PyPy'",
-    "numpy==1.26.0; python_version=='3.12' and platform_python_implementation != 'PyPy'",
-    # For Python versions which aren't yet officially supported, we specify an
-    # unpinned NumPy which allows source distributions to be used and allows
-    # wheels to be used as soon as they become available.
-    "numpy; python_version>='3.13'",
-    "numpy; python_version>='3.9' and platform_python_implementation=='PyPy'",
+    "numpy",
 ]
 
 [project]
@@ -40,7 +26,7 @@
 dependencies = [
     # TODO: update to "pin-compatible" once possible, see
     # https://github.com/FFY00/meson-python/issues/29
-    "numpy>=1.22.4,<2.0",
+    "numpy>=1.22.4",
 ]
 readme = "README.rst"
 classifiers = [
