--- src/Makevars.in.orig	2019-04-27 11:35:06.000000000 +0800
+++ src/Makevars.in	2023-07-11 12:46:28.000000000 +0800
@@ -1,9 +1,8 @@
 ## -*- mode: makefile; -*-
 
-PKG_CXXFLAGS = -I../inst/include @OPENMP_FLAG@
-PKG_LIBS= @OPENMP_FLAG@ $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
+PKG_CXXFLAGS = -DARMA_USE_OPENMP=1 -I../inst/include -fopenmp
+PKG_LIBS= -fopenmp $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
 
 ## And with R 3.4.0, and RcppArmadillo 0.7.960.*, we turn C++11 on as OpenMP
 ## support within Armadillo prefers / requires it
 CXX_STD = CXX11
-
