--- autogen.sh.orig	2019-01-05 14:46:08.000000000 -0800
+++ autogen.sh	2019-12-26 21:41:55.000000000 -0800
@@ -14,8 +14,8 @@
 DIE=0
 
 have_libtool=false
-if libtool --version < /dev/null > /dev/null 2>&1 ; then
-	libtool_version=`libtoolize --version |	 libtoolize --version | sed 's/^[^0-9]*\([0-9].[0-9.]*\).*/\1/'`
+if glibtool --version < /dev/null > /dev/null 2>&1 ; then
+	libtool_version=`glibtoolize --version | glibtoolize --version | sed 's/^[^0-9]*\([0-9].[0-9.]*\).*/\1/'`
 	case $libtool_version in
 	    1.4*|1.5*|2.*)
 		have_libtool=true
@@ -65,7 +65,7 @@
         echo "to pass any to it, please specify them on the $0 command line."
 fi
 
-libtoolize --copy --force
+glibtoolize --copy --force
 aclocal $ACLOCAL_FLAGS
 
 # optionally run autoheader
