diff -ur CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt	2021-03-22 08:05:04.000000000 -0400
+++ CMakeLists.txt	2021-08-18 14:35:27.167498889 -0400
@@ -83,8 +83,6 @@
     # for code coverage we want exactly one binary to be produced and exercised
     if(NOT DEFINED ENV{CODE_COVERAGE})
         add_subdirectory(examples/exe_with_static_libs)
-        add_subdirectory(examples/executable_dll_and_plugin)
-        add_subdirectory(examples/combining_the_same_tests_built_differently_in_multiple_shared_objects)
         add_subdirectory(scripts/playground)
         add_subdirectory(examples/mpi)
     endif()
