--- fancycompleter.py.orig	2012-03-16 10:54:45.000000000 -0400
+++ fancycompleter.py	2012-03-16 10:55:01.000000000 -0400
@@ -140,7 +140,7 @@
             try:
                 my_execfile(rcfile, mydict)
                 return mydict['Config']()
-            except Exception as e:
+            except Exception, e:
                 print('** error when importing %s: %s **' % (filename, e))
         return self.DefaultConfig()
 
