--- Makefile.in.orig	Tue Dec 14 11:43:01 2004
+++ Makefile.in	Tue Dec 14 11:44:12 2004
@@ -65,7 +65,7 @@
 EXEEXT = @EXEEXT@
 FBGDATA_SOUND = @FBGDATA_SOUND@
 FBG_SOUND_OBJECTS = @FBG_SOUND_OBJECTS@
-GL_LIBS = @GL_LIBS@
+GL_LIBS = 
 INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
@@ -159,7 +159,7 @@
 fbg_DEPENDENCIES = $(FBG_SOUND_OBJECTS)
 fbgdatadir = $(pkgdatadir)/data
 dist_fbgdata_DATA = data/default.pk3 data/Arctic.pk3 $(FBGDATA_SOUND)
-docdir = $(prefix)/doc/fbg
+docdir = $(prefix)/share/doc/fbg
 doc_DATA = README COPYING
 
 distname = fbg-0.9
@@ -183,7 +183,7 @@
 @BUILD_STARTFBG_FALSE@fbglaunch_DEPENDENCIES =
 fbglaunch_LDFLAGS =
 
-DEFAULT_INCLUDES =  -I. -I$(srcdir)
+DEFAULT_INCLUDES =  -I. -I$(srcdir) $(CFLAGS)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 @AMDEP_TRUE@DEP_FILES = src/$(DEPDIR)/fbgBlock.Po \
