--- configure.orig	Tue Dec 14 12:01:32 2004
+++ configure	Tue Dec 14 12:02:00 2004
@@ -5944,8 +5944,8 @@
 
 
 HAVE_OPENGL=no
-echo "$as_me:$LINENO: checking for GL/gl.h" >&5
-echo $ECHO_N "checking for GL/gl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for gl.h" >&5
+echo $ECHO_N "checking for gl.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_gl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -5958,7 +5958,7 @@
 /* end confdefs.h.  */
 $SYS_GL_PRE_INCLUDES
 
-#include <GL/gl.h>
+#include <gl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -5984,8 +5984,8 @@
 echo "$as_me:$LINENO: result: $ac_cv_header_GL_gl_h" >&5
 echo "${ECHO_T}$ac_cv_header_GL_gl_h" >&6
 if test $ac_cv_header_GL_gl_h = yes; then
-  echo "$as_me:$LINENO: checking for GL/glu.h" >&5
-echo $ECHO_N "checking for GL/glu.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for glu.h" >&5
+echo $ECHO_N "checking for glu.h... $ECHO_C" >&6
 if test "${ac_cv_header_GL_glu_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -5998,7 +5998,7 @@
 /* end confdefs.h.  */
 $SYS_GL_PRE_INCLUDES
 
-#include <GL/glu.h>
+#include <glu.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -6761,13 +6761,13 @@
 test "$prefix" = "NONE" && prefix=/usr/local
 test "$exec_prefix" = "NONE" && exec_prefix=$prefix
 
-eval FBGDATADIR="$prefix/games/$PACKAGE"
+eval FBGDATADIR="$prefix/share/$PACKAGE"
 cat >>confdefs.h <<_ACEOF
 #define FBGDATADIR "$FBGDATADIR"
 _ACEOF
 
 
-datadir="\$(prefix)/games"
+datadir="\$(prefix)/share"
 
 
           ac_config_files="$ac_config_files Makefile"
