--- Makefile.orig	2006-01-12 10:55:15.000000000 -0700
+++ Makefile	2008-11-15 15:43:03.000000000 -0700
@@ -28,13 +28,10 @@
     #CPPFLAGS += -DUSE_IOCTL_FREEBSD
   endif
   ifeq ($(system),Darwin)
-    CPPFLAGS += -DUSE_MACOS_TYPE -DUSE_IOCTL_DARWIN
+    CPPFLAGS += -DUSE_MACOS_TYPE -DUSE_IOCTL_DARWIN @@UNIVERSAL_CPPFLAGS@@
+    CFLAGS   += @@UNIVERSAL_CFLAGS@@
+    LDFLAGS  += @@UNIVERSAL_LDFLAGS@@
     LIBS     += -framework CoreServices
-    ifeq (/Developer/SDKs/MacOSX10.4u.sdk,$(wildcard /Developer/SDKs/MacOSX10.4u.sdk))
-      CPPFLAGS += -isysroot /Developer/SDKs/MacOSX10.4u.sdk
-      CFLAGS   += -arch i386 -arch ppc
-      LDFLAGS  += -arch i386 -arch ppc -Wl,-syslibroot,/Developer/SDKs/MacOSX10.4u.sdk
-    endif
   endif
   ifeq ($(system),AmigaOS)
     CC       += -noixemul
