https://github.com/geany/geany/commit/d11f9a51b939bf39c3c1676ab823147d460ede75
--- src/Makefile.in.orig	2012-06-18 09:13:57.000000000 -0500
+++ src/Makefile.in	2012-10-24 21:28:48.000000000 -0500
@@ -468,7 +468,6 @@
 @MINGW_TRUE@			-DGTK \
 @MINGW_TRUE@			-DG_LOG_DOMAIN=\""Geany"\"
 
-@MINGW_FALSE@geany_LDFLAGS = -Wl,--export-dynamic
 @MINGW_TRUE@geany_LDFLAGS = -mwindows -mms-bitfields
 @MINGW_TRUE@WINDRES = $(host_alias)-windres
 all: all-am
--- wscript.orig	2012-06-17 17:24:32.000000000 -0500
+++ wscript	2012-10-24 21:28:41.000000000 -0500
@@ -309,7 +309,6 @@
         source          = geany_sources,
         includes        = ['.', 'scintilla/include/', 'tagmanager/include/'],
         defines         = ['G_LOG_DOMAIN="Geany"', 'GEANY_PRIVATE'],
-        linkflags       = [] if is_win32 else ['-Wl,--export-dynamic'],
         uselib          = ['GTK', 'GLIB', 'GMODULE', 'GIO', 'GTHREAD', 'WIN32', 'SUNOS_SOCKET'],
         use             = ['scintilla', 'tagmanager', 'mio'])
 
