--- src/vte.c.orig	2014-04-16 08:20:13.000000000 -0700
+++ src/vte.c	2014-09-22 09:04:22.000000000 -0700
@@ -203,9 +203,9 @@
 		gint i;
 		const gchar *sonames[] = {
 #if GTK_CHECK_VERSION(3, 0, 0)
-			"libvte2_90.so", "libvte2_90.so.9",
+			"@@MP_PREFIX@@/lib/libvte2_90.dylib",
 #else
-			"libvte.so", "libvte.so.4", "libvte.so.8", "libvte.so.9",
+			"@@MP_PREFIX@@/lib/libvte.dylib",
 #endif
 			NULL
 		};
@@ -219,7 +219,7 @@
 	if (module == NULL)
 	{
 		vte_info.have_vte = FALSE;
-		geany_debug("Could not load libvte.so, embedded terminal support disabled");
+		geany_debug("Could not load libvte.dylib, embedded terminal support disabled");
 		return;
 	}
 	else
--- src/main.c.orig	2014-04-16 08:20:13.000000000 -0700
+++ src/main.c	2014-09-22 09:08:22.000000000 -0700
@@ -140,7 +140,7 @@
 	{ "no-session", 's', G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE, &cl_options.load_session, N_("Don't load the previous session's files"), NULL },
 #ifdef HAVE_VTE
 	{ "no-terminal", 't', 0, G_OPTION_ARG_NONE, &no_vte, N_("Don't load terminal support"), NULL },
-	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Filename of libvte.so"), NULL },
+	{ "vte-lib", 0, 0, G_OPTION_ARG_FILENAME, &lib_vte, N_("Filename of libvte.dylib"), NULL },
 #endif
 	{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose_mode, N_("Be verbose"), NULL },
 	{ "version", 'V', 0, G_OPTION_ARG_NONE, &show_version, N_("Show version and exit"), NULL },
