--- src/af/util/xp/ut_jpeg.cpp.orig	2016-11-20 14:52:27.000000000 +1100
+++ src/af/util/xp/ut_jpeg.cpp	2016-11-20 14:53:21.000000000 +1100
@@ -102,7 +102,7 @@
 	src->pub.next_input_byte = src->sourceBuf->getPointer (src->pos);
 	src->pub.bytes_in_buffer = src->sourceBuf->getLength ();
 	
-	return TRUE;
+	return (boolean)true;
 }
 
 /*
@@ -161,7 +161,7 @@
 	/* set the data source */
 	_JPEG_ByteBufSrc (&cinfo, pBB);
 
-	jpeg_read_header(&cinfo, TRUE);
+	jpeg_read_header(&cinfo, (boolean)true);
 	jpeg_start_decompress(&cinfo);
     iImageWidth = cinfo.output_width;
     iImageHeight = cinfo.output_height;
@@ -189,7 +189,7 @@
 	/* set the data source */
 	_JPEG_ByteBufSrc (&cinfo, pBB);
 
-	jpeg_read_header(&cinfo, TRUE);
+	jpeg_read_header(&cinfo, (boolean)true);
 	jpeg_start_decompress(&cinfo);
     
 	int row_stride = cinfo.output_width * cinfo.output_components;
