--- src/texture.c	2008-02-13 08:49:51.000000000 -0500
+++ src/texture.c	2010-03-11 15:34:17.000000000 -0500
@@ -39,14 +39,14 @@
 
 /* 
  This variable is used as a temporary global value which
- is used to communicate with qsort and psort since we have no
+ is used to communicate with qsort and p_sort since we have no
  provision from textur to pass additional arguments.
 
  In a multi-threaded version, we would need to protect this.
 */
 static ggobid *CurrentGGobi;
 gint
-psort (const void *arg1, const void *arg2)
+p_sort (const void *arg1, const void *arg2)
 {
   ggobid *gg = CurrentGGobi;
 
@@ -195,7 +195,7 @@
 
   indx = (gint *) g_malloc (ny * sizeof (gint));
 /*
- * gy is needed solely for the psort routine:  psort is used by
+ * gy is needed solely for the p_sort routine:  p_sort is used by
  * qsort to put an index vector in the order that yy will assume.
 */
   gg->p1d.gy = (gfloat *) g_malloc (ny * sizeof (gfloat));
@@ -208,7 +208,7 @@
 
   CurrentGGobi = gg;
 
-  qsort ((void *) indx, (gsize) ny, sizeof (gint), psort);
+  qsort ((void *) indx, (gsize) ny, sizeof (gint), p_sort);
   qsort ((void *) yy, (gsize) ny, sizeof (gfloat), fcompare);
   CurrentGGobi = NULL;
 /*
