--- xulrunner/app/Makefile.in.orig	2009-04-07 22:17:16.000000000 -0700
+++ xulrunner/app/Makefile.in	2009-04-07 22:22:16.000000000 -0700
@@ -87,12 +87,16 @@ CPPSRCS = nsXULRunnerApp.cpp
 ifeq ($(OS_ARCH),WINNT)
 CPPSRCS += nsRegisterGREWin.cpp
 else
-ifneq (,$(filter OS2 Darwin,$(OS_ARCH)))
+ifeq (OS2,$(OS_ARCH))
+CPPSRCS += nsRegisterGREVoid.cpp
+else
+ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 CPPSRCS += nsRegisterGREVoid.cpp
 else
 CPPSRCS += nsRegisterGREUnix.cpp
 endif
 endif
+endif
 
 LOCAL_INCLUDES += \
         -I$(topsrcdir)/toolkit/xre \
@@ -104,7 +108,7 @@ TK_LIBS := -framework Cocoa $(TK_LIBS)
 endif
 
 ifndef MOZ_ENABLE_LIBXUL
-ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
+ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 LIBS += $(DIST)/bin/XUL
 else
 EXTRA_DSO_LIBS += xul
@@ -202,8 +206,8 @@ LDFLAGS += /HEAP:0x40000
 endif
 endif
 
-ifneq (,$(filter-out OS2 WINNT Darwin BeOS,$(OS_ARCH)))
-
+ifneq (,$(filter-out OS2 WINNT BeOS,$(OS_ARCH)))
+ifneq (cocoa, $(MOZ_WIDGET_TOOLKIT))
 xulrunner:: mozilla.in Makefile.in Makefile $(DEPTH)/config/autoconf.mk
 	cat $< | sed -e "s|%MOZAPPDIR%|$(mozappdir)|" \
                 -e "s|%MOZ_USER_DIR%|.mozilla/xulrunner|" \
@@ -218,6 +222,7 @@ install:: xulrunner
 
 GARBAGE += xulrunner
 endif
+endif
 
 ifneq (,$(filter gtk gtk2,$(MOZ_WIDGET_TOOLKIT)))
 
