--- configure.ac	2003-08-09 20:04:33.000000000 +0300
+++ configure.ac	2011-05-21 22:54:12.000000000 +0300
@@ -89,6 +89,10 @@
 Freedroid to be able to play Ogg files (e.g. the Intro theme)
 --------------------------------------------------]))
 
+AC_CHECK_LIB([SDL_gfx], [zoomSurface],, AC_MSG_ERROR([
+--------------------------------------------------
+SDL_gfx is needed for Freedroid
+--------------------------------------------------]))
 
 # Checks for header files.
 AC_HEADER_STDC
--- src/Makefile.am	2003-08-09 20:06:44.000000000 +0300
+++ src/Makefile.am	2011-05-21 22:53:38.000000000 +0300
@@ -16,8 +16,7 @@
 	enemy.c input.c  main.c  view.c BFont.c defs.h \
 	map.h vars.h takeover.h global.h \
 	text.h proto.h struct.h system.h BFont.h maped.h \
-	SDL_rotozoom.c SDL_rotozoom.h highscore.c \
-	getopt.c getopt1.c getopt.h \
+	highscore.c getopt.c getopt1.c getopt.h \
 	$(win32_res)
 
 
