--- setup.py.orig	2012-03-19 22:15:57.000000000 +0400
+++ setup.py	2012-03-19 22:17:01.000000000 +0400
@@ -18,7 +18,10 @@
 
 
 execfile('imread/imread_version.py')
-long_description = file('docs/source/readme.rst').read()
+try:
+    long_description = file('docs/source/readme.rst').read()
+except:
+    long_description = 'imread: Image reading library'
 
 undef_macros=[]
 if os.environ.get('DEBUG'):
