--- setup.py.orig	2013-06-20 16:28:23.000000000 +0400
+++ setup.py	2013-06-20 16:29:00.000000000 +0400
@@ -33,14 +33,6 @@
 include_dirs = []
 library_dirs = []
 
-for pth in ('/usr/local/include', '/usr/X11/include'):
-    if os.path.isdir(pth):
-        include_dirs.append(pth)
-
-for pth in ('/usr/local/lib', '/usr/X11/lib'):
-    if os.path.isdir(pth):
-        library_dirs.append(pth)
-
 extensions = {
     'imread._imread': [
         'imread/_imread.cpp',
