--- configure.orig	2015-11-26 16:20:24.000000000 +0100
+++ configure	2015-11-26 17:11:50.000000000 +0100
@@ -19464,7 +19464,6 @@ case "$host_os" in
 		else
 			developer_path="/Developer"
 			fi
-		OS_SPECIFIC_CFLAGS="-I${developer_path}/Headers/FlatCarbon"
 		OS_SPECIFIC_LINKS="-framework CoreAudio -framework AudioToolbox -framework CoreFoundation"
 		;;
 	mingw*)
--- man/sndfile-play.1.orig	2015-11-26 17:09:22.000000000 +0100
+++ man/sndfile-play.1	2015-11-26 17:08:28.000000000 +0100
@@ -22,7 +22,7 @@ sndio
 .It Solaris
 /dev/audio
 .It MacOSX 10.6
-Carbon and CoreAudio
+CoreAudio
 .It MacOSX 10.7
 AudioToolbox
 .It Win32
@@ -30,6 +30,5 @@ waveOut
 .El
 .Sh SEE ALSO
 .Lk http://www.mega-nerd.com/libsndfile/
-.\" FIXME: links to ALSA, OSS, sndio, Sun audio, MacOSX audio
 .Sh AUTHORS
 .An Erik de Castro Lopo Aq Mt erikd@mega-nerd.com
--- programs/sndfile-play.c.orig	2015-11-26 16:26:09.000000000 +0100
+++ programs/sndfile-play.c	2015-11-26 17:08:58.000000000 +0100
@@ -68,7 +68,6 @@
 	#elif (OSX_DARWIN_VERSION == 11)
 		#include <AudioToolbox/AudioToolbox.h>
 	#elif (OSX_DARWIN_VERSION > 0 && OSX_DARWIN_VERSION <= 10)
-		#include <Carbon.h>
 		#include <CoreAudio/AudioHardware.h>
 	#endif
 
@@ -648,7 +647,7 @@ macosx_play (int argc, char *argv [])
 #endif /* OSX_DARWIN_VERSION == 11 */
 
 #if (OSX_DARWIN_VERSION > 0 && OSX_DARWIN_VERSION <= 10)
-/* MacOSX 10.6 or earlier, use Carbon and AudioHardware API */
+/* MacOSX 10.6 or earlier, use AudioHardware API */
 
 typedef struct
 {	AudioStreamBasicDescription		format ;
