--- configure.orig	2017-04-26 10:10:42.000000000 +0200
+++ configure	2017-04-26 10:10:56.000000000 +0200
@@ -20828,7 +20828,7 @@ rm -f core conftest.err conftest.$ac_obj
 	common_flags="-Wcast-align -Wcast-qual -Wshadow -Wwrite-strings -Wundef -Wuninitialized -Winit-self"
 
 	# -Winline -Wconversion "
-	CFLAGS="$CFLAGS     $common_flags -Wbad-function-cast -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Waggregate-return -Wvla"
+	CFLAGS="$CFLAGS     $common_flags -Wbad-function-cast -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Waggregate-return"
 	CXXFLAGS="$CXXFLAGS $common_flags -Wctor-dtor-privacy -Wnon-virtual-dtor -Woverloaded-virtual -Wreorder -Wsign-promo"
 
 	if test "x$enable_gcc_opt" = "xno" ; then
