--- configure.orig	2010-07-29 02:31:57.000000000 +0200
+++ configure	2011-01-31 23:29:54.000000000 +0100
@@ -4373,6 +4373,9 @@
 
 
 echocheck "X11"
+if test "$_x11" = yes ; then
+  libs_mplayer="$libs_mplayer -lXext -lX11"
+fi
 if test "$_x11" = auto && test "$_x11_headers" = yes ; then
   cat > $TMPC <<EOF
 #include <X11/Xlib.h>
@@ -7104,6 +7107,7 @@
 if test "$_qtx" = yes ; then
     def_qtx='#define CONFIG_QTX_CODECS 1'
     win32 && _qtx_codecs_win32=yes && def_qtx_win32='#define CONFIG_QTX_CODECS_WIN32 1'
+    darwin && extra_ldflags="$extra_ldflags -framework Carbon -framework QuickTime" && def_quicktime='#define CONFIG_QUICKTIME 1'
     codecmodules="qtx $codecmodules"
     darwin || win32 || _qtx_emulation=yes
 else
