--- makefile.orig	2025-03-07 10:36:59.000000000 -0500
+++ makefile	2025-03-07 10:37:20.000000000 -0500
@@ -50,14 +50,12 @@
 release-windows release-darwin: $(GENIE)
 	$(GENIE) release
 	$(SILENT) $(MAKE) -C build/$(PROJECT_TYPE).$(OS) clean all
-	$(SILENT) git checkout src/host/version.h
 
 release-linux: $(GENIE)
 	$(SILENT) $(GENIE) release
 	$(SILENT) $(MAKE) -C build/$(PROJECT_TYPE).darwin  clean all CC=arm64-apple-darwin24.1-clang CFLAGS="-arch arm64" LDFLAGS="-arch arm64"
 	$(SILENT) $(MAKE) -C build/$(PROJECT_TYPE).linux   clean all
 	$(SILENT) $(MAKE) -C build/$(PROJECT_TYPE).windows clean all CC=x86_64-w64-mingw32-gcc
-	$(SILENT) git checkout src/host/version.h
 
 release: release-$(OS)
 
