--- scripts/release.lua.orig	2021-11-25 13:26:19.000000000 -0500
+++ scripts/release.lua	2021-11-25 13:25:52.000000000 -0500
@@ -15,16 +15,16 @@
 
 	print("Updating version number...")
 
-	local f = io.popen("git rev-list --count HEAD")
-	local rev = string.match(f:read("*a"), ".*%S")
-	f:close()
-	f = io.popen("git log --format=format:%H -1")
-	local sha1 = f:read("*a")
-	f:close()
-	io.output("src/host/version.h")
-	io.write("#define VERSION " ..rev .. "\n")
-	io.write("#define VERSION_STR \"version " ..rev .. " (commit " .. sha1 .. ")\"\n")
-	io.close()
+--	local f = io.popen("git rev-list --count HEAD")
+--	local rev = string.match(f:read("*a"), ".*%S")
+--	f:close()
+--	f = io.popen("git log --format=format:%H -1")
+--	local sha1 = f:read("*a")
+--	f:close()
+--	io.output("src/host/version.h")
+--	io.write("#define VERSION " ..rev .. "\n")
+--	io.write("#define VERSION_STR \"version " ..rev .. " (commit " .. sha1 .. ")\"\n")
+--	io.close()
 
 
 	print("Updating embedded scripts...")
