--- test/Multiyearstat.test.in.orig	2022-08-11 09:06:17.000000000 +0200
+++ test/Multiyearstat.test.in	2022-10-17 22:25:01.000000000 +0200
@@ -18,6 +18,10 @@
 for TYPE in $OPERTYPES; do
     for STAT in $STATS; do
         if [ -z "$1" ] || [ $NTEST == $1 ]; then
+        if [ "$TYPE" = "yseas" ] ; then
+            CDOTEST="${TYPE}$STAT"
+            test $RSTAT -eq 0 && echo "ok $NTEST - $CDOTEST # SKIP test not enabled"
+        else
             RSTAT=0
             #    if [ $TYPE = yday -o $TYPE = yhour ] ; then
             if [ $TYPE = yhour ] ; then
@@ -46,6 +50,7 @@
             test $RSTAT -eq 0 || echo "not ok $NTEST - $CDOTEST"
             rm -f $OFILE
         fi
+        fi
         (( NTEST+=1 ))
     done
 done
@@ -55,6 +60,10 @@
 for TYPE in $OPERTYPES; do
     for STAT in $STATS; do
         if [ -z "$1" ] || [ $NTEST == $1 ]; then
+        if [ "$TYPE" = "yseas" ] ; then
+            CDOTEST="${TYPE}$STAT with missing value"
+            test $RSTAT -eq 0 && echo "ok $NTEST - $CDOTEST # SKIP test not enabled"
+        else
             RSTAT=0
             #    if [ $TYPE = yday -o $TYPE = yhour ] ; then
             if [ $TYPE = yhour ] ; then
@@ -83,6 +92,7 @@
             test $RSTAT -eq 0 || echo "not ok $NTEST - $CDOTEST"
             rm -f $OFILE
         fi
+        fi
         (( NTEST+=1 ))
     done
 done
