--- src/getRSS.c.orig	2022-11-27 19:13:38.000000000 -0800
+++ src/getRSS.c	2022-11-27 19:14:59.000000000 -0800
@@ -16,6 +16,14 @@
 #if defined(__APPLE__) && defined(__MACH__)
 #include <mach/mach.h>
 
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1080
+#  define MACH_TASK_BASIC_INFO_COUNT TASK_BASIC_INFO_COUNT
+#  define mach_task_basic_info_data_t task_basic_info_data_t
+#  define MACH_TASK_BASIC_INFO TASK_BASIC_INFO
+#  define mach_task_basic_info task_basic_info
+#endif
+
 #elif (defined(_AIX) || defined(__TOS__AIX__)) || (defined(__sun__) || defined(__sun) || defined(sun) && (defined(__SVR4) || defined(__svr4__)))
 #include <fcntl.h>
 #include <procfs.h>
