--- Makefile.orig	2009-03-16 11:48:17.000000000 -0500
+++ Makefile	2009-03-22 20:21:57.000000000 -0500
@@ -75,8 +75,8 @@
 	rm -r *.aux *.log *.out
 
 install:
-	test -d "$(DESTDIR)$(libdir)/pure" || mkdir -p "$(DESTDIR)$(libdir)/pure"
-	cp csv.pure csv$(DLL) pure-csv.html "$(DESTDIR)$(libdir)/pure"
+	test -d "$(DESTDIR)`readlink $(libdir)/pure`" || mkdir -p "$(DESTDIR)`readlink $(libdir)/pure`"
+	cp csv.pure csv$(DLL) pure-csv.html "$(DESTDIR)`readlink $(libdir)/pure`"
 
 uninstall:
 	rm -f "$(DESTDIR)$(libdir)/pure/csv.pure" "$(DESTDIR)$(libdir)/pure/csv$(DLL)" "$(DESTDIR)$(libdir)/pure/pure-csv.html"
