diff --git docker/Dockerfile docker/Dockerfile
index 375ac746e2..1c0a242c2f 100644
--- docker/Dockerfile
+++ docker/Dockerfile
@@ -19,7 +19,7 @@ RUN export DEBIAN_FRONTEND=noninteractive \
 # Update local opam repository
 USER opam
 WORKDIR /home/opam/opam-repository
-RUN git fetch origin master && git checkout a4ce2314f5990609c9ee2133734cc49d9606646f && opam update
+RUN git fetch origin master && git checkout b69889513d1dfe730791f2fffabe24ca1d944b5f && opam update
 
 # Initialize OPAM
 WORKDIR /home/opam
diff --git dune-project dune-project
index ce31cce174..576bf4eacf 100644
--- dune-project
+++ dune-project
@@ -39,7 +39,7 @@
     ppx_deriving
     ppx_import
     stdlib-shims
-    (unidecode (= "0.2.0"))
+    (unidecode (>= "0.5.0"))
     uri
     uucp
     uunf
diff --git geneweb.opam geneweb.opam
index 5287f47e77..47b9be302f 100644
--- geneweb.opam
+++ geneweb.opam
@@ -33,7 +33,7 @@ depends: [
   "ppx_deriving"
   "ppx_import"
   "stdlib-shims"
-  "unidecode" {= "0.2.0"}
+  "unidecode" {>= "0.5.0"}
   "uri"
   "uucp"
   "uunf"
diff --git geneweb_colab.ipynb geneweb_colab.ipynb
index 3f44a30ad3..a6bd87f78f 100644
--- geneweb_colab.ipynb
+++ geneweb_colab.ipynb
@@ -75,7 +75,7 @@
         "!opam -y init --bare --shell-setup\n",
         "!opam switch create 4.14.2\n",
         "!eval $(opam env --switch=4.14.2)\n",
-        "!opam install -y calendars.2.0.0 camlp-streams camlp5 cppo decompress digestif dune jingoo markup oUnit pp_loc ppx_blob ppx_deriving ppx_import stdlib-shims logs logs-syslog unidecode.0.2.0 uri uucp uutf uunf yojson"
+        "!opam install -y calendars.2.0.0 camlp-streams camlp5 cppo decompress digestif dune jingoo markup oUnit pp_loc ppx_blob ppx_deriving ppx_import stdlib-shims logs logs-syslog unidecode.0.5.0 uri uucp uutf uunf yojson"
       ]
     },
     {
diff --git nix/unidecode.nix nix/unidecode.nix
index 5243d29ece..ac3c0906c1 100644
--- nix/unidecode.nix
+++ nix/unidecode.nix
@@ -3,11 +3,11 @@
 
 buildDunePackage {
   pname = "unidecode";
-  version = "0.4.0";
+  version = "0.5.0";
   duneVersion = "3";
 
   src = fetchTarball {
     url = "https://github.com/geneweb/unidecode/archive/refs/tags/v0.4.0.tar.gz";
-    sha256 = "1fahjzgr2zw8cplw5azd39z3m0dfv7wjbrzpb8953nadyzk87cjl";
+    sha256 = "87ccf6a1da1aa7a8d3c0278940934df7c887a50d28485b668dea1d654d066631";
   };
 }