--- Makefile.PL.orig	2016-08-14 01:38:28.000000000 -0700
+++ Makefile.PL	2016-08-14 01:39:09.000000000 -0700
@@ -6,8 +6,8 @@
     'VERSION_FROM'	=> 'Cisco.pm', # finds $VERSION
     'PREREQ_PM'		=> { Net::Telnet => 3.02,
 			     Term::ReadKey => 2,
-			     Test::More => undef,
-			     Cwd => undef,
+			     Test::More => 0,
+			     Cwd => 0,
 			   }, # e.g., Module::Name => 1.1
      ($] ge '5.005') ? (
 			'AUTHOR' => 'Joshua Keroes (joshua@cpan.org)',
