--- configure.orig	2026-01-05 06:47:16
+++ configure	2026-01-22 12:43:53
@@ -5504,7 +5504,7 @@
     sitedir = sysconfig.get_path('purelib', vars={'base':'$am_py_prefix'})
 else:
   from distutils import sysconfig
-  sitedir = sysconfig.get_python_lib(0, 0, prefix='$am_py_prefix')
+  sitedir = sysconfig.get_python_lib()
 sys.stdout.write(sitedir)"`
    #
    case $am_cv_python_pythondir in
@@ -5552,7 +5552,7 @@
     sitedir = sysconfig.get_path('platlib', vars={'platbase':'$am_py_exec_prefix'})
 else:
   from distutils import sysconfig
-  sitedir = sysconfig.get_python_lib(1, 0, prefix='$am_py_exec_prefix')
+  sitedir = sysconfig.get_python_lib()
 sys.stdout.write(sitedir)"`
    #
    case $am_cv_python_pyexecdir in
