--- mk/beforeauto.mk.in.orig	2014-04-17 10:40:15.000000000 -0500
+++ mk/beforeauto.mk.in	2014-09-08 03:21:00.000000000 -0500
@@ -1053,7 +1053,7 @@
 OMNITHREAD_POSIX_CPPFLAGS = -DPthreadDraftVersion=10 \
                             -DPthreadSupportThreadPriority -DNoNanoSleep
 
-CXXOPTIONS = -fno-common -bind_at_load @EXTRA_GCC_CXXFLAGS@
+CXXOPTIONS = -fno-common @EXTRA_GCC_CXXFLAGS@
 SHAREDLIB_SUFFIX = dylib
 
 SharedLibraryFullNameTemplate = lib$$1$$2.$$3.$$4.$(SHAREDLIB_SUFFIX)
@@ -1064,8 +1064,7 @@
                                          -flat_namespace \
                                          -undefined suppress
 PythonLibraryPlatformLinkFlagsTemplate = -bundle -flat_namespace \
-                                         -undefined suppress \
-                                         -bind_at_load
+                                         -undefined suppress
 PythonSHAREDLIB_SUFFIX = so
 
 
