--- buildtools/config.py.orig
+++ buildtools/config.py
@@ -840,7 +840,7 @@
                 line = line.strip()
                 endPos = line.rfind(' (')
                 curName = line[:endPos]
-                newName = '@loader_path/' + os.path.basename(curName)
+                newName = '@loader_path/../../../' + os.path.basename(curName)
                 cmd = 'install_name_tool -change %s %s %s' % (curName, newName, filename)
                 os.system(cmd)
 
@@ -1074,13 +1074,9 @@
         copy_file(opj(cfg.WXDIR, 'docs', filename), opj('license',filename),
                       update=1, verbose=1)
 
-    # Get the sip license too
-    copy_file(opj('sip', 'siplib', 'LICENSE'), opj('license', 'sip-license.txt'),
-              update=1, verbose=1)
-
     # Combine the relevant files into a single LICENSE.txt file
     text = ''
-    for filename in ['preamble.txt', 'licence.txt', 'lgpl.txt', 'sip-license.txt']:
+    for filename in ['preamble.txt', 'licence.txt', 'lgpl.txt']:
         with open(opj('license', filename), 'r', encoding='utf-8') as f:
             text += f.read() + '\n\n'
     with open('LICENSE.txt', 'w') as f:

