--- fract4d/fractconfig.py.orig	2008-02-14 01:04:32.000000000 +1100
+++ fract4d/fractconfig.py	2008-11-01 21:43:41.000000000 +1100
@@ -212,7 +212,7 @@ class DarwinConfig(T):
         return "open %s"
 
     def get_default_compiler_options(self):
-        return "-fPIC -DPIC -D_REENTRANT -O2 -dynamiclib -flat_namespace -undefined suppress -ffast-math"
+        return "-fPIC -DPIC -D_REENTRANT -O2 -dynamiclib -undefined dynamic_lookup -ffast-math"
         
 config_file = "~/.gnofract4d"
 if sys.platform[:6] == "darwin":
