diff --git a/hw/xquartz/GL/visualConfigs.c b/hw/xquartz/GL/visualConfigs.c
index 69b18f6a0..79a4cce58 100644
--- a/hw/xquartz/GL/visualConfigs.c
+++ b/hw/xquartz/GL/visualConfigs.c
@@ -59,7 +59,7 @@
 /* Based originally on code from indirect.c which was based on code from i830_dri.c. */
 __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber) {
     int numConfigs = 0;
-    __GLXconfig *visualConfigs, *c;
+    __GLXconfig *visualConfigs, *c, *last_c;
     struct glCapabilities caps;
     struct glCapabilitiesConfig *conf;
     int stereo, depth, aux, buffers, stencil, accum, color, msample;
@@ -113,7 +113,7 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
     if(numConfigsPtr)
         *numConfigsPtr = numConfigs;
 
-    visualConfigs = calloc(sizeof(*visualConfigs), numConfigs);
+    visualConfigs = calloc(sizeof(*visualConfigs),1);
 
     if(NULL == visualConfigs) {
         ErrorF("xcalloc failure when allocating visualConfigs\n");
@@ -122,6 +122,7 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
     }
 
     c = visualConfigs; /* current buffer */
+    int confCount = 0;
     for(conf = caps.configurations; conf; conf = conf->next) {
         for(stereo = 0; stereo < (conf->stereo ? 2 : 1); ++stereo) {
             for(aux = 0; aux < (conf->aux_buffers ? 2 : 1); ++aux) {
@@ -134,10 +135,12 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
                                 for(depth = 0; depth < conf->total_depth_buffer_depths; ++depth) {
                                     for(msample = 0; msample < (conf->multisample_buffers + 1); ++msample) {
 
+                                        ++confCount;
+
                                         // Global
                                         c->visualID = -1;
                                         c->visualType = GLX_TRUE_COLOR;
-                                        c->next = c + 1;
+                                        c->next = calloc(sizeof(*visualConfigs),1);
 
                                         c->level = 0;
                                         c->indexBits = 0;
@@ -260,6 +263,7 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
                                         /* EXT_framebuffer_sRGB */
                                         c->sRGBCapable = GL_FALSE;
 
+                                        last_c = c;
                                         c = c->next;
                                     }
                                 }
@@ -271,10 +275,11 @@ __GLXconfig *__glXAquaCreateVisualConfigs(int *numConfigsPtr, int screenNumber)
         }
     }
 
-    (c-1)->next = NULL;
+    free(last_c->next);
+    last_c->next = NULL;
 
-    if (c - visualConfigs != numConfigs) {
-        FatalError("numConfigs calculation error in setVisualConfigs!  numConfigs is %d  i is %d\n", numConfigs, (int)(c - visualConfigs));
+    if (confCount != numConfigs) {
+        FatalError("numConfigs calculation error in setVisualConfigs!  numConfigs is %d count is %d\n", numConfigs, confCount);
     }
 
     freeGlCapabilities(&caps);
