From 15fc56addcc906592af7c0f4c0a5ac906fa5c389 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@freedesktop.org>
Date: Mon, 09 Nov 2009 04:25:42 +0000
Subject: XQuartz: Buildfix for Leopard and older

Signed-off-by: Jeremy Huddleston <jeremyhu@freedesktop.org>
---
diff --git a/hw/xquartz/X11Controller.h b/hw/xquartz/X11Controller.h
index bb28447..9e16f75 100644
--- a/hw/xquartz/X11Controller.h
+++ b/hw/xquartz/X11Controller.h
@@ -55,7 +55,11 @@ typedef unsigned int NSUInteger;
 #endif
 #endif
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 @interface X11Controller : NSObject <NSTableViewDataSource>
+#else
+@interface X11Controller : NSObject
+#endif
 {
     IBOutlet NSPanel *prefs_panel;
 
diff --git a/hw/xquartz/quartzKeyboard.c b/hw/xquartz/quartzKeyboard.c
index 90f6610..c504762 100644
--- a/hw/xquartz/quartzKeyboard.c
+++ b/hw/xquartz/quartzKeyboard.c
@@ -40,6 +40,7 @@
 #define HACK_MISSING 1
 #define HACK_KEYPAD 1
 
+#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
--
cgit v0.8.2
