--- makefile.orig	2013-03-27 20:08:01.000000000 +0100
+++ makefile	2013-03-27 22:33:43.000000000 +0100
@@ -26,12 +26,12 @@
 CC = gcc
 
 # For Linux
-DLLFLAGS = -shared -fPIC
-ENV = 
+#DLLFLAGS = -shared -fPIC
+#ENV = 
 
 # For Mac OS
 # ENV = MACOSX_DEPLOYMENT_TARGET=10.4
-# DLLFLAGS = -bundle -undefined dynamic_lookup
+ DLLFLAGS = -bundle -undefined dynamic_lookup
 
 
 FILES = lpvm.o lpcap.o lptree.o lpcode.o lpprint.o
