--- DESCRIPTION	2023-10-27 01:00:02.000000000 +0800
+++ DESCRIPTION	2023-10-27 01:55:45.000000000 +0800
@@ -22,7 +22,7 @@
 Imports: caret, data.table, doParallel, foreach, ggplot2, glmnet,
         matrixStats, matrixTests, methods, parallel, pROC, Rfast, rlang
 RoxygenNote: 7.2.3
-Suggests: Boruta, CORElearn, fastshap, gbm, ggbeeswarm, ggpubr, hsstan,
+Suggests: CORElearn, fastshap, gbm, ggbeeswarm, ggpubr, hsstan,
         mda, mlbench, pbapply, randomForest, ranger, RcppEigen,
         rmarkdown, knitr, SuperLearner
 VignetteBuilder: knitr
