--- advance/menu.mak.orig	2018-09-08 22:58:20.000000000 +1000
+++ advance/menu.mak	2021-11-29 09:54:04.000000000 +1100
@@ -510,7 +510,7 @@
 
 $(MENUOBJ)/%.o: $(srcdir)/advance/%.cc
 	$(ECHO) $@ $(MSG)
-	$(CXX) $(CFLAGS) $(MENUCFLAGS) -c $< -o $@
+	$(CXX) $(CXXFLAGS) $(CFLAGS) $(MENUCFLAGS) -c $< -o $@
 
 $(MENUOBJ)/%.o: $(srcdir)/advance/%.c
 	$(ECHO) $@ $(MSG)
@@ -530,7 +530,7 @@
 
 $(MENUOBJ)/advmenu$(EXE) : $(sort $(MENUOBJDIRS)) $(MENUOBJS)
 	$(ECHO) $@ $(MSG)
-	$(LDXX) $(MENUOBJS) $(MENULDFLAGS) $(LDFLAGS) $(MENULIBS) $(LIBS) -o $@
+	$(LDXX) $(CXXFLAGS) $(MENUOBJS) $(MENULDFLAGS) $(LDFLAGS) $(MENULIBS) $(LIBS) -o $@
 ifeq ($(CONF_PERF),yes)
 	$(RM) advmenup$(EXE)
 	$(LN_S) $@ advmenup$(EXE)
--- Makefile.in.orig	2021-11-29 09:46:46.000000000 +1100
+++ Makefile.in	2021-11-29 09:57:14.000000000 +1100
@@ -91,6 +91,7 @@
 LD_FOR_BUILD=@CC_FOR_BUILD@
 CXX_FOR_BUILD=@CXX_FOR_BUILD@
 LDXX_FOR_BUILD=@CXX_FOR_BUILD@
+CXXFLAGS=@CXXFLAGS@
 EXE=@EXE@
 EXE_FOR_BUILD=@EXE_FOR_BUILD@
 SDLCFLAGS=@SDLCFLAGS@
