diff --git a/build/cmake/config.cmake b/build/cmake/config.cmake
index 90277f3955..966be07ec8 100644
--- a/build/cmake/config.cmake
+++ b/build/cmake/config.cmake
@@ -33,7 +33,12 @@ macro(wx_get_dependencies var lib)
         get_target_property(deps ${lib} LINK_LIBRARIES)
         foreach(dep IN LISTS deps)
             if(TARGET ${dep})
-                get_target_property(dep_name ${dep} OUTPUT_NAME)
+                get_target_property(dep_type ${dep} TYPE)
+                if (dep_type STREQUAL "INTERFACE_LIBRARY")
+                    get_target_property(dep_name ${dep} INTERFACE_OUTPUT_NAME)
+                else()
+                    get_target_property(dep_name ${dep} OUTPUT_NAME)
+                endif()
                 set(dep_name "-l${dep_name}")
             else()
                 get_filename_component(dep_name ${dep} NAME)
