diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8909de65d2d6d8aebbff41a0e22f9b152def8075..44ab8c37e046177d04379a1e6b4fead2e6bbb8db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,3 +72,10 @@ include(build/cmake/main.cmake)
 if(wxBUILD_SAMPLES AND wxUSE_GUI)
     set_directory_properties(PROPERTIES VS_STARTUP_PROJECT minimal)
 endif()
+
+if( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
+      # When accessibility is enabled, the build will fail in "wx/chkconf.h"
+      # since wxRegex compile defines do not include __WXOSX_COCOA__.  So,
+      # add it here.
+      target_compile_definitions( wxregex PRIVATE "__WXOSX_COCOA__" )
+endif()
