--- DESCRIPTION.orig	2023-02-23 20:10:02.000000000 +0800
+++ DESCRIPTION	2023-02-24 07:24:10.000000000 +0800
@@ -27,7 +27,7 @@
 Depends: R (>= 3.0)
 Imports: future.apply, grDevices, graphics, methods, numDeriv,
         progressr, stats, survival, SQUAREM, utils
-Suggests: KernSmooth, Matrix, Rgraphviz, data.table, ellipse, fields,
+Suggests: KernSmooth, Matrix, data.table, ellipse, fields,
         geepack, graph, knitr, bookdown, rmarkdown, igraph (>= 0.6),
         lavaSearch2, lme4, mets (>= 1.1), nlme, optimx, polycor,
         quantreg, rgl, R.rsp (>= 0.40), targeted (>= 0.2), testthat (>=
