https://sourceforge.net/p/quantlib/bugs/181/
--- acinclude.m4.orig	2015-11-16 16:35:18.000000000 -0600
+++ acinclude.m4	2015-12-26 04:51:17.000000000 -0600
@@ -179,7 +179,7 @@
  AC_REQUIRE([AC_PROG_CC])
  ql_original_LIBS=$LIBS
  boost_thread_found=no
- boost_thread_lib="-lboost_thread -lboost_system -lpthread"
+ boost_thread_lib="-lboost_thread-mt -lboost_system-mt -lpthread"
  LIBS="$ql_original_LIBS $boost_thread_lib"
  AC_LINK_IFELSE([AC_LANG_SOURCE(
      [@%:@include <boost/thread/locks.hpp>
