https://sourceforge.net/p/quantlib/bugs/181/
--- Examples/FRA/Makefile.am.orig	2015-11-16 16:35:18.000000000 -0600
+++ Examples/FRA/Makefile.am	2015-12-26 13:18:55.000000000 -0600
@@ -8,7 +8,7 @@
 noinst_PROGRAMS = FRA
 endif
 FRA_SOURCES = FRA.cpp
-FRA_LDADD = ../../ql/libQuantLib.la -lboost_thread ${BOOST_THREAD_LIB}
+FRA_LDADD = ../../ql/libQuantLib.la ${BOOST_THREAD_LIB}
 
 EXTRA_DIST = \
     FRA.dev \
