--- rtgui/options.cc	2023-05-28 06:24:25.000000000 +0800
+++ rtgui/options.cc	2024-04-26 12:49:58.000000000 +0800
@@ -587,9 +587,9 @@
     }
 
 #elif defined __APPLE__
-    rtSettings.iccDirectory = "/library/ColorSync/Profiles/Displays";
+    rtSettings.iccDirectory = "/Library/ColorSync/Profiles/Displays";
 #else
-    rtSettings.iccDirectory = "/usr/share/color/icc";
+    rtSettings.iccDirectory = "@PREFIX@/share/rawtherapee/color/icc";
 #endif
 //   rtSettings.viewingdevice = 0;
 //   rtSettings.viewingdevicegrey = 3;
@@ -2798,11 +2798,7 @@
         }
 
 #else
-    #ifdef __APPLE__
-        rtdir = Glib::build_filename(Glib::ustring(g_get_home_dir()), "/Library/Application Support/", Glib::ustring(CACHEFOLDERNAME), "/config/");
-    #else
         rtdir = Glib::build_filename(Glib::ustring(g_get_user_config_dir()), Glib::ustring(CACHEFOLDERNAME));
-    #endif
 #endif
     }
 
@@ -2841,11 +2837,7 @@
 #ifdef _WIN32
         cacheBaseDir = Glib::build_filename(rtdir, "cache");
 #else
-    #ifdef __APPLE__
-        cacheBaseDir = Glib::build_filename(Glib::ustring(g_get_home_dir()), "/Library/Application Support/", Glib::ustring(CACHEFOLDERNAME), "/cache/");
-    #else
         cacheBaseDir = Glib::build_filename(Glib::ustring(g_get_user_cache_dir()), Glib::ustring(CACHEFOLDERNAME));
-    #endif
 #endif
     }
 
