--- src/platformsupport/fontdatabases/mac/coretext.pri.orig	2016-09-15 22:49:42.000000000 -0700
+++ src/platformsupport/fontdatabases/mac/coretext.pri	2017-07-07 07:53:26.000000000 -0700
@@ -1,7 +1,7 @@
 HEADERS += $$PWD/qcoretextfontdatabase_p.h $$PWD/qfontengine_coretext_p.h
 OBJECTIVE_SOURCES += $$PWD/qfontengine_coretext.mm $$PWD/qcoretextfontdatabase.mm
 
-contains(QT_CONFIG, freetype) {
+contains(QT_CONFIG, freetype)|contains(QT_CONFIG, system-freetype) {
     include($$QT_SOURCE_TREE/src/3rdparty/freetype_dependency.pri)
     HEADERS += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft_p.h
     SOURCES += $$QT_SOURCE_TREE/src/gui/text/qfontengine_ft.cpp
--- src/3rdparty/freetype_dependency.pri.orig	2016-12-01 01:17:04.000000000 -0700
+++ src/3rdparty/freetype_dependency.pri	2017-07-07 09:06:25.000000000 -0700
@@ -2,6 +2,6 @@
     INCLUDEPATH += $$PWD/freetype/include
     LIBS_PRIVATE += -L$$QT_BUILD_TREE/lib -lqtfreetype$$qtPlatformTargetSuffix()
 } else:contains(QT_CONFIG, system-freetype) {
-    # pull in the proper freetype2 include directory
-    include($$QT_SOURCE_TREE/config.tests/unix/freetype/freetype.pri)
+    INCLUDEPATH += __MACPORTS_PREFIX__/include/freetype2
+    LIBS_PRIVATE += -L__MACPORTS_PREFIX__/lib -lfreetype
 }
