--- build.sh.orig	2022-02-17 13:27:30.000000000 +0100
+++ build.sh	2022-02-27 15:07:28.000000000 +0100
@@ -36,6 +36,7 @@
 cp bazel-out/*-opt-*/bin/bazelisk-linux_amd64 bin/bazelisk-linux-amd64
 cp bazel-out/*-opt-*/bin/bazelisk-linux_arm64 bin/bazelisk-linux-arm64
 cp bazel-out/*-opt-*/bin/bazelisk-windows_amd64.exe bin/bazelisk-windows-amd64.exe
+./bazelisk clean --expunge
 rm -f bazelisk
 
 ### Build release artifacts using `go build`.
@@ -57,6 +58,8 @@
 "bin/bazelisk-$(uname -s | tr [:upper:] [:lower:])-amd64" version | grep "Bazelisk version"
 echo
 
+"bin/bazelisk-$(uname -s | tr [:upper:] [:lower:])-amd64" clean --expunge
+
 # Non-googlers: you should run this script with NPM_REGISTRY=https://registry.npmjs.org
 readonly REGISTRY=${NPM_REGISTRY:-https://wombat-dressing-room.appspot.com}
 echo "== NPM releases =="
