--- ./lib/util/charset/wscript_configure
+++ ./lib/util/charset/wscript_configure
@@ -41,6 +41,7 @@
                args='--cflags --libs',
                msg='Checking for icu-i18n',
                uselib_store='ICU_I18N'):
+    conf.env['LIB_ICU_I18N'].append('icuuc')
     for lib in conf.env['LIB_ICU_I18N']:
         conf.CHECK_LIB(lib, shlib=True, mandatory=True)
     conf.env['icu-libs'] = ' '.join(conf.env['LIB_ICU_I18N'])
