--- buildtools/wafsamba/samba_conftests.py.orig	2021-08-09 08:38:35.000000000 -0500
+++ buildtools/wafsamba/samba_conftests.py	2022-01-11 20:36:04.000000000 -0600
@@ -484,7 +484,7 @@
 
 
     if not conf.CONFIG_SET('XSLTPROC'):
-        conf.find_program('xsltproc', var='XSLTPROC')
+        conf.find_program('NOTxsltproc', var='XSLTPROC')
     if not conf.CONFIG_SET('XSLTPROC'):
         return False
 
--- ctdb/wscript.orig	2021-08-09 08:38:36.000000000 -0500
+++ ctdb/wscript	2022-01-11 20:36:04.000000000 -0600
@@ -1183,7 +1183,7 @@
     BASE_URL = 'http://docbook.sourceforge.net/release/xsl/current'
     MAN_XSL = '%s/manpages/docbook.xsl' % BASE_URL
     HTML_XSL = '%s/html/docbook.xsl' % BASE_URL
-    CMD_TEMPLATE = 'xsltproc --xinclude -o %s --nonet %s %s'
+    CMD_TEMPLATE = 'NOTxsltproc --xinclude -o %s --nonet %s %s'
     manpages = manpages_binary + manpages_misc + manpages_etcd + manpages_ceph
     for t in manpages:
         cmd = CMD_TEMPLATE % ('doc/%s' % t, MAN_XSL, 'doc/%s.xml' % t)
--- lib/ldb/wscript.orig	2021-10-27 07:53:55.000000000 -0500
+++ lib/ldb/wscript	2022-01-11 20:36:04.000000000 -0600
@@ -60,7 +60,7 @@
             conf.define('USING_SYSTEM_CMOCKA', 1)
 
     conf.RECURSE('lib/replace')
-    conf.find_program('xsltproc', var='XSLTPROC')
+    conf.find_program('NOTxsltproc', var='XSLTPROC')
     conf.SAMBA_CHECK_PYTHON()
     conf.SAMBA_CHECK_PYTHON_HEADERS()
 
--- wscript.orig	2021-08-26 04:01:35.000000000 -0500
+++ wscript	2022-01-11 20:36:04.000000000 -0600
@@ -166,7 +166,7 @@
     conf.RECURSE('examples/winexe')
 
     conf.SAMBA_CHECK_PERL(mandatory=True)
-    conf.find_program('xsltproc', var='XSLTPROC')
+    conf.find_program('NOTxsltproc', var='XSLTPROC')
 
     if conf.env.disable_python:
         if not (Options.options.without_ad_dc):
