--- configure.orig	2005-11-10 00:54:23.000000000 -0600
+++ configure	2023-07-30 02:34:47.000000000 -0500
@@ -3262,6 +3262,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -4014,6 +4015,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <signal.h>
+#include <stdlib.h>
 int main(int argc,char**argv) {
     sigset_t set;
     exit( (sigaddset(&set,-1) == -1) ? 0 : -1);
@@ -4061,6 +4063,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <signal.h>
+#include <stdlib.h>
 #undef sigaddset
 int main(int argc,char**argv) {
     sigset_t set;
@@ -4114,9 +4117,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <stdarg.h>
-int
-main ()
-{
+#include <stdlib.h>
 int test_va(int i, ...) {
     va_list ap,ac;
     int j=0;
@@ -4133,9 +4134,6 @@
 int main(int argc,char**argv) {
     exit( (test_va(1,2) == 2) ? 0 : -1);
 }
-  ;
-  return 0;
-}
 _ACEOF
 rm -f conftest.$ac_objext conftest$ac_exeext
 if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
