--- src/fd.c.orig	2005-11-07 12:45:58.000000000 +1100
+++ src/fd.c	2013-09-23 18:48:38.000000000 +1000
@@ -165,6 +165,7 @@ int UT_fd_init() {
     UT_mem_pool_create( FD_POOL, sizeof(UT_fd), 10 );
     UT_shl_cmd_create( "fds", "list selected file descriptors", 
             (UT_shlcmd *)UT_fds_shlcmd, NULL);
+    return 0;
 }
 
 /*******************************************************************************
