--- src/loop.c.orig	2005-10-24 07:12:36.000000000 +1000
+++ src/loop.c	2013-09-23 23:36:32.000000000 +1000
@@ -103,6 +103,9 @@ struct timeval UT_figure_select_timeout(
     return tv;
 }
 
+int UT_invoke_per_loop_cbs();
+int UT_signal_update();
+
 /******************************************************************************
  * UT_event_once()                                                       *
  * This function is one of the most important in libut. It executes one "loop"*
@@ -215,10 +218,12 @@ int UT_invoke_per_loop_cbs() {
 
     /* Free the copies we've made.*/
     UT_mem_free(PER_LOOP_CB_POOL, pscb_tmp, i);
+    return 0;
 }
 
 int UT_loop_init() {
     UT_mem_pool_create(PER_LOOP_CB_POOL, sizeof(UT_per_loop_cb_t), 2);
+    return 0;
 }
 
 /******************************************************************************
