--- src/mem_shl.c.orig	2005-11-09 16:43:31.000000000 +1100
+++ src/mem_shl.c	2013-09-23 23:14:22.000000000 +1000
@@ -28,6 +28,8 @@ NEGLIGENCE OR OTHERWISE) ARISING IN ANY 
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+#include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include "libut/ut_internal.h"
 
@@ -39,6 +41,9 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
 *******************************************************************************/
 extern UT_mem_global_type UT_mem_global;
 
+int UT_hash_dump();
+int UT_hash_shlcmd(int argc, char *argv[]);
+int UT_mem_usage(int argc, char *argv[]);
 
 int UT_mem_shlcmd(   int argc, char *argv[]) {
     UT_mem_pool *pool;
@@ -149,7 +154,7 @@ int UT_mem_shlcmd(   int argc, char *arg
     case 3:
             
             if (strcmp(argv[1],"-d") != 0) {
-                    UT_mem_usage(NULL);
+                    UT_mem_usage(0,NULL);
                     return SHL_OK;
             }
 
@@ -191,6 +196,7 @@ int UT_mem_usage(int argc,char*argv[]) {
     UT_shlf("mem -d poolname  --   detailed info on a pool\n");
     UT_shlf("mem -h           --   display #items/bucket histogram for each hash\n");
     UT_shlf("mem -H           --   ascii/hex dump of each hash by bucket\n");
+    return 0;
 }
 
 /*******************************************************************************
