--- tests/rqst/srvr.c.orig	2005-10-23 13:36:45.000000000 +1000
+++ tests/rqst/srvr.c	2013-09-24 00:25:45.000000000 +1000
@@ -29,10 +29,12 @@ int srvr_cb(int fd, char *name, int flag
         UT_fd_unreg(fd);
         close(fd);
     }
+    return 0;
 }
 
 int sig_cb(int signum) {
     UT_LOG(Info,"srvr got signal %d", signum);
+    return 0;
 }
 
 int main(int argc, char **argv) {
@@ -41,4 +43,5 @@ int main(int argc, char **argv) {
     UT_signal_reg( sig_cb );
     UT_net_listen("srvr", "*:2222", (UT_fd_cb*)srvr_cb, NULL);
     UT_event_loop();
+    return 0;
 }
