--- tests/rqst/upld.c.orig	2005-10-25 14:52:06.000000000 +1000
+++ tests/rqst/upld.c	2013-09-24 00:20:30.000000000 +1000
@@ -16,7 +16,7 @@ int upld_done_cb(char *name, void *data,
 
     if (flags & UTFD_IS_REQFAILURE) {
         UT_LOG(Error, "upld failed");
-        return;
+        return -1;
     }
 
     /* can assume REQSUCCESS since we did not have REQFALURE */
@@ -24,6 +24,7 @@ int upld_done_cb(char *name, void *data,
 
     /* the iobs are our responsibility to free */
     UT_iob_free(io[0]);
+    return 0;
 }
 
 UT_iob *file_to_iob(char *filename) {
@@ -110,4 +111,5 @@ int main(int argc, char **argv) {
     UT_shl_cmd_create("upld", "upload <file> to <host> <port>",upld_test_cmd,NULL);
     UT_shl_cmd_create("dns", "translate host:port to ip:port", dns_test_cmd, NULL);
 	UT_event_loop();
+	return 0;
 }
