--- src/var.c.orig	2005-10-24 11:40:32.000000000 +1000
+++ src/var.c	2013-09-23 23:46:28.000000000 +1000
@@ -45,11 +45,13 @@ UT_var_global_type UT_var_global = { 
     .vars  = NULL
 };
 
+void UT_var_shl_init();
 
 int UT_var_init(void) {
     UT_mem_pool_create( UTSHL_VARS, sizeof(UT_var), 10);
     UT_mem_pool_create( CBDATA, sizeof(UT_callback), 10 );
     UT_var_shl_init();
+    return 0;
 }
 
 UT_API int UT_var_create(char *name, char *desc, UT_var_type type, ...) {
@@ -174,6 +176,7 @@ int UT_var_set_cvar(UT_var *var) {
             UT_LOG(Fatal, "Unknown var type %d", var->type);
             break;
     }
+    return 0;
 }
 
 UT_API int UT_var_get(char *name, void *val) {
