--- build.sh
+++ build.sh
@@ -1,4 +1,4 @@
-#!/bin/sh -e
+#!/bin/sh -ex
 
 format=0
 console=0
@@ -38,14 +38,14 @@ while [ $# -gt 0 ]; do
 	esac
 done
 
-echo "Cleaning.."
+echo "Cleaning"
 rm -f ./bin/*
 
 # When clang-format is present
 
 if [ $format = 1 ];
 then
-	echo "Formatting.."
+	echo "Formatting"
 	clang-format -i src/devices/system.h
 	clang-format -i src/devices/system.c
 	clang-format -i src/devices/screen.h
@@ -66,8 +66,6 @@ then
 fi
 
 mkdir -p bin
-CC="${CC:-cc}"
-CFLAGS="${CFLAGS:--std=c89 -Wall -Wno-unknown-pragmas}"
 case "$(uname -s 2>/dev/null)" in
 MSYS_NT*|MINGW*) # MSYS2 on Windows
 	FILE_LDFLAGS="-liberty"
@@ -79,8 +77,7 @@ MSYS_NT*|MINGW*) # MSYS2 on Windows
 	fi
 	;;
 Darwin) # macOS
-	CFLAGS="${CFLAGS} -Wno-typedef-redefinition"
-	UXNEMU_LDFLAGS="$(brew --prefix)/lib/libSDL2.a $(sdl2-config --cflags --static-libs | sed -e 's/-lSDL2 //')"
+	UXNEMU_LDFLAGS="-L@PREFIX@/lib/ $(sdl2-config --cflags --libs)"
 	;;
 Linux|*)
 	UXNEMU_LDFLAGS="-L/usr/local/lib $(sdl2-config --cflags --libs)"
@@ -93,14 +90,13 @@ then
 	CFLAGS="${CFLAGS} -DDEBUG -Wpedantic -Wshadow -Wextra -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=int-conversion -Wvla -g -Og -fsanitize=address -fsanitize=undefined"
 	CORE='src/uxn.c'
 else
-	CFLAGS="${CFLAGS} -DNDEBUG -Os -g0 -s"
 	CORE='src/uxn.c'
 fi
 
-echo "Building.."
-${CC} ${CFLAGS} src/uxnasm.c -o bin/uxnasm
-${CC} ${CFLAGS} ${CORE} src/devices/system.c src/devices/file.c src/devices/datetime.c src/devices/mouse.c src/devices/controller.c src/devices/screen.c src/devices/audio.c src/uxnemu.c ${UXNEMU_LDFLAGS} ${FILE_LDFLAGS} -o bin/uxnemu
-${CC} ${CFLAGS} ${CORE} src/devices/system.c src/devices/file.c src/devices/datetime.c src/uxncli.c ${FILE_LDFLAGS} -o bin/uxncli
+echo "Building"
+${CC} ${CFLAGS} ${LDFLAGS} -std=c89 src/uxnasm.c -o bin/uxnasm
+${CC} ${CFLAGS} ${CORE} ${LDFLAGS} -std=c89 src/devices/system.c src/devices/file.c src/devices/datetime.c src/devices/mouse.c src/devices/controller.c src/devices/screen.c src/devices/audio.c src/uxnemu.c ${UXNEMU_LDFLAGS} ${FILE_LDFLAGS} -o bin/uxnemu
+${CC} ${CFLAGS} ${CORE} ${LDFLAGS} -std=c89 src/devices/system.c src/devices/file.c src/devices/datetime.c src/uxncli.c ${FILE_LDFLAGS} -o bin/uxncli
 
 if [ $install = 1 ]
 then
@@ -108,17 +104,15 @@ then
 	cp bin/uxnemu bin/uxnasm bin/uxncli $HOME/bin/
 fi
 
-echo "Assembling(launcher).."
+echo "Assembling(launcher)"
 ./bin/uxnasm projects/software/launcher.tal bin/launcher.rom
-echo "Assembling(asma).."
-./bin/uxnasm projects/software/asma.tal bin/asma.rom
 
 if [ $norun = 1 ]; then exit; fi
 
-echo "Assembling(piano).."
+echo "Assembling(piano)"
 ./bin/uxnasm projects/software/piano.tal bin/piano.rom
 
-echo "Running.."
+echo "Running"
 ./bin/uxnemu bin/piano.rom
 
-echo "Done."
+echo "Done"
