--- cmake/XRootDOSDefs.cmake.orig	2013-03-09 22:33:45.000000000 +0000
+++ cmake/XRootDOSDefs.cmake	2013-03-09 22:34:17.000000000 +0000
@@ -30,7 +30,7 @@
   add_definitions( -DLT_MODULE_EXT=".dylib" )
   set( CMAKE_INSTALL_LIBDIR "lib" )
   set( CMAKE_INSTALL_BINDIR "bin" )
-  set( CMAKE_INSTALL_MANDIR "man" )
+  set( CMAKE_INSTALL_MANDIR "share/man" )
   set( CMAKE_INSTALL_INCLUDEDIR "include" )
   set( CMAKE_INSTALL_DATADIR "share" )
 endif()
