--- imap/cvt_cyrusdb.c.orig	Fri May 21 23:45:48 2004
+++ imap/cvt_cyrusdb.c	Fri Mar  4 15:13:18 2005
@@ -60,7 +60,6 @@
 #include <syslog.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include "acl.h"
 #include "assert.h"

--- imap/mboxlist.c.orig	Tue Jun 21 00:34:44 2005
+++ imap/mboxlist.c	Tue Jun 21 00:36:53 2005
@@ -60,7 +60,6 @@
 #include <syslog.h>
 
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include "acl.h"
 #include "annotate.h"

--- imtest/imtest.c.orig	Tue Dec  7 15:26:22 2004
+++ imtest/imtest.c	Fri Mar  4 15:22:16 2005
@@ -47,7 +47,6 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 #include <sys/stat.h>
 #include <fcntl.h>

--- installsieve/installscript.c.orig	Thu Feb 13 16:15:37 2003
+++ installsieve/installscript.c	Tue Jun 21 01:30:34 2005
@@ -47,7 +47,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include <stdlib.h>
 #include <stdio.h>
 
--- installsieve/request.c.orig	Tue Jun 21 01:20:13 2005
+++ installsieve/request.c	Tue Jun 21 01:21:04 2005
@@ -47,7 +47,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include <stdlib.h>
 #include <stdio.h>

--- netnews/remotepurge.c.orig	Thu Feb 13 16:15:47 2003
+++ netnews/remotepurge.c	Tue Jun 21 01:35:12 2005
@@ -48,7 +48,6 @@
 
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <ctype.h>

--- perl/sieve/lib/request.c.orig	Thu Feb 13 16:15:53 2003
+++ perl/sieve/lib/request.c	Tue Jun 21 01:38:26 2005
@@ -51,7 +51,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
 
 #include <stdlib.h>
 #include <stdio.h>
