--- configure.orig	2023-12-04 18:50:27
+++ configure	2024-12-02 15:48:58
@@ -26,7 +26,7 @@
   xml2-config --version >/dev/null 2>&1
   if [ $? -eq 0 ]; then
     PKGCONFIG_CFLAGS=`xml2-config --cflags`
-    PKGCONFIG_LIBS=`xml2-config --libs`
+    PKGCONFIG_LIBS=`xml2-config --libs --dynamic`
 
     # MacOS versions El Capitan and later ship a xml2-config which appends `xcrun
     # --show-sdk-path` to the xml2-config. So we remove it if it is present.
