--- setup.py.orig	2018-03-11 16:15:12.000000000 +0300
+++ setup.py	2018-03-11 16:20:05.000000000 +0300
@@ -69,7 +69,8 @@
 
 cpython = platform.python_implementation() == 'CPython'
 
-ext_modules = [Extension('logbook._speedups', sources=['logbook/_speedups.c'])]
+from Cython.Build import cythonize
+ext_modules = cythonize([Extension('logbook._speedups', sources=['logbook/_speedups.pyx'])])
 
 ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError)
 if sys.platform == 'win32':
