--- src/gnutls-callbacks.cc.orig	2020-04-20 14:02:40.000000000 -0500
+++ src/gnutls-callbacks.cc	2020-04-20 15:55:26.000000000 -0500
@@ -32,18 +32,12 @@
 
 #else
 
-#ifdef HAVE_MALLOC_H
-#include <malloc.h>
-#elif HAVE_SYS_MALLOC_H
-#include <sys/malloc.h>
-#else
-#error "No malloc.h present."
-#endif
+#include <malloc/malloc.h>
 
 static inline
 size_t malloc_usable_size_wrapper (void *ptr)
 {
-  return malloc_usable_size (ptr);
+  return malloc_size (ptr);
 }
 
 #endif
