--- ChangeLog.old	2015-01-16 19:15:05.000000000 +0100
+++ ChangeLog	2015-01-16 18:57:08.000000000 +0100
@@ -1,3 +1,9 @@
+2013-10-15 Borut Razem <borutr@users.sourceforge.net>
+	* modules/stimuli.cc, modules/stimuli.h, modules/switch.h,
+	  modules/switch.cc, gui/gui_stopwatch.cc: applied patch
+	  #32 Make gpsim compile with clang,
+	  thanks to Ralf Horstmann
+
 2011-04-27 Roy Rankin <rrankin@ihug.com.au>
 	* regression/p16f690/Makefile : use local lkr file
 	* regression/Makefile.am : tarball fix
--- gui/gui_stopwatch.cc.old	2015-01-16 19:15:05.000000000 +0100
+++ gui/gui_stopwatch.cc	2015-01-16 18:57:08.000000000 +0100
@@ -179,7 +179,7 @@
     }
     if(!sww->IsUpdate() && (text=gtk_entry_get_text (GTK_ENTRY(widget)))!=0)
     {
-        long long v = strtoll(text,0,10);
+        long long v = ::strtoll(text,0,10);
 	if(v!=(sww->cyclecounter-sww->offset)%sww->rollover)
 	{
             v=(v+sww->offset)%sww->rollover;
@@ -201,7 +201,7 @@
     }
     if(!sww->IsUpdate() && (text=gtk_entry_get_text (GTK_ENTRY(widget)))!=0)
     {
-        long long v = strtoll(text,0,10);
+        long long v = ::strtoll(text,0,10);
 
 	if(v!=sww->offset)
 	{
@@ -221,7 +221,7 @@
     }
     if(!sww->IsUpdate() && (text=gtk_entry_get_text (GTK_ENTRY(widget)))!=0)
     {
-        long long v = strtoll(text,0,10);
+        long long v = ::strtoll(text,0,10);
 
 	if(v!=sww->rollover)
 	{
@@ -403,7 +403,7 @@
 
   get_config();
   if(config_get_string(name(),"rollover",&string))
-    rollover = strtoll(string,0,10);
+    rollover = ::strtoll(string,0,10);
   config_get_variable(name(),"count_dir",&count_dir);
     
   if(enabled)
--- modules/stimuli.cc.old	2015-01-16 19:15:05.000000000 +0100
+++ modules/stimuli.cc	2015-01-16 18:57:08.000000000 +0100
@@ -216,7 +216,7 @@
   //----------------------------------------------------------------------
   //----------------------------------------------------------------------
 
-  PulseGen::PulseGen(const char *_name=0)
+  PulseGen::PulseGen(const char *_name)
     : StimulusBase(_name, "\
 Pulse Generator\n\
  Attributes:\n\
--- modules/stimuli.h.old	2015-01-16 19:15:05.000000000 +0100
+++ modules/stimuli.h	2015-01-16 19:14:21.000000000 +0100
@@ -48,10 +48,10 @@
     gint64 time;
     Value  *v;
 
-    inline bool operator < (ValueStimulusData &rValue) {
+    inline bool operator < (const ValueStimulusData &rValue) const {
       return time < rValue.time;
     }
-    inline bool operator == (ValueStimulusData rValue) {
+    inline bool operator == (const ValueStimulusData rValue) const {
       return time == rValue.time;
     }
   };
@@ -72,7 +72,7 @@
   {
   public:
     static Module *construct(const char *new_name);
-    PulseGen(const char *_name);
+    PulseGen(const char *_name=0);
     ~PulseGen();
 
     virtual void callback();
--- modules/switch.cc.old	2015-01-16 19:15:05.000000000 +0100
+++ modules/switch.cc	2015-01-16 18:57:08.000000000 +0100
@@ -582,7 +582,7 @@
 
   //========================================================================
 
-  Switch::Switch(const char *_new_name=0)
+  Switch::Switch(const char *_new_name)
     : SwitchBase(_new_name, "\
 Two port switch\n\
  Attributes:\n\
--- modules/switch.h.old	2015-01-16 19:15:05.000000000 +0100
+++ modules/switch.h	2015-01-16 18:57:08.000000000 +0100
@@ -88,7 +88,7 @@
   class Switch : public SwitchBase
   {
   public:
-    Switch(const char *_name);
+    Switch(const char *_name=0);
     ~Switch();
 
     virtual void setState(bool);
