--- Makefile.PL.orig	2018-06-29 05:32:54.000000000 -0700
+++ Makefile.PL	2018-09-20 14:31:16.000000000 -0700
@@ -131,7 +131,7 @@
 # look for already existing configuration file
 my $config_file_message = -f "$MY::variables{SYSCONFDIR}/agent.cfg" ?
     "previous configuration file found, new one will be installed as agent.cfg.new" :
-    "no previous configuration file found, new one will be installed as agent.cfg";
+    "no previous configuration file found, new one will be installed as agent.cfg.new";
 
 print <<EOF;
 
@@ -240,7 +240,7 @@
 	if [ -f $(DESTDIR)/$(SYSCONFDIR)/agent.cfg ]; then \
 	    install -m 644 etc/agent.cfg $(DESTDIR)$(SYSCONFDIR)/agent.cfg.new; \
 	else \
-	    install -m 644 etc/agent.cfg $(DESTDIR)$(SYSCONFDIR)/agent.cfg; \
+	    install -m 644 etc/agent.cfg $(DESTDIR)$(SYSCONFDIR)/agent.cfg.new; \
 	fi
 	$(ABSPERLRUN) -pi \
 	    -e "s|=> undef, # SYSCONFDIR.*|=> '$(SYSCONFDIR)',|;" \
