--- Makefile.PL.orig	2020-09-28 02:55:10.000000000 -0700
+++ Makefile.PL	2020-11-24 15:21:16.000000000 -0800
@@ -135,7 +135,7 @@
 # look for already existing configuration file
 my $config_file_message = -f "$MY::variables{SYSCONFDIR}/agent.cfg" ?
     "previous configuration file found, new one will be installed as agent.cfg.new" :
-    "no previous configuration file found, new one will be installed as agent.cfg";
+    "no previous configuration file found, new one will be installed as agent.cfg.new";
 
 print <<EOF;
 
@@ -249,8 +249,8 @@
 	    $(CP) etc/$$config $(DESTDIR)$(SYSCONFDIR)/$$config.new; \
 	    $(CHMOD) $(PERM_RW) $(DESTDIR)$(SYSCONFDIR)/$$config.new; \
 	  else \
-	    $(CP) etc/$$config $(DESTDIR)$(SYSCONFDIR)/$$config; \
-	    $(CHMOD) $(PERM_RW) $(DESTDIR)$(SYSCONFDIR)/$$config; \
+	    $(CP) etc/$$config $(DESTDIR)$(SYSCONFDIR)/$$config.new; \
+	    $(CHMOD) $(PERM_RW) $(DESTDIR)$(SYSCONFDIR)/$$config.new; \
 	  fi; \
 	done
 	$(ABSPERLRUN) -pi \
