From 37341e9dbd382d51f00c1a3bdb3cd494cbdc04ce Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 20 Apr 2024 02:15:10 +0800
Subject: [PATCH] Do not use precomp headers target with GCC on Apple

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 236dbd2..a118b10 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -66,7 +66,9 @@ if (ENABLE_QRCODE_LOGIN)
     target_compile_definitions(abaddon PRIVATE WITH_QRLOGIN)
 endif ()
 
-target_precompile_headers(abaddon PRIVATE <gtkmm.h> src/abaddon.hpp src/util.hpp)
+if (NOT (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
+    target_precompile_headers(abaddon PRIVATE <gtkmm.h> src/abaddon.hpp src/util.hpp)
+endif()
 
 if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") OR
 (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
