--- gdal/CMakeLists.txt.orig	2015-01-08 01:48:57.000000000 -0600
+++ gdal/CMakeLists.txt	2015-08-22 21:40:49.000000000 -0500
@@ -45,5 +45,5 @@
 
 ###############################################################################
 # Installation 
-install (TARGETS ${LIBKEA_GDAL_DRIVER} DESTINATION gdalplugins)
-###############################################################################
\ No newline at end of file
+install (TARGETS ${LIBKEA_GDAL_DRIVER} DESTINATION lib/gdalplugins)
+###############################################################################
