--- krusader/CMakeLists.txt.orig	2012-09-04 11:18:47.000000000 +0200
+++ krusader/CMakeLists.txt	2012-09-04 11:19:51.000000000 +0200
@@ -44,7 +44,7 @@
 
 kde4_add_executable(krusader ${krusader_SRCS})
 
-target_link_libraries(krusader BookMan Dialogs DiskUsage GUI Konfigurator KViewer MountMan Panel Queue VFS Search Splitter Synchronizer UserMenu Locate UserAction ActionMan KViewer Filter Dialogs GUI ${KDE4_KPARTS_LIBS} ${KDE4_KHTML_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS}  ${QT_QT3SUPPORT_LIBRARY} ${KDE4_KDE3SUPPORT_LIBS} ${KDE4_KDECORE_LIBS} )
+target_link_libraries(krusader BookMan Dialogs DiskUsage GUI Konfigurator KViewer MountMan Panel Queue VFS Search Splitter Synchronizer UserMenu Locate UserAction ActionMan KViewer Filter Dialogs GUI ${KDE4_KPARTS_LIBS} ${KDE4_KHTML_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_KFILE_LIBS}  ${QT_QT3SUPPORT_LIBRARY} ${KDE4_KDE3SUPPORT_LIBS} ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} )
 
 install(TARGETS krusader ${INSTALL_TARGETS_DEFAULT_ARGS})
 
