Index: aclocal.m4
diff --git aclocal.m4 aclocal.m4.orig
index 1228f6ca29c6208a89c19bdf8aee3f651a9c3dcb..7295aa9cf6126b121cac9b57db3c74ca76989f95 100644
--- aclocal.m4
+++ aclocal.m4.orig
@@ -1121,17 +1121,37 @@ AC_DEFUN(STEPMAKE_PYTHON_DEVEL, [
     AC_CHECK_PROGS(PYTHON_CONFIG, `basename $PYTHON`-config, no)
 
     if test -z "$PYTHON_CFLAGS" -a "$PYTHON_CONFIG" != "no"; then
-        # Clean out junk: http://bugs.python.org/issue3290
-	# Python headers may need some -f* flags, leave them in.
-	# We want the sed commands to look like 's/-[WDOm][[:alnum:][:punct:]][[:alnum:][:punct:]]*//g' and 's/-arch [^[:space:]]*//g', but automake eats brackets.
-	 #PYTHON_CFLAGS=`$PYTHON_CONFIG --cflags | sed -e 's/-[[WDOm]][[[:alnum:][:punct:]]][[[:alnum:][:punct:]]]*//g' | sed -e 's/-arch @<:@^@<:@:space:@:>@@:>@*//g'`
-	 # The above sed BRE matches parts of legal options, stipping down part of that option, resulting in invalid gcc arguments. Gentoo Bug #415793
-	 # For instance, '-floop-stip-mime' becomes '-floop-strip', and '-fvect-cost-model' becomes '-fvect-cost'.
-	 # Tentative fix to require a non alphanumeric character before the initial hyphen of the BRE or the hyphen being the first character in the string.
-	 PYTHON_CFLAGS=`$PYTHON_CONFIG --cflags | sed -e 's/\(^\|[[^[:alnum:]]]\)-[[WDOm]][[[:alnum:][:punct:]]][[[:alnum:][:punct:]]]*//g' | sed -e 's/-arch @<:@^@<:@:space:@:>@@:>@*//g'`
+	# We massage $PYTHON_CFLAGS:
+	#
+	# . Clean out junk (http://bugs.python.org/issue3290).
+	# . Python headers may need some '-f*' flags, leave them in.
+	# . Since we are later on adding $PYTHON_CFLAGS to $CPPFLAGS, remove
+	#   the '-Wstrict-prototypes' warning, which is not valid for C++,
+	#   and which some ports of Python add (for example macports).
+	#
+	# This leads to the following BREs for sed:
+	#
+	#   s/-[WDOm][[:alnum:][:punct:]][[:alnum:][:punct:]]*//g
+	#   s/-arch [^[:space:]]*//g
+	#   s/-Wstrict-prototypes//g
+	#
+	# However, the first BRE strips off parts of valid options, thus
+	# resulting in invalid gcc arguments (Gentoo bug #415793).  For
+	# instance, '-floop-strip-mime' becomes '-floop-strip', and
+	# '-fvect-cost-model' becomes '-fvect-cost'.
+	#
+	# Our tentative fix is to require a non-alphanumeric character
+	# before the initial hyphen of the BRE or the hyphen being the first
+	# character in the string.
+dnl To prevent m4 eating brackets we add one more level of brackets
+dnl for proper quoting.
+	[PYTHON_CFLAGS=`$PYTHON_CONFIG --cflags \
+	  | sed -e 's/\(^\|[^[:alnum:]]\)-[WDOm][[:alnum:][:punct:]][[:alnum:][:punct:]]*//g' \
+		-e 's/-arch [^[:space:]]*//g' \
+		-e 's/-Wstrict-prototypes//g'`]
 	PYTHON_LDFLAGS=`$PYTHON_CONFIG --ldflags`
     fi
-    
+
     if test -z "$PYTHON_CFLAGS" -a "$cross_compiling" = "no"; then
 	changequote(<<, >>)#dnl
 	# alternatively, for python >= 2.0
