--- scripts/build/mf2pt1.pl.orig	2012-03-27 19:42:13.000000000 -0400
+++ scripts/build/mf2pt1.pl	2012-03-30 15:16:48.000000000 -0400
@@ -426,7 +426,7 @@
 # parameters, too.
 sub get_bboxes ($)
 {
-    execute_command 1, ("mpost", "-mem=mf2pt1", "-progname=mpost",
+    execute_command 1, ("mpost", "-mem=../../mf2pt1", "-progname=mpost",
                         "\\mode:=localfont; mag:=$mag; bpppix $bpppix; nonstopmode; input $mffile");
     opendir (CURDIR, ".") || die "${progname}: $! ($filedir)\n";
     @charfiles = sort
