diff --git a/src/MemoryX.h b/src/MemoryX.h
index 56e99171a9eb87fe574291d0ec9d086efb0c1fbb..621afa3001ba354f36e80edb86c72a45dca426cb 100644
--- a/src/MemoryX.h
+++ b/src/MemoryX.h
@@ -1177,7 +1179,7 @@ namespace std
    namespace tr1
    {
 #endif
-      template<typename T> struct hash;
+
       template<> struct hash< wxString > {
          size_t operator () (const wxString &str) const // noexcept
          {
diff --git a/src/commands/CommandManager.h b/src/commands/CommandManager.h
index 7668506f2698240a1efbace1521401040a7396e4..7fae64ec06a48ce4ae7b77c2489a17ed0e2d89da 100644
--- a/src/commands/CommandManager.h
+++ b/src/commands/CommandManager.h
@@ -104,7 +104,7 @@ namespace std
    namespace tr1
    {
 #endif
-      template<typename T> struct hash;
+
       template<> struct hash< NormalizedKeyString > {
          size_t operator () (const NormalizedKeyString &str) const // noexcept
          {
