diff -NaurB ./cmake/OSDep.cmake ./cmake/OSDep.cmake
--- ./cmake/OSDep.cmake	2024-05-13 18:12:22
+++ ./cmake/OSDep.cmake	2024-05-13 18:12:08
@@ -23,7 +23,9 @@
     ADD_COMPILE_OPTIONS(-D_BSD_SOURCE -DDARWIN)
     SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS} -undefined dynamic_lookup")
     IF(ENABLE_LUAJIT MATCHES "ON")
-        SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pagezero_size 10000 -image_base 100000000")
+        IF ("${ARCH}" STREQUAL "x86_64")
+            SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pagezero_size 10000 -image_base 100000000")
+        ENDIF()
     ENDIF(ENABLE_LUAJIT MATCHES "ON")
     MESSAGE(STATUS "Configuring for Darwin")
     SET(TAR "gnutar")
