--- a/hw/xquartz/X11Application.m	2020-12-09 09:26:40.000000000 +0000
+++ b/hw/xquartz/X11Application.m	2020-12-09 09:27:38.000000000 +0000
@@ -426,11 +426,6 @@
                 BOOL order_all_windows = YES, workspaces, ok;
                 for_appkit = NO;
 
-                /* FIXME: This is a hack to avoid passing the event to AppKit which
-                 *        would result in it raising one of its windows.
-                 */
-                _appFlags._active = YES;
-
                 [self set_front_process:nil];
 
                 /* Get the Spaces preference for SwitchOnActivate */
