--- a2x.orig	2009-02-26 15:27:38.000000000 +0100
+++ a2x	2009-02-26 15:53:54.000000000 +0100
@@ -8,12 +8,12 @@
 #
 
 VERSION=1.0.0
-BASENAME=$(basename "$0")
+BASENAME=$(gbasename "$0")
 REALNAME="$0"
 if [ ! -e "$REALNAME" ]; then
     REALNAME=$(which "$REALNAME")
 fi
-REALNAME="$(readlink -f "$REALNAME")"
+REALNAME="$(greadlink -f "$REALNAME")"
 CONF_DIR=/etc/asciidoc
 FOP_COMMAND="fop.sh"    # FOP executable.
 
@@ -161,7 +161,7 @@
 # Echo the total size in bytes of file name arguments.
 function file_size()
 {
-    echo $(du -cb "$@" | tail -1 | awk '{print $1}')
+    echo $(gdu -cb "$@" | gtail -1 | awk '{print $1}')
 }
 
 #--------------------------------------------------------------------
@@ -223,7 +223,7 @@
 {
     local result dir
     # First look in same directory as asciidoc executable.
-    dir="$(dirname "$REALNAME")"
+    dir="$(gdirname "$REALNAME")"
     if [ ! -f "$dir/$1" -a -d $CONF_DIR ]; then
         dir=$CONF_DIR
     fi
@@ -334,7 +334,7 @@
         quit "source file not found: $1"
     fi
     SRC_FILE=$1
-    SRC_DIR=$(dirname "$1")
+    SRC_DIR=$(gdirname "$1")
     SRC_NAME=$1
     SRC_NAME=${SRC_NAME##*/}    # Strip path.
     SRC_NAME=${SRC_NAME%.*}     # Strip extension.
@@ -415,7 +415,7 @@
     if [[ "$src" -ef "$dst" ]]; then
         return
     fi
-    execute_command_2 "cp -u \"$src\" \"$dst\""
+    execute_command_2 "gcp -u \"$src\" \"$dst\""
 }
 
 # Copy distribution admonition and navigation icons to destination directory
@@ -429,7 +429,7 @@
     if [ ! -r "$src" ]; then
         quit "file not found: $src"
     fi
-    src=$(dirname "$src")
+    src=$(gdirname "$src")
     # Check we're not trying to copy the file onto itself.
     if [[ "$src" -ef "$dst" ]]; then
         return
@@ -439,9 +439,9 @@
             quit "icon destination must be a directory: $dst"
         fi
     else
-        execute_command_2 "mkdir -p \"$dst\""
+        execute_command_2 "gmkdir -p \"$dst\""
     fi
-    execute_command_2 "cp -rfu  \"$src/\"* \"$dst\""
+    execute_command_2 "gcp -rfu  \"$src/\"* \"$dst\""
 }
 
 # Run xmllint(1) checking of $1 file if xmllint checking is enabled.
@@ -486,7 +486,7 @@
         quit "file not found: $xsl"
     fi
     to_docbook
-    xml=$(readlink -f "$SRC_DIR/$SRC_NAME.xml")
+    xml=$(greadlink -f "$SRC_DIR/$SRC_NAME.xml")
     html="$SRC_NAME.html"
     copy_stylesheet_and_icons "$DESTINATION_DIR"
     execute_command_2 "cd \"$DESTINATION_DIR\""
@@ -519,7 +519,7 @@
         quit "file not found: $xsl"
     fi
     to_docbook
-    xml=$(readlink -f "$SRC_DIR/$SRC_NAME.xml")
+    xml=$(greadlink -f "$SRC_DIR/$SRC_NAME.xml")
     if [ ! -d "$chunkdir" ]; then
         execute_command_2 "mkdir \"$chunkdir\""
     fi
@@ -541,7 +541,7 @@
         quit "file not found: $xsl"
     fi
     to_docbook "-d manpage"
-    xml=$(readlink -f "$SRC_DIR/$SRC_NAME.xml")
+    xml=$(greadlink -f "$SRC_DIR/$SRC_NAME.xml")
     execute_command_2 "cd \"$DESTINATION_DIR\""
     execute_command_2 "xsltproc $XSLTPROC_OPTS --nonet \
         \"$xsl\" \"$xml\""
