--- dlls/gdi32/freetype.c.orig	2012-02-12 23:04:21.000000000 -0800
+++ dlls/gdi32/freetype.c	2012-02-12 23:04:58.000000000 -0800
@@ -40,6 +40,7 @@
 #include <assert.h>
 
 #ifdef HAVE_CARBON_CARBON_H
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -57,6 +58,7 @@
 #define ResizePalette __carbon_ResizePalette
 #define SetRectRgn __carbon_SetRectRgn
 #include <Carbon/Carbon.h>
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/dinput/joystick_osx.c.orig	2012-02-13 11:57:04.000000000 -0800
+++ dlls/dinput/joystick_osx.c	2012-02-13 11:57:27.000000000 -0800
@@ -24,6 +24,7 @@
 #include "wine/port.h"
 
 #if defined(HAVE_CARBON_CARBON_H) && defined(HAVE_IOKIT_HID_IOHIDLIB_H)
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -62,6 +63,7 @@
 #define STDMETHODCALLTYPE __carbon_STDMETHODCALLTYPE
 #include <Carbon/Carbon.h>
 #include <IOKit/hid/IOHIDLib.h>
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/windowscodecs/icnsformat.c.orig	2012-02-13 12:58:07.000000000 -0800
+++ dlls/windowscodecs/icnsformat.c	2012-02-13 12:58:31.000000000 -0800
@@ -22,6 +22,7 @@
 #include <stdarg.h>
 
 #ifdef HAVE_APPLICATIONSERVICES_APPLICATIONSERVICES_H
+#define BOOL BOOL_Mac
 #define GetCurrentProcess GetCurrentProcess_Mac
 #define GetCurrentThread GetCurrentThread_Mac
 #define LoadResource LoadResource_Mac
@@ -37,6 +38,7 @@
 #define ShowCursor ShowCursor_Mac
 #define UnionRect UnionRect_Mac
 #include <ApplicationServices/ApplicationServices.h>
+#undef BOOL
 #undef GetCurrentProcess
 #undef GetCurrentThread
 #undef LoadResource
--- dlls/wineosxime.drv/mactext.c.orig	2012-02-13 14:31:06.000000000 -0800
+++ dlls/wineosxime.drv/mactext.c	2012-02-13 14:31:58.000000000 -0800
@@ -50,7 +50,9 @@
 #ifdef __APPLE__
 
 #include "wine/debug.h"
+#define BOOL __carbon_BOOL
 #include <Carbon/Carbon.h>
+#undef BOOL
 #include <dlfcn.h>
 #include "IMKClient.h"
 
--- dlls/wineqtdecoder/qtvdecoder.c.orig	2012-02-13 14:36:12.000000000 -0800
+++ dlls/wineqtdecoder/qtvdecoder.c	2012-02-13 14:36:46.000000000 -0800
@@ -22,6 +22,7 @@
 
 #define ULONG CoreFoundation_ULONG
 #define HRESULT CoreFoundation_HRESULT
+#define BOOL CoreFoundation_BOOL
 
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
@@ -103,6 +104,8 @@
 
 #undef ULONG
 #undef HRESULT
+#undef BOOL
+
 #undef DPRINTF
 #undef STDMETHODCALLTYPE
 
