--- dlls/gdi32/freetype.c.orig	2012-02-12 23:04:21.000000000 -0800
+++ dlls/gdi32/freetype.c	2012-02-12 23:04:58.000000000 -0800
@@ -40,6 +40,7 @@
 #include <assert.h>
 
 #ifdef HAVE_CARBON_CARBON_H
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -57,6 +58,7 @@
 #define ResizePalette __carbon_ResizePalette
 #define SetRectRgn __carbon_SetRectRgn
 #include <Carbon/Carbon.h>
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/windowscodecs/icnsformat.c.orig	2012-02-12 23:54:58.000000000 -0800
+++ dlls/windowscodecs/icnsformat.c	2012-02-12 23:55:20.000000000 -0800
@@ -22,6 +22,7 @@
 #include <stdarg.h>
 
 #ifdef HAVE_APPLICATIONSERVICES_APPLICATIONSERVICES_H
+#define BOOL BOOL_Mac
 #define GetCurrentProcess GetCurrentProcess_Mac
 #define GetCurrentThread GetCurrentThread_Mac
 #define LoadResource LoadResource_Mac
@@ -49,6 +50,7 @@
 #define ShowCursor ShowCursor_Mac
 #define UnionRect UnionRect_Mac
 #include <ApplicationServices/ApplicationServices.h>
+#undef BOOL
 #undef GetCurrentProcess
 #undef GetCurrentThread
 #undef LoadResource
--- dlls/wineqtdecoder/qtdatahandler.c.orig	2012-02-13 00:04:58.000000000 -0800
+++ dlls/wineqtdecoder/qtdatahandler.c	2012-02-13 00:05:24.000000000 -0800
@@ -23,6 +23,7 @@
 #define ULONG CoreFoundation_ULONG
 #define HRESULT CoreFoundation_HRESULT
 
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -62,6 +63,7 @@
 
 #include <QuickTime/QuickTimeComponents.h>
 
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/wineqtdecoder/qtsplitter.c.orig	2012-02-13 00:19:05.000000000 -0800
+++ dlls/wineqtdecoder/qtsplitter.c	2012-02-13 00:19:26.000000000 -0800
@@ -23,6 +23,7 @@
 #define ULONG CoreFoundation_ULONG
 #define HRESULT CoreFoundation_HRESULT
 
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -63,6 +64,7 @@
 #include <QuickTime/Movies.h>
 #include <QuickTime/QuickTimeComponents.h>
 
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/wineqtdecoder/qtutils.c.orig	2012-02-13 00:31:37.000000000 -0800
+++ dlls/wineqtdecoder/qtutils.c	2012-02-13 00:32:14.000000000 -0800
@@ -23,6 +23,7 @@
 #define ULONG CoreFoundation_ULONG
 #define HRESULT CoreFoundation_HRESULT
 
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -62,6 +63,7 @@
 
 #include <CoreVideo/CVPixelBuffer.h>
 
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/wineqtdecoder/qtvdecoder.c.orig	2012-02-13 00:31:42.000000000 -0800
+++ dlls/wineqtdecoder/qtvdecoder.c	2012-02-13 00:32:30.000000000 -0800
@@ -23,6 +23,7 @@
 #define ULONG CoreFoundation_ULONG
 #define HRESULT CoreFoundation_HRESULT
 
+#define BOOL __carbon_BOOL
 #define LoadResource __carbon_LoadResource
 #define CompareString __carbon_CompareString
 #define GetCurrentThread __carbon_GetCurrentThread
@@ -63,6 +64,7 @@
 #include <QuickTime/ImageCompression.h>
 #include <CoreVideo/CVPixelBuffer.h>
 
+#undef BOOL
 #undef LoadResource
 #undef CompareString
 #undef GetCurrentThread
--- dlls/dinput/joystick_osx.c.orig	2013-05-25 11:13:00.000000000 +0300
+++ dlls/dinput/joystick_osx.c	2013-05-25 11:13:00.000000000 +0300
@@ -29,6 +29,7 @@
 #define LONG SInt32
 #define LPLONG SInt32*
 #define E_PENDING __carbon_E_PENDING
+#define BOOL __carbon_BOOL
 #define ULONG __carbon_ULONG
 #define E_INVALIDARG __carbon_E_INVALIDARG
 #define E_OUTOFMEMORY __carbon_E_OUTOFMEMORY
@@ -52,6 +53,7 @@
 #include <IOKit/IOKitLib.h>
 #include <IOKit/hid/IOHIDLib.h>
 #include <ForceFeedback/ForceFeedback.h>
+#undef BOOL
 #undef ULONG
 #undef E_INVALIDARG
 #undef E_OUTOFMEMORY
