Patch provided by Josh Dubois, CrossOver product manager, CodeWeavers.
--- dlls/winemac.drv/winemac.rc.orig	2013-11-07 15:40:10.000000000 -0600
+++ dlls/winemac.drv/winemac.rc	2013-12-07 09:42:14.000000000 -0600
@@ -19,7 +19,6 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
  */
 
-FALLBACK_ICON RCDATA ../../../macosx/CrossOver/images/exeIcon.icns
 
 #define WINE_FILEDESCRIPTION_STR "Wine Mac driver"
 #define WINE_FILENAME_STR "winemac.drv"
