--- configure.ac.orig	2011-04-28 14:52:28.000000000 -0500
+++ configure.ac	2011-10-09 08:25:51.000000000 -0500
@@ -713,6 +713,10 @@
     LDRPATH_LOCAL="&& install_name_tool -change @executable_path/\`\$(RELPATH) \$(bindir) \$(libdir)\`/libwine.1.dylib @executable_path/\$(top_builddir)/libs/wine/libwine.1.dylib \$@ || \$(RM) \$@"
     dnl declare needed frameworks
     AC_SUBST(SECURITYLIB,"-framework Security -framework CoreFoundation")
+    ac_save_LIBS="$LIBS"
+    LIBS="$LIBS $SECURITYLIB"
+    AC_CHECK_FUNCS(SSLCopyPeerCertificates)
+    LIBS="$ac_save_LIBS"
     AC_SUBST(COREFOUNDATIONLIB,"-framework CoreFoundation")
     AC_SUBST(IOKITLIB,"-framework IOKit -framework CoreFoundation")
     AC_SUBST(APPKITLIB,"-framework AppKit")
--- dlls/secur32/schannel.c.orig	2011-04-28 14:52:45.000000000 -0500
+++ dlls/secur32/schannel.c	2011-10-09 08:29:57.000000000 -0500
@@ -25,7 +25,7 @@
 #include <stdarg.h>
 #include <errno.h>
 #include <limits.h>
-#ifdef HAVE_SECURITY_SECURITY_H
+#if defined(HAVE_SECURITY_SECURITY_H) && defined(HAVE_SSLCOPYPEERCERTIFICATES)
 #include <Security/Security.h>
 #define GetCurrentThread GetCurrentThread_Mac
 #define LoadResource LoadResource_Mac
@@ -48,7 +48,7 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(secur32);
 
-#ifdef HAVE_SECURITY_SECURITY_H
+#if defined(HAVE_SECURITY_SECURITY_H) && defined(HAVE_SSLCOPYPEERCERTIFICATES)
 
 typedef void *schan_imp_transport;
 static int schan_pull(schan_imp_transport transport, void *buff, size_t *buff_len);
@@ -3767,7 +3767,7 @@
     wine_dlclose(libgnutls_handle, NULL, 0);
 }
 
-#else /* HAVE_SECURITY_SECURITY_H || SONAME_LIBGNUTLS */
+#else /* (HAVE_SECURITY_SECURITY_H && HAVE_SSLCOPYPEERCERTIFICATES) || SONAME_LIBGNUTLS */
 
 void SECUR32_initSchannelSP(void)
 {
@@ -3776,4 +3776,4 @@
 
 void SECUR32_deinitSchannelSP(void) {}
 
-#endif /* HAVE_SECURITY_SECURITY_H || SONAME_LIBGNUTLS */
+#endif /* (HAVE_SECURITY_SECURITY_H && HAVE_SSLCOPYPEERCERTIFICATES) || SONAME_LIBGNUTLS */
