diff -ur ../OpenBLAS-0.2.14-orig/cpuid_x86.c ./cpuid_x86.c
--- ../OpenBLAS-0.2.14-orig/cpuid_x86.c	2015-03-25 05:07:07.000000000 +0900
+++ ./cpuid_x86.c	2015-08-08 22:26:31.000000000 +0900
@@ -1101,6 +1101,16 @@
 #endif
           else
 	    return CPUTYPE_NEHALEM;
+	case 13:
+	  //Broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CPUTYPE_HASWELL;
+#else
+	    return CPUTYPE_SANDYBRIDGE;
+#endif
+          else
+	    return CPUTYPE_NEHALEM;
 	}
 	break;
       case 4:
@@ -1115,8 +1125,33 @@
 #endif
           else
 	    return CPUTYPE_NEHALEM;
+	case 7:
+	case 15:
+	  //Broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CPUTYPE_HASWELL;
+#else
+	    return CPUTYPE_SANDYBRIDGE;
+#endif
+          else
+	    return CPUTYPE_NEHALEM;
         }
         break;
+      case 5:
+        switch (model) {
+	case 6:
+	  //Broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CPUTYPE_HASWELL;
+#else
+	    return CPUTYPE_SANDYBRIDGE;
+#endif
+          else
+	    return CPUTYPE_NEHALEM;
+	}
+	break;
       }
       break;
     case 0x7:
@@ -1528,6 +1563,16 @@
 #endif
           else
 	    return CORE_NEHALEM;
+	case 13:
+	  //broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CORE_HASWELL;
+#else
+	    return CORE_SANDYBRIDGE;
+#endif
+          else
+	    return CORE_NEHALEM;
 	}
 	break;
       case 4:
@@ -1542,8 +1587,33 @@
 #endif
           else
 	    return CORE_NEHALEM;
+	case 7:
+	case 15:
+	  //broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CORE_HASWELL;
+#else
+	    return CORE_SANDYBRIDGE;
+#endif
+          else
+	    return CORE_NEHALEM;
         }
         break;
+      case 5:
+        switch (model) {
+	case 6:
+	  //broadwell
+          if(support_avx())
+#ifndef NO_AVX2
+            return CORE_HASWELL;
+#else
+	    return CORE_SANDYBRIDGE;
+#endif
+          else
+	    return CORE_NEHALEM;
+	}
+	break;
       }
       break;
 
diff -ur ../OpenBLAS-0.2.14-orig/driver/others/dynamic.c ./driver/others/dynamic.c
--- ../OpenBLAS-0.2.14-orig/driver/others/dynamic.c	2015-03-25 05:07:07.000000000 +0900
+++ ./driver/others/dynamic.c	2015-08-08 22:26:45.000000000 +0900
@@ -221,6 +221,15 @@
 	    return &gotoblas_NEHALEM; //OS doesn't support AVX. Use old kernels.
 	  }
 	}
+	//Intel Broadwell
+	if (model == 13) {
+	  if(support_avx())
+	    return &gotoblas_HASWELL;
+	  else{
+	    openblas_warning(FALLBACK_VERBOSE, NEHALEM_FALLBACK);
+	    return &gotoblas_NEHALEM; //OS doesn't support AVX. Use old kernels.
+	  }
+	}
 	return NULL;
       case 4:
 		//Intel Haswell
@@ -232,6 +241,26 @@
 	    return &gotoblas_NEHALEM; //OS doesn't support AVX. Use old kernels.
 	  }
 	}
+	//Intel Broadwell
+	if (model == 7 || model == 15) {
+	  if(support_avx())
+	    return &gotoblas_HASWELL;
+	  else{
+	    openblas_warning(FALLBACK_VERBOSE, NEHALEM_FALLBACK);
+	    return &gotoblas_NEHALEM; //OS doesn't support AVX. Use old kernels.
+	  }
+	}
+	return NULL;
+      case 5:
+	//Intel Broadwell
+	if (model == 6) {
+	  if(support_avx())
+	    return &gotoblas_HASWELL;
+	  else{
+	    openblas_warning(FALLBACK_VERBOSE, NEHALEM_FALLBACK);
+	    return &gotoblas_NEHALEM; //OS doesn't support AVX. Use old kernels.
+	  }
+	}
 	return NULL;
       }
       case 0xf:
Only in .: patch-broadwell.diff
