--- src/dxfacad.h.orig	2019-01-06 12:59:59.000000000 +0100
+++ src/dxfacad.h	2019-07-21 00:50:07.000000000 +0200
@@ -1,3 +1,4 @@
+#include <cmath>
 static const char  dxf14acadheader_prelayer1 [] =
 "  0\n"
 "SECTION\n"
@@ -245,7 +246,7 @@
 	friend ostream & operator << (ostream & out, const DXF_LineType & lt) {
 
 		double length = 0.0;
-		for (auto l : lt.m_pattern) length += abs(l);
+		for (auto l : lt.m_pattern) length += std::abs(l);
 		
 		out <<
 			"  0\n"
