--- gimp-gap-2.6.0/configure.in.orig	2009-06-04 13:40:33.000000000 -0700
+++ gimp-gap-2.6.0/configure.in	2015-04-28 21:15:16.000000000 -0700
@@ -447,7 +447,7 @@
       fi
 
       if test "x$CC" != "x"; then
-      	 ff_pass_through_flags=" $ff_pass_through_flags --cc=$CC "
+      	 ff_pass_through_flags=" $ff_pass_through_flags --cc=\"$CC\" "
       fi
       
       FFMPEG_CONFIGURE_OPTIONS=" $ff_pass_through_flags $FFMPEG_CONFIGURE_OPTIONS"
