--- gimp-gap-2.6.0/extern_libs/Makefile.in.orig	2013-12-25 01:08:06.000000000 -0800
+++ gimp-gap-2.6.0/extern_libs/Makefile.in	2013-12-25 01:08:48.000000000 -0800
@@ -458,9 +458,9 @@
 @BUILD_LIBMPEG3_LIB_TRUE@	$(EXGMAKE)
 
 install-data-hook:
-	rm -f $(GAPLIBDIR)/libavformat.a \
-	rm -f $(GAPLIBDIR)/libavcodec.a \
-	rm -f $(GAPLIBDIR)/libmpeg3.a
+	rm -f ${DESTDIR}/$(GAPLIBDIR)/libavformat.a \
+	rm -f ${DESTDIR}/$(GAPLIBDIR)/libavcodec.a \
+	rm -f ${DESTDIR}/$(GAPLIBDIR)/libmpeg3.a
 
 # eof (without this line Makefile generation produces syntax error)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
