--- gap/gap_dbbrowser_utils.c.orig	2008-10-17 12:25:36.000000000 -0700
+++ gap/gap_dbbrowser_utils.c	2008-10-17 12:26:26.000000000 -0700
@@ -167,8 +167,8 @@
                     GimpPDBProcType  type,
                     gint             n_params,
                     gint             n_return_vals,
-                    GimpParamDef    *params,
-                    GimpParamDef    *return_vals)
+                    const GimpParamDef    *params,
+                    const GimpParamDef    *return_vals)
 {
   GtkWidget    *main_vbox;
   GtkWidget    *frame;
