From 3be8d6e6de76ca3f83368df326497661b427b41c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 2 Nov 2024 10:33:19 +0800
Subject: [PATCH] timg.cc: define __STDC_FORMAT_MACROS if undefined

---
 src/timg.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/timg.cc src/timg.cc
index 77c08e7..5afef64 100644
--- src/timg.cc
+++ src/timg.cc
@@ -61,10 +61,14 @@
 #    include <poppler.h>
 #endif
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
 #include <errno.h>
 #include <fcntl.h>
 #include <getopt.h>
-#include <inttypes.h>
+#include <cinttypes>
 #include <libswscale/swscale.h>  // Only needed for version.
 #include <math.h>
 #include <signal.h>
