--- lib/Json.hpp.orig	2019-10-10 13:05:56.000000000 -0400
+++ lib/Json.hpp	2019-10-10 13:05:51.000000000 -0400
@@ -3723,7 +3723,7 @@
 
 bool Json::iterator::is_original_(const Jnode & jn, const WalkStep &ws) {
  // match first-seen jsons: keep track of matched jsons in the namespaces
- static const Jnode not_original;
+ static const Jnode not_original{};
  std::string prefix = ws.stripped.front() + "\n";               // prevent clashing with user's NS
 
  if(json().ojn_ptr_ == nullptr) {                               // first time, prepare
@@ -3748,7 +3748,7 @@
 
 bool Json::iterator::is_duplicate_(const Jnode & jn, const WalkStep &ws) {
  // match duplicate jsons: keep track of matched jsons in the namespaces
- static const Jnode original;
+ static const Jnode original{};
  std::string prefix = ws.stripped.front() + "\n";               // prevent clashing with user's NS
 
  if(json().djn_ptr_ == nullptr) {                               // first time, prepare
