--- jtc.cpp.orig	2019-10-09 15:05:46.000000000 -0400
+++ jtc.cpp	2019-10-10 09:22:38.000000000 -0400
@@ -239,7 +239,7 @@
                                             opt_[CHR(OPT_TMP)].hits() == opt_[CHR(OPT_WLK)].hits();
                          // ready jinp_
                          jinp_.tab(opt_[CHR(OPT_IND)].hits() > 0 or not opt_[CHR(OPT_RAW)]?
-                                    abs(opt_[CHR(OPT_IND)]): 1)
+                                    abs(static_cast<int>(opt_[CHR(OPT_IND)].hits())): 1)
                               .raw(opt_[CHR(OPT_RAW)])
                               .quote_solidus(opt_[CHR(OPT_QUT)].hits() % 2 == 1);
 
