--- ./lib/Streamstr.hpp.orig	2019-10-09 15:09:01.000000000 -0400
+++ ./lib/Streamstr.hpp	2019-10-09 15:08:53.000000000 -0400
@@ -42,7 +42,7 @@
                              size_t offset = len <= cur_? hb_.size(): cur_ + hb_.size() - len;
                              std::string s(&hb_[offset], hb_.size() - offset);
                              offset = len <= cur_? cur_ - len: 0;
-                             s += {&hb_[offset], cur_ - offset};
+                             s += std::string{&hb_[offset], cur_ - offset};
                              return s;
                             }
         const char &        chr(size_t offset = 0) const {
