--- CMakeLists.txt.orig	2019-07-02 08:13:37.000000000 -0700
+++ CMakeLists.txt	2019-07-02 08:15:42.000000000 -0700
@@ -58,7 +58,7 @@
 find_package(TIFF)
 if (TIFF_FOUND)
   add_definitions(-DXDMF_BUILD_TIFF)
-  include_directories(${TIFF_INCLUDE_DIR})
+  include_directories(SYSTEM ${TIFF_INCLUDE_DIR})
 endif (TIFF_FOUND)
 
 # Test for DSM
@@ -67,7 +67,7 @@
 if(XDMF_BUILD_DSM)
   find_package(MPI REQUIRED)
   if(MPI_FOUND)
-    include_directories(${MPI_INCLUDE_PATH})
+    include_directories(SYSTEM ${MPI_INCLUDE_PATH})
     # The Intel compiler requires an extra tag for CXX_FLAGS in order to properly build.
     if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
       add_definitions(-DMPICH_IGNORE_CXX_SEEK)
@@ -99,9 +99,9 @@
         set(PYTHON_INCLUDE_MPI4PY_DIR ${MPI4PY_INCLUDE_DIR})
       endif ()
     endif ()
-    include_directories(${PYTHON_INCLUDE_DIRS} ${MPI_CXX_INCLUDE_PATH} ${PYTHON_INCLUDE_MPI4PY_DIR})
+    include_directories(SYSTEM ${PYTHON_INCLUDE_DIRS} ${MPI_CXX_INCLUDE_PATH} ${PYTHON_INCLUDE_MPI4PY_DIR})
   else ()
-    include_directories(${PYTHON_INCLUDE_DIRS})
+    include_directories(SYSTEM ${PYTHON_INCLUDE_DIRS})
   endif ()
 
   # A macro to swig and create the python files
@@ -181,7 +181,7 @@
 if(XDMF_WRAP_JAVA)
   find_package(Java REQUIRED)
   find_package(JNI REQUIRED)
-  include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+  include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}
     ${JAVA_INCLUDE_PATH}
     ${JAVA_INCLUDE_PATH2})
 
