--- tests/C/CMakeLists.txt.orig	2023-05-23 09:37:42
+++ tests/C/CMakeLists.txt	2023-05-23 09:48:28
@@ -23,6 +23,7 @@
 ADD_TEST_C(CTestXdmfDomain)
 if (TIFF_FOUND)
   ADD_TEST_C(CTestXdmfTIFFReadWriteCompressed)
+  target_link_libraries(CTestXdmfTIFFReadWriteCompressed PRIVATE ${TIFF_LIBRARIES})
 endif ()
 ADD_TEST_C(CTestXdmfTopology)
 ADD_TEST_C(CTestXdmfGeometry)
--- tests/Cxx/CMakeLists.txt.orig	2023-05-23 09:50:13
+++ tests/Cxx/CMakeLists.txt	2023-05-23 09:51:44
@@ -40,6 +40,7 @@
 ADD_TEST_CXX(TestXdmfTime)
 if (TIFF_FOUND)
   ADD_TEST_CXX(TestXdmfTIFFReadWriteCompressed)
+  target_link_libraries(TestXdmfTIFFReadWriteCompressed PRIVATE ${TIFF_LIBRARIES})
 endif (TIFF_FOUND)
 ADD_TEST_CXX(TestXdmfTopology)
 ADD_TEST_CXX(TestXdmfTopologyMixed)
--- core/tests/C/CMakeLists.txt.orig	2023-05-26 06:45:54
+++ core/tests/C/CMakeLists.txt	2023-05-26 06:46:28
@@ -6,10 +6,10 @@
 ADD_TEST_C_DEPENDENCIES("XdmfCore")
 
 # Add any ldpath directories that the cxx tests may need
-ADD_TEST_C_LDPATH("${CMAKE_BINARY_DIR}")
+ADD_TEST_C_LDPATH("${CMAKE_BINARY_DIR}/lib")
 
 # Add any path directoreis that the Cxx tests may need
-ADD_TEST_C_PATH("${CMAKE_BINARY_DIR}")
+ADD_TEST_C_PATH("${CMAKE_BINARY_DIR}/bin")
 
 # Add any cxx tests here:
 # Note: We don't want to use a foreach loop to test the files incase we
--- core/tests/Cxx/CMakeLists.txt.orig	2023-05-26 06:45:54
+++ core/tests/Cxx/CMakeLists.txt	2023-05-26 06:46:48
@@ -6,10 +6,10 @@
 ADD_TEST_CXX_DEPENDENCIES("XdmfCore")
 
 # Add any ldpath directories that the cxx tests may need
-ADD_TEST_CXX_LDPATH("${CMAKE_BINARY_DIR}")
+ADD_TEST_CXX_LDPATH("${CMAKE_BINARY_DIR}/lib")
 
 # Add any path directoreis that the Cxx tests may need
-ADD_TEST_CXX_PATH("${CMAKE_BINARY_DIR}")
+ADD_TEST_CXX_PATH("${CMAKE_BINARY_DIR}/bin")
 
 # Add any cxx tests here:
 # Note: We don't want to use a foreach loop to test the files incase we
