--- DESCRIPTION	2024-04-05 23:10:02.000000000 +0800
+++ DESCRIPTION	2024-04-07 14:23:06.000000000 +0800
@@ -30,9 +30,9 @@
 Imports: glue, grDevices, htmltools (>= 0.2.6), htmlwidgets (>= 0.8),
         jsonlite, magrittr, purrr (>= 0.2.2), R6, semantic.assets (>=
         1.1.0), shiny (>= 0.12.1), stats
-Suggests: covr, chromote, dplyr, DT, gapminder, knitr, leaflet, lintr,
+Suggests: covr, dplyr, DT, gapminder, knitr, leaflet, lintr,
         markdown, mockery, plotly, rcmdcheck, rmarkdown, testthat,
-        shinytest2, tibble, withr
+        tibble, withr
 VignetteBuilder: knitr
 Encoding: UTF-8
 Language: en-US
