--- DESCRIPTION.orig	2021-11-08 00:00:02.000000000 +0800
+++ DESCRIPTION	2023-06-28 06:26:32.000000000 +0800
@@ -31,7 +31,7 @@
 Imports: shiny (>= 0.12.1), htmltools (>= 0.2.6), htmlwidgets (>= 0.8),
         purrr (>= 0.2.2), stats, magrittr, jsonlite, grDevices, glue,
         R6
-Suggests: dplyr, tibble, knitr, testthat, lintr, DT, covr, leaflet,
+Suggests: dplyr, tibble, knitr, testthat, lintr, DT, covr,
         plotly, rmarkdown, markdown
 RoxygenNote: 7.1.2
 NeedsCompilation: no
