--- cmake/modules/FindPETSc.cmake.orig	2014-06-10 01:10:50.000000000 -0500
+++ cmake/modules/FindPETSc.cmake	2014-06-10 01:11:08.000000000 -0500
@@ -164,32 +164,6 @@
   resolve_libraries(PETSC_LIBRARIES "${PETSC_LIB}")
   resolve_libraries(PETSC_EXTERNAL_LIBRARIES "${PETSC_EXTERNAL_LIB_BASIC}")
 
-  # Add some extra libraries on OSX
-  if (APPLE)
-
-    # CMake will have troubel finding the gfortan libraries if compiling
-    # with clang (the libs may be required by 3rd party Fortran libraries)
-    find_program(GFORTRAN_EXECUTABLE gfortran)
-    if (GFORTRAN_EXECUTABLE)
-      execute_process(COMMAND ${GFORTRAN_EXECUTABLE} -print-file-name=libgfortran.dylib
-      OUTPUT_VARIABLE GFORTRAN_LIBRARY
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-      if (EXISTS "${GFORTRAN_LIBRARY}")
-        list(APPEND PETSC_EXTERNAL_LIBRARIES ${GFORTRAN_LIBRARY})
-      endif()
-    endif()
-
-    find_package(X11)
-    list(APPEND PETSC_INCLUDE_DIRS ${X11_X11_INCLUDE_PATH})
-    list(APPEND PETSC_EXTERNAL_LIBRARIES ${X11_LIBRARIES})
-
-    # ResolveCompilerPaths strips OSX frameworks, so add BLAS here for OSX
-    petsc_get_variable(PETSC_BLASLAPACK_LIB BLASLAPACK_LIB)
-    list(APPEND PETSC_EXTERNAL_LIBRARIES ${PETSC_BLASLAPACK_LIB})
-
-  endif()
-
   # Remove temporary Makefile
   file(REMOVE ${petsc_config_makefile})
 
