--- ./deps/v8/src/base/platform/platform-macos.cc	2025-07-14 17:30:53.000000000 -0600
+++ ./deps/v8/src/base/platform/platform-macos.cc	2025-07-14 17:35:04.000000000 -0600
@@ -49,11 +49,11 @@
     const mach_header* header = _dyld_get_image_header(i);
     if (header == nullptr) continue;
 #if V8_HOST_ARCH_I32
-    unsigned int size;
-    char* code_ptr = getsectdatafromheader(header, SEG_TEXT, SECT_TEXT, &size);
+    uintptr_t size;
+    const void* code_ptr = getsectdatafromheader(header, SEG_TEXT, SECT_TEXT, &size);
 #else
     uint64_t size;
-    char* code_ptr = getsectdatafromheader_64(
+    const void* code_ptr = getsectdatafromheader_64(
         reinterpret_cast<const mach_header_64*>(header), SEG_TEXT, SECT_TEXT,
         &size);
 #endif
