From 6337510b45f6880f6d385982b96539544ca615e2 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Mon, 10 Jul 2023 23:07:13 +0800
Subject: [PATCH 3/7] async-io-unix.c++: fix for missing SOL_LOCAL

---
 c++/src/kj/async-io-unix.c++ | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git c++/src/kj/async-io-unix.c++ c++/src/kj/async-io-unix.c++
index 3da285af..62ce2132 100644
--- src/kj/async-io-unix.c++
+++ src/kj/async-io-unix.c++
@@ -66,8 +66,8 @@
 #include <sys/ucred.h>
 #endif
 
-#if !defined(SOL_LOCAL) && (__FreeBSD__ || __DragonflyBSD__)
-// On DragonFly or FreeBSD < 12.2 you're supposed to use 0 for SOL_LOCAL.
+#if !defined(SOL_LOCAL) && (__FreeBSD__ || __DragonflyBSD__ || __APPLE__)
+// On DragonFly, FreeBSD < 12.2 and older Darwin you're supposed to use 0 for SOL_LOCAL.
 #define SOL_LOCAL 0
 #endif
 
