From a245f0b756d4dfd0aeee4315ee64657ee7a71190 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 11 Jul 2023 09:46:00 +0800
Subject: [PATCH 7/7] mutex.c++: fix for Apple

---
 c++/src/kj/mutex.c++ | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git c++/src/kj/mutex.c++ c++/src/kj/mutex.c++
index 1ddd4921..8f0534f7 100644
--- src/kj/mutex.c++
+++ src/kj/mutex.c++
@@ -866,11 +866,17 @@ void Once::reset() {
   { \
     int pthreadError = code; \
     if (pthreadError != 0) { \
-      KJ_LOG(ERROR, #code, strerror(pthreadError)); \
+      KJ_LOG(WARNING, #code, strerror(pthreadError)); \
     } \
   }
 
+#if defined(__APPLE__) && !defined(__clang__)
+Mutex::Mutex() {
+  KJ_PTHREAD_CALL(pthread_rwlock_init(&mutex, nullptr));
+}
+#else
 Mutex::Mutex(): mutex(PTHREAD_RWLOCK_INITIALIZER) {}
+#endif
 Mutex::~Mutex() {
   KJ_PTHREAD_CLEANUP(pthread_rwlock_destroy(&mutex));
 }
@@ -1057,9 +1063,16 @@ void Mutex::induceSpuriousWakeupForTest() {
   }
 }
 
+#if defined(__APPLE__) && !defined(__clang__)
+Once::Once(bool startInitialized)
+    : state(startInitialized ? INITIALIZED : UNINITIALIZED) {
+      KJ_PTHREAD_CALL(pthread_mutex_init(&mutex, nullptr));
+    }
+#else
 Once::Once(bool startInitialized)
     : state(startInitialized ? INITIALIZED : UNINITIALIZED),
       mutex(PTHREAD_MUTEX_INITIALIZER) {}
+#endif
 Once::~Once() {
   KJ_PTHREAD_CLEANUP(pthread_mutex_destroy(&mutex));
 }
