--- src/makefile.orig	2012-12-29 09:22:58.000000000 -0500
+++ src/makefile	2013-05-31 10:54:46.000000000 -0400
@@ -1,6 +1,6 @@
 #  This is the makefile for edbrowse.
 
-prefix = /usr/local
+prefix = @@PREFIX@@
 bindir = $(prefix)/bin
 
 #  Flags for gcc compilation.
@@ -15,7 +15,7 @@
 
 # C preprocessor flags.  CPPFLAGS is used in the %.o: %.c implicit rule,
 # available in GNU make.
-CPPFLAGS = $(JS_CPPFLAGS)
+#CPPFLAGS = $(JS_CPPFLAGS)
 
 # By default, we strip the executables.
 # Override this behavior on the command line, by setting STRIP to the
@@ -59,7 +59,7 @@
 
 #  You probably need to be root to do this.
 install:
-	install -Dm755 edbrowse $(DESTDIR)$(bindir)/edbrowse
+	install -m755 edbrowse $(DESTDIR)$(bindir)/edbrowse
 
 #  If you had to build the javascript library yourself,
 #  link it into /usr/lib.
@@ -102,7 +102,7 @@
 
 STATICLIBS = -lodbc -lpcre -lm -ljs -lcurl -lssl -lc -lcrypto -lpthread -lm -ldl -lrt
 edbrowse.static: $(EBOBJS) tcp.o dbops.o dbodbc.o
-	cc --static $(LFLAGS) -o edbrowse.static $(EBOBJS) tcp.o dbops.o dbodbc.o $(STATICLIBS)
+	$(CC) --static $(LFLAGS) -o edbrowse.static $(EBOBJS) tcp.o dbops.o dbodbc.o $(STATICLIBS)
 
 clean:
 	rm -f *.o edbrowse edbrowseinf edbrowseodbc edbrowse.static
