--- src/libscotch/Makefile.orig     2014-01-12 21:15:13.000000000 -0600
+++ src/libscotch/Makefile  2014-01-12 21:16:04.000000000 -0600
@@ -100,9 +100,11 @@
 
 $(libdir)/libscotch$(LIB)	:	libscotch$(LIB)
 					-$(CP) libscotch$(LIB) libscotcherr*$(LIB) $(libdir)
+					-$(CP) libscotch.dylib libscotcherr*.dylib $(libdir)
 
 $(libdir)/libptscotch$(LIB)	:	libptscotch$(LIB)
 					-$(CP) libptscotch$(LIB) libptscotcherr*$(LIB) $(libdir)
+					-$(CP) libptscotch.dylib libptscotcherr*.dylib $(libdir)
 
 ##
 ##  Library dependencies.
@@ -3064,26 +3066,32 @@
 					library_pt_f.h
 					./ptdummysizes$(EXE) "-s$(SCOTCH_NAME_SUFFIX)" library_pt_f.h ptscotchf.h
 
-libscotch$(LIB)			:	$(LIBSCOTCHDEPS)
+libscotch$(LIB)			:	$(LIBSCOTCHDEPS) libscotcherr$(LIB)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) libscotcherr.dylib $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libscotch.dylib -o libscotch.dylib
 
 libscotcherr$(LIB)		:	library_error$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libscotcherr.dylib -o libscotcherr.dylib
 
 libscotcherrexit$(LIB)		:	library_error_exit$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libscotcherrexit.dylib -o libscotcherrexit.dylib
 
-libptscotch$(LIB)		:	$(LIBPTSCOTCHDEPS)
+libptscotch$(LIB)		:	$(LIBPTSCOTCHDEPS) libscotch$(LIB)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) libscotch.dylib libscotcherr.dylib $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libptscotch.dylib -o libptscotch.dylib
 
 libptscotcherr$(LIB)		:	library_error_pt$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libptscotcherr.dylib -o libptscotcherr.dylib
 
 libptscotcherrexit$(LIB)	:	library_error_exit_pt$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(?)
 					-$(RANLIB) $(@)
+					$(CC) $(?) $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libptscotcherrexit.dylib -o libptscotcherrexit.dylib
