--- src/Makefile.orig	2018-07-14 08:24:49.000000000 -0700
+++ src/Makefile	2018-07-27 19:36:21.000000000 -0700
@@ -120,6 +120,8 @@
 					-$(CP) -f ../bin/d[agm]*$(EXE) $(bindir)
 					-$(CP) -f ../include/*esmumps*.h $(includedir)
 					-$(CP) -f ../include/*scotch*.h $(includedir)
+					-$(CP) -f ../lib/*scotch*.dylib $(libdir)
+					-$(CP) -f ../lib/*esmumps*.dylib $(libdir)
 					-$(CP) -f ../lib/*esmumps*$(LIB) $(libdir)
 					-$(CP) -f ../lib/*scotch*$(LIB) $(libdir)
 					-$(CP) -Rf ../man/* $(mandir)
