--- src/libscotchmetis/Makefile.orig	2014-01-12 20:59:29.000000000 -0600
+++ src/libscotchmetis/Makefile	2014-01-12 21:07:57.000000000 -0600
@@ -65,6 +65,7 @@
 install				:	scotch
 					-$(CP) metis.h $(includedir)
 					-$(CP) libscotchmetis$(LIB) $(libdir)
+					-$(CP) libscotchmetis.dylib $(libdir)
 
 ptinstall			:	ptscotch
 					-$(CP) parmetis.h $(includedir)
@@ -141,6 +142,7 @@
 					parmetis_dgraph_part_f$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(^)
 					-$(RANLIB) $(@)
+					$(CC) $(^) ../libscotch/libptscotch.dylib ../libscotch/libscotch.dylib ../libscotch/libscotcherr.dylib $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libptscotchparmetis.dylib -o libptscotchparmetis.dylib
 
 libscotchmetis$(LIB)		:	metis_graph_order$(OBJ)			\
 					metis_graph_order_f$(OBJ)		\
@@ -148,3 +150,4 @@
 					metis_graph_part_f$(OBJ)
 					$(AR) $(ARFLAGS) $(@) $(^)
 					-$(RANLIB) $(@)
+					$(CC) $(^) ../libscotch/libscotch.dylib ../libscotch/libscotcherr.dylib $(LDFLAGS) -dynamiclib -install_name $(prefix)/lib/libscotchmetis.dylib -o libscotchmetis.dylib
