--- build/npm/postinstall.js.orig	2022-01-06 22:30:54.000000000 +0800
+++ build/npm/postinstall.js	2022-01-07 17:28:45.000000000 +0800
@@ -90,5 +90,5 @@
 	yarnInstall(watchPath);
 }
 
-cp.execSync('git config pull.rebase merges');
-cp.execSync('git config blame.ignoreRevsFile .git-blame-ignore');
+// cp.execSync('git config pull.rebase merges');
+// cp.execSync('git config blame.ignoreRevsFile .git-blame-ignore');
