--- Makefile.orig	1995-12-05 04:26:11.000000000 -0800
+++ Makefile	2014-12-29 15:30:05.000000000 -0800
@@ -13,13 +13,13 @@
 #
 
 # Where is the binary for perl5 located on your system?
-PERL = /usr/bin/perl
+PERL = __MP_PERL__
 
 # Where do you want the sgmlspl executable script to be installed?
-BINDIR = /usr/local/bin
+BINDIR = __MP_BINDIR__
 
 # Where do you put local perl5 libaries?
-PERL5DIR = /usr/local/lib/perl5
+PERL5DIR = __MP_PERL5DIR__
 MODULEDIR = ${PERL5DIR}/SGMLS
 
 # Where do you want to put sgmlspl specifications?
@@ -29,14 +29,14 @@
 # do you intend to put it?  'make html' will create two
 # subdirectories, ${HTMLDIR}/SGMLSpm and ${HTMLDIR}/sgmlspl, and place
 # its files there.
-HTMLDIR = /usr/local/lib/www/docs
+HTMLDIR = __MP_HTMLDIR__
 
 
 #
 # End of user options.
 #
 
-HTML_SOURCES = DOC/HTML/SGMLSpm/ DOC/HTML/sgmlspl/
+HTML_SOURCES = DOC/HTML/SGMLSpm DOC/HTML/sgmlspl
 
 FILES =	${BINDIR}/sgmlspl \
 	${PERL5DIR}/SGMLS.pm \
@@ -46,13 +46,17 @@
 
 all: install docs
 
-install: install_system # install_html
+install: install_system install_html
 
 install_system: ${MODULEDIR} ${FILES}
 
 dist: ${DIST}
 
 ${BINDIR}/sgmlspl: sgmlspl.pl
+	if [ ! -d ${BINDIR} ]; then\
+	  mkdir -p ${BINDIR}; \
+	  chmod a+x ${BINDIR}; \
+	fi
 	sed -e 's!/usr/bin/perl!${PERL}!' sgmlspl.pl > ${BINDIR}/sgmlspl
 	chmod a+x,a+r ${BINDIR}/sgmlspl
 
@@ -62,7 +66,7 @@
 
 ${MODULEDIR}:
 	if [ ! -d ${MODULEDIR} ]; then\
-	  mkdir ${MODULEDIR}; \
+	  mkdir -p ${MODULEDIR}; \
 	  chmod a+x ${MODULEDIR}; \
 	fi
 
@@ -80,8 +84,12 @@
 
 install_html: ${HTML_SOURCES}
 	cd DOC; make html
+	if [ ! -d ${HTMLDIR} ]; then\
+	  mkdir -p ${HTMLDIR}; \
+	  chmod a+x ${HTMLDIR}; \
+	fi
 	rm -rf ${HTMLDIR}/SGMLSpm ${HTMLDIR}/sgmlspl
-	cp -r ${HTML_SOURCES} ${HTMLDIR}
+	cp -R ${HTML_SOURCES} ${HTMLDIR}
 	chmod a+x,a+r ${HTMLDIR}/SGMLSpm ${HTMLDIR}/sgmlspl
 	chmod a+r ${HTMLDIR}/SGMLSpm/* ${HTMLDIR}/sgmlspl/*
 
