--- Build.PL.orig
+++ Build.PL
@@ -11,7 +11,7 @@ my $build = Module::Build->new
   dist_author => "David Megginson",
   dist_abstract => "a perl5 class library for parsing the output from James Clark's SGMLS and NSGMLS parsers.",
   create_makefile_pl => 'traditional',
-  script_files =>  [ qw(script/sgmlspl.pl) ]
+  script_files =>  [ qw(script/sgmlspl) ]
  );
 
 $build->add_build_element('pl');
--- MANIFEST.orig
+++ MANIFEST
@@ -67,6 +67,6 @@ MANIFEST.SKIP
 META.yml
 MYMETA.yml
 README
-script/sgmlspl.pl
+script/sgmlspl
 t/test-SGMLS.pl
 TODO
