--- virtManager/createvm.py.orig	2022-02-12 10:13:59.000000000 -0600
+++ virtManager/createvm.py	2022-06-02 19:31:22.000000000 -0500
@@ -7,6 +7,7 @@
 import io
 import pkgutil
 import os
+import sys
 import threading
 import time
 
@@ -602,7 +603,8 @@
         uiutil.set_grid_row_visible(self.widget("arch-expander"), show_arch)
 
         if self.conn.is_qemu():
-            if not self._capsinfo.guest.is_kvm_available():
+            if (sys.platform == 'linux' and
+                not self._capsinfo.guest.is_kvm_available()):
                 error = _("KVM is not available. This may mean the KVM "
                  "package is not installed, or the KVM kernel modules "
                  "are not loaded. Your virtual machines may perform poorly.")
