--- configure.orig	2021-09-04 13:49:21.000000000 +1000
+++ configure	2021-10-10 06:15:43.000000000 +1100
@@ -3181,6 +3181,8 @@
 			;;
 		esac
 
+		FRAMEWORKINSTALLAPPSPREFIX="@@APPLICATIONS_DIR@@"
+
 		prefix=$PYTHONFRAMEWORKINSTALLDIR/Versions/$VERSION
 
 		# Add files for Mac specific code to the list of output
@@ -6444,7 +6446,7 @@
 # non-suffixed name in their versioned llvm directory.
 
 llvm_bin_dir=''
-llvm_path="${PATH}"
+llvm_path="/bin:/sbin:/usr/bin:/usr/sbin"
 if test "${CC}" = "clang"
 then
   clang_bin=`which clang`
@@ -7452,6 +7454,10 @@
                ;;
             esac
 
+            UNIVERSAL_ARCH_FLAGS="@@UNIVERSAL_ARCH_FLAGS@@"
+            LIPO_32BIT_FLAGS="@@LIPO_32BIT_FLAGS@@"
+            ARCH_RUN_32BIT="@@ARCH_RUN_32BIT@@"
+
             if test "${UNIVERSALSDK}" != "/"
             then
                 CFLAGS="${UNIVERSAL_ARCH_FLAGS} -isysroot ${UNIVERSALSDK} ${CFLAGS}"
