--- packages/base/Foreign/Storable.hs.orig	2006-09-21 06:28:34.000000000 +0800
+++ packages/base/Foreign/Storable.hs	2022-10-08 14:29:09.000000000 +0800
@@ -38,6 +38,7 @@
 
 #include "MachDeps.h"
 #include "HsBaseConfig.h"
+#include "CTypes.h"
 
 #ifdef __GLASGOW_HASKELL__
 import GHC.Storable
@@ -63,6 +64,10 @@
 import Hugs.Storable
 #endif
 
+#ifndef HTYPE_INT
+#define HTYPE_INT Int32
+#endif
+
 {- |
 The member functions of this class facilitate writing values of
 primitive types to raw memory (which may have been allocated with the
