--- lib/sqlite/sqlite3.c.orig	2011-07-06 19:03:19.000000000 -0700
+++ lib/sqlite/sqlite3.c	2011-07-06 19:03:43.000000000 -0700
@@ -330,7 +330,7 @@
  * requires the #include of specific headers which differs between
  * platforms.
  */
-#define SQLITE_INT_TO_PTR(X)   ((void*)&((char*)0)[X])
+#define SQLITE_INT_TO_PTR(X)   ((void*)(X))
 #define SQLITE_PTR_TO_INT(X)   ((int)(((char*)X)-(char*)0))
 
 /*
