--- src/qt3support/other/q3process_unix.cpp.orig	2008-05-16 09:28:53.000000000 -0600
+++ src/qt3support/other/q3process_unix.cpp	2008-05-16 09:30:10.000000000 -0600
@@ -772,7 +772,7 @@
 #endif
 	i++;
     }
-#ifdef Q_OS_MACX
+#ifdef Q_WS_MAC
     if(i) {
 	Q3CString arg_bundle = arglistQ[0];
 	QFileInfo fi(QString::fromUtf8(arg_bundle.constData()));
@@ -869,7 +869,7 @@
 		    QStringList pathList = QStringList::split( QLatin1Char(':'), QString::fromLocal8Bit(getenv( "PATH" )) );
 		    for (QStringList::Iterator it = pathList.begin(); it != pathList.end(); ++it ) {
 			QString dir = *it;
-#if defined(Q_OS_MACX) //look in a bundle
+#if defined(Q_WS_MAC) //look in a bundle
 			if(!QFile::exists(dir + QLatin1Char('/') + command) && QFile::exists(dir + QLatin1Char('/') + command + QLatin1String(".app")))
 			    dir += QLatin1Char('/') + command + QLatin1String(".app/Contents/MacOS");
 #endif
@@ -879,7 +879,7 @@
 			QFileInfo fileInfo( dir + "/" + command );
 #endif
 			if ( fileInfo.isExecutable() ) {
-#if defined(Q_OS_MACX)
+#if defined(Q_WS_MAC)
 			    arglistQ[0] = fileInfo.absFilePath().local8Bit();
 #else
 			    arglistQ[0] = fileInfo.filePath().local8Bit();
