--- src/qt3support/other/q3process_unix.cpp.orig	2008-02-20 19:52:59.000000000 -0700
+++ src/qt3support/other/q3process_unix.cpp	2008-02-20 19:53:34.000000000 -0700
@@ -761,7 +761,7 @@
 #endif
 	i++;
     }
-#ifdef Q_OS_MACX
+#ifdef Q_WS_MACX
     if(i) {
 	Q3CString arg_bundle = arglistQ[0];
 	QFileInfo fi(QString::fromUtf8(arg_bundle.constData()));
@@ -858,7 +858,7 @@
 		    QStringList pathList = QStringList::split( QLatin1Char(':'), QString::fromLocal8Bit(getenv( "PATH" )) );
 		    for (QStringList::Iterator it = pathList.begin(); it != pathList.end(); ++it ) {
 			QString dir = *it;
-#if defined(Q_OS_MACX) //look in a bundle
+#if defined(Q_WS_MACX) //look in a bundle
 			if(!QFile::exists(dir + QLatin1Char('/') + command) && QFile::exists(dir + QLatin1Char('/') + command + QLatin1String(".app")))
 			    dir += QLatin1Char('/') + command + QLatin1String(".app/Contents/MacOS");
 #endif
@@ -868,7 +868,7 @@
 			QFileInfo fileInfo( dir + "/" + command );
 #endif
 			if ( fileInfo.isExecutable() ) {
-#if defined(Q_OS_MACX)
+#if defined(Q_WS_MACX)
 			    arglistQ[0] = fileInfo.absFilePath().local8Bit();
 #else
 			    arglistQ[0] = fileInfo.filePath().local8Bit();
