From 2cac96b557d2d7441473668bd6678e4d22254246 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 15 Apr 2024 18:09:20 +0800
Subject: [PATCH 03/13] Avoid building tests

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git Makefile.am Makefile.am
index 51791983c9..a697d5707a 100644
--- Makefile.am
+++ Makefile.am
@@ -6,7 +6,7 @@
 # something. DIST_SUBDIRS stores the directories where nothing is built but
 # which have makefiles with distribution information.
 #  - src (libvlccore) is nedeed by modules
-SUBDIRS = compat doc po share src modules lib bin test
+SUBDIRS = compat doc po share src modules lib bin
 DIST_SUBDIRS = m4 $(SUBDIRS)
 
 EXTRA_DIST = \
