From e1c0b6a0957fdf85bf178ff6288413342ab71b52 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 15 Apr 2024 17:44:50 +0800
Subject: [PATCH 07/13] Sort out HAVE_OSX and HAVE_DARWIN macros to work
 correctly

diff --git configure.ac configure.ac
index fa6fa5929c..285550d18e 100644
--- configure.ac
+++ configure.ac
@@ -366,7 +366,7 @@ AM_CONDITIONAL(HAVE_OS2,     test "${SYS}" = "os2")
 
 AM_CONDITIONAL(HAVE_DARWIN,  test "${SYS}" = "darwin")
 AM_CONDITIONAL(HAVE_IOS,     test "${HAVE_IOS}" = "1")
-AM_CONDITIONAL(HAVE_OSX,     test "${HAVE_OSX}" = "1")
+AM_CONDITIONAL(HAVE_OSX,     [test "$enable_osx" != "no" -a "${SYS}" = "darwin"])
 AM_CONDITIONAL(HAVE_TVOS,    test "${HAVE_TVOS}" = "1")
 
 AM_CONDITIONAL(HAVE_NACL,    test "${SYS}" = "nacl")
diff --git modules/codec/Makefile.am modules/codec/Makefile.am
index 6d9465fdae..1ca656dba5 100644
--- modules/codec/Makefile.am
+++ modules/codec/Makefile.am
@@ -63,7 +63,7 @@ endif
 libaudiotoolboxmidi_plugin_la_SOURCES = codec/audiotoolbox_midi.c
 libaudiotoolboxmidi_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(codecdir)'
 libaudiotoolboxmidi_plugin_la_LDFLAGS += -Wl,-framework,CoreFoundation,-framework,AudioToolbox
-if HAVE_OSX
+if HAVE_DARWIN
 libaudiotoolboxmidi_plugin_la_LDFLAGS += -Wl,-framework,AudioUnit
 endif
 EXTRA_LTLIBRARIES += libaudiotoolboxmidi_plugin.la
diff --git modules/misc/Makefile.am modules/misc/Makefile.am
index 7c187d2e2a..b26f60a98f 100644
--- modules/misc/Makefile.am
+++ modules/misc/Makefile.am
@@ -38,7 +38,7 @@ endif
 EXTRA_LTLIBRARIES += libgnutls_plugin.la
 misc_LTLIBRARIES += $(LTLIBgnutls)
 
-if HAVE_DARWIN
+if HAVE_OSX
 libsecuretransport_plugin_la_SOURCES = misc/securetransport.c
 libsecuretransport_plugin_la_CFLAGS = $(AM_CFLAGS) $(SECURETRANSPORT_CFLAGS)
 libsecuretransport_plugin_la_LIBADD = $(SECURETRANSPORT_LIBS)
diff --git modules/services_discovery/Makefile.am modules/services_discovery/Makefile.am
index e070d6463b..216a43525b 100644
--- modules/services_discovery/Makefile.am
+++ modules/services_discovery/Makefile.am
@@ -33,7 +33,7 @@ libupnp_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(sddir)'
 libupnp_plugin_la_LIBADD = $(UPNP_LIBS)
 EXTRA_LTLIBRARIES += libupnp_plugin.la
 sd_LTLIBRARIES += $(LTLIBupnp)
-if HAVE_OSX
+if HAVE_DARWIN
 libupnp_plugin_la_LDFLAGS += -Wl,-framework,CoreFoundation,-framework,SystemConfiguration
 endif
 
@@ -85,6 +85,6 @@ EXTRA_LTLIBRARIES += libmicrodns_plugin.la
 libbonjour_plugin_la_SOURCES = services_discovery/bonjour.m
 libbonjour_plugin_la_OBJCFLAGS = $(AM_OBJCFLAGS) -fobjc-arc
 libbonjour_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(sddir)' -Wl,-framework,Foundation
-if HAVE_DARWIN
+if HAVE_OSX
 sd_LTLIBRARIES += libbonjour_plugin.la
 endif
diff --git src/Makefile.am src/Makefile.am
index 03f68947b2..49f8703743 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -481,7 +481,7 @@ libvlccore_la_LIBADD += $(DBUS_LIBS)
 endif
 if HAVE_DARWIN
 libvlccore_la_LDFLAGS += -Xlinker -install_name -Xlinker @rpath/libvlccore.dylib
-if !HAVE_OSX
+if HAVE_IOS
 libvlccore_la_LDFLAGS += -Wl,-framework,CFNetwork
 endif
 endif

diff --git modules/audio_output/Makefile.am modules/audio_output/Makefile.am
index bb4ac656d0..17fb2c5452 100644
--- modules/audio_output/Makefile.am
+++ modules/audio_output/Makefile.am
@@ -104,7 +104,7 @@ libauhal_plugin_la_SOURCES = audio_output/auhal.c \
 	audio_output/coreaudio_common.c audio_output/coreaudio_common.h
 libauhal_plugin_la_LDFLAGS = $(AM_LDFLAGS) -rpath '$(aoutdir)' \
 	-Wl,-framework,CoreAudio,-framework,AudioUnit,-framework,AudioToolbox,-framework,CoreServices
-if HAVE_OSX
+if HAVE_DARWIN
 aout_LTLIBRARIES += libauhal_plugin.la
 endif
 libaudiounit_ios_plugin_la_SOURCES = audio_output/audiounit_ios.m \
