http://svn.php.net/viewvc?view=revision&revision=323500
--- xrange.c	2012/02/25 11:48:49	323499
+++ xrange.c	2012/02/25 12:10:07	323500
@@ -62,7 +62,6 @@
 /* }}} */
 
 /* {{{ XRangeIterator seek method argument info */
-static
 ZEND_BEGIN_ARG_INFO(arginfo_xrange_xri_seek, 0) 
 	ZEND_ARG_INFO(0, position)
 ZEND_END_ARG_INFO();
@@ -130,17 +129,23 @@
 zend_object_value xrange_xri_object_new(zend_class_entry *class_type TSRMLS_DC)
 {
 	zend_object_value retval;
+#if PHP_VERSION_ID < 50399
 	zval *tmp;
+#endif
 
 	xrange_module_storage *internalStorage =
 		ecalloc(1, sizeof(xrange_module_storage));
 	zend_object_std_init(&internalStorage->obj, class_type TSRMLS_CC);
+#if PHP_VERSION_ID < 50399
 	zend_hash_copy(
 		internalStorage->obj.properties,
 		&php_xrange_xri_entry->default_properties,
 		(copy_ctor_func_t) zval_add_ref,
 		(void *) &tmp, sizeof(zval *)
 	);
+#else
+	object_properties_init(&internalStorage->obj, php_xrange_xri_entry);
+#endif
 
 	retval.handle = zend_objects_store_put(
 		internalStorage,
@@ -233,12 +238,11 @@
 		}
 
 		/* shove new XRangeIterator into return_value */
-		return_value->type = IS_OBJECT;
+		Z_TYPE_P(return_value) = IS_OBJECT;
 		return_value->value.obj = xrange_xri_object_new(
 			php_xrange_xri_entry TSRMLS_CC
 		);
-		return_value->refcount = 1;
-		return_value->is_ref = 0;
+		Z_ADDREF_P(return_value);
 
 		/* setup call to XRangeIterator's constructor (must do manually) */
 		zval *retval = NULL;
