--- boost/container_hash/hash.hpp.orig	2019-08-14 16:02:24
+++ boost/container_hash/hash.hpp	2024-12-22 19:24:16
@@ -118,17 +118,12 @@
 {
     namespace hash_detail
     {
-#if defined(_HAS_AUTO_PTR_ETC) && !_HAS_AUTO_PTR_ETC
         template <typename T>
         struct hash_base
         {
             typedef T argument_type;
             typedef std::size_t result_type;
         };
-#else
-        template <typename T>
-        struct hash_base : std::unary_function<T, std::size_t> {};
-#endif
 
         struct enable_hash_value { typedef std::size_t type; };
 
