--- tools/build/src/tools/clang-darwin.jam.orig	2019-08-14 05:03:50.000000000 -0700
+++ tools/build/src/tools/clang-darwin.jam	2019-08-28 08:47:22.000000000 -0700
@@ -49,7 +49,7 @@
 #   compile and link options allow you to specify addition command line options for each version
 rule init ( version ? :  command * : options * )
 {
-    command = [ common.get-invocation-command clang-darwin : clang++ 
+    command = [ common.get-invocation-command clang-darwin : __MACPORTS_CXX__
         : $(command) : /usr/bin /usr/local/bin ] ;
 
     # Determine the version
