--- Makefile.unix.orig	2014-11-27 14:40:14.000000000 -0800
+++ Makefile.unix	2014-11-27 14:40:53.000000000 -0800
@@ -10,7 +10,7 @@
 	tktable tkcon xmlrpc tkblt \
 	tclxml tkimg tkmpeg tkhtml \
 	xpa tcliis tclcheckdns signal funtools \
-	ast wcssubs \
+	wcssubs \
 	rice hcompress plio \
 	saotk zvfs ds9
 
@@ -113,7 +113,7 @@
 	tktableclean tkbltclean \
 	tclxmlclean tkimgclean tkmpegclean tkhtmlclean \
 	xpaclean tcliisclean tclcheckdnsclean signalclean funtoolsclean \
-	astclean wcssubsclean \
+	wcssubsclean \
 	riceclean hcompressclean plioclean \
 	saotkclean zvfsclean ds9clean srcclean \
 	filesclean dirsclean
