--- ds9/unix/Makefile.in.orig	2016-01-07 00:02:12.000000000 -0500
+++ ds9/unix/Makefile.in	2016-01-07 01:08:35.000000000 -0500
@@ -52,10 +52,6 @@
 
 LLIBS	= \
 	../../lib/$(TKSAODIR)/libtksao1.0.a \
-	../../lib/$(TKSAODIR)/libfuntools.a \
-	../../lib/$(TKSAODIR)/libast.a \
-	../../lib/$(TKSAODIR)/libast_err.a \
-	../../lib/$(TKSAODIR)/libast_pal.a \
 	../../lib/$(TKSAODIR)/libtksao1.0.a \
 	../../lib/$(TKHTMLDIR)/libtkhtml1.0.a \
 	../../lib/$(TKMPEGDIR)/libtkmpeg1.0.a \
@@ -73,7 +69,6 @@
 	../../lib/$(TKIMGVER)/libtkimg1.4.a \
 	../../lib/libtiff.a \
 	../../lib/$(TCLZVFSDIR)/libtclzvfs1.0.a \
-	../../lib/libxpa.a \
 	../../lib/$(TCLIISDIR)/libtcliis1.0.a \
 	../../lib/$(TCLCHECKDNSDIR)/libtclcheckdns1.1.a \
 	../../lib/$(TCLSIGNALDIR)/libtclsignal1.4.a \
@@ -112,29 +107,51 @@
 	strip ds9Base
 	cp ds9Base ds9
 	cp ds9 ../../bin/.
-	cp ds9.zip ../../bin/.
-	cd ../../bin; codesign -s "SAOImage DS9" ds9
 
 dds9zip: ds9Base ds9.zip
 	$(RM) $@
 	cp ds9Base ds9
 	cp ds9 ../../bin/.
 	cp ds9.zip ../../bin/.
-	cd ../../bin; codesign -s "SAOImage DS9" ds9
 
 ds9Base	: $(OBJS) $(LLIBS)
 	$(RM) $@
 	$(CC) $(CPPFLAGS) $(CFLAGS) -o $@ \
 	$(OBJS) \
-	$(LLIBS) \
-	$(LIBS) \
+	../../lib/$(TKSAODIR)/libtksao1.0.a \
+	-lfuntools \
+	@prefix@/lib/libast.a \
+	@prefix@/lib/libast_err.a \
+	@prefix@/lib/libast_pal.a \
+	../../lib/$(TKSAODIR)/libtksao1.0.a \
+	../../lib/$(TKHTMLDIR)/libtkhtml1.0.a \
+	../../lib/$(TKMPEGDIR)/libtkmpeg1.0.a \
+	../../lib/$(TCLXMLVER)/libTclxml3.2.a \
+	../../lib/$(TKTABLEVER)/libTktable2.10.a \
+	../../lib/$(TKIMGVER)/libtkimgpng1.4.a \
+	../../lib/$(TKIMGVER)/libpngtcl1.4.3.a \
+	../../lib/$(TKIMGVER)/libtkimgtiff1.4.a \
+	../../lib/$(TKIMGVER)/libtifftcl3.9.4.a \
+	../../lib/$(TKIMGVER)/libtkimgjpeg1.4.a \
+	../../lib/$(TKIMGVER)/libjpegtcl8.2.a \
+	../../lib/$(TKIMGVER)/libtkimggif1.4.a \
+	../../lib/$(TKIMGVER)/libtkimgwindow1.4.a \
+	../../lib/$(TKIMGVER)/libzlibtcl1.2.5.a \
+	../../lib/$(TKIMGVER)/libtkimg1.4.a \
+	../../lib/libtiff.a \
+	../../lib/$(TCLZVFSDIR)/libtclzvfs1.0.a \
+	-lxpa \
+	../../lib/$(TCLIISDIR)/libtcliis1.0.a \
+	../../lib/$(TCLCHECKDNSDIR)/libtclcheckdns1.1.a \
+	../../lib/$(TCLSIGNALDIR)/libtclsignal1.4.a \
+	../../lib/$(TKBLTDIR)/libtkblt3.0.a \
+	$(LIBS) -ljbig \
 	$(TK_STUB_LIB_SPEC) $(TK_LIB_SPEC) \
 	$(TCL_STUB_LIB_SPEC) $(TCL_LIB_SPEC) \
 	$(TK_LIBS)
 
 ds9.zip	: $(FILES)
 	cd zipdir; find . -depth -name "CVS" -exec rm -r {} \;
-	cd zipdir; zip -r ../ds9.zip *	
 
 zipdir	: 
 	mkdir zipdir
