--- tksao/configure.ac.orig	2019-11-15 11:07:34.000000000 -0500
+++ tksao/configure.ac	2019-11-15 11:09:31.000000000 -0500
@@ -59,6 +59,7 @@
 # necessary to compile executables.
 #-----------------------------------------------------------------------
 
+AC_PROG_CXX
 TEA_SETUP_COMPILER
 
 #-----------------------------------------------------------------------
@@ -325,7 +326,7 @@
     :
     #TEA_ADD_SOURCES([unix/unixFile.c])
     #TEA_ADD_LIBS([-lsuperfly])
-    TEA_ADD_INCLUDES([-I/usr/include/libxml2])
+    TEA_ADD_INCLUDES([-I@prefix@/include/libxml2])
 fi
 
 #--------------------------------------------------------------------
