--- ds9/Makefile.orig	2014-04-18 00:11:43.000000000 -0700
+++ ds9/Makefile	2014-04-18 01:17:01.000000000 -0700
@@ -50,14 +50,9 @@
 	../lib/librice.a \
 	../lib/libhcomp.a \
 	../lib/libplio.a \
-	../lib/libast.a \
-	../lib/libast_err.a \
-	../lib/libast_pal.a \
 	../lib/libsaotk.a \
 	../lib/libwcs.a \
 	../lib/libzvfs.a \
-	../lib/libzip.a \
-	../lib/libz.a \
 	../lib/libxpa.a \
 	../lib/libiis.a \
 	../lib/libcheckdns.a \
@@ -142,7 +137,7 @@
 	cp ds9Base ds9
 
 ds9.zip	: $(FILES)
-	cd zipdir; ../../bin/zip -r9 ../ds9.zip *	
+	cd zipdir; zip -r9 ../ds9.zip *	
 
 else
 
@@ -374,9 +369,47 @@
 ds9Base	: $(OBJS) $(LIBS)
 	$(RM) $@
 	$(CXX) ${OPTS} \
-	-o $@ $(OBJS) $(LIBS) \
+	-o $@ $(OBJS) \
+	../lib/libsaotk.a \
+	../lib/libtkhtml.a \
+	../lib/libtkmpeg.a \
+	../lib/$(TCLXMLVER)/libTclxml3.2.a \
+	../lib/$(TKTABLEVER)/libTktable2.10.a \
+	../lib/$(TKIMGVER)/libtkimgpng1.4.a \
+	../lib/$(TKIMGVER)/libpngtcl1.4.3.a \
+	../lib/$(TKIMGVER)/libtkimgtiff1.4.a \
+	../lib/$(TKIMGVER)/libtifftcl3.9.4.a \
+	../lib/$(TKIMGVER)/libtkimgjpeg1.4.a \
+	../lib/$(TKIMGVER)/libjpegtcl8.2.a \
+	../lib/$(TKIMGVER)/libtkimggif1.4.a \
+	../lib/$(TKIMGVER)/libtkimgppm1.4.a \
+	../lib/$(TKIMGVER)/libtkimgwindow1.4.a \
+	../lib/$(TKIMGVER)/libzlibtcl1.2.5.a \
+	../lib/$(TKIMGVER)/libtkimg1.4.a \
+	../lib/libtiff.a \
+	../lib/libfuntools.a \
+	../lib/librice.a \
+	../lib/libhcomp.a \
+	../lib/libplio.a \
+	@prefix@/lib/libast.a \
+	@prefix@/lib/libast_err.a \
+	@prefix@/lib/libast_pal.a \
+	../lib/libsaotk.a \
+	../lib/libwcs.a \
+	../lib/libzvfs.a \
+	-lzip \
+	-lz \
+	../lib/libxpa.a \
+	../lib/libiis.a \
+	../lib/libcheckdns.a \
+	../lib/libsignal_ext.a \
+	../lib/libxxlib.a \
+	../lib/libBLTX30.a \
+	../lib/libBLTCore30.a \
+	../lib/libtk8.5.a \
+	../lib/libtcl8.5.a \
 	-L$(X11LIB) -lX11 -lXext -lXft -lXrender -lXss -lfontconfig -lfreetype \
-	-lxml2 \
+	-lxml2 -ljbig \
 	-lstdc++ -liconv
 endif
 
