--- ds9/macosx/Makefile.in.orig	2016-12-22 13:01:26.000000000 -0500
+++ ds9/macosx/Makefile.in	2016-12-22 13:02:01.000000000 -0500
@@ -38,10 +38,6 @@
 
 LLIBS	= \
 	../../lib/tksao$(TKSAOVER)/libtksao$(TKSAOVER).a \
-	../../lib/libfuntools.a \
-	../../lib/libast.a \
-	../../lib/libast_err.a \
-	../../lib/libast_pal.a \
 	../../lib/tksao$(TKSAOVER)/libtksao$(TKSAOVER).a \
 	../../lib/tkhtml1$(TKHTMLVER)/libtkhtml1$(TKHTMLVER).a \
 	../../lib/tkmpeg$(TKMPEGVER)/libtkmpeg$(TKMPEGVER).a \
@@ -57,7 +53,6 @@
 	../../lib/Img$(TKIMGVER)/libtkimgwindow1.4.2.a \
 	../../lib/Img$(TKIMGVER)/libzlibtcl1.2.8.a \
 	../../lib/Img$(TKIMGVER)/libtkimg1.4.2.a \
-	../../lib/libxpa.a \
 	../../lib/tcliis$(TCLIISVER)/libtcliis$(TCLIISVER).a \
 	../../lib/tclcheckdns$(TCLCHECKDNSVER)/libtclcheckdns$(TCLCHECKDNSVER).a \
 	../../lib/tclsignal$(TCLSIGNALVER)/libtclsignal$(TCLSIGNALVER).a \
@@ -79,7 +74,7 @@
 .PHONY	: ds9 debug appdir framework tcl tcl_debug tk tk_debug tcllib tkcon
 .PHONY	: tkblt xpa compiler cleanup
 
-ds9	: ds9base appdir framework tcl tk tcllib tkcon tkblt xpa compiler cleanup
+ds9	: ds9base appdir framework tcl tk tcllib tkcon tkblt compiler cleanup
 	cp -p ds9base $(APPDIR)/Contents/MacOS/ds9
 	strip $(APPDIR)/Contents/MacOS/ds9
 	install_name_tool -change \
@@ -90,7 +85,6 @@
 	/Library/Frameworks/Tk.framework/Versions/8.6/Tk \
 	@executable_path/../Frameworks/Tk.framework/Tk \
 	$(APPDIR)/Contents/MacOS/ds9
-	cd ../../bin; codesign --deep -f -s "$(ID)" SAOImage\ DS9.app
 
 debug	: ds9base appdir framework tcl tk tcllib tkcon tkblt xpa compiler cleanup
 	cp -p ds9base $(APPDIR)/Contents/MacOS/ds9
@@ -115,8 +109,34 @@
 	$(RM) $@
 	$(CC) $(CFLAGS) -o $@ \
 	$(OBJS) \
-	$(LLIBS) \
-	$(LIBS) \
+	../../lib/tksao$(TKSAOVER)/libtksao$(TKSAOVER).a \
+	-lfuntools \
+	/opt/local/lib/libast.a \
+	/opt/local/lib/libast_err.a \
+	/opt/local/lib/libast_pal.a \
+	../../lib/tksao$(TKSAOVER)/libtksao$(TKSAOVER).a \
+	../../lib/tkhtml1$(TKHTMLVER)/libtkhtml1$(TKHTMLVER).a \
+	../../lib/tkmpeg$(TKMPEGVER)/libtkmpeg$(TKMPEGVER).a \
+	../../lib/Tclxml$(TCLXMLVER)/libTclxml$(TCLXMLVER).a \
+	../../lib/Tktable$(TKTABLEVER)/libTktable$(TKTABLEVER).a \
+	../../lib/Img$(TKIMGVER)/libtkimgpng1.4.2.a \
+	../../lib/Img$(TKIMGVER)/libpngtcl1.4.12.a \
+	../../lib/Img$(TKIMGVER)/libtkimgtiff1.4.2.a \
+	../../lib/Img$(TKIMGVER)/libtifftcl3.9.4.a \
+	../../lib/Img$(TKIMGVER)/libtkimgjpeg1.4.2.a \
+	../../lib/Img$(TKIMGVER)/libjpegtcl8.4.a \
+	../../lib/Img$(TKIMGVER)/libtkimggif1.4.2.a \
+	../../lib/Img$(TKIMGVER)/libtkimgwindow1.4.2.a \
+	../../lib/Img$(TKIMGVER)/libzlibtcl1.2.8.a \
+	../../lib/Img$(TKIMGVER)/libtkimg1.4.2.a \
+	-lxpa \
+	../../lib/tcliis$(TCLIISVER)/libtcliis$(TCLIISVER).a \
+	../../lib/tclcheckdns$(TCLCHECKDNSVER)/libtclcheckdns$(TCLCHECKDNSVER).a \
+	../../lib/tclsignal$(TCLSIGNALVER)/libtclsignal$(TCLSIGNALVER).a \
+	../../lib/tkblt$(TKBLTVER)/libtkbltstub$(TKBLTVER).a \
+	../../lib/tkblt$(TKBLTVER)/libtkblt$(TKBLTVER).a \
+	../../lib/tkmacosx$(TKMACOSXVER)/libtkmacosx$(TKMACOSXVER).a \
+	$(LIBS)$ -ljbig \
 	$(TK_STUB_LIB_SPEC) $(TK_LIB_SPEC) \
 	$(TCL_STUB_LIB_SPEC) $(TCL_LIB_SPEC) \
 	$(TK_LIBS)
