--- ds9/macos/Makefile.in.orig	2022-03-29 12:26:36.000000000 -0400
+++ ds9/macos/Makefile.in	2022-03-29 12:28:45.000000000 -0400
@@ -119,9 +119,6 @@
 
 LLIBS	= \
 	$(subst /$(space),/,$(patsubst -L%,%/,$(patsubst -l%,lib%.a,$(SPECS)))) \
-	$(libdir)/libast.a \
-	$(libdir)/libast_err.a \
-	$(libdir)/libast_pal.a \
 	$(libdir)/libfuntools.a \
 	$(libdir)/libxpa.a
 
@@ -187,6 +184,9 @@
 	$(RM) $@
 	$(CXX) $(CFLAGS) -o $@ \
 	$(OBJS) \
+	-last \
+	-last_err \
+	-last_pal \
 	$(LLIBS) \
 	-L @macports_prefix@/lib/openssl-1.0 -lssl -lcrypto \
 	$(TK_LIB_SPEC) \
--- ds9/unix/Makefile.in.orig	2022-03-29 12:26:36.000000000 -0400
+++ ds9/unix/Makefile.in	2022-03-29 12:30:27.000000000 -0400
@@ -118,9 +118,6 @@
 
 LLIBS	= \
 	$(subst /$(space),/,$(patsubst -L%,%/,$(patsubst -l%,lib%.a,$(SPECS)))) \
-	$(libdir)/libast.a \
-	$(libdir)/libast_err.a \
-	$(libdir)/libast_pal.a \
 	$(subst /$(space),/,$(patsubst -L%,%/,$(patsubst -l%,lib%.a,$(tksao_LIB_SPEC)))) \
 	$(libdir)/libfuntools.a \
 	$(libdir)/libxpa.a
@@ -192,6 +189,9 @@
 	$(RM) $@
 	$(CXX) $(CFLAGS) -o $@ \
 	$(OBJS) \
+	-last \
+	-last_err \
+	-last_pal \
 	$(LLIBS) \
 	-L @macports_prefix@/lib/openssl-1.0 -lssl -lcrypto \
 	$(TK_LIBS) \
--- make.include.orig	2022-03-29 12:26:36.000000000 -0400
+++ make.include	2022-03-29 12:32:32.000000000 -0400
@@ -383,54 +383,6 @@
 
 #-------------------------- ast
 
-ast	: ast/Makefile ast/ast.h ast/.libs/libast.a ast/.libs/libast_err.a ast/.libs/libast_pal.a
-
-ast/Makefile :
-	@echo ""
-	@echo "*** $@ ***"
-	cd ast; \
-	./configure \
-	$(ASTFLAGS) \
-	--without-pthreads --enable-shared=no \
-	star_cv_cnf_trail_type=long star_cv_cnf_f2c_compatible=no \
-	$(CONFIGFLAGS) \
-	$(PREFIX)
-	cd ast; touch configure.ac aclocal.m4 configure Makefile.am Makefile.in
-
-ast/ast.h :
-	@echo ""
-	@echo "*** $@ ***"
-	$(MAKE) -C ast -j ast.h
-	$(MAKE) -C ast -j $(JOBS) libast.la
-	$(MAKE) -C ast -j $(JOBS) libast_err.la
-	$(MAKE) -C ast -j $(JOBS) libast_pal.la
-	cp ast/ast.h $(includedir)/.
-	cp ast/src/grf.h $(includedir)/.
-	cp ast/src/grf3d.h $(includedir)/.
-	cp ast/.libs/libast.a $(libdir)/.
-	cp ast/.libs/libast_err.a $(libdir)/.
-	cp ast/.libs/libast_pal.a $(libdir)/.
-
-ast/.libs/libast.a :
-	@echo ""
-	@echo "*** $@ ***"
-	$(MAKE) -C ast -j ast.h
-	cp ast/ast.h $(includedir)/.
-	cp ast/grf.h $(includedir)/.
-	cp ast/grf3d.h $(includedir)/.
-
-ast/.libs/libast_err.a :
-	@echo ""
-	@echo "*** $@ ***"
-	$(MAKE) -C ast -j $(JOBS) libast_err.la
-	cp ast/.libs/libast_err.a $(libdir)/.
-
-ast/.libs/libast_pal.a :
-	@echo ""
-	@echo "*** $@ ***"
-	$(MAKE) -C ast -j $(JOBS) libast_pal.la
-	cp ast/.libs/libast_pal.a $(libdir)/.
-
 #-------------------------- funtools
 
 funtools : funtools/libfuntools.a
