--- DESCRIPTION.orig	2023-09-22 17:20:02.000000000 +0700
+++ DESCRIPTION	2023-09-22 23:39:30.000000000 +0700
@@ -6,7 +6,6 @@
 Description: Some utility functions for validation, data manipulation or color palettes. These functions can be helpful to reduce internal codes everywhere in package development.
 Depends: R (>= 4.1.0), methods
 Imports: dplyr, fastmatch, Polychrome, rlang, scales
-Suggests: randomcoloR
 License: MIT + file LICENSE
 Encoding: UTF-8
 URL: https://github.com/ycli1995/easy.utils
