--- gcr/Makefile.am.orig	2014-09-23 00:40:08.000000000 -0700
+++ gcr/Makefile.am	2014-12-04 21:59:38.000000000 -0800
@@ -159,13 +159,13 @@ gcr/gcr-dbus-generated.c: $(DBUS_XML_DEF
 	$(AM_V_GEN) $(GDBUS_CODEGEN) --interface-prefix org.gnome.keyring.internal. \
 		--generate-c-code gcr/gcr-dbus-generated --c-namespace GcrDBus $^
 	$(AM_V_GEN) sed -e 's/gcr_dbus/_gcr_dbus/g' \
+		gcr/gcr-dbus-generated.h > gcr/gcr-dbus-generated.h.tmp && \
+	mv gcr/gcr-dbus-generated.h.tmp gcr/gcr-dbus-generated.h
+	$(AM_V_GEN) sed -e 's/gcr_dbus/_gcr_dbus/g' \
 		gcr/gcr-dbus-generated.c > gcr/gcr-dbus-generated.c.tmp && \
 	mv gcr/gcr-dbus-generated.c.tmp gcr/gcr-dbus-generated.c
 
 gcr/gcr-dbus-generated.h: gcr/gcr-dbus-generated.c
-	$(AM_V_GEN) sed -e 's/gcr_dbus/_gcr_dbus/g' \
-		gcr/gcr-dbus-generated.h > gcr/gcr-dbus-generated.h.tmp && \
-	mv gcr/gcr-dbus-generated.h.tmp gcr/gcr-dbus-generated.h
 
 pkgconfig_DATA += \
 	gcr-base-$(GCR_MAJOR).pc
