From 8b1c65e2a07c4c8eb14a55777bfcffb30cf7f6ab Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Sat, 1 Jul 2017 19:28:32 +0200
Subject: grid view: activate smooth scrolling to get the overshoot effect

---
 gthumb/gth-grid-view.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gthumb/gth-grid-view.c b/gthumb/gth-grid-view.c
index f110918..ba5846a 100644
--- gthumb/gth-grid-view.c
+++ gthumb/gth-grid-view.c
@@ -1042,6 +1042,7 @@ gth_grid_view_realize (GtkWidget *widget)
 	attributes.height = MAX (self->priv->height, allocation.height);
 	attributes.event_mask = (GDK_EXPOSURE_MASK
 				 | GDK_SCROLL_MASK
+				 | GDK_SMOOTH_SCROLL_MASK
 				 | GDK_POINTER_MOTION_MASK
 				 | GDK_ENTER_NOTIFY_MASK
 				 | GDK_LEAVE_NOTIFY_MASK
-- 
cgit v0.12

