--- gthumb/make-header.sh.orig	2010-07-15 11:45:39.000000000 -0700
+++ gthumb/make-header.sh	2010-10-10 09:24:37.000000000 -0700
@@ -3,9 +3,9 @@
 template="$1"
 
 shift
-includes="\n"
+includes="@"
 for i in "$@"; do
-	includes="${includes}#include <gthumb/"$i">\n"
+	includes="${includes}#include <gthumb/"$i">@"
 done
 
-sed -e 's|@@|'"$includes"'|' $template
+sed -e 's|@@|'"$includes"'|' $template | tr @ "\n"
