From a5f0ad8c2ab916846b83caa9904942f5c15e53d7 Mon Sep 17 00:00:00 2001
From: Paolo Bacchilega <paobac@src.gnome.org>
Date: Mon, 31 Jul 2017 08:02:00 +0200
Subject: use autoaudiosink instead of gsettingsaudiosink

[bug #785612]
---
 extensions/slideshow/gth-slideshow.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extensions/slideshow/gth-slideshow.c b/extensions/slideshow/gth-slideshow.c
index 8a2724e..f1dc0de 100644
--- extensions/slideshow/gth-slideshow.c
+++ extensions/slideshow/gth-slideshow.c
@@ -434,7 +434,7 @@ gth_slideshow_show_cb (GtkWidget    *widget,
 
 			self->priv->playbin = gst_element_factory_make ("playbin", "playbin");
 			g_object_set (self->priv->playbin,
-				      "audio-sink", gst_element_factory_make ("gsettingsaudiosink", "audiosink"),
+				      "audio-sink", gst_element_factory_make ("autoaudiosink", "audiosink"),
 				      "flags", _GST_PLAY_FLAG_AUDIO,
 				      "volume", 1.0,
 				      NULL);
-- 
cgit v0.12

