--- extensions/gstreamer_tools/gth-media-viewer-page.c.orig	2015-12-23 07:46:18.000000000 -0800
+++ extensions/gstreamer_tools/gth-media-viewer-page.c	2018-12-23 18:58:50.000000000 -0800
@@ -22,7 +22,7 @@
 #include <config.h>
 #include <math.h>
 #include <gdk/gdkkeysyms.h>
-#include <gdk/gdkx.h>
+//#include <gdk/gdkx.h>
 #include <gst/gst.h>
 #include <gst/video/videooverlay.h>
 #include <gthumb.h>
