--- configure.ac.orig	2015-04-14 08:12:09.000000000 -0700
+++ configure.ac	2015-04-24 14:26:29.000000000 -0700
@@ -77,7 +77,7 @@
 dnl ===========================================================================
 
 CFLAGS=${CFLAGS-:""}
-CFLAGS="$CFLAGS -fPIC -DPIC -Wl,--as-needed"
+CFLAGS="$CFLAGS -fPIC -DPIC"
 
 AC_ARG_ENABLE(deprecated,
 	      AS_HELP_STRING([--disable-deprecated], [do not use deprecated API]),
@@ -106,7 +106,7 @@
 dnl ===========================================================================
 
 if test x$ENABLE_DEBUG = xyes; then
-WARN_CFLAGS="-Wall -Wcast-align -Wtype-limits -Wclobbered -Wempty-body -Wignored-qualifiers -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-sign-compare -Wformat-security"
+WARN_CFLAGS="-Wall -Wcast-align -Wtype-limits -Wempty-body -Wignored-qualifiers -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-sign-compare -Wformat-security"
 else
 WARN_CFLAGS="-Wall"
 fi
