--- a/pdftk/report.cc
+++ b/pdftk/report.cc
@@ -1233,7 +1233,7 @@
 								string_to_jcharstring( jvs, jvs_size, &jvs_len, it->second );
 
 								info_p->put( new itext::PdfName( JvNewStringLatin1(it->first.c_str()) ),
-														 new itext::PdfString( JvNewString(jvs, jvs_len), itext::PdfObject::TEXT_UNICODE ) );
+														 new itext::PdfString( JvNewStringUTF((char* )it->second.c_str()), (strcmp(it->first.c_str(), "ModDate") && strcmp(it->first.c_str(), "CreationDate")) ? itext::PdfObject::TEXT_UNICODE : itext::PdfObject::TEXT_PDFDOCENCODING ) );
 							}
 						}
 				}
