--- xraylib-2.16.0-released/configure	2012-09-17 17:19:28.000000000 +0200
+++ xraylib-2.16.0/configure	2012-09-19 17:27:05.000000000 +0200
@@ -17732,20 +17732,23 @@
 		# credits goes to momjian, I think. I'd like to put the right name
 		# in the credits, if someone can point me in the right direction... ?
 		#
-		if test -n "$ac_python_libdir" -a -n "$ac_python_library" \
-			-a x"$ac_python_library" != x"$ac_python_soname"
-		then
+		#echo "ac_python_libdir: x$ac_python_libdirx"
+		#echo "ac_python_library: x$ac_python_libraryx"
+		#echo "ac_python_soname: x$ac_python_sonamex"
+		#if test -n "$ac_python_libdir" -a -n "$ac_python_library" \
+		#	-a x"$ac_python_library" != x"$ac_python_soname"
+		#then
 			# use the official shared library
 			ac_python_library=`echo "$ac_python_library" | sed "s/^lib//"`
-			PYTHON_LDFLAGS="-L$ac_python_libdir -l$ac_python_library"
-		else
-			# old way: use libpython from python_configdir
-			ac_python_libdir=`$PYTHON -c \
-			  "from distutils.sysconfig import get_python_lib as f; \
-			  import os; \
-			  print (os.path.join(f(plat_specific=1, standard_lib=1), 'config'));"`
 			PYTHON_LDFLAGS="-L$ac_python_libdir -lpython$ac_python_version"
-		fi
+		#else
+		#	# old way: use libpython from python_configdir
+		#	ac_python_libdir=`$PYTHON -c \
+		#	  "from distutils.sysconfig import get_python_lib as f; \
+		#	  import os; \
+		#	  print (os.path.join(f(plat_specific=1, standard_lib=1), 'config'));"`
+		#	PYTHON_LDFLAGS="-L$ac_python_libdir -lpython$ac_python_version"
+		#fi
 
 		if test -z "PYTHON_LDFLAGS"; then
 			{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING:
@@ -17800,7 +17803,7 @@
 	if test -z "$PYTHON_EXTRA_LIBS"; then
 	   PYTHON_EXTRA_LIBS=`$PYTHON -c "import distutils.sysconfig; \
                 conf = distutils.sysconfig.get_config_var; \
-                print (conf('LOCALMODLIBS') + ' ' + conf('LIBS'))"`
+                print (conf('LOCALMODLIBS') + ' ' + conf('LDFLAGS') + ' ' + conf('LIBS'))"`
 	fi
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_EXTRA_LIBS" >&5
 $as_echo "$PYTHON_EXTRA_LIBS" >&6; }
@@ -18080,7 +18083,7 @@
 JAVA_TEST=Test.java
 CLASS_TEST=Test.class
 cat << \EOF > $JAVA_TEST
-/* #line 18083 "configure" */
+/* #line 18086 "configure" */
 public class Test {
 }
 EOF
@@ -18267,7 +18270,7 @@
 if $UUDECODE Test.uue; then
         ac_cv_prog_uudecode_base64=yes
 else
-        echo "configure: 18270: uudecode had trouble decoding base 64 file 'Test.uue'" >&5
+        echo "configure: 18273: uudecode had trouble decoding base 64 file 'Test.uue'" >&5
         echo "configure: failed file was:" >&5
         cat Test.uue >&5
         ac_cv_prog_uudecode_base64=no
@@ -18298,7 +18301,7 @@
 CLASS_TEST=Test.class
 TEST=Test
 cat << \EOF > $JAVA_TEST
-/* [#]line 18301 "configure" */
+/* [#]line 18304 "configure" */
 public class Test {
 public static void main (String args[]) {
         System.exit (0);
