--- xraylib-2.16.0-orig/python/xrlsetup.py.in	2013-05-28 15:45:26.000000000 +0200
+++ xraylib-2.16.0/python/xrlsetup.py.in	2013-05-28 16:28:08.000000000 +0200
@@ -20,6 +20,6 @@
 from distutils.core import setup, Extension
 
 
-xraylib_module = Extension('_xraylib', sources=['xraylib_wrap.c'],libraries = ['xrl', @PYTHON_LIBS_FIXED@], include_dirs = [@PYTHON_INCLUDE_FIXED@], library_dirs=['@abs_top_builddir@/src/.libs'])
+xraylib_module = Extension('_xraylib', sources=['xraylib_wrap.c'],libraries = ['xrl'], library_dirs=['@abs_top_builddir@/src/.libs'])
 
 setup (name = 'xraylib' ,version = '@PYTHON_XRL_VERSION@' , author = "Tom Schoonjans", description = """python bindings of xraylib""", ext_modules = [xraylib_module], py_modules = ["xraylib"])
