--- xraylib-2.16.0-orig/src/xraylib.i	2013-05-28 15:45:26.000000000 +0200
+++ xraylib-2.16.0/src/xraylib.i	2013-05-28 16:41:24.000000000 +0200
@@ -351,7 +351,7 @@
                 lua_getfield(L, $input,"atom");
                 if (lua_istable(L,-1)) {
                         /* count number of elements */
-                        size_t n_atom = lua_objlen(L, -1);
+                        size_t n_atom = lua_rawlen(L, -1);
                         if (n_atom != cs->n_atom) {
                                 SWIG_exception(SWIG_RuntimeError,"n_atom hash value differs from number of elements");
                         }
