--- src/osx/cocoa/mediactrl.mm.orig	2017-05-15 13:54:48.000000000 -0700
+++ src/osx/cocoa/mediactrl.mm	2017-11-08 04:36:53.000000000 -0700
@@ -52,7 +52,7 @@
     #endif
 #endif
 
-#if wxOSX_USE_AVFOUNDATION && wxOSX_USE_COCOA && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9
+#if wxOSX_USE_AVFOUNDATION && wxOSX_USE_COCOA && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9 && defined(__LP64__)
     #define wxOSX_USE_AVKIT 1
 #else
     #define wxOSX_USE_AVKIT 0
