--- sounddecompress.py.orig	2010-03-08 12:32:51.000000000 -0600
+++ sounddecompress.py	2010-09-25 02:15:46.000000000 -0500
@@ -87,9 +87,11 @@
                 ##  which are not set to StreamCompressed
                 for pageNum in range(0, age.getNumPages()):
                     page = plResMgr.FindPage(age.getPageLoc(pageNum, plResMgr.getVer()))
+                    if (page == None):
+                        raise Exception("Unable to completely load age "+age.name+": Can't find page "+str(age.getPageLoc(pageNum, plResMgr.getVer())))
                     if PyHSPlasma.plFactory.kSoundBuffer in plResMgr.getTypes(page.location):
                         for key in plResMgr.getKeys(page.location, PyHSPlasma.plFactory.kSoundBuffer):	            
-                            soundBuffer = PyHSPlasma.plSoundBuffer.Convert(key.object)
+                            soundBuffer = key.object
 
                             if soundBuffer.fileName in queue.keys():
                                 channelOptions = queue[soundBuffer.fileName]
