--- a/src/dispatch_common.c	2016-07-29 10:24:17.000000000 -0700
+++ b/src/dispatch_common.c	2016-07-29 10:25:06.000000000 -0700
@@ -348,7 +348,12 @@
 epoxy_extension_in_string(const char *extension_list, const char *ext)
 {
     const char *ptr = extension_list;
-    int len = strlen(ext);
+    int len;
+    
+    if (!extension_list)
+        return false;
+        
+    len = strlen(ext);
 
     /* Make sure that don't just find an extension with our name as a prefix. */
     while (true) {
