--- src/dispatch_common.h.orig	2022-02-17 19:56:12.000000000 +0700
+++ src/dispatch_common.h	2022-09-01 21:11:53.000000000 +0700
@@ -28,7 +28,7 @@
 #define PLATFORM_HAS_GLX ENABLE_GLX
 #define PLATFORM_HAS_WGL 1
 #elif defined(__APPLE__)
-#define PLATFORM_HAS_EGL 0 
+#define PLATFORM_HAS_EGL ENABLE_EGL
 #define PLATFORM_HAS_GLX ENABLE_GLX
 #define PLATFORM_HAS_WGL 0
 #elif defined(ANDROID)

--- src/dispatch_common.c.orig	2022-02-17 19:56:12.000000000 +0700
+++ src/dispatch_common.c	2022-09-01 21:11:35.000000000 +0700
@@ -174,7 +174,8 @@
 #include "dispatch_common.h"
 
 #if defined(__APPLE__)
-#define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
+#define GLX_LIB "@PREFIX@/lib/libGL.1.dylib"
+#define EGL_LIB "@PREFIX@/lib/libEGL.1.dylib"
 #define OPENGL_LIB "/System/Library/Frameworks/OpenGL.framework/Versions/Current/OpenGL"
 #define GLES1_LIB "libGLESv1_CM.so"
 #define GLES2_LIB "libGLESv2.so"
