--- BlockOut/Makefile.orig	2008-01-09 12:29:06.000000000 -0600
+++ BlockOut/Makefile	2013-09-30 04:49:38.000000000 -0500
@@ -34,17 +34,17 @@
 
 #---------------------------------------------------------------------
 
-SDL_ROOT	= /users/pons/test/opengl/SDL
-IMGLIB_ROOT	= ../ImageLib
+SDL_ROOT	= $(PREFIX)
+IMGLIB_ROOT	= ../ImageLib/src
 
 CXX		= g++
 
-CXXFLAGS =     -g -D_DEBUG -Dlinux -c \
-               -I$(SDL_ROOT)/include/SDL -I$(IMGLIB_ROOT) -I.
+CXXFLAGS =     -g -D_DEBUG -c \
+               -I$(SDL_ROOT)/include/SDL -I$(IMGLIB_ROOT) -I. -I/usr/include -D_GNU_SOURCE=1 -D_THREAD_SAFE 
 
-LIBS	 =     -L$(SDL_ROOT)/lib -L$(IMGLIB_ROOT)
+LIBS	 =     -L$(SDL_ROOT)/lib -L$(IMGLIB_ROOT) -lSDLmain -Wl,-framework,AppKit -lSDL -Wl,-framework,Cocoa
 
-LFLAGS   =      $(LIBS) -limagelib -lGL -lGLU -lSDL_mixer -lSDL -lpthread -lm -ldl 
+LFLAGS   =      $(LIBS) -lGL -lGLU -limagelib -lSDL_mixer -lSDL -lSDLmain -lpthread -lm -ldl 
 
 #--------------------------------------------------------------------
 
