Addressed upstream: https://github.com/BIC-MNI/libminc/pull/147

--- CMakeLists.txt.orig	2026-04-24 06:00:26
+++ CMakeLists.txt	2026-05-18 11:42:45
@@ -631,10 +631,6 @@
   export(TARGETS ${LIBMINC_LIBRARY} FILE "${LIBMINC_EXPORTED_TARGETS}.cmake")
 
   # config for install dir
-  set(LIBMINC_USE_FILE_CONFIG     "\${LIBMINC_INSTALL_PREFIX}/lib/cmake/Use${LIBMINC_EXTERNAL_LIB_PREFIX}LIBMINC.cmake" )
-  set(LIBMINC_INCLUDE_DIRS_CONFIG "\${LIBMINC_INSTALL_PREFIX}/include" )
-  set(LIBMINC_LIBRARY_DIRS_CONFIG "\${LIBMINC_INSTALL_PREFIX}/lib" )
-  set(LIBMINC_STATIC_LIBRARIES_CONFIG   "" )
   set(VOLUME_IO_LIBRARY_STATIC "")
 
   configure_file(LIBMINCConfig.cmake.in
@@ -733,7 +729,7 @@
     install(
       EXPORT ${LIBMINC_EXPORTED_TARGETS}
       NAMESPACE LIBMINC::
-      DESTINATION ${LIBMINC_INSTALL_LIB_DIR}/cmake
+      DESTINATION ${LIBMINC_INSTALL_LIB_DIR}/cmake/LIBMINC
       COMPONENT Development
     )
 
@@ -753,7 +749,7 @@
       FILES
        ${_LIBMINC_CONFIG_INSTALL_FILES}
       DESTINATION
-       ${LIBMINC_INSTALL_LIB_DIR}/cmake
+       ${LIBMINC_INSTALL_LIB_DIR}/cmake/LIBMINC
       COMPONENT Development)
   endif()
 
--- LIBMINCConfig.cmake.in.orig	2026-04-24 06:00:26
+++ LIBMINCConfig.cmake.in	2026-05-18 11:42:27
@@ -21,7 +21,6 @@
 
 include(CMakeFindDependencyMacro)
 
-get_filename_component(LIBMINC_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)
 get_filename_component(LIBMINC_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 
 # ----- Feature flags (substituted at configure time) -----
