--- mythtv/libs/libmythtv/videosource.cpp.orig	2019-11-25 17:13:24.000000000 -0700
+++ mythtv/libs/libmythtv/videosource.cpp	2019-11-25 17:15:07.000000000 -0700
@@ -396,7 +396,7 @@
 void DataDirect_config::Load()
 {
     VerticalConfigurationGroup::Load();
-    bool is_sd_userid = userid->getValue().contains('@') > 0;
+    bool is_sd_userid = userid->getValue().contains('@') != 0;
     bool match = ((is_sd_userid  && (source == DD_SCHEDULES_DIRECT)) ||
                   (!is_sd_userid && (source == DD_ZAP2IT)));
     if (((userid->getValue() != lastloadeduserid) ||
