--- mythtv/libs/libmythtv/libmythtv.pro.orig	2018-02-01 05:20:06.000000000 -0700
+++ mythtv/libs/libmythtv/libmythtv.pro	2019-11-25 17:50:40.000000000 -0700
@@ -53,7 +53,7 @@
 
 macx {
     # Mac OS X Frameworks
-    FWKS = AGL ApplicationServices Carbon Cocoa CoreServices CoreFoundation OpenGL QuickTime IOKit
+    FWKS = AGL ApplicationServices Carbon Cocoa CoreServices CoreFoundation OpenGL IOKit
     using_quartz_video {
         FWKS += QuartzCore
     } else {
