--- mythtv/libs/libmythtv/tv_play.cpp.orig	2018-02-01 05:20:06.000000000 -0700
+++ mythtv/libs/libmythtv/tv_play.cpp	2019-11-25 17:27:29.000000000 -0700
@@ -196,12 +196,17 @@
     QWaitCondition m_wait;
 };
 
+static const MenuBase dummy_menubase;
+
 class MenuNodeTuple
 {
 public:
     MenuNodeTuple(const MenuBase &menu, const QDomNode &node) :
         m_menu(menu), m_node(node) {}
-    MenuNodeTuple(void) : m_menu(MenuBase()) {}
+    MenuNodeTuple(void) : m_menu(dummy_menubase)
+    {
+        assert("Should never be reached.");
+    }
     const MenuBase &m_menu;
     const QDomNode  m_node;
 };
