--- mythtv/programs/mythbackend/httpstatus.cpp.orig	2018-02-01 05:20:06.000000000 -0700
+++ mythtv/programs/mythbackend/httpstatus.cpp	2019-11-25 18:12:44.000000000 -0700
@@ -1478,8 +1478,8 @@
 
             // Only include HTML line break if display value doesn't already
             // contain breaks.
-            if ((display.contains("<p>", Qt::CaseInsensitive) > 0) ||
-                (display.contains("<br", Qt::CaseInsensitive) > 0))
+            if ((display.contains("<p>", Qt::CaseInsensitive) != 0) ||
+                (display.contains("<br", Qt::CaseInsensitive) != 0))
             {
                 // matches <BR> or <br /
                 linebreak = "\r\n";
