From d0193f0ec7c6817c5d6b6228e13d04528cfb53d0 Mon Sep 17 00:00:00 2001
From: Aaron Madlon-Kay <aaron@madlon-kay.com>
Date: Sat, 28 Oct 2023 21:37:14 +0900
Subject: [PATCH] Loosen build requirements

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 925778c..5a131ba 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools ~= 62.6", "setuptools-scm[toml] ~= 7.0", "wheel ~= 0.37.0"]
+requires = ["setuptools >= 62.6", "setuptools-scm[toml] >= 7.0", "wheel >= 0.37.0"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
-- 
2.42.0

