--- default.build.orig	2007-12-12 12:00:49.000000000 +0100
+++ default.build       2007-12-12 12:05:11.000000000 +0100
@@ -17,7 +17,7 @@
 	<property name="skip.ngen" value="True" />
 	<property name="skip.antlr" value="False" />
 	<property name="skip.ast" value="False" />
-	<property name="skip.vs2005" value="False" />
+	<property name="skip.vs2005" value="True" />
 	
 	<property name="build.dir" value="build" dynamic="True"/>
 	<property name="distrobuild.dir" value="distrobuild"/>
@@ -478,9 +478,9 @@
 		key files for mime detection, etc
 		-->
 
-		<property name="sharedmime.prefix" value="${pkg-config::get-variable('shared-mime-info','prefix')}" />
+		<property name="sharedmime.prefix" value="${install.prefix}" />
 		<property name="fakeroot.sharedmime" value="${install.destdir}/${sharedmime.prefix}" />
-		<property name="gsv.prefix" value="${pkg-config::get-variable('gtksourceview-1.0','prefix')}" />
+		<property name="gsv.prefix" value="/tmp" />
 		<property name="fakeroot.gsv" value="${install.destdir}/${gsv.prefix}" />
 
 		<mkdir dir="${fakeroot.boolib}"/>
