--- Makefile.orig	2019-07-22 12:17:19.000000000 +0200
+++ Makefile	2019-07-22 12:18:00.000000000 +0200
@@ -181,9 +181,10 @@
 	ln -s xca.html $(DMGSTAGE)/manual/index.html
-	otool -l $(DMGSTAGE)/xca.app/Contents/MacOS/xca | grep -e "chris\|Users" >&2
-	$(MACDEPLOYQT) $(DMGSTAGE)/xca.app
-	rpath="`otool -l $(DMGSTAGE)/xca.app/Contents/MacOS/xca | grep -e "chris\|Users"`" && \
-	if test -n "$$rpath"; then echo "  ERROR $$rpath"; false; fi
-	-codesign --force --deep --signature-size=96000 -s "Christian Hohnstaedt" $(DMGSTAGE)/xca.app --timestamp
+	#otool -l $(DMGSTAGE)/xca.app/Contents/MacOS/xca | grep -e "chris\|Users" >&2
+	$(MACDEPLOYQT) $(DMGSTAGE)/xca.app
+############# disable code signing by original author ###########################################################
+#	rpath="`otool -l $(DMGSTAGE)/xca.app/Contents/MacOS/xca | grep -e "chris\|Users"`" && \
+#	if test -n "$$rpath"; then echo "  ERROR $$rpath"; false; fi
+#	-codesign --force --deep --signature-size=96000 -s "Christian Hohnstaedt" $(DMGSTAGE)/xca.app --timestamp
 
 xca.dmg: $(MACTARGET).dmg
 
