--- Makefile.orig	2021-05-07 22:40:29.000000000 +0200
+++ Makefile	2021-06-10 13:53:45.000000000 +0200
@@ -124,7 +124,7 @@
 	test ! -z "$(TVERSION)"
 	git archive --format=tar --prefix=$(TARGET)/ $(TAG) > _$@
 	V=`tar xf _$@ -O $(TARGET)/VERSION` && \
-		test "$(TVERSION)" = "$$V" && echo "$$V" > VERSION
+		test "$(TVERSION)" = "$$V" && echo "$$V" > VERSION.txt
 	./bootstrap "$(TARGET)"
 	tar -rf _$@ "$(TARGET)/configure"
 	rm -rf "$(TARGET)"
