--- config.tests/unix/compile.test.old	2016-12-01 09:17:04.000000000 +0100
+++ config.tests/unix/compile.test	2017-05-14 14:05:41.000000000 +0200
@@ -31,6 +31,21 @@ while [ "$#" -gt 0 ]; do
         LFLAGS="$LFLAGS -framework \"$2\""
         shift
         ;;
+    -Wl,-framework,*)
+        LFLAGS="$LFLAGS -framework \"${PARAM#-Wl,-framework,}\""
+        ;;
+    -Wl,-framework)
+        # -Wl,-framework,... cought above
+        case $2 in
+        -Wl,*)
+            LFLAGS="$LFLAGS -framework \"${2#-Wl,}\""
+            shift
+            ;;
+        *)
+            exit 1
+            ;;
+        esac
+        ;;
     -arch)
         MAC_ARCH_CFLAGS="$MAC_ARCH_CFLAGS -arch $2"
         MAC_ARCH_CXXFLAGS="$MAC_ARCH_CXXFLAGS -arch $2"
