--- src/plugins/avfoundation/camera/avfcamerautility.mm.orig	2016-09-12 03:58:51.000000000 -0700
+++ src/plugins/avfoundation/camera/avfcamerautility.mm	2016-12-28 09:44:52.000000000 -0700
@@ -48,6 +48,13 @@
 #include <algorithm>
 #include <limits>
 
+// see https://trac.macports.org/ticket/52922
+// see https://trac.macports.org/ticket/53949
+// from the 10.9 SDK
+@interface NSArray (MacPortsCompatibilityFirstObject)
+- (id)firstObject NS_AVAILABLE(10_6, 4_0);
+@end
+
 QT_BEGIN_NAMESPACE
 
 AVFPSRange qt_connection_framerates(AVCaptureConnection *videoConnection)
