--- nethack-3.4.3/sys/unix/Makefile.top.orig	Mon Sep  2 17:44:55 2002
+++ nethack-3.4.3/sys/unix/Makefile.top	Mon Sep  2 17:46:44 2002
@@ -40,8 +40,8 @@
 SHELLDIR = $(PREFIX)/bin
 
 # per discussion in Install.X11 and Install.Qt
-VARDATND = 
-# VARDATND = x11tiles NetHack.ad pet_mark.xbm
+# VARDATND = 
+VARDATND = x11tiles NetHack.ad pet_mark.xbm
 # VARDATND = x11tiles NetHack.ad pet_mark.xbm rip.xpm
 # for Atari/Gem
 # VARDATND = nh16.img title.img GEM_RSC.RSC rip.img
