--- nethack-3.4.3/include/config.h.orig	Mon Sep  2 17:43:56 2002
+++ nethack-3.4.3/include/config.h	Mon Sep  2 17:44:14 2002
@@ -43,7 +43,7 @@
  * Some combinations make no sense.  See the installation document.
  */
 #define TTY_GRAPHICS	/* good old tty based graphics */
-/* #define X11_GRAPHICS */	/* X11 interface */
+#define X11_GRAPHICS	/* X11 interface */
 /* #define QT_GRAPHICS */	/* Qt interface */
 /* #define GNOME_GRAPHICS */	/* Gnome interface */
 /* #define MSWIN_GRAPHICS */	/* Windows NT, CE, Graphics */
