diff --git src/unix/fsevents.c src/unix/fsevents.c
index c430562..7ab55c1 100644
--- src/unix/fsevents.c
+++ src/unix/fsevents.c
@@ -40,6 +40,7 @@ void uv__fsevents_loop_delete(uv_loop_t* loop) {
 
 #else /* TARGET_OS_IPHONE */
 
+
 #include <dlfcn.h>
 #include <assert.h>
 #include <stdlib.h>
@@ -48,6 +49,22 @@ void uv__fsevents_loop_delete(uv_loop_t* loop) {
 #include <CoreFoundation/CFRunLoop.h>
 #include <CoreServices/CoreServices.h>
 
+#if ( MAC_OS_X_VERSION_MAX_ALLOWED < 1070 )
+  #define  kFSEventStreamEventFlagItemCreated       0x00000100
+  #define  kFSEventStreamEventFlagItemRemoved       0x00000200
+  #define  kFSEventStreamEventFlagItemInodeMetaMod  0x00000400
+  #define  kFSEventStreamEventFlagItemRenamed       0x00000800
+  #define  kFSEventStreamEventFlagItemModified      0x00001000
+  #define  kFSEventStreamEventFlagItemFinderInfoMod 0x00002000
+  #define  kFSEventStreamEventFlagItemChangeOwner   0x00004000
+  #define  kFSEventStreamEventFlagItemXattrMod      0x00008000
+  #define  kFSEventStreamEventFlagItemIsFile        0x00010000
+  #define  kFSEventStreamEventFlagItemIsDir         0x00020000
+  #define  kFSEventStreamEventFlagItemIsSymlink     0x00040000
+  #define  kFSEventStreamCreateFlagFileEvents       0x00000010
+#endif
+
+
 /* These are macros to avoid "initializer element is not constant" errors
  * with old versions of gcc.
  */
