--- src/htsweb.c.orig	2013-08-18 12:25:48.000000000 -0400
+++ src/htsweb.c	2026-08-09 00:21:58.463900406 -0400
@@ -186,7 +186,7 @@
             "usage: %s [--port <port>] [--bind <address>] [--ppid parent-pid] "
             "<path-to-html-root-dir> [key value [key value]..]\n",
             argv[0]);
-    fprintf(stderr, "example: %s /usr/share/httrack/\n", argv[0]);
+    fprintf(stderr, "example: %s @PREFIX@/bin/httrack/\n", argv[0]);
     return 1;
   }
 
