--- Makefile.orig	2010-11-25 14:08:08.000000000 +0100
+++ Makefile	2010-11-25 14:10:45.000000000 +0100
@@ -14,16 +14,17 @@
 fcgi: fastcgi
 
 src/fastcgi/lfcgi.so: src/fastcgi/lfcgi.o src/fastcgi/lfcgi.h
-	$(CC) $(CFLAGS) $(LIB_OPTION) -o src/fastcgi/lfcgi.so src/fastcgi/lfcgi.o -lfcgi 
+	$(CC) $(CFLAGS) $(LIB_OPTION) -o src/fastcgi/lfcgi.so src/fastcgi/lfcgi.o -lfcgi -llua
 
 install:
-	mkdir -p $(LUA_DIR)/wsapi
-	cp src/wsapi/*.lua $(LUA_DIR)/wsapi
-	cp src/launcher/wsapi.cgi $(BIN_DIR)/
-	cp src/launcher/wsapi.fcgi $(BIN_DIR)/
+	mkdir -p $(DESTDIR)/$(LUA_DIR)/wsapi
+	cp src/wsapi/*.lua $(DESTDIR)/$(LUA_DIR)/wsapi
+	cp src/launcher/wsapi.cgi $(DESTDIR)/$(BIN_DIR)/
+	cp src/launcher/wsapi.fcgi $(DESTDIR)/$(BIN_DIR)/
 
 install-fcgi:
-	cp src/fastcgi/lfcgi.so $(LUA_LIBDIR)/
+	mkdir -p $(DESTDIR)/$(LUA_LIBDIR)
+	cp src/fastcgi/lfcgi.so $(DESTDIR)/$(LUA_LIBDIR)/
 
 install-rocks: install
 	mkdir -p $(PREFIX)/samples
