--- plparse/plparser.symbols.orig	2013-11-25 14:26:38.000000000 -0800
+++ plparse/plparser.symbols	2013-11-25 14:27:22.000000000 -0800
@@ -4,7 +4,6 @@
 totem_cd_get_human_readable_name
 totem_cd_has_medium
 totem_cd_mrl_from_type
-totem_disc_media_type_get_type
 totem_disc_media_type_quark
 totem_pl_parser_add_ignored_mimetype
 totem_pl_parser_add_ignored_scheme
@@ -14,7 +13,7 @@
 totem_pl_parser_error_get_type
 totem_pl_parser_error_quark
 totem_pl_parser_get_type
-totemplparser_marshal_VOID__STRING_STRING_STRING
+_totemplparser_marshal_VOID__STRING_STRING_STRING
 totem_pl_parser_new
 totem_pl_parser_parse
 totem_pl_parser_parse_async
