--- libraries/libmbwrap/metiswrap.cc.orig	2017-01-12 08:44:25
+++ libraries/libmbwrap/metiswrap.cc	2026-04-05 15:29:52
@@ -36,6 +36,8 @@
 #undef ASSERT /* kill annoying redefiniton message */
 #include "mynewmem.h"
 
+typedef idx_t idxtype;
+
 int
 mbdyn_METIS_PartGraph(int iTotVertices,
 		int *pXadj,
@@ -112,7 +114,7 @@
 		wgtflag = 0;
 	}
 
-	METIS_PartGraphVKway(&iTotVertices,
+	METIS_PartGraphKway(&iTotVertices,
 			xadj,
 			adjncy,
 			vwgt,
@@ -120,6 +122,8 @@
 			&wgtflag,
 			&numflag,
 			&DataCommSize,
+			NULL,
+			NULL,
 			options,
 			&volume,
 			part);
