--- makefile.orig	2026-01-21 14:07:44
+++ makefile	2026-01-21 14:11:45
@@ -21,28 +21,13 @@
             -L$(SDL2_PATH)/lib/ \
             -L$(SDL2_TTF_PATH)/lib/
 else
-  CXX = g++
-  LDLIBS = -pthread
+  CXX ?= g++
 endif

-ifneq "$(wildcard /opt/homebrew)" ""
-  CXXFLAGS += -I/opt/homebrew/include/
-  LDLIBS += -L/opt/homebrew/lib/
-  BINDIR = /opt/homebrew/bin/
-else ifneq "$(wildcard /opt/local)" ""
-  CXXFLAGS += -I/opt/local/include/
-  LDLIBS += -L/opt/local/lib/
-  BINDIR = /opt/local/bin/
-else
-  CXXFLAGS += -I/usr/local/include/
-  LDLIBS += -L/usr/local/lib/
-  BINDIR = /usr/local/bin/
-endif
+CXXFLAGS = -I@@MP_PREFIX@@/libexec/ffmpeg8/include -I@@MP_PREFIX@@/include/
+LDLIBS = -L@@MP_PREFIX@@/libexec/ffmpeg8/lib -L@@MP_PREFIX@@/lib/ -pthread
+BINDIR = @@MP_PREFIX@@bin/

-ifneq "$(wildcard /usr/include/ffmpeg)" ""
-  CXXFLAGS += -I/usr/include/ffmpeg
-endif
-
 # Default: don't use pkg-config unless user explicitly enables it
 # Usage: make USE_PKG_CONFIG=1
 USE_PKG_CONFIG ?= 0
