--- Makefile.orig	2011-11-21 09:15:43.000000000 -0600
+++ Makefile	2012-01-19 21:43:19.000000000 -0600
@@ -18,7 +18,7 @@
 # NOTE for java compiles the environment variable $JAVA_HOME must be set
 # example: export JAVA_HOME=/usr/lib/jvm/java-6-sun-1.6.0.00
 ifndef JAVA_HOME
-    JAVA_HOME = /usr/lib/jvm/default-java
+    JAVA_HOME = /System/Library/Frameworks/JavaVM.framework/Versions/CurrentJDK
 endif
 
 # set prefix if not defined on the command line
@@ -37,9 +37,9 @@
 objdir = obj
 ipath=$(prefix)/bin
 CBASEFLAGS = -W -Wall -fno-rtti -fno-exceptions
-JAVAINCS   = -I$(JAVA_HOME)/include
+JAVAINCS   = -I$(JAVA_HOME)/Headers
 CXX = g++
-INSTALL=install -o $(USER) -g $(USER)
+INSTALL=install
 # INSTALL=install -o 0 -g 0
 
 # for testing on linux
@@ -186,19 +186,24 @@
 cleanobj:
 	rm -f $(objdir)/*.o
 
-install:
-	$(INSTALL) -m 755 -d $(ipath)
-	@$(INSTALL) -m 755 $(bindir)/astyle  $(ipath)
+install:  release
+	$(INSTALL) -m 755 -d $(DESTDIR)$(ipath)
+	$(INSTALL) -m 755 $(bindir)/astyle  $(DESTDIR)$(ipath)
 
-	$(INSTALL) -m 755 -d $(SYSCONF_PATH)
-	@mkdir -p $(SYSCONF_PATH)/html;
+	$(INSTALL) -m 755 -d $(DESTDIR)$(SYSCONF_PATH)
+	@mkdir -p $(DESTDIR)$(SYSCONF_PATH)/html;
 	@for files in ../../doc/*.html  ../../doc/*.css; \
 	do \
-		$(INSTALL)  -m 644  $$files  $(SYSCONF_PATH)/html; \
+		$(INSTALL)  -m 644  $$files  $(DESTDIR)$(SYSCONF_PATH)/html; \
 	done
-	@if [ -d $(SYSCONF_PATH_OLD) ];  then \
-		rm -rf $(SYSCONF_PATH_OLD); \
-	fi
+
+install-lib:  libastyle.a libastyle.dylib
+	$(INSTALL) -m 755 -d $(DESTDIR)$(prefix)/lib
+	$(INSTALL) $(bindir)/libastyle.a $(bindir)/libastyle.dylib $(DESTDIR)$(prefix)/lib
+
+install-jnilib:  libastylej.dylib
+	$(INSTALL) -m 755 -d $(DESTDIR)$(prefix)/lib
+	$(INSTALL) $(bindir)/libastylej.dylib $(DESTDIR)$(prefix)/lib
 
 uninstall:
 	rm -f $(ipath)/astyle
