--- Makefile.am.orig	2016-02-25 04:16:29.000000000 -0700
+++ Makefile.am	2018-06-18 03:37:19.000000000 -0700
@@ -1,5 +1,5 @@
 ACLOCAL_AMFLAGS = ${ACLOCAL_FLAGS} -I m4
-COLPACK_COMMON_FLAGS = $(EXTRA_FLAGS)  -O3
+COLPACK_COMMON_FLAGS = $(EXTRA_FLAGS)  
 
 AM_CPPFLAGS =  $(COLPACK_COMMON_FLAGS) \
 		-I$(top_srcdir)/Main \
@@ -76,7 +76,7 @@
 
 
 
-examplesdir = ${prefix}/examples
+examplesdir = ${prefix}/share/examples/ColPack
 basic_examplesdir = ${examplesdir}/Basic
 basic_examples_PROGRAMS = \
 			SampleDrivers/Basic/color_bipartite_graph_using_BipartiteGraphBicoloringInterface \
