--- dapi/src/nt/playaud.c.orig	2023-10-31 02:42:54.000000000 +0800
+++ dapi/src/nt/playaud.c	2024-11-04 11:02:24.000000000 +0800
@@ -5934,7 +5934,7 @@
     mmStatus = waveOutOpen( &(pPlayAudio->hWaveOut),
                             pPlayAudio->uiSelectedDeviceID,
                             (LPWAVEFORMATEX)pWaveFormat,
-                            (DWORD)WaveOutCallbackRoutine,
+                            WaveOutCallbackRoutine,
                             (DWORD)pPlayAudio,
                             dwOpenFlags ))
 #endif
@@ -5997,7 +5997,7 @@
       mmStatus = waveOutOpen( &(pPlayAudio->hWaveOut),
                               pPlayAudio->uiSelectedDeviceID,
                               (LPWAVEFORMATEX)pWaveFormat,
-                              (DWORD)WaveOutCallbackRoutine,
+                              WaveOutCallbackRoutine,
                               (DWORD)pPlayAudio,
                               dwOpenFlags ))
 #endif
