--- source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm.old	2016-12-03 16:23:57.000000000 -0800
+++ source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm	2016-12-03 16:19:08.000000000 -0800
@@ -496,7 +496,7 @@
 		{
 			[[NSAutoreleasePool alloc] init];
 			[NSApplication sharedApplication];
-			[NSApp setDelegate:(id<NSFileManagerDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
+			[NSApp setDelegate:(id<NSApplicationDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
 			[NSBundle loadNibNamed:@"MainMenu" owner:[NSApp delegate]];
 			[NSApp finishLaunching];
 		}
