https://trac.macports.org/ticket/67371

--- src/fpmath.h.roig	2022-01-19 16:43:52.000000000 -0700
+++ src/fpmath.h	2023-05-05 08:15:20.000000000 -0700
@@ -37,7 +37,7 @@
 #else 
 #include "i386_fpmath.h"
 #endif
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__ppc__)
 #include "powerpc_fpmath.h"
 #elif defined(__mips__)
 #include "mips_fpmath.h"
--- include/openlibm_fenv.h.orig	2022-01-19 16:43:52.000000000 -0700
+++ include/openlibm_fenv.h	2023-05-05 09:59:34.000000000 -0700
@@ -8,7 +8,7 @@
 #include <openlibm_fenv_amd64.h>
 #elif defined(__i386__)
 #include <openlibm_fenv_i387.h>
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__ppc__)
 #include <openlibm_fenv_powerpc.h>
 #elif defined(__mips__)
 #include <openlibm_fenv_mips.h>
