--- a/Changes	2012-04-20 15:25:23.000000000 -0700
+++ b/Changes	2012-04-20 15:25:36.000000000 -0700
@@ -1,3 +1,10 @@
+* 1.6
+    - blacklist: blacklist file is now human-readable/editable (thanks Armando Miraglia)
+    - fix Log Sucker bug with filesystems with gazillion files (thanks Johann H. Hauschild)
+    - fix syslog format detection on hosts with undefined hostname
+    - fix ipfw backend crashing when blacklist is enabled but empty on start-up (thanks Jin Choi)
+    - hosts backend: ensure safe permission for tcpwrappers file regardless of the local umask
+
 * 1.5       Feb 2011
     - logsucker: sshguard polls multiple log files at once
     - recognize syslog's "last message repeated N times" contextually and per-source
