--- a/src/fwalls/ipfw.c	2012-04-20 15:25:23.000000000 -0700
+++ b/src/fwalls/ipfw.c	2012-04-20 15:25:36.000000000 -0700
@@ -129,6 +129,8 @@
     assert(addresses != NULL);
     assert(service_codes != NULL);
 
+    if (addresses[0] == NULL) return FWALL_OK;
+
     ruleno = ipfwmod_getrulenumber();
     /* insert rules under this rule number (in chunks of max_addresses_per_rule) */
     if (ipfwmod_buildblockcommand(ruleno, addresses, addrkind, command, args) != FWALL_OK)
