--- a/src/sshguard_options.c	2012-04-20 15:25:23.000000000 -0700
+++ b/src/sshguard_options.c	2012-04-20 15:25:36.000000000 -0700
@@ -161,8 +161,8 @@
 static void usage(void) {
     fprintf(stderr, "Usage:\nsshguard [-b <thr:file>] [-w <whlst>]{0,n} [-a num] [-p sec] [-s sec]\n\t[-l <source>] [-f <srv:pidfile>]{0,n} [-i <pidfile>] [-v]\n");
     /* fprintf(stderr, "\t-d\tDebugging mode: don't fork to background, and dump activity to stderr.\n"); */
-    fprintf(stderr, "\t-b\tBlacklist: thr = number of abuses before blacklisting, file = blacklist filename.\n");
-    fprintf(stderr, "\t-a\tNumber of hits after which blocking an address (%d)\n", DEFAULT_ABUSE_THRESHOLD);
+    fprintf(stderr, "\t-b\tBlacklist: thr = blacklist from this total incurred danger, file = blacklist filename.\n");
+    fprintf(stderr, "\t-a\tIncurred danger score after which blocking an address (%d)\n", DEFAULT_ABUSE_THRESHOLD);
     fprintf(stderr, "\t-p\tSeconds after which unblocking a blocked address (%d)\n", DEFAULT_PARDON_THRESHOLD);
     fprintf(stderr, "\t-w\tWhitelisting of addr/host/block, or take from file if starts with \"/\" or \".\" (repeatable)\n");
     fprintf(stderr, "\t-s\tSeconds after which forgetting about a cracker candidate (%d)\n", DEFAULT_STALE_THRESHOLD);
