--- a/src/Makefile.in	2012-04-20 15:25:23.000000000 -0700
+++ b/src/Makefile.in	2012-04-20 15:25:36.000000000 -0700
@@ -35,7 +35,7 @@
 build_triplet = @build@
 
 # on solaris, the compiler refuses to use C99 for compiling pre-POSIX.1-2001 stuff (ew!)
-@SOLARIS_TRUE@am__append_1 = -D_XOPEN_SOURCE=600 
+@SOLARIS_TRUE@am__append_1 = -D_XOPEN_SOURCE=600
 @SOLARIS_FALSE@am__append_2 = -D_XOPEN_SOURCE
 @DEBUG_TRUE@am__append_3 = -g
 sbin_PROGRAMS = sshguard$(EXEEXT)
@@ -216,8 +216,9 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = parser fwalls
-AM_CFLAGS = -I. @OPTIMIZER_CFLAGS@ @WARNING_CFLAGS@ @STD99_CFLAGS@ \
-	$(am__append_1) $(am__append_2) $(am__append_3)
+AM_CFLAGS = -I. -DSIMCLIST_NO_DUMPRESTORE @OPTIMIZER_CFLAGS@ \
+	@WARNING_CFLAGS@ @STD99_CFLAGS@ $(am__append_1) \
+	$(am__append_2) $(am__append_3)
 sshguard_SOURCES = sshguard.c seekers.c sshguard_whitelist.c sshguard_log.c sshguard_procauth.c sshguard_blacklist.c sshguard_options.c sshguard_logsuck.c simclist.c hash_32a.c
 sshguard_LDADD = parser/libparser.a fwalls/libfwall.a
 all: config.h
