--- src/gui/CMakeLists.txt
+++ src/gui/CMakeLists.txt
@@ -258,7 +259,7 @@ function(generate_sized_png_from_svg icon_path size)
   set(icon_output_name "${size}-${icon_name_wle}.png")
   message(STATUS "Generate ${icon_output_name}")
   execute_process(COMMAND
-    "${INKSCAPE}" -w ${size} -h ${size} "${icon_path}" -o "${icon_output_name}"
+    "${INKSCAPE}" -w ${size} -h ${size} --export-png "${icon_output_name}" "${icon_path}" 
     WORKING_DIRECTORY "${icon_name_dir}"
     RESULT_VARIABLE
     INKSCAPE_SIDEBAR_ERROR
