--- src/crashreporter/CMakeLists.txt.old	2020-08-15 10:34:50.000000000 +0200
+++ src/crashreporter/CMakeLists.txt	2020-08-15 10:37:29.000000000 +0200
@@ -48,16 +48,4 @@
     #FIXME: find a nice solution to make the second if(BUILD_OWNCLOUD_OSX_BUNDLE) unnecessary
     # currently it needs to be done because the code right above needs to be executed no matter
     # if building a bundle or not and the install_qt4_executable needs to be called afterwards
-    if(BUILD_OWNCLOUD_OSX_BUNDLE)
-        get_target_property (QT_QMAKE_EXECUTABLE Qt5::qmake IMPORTED_LOCATION)
-        get_filename_component(QT_BIN_DIR "${QT_QMAKE_EXECUTABLE}" DIRECTORY)
-        find_program(MACDEPLOYQT_EXECUTABLE macdeployqt HINTS "${QT_BIN_DIR}")
-
-        add_custom_command(TARGET ${CRASHREPORTER_EXECUTABLE} POST_BUILD
-            COMMAND "${MACDEPLOYQT_EXECUTABLE}"
-                "$<TARGET_FILE_DIR:${CRASHREPORTER_EXECUTABLE}>/../.."
-                -always-overwrite
-            COMMENT "Running macdeployqt..."
-        )
-    endif()
 endif()

--- src/gui/CMakeLists.txt.old	2020-08-15 10:38:01.000000000 +0200
+++ src/gui/CMakeLists.txt	2020-08-15 10:39:56.000000000 +0200
@@ -387,21 +387,6 @@
 # if building a bundle or not and the install_qt4_executable needs to be called afterwards
 #
 # OSX: Run macdeployqt for src/gui and for src/cmd using the -executable option
-if(BUILD_OWNCLOUD_OSX_BUNDLE AND NOT BUILD_LIBRARIES_ONLY)
-    get_target_property (QT_QMAKE_EXECUTABLE Qt5::qmake IMPORTED_LOCATION)
-    get_filename_component(QT_BIN_DIR "${QT_QMAKE_EXECUTABLE}" DIRECTORY)
-    find_program(MACDEPLOYQT_EXECUTABLE macdeployqt HINTS "${QT_BIN_DIR}")
-
-    set(cmd_NAME ${APPLICATION_EXECUTABLE}cmd)
-
-    add_custom_command(TARGET ${APPLICATION_EXECUTABLE} POST_BUILD
-        COMMAND "${MACDEPLOYQT_EXECUTABLE}"
-            "$<TARGET_FILE_DIR:${APPLICATION_EXECUTABLE}>/../.."
-            -always-overwrite
-            -executable="$<TARGET_FILE_DIR:${APPLICATION_EXECUTABLE}>/${cmd_NAME}"
-        COMMENT "Running macdeployqt..."
-    )
-endif()
 
 if(NOT BUILD_OWNCLOUD_OSX_BUNDLE AND NOT WIN32)
     configure_file(${CMAKE_SOURCE_DIR}/mirall.desktop.in
