diff --git doc/nested/Makefile doc/nested/Makefile
index 4691168..cea3c76 100644
--- doc/nested/Makefile
+++ doc/nested/Makefile
@@ -2,7 +2,7 @@ HTML=html	# default to be replaced when makefile is called
 BK=../../src/bk
 
 ../www/nested.html: nested.doc ../www/template.html ../bin/pod2html.l
-	$(BK) tclsh ../bin/pod2html.l \
+	$(BK) little ../bin/pod2html.l \
 		--title="BitKeeper Nested Overview" \
 	    --template=../www/template.html -- nested.doc > ../www/nested.html
 
diff --git man/man2html/Makefile man/man2html/Makefile
index 454a0df..589cb10 100644
--- man/man2html/Makefile
+++ man/man2html/Makefile
@@ -5,7 +5,7 @@ all: html
 html: man2html.l BitKeeper_SN_Blue.png BitKeeper_SN_SVC_Blue.png
 	@test -d www || mkdir www
 	@rm -f www/*
-	$(BK) tclsh man2html.l
+	$(BK) little man2html.l
 	@cp *.png *.js www
 
 .PHONY: clean
