--- configure.orig	Sat Mar 27 11:34:18 2004
+++ configure		Wed Apr 21 23:39:46 2004
@@ -1404,7 +1404,7 @@
     # OSF1 and SCO ODT 3.0 have their own names for install.
     # Don't use installbsd from OSF since it installs stuff as root
     # by default.
-    for ac_prog in ginstall scoinst install; do
+    for ac_prog in scoinst install; do
       for ac_exec_ext in '' $ac_executable_extensions; do
         if $as_executable_p "$as_dir/$ac_prog$ac_exec_ext"; then
           if test $ac_prog = install &&
