--- export.c.orig	2004-04-02 19:40:40.000000000 +1000
+++ export.c	2011-02-07 22:53:47.000000000 +1100
@@ -123,8 +123,8 @@ WritePNG (const char *sz, unsigned char 
   atext[1].compression = PNG_TEXT_COMPRESSION_NONE;
 
 #ifdef PNG_iTXt_SUPPORTED
-  text_ptr[0].lang = NULL;
-  text_ptr[1].lang = NULL;
+  atext[0].lang = NULL;
+  atext[1].lang = NULL;
 #endif
   png_set_text (ppng, pinfo, atext, 2);
 
