--- src/metaio.c
+++ src/metaio.c
@@ -545,7 +545,7 @@ int character_entity(const char* const s)
 static
 int get_char(MetaioFile const f)
 {
-    const int c = gzgetc(f->fp);
+    const int c = gzgetc((gzFile)f->fp);
 
     if (c == '\n')
     {
@@ -794,7 +794,7 @@ int fscanf_ilwd_char_u(MetaioFile f, struct MetaioStringU *b,
 
     /*-- Get first character which is not a space --*/
     do
-        val = gzgetc(f->fp);
+        val = gzgetc((gzFile)f->fp);
     while(val == ' ');
 
     while ((count == 1) && (val <= 256) && ! strchr(terminators, val) && (val != EOF))
@@ -840,7 +840,7 @@ int fscanf_ilwd_char_u(MetaioFile f, struct MetaioStringU *b,
 
         /*-- Get next character, skipping any spaces. --*/
         do
-            val = gzgetc(f->fp);
+            val = gzgetc((gzFile)f->fp);
         while(val == ' ');
     }
     
