--- CMakeLists.txt.orig	2025-12-03 10:37:41
+++ CMakeLists.txt	2025-12-03 10:38:09
@@ -173,11 +173,12 @@
 	SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/cmake/test-charconv.cpp)
 
 if(NOT STD_CHARCONV_COMPILING)
-	message(NOTICE "libcifpp: Using fast_float for std::from_chars")
-	FetchContent_Declare(fastfloat
-		GIT_REPOSITORY "https://github.com/fastfloat/fast_float"
-		GIT_TAG v8.0.2)
-	FetchContent_MakeAvailable(fastfloat)
+    message(NOTICE "libcifpp: Using fast_float for std::from_chars")
+    find_package(FASTFLOAT)
+	##FetchContent_Declare(fastfloat
+	##	GIT_REPOSITORY "https://github.com/fastfloat/fast_float"
+	##	GIT_TAG v8.0.2)
+	##FetchContent_MakeAvailable(fastfloat)
 endif()
 
 find_package(Threads)
