--- a/site.cfg	2011-12-12 12:05:40.000000000 -0800
+++ b/site.cfg	2011-12-12 12:14:01.000000000 -0800
@@ -10,7 +10,7 @@
 # The path to mysql_config.
 # Only use this if mysql_config is not on your PATH, or you have some weird
 # setup that requires it.
-#mysql_config = /usr/local/bin/mysql_config
+mysql_config = @MYSQL_CONFIG@
 
 # The Windows registry key for MySQL.
 # This has to be set for Windows builds to work.
