--- src/base/PdfInputDevice.h.orig
+++ src/base/PdfInputDevice.h
@@ -22,6 +22,7 @@
 #define _PDF_INPUT_DEVICE_H_
 
 #include <istream>
+#include <ios>
 
 #include "PdfDefines.h"
 #include "PdfLocale.h"
--- src/base/PdfLocale.h.orig
+++ src/base/PdfLocale.h
@@ -1,7 +1,7 @@
 #ifndef PODOFO_PDFLOCALE_H
 #define PODOFO_PDFLOCALE_H
 
-namespace std { class ios_base; };
+#include <ios>
 
 namespace PoDoFo {
 
