--- configure_orig	2010-08-04 20:17:53.000000000 -0400
+++ configure	2010-08-04 20:22:54.000000000 -0400
@@ -537,18 +537,7 @@
 // based on crypto.prf. any changes made to that file need to be tracked here.
 static QString internal_crypto_prf(const QString &incdir, const QString &libdir)
 {
-	QString out = QString(
-	"CONFIG *= qt\n"
-	"INCLUDEPATH += %1/QtCrypto\n"
-	"LIBS += -L%2\n"
-	"\n"
-	"LINKAGE = -lqca\n"
-	"CONFIG(debug, debug|release) {\n"
-        "	windows:LINKAGE = -lqcad\n"
-        "	mac:LINKAGE = -lqca_debug\n"
-	"}\n"
-	"LIBS += \$\$LINKAGE\n"
-	).arg(incdir, libdir);
+	QString out = QString("CONFIG *= crypto\n");
 	return out;
 }
 
