--- src/luarocks/cfg.lua.orig	2012-02-15 14:48:44.000000000 +0100
+++ src/luarocks/cfg.lua	2012-02-15 14:53:14.000000000 +0100
@@ -308,10 +308,10 @@
    defaults.lib_extension = "so"
    defaults.external_lib_extension = "so"
    defaults.obj_extension = "o"
-   defaults.external_deps_dirs = { "/usr/local", "/usr" }
-   defaults.variables.LUA_BINDIR = site_config.LUA_BINDIR or "/usr/local/bin"
-   defaults.variables.LUA_INCDIR = site_config.LUA_INCDIR or "/usr/local/include"
-   defaults.variables.LUA_LIBDIR = site_config.LUA_LIBDIR or "/usr/local/lib"
+   defaults.external_deps_dirs = { "%PREFIX%", "/usr/local", "/usr" }
+   defaults.variables.LUA_BINDIR = site_config.LUA_BINDIR or "%PREFIX%/bin"
+   defaults.variables.LUA_INCDIR = site_config.LUA_INCDIR or "%PREFIX%/include"
+   defaults.variables.LUA_LIBDIR = site_config.LUA_LIBDIR or "%PREFIX%/lib"
    defaults.variables.CFLAGS = "-O2"
    defaults.cmake_generator = "Unix Makefiles"
    defaults.platforms = { "unix" }
@@ -352,8 +352,8 @@
    defaults.external_lib_extension = "dylib"
    defaults.arch = "macosx-"..proc
    defaults.platforms = {"unix", "bsd", "macosx"}
-   defaults.variables.CC = "export MACOSX_DEPLOYMENT_TARGET=10.3; gcc"
-   defaults.variables.LD = "export MACOSX_DEPLOYMENT_TARGET=10.3; gcc"
+   defaults.variables.CC = "export MACOSX_DEPLOYMENT_TARGET=10.5; %COMPILER%"
+   defaults.variables.LD = "export MACOSX_DEPLOYMENT_TARGET=10.5; %COMPILER%"
    defaults.variables.LIBFLAG = "-bundle -undefined dynamic_lookup -all_load"
 end
 
