--- src/modules/rtaudio/RtAudio.cpp.orig	2020-12-31 11:47:28.000000000 -0800
+++ src/modules/rtaudio/RtAudio.cpp	2020-12-31 11:48:08.000000000 -0800
@@ -46,6 +46,7 @@
 #include <cstring>
 #include <climits>
 #include <algorithm>
+#include <unistd.h>
 
 // Static variable definitions.
 const unsigned int RtApi::MAX_SAMPLE_RATES = 14;
@@ -10227,4 +10228,3 @@
   // End:
   //
   // vim: et sts=2 sw=2
-
