diff --git a/src/modules/qimage/configure b/src/modules/qimage/configure
index 0e9b906..0466423 100755
--- a/src/modules/qimage/configure
+++ b/src/modules/qimage/configure
@@ -85,7 +85,7 @@ else
 	if [ -d "$qimage_libdir" -a -d "$qimage_includedir" ]
 	then
 		# test if we have a Qt3 or Qt4
-		if [ -f "$qimage_libdir/libQtCore.so" ] || [ -d "$qimage_libdir/QtGui.framework" ] || [ -f "$qimage_libdir/libQtCore4.a" ] && [ "$force_qt3" = "" ]
+		if [ -f "$qimage_libdir/libQtCore.dylib" ] || [ -d "$qimage_libdir/QtGui.framework" ] || [ -f "$qimage_libdir/libQtCore4.a" ] && [ "$force_qt3" = "" ]
 		then
 			echo "Qt version 4.x detected, will compile Qt4 qimage producer"
 			qt4_found=true
