diff --git src/modules/qt/graph.cpp src/modules/qt/graph.cpp
index 6a97011..54f7582 100644
--- src/modules/qt/graph.cpp
+++ src/modules/qt/graph.cpp
@@ -19,6 +19,7 @@
 
 #include "graph.h"
 #include <QVector>
+#include <QtGui/QPainterPath>
 #include <math.h>
 
 /*
diff --git src/modules/qt/kdenlivetitle_wrapper.cpp src/modules/qt/kdenlivetitle_wrapper.cpp
index af65519..bd06277 100755
--- src/modules/qt/kdenlivetitle_wrapper.cpp
+++ src/modules/qt/kdenlivetitle_wrapper.cpp
@@ -45,6 +45,7 @@
 #include <QGraphicsEffect>
 #include <QGraphicsBlurEffect>
 #include <QGraphicsDropShadowEffect>
+#include <QtGui/QPainterPath>
 #endif
 
 Q_DECLARE_METATYPE(QTextCursor);
diff --git src/modules/qt/producer_qtext.cpp src/modules/qt/producer_qtext.cpp
index f3b0915..a7339fd 100644
--- src/modules/qt/producer_qtext.cpp
+++ src/modules/qt/producer_qtext.cpp
@@ -30,6 +30,7 @@
 #include <QString>
 #include <QTextCodec>
 #include <QTextDecoder>
+#include <QtGui/QPainterPath>
 
 static void close_qimg( void* qimg )
 {
