diff --git configure configure
index 46cdb6063fc4d308e25c4491949b099980d68d44..47bf2392931a4a0cc64b39a13d9f537b085903c3 100755
--- configure
+++ configure
@@ -94,9 +94,9 @@ build_config()
 
 		case $targetos in
 		Darwin)
-                echo "CFLAGS+=-fPIC `sdl-config --cflags`"
+                echo "CFLAGS+=-fPIC"
 		echo "SHFLAGS=-dynamiclib"
-		echo "LDFLAGS+=`sdl-config --libs`"
+		echo "LDFLAGS+=-framework AppKit -framework Cocoa"
 		;;
 		Linux|GNU/kFreeBSD|GNU)
 		[ "$optimisations" = "true" ] &&
