--- lsdvd.c.orig	2007-11-27 22:57:02.000000000 -0500
+++ lsdvd.c	2007-11-27 22:57:20.000000000 -0500
@@ -13,13 +13,13 @@
  *  2003-04-19  Cleanups get_title_name, added dvdtime2msec, added helper macros,
  *			  output info structures in form of a Perl module, by Henk Vergonet.
  */
+#include <stdint.h>
 #include <dvdread/ifo_read.h>
 #include <string.h>
 #include <sys/stat.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <stdint.h>
 #include "lsdvd.h"
 #include "ocode.h"
 
