--- makefile	2023-07-09 14:26:16.000000000 +0800
+++ makefile	2023-07-09 14:42:03.000000000 +0800
@@ -2,7 +2,7 @@
 
 # Compiler command
 ifeq (${MPICXX},)
-   MPICXX=mpicxx
+   MPICXX=@MPICXX@
 endif
 CFLAGS=-O3 -std=c++0x -w -fopenmp
 
@@ -20,10 +20,10 @@
 OBJS:=$(addprefix $(ODIR)/,$(notdir $(SRCS:.cpp=.o)))
 
 # Include and library file
-INC=-I$(HDF5_HOME)/include
-LIBDIR=-L$(HDF5_HOME)/lib
+INC=-I@PREFIX@/include
+LIBDIR=-L@PREFIX@/lib
 ifeq (${LIB},)
-   LIB=-lhdf5 -llapack
+   LIB=-lhdf5 @LAPACK@
 endif
 
 ifneq (${LAPACK_DIR},)
