--- configure.orig	2001-12-28 08:55:46.000000000 +1100
+++ configure	2008-08-06 17:33:48.000000000 +1000
@@ -3038,6 +3038,7 @@
 	# crt1.o) gets erroneously defined as common, which breaks dynamic
 	# loading of any modules which reference it in System.framework
 	next/4*|next/5*) LINKFORSHARED="-u __dummy -framework System" ;;
+	Darwin/9*) LINKFORSHARED="-framework System -framework Foundation" ;;
 	Darwin/*) LINKFORSHARED="-u __dummy -framework System -framework Foundation" ;;
 	UnixWare*) LINKFORSHARED="-dy -Bdynamic -Wl,-Bexport";;
 	SCO_SV*) LINKFORSHARED="-Bdynamic -dy -Wl,-Bexport";;
--- configure.in.orig	2001-12-28 08:55:46.000000000 +1100
+++ configure.in	2008-08-06 17:33:27.000000000 +1000
@@ -692,6 +692,7 @@
 	# crt1.o) gets erroneously defined as common, which breaks dynamic
 	# loading of any modules which reference it in System.framework
 	next/4*|next/5*) LINKFORSHARED="-u __dummy -framework System" ;;
+	Darwin/9*) LINKFORSHARED="-framework System -framework Foundation" ;;
 	Darwin/*) LINKFORSHARED="-u __dummy -framework System -framework Foundation" ;;
 	UnixWare*) LINKFORSHARED="-dy -Bdynamic -Wl,-Bexport";;
 	SCO_SV*) LINKFORSHARED="-Bdynamic -dy -Wl,-Bexport";;
