--- src/tools/install.neko.orig	2011-09-10 03:52:00.000000000 -0500
+++ src/tools/install.neko	2012-07-22 09:21:11.000000000 -0500
@@ -39,11 +39,6 @@
 // LIBS DATAS
 
 libs = {
-	mod_neko => {
-		src => $array("../../vm/stats","mod_neko","cgi"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x"
-	},
 	mod_neko2 => {
 		src => $array("../../vm/stats","mod_neko","cgi"),
 		inc => $array("httpd.h","apr.h"),
@@ -52,13 +47,6 @@
 		realdir => "mod_neko",
 		apache2 => true,
 	},
-	mysql => {
-		src => $array("mysql"),
-		inc => "mysql.h",
-		incname => "MySQL 4.+"
-		lib => "libmysqlclient_r.a",
-		lparams => "-lz -lssl"
-	},
 	mysql5 => {
 		src => $array("../common/sha1","../common/socket","my_proto/my_proto","my_proto/my_api","mysql"),
 		inc => $array(),
@@ -90,12 +78,6 @@
 		cflags => switch system { "Mac" => "" default => "`pkg-config --cflags gtk+-2.0`" },
 		lparams => switch system { "Mac" => "-framework Carbon" default => "`pkg-config --libs gtk+-2.0` -lgthread-2.0" },
 	},
-	mod_tora => {
-		src => $array("../common/socket","protocol","mod_tora"),
-		inc => "httpd.h",
-		incname => "Apache 1.3.x",
-		cflags => "-I../common",
-	},
 	mod_tora2 => {
 		src => $array("../common/socket","protocol","mod_tora"),
 		inc => $array("httpd.h","apr.h"),
@@ -140,38 +122,20 @@
 
 search_includes = function(isap2) {
 	var inc1 = $array(
+		"@PREFIX@/include",
+		"@PREFIX@/include/gtk-2.0",
+		"@PREFIX@/include/@MYSQL@/mysql",
 		"/usr/include",
-		"/usr/local/include/mysql",
-		"/usr/include/mysql",
-		"/usr/local/include",
-		"/usr/include/gtk-2.0",
-		"/opt/local/include",
-		"/opt/local/include/mysql",
-		"/Developer/Headers/FlatCarbon",
+		"/System/Library/Frameworks/Carbon.framework/Headers",
 	);
 	var inc2;
 	if( isap2 )
 		inc2 = $array(
-			"/usr/local/apache2/include",
-			"/usr/include/apache2",
-			"/opt/local/apache2/include",
-			"/usr/include/apr-1",
-			"/usr/include/apr-1.0",
-			"/usr/include/apr-0",
-			"/usr/local/include/apr-1",
-			"/usr/local/include/apr-1.0",
-			"/usr/local/include/apr-0",
-			"/opt/local/include/apr-1",
-			"/opt/local/include/apr-1.0",
-			"/opt/local/include/apr-0"
+			"@PREFIX@/apache2/include",
+			"@PREFIX@/include/apr-1",
 		);
 	else
 		inc2 = $array(
-			"/usr/apache/include",
-			"/usr/include/apache-1.3",
-			"/usr/include/httpd",
-			"/opt/local/include/apache",
-			"/usr/local/apache/include"
 		);
 	return $aconcat($array(inc1,inc2));
 }
@@ -180,12 +144,9 @@
 cwd = $ssub(cwd,0,$ssize(cwd)-1);
 libraries = $array(
 	cwd,
+	"@PREFIX@/lib",
+	"@PREFIX@/lib/@MYSQL@/mysql",
 	"/usr/lib",
-	"/usr/lib/mysql",
-	"/usr/local/lib",
-	"/usr/local/lib/mysql",
-	"/opt/local/lib",
-	"/opt/local/lib/mysql",
 );
 
 exec = function(cmd) {
