--- psi4/src/psi4/libqt/CMakeLists.txt.orig	2026-09-17 13:33:13
+++ psi4/src/psi4/libqt/CMakeLists.txt	2026-09-17 13:44:38
@@ -26,15 +26,28 @@
 #   and that's tough with multiple BLAS/LAPACK backends. So, we'll see if this is robust enough.
 include(CMakePushCheckState)
 include(CheckFunctionExists)
+include(CheckSymbolExists)
 cmake_push_check_state()
 list(APPEND CMAKE_REQUIRED_LIBRARIES tgt::lapack)
-check_function_exists(daxpby _has_daxpby)
+
+if(APPLE)
+    check_symbol_exists (daxpby_ "Accelerate/Accelerate.h" _has_daxpby)
+else()
+    check_function_exists(daxpby _has_daxpby)
+endif()
+
 if(NOT _has_daxpby)
   message(WARNING "${Yellow}Your BLAS/LAPACK library does not seem to be providing the DAXPBY subroutine. DAXPBY calls will be re-routed through DSCAL and DAXPY. If you are seeing this message, and you are not building with Apple Accelerate, you may want to re-check that the correct BLAS/LAPACK libraries are being used by the build system. Redo CMake configuration for changes to register.${ColourReset}")
 endif()
 
-check_function_exists(dggsvd3 _has_dggsvd3)
-check_function_exists(dggsvp3 _has_dggsvp3)
+if(APPLE)
+    check_symbol_exists (dggsvd3_ "Accelerate/Accelerate.h" _has_dggsvd3)
+    check_symbol_exists (dggsvp3_ "Accelerate/Accelerate.h" _has_dggsvp3)
+else()
+    check_function_exists(dggsvd3 _has_dggsvd3)
+    check_function_exists(dggsvp3 _has_dggsvp3)
+endif()
+
 if(NOT (_has_dggsvd3 AND _has_dggsvp3))
   message(WARNING "${Yellow}Your BLAS/LAPACK library does not seem to be providing the DGGSVD3 and DGGSVP3 subroutines. No re-routing is available. If you are seeing this message, and you are not building with Apple Accelerate, you may want to re-check that the correct BLAS/LAPACK libraries are being used by the build system. Redo CMake configuration for changes to register.${ColourReset}")
 endif()
