--- a/cppconn/CMakeLists.txt	2012-08-16 12:15:30.000000000 -0700
+++ b/cppconn/CMakeLists.txt	2012-08-23 07:14:51.000000000 -0700
@@ -58,7 +58,6 @@
 
 SET(MYSQLCPPCONN_INSTALL_HEADERS
 		build_config.h
-		config.h
 		connection.h
 		datatype.h
 		driver.h
@@ -72,4 +71,5 @@
 		sqlstring.h
 		warning.h)
 
-INSTALL(FILES ${MYSQLCPPCONN_INSTALL_HEADERS} DESTINATION include/cppconn)
+INSTALL(FILES ${MYSQLCPPCONN_INSTALL_HEADERS} DESTINATION include@NAME@/cppconn)
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/config.h DESTINATION include@NAME@/cppconn)
