--- src/Makefile_orig.in	2025-10-04 22:20:30.000000000 +0200
+++ src/Makefile.in	2025-10-04 22:22:29.000000000 +0200
@@ -784,7 +784,7 @@
 	$(INSTALL_DATA) figures $(DESTDIR)/$(datadir)/ltris2/figures
 	$(mkinstalldirs) $(DESTDIR)/$(hiscoredir)
 	if ! test -f $(DESTDIR)/$(hiscoredir)/ltris2.hscr; then \
-	  $(INSTALL_DATA) --mode=666 empty.hscr $(DESTDIR)/$(hiscoredir)/ltris2.hscr; \
+	  $(INSTALL_DATA) empty.hscr $(DESTDIR)/$(hiscoredir)/ltris2.hscr; \
 	fi;
 
 uninstall-local: