error: use of undeclared identifier 'CLOCK_UPTIME_RAW'

--- mozglue/misc/AwakeTimeStamp.cpp.orig
+++ mozglue/misc/AwakeTimeStamp.cpp
@@ -58,6 +58,9 @@
 #  include <sys/time.h>
 #  include <sys/types.h>
 #  include <mach/mach_time.h>
+#endif
+
+#if defined(__APPLE__) && defined(__MACH__) && defined(CLOCK_UPTIME_RAW)
 
 AwakeTimeStamp AwakeTimeStamp::NowLoRes() {
   return AwakeTimeStamp(clock_gettime_nsec_np(CLOCK_UPTIME_RAW) / kNSperUS);
--- mozglue/misc/Uptime.cpp.orig
+++ mozglue/misc/Uptime.cpp
@@ -32,7 +32,9 @@
 #  include <sys/time.h>
 #  include <sys/types.h>
 #  include <mach/mach_time.h>
+#endif
 
+#if defined(__APPLE__) && defined(__MACH__) && defined(CLOCK_UPTIME_RAW) && defined(CLOCK_MONOTONIC_RAW)
 const uint64_t kNSperMS = 1000000;
 
 Maybe<uint64_t> NowExcludingSuspendMs() {
