--- src/Makefile.in.orig	2018-06-06 08:48:52.000000000 -0500
+++ src/Makefile.in	2018-06-13 14:14:53.000000000 -0500
@@ -90,7 +90,7 @@
 prefix	= @prefix@
 mandir	= @mandir@
 install:
-	mkdir -p /etc/fsvs /var/spool/fsvs $(bindir) /etc/fsvs/svn/auth/svn.{simple,ssl.server,ssl.client-passphrase}
+	mkdir -p $(DESTDIR)$(prefix)/etc/fsvs $(DESTDIR)$(prefix)/var/spool/fsvs $(DESTDIR)$(bindir) $(DESTDIR)$(prefix)/etc/fsvs/svn/auth/svn.{simple,ssl.server,ssl.client-passphrase}
 	install -m 0755 $(DEST) $(DESTDIR)/$(bindir)
 #	install -m 0644 ../doc/fsvs.1 $(DESTDIR)/(mandir)
 
@@ -148,13 +148,13 @@
 ################################ Rules ######################################
 %.o:	%.c
 	@echo "     CC $<"
-	@$(CC) $(CFLAGS) -c -o $@ $<
+	$(CC) $(CFLAGS) -c -o $@ $<
 
 # if the Makefile has changed, the output will (at least sometimes)
 # change, too.
 $(DEST): $(C_FILES:%.c=%.o)
 	@echo "     Link $@"
-	@$(CC) $(FSVS_LDFLAGS) $(LDLIBS) $(LIBS) $(EXTRALIBS) -o $@ $^
+	$(CC) $(FSVS_LDFLAGS) $(LDLIBS) $(LIBS) $(EXTRALIBS) -o $@ $^
 ifeq (@ENABLE_RELEASE@, 1)
 	-strip $@
 endif
@@ -171,7 +171,7 @@
 deps:	$(D_FILES)
 .%.d:	%.c
 	@echo "     deps for $<"
-	@$(CC) $(CFLAGS) -MM $< | perl -pe 's#\bdoc.g-c\b##' > $@
+	$(CC) $(CFLAGS) -MM $< | perl -pe 's#\bdoc.g-c\b##' > $@
 include $(D_FILES)
 
 tools/fsvs-chrooter: tools/fsvs-chrooter.c
