--- configure.orig	2015-07-25 09:35:10.000000000 -0500
+++ configure	2015-07-26 12:30:44.000000000 -0500
@@ -4251,7 +4251,7 @@
 
 
 ##################################### Header files
-INCDIRS="/usr/local/include /usr/include /openpkg/include "
+INCDIRS=""
 # The subversion headers do a #include <apr_pools.h>, so the APR libraries
 # *have* to be directly specified.
 # Furthermore there's apr-1/ as directory name, depending on apr version.
@@ -4333,7 +4333,7 @@
 $as_echo "$as_me: \"CFLAGS=$CFLAGS\"" >&6;}
 
 ##################################### Linker
-LIBDIRS="/usr/local/lib /openpkg/lib"
+LIBDIRS=""
 
 
 # Check whether --with-aprlib was given.
