diff -u libhdhomerun.orig/Makefile libhdhomerun/Makefile
--- libhdhomerun/Makefile.orig	2014-10-12 10:18:16.000000000 -0400
+++ libhdhomerun/Makefile	2014-10-12 10:22:19.000000000 -0400
@@ -32,7 +32,6 @@
     LDFLAGS += -lsocket
   endif
   ifeq ($(OS),Darwin)
-    CFLAGS += -arch i386 -arch x86_64
     LIBEXT := .dylib
     SHARED := -dynamiclib -install_name @PREFIX@/lib/libhdhomerun$(LIBEXT)
   endif
