--- configure.py.orig
+++ configure.py
@@ -788,7 +788,7 @@

         g = optparse.OptionGroup(p, title="MacOS X Configuration")
         g.add_option("--arch", action="append", default=[], dest="arch",
-                choices=["i386", "x86_64", "ppc"],
+                choices=["i386", "x86_64", "ppc", "ppc64", "arm64"],
                 help="build for architecture ARCH")
         g.add_option("--deployment-target", action="store", default='',
                 metavar="VERSION", dest="deployment_target",
