--- main.c.org	2004-12-09 02:48:49.000000000 +0100
+++ main.c	2011-02-12 21:50:31.000000000 +0100
@@ -151,7 +151,7 @@
     }
     /* this looks different from the others because 77E 
        parses as scientific notation */
-    if (1 == sscanf(argv[i], "%lf%", &temp, &hemisphere)
+    if (2 == sscanf(argv[i], "%lf%1[Ee]", &temp, &hemisphere)
 	&& (argv[i][strlen(argv[i])-1] == 'E' ||
 	    argv[i][strlen(argv[i])-1] == 'e')) {
       lon = temp;
