--- gdk/x11/gdkapplaunchcontext-x11.c.old       2016-12-30 15:55:56.000000000 +0100
+++ gdk/x11/gdkapplaunchcontext-x11.c   2017-04-03 08:42:14.000000000 +0200
@@ -27,7 +27,11 @@
 #include "gdkprivate-x11.h"
 
 #include <glib.h>
+#if defined(__APPLE__) && GLIB_CHECK_VERSION(2,52,0)
+#include <gio/gosxappinfo.h>
+#else
 #include <gio/gdesktopappinfo.h>
+#endif
 
 #include <string.h>
 #include <unistd.h>
