--- Makefile.orig
+++ Makefile
@@ -16,6 +16,6 @@ BYTEOBJS=mpi.cmo
 NATOBJS=$(BYTEOBJS:.cmo=.cmx)

-all: byte
+all: byte opt

 byte: libcamlmpi.a mpi.cma

@@ -22,7 +22,7 @@ opt: libcamlmpi.a mpi.cmxa

 install:
-	ocamlfind install mpi META mpi.mli mpi.cmi mpi.cmti \
+	ocamlfind install $(OCAMLFIND_INSTFLAGS) mpi META mpi.mli mpi.cmi mpi.cmti \
           $(wildcard mpi*.cmx) $(wildcard mpi.cm*a) \
           $(wildcard *mpi.a) $(wildcard *mpi.so)

