--- libemos-dp/CMakeLists.txt.orig	2015-10-27 23:05:24.000000000 +0900
+++ libemos-dp/CMakeLists.txt	2016-01-30 12:49:59.000000000 +0900
@@ -41,7 +41,7 @@
 #  install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink
 #               libemosR64.so ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/libemos.R64.D64.I32.so)" )
 #else()
-  install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink
-               libemosR64.a ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/libemos.R64.D64.I32.a)" )
+#  install(CODE "execute_process(COMMAND \"${CMAKE_COMMAND}\" -E create_symlink
+#               libemosR64.a ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}/libemos.R64.D64.I32.a)" )
 #endif()
 
