--- configure.orig	2010-08-19 15:45:58.000000000 -0700
+++ configure	2010-08-19 15:46:24.000000000 -0700
@@ -12940,7 +12940,7 @@
    { (exit 1); exit 1; }; }
 fi
 PYTHON=
-for ac_prog in python2.5 python2.4 python2.3 python
+for ac_prog in python2.6 python2.5 python2.4 python2.3 python
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
