--- src/Makefile.orig	2010-01-27 21:40:01.000000000 -0700
+++ src/Makefile	2018-11-10 14:12:55.000000000 -0700
@@ -51,13 +51,17 @@
 default:
 	@echo "Specify a target configuration"
 
-targets: $(LIBDIR)/$(ANNLIB)
+targets: $(LIBDIR)/$(ANNLIB) $(LIBDIR)/$(ANNLIB:.a=.dylib)
 
 $(LIBDIR)/$(ANNLIB): $(OBJECTS)
 	$(MAKELIB) $(ANNLIB) $(OBJECTS)
 	$(RANLIB) $(ANNLIB)
 	mv $(ANNLIB) $(LIBDIR)
 
+$(LIBDIR)/$(ANNLIB:.a=.dylib): $(OBJECTS)
+	$(C++) $(CFLAGS) -dynamiclib -o $(ANNLIB:.a=.dylib) $(OBJECTS) -install_name __MACPORTS_PREFIX__/lib/$(ANNLIB:.a=.dylib)
+	mv $(ANNLIB:.a=.dylib) $(LIBDIR)
+
 #-----------------------------------------------------------------------------
 # Make object files
 #-----------------------------------------------------------------------------
