--- src/meson.build
+++ #<buffer meson.build>
@@ -89,7 +89,7 @@
 endif
 
 gvnc_link_args = []
-if host_machine.system() != 'windows'
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
   gvnc_link_args += ['-Wl,--no-undefined']
 endif
 
@@ -178,7 +178,7 @@
   ]
 
   gvncpulse_link_args = []
-  if host_machine.system() != 'windows'
+  if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
     gvncpulse_link_args += ['-Wl,--no-undefined']
   endif
 
@@ -337,7 +337,7 @@
 
 
 gtk_vnc_link_args = []
-if host_machine.system() != 'windows'
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
   gtk_vnc_link_args += ['-Wl,--no-undefined']
 endif
 
