--- gdl/gdl-dock-object.c.orig	2012-07-15 21:51:41.000000000 -0500
+++ gdl/gdl-dock-object.c	2012-07-15 21:59:15.000000000 -0500
@@ -892,7 +892,7 @@
     g_relation_insert (dock_register, "placeholder", (gpointer) GDL_TYPE_DOCK_PLACEHOLDER);
 }
 
-G_CONST_RETURN gchar *
+const gchar *
 gdl_dock_object_nick_from_type (GType type)
 {
     GTuples *tuples;
--- gdl/gdl-dock-object.h.orig	2012-07-15 21:51:52.000000000 -0500
+++ gdl/gdl-dock-object.h	2012-07-15 21:59:31.000000000 -0500
@@ -195,7 +195,7 @@
 GType gdl_dock_param_get_type (void);
 
 /* functions for setting/retrieving nick names for serializing GdlDockObject types */
-G_CONST_RETURN gchar *gdl_dock_object_nick_from_type    (GType        type);
+const gchar *gdl_dock_object_nick_from_type    (GType        type);
 GType                 gdl_dock_object_type_from_nick    (const gchar *nick);
 GType                 gdl_dock_object_set_type_for_nick (const gchar *nick,
                                                          GType        type);
