--- gamt.c.old	2016-02-25 02:24:15.000000000 +0100
+++ gamt.c	2016-02-25 02:32:12.000000000 +0100
@@ -32,6 +32,7 @@
 #include <gdk/gdkx.h>
 #include <gtk/gtk.h>
 #include <vte/vte.h>
+#include <pango/pango-font.h>
 
 #include "parseconfig.h"
 #include "redir.h"
@@ -271,7 +272,8 @@ static void menu_cb_config_font(GtkActio
             /* fall through */
         case GTK_RESPONSE_APPLY:
             fontname = gtk_font_chooser_get_font(GTK_FONT_CHOOSER(dialog));
-            vte_terminal_set_font_from_string(VTE_TERMINAL(gamt->vte), fontname);
+            PangoFontDescription *font_desc = pango_font_description_from_string(fontname);
+            vte_terminal_set_font(VTE_TERMINAL(gamt->vte), font_desc);
             cfg_set_str(CFG_FONT, fontname);
             break;
         default:
@@ -309,7 +311,7 @@ static void menu_cb_config_fg(GtkAction 
     gdk_rgba_parse(&color, cfg_get_str(CFG_FOREGROUND));
     if (0 != pickcolor("Text color", &color))
 	    return;
-    vte_terminal_set_color_foreground_rgba(VTE_TERMINAL(gamt->vte), &color);
+    vte_terminal_set_color_foreground(VTE_TERMINAL(gamt->vte), &color);
     snprintf(name, sizeof(name), "#%04x%04x%04x",
 	     (int)(color.red * 65535),
              (int)(color.green * 65535),
@@ -326,7 +328,7 @@ static void menu_cb_config_bg(GtkAction 
     gdk_rgba_parse(&color, cfg_get_str(CFG_BACKGROUND));
     if (0 != pickcolor("Background color", &color))
 	return;
-    vte_terminal_set_color_background_rgba(VTE_TERMINAL(gamt->vte), &color);
+    vte_terminal_set_color_background(VTE_TERMINAL(gamt->vte), &color);
     snprintf(name, sizeof(name), "#%04x%04x%04x",
              (int)(color.red * 65535),
              (int)(color.green * 65535),
@@ -338,7 +340,7 @@ static void menu_cb_blink_cursor(GtkTogg
 {
     struct gamt_window *gamt = user_data;
     gboolean state = gtk_toggle_action_get_active(action);
-    VteTerminalCursorBlinkMode blink;
+    VteCursorBlinkMode blink;
 
     if (amt_debug)
 	fprintf(stderr, "%s: %s\n", __FUNCTION__, state ? "on" : "off");
@@ -781,7 +783,7 @@ static int gamt_connect(struct gamt_wind
 static struct gamt_window *gamt_window()
 {
     GtkWidget *vbox, *hbox, *frame, *item;
-    VteTerminalCursorBlinkMode blink;
+    VteCursorBlinkMode blink;
     GdkRGBA color;
     GError *err;
     gboolean state;
@@ -823,7 +825,8 @@ static struct gamt_window *gamt_window()
     g_signal_connect(gamt->vte, "commit", G_CALLBACK(user_input), gamt);
     vte_terminal_set_scrollback_lines(VTE_TERMINAL(gamt->vte), 4096);
     str = cfg_get_str(CFG_FONT);
-    vte_terminal_set_font_from_string(VTE_TERMINAL(gamt->vte), str);
+    PangoFontDescription *font_desc = pango_font_description_from_string(str);
+    vte_terminal_set_font(VTE_TERMINAL(gamt->vte), font_desc);
 
     /* FIXME: make configurable */
     vte_terminal_set_backspace_binding(VTE_TERMINAL(gamt->vte),
@@ -872,12 +875,12 @@ static struct gamt_window *gamt_window()
     str = cfg_get_str(CFG_FOREGROUND);
     if (str) {
 	gdk_rgba_parse(&color, str);
-	vte_terminal_set_color_foreground_rgba(VTE_TERMINAL(gamt->vte), &color);
+	vte_terminal_set_color_foreground(VTE_TERMINAL(gamt->vte), &color);
     }
     str = cfg_get_str(CFG_BACKGROUND);
     if (str) {
 	gdk_rgba_parse(&color, str);
-	vte_terminal_set_color_background_rgba(VTE_TERMINAL(gamt->vte), &color);
+	vte_terminal_set_color_background(VTE_TERMINAL(gamt->vte), &color);
     }
 
     return gamt;
