--- GNUmakefile.old	2017-02-27 11:24:26.000000000 +0100
+++ GNUmakefile	2019-04-23 08:27:37.000000000 +0200
@@ -26,18 +26,6 @@ endef
 
 #################################################################
 
-# build gamt?
-ifeq ($(HAVE_GTK)$(HAVE_VTE),yesyes)
-  TARGETS += gamt
-  gamt : CFLAGS += -Wno-strict-prototypes
-  gamt : pkglst += gtk+-3.0 gdk-3.0 vte-2.91
-endif
-
-CFLAGS += $(shell test "$(pkglst)" != "" && pkg-config --cflags $(pkglst))
-LDLIBS += $(shell test "$(pkglst)" != "" && pkg-config --libs   $(pkglst))
-
-#################################################################
-
 build: $(TARGETS)
 
 install: build
