--- GNUmakefile.old	2016-02-25 00:39:56.000000000 +0100
+++ GNUmakefile	2016-02-25 00:41:08.000000000 +0100
@@ -19,7 +19,7 @@ include mk/Autoconf.mk
 
 define make-config
 LIB		:= $(LIB)
-HAVE_GTK	:= $(call ac_pkg_config,gtk+-2.0)
+HAVE_GTK	:= $(call ac_pkg_config,gtk+-3.0)
 HAVE_VTE	:= $(call ac_pkg_config,vte)
 endef
 
@@ -29,7 +29,7 @@ endef
 ifeq ($(HAVE_GTK)$(HAVE_VTE),yesyes)
   TARGETS += gamt
   gamt : CFLAGS += -Wno-strict-prototypes
-  gamt : pkglst += gtk+-2.0 vte
+  gamt : pkglst += gtk+-3.0 vte
 endif
 
 CFLAGS += $(shell test "$(pkglst)" != "" && pkg-config --cflags $(pkglst))
