--- BLAS/CMakeLists.txt.orig	2010-12-16 03:44:42.000000000 +0900
+++ BLAS/CMakeLists.txt	2016-03-13 17:51:33.000000000 +0900
@@ -2,7 +2,7 @@
 if(BUILD_TESTING)
 add_subdirectory(TESTING)
 endif(BUILD_TESTING)
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/blas.pc.in ${CMAKE_CURRENT_BINARY_DIR}/blas.pc)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/blas.pc.in ${CMAKE_CURRENT_BINARY_DIR}/blas.pc @ONLY)
 install(FILES
   ${CMAKE_CURRENT_BINARY_DIR}/blas.pc
   DESTINATION ${PKG_CONFIG_DIR}
