--- main.c.orig.2	2021-11-28 20:33:30.000000000 -0500
+++ main.c	2021-11-28 20:37:53.000000000 -0500
@@ -74,6 +74,7 @@
 #include <string.h>
 #include <assert.h>
 #include <math.h>
+#include <stdint.h>
 
 #ifndef USE_FD
 #include <unistd.h>
@@ -103,36 +104,22 @@
 	#define unlikely(x) (x)
 #endif
 
-// user defined types
-//typedef unsigned int		uint;
-typedef unsigned char		uint8;
-typedef unsigned short		uint16;
-typedef unsigned int		uint32;
-#ifdef WIN
-typedef __int64				uint64;
-#else
-typedef unsigned long long	uint64;
-#endif
-
-typedef char				int8;
-typedef short				int16;
-typedef int					int32;
-#ifdef WIN
-typedef __int64				int64;
-#else
-typedef long long			int64;
-#endif
-
-
-typedef signed int			sint;
-typedef signed char			sint8;
-typedef signed short		sint16;
-typedef signed int			sint32;
-#ifdef WIN
-typedef __int64				sint64;
-#else
-typedef signed long long	sint64;
-#endif
+typedef unsigned int uint;
+typedef uint8_t uint8;
+typedef uint16_t uint16;
+typedef uint32_t uint32;
+typedef uint64_t uint64;
+
+typedef int8_t				int8;
+typedef int16_t				int16;
+typedef int32_t				int32;
+typedef int64_t			    int64;
+
+typedef signed int		sint;
+typedef int8_t			sint8;
+typedef int16_t    		sint16;
+typedef int32_t			sint32;
+typedef int64_t			sint64;
 
 #define BITS_IN_BUF (8)
 
