diff --git src/init.lisp src/init.lisp
index 9dd202a..1997022 100644
--- src/init.lisp
+++ src/init.lisp
@@ -231,7 +231,7 @@ OpenSSL shared libraries have not changed.
 If they have changed, you may get errors, as users report:
 https://github.com/cl-plus-ssl/cl-plus-ssl/issues/167
 "
-  (detect-custom-openssl-installations-if-macos)
+  ;; (detect-custom-openssl-installations-if-macos)
   (unless (member :cl+ssl-foreign-libs-already-loaded
                   *features*)
     (cffi:use-foreign-library libcrypto)
diff --git src/reload.lisp src/reload.lisp
index 05ab883..cff26a3 100644
--- src/reload.lisp
+++ src/reload.lisp
@@ -48,7 +48,7 @@
   (when (member :darwin *features*)
     (detect-macos-custom-openssl-installations)))
 
-(detect-custom-openssl-installations-if-macos)
+;; (detect-custom-openssl-installations-if-macos)
 
 #|
 
@@ -123,11 +123,11 @@ sudo rm /usr/local/lib/libcrypto.dylib /usr/local/lib/libssl.dylib
                   ;; load them via dlopen will work. This is because macOS ships
                   ;; all system-provided libraries as a single dyld_shared_cache
                   ;; bundle.
-                  "/usr/lib/libcrypto.46.dylib"
-                  "/usr/lib/libcrypto.44.dylib"
-                  "/usr/lib/libcrypto.42.dylib"
-                  "/usr/lib/libcrypto.41.dylib"
-                  "/usr/lib/libcrypto.35.dylib"
+                  ;; "/usr/lib/libcrypto.46.dylib"
+                  ;; "/usr/lib/libcrypto.44.dylib"
+                  ;; "/usr/lib/libcrypto.42.dylib"
+                  ;; "/usr/lib/libcrypto.41.dylib"
+                  ;; "/usr/lib/libcrypto.35.dylib"
 
                   ;; The default old system libcrypto, versionless file name,
                   ;; which may have insufficient crypto and can cause process
@@ -138,8 +138,8 @@ sudo rm /usr/local/lib/libcrypto.dylib /usr/local/lib/libssl.dylib
                   ;; delete the unversioned paths. Keeping them for a while for
                   ;; compatibility. See
                   ;; https://github.com/cl-plus-ssl/cl-plus-ssl/pull/115
-                  "libcrypto.dylib"
-                  "/usr/lib/libcrypto.dylib"))
+                  ;; "libcrypto.dylib"
+                  "@@MACPORTS_OPENSSL_LIB@@/libcrypto.dylib"))
     ((and :unix (not :cygwin)) (:or "libcrypto.so.1.1"
                                     "libcrypto.so.1.0.0"
                                     "libcrypto.so.3"
@@ -162,16 +162,16 @@ sudo rm /usr/local/lib/libcrypto.dylib /usr/local/lib/libssl.dylib
     ((:and :darwin :cl+ssl-personalized-install-found) "/usr/local/lib/libssl.dylib")
     (:darwin (:or ;; System-provided libraries, with version in the file name.
                   ;; See the comment for the libcryto equivalents above.
-                  "/usr/lib/libssl.48.dylib"
-                  "/usr/lib/libssl.46.dylib"
-                  "/usr/lib/libssl.44.dylib"
-                  "/usr/lib/libssl.43.dylib"
-                  "/usr/lib/libssl.35.dylib"
+                  ;; "/usr/lib/libssl.48.dylib"
+                  ;; "/usr/lib/libssl.46.dylib"
+                  ;; "/usr/lib/libssl.44.dylib"
+                  ;; "/usr/lib/libssl.43.dylib"
+                  ;; "/usr/lib/libssl.35.dylib"
 
                   ;; Default system libssl, versionless file name.
                   ;; See the coment for the corresponding libcrypto.
-                  "libssl.dylib"
-                  "/usr/lib/libssl.dylib"))
+                  ;; "libssl.dylib"
+                  "@@MACPORTS_OPENSSL_LIB@@/libssl.dylib"))
     (:solaris (:or "/lib/64/libssl.so"
                    "libssl.so.0.9.8" "libssl.so" "libssl.so.4"))
     ;; Unlike some other systems, OpenBSD linker,
