--- configure.in.orig	2012-11-18 09:42:02.000000000 -0600
+++ configure.in	2012-11-18 10:08:03.000000000 -0600
@@ -700,8 +700,8 @@
 
 if test "x$with_mpich" != "xno"; then
   if test "x$with_mpich" != "xyes"; then
-    LIBS="$LIBS -L$with_mpich/lib -lmpich"
-    INCLUDES="$INCLUDES -I$with_mpich/include"
+    LIBS="$LIBS -L$with_mpich/lib -lpmpich -lmpich -lopa -lmpl"
+    INCLUDES="$INCLUDES -I$with_mpich/include/mpich"
   else
     INCLUDES="$INCLUDES -I/usr/include"
     LIBS="$LIBS -lmpich"
