--- frontend/readline.c.orig
+++ frontend/readline.c
@@ -8,9 +8,20 @@
 #include <signal.h>
 #include <unistd.h>
 #if defined(HAVE_READLINE)
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#endif
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wstrict-prototypes"
 #include <readline/history.h>
 #include <readline/readline.h>
+#pragma clang diagnostic pop
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
+#pragma GCC diagnostic pop
+#endif
 #elif defined(HAVE_EDITLINE)
+#define HAVE_WCSDUP 1
 #include <editline/readline.h>
 #include <histedit.h>
 #endif
