--- qtiplot/src/core/ApplicationWindow.cpp.orig	2011-11-01 14:21:14.000000000 +0900
+++ qtiplot/src/core/ApplicationWindow.cpp	2012-04-19 21:03:12.000000000 +0900
@@ -174,7 +174,7 @@
 #include <QVarLengthArray>
 #include <QList>
 #include <QUrl>
-#include <QAssistantClient>
+//#include <QAssistantClient>
 #include <QFontComboBox>
 #include <QSpinBox>
 #include <QMdiArea>
@@ -349,7 +349,7 @@
 	insertTranslatedStrings();
 	disableToolbars();
 
-	assistant = new QAssistantClient( QString(), this );
+	//assistant = new QAssistantClient( QString(), this );
 
 	connect(tablesDepend, SIGNAL(activated(int)), this, SLOT(showTable(int)));
 
@@ -11334,10 +11334,10 @@
 	}
 
 	QStringList cmdLst = QStringList() << "-profile" << profilePath;
-	QAssistantClient *assist = new QAssistantClient( QString(), 0);
-	assist->setArguments( cmdLst );
-	assist->showPage(helpPath);
-	connect(assist, SIGNAL(assistantClosed()), qApp, SLOT(quit()) );
+	//QAssistantClient *assist = new QAssistantClient( QString(), 0);
+	//assist->setArguments( cmdLst );
+	//assist->showPage(helpPath);
+	//connect(assist, SIGNAL(assistantClosed()), qApp, SLOT(quit()) );
 }
 
 void ApplicationWindow::showHelp()
@@ -11368,8 +11368,8 @@
 	}
 
 	QStringList cmdLst = QStringList() << "-profile" << profilePath;
-	assistant->setArguments( cmdLst );
-	assistant->showPage(helpFilePath);
+	//assistant->setArguments( cmdLst );
+	//assistant->showPage(helpFilePath);
 }
 
 void ApplicationWindow::showPlotWizard()
@@ -16228,7 +16228,7 @@
 	if (hostInfo.error() != QHostInfo::NoError){
 		QApplication::restoreOverrideCursor();
 		QMessageBox::critical(this, tr("QtiPlot - Error"), qtiplotWeb + ": " + hostInfo.errorString());
-		exit(0);
+//		exit(0);
 	}
 
 	QApplication::restoreOverrideCursor();
