--- qtiplot/src/origin/importOPJ.cpp.orig	2010-10-12 08:53:22.000000000 +0200
+++ qtiplot/src/origin/importOPJ.cpp	2010-10-24 01:50:36.000000000 +0200
@@ -790,7 +790,7 @@
 					case Origin::GraphCurve::TextPlot:
 						style = Origin::GraphCurve::TextPlot;
 						break;
-					case Origin::GraphCurve::XYZContour:
+/*					case Origin::GraphCurve::XYZContour:*/
 					case Origin::GraphCurve::Contour:
 						style = Origin::GraphCurve::Contour;
 						break;
