--- qtiplot/src/scripting/scripting.pri.orig	2011-07-01 17:56:54.000000000 +0200
+++ qtiplot/src/scripting/scripting.pri	2011-07-22 17:30:27.000000000 +0200
@@ -115,8 +115,7 @@
 			 $${SIP_DIR}/sipqtiStatisticTest.cpp \
 			 $${SIP_DIR}/sipqtitTest.cpp \
 			 $${SIP_DIR}/sipqtiShapiroWilkTest.cpp \
-			 $${SIP_DIR}/sipqtiChiSquareTest.cpp \
-			 $${SIP_DIR}/sipqtiAnova.cpp \
+			 $${SIP_DIR}/sipqtiChiSquareTest.cpp
 
 exists(../../$${SIP_DIR}/sipqtiQList.cpp) {
   # SIP < 4.9
