--- ./app/Makefile	2022-12-23 04:11:10.000000000 -0500
+++ ./app/Makefile	2022-12-23 04:11:44.000000000 -0500
@@ -258,12 +258,12 @@
 	${MAKE} -C ../src CONFIGFILE=${CONFIGFILEPATH} install DEBUG=${DEBUG}
 
 ${ZSV}: ${CLI}
-	@mkdir -p `dirname "$@"`
+	@mkdir -p `dirname ${DESTDIR}/"$@"`
 ifneq ($(findstring emcc,$(CC)),) # emcc
-	cp -p $< `dirname "$@"`/
-	cp -p ${CLI_ADDITIONAL} `dirname "$@"`/
+	cp -p $< `dirname ${DESTDIR}/"$@"`/
+	cp -p ${CLI_ADDITIONAL} `dirname ${DESTDIR}/"$@"`/
 else
-	cp -p $< $@
+	cp -p $< ${DESTDIR}/$@
 endif
 
 cli: build-cli
