diff -Naur welle.io-2.0~beta3/src/welle-gui/QML/components/MessagePopup.qml welle.io-2.0~beta3-kde/src/welle-gui/QML/components/MessagePopup.qml
--- welle.io-2.0~beta3/src/welle-gui/QML/components/MessagePopup.qml	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0~beta3-kde/src/welle-gui/QML/components/MessagePopup.qml	2019-09-27 10:00:18.111673873 +0200
@@ -1,5 +1,6 @@
 import QtQuick 2.5
 import QtQuick.Controls 2.3
+import QtGraphicalEffects 1.0
 
 import "../texts"
 
@@ -37,18 +38,20 @@
             }
         }
 
-        Button {
+        Image {
+            id:exitImage
             anchors.left: parent.left
             anchors.verticalCenter: parent.verticalCenter
-            icon.name: "exit"
-            icon.width: width
-            icon.height: width
-            icon.color: "white"
-            padding: 0
-            background: null
+            source: "qrc:/icon/welle_io_icons/20x20@2/exit.png"
             height: Units.dp(30)
             width: height
         }
+        
+        ColorOverlay {
+            anchors.fill: exitImage
+            source: exitImage
+            color: "white"
+        }
 
         MouseArea {
             anchors.fill: parent
diff -Naur welle.io-2.0~beta3/src/welle-gui/QML/components/ViewBaseFrame.qml welle.io-2.0~beta3-kde/src/welle-gui/QML/components/ViewBaseFrame.qml
--- welle.io-2.0~beta3/src/welle-gui/QML/components/ViewBaseFrame.qml	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0~beta3-kde/src/welle-gui/QML/components/ViewBaseFrame.qml	2019-09-27 10:00:18.111673873 +0200
@@ -41,9 +41,11 @@
 
         ToolBar {
             Layout.fillWidth: true
+            Layout.margins: Units.dp(1)
             Layout.topMargin: Units.dp(5)
             visible: isExpert
-
+            implicitHeight: Units.dp(20)
+            
             MouseArea {
                 id: mouseArea
                 anchors.fill: parent
@@ -71,28 +73,32 @@
                 }
             }
 
+            //Uncomment if we want the header of the View to be black text over white background
+/*
             background: Rectangle {
                 implicitHeight: menuButton.height
-                opacity: 0
+                color: "white"
             }
-
+*/
             Label {
                 id: label
                 anchors.horizontalCenter: parent.horizontalCenter
                 anchors.verticalCenter: parent.verticalCenter
                 font.pixelSize: TextStyle.textStandartSize
+                //color: TextStyle.textColor //Uncomment if we want the header of the View to be black text over white background
             }
 
-            ToolButton {
+            Button {
                 id: menuButton
                 anchors.right: parent.right
+                anchors.verticalCenter: parent.verticalCenter
+                implicitWidth: icon.width + Units.dp(20)
+
                 icon.name: "menu"
                 icon.height: Units.dp(10)
                 icon.width: Units.dp(10)
-                background: Rectangle {
-                    color: menuButton.pressed ? "lightgrey" : "white"
-                    opacity: menuButton.pressed ? 100 : 0
-                }
+                //icon.color: "black"  //Uncomment if we want the header of the View to be black text over white background
+                flat:true
 
                 onClicked: optionsMenu.open()
 
diff -Naur welle.io-2.0~beta3/src/welle-gui/QML/expertviews/TextOutputView.qml welle.io-2.0~beta3-kde/src/welle-gui/QML/expertviews/TextOutputView.qml
--- welle.io-2.0~beta3/src/welle-gui/QML/expertviews/TextOutputView.qml	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0~beta3-kde/src/welle-gui/QML/expertviews/TextOutputView.qml	2019-09-27 10:00:18.111673873 +0200
@@ -11,7 +11,6 @@
     labelText: qsTr("Console Output")
 
     content:  Rectangle {
-        color: "black"
         anchors.fill: parent
         Flickable {
             id: flick
@@ -22,6 +21,7 @@
                 id: textField
                 font.family: "Monospace"
                 font.pixelSize: Units.em(0.9)
+                background: Rectangle { color: "black" }
                 color: "white"
                 width: flick.width
                 wrapMode: TextEdit.Wrap
diff -Naur welle.io-2.0~beta3/src/welle-gui/QML/MainView.qml welle.io-2.0~beta3-kde/src/welle-gui/QML/MainView.qml
--- welle.io-2.0~beta3/src/welle-gui/QML/MainView.qml	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0~beta3-kde/src/welle-gui/QML/MainView.qml	2019-09-27 10:00:18.111673873 +0200
@@ -154,6 +154,8 @@
 
             ToolButton {
                 icon.name: "menu"
+                icon.width: Units.dp(20)
+                icon.height: Units.dp(20)
                 onClicked: optionsMenu.open()
 
                 Menu {
@@ -220,8 +222,8 @@
                 WComboBox {
                     id: stationListBox
                     Layout.preferredWidth: Units.dp(200)
-                    background: Rectangle { color: "white" }
-
+                    Layout.preferredHeight: Units.dp(25)
+                    
                     model:  [qsTr("All stations"), qsTr("Favorites")]
 
                     onCurrentIndexChanged: {
@@ -241,10 +243,7 @@
                     icon.name: "menu"
                     icon.height: Units.dp(15)
                     icon.width: Units.dp(15)
-                    background: Rectangle {
-                        color: menuButton.pressed ? "lightgrey" : "white"
-                        opacity: menuButton.pressed ? 100 : 0
-                    }
+                    flat:true
                     onClicked: stationMenu.open()
                     implicitWidth: contentItem.implicitWidth + Units.dp(15)
 
diff -Naur welle.io-2.0~beta3/src/welle-gui/QML/RadioView.qml welle.io-2.0~beta3-kde/src/welle-gui/QML/RadioView.qml
--- welle.io-2.0~beta3/src/welle-gui/QML/RadioView.qml	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0~beta3-kde/src/welle-gui/QML/RadioView.qml	2019-09-27 10:01:47.739671215 +0200
@@ -18,29 +18,48 @@
         text: radioController.ensemble.trim()
     }
 
-    // Use a button to display a icon
-    Button {
-        anchors.top: parent.top
+    // Use 2 Images to switch between speaker & speaker_mute icon (instead of toggle button). 
+    // Permits use of color with org.kde.desktop style
+    Image {
+        id: speakerIcon
+        anchors.verticalCenter: signalStrength.verticalCenter
         anchors.right: parent.right
-        anchors.topMargin: Units.dp(-20) // ToDo Hack!
-        anchors.rightMargin: Units.dp(-10) // ToDo Hack!
+        width: Units.dp(30)
+        height: Units.dp(30)
+        visible: true
+        source: "qrc:/icon/welle_io_icons/20x20@2/speaker.png"
+
+        MouseArea {
+            anchors.fill: parent
+            onClicked: {radioController.setVolume(0); speakerIconMutedRed.visible = true; speakerIcon.visible = false}
+        }
+    }
 
-        background: Rectangle { opacity: 0} // Hack to disable the pressed color
-        checkable: true
-        flat: true
-
-        icon.name: checked ? "speaker_mute" : "speaker"
-        icon.width: Units.dp(30)
-        icon.height: Units.dp(30)
-        icon.color: checked ? "red" : "transparent"
+    Image {
+        id: speakerIconMuted
+        anchors.verticalCenter: signalStrength.verticalCenter
+        anchors.right: parent.right
+        width: Units.dp(30)
+        height: Units.dp(30)
+        visible: false
 
-        implicitWidth: contentItem.implicitWidth + Units.dp(30)
-        implicitHeight: implicitWidth
+        source: "qrc:/icon/welle_io_icons/20x20@2/speaker_mute.png"
+    }
 
-        onCheckedChanged: checked ? radioController.setVolume(0) : radioController.setVolume(100)
+    ColorOverlay {
+        id: speakerIconMutedRed
+        visible: false
+        anchors.fill: speakerIconMuted
+        source: speakerIconMuted
+        color: "red"
+        MouseArea {
+            anchors.fill: parent
+            onClicked: {radioController.setVolume(100); speakerIconMutedRed.visible = false; speakerIcon.visible = true}
+        }
     }
 
     RowLayout{
+        id: signalStrength
         anchors.top: parent.top
         anchors.left: parent.left
         anchors.leftMargin: Units.dp(5)
