diff -Naur welle.io-2.0-beta3/src/welle-gui/main.cpp welle.io-2.0-beta3-MainTr/src/welle-gui/main.cpp
--- welle.io-2.0-beta3/src/welle-gui/main.cpp	2019-09-08 20:37:15.000000000 +0200
+++ welle.io-2.0-beta3-MainTr/src/welle-gui/main.cpp	2019-09-19 21:33:19.272993768 +0200
@@ -71,6 +71,18 @@
     // Create new QT application
     QApplication app(argc, argv);
 
+    //Initialise translation
+    QTranslator *translator = new QTranslator;
+    QString Language = QLocale::system().name();
+    qDebug() << "main:" <<  "Set language" << Language;
+    bool isTranslation = translator->load(QString(":/i18n/") + Language);
+    if(!isTranslation && Language != "en_GB")
+    {
+        qDebug() << "main:" <<  "Error while loading language" << Language << "use English \"en_GB\" instead";
+        Language = "en_GB";
+    }
+    QCoreApplication::installTranslator(translator);
+    
     // Register waterfall diagram
     qmlRegisterType<WaterfallItem>("io.welle", 1, 0, "Waterfall");
 
