--- components/editor/Makefile.am.orig	2014-04-30 00:26:54.000000000 -0700
+++ components/editor/Makefile.am	2014-04-30 00:27:18.000000000 -0700
@@ -132,7 +132,7 @@
 	gtkhtml-color-swatch.c			\
 	gtkhtml-combo-box.c
 
-libglade_gtkhtml_editor_la_LDFLAGS = -module -avoid-version -no-undefined
+libglade_gtkhtml_editor_la_LDFLAGS = -module -avoid-version
 
 libglade_gtkhtml_editor_la_LIBADD = @GTKHTML_CFLAGS@
 
