--- config/autotools/make/python.mk
+++ config/autotools/make/python.mk
@@ -28,12 +28,15 @@ clean-local-python:
 %_python.stamp: $$($$*_SWIG) Makefile $$($$*_SWIG_INCLUDES) $$($$*_SWIG_AUXDEPS)
 	@rm -f $*_python.stamp
 	@touch $*_python.stamp.tmp
-	$(AM_V_GEN)$(SWIG) $(PKG_SWIGFLAGS) $(SWIGFLAGS) \
-		-shadow -c++ -importall -Wall \
+	$(AM_V_GEN)$(SWIG) $(SWIGFLAGS) \
+		-shadow \
+	        -c++ \
+	        -importall \
+	        -Wall \
+		-ignoremissing \
 		-I$(srcdir) \
 		-I$(top_srcdir)/config/swig $(PYTHON_INCLUDES) \
-		 $(AM_CPPFLAGS) \
-		-c++ \
+		 $(AM_CPPFLAGS) $(PKG_SWIGFLAGS) \
 		-python -classic \
 		-o $*_python_wrap.cc $<
 	@mv $*_python.stamp.tmp $*_python.stamp
