--- mythtv/configure.prev	2016-08-17 13:26:45.000000000 -0400
+++ mythtv/configure	2016-08-17 13:27:12.000000000 -0400
@@ -6819,9 +6819,7 @@
                              { check_cpp_condition "OpenCL/cl.h" "defined(CL_VERSION_1_2)" ||
                                check_cpp_condition "CL/cl.h" "defined(CL_VERSION_1_2)" ||
                                die "ERROR: opencl must be installed and version must be 1.2 or compatible"; }
-enabled opengl            && { check_lib GL/glx.h glXGetProcAddress "-lGL" ||
-                               check_lib2 windows.h wglGetProcAddress "-lopengl32 -lgdi32" ||
-                               check_lib2 OpenGL/gl3.h glGetError "-Wl,-framework,OpenGL" ||
+enabled opengl            && { check_lib2 OpenGL/gl3.h glGetError "-Wl,-framework,OpenGL" ||
                                check_lib2 ES2/gl.h glGetError "-isysroot=${sysroot} -Wl,-framework,OpenGLES" ||
                                check_lib2 GLES/gl.h glGetError "-Wl,-framework,OpenGLES" ||
                                { check_lib2 GLES2/gl2.h glGetError "-lGLESv2" &&
