--- Makefile.orig	2014-12-21 04:26:26.000000000 -0600
+++ Makefile	2014-12-21 19:54:36.000000000 -0600
@@ -16,8 +16,6 @@
 LDLIBS_PTHREAD = -lpthread
 
 
-# Optimization flags
-CXXFLAGS = -O2
 
 #CXXFLAGS += -g -Wall
 #CXXFLAGS += -ansi
@@ -49,8 +47,6 @@
 CXXFLAGS += $(CXXFLAGS_PTHREAD)
 
 
-# Linker flags
-LDFLAGS =
 
 
 # External libraries
@@ -60,6 +56,7 @@
 
 # Where you want pbzip2 installed when you do 'make install'
 PREFIX = /usr
+DESTDIR =
 
 
 all: pbzip2
@@ -74,16 +71,16 @@
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
-	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
-	if ( test ! -d $(PREFIX)/share ) ; then mkdir -p $(PREFIX)/share ; fi
-	if ( test ! -d $(PREFIX)/share/man ) ; then mkdir -p $(PREFIX)/share/man ; fi
-	if ( test ! -d $(PREFIX)/share/man/man1 ) ; then mkdir -p $(PREFIX)/share/man/man1 ; fi
-	cp -f pbzip2 $(PREFIX)/bin/pbzip2
-	chmod a+x $(PREFIX)/bin/pbzip2
-	ln -s -f pbzip2 $(PREFIX)/bin/pbunzip2
-	ln -s -f pbzip2 $(PREFIX)/bin/pbzcat
-	cp -f pbzip2.1 $(PREFIX)/share/man/man1
-	chmod a+r $(PREFIX)/share/man/man1/pbzip2.1
+	if ( test ! -d $(DESTDIR)$(PREFIX)/bin ) ; then mkdir -p $(DESTDIR)$(PREFIX)/bin ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man ; fi
+	if ( test ! -d $(DESTDIR)$(PREFIX)/share/man/man1 ) ; then mkdir -p $(DESTDIR)$(PREFIX)/share/man/man1 ; fi
+	cp -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbzip2
+	chmod a+x $(DESTDIR)$(PREFIX)/bin/pbzip2
+	ln -s -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbunzip2
+	ln -s -f pbzip2 $(DESTDIR)$(PREFIX)/bin/pbzcat
+	cp -f pbzip2.1 $(DESTDIR)$(PREFIX)/share/man/man1
+	chmod a+r $(DESTDIR)$(PREFIX)/share/man/man1/pbzip2.1
 
 clean:
 	rm -f *.o pbzip2
