--- setup.py.orig	2013-03-30 17:53:02.000000000 +0400
+++ setup.py	2013-03-30 17:54:16.000000000 +0400
@@ -15,7 +15,7 @@
 from glob import glob
 
 # some config settings
-bind_libsvm = 'local' # choices: 'local', 'system', None
+bind_libsvm = 'system' # choices: 'local', 'system', None
 
 libsvmc_extra_sources = []
 libsvmc_include_dirs = []
@@ -63,12 +63,7 @@
     libsvmc_libraries += ['svm']
     if not sys.platform.startswith('win'):
         libsvmc_include_dirs += [
-            '/usr/include/libsvm-3.0/libsvm',
-            '/usr/include/libsvm-2.0/libsvm',
-            '/usr/include/libsvm',
-            '/usr/local/include/libsvm',
-            '/usr/local/include/libsvm-2.0/libsvm',
-            '/usr/local/include']
+            '@@PREFIX@@/include']
     else:
         # no clue on windows
         pass
