--- src/src/collective/communicator-inl.h.orig	2023-01-15 03:02:17.000000000 +0700
+++ src/src/collective/communicator-inl.h	2023-03-23 15:20:46.000000000 +0700
@@ -197,7 +197,7 @@
 template <Operation op, typename T,
           typename = std::enable_if_t<std::is_same<size_t, T>{} && !std::is_same<uint64_t, T>{}> >
 inline void Allreduce(T *send_receive_buffer, size_t count) {
-  static_assert(sizeof(T) == sizeof(uint64_t), "");
+// static_assert(sizeof(T) == sizeof(uint64_t), "");
   Communicator::Get()->AllReduce(send_receive_buffer, count, DataType::kUInt64, op);
 }
 
