--- configure.orig	2023-03-31 01:10:42.000000000 +0700
+++ configure	2023-03-31 03:58:35.000000000 +0700
@@ -2834,15 +2834,9 @@
 
 if test `uname -s` = "Darwin"
 then
-  if command -v brew &> /dev/null
-  then
-    HOMEBREW_LIBOMP_PREFIX=`brew --prefix libomp`
-  else
-    # Homebrew not found
-    HOMEBREW_LIBOMP_PREFIX=''
-  fi
-  OPENMP_CXXFLAGS="-Xpreprocessor -fopenmp -I${HOMEBREW_LIBOMP_PREFIX}/include"
-  OPENMP_LIB="-lomp -L${HOMEBREW_LIBOMP_PREFIX}/lib"
+  LIBOMP_PREFIX=@OMP_PREFIX@
+  OPENMP_CXXFLAGS="-Xpreprocessor -fopenmp -I${LIBOMP_PREFIX}/@INC@"
+  OPENMP_LIB="-L${LIBOMP_PREFIX}/@LIB@ @OMP@"
   ac_pkg_openmp=no
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether OpenMP will work in a package" >&5
 printf %s "checking whether OpenMP will work in a package... " >&6; }
@@ -2865,8 +2859,7 @@
     OPENMP_LIB=''
     echo '*****************************************************************************************'
     echo '         OpenMP is unavailable on this Mac OSX system. Training speed may be suboptimal.'
-    echo '         To use all CPU cores for training jobs, you should install OpenMP by running\n'
-    echo '             brew install libomp'
+    echo '         To use all CPU cores for training jobs, you should install OpenMP.'
     echo '*****************************************************************************************'
   fi
 fi
