--- configure.orig	2020-03-31 21:15:58.000000000 +0900
+++ configure	2020-12-30 08:55:14.000000000 +0900
@@ -27747,11 +27747,11 @@
 _ACEOF
 
     cat >>confdefs.h <<_ACEOF
-#define RUBY_ARCH "universal-"RUBY_PLATFORM_OS
+#define RUBY_ARCH "universal-" RUBY_PLATFORM_OS
 _ACEOF
 
     cat >>confdefs.h <<_ACEOF
-#define RUBY_PLATFORM "universal."RUBY_PLATFORM_CPU"-"RUBY_PLATFORM_OS
+#define RUBY_PLATFORM "universal." RUBY_PLATFORM_CPU "-" RUBY_PLATFORM_OS
 _ACEOF
 
 
