From e0bb397409adb1a6146dd5c1e044e0614070920e Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 20 Jan 2015 00:09:16 -0800
Subject: [PATCH 1005/1005] Default to fragile ObjC runtime when targeting
 darwin/ppc

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 lib/Driver/ToolChains.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git llvm_master/tools/clang/lib/Driver/ToolChains.h macports_master/tools/clang/lib/Driver/ToolChains.h
index a5f5473..4f53d95 100644
--- llvm_master/tools/clang/lib/Driver/ToolChains.h
+++ macports_master/tools/clang/lib/Driver/ToolChains.h
@@ -324,8 +324,10 @@ public:
   bool IsEncodeExtendedBlockSignatureDefault() const override { return true; }
 
   bool IsObjCNonFragileABIDefault() const override {
-    // Non-fragile ABI is default for everything but i386.
-    return getTriple().getArch() != llvm::Triple::x86;
+    // Non-fragile ABI is default for legacy architectures
+    return getTriple().getArch() != llvm::Triple::x86 &&
+           getTriple().getArch() != llvm::Triple::ppc &&
+           getTriple().getArch() != llvm::Triple::ppc64;
   }
 
   bool UseObjCMixedDispatch() const override { return true; }
-- 
2.7.0

