diff --git a/cmake/Modules/CompilerRTDarwinUtils.cmake b/cmake/Modules/CompilerRTDarwinUtils.cmake
index c1347de..33e457c 100644
--- llvm_release_38/projects/compiler-rt/cmake/Modules/CompilerRTDarwinUtils.cmake
+++ macports_release_38/projects/compiler-rt/cmake/Modules/CompilerRTDarwinUtils.cmake
@@ -31,7 +31,7 @@ function(darwin_get_toolchain_supported_archs output_var)
   else()
     # If auto-detecting fails, fall back to a default set
     message(WARNING "Detecting supported architectures from 'ld -v' failed. Returning default set.")
-    set(ARCHES "i386;x86_64;armv7;armv7s;arm64")
+    set(ARCHES "i386;x86_64;ppc;ppc64")
   endif()
   
   set(${output_var} ${ARCHES} PARENT_SCOPE)

