--- a/projects/compiler-rt/cmake/config-ix.cmake	2015-11-29 22:17:04.000000000 -0800
+++ b/projects/compiler-rt/cmake/config-ix.cmake	2015-11-29 22:19:16.000000000 -0800
@@ -313,9 +313,6 @@ if(APPLE)
     else()
       set(SANITIZER_MIN_OSX_VERSION 10.9)
     endif()
-    if(SANITIZER_MIN_OSX_VERSION VERSION_LESS "10.7")
-      message(FATAL_ERROR "Too old OS X version: ${SANITIZER_MIN_OSX_VERSION}")
-    endif()
   endif()
 
   # We're setting the flag manually for each target OS
