--- app/ige-mac-menu.c.orig	2009-10-20 20:36:40.000000000 +0200
+++ app/ige-mac-menu.c	2009-10-20 20:37:23.000000000 +0200
@@ -21,10 +21,10 @@
  * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  */
-#ifdef __APPLE__
 #include "config.h"
 
 #include <gtk/gtk.h>
+#ifdef GDK_WINDOWING_QUARTZ
 #include <gdk/gdkkeysyms.h>
 #include <gdk/gdkquartz.h>
 #include <Carbon/Carbon.h>
