--- Makefile.orig	2021-02-16 20:31:03.000000000 -0500
+++ Makefile	2021-02-16 20:33:25.000000000 -0500
@@ -27,15 +27,11 @@
 CP ?= cp
 OD ?= od
 
-OPT ?= -O3
-
 PREFIX ?= /usr/local
 
-CXXFLAGS ?= -g $(OPT) -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++0x -fPIC
-CXXFLAGS += -Iinclude -Ithird_party/md5 -Ithird_party/json
-CFLAGS ?= -g $(OPT) -Wall -Wextra -pedantic -std=c99 -fPIC
-CFLAGS += -Iinclude
-MAKEDEPENDFLAGS += -Iinclude -Ithird_party/md5 -Ithird_party/json
+CXXFLAGS += -g -Wall -Wextra -Woverloaded-virtual -pedantic -std=c++0x -fPIC -Iinclude -Ithird_party/md5 -I@@PREFIX@@/include/nlohmann
+CFLAGS ?= -g -Wall -Wextra -pedantic -std=c99 -fPIC -Iinclude -I@@PREFIX@@/include/nlohmann
+MAKEDEPENDFLAGS += -Iinclude -Ithird_party/md5 -I@@PREFIX@@/include/nhlomann
 EMCXXFLAGS = $(CXXFLAGS) --memory-init-file 0 -s DISABLE_EXCEPTION_CATCHING=0 -s INLINING_LIMIT=50 -s RESERVED_FUNCTION_POINTERS=20 -s ASSERTIONS=1 -s ALLOW_MEMORY_GROWTH=1
 EMCFLAGS = $(CFLAGS) --memory-init-file 0 -s DISABLE_EXCEPTION_CATCHING=0 -s ASSERTIONS=1 -s ALLOW_MEMORY_GROWTH=1
 LDFLAGS ?=
