--- configure.ac.orig	2014-05-14 14:51:10.000000000 -0700
+++ configure.ac	2014-05-14 14:52:07.000000000 -0700
@@ -174,7 +174,7 @@
 dnl **********************************
 dnl GTKHTML modules
 dnl **********************************
-GTKHTML_MODULES="$GTK_PACKAGE >= $gtk_minimum_version enchant >= enchant_minimum_version gconf-2.0 iso-codes"
+GTKHTML_MODULES="$GTK_PACKAGE >= $gtk_minimum_version enchant >= enchant_minimum_version gconf-2.0 gthread-2.0 iso-codes"
 PKG_CHECK_MODULES(GTKHTML, $GTKHTML_MODULES)
 AC_SUBST(GTKHTML_CFLAGS)
 AC_SUBST(GTKHTML_LIBS)
From ba580fac5e743461f8d7190919851f026d0fc050 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 2 May 2014 12:57:19 +0200
Subject: Bug #729020 - Drop gnome-icon-theme dependency

gnome-icon-theme got renamed to adwaita-icon-theme. And the
dependency is not really needed; depending on GTK+ should be
enough to guarantee the presence of a spec-compliant icon
theme.
---
 configure.ac | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2063ebe..04985a1 100644
--- configure.ac.orig	2015-10-05 15:13:55.000000000 -0700
+++ configure.ac	2015-10-05 15:15:12.000000000 -0700
@@ -10,7 +10,6 @@
 
 dnl Required Package Versions
 m4_define([enchant_minimum_version], [1.1.7])
-m4_define([gnome_icon_theme_minimum_version], [2.22.0])
 
 dnl **********************************
 dnl GTK version
@@ -215,11 +214,6 @@
 AM_CONDITIONAL(HAVE_NEWSOUP, test x$HAVE_NEWSOUP != xno)
 
 dnl **************************************************
-dnl Gnome Icon Theme
-dnl **************************************************
-PKG_CHECK_MODULES(GIT, gnome-icon-theme >= gnome_icon_theme_minimum_version)
-
-dnl **************************************************
 dnl iso-codes
 dnl **************************************************
 AC_MSG_CHECKING([for iso-codes package])

