--- Makefile.orig	2011-03-18 22:25:20.000000000 +0100
+++ Makefile	2011-04-05 09:51:09.000000000 +0200
@@ -1,4 +1,5 @@
-WAF=python tools/waf-light
+PYTHON=python
+WAF=$(PYTHON) tools/waf-light
 
 web_root = ryan@nodejs.org:~/web/nodejs.org/
 
@@ -23,34 +24,34 @@
 	@$(WAF) uninstall
 
 test: all
-	python tools/test.py --mode=release simple message
+	$(PYTHON) tools/test.py --mode=release simple message
 
 test-valgrind: all
-	python tools/test.py --mode=release --valgrind simple message
+	$(PYTHON) tools/test.py --mode=release --valgrind simple message
 
 test-all: all
-	python tools/test.py --mode=debug,release
+	$(PYTHON) tools/test.py --mode=debug,release
 
 test-all-valgrind: all
-	python tools/test.py --mode=debug,release --valgrind
+	$(PYTHON) tools/test.py --mode=debug,release --valgrind
 
 test-release: all
-	python tools/test.py --mode=release
+	$(PYTHON) tools/test.py --mode=release
 
 test-debug: all
-	python tools/test.py --mode=debug
+	$(PYTHON) tools/test.py --mode=debug
 
 test-message: all
-	python tools/test.py message
+	$(PYTHON) tools/test.py message
 
 test-simple: all
-	python tools/test.py simple
+	$(PYTHON) tools/test.py simple
 
 test-pummel: all
-	python tools/test.py pummel
+	$(PYTHON) tools/test.py pummel
 
 test-internet: all
-	python tools/test.py internet
+	$(PYTHON) tools/test.py internet
 
 build/default/node: all
 
