--- wscript.orig	2011-11-25 23:41:57.000000000 +0100
+++ wscript	2011-11-25 23:43:19.000000000 +0100
@@ -969,7 +969,6 @@
   bld.install_files('${PREFIX}/bin/', 'tools/node-waf', chmod=0755)
   bld.install_files('${LIBDIR}/node/wafadmin', 'tools/wafadmin/*.py')
   bld.install_files('${LIBDIR}/node/wafadmin/Tools', 'tools/wafadmin/Tools/*.py')
-  install_npm(bld)
 
 def install_npm(bld):
   start_dir = bld.path.find_dir('deps/npm')
