--- ./setup.py	2026-09-15 08:24:18
+++ ./setup.py	2026-09-25 14:05:33
@@ -310,11 +310,15 @@
     use_swig = get_flag('MUPDF_SETUP_USE_SWIG', True)
 
     b = ''
+    skip_cxx = os.environ.get('MUPDF_SETUP_SKIP_CXX') == '1'
     if not windows():
         b = 'm'     # Build C library.
-    if use_clang_python:
-        b += '0'    # Build C++ source.
-    b += '1'        # Build C++ library (also contains C library on Windows).
+    if skip_cxx:
+        b = '0'     # Only build C++ source (generate generated.pickle).
+    else:
+        if use_clang_python:
+            b += '0'    # Build C++ source.
+        b += '1'        # Build C++ library (also contains C library on Windows).
     if use_swig:
         b += '2'    # Build SWIG-generated source.
     b += '3'        # Build SWIG library _mupdf.so.
@@ -330,6 +334,9 @@
     if do_build == '0':
         # This is a hack for testing.
         log(f'Not doing build because $MUPDF_SETUP_DO_BUILD={do_build}')
+    elif skip_cxx:
+        log(f'build(): Building Python libraries only (C/C++ libs expected to be pre-installed) with: {command}')
+        subprocess.check_call(command, shell=True)
     else:
         log(f'build(): Building MuPDF C, C++ and Python libraries with: {command}')
         subprocess.check_call(command, shell=True)
