--- Makefile.orig	2009-02-05 09:06:54.000000000 -0600
+++ Makefile	2009-02-07 03:22:35.000000000 -0600
@@ -60,8 +60,8 @@
 	rm -f *$(DLL) *~ *.a *.o
 
 install:
-	test -d "$(DESTDIR)$(libdir)/pure" || mkdir -p "$(DESTDIR)$(libdir)/pure"
-	cp ffi.pure ffi$(DLL) "$(DESTDIR)$(libdir)/pure"
+	test -d "$(DESTDIR)`readlink $(libdir)/pure`" || mkdir -p "$(DESTDIR)`readlink $(libdir)/pure`"
+	cp ffi.pure ffi$(DLL) "$(DESTDIR)`readlink $(libdir)/pure`"
 
 uninstall:
 	rm -f "$(DESTDIR)$(libdir)/pure/ffi.pure" "$(DESTDIR)$(libdir)/pure/ffi$(DLL)"
