diff -ru ../Ice-3.3.0.orig/cpp/config/Make.rules ./cpp/config/Make.rules
--- ../Ice-3.3.0.orig/cpp/config/Make.rules	2008-05-16 10:24:06.000000000 -0700
+++ ./cpp/config/Make.rules	2008-05-23 12:40:52.000000000 -0700
@@ -163,9 +163,9 @@
 BZIP2_RPATH_LINK        = $(if $(BZIP2_HOME),$(call rpathlink,$(BZIP2_HOME)/$(libsubdir)))
 
 ifneq ($(DB_HOME),)
-   DB_FLAGS             = -I$(DB_HOME)/include
-   DB_LIBS              = -L$(DB_HOME)/$(libsubdir) -ldb_cxx
-   DB_RPATH_LINK        = $(call rpathlink,$(DB_HOME)/$(libsubdir))
+   DB_FLAGS             = -I$(DB_HOME)/include/db46
+   DB_LIBS              = -L$(DB_HOME)/$(libsubdir)/db46 -ldb_cxx
+   DB_RPATH_LINK        = $(call rpathlink,$(DB_HOME)/$(libsubdir)/db46)
 else
    ifeq ($(shell if [ -d /usr/include/db46 -a -d /usr/$(libsubdir)/db46 ] ; then echo yes; fi), yes)
       DB_FLAGS          = -I/usr/include/db46
