--- mico/include/mico/os-math.h.orig	2008-07-25 14:41:44.000000000 +0100
+++ mico/include/mico/os-math.h	2019-07-20 23:38:56.000000000 +0100
@@ -332,24 +332,6 @@
 #include <sunmath.h>
 #endif
 
-#if defined(__APPLE__) && defined(__MACH__)
-// it _IS_ defined on 10.4, 10.5
-#ifndef isinf
-#define isinf( x ) ( ( sizeof ( x ) == sizeof(double) ) ?           \
-                   __isinfd ( x ) :                                 \
-                   ( sizeof ( x ) == sizeof( float) ) ?             \
-                   __isinff ( x ) :                                 \
-                   __isinf  ( x ) )
-#endif
-#ifndef isnan
-#define isnan( x ) ( ( sizeof ( x ) == sizeof(double) ) ?           \
-                   __isnand ( x ) :                                 \
-                   ( sizeof ( x ) == sizeof( float) ) ?             \
-                   __isnanf ( x ) :                                 \
-                   __isnan  ( x ) )
-#endif
-#endif // __APPLE__ && __MACH__
-
 #include <mico/lmath.h>
 
 class OSMath {
