--- mico/include/mico/os-math.h	2005-11-25 12:15:52.000000000 +0100
+++ mico-mod/include/mico/os-math.h	2006-12-18 19:16:39.000000000 +0100
@@ -333,16 +333,21 @@
 #endif
 
 #if defined(__APPLE__) && defined(__MACH__)
+// it _IS_ defined on 10.4
+#ifndef isinf
 #define isinf( x ) ( ( sizeof ( x ) == sizeof(double) ) ?           \
                    __isinfd ( x ) :                                 \
                    ( sizeof ( x ) == sizeof( float) ) ?             \
                    __isinff ( x ) :                                 \
                    __isinf  ( x ) )
+#endif
+#ifndef isnan
 #define isnan( x ) ( ( sizeof ( x ) == sizeof(double) ) ?           \
                    __isnand ( x ) :                                 \
                    ( sizeof ( x ) == sizeof( float) ) ?             \
                    __isnanf ( x ) :                                 \
                    __isnan  ( x ) )
+#endif
 #endif // __APPLE__ && __MACH__
 
 #include <mico/lmath.h>
