--- /dev/null	2016-02-18 23:10:00.000000000 +0100
+++ ipe/mainmenu.xib	2016-02-18 23:07:52.000000000 +0100
@@ -0,0 +1,173 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<document type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="3.0"
+	  toolsVersion="9060" systemVersion="15B42"
+	  targetRuntime="MacOSX.Cocoa" propertyAccessControl="none"
+	  useAutolayout="YES" customObjectInstantitationMethod="direct">
+    <dependencies>
+      <deployment identifier="macosx"/>
+      <plugIn identifier="com.apple.InterfaceBuilder.CocoaPlugin"
+	      version="9060"/>
+    </dependencies>
+    <objects>
+      <customObject id="-2" userLabel="File's Owner" 
+		    customClass="NSApplication">
+        <connections>
+          <outlet property="delegate" destination="appdel" id="appdel-outlet"/>
+        </connections>
+      </customObject>
+      <customObject id="-1" userLabel="First Responder"
+		    customClass="FirstResponder"/>
+      <customObject id="-3" userLabel="Application" customClass="NSObject"/>
+      <customObject id="appdel" customClass="AppDelegate"/>
+      <menu title="Main Menu" systemMenu="main" id="main-menu">
+        <items>
+          <menuItem title="Ipe" id="apple-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Ipe" systemMenu="apple">
+              <items>
+                <menuItem title="About Ipe" id="about">
+                  <modifierMask key="keyEquivalentModifierMask"/>
+                  <connections>
+                    <action selector="aboutIpe:" target="-1" id="about-action"/>
+                  </connections>
+                </menuItem>
+                <menuItem isSeparatorItem="YES"/>
+                <menuItem title="Preferences…" keyEquivalent=","
+			  id="preferences"/>
+                <menuItem isSeparatorItem="YES"/>
+                <menuItem title="Services" id="services">
+                  <modifierMask key="keyEquivalentModifierMask"/>
+                  <menu key="submenu" title="Services"
+			systemMenu="services" id="services-submenu"/>
+                </menuItem>
+                <menuItem isSeparatorItem="YES"/>
+                <menuItem title="Hide Ipe" keyEquivalent="h" id="hide">
+                  <connections>
+                    <action selector="hide:" target="-1" id="hide-action"/>
+                  </connections>
+                </menuItem>
+                <menuItem title="Hide Others" keyEquivalent="h" 
+			  id="hide-others">
+                  <modifierMask key="keyEquivalentModifierMask"
+				option="YES" command="YES"/>
+                  <connections>
+                    <action selector="hideOtherApplications:"
+			    target="-1" id="hide-others-action"/>
+                  </connections>
+                </menuItem>
+                <menuItem title="Show All" id="show-all">
+                  <modifierMask key="keyEquivalentModifierMask"/>
+                  <connections>
+                    <action selector="unhideAllApplications:"
+			    target="-1" id="show-all-action"/>
+                  </connections>
+                </menuItem>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="File" id="file-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="File">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Edit" id="edit-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Edit">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Properties" id="properties-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Properties">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Snap" id="snap-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Snap">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Mode" id="mode-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Mode">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Zoom" id="zoom-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Zoom">
+              <items>
+                <menuItem title="Show Toolbar" keyEquivalent="t"
+			  id="show-toolbar">
+                  <modifierMask key="keyEquivalentModifierMask"
+				option="YES" command="YES"/>
+                  <connections>
+                    <action selector="toggleToolbarShown:" target="-1"
+			    id="show-toolbar-action"/>
+                  </connections>
+                </menuItem>
+                <menuItem title="Show Snap Toolbar" keyEquivalent="s"
+			  id="show-snapbar">
+                  <modifierMask key="keyEquivalentModifierMask"
+				option="YES" command="YES"/>
+                  <connections>
+                    <action selector="toggleSnapbarShown:" target="-1"
+			    id="show-snapbar-action"/>
+                  </connections>
+                </menuItem>
+                <menuItem title="Customize Toolbar…" id="customize-toolbar">
+                  <connections>
+                    <action selector="runToolbarCustomizationPalette:"
+			    target="-1" id="customize-toolbar-action"/>
+                  </connections>
+                </menuItem>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Layers" id="layers-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Layers">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Views" id="views-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Views">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Pages" id="pages-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Pages">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Ipelets" id="ipelets-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Ipelets">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+          <menuItem title="Help" id="help-menu">
+            <modifierMask key="keyEquivalentModifierMask"/>
+            <menu key="submenu" title="Help">
+              <items>
+              </items>
+            </menu>
+          </menuItem>
+        </items>
+      </menu>
+    </objects>
+  </document>
+  
