--- config.mak.orig	2009-11-04 20:56:07.000000000 +0100
+++ config.mak	2009-11-07 02:18:42.000000000 +0100
@@ -17,11 +17,13 @@
 FREETYPE_LIBS ?= $(shell pkg-config --libs freetype2)
 CAIRO_CFLAGS  ?= $(shell pkg-config --cflags cairo)
 CAIRO_LIBS    ?= $(shell pkg-config --libs cairo)
-LUA_CFLAGS    ?= $(shell pkg-config --cflags lua5.1)
-LUA_LIBS      ?= $(shell pkg-config --libs lua5.1)
+LUA_CFLAGS    ?= $(shell pkg-config --cflags lua)
+LUA_LIBS      ?= $(shell pkg-config --libs lua)
 QT_CFLAGS     ?= $(shell pkg-config --cflags QtGui QtCore)
 QT_LIBS	      ?= $(shell pkg-config --libs QtGui QtCore)
-MOC	      ?= moc
+ifeq "$(MOC)" ""
+MOC	?= moc
+endif
 #
 # --------------------------------------------------------------------
 #
@@ -39,6 +41,7 @@
 # Installing Ipe:
 #
 IPEVERS = 7.0.8
+INSTALL_DIR = $(DESTDIR)
 #
 # IPEPREFIX is the global prefix for the Ipe directory structure, which
 # you can override individually for any of the specific directories.
