--- lang/CMakeLists.txt.orig	2016-06-19 10:01:08.000000000 -0700
+++ lang/CMakeLists.txt	2016-12-17 15:47:57.000000000 -0700
@@ -39,7 +39,7 @@
 
 	include_directories(${PYTHON_INCLUDE_DIR}  ${NUMPY_INCLUDE_PATH})
 	execute_process(
-		COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}')"
+		COMMAND ${PYTHON_EXECUTABLE} -c "from distutils.sysconfig import get_python_lib; print get_python_lib()"
 		OUTPUT_VARIABLE MGL_PYTHON_SITE_PACKAGES
 		OUTPUT_STRIP_TRAILING_WHITESPACE)
 	set(SWIG_MODULE_mathgl_EXTRA_DEPS numpy.i ${src_imp_dep})
