--- CMakeLists.txt	2024-04-18 20:36:35.000000000 +0800
+++ CMakeLists.txt	2024-05-18 12:43:03.000000000 +0800
@@ -1,10 +1,6 @@
 cmake_minimum_required(VERSION 2.8.6 FATAL_ERROR)
 project(qmmp)
 
-#freebsd support
-include_directories(SYSTEM /usr/local/include)
-SET(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} /usr/local/include)
-
 #macports support
 IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 	SET(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} /opt/local/include)
