--- CMakeLists.txt	2024-04-18 20:36:35.000000000 +0800
+++ CMakeLists.txt	2024-05-18 10:49:51.000000000 +0800
@@ -84,10 +84,9 @@
 execute_process(COMMAND find ${CMAKE_CURRENT_SOURCE_DIR} -name *.ts COMMAND xargs ${QT_LRELEASE_EXECUTABLE} -silent)
 ENDIF(SVN_VERSION)
 
-
-IF(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
-	SET(CMAKE_INSTALL_LIBDIR "qmmp.app/Contents/Frameworks" CACHE)
-ENDIF()
+#IF(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
+#	SET(CMAKE_INSTALL_LIBDIR "qmmp.app/Contents/Frameworks" CACHE)
+#ENDIF()
 
 SET(PLUGIN_DIR "${CMAKE_INSTALL_LIBDIR}/qmmp-${QMMP_VERSION_MAJOR}.${QMMP_VERSION_MINOR}" CACHE STRING "plugin path")
 
@@ -101,10 +100,10 @@
 add_subdirectory(src/qmmpui)
 add_subdirectory(src/plugins)
 add_subdirectory(src/app)
-IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-	#Hack for MacOS X to run fix_mac_libs.sh script after installation
-	add_subdirectory(src/mac)
-ENDIF()
+#IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+#	#Hack for MacOS X to run fix_mac_libs.sh script after installation
+#	add_subdirectory(src/mac)
+#ENDIF()
 
 IF(MAD_FOUND)
   SET(MPEG_DECODERS "MAD")

--- src/app/CMakeLists.txt	2024-04-18 20:36:35.000000000 +0800
+++ src/app/CMakeLists.txt	2024-05-18 10:59:23.000000000 +0800
@@ -39,17 +39,17 @@
 # the UI file won't be wrapped!
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-ADD_EXECUTABLE(qmmp MACOSX_BUNDLE ${app_SRCS} ${app_RCC_SRCS} ${app_HDRS})
+ADD_EXECUTABLE(qmmp ${app_SRCS} ${app_RCC_SRCS} ${app_HDRS})
 target_link_libraries(qmmp ${QT_LIBRARIES} libqmmp libqmmpui)
 SET_TARGET_PROPERTIES(qmmp PROPERTIES OUTPUT_NAME qmmp${APP_NAME_SUFFIX})
 add_dependencies(qmmp libqmmpui libqmmp)
 
-IF(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
-	set(MACOSX_BUNDLE_GUI_IDENTIFIER "com.ylsoftware.qmmp")
-	set(MACOSX_BUNDLE_ICON_FILE "icons.icns")
-	install(TARGETS qmmp${APP_NAME_SUFFIX} DESTINATION .)
-	install(FILES images/mac/icons.icns DESTINATION ./qmmp.app/Contents/Resources)
-ELSE()
+#IF(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
+#	set(MACOSX_BUNDLE_GUI_IDENTIFIER "com.ylsoftware.qmmp")
+#	set(MACOSX_BUNDLE_ICON_FILE "icons.icns")
+#	install(TARGETS qmmp${APP_NAME_SUFFIX} DESTINATION .)
+#	install(FILES images/mac/icons.icns DESTINATION ./qmmp.app/Contents/Resources)
+#ELSE()
 	install(TARGETS qmmp DESTINATION ${CMAKE_INSTALL_BINDIR})
     install(FILES qmmp${APP_NAME_SUFFIX}.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
     install(FILES images/16x16/qmmp${APP_NAME_SUFFIX}.png DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/16x16/apps)
@@ -65,4 +65,4 @@
 		install(FILES qmmp-enqueue${APP_NAME_SUFFIX}.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
         install(FILES qmmp-dir${APP_NAME_SUFFIX}.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
 	ENDIF(USE_DIR_ASSOC)
-ENDIF()
+#ENDIF()

--- src/qmmp/CMakeLists.txt	2024-04-18 20:36:35.000000000 +0800
+++ src/qmmp/CMakeLists.txt	2024-05-18 12:40:28.000000000 +0800
@@ -148,7 +148,7 @@
                         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
                         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
-IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES Darwin)
+#IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES Darwin)
     IF(APP_NAME_SUFFIX)
         install(FILES ${libqmmp_DEVEL_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/qmmp${APP_NAME_SUFFIX}/qmmp)
         SET(QMMP_INCLUDE_DIR ${CMAKE_INSTALL_FULL_INCLUDEDIR}/qmmp${APP_NAME_SUFFIX})
@@ -158,4 +158,4 @@
     ENDIF()
 	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/qmmp.pc.in ${CMAKE_CURRENT_BINARY_DIR}/qmmp${APP_NAME_SUFFIX}.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qmmp${APP_NAME_SUFFIX}.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-ENDIF()
+#ENDIF()

--- src/qmmpui/CMakeLists.txt	2024-04-18 20:36:35.000000000 +0800
+++ src/qmmpui/CMakeLists.txt	2024-05-18 12:38:11.000000000 +0800
@@ -152,7 +152,7 @@
                           RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
                           ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
-IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES Darwin)
+#IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES Darwin)
     IF(APP_NAME_SUFFIX)
          install(FILES ${libqmmpui_DEVEL_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/qmmp${APP_NAME_SUFFIX}/qmmpui)
          SET(QMMP_INCLUDE_DIR ${CMAKE_INSTALL_FULL_INCLUDEDIR}/qmmp${APP_NAME_SUFFIX})
@@ -162,4 +162,4 @@
     ENDIF()
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/qmmpui.pc.in ${CMAKE_CURRENT_BINARY_DIR}/qmmpui${APP_NAME_SUFFIX}.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/qmmpui${APP_NAME_SUFFIX}.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-ENDIF()
+#ENDIF()
