--- Canna.conf.orig	2004-05-20 00:48:47.000000000 +0900
+++ Canna.conf	2007-10-12 17:20:00.000000000 +0900
@@ -109,7 +109,7 @@
 # define DefErrDir $(cannaPrefix)/log
 # define DefLibCannaDir $(cannaExecPrefix)/lib
 #else 
-# define DefCannaPrefix /usr/local/canna
+# define DefCannaPrefix __PREFIX__
 # ifdef ObsoleteInstallDir
 #  define DefErrDir /usr/spool/canna
 #  define DefLibCannaDir /usr/lib
@@ -147,12 +147,12 @@
 XCOMM cannaLibDir = /var/lib/canna
 XCOMM cannaLibDir = /var/db/canna
 cannaLibDir = DefCannaLibDir
-XCOMM cannaManDir = $(cannaPrefix)/share/man
-cannaManDir = $(cannaPrefix)/man
+cannaManDir = $(cannaPrefix)/share/man
+XCOMM cannaManDir = $(cannaPrefix)/man
 cannaIncDir = $(cannaPrefix)/include/canna
 
 libCannaDir = DefLibCannaDir
-ErrDir  = DefErrDir
+ErrDir  = ${cannaPrefix}/var/log/canna
 
 /* 旧バージョンとの互換APIを無効にする場合は0と定義してください */
 #define SupportOldWchar 1
@@ -169,8 +169,8 @@
 #define ModernElfLink NO /* experimental */
 
 
-cannaOwner = bin
-cannaGroup = bin
+cannaOwner = canna
+cannaGroup = canna
 
 #ifdef InstallAsUser
 cannaOwnerGroup =
@@ -205,7 +205,7 @@
 #endif
 
 /* 日本語マニュアルを使わないのであればコメントアウト */
-#define JAPANESEMAN
+/* #define JAPANESEMAN */
 
 /* #define engineSwitch */
 
@@ -237,7 +237,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = ${cannaPrefix}/var/db/canna/dic
 
 /* ここから下は変更不要です */
 
@@ -343,7 +343,7 @@
 AccessFile = $(cannaPrefix)/etc/hosts.canna
 #else
 UnixSockDir = /tmp/.iroha_unix
-AccessFile = /etc/hosts.canna
+AccessFile = $(cannaPrefix)/etc/hosts.canna
 #endif
 UnixSockName = IROHA
 
