# https://github.com/helix-editor/helix/pull/8932

--- ./languages.toml	2023-12-05 01:38:58
+++ ./languages.toml	2023-12-05 01:39:07
@@ -2998,7 +2998,7 @@
 
 [[grammar]]
 name = "gemini"
-source = { git = "https://git.sr.ht/~sfr/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
+source = { git = "https://git.sr.ht/~nbsp/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
 
 [[language]]
 name = "templ"
