--- lzma-4.42.0alpha6/src/lzmadec/Makefile.am.orig	2007-08-08 17:03:37.000000000 +0200
+++ lzma-4.42.0alpha6/src/lzmadec/Makefile.am	2007-10-03 12:13:21.000000000 +0200
@@ -17,7 +17,7 @@ bin_PROGRAMS = lzmadec
 lzmadec_SOURCES = lzmadec.c
 lzmadec_CPPFLAGS = -I@top_srcdir@/src/common -I@top_builddir@/lib
 lzmadec_LDFLAGS = -static
-lzmadec_LDADD = @top_builddir@/src/liblzma/liblzma.la
+lzmadec_LDADD = @top_builddir@/src/liblzma/liblzma.la @LTLIBINTL@ @PTHREAD_LIBS@
 
 if COND_GNULIB
 lzmadec_LDADD += @top_builddir@/lib/libgnu.a
--- lzma-4.42.0alpha6/src/lzmadec/Makefile.in.orig	2007-08-12 16:04:09.000000000 +0200
+++ lzma-4.42.0alpha6/src/lzmadec/Makefile.in	2007-10-03 12:14:50.000000000 +0200
@@ -208,7 +208,7 @@ top_srcdir = @top_srcdir@
 lzmadec_SOURCES = lzmadec.c
 lzmadec_CPPFLAGS = -I@top_srcdir@/src/common -I@top_builddir@/lib
 lzmadec_LDFLAGS = -static
-lzmadec_LDADD = @top_builddir@/src/liblzma/liblzma.la $(am__append_1)
+lzmadec_LDADD = @top_builddir@/src/liblzma/liblzma.la @LTLIBINTL@ @PTHREAD_LIBS@ $(am__append_1)
 all: all-am
 
 .SUFFIXES:
