--- configure.orig	2017-07-18 20:52:12.000000000 -0700
+++ configure	2017-11-25 23:46:25.000000000 -0700
@@ -178,22 +178,8 @@
   tcl_version=''
   tcl_version=`sh ./runtest $tk_defs tclversion.c`
   for tk_incs in \
-    "-I/usr/local/include" \
-    "-I/opt/local/include" \
-    "-I/sw/include" \
-    "-I/usr/pkg/include" \
-    "-I/usr/include" \
-    "-I/usr/X11/include" \
-    "-I/usr/local/include/tcl8.6 -I/usr/local/include/tk8.6" \
-    "-I/usr/include/tcl8.6 -I/usr/include/tk8.6" \
-    "-I/usr/local/include/tcl8.5 -I/usr/local/include/tk8.5" \
-    "-I/usr/include/tcl8.5 -I/usr/include/tk8.5" \
-    "-I/usr/local/include/tcl8.4 -I/usr/local/include/tk8.4" \
-    "-I/usr/include/tcl8.4 -I/usr/include/tk8.4" \
-    "-I/usr/local/include/tcl8.3 -I/usr/local/include/tk8.3" \
-    "-I/usr/include/tcl8.3 -I/usr/include/tk8.3" \
-    "-I/usr/local/include/tcl8.2 -I/usr/local/include/tk8.2" \
-    "-I/usr/include/tcl8.2 -I/usr/include/tk8.2"
+    "-I__MACPORTS_PREFIX__/include" \
+    "-I/usr/include"
   do if test -z "$tcl_version"; then
     tk_defs="$tk_incs $tk_x11_include"
     tcl_version=`sh ./runtest $tk_defs tclversion.c`
@@ -223,11 +209,7 @@
 if test $has_tk = true && test -z "$tk_libs"; then
   tklibdir=""
   case "$tk_defs" in
-  -I/opt/local/include*) tklibdir="/opt/local/lib" ;;
-  -I/usr/local/include*) tklibdir="/usr/local/lib" ;;
-  -I/sw/include*) tklibdir="/sw/lib" ;;
-  -I/usr/pkg/include*) tklibdir="/usr/pkg/lib" ;;
-  -I/usr/X11/include*) tklibdir="/usr/X11/lib" ;;
+  -I__MACPORTS_PREFIX__/include*) tklibdir="__MACPORTS_PREFIX__/lib" ;;
   esac
   if test -n "$tklibdir"; then
     case "$system" in
@@ -269,7 +251,7 @@
   else
     has_tk=false
     for tklibdir in \
-      "/usr/local/lib" "/opt/local/lib" "/sw/lib" "/usr/pkg/lib"  "/usr/lib";
+      "__MACPORTS_PREFIX__/lib" "/usr/lib";
     do
       if test $has_tk = false &&
          sh ./hasgot -L$tklibdir $tk_libs $tk_x11_libs $tkauxlibs Tk_SetGrid
