--- man/Makefile.in.orig	2010-04-08 13:34:43.000000000 -0500
+++ man/Makefile.in	2010-12-14 19:16:52.000000000 -0600
@@ -533,7 +533,7 @@
 
 dist_win: $(man_MANS)
 	groff -Tascii -man $(man_MANS) > Manpage
-	C_INCLUDE_PATH="" LIBRARY_PATH="" gcc cleaner.c -o cleaner
+	C_INCLUDE_PATH="" LIBRARY_PATH="" $(CC) cleaner.c -o cleaner
 	./cleaner Manpage
 	rm Manpage cleaner
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
