--- qiconfix/qiconloader_qt4.cpp	2015-03-29 17:30:27.000000000 +0800
+++ qiconfix/qiconloader_qt4.cpp	2023-08-27 15:44:34.000000000 +0800
@@ -492,9 +492,9 @@
                              QIcon::Mode mode, QIcon::State state)
 {
     QSize pixmapSize = rect.size();
-#if defined(Q_WS_MAC)
-    pixmapSize *= qt_mac_get_scalefactor();
-#endif
+//#if defined(Q_WS_MAC)
+//    pixmapSize *= qt_mac_get_scalefactor();
+//#endif
     painter->drawPixmap(rect, pixmap(pixmapSize, mode, state));
 }
 
