Index: configure.ac
===================================================================
--- configure.ac	(revision 10804)
+++ configure.ac	(revision 10820)
@@ -652,14 +652,11 @@
 
 # Before attempting to find valid compilers, set the corresponding precious
-# shell variable to "false" (the program/builtin that always fails) for any
-# languages that have been disabled by the user with "--disable-LANG".  This
-# should save a bit of configure time and also prevent user complaints like
-# ticket #1570.
-AS_IF([test "x$enable_f77" = "xno"],[F77=false])
-AS_IF([test "x$enable_fc"  = "xno"],[FC=false])
-# Set CXXCPP=$CPP so that we don't fail the (fatal) preprocessor sanity check.
-# At least as of autoconf-2.69, this sanity test does not check for any
-# C++-specific headers.
-AS_IF([test "x$enable_cxx" = "xno"],[CXX=false ; CXXCPP=$CPP])
+# shell variable to "no" for any languages that have been disabled by the user
+# with "--disable-LANG".  Libtool understands this as a request to disable
+# support for this language. This should save a bit of configure time and also
+# prevent user complaints like ticket #1570.
+AS_IF([test "x$enable_f77" = "xno"],[F77=no])
+AS_IF([test "x$enable_fc"  = "xno"],[FC=no])
+AS_IF([test "x$enable_cxx" = "xno"],[CXX=no])
 
 # suppress default "-g -O2" from AC_PROG_CXX
