--- src/util/mpir_cvars.c.orig	2021-02-19 13:22:26.000000000 -0600
+++ src/util/mpir_cvars.c	2021-02-19 13:23:31.000000000 -0600
@@ -6778,6 +6778,9 @@
     rc = MPL_env2bool("MPIR_CVAR_OFI_SKIP_IPV6", &(MPIR_CVAR_OFI_SKIP_IPV6));
     MPIR_ERR_CHKANDJUMP1((-1 == rc),mpi_errno,MPI_ERR_OTHER,"**envvarparse","**envvarparse %s","MPIR_CVAR_OFI_SKIP_IPV6");
 
+    /* DISABLE IPV6 FOR NOW. https://github.com/pmodels/mpich/issues/5041 */
+    MPIR_CVAR_OFI_SKIP_IPV6 = 1;
+
     defaultval.d = -1;
     MPIR_T_CVAR_REGISTER_STATIC(
         MPI_INT,
