--- src/mpi/datatype/typerep/src/typerep_util.h.orig
+++ src/mpi/datatype/typerep/src/typerep_util.h
@@ -48,6 +49,17 @@
 #define __BYTE_ORDER BYTE_ORDER
 #endif
 
+#if defined __APPLE__
+#if defined (__ppc__) || defined(__ppc64__)
+#ifndef __BYTE_ORDER
+#define __BYTE_ORDER __DARWIN_BYTE_ORDER
+#endif
+#ifndef __BIG_ENDIAN
+#define __BIG_ENDIAN __DARWIN_BIG_ENDIAN
+#endif
+#endif
+#endif
+
 #if defined(WORDS_BIGENDIAN)
 #define BLENDIAN 0
 #elif defined(WORDS_LITTLEENDIAN)
