--- Makefile.am.orig	2009-10-20 23:23:32.000000000 -0500
+++ Makefile.am	2009-10-20 23:25:12.000000000 -0500
@@ -80,8 +80,10 @@
 	src/tag.c
 
 src_libmpdclient_la_LDFLAGS = -version-info @LIBMPDCLIENT_LIBTOOL_VERSION@ \
-	-no-undefined \
-	-Wl,--version-script=$(srcdir)/libmpdclient.ld
+	-no-undefined
+if HAVE_GNU_LD
+src_libmpdclient_la_LDFLAGS += 	-Wl,--version-script=$(srcdir)/libmpdclient.ld
+endif
 
 noinst_PROGRAMS = src/example test/test
 
