From 1090f554654d563ce1477f52117fa3deb6ea011f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 20 Jan 2023 03:28:23 +0800
Subject: [PATCH] Do not use unsupported -march flag on PPC

Signed-off-by: Sergey Fedorov <vital.had@gmail.com>
---
 conf/gkbuild.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/conf/gkbuild.cmake b/conf/gkbuild.cmake
index 0e70a8e..591ece4 100644
--- conf/gkbuild.cmake
+++ conf/gkbuild.cmake
@@ -31,8 +31,13 @@ if(CMAKE_COMPILER_IS_GNUCC)
   set(GK_COPTIONS "${GK_COPTIONS} -std=c99 -fno-strict-aliasing")
 if(VALGRIND)
   set(GK_COPTIONS "${GK_COPTIONS} -march=x86-64 -mtune=generic")
+else()
+# -march=native is not a valid flag on PPC:
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "power|ppc|powerpc|ppc64|powerpc64" OR (APPLE AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc|ppc64"))
+  set(GK_COPTIONS "${GK_COPTIONS} -mtune=native")
 else()
   set(GK_COPTIONS "${GK_COPTIONS} -march=native")
+endif()
 endif(VALGRIND)
   if(NOT MINGW)
       set(GK_COPTIONS "${GK_COPTIONS} -fPIC")
