From b89fadcc31b6524f7b11af24e9feef5b9844c04c Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Wed, 17 Aug 2022 13:49:38 +0200
Subject: [PATCH] lib/Support/Unix/Path.inc: define COPYFILE_CLONE if needed

---
 llvm/lib/Support/Unix/Path.inc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/llvm/lib/Support/Unix/Path.inc b/llvm/lib/Support/Unix/Path.inc
index d91b269cc6d3..9be4eac0a1b5 100644
--- a/llvm/lib/Support/Unix/Path.inc
+++ b/llvm/lib/Support/Unix/Path.inc
@@ -38,6 +38,9 @@
 #include <mach-o/dyld.h>
 #include <sys/attr.h>
 #include <copyfile.h>
+#ifndef COPYFILE_CLONE
+#define COPYFILE_CLONE		(1<<24)
+#endif
 #elif defined(__FreeBSD__)
 #include <osreldate.h>
 #if __FreeBSD_version >= 1300057
-- 
2.37.2

