--- gcc/c-incpath.c.orig	2011-11-06 23:49:39.000000000 -0500
+++ gcc/c-incpath.c	2011-11-06 23:49:39.000000000 -0500
@@ -164,10 +164,30 @@
 	  char *str;
 
 	  /* Should this directory start with the sysroot?  */
-	  if (sysroot && p->add_sysroot)
-	    str = concat (sysroot, p->fname, NULL);
-	  else
+	  if (sysroot && p->add_sysroot) {
+        if (false) {
+        }
+        /* Don't append the MacPorts prefix to the sysroot, and instead use
+           "/usr/include". This is because the "-isysroot" option almost always
+           refers to an Mac OS X SDK which is structured that way. */
+        #ifdef PREFIX_INCLUDE_DIR
+        else if (!strcmp (p->fname, PREFIX_INCLUDE_DIR)) {
+          str = concat (sysroot, "/usr/include", NULL);
+        }
+        #endif
+        /* Don't append the MacPorts compiler's include directory to the
+           sysroot, and instead use it as-is. This is because the "-isysroot"
+           option almost always refers to an Mac OS X SDK which surely does not
+           contain a hierarchy for the compiler in question. */
+        else if (!strcmp (p->fname, cpp_GCC_INCLUDE_DIR)) {
+           str = (char *) cpp_GCC_INCLUDE_DIR;
+        }
+        else {
+          str = concat (sysroot, p->fname, NULL);
+        }
+      } else {
 	    str = update_path (p->fname, p->component);
+      }
 
 	  if (p->multilib && imultilib)
 	    str = concat (str, dir_separator_str, imultilib, NULL);
