--- gcc/c-incpath.c.orig	2012-04-27 11:33:12.000000000 -0700
+++ gcc/c-incpath.c	2012-04-27 11:34:17.000000000 -0700
@@ -164,7 +164,7 @@ add_standard_paths (const char *sysroot,
 	  char *str;
 
 	  /* Should this directory start with the sysroot?  */
-	  if (sysroot && p->add_sysroot)
+	  if (sysroot && p->add_sysroot && strstr(p->fname, "apple-gcc42") == NULL)
 	    str = concat (sysroot, p->fname, NULL);
 	  else
 	    str = update_path (p->fname, p->component);
