--- build_gcc.orig	2011-07-12 22:44:19.000000000 -0700
+++ build_gcc	2011-07-12 22:45:31.000000000 -0700
@@ -174,7 +174,10 @@ CONFIGFLAGS="--disable-checking --enable
   --prefix=$DEST_ROOT \
   --mandir=\${prefix}/share/man \
   --enable-languages=$LANGUAGES \
-  --program-transform-name=/^[cg][^.-]*$/s/$/-$MAJ_VERS/ \
+  --libexecdir=$DEST_ROOT/libexec/apple-gcc42 \
+  --libdir=$DEST_ROOT/lib/apple-gcc42 \
+  --includedir=$DEST_ROOT/include/apple-gcc42 \
+  --program-suffix=-apple-$MAJ_VERS \
   --with-slibdir=/usr/lib \
   --build=$BUILD-apple-darwin$DARWIN_VERS"
 
@@ -227,7 +230,7 @@ make $MAKEFLAGS DESTDIR=$DIR/dst-$BUILD-
 
 # Add the compiler we just built to the path, giving it appropriate names.
 D=$DIR/dst-$BUILD-$BUILD$DEST_ROOT/bin
-ln -f $D/gcc-$MAJ_VERS $D/gcc || exit 1
+ln -f $D/gcc-apple-$MAJ_VERS $D/gcc || exit 1
 ln -f $D/gcc $D/$BUILD-apple-darwin$DARWIN_VERS-gcc || exit 1
 PATH=$DIR/dst-$BUILD-$BUILD$DEST_ROOT/bin:$PATH
 
@@ -320,7 +323,7 @@ done
 # Rearrange various libraries, for no really good reason.
 for t in $CROSS_TARGETS ; do
   DT=$DIR/dst-$BUILD-$t
-  D=`echo $DT$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS`
+  D=`echo $DT$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS`
   mv $D/static/libgcc.a $D/libgcc_static.a || exit 1
   mv $D/kext/libgcc.a $D/libcc_kext.a || exit 1
   rm -r $D/static $D/kext || exit 1
@@ -408,13 +411,13 @@ cp -Rp $DIR/dst-$BUILD-$BUILD$DEST_ROOT/
 rm -rf .$DEST_ROOT/share/man/man7
 
 # libexec
-cd $DIR/dst-$BUILD-$BUILD$DEST_ROOT/libexec/gcc/$BUILD-apple-darwin$DARWIN_VERS/$VERS \
+cd $DIR/dst-$BUILD-$BUILD$DEST_ROOT/libexec/apple-gcc42/gcc/$BUILD-apple-darwin$DARWIN_VERS/$VERS \
   || exit 1
 LIBEXEC_FILES=`find . -type f -print || exit 1`
 LIBEXEC_DIRS=`find . -type d -print || exit 1`
 cd $DEST_DIR || exit 1
 for t in $TARGETS ; do
-  DL=$DEST_ROOT/libexec/gcc/$t-apple-darwin$DARWIN_VERS/$VERS
+  DL=$DEST_ROOT/libexec/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS
   for d in $LIBEXEC_DIRS ; do
     mkdir -p .$DL/$d || exit 1
   done
@@ -431,7 +434,7 @@ done
 
 # bin
 # The native drivers ('native' is different in different architectures).
-BIN_FILES=`ls $DIR/dst-$BUILD-$BUILD$DEST_ROOT/bin | grep '^[^-]*-[0-9.]*$' \
+BIN_FILES=`ls $DIR/dst-$BUILD-$BUILD$DEST_ROOT/bin | grep '^[^-]*-apple-[0-9.]*$' \
   | grep -v gccbug | grep -v gcov || exit 1`
 mkdir .$DEST_ROOT/bin
 for f in $BIN_FILES ; do
@@ -440,25 +443,25 @@ done
 # gcov, which is special only because it gets built multiple times and lipo
 # will complain if we try to add two architectures into the same output.
 TARG0=`echo $TARGETS | cut -d ' ' -f 1`
-lipo -output .$DEST_ROOT/bin/gcov-$MAJ_VERS -create \
+lipo -output .$DEST_ROOT/bin/gcov-apple-$MAJ_VERS -create \
   $DIR/dst-*-$TARG0$DEST_ROOT/bin/*gcov* || exit 1
 # The fully-named drivers, which have the same target on every host.
 for t in $TARGETS ; do
-  lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-gcc-$VERS -create \
+  lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-gcc-apple-$VERS -create \
     $DIR/dst-*-$t$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-gcc-$VERS || exit 1
-  lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-cpp-$VERS -create \
+  lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-cpp-apple-$VERS -create \
     $DIR/dst-*-$t$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-cpp-$VERS || exit 1
   if [ $BUILD_CXX -eq 1 ]; then
-    lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-g++-$VERS -create \
+    lipo -output .$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-g++-apple-$VERS -create \
     $DIR/dst-*-$t$DEST_ROOT/bin/$t-apple-darwin$DARWIN_VERS-g++* || exit 1
   fi
 done
 
 # lib
-mkdir -p .$DEST_ROOT/lib/gcc || exit 1
+mkdir -p .$DEST_ROOT/lib/apple-gcc42/gcc || exit 1
 for t in $TARGETS ; do
-  cp -Rp $DIR/dst-$BUILD-$t$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS \
-    .$DEST_ROOT/lib/gcc || exit 1
+  cp -Rp $DIR/dst-$BUILD-$t$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS \
+    .$DEST_ROOT/lib/apple-gcc42/gcc || exit 1
 done
 
 # APPLE LOCAL begin native compiler support
@@ -470,20 +473,20 @@ LIBGOMP_HOSTS=`echo $HOSTS | $OMIT_X86_6
 for t in $LIBGOMP_TARGETS ; do
     for h in $LIBGOMP_HOSTS ; do
 	if [ $h = $t ] ; then
-	    cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/libgomp.a \
-		.$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ || exit 1
-	    cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/libgomp.spec \
-		.$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ || exit 1
+	    cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/libgomp.a \
+		.$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ || exit 1
+	    cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/libgomp.spec \
+		.$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ || exit 1
 	    if [ $h = 'powerpc' ] ; then
-		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/ppc64/libgomp.a \
-		    .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ppc64/ || exit 1
-		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/ppc64/libgomp.spec \
-		    .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ppc64/ || exit 1
+		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/ppc64/libgomp.a \
+		    .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ppc64/ || exit 1
+		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/ppc64/libgomp.spec \
+		    .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/ppc64/ || exit 1
 	    elif [ $h = 'i686' ] ; then
-		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/x86_64/libgomp.a \
-		    .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/x86_64/ || exit 1
-		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/x86_64/libgomp.spec \
-		    .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/x86_64/ || exit 1
+		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/x86_64/libgomp.a \
+		    .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/x86_64/ || exit 1
+		cp -p $DIR/dst-$h-$t$DEST_ROOT/lib/apple-gcc42/x86_64/libgomp.spec \
+		    .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/x86_64/ || exit 1
 	    fi
 	fi
     done
@@ -494,19 +497,19 @@ if [ $BUILD_CXX -eq 1 ]; then
 for t in $TARGETS ; do
   if [ "$t" == 'arm' ] ; then
     cp -p $ARM_SYSROOT/usr/lib/libstdc++.6.dylib \
-      .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib \
+      .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib \
       || exit 1
   else
     cp -p /usr/lib/libstdc++.6.dylib \
-      .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib \
+      .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib \
       || exit 1
   fi
-  strip -x -c .$DEST_ROOT/lib/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib || exit 1
+  strip -x -c .$DEST_ROOT/lib/apple-gcc42/gcc/$t-apple-darwin$DARWIN_VERS/$VERS/libstdc++.dylib || exit 1
 done
 fi
 
 # include
-HEADERPATH=$DEST_ROOT/include/gcc/darwin/$MAJ_VERS
+HEADERPATH=$DEST_ROOT/include/gcc/darwin/apple-gcc42
 mkdir -p .$HEADERPATH || exit 1
 
 # Some headers are installed from more-hdrs/.  They all share
@@ -521,9 +524,9 @@ for h in `echo *.h` ; do
   if [ ! -f /usr/include/$h -o -L /usr/include/$h ] ; then
     cp -R $h $DEST_DIR$HEADERPATH/$h || exit 1
     for t in $TARGETS ; do
-      THEADERPATH=$DEST_DIR$DEST_ROOT/lib/gcc/${t}-apple-darwin$DARWIN_VERS/$VERS/include
+      THEADERPATH=$DEST_DIR$DEST_ROOT/lib/apple-gcc42/gcc/${t}-apple-darwin$DARWIN_VERS/$VERS/include
       [ -f $THEADERPATH/$h ] || \
-        ln -s ../../../../../include/gcc/darwin/$MAJ_VERS/$h $THEADERPATH/$h || \
+        ln -s ../../../../../../include/gcc/darwin/apple-gcc42/$h $THEADERPATH/$h || \
         exit 1
     done
   fi
@@ -532,15 +535,15 @@ done
 # Add extra man page symlinks for 'c++' and for arch-specific names.
 MDIR=$DEST_DIR$DEST_ROOT/share/man/man1
 if [ $BUILD_CXX -eq 1 ]; then
-  ln -f $MDIR/g++-$MAJ_VERS.1 $MDIR/c++-$MAJ_VERS.1 || exit 1
+  ln -f $MDIR/g++-apple-$MAJ_VERS.1 $MDIR/c++-apple-$MAJ_VERS.1 || exit 1
 fi
 for t in $TARGETS ; do
-  ln -f $MDIR/gcc-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-gcc-$VERS.1 \
+  ln -f $MDIR/gcc-apple-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-gcc-apple-$VERS.1 \
       || exit 1
-  ln -f $MDIR/cpp-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-cpp-$VERS.1 \
+  ln -f $MDIR/cpp-apple-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-cpp-apple-$VERS.1 \
       || exit 1
   if [ $BUILD_CXX -eq 1 ]; then
-    ln -f $MDIR/g++-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-g++-$VERS.1 \
+    ln -f $MDIR/g++-apple-$MAJ_VERS.1 $MDIR/$t-apple-darwin$DARWIN_VERS-g++-apple-$VERS.1 \
 	|| exit 1
   fi
 done
@@ -549,7 +552,7 @@ done
 for h in $HOSTS ; do
     $DIR/dst-$BUILD-$h$DEST_ROOT/bin/$h-apple-darwin$DARWIN_VERS-gcc-$VERS     \
 	$ORIG_SRC_DIR/driverdriver.c                               \
-	-DPDN="\"-apple-darwin$DARWIN_VERS-gcc-$VERS\""                                    \
+	-DPDN="\"-apple-darwin$DARWIN_VERS-gcc-apple-$VERS\""                                    \
 	-DIL="\"$DEST_ROOT/bin/\"" -I  $ORIG_SRC_DIR/include                   \
 	-I  $ORIG_SRC_DIR/gcc -I  $ORIG_SRC_DIR/gcc/config                     \
 	-liberty -L$DIR/dst-$BUILD-$h$DEST_ROOT/lib/                           \
@@ -558,7 +561,7 @@ for h in $HOSTS ; do
 	-o $DEST_DIR/$DEST_ROOT/bin/tmp-$h-gcc-$MAJ_VERS || exit 1
     $DIR/dst-$BUILD-$h$DEST_ROOT/bin/$h-apple-darwin$DARWIN_VERS-gcc-$VERS     \
 	$ORIG_SRC_DIR/driverdriver.c                               \
-	-DPDN="\"-apple-darwin$DARWIN_VERS-cpp-$VERS\""                                    \
+	-DPDN="\"-apple-darwin$DARWIN_VERS-cpp-apple-$VERS\""                                    \
 	-DIL="\"$DEST_ROOT/bin/\"" -I  $ORIG_SRC_DIR/include                   \
 	-I  $ORIG_SRC_DIR/gcc -I  $ORIG_SRC_DIR/gcc/config                     \
 	-liberty -L$DIR/dst-$BUILD-$h$DEST_ROOT/lib/                           \
@@ -568,7 +571,7 @@ for h in $HOSTS ; do
     if [ $BUILD_CXX -eq 1 ]; then
 	$DIR/dst-$BUILD-$h$DEST_ROOT/bin/$h-apple-darwin$DARWIN_VERS-gcc-$VERS     \
 	    $ORIG_SRC_DIR/driverdriver.c                               \
-	    -DPDN="\"-apple-darwin$DARWIN_VERS-g++-$VERS\""                                    \
+	    -DPDN="\"-apple-darwin$DARWIN_VERS-g++-apple-$VERS\""                                    \
 	    -DIL="\"$DEST_ROOT/bin/\"" -I  $ORIG_SRC_DIR/include                   \
 	    -I  $ORIG_SRC_DIR/gcc -I  $ORIG_SRC_DIR/gcc/config                     \
 	    -liberty -L$DIR/dst-$BUILD-$h$DEST_ROOT/lib/                           \
@@ -578,21 +581,21 @@ for h in $HOSTS ; do
     fi
 done
 
-lipo -output $DEST_DIR/$DEST_ROOT/bin/gcc-$MAJ_VERS -create \
+lipo -output $DEST_DIR/$DEST_ROOT/bin/gcc-apple-$MAJ_VERS -create \
   $DEST_DIR/$DEST_ROOT/bin/tmp-*-gcc-$MAJ_VERS || exit 1
 rm $DEST_DIR/$DEST_ROOT/bin/tmp-*-gcc-$MAJ_VERS || exit 1
-lipo -output $DEST_DIR/$DEST_ROOT/bin/cpp-$MAJ_VERS -create \
+lipo -output $DEST_DIR/$DEST_ROOT/bin/cpp-apple-$MAJ_VERS -create \
   $DEST_DIR/$DEST_ROOT/bin/tmp-*-cpp-$MAJ_VERS || exit 1
 rm $DEST_DIR/$DEST_ROOT/bin/tmp-*-cpp-$MAJ_VERS || exit 1
 
 if [ $BUILD_CXX -eq 1 ]; then
-  lipo -output $DEST_DIR/$DEST_ROOT/bin/g++-$MAJ_VERS -create \
+  lipo -output $DEST_DIR/$DEST_ROOT/bin/g++-apple-$MAJ_VERS -create \
        $DEST_DIR/$DEST_ROOT/bin/tmp-*-g++-$MAJ_VERS || exit 1
-  ln -f $DEST_DIR/$DEST_ROOT/bin/g++-$MAJ_VERS $DEST_DIR/$DEST_ROOT/bin/c++-$MAJ_VERS || exit 1
+  ln -f $DEST_DIR/$DEST_ROOT/bin/g++-apple-$MAJ_VERS $DEST_DIR/$DEST_ROOT/bin/c++-apple-$MAJ_VERS || exit 1
   rm $DEST_DIR/$DEST_ROOT/bin/tmp-*-g++-$MAJ_VERS || exit 1
 
   # Remove extraneous stuff
-  rm -rf $DEST_DIR/$DEST_ROOT/lib/gcc/*/*/include/c++
+  rm -rf $DEST_DIR/$DEST_ROOT/lib/apple-gcc42/gcc/*/*/include/c++
 fi
 
 
