--- build_gcc.orig	2011-07-06 17:54:57.000000000 -0700
+++ build_gcc	2011-07-06 17:55:55.000000000 -0700
@@ -81,10 +81,10 @@ MAJ_VERS=`echo $VERS | sed 's/\([0-9]*\.
 
 # This is the libstdc++ version to use.
 LIBSTDCXX_VERSION=4.2.1
-if [ ! -d "$DEST_ROOT/include/c++/$LIBSTDCXX_VERSION" ]; then
+if [ ! -d "/usr/include/c++/$LIBSTDCXX_VERSION" ]; then
   LIBSTDCXX_VERSION=4.0.0
 fi
-NON_ARM_CONFIGFLAGS="--with-gxx-include-dir=\${prefix}/include/c++/$LIBSTDCXX_VERSION"
+NON_ARM_CONFIGFLAGS="--with-gxx-include-dir=/usr/include/c++/$LIBSTDCXX_VERSION"
 
 # Build against the MacOSX10.5 SDK for PowerPC.
 PPC_SYSROOT=/Developer/SDKs/MacOSX10.5.sdk
