--- brut.apktool/apktool-lib/build.gradle	2020-12-02 08:04:44.000000000 +0800
+++ brut.apktool/apktool-lib/build.gradle	2021-06-25 01:41:17.000000000 +0800
@@ -19,11 +19,13 @@
 
 processResources {
   from('src/main/resources/properties') {
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
     include '**/*.properties'
     into 'properties'
     filter(ReplaceTokens, tokens: [version: project.apktool_version, gitrev: project.hash] )
   }
   from('src/main/resources/') {
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
     include '**/*.jar'
   }
   
