--- Tools/CMakeLists.txt.orig	2011-05-08 20:28:05.000000000 +0200
+++ Tools/CMakeLists.txt	2011-11-12 18:35:58.000000000 +0100
@@ -14,3 +14,15 @@
   add_subdirectory(XMLConverter)
   add_subdirectory(MeshUpgrader)
 endif (NOT OGRE_BUILD_PLATFORM_IPHONE)
+
+if (MACPORTS)
+  install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
+    DESTINATION ${MACPORTS_OGRE_TOOLS_PATH}
+    PATTERN ".svn" EXCLUDE
+    PATTERN "src" EXCLUDE
+    PATTERN "CMakeFiles" EXCLUDE
+    PATTERN "cmake_install.cmake" EXCLUDE
+    PATTERN "CMakeLists.txt" EXCLUDE
+    PATTERN "Makefile" EXCLUDE
+  )
+endif (MACPORTS)
