--- Samples/Browser/CMakeLists.txt.orig	2011-05-08 20:28:04.000000000 +0200
+++ Samples/Browser/CMakeLists.txt	2011-11-11 05:22:17.000000000 +0100
@@ -218,18 +218,25 @@
 		${OGRE_SAMPLE_CONTENTS_PATH}/Frameworks/
 		)
 	# now cfg files
+	if(OGRE_SDK_BUILD AND MACPORTS)
+	  set(OGRE_CFGFILES_PATH "${MACPORTS_CONFIGFILE_PATH}")
+	else()
+	  set(OGRE_CFGFILES_PATH "${OGRE_BINARY_DIR}/inst/bin/release")
+	endif()
+
 	add_custom_command(TARGET SampleBrowser POST_BUILD
 		COMMAND mkdir ARGS -p ${OGRE_SAMPLE_CONTENTS_PATH}/Resources
-		COMMAND ln ARGS -s -f ${OGRE_BINARY_DIR}/bin/plugins.cfg 
+		COMMAND cp ARGS ${OGRE_CFGFILES_PATH}/plugins.cfg 
 		${OGRE_SAMPLE_CONTENTS_PATH}/Resources/
-		COMMAND ln ARGS -s -f ${OGRE_BINARY_DIR}/bin/resources.cfg 
+		COMMAND cp ARGS ${OGRE_CFGFILES_PATH}/resources.cfg 
 		${OGRE_SAMPLE_CONTENTS_PATH}/Resources/
-		COMMAND ln ARGS -s -f ${OGRE_BINARY_DIR}/bin/samples.cfg 
+		COMMAND cp ARGS ${OGRE_CFGFILES_PATH}/samples.cfg 
 		${OGRE_SAMPLE_CONTENTS_PATH}/Resources/
-		COMMAND ln ARGS -s -f ${OGRE_BINARY_DIR}/bin/quakemap.cfg 
+		COMMAND cp ARGS ${OGRE_CFGFILES_PATH}/quakemap.cfg 
 		${OGRE_SAMPLE_CONTENTS_PATH}/Resources/
 		)
 
+	if (NOT OGRE_STATIC)
 	# now plugins
 	add_custom_command(TARGET SampleBrowser POST_BUILD
 		COMMAND mkdir ARGS -p ${OGRE_SAMPLE_CONTENTS_PATH}/Plugins)
@@ -313,6 +320,7 @@
 		${OGRE_SAMPLE_CONTENTS_PATH}/Plugins/
 		)
 	ENDIF()
+	endif(NOT OGRE_STATIC)
   endif()
 endif (APPLE)
 
