--- Tools/XMLConverter/CMakeLists.txt.orig	2011-08-05 09:53:08.000000000 -0500
+++ Tools/XMLConverter/CMakeLists.txt	2011-08-05 09:54:11.000000000 -0500
@@ -27,7 +27,7 @@
   src/tinyxmlparser.cpp
 )
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/include)
 add_definitions(-DTIXML_USE_STL)
 add_executable(OgreXMLConverter ${HEADER_FILES} ${SOURCE_FILES})
 target_link_libraries(OgreXMLConverter ${OGRE_LIBRARIES})
