--- json_spirit_v4.03/CMakeLists.txt	2010-01-09 20:35:31.000000000 +0100
+++ json_spirit_v4.03.patched/CMakeLists.txt	2010-07-05 22:41:58.000000000 +0200
@@ -11,6 +11,7 @@
   ${CMAKE_SOURCE_DIR}/json_spirit/json_spirit_utils.h
   ${CMAKE_SOURCE_DIR}/json_spirit/json_spirit_value.h
   ${CMAKE_SOURCE_DIR}/json_spirit/json_spirit_writer.h
+  ${CMAKE_SOURCE_DIR}/json_spirit/json_spirit_error_position.h
   DESTINATION include)
 
 INSTALL(
