diff --git requirements.txt requirements.txt
index 1c49ba0..3b659ad 100644
--- requirements.txt
+++ requirements.txt
@@ -1,5 +1,5 @@
-pyserial==3.5
-PyQt5==5.15.4
-numpy==1.21.1
-scipy==1.7.1
-cython==0.29.24
+pyserial>=3.5
+PyQt5>=5.15.4
+numpy>=1.21.1
+scipy>=1.7.1
+cython>=0.29.24
diff --git setup.cfg setup.cfg
index 646e60b..d7a0dd7 100644
--- setup.cfg
+++ setup.cfg
@@ -15,11 +15,11 @@ version = attr: NanoVNASaver.About.VERSION
 [options]
 packages = find_namespace:
 install_requires=
-    pyserial==3.5
-    PyQt5==5.15.4
-    numpy==1.21.1
-    scipy==1.7.1
-    cython==0.29.24
+    pyserial>=3.5
+    PyQt5>=5.15.4
+    numpy>=1.21.1
+    scipy>=1.7.1
+    cython>=0.29.24
 python_requires = >=3.8, <4
 
 [options.entry_points]
