--- __init__.py	2009-03-17 18:41:36.000000000 -0500
+++ __init__.py	2009-03-17 18:42:04.000000000 -0500
@@ -34,7 +34,7 @@
 __author__ = 'Jelmer Vernooij <jelmer@samba.org>'
 
 COMPATIBLE_BZR_VERSIONS = [(1, 6, 0), (1, 7, 0), (1, 8, 0), (1, 9, 0),
-                           (1, 10, 0), (1, 11, 0), (1, 12, 0)]
+                           (1, 10, 0), (1, 11, 0), (1, 12, 0), (1, 13, 0)]
 
 bzrlib.api.require_any_api(bzrlib, COMPATIBLE_BZR_VERSIONS)
 
