--- install.py.orig	2019-09-18 13:44:14.000000000 +0200
+++ install.py	2019-09-18 13:47:56.000000000 +0200
@@ -1213,11 +1213,12 @@
         if not os.path.exists(directory):
             os.makedirs(directory)
     
-    if macPythonExe == defaultMacPythonExe:
-        starter = os.path.join(directories["exe"], "eric")
-        os.symlink(macPythonExe, starter)
-    else:
-        starter = "python{0}".format(sys.version_info.major)
+##    if macPythonExe == defaultMacPythonExe:
+##        starter = os.path.join(directories["exe"], "eric")
+##        os.symlink(macPythonExe, starter)
+##    else:
+##        starter = "python{0}".format(sys.version_info.major)
+    starter = "@@PYTHON_BIN_NAME@@"
     
     wname = os.path.join(directories["exe"], "eric6")
     
@@ -1246,13 +1247,15 @@
                 pathlist_n.append(path_)
         pathLine = "PATH={0}\n".format(os.pathsep.join(pathlist_n))
     
+    pydir = "@@PYTHON_PKG_DIR@@"
+    pyqt_ver = "@@PYQT_VERSION@@"
     # create the wrapper script
     wrapper = ('''#!/bin/sh\n'''
                '''\n'''
                '''{0}'''
                '''{1}'''
-               '''exec "{2}" "{3}/{4}.py" "$@"\n'''
-               .format(pathLine, dyldLine, starter, pydir, "eric6"))
+               '''exec "{2}" "{3}/{4}.py" "{5}" "$@"\n'''
+               .format(pathLine, dyldLine, starter, pydir, "eric6", pyqt_ver))
     copyToFile(wname, wrapper)
     os.chmod(wname, 0o755)
     
