--- setup.py.orig	2019-01-14 16:56:31.000000000 +0300
+++ setup.py	2019-01-14 17:00:37.000000000 +0300
@@ -6,12 +6,6 @@
 from setuptools import Extension
 from setuptools import setup
 
-try:
-    from wheel.bdist_wheel import bdist_wheel
-except ImportError:
-    bdist_wheel = None
-
-
 def get_arch():
     if sys.maxsize > 2**32:
         arch = 64
@@ -204,21 +198,6 @@
 
 extensions = [Extension("pymunk.chipmunk", sources)]
 
-class bdist_wheel_universal_extension(bdist_wheel):
-    """
-    bdist_wheel give overly strict tags for python packages that uses native 
-    dynamic linked library called from cffi at runtime.
-
-    References
-    https://www.python.org/dev/peps/pep-0491/
-    https://www.python.org/dev/peps/pep-0427/
-    https://www.python.org/dev/peps/pep-0425/
-    https://github.com/getsentry/milksnake
-    """
-    def get_tag(self):
-        rv = bdist_wheel.get_tag(self)
-        return ('py2.py3', 'none',) + rv[2:]
-
 setup(
     name = 'pymunk',
     url = 'http://www.pymunk.org',
@@ -233,7 +212,6 @@
     classifiers = classifiers,
     cmdclass = {
         'build_ext': build_chipmunk,
-        'bdist_wheel': bdist_wheel_universal_extension
     },
     install_requires = ['cffi'],
     extras_require = {'dev': ['pyglet','pygame','sphinx']},    
