--- setup.py.orig	2012-03-26 16:51:00.000000000 +0400
+++ setup.py	2012-03-26 16:51:53.000000000 +0400
@@ -102,11 +102,7 @@
     , description='pymunk is a easy-to-use pythonic 2d physics library built on top of Chipmunk'
     , long_description=open('README.txt').read()
     , packages=['pymunk']
-    , package_data = {'pymunk': ['chipmunk.dll'
-                                , 'chipmunk64.dll'
-                                , 'libchipmunk.so'
-                                , 'libchipmunk64.so'
-                                , 'libchipmunk.dylib']}
+    , package_data = {'pymunk': ['libchipmunk.dylib']}
     , license='MIT License'
     , classifiers=classifiers
     , cmdclass={'build_chipmunk':build_chipmunk}
