From 3aa6d1ee974918c0b34e4cd88102d976203c85fc Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Mon, 15 Apr 2013 22:38:18 -0700
Subject: [PATCH 2001/2012] MacPorts Only: Comment out SL cctools workaround

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 make/platform/clang_darwin.mk | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git llvm_release_38/projects/compiler-rt/make/platform/clang_darwin.mk macports_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
index 9944481..b0805d9 100644
--- llvm_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
+++ macports_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -125,11 +125,11 @@ UniversalArchs.ubsan_iossim_dynamic := $(call CheckArches,i386 x86_64,ubsan_ioss
 # object files. If we are on that platform, strip out all ARM archs. We still
 # build the libraries themselves so that Clang can find them where it expects
 # them, even though they might not have an expected slice.
-ifneq ($(shell test -x /usr/bin/sw_vers && sw_vers -productVersion | grep 10.6),)
-UniversalArchs.ios := $(filter-out armv7, $(UniversalArchs.ios))
-UniversalArchs.cc_kext_ios := $(filter-out armv7, $(UniversalArchs.cc_kext_ios))
-UniversalArchs.profile_ios := $(filter-out armv7, $(UniversalArchs.profile_ios))
-endif
+#ifneq ($(shell test -x /usr/bin/sw_vers && sw_vers -productVersion | grep 10.6),)
+#UniversalArchs.ios := $(filter-out armv7, $(UniversalArchs.ios))
+#UniversalArchs.cc_kext_ios := $(filter-out armv7, $(UniversalArchs.cc_kext_ios))
+#UniversalArchs.profile_ios := $(filter-out armv7, $(UniversalArchs.profile_ios))
+#endif
 
 # If RC_SUPPORTED_ARCHS is defined, treat it as a list of the architectures we
 # are intended to support and limit what we try to build to that.
-- 
2.10.1

