diff --git a/unittests/Makefile.unittest b/unittests/Makefile.unittest
index bd32aed..993b69a 100644
--- a/unittests/Makefile.unittest
+++ b/unittests/Makefile.unittest
@@ -54,7 +54,7 @@ $(LLVMUnitTestExe): $(ObjectsO) $(ProjLibsPaths) $(LLVMLibsPaths)
 	$(Echo) ======= Finished Linking $(BuildMode) Unit test $(TESTNAME) \
           $(StripWarnMsg)
 
-all:: $(LLVMUnitTestExe)
+all::
 
 unitcheck:: $(LLVMUnitTestExe)
 	$(Run.Shared) $(LLVMUnitTestExe)
