--- check/exam_clifford.cpp.orig	2013-12-02 12:05:05.000000000 -0500
+++ check/exam_clifford.cpp	2013-12-02 12:05:20.000000000 -0500
@@ -546,7 +546,7 @@
 	realsymbol a("a");
 	varidx mu(symbol("mu", "\\mu"), 1);
 
-	ex e = clifford_unit(mu, diag_matrix(lst(-1))), e0 = e.subs(mu==0);
+	ex e = clifford_unit(mu, diag_matrix(lst(ex(-1)))), e0 = e.subs(mu==0);
 	result += ( exp(a*e0)*e0*e0 == -exp(e0*a) ) ? 0 : 1;
 
 	return result;
