--- check/exam_inifcns_nstdsums.cpp.orig	2013-12-02 11:55:47.000000000 -0500
+++ check/exam_inifcns_nstdsums.cpp	2013-12-02 12:03:09.000000000 -0500
@@ -192,10 +192,10 @@
 	res.append(zeta(lst(4,2,3)) - (-zeta(9)*59 + zeta(2)*zeta(7)*28 + pow(zeta(2),2)*zeta(5)*4 -
 	                               pow(zeta(3),3)/3 + pow(zeta(2),3)*zeta(3)*8/21));
 	res.append(zeta(lst(3,1,3,1,3,1,3,1)) - (2*pow(Pi,16)/factorial(18)));
-	res.append(zeta(lst(2),lst(-1)) - -zeta(2)/2);
-	res.append(zeta(lst(1,2),lst(-1,1)) - (-zeta(3)/4 - zeta(lst(1),lst(-1))*zeta(2)/2));
-	res.append(zeta(lst(2,1,1),lst(-1,-1,1)) - (-pow(zeta(2),2)*23/40 - pow(zeta(lst(1),lst(-1)),2)*zeta(2)*3/4
-	                                            - zeta(lst(3,1),lst(-1,1))*3/2 - zeta(lst(1),lst(-1))*zeta(3)*21/8));
+	res.append(zeta(lst(ex(2)),lst(ex(-1))) - -zeta(2)/2);
+	res.append(zeta(lst(1,2),lst(-1,1)) - (-zeta(3)/4 - zeta(lst(ex(1)),lst(ex(-1)))*zeta(2)/2));
+	res.append(zeta(lst(2,1,1),lst(-1,-1,1)) - (-pow(zeta(2),2)*23/40 - pow(zeta(lst(ex(1)),lst(ex(-1))),2)*zeta(2)*3/4
+	                                            - zeta(lst(3,1),lst(-1,1))*3/2 - zeta(lst(ex(1)),lst(ex(-1)))*zeta(3)*21/8));
 	
 	for (lst::const_iterator it = res.begin(); it != res.end(); it++) {
 		Digits = 17;
@@ -241,7 +241,7 @@
 
 	lst res;
 	
-	res.append(Li(lst(4), lst(6)).hold() - Li(4, 6.0));
+	res.append(Li(lst(ex(4)), lst(ex(6))).hold() - Li(4, 6.0));
 	res.append(G(lst(0,0,5.0,0,2.0,0,0,0,3.0),0.5).hold()
 	           + Li(lst(3,2,4), lst(numeric(1,10), numeric(5,2), numeric(2,3))));
 	res.append(Li(lst(2,1,1), lst(almostone, almostone, almostone)) - zeta(lst(2,1,1)));
@@ -258,20 +258,20 @@
 	res.append(s1.subs(lst(x==2, y==numeric(4)/5)) - s2.subs(lst(x==2, y==numeric(4)/5)));
 
 	// shuffle and quasi-shuffle identities
-	res.append(G(lst(0,0.2),1).hold() * G(lst(0.5),1).hold() - G(lst(0.5,0,0.2),1).hold()
-			- G(lst(0,0.5,0.2),1).hold() - G(lst(0,0.2,0.5),1).hold());
-	res.append(G(lst(0,0.5),1).hold() * G(lst(0.6),1).hold() - G(lst(0,0.5,0.5*0.6),1).hold()
+	res.append(G(lst(ex(0),0.2),1).hold() * G(lst(ex(0.5)),1).hold() - G(lst(ex(0.5),0,0.2),1).hold()
+			- G(lst(ex(0),0.5,0.2),1).hold() - G(lst(ex(0),0.2,0.5),1).hold());
+	res.append(G(lst(ex(0),0.5),1).hold() * G(lst(ex(0.6)),1).hold() - G(lst(0,0.5,0.5*0.6),1).hold()
 			- G(lst(0.6,0,0.5*0.6),1).hold() + G(lst(0,0,0.5*0.6),1).hold());
-	res.append(Li(lst(2),lst(numeric(1,5))).hold() * Li(lst(3),lst(7)).hold() - Li(lst(2,3),lst(numeric(1,5),7)).hold()
-			- Li(lst(3,2),lst(7,numeric(1,5))).hold() - Li(lst(5),lst(numeric(7,5))).hold());
+	res.append(Li(lst(ex(2)),lst(numeric(1,5))).hold() * Li(lst(ex(3)),lst(ex(7))).hold() - Li(lst(2,3),lst(numeric(1,5),7)).hold()
+			- Li(lst(ex(3),2),lst(ex(7),numeric(1,5))).hold() - Li(lst(ex(5)),lst(numeric(7,5))).hold());
 	symbol a1, a2, a3, a4;
 	res.append((G(lst(a1,a2),1) * G(lst(a3,a4),1) - G(lst(a1,a2,a3,a4),1)
 			- G(lst(a1,a3,a2,a4),1) - G(lst(a3,a1,a2,a4),1)
 			- G(lst(a1,a3,a4,a2),1) - G(lst(a3,a1,a4,a2),1) - G(lst(a3,a4,a1,a2),1))
 				.subs(lst(a1==numeric(1)/10, a2==numeric(3)/10, a3==numeric(7)/10, a4==5)));
-	res.append(G(lst(-0.009),1).hold() * G(lst(-8,1.4999),1).hold() - G(lst(-0.009,-8,1.4999),1).hold()
+	res.append(G(lst(ex(-0.009)),1).hold() * G(lst(ex(-8),1.4999),1).hold() - G(lst(-0.009,-8,1.4999),1).hold()
 			- G(lst(-8,-0.009,1.4999),1).hold() - G(lst(-8,1.4999,-0.009),1).hold());
-	res.append(G(lst(sqrt(numeric(1)/2)+I*sqrt(numeric(1)/2)),1).hold() * G(lst(1.51,-0.999),1).hold()
+	res.append(G(lst(sqrt(numeric(1)/2)+I*sqrt(numeric(1)/2)),1).hold() * G(lst(ex(1.51),-0.999),1).hold()
 			- G(lst(sqrt(numeric(1)/2)+I*sqrt(numeric(1)/2),1.51,-0.999),1).hold()
 			- G(lst(1.51,sqrt(numeric(1)/2)+I*sqrt(numeric(1)/2),-0.999),1).hold()
 			- G(lst(1.51,-0.999,sqrt(numeric(1)/2)+I*sqrt(numeric(1)/2)),1).hold());
@@ -313,15 +313,15 @@
 
 	ex x1 = exp(2*Pi*I/13).evalf();
 	ex x2 = exp(24*Pi*I/13).evalf();
-  	ex r2 = Li(lst(2),lst(x1)).hold().evalf();
-	ex r3 = Li(lst(2),lst(x2)).hold().evalf();
+  	ex r2 = Li(lst(ex(2)),lst(x1)).hold().evalf();
+	ex r3 = Li(lst(ex(2)),lst(x2)).hold().evalf();
 	if ( abs(r2-conjugate(r3)) > prec ) {
 		clog << "Legacy test 2 seems to be wrong." << endl;
 		result++;
 	}
 
   	ex x3 = exp(5*Pi*I/3).evalf();
-	ex r4 = Li(lst(3),lst(x3)).hold().evalf();
+	ex r4 = Li(lst(ex(3)),lst(x3)).hold().evalf();
 	if ( abs(r4 - numeric("0.40068563438653142847-0.95698384815740185713*I")) > prec ) {
 		clog << "Legacy test 3 seems to be wrong." << endl;
 		result++;
@@ -353,10 +353,10 @@
 		+Li(lst(1,1,1,1),lst(x2,x4,x3,x0)).hold().evalf()
 		+Li(lst(1,1,1,1),lst(x2,x3,x4,x0)).hold().evalf()
 		+Li(lst(1,1,1,1),lst(x2,x3,x0,x4)).hold().evalf()
-		+Li(lst(2,2),lst(x5,x4)).hold().evalf()
+		+Li(lst(ex(2),2),lst(x5,x4)).hold().evalf()
 		+Li(lst(2,1,1),lst(x5,x0,x4)).hold().evalf()
 		+Li(lst(2,1,1),lst(x5,x4,x0)).hold().evalf()
-		-Li(lst(1,1),lst(x3,x0)).hold().evalf()*Li(lst(1,1),lst(x2,x4)).hold().evalf();
+		-Li(lst(ex(1),1),lst(x3,x0)).hold().evalf()*Li(lst(ex(1),1),lst(x2,x4)).hold().evalf();
 	if ( abs(r7) > prec ) {
 		clog << "Legacy test 5 seems to be wrong." << endl;
 		result++;
