--- ginac/compiler.h.orig	2016-04-05 15:45:39.000000000 -0500
+++ ginac/compiler.h	2016-05-10 13:50:28.000000000 -0500
@@ -26,11 +26,11 @@
 #ifdef __GNUC__
 #define unlikely(cond) __builtin_expect((cond), 0)
 #define likely(cond) __builtin_expect((cond), 1)
-#define deprecated __attribute__ ((deprecated));
+#define GiNaC_deprecated __attribute__ ((GiNaC_deprecated));
 #else
 #define unlikely(cond) (cond)
 #define likely(cond) (cond)
-#define deprecated
+#define GiNaC_deprecated
 #endif
 
 #ifdef _MSC_VER
