--- tksrc/Makefile.in.orig
+++ tksrc/Makefile.in
@@ -846,7 +846,7 @@ gmtkMMItrain_SOURCES = gmtkMMItrain.cc
 #	$(compile.cc.link) -Wall -g -DMAIN  -o $@ gmtkUnroll.o $(LIBS)
 gmtkViz_SOURCES = gmtkViz.cc
 gmtkViz_CXXFLAGS = $(VERSIONFLAG) $(VIZCXXFLAGS) $(DEBUGFLAGS) $(OPTFLAGS) $(X_CFLAGS) $(EXTRA_WX_FLAGS)
-gmtkViz_LDADD = $(LDADD) $(VIZLDFLAGS) $(X_PRE_LIBS) $(X_LIBS) $(X_EXTRA_LIBS) -lX11 $(OSXLDFLAGS)
+gmtkViz_LDADD = $(LDADD) $(VIZLDFLAGS) $(X_PRE_LIBS) $(X_LIBS) $(X_EXTRA_LIBS) $(OSXLDFLAGS)
 gmtkViz_app_SOURCES = Info.plist.in wxmac.icns
 @MAKE_BUNDLE_TRUE@app_verbose = $(app_verbose_@AM_V@)
 @MAKE_BUNDLE_TRUE@app_verbose_ = $(app_verbose_@AM_DEFAULT_V@)
