--- topinambour-2.0.6.gemspec
+++ topinambour-2.0.6.gemspec	2024-05-19 15:00:46
@@ -23,6 +23,6 @@
 
   s.specification_version = 4
 
-  s.add_runtime_dependency(%q<gtk3>.freeze, ["~> 3.2".freeze, ">= 3.2.6".freeze])
-  s.add_runtime_dependency(%q<vte3>.freeze, ["~> 3.0".freeze, ">= 3.1.0".freeze])
+  s.add_runtime_dependency(%q<gtk3>.freeze, [">= 3.2.6".freeze])
+  s.add_runtime_dependency(%q<vte3>.freeze, [">= 3.1.0".freeze])
 end
