--- topinambour-2.0.6.gemspec
+++ topinambour-2.0.6.gemspec	2024-05-19 15:15:09
@@ -26,10 +26,10 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<gtk3>.freeze, ["~> 3.2", ">= 3.2.6"])
-    s.add_runtime_dependency(%q<vte3>.freeze, ["~> 3.0", ">= 3.1.0"])
+    s.add_runtime_dependency(%q<gtk3>.freeze, [">= 3.2.6"])
+    s.add_runtime_dependency(%q<vte3>.freeze, [">= 3.1.0"])
   else
-    s.add_dependency(%q<gtk3>.freeze, ["~> 3.2", ">= 3.2.6"])
-    s.add_dependency(%q<vte3>.freeze, ["~> 3.0", ">= 3.1.0"])
+    s.add_dependency(%q<gtk3>.freeze, [">= 3.2.6"])
+    s.add_dependency(%q<vte3>.freeze, [">= 3.1.0"])
   end
 end
