--- gcc/Makefile.in.orig	2023-09-27 12:34:24
+++ gcc/Makefile.in	2023-09-27 12:35:59
@@ -3280,7 +3280,7 @@
 	      echo "  ${BUILD_SYSTEM_HEADER_DIR}" >&2 ; \
 	      tooldir_sysinc=`echo "${gcc_tooldir}/sys-include" | sed -e :a -e "s,[^/]*/\.\.\/,," -e ta`; \
 	      if test "x${BUILD_SYSTEM_HEADER_DIR}" = "x$${tooldir_sysinc}"; \
-	      then sleep 1; else exit 1; fi; \
+	      then sleep 1; else exit 0; fi; \
 	    fi; \
 	    $(mkinstalldirs) $${fix_dir}; \
 	    chmod a+rx $${fix_dir} || true; \
