--- configure.ac.orig	2023-01-10 18:01:14.000000000 +0800
+++ configure.ac	2023-06-22 15:13:49.000000000 +0800
@@ -127,7 +127,7 @@
 	AC_DEFINE_UNQUOTED([WD_ROOT], ["${with_fake_prefix}/${wireddir}"], [Root directory])
 else
 	AC_SUBST([fake_prefix], ['${prefix}'])
-	AC_DEFINE_UNQUOTED([WD_ROOT], ["/usr/local/${wireddir}"], [Root directory])
+	AC_DEFINE_UNQUOTED([WD_ROOT], ["__PREFIX__/${wireddir}"], [Root directory])
 fi
 
 
@@ -136,10 +136,8 @@
 
 if test -n "$with_objdir"; then
 	AC_SUBST([objdir], ["$with_objdir"])
-	wd_libwired_objdir="$with_objdir"
 else
 	AC_SUBST([objdir], ["obj"])
-	wd_libwired_objdir="../obj"
 fi
 
 
@@ -148,10 +146,8 @@
 
 if test -n "$with_rundir"; then
 	AC_SUBST([rundir], ["$with_rundir"])
-	wd_libwired_rundir="$with_rundir/libwired"
 else
 	AC_SUBST([rundir], ["run"])
-	wd_libwired_rundir="../run/libwired"
 fi
 
 
@@ -200,19 +196,6 @@
 
 AC_CONFIG_FILES([Makefile])
 
-# Configure libwired
-AC_ARG_WITH([libwired], , [
-	if test "$with_libwired" = "yes"; then
-		wd_libwired=1
-	fi
-], [
-	wd_libwired=1
-])
-
-if test "$wd_libwired"; then
-	ac_configure_args="$ac_configure_args --enable-pthreads --enable-libxml2 --enable-sqlite3 --enable-p7 --with-rundir=$wd_libwired_rundir --with-objdir=$wd_libwired_objdir"
-	AC_CONFIG_SUBDIRS([libwired])
-fi
 
 AC_OUTPUT
 
