--- configure.ac.orig	Sun Feb  8 08:28:16 2004
+++ configure.ac	Sun Feb  8 08:28:47 2004
@@ -345,7 +345,7 @@
 
 	AC_MSG_CHECKING(for $dir/include in INCPATH)
 	if test -d $dir/include 2>/dev/null ; then
-		if [ "$dir" != "/usr/local" ] ; then
+		if [[ "$dir" != "/usr/local" ]] ; then
 		    INCPATH="$INCPATH -I$dir/include"
 		fi
 		AC_MSG_RESULT(yes)
@@ -859,6 +859,18 @@
 WITH_PYTHON_VERSION=$withval
 if test $withval = auto ; then
 
+  AC_MSG_CHECKING(for python 2.3)
+  AC_TRY_RUN([
+#include <python2.3/Python.h>
+main() {
+  exit(strncmp("2.3", PY_VERSION, 3));
+} ],
+  withval=yes, withval=no, withval=yes)
+  AC_MSG_RESULT($withval)
+  if test $withval = yes ; then
+    WITH_PYTHON_VERSION="2.3"
+  else
+
   AC_MSG_CHECKING(for python 2.2)
   AC_TRY_RUN([
 #include <python2.2/Python.h>
@@ -883,7 +895,7 @@
       WITH_PYTHON_VERSION="1.5"
     fi
   fi
-
+  fi
 fi
 
 if test "$WITH_PYTHON_VERSION" != no ; then
