--- rpmrc.in.orig	Tue Mar  4 11:04:50 2003
+++ rpmrc.in	Fri Nov 28 06:54:51 2003
@@ -213,7 +213,7 @@
 arch_compat: i686: i586
 arch_compat: i586: i486
 arch_compat: i486: i386
-arch_compat: i386: noarch
+arch_compat: i386: noarch fat
 
 arch_compat: osfmach3_i686: i686 osfmach3_i586
 arch_compat: osfmach3_i586: i586 osfmach3_i486
@@ -227,7 +227,7 @@
 arch_compat: ppcpseries: ppc
 arch_compat: ppc64: ppc
 arch_compat: ppc: rs6000
-arch_compat: rs6000: noarch
+arch_compat: rs6000: noarch fat
 
 arch_compat: sun4c: sparc
 arch_compat: sun4d: sparc
@@ -299,7 +299,7 @@
 buildarch_compat: i686: i586
 buildarch_compat: i586: i486
 buildarch_compat: i486: i386
-buildarch_compat: i386: noarch
+buildarch_compat: i386: noarch fat
 
 buildarch_compat: sun4c: noarch
 buildarch_compat: sun4d: noarch
@@ -318,10 +318,10 @@
 
 buildarch_compat: m68k: noarch
 
-buildarch_compat: ppciseries: noarch
-buildarch_compat: ppcpseries: noarch
-buildarch_compat: ppc: noarch
-buildarch_compat: ppc64: noarch
+buildarch_compat: ppciseries: noarch fat
+buildarch_compat: ppcpseries: noarch fat
+buildarch_compat: ppc: noarch fat
+buildarch_compat: ppc64: noarch fat
 
 buildarch_compat: mips: noarch
 buildarch_compat: mipsel: noarch
