--- python/Makefile.in.orig	Sun Feb  8 10:32:20 2004
+++ python/Makefile.in	Sun Feb  8 10:33:04 2004
@@ -224,6 +224,7 @@
 	-I$(top_srcdir)/rpmio \
 	-I$(top_srcdir)/beecrypt \
 	-I$(top_srcdir)/popt \
+	-I$(top_srcdir)/misc \
 	-I$(pyincdir) \
 	@WITH_LIBELF_INCLUDE@ \
 	@INCPATH@
@@ -235,30 +236,30 @@
 
 
 mylibs = \
-	$(top_builddir)/lib/.libs/librpm.so \
-	$(top_builddir)/rpmdb/.libs/librpmdb.so \
-	$(top_builddir)/rpmio/.libs/librpmio.so \
-	$(top_builddir)/popt/.libs/libpopt.so
+	$(top_builddir)/lib/librpm.la \
+	$(top_builddir)/rpmdb/librpmdb.la \
+	$(top_builddir)/rpmio/librpmio.la \
+	$(top_builddir)/popt/libpopt.la
 
 
 LDADD = 
 
 pythondir = $(pylibdir)/site-packages
-python_PROGRAMS = rpmmodule.so
+python_LTLIBRARIES = rpmmodule.so
 
 rpmdbdir = $(pylibdir)/site-packages/rpmdb
-rpmdb_PROGRAMS = _rpmdb.so
+rpmdb_LTLIBRARIES = _rpmdb.so
 
 noinst_PROGRAMS = poptmodule.so
 
 rpmmodule_so_SOURCES = 
-rpmmodule_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,rpmmodule.so
+rpmmodule_so_LDFLAGS = $(mylibs) $(LIBS) -bundle -bundle_loader $(__PYTHON)
 
 _rpmdb_so_SOURCES = _rpmdb.c
-_rpmdb_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,_rpmdb.so
+_rpmdb_so_LDFLAGS = $(mylibs) $(LIBS) -bundle -bundle_loader $(__PYTHON)
 
 poptmodule_so_SOURCES = poptmodule.c
-poptmodule_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,poptmodule.so
+poptmodule_so_LDFLAGS = $(mylibs) $(LIBS) -bundle -bundle_loader $(__PYTHON)
 
 noinst_LTLIBRARIES = librpmmodule.la
 librpmmodule_la_SOURCES = rpmmodule.c header-py.c \
@@ -707,13 +708,13 @@
 
 
 rpmmodule.so$(EXEEXT): $(librpmmodule_la_OBJECTS)
-	$(CC) -o $@ $(librpmmodule_la_OBJECTS) $(rpmmodule_so_LDFLAGS)
+	$(LINK) $(librpmmodule_la_OBJECTS) $(rpmmodule_so_LDFLAGS)
 
 _rpmdb.so$(EXEEXT): _rpmdb.lo
-	$(CC) -o $@ _rpmdb.lo $(_rpmdb_so_LDFLAGS)
+	$(LINK) _rpmdb.lo $(_rpmdb_so_LDFLAGS)
 
 poptmodule.so$(EXEEXT): poptmodule.lo
-	$(CC) -o $@ poptmodule.lo $(poptmodule_so_LDFLAGS)
+	$(LINK) poptmodule.lo $(poptmodule_so_LDFLAGS)
 
 .PHONY:	lint
 lint:
