--- rpmio/rpmrpc.c.orig	Sat Nov  8 20:58:20 2003
+++ rpmio/rpmrpc.c	Sat Nov  8 22:15:26 2003
@@ -1102,7 +1102,7 @@
 # define DT_WHT		14
 typedef struct __dirstream *	FTPDIR;
 #else
-typedef DIR *			FTPDIR;
+typedef struct __dirstream *	FTPDIR;
 #endif
 
 /*@unchecked@*/
@@ -1287,7 +1287,9 @@
     dp->d_reclen = 0;		/* W2DO? */
 
 #if !defined(hpux) && !defined(sun)
+#if 0
     dp->d_off = 0;		/* W2DO? */
+#endif
 /*@-boundsread@*/
     dp->d_type = dt[i];
 /*@=boundsread@*/
