--- configure.orig	2008-01-10 23:34:23.000000000 +1100
+++ configure	2008-01-26 15:32:17.000000000 +1100
@@ -25837,7 +25837,12 @@
     if (getrlimit(RLIMIT_NOFILE, &rl) < 0) {
 	perror("getrlimit: RLIMIT_NOFILE");
     } else {
+#if defined(__APPLE__)
+	/* asking for more than OPEN_MAX fails on Leopard */
+	rl.rlim_cur = (OPEN_MAX < rl.rlim_max ? OPEN_MAX : rl.rlim_max);
+#else
 	rl.rlim_cur = rl.rlim_max;      /* set it to the max */
+#endif
 	if (setrlimit(RLIMIT_NOFILE, &rl) < 0) {
 	    perror("setrlimit: RLIMIT_NOFILE");
 	}
