--- configure.orig	2011-08-24 16:29:48.000000000 -0500
+++ configure	2011-09-13 23:06:28.000000000 -0500
@@ -20011,7 +20011,7 @@
 		echo "   (note that newer gcc 4.x versions do not use/need the align check)"
 	fi
 fi
-echo -n "
+echo "
   Core libmpg123 features:
   Integer conversion ...... $integers
   Layer I ................. $layer1
@@ -20032,8 +20032,6 @@
   32/24 bit integer ....... $int32
   real (32 bit float) ..... $real
 "
-# just an empty line
-echo
 
 echo "  Modules ................. $modules"
 echo "  Module suffix ........... $with_module_suffix"
