diff --git xvjp2k.c xvjp2k.c
index da763fc..41b5554 100644
--- xvjp2k.c
+++ xvjp2k.c
@@ -76,7 +76,7 @@ static const char *fbasename,  /* File's base name, for error/warning msgs */
 */
 int jas_getdbglevel(void) {return 0;}
 int jas_setdbglevel(int n) {return 0;}
-int jas_memdump(FILE *fp,void *data,size_t len) {return 0;}
+int jas_memdump(FILE *fp,const void *data,size_t len) {return 0;}
 
 int jas_eprintf(const char *fmt,...)         /* Handle JasPer Library message */
 {
