--- test/ttest.sh	2019-07-01 12:37:08.000000000 -0400
+++ test/ttest.sh	2019-07-03 22:14:14.000000000 -0400
@@ -5,7 +5,7 @@
 
 check_opts=('' '-r' '-v' '-D' '-S')
 
-shc=${1-shc}
+shc=./src/shc
 
 txtred='\e[0;31m' # Red
 txtgrn='\e[0;32m' # Green
@@ -17,6 +17,9 @@
 echo
 echo "== Running tests ..."
 for shell in ${shells[@]}; do
+    if ! [[ -x "$shell" ]]; then
+        continue
+    fi
     for opt in "${check_opts[@]}"; do
         tmpd=$(mktemp -d)
         tmpf="$tmpd/test.$(basename $shell)"
