https://github.com/mde/jake/issues/85
--- Makefile.orig	2012-01-15 20:52:06.000000000 -0600
+++ Makefile	2012-01-18 06:03:58.000000000 -0600
@@ -29,7 +29,7 @@
 	@mkdir -p $(DESTDIR)$(PREFIX)/bin \
     mkdir -p $(DESTDIR)$(PREFIX)/lib/node_modules/jake && \
 		cp -R ./* $(DESTDIR)$(PREFIX)/lib/node_modules/jake/ && \
-		ln -snf $(DESTDIR)$(PREFIX)/lib/node_modules/jake/bin/cli.js $(DESTDIR)$(PREFIX)/bin/jake && \
+		ln -snf ../lib/node_modules/jake/bin/cli.js $(DESTDIR)$(PREFIX)/bin/jake && \
 		chmod 755 $(DESTDIR)$(PREFIX)/lib/node_modules/jake/bin/cli.js && \
 		echo 'Jake installed.'
 
