--- src/tap/Makefile.orig
+++ src/tap/Makefile
@@ -22,15 +22,15 @@ BUNDLE_VERSION = $(TAP_KEXT_VERSION)
 INCLUDE = -I.. -I/System/Library/Frameworks/Kernel.framework/Headers
 CFLAGS = -Wall -Werror -mkernel -force_cpusubtype_ALL \
 	-nostdinc -fno-builtin -fno-stack-protector -msoft-float -fno-common \
-	-arch x86_64 \
+	@ARCHFLAGS@ \
 	-DKERNEL -DAPPLE -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTAP_KEXT_VERSION=\"$(TAP_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS)
-LDFLAGS = -Wall -Werror -arch x86_64 -Xlinker -kext -nostdlib -lkmodc++ -lkmod -lcc_kext
+LDFLAGS = -Wall -Werror @ARCHFLAGS@ -Xlinker -kext -nostdlib -lkmodc++ -lkmod -lcc_kext
 
-CCP = clang -x c++
-CC = clang -x c
-LD = clang
+CCP = @CXX@
+CC = @CC@
+LD = @CC@
 
 all: $(KMOD_BIN) bundle
 
