--- src/tap/Makefile.orig
+++ src/tap/Makefile
@@ -26,7 +26,7 @@ CFLAGS = -Wall -Werror -mkernel -force_cpusubtype_ALL \
 	-DKERNEL -DAPPLE -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTAP_KEXT_VERSION=\"$(TAP_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS)
-LDFLAGS = -Wall -Werror @ARCHFLAGS@ -Xlinker -kext -nostdlib -lkmodc++ -lkmod -lcc_kext
+LDFLAGS = -Wall -Werror @ARCHFLAGS@ -nostdlib -lkmodc++ -lkmod -lcc_kext
 
 CCP = @CXX@
 CC = @CC@
--- src/tun/Makefile.orig
+++ src/tun/Makefile
@@ -27,7 +27,7 @@ CFLAGS = -Wall -Werror -mkernel -force_cpusubtype_ALL \
 	-DKERNEL -DAPPLE -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTUN_KEXT_VERSION=\"$(TUN_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS)
-LDFLAGS = -Wall -Werror @ARCHFLAGS@ -Xlinker -kext -nostdlib -lkmodc++ -lkmod -lcc_kext
+LDFLAGS = -Wall -Werror @ARCHFLAGS@ -nostdlib -lkmodc++ -lkmod -lcc_kext
 
 CCP = @CXX@
 CC = @CC@
--- src/tuntap.cc.orig
+++ src/tuntap.cc
@@ -75,7 +75,7 @@ tuntap_if_output(ifnet_t ifp, mbuf_t m)
 }
 
 errno_t
-tuntap_if_ioctl(ifnet_t ifp, long unsigned int cmd, void *arg)
+tuntap_if_ioctl(ifnet_t ifp, uint32_t cmd, void *arg)
 {
 	if (ifp != NULL) {
 		tuntap_interface *ttif = (tuntap_interface *) ifnet_softc(ifp);
--- src/tuntap.h.orig
+++ src/tuntap.h
@@ -53,7 +53,7 @@ extern "C" {
 extern "C" {
 
 errno_t tuntap_if_output(ifnet_t ifp, mbuf_t m);
-errno_t tuntap_if_ioctl(ifnet_t ifp, long unsigned int cmd, void *arg);
+errno_t tuntap_if_ioctl(ifnet_t ifp, uint32_t cmd, void *arg);
 errno_t tuntap_if_set_bpf_tap(ifnet_t ifp, bpf_tap_mode mode, int (*cb)(ifnet_t, mbuf_t));
 errno_t tuntap_if_demux(ifnet_t ifp, mbuf_t m, char *header, protocol_family_t *proto);
 errno_t tuntap_if_framer(ifnet_t ifp, mbuf_t *m, const struct sockaddr *dest,
@@ -268,7 +268,7 @@ class tuntap_interface {
 
 		/* interface functions. friends and implementation methods */
 		friend errno_t tuntap_if_output(ifnet_t ifp, mbuf_t m);
-		friend errno_t tuntap_if_ioctl(ifnet_t ifp, long unsigned int cmd, void *arg);
+		friend errno_t tuntap_if_ioctl(ifnet_t ifp, uint32_t cmd, void *arg);
 		friend errno_t tuntap_if_set_bpf_tap(ifnet_t ifp, bpf_tap_mode mode,
 				int (*cb)(ifnet_t, mbuf_t));
 		friend errno_t tuntap_if_demux(ifnet_t ifp, mbuf_t m, char *header,
