--- tests/test_black.py
+++ tests/test_black.py
@@ -1934,8 +1934,8 @@ class BlackTestCase(unittest.TestCase):
             self.assertEqual(black.find_project_root((src_dir,)), src_dir.resolve())
             self.assertEqual(black.find_project_root((src_python,)), src_dir.resolve())
 
-
-class BlackDTestCase(AioHTTPTestCase):
+if has_blackd_deps:
+  class BlackDTestCase(AioHTTPTestCase):
     async def get_application(self) -> web.Application:
         return blackd.make_app()
 
