--- stm32.c.orig	2015-01-08 13:12:24.000000000 +0100
+++ stm32.c	2015-01-08 13:13:48.000000000 +0100
@@ -604,7 +604,7 @@
 		return STM32_ERR_UNKNOWN;
 
 	s_err = stm32_get_ack_timeout(stm, STM32_WUNPROT_TIMEOUT);
-	if (s_err == STM32_NACK) {
+	if (s_err == STM32_ERR_NACK) {
 		fprintf(stderr, "Error: Failed to WRITE UNPROTECT\n");
 		return STM32_ERR_UNKNOWN;
 	}
@@ -631,7 +631,7 @@
 		return STM32_ERR_UNKNOWN;
 
 	s_err = stm32_get_ack_timeout(stm, STM32_WPROT_TIMEOUT);
-	if (s_err == STM32_NACK) {
+	if (s_err == STM32_ERR_NACK) {
 		fprintf(stderr, "Error: Failed to WRITE PROTECT\n");
 		return STM32_ERR_UNKNOWN;
 	}
@@ -658,7 +658,7 @@
 		return STM32_ERR_UNKNOWN;
 
 	s_err = stm32_get_ack_timeout(stm, STM32_MASSERASE_TIMEOUT);
-	if (s_err == STM32_NACK) {
+	if (s_err == STM32_ERR_NACK) {
 		fprintf(stderr, "Error: Failed to READOUT UNPROTECT\n");
 		return STM32_ERR_UNKNOWN;
 	}
@@ -685,7 +685,7 @@
 		return STM32_ERR_UNKNOWN;
 
 	s_err = stm32_get_ack_timeout(stm, STM32_RPROT_TIMEOUT);
-	if (s_err == STM32_NACK) {
+	if (s_err == STM32_ERR_NACK) {
 		fprintf(stderr, "Error: Failed to READOUT PROTECT\n");
 		return STM32_ERR_UNKNOWN;
 	}
