--- dot_parser.py.orig	2012-01-03 08:23:26.000000000 +0900
+++ dot_parser.py	2014-02-14 00:39:04.000000000 +0900
@@ -25,8 +25,9 @@
 from pyparsing import ( nestedExpr, Literal, CaselessLiteral, Word, Upcase, OneOrMore, ZeroOrMore,
     Forward, NotAny, delimitedList, oneOf, Group, Optional, Combine, alphas, nums,
     restOfLine, cStyleComment, nums, alphanums, printables, empty, quotedString,
-    ParseException, ParseResults, CharsNotIn, _noncomma, dblQuotedString, QuotedString, ParserElement )
+    ParseException, ParseResults, CharsNotIn, dblQuotedString, QuotedString, ParserElement )
 
+_noncomma = "".join( [ c for c in printables if c != "," ] )
 
 class P_AttrList:
 
