--- configure.ac.orig	2023-03-22 03:51:54.000000000 -0500
+++ configure.ac	2023-03-22 05:52:32.000000000 -0500
@@ -55,11 +55,7 @@
     CFLAGS="$CFLAGS -DUSE_NO_PROC"
   fi],
   [case $host_os in
-    darwin*) CFLAGS="$CFLAGS -I/opt/local/include"
-             # Starting with Mac OS X 10.11 (El Capitan) the OpenSSL headers
-             # are in /usr/local/opt/openssl/include
-             CFLAGS="$CFLAGS -DUSE_MACH_PROC -I/usr/local/opt/openssl/include"
-             LDFLAGS="$LDFLAGS -L/usr/local/opt/openssl/lib"
+    darwin*) CFLAGS="$CFLAGS -DUSE_MACH_PROC"
              posix=true
              proc_interface=mach
              jemalloc_prefix=je_ ;;
