https://github.com/egallesio/STklos/pull/512
--- utils/shlib-options.orig	2021-05-03 05:02:46.000000000 -0500
+++ utils/shlib-options	2023-03-15 18:02:56.000000000 -0500
@@ -151,7 +151,7 @@
   Darwin*)
       OS=DARWIN;
           SH_COMP_FLAGS="-fPIC -fno-common"
-      SH_LOAD_FLAGS='-bundle -flat_namespace -undefined suppress -o'
+      SH_LOAD_FLAGS='-bundle -undefined dynamic_lookup -o'
           SH_LOADER="$CC"
           SH_SUFFIX='so'
           SH_LIB_SUFFIX='dylib'
