--- configure	2008-05-03 22:43:30.000000000 +0200
+++ configure	2009-01-12 19:26:02.000000000 +0100
@@ -6731,7 +6731,7 @@
 if test "$HAVE_GC" = "no"
 then
   echo "No Boehm-Demers-Weiser GC library. Using the one provided with the package"
-  (cd gc; CC="$CC" CFLAGS="$CFLAGS" sh ./configure --prefix=$prefix \
+  (cd gc; CC="$CC" CFLAGS="$CFLAGS" sh ./configure --prefix=$prefix --libdir=$libdir \
           --includedir=$prefix/include/stklos \
           --enable-threads=$THREADS) || { echo "Cannot configure the GC"; exit; }
 
@@ -6793,7 +6793,7 @@
   echo "Configuring libffi"
   (cd ffi; CC=${CC} CFLAGS=${CFLAGS} ./configure \
            --disable-structs --disable-raw-api --disable-shared \
-	   --prefix=$prefix)
+	   --prefix=$prefix --libdir=$libdir)
   LIBS="$LIBS ../ffi/.libs/libffi.a"
   CFLAGS="$CFLAGS -I../ffi/include"
   LIBFFI="ffi"
