diff -rwub postgresql-7.2.1/Makefile postgresql-7.2.1-new/Makefile
--- Makefile	Fri Feb  9 18:31:25 2001
+++ Makefile	Mon Sep  2 05:37:22 2002
@@ -11,7 +11,7 @@
 # GNUmakefile won't exist yet, so we catch that case as well.
 
 
-all check install installdirs install-all-headers installcheck uninstall dep depend clean distclean maintainer-clean:
+.PHONY all check install installdirs install-all-headers installcheck uninstall dep depend clean distclean maintainer-clean:
 	@if [ ! -f GNUmakefile ] ; then \
 	   echo "You need to run the 'configure' program first. See the file"; \
 	   echo "'INSTALL' for installation instructions." ; \
