--- magics-config.in.orig	2009-12-10 15:45:26.000000000 -1000
+++ magics-config.in	2009-12-10 16:08:07.000000000 -1000
@@ -6,6 +6,8 @@
 includedir=@includedir@
 F77="@F77@"
 CC="@CC@"
+CXX="@CXX@"
+CXXFLAGS="@CXXFLAGS@"
 AXX="@AXX@"
 FFLAGS="@FFLAGS@"
 CPPLIBS="@CPPLIBS@"
@@ -154,7 +156,7 @@
 			# Intel compiler
 			# Portland pgf90
 			# IBM fortran compiler
-			ifc | ifort | pgf90 | pgf77 | xlf | xlf90 | xlf_r | xlf90_r )
+			ifc | ifort | pgf90 | pgf77 | xlf | xlf90 | xlf_r | xlf90_r | g95 )
 				FXX="-r8 ${FXX}"
 				;;
 		esac
@@ -179,10 +181,10 @@
 
 		case "${CC}" in
 			icc | icpc | gcc | g++ )
-				CXX="-m64 ${CXX}"
+				CXXFLAGS="-m64 ${CXXFLAGS}"
 				;;
 			pgc | pgcpp )
-				CXX="-tp amd64 ${CXX}"
+				CXXFLAGS="-tp amd64 ${CXXFLAGS}"
 				;;
 		esac
 		;; 
@@ -205,7 +207,7 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-	if test -f ${libdir}/libMagPlus.so ; then
+	if test -f ${libdir}/libMagPlus.dylib ; then
 		my_libs="-L${libdir} -lMagPlus"
 	else
 		my_libs="${libdir}/libMagPlus.a"
@@ -237,7 +239,7 @@
 fi
 
 if test "$echo_shared" = "yes"; then
-	if test -f ${libdir}/libMagPlus.so ; then
+	if test -f ${libdir}/libMagPlus.dylib ; then
 	  shared="${precision} -L${libdir} -lMagPlus @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
 	  echo ${shared}
 	else
@@ -252,16 +254,17 @@
 if test "$compile" = "yes"; then
 	out=""
 	if test "${suffix}x" = "x"; then
-	   suffix="f"
+	   suffix="o"
 	fi
 
-	name="`basename $f77_file .${suffix}`"
+	name=${f77_file%.*}
 
 	if test ${name} != ${f77_file} ; then
 	      out="-o $name "              ### avoid overriding source file
 	fi
-	$F77 ${out}$f77_file ${FFLAGS} ${precision} -L${libdir} -Wl,-rpath,${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS
-	echo "$F77 ${out}$f77_file ${FFLAGS} ${precision} -L${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
+	$F77 ${f77_file} -c && \
+	${CXX} ${CXXFLAGS} ${out}${name}.o ${FFLAGS} ${precision} -L${libdir} -Wl,-rpath,${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS
+	echo "$F77 ${f77_file} -c && ${CXX} ${CXXFLAGS} ${out}${name}.o ${FFLAGS} ${precision} -L${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
 fi
 
 if test "$compileC" = "yes"; then
