--- src/drivers/CairoDriver.cc.orig	2015-12-15 18:33:04.000000000 +0900
+++ src/drivers/CairoDriver.cc	2016-01-31 15:10:53.000000000 +0900
@@ -539,6 +539,7 @@
 
     // DPI
     TIFFSetField(tif, TIFFTAG_RESOLUTIONUNIT, RESUNIT_INCH);
+    const double resolution_ = 72.;
     TIFFSetField(tif, TIFFTAG_XRESOLUTION, (float) resolution_);
     TIFFSetField(tif, TIFFTAG_YRESOLUTION, (float) resolution_);
 
