--- CMakeLists.txt.orig	2020-10-15 20:46:30.000000000 +0900
+++ CMakeLists.txt	2020-11-07 10:06:18.000000000 +0900
@@ -157,7 +157,7 @@
 find_package( EXPAT REQUIRED )   # for MagML
 
 if(APPLE)
-  find_package( zlib REQUIRED )  # dependency of minizip - KMZ support
+  find_package( ZLIB REQUIRED )  # dependency of minizip - KMZ support
 endif(APPLE)
 
 ###############################################################################
