--- magics-config.in.orig	2010-08-29 21:59:57.000000000 +0900
+++ magics-config.in	2010-08-29 22:05:06.000000000 +0900
@@ -6,6 +6,8 @@
 includedir=@includedir@
 F77="@F77@"
 CC="@CC@"
+CXX="@CXX@"
+CXXFLAGS="@CXXFLAGS@"
 AXX="@AXX@"
 FFLAGS="@FFLAGS@"
 CPPLIBS="@CPPLIBS@"
@@ -200,7 +202,7 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-	if test -f ${libdir}/libMagPlus.so ; then
+	if test -f ${libdir}/libMagPlus.dylib ; then
 		my_libs="-L${libdir} -lMagPlus"
 	else
 		my_libs="${libdir}/libMagPlus.a"
@@ -232,7 +234,7 @@
 fi
 
 if test "$echo_shared" = "yes"; then
-	if test -f ${libdir}/libMagPlus.so ; then
+	if test -f ${libdir}/libMagPlus.dylib ; then
 	  shared="${precision} -L${libdir} -lMagPlus @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
 	  echo ${shared}
 	else
@@ -247,16 +249,17 @@
 if test "$compile" = "yes"; then
 	out=""
 	if test "${suffix}x" = "x"; then
-	   suffix="f"
+	   suffix="o"
 	fi
 
-	name="`basename $f77_file .${suffix}`"
+        name=${f77_file%.*}
 
 	if test ${name} != ${f77_file} ; then
 	      out="-o $name "              ### avoid overriding source file
 	fi
-	$F77 ${out}$f77_file ${FFLAGS} ${precision} -L${libdir} -Wl,-rpath,${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS
-	echo "$F77 ${out}$f77_file ${FFLAGS} ${precision} -L${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
+       $F77 ${f77_file} -c && \
+       ${CXX} ${CXXFLAGS} ${out}${name}.o ${FFLAGS} ${precision} -L${libdir} -Wl,-rpath,${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS
+       echo "$F77 ${f77_file} -c && ${CXX} ${CXXFLAGS} ${out}${name}.o ${FFLAGS} ${precision} -L${libdir} -lMagPlus @LDFLAGS@ @MAGICS_3RDPARTY_LIBS@ $CPPLIBS"
 fi
 
 if test "$compileC" = "yes"; then
