--- setup.py.orig	2022-01-15 14:15:49.000000000 -0500
+++ setup.py	2022-01-15 14:17:30.000000000 -0500
@@ -15,7 +15,7 @@
     author_email='harry.jubb@sanger.ac.uk',
     license='GNU General Public License v3.0',
     keywords='arpeggio PDB mmCIF protein ligand interactions CREDO',
-    packages=find_namespace_packages(),
+    packages=find_namespace_packages(exclude=['build.*']),
     scripts=['bin/arpeggio'],
     classifiers=[
         "Programming Language :: Python",
