diff -ur ../khangman-4.14.3-orig/src/desktop/khangmanview.cpp ./src/desktop/khangmanview.cpp
--- ../khangman-4.14.3-orig/src/desktop/khangmanview.cpp	2014-08-27 14:30:36.000000000 +0900
+++ ./src/desktop/khangmanview.cpp	2020-04-15 23:02:38.000000000 +0900
@@ -510,7 +510,7 @@
 
         int  x = 0;
         int  y = 0;
-        if (m_missedLetters.contains(guess) > 0) {
+        if (m_missedLetters.contains(guess) != NULL) {
             // FIXME: popup should be better placed.
 
             QPoint abspos = popup->pos();
@@ -527,7 +527,7 @@
             m_letterInput->setEnabled(false);
         }
 
-        if (goodWord.contains(guess) > 0) {
+        if (goodWord.contains(guess) != NULL) {
             point = m_theme->goodWordPos(size(), popup->pos());
             
             QTimer::singleShot( Prefs::missedTimer() * 1000,
