--- Makefile.orig	2018-04-30 11:44:19.000000000 -0300
+++ Makefile	2018-04-30 11:48:17.000000000 -0300
@@ -131,14 +131,14 @@
 	ctags $^
 
 install: library
-	$(INSTALL) -D  http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
-	$(INSTALL) -D $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
+	$(INSTALL) -c  http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
+	$(INSTALL) -c $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	ln -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	ln -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SOLIBNAME).$(SOEXT)
 
 install-strip: library
-	$(INSTALL) -D  http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
-	$(INSTALL) -D -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
+	$(INSTALL) -c  http_parser.h $(DESTDIR)$(INCLUDEDIR)/http_parser.h
+	$(INSTALL) -c -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(LIBNAME)
 	ln -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SONAME)
 	ln -s $(LIBNAME) $(DESTDIR)$(LIBDIR)/$(SOLIBNAME).$(SOEXT)
 
