--- dev/CMakeLists.txt.orig	2021-11-19 07:33:54.000000000 -0600
+++ dev/CMakeLists.txt	2022-01-07 12:09:03.000000000 -0600
@@ -132,65 +127,6 @@
 	IF (RESTINIO_USE_EXTERNAL_VARIANT_LITE)
 		find_package(variant-lite REQUIRED)
 	ENDIF()
-
-	# ------------------------------------------------------------------------------
-
-	# ------------------------------------------------------------------------------
-	# Tests, samples and benchmark dependencies:
-	# OpenSSL
-	find_package(OpenSSL)
-	IF ( OPENSSL_FOUND )
-		message("OpenSSL include dir: ${OPENSSL_INCLUDE_DIR}")
-		message("OpenSSL libraries: ${OPENSSL_LIBRARIES}")
-	ENDIF ( OPENSSL_FOUND )
-
-	# PCRE
-	find_package(PCRE)
-	IF (PCRE_FOUND)
-		message( STATUS "PCRE_LIBRARIES='" ${PCRE_LIBRARIES} "'" )
-		message( STATUS "PCRE_INCLUDE_DIRS='" ${PCRE_INCLUDE_DIRS} "'" )
-	ENDIF ()
-
-	# PCRE2
-	find_package(PCRE2)
-	IF (PCRE2_FOUND)
-		message( STATUS "PCRE2_LIBRARIES='" ${PCRE2_LIBRARIES} "'" )
-		message( STATUS "PCRE2_INCLUDE_DIRS='" ${PCRE2_INCLUDE_DIRS} "'" )
-	ENDIF ()
-
-	# PCRE2
-	find_package(PCRE2)
-	IF (PCRE2_FOUND)
-		message( STATUS "PCRE2_LIBRARIES='" ${PCRE2_LIBRARIES} "'" )
-		message( STATUS "PCRE2_INCLUDE_DIRS='" ${PCRE2_INCLUDE_DIRS} "'" )
-	ENDIF ()
-
-	# ------------------------------------------------------------------------------
-	# SObjectizer
-	IF (RESTINIO_SOBJECTIZER_ENABLED)
-		IF (RESTINIO_USE_EXTERNAL_SOBJECTIZER)
-			find_package(sobjectizer)
-		ELSE ()
-			SET(SOBJECTIZER_BUILD_STATIC ON)
-			add_subdirectory(so_5)
-		ENDIF ()
-		SET(SOBJECTIZER_LIBS sobjectizer::StaticLib)
-	ENDIF()
-
-
-	# ------------------------------------------------------------------------------
-	# Zlib
-	find_package(ZLIB)
-
-	IF (NOT ZLIB_FOUND)
-		add_subdirectory(restinio/third_party/zlib)
-		message( STATUS "USE OWN ZLIB SOURCES")
-		SET(ZLIB_LIBRARIES zlibstatic)
-		SET(ZLIB_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/restinio/third_party/zlib)
-	ENDIF ()
-
-	message( STATUS "ZLIB_LIBRARIES='" ${ZLIB_LIBRARIES} "'" )
-	message( STATUS "ZLIB_INCLUDE_DIRS='" ${ZLIB_INCLUDE_DIRS} "'" )
 ENDIF (RESTINIO_MASTER_PROJECT)
 
 # RESTinio itself.
