--- dev/nodejs/http_parser/CMakeLists.txt.orig	2018-04-30 13:04:01.000000000 -0300
+++ dev/nodejs/http_parser/CMakeLists.txt	2018-04-30 13:08:03.000000000 -0300
@@ -1,6 +1,3 @@
-project(nodejs_http_parser)
-
-add_library(nodejs_http_parser http_parser.c)
-
-target_include_directories(nodejs_http_parser PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-
+project(nodejs_http_parser)
+
+add_library(libhttp_parser SHARED IMPORTED)
\ No newline at end of file
