--- run-dnn.sh.orig	2014-01-30 20:18:17.000000000 +0900
+++ run-dnn.sh	2014-02-02 23:44:48.000000000 +0900
@@ -2,8 +2,8 @@
 
 ./bin/julius -C main.jconf -C am-dnn.jconf -demo $* &
 sleep 10
-xterm -e python ./bin/dnnclient.py dnnclient.conf &
+osascript -e 'tell application "Terminal" to do script "python ./bin/dnnclient.py dnnclient.conf &"'
 sleep 2
-xterm -e ./bin/adintool -in mic -out vecnet -server 127.0.0.1 -paramtype FBANK_D_A_Z -veclen 120 -htkconf model/dnn/config.lmfb.40ch.jnas -port 5532 -cvn -cmnload model/dnn/norm.jnas
+osascript -e 'tell application "Terminal" to do script "./bin/adintool -in mic -out vecnet -server 127.0.0.1 -paramtype FBANK_D_A_Z -veclen 120 -htkconf model/dnn/config.lmfb.40ch.jnas -port 5532 -cvn -cmnload model/dnn/norm.jnas"'
 
 kill 0
