--- src/gui/menu_osx.mm.orig	2020-08-28 22:30:31.000000000 -0700
+++ src/gui/menu_osx.mm	2020-08-28 22:31:44.000000000 -0700
@@ -9,7 +9,9 @@
 #include "SDL_syswm.h"
 
 #if DOSBOXMENU_TYPE == DOSBOXMENU_NSMENU /* Mac OS X NSMenu / NSMenuItem handle */
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
 # include <MacTypes.h>
+#endif
 # include <Cocoa/Cocoa.h>
 # include <Foundation/NSString.h>
 # include <ApplicationServices/ApplicationServices.h>
