--- src/dos/drive_local.cpp.orig	2021-10-12 22:39:22.000000000 -0700
+++ src/dos/drive_local.cpp	2021-10-12 22:41:38.000000000 -0700
@@ -107,6 +107,16 @@
 #define MAX_PATH PATH_MAX
 #endif
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 101300
+#include <sys/time.h>
+int futimens(int filedes, const struct timespec ftsp[2]) {
+    struct timeval ftvp[2];
+    TIMESPEC_TO_TIMEVAL(ftvp, ftsp);
+
+    return (futimes(filedes, ftvp));
+}
+#endif
+
 #if defined(WIN32)
 // Windows: Use UTF-16 (wide char)
 // TODO: Offer an option to NOT use wide char on Windows if directed by config.h
