--- configure.ac.orig	2025-03-28 00:00:00.000000000 +0000
+++ configure.ac	2025-03-28 00:00:00.000000000 +0000
@@ -674,18 +674,23 @@ AC_MSG_CHECKING([for Metal support])
 case "$host_os" in
   darwin*)
     AC_LANG_PUSH([Objective C++])

+    saved_LIBS="$LIBS"
+    LIBS="$LIBS -framework Metal -framework QuartzCore"
+
     AC_LINK_IFELSE(
       [AC_LANG_PROGRAM([[
         #import <Metal/Metal.h>
       ]], [[
         id<MTLDevice> device = MTLCreateSystemDefaultDevice();
         return (device == nil);
       ]])],
       [have_metal=yes],
       [have_metal=no]
     )
+
+    LIBS="$saved_LIBS"

     AC_LANG_POP
     ;;
 esac
