--- ./src/gui/menu_osx.mm.orig	2020-09-02 23:45:46.000000000 -0700
+++ ./src/gui/menu_osx.mm	2020-09-02 23:46:11.000000000 -0700
@@ -517,7 +517,7 @@
             pt.y = (prct.origin.y + prct.size.height) - pt.y;
         }
 
-        err = CGGetDisplaysWithPoint(pt,1,&did,&cnt);
+        err = CGGetDisplaysWithPoint(CGPointMake(pt.x,pt.y),1,&did,&cnt);
 
         /* This might happen if our window is so far off the screen that the center point does not match any monitor */
         if (err != kCGErrorSuccess) {
