--- loudmouth/lm-sock.c.orig	2014-05-05 11:50:37.000000000 +0900
+++ loudmouth/lm-sock.c	2014-05-05 11:50:49.000000000 +0900
@@ -325,10 +325,10 @@
 		return FALSE;
 	}
 
-	opt = delay;
+/*	opt = delay;
 	if (setsockopt (sock, IPPROTO_TCP, TCP_KEEPIDLE, &opt, sizeof (opt)) < 0) {
 		return FALSE;
-	}
+	}*/
 
 	opt = delay; 
 	if (setsockopt (sock, IPPROTO_TCP, TCP_KEEPINTVL, &opt, sizeof (opt)) < 0) {
