--- type-level.cabal.sav	2010-03-18 10:53:56.000000000 -0400
+++ type-level.cabal	2010-03-18 10:56:13.000000000 -0400
@@ -43,7 +43,7 @@
 
 -- depends on ghc due to the use of infix type constructors and template haskell
 Library
-  build-depends:   base >=4 && <6, template-haskell > 2.0, syb
+  build-depends:   base >=4 && <6, template-haskell > 2.0, syb == 0.1.0.1
   hs-source-dirs:  src
   exposed-modules: Data.TypeLevel,
                    Data.TypeLevel.Bool,
