From da4ba02ea7a55fe8178a1a0d9280ce0b0305ec2a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 10 Mar 2024 01:17:32 +0700
Subject: [PATCH] CMakeLists: do not use Linux linker flag on macOS

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 488b464..f00823c 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -27,7 +27,7 @@ if(FAST_DOUBLE_PARSER_SANITIZE)
   target_compile_options(unit PUBLIC -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
   target_link_options(unit PUBLIC -fsanitize=address -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all)
   # Ubuntu bug for GCC 5.0+ (safe for all versions)
-  if (CMAKE_COMPILER_IS_GNUCC)
+  if (CMAKE_COMPILER_IS_GNUCC AND NOT APPLE)
     target_link_libraries(unit PUBLIC -fuse-ld=gold)
   endif()
 endif()
