--- configure.pl.orig	2020-02-17 11:40:13.000000000 -0800
+++ configure.pl	2020-05-08 22:36:34.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!@@PERL_FULL@@
 #============================================================= -*-perl-*-
 #
 # configure.pl: Configuration and installation program for BackupPC
@@ -363,9 +363,9 @@
 # These are the programs whose paths we need to find
 #
 my %Programs = (
-    perl           => "PerlPath",
+    '@@PERL@@'     => "PerlPath",
     rsync_bpc      => "RsyncBackupPCPath",
-    'gtar/tar'     => "TarClientPath",
+    gnutar         => "TarClientPath",
     smbclient      => "SmbClientPath",
     nmblookup      => "NmbLookupPath",
     rsync          => "RsyncClientPath",
