--- configure.pl.orig
+++ configure.pl
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!@@PERL_FULL@@
 #============================================================= -*-perl-*-
 #
 # configure.pl: Configuration and installation program for BackupPC
@@ -247,8 +247,8 @@ if ( $opts{fhs} ) {
 # These are the programs whose paths we need to find
 #
 my %Programs = (
-    perl           => "PerlPath",
-    'gtar/tar'     => "TarClientPath",
+    '@@PERL@@'     => "PerlPath",
+    gnutar         => "TarClientPath",
     smbclient      => "SmbClientPath",
     nmblookup      => "NmbLookupPath",
     rsync          => "RsyncClientPath",
