--- configure.pl.orig	2020-06-20 15:55:58.000000000 -0700
+++ configure.pl	2020-06-21 22:45:48.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!@@PERL_FULL@@
 #============================================================= -*-perl-*-
 #
 # configure.pl: Configuration and installation program for BackupPC
@@ -347,9 +347,9 @@
 # These are the programs whose paths we need to find
 #
 my %Programs = (
-    perl       => "PerlPath",
+    '@@PERL@@' => "PerlPath",
     rsync_bpc  => "RsyncBackupPCPath",
-    'gtar/tar' => "TarClientPath",
+    gnutar     => "TarClientPath",
     smbclient  => "SmbClientPath",
     nmblookup  => "NmbLookupPath",
     rsync      => "RsyncClientPath",
