--- configure.pl.orig
+++ configure.pl
@@ -522,7 +522,7 @@ exit unless prompt("--> Do you want to continue?", "y") =~ /y/i;
 #
 # Create install directories
 #
-foreach my $dir ( qw(bin doc
+foreach my $dir ( qw(bin share/doc/BackupPC
 		     lib/BackupPC/CGI
 		     lib/BackupPC/Config
 		     lib/BackupPC/Lang
@@ -693,9 +693,9 @@ foreach my $init ( qw(gentoo-backuppc gentoo-backuppc.conf linux-backuppc
 printf("Making Apache configuration file for suid-perl\n");
 InstallFile("httpd/src/BackupPC.conf", "httpd/BackupPC.conf", 0644);
 
-printf("Installing docs in $DestDir$Conf{InstallDir}/doc\n");
+printf("Installing docs in $DestDir$Conf{InstallDir}/share/doc/BackupPC\n");
 foreach my $doc ( qw(BackupPC.pod BackupPC.html) ) {
-    InstallFile("doc/$doc", "$DestDir$Conf{InstallDir}/doc/$doc", 0444);
+    InstallFile("doc/$doc", "$DestDir$Conf{InstallDir}/share/doc/BackupPC/$doc", 0444);
 }
 
 printf("Installing config.pl and hosts in $DestDir$Conf{ConfDir}\n");
@@ -948,7 +948,7 @@ will need to do:
 
   - Edit the list of hosts to backup in $Conf{ConfDir}/hosts.
 
-  - Read the documentation in $Conf{InstallDir}/doc/BackupPC.html.
+  - Read the documentation in $Conf{InstallDir}/share/doc/BackupPC/doc/BackupPC.html.
     Please pay special attention to the security section.
 
   - Verify that the CGI script BackupPC_Admin runs correctly.  You might
--- lib/BackupPC/CGI/View.pm.orig
+++ lib/BackupPC/CGI/View.pm
@@ -89,7 +89,7 @@ sub action
         $file = $bpc->ConfDir() . "/hosts";
         $linkHosts = 1;
     } elsif ( $type eq "docs" ) {
-        $file = $bpc->InstallDir() . "/doc/BackupPC.html";
+        $file = $bpc->InstallDir() . "/share/doc/BackupPC/BackupPC.html";
     } elsif ( $host ne "" ) {
         if ( !defined($In{num}) ) {
             # get the latest LOG file
