--- FETK-1.9.3/punc/CMakeLists.txt.orig	2024-04-16 14:10:11
+++ FETK-1.9.3/punc/CMakeLists.txt	2024-04-16 14:11:00
@@ -122,9 +122,10 @@
 
 # maloc library
 if(NOT TARGET maloc)
-    message(FATAL_ERROR "Maloc target not found")
+    list(APPEND PUNC_EXT_LIBS maloc)
+else()
+    list(APPEND PUNC_INT_LIBS maloc)
 endif()
-list(APPEND PUNC_INT_LIBS maloc)
 
 
 ################################################################################
--- FETK-1.9.3/mc/CMakeLists.txt.orig	2024-04-16 14:08:40
+++ FETK-1.9.3/mc/CMakeLists.txt	2024-04-16 14:09:41
@@ -65,9 +65,10 @@
 
 # maloc library
 if(NOT TARGET maloc)
-    message(FATAL_ERROR "Maloc target not found")
+     list(APPEND MC_EXT_LIBS maloc)
+else()
+     list(APPEND MC_INT_LIBS maloc)
 endif()
-list(APPEND MC_INT_LIBS maloc)
 
 # punc library
 if(NOT TARGET punc)
--- FETK-1.9.3/gamer/CMakeLists.txt.orig	2024-04-16 14:11:32
+++ FETK-1.9.3/gamer/CMakeLists.txt	2024-04-16 14:12:59
@@ -51,7 +51,9 @@
 
 # maloc library
 if(NOT TARGET maloc)
-    message(FATAL_ERROR "Maloc target not found")
+    list(APPEND GAMER_EXT_LIBS maloc)
+else()
+    list(APPEND GAMER_INT_LIBS maloc)
 endif()
 #include_directories(
 #    ${CMAKE_CURRENT_SOURCE_DIR}/../maloc/src/base
@@ -59,7 +61,6 @@
 #    ${CMAKE_CURRENT_SOURCE_DIR}/../maloc/src/vsh
 #    ${CMAKE_CURRENT_SOURCE_DIR}/../maloc/src/vsys
 #)
-list(APPEND GAMER_INT_LIBS maloc)
 
 
 ################################################################################
--- FETK-1.9.3/CMakeLists.txt.orig	2024-04-16 14:30:41
+++ FETK-1.9.3/CMakeLists.txt	2024-04-16 14:28:59
@@ -67,7 +67,7 @@
 endif()
 
 if(BUILD_PUNC OR BUILD_GAMER) # SG would be included in this if statement
-    set(BUILD_MALOC TRUE)
+    set(BUILD_MALOC FALSE)
 endif()
 
 
