--- CMakeLists.txt.orig	2016-06-07 15:34:23.000000000 -0400
+++ CMakeLists.txt	2016-06-07 15:31:19.000000000 -0400
@@ -214,8 +214,8 @@
     if(HAVE_GFORTRAN GREATER 0)
         message(FATAL_ERROR gfortran is currently required for using BEM/TABIPB in APBS)
     endif()
-    get_filename_component(LIBGFORTRANPATH ${LIBGFORTRANPATH} PATH)
-    find_path(LIBGFORTRAN_PATH ${LIBGFORTRAN_NAME} PATH ${LIBGFORTRANPATH})
+    #get_filename_component(LIBGFORTRANPATH ${LIBGFORTRANPATH} PATH)
+    #find_path(LIBGFORTRAN_PATH ${LIBGFORTRAN_NAME} PATH ${LIBGFORTRANPATH})
     list(APPEND APBS_LIBS "-L${APBS_ROOT}/${LIBRARY_INSTALL_PATH} -L${LIBGFORTRAN_PATH} -lgfortran -ltabipb -lstdc++" )
 endif() # ENABLE_BEM
 
@@ -277,7 +277,7 @@
     )
     set(MALOC_LIBRARY ${LIBRARY_OUTPUT_PATH}/${MALOC_LIBRARY_BASENAME})
 endif()
-list(APPEND APBS_LIBS -lmaloc)
+list(APPEND APBS_LIBS -lmaloc -lreadline)
 
 
 ##############################################################################
@@ -676,3 +676,4 @@
                 DESTINATION ${EXECUTABLE_INSTALL_PATH})
     endif()
 endif()
+include_directories(${APBS_ROOT}/include)
