--- tests/test_cases.cfg.orig	2024-04-16 10:18:48
+++ tests/test_cases.cfg	2024-04-16 10:21:19
@@ -76,36 +76,36 @@
 apbs-smol          : 1.847860440020E+03 1.885436377745E+03 2.734040568569E+03 3.125279428954E+03 -3.757593797629E+01 -3.912388198513E+02 -3.536628818750E+02
 
 
-[geoflow]
-input_dir   : ../examples/geoflow
-imidazole   : -1.030222099963E+01 5.417419E-01
+#[geoflow]
+#input_dir   : ../examples/geoflow
+#imidazole   : -1.030222099963E+01 5.417419E-01
 
 
-[bem]
-input_dir   : ../examples/bem
-451c_order1 : -4.678324477794E+03
-451c_order5 : -4.678190000000e+03
+#[bem]
+#input_dir   : ../examples/bem
+#451c_order1 : -4.678324477794E+03
+#451c_order5 : -4.678190000000e+03
 
-[bem-pKa]
-input_dir     : ../examples/bem-pKa
-ASH66         : -4.005763343995E+00
-2LZT-ASH66    : -3.636994149197E+02
-2LZT-noASH66  : -3.625121468362E+02
+#[bem-pKa]
+#input_dir     : ../examples/bem-pKa
+#ASH66         : -4.005763343995E+00
+#2LZT-ASH66    : -3.636994149197E+02
+#2LZT-noASH66  : -3.625121468362E+02
 
-[bem-binding-energy]
-input_dir     : ../examples/bem-binding-energy
-1d30          : -2.197264516397E+04
-1d30_monomer1 : -2.608072100934E+04
-1d30_monomer2 : -7.676311645066E+02
+#[bem-binding-energy]
+#input_dir     : ../examples/bem-binding-energy
+#1d30          : -2.197264516397E+04
+#1d30_monomer1 : -2.608072100934E+04
+#1d30_monomer2 : -7.676311645066E+02
 
 
-[pbam]
-input_dir       : ../examples/pbam
-toy_energyforce : -4.965018552290E+01 -4.965018552290E+01
+#[pbam]
+#input_dir       : ../examples/pbam
+#toy_energyforce : -4.965018552290E+01 -4.965018552290E+01
 
-[pbsam]
-input_dir       : ../examples/pbsam-gly
-gly_energyforce : 6.176608555569E-05 6.210593742215E-05
+#[pbsam]
+#input_dir       : ../examples/pbsam-gly
+#gly_energyforce : 6.176608555569E-05 6.210593742215E-05
 
 # Very slow test
 #[protein-rna]
