--- tests/apbs_tester.py.orig	2024-04-17 07:15:37
+++ tests/apbs_tester.py	2024-04-17 07:15:59
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 
 """
 Provides utility for testing apbs against examples and known results
--- tests/apbs_logger.py.orig	2024-04-17 07:17:38
+++ tests/apbs_logger.py	2024-04-17 07:17:45
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 
 """
 Provides a logger that prints regular messages and logger messages.
--- tests/apbs_check_forces.py.orig	2024-04-17 07:18:31
+++ tests/apbs_check_forces.py	2024-04-17 07:18:40
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 
 """
 Checks computed forces from an apbs run
--- tools/python/noinput.py.orig	2024-04-17 07:19:08
+++ tools/python/noinput.py	2024-04-17 07:19:19
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python3
 """ Python APBS No Input Driver File
 
     This module allows a user to run APBS through Python. Use this module if
--- examples/protein-rna/fit.py.orig	2024-04-17 07:20:08
+++ examples/protein-rna/fit.py	2024-04-17 07:20:16
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Performs unweighted linear regression.  Can be invoked from command line by
 providing data pairs through stdin.
 
--- tests/apbs_check_results.py.orig	2024-04-17 07:20:51
+++ tests/apbs_check_results.py	2024-04-17 07:20:58
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 
 """
 Provides functions for verifying results from a test run
--- examples/protein-rna/apbs_dx.py.orig	2024-04-17 07:21:38
+++ examples/protein-rna/apbs_dx.py	2024-04-17 07:22:03
@@ -1,4 +1,4 @@
-#!/bin/python
+#!/usr/bin/env python3
 
 """
 Create files for the protein-rna tests
--- tests/apbs_check_intermediate_energies.py.orig	2024-04-17 07:23:23
+++ tests/apbs_check_intermediate_energies.py	2024-04-17 07:23:28
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 Check intermediate computed energies form an APBS run
