--- src/GUI/app.cpp.orig
+++ src/GUI/app.cpp
@@ -43,7 +43,7 @@ App::App(int &argc, char **argv) : QApplication(argc, argv)
 			installTranslator(app);
 
 	QTranslator *qt = new QTranslator(this);
-#if defined(Q_OS_WIN32) || defined(Q_OS_MACOS)
+#if defined(Q_OS_WIN32)
 	if (qt->load(QLocale::system(), "qt", "_", ProgramPaths::translationsDir()))
 #else // Q_OS_WIN32 || Q_OS_MACOS
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
