--- crude.cfg.orig
+++ crude.cfg
@@ -7,6 +7,7 @@

 # THIS CONFIGURATION FILE USES SUBSTITUTES IN THE INCOMPLETE TONE BANK!

+dir @PREFIX@/share/freepats

 # Bank 0: Melodic 0-based patches
 bank 0
--- freepats.cfg.orig
+++ freepats.cfg
@@ -1,6 +1,8 @@
 # Automatically generated on Sun Feb 19 19:22:39 EST 2006
 # by http://freepats.opensrc.org/mkcfg.sh.txt

+dir @PREFIX@/share/freepats
+
 drumset 0

  25	Drum_000/025_Snare_Roll.pat 
