--- scripts/configure.orig	2016-07-21 10:25:36.000000000 -0500
+++ scripts/configure	2017-02-10 05:19:07.000000000 -0600
@@ -5028,7 +5028,6 @@
        /usr/lib64 \
        /usr/local \
        `ls -dr /usr/lib/tcl[7-9].[0-9]* 2>/dev/null` \
-       /sw/lib \
        /usr
       do
         if test -r "$dir/tclConfig.sh" ; then
@@ -5117,7 +5116,6 @@
        /usr/local \
        `ls -dr /usr/lib/tcl[7-9].[0-9]* 2>/dev/null` \
        `ls -dr /usr/lib/tk[7-9].[0-9]* 2>/dev/null` \
-       /sw/lib \
        ${x_libraries} \
        /usr/lib \
        /usr
@@ -6422,7 +6420,7 @@
       SHDLIB_EXT=".dylib"
       LDDL_FLAGS="-dynamiclib -flat_namespace -undefined suppress -noprebind"
       LDFLAGS="${LDFLAGS} ${LIB_SPECS}"
-      CFLAGS="${CFLAGS} ${X_CFLAGS} ${INC_SPECS} -I/sw/include -fno-common"
+      CFLAGS="${CFLAGS} ${X_CFLAGS} ${INC_SPECS} -fno-common"
       ;;
 
     *cygwin*)
