--- style/fonts.mss.orig	2021-08-21 18:35:03.000000000 +0100
+++ style/fonts.mss	2021-08-21 18:37:38.000000000 +0100
@@ -189,4 +189,4 @@
 Italics are only available for the base font, not the other scripts.
 For a considerable number of labels this style will make no difference to the regular style.
 */
-@oblique-fonts: "Noto Sans Italic", @book-fonts;
+@oblique-fonts: "Noto Sans Italic", "DejaVu Sans Oblique", @book-fonts;
