--- setup.py.orig
+++ setup.py
@@ -274,7 +274,7 @@ def init_env(
         cppflags.append('-DDEBUG_{}'.format(el.upper().replace('-', '_')))
     cflags_ = os.environ.get(
         'OVERRIDE_CFLAGS', (
-            '-Wextra {} -Wno-missing-field-initializers -Wall -Wstrict-prototypes {}'
+            '-Wextra {} -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wstrict-prototypes {}'
             ' -pedantic-errors -Werror {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden {}'
         ).format(
             float_conversion,
