--- setup.py.orig
+++ setup.py
@@ -529,7 +529,7 @@ def init_env(
 
     cflags_ = os.environ.get(
         'OVERRIDE_CFLAGS', (
-            f'-Wextra {float_conversion} -Wno-missing-field-initializers -Wall -Wstrict-prototypes {c_std}'
+            f'-Wextra {float_conversion} -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wstrict-prototypes {c_std}'
             f' {werror} {optimize} {sanitize_flag} -fwrapv {stack_protector} {missing_braces}'
             f' -pipe -fvisibility=hidden {no_plt}'
         )
