--- setup.py.orig	2020-03-24 20:44:00.000000000 -0500
+++ setup.py	2020-03-24 20:44:45.000000000 -0500
@@ -256,7 +256,7 @@
         cppflags.append('-DDEBUG_{}'.format(el.upper().replace('-', '_')))
     cflags_ = os.environ.get(
         'OVERRIDE_CFLAGS', (
-            '-Wextra {} -Wno-missing-field-initializers -Wall -Wstrict-prototypes -std=c99'
+            '-Wextra {} -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wstrict-prototypes -std=c99'
             ' -pedantic-errors -Werror {} {} -fwrapv {} {} -pipe {} -fvisibility=hidden {}'
         ).format(
             float_conversion,
