diff --git src/tbbmalloc/proxy_overload_osx.h src/tbbmalloc/proxy_overload_osx.h
index 53afe13..98f622d 100644
--- src/tbbmalloc/proxy_overload_osx.h
+++ src/tbbmalloc/proxy_overload_osx.h
@@ -139,10 +139,11 @@ struct DoMallocReplacement {
         introspect.force_unlock = &zone_force_unlock;
         introspect.statistics = zone_statistics;
         introspect.zone_locked = &zone_locked;
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         introspect.enable_discharge_checking = &impl_zone_enable_discharge_checking;
         introspect.disable_discharge_checking = &impl_zone_disable_discharge_checking;
         introspect.discharge = &impl_zone_discharge;
-
+#endif
         zone.size = &impl_malloc_usable_size;
         zone.malloc = &impl_malloc;
         zone.calloc = &impl_calloc;
@@ -155,8 +156,9 @@ struct DoMallocReplacement {
         zone.version = 8;
         zone.memalign = impl_memalign;
         zone.free_definite_size = &impl_free_definite_size;
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         zone.pressure_relief = &impl_pressure_relief;
-
+#endif
         // make sure that default purgeable zone is initialized
         malloc_default_purgeable_zone();
         void* ptr = malloc(1);
