--- tools/Makefile.in.old	2016-11-18 16:53:58.000000000 +0100
+++ tools/Makefile.in	2016-11-24 15:36:03.000000000 +0100
@@ -669,7 +669,7 @@ gpg_wks_client_SOURCES = \
 gpg_wks_client_CFLAGS = $(LIBASSUAN_CFLAGS) $(GPG_ERROR_CFLAGS) $(INCICONV)
 gpg_wks_client_LDADD = $(libcommon) \
 		       $(LIBASSUAN_LIBS) $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS) \
-		       $(LIBICONV)
+		       $(LIBICONV) $(LIBINTL)
 
 all: all-am
 
