Install the libxvidcore.dylib symlink.
--- configure.in.orig	2019-12-28 08:19:24.000000000 -0600
+++ configure.in	2024-04-29 02:28:42.000000000 -0500
@@ -329,6 +329,7 @@
 	;;
      darwin*)
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
+	SO_LINK="libxvidcore.\$(SHARED_EXTENSION)"
 	SPECIFIC_CFLAGS="-fPIC -fno-common -no-cpp-precomp"
 	if test x"$macosx_module" = x"no"; then
 	   AC_MSG_RESULT([dylib options])
