From e483556bd5aa9a72fbf36cb980abed4bee0f3198 Mon Sep 17 00:00:00 2001
From: Aaron Madlon-Kay <aaron@madlon-kay.com>
Date: Thu, 17 Mar 2022 09:37:06 +0900
Subject: [PATCH] Replace python with token to allow replacing with
 ${python.bin}

---
 bin/cask | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/cask b/bin/cask
index 4a2c4fa..da87d01 100755
--- a/bin/cask
+++ b/bin/cask
@@ -38,7 +38,7 @@ if [ -z "$SRCDIR__" ] ; then
 fi
 set -eu
 if [ -z "$SRCDIR__" ]; then
-    SRCDIR__=$(python -c "import os, sys; print(os.path.realpath(sys.argv[1]))" "$CASK")
+    SRCDIR__=$(@PYTHON_BIN@ -c "import os, sys; print(os.path.realpath(sys.argv[1]))" "$CASK")
 fi
 SRCDIR_=$(dirname "$SRCDIR__")
 SRCDIR=$(dirname "$SRCDIR_")
-- 
2.41.0

