--- configure.ac.orig	2011-02-16 13:43:26.000000000 -0600
+++ configure.ac	2012-09-07 03:36:03.000000000 -0500
@@ -121,10 +121,10 @@
 # --------------------------------------------------------------------
 AC_MSG_NOTICE([])
 AC_MSG_NOTICE([--> checking for Xft configuration])
-AC_CHECK_PROG([XO_XFT_CFLAGS],[xft-config],[`xft-config --cflags`],
-	[AC_MSG_ERROR([unable to find xft-config])])
-AC_CHECK_PROG([XO_XFT_LIBS],[xft-config],[`xft-config --libs`],
-	[AC_MSG_ERROR([unable to find xft-config])])
+AC_CHECK_PROG([XO_XFT_CFLAGS],[pkg-config],[`pkg-config xft --cflags`],
+	[AC_MSG_ERROR([unable to find pkg-config])])
+AC_CHECK_PROG([XO_XFT_LIBS],[pkg-config],[`pkg-config xft --libs`],
+	[AC_MSG_ERROR([unable to find pkg-config])])
 
 
 # --------------------------------------------------------------------
@@ -146,7 +146,7 @@
 AC_DEFUN([XO_TMP], [tmpx_$1])
 AC_DEFUN([XO_UNIQ],
 [(\
-	echo -n '' > XO_TMP($1)
+	echo '' > XO_TMP($1)
 	xo_buf=""
 	for i in $$1; do
 	    j=`echo $i | sed -e 's/^-.*//g'`
@@ -165,7 +165,7 @@
 	if [[ ! -z "$xo_buf" ]]; then
 	   echo $xo_buf >> XO_TMP($1)
 	fi
-	echo -n `cat XO_TMP($1) | sort -b | uniq` > XO_TMP($1)
+	echo `cat XO_TMP($1) | sort -b | uniq` > XO_TMP($1)
 )
 $1=`cat XO_TMP($1)`
 rm -f XO_TMP($1)
