--- syscall.c.orig
+++ syscall.c
@@ -126,1 +126,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -221,1 +221,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -346,1 +346,1 @@
-#if defined AT_FDCWD && defined HAVE_LINKAT
+#if defined AT_FDCWD && defined HAVE_LINKAT && !defined RSYNC_NO_AT_SYSCALLS
@@ -478,1 +478,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -535,1 +535,1 @@
-	return do_lchown(fname, owner, group);
+	return do_lchown(fname, owner, group, mode, file_flags);
@@ -712,1 +712,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -802,1 +802,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -945,1 +945,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -1006,1 +1006,1 @@
-	return do_chmod(fname, mode);
+	return do_chmod(fname, mode, file_flags);
@@ -1066,1 +1066,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -1144,1 +1144,1 @@
-	return do_rename(old_path, new_path);
+	return do_rename(old_path, new_path, mode, file_flags);
@@ -1205,1 +1205,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -1318,1 +1318,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
@@ -1560,1 +1560,1 @@
-#ifdef AT_FDCWD
+#if defined AT_FDCWD && !defined RSYNC_NO_AT_SYSCALLS
