Index: pre-config.rb
===================================================================
--- pre-config.rb	(revision 2267)
+++ pre-config.rb	(working copy)
@@ -82,7 +82,7 @@
 config_ary << [ :other_cflags, cflags ]
 config_ary << [ :other_ldflags, ldflags ]
 config_ary << [ :target_archs, archs.size > 0 ? archs : '$NATIVE_ARCH' ]
-config_ary << [ :arch_flags, arch_flags ]
+config_ary << [ :arch_flags, archs.size > 0 ? arch_flags : '' ]
 
 target_files.each do |dst_name|
   src_name = dst_name + '.in'
