--- setup.py.orig	2025-11-04 10:40:13
+++ setup.py	2025-11-04 10:40:32
@@ -12,7 +12,6 @@
 from distutils.core import Extension
 from distutils.command.sdist import sdist
 from distutils.command.build_ext import build_ext
-from distutils.command.upload import upload
 from distutils import cygwinccompiler
 
 
@@ -111,20 +110,6 @@
             return build_ext.build_extension(self, ext)
 
 
-class custom_upload(upload):
-    """Upload binary package to PyPI with credentials obtained from environment overriding pypirc.
-
-    This is used by Appveyor to upload binary wheels to PyPI.
-    """
-
-    def finalize_options(self):
-        import os
-        upload.finalize_options(self)
-        overrides = ((k[5:].lower(), v) for k, v in os.environ.iteritems() if k.startswith('PYPI_'))
-        for k, v in overrides:
-            setattr(self, k, v)
-
-
 def read_long_description():
     import io
     parts = []
@@ -137,7 +122,7 @@
 metadata = dict(
     author           = "Stefano Taschini",
     author_email     = 'taschini@gmail.com',
-    cmdclass         = {'sdist': sdist, 'build_ext': custom_build_ext, 'upload': custom_upload},
+    cmdclass         = {'sdist': sdist, 'build_ext': custom_build_ext},
     description      = "Python bindings for CRlibm, an efficient and proven correctly-rounded mathematical library",
     install_requires = [],
     keywords         = 'crlibm',
