--- CMakeLists.txt.orig	2022-11-30 00:19:21.000000000 +0800
+++ CMakeLists.txt	2023-05-24 17:52:16.000000000 +0800
@@ -3,7 +3,7 @@
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 # Has to be set before `project()`, and ignored on non-macos:
-set(CMAKE_OSX_DEPLOYMENT_TARGET 10.12 CACHE STRING "macOS deployment target (Apple clang only)")
+set(CMAKE_OSX_DEPLOYMENT_TARGET @MACOS@ CACHE STRING "macOS deployment target (Apple clang only)")
 
 find_program(CCACHE_PROGRAM ccache)
 if(CCACHE_PROGRAM)

