https://sources.debian.org/patches/lsh-utils/2.1-12/nettle3.4.patch/
--- src/dummy.c.orig
+++ src/dummy.c
@@ -113,14 +113,14 @@ base64_encode_init(struct base64_encode_ctx *ctx UNUSED)
 
 size_t
 base64_encode_update(struct base64_encode_ctx *ctx UNUSED,
-		     uint8_t *dst UNUSED,
+		     char *dst UNUSED,
 		     size_t length UNUSED,
 		     const uint8_t *src UNUSED)
 { abort(); }
 
 size_t
 base64_encode_final(struct base64_encode_ctx *ctx UNUSED,
-		    uint8_t *dst UNUSED)
+		    char *dst UNUSED)
 { abort(); }
 
 void
@@ -132,7 +132,7 @@ base64_decode_update(struct base64_decode_ctx *ctx UNUSED,
 		     size_t *dst_length UNUSED,
 		     uint8_t *dst UNUSED,
 		     size_t src_length UNUSED,
-		     const uint8_t *src UNUSED)
+		     const char *src UNUSED)
 { abort(); }
 
 int
