--- src/spki/configure.orig	Tue Feb 24 10:21:49 2004
+++ src/spki/configure	Tue Feb 24 10:22:00 2004
@@ -1755,7 +1755,7 @@
 fi;
 
 if test x$with_include_path != x ; then
-  CPPFLAGS="$CPPFLAGS -I `echo $with_include_path | sed 's/:/ -I /g'`"
+  CPPFLAGS="$CPPFLAGS -I`echo $with_include_path | sed 's/:/ -I/g'`"
 fi
 
 
@@ -1768,7 +1768,7 @@
 fi;
 
 if test x$with_lib_path != x ; then
-  LDFLAGS="$LDFLAGS -L `echo $with_lib_path | sed 's/:/ -L /g'`"
+  LDFLAGS="$LDFLAGS -L`echo $with_lib_path | sed 's/:/ -L/g'`"
 fi
 
 echo "$as_me:$LINENO: checking for -R flag" >&5
