--- src/argp/argp-pvh.c.orig	Mon Nov 27 11:26:07 2000
+++ src/argp/argp-pvh.c	Tue Feb 24 10:48:31 2004
@@ -29,4 +29,4 @@
    this function with a stream to print the version to and a pointer to the
    current parsing state, and then exits (unless the ARGP_NO_EXIT flag is
    used).  This variable takes precedent over ARGP_PROGRAM_VERSION.  */
-void (*argp_program_version_hook) (FILE *stream, struct argp_state *state);
+void (*argp_program_version_hook) (FILE *stream, struct argp_state *state) = 0;
