--- source/fitz/font.c.orig
+++ source/fitz/font.c
@@ -4,7 +4,7 @@
 #include "font-imp.h"
 #include "draw-imp.h"
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include "hb.h"
 #include "hb-ft.h"
 
--- source/fitz/stext-output.c.orig
+++ source/fitz/stext-output.c
@@ -3,7 +3,7 @@
 #define SUBSCRIPT_OFFSET 0.2f
 #define SUPERSCRIPT_OFFSET -0.2f
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_FREETYPE_H
 
 /* HTML output (visual formatting with preserved layout) */
--- source/html/html-layout.c.orig
+++ source/html/html-layout.c
@@ -4,7 +4,7 @@
 
 #include "hb.h"
 #include "hb-ft.h"
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 
 #include <math.h>
 
--- source/pdf/pdf-appearance.c.orig
+++ source/pdf/pdf-appearance.c
@@ -5,7 +5,7 @@
 #include <limits.h>
 #include <math.h>
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_FREETYPE_H
 #include FT_ADVANCES_H
 
--- source/pdf/pdf-device.c.orig
+++ source/pdf/pdf-device.c
@@ -1,7 +1,7 @@
 #include "mupdf/fitz.h"
 #include "mupdf/pdf.h"
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_FREETYPE_H
 #include FT_ADVANCES_H
 
--- source/pdf/pdf-font.c.orig
+++ source/pdf/pdf-font.c
@@ -6,7 +6,7 @@
 
 #include <assert.h>
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_FREETYPE_H
 #include FT_ADVANCES_H
 #ifdef FT_FONT_FORMATS_H
--- source/xps/xps-glyphs.c.orig
+++ source/xps/xps-glyphs.c
@@ -2,7 +2,7 @@
 #include "xps-imp.h"
 #include "../fitz/fitz-imp.h"
 
-#include <ft2build.h>
+#include <freetype2/ft2build.h>
 #include FT_FREETYPE_H
 #include FT_ADVANCES_H
 
