https://bugs.ghostscript.com/show_bug.cgi?id=699726

--- platform/gl/gl-main.c.orig
+++ platform/gl/gl-main.c
@@ -1673,9 +1673,9 @@ int main(int argc, char **argv)
 	search_input.end = search_input.p;
 
 	/* Init GLUT */
-
+#if defined(GLUT_ACTION_ON_WINDOW_CLOSE)
 	glutSetOption(GLUT_ACTION_ON_WINDOW_CLOSE, GLUT_ACTION_GLUTMAINLOOP_RETURNS);
-
+#endif
 	glutInitErrorFunc(on_error);
 	glutInitWarningFunc(on_warning);
 	glutInitDisplayMode(GLUT_RGBA | GLUT_DOUBLE);
