--- Makerules.orig	2011-09-05 06:55:39.000000000 -0500
+++ Makerules	2011-10-26 03:20:01.000000000 -0500
@@ -18,31 +18,10 @@
 $(error unknown build setting: '$(build)')
 endif
 
-ifeq "$(OS)" "Linux"
 SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
 X11_LIBS := -lX11 -lXext
-endif
 
-ifeq "$(OS)" "FreeBSD"
-SYS_FREETYPE_INC := `pkg-config --cflags freetype2`
-LDFLAGS += -L/usr/local/lib
-X11_LIBS := -lX11 -lXext
-endif
 
-# Mac OS X build depends on some thirdparty libs
-ifeq "$(OS)" "Darwin"
-SYS_FREETYPE_INC := -I/usr/X11R6/include/freetype2
-CFLAGS += -I/usr/X11R6/include
-LDFLAGS += -L/usr/X11R6/lib
-X11_LIBS := -lX11 -lXext
-ifeq "$(arch)" "amd64"
-CFLAGS += -m64
-LDFLAGS += -m64
-else
-CFLAGS += -m32
-LDFLAGS += -m32
-endif
-endif
 
 # The following section is an example of how to simply do cross-compilation
 # using these Makefiles. It builds for a beagleboard running ARM linux,
