--- Makethird	2024-05-08 18:53:12.000000000 +0800
+++ Makethird	2024-05-25 06:25:49.000000000 +0800
@@ -52,12 +52,6 @@
   endif
 endif
 
-ifeq ($(USE_TESSERACT),yes)
-  ifneq ($(HAVE_LEPTONICA),yes)
-    echo "Build will fail! Tesseract requires LEPTONICA"
-  endif
-endif
-
 # --- Include source file lists ---
 
 include Makelists
@@ -240,7 +234,7 @@
   THIRD_CFLAGS += -DHAVE_TESSERACT
 ifeq ($(USE_SYSTEM_TESSERACT),yes)
   THIRD_CFLAGS += $(SYS_TESSERACT_CFLAGS)
-  THIRD_LIBS += $(SYS_TESSERACT_LIBS) -lstdc++
+  THIRD_LIBS += $(SYS_TESSERACT_LIBS)
 else
   THIRD_CFLAGS += $(TESSERACT_CFLAGS)
   THIRD_LIBS += $(TESSERACT_LIBS)
@@ -290,7 +284,7 @@
   THIRD_CFLAGS += -DHAVE_LIBARCHIVE
 ifeq ($(USE_SYSTEM_LIBARCHIVE),yes)
   THIRD_CFLAGS += $(SYS_LIBARCHIVE_CFLAGS)
-  THIRD_LIBS += $(SYS_LIBARCHIVE_LIBS) -lstdc++
+  THIRD_LIBS += $(SYS_LIBARCHIVE_LIBS)
 else
   THIRD_CFLAGS += $(LIBARCHIVE_CFLAGS)
   THIRD_LIBS += $(LIBARCHIVE_LIBS)

--- Makefile	2024-05-08 18:53:12.000000000 +0800
+++ Makefile	2024-05-25 05:51:49.000000000 +0800
@@ -73,7 +73,7 @@
 ifdef RANLIB
   RANLIB_CMD = $(QUIET_RANLIB) $(RANLIB) $@
 endif
-LINK_CMD = $(QUIET_LINK) $(MKTGTDIR) ; $(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
+LINK_CMD = $(QUIET_LINK) $(MKTGTDIR) ; $(CXX) $(LDFLAGS) -o $@ $^ $(LIBS)
 TAGS_CMD = $(QUIET_TAGS) ctags
 WINDRES_CMD = $(QUIET_WINDRES) $(MKTGTDIR) ; $(WINDRES) $< $@
 OBJCOPY_CMD = $(QUIET_OBJCOPY) $(MKTGTDIR) ; $(LD) -r -b binary -z noexecstack -o $@ $<
@@ -81,7 +81,7 @@
 DLLTOOL_CMD = $(QUIET_DLLTOOL) dlltool -d $< -D $(notdir $(^:%.def=%.dll)) -l $@
 
 ifeq ($(shared),yes)
-LINK_CMD = $(QUIET_LINK) $(MKTGTDIR) ; $(CC) $(LDFLAGS) -o $@ \
+LINK_CMD = $(QUIET_LINK) $(MKTGTDIR) ; $(CXX) $(LDFLAGS) -o $@ \
 	$(filter-out %.$(SO)$(SO_VERSION),$^) \
 	$(sort $(patsubst %,-L%,$(dir $(filter %.$(SO)$(SO_VERSION),$^)))) \
 	$(patsubst lib%.$(SO)$(SO_VERSION),-l%,$(notdir $(filter %.$(SO)$(SO_VERSION),$^))) \
