--- Makerules.orig	2016-07-25 12:09:24.000000000 -0500
+++ Makerules	2016-07-25 11:58:29.000000000 -0500
@@ -65,20 +65,22 @@
 
 SYS_CURL_DEPS = -lpthread
 
-SYS_X11_CFLAGS = -I/usr/X11R6/include
-SYS_X11_LIBS = -L/usr/X11R6/lib -lX11 -lXext
+SYS_X11_CFLAGS = $(shell pkg-config --cflags x11 xext)
+SYS_X11_LIBS = $(shell pkg-config --libs x11 xext)
 
-SYS_FREETYPE_CFLAGS = $(shell freetype-config --cflags)
-SYS_FREETYPE_LIBS = $(shell freetype-config --libs)
+SYS_FREETYPE_CFLAGS = $(shell pkg-config --cflags freetype2)
+SYS_FREETYPE_LIBS = $(shell pkg-config --libs freetype2)
 SYS_OPENJPEG_LIBS = -lopenjpeg
+SYS_HARFBUZZ_CFLAGS = $(shell pkg-config --cflags harfbuzz)
+SYS_HARFBUZZ_LIBS = $(shell pkg-config --libs harfbuzz)
 SYS_JBIG2DEC_LIBS = -ljbig2dec
 SYS_JPEG_LIBS = -ljpeg
 SYS_ZLIB_LIBS = -lz
 
-CC = xcrun cc
-AR = xcrun ar
-LD = xcrun ld
-RANLIB_CMD = xcrun ranlib $@
+CC = cc
+AR = ar
+LD = ld
+RANLIB_CMD = ranlib $@
 
 # Linux uses pkg-config for system libraries.
 else ifeq "$(OS)" "Linux"
