index 7047b72422fb36c138128ade03d1faa4a5413698..90c61b98a1bf3f1ca65aa002ca64f8f596d62828 100644 (file)
--- libavcodec/libsvtav1.c
+++ libavcodec/libsvtav1.c
@@ -237,7 +237,11 @@ static int config_enc_params(EbSvtAv1EncConfiguration *param,
     } else if (svt_enc->qp > 0) {
         param->qp                   = svt_enc->qp;
         param->rate_control_mode    = 0;
+#if SVT_AV1_CHECK_VERSION(4, 0, 0)
+        param->aq_mode = 0;
+#else
         param->enable_adaptive_quantization = 0;
+#endif
     }
 
     desc = av_pix_fmt_desc_get(avctx->pix_fmt);