--- config/mkcommon.inc.in.orig	2010-10-18 11:28:33.000000000 +0200
+++ config/mkcommon.inc.in	2010-10-18 11:29:11.000000000 +0200
@@ -64,8 +64,8 @@
 RPATH              = @RPATH@
 
 LIBTOOL            = $(top_builddir)/libtool --silent
-LIBTOOL_CC         = $(LIBTOOL) --mode=compile
-LIBTOOL_LD         = $(LIBTOOL) --mode=link
+LIBTOOL_CC         = $(LIBTOOL) --mode=compile --tag=CC
+LIBTOOL_LD         = $(LIBTOOL) --mode=link --tag=CC
 LIBTOOL_EXE        = $(LIBTOOL) --mode=execute -dlopen $(LIBNAME)
 
 # Java
