diff --git a/nonstatgab/nsdgt.m b/nonstatgab/nsdgt.m
index 887597fc..519f7c31 100644
--- inst/nonstatgab/nsdgt.m
+++ inst/nonstatgab/nsdgt.m
@@ -124,8 +124,8 @@
         col = ceil(Lg/M(ii));
         
         temp = zeros(col*M(ii),W,assert_classname(f,g{1}));
-        temp([end-floor(Lg/2)+1:end,1:ceil(Lg/2)],:) = bsxfun(@ ...
-                                                          times,f(win_range,:),g{ii}(idx));
+        temp([end-floor(Lg/2)+1:end,1:ceil(Lg/2)],:) = bsxfun(@times, ...
+                                                          f(win_range,:),g{ii}(idx));
         
         temp = reshape(temp,M(ii),col,W);
         X = squeeze(fft(sum(temp,2)));
