--- ../conf/workers2.properties.minimal.orig	Tue Oct 19 20:52:54 2004
+++ ../conf/workers2.properties.minimal	Tue Oct 19 21:04:08 2004
@@ -17,7 +17,7 @@
 
 [shm:]
 info=Scoreboard. Required for reconfiguration and status with multiprocess servers
-file=anonymous
+file=${serverRoot}/logs/shm.file
 debug=0
 
 [workerEnv:]
@@ -27,29 +27,51 @@
 
 [lb:lb]
 info=Default load balancer.
+disabled=0
 debug=0
 
+# TCP channel on localhost:8009
 [channel.socket:localhost:8009]
 info=Ajp13 forwarding over socket
+disabled=1
 debug=0
-tomcatId=localhost:8009
 
-[uri:/admin]
-info=Tomcat HTML based administration web application.
+# Unix domain socket channel
+[channel.un:unixsocket1]
+file=/opt/local/share/java/tomcat5/logs/jk2.socket
+info=Ajp13 over a Unix domain socket
+disabled=0
 debug=0
 
-[uri:/manager]
-info=A scriptable management web application for the Tomcat Web Server.
+# Status worker
+[status:]
+info=Status worker, displays runtime informations
+disabled=0
+
+# Do some URI Mapping
+#
+# You may also, or alternatively, use mapping directives of the form:
+#	JkUriSet group ajp13:unixsocket1
+#	Inside containers in apache2 httpd.conf
+
+# Map to the status worker
+[uri:/jkstatus/*]
+info=Display status information and checks the config file for changes.
+group=status:
+
+[uri:/admin/*]
+info=Tomcat HTML based administration web application.
 debug=0
 
-[uri:/jsp-examples]
+[uri:/jsp-examples/*]
 info=JSP 2.0 Examples.
 debug=0
+#group=ajp13:unixsocket1
 
-[uri:/servlets-examples]
+[uri:/servlets-examples/*]
 info=Servlet 2.4 Examples.
 debug=0
-
+ 
 [uri:/*.jsp]
 info=JSP Extension mapping.
 debug=0
