--- infrastructure/BoxPlatform.pm.in.org	2006-02-23 12:47:36.000000000 -0800
+++ infrastructure/BoxPlatform.pm.in	2006-03-22 23:29:51.000000000 -0800
@@ -41,6 +41,11 @@
 	# if it's Darwin,
 	if($build_os eq 'Darwin')
 	{
+		# Use headers and libraries from Darwinports
+		$platform_compile_line_extra = '-I__PREFIX__/include ';
+		$platform_link_line_extra = '-L__PREFIX__/lib ';
+
+
 		# see how many processors there are, and set make flags accordingly
 		my $cpus = `sysctl hw.ncpu`;
 		if($cpus =~ m/hw.ncpu:\s(\d+)/ && $1 > 1)
