--- install.sh.orig	2007-12-17 22:22:50.000000000 -0500
+++ install.sh	2007-12-17 22:42:11.000000000 -0500
@@ -2,13 +2,12 @@
 #
 # Create ~/.mediaserv and copy the static files we care about into it.
 
-STATIC_DIR=`dirname "$0"`
-STATIC_DIR=`cd "$STATIC_DIR"; pwd`
+STATIC_DIR="@@STATIC_DIR@@"
 
 [ -d ~/.mediaserv ] || mkdir ~/.mediaserv || exit
 [ -f ~/.mediaserv/config ] && touch ~/.mediaserv/config
 
-cp -vup "${STATIC_DIR}/resources/"* ~/.mediaserv/
+cp -vrp "${STATIC_DIR}/"* ~/.mediaserv/
 
 cp ~/.mediaserv/config ~/.mediaserv/config~
 perl -e '
@@ -20,7 +19,7 @@
     print;
     print "$1 = $settings{$1}\n" if /(\w+)\s*=\s*/ and $settings{$1};
   }
-' "${STATIC_DIR}/resources/config" >~/.mediaserv/config
+' "${STATIC_DIR}/config" >~/.mediaserv/config
 
 echo "Install of mediaserv complete."
 
