--- src/auth/basic/NIS/nis_support.cc.orig	2018-07-02 13:26:07.000000000 +1000
+++ src/auth/basic/NIS/nis_support.cc	2018-07-22 14:15:33.000000000 +1000
@@ -23,7 +23,7 @@
 #include <rpc/rpc.h>
 #endif
 
-#if _SQUID_FREEBSD_  && !defined(BOOL_DEFINED)
+#ifndef BOOL_DEFINED
 // BUG: FreeBSD rpcsvc/yp_* headers try to redefine bool unless we match their non-standard hack.
 #define BOOL_DEFINED
 #endif
