--- tuxcap/CMakeModules/FindImageMagickLib.cmake.orig	2009-05-12 13:26:02.000000000 -0500
+++ tuxcap/CMakeModules/FindImageMagickLib.cmake	2013-05-05 04:03:36.000000000 -0500
@@ -27,50 +27,27 @@
 
 # find the libraries main include header file
 FIND_PATH(IMAGEMAGICKLIB_INCLUDE_DIR Magick++.h
-  "${IMAGEMAGICKLIB_DIR}/include"
-  "$ENV{IMAGEMAGICKLIB_DIR}/include"
-  "$ENV{MAGICK_DIR}/include"
-  "${IMAGEMAGICKLIB_HOME}/include"
-  "$ENV{IMAGEMAGICKLIB_HOME}/include"
-  "$ENV{MAGICK_HOME}/include"
-  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/include"
-  /usr/local/include
-  /usr/include
-  /usr/include/ImageMagick
-  /opt/net/gcc41/ImageMagick/include
-  /opt/net/gcc33/ImageMagick/include
-  )
+  @PREFIX@/include/ImageMagick-6
+)
 #MESSAGE("DBG IMAGEMAGICKLIB_INCLUDE_DIR=${IMAGEMAGICKLIB_INCLUDE_DIR}")
 
 
 # set directories to search for libraries: 
 SET(IMAGEMAGICKLIB_POSSIBLE_LIBRARY_PATHS
-  "${IMAGEMAGICKLIB_DIR}/lib"
-  "$ENV{IMAGEMAGICKLIB_DIR}/lib"
-  "$ENV{MAGICK_DIR}/lib"
-  "${IMAGEMAGICKLIB_HOME}/lib"
-  "$ENV{IMAGEMAGICKLIB_HOME}/lib"
-  "$ENV{MAGICK_HOME}/lib"
-  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;LibPath]/lib"
-  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;LibPath]"
-  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/lib"  
-  /usr/local/lib
-  /usr/lib
-  /opt/net/gcc41/ImageMagick/lib  
-  /opt/net/gcc33/ImageMagick/lib  
-  )
+  @PREFIX@/lib
+)
 #MESSAGE("DBG IMAGEMAGICKLIB_POSSIBLE_LIBRARY_PATHS=${IMAGEMAGICKLIB_POSSIBLE_LIBRARY_PATHS}")
 
 
 # find provided .lib import library files 
 FIND_LIBRARY(IMAGEMAGICKLIB_CXX_LIBRARY
-  NAMES Magick++ CORE_RL_Magick++_ CORE_DB_Magick++_
+  NAMES Magick++-6.Q16
   PATHS ${IMAGEMAGICKLIB_POSSIBLE_LIBRARY_PATHS} )
 #MESSAGE("DBG IMAGEMAGICKLIB_CXX_LIBRARY=${IMAGEMAGICKLIB_CXX_LIBRARY}")
 
 
 FIND_LIBRARY(IMAGEMAGICKLIB_C_LIBRARY
-  NAMES Magick CORE_RL_magick_ CORE_DB_magick_ MagickCore
+  NAMES MagickCore-6.Q16
   PATHS ${IMAGEMAGICKLIB_POSSIBLE_LIBRARY_PATHS} )
 #MESSAGE("DBG IMAGEMAGICKLIB_C_LIBRARY=${IMAGEMAGICKLIB_C_LIBRARY}")
 
