--- configure.in.orig	2012-03-25 02:42:42.000000000 -0700
+++ configure.in	2012-03-25 02:43:20.000000000 -0700
@@ -225,9 +225,6 @@ AC_HELP_STRING([--disable-xmms-plugin], 
 	no)  enable_xmms_plugin=false ;;
 	*) AC_MSG_ERROR(bad value ${enableval} for --enable-xmms-plugin) ;;
 esac],[enable_xmms_plugin=true])
-if test "x$enable_xmms_plugin" != xfalse ; then
-	AM_PATH_XMMS(0.9.5.1, , AC_MSG_WARN([*** XMMS >= 0.9.5.1 not installed - XMMS support will not be built]))
-fi
 AM_CONDITIONAL(FLaC__HAS_XMMS, test -n "$XMMS_INPUT_PLUGIN_DIR")
 
 dnl build FLAC++ or not
