--- strigiutils/bin/xmlindexer/CMakeLists.txt.orig	2011-04-23 13:00:17.000000000 +0200
+++ strigiutils/bin/xmlindexer/CMakeLists.txt	2011-08-22 16:32:35.000000000 +0200
@@ -1,11 +1,11 @@
-add_library(librdfindexer
+add_library(librdfindexer STATIC
         rdfindexer.cpp
         tagmapping.cpp
 	rdfindexwriter.cpp
 )
 target_link_libraries(librdfindexer ${LIBSTREAMANALYZER_LIBRARIES} ${LIBSTREAMS_LIBRARIES})
 
-add_library(libxmlindexer
+add_library(libxmlindexer STATIC
         xmlindexer.cpp
         tagmapping.cpp
 	xmlindexwriter.cpp
