--- strigidaemon/bin/daemon/xesam/xesamsearch.h.orig       2013-01-25 12:49:35.000000000 -0500
+++ strigidaemon/bin/daemon/xesam/xesamsearch.h    2015-04-17 19:57:05.000000000 -0400
@@ -43,7 +43,7 @@
     XesamSearch(Private* p);
     ~XesamSearch();
     void operator=(const XesamSearch& xs);
-    bool operator==(const XesamSearch& xs) { return p == xs.p; }
+    bool operator==(const XesamSearch& xs) const { return p == xs.p; }
     void startSearch();
     void getHitCount(void* msg);
     void getHits(void* msg, uint32_t num);
