--- VAX/vax_cpu.c.sav	2008-06-25 14:08:05.000000000 -0400
+++ VAX/vax_cpu.c	2008-06-25 14:08:15.000000000 -0400
@@ -372,7 +372,7 @@
 t_stat cpu_set_idle (UNIT *uptr, int32 val, char *cptr, void *desc);
 t_stat cpu_show_idle (FILE *st, UNIT *uptr, int32 val, void *desc);
 int32 cpu_get_vsw (int32 sw);
-SIM_INLINE int32 get_istr (int32 lnt, int32 acc);
+SIM_INLINE_DECL int32 get_istr (int32 lnt, int32 acc);
 int32 ReadOcta (int32 va, int32 *opnd, int32 j, int32 acc);
 t_bool cpu_show_opnd (FILE *st, InstHistory *h, int32 line);
 int32 cpu_psl_ipl_idle (int32 newpsl);
