From 58c96da9283d3c2f65c818bd952f611ec4e4dad0 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@freedesktop.org>
Date: Fri, 11 Dec 2009 17:15:16 -0800
Subject: [PATCH] darwin: xnu doesn't support poll on ttys on the master side.

<rdar://problem/7360546>

Signed-off-by: Jeremy Huddleston <jeremyhu@freedesktop.org>
---
 configure.ac |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index f535a06..67d5a95 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,9 +121,8 @@ AC_PREREQ([2.59c], [], [AC_SUBST([htmldir], [m4_ifset([AC_PACKAGE_TARNAME],
 XCB_CHECK_DOXYGEN()
 
 case $host_os in
-	# darwin has poll() but can't be used to poll character devices
-	# darwin10 (SnowLeopard) should be tested as well once released
-	darwin7*) ;; darwin8*) ;; darwin9*) ;;
+	# darwin has poll() but can't be used to poll character devices (atleast through SnowLeopard)
+	darwin*) ;;
 	*)
 		AC_CHECK_FUNC(poll, [AC_DEFINE(USE_POLL, 1, [poll() function is available])], )
 	;;
-- 
1.6.5.3

