--- src/gui/debugger/dbg_stack.c.orig	2010-08-19 12:25:41.000000000 +0200
+++ src/gui/debugger/dbg_stack.c	2010-08-19 12:31:13.000000000 +0200
@@ -197,7 +197,7 @@
 		gtk_window_iconify(GTK_WINDOW(dbgw.stack));
 #endif
 
-	if(!GTK_WIDGET_VISIBLE(dbgw.stack) && !options3.stack.closed)
+	if(!gtk_widget_get_visible(dbgw.stack) && !options3.stack.closed)
 		gtk_widget_show(dbgw.stack);
 
 	return dbgw.stack;
@@ -422,4 +422,4 @@
                                         gpointer         user_data)
 {
 	dbgmem_add_tab(value);
-}
\ No newline at end of file
+}
