--- src/gui/debugger/dbg_mem.c.orig	2010-08-19 12:25:41.000000000 +0200
+++ src/gui/debugger/dbg_mem.c	2010-08-19 12:31:13.000000000 +0200
@@ -435,7 +435,7 @@
 		gtk_window_iconify(GTK_WINDOW(dbgw.mem));
 #endif
 
-	if(!GTK_WIDGET_VISIBLE(dbgw.mem) && !options3.mem.closed)
+	if(!gtk_widget_get_visible(dbgw.mem) && !options3.mem.closed)
 		gtk_widget_show(dbgw.mem);
 
     return dbgw.mem;
@@ -593,7 +593,7 @@
 
 GLADE_CB void
 on_notebook1_switch_page               (GtkNotebook     *notebook,
-                                        GtkNotebookPage *page,
+                                        gpointer         page,
                                         guint            page_num,
                                         gpointer         user_data)
 {
