--- makefile.orig	2014-04-03 11:32:44.000000000 +0200
+++ makefile	2014-04-03 11:33:27.000000000 +0200
@@ -27,8 +27,10 @@
 	$(CC) $(CF) -o $@ src/$(T)lib.c $(LF)
 
 install:
-	$(INSTALL_PROGRAM) -D src/$(LIBNAME) $(DESTDIR)$(LUA_CDIR)/$(LIBNAME)
-	$(INSTALL_PROGRAM) -D src/$T/lom.lua $(DESTDIR)$(LUA_LDIR)/$T/lom.lua
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(LUA_CDIR)/
+	$(INSTALL_PROGRAM) -d $(DESTDIR)$(LUA_LDIR)/$T
+	$(INSTALL_DATA) src/$(LIBNAME) $(DESTDIR)$(LUA_CDIR)/$(LIBNAME)
+	$(INSTALL_DATA) src/$T/lom.lua $(DESTDIR)$(LUA_LDIR)/$T/lom.lua
 
 clean:
 	$(RM) src/$(LIBNAME) $(OBJS)
