--- ./setup.py	2020-07-24 08:14:22.000000000 -0400
+++ ./setup.py	2020-07-24 08:24:54.000000000 -0400
@@ -148,10 +148,4 @@
     package_data={
         'magenta': ['models/image_stylization/evaluation_images/*.jpg'],
     },
-    setup_requires=['pytest-runner', 'pytest-pylint'],
-    tests_require=[
-        'pytest >= 5.2.0',
-        'pytest-xdist < 1.30.0',  # 1.30 has problems working with pylint plugin
-        'pylint >= 2.4.2',
-    ],
 )
