--- Makefile	2006-08-17 20:03:52.000000000 +0000
+++ Makefile.new	2007-06-13 05:57:57.000000000 +0000
@@ -16,8 +16,8 @@
 	$(MAKE) -f Makefile.qmake
 
 install: build
-	$(INSTALL) -m755 -D mimms   $(DESTDIR)/$(BINDIR)/mimms
-	$(INSTALL) -m644 -D mimms.1 $(DESTDIR)/$(MANDIR)/man1/mimms.1
+	$(INSTALL) -m755 mimms.app/Contents/MacOS/mimms $(DESTDIR)$(BINDIR)/mimms
+	$(INSTALL) -m644 mimms.1 $(DESTDIR)$(MANDIR)/man1/mimms.1
 
 clean:
 	rm -f Makefile.qmake *.o mimms *~ 
