Don't use conan.
--- Rgb2Hires_PC/CMakeLists.txt.orig	2023-10-31 10:20:32.000000000 -0500
+++ Rgb2Hires_PC/CMakeLists.txt	2023-10-31 12:49:26.000000000 -0500
@@ -16,9 +16,6 @@
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
-# Conan init
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
 
 # directories
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_CURRENT_SOURCE_DIR}/bin/debug)
@@ -31,9 +28,6 @@
 							src/ImageQuantized.cpp
 							src/ImageQuantized.h
 )
-## dependencies
-conan_set_find_library_paths(${PROJECT_NAME})
-conan_target_link_libraries(${PROJECT_NAME})
 
 
 # Application Picture
@@ -48,8 +42,6 @@
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "9")
 	target_link_libraries(Picture stdc++fs) # filesystem lib not included in stdc++ for gcc < 9
 endif()
-conan_set_find_library_paths(Picture)
-conan_target_link_libraries(Picture)
 target_link_libraries(Picture ${PROJECT_NAME})
 ## output
 set_property(TARGET Picture PROPERTY
@@ -64,8 +56,6 @@
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS "9")
 	target_link_libraries(Tile stdc++fs) # filesystem lib not included in stdc++ for gcc < 9
 endif()
-conan_set_find_library_paths(Tile)
-conan_target_link_libraries(Tile)
 target_link_libraries(Tile ${PROJECT_NAME})
 ## output
 set_property(TARGET Tile PROPERTY
