--- a/plugin/percona-pam-for-mysql/src/auth_pam_common.h	2012-06-24 11:50:56.000000000 -0700
+++ b/plugin/percona-pam-for-mysql/src/auth_pam_common.h	2012-06-24 13:47:31.000000000 -0700
@@ -31,7 +31,11 @@
 
 #include <security/pam_appl.h>
 #include <security/pam_modules.h>
+#if HAVE_SECURITY_PAM_MISC_H
 #include <security/pam_misc.h>
+#elif HAVE_SECURITY_OPENPAM_H
+#include <security/openpam.h>
+#endif
 
 #include <mysql/plugin.h>
 #include <mysql/plugin_auth.h>
