--- code/hardware/eeg/pulse/Makefile.orig	2007-04-25 06:06:35.000000000 +1000
+++ code/hardware/eeg/pulse/Makefile	2011-01-02 04:11:07.000000000 +1100
@@ -32,9 +32,9 @@
 LIBPY=$(shell python -c "import distutils.sysconfig;print distutils.sysconfig.get_config_var('LIBPL')")
 PYTHON_INC = $(INCLUDEPY)
 PYTHON_LIB = $(LIBPY)
-FLAGS1 = -Wno-long-double -DHAVE_CONFIG_H
-FLAGS2 = -bundle -undefined suppress -flat_namespace -lpthread
-FLAGS3 = -fno-strict-aliasing -Wno-long-double -no-cpp-precomp	\
+FLAGS1 = -DHAVE_CONFIG_H
+FLAGS2 = -bundle -undefined dynamic_lookup -lpthread
+FLAGS3 = -fno-strict-aliasing	\
 -mno-fused-madd -fno-common -dynamic -DNDEBUG -O2 -Wall	
 
 _awCard.so: awCard.o awCard.i 
--- code/hardware/rt/Makefile.orig	2005-11-19 03:40:54.000000000 +1100
+++ code/hardware/rt/Makefile	2011-01-02 04:18:37.000000000 +1100
@@ -27,11 +27,11 @@
 
 _realtime.so: realtime.o realtime.i 
 	swig -c++ -python realtime.i
-	g++ -c  -Wno-long-double realtime_wrap.cxx -DHAVE_CONFIG_H -I$(INCLUDEPY)
-	g++ -bundle -undefined suppress -flat_namespace realtime.o realtime_wrap.o -o _realtime.so -lpthread
+	g++ -c realtime_wrap.cxx -DHAVE_CONFIG_H -I$(INCLUDEPY)
+	g++ -bundle -undefined dynamic_lookup realtime.o realtime_wrap.o -o _realtime.so -lpthread
 
 realtime.o: realtime.cpp
-	g++ -fno-strict-aliasing -Wno-long-double -no-cpp-precomp -mno-fused-madd -fno-common -dynamic -DNDEBUG -g -O3 -Wall -Wstrict-prototypes -D__MACOSX_CORE__ -I$(INCLUDEPY) -c realtime.cpp -o realtime.o
+	g++ -fno-strict-aliasing -mno-fused-madd -fno-common -dynamic -DNDEBUG -g -O3 -Wall -Wstrict-prototypes -D__MACOSX_CORE__ -I$(INCLUDEPY) -c realtime.cpp -o realtime.o
 
 
 endif
--- code/hardware/sound/Makefile.orig	2011-01-02 04:09:55.000000000 +1100
+++ code/hardware/sound/Makefile	2011-01-02 04:23:09.000000000 +1100
@@ -59,9 +59,9 @@
 LIBPY=$(shell python -c "import distutils.sysconfig;print distutils.sysconfig.get_config_var('LIBPL')")
 PYTHON_INC = $(INCLUDEPY)
 PYTHON_LIB = $(LIBPY)
-FLAGS1 = -Wno-long-double -DHAVE_CONFIG_H
-FLAGS2 = -bundle -undefined suppress -flat_namespace -lpthread -framework CoreAudio
-FLAGS3 = -fno-strict-aliasing -Wno-long-double -no-cpp-precomp	\
+FLAGS1 = -DHAVE_CONFIG_H
+FLAGS2 = -bundle -undefined dynamic_lookup -lpthread -framework CoreAudio
+FLAGS3 = -fno-strict-aliasing	\
 -mno-fused-madd -fno-common -dynamic -DNDEBUG -O2 -Wall	
 
 _eplSound.so: eplSound.o eplSound.i RtAudio.o fifo.o
@@ -84,7 +84,7 @@
 _soundFile.so: soundFile.i soundFile.o
 	swig -c++ -python soundFile.i		
 	g++ -c soundFile_wrap.cxx -DHAVE_CONFIG_H -I$(PYTHON_INC) $(INCLUDEDIR)
-	g++ -bundle -undefined suppress -flat_namespace soundFile_wrap.o soundFile.o -lsndfile -lsamplerate -o _soundFile.so $(LIBDIR) $(INCLUDEDIR)
+	g++ -bundle -undefined dynamic_lookup soundFile_wrap.o soundFile.o -lsndfile -lsamplerate -o _soundFile.so $(LIBDIR) $(INCLUDEDIR)
 
 test_epl: eplSound.o RtAudio.o test_epl.cpp
 	g++ -O2 -I../  -D__MACOSX_CORE__ -o test_epl test_epl.cpp eplSound.o RtAudio.o -lstdc++ -lpthread  -framework CoreAudio
