--- code/hardware/Makefile.old	2009-07-18 16:55:42.000000000 -0700
+++ code/hardware/Makefile	2009-07-18 16:56:15.000000000 -0700
@@ -7,7 +7,7 @@
 # Distributed under the terms of the GNU Lesser General Public License
 # (LGPL). See the license.txt that came with this file.
 
-all: eeg sound vr rt eventpoll.so timing.so keyboard.so joystick.so mouse.so
+all: sound vr rt eventpoll.so timing.so keyboard.so joystick.so mouse.so
 
 eeg: FORCE
 	$(MAKE) -C eeg
@@ -31,7 +31,7 @@
 	gcc -shared eventpoll.o -o eventpoll.so
 
 eventpoll.o: eventpoll.c
-	gcc -c -fPIC -I/usr/include/python2.3/ eventpoll.c
+	gcc -c -fPIC -I@@PYTHON.INC@@ eventpoll.c
 
 eventpoll.c: eventpoll.pyx
 	pyrexc eventpoll.pyx
@@ -40,7 +40,7 @@
 	gcc -shared timing.o -o timing.so
 
 timing.o: timing.c
-	gcc -c -fPIC -I/usr/include/python2.3/ timing.c
+	gcc -c -fPIC -I@@PYTHON.INC@@ timing.c
 
 timing.c: timing.pyx
 	pyrexc timing.pyx
@@ -49,7 +49,7 @@
 	gcc -shared keyboard.o -o keyboard.so
 
 keyboard.o: keyboard.c
-	gcc -c -fPIC -I/usr/include/python2.3/ keyboard.c
+	gcc -c -fPIC -I@@PYTHON.INC@@ keyboard.c
 
 keyboard.c: keyboard.pyx
 	pyrexc keyboard.pyx
@@ -58,7 +58,7 @@
 	gcc -shared joystick.o -o joystick.so
 
 joystick.o: joystick.c
-	gcc -c -fPIC -I/usr/include/python2.3/ joystick.c
+	gcc -c -fPIC -I@@PYTHON.INC@@ joystick.c
 
 joystick.c: joystick.pyx
 	pyrexc joystick.pyx
@@ -67,7 +67,7 @@
 	gcc -shared mouse.o -o mouse.so
 
 mouse.o: mouse.c
-	gcc -c -fPIC -I/usr/include/python2.3/ mouse.c
+	gcc -c -fPIC -I@@PYTHON.INC@@ mouse.c
 
 mouse.c: mouse.pyx
 	pyrexc mouse.pyx
@@ -80,7 +80,7 @@
 	gcc -bundle -flat_namespace -undefined suppress eventpoll.o -o eventpoll.so
 
 eventpoll.o: eventpoll.c
-	gcc -fPIC -c eventpoll.c -I/System/Library/Frameworks/Python.framework/Versions/2.3/include/python2.3//
+	gcc -fPIC -c eventpoll.c -I@@PYTHON.INC@@
 
 eventpoll.c: eventpoll.pyx
 	pyrexc eventpoll.pyx
@@ -89,7 +89,7 @@
 	gcc -bundle -flat_namespace -undefined suppress timing.o -o timing.so
 
 timing.o: timing.c
-	gcc -fPIC -c timing.c -I/System/Library/Frameworks/Python.framework/Versions/2.3/include/python2.3//
+	gcc -fPIC -c timing.c -I@@PYTHON.INC@@
 
 timing.c: timing.pyx
 	pyrexc timing.pyx
@@ -98,7 +98,7 @@
 	gcc -bundle -flat_namespace -undefined suppress keyboard.o -o keyboard.so
 
 keyboard.o: keyboard.c
-	gcc -fPIC -c keyboard.c -I/System/Library/Frameworks/Python.framework/Versions/2.3/include/python2.3//
+	gcc -fPIC -c keyboard.c -I@@PYTHON.INC@@
 
 keyboard.c: keyboard.pyx
 	pyrexc keyboard.pyx
@@ -107,7 +107,7 @@
 	gcc -bundle -flat_namespace -undefined suppress joystick.o -o joystick.so
 
 joystick.o: joystick.c
-	gcc -fPIC -c joystick.c -I/System/Library/Frameworks/Python.framework/Versions/2.3/include/python2.3//
+	gcc -fPIC -c joystick.c -I@@PYTHON.INC@@
 
 joystick.c: joystick.pyx
 	pyrexc joystick.pyx
@@ -116,7 +116,7 @@
 	gcc -bundle -flat_namespace -undefined suppress mouse.o -o mouse.so
 
 mouse.o: mouse.c
-	gcc -fPIC -c mouse.c -I/System/Library/Frameworks/Python.framework/Versions/2.3/include/python2.3//
+	gcc -fPIC -c mouse.c -I@@PYTHON.INC@@
 
 mouse.c: mouse.pyx
 	pyrexc mouse.pyx
