--- src/video/cocoa/cocoa_v.mm.orig	2013-12-25 20:47:32.000000000 +0100
+++ src/video/cocoa/cocoa_v.mm	2013-12-25 20:48:13.000000000 +0100
@@ -398,7 +398,9 @@
 		ret = QZ_CreateWindowSubdriver(width, height, bpp);
 		if (ret != NULL && fullscreen) ret->ToggleFullscreen();
 	} else {
+#if (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_9)
 		ret = fullscreen ? QZ_CreateFullscreenSubdriver(width, height, bpp) : QZ_CreateWindowSubdriver(width, height, bpp);
+#endif /* (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_9) */
 	}
 
 	if (ret != NULL) return ret;
