Upstream-Status: Submitted [https://bugs.openttd.org/task/6380]
--- src/video/cocoa/wnd_quartz.mm.orig	2015-10-18 18:11:44.000000000 +0200
+++ src/video/cocoa/wnd_quartz.mm	2015-10-18 18:14:30.000000000 +0200
@@ -110,12 +110,7 @@
 	static CGColorSpaceRef colorSpace = NULL;
 
 	if (colorSpace == NULL) {
-		CMProfileRef sysProfile;
-
-		if (CMGetSystemProfile(&sysProfile) == noErr) {
-			colorSpace = CGColorSpaceCreateWithPlatformColorSpace(sysProfile);
-			CMCloseProfile(sysProfile);
-		} else {
+		if (!(colorSpace = CGDisplayCopyColorSpace(CGMainDisplayID()))) {
 			colorSpace = CGColorSpaceCreateDeviceRGB();
 		}
 
