--- src/string.cpp.orig	2019-07-07 13:57:48.000000000 -0700
+++ src/string.cpp	2019-10-04 03:42:57.000000000 -0700
@@ -33,10 +33,6 @@
 #include "os/windows/string_uniscribe.h"
 #endif
 
-#if defined(WITH_COCOA)
-#include "os/macosx/string_osx.h"
-#endif
-
 #ifdef WITH_ICU_SORT
 /* Required by strnatcmp. */
 #include <unicode/ustring.h>
@@ -44,6 +40,10 @@
 #include "gfx_func.h"
 #endif /* WITH_ICU_SORT */
 
+#if defined(WITH_COCOA)
+#include "os/macosx/string_osx.h"
+#endif
+
 /* The function vsnprintf is used internally to perform the required formatting
  * tasks. As such this one must be allowed, and makes sure it's terminated. */
 #include "safeguards.h"
