--- cmake/FindSpinlock.cmake.orig	2013-12-19 01:25:15.000000000 +0400
+++ cmake/FindSpinlock.cmake	2013-12-19 01:34:27.000000000 +0400
@@ -1,10 +1,10 @@
 IF (CMAKE_USE_PTHREADS_INIT)
 	SET(_bindir "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}")
 	IF (DARWIN)
-		TRY_COMPILE(HAVE_SPINLOCK "${_bindir}" "${CMAKE_MODULE_PATH}/spinlock-test-darwin.cpp")
+		TRY_COMPILE(HAVE_SPINLOCK "${_bindir}" "${CMAKE_CURRENT_LIST_DIR}/spinlock-test-darwin.cpp")
 	ELSE ()
 		TRY_COMPILE(HAVE_SPINLOCK "${_bindir}"
-			"${CMAKE_MODULE_PATH}/spinlock-test.cpp"
+			"${CMAKE_CURRENT_LIST_DIR}/spinlock-test.cpp"
 			CMAKE_FLAGS "-DLINK_LIBRARIES:STRING=${CMAKE_THREAD_LIBS_INIT}"
 		)
 	ENDIF ()
