--- Makefile.orig	2005-10-09 13:00:29.000000000 -0700
+++ Makefile	2008-11-29 08:45:49.000000000 -0800
@@ -1,13 +1,13 @@
 
-MANDIR=$(DESTDIR)/usr/share/man/man6/
-BINDIR=$(DESTDIR)/usr/games/
+MANDIR=$(DESTDIR)__PREFIX__/share/man/man6
+BINDIR=$(DESTDIR)__PREFIX__/bin
 
-CC=gcc
+CC=__CC__
 MKDIR=mkdir -p
 INSTALL=install
 
-CFLAGS=-Wall -fomit-frame-pointer -O3
-LIBS=-lm -lncurses
+CFLAGS=-Wall __CFLAGS__
+LIBS=-lncurses
 
 OBJS=ctris.o game.o screen.o brick.o highscore.o
 OUTPUT=ctris
@@ -35,7 +35,7 @@
 	$(RM) $(OBJS) $(OUTPUT) config.h
 
 
-install:
+install: all
 	$(MKDIR) $(BINDIR) $(MANDIR)
 	$(INSTALL) $(OUTPUT) $(BINDIR)
 	$(INSTALL) -m 644 $(MANS) $(MANDIR)
