--- src/3rdparty/chromium/base/mac/sdk_forward_declarations.h.orig	2018-11-13 11:25:11.000000000 -0700
+++ src/3rdparty/chromium/base/mac/sdk_forward_declarations.h	2018-12-09 08:42:04.000000000 -0700
@@ -262,8 +262,8 @@
 
 // Once Chrome no longer supports OSX 10.12, everything within this
 // preprocessor block can be removed.
-#if !defined(MAC_OS_X_VERSION_10_13) || \
-    MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_13
+//#if !defined(MAC_OS_X_VERSION_10_13) ||			\
+//    MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_13
 
 // VNRequest forward declarations.
 @class VNRequest;
@@ -328,7 +328,7 @@
 @property(readonly, nonatomic, strong) VNFaceLandmarks2D* landmarks;
 @end
 
-#endif  // MAC_OS_X_VERSION_10_13
+//#endif  // MAC_OS_X_VERSION_10_13
 // ----------------------------------------------------------------------------
 // The symbol for kCWSSIDDidChangeNotification is available in the
 // CoreWLAN.framework for OSX versions 10.6 through 10.10. The symbol is not
