diff --git src/3rdparty/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc src/3rdparty/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
index e41a894fc..be79bf7cf 100644
--- src/3rdparty/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
+++ src/3rdparty/chromium/third_party/blink/renderer/core/editing/finder/text_finder.cc
@@ -629,7 +629,10 @@ gfx::RectF TextFinder::ActiveFindMatchRect() {
   if (!current_active_match_frame_ || !active_match_)
     return gfx::RectF();
 
-  return gfx::RectF(FindInPageRectFromRange(EphemeralRange(ActiveMatch())));
+//  return gfx::RectF(FindInPageRectFromRange(EphemeralRange(ActiveMatch())));
+  CGRect r = FindInPageRectFromRange(EphemeralRange(ActiveMatch()));
+  return gfx::RectF(r);
+
 }
 
 Vector<gfx::RectF> TextFinder::FindMatchRects() {
@@ -639,7 +642,10 @@ Vector<gfx::RectF> TextFinder::FindMatchRects() {
   match_rects.ReserveCapacity(match_rects.size() + find_matches_cache_.size());
   for (const FindMatch& match : find_matches_cache_) {
     DCHECK(!match.rect_.IsEmpty());
-    match_rects.push_back(match.rect_);
+//    match_rects.push_back(match.rect_);
+  CGRect r = match.rect_;
+  match_rects.push_back(r);
+
   }
 
   return match_rects;
diff --git src/3rdparty/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc src/3rdparty/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
index 48d06120e..e7963b327 100644
--- src/3rdparty/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
+++ src/3rdparty/chromium/third_party/blink/renderer/core/paint/paint_timing_detector.cc
@@ -320,7 +320,9 @@ FloatRect PaintTimingDetector::BlinkSpaceToDIPs(
   // May be nullptr in tests.
   if (!widget)
     return float_rect;
-  return FloatRect(widget->BlinkSpaceToDIPs(gfx::RectF(float_rect)));
+//  return FloatRect(widget->BlinkSpaceToDIPs(gfx::RectF(float_rect)));
+  CGRect r = float_rect;
+  return FloatRect(widget->BlinkSpaceToDIPs(gfx::RectF(r)));
 }
 
 FloatRect PaintTimingDetector::CalculateVisualRect(
diff --git src/3rdparty/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc src/3rdparty/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
index f7a348c81..45de8d332 100644
--- src/3rdparty/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
+++ src/3rdparty/chromium/third_party/blink/renderer/modules/exported/web_ax_object.cc
@@ -1480,7 +1480,9 @@ void WebAXObject::GetRelativeBounds(WebAXObject& offset_container,
   private_->GetRelativeBounds(&container, bounds, container_transform,
                               clips_children);
   offset_container = WebAXObject(container);
-  bounds_in_container = gfx::RectF(bounds);
+  //bounds_in_container = gfx::RectF(bounds);
+  CGRect r = bounds;
+  bounds_in_container = gfx::RectF(r);
 }
 
 void WebAXObject::GetAllObjectsWithChangedBounds(
