--- src/buildtools/config/mac_osx.pri.orig	2020-03-24 02:16:30.000000000 -0700
+++ src/buildtools/config/mac_osx.pri	2020-04-27 18:56:04.000000000 -0700
@@ -28,7 +28,7 @@
     clang_base_path=\"$${QMAKE_CLANG_DIR}\" \
     clang_use_chrome_plugins=false \
     mac_deployment_target=\"$${QMAKE_MACOSX_DEPLOYMENT_TARGET}\" \
-    mac_sdk_min=\"$${QMAKE_MAC_SDK_VERSION}\" \
+    mac_sdk_min=\"$$section(QMAKE_MAC_SDK_VERSION, ".", 0, 1)\" \
     use_external_popup_menu=false
 
 qtConfig(webengine-spellchecker) {
