diff --git a/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h b/qtwebengine/src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
index e45ab43..1ba35cb 100644
--- src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
+++ src/3rdparty/chromium/base/mac/sdk_forward_declarations.h
@@ -15,6 +15,7 @@
 #import <CoreWLAN/CoreWLAN.h>
 #import <ImageCaptureCore/ImageCaptureCore.h>
 #import <IOBluetooth/IOBluetooth.h>
+#import <CoreBluetooth/CoreBluetooth.h>
 
 #include "base/base_export.h"
 
