--- src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm.orig	2017-10-02 01:43:38.000000000 -0700
+++ src/platformsupport/fontdatabases/mac/qfontengine_coretext.mm	2017-10-24 16:36:13.000000000 -0700
@@ -830,7 +830,7 @@
 
 QFixed QCoreTextFontEngine::emSquareSize() const
 {
-    return QFixed::QFixed(int(CTFontGetUnitsPerEm(ctfont)));
+    return QFixed(int(CTFontGetUnitsPerEm(ctfont)));
 }
 
 QFontEngine *QCoreTextFontEngine::cloneWithSize(qreal pixelSize) const
