--- configure.orig	2022-08-05 21:19:35.000000000 +0800
+++ configure	2023-02-03 01:48:20.000000000 +0800
@@ -2898,17 +2898,17 @@
 esac
 case $includedir in
    \/* )   eval "INSTALL_INCLUDEDIR=$includedir";;
-   * ) eval "INSTALL_INCLUDEDIR=$INSTALL_DIR/include";;
+   * ) eval "INSTALL_INCLUDEDIR=$INSTALL_DIR/include/psblas";;
 esac
 case $docsdir in
    \/* )   eval "INSTALL_DOCSDIR=$docsdir";;
-   * ) eval "INSTALL_DOCSDIR=$INSTALL_DIR/docs";;
+   * ) eval "INSTALL_DOCSDIR=$INSTALL_DIR/share/psblas/docs";;
 esac
 case $samplesdir in
    \/* )   eval "INSTALL_SAMPLESDIR=$samplesdir";;
-   * ) eval "INSTALL_SAMPLESDIR=$INSTALL_DIR/samples";;
+   * ) eval "INSTALL_SAMPLESDIR=$INSTALL_DIR/share/psblas/samples";;
 esac
-INSTALL_MODULESDIR=$INSTALL_DIR/modules
+INSTALL_MODULESDIR=$INSTALL_DIR/include/psblas
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INSTALL_DIR $INSTALL_INCLUDEDIR $INSTALL_MODULESDIR $INSTALL_LIBDIR $INSTALL_DOCSDIR $INSTALL_SAMPLESDIR" >&5
 $as_echo "$INSTALL_DIR $INSTALL_INCLUDEDIR $INSTALL_MODULESDIR $INSTALL_LIBDIR $INSTALL_DOCSDIR $INSTALL_SAMPLESDIR" >&6; }
 
