diff -ur src.orig/http_query.cc src/http_query.cc
--- src.orig/http_query.cc	2014-08-03 13:33:49.000000000 -0700
+++ src/http_query.cc	2014-08-03 13:34:33.000000000 -0700
@@ -33,8 +33,6 @@
 #include <stdlib.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
 
 #include "http_query.h"
 #include "ps_config.h"
diff -ur src.orig/http_query.h src/http_query.h
--- src.orig/http_query.h	2014-08-03 13:33:49.000000000 -0700
+++ src/http_query.h	2014-08-03 13:35:07.000000000 -0700
@@ -36,7 +36,6 @@
 #include <iostream>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 
 class HttpQuery
 {
diff -ur src.orig/vantaged.cc src/vantaged.cc
--- src.orig/vantaged.cc	2014-08-03 13:33:49.000000000 -0700
+++ src/vantaged.cc	2014-08-03 13:35:29.000000000 -0700
@@ -52,8 +52,6 @@
 #include <vector>
 
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
 
 #include "ps_poller.h"
 #include "ps_defs.h"
