--- ltmain.sh.orig	2011-10-02 06:54:21.000000000 -0500
+++ ltmain.sh	2011-10-02 06:54:58.000000000 -0500
@@ -979,6 +979,13 @@
 	  prev=
 	  continue
 	  ;;
+        xarch)
+	  linkopts="$linkopts $arg"
+	  compile_command="$compile_command $arg"
+	  finalize_command="$finalize_command $arg"
+	  prev=
+	  continue
+          ;;
 	*)
 	  eval "$prev=\"\$arg\""
 	  prev=
@@ -1004,6 +1011,16 @@
 	continue
 	;;
 
+      # Darwin uses the -arch flag to determine output architecture.
+      # Add it to the link options and compile and finalize commands.
+      -arch)
+	linkopts="$linkopts $arg"
+	compile_command="$compile_command $arg"
+	finalize_command="$finalize_command $arg"
+	prev=xarch
+	continue
+	;;
+
       -avoid-version)
 	avoid_version=yes
 	continue
