--- Makefile.rev2	2022-07-29 11:13:10.000000000 -0400
+++ Makefile	2022-07-29 11:26:03.000000000 -0400
@@ -1,8 +1,8 @@
 CXX := clang++
 CC  := clang
-OBJ := main.o UnAlz.o UnAlzUtils.o UnAlzBz2decompress.o UnAlzBzip2.o UnAlzbzlib.o bzip2/blocksort.o bzip2/compress.o bzip2/crctable.o bzip2/huffman.o bzip2/randtable.o
+OBJ := main.o UnAlz.o UnAlzUtils.o UnAlzBz2decompress.o UnAlzBzip2.o
 BIN := unalz
-LDFLAGS += -lz
+LDFLAGS += -lz -lbz2
 CFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 CXXFLAGS += -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 
 
--- UnAlzBzip2.cpp.rev1	2022-07-29 11:17:40.000000000 -0400
+++ UnAlzBzip2.cpp	2022-07-29 11:32:37.000000000 -0400
@@ -11,10 +11,20 @@
 
 //#include "stdafx.h"
 #include <zlib.h>
-#include "bzip2/bzlib.h"
-#include "bzip2/bzlib_private.h"
+#include <bzlib.h>
 #include "UnAlz.h"
 
+// from bzlib_private.h of bzip2
+typedef char            Char;
+typedef unsigned char   Bool;
+typedef unsigned char   UChar;
+typedef int             Int32;
+typedef unsigned int    UInt32;
+typedef short           Int16;
+typedef unsigned short  UInt16;
+
+#define True  ((Bool)1)
+#define False ((Bool)0)
 
 typedef struct {
   CUnAlz*    handle;
