--- UnAlz.cpp.rev2	2022-07-29 12:03:49.000000000 -0400
+++ UnAlz.cpp	2022-07-29 12:10:15.000000000 -0400
@@ -431,6 +431,11 @@
     zipHeader.uncompressedSize      =   unalz_le64toh(zipHeader.uncompressedSize); 
 
 	// FILE NAME
+	if(zipHeader.head.fileNameLength<=0)
+	{
+			m_nErr = ERR_INVALID_FILENAME_LENGTH;
+			return FALSE;
+	}		
 	zipHeader.fileName = (char*)malloc(zipHeader.head.fileNameLength+sizeof(char));
 	if(zipHeader.fileName==NULL)
 	{
