--- Makefile.rev3	2022-07-30 09:14:01.000000000 -0400
+++ Makefile	2022-07-30 10:04:51.000000000 -0400
@@ -25,19 +25,19 @@
 
 posix: unalz
 	$(CXX) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp -D_UNALZ_ICONV $(CXXFLAGS) 
-	$(CXX) $(OBJ) $(LDFLAGS) $(CXXFLAGS) -liconv -o $(BIN) 
+	$(CXX) $(OBJ) $(LDFLAGS) -liconv -o $(BIN) 
 
 posix-utf8: unalz
 	$(CXX) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp -D_UNALZ_ICONV -D_UNALZ_UTF8 $(CXXFLAGS)
-	$(CXX) $(OBJ) $(LDFLAGS) $(CXXFLAGS) -liconv -o $(BIN)
+	$(CXX) $(OBJ) $(LDFLAGS) -liconv -o $(BIN)
 
 posix-noiconv: unalz
 	$(CXX) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp $(CXXFLAGS)
-	$(CXX) $(OBJ) $(LDFLAGS) $(CXXFLAGS) -o $(BIN)
+	$(CXX) $(OBJ) $(LDFLAGS) -o $(BIN)
 
 linux-utf8: unalz
 	$(CXX) -c UnAlz.cpp -c UnAlzUtils.cpp -c main.cpp -D_UNALZ_ICONV -D_UNALZ_UTF8 $(CXXFLAGS)
-	$(CXX) $(OBJ) $(LDFLAGS) $(CXXFLAGS) -o $(BIN)
+	$(CXX) $(OBJ) $(LDFLAGS) -o $(BIN)
 
 install:
 	cp unalz /usr/local/bin/
