--- setup.py.orig	2026-03-27 15:49:51
+++ setup.py	2026-03-27 15:52:03
@@ -203,7 +203,7 @@
            and not platform.uname()[4].startswith(('arm', 'aarch64')):
             # Raspberry Pi doesn't have or need this switch
             compflags = ['-m{0}'.format(bit)] + compflags
-        comp_candidates = ['gfortran']
+        comp_candidates = ['@@MP_FC@@']
         if sys.platform == 'win32':
             comp_candidates.insert(0, 'gfortran.exe')
         if 'FC' in os.environ:
@@ -409,7 +409,7 @@
     if not os.path.exists(outlibdir):
         os.makedirs(outlibdir, exist_ok=True)
     for f in libneeded:
-        p = subprocess.Popen(['gfortran', f'--print-file-name={f}.dylib'],
+        p = subprocess.Popen(['@@MP_FC@@', f'--print-file-name={f}.dylib'],
                          stdout=subprocess.PIPE)
         out, err = p.communicate()
         if p.returncode:
