--- mindy/compat/Makefile.in.orig	Sat Oct  4 13:20:58 2003
+++ mindy/compat/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -49,7 +49,7 @@
 AR      =               ar
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DTARGET=\"@d2c_target@\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DTARGET=\"@d2c_target@\"
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
