diff -Naur phpext.c.orig phpext.c
--- phpext.c.orig	2003-09-24 08:36:09.000000000 -0600
+++ phpext.c	2007-04-02 09:01:29.000000000 -0600
@@ -21,7 +21,7 @@
 static double zero()    { return 0.0; }
 static double one() { return 1.0; }
 static double inf() { return one() / zero(); }
-static double nan() { return zero() / zero(); }
+static double nanphp() { return zero() / zero(); }
 
 /* {{{ syck_functions[]
  *
@@ -220,7 +220,7 @@
             }
             else if ( strcmp( n->type_id, "float#nan" ) == 0 )
             {
-                ZVAL_DOUBLE( o, nan() );
+                ZVAL_DOUBLE( o, nanphp() );
             }
             else if ( strcmp( n->type_id, "float#inf" ) == 0 )
             {
