--- docs/conf/httpd.conf.in.orig	2012-02-06 10:54:24.000000000 -0600
+++ docs/conf/httpd.conf.in	2013-01-03 02:10:50.000000000 -0600
@@ -169,14 +169,24 @@
 </IfModule>
 
 #
-# The following lines prevent .htaccess and .htpasswd files from being 
-# viewed by Web clients. 
+# The following lines prevent .htaccess, .htpasswd and .DS_Store files and
+# Mac resource forks and named forks from being viewed by Web clients.
 #
-<FilesMatch "^\.ht">
+<Files ~ "^\.([Hh][Tt]|[Dd][Ss]_[Ss])">
     Order allow,deny
     Deny from all
     Satisfy All
-</FilesMatch>
+</Files>
+<Files "rsrc">
+    Order allow,deny
+    Deny from all
+    Satisfy All
+</Files>
+<DirectoryMatch ".*\.\.namedfork">
+    Order allow,deny
+    Deny from all
+    Satisfy All
+</DirectoryMatch>
 
 #
 # ErrorLog: The location of the error log file.
