--- Deriv.pro.orig	2023-06-26 08:09:03.000000000 +0800
+++ Deriv.pro	2023-06-26 08:14:14.000000000 +0800
@@ -88,12 +88,12 @@
     dreditconnectionwindow.ui \
     drserverinfodialog.ui
 
-macx: LIBS += -L$$PWD/libwired/osx/lib/ -L$$PWD/qtkeychain-build -lwired -lxml2 -lssl -lcrypto -lsqlite3 -lz -liconv -lqt5keychain -framework Carbon
-macx: INCLUDEPATH += libwired/osx/include
+macx: LIBS += -L@PREFIX@/libexec/openssl10/lib -lssl -lcrypto @PREFIX@/lib/libwired.a -L@PREFIX@/lib -lqtkeychain -lxml2 -lsqlite3 -lz -liconv -lncurses -framework Carbon
+macx: INCLUDEPATH += @PREFIX@/include/wired
 macx: INCLUDEPATH += qtkeychain-build
 
-unix: LIBS += -L$$PWD/libwired/linux/lib/ -L$$PWD/qtkeychain-build -lwired -lxml2 -lssl -lcrypto -lsqlite3 -lz -lqt5keychain
-unix: INCLUDEPATH += libwired/linux/include
+unix: LIBS += -L@PREFIX@/libexec/openssl10/lib -lssl -lcrypto @PREFIX@/lib/libwired.a -L@PREFIX@/lib -lqtkeychain -lxml2 -lsqlite3 -lz -lncurses
+unix: INCLUDEPATH += @PREFIX@/include/wired
 unix: INCLUDEPATH += qtkeychain-build
 
 RESOURCES += \

