--- hints/darwin.sh.orig	2017-09-16 15:49:34.000000000 -0700
+++ hints/darwin.sh	2017-09-16 15:53:51.000000000 -0700
@@ -346,7 +346,9 @@
     prodvers_minor=$(echo $prodvers|awk -F. '{print $2}')
 
     # macOS (10.12) deprecated syscall().
-    if [ "$prodvers_minor" -ge 12 ]; then
+    # but it's still available
+    # for compatibility with perl5.24 enable syscall() on Sierra
+    if [ "$prodvers_minor" -ge 13 ]; then
         d_syscall='undef'
     fi
 
