--- Makefile~	2004-09-08 13:31:40.000000000 -0700
+++ Makefile	2007-01-22 14:27:09.000000000 -0800
@@ -6,12 +6,12 @@
 #X11_PATH= -L/usr/X11/lib
 #X11_PATH= -L/usr/X11R6/lib
 
-OGL_LIBS= -lGL -lm
+OGL_LIBS= -framework OpenGL -lm
 #OGL_LIBS= -lm                                                # Think Different
 
 #------------------------------------------------------------------------------
 
-CFLAGS= -Wall -O3 -ansi $(shell sdl-config --cflags)
+CFLAGS= -Wall -O3 -ansi $(shell sdl-config --cflags) -FOpenGL
 #CFLAGS= -Wall -g -ansi $(shell sdl-config --cflags)
 #CFLAGS= -Wall -pg -ansi $(shell sdl-config --cflags)
 
