--- configure.ac.orig	2010-11-13 04:24:43.000000000 -0800
+++ configure.ac	2013-05-13 13:57:40.000000000 -0700
@@ -92,7 +92,7 @@
   [AS_HELP_STRING([--with-smclient-backend],[which smclient backend to use (xsmp/win32/quartz)])],
   [],
   [case "$GDK_TARGET" in
-    x11) with_smclient=xsmp SMCLIENT_PKGS="sm" ;;
+    x11) with_smclient=xsmp SMCLIENT_PKGS="sm ice" ;;
     win32|quartz) with_smclient=$GDK_TARGET ;;
     *) with_smclient=no ;;
    esac])
