--- src/terminal-screen.c.orig	2014-08-17 09:47:20.000000000 -0700
+++ src/terminal-screen.c	2014-08-18 22:32:33.000000000 -0700
@@ -27,7 +27,7 @@
 #include <unistd.h>
 #include <sys/wait.h>
 #include <fcntl.h>
-#include <uuid.h>
+#include <uuid/uuid.h>
 
 #include <glib.h>
 #include <glib/gi18n.h>
@@ -1199,7 +1199,12 @@
       for (j = 0; j < n_fds; j++) {
         if (fds[j] == target_fd) {
           do {
+#ifdef F_DUPFD_CLOEXEC
             fd = fcntl (fds[j], F_DUPFD_CLOEXEC, 3);
+#else
+            fd = fcntl (fds[j], F_DUPFD, 3);
+            fcntl(fd, F_SETFD, FD_CLOEXEC);
+#endif
           } while (fd == -1 && errno == EINTR);
           if (fd == -1)
             _exit (127);
