--- src/terminal.c.orig	2009-06-29 06:41:51.000000000 -0700
+++ src/terminal.c	2009-06-29 13:09:42.000000000 -0700
@@ -450,12 +450,13 @@
    * session bus running), we'd crash later on. Tell the user about it
    * now, and exit. See bug #561663.
    */
+/*
   if (!gconf_ping_daemon ())
     {
       g_printerr ("Failed to contact the GConf daemon; exiting.\n");
       exit (1);
     }
-
+*/
   gtk_window_set_default_icon_name (GNOME_TERMINAL_ICON_NAME);
 
   gtk_about_dialog_set_url_hook (about_url_hook, NULL, NULL);
