--- dao/Makefile.in.orig	Sun Jun  6 19:57:21 2004
+++ dao/Makefile.in	Sat Nov 27 14:51:20 2004
@@ -698,7 +698,7 @@
 
 
 install-data-hook:
-	mv $(CDRDAO_DATA_DIR)/cdrdao.drivers $(DRIVER_TABLE_FILE)
+	mv $(DESTDIR)$(CDRDAO_DATA_DIR)/cdrdao.drivers $(DESTDIR)$(DRIVER_TABLE_FILE)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
