From f929eff2d940a553f2eb63a14d029256c0664557 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 24 May 2024 23:20:24 +0800
Subject: [PATCH] CMakeLists: fix optflags for PowerPC case

---
 CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index c9a5c6a..00e4781 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -39,7 +39,7 @@ message(STATUS "xvega v${${PROJECT_NAME}_VERSION}")
 # Build options
 # =============
 
-OPTION(XVEGA_DISABLE_ARCH_NATIVE "disable -march=native flag" OFF)
+OPTION(XVEGA_DISABLE_OPT_NATIVE "disable -march=native / -mtune=native flag" OFF)
 OPTION(XVEGA_DISABLE_TUNE_GENERIC "disable -mtune=generic flag" OFF)
 
 OPTION(XVEGA_BUILD_STATIC "Build xvega static library" ON)
@@ -361,11 +361,17 @@ macro(xvega_set_common_options target_name)
         target_compile_options(${target_name} PUBLIC -Wunused-parameter -Wextra -Wreorder)
     
         # Mtune generic/native
-        if (XVEGA_DISABLE_ARCH_NATIVE AND NOT XVEGA_DISABLE_TUNE_GENERIC)
+        if (XVEGA_DISABLE_OPT_NATIVE AND NOT XVEGA_DISABLE_TUNE_GENERIC)
             target_compile_options(${target_name} PUBLIC -mtune=generic)
         elseif (XVEGA_DISABLE_TUNE_GENERIC)
+            # Do not pass optflags.
         else ()
-            target_compile_options(${target_name} PUBLIC -march=native)
+            # PowerPC does not support -march
+            if (CMAKE_SYSTEM_PROCESSOR MATCHES "ppc|powerpc")
+                target_compile_options(${target_name} PUBLIC -mtune=native)
+            else ()
+                target_compile_options(${target_name} PUBLIC -march=native)
+            endif ()
         endif ()
 
         # C++14 flag
