--- Modules/_testexternalinspection.c
+++ Modules/_testexternalinspection.c
@@ -17,10 +17,18 @@
 
 #if defined(__APPLE__)
 #  include <TargetConditionals.h>
+#  include <AvailabilityMacros.h>
 // Older macOS SDKs do not define TARGET_OS_OSX
 #  if !defined(TARGET_OS_OSX)
 #     define TARGET_OS_OSX 1
 #  endif
+
+// Tiger does not have libproc, so use the non-osx fallback
+#  if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#    undef    TARGET_OS_OSX
+#    define   TARGET_OS_OSX 0
+#  endif
+
 #  if TARGET_OS_OSX
 #    include <libproc.h>
 #    include <mach-o/fat.h>
