--- cairomm/context.cc.orig	2008-03-11 04:36:32.000000000 +0100
+++ cairomm/context.cc	2008-04-19 07:20:31.000000000 +0200
@@ -21,8 +21,17 @@
 #include <cairomm/private.h>
 #include <cairomm/surface.h>
 #include <cairomm/win32_surface.h>
-#include <cairomm/xlib_surface.h>
+
+#define Cursor Quartz_Cursor
+#define Picture Quartz_Picture
+#define Status Quartz_Status 
 #include <cairomm/quartz_surface.h>
+#undef Cursor
+#undef Picture
+#undef Status
+
+#include <cairomm/xlib_surface.h>
+
 
 /* M_PI is defined in math.h in the case of Microsoft Visual C++ */
 #if defined(_MSC_VER)
