--- cairomm/context.cc.orig	2008-03-11 04:36:32.000000000 +0100
+++ cairomm/context.cc	2008-04-17 06:36:51.000000000 +0200
@@ -21,7 +21,11 @@
 #include <cairomm/private.h>
 #include <cairomm/surface.h>
 #include <cairomm/win32_surface.h>
+#define Cursor X11_Cursor
+#define Picture X11_Picture
 #include <cairomm/xlib_surface.h>
+#undef Cursor
+#undef Picture
 #include <cairomm/quartz_surface.h>
 
 /* M_PI is defined in math.h in the case of Microsoft Visual C++ */
