--- configure.ac.orig	2016-09-22 16:09:36.000000000 -0700
+++ configure.ac	2016-09-22 16:10:42.000000000 -0700
@@ -256,18 +256,7 @@
 # ***********
 
 CC_CHECK_CFLAGS_APPEND([ \
-  -Wall -Wextra \
-  -Wformat-nonliteral -Werror=format-security \
-  -Wsign-compare -Werror=implicit-function-declaration \
-  -Wpointer-arith -Wwrite-strings -Wstrict-prototypes \
-  -Waggregate-return -Wcast-align -Wimplicit -Wuninitialized \
-  -Wmissing-prototypes -Wmissing-declarations -Wnested-externs \
-  -Wpacked -Wmissing-format-attribute -Wshadow \
-  -Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
-  -Wdeclaration-after-statement -Wold-style-definition \
-  -Wno-missing-field-initializers -Wno-unused-parameter \
-  -Wformat-signedness \
-  -fno-common -fno-strict-aliasing -Wno-switch-enum])
+  -Wall -fno-common -fno-strict-aliasing])
 
 # ********
 # Bindings
