diff -Nurd kdeaddons-3.5.9/atlantikdesigner/designer/Makefile.am kdeaddons-3.5.9-new/atlantikdesigner/designer/Makefile.am
--- kdeaddons-3.5.9/atlantikdesigner/designer/Makefile.am	2005-09-10 04:14:16.000000000 -0400
+++ kdeaddons-3.5.9-new/atlantikdesigner/designer/Makefile.am	2008-02-13 22:06:55.000000000 -0500
@@ -1,10 +1,14 @@
-bin_PROGRAMS = atlantikdesigner
+
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = atlantikdesigner.la
+
 INCLUDES = -I$(kde_includes)/atlantic $(all_includes)
-AM_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-LDADD = -latlantikui -lkdegames $(LIB_KIO) 
 
 SUBDIRS = 
 
-atlantikdesigner_SOURCES = boardinfo.cpp designer.cpp editor.cpp main.cpp group.cpp
+atlantikdesigner_la_SOURCES = boardinfo.cpp designer.cpp editor.cpp main.cpp group.cpp
+atlantikdesigner_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+atlantikdesigner_la_LIBADD = -latlantikui -lkdegames $(LIB_KIO) 
 
 METASOURCES = AUTO 
diff -Nurd kdeaddons-3.5.9/atlantikdesigner/designer/main.cpp kdeaddons-3.5.9-new/atlantikdesigner/designer/main.cpp
--- kdeaddons-3.5.9/atlantikdesigner/designer/main.cpp	2005-09-10 04:14:16.000000000 -0400
+++ kdeaddons-3.5.9-new/atlantikdesigner/designer/main.cpp	2008-02-13 22:06:55.000000000 -0500
@@ -5,7 +5,7 @@
 
 #include "designer.h"
 
-int main(int argc, char *argv[])
+extern "C" KDE_EXPORT int kdemain(int argc, char *argv[])
 {
 	KAboutData aboutData(
 	           "atlantikdesigner",
diff -Nurd kdeaddons-3.5.9/kfile-plugins/lnk/Makefile.am kdeaddons-3.5.9-new/kfile-plugins/lnk/Makefile.am
--- kdeaddons-3.5.9/kfile-plugins/lnk/Makefile.am	2005-09-10 04:14:17.000000000 -0400
+++ kdeaddons-3.5.9-new/kfile-plugins/lnk/Makefile.am	2008-02-13 22:06:55.000000000 -0500
@@ -2,12 +2,14 @@
 
 INCLUDES         = $(all_includes)
 
-bin_PROGRAMS = lnkforward
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = lnkforward.la
 
-lnkforward_SOURCES = lnkforward.cpp
-lnkforward_LDADD = $(LIB_KDECORE) $(LIB_KIO)
-lnkforward_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-lnkforward_METASOURCES = lnkforward.moc
+lnkforward_la_SOURCES = lnkforward.cpp
+lnkforward_la_LIBADD = $(LIB_KDECORE) $(LIB_KIO)
+lnkforward_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+lnkforward_la_METASOURCES = lnkforward.moc
 
 lnkforward_DATA = lnkforward.desktop
 lnkforwarddir = $(kde_appsdir)/.hidden
diff -Nurd kdeaddons-3.5.9/kfile-plugins/lnk/lnkforward.cpp kdeaddons-3.5.9-new/kfile-plugins/lnk/lnkforward.cpp
--- kdeaddons-3.5.9/kfile-plugins/lnk/lnkforward.cpp	2005-10-10 10:53:15.000000000 -0400
+++ kdeaddons-3.5.9-new/kfile-plugins/lnk/lnkforward.cpp	2008-02-13 22:06:55.000000000 -0500
@@ -50,7 +50,7 @@
 
 //--------------------------------------------------------------------------------
 
-int main(int argc, char **argv)
+extern "C" KDE_EXPORT int kdemain(int argc, char **argv)
 {
   KCmdLineArgs::init(argc, argv, appName, programName, description, version, false);
 
diff -Nurd kdeaddons-3.5.9/ksig/Makefile.am kdeaddons-3.5.9-new/ksig/Makefile.am
--- kdeaddons-3.5.9/ksig/Makefile.am	2005-09-10 04:14:16.000000000 -0400
+++ kdeaddons-3.5.9-new/ksig/Makefile.am	2008-02-13 22:06:55.000000000 -0500
@@ -1,6 +1,9 @@
-bin_PROGRAMS = ksig
 
-ksig_SOURCES = siggenerator.cpp siglistview.cpp siglistviewitem.cpp ksig.cpp \
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = ksig.la
+
+ksig_la_SOURCES = siggenerator.cpp siglistview.cpp siglistviewitem.cpp ksig.cpp \
 	standardtextbase.ui standardtext.cpp main.cpp 
 
 INCLUDES = $(all_includes)
@@ -8,8 +11,8 @@
 METASOURCES = AUTO
 KDE_ICON = AUTO
 
-ksig_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-ksig_LDADD   =  $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
+ksig_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+ksig_la_LIBADD   =  $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI)
 
 rcdir = $(kde_datadir)/ksig
 rc_DATA = ksigui.rc
diff -Nurd kdeaddons-3.5.9/ksig/main.cpp kdeaddons-3.5.9-new/ksig/main.cpp
--- kdeaddons-3.5.9/ksig/main.cpp	2005-09-10 04:14:16.000000000 -0400
+++ kdeaddons-3.5.9-new/ksig/main.cpp	2008-02-13 22:06:55.000000000 -0500
@@ -33,7 +33,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" KDE_EXPORT int kdemain(int argc, char *argv[])
 {
 
     KAboutData aboutData("ksig", I18N_NOOP("KSig"),
