--- ffi-tk-2018.02.20.gemspec
+++ ffi-tk-2018.02.20.gemspec	2024-04-15 12:44:02
@@ -23,10 +23,10 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<ffi>.freeze, ["~> 1.9"])
+    s.add_runtime_dependency(%q<ffi>.freeze, [">= 1.9"])
     s.add_development_dependency(%q<bacon>.freeze, ["~> 1.2"])
   else
-    s.add_dependency(%q<ffi>.freeze, ["~> 1.9"])
+    s.add_dependency(%q<ffi>.freeze, [">= 1.9"])
     s.add_dependency(%q<bacon>.freeze, ["~> 1.2"])
   end
 end
