--- SConstruct.orig	2015-10-16 08:03:33.000000000 +0200
+++ SConstruct	2016-01-02 11:40:21.000000000 +0100
@@ -67,10 +67,10 @@
 env['CPPPATH']        = []
 env['LIBPATH']        = []
 env['RPATH']          = []
-env['WITH_PYTHON']    = 'auto'
-env['WITH_LUA']       = 'auto'
-env['WITH_TBB']       = 'auto'
-env['WITH_CPPUNIT']   = 'auto'
+env['WITH_PYTHON']    = None
+env['WITH_LUA']       = None
+env['WITH_TBB']       = None
+env['WITH_CPPUNIT']   = None
 
 if GetOption("build_dir") == "static":
     env['CXXFLAGS']  = ['-std=c++11', '-O3', '-Wall']
