--- hints/darwin.sh.orig	2013-05-08 11:13:45.000000000 -0600
+++ hints/darwin.sh	2013-05-08 11:15:04.000000000 -0600
@@ -129,7 +129,7 @@
 
 # Avoid Apple's cpp precompiler, better for extensions
 if [ "X`echo | ${cc} -no-cpp-precomp -E - 2>&1 >/dev/null`" = "X" ]; then
-    cppflags="${cppflags} -no-cpp-precomp"
+    #cppflags="${cppflags} -no-cpp-precomp"
 
     # This is necessary because perl's build system doesn't
     # apply cppflags to cc compile lines as it should.
@@ -143,7 +143,6 @@
 
 # Shared library extension is .dylib.
 # Bundle extension is .bundle.
-ld='cc';
 so='dylib';
 dlext='bundle';
 usedl='define';
@@ -181,6 +180,34 @@
    ldflags="${ldflags} -flat_namespace"
    lddlflags="${ldflags} -bundle -undefined suppress"
    ;;
+9.*)
+   lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
+   case "$ld" in
+       *MACOSX_DEVELOPMENT_TARGET*) ;;
+       *) ld="env MACOSX_DEPLOYMENT_TARGET=10.5 ${ld}" ;;
+   esac
+   ;;
+10.*)
+   lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
+   case "$ld" in
+       *MACOSX_DEVELOPMENT_TARGET*) ;;
+       *) ld="env MACOSX_DEPLOYMENT_TARGET=10.6 ${ld}" ;;
+   esac
+   ;;
+11.*)
+   lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
+   case "$ld" in
+       *MACOSX_DEVELOPMENT_TARGET*) ;;
+       *) ld="env MACOSX_DEPLOYMENT_TARGET=10.7 ${ld}" ;;
+   esac
+   ;;
+12.*)
+   lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
+   case "$ld" in
+       *MACOSX_DEVELOPMENT_TARGET*) ;;
+       *) ld="env MACOSX_DEPLOYMENT_TARGET=10.8 ${ld}" ;;
+   esac
+   ;;
 *) 
    lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
    case "$ld" in
@@ -262,6 +289,10 @@
 ;;
 esac
 
+# When the bind9 port is installed, its libbind includes dups from
+# /usr/lib/libdl, so remove libbind
+libswanted=`echo $libswanted | sed 's/ bind / /'`
+
 ##
 # System libraries
 ##
@@ -326,3 +357,5 @@
 # makefile in the same place.  Since Darwin uses GNU make, this dodges
 # the problem.
 firstmakefile=GNUmakefile;
+
+usenm='false'
