--- generator/CMakeLists.txt	2012-08-08 11:38:27.000000000 -0400
+++ generator/CMakeLists.txt	2012-08-08 11:39:15.000000000 -0400
@@ -2,11 +2,11 @@

 set(shiboken_SRC
 generator.cpp
-shiboken/cppgenerator.cpp
-shiboken/headergenerator.cpp
-shiboken/overloaddata.cpp
-shiboken/shibokengenerator.cpp
-shiboken/shibokennormalize.cpp
+shiboken.d/cppgenerator.cpp
+shiboken.d/headergenerator.cpp
+shiboken.d/overloaddata.cpp
+shiboken.d/shibokengenerator.cpp
+shiboken.d/shibokennormalize.cpp
 main.cpp
 )

@@ -15,7 +15,7 @@
     add_definitions(-DDOCSTRINGS_ENABLED)
 endif()

-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/shiboken
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/shiboken.d
                     ${CMAKE_CURRENT_SOURCE_DIR}/qtdoc
                     ${CMAKE_CURRENT_SOURCE_DIR}
                     ${CMAKE_CURRENT_BINARY_DIR}
