--- source/generic/pgf/c/graphdrawing/pgf/gd/interface/c/Makefile.orig
+++ source/generic/pgf/c/graphdrawing/pgf/gd/interface/c/Makefile
@@ -1,7 +1,7 @@
 CONFIGDIR=../../../../../config
 include $(CONFIGDIR)/MakefileConfig.mk
 
-FLAGS=$(MYCFLAGS) $(ARCHFLAGS) -O2 -Wall -I$(LUAINCLUDES) -I$(PGFINCLUDES)
+FLAGS=$(MYCFLAGS) $(ARCHFLAGS) -Wall -I$(LUAINCLUDES) -I$(PGFINCLUDES)
 
 all: InterfaceFromC.o InterfaceFromC++.o
 
--- source/generic/pgf/c/graphdrawing/pgf/gd/examples/c/Makefile.orig
+++ source/generic/pgf/c/graphdrawing/pgf/gd/examples/c/Makefile
@@ -31,5 +31,5 @@ SimpleDemoCPlusPlus.so: SimpleDemoCPlusPlus.o ../../interface/c/InterfaceFromC++
 	SimpleDemoCPlusPlus.o ../../interface/c/InterfaceFromC++.o ../../interface/c/InterfaceFromC.o
 
 SimpleDemoCPlusPlus.o: SimpleDemoCPlusPlus.c++ ../../interface/c/InterfaceFromC++.h  ../../interface/c/InterfaceFromC.h
-	g++ $(FLAGS) -c -o SimpleDemoCPlusPlus.o SimpleDemoCPlusPlus.c++
+	$(CC) $(FLAGS) -c -o SimpleDemoCPlusPlus.o SimpleDemoCPlusPlus.c++
 
