--- macosx/mac-app.py.orig	2015-01-02 16:48:00.000000000 +0100
+++ macosx/mac-app.py	2015-01-02 16:45:35.000000000 +0100
@@ -17,10 +17,14 @@
 macosx = os.path.realpath(os.path.dirname(__file__))
 root = os.path.dirname(macosx)
 
-sys.path.append(root)
+sys.path.insert(0, root)
 
 from frescobaldi_app import info
-from frescobaldi_app.portmidi import pm_ctypes
+try:
+    from frescobaldi_app.portmidi import pm_ctypes
+    dylib_name = pm_ctypes.dll_name
+except ImportError:
+    dylib_name = None
 
 icon = '{0}/icons/{1}.icns'.format(macosx, info.name)
 ipstrings = '{0}/app_resources/InfoPlist.strings'.format(macosx)
@@ -40,7 +44,7 @@
   (WARNING: some manual steps are required after the execution of this script)')
 parser.add_argument('-p', '--portmidi', \
   help = 'full path of PortMIDI library (used only with \'-a\')', \
-  default = pm_ctypes.dll_name)
+  default = dylib_name)
 parser.add_argument('-r', '--arch', \
   help = 'architecture set to include, e.g. i386, x86_64, intel; \
   if the value is None, the architecture of the current Python binary is used \
@@ -52,7 +56,7 @@
 If you really want to point the application bundle to \'{0}\',\n\
 use the \'-f\' or \'--force\' flag.'.format(args.script))
 
-if args.standalone and not os.path.isfile(args.portmidi):
+if args.standalone and not (isinstance(args.portmidi, basestring) and os.path.isfile(args.portmidi)):
     sys.exit('Error: \'{0}\' does not exist or is not a file.'.format(args.portmidi))
 
 plist = dict(
