--- newmat6.cpp.orig	2006-09-09 19:51:25.000000000 +0700
+++ newmat6.cpp	2022-12-29 22:11:03.000000000 +0700
@@ -11,12 +11,13 @@
 #include "newmat.h"
 #include "newmatrc.h"
 
+#include <iostream>
+
 #ifdef use_namespace
 namespace NEWMAT {
 #endif
 
 
-
 #ifdef DO_REPORT
 #define REPORT { static ExeCounter ExeCount(__LINE__,6); ++ExeCount; }
 #else
@@ -428,7 +429,7 @@
 {
    if (&gm == this) { REPORT tag_val = -1; return; }
    REPORT
-   if (indx > 0) { delete [] indx; indx = 0; }
+   if (indx != nullptr) { delete [] indx; indx = 0; }
    ((CroutMatrix&)gm).get_aux(*this);
    Eq(gm);
 }
