From f0bf1d6eec7a92ea809c142f4d1718867be42d68 Mon Sep 17 00:00:00 2001
Date: Sun, 16 May 2021 12:25:16 -0700
Subject: [PATCH 23/24] 10.5 and less: use older cfname and fix uuid on leopard

---
 llvm/tools/dsymutil/SymbolMap.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/llvm/tools/dsymutil/SymbolMap.cpp b/llvm/tools/dsymutil/SymbolMap.cpp
index 07a54795a841..595eaa281b57 100644
--- a/llvm/tools/dsymutil/SymbolMap.cpp
+++ b/llvm/tools/dsymutil/SymbolMap.cpp
@@ -17,6 +17,14 @@
 #ifdef __APPLE__
 #include <CoreFoundation/CoreFoundation.h>
 #include <uuid/uuid.h>
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
+/* declare a missing reference not found in SDK < 10.6 for function called below */
+typedef char * uuid_string_t;
+typedef struct __CFError * CFErrorRef;
+#define CFPropertyListCreateWithStream(A,B,C,D,E,F) CFPropertyListCreateFromStream(A,B,C,D,E,F)
+#endif
+
 #endif
 
 namespace llvm {
-- 
2.21.1 (Apple Git-122.3)

