--- xlsx2csv.sh.orig	2010-08-30 09:50:50.000000000 -0500
+++ xlsx2csv.sh	2010-09-10 17:20:32.000000000 -0500
@@ -155,7 +155,7 @@
 #set up some platform specific utilities
 gnused=$(which sed)
 platform=$(uname -s)
-[[ "${platform}" == "Darwin" ]] && gnused=/opt/local/bin/gsed
+[[ "${platform}" == "Darwin" ]] && gnused=@PREFIX@/bin/gsed
 [[ -f "$gnused" ]] || {
   echo "Can't find a copy of GNU sed"
   exit
