--- ./distribution/archives/build.gradle	2026-10-06 17:21:21
+++ ./distribution/archives/build.gradle	2026-10-06 17:24:42
@@ -38,7 +38,8 @@
       }
       into("darwin".equals(os) ? 'jdk.app' : 'jdk') {
         if (isTestDistro == false) {
-          with jdkFiles(project, os, architecture)
+          // no jdk download, no bundled jdk
+          // with jdkFiles(project, os, architecture)
         }
       }
       into('') {
