--- bin/elasticsearch-env
+++ bin/elasticsearch-env	2020-09-02 17:12:34.000000000 +0900
@@ -19,34 +19,33 @@
   fi
 done
 
-# determine Elasticsearch home; to do this, we strip from the path until we find
-# bin, and then strip bin (there is an assumption here that there is no nested
-# directory under bin also named bin)
-ES_HOME=`dirname "$SCRIPT"`
-
-# now make ES_HOME absolute
-ES_HOME=`cd "$ES_HOME"; pwd`
-
-while [ "`basename "$ES_HOME"`" != "bin" ]; do
-  ES_HOME=`dirname "$ES_HOME"`
-done
-ES_HOME=`dirname "$ES_HOME"`
+ES_HOME="@PREFIX@/share/elasticsearch"
+ES_PATH_CONF="@PREFIX@/etc/elasticsearch"
 
 # now set the classpath
 ES_CLASSPATH="$ES_HOME/lib/*"
 
 # now set the path to java
+if [ -z "$JAVA_HOME" ]; then
+  if [ -x /usr/libexec/java_home ]; then
+    export JAVA_HOME=$(/usr/libexec/java_home)
+    export KEYTOOL="$JAVA_HOME"/jre/bin
+  fi
+fi
+
 if [ ! -z "$JAVA_HOME" ]; then
   JAVA="$JAVA_HOME/bin/java"
   JAVA_TYPE="JAVA_HOME"
-else
-  if [ "$(uname -s)" = "Darwin" ]; then
-    # macOS has a different structure
-    JAVA="$ES_HOME/jdk.app/Contents/Home/bin/java"
-  else
-    JAVA="$ES_HOME/jdk/bin/java"
-  fi
-  JAVA_TYPE="bundled jdk"
+# MacPorts' port:elasticsearch includes port:openjdk13 as a dependency;
+# no need to include this bundled Elasticsearch copy of OpenJDK13
+## else
+##   if [ "$(uname -s)" = "Darwin" ]; then
+##     # macOS has a different structure
+##     JAVA="$ES_HOME/jdk.app/Contents/Home/bin/java"
+##   else
+##     JAVA="$ES_HOME/jdk/bin/java"
+##   fi
+##   JAVA_TYPE="bundled jdk"
 fi
 
 if [ ! -x "$JAVA" ]; then
@@ -78,8 +77,6 @@
 
 export HOSTNAME=$HOSTNAME
 
-if [ -z "$ES_PATH_CONF" ]; then ES_PATH_CONF="$ES_HOME"/config; fi
-
 if [ -z "$ES_PATH_CONF" ]; then
   echo "ES_PATH_CONF must be set to the configuration path"
   exit 1
