From d65a73f5f828153aa7337e9ca27452a5b70a1c05 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Wed, 25 Dec 2019 20:11:09 +0000
Subject: [PATCH] [7.1.1 PowerPC Darwin] Update version.

diff --git a/llvm/CMakeLists.txt b/llvm/CMakeLists.txt
index a3d3013cedc..99d0edfea93 100644
--- a/llvm/CMakeLists.txt
+++ b/llvm/CMakeLists.txt
@@ -29,7 +29,7 @@ if(NOT DEFINED LLVM_VERSION_MINOR)
   set(LLVM_VERSION_MINOR 1)
 endif()
 if(NOT DEFINED LLVM_VERSION_PATCH)
-  set(LLVM_VERSION_PATCH 0)
+  set(LLVM_VERSION_PATCH 1)
 endif()
 if(NOT DEFINED LLVM_VERSION_SUFFIX)
   set(LLVM_VERSION_SUFFIX "")
