diff --git a/src/AdornedRulerPanel.cpp b/src/AdornedRulerPanel.cpp
index 1aedcb99894fa038e790bc7af18068226056ed6b..a41c696e2ecd4e8381974ac998c3e6ded3595793 100644
--- a/src/AdornedRulerPanel.cpp
+++ b/src/AdornedRulerPanel.cpp
@@ -215,6 +215,11 @@ public:
    , mNumGuides{ numGuides }
    {}
 
+   void SetCursor(wxCursor cursor)
+   {
+      mCursor = cursor;
+   }
+
    HitTestPreview Preview(
       const TrackPanelMouseState &state, AudacityProject *pProject)
    override
@@ -767,14 +772,17 @@ private:
 };
 #endif
 
-static auto handOpenCursor =
-    MakeCursor(wxCURSOR_HAND, RearrangeCursorXpm, 16, 16);
+static std::unique_ptr<wxCursor> handOpenCursor = nullptr;
 
 class AdornedRulerPanel::MovePlayRegionHandle final : public PlayRegionAdjustingHandle {
 public:
    MovePlayRegionHandle( AdornedRulerPanel *pParent, wxCoord xx )
    : PlayRegionAdjustingHandle( pParent, xx, MenuChoice::QuickPlay, *handOpenCursor, 2 )
    {
+      if (!handOpenCursor) {
+         handOpenCursor = MakeCursor(wxCURSOR_HAND, RearrangeCursorXpm, 16, 16);
+         SetCursor(*handOpenCursor);
+      }
    }
 
    ~MovePlayRegionHandle()
