diff --git a/src/AdornedRulerPanel.cpp b/src/AdornedRulerPanel.cpp
index 0d9185fe..c5e8a32b 100644
--- a/src/AdornedRulerPanel.cpp
+++ b/src/AdornedRulerPanel.cpp
@@ -234,6 +234,11 @@ public:
    , mNumGuides{ numGuides }
    {}
 
+   void SetCursor(wxCursor cursor)
+   {
+      mCursor = cursor;
+   }
+
    HitTestPreview Preview(
       const TrackPanelMouseState &state, AudacityProject *pProject)
    override
@@ -790,14 +795,17 @@ private:
 };
 #endif
 
-static auto handOpenCursor =
-    MakeCursor(wxCURSOR_HAND, RearrangeCursorXpm, 16, 16);
+static std::unique_ptr<wxCursor> handOpenCursor = nullptr;
 
 class AdornedRulerPanel::MovePlayRegionHandle final : public PlayRegionAdjustingHandle {
 public:
    MovePlayRegionHandle( AdornedRulerPanel *pParent, wxCoord xx )
    : PlayRegionAdjustingHandle( pParent, xx, MenuChoice::QuickPlay, *handOpenCursor, 2 )
    {
+      if (!handOpenCursor) {
+         handOpenCursor = MakeCursor(wxCURSOR_HAND, RearrangeCursorXpm, 16, 16);
+         SetCursor(*handOpenCursor);
+      }
    }
 
    ~MovePlayRegionHandle()
