--- sipbuild/generator/outputs/code.py	Thu Nov 30 09:55:35 2023 +0000
+++ sipbuild/generator/outputs/code.py	Thu Nov 30 21:16:18 2023 +0000
@@ -8408,7 +8408,7 @@
     else:
         is_first = True

-    if klass.docstring is None or klass.docstring.signature is not SocstringSignature.DISCARDED:
+    if klass.docstring is None or klass.docstring.signature is not DocstringSignature.DISCARDED:
         for ctor in klass.ctors:
             if ctor.access_specifier is AccessSpecifier.PRIVATE:
                 continue
