--- siputils.py.orig	2010-09-03 03:26:37.000000000 +1000
+++ siputils.py	2010-10-31 02:55:12.000000000 +1100
@@ -1415,8 +1415,8 @@
             # 'real_prefix' exists if virtualenv is being used.
             dl = getattr(sys, 'real_prefix', sys.exec_prefix).split(os.sep)
 
-            if "Python.framework" not in dl:
-                error("SIP requires Python to be built as a framework")
+            #if "Python.framework" not in dl:
+            #    error("SIP requires Python to be built as a framework")
 
             self.LFLAGS.append("-undefined dynamic_lookup")
 
