diff -u -r rom.orig/SConscript rom/SConscript
--- rom.orig/SConscript	2008-08-24 02:37:19.000000000 +0200
+++ rom/SConscript	2009-03-06 18:18:50.000000000 +0100
@@ -26,7 +26,8 @@
 
 roms = Split ("""21.obj 25.obj
                  32e.obj 33c.obj 34c.obj 37e.obj 38c.obj 38e.obj
-                 41cv.mod 41cx.mod 82143a.mod 82182a.mod""")
+                 41cv.mod 41cx.mod 82143a.mod 82182a.mod
+                 11c.obj 12c.obj 15c.obj 16c.obj""")
 
 env.Alias (target = 'install',
            source = env.Install (dir = env ['destdir'] + env ['libdir'],
