diff -u -r src.orig/util.c src/util.c
--- src.orig/util.c	2008-08-24 02:37:18.000000000 +0200
+++ src/util.c	2009-03-05 17:53:26.000000000 +0100
@@ -133,12 +133,12 @@
 // strlcpy will copy as much of src into dest as it can, up to one less than
 // the maximum length of dest specified by the argument l.  Unlike strncpy(),
 // strlcpy() will always leave dest NULL-terminated on return.
-char *strlcpy (char *dest, const char *src, size_t l)
-{
-  strncpy (dest, src, l);
-  dest [l - 1] = '\0';
-  return dest;
-}
+/*char *strlcpy (char *dest, const char *src, size_t l)*/
+/*{*/
+  /*strncpy (dest, src, l);*/
+  /*dest [l - 1] = '\0';*/
+  /*return dest;*/
+/*}*/
 
 
 // strlncpy will copy up to n characters from src to dest, but not more than
diff -u -r src.orig/util.h src/util.h
--- src.orig/util.h	2008-08-24 02:37:18.000000000 +0200
+++ src/util.h	2009-03-05 17:53:37.000000000 +0100
@@ -56,7 +56,7 @@
 // strlcpy will copy as much of src into dest as it can, up to one less than
 // the maximum length of dest specified by the argument l.  Unlike strncpy(),
 // strlcpy() will always leave dest NULL-terminated on return.
-char *strlcpy (char *dest, const char *src, size_t l);
+/*char *strlcpy (char *dest, const char *src, size_t l);*/
 
 
 // strlncpy will copy up to n characters from src to dest, but not more than
