--- fluid/Makefile.orig	2012-09-09 10:28:39.000000000 -0500
+++ fluid/Makefile	2012-09-09 10:30:39.000000000 -0500
@@ -54,7 +54,7 @@
 fluid-shared$(EXEEXT):	$(OBJECTS) ../src/$(DSONAME) ../src/$(FLDSONAME) \
 			../src/$(IMGDSONAME)
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) $(LINKSHARED) $(LDLIBS)
+	$(CXX) $(ARCHFLAGS) -o $@ $(OBJECTS) $(LINKSHARED) $(LDLIBS) $(LDFLAGS)
 
 clean:
 	-$(RM) *.o core.* *~ *.bck *.bck
