https://github.com/sdispater/clikit/pull/10

diff --git a/src/clikit/adapter/style_converter.py b/src/clikit/adapter/style_converter.py
index a81d620..a2062e5 100644
--- src/clikit/adapter/style_converter.py
+++ src/clikit/adapter/style_converter.py
@@ -16,7 +16,7 @@ def convert(cls, style):  # type: (Style) -> PastelStyle
             options.append("bold")
 
         if style.is_underlined():
-            options.append("underscore")
+            options.append("underline")
 
         if style.is_blinking():
             options.append("blink")
