--- autogen.sh.orig	2017-03-13 12:03:45.000000000 -0700
+++ autogen.sh	2017-03-13 12:04:30.000000000 -0700
@@ -5,8 +5,8 @@
 cd ..
 autoreconf -f -i
 
-curl "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD" > config.guess
-curl "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD" > config.sub
+#curl "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD" > config.guess
+#curl "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD" > config.sub
 
 #for f in "lt~obsolete.m4" "ltversion.m4" "ltsugar.m4" "ltoptions.m4" "libtool.m4"; do
 	#curl "http://git.savannah.gnu.org/cgit/libtool.git/plain/m4/${f}?id=v2.4.2.418" > m4/${f}
