--- pyproject.toml.orig	2024-01-13 20:13:08
+++ pyproject.toml	2024-03-18 11:45:39
@@ -139,7 +139,7 @@
 # Runtime dependency for this build hook
 # We need jupyterlab as build time depdendency to get jlpm (wrapper around yarn)
 dependencies = [
-    "hatch-jupyter-builder>=0.5", "jupyterlab>=4"
+    "hatch-jupyter-builder>=0.5"
 ]
 # We use npm_builder to build the jupyterlab extension
 build-function = "hatch_jupyter_builder.npm_builder"
