--- src/prefs/GUIPrefs.cpp.orig	2024-04-24 15:00:43
+++ src/prefs/GUIPrefs.cpp	2024-05-09 22:12:24
@@ -28,6 +28,7 @@
 #include "Theme.h"
 #include "Prefs.h"
 #include "ShuttleGui.h"
+#include "HelpSystem.h"
 
 #include "Decibels.h"
 #include "Beats.h"
@@ -147,7 +148,7 @@
       S.StartMultiColumn(2);
       {
          S.TieChoice( XXO("&Language:"), LanguageSetting);
-         // S.TieChoice( XXO("Location of &Manual:"), GUIManualLocation);
+         S.TieChoice( XXO("Location of &Manual:"), GUIManualLocation);
          S.TieChoice( XXO("Th&eme:"), GUITheme());
          S.TieChoice( XXO("Meter dB &range:"), DBSetting);
       }
