--- minitar-cli-0.9.gemspec
+++ minitar-cli-0.9.gemspec
@@ -26,8 +26,8 @@
 
   s.specification_version = 4
 
-  s.add_runtime_dependency(%q<minitar>.freeze, ["~> 0.9.0"])
-  s.add_runtime_dependency(%q<powerbar>.freeze, ["~> 1.0"])
+  s.add_runtime_dependency(%q<minitar>.freeze, [">= 0.9.0"])
+  s.add_runtime_dependency(%q<powerbar>.freeze, [">= 1.0"])
   s.add_development_dependency(%q<minitest>.freeze, ["~> 5.24"])
   s.add_development_dependency(%q<hoe-doofus>.freeze, ["~> 1.0"])
   s.add_development_dependency(%q<hoe-gemspec2>.freeze, ["~> 1.1"])
