--- minitar-cli-0.8.gemspec
+++ minitar-cli-0.8.gemspec
@@ -25,16 +25,16 @@
 
   s.specification_version = 4
 
-  s.add_runtime_dependency(%q<minitar>.freeze, ["~> 0.8.0".freeze])
-  s.add_runtime_dependency(%q<powerbar>.freeze, ["~> 1.0".freeze])
-  s.add_development_dependency(%q<minitest>.freeze, ["~> 5.11".freeze])
-  s.add_development_dependency(%q<hoe-doofus>.freeze, ["~> 1.0".freeze])
-  s.add_development_dependency(%q<hoe-gemspec2>.freeze, ["~> 1.1".freeze])
-  s.add_development_dependency(%q<hoe-git>.freeze, ["~> 1.6".freeze])
-  s.add_development_dependency(%q<hoe-rubygems>.freeze, ["~> 1.0".freeze])
-  s.add_development_dependency(%q<hoe-travis>.freeze, ["~> 1.2".freeze])
-  s.add_development_dependency(%q<minitest-autotest>.freeze, [">= 1.0".freeze, "< 2".freeze])
-  s.add_development_dependency(%q<rake>.freeze, [">= 10.0".freeze, "< 12".freeze])
+  s.add_runtime_dependency(%q<minitar>.freeze, [">= 0.8.0".freeze])
+  s.add_runtime_dependency(%q<powerbar>.freeze, [">= 1.0".freeze])
+  s.add_development_dependency(%q<minitest>.freeze, [">= 5.11".freeze])
+  s.add_development_dependency(%q<hoe-doofus>.freeze, [">= 1.0".freeze])
+  s.add_development_dependency(%q<hoe-gemspec2>.freeze, [">= 1.1".freeze])
+  s.add_development_dependency(%q<hoe-git>.freeze, [">= 1.6".freeze])
+  s.add_development_dependency(%q<hoe-rubygems>.freeze, [">= 1.0".freeze])
+  s.add_development_dependency(%q<hoe-travis>.freeze, [">= 1.2".freeze])
+  s.add_development_dependency(%q<minitest-autotest>.freeze, [">= 1.0".freeze])
+  s.add_development_dependency(%q<rake>.freeze, [">= 10.0".freeze])
   s.add_development_dependency(%q<rdoc>.freeze, [">= 0.0".freeze])
-  s.add_development_dependency(%q<hoe>.freeze, ["~> 3.17".freeze])
+  s.add_development_dependency(%q<hoe>.freeze, [">= 3.17".freeze])
 end
