--- minitar-cli-0.9.gemspec
+++ minitar-cli-0.9.gemspec
@@ -29,8 +29,8 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<minitar>.freeze, ["~> 0.9.0"])
-    s.add_runtime_dependency(%q<powerbar>.freeze, ["~> 1.0"])
+    s.add_runtime_dependency(%q<minitar>.freeze, [">= 0.9.0"])
+    s.add_runtime_dependency(%q<powerbar>.freeze, [">= 1.0"])
     s.add_development_dependency(%q<minitest>.freeze, ["~> 5.24"])
     s.add_development_dependency(%q<hoe-doofus>.freeze, ["~> 1.0"])
     s.add_development_dependency(%q<hoe-gemspec2>.freeze, ["~> 1.1"])
@@ -42,8 +42,8 @@
     s.add_development_dependency(%q<rdoc>.freeze, [">= 4.0", "< 7"])
     s.add_development_dependency(%q<hoe>.freeze, ["~> 4.2"])
   else
-    s.add_dependency(%q<minitar>.freeze, ["~> 0.9.0"])
-    s.add_dependency(%q<powerbar>.freeze, ["~> 1.0"])
+    s.add_dependency(%q<minitar>.freeze, [">= 0.9.0"])
+    s.add_dependency(%q<powerbar>.freeze, [">= 1.0"])
     s.add_dependency(%q<minitest>.freeze, ["~> 5.24"])
     s.add_dependency(%q<hoe-doofus>.freeze, ["~> 1.0"])
     s.add_dependency(%q<hoe-gemspec2>.freeze, ["~> 1.1"])
