--- config/apple.orig	2011-02-14 10:23:41.000000000 -0600
+++ config/apple	2011-02-26 15:11:09.000000000 -0600
@@ -49,11 +49,11 @@
 
 # Fortran doesn't work with shared libraries
 # 
-if test "X-" != "X-$enable_fortran"; then
-        echo '    Warning: shared libraries are not supported with Fortran'
-        echo '    Disabling shared libraries'
-        enable_shared="no"
-fi
+#if test "X-yes" = "X-$enable_fortran"; then
+#        echo '    Warning: shared libraries are not supported with Fortran'
+#        echo '    Disabling shared libraries'
+#        enable_shared="no"
+#fi
 
 # compiler version strings
 case $CC in
