--- config/apple.orig	2009-03-03 10:59:12.000000000 -0600
+++ config/apple	2009-03-03 10:59:29.000000000 -0600
@@ -64,9 +64,9 @@
 esac  
 # Fortran doesn't support shared libraries
 # Works with static executable only.
-if test "X-" != "X-$enable_fortran"; then
-        echo '    warning: shared libraries are not supported with Fortran'
-        echo '    disabling shared libraries'
-        enable_shared="no"
-fi
+# if test "X-" != "X-$enable_fortran"; then
+#         echo '    warning: shared libraries are not supported with Fortran'
+#         echo '    disabling shared libraries'
+#         enable_shared="no"
+# fi
 
