--- a/cmake/Modules/PothosStandardFlags.cmake	2021-01-25 11:32:12.000000000 +0800
+++ b/cmake/Modules/PothosStandardFlags.cmake	2023-07-13 18:47:50.000000000 +0800
@@ -27,9 +27,9 @@
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
     endif()
 
-    #force a compile-time error when symbols are missing
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
-    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-undefined")
+    # This flag is unsupported and breaks the build.
+    # set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
+    # set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-undefined")
 
     #these warnings are caused by static warnings used throughout the code
     add_compile_options(-Wno-unused-local-typedefs)
