Index: /AppTools/trunk/enthought/preferences/scoped_preferences.py
===================================================================
--- /AppTools/trunk/enthought/preferences/scoped_preferences.py (revision 23752)
+++ /AppTools/trunk/enthought/preferences/scoped_preferences.py (revision 24035)
@@ -321,5 +321,6 @@
         application_scope = Preferences(
             name     = 'application',
-            filename = join(ETSConfig.application_home, 'preferences.ini')
+            filename = join(ETSConfig.get_application_home(create=False), 
+                            'preferences.ini')
         )
 
