--- setup.py	2019-02-02 00:44:04.000000000 +0100
+++ setup.fixed.py	2019-09-03 14:51:56.000000000 +0200
@@ -23,13 +23,13 @@
             'pyspf-milter = spf_engine.milter_spf:main',
         ]},
     include_package_data=True,
-    data_files=[(os.path.join('share', 'man', 'man1'),
-        ['policyd-spf.1']), (os.path.join('share', 'man', 'man5'),
-        ['policyd-spf.conf.5']), (os.path.join('etc', 'python-policyd-spf'),
-        ['policyd-spf.conf']), (os.path.join('share', 'man', 'man5'),
-        ['policyd-spf.peruser.5']), (os.path.join('lib', 'systemd', 'system'),
-        ['system/pyspf-milter.service']),(os.path.join('etc', 'init.d'),
-        ['system/pyspf-milter']) ],
+    data_files=[
+        (os.path.join('@@prefix@@','share', 'man', 'man1'),['policyd-spf.1']), 
+        (os.path.join('@@prefix@@','share', 'man', 'man5'),['policyd-spf.conf.5']), 
+        (os.path.join('@@prefix@@','etc', 'python-policyd-spf'),['policyd-spf.conf']), 
+        (os.path.join('@@prefix@@','etc', 'python-policyd-spf'),['policyd-spf.conf.commented']), 
+        (os.path.join('@@prefix@@','share', 'man', 'man5'),['policyd-spf.peruser.5']) 
+    ],
     classifiers = [
         'Development Status :: 5 - Production/Stable',
         'Environment :: No Input/Output (Daemon)',
