diff -u a/Monitors/PulsarMon/PulsarMon.cc b/Monitors/PulsarMon/PulsarMon.cc
--- Monitors/PulsarMon/PulsarMon.cc
+++ Monitors/PulsarMon/PulsarMon.cc
@@ -261,11 +261,11 @@
 	FILE* tmp;
         if (mOutput.error_log) {
 	    tmp = freopen(mOutput.error_file_name.c_str(), "w", stderr);
-	    if (tmp < 0) perror("Unable to divert stderr");
+	    if (tmp == (FILE*)NULL) perror("Unable to divert stderr");
 	}
         if (mOutput.write_log) {
 	    tmp = freopen(mOutput.log_file_name.c_str(), "w", stdout);	
-	    if (tmp < 0) perror("Unable to divert stdout");
+	    if (tmp == (FILE*)NULL) perror("Unable to divert stdout");
 	}
 
         //---------- Set up array that will hold data for DMTViewer history 
