Don't use rpath.

https://github.com/libLAS/libLAS/pull/185
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -67,7 +67,6 @@ mark_as_advanced(CMAKE_VERBOSE_MAKEFILE)
 
 # Path to additional CMake modules
 set(CMAKE_MODULE_PATH "${libLAS_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
-set(CMAKE_MACOSX_RPATH OFF)
 
 ###############################################################################
 # General build settings
--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -248,17 +248,6 @@ target_include_directories(las_c
     $<BUILD_INTERFACE:${libLAS_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:${LIBLAS_INCLUDE_DIR}>)
 
-if (APPLE)
-  set_target_properties(
-    las_c
-    PROPERTIES
-    INSTALL_NAME_DIR "@rpath" BUILD_WITH_INSTALL_RPATH ON)
-  set_target_properties(
-    las
-    PROPERTIES
-    INSTALL_NAME_DIR "@rpath" BUILD_WITH_INSTALL_RPATH ON)
-endif()
-
 ###############################################################################
 # Targets installation.  The EXPORT clause specifies a depends target
 # which packages up information about the libraries for
