--- src/gt_citation.cpp.old	2016-08-22 15:06:34.000000000 +0200
+++ src/gt_citation.cpp	2018-10-16 14:08:18.000000000 +0200
@@ -155,7 +155,7 @@
         if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
           p1 --;
         p2 = p1 - 1;
-        while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+        while( p2>(char *)0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
           p2--;
         if(p2 != p1 - 1)
           p1 = p2;
@@ -198,7 +198,7 @@
         if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
           p1 --;
         char* p2 = p1 - 1;
-        while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+        while( p2>(char *)0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
           p2--;
         if(p2 != p1 - 1)
           p1 = p2;
@@ -387,10 +387,10 @@
         osCitation += primemName;
         bRewriteGeogCitation = TRUE;
 
-        double primemValue = poSRS->GetPrimeMeridian(NULL);
+        double primemValue = poSRS->GetPrimeMeridian((char **)NULL);
         if(angUnitName && !EQUAL(angUnitName, "Degree"))
         {
-            double aUnit = poSRS->GetAngularUnits(NULL);
+            double aUnit = poSRS->GetAngularUnits((char **)NULL);
             primemValue *= aUnit;
         }
         GTIFKeySet( psGTIF, GeogPrimeMeridianLongGeoKey, TYPE_DOUBLE, 1, 
