--- xmore.c.orig	2010-07-22 22:54:24.000000000 -0700
+++ xmore.c	2010-07-22 22:54:50.000000000 -0700
@@ -147,14 +147,14 @@ quitXtProc(Widget w, XtPointer client_da
     XtCallActionProc(w, "quit", NULL, NULL, 0);
 }
 
+#ifdef USE_XPRINT
+
 static void
 printshellDestroyXtProc(Widget w, XtPointer client_data, XtPointer callData)
 {
     XawPrintDialogClosePrinterConnection(printdialog, False);
 }
 
-#ifdef USE_XPRINT
-
 static void
 printOKXtProc(Widget w, XtPointer client_data, XtPointer callData)
 {
