--- host/CMakeLists.txt.orig
+++ host/CMakeLists.txt
@@ -152,6 +152,11 @@
             ${CMAKE_CURRENT_LIST_DIR}/common/include/osx
     )
 
+    # 'clock_gettime' and related are defined on OSX as of 10.12.0 (16.0.0)
+    if(${CMAKE_SYSTEM_VERSION} VERSION_LESS 16.0.0)
+        add_definitions(-D_OSX_NEED_CLOCK_GETTIME)
+    endif()
+
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     set(BLADERF_OS_WINDOWS 1)
 
--- host/common/include/osx/clock_gettime.h.orig
+++ host/common/include/osx/clock_gettime.h
@@ -35,10 +35,12 @@
 #   error "This file is intended for use with OSX systems only."
 #endif
 
+#ifdef _OSX_NEED_CLOCK_GETTIME
 typedef int clockid_t;
 #define CLOCK_REALTIME 0
 
 int clock_gettime(clockid_t clk_id, struct timespec *tp);
+#endif /* _OSX_NEED_CLOCK_GETTIME */
 
 #ifdef __cplusplus
 } /* extern "C" */
