--- Makefile	2025-11-04 17:58:08
+++ Makefile.new	2025-11-17 13:40:45
@@ -1,9 +1,9 @@
 OS = $(shell uname -s)
-KRUNKIT_RELEASE = target/release/krunkit
-KRUNKIT_DEBUG = target/debug/krunkit
+KRUNKIT_RELEASE = target/$(CARGO_BUILD_TARGET)/release/krunkit
+KRUNKIT_DEBUG = target/$(CARGO_BUILD_TARGET)/debug/krunkit
 
 ifeq ($(PREFIX),)
-    PREFIX := /usr/local
+    PREFIX := @@DEFAULT_PREFIX@@
 endif
 
 .PHONY: install clean $(KRUNKIT_RELEASE) $(KRUNKIT_DEBUG)
@@ -15,6 +15,7 @@
 $(KRUNKIT_RELEASE):
 	cargo build --release
 ifeq ($(OS),Darwin)
+	install_name_tool -add_rpath $(PREFIX)/lib $@
 	codesign --entitlements krunkit.entitlements --force -s - $@
 endif
 
