--- Makefile.orig	2012-02-24 12:55:04.000000000 -0600
+++ Makefile	2014-04-29 21:24:01.000000000 -0500
@@ -1,6 +1,6 @@
 PREFIX?=	/usr/local
 LDLIBS=		-lz
-CFLAGS+=	-Wsystem-headers -Wno-format-y2k -W -Werror \
+CFLAGS+=	-Wsystem-headers -Wno-format-y2k -W \
 		-Wno-unused-parameter -Wstrict-prototypes \
 		-Wmissing-prototypes -Wpointer-arith -Wreturn-type \
 		-Wcast-qual -Wwrite-strings -Wswitch -Wshadow -Wcast-align \
@@ -28,4 +28,4 @@
 
 install:
 	install -s vmdktool ${DESTDIR}${PREFIX}/bin/
-	install vmdktool.8 ${DESTDIR}${PREFIX}/man/man8/
+	install vmdktool.8 ${DESTDIR}${PREFIX}/share/man/man8/
