--- Makefile.orig	2009-03-03 02:19:56.000000000 +0100
+++ Makefile	2009-03-03 02:23:11.000000000 +0100
@@ -2,9 +2,9 @@
 # $Id: Makefile 761 2009-02-24 08:20:20Z x $
 # 
 #
-CFLAGS = -Wall -ggdb -DGUI
+CFLAGS = -Wall -ggdb -DGUI -I@PREFIX@/include
 #CFLAGS = -Wall -ggdb -DGUI -DDEBUG
-LDFLAGS = -lcrypto -lz -lresolv -lexpat -lvorbisfile 
+LDFLAGS = -lcrypto -lz -lresolv -lexpat -lvorbisfile -L@PREFIX@/lib
 
 CORE_OBJS = aes.o auth.o buffer.o channel.o commands.o dns.o event.o handlers.o keyexchange.o packet.o playlist.o puzzle.o session.o shn.o util.o xml.o
 DESPOTIFY_OBJS = ADMclubbing.o audio.o despotify.o ui-core.o ui-help.o ui-player.o ui-playlist.o ui-search.o ui-userdata.o sndqueue.o
