--- configure.orig	2008-09-22 05:37:14.000000000 -0700
+++ configure	2008-09-30 16:21:00.000000000 -0700
@@ -24787,11 +24787,6 @@
   ;;
 esac
 
-if test "x$LIBM" = "x"; then
-	{ { $as_echo "$as_me:$LINENO: error: You need -lm(mathematic library) for this version of libgnomeprint." >&5
-$as_echo "$as_me: error: You need -lm(mathematic library) for this version of libgnomeprint." >&2;}
-   { (exit 1); exit 1; }; }
-fi
 LDFLAGS="$LDFLAGS $LIBM"
 
 FREETYPE_LIBS=
