diff -Naurp drivers/coreaudio.orig/Makefile.am drivers/coreaudio/Makefile.am
--- drivers/coreaudio.orig/Makefile.am	2008-05-29 05:26:08.000000000 -0700
+++ drivers/coreaudio/Makefile.am	2015-06-12 00:10:18.000000000 -0700
@@ -1,7 +1,6 @@
 MAINTAINERCLEANFILES=Makefile.in
 
-AM_CFLAGS = $(JACK_CFLAGS) -include /System/Library/Frameworks/Carbon.framework/Headers/Carbon.h
-
+AM_CFLAGS = $(JACK_CFLAGS)
 plugindir = $(ADDON_DIR)
 
 plugin_LTLIBRARIES = jack_coreaudio.la
diff -Naurp drivers/coreaudio.orig/Makefile.in drivers/coreaudio/Makefile.in
--- drivers/coreaudio.orig/Makefile.in	2011-09-28 06:10:12.000000000 -0700
+++ drivers/coreaudio/Makefile.in	2015-06-12 00:10:29.000000000 -0700
@@ -249,7 +249,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 MAINTAINERCLEANFILES = Makefile.in
-AM_CFLAGS = $(JACK_CFLAGS) -include /System/Library/Frameworks/Carbon.framework/Headers/Carbon.h
+AM_CFLAGS = $(JACK_CFLAGS)
 plugindir = $(ADDON_DIR)
 plugin_LTLIBRARIES = jack_coreaudio.la
 jack_coreaudio_la_LDFLAGS = -module -avoid-version @OS_LDFLAGS@
diff -Naurp drivers/coreaudio.orig/coreaudio_driver.h drivers/coreaudio/coreaudio_driver.h
--- drivers/coreaudio.orig/coreaudio_driver.h	2008-05-29 05:26:08.000000000 -0700
+++ drivers/coreaudio/coreaudio_driver.h	2015-06-12 00:11:56.000000000 -0700
@@ -28,6 +28,7 @@
 #ifndef __jack_coreaudio_driver_h__
 #define __jack_coreaudio_driver_h__
 
+#include <Carbon/Carbon.h>
 #include <CoreAudio/CoreAudio.h>
 #include <AudioToolbox/AudioConverter.h>
 #include <AudioUnit/AudioUnit.h>
