--- configure.ac.orig	2016-03-20 19:19:45.000000000 -0700
+++ configure.ac	2016-03-20 19:19:55.000000000 -0700
@@ -309,11 +309,6 @@ AC_ARG_ENABLE(ancient_libc,
 	 
 JACK_CORE_CFLAGS="$JACK_CORE_CFLAGS $JACK_LIBC_HELPER_FLAGS"   
 
-case $build_os in
-     # we need weak linkage which appeared in 10.2, but lets ask for 10.4 anyway
-     darwin*) JACK_CORE_CFLAGS="$JACK_CORE_CFLAGS -mmacosx-version-min=10.4" ;;
-esac    
-
 AC_ARG_WITH(cpu-target,
             [  --with-cpu-target=cpu-type    explicit, overriding argument for gcc -march= flag])
 
