--- config/os/macosx/pThreadUtilities.h.orig	2013-07-03 21:52:49.000000000 -0700
+++ config/os/macosx/pThreadUtilities.h	2013-07-03 21:53:03.000000000 -0700
@@ -66,7 +66,7 @@
 #define __PTHREADUTILITIES_H__
 
 #import "pthread.h"
-#import <CoreServices/../Frameworks/CarbonCore.framework/Headers/MacTypes.h>
+#import <MacTypes.h>
 
 #define THREAD_SET_PRIORITY			0
 #define THREAD_SCHEDULED_PRIORITY		1
