--- macosx/JackMachThread.h.orig	2022-04-15 14:14:06.000000000 -0500
+++ macosx/JackMachThread.h	2022-04-23 02:00:32.000000000 -0500
@@ -70,7 +70,7 @@
 
 #include "JackPosixThread.h"
 #ifndef MY_TARGET_OS_IPHONE
-#include <MacTypes.h>
+#include <CoreServices/../Frameworks/CarbonCore.framework/Headers/MacTypes.h>
 #endif
 
 #include <mach/thread_policy.h>
--- macosx/coremidi/JackCoreMidiUtil.h.orig	2022-04-15 14:14:06.000000000 -0500
+++ macosx/coremidi/JackCoreMidiUtil.h	2022-04-23 02:00:32.000000000 -0500
@@ -22,7 +22,7 @@
 
 #include <string>
 
-#include <MacTypes.h>
+#include <CoreServices/../Frameworks/CarbonCore.framework/Headers/MacTypes.h>
 
 namespace Jack {
 
