--- bin/fcat.orig	2012-06-18 07:25:41.000000000 -0700
+++ bin/fcat	2021-04-25 13:58:44.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!__MACPORTS_PERL5__
 
 #c
 #c                 / ========================================== \ 
@@ -422,7 +422,7 @@
 	    /^(\s*intrinsic\s+)/i || 
 	    /^(\s*include\s*)/i || 	    /^(\s*common\s+)/i || 	    /^(\s*common\s*\/)/i ||
 	    /^(\s*intent\s+)/i || 	    /^(\s*intent\s*\()/i ||
-	    /^(\s*dimension\s*\()/i ||
+	    /^(\s*dimension\s*\()/i ||  /^(\s*dimension\s+)/i ||
 	    /^(\s*type\s*\()/i || 	    /^(\s*parameter\s*\()/i ||
 	    /^(\s*data\s+)/i ||
 	    /^(\s*namelist\s+)/i ||
