From 8a94aef1b7e7311ea507f04ec6dfc2721bdebb92 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Wed, 14 Jun 2023 22:39:35 +0800
Subject: [PATCH] ProcessorTargets: fix native flags

---
 ProcessorTargets.cmake | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git ProcessorTargets.cmake ProcessorTargets.cmake
index b68d047..0146a26 100644
--- ProcessorTargets.cmake
+++ ProcessorTargets.cmake
@@ -6,5 +6,8 @@ set(PROC_TARGET_1_FLAGS "-mtune=generic" CACHE STRING "Processor-1 flags")
 
 # This second choice should be used for your own build only
 set(PROC_TARGET_2_LABEL native CACHE STRING "Processor-2 label - use it for your own build")
-set(PROC_TARGET_2_FLAGS "-march=native" CACHE STRING "Processor-2 flags")
-
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "ppc|ppc64|powerpc|powerpc64" OR (APPLE AND CMAKE_OSX_ARCHITECTURES MATCHES "ppc|ppc64"))
+  set(PROC_TARGET_2_FLAGS "-mtune=native" CACHE STRING "Processor-2 flags")
+else()
+  set(PROC_TARGET_2_FLAGS "-march=native" CACHE STRING "Processor-2 flags")
+endif()
