--- configure	2015-10-09 12:11:54.000000000 -0400
+++ configure.new	2015-10-09 12:11:38.000000000 -0400
@@ -1582,7 +1582,7 @@
                           Directory where PNFFT Fortran include files were
                           installed.
   --with-pspio-prefix=DIR Directory where pspio was installed.
-  --with-metis-prefix     Directory where external METIS library was installed
+  --with-metis-prefix=DIR Directory where external METIS library was installed
                           (must be single-precision)
   --with-parmetis-prefix  Directory where ParMETIS library was installed
 
@@ -12691,11 +12691,11 @@
 
 
   case $with_metis_prefix in
-    no ) acx_external_metis=disabled ;;
+    no ) acx_external_metis=no ;;
     "") with_metis_prefix="/usr" ;;
   esac
 
-  if test x"$acx_external_metis" != xdisabled; then
+  if test x"$acx_external_metis" != xno; then
 
         acx_metis_save_CFLAGS="$CFLAGS"
     acx_metis_save_LIBS="$LIBS"
@@ -12775,6 +12775,9 @@
 
     CFLAGS="$acx_metis_save_CFLAGS"
     LIBS="$acx_metis_save_LIBS"
+  else
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled" >&5
+$as_echo "disabled" >&6; }
   fi
 
   if test x"$acx_external_metis" = xno ; then
