--- configure	2015-11-08 13:28:49.000000000 -0500
+++ configure.newer	2015-11-08 13:25:15.000000000 -0500
@@ -10214,7 +10214,8 @@
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gsl_major_version.$gsl_minor_version.$gsl_micro_version" >&5
 $as_echo "$gsl_major_version.$gsl_minor_version.$gsl_micro_version" >&6; }
-    if test $gsl_major_version -lt $min_gsl_major_version -o $gsl_minor_version -lt $min_gsl_minor_version; then
+    if test $gsl_major_version -lt $min_gsl_major_version -o \
+            $gsl_major_version -eq $min_gsl_major_version -a $gsl_minor_version -lt $min_gsl_minor_version; then
       as_fn_error $? "could not find required GSL library" "$LINENO" 5
     fi
 
@@ -11167,11 +11168,9 @@
   withval=$with_berkeleygw_prefix;
 fi
 
-case $with_berkeleygw_prefix in
-  no ) acx_berkeleygw_ok=disabled ;;
-  *) LIBS_BERKELEYGW="-L$with_berkeleygw_prefix/library -lBGW_wfn";
-     FCFLAGS_BERKELEYGW="$ax_cv_f90_modflag$with_berkeleygw_prefix/library" ;;
-esac
+if test "x$with_berkeleygw_prefix" = xno; then
+  acx_berkeleygw_ok=disabled
+fi
 
 acx_berkeleygw_save_LIBS="$LIBS"
 acx_berkeleygw_save_FCFLAGS="$FCFLAGS"
@@ -11179,21 +11177,31 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for BerkeleyGW" >&5
 $as_echo_n "checking for BerkeleyGW... " >&6; }
 if test "$acx_berkeleygw_ok" != disabled; then
-  FCFLAGS="$FCFLAGS_BERKELEYGW $acx_berkeleygw_save_FCFLAGS"
-  LIBS="$LIBS_BERKELEYGW $acx_berkeleygw_save_LIBS"
-  cat > conftest.$ac_ext <<_ACEOF
+  for location in "source" "install"; do
+    if test "$location" = "source"; then
+      LIBS_BERKELEYGW="-L$with_berkeleygw_prefix/library -lBGW_wfn";
+      FCFLAGS_BERKELEYGW="$ax_cv_f90_modflag$with_berkeleygw_prefix/library";
+    else
+      LIBS_BERKELEYGW="-L$with_berkeleygw_prefix/lib -lBGW_wfn";
+      FCFLAGS_BERKELEYGW="$ax_cv_f90_modflag$with_berkeleygw_prefix/include";
+    fi
+
+    FCFLAGS="$FCFLAGS_BERKELEYGW $acx_berkeleygw_save_FCFLAGS"
+    LIBS="$LIBS_BERKELEYGW $acx_berkeleygw_save_LIBS"
+    cat > conftest.$ac_ext <<_ACEOF
       program main
 
-    use wfn_rho_vxc_io_m
-    call bgw_conf_test()
+      use wfn_rho_vxc_io_m
+      call bgw_conf_test()
 
       end
 _ACEOF
 if ac_fn_fc_try_link "$LINENO"; then :
-  acx_berkeleygw_ok=yes
+  acx_berkeleygw_ok=yes; break
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
+  done
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $acx_berkeleygw_ok ($FCFLAGS_BERKELEYGW $LIBS_BERKELEYGW)" >&5
 $as_echo "$acx_berkeleygw_ok ($FCFLAGS_BERKELEYGW $LIBS_BERKELEYGW)" >&6; }
@@ -12655,6 +12664,10 @@
 
     CFLAGS="$acx_metis_save_CFLAGS"
     LIBS="$acx_metis_save_LIBS"
+  else
+    { $as_echo "$as_me:${as_lineno-$LINENO}: result: disabled" >&5
+$as_echo "disabled" >&6; }
+    acx_external_metis=no
   fi
 
   if test x"$acx_external_metis" = xno ; then
