--- util.c.orig	2003-09-16 09:37:19.000000000 +1000
+++ util.c	2012-07-26 19:36:52.000000000 +1000
@@ -59,7 +59,7 @@ powmod(int x, int y, int q)
 	}
 	return ret;
 }
-
+/*
 const char *
 memmem(const char *T, size_t n, const char *P, size_t m)
 {
@@ -91,5 +91,5 @@ memmem(const char *T, size_t n, const ch
 			t = (d * (t - T[s] * h) + T[s + m]) % q;
 	}
 	return NULL;
-}
+}*/
 #endif
--- header.c.orig	2003-09-16 09:37:19.000000000 +1000
+++ header.c	2012-07-26 19:38:00.000000000 +1000
@@ -31,7 +31,6 @@
 #include <string.h>
 
 #include <mimemisc.h>
-#include <util.h>
 
 #include <mime_priv.h>
 
--- message.c.orig	2003-09-16 09:37:19.000000000 +1000
+++ message.c	2012-07-26 19:37:43.000000000 +1000
@@ -33,7 +33,6 @@
 
 #include <mime_priv.h>
 #include <ct.h>
-#include <util.h>
 
 static struct mime_message *
 getmminit()
