--- deluge/i18n/util.py.orig	2019-08-13 12:30:04.000000000 -0700
+++ deluge/i18n/util.py	2019-08-13 12:32:11.000000000 -0700
@@ -132,7 +132,7 @@
                         log.error(exception)
                         setup_mock_translation()
         elif deluge.common.osx_check():
-            libintl = ctypes.cdll.LoadLibrary('libintl.dylib')
+            libintl = ctypes.cdll.LoadLibrary('@@MP_PREFIX@@/lib/libintl.dylib')
 
         if libintl:
             libintl.bindtextdomain(
