--- Modules/objc/objc_inject.m.orig	2007-11-02 00:04:34.000000000 +1100
+++ Modules/objc/objc_inject.m	2009-09-14 21:44:14.000000000 +1000
@@ -39,6 +39,9 @@
 static void INJECT_test_func(void);
 static pascal void INJECT_EventLoopTimerEntry(EventLoopTimerRef inTimer, void *p);
 
+#ifdef snprintf
+#undef snprintf
+#endif
 #define DEFWRAP(func) __typeof__(&func) func
 typedef struct {
     /* internal */
