--- Makefile.orig	2016-06-18 12:35:35.000000000 -0700
+++ Makefile	2018-11-11 14:22:16.000000000 -0700
@@ -19,31 +19,7 @@
 cc-option = $(shell if $(CC) $(1) -S -o /dev/null -xc /dev/null \
             > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi ;)
 
-ifeq "$(X_ARCH)" "i686"
-CFLAGS += -march=i686
-CFLAGS += $(call cc-option,-mtune=pentium4,)
-CFLAGS += $(call cc-option,-mfpmath=sse,)
-endif
-
-# TODO: improve to match i[3456]86
-ifneq  "$(X_ARCH)" "i686"
-CFLAGS += -fPIC
-endif
-
-ifeq "$(X_ARCH)" "sparc64"
-CFLAGS += -mcpu=ultrasparc
-CFLAGS += $(call cc-option,-mtune=niagara,)
-endif
-
-CFLAGS += $(call cc-option,-march=native)
-
 CFLAGS += -funroll-loops -fomit-frame-pointer
-CFLAGS += $(call cc-option,-funswitch-loops,)
-CFLAGS += $(call cc-option,-fpeel-loops,)
-CFLAGS += $(call cc-option,-ftracer,)
-CFLAGS += $(call cc-option,-funit-at-a-time,)
-CFLAGS += $(call cc-option,-frename-registers,)
-CFLAGS += $(call cc-option,-ftree-vectorize,)
 
 # we have some unimplemented colorspaces in the Image::iterator :-(
 CFLAGS += $(call cc-option,-Wno-switch -Wno-switch-enum,)
