--- Makefile.orig	2005-10-09 13:00:29.000000000 -0700
+++ Makefile	2008-11-11 01:11:39.000000000 -0800
@@ -1,6 +1,6 @@
 
-MANDIR=$(DESTDIR)/usr/share/man/man6/
-BINDIR=$(DESTDIR)/usr/games/
+MANDIR=$(DESTDIR)__PREFIX__/share/man/man6
+BINDIR=$(DESTDIR)__PREFIX__/bin
 
 CC=gcc
 MKDIR=mkdir -p
@@ -35,7 +35,7 @@
 	$(RM) $(OBJS) $(OUTPUT) config.h
 
 
-install:
+install: all
 	$(MKDIR) $(BINDIR) $(MANDIR)
 	$(INSTALL) $(OUTPUT) $(BINDIR)
 	$(INSTALL) -m 644 $(MANS) $(MANDIR)
