diff --git a/setup.py b/setup.py
index 5f2e4a5..9163cab 100644
--- setup.py
+++ setup.py
@@ -420,6 +420,11 @@ class BuildConfiguration(object):
                     extra_libraries = os.environ["IGRAPH_EXTRA_DYNAMIC_LIBRARIES"].split(',')
                     buildcfg.libraries.extend(extra_libraries)
 
+                # Remove C++ standard library as we will use the C++ linker
+                for lib in ("c++", "stdc++"):
+                    if lib in buildcfg.libraries:
+                        buildcfg.libraries.remove(lib)
+
                 # Prints basic build information
                 buildcfg.print_build_info()
 
@@ -664,11 +669,6 @@ class BuildConfiguration(object):
     def replace_static_libraries(self, only=None, exclusions=None):
         """Replaces references to libraries with full paths to their static
         versions if the static version is to be found on the library path."""
-        building_on_windows = building_on_windows_msvc()
-
-        if not building_on_windows and "stdc++" not in self.libraries:
-            self.libraries.append("stdc++")
-
         if exclusions is None:
             exclusions = []
 
@@ -757,6 +757,7 @@ buildcfg.process_args_from_command_line()
 
 # Define the extension
 sources = glob.glob(os.path.join("src", "_igraph", "*.c"))
+sources.append(os.path.join("src", "_igraph", "force_cpp_linker.cpp"))
 igraph_extension = Extension("igraph._igraph", sources)
 
 description = """Python interface to the igraph high performance graph
diff --git a/src/_igraph/force_cpp_linker.cpp b/src/_igraph/force_cpp_linker.cpp
new file mode 100644
index 0000000..d78b8f3
--- /dev/null
+++ src/_igraph/force_cpp_linker.cpp
@@ -0,0 +1,4 @@
+/* The purpose of this file is to make Python use the C++ linker instead of
+ * the standard C linker because igraph's core static library needs the C++
+ * standard library */
+
