--- setup.py.orig	2013-11-30 18:55:31.000000000 -0500
+++ setup.py	2014-03-10 09:54:35.000000000 -0400
@@ -44,7 +44,7 @@
 
 ###########################################################################
 
-LIBIGRAPH_FALLBACK_INCLUDE_DIRS = ['/usr/include/igraph', '/usr/local/include/igraph']
+LIBIGRAPH_FALLBACK_INCLUDE_DIRS = ['@PREFIX@/include/igraph']
 LIBIGRAPH_FALLBACK_LIBRARIES = ['igraph']
 LIBIGRAPH_FALLBACK_LIBRARY_DIRS = []
 
@@ -639,11 +639,11 @@
     }
 )
 
-if "macosx" in get_platform() and "bdist_mpkg" in sys.argv:
-    # OS X specific stuff to build the .mpkg installer
-    options["data_files"] = [ \
-            ('/usr/local/lib', [os.path.join('..', '..', 'fatbuild', 'libigraph.0.dylib')])
-    ]
+#if "macosx" in get_platform() and "bdist_mpkg" in sys.argv:
+#    # OS X specific stuff to build the .mpkg installer
+#    options["data_files"] = [ \
+#            ('/usr/local/lib', [os.path.join('..', '..', 'fatbuild', 'libigraph.0.dylib')])
+#    ]
 
 if sys.version_info > (3, 0):
     if build_py is None:
