--- rest-client-2.1.0.gemspec	2024-02-18 20:52:52
+++ rest-client-2.1.0.gemspec	2024-02-18 20:56:05
@@ -27,26 +27,26 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_development_dependency(%q<webmock>.freeze, ["~> 2.0"])
-    s.add_development_dependency(%q<rspec>.freeze, ["~> 3.0"])
-    s.add_development_dependency(%q<pry>.freeze, ["~> 0"])
-    s.add_development_dependency(%q<pry-doc>.freeze, ["~> 0"])
-    s.add_development_dependency(%q<rdoc>.freeze, [">= 2.4.2", "< 6.0"])
-    s.add_development_dependency(%q<rubocop>.freeze, ["~> 0.49"])
-    s.add_runtime_dependency(%q<http-accept>.freeze, [">= 1.7.0", "< 2.0"])
-    s.add_runtime_dependency(%q<http-cookie>.freeze, [">= 1.0.2", "< 2.0"])
-    s.add_runtime_dependency(%q<mime-types>.freeze, [">= 1.16", "< 4.0"])
-    s.add_runtime_dependency(%q<netrc>.freeze, ["~> 0.8"])
+    s.add_development_dependency(%q<webmock>.freeze, [">= 2.0"])
+    s.add_development_dependency(%q<rspec>.freeze, [">= 3.0"])
+    s.add_development_dependency(%q<pry>.freeze, [">= 0"])
+    s.add_development_dependency(%q<pry-doc>.freeze, [">= 0"])
+    s.add_development_dependency(%q<rdoc>.freeze, [">= 2.4.2"])
+    s.add_development_dependency(%q<rubocop>.freeze, [">= 0.49"])
+    s.add_runtime_dependency(%q<http-accept>.freeze, [">= 1.7.0"])
+    s.add_runtime_dependency(%q<http-cookie>.freeze, [">= 1.0.2"])
+    s.add_runtime_dependency(%q<mime-types>.freeze, [">= 1.16"])
+    s.add_runtime_dependency(%q<netrc>.freeze, [">= 0.8"])
   else
-    s.add_dependency(%q<webmock>.freeze, ["~> 2.0"])
-    s.add_dependency(%q<rspec>.freeze, ["~> 3.0"])
-    s.add_dependency(%q<pry>.freeze, ["~> 0"])
-    s.add_dependency(%q<pry-doc>.freeze, ["~> 0"])
-    s.add_dependency(%q<rdoc>.freeze, [">= 2.4.2", "< 6.0"])
-    s.add_dependency(%q<rubocop>.freeze, ["~> 0.49"])
-    s.add_dependency(%q<http-accept>.freeze, [">= 1.7.0", "< 2.0"])
-    s.add_dependency(%q<http-cookie>.freeze, [">= 1.0.2", "< 2.0"])
-    s.add_dependency(%q<mime-types>.freeze, [">= 1.16", "< 4.0"])
-    s.add_dependency(%q<netrc>.freeze, ["~> 0.8"])
+    s.add_dependency(%q<webmock>.freeze, [">= 2.0"])
+    s.add_dependency(%q<rspec>.freeze, [">= 3.0"])
+    s.add_dependency(%q<pry>.freeze, [">= 0"])
+    s.add_dependency(%q<pry-doc>.freeze, [">= 0"])
+    s.add_dependency(%q<rdoc>.freeze, [">= 2.4.2"])
+    s.add_dependency(%q<rubocop>.freeze, [">= 0.49"])
+    s.add_dependency(%q<http-accept>.freeze, [">= 1.7.0"])
+    s.add_dependency(%q<http-cookie>.freeze, [">= 1.0.2"])
+    s.add_dependency(%q<mime-types>.freeze, [">= 1.16"])
+    s.add_dependency(%q<netrc>.freeze, [">= 0.8"])
   end
 end
