--- Platform/Linux/CreateRedist/install.sh.orig	2012-01-04 05:35:46.000000000 -0600
+++ Platform/Linux/CreateRedist/install.sh	2012-08-03 18:06:13.000000000 -0500
@@ -71,11 +71,11 @@
 	exit 1
 fi
 
-INSTALL_LIB=$rootfs/usr/lib
-INSTALL_BIN=$rootfs/usr/bin
-INSTALL_INC=$rootfs/usr/include/ni
-INSTALL_VAR=$rootfs/var/lib/ni
-INSTALL_JAR=$rootfs/usr/share/java
+INSTALL_LIB=$rootfs@PREFIX@/lib
+INSTALL_BIN=$rootfs@PREFIX@/bin
+INSTALL_INC=$rootfs@PREFIX@/include/ni
+INSTALL_VAR=$rootfs@PREFIX@/var/lib/ni
+INSTALL_JAR=$rootfs@PREFIX@/share/java
 
 # make all calls into OpenNI run in this filesystem
 export OPEN_NI_INSTALL_PATH=$rootfs
@@ -107,12 +107,6 @@
 	mkdir -p $INSTALL_VAR
 	printf "OK\n"
 
-	# register modules
-	for module in $MODULES; do
-		printf "registering module '$module'..."
-		$INSTALL_BIN/niReg -r $INSTALL_LIB/$module
-		printf "OK\n"
-	done
 
 	# mono
 	if [ -f $rootfs/usr/bin/gmcs -a -f Bin/OpenNI.net.dll ]; then
