--- barcode-0.98/main.c.orig	2002-03-02 08:31:07.000000000 +1100
+++ barcode-0.98/main.c	2012-07-16 12:40:37.000000000 +1000
@@ -299,6 +299,9 @@ int get_page_geometry(void *arg)
     int n;
     double dpw, dph; /* page width, height in mm or inches */
     static char tmpstr[20];
+    if (!arg) {
+        return -1;
+    }
     page_name = arg; /* if undecipherable, we won't run the program :) */
     /*
      * try to decode a "mm" string (eg. "210mmx297mm" or "210x297mm")
