diff --git GitUp/Application/AppDelegate.m GitUp/Application/AppDelegate.m
index 1042982..578dd3e 100644
--- GitUp/Application/AppDelegate.m
+++ GitUp/Application/AppDelegate.m
@@ -253,6 +253,9 @@ - (void)applicationDidFinishLaunching:(NSNotification*)notification {
   // First launch has completed
   [[NSUserDefaults standardUserDefaults] setBool:NO forKey:kUserDefaultsKey_FirstLaunch];

+  // Skip command line tool installation
+  [[NSUserDefaults standardUserDefaults] setBool:true forKey:kUserDefaultsKey_SkipInstallCLT];
+
   // Create tool message port
   CFMessagePortContext context = {0, (__bridge void*)self, NULL, NULL, NULL};
   _messagePort = CFMessagePortCreateLocal(kCFAllocatorDefault, CFSTR(kToolPortName), _MessagePortCallBack, &context, NULL);
diff --git GitUp/Application/Base.lproj/MainMenu.xib GitUp/Application/Base.lproj/MainMenu.xib
index 528efb4..3602656 100644
--- GitUp/Application/Base.lproj/MainMenu.xib
+++ GitUp/Application/Base.lproj/MainMenu.xib
@@ -31,13 +31,6 @@
                                     <action selector="checkForUpdates:" target="Voe-Tx-rLC" id="Bx2-39-dyN"/>
                                 </connections>
                             </menuItem>
-                            <menuItem isSeparatorItem="YES" id="gLG-bh-eJk"/>
-                            <menuItem title="Install Command Line Tool…" id="f9m-On-7Ly">
-                                <modifierMask key="keyEquivalentModifierMask"/>
-                                <connections>
-                                    <action selector="installTool:" target="Voe-Tx-rLC" id="vFd-eA-ejg"/>
-                                </connections>
-                            </menuItem>
                             <menuItem isSeparatorItem="YES" id="dst-2l-Azr"/>
                             <menuItem title="Preferences…" keyEquivalent="," id="smQ-Fd-MMe">
                                 <connections>
