--- CMakeLists.txt	2021-11-20 12:44:53
+++ CMakeLists.txt	2024-05-12 04:28:29
@@ -9,20 +9,12 @@
 
 # ftxui
 
-FetchContent_Declare(ftxui
-  GIT_REPOSITORY https://github.com/ArthurSonzogni/FTXUI
-  GIT_TAG 4188ee2c046ced00cd3accba4cde2318587f8c8a
-)
+find_package(ftxui)
 
 # fmt
 
-FetchContent_Declare(fmt
-  GIT_REPOSITORY https://github.com/fmtlib/fmt
-  GIT_TAG d141cdbeb0fb422a3fb7173b285fd38e0d1772dc
-)
+find_package(fmt)
 
-FetchContent_MakeAvailable(ftxui fmt)
-
 # TermME
 
 set(SRCS
@@ -32,8 +24,8 @@
 add_executable(TermME ${SRCS})
 target_include_directories(TermME PRIVATE include)
 target_link_libraries(TermME
-  PRIVATE ftxui::screen
-  PRIVATE ftxui::dom
-  PRIVATE ftxui::component
-  PRIVATE fmt::fmt
+  PUBLIC ftxui::screen
+  PUBLIC ftxui::dom
+  PUBLIC ftxui::component
+  PUBLIC fmt::fmt
 )
