--- src/org/apache/html/dom/HTMLElementImpl.java.orig	2013-01-21 10:27:18.000000000 -0800
+++ src/org/apache/html/dom/HTMLElementImpl.java	2013-01-21 10:31:36.000000000 -0800
@@ -20,6 +20,7 @@
 
 import org.apache.xerces.dom.ElementImpl;
 import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
 import org.w3c.dom.html.HTMLElement;
@@ -254,4 +255,7 @@
         return null;
     }
 
+    public Document getContentDocument() {
+        throw new UnsupportedOperationException();
+    }
 }
