diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index a91524b9664..09005661e6c 100644
--- Source/WTF/wtf/Platform.h
+++ Source/WTF/wtf/Platform.h
@@ -662,10 +662,6 @@
 #define HAVE_READLINE 1
 #define HAVE_SYS_TIMEB_H 1

-#if __has_include(<mach/mach_exc.defs>) && !(PLATFORM(WATCHOS) || PLATFORM(APPLETV))
-#define HAVE_MACH_EXCEPTIONS 1
-#endif
-
 #if !PLATFORM(GTK)
 #define USE_ACCELERATE 1
 #endif
