--- Source/bmalloc/bmalloc/BPlatform.h.orig	2026-04-29 22:00:00
+++ Source/bmalloc/bmalloc/BPlatform.h	2026-04-29 22:00:00
@@ -429,9 +429,5 @@
 #endif
 
 #if !defined(BUSE_TZONE)
-#if BUSE(LIBPAS) && BOS(DARWIN) && (BCPU(ARM64) || BCPU(X86_64))
-#define BUSE_TZONE 1
-#else
 #define BUSE_TZONE 0
 #endif
-#endif
--- Source/WTF/wtf/PlatformUse.h.orig	2026-04-29 22:00:00
+++ Source/WTF/wtf/PlatformUse.h	2026-04-29 22:00:00
@@ -305,12 +305,8 @@
 #endif
 
 #if !defined(USE_TZONE_MALLOC)
-#if (CPU(ARM64) || CPU(X86_64)) && OS(DARWIN) && (__SIZEOF_POINTER__ == 8)
-#define USE_TZONE_MALLOC 1
-#else
 #define USE_TZONE_MALLOC 0
 #endif
-#endif
 
 #if !defined(USE_PROTECTED_JIT)
 #if CPU(ADDRESS64) && OS(DARWIN) && USE(APPLE_INTERNAL_SDK)
