--- Source/WebCore/page/SecurityOriginData.cpp.orig	2018-06-13 14:43:40.000000000 +0800
+++ Source/WebCore/page/SecurityOriginData.cpp	2018-06-13 14:37:12.000000000 +0800
@@ -33,8 +33,6 @@
 #include <wtf/text/CString.h>
 #include <wtf/text/StringBuilder.h>
 
-using namespace WebCore;
-
 namespace WebCore {
 
 SecurityOriginData SecurityOriginData::fromSecurityOrigin(const SecurityOrigin& securityOrigin)
--- Source/WebCore/page/TextIndicator.cpp.orig	2018-06-13 21:38:47.000000000 +0800
+++ Source/WebCore/page/TextIndicator.cpp	2018-06-13 21:39:03.000000000 +0800
@@ -50,8 +50,6 @@
 #include "SelectionRect.h"
 #endif
 
-using namespace WebCore;
-
 namespace WebCore {
 
 static bool initializeIndicator(TextIndicatorData&, Frame&, const Range&, FloatSize margin, bool indicatesCurrentSelection);
--- Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectEventListener.cpp.orig	2018-06-13 23:03:53.000000000 +0800
+++ Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectEventListener.cpp	2018-06-13 23:17:03.000000000 +0800
@@ -25,7 +25,9 @@
 #include <WebCore/Event.h>
 #include <wtf/HashMap.h>
 
-using namespace WebCore;
+using WebCore::Event;
+using WebCore::EventTarget;
+using WebCore::ScriptExecutionContext;
 
 namespace WebKit {
 
--- Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectNodeFilterCondition.cpp.orig	2018-06-13 23:18:03.000000000 +0800
+++ Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/GObjectNodeFilterCondition.cpp	2018-06-13 23:29:20.000000000 +0800
@@ -22,7 +22,7 @@
 #include "WebKitDOMNodePrivate.h"
 #include <WebCore/NodeFilter.h>
 
-using namespace WebCore;
+using WebCore::Node;
 
 namespace WebKit {
 
