diff --git Source/JavaScriptCore/tools/JSDollarVM.cpp Source/JavaScriptCore/tools/JSDollarVM.cpp
index 92ffb85b..481763fa 100644
--- Source/JavaScriptCore/tools/JSDollarVM.cpp
+++ Source/JavaScriptCore/tools/JSDollarVM.cpp
@@ -179,7 +179,7 @@ private:
 class ElementHandleOwner : public WeakHandleOwner {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    bool isReachableFromOpaqueRoots(Handle<JSC::Unknown> handle, void*, SlotVisitor& visitor, const char** reason) override
+    bool isReachableFromOpaqueRoots(JSC::Handle<JSC::Unknown> handle, void*, SlotVisitor& visitor, const char** reason) override
     {
         if (UNLIKELY(reason))
             *reason = "JSC::Element is opaque root";
