--- vsftpd.conf.orig	2005-05-21 00:52:54.000000000 +0200
+++ vsftpd.conf		2006-01-09 20:51:29.000000000 +0100
@@ -100,4 +100,13 @@
 # sites. However, some broken FTP clients such as "ncftp" and "mirror" assume
 # the presence of the "-R" option, so there is a strong case for enabling it.
 #ls_recurse_enable=YES
-
+#
+# Name of pam module
+pam_service_name=ftpd
+#
+# Chroot local users by default
+chroot_local_user=YES
+#
+# Turn on vsftpd into the standalone mode
+# If you run vsftpd through a super-server, change to NO.
+listen=YES
