--- vsftpd.conf.old	2010-02-20 13:55:06.000000000 -0800
+++ vsftpd.conf	2010-02-20 13:56:58.000000000 -0800
@@ -92,7 +92,7 @@
 # You may specify an explicit list of local users to chroot() to their home
 # directory. If chroot_local_user is YES, then this list becomes a list of
 # users to NOT chroot().
-#chroot_local_user=YES
+chroot_local_user=YES
 #chroot_list_enable=YES
 # (default follows)
 #chroot_list_file=/etc/vsftpd.chroot_list
@@ -112,3 +112,6 @@
 # sockets, you must run two copies of vsftpd with two configuration files.
 # Make sure, that one of the listen options is commented !!
 #listen_ipv6=YES
+#
+# Name of pam module
+pam_service_name=ftpd
