--- vsftpd.conf.old	2013-07-18 16:57:30.000000000 +0200
+++ vsftpd.conf	2013-07-18 16:58:25.000000000 +0200
@@ -95,7 +95,7 @@
 # (Warning! chroot'ing can be very dangerous. If using chroot, make sure that
 # the user does not have write access to the top level directory within the
 # chroot)
-#chroot_local_user=YES
+chroot_local_user=YES
 #chroot_list_enable=YES
 # (default follows)
 #chroot_list_file=/etc/vsftpd.chroot_list
@@ -115,3 +115,6 @@
 # sockets, you must run two copies of vsftpd with two configuration files.
 # Make sure, that one of the listen options is commented !!
 #listen_ipv6=YES
+#
+# Name of pam module
+pam_service_name=ftpd
