--- setup.py.orig	2011-11-19 20:48:45.000000000 -0500
+++ setup.py	2012-03-20 14:31:50.000000000 -0400
@@ -15,8 +15,7 @@
         osx_ver = float('.'.join(osx_ver.split('.')[:2]))
 
         default_libs = []
-        default_cxxflags = ['-arch', 'i386', '-arch', 'x86_64',
-                '-isysroot', '/Developer/SDKs/MacOSX%s.sdk' % osx_ver]
+        default_cxxflags = []
         default_ldflags = default_cxxflags[:] + ["-Wl,-framework,OpenCL"]
 
     else:
