--- config.sh	2021-03-22 07:05:13.000000000 -0400
+++ config.sh	2021-03-22 07:06:15.000000000 -0400
@@ -101,10 +101,10 @@
 check_pkg "libcurl" || check_custom "libcurl" "curl-config" || fail "libcurl"
 check_pkg "libxml-2.0" || check_custom "libxml2" "xml2-config" || fail "libxml2"
 check_pkg "stfl" || fail "stfl"
+check_pkg "ncurses" || fail "ncurses"
 ( check_pkg "json" "" 0.11 || check_pkg "json-c" "" 0.11 ) || fail "json-c"
 
 if [ `uname -s` = "Darwin" ]; then
-	check_custom "ncurses5.4" "ncurses5.4-config" || fail "ncurses5.4"
     # rand crate needs Security framework, and rustc doesn't (can't) link it
     # into libnewsboat.a
     echo 'LDFLAGS+=-framework Security' >> config.mk
