--- eckit/src/eckit/sql/sqly.y.orig	2020-07-22 22:45:15.000000000 +0900
+++ eckit/src/eckit/sql/sqly.y	2020-08-19 16:08:52.000000000 +0900
@@ -1,7 +1,7 @@
-%pure-parser
-%lex-param {yyscan_t scanner}
+%define api.pure
+%lex-param {void *scanner}
 %lex-param {eckit::sql::SQLSession* session}
-%parse-param {yyscan_t scanner}
+%parse-param {void *scanner}
 %parse-param {eckit::sql::SQLSession* session}
 
 %{
