--- ./apt-pkg/rpm/rpmpm.cc.orig	2007-06-20 13:16:55.000000000 +0200
+++ ./apt-pkg/rpm/rpmpm.cc	2007-06-20 13:21:31.000000000 +0200
@@ -850,7 +850,7 @@
    if (_config->FindB("RPM::NoDeps", false) == false) {
       rc = rpmtsCheck(TS);
       probs = rpmtsProblems(TS);
-      if (rc || probs->numProblems > 0) {
+      if (rc || rpmpsNumProblems(probs) > 0) {
 	 rpmpsPrint(NULL, probs);
 	 rpmpsFree(probs);
 	 _error->Error(_("Transaction set check failed"));
@@ -906,7 +906,7 @@
 
    if (rc > 0) {
       _error->Error(_("Error while running transaction"));
-      if (probs->numProblems > 0)
+      if (rpmpsNumProblems(probs) > 0)
 	 rpmpsPrint(stderr, probs);
    } else {
       Success = true;
