--- Makefile.in.orig	2011-03-31 14:08:36.000000000 -0500
+++ Makefile.in	2012-05-19 17:51:45.000000000 -0500
@@ -74,7 +74,7 @@
 SRCREL_VERSION      = 1.0.0
 SRCREL_ABI_VERSION  := $(word 1,$(subst ., ,$(SRCREL_VERSION))).$(word 2,$(subst ., ,$(SRCREL_VERSION)))
 
-MODULE		    = js
+MODULE		    = mozjs185
 ifeq (,$(HOST_BIN_SUFFIX))
 LIBRARY_NAME	    = mozjs185
 else
--- mozjs185.pc.in.orig	2011-03-31 14:08:36.000000000 -0500
+++ mozjs185.pc.in	2012-05-19 17:37:45.000000000 -0500
@@ -7,4 +7,4 @@
 Version: %MOZILLA_VERSION%
 Requires: nspr >= 4.7
 Libs: -L${libdir} -lmozjs185
-Cflags: -I${includedir}/js
+Cflags: -I${includedir}/mozjs185
