From 44f8d4d1e5b938cd998b90b21c2ca7163ad84fed Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 18 Mar 2023 23:47:38 +0700
Subject: [PATCH] Fix big endian for PowerPC

---
 src/bytes.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/bytes.cpp b/src/bytes.cpp
index 28277be..871e15b 100644
--- src/bytes.cpp
+++ src/bytes.cpp
@@ -4,7 +4,7 @@
 using namespace Rcpp;
 
 // good enough for now, I suppose
-#if defined(__sparc__) || defined(__sparc)
+#if defined(__sparc__) || defined(__sparc) || defined(__ppc__) || defined(__ppc64__)
 #define IS_BIG_ENDIAN true
 #else
 #define IS_BIG_ENDIAN false
