https://trac.macports.org/ticket/49317

Upstream-Status: Inappropriate [enable feature]

Index: setup.py
===================================================================
--- setup.py.orig
+++ setup.py
@@ -1158,7 +1158,6 @@ class PyBuildExt(build_ext):
                 sqlite_defines.append(('MODULE_NAME', '\\"sqlite3\\"'))
 
             # Comment this out if you want the sqlite3 module to be able to load extensions.
-            sqlite_defines.append(("SQLITE_OMIT_LOAD_EXTENSION", "1"))
 
             if host_platform == 'darwin':
                 # In every directory on the search path search for a dynamic
