--- Lib/sysconfig.py	2010-07-06 19:56:06.000000000 +0300
+++ Lib/sysconfig.py	2010-07-06 19:56:45.000000000 +0300
@@ -646,7 +646,7 @@
 
                 archs = re.findall('-arch\s+(\S+)', cflags)
                 archs.sort()
-                archs = tuple(archs)
+                archs = tuple(sorted(list(set(archs))))
 
                 if len(archs) == 1:
                     machine = archs[0]
