bll-g7:bll$ hg diff CMakeLists.txt
diff -r 29947e7eac91 CMakeLists.txt
--- CMakeLists.txt    Mon Nov 03 07:42:19 2025 -0800
+++ CMakeLists.txt    Mon Nov 03 14:45:40 2025 -0800
@@ -835,7 +835,7 @@
   set (CMAKE_REQUIRED_LIBRARIES ${Intl_LIBRARY})
 endif()
 if (Iconv_LIBRARY AND NOT Iconv_LIBRARY STREQUAL "NOTFOUND")
-  set (CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} ${Iconv_LIBRARY}")
+  set (CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES};${Iconv_LIBRARY}")
 endif()
 check_symbol_exists (bindtextdomain libintl.h _lib_bindtextdomain)
 check_symbol_exists (gettext libintl.h _lib_gettext)
