--- CMakeLists.txt.orig	2017-06-20 19:46:55.000000000 -0700
+++ CMakeLists.txt	2017-06-20 19:47:56.000000000 -0700
@@ -126,19 +126,6 @@
 endif()
 
 
-#build engine without freepascal
-if(BUILD_ENGINE_C)
-    find_package(Clang REQUIRED)
-
-    if(${CLANG_VERSION} VERSION_LESS "3.0")
-        message(FATAL_ERROR "LLVM/Clang compiler required version is 3.0 but version ${CLANG_VERSION} was found!")
-    endif()
-
-    set(CMAKE_C_COMPILER ${CLANG_EXECUTABLE})
-    set(CMAKE_CXX_COMPILER ${CLANG_EXECUTABLE})
-endif()
-
-
 #server
 if(NOT NOSERVER)
     add_subdirectory(gameServer)
