--- packages/plot/gnuintfc.red.orig	2022-07-22 09:36:50.000000000 -0500
+++ packages/plot/gnuintfc.red	2022-07-22 09:56:19.000000000 -0500
@@ -128,7 +128,7 @@
                 ("vt102" . "tek40xx")
            ));
 
-   plotcommand!* := find!-gnuplot();
+   plotcommand!* := "@PREFIX@/bin/gnuplot";
 
    plotcleanup!* :=                  % delete scratch files
        {bldmsg("rm %wplotdt*",plottmp!*),bldmsg("rm %wplotcmds*",plottmp!*)};
@@ -259,7 +259,7 @@
   !*plotusepipe := t;
 % find-gnuplot returns the full name of a version of gnuplot (if it can
 % find one).
-  plotcommand!* := find!-gnuplot();
+  plotcommand!* := "@PREFIX@/bin/gnuplot";
 % The Cygwin case can be "funny" here, and the case I wish to trap is
 % where a Cygwin version of Reduce is trying to use the native Windows
 % version of gnuplot. I believe I can detect this by seeing of
