--- configure.ac.orig	2016-08-15 10:54:00.000000000 -0700
+++ configure.ac	2016-08-15 13:49:15.000000000 -0700
@@ -137,10 +137,10 @@
 if test "x$found_introspection" = "xyes"; then
     # Used to determine where Python wants the .gi file to be installed ...
     # if Python is not installed, PYTHON2 and PYTHON3 will be empty
-    PYTHON="import gi; print(gi._overridesdir)"
+    PYTHON_SCRIPT="import gi; print(gi._overridesdir)"
     AS_IF([test "x$ac_py2girdir" = "x"],
     [
-        PYTHON2_GIDIR=$(python2 -c "$PYTHON")
+        PYTHON2_GIDIR=$("$PYTHON2" -c "$PYTHON_SCRIPT")
     ],
     [
         PYTHON2_GIDIR=$ac_py2girdir
@@ -149,7 +149,7 @@
 
     AS_IF([test "x$ac_py3girdir" = "x"],
     [
-        PYTHON3_GIDIR=$(python3 -c "$PYTHON")
+        PYTHON3_GIDIR=$("$PYTHON3" -c "$PYTHON_SCRIPT")
     ],
     [
         PYTHON3_GIDIR=$ac_py3girdir
