--- meson.build.orig	2020-08-30 20:01:01.000000000 -0700
+++ meson.build	2020-08-30 20:02:07.000000000 -0700
@@ -36,7 +36,7 @@
 
 if introspection_available
   if python2_girdir != 'no'
-    python2 = find_program('python2', required : false)
+    python2 = find_program('@@PYTHON2_BIN@@', required : false)
     if python2.found()
       py2gi = run_command(python2, python_gi_test_args)
       if py2gi.returncode() == 0
@@ -50,7 +50,7 @@
   endif
 
   if python3_girdir != 'no'
-    python3 = find_program('python3', required : false)
+    python3 = find_program('@@PYTHON3_BIN@@', required : false)
     if python3.found()
       py3gi = run_command(python3, python_gi_test_args)
       if py3gi.returncode() == 0
