--- meson.build.orig	2024-04-19 13:36:49.000000000 -0400
+++ meson.build	2024-04-19 13:37:45.000000000 -0400
@@ -7,7 +7,7 @@
                               # and meson.add_install_script(python3, ...)
 )
 
-python3 = find_program('python3', version: '>= 3.5')
+python3 = find_program('@@PYTHON3_BIN@@', version: '>= 3.5')
 
 # Use these instead of meson.source_root() and meson.build_root().
 # source_root() and build_root() are not useful, if this is a subproject.
--- skeletonmm/meson.build.orig	2024-04-19 13:40:40.000000000 -0400
+++ skeletonmm/meson.build	2024-04-19 13:41:00.000000000 -0400
@@ -41,7 +41,7 @@
 project_source_root = meson.project_source_root()
 project_build_root = meson.project_build_root()
 
-python3 = find_program('python3', version: '>= 3.5')
+python3 = find_program('@@PYTHON3_BIN@@', version: '>= 3.5')
 
 # Do we build from a git repository?
 # Suppose we do if and only if the meson.build file is tracked by git.
