diff --git CMakeLists.txt.orig CMakeLists.txt
index 25ded3c..d7a2160 100644
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -261,11 +261,11 @@ configure_file(tesseract.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc @ONLY
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib/pkgconfig)
 install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
 install(TARGETS libtesseract EXPORT TesseractTargets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
-install(EXPORT TesseractTargets DESTINATION cmake)
+install(EXPORT TesseractTargets DESTINATION lib/cmake/tesseract)
 install(FILES
     ${CMAKE_BINARY_DIR}/TesseractConfig.cmake
     ${CMAKE_BINARY_DIR}/TesseractConfig-version.cmake
-    DESTINATION cmake)
+    DESTINATION lib/cmake/tesseract)
 
 install(FILES
     # from api/makefile.am
