--- vidoas.orig
+++ vidoas
@@ -15,17 +15,17 @@
 # TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 # PERFORMANCE OF THIS SOFTWARE.
 
 # Edit a temporary copy of the doas.conf file and check it for syntax
 # errors before installing it as the actual doas.conf file.
 
 set -eu
 
-PATH=/bin:/usr/bin:/usr/local/bin
+PATH=/bin:/usr/bin:/usr/local/bin:@PREFIX@/bin
 export PATH
 
 PROG="${0##*/}"
 
 umask 022
 
 DOAS_CONF=@DOAS_CONF@
 doas_conf_mode="0600"
