From c730618c01f7bf34b635056881d8ec271c2fe75d Mon Sep 17 00:00:00 2001
From: Davide Gerhard <rainbow@irh.it>
Date: Sun, 19 Jul 2020 19:47:02 +0200
Subject: [PATCH] Makefile.mac: add subrx and css

---
 Makefile.mac | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git Makefile.mac Makefile.mac
index 4e15ce6..cbd1190 100644
--- Makefile.mac
+++ Makefile.mac
@@ -61,6 +61,7 @@ PROGRAM=linhpsdr
 
 SOURCES=\
 main.c\
+css.c\
 audio.c\
 version.c\
 discovered.c\
@@ -106,10 +107,12 @@ frequency.c\
 rigctl.c\
 error_handler.c\
 radio_info.c\
-bpsk.c
+bpsk.c\
+subrx.c
 
 HEADERS=\
 main.h\
+css.h\
 audio.h\
 version.h\
 discovered.h\
@@ -156,10 +159,12 @@ frequency.h\
 rigctl.h\
 error_handler.h\
 radio_info.h\
-bpsk.h
+bpsk.h\
+subrx.h
 
 OBJS=\
 main.o\
+css.o\
 audio.o\
 version.o\
 discovered.o\
@@ -205,7 +210,8 @@ frequency.o\
 rigctl.o\
 error_handler.o\
 radio_info.o\
-bpsk.o
+bpsk.o\
+subrx.o
 
 all: prebuild  $(PROGRAM) $(HEADERS) $(SOURCES) $(SOAPYSDR_SOURCES)
 
-- 
2.27.0

