Fix docdir.
--- Makefile.am.orig	2015-08-04 05:41:54.000000000 -0700
+++ Makefile.am	2015-08-04 05:44:58.000000000 -0700
@@ -13,7 +13,7 @@
 	ui 
 
 # Lugar de instalación de la documentación estandar
-gelidedocdir = ${prefix}/doc/gelide
+gelidedocdir = ${prefix}/share/doc/gelide
 # Archivos de documentación estandar a instalar
 gelidedoc_DATA = \
 	AUTHORS \
--- doc/Makefile.am.orig	2015-08-04 05:43:15.000000000 -0700
+++ doc/Makefile.am	2015-08-04 05:45:25.000000000 -0700
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 # Lugar de instalación de la documentación
-gelide_docdir = ${prefix}/doc/gelide
+gelide_docdir = ${prefix}/share/doc/gelide
 gelide_doc_DATA = \
 	SystemsAndEmulators-HowTo.es_ES.txt \
 	Gelide-configuration-file-format.txt
