--- Makefile.in.orig	2009-04-25 12:05:57.000000000 -0500
+++ Makefile.in	2010-02-20 00:27:56.000000000 -0600
@@ -242,7 +242,7 @@
 
 
 # Lugar de instalación de la documentación estandar
-gelidedocdir = ${prefix}/doc/gelide
+gelidedocdir = ${prefix}/share/doc/gelide
 # Archivos de documentación estandar a instalar
 gelidedoc_DATA = \
 	README \
--- doc/Makefile.in.orig	2009-04-25 12:05:56.000000000 -0500
+++ doc/Makefile.in	2010-02-20 00:46:49.000000000 -0600
@@ -199,7 +199,7 @@
 top_srcdir = @top_srcdir@
 
 # Lugar de instalación de la documentación
-gelidedocdir = ${prefix}/doc/gelide
+gelidedocdir = ${prefix}/share/doc/gelide
 gelidedoc_DATA = \
 	Systems&Emulators-HowTo.es_ES.txt
 
