--- configure.py.orig
+++ configure.py
@@ -2147,7 +2147,7 @@
     # Instead we look where DBus itself is installed - which in most cases will
     # be where dbus-python is also installed.
     if target_config.pydbus_inc_dir != '':
-        target_config.dbus_inc_dirs = [target_config.pydbus_inc_dir]
+        target_config.dbus_inc_dirs.append(target_config.pydbus_inc_dir)
 
     for d in target_config.dbus_inc_dirs:
         if os.access(os.path.join(d, 'dbus', 'dbus-python.h'), os.F_OK):
