--- configure.py.orig	2020-09-12 00:00:00.000000000 +0000
+++ configure.py	2020-09-14 19:28:50.000000000 -0500
@@ -2333,7 +2333,7 @@
     # Instead we look where DBus itself is installed - which in most cases will
     # be where dbus-python is also installed.
     if target_config.pydbus_inc_dir != '':
-        target_config.dbus_inc_dirs = [target_config.pydbus_inc_dir]
+        target_config.dbus_inc_dirs.append(target_config.pydbus_inc_dir)
 
     for d in target_config.dbus_inc_dirs:
         if os.access(os.path.join(d, 'dbus', 'dbus-python.h'), os.F_OK):
