--- orig.configure.py	2018-12-15 17:42:37.000000000 +0100
+++ configure.py	2018-12-15 17:50:21.000000000 +0100
@@ -2440,7 +2440,7 @@
     the target configuration.
     """
 
-    sip_flags = ['-n', 'PyQt5.sip']
+    sip_flags = ['-n', 'sip']
 
     # If we don't check for signed interpreters, we exclude the 'VendorID'
     # feature
--- designer/orig.pluginloader.cpp	2018-10-01 15:38:22.000000000 +0200
+++ designer/pluginloader.cpp	2018-12-15 17:50:50.000000000 +0100
@@ -167,7 +167,7 @@
     // Make sure we have sip.unwrapinstance.
     if (!sip_unwrapinstance)
     {
-        sip_unwrapinstance = getModuleAttr("PyQt5.sip", "unwrapinstance");
+        sip_unwrapinstance = getModuleAttr("sip", "unwrapinstance");
 
         if (!sip_unwrapinstance)
             return true;
--- qmlscene/orig.pluginloader.cpp	2018-10-01 15:38:22.000000000 +0200
+++ qmlscene/pluginloader.cpp	2018-12-15 17:51:32.000000000 +0100
@@ -412,9 +412,9 @@
 void PyQt5QmlPlugin::getSipAPI()
 {
 #if defined(SIP_USE_PYCAPSULE)
-    sip = (const sipAPIDef *)PyCapsule_Import("PyQt5.sip._C_API", 0);
+    sip = (const sipAPIDef *)PyCapsule_Import("sip._C_API", 0);
 #else
-    PyObject *c_api = getModuleAttr("PyQt5.sip", "_C_API");
+    PyObject *c_api = getModuleAttr("sip", "_C_API");
 
     if (c_api)
     {
