--- mk/elftoolchain.inc.mk	2012-02-24 23:30:51.000000000 +0100
+++ mk/elftoolchain.inc.mk	2012-02-24 23:32:03.000000000 +0100
@@ -11,7 +11,7 @@
 
 .include <bsd.own.mk>
 
-.if ${OS_HOST} == "FreeBSD"
+.if ${OS_HOST} == Darwin || ${OS_HOST} == FreeBSD
 # Simulate <bsd.inc.mk>.
 .PHONY:		incinstall
 includes:	${INCS}	incinstall
--- mk/elftoolchain.subdir.mk	2012-02-25 01:46:53.000000000 +0100
+++ mk/elftoolchain.subdir.mk	2012-02-25 01:47:40.000000000 +0100
@@ -6,7 +6,7 @@
 
 .include "$(TOP)/mk/elftoolchain.os.mk"
 
-.if ${OS_HOST} == FreeBSD
+.if ${OS_HOST} == Darwin || ${OS_HOST} == FreeBSD
 SUBDIR_TARGETS+=	clobber test
 .else
 TARGETS+=	clobber test
--- common/Makefile	2012-02-25 02:09:41.000000000 +0100
+++ common/Makefile	2012-02-25 02:09:58.000000000 +0100
@@ -5,9 +5,9 @@
 INCS=	elfdefinitions.h
 INCSDIR?=	/usr/include
 
-.PHONY:	all clean clobber depend obj
+.PHONY:	all clean install clobber depend obj
 
-all depend obj:
+all install depend obj:
 
 clean clobber:
 	rm -f ${CLEANFILES}
--- mk/elftoolchain.prog.mk	2012-03-10 13:03:40.000000000 +0100
+++ mk/elftoolchain.prog.mk	2012-03-10 13:04:46.000000000 +0100
@@ -36,6 +36,11 @@
 .endif
 .endif
 
+.if ${OS_HOST} == "Darwin"
+CFLAGS+=       -I@PREFIX@/include
+LDFLAGS+=      -L@PREFIX@/lib
+.endif
+
 #
 # Handle lex(1) and yacc(1) in a portable fashion.
 #
--- mk/elftoolchain.lib.mk	2012-03-10 13:26:28.000000000 +0100
+++ mk/elftoolchain.lib.mk	2012-03-10 13:29:02.000000000 +0100
@@ -8,6 +8,10 @@
 
 .include "${TOP}/mk/elftoolchain.os.mk"
 
+.if ${OS_HOST} == "Darwin" && !defined(SONAME)
+SONAME=	${SHLIBDIR}/${SHLIB_NAME}
+.endif
+
 .include <bsd.lib.mk>
 
 # Support a 'clobber' target.
