--- ./src/FilePanel.cpp.orig	2013-08-21 13:32:33.000000000 +0200
+++ ./src/FilePanel.cpp	2013-09-08 15:41:26.000000000 +0200
@@ -4546,6 +4546,7 @@ long FilePanel::onCmdMount(FXObject*,FXS
 
     return 1;
 }
+#endif // End #if defined(linux)
 
 
 // Run script
@@ -4650,6 +4651,7 @@ long FilePanel::onCmdGoScriptDir(FXObjec
 }
 
 
+#if defined(linux)
 // Update the Mount button
 long FilePanel::onUpdMount(FXObject* o,FXSelector sel,void*)
 {
