--- mesonpy/__init__.py.orig	2023-10-26 04:49:15
+++ mesonpy/__init__.py	2023-10-29 13:21:10
@@ -937,7 +937,7 @@
     if meson.endswith('.py'):
         cmd = [sys.executable, meson]
     else:
-        cmd = [meson]
+        cmd = ["meson-@@PYVER@@"]

     # The meson Python package is a dependency of the meson-python Python
     # package, however, it may occur that the meson Python package is installed
