--- configure.orig	Sat Oct 23 06:24:51 2004
+++ configure	Sat Oct 23 06:37:44 2004
@@ -474,17 +474,12 @@
     troff="troff -mandoc"
     echo "Warning: could not find groff"
   else
-    if test $Fnroff = "missing"
-    then
-      nroff="$Fgroff -Tlatin1 -mandoc"
-    else
-      nroff="$Fnroff -Tlatin1 -mandoc"
-    fi
+    nroff="$Fgroff -P-c -Wall -mtty-char -Tascii -mandoc"
     troff="$Fgroff -Tps -mandoc"
     jnroff="$Fgroff -Tnippon -mandocj"
   fi
   eqn="$Fgeqn -Tps"
-  neqn="$Fgeqn -Tlatin1"
+  neqn="$Fgeqn -Tascii"
   jneqn="$Fgeqn -Tnippon"
   tbl="$Fgtbl"
   col="$Fcol"
