--- src/manfile.c.orig	2006-06-01 17:16:00.000000000 -0700
+++ src/manfile.c	2006-06-01 17:37:31.000000000 -0700
@@ -147,10 +147,14 @@
 static char **
 glob_for_file (const char *dir, const char *sec, const char *name, int type) {
      char **names;
+     char shortsec[2];
 
      if (debug >= 2)
 	 gripe(CALLTRACE2, dir, sec, name, type);
 
+     shortsec[0] = sec[0];
+     shortsec[1] = '\0';
+
      if (standards & DO_IRIX) {
 	  /* try first without `sec' extension */
 	  /* maybe this should be done only for cat pages? */
@@ -158,7 +162,7 @@
      }
 
      /* try /usr/X11R6/man/man3x/XSetFont.3x */
-     names = glob_for_file_ext (dir, sec, name, sec, type);
+     names = glob_for_file_ext (dir, shortsec, name, sec, type);
 
      if (!names)
 	  return 0;		/* out-of-memory or error */
