--- man/en/whatis.man.orig	2006-06-02 12:25:41.000000000 -0700
+++ man/en/whatis.man	2006-06-02 12:25:59.000000000 -0700
@@ -24,7 +24,7 @@
 of system commands for keywords and displays the result on the
 standard output.  Only complete word matches are displayed.
 
-The whatis database is created using the command @makewhatis@.
+The whatis database is created using the command /usr/libexec/makewhatis.
 .SH AUTHOR
 John W. Eaton was the original author of
 .BR "man" .
