--- configure.orig	2008-11-22 12:14:41.000000000 +1100
+++ configure	2020-01-29 13:19:23.000000000 +1100
@@ -23294,9 +23294,9 @@
 if test -n $sslincludedir; then
      CPPFLAGS="$CPPFLAGS $sslincludedir"
 
-{ echo "$as_me:$LINENO: checking for CRYPTO_num_locks in -lcrypto" >&5
-echo $ECHO_N "checking for CRYPTO_num_locks in -lcrypto... $ECHO_C" >&6; }
-if test "${ac_cv_lib_crypto_CRYPTO_num_locks+set}" = set; then
+{ echo "$as_me:$LINENO: checking for OPENSSL_init_crypto in -lcrypto" >&5
+echo $ECHO_N "checking for OPENSSL_init_crypto in -lcrypto... $ECHO_C" >&6; }
+if test "${ac_cv_lib_crypto_OPENSSL_init_crypto+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -23314,11 +23314,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char CRYPTO_num_locks ();
+char OPENSSL_init_crypto ();
 int
 main ()
 {
-return CRYPTO_num_locks ();
+return OPENSSL_init_crypto ();
   ;
   return 0;
 }
@@ -23341,21 +23341,21 @@
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext &&
        $as_test_x conftest$ac_exeext; then
-  ac_cv_lib_crypto_CRYPTO_num_locks=yes
+  ac_cv_lib_crypto_OPENSSL_init_crypto=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	ac_cv_lib_crypto_CRYPTO_num_locks=no
+	ac_cv_lib_crypto_OPENSSL_init_crypto=no
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_CRYPTO_num_locks" >&5
-echo "${ECHO_T}$ac_cv_lib_crypto_CRYPTO_num_locks" >&6; }
-if test $ac_cv_lib_crypto_CRYPTO_num_locks = yes; then
+{ echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_OPENSSL_init_crypto" >&5
+echo "${ECHO_T}$ac_cv_lib_crypto_OPENSSL_init_crypto" >&6; }
+if test $ac_cv_lib_crypto_OPENSSL_init_crypto = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBCRYPTO 1
 _ACEOF
@@ -23364,16 +23364,16 @@
 
 fi
 
-     if test "$ac_cv_lib_crypto_CRYPTO_num_locks" != "yes"
+     if test "$ac_cv_lib_crypto_OPENSSL_init_crypto" != "yes"
      then
              { { echo "$as_me:$LINENO: error: ** Unable to find OpenSSL libraries!" >&5
 echo "$as_me: error: ** Unable to find OpenSSL libraries!" >&2;}
    { (exit 1); exit 1; }; }
      fi
 
-{ echo "$as_me:$LINENO: checking for SSL_library_init in -lssl" >&5
-echo $ECHO_N "checking for SSL_library_init in -lssl... $ECHO_C" >&6; }
-if test "${ac_cv_lib_ssl_SSL_library_init+set}" = set; then
+{ echo "$as_me:$LINENO: checking for OPENSSL_init_ssl in -lssl" >&5
+echo $ECHO_N "checking for OPENSSL_init_ssl in -lssl... $ECHO_C" >&6; }
+if test "${ac_cv_lib_ssl_OPENSSL_init_ssl+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -23391,11 +23391,11 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char SSL_library_init ();
+char OPENSSL_init_ssl ();
 int
 main ()
 {
-return SSL_library_init ();
+return OPENSSL_init_ssl ();
   ;
   return 0;
 }
@@ -23418,21 +23418,21 @@
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext &&
        $as_test_x conftest$ac_exeext; then
-  ac_cv_lib_ssl_SSL_library_init=yes
+  ac_cv_lib_ssl_OPENSSL_init_ssl=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-	ac_cv_lib_ssl_SSL_library_init=no
+	ac_cv_lib_ssl_OPENSSL_init_ssl=no
 fi
 
 rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ echo "$as_me:$LINENO: result: $ac_cv_lib_ssl_SSL_library_init" >&5
-echo "${ECHO_T}$ac_cv_lib_ssl_SSL_library_init" >&6; }
-if test $ac_cv_lib_ssl_SSL_library_init = yes; then
+{ echo "$as_me:$LINENO: result: $ac_cv_lib_ssl_OPENSSL_init_ssl" >&5
+echo "${ECHO_T}$ac_cv_lib_ssl_OPENSSL_init_ssl" >&6; }
+if test $ac_cv_lib_ssl_OPENSSL_init_ssl = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSSL 1
 _ACEOF
@@ -23441,7 +23441,7 @@
 
 fi
 
-     if test "$ac_cv_lib_ssl_SSL_library_init" != "yes"
+     if test "$ac_cv_lib_ssl_OPENSSL_init_ssl" != "yes"
      then
              { { echo "$as_me:$LINENO: error: ** Unable to find OpenSSL libraries!" >&5
 echo "$as_me: error: ** Unable to find OpenSSL libraries!" >&2;}
