--- config/BuildSystem/config/packages/sundials2.py.orig	2023-09-29 11:04:42
+++ config/BuildSystem/config/packages/sundials2.py	2023-12-28 22:48:05
@@ -22,7 +22,8 @@
     self.blasLapack = framework.require('config.packages.BlasLapack',self)
     self.mpi        = framework.require('config.packages.MPI',self)
     self.mathlib    = framework.require('config.packages.mathlib',self)
-    self.deps       = [self.mpi,self.blasLapack,self.mathlib]
+    self.suiteSparse = framework.require('config.packages.SuiteSparse',self)
+    self.deps       = [self.mpi,self.blasLapack,self.mathlib,self.suiteSparse]
 
   def formGNUConfigureArgs(self):
     import os
