--- config/BuildSystem/config/packages/sundials2.py.orig	2023-09-29 11:04:42
+++ config/BuildSystem/config/packages/sundials2.py	2023-12-28 22:45:39
@@ -4,7 +4,7 @@
   def __init__(self, framework):
     config.package.GNUPackage.__init__(self, framework)
     self.version          = '2.5.0'
-    self.maxversion       = '2.5.100000'
+    self.maxversion       = '2.9.100000'
     self.versionname      = 'SUNDIALS_PACKAGE_VERSION'
     self.download         = ['https://web.cels.anl.gov/projects/petsc/download/externalpackages/sundials-'+self.version+'p1.tar.gz']
     self.downloaddirnames = ['sundials']
