--- Makefile.PL.orig	2021-11-26 11:52:45.000000000 -0800
+++ Makefile.PL	2021-11-26 12:05:12.000000000 -0800
@@ -124,8 +124,11 @@
 	\$(XSUBPPRUN) \$(XSPROTOARG) \$(XSUBPPARGS) \$(XSUBPP_EXTRA_ARGS) \$*.xs > \$*.xsc
 	\$(MV) \$*.xsc \$*.c
 
-constant-c.inc constant-h.inc constant-xs.inc : Constant.PL
-	\$(PERLRUN) Constant.PL
+# disable developer code to regenerate constant-*.inc based on Constant.PL
+# generated files are included in distro and Constant.PL is not
+# prevents errors during configure and build due to missing Constant.PL
+# constant-c.inc constant-h.inc constant-xs.inc : Constant.PL
+#	\$(PERLRUN) Constant.PL
 EOD
 }
 
