diff -ur enable-4.1.0/kiva/agg/agg-24/include/agg_renderer_outline_aa.h enable-4.1.0_clang_patched/kiva/agg/agg-24/include/agg_renderer_outline_aa.h
--- enable-4.1.0/kiva/agg/agg-24/include/agg_renderer_outline_aa.h	2011-12-21 16:11:25.000000000 -0700
+++ enable-4.1.0_clang_patched/kiva/agg/agg-24/include/agg_renderer_outline_aa.h	2012-04-24 08:52:30.000000000 -0600
@@ -1365,7 +1365,7 @@
         //---------------------------------------------------------------------
         void profile(const line_profile_aa& prof) { m_profile = &prof; }
         const line_profile_aa& profile() const { return *m_profile; }
-        line_profile_aa& profile() { return *m_profile; }
+        const line_profile_aa& profile() { return *m_profile; }
 
         //---------------------------------------------------------------------
         int subpixel_width() const { return m_profile->subpixel_width(); }
diff -ur enable-4.1.0/kiva/agg/agg-24/include/agg_scanline_u.h enable-4.1.0_clang_patched/kiva/agg/agg-24/include/agg_scanline_u.h
--- enable-4.1.0/kiva/agg/agg-24/include/agg_scanline_u.h	2011-12-21 16:11:25.000000000 -0700
+++ enable-4.1.0_clang_patched/kiva/agg/agg-24/include/agg_scanline_u.h	2012-04-24 08:54:20.000000000 -0600
@@ -458,7 +458,7 @@
     class scanline32_u8_am : public scanline32_u8
     {
     public:
-        typedef scanline_u8           base_type;
+        typedef scanline32_u8           base_type;
         typedef AlphaMask             alpha_mask_type;
         typedef base_type::cover_type cover_type;
         typedef base_type::coord_type coord_type;
