--- enable-4.1.0/kiva/setup.py	2011-12-22 00:11:26.000000000 +0100
+++ enable-4.1.0/kiva/setup_patched.py	2012-04-30 08:55:58.000000000 +0200
@@ -13,7 +13,7 @@
 # Description: <Enthought kiva package project>
 #------------------------------------------------------------------------------
 
-import sys
+import platform
 import os
 
 
@@ -34,7 +34,7 @@
     config.add_subpackage('trait_defs.ui')
     config.add_subpackage('trait_defs.ui.*')
 
-    if sys.platform == 'darwin':
+    if platform.system() == 'Darwin' and platform.architecture()[0]=='32bit':
         config.add_subpackage('quartz')
 
     config.get_version()
