From 54a27de078eab56138ff4bcb31c185df9f6e3609 Mon Sep 17 00:00:00 2001
From: Robert Kern <rkern@enthought.com>
Date: Fri, 15 Jun 2012 17:51:19 +0100
Subject: [PATCH] BUG: Avoid a PySide-only API.

---
 enable/qt4/base_window.py |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/enable/qt4/base_window.py b/enable/qt4/base_window.py
index 8e30688..81cb250 100644
--- a/enable/qt4/base_window.py
+++ b/enable/qt4/base_window.py
@@ -138,7 +138,8 @@ def sizeHint(self, qt_size_hint):
         """
 
         preferred_size = self._enable_window.component.get_preferred_size()
-        window_size = self._enable_window.control.size().toTuple()
+        q_size = self._enable_window.control.size()
+        window_size = (q_size.width(), q_size.height())
 
         if qt_size_hint.width() < 0:
             width = max(preferred_size[0], window_size[0])
-- 
1.7.10

