--- makefile.orig	2015-03-04 09:34:17.000000000 -0800
+++ makefile	2015-09-14 13:11:45.000000000 -0700
@@ -22,8 +22,8 @@
 # -Wunreachable-code \
 
 
-CFLAGS = $(CWARNS) $(COPT) -std=c99 -I$(LUADIR) -fPIC
-CC = gcc
+CFLAGS = $(CWARNS) $(COPT) -std=c99 -I$(LUADIR) -fPIC %%CFLAGS%%
+#CC = gcc
 
 FILES = lpvm.o lpcap.o lptree.o lpcode.o lpprint.o
 
@@ -33,7 +33,7 @@
 
 # For Mac OS
 macosx:
-	make lpeg.so "DLLFLAGS = -bundle -undefined dynamic_lookup"
+	make lpeg.so "DLLFLAGS = -bundle -undefined dynamic_lookup %%LDFLAGS%%"
 
 lpeg.so: $(FILES)
 	env $(CC) $(DLLFLAGS) $(FILES) -o lpeg.so
