--- easyrsa	2023-06-10 21:58:12.000000000 +0800
+++ easyrsa	2023-07-17 01:43:06.000000000 +0800
@@ -1309,8 +1309,7 @@
 	# "$EASYRSA" - Old default and Windows
 	# "$PWD" - Usually the same as above, avoid
 	# "${0%/*}" - Usually the same as above, avoid
-	# '/usr/local/share/easy-rsa' - Default user installed
-	# '/usr/share/easy-rsa' - Default system installed
+	# '@PREFIX@/share/easy-rsa' - @PREFIX@
 	# Room for more..
 	# '/etc/easy-rsa' - Last resort
 
@@ -1320,8 +1319,7 @@
 		"$EASYRSA" \
 		"$PWD" \
 		"${0%/*}" \
-		'/usr/local/share/easy-rsa' \
-		'/usr/share/easy-rsa' \
+		'@PREFIX@/share/easy-rsa' \
 		'/etc/easy-rsa' \
 		# EOL
 	do
@@ -5226,8 +5224,8 @@
 			easyrsa_shell="$SHELL (Git)"
 			easyrsa_win_git_bash="${EXEPATH}"
 			# If found then set openssl NOW!
-			#[ -e /usr/bin/openssl ] && \
-			#	set_var EASYRSA_OPENSSL /usr/bin/openssl
+			#[ -e @PREFIX@/bin/openssl ] && \
+			#	set_var EASYRSA_OPENSSL @PREFIX@/bin/openssl
 		fi
 	else
 		easyrsa_host_os=nix
@@ -5402,7 +5400,7 @@
 		fi
 
 		# Program dir vars - This location is least wanted.
-		prog_vars="${prog_dir}/vars"
+		prog_vars="@PREFIX@/share/easy-rsa/vars"
 
 		# set up PKI path vars - Top preference
 		pki_vars="${EASYRSA_PKI:-$PWD/pki}/vars"
@@ -5563,9 +5561,9 @@
 	fi
 
 	# Set defaults, preferring existing env-vars if present
-	set_var EASYRSA					"$PWD"
+	set_var EASYRSA					"@PREFIX@/share/easy-rsa"
 	set_var EASYRSA_OPENSSL			openssl
-	set_var EASYRSA_PKI				"$EASYRSA/pki"
+	set_var EASYRSA_PKI				"$PWD/pki"
 	set_var EASYRSA_DN				cn_only
 	set_var EASYRSA_REQ_COUNTRY		"US"
 	set_var EASYRSA_REQ_PROVINCE	"California"

--- vars.example	2023-06-10 21:58:12.000000000 +0800
+++ vars.example	2023-07-17 01:30:50.000000000 +0800
@@ -47,7 +47,7 @@
 # itself, which is also where the configuration files are located in the
 # easy-rsa tree.
 #
-#set_var EASYRSA	"${0%/*}"
+#set_var EASYRSA	"@PREFIX@/share/easy-rsa"
 
 # If your OpenSSL command is not in the system PATH, you will need to define
 # the path here. Normally this means a full path to the executable, otherwise
