--- src/osx/syncmenu/Makefile.am.orig	2008-08-22 23:29:53.000000000 +0200
+++ src/osx/syncmenu/Makefile.am	2008-08-22 23:30:22.000000000 +0200
@@ -10,8 +10,8 @@
 syncmenulib_LTLIBRARIES = syncmenu.la
 syncmenulibdir = $(libdir)/gajim
 syncmenu_la_SOURCES = sync-menu.c pysyncmenu.c
-syncmenu_la_LDFLAGS = -module -avoid-version -Xcompiler -isysroot -Xcompiler /Developer/SDKs/MacOSX10.4u.sdk -Xcompiler -arch -Xcompiler ppc -Xcompiler -arch -Xcompiler i386 $(GTKLDFLAGS)
-syncmenu_la_CFLAGS = -Xcompiler -isysroot -Xcompiler /Developer/SDKs/MacOSX10.4u.sdk -Xcompiler -arch -Xcompiler ppc -Xcompiler -arch -Xcompiler i386 $(GTKCFLAGS) -Wall -g $(INCLUDES)
+syncmenu_la_LDFLAGS = -module -avoid-version
+syncmenu_la_CFLAGS = $(GTKCFLAGS) -Wall -g $(INCLUDES)
 endif
 
 DISTCLEANFILES = 
