--- CMakeLists.txt.orig	2014-06-14 00:02:49.000000000 +0900
+++ CMakeLists.txt	2014-06-14 00:03:55.000000000 +0900
@@ -22,7 +22,7 @@
 set(MOBILE_BACKEND "kde" CACHE STRING "Backend to install, currently. Check /mobile/plugins/widgets/*")
 
 if(NOT MOBILE_BACKEND STREQUAL "harmattan")
-  add_subdirectory(doc)
+#  add_subdirectory(doc)
   add_subdirectory(src)
   add_subdirectory(plasmoids)
 endif(NOT MOBILE_BACKEND STREQUAL "harmattan")
