--- Makefile.fip.orig	2010-12-09 00:12:12.000000000 -0600
+++ Makefile.fip	2012-08-07 01:35:53.000000000 -0500
@@ -67,6 +67,8 @@
 	install -m 644 -o root -g root $(HEADERFIP) $(INCDIR)
 	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)
 	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)
+	ln -sf $(SHAREDLIB) $(INSTALLDIR)/$(VERLIBNAME)
+	ln -sf $(VERLIBNAME) $(INSTALLDIR)/$(LIBNAME)	
 
 clean:
 	rm -f core Dist/*.* u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
