--- Makefile.osx.orig	2007-08-25 17:00:38.000000000 +0200
+++ Makefile.osx	2008-02-15 15:41:15.000000000 +0100
@@ -14,14 +14,14 @@
 COMPILERFLAGS_I386 = -arch i386
 COMPILERPPFLAGS = -Wno-ctor-dtor-privacy
 INCLUDE += 
-INCLUDE_PPC = -isysroot /Developer/SDKs/MacOSX10.3.9.sdk
+INCLUDE_PPC = -isysroot /Developer/SDKs/MacOSX10.4u.sdk
 INCLUDE_I386 = -isysroot /Developer/SDKs/MacOSX10.4u.sdk
 CFLAGS_PPC = $(COMPILERFLAGS) $(COMPILERFLAGS_PPC) $(INCLUDE) $(INCLUDE_PPC)
 CFLAGS_I386 = $(COMPILERFLAGS) $(COMPILERFLAGS_I386) $(INCLUDE) $(INCLUDE_I386)
 CPPFLAGS_PPC = $(COMPILERPPFLAGS) $(CFLAGS_PPC)
 CPPFLAGS_I386 = $(COMPILERPPFLAGS) $(CFLAGS_I386)
-LIBRARIES_PPC = -Wl,-syslibroot /Developer/SDKs/MacOSX10.3.9.sdk
-LIBRARIES_I386 = -Wl,-syslibroot /Developer/SDKs/MacOSX10.4u.sdk
+LIBRARIES_PPC = -Wl,-syslibroot /Developer/SDKs/MacOSX10.4u.sdk/usr/lib
+LIBRARIES_I386 = -Wl,-syslibroot /Developer/SDKs/MacOSX10.4u.sdk/usr/lib
 LIBTOOL = libtool
 LIPO = lipo
 
