# https://github.com/aristocratos/bpytop/pull/413
--- bpytop.py	2021-12-22 14:22:48
+++ bpytop.py	2024-03-14 02:13:56
@@ -840,7 +840,7 @@
 	mouse: Dict[str, List[List[int]]] = {}
 	mouse_pos: Tuple[int, int] = (0, 0)
 	escape: Dict[Union[str, Tuple[str, str]], str] = {
-		"\n" :					"enter",
+		("\n", "\r") :			"enter",
 		("\x7f", "\x08") :		"backspace",
 		("[A", "OA") :			"up",
 		("[B", "OB") :			"down",
