--- Makefile.orig	2022-05-14 06:05:16.000000000 +0800
+++ Makefile	2023-02-24 18:23:58.000000000 +0800
@@ -1,14 +1,14 @@
 .POSIX:
 .SUFFIXES:
 
-FC      = gfortran
-CC      = gcc
+FC      = @FC@
+CC      = @CC@
 AR      = ar
-PREFIX  = /usr/local
+PREFIX  = @PREFIX@
 DEBUG   = #-ggdb3 -O0
 
-FFLAGS  = $(DEBUG) -Wall -Wno-unused-dummy-argument -std=f2008 -fmax-errors=1 -fcheck=all
-CFLAGS  = $(DEBUG) -Wall
+FFLAGS  = @FFLAGS@ $(DEBUG) -Wall -Wno-unused-dummy-argument -std=f2008 -fmax-errors=1 -fcheck=all
+CFLAGS  = @CFLAGS@ $(DEBUG) -Wall
 LDFLAGS = -I$(PREFIX)/include/ -L$(PREFIX)/lib/
 LDLIBS  = -lcurl
 ARFLAGS = rcs
