--- configure	2013-07-19 01:26:22.000000000 +0200
+++ configure.new	2014-04-25 12:08:31.000000000 +0200
@@ -3582,7 +3582,7 @@
 have_libusb_1_0=no
 if test "$disable_libusb_1_0" = "no"; then
   have_libusb_1_0=yes
-  LIBUSB_1_0_CFLAGS=-I/usr/include/libusb-1.0
+  LIBUSB_1_0_CFLAGS="-I$prefix/include/libusb-1.0"
   LIBUSB_1_0_LIBS=-lusb-1.0
   if test "$have_libusb_1_0" = "yes"; then
     $as_echo "using libusb_1.0";
