diff -Nurd -x'*~' gdm-2.20.3.orig/daemon/gdm-xdmcp-manager.c gdm-2.20.3/daemon/gdm-xdmcp-manager.c
--- daemon/gdm-xdmcp-display-factory.c	2008-01-07 14:39:37.000000000 -0500
+++ daemon/gdm-xdmcp-display-factory.c	2008-01-09 17:33:11.000000000 -0500
@@ -368,7 +368,7 @@
                 if (ai->ai_flags & AI_NUMERICHOST) {
                         g_string_append (str, "numhost ");
                 }
-                if (ai->ai_flags & AI_NUMERICSERV) {
+                if (ai->ai_flags & NI_NUMERICSERV) {
                         g_string_append (str, "numserv ");
                 }
                 if (ai->ai_flags & AI_V4MAPPED) {
diff -Nurd -x'*~' gdm-2.20.3.orig/daemon/slave.c gdm-2.20.3/daemon/slave.c
--- daemon/gdm-session-record.c	2008-01-07 14:39:38.000000000 -0500
+++ daemon/gdm-session-record.c	2008-01-09 17:33:11.000000000 -0500
@@ -322,8 +322,10 @@
 #elif defined(HAVE_UT_UT_TIME)
                 u->ut_time = session_record.ut_time;
 #endif
+#ifndef __APPLE__
                 u->ut_exit.e_termination = 0;
                 u->ut_exit.e_exit = 0;
+#endif
 
                 pututxline (u);
 
