diff -Nurd -x'*~' gdm-2.24.1.orig/daemon/gdm-session-worker.c gdm-2.24.1/daemon/gdm-session-worker.c
--- daemon/gdm-session-worker.c	2008-11-18 17:38:44.000000000 -0500
+++ daemon/gdm-session-worker.c	2009-01-20 22:14:05.000000000 -0500
@@ -32,7 +32,7 @@
 #include <grp.h>
 #include <pwd.h>
 
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 
 #include <glib.h>
 #include <glib/gi18n.h>
diff -Nurd -x'*~' gdm-2.24.1.orig/gui/simple-greeter/gdm-greeter-login-window.c gdm-2.24.1/gui/simple-greeter/gdm-greeter-login-window.c
--- gui/simple-greeter/gdm-greeter-login-window.c	2008-11-18 17:49:12.000000000 -0500
+++ gui/simple-greeter/gdm-greeter-login-window.c	2009-01-20 22:14:20.000000000 -0500
@@ -68,7 +68,7 @@
 #include "gdm-user-chooser-widget.h"
 
 #ifdef HAVE_PAM
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 #define PW_ENTRY_SIZE PAM_MAX_RESP_SIZE
 #else
 #define PW_ENTRY_SIZE GDM_MAX_PASS
