--- configure.orig	2011-04-07 10:31:30.000000000 -0700
+++ configure	2011-04-11 16:30:13.000000000 -0700
@@ -31203,6 +31203,7 @@
 fi
 
 
+ac_pam_includes="/usr/include"
 if test "x$have_pam" = "xyes"; then
    pam_appl_h="$ac_pam_includes/security/pam_appl.h"
    { $as_echo "$as_me:$LINENO: checking for const pam_message" >&5
@@ -35935,7 +35936,12 @@
 # and /usr/X11 since they often symlink to each other, and configure
 # should use the more stable location (the real directory) if possible.
 #
-if test -x /usr/X11/bin/Xserver; then
+if test -x @MP_PREFIX@/bin/Xquartz; then
+   X_PATH="@MP_PREFIX@/bin"
+   X_SERVER_PATH="@MP_PREFIX@/bin"
+   X_SERVER="@MP_PREFIX@/bin/Xquartz"
+   X_CONFIG_OPTIONS="-audit 0"
+elif test -x /usr/X11/bin/Xserver; then
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/Xserver"
