--- daemon/gdm-session-worker.c.orig	2013-12-18 12:40:11.000000000 -0800
+++ daemon/gdm-session-worker.c	2013-12-18 12:46:46.000000000 -0800
@@ -740,6 +740,25 @@
         return utf8;
 }
 
+#ifdef __APPLE__
+#include <Availability.h>
+#if __MAC_OS_X_VERSION_MIN_REQUIRED <= 1060
+static char * strndup (char const *s, size_t n);
+
+static char *
+strndup (char const *s, size_t n)
+{
+  size_t len = strnlen (s, n);
+  char *new = malloc (len + 1);
+
+  if (new == NULL)
+    return NULL;
+
+  new[len] = '\0';
+  return memcpy (new, s, len);
+}
+#endif
+#endif
 static gboolean
 gdm_session_worker_process_pam_message (GdmSessionWorker          *worker,
                                         const struct pam_message  *query,
