--- daemon/gdm-session.c.orig	2013-09-19 12:29:48.000000000 -0700
+++ daemon/gdm-session.c	2013-11-08 21:16:43.000000000 -0800
@@ -979,6 +979,9 @@
 #ifdef __linux__
         native_credentials = g_credentials_get_native (credentials, G_CREDENTIALS_TYPE_LINUX_UCRED);
         pid = (GPid) ((struct ucred *) native_credentials)->pid;
+#elif defined (__APPLE__)
+        native_credentials = g_credentials_get_native (credentials, G_CREDENTIALS_TYPE_INVALID);
+        pid = (GPid) (0);
 #elif defined (__FreeBSD__)
         native_credentials = g_credentials_get_native (credentials, G_CREDENTIALS_TYPE_OPENBSD_SOCKPEERCRED);
         pid = (GPid) ((struct cmsgcred *) native_credentials)->cmcred_pid;
