--- configure.orig	2013-10-16 07:46:13.000000000 -0700
+++ configure	2013-11-08 21:31:12.000000000 -0800
@@ -23256,6 +23256,7 @@
 fi
 
 
+ac_pam_includes="/usr/include"
 if test "x$have_pam" = "xyes"; then
    pam_appl_h="$ac_pam_includes/security/pam_appl.h"
    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for const pam_message" >&5
@@ -25473,7 +25474,17 @@
 # and /usr/X11 since they often symlink to each other, and configure
 # should use the more stable location (the real directory) if possible.
 #
-if test -x /usr/X11/bin/Xserver; then
+if test -x @MP_PREFIX@/bin/Xquartz; then
+   X_PATH="/opt/local/bin"
+   X_SERVER_PATH="/opt/local/bin"
+   X_SERVER="/opt/local/bin/Xquartz"
+   X_CONFIG_OPTIONS="-audit 0"
+elif test -x /opt/X11/bin/Xquartz; then
+   X_PATH="/opt/X11/bin"
+   X_SERVER_PATH="/opt/X11/bin"
+   X_SERVER="/opt/X11/bin/Xquartz"
+   X_CONFIG_OPTIONS="-audit 0"
+elif test -x /usr/X11/bin/Xserver; then
    X_PATH="/usr/X11/bin"
    X_SERVER_PATH="/usr/X11/bin"
    X_SERVER="/usr/X11/bin/Xserver"
