--- libtlog/src_threads/Makefile.in.orig	2016-07-05 19:19:33.000000000 +0900
+++ libtlog/src_threads/Makefile.in	2016-07-18 09:43:14.000000000 +0900
@@ -27,7 +27,7 @@
 	rm -f $(LIBTLOG) *.o
 
 install: $(LIBTLOG) $(HEADERS)
-	$(MKDIR_P) $(LIBDIR) $(INCDIR)
-	$(INSTALL) $(LIBTLOG) $(LIBDIR)
-	$(INSTALL) $(HEADERS) $(INCDIR)
+	$(MKDIR_P) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCDIR)
+	$(INSTALL) $(LIBTLOG) $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(HEADERS) $(DESTDIR)$(INCDIR)
 
