--- F-FrontEnd/src/Makefile.in.orig	2016-07-05 19:19:33.000000000 +0900
+++ F-FrontEnd/src/Makefile.in	2016-07-18 08:47:39.000000000 +0900
@@ -46,8 +46,8 @@
 	$(AWK) -f C-exprcode.c.awk < C-exprcode.def > C-exprcode.c
 
 install: $(F_FRONT)
-	$(MKDIR_P) $(BINDIR)
-	$(INSTALL) $(F_FRONT) $(BINDIR)
+	$(MKDIR_P) $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(F_FRONT) $(DESTDIR)$(BINDIR)
 
 clean:
 	rm -f *.o $(CREATE_FILES) $(F_FRONT)
