--- Driver/src/Makefile.in.orig	2016-07-05 19:19:33.000000000 +0900
+++ Driver/src/Makefile.in	2016-07-18 08:47:39.000000000 +0900
@@ -20,10 +20,10 @@
 	$(CC) $^ -o $@
 
 install: $(bin_SCRIPTS) $(etc_DATA) $(libexec_SCRIPTS)
-	$(MKDIR_P) $(BINDIR) $(ETCDIR) $(LIBEXECDIR)
-	$(INSTALL) $(bin_SCRIPTS) $(BINDIR)
-	$(INSTALL) $(etc_DATA) $(ETCDIR)
-	$(INSTALL) $(libexec_SCRIPTS) $(LIBEXECDIR)
+	$(MKDIR_P) $(DESTDIR)$(BINDIR) $(DESTDIR)$(ETCDIR) $(DESTDIR)$(LIBEXECDIR)
+	$(INSTALL) $(bin_SCRIPTS) $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(etc_DATA) $(DESTDIR)$(ETCDIR)
+	$(INSTALL) $(libexec_SCRIPTS) $(DESTDIR)$(LIBEXECDIR)
 
 clean:
 	rm -f om_replace_pragma *.o
