--- Makefile.orig	2016-12-16 20:38:38.000000000 +0000
+++ Makefile	2017-04-09 02:49:21.000000000 +0000
@@ -68,7 +68,7 @@
 CC=cc
 CXX=c++
 CXXFLAGS=$(CFLAGS)
-CLD=c++ $(CFLAGS) -L/usr/local/lib $(OPTLOCALLFLAGS) $(OPENSSLLFLAGS)
+CLD=$(CXX) $(CFLAGS) -L@@PREFIX@@/lib
 
 ifeq ($(OS),osx)
   CLDOSFLAGS=-bind_at_load
@@ -108,9 +108,7 @@
 #   -DHUSH
 CFLAGS+= $(COSFLAGS) -ffast-math \
 	-funsigned-char \
-	-I/usr/local/include \
-	$(OPTLOCALIFLAGS) \
-	$(OPENSSLIFLAGS) \
+	-I@@PREFIX@@/include \
 	$(CURLINC) \
 	-I$(INCLUDE) \
 	-Ioutside/$(LIBUV_VER)/include \
