From 06bcfebf058c083153a1a206ad84e600a4e7d0b1 Mon Sep 17 00:00:00 2001
From: MarcusCalhoun-Lopez <MarcusCalhoun-Lopez@users.noreply.github.com>
Date: Sun, 26 Apr 2020 19:00:20 -0700
Subject: [PATCH 1/3] Use standard name for C++ compiler flags

---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 57f0f03..ecf8e46 100644
--- Makefile
+++ Makefile
@@ -26,9 +26,9 @@
 #CC=g++
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
-   CCFLAGS += -lboost_regex -lboost_program_options -std=c++0x
+   CXXFLAGS += -lboost_regex -lboost_program_options -std=c++0x
 else ifeq ($(UNAME_S),Darwin)
-   CCFLAGS += -L/opt/local/lib -lboost_regex-mt -lboost_program_options-mt -std=c++0x -I/opt/local/include -Wno-deprecated-register
+   CXXFLAGS += -L/opt/local/lib -lboost_regex-mt -lboost_program_options-mt -std=c++0x -I/opt/local/include -Wno-deprecated-register
 endif
 EXEC=logcat-colorize
 DEPS=logcat-colorize.cpp
@@ -40,7 +40,7 @@ endif
 INSTALLDIR=$(DESTDIR)/$(PREFIX)/bin
 
 $(EXEC): $(DEPS)
-	$(CXX) $(DEPS) -o $(EXEC) $(CCFLAGS)
+	$(CXX) $(DEPS) -o $(EXEC) $(CXXFLAGS)
 
 $(INSTALLDIR):
 	mkdir -pv $(INSTALLDIR)

From 79dbba6d068715b19ae7b3e587650cfc12c7b82d Mon Sep 17 00:00:00 2001
From: MarcusCalhoun-Lopez <MarcusCalhoun-Lopez@users.noreply.github.com>
Date: Sun, 26 Apr 2020 19:03:15 -0700
Subject: [PATCH 2/3] Conform to standard use of DESTDIR

---
 Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index ecf8e46..97fc3b9 100644
--- Makefile
+++ Makefile
@@ -33,11 +33,9 @@ endif
 EXEC=logcat-colorize
 DEPS=logcat-colorize.cpp
 
-ifndef PREFIX
-	PREFIX=usr
-endif
+PREFIX ?= /usr
 
-INSTALLDIR=$(DESTDIR)/$(PREFIX)/bin
+INSTALLDIR=$(DESTDIR)$(PREFIX)/bin
 
 $(EXEC): $(DEPS)
 	$(CXX) $(DEPS) -o $(EXEC) $(CXXFLAGS)

From 4bf7d828115dec6b14e88dd3de92c420559e3aef Mon Sep 17 00:00:00 2001
From: MarcusCalhoun-Lopez <MarcusCalhoun-Lopez@users.noreply.github.com>
Date: Sun, 26 Apr 2020 19:07:08 -0700
Subject: [PATCH 3/3] Support nonstandard  MacPorts location

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 97fc3b9..77e00ff 100644
--- Makefile
+++ Makefile
@@ -28,7 +28,8 @@ UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Linux)
    CXXFLAGS += -lboost_regex -lboost_program_options -std=c++0x
 else ifeq ($(UNAME_S),Darwin)
-   CXXFLAGS += -L/opt/local/lib -lboost_regex-mt -lboost_program_options-mt -std=c++0x -I/opt/local/include -Wno-deprecated-register
+   BOOSTDIR ?= /opt/local
+   CXXFLAGS += -L$(BOOSTDIR)/lib -lboost_regex-mt -lboost_program_options-mt -std=c++0x -I$(BOOSTDIR)/include -Wno-deprecated-register
 endif
 EXEC=logcat-colorize
 DEPS=logcat-colorize.cpp
