--- doc/create-doxygen-lua-api.sh.orig	2022-03-14 20:12:17.000000000 -0400
+++ doc/create-doxygen-lua-api.sh	2022-03-14 20:16:54.000000000 -0400
@@ -33,13 +33,13 @@
 ##
 
 export PROJECT="Geeqie"
-export VERSION=$(git tag --list v[1-9]* | tail -1)
+export VERSION="@@VERSION@@"
 export SRCDIR="$PWD/.."
 export DOCDIR="$PWD/html/lua-api"
 export INLINE_SOURCES=NO
 export STRIP_CODE_COMMENTS=YES
 
-TMPFILE=$(mktemp) || exit 1
+TMPFILE=$(mktemp "$TMPDIR/geeqie.XXXXXXXX") || exit 1
 
 # Modify the Geeqie doxygen.conf file to produce
 # only the data needed for the lua API document
