From 87042fa51da7c14a7600bbf8420105dd91675757 Mon Sep 17 00:00:00 2001
From: Dmitry Marakasov <amdmi3@amdmi3.ru>
Date: Wed, 31 Aug 2022 19:14:36 +0300
Subject: [PATCH] Detect strverscmp support properly

---
 config.h.in | 2 +-
 meson.build | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/config.h.in b/config.h.in
index 67e68216..ab5e2825 100644
--- config.h.in
+++ config.h.in
@@ -161,7 +161,7 @@
 #define HAVE_STRING_H 1
 
 /* Define to 1 if you have the `strverscmp' function. */
-#define HAVE_STRVERSCMP 1
+#mesondefine HAVE_STRVERSCMP
 
 /* Define to 1 if you have the <sys/stat.h> header file. */
 #define HAVE_SYS_STAT_H 1
diff --git a/meson.build b/meson.build
index e44cf8c3..38aaa627 100644
--- meson.build
+++ meson.build
@@ -446,6 +446,11 @@ else
     summary({'nl_langinfo' : ['nl_langinfo not found - first weekday depends on locale:', false, 'first weekday defaults to Monday']}, section : 'Documentation', bool_yn : true)
 endif
 
+result = cc.has_function('strverscmp', prefix : '#include <string.h>')
+if result
+    conf_data.set('HAVE_STRVERSCMP', 1)
+endif
+
 conf_data.set_quoted('GETTEXT_PACKAGE', meson.project_name())
 conf_data.set_quoted('GQ_APPDIR', gq_appdir)
 conf_data.set_quoted('GQ_BINDIR', gq_bindir)
