diff --git clutter/osx/clutter-device-manager-osx.c clutter/osx/clutter-device-manager-osx.c
index 8480da6..91193f9 100644
--- clutter/osx/clutter-device-manager-osx.c
+++ clutter/osx/clutter-device-manager-osx.c
@@ -34,6 +34,8 @@
 #include "clutter-debug.h"
 #include "clutter-device-manager-private.h"
 #include "clutter-private.h"
+#include "clutter-stage-private.h"
+
 
 enum
 {
diff --git clutter/osx/clutter-event-loop-osx.c clutter/osx/clutter-event-loop-osx.c
index 77a4330..54e693b 100644
--- clutter/osx/clutter-event-loop-osx.c
+++ clutter/osx/clutter-event-loop-osx.c
@@ -35,6 +35,8 @@
 
 #include "clutter-debug.h"
 #include "clutter-private.h"
+#include "clutter-stage-private.h"
+
 
 /* 
  * This file implementations integration between the GLib main loop and
diff --git clutter/osx/clutter-event-osx.c clutter/osx/clutter-event-osx.c
index 4b53fef..4aa5f71 100644
--- clutter/osx/clutter-event-osx.c
+++ clutter/osx/clutter-event-osx.c
@@ -34,6 +34,7 @@
 
 #include "clutter-debug.h"
 #include "clutter-device-manager.h"
+#include "clutter-device-manager-private.h"
 #include "clutter-event-private.h"
 #include "clutter-keysyms.h"
 #include "clutter-private.h"
