--- clutter/clutter-backend.c.orig	2015-07-10 18:13:50.000000000 -0700
+++ clutter/clutter-backend.c	2015-07-10 18:17:25.000000000 -0700
@@ -512,6 +512,11 @@
 	backend = g_intern_string (backend_env);
     }
 
+#ifdef CLUTTER_WINDOWING_X11
+  if (backend == NULL || backend == I_(CLUTTER_WINDOWING_X11))
+    retval = g_object_new (CLUTTER_TYPE_BACKEND_X11, NULL);
+  else
+#endif
 #ifdef CLUTTER_WINDOWING_OSX
   if (backend == NULL || backend == I_(CLUTTER_WINDOWING_OSX))
     retval = g_object_new (CLUTTER_TYPE_BACKEND_OSX, NULL);
@@ -527,11 +533,6 @@
     retval = g_object_new (CLUTTER_TYPE_BACKEND_GDK, NULL);
   else
 #endif
-#ifdef CLUTTER_WINDOWING_X11
-  if (backend == NULL || backend == I_(CLUTTER_WINDOWING_X11))
-    retval = g_object_new (CLUTTER_TYPE_BACKEND_X11, NULL);
-  else
-#endif
 #ifdef CLUTTER_WINDOWING_WAYLAND
   if (backend == NULL || backend == I_(CLUTTER_WINDOWING_WAYLAND))
     retval = g_object_new (CLUTTER_TYPE_BACKEND_WAYLAND, NULL);
