https://gitlab.freedesktop.org/spice/spice/-/issues/84
--- server/inputs-channel.h.orig
+++ server/inputs-channel.h
@@ -33,7 +33,7 @@ class InputsChannelClient;
 class InputsChannel final: public RedChannel
 {
     friend class InputsChannelClient;
-    friend int spice_server_kbd_leds(SpiceKbdInstance *sin, int leds);
+    friend SPICE_GNUC_VISIBLE int spice_server_kbd_leds(SpiceKbdInstance *sin, int leds);
 
 public:
     InputsChannel(RedsState *reds);
