--- configure.orig	2011-08-23 23:28:30.000000000 +0900
+++ configure	2011-10-31 22:26:40.000000000 +0900
@@ -14292,7 +14292,9 @@
 $as_echo "#define FFI_EXEC_TRAMPOLINE_TABLE 1" >>confdefs.h
 
      ;;
-     *-apple-darwin10* | *-*-freebsd* | *-*-openbsd* | *-pc-solaris*)
+     #*-apple-darwin10* | *-*-freebsd* | *-*-openbsd* | *-pc-solaris*)
+     # enable FFI_MMAP_EXEC_WRIT on 10.7 Lion as same as 10.6 Snow Leopard
+     *-apple-darwin1[01]* | *-*-freebsd* | *-*-openbsd* | *-pc-solaris*)
 
 $as_echo "#define FFI_MMAP_EXEC_WRIT 1" >>confdefs.h
 
