From ac0fc79c76fc92783d2a5267082a1f8f9c28df22 Mon Sep 17 00:00:00 2001
From: Gustav <sir.gustav.the.coder@gmail.com>
Date: Mon, 18 May 2020 14:06:39 +0200
Subject: [PATCH] Fixes issue #1718

---
 include/rapidjson/writer.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/rapidjson/writer.h b/include/rapidjson/writer.h
index e7fb873a4..51dd86d58 100644
--- include/rapidjson/writer.h
+++ include/rapidjson/writer.h
@@ -283,6 +283,8 @@ class Writer {
         os_->Flush();
     }
 
+    static const size_t kDefaultLevelDepth = 32;
+
 protected:
     //! Information for each nested level
     struct Level {
@@ -291,8 +293,6 @@ class Writer {
         bool inArray;       //!< true if in array, otherwise in object
     };
 
-    static const size_t kDefaultLevelDepth = 32;
-
     bool WriteNull()  {
         PutReserve(*os_, 4);
         PutUnsafe(*os_, 'n'); PutUnsafe(*os_, 'u'); PutUnsafe(*os_, 'l'); PutUnsafe(*os_, 'l'); return true;
