--- configure.orig	2009-06-17 17:14:14.000000000 -0700
+++ configure	2009-06-17 17:15:10.000000000 -0700
@@ -20725,7 +20725,7 @@
         python_path=`dirname $PYTHON | sed "s,/bin.*$,,"`
         for i in "$python_path/lib/python$PYTHON_VERSION/config/" "$python_path/lib/python$PYTHON_VERSION/" "$python_path/lib/python/config/" "$python_path/lib/python/" "$python_path/" ; do
 		if test -e "$i"; then
-	                python_path=`find $i -type f -name libpython$PYTHON_VERSION.* -print | sed "1q"`
+	                python_path=`find $i -type l -name libpython$PYTHON_VERSION.* -print | sed "1q"`
         	        if test -n "$python_path" ; then
 				python_lib="python$PYTHON_VERSION"
                 	        break
