--- configure.ac.orig	2004-10-16 17:26:12.000000000 +0200
+++ configure.ac	2011-05-09 13:08:24.000000000 +0200
@@ -49,7 +49,8 @@
 
 AM_ICONV
 
-AC_CHECK_FUNCS(iconv_open,,AC_MSG_ERROR([
+LIBS="$LIBS -liconv"
+AC_CHECK_FUNCS(libiconv_open,,AC_MSG_ERROR([
 iconv_open not found try to install replacement from
 http://www.gnu.org/software/libiconv/
 ftp://ftp.gnu.org/pub/gnu/libiconv/libiconv-1.9.1.tar.gz]))
