diff -ur vala.orig/libdmapsharing-3.0-daap/libdmapsharing-3.0-uninstalled.files.in vala/libdmapsharing-3.0-daap/libdmapsharing-3.0-uninstalled.files.in
--- vala.orig/libdmapsharing-3.0-daap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:24.000000000 -0700
+++ vala/libdmapsharing-3.0-daap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:34.000000000 -0700
@@ -1,2 +1,2 @@
 @srcdir@/../libdmapsharing/
-@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.so
+@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.dylib
diff -ur vala.orig/libdmapsharing-3.0-dacp/libdmapsharing-3.0-uninstalled.files.in vala/libdmapsharing-3.0-dacp/libdmapsharing-3.0-uninstalled.files.in
--- vala.orig/libdmapsharing-3.0-dacp/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:24.000000000 -0700
+++ vala/libdmapsharing-3.0-dacp/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:22:13.000000000 -0700
@@ -1,2 +1,2 @@
 @srcdir@/../libdmapsharing/
-@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.so
+@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.dylib
diff -ur vala.orig/libdmapsharing-3.0-dmap/libdmapsharing-3.0-uninstalled.files.in vala/libdmapsharing-3.0-dmap/libdmapsharing-3.0-uninstalled.files.in
--- vala.orig/libdmapsharing-3.0-dmap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:24.000000000 -0700
+++ vala/libdmapsharing-3.0-dmap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:34.000000000 -0700
@@ -1,2 +1,2 @@
 @srcdir@/../libdmapsharing/
-@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.so
+@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.dylib
diff -ur vala.orig/libdmapsharing-3.0-dpap/libdmapsharing-3.0-uninstalled.files.in vala/libdmapsharing-3.0-dpap/libdmapsharing-3.0-uninstalled.files.in
--- vala.orig/libdmapsharing-3.0-dpap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:24.000000000 -0700
+++ vala/libdmapsharing-3.0-dpap/libdmapsharing-3.0-uninstalled.files.in	2014-08-12 09:21:34.000000000 -0700
@@ -1,2 +1,2 @@
 @srcdir@/../libdmapsharing/
-@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.so
+@builddir@/../libdmapsharing/.libs/libdmapsharing-3.0.dylib
