--- configure.ac.orig	2014-05-22 01:50:24.000000000 -0700
+++ configure.ac	2014-05-22 01:52:18.000000000 -0700
@@ -339,12 +339,15 @@
 AC_SUBST(GSTREAMERAPP_LIBS)
 
 AM_PROG_VALAC([0.11.4])
+VALAC=no
 AM_CONDITIONAL(HAVE_VALAC, test -x "$VALAC")
 
 AC_PATH_PROG(VALA_GEN_INTROSPECT, vala-gen-introspect)
 AC_PATH_PROG(VAPIGEN, vapigen)
-AM_CONDITIONAL(BUILD_VAPI, test -n "$VALA_GEN_INTROSPECT" -a -n "$VAPIGEN")
-if test -n "$BUILD_VAPI" ; then
+dnl AM_CONDITIONAL(BUILD_VAPI, test -n "$VALA_GEN_INTROSPECT" -a -n "$VAPIGEN")
+BUILD_VAPI=no
+AM_CONDITIONAL(BUILD_VAPI, test x"$BUILD_VAPI" = "xyes")
+if test x"$BUILD_VAPI" = "xno" ; then
 	AC_WARN([will not build vapi])
 fi
 
