--- src/defs.h.orig	2005-05-11 06:01:36.000000000 -0500
+++ src/defs.h	2010-03-26 17:47:23.000000000 -0500
@@ -225,10 +225,10 @@
 
 	bool GetPowerups() { return powerups; }
 	void SetPowerups(bool x) { powerups=x; }
-	rectangle defines::UpperBound();
-	rectangle defines::LowerBound();
-	rectangle defines::RightBound();
-	rectangle defines::LeftBound();
+	rectangle UpperBound();
+	rectangle LowerBound();
+	rectangle RightBound();
+	rectangle LeftBound();
 
 private:
 	int maxballs;
