--- src/Makefile.orig	2005-05-11 06:01:36.000000000 -0500
+++ src/Makefile	2010-03-26 17:44:01.000000000 -0500
@@ -4,11 +4,11 @@
 CXX      = g++
 LEX      = flex
 YACC     = yacc
-CFLAGS   = -pipe -fno-exceptions -Wall -W -O2 -march=i486 -mcpu=i686 -D_REENTRANT
-CXXFLAGS = -pipe -fno-exceptions -Wall -W -O2 -march=i486 -mcpu=i686 -D_REENTRANT
+CFLAGS   = -pipe -fno-exceptions -Wall -W -O2 -D_REENTRANT
+CXXFLAGS = -pipe -fno-exceptions -Wall -W -O2 -D_REENTRANT
 LEXFLAGS = 
 YACCFLAGS= -d
-INCPATH  = `xml2-config --cflags` `sdl-config --cflags`
+INCPATH  = `xml2-config --cflags` `sdl-config --cflags` $(CPPFLAGS)
 LINK     = g++
 LFLAGS   = -fno-exceptions  -Wl,-rpath,/usr/lib/qt/lib
 LIBS     = `xml2-config --libs` `sdl-config --libs` -lSDL_net -lSDL_mixer
