--- bindings/python/notmuch/globals.py.orig	2015-08-21 11:13:03.000000000 -0400
+++ bindings/python/notmuch/globals.py	2015-08-21 12:13:27.000000000 -0400
@@ -25,7 +25,7 @@
 try:
     from os import uname
     if uname()[0] == 'Darwin':
-        nmlib = CDLL("libnotmuch.{0:s}.dylib".format(SOVERSION))
+        nmlib = CDLL("@PREFIX@/lib/"+"libnotmuch.{0:s}.dylib".format(SOVERSION))
     else:
         nmlib = CDLL("libnotmuch.so.{0:s}".format(SOVERSION))
 except:
