--- os/darwin/ccconfig.h.orig	2014-06-03 23:43:50.000000000 -0700
+++ os/darwin/ccconfig.h	2018-11-04 01:41:27.000000000 -0700
@@ -33,14 +33,14 @@
 #define	CPPADD		{ "-D__Darwin__", "-D__APPLE__", "-D__MACH__", "-D__APPLE_CPP__", NULL }
 #define	CRT0		"crt1.o"
 #define GCRT0		"gcrt1.o"
-#define CRTBEGIN_T	0
-#define CRTEND_T	0
-#define CRTBEGIN	0
-#define CRTEND		0
+#define CRTBEGIN_T	""
+#define CRTEND_T	""
+#define CRTBEGIN	""
+#define CRTEND		""
 #define CRTBEGIN_S	"dylib1.o"
-#define CRTEND_S	0
-#define CRTI		0
-#define CRTN		0
+#define CRTEND_S	""
+#define CRTI		""
+#define CRTN		""
 #define DEFLIBS		{ "-lSystem", "-lpcc", NULL }
 #define DEFPROFLIBS	{ "-lSystem_profile", "-lpcc", NULL }
 #define STARTLABEL "start"
