--- dirac.pc.in.orig	2014-01-01 00:26:12.000000000 -0800
+++ dirac.pc.in	2014-01-01 00:26:20.000000000 -0800
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: @GENERIC_LIBRARY_NAME@
 Description: The BBC Dirac Video Codec
 Version: @VERSION@
-Libs: -L${libdir} -l@GENERIC_LIBRARY_NAME@_encoder -l@GENERIC_LIBRARY_NAME@_decoder @CONFIG_MATH_LIB@ -lstdc++
+Libs: -L${libdir} -l@GENERIC_LIBRARY_NAME@_encoder -l@GENERIC_LIBRARY_NAME@_decoder @CONFIG_MATH_LIB@
 Cflags: -I${includedir}/@GENERIC_LIBRARY_NAME@
 
