--- doc/Makefile.in.orig	2009-02-10 17:43:37.000000000 -0800
+++ doc/Makefile.in	2009-02-18 08:12:47.000000000 -0800
@@ -390,7 +390,7 @@
 	doxygen $(srcdir)/dirac_api.doxygen
 
 install-data-hook:
-	cp -dR html/* $(DESTDIR)$(htmldir)
+	cp -R html/* $(DESTDIR)$(htmldir)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
