--- bindings/python/gtkosx_application/Makefile.in.old	2013-01-25 08:40:08.000000000 +0100
+++ bindings/python/gtkosx_application/Makefile.in	2013-01-25 08:41:39.000000000 +0100
@@ -252,7 +252,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 INCLUDES = -I$(top_srcdir)/src $(PYTHON_INCLUDES)
-gtkosx_applicationdir = $(libdir)/python$(PYTHON_VERSION)/site-packages/gtkosx_application
+gtkosx_applicationdir = @pythondir@/gtkosx_application
 gtkosx_application_PYTHON = __init__.py
 gtkosx_application_LTLIBRARIES = _gtkosx_application.la
 headers = $(top_srcdir)/src/gtkosxapplication.h
