--- src/GtkApplicationDelegate.c.orig	2024-12-20 11:03:23.000000000 +0100
+++ src/GtkApplicationDelegate.c	2024-12-20 11:03:44.000000000 +0100
@@ -73,7 +73,7 @@
   return result;
 }
 
--(void) application: (NSApplication*)theApplication openFiless: (NSArray<NSString *> *) files
+-(void) application: (NSApplication*)theApplication openFiless: (NSArray *) files
 {
   gboolean overall_result = TRUE;
   guint sig = g_signal_lookup ("NSApplicationOpenFile",
