--- bindings/python/gtkmacintegration/Makefile.in.orig	2015-06-22 19:51:48.000000000 +0200
+++ bindings/python/gtkmacintegration/Makefile.in	2015-06-22 19:48:03.000000000 +0200
@@ -399,7 +399,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I$(top_srcdir)/src $(PYTHON_INCLUDES)
-gtkmacintegrationdir = $(libdir)/python$(PYTHON_VERSION)/site-packages/gtkmacintegration
+gtkmacintegrationdir = @pythondir@/gtkmacintegration
 gtkmacintegration_PYTHON = __init__.py
 gtkmacintegration_LTLIBRARIES = $(am__append_1) $(am__append_2)
 headers = \
