--- configure.orig	2013-10-26 06:41:18.000000000 +0900
+++ configure	2013-10-26 16:44:10.000000000 +0900
@@ -15774,7 +15774,7 @@
 
 # Check for binaries...
 # Extract the first word of "rst2man.py", so it can be a program name with args.
-set dummy rst2man.py; ac_word=$2
+set dummy rst2man-2.7.py; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_path_RST2MAN+:} false; then :
@@ -15824,7 +15824,7 @@
 
 
 # Extract the first word of "rst2html.py", so it can be a program name with args.
-set dummy rst2html.py; ac_word=$2
+set dummy rst2html-2.7.py; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_path_RST2HTML+:} false; then :
