Patch submitted upstreams: https://github.com/qgis/QGIS/pull/54940

--- cmake/FindPDAL.cmake.orig
+++ cmake/FindPDAL.cmake
@@ -18,66 +18,68 @@
 # locations. When an earlier FIND_* succeeds, subsequent FIND_*s
 # searching for the same item do nothing.
 
-FIND_PATH(PDAL_INCLUDE_DIR pdal/pdal.hpp
-  "$ENV{LIB_DIR}/include"
-  "/usr/include"
-  c:/msys/local/include
-  NO_DEFAULT_PATH
-  )
-FIND_PATH(PDAL_INCLUDE_DIR pdal/pdal.hpp)
+find_program(PDAL_BIN pdal $ENV{LIB_DIR}/bin /usr/local/bin/ /usr/bin/
+             NO_DEFAULT_PATH)
+find_program(PDAL_BIN pdal)
 
-FIND_LIBRARY(PDAL_CPP_LIBRARY NAMES pdalcpp libpdalcpp PATHS
-  "$ENV{LIB_DIR}/lib"
-  c:/msys/local/lib
-  NO_DEFAULT_PATH
-  )
-FIND_LIBRARY(PDAL_CPP_LIBRARY NAMES pdalcpp libpdalcpp)
+if(PDAL_BIN)
+  # extract PDAL version
+  exec_program(
+    ${PDAL_BIN} ARGS
+    --version
+    OUTPUT_VARIABLE PDAL_VERSION_OUT)
+  string(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\1" PDAL_VERSION_MAJOR "${PDAL_VERSION_OUT}")
+  string(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\2" PDAL_VERSION_MINOR "${PDAL_VERSION_OUT}")
+  string(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\3" PDAL_VERSION_MICRO "${PDAL_VERSION_OUT}")
+  string(CONCAT PDAL_VERSION ${PDAL_VERSION_MAJOR} "." ${PDAL_VERSION_MINOR} "." ${PDAL_VERSION_MICRO})
+endif()
 
-FIND_LIBRARY(PDAL_UTIL_LIBRARY NAMES pdal_util libpdal_util PATHS
-  "$ENV{LIB_DIR}/lib"
-  c:/msys/local/lib
-  NO_DEFAULT_PATH
-  )
-FIND_LIBRARY(PDAL_UTIL_LIBRARY NAMES pdal_util libpdal_util)
+find_path(PDAL_INCLUDE_DIR pdal/pdal.hpp "$ENV{LIB_DIR}/include" "/usr/include"
+          c:/msys/local/include NO_DEFAULT_PATH)
+find_path(PDAL_INCLUDE_DIR pdal/pdal.hpp)
 
-FIND_PROGRAM(PDAL_BIN pdal
-    $ENV{LIB_DIR}/bin
-    /usr/local/bin/
-    /usr/bin/
-    NO_DEFAULT_PATH
-    )
-FIND_PROGRAM(PDAL_BIN pdal)
+find_library(
+  PDAL_CPP_LIBRARY
+  NAMES pdalcpp libpdalcpp
+  PATHS "$ENV{LIB_DIR}/lib" c:/msys/local/lib
+  NO_DEFAULT_PATH)
+find_library(PDAL_CPP_LIBRARY NAMES pdalcpp libpdalcpp)
 
-IF (PDAL_INCLUDE_DIR AND PDAL_CPP_LIBRARY AND PDAL_UTIL_LIBRARY AND PDAL_BIN)
-   SET(PDAL_FOUND TRUE)
-   SET(PDAL_LIBRARIES ${PDAL_CPP_LIBRARY} ${PDAL_UTIL_LIBRARY})
-ENDIF (PDAL_INCLUDE_DIR AND PDAL_CPP_LIBRARY AND PDAL_UTIL_LIBRARY AND PDAL_BIN)
+# with PDAL 2.6 the pdal_util library is removed and hence part of pdalcpp
+if(PDAL_VERSION VERSION_LESS "2.6.0" AND PDAL_BIN)
+  find_library(
+    PDAL_UTIL_LIBRARY
+    NAMES pdal_util libpdal_util
+    PATHS "$ENV{LIB_DIR}/lib" c:/msys/local/lib
+    NO_DEFAULT_PATH)
+  find_library(PDAL_UTIL_LIBRARY NAMES pdal_util libpdal_util)
+endif()
 
-IF (PDAL_FOUND)
-    # extract PDAL version
-    EXEC_PROGRAM(${PDAL_BIN}
-	ARGS --version
-	OUTPUT_VARIABLE PDAL_VERSION_OUT )
-    STRING(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\1" PDAL_VERSION_MAJOR "${PDAL_VERSION_OUT}")
-    STRING(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\2" PDAL_VERSION_MINOR "${PDAL_VERSION_OUT}")
-    STRING(REGEX REPLACE "^.*([0-9]+)\\.([0-9]+)\\.([0-9]+).*$" "\\3" PDAL_VERSION_MICRO "${PDAL_VERSION_OUT}")
-    STRING(CONCAT PDAL_VERSION ${PDAL_VERSION_MAJOR} "." ${PDAL_VERSION_MINOR} "." ${PDAL_VERSION_MICRO})
+if(PDAL_INCLUDE_DIR AND PDAL_CPP_LIBRARY AND PDAL_UTIL_LIBRARY AND PDAL_BIN
+   AND PDAL_VERSION VERSION_LESS "2.6.0")
+  set(PDAL_FOUND TRUE)
+  set(PDAL_LIBRARIES ${PDAL_CPP_LIBRARY} ${PDAL_UTIL_LIBRARY})
+elseif(PDAL_INCLUDE_DIR AND PDAL_CPP_LIBRARY AND PDAL_BIN)
+  set(PDAL_FOUND TRUE)
+  set(PDAL_LIBRARIES ${PDAL_CPP_LIBRARY})
+endif()
 
-   IF (NOT PDAL_FIND_QUIETLY)
-      MESSAGE(STATUS "Found PDAL: ${PDAL_LIBRARIES} (${PDAL_VERSION})")
-   ENDIF (NOT PDAL_FIND_QUIETLY)
+if(PDAL_FOUND)
+  if(NOT PDAL_FIND_QUIETLY)
+    message(STATUS "Found PDAL: ${PDAL_LIBRARIES} (${PDAL_VERSION})")
+  endif()
 
-   IF ((PDAL_VERSION_MAJOR EQUAL 1) AND (PDAL_VERSION_MINOR LESS 7))
-      MESSAGE (FATAL_ERROR "PDAL version is too old (${PDAL_VERSION}). Use 1.7 or higher.")
-   ENDIF()
-
-ELSE (PDAL_FOUND)
-   IF (PDAL_FIND_REQUIRED)
-     MESSAGE(FATAL_ERROR "Could not find PDAL")
-   ELSE (PDAL_FIND_REQUIRED)
-     IF (NOT PDAL_FIND_QUIETLY)
-        MESSAGE(STATUS "Could not find PDAL")
-     ENDIF (NOT PDAL_FIND_QUIETLY)
-   ENDIF (PDAL_FIND_REQUIRED)
-
-ENDIF (PDAL_FOUND)
+  if((PDAL_VERSION_MAJOR EQUAL 1) AND (PDAL_VERSION_MINOR LESS 7))
+    message(
+      FATAL_ERROR
+        "PDAL version is too old (${PDAL_VERSION}). Use 1.7 or higher.")
+  endif()
+else()
+  if(PDAL_FIND_REQUIRED)
+    message(FATAL_ERROR "Could not find PDAL")
+  else()
+    if(NOT PDAL_FIND_QUIETLY)
+      message(STATUS "Could not find PDAL")
+    endif()
+  endif()
+endif()
