--- Sources/TSDocument.m.orig	2017-09-21 02:08:30.000000000 -0500
+++ Sources/TSDocument.m	2017-10-20 20:32:15.000000000 -0500
@@ -1790,7 +1790,7 @@
     
         if (dispatch_source != NULL) {
                 dispatch_source_cancel(dispatch_source);
-                dispatch_release(dispatch_source);
+               // dispatch_release(dispatch_source);
                 dispatch_source = NULL;
             }
     }
