From 66490f3ef3d4f808976f43f26198e3e2bc1b6d9f Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Wed, 28 Sep 2022 21:01:43 +0200
Subject: [PATCH] Disable C++11 warnings

---
 hotspot/make/bsd/makefiles/gcc.make | 2 ++
 1 file changed, 2 insertions(+)

diff --git hotspot/make/bsd/makefiles/gcc.make hotspot/make/bsd/makefiles/gcc.make
index 65b31149756..1c6dcb19062 100644
--- hotspot/make/bsd/makefiles/gcc.make
+++ hotspot/make/bsd/makefiles/gcc.make
@@ -258,6 +258,8 @@ ifeq ($(USE_CLANG), true)
 #  WARNINGS_ARE_ERRORS += -Wno-tautological-constant-out-of-range-compare
   WARNINGS_ARE_ERRORS += -Wno-delete-non-virtual-dtor -Wno-deprecated -Wno-format -Wno-dynamic-class-memaccess
   WARNINGS_ARE_ERRORS += -Wno-empty-body
+  WARNINGS_ARE_ERRORS += -Wno-reserved-user-defined-literal
+  WARNINGS_ARE_ERRORS += -Wno-c++11-narrowing
 endif
 
 WARNING_FLAGS = -Wpointer-arith -Wsign-compare -Wundef -Wunused-function -Wformat=2
-- 
2.37.3

