From 8f1e4d58649ff043b83dae8c37feae0c536dae9c Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Wed, 6 Jan 2016 22:33:12 -0800
Subject: [PATCH] darwin: Fixup vendorShellWidgetClass

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 src/Vendor.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/Vendor.c b/src/Vendor.c
index 8899cd1..a18b9f3 100644
--- a/src/Vendor.c
+++ b/src/Vendor.c
@@ -110,12 +110,21 @@ static void XawVendorShellClassPartInit(WidgetClass);
 void XawVendorShellExtResize(Widget);
 #endif
 
-#if defined(__UNIXOS2__) || defined(__CYGWIN__) || defined(__MINGW32__)
+#if defined(__UNIXOS2__) || defined(__CYGWIN__) || defined(__MINGW32__) || defined(__APPLE__)
 /* to fix the EditRes problem because of wrong linker semantics */
 extern WidgetClass vendorShellWidgetClass; /* from Xt/Vendor.c */
 extern VendorShellClassRec _XawVendorShellClassRec;
 void _XawFixupVendorShell(void);
 
+#if defined(__APPLE__)
+__attribute__((constructor))
+static void __VendorShellHack(void)
+{
+    vendorShellWidgetClass = (WidgetClass)(&_XawVendorShellClassRec);
+    _XawFixupVendorShell();
+}
+#endif
+
 #if defined(__UNIXOS2__)
 unsigned long _DLL_InitTerm(unsigned long mod,unsigned long flag)
 {
@@ -475,7 +484,7 @@ XawVendorShellClassPartInit(WidgetClass class)
 }
 #endif
 
-#if defined(__osf__) || defined(__UNIXOS2__) || defined(__CYGWIN__) || defined(__MINGW32__)
+#if defined(__osf__) || defined(__UNIXOS2__) || defined(__CYGWIN__) || defined(__MINGW32__) || defined(__APPLE__)
 /* stupid OSF/1 shared libraries have the wrong semantics */
 /* symbols do not get resolved external to the shared library */
 void
-- 
2.6.4

