From b093ed77fbab09bd474094898719ed6346d30d45 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Sun, 22 Jan 2012 22:01:07 -0800
Subject: [PATCH 3/3] Build fix for int-pointer cast errors

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 src/XawIm.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/XawIm.c b/src/XawIm.c
index b01f0d0..82a94b1 100644
--- a/src/XawIm.c
+++ b/src/XawIm.c
@@ -141,14 +141,14 @@ SearchVendorShell(Widget w)
     return(NULL);
 }
 
-static XContext extContext = (XContext)NULL;
+static XContext extContext = (XContext)0;
 
 static XawVendorShellExtPart *
 SetExtPart(VendorShellWidget w, XawVendorShellExtWidget vew)
 {
     contextDataRec *contextData;
 
-    if (extContext == (XContext)NULL) extContext = XUniqueContext();
+    if (extContext == (XContext)0) extContext = XUniqueContext();
 
     contextData = XtNew(contextDataRec);
     contextData->parent = (Widget)w;
@@ -252,14 +252,14 @@ ConfigureCB(Widget w, XtPointer closure, XEvent *event)
     }
 }
 
-static XContext errContext = (XContext)NULL;
+static XContext errContext = (XContext)0;
 
 static Widget
 SetErrCnxt(Widget w, XIM xim)
 {
     contextErrDataRec *contextErrData;
 
-    if (errContext == (XContext)NULL) errContext = XUniqueContext();
+    if (errContext == (XContext)0) errContext = XUniqueContext();
 
     contextErrData = XtNew(contextErrDataRec);
     contextErrData->widget = w;
@@ -1321,12 +1321,12 @@ Destroy(Widget w, XawVendorShellExtPart *ve)
 	return;
     XtFree( (char*) ve->im.resources );
 
-    if (extContext != (XContext)NULL &&
+    if (extContext != (XContext)0 &&
 	!XFindContext (XtDisplay (w), (Window)w,
 		       extContext, (XPointer*)&contextData))
         XtFree( (char*) contextData );
 
-    if (errContext != (XContext)NULL &&
+    if (errContext != (XContext)0 &&
 	!XFindContext (XDisplayOfIM( ve->im.xim ), (Window) ve->im.xim,
 		       errContext, (XPointer*) &contextErrData))
         XtFree( (char*) contextErrData );
-- 
1.7.7.4

