From 93306e5b20dd85aeb98d42d06dffefc3c35fe680 Mon Sep 17 00:00:00 2001
From: Markus Steinborn <gnugv_maintainer@yahoo.de>
Date: Thu, 1 Mar 2012 18:08:00 +0100
Subject: [PATCH 6/9] Xaw3d-1.6.1-i18n.patch

---
 src/MultiSrc.c |    1 +
 1 file changed, 1 insertion(+)

diff --git libXaw3d-1.6.1/src/MultiSrc.c libXaw3d-1.6.1/src/MultiSrc.c
index b53fa50..1ac3283 100644
--- libXaw3d-1.6.1/src/MultiSrc.c
+++ libXaw3d-1.6.1/src/MultiSrc.c
@@ -728,6 +728,7 @@ SetValues(Widget current, Widget request, Widget new, ArgList args, Cardinal *nu
 
   if ( string_set || (old_src->multi_src.type != src->multi_src.type) ) {
     RemoveOldStringOrFile(old_src, string_set);
+    src->multi_src.allocated_string = old_src->multi_src.allocated_string;
     file = InitStringOrFile(src, string_set);
 
     /* Load pieces does this logic for us, but it shouldn't.  Its messy.*/
-- 
1.7.9.4

