--- aqbanking-config.cmake.in	2016-10-21 01:52:57.000000000 +0200
+++ aqbanking-config.cmake.in.new	2016-10-21 01:55:25.000000000 +0200
@@ -22,7 +22,11 @@
 set_and_check(prefix "@prefix@")
 set_and_check(exec_prefix "@exec_prefix@")
 set_and_check(AQBANKING_INCLUDE_DIRS "@includedir@/aqbanking@AQBANKING_VERSION_MAJOR@")
-set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so")
+if(APPLE)
+  set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.dylib")
+else()
+  set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so")
+endif(APPLE)
 set(AQBANKING_FOUND true)
 
 # The target is available since aqbanking 5.6.5
