--- aqbanking-config.cmake.in	2014-08-03 23:06:06.000000000 +0200
+++ aqbanking-config.cmake.in.new	2015-06-26 23:22:32.000000000 +0200
@@ -21,7 +21,11 @@
 set_and_check(prefix "@prefix@")
 set_and_check(exec_prefix "@exec_prefix@")
 set_and_check(AQBANKING_INCLUDE_DIRS "@includedir@/aqbanking@AQBANKING_VERSION_MAJOR@")
-set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so")
+if(APPLE)
+  set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.dylib")
+else()
+  set_and_check(AQBANKING_LIBRARIES "@libdir@/libaqbanking.so")
+endif(APPLE)
 set(AQBANKING_LIBRARIES ${AQBANKING_LIBRARIES} ${GWENHYWFAR_LIBRARIES})
 
 mark_as_advanced( AQBANKING_INCLUDE_DIRS AQBANKING_LIBRARIES )
