--- Makefile.old	2013-09-18 08:32:05.000000000 +0200
+++ Makefile	2013-09-18 08:33:14.000000000 +0200
@@ -1365,7 +1365,8 @@
 	for f in `find . -name \*.c -o -name \*.cpp`; do cppcheck $$f; done
 
 ChangeLog:
-	git log -n 100 --pretty="format:------------------------------------------------------------------%n%ai %an%n%s%n%b" >ChangeLog
+#	git log -n 100 --pretty="format:------------------------------------------------------------------%n%ai %an%n%s%n%b" >ChangeLog
+	echo "Skipped ChangeLog creation on MacPorts." >ChangeLog
 
 dist-hook:
 	rm -f ChangeLog 2>/dev/null
