--- Makefile.am.old	2013-09-18 09:06:35.000000000 +0200
+++ Makefile.am	2013-09-18 09:07:04.000000000 +0200
@@ -245,7 +245,7 @@
 	for f in `find . -name \*.c -o -name \*.cpp`; do cppcheck $$f; done
 
 ChangeLog:
-	git log -n 100 --pretty="format:------------------------------------------------------------------%n%ai %an%n%s%n%b" >ChangeLog
+	echo "Skipping ChangeLog creation on MacPorts" >ChangeLog
 
 dist-hook:
 	rm -f ChangeLog 2>/dev/null
