--- lib/Makefile.in.orig	2018-10-25 18:30:09.000000000 +1100
+++ lib/Makefile.in	2020-01-28 13:42:34.000000000 +1100
@@ -166,7 +166,7 @@
 target_vendor = @target_vendor@
 noinst_LIBRARIES = libgvpe.a
 
-libgvpe_a_SOURCES = pidfile.c getopt1.c dropin.c
+libgvpe_a_SOURCES = pidfile.c dropin.c
 
 libgvpe_a_LIBADD = @LIBOBJS@ @ALLOCA@
 libgvpe_a_DEPENDENCIES = $(libgvpe_a_LIBADD)
@@ -180,7 +180,7 @@
 LIBRARIES = $(noinst_LIBRARIES)
 
 libgvpe_a_AR = $(AR) cru
-am_libgvpe_a_OBJECTS = pidfile.$(OBJEXT) getopt1.$(OBJEXT) \
+am_libgvpe_a_OBJECTS = pidfile.$(OBJEXT) \
 	dropin.$(OBJEXT)
 libgvpe_a_OBJECTS = $(am_libgvpe_a_OBJECTS)
 
@@ -188,7 +188,7 @@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 @AMDEP_TRUE@DEP_FILES = $(DEPDIR)/alloca.Po ./$(DEPDIR)/dropin.Po \
-@AMDEP_TRUE@	./$(DEPDIR)/getopt1.Po ./$(DEPDIR)/pidfile.Po
+@AMDEP_TRUE@	./$(DEPDIR)/pidfile.Po
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
@@ -227,7 +227,6 @@
 
 @AMDEP_TRUE@@am__include@ @am__quote@$(DEPDIR)/alloca.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/dropin.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getopt1.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pidfile.Po@am__quote@
 
 .c.o:
