--- ../include/wx/mac/menu.h	Sat Feb  7 13:22:19 2004
+++ ../include/wx/mac/menu.h.new	Sat Feb  7 13:20:36 2004
@@ -44,6 +44,7 @@
     virtual bool DoAppend(wxMenuItem *item);
     virtual bool DoInsert(size_t pos, wxMenuItem *item);
     virtual wxMenuItem *DoRemove(wxMenuItem *item);
+    virtual void Attach(wxMenuBarBase *menubar) ;
 
     virtual void Break();
 
@@ -94,8 +95,14 @@
     // common part of Append/Insert (behaves as Append is pos == (size_t)-1)
     bool DoInsertOrAppend(wxMenuItem *item, size_t pos = (size_t)-1);
 
+    // terminate the current radio group, if any
+    void EndRadioGroup();
+
     // if TRUE, insert a breal before appending the next item
     bool m_doBreak;
+
+    // the position of the first item in the current radio group or -1
+    int m_startRadioGroup;
 
     // the menu handle of this menu
     WXHMENU m_hMenu;
