--- cmake/FindFFmpeg.cmake.orig	2025-05-01 21:05:30
+++ cmake/FindFFmpeg.cmake	2025-05-01 21:06:31
@@ -96,12 +96,12 @@
     endif()
   endif(NOT WIN32)
 
-  find_path(${_component}_INCLUDE_DIRS ${_header} HINTS ${PC_LIB${_component}_INCLUDEDIR}
-                                                        ${PC_LIB${_component}_INCLUDE_DIRS} PATH_SUFFIXES ffmpeg
+  find_path(${_component}_INCLUDE_DIRS ${_header} HINTS ${PC_${_component}_INCLUDEDIR}
+                                                        ${PC_${_component}_INCLUDE_DIRS} PATH_SUFFIXES ffmpeg
   )
 
   find_library(
-    ${_component}_LIBRARIES NAMES ${_library} HINTS ${PC_LIB${_component}_LIBDIR} ${PC_LIB${_component}_LIBRARY_DIRS}
+    ${_component}_LIBRARIES NAMES ${_library} HINTS ${PC_${_component}_LIBDIR} ${PC_${_component}_LIBRARY_DIRS}
   )
 
   set(${_component}_DEFINITIONS ${PC_${_component}_CFLAGS_OTHER} CACHE STRING "The ${_component} CFLAGS.")
