--- client/Mac/MRDPView.m.orig	2014-03-02 23:33:18.000000000 -0800
+++ client/Mac/MRDPView.m	2014-03-02 23:37:14.000000000 -0800
@@ -68,6 +68,9 @@
  Forward declarations
  ******************************************/
 
+int mac_receive_channel_data(freerdp* instance, int chan_id, BYTE* data, int size, int flags, int total_size);
+void windows_to_apple_cords(MRDPView* view, NSRect* r);
+int register_channels_fds(freerdp* instance);
 
 void mf_Pointer_New(rdpContext* context, rdpPointer* pointer);
 void mf_Pointer_Free(rdpContext* context, rdpPointer* pointer);
