--- client/Mac/MRDPView.m.orig	2013-08-06 13:59:24.000000000 +0200
+++ client/Mac/MRDPView.m	2013-08-06 14:08:26.000000000 +0200
@@ -101,6 +101,9 @@
 void cliprdr_send_supported_format_list(freerdp* instance);
 int register_channel_fds(int* fds, int count, freerdp* instance);
 
+void windows_to_apple_cords(MRDPView* view, NSRect* r);
+int register_channels_fds(freerdp* instance);
+
 struct cursor
 {
 	rdpPointer* pointer;
