diff --git Makefile.am Makefile.am
index 5db5804..4b3ecd6 100755
--- Makefile.am
+++ Makefile.am
@@ -45,7 +45,6 @@ par2_SOURCES = par2cmdline.cpp par2cmdline.h \
 	verificationhashtable.cpp verificationhashtable.h \
 	verificationpacket.cpp verificationpacket.h
 
-LDADD = -lstdc++
 AM_CXXFLAGS = -Wall
 
 EXTRA_DIST = PORTING ROADMAP par2cmdline.sln par2cmdline.vcproj \
diff --git Makefile.in Makefile.in
index 1cace84..bbc82c8 100644
--- Makefile.in
+++ Makefile.in
@@ -227,7 +227,6 @@ par2_SOURCES = par2cmdline.cpp par2cmdline.h \
 	verificationhashtable.cpp verificationhashtable.h \
 	verificationpacket.cpp verificationpacket.h
 
-LDADD = -lstdc++
 AM_CXXFLAGS = -Wall
 EXTRA_DIST = PORTING ROADMAP par2cmdline.sln par2cmdline.vcproj \
 	testdata.tar.gz pretest test1 test2 test3 test4 test5 test6 \
