--- commandline.cpp~	Tue Jun 24 09:11:02 2003
+++ commandline.cpp	Tue Nov  4 19:28:09 2003
@@ -27,6 +27,8 @@
 #endif
 #endif
 
+#define stricmp(s1,s2) strcasecmp(s1,s2)
+
 CommandLine::ExtraFile::ExtraFile(void)
 : filename()
 , filesize(0)
--- par2repairer.cpp~	Mon May 26 14:01:26 2003
+++ par2repairer.cpp	Tue Nov  4 19:38:53 2003
@@ -27,6 +27,8 @@
 #endif
 #endif
 
+#define stricmp(s1,s2) strcasecmp(s1,s2)
+
 Par2Repairer::Par2Repairer(void)
 {
   firstpacket = true;
