--- src/util.h.orig	2014-10-27 11:12:00.000000000 -0400
+++ src/util.h	2014-10-27 11:12:22.000000000 -0400
@@ -219,6 +219,7 @@
 
 void XC(gga_c_pw91_func)(const XC(func_type) *p, XC(gga_work_c_t) *r);
 void XC(gga_c_pbe_func) (const XC(func_type) *p, XC(gga_work_c_t) *r);
+void XC(gga_c_q2d_func) (const XC(func_type) *p, XC(gga_work_c_t) *r);
 
 /* meta GGAs */
 typedef struct XC(mgga_work_x_t) {
