--- DESCRIPTION	2023-12-11 06:33:58.000000000 +0800
+++ DESCRIPTION	2023-12-12 01:11:33.000000000 +0800
@@ -36,7 +36,6 @@
     utils,
     zip (>= 2.3.0)
 Suggests:
-    asciicast (>= 2.2.0.9000),
     codetools,
     covr,
     debugme,
@@ -56,10 +55,8 @@
     webfakes (>= 1.1.5.9000),
     withr (>= 2.1.1)
 Config/Needs/coverage:
-    r-lib/asciicast,
     covr
 Config/Needs/website:
-    r-lib/asciicast,
     pkgdown (>= 2.0.2),
     tidyverse/tidytemplate
 Config/testthat/edition: 3
