--- configure.orig	2009-11-04 16:56:19.000000000 +0900
+++ configure	2009-11-04 16:59:59.000000000 +0900
@@ -7053,10 +7053,7 @@
 fi
 if test $do_udunits = true; then
 	echo "****************************************************************************"
-	echo "Udunits support enabled.   Note: Udunits did not used to work with MAC OS X"
-	echo "If you are using Mac OS X and have trouble, make sure you are using the"
-	echo "latest udunits package.  If it still doesn't work, remove the udunits library and remake ncview"
-	echo "udunits dirs: include: $UDUNITS_INCDIR  library: $UDUNITS_LIBDIR  libname: $UDUNITS_LIBNAME"
+	echo "Udunits support enabled."
 	echo "****************************************************************************"
 	DO_UDUNITS_LINE="INC_UDUNITS   = -DINC_UDUNITS"
 	DO_UDUNITS_INCDIR="UDUNITSINCDIR = -I$UDUNITS_INCDIR"
@@ -7678,7 +7675,7 @@
 
 
 fi
-PPM_LIBNAME=libppm.so
+PPM_LIBNAME=libppm.dylib
 
 # Check whether --with-ppm_libdir or --without-ppm_libdir was given.
 if test "${with_ppm_libdir+set}" = set; then
@@ -7883,7 +7880,7 @@
 
 fi
 if test x$PPM_LIBDIR = x; then
-PPM_LIBNAME=libnetpbm.so
+PPM_LIBNAME=libnetpbm.dylib
 if test x$PPM_LIBDIR = x; then
 	echo "$as_me:$LINENO: checking for ppm_writeppm in -lnetpbm" >&5
 echo $ECHO_N "checking for ppm_writeppm in -lnetpbm... $ECHO_C" >&6
@@ -8051,7 +8048,7 @@
 
 fi
 fi
-PPM_LIBNAME=`echo $PPM_LIBNAME | sed s/lib// | sed s/\.so//`
+PPM_LIBNAME=`echo $PPM_LIBNAME | sed s/lib// | sed s/\.dylib//`
 
 
 
