--- src/cf.data.pre.orig	2009-12-03 17:49:37.000000000 -0500
+++ src/cf.data.pre	2009-12-03 17:50:05.000000000 -0500
@@ -855,12 +855,12 @@
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
 # one who can access services on "localhost" is a local user
-#http_access deny to_localhost
+http_access deny to_localhost
 
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 #
-
+http_access allow localhost
 # Example rule allowing access from your local networks.
 # Adapt localnet in the ACL section to list your (internal) IP networks
 # from where browsing should be allowed
