--- setup.py.orig	2013-01-14 21:25:04.000000000 +0100
+++ setup.py	2013-01-14 21:25:39.000000000 +0100
@@ -45,7 +45,8 @@
 				  'scikits/bvp_solver/lib/BVP_LA.f',
 				  'scikits/bvp_solver/lib/BVP_M.f90',
                                   'scikits/bvp_solver/lib/BVP_INTERFACE.f90',
-				  'scikits/bvp_solver/lib/bvp_interface.pyf'])
+				  'scikits/bvp_solver/lib/bvp_interface.pyf'],
+                  extra_link_args=['XXX'])
 
     config.add_data_files('scikits/bvp_solver/examples/*.*')
     config.add_data_files('scikits/bvp_solver/lib/BVP_SOLVER_License.txt')
