--- generic/tkTable.h.orig	2020-11-13 05:04:00.000000000 -0600
+++ generic/tkTable.h	2020-11-13 20:50:06.000000000 -0600
@@ -22,11 +22,11 @@
 # include <inttypes.h>
 #endif
 #include <tk.h>
-#ifdef MAC_TCL
-# include <Xatom.h>
+#ifdef MAC_OSX_TK
+# include <X11_tk/Xatom.h>
 #else
 # include <X11/Xatom.h>
-#endif /* MAC_TCL */
+#endif /* MAC_OSX_TK */
 
 #if (TCL_MAJOR_VERSION == 8) && (TCL_MINOR_VERSION == 0) /* Tcl8.0 stuff */
 #define Tcl_GetString(objPtr)	Tcl_GetStringFromObj(objPtr, (int *)NULL)
