--- Makefile.in.orig	2015-03-17 01:40:48.000000000 -0500
+++ Makefile.in	2015-03-18 23:52:33.000000000 -0500
@@ -825,32 +825,19 @@
 run: all
 	$(MAKE) -C src run
 
-MP4Joiner.app: Info.plist src/mp4joiner
+MP4Joiner.app: Info.plist all
 	-mkdir MP4Joiner.app
 	-mkdir MP4Joiner.app/Contents
 	-mkdir MP4Joiner.app/Contents/MacOS
 	-mkdir MP4Joiner.app/Contents/Resources
 	-mkdir MP4Joiner.app/Contents/Resources/English.lproj
 	cp Info.plist MP4Joiner.app/Contents/
-	echo -n "APPL????" > MP4Joiner.app/Contents/PkgInfo
+	printf "APPL????" > MP4Joiner.app/Contents/PkgInfo
 	cp src/mp4joiner MP4Joiner.app/Contents/MacOS/MP4Joiner
 	cp resources/mp4joiner.icns MP4Joiner.app/Contents/Resources/MP4Joiner.icns
 	-mkdir MP4Joiner.app/Contents/locale
 	(cd locale && cp -r de es pt ru ../MP4Joiner.app/Contents/locale/)
-	cp /opt/local/bin/ffmpeg MP4Joiner.app/Contents/MacOS/
 	strip MP4Joiner.app/Contents/MacOS/MP4Joiner
-	strip MP4Joiner.app/Contents/MacOS/ffmpeg
-	cp /Applications/Osmo4.app/Contents/MacOS/MP4Box MP4Joiner.app/Contents/MacOS/
-	-mkdir MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libcrypto.1.0.0.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libgpac.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libjpeg.8.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libjs.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libnspr4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libplc4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libplds4.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libpng15.15.dylib MP4Joiner.app/Contents/MacOS/lib/
-	cp /Applications/Osmo4.app/Contents/MacOS/lib/libssl.1.0.0.dylib MP4Joiner.app/Contents/MacOS/lib/
 
 libs:
 	dylibbundler -od -b -x ./MP4Joiner.app/Contents/MacOS/MP4Joiner -d ./MP4Joiner.app/Contents/libs/
