--- index/mod_dom.c	2022-12-16 03:50:29
+++ index/mod_dom.c	2024-06-06 13:30:30
@@ -20,6 +20,8 @@
 #if HAVE_CONFIG_H
 #include <config.h>
 #endif
+
+#include <stdlib.h>
 #include <stdio.h>
 #include <assert.h>
 #include <ctype.h>

--- index/mod_alvis.c	2022-12-16 03:50:29
+++ index/mod_alvis.c	2024-06-06 13:42:09
@@ -20,6 +20,8 @@
 #if HAVE_CONFIG_H
 #include <config.h>
 #endif
+
+#include <stdlib.h>
 #include <stdio.h>
 #include <assert.h>
 #include <ctype.h>
