diff --git a/mozilla/EventContext.cpp b/mozilla/EventContext.cpp
index a2a0556..7b22394 100644
--- a/mozilla/EventContext.cpp
+++ b/mozilla/EventContext.cpp
@@ -492,12 +492,12 @@ nsresult EventContext::GetCSSBackground (nsIDOMNode *node, nsAString& url)
 	nsCOMPtr<nsIDOMCSSStyleDeclaration> decl;
 	rv = viewCSS->GetComputedStyle (element, NS_LITERAL_STRING(""),
 					    getter_AddRefs (decl));
-	if (NS_FAILED (rv)) return rv;
+	if (NS_FAILED (rv) || !decl) return rv;
 
 	nsCOMPtr<nsIDOMCSSValue> CSSValue;
-	decl->GetPropertyCSSValue (NS_LITERAL_STRING ("background-image"),
+	rv = decl->GetPropertyCSSValue (NS_LITERAL_STRING ("background-image"),
 				   getter_AddRefs (CSSValue));
-	if (!CSSValue) return NS_ERROR_FAILURE;
+	if (NS_FAILED (rv) || !CSSValue) return rv;
 
 	nsCOMPtr<nsIDOMCSSPrimitiveValue> primitiveValue = 
 		do_QueryInterface (CSSValue);
