diff -ur galeon-2.0.7/libegg/dock/egg-dock-box.c galeon-2.0.7.dockcrash/libegg/dock/egg-dock-box.c
--- galeon-2.0.7/libegg/dock/egg-dock-box.c	2008-07-04 19:30:51.000000000 +0300
+++ galeon-2.0.7.dockcrash/libegg/dock/egg-dock-box.c	2009-09-24 13:20:02.000000000 +0300
@@ -425,9 +425,6 @@
 
     box = GTK_BOX (EGG_DOCK_ITEM (object)->child);
 
-    if (other_data) g_print("has somesort of value\n");
-    if (G_VALUE_HOLDS(other_data, G_TYPE_FLAGS)) g_print("is a flag\n");
-
     /* see if we can dock the item in our box */
     switch (EGG_DOCK_ITEM (object)->orientation) {
         case GTK_ORIENTATION_HORIZONTAL:
