diff --git a/mozilla/EphyXULAppInfo.cpp b/mozilla/EphyXULAppInfo.cpp
index 1319431..299cbbf 100644
--- a/mozilla/EphyXULAppInfo.cpp
+++ b/mozilla/EphyXULAppInfo.cpp
@@ -128,3 +128,12 @@ EphyXULAppInfo::GetXPCOMABI(nsACString & aXPCOMABI)
   aXPCOMABI.Assign (EPHY_HOST_CPU "-gcc3");
   return NS_OK;
 }
+
+/* readonly attribute AUTF8String WidgetToolkit; */
+NS_IMETHODIMP
+EphyXULAppInfo::GetWidgetToolkit(nsACString & aWidgetToolkit)
+{
+  aWidgetToolkit.Assign ("gtk");
+  return NS_OK;
+}
+
diff --git a/mozilla/ExternalProtocolService.cpp b/mozilla/ExternalProtocolService.cpp
index 4554998..85ba677 100644
--- a/mozilla/ExternalProtocolService.cpp
+++ b/mozilla/ExternalProtocolService.cpp
@@ -185,3 +185,12 @@ NS_IMETHODIMP GExternalProtocolService::GetProtocolHandlerInfo(const nsACString
 	return NS_ERROR_NOT_IMPLEMENTED;
 }
 #endif
+
+NS_IMETHODIMP GExternalProtocolService::GetProtocolHandlerInfoFromOS(const nsACString & aProtocolScheme, PRBool*, nsIHandlerInfo**)
+{
+	return NS_ERROR_NOT_IMPLEMENTED;
+}
+NS_IMETHODIMP GExternalProtocolService::SetProtocolHandlerDefaults(nsIHandlerInfo*, int)
+{
+	return NS_ERROR_NOT_IMPLEMENTED;
+}
