--- mozilla/ExternalProtocolService.cpp.orig
+++ mozilla/ExternalProtocolService.cpp
@@ -185,3 +185,12 @@ NS_IMETHODIMP GExternalProtocolService::GetProtocolHandlerInfo(const nsACString
 	return NS_ERROR_NOT_IMPLEMENTED;
 }
 #endif
+
+NS_IMETHODIMP GExternalProtocolService::GetProtocolHandlerInfoFromOS(const nsACString & aProtocolScheme, PRBool*, nsIHandlerInfo**)
+{
+	return NS_ERROR_NOT_IMPLEMENTED;
+}
+NS_IMETHODIMP GExternalProtocolService::SetProtocolHandlerDefaults(nsIHandlerInfo*, int)
+{
+	return NS_ERROR_NOT_IMPLEMENTED;
+}
