--- ext/gd/config.m4.orig	2015-07-21 14:00:36.000000000 -0500
+++ ext/gd/config.m4	2015-07-26 13:56:55.000000000 -0500
@@ -72,7 +72,7 @@
 AC_DEFUN([PHP_GD_WEBP],[
   if test "$PHP_WEBP_DIR" != "no"; then
 
-    for i in $PHP_WEBP_DIR /usr/local /usr; do
+    for i in $PHP_WEBP_DIR; do
       test -f $i/include/webp/decode.h && GD_WEBP_DIR=$i && break
     done
 
@@ -80,7 +80,7 @@
       AC_MSG_ERROR([webp/decode.h not found.])
     fi
 
-    for i in $PHP_WEBP_DIR /usr/local /usr; do
+    for i in $PHP_WEBP_DIR; do
       test -f $i/include/webp/encode.h && GD_WEBP_DIR=$i && break
     done
 
@@ -106,7 +106,7 @@
 AC_DEFUN([PHP_GD_JPEG],[
   if test "$PHP_JPEG_DIR" != "no"; then
 
-    for i in $PHP_JPEG_DIR /usr/local /usr; do
+    for i in $PHP_JPEG_DIR; do
       test -f $i/include/jpeglib.h && GD_JPEG_DIR=$i && break
     done
 
@@ -131,7 +131,7 @@
 AC_DEFUN([PHP_GD_PNG],[
   if test "$PHP_PNG_DIR" != "no"; then
 
-    for i in $PHP_PNG_DIR /usr/local /usr; do
+    for i in $PHP_PNG_DIR; do
       test -f $i/include/png.h && GD_PNG_DIR=$i && break
     done
 
@@ -162,7 +162,7 @@
 AC_DEFUN([PHP_GD_XPM],[
   if test "$PHP_XPM_DIR" != "no"; then
 
-    for i in $PHP_XPM_DIR /usr/local /usr/X11R6 /usr; do
+    for i in $PHP_XPM_DIR; do
       test -f $i/include/xpm.h && GD_XPM_DIR=$i && GD_XPM_INC=$i && break
       test -f $i/include/X11/xpm.h && GD_XPM_DIR=$i && GD_XPM_INC=$i/X11 && break
     done
@@ -189,7 +189,7 @@
 AC_DEFUN([PHP_GD_FREETYPE2],[
   if test "$PHP_FREETYPE_DIR" != "no"; then
 
-    for i in $PHP_FREETYPE_DIR /usr/local /usr; do
+    for i in $PHP_FREETYPE_DIR; do
       if test -f "$i/bin/freetype-config"; then
         FREETYPE2_DIR=$i
         FREETYPE2_CONFIG="$i/bin/freetype-config"
