--- ext/odbc/config.m4.orig	2024-09-24 10:59:52.000000000 -0500
+++ ext/odbc/config.m4	2024-09-30 03:04:19.000000000 -0500
@@ -113,6 +113,7 @@
     be defined.])],
   [0x0350])
 
+:<<'MACPORTS_DISABLED'
 AS_VAR_IF([ODBC_TYPE],, [
 PHP_ARG_WITH([adabas],
   [for Adabas support],
@@ -339,6 +340,7 @@
   ])
 ])
 
+MACPORTS_DISABLED
 AS_VAR_IF([ODBC_TYPE],, [
 PHP_ARG_WITH([unixODBC],
   [whether to build with unixODBC support],
