--- ext/openssl/config.m4.orig	2019-01-09 17:54:13.000000000 +0800
+++ ext/openssl/config.m4	2019-05-11 16:52:27.000000000 +0800
@@ -32,4 +32,10 @@
   if test "$PHP_SYSTEM_CIPHERS" != "no"; then
     AC_DEFINE(USE_OPENSSL_SYSTEM_CIPHERS,1,[ Use system default cipher list instead of hardcoded value ])
   fi
+
+  INCLUDES=`echo "$INCLUDES" | sed "s|-I@PREFIX@/include ||g"`
+  INCLUDES="$INCLUDES -I@PREFIX@/include"
+
+  LDFLAGS=`echo "$LDFLAGS" | sed "s|-L@PREFIX@/lib ||g"`
+  OPENSSL_SHARED_LIBADD="$OPENSSL_SHARED_LIBADD -L@PREFIX@/lib"
 fi
