--- ext/curl/config.m4.orig	2014-08-13 20:22:50.000000000 +0100
+++ ext/curl/config.m4	2021-07-23 02:41:22.000000000 +0100
@@ -65,6 +65,7 @@
     AC_PROG_CPP
     AC_MSG_CHECKING([for openssl support in libcurl])
     AC_TRY_RUN([
+#include <strings.h>
 #include <curl/curl.h>
 
 int main(int argc, char *argv[])
@@ -92,6 +93,7 @@
    
     AC_MSG_CHECKING([for gnutls support in libcurl])
     AC_TRY_RUN([
+#include <strings.h>
 #include <curl/curl.h>
 
 int main(int argc, char *argv[])
