commit 4e6d54f5a7003b73f12d86d7f5cba0a37ce40930
Author: Pierre Joye <pierre.php@gmail.com>
Date:   Fri Mar 22 08:28:11 2013 +0100

    - fix regression (imagerotate_overflow.phpt)

diff --git a/ext/gd/gd.c b/ext/gd/gd.c
index caf84e304e..e291793fd7 100644
--- ext/gd/gd.c.orig
+++ ext/gd/gd.c
@@ -84,6 +84,10 @@
 # endif
 #endif
 
+#if defined(HAVE_GD_XPM) && defined(HAVE_GD_BUNDLED)
+# include "X11/xpm.h"
+#endif
+
 #ifndef M_PI
 #define M_PI 3.14159265358979323846
 #endif
@@ -125,6 +129,10 @@
 #define gdNewDynamicCtxEx(len, data, val) gdNewDynamicCtx(len, data)
 #endif
 
+/* as it is not really public, duplicate declaration here to avoid 
+   pointless warnings */
+int overflow2(int a, int b);
+
 /* Section Filters Declarations */
 /* IMPORTANT NOTE FOR NEW FILTER
  * Do not forget to update:
