Don't overlink with libxml2's dependencies.
--- a/acinclude.m4	2021-11-16 05:18:41.000000000 -0600
+++ b/acinclude.m4	2024-01-10 03:39:13.000000000 -0600
@@ -2564,7 +2564,7 @@
     LIBXML_VERSION=`expr [$]1 \* 1000000 + [$]2 \* 1000 + [$]3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       AC_MSG_ERROR([libxml2 version 2.6.11 or greater required.])
--- a/configure	2021-11-16 05:18:29.000000000 -0600
+++ b/configure	2024-01-10 03:40:01.000000000 -0600
@@ -17478,7 +17478,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -29756,7 +29756,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -75716,7 +75716,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -77260,7 +77260,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -85861,7 +85861,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -86735,7 +86735,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -87552,7 +87552,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -88293,7 +88293,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
@@ -90619,7 +90619,7 @@
     LIBXML_VERSION=`expr $1 \* 1000000 + $2 \* 1000 + $3`
     if test "$LIBXML_VERSION" -ge "2006011"; then
       found_libxml=yes
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
+      LIBXML_LIBS=`$XML2_CONFIG --libs --dynamic`
       LIBXML_INCS=`$XML2_CONFIG --cflags`
     else
       as_fn_error $? "libxml2 version 2.6.11 or greater required." "$LINENO" 5
