https://bugs.php.net/bug.php?id=78750
--- a/configure.orig	2019-12-24 16:15:30.000000000 -0600
+++ b/configure	2019-12-24 16:21:04.000000000 -0600
@@ -5377,6 +5377,7 @@
 $as_echo_n "checking for bison version... " >&6; }
 
     php_bison_version=$($YACC --version 2> /dev/null | grep 'GNU Bison' | cut -d ' ' -f 4 | tr -d a-z)
+    [ -z "$php_bison_version" ] && php_bison_version=0.0.0
     ac_IFS=$IFS; IFS="."
     set $php_bison_version
     IFS=$ac_IFS
@@ -5474,6 +5475,7 @@
 $as_echo_n "checking for re2c version... " >&6; }
 
     php_re2c_version=$($RE2C --version | cut -d ' ' -f 2 2>/dev/null)
+    [ -z "$php_re2c_version" ] && php_re2c_version=0.0.0
     ac_IFS=$IFS; IFS="."
     set $php_re2c_version
     IFS=$ac_IFS
