--- ext/odbc/config.m4.orig	2015-04-15 13:05:57.000000000 -0500
+++ ext/odbc/config.m4	2015-04-17 20:15:12.000000000 -0500
@@ -104,6 +104,7 @@
 [  --with-odbcver[=HEX]      Force support for the passed ODBC version. A hex number is expected, default 0x0300.
                              Use the special value of 0 to prevent an explicit ODBCVER to be defined. ], 0x0300)
 
+:<<'MACPORTS_DISABLED'
 if test -z "$ODBC_TYPE"; then
 PHP_ARG_WITH(adabas,,
 [  --with-adabas[=DIR]       Include Adabas D support [/usr/local]])
@@ -386,6 +387,7 @@
   fi
 fi
 
+MACPORTS_DISABLED
 if test -z "$ODBC_TYPE"; then
 PHP_ARG_WITH(iodbc,,
 [  --with-iodbc[=DIR]        Include iODBC support [/usr/local]])
