--- ext/odbc/config.m4.orig	2020-11-24 05:10:57.000000000 -0600
+++ ext/odbc/config.m4	2020-12-02 15:58:35.000000000 -0600
@@ -102,6 +102,7 @@
 [  --with-odbcver[=HEX]      Force support for the passed ODBC version. A hex number is expected, default 0x0350.
                           Use the special value of 0 to prevent an explicit ODBCVER to be defined. ], 0x0350)
 
+:<<'MACPORTS_DISABLED'
 if test -z "$ODBC_TYPE"; then
 PHP_ARG_WITH(adabas,,
 [  --with-adabas[=DIR]       Include Adabas D support [/usr/local]])
@@ -372,6 +373,7 @@
   fi
 fi
 
+MACPORTS_DISABLED
 if test -z "$ODBC_TYPE"; then
 PHP_ARG_WITH(unixODBC,,
 [  --with-unixODBC[=DIR]     Include unixODBC support [/usr/local]])
