--- Makefile.PL.orig	2019-11-01 22:17:15.000000000 -0700
+++ Makefile.PL	2019-11-01 22:17:45.000000000 -0700
@@ -1,6 +1,7 @@
 #!/usr/bin/perl
 my $required_svn = '1.2.0';
 
+use lib '.';
 use inc::Module::Install;
 
 name		('SVN-Mirror');
@@ -33,5 +34,8 @@
 #sign(1);
 WriteAll( sign => 1);
 
+my ($required_version_major, $required_version_minor) = split(/./, $required_svn) ;
+my ($core_version_major, $core_version_minor) = split(/./, $SVN::Core::VERSION);
+
 die "SVN::Core $required_svn must be manually installed"
-    unless eval {require SVN::Core} && $SVN::Core::VERSION >= $required_svn;
+    unless eval {require SVN::Core} && $core_version_major == $required_version_major && $core_version_minor >= $required_version_minor;
