--- perl-module/Makefile.PL.orig	2010-06-09 13:05:19.000000000 -0400
+++ perl-module/Makefile.PL	2010-06-09 13:05:30.000000000 -0400
@@ -22,7 +22,7 @@
       (ABSTRACT_FROM  => 'lib/PSPP.pm', # retrieve abstract from module
        AUTHOR         => 'John Darrington <john@darrington.wattle.id.au>') : ()),
     INC       => "-I $Locations{SourceDir}  -I $Locations{SourceDir}/src -I $Locations{SourceDir}/gl -I $Locations{BuildDir}/gl -I $Locations{BuildDir}", 
-    MYEXTLIB  => "$Locations{BuildDir}/src/.libs/libpspp-core.\$(DLEXT)",
+    MYEXTLIB  => "$Locations{BuildDir}/src/.libs/libpspp-core.\$(SO)",
     MAN3PODS  => {"lib/PSPP.pm", "\$(INST_MAN3DIR)/PSPP.3pm",
 	    "Examples.pod", "\$(INST_MAN3DIR)/PSPP::Examples.3pm"}
 );
