--- nepomuk/services/storage/resourcemerger.cpp.orig	2012-05-05 16:10:07.000000000 -0400
+++ nepomuk/services/storage/resourcemerger.cpp	2012-05-05 16:10:42.000000000 -0400
@@ -506,7 +506,7 @@
     m_mappings.insert( QUrl(node.toN3()), newUri );
 
     // FIXME: trueg: IMHO these statements should instead be added to the list of all statements so there is only one place where anything is actually added to the model
-    Soprano::Node dateTime( Soprano::LiteralValue( QDateTime::currentDateTime() ) );
+    Soprano::Node dateTime = ( Soprano::LiteralValue( QDateTime::currentDateTime() ) );
     m_model->addStatement( newUri, NAO::created(), dateTime, m_graph );
     m_model->addStatement( newUri, NAO::lastModified(), dateTime, m_graph );
 
