--- kglobalaccel/globalshortcutsregistry.cpp.orig	2012-11-30 16:26:48.000000000 +0900
+++ kglobalaccel/globalshortcutsregistry.cpp	2013-01-11 14:33:56.000000000 +0900
@@ -33,7 +33,7 @@
 #include "kglobalaccel_x11.h"
 #include <QX11Info>
 #include <QApplication>
-#elif defined(Q_WS_MACX)
+#elif defined(Q_WS_MAC)
 #include "kglobalaccel_mac.h"
 #elif defined(Q_WS_WIN)
 #include "kglobalaccel_win.h"
