--- configure.ac.old	2018-04-04 23:30:08.000000000 -0700
+++ configure.ac	2018-04-04 23:32:33.000000000 -0700
@@ -52,7 +52,7 @@
   fi
   if test -d "${with_libintl_prefix}/lib" && test -f "${with_libintl_prefix}/lib/libintl.a" ; then
     LIBINTL="${with_libintl_prefix}/lib/libintl.a"
-    LIBS="$LIBINTL /usr/lib/libiconv.dylib $LIBS"
+    LIBS="-lintl -licuuc -liconv $LIBS"
   else
     AC_MSG_ERROR([The specified --with-libintl-prefix did not contain lib/libintl.a which is required on MacOS for static link])
   fi
