--- ./doc/meson.build	2023-06-27 04:27:14.000000000 -0400
+++ ./doc/meson.build	2023-06-27 04:27:39.000000000 -0400
@@ -4,7 +4,7 @@
 
     # This is *completely* ridiculous, but apparently this thing has THAT many faces.
     basenames = ['rst2html5', 'rst2html', 'rst2html4']
-    suffixes = ['', '-3', '3', '-docutils']
+    suffixes = ['', '-3', '3', '@@PY_VERSION@@', '-@@PY_VERSION@@', '-docutils']
 
     foreach basename : basenames
         foreach suffix : suffixes
