--- ./CMakeLists.txt	2026-04-05 08:16:47
+++ ./CMakeLists.txt	2026-04-05 08:36:51
@@ -24,7 +24,7 @@
 # https://reproducible-builds.org/docs/source-date-epoch/
 if(DEFINED ENV{SOURCE_DATE_EPOCH})
   execute_process(
-    COMMAND "date" "-u" "-d" "@$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d"
+    COMMAND "date" "-u" "-r" "$ENV{SOURCE_DATE_EPOCH}" "+%Y-%m-%d"
     OUTPUT_VARIABLE BUILD_DATE
     OUTPUT_STRIP_TRAILING_WHITESPACE)
 else()
@@ -426,7 +426,7 @@
   # find system nanobind
   if(NOT DEFINED nanobind_DIR OR "${nanobind_DIR}" STREQUAL "")
     execute_process(
-      COMMAND ${Python_EXECUTABLE} -m nanobind --cmake-dir
+      COMMAND ${Python_EXECUTABLE} -m nanobind
       RESULT_VARIABLE NANOBIND_CMAKE_RESULT
       OUTPUT_VARIABLE NANOBIND_CMAKE_DIR
       OUTPUT_STRIP_TRAILING_WHITESPACE
