--- configure.ac.orig	2014-02-21 06:00:00.000000000 -0800
+++ configure.ac	2014-06-23 03:42:50.000000000 -0700
@@ -127,33 +127,12 @@
     ;;
   darwin*)
 
-    dnl Force gcc "-arch" flag
-    ARCH_flag=""
-    case "${host}" in
-      i?86*)
-        ARCH_flag="-arch i386"
-      ;;
-      ppc64*)
-        ARCH_flag="-arch ppc64"
-      ;;
-      ppc*)
-        ARCH_flag="-arch ppc"
-      ;;
-      x86_64*)
-        ARCH_flag="-arch x86_64"
-      ;;
-      arm*)
-        ac_cv_c_bigendian="no"
-        ac_cv_c_attribute_packed="no"
-      ;;
-    esac
-
     SYS=darwin
-    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS ${ARCH_flag}"
-    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS ${ARCH_flag}"
-    CPPFLAGS="${CPPFLAGS} ${ARCH_flag}"
-    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu99 ${ARCH_flag}"
-    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names ${ARCH_flag}"
+    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS"
+    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS"
+    CPPFLAGS="${CPPFLAGS}"
+    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu99"
+    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names"
     VLC_ADD_LIBS([motion rotate dvdread dvdnav], [-Wl,-framework,IOKit,-framework,CoreFoundation])
     VLC_ADD_LIBS([libvlc vlc],[-Wl,-undefined,dynamic_lookup,-framework,AppKit])
     VLC_ADD_LIBS([avcodec access_avio swscale postproc i420_rgb_mmx x264 x26410b],[-Wl,-read_only_relocs,suppress])
