diff --git configure.ac configure.ac
index be65881..704e92a 100644
--- configure.ac
+++ configure.ac
@@ -138,20 +138,12 @@ case "${host_os}" in
     ;;
   darwin*)
 
-    dnl Force gcc "-arch" flag
-    ARCH_flag=""
     case "${host}" in
       i?86*)
-        ARCH_flag="-arch i386"
-      ;;
-      ppc64*)
-        ARCH_flag="-arch ppc64"
-      ;;
-      ppc*)
-        ARCH_flag="-arch ppc"
+        YASMFLAGS="-f macho32"
       ;;
       x86_64*)
-        ARCH_flag="-arch x86_64"
+        YASMFLAGS="-f macho64"
       ;;
       arm*)
         ac_cv_c_bigendian="no"
@@ -160,12 +152,12 @@ case "${host_os}" in
     esac
 
     SYS=darwin
-    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu11 ${ARCH_flag}"
+    CFLAGS="${CFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu11"
     CXX="${CXX} -stdlib=libc++ -std=c++11"
-    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS ${ARCH_flag}"
-    CPPFLAGS="${CPPFLAGS} ${ARCH_flag}"
-    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu11 ${ARCH_flag} -Werror=partial-availability"
-    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names ${ARCH_flag}"
+    CXXFLAGS="${CXXFLAGS} -D_INTL_REDIRECT_MACROS"
+dnl     CPPFLAGS="${CPPFLAGS} ${ARCH_flag}"
+    OBJCFLAGS="${OBJCFLAGS} -D_INTL_REDIRECT_MACROS -std=gnu11 -Werror=partial-availability"
+    LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names"
     VLC_ADD_LIBS([libvlc vlc],[-Wl,-undefined,dynamic_lookup,-framework,AppKit])
     VLC_ADD_LIBS([libvlccore],[-Wl,-framework,CoreFoundation,-framework,CoreServices])
 
