diff --git modules/gui/macosx/VLCPlaylistInfo.h modules/gui/macosx/VLCPlaylistInfo.h
index e7e1ffd4a5a6e9f9917c7afb526d4c795ede6935..048b08b4a0a0cff72116991c2d0b83b3ab85da84 100644
--- modules/gui/macosx/VLCPlaylistInfo.h
+++ modules/gui/macosx/VLCPlaylistInfo.h
@@ -106,8 +106,8 @@
  */
 @interface VLCInfoTreeItem : NSObject
 
-@property (readwrite) NSString *name;
-@property (readwrite) NSString *value;
+@property (readwrite) NSString *propertyName;
+@property (readwrite) NSString *propertyValue;
 
 @property (readwrite) NSArray *children;
 
diff --git modules/gui/macosx/VLCPlaylistInfo.m modules/gui/macosx/VLCPlaylistInfo.m
index 6d5c8c2906b9d0eb45cdad64a705bcdae0091fe0..eb4e481a666c70c48087ae707405a87529ec9b2b 100644
--- modules/gui/macosx/VLCPlaylistInfo.m
+++ modules/gui/macosx/VLCPlaylistInfo.m
@@ -303,15 +303,15 @@ FREENULL( psz_##foo );
             info_category_t *cat = p_item->pp_categories[i];
 
             VLCInfoTreeItem *subItem = [[VLCInfoTreeItem alloc] init];
-            subItem.name = toNSStr(cat->psz_name);
+            subItem.propertyName = toNSStr(cat->psz_name);
 
             // Build list of codec details
             NSMutableArray *infos = [NSMutableArray array];
 
             for (int j = 0; j < cat->i_infos; j++) {
                 VLCInfoTreeItem *infoItem = [[VLCInfoTreeItem alloc] init];
-                infoItem.name = toNSStr(cat->pp_infos[j]->psz_name);
-                infoItem.value = toNSStr(cat->pp_infos[j]->psz_value);
+                infoItem.propertyName = toNSStr(cat->pp_infos[j]->psz_name);
+                infoItem.propertyValue = toNSStr(cat->pp_infos[j]->psz_value);
                 [infos addObject:infoItem];
             }
 
@@ -397,9 +397,9 @@ FREENULL( psz_##foo );
         return @"";
 
     if ([[tableColumn identifier] isEqualToString:@"0"])
-        return [item name];
+        return [item propertyName];
     else
-        return [item value];
+        return [item propertyValue];
 }
 
 @end
