--- Makefile.am.orig	2009-07-07 17:34:22.000000000 +0200
+++ Makefile.am	2009-07-07 17:38:50.000000000 +0200
@@ -411,13 +411,6 @@
 vlc$(EXEEXT):
 	ln -sf bin/vlc-static$(EXEEXT) vlc$(EXEEXT)
 
-if HAVE_DARWIN
-if BUILD_VLC
-# Create the MacOS X app
-noinst_DATA = VLC.app
-endif
-endif
-
 # VLC-release.app for packaging and giving it to your friends
 # use package-macosx to get a nice dmg
 VLC-release.app: vlc
@@ -428,7 +421,7 @@
 	for i in vlc.xcodeproj Resources README.MacOSX.rtf ; do \
 	  cp -R $(srcdir)/extras/package/macosx/$$i $(top_builddir)/tmp/extras/package/macosx/; \
 	done
-	REVISION=`git describe --always` && \
+	REVISION="$(VERISON_REVISION)" && \
 	cat $(top_builddir)/extras/package/macosx/Info.plist | \
 	sed "s/#REVISION#/$$REVISION/g" > $(top_builddir)/tmp/extras/package/macosx/Info.plist
 	cp -R $(top_builddir)/extras/package/macosx/Resources $(top_builddir)/tmp/extras/package/macosx/
@@ -582,7 +575,7 @@
 	for i in vlc.xcodeproj Resources README.MacOSX.rtf; do \
 	  cp -R $(srcdir)/extras/package/macosx/$$i $(top_builddir)/tmp/extras/package/macosx/; \
 	done
-	REVISION=`git describe --always` && \
+	REVISION="$(VERSION_REVISION)" && \
 	cat $(top_builddir)/extras/package/macosx/Info.plist | \
 	sed "s/#REVISION#/$$REVISION/g" > $(top_builddir)/tmp/extras/package/macosx/Info.plist
 	cp -R $(top_builddir)/extras/package/macosx/Resources $(top_builddir)/tmp/extras/package/macosx/
