--- modules/access/bluray.c.orig	2015-12-06 06:41:52.000000000 -0800
+++ modules/access/bluray.c	2015-12-06 06:43:19.000000000 -0800
@@ -43,6 +43,11 @@
 #include <vlc_plugin.h>
 #include <vlc_demux.h>                      /* demux_t */
 #include <vlc_input.h>                      /* Seekpoints, chapters */
+
+#ifndef __STDC_NO_ATOMICS__
+#define __STDC_NO_ATOMICS__ 1
+#endif
+
 #include <vlc_atomic.h>
 #include <vlc_dialog.h>                     /* BD+/AACS warnings */
 #include <vlc_vout.h>                       /* vout_PutSubpicture / subpicture_t */
