diff --git src/darwin/dirs.c src/darwin/dirs.c
index d0489fd21cbb56a0730feb68e8fdbe8066728c0a..fa701de23a3f7cbe0e1a6e8ca4276cf12d2f729d 100644
--- src/darwin/dirs.c
+++ src/darwin/dirs.c
@@ -36,14 +36,20 @@
 
 #include <CoreFoundation/CoreFoundation.h>
 
-char *config_GetLibDir (void)
+// 20150205: we should do case-insensitive filename comparisons. HFS is case-insensitive by default,
+// and while it strives to preserve case there is no guarantee that case folding will never occur,
+// especially in directory components.
+
+// @p libDirIsPosix is set to true in if getLibDir() detects we're installed
+// in a (linux-specific) posix way, like e.g. MacPorts would do.
+char *getLibDir (bool *libDirIsPosix)
 {
+    *libDirIsPosix = FALSE;
     /* Get the full program path and name */
     /* First try to see if we are linked to the framework */
-    for (unsigned i = 0; i < _dyld_image_count(); i++)
-    {
+    for (unsigned i = 0; i < _dyld_image_count(); i++) {
         const char *psz_img_name = _dyld_get_image_name(i);
-        const char *p = strstr( psz_img_name, "VLCKit.framework/Versions/" );
+        const char *p = strcasestr( psz_img_name, "VLCKit.framework/Versions/" );
 
         /* Check for "VLCKit.framework/Versions/Current/VLCKit",
          * as well as "VLCKit.framework/Versions/A/VLCKit" and
@@ -54,18 +60,18 @@ char *config_GetLibDir (void)
             p += strcspn( p, "/" );
 
             /* If the string ends with VLCKit then we've found a winner */
-            if (!strcmp( p, "/VLCKit"))
+            if (!strcasecmp( p, "/VLCKit"))
                 return strdup( dirname(psz_img_name) );
         }
 
-        /* Do we end by "VLC"? If so we are the legacy VLC.app that doesn't
+        /* Do we end by "MacOS/VLC"? If so we are the legacy VLC.app that doesn't
          * link to VLCKit. */
         size_t len = strlen(psz_img_name);
-        if (len >= 3 && !strcmp( psz_img_name + len - 3, "VLC"))
+        if (len >= 9 && !strcasecmp( psz_img_name + len - 9, "MacOS/VLC"))
             return strdup( dirname(psz_img_name) );
 
         /* Do we end by "VLC-Plugin"? oh, we must be the NPAPI plugin */
-        if (len >= 10 && !strcmp( psz_img_name + len - 10, "VLC-Plugin"))
+        if (len >= 10 && !strcasecmp( psz_img_name + len - 10, "VLC-Plugin"))
             return strdup( dirname(psz_img_name) );
 
         /* Do we end by "VLC for iOS"? so we are the iOS app */
@@ -85,6 +91,7 @@ char *config_GetLibDir (void)
         char *path = NULL;
         asprintf(&path, "%s/"PACKAGE, incompletepath);
         free(incompletepath);
+        *libDirIsPosix = TRUE;
         return path;
     }
 
@@ -92,15 +99,37 @@ char *config_GetLibDir (void)
     abort ();
 }
 
+char *config_GetLibDir (void)
+{
+    bool isPosixStub;
+    return getLibDir (&isPosixStub);
+}
+
 char *config_GetDataDir (void)
 {
     const char *path = getenv ("VLC_DATA_PATH");
     if (path)
         return strdup (path);
 
-    char *vlcpath = config_GetLibDir ();
-    char *datadir;
+    bool libDirIsPosix;
+    char *vlcpath = getLibDir (&libDirIsPosix);
+    char *datadir = NULL;
 
+    if (libDirIsPosix) {
+        // vlcpath should point to something like /opt/local/lib/vlc
+        // if so, we can chop off the /lib/vlc bit, and then add /share
+        // like we would otherwise.
+        char *p = strcasestr(vlcpath, "/lib/");
+        if (p != NULL) {
+            *p = '\0';
+        }
+        else {
+            // something else ... return PKGDATADIR
+            free(vlcpath);
+            datadir = strdup(PKGDATADIR);
+            return datadir;
+        }
+    }
     if (asprintf (&datadir, "%s/share", vlcpath) == -1)
         datadir = NULL;
 
@@ -141,19 +170,27 @@ static char *getAppDependentDir(vlc_userdir_t type)
     const char *fallback = "org.videolan.vlc";
     char *name = NULL;
 
-    CFBundleRef mainBundle = CFBundleGetMainBundle();
-    if (mainBundle) {
-        CFStringRef identifierAsNS = CFBundleGetIdentifier(mainBundle);
-        if (identifierAsNS) {
-            CFIndex len = CFStringGetLength(identifierAsNS);
-            CFIndex size = CFStringGetMaximumSizeForEncoding(len, kCFStringEncodingUTF8);
-            char *identifier = calloc(len + 1, sizeof(char));
-            if (identifier != NULL) {
-                Boolean ret = CFStringGetCString(identifierAsNS, identifier, size, kCFStringEncodingUTF8);
-                if (ret)
-                    name = identifier;
-                else
-                    free(identifier);
+    bool libDirIsPosix;
+    char *vlcpath = getLibDir(&libDirIsPosix);
+    if (vlcpath) {
+        free(vlcpath);
+    }
+
+    if (!libDirIsPosix) {
+        CFBundleRef mainBundle = CFBundleGetMainBundle();
+        if (mainBundle) {
+            CFStringRef identifierAsNS = CFBundleGetIdentifier(mainBundle);
+            if (identifierAsNS) {
+                CFIndex len = CFStringGetLength(identifierAsNS);
+                CFIndex size = CFStringGetMaximumSizeForEncoding(len, kCFStringEncodingUTF8);
+                char *identifier = calloc(len + 1, sizeof(char));
+                if (identifier != NULL) {
+                    Boolean ret = CFStringGetCString(identifierAsNS, identifier, size, kCFStringEncodingUTF8);
+                    if (ret)
+                        name = identifier;
+                    else
+                        free(identifier);
+                }
             }
         }
     }
diff --git bin/Makefile.am bin/Makefile.am
index 6033e50..ea7b544 100644
--- bin/Makefile.am
+++ bin/Makefile.am
@@ -132,7 +132,7 @@ MOSTLYCLEANFILES = $(noinst_DATA)
 if HAVE_OSX
 if BUILD_VLC
 install-data-local:
-	cd $(bindir); mv vlc-osx vlc
+	cd $(DESTDIR)/$(bindir); mv vlc-osx vlc
 
 endif
 endif
