diff --git modules/access/qtsound.m modules/access/qtsound.m
index af190d1..3f0fc4f 100644
--- modules/access/orig.qtsound.m	2018-10-28 12:11:19.000000000 +0100
+++ modules/access/qtsound.m	2019-02-18 00:41:40.000000000 +0100
@@ -45,6 +45,7 @@
 #import <QTKit/QTKit.h>
 #import <AVFoundation/AVFoundation.h>
 
+#ifdef MAC_OS_X_VERSION_10_13
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
 
@@ -52,6 +53,7 @@
 
 @end
 #endif
+#endif
 
 /*****************************************************************************
  * Local prototypes.
@@ -318,6 +320,7 @@
             goto error;
         }
 
+#ifdef MAC_OS_X_VERSION_10_13
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the audio device");
 
@@ -334,6 +337,7 @@
                 goto error;
             }
         }
+#endif
 
         audioInput = [[QTCaptureDeviceInput alloc] initWithDevice: p_sys->audiodevice];
         if(!audioInput) {
diff --get modules/access/avcapture.m modules/access/avcapture.m
index af190d1..3f0fc4f 100644
--- modules/access/orig.avcapture.m	2018-10-28 12:11:19.000000000 +0100
+++ modules/access/avcapture.m	2019-02-18 00:27:36.000000000 +0100
@@ -43,6 +43,7 @@
 #import <AVFoundation/AVFoundation.h>
 #import <CoreMedia/CoreMedia.h>
 
+#ifdef MAC_OS_X_VERSION_10_13
 #ifndef MAC_OS_X_VERSION_10_14
 @interface AVCaptureDevice (AVCaptureDeviceAuthorizationSince10_14)
 
@@ -50,6 +51,7 @@
 
 @end
 #endif
+#endif
 
 /*****************************************************************************
 * Local prototypes
@@ -334,6 +336,7 @@ static int Open(vlc_object_t *p_this)
             goto error;
         }
 
+#ifdef MAC_OS_X_VERSION_10_13
         if (@available(macOS 10.14, *)) {
             msg_Dbg(p_demux, "Check user consent for access to the video device");
 
@@ -350,6 +353,7 @@ static int Open(vlc_object_t *p_this)
                 goto error;
             }
         }
+#endif
 
         input = [AVCaptureDeviceInput deviceInputWithDevice:(__bridge AVCaptureDevice *)p_sys->device error:&o_returnedError];
 
diff --git modules/video_output/macosx.m modules/video_output/macosx.m
index 20aedfc63e440cbb16ec3d0078d7d433434b8131..26ca20e4b449a52b7650a3ef7f2c1337f824bc1e 100644
--- modules/video_output/macosx.m
+++ modules/video_output/macosx.m
@@ -339,6 +339,7 @@ static void PictureDisplay (vout_display_t *vd, picture_t *pic, subpicture_t *su
     [sys->glView setVoutFlushing:YES];
     if (vlc_gl_MakeCurrent(sys->gl) == VLC_SUCCESS)
     {
+#ifdef MAC_OS_X_VERSION_10_13
         if (@available(macOS 10.14, *)) {
             vout_display_place_t place;
             vout_display_PlacePicture(&place, &vd->source, vd->cfg, false);
@@ -346,6 +347,7 @@ static void PictureDisplay (vout_display_t *vd, picture_t *pic, subpicture_t *su
                                          vd->cfg->display.height - (place.y + place.height),
                                          place.width, place.height);
         }
+#endif
 
         vout_display_opengl_Display (sys->vgl, &vd->source);
         vlc_gl_ReleaseCurrent(sys->gl);
