--- modules/access/Modules.am
+++ modules/access/Modules.am
@@ -1,3 +1,5 @@
+AM_LIBTOOLFLAGS=--tag=CC
+
 SUBDIRS = rtp
 noinst_LTLIBRARIES =
 
--- modules/access/qtsound.m
+++ modules/access/qtsound.m
@@ -40,7 +40,7 @@
 #include <vlc_dialog.h>
 
 //#define QTKIT_VERSION_MIN_REQUIRED 70603
-#define QTKIT_VERSION_MAX_ALLOWED 70700
+//#define QTKIT_VERSION_MAX_ALLOWED 70700
 
 #import <QTKit/QTKit.h>
 
