--- Makefile.in.orig	2009-04-04 03:32:07.000000000 +0200
+++ Makefile.in	2009-04-04 03:33:06.000000000 +0200
@@ -928,7 +928,7 @@
 DISTCLEANFILES = $(BUILT_SOURCES_distclean) vlc-config.in compile ChangeLog
 
 # Create the MacOS X app
-@BUILD_VLC_TRUE@@HAVE_DARWIN_TRUE@noinst_DATA = VLC.app
+@BUILD_VLC_TRUE@@HAVE_DARWIN_TRUE@noinst_DATA =
 @HAVE_BEOS_TRUE@noinst_DATA = vlc-bundle
 
 ###############################################################################
@@ -1539,7 +1539,7 @@
 	for i in vlc.xcodeproj Resources README.MacOSX.rtf ; do \
 	  cp -R $(srcdir)/extras/package/macosx/$$i $(top_builddir)/tmp/extras/package/macosx/; \
 	done
-	REVISION=`git describe` && \
+	REVISION="$(VERSION_REVISION)" && \
 	cat $(top_builddir)/extras/package/macosx/Info.plist | \
 	sed "s/#REVISION#/$$REVISION/g" > $(top_builddir)/tmp/extras/package/macosx/Info.plist
 	cp -R $(top_builddir)/extras/package/macosx/Resources $(top_builddir)/tmp/extras/package/macosx/
@@ -1706,7 +1706,7 @@
 	for i in vlc.xcodeproj Resources README.MacOSX.rtf; do \
 	  cp -R $(srcdir)/extras/package/macosx/$$i $(top_builddir)/tmp/extras/package/macosx/; \
 	done
-	REVISION=`git describe` && \
+	REVISION="$(VERSION_REVISION)" && \
 	cat $(top_builddir)/extras/package/macosx/Info.plist | \
 	sed "s/#REVISION#/$$REVISION/g" > $(top_builddir)/tmp/extras/package/macosx/Info.plist
 	cp -R $(top_builddir)/extras/package/macosx/Resources $(top_builddir)/tmp/extras/package/macosx/
