--- modules/codec/x264.c.orig	2009-04-04 02:07:20.000000000 +0200
+++ modules/codec/x264.c	2009-04-04 02:09:00.000000000 +0200
@@ -957,9 +957,12 @@
         p_sys->param.i_scenecut_threshold = val.i_int;
 #endif
 
-#if X264_BUILD >= 55 /* r607 */
+#if X264_BUILD >= 55 /* r607 */ && X264_BUILD < 67 /* r1117 */
     var_Get( p_enc, SOUT_CFG_PREFIX "pre-scenecut", &val );
     p_sys->param.b_pre_scenecut = val.b_bool;
+#endif
+
+#if X264_BUILD >= 55 /* r607 */
     var_Get( p_enc, SOUT_CFG_PREFIX "non-deterministic", &val );
     p_sys->param.b_deterministic = val.b_bool;
 #endif
