--- src/_macosx.m.orig	2018-08-09 23:01:31.000000000 -0400
+++ src/_macosx.m	2018-10-04 10:20:58.000000000 -0400
@@ -1,5 +1,8 @@
 #include <Cocoa/Cocoa.h>
 #include <ApplicationServices/ApplicationServices.h>
+#ifndef kCTForegroundColorFromContextAttributeName
+extern const CFStringRef kCTForegroundColorFromContextAttributeName AVAILABLE_MAC_OS_X_VERSION_10_5_AND_LATER;
+#endif
 #include <sys/socket.h>
 #include <Python.h>
 
