--- setupext.py.orig	2023-11-18 21:55:31
+++ setupext.py	2023-11-18 21:55:57
@@ -803,7 +803,7 @@
             'matplotlib.backends._macosx', [
                 'src/_macosx.m'
             ])
-        ext.extra_compile_args.extend(['-Werror'])
+        ext.extra_compile_args.extend(['-Wextra'])
         ext.extra_link_args.extend(['-framework', 'Cocoa'])
         if platform.python_implementation().lower() == 'pypy':
             ext.extra_compile_args.append('-DPYPY=1')
