--- setup.cfg	2013-08-07 11:50:33.000000000 +0400
+++ setup.cfg	2013-08-07 11:54:34.000000000 +0400
@@ -0,0 +1,24 @@
+[gui_support]
+agg=False
+tkagg=False
+gtk=False
+gtkagg=False
+gtk3agg=False
+gtk3cairo=False
+wxagg=False
+macosx=True
+windowing=False
+qt4agg=False
+pyside=False
+cairo=False
+
+[packages]
+sample_data=True
+toolkits=True
+tests=False
+
+[rc_options]
+backend=MacOSX
+
+[directories]
+basedirlist=@@PREFIX@@
