--- pyproject.toml.orig	2024-12-14 22:19:39
+++ pyproject.toml	2024-12-14 22:20:03
@@ -48,7 +48,7 @@
 [project.optional-dependencies]
 # Should be a copy of the build dependencies below.
 dev = [
-    "meson-python>=0.13.1,<0.17.0",
+    "meson-python>=0.13.1",
     "numpy>=1.25",
     "pybind11>=2.6,!=2.13.3",
     "setuptools_scm>=7",
@@ -73,7 +73,7 @@
 build-backend = "mesonpy"
 # Also keep in sync with optional dependencies above.
 requires = [
-    "meson-python>=0.13.1,<0.17.0",
+    "meson-python>=0.13.1",
     "pybind11>=2.6,!=2.13.3",
     "setuptools_scm>=7",
 
@@ -87,7 +87,7 @@
     #      and disabling build isolation.
     #   3. The <2.3 upper bound is for matching the numpy deprecation policy,
     #      it should not be loosened.
-    "numpy>=2.0.0rc1,<2.3",
+    "numpy",
 ]
 
 [tool.meson-python.args]
