--- src/_macosx.m.orig	2015-11-29 10:44:55.000000000 -0700
+++ src/_macosx.m	2015-11-29 11:02:16.000000000 -0700
@@ -1,5 +1,8 @@
 #include <Cocoa/Cocoa.h>
 #include <ApplicationServices/ApplicationServices.h>
+#ifndef kCTForegroundColorFromContextAttributeName
+extern const CFStringRef kCTForegroundColorFromContextAttributeName AVAILABLE_MAC_OS_X_VERSION_10_5_AND_LATER;
+#endif
 #include <sys/socket.h>
 #include <Python.h>
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
