--- setupext.py.orig	2018-10-04 10:20:58.000000000 -0400
+++ setupext.py	2018-10-04 10:20:58.000000000 -0400
@@ -1453,10 +1453,10 @@
 
 class BackendTkAgg(OptionalBackendPackage):
     name = "tkagg"
-    force = True
-
-    def check(self):
-        return "installing; run-time loading from Python Tcl / Tk"
+##    force = True
+##
+##    def check(self):
+##        return "installing; run-time loading from Python Tcl / Tk"
 
     def runtime_check(self):
         """ Checks whether TkAgg runtime dependencies are met
