--- setup.py.orig	2020-03-03 08:42:24.000000000 -0500
+++ setup.py	2020-03-06 11:44:43.000000000 -0500
@@ -163,9 +163,9 @@
         _download_jquery_to(Path("lib/matplotlib/backends/web_backend/"))
 
 
-cmdclass['sdist'] = sdist_with_jquery
-cmdclass['install_lib'] = install_lib_with_jquery
-cmdclass['develop'] = develop_with_jquery
+##cmdclass['sdist'] = sdist_with_jquery
+##cmdclass['install_lib'] = install_lib_with_jquery
+##cmdclass['develop'] = develop_with_jquery
 
 
 # One doesn't normally see `if __name__ == '__main__'` blocks in a setup.py,
