--- setupext.py.orig	2007-02-01 01:32:51.000000000 -0800
+++ setupext.py	2007-03-30 15:51:37.000000000 -0700
@@ -48,8 +48,7 @@
     'linux2' : ['/usr/local', '/usr',],
     'linux'  : ['/usr/local', '/usr',],
     'cygwin' : ['/usr/local', '/usr',],
-    'darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
-                '/usr', '/sw'],
+    'darwin' : ['@@DPORTS_PREFIX@@', '/usr'],
     'freebsd4' : ['/usr/local', '/usr'],
     'freebsd5' : ['/usr/local', '/usr'],
     'freebsd6' : ['/usr/local', '/usr'],
@@ -132,6 +131,7 @@
                      [os.path.join(p, 'lib64')     for p in basedir[sys.platform] ] )
 
     module.include_dirs.extend(incdirs)
+    module.include_dirs.extend(['@@DPORTS_PREFIX@@/include/python2.4'])
     module.include_dirs.append('.')
     module.library_dirs.extend(libdirs)
 
