--- setupext.py.bak	Wed Mar  9 11:28:15 2005
+++ setupext.py	Wed Mar  9 11:28:48 2005
@@ -35,7 +35,7 @@
     'win32'  : ['win32_static',],
     'linux2' : ['/usr/local', '/usr',],
     'linux'  : ['/usr/local', '/usr',],
-    'darwin' : ['/usr/local', '/usr', '/sw', '/usr/X11R6'],
+    'darwin' : ['@@DPORTS_PREFIX@@', '/usr', '/usr/X11R6'],
     'freebsd4' : ['/usr/local', '/usr'],
     'sunos5' : [os.getenv('MPLIB_BASE') or '/usr/local',],
 }
