--- setupext.py.orig	2018-10-04 11:17:33.000000000 -0400
+++ setupext.py	2018-10-04 11:20:29.000000000 -0400
@@ -1676,35 +1676,36 @@
 
 
 def backend_gtk3cairo_internal_check(x):
-    try:
-        import cairocffi
-    except ImportError:
-        try:
-            import cairo
-        except ImportError:
-            return (False, "Requires cairocffi or pycairo to be installed.")
-
-    try:
-        import gi
-    except ImportError:
-        return (False, "Requires pygobject to be installed.")
-
-    try:
-        gi.require_version("Gtk", "3.0")
-    except ValueError:
-        return (False, "Requires gtk3 development files to be installed.")
-    except AttributeError:
-        return (False, "pygobject version too old.")
-
-    try:
-        from gi.repository import Gtk, Gdk, GObject
-    except (RuntimeError, ImportError):
-        return (False, "Requires pygobject to be installed.")
-
-    return (True, "version %s.%s.%s" % (
-        Gtk.get_major_version(),
-        Gtk.get_micro_version(),
-        Gtk.get_minor_version()))
+    return (True, "Gtk and cairo installed by MacPorts")
+##    try:
+##        import cairocffi
+##    except ImportError:
+##        try:
+##            import cairo
+##        except ImportError:
+##            return (False, "Requires cairocffi or pycairo to be installed.")
+##
+##    try:
+##        import gi
+##    except ImportError:
+##        return (False, "Requires pygobject to be installed.")
+##
+##    try:
+##        gi.require_version("Gtk", "3.0")
+##    except ValueError:
+##        return (False, "Requires gtk3 development files to be installed.")
+##    except AttributeError:
+##        return (False, "pygobject version too old.")
+##
+##    try:
+##        from gi.repository import Gtk, Gdk, GObject
+##    except (RuntimeError, ImportError):
+##        return (False, "Requires pygobject to be installed.")
+##
+##    return (True, "version %s.%s.%s" % (
+##        Gtk.get_major_version(),
+##        Gtk.get_micro_version(),
+##        Gtk.get_minor_version()))
 
 
 class BackendGtk3Cairo(OptionalBackendPackage):
