--- setupext.py.orig	2018-10-04 12:24:24.000000000 -0400
+++ setupext.py	2018-10-04 12:24:24.000000000 -0400
@@ -1343,10 +1343,10 @@
 
 class BackendTkAgg(OptionalBackendPackage):
     name = "tkagg"
-    force = True
-
-    def check(self):
-        return "installing; run-time loading from Python Tcl / Tk"
+##    force = True
+##
+##    def check(self):
+##        return "installing; run-time loading from Python Tcl / Tk"
 
     def get_extension(self):
         sources = [
