--- src/astar/src/astar.h.orig	2014-11-04 14:11:56.000000000 +0100
+++ src/astar/src/astar.h	2014-11-04 14:12:19.000000000 +0100
@@ -21,6 +21,7 @@
  
 #define _ASTAR_H
 
+#include <unistd.h>
 #include "postgres.h"
 #include "dijkstra.h"
 
--- src/dijkstra/src/dijkstra.h.orig	2014-11-04 14:13:53.000000000 +0100
+++ src/dijkstra/src/dijkstra.h	2014-11-04 14:14:20.000000000 +0100
@@ -22,6 +22,7 @@
 #ifndef _DIJKSTRA_H
 #define _DIJKSTRA_H
 
+#include <unistd.h>
 #include "postgres.h"
 
 typedef struct edge 
--- src/driving_distance/src/drivedist.h.orig	2014-11-04 14:15:06.000000000 +0100
+++ src/driving_distance/src/drivedist.h	2014-11-04 14:15:24.000000000 +0100
@@ -22,6 +22,7 @@
 #ifndef _DRIVEDIST_H
 #define _DRIVEDIST_H
 
+#include <unistd.h>
 #include "postgres.h"
 #include "dijkstra.h"
 
