--- DESCRIPTION	2024-05-20 11:50:02.000000000 +0800
+++ DESCRIPTION	2024-05-21 20:04:43.000000000 +0800
@@ -24,7 +24,7 @@
         0.4-1), lme4, lmerTest (>= 2.0.32), MASS, MuMIn, rsm, knitr,
         rmarkdown, sandwich, scales, splines, testthat, tibble, xtable
         (>= 1.8-2)
-Enhances: CARBayes, coxme, gee, geepack, MCMCglmm, MCMCpack, mice,
+Enhances: coxme, gee, geepack, MCMCglmm, MCMCpack, mice,
         nnet, pscl, rstanarm, sommer, survival
 URL:
         https://rvlenth.github.io/emmeans/,https://rvlenth.github.io/emmeans/
