From 4b421119737dfdc925baafce7b30433f9836010f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 6 Aug 2024 09:31:41 +0800
Subject: [PATCH 02/21] version.c: add a missing macro for powerpc

---
 src/detection/version/version.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/detection/version/version.c b/src/detection/version/version.c
index fe87e2e7..9f4b238c 100644
--- a/src/detection/version/version.c
+++ b/src/detection/version/version.c
@@ -10,7 +10,7 @@
     #define FF_ARCHITECTURE "arm"
 #elif defined(__mips__)
     #define FF_ARCHITECTURE "mips"
-#elif defined(__powerpc__) || defined(__powerpc)
+#elif defined(__powerpc__) || defined(__powerpc) || defined(__POWERPC__)
     #define FF_ARCHITECTURE "powerpc"
 #elif defined(__riscv__) || defined(__riscv)
     #define FF_ARCHITECTURE "riscv"
-- 
2.46.0

