From d3df0d80b4a121553ee52dbd280a2fd8addcbe4e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 Aug 2024 00:59:49 +0800
Subject: [PATCH 05/19] monitor_apple.m: no HDR before 10.11

---
 src/detection/monitor/monitor_apple.m | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git src/detection/monitor/monitor_apple.m src/detection/monitor/monitor_apple.m
index 2c4a1e3c..c0169c29 100644
--- src/detection/monitor/monitor_apple.m
+++ src/detection/monitor/monitor_apple.m
@@ -3,20 +3,25 @@
 #include "util/apple/cf_helpers.h"
 #include "util/edidHelper.h"
 
-#import <AppKit/NSScreen.h>
-#import <Foundation/Foundation.h>
-#import <CoreGraphics/CoreGraphics.h>
+#include <AppKit/NSScreen.h>
+#include <Foundation/Foundation.h>
+#include <CoreGraphics/CoreGraphics.h>
+#include <AvailabilityMacros.h>
 
 extern CFDictionaryRef CoreDisplay_DisplayCreateInfoDictionary(CGDirectDisplayID display) __attribute__((weak_import));
 
 #ifndef MAC_OS_X_VERSION_10_15
-#import <IOKit/graphics/IOGraphicsLib.h>
+#include <IOKit/graphics/IOGraphicsLib.h>
+#ifndef MAC_OS_X_VERSION_10_8
+#define CoreDisplay_IODisplayCreateInfoDictionary IODisplayCreateInfoDictionary
+#endif
 extern CFDictionaryRef CoreDisplay_IODisplayCreateInfoDictionary(io_service_t framebuffer, IOOptionBits options)  __attribute__((weak_import));
 #endif
 
 static bool detectHdrSupportWithNSScreen(FFDisplayResult* display)
 {
     NSScreen* mainScreen = NSScreen.mainScreen;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
     if (display->primary)
     {
         #ifdef MAC_OS_X_VERSION_10_15
@@ -41,6 +46,7 @@ static bool detectHdrSupportWithNSScreen(FFDisplayResult* display)
             }
         }
     }
+#endif
     return false;
 }
 
-- 
2.46.0

