From 623bcc5045ee78e8759dd34a672d5702c8dc30c4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 Aug 2024 10:12:44 +0800
Subject: [PATCH] Fix PowerPC macros

diff --git src/3rdparty/yyjson/yyjson.c src/3rdparty/yyjson/yyjson.c
index 16ecbacf..298a56c0 100644
--- src/3rdparty/yyjson/yyjson.c
+++ src/3rdparty/yyjson/yyjson.c
@@ -236,7 +236,7 @@ uint32_t yyjson_version(void) {
 #elif (defined(__BIG_ENDIAN__) && __BIG_ENDIAN__ == 1) || \
     defined(__ARMEB__) || defined(__THUMBEB__) || defined(__AARCH64EB__) || \
     defined(_MIPSEB) || defined(__MIPSEB) || defined(__MIPSEB__) || \
-    defined(__or1k__) || defined(__OR1K__)
+    defined(__or1k__) || defined(__OR1K__) || defined(__POWERPC__)
 #   define YYJSON_ENDIAN YYJSON_BIG_ENDIAN
 
 #else
diff --git src/detection/version/version.c src/detection/version/version.c
index b6d6e7d7..7d456313 100644
--- src/detection/version/version.c
+++ src/detection/version/version.c
@@ -10,7 +10,7 @@
     #define FF_ARCHITECTURE "arm"
 #elif defined(__mips__)
     #define FF_ARCHITECTURE "mips"
-#elif defined(__powerpc__) || defined(__powerpc)
+#elif defined(__powerpc__) || defined(__powerpc) || defined(__POWERPC__)
     #define FF_ARCHITECTURE "powerpc"
 #elif defined(__riscv__) || defined(__riscv)
     #define FF_ARCHITECTURE "riscv"
