From b425aae0e3f0c1211a267432018e1cfb58481911 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 Aug 2024 19:58:34 +0800
Subject: [PATCH] disk_bsd: no support for creation time

diff --git src/detection/disk/disk_bsd.c src/detection/disk/disk_bsd.c
index 81be9c83..4c35f50c 100644
--- src/detection/disk/disk_bsd.c
+++ src/detection/disk/disk_bsd.c
@@ -65,6 +65,7 @@ static void detectFsInfo(struct statfs* fs, FFDisk* disk)
 
 #include <sys/attr.h>
 #include <unistd.h>
+#include <AvailabilityMacros.h>
 
 #ifndef MAC_OS_X_VERSION_10_15
     #define MNT_REMOVABLE 0x00000200
@@ -143,9 +144,11 @@ const char* ffDetectDisksImpl(FFDiskOptions* options, FFlist* disks)
         if(fs->f_flags & MNT_RDONLY)
             disk->type |= FF_DISK_VOLUME_TYPE_READONLY_BIT;
 
+#ifdef MAC_OS_X_VERSION_10_15
         struct stat st;
         if(stat(fs->f_mntonname, &st) == 0 && st.st_birthtimespec.tv_sec > 0)
             disk->createTime = (uint64_t)((st.st_birthtimespec.tv_sec * 1000) + (st.st_birthtimespec.tv_nsec / 1000000));
+#endif
     }
 
     return NULL;
