From c8c6959a0143624772b2837e7b1de54f039e5dba Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 27 Aug 2024 16:40:51 +0800
Subject: [PATCH 20/21] swap_fat_arch_64 does not exist in 10.6

---
 src/util/binary_apple.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/util/binary_apple.c b/src/util/binary_apple.c
index 6efeb5d5..c3fa3328 100644
--- a/src/util/binary_apple.c
+++ b/src/util/binary_apple.c
@@ -142,13 +142,12 @@ static const char* dumpFatHeader(FILE *objFile, bool (*cb)(const char *str, uint
         }
         else
         {
-            struct fat_arch_64 arch;
+            struct fat_arch arch;
             if (!readData(objFile, &arch, sizeof(arch), (off_t) (sizeof(header) + i * sizeof(arch))))
                 continue;
 
             if (needSwap)
-                swap_fat_arch_64(&arch, 1, NX_UnknownByteOrder);
-
+                swap_fat_arch(&arch, 1, NX_UnknownByteOrder);
             machHeaderOffset = (off_t)arch.offset;
         }
 
-- 
2.46.0

