--- rnnlmlib.cpp.orig	2013-11-14 02:25:40.000000000 +0900
+++ rnnlmlib.cpp	2014-02-16 21:00:51.000000000 +0900
@@ -1644,7 +1644,7 @@
         
         fprintf(flog, "\niter: %d\n", iter);
         fprintf(flog, "valid log probability: %f\n", logp);
-        fprintf(flog, "PPL net: %f\n", exp10(-logp/(real)wordcn));
+        fprintf(flog, "PPL net: %f\n", pow(10, -logp/(real)wordcn));
         
         fclose(flog);
     
@@ -1791,10 +1791,10 @@
     	    fprintf(flog, "test log probability %f*rnn + %f*other_lm: %f\n", lambda, 1-lambda, log_combine);
 	}
 
-	fprintf(flog, "\nPPL net: %f\n", exp10(-logp/(real)wordcn));
+	fprintf(flog, "\nPPL net: %f\n", pow(10, -logp/(real)wordcn));
 	if (use_lmprob) {
-    	    fprintf(flog, "PPL other: %f\n", exp10(-log_other/(real)wordcn));
-    	    fprintf(flog, "PPL combine: %f\n", exp10(-log_combine/(real)wordcn));
+           fprintf(flog, "PPL other: %f\n", pow(10, -log_other/(real)wordcn));
+           fprintf(flog, "PPL combine: %f\n", pow(10, -log_combine/(real)wordcn));
 	}
     }
     
@@ -1929,10 +1929,10 @@
     	    printf("test log probability %f*rnn + %f*other_lm: %f\n", lambda, 1-lambda, log_combine);
 	}
 
-	printf("\nPPL net: %f\n", exp10(-logp/(real)wordcn));
+	printf("\nPPL net: %f\n", pow(10, -logp/(real)wordcn));
 	if (use_lmprob) {
-    	    printf("PPL other: %f\n", exp10(-log_other/(real)wordcn));
-    	    printf("PPL combine: %f\n", exp10(-log_combine/(real)wordcn));
+           printf("PPL other: %f\n", pow(10, -log_other/(real)wordcn));
+    	    printf("PPL combine: %f\n", pow(10, -log_combine/(real)wordcn));
 	}
     }
 
