--- libgnome-desktop/gnome-rr.c.orig	2009-04-13 15:00:20.000000000 -0700
+++ libgnome-desktop/gnome-rr.c	2009-04-19 17:23:31.000000000 -0700
@@ -28,15 +28,19 @@
 #include <glib/gi18n-lib.h>
 #include "libgnomeui/gnome-rr.h"
 #include <string.h>
+#include <gtk/gtk.h>
+#include <gdk/gdk.h>
+#ifndef GDK_WINDOWING_QUARTZ
 #include <X11/Xlib.h>
 #include <X11/extensions/Xrandr.h>
-#include <gtk/gtk.h>
 #include <gdk/gdkx.h>
 #include <X11/Xatom.h>
+#endif
 
 #include "private.h"
 #include "gnome-rr-private.h"
 
+#ifndef GDK_WINDOWING_QUARTZ
 #define DISPLAY(o) ((o)->info->screen->xdisplay)
 
 struct GnomeRROutput
@@ -1367,3 +1371,4 @@
     g_free (mode->name);
     g_free (mode);
 }
+#endif
