--- ext/vp8/gstvp8utils.h.orig	2015-06-01 07:13:21.000000000 -0700
+++ ext/vp8/gstvp8utils.h	2015-06-01 07:27:15.000000000 -0700
@@ -26,6 +26,19 @@
 G_BEGIN_DECLS
 
 /* Some compatibility defines for older libvpx versions */
+
+/* The following #defines break the build for libvpx 1.40+ because
+ * the deprecated symbols IMG_FMT_* PLANE_* have been removed entirely from libvpx
+ * and the newer VPX_IMG_FMT_* VPX_PLANE_* are now implemented as an enum typdef
+ * rather than a list of #defines as in libvpx 1.30 and earlier.. 
+ *
+ * The result is that VPX_IMG_FMT_* VPX_PLANE_* are not "#defined" and and 
+ * IMG_FMT_* PLANE_* are just gone
+ *
+ */
+
+/*
+
 #ifndef VPX_IMG_FMT_I420
 #define VPX_IMG_FMT_I420 IMG_FMT_I420
 #endif
@@ -42,6 +55,8 @@
 #define VPX_PLANE_V PLANE_V
 #endif
 
+*/ 
+
 const char * gst_vpx_error_name (vpx_codec_err_t status);
 
 G_END_DECLS
