--- setup.py.orig	2010-11-01 10:38:32.000000000 +0100
+++ setup.py	2010-11-01 10:38:44.000000000 +0100
@@ -11,6 +11,6 @@
       license='GPL-3',
       py_modules=['djvubind/__init__', 'djvubind/encode', 'djvubind/ocr', 'djvubind/organizer', 'djvubind/utils'],
       data_files=[('bin', ['bin/djvubind']),
-                  ('/etc/djvubind', ['docs/config']),
+                  ('etc/djvubind', ['docs/config']),
                   ('share/man/man1', ['docs/djvubind.1.gz'])]
 )
