--- ./configure	2024-05-10 20:53:39
+++ ./configure	2024-05-10 20:55:58
@@ -444,7 +444,7 @@
 }
 EOF
 
-	for liblua in lua lua5.4 lua5.3 lua5.2 lua-5.4 lua-5.3 lua-5.2 lua54 lua53 lua52; do
+	for liblua in lua@@LUA_VERSION@@ ; do
 		printf " checking for %s... " "$liblua"
 
 		if test "$have_pkgconfig" = "yes" ; then
@@ -505,7 +505,7 @@
 }
 EOF
 
-	for liblpeg in lpeg lua5.3-lpeg lua5.2-lpeg; do
+	for liblpeg in lua@@LUA_VERSION@@-lpeg lpeg; do
 		printf " checking for static %s... " "$liblpeg"
 
 		if test "$have_pkgconfig" = "yes" ; then
