--- src/configure.ac.orig	2018-06-18 09:57:53.000000000 -0500
+++ src/configure.ac	2022-11-16 05:31:17.000000000 -0600
@@ -86,7 +86,7 @@
   [AC_LANG_SOURCE(
      [[
 #include "ov-scalar.h"
-main()
+int main()
 {
   octave_scalar x (1.234);
   bool y = x.isreal ();
@@ -106,7 +106,7 @@
 #include "ov-scalar.h"
 #include "ovl.h"
 #include "parse.h"
-main()
+int main()
 {
   octave_value_list ovl = octave::feval ("sin", octave_value (1.234));
   return 0;
