--- ./Makefile	2022-04-04 14:39:42.000000000 -0400
+++ ./Makefile	2022-04-04 14:44:45.000000000 -0400
@@ -11,7 +11,7 @@
 			 $(SPECIFIC_CFLAGS) \
 	     -DNODEBUG $(INCLUDE_PATHS)
 
-VERSION=$(shell git describe)
+VERSION=@VERSION@
 ifdef TRAVIS_OS_NAME
   OS=$(TRAVIS_OS_NAME)
 else
@@ -25,14 +25,14 @@
 all: optar unoptar
 
 install:
-	install optar /usr/local/bin/
-	install unoptar /usr/local/bin
-	install pgm2ps /usr/local/bin
+	install optar @DESTROOT@@PREFIX@/bin/
+	install unoptar @DESTROOT@@PREFIX@/bin
+	install pgm2ps @DESTROOT@@PREFIX@/bin
 
 uninstall:
-	rm /usr/local/bin/optar
-	rm /usr/local/bin/unoptar
-	rm /usr/local/bin/pgm2ps
+	rm @DESTROOT@@PREFIX@/bin/optar
+	rm @DESTROOT@@PREFIX@/bin/unoptar
+	rm @DESTROOT@@PREFIX@/bin/pgm2ps
 
 clean:
 	rm -f $(BINARIES) optar-*.tar.gz golay_codes.c *.o
@@ -65,7 +65,7 @@
 	$(CC) $(LDFLAGS) -o $@ $^
 
 unoptar: unoptar.o common.o golay_codes.o parity.o
-	$(CC) -o $@ -L/usr/local/lib $^ -lm -lpng -lz
+	$(CC) -o $@ -L@PREFIX@/lib $^ -lm -lpng -lz
 
 archive: $(ARCHIVE_PATH)
 
