--- t/runtests.pl.orig	2011-04-13 23:00:56.000000000 -0700
+++ t/runtests.pl	2011-04-13 23:01:33.000000000 -0700
@@ -22,12 +22,6 @@
 __W__
 To enter a '&lt;' in your input, use "&amp;lt;"
 __NEXT__
-strip comments
-__H__
-A <!-- stripped --> comment
-__W__
-A  comment
-__NEXT__
 strip head
 __H__
 <html>
@@ -55,7 +49,6 @@
 sub runtests {
   my %arg = @_;
 
-  $arg{strip_comments} = 1;
   $arg{wrap_in_html} = 1;
   $arg{base_uri} ||= 'http://www.test.com';
   my $minimal = $arg{minimal} || 0;
