--- t/03_OO.t.orig	2022-03-17 20:06:37.000000000 -0700
+++ t/03_OO.t	2022-03-17 20:10:18.000000000 -0700
@@ -14,7 +14,8 @@
 
   $ENV{XDG_DATA_DIRS} = '';
   is_deeply( [$conf->xdg_data_dirs()],
-             [ File::Spec->catdir($rootdir, qw/usr local share/),
+             [ File::Spec->catdir($rootdir, qw/@@MP_PREFIX_WORDS@@ share/),
+               File::Spec->catdir($rootdir, qw/usr local share/),
                File::Spec->catdir($rootdir, qw/usr share/)         ],
        'xdg_data_dirs default - OO');
 }
