--- a/src/lib-storage/index/dbox-common/dbox-storage.c.orig	2023-09-14 23:17:46
+++ b/src/lib-storage/index/dbox-common/dbox-storage.c	2023-12-18 20:59:07
@@ -293,8 +293,8 @@ int dbox_mailbox_list_cleanup(struct mail_user *user, 
 		   if the directory exists. In case, get also the ctime */
 		struct stat stats;
 		if (stat(path, &stats) == 0) {
-			last_temp_file_scan = stats.st_atim.tv_sec;
-			change_time = stats.st_ctim.tv_sec;
+			last_temp_file_scan = stats.st_atime;
+			change_time = stats.st_ctime;
 		} else {
 			if (errno != ENOENT)
 				e_error(user->event, "stat(%s) failed: %m", path);
