--- include/mach-o/dyld_priv.h.orig	2024-03-23 12:09:17.000000000 -0700
+++ include/mach-o/dyld_priv.h	2024-03-23 12:11:07.000000000 -0700
@@ -32,6 +32,14 @@
 #include <mach-o/dyld_images.h>
 #include <uuid/uuid.h>
 
+#ifndef TARGET_OS_WATCH
+#define TARGET_OS_WATCH 0
+#endif 
+
+#ifndef TARGET_OS_BRIDGE
+#define TARGET_OS_BRIDGE 0
+#endif 
+
 #if __cplusplus
 extern "C" {
 #endif /* __cplusplus */
