--- Makefile-133	2012-09-04 14:50:18.000000000 -0700
+++ Makefile	2012-09-04 14:49:26.000000000 -0700
@@ -20,17 +20,10 @@ endif
 
 all : ObjectDump dyldinfo ld machocheck rebase unwinddump libprunetrie.a
 
-src/ld/Snapshot.o : src/ld/compile_stubs.h
-src/ld/compile_stubs.h : compile_stubs
-	echo "static const char *compile_stubs = " > $@
-	cat $^ | sed s/\"/\\\\\"/g | sed s/^/\"/ | sed s/$$/\\\\n\"/ >> $@
-	echo ";" >> $@
-
 src/ld/ld.o : src/ld/configure.h
-src/other/ObjectDump.o : src/ld/configure.h
 src/ld/Options.o : src/ld/configure.h
-src/ld/configure.h : src/create_configure
-	DERIVED_SOURCES_DIR=src/ld DERIVED_FILE_DIR=src/ld $^ > $@
+src/ld/configure.h :
+	touch $@
 
 ObjectDump : src/ld/debugline.o
 ObjectDump : src/ld/parsers/macho_relocatable_file.o 
@@ -47,7 +40,6 @@ ld : src/ld/InputFiles.o
 ld : src/ld/Options.o
 ld : src/ld/OutputFile.o
 ld : src/ld/Resolver.o
-ld : src/ld/Snapshot.o
 ld : src/ld/SymbolTable.o
 ld : src/ld/parsers/archive_file.o
 ld : src/ld/parsers/lto_file.o
@@ -62,7 +54,7 @@ ld : src/ld/passes/dylibs.o
 ld : src/ld/passes/got.o
 ld : src/ld/passes/huge.o
 ld : src/ld/passes/objc.o
-ld : src/ld/passes/order.o
+ld : src/ld/passes/order_file.o
 ld : src/ld/passes/tlvp.o
 ld : src/ld/passes/stubs/stubs.o
 	$(CXX) $(LLVM_LDFLAGS) $(LDFLAGS) $(OTHER_LDFLAGS_LD64) -Wl,-exported_symbol,__mh_execute_header $^ -o $@
