--- ld64.xcodeproj/project.pbxproj.orig	2012-04-09 23:17:48.000000000 -0700
+++ ld64.xcodeproj/project.pbxproj	2012-04-09 23:21:25.000000000 -0700
@@ -463,7 +463,6 @@
 				PREBINDING = NO;
 				PRODUCT_NAME = ld64;
 				SECTORDER_FLAGS = "";
-				VERSIONING_SYSTEM = "apple-generic";
 				WARNING_CFLAGS = "-Wall";
 			};
 			name = Debug;
@@ -511,7 +510,6 @@
 				PRODUCT_NAME = ld64;
 				SECTORDER_FLAGS = "";
 				VALID_ARCHS = "i386 ppc";
-				VERSIONING_SYSTEM = "apple-generic";
 				WARNING_CFLAGS = "-Wall";
 			};
 			name = Release;
--- src/ld.cpp.orig	2012-04-09 23:23:58.000000000 -0700
+++ src/ld.cpp	2012-04-09 23:25:34.000000000 -0700
@@ -57,6 +57,8 @@
 
 #include "SectCreate.h"
 
+const double ld64VersionNumber = (double)62.1;
+
 #if 0
 static void dumpAtom(ObjectFile::Atom* atom)
 {
@@ -2629,7 +2632,6 @@ int main(int argc, const char* argv[])
 		ld.link();
 	}
 	catch (const char* msg) {
-		extern const double ld64VersionNumber;
 		if ( archInferred )
 			fprintf(stderr, "ld64-%g failed: %s for inferred architecture %s\n", ld64VersionNumber, msg, archName);
 		else if ( showArch )
--- src/Options.cpp.orig	2012-04-09 23:30:52.000000000 -0700
+++ src/Options.cpp	2012-04-09 23:31:07.000000000 -0700
@@ -31,6 +31,8 @@
 
 #include "Options.h"
 
+const char *ld64VersionString = "@(#)PROGRAM:ld64  PROJECT:ld64-62.1\n";
+
 void throwf(const char* format, ...)
 {
 	va_list	list;
@@ -1456,7 +1458,6 @@ void Options::buildSearchPaths(int argc,
 			addStandardLibraryDirectories = false;
 		else if ( strcmp(argv[i], "-v") == 0 ) {
 			fVerbose = true;
-			extern const char ld64VersionString[];
 			fprintf(stderr, "%s", ld64VersionString);
 			 // if only -v specified, exit cleanly
 			 if ( argc == 2 )
