--- src/ld/ld.cpp.orig	2012-04-21 16:07:59.000000000 -0700
+++ src/ld/ld.cpp	2012-04-21 16:08:44.000000000 -0700
@@ -3206,11 +3206,11 @@ const char* Linker::fileArch(const void*
 	result = mach_o::relocatable::Reader<arm>::fileKind(bytes);
 	if ( result != NULL  )
 		 return result;
-		 
+#if LTO_SUPPORT
 	result = lto::Reader::fileKind(bytes);
 	if ( result != NULL  )
 		 return result;
-	
+#endif	
 	return "unsupported file format";	 
 }
 
--- src/other/ObjectDump.cpp.orig	2012-04-28 11:18:03.000000000 -0700
+++ src/other/ObjectDump.cpp	2012-04-28 11:22:00.000000000 -0700
@@ -30,8 +30,6 @@
 
 #include "MachOReaderRelocatable.hpp"
 
-#define LTO_SUPPORT 1
-
 #if LTO_SUPPORT
 	#include "LTOReader.hpp"
 #endif
