diff --git libtapi-1.30/lib/Core/MachODylibReader.cpp libtapi-1.30/lib/Core/MachODylibReader.cpp
index edaa7c9..c260b61 100644
--- libtapi-1.30/lib/Core/MachODylibReader.cpp
+++ libtapi-1.30/lib/Core/MachODylibReader.cpp
@@ -148,12 +148,12 @@ void load(MachOObjectFile *object, InterfaceFile *file) {
     case MachO::LC_VERSION_MIN_IPHONEOS:
       file->setPlatform(Platform::iOS);
       break;
-    case MachO::LC_VERSION_MIN_WATCHOS:
-      file->setPlatform(Platform::watchOS);
-      break;
-    case MachO::LC_VERSION_MIN_TVOS:
-      file->setPlatform(Platform::tvOS);
-      break;
+ //   case MachO::LC_VERSION_MIN_WATCHOS:
+ //     file->setPlatform(Platform::watchOS);
+ //     break;
+ //   case MachO::LC_VERSION_MIN_TVOS:
+ //     file->setPlatform(Platform::tvOS);
+ //     break;
     default:
       break;
     }
