--- src/ld/passes/bitcode_bundle.cpp.orig	2015-08-25 10:19:19.000000000 -0700
+++ src/ld/passes/bitcode_bundle.cpp	2015-10-20 09:59:42.000000000 -0700
@@ -252,6 +252,9 @@ BitcodeObfuscator::BitcodeObfuscator()
     _lto_hide_symbols = (lto_codegen_func_t) dlsym(RTLD_DEFAULT, "lto_codegen_hide_symbols");
     _lto_write_reverse_map = (lto_codegen_output_t) dlsym(RTLD_DEFAULT, "lto_codegen_write_symbol_reverse_map");
     _lto_reset_context = (lto_codegen_func_t) dlsym(RTLD_DEFAULT, "lto_codegen_reset_context");
+#if LTO_API_VERSION < 12
+    throwf("compile-time libLTO didn't support -bitcode_hide_symbols: %d", LTO_API_VERSION);
+#else
     if ( _lto_hide_symbols == NULL || _lto_write_reverse_map == NULL ||
         _lto_reset_context == NULL || ::lto_api_version() < 14 )
         throwf("loaded libLTO doesn't support -bitcode_hide_symbols: %s", ::lto_get_version());
@@ -259,6 +262,7 @@ BitcodeObfuscator::BitcodeObfuscator()
 #if LTO_API_VERSION >= 14
     lto_codegen_set_should_internalize(_obfuscator, false);
 #endif
+#endif
 }
 
 BitcodeObfuscator::~BitcodeObfuscator()
