--- src/ld/Options.cpp.orig	2015-10-20 09:15:30.000000000 -0700
+++ src/ld/Options.cpp	2015-10-20 09:15:45.000000000 -0700
@@ -5292,7 +5292,6 @@ void Options::checkIllegalOptionCombinat
 		throw "-segment_order can only used used with -preload output";
 
 	// <rdar://problem/17598404> warn if building an embedded iOS dylib for pre-iOS 8
-	// <rdar://problem/18935714> How can we suppress "ld: warning: embedded dylibs/frameworks only run on iOS 8 or later” when building XCTest?
 	if ( (fOutputKind == Options::kDynamicLibrary) && (fIOSVersionMin != ld::iOSVersionUnset) && (fDylibInstallName != NULL) ) {
 		if ( !min_iOS(ld::iOS_8_0) && (fDylibInstallName[0] == '@') && !fEncryptableForceOff )
 			warning("embedded dylibs/frameworks only run on iOS 8 or later");
