--- src/ld/passes/bitcode_bundle.cpp.orig	2015-10-19 17:17:58.000000000 -0700
+++ src/ld/passes/bitcode_bundle.cpp	2016-05-07 01:11:11.000000000 -0700
@@ -277,6 +277,9 @@ BitcodeObfuscator::BitcodeObfuscator()
     _lto_lookup_hidden_name = (lto_codegen_lookup_t) dlsym(RTLD_DEFAULT, "lto_codegen_lookup_hidden_name");
     _lto_get_asm_symbol_num = (lto_module_num_symbols) dlsym(RTLD_DEFAULT, "lto_module_get_num_asm_symbols");
     _lto_get_asm_symbol_name = (lto_module_symbol_name) dlsym(RTLD_DEFAULT, "lto_module_get_asm_symbol_name");
+#if LTO_API_VERSION < 12
+    throwf("compile-time libLTO didn't support -bitcode_hide_symbols: %d", LTO_API_VERSION);
+#else
     if ( _lto_hide_symbols == NULL || _lto_write_reverse_map == NULL ||
         _lto_reset_context == NULL || _lto_lookup_hidden_name == NULL ||
         _lto_get_asm_symbol_num == NULL || _lto_get_asm_symbol_name == NULL || ::lto_api_version() < 14 )
@@ -285,6 +288,7 @@ BitcodeObfuscator::BitcodeObfuscator()
 #if LTO_API_VERSION >= 14
     lto_codegen_set_should_internalize(_obfuscator, false);
 #endif
+#endif
 }
 
 BitcodeObfuscator::~BitcodeObfuscator()
