--- src/other/ObjectDump.cpp.orig	2016-03-09 13:27:56.000000000 -0800
+++ src/other/ObjectDump.cpp	2016-12-18 00:12:47.000000000 -0800
@@ -843,9 +843,6 @@ void dumper::dumpFixup(const ld::Fixup* 
 		case ld::Fixup::kindStoreARM64PCRelToGOT:
 			printf(", then store as 32-bit delta to GOT entry");
 			break;
-		case ld::Fixup::kindStoreARM64PointerToGOT32:
-			printf(", then store as 32-bit pointer to GOT entry");
-			break;
 		case ld::Fixup::kindDtraceExtra:
 			printf("dtrace static probe extra info");
 			break;
