--- a/xtask/src/commands/macos/bundle.rs.orig	2026-08-29 18:35:07
+++ b/xtask/src/commands/macos/bundle.rs	2026-09-01 15:25:00
@@ -64,8 +64,13 @@ fn run_with_channel(
     let sh = Shell::new()?;
     let _repo = sh.push_dir(&root);
     let xcode_env = xcode::env()?;
-    let target_triple = target.map(DistributionTarget::triple);
+    let env_target = env::var("CARGO_BUILD_TARGET")
+        .ok()
+        .filter(|target| !target.is_empty());
+    let target_triple = target
+        .map(DistributionTarget::triple)
+        .or(env_target.as_deref());
     let release_dir = release_dir(&root, target_triple);
 
     println!("==> app icon");
