wrong patch, but a temporary workaround for clang
--- src/foreign/vigra/vigra/cachedfileimage.hxx
+++ src/foreign/vigra/vigra/cachedfileimage.hxx
@@ -555,7 +555,7 @@ template <class T, class Notify>
 class NotifyingDirectionSelector<UnstridedArrayTag, T, Notify>
 {
 #ifdef __GNUC__
-friend class Notify::self_type;
+friend typename Notify::self_type;
 #else
 friend typename Notify::self_type;
 #endif
