--- src/hugin1/stitch_project/CMakeLists.txt.orig	2016-07-20 07:54:34.000000000 -0700
+++ src/hugin1/stitch_project/CMakeLists.txt	2017-07-27 16:44:15.000000000 -0700
@@ -86,6 +86,11 @@
     ADD_EXECUTABLE( ${PROGNAME} MACOSX_BUNDLE hugin_stitch_project.cpp 
                     ${ICON_FILES})
 
+    SET_SOURCE_FILES_PROPERTIES(
+          ${ICON_FILES}
+          PROPERTIES
+          MACOSX_PACKAGE_LOCATION Resources)
+
   ENDIF(MAC_SELF_CONTAINED_BUNDLE)
      
 
--- src/hugin1/ptbatcher/CMakeLists.txt.orig	2017-04-22 10:03:06.000000000 -0700
+++ src/hugin1/ptbatcher/CMakeLists.txt	2017-07-27 17:17:14.000000000 -0700
@@ -135,6 +135,11 @@
     set( ICON_FILES ${MACOSX_BUNDLE_ICON_FILE} ${MACOSX_BUNDLE_TYPE_ICON_FILE} )
     ADD_EXECUTABLE( ${PROGNAME} MACOSX_BUNDLE ${PTBATCHERGUI_SOURCE} ${PTBATCHERGUI_HEADER} ${ICON_FILES} ${OTHER})
 
+    SET_SOURCE_FILES_PROPERTIES(
+          ${ICON_FILES}
+          PROPERTIES
+          MACOSX_PACKAGE_LOCATION Resources)
+
   ENDIF(MAC_SELF_CONTAINED_BUNDLE)
      
 
