--- CMakeLists.txt.orig	2017-07-30 07:51:52.000000000 -0700
+++ CMakeLists.txt	2017-07-30 07:52:24.000000000 -0700
@@ -387,12 +387,12 @@
 ENDIF()
 
 # Check for OpenMP
-IF(APPLE)
+#IF(APPLE)
   #we need a special search routine for mac, see changes in file CMakeModules/FindOpenMPMac.cmake
-  FIND_PACKAGE(OpenMPMac)
-ELSE()
-  FIND_PACKAGE(OpenMP)
-ENDIF()
+  #FIND_PACKAGE(OpenMPMac)
+#ELSE()
+  ___MACPORTS_FIND_OPENMP__
+#ENDIF()
 IF(OPENMP_FOUND)
   MESSAGE(STATUS "Compiler supports OpenMP. Activating support for it.")
   ADD_COMPILE_OPTIONS(${OpenMP_CXX_FLAGS})
