Index: src/hugin_base/algorithms/panorama_makefile/PanoramaMakefileExport.cpp
===================================================================
--- src/hugin_base/algorithms/panorama_makefile/PanoramaMakefileExport.cpp	(revision 2903)
+++ src/hugin_base/algorithms/panorama_makefile/PanoramaMakefileExport.cpp	(working copy)
@@ -196,15 +196,15 @@
     o << endl;
 
     o << "ENBLEND_LDR_COMP=";
-    if (opts.outputImageTypeCompression.size() != 0) {
-        o << "--compression " << opts.outputImageTypeCompression;
-    }
+//    if (opts.outputImageTypeCompression.size() != 0) {
+//        o << "--compression " << opts.outputImageTypeCompression;
+//    }
     o << endl;
 
     o << "ENBLEND_HDR_COMP=";
-    if (opts.outputImageTypeHDRCompression.size() != 0) {
-        o << "--compression " << opts.outputImageTypeHDRCompression;
-    }
+//    if (opts.outputImageTypeHDRCompression.size() != 0) {
+//        o << "--compression " << opts.outputImageTypeHDRCompression;
+//    }
     o << endl;
 
 
