--- configure	2023-12-04 20:28:20.000000000 +0800
+++ configure	2023-12-10 00:54:49.000000000 +0800
@@ -8,28 +8,14 @@
 PKG_DEB_NAME="libjq-dev"
 PKG_RPM_NAME="jq-devel"
 PKG_CSW_NAME="libjq_dev"
-PKG_BREW_NAME="jq"
+PKG_MACPORTS_NAME="jq"
 PKG_TEST_HEADER="<jq.h>"
 PKG_LIBS="-ljq"
 PKG_LIBS_STATIC="-ljq -lonig"
 PKG_CFLAGS=""
 
-# Note that cflags may be empty in case of success
-if [ "$INCLUDE_DIR" ] || [ "$LIB_DIR" ]; then
-  echo "Found INCLUDE_DIR and/or LIB_DIR!"
-  PKG_CFLAGS="-I$INCLUDE_DIR $PKG_CFLAGS"
-  PKG_LIBS="-L$LIB_DIR $PKG_LIBS"
-elif [ `uname` = "Darwin" ]; then
-  brew --version 2>/dev/null
-  if [ $? -eq 0 ]; then
-    BREWDIR=`brew --prefix`
-    PKG_CFLAGS="-I$BREWDIR/opt/$PKG_BREW_NAME/include"
-    PKG_LIBS="-L$BREWDIR/lib $PKG_LIBS"
-  else
-    curl -sfL "https://autobrew.github.io/scripts/jq" > autobrew
-    . ./autobrew
-  fi
-fi
+PKG_CFLAGS="-I@PREFIX@/include $PKG_CFLAGS"
+PKG_LIBS="-L@PREFIX@/lib $PKG_LIBS"
 
 # For debugging
 echo "Using PKG_CFLAGS=$PKG_CFLAGS"
@@ -50,7 +36,7 @@
   echo " * deb: $PKG_DEB_NAME (Debian, Ubuntu)".
   echo " * rpm: $PKG_RPM_NAME (Fedora, EPEL)"
   echo " * csw: $PKG_CSW_NAME (Solaris)"
-  echo " * brew: $PKG_BREW_NAME (OSX)"
+  echo " * port: $PKG_MACPORTS_NAME (macOS)"
   echo "If $PKG_CONFIG_NAME is already installed set INCLUDE_DIR and LIB_DIR manually via:"
   echo "R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=...'"
   echo "-------------------------- [ERROR MESSAGE] ---------------------------"
