--- hints/darwin.sh.orig
+++ hints/darwin.sh
@@ -143,7 +143,7 @@ esac
 
 # Shared library extension is .dylib.
 # Bundle extension is .bundle.
-ld='cc';
+
 so='dylib';
 dlext='bundle';
 usedl='define';
@@ -184,7 +184,7 @@ case "$osvers" in
 *) 
    lddlflags="${ldflags} -bundle -undefined dynamic_lookup"
    case "$ld" in
-       *MACOSX_DEVELOPMENT_TARGET*) ;;
+       *MACOSX_DEPLOYMENT_TARGET*) ;;
        *) ld="env MACOSX_DEPLOYMENT_TARGET=10.3 ${ld}" ;;
    esac
    ;;
