--- dselect/Makefile.in.orig	2010-03-09 07:32:20.000000000 +1100
+++ dselect/Makefile.in	2011-03-11 12:01:25.000000000 +1100
@@ -640,7 +640,7 @@
 
 curkeys.$(OBJEXT): curkeys.h
 curkeys.h: $(srcdir)/keyoverride $(srcdir)/mkcurkeys.pl
-	cursesfile=`echo '#include <ncursesw/curses.h>' | \
+	cursesfile=`echo '#include <curses.h>' | \
 		$(CC) -E - | grep 'curses.h' | head -n 1 | \
 		sed -e 's/^[^"]*"//; s/".*$$//'`; \
 	if [ "$$cursesfile" = "" ]; then \
--- dselect/dselect.h.orig	2010-03-09 07:27:35.000000000 +1100
+++ dselect/dselect.h	2011-03-11 12:01:43.000000000 +1100
@@ -28,7 +28,7 @@
 
 #include <signal.h>
 #undef ERR
-#include <ncursesw/curses.h>
+#include <curses.h>
 
 struct helpmenuentry {
   char key;
--- dselect/main.cc.orig	2010-03-09 07:27:35.000000000 +1100
+++ dselect/main.cc	2011-03-11 12:01:36.000000000 +1100
@@ -37,7 +37,7 @@ extern "C" {
 #include <ctype.h>
 #include <assert.h>
 
-#include <ncursesw/term.h>
+#include <term.h>
 
 extern "C" {
 #include <dpkg.h>
