--- scripts/Dpkg/Source/Archive.pm.orig	2015-11-19 21:10:57.000000000 -0800
+++ scripts/Dpkg/Source/Archive.pm	2015-11-19 21:18:41.000000000 -0800
@@ -48,7 +48,7 @@
     $spawn_opts{from_pipe} = \*$self->{tar_input};
     # Call tar creation process
     $spawn_opts{delete_env} = [ 'TAR_OPTIONS' ];
-    $spawn_opts{exec} = [ 'tar', '-cf', '-', '--format=gnu', '--null',
+    $spawn_opts{exec} = [ 'gnutar', '-cf', '-', '--format=gnu', '--null',
                           '--numeric-owner', '--owner=0', '--group=0',
                           @{$opts{options}}, '-T', '-' ];
     *$self->{pid} = spawn(%spawn_opts);
@@ -90,7 +90,7 @@
     my $self = shift;
 
     close(*$self->{tar_input}) or syserr(g_('close on tar input'));
-    wait_child(*$self->{pid}, cmdline => 'tar -cf -');
+    wait_child(*$self->{pid}, cmdline => 'gnutar -cf -');
     delete *$self->{pid};
     delete *$self->{tar_input};
     delete *$self->{cwd};
@@ -126,7 +126,7 @@
 
     # Call tar extraction process
     $spawn_opts{delete_env} = [ 'TAR_OPTIONS' ];
-    $spawn_opts{exec} = [ 'tar', '-xf', '-', '--no-same-permissions',
+    $spawn_opts{exec} = [ 'gnutar', '-xf', '-', '--no-same-permissions',
                           '--no-same-owner', @{$opts{options}} ];
     spawn(%spawn_opts);
     $self->close();
