--- libgnomecups/gnome-cups-request.c.orig	2008-09-11 10:24:30.000000000 +0200
+++ libgnomecups/gnome-cups-request.c	2008-09-11 10:25:30.000000000 +0200
@@ -399,9 +399,14 @@
 	cups_lang_t *language;
 	ipp_t *request;
 	
-	request = ippNewRequest (operation_id);
-	
+	request = ippNew ();
+	request->request.op.operation_id = operation_id;
+	request->request.op.request_id = 1;
+
 	language = cupsLangDefault ();
+	ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_CHARSET,
+		      "attributes-charset",
+		      NULL, "utf-8");
 	ippAddString (request, IPP_TAG_OPERATION, IPP_TAG_LANGUAGE,
 		      "attributes-natural-language", 
 		      NULL, language->language);
