diff --git a/configure.ac b/configure.ac
index 60c177f..c5d1236 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,4 @@
-AC_INIT([libosmocore],
-	m4_esyscmd([./git-version-gen .tarball-version]),
-	[openbsc@lists.osmocom.org])
+AC_INIT([libosmocore],[m4_esyscmd(./git-version-gen .tarball-version)],[openbsc@lists.osmocom.org])
 
 dnl *This* is the root dir, even if an install-sh exists in ../ or ../../
 AC_CONFIG_AUX_DIR([.])
@@ -318,8 +316,7 @@ dnl Check if the compiler supports runtime SIMD detection
 CHECK_BUILTIN_SUPPORT([__builtin_cpu_supports],
   [Runtime SIMD detection will be disabled])
 
-AC_OUTPUT(
-	libosmocore.pc
+AC_CONFIG_FILES([libosmocore.pc
 	libosmocodec.pc
 	libosmocoding.pc
 	libosmovty.pc
@@ -346,4 +343,5 @@ AC_OUTPUT(
 	Doxyfile.codec
 	Doxyfile.coding
 	Doxyfile.gb
-	Makefile)
+	Makefile])
+AC_OUTPUT
