diff --git a/av1an-core/src/context.rs b/av1an-core/src/context.rs
--- a/av1an-core/src/context.rs
+++ b/av1an-core/src/context.rs
@@ -572,6 +572,7 @@ impl Av1anContext {
                     }
                     command
                         .args(args)
+                        .stdin(Stdio::null())
                         .stdout(Stdio::piped())
                         .stderr(Stdio::piped())
                         .spawn()
diff --git a/av1an-core/src/target_quality.rs b/av1an-core/src/target_quality.rs
--- a/av1an-core/src/target_quality.rs
+++ b/av1an-core/src/target_quality.rs
@@ -555,6 +555,7 @@ impl TargetQuality {
             let mut source = if let [pipe_cmd, args @ ..] = &*source_cmd {
                 std::process::Command::new(pipe_cmd)
                     .args(args)
+                    .stdin(std::process::Stdio::null())
                     .stderr(std::process::Stdio::piped())
                     .stdout(std::process::Stdio::piped())
                     .spawn()
diff --git a/av1an-core/src/metrics/vmaf.rs b/av1an-core/src/metrics/vmaf.rs
--- a/av1an-core/src/metrics/vmaf.rs
+++ b/av1an-core/src/metrics/vmaf.rs
@@ -277,6 +277,7 @@ pub fn calculate(
             source_pipe.args(["-a", &arg]);
         }
         source_pipe.args(args);
+        source_pipe.stdin(Stdio::null());
         source_pipe.stdout(Stdio::piped());
         source_pipe.stderr(Stdio::null());
         source_pipe.spawn()?
@@ -393,6 +394,7 @@ pub fn plot(
             source_pipe.args(["-a", &arg]);
         }
         source_pipe.args(args);
+        source_pipe.stdin(Stdio::null());
         source_pipe.stdout(Stdio::piped());
         source_pipe.stderr(Stdio::null());
         source_pipe.spawn()?
diff --git a/av1an-core/src/metrics/xpsnr.rs b/av1an-core/src/metrics/xpsnr.rs
--- a/av1an-core/src/metrics/xpsnr.rs
+++ b/av1an-core/src/metrics/xpsnr.rs
@@ -73,6 +73,7 @@ pub fn calculate(
             source_pipe.args(["-a", &arg]);
         }
         source_pipe.args(args);
+        source_pipe.stdin(Stdio::null());
         source_pipe.stdout(Stdio::piped());
         source_pipe.stderr(Stdio::null());
         source_pipe.spawn()?
