--- archive-tar-minitar-0.8.gemspec
+++ archive-tar-minitar-0.8.gemspec
@@ -25,10 +25,10 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<minitar>.freeze, ["~> 0.8"])
-    s.add_runtime_dependency(%q<minitar-cli>.freeze, ["~> 0.8"])
+    s.add_runtime_dependency(%q<minitar>.freeze, [">= 0.8"])
+    s.add_runtime_dependency(%q<minitar-cli>.freeze, [">= 0.8"])
   else
-    s.add_dependency(%q<minitar>.freeze, ["~> 0.8"])
-    s.add_dependency(%q<minitar-cli>.freeze, ["~> 0.8"])
+    s.add_dependency(%q<minitar>.freeze, [">= 0.8"])
+    s.add_dependency(%q<minitar-cli>.freeze, [">= 0.8"])
   end
 end
