--- a/Makefile.orig	2018-02-08 16:14:10.000000000 +0100
+++ b/Makefile	2018-09-14 22:39:48.000000000 +0200
@@ -33,11 +33,7 @@
 MAN3DIR=$(MANDIR)/man3
 
 ifneq ($(OS),Windows_NT)
-  TERMINFO_DIRS="$(shell ncursesw6-config --terminfo-dirs 2>/dev/null || \
-                         ncurses6-config  --terminfo-dirs 2>/dev/null || \
-                         ncursesw5-config --terminfo-dirs 2>/dev/null || \
-                         ncurses5-config  --terminfo-dirs 2>/dev/null || \
-                         echo "/etc/terminfo:/lib/terminfo:/usr/share/terminfo:/usr/lib/terminfo:/usr/local/share/terminfo:/usr/local/lib/terminfo")"
+  TERMINFO_DIRS="/etc/terminfo:$(PREFIX)/share/terminfo:/usr/local/share/terminfo:/usr/share/terminfo"
 else
   TERMINFO_DIRS=""
 endif
