--- autogen.sh.orig	2016-04-07 02:29:15.000000000 -0500
+++ autogen.sh	2026-03-20 22:46:01.000000000 -0500
@@ -15,12 +15,15 @@
 	chmod 755 ${i}
 done
 
-libtoolize --force --copy
-aclocal
-autoheader
+touch config.rpath
+
+prefix=$(dirname $(dirname $(command -v autopoint)))
+glibtoolize --force --copy
+aclocal --force -I"$prefix/share/gettext/m4"
+autoheader --force
 automake --verbose --copy --force --add-missing
 intltoolize --force --copy
-autoconf
+autoconf --force
 
 # For the Debian build
 test -d debian && {
