--- setup.py.orig	2022-04-03 11:08:31.000000000 -0400
+++ setup.py	2022-04-03 11:08:41.000000000 -0400
@@ -42,7 +42,7 @@
     'ipykernel<5; python_version<"3"',
     'ipykernel>=6.9.2; python_version>="3"',
     'ipython<6; python_version<"3"',
-    'ipython>=7.31.1,<8; python_version>="3"',
+    'ipython>=7.31.1; python_version>="3"',
     'jupyter-client>=5.3.4,<6; python_version<"3"',
     'jupyter-client>=7.1.0; python_version>="3"',
     'pyzmq>=17,<20; python_version<"3"',
