--- src/aaa_lib/Makefile.in.orig	2021-05-27 16:30:23.000000000 -0400
+++ src/aaa_lib/Makefile.in	2021-05-27 16:30:53.000000000 -0400
@@ -310,7 +310,7 @@
 @BUILD_EFENCE_TRUE@LIBEFENCE = libefence.la
 lib_LTLIBRARIES = libmaloc.la ${LIBEFENCE}
 libmaloc_la_SOURCES = 
-libmaloc_la_LIBADD = ${BASE_LIBS} ${VSYS_LIBS} ${VSH_LIBS} ${PSH_LIBS}
+libmaloc_la_LIBADD = ${BASE_LIBS} ${VSYS_LIBS} ${VSH_LIBS} ${PSH_LIBS} $(rl_lib) $(math_lib)
 libmaloc_la_LDFLAGS = -version-info ${FETK_VERSION}
 libefence_la_SOURCES = 
 libefence_la_LIBADD = ${EFENCE_LIBS}
