--- t/100_basic.t.orig	2011-04-06 20:08:51.000000000 -0700
+++ t/100_basic.t	2011-04-06 20:09:31.000000000 -0700
@@ -35,7 +35,7 @@
     
     ok(!$sock->getsockopt(ZMQ_RCVMORE), "no ZMQ_RCVMORE set");
     ok($sock->getsockopt(ZMQ_AFFINITY) == 0, "no ZMQ_AFFINITY");
-    ok($sock->getsockopt(ZMQ_RATE) == 100, "ZMQ_RATE is at default 100");
+    ok($sock->getsockopt(ZMQ_RATE) == 40_000, "ZMQ_RATE is at default 40 000");
 
     my $msg = $sock->recv();
     ok(defined $msg, "received defined msg");
