--- src/Makefile.orig	2020-05-16 05:34:58.000000000 -0500
+++ src/Makefile	2020-05-24 06:20:27.000000000 -0500
@@ -1,17 +1,18 @@
 # -*- coding: utf-8 -*-
 #Compiler
-MKOCTFILE?=mkoctfile
+#MKOCTFILE?=mkoctfile
 #Common warning flags for C and C++
-FLAGSCOMW=-Wall -Wextra -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings
+#FLAGSCOMW=-Wall -Wextra -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings
 #Common optimization flags for C and C++
-FLAGSCOMO=-O2 -funroll-loops -fno-common -fshort-enums
+#FLAGSCOMO=-O2 -funroll-loops -fno-common -fshort-enums
 #Flags for C
-CFLAGS=-std=c99 -pedantic $(FLAGSCOMW) -Wconversion -Wmissing-prototypes
-CFLAGS+=-Wstrict-prototypes -Wnested-externs $(FLAGSCOMO)
+CFLAGS=-std=c99
+#CFLAGS=-std=c99 -pedantic $(FLAGSCOMW) -Wconversion -Wmissing-prototypes
+#CFLAGS+=-Wstrict-prototypes -Wnested-externs $(FLAGSCOMO)
 #Flags for C++
-CXXFLAGS=$(FLAGSCOMW) $(FLAGSCOMO)
+#CXXFLAGS=$(FLAGSCOMW) $(FLAGSCOMO)
 #Export flags for compilers and linker
-export CFLAGS CXXFLAGS
+#export CFLAGS CXXFLAGS
 
 .PHONY: all
 all: compile
