--- CMakeLists.txt.orig	2016-11-29 02:14:16.000000000 -0600
+++ CMakeLists.txt	2017-01-05 13:00:17.000000000 -0600
@@ -40,8 +40,6 @@
 endif()
 
 if (${CMAKE_SYSTEM} MATCHES "Darwin")
-    set(CMAKE_CXX_FLAGS "-Wno-deprecated-register -mmacosx-version-min=10.5 ${CMAKE_CXX_FLAGS}")
-    set(CMAKE_C_FLAGS "-Wno-deprecated-register -mmacosx-version-min=10.5 ${CMAKE_C_FLAGS}")
     find_library(CORESERVICES_LIB CoreServices)
     set(EXTRA_LIBS ${CORESERVICES_LIB})
 endif()
