--- a/qtools/qglobal.h.orig	2011-06-17 00:14:03.000000000 -0700
+++ a/qtools/qglobal.h	2011-06-17 00:19:32.000000000 -0700
@@ -78,19 +78,22 @@
 #  define MAC_OS_X_VERSION_MIN_REQUIRED MAC_OS_X_VERSION_10_3
 #  include <AvailabilityMacros.h>
 #  if !defined(MAC_OS_X_VERSION_10_3)
-#     define MAC_OS_X_VERSION_10_3 MAC_OS_X_VERSION_10_2 + 1
+#     define MAC_OS_X_VERSION_10_3 MAC_OS_X_VERSION_10_2 + 10
 #  endif
 #  if !defined(MAC_OS_X_VERSION_10_4)
-#       define MAC_OS_X_VERSION_10_4 MAC_OS_X_VERSION_10_3 + 1
+#       define MAC_OS_X_VERSION_10_4 MAC_OS_X_VERSION_10_3 + 10
 #  endif
 #  if !defined(MAC_OS_X_VERSION_10_5)
-#       define MAC_OS_X_VERSION_10_5 MAC_OS_X_VERSION_10_4 + 1
+#       define MAC_OS_X_VERSION_10_5 MAC_OS_X_VERSION_10_4 + 10
 #  endif
 #  if !defined(MAC_OS_X_VERSION_10_6)
-#       define MAC_OS_X_VERSION_10_6 MAC_OS_X_VERSION_10_5 + 1
+#       define MAC_OS_X_VERSION_10_6 MAC_OS_X_VERSION_10_5 + 10
 #  endif
-#  if (MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_6)
-#    error "This version of Mac OS X is unsupported"
+#  if !defined(MAC_OS_X_VERSION_10_7)
+#       define MAC_OS_X_VERSION_10_7 MAC_OS_X_VERSION_10_6 + 10
+#  endif
+#  if (MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_7)
+#    warning "This version of Mac OS X is unsupported"
 #  endif
 #elif defined(MSDOS) || defined(_MSDOS) || defined(__MSDOS__)
 #define _OS_MSDOS_
