--- configure.orig	2012-12-27 06:24:51.000000000 -0500
+++ configure	2013-03-07 16:41:38.000000000 -0500
@@ -379,27 +379,6 @@
 echo "using $f_insttool";
 
 
-# - check for dot ------------------------------------------------------------
-
-echo -n "  Checking for dot (part of GraphViz)... "
-if test "$f_dot" = NO; then
-  dot_dirs="$bin_dirs /usr/bin /usr/local/bin /bin /sbin"
-  dot_prog=NO
-  for j in $dot_dirs; do
-    if test -x "$j/dot"; then
-      dot_prog="$j/dot"
-      break 2
-    fi
-  done
-  f_dot="$dot_prog" 
-fi
-
-if test "$f_dot" = NO; then
-  echo "not found!";
-else
-  echo "using $f_dot"
-fi
-
 # - check for perl ------------------------------------------------------------
 
 echo -n "  Checking for perl... "
