--- examples/CMakeLists.txt.orig	2015-06-20 16:50:23.000000000 -0500
+++ examples/CMakeLists.txt	2015-07-04 09:59:18.000000000 -0500
@@ -10,11 +10,9 @@
 
 # copy all example files
 add_custom_target(copy_examples)
-foreach(example_file ${EXAMPLE_FILES})
-  add_custom_command(TARGET copy_examples PRE_BUILD
+add_custom_command(TARGET copy_examples PRE_BUILD
                      COMMAND ${CMAKE_COMMAND} -E
-                         copy ${example_file} ${PROJECT_BINARY_DIR}/examples)
-endforeach()
+                         copy_directory . ${PROJECT_BINARY_DIR}/examples)
 
 add_custom_target(examples
 	DEPENDS doxygen copy_examples
