--- nwnnsscomp/python/Makefile.in.orig	2006-02-01 04:08:31.000000000 +1100
+++ nwnnsscomp/python/Makefile.in	2014-12-17 15:53:57.000000000 +1100
@@ -302,10 +302,10 @@
 	pdf-am ps ps-am uninstall uninstall-am uninstall-info-am
 
 all-local: setup.py compiler.cpp __init__.py
-	$(PYTHON) setup.py build
+	$(PYTHON) setup.py --no-user-cfg build
 
 install-exec-local: 
-	$(PYTHON) setup.py install --prefix=${DESTDIR}${prefix}
+	$(PYTHON) setup.py --no-user-cfg install --root=$(DESTDIR)
 
 distclean: clean
 
