--- _MathLib/Makefile.in.orig	2006-01-31 11:39:10.000000000 +1100
+++ _MathLib/Makefile.in	2014-12-17 14:02:09.000000000 +1100
@@ -193,7 +193,6 @@
 	Vector4.h
 
 libmath_a_LIBADD = @LIBOBJS@
-AM_CXXFLAGS = -fno-default-inline
 EXTRA_DIST = \
 	_MathLib.vcproj
 
--- _NscLib/Makefile.in.orig	2006-01-31 11:39:10.000000000 +1100
+++ _NscLib/Makefile.in	2014-12-17 14:04:25.000000000 +1100
@@ -178,7 +178,6 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CXXFLAGS = -fno-default-inline
 AM_YFLAGS = --defines=NscParser.h --skeleton=lalr1.cc
 noinst_LIBRARIES = libnsc.a
 libnsc_a_SOURCES = \
--- _NwnLib/Makefile.in.orig	2006-01-31 11:39:10.000000000 +1100
+++ _NwnLib/Makefile.in	2014-12-17 14:03:39.000000000 +1100
@@ -212,7 +212,6 @@
 	NwnTexturePlt.h
 
 libnwn_a_LIBADD = @LIBOBJS@
-AM_CXXFLAGS = -fno-default-inline
 EXTRA_DIST = \
 	_NwnLib.vcproj
 
--- nwnmdlcomp/Makefile.in.orig	2006-01-31 11:39:10.000000000 +1100
+++ nwnmdlcomp/Makefile.in	2014-12-17 15:18:37.000000000 +1100
@@ -172,7 +172,6 @@
 target_alias = @target_alias@
 nwnmdlcomp_SOURCES = nwnmdlcomp.cpp
 nwnmdlcomp_LDADD = ../_NmcLib/libnmc.a ../_NwnLib/libnwn.a ../_MathLib/libmath.a
-AM_CXXFLAGS = -fno-default-inline
 EXTRA_DIST = \
 		nwnmdlcomp.vcproj
 
--- nwnnsscomp/Makefile.in.orig	2006-01-31 11:39:11.000000000 +1100
+++ nwnnsscomp/Makefile.in	2014-12-17 15:17:23.000000000 +1100
@@ -179,7 +179,6 @@
 SUBDIRS = python
 nwnnsscomp_SOURCES = nwnnsscomp.cpp
 nwnnsscomp_LDADD = ../_NwnLib/libnwn.a ../_NscLib/libnsc.a
-AM_CXXFLAGS = -fno-default-inline
 EXTRA_DIST = \
 	nwnnsscomp.vcproj
 
