--- /Makefile.in.orig	2014-10-04 19:23:05.000000000 -0400
+++ Makefile.in	2014-10-04 19:22:31.000000000 -0400
@@ -2566,7 +2566,7 @@
 # Use 'ginstall' in the definition of PROGRAMS and in dependencies to avoid
 # confusion with the 'install' target.  The install rule transforms 'ginstall'
 # to install before applying any user-specified name transformations.
-transform = s/ginstall/install/; $(program_transform_name)
+transform = s/ginstall/install/; /libstdbuf/!$(program_transform_name)
 ACLOCAL = @ACLOCAL@
 ALLOCA = @ALLOCA@
 ALLOCA_H = @ALLOCA_H@
