--- configure	2024-10-03 22:13:06.000000000 +0800
+++ configure	2024-11-04 02:41:05.000000000 +0800
@@ -8,7 +8,7 @@
 PKG_CONFIG_NAME="protobuf"
 PKG_DEB_NAME="libprotobuf-dev"
 PKG_RPM_NAME="protobuf-devel"
-PKG_BREW_NAME="protobuf"
+PKG_MP_NAME="protobuf"
 PKG_TEST_HEADER="<google/protobuf/message.h>"
 PKG_LIBS="-lprotobuf"
 PKG_CFLAGS=""
@@ -29,17 +29,6 @@
   echo "Found pkg-config cflags and libs!"
   PKG_CFLAGS=${PKGCONFIG_CFLAGS}
   PKG_LIBS=${PKGCONFIG_LIBS}
-elif [ `uname` = "Darwin" ]; then
-  test ! "$CI" && brew --version 2>/dev/null
-  if [ $? -eq 0 ]; then
-    BREWDIR=`brew --prefix`
-    PATH="$BREWDIR/bin:$PATH"
-    PKG_CFLAGS="-I$BREWDIR/include"
-    PKG_LIBS="-L$BREWDIR/lib $PKG_LIBS"
-  else
-    curl -sfL "https://autobrew.github.io/scripts/protobuf" > autobrew
-    . ./autobrew
-  fi
 fi
 
 # For debugging
@@ -69,7 +58,7 @@
   echo "Configuration failed to find $PKG_CONFIG_NAME. Try installing:"
   echo " * deb: $PKG_DEB_NAME (Debian, Ubuntu, etc)"
   echo " * rpm: $PKG_RPM_NAME (Fedora, EPEL)"
-  echo " * brew: $PKG_BREW_NAME (OSX)"
+  echo " * port: $PKG_MP_NAME (OSX)"
   echo "If $PKG_CONFIG_NAME is already installed, check that 'pkg-config' is in your"
   echo "PATH and PKG_CONFIG_PATH contains a $PKG_CONFIG_NAME.pc file. If pkg-config"
   echo "is unavailable you can set INCLUDE_DIR and LIB_DIR manually via:"
