--- src/c/Makefile.in	2008-09-24 20:40:10.000000000 +0200
+++ src/c/Makefile.in	2008-09-24 20:41:41.000000000 +0200
@@ -75,7 +75,7 @@
 
 $(DPP): $(srcdir)/dpp.c $(srcdir)/symbols_list2.h
 	if test -f ../CROSS-DPP; then touch dpp; else \
-	$(TRUE_CC) @CFLAGS@ @ECL_CFLAGS@ -I$(srcdir) -I@true_builddir@ -I./ $(DEFS) $(srcdir)/dpp.c  -o $@ ; \
+	$(TRUE_CC) @CPPFLAGS@ @CFLAGS@ @ECL_CFLAGS@ -I$(srcdir) -I@true_builddir@ -I./ $(DEFS) $(srcdir)/dpp.c  -o $@ ; \
 	fi
 $(srcdir)/symbols_list2.h: $(srcdir)/symbols_list.h Makefile
 	cat $(srcdir)/symbols_list.h | \
