--- src/Makefile.orig	2015-06-16 08:23:41.000000000 -0500
+++ src/Makefile	2018-03-27 09:33:06.000000000 -0500
@@ -1,19 +1,20 @@
 # -*- coding: utf-8 -*-
 #Compiler
-MKOCTFILE=mkoctfile
+#MKOCTFILE=mkoctfile
 #Common warning flags for C and C++
-FLAGSCOMW=-Wall -Wextra -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings
+#FLAGSCOMW=-Wall -Wextra -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings
 #Common optimization flags for C and C++
-FLAGSCOMO=-O2 -funroll-loops -fno-common -fshort-enums
+#FLAGSCOMO=-O2 -funroll-loops -fno-common -fshort-enums
 #Flags for C
-CFLAGS=-std=c99 -pedantic $(FLAGSCOMW) -Wconversion -Wmissing-prototypes
-CFLAGS+=-Wstrict-prototypes -Wnested-externs $(FLAGSCOMO)
+#CFLAGS=-std=c99 -pedantic $(FLAGSCOMW) -Wconversion -Wmissing-prototypes
+#CFLAGS+=-Wstrict-prototypes -Wnested-externs $(FLAGSCOMO)
 #Flags for C++
-CXXFLAGS=$(FLAGSCOMW) $(FLAGSCOMO)
+#CXXFLAGS=$(FLAGSCOMW) $(FLAGSCOMO)
 #Flags for the linker
-LDFLAGS=-lproj
+LDFLAGS+=-L%PREFIX%/lib/proj49/lib -lproj
 #Export flags for compilers and linker
-export CFLAGS CXXFLAGS
+#export CFLAGS CXXFLAGS
+CPPFLAGS+=-I%PREFIX%/lib/proj49/include
 
 .PHONY: all
 all: compile
