--- Makefile_libglom.am
+++ Makefile_libglom.am
@@ -97,7 +97,7 @@ pymod_ldflags = -module -avoid-version -no-undefined -shrext .pyd
 else
 pymod_ldflags = -module -avoid-version -no-undefined
 endif
-glom_pyexport = -export-symbols-regex '^_*initglom'
+glom_pyexport = -export-symbols-regex 'glom_@GLOM_ABI_VERSION_UNDERLINED@'
 glom_python_embed_python_module_glom_@GLOM_ABI_VERSION_UNDERLINED@_la_LDFLAGS = $(pymod_ldflags) $(glom_pyexport)
 glom_python_embed_python_module_glom_@GLOM_ABI_VERSION_UNDERLINED@_la_CPPFLAGS = $(glom_includes) $(LIBGLOM_CFLAGS) $(boost_python_cflags)
 
