--- glom/frame_glom.cc.orig	2013-04-16 00:35:35.000000000 -0700
+++ glom/frame_glom.cc	2013-04-16 00:35:55.000000000 -0700
@@ -1576,7 +1576,7 @@
   //Get the fields information from the database:
   DbUtils::type_vec_fields fieldsDatabase = DbUtils::get_fields_for_table_from_database(m_table_name);
 
-  Document* pDoc = dynamic_cast<const Document*>(get_document());
+  Document* pDoc = dynamic_cast<Document*>(get_document());
   if(pDoc)
   {
     bool document_must_be_updated = false;
