--- beanstalkd-0.11.orig/Makefile	2008-04-17 00:18:02.000000000 +0100
+++ beanstalkd-0.11/Makefile	2008-04-17 16:15:53.000000000 +0100
@@ -1,12 +1,14 @@
 program := beanstalkd
-export CFLAGS := $(CFLAGS) -Wall -Werror
-export LDFLAGS := $(LDFLAGS) -levent
+export CPPFLAGS := $(CPPFLAGS) -I__PREFIX__/include
+export CFLAGS := $(CFLAGS) -Wall -Werror -I__PREFIX__/include
+export LDFLAGS := $(LDFLAGS) -levent -L__PREFIX__/lib
 
 sources := $(shell ls *.c | fgrep -v $(program))
 objects := $(sources:.c=.o)
 tests := $(sources:%=tests/test_%)
 
-all: export CFLAGS := $(CFLAGS) -O2
+all: export CFLAGS := $(CFLAGS) -O2 -I__PREFIX__/include
+all: export LDFLAGS := $(LDFLAGS) -levent -L__PREFIX__/lib
 all: $(program)
 
 debug: export CFLAGS := $(CFLAGS) -g -pg -DDEBUG
