diff -u -r beanstalkd-0.9/prot.c.orig beanstalkd-0.9/prot.c
--- beanstalkd-0.9/prot.c.orig	2008-03-21 00:58:46.000000000 +0000
+++ beanstalkd-0.9/prot.c	2008-03-21 00:59:03.000000000 +0000
@@ -738,7 +738,7 @@
             list_watched_tubes_ct,
             timeout_ct,
             global_stat.total_jobs_ct,
-            tubes.used,
+            (unsigned int)tubes.used,
             count_cur_conns(),
             count_cur_producers(),
             count_cur_workers(),

