--- pyproject.toml.orig	2025-10-17 10:47:42
+++ pyproject.toml	2025-10-17 10:48:37
@@ -38,7 +38,7 @@
     "Topic :: Utilities",
 ]
 requires-python = ">=3.9, <4"
-dependencies = ["h5py >= 3.6", "NumPy >= 1.22", "QtPy >= 1.9", "requests", "tomli"]
+dependencies = ["h5py >= 3.6", "NumPy >= 1.22", "QtPy >= 1.9", "requests", "tomli; python_version < '3.11'"]
 dynamic = ["version"]
 
 [project.gui-scripts]
