--- Imakefile.orig	2008-03-23 06:23:09.000000000 +1100
+++ Imakefile	2009-11-25 14:27:47.000000000 +1100
@@ -141,21 +141,14 @@
 #endif
 #endif
 
-# Definitions for CBFLIB
-#
-CBFLIB_VERSION = 0.7.9
-CBFLIB_TARBALL = CBFlib_$(CBFLIB_VERSION).tar.gz
-CBFLIB_TARBALLDIR = ../
-CBFLIB_TARBALL_URL = http://www.bernstein-plus-sons.com/software/$(CBFLIB_TARBALL)
-CBFLIBDIR = $(CBFLIB_TARBALLDIR)/CBFlib_$(CBFLIB_VERSION)
 
 # RasMol's on-line help direcory
 #   e.g. /usr/local/lib/rasmol/rasmol.hlp
-RASMOLDIR = $(USRLIBDIR)/rasmol/
+RASMOLDIR = @PREFIX@/share/rasmol/
 
 
 DEFINES = $(GTKCFLAGS) $(GUIDEF) -DRASMOLDIR=\"$(RASMOLDIR)\" $(DEPTHDEF) $(LOCALEDEF) \
-          -D_USE_RASMOL_H_ $(GUIOPTION) -I$(CBFLIBDIR)/include
+          -D_USE_RASMOL_H_ $(GUIOPTION)
 
 SRCS = rasmol.c rasmol.h molecule.c molecule.h abstree.c abstree.h \
        cmndline.c cmndline.h command.c command.h tokens.h transfor.c \
@@ -177,8 +170,7 @@
 # Additional RS6000 AIX MITSHM Library
 # LDLIBS = -lm -lXi -lXextSam $(XLIB)
 
-LDLIBS = -lm -lXi $(XLIB) $(EXTRA_LIBRARIES) $(XFORMSLIB) $(GTKLIBS) \
-			-L$(CBFLIBDIR)/lib -lcbf
+LDLIBS = -lm -lXi $(XLIB) $(EXTRA_LIBRARIES) $(XFORMSLIB) $(GTKLIBS)
 
 
 
@@ -232,16 +224,6 @@
 ../doc/rasmol.1:	../doc/rasmol.1.gz
 		gunzip  <../doc/rasmol.1.gz > ../doc/rasmol.1
 		
-$(CBFLIBDIR):
-		(cd $(CBFLIB_TARBALLDIR);\
-		  wget $(CBFLIB_TARBALL_URL);\
-		  gunzip < $(CBFLIB_TARBALL) | tar xvf -)
-		  
-$(CBFLIBDIR)/lib/libcbf.a: $(CBFLIBDIR)
-		(cd $(CBFLIBDIR); make lib/libcbf.a)
-		  
-
-$(OBJS):  $(CBFLIBDIR)/lib/libcbf.a
 
 gtkwin.c: gtkui.h
 
