--- rasmol.h.orig	2008-03-23 06:23:10.000000000 +1100
+++ rasmol.h	2009-11-25 13:26:27.000000000 +1100
@@ -83,17 +83,17 @@
 /* #define X11WIN       */    /* X11WIN or GTKWIN defined in Imakefile */
 #define UNIX
 
-/* #define DIALBOX      */
+#define DIALBOX
 #ifndef GTKWIN
 #define SOCKETS
 #endif
 #define TERMIOS
-#define PROFILE
+/*#define PROFILE*/
 #define MITSHM
 
-/* #define HAVEZLIB     */
-/* #define HAVELIBJPEG  */
-/* #define HAVELIBPNG   */
+#define HAVEZLIB
+#define HAVELIBJPEG
+#define HAVELIBPNG
 
 /* #define HPUX_LEGACY  */     /* define this for old-style hpux select */
 
@@ -109,7 +109,7 @@
 /* #define XPROCARCH    */
 #define USE_UNAME
 
-#define USE_CBFLIB
+/* #define USE_CBFLIB */
 
 
 /*========================*/
