--- meson.build.orig	2022-01-14 08:05:54.000000000 -0600
+++ meson.build	2022-01-21 09:14:04.000000000 -0600
@@ -787,7 +787,7 @@
 ]
 
 required_programs_groups = [
-  { 'name': 'rpcgen', 'prog': [ 'rpcgen', 'portable-rpcgen' ] },
+  { 'name': 'rpcgen', 'prog': [ 'rpcgen-mt', 'rpcgen', 'portable-rpcgen' ] },
   { 'name': 'rst2html5', 'prog': [ 'rst2html5', 'rst2html5.py', 'rst2html5-3' ] },
   { 'name': 'rst2man', 'prog': [ 'rst2man', 'rst2man.py', 'rst2man-3' ] },
 ]
