https://bugzilla.redhat.com/show_bug.cgi?id=1071777
--- src/Makefile.in.orig	2014-03-07 08:08:47.000000000 -0800
+++ src/Makefile.in	2014-03-07 08:10:53.000000000 -0800
@@ -288,9 +288,9 @@
 @WITH_LIBVIRTD_TRUE@am__append_159 = lockd.la
 @WITH_LIBVIRTD_TRUE@am__append_160 = locking/libvirt_lockd.aug \
 @WITH_LIBVIRTD_TRUE@	locking/virtlockd.aug
-@WITH_LIBVIRTD_TRUE@am__append_161 = test_libvirt_lockd.aug \
-@WITH_LIBVIRTD_TRUE@	test_virtlockd.aug
-@WITH_LIBVIRTD_TRUE@am__append_162 = test_libvirt_lockd.aug
+@WITH_LIBVIRTD_TRUE@@WITH_QEMU_TRUE@am__append_161 = test_libvirt_lockd.aug \
+@WITH_LIBVIRTD_TRUE@@WITH_QEMU_TRUE@	test_virtlockd.aug
+@WITH_LIBVIRTD_TRUE@@WITH_QEMU_TRUE@am__append_162 = test_libvirt_lockd.aug
 @WITH_DTRACE_PROBES_TRUE@@WITH_LIBVIRTD_TRUE@am__append_163 = libvirt_probes.lo
 @WITH_LIBVIRTD_TRUE@@WITH_QEMU_TRUE@am__append_164 = locking/qemu-lockd.conf
 @WITH_LIBVIRTD_TRUE@@WITH_QEMU_TRUE@am__append_165 = locking/qemu-lockd.conf
@@ -9678,6 +9678,8 @@
 @WITH_QEMU_TRUE@		locking/qemu-lockd.conf $(AUG_GENTEST)
 @WITH_QEMU_TRUE@	$(AM_V_GEN)$(AUG_GENTEST) locking/qemu-lockd.conf $< $@
 
+@WITH_QEMU_FALSE@test_libvirt_lockd.aug:
+
 test_virtlockd.aug: locking/test_virtlockd.aug.in \
 		locking/virtlockd.conf $(AUG_GENTEST)
 	$(AM_V_GEN)$(AUG_GENTEST) $(srcdir)/locking/virtlockd.conf $< $@

