--- src/util/virfile.h.orig	2012-02-13 07:06:23.000000000 +0000
+++ src/util/virfile.h	2012-03-06 19:10:43.000000000 +0000
@@ -58,10 +58,10 @@
 
 int virFileDirectFdFlag(void);
 
-enum {
+enum virFileWrapperFdFlags {
     VIR_FILE_WRAPPER_BYPASS_CACHE   = (1 << 0),
     VIR_FILE_WRAPPER_NON_BLOCKING   = (1 << 1),
-} virFileWrapperFdFlags;
+};
 
 virFileWrapperFdPtr virFileWrapperFdNew(int *fd,
                                         const char *name,
