--- configure.orig	2013-05-15 11:40:32.000000000 -0700
+++ configure	2013-05-15 11:41:24.000000000 -0700
@@ -2659,7 +2659,7 @@
 GTK_REQUIRED=2.22.0
 KEYRING_REQUIRED=2.26.0
 LIBCANBERRA_GTK_REQUIRED=0.4
-LIBNOTIFY_REQUIRED=0.7.0
+LIBNOTIFY_REQUIRED=0.5.1
 TELEPATHY_FARSIGHT_REQUIRED=0.0.14
 TELEPATHY_GLIB_REQUIRED=0.14.1
 TELEPATHY_LOGGER=0.2.0
--- src/empathy-call-observer.c	2013-05-15 11:28:28.000000000 -0700
+++ src/empathy-call-observer.c	2013-05-15 11:30:55.000000000 -0700
@@ -135,7 +135,7 @@
       _("%s just tried to call you, but you were in another call."),
       tp_contact_get_alias (contact));
 
-  notification = notify_notification_new (summary, body, NULL);
+  notification = notify_notification_new (summary, body, NULL, NULL);
 
   emp_contact = empathy_contact_dup_from_tp_contact (contact);
   pixbuf = empathy_notify_manager_get_pixbuf_for_notification (
--- src/empathy-chat-window.c	2013-05-15 11:28:28.000000000 -0700
+++ src/empathy-chat-window.c	2013-05-15 11:33:37.000000000 -0700
@@ -1325,7 +1325,7 @@
 		   to an existing notification with the same title.
 		   In this way the previous message will not be lost: the new
 		   message will appear below it, in the same notification */
-		notification = notify_notification_new (header, escaped, NULL);
+		notification = notify_notification_new (header, escaped, NULL, NULL);
 
 		if (priv->notification == NULL) {
 			priv->notification = notification;
--- src/empathy-status-icon.c	2013-05-15 11:28:28.000000000 -0700
+++ src/empathy-status-icon.c	2013-05-15 11:36:27.000000000 -0700
@@ -241,7 +241,7 @@
 			   In this way the previous message will not be lost: the new
 			   message will appear below it, in the same notification */
 			notification = notify_notification_new
-				(priv->event->header, message_esc, NULL);
+				(priv->event->header, message_esc, NULL, NULL);
 
 			if (priv->notification == NULL) {
 				priv->notification = notification;
