--- Makefile.in.orig	2023-02-09 12:53:32.000000000 +0800
+++ Makefile.in	2023-04-18 09:49:45.000000000 +0800
@@ -10,10 +10,10 @@
 GCOPTIONS = @GCOPTIONS@
 GCLIB = @GCLIB@
 GCPPLIB = @GCPPLIB@
-GCLIBS = $(GCPPLIB) $(GCLIB)
+GCLIBS = $(GCPPLIB) $(GCLIB) -lgccpp
 LFLAGS = @LDFLAGS@ -LLspCpp -LLspCpp/third_party/uri/src
 LIBS = $(LFLAGS) @PTHREAD_LIBS@ @GLEW@ @LIBS@ $(GCLIBS) @LSPLIBS@
-DOSLIBS = $(subst -lncurses, -ltermcap, $(LIBS)) -lwolfssl -lgdi32 -lwinmm -s -static
+DOSLIBS = $(subst -lncurses, -ltermcap, $(LIBS)) -lgdi32 -lwinmm -s -static
 
 CMAKE ?= cmake
 PERL ?= perl
@@ -149,7 +149,7 @@
 	echo @set Datadir @datadir@ >> doc/version.texi
 	echo @set Docdir @docdir@ >> doc/version.texi
 
-	echo "#!/usr/bin/env python3" > GUI/xasyVersion.py
+	echo "#!@@PYTHON@@" > GUI/xasyVersion.py
 	echo xasyVersion = \"$(revision)\" >> GUI/xasyVersion.py
 
 	if test -n "$(MSDOS)"; then \
