https://gna.org/bugs/?16696
--- configure.ac.orig	2010-08-08 10:53:09.000000000 -0500
+++ configure.ac	2010-09-14 15:16:43.000000000 -0500
@@ -114,14 +114,8 @@
 	CXXFLAGS="-Werror -Wno-unused-parameter -Wno-strict-aliasing $CXXFLAGS"
 fi
 
-# Make tests default in svn version
-svn_in_version=`expr match "$WESNOTH_VERSION" '.*svn'`
-test_build=yes
-if test $svn_in_version = 0 
-then
 	#disabling building tests for release version
 	test_build=no
-fi
 
 AC_ARG_ENABLE([tests],
        	      AS_HELP_STRING([--enable-tests], [build unit tests]),
