--- m4/boost.m4.orig	2008-03-24 12:43:24.000000000 -0500
+++ m4/boost.m4	2010-09-13 10:32:57.000000000 -0500
@@ -89,7 +89,6 @@
 && B_V_MIN == $boost_version_minor \
 && B_V_SUB < $boost_version_subminor)"
     for boost_inc in "$with_boost/include" '' \
-             /opt/local/include /usr/local/include /opt/include /usr/include \
              "$with_boost" C:/Boost/include
     do
       test -e "$boost_inc" || continue
@@ -269,7 +268,6 @@
     boost_tmp_lib=$with_boost
     test x"$with_boost" = x && boost_tmp_lib=${boost_cv_inc_path%/include}
     for boost_ldpath in "$boost_tmp_lib/lib" '' \
-             /opt/local/lib /usr/local/lib /opt/lib /usr/lib \
              "$with_boost" C:/Boost/lib /lib /usr/lib64 /lib64
     do
       test -e "$boost_ldpath" || continue
