--- src/desktop/apple_notification.mm.orig
+++ src/desktop/apple_notification.mm
@@ -46,12 +46,14 @@
 void send_cocoa_notification(const std::string& owner, const std::string& message) {
     NSString *title = [NSString stringWithCString:owner.c_str() encoding:NSUTF8StringEncoding];
     NSString *description = [NSString stringWithCString:message.c_str() encoding:NSUTF8StringEncoding];
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1080
     NSUserNotification *notification = [[NSUserNotification alloc] init];
     notification.title = title;
     notification.informativeText = description;
     notification.deliveryDate = [NSDate date];
 
     [[NSUserNotificationCenter defaultUserNotificationCenter] scheduleNotification:notification];
+#endif
 }
 
 }
