--- ffi-openmpt-0.5.0.gemspec
+++ ffi-openmpt-0.5.0.gemspec	2024-04-15 12:59:24
@@ -25,7 +25,7 @@
   end
 
   if s.respond_to? :add_runtime_dependency then
-    s.add_runtime_dependency(%q<ffi>.freeze, ["~> 1.9"])
+    s.add_runtime_dependency(%q<ffi>.freeze, [">= 1.9"])
     s.add_development_dependency(%q<bundler>.freeze, ["~> 2.0.1"])
     s.add_development_dependency(%q<coveralls>.freeze, ["~> 0.8"])
     s.add_development_dependency(%q<minitest>.freeze, ["~> 5.0"])
@@ -33,7 +33,7 @@
     s.add_development_dependency(%q<rubocop>.freeze, ["= 0.67.2"])
     s.add_development_dependency(%q<rubocop-performance>.freeze, ["= 1.1.0"])
   else
-    s.add_dependency(%q<ffi>.freeze, ["~> 1.9"])
+    s.add_dependency(%q<ffi>.freeze, [">= 1.9"])
     s.add_dependency(%q<bundler>.freeze, ["~> 2.0.1"])
     s.add_dependency(%q<coveralls>.freeze, ["~> 0.8"])
     s.add_dependency(%q<minitest>.freeze, ["~> 5.0"])
