--- util/sdest.f.orig	2000-08-06 04:28:17.000000000 +0900
+++ util/sdest.f	2020-11-21 16:29:45.000000000 +0900
@@ -349,7 +349,7 @@
 C                                                                       
       IMPLICIT INTEGER (A-Z)                                            
       DIMENSION STR(*)
-      DATA ZERO/'30'X/
+      DATA ZERO/X'30'/
 C                                                                       
 C                                                                       
       IF (SIZE.LE.0.OR.POS.LT.0) GOTO 9                                 
