--- util/maketopo.c.orig	2000-10-28 11:56:04.000000000 +0900
+++ util/maketopo.c	2020-11-21 17:15:23.000000000 +0900
@@ -71,6 +71,8 @@
 
 
 
+#include <stdlib.h>
+#include <string.h>
 #include <stdio.h>
 #ifdef HAVE_SYS_TYPES_H
 #  include <sys/types.h>
@@ -97,12 +99,14 @@
 #define FILENAME "sample.topo"   /* use the .topo suffix for consistency */
 
 
+int write_topo( char* );
+
 float Topo[ROWS][COLUMNS];
 char  Water[ROWS][COLUMNS];
 
 
 
-main()
+int main()
 {
    int r, c;
 
@@ -184,7 +188,7 @@
 
 
 /** write the topofile and return 1 for success, 0 for error **/
-write_topo( filename )
+int write_topo( filename )
 char filename[];
 {
    int f;
