--- src/ls-hpack/lshpack.c
+++ src/ls-hpack/lshpack.c
@@ -691,6 +691,11 @@ lshpack_enc_enc_int (unsigned char *dst, unsigned char *const end,
 #endif
 
 
+#if __GNUC__ && !defined(__COVERITY__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#pragma GCC diagnostic ignored "-Wuninitialized"
+#endif
 static int
 lshpack_enc_huff_encode (const unsigned char *src,
     const unsigned char *const src_end, unsigned char *const dst, int dst_len)
@@ -701,9 +706,6 @@ lshpack_enc_huff_encode (const unsigned char *src,
     unsigned bits_used = 0, adj;
     struct encode_el cur_enc_code;
 #if __GNUC__ && !defined(__COVERITY__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
-#pragma GCC diagnostic ignored "-Wuninitialized"
 #else
     bits = 0;
 #endif
@@ -817,10 +819,10 @@ lshpack_enc_huff_encode (const unsigned char *src,
         return p_dst - dst;
     else
         return -1;
-#if __GNUC__
+}
+#if __GNUC__ && !defined(__COVERITY__)
 #pragma GCC diagnostic pop
 #endif
-}
 
 
 #if !LS_HPACK_EMIT_TEST_CODE
@@ -1853,6 +1855,11 @@ need_more_buf:
 #define SHORTEST_CODE 5
 
 
+#if __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#pragma GCC diagnostic ignored "-Wuninitialized"
+#endif
 /* The decoder is optimized for the common case.  Most of the time, we decode
  * data whose encoding is 16 bits or shorter.  This lets us use a 64 KB table
  * indexed by two bytes of input and outputs 1, 2, or 3 bytes at a time.
@@ -1874,10 +1881,6 @@ lshpack_dec_huff_decode (const unsigned char *src, int src_len,
     int r;
 
 #if __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
-#pragma GCC diagnostic ignored "-Wuninitialized"
-#else
     buf = 0;
 #endif
 
@@ -2038,9 +2041,6 @@ lshpack_dec_huff_decode (const unsigned char *src, int src_len,
         if (((1u << avail_bits) - 1) != (buf & ((1u << avail_bits) - 1)))
             return -1;  /* Not EOF as expected */
     }
-#if __GNUC__
-#pragma GCC diagnostic pop
-#endif
 
   end:
     return dst - orig_dst;
@@ -2057,7 +2057,11 @@ lshpack_dec_huff_decode (const unsigned char *src, int src_len,
     else
         return r;
 }
+#if __GNUC__
+#pragma GCC diagnostic pop
+#endif
 #endif
+
 #if __GNUC__
 #pragma GCC diagnostic pop  /* -Wunknown-pragmas */
 #endif
