--- doc/config/conf.d/auth.conf.orig	2020-12-17 03:11:11.000000000 -0600
+++ doc/config/conf.d/auth.conf	2020-12-18 01:16:22.000000000 -0600
@@ -11,7 +11,7 @@
 
 #server.modules += ( "mod_authn_file" )
 #auth.backend                 = "plain"
-#auth.backend.plain.userfile  = "/etc/lighttpd/lighttpd.user"
+#auth.backend.plain.userfile  = conf_dir + "/lighttpd.user"
 
 #server.modules += ( "mod_authn_ldap" )
 #auth.backend               = "ldap"

--- doc/config/conf.d/cgi.conf.orig	2010-07-11 12:01:32.000000000 -0500
+++ doc/config/conf.d/cgi.conf	2010-10-17 05:19:33.000000000 -0500
@@ -12,11 +12,11 @@
 ##
 ## For PHP don't forget to set cgi.fix_pathinfo = 1 in the php.ini.
 ##
-cgi.assign                 = ( ".pl"  => "/usr/bin/perl",
-                               ".cgi" => "/usr/bin/perl",
-                               ".rb"  => "/usr/bin/ruby",
-                               ".erb" => "/usr/bin/eruby",
-                               ".py"  => "/usr/bin/python" )
+cgi.assign                 = ( ".pl"  => "@PREFIX@/bin/perl",
+                               ".cgi" => "@PREFIX@/bin/perl",
+                               ".rb"  => "@PREFIX@/bin/ruby",
+                               ".erb" => "@PREFIX@/bin/eruby",
+                               ".py"  => "@PREFIX@/bin/python2.6" )
 
 ##
 ## to get the old cgi-bin behavior of apache
--- doc/config/conf.d/fastcgi.conf.orig	2017-11-11 10:30:25.000000000 -0600
+++ doc/config/conf.d/fastcgi.conf	2017-11-12 09:42:40.000000000 -0600
@@ -23,7 +23,7 @@
 #                   ( "php-local" =>
 #                     (
 #                       "socket" => socket_dir + "/php-fastcgi-1.socket",
-#                       "bin-path" => server_root + "/cgi-bin/php5",
+#                       "bin-path" => "@PREFIX@/bin/php-cgi",
 #                       "max-procs" => 1,
 #                       "broken-scriptfilename" => "enable",
 #                     ),
@@ -37,7 +37,7 @@
 #                     "php-num-procs" =>
 #                     (
 #                       "socket" => socket_dir + "/php-fastcgi-2.socket",
-#                       "bin-path" => server_root + "/cgi-bin/php5",
+#                       "bin-path" => "@PREFIX@/bin/php",
 #                       "bin-environment" => (
 #                         "PHP_FCGI_CHILDREN" => "16",
 #                         "PHP_FCGI_MAX_REQUESTS" => "10000",
@@ -112,7 +112,7 @@
 
 ## chrooted webserver + external PHP
 ##
-## $ spawn-fcgi -f /usr/bin/php-cgi -p 2000 -a 127.0.0.1 -C 8
+## $ spawn-fcgi -f @PREFIX@/bin/php-cgi -p 2000 -a 127.0.0.1 -C 8
 ##
 ## webserver chrooted to /srv/www/
 ## php running outside the chroot
@@ -125,7 +125,7 @@
 #  )))
 #
 #server.chroot = "/srv/www"
-#server.document-root = "/servers/wwww.example.org/htdocs/"
+#server.document-root = "/servers/www.example.org/htdocs/"
 #
 
 ##
--- doc/config/conf.d/mysql_vhost.conf.orig	2010-07-11 12:01:32.000000000 -0500
+++ doc/config/conf.d/mysql_vhost.conf	2010-10-17 05:14:56.000000000 -0500
@@ -12,7 +12,7 @@
 ##
 ## Local path to the mysql socket
 ##
-#mysql-vhost.sock           = "/var/lib/mysql/mysql.sock"
+mysql-vhost.sock           = "@PREFIX@/var/run/mysql5/mysqld.sock"
 
 ##
 ## Host of the MySQL server. 
--- doc/config/conf.d/rrdtool.conf.orig	2010-07-11 12:01:32.000000000 -0500
+++ doc/config/conf.d/rrdtool.conf	2010-10-17 05:07:35.000000000 -0500
@@ -10,12 +10,12 @@
 ##
 ## Path to the rrdtool binary.
 ##
-rrdtool.binary             = "/usr/bin/rrdtool"
+rrdtool.binary             = "@PREFIX@/bin/rrdtool"
 
 ##
 ## Path to the rrdtool database. You can override this in conditionals.
 ##
-rrdtool.db-name            = "/var/lib/lighttpd/lighttpd.rrd"
+rrdtool.db-name            = home_dir + "/lighttpd.rrd"
 
 ##
 #######################################################################
--- doc/config/lighttpd.conf.orig	2020-12-17 03:11:11.000000000 -0600
+++ doc/config/lighttpd.conf	2020-12-18 01:14:46.000000000 -0600
@@ -13,11 +13,11 @@
 ## if you add a variable here. Add the corresponding variable in the
 ## chroot example as well.
 ##
-var.log_root    = "/var/log/lighttpd"
-var.server_root = "/srv/www"
-var.state_dir   = "/run"
-var.home_dir    = "/var/lib/lighttpd"
-var.conf_dir    = "/etc/lighttpd"
+var.log_root    = "@PREFIX@/var/log/lighttpd"
+var.server_root = "@PREFIX@/www"
+var.state_dir   = "@PREFIX@/var/run/lighttpd"
+var.home_dir    = "@PREFIX@/var/lib/lighttpd"
+var.conf_dir    = "@PREFIX@/etc/lighttpd"
 
 ## 
 ## run the server chrooted.
@@ -58,7 +58,7 @@
 ## used in:
 ## conf.d/deflate.conf
 ##
-var.cache_dir   = "/var/cache/lighttpd"
+var.cache_dir   = "@PREFIX@/var/cache/lighttpd"
 
 ##
 ## Base directory for sockets.
@@ -385,7 +385,7 @@
 ## Format: <errorfile-prefix><status-code>.html
 ## -> ..../status-404.html for 'File not found'
 ##
-#server.errorfile-prefix    = "/srv/www/htdocs/errors/status-"
+#server.errorfile-prefix    = server.document-root + /errors/status-"
 
 ##
 ## mimetype mapping
