--- doc/lighttpd.conf.orig	2009-04-09 15:02:00.000000000 -0500
+++ doc/lighttpd.conf	2009-09-23 19:20:27.000000000 -0500
@@ -36,17 +36,17 @@
 
 ## A static document-root. For virtual hosting take a look at the
 ## mod_simple_vhost module.
-server.document-root        = "/srv/www/htdocs/"
+server.document-root        = "__PREFIX/www/htdocs/"
 
 ## where to send error-messages to
-server.errorlog             = "/var/log/lighttpd/error.log"
+server.errorlog             = "__PREFIX/var/log/lighttpd/error.log"
 
 # files to check for if .../ is requested
 index-file.names            = ( "index.php", "index.html",
                                 "index.htm", "default.htm" )
 
 ## set the event-handler (read the performance section in the manual)
-# server.event-handler = "freebsd-kqueue" # needed on OS X
+server.event-handler = "freebsd-kqueue" # needed on OS X
 
 # mimetype mapping
 mimetype.assign             = (
@@ -115,7 +115,7 @@
 # server.tag                 = "lighttpd"
 
 #### accesslog module
-accesslog.filename          = "/var/log/lighttpd/access.log"
+accesslog.filename          = "__PREFIX/var/log/lighttpd/access.log"
 
 ## deny access the file-extensions
 #
@@ -147,7 +147,7 @@
 #server.error-handler-404   = "/error-handler.php"
 
 ## to help the rc.scripts
-#server.pid-file            = "/var/run/lighttpd.pid"
+server.pid-file            = "__PREFIX/var/run/lighttpd.pid"
 
 
 ###### virtual hosts
@@ -160,7 +160,7 @@
 ## or
 ##   virtual-server-root + http-host + virtual-server-docroot
 ##
-#simple-vhost.server-root   = "/srv/www/vhosts/"
+#simple-vhost.server-root   = "__PREFIX/www/vhosts/"
 #simple-vhost.default-host  = "www.example.org"
 #simple-vhost.document-root = "/htdocs/"
 
@@ -168,8 +168,8 @@
 ##
 ## Format: <errorfile-prefix><status-code>.html
 ## -> ..../status-404.html for 'File not found'
-#server.errorfile-prefix    = "/usr/share/lighttpd/errors/status-"
-#server.errorfile-prefix    = "/srv/www/errors/status-"
+#server.errorfile-prefix    = "__PREFIX/share/lighttpd/errors/status-"
+#server.errorfile-prefix    = "__PREFIX/www/errors/status-"
 
 ## virtual directory listings
 #dir-listing.activate       = "enable"
@@ -188,13 +188,13 @@
 #server.chroot              = "/"
 
 ## change uid to <uid> (default: don't care)
-#server.username            = "wwwrun"
+server.username            = "www"
 
 ## change uid to <uid> (default: don't care)
-#server.groupname           = "wwwrun"
+server.groupname           = "www"
 
 #### compress module
-#compress.cache-dir         = "/var/cache/lighttpd/compress/"
+#compress.cache-dir         = "__PREFIX/var/cache/lighttpd/compress/"
 #compress.filetype          = ("text/plain", "text/html")
 
 #### proxy module
@@ -214,20 +214,20 @@
 #fastcgi.server             = ( ".php" =>
 #                               ( "localhost" =>
 #                                 (
-#                                   "socket" => "/var/run/lighttpd/php-fastcgi.socket",
-#                                   "bin-path" => "/usr/local/bin/php-cgi"
+#                                   "socket" => "__PREFIX/var/run/lighttpd/php-fastcgi.socket",
+#                                   "bin-path" => "__PREFIX/bin/php-cgi"
 #                                 )
 #                               )
 #                            )
 
 #### CGI module
-#cgi.assign                 = ( ".pl"  => "/usr/bin/perl",
-#                               ".cgi" => "/usr/bin/perl" )
+#cgi.assign                 = ( ".pl"  => "__PREFIX/bin/perl",
+#                               ".cgi" => "__PREFIX/bin/perl" )
 #
 
 #### SSL engine
 #ssl.engine                 = "enable"
-#ssl.pemfile                = "/etc/ssl/private/lighttpd.pem"
+#ssl.pemfile                = "__PREFIX/etc/ssl/private/lighttpd.pem"
 
 #### status module
 #status.status-url          = "/server-status"
@@ -274,7 +274,7 @@
 # %3 => subdomain 1 name
 # %4 => subdomain 2 name
 #
-#evhost.path-pattern        = "/srv/www/vhosts/%3/htdocs/"
+#evhost.path-pattern        = "__PREFIX/www/vhosts/%3/htdocs/"
 
 #### expire module
 #expire.url                 = ( "/buggy/" => "access 2 hours", "/asdhas/" => "access plus 1 seconds 2 minutes")
@@ -283,15 +283,15 @@
 #ssi.extension              = ( ".shtml" )
 
 #### rrdtool
-#rrdtool.binary             = "/usr/bin/rrdtool"
-#rrdtool.db-name            = "/var/lib/lighttpd/lighttpd.rrd"
+#rrdtool.binary             = "__PREFIX/bin/rrdtool"
+#rrdtool.db-name            = "__PREFIX/var/lib/lighttpd/lighttpd.rrd"
 
 #### setenv
 #setenv.add-request-header  = ( "TRAV_ENV" => "mysql://user@host/db" )
 #setenv.add-response-header = ( "X-Secret-Message" => "42" )
 
 ## for mod_trigger_b4_dl
-# trigger-before-download.gdbm-filename = "/var/lib/lighttpd/trigger.db"
+# trigger-before-download.gdbm-filename = "__PREFIX/var/lib/lighttpd/trigger.db"
 # trigger-before-download.memcache-hosts = ( "127.0.0.1:11211" )
 # trigger-before-download.trigger-url = "^/trigger/"
 # trigger-before-download.download-url = "^/download/"
@@ -312,8 +312,8 @@
 #index-file.names += (foo + ".php")
 
 #### include
-#include /etc/lighttpd/lighttpd-inc.conf
-## same as above if you run: "lighttpd -f /etc/lighttpd/lighttpd.conf"
+#include __PREFIX/etc/lighttpd/lighttpd-inc.conf
+## same as above if you run: "lighttpd -f __PREFIX/etc/lighttpd/lighttpd.conf"
 #include "lighttpd-inc.conf"
 
 #### include_shell
