--- doc/lighttpd.conf.orig	2009-08-17 15:34:32.000000000 -0400
+++ doc/lighttpd.conf	2009-08-17 15:34:49.000000000 -0400
@@ -37,17 +37,17 @@
 
 ## a static document-root, for virtual-hosting take look at the
 ## server.virtual-* options
-server.document-root        = "/srv/www/htdocs/"
+server.document-root        = "__PREFIX/www/htdocs/"
 
 ## where to send error-messages to
-server.errorlog             = "/var/log/lighttpd/error.log"
+server.errorlog             = "__PREFIX/var/log/lighttpd/error.log"
 
 # files to check for if .../ is requested
 index-file.names            = ( "index.php", "index.html",
                                 "index.htm", "default.htm" )
 
 ## set the event-handler (read the performance section in the manual)
-# server.event-handler = "freebsd-kqueue" # needed on OS X
+server.event-handler = "freebsd-kqueue" # needed on OS X
 
 # mimetype mapping
 mimetype.assign             = (
@@ -116,7 +116,7 @@
 # server.tag                 = "lighttpd"
 
 #### accesslog module
-accesslog.filename          = "/var/log/lighttpd/access.log"
+accesslog.filename          = "__PREFIX/var/log/lighttpd/access.log"
 
 ## deny access the file-extensions
 #
@@ -148,7 +148,7 @@
 #server.error-handler-404   = "/error-handler.php"
 
 ## to help the rc.scripts
-#server.pid-file            = "/var/run/lighttpd.pid"
+server.pid-file            = "__PREFIX/var/run/lighttpd.pid"
 
 
 ###### virtual hosts
@@ -189,13 +189,13 @@
 #server.chroot              = "/"
 
 ## change uid to <uid> (default: don't care)
-#server.username            = "wwwrun"
+server.username            = "www"
 
 ## change uid to <uid> (default: don't care)
-#server.groupname           = "wwwrun"
+server.groupname           = "www"
 
 #### compress module
-#compress.cache-dir         = "/var/cache/lighttpd/compress/"
+#compress.cache-dir         = "__PREFIX/var/cache/lighttpd/compress/"
 #compress.filetype          = ("text/plain", "text/html")
 
 #### proxy module
@@ -215,15 +215,15 @@
 #fastcgi.server             = ( ".php" =>
 #                               ( "localhost" =>
 #                                 (
-#                                   "socket" => "/var/run/lighttpd/php-fastcgi.socket",
-#                                   "bin-path" => "/usr/local/bin/php-cgi"
+#                                   "socket" => "__PREFIX/var/run/lighttpd/php-fastcgi.socket",
+#                                   "bin-path" => "__PREFIX/bin/php-cgi"
 #                                 )
 #                               )
 #                            )
 
 #### CGI module
-#cgi.assign                 = ( ".pl"  => "/usr/bin/perl",
-#                               ".cgi" => "/usr/bin/perl" )
+#cgi.assign                 = ( ".pl"  => "__PREFIX/bin/perl",
+#                               ".cgi" => "__PREFIX/bin/perl" )
 #
 
 #### SSL engine
@@ -284,15 +284,15 @@
 #ssi.extension              = ( ".shtml" )
 
 #### rrdtool
-#rrdtool.binary             = "/usr/bin/rrdtool"
-#rrdtool.db-name            = "/var/lib/lighttpd/lighttpd.rrd"
+#rrdtool.binary             = "__PREFIX/bin/rrdtool"
+#rrdtool.db-name            = "__PREFIX/var/lib/lighttpd/lighttpd.rrd"
 
 #### setenv
 #setenv.add-request-header  = ( "TRAV_ENV" => "mysql://user@host/db" )
 #setenv.add-response-header = ( "X-Secret-Message" => "42" )
 
 ## for mod_trigger_b4_dl
-# trigger-before-download.gdbm-filename = "/var/lib/lighttpd/trigger.db"
+# trigger-before-download.gdbm-filename = "__PREFIX/var/lib/lighttpd/trigger.db"
 # trigger-before-download.memcache-hosts = ( "127.0.0.1:11211" )
 # trigger-before-download.trigger-url = "^/trigger/"
 # trigger-before-download.download-url = "^/download/"
