--- abcde.orig	2010-05-29 04:02:44.000000000 -0500
+++ abcde	2011-03-11 16:47:41.000000000 -0600
@@ -1730,7 +1730,7 @@
 		vecho -n "Getting CD track info... "
 		# In OSX, unmount the disc before a query
 		if [ "$OSFLAVOUR" = "OSX" ]; then
-			disktool -u ${CDROM#/dev/}
+			diskutil unmount ${CDROM#/dev/}
 		fi
 		case "$CDROMREADERSYNTAX" in
 			flac)
@@ -1803,7 +1803,7 @@
 		fi
 		# In OSX, remount the disc again
 		if [ "$OSFLAVOUR" = "OSX" ]; then
-			disktool -m ${CDROM#/dev/}
+			diskutil mount ${CDROM#/dev/}
 		fi
 		WEHAVEACD=y
 		DISCID=$(echo $TRACKINFO | cut -f1 -d' ')
@@ -2815,7 +2815,7 @@
 		cdda2wav | icedax)
 			if [ "$OSFLAVOUR" = "OSX" ] ; then
 				# Hei, we have to unmount the device before running anything like cdda2wav/icedax in OSX
-				disktool -u ${CDROM#/dev/} 0
+				diskutil unmount ${CDROM#/dev/} 0
 				# Also, in OSX the cdrom device for cdda2wav/icedax changes...
 				CDDA2WAVCDROM="IODVDServices"
 			elif [ "$OSFLAVOUR" = "FBSD" ] ; then
@@ -3152,8 +3152,8 @@
 elif [ X$(uname) = "XDarwin" ] ; then
 	HTTPGET=curl
 	OSFLAVOUR=OSX
-	# We should have disktool in OSX, but let's be sure...
-	NEEDDISKTOOL=y
+	# We should have diskutil in OSX, but let's be sure...
+	NEEDDISKUTIL=y
 	CDROMREADERSYNTAX=cddafs
 elif [ X$(uname) = "XOpenBSD" ] ; then
 	HTTPGET=wget
@@ -3840,7 +3840,7 @@
 	${NEEDHTTPGET+$HTTPGET} ${NEEDDISTMP3+$DISTMP3} \
 	${NEEDCOMMENTER+$VORBISCOMMENT} ${NEEDMETAFLAC+$METAFLAC} \
 	${NEEDNORMALIZER+$NORMALIZER} ${NEEDEJECT+$EJECT} \
-	${NEEDDISKTOOL+disktool} ${NEEDCDSPEED+$CDSPEED} \
+	${NEEDDISKUTIL+diskutil} ${NEEDCDSPEED+$CDSPEED} \
 	${NEEDVORBISGAIN+$VORBISGAIN} ${NEEDMP3GAIN+$MP3GAIN} \
 	${NEEDMPPGAIN+$MPPGAIN} ${NEEDCUEREADER+$CUEREADER} \
 	${NEEDCUE2DISCID+$CUE2DISCID}
@@ -4067,7 +4067,7 @@
 		cd="$(echo $CDROM | sed -e 's=.*/==;s=[a-h]$==;')"
 		$eject $EJECTOPTS $cd
 	elif [ X"$(uname)" = X"Darwin" ] ; then
-		disktool -e ${CDROM#/dev/} 0
+		diskutil eject ${CDROM#/dev/} 0
 	else
 		$EJECT $EJECTOPTS "$CDROM"
 	fi
