--- Makefile	2016-04-06 04:47:56.000000000 -0400
+++ Makefile	2016-11-13 10:17:29.000000000 -0500
@@ -1,12 +1,11 @@
 abcde_version = abcde-2.7.2
 INSTALL = /usr/bin/install -c
 
-prefix = /usr/local
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
-sysconfdir = /etc
+sysconfdir = $(prefix)/etc
 datarootdir = $(prefix)/share
-docdir = $(datarootdir)/doc/$(abcde_version)
+docdir = $(datarootdir)/doc/abcde
 mandir = $(datarootdir)/man
 DESTDIR =
 
