--- cmake/Modules.FindLua.cmake.orig	2017-12-08 11:49:52.000000000 -0800
+++ cmake/Modules/FindLua.cmake	2017-12-08 11:50:05.000000000 -0800
@@ -11,6 +11,7 @@
 	find_package(LuaJIT)
 	if(LUAJIT_FOUND)
 		set(USE_LUAJIT TRUE)
+		include_directories(BEFORE ${LUA_INCLUDE_DIR})
 		message (STATUS "Using LuaJIT provided by system.")
 	elseif(REQUIRE_LUAJIT)
 		message(FATAL_ERROR "LuaJIT not found whereas REQUIRE_LUAJIT=\"TRUE\" is used.\n"
