--- plugins/slideshowshuffle/slideshowshuffle.py.orig	2015-07-18 01:31:58.000000000 -0700
+++ plugins/slideshowshuffle/slideshowshuffle.py	2015-07-18 01:34:15.000000000 -0700
@@ -14,14 +14,18 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import gi
 
+gi.require_version('Gtk', '3.0')
+gi.require_version('Eog', '3.0')
 from gi.repository import GObject, Gtk, Eog
+
 import random
 
 class SlideshowShufflePlugin(GObject.Object, Eog.WindowActivatable):
 
     # Override EogWindowActivatable's window property
-    window = GObject.property(type=Eog.Window)
+    window = GObject.Property(type=Eog.Window)
 
     def __init__(self):
         GObject.Object.__init__(self)
