From 4d024133af07ba738542e23917c31d7248b6ef2a Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date: Fri, 28 Mar 2014 10:26:24 +0100
Subject: Switch slideshowshuffle to use the python3 loader.

https://bugzilla.gnome.org/show_bug.cgi?id=727222

diff --git a/plugins/slideshowshuffle/slideshowshuffle.plugin.desktop.in b/plugins/slideshowshuffle/slideshowshuffle.plugin.desktop.in
index 986de8b..ca53c60 100644
--- plugins/slideshowshuffle/slideshowshuffle.plugin.desktop.in
+++ plugins/slideshowshuffle/slideshowshuffle.plugin.desktop.in
@@ -1,5 +1,5 @@
 [Plugin]
-Loader=python
+Loader=python3
 Module=slideshowshuffle
 IAge=2
 _Name=Slideshow Shuffle
-- 
cgit v0.10.1
