--- plugins/pythonconsole/__init__.py.orig	2015-03-23 15:21:26.000000000 -0700
+++ plugins/pythonconsole/__init__.py	2015-09-28 13:38:11.000000000 -0700
@@ -25,6 +25,11 @@
 #     Copyright (C), 2005 Adam Hooper <adamh@densi.com>
 #     Copyrignt (C), 2005 Raphaël Slinckx
 
+import gi
+
+gi.require_version('Gtk', '3.0')
+gi.require_version('Eog', '3.0')
+gi.require_version('PeasGtk', '1.0')
 from gi.repository import GObject, Gio, GLib, Gtk, Eog, PeasGtk
 
 from .console import PythonConsole
@@ -41,7 +46,7 @@
                           PeasGtk.Configurable):
 
     # Override EogWindowActivatable's window property
-    window = GObject.property(type=Eog.Window)
+    window = GObject.Property(type=Eog.Window)
     action_group = None
 
     def __init__(self):
